// @backbase/consent-journey-ang/esm2015/src/views/consent-request/consent-request.component.js
import { Component } from '@angular/core';
import { Subject } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "../../services/consent-journey-config.service";
import * as i2 from "@angular/router";
import * as i3 from "@backbase/consent-request-widget-ang";
import * as i4 from "@backbase/ui-ang/button";
export class ConsentRequestComponent {
    constructor(config, router, activeRoute) {
        this.config = config;
        this.router = router;
        this.activeRoute = activeRoute;
        this.destroy$ = new Subject();
    }
    requestContext() {
        this.router.navigate([], {
            queryParams: {
                consentRequestId: '1231313',
            },
            skipLocationChange: false,
            relativeTo: this.activeRoute,
        });
    }
    handleAllowRequest(event) {
        this.router.navigate(['../consent-redirect', { allowConsentRequestRedirectUrl: event }], {
            relativeTo: this.activeRoute,
        });
    }
    handleRejectRequest(event) {
        this.router.navigate(['../consent-redirect', { rejectConsentRequestRedirectUrl: event }], {
            relativeTo: this.activeRoute,
        });
    }
    ngOnDestroy() {
        this.destroy$.next();
    }
}
ConsentRequestComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentRequestComponent, deps: [{ token: i1.ConsentJourneyConfigurationService }, { token: i2.Router }, { token: i2.ActivatedRoute }], target: i0.ɵɵFactoryTarget.Component });
ConsentRequestComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: ConsentRequestComponent, selector: "bb-consent-request-view", ngImport: i0, template: `
    <div class="container--fixed-width-small mx-auto">
      <div class="bb-card">
        <div class="bb-card__body">
          <div class="mb-5 d-flex justify-content-center">
            <button
              bbButton
              (click)="requestContext()"
              i18n="Simulate Consent Request@@consent-journey.label.simulate-consent-request">
              Simulate Consent Request
            </button>
          </div>
          <bb-consent-request-widget
            [multiAccountSelection]="config.multiAccountSelection"
            (allowConsentRequestRedirectUrl)="handleAllowRequest($event)"
            (rejectConsentRequestRedirectUrl)="handleRejectRequest($event)"></bb-consent-request-widget>
        </div>
      </div>
    </div>
  `, isInline: true, components: [{ type: i3.ConsentRequestWidgetComponent, selector: "bb-consent-request-widget", inputs: ["multiAccountSelection"], outputs: ["allowConsentRequestRedirectUrl", "rejectConsentRequestRedirectUrl"] }], directives: [{ type: i4.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentRequestComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-consent-request-view',
                    template: `
    <div class="container--fixed-width-small mx-auto">
      <div class="bb-card">
        <div class="bb-card__body">
          <div class="mb-5 d-flex justify-content-center">
            <button
              bbButton
              (click)="requestContext()"
              i18n="Simulate Consent Request@@consent-journey.label.simulate-consent-request">
              Simulate Consent Request
            </button>
          </div>
          <bb-consent-request-widget
            [multiAccountSelection]="config.multiAccountSelection"
            (allowConsentRequestRedirectUrl)="handleAllowRequest($event)"
            (rejectConsentRequestRedirectUrl)="handleRejectRequest($event)"></bb-consent-request-widget>
        </div>
      </div>
    </div>
  `,
                }]
        }], ctorParameters: function () { return [{ type: i1.ConsentJourneyConfigurationService }, { type: i2.Router }, { type: i2.ActivatedRoute }]; } });
