// @backbase/payment-orders-ang/esm2015/lib/ui/components/dynamic-label/dynamic-label.component.js
import { Component, Input } from '@angular/core';
import { getLabelFromPaymentConfig } from '../../helpers/payment-config-helpers';
import * as i0 from "@angular/core";
export class PayordDynamicLabelComponent {
    constructor() {
        /**
         * Fallback label name
         */
        this.defaultLabel = '';
        /**
         * Parent group name of the form field
         */
        this.parentName = '';
        this.label = '';
    }
    ngOnInit() {
        this.label = getLabelFromPaymentConfig(this.name, this.config, this.parentName);
    }
}
/** @nocollapse */ PayordDynamicLabelComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordDynamicLabelComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ PayordDynamicLabelComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: PayordDynamicLabelComponent, selector: "bb-payord-dynamic-label", inputs: { config: "config", name: "name", defaultLabel: "defaultLabel", parentName: "parentName" }, ngImport: i0, template: ` <ng-container>{{ label || defaultLabel }}</ng-container> `, isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordDynamicLabelComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-payord-dynamic-label',
                    template: ` <ng-container>{{ label || defaultLabel }}</ng-container> `,
                }]
        }], propDecorators: { config: [{
                type: Input
            }], name: [{
                type: Input
            }], defaultLabel: [{
                type: Input
            }], parentName: [{
                type: Input
            }] } });
