// @backbase/self-enrollment-http-ang/bundles/backbase-self-enrollment-http-ang.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common/http'), require('@backbase/foundation-ang/data-http')) :
    typeof define === 'function' && define.amd ? define('@backbase/self-enrollment-http-ang', ['exports', '@angular/core', '@angular/common/http', '@backbase/foundation-ang/data-http'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["self-enrollment-http-ang"] = {}), global.ng.core, global.ng.common.http, global.i2));
})(this, (function (exports, i0, i1, i2) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);

    var SELF_ENROLLMENT_BASE_PATH = new i0.InjectionToken('SELF_ENROLLMENT_BASE_PATH');
    var COLLECTION_FORMATS = {
        'csv': ',',
        'tsv': '   ',
        'ssv': ' ',
        'pipes': '|'
    };

    var SelfEnrollmentConfiguration = /** @class */ (function () {
        function SelfEnrollmentConfiguration(configurationParameters) {
            if (configurationParameters === void 0) { configurationParameters = {}; }
            this.apiKeys = configurationParameters.apiKeys;
            this.username = configurationParameters.username;
            this.password = configurationParameters.password;
            this.accessToken = configurationParameters.accessToken;
            this.basePath = configurationParameters.basePath;
            this.withCredentials = configurationParameters.withCredentials;
            this.encoder = configurationParameters.encoder;
            if (configurationParameters.credentials) {
                this.credentials = configurationParameters.credentials;
            }
            else {
                this.credentials = {};
            }
        }
        SelfEnrollmentConfiguration.prototype.selectHeaderContentType = function (contentTypes) {
            var _this = this;
            if (contentTypes.length === 0) {
                return undefined;
            }
            var type = contentTypes.find(function (x) { return _this.isJsonMime(x); });
            if (type === undefined) {
                return contentTypes[0];
            }
            return type;
        };
        SelfEnrollmentConfiguration.prototype.selectHeaderAccept = function (accepts) {
            var _this = this;
            if (accepts.length === 0) {
                return undefined;
            }
            var type = accepts.find(function (x) { return _this.isJsonMime(x); });
            if (type === undefined) {
                return accepts[0];
            }
            return type;
        };
        SelfEnrollmentConfiguration.prototype.isJsonMime = function (mime) {
            var jsonMime = new RegExp('^(application\/json|[^;/ \t]+\/[^;/ \t]+[+]json)[ \t]*(;.*)?$', 'i');
            return mime !== null && (jsonMime.test(mime) || mime.toLowerCase() === 'application/json-patch+json');
        };
        SelfEnrollmentConfiguration.prototype.lookupCredential = function (key) {
            var value = this.credentials[key];
            return typeof value === 'function'
                ? value()
                : value;
        };
        return SelfEnrollmentConfiguration;
    }());
    var CONFIG_TOKEN = new i0.InjectionToken('api-data-module SelfEnrollmentConfiguration');

    var SelfEnrollmentApiModule = /** @class */ (function () {
        function SelfEnrollmentApiModule(parentModule, http, dataModulesManager, config) {
            if (parentModule) {
                throw new Error('SelfEnrollmentApiModule is already loaded. Import in your base AppModule only.');
            }
            if (!http) {
                throw new Error('You need to import the HttpClientModule in your AppModule! \n' +
                    'See also https://github.com/angular/angular/issues/20575');
            }
            if (dataModulesManager) {
                dataModulesManager.setModuleConfig(CONFIG_TOKEN, {
                    apiRoot: '',
                    servicePath: config.basePath || '',
                    headers: {},
                });
            }
        }
        SelfEnrollmentApiModule.forRoot = function (configurationFactory) {
            return {
                ngModule: SelfEnrollmentApiModule,
                providers: [{ provide: SelfEnrollmentConfiguration, useFactory: configurationFactory }]
            };
        };
        return SelfEnrollmentApiModule;
    }());
    SelfEnrollmentApiModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: SelfEnrollmentApiModule, deps: [{ token: SelfEnrollmentApiModule, optional: true, skipSelf: true }, { token: i1__namespace.HttpClient, optional: true }, { token: i2__namespace.DataModulesManager, optional: true }, { token: SelfEnrollmentConfiguration }], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    SelfEnrollmentApiModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: SelfEnrollmentApiModule });
    SelfEnrollmentApiModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: SelfEnrollmentApiModule, providers: [], imports: [[]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: SelfEnrollmentApiModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [],
                        declarations: [],
                        exports: [],
                        providers: []
                    }]
            }], ctorParameters: function () {
            return [{ type: SelfEnrollmentApiModule, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.SkipSelf
                        }] }, { type: i1__namespace.HttpClient, decorators: [{
                            type: i0.Optional
                        }] }, { type: i2__namespace.DataModulesManager, decorators: [{
                            type: i0.Optional
                        }] }, { type: SelfEnrollmentConfiguration }];
        } });

    var CustomHttpParameterCodec = /** @class */ (function () {
        function CustomHttpParameterCodec() {
        }
        CustomHttpParameterCodec.prototype.encodeKey = function (k) {
            return encodeURIComponent(k);
        };
        CustomHttpParameterCodec.prototype.encodeValue = function (v) {
            return encodeURIComponent(v);
        };
        CustomHttpParameterCodec.prototype.decodeKey = function (k) {
            return decodeURIComponent(k);
        };
        CustomHttpParameterCodec.prototype.decodeValue = function (v) {
            return decodeURIComponent(v);
        };
        return CustomHttpParameterCodec;
    }());

    var SelfEnrollmentService = /** @class */ (function () {
        function SelfEnrollmentService(httpClient, basePath, configuration) {
            this.httpClient = httpClient;
            this.basePath = '';
            this.defaultHeaders = new i1.HttpHeaders();
            this.configuration = new SelfEnrollmentConfiguration();
            if (configuration) {
                this.configuration = configuration;
            }
            if (typeof this.configuration.basePath !== 'string') {
                if (typeof basePath !== 'string') {
                    basePath = this.basePath;
                }
                this.configuration.basePath = basePath;
            }
            this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
        }
        SelfEnrollmentService.prototype.addToHttpParams = function (httpParams, value, key) {
            if (typeof value === "object" && value instanceof Date === false) {
                httpParams = this.addToHttpParamsRecursive(httpParams, value);
            }
            else {
                httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
            }
            return httpParams;
        };
        SelfEnrollmentService.prototype.addToHttpParamsRecursive = function (httpParams, value, key) {
            var _this = this;
            if (value == null) {
                return httpParams;
            }
            if (typeof value === "object") {
                if (Array.isArray(value)) {
                    value.forEach(function (elem) { return httpParams = _this.addToHttpParamsRecursive(httpParams, elem, key); });
                }
                else if (value instanceof Date) {
                    if (key != null) {
                        httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                    }
                    else {
                        throw Error("key may not be null if value is Date");
                    }
                }
                else {
                    Object.keys(value).forEach(function (k) { return httpParams = _this.addToHttpParamsRecursive(httpParams, value[k], key != null ? key + "." + k : k); });
                }
            }
            else if (key != null) {
                httpParams = httpParams.append(key, value);
            }
            else {
                throw Error("key may not be null if value is not object or array");
            }
            return httpParams;
        };
        SelfEnrollmentService.prototype.getStatus = function (observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.get(this.configuration.basePath + "/client-api/v2/status", {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        SelfEnrollmentService.prototype.getStatusUrl = function () {
            return this.configuration.basePath + "/client-api/v2/status";
        };
        SelfEnrollmentService.prototype.getUsernameByUsername = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _username = requestParameters["username"];
            if (_username === null || _username === undefined) {
                throw new Error('Required parameter username was null or undefined when calling getUsernameByUsername.');
            }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.get(this.configuration.basePath + "/client-api/v2/credentials/existing/username/" + encodeURIComponent(String(_username)), {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        SelfEnrollmentService.prototype.getUsernameByUsernameUrl = function (requestParameters) {
            var _username = requestParameters["username"];
            if (_username === null || _username === undefined) {
                throw new Error('Required parameter username was null or undefined when calling getUsernameByUsername.');
            }
            return this.configuration.basePath + "/client-api/v2/credentials/existing/username/" + encodeURIComponent(String(_username));
        };
        SelfEnrollmentService.prototype.getVerifications = function (observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.get(this.configuration.basePath + "/client-api/v2/verifications", {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        SelfEnrollmentService.prototype.getVerificationsUrl = function () {
            return this.configuration.basePath + "/client-api/v2/verifications";
        };
        SelfEnrollmentService.prototype.postCredentials = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _createCredentials = requestParameters["createCredentials"];
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var consumes = [
                'application/json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected !== undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.post(this.configuration.basePath + "/client-api/v2/credentials", _createCredentials, {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        SelfEnrollmentService.prototype.postCredentialsUrl = function (requestParameters) {
            var _createCredentials = requestParameters["createCredentials"];
            return this.configuration.basePath + "/client-api/v2/credentials";
        };
        SelfEnrollmentService.prototype.postEnrollments = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _createEnrollment = requestParameters["createEnrollment"];
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var consumes = [
                'application/json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected !== undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.post(this.configuration.basePath + "/client-api/v2/enrollments", _createEnrollment, {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        SelfEnrollmentService.prototype.postEnrollmentsUrl = function (requestParameters) {
            var _createEnrollment = requestParameters["createEnrollment"];
            return this.configuration.basePath + "/client-api/v2/enrollments";
        };
        SelfEnrollmentService.prototype.postVerifications = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _verificationRequestItem = requestParameters["verificationRequestItem"];
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var consumes = [
                'application/json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected !== undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.post(this.configuration.basePath + "/client-api/v2/verifications", _verificationRequestItem, {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        SelfEnrollmentService.prototype.postVerificationsUrl = function (requestParameters) {
            var _verificationRequestItem = requestParameters["verificationRequestItem"];
            return this.configuration.basePath + "/client-api/v2/verifications";
        };
        return SelfEnrollmentService;
    }());
    SelfEnrollmentService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: SelfEnrollmentService, deps: [{ token: i1__namespace.HttpClient }, { token: SELF_ENROLLMENT_BASE_PATH, optional: true }, { token: SelfEnrollmentConfiguration, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    SelfEnrollmentService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: SelfEnrollmentService, providedIn: 'root' });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: SelfEnrollmentService, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root'
                    }]
            }], ctorParameters: function () {
            return [{ type: i1__namespace.HttpClient }, { type: undefined, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.Inject,
                            args: [SELF_ENROLLMENT_BASE_PATH]
                        }] }, { type: SelfEnrollmentConfiguration, decorators: [{
                            type: i0.Optional
                        }] }];
        } });

    var SelfEnrollmentServiceGetStatusMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v2/status",
            method: "GET",
            responses: [
                {
                    status: 200,
                    body: {
                        "status": "IDENTIFIED"
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
            ]
        }]);
    var SelfEnrollmentServiceGetUsernameByUsernameMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v2/credentials/existing/username/{username}",
            method: "GET",
            responses: [
                {
                    status: 204,
                    body: ""
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
            ]
        }]);
    var SelfEnrollmentServiceGetVerificationsMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v2/verifications",
            method: "GET",
            responses: [
                {
                    status: 200,
                    body: [{
                            "question": {
                                "id": "12ghr44356-54mpkm65",
                                "text": "When did you open your Loan account?"
                            },
                            "answers": [{
                                    "id": "wrghr44356-54mpkm65",
                                    "text": "January 2017"
                                }, {
                                    "id": "ds-0h-903285hb6",
                                    "text": "March 2017"
                                }, {
                                    "id": "34bkbdfs89sdfnj",
                                    "text": "July 2017"
                                }, {
                                    "id": "123-3456-nkbjhbdf",
                                    "text": "None of the above"
                                }]
                        }, {
                            "question": {
                                "id": "12ghr44356-54mpkerwfvd5",
                                "text": "What state was your SSN issued in?"
                            },
                            "answers": [{
                                    "id": "123kbbb43500fds",
                                    "text": "New York"
                                }, {
                                    "id": "453-65njb453-45657",
                                    "text": "Virginia"
                                }, {
                                    "id": "csgdf-35436njgf",
                                    "text": "Missouri"
                                }, {
                                    "id": "aaa-23454363jb634",
                                    "text": "None of the above"
                                }]
                        }, {
                            "question": {
                                "id": "18hgdd8756-54botdf8876",
                                "text": "In 2017, your Car Loan was approved by which bank?"
                            },
                            "answers": [{
                                    "id": "1mlfjur-i094s",
                                    "text": "Bank of America"
                                }, {
                                    "id": "473-65iiktnd93-433457",
                                    "text": "Wells Fargo"
                                }, {
                                    "id": "cujhf-356657ytjgf",
                                    "text": "Chase"
                                }, {
                                    "id": "abb-236678435jb634",
                                    "text": "None of the above"
                                }]
                        }, {
                            "question": {
                                "id": "109jjhbft6-590-ouhgj345",
                                "text": "Which of the following cities has a current or former association to you?"
                            },
                            "answers": [{
                                    "id": "19874yhrgsvdr5",
                                    "text": "Houston"
                                }, {
                                    "id": "4577-8576ijuth-45659",
                                    "text": "Los Angeles"
                                }, {
                                    "id": "cascadf-3554r66t-uu",
                                    "text": "Chicago"
                                }, {
                                    "id": "aga-264547789-jb634",
                                    "text": "None of the above"
                                }]
                        }]
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
            ]
        }]);
    var SelfEnrollmentServicePostCredentialsMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v2/credentials",
            method: "POST",
            responses: [
                {
                    status: 204,
                    body: ""
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
            ]
        }]);
    var SelfEnrollmentServicePostEnrollmentsMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v2/enrollments",
            method: "POST",
            responses: [
                {
                    status: 204,
                    body: ""
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
            ]
        }]);
    var SelfEnrollmentServicePostVerificationsMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v2/verifications",
            method: "POST",
            responses: [
                {
                    status: 204,
                    body: ""
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
            ]
        }]);
    var SelfEnrollmentServiceMocksProvider = i2.createMocks([
        {
            urlPattern: "/client-api/v2/status",
            method: "GET",
            responses: [
                {
                    status: 200,
                    body: {
                        "status": "IDENTIFIED"
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
            ]
        },
        {
            urlPattern: "/client-api/v2/credentials/existing/username/{username}",
            method: "GET",
            responses: [
                {
                    status: 204,
                    body: ""
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
            ]
        },
        {
            urlPattern: "/client-api/v2/verifications",
            method: "GET",
            responses: [
                {
                    status: 200,
                    body: [{
                            "question": {
                                "id": "12ghr44356-54mpkm65",
                                "text": "When did you open your Loan account?"
                            },
                            "answers": [{
                                    "id": "wrghr44356-54mpkm65",
                                    "text": "January 2017"
                                }, {
                                    "id": "ds-0h-903285hb6",
                                    "text": "March 2017"
                                }, {
                                    "id": "34bkbdfs89sdfnj",
                                    "text": "July 2017"
                                }, {
                                    "id": "123-3456-nkbjhbdf",
                                    "text": "None of the above"
                                }]
                        }, {
                            "question": {
                                "id": "12ghr44356-54mpkerwfvd5",
                                "text": "What state was your SSN issued in?"
                            },
                            "answers": [{
                                    "id": "123kbbb43500fds",
                                    "text": "New York"
                                }, {
                                    "id": "453-65njb453-45657",
                                    "text": "Virginia"
                                }, {
                                    "id": "csgdf-35436njgf",
                                    "text": "Missouri"
                                }, {
                                    "id": "aaa-23454363jb634",
                                    "text": "None of the above"
                                }]
                        }, {
                            "question": {
                                "id": "18hgdd8756-54botdf8876",
                                "text": "In 2017, your Car Loan was approved by which bank?"
                            },
                            "answers": [{
                                    "id": "1mlfjur-i094s",
                                    "text": "Bank of America"
                                }, {
                                    "id": "473-65iiktnd93-433457",
                                    "text": "Wells Fargo"
                                }, {
                                    "id": "cujhf-356657ytjgf",
                                    "text": "Chase"
                                }, {
                                    "id": "abb-236678435jb634",
                                    "text": "None of the above"
                                }]
                        }, {
                            "question": {
                                "id": "109jjhbft6-590-ouhgj345",
                                "text": "Which of the following cities has a current or former association to you?"
                            },
                            "answers": [{
                                    "id": "19874yhrgsvdr5",
                                    "text": "Houston"
                                }, {
                                    "id": "4577-8576ijuth-45659",
                                    "text": "Los Angeles"
                                }, {
                                    "id": "cascadf-3554r66t-uu",
                                    "text": "Chicago"
                                }, {
                                    "id": "aga-264547789-jb634",
                                    "text": "None of the above"
                                }]
                        }]
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
            ]
        },
        {
            urlPattern: "/client-api/v2/credentials",
            method: "POST",
            responses: [
                {
                    status: 204,
                    body: ""
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
            ]
        },
        {
            urlPattern: "/client-api/v2/enrollments",
            method: "POST",
            responses: [
                {
                    status: 204,
                    body: ""
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
            ]
        },
        {
            urlPattern: "/client-api/v2/verifications",
            method: "POST",
            responses: [
                {
                    status: 204,
                    body: ""
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
            ]
        },
    ]);

    exports.COLLECTION_FORMATS = COLLECTION_FORMATS;
    exports.CONFIG_TOKEN = CONFIG_TOKEN;
    exports.SELF_ENROLLMENT_BASE_PATH = SELF_ENROLLMENT_BASE_PATH;
    exports.SelfEnrollmentApiModule = SelfEnrollmentApiModule;
    exports.SelfEnrollmentConfiguration = SelfEnrollmentConfiguration;
    exports.SelfEnrollmentService = SelfEnrollmentService;
    exports.SelfEnrollmentServiceGetStatusMocksProvider = SelfEnrollmentServiceGetStatusMocksProvider;
    exports.SelfEnrollmentServiceGetUsernameByUsernameMocksProvider = SelfEnrollmentServiceGetUsernameByUsernameMocksProvider;
    exports.SelfEnrollmentServiceGetVerificationsMocksProvider = SelfEnrollmentServiceGetVerificationsMocksProvider;
    exports.SelfEnrollmentServiceMocksProvider = SelfEnrollmentServiceMocksProvider;
    exports.SelfEnrollmentServicePostCredentialsMocksProvider = SelfEnrollmentServicePostCredentialsMocksProvider;
    exports.SelfEnrollmentServicePostEnrollmentsMocksProvider = SelfEnrollmentServicePostEnrollmentsMocksProvider;
    exports.SelfEnrollmentServicePostVerificationsMocksProvider = SelfEnrollmentServicePostVerificationsMocksProvider;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
