// @backbase/contact-common-ang/esm2015/lib/components/request-list-base/request-list-base.component.js
import { Directive, EventEmitter, Output } from '@angular/core';
import { ApprovalSort } from '@backbase/data-ang/contact-manager';
import * as i0 from "@angular/core";
/**
 * Abstract Base Request List Component
 *
 */
export class RequestListBaseComponent {
    constructor() {
        /**
         * Direction of current sorting
         */
        this.sortDirection = 'asc';
        /**
         * Enum with names of sorting columns
         */
        this.approvalSortEnum = ApprovalSort;
        /**
         * Event emitted when column is clicked
         */
        this.sort = new EventEmitter();
    }
    /**
     * Sort column method
     *
     * @param {string} event - object with column and direction properites
     */
    onSort(event) {
        this.sortColumn = event.direction ? event.column : null;
        this.sortDirection = event.direction || 'desc';
        this.sort.emit(event);
    }
}
/** @nocollapse */ RequestListBaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RequestListBaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Directive });
/** @nocollapse */ RequestListBaseComponent.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: RequestListBaseComponent, outputs: { sort: "sort" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RequestListBaseComponent, decorators: [{
            type: Directive
        }], propDecorators: { sort: [{
                type: Output
            }] } });
