// @backbase/payment-orders-ang/esm2015/lib/ui/components/confirmation-dialog/confirmation-dialog.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ConfirmationDialogComponent } from './confirmation-dialog.component';
import { ModalModule } from '@backbase/ui-ang/modal';
import { TextareaModule } from '@backbase/ui-ang/textarea';
import { HeaderModule } from '@backbase/ui-ang/header';
import { ButtonModule } from '@backbase/ui-ang/button';
import { FormsModule } from '@angular/forms';
import * as i0 from "@angular/core";
export class ConfirmationDialogModule {
}
/** @nocollapse */ ConfirmationDialogModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ConfirmationDialogModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ ConfirmationDialogModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ConfirmationDialogModule, declarations: [ConfirmationDialogComponent], imports: [CommonModule,
        HeaderModule,
        ModalModule,
        TextareaModule,
        ButtonModule,
        FormsModule], exports: [ConfirmationDialogComponent] });
/** @nocollapse */ ConfirmationDialogModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ConfirmationDialogModule, imports: [[
            CommonModule,
            HeaderModule,
            ModalModule,
            TextareaModule,
            ButtonModule,
            FormsModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ConfirmationDialogModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ConfirmationDialogComponent],
                    imports: [
                        CommonModule,
                        HeaderModule,
                        ModalModule,
                        TextareaModule,
                        ButtonModule,
                        FormsModule,
                    ],
                    exports: [ConfirmationDialogComponent],
                }]
        }] });
