// @backbase/ui-ang/esm2015/currency-input/currency-input.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { CurrencyInputComponent } from './currency-input.component';
import { DropdownSingleSelectModule } from '@backbase/ui-ang/dropdown-single-select';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import * as i0 from "@angular/core";
export class CurrencyInputModule {
}
CurrencyInputModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CurrencyInputModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CurrencyInputModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CurrencyInputModule, declarations: [CurrencyInputComponent], imports: [CommonModule, ReactiveFormsModule, DropdownSingleSelectModule], exports: [CurrencyInputComponent,
        // exporting validation message as it can be injected into the component template
        InputValidationMessageModule,
        /**
         * @deprecated
         * Exporting `FormsModule` is deprecated
         **/
        FormsModule] });
CurrencyInputModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CurrencyInputModule, imports: [[CommonModule, ReactiveFormsModule, DropdownSingleSelectModule], 
        // exporting validation message as it can be injected into the component template
        InputValidationMessageModule,
        /**
         * @deprecated
         * Exporting `FormsModule` is deprecated
         **/
        FormsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CurrencyInputModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ReactiveFormsModule, DropdownSingleSelectModule],
                    declarations: [CurrencyInputComponent],
                    exports: [
                        CurrencyInputComponent,
                        // exporting validation message as it can be injected into the component template
                        InputValidationMessageModule,
                        /**
                         * @deprecated
                         * Exporting `FormsModule` is deprecated
                         **/
                        FormsModule,
                    ],
                }]
        }] });
