// @backbase/messages-client-inbox-journey-ang/bundles/backbase-messages-client-inbox-journey-ang.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('@angular/localize/init'), require('@backbase/foundation-ang/web-sdk'), require('@angular/router'), require('@backbase/messages-create-message-modal-widget-ang'), require('@backbase/messages-ui-ang'), require('@backbase/foundation-ang/core'), require('rxjs'), require('@backbase/messages-conversations-list-widget-ang'), require('@backbase/messages-conversation-thread-widget-ang'), require('rxjs/operators')) :
    typeof define === 'function' && define.amd ? define('@backbase/messages-client-inbox-journey-ang', ['exports', '@angular/core', '@angular/common', '@angular/localize/init', '@backbase/foundation-ang/web-sdk', '@angular/router', '@backbase/messages-create-message-modal-widget-ang', '@backbase/messages-ui-ang', '@backbase/foundation-ang/core', 'rxjs', '@backbase/messages-conversations-list-widget-ang', '@backbase/messages-conversation-thread-widget-ang', 'rxjs/operators'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["messages-client-inbox-journey-ang"] = {}), global.ng.core, global.ng.common, global.ng.localize.init, global.webSdk, global.ng.router, global.i3, global.i4, global.i1, global.rxjs, global.i3$1, global.i3$2, global.rxjs.operators));
})(this, (function (exports, i0, i4$1, init, webSdk, i2, i3, i4, i1, rxjs, i3$1, i3$2, operators) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i4__namespace$1 = /*#__PURE__*/_interopNamespace(i4$1);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i3__namespace = /*#__PURE__*/_interopNamespace(i3);
    var i4__namespace = /*#__PURE__*/_interopNamespace(i4);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i3__namespace$1 = /*#__PURE__*/_interopNamespace(i3$1);
    var i3__namespace$2 = /*#__PURE__*/_interopNamespace(i3$2);

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from, pack) {
        if (pack || arguments.length === 2)
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar)
                        ar = Array.prototype.slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
        return to.concat(ar || Array.prototype.slice.call(from));
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    var MessagesClientInboxJourneyConfigurationToken = new i0.InjectionToken('MessagesClientInboxJourneyConfiguration injection token');
    var configDefaults = {
        headingTitle: $localize(templateObject_1$2 || (templateObject_1$2 = __makeTemplateObject([":Client Inbox Journey - Heading title@@messages-client-inbox-journey.heading-title:Messages"], [":Client Inbox Journey - Heading title@@messages-client-inbox-journey.heading-title:Messages"]))),
        headingType: 'h1',
        headingWrapperClasses: '',
        // eslint-disable-next-line max-len
        buttonText: $localize(templateObject_2$1 || (templateObject_2$1 = __makeTemplateObject([":Client Inbox Journey - Compose button text@@messages-client-inbox-journey.compose-button-text:Compose"], [":Client Inbox Journey - Compose button text@@messages-client-inbox-journey.compose-button-text:Compose"]))),
        createMessageOpenEventName: 'bb.event.messages.create.message.open',
        createMessageClosedEventName: 'bb.event.messages.create.message.close',
        itemsPerPage: 10,
        // eslint-disable-next-line max-len
        customerServiceTitle: $localize(templateObject_3$1 || (templateObject_3$1 = __makeTemplateObject([":Client Inbox Journey - Customer Service title@@messages-client-inbox-journey.customer-service-title:Customer Service"], [":Client Inbox Journey - Customer Service title@@messages-client-inbox-journey.customer-service-title:Customer Service"]))),
        hideAssignedToColumn: false,
        maxAttachmentSize: '10',
        replyMessageMaxLength: '300',
        maxSubjectLength: 100,
        maxMessageLength: 300,
        // eslint-disable-next-line max-len
        modalHeader: $localize(templateObject_4 || (templateObject_4 = __makeTemplateObject([":Client Inbox Journey - Create new message modal header@@messages-client-inbox-journey.create-new-message-modal-header:New message"], [":Client Inbox Journey - Create new message modal header@@messages-client-inbox-journey.create-new-message-modal-header:New message"]))),
        hideComposeButton: true,
    };
    var MessagesClientInboxJourneyConfigurationProvider = {
        provide: MessagesClientInboxJourneyConfigurationToken,
        useValue: configDefaults,
    };
    var MessagesClientInboxJourneyConfigService = /** @class */ (function () {
        function MessagesClientInboxJourneyConfigService(staticConfig) {
            this._config = Object.assign(Object.assign({}, configDefaults), staticConfig);
        }
        Object.defineProperty(MessagesClientInboxJourneyConfigService.prototype, "headingTitle", {
            get: function () {
                return this._config.headingTitle;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(MessagesClientInboxJourneyConfigService.prototype, "headingType", {
            get: function () {
                return this._config.headingType;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(MessagesClientInboxJourneyConfigService.prototype, "headingWrapperClasses", {
            get: function () {
                return this._config.headingWrapperClasses;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(MessagesClientInboxJourneyConfigService.prototype, "buttonText", {
            get: function () {
                return this._config.buttonText;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(MessagesClientInboxJourneyConfigService.prototype, "createMessageOpenEventName", {
            get: function () {
                return this._config.createMessageOpenEventName;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(MessagesClientInboxJourneyConfigService.prototype, "createMessageClosedEventName", {
            get: function () {
                return this._config.createMessageClosedEventName;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(MessagesClientInboxJourneyConfigService.prototype, "itemsPerPage", {
            get: function () {
                return this._config.itemsPerPage;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(MessagesClientInboxJourneyConfigService.prototype, "customerServiceTitle", {
            get: function () {
                return this._config.customerServiceTitle;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(MessagesClientInboxJourneyConfigService.prototype, "hideAssignedToColumn", {
            get: function () {
                return this._config.hideAssignedToColumn;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(MessagesClientInboxJourneyConfigService.prototype, "maxAttachmentSize", {
            get: function () {
                return this._config.maxAttachmentSize;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(MessagesClientInboxJourneyConfigService.prototype, "replyMessageMaxLength", {
            get: function () {
                return this._config.replyMessageMaxLength;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(MessagesClientInboxJourneyConfigService.prototype, "maxSubjectLength", {
            get: function () {
                return this._config.maxSubjectLength;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(MessagesClientInboxJourneyConfigService.prototype, "maxMessageLength", {
            get: function () {
                return this._config.maxMessageLength;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(MessagesClientInboxJourneyConfigService.prototype, "modalHeader", {
            get: function () {
                return this._config.modalHeader;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(MessagesClientInboxJourneyConfigService.prototype, "hideComposeButton", {
            get: function () {
                return this._config.hideComposeButton;
            },
            enumerable: false,
            configurable: true
        });
        return MessagesClientInboxJourneyConfigService;
    }());
    MessagesClientInboxJourneyConfigService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: MessagesClientInboxJourneyConfigService, deps: [{ token: MessagesClientInboxJourneyConfigurationToken, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    MessagesClientInboxJourneyConfigService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: MessagesClientInboxJourneyConfigService });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: MessagesClientInboxJourneyConfigService, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () {
            return [{ type: undefined, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.Inject,
                            args: [MessagesClientInboxJourneyConfigurationToken]
                        }] }];
        } });
    var templateObject_1$2, templateObject_2$1, templateObject_3$1, templateObject_4;

    var MessagesClientInboxJourneyComponent = /** @class */ (function () {
        function MessagesClientInboxJourneyComponent(config, eventBus, route) {
            this.config = config;
            this.eventBus = eventBus;
            this.route = route;
            /**
             * Collection of Tabs to be rendered according to the journey layout and routing.
             */
            this.tabs = [];
            var routeConfig = this.route.snapshot.routeConfig;
            if (routeConfig && routeConfig.children) {
                this.tabs = (routeConfig === null || routeConfig === void 0 ? void 0 : routeConfig.children) ? this.initializeTabs(routeConfig) : [];
            }
        }
        MessagesClientInboxJourneyComponent.prototype.initializeTabs = function (_b) {
            var _c = _b.children, children = _c === void 0 ? [] : _c;
            return children
                .filter(function (routeItem) { return routeItem.path && routeItem.data; })
                .map(function (routeItem, index) {
                var _a;
                return ({
                    title: ((_a = routeItem.data) === null || _a === void 0 ? void 0 : _a.title) || $localize(templateObject_1$1 || (templateObject_1$1 = __makeTemplateObject(["Default unnamed tab title@@messages-client-inbox-journey.default-tab.title:Tab", ""], ["Default unnamed tab title@@messages-client-inbox-journey.default-tab.title:Tab", ""])), index + 1),
                    route: routeItem.path || "" + (index + 1),
                });
            });
        };
        /**
         * Handler for the header button to open the upload modal
         */
        MessagesClientInboxJourneyComponent.prototype.openComposeModal = function () {
            this.eventBus.publish(this.config.createMessageOpenEventName, undefined);
        };
        return MessagesClientInboxJourneyComponent;
    }());
    MessagesClientInboxJourneyComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: MessagesClientInboxJourneyComponent, deps: [{ token: MessagesClientInboxJourneyConfigService }, { token: webSdk.PUBSUB }, { token: i2__namespace.ActivatedRoute }], target: i0__namespace.ɵɵFactoryTarget.Component });
    MessagesClientInboxJourneyComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.14", type: MessagesClientInboxJourneyComponent, selector: "bb-messages-client-inbox-journey", ngImport: i0__namespace, template: "<bb-messages-create-message-modal-widget-ang\n  [maxAttachmentSize]=\"config.maxAttachmentSize\"\n  [maxSubjectLength]=\"config.maxSubjectLength\"\n  [maxMessageLength]=\"config.maxMessageLength\"\n  [modalHeader]=\"config.modalHeader\"\n  [hideComposeButton]=\"config.hideComposeButton\"\n  [createMessageOpenEventName]=\"config.createMessageOpenEventName\"\n  [createMessageClosedEventName]=\"config.createMessageClosedEventName\"\n>\n</bb-messages-create-message-modal-widget-ang>\n<bb-messages-heading\n  [headingTitle]=\"config.headingTitle\"\n  [headingType]=\"config.headingType\"\n  [headingWrapperClasses]=\"config.headingWrapperClasses\"\n  [buttonText]=\"config.buttonText\"\n  (buttonAction)=\"openComposeModal()\"\n>\n</bb-messages-heading>\n<div class=\"bb-messages bb-messages__tabs-wrapper\">\n  <bb-messages-tabs data-role=\"client-inbox__tabs\" [tabs]=\"tabs\"></bb-messages-tabs>\n</div>\n", components: [{ type: i3__namespace.MessagesCreateMessageModalWidgetComponent, selector: "bb-messages-create-message-modal-widget-ang" }, { type: i4__namespace.MessagesHeadingComponent, selector: "bb-messages-heading", inputs: ["headingTitle", "headingType", "headingClasses", "headingWrapperClasses", "buttonText", "showButton"], outputs: ["buttonAction"] }, { type: i4__namespace.MessagesTabsComponent, selector: "bb-messages-tabs", inputs: ["tabs"] }], changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: MessagesClientInboxJourneyComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-messages-client-inbox-journey',
                        templateUrl: './messages-client-inbox-journey.component.html',
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], ctorParameters: function () {
            return [{ type: MessagesClientInboxJourneyConfigService }, { type: undefined, decorators: [{
                            type: i0.Inject,
                            args: [webSdk.PUBSUB]
                        }] }, { type: i2__namespace.ActivatedRoute }];
        } });
    var templateObject_1$1;

    var MessagesClientInboxListComponent = /** @class */ (function () {
        function MessagesClientInboxListComponent(config, route, router) {
            var _a;
            this.config = config;
            this.route = route;
            this.router = router;
            this.mailboxType$$ = new rxjs.BehaviorSubject(undefined);
            /**
             * Mailbox type title
             */
            this.mailboxType$ = this.mailboxType$$.asObservable();
            /**
             * List of shown tabs
             */
            this.tabs = [];
            var routeConfig = this.route.snapshot.routeConfig;
            this.mailboxType$$.next((_a = routeConfig === null || routeConfig === void 0 ? void 0 : routeConfig.data) === null || _a === void 0 ? void 0 : _a.mailboxType);
        }
        /**
         * Open conversation thread
         */
        MessagesClientInboxListComponent.prototype.openConversation = function (id) {
            this.router.navigate(['../conversation', { id: id }], { relativeTo: this.route });
        };
        return MessagesClientInboxListComponent;
    }());
    MessagesClientInboxListComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: MessagesClientInboxListComponent, deps: [{ token: MessagesClientInboxJourneyConfigService }, { token: i2__namespace.ActivatedRoute }, { token: i2__namespace.Router }], target: i0__namespace.ɵɵFactoryTarget.Component });
    MessagesClientInboxListComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.14", type: MessagesClientInboxListComponent, selector: "bb-messages-client-inbox-list", ngImport: i0__namespace, template: "<bb-messages-conversations-list-widget-ang\n  [mailboxType]=\"mailboxType$ | async\"\n  (conversationId)=\"openConversation($event)\"\n  [itemsPerPage]=\"config.itemsPerPage\"\n  [customerServiceTitle]=\"config.customerServiceTitle\"\n  [createMessageClosedEventName]=\"config.createMessageClosedEventName\"\n>\n</bb-messages-conversations-list-widget-ang>\n", components: [{ type: i3__namespace$1.MessagesConversationsListWidgetComponent, selector: "bb-messages-conversations-list-widget-ang", inputs: ["mailboxType", "itemsPerPage", "createMessageClosedEventName", "customerServiceTitle"] }], pipes: { "async": i4__namespace$1.AsyncPipe }, changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: MessagesClientInboxListComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-messages-client-inbox-list',
                        templateUrl: './messages-client-inbox-list.component.html',
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], ctorParameters: function () { return [{ type: MessagesClientInboxJourneyConfigService }, { type: i2__namespace.ActivatedRoute }, { type: i2__namespace.Router }]; } });

    var MessagesClientInboxThreadComponent = /** @class */ (function () {
        function MessagesClientInboxThreadComponent(config, route, router) {
            var _a;
            this.config = config;
            this.route = route;
            this.router = router;
            this.conversationId$$ = new rxjs.BehaviorSubject('');
            /**
             * Selected conversation id stream
             */
            this.conversationId$ = this.conversationId$$.pipe(operators.distinctUntilChanged());
            this.destroy$$ = new rxjs.Subject();
            var routeConfig = this.route.snapshot.routeConfig;
            this.mailboxTypeTitle = (_a = routeConfig === null || routeConfig === void 0 ? void 0 : routeConfig.data) === null || _a === void 0 ? void 0 : _a.mailboxTypeTitle;
        }
        MessagesClientInboxThreadComponent.prototype.ngOnInit = function () {
            var _this = this;
            this.route.params.pipe(operators.takeUntil(this.destroy$$)).subscribe(function (_b) {
                var id = _b.id;
                _this.conversationId$$.next(id || '');
            });
        };
        /**
         * Navigates back
         */
        MessagesClientInboxThreadComponent.prototype.navigateBack = function () {
            this.router.navigate(["../"], { relativeTo: this.route });
        };
        MessagesClientInboxThreadComponent.prototype.ngOnDestroy = function () {
            this.destroy$$.next();
        };
        return MessagesClientInboxThreadComponent;
    }());
    MessagesClientInboxThreadComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: MessagesClientInboxThreadComponent, deps: [{ token: MessagesClientInboxJourneyConfigService }, { token: i2__namespace.ActivatedRoute }, { token: i2__namespace.Router }], target: i0__namespace.ɵɵFactoryTarget.Component });
    MessagesClientInboxThreadComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.14", type: MessagesClientInboxThreadComponent, selector: "bb-messages-client-inbox-thread", ngImport: i0__namespace, template: "<bb-messages-conversation-thread-widget-ang\n  [selectedConversationId]=\"conversationId$ | async\"\n  [customerServiceTitle]=\"config.customerServiceTitle\"\n  [replyMessageMaxLength]=\"config.replyMessageMaxLength\"\n  [maxAttachmentSize]=\"config.maxAttachmentSize\"\n  [mailboxTypeTitle]=\"mailboxTypeTitle\"\n  (back)=\"navigateBack()\"\n></bb-messages-conversation-thread-widget-ang>\n", components: [{ type: i3__namespace$2.MessagesConversationThreadWidgetComponent, selector: "bb-messages-conversation-thread-widget-ang", inputs: ["mailboxTypeTitle", "maxAttachmentSize", "replyMessageMaxLength", "customerServiceTitle"] }], pipes: { "async": i4__namespace$1.AsyncPipe }, changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: MessagesClientInboxThreadComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-messages-client-inbox-thread',
                        templateUrl: './messages-client-inbox-thread.component.html',
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], ctorParameters: function () { return [{ type: MessagesClientInboxJourneyConfigService }, { type: i2__namespace.ActivatedRoute }, { type: i2__namespace.Router }]; } });

    var MessagesManageConversationsComponent = /** @class */ (function () {
        function MessagesManageConversationsComponent() {
        }
        return MessagesManageConversationsComponent;
    }());
    MessagesManageConversationsComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: MessagesManageConversationsComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    MessagesManageConversationsComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.14", type: MessagesManageConversationsComponent, selector: "bb-messages-manage-conversations-view", ngImport: i0__namespace, template: "<router-outlet></router-outlet>", isInline: true, directives: [{ type: i2__namespace.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: MessagesManageConversationsComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-messages-manage-conversations-view',
                        template: "<router-outlet></router-outlet>",
                    }]
            }] });

    var mailboxTypeTitles = {
        inbox: $localize(templateObject_1 || (templateObject_1 = __makeTemplateObject([":Client Inbox Journey - Inbox mailbox type@@messages-client-inbox-journey.inbox.mailbox-type:Inbox"], [":Client Inbox Journey - Inbox mailbox type@@messages-client-inbox-journey.inbox.mailbox-type:Inbox"]))),
        // eslint-disable-next-line max-len
        outbox: $localize(templateObject_2 || (templateObject_2 = __makeTemplateObject([":Client Inbox Journey - Outbox mailbox type@@messages-client-inbox-journey.outbox.mailbox-type:Outbox"], [":Client Inbox Journey - Outbox mailbox type@@messages-client-inbox-journey.outbox.mailbox-type:Outbox"]))),
        // eslint-disable-next-line max-len
        drafts: $localize(templateObject_3 || (templateObject_3 = __makeTemplateObject([":Client Inbox Journey - Drafts mailbox type@@messages-client-inbox-journey.drafts.mailbox-type:Drafts"], [":Client Inbox Journey - Drafts mailbox type@@messages-client-inbox-journey.drafts.mailbox-type:Drafts"]))),
    };
    var defaultRoute = {
        path: '',
        component: MessagesClientInboxJourneyComponent,
        children: [
            { path: '', redirectTo: 'inbox/list' },
            {
                path: 'inbox',
                data: { title: mailboxTypeTitles.inbox },
                component: MessagesManageConversationsComponent,
                children: [
                    { path: '', redirectTo: 'list', pathMatch: 'full' },
                    { path: 'list', component: MessagesClientInboxListComponent, data: { mailboxType: 'inbox' } },
                    {
                        path: 'conversation',
                        component: MessagesClientInboxThreadComponent,
                        data: { mailboxTypeTitle: mailboxTypeTitles.inbox },
                    },
                ],
            },
            {
                path: 'outbox',
                data: { title: mailboxTypeTitles.outbox },
                component: MessagesManageConversationsComponent,
                children: [
                    { path: '', redirectTo: 'list', pathMatch: 'full' },
                    { path: 'list', component: MessagesClientInboxListComponent, data: { mailboxType: 'outbox' } },
                    {
                        path: 'conversation',
                        component: MessagesClientInboxThreadComponent,
                        data: { mailboxTypeTitle: mailboxTypeTitles.outbox },
                    },
                ],
            },
            {
                path: 'drafts',
                data: { title: mailboxTypeTitles.drafts },
                component: MessagesManageConversationsComponent,
                children: [
                    { path: '', redirectTo: 'list', pathMatch: 'full' },
                    { path: 'list', component: MessagesClientInboxListComponent, data: { mailboxType: 'drafts' } },
                    {
                        path: 'conversation',
                        component: MessagesClientInboxThreadComponent,
                        data: { mailboxTypeTitle: mailboxTypeTitles.drafts },
                    },
                ],
            },
        ],
    };
    var templateObject_1, templateObject_2, templateObject_3;

    var components = [
        MessagesClientInboxJourneyComponent,
        MessagesClientInboxListComponent,
        MessagesClientInboxThreadComponent,
        MessagesManageConversationsComponent,
    ];
    var MessagesClientInboxJourneyModule = /** @class */ (function () {
        function MessagesClientInboxJourneyModule() {
        }
        MessagesClientInboxJourneyModule.forRoot = function (data) {
            if (data === void 0) { data = { routes: defaultRoute }; }
            return {
                ngModule: MessagesClientInboxJourneyModule,
                providers: [i2.provideRoutes([data.routes])],
            };
        };
        return MessagesClientInboxJourneyModule;
    }());
    MessagesClientInboxJourneyModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: MessagesClientInboxJourneyModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    MessagesClientInboxJourneyModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: MessagesClientInboxJourneyModule, declarations: [MessagesClientInboxJourneyComponent,
            MessagesClientInboxListComponent,
            MessagesClientInboxThreadComponent,
            MessagesManageConversationsComponent], imports: [i4$1.CommonModule,
            i2.RouterModule,
            i3.MessagesCreateMessageModalWidgetModule,
            i3$1.MessagesConversationsListWidgetModule,
            i3$2.MessagesConversationThreadWidgetModule,
            i4.MessagesCollectionUiAngModule, i1__namespace.BackbaseCoreModule] });
    MessagesClientInboxJourneyModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: MessagesClientInboxJourneyModule, providers: [MessagesClientInboxJourneyConfigService, i1.TemplateRegistry], imports: [[
                i4$1.CommonModule,
                i2.RouterModule,
                i3.MessagesCreateMessageModalWidgetModule,
                i3$1.MessagesConversationsListWidgetModule,
                i3$2.MessagesConversationThreadWidgetModule,
                i4.MessagesCollectionUiAngModule,
                i1.BackbaseCoreModule.withConfig({
                    classMap: {
                        MessagesClientInboxJourneyComponent: MessagesClientInboxJourneyComponent,
                    },
                }),
            ]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: MessagesClientInboxJourneyModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: __spreadArray([], __read(components)),
                        imports: [
                            i4$1.CommonModule,
                            i2.RouterModule,
                            i3.MessagesCreateMessageModalWidgetModule,
                            i3$1.MessagesConversationsListWidgetModule,
                            i3$2.MessagesConversationThreadWidgetModule,
                            i4.MessagesCollectionUiAngModule,
                            i1.BackbaseCoreModule.withConfig({
                                classMap: {
                                    MessagesClientInboxJourneyComponent: MessagesClientInboxJourneyComponent,
                                },
                            }),
                        ],
                        providers: [MessagesClientInboxJourneyConfigService, i1.TemplateRegistry],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.MessagesClientInboxJourneyComponent = MessagesClientInboxJourneyComponent;
    exports.MessagesClientInboxJourneyConfigService = MessagesClientInboxJourneyConfigService;
    exports.MessagesClientInboxJourneyConfigurationProvider = MessagesClientInboxJourneyConfigurationProvider;
    exports.MessagesClientInboxJourneyConfigurationToken = MessagesClientInboxJourneyConfigurationToken;
    exports.MessagesClientInboxJourneyModule = MessagesClientInboxJourneyModule;
    exports.MessagesClientInboxListComponent = MessagesClientInboxListComponent;
    exports.MessagesClientInboxThreadComponent = MessagesClientInboxThreadComponent;
    exports.MessagesManageConversationsComponent = MessagesManageConversationsComponent;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
