// @backbase/notifications-common-ang/bundles/backbase-notifications-common-ang.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('@backbase/foundation-ang/core'), require('@backbase/ui-ang/icon'), require('rxjs'), require('rxjs/operators')) :
    typeof define === 'function' && define.amd ? define('@backbase/notifications-common-ang', ['exports', '@angular/core', '@angular/common', '@backbase/foundation-ang/core', '@backbase/ui-ang/icon', 'rxjs', 'rxjs/operators'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["notifications-common-ang"] = {}), global.ng.core, global.ng.common, global.i1$1, global.i1, global.rxjs, global.rxjs.operators));
})(this, (function (exports, i0, i2, i1$1, i1, rxjs, operators) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i1__namespace$1 = /*#__PURE__*/_interopNamespace(i1$1);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from, pack) {
        if (pack || arguments.length === 2)
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar)
                        ar = Array.prototype.slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
        return to.concat(ar || Array.prototype.slice.call(from));
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    var NotificationLocalizationPipe = /** @class */ (function () {
        function NotificationLocalizationPipe(locale) {
            this.locale = locale;
        }
        NotificationLocalizationPipe.prototype.transform = function (notification, property) {
            return (((notification.translations || {})[this.locale] || notification)[property] || '');
        };
        return NotificationLocalizationPipe;
    }());
    /** @nocollapse */ NotificationLocalizationPipe.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: NotificationLocalizationPipe, deps: [{ token: i0.LOCALE_ID }], target: i0__namespace.ɵɵFactoryTarget.Pipe });
    /** @nocollapse */ NotificationLocalizationPipe.ɵpipe = i0__namespace.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: NotificationLocalizationPipe, name: "notificationLocalization" });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: NotificationLocalizationPipe, decorators: [{
                type: i0.Pipe,
                args: [{
                        name: 'notificationLocalization',
                    }]
            }], ctorParameters: function () {
            return [{ type: undefined, decorators: [{
                            type: i0.Inject,
                            args: [i0.LOCALE_ID]
                        }] }];
        } });

    var linkRegexp = /{{(.*)}}/;
    var InterpolateLinkPipe = /** @class */ (function () {
        function InterpolateLinkPipe() {
        }
        InterpolateLinkPipe.prototype.transform = function (template, value, focusableLink, classLink) {
            if (focusableLink === void 0) { focusableLink = true; }
            if (classLink === void 0) { classLink = ''; }
            return template.replace(linkRegexp, function (_, linkText) { return "<a" + (classLink && ' class="' + classLink + '"') + " href=\"" + value.link + "\" target=\"_blank\" rel=\"noopener noreferrer\" tabindex=\"" + (focusableLink ? '0' : '-1') + "\">" + (linkText || value.link) + "</a>"; });
        };
        return InterpolateLinkPipe;
    }());
    /** @nocollapse */ InterpolateLinkPipe.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: InterpolateLinkPipe, deps: [], target: i0__namespace.ɵɵFactoryTarget.Pipe });
    /** @nocollapse */ InterpolateLinkPipe.ɵpipe = i0__namespace.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: InterpolateLinkPipe, name: "interpolateLink" });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: InterpolateLinkPipe, decorators: [{
                type: i0.Pipe,
                args: [{
                        name: 'interpolateLink',
                    }]
            }] });

    exports.NotificationOrigins = void 0;
    (function (NotificationOrigins) {
        NotificationOrigins["Accounts"] = "Accounts";
        NotificationOrigins["Transactions"] = "Transaction";
        NotificationOrigins["Messages"] = "MessageCenter";
        NotificationOrigins["Payments"] = "Payment";
        NotificationOrigins["Limits"] = "Limits";
        NotificationOrigins["Contact"] = "Contact";
        NotificationOrigins["SavingGoals"] = "PFM";
        NotificationOrigins["AccountStatement"] = "AccountStatement";
        NotificationOrigins["FinancialInsights"] = "FinancialInsights";
    })(exports.NotificationOrigins || (exports.NotificationOrigins = {}));

    var _a;
    var NOTIFICAION_ORIGIN_ICONS = (_a = {},
        _a[exports.NotificationOrigins.Accounts] = 'current-accounts',
        _a[exports.NotificationOrigins.Transactions] = 'swap-horiz',
        _a[exports.NotificationOrigins.Messages] = 'mail-outline',
        _a[exports.NotificationOrigins.SavingGoals] = 'donut-large',
        _a[exports.NotificationOrigins.Contact] = 'contacts',
        _a[exports.NotificationOrigins.Payments] = 'payments',
        _a[exports.NotificationOrigins.AccountStatement] = 'description',
        _a[exports.NotificationOrigins.Limits] = 'how-to-reg',
        _a[exports.NotificationOrigins.FinancialInsights] = 'pie_chart_outline',
        _a);

    var NotificationsCapabilityIconComponent = /** @class */ (function () {
        function NotificationsCapabilityIconComponent() {
            /**
             * Notification origin types
             */
            this.originTypes = exports.NotificationOrigins;
            /**
             * Notification origin icons
             */
            this.originIcons = NOTIFICAION_ORIGIN_ICONS;
        }
        return NotificationsCapabilityIconComponent;
    }());
    /** @nocollapse */ NotificationsCapabilityIconComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: NotificationsCapabilityIconComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    /** @nocollapse */ NotificationsCapabilityIconComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: NotificationsCapabilityIconComponent, selector: "bb-notifications-capability-icon", inputs: { origin: "origin" }, ngImport: i0__namespace, template: "<ng-container\n  bbNotificationsCapabilityIconCustomizable\n  [bbHostRef]=\"this\"\n  [bbTemplateContext]=\"{\n    origin: origin\n  }\"\n></ng-container>\n\n<ng-template bbNotificationsCapabilityIconCustomizable let-origin=\"context.origin\" let-host>\n  <div *ngIf=\"host.originIcons[origin]\" class=\"bb-stack bb-stack--align-center bb-text-support\">\n    <bb-icon-ui\n      size=\"sm\"\n      class=\"bb-stack__item bb-stack__item--spacing-sm\"\n      [name]=\"host.originIcons[origin]\"\n      data-role=\"notifications-capability-icon\"\n    ></bb-icon-ui>\n    <span\n      class=\"bb-stack__item bb-subheader bb-subheader--regular\"\n      data-role=\"notifications-capability-icon-label\"\n      i18n=\"Notification origin label@@notifications-capability-icon.origin.label\"\n    >\n      { origin, select, Accounts {Accounts} Transaction {Transactions} MessageCenter {Messages} Limits {Limits} Payment\n      {Payments} AccountStatement {Account Statement} Contact {Contact} PFM {Saving Goals} FinancialInsights {Financial\n      Insights} }\n    </span>\n  </div>\n</ng-template>\n", components: [{ type: i0__namespace.forwardRef(function () { return i1__namespace.IconComponent; }), selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], directives: [{ type: i0__namespace.forwardRef(function () { return NotificationsCapabilityIconCustomizableDirective; }), selector: "[bbNotificationsCapabilityIconCustomizable]" }, { type: i0__namespace.forwardRef(function () { return i2__namespace.NgIf; }), selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: NotificationsCapabilityIconComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-notifications-capability-icon',
                        templateUrl: './notifications-capability-icon.component.html',
                    }]
            }], propDecorators: { origin: [{
                    type: i0.Input
                }] } });
    var NotificationsCapabilityIconCustomizableDirective = /** @class */ (function (_super) {
        __extends(NotificationsCapabilityIconCustomizableDirective, _super);
        function NotificationsCapabilityIconCustomizableDirective() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return NotificationsCapabilityIconCustomizableDirective;
    }(i1$1.BbTemplate));
    /** @nocollapse */ NotificationsCapabilityIconCustomizableDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: NotificationsCapabilityIconCustomizableDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    /** @nocollapse */ NotificationsCapabilityIconCustomizableDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: NotificationsCapabilityIconCustomizableDirective, selector: "[bbNotificationsCapabilityIconCustomizable]", usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: NotificationsCapabilityIconCustomizableDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: '[bbNotificationsCapabilityIconCustomizable]',
                    }]
            }] });

    /**
     * Check if element is truncated with overflow:ellipsis
     */
    function isElementContentTruncated(_a) {
        var offsetHeight = _a.offsetHeight, scrollHeight = _a.scrollHeight;
        return offsetHeight < scrollHeight;
    }

    var IsElementContentTruncatedDirective = /** @class */ (function () {
        function IsElementContentTruncatedDirective(elementRef, cd) {
            this.elementRef = elementRef;
            this.cd = cd;
        }
        IsElementContentTruncatedDirective.prototype.ngAfterViewInit = function () {
            var _this = this;
            setTimeout(function () {
                _this.setIsContentTruncatedInParentComponent();
            });
        };
        IsElementContentTruncatedDirective.prototype.setIsContentTruncatedInParentComponent = function () {
            var isContentTruncated = isElementContentTruncated(this.elementRef.nativeElement);
            this.hostRef.setIsContentTruncated(isContentTruncated);
            this.cd.markForCheck();
        };
        return IsElementContentTruncatedDirective;
    }());
    /** @nocollapse */ IsElementContentTruncatedDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: IsElementContentTruncatedDirective, deps: [{ token: i0__namespace.ElementRef }, { token: i0__namespace.ChangeDetectorRef }], target: i0__namespace.ɵɵFactoryTarget.Directive });
    /** @nocollapse */ IsElementContentTruncatedDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: IsElementContentTruncatedDirective, selector: "[bbIsElementContentTruncated]", inputs: { hostRef: "hostRef" }, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: IsElementContentTruncatedDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: '[bbIsElementContentTruncated]',
                    }]
            }], ctorParameters: function () { return [{ type: i0__namespace.ElementRef }, { type: i0__namespace.ChangeDetectorRef }]; }, propDecorators: { hostRef: [{
                    type: i0.Input
                }] } });

    var pipes = [NotificationLocalizationPipe, InterpolateLinkPipe];
    var components = [NotificationsCapabilityIconComponent];
    var directives = [NotificationsCapabilityIconCustomizableDirective, IsElementContentTruncatedDirective];
    var uiModules = [i1.IconModule];
    var NotificationsCommonModule = /** @class */ (function () {
        function NotificationsCommonModule() {
        }
        return NotificationsCommonModule;
    }());
    /** @nocollapse */ NotificationsCommonModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: NotificationsCommonModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    /** @nocollapse */ NotificationsCommonModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: NotificationsCommonModule, declarations: [NotificationLocalizationPipe, InterpolateLinkPipe, NotificationsCapabilityIconComponent, NotificationsCapabilityIconCustomizableDirective, IsElementContentTruncatedDirective], imports: [i2.CommonModule, i1.IconModule], exports: [NotificationLocalizationPipe, InterpolateLinkPipe, NotificationsCapabilityIconComponent, NotificationsCapabilityIconCustomizableDirective, IsElementContentTruncatedDirective, i1.IconModule] });
    /** @nocollapse */ NotificationsCommonModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: NotificationsCommonModule, imports: [__spreadArray([i2.CommonModule], __read(uiModules)), i1.IconModule] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: NotificationsCommonModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: __spreadArray([i2.CommonModule], __read(uiModules)),
                        declarations: __spreadArray(__spreadArray(__spreadArray([], __read(pipes)), __read(components)), __read(directives)),
                        exports: __spreadArray(__spreadArray(__spreadArray(__spreadArray([], __read(pipes)), __read(components)), __read(directives)), __read(uiModules)),
                    }]
            }] });

    var NOTIFICATIONS_BADGE_FORCE_UPDATE_EVENT = 'bb.event.notifications-badge.force-update';
    var NOTIFICATIONS_CREATE_LOCAL_EVENT = 'bb.event.notifications-popups.notify';

    var getPropertyValue = function (input, prop) { return rxjs.combineLatest([input, prop]).pipe(operators.map(function (_a) {
        var _b = __read(_a, 2), inputVal = _b[0], propVal = _b[1];
        return (inputVal !== undefined ? inputVal : propVal);
    })); };

    function propertiesListToArray() {
        return function (source$) { return source$.pipe(operators.map(function (properties) { return properties.replace(/\s/g, '').split(','); }), operators.map(function (properties) { return properties.filter(Boolean); })); };
    }
    var shareReplayOnce = function () { return function (source) {
        // use refCount: true to prevent memory leaks issue with shareReplay
        return source.pipe(operators.shareReplay({ refCount: true, bufferSize: 1 }));
    }; };

    var modelItemToBoolean = function (value) {
        if (typeof value === 'boolean') {
            return value;
        }
        return value.trim() === 'true';
    };
    var modelItemToArray = function (value) { return value.replace(/\s/g, '').split(','); };
    var secondsToMilliseconds = function (seconds) { return seconds * 1000; };

    var DefaultBaseWidgetProperties = {
        allowedRoutes: [],
        origins: [],
        severityLevels: [],
    };
    var BaseWidgetPropertiesService = /** @class */ (function () {
        function BaseWidgetPropertiesService(itemModel) {
            this.itemModel = itemModel;
            this.allowedRoutes$$ = new rxjs.BehaviorSubject(undefined);
            this.origins$$ = new rxjs.BehaviorSubject(undefined);
            this.severityLevels$$ = new rxjs.BehaviorSubject(undefined);
            this.allowedRoutes$ = getPropertyValue(this.allowedRoutes$$, this.allowedRoutes);
            this.origins$ = getPropertyValue(this.origins$$, this.origins);
            this.severityLevels$ = getPropertyValue(this.severityLevels$$, this.severityLevels);
        }
        Object.defineProperty(BaseWidgetPropertiesService.prototype, "allowedRoutes", {
            get: function () {
                if (!this.itemModel) {
                    return this.defaultModelObservable(DefaultBaseWidgetProperties.allowedRoutes);
                }
                return this.itemModel.property('allowedRoutes', '').pipe(operators.first(), propertiesListToArray());
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(BaseWidgetPropertiesService.prototype, "origins", {
            get: function () {
                if (!this.itemModel) {
                    return this.defaultModelObservable(DefaultBaseWidgetProperties.origins);
                }
                return this.itemModel.property('origins', '').pipe(propertiesListToArray());
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(BaseWidgetPropertiesService.prototype, "severityLevels", {
            get: function () {
                if (!this.itemModel) {
                    return this.defaultModelObservable(DefaultBaseWidgetProperties.severityLevels);
                }
                return this.itemModel.property('severityLevels', '').pipe(propertiesListToArray());
            },
            enumerable: false,
            configurable: true
        });
        BaseWidgetPropertiesService.prototype.setAllowedRoutes = function (value) {
            this.allowedRoutes$$.next(modelItemToArray(value));
        };
        BaseWidgetPropertiesService.prototype.setOrigins = function (value) {
            this.origins$$.next(modelItemToArray(value));
        };
        BaseWidgetPropertiesService.prototype.setSeverityLevels = function (value) {
            this.severityLevels$$.next(modelItemToArray(value));
        };
        BaseWidgetPropertiesService.prototype.defaultModelObservable = function (value) {
            return rxjs.defer(function () { return rxjs.of(value); }).pipe(operators.publishReplay(1), operators.refCount());
        };
        return BaseWidgetPropertiesService;
    }());
    /** @nocollapse */ BaseWidgetPropertiesService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BaseWidgetPropertiesService, deps: [{ token: i1__namespace$1.ItemModel, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    /** @nocollapse */ BaseWidgetPropertiesService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BaseWidgetPropertiesService });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BaseWidgetPropertiesService, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () {
            return [{ type: i1__namespace$1.ItemModel, decorators: [{
                            type: i0.Optional
                        }] }];
        } });

    function toISOExtendedFormat(value) {
        // eslint-disable-next-line max-len
        // The date's format `2020-04-09T10:55:03.576+0000` from BE doesn't support in IE, Safari. Prevent `Invalid Date` result
        var offsetSign = value.slice(-5, -4);
        return offsetSign === '+' || '-' === offsetSign ? value.slice(0, -2) + ":" + value.slice(-2) : value;
    }

    var LeaveZoneScheduler = /** @class */ (function () {
        function LeaveZoneScheduler(zone, scheduler) {
            this.zone = zone;
            this.scheduler = scheduler;
        }
        LeaveZoneScheduler.prototype.schedule = function () {
            var _this = this;
            var args = [];
            for (var _i = 0; _i < arguments.length; _i++) {
                args[_i] = arguments[_i];
            }
            // eslint-disable-next-line prefer-spread
            return this.zone.runOutsideAngular(function () { return _this.scheduler.schedule.apply(_this.scheduler, args); });
        };
        LeaveZoneScheduler.prototype.now = function () {
            return this.scheduler.now();
        };
        return LeaveZoneScheduler;
    }());
    var EnterZoneScheduler = /** @class */ (function () {
        function EnterZoneScheduler(zone, scheduler) {
            this.zone = zone;
            this.scheduler = scheduler;
        }
        EnterZoneScheduler.prototype.schedule = function () {
            var _this = this;
            var args = [];
            for (var _i = 0; _i < arguments.length; _i++) {
                args[_i] = arguments[_i];
            }
            // eslint-disable-next-line prefer-spread
            return this.zone.run(function () { return _this.scheduler.schedule.apply(_this.scheduler, args); });
        };
        EnterZoneScheduler.prototype.now = function () {
            return this.scheduler.now();
        };
        return EnterZoneScheduler;
    }());
    function leaveZone(zone, scheduler) {
        return new LeaveZoneScheduler(zone, scheduler);
    }
    function enterZone(zone, scheduler) {
        return new EnterZoneScheduler(zone, scheduler);
    }

    /**
     * Generated bundle index. Do not edit.
     */

    exports.BaseWidgetPropertiesService = BaseWidgetPropertiesService;
    exports.NOTIFICATIONS_BADGE_FORCE_UPDATE_EVENT = NOTIFICATIONS_BADGE_FORCE_UPDATE_EVENT;
    exports.NOTIFICATIONS_CREATE_LOCAL_EVENT = NOTIFICATIONS_CREATE_LOCAL_EVENT;
    exports.NotificationsCapabilityIconComponent = NotificationsCapabilityIconComponent;
    exports.NotificationsCapabilityIconCustomizableDirective = NotificationsCapabilityIconCustomizableDirective;
    exports.NotificationsCommonModule = NotificationsCommonModule;
    exports.enterZone = enterZone;
    exports.getPropertyValue = getPropertyValue;
    exports.isElementContentTruncated = isElementContentTruncated;
    exports.leaveZone = leaveZone;
    exports.modelItemToArray = modelItemToArray;
    exports.modelItemToBoolean = modelItemToBoolean;
    exports.propertiesListToArray = propertiesListToArray;
    exports.secondsToMilliseconds = secondsToMilliseconds;
    exports.shareReplayOnce = shareReplayOnce;
    exports.toISOExtendedFormat = toISOExtendedFormat;
    exports["ӨInterpolateLinkPipe"] = InterpolateLinkPipe;
    exports["ӨIsElementContentTruncatedDirective"] = IsElementContentTruncatedDirective;
    exports["ӨNotificationLocalizationPipe"] = NotificationLocalizationPipe;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
