// @backbase/retail-loans-journey-ang/esm2015/lib/loans-journey.routes.js
import '@angular/localize/init';
import { LoanInfoWrapperComponent } from './components/loan-details/tabs/loan-info-wrapper/loan-info-wrapper.component';
import { LoanDetailsWrapperComponent } from './components/loan-details/loan-details-wrapper/loan-details-wrapper.component';
import { LoanTransactionsWrapperComponent } from './components/loan-details/tabs/loan-transactions-wrapper/loan-transactions-wrapper.component';
import { PaymentScheduleWrapperComponent } from './components/loan-details/tabs/payment-schedule-wrapper/payment-schedule-wrapper.component';
import { PaymentScheduleDetailsWrapperComponent } from './components/loan-details/tabs/payment-schedule-wrapper/payment-schedule-details-wrapper/payment-schedule-details-wrapper.component';
import { loanIdParamKey, scheduleIdParam } from './config/route-params.model';
// To prevent error on package:apps process
export const tabs = {
    transactions: () => $localize `:Loan details transactions tab title@@loans.details.tab.transactions.title:Transactions`,
    statements: () => $localize `:Loan details account statements tab title@@loans.details.tab.statements.title:Statements`,
    loanInfo: () => $localize `:Loan details info tab title@@loans.details.tab.loan-info.title:Loan info`,
    paymentSchedule: () => $localize `:Loan details payment schedule tab title@@loans.details.tab.payment-schedule.title:Payment schedule`,
};
export const loansRoutes = [
    {
        path: `:${loanIdParamKey}`,
        component: LoanDetailsWrapperComponent,
        children: [
            {
                path: '',
                redirectTo: 'transactions',
                pathMatch: 'full',
            },
            {
                path: 'transactions',
                component: LoanTransactionsWrapperComponent,
                data: { titleFn: tabs.transactions },
            },
            {
                path: 'info',
                component: LoanInfoWrapperComponent,
                data: { titleFn: tabs.loanInfo },
            },
            {
                path: 'payment-schedule',
                component: PaymentScheduleWrapperComponent,
                data: { titleFn: tabs.paymentSchedule },
                children: [{ path: `:${scheduleIdParam}`, component: PaymentScheduleDetailsWrapperComponent }],
            },
        ],
    },
];
