// @backbase/retail-loans-journey-ang/esm2015/lib/loans-journey.module.js
import { provideRoutes, RouterModule } from '@angular/router';
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { EntitlementsModule } from '@backbase/foundation-ang/entitlements';
import { LoanInfoItemModule, LoanInfoPrintButtonModule, LoanItemNameModule, LoansCollectionUiAngModule, LoansHeadingModule, } from '@backbase/loans-collection-ui-ang';
import { AccountStatementBusinessWidgetModule } from '@backbase/account-statement-business-widget-ang';
import { LoansJourneyConfigurationService } from './loans-journey-configuration';
import { loansRoutes } from './loans-journey.routes';
import { LoansStoreModule } from './store/loans/store.module';
import { PaymentScheduleStoreModule } from './store/payment-schedule/store.module';
import { LoansService } from './services/loans.service';
import { LoanMapperService } from './services/loan-mapper.service';
import { ArrangementsService } from './services/arrangements.service';
import { LoanSummaryComponent } from './components/loan-details/loan-summary/loan-summary.component';
import { LoanInfoWrapperComponent } from './components/loan-details/tabs/loan-info-wrapper/loan-info-wrapper.component';
import { LoanDetailsWrapperComponent } from './components/loan-details/loan-details-wrapper/loan-details-wrapper.component';
import { LoanTransactionsWrapperComponent } from './components/loan-details/tabs/loan-transactions-wrapper/loan-transactions-wrapper.component';
import { PaymentScheduleWrapperComponent } from './components/loan-details/tabs/payment-schedule-wrapper/payment-schedule-wrapper.component';
import { PaymentScheduleDetailsWrapperComponent } from './components/loan-details/tabs/payment-schedule-wrapper/payment-schedule-details-wrapper/payment-schedule-details-wrapper.component';
import { PaymentScheduleDetailsModalHeaderComponent } from './components/loan-details/tabs/payment-schedule-wrapper/payment-schedule-details-wrapper/payment-schedule-details-modal-header/payment-schedule-details-modal-header.component';
import { LoanInfoRetailComponent } from './components/loan-details/tabs/loan-info-wrapper/loan-info-retail/loan-info-retail.component';
import { TransactionsListWidgetModule } from '@backbase/transactions-list-widget-ang';
import * as i0 from "@angular/core";
const components = [
    LoanDetailsWrapperComponent,
    LoanSummaryComponent,
    LoanTransactionsWrapperComponent,
    LoanInfoWrapperComponent,
    PaymentScheduleWrapperComponent,
    PaymentScheduleDetailsWrapperComponent,
    PaymentScheduleDetailsModalHeaderComponent,
    LoanInfoRetailComponent,
];
const modules = [
    LoanInfoPrintButtonModule,
    LoanItemNameModule,
    TransactionsListWidgetModule,
    LoanInfoItemModule,
    LoansHeadingModule,
    LoansCollectionUiAngModule,
    AccountStatementBusinessWidgetModule,
];
export class LoansJourneyModule {
    static forRoot(data = { routes: loansRoutes }) {
        return {
            ngModule: LoansJourneyModule,
            providers: [provideRoutes(data.routes)],
        };
    }
}
/** @nocollapse */ LoansJourneyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoansJourneyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoansJourneyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoansJourneyModule, declarations: [LoanDetailsWrapperComponent,
        LoanSummaryComponent,
        LoanTransactionsWrapperComponent,
        LoanInfoWrapperComponent,
        PaymentScheduleWrapperComponent,
        PaymentScheduleDetailsWrapperComponent,
        PaymentScheduleDetailsModalHeaderComponent,
        LoanInfoRetailComponent], imports: [CommonModule, RouterModule, LoansStoreModule, PaymentScheduleStoreModule, EntitlementsModule, LoanInfoPrintButtonModule,
        LoanItemNameModule,
        TransactionsListWidgetModule,
        LoanInfoItemModule,
        LoansHeadingModule,
        LoansCollectionUiAngModule,
        AccountStatementBusinessWidgetModule], exports: [LoanDetailsWrapperComponent,
        LoanSummaryComponent,
        LoanTransactionsWrapperComponent,
        LoanInfoWrapperComponent,
        PaymentScheduleWrapperComponent,
        PaymentScheduleDetailsWrapperComponent,
        PaymentScheduleDetailsModalHeaderComponent,
        LoanInfoRetailComponent, LoanInfoPrintButtonModule,
        LoanItemNameModule,
        TransactionsListWidgetModule,
        LoanInfoItemModule,
        LoansHeadingModule,
        LoansCollectionUiAngModule,
        AccountStatementBusinessWidgetModule] });
/** @nocollapse */ LoansJourneyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoansJourneyModule, providers: [LoansJourneyConfigurationService, LoanMapperService, LoansService, ArrangementsService], imports: [[CommonModule, RouterModule, LoansStoreModule, PaymentScheduleStoreModule, EntitlementsModule, ...modules], LoanInfoPrintButtonModule,
        LoanItemNameModule,
        TransactionsListWidgetModule,
        LoanInfoItemModule,
        LoansHeadingModule,
        LoansCollectionUiAngModule,
        AccountStatementBusinessWidgetModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoansJourneyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components],
                    imports: [CommonModule, RouterModule, LoansStoreModule, PaymentScheduleStoreModule, EntitlementsModule, ...modules],
                    exports: [...components, ...modules],
                    providers: [LoansJourneyConfigurationService, LoanMapperService, LoansService, ArrangementsService],
                }]
        }] });
