// @backbase/messages-common/esm2015/src/services/messages-unread-conversation-count/messages-unread-conversation-count.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { MessagesUnreadConversationCountService } from './messages-unread-conversation-count.service';
import { MessagecenterHttpService } from '@backbase/messages-v5-http-ang';
import { PUBSUB } from '@backbase/foundation-ang/web-sdk';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
export class MessagesUnreadConversationCountModule {
}
MessagesUnreadConversationCountModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesUnreadConversationCountModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MessagesUnreadConversationCountModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesUnreadConversationCountModule, imports: [CommonModule, i1.BackbaseCoreModule] });
MessagesUnreadConversationCountModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesUnreadConversationCountModule, providers: [MessagesUnreadConversationCountService], imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                pipelineProcessors: [
                    {
                        name: 'counters-badge',
                        provider: {
                            provide: MessagesUnreadConversationCountService,
                            useClass: MessagesUnreadConversationCountService,
                            deps: [MessagecenterHttpService, PUBSUB],
                        },
                    },
                ],
            }),
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesUnreadConversationCountModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            pipelineProcessors: [
                                {
                                    name: 'counters-badge',
                                    provider: {
                                        provide: MessagesUnreadConversationCountService,
                                        useClass: MessagesUnreadConversationCountService,
                                        deps: [MessagecenterHttpService, PUBSUB],
                                    },
                                },
                            ],
                        }),
                    ],
                    providers: [MessagesUnreadConversationCountService],
                }]
        }] });
