// @backbase/messages-common/esm2015/src/services/messages-unread-conversation-count/messages-unread-conversation-count.service.js
import { Inject, Injectable } from '@angular/core';
import { merge, of, Subject, timer } from 'rxjs';
import { concatMap, map, switchMap } from 'rxjs/operators';
import { PUBSUB } from '@backbase/foundation-ang/web-sdk';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/messages-v5-http-ang";
const MessagesToken = 'messages';
const MessagesPollPeriod = 5000;
const bbEventUnreadConversationCountUpdate = 'bb.event.unread-conversation-count.update';
export class MessagesUnreadConversationCountService {
    constructor(messagecenterService, pubSub) {
        this.messagecenterService = messagecenterService;
        this.pubSub = pubSub;
        this.resetSubject = new Subject();
        this.subscriptionHandler = this.reset.bind(this);
        this.pubSub.subscribe(bbEventUnreadConversationCountUpdate, this.subscriptionHandler);
    }
    /**
     * Conversations count poll source
     */
    source() {
        return () => merge(this.resetSubject, of(undefined)).pipe(switchMap(() => timer(0, MessagesPollPeriod).pipe(concatMap(() => this.getUnreadConversationCount()), map(body => ({ token: MessagesToken, count: body ? body.unreadCount : 0 })))));
    }
    /**
     * Get conversations count
     *
     * @returns Get conversations count stream
     */
    getUnreadConversationCount() {
        return this.messagecenterService.getUnreadConversationsCount();
    }
    reset() {
        this.resetSubject.next();
    }
}
MessagesUnreadConversationCountService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesUnreadConversationCountService, deps: [{ token: i1.MessagecenterHttpService }, { token: PUBSUB }], target: i0.ɵɵFactoryTarget.Injectable });
MessagesUnreadConversationCountService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesUnreadConversationCountService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesUnreadConversationCountService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.MessagecenterHttpService }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [PUBSUB]
                }] }]; } });
