// @backbase/loans-collection-ui-ang/esm2015/lib/loan-info/loan-info-print-button/loan-info-print-button.component.js
import { Component, Inject } from '@angular/core';
import { DOCUMENT } from '@angular/common';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/icon";
import * as i2 from "@backbase/ui-ang/button";
// @dynamic
export class LoanInfoPrintButtonComponent {
    constructor(document) {
        this.document = document;
    }
    /**
     * Print the document
     */
    print() {
        var _a;
        (_a = this.document.defaultView) === null || _a === void 0 ? void 0 : _a.print();
    }
}
/** @nocollapse */ LoanInfoPrintButtonComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoPrintButtonComponent, deps: [{ token: DOCUMENT }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ LoanInfoPrintButtonComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: LoanInfoPrintButtonComponent, selector: "bb-loan-info-print-button", ngImport: i0, template: "<button\n  data-role=\"bb-loans-list-info-print-btn\"\n  bbButton\n  class=\"d-print-none\"\n  color=\"secondary\"\n  aria-label=\"Print page\"\n  i18n-aria-label=\"Loan info print button aria label@@loans.loan-info.print.button.aria-label\"\n  type=\"button\"\n  [circle]=\"true\"\n  (click)=\"print()\"\n>\n  <bb-icon-ui name=\"print\"></bb-icon-ui>\n</button>\n", components: [{ type: i1.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], directives: [{ type: i2.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoPrintButtonComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-loan-info-print-button',
                    templateUrl: './loan-info-print-button.component.html',
                }]
        }], ctorParameters: function () { return [{ type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }]; } });
