// @backbase/loans-collection-ui-ang/esm2015/lib/loan-info/loan-info-print-button/loan-info-print-button.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ButtonModule } from '@backbase/ui-ang/button';
import { IconModule } from '@backbase/ui-ang/icon';
import { LoanInfoPrintButtonComponent } from './loan-info-print-button.component';
import * as i0 from "@angular/core";
export class LoanInfoPrintButtonModule {
}
/** @nocollapse */ LoanInfoPrintButtonModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoPrintButtonModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanInfoPrintButtonModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoPrintButtonModule, declarations: [LoanInfoPrintButtonComponent], imports: [CommonModule, ButtonModule, IconModule], exports: [LoanInfoPrintButtonComponent] });
/** @nocollapse */ LoanInfoPrintButtonModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoPrintButtonModule, imports: [[CommonModule, ButtonModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoPrintButtonModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanInfoPrintButtonComponent],
                    imports: [CommonModule, ButtonModule, IconModule],
                    exports: [LoanInfoPrintButtonComponent],
                }]
        }] });
