// @backbase/ui-ang/esm2015/_example/dropdown-single-select/showcase/dropdown-single-select-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { DropdownSingleSelectShowcaseComponent } from './dropdown-single-select-showcase.component';
import { DropdownSingleSelectModule } from '@backbase/ui-ang/dropdown-single-select';
import { ReactiveFormsModule } from '@angular/forms';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import * as i0 from "@angular/core";
export class DropdownSingleSelectShowcaseModule {
}
DropdownSingleSelectShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownSingleSelectShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
DropdownSingleSelectShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownSingleSelectShowcaseModule, declarations: [DropdownSingleSelectShowcaseComponent], imports: [CommonModule, DropdownSingleSelectModule, ReactiveFormsModule, InputValidationMessageModule] });
DropdownSingleSelectShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownSingleSelectShowcaseModule, imports: [[CommonModule, DropdownSingleSelectModule, ReactiveFormsModule, InputValidationMessageModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownSingleSelectShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, DropdownSingleSelectModule, ReactiveFormsModule, InputValidationMessageModule],
                    declarations: [DropdownSingleSelectShowcaseComponent],
                    entryComponents: [DropdownSingleSelectShowcaseComponent],
                }]
        }] });
