// @backbase/ui-ang/esm2015/control-error-handler/control-error-handler.const.js
import { InjectionToken } from '@angular/core';
import { ValidationMessageComponent } from './components/validation-message/validation-message.component';
import '@angular/localize/init';
export const BB_VALIDATION_ERRORS = new InjectionToken('BB_VALIDATION_ERRORS', {
    providedIn: 'root',
    factory: () => defaultFromErrors,
});
export const BB_DYNAMIC_VALIDATION_ERROR_TMPL = new InjectionToken('BB_DYNAMIC_VALIDATION_ERROR_TMPL', {
    providedIn: 'root',
    factory: () => ValidationMessageComponent,
});
export const defaultFromErrors = {
    invalid: () => $localize `:@@validation-form-invalid-error:Field is invalid`,
    required: () => $localize `:@@validation-form-required-error:Field is required`,
    minlength: ({ requiredLength }) => {
        const singleStr = $localize `:@@validation-form-minlength-error-single:
    Must be shorter then ${requiredLength} character`;
        const multipleStr = $localize `:@@validation-form-minlength-error-multiple:
    Must be shorter then ${requiredLength} characters`;
        return requiredLength === 1 ? singleStr : multipleStr;
    },
    maxlength: ({ requiredLength }) => {
        const singleStr = $localize `:@@validation-form-minlength-error-single:
    Must be at least ${requiredLength} character long`;
        const multipleStr = $localize `:@@validation-form-minlength-error-multiple:
    Must be at least ${requiredLength} characters long`;
        return requiredLength === 1 ? singleStr : multipleStr;
    },
    min: ({ min }) => $localize `:@@validation-form-min-error:Expect value to be bigger than ${min}`,
    max: ({ max }) => $localize `:@@validation-form-max-error:Expect value to be less than ${max}`,
    email: () => $localize `:@@validation-form-email-error:Not a valid email`,
};
