// @backbase/ui-ang/esm2015/control-error-handler/control-error-handler.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ValidationMessageComponent } from './components/validation-message/validation-message.component';
import { ControlErrorContainerDirective } from './directives/control-error-container.directive';
import { FormSubmitDirective } from './directives/form-submit.directive';
import { ValidationErrorsDirective } from './directives/validation-errors.directive';
import * as i0 from "@angular/core";
const directives = [ValidationErrorsDirective, FormSubmitDirective, ControlErrorContainerDirective];
const components = [ValidationMessageComponent];
export class ControlErrorHandlerModule {
}
ControlErrorHandlerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ControlErrorHandlerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ControlErrorHandlerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ControlErrorHandlerModule, declarations: [ValidationErrorsDirective, FormSubmitDirective, ControlErrorContainerDirective, ValidationMessageComponent], imports: [CommonModule], exports: [ValidationErrorsDirective, FormSubmitDirective, ControlErrorContainerDirective, ValidationMessageComponent] });
ControlErrorHandlerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ControlErrorHandlerModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ControlErrorHandlerModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [...directives, ...components],
                    exports: [...directives, ...components],
                }]
        }] });
