// @backbase/ui-ang/esm2015/_example/account-number-pipe/showcase/account-number-pipe-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { AccountNumberPipeModule } from '@backbase/ui-ang/account-number-pipe';
import { AccountNumberPipeShowcaseComponent } from './account-number-pipe-showcase.component';
import { ACCOUNTS_DISPLAYING_FORMAT, } from '@backbase/ui-ang/account-number-pipe';
import * as i0 from "@angular/core";
export const accountsDisplayingFormatConfiguration = {
    iban: { segments: 4 },
    bban: { segments: 4 },
    cardNumber: {
        segments: [6, 8, 6],
        cardProviders: {
            visa: { segments: [3, 9, 9] },
        },
    },
};
export class AccountNumberPipeShowcaseModule {
}
AccountNumberPipeShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountNumberPipeShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AccountNumberPipeShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountNumberPipeShowcaseModule, declarations: [AccountNumberPipeShowcaseComponent], imports: [CommonModule, FormsModule, AccountNumberPipeModule] });
AccountNumberPipeShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountNumberPipeShowcaseModule, providers: [{ provide: ACCOUNTS_DISPLAYING_FORMAT, useValue: accountsDisplayingFormatConfiguration }], imports: [[CommonModule, FormsModule, AccountNumberPipeModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountNumberPipeShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, FormsModule, AccountNumberPipeModule],
                    declarations: [AccountNumberPipeShowcaseComponent],
                    entryComponents: [AccountNumberPipeShowcaseComponent],
                    providers: [{ provide: ACCOUNTS_DISPLAYING_FORMAT, useValue: accountsDisplayingFormatConfiguration }],
                }]
        }] });
