// @backbase/ui-ang/esm2015/_example/account-number-pipe/showcase/account-number-pipe-showcase.component.js
import { __decorate } from "tslib";
import { Component, Inject } from '@angular/core';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import { ACCOUNTS_DISPLAYING_FORMAT, } from '@backbase/ui-ang/account-number-pipe';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "@backbase/ui-ang/account-number-pipe";
let AccountNumberPipeShowcaseComponent = class AccountNumberPipeShowcaseComponent {
    constructor(configuration) {
        this.configuration = configuration;
        this.customConfiguration = { segments: [1, 2, 3, 1, 2, 3, 1, 2, 3] };
        this.accounts = [
            {
                description: 'iban',
                typeOrFormat: 'iban',
                number: 'IBAN1234567891234567',
            },
            {
                description: 'bban',
                typeOrFormat: 'bban',
                number: 'BBAN1234567891234567',
            },
            {
                description: 'cardNumber',
                typeOrFormat: 'cardNumber',
                number: 'num1234567891234567',
            },
            {
                description: 'cardNumber (visa provider)',
                typeOrFormat: 'cardNumber',
                number: 'num1234567891234567',
                cardProviderName: 'Visa',
            },
            {
                description: `Custom format [${this.customConfiguration.segments}]`,
                typeOrFormat: this.customConfiguration,
                number: 'num1234567891234567',
            },
        ];
    }
};
AccountNumberPipeShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountNumberPipeShowcaseComponent, deps: [{ token: ACCOUNTS_DISPLAYING_FORMAT }], target: i0.ɵɵFactoryTarget.Component });
AccountNumberPipeShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AccountNumberPipeShowcaseComponent, selector: "bb-account-number-pipe-ui-showcase", ngImport: i0, template: "<div class=\"bb-block bb-block--lg\">\n  <h2>App configuration</h2>\n  <div>\n    <code>iban: {{ configuration.iban | json }}</code>\n  </div>\n  <div>\n    <code>bban: {{ configuration.bban | json }}</code>\n  </div>\n  <div>\n    <code>cardNumber: {{ configuration.cardNumber | json }}</code>\n  </div>\n</div>\n\n<div class=\"bb-block bb-block--lg\">\n  <ng-container *ngFor=\"let account of accounts\">\n    <h4>{{ account.description }}</h4>\n    <p>{{ account.number | bbAccountNumber: account.typeOrFormat:account.cardProviderName }}</p>\n  </ng-container>\n</div>\n", directives: [{ type: i1.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }], pipes: { "json": i1.JsonPipe, "bbAccountNumber": i2.AccountNumberPipe } });
AccountNumberPipeShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
], AccountNumberPipeShowcaseComponent);
export { AccountNumberPipeShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountNumberPipeShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-account-number-pipe-ui-showcase',
                    templateUrl: './account-number-pipe-showcase.component.html',
                }]
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [ACCOUNTS_DISPLAYING_FORMAT]
                }] }]; } });
