// @backbase/notifications-ang/esm2015/lib/notifications-popups-ang/src/components/notifications-popups.component.js
import { Component, Input, Optional } from '@angular/core';
import { NotificationsCommunicationService } from '../../../notifications-common-ang/src/services/notifications-communication.service';
import { titles } from '../../../notifications-common-ang/src/config/titles.config';
import * as i0 from "@angular/core";
import * as i1 from "../../../notifications-common-ang/src/services/notifications-communication.service";
import * as i2 from "@backbase/notifications-popups-widget-ang";
export class NotificationsPopupsComponent {
    constructor(communicationService) {
        this.communicationService = communicationService;
    }
    /**
     * Method that emits navigation data and triggers navigation.
     */
    navigate(navigationData) {
        if (this.communicationService) {
            this.communicationService.notificationNavigation(navigationData);
        }
        else {
            this.showNoCommunicationServiceProvidedError();
        }
    }
    showNoCommunicationServiceProvidedError() {
        console.warn(titles.noCommunicationServiceProvidedError);
    }
}
/** @nocollapse */ NotificationsPopupsComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsPopupsComponent, deps: [{ token: i1.NotificationsCommunicationService, optional: true }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ NotificationsPopupsComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: NotificationsPopupsComponent, selector: "bb-notifications-popups", inputs: { origins: "origins", severityLevels: "severityLevels", pollingInterval: "pollingInterval", allowedRoutes: "allowedRoutes", autofocus: "autofocus", displayShowMoreButtonInPopup: "displayShowMoreButtonInPopup", popupMessageLineAmount: "popupMessageLineAmount" }, ngImport: i0, template: "<bb-notifications-popups-widget\n  [origins]=\"origins\"\n  [severityLevels]=\"severityLevels\"\n  [pollingInterval]=\"pollingInterval\"\n  [allowedRoutes]=\"allowedRoutes\"\n  [autofocus]=\"autofocus\"\n  [displayShowMoreButtonInPopup]=\"displayShowMoreButtonInPopup\"\n  [popupMessageLineAmount]=\"popupMessageLineAmount\"\n  (notificationRouting)=\"navigate($event)\"\n></bb-notifications-popups-widget>\n", components: [{ type: i2.NotificationsPopupsWidgetComponent, selector: "bb-notifications-popups-widget", inputs: ["origins", "severityLevels", "pollingInterval", "allowedRoutes", "autofocus", "displayShowMoreButtonInPopup", "popupMessageLineAmount"], outputs: ["notificationRouting"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsPopupsComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-notifications-popups',
                    templateUrl: './notifications-popups.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.NotificationsCommunicationService, decorators: [{
                    type: Optional
                }] }]; }, propDecorators: { origins: [{
                type: Input
            }], severityLevels: [{
                type: Input
            }], pollingInterval: [{
                type: Input
            }], allowedRoutes: [{
                type: Input
            }], autofocus: [{
                type: Input
            }], displayShowMoreButtonInPopup: [{
                type: Input
            }], popupMessageLineAmount: [{
                type: Input
            }] } });
