// @backbase/loans-collection-ui-ang/esm2015/lib/custom-loan-payments/custom-loan-payments.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CustomReviewScreenComponent } from './custom-review-screen/custom-review-screen.component';
import { ReviewPaymentModule } from '@backbase/payments-collection-ui-ang';
import { PaymentOrdersUiModule } from '@backbase/payment-orders-ang';
import { CustomScheduleDetailsComponent } from './custom-schedule-details/custom-schedule-details.component';
import * as i0 from "@angular/core";
export class CustomLoanPaymentsModule {
}
/** @nocollapse */ CustomLoanPaymentsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomLoanPaymentsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ CustomLoanPaymentsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomLoanPaymentsModule, declarations: [CustomReviewScreenComponent, CustomScheduleDetailsComponent], imports: [CommonModule, ReviewPaymentModule, PaymentOrdersUiModule] });
/** @nocollapse */ CustomLoanPaymentsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomLoanPaymentsModule, imports: [[CommonModule, ReviewPaymentModule, PaymentOrdersUiModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomLoanPaymentsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ReviewPaymentModule, PaymentOrdersUiModule],
                    declarations: [CustomReviewScreenComponent, CustomScheduleDetailsComponent],
                    schemas: [],
                }]
        }] });
