// @backbase/manage-pockets-journey-ang/esm2015/src/views/manage-pockets-create-view.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../manage-pockets-common/services/manage-pockets.service";
import * as i2 from "@angular/router";
import * as i3 from "@backbase/ui-ang/notification";
import * as i4 from "../manage-pockets-journey-config.service";
import * as i5 from "@backbase/ui-ang/header";
import * as i6 from "../manage-pockets-create/components/manage-pockets-create-wrapper/manage-pockets-create-wrapper.component";
import * as i7 from "@angular/common";
export class ManagePocketsCreateViewComponent {
    constructor(managePocketsService, router, route, cd, notificationService, configService) {
        this.managePocketsService = managePocketsService;
        this.router = router;
        this.route = route;
        this.cd = cd;
        this.notificationService = notificationService;
        this.configService = configService;
        this.awaitingCreateResponse = false;
    }
    abort() {
        this.navigateToList();
    }
    saveNewPocket(newPocketItem) {
        this.awaitingCreateResponse = true;
        const onSuccess = () => {
            this.awaitingCreateResponse = false;
            this.cd.markForCheck();
            this.pushNotification($localize `:Success pocket creation toast message@@managePockets.create.toast.success.message:Your "${newPocketItem.name}:pocket name:" pocket is created.`, 'success', $localize `:Success pocket creation toast header@@managePockets.create.toast.success.header:Your Pocket is Ready!`);
            this.navigateToList();
        };
        const onError = () => {
            this.awaitingCreateResponse = false;
            this.cd.markForCheck();
            this.pushNotification($localize `:Error pocket creation toast message@@managePockets.create.toast.error.message:Unable to proceed, please try again.`, 'error', $localize `:Error pocket creation toast header@@managePockets.create.toast.error.header:Failed to Create Pocket`);
        };
        this.managePocketsService.postPocket(newPocketItem).subscribe(onSuccess, onError);
    }
    navigateToList() {
        this.router.navigate(['../list'], { relativeTo: this.route });
    }
    pushNotification(message, modifier, header) {
        this.notificationService.showNotification({
            header,
            message,
            modifier,
            dismissible: true,
            ttl: 5000,
        });
    }
}
ManagePocketsCreateViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsCreateViewComponent, deps: [{ token: i1.ManagePocketsService }, { token: i2.Router }, { token: i2.ActivatedRoute }, { token: i0.ChangeDetectorRef }, { token: i3.NotificationService }, { token: i4.ManagePocketsJourneyConfigService }], target: i0.ɵɵFactoryTarget.Component });
ManagePocketsCreateViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ManagePocketsCreateViewComponent, selector: "bb-manage-pockets-create-view", ngImport: i0, template: "<div class=\"heading-widget bb-heading-widget px-0\" *ngIf=\"configService.displayHeaders\">\n  <bb-header-ui\n    headingType=\"h1\"\n    headingClasses=\"bb-heading-widget__heading\"\n    heading=\"Create Pocket\"\n    i18n-heading=\"Create Pocket heading@@managePockets.createView.heading.title\"\n  ></bb-header-ui>\n</div>\n\n<bb-manage-pockets-create-wrapper\n  (cancel)=\"abort()\"\n  (newPocket)=\"saveNewPocket($event)\"\n  [pocketImageList]=\"configService.pocketImageList\"\n  [journeyCurrency]=\"configService.journeyCurrency\"\n  [awaitingCreateResponse]=\"awaitingCreateResponse\"\n></bb-manage-pockets-create-wrapper>", components: [{ type: i5.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i6.ManagePocketsCreateWrapperComponent, selector: "bb-manage-pockets-create-wrapper", inputs: ["pocketImageList", "awaitingCreateResponse"], outputs: ["newPocket"] }], directives: [{ type: i7.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsCreateViewComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-manage-pockets-create-view',
                    templateUrl: './manage-pockets-create-view.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.ManagePocketsService }, { type: i2.Router }, { type: i2.ActivatedRoute }, { type: i0.ChangeDetectorRef }, { type: i3.NotificationService }, { type: i4.ManagePocketsJourneyConfigService }]; } });
