// @backbase/manage-pockets-journey-ang/esm2015/src/views/manage-pockets-list-view.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../manage-pockets-common/services/manage-pockets.service";
import * as i2 from "@backbase/foundation-ang/core";
import * as i3 from "@angular/router";
import * as i4 from "../manage-pockets-journey-config.service";
import * as i5 from "@backbase/ui-ang/header";
import * as i6 from "../manage-pockets-list/components/manage-pockets-list-wrapper/manage-pockets-list-wrapper.component";
import * as i7 from "@backbase/ui-ang/loading-indicator";
import * as i8 from "@angular/common";
export class ManagePocketsListViewComponent {
    constructor(managePocketsService, assetsService, router, route, configService) {
        this.managePocketsService = managePocketsService;
        this.assetsService = assetsService;
        this.router = router;
        this.route = route;
        this.configService = configService;
        /**
         * Api Error status observable. Will return `true` is API fails for any reason
         */
        this.apiErrorStatus$ = this.managePocketsService.pocketListApiError$;
        this.resolveAsset = this.assetsService.createAssetsResolver({
            widgetName: 'bb-manage-pockets-journey-ang',
            getURIFromAssetID: (assetId) => assetId,
        });
    }
    ngOnInit() {
        this.refreshPocketList();
    }
    refreshPocketList() {
        this.pocketList$ = this.managePocketsService.retrievePocketList();
    }
    goToCreatePocket() {
        this.router.navigate(['../create'], { relativeTo: this.route });
    }
    goToPocketTransfer(pocketTransferItem) {
        this.router.navigate(['../transfer'], { state: pocketTransferItem, relativeTo: this.route });
    }
    get imageEmptyStateUrl() {
        return this.resolveAsset('manage-pockets-empty-state-img.webp');
    }
}
ManagePocketsListViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsListViewComponent, deps: [{ token: i1.ManagePocketsService }, { token: i2.AssetsService }, { token: i3.Router }, { token: i3.ActivatedRoute }, { token: i4.ManagePocketsJourneyConfigService }], target: i0.ɵɵFactoryTarget.Component });
ManagePocketsListViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ManagePocketsListViewComponent, selector: "bb-manage-pockets-list-view", ngImport: i0, template: "<div class=\"heading-widget bb-heading-widget px-0\" *ngIf=\"configService.displayHeaders\">\n  <bb-header-ui\n    headingType=\"h1\"\n    headingClasses=\"bb-heading-widget__heading\"\n    heading=\"Pockets\"\n    i18n-heading=\"Pockets heading@@managePockets.listView.heading.title\"\n  ></bb-header-ui>\n</div>\n\n<bb-manage-pockets-list-wrapper\n  *ngIf=\"pocketList$ | async as pocketList; else loadingState\"\n  [pocketList]=\"pocketList\"\n  [pocketListApiError]=\"apiErrorStatus$ | async\"\n  [emptyStateImageUrl]=\"imageEmptyStateUrl | async\"\n  (createPocket)=\"goToCreatePocket()\"\n  (pocketTransfer)=\"goToPocketTransfer($event)\"\n  (refreshData)=\"refreshPocketList()\"\n></bb-manage-pockets-list-wrapper>\n\n<ng-template #loadingState>\n  <bb-loading-indicator-ui \n    loaderSize=\"lg\" \n    text=\"Loading your pockets...\"\n    i18n-text=\"Loading your pockets...|Text for the loading indicator@@managePockets.listView.loading.indicator.text\"\n  ></bb-loading-indicator-ui>\n</ng-template>", components: [{ type: i5.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i6.ManagePocketsListWrapperComponent, selector: "bb-manage-pockets-list-wrapper", inputs: ["pocketList", "pocketListApiError", "emptyStateImageUrl"], outputs: ["refreshData", "createPocket", "pocketTransfer"] }, { type: i7.LoadingIndicatorComponent, selector: "bb-loading-indicator-ui", inputs: ["text", "loaderSize", "showDelay", "hasBackground", "inline"] }], directives: [{ type: i8.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "async": i8.AsyncPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsListViewComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-manage-pockets-list-view',
                    templateUrl: './manage-pockets-list-view.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.ManagePocketsService }, { type: i2.AssetsService }, { type: i3.Router }, { type: i3.ActivatedRoute }, { type: i4.ManagePocketsJourneyConfigService }]; } });
