// @backbase/accounts-transactions-journeys-shared-ui-transform-ang/esm2015/lib/to-number-pipe/to-number.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ToNumberPipe } from './to-number.pipe';
import * as i0 from "@angular/core";
const pipes = [ToNumberPipe];
export class ToNumberPipeModule {
}
/** @nocollapse */ ToNumberPipeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ToNumberPipeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ ToNumberPipeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ToNumberPipeModule, declarations: [ToNumberPipe], imports: [CommonModule], exports: [ToNumberPipe] });
/** @nocollapse */ ToNumberPipeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ToNumberPipeModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ToNumberPipeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [...pipes],
                    exports: [...pipes],
                }]
        }] });
