// @backbase/income-spending-analysis-journey-ang/bundles/backbase-income-spending-analysis-journey-ang.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core'), require('@angular/router'), require('@backbase/foundation-ang/core'), require('@backbase/income-spending-analysis-widget-ang'), require('@backbase/product-summary-account-selector-widget-ang'), require('@backbase/ui-ang/header')) :
    typeof define === 'function' && define.amd ? define('@backbase/income-spending-analysis-journey-ang', ['exports', '@angular/common', '@angular/core', '@angular/router', '@backbase/foundation-ang/core', '@backbase/income-spending-analysis-widget-ang', '@backbase/product-summary-account-selector-widget-ang', '@backbase/ui-ang/header'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["income-spending-analysis-journey-ang"] = {}), global.ng.common, global.ng.core, global.ng.router, global.i1$1, global.i5, global.i4, global.i3));
})(this, (function (exports, i6, i0, i1, i1$1, i5, i4, i3) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i6__namespace = /*#__PURE__*/_interopNamespace(i6);
    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i1__namespace$1 = /*#__PURE__*/_interopNamespace(i1$1);
    var i5__namespace = /*#__PURE__*/_interopNamespace(i5);
    var i4__namespace = /*#__PURE__*/_interopNamespace(i4);
    var i3__namespace = /*#__PURE__*/_interopNamespace(i3);

    exports.AnalysisType = void 0;
    (function (AnalysisType) {
        AnalysisType["INCOME"] = "CRDT";
        AnalysisType["SPENDING"] = "DBIT";
    })(exports.AnalysisType || (exports.AnalysisType = {}));
    var IncomeSpendingAnalysisJourneyConfigurationToken = new i0.InjectionToken('IncomeSpendingAnalysisJourneyConfiguration injection token');
    var configDefaults = {
        analysisType: exports.AnalysisType.SPENDING,
    };
    var IncomeSpendingAnalysisJourneyService = /** @class */ (function () {
        function IncomeSpendingAnalysisJourneyService(config) {
            config = config || {};
            this._config = Object.assign(Object.assign({}, configDefaults), config);
        }
        Object.defineProperty(IncomeSpendingAnalysisJourneyService.prototype, "defaults", {
            get: function () {
                return configDefaults;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(IncomeSpendingAnalysisJourneyService.prototype, "analysisType", {
            get: function () {
                return this._config.analysisType;
            },
            enumerable: false,
            configurable: true
        });
        return IncomeSpendingAnalysisJourneyService;
    }());
    IncomeSpendingAnalysisJourneyService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: IncomeSpendingAnalysisJourneyService, deps: [{ token: IncomeSpendingAnalysisJourneyConfigurationToken, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    IncomeSpendingAnalysisJourneyService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: IncomeSpendingAnalysisJourneyService });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: IncomeSpendingAnalysisJourneyService, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () {
            return [{ type: undefined, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.Inject,
                            args: [IncomeSpendingAnalysisJourneyConfigurationToken]
                        }] }];
        } });

    var IncomeSpendingAnalysisJourneyComponent = /** @class */ (function () {
        function IncomeSpendingAnalysisJourneyComponent() {
        }
        return IncomeSpendingAnalysisJourneyComponent;
    }());
    IncomeSpendingAnalysisJourneyComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: IncomeSpendingAnalysisJourneyComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    IncomeSpendingAnalysisJourneyComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: IncomeSpendingAnalysisJourneyComponent, selector: "bb-income-spending-analysis-journey", providers: [IncomeSpendingAnalysisJourneyService], ngImport: i0__namespace, template: '<router-outlet></router-outlet>', isInline: true, directives: [{ type: i1__namespace.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: IncomeSpendingAnalysisJourneyComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-income-spending-analysis-journey',
                        template: '<router-outlet></router-outlet>',
                        providers: [IncomeSpendingAnalysisJourneyService],
                    }]
            }] });

    var IncomeSpendingAnalysisViewComponent = /** @class */ (function () {
        function IncomeSpendingAnalysisViewComponent(router, route, config) {
            this.router = router;
            this.route = route;
            this.config = config;
            this.isSpendingAnalysis = this.config.analysisType === exports.AnalysisType.SPENDING;
        }
        IncomeSpendingAnalysisViewComponent.prototype.onItemSelect = function (selectedAccount) {
            this.router.navigate(['.', { selectedAccount: selectedAccount }], { relativeTo: this.route });
        };
        return IncomeSpendingAnalysisViewComponent;
    }());
    IncomeSpendingAnalysisViewComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: IncomeSpendingAnalysisViewComponent, deps: [{ token: i1__namespace.Router }, { token: i1__namespace.ActivatedRoute }, { token: IncomeSpendingAnalysisJourneyService }], target: i0__namespace.ɵɵFactoryTarget.Component });
    IncomeSpendingAnalysisViewComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: IncomeSpendingAnalysisViewComponent, selector: "bb-income-spending-analysis-view", ngImport: i0__namespace, template: "\n    <bb-header-ui\n      *ngIf=\"isSpendingAnalysis\"\n      headingType=\"h1\"\n      headingClasses=\"bb-heading-widget__heading px-0 pb-5\"\n      heading=\"Spending Analysis\"\n      i18n-heading=\"Spending analysis|Heading for Spending analysis journey@@spending.analysis.journey.heading.text\"\n    ></bb-header-ui>\n    <bb-header-ui\n      *ngIf=\"!isSpendingAnalysis\"\n      headingType=\"h1\"\n      headingClasses=\"bb-heading-widget__heading px-0 pb-5\"\n      heading=\"Income Analysis\"\n      i18n-heading=\"Income analysis|Heading for Income analysis journey@@income.analysis.journey.heading.text\"\n    ></bb-header-ui>\n    <div class=\"bb-block--xl\">\n      <bb-product-summary-account-selector-widget\n        (selectedAccount)=\"onItemSelect($event)\"\n      ></bb-product-summary-account-selector-widget>\n    </div>\n    <div>\n      <bb-income-spending-analysis-widget [analysisType]=\"config.analysisType\"></bb-income-spending-analysis-widget>\n    </div>\n  ", isInline: true, components: [{ type: i3__namespace.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i4__namespace.ProductSummaryAccountSelectorWidgetComponent, selector: "bb-product-summary-account-selector-widget", inputs: ["event.refreshAccountsList", "multiple", "filterItems", "showAccountState"], outputs: ["selectedAccount", "selectedKind"] }, { type: i5__namespace.IncomeSpendingAnalysisWidgetComponent, selector: "bb-income-spending-analysis-widget", inputs: ["analysisType"] }], directives: [{ type: i6__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: IncomeSpendingAnalysisViewComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-income-spending-analysis-view',
                        template: "\n    <bb-header-ui\n      *ngIf=\"isSpendingAnalysis\"\n      headingType=\"h1\"\n      headingClasses=\"bb-heading-widget__heading px-0 pb-5\"\n      heading=\"Spending Analysis\"\n      i18n-heading=\"Spending analysis|Heading for Spending analysis journey@@spending.analysis.journey.heading.text\"\n    ></bb-header-ui>\n    <bb-header-ui\n      *ngIf=\"!isSpendingAnalysis\"\n      headingType=\"h1\"\n      headingClasses=\"bb-heading-widget__heading px-0 pb-5\"\n      heading=\"Income Analysis\"\n      i18n-heading=\"Income analysis|Heading for Income analysis journey@@income.analysis.journey.heading.text\"\n    ></bb-header-ui>\n    <div class=\"bb-block--xl\">\n      <bb-product-summary-account-selector-widget\n        (selectedAccount)=\"onItemSelect($event)\"\n      ></bb-product-summary-account-selector-widget>\n    </div>\n    <div>\n      <bb-income-spending-analysis-widget [analysisType]=\"config.analysisType\"></bb-income-spending-analysis-widget>\n    </div>\n  ",
                    }]
            }], ctorParameters: function () { return [{ type: i1__namespace.Router }, { type: i1__namespace.ActivatedRoute }, { type: IncomeSpendingAnalysisJourneyService }]; } });

    var defaultRoute = {
        path: '',
        component: IncomeSpendingAnalysisJourneyComponent,
        children: [
            {
                path: '',
                component: IncomeSpendingAnalysisViewComponent,
            },
        ],
    };
    var IncomeSpendingAnalysisJourneyModule = /** @class */ (function () {
        function IncomeSpendingAnalysisJourneyModule() {
        }
        IncomeSpendingAnalysisJourneyModule.forRoot = function (data) {
            if (data === void 0) { data = { route: defaultRoute }; }
            return {
                ngModule: IncomeSpendingAnalysisJourneyModule,
                providers: [i1.provideRoutes([data.route])],
            };
        };
        return IncomeSpendingAnalysisJourneyModule;
    }());
    IncomeSpendingAnalysisJourneyModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: IncomeSpendingAnalysisJourneyModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    IncomeSpendingAnalysisJourneyModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: IncomeSpendingAnalysisJourneyModule, declarations: [IncomeSpendingAnalysisJourneyComponent, IncomeSpendingAnalysisViewComponent], imports: [i6.CommonModule, i1__namespace$1.BackbaseCoreModule, i1.RouterModule,
            i3.HeaderModule,
            i4.ProductSummaryAccountSelectorWidgetModule,
            i5.IncomeSpendingAnalysisWidgetModule] });
    IncomeSpendingAnalysisJourneyModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: IncomeSpendingAnalysisJourneyModule, imports: [[
                i6.CommonModule,
                i1$1.BackbaseCoreModule.withConfig({
                    classMap: { IncomeSpendingAnalysisJourneyComponent: IncomeSpendingAnalysisJourneyComponent },
                }),
                i1.RouterModule,
                i3.HeaderModule,
                i4.ProductSummaryAccountSelectorWidgetModule,
                i5.IncomeSpendingAnalysisWidgetModule,
            ]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: IncomeSpendingAnalysisJourneyModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [IncomeSpendingAnalysisJourneyComponent, IncomeSpendingAnalysisViewComponent],
                        imports: [
                            i6.CommonModule,
                            i1$1.BackbaseCoreModule.withConfig({
                                classMap: { IncomeSpendingAnalysisJourneyComponent: IncomeSpendingAnalysisJourneyComponent },
                            }),
                            i1.RouterModule,
                            i3.HeaderModule,
                            i4.ProductSummaryAccountSelectorWidgetModule,
                            i5.IncomeSpendingAnalysisWidgetModule,
                        ],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.IncomeSpendingAnalysisJourneyComponent = IncomeSpendingAnalysisJourneyComponent;
    exports.IncomeSpendingAnalysisJourneyConfigurationToken = IncomeSpendingAnalysisJourneyConfigurationToken;
    exports.IncomeSpendingAnalysisJourneyModule = IncomeSpendingAnalysisJourneyModule;
    exports.IncomeSpendingAnalysisJourneyService = IncomeSpendingAnalysisJourneyService;
    exports.IncomeSpendingAnalysisViewComponent = IncomeSpendingAnalysisViewComponent;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
