// @backbase/product-summary-common-ang/esm2015/src/product-summary-common-ang.module.js
import '@angular/localize/init';
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NotificationModule } from '@backbase/ui-ang/notification';
import { ProductSummaryAccountDetailsBaseModule } from './classes/product-summary-account-details-base/product-summary-account-details-base.module';
import { ResponsiveContainerModule } from './classes/responsive-container/responsive-container.module';
import { WindowProvider } from './model/providers';
import * as i0 from "@angular/core";
const sharedModules = [NotificationModule, ProductSummaryAccountDetailsBaseModule, ResponsiveContainerModule];
export class ProductSummaryCommon {
}
ProductSummaryCommon.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryCommon, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductSummaryCommon.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryCommon, imports: [CommonModule, NotificationModule, ProductSummaryAccountDetailsBaseModule, ResponsiveContainerModule], exports: [NotificationModule, ProductSummaryAccountDetailsBaseModule, ResponsiveContainerModule] });
ProductSummaryCommon.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryCommon, providers: [WindowProvider], imports: [[CommonModule, ...sharedModules], NotificationModule, ProductSummaryAccountDetailsBaseModule, ResponsiveContainerModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryCommon, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...sharedModules],
                    exports: [...sharedModules],
                    providers: [WindowProvider],
                }]
        }] });
