// @backbase/places-journey-ang/esm2015/lib/components/work-times/work-times.component.js
import { Component, Input, ChangeDetectionStrategy } from '@angular/core';
import { DatePipe } from '@angular/common';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class WorkTimesComponent {
    constructor(datePipe) {
        this.datePipe = datePipe;
        /**
         * The day number based on WeekDay enum
         */
        this.day = 0;
        /**
         * Hours related to this day
         */
        this.hours = [];
    }
    toLocaleTime(hour) {
        return (this.datePipe.transform(new Date('2000-01-01T' + hour), 'shortTime') ||
            hour);
    }
    isToday(day, today = new Date().getDay()) {
        today = today || 7;
        return day === today;
    }
}
WorkTimesComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: WorkTimesComponent, deps: [{ token: i1.DatePipe }], target: i0.ɵɵFactoryTarget.Component });
WorkTimesComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: WorkTimesComponent, selector: "bb-work-times", inputs: { day: "day", hours: "hours" }, providers: [DatePipe], ngImport: i0, template: "<div\n  class=\"row bb-place-timetable\"\n  [ngClass]=\"{\n    'bb-text-semi-bold': this.isToday(day),\n    'bb-text-support': !this.isToday(day)\n  }\"\n>\n  <span\n    class=\"col-md-5\"\n    i18n=\"\n      Week day name|label for week day name based on work\n      times@@places.workTimes.dayName\"\n  >\n    { day, select, 1 {Monday} 2 {Tuesday} 3 {Wednesday} 4 {Thursday} 5 {Friday}\n    6 {Saturday} 7 {Sunday} }\n  </span>\n  <span class=\"col-md-7\">\n    <ng-container *ngIf=\"this.hours.length; else closed\">\n      <ng-container *ngFor=\"let workHour of this.hours; let i = index\"\n        >{{ this.toLocaleTime(workHour.open) }} -\n        {{ this.toLocaleTime(workHour.close) }}<br\n      /></ng-container>\n    </ng-container>\n  </span>\n</div>\n\n<ng-template #closed>\n  <span i18n=\"Closed|label for closed day@@places.workTimes.closed\"\n    >Closed</span\n  >\n</ng-template>\n", directives: [{ type: i1.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i1.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: WorkTimesComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-work-times',
                    templateUrl: './work-times.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [DatePipe],
                }]
        }], ctorParameters: function () { return [{ type: i1.DatePipe }]; }, propDecorators: { day: [{
                type: Input
            }], hours: [{
                type: Input
            }] } });
