// @backbase/contact-manager-journey-ang/esm2015/lib/views/contact-manager-detail/contact-manager-detail.component.js
import { ChangeDetectionStrategy, Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/contact-manager-widget-ang";
/**
 * The wrappper component will show detail of selected contact
 *
 * @usageNotes
 *
 * ### Display contact detail
 *
 * ```route
 * export const routes: Routes = [
 *   {
 *     path: '',
 *     component: ContactManagerJourneyComponent,
 *     children: [
 *       {
 *         path: '',
 *         component: ContactManagerDetailViewComponent,
 *       },
 *       {
 *         path: 'select',
 *         component: ContactManagerTypeListViewComponent,
 *       },
 *       {
 *         path: ':id',
 *         component: ContactManagerDetailViewComponent,
 *       },
 *       {
 *         path: 'edit/:id/:type',
 *         canDeactivate: [CanDeactivateGuard],
 *         component: ContactManagerFormViewComponent,
 *       },
 *     ],
 *   },
 * ];
 * };
 * @NgModule({
 *   imports: [ContactManagerJourneyModule.forRoot({ routes })],
 * })
 * export class ContactManagerJourneyBundleModule {}
 * ```
 * @ngModule ContactManagerJourneyModule
 */
export class ContactManagerDetailViewComponent {
}
/** @nocollapse */ ContactManagerDetailViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactManagerDetailViewComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ ContactManagerDetailViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ContactManagerDetailViewComponent, selector: "bb-contact-manager-detail-view", ngImport: i0, template: "<bb-contacts-detail-container></bb-contacts-detail-container>\n", components: [{ type: i1.ContactsDetailContainerComponent, selector: "bb-contacts-detail-container" }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactManagerDetailViewComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-contact-manager-detail-view',
                    templateUrl: './contact-manager-detail.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }] });
