// @backbase/ui-ang/esm2015/input-number/digit-only.directive.js
import { Directive, HostListener, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class DigitOnlyDirective {
    constructor(el) {
        this.el = el;
        this.navigationKeys = [
            'Backspace',
            'Delete',
            'Tab',
            'Escape',
            'Enter',
            'Home',
            'End',
            'ArrowLeft',
            'ArrowRight',
            'Clear',
            'Copy',
            'Paste',
        ];
        this.isEnabled = true;
        this.allowNegativeSign = false;
        this.inputElement = el.nativeElement;
    }
    /* eslint-disable complexity */
    onKeyDown(e) {
        const array = ['KeyA', 'KeyC', 'KeyV', 'KeyX'];
        const matchPattern = this.allowNegativeSign ? /^-|\d$/ : /^\d$/;
        if (!this.isEnabled ||
            this.navigationKeys.indexOf(e.key) > -1 ||
            (array.includes(e.code) && (e.ctrlKey || e.metaKey))) {
            this.previousKey = e.key;
            return;
        }
        const preventEventDefault = 
        // it is other than pattern
        !matchPattern.test(e.key) ||
            // try to add negative when there is already a value
            (e.key === '-' && this.inputElement.value.indexOf('-') > -1) ||
            // try to add negative sign when you hit negative sign just before
            (e.key === '-' && this.previousKey === '-');
        if (!this.inputElement.value && this.previousKey !== '-') {
            this.inputElement.value = '';
        }
        this.previousKey = e.key;
        if (preventEventDefault) {
            e.preventDefault();
        }
    }
}
DigitOnlyDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DigitOnlyDirective, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive });
DigitOnlyDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: DigitOnlyDirective, selector: "[bbDigitOnly]", inputs: { isEnabled: ["bbDigitOnly", "isEnabled"], allowNegativeSign: "allowNegativeSign" }, host: { listeners: { "keydown": "onKeyDown($event)" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DigitOnlyDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbDigitOnly]',
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { isEnabled: [{
                type: Input,
                args: ['bbDigitOnly']
            }], allowNegativeSign: [{
                type: Input
            }], onKeyDown: [{
                type: HostListener,
                args: ['keydown', ['$event']]
            }] } });
