// @backbase/ui-ang/esm2015/input-number/input-number.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { BaseClassesModule } from '@backbase/ui-ang/base-classes';
import { InputNumberComponent } from './input-number.component';
import { FocusModule } from '@backbase/ui-ang/focus';
import { DigitOnlyDirective } from './digit-only.directive';
import * as i0 from "@angular/core";
export class InputNumberModule {
}
InputNumberModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputNumberModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputNumberModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputNumberModule, declarations: [InputNumberComponent, DigitOnlyDirective], imports: [CommonModule, FormsModule, BaseClassesModule, FocusModule], exports: [InputNumberComponent] });
InputNumberModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputNumberModule, imports: [[CommonModule, FormsModule, BaseClassesModule, FocusModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputNumberModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, FormsModule, BaseClassesModule, FocusModule],
                    declarations: [InputNumberComponent, DigitOnlyDirective],
                    exports: [InputNumberComponent],
                }]
        }] });
