// @backbase/loans-collection-ui-ang/esm2015/lib/loan-info/loan-info-custom-section/loan-info-custom-section.directive.js
import { ComponentFactoryResolver, Directive, Input, Type, ViewContainerRef } from '@angular/core';
import * as i0 from "@angular/core";
export class LoanInfoCustomSectionDirective {
    constructor(viewContainerRef, componentFactoryResolver) {
        this.viewContainerRef = viewContainerRef;
        this.componentFactoryResolver = componentFactoryResolver;
    }
    ngOnInit() {
        const componentFactory = this.componentFactoryResolver.resolveComponentFactory(this.componentRef);
        this.viewContainerRef.clear();
        const componentRef = this.viewContainerRef.createComponent(componentFactory);
        componentRef.instance.loan = this.loan;
    }
}
/** @nocollapse */ LoanInfoCustomSectionDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoCustomSectionDirective, deps: [{ token: i0.ViewContainerRef }, { token: i0.ComponentFactoryResolver }], target: i0.ɵɵFactoryTarget.Directive });
/** @nocollapse */ LoanInfoCustomSectionDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: LoanInfoCustomSectionDirective, selector: "[bbLoanInfoCustomSection]", inputs: { componentRef: "componentRef", loan: "loan" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoCustomSectionDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbLoanInfoCustomSection]',
                }]
        }], ctorParameters: function () { return [{ type: i0.ViewContainerRef }, { type: i0.ComponentFactoryResolver }]; }, propDecorators: { componentRef: [{
                type: Input
            }], loan: [{
                type: Input
            }] } });
