// @backbase/ui-ang/esm2015/_example/account-number-pipe/examples/account-number-pipe.example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/account-number-pipe";
export class AccountNumberPipeExampleComponent {
    constructor() {
        this.productNumber = 'IBAN1234567891234567';
        this.numberType = 'iban';
    }
}
AccountNumberPipeExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountNumberPipeExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
AccountNumberPipeExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AccountNumberPipeExampleComponent, selector: "bb-account-number-pipe-example-ui", ngImport: i0, template: "<p>{{ productNumber | bbAccountNumber: numberType }}</p>\n", pipes: { "bbAccountNumber": i1.AccountNumberPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountNumberPipeExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-account-number-pipe-example-ui',
                    templateUrl: './account-number-pipe.example.component.html',
                }]
        }] });
