// @backbase/account-statement-business-widget-ang/esm2015/src/components/account-statement-view-online-business-wrapper/account-statement-view-online-business-wrapper.component.js
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/account-statement-common-ang";
import * as i2 from "@angular/common";
export class AccountStatementViewOnlineBusinessWrapperComponent {
    constructor() {
        this.describedByElementId = '';
    }
    set documents(documents) {
        this.documentPDF = documents.find(document => document.contentType === 'application/pdf');
    }
}
AccountStatementViewOnlineBusinessWrapperComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementViewOnlineBusinessWrapperComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
AccountStatementViewOnlineBusinessWrapperComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: AccountStatementViewOnlineBusinessWrapperComponent, selector: "bb-account-statement-view-online-business-wrapper", inputs: { documents: "documents", describedByElementId: "describedByElementId" }, ngImport: i0, template: "<bb-account-statement-view-online\n  *ngIf=\"documentPDF\"\n  [href]=\"documentPDF.url!\"\n  [describedByElementId]=\"describedByElementId\"\n></bb-account-statement-view-online>\n", components: [{ type: i1.ӨAccountStatementViewOnlineComponent, selector: "bb-account-statement-view-online", inputs: ["href", "describedByElementId"] }], directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementViewOnlineBusinessWrapperComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-account-statement-view-online-business-wrapper',
                    templateUrl: './account-statement-view-online-business-wrapper.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { documents: [{
                type: Input
            }], describedByElementId: [{
                type: Input
            }] } });
