// @backbase/select-context-widget-ang/esm2015/lib/select-context-widget.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BackbaseCoreModule, FutureFeaturesModule } from '@backbase/foundation-ang/core';
import { AccessgroupUiModule } from '@backbase/accessgroup-ui';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { HeaderModule } from '@backbase/ui-ang/header';
import { IconModule } from '@backbase/ui-ang/icon';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { SearchBoxModule } from '@backbase/ui-ang/search-box';
import { AlertModule } from '@backbase/ui-ang/alert';
import { ButtonModule } from '@backbase/ui-ang/button';
import { InfiniteScrollModule } from '@backbase/ui-ang/infinite-scroll';
import { DefaultSelectContextRedirectService, SELECT_CONTEXT_URL_PROCESSOR_SERVICE_TOKEN, } from './select-context-widget/services/default-select-context-redirect.service';
import { SelectContextContainerComponent, SelectContextContainerCustomizableDirective, SelectContextContainerLoadingCustomizableDirective, SelectContextContainerSavingCustomizableDirective, SELECT_CONTEXT_INNER_CONTAINER, } from './select-context-widget/select-context-container/select-context-container.component';
import { SelectContextServiceAgreementListCustomizableDirective, ServiceAgreementListComponent, } from './select-context-widget/service-agreement-list/service-agreement-list.component';
import { SelectContextErrorComponent } from './select-context-widget/select-context-error/select-context-error.component';
import { SelectContextServiceAgreementItemCustomizableDirective, ServiceAgreementItemComponent, } from './select-context-widget/service-agreement-item/service-agreement-item.component';
import { SelectContextWidgetComponent } from './select-context-widget.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const components = [
    SelectContextWidgetComponent,
    ServiceAgreementItemComponent,
    SelectContextErrorComponent,
    ServiceAgreementListComponent,
    SelectContextContainerComponent,
];
const directives = [
    SelectContextServiceAgreementListCustomizableDirective,
    SelectContextServiceAgreementItemCustomizableDirective,
    SelectContextContainerCustomizableDirective,
    SelectContextContainerLoadingCustomizableDirective,
    SelectContextContainerSavingCustomizableDirective,
];
const uiModules = [
    EmptyStateModule,
    HeaderModule,
    IconModule,
    LoadingIndicatorModule,
    SearchBoxModule,
    AlertModule,
    ButtonModule,
    InfiniteScrollModule,
    AccessgroupUiModule,
];
export class SelectContextWidgetModule {
}
/** @nocollapse */ SelectContextWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SelectContextWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ SelectContextWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SelectContextWidgetModule, declarations: [SelectContextWidgetComponent,
        ServiceAgreementItemComponent,
        SelectContextErrorComponent,
        ServiceAgreementListComponent,
        SelectContextContainerComponent, SelectContextServiceAgreementListCustomizableDirective,
        SelectContextServiceAgreementItemCustomizableDirective,
        SelectContextContainerCustomizableDirective,
        SelectContextContainerLoadingCustomizableDirective,
        SelectContextContainerSavingCustomizableDirective], imports: [CommonModule, i1.BackbaseCoreModule, i1.FutureFeaturesModule, EmptyStateModule,
        HeaderModule,
        IconModule,
        LoadingIndicatorModule,
        SearchBoxModule,
        AlertModule,
        ButtonModule,
        InfiniteScrollModule,
        AccessgroupUiModule], exports: [SelectContextWidgetComponent,
        ServiceAgreementItemComponent,
        SelectContextErrorComponent,
        ServiceAgreementListComponent,
        SelectContextContainerComponent, SelectContextServiceAgreementListCustomizableDirective,
        SelectContextServiceAgreementItemCustomizableDirective,
        SelectContextContainerCustomizableDirective,
        SelectContextContainerLoadingCustomizableDirective,
        SelectContextContainerSavingCustomizableDirective, EmptyStateModule,
        HeaderModule,
        IconModule,
        LoadingIndicatorModule,
        SearchBoxModule,
        AlertModule,
        ButtonModule,
        InfiniteScrollModule,
        AccessgroupUiModule] });
/** @nocollapse */ SelectContextWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SelectContextWidgetModule, providers: [
        {
            provide: SELECT_CONTEXT_URL_PROCESSOR_SERVICE_TOKEN,
            useClass: DefaultSelectContextRedirectService,
        },
    ], imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { SelectContextWidgetComponent },
            }),
            FutureFeaturesModule.allowOptIn([SELECT_CONTEXT_INNER_CONTAINER]),
            ...uiModules,
        ], EmptyStateModule,
        HeaderModule,
        IconModule,
        LoadingIndicatorModule,
        SearchBoxModule,
        AlertModule,
        ButtonModule,
        InfiniteScrollModule,
        AccessgroupUiModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SelectContextWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { SelectContextWidgetComponent },
                        }),
                        FutureFeaturesModule.allowOptIn([SELECT_CONTEXT_INNER_CONTAINER]),
                        ...uiModules,
                    ],
                    providers: [
                        {
                            provide: SELECT_CONTEXT_URL_PROCESSOR_SERVICE_TOKEN,
                            useClass: DefaultSelectContextRedirectService,
                        },
                    ],
                    declarations: [...components, ...directives],
                    exports: [...components, ...directives, ...uiModules],
                }]
        }] });
