// @backbase/places-http-ang/esm2015/api/places.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { PLACES_BASE_PATH } from '../variables';
import { PlacesConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class PlacesHttpService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new PlacesConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    getPlaces(requestParameters, observe = 'body', reportProgress = false, options) {
        const _latitude = requestParameters["latitude"];
        if (_latitude === null || _latitude === undefined) {
            throw new Error('Required parameter latitude was null or undefined when calling getPlaces.');
        }
        const _longitude = requestParameters["longitude"];
        if (_longitude === null || _longitude === undefined) {
            throw new Error('Required parameter longitude was null or undefined when calling getPlaces.');
        }
        const _radius = requestParameters["radius"];
        if (_radius === null || _radius === undefined) {
            throw new Error('Required parameter radius was null or undefined when calling getPlaces.');
        }
        const _type = requestParameters["type"];
        const _types = requestParameters["types"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_latitude !== undefined && _latitude !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _latitude, 'latitude');
        }
        if (_longitude !== undefined && _longitude !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _longitude, 'longitude');
        }
        if (_radius !== undefined && _radius !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _radius, 'radius');
        }
        if (_type !== undefined && _type !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _type, 'type');
        }
        if (_types) {
            _types.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'types');
            });
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/places`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getPlacesUrl(requestParameters) {
        const _latitude = requestParameters["latitude"];
        if (_latitude === null || _latitude === undefined) {
            throw new Error('Required parameter latitude was null or undefined when calling getPlaces.');
        }
        const _longitude = requestParameters["longitude"];
        if (_longitude === null || _longitude === undefined) {
            throw new Error('Required parameter longitude was null or undefined when calling getPlaces.');
        }
        const _radius = requestParameters["radius"];
        if (_radius === null || _radius === undefined) {
            throw new Error('Required parameter radius was null or undefined when calling getPlaces.');
        }
        const _type = requestParameters["type"];
        const _types = requestParameters["types"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_latitude !== undefined && _latitude !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _latitude, 'latitude');
        }
        if (_longitude !== undefined && _longitude !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _longitude, 'longitude');
        }
        if (_radius !== undefined && _radius !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _radius, 'radius');
        }
        if (_type !== undefined && _type !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _type, 'type');
        }
        if (_types) {
            _types.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'types');
            });
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/places${queryString ? `?${queryString}` : ''}`;
    }
}
PlacesHttpService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PlacesHttpService, deps: [{ token: i1.HttpClient }, { token: PLACES_BASE_PATH, optional: true }, { token: i2.PlacesConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
PlacesHttpService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PlacesHttpService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PlacesHttpService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [PLACES_BASE_PATH]
                }] }, { type: i2.PlacesConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
