// @backbase/user-context-menu-widget-ang/bundles/backbase-user-context-menu-widget-ang.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('@backbase/foundation-ang/core'), require('@ng-bootstrap/ng-bootstrap'), require('@backbase/accessgroup-ui'), require('@backbase/select-context-widget-ang'), require('@backbase/ui-ang/avatar'), require('@backbase/ui-ang/icon'), require('@backbase/ui-ang/modal'), require('@backbase/ui-ang/button'), require('@backbase/ui-ang/dropdown-menu'), require('@backbase/foundation-ang/web-sdk'), require('rxjs'), require('rxjs/operators'), require('@backbase/data-ang/accesscontrol'), require('@backbase/data-ang/user')) :
    typeof define === 'function' && define.amd ? define('@backbase/user-context-menu-widget-ang', ['exports', '@angular/core', '@angular/common', '@backbase/foundation-ang/core', '@ng-bootstrap/ng-bootstrap', '@backbase/accessgroup-ui', '@backbase/select-context-widget-ang', '@backbase/ui-ang/avatar', '@backbase/ui-ang/icon', '@backbase/ui-ang/modal', '@backbase/ui-ang/button', '@backbase/ui-ang/dropdown-menu', '@backbase/foundation-ang/web-sdk', 'rxjs', 'rxjs/operators', '@backbase/data-ang/accesscontrol', '@backbase/data-ang/user'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["user-context-menu-widget-ang"] = {}), global.ng.core, global.ng.common, global.i1$1, global.ngBootstrap, global.i1$2, global.i8, global.i1, global.i5, global.i7, global.i10, global.i3, global.webSdk, global.rxjs, global.rxjs.operators, global.i1$3, global.i2$1));
})(this, (function (exports, i0, i2, i1$1, ngBootstrap, i1$2, i8, i1, i5, i7, i10, i3, webSdk, rxjs, operators, i1$3, i2$1) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i1__namespace$3 = /*#__PURE__*/_interopNamespace(i1$1);
    var i1__namespace$1 = /*#__PURE__*/_interopNamespace(i1$2);
    var i8__namespace = /*#__PURE__*/_interopNamespace(i8);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i5__namespace = /*#__PURE__*/_interopNamespace(i5);
    var i7__namespace = /*#__PURE__*/_interopNamespace(i7);
    var i10__namespace = /*#__PURE__*/_interopNamespace(i10);
    var i3__namespace = /*#__PURE__*/_interopNamespace(i3);
    var i1__namespace$2 = /*#__PURE__*/_interopNamespace(i1$3);
    var i2__namespace$1 = /*#__PURE__*/_interopNamespace(i2$1);

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from, pack) {
        if (pack || arguments.length === 2)
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar)
                        ar = Array.prototype.slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
        return to.concat(ar || Array.prototype.slice.call(from));
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    var UserContextMenuInfoComponent = /** @class */ (function () {
        function UserContextMenuInfoComponent() {
            this.hostRef = this;
        }
        return UserContextMenuInfoComponent;
    }());
    /** @nocollapse */ UserContextMenuInfoComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: UserContextMenuInfoComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    /** @nocollapse */ UserContextMenuInfoComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: UserContextMenuInfoComponent, selector: "bb-user-context-menu-info", inputs: { userInfo: "userInfo", currentContext: "currentContext" }, ngImport: i0__namespace, template: "<ng-container\n  bbUserContextMenuInfoCustomizable\n  [bbHostRef]=\"hostRef\"\n  [bbTemplateContext]=\"{\n    userInfo: userInfo,\n    currentContext: currentContext\n  }\">\n</ng-container>\n\n<ng-template\n  bbUserContextMenuInfoCustomizable\n  let-hostRef\n  let-userInfo=\"context.userInfo\"\n  let-currentContext=\"context.currentContext\">\n  <div class=\"user-context-menu-info\">\n    <bb-avatar-ui *ngIf=\"userInfo\" [name]=\"userInfo?.fullName\" class=\"user-context-menu-info__avatar\"></bb-avatar-ui>\n    <div class=\"user-context-menu-info__description\">\n      <div *ngIf=\"userInfo\" class=\"user-context-menu-info__name\" data-role=\"user-full-name\">\n        {{ userInfo?.fullName }}\n      </div>\n      <div class=\"user-context-menu-info__context\" data-role=\"current-context\">\n        {{ currentContext?.name }}\n      </div>\n    </div>\n  </div>\n</ng-template>\n", components: [{ type: i0__namespace.forwardRef(function () { return i1__namespace.AvatarComponent; }), selector: "bb-avatar-ui", inputs: ["image", "name", "imgAlt", "size"] }], directives: [{ type: i0__namespace.forwardRef(function () { return UserContextMenuInfoCustomizableDirective; }), selector: "[bbUserContextMenuInfoCustomizable]" }, { type: i0__namespace.forwardRef(function () { return i2__namespace.NgIf; }), selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: UserContextMenuInfoComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-user-context-menu-info',
                        templateUrl: './user-context-menu-info.component.html',
                    }]
            }], propDecorators: { userInfo: [{
                    type: i0.Input
                }], currentContext: [{
                    type: i0.Input
                }] } });
    var UserContextMenuInfoCustomizableDirective = /** @class */ (function (_super) {
        __extends(UserContextMenuInfoCustomizableDirective, _super);
        function UserContextMenuInfoCustomizableDirective() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return UserContextMenuInfoCustomizableDirective;
    }(i1$1.BbTemplate));
    /** @nocollapse */ UserContextMenuInfoCustomizableDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: UserContextMenuInfoCustomizableDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    /** @nocollapse */ UserContextMenuInfoCustomizableDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: UserContextMenuInfoCustomizableDirective, selector: "[bbUserContextMenuInfoCustomizable]", usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: UserContextMenuInfoCustomizableDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: '[bbUserContextMenuInfoCustomizable]',
                    }]
            }] });

    var UserContextMenuDropdownComponent = /** @class */ (function () {
        function UserContextMenuDropdownComponent() {
            this.hostRef = this;
            this.canSelectContext = false;
            this.changeContext = new i0.EventEmitter();
            this.logout = new i0.EventEmitter();
        }
        UserContextMenuDropdownComponent.prototype.onChangeContext = function () {
            if (this.canSelectContext) {
                this.changeContext.emit();
            }
        };
        UserContextMenuDropdownComponent.prototype.onLogout = function () {
            this.logout.emit();
        };
        return UserContextMenuDropdownComponent;
    }());
    /** @nocollapse */ UserContextMenuDropdownComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: UserContextMenuDropdownComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    /** @nocollapse */ UserContextMenuDropdownComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: UserContextMenuDropdownComponent, selector: "bb-user-context-menu-dropdown", inputs: { currentContext: "currentContext", showSingleContext: "showSingleContext", canSelectContext: "canSelectContext" }, outputs: { changeContext: "changeContext", logout: "logout" }, ngImport: i0__namespace, template: "<ng-container\n  bbUserContextMenuDropdownCustomizable\n  [bbHostRef]=\"hostRef\"\n  [bbTemplateContext]=\"{\n    currentContext: currentContext,\n    canSelectContext: canSelectContext,\n    showSingleContext: showSingleContext\n  }\">\n</ng-container>\n\n<ng-template\n  bbUserContextMenuDropdownCustomizable\n  let-hostRef\n  let-currentContext=\"context.currentContext\"\n  let-canSelectContext=\"context.canSelectContext\"\n  let-showSingleContext=\"context.showSingleContext\">\n  <ng-container *ngIf=\"canSelectContext || showSingleContext\">\n    <button\n      class=\"dropdown-item bb-stack\"\n      role=\"menuitem\"\n      data-role=\"user-context-menu-dropdown-change-context-btn\"\n      (click)=\"hostRef.onChangeContext()\">\n      <div class=\"bb-stack__item\">\n        <bb-user-context-icon-ui\n          [isMaster]=\"currentContext?.isMaster\"\n          [name]=\"currentContext?.name\"></bb-user-context-icon-ui>\n      </div>\n      <div class=\"bb-stack__item\">\n        <div class=\"bb-text-bold\" data-role=\"user-context\">\n          {{ currentContext?.name }}\n        </div>\n        <div\n          class=\"bb-text-support\"\n          data-role=\"change-context\"\n          *ngIf=\"canSelectContext\"\n          i18n=\"Change context|Click to change context label@@user-context-menu.label.clickToChange\">\n          (click to change context)\n        </div>\n      </div>\n    </button>\n\n    <div class=\"dropdown-divider\"></div>\n  </ng-container>\n\n  <button role=\"menuitem\" class=\"dropdown-item\" data-role=\"logout\" (click)=\"hostRef.onLogout()\">\n    <div class=\"user-context-menu-info\" i18n=\"Log out|Click to log out label@@user-context-menu.label.logout\">\n      Log out\n    </div>\n  </button>\n</ng-template>\n", components: [{ type: i0__namespace.forwardRef(function () { return i1__namespace$1.UserContextIconComponent; }), selector: "bb-user-context-icon-ui", inputs: ["isMaster", "name"] }], directives: [{ type: i0__namespace.forwardRef(function () { return UserContextMenuDropdownCustomizableDirective; }), selector: "[bbUserContextMenuDropdownCustomizable]" }, { type: i0__namespace.forwardRef(function () { return i2__namespace.NgIf; }), selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: UserContextMenuDropdownComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-user-context-menu-dropdown',
                        templateUrl: './user-context-menu-dropdown.component.html',
                    }]
            }], propDecorators: { currentContext: [{
                    type: i0.Input
                }], showSingleContext: [{
                    type: i0.Input
                }], canSelectContext: [{
                    type: i0.Input
                }], changeContext: [{
                    type: i0.Output
                }], logout: [{
                    type: i0.Output
                }] } });
    var UserContextMenuDropdownCustomizableDirective = /** @class */ (function (_super) {
        __extends(UserContextMenuDropdownCustomizableDirective, _super);
        function UserContextMenuDropdownCustomizableDirective() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return UserContextMenuDropdownCustomizableDirective;
    }(i1$1.BbTemplate));
    /** @nocollapse */ UserContextMenuDropdownCustomizableDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: UserContextMenuDropdownCustomizableDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    /** @nocollapse */ UserContextMenuDropdownCustomizableDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: UserContextMenuDropdownCustomizableDirective, selector: "[bbUserContextMenuDropdownCustomizable]", usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: UserContextMenuDropdownCustomizableDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: '[bbUserContextMenuDropdownCustomizable]',
                    }]
            }] });

    var UserContextMenuService = /** @class */ (function () {
        function UserContextMenuService(userContextService, serviceAgreementService, userService) {
            this.userContextService = userContextService;
            this.serviceAgreementService = serviceAgreementService;
            this.userService = userService;
            this.currentContext = this.serviceAgreementService
                .getServiceAgreementContext()
                .pipe(operators.catchError(function () { return rxjs.of(undefined); }));
            this.canSelectContext = this.getTotalServiceAgreements();
            this.userInfo = this.userService
                .getOwnUser()
                .pipe(operators.catchError(function () { return rxjs.of(undefined); }));
        }
        UserContextMenuService.prototype.getTotalServiceAgreements = function () {
            return this.userContextService.getUserContextServiceAgreements({ from: 0, size: 2 }, 'response').pipe(operators.map(function (response) {
                var items = response.body || [];
                return (parseInt(response.headers.get('x-total-count') || '0', 10) || items.length) > 1;
            }), operators.catchError(function () {
                return rxjs.of(false);
            }));
        };
        return UserContextMenuService;
    }());
    /** @nocollapse */ UserContextMenuService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: UserContextMenuService, deps: [{ token: i1__namespace$2.UserContextHttpService }, { token: i1__namespace$2.ServiceAgreementHttpService }, { token: i2__namespace$1.UserManagementService }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    /** @nocollapse */ UserContextMenuService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: UserContextMenuService });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: UserContextMenuService, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () { return [{ type: i1__namespace$2.UserContextHttpService }, { type: i1__namespace$2.ServiceAgreementHttpService }, { type: i2__namespace$1.UserManagementService }]; } });

    var WidgetProperty = {
        PAGE_SIZE: 'pageSize',
        INCLUDE_CURRENT_CONTEXT: 'includeCurrentContext',
        SHOW_SEARCH_AFTER: 'showSearchAfter',
        SHOW_SINGLE_CONTEXT: 'showSingleContext',
        REDIRECT_PAGE: 'selectContextRedirectPage',
    };
    var DefaultWidgetProperties = {
        pageSize: 7,
        includeCurrentContext: true,
        showSearchAfter: 7,
        selectContextRedirectPage: '',
        showSingleContext: true,
    };
    var UserContextMenuWidgetPropertiesService = /** @class */ (function () {
        function UserContextMenuWidgetPropertiesService(itemModel) {
            this.itemModel = itemModel;
            this.pageSizeSubject$ = new rxjs.BehaviorSubject(undefined);
            this.includeCurrentContextSubject$ = new rxjs.BehaviorSubject(undefined);
            this.showSearchAfterSubject$ = new rxjs.BehaviorSubject(undefined);
            this.selectContextRedirectPageSubject$ = new rxjs.BehaviorSubject(undefined);
            this.showSingleContextSubject$ = new rxjs.BehaviorSubject(undefined);
            this.pageSize = this.itemModel
                ? this.itemModel
                    .property(WidgetProperty.PAGE_SIZE, '7')
                    .pipe(operators.map(function (propertyValue) { return parseInt(propertyValue, 10); }))
                : rxjs.of(DefaultWidgetProperties.pageSize);
            this.includeCurrentContext = this.itemModel
                ? this.itemModel
                    .property(WidgetProperty.INCLUDE_CURRENT_CONTEXT, 'true')
                    .pipe(operators.map(function (propertyValue) { return "" + propertyValue === 'true'; }))
                : rxjs.of(DefaultWidgetProperties.includeCurrentContext);
            this.showSearchAfter = this.itemModel
                ? this.itemModel
                    .property(WidgetProperty.SHOW_SEARCH_AFTER, '7')
                    .pipe(operators.map(function (propertyValue) { return parseInt(propertyValue, 10); }))
                : rxjs.of(DefaultWidgetProperties.showSearchAfter);
            this.selectContextRedirectPage = this.itemModel
                ? this.itemModel.property(WidgetProperty.REDIRECT_PAGE, '')
                : rxjs.of(DefaultWidgetProperties.selectContextRedirectPage);
            this.showSingleContext = this.itemModel
                ? this.itemModel
                    .property(WidgetProperty.SHOW_SINGLE_CONTEXT, 'true')
                    .pipe(operators.map(function (propertyValue) { return "" + propertyValue === 'true'; }))
                : rxjs.of(DefaultWidgetProperties.showSingleContext);
            this.pageSize$ = this.getProperty(this.pageSizeSubject$, this.pageSize).pipe(operators.map(Number));
            this.includeCurrentContext$ = this.getProperty(this.includeCurrentContextSubject$, this.includeCurrentContext).pipe(operators.map(this.toBoolean));
            this.showSearchAfter$ = this.getProperty(this.showSearchAfterSubject$, this.showSearchAfter).pipe(operators.map(Number));
            this.selectContextRedirectPage$ = this.getProperty(this.selectContextRedirectPageSubject$, this.selectContextRedirectPage);
            this.showSingleContext$ = this.getProperty(this.showSingleContextSubject$, this.showSingleContext);
            this.properties$ = rxjs.combineLatest([
                this.pageSize$,
                this.includeCurrentContext$,
                this.showSearchAfter$,
                this.selectContextRedirectPage$,
                this.showSingleContext$,
            ]).pipe(operators.map(function (_a) {
                var _b = __read(_a, 5), pageSize = _b[0], includeCurrentContext = _b[1], showSearchAfter = _b[2], selectContextRedirectPage = _b[3], showSingleContext = _b[4];
                return ({
                    pageSize: pageSize,
                    includeCurrentContext: includeCurrentContext,
                    showSearchAfter: showSearchAfter,
                    selectContextRedirectPage: selectContextRedirectPage,
                    showSingleContext: showSingleContext,
                });
            }));
        }
        UserContextMenuWidgetPropertiesService.prototype.getProperty = function (input$, itemModel) {
            return rxjs.combineLatest([input$, itemModel]).pipe(operators.map(function (_a) {
                var _b = __read(_a, 2), value = _b[0], valueIM = _b[1];
                return (value === undefined ? valueIM : value);
            }));
        };
        UserContextMenuWidgetPropertiesService.prototype.toBoolean = function (value) {
            return value === 'true' || value === true;
        };
        UserContextMenuWidgetPropertiesService.prototype.setPageSize = function (value) {
            this.pageSizeSubject$.next(value);
        };
        UserContextMenuWidgetPropertiesService.prototype.setIncludeCurrentContext = function (value) {
            this.includeCurrentContextSubject$.next(value);
        };
        UserContextMenuWidgetPropertiesService.prototype.setShowSearchAfter = function (value) {
            this.showSearchAfterSubject$.next(value);
        };
        UserContextMenuWidgetPropertiesService.prototype.setSelectContextRedirectPage = function (value) {
            this.selectContextRedirectPageSubject$.next(value);
        };
        UserContextMenuWidgetPropertiesService.prototype.setShowSingleContext = function (value) {
            this.showSingleContextSubject$.next(value);
        };
        return UserContextMenuWidgetPropertiesService;
    }());
    /** @nocollapse */ UserContextMenuWidgetPropertiesService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: UserContextMenuWidgetPropertiesService, deps: [{ token: i1__namespace$3.ItemModel, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    /** @nocollapse */ UserContextMenuWidgetPropertiesService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: UserContextMenuWidgetPropertiesService });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: UserContextMenuWidgetPropertiesService, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () {
            return [{ type: i1__namespace$3.ItemModel, decorators: [{
                            type: i0.Optional
                        }] }];
        } });

    var UserContextMenuContainerComponent = /** @class */ (function () {
        function UserContextMenuContainerComponent(service, widgetPropertiesService, logoutService) {
            this.service = service;
            this.widgetPropertiesService = widgetPropertiesService;
            this.logoutService = logoutService;
            this.hostRef = this;
            this.currentContext = this.service.currentContext;
            this.showSingleContext = this.widgetPropertiesService.showSingleContext$;
            this.widgetProperties$ = this.widgetPropertiesService.properties$;
            this.canSelectContext = this.service.canSelectContext;
            this.userInfo = this.service.userInfo;
            this.showSwitchContextModal = false;
            this.renderContextSelect = false;
            this.showCancelButton = false;
        }
        UserContextMenuContainerComponent.prototype.openSwitchContextModal = function () {
            this.showSwitchContextModal = true;
            this.renderContextSelect = true;
        };
        UserContextMenuContainerComponent.prototype.logout = function () {
            var _this = this;
            this.logoutService
                .logout()
                .catch(function () { })
                .then(function () {
                _this.logoutService.goToLoginPage();
            });
        };
        /**
         * Set flag for show cancel button when loading finished
         */
        UserContextMenuContainerComponent.prototype.loadingChanged = function (event) {
            this.showCancelButton = !event;
        };
        return UserContextMenuContainerComponent;
    }());
    /** @nocollapse */ UserContextMenuContainerComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: UserContextMenuContainerComponent, deps: [{ token: UserContextMenuService }, { token: UserContextMenuWidgetPropertiesService }, { token: webSdk.LOGOUT }], target: i0__namespace.ɵɵFactoryTarget.Component });
    /** @nocollapse */ UserContextMenuContainerComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: UserContextMenuContainerComponent, selector: "bb-user-context-menu-container", inputs: { dropdownMenuPosition: "dropdownMenuPosition" }, ngImport: i0__namespace, template: "<ng-container\n  bbUserContextMenuContainerCustomizable\n  [bbHostRef]=\"hostRef\"\n  [bbTemplateContext]=\"{\n    currentContext: currentContext | async,\n    showSingleContext: showSingleContext | async,\n    dropdownMenuPosition: dropdownMenuPosition,\n    canSelectContext: canSelectContext | async,\n    userInfo: userInfo | async,\n    widgetProperties: widgetProperties$ | async\n  }\">\n</ng-container>\n\n<ng-template\n  bbUserContextMenuContainerCustomizable\n  let-hostRef\n  let-currentContext=\"context.currentContext\"\n  let-showSingleContext=\"context.showSingleContext\"\n  let-dropdownMenuPosition=\"context.dropdownMenuPosition\"\n  let-canSelectContext=\"context.canSelectContext\"\n  let-userInfo=\"context.userInfo\"\n  let-widgetProperties=\"context.widgetProperties\">\n  <div class=\"user-context-dropdown\">\n    <bb-dropdown-menu-ui\n      [position]=\"dropdownMenuPosition\"\n      btnColor=\"unstyled\"\n      container=\"body\"\n      [fullWidth]=\"true\"\n      ariaLabel=\"User Context Menu\"\n      i18n-ariaLabel=\"User Context Menu ariaLabel@@user-context-menu.ariaLabel.text\">\n      <ng-template bbDropdownLabel>\n        <div role=\"button\" data-role=\"context-navigation\" class=\"user-context-dropdown__selector\">\n          <bb-user-context-menu-info\n            [userInfo]=\"userInfo\"\n            [currentContext]=\"currentContext\"></bb-user-context-menu-info>\n          <div class=\"user-context-dropdown__toggle\">\n            <bb-icon-ui name=\"toggle-down\"></bb-icon-ui>\n          </div>\n        </div>\n      </ng-template>\n      <ng-template bbDropdownMenuItem>\n        <bb-user-context-menu-dropdown\n          [currentContext]=\"currentContext\"\n          [showSingleContext]=\"showSingleContext\"\n          [canSelectContext]=\"canSelectContext\"\n          (changeContext)=\"hostRef.openSwitchContextModal()\"\n          (logout)=\"hostRef.logout()\"></bb-user-context-menu-dropdown>\n      </ng-template>\n    </bb-dropdown-menu-ui>\n  </div>\n\n  <bb-modal-ui [(isOpen)]=\"hostRef.showSwitchContextModal\">\n    <bb-modal-body-ui>\n      <ng-template bbCustomModalBody>\n        <bb-select-context-widget\n          (loading)=\"hostRef.loadingChanged($event)\"\n          *ngIf=\"hostRef.renderContextSelect\"\n          [pageSize]=\"widgetProperties.pageSize\"\n          [includeCurrentContext]=\"widgetProperties.includeCurrentContext\"\n          [showSearchAfter]=\"widgetProperties.showSearchAfter\"\n          [selectContextRedirectPage]=\"widgetProperties.selectContextRedirectPage\"></bb-select-context-widget>\n      </ng-template>\n    </bb-modal-body-ui>\n    <bb-modal-footer-ui>\n      <ng-template bbCustomModalFooter>\n        <button\n          bbButton\n          *ngIf=\"hostRef.showCancelButton\"\n          (click)=\"hostRef.showSwitchContextModal = false\"\n          color=\"secondary\"\n          data-role=\"cancel-button\"\n          i18n=\"Cancel button text@@user-context-menu.label.cancelButton\">\n          Cancel\n        </button>\n      </ng-template>\n    </bb-modal-footer-ui>\n  </bb-modal-ui>\n</ng-template>\n", components: [{ type: i0__namespace.forwardRef(function () { return i3__namespace.DropdownMenuComponent; }), selector: "bb-dropdown-menu-ui", inputs: ["label", "disabled", "options", "icon", "iconSize", "iconColor", "btnColor", "btnCircle", "optionLabelKey", "fullWidth", "autoClose", "container", "position", "ariaLabel", "dropDownMenuRole", "dropDownMenuId", "buttonSize", "dropDownButtonLabel", "visibleElementDependingOnViewport", "role"], outputs: ["select"] }, { type: i0__namespace.forwardRef(function () { return UserContextMenuInfoComponent; }), selector: "bb-user-context-menu-info", inputs: ["userInfo", "currentContext"] }, { type: i0__namespace.forwardRef(function () { return i5__namespace.IconComponent; }), selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i0__namespace.forwardRef(function () { return UserContextMenuDropdownComponent; }), selector: "bb-user-context-menu-dropdown", inputs: ["currentContext", "showSingleContext", "canSelectContext"], outputs: ["changeContext", "logout"] }, { type: i0__namespace.forwardRef(function () { return i7__namespace.ModalComponent; }), selector: "bb-modal-ui", inputs: ["isOpen", "aria-labelledby", "aria-describedby", "modalOptions"], outputs: ["confirm", "cancel", "isOpenChange"] }, { type: i0__namespace.forwardRef(function () { return i7__namespace.ModalBodyComponent; }), selector: "bb-modal-body-ui" }, { type: i0__namespace.forwardRef(function () { return i8__namespace.SelectContextWidgetComponent; }), selector: "bb-select-context-widget", inputs: ["pageSize", "includeCurrentContext", "showSearchAfter", "selectContextRedirectPage", "backOutput"], outputs: ["back", "loading"] }, { type: i0__namespace.forwardRef(function () { return i7__namespace.ModalFooterComponent; }), selector: "bb-modal-footer-ui", inputs: ["confirmText", "cancelText"], outputs: ["confirm", "cancel"] }], directives: [{ type: i0__namespace.forwardRef(function () { return UserContextMenuContainerCustomizableDirective; }), selector: "[bbUserContextMenuContainerCustomizable]" }, { type: i0__namespace.forwardRef(function () { return i3__namespace.DropdownLabelDirective; }), selector: "ng-template[bbDropdownLabel]" }, { type: i0__namespace.forwardRef(function () { return i3__namespace.DropdownMenuItemDirective; }), selector: "ng-template[bbDropdownMenuItem]" }, { type: i0__namespace.forwardRef(function () { return i7__namespace.CustomModalBodyDirective; }), selector: "ng-template[bbCustomModalBody]" }, { type: i0__namespace.forwardRef(function () { return i2__namespace.NgIf; }), selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i0__namespace.forwardRef(function () { return i7__namespace.CustomModalFooterDirective; }), selector: "ng-template[bbCustomModalFooter]" }, { type: i0__namespace.forwardRef(function () { return i10__namespace.ButtonDirective; }), selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }], pipes: { "async": i0__namespace.forwardRef(function () { return i2__namespace.AsyncPipe; }) } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: UserContextMenuContainerComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-user-context-menu-container',
                        templateUrl: './user-context-menu-container.component.html',
                    }]
            }], ctorParameters: function () {
            return [{ type: UserContextMenuService }, { type: UserContextMenuWidgetPropertiesService }, { type: undefined, decorators: [{
                            type: i0.Inject,
                            args: [webSdk.LOGOUT]
                        }] }];
        }, propDecorators: { dropdownMenuPosition: [{
                    type: i0.Input
                }] } });
    var UserContextMenuContainerCustomizableDirective = /** @class */ (function (_super) {
        __extends(UserContextMenuContainerCustomizableDirective, _super);
        function UserContextMenuContainerCustomizableDirective() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return UserContextMenuContainerCustomizableDirective;
    }(i1$1.BbTemplate));
    /** @nocollapse */ UserContextMenuContainerCustomizableDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: UserContextMenuContainerCustomizableDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    /** @nocollapse */ UserContextMenuContainerCustomizableDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: UserContextMenuContainerCustomizableDirective, selector: "[bbUserContextMenuContainerCustomizable]", usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: UserContextMenuContainerCustomizableDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: '[bbUserContextMenuContainerCustomizable]',
                    }]
            }] });

    var UserContextMenuWidgetComponent = /** @class */ (function () {
        function UserContextMenuWidgetComponent(userContextMenuWidgetPropertiesService) {
            this.userContextMenuWidgetPropertiesService = userContextMenuWidgetPropertiesService;
            /**
             * Dropdown menu postion
             */
            this.dropdownMenuPosition = 'bottom-right';
        }
        Object.defineProperty(UserContextMenuWidgetComponent.prototype, "pageSize", {
            /**
             * Number of service agreements to load per page
             */
            set: function (value) {
                this.userContextMenuWidgetPropertiesService.setPageSize(value);
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(UserContextMenuWidgetComponent.prototype, "includeCurrentContext", {
            /**
             * Include current context
             */
            set: function (value) {
                this.userContextMenuWidgetPropertiesService.setIncludeCurrentContext(value);
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(UserContextMenuWidgetComponent.prototype, "showSearchAfter", {
            /**
             * Show search if more than service agreements
             */
            set: function (value) {
                this.userContextMenuWidgetPropertiesService.setShowSearchAfter(value);
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(UserContextMenuWidgetComponent.prototype, "selectContextRedirectPage", {
            /**
             * Redirect after successful context select
             */
            set: function (value) {
                this.userContextMenuWidgetPropertiesService.setSelectContextRedirectPage(value);
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(UserContextMenuWidgetComponent.prototype, "showSingleContext", {
            /**
             * Show current context in context switcher when there is only one context
             */
            set: function (value) {
                this.userContextMenuWidgetPropertiesService.setShowSingleContext(value);
            },
            enumerable: false,
            configurable: true
        });
        return UserContextMenuWidgetComponent;
    }());
    /** @nocollapse */ UserContextMenuWidgetComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: UserContextMenuWidgetComponent, deps: [{ token: UserContextMenuWidgetPropertiesService }], target: i0__namespace.ɵɵFactoryTarget.Component });
    /** @nocollapse */ UserContextMenuWidgetComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: UserContextMenuWidgetComponent, selector: "bb-user-context-menu-widget", inputs: { pageSize: "pageSize", includeCurrentContext: "includeCurrentContext", showSearchAfter: "showSearchAfter", selectContextRedirectPage: "selectContextRedirectPage", showSingleContext: "showSingleContext", dropdownMenuPosition: "dropdownMenuPosition" }, providers: [UserContextMenuService, UserContextMenuWidgetPropertiesService], ngImport: i0__namespace, template: "\n    <bb-user-context-menu-container [dropdownMenuPosition]=\"dropdownMenuPosition\"></bb-user-context-menu-container>\n  ", isInline: true, components: [{ type: UserContextMenuContainerComponent, selector: "bb-user-context-menu-container", inputs: ["dropdownMenuPosition"] }] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: UserContextMenuWidgetComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-user-context-menu-widget',
                        template: "\n    <bb-user-context-menu-container [dropdownMenuPosition]=\"dropdownMenuPosition\"></bb-user-context-menu-container>\n  ",
                        providers: [UserContextMenuService, UserContextMenuWidgetPropertiesService],
                    }]
            }], ctorParameters: function () { return [{ type: UserContextMenuWidgetPropertiesService }]; }, propDecorators: { pageSize: [{
                    type: i0.Input
                }], includeCurrentContext: [{
                    type: i0.Input
                }], showSearchAfter: [{
                    type: i0.Input
                }], selectContextRedirectPage: [{
                    type: i0.Input
                }], showSingleContext: [{
                    type: i0.Input
                }], dropdownMenuPosition: [{
                    type: i0.Input
                }] } });

    var components = [
        UserContextMenuWidgetComponent,
        UserContextMenuContainerComponent,
        UserContextMenuDropdownComponent,
        UserContextMenuInfoComponent,
    ];
    var directives = [
        UserContextMenuInfoCustomizableDirective,
        UserContextMenuDropdownCustomizableDirective,
        UserContextMenuContainerCustomizableDirective,
    ];
    var uiModules = [
        i1$2.AccessgroupUiModule,
        i8.SelectContextWidgetModule,
        i1.AvatarModule,
        i5.IconModule,
        i7.ModalModule,
        i10.ButtonModule,
        ngBootstrap.NgbDropdownModule,
        i3.DropdownMenuModule,
    ];
    var UserContextMenuWidgetModule = /** @class */ (function () {
        function UserContextMenuWidgetModule() {
        }
        return UserContextMenuWidgetModule;
    }());
    /** @nocollapse */ UserContextMenuWidgetModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: UserContextMenuWidgetModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    /** @nocollapse */ UserContextMenuWidgetModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: UserContextMenuWidgetModule, declarations: [UserContextMenuWidgetComponent,
            UserContextMenuContainerComponent,
            UserContextMenuDropdownComponent,
            UserContextMenuInfoComponent, UserContextMenuInfoCustomizableDirective,
            UserContextMenuDropdownCustomizableDirective,
            UserContextMenuContainerCustomizableDirective], imports: [i2.CommonModule, i1__namespace$3.BackbaseCoreModule, i1$2.AccessgroupUiModule,
            i8.SelectContextWidgetModule,
            i1.AvatarModule,
            i5.IconModule,
            i7.ModalModule,
            i10.ButtonModule,
            ngBootstrap.NgbDropdownModule,
            i3.DropdownMenuModule], exports: [UserContextMenuWidgetComponent,
            UserContextMenuContainerComponent,
            UserContextMenuDropdownComponent,
            UserContextMenuInfoComponent, UserContextMenuInfoCustomizableDirective,
            UserContextMenuDropdownCustomizableDirective,
            UserContextMenuContainerCustomizableDirective, i1$2.AccessgroupUiModule,
            i8.SelectContextWidgetModule,
            i1.AvatarModule,
            i5.IconModule,
            i7.ModalModule,
            i10.ButtonModule,
            ngBootstrap.NgbDropdownModule,
            i3.DropdownMenuModule] });
    /** @nocollapse */ UserContextMenuWidgetModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: UserContextMenuWidgetModule, imports: [__spreadArray([
                i2.CommonModule,
                i1$1.BackbaseCoreModule.withConfig({
                    classMap: { UserContextMenuWidgetComponent: UserContextMenuWidgetComponent },
                })
            ], __read(uiModules)), i1$2.AccessgroupUiModule,
            i8.SelectContextWidgetModule,
            i1.AvatarModule,
            i5.IconModule,
            i7.ModalModule,
            i10.ButtonModule,
            ngBootstrap.NgbDropdownModule,
            i3.DropdownMenuModule] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: UserContextMenuWidgetModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: __spreadArray(__spreadArray([], __read(components)), __read(directives)),
                        imports: __spreadArray([
                            i2.CommonModule,
                            i1$1.BackbaseCoreModule.withConfig({
                                classMap: { UserContextMenuWidgetComponent: UserContextMenuWidgetComponent },
                            })
                        ], __read(uiModules)),
                        exports: __spreadArray(__spreadArray(__spreadArray([], __read(components)), __read(directives)), __read(uiModules)),
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.UserContextMenuWidgetComponent = UserContextMenuWidgetComponent;
    exports.UserContextMenuWidgetModule = UserContextMenuWidgetModule;
    exports["ӨUserContextMenuContainerComponent"] = UserContextMenuContainerComponent;
    exports["ӨUserContextMenuContainerCustomizableDirective"] = UserContextMenuContainerCustomizableDirective;
    exports["ӨUserContextMenuDropdownComponent"] = UserContextMenuDropdownComponent;
    exports["ӨUserContextMenuDropdownCustomizableDirective"] = UserContextMenuDropdownCustomizableDirective;
    exports["ӨUserContextMenuInfoComponent"] = UserContextMenuInfoComponent;
    exports["ӨUserContextMenuInfoCustomizableDirective"] = UserContextMenuInfoCustomizableDirective;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
