// @backbase/product-summary-ui/esm2015/src/bb-product-kind-name/bb-product-kind-name.component.js
import { Component, Directive, Input } from '@angular/core';
import { BbTemplate } from '@backbase/foundation-ang/core';
import { ProductKindNames, ProductKindUri } from '@backbase/product-summary-common-ang';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class ProductKindNameComponent {
    constructor() {
        this.hostRef = this;
        this.productKindNames = ProductKindNames;
        this.productKindUri = ProductKindUri;
        /**
         * Product kind name for which display a localized version
         *
         * @default ''
         */
        this.name = '';
        /**
         * Defines whether product kind name is plural
         *
         * @default false
         */
        this.isPlural = false;
    }
}
ProductKindNameComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductKindNameComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ProductKindNameComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ProductKindNameComponent, selector: "bb-product-kind-name", inputs: { name: "name", isPlural: "isPlural" }, ngImport: i0, template: "<ng-container\n  bbProductKindNameCustomizable\n  [bbHostRef]=\"hostRef\"></ng-container>\n\n<ng-template\n  bbProductKindNameCustomizable\n  let-hostRef>\n  <ng-container [ngSwitch]=\"hostRef.name\">\n    <ng-container *ngSwitchCase=\"hostRef.productKindUri.CURRENT_ACCOUNT\">\n      <span i18n=\"Current Account|Product kind name@@product-summary.product-kind-name.current-account\">\n        {+hostRef.isPlural, plural, =0 {Current Account} =1 {Current Accounts}}\n      </span>\n    </ng-container>\n    <ng-container *ngSwitchCase=\"hostRef.productKindUri.SAVINGS_ACCOUNT\">\n      <span i18n=\"Savings Account|Product kind name@@product-summary.product-kind-name.savings-account\">\n        {+hostRef.isPlural, plural, =0 {Savings Account} =1 {Savings Accounts}}\n      </span>\n    </ng-container>\n    <ng-container *ngSwitchCase=\"hostRef.productKindUri.TERM_DEPOSIT\">\n      <span i18n=\"Terms Deposit|Product kind name@@product-summary.product-kind-name.term-deposit\">\n        {+hostRef.isPlural, plural, =0 {Term Deposit} =1 {Term Deposits}}\n      </span>\n    </ng-container>\n    <ng-container *ngSwitchCase=\"hostRef.productKindUri.LOAN\">\n      <span i18n=\"Loan|Product kind name@@product-summary.product-kind-name.loan\">\n        {+hostRef.isPlural, plural, =0 {Loan} =1 {Loans}}\n      </span>\n    </ng-container>\n    <ng-container *ngSwitchCase=\"hostRef.productKindUri.CREDIT_CARD\">\n      <span i18n=\"Credit Card|Product kind name@@product-summary.product-kind-name.credit-card\">\n        {+hostRef.isPlural, plural, =0 {Credit Card} =1 {Credit Cards}}\n      </span>\n    </ng-container>\n    <ng-container *ngSwitchCase=\"hostRef.productKindUri.DEBIT_CARD\">\n      <span i18n=\"Debit card|Product kind name@@product-summary.product-kind-name.debit-card\">\n        {+hostRef.isPlural, plural, =0 {Debit Card} =1 {Debit Cards}}\n      </span>\n    </ng-container>\n    <ng-container *ngSwitchCase=\"hostRef.productKindUri.INVESTMENT_ACCOUNT\">\n      <span i18n=\"Investment Account|Product kind name@@product-summary.product-kind-name.investment-account\">\n        {+hostRef.isPlural, plural, =0 {Investment Account} =1 {Investment Accounts}}\n      </span>\n    </ng-container>\n    <ng-container *ngSwitchDefault>\n      <span>{{hostRef.name}}</span>\n    </ng-container>\n  </ng-container>\n</ng-template>\n\n", directives: [{ type: i0.forwardRef(function () { return ProductKindNameCustomizableDirective; }), selector: "[bbProductKindNameCustomizable]" }, { type: i0.forwardRef(function () { return i1.NgSwitch; }), selector: "[ngSwitch]", inputs: ["ngSwitch"] }, { type: i0.forwardRef(function () { return i1.NgSwitchCase; }), selector: "[ngSwitchCase]", inputs: ["ngSwitchCase"] }, { type: i0.forwardRef(function () { return i1.NgSwitchDefault; }), selector: "[ngSwitchDefault]" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductKindNameComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-product-kind-name',
                    templateUrl: 'bb-product-kind-name.component.html',
                }]
        }], propDecorators: { name: [{
                type: Input
            }], isPlural: [{
                type: Input
            }] } });
/**
 * This is an extension slot to be used for customizing the entire component
 *
 * @usageNotes
 *
 * ### Customize details
 *
 * ```html
 * <ng-template
 *   bbProductKindNameCustomizable
 *   let-hostRef>
 *   Your custom template here
 * </ng-template>
 * ```
 *
 * @NgModule ProductSummaryAccountDetailsWidgetModule
 */
export class ProductKindNameCustomizableDirective extends BbTemplate {
}
ProductKindNameCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductKindNameCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
ProductKindNameCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: ProductKindNameCustomizableDirective, selector: "[bbProductKindNameCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductKindNameCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbProductKindNameCustomizable]',
                }]
        }] });
