// @backbase/product-summary-ui/esm2015/src/bb-product-kind-name/bb-product-kind-name.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ProductKindNameComponent, ProductKindNameCustomizableDirective } from './bb-product-kind-name.component';
import * as i0 from "@angular/core";
export class ProductKindNameModule {
}
ProductKindNameModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductKindNameModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductKindNameModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductKindNameModule, declarations: [ProductKindNameComponent, ProductKindNameCustomizableDirective], imports: [CommonModule], exports: [ProductKindNameComponent, ProductKindNameCustomizableDirective] });
ProductKindNameModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductKindNameModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductKindNameModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [ProductKindNameComponent, ProductKindNameCustomizableDirective],
                    exports: [ProductKindNameComponent, ProductKindNameCustomizableDirective],
                }]
        }] });
