// @backbase/messages-common/esm2015/src/components/create-message/employee-form/employee-create-message-form.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { ButtonModule } from '@backbase/ui-ang/button';
import { DropdownSingleSelectModule } from '@backbase/ui-ang/dropdown-single-select';
import { InputCheckboxModule } from '@backbase/ui-ang/input-checkbox';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { RichTextEditorModule } from '@backbase/ui-ang/rich-text-editor';
import { SearchBoxModule } from '@backbase/ui-ang/search-box';
import { MessagesManipulationConfirmModalModule } from '../../messages-manipulation-confirm-modal/messages-manipulation-confirm-modal.module';
import { MessagesUploadAttachmentsModule } from '../../upload-attachments/messages-upload-attachments.module';
import { MessagesEncodingService } from '../../../services/messages-encoding/messages-encoding.service';
import { BaseCreateMessageFormModule } from '../base-form/base-create-message-form.module';
import { EmployeeCreateMessageFormComponent, EmployeeCreateMessageFormCustomizableDirective, } from './employee-create-message-form.component';
import { EmployeeCreateMessageFormService } from './services/employee-create-message-form.service';
import { PreventBubbleDownModule } from '../../../directives/prevent-bubble-down/prevent-bubble-down.module';
import * as i0 from "@angular/core";
const components = [EmployeeCreateMessageFormComponent];
const directives = [EmployeeCreateMessageFormCustomizableDirective];
const uiModules = [
    DropdownSingleSelectModule,
    InputValidationMessageModule,
    SearchBoxModule,
    InputTextModule,
    RichTextEditorModule,
    ButtonModule,
    InputCheckboxModule,
    LoadButtonModule,
];
const innerModules = [
    MessagesManipulationConfirmModalModule,
    MessagesUploadAttachmentsModule,
    BaseCreateMessageFormModule,
];
export class EmployeeCreateMessageFormModule {
}
EmployeeCreateMessageFormModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: EmployeeCreateMessageFormModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
EmployeeCreateMessageFormModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: EmployeeCreateMessageFormModule, declarations: [EmployeeCreateMessageFormComponent, EmployeeCreateMessageFormCustomizableDirective], imports: [CommonModule, ReactiveFormsModule, PreventBubbleDownModule, DropdownSingleSelectModule,
        InputValidationMessageModule,
        SearchBoxModule,
        InputTextModule,
        RichTextEditorModule,
        ButtonModule,
        InputCheckboxModule,
        LoadButtonModule, MessagesManipulationConfirmModalModule,
        MessagesUploadAttachmentsModule,
        BaseCreateMessageFormModule], exports: [ReactiveFormsModule, EmployeeCreateMessageFormComponent, EmployeeCreateMessageFormCustomizableDirective, DropdownSingleSelectModule,
        InputValidationMessageModule,
        SearchBoxModule,
        InputTextModule,
        RichTextEditorModule,
        ButtonModule,
        InputCheckboxModule,
        LoadButtonModule, MessagesManipulationConfirmModalModule,
        MessagesUploadAttachmentsModule,
        BaseCreateMessageFormModule] });
EmployeeCreateMessageFormModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: EmployeeCreateMessageFormModule, providers: [EmployeeCreateMessageFormService, MessagesEncodingService], imports: [[CommonModule, ReactiveFormsModule, PreventBubbleDownModule, ...uiModules, ...innerModules], ReactiveFormsModule, DropdownSingleSelectModule,
        InputValidationMessageModule,
        SearchBoxModule,
        InputTextModule,
        RichTextEditorModule,
        ButtonModule,
        InputCheckboxModule,
        LoadButtonModule, MessagesManipulationConfirmModalModule,
        MessagesUploadAttachmentsModule,
        BaseCreateMessageFormModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: EmployeeCreateMessageFormModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ReactiveFormsModule, PreventBubbleDownModule, ...uiModules, ...innerModules],
                    exports: [ReactiveFormsModule, ...components, ...directives, ...uiModules, ...innerModules],
                    declarations: [...components, ...directives],
                    providers: [EmployeeCreateMessageFormService, MessagesEncodingService],
                }]
        }] });
