// @backbase/payment-orders-ang/esm2015/lib/ui/components/schedule/schedule.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ScheduleComponent, ScheduleCustomizableDirective, } from './schedule.component';
import { InputDatepickerModule } from '@backbase/ui-ang/input-datepicker';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { DropdownSingleSelectModule } from '@backbase/ui-ang/dropdown-single-select';
import { InputRadioGroupModule } from '@backbase/ui-ang/input-radio-group';
import { InputNumberModule } from '@backbase/ui-ang/input-number';
import { ReactiveFormsModule } from '@angular/forms';
import * as i0 from "@angular/core";
const uiModules = [
    InputDatepickerModule,
    InputValidationMessageModule,
    DropdownSingleSelectModule,
    InputNumberModule,
    InputRadioGroupModule,
];
export class ScheduleModule {
}
/** @nocollapse */ ScheduleModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ScheduleModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ ScheduleModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ScheduleModule, declarations: [ScheduleComponent, ScheduleCustomizableDirective], imports: [CommonModule, ReactiveFormsModule, InputDatepickerModule,
        InputValidationMessageModule,
        DropdownSingleSelectModule,
        InputNumberModule,
        InputRadioGroupModule], exports: [ScheduleComponent, ScheduleCustomizableDirective] });
/** @nocollapse */ ScheduleModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ScheduleModule, imports: [[CommonModule, ReactiveFormsModule, ...uiModules]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ScheduleModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ScheduleComponent, ScheduleCustomizableDirective],
                    imports: [CommonModule, ReactiveFormsModule, ...uiModules],
                    exports: [ScheduleComponent, ScheduleCustomizableDirective],
                }]
        }] });
