// @backbase/payord-stop-checks-list-widget-ang/bundles/backbase-payord-stop-checks-list-widget-ang.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('@backbase/foundation-ang/core'), require('rxjs'), require('rxjs/operators'), require('@backbase/payment-orders-ang'), require('@backbase/data-ang/stop-checks'), require('@backbase/ui-ang/table'), require('@backbase/ui-ang/pagination'), require('@backbase/ui-ang/load-button'), require('@backbase/ui-ang/loading-indicator'), require('@backbase/ui-ang/empty-state'), require('@backbase/ui-ang/modal'), require('@backbase/ui-ang/header'), require('@backbase/ui-ang/icon'), require('@backbase/ui-ang/amount'), require('@backbase/ui-ang/button'), require('@backbase/ui-ang/badge'), require('@backbase/ui-ang/common-error-state'), require('@backbase/ui-ang/tooltip-directive')) :
    typeof define === 'function' && define.amd ? define('@backbase/payord-stop-checks-list-widget-ang', ['exports', '@angular/core', '@angular/common', '@backbase/foundation-ang/core', 'rxjs', 'rxjs/operators', '@backbase/payment-orders-ang', '@backbase/data-ang/stop-checks', '@backbase/ui-ang/table', '@backbase/ui-ang/pagination', '@backbase/ui-ang/load-button', '@backbase/ui-ang/loading-indicator', '@backbase/ui-ang/empty-state', '@backbase/ui-ang/modal', '@backbase/ui-ang/header', '@backbase/ui-ang/icon', '@backbase/ui-ang/amount', '@backbase/ui-ang/button', '@backbase/ui-ang/badge', '@backbase/ui-ang/common-error-state', '@backbase/ui-ang/tooltip-directive'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["payord-stop-checks-list-widget-ang"] = {}), global.ng.core, global.ng.common, global.i1$1, global.rxjs, global.rxjs.operators, global.i1, global.i1$2, global.i2, global.i5$1, global.i6$1, global.i6, global.i1$3, global.i1$4, global.i2$1, global.i4, global.i5, global.i8, global.badge, global.commonErrorState, global.tooltipDirective));
})(this, (function (exports, i0, i3, i1$1, rxjs, operators, i1, i1$2, i2, i5$1, i6$1, i6, i1$3, i1$4, i2$1, i4, i5, i8, badge, commonErrorState, tooltipDirective) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i3__namespace = /*#__PURE__*/_interopNamespace(i3);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1$1);
    var i1__namespace$2 = /*#__PURE__*/_interopNamespace(i1);
    var i1__namespace$1 = /*#__PURE__*/_interopNamespace(i1$2);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i5__namespace$1 = /*#__PURE__*/_interopNamespace(i5$1);
    var i6__namespace$1 = /*#__PURE__*/_interopNamespace(i6$1);
    var i6__namespace = /*#__PURE__*/_interopNamespace(i6);
    var i1__namespace$3 = /*#__PURE__*/_interopNamespace(i1$3);
    var i1__namespace$4 = /*#__PURE__*/_interopNamespace(i1$4);
    var i2__namespace$1 = /*#__PURE__*/_interopNamespace(i2$1);
    var i4__namespace = /*#__PURE__*/_interopNamespace(i4);
    var i5__namespace = /*#__PURE__*/_interopNamespace(i5);
    var i8__namespace = /*#__PURE__*/_interopNamespace(i8);

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from, pack) {
        if (pack || arguments.length === 2)
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar)
                        ar = Array.prototype.slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
        return to.concat(ar || Array.prototype.slice.call(from));
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    var defaultChecksParams = {
        size: 50,
        paginationType: i1.LoadMoreMethod.PAGINATION,
        maxNavPages: 3,
        currentPage: 0,
    };

    var StopChecksListPropertiesService = /** @class */ (function () {
        function StopChecksListPropertiesService(itemModel) {
            this.itemModel = itemModel;
            this.pageSizeSubject$ = new rxjs.BehaviorSubject(undefined);
            this.maxNavPagesSubject$ = new rxjs.BehaviorSubject(undefined);
            this.paginationTypeSubject$ = new rxjs.BehaviorSubject(undefined);
            this.pageSize$ = rxjs.combineLatest([
                this.pageSizeSubject$,
                this.pageSizeIM,
            ]).pipe(operators.map(function (_b) {
                var _c = __read(_b, 2), value = _c[0], valueIM = _c[1];
                return value !== null && value !== void 0 ? value : valueIM;
            }));
            this.maxNavPages$ = rxjs.combineLatest([
                this.maxNavPagesSubject$,
                this.maxNavPagesIM,
            ]).pipe(operators.map(function (_b) {
                var _c = __read(_b, 2), value = _c[0], valueIM = _c[1];
                return value !== null && value !== void 0 ? value : valueIM;
            }));
            this.paginationType$ = rxjs.combineLatest([
                this.paginationTypeSubject$,
                this.paginationTypeIM,
            ]).pipe(operators.map(function (_b) {
                var _c = __read(_b, 2), value = _c[0], valueIM = _c[1];
                return value !== null && value !== void 0 ? value : valueIM;
            }));
        }
        Object.defineProperty(StopChecksListPropertiesService.prototype, "pageSizeIM", {
            get: function () {
                var _a;
                return (((_a = this.itemModel) === null || _a === void 0 ? void 0 : _a.property('pageSize', defaultChecksParams.size).pipe(operators.map(i1.toInteger))) || rxjs.of(defaultChecksParams.size));
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(StopChecksListPropertiesService.prototype, "maxNavPagesIM", {
            get: function () {
                var _a;
                return (((_a = this.itemModel) === null || _a === void 0 ? void 0 : _a.property('maxNavPages', defaultChecksParams.maxNavPages).pipe(operators.map(i1.toInteger))) || rxjs.of(defaultChecksParams.maxNavPages));
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(StopChecksListPropertiesService.prototype, "paginationTypeIM", {
            get: function () {
                var _a;
                return (((_a = this.itemModel) === null || _a === void 0 ? void 0 : _a.property('paginationType', defaultChecksParams.paginationType)) || rxjs.of(defaultChecksParams.paginationType));
            },
            enumerable: false,
            configurable: true
        });
        StopChecksListPropertiesService.prototype.setPageSize = function (value) {
            this.pageSizeSubject$.next(value);
        };
        StopChecksListPropertiesService.prototype.setMaxNavPages = function (value) {
            this.maxNavPagesSubject$.next(value);
        };
        StopChecksListPropertiesService.prototype.setPaginationType = function (value) {
            this.paginationTypeSubject$.next(value);
        };
        return StopChecksListPropertiesService;
    }());
    /** @nocollapse */ StopChecksListPropertiesService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StopChecksListPropertiesService, deps: [{ token: i1__namespace.ItemModel, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    /** @nocollapse */ StopChecksListPropertiesService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StopChecksListPropertiesService });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StopChecksListPropertiesService, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () {
            return [{ type: i1__namespace.ItemModel, decorators: [{
                            type: i0.Optional
                        }] }];
        } });

    var StopChecksListService = /** @class */ (function () {
        function StopChecksListService(stopChecksDataHttpService, preferences) {
            var _this = this;
            this.stopChecksDataHttpService = stopChecksDataHttpService;
            this.preferences = preferences;
            this.currentPage$ = new rxjs.BehaviorSubject(0);
            this.params$ = rxjs.combineLatest([
                this.currentPage$,
                this.preferences.pageSize$,
            ]).pipe(operators.map(function (_a) {
                var _b = __read(_a, 2), currentPage = _b[0], size = _b[1];
                var from = currentPage * size;
                return { from: from, size: size };
            }));
            this.stopChecksData$ = this.params$.pipe(operators.tap(function () { return _this.loading$.next(true); }), operators.switchMap(function (config) { return _this.getStopChecks(config).pipe(operators.catchError(function (error) {
                _this.error$.next(error);
                _this.loading$.next(false);
                return rxjs.EMPTY;
            }), operators.tap(function () {
                _this.loading$.next(false);
            })); }), operators.shareReplay({ bufferSize: 1, refCount: true }));
            this.totalItems$ = this.stopChecksData$.pipe(operators.map(function (res) { return res.totalStopChecks; }));
            this.error$ = new rxjs.Subject();
            this.loading$ = new rxjs.BehaviorSubject(true);
            this.getData$ = this.stopChecksData$.pipe(operators.map(function (res) { return res.stopChecks; }), operators.withLatestFrom(this.preferences.paginationType$), operators.scan(function (acc, _a) {
                var _b = __read(_a, 2), data = _b[0], paginationType = _b[1];
                if (paginationType !== i1.LoadMoreMethod.PAGINATION) {
                    return acc.concat(data);
                }
                return data;
            }, []));
            this.paginationConfig$ = rxjs.combineLatest([
                this.currentPage$,
                this.totalItems$.pipe(operators.startWith(undefined)),
                this.preferences.maxNavPages$,
                this.preferences.pageSize$,
                this.preferences.paginationType$,
            ]).pipe(operators.map(function (_a) {
                var _b = __read(_a, 5), currentPage = _b[0], totalItems = _b[1], maxNavPages = _b[2], size = _b[3], paginationType = _b[4];
                return ({
                    currentPage: currentPage,
                    totalItems: totalItems,
                    maxNavPages: maxNavPages,
                    size: size,
                    paginationType: paginationType,
                });
            }));
        }
        StopChecksListService.prototype.getStopChecks = function (params) {
            return this.stopChecksDataHttpService
                .getStopChecks(params, 'response')
                .pipe(operators.map(i1.fromResponse));
        };
        StopChecksListService.prototype.loadMore = function (page) {
            this.currentPage$.next(page);
        };
        return StopChecksListService;
    }());
    /** @nocollapse */ StopChecksListService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StopChecksListService, deps: [{ token: i1__namespace$1.StopChecksHttpService }, { token: StopChecksListPropertiesService }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    /** @nocollapse */ StopChecksListService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StopChecksListService });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StopChecksListService, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () { return [{ type: i1__namespace$1.StopChecksHttpService }, { type: StopChecksListPropertiesService }]; } });

    var StopChecksListCommunicationService = /** @class */ (function () {
        function StopChecksListCommunicationService() {
            this.refreshListSub = new rxjs.Subject();
            this.refreshList$ = this.refreshListSub.asObservable();
        }
        return StopChecksListCommunicationService;
    }());

    var StopChecksListComponent = /** @class */ (function () {
        function StopChecksListComponent() {
            /**
             * Reference to instance of customizable component.
             */
            this.hostRef = this;
            /**
             * Changes will be triggered on paymentItemClick event emit.
             */
            this.checkItemClick = new i0.EventEmitter();
        }
        StopChecksListComponent.prototype.onPaymentItemClick = function (checkItem) {
            if (checkItem) {
                this.checkItemClick.emit(checkItem);
            }
        };
        return StopChecksListComponent;
    }());
    /** @nocollapse */ StopChecksListComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StopChecksListComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    /** @nocollapse */ StopChecksListComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: StopChecksListComponent, selector: "bb-stop-checks-list", inputs: { checks: "checks" }, outputs: { checkItemClick: "checkItemClick" }, ngImport: i0__namespace, template: "<ng-container\n  bbStopChecksListCustomizableDirective\n  [bbHostRef]=\"hostRef\"\n></ng-container>\n<ng-template bbStopChecksListCustomizableDirective let-hostRef>\n  <div class=\"table-responsive\">\n    <table\n      class=\"table table-hover bb-table bb-block--no-margin\"\n      [bbTable]=\"hostRef.checks\"\n      bbTableFocus\n    >\n      <thead>\n        <tr>\n          <th data-role=\"stop-checks-header-account\" scope=\"col\">\n            <span i18n=\"@@stop-checks.header.account\"> Check account </span>\n          </th>\n          <th data-role=\"stop-checks-header-check-numbers\" scope=\"col\">\n            <span i18n=\"@@stop-checks.header.numbers\"> Check number(s) </span>\n          </th>\n          <th data-role=\"stop-checks-header-check-submitted-date\" scope=\"col\">\n            <span i18n=\"@@stop-checks.header.date.submitted\">\n              Date submitted\n            </span>\n          </th>\n          <th\n            scope=\"col\"\n            data-role=\"stop-checks-header-check-expiry-date\"\n            i18n=\"@@stop-checks.header.expiry.date\"\n          >\n            <span>Expiry date</span>\n          </th>\n          <th\n            scope=\"col\"\n            data-role=\"stop-checks-header-status\"\n            i18n=\"@@stop-checks.header.status\"\n          >\n            <span>Status</span>\n          </th>\n        </tr>\n      </thead>\n      <tbody>\n        <tr\n          *ngFor=\"let check of hostRef.checks\"\n          data-role=\"stop-checks-row\"\n          [ngClass]=\"{ 'text-muted': check.status.toUpperCase() === 'EXPIRED' }\"\n          (keydown.enter)=\"hostRef.onPaymentItemClick(check)\"\n          (keydown.space)=\"hostRef.onPaymentItemClick(check)\"\n          (click)=\"hostRef.onPaymentItemClick(check)\"\n          tabindex=\"0\"\n        >\n          <td class=\"d-none d-md-table-cell\">\n            <div data-role=\"stop-check-originator-account-name\">\n              {{ check.originatorAccount.name }}\n            </div>\n            <div\n              class=\"bb-text-support\"\n              data-role=\"stop-check-originator-account-number\"\n            >\n              {{ check.originatorAccount.identification }}\n            </div>\n          </td>\n          <td>\n            <div class=\"d-block d-md-none bb-block--sm\"></div>\n            <div class=\"d-block d-md-none bb-block--sm\">\n              <bb-payord-status-badge\n                [status]=\"check.status\"\n              ></bb-payord-status-badge>\n            </div>\n\n            <div class=\"d-block d-md-none\">\n              <div class=\"bb-block--sm\">\n                <div\n                  class=\"bb-text-bold\"\n                  i18n=\"@@stop-checks.header.small.account\"\n                >\n                  Check account\n                </div>\n                <div data-role=\"stop-check-originator-account-name\">\n                  {{ check.originatorAccount.name }}\n                </div>\n                <div\n                  class=\"bb-text-support\"\n                  data-role=\"stop-check-originator-account-number\"\n                >\n                  {{ check.originatorAccount.identification }}\n                </div>\n              </div>\n\n              <div class=\"bb-block--sm\">\n                <div\n                  class=\"bb-text-bold\"\n                  i18n=\"@@stop-checks.header.small.numbers\"\n                >\n                  Check number\n                </div>\n                <ng-container *ngIf=\"check.check\">{{\n                  check.check.number\n                }}</ng-container>\n                <ng-container *ngIf=\"check.checkNumberRange\">\n                  {{ check.checkNumberRange.firstNumber }}\n                  <span i18n=\"@@stop-checks.table.label.to\">to</span>\n                  {{ check.checkNumberRange.lastNumber }}\n                </ng-container>\n              </div>\n\n              <div class=\"row\">\n                <div class=\"col-6\">\n                  <div\n                    class=\"bb-text-bold\"\n                    i18n=\"@@stop-checks.header.small.date.submitted\"\n                  >\n                    Date submitted\n                  </div>\n                  <div>{{ check.dateOfRequest | date: 'MM/dd/y' }}</div>\n                </div>\n                <div class=\"col-6\">\n                  <div\n                    class=\"bb-text-bold\"\n                    i18n=\"@@stop-checks.header.small.date.expired\"\n                  >\n                    Date expired\n                  </div>\n                  <div *ngIf=\"check.expirationDate; else noValue\">\n                    {{ check.expirationDate | date: 'MM/dd/y' }}\n                  </div>\n                </div>\n              </div>\n            </div>\n\n            <div class=\"d-none d-md-block\">\n              <ng-container *ngIf=\"check.check\">{{\n                check.check.number\n              }}</ng-container>\n              <ng-container *ngIf=\"check.checkNumberRange\">\n                {{ check.checkNumberRange.firstNumber }}\n                <span i18n=\"@@stop-checks.table.label.to\">to</span>\n                {{ check.checkNumberRange.lastNumber }}\n              </ng-container>\n            </div>\n          </td>\n          <td class=\"d-none d-md-table-cell\" data-role=\"stop-check-col-date\">\n            {{ check.dateOfRequest | date: 'MM/dd/y' }}\n          </td>\n          <td class=\"d-none d-md-table-cell\" data-role=\"stop-check-col-date\">\n            <span *ngIf=\"check.expirationDate; else noValue\">{{\n              check.expirationDate | date: 'MM/dd/y'\n            }}</span>\n          </td>\n          <td class=\"d-none d-md-table-cell\" data-role=\"stop-check-col-status\">\n            <bb-payord-status-badge\n              [status]=\"check.status\"\n            ></bb-payord-status-badge>\n          </td>\n        </tr>\n      </tbody>\n    </table>\n  </div>\n</ng-template>\n\n<ng-template #noValue>-</ng-template>\n", components: [{ type: i0__namespace.forwardRef(function () { return i1__namespace$2.PayordStatusBadgeComponent; }), selector: "bb-payord-status-badge", inputs: ["status", "statusColorMap", "statusColorDefault", "showAbbreviated"] }], directives: [{ type: i0__namespace.forwardRef(function () { return StopChecksListCustomizableDirective; }), selector: "[bbStopChecksListCustomizableDirective]" }, { type: i0__namespace.forwardRef(function () { return i2__namespace.TableDirective; }), selector: "table[bbTable]", inputs: ["bbTable"], outputs: ["selectRow", "rowClick", "sort"] }, { type: i0__namespace.forwardRef(function () { return i2__namespace.TableFocusDirective; }), selector: "[bbTableFocus]", inputs: ["focusFirstRowOnChanges"] }, { type: i0__namespace.forwardRef(function () { return i3__namespace.NgForOf; }), selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { type: i0__namespace.forwardRef(function () { return i3__namespace.NgClass; }), selector: "[ngClass]", inputs: ["class", "ngClass"] }, { type: i0__namespace.forwardRef(function () { return i3__namespace.NgIf; }), selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "date": i0__namespace.forwardRef(function () { return i3__namespace.DatePipe; }) }, changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StopChecksListComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-stop-checks-list',
                        templateUrl: './stop-checks-list.component.html',
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], propDecorators: { checks: [{
                    type: i0.Input
                }], checkItemClick: [{
                    type: i0.Output
                }] } });
    var StopChecksListCustomizableDirective = /** @class */ (function (_super) {
        __extends(StopChecksListCustomizableDirective, _super);
        function StopChecksListCustomizableDirective() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return StopChecksListCustomizableDirective;
    }(i1$1.BbTemplate));
    /** @nocollapse */ StopChecksListCustomizableDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StopChecksListCustomizableDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    /** @nocollapse */ StopChecksListCustomizableDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: StopChecksListCustomizableDirective, selector: "[bbStopChecksListCustomizableDirective]", usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StopChecksListCustomizableDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: '[bbStopChecksListCustomizableDirective]',
                    }]
            }] });

    var StopChecksEmptyListComponent = /** @class */ (function () {
        function StopChecksEmptyListComponent() {
            /**
             * Reference to instance of customizable component.
             */
            this.hostRef = this;
            /**
             * Indicates if empty result caused by user search
             * Default value is false
             */
            this.isSearch = false;
        }
        return StopChecksEmptyListComponent;
    }());
    /** @nocollapse */ StopChecksEmptyListComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StopChecksEmptyListComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    /** @nocollapse */ StopChecksEmptyListComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: StopChecksEmptyListComponent, selector: "bb-empty-stop-checks-list", inputs: { isSearch: "isSearch" }, ngImport: i0__namespace, template: "\n    <ng-container\n      bbStopChecksEmptyListCustomizableDirective\n      [bbHostRef]=\"hostRef\"\n    >\n    </ng-container>\n    <ng-template bbStopChecksEmptyListCustomizableDirective let-hostRef>\n      <div class=\"bb-state-container\">\n        <bb-empty-state-ui\n          *ngIf=\"!hostRef.isSearch\"\n          title=\"No stop check requests\"\n          i18n-title=\"@@stop-checks.no.stop.checks.title\"\n          subtitle=\"A stop check payment alerts us that you want to prevent a check you have written from being cashed.\"\n          i18n-subtitle=\"@@stop-checks.no.stop.checks.subTitle\"\n          iconSize=\"xl\"\n          iconModifier=\"payments\"\n        ></bb-empty-state-ui>\n      </div>\n    </ng-template>\n  ", isInline: true, components: [{ type: i0__namespace.forwardRef(function () { return i1__namespace$3.EmptyStateComponent; }), selector: "bb-empty-state-ui", inputs: ["title", "subtitle", "showIcon", "iconClasses", "iconModifier", "iconSize", "iconColor"] }], directives: [{ type: i0__namespace.forwardRef(function () { return StopChecksEmptyListCustomizableDirective; }), selector: "[bbStopChecksEmptyListCustomizableDirective]" }, { type: i0__namespace.forwardRef(function () { return i3__namespace.NgIf; }), selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StopChecksEmptyListComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-empty-stop-checks-list',
                        template: "\n    <ng-container\n      bbStopChecksEmptyListCustomizableDirective\n      [bbHostRef]=\"hostRef\"\n    >\n    </ng-container>\n    <ng-template bbStopChecksEmptyListCustomizableDirective let-hostRef>\n      <div class=\"bb-state-container\">\n        <bb-empty-state-ui\n          *ngIf=\"!hostRef.isSearch\"\n          title=\"No stop check requests\"\n          i18n-title=\"@@stop-checks.no.stop.checks.title\"\n          subtitle=\"A stop check payment alerts us that you want to prevent a check you have written from being cashed.\"\n          i18n-subtitle=\"@@stop-checks.no.stop.checks.subTitle\"\n          iconSize=\"xl\"\n          iconModifier=\"payments\"\n        ></bb-empty-state-ui>\n      </div>\n    </ng-template>\n  ",
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], propDecorators: { isSearch: [{
                    type: i0.Input
                }] } });
    var StopChecksEmptyListCustomizableDirective = /** @class */ (function (_super) {
        __extends(StopChecksEmptyListCustomizableDirective, _super);
        function StopChecksEmptyListCustomizableDirective() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return StopChecksEmptyListCustomizableDirective;
    }(i1$1.BbTemplate));
    /** @nocollapse */ StopChecksEmptyListCustomizableDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StopChecksEmptyListCustomizableDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    /** @nocollapse */ StopChecksEmptyListCustomizableDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: StopChecksEmptyListCustomizableDirective, selector: "[bbStopChecksEmptyListCustomizableDirective]", usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StopChecksEmptyListCustomizableDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: '[bbStopChecksEmptyListCustomizableDirective]',
                    }]
            }] });

    var StopChecksDetailsComponent = /** @class */ (function () {
        function StopChecksDetailsComponent() {
            /**
             * Reference to instance of customizable component.
             */
            this.hostRef = this;
            /**
             * Indicator for details dialog open state.
             * Default value is `false`.
             */
            this.isOpen = false;
            /**
             * Changes will be trigged on close event emit.
             */
            // eslint-disable-next-line @angular-eslint/no-output-native
            this.close = new i0.EventEmitter();
        }
        StopChecksDetailsComponent.prototype.onClose = function () {
            this.isOpen = false;
            this.close.emit();
        };
        StopChecksDetailsComponent.prototype.onPrint = function () {
            window.print();
        };
        return StopChecksDetailsComponent;
    }());
    /** @nocollapse */ StopChecksDetailsComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StopChecksDetailsComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    /** @nocollapse */ StopChecksDetailsComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: StopChecksDetailsComponent, selector: "bb-stop-checks-details", inputs: { isOpen: "isOpen", check: "check" }, outputs: { close: "close" }, ngImport: i0__namespace, template: "<ng-container\n  bbStopChecksDetailsCustomizableDirective\n  [bbHostRef]=\"hostRef\"\n></ng-container>\n<ng-template bbStopChecksDetailsCustomizableDirective let-hostRef>\n  <bb-modal-ui\n    [isOpen]=\"hostRef.isOpen\"\n    (cancel)=\"hostRef.onClose()\"\n    (confirm)=\"hostRef.onClose()\"\n  >\n    <ng-container *ngIf=\"hostRef.check; else loading\">\n      <bb-modal-header-ui>\n        <ng-template bbCustomModalHeader>\n          <div class=\"bb-stack\" data-role=\"stop-checks-details-header\">\n            <div class=\"bb-stack__item\">\n              <bb-header-ui\n                headingType=\"h2\"\n                heading=\"Stop check request\"\n                i18n-heading=\"@@stop-checks.details.header.title\"\n              ></bb-header-ui>\n              <bb-payord-status-badge\n                [status]=\"hostRef.check.status\"\n              ></bb-payord-status-badge>\n              <span class=\"text-small bb-text-support\">\n                <span i18n=\"@@stop-checks.details.submitted.on\">\n                  Submitted on\n                </span>\n                <span>{{ hostRef.check.dateOfRequest | date: 'MM/dd/y' }}</span>\n              </span>\n            </div>\n\n            <div\n              class=\"bb-stack__item bb-stack__item--push-right bb-stack__item--align-top d-print-none\"\n            >\n              <div class=\"bb-stack\">\n                <div class=\"bb-stack__item\">\n                  <button\n                    bbButton\n                    [circle]=\"true\"\n                    buttonSize=\"sm\"\n                    color=\"link-dark\"\n                    (click)=\"hostRef.onPrint()\"\n                    (keydown.enter)=\"hostRef.onPrint()\"\n                    type=\"button\"\n                    aria-label=\"Print stop check detail dialog\"\n                    data-role=\"print-stop-checks-details-modal\"\n                    i18n-aria-label=\"@@stop-checks.check.header.print\"\n                  >\n                    <bb-icon-ui color=\"muted\" name=\"print\"></bb-icon-ui>\n                  </button>\n                </div>\n                <div class=\"bb-stack__item\">\n                  <button\n                    bbButton\n                    [circle]=\"true\"\n                    buttonSize=\"sm\"\n                    color=\"unstyled\"\n                    (click)=\"hostRef.onClose()\"\n                    (keydown.enter)=\"hostRef.onClose()\"\n                    aria-label=\"Close stop check detail dialog\"\n                    data-role=\"close-stop-checks-details-modal\"\n                    i18n-aria-label=\"@@stop-checks.stop-checks.header.close\"\n                  >\n                    <bb-icon-ui color=\"muted\" name=\"clear\"></bb-icon-ui>\n                  </button>\n                </div>\n              </div>\n            </div>\n          </div>\n        </ng-template>\n      </bb-modal-header-ui>\n\n      <bb-modal-body-ui>\n        <ng-template bbCustomModalBody>\n          <div class=\"bb-block--md\">\n            <bb-header-ui\n              headingType=\"h4\"\n              heading=\"Check details\"\n              i18n-heading=\"@@stop-checks.details.body.title\"\n            ></bb-header-ui>\n          </div>\n\n          <div class=\"bb-block--md\">\n            <div\n              class=\"bb-label bb-block--xs\"\n              i18n=\"@@stop-checks.details.body.account.label\"\n            >\n              Check Account\n            </div>\n            <div>{{ hostRef.check.originatorAccount.name }}</div>\n            <div class=\"bb-text-support\">\n              {{ hostRef.check.originatorAccount.identification }}\n            </div>\n          </div>\n\n          <ng-container *ngIf=\"hostRef.check.check; else numbersRange\">\n            <div class=\"bb-stack row break-word\">\n              <div\n                class=\"bb-stack__item col-12 bb-block--md\"\n                [ngClass]=\"{ 'col-md-6': hostRef.check.check.amount }\"\n              >\n                <div\n                  class=\"bb-label bb-block--xs\"\n                  i18n=\"@@stop-checks.details.body.check.number\"\n                >\n                  Check number\n                </div>\n                <div>{{ hostRef.check.check.number }}</div>\n              </div>\n              <div\n                *ngIf=\"hostRef.check.check.amount\"\n                class=\"bb-stack__item col-12 col-md-6 bb-block--md\"\n              >\n                <div\n                  class=\"bb-label bb-block--xs\"\n                  i18n=\"@@stop-checks.details.body.check.amount\"\n                >\n                  Check amount\n                </div>\n                <bb-amount-ui\n                  [amount]=\"hostRef.check.check.amount?.amount\"\n                  [currency]=\"hostRef.check.check.amount?.currencyCode\"\n                ></bb-amount-ui>\n              </div>\n            </div>\n\n            <div class=\"bb-stack row break-word\">\n              <div\n                *ngIf=\"hostRef.check.check.dateIssued\"\n                class=\"bb-stack__item col-12 bb-block--md\"\n                [ngClass]=\"{ 'col-md-6': hostRef.check.check.payeeName }\"\n              >\n                <div\n                  class=\"bb-label bb-block--xs\"\n                  i18n=\"@@stop-checks.details.body.check.date\"\n                >\n                  Check date\n                </div>\n                <div>\n                  {{ hostRef.check.check.dateIssued | date: 'MM/dd/y' }}\n                </div>\n              </div>\n              <div\n                *ngIf=\"hostRef.check.check.payeeName\"\n                class=\"bb-stack__item col-12 col-md-6 bb-block--md\"\n              >\n                <div\n                  class=\"bb-label bb-block--xs\"\n                  i18n=\"@@stop-checks.details.body.check.payee.name\"\n                >\n                  Payee name\n                </div>\n                <div>{{ hostRef.check.check.payeeName }}</div>\n              </div>\n            </div>\n          </ng-container>\n\n          <div class=\"bb-stack row break-word\">\n            <div\n              *ngIf=\"hostRef.check.reason\"\n              class=\"bb-stack__item\"\n              [ngClass]=\"{ 'col-md-6': hostRef.check.expirationDate }\"\n            >\n              <div\n                class=\"bb-label bb-block--xs\"\n                i18n=\"@@stop-checks.details.body.stop.reason\"\n              >\n                Stop reason\n              </div>\n              <div>{{ hostRef.check.reason }}</div>\n            </div>\n            <div class=\"d-md-none\">&nbsp;</div>\n            <div\n              *ngIf=\"hostRef.check.expirationDate\"\n              class=\"bb-stack__item col-md-6\"\n            >\n              <div\n                class=\"bb-label bb-block--xs\"\n                i18n=\"@@stop-checks.details.body.expiration.date\"\n              >\n                Stop expiration date\n              </div>\n              <div>{{ hostRef.check.expirationDate | date: 'MM/dd/y' }}</div>\n            </div>\n          </div>\n        </ng-template>\n      </bb-modal-body-ui>\n    </ng-container>\n\n    <ng-template #numbersRange>\n      <div class=\"bb-stack row break-word\">\n        <div class=\"bb-stack__item col-12 col-md-6 bb-block--md\">\n          <div\n            class=\"bb-label bb-block--xs\"\n            i18n=\"@@stop-checks.details.body.range.first.label\"\n          >\n            First check number\n          </div>\n          <div>{{ hostRef.check.checkNumberRange.firstNumber }}</div>\n        </div>\n        <div class=\"bb-stack__item col-12 col-md-6 bb-block--md\">\n          <div\n            class=\"bb-label bb-block--xs\"\n            i18n=\"@@stop-checks.details.body.range.last.label\"\n          >\n            Last check number\n          </div>\n          <div>{{ hostRef.check.checkNumberRange.lastNumber }}</div>\n        </div>\n      </div>\n    </ng-template>\n\n    <ng-template #loading>\n      <bb-loading-indicator-ui></bb-loading-indicator-ui>\n    </ng-template>\n  </bb-modal-ui>\n</ng-template>\n", components: [{ type: i0__namespace.forwardRef(function () { return i1__namespace$4.ModalComponent; }), selector: "bb-modal-ui", inputs: ["isOpen", "aria-labelledby", "aria-describedby", "modalOptions"], outputs: ["confirm", "cancel", "isOpenChange"] }, { type: i0__namespace.forwardRef(function () { return i1__namespace$4.ModalHeaderComponent; }), selector: "bb-modal-header-ui", inputs: ["title"], outputs: ["close"] }, { type: i0__namespace.forwardRef(function () { return i2__namespace$1.HeaderComponent; }), selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i0__namespace.forwardRef(function () { return i1__namespace$2.PayordStatusBadgeComponent; }), selector: "bb-payord-status-badge", inputs: ["status", "statusColorMap", "statusColorDefault", "showAbbreviated"] }, { type: i0__namespace.forwardRef(function () { return i4__namespace.IconComponent; }), selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i0__namespace.forwardRef(function () { return i1__namespace$4.ModalBodyComponent; }), selector: "bb-modal-body-ui" }, { type: i0__namespace.forwardRef(function () { return i5__namespace.AmountComponent; }), selector: "bb-amount-ui", inputs: ["currency", "showPlusSign", "mapCurrency", "showPercent", "abbreviate", "decimalPlaces", "trailingZeroes", "amount"] }, { type: i0__namespace.forwardRef(function () { return i6__namespace.LoadingIndicatorComponent; }), selector: "bb-loading-indicator-ui", inputs: ["text", "loaderSize", "showDelay", "hasBackground", "inline"] }], directives: [{ type: i0__namespace.forwardRef(function () { return StopChecksDetailsCustomizableDirective; }), selector: "[bbStopChecksDetailsCustomizableDirective]" }, { type: i0__namespace.forwardRef(function () { return i3__namespace.NgIf; }), selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i0__namespace.forwardRef(function () { return i1__namespace$4.CustomModalHeaderDirective; }), selector: "ng-template[bbCustomModalHeader]" }, { type: i0__namespace.forwardRef(function () { return i8__namespace.ButtonDirective; }), selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }, { type: i0__namespace.forwardRef(function () { return i1__namespace$4.CustomModalBodyDirective; }), selector: "ng-template[bbCustomModalBody]" }, { type: i0__namespace.forwardRef(function () { return i3__namespace.NgClass; }), selector: "[ngClass]", inputs: ["class", "ngClass"] }], pipes: { "date": i0__namespace.forwardRef(function () { return i3__namespace.DatePipe; }) }, changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StopChecksDetailsComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-stop-checks-details',
                        templateUrl: './stop-checks-details.component.html',
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], propDecorators: { isOpen: [{
                    type: i0.Input
                }], check: [{
                    type: i0.Input
                }], close: [{
                    type: i0.Output
                }] } });
    var StopChecksDetailsCustomizableDirective = /** @class */ (function (_super) {
        __extends(StopChecksDetailsCustomizableDirective, _super);
        function StopChecksDetailsCustomizableDirective() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return StopChecksDetailsCustomizableDirective;
    }(i1$1.BbTemplate));
    /** @nocollapse */ StopChecksDetailsCustomizableDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StopChecksDetailsCustomizableDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    /** @nocollapse */ StopChecksDetailsCustomizableDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: StopChecksDetailsCustomizableDirective, selector: "[bbStopChecksDetailsCustomizableDirective]", usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StopChecksDetailsCustomizableDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: '[bbStopChecksDetailsCustomizableDirective]',
                    }]
            }] });

    var StopChecksListContainerComponent = /** @class */ (function () {
        function StopChecksListContainerComponent(communicationService, model, checkService) {
            var _this = this;
            var _a;
            this.communicationService = communicationService;
            this.model = model;
            this.checkService = checkService;
            this.destroyed$ = new rxjs.Subject();
            /**
             * Reference to instance of customizable component.
             */
            this.hostRef = this;
            this.loading$ = this.checkService.loading$;
            this.paginationConfig$ = this.checkService.paginationConfig$;
            this.checksList$ = this.checkService.getData$;
            this.openDetails = false;
            (_a = this.communicationService) === null || _a === void 0 ? void 0 : _a.refreshList$.subscribe(function () { return _this.checkService.loadMore(0); });
            this.checkService.error$
                .pipe(operators.takeUntil(this.destroyed$))
                .subscribe(function (val) {
                _this.error = val;
            });
            this.checksList$.pipe(operators.takeUntil(this.destroyed$)).subscribe(function (res) {
                _this.checksList = res;
            });
        }
        StopChecksListContainerComponent.prototype.ngOnDestroy = function () {
            this.destroyed$.next();
        };
        StopChecksListContainerComponent.prototype.onCheckItemClick = function (selectedCheckItem) {
            this.openDetails = true;
            this.selectedCheck = selectedCheckItem;
        };
        StopChecksListContainerComponent.prototype.onCloseDetails = function () {
            this.openDetails = false;
        };
        StopChecksListContainerComponent.prototype.onPageChange = function (page) {
            this.checkService.loadMore(page++);
        };
        StopChecksListContainerComponent.prototype.onLoadMore = function (params) {
            this.checkService.loadMore(params.from);
        };
        StopChecksListContainerComponent.prototype.isPagination = function (paginationType) {
            return paginationType === i1.LoadMoreMethod.PAGINATION;
        };
        StopChecksListContainerComponent.prototype.showListLoading = function (isLoading, _b) {
            var paginationType = _b.paginationType, currentPage = _b.currentPage;
            var isFirstPage = currentPage === 0;
            return isLoading && (this.isPagination(paginationType) || isFirstPage);
        };
        return StopChecksListContainerComponent;
    }());
    /** @nocollapse */ StopChecksListContainerComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StopChecksListContainerComponent, deps: [{ token: StopChecksListCommunicationService, optional: true }, { token: i1__namespace.ItemModel, optional: true }, { token: StopChecksListService }], target: i0__namespace.ɵɵFactoryTarget.Component });
    /** @nocollapse */ StopChecksListContainerComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: StopChecksListContainerComponent, selector: "bb-stop-checks-list-container", ngImport: i0__namespace, template: "<ng-container\n  bbStopChecksListContainerCustomizableDirective\n  [bbTemplateContext]=\"{\n    isLoading: loading$ | async,\n    paginationConfig: paginationConfig$ | async\n  }\"\n  [bbHostRef]=\"hostRef\"\n></ng-container>\n\n<ng-template\n  bbStopChecksListContainerCustomizableDirective\n  let-hostRef\n  let-isLoading=\"context.isLoading\"\n  let-paginationConfig=\"context.paginationConfig\"\n>\n  <ng-container\n    *ngIf=\"hostRef.showListLoading(isLoading, paginationConfig); else list\"\n  >\n    <ng-container *ngTemplateOutlet=\"loading\"></ng-container>\n  </ng-container>\n\n  <ng-template #list>\n    <ng-container *ngIf=\"!hostRef.error; else errorState\">\n      <ng-container *ngIf=\"hostRef.checksList?.length; else emptyState\">\n        <div class=\"bb-block bb-block--md\">\n          <bb-stop-checks-list\n            [checks]=\"hostRef.checksList\"\n            (checkItemClick)=\"hostRef.onCheckItemClick($event)\"\n          ></bb-stop-checks-list>\n        </div>\n\n        <div data-role=\"paging-method-wrapper\">\n          <ng-container\n            *ngIf=\"\n              hostRef.isPagination(paginationConfig.paginationType);\n              else loadMore\n            \"\n          >\n            <bb-pagination-ui\n              *ngIf=\"!isLoading\"\n              [page]=\"paginationConfig.currentPage\"\n              [maxNavPages]=\"paginationConfig.maxNavPages\"\n              [totalItems]=\"paginationConfig.totalItems\"\n              [itemsPerPage]=\"paginationConfig.size\"\n              (pageChange)=\"hostRef.onPageChange($event)\"\n            ></bb-pagination-ui>\n          </ng-container>\n\n          <ng-template #loadMore>\n            <bb-load-button-ui\n              class=\"bb-pagination\"\n              color=\"secondary\"\n              i18n=\"\n                Load more payment orders|Load more button for loading more\n                payment orders@@manage-payments-list.load.more\"\n              [isLoading]=\"isLoading\"\n              [size]=\"paginationConfig.size\"\n              [from]=\"paginationConfig.currentPage\"\n              [totalCount]=\"paginationConfig.totalItems\"\n              (paramChange)=\"hostRef.onLoadMore($event)\"\n            >\n              Load more\n            </bb-load-button-ui>\n          </ng-template>\n        </div>\n      </ng-container>\n    </ng-container>\n  </ng-template>\n\n  <ng-template #loading>\n    <bb-loading-indicator-ui\n      i18n-text=\"@@manage-stop-checks.loading.text\"\n      text=\"Loading data...\"\n    >\n    </bb-loading-indicator-ui>\n  </ng-template>\n\n  <ng-template #emptyState>\n    <bb-empty-stop-checks-list></bb-empty-stop-checks-list>\n  </ng-template>\n\n  <ng-template #errorState>\n    <div class=\"bb-stop-checks-error__container\">\n      <bb-empty-state-ui\n        data-role=\"stop-checks-list-error-state\"\n        i18n-title=\"@@manage-stop-checks.error.generic.title\"\n        title=\"Page couldn't be load!\"\n        i18n-subtitle=\"@@manage-stop-checks.error.generic.subtitle\"\n        subtitle=\"Try to reload the page or contact the system administrator.\"\n        iconSize=\"xxl\"\n        iconModifier=\"error\"\n      ></bb-empty-state-ui>\n    </div>\n  </ng-template>\n\n  <bb-stop-checks-details\n    [isOpen]=\"hostRef.openDetails\"\n    (close)=\"hostRef.onCloseDetails()\"\n    [check]=\"hostRef.selectedCheck\"\n  ></bb-stop-checks-details>\n</ng-template>\n", components: [{ type: i0__namespace.forwardRef(function () { return StopChecksListComponent; }), selector: "bb-stop-checks-list", inputs: ["checks"], outputs: ["checkItemClick"] }, { type: i0__namespace.forwardRef(function () { return i5__namespace$1.PaginatorComponent; }), selector: "bb-pagination-ui", inputs: ["page", "boundaryLinks", "directionLinks", "totalItems", "itemsPerPage", "maxNavPages"], outputs: ["pageChange"] }, { type: i0__namespace.forwardRef(function () { return i6__namespace$1.LoadButtonComponent; }), selector: "bb-load-button-ui", inputs: ["isLoading", "from", "size", "circle", "totalCount"], outputs: ["paramChange"] }, { type: i0__namespace.forwardRef(function () { return i6__namespace.LoadingIndicatorComponent; }), selector: "bb-loading-indicator-ui", inputs: ["text", "loaderSize", "showDelay", "hasBackground", "inline"] }, { type: i0__namespace.forwardRef(function () { return StopChecksEmptyListComponent; }), selector: "bb-empty-stop-checks-list", inputs: ["isSearch"] }, { type: i0__namespace.forwardRef(function () { return i1__namespace$3.EmptyStateComponent; }), selector: "bb-empty-state-ui", inputs: ["title", "subtitle", "showIcon", "iconClasses", "iconModifier", "iconSize", "iconColor"] }, { type: i0__namespace.forwardRef(function () { return StopChecksDetailsComponent; }), selector: "bb-stop-checks-details", inputs: ["isOpen", "check"], outputs: ["close"] }], directives: [{ type: i0__namespace.forwardRef(function () { return StopChecksListContainerCustomizableDirective; }), selector: "[bbStopChecksListContainerCustomizableDirective]" }, { type: i0__namespace.forwardRef(function () { return i3__namespace.NgIf; }), selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i0__namespace.forwardRef(function () { return i3__namespace.NgTemplateOutlet; }), selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }], pipes: { "async": i0__namespace.forwardRef(function () { return i3__namespace.AsyncPipe; }) } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StopChecksListContainerComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-stop-checks-list-container',
                        templateUrl: 'stop-checks-list-container.component.html',
                    }]
            }], ctorParameters: function () {
            return [{ type: StopChecksListCommunicationService, decorators: [{
                            type: i0.Optional
                        }] }, { type: i1__namespace.ItemModel, decorators: [{
                            type: i0.Optional
                        }] }, { type: StopChecksListService }];
        } });
    var StopChecksListContainerCustomizableDirective = /** @class */ (function (_super) {
        __extends(StopChecksListContainerCustomizableDirective, _super);
        function StopChecksListContainerCustomizableDirective() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return StopChecksListContainerCustomizableDirective;
    }(i1$1.BbTemplate));
    /** @nocollapse */ StopChecksListContainerCustomizableDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StopChecksListContainerCustomizableDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    /** @nocollapse */ StopChecksListContainerCustomizableDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: StopChecksListContainerCustomizableDirective, selector: "[bbStopChecksListContainerCustomizableDirective]", usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StopChecksListContainerCustomizableDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: '[bbStopChecksListContainerCustomizableDirective]',
                    }]
            }] });

    var PayordStopChecksListWidgetAngComponent = /** @class */ (function () {
        function PayordStopChecksListWidgetAngComponent(preferences) {
            this.preferences = preferences;
        }
        Object.defineProperty(PayordStopChecksListWidgetAngComponent.prototype, "pageSize", {
            /**
             * Number of payments per page
             */
            set: function (value) {
                this.preferences.setPageSize(value);
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(PayordStopChecksListWidgetAngComponent.prototype, "maxNavPages", {
            /**
             * Max. number of navigation pages
             */
            set: function (value) {
                this.preferences.setMaxNavPages(value);
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(PayordStopChecksListWidgetAngComponent.prototype, "paginationType", {
            /**
             * Number of payments per page
             */
            set: function (value) {
                this.preferences.setPaginationType(value);
            },
            enumerable: false,
            configurable: true
        });
        return PayordStopChecksListWidgetAngComponent;
    }());
    /** @nocollapse */ PayordStopChecksListWidgetAngComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: PayordStopChecksListWidgetAngComponent, deps: [{ token: StopChecksListPropertiesService }], target: i0__namespace.ɵɵFactoryTarget.Component });
    /** @nocollapse */ PayordStopChecksListWidgetAngComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: PayordStopChecksListWidgetAngComponent, selector: "bb-payord-stop-checks-list-widget-ang", inputs: { pageSize: "pageSize", maxNavPages: "maxNavPages", paginationType: "paginationType" }, providers: [StopChecksListService, StopChecksListPropertiesService], ngImport: i0__namespace, template: " <bb-stop-checks-list-container></bb-stop-checks-list-container> ", isInline: true, components: [{ type: StopChecksListContainerComponent, selector: "bb-stop-checks-list-container" }] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: PayordStopChecksListWidgetAngComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-payord-stop-checks-list-widget-ang',
                        template: " <bb-stop-checks-list-container></bb-stop-checks-list-container> ",
                        providers: [StopChecksListService, StopChecksListPropertiesService],
                    }]
            }], ctorParameters: function () { return [{ type: StopChecksListPropertiesService }]; }, propDecorators: { pageSize: [{
                    type: i0.Input
                }], maxNavPages: [{
                    type: i0.Input
                }], paginationType: [{
                    type: i0.Input
                }] } });

    var components = [
        PayordStopChecksListWidgetAngComponent,
        StopChecksListContainerComponent,
        StopChecksEmptyListComponent,
        StopChecksListComponent,
        StopChecksDetailsComponent,
    ];
    var directives = [
        StopChecksListContainerCustomizableDirective,
        StopChecksEmptyListCustomizableDirective,
        StopChecksListCustomizableDirective,
        StopChecksDetailsCustomizableDirective,
    ];
    var uiModules = [
        i1$3.EmptyStateModule,
        i8.ButtonModule,
        i6.LoadingIndicatorModule,
        i2.TableModule,
        badge.BadgeModule,
        i4.IconModule,
        commonErrorState.ErrorCommonStateModule,
        i5.AmountModule,
        tooltipDirective.TooltipModule,
        i1$4.ModalModule,
        i2$1.HeaderModule,
        i5$1.PaginationModule,
        i6$1.LoadButtonModule,
    ];
    /**
     * Exports the UI components, directives and modules for the stop checks list widget
     */
    var PayordStopChecksListWidgetAngModule = /** @class */ (function () {
        function PayordStopChecksListWidgetAngModule() {
        }
        return PayordStopChecksListWidgetAngModule;
    }());
    /** @nocollapse */ PayordStopChecksListWidgetAngModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: PayordStopChecksListWidgetAngModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    /** @nocollapse */ PayordStopChecksListWidgetAngModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: PayordStopChecksListWidgetAngModule, declarations: [PayordStopChecksListWidgetAngComponent,
            StopChecksListContainerComponent,
            StopChecksEmptyListComponent,
            StopChecksListComponent,
            StopChecksDetailsComponent, StopChecksListContainerCustomizableDirective,
            StopChecksEmptyListCustomizableDirective,
            StopChecksListCustomizableDirective,
            StopChecksDetailsCustomizableDirective], imports: [i3.CommonModule, i1__namespace.BackbaseCoreModule, i1.PaymentOrdersUiModule, i1$3.EmptyStateModule,
            i8.ButtonModule,
            i6.LoadingIndicatorModule,
            i2.TableModule,
            badge.BadgeModule,
            i4.IconModule,
            commonErrorState.ErrorCommonStateModule,
            i5.AmountModule,
            tooltipDirective.TooltipModule,
            i1$4.ModalModule,
            i2$1.HeaderModule,
            i5$1.PaginationModule,
            i6$1.LoadButtonModule], exports: [PayordStopChecksListWidgetAngComponent,
            StopChecksListContainerComponent,
            StopChecksEmptyListComponent,
            StopChecksListComponent,
            StopChecksDetailsComponent, StopChecksListContainerCustomizableDirective,
            StopChecksEmptyListCustomizableDirective,
            StopChecksListCustomizableDirective,
            StopChecksDetailsCustomizableDirective, i1.PaymentOrdersUiModule, i1$3.EmptyStateModule,
            i8.ButtonModule,
            i6.LoadingIndicatorModule,
            i2.TableModule,
            badge.BadgeModule,
            i4.IconModule,
            commonErrorState.ErrorCommonStateModule,
            i5.AmountModule,
            tooltipDirective.TooltipModule,
            i1$4.ModalModule,
            i2$1.HeaderModule,
            i5$1.PaginationModule,
            i6$1.LoadButtonModule] });
    /** @nocollapse */ PayordStopChecksListWidgetAngModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: PayordStopChecksListWidgetAngModule, imports: [__spreadArray([
                i3.CommonModule,
                i1$1.BackbaseCoreModule.withConfig({
                    classMap: { PayordStopChecksListWidgetAngComponent: PayordStopChecksListWidgetAngComponent },
                }),
                i1.PaymentOrdersUiModule
            ], __read(uiModules)), i1.PaymentOrdersUiModule, i1$3.EmptyStateModule,
            i8.ButtonModule,
            i6.LoadingIndicatorModule,
            i2.TableModule,
            badge.BadgeModule,
            i4.IconModule,
            commonErrorState.ErrorCommonStateModule,
            i5.AmountModule,
            tooltipDirective.TooltipModule,
            i1$4.ModalModule,
            i2$1.HeaderModule,
            i5$1.PaginationModule,
            i6$1.LoadButtonModule] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: PayordStopChecksListWidgetAngModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: __spreadArray([
                            i3.CommonModule,
                            i1$1.BackbaseCoreModule.withConfig({
                                classMap: { PayordStopChecksListWidgetAngComponent: PayordStopChecksListWidgetAngComponent },
                            }),
                            i1.PaymentOrdersUiModule
                        ], __read(uiModules)),
                        declarations: __spreadArray(__spreadArray([], __read(components)), __read(directives)),
                        exports: __spreadArray(__spreadArray(__spreadArray(__spreadArray([], __read(components)), __read(directives)), [i1.PaymentOrdersUiModule]), __read(uiModules)),
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.PayordStopChecksListWidgetAngComponent = PayordStopChecksListWidgetAngComponent;
    exports.PayordStopChecksListWidgetAngModule = PayordStopChecksListWidgetAngModule;
    exports.StopChecksListCommunicationService = StopChecksListCommunicationService;
    exports["ӨStopChecksDetailsComponent"] = StopChecksDetailsComponent;
    exports["ӨStopChecksDetailsCustomizableDirective"] = StopChecksDetailsCustomizableDirective;
    exports["ӨStopChecksEmptyListComponent"] = StopChecksEmptyListComponent;
    exports["ӨStopChecksEmptyListCustomizableDirective"] = StopChecksEmptyListCustomizableDirective;
    exports["ӨStopChecksListComponent"] = StopChecksListComponent;
    exports["ӨStopChecksListContainerComponent"] = StopChecksListContainerComponent;
    exports["ӨStopChecksListContainerCustomizableDirective"] = StopChecksListContainerCustomizableDirective;
    exports["ӨStopChecksListCustomizableDirective"] = StopChecksListCustomizableDirective;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
