// @backbase/ui-ang/esm2015/_example/infinite-scroll/examples/infinite-scroll.example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/infinite-scroll";
import * as i2 from "@backbase/ui-ang/loading-indicator";
import * as i3 from "@angular/common";
export class InfiniteScrollExampleComponent {
    constructor() {
        this.items = [1, 2, 3, 4];
        this.complete = false;
        this.loading = false;
        this.maxCount = 42;
    }
    loadMore() {
        if (!this.loading) {
            this.loading = true;
            window.setTimeout(() => {
                this.loading = false;
                this.complete = this.items.length >= this.maxCount;
                this.generateMore(5);
            }, 1000);
        }
    }
    generateMore(count) {
        const length = Math.max(0, Math.min(this.maxCount, this.items.length + count) -
            this.items.length);
        const start = 1 + (this.items[this.items.length - 1] || 0);
        console.log(start, length);
        this.items = [
            ...this.items,
            ...new Array(length).fill(0).map((_, index) => start + index),
        ];
    }
}
InfiniteScrollExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InfiniteScrollExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
InfiniteScrollExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: InfiniteScrollExampleComponent, selector: "bb-infinite-scroll-example-ui", ngImport: i0, template: "<div style=\"max-height: 320px; overflow: scroll\">\n  <bb-infinite-scroll-ui\n    [disableScrollEnd]=\"complete || loading\"\n    (scrollEnd)=\"loadMore()\"\n  >\n    <ul>\n      <li *ngFor=\"let item of items\">Item {{ item }}</li>\n    </ul>\n    <p *ngIf=\"loading\">\n      <bb-loading-indicator-ui></bb-loading-indicator-ui>\n    </p>\n    <p *ngIf=\"complete\">All items loaded</p>\n  </bb-infinite-scroll-ui>\n</div>\n", components: [{ type: i1.InfiniteScrollComponent, selector: "bb-infinite-scroll-ui", inputs: ["offset", "debounce", "disableScrollEnd"], outputs: ["scrollEnd"] }, { type: i2.LoadingIndicatorComponent, selector: "bb-loading-indicator-ui", inputs: ["text", "loaderSize", "showDelay", "hasBackground", "inline"] }], directives: [{ type: i3.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InfiniteScrollExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-infinite-scroll-example-ui',
                    templateUrl: './infinite-scroll.example.component.html',
                }]
        }] });
