// @backbase/ui-ang/esm2015/common-error-state/common-error-state.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ErrorCommonStateComponent } from './common-error-state.component';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import * as i0 from "@angular/core";
export class ErrorCommonStateModule {
}
ErrorCommonStateModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ErrorCommonStateModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ErrorCommonStateModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ErrorCommonStateModule, declarations: [ErrorCommonStateComponent], imports: [CommonModule, EmptyStateModule], exports: [ErrorCommonStateComponent] });
ErrorCommonStateModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ErrorCommonStateModule, imports: [[CommonModule, EmptyStateModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ErrorCommonStateModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, EmptyStateModule],
                    declarations: [ErrorCommonStateComponent],
                    exports: [ErrorCommonStateComponent],
                }]
        }] });
