// @backbase/budget-widget-ang/esm2015/src/budgets-header/budgets-header.component.js
import { Component, EventEmitter, Output } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/icon";
import * as i2 from "@backbase/ui-ang/button";
export class BudgetsHeaderComponent {
    constructor() {
        /**
         * EventEmitter for triggering a createBudget event.
         */
        this.createBudget = new EventEmitter();
        this.currentDate = new Date();
    }
    triggerCreateBudget() {
        this.createBudget.emit();
    }
}
BudgetsHeaderComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetsHeaderComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
BudgetsHeaderComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: BudgetsHeaderComponent, selector: "bb-budgets-header-ui", outputs: { createBudget: "createBudget" }, ngImport: i0, template: "<div class=\"bb-block bb-block--xl\">\n  <div class=\"bb-stack\">\n    <button \n      bbButton \n      color=\"primary\"\n      class=\"bb-stack__item bb-stack__item--push-right\" \n      (click)=\"triggerCreateBudget()\"\n    >\n      <bb-icon-ui \n        name=\"add\" \n        size=\"md\" \n        class=\"bb-stack__item bb-stack__item--spacing-xs\"\n      ></bb-icon-ui>\n      <span i18n=\"New budget|Label for a button that create a new budget@@budgets.new.button.lable\">New budget</span>\n    </button>\n  </div>\n</div>", components: [{ type: i1.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], directives: [{ type: i2.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetsHeaderComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-budgets-header-ui',
                    templateUrl: './budgets.header.component.html',
                }]
        }], propDecorators: { createBudget: [{
                type: Output
            }] } });
