// @backbase/product-summary-ui/esm2015/src/bb-product-needs-attention-item/product-needs-attention-item.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { AmountModule } from '@backbase/ui-ang/amount';
import { IconModule } from '@backbase/ui-ang/icon';
import { KeyboardClickModule } from '@backbase/ui-ang/keyboard-click-directive';
import { BadgeModule } from '@backbase/ui-ang/badge';
import { ProductNeedsAttentionItemComponent, ProductNeedsAttentionItemCustomizableDirective, } from './product-needs-attention-item.component';
import { ProductSummaryAccountLogoModule } from '../bb-product-summary-account-logo-ui/bb-product-summary-account-logo.module';
import { ProductNumberModule } from '../bb-product-number/product-number.module';
import { PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import { ButtonModule } from '@backbase/ui-ang/button';
import * as i0 from "@angular/core";
const uiModules = [
    IconModule,
    AmountModule,
    KeyboardClickModule,
    NgbModule,
    BadgeModule,
    ProductSummaryAccountLogoModule,
    ProductNumberModule,
    PaymentCardNumberModule,
    ButtonModule,
];
const components = [ProductNeedsAttentionItemComponent, ProductNeedsAttentionItemCustomizableDirective];
export class ProductNeedsAttentionItemModule {
}
ProductNeedsAttentionItemModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductNeedsAttentionItemModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductNeedsAttentionItemModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductNeedsAttentionItemModule, declarations: [ProductNeedsAttentionItemComponent, ProductNeedsAttentionItemCustomizableDirective], imports: [CommonModule, IconModule,
        AmountModule,
        KeyboardClickModule,
        NgbModule,
        BadgeModule,
        ProductSummaryAccountLogoModule,
        ProductNumberModule,
        PaymentCardNumberModule,
        ButtonModule], exports: [ProductNeedsAttentionItemComponent, ProductNeedsAttentionItemCustomizableDirective, IconModule,
        AmountModule,
        KeyboardClickModule,
        NgbModule,
        BadgeModule,
        ProductSummaryAccountLogoModule,
        ProductNumberModule,
        PaymentCardNumberModule,
        ButtonModule] });
ProductNeedsAttentionItemModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductNeedsAttentionItemModule, imports: [[CommonModule, ...uiModules], IconModule,
        AmountModule,
        KeyboardClickModule,
        NgbModule,
        BadgeModule,
        ProductSummaryAccountLogoModule,
        ProductNumberModule,
        PaymentCardNumberModule,
        ButtonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductNeedsAttentionItemModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules],
                    declarations: [...components],
                    exports: [...components, ...uiModules],
                }]
        }] });
