// @backbase/ui-ang/esm2015/_example/table/examples/table.example.component.js
import { Component } from '@angular/core';
import { of } from 'rxjs';
import { countryData } from './countries';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/table";
import * as i2 from "@angular/common";
const sortInteger = {
    asc: 1,
    desc: -1,
};
export class TableExampleComponent {
    constructor() {
        this.countries = countryData;
        this.countries$ = of(this.countries);
        this.sortDirection = 0;
        this.sortKey = '';
        this.pathFlagAssets = 'https://upload.wikimedia.org/wikipedia/commons/';
    }
    onRowClick(country) {
        alert(`You clicked on ${country.name}`);
    }
    onSort({ column, direction }) {
        this.sortDirection = sortInteger[direction] || 0;
        this.sortKey = column;
        this.countries$ = of(this.sort(this.countries, this.sortKey, this.sortDirection));
    }
    sort(countries, column, direction) {
        return [...countries].sort((a, b) => direction * this.compare(a[column], b[column]));
    }
    compare(v1, v2) {
        return v1 === v2 ? 0 : v1 < v2 ? -1 : 1;
    }
    getAriaSort(key) {
        if (this.sortKey === key) {
            if (this.sortDirection < 0) {
                return 'descending';
            }
            else if (this.sortDirection > 0) {
                return 'ascending';
            }
        }
        return null;
    }
}
TableExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
TableExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: TableExampleComponent, selector: "bb-table-example-ui", ngImport: i0, template: "<table\n  [bbTable]=\"countries$ | async\"\n  bbTableFocus\n  (rowClick)=\"onRowClick($event)\"\n  (sort)=\"onSort($event)\"\n  class=\"table table-hover\"\n>\n  <thead>\n    <tr>\n      <th\n        [attr.aria-sort]=\"getAriaSort('name')\"\n        [bbSortable]=\"'name'\"\n        scope=\"col\"\n        tabindex=\"0\"\n      >\n        Country\n      </th>\n      <th\n        [attr.aria-sort]=\"getAriaSort('area')\"\n        [bbSortable]=\"'area'\"\n        scope=\"col\"\n        tabindex=\"0\"\n      >\n        Area (km\u00B2)\n      </th>\n      <th\n        [attr.aria-sort]=\"getAriaSort('population')\"\n        [bbSortable]=\"'population'\"\n        scope=\"col\"\n        tabindex=\"0\"\n      >\n        Population\n      </th>\n      <th></th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr *ngFor=\"let country of countries$ | async\" [bbRow]=\"country\">\n      <td>\n        {{ country.name }}\n      </td>\n      <td>\n        {{ country.area | number }}\n      </td>\n      <td>\n        {{ country.population | number }}\n      </td>\n      <td>\n        <img\n          [alt]=\"'Flag of ' + country.name\"\n          [src]=\"pathFlagAssets + country.flag\"\n          width=\"32\"\n        />\n      </td>\n    </tr>\n  </tbody>\n</table>\n", directives: [{ type: i1.TableDirective, selector: "table[bbTable]", inputs: ["bbTable"], outputs: ["selectRow", "rowClick", "sort"] }, { type: i1.TableFocusDirective, selector: "[bbTableFocus]", inputs: ["focusFirstRowOnChanges"] }, { type: i1.TableSortableDirective, selector: "th[bbSortable]", inputs: ["bbSortable", "direction"], outputs: ["sort"] }, { type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { type: i1.TableRowDirective, selector: "tr[bbRow]", inputs: ["bbRow"] }], pipes: { "async": i2.AsyncPipe, "number": i2.DecimalPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-table-example-ui',
                    templateUrl: './table.example.component.html',
                }]
        }] });
