// @backbase/ui-ang/esm2015/iban-directive/iban-validator.directive.js
import { Directive } from '@angular/core';
import { NG_VALIDATORS } from '@angular/forms';
import { isValidIBANFormControl } from '@backbase/ui-ang/iban-lib';
import * as i0 from "@angular/core";
export class IbanValidatorDirective {
    // emits invalidIban
    validate(control) {
        return control && control.value ? isValidIBANFormControl(control.value) : null;
    }
}
IbanValidatorDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IbanValidatorDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
IbanValidatorDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: IbanValidatorDirective, selector: "[bbIbanValidatorDir]", providers: [{ provide: NG_VALIDATORS, useExisting: IbanValidatorDirective, multi: true }], ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IbanValidatorDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbIbanValidatorDir]',
                    providers: [{ provide: NG_VALIDATORS, useExisting: IbanValidatorDirective, multi: true }],
                }]
        }] });
