// @backbase/ui-ang/esm2015/_example/product-item-term-deposit/showcase/product-item-term-deposit-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ProductItemTermDepositShowcaseComponent } from './product-item-term-deposit-showcase.component';
import { ProductItemTermDepositModule } from '@backbase/ui-ang/product-item-term-deposit';
import * as i0 from "@angular/core";
export class ProductItemTermDepositShowcaseModule {
}
ProductItemTermDepositShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemTermDepositShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductItemTermDepositShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemTermDepositShowcaseModule, declarations: [ProductItemTermDepositShowcaseComponent], imports: [CommonModule, ProductItemTermDepositModule] });
ProductItemTermDepositShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemTermDepositShowcaseModule, imports: [[CommonModule, ProductItemTermDepositModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemTermDepositShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ProductItemTermDepositModule],
                    declarations: [ProductItemTermDepositShowcaseComponent],
                    entryComponents: [ProductItemTermDepositShowcaseComponent],
                }]
        }] });
