// @backbase/ui-ang/esm2015/_example/input-number/examples/input-number.example.component.js
import { Component } from '@angular/core';
import { Validators } from '@angular/forms';
import * as i0 from "@angular/core";
import * as i1 from "@angular/forms";
import * as i2 from "@backbase/ui-ang/input-number";
import * as i3 from "@backbase/ui-ang/input-validation-message";
import * as i4 from "@angular/common";
export class InputNumberExampleComponent {
    constructor(formBuilder) {
        this.formBuilder = formBuilder;
        this.vForm = this.formBuilder.group({
            count: [
                '',
                Validators.compose([
                    Validators.required,
                    Validators.min(2),
                    Validators.max(42),
                ]),
            ],
        });
    }
    isFieldValid(fieldName) {
        const control = this.getControl(fieldName);
        return control && (control.valid || control.untouched);
    }
    getControl(fieldName) {
        var _a;
        return (_a = this.vForm) === null || _a === void 0 ? void 0 : _a.controls[fieldName];
    }
    hasError(field, type) {
        const fieldControl = this.getControl(field);
        if (!fieldControl || !fieldControl.errors) {
            return;
        }
        return fieldControl.errors[type];
    }
    hasRequiredError(field) {
        const fieldControl = this.getControl(field);
        if (!fieldControl || !fieldControl.errors) {
            return;
        }
        return fieldControl.errors.required;
    }
    hasMinError(field) {
        return this.hasError(field, 'min');
    }
    hasMaxError(field) {
        return this.hasError(field, 'max');
    }
}
InputNumberExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputNumberExampleComponent, deps: [{ token: i1.FormBuilder }], target: i0.ɵɵFactoryTarget.Component });
InputNumberExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: InputNumberExampleComponent, selector: "bb-input-number-example-ui", ngImport: i0, template: "<form [formGroup]=\"vForm\">\n  <bb-input-number-ui\n    label=\"Count\"\n    placeholder=\"Enter number...\"\n    formControlName=\"count\"\n    [aria-describedby]=\"'input-number-showcase-error'\"\n    [min]=\"2\"\n    [max]=\"42\"\n    [required]=\"true\"\n    #numberInput\n  >\n  </bb-input-number-ui>\n  <bb-input-validation-message-ui\n    id=\"input-number-showcase-error\"\n    [showErrors]=\"vForm?.dirty && vForm?.get('count')?.errors\"\n  >\n    <p *ngIf=\"hasRequiredError('count')\">Field is required.</p>\n    <p *ngIf=\"hasMinError('count')\">\n      Minimum number allowed is: {{ numberInput.min }}\n    </p>\n    <p *ngIf=\"hasMaxError('count')\">\n      Maximum number allowed is: {{ numberInput.max }}\n    </p>\n  </bb-input-validation-message-ui>\n</form>\n", components: [{ type: i2.InputNumberComponent, selector: "bb-input-number-ui", inputs: ["placeholder", "max", "min", "step", "readonly", "autocomplete", "ariaLabel", "allowNonDigits", "allowNegativeSign"] }, { type: i3.InputValidationMessageComponent, selector: "bb-input-validation-message-ui", inputs: ["showErrors", "messageId"] }], directives: [{ type: i1.ɵNgNoValidate, selector: "form:not([ngNoForm]):not([ngNativeValidate])" }, { type: i1.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { type: i1.FormGroupDirective, selector: "[formGroup]", inputs: ["formGroup"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { type: i1.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { type: i1.FormControlName, selector: "[formControlName]", inputs: ["disabled", "formControlName", "ngModel"], outputs: ["ngModelChange"] }, { type: i1.RequiredValidator, selector: ":not([type=checkbox])[required][formControlName],:not([type=checkbox])[required][formControl],:not([type=checkbox])[required][ngModel]", inputs: ["required"] }, { type: i4.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputNumberExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-input-number-example-ui',
                    templateUrl: './input-number.example.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.FormBuilder }]; } });
