// @backbase/messages-create-message-modal-widget-ang/esm2015/src/create-message-form/create-message-form.service.js
import { Injectable } from '@angular/core';
import { catchError } from 'rxjs/operators';
import { of } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/messages-v5-http-ang";
export class CreateMessageFormService {
    constructor(messagesService) {
        this.messagesService = messagesService;
    }
    /**
     * Get topics
     *
     * @returns Topics stream
     */
    getTopics() {
        return this.messagesService.getTopics().pipe(catchError(() => of([])));
    }
    /**
     * Create message
     *
     * @param params Create message request body
     * @returns Post message stream
     */
    createMessage(params) {
        return this.messagesService.postMessages({ presentationMessagePostRequestBody: params });
    }
    /**
     * Save message draft
     *
     * @param body Save draft request body
     * @returns Safe draft stream
     */
    saveDraft(body) {
        return this.messagesService.postMessageDrafts({ messageDraftsPostRequestBody: body });
    }
    /**
     * Get message draft
     *
     * @param draftId Draft id
     * @returns Draft stream
     */
    getDraft(draftId) {
        return this.messagesService.getMessageDraft({ draftId });
    }
    /**
     * Delete draft
     *
     * @param draftId Draft id
     * @returns Delete draft stream
     */
    deleteDraft(draftId) {
        return this.messagesService.deleteMessageDrafts({ ids: [draftId] });
    }
    /**
     * Update draft
     *
     * @param draftId Draft id
     * @param body Update draft request body
     * @returns Update draft stream
     */
    updateDraft(draftId, body) {
        return this.messagesService.putMessageDraft({ draftId, messageDraftPutRequestBody: body });
    }
}
CreateMessageFormService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: CreateMessageFormService, deps: [{ token: i1.MessagecenterHttpService }], target: i0.ɵɵFactoryTarget.Injectable });
CreateMessageFormService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: CreateMessageFormService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: CreateMessageFormService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.MessagecenterHttpService }]; } });
