// @backbase/messages-create-message-modal-widget-ang/esm2015/src/create-message-form/create-message-form.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { MessagesEncodingService, MessagesManipulationConfirmModalModule, MessagesUploadAttachmentsModule, PreventBubbleDownModule, } from '@backbase/messages-common';
import { ButtonModule } from '@backbase/ui-ang/button';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import { DropdownSingleSelectModule } from '@backbase/ui-ang/dropdown-single-select';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { RichTextEditorModule } from '@backbase/ui-ang/rich-text-editor';
import { CreateMessageFormComponent, CreateMessageFormCustomizableDirective } from './create-message-form.component';
import * as i0 from "@angular/core";
const components = [CreateMessageFormComponent];
const directives = [CreateMessageFormCustomizableDirective];
const uiModules = [
    ButtonModule,
    DropdownSingleSelectModule,
    InputValidationMessageModule,
    InputTextModule,
    LoadButtonModule,
    RichTextEditorModule,
];
const innerModules = [MessagesManipulationConfirmModalModule, MessagesUploadAttachmentsModule];
export class CreateMessageFormModule {
}
CreateMessageFormModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: CreateMessageFormModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CreateMessageFormModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: CreateMessageFormModule, declarations: [CreateMessageFormComponent, CreateMessageFormCustomizableDirective], imports: [CommonModule, ReactiveFormsModule, PreventBubbleDownModule, ButtonModule,
        DropdownSingleSelectModule,
        InputValidationMessageModule,
        InputTextModule,
        LoadButtonModule,
        RichTextEditorModule, MessagesManipulationConfirmModalModule, MessagesUploadAttachmentsModule], exports: [CreateMessageFormComponent, CreateMessageFormCustomizableDirective, ButtonModule,
        DropdownSingleSelectModule,
        InputValidationMessageModule,
        InputTextModule,
        LoadButtonModule,
        RichTextEditorModule, MessagesManipulationConfirmModalModule, MessagesUploadAttachmentsModule] });
CreateMessageFormModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: CreateMessageFormModule, providers: [MessagesEncodingService], imports: [[CommonModule, ReactiveFormsModule, PreventBubbleDownModule, ...uiModules, ...innerModules], ButtonModule,
        DropdownSingleSelectModule,
        InputValidationMessageModule,
        InputTextModule,
        LoadButtonModule,
        RichTextEditorModule, MessagesManipulationConfirmModalModule, MessagesUploadAttachmentsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: CreateMessageFormModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ReactiveFormsModule, PreventBubbleDownModule, ...uiModules, ...innerModules],
                    declarations: [...components, ...directives],
                    exports: [...components, ...directives, ...uiModules, ...innerModules],
                    providers: [MessagesEncodingService],
                }]
        }] });
