// @backbase/retail-loans-journey-ang/esm2015/lib/components/loan-details/tabs/loan-info-wrapper/loan-info-wrapper.component.js
import { Component } from '@angular/core';
import { getLoanInfoDataConfig } from './loan-info-wrapper.config';
import { LoanDetailsBaseComponent } from '../../../base/loan-details/loan-details-base-component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/loans-collection-ui-ang";
import * as i2 from "./loan-info-retail/loan-info-retail.component";
import * as i3 from "@angular/common";
export class LoanInfoWrapperComponent extends LoanDetailsBaseComponent {
    /**
     * Configuration for loan info container
     */
    getDataConfig({ viewType }) {
        return getLoanInfoDataConfig(viewType);
    }
}
/** @nocollapse */ LoanInfoWrapperComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoWrapperComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ LoanInfoWrapperComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: LoanInfoWrapperComponent, selector: "bb-loan-info-wrapper", usesInheritance: true, ngImport: i0, template: "<bb-loans-main-container\n  [isLoadingState]=\"isLoading$ | async\"\n  [isErrorState]=\"isErrorState$ | async\"\n  rootWrapperClasses=\"\"\n  contentWrapperClasses=\"\"\n>\n  <bb-loan-info-retail\n    *ngIf=\"loan$ | async as loan\"\n    [loan]=\"loan\"\n    [dataConfig]=\"getDataConfig(loan)\"\n    [isCurrencySymbolUsed]=\"true\"\n  ></bb-loan-info-retail>\n</bb-loans-main-container>\n", components: [{ type: i1.LoansMainContainerComponent, selector: "bb-loans-main-container", inputs: ["isErrorState", "isLoadingState", "isEmptyState", "containerHeightFitToContent", "contentWrapperClasses", "rootWrapperClasses", "customErrorState", "customLoadingState", "customEmptyState"] }, { type: i2.LoanInfoRetailComponent, selector: "bb-loan-info-retail" }], directives: [{ type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "async": i3.AsyncPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoWrapperComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-loan-info-wrapper',
                    templateUrl: './loan-info-wrapper.component.html',
                }]
        }] });
