// @backbase/retail-loans-journey-ang/esm2015/lib/components/loan-details/tabs/loan-info-wrapper/loan-info-wrapper.config.js
import { LoanDefaultViewType, LoanInfoCollateralSectionComponent, LoanInfoDocumentsSectionComponent, LoanInfoEscrowSectionComponent, LoanInfoItemType, LoanInfoTermItemComponent, } from '@backbase/loans-collection-ui-ang';
import { titles } from '../../../../config/titles.config';
export const showAmountItem = (amountKey) => (loan) => Boolean(loan.currencyCode && loan[amountKey]);
export const showCollaterals = (item) => { var _a; return Boolean((_a = item.collateral) === null || _a === void 0 ? void 0 : _a.some((collateralItem) => Object.values(collateralItem).some(Boolean))); };
export const showBorrowers = (item) => { var _a; return Boolean((_a = item.borrower) === null || _a === void 0 ? void 0 : _a.length); };
export const showDocuments = (item) => { var _a; return Boolean((_a = item.documents) === null || _a === void 0 ? void 0 : _a.some(({ documentId }) => documentId)); };
export const showTerm = (item) => Boolean(item.termUnit && item.termCount);
export const showDefaultSettlementAccountName = (item) => Boolean(item.defaultSettlementAccountName && item.defaultSettlementAccountNumber);
export const showEscrowSection = (item) => Boolean(item.currencyCode && item.escrow && Object.values(item.escrow).some(Boolean));
const cssClasses = 'col-sm-6 col-12';
const generalSection = {
    title: titles.sections.general,
    items: [
        {
            cssClasses,
            title: titles.items.name,
            fieldKey: 'name',
            fieldType: LoanInfoItemType.String,
        },
        {
            cssClasses,
            title: titles.items.type,
            fieldKey: 'loanTypeAlias',
            fieldType: LoanInfoItemType.String,
        },
        {
            cssClasses,
            title: titles.items.contractNumber,
            fieldKey: 'contractNumber',
            fieldType: LoanInfoItemType.String,
        },
        {
            cssClasses,
            title: titles.items.accountNumber,
            fieldKey: 'accountNumber',
            fieldType: LoanInfoItemType.String,
        },
        {
            cssClasses,
            title: titles.items.branchCode,
            fieldKey: 'branchCode',
            fieldType: LoanInfoItemType.String,
        },
        {
            cssClasses,
            title: titles.items.currency,
            fieldKey: 'currencyCode',
            fieldType: LoanInfoItemType.String,
        },
        {
            cssClasses,
            title: titles.items.defaultSettlementAccountName,
            fieldKey: 'defaultSettlementAccountName',
            fieldType: LoanInfoItemType.String,
            showItem: showDefaultSettlementAccountName,
        },
        {
            cssClasses,
            title: titles.items.defaultSettlementAccountNumber,
            fieldKey: 'defaultSettlementAccountNumber',
            fieldType: LoanInfoItemType.String,
        },
        {
            cssClasses,
            title: titles.items.borrowers,
            fieldKey: 'borrower',
            fieldType: LoanInfoItemType.StringArray,
            showItem: showBorrowers,
        },
    ],
};
const statusSectionBaseItems = [
    {
        cssClasses,
        title: titles.items.status,
        fieldKey: 'loanStatus',
        fieldType: LoanInfoItemType.String,
    },
    {
        cssClasses,
        title: titles.items.startDate,
        fieldKey: 'startDate',
        fieldType: LoanInfoItemType.Date,
    },
];
const paymentInterestsSectionBaseItems = [
    {
        cssClasses,
        title: titles.items.feesDue,
        fieldKey: 'feesDue',
        fieldType: LoanInfoItemType.Amount,
        showItem: showAmountItem('feesDue'),
    },
    {
        cssClasses,
        title: titles.items.accruedInterestAmount,
        fieldKey: 'interestDue',
        fieldType: LoanInfoItemType.Amount,
        showItem: showAmountItem('interestDue'),
    },
    {
        cssClasses,
        title: titles.items.interestRate,
        fieldKey: 'interestRate',
        fieldType: LoanInfoItemType.Percent,
    },
    {
        cssClasses,
        title: titles.items.principalPaymentFrequency,
        fieldKey: 'paymentFrequency',
        fieldType: LoanInfoItemType.PaymentFrequency,
    },
    {
        cssClasses,
        title: titles.items.interestPaymentFrequency,
        fieldKey: 'interestPaymentFrequency',
        fieldType: LoanInfoItemType.PaymentFrequency,
    },
];
const outstandingAmountItem = {
    cssClasses,
    title: titles.items.outstandingAmount,
    fieldKey: 'outstandingAmount',
    fieldType: LoanInfoItemType.Amount,
    showItem: showAmountItem('outstandingAmount'),
};
const overdueAmountItem = {
    cssClasses,
    title: titles.items.overdueAmount,
    fieldKey: 'inArrearsAmount',
    fieldType: LoanInfoItemType.Amount,
    showItem: showAmountItem('inArrearsAmount'),
};
const documentsSection = {
    title: titles.sections.documents,
    items: [
        {
            cssClasses: 'col-sm-10 col-12',
            componentRef: LoanInfoDocumentsSectionComponent,
            showItem: showDocuments,
        },
    ],
};
const statusMortgageTermSection = {
    title: titles.sections.status,
    items: [
        ...statusSectionBaseItems,
        {
            cssClasses,
            title: titles.items.maturityDate,
            fieldKey: 'endDate',
            fieldType: LoanInfoItemType.Date,
        },
        {
            cssClasses: 'col-6',
            componentRef: LoanInfoTermItemComponent,
            showItem: showTerm,
        },
    ],
};
const financialDetailsMortgageTermSection = {
    title: titles.sections.financialDetails,
    items: [
        outstandingAmountItem,
        {
            cssClasses,
            title: titles.items.remainingPrincipal,
            fieldKey: 'drawnAmount',
            fieldType: LoanInfoItemType.Amount,
            showItem: showAmountItem('drawnAmount'),
        },
        {
            cssClasses,
            title: titles.items.amountRepaid,
            fieldKey: 'paidAmount',
            fieldType: LoanInfoItemType.Amount,
            showItem: showAmountItem('paidAmount'),
        },
        overdueAmountItem,
        {
            cssClasses,
            title: titles.items.originalPrincipal,
            fieldKey: 'creditLimit',
            fieldType: LoanInfoItemType.Amount,
            showItem: showAmountItem('creditLimit'),
        },
        ...paymentInterestsSectionBaseItems,
    ],
};
const colleteralSection = {
    title: titles.sections.collateral,
    items: [
        {
            cssClasses: 'col-12',
            componentRef: LoanInfoCollateralSectionComponent,
            showItem: showCollaterals,
        },
    ],
};
/**
 * Loan info config for LineOfCredit type
 */
export const loanInfoLineOfCreditDataConfig = [
    generalSection,
    {
        title: titles.sections.status,
        items: [
            ...statusSectionBaseItems,
            {
                cssClasses,
                title: titles.items.endDate,
                fieldKey: 'endDate',
                fieldType: LoanInfoItemType.Date,
            },
        ],
    },
    {
        title: titles.sections.financialDetails,
        items: [
            outstandingAmountItem,
            overdueAmountItem,
            {
                cssClasses,
                title: titles.items.availableBalance,
                fieldKey: 'availableBalance',
                fieldType: LoanInfoItemType.Amount,
                showItem: showAmountItem('availableBalance'),
            },
            {
                cssClasses,
                title: titles.items.creditLimit,
                fieldKey: 'creditLimit',
                fieldType: LoanInfoItemType.Amount,
                showItem: showAmountItem('creditLimit'),
            },
            {
                cssClasses,
                title: titles.items.drawnAmount,
                fieldKey: 'drawnAmount',
                fieldType: LoanInfoItemType.Amount,
                showItem: showAmountItem('drawnAmount'),
            },
            ...paymentInterestsSectionBaseItems,
        ],
    },
    documentsSection,
];
/**
 * Loan info config for Mortgage loan
 */
export const loanInfoMortgageDataConfig = [
    generalSection,
    statusMortgageTermSection,
    financialDetailsMortgageTermSection,
    {
        title: titles.sections.escrow,
        items: [
            {
                cssClasses: 'col-12',
                componentRef: LoanInfoEscrowSectionComponent,
                showItem: showEscrowSection,
            },
        ],
    },
    colleteralSection,
    documentsSection,
];
/**
 * Loan info config for Term loan
 */
export const loanInfoTermDataConfig = [
    generalSection,
    statusMortgageTermSection,
    financialDetailsMortgageTermSection,
    colleteralSection,
    documentsSection,
];
export function getLoanInfoDataConfig(viewType) {
    switch (viewType) {
        case LoanDefaultViewType.LineOfCredit:
            return loanInfoLineOfCreditDataConfig;
        case LoanDefaultViewType.TermLoan:
            return loanInfoTermDataConfig;
        case LoanDefaultViewType.Mortgage:
            return loanInfoMortgageDataConfig;
        default:
            throw new Error(`[bb-loans]: "${viewType}" view type is not configured`);
    }
}
