// @backbase/contact-common-ang/esm2015/lib/components/account-card-view/account-card-view.component.js
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { addressFromAccountHolder, addressFromBank, isObjectTruthy } from '../../helpers/address';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/icon";
import * as i2 from "@backbase/ui-ang/header";
import * as i3 from "../contact-full-address/contact-full-address.component";
import * as i4 from "./account-card-view.directive";
import * as i5 from "@angular/common";
import * as i6 from "@backbase/ui-ang/button";
import * as i7 from "@backbase/ui-ang/tooltip-directive";
import * as i8 from "@backbase/ui-ang/iban-pipe";
/**
 * The component is for showing contact account in a card view
 *
 * The following extension slots are available in this component:
 *  - `AccountCardViewCustomizableDirective`
 *
 * @see AccountCardViewCustomizableDirective
 * @usageNotes
 *
 * ### Display contact account data in card
 *
 * ```html
 *  <bb-account-card-view
 *   [index]="hostRef.index"
 *   [isEditMode]="true"
 *   [account]="hostRef.account"
 *   [bankAddress]="hostRef.bankAddress"
 *   [accountHolderAddress]="hostRef.accountHolderAddress"
 *   (deleteAccount)="hostRef.onDeleteAccount($event)"
 *   (editAccount)="hostRef.onEditAccount($event)">
 * </bb-account-card-view>
 * ```
 * @ngModule AccountCardViewModule
 */
export class AccountCardViewComponent {
    constructor() {
        this.hostRef = this;
        /**
         * Current index of the account Information
         */
        this.index = 0;
        /**
         * Flag that denotes if the account information can be edited. Defaults to `false`
         */
        this.isEditMode = false;
        /**
         * Event emitted when account is editted
         */
        this.editAccount = new EventEmitter();
        /**
         * Event emitted when account is deleted
         */
        this.deleteAccount = new EventEmitter();
    }
    /**
     * OnInit - set up initial values.
     */
    ngOnInit() {
        this.bankAddress = this.bankAddress || addressFromBank(this.account);
        this.accountHolderAddress = this.accountHolderAddress || addressFromAccountHolder(this.account);
    }
    /**
     * Emits the edit Account event
     */
    onEditAccount() {
        this.editAccount.emit(this.index);
    }
    /**
     * Emits the delete account event
     */
    onDeleteAccount() {
        this.deleteAccount.emit(this.index);
    }
    /**
     * Verifies if the address is valid.
     *
     * @param {Address} address object
     * @returns {boolean} valid or not.
     */
    isValidAddress(address) {
        return isObjectTruthy(address);
    }
}
/** @nocollapse */ AccountCardViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountCardViewComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ AccountCardViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AccountCardViewComponent, selector: "bb-account-card-view", inputs: { index: "index", account: "account", isEditMode: "isEditMode", bankAddress: "bankAddress", accountHolderAddress: "accountHolderAddress" }, outputs: { editAccount: "editAccount", deleteAccount: "deleteAccount" }, ngImport: i0, template: "<ng-container bbAccountCardViewCustomizable [bbHostRef]=\"hostRef\"></ng-container>\n<ng-template bbAccountCardViewCustomizable let-hostRef>\n  <div class=\"bb-stack\">\n    <bb-icon-ui class=\"bb-stack__item bb-stack__item--align-top mt-1\" name=\"accounts\"></bb-icon-ui>\n    <div class=\"bb-stack__item bb-stack__item--fill\">\n      <div class=\"bb-stack bb-block--md\">\n        <div class=\"bb-stack__item bb-stack__item--fill\">\n          <bb-header-ui\n            headingType=\"h3\"\n            [heading]=\"hostRef.account.name\"\n            headingClasses=\"bb-stack__item bb-stack__item--fill break-word\"\n            data-role=\"contact-details-account-name\"\n          >\n          </bb-header-ui>\n          <div class=\"bb-text-support\">{{hostRef.account.alias}}</div>\n        </div>\n        <ng-container *ngIf=\"hostRef.isEditMode\">\n          <button\n            bbButton\n            data-role=\"edit-account-button\"\n            [circle]=\"true\"\n            size=\"sm\"\n            color=\"link\"\n            (click)=\"hostRef.onEditAccount()\"\n            bbTooltip=\"Edit account\"\n            triggers=\"hover focus\"\n            i18n-bbTooltip=\"@@contact-address-data.contactaddress.card.btn.edit.label\"\n          >\n            <bb-icon-ui class=\"text-dark\" name=\"edit\"></bb-icon-ui>\n          </button>\n          <button\n            bbButton\n            [circle]=\"true\"\n            size=\"sm\"\n            color=\"link\"\n            (click)=\"hostRef.onDeleteAccount()\"\n            bbTooltip=\"Delete account\"\n            triggers=\"hover focus\"\n            i18n-bbTooltip=\"@@contact-address-data.contactaddress.card.btn.delete.label\"\n            data-role=\"account-contact-delete\"\n          >\n            <bb-icon-ui class=\"text-dark\" name=\"delete\"></bb-icon-ui>\n          </button>\n        </ng-container>\n      </div>\n      <dl>\n        <div class=\"bb-block bb-block--md\">\n          <div *ngIf=\"hostRef.account.IBAN\" class=\"bb-inline-stack bb-block--xs bb-block--full-width text-dark\">\n            <dt class=\"bb-inline-stack__item bb-text-bold\" i18n=\"@@bb-account-card-view.iban.label\">IBAN:</dt>\n            <dd class=\"bb-inline-stack__item break-word bb-block--no-margin\" data-role=\"contact-details-account-number\">\n              {{hostRef.account.IBAN | bbIban}}\n            </dd>\n          </div>\n          <div *ngIf=\"hostRef.account.accountNumber\" class=\"bb-stack bb-block--xs bb-block--full-width text-dark\">\n            <dt class=\"bb-inline-stack__item bb-text-bold\" i18n=\"@@bb-account-card-view.bban.label\">BBAN:</dt>\n            <dd class=\"bb-inline-stack__item break-word bb-block--no-margin\" data-role=\"contact-details-account-number\">\n              {{hostRef.account.accountNumber}}\n            </dd>\n          </div>\n          <div *ngIf=\"hostRef.account.accountType\" class=\"bb-inline-stack bb-block--xs bb-block--full-width text-dark\">\n            <dt class=\"bb-inline-stack__item bb-text-bold\" i18n=\"@@bb-account-card-view.account-type.label\">\n              Account type:\n            </dt>\n            <dd class=\"break-word bb-inline-stack__item bb-block--no-margin\">{{hostRef.account.accountType}}</dd>\n          </div>\n          <div *ngIf=\"hostRef.account.bankName\" class=\"bb-inline-stack bb-block--xs bb-block--full-width text-dark\">\n            <dt class=\"bb-inline-stack__item bb-text-bold\" i18n=\"@@bb-account-card-view.bank-name.label\">Bank:</dt>\n            <dd data-role=\"contact-details-bank-name\" class=\"bb-inline-stack__item break-word bb-block--no-margin\">\n              {{hostRef.account.bankName}}\n            </dd>\n          </div>\n          <div *ngIf=\"hostRef.account.bankCode\" class=\"bb-inline-stack bb-block--xs bb-block--full-width text-dark\">\n            <dt class=\"bb-inline-stack__item bb-text-bold\" i18n=\"@@bb-account-card-view.bank-code.label\">\n              Bank branch code:\n            </dt>\n            <dd data-role=\"contact-details-bank-code\" class=\"bb-inline-stack__item break-word bb-block--no-margin\">\n              {{hostRef.account.bankCode}}\n            </dd>\n          </div>\n          <div *ngIf=\"hostRef.account.BIC\" class=\"bb-inline-stack bb-block--xs bb-block--full-width text-dark\">\n            <dt class=\"bb-inline-stack__item bb-text-bold\" i18n=\"@@bb-account-card-view.bic.label\">BIC:</dt>\n            <dd class=\"bb-inline-stack__item break-word bb-block--no-margin\">{{hostRef.account.BIC}}</dd>\n          </div>\n        </div>\n        <div class=\"bb-block bb-block--xs\" *ngIf=\"hostRef.isValidAddress(hostRef.bankAddress)\">\n          <dt class=\"bb-text-bold\" i18n=\"@@bb-account-card-view.bank.address.label\">Bank branch address</dt>\n          <dd class=\"break-word\" data-role=\"contact-details-account-bank-address\">\n            <bb-contact-full-address [address]=\"hostRef.bankAddress\"></bb-contact-full-address>\n          </dd>\n        </div>\n        <div class=\"bb-block bb-block--xs\" *ngIf=\"hostRef.isValidAddress(hostRef.accountHolderAddress)\">\n          <dt class=\"bb-text-bold\" i18n=\"@@bb-account-card-view.accountHolder.address.label\">\n            Account holder's address\n          </dt>\n          <dd class=\"break-word\" data-role=\"contact-details-account-holder-address\">\n            <bb-contact-full-address [address]=\"hostRef.accountHolderAddress\"></bb-contact-full-address>\n          </dd>\n        </div>\n      </dl>\n    </div>\n  </div>\n</ng-template>\n", components: [{ type: i1.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i2.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i3.ContactFullAddressComponent, selector: "bb-contact-full-address", inputs: ["address"] }], directives: [{ type: i4.AccountCardViewCustomizableDirective, selector: "[bbAccountCardViewCustomizable]" }, { type: i5.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i6.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }, { type: i7.TooltipDirective, selector: "[bbTooltip]", inputs: ["bbTooltip", "triggers"] }], pipes: { "bbIban": i8.BbIbanPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountCardViewComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-account-card-view',
                    templateUrl: './account-card-view.component.html',
                }]
        }], propDecorators: { index: [{
                type: Input
            }], account: [{
                type: Input
            }], isEditMode: [{
                type: Input
            }], editAccount: [{
                type: Output
            }], deleteAccount: [{
                type: Output
            }], bankAddress: [{
                type: Input
            }], accountHolderAddress: [{
                type: Input
            }] } });
