// @backbase/contact-common-ang/esm2015/lib/components/account-card-view/account-card-view.module.js
import { NgModule } from '@angular/core';
import { AccountCardViewComponent } from './account-card-view.component';
import { AccountCardViewCustomizableDirective } from './account-card-view.directive';
import { ContactFullAddressModule } from '../contact-full-address/contact-full-address.module';
import { CommonModule } from '@angular/common';
import { HeaderModule } from '@backbase/ui-ang/header';
import { ButtonModule } from '@backbase/ui-ang/button';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { BbIbanPipeModule } from '@backbase/ui-ang/iban-pipe';
import { IconModule } from '@backbase/ui-ang/icon';
import * as i0 from "@angular/core";
/**
 * Exports:
 * - AccountCardViewComponent
 * - AccountCardViewCustomizableDirective
 */
export class AccountCardViewModule {
}
/** @nocollapse */ AccountCardViewModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountCardViewModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ AccountCardViewModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountCardViewModule, declarations: [AccountCardViewComponent, AccountCardViewCustomizableDirective], imports: [IconModule,
        HeaderModule,
        ButtonModule,
        BbIbanPipeModule,
        ContactFullAddressModule,
        CommonModule,
        TooltipModule], exports: [AccountCardViewComponent, AccountCardViewCustomizableDirective] });
/** @nocollapse */ AccountCardViewModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountCardViewModule, imports: [[
            IconModule,
            HeaderModule,
            ButtonModule,
            BbIbanPipeModule,
            ContactFullAddressModule,
            CommonModule,
            TooltipModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountCardViewModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [AccountCardViewComponent, AccountCardViewCustomizableDirective],
                    exports: [AccountCardViewComponent, AccountCardViewCustomizableDirective],
                    imports: [
                        IconModule,
                        HeaderModule,
                        ButtonModule,
                        BbIbanPipeModule,
                        ContactFullAddressModule,
                        CommonModule,
                        TooltipModule,
                    ],
                }]
        }] });
