// @backbase/ui-ang/esm2015/_example/product-item-investment-account/examples/product-item-investment-account.example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/product-item-investment-account";
export class ProductItemInvestmentAccountExampleComponent {
    constructor() {
        this.product = {
            id: 'b387bfa5-75fd-410e-bb86-7f764f8ef69c',
            name: 'Ricardo Norton\'s Savings Account',
            currency: 'EUR',
            amount: '83652.32',
            productNumber: 'NL75INGB3400120056',
        };
        this.productDetails = {
            owner: 'Richard Norton',
        };
        this.productNumberFormat = {
            length: 18,
            maskRange: [0, 0],
            segments: 4,
        };
    }
}
ProductItemInvestmentAccountExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemInvestmentAccountExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ProductItemInvestmentAccountExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ProductItemInvestmentAccountExampleComponent, selector: "bb-product-item-investment-account-example-ui", ngImport: i0, template: "<bb-product-item-investment-account-ui\n  [title]=\"product.name\"\n  [amount]=\"product.amount\"\n  [productNumber]=\"product.productNumber\"\n  [currency]=\"product.currency\"\n  [productNumberFormat]=\"productNumberFormat\"\n  [productDetailsInfo]=\"productDetails\"\n  [highlight]=\"false\"\n  [favorite]=\"false\"\n  [active]=\"false\"\n></bb-product-item-investment-account-ui>\n", components: [{ type: i1.ProductItemInvestmentAccountComponent, selector: "bb-product-item-investment-account-ui", inputs: ["title", "productNumber", "productNumberFormat", "currency", "amount", "highlight", "productDetailsInfo", "active", "favorite", "favoriteIconColor"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemInvestmentAccountExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-product-item-investment-account-example-ui',
                    templateUrl: './product-item-investment-account.example.component.html',
                }]
        }] });
