// @backbase/authorized-users-journey-ang/esm2015/src/common/components/authorized-users-base-dialog/authorized-users-base-dialog.component.js
import { Directive, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
// eslint-disable-next-line @angular-eslint/directive-class-suffix
export class AuthorizedUsersBaseDialogComponent {
    constructor() {
        this.dialogOpen = false;
        this.confirm = new EventEmitter();
        this.closed = new EventEmitter();
        this.onClose = () => {
            this.closed.emit();
        };
    }
    onConfirm() {
        this.confirm.emit();
    }
}
AuthorizedUsersBaseDialogComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersBaseDialogComponent, deps: [], target: i0.ɵɵFactoryTarget.Directive });
AuthorizedUsersBaseDialogComponent.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: AuthorizedUsersBaseDialogComponent, inputs: { dialogOpen: "dialogOpen" }, outputs: { confirm: "confirm", closed: "closed" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersBaseDialogComponent, decorators: [{
            type: Directive
        }], propDecorators: { dialogOpen: [{
                type: Input
            }], confirm: [{
                type: Output
            }], closed: [{
                type: Output
            }] } });
