// @backbase/contact-manager-widget-ang/esm2015/lib/components/contacts-detail/contacts-detail.component.js
import { ChangeDetectionStrategy, Component, Directive, EventEmitter, Input, Output } from '@angular/core';
import { BbTemplate } from '@backbase/foundation-ang/core';
import { ContactsService } from '../../services/contacts.service';
import { AccountCreateType } from '../../models/types';
import * as i0 from "@angular/core";
import * as i1 from "../../services/contacts.service";
import * as i2 from "@backbase/ui-ang/header";
import * as i3 from "@backbase/ui-ang/icon";
import * as i4 from "@backbase/contact-common-ang";
import * as i5 from "@angular/common";
import * as i6 from "@backbase/ui-ang/button";
import * as i7 from "../../directives/contacts-delete-confirmation-customizable.directive";
import * as i8 from "@backbase/ui-ang/iban-pipe";
/**
 * Displays a detail of contact selected.
 *
 * The following extension slots are available in this component:
 *  - `ContactsDetailCustomizableDirective`
 *
 * @see ContactsDetailCustomizableDirective
 * @usageNotes
 *
 * ### Display a details of contacts
 *
 * ```html
 * <bb-contacts-detail
 *    [showDetail]="false"
 *    [item]="hostRef.item"
 *    [type]="hostRef.type"
 *    (deleteContactsItem)="hostRef.deleteItemEmit()"
 *    (editContactsItem)="hostRef.editContactsItem()"
 * ></bb-contacts-detail>
 * ```
 * @ngModule ContactManagerWidgetModule
 */
export class ContactsDetailComponent {
    constructor(service) {
        this.service = service;
        this.hostRef = this;
        /**
         * Wether to show user details or not
         */
        this.showDetail = true;
        /**
         * Event emitted when contact is edited
         */
        this.editContactsItem = new EventEmitter();
        /**
         * Event emitted when contact is deleted
         */
        this.deleteContactsItem = new EventEmitter();
        this.isDeleteConfirmOpened = false;
    }
    /**
     * OnInit - lifecycle hook.
     */
    ngOnInit() {
        if (this.item === undefined) {
            throw new Error('`item` input is required in `bb-contacts-detail` component');
        }
    }
    /**
     *  Will emit the event when contact is edited.
     *
     * @param {string} id the contact id that has to be edited.
     */
    edit(id) {
        this.editContactsItem.emit({ id: id, type: this.type });
    }
    /**
     * opens modal for delete confirmation
     */
    openDeleteModal() {
        this.isDeleteConfirmOpened = true;
    }
    /**
     * event handler for confirming delete
     */
    onConfirmDelete() {
        this.deleteContactsItem.emit(this.item.id);
        this.isDeleteConfirmOpened = false;
    }
    /**
     * event handler for rejecting delete
     */
    onCancelDelete() {
        this.isDeleteConfirmOpened = false;
    }
}
/** @nocollapse */ ContactsDetailComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactsDetailComponent, deps: [{ token: i1.ContactsService }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ ContactsDetailComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ContactsDetailComponent, selector: "bb-contacts-detail", inputs: { type: "type", showDetail: "showDetail", item: "item" }, outputs: { editContactsItem: "editContactsItem", deleteContactsItem: "deleteContactsItem" }, ngImport: i0, template: "<ng-container bbContactsDetailCustomizable [bbHostRef]=\"hostRef\" [bbTemplateContext]=\"item\"></ng-container>\n<ng-template bbContactsDetailCustomizable let-hostRef let-item=\"item\">\n  <div class=\"bb-block bb-block--xl\">\n    <div class=\"bb-stack bb-stack--align-bottom\">\n      <div class=\"bb-stack__item bb-stack__item--fill\">\n        <div class=\"bb-block\" data-role=\"contact-details-info-name\">\n          <span\n            class=\"d-none\"\n            data-role=\"create-contact-label\"\n            aria-hidden=\"true\"\n            i18n=\"Create contact heading | Title shown when user is in create contact@@contact-manager-detail.create.heading\"\n            #heading\n          >\n            Create contact\n          </span>\n          <bb-header-ui\n            data-role=\"create-new-contact-label\"\n            headingType=\"h3\"\n            headingClasses=\"break-word\"\n            [heading]=\"hostRef.item.name || heading.textContent\"\n          >\n          </bb-header-ui>\n        </div>\n      </div>\n      <div *ngIf=\"hostRef.item.name\" class=\"bb-stack__item bb-stack__item--push-right\">\n        <span bbTooltip=\"Edit\" triggers=\"hover focus\" i18n-bbTooltip=\"@@contact-manager-detail.button.edit.tooltip\">\n          <button\n            data-role=\"contact-edit-button\"\n            bbButton\n            circle=\"true\"\n            color=\"link\"\n            (click)=\"hostRef.edit(hostRef.item.id)\"\n            i18n=\"Edit Contact Button | Button for editing an existing contact@@contact-manager-detail.button.edit\"\n            aria-label=\"Edit Contact\"\n            i18n-aria-label=\"Edit Contact | Aria label edit contact menu@@contact-manager-detail.label.actions.edit\"\n          >\n            <bb-icon-ui name=\"edit\" color=\"dark\"></bb-icon-ui>\n          </button>\n        </span>\n        <span bbTooltip=\"Delete\" triggers=\"hover focus\" i18n-bbTooltip=\"@@contact-manager-detail.button.delete.tooltip\">\n          <button\n            bbButton\n            circle=\"true\"\n            color=\"link\"\n            (click)=\"hostRef.openDeleteModal()\"\n            i18n=\"Delete Contact Button | Button for deleting an existing contact@@contact-manager-detail.detail.button.delete\"\n            data-role=\"delete-button\"\n            aria-label=\"Delete Contact\"\n            i18n-aria-label=\"Delete Contact | Aria label delete contact menu@@contact-manager-detail.label.actions.delete\"\n          >\n            <bb-icon-ui name=\"delete\" color=\"dark\"></bb-icon-ui>\n          </button>\n        </span>\n      </div>\n    </div>\n    <hr />\n  </div>\n  <div class=\"bb-lock bb-lock bb-block--lg\" *ngIf=\"hostRef.showDetail\">\n    <span\n      class=\"d-none\"\n      aria-hidden=\"true\"\n      i18n=\"Account number | Heading for account number or IBAN or Email or Phone@@contact-manager-detail.label.account.type.heading\"\n      #heading\n    >\n      {hostRef.type, select, IBAN {IBAN} accountNumber {Account number} email { Email} phoneNumber { Phone number} other\n      {IBAN} }\n    </span>\n    <bb-header-ui headingType=\"h5\" [heading]=\"heading.textContent\" class=\"bb-block bb-block-md\"></bb-header-ui>\n    <div class=\"bb-bock bb-block-md\" data-role=\"contact-details-info\">\n      <ng-container *ngIf=\"hostRef.item?.accounts[0] as account\">\n        <ng-container *ngIf=\"hostRef.type === 'IBAN'; else notIban\">\n          {{ account[hostRef.type] | bbIban }}\n        </ng-container>\n        <ng-template #notIban>{{ account[hostRef.type] }} </ng-template>\n      </ng-container>\n    </div>\n  </div>\n</ng-template>\n\n<ng-container bbContactsDeleteConfirmationCustomizable [bbHostRef]=\"hostRef\"></ng-container>\n<ng-template bbContactsDeleteConfirmationCustomizable let-hostRef>\n  <bb-confirm-action-dialog\n    [isOpen]=\"hostRef.isDeleteConfirmOpened\"\n    [confirmButtonColor]=\"'danger'\"\n    (confirm)=\"hostRef.onConfirmDelete()\"\n    (cancel)=\"hostRef.onCancelDelete()\"\n  >\n    <span title i18n=\"@@contact-manager-detail.header.delete.contact\" data-role=\"delete-contact-label\"\n      >Delete Contact</span\n    >\n    <span body i18n=\"@@contact-manager-detail.delete.confirmation\">\n      Are you sure you want to delete {{ hostRef.item.name }} from your contacts? This action cannot be undone.\n    </span>\n    <span cancelButtonLabel i18n=\"@@contact-manager-detail.button.cancel\">Not now</span>\n    <span confirmButtonLabel i18n=\"@@contact-manager-detail.button.delete\" data-role=\"confirm-delete-button\"\n      >Delete</span\n    >\n  </bb-confirm-action-dialog>\n</ng-template>\n", components: [{ type: i0.forwardRef(function () { return i2.HeaderComponent; }), selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i0.forwardRef(function () { return i3.IconComponent; }), selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i0.forwardRef(function () { return i4.ConfirmActionDialogComponent; }), selector: "bb-confirm-action-dialog", inputs: ["confirmButtonColor", "cancelButtonColor", "isOpen", "attachComment", "commentFieldLabel"], outputs: ["confirm", "cancel"] }], directives: [{ type: i0.forwardRef(function () { return ContactsDetailCustomizableDirective; }), selector: "[bbContactsDetailCustomizable]" }, { type: i0.forwardRef(function () { return i5.NgIf; }), selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i0.forwardRef(function () { return i6.ButtonDirective; }), selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }, { type: i0.forwardRef(function () { return i7.ContactsDeleteConfirmationCustomizableDirective; }), selector: "[bbContactsDeleteConfirmationCustomizable]" }], pipes: { "bbIban": i0.forwardRef(function () { return i8.BbIbanPipe; }) }, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactsDetailComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-contacts-detail',
                    templateUrl: './contacts-detail.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.ContactsService }]; }, propDecorators: { type: [{
                type: Input
            }], showDetail: [{
                type: Input
            }], item: [{
                type: Input
            }], editContactsItem: [{
                type: Output
            }], deleteContactsItem: [{
                type: Output
            }] } });
/**
 * This is an extension slot to be used for customizing an item of the contact details.
 *
 * @usageNotes
 *
 * ### Display a contact details
 *
 * ```html
 * <ng-template bbContactsDetailCustomizable let-hostRef let-item="item">
 * <div class="bb-block bb-block--xl">
 *   <div class="bb-stack bb-stack--align-bottom">
 *     <div class="bb-stack__item bb-stack__item--fill">
 *       <div
 *         class="bb-block"
 *         data-role="contact-details-info-name"
 *       >
 *         <span
 *           class="d-none"
 *           data-role="create-contact-label"
 *           aria-hidden="true"
 *           i18n-heading="Create contact heading | Title shown when user is in
 *                        create contact@@contact-manager-detail.create.heading"
 *          #heading
 *         >
 *           Create contact
 *         </span>
 *         <bb-header-ui
 *           data-role="create-new-contact-label"
 *           headingType="h3"
 *           [heading]="hostRef.item.name || heading.textContent"
 *         >
 *         </bb-header-ui>
 *       </div>
 *     </div>
 *     <div
 *      *ngIf="hostRef.item.name"
 *       class="bb-stack__item bb-stack__item--push-right"
 *     >
 *       <span
 *         bbTooltip="Edit"
 *         triggers="hover focus"
 *       >
 *         <button
 *           data-role="contact-edit-button"
 *           bbButton
 *           circle="true"
 *           color="link"
 *           (click)="hostRef.edit(hostRef.item.id)"
 *           i18n="Edit Contact Button | Button for editing an existing contact@@contact-manager-detail.button.edit"
 *           aria-label="Edit Contact"
 *           i18n-aria-label="Edit Contact | Aria label edit contact menu@@contact-manager-detail.label.actions.edit"
 *         >
 *           <bb-icon-ui
 *             name="edit"
 *             color="dark"
 *           ></bb-icon-ui>
 *         </button>
 *       </span>
 *       <span
 *         bbTooltip="Delete"
 *         triggers="hover focus"
 *       >
 *         <button data-role="contact-delete-button"
 *           bbButton
 *           circle="true"
 *           color="link"
 *           (click)="hostRef.openDeleteModal(hostRef.modalContent)"
 *           i18n="Delete Contact Button | Button for deleting an existing contact@@contact-manager-detail.detail.button.delete"
 *           data-role="delete-button"
 *           aria-label="Delete Contact"
 *           i18n-aria-label="Delete Contact | Aria label delete contact menu@@contact-manager-detail.label.actions.delete"
 *         >
 *           <bb-icon-ui name="delete" color="dark"></bb-icon-ui>
 *         </button>
 *       </span>
 *     </div>
 *   </div>
 *   <hr/>
 * </div>
 * <div
 *   class="bb-lock bb-lock bb-block--lg"
 *  *ngIf="hostRef.showDetail"
 * >
 *   <span
 *     class="d-none"
 *     aria-hidden="true"
 *     i18n="Account number | Heading for account number or
 *      IBAN or Email or Phone@@contact-manager-detail.label.account.type.heading"
 *     #heading
 *   >
 *    {hostRef.type, select, IBAN {IBAN}  accountNumber {Account number} email { Email} phoneNumber { Phone number} other {IBAN} }
 *   </span>
 *   <bb-header-ui
 *     headingType="h5"
 *     [heading]="heading.textContent"
 *     class="bb-block bb-block-md"
 *   >
 *   </bb-header-ui>
 *   <div
 *     class="bb-bock bb-block-md"
 *     data-role="contact-details-info-alias"
 *   >
 *   <span *ngIf="hostRef.item?.accounts[0] as account">
 *     {{ account[hostRef.type] }}
 *   </span>
 *   </div>
 * </div>
 *</ng-template>
 * ```
 * @ngModule ContactManagerWidgetModule
 */
export class ContactsDetailCustomizableDirective extends BbTemplate {
}
/** @nocollapse */ ContactsDetailCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactsDetailCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
/** @nocollapse */ ContactsDetailCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: ContactsDetailCustomizableDirective, selector: "[bbContactsDetailCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactsDetailCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbContactsDetailCustomizable]',
                }]
        }] });
