// @backbase/ui-ang/esm2015/load-button/load-button.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ButtonModule } from '@backbase/ui-ang/button';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { LoadButtonComponent } from './load-button.component';
import * as i0 from "@angular/core";
export class LoadButtonModule {
}
LoadButtonModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoadButtonModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LoadButtonModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoadButtonModule, declarations: [LoadButtonComponent], imports: [CommonModule, ButtonModule, LoadingIndicatorModule], exports: [LoadButtonComponent] });
LoadButtonModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoadButtonModule, imports: [[CommonModule, ButtonModule, LoadingIndicatorModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoadButtonModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ButtonModule, LoadingIndicatorModule],
                    declarations: [LoadButtonComponent],
                    exports: [LoadButtonComponent],
                }]
        }] });
