// @backbase/portfolio-summary-http-ang/esm2015/api/portfolioSummaryClientApi.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { PORTFOLIO_SUMMARY_BASE_PATH } from '../variables';
import { PortfolioSummaryConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class PortfolioSummaryClientApiHttpService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new PortfolioSummaryConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    deleteFavoriteByIban(requestParameters, observe = 'body', reportProgress = false, options) {
        const _iban = requestParameters["iban"];
        if (_iban === null || _iban === undefined) {
            throw new Error('Required parameter iban was null or undefined when calling deleteFavoriteByIban.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.delete(`${this.configuration.basePath}/client-api/v2/portfoliosummary/favorites/${encodeURIComponent(String(_iban))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    deleteFavoriteByIbanUrl(requestParameters) {
        const _iban = requestParameters["iban"];
        if (_iban === null || _iban === undefined) {
            throw new Error('Required parameter iban was null or undefined when calling deleteFavoriteByIban.');
        }
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/favorites/${encodeURIComponent(String(_iban))}`;
    }
    deleteFavoriteByPortfolioId(requestParameters, observe = 'body', reportProgress = false, options) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling deleteFavoriteByPortfolioId.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.delete(`${this.configuration.basePath}/client-api/v2/portfoliosummary/${encodeURIComponent(String(_portfolioId))}/favorite`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    deleteFavoriteByPortfolioIdUrl(requestParameters) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling deleteFavoriteByPortfolioId.');
        }
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/${encodeURIComponent(String(_portfolioId))}/favorite`;
    }
    deleteRemovePortfoliosFromCustomListByUuidAndPortfolioIbanList(requestParameters, observe = 'body', reportProgress = false, options) {
        const _portfolioIbanList = requestParameters["portfolioIbanList"];
        if (_portfolioIbanList === null || _portfolioIbanList === undefined) {
            throw new Error('Required parameter portfolioIbanList was null or undefined when calling deleteRemovePortfoliosFromCustomListByUuidAndPortfolioIbanList.');
        }
        const _uuid = requestParameters["uuid"];
        if (_uuid === null || _uuid === undefined) {
            throw new Error('Required parameter uuid was null or undefined when calling deleteRemovePortfoliosFromCustomListByUuidAndPortfolioIbanList.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.delete(`${this.configuration.basePath}/client-api/v2/portfoliosummary/custom-list/${encodeURIComponent(String(_uuid))}/portfolio/${encodeURIComponent(String(_portfolioIbanList))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    deleteRemovePortfoliosFromCustomListByUuidAndPortfolioIbanListUrl(requestParameters) {
        const _portfolioIbanList = requestParameters["portfolioIbanList"];
        if (_portfolioIbanList === null || _portfolioIbanList === undefined) {
            throw new Error('Required parameter portfolioIbanList was null or undefined when calling deleteRemovePortfoliosFromCustomListByUuidAndPortfolioIbanList.');
        }
        const _uuid = requestParameters["uuid"];
        if (_uuid === null || _uuid === undefined) {
            throw new Error('Required parameter uuid was null or undefined when calling deleteRemovePortfoliosFromCustomListByUuidAndPortfolioIbanList.');
        }
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/custom-list/${encodeURIComponent(String(_uuid))}/portfolio/${encodeURIComponent(String(_portfolioIbanList))}`;
    }
    deleteUuidByUuid(requestParameters, observe = 'body', reportProgress = false, options) {
        const _uuid = requestParameters["uuid"];
        if (_uuid === null || _uuid === undefined) {
            throw new Error('Required parameter uuid was null or undefined when calling deleteUuidByUuid.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.delete(`${this.configuration.basePath}/client-api/v2/portfoliosummary/custom-list/${encodeURIComponent(String(_uuid))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    deleteUuidByUuidUrl(requestParameters) {
        const _uuid = requestParameters["uuid"];
        if (_uuid === null || _uuid === undefined) {
            throw new Error('Required parameter uuid was null or undefined when calling deleteUuidByUuid.');
        }
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/custom-list/${encodeURIComponent(String(_uuid))}`;
    }
    exportPortfolioTransactions(requestParameters, observe = 'body', reportProgress = false, options) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling exportPortfolioTransactions.');
        }
        const _fileType = requestParameters["fileType"];
        if (_fileType === null || _fileType === undefined) {
            throw new Error('Required parameter fileType was null or undefined when calling exportPortfolioTransactions.');
        }
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_fileType !== undefined && _fileType !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _fileType, 'fileType');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'text/csv',
                'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/portfoliosummary/export/${encodeURIComponent(String(_portfolioId))}/transactions`, {
            params: queryParameters,
            responseType: "blob",
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    exportPortfolioTransactionsUrl(requestParameters) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling exportPortfolioTransactions.');
        }
        const _fileType = requestParameters["fileType"];
        if (_fileType === null || _fileType === undefined) {
            throw new Error('Required parameter fileType was null or undefined when calling exportPortfolioTransactions.');
        }
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_fileType !== undefined && _fileType !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _fileType, 'fileType');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/export/${encodeURIComponent(String(_portfolioId))}/transactions${queryString ? `?${queryString}` : ''}`;
    }
    getAggregation(requestParameters, observe = 'body', reportProgress = false, options) {
        const _portfolioIds = requestParameters["portfolioIds"];
        const _currencyCode = requestParameters["currencyCode"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_portfolioIds) {
            _portfolioIds.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'portfolioIds');
            });
        }
        if (_currencyCode !== undefined && _currencyCode !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _currencyCode, 'currencyCode');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/portfoliosummary/aggregation`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getAggregationUrl(requestParameters) {
        const _portfolioIds = requestParameters["portfolioIds"];
        const _currencyCode = requestParameters["currencyCode"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_portfolioIds) {
            _portfolioIds.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'portfolioIds');
            });
        }
        if (_currencyCode !== undefined && _currencyCode !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _currencyCode, 'currencyCode');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/aggregation${queryString ? `?${queryString}` : ''}`;
    }
    getBenchmarkDataByPortfolioId(requestParameters, observe = 'body', reportProgress = false, options) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling getBenchmarkDataByPortfolioId.');
        }
        const _uuid = requestParameters["uuid"];
        if (_uuid === null || _uuid === undefined) {
            throw new Error('Required parameter uuid was null or undefined when calling getBenchmarkDataByPortfolioId.');
        }
        const _fromDate = requestParameters["fromDate"];
        if (_fromDate === null || _fromDate === undefined) {
            throw new Error('Required parameter fromDate was null or undefined when calling getBenchmarkDataByPortfolioId.');
        }
        const _toDate = requestParameters["toDate"];
        if (_toDate === null || _toDate === undefined) {
            throw new Error('Required parameter toDate was null or undefined when calling getBenchmarkDataByPortfolioId.');
        }
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_uuid !== undefined && _uuid !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _uuid, 'uuid');
        }
        if (_fromDate !== undefined && _fromDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _fromDate, 'fromDate');
        }
        if (_toDate !== undefined && _toDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _toDate, 'toDate');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/portfoliosummary/${encodeURIComponent(String(_portfolioId))}/performance-benchmark-history`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getBenchmarkDataByPortfolioIdUrl(requestParameters) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling getBenchmarkDataByPortfolioId.');
        }
        const _uuid = requestParameters["uuid"];
        if (_uuid === null || _uuid === undefined) {
            throw new Error('Required parameter uuid was null or undefined when calling getBenchmarkDataByPortfolioId.');
        }
        const _fromDate = requestParameters["fromDate"];
        if (_fromDate === null || _fromDate === undefined) {
            throw new Error('Required parameter fromDate was null or undefined when calling getBenchmarkDataByPortfolioId.');
        }
        const _toDate = requestParameters["toDate"];
        if (_toDate === null || _toDate === undefined) {
            throw new Error('Required parameter toDate was null or undefined when calling getBenchmarkDataByPortfolioId.');
        }
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_uuid !== undefined && _uuid !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _uuid, 'uuid');
        }
        if (_fromDate !== undefined && _fromDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _fromDate, 'fromDate');
        }
        if (_toDate !== undefined && _toDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _toDate, 'toDate');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/${encodeURIComponent(String(_portfolioId))}/performance-benchmark-history${queryString ? `?${queryString}` : ''}`;
    }
    getChartDataByPortfolioId(requestParameters, observe = 'body', reportProgress = false, options) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling getChartDataByPortfolioId.');
        }
        const _fromDate = requestParameters["fromDate"];
        if (_fromDate === null || _fromDate === undefined) {
            throw new Error('Required parameter fromDate was null or undefined when calling getChartDataByPortfolioId.');
        }
        const _toDate = requestParameters["toDate"];
        if (_toDate === null || _toDate === undefined) {
            throw new Error('Required parameter toDate was null or undefined when calling getChartDataByPortfolioId.');
        }
        const _granularity = requestParameters["granularity"];
        if (_granularity === null || _granularity === undefined) {
            throw new Error('Required parameter granularity was null or undefined when calling getChartDataByPortfolioId.');
        }
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_fromDate !== undefined && _fromDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _fromDate, 'fromDate');
        }
        if (_toDate !== undefined && _toDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _toDate, 'toDate');
        }
        if (_granularity !== undefined && _granularity !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _granularity, 'granularity');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/portfoliosummary/${encodeURIComponent(String(_portfolioId))}/valuation-chart`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getChartDataByPortfolioIdUrl(requestParameters) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling getChartDataByPortfolioId.');
        }
        const _fromDate = requestParameters["fromDate"];
        if (_fromDate === null || _fromDate === undefined) {
            throw new Error('Required parameter fromDate was null or undefined when calling getChartDataByPortfolioId.');
        }
        const _toDate = requestParameters["toDate"];
        if (_toDate === null || _toDate === undefined) {
            throw new Error('Required parameter toDate was null or undefined when calling getChartDataByPortfolioId.');
        }
        const _granularity = requestParameters["granularity"];
        if (_granularity === null || _granularity === undefined) {
            throw new Error('Required parameter granularity was null or undefined when calling getChartDataByPortfolioId.');
        }
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_fromDate !== undefined && _fromDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _fromDate, 'fromDate');
        }
        if (_toDate !== undefined && _toDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _toDate, 'toDate');
        }
        if (_granularity !== undefined && _granularity !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _granularity, 'granularity');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/${encodeURIComponent(String(_portfolioId))}/valuation-chart${queryString ? `?${queryString}` : ''}`;
    }
    getChartDataByPortfolioIdChart(requestParameters, observe = 'body', reportProgress = false, options) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling getChartDataByPortfolioIdChart.');
        }
        const _fromDate = requestParameters["fromDate"];
        if (_fromDate === null || _fromDate === undefined) {
            throw new Error('Required parameter fromDate was null or undefined when calling getChartDataByPortfolioIdChart.');
        }
        const _toDate = requestParameters["toDate"];
        if (_toDate === null || _toDate === undefined) {
            throw new Error('Required parameter toDate was null or undefined when calling getChartDataByPortfolioIdChart.');
        }
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_fromDate !== undefined && _fromDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _fromDate, 'fromDate');
        }
        if (_toDate !== undefined && _toDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _toDate, 'toDate');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/portfoliosummary/${encodeURIComponent(String(_portfolioId))}/performance-chart`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getChartDataByPortfolioIdChartUrl(requestParameters) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling getChartDataByPortfolioIdChart.');
        }
        const _fromDate = requestParameters["fromDate"];
        if (_fromDate === null || _fromDate === undefined) {
            throw new Error('Required parameter fromDate was null or undefined when calling getChartDataByPortfolioIdChart.');
        }
        const _toDate = requestParameters["toDate"];
        if (_toDate === null || _toDate === undefined) {
            throw new Error('Required parameter toDate was null or undefined when calling getChartDataByPortfolioIdChart.');
        }
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_fromDate !== undefined && _fromDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _fromDate, 'fromDate');
        }
        if (_toDate !== undefined && _toDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _toDate, 'toDate');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/${encodeURIComponent(String(_portfolioId))}/performance-chart${queryString ? `?${queryString}` : ''}`;
    }
    getCustomlist(observe = 'body', reportProgress = false, options) {
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/portfoliosummary/custom-list`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getCustomlistUrl() {
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/custom-list`;
    }
    getPerformancebenchmarksByPortfolioId(requestParameters, observe = 'body', reportProgress = false, options) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling getPerformancebenchmarksByPortfolioId.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/portfoliosummary/${encodeURIComponent(String(_portfolioId))}/performance-benchmarks`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getPerformancebenchmarksByPortfolioIdUrl(requestParameters) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling getPerformancebenchmarksByPortfolioId.');
        }
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/${encodeURIComponent(String(_portfolioId))}/performance-benchmarks`;
    }
    getPortfolioAllocationsByAssetClassByPortfolioId(requestParameters, observe = 'body', reportProgress = false, options) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling getPortfolioAllocationsByAssetClassByPortfolioId.');
        }
        const _currencyCode = requestParameters["currencyCode"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_currencyCode !== undefined && _currencyCode !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _currencyCode, 'currencyCode');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/portfoliosummary/${encodeURIComponent(String(_portfolioId))}/allocations-asset`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getPortfolioAllocationsByAssetClassByPortfolioIdUrl(requestParameters) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling getPortfolioAllocationsByAssetClassByPortfolioId.');
        }
        const _currencyCode = requestParameters["currencyCode"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_currencyCode !== undefined && _currencyCode !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _currencyCode, 'currencyCode');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/${encodeURIComponent(String(_portfolioId))}/allocations-asset${queryString ? `?${queryString}` : ''}`;
    }
    getPortfolioAllocationsByCurrencyByPortfolioId(requestParameters, observe = 'body', reportProgress = false, options) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling getPortfolioAllocationsByCurrencyByPortfolioId.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/portfoliosummary/${encodeURIComponent(String(_portfolioId))}/allocations-currency`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getPortfolioAllocationsByCurrencyByPortfolioIdUrl(requestParameters) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling getPortfolioAllocationsByCurrencyByPortfolioId.');
        }
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/${encodeURIComponent(String(_portfolioId))}/allocations-currency`;
    }
    getPortfolioIdByPortfolioId(requestParameters, observe = 'body', reportProgress = false, options) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling getPortfolioIdByPortfolioId.');
        }
        const _currencyCode = requestParameters["currencyCode"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_currencyCode !== undefined && _currencyCode !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _currencyCode, 'currencyCode');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/portfoliosummary/${encodeURIComponent(String(_portfolioId))}`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getPortfolioIdByPortfolioIdUrl(requestParameters) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling getPortfolioIdByPortfolioId.');
        }
        const _currencyCode = requestParameters["currencyCode"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_currencyCode !== undefined && _currencyCode !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _currencyCode, 'currencyCode');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/${encodeURIComponent(String(_portfolioId))}${queryString ? `?${queryString}` : ''}`;
    }
    getPortfolioPositionsByPortfolioId(requestParameters, observe = 'body', reportProgress = false, options) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling getPortfolioPositionsByPortfolioId.');
        }
        const _search = requestParameters["search"];
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        const _currencyCode = requestParameters["currencyCode"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_search !== undefined && _search !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _search, 'search');
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_currencyCode !== undefined && _currencyCode !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _currencyCode, 'currencyCode');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/portfoliosummary/${encodeURIComponent(String(_portfolioId))}/positions`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getPortfolioPositionsByPortfolioIdUrl(requestParameters) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling getPortfolioPositionsByPortfolioId.');
        }
        const _search = requestParameters["search"];
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        const _currencyCode = requestParameters["currencyCode"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_search !== undefined && _search !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _search, 'search');
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_currencyCode !== undefined && _currencyCode !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _currencyCode, 'currencyCode');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/${encodeURIComponent(String(_portfolioId))}/positions${queryString ? `?${queryString}` : ''}`;
    }
    getPortfoliosummary(requestParameters, observe = 'body', reportProgress = false, options) {
        const _query = requestParameters["query"];
        const _customListUuid = requestParameters["customListUuid"];
        const _justFavorites = requestParameters["justFavorites"];
        const _from = requestParameters["from"];
        const _cursor = requestParameters["cursor"];
        const _size = requestParameters["size"];
        const _orderBy = requestParameters["orderBy"];
        const _direction = requestParameters["direction"];
        const _currencyCode = requestParameters["currencyCode"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_query !== undefined && _query !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _query, 'query');
        }
        if (_customListUuid !== undefined && _customListUuid !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _customListUuid, 'customListUuid');
        }
        if (_justFavorites !== undefined && _justFavorites !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _justFavorites, 'justFavorites');
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_cursor !== undefined && _cursor !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _cursor, 'cursor');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_orderBy !== undefined && _orderBy !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _orderBy, 'orderBy');
        }
        if (_direction !== undefined && _direction !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _direction, 'direction');
        }
        if (_currencyCode !== undefined && _currencyCode !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _currencyCode, 'currencyCode');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/portfoliosummary`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getPortfoliosummaryUrl(requestParameters) {
        const _query = requestParameters["query"];
        const _customListUuid = requestParameters["customListUuid"];
        const _justFavorites = requestParameters["justFavorites"];
        const _from = requestParameters["from"];
        const _cursor = requestParameters["cursor"];
        const _size = requestParameters["size"];
        const _orderBy = requestParameters["orderBy"];
        const _direction = requestParameters["direction"];
        const _currencyCode = requestParameters["currencyCode"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_query !== undefined && _query !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _query, 'query');
        }
        if (_customListUuid !== undefined && _customListUuid !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _customListUuid, 'customListUuid');
        }
        if (_justFavorites !== undefined && _justFavorites !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _justFavorites, 'justFavorites');
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_cursor !== undefined && _cursor !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _cursor, 'cursor');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_orderBy !== undefined && _orderBy !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _orderBy, 'orderBy');
        }
        if (_direction !== undefined && _direction !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _direction, 'direction');
        }
        if (_currencyCode !== undefined && _currencyCode !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _currencyCode, 'currencyCode');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary${queryString ? `?${queryString}` : ''}`;
    }
    getTransactionFilterOptionsByPortfolioId(requestParameters, observe = 'body', reportProgress = false, options) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling getTransactionFilterOptionsByPortfolioId.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/portfoliosummary/${encodeURIComponent(String(_portfolioId))}/transaction-filter-options`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getTransactionFilterOptionsByPortfolioIdUrl(requestParameters) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling getTransactionFilterOptionsByPortfolioId.');
        }
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/${encodeURIComponent(String(_portfolioId))}/transaction-filter-options`;
    }
    getTransactionIdByPortfolioIdAndTransactionId(requestParameters, observe = 'body', reportProgress = false, options) {
        const _transactionId = requestParameters["transactionId"];
        if (_transactionId === null || _transactionId === undefined) {
            throw new Error('Required parameter transactionId was null or undefined when calling getTransactionIdByPortfolioIdAndTransactionId.');
        }
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling getTransactionIdByPortfolioIdAndTransactionId.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/portfoliosummary/${encodeURIComponent(String(_portfolioId))}/transactions/${encodeURIComponent(String(_transactionId))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getTransactionIdByPortfolioIdAndTransactionIdUrl(requestParameters) {
        const _transactionId = requestParameters["transactionId"];
        if (_transactionId === null || _transactionId === undefined) {
            throw new Error('Required parameter transactionId was null or undefined when calling getTransactionIdByPortfolioIdAndTransactionId.');
        }
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling getTransactionIdByPortfolioIdAndTransactionId.');
        }
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/${encodeURIComponent(String(_portfolioId))}/transactions/${encodeURIComponent(String(_transactionId))}`;
    }
    getTransactionsByPortfolioId(requestParameters, observe = 'body', reportProgress = false, options) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling getTransactionsByPortfolioId.');
        }
        const _size = requestParameters["size"];
        if (_size === null || _size === undefined) {
            throw new Error('Required parameter size was null or undefined when calling getTransactionsByPortfolioId.');
        }
        const _from = requestParameters["from"];
        if (_from === null || _from === undefined) {
            throw new Error('Required parameter from was null or undefined when calling getTransactionsByPortfolioId.');
        }
        const _search = requestParameters["search"];
        const _fromDate = requestParameters["fromDate"];
        const _toDate = requestParameters["toDate"];
        const _fromAmount = requestParameters["fromAmount"];
        const _toAmount = requestParameters["toAmount"];
        const _category = requestParameters["category"];
        const _instrumentName = requestParameters["instrumentName"];
        const _counterpartyName = requestParameters["counterpartyName"];
        const _counterpartyAccount = requestParameters["counterpartyAccount"];
        const _subPortfolioId = requestParameters["subPortfolioId"];
        const _iSIN = requestParameters["iSIN"];
        const _assetKey = requestParameters["assetKey"];
        const _assetClass = requestParameters["assetClass"];
        const _positionType = requestParameters["positionType"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_search !== undefined && _search !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _search, 'search');
        }
        if (_fromDate !== undefined && _fromDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _fromDate, 'fromDate');
        }
        if (_toDate !== undefined && _toDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _toDate, 'toDate');
        }
        if (_fromAmount !== undefined && _fromAmount !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _fromAmount, 'fromAmount');
        }
        if (_toAmount !== undefined && _toAmount !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _toAmount, 'toAmount');
        }
        if (_category !== undefined && _category !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _category, 'category');
        }
        if (_instrumentName !== undefined && _instrumentName !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _instrumentName, 'instrumentName');
        }
        if (_counterpartyName !== undefined && _counterpartyName !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _counterpartyName, 'counterpartyName');
        }
        if (_counterpartyAccount !== undefined && _counterpartyAccount !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _counterpartyAccount, 'counterpartyAccount');
        }
        if (_subPortfolioId !== undefined && _subPortfolioId !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _subPortfolioId, 'subPortfolioId');
        }
        if (_iSIN !== undefined && _iSIN !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _iSIN, 'iSIN');
        }
        if (_assetKey !== undefined && _assetKey !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _assetKey, 'assetKey');
        }
        if (_assetClass !== undefined && _assetClass !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _assetClass, 'assetClass');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_positionType !== undefined && _positionType !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _positionType, 'positionType');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/portfoliosummary/${encodeURIComponent(String(_portfolioId))}/transactions`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getTransactionsByPortfolioIdUrl(requestParameters) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling getTransactionsByPortfolioId.');
        }
        const _size = requestParameters["size"];
        if (_size === null || _size === undefined) {
            throw new Error('Required parameter size was null or undefined when calling getTransactionsByPortfolioId.');
        }
        const _from = requestParameters["from"];
        if (_from === null || _from === undefined) {
            throw new Error('Required parameter from was null or undefined when calling getTransactionsByPortfolioId.');
        }
        const _search = requestParameters["search"];
        const _fromDate = requestParameters["fromDate"];
        const _toDate = requestParameters["toDate"];
        const _fromAmount = requestParameters["fromAmount"];
        const _toAmount = requestParameters["toAmount"];
        const _category = requestParameters["category"];
        const _instrumentName = requestParameters["instrumentName"];
        const _counterpartyName = requestParameters["counterpartyName"];
        const _counterpartyAccount = requestParameters["counterpartyAccount"];
        const _subPortfolioId = requestParameters["subPortfolioId"];
        const _iSIN = requestParameters["iSIN"];
        const _assetKey = requestParameters["assetKey"];
        const _assetClass = requestParameters["assetClass"];
        const _positionType = requestParameters["positionType"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_search !== undefined && _search !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _search, 'search');
        }
        if (_fromDate !== undefined && _fromDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _fromDate, 'fromDate');
        }
        if (_toDate !== undefined && _toDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _toDate, 'toDate');
        }
        if (_fromAmount !== undefined && _fromAmount !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _fromAmount, 'fromAmount');
        }
        if (_toAmount !== undefined && _toAmount !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _toAmount, 'toAmount');
        }
        if (_category !== undefined && _category !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _category, 'category');
        }
        if (_instrumentName !== undefined && _instrumentName !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _instrumentName, 'instrumentName');
        }
        if (_counterpartyName !== undefined && _counterpartyName !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _counterpartyName, 'counterpartyName');
        }
        if (_counterpartyAccount !== undefined && _counterpartyAccount !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _counterpartyAccount, 'counterpartyAccount');
        }
        if (_subPortfolioId !== undefined && _subPortfolioId !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _subPortfolioId, 'subPortfolioId');
        }
        if (_iSIN !== undefined && _iSIN !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _iSIN, 'iSIN');
        }
        if (_assetKey !== undefined && _assetKey !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _assetKey, 'assetKey');
        }
        if (_assetClass !== undefined && _assetClass !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _assetClass, 'assetClass');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_positionType !== undefined && _positionType !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _positionType, 'positionType');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/${encodeURIComponent(String(_portfolioId))}/transactions${queryString ? `?${queryString}` : ''}`;
    }
    postAddPortfoliosToCustomListByUuid(requestParameters, observe = 'body', reportProgress = false, options) {
        const _uuid = requestParameters["uuid"];
        if (_uuid === null || _uuid === undefined) {
            throw new Error('Required parameter uuid was null or undefined when calling postAddPortfoliosToCustomListByUuid.');
        }
        const _customListAddPortfolios = requestParameters["customListAddPortfolios"];
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v2/portfoliosummary/custom-list/${encodeURIComponent(String(_uuid))}/portfolio`, _customListAddPortfolios, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    postAddPortfoliosToCustomListByUuidUrl(requestParameters) {
        const _uuid = requestParameters["uuid"];
        if (_uuid === null || _uuid === undefined) {
            throw new Error('Required parameter uuid was null or undefined when calling postAddPortfoliosToCustomListByUuid.');
        }
        const _customListAddPortfolios = requestParameters["customListAddPortfolios"];
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/custom-list/${encodeURIComponent(String(_uuid))}/portfolio`;
    }
    postCreatePortfolioCustomList(requestParameters, observe = 'body', reportProgress = false, options) {
        const _createPortfolioCustomListRequest = requestParameters["createPortfolioCustomListRequest"];
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v2/portfoliosummary/custom-list`, _createPortfolioCustomListRequest, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    postCreatePortfolioCustomListUrl(requestParameters) {
        const _createPortfolioCustomListRequest = requestParameters["createPortfolioCustomListRequest"];
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/custom-list`;
    }
    postSavePortfolioAlias(requestParameters, observe = 'body', reportProgress = false, options) {
        const _postPortfolioAlias = requestParameters["postPortfolioAlias"];
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v2/portfoliosummary/alias`, _postPortfolioAlias, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    postSavePortfolioAliasUrl(requestParameters) {
        const _postPortfolioAlias = requestParameters["postPortfolioAlias"];
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/alias`;
    }
    postSavePortfoliosOrdering(requestParameters, observe = 'body', reportProgress = false, options) {
        const _portfolioClientOrder = requestParameters["portfolioClientOrder"];
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v2/portfoliosummary/client-order`, _portfolioClientOrder, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    postSavePortfoliosOrderingUrl(requestParameters) {
        const _portfolioClientOrder = requestParameters["portfolioClientOrder"];
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/client-order`;
    }
    putMarkPortfolioAsFavorite(requestParameters, observe = 'body', reportProgress = false, options) {
        const _markPortfolioAsFavoriteRequest = requestParameters["markPortfolioAsFavoriteRequest"];
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v2/portfoliosummary/favorites`, _markPortfolioAsFavoriteRequest, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    putMarkPortfolioAsFavoriteUrl(requestParameters) {
        const _markPortfolioAsFavoriteRequest = requestParameters["markPortfolioAsFavoriteRequest"];
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/favorites`;
    }
    putMarkPortfolioAsFavoriteUnfavorite(requestParameters, observe = 'body', reportProgress = false, options) {
        const _markPortfolioFavoriteItemRequest = requestParameters["markPortfolioFavoriteItemRequest"];
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v2/portfoliosummary/mark-favorites`, _markPortfolioFavoriteItemRequest, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    putMarkPortfolioAsFavoriteUnfavoriteUrl(requestParameters) {
        const _markPortfolioFavoriteItemRequest = requestParameters["markPortfolioFavoriteItemRequest"];
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/mark-favorites`;
    }
    putRenameTheCustomListByUuid(requestParameters, observe = 'body', reportProgress = false, options) {
        const _uuid = requestParameters["uuid"];
        if (_uuid === null || _uuid === undefined) {
            throw new Error('Required parameter uuid was null or undefined when calling putRenameTheCustomListByUuid.');
        }
        const _customListName = requestParameters["customListName"];
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v2/portfoliosummary/custom-list/${encodeURIComponent(String(_uuid))}`, _customListName, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    putRenameTheCustomListByUuidUrl(requestParameters) {
        const _uuid = requestParameters["uuid"];
        if (_uuid === null || _uuid === undefined) {
            throw new Error('Required parameter uuid was null or undefined when calling putRenameTheCustomListByUuid.');
        }
        const _customListName = requestParameters["customListName"];
        return `${this.configuration.basePath}/client-api/v2/portfoliosummary/custom-list/${encodeURIComponent(String(_uuid))}`;
    }
}
PortfolioSummaryClientApiHttpService.fac = i0.ngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PortfolioSummaryClientApiHttpService, deps: [{ token: i1.HttpClient }, { token: PORTFOLIO_SUMMARY_BASE_PATH, optional: true }, { token: i2.PortfolioSummaryConfiguration, optional: true }], target: i0.FactoryTarget.Injectable });
PortfolioSummaryClientApiHttpService.prov = i0.ngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PortfolioSummaryClientApiHttpService, providedIn: 'root' });
i0.ngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PortfolioSummaryClientApiHttpService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [PORTFOLIO_SUMMARY_BASE_PATH]
                }] }, { type: i2.PortfolioSummaryConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
