// @backbase/portfolio-summary-http-ang/esm2015/api/transactionCategoriesClient.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const TransactionCategoriesClientHttpServiceGetTransactionCategoriesMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/transaction-categories",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "key": "purchase",
                        "alias": "Purchase",
                        "description": "Additional info"
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const TransactionCategoriesClientHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v2/transaction-categories",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "key": "purchase",
                        "alias": "Purchase",
                        "description": "Additional info"
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
]);
