// @backbase/loans-collection-ui-ang/esm2015/lib/loan-summary/loan-summary-amount-item/loan-summary-amount-item.component.js
import { Component, ChangeDetectionStrategy, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/amount";
import * as i2 from "@angular/common";
export class LoanSummaryAmountItemComponent {
    constructor() {
        /**
         * Whether amount will be as percentage
         */
        this.showPercent = false;
    }
}
/** @nocollapse */ LoanSummaryAmountItemComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryAmountItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ LoanSummaryAmountItemComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: LoanSummaryAmountItemComponent, selector: "bb-loan-summary-amount-item", inputs: { title: "title", amount: "amount", currency: "currency", isCurrencySymbolUsed: "isCurrencySymbolUsed", showPercent: "showPercent", titleClasses: "titleClasses", amountClasses: "amountClasses", totalAmount: "totalAmount", showLabel: "showLabel" }, ngImport: i0, template: "<div\n  *ngIf=\"amount && (currency || showPercent)\"\n  class=\"bb-loans-summary-amount-item\"\n  data-role=\"loan-summary-amount-item__wrapper\"\n>\n  <div\n    class=\"bb-subtitle bb-loans-summary-amount-item__title text-capitalize\"\n    [ngClass]=\"titleClasses\"\n    data-role=\"loan-summary-amount-item__title\"\n  >\n    {{ title }}\n  </div>\n  <div class=\"bb-text-bold bb-heading-3\" [ngClass]=\"amountClasses\" data-role=\"loan-summary-amount-item__amount-wrapper\">\n    <bb-amount-ui\n      [amount]=\"amount\"\n      [currency]=\"currency || ''\"\n      [mapCurrency]=\"!!isCurrencySymbolUsed\"\n      [showPercent]=\"showPercent\"\n    ></bb-amount-ui>\n\n    <span *ngIf=\"totalAmount && showLabel\" class=\"bb-loans-summary-amount-item__label bb-subtitle\">\n      of\n      <bb-amount-ui\n        [amount]=\"totalAmount\"\n        [currency]=\"currency || ''\"\n        [mapCurrency]=\"!!isCurrencySymbolUsed\"\n        [showPercent]=\"showPercent\"\n      ></bb-amount-ui>\n    </span>\n  </div>\n  <span class=\"sr-only\">,</span>\n</div>\n", components: [{ type: i1.AmountComponent, selector: "bb-amount-ui", inputs: ["currency", "showPlusSign", "mapCurrency", "showPercent", "abbreviate", "decimalPlaces", "trailingZeroes", "amount"] }], directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i2.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryAmountItemComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-loan-summary-amount-item',
                    templateUrl: './loan-summary-amount-item.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { title: [{
                type: Input
            }], amount: [{
                type: Input
            }], currency: [{
                type: Input
            }], isCurrencySymbolUsed: [{
                type: Input
            }], showPercent: [{
                type: Input
            }], titleClasses: [{
                type: Input
            }], amountClasses: [{
                type: Input
            }], totalAmount: [{
                type: Input
            }], showLabel: [{
                type: Input
            }] } });
