// @backbase/loans-collection-ui-ang/esm2015/lib/loan-summary/loan-summary-amount-item/loan-summary-amount-item.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AmountModule } from '@backbase/ui-ang/amount';
import { LoanSummaryAmountItemComponent } from './loan-summary-amount-item.component';
import * as i0 from "@angular/core";
export class LoanSummaryAmountItemModule {
}
/** @nocollapse */ LoanSummaryAmountItemModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryAmountItemModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanSummaryAmountItemModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryAmountItemModule, declarations: [LoanSummaryAmountItemComponent], imports: [CommonModule, AmountModule], exports: [LoanSummaryAmountItemComponent] });
/** @nocollapse */ LoanSummaryAmountItemModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryAmountItemModule, imports: [[CommonModule, AmountModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryAmountItemModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanSummaryAmountItemComponent],
                    imports: [CommonModule, AmountModule],
                    exports: [LoanSummaryAmountItemComponent],
                }]
        }] });
