// @backbase/payord-form-builder-ang/esm2015/lib/services/components-factory.service.js
import { PaymentComponentsMap } from '../models/components-map.model';
import { PayordTextboxComponent } from '../components/textbox/payord-text.component';
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
export class ComponentsFactoryService {
    constructor() {
        this.componentsMap = Object.assign({}, PaymentComponentsMap);
    }
    /**
     * Registers custom fields
     *
     * @param components Payment components
     * @returns
     */
    registerCustomFields(components = {}) {
        for (const key in components) {
            if (components.hasOwnProperty(key)) {
                this.componentsMap[key] = components[key];
            }
        }
    }
    /**
     * Registers custom fields for payment configs
     *
     * @param paymentConfigs Payment configurations
     * @returns
     */
    registerCustomFieldsForPaymentConfigs(paymentConfigs) {
        for (const config of paymentConfigs) {
            if (config.customFields) {
                this.registerCustomFields(config.customFields);
            }
        }
    }
    /**
     * Gets a form field
     *
     * @param formFieldType Type of the form field
     * @returns
     */
    getFormField(formFieldType) {
        return this.componentsMap[formFieldType] || PayordTextboxComponent;
    }
}
/** @nocollapse */ ComponentsFactoryService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ComponentsFactoryService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
/** @nocollapse */ ComponentsFactoryService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ComponentsFactoryService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ComponentsFactoryService, decorators: [{
            type: Injectable
        }] });
