// @backbase/account-statement-common-ang/esm2015/src/components/account-statement-base-widget/account-statement-base-widget.component.js
import { Directive, Input } from '@angular/core';
import * as i0 from "@angular/core";
// @dynamic
// eslint-disable-next-line @angular-eslint/directive-class-suffix
export class AccountStatementBaseWidgetComponent {
    constructor(propertiesService) {
        this.propertiesService = propertiesService;
    }
    /* eslint-disable @angular-eslint/no-input-rename */
    /**
     * Number of statements per page
     */
    set pageSizeValue(value) {
        this.propertiesService.setPageSize(value);
    }
    /**
     * Sorting order (date, description, category)
     */
    set sortingOrderValue(value) {
        this.propertiesService.setSortingOrder(value);
    }
    /**
     * Sorting direction (ASC, DESC)
     */
    set sortingDirectionValue(value) {
        this.propertiesService.setSortingDirection(value);
    }
    /**
     * Flag to show/hide categories
     */
    set showCategoryValue(value) {
        this.propertiesService.setShowCategory(value);
    }
    /* eslint-enable @angular-eslint/no-input-rename */
    /**
     * Notification dismiss time in seconds
     */
    set notificationDismissTime(value) {
        this.propertiesService.setNotificationDismissTime(value);
    }
    /**
     * Relative filter datepicker limit (months)
     */
    set relativeDateLimit(value) {
        this.propertiesService.setRelativeDateLimit(value);
    }
    /**
     * Absolute filter datepicker limit (yyyy-mm-dd)
     */
    set absoluteDateLimit(value) {
        this.propertiesService.setAbsoluteDateLimit(value);
    }
    /**
     * Pagination type (PAGINATION or LOAD_MORE)
     */
    set paginationType(value) {
        this.propertiesService.setPaginationType(value);
    }
}
AccountStatementBaseWidgetComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementBaseWidgetComponent, deps: "invalid", target: i0.ɵɵFactoryTarget.Directive });
AccountStatementBaseWidgetComponent.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.15", type: AccountStatementBaseWidgetComponent, inputs: { pageSizeValue: ["pageSize", "pageSizeValue"], sortingOrderValue: ["sortingOrder", "sortingOrderValue"], sortingDirectionValue: ["sortingDirection", "sortingDirectionValue"], showCategoryValue: ["showCategory", "showCategoryValue"], notificationDismissTime: "notificationDismissTime", relativeDateLimit: "relativeDateLimit", absoluteDateLimit: "absoluteDateLimit", paginationType: "paginationType" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementBaseWidgetComponent, decorators: [{
            type: Directive
        }], ctorParameters: function () { return [{ type: undefined }]; }, propDecorators: { pageSizeValue: [{
                type: Input,
                args: ['pageSize']
            }], sortingOrderValue: [{
                type: Input,
                args: ['sortingOrder']
            }], sortingDirectionValue: [{
                type: Input,
                args: ['sortingDirection']
            }], showCategoryValue: [{
                type: Input,
                args: ['showCategory']
            }], notificationDismissTime: [{
                type: Input
            }], relativeDateLimit: [{
                type: Input
            }], absoluteDateLimit: [{
                type: Input
            }], paginationType: [{
                type: Input
            }] } });
