// @backbase/accessgroup-ui/esm2015/lib/accessgroup-approval-log/accessgroup-approval-log.component.js
import { Component, Input } from '@angular/core';
import '@angular/localize/init';
import { ApprovalStatus } from '../model/approval-status.enum';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/icon";
import * as i2 from "@angular/common";
const translations = {
    approved: $localize `:Approved status@@approvals-journey.approval-log.status.approved:Approved`,
    rejected: $localize `:Rejected status@@approvals-journey.approval-log.status.rejected:Rejected`,
};
export class AccessgroupApprovalLogComponent {
    constructor() {
        /** Is pending status */
        this.isPending = false;
        this.records = [];
        /** Status name (translated) reference object */
        this.statusName = {
            [ApprovalStatus.REJECTED]: translations.rejected,
            [ApprovalStatus.APPROVED]: translations.approved,
        };
        /** Icon name reference object */
        this.iconName = {
            [ApprovalStatus.REJECTED]: 'cancel',
            [ApprovalStatus.APPROVED]: 'success',
        };
        /** Icon color reference object */
        this.iconColor = {
            [ApprovalStatus.REJECTED]: 'danger',
            [ApprovalStatus.APPROVED]: 'success',
        };
    }
    /** Approval info setter */
    set approvalInfo(value) {
        this.records = [];
        if (value.records) {
            value.records.forEach((el) => {
                var _a, _b;
                el.approvalLevel = (_b = (_a = value === null || value === void 0 ? void 0 : value.approvalTypes) === null || _a === void 0 ? void 0 : _a.find(item => item.id === el.approvalTypeId)) === null || _b === void 0 ? void 0 : _b.name;
                this.records.push(el);
            });
        }
    }
}
/** @nocollapse */ AccessgroupApprovalLogComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccessgroupApprovalLogComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ AccessgroupApprovalLogComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AccessgroupApprovalLogComponent, selector: "bb-accessgroup-approval-log", inputs: { isPending: "isPending", approvalInfo: "approvalInfo" }, ngImport: i0, template: "<div class=\"bb-item-log bb-accessgroup-approval-log\">\n  <div *ngFor=\"let itemLogRecord of records\" class=\"bb-item-log__record\" data-role=\"approval-log-record\">\n    <div class=\"bb-stack bb-stack--align-top bb-stack--wrap\">\n      <bb-icon-ui\n        class=\"bb-item-log-record__icon\"\n        size=\"md\"\n        [name]=\"iconName[itemLogRecord.status]\"\n        [color]=\"iconColor[itemLogRecord.status]\"></bb-icon-ui>\n      <div class=\"bb-item-log-record__details\">\n        <span class=\"bb-text-bold\">{{ statusName[itemLogRecord.status] }} </span>\n\n        <!-- TODO: Replace letter with component -->\n        <span\n          *ngIf=\"itemLogRecord.approvalLevel\"\n          class=\"d-inline-block px-2 mx-1\"\n          style=\"background-color: #dedede; border-radius: 4px\"\n          >{{ itemLogRecord.approvalLevel }}</span\n        >\n\n        <small>\n          <ng-container> {{ itemLogRecord.userFullName }}</ng-container>\n          <ng-container *ngIf=\"itemLogRecord.externalUserId\"> ({{ itemLogRecord.externalUserId }})</ng-container>\n        </small>\n\n        <div class=\"bb-item-log-record__time\">\n          <time> {{ itemLogRecord.createdAt | date: 'd MMM YYYY' }}</time>\n          <ng-container i18n=\"At (used in date info)@@approvals-journey.approval-log.date.at\"> at</ng-container>\n          <time> {{ itemLogRecord.createdAt | date: 'shortTime' }}</time>\n        </div>\n      </div>\n\n      <div *ngIf=\"itemLogRecord.comment\" class=\"w-100 mt-3 mr-4 mb-2 ml-4\">\n        <div class=\"bb-card bb-card--sm bb-card--shadowless ml-2 bb-accessgroup-approval-log__comment\">\n          <div class=\"bb-card__body\">\n            <small>{{ itemLogRecord.comment }}</small>\n          </div>\n        </div>\n      </div>\n    </div>\n  </div>\n  <div *ngIf=\"isPending\" class=\"bb-item-log__record\" data-role=\"approval-log-pending\">\n    <bb-icon-ui class=\"bb-text-support bb-item-log-record__icon\" name=\"watch-later\" size=\"md\"></bb-icon-ui>\n    <div\n      class=\"bb-item-log__pending-record--details bb-text-bold\"\n      i18n=\"Waiting for further approval(s)@@approvals-journey.approval-log.pending.title\">\n      Waiting for further approval(s)\n    </div>\n  </div>\n</div>\n", components: [{ type: i1.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], directives: [{ type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "date": i2.DatePipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccessgroupApprovalLogComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-accessgroup-approval-log',
                    templateUrl: './accessgroup-approval-log.component.html',
                }]
        }], propDecorators: { isPending: [{
                type: Input
            }], approvalInfo: [{
                type: Input
            }] } });
