// @backbase/payord-form-builder-ang/esm2015/lib/directives/form-field.directive.js
import { Directive, Input, ComponentFactoryResolver, ViewContainerRef, } from '@angular/core';
import { FormGroup } from '@angular/forms';
import { ComponentsFactoryService } from '../services/components-factory.service';
import * as i0 from "@angular/core";
import * as i1 from "../services/components-factory.service";
export class PaymentFormFieldDirective {
    constructor(resolver, container, componentsFactory) {
        this.resolver = resolver;
        this.container = container;
        this.componentsFactory = componentsFactory;
    }
    ngOnInit() {
        const component = this.componentsFactory.getFormField(this.config.type);
        const factory = this.resolver.resolveComponentFactory(component);
        this.component = this.container.createComponent(factory);
        this.component.instance.config = this.config;
        this.component.instance.group = this.group;
        this.component.instance.options = this.options;
    }
}
/** @nocollapse */ PaymentFormFieldDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentFormFieldDirective, deps: [{ token: i0.ComponentFactoryResolver }, { token: i0.ViewContainerRef }, { token: i1.ComponentsFactoryService }], target: i0.ɵɵFactoryTarget.Directive });
/** @nocollapse */ PaymentFormFieldDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: PaymentFormFieldDirective, selector: "[bbPaymentFormField]", inputs: { options: "options", config: "config", group: "group" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentFormFieldDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbPaymentFormField]',
                }]
        }], ctorParameters: function () { return [{ type: i0.ComponentFactoryResolver }, { type: i0.ViewContainerRef }, { type: i1.ComponentsFactoryService }]; }, propDecorators: { options: [{
                type: Input
            }], config: [{
                type: Input
            }], group: [{
                type: Input
            }] } });
