// @backbase/ui-ang/esm2015/_example/search-box/showcase/search-box-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SearchBoxShowcaseComponent } from './search-box-showcase.component';
import { SearchBoxModule } from '@backbase/ui-ang/search-box';
import { ReactiveFormsModule } from '@angular/forms';
import * as i0 from "@angular/core";
export class SearchBoxShowcaseModule {
}
SearchBoxShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SearchBoxShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
SearchBoxShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SearchBoxShowcaseModule, declarations: [SearchBoxShowcaseComponent], imports: [CommonModule, SearchBoxModule, ReactiveFormsModule] });
SearchBoxShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SearchBoxShowcaseModule, imports: [[CommonModule, SearchBoxModule, ReactiveFormsModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SearchBoxShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, SearchBoxModule, ReactiveFormsModule],
                    declarations: [SearchBoxShowcaseComponent],
                    entryComponents: [SearchBoxShowcaseComponent],
                }]
        }] });
