// @backbase/cards-travel-notice-widget-ang/esm2015/src/components/cards-travel-notice-payment-card/cards-travel-notice-payment-card.component.js
import { Component, Input } from '@angular/core';
import { PaymentCardLockStatus, PaymentCardStatus, } from '@backbase/cards-common-ang';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/cards-common-ang";
import * as i2 from "@backbase/ui-ang/payment-card";
export class CardsTravelNoticePaymentCardComponent {
    getDisplayState() {
        if (this.paymentCard && this.paymentCard.status === PaymentCardStatus.Inactive) {
            return 'INACTIVE';
        }
        if (this.paymentCard && this.paymentCard.status === PaymentCardStatus.Canceled) {
            return 'CANCELLED';
        }
        if (this.paymentCard && this.paymentCard.status === PaymentCardStatus.Expired) {
            return 'EXPIRED';
        }
        if (this.paymentCard && this.paymentCard.lockStatus === PaymentCardLockStatus.Locked) {
            return 'LOCKED';
        }
        return undefined;
    }
}
CardsTravelNoticePaymentCardComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardsTravelNoticePaymentCardComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
CardsTravelNoticePaymentCardComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: CardsTravelNoticePaymentCardComponent, selector: "bb-cards-travel-notice-payment-card", inputs: { paymentCard: "paymentCard" }, ngImport: i0, template: "<bb-payment-card-state-wrapper-ui [state]=\"getDisplayState()\">\n  <bb-payment-card-ui [paymentCard]=\"paymentCard\"></bb-payment-card-ui>\n</bb-payment-card-state-wrapper-ui>", components: [{ type: i1.PaymentCardStateWrapperComponent, selector: "bb-payment-card-state-wrapper,bb-payment-card-state-wrapper-ui", inputs: ["state"] }, { type: i2.PaymentCardComponent, selector: "bb-payment-card-ui", inputs: ["paymentCard", "paymentCardNumberFormat"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardsTravelNoticePaymentCardComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-cards-travel-notice-payment-card',
                    templateUrl: 'cards-travel-notice-payment-card.component.html',
                }]
        }], propDecorators: { paymentCard: [{
                type: Input
            }] } });
