// @backbase/ui-ang/esm2015/input-inline-edit/input-inline-edit.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { IconModule } from '@backbase/ui-ang/icon';
import { BaseClassesModule } from '@backbase/ui-ang/base-classes';
import { ReactiveFormsModule } from '@angular/forms';
import { ButtonModule } from '@backbase/ui-ang/button';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import { InputInlineEditComponent } from './input-inline-edit.component';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import * as i0 from "@angular/core";
export class InputInlineEditModule {
}
InputInlineEditModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputInlineEditModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputInlineEditModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputInlineEditModule, declarations: [InputInlineEditComponent], imports: [CommonModule,
        IconModule,
        ReactiveFormsModule,
        BaseClassesModule,
        ButtonModule,
        InputTextModule,
        InputValidationMessageModule,
        LoadingIndicatorModule,
        TooltipModule], exports: [InputInlineEditComponent] });
InputInlineEditModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputInlineEditModule, imports: [[
            CommonModule,
            IconModule,
            ReactiveFormsModule,
            BaseClassesModule,
            ButtonModule,
            InputTextModule,
            InputValidationMessageModule,
            LoadingIndicatorModule,
            TooltipModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputInlineEditModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        IconModule,
                        ReactiveFormsModule,
                        BaseClassesModule,
                        ButtonModule,
                        InputTextModule,
                        InputValidationMessageModule,
                        LoadingIndicatorModule,
                        TooltipModule,
                    ],
                    declarations: [InputInlineEditComponent],
                    exports: [InputInlineEditComponent],
                }]
        }] });
