// @backbase/ui-ang/esm2015/action-status/action-status.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { IconModule } from '@backbase/ui-ang/icon';
import { ActionStatusComponent } from './action-status.component';
import * as i0 from "@angular/core";
export class ActionStatusModule {
}
ActionStatusModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ActionStatusModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ActionStatusModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ActionStatusModule, declarations: [ActionStatusComponent], imports: [CommonModule, LoadingIndicatorModule, IconModule], exports: [ActionStatusComponent] });
ActionStatusModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ActionStatusModule, imports: [[CommonModule, LoadingIndicatorModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ActionStatusModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, LoadingIndicatorModule, IconModule],
                    declarations: [ActionStatusComponent],
                    exports: [ActionStatusComponent],
                }]
        }] });
