// @backbase/ui-ang/esm2015/services/browser.service.js
import { Inject, Injectable, InjectionToken } from '@angular/core';
import * as i0 from "@angular/core";
export function navigatorFactory() {
    return navigator;
}
export const NAVIGATOR_TOKEN = new InjectionToken('Navigator object');
export class BrowserService {
    // 'any' due to issue https://github.com/angular/angular/issues/15640
    constructor(navigator) {
        this.navigator = navigator;
    }
    browserIsMSIE() {
        return this.navigator.userAgent.indexOf('MSIE') > -1 || this.navigator.userAgent.indexOf('Trident/') > -1;
    }
}
BrowserService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BrowserService, deps: [{ token: NAVIGATOR_TOKEN }], target: i0.ɵɵFactoryTarget.Injectable });
BrowserService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BrowserService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BrowserService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [NAVIGATOR_TOKEN]
                }] }]; } });
