// @backbase/ui-ang/esm2015/_example/phone-number-format-pipe/examples/phone-number-pipe.example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/phone-number-format-pipe";
export class PhoneNumberPipeExampleComponent {
    constructor() {
        this.workPhone = '4151234567';
    }
}
PhoneNumberPipeExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PhoneNumberPipeExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
PhoneNumberPipeExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: PhoneNumberPipeExampleComponent, selector: "bb-phone-number-pipe-example-ui", ngImport: i0, template: "<p>{{ workPhone | phoneNumber: 'US' }}</p>\n", pipes: { "phoneNumber": i1.PhoneNumberPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PhoneNumberPipeExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-phone-number-pipe-example-ui',
                    templateUrl: './phone-number-pipe.example.component.html',
                }]
        }] });
