// @backbase/contact-manager-widget-ang/esm2015/lib/containers/contacts-list-container/contacts-list-container.directive.js
import { Directive } from '@angular/core';
import { BbTemplate } from '@backbase/foundation-ang/core';
import * as i0 from "@angular/core";
/**
 * This is an extension slot to be used for customizing contact manager widget..
 *
 * @usageNotes
 *
 * ### Display a single contact detail
 *
 * ```html
 *  <ng-template bbContactsListContainerCustomizable let-hostRef let-items="context">
 *   <div
 *     class="col"
 *     [ngClass]="{
 *       'd-block': hostRef.isListView | async,
 *       'd-lg-block d-none': !(hostRef.isListView | async)
 *     }"
 *   >
 *     <bb-contacts-list-widget
 *       [items]="items.contacts"
 *       (selectedChange)="hostRef.select($event)"
 *       (formItemChange)="hostRef.edit($event)"
 *       [showAlias]="false"
 *     >
 *     </bb-contacts-list-widget>
 *   </div>
 * </ng-template>
 * ```
 * @ngModule ContactManagerWidgetModule
 */
export class ContactsListContainerCustomizableDirective extends BbTemplate {
}
/** @nocollapse */ ContactsListContainerCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactsListContainerCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
/** @nocollapse */ ContactsListContainerCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: ContactsListContainerCustomizableDirective, selector: "[bbContactsListContainerCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactsListContainerCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbContactsListContainerCustomizable]',
                }]
        }] });
/**
 * This is an extension slot to be used for customizing contact manager header section
 *
 * @usageNotes
 *
 * ### Display a header section with search box
 *
 * ```html
 *  <ng-template bbContactsTopSectionCustomizable let-hostRef let-item="context">
 *   <div class="col bb-stack">
 *     <div class="bb-stack__item bb-stack__item--fill">
 *       <bb-search-box-ui
 *         class="bb-toolbar__item"
 *         data-role="search-contact-textbox"
 *         placeholder="Search"
 *         i18n-placeholder="Search box placeholder | Placeholder for search box@@contact-manager.placeholder.search"
 *         [showSearch]="true"
 *         [showClear]="true"
 *         (submit)="hostRef.searchBoxItemSelected()"
 *         (clear)="hostRef.searchBoxItemSelected(true)"
 *         [formControl]="hostRef.searchTerm"
 *         (keyUp.enter)="hostRef.searchBoxItemSelected()"
 *       ></bb-search-box-ui>
 *     </div>
 *     <div class="bb-stack__item">
 *       <button
 *         bbButton
 *         i18n="New Contact Button | Button for creating a new contact@@contact-manager.button.newContact"
 *         data-role="new-contact-button"
 *         class="d-none d-lg-block bb-button-bar__button"
 *         [disabled]="!(hostRef.newContactAllowed | async)"
 *         (click)="hostRef.newContact(item.contacts.length)"
 *         type="button"
 *       >
 *         New Contact
 *       </button>
 *       <button
 *         bbButton
 *         i18n="New Contact Button | Button for creating a new contact@@contact-manager.button.newContact"
 *         data-role="new-contact-button"
 *         class="d-block d-lg-none bb-button-bar__button"
 *         [disabled]="!hostRef.newContactAllowed"
 *         (click)="hostRef.newContact(item.contacts.length)"
 *         type="button"
 *       >
 *         New Contact
 *       </button>
 *     </div>
 *   </div>
 *   <ng-template #rt let-result="result" let-term="term">
 *     <div class="text-truncate" [title]="result">
 *       <ngb-highlight [result]="result" [term]="term"></ngb-highlight>
 *     </div>
 *   </ng-template>
 * </ng-template>
 * ```
 * @ngModule ContactManagerWidgetModule
 */
export class ContactsTopSectionCustomizableDirective extends BbTemplate {
}
/** @nocollapse */ ContactsTopSectionCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactsTopSectionCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
/** @nocollapse */ ContactsTopSectionCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: ContactsTopSectionCustomizableDirective, selector: "[bbContactsTopSectionCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactsTopSectionCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbContactsTopSectionCustomizable]',
                }]
        }] });
/**
 * This is an extension slot to be used for customizing contact manager pagination
 *
 * @usageNotes
 *
 * ### Display the pagination
 *
 * ```html
 *   <ng-template bbContactsLoadMoreCustomizable let-hostRef let-items="context">
 *   <bb-pagination-ui
 *     *ngIf="(hostRef.paginationType | async) !== 'load-more'"
 *     class="bb-pagination"
 *     [page]="(hostRef.pagination | async)?.from"
 *     [boundaryLinks]="false"
 *     [directionLinks]="true"
 *     [totalItems]="items?.count"
 *     [itemsPerPage]="hostRef.pageSize | async"
 *     [maxNavPages]="hostRef.maxNavPages | async"
 *     (pageChange)="hostRef.onPageChange($event)"
 *   ></bb-pagination-ui>
 *   <div class="p-5 text-center" *ngIf="(hostRef.paginationType | async) === 'load-more'">
 *     <button
 *       class="border text-dark btn btn-secondary"
 *       (click)="hostRef.loadMore()"
 *       [disabled]="items.count && items.count <= items.contacts.length"
 *     >
 *       <ng-container
 *          i18n="Load more button label | Load more button shown down bellow contact list@@contact-manager.load-more.label">
 *         Load more
 *       </ng-container>
 *     </button>
 *   </div>
 * </ng-template>
 * ```
 * @ngModule ContactManagerWidgetModule
 */
export class ContactsLoadMoreCustomizableDirective extends BbTemplate {
}
/** @nocollapse */ ContactsLoadMoreCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactsLoadMoreCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
/** @nocollapse */ ContactsLoadMoreCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: ContactsLoadMoreCustomizableDirective, selector: "[bbContactsLoadMoreCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactsLoadMoreCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbContactsLoadMoreCustomizable]',
                }]
        }] });
/**
 * This is an extension slot to be used for customizing contact manager widget loading view
 *
 * @usageNotes
 *
 * ### Display loading icon
 *
 * ```html
 *    <ng-template bbContactsLoadingCustomizable let-hostRef>
 *   <div class="bb-state-container">
 *     <bb-loading-indicator-ui
 *       i18n-text="@@contact-manager.loading.label"
 *       text="Loading contacts..."
 *     ></bb-loading-indicator-ui>
 *   </div>
 * </ng-template>
 * ```
 * @ngModule ContactManagerWidgetModule
 */
export class ContactsLoadingCustomizableDirective extends BbTemplate {
}
/** @nocollapse */ ContactsLoadingCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactsLoadingCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
/** @nocollapse */ ContactsLoadingCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: ContactsLoadingCustomizableDirective, selector: "[bbContactsLoadingCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactsLoadingCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbContactsLoadingCustomizable]',
                }]
        }] });
/**
 * This is an extension slot to be used for customizing contact manager widget empty view
 *
 * @usageNotes
 *
 * ### Display empty message
 *
 * ```html
 *  <ng-template bbContactsEmptyStateCustomizable let-hostRef>
 *   <div class="bb-state-container">
 *     <bb-empty-state-ui
 *       i18n-title="@@contact-manager.state.noContactsLoaded.title"
 *       title="No contacts"
 *       i18n-subtitle="@@contact-manager.state.noContactsLoaded.subtitle"
 *       subtitle="You don't have any contacts to display yet.<br>Try adding a new contact using the 'New Contact' button"
 *       iconClasses="d-flex justify-content-center mb-3"
 *       iconModifier="contacts"
 *       iconColor="primary"
 *       iconSize="xxl"
 *     ></bb-empty-state-ui>
 *   </div>
 * </ng-template>
 * ```
 * @ngModule ContactManagerWidgetModule
 */
export class ContactsEmptyStateCustomizableDirective extends BbTemplate {
}
/** @nocollapse */ ContactsEmptyStateCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactsEmptyStateCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
/** @nocollapse */ ContactsEmptyStateCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: ContactsEmptyStateCustomizableDirective, selector: "[bbContactsEmptyStateCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactsEmptyStateCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbContactsEmptyStateCustomizable]',
                }]
        }] });
/**
 * This is an extension slot to be used for customizing contact manager Empty view on search
 *
 * @usageNotes
 *
 * ### Display no search result found message.
 *
 * ```html
 *  <ng-template bbContactsEmptySearchStateCustomizable let-hostRef let-term="context">
 *   <div class="bb-state-container">
 *     <bb-empty-state-ui
 *       i18n-title="@@contact-manager.state.empty-search.title"
 *       title="No results found"
 *       i18n-subtitle="@@contact-manager.state.empty-search.subtitle"
 *       subtitle="We can't find what you're looking for. Please try a different criteria."
 *       iconClasses="d-flex justify-content-center mb-3"
 *       iconModifier="search"
 *       iconColor="primary"
 *       iconSize="xxl"
 *     ></bb-empty-state-ui>
 *   </div>
 * </ng-template>
 * ```
 * @ngModule ContactManagerWidgetModule
 */
export class ContactsEmptySearchStateCustomizableDirective extends BbTemplate {
}
/** @nocollapse */ ContactsEmptySearchStateCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactsEmptySearchStateCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
/** @nocollapse */ ContactsEmptySearchStateCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: ContactsEmptySearchStateCustomizableDirective, selector: "[bbContactsEmptySearchStateCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactsEmptySearchStateCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbContactsEmptySearchStateCustomizable]',
                }]
        }] });
/**
 * This is an extension slot to be used for customizing contact Error state
 *
 * @usageNotes
 *
 * ### Display error view
 *
 * ```html
 *  <ng-template bbContactsErrorStateCustomizable let-hostRef let-error="context">
 *   <div class="bb-state-container">
 *     <bb-common-error-state-ui [error]="error"></bb-common-error-state-ui>
 *   </div>
 * </ng-template>
 * ```
 * @ngModule ContactManagerWidgetModule
 */
export class ContactsErrorStateCustomizableDirective extends BbTemplate {
}
/** @nocollapse */ ContactsErrorStateCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactsErrorStateCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
/** @nocollapse */ ContactsErrorStateCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: ContactsErrorStateCustomizableDirective, selector: "[bbContactsErrorStateCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactsErrorStateCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbContactsErrorStateCustomizable]',
                }]
        }] });
