// @backbase/accesscontrol-http-ang/esm2015/api/approvals.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const ApprovalsHttpServiceGetApprovalsMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/approvals",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "creatorUserFullName": "User 1",
                        "creatorUserId": "5643e686d31e4216b3ff5d66161d1234",
                        "createdAt": "2019-01-01T12:00:00Z",
                        "action": "EDIT",
                        "category": "Assign Permissions",
                        "approvalId": "2df100c6-b12b-4102-8831-db417acf12b6"
                    }, {
                        "creatorUserFullName": "User 2",
                        "creatorUserId": "5643e686d31e4216b3ff5d66161d1234",
                        "createdAt": "2019-01-01T12:00:00Z",
                        "action": "EDIT",
                        "category": "Manage Data Groups",
                        "approvalId": "2df100c6-b12b-4102-8831-db417acf12b7"
                    }, {
                        "creatorUserFullName": "User 3",
                        "creatorUserId": "5643e686d31e4216b3ff5d66161d1234",
                        "createdAt": "2019-01-01T12:00:00Z",
                        "action": "EDIT",
                        "category": "Unlock User",
                        "approvalId": "2df100c6-b12b-4102-8831-db417acf12b8"
                    }, {
                        "creatorUserFullName": "User 4",
                        "creatorUserId": "5643e686d31e4216b3ff5d66161d1234",
                        "createdAt": "2019-01-01T12:00:00Z",
                        "action": "EDIT",
                        "category": "Manage Limits",
                        "approvalId": "2df100c6-b12b-4102-8831-db417acf12b9"
                    }, {
                        "creatorUserFullName": "User 5",
                        "creatorUserId": "5643e686d31e4216b3ff5d66161d1234",
                        "createdAt": "2019-01-01T12:00:00Z",
                        "action": "EDIT",
                        "category": "Manage Shadow Limits",
                        "approvalId": "2df100c6-b12b-4102-8831-db417acf12b1"
                    }, {
                        "creatorUserFullName": "User 6",
                        "creatorUserId": "5643e686d31e3567b3ff5d66161d1234",
                        "createdAt": "2019-01-01T12:00:00Z",
                        "action": "EDIT",
                        "category": "Manage Function Groups",
                        "approvalId": "2df100c6-b12b-4102-8831-db417acf12c4"
                    }]
            },
        ]
    }]);
export const ApprovalsHttpServiceGetPresentationDataGroupApprovalDetailsItemMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/approvals/{approvalId}/data-group",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "dataGroupId": "0955e686d31e4216b3dd5d66161d536d",
                    "serviceAgreementId": "0889e686d31e4216b3dd5d66163d2b14",
                    "serviceAgreementName": "Sa name",
                    "type": "ARRANGEMENTS",
                    "approvalId": "606d4532-f8d9-4a5f-36kl-887baf88fa24",
                    "action": "EDIT",
                    "oldState": {
                        "name": "DG name",
                        "description": "DG desc"
                    },
                    "newState": {
                        "name": "new DG name",
                        "description": "new DG desc"
                    },
                    "addedDataItems": ["1cdb2224-8926-4b4d-a99f-1c9dfbbb4612", "1cdb2224-8926-4b4d-a99f-1c9dfbbb4613"],
                    "removedDataItems": ["1cdb2224-8926-4b4d-a99f-1c9dfbbb4699", "1cdb2224-8926-4b4d-a99f-1c9dfbbb4614"],
                    "unmodifiedDataItems": ["1cdb2224-8926-4b4d-a99f-1c9dfbbb4615", "1cdb2224-8926-4b4d-a99f-1c9dfbbb4611"],
                    "legalEntityIds": ["cd0968b3-3520-4280-8699-cd46c4d6df72", "3c684e80-c646-4148-bc20-37718eab4d97"]
                }
            },
        ]
    }]);
export const ApprovalsHttpServiceGetPresentationFunctionGroupApprovalDetailsItemMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/approvals/{approvalId}/function-group",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "functionGroupId": "0955e686d31e4216b3dd5d66161d536d",
                    "serviceAgreementId": "0889e686d31e4216b3dd5d66163d2b14",
                    "serviceAgreementName": "saName",
                    "approvalId": "606d4532-f8d9-4a5f-36kl-887baf88fa24",
                    "action": "EDIT",
                    "oldState": {
                        "name": "FG1",
                        "description": "FG1 description",
                        "approvalTypeId": "approvalTypeIdA",
                        "validFromDate": "2020-01-01",
                        "validFromTime": "00:00:00",
                        "validUntilDate": "2025-01-01",
                        "validUntilTime": "23:59:59"
                    },
                    "newState": {
                        "name": "FG2",
                        "description": "FG2 description",
                        "approvalTypeId": "approvalTypeIdB",
                        "validFromDate": "2020-06-10",
                        "validFromTime": "04:32:24",
                        "validUntilDate": "2023-01-01",
                        "validUntilTime": "23:53:51"
                    },
                    "permissionMatrix": [{
                            "functionId": "1012",
                            "functionCode": "manage.limits",
                            "resource": "Limits",
                            "name": "Manage Limits",
                            "action": "changed",
                            "privileges": {
                                "view": {
                                    "old": true,
                                    "new": false
                                },
                                "create": {
                                    "old": false,
                                    "new": true
                                },
                                "edit": {
                                    "old": true,
                                    "new": false
                                },
                                "delete": {
                                    "old": false,
                                    "new": false
                                },
                                "approve": {
                                    "old": true,
                                    "new": true
                                },
                                "cancel": {
                                    "old": false,
                                    "new": true
                                },
                                "execute": {
                                    "old": false,
                                    "new": true
                                }
                            }
                        }, {
                            "functionId": "1009",
                            "functionCode": "assign.permissions",
                            "resource": "Service Agreement",
                            "name": "Assign Permissions",
                            "action": "added",
                            "privileges": {
                                "view": {
                                    "new": true
                                },
                                "create": {
                                    "new": true
                                },
                                "edit": {
                                    "new": false
                                },
                                "delete": {
                                    "new": true
                                },
                                "approve": {
                                    "new": false
                                },
                                "cancel": {
                                    "new": true
                                },
                                "execute": {
                                    "new": true
                                }
                            }
                        }, {
                            "functionId": "1046",
                            "functionCode": "ach.debit",
                            "resource": "Payments",
                            "name": "ACH Debit",
                            "action": "removed",
                            "privileges": {
                                "view": {
                                    "old": true
                                },
                                "create": {
                                    "old": true
                                },
                                "edit": {
                                    "old": false
                                },
                                "delete": {
                                    "old": true
                                },
                                "approve": {
                                    "old": false
                                },
                                "cancel": {
                                    "old": true
                                },
                                "execute": {
                                    "old": true
                                }
                            }
                        }, {
                            "functionId": "1010",
                            "functionCode": "manage.users",
                            "resource": "User",
                            "name": "Manage Users",
                            "action": "unchanged",
                            "privileges": {
                                "view": {
                                    "old": true,
                                    "new": true
                                },
                                "create": {
                                    "old": true,
                                    "new": true
                                },
                                "edit": {
                                    "old": true,
                                    "new": true
                                },
                                "delete": {
                                    "old": false,
                                    "new": false
                                },
                                "approve": {
                                    "old": true,
                                    "new": true
                                },
                                "cancel": {
                                    "old": false,
                                    "new": false
                                }
                            }
                        }]
                }
            },
        ]
    }]);
export const ApprovalsHttpServiceGetPresentationPermissionsApprovalDetailsItemMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/approvals/{approvalId}/permissions",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "approvalId": "606d4532-f8d9-4a5f-36kl-887baf88fa24",
                    "creatorUserFullName": "John Snow",
                    "creatorUserId": "69ef5b3c-3997-4ee9-9f11-cb14447cf572",
                    "createdAt": "2018-09-28T16:41:41.090Z",
                    "userId": "606d4532-f8d9-4a5f-840a-887baf88fa24",
                    "userFullName": "Jack Sparrow",
                    "serviceAgreementId": "606d4532-f8d9-4a5f-840a-887baf88fa24",
                    "serviceAgreementName": "Backbase SA",
                    "serviceAgreementDescription": "Detailed desscription of the SA.",
                    "action": "EDIT",
                    "category": "Assign Permissions",
                    "newFunctionGroups": [{
                            "id": "c18dda63-27fa-4e4f-8882-24b24b0d46df",
                            "name": "FGName",
                            "type": "REGULAR",
                            "description": "Function Group Description",
                            "dataGroups": [{
                                    "id": "c18dda63-27fa-4e4f-8882-24b24b0d46df",
                                    "name": "DGName",
                                    "description": "Function Group Description",
                                    "type": "PAYEES"
                                }],
                            "selfApprovalPolicies": [{
                                    "businessFunctionCode": "manage.users",
                                    "canSelfApprove": false
                                }]
                        }],
                    "removedFunctionGroups": [{
                            "id": "0955e686-d31e-4216-b3dd-5d66161d536d",
                            "name": "FGName",
                            "type": "TEMPLATE",
                            "description": "Function Group Description",
                            "dataGroups": [{
                                    "id": "d9b299b0-ca28-4683-a3c1-9a339abfb5e4",
                                    "name": "DGName",
                                    "description": "Data Group Description",
                                    "type": "CONTACTS"
                                }],
                            "selfApprovalPolicies": [{
                                    "businessFunctionCode": "manage.users",
                                    "canSelfApprove": false
                                }]
                        }],
                    "modifiedFunctionGroups": [{
                            "id": "d9b299b0-ca28-4683-a3c1-9a339abfb5e4",
                            "name": "FGName",
                            "type": "TEMPLATE",
                            "description": "Function Group Description",
                            "newDataGroups": [{
                                    "id": "5690bd25-b9db-493a-9659-3d5e505c7358",
                                    "name": "DGName",
                                    "description": "Data Group Description",
                                    "type": "CONTACTS"
                                }],
                            "removedDataGroups": [{
                                    "id": "5f5c3714-289e-4f0c-bdb0-0b5a749f8beb",
                                    "name": "DGName",
                                    "description": "Data Group Description",
                                    "type": "CUSTOMERS"
                                }],
                            "unmodifiedDataGroups": [{
                                    "id": "10c387f2-4348-483e-ba98-f9b73926df4b",
                                    "name": "DGName",
                                    "description": "Data Group Description",
                                    "type": "ARRANGEMENTS"
                                }]
                        }],
                    "unmodifiedFunctionGroups": [{
                            "id": "08cb2ed4-2691-4a28-8219-37378917e58d",
                            "name": "FGName",
                            "type": "REGULAR",
                            "description": "Function Group Description",
                            "dataGroups": [{
                                    "id": "619f3d39-9647-46c7-ac70-6218382ebc4d",
                                    "name": "DGName",
                                    "description": "Data Group Description",
                                    "type": "CUSTOMERS"
                                }],
                            "selfApprovalPolicies": [{
                                    "businessFunctionCode": "manage.users",
                                    "canSelfApprove": false
                                }]
                        }],
                    "requiredApproves": 3,
                    "completedApproves": 1,
                    "approvalLog": [{
                            "approverId": "619f3d39-9647-46c7-ac70-6218382ebc4d",
                            "approverFullName": "Donald Trump",
                            "approvedAt": "2018-10-15T16:41:41.090Z"
                        }]
                }
            },
        ]
    }]);
export const ApprovalsHttpServiceGetServiceAgreementApprovalDetailsItemMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/approvals/{approvalId}/service-agreement",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "serviceAgreementId": "0889e686d31e4216b3dd5d66163d2b14",
                    "approvalId": "606d4532-f8d9-4a5f-36kl-887baf88fa24",
                    "action": "EDIT",
                    "oldState": {
                        "name": "FG1",
                        "description": "FG1 description",
                        "validFromDate": "2020-01-01",
                        "validFromTime": "00:00:00",
                        "validUntilDate": "2025-01-01",
                        "validUntilTime": "23:59:59",
                        "admins": ["0955e686d31e4216b3dd5d76561d536d", "0955e686d31e4216b3dd5d76164d536d", "0955e686d31e4216b3aa5d76161d536d"],
                        "legalEntities": [{
                                "name": "LEName1",
                                "id": "f41c4f3fea444cd2bfcf861dd13d2a1c",
                                "externalId": "C000001",
                                "contributeUsers": true,
                                "contributeAccount": true
                            }, {
                                "name": "LEName2",
                                "id": "f41c4f3fea444cd2bfcf861dd13d2a2c",
                                "externalId": "C000002",
                                "contributeUsers": true,
                                "contributeAccount": false
                            }]
                    },
                    "newState": {
                        "name": "FG1",
                        "description": "FG1 description",
                        "validFromDate": "2020-01-01",
                        "validFromTime": "00:00:00",
                        "validUntilDate": "2025-01-01",
                        "validUntilTime": "23:59:59",
                        "admins": ["0955e686d31e4216b3dd5d76561d536d", "0955e686d31e4216b3dd5d76164d536d", "0955e686d31e4216b3aa5d76161d522d"],
                        "legalEntities": [{
                                "name": "LEName1",
                                "id": "f41c4f3fea444cd2bfcf861dd13d2a1c",
                                "externalId": "C000001",
                                "contributeUsers": true,
                                "contributeAccount": true
                            }, {
                                "name": "LEName2",
                                "id": "f41c4f3fea444cd2bfcf861dd13d2a2c",
                                "externalId": "C000002",
                                "contributeUsers": true,
                                "contributeAccount": false
                            }, {
                                "name": "LEName3",
                                "id": "f41c4f3fea444cd2bfcf861dd13d2a4c",
                                "externalId": "C000003",
                                "contributeUsers": false,
                                "contributeAccount": true
                            }]
                    }
                }
            },
        ]
    }]);
export const ApprovalsHttpServiceGetWizardApprovalDetailsMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/approvals/{approvalId}/wizard-setup",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "approvalId": "test",
                    "action": "CREATE",
                    "category": "Assign Permissions by Wizard",
                    "users": [{
                            "userId": "c7a382786d514262b75ab9531b749a2b",
                            "userFullName": "Paddy Brian"
                        }],
                    "functionGroups": [{
                            "functionGroupId": "f34f7587-5ded-4f51-94c3-3a89d1b645392"
                        }, {
                            "functionGroupId": "f34f7587-5ded-4f51-94c3-3a89d1b6453812"
                        }],
                    "serviceAgreementId": "606d4532-f8d9-4a5f-840a-887baf88fa24",
                    "serviceAgreementName": "AIB",
                    "accounts": ["3cdb2224-8926-4b4d-a99f-1c9dfbbb4699", "6c1261e8-7b6b-4c14-9bb4-e976d834c730", "e6659eea893c4da48f9b7fd425de4677", "8eb58b1d048d4112a2f7591d465c3f14", "8a48e7546e11f733016e2249604f4076", "8a48e7546e11f733016e22495b6a4067", "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699", "8a48e7546e11f733016e22496d7e409d", "8a48e7546e11f733016e2249734640af", "8a48e7546e11f733016e22496ba94097", "8a48e7546e11f733016e22494c19403a", "8a48e7546e11f733016e2249715a40a9", "8a48e7546e11f733016e2249693f4091", "8a48e7546e11f733016e2249771240bb", "8a48e7546e11f733016e22496f5f40a3"],
                    "limit": {
                        "currency": "EUR",
                        "periodicLimitsBounds": {
                            "customPeriods": [{
                                    "amount": 100,
                                    "period": "biweekly"
                                }],
                            "daily": 100,
                            "monthly": 10000,
                            "quarterly": 100000,
                            "weekly": 1000
                        },
                        "transactionalLimitsBounds": {
                            "amount": 1000
                        }
                    },
                    "removeExistingPermissions": false
                }
            },
        ]
    }]);
export const ApprovalsHttpServicePostApproveApprovalRequestMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/approvals/{approvalId}/approve",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "approvalStatus": "PENDING"
                }
            },
        ]
    }]);
export const ApprovalsHttpServicePostRejectApprovalRequestMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/approvals/{approvalId}/reject",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "approvalStatus": "PENDING"
                }
            },
        ]
    }]);
export const ApprovalsHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/approvals",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "creatorUserFullName": "User 1",
                        "creatorUserId": "5643e686d31e4216b3ff5d66161d1234",
                        "createdAt": "2019-01-01T12:00:00Z",
                        "action": "EDIT",
                        "category": "Assign Permissions",
                        "approvalId": "2df100c6-b12b-4102-8831-db417acf12b6"
                    }, {
                        "creatorUserFullName": "User 2",
                        "creatorUserId": "5643e686d31e4216b3ff5d66161d1234",
                        "createdAt": "2019-01-01T12:00:00Z",
                        "action": "EDIT",
                        "category": "Manage Data Groups",
                        "approvalId": "2df100c6-b12b-4102-8831-db417acf12b7"
                    }, {
                        "creatorUserFullName": "User 3",
                        "creatorUserId": "5643e686d31e4216b3ff5d66161d1234",
                        "createdAt": "2019-01-01T12:00:00Z",
                        "action": "EDIT",
                        "category": "Unlock User",
                        "approvalId": "2df100c6-b12b-4102-8831-db417acf12b8"
                    }, {
                        "creatorUserFullName": "User 4",
                        "creatorUserId": "5643e686d31e4216b3ff5d66161d1234",
                        "createdAt": "2019-01-01T12:00:00Z",
                        "action": "EDIT",
                        "category": "Manage Limits",
                        "approvalId": "2df100c6-b12b-4102-8831-db417acf12b9"
                    }, {
                        "creatorUserFullName": "User 5",
                        "creatorUserId": "5643e686d31e4216b3ff5d66161d1234",
                        "createdAt": "2019-01-01T12:00:00Z",
                        "action": "EDIT",
                        "category": "Manage Shadow Limits",
                        "approvalId": "2df100c6-b12b-4102-8831-db417acf12b1"
                    }, {
                        "creatorUserFullName": "User 6",
                        "creatorUserId": "5643e686d31e3567b3ff5d66161d1234",
                        "createdAt": "2019-01-01T12:00:00Z",
                        "action": "EDIT",
                        "category": "Manage Function Groups",
                        "approvalId": "2df100c6-b12b-4102-8831-db417acf12c4"
                    }]
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/approvals/{approvalId}/data-group",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "dataGroupId": "0955e686d31e4216b3dd5d66161d536d",
                    "serviceAgreementId": "0889e686d31e4216b3dd5d66163d2b14",
                    "serviceAgreementName": "Sa name",
                    "type": "ARRANGEMENTS",
                    "approvalId": "606d4532-f8d9-4a5f-36kl-887baf88fa24",
                    "action": "EDIT",
                    "oldState": {
                        "name": "DG name",
                        "description": "DG desc"
                    },
                    "newState": {
                        "name": "new DG name",
                        "description": "new DG desc"
                    },
                    "addedDataItems": ["1cdb2224-8926-4b4d-a99f-1c9dfbbb4612", "1cdb2224-8926-4b4d-a99f-1c9dfbbb4613"],
                    "removedDataItems": ["1cdb2224-8926-4b4d-a99f-1c9dfbbb4699", "1cdb2224-8926-4b4d-a99f-1c9dfbbb4614"],
                    "unmodifiedDataItems": ["1cdb2224-8926-4b4d-a99f-1c9dfbbb4615", "1cdb2224-8926-4b4d-a99f-1c9dfbbb4611"],
                    "legalEntityIds": ["cd0968b3-3520-4280-8699-cd46c4d6df72", "3c684e80-c646-4148-bc20-37718eab4d97"]
                }
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/approvals/{approvalId}/function-group",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "functionGroupId": "0955e686d31e4216b3dd5d66161d536d",
                    "serviceAgreementId": "0889e686d31e4216b3dd5d66163d2b14",
                    "serviceAgreementName": "saName",
                    "approvalId": "606d4532-f8d9-4a5f-36kl-887baf88fa24",
                    "action": "EDIT",
                    "oldState": {
                        "name": "FG1",
                        "description": "FG1 description",
                        "approvalTypeId": "approvalTypeIdA",
                        "validFromDate": "2020-01-01",
                        "validFromTime": "00:00:00",
                        "validUntilDate": "2025-01-01",
                        "validUntilTime": "23:59:59"
                    },
                    "newState": {
                        "name": "FG2",
                        "description": "FG2 description",
                        "approvalTypeId": "approvalTypeIdB",
                        "validFromDate": "2020-06-10",
                        "validFromTime": "04:32:24",
                        "validUntilDate": "2023-01-01",
                        "validUntilTime": "23:53:51"
                    },
                    "permissionMatrix": [{
                            "functionId": "1012",
                            "functionCode": "manage.limits",
                            "resource": "Limits",
                            "name": "Manage Limits",
                            "action": "changed",
                            "privileges": {
                                "view": {
                                    "old": true,
                                    "new": false
                                },
                                "create": {
                                    "old": false,
                                    "new": true
                                },
                                "edit": {
                                    "old": true,
                                    "new": false
                                },
                                "delete": {
                                    "old": false,
                                    "new": false
                                },
                                "approve": {
                                    "old": true,
                                    "new": true
                                },
                                "cancel": {
                                    "old": false,
                                    "new": true
                                },
                                "execute": {
                                    "old": false,
                                    "new": true
                                }
                            }
                        }, {
                            "functionId": "1009",
                            "functionCode": "assign.permissions",
                            "resource": "Service Agreement",
                            "name": "Assign Permissions",
                            "action": "added",
                            "privileges": {
                                "view": {
                                    "new": true
                                },
                                "create": {
                                    "new": true
                                },
                                "edit": {
                                    "new": false
                                },
                                "delete": {
                                    "new": true
                                },
                                "approve": {
                                    "new": false
                                },
                                "cancel": {
                                    "new": true
                                },
                                "execute": {
                                    "new": true
                                }
                            }
                        }, {
                            "functionId": "1046",
                            "functionCode": "ach.debit",
                            "resource": "Payments",
                            "name": "ACH Debit",
                            "action": "removed",
                            "privileges": {
                                "view": {
                                    "old": true
                                },
                                "create": {
                                    "old": true
                                },
                                "edit": {
                                    "old": false
                                },
                                "delete": {
                                    "old": true
                                },
                                "approve": {
                                    "old": false
                                },
                                "cancel": {
                                    "old": true
                                },
                                "execute": {
                                    "old": true
                                }
                            }
                        }, {
                            "functionId": "1010",
                            "functionCode": "manage.users",
                            "resource": "User",
                            "name": "Manage Users",
                            "action": "unchanged",
                            "privileges": {
                                "view": {
                                    "old": true,
                                    "new": true
                                },
                                "create": {
                                    "old": true,
                                    "new": true
                                },
                                "edit": {
                                    "old": true,
                                    "new": true
                                },
                                "delete": {
                                    "old": false,
                                    "new": false
                                },
                                "approve": {
                                    "old": true,
                                    "new": true
                                },
                                "cancel": {
                                    "old": false,
                                    "new": false
                                }
                            }
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/approvals/{approvalId}/permissions",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "approvalId": "606d4532-f8d9-4a5f-36kl-887baf88fa24",
                    "creatorUserFullName": "John Snow",
                    "creatorUserId": "69ef5b3c-3997-4ee9-9f11-cb14447cf572",
                    "createdAt": "2018-09-28T16:41:41.090Z",
                    "userId": "606d4532-f8d9-4a5f-840a-887baf88fa24",
                    "userFullName": "Jack Sparrow",
                    "serviceAgreementId": "606d4532-f8d9-4a5f-840a-887baf88fa24",
                    "serviceAgreementName": "Backbase SA",
                    "serviceAgreementDescription": "Detailed desscription of the SA.",
                    "action": "EDIT",
                    "category": "Assign Permissions",
                    "newFunctionGroups": [{
                            "id": "c18dda63-27fa-4e4f-8882-24b24b0d46df",
                            "name": "FGName",
                            "type": "REGULAR",
                            "description": "Function Group Description",
                            "dataGroups": [{
                                    "id": "c18dda63-27fa-4e4f-8882-24b24b0d46df",
                                    "name": "DGName",
                                    "description": "Function Group Description",
                                    "type": "PAYEES"
                                }],
                            "selfApprovalPolicies": [{
                                    "businessFunctionCode": "manage.users",
                                    "canSelfApprove": false
                                }]
                        }],
                    "removedFunctionGroups": [{
                            "id": "0955e686-d31e-4216-b3dd-5d66161d536d",
                            "name": "FGName",
                            "type": "TEMPLATE",
                            "description": "Function Group Description",
                            "dataGroups": [{
                                    "id": "d9b299b0-ca28-4683-a3c1-9a339abfb5e4",
                                    "name": "DGName",
                                    "description": "Data Group Description",
                                    "type": "CONTACTS"
                                }],
                            "selfApprovalPolicies": [{
                                    "businessFunctionCode": "manage.users",
                                    "canSelfApprove": false
                                }]
                        }],
                    "modifiedFunctionGroups": [{
                            "id": "d9b299b0-ca28-4683-a3c1-9a339abfb5e4",
                            "name": "FGName",
                            "type": "TEMPLATE",
                            "description": "Function Group Description",
                            "newDataGroups": [{
                                    "id": "5690bd25-b9db-493a-9659-3d5e505c7358",
                                    "name": "DGName",
                                    "description": "Data Group Description",
                                    "type": "CONTACTS"
                                }],
                            "removedDataGroups": [{
                                    "id": "5f5c3714-289e-4f0c-bdb0-0b5a749f8beb",
                                    "name": "DGName",
                                    "description": "Data Group Description",
                                    "type": "CUSTOMERS"
                                }],
                            "unmodifiedDataGroups": [{
                                    "id": "10c387f2-4348-483e-ba98-f9b73926df4b",
                                    "name": "DGName",
                                    "description": "Data Group Description",
                                    "type": "ARRANGEMENTS"
                                }]
                        }],
                    "unmodifiedFunctionGroups": [{
                            "id": "08cb2ed4-2691-4a28-8219-37378917e58d",
                            "name": "FGName",
                            "type": "REGULAR",
                            "description": "Function Group Description",
                            "dataGroups": [{
                                    "id": "619f3d39-9647-46c7-ac70-6218382ebc4d",
                                    "name": "DGName",
                                    "description": "Data Group Description",
                                    "type": "CUSTOMERS"
                                }],
                            "selfApprovalPolicies": [{
                                    "businessFunctionCode": "manage.users",
                                    "canSelfApprove": false
                                }]
                        }],
                    "requiredApproves": 3,
                    "completedApproves": 1,
                    "approvalLog": [{
                            "approverId": "619f3d39-9647-46c7-ac70-6218382ebc4d",
                            "approverFullName": "Donald Trump",
                            "approvedAt": "2018-10-15T16:41:41.090Z"
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/approvals/{approvalId}/service-agreement",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "serviceAgreementId": "0889e686d31e4216b3dd5d66163d2b14",
                    "approvalId": "606d4532-f8d9-4a5f-36kl-887baf88fa24",
                    "action": "EDIT",
                    "oldState": {
                        "name": "FG1",
                        "description": "FG1 description",
                        "validFromDate": "2020-01-01",
                        "validFromTime": "00:00:00",
                        "validUntilDate": "2025-01-01",
                        "validUntilTime": "23:59:59",
                        "admins": ["0955e686d31e4216b3dd5d76561d536d", "0955e686d31e4216b3dd5d76164d536d", "0955e686d31e4216b3aa5d76161d536d"],
                        "legalEntities": [{
                                "name": "LEName1",
                                "id": "f41c4f3fea444cd2bfcf861dd13d2a1c",
                                "externalId": "C000001",
                                "contributeUsers": true,
                                "contributeAccount": true
                            }, {
                                "name": "LEName2",
                                "id": "f41c4f3fea444cd2bfcf861dd13d2a2c",
                                "externalId": "C000002",
                                "contributeUsers": true,
                                "contributeAccount": false
                            }]
                    },
                    "newState": {
                        "name": "FG1",
                        "description": "FG1 description",
                        "validFromDate": "2020-01-01",
                        "validFromTime": "00:00:00",
                        "validUntilDate": "2025-01-01",
                        "validUntilTime": "23:59:59",
                        "admins": ["0955e686d31e4216b3dd5d76561d536d", "0955e686d31e4216b3dd5d76164d536d", "0955e686d31e4216b3aa5d76161d522d"],
                        "legalEntities": [{
                                "name": "LEName1",
                                "id": "f41c4f3fea444cd2bfcf861dd13d2a1c",
                                "externalId": "C000001",
                                "contributeUsers": true,
                                "contributeAccount": true
                            }, {
                                "name": "LEName2",
                                "id": "f41c4f3fea444cd2bfcf861dd13d2a2c",
                                "externalId": "C000002",
                                "contributeUsers": true,
                                "contributeAccount": false
                            }, {
                                "name": "LEName3",
                                "id": "f41c4f3fea444cd2bfcf861dd13d2a4c",
                                "externalId": "C000003",
                                "contributeUsers": false,
                                "contributeAccount": true
                            }]
                    }
                }
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/approvals/{approvalId}/wizard-setup",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "approvalId": "test",
                    "action": "CREATE",
                    "category": "Assign Permissions by Wizard",
                    "users": [{
                            "userId": "c7a382786d514262b75ab9531b749a2b",
                            "userFullName": "Paddy Brian"
                        }],
                    "functionGroups": [{
                            "functionGroupId": "f34f7587-5ded-4f51-94c3-3a89d1b645392"
                        }, {
                            "functionGroupId": "f34f7587-5ded-4f51-94c3-3a89d1b6453812"
                        }],
                    "serviceAgreementId": "606d4532-f8d9-4a5f-840a-887baf88fa24",
                    "serviceAgreementName": "AIB",
                    "accounts": ["3cdb2224-8926-4b4d-a99f-1c9dfbbb4699", "6c1261e8-7b6b-4c14-9bb4-e976d834c730", "e6659eea893c4da48f9b7fd425de4677", "8eb58b1d048d4112a2f7591d465c3f14", "8a48e7546e11f733016e2249604f4076", "8a48e7546e11f733016e22495b6a4067", "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699", "8a48e7546e11f733016e22496d7e409d", "8a48e7546e11f733016e2249734640af", "8a48e7546e11f733016e22496ba94097", "8a48e7546e11f733016e22494c19403a", "8a48e7546e11f733016e2249715a40a9", "8a48e7546e11f733016e2249693f4091", "8a48e7546e11f733016e2249771240bb", "8a48e7546e11f733016e22496f5f40a3"],
                    "limit": {
                        "currency": "EUR",
                        "periodicLimitsBounds": {
                            "customPeriods": [{
                                    "amount": 100,
                                    "period": "biweekly"
                                }],
                            "daily": 100,
                            "monthly": 10000,
                            "quarterly": 100000,
                            "weekly": 1000
                        },
                        "transactionalLimitsBounds": {
                            "amount": 1000
                        }
                    },
                    "removeExistingPermissions": false
                }
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/approvals/{approvalId}/approve",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "approvalStatus": "PENDING"
                }
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/approvals/{approvalId}/reject",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "approvalStatus": "PENDING"
                }
            },
        ]
    },
]);
