// @backbase/accesscontrol-http-ang/esm2015/api/functionGroups.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const FunctionGroupsHttpServiceDeleteFunctionGroupByIdMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/function-groups/{id}",
        method: "DELETE",
        responses: [
            {
                status: 202,
                body: {}
            },
            {
                status: 200,
                body: {}
            },
        ]
    }]);
export const FunctionGroupsHttpServiceGetFunctionGroupByIdMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/function-groups/{id}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "f34f7587-5ded-4f51-94c3-3a89d1b64539",
                    "name": "Manage Entitlements",
                    "description": "This profile will control managing of entitlements",
                    "serviceAgreementId": "1642b2e1-61c4-4ce3-8bd9-877327587f4b",
                    "permissions": [{
                            "functionId": "1012",
                            "assignedPrivileges": [{
                                    "privilege": "view"
                                }]
                        }, {
                            "functionId": "1011",
                            "assignedPrivileges": [{
                                    "privilege": "view"
                                }]
                        }, {
                            "functionId": "1010",
                            "assignedPrivileges": [{
                                    "privilege": "view"
                                }]
                        }, {
                            "functionId": "1009",
                            "assignedPrivileges": [{
                                    "privilege": "create"
                                }, {
                                    "privilege": "view"
                                }, {
                                    "privilege": "edit"
                                }]
                        }]
                }
            },
        ]
    }]);
export const FunctionGroupsHttpServiceGetFunctionGroupsMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/function-groups",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "00779897-253c-4bbb-b0c7-3c47fd0b0b08",
                        "name": "Manage Entitlements",
                        "description": "This profile will control managing of entitlements",
                        "serviceAgreementId": "51c95177-0d29-43f6-978c-be8c0b9126ab",
                        "permissions": [{
                                "functionId": "1012",
                                "assignedPrivileges": [{
                                        "privilege": "view"
                                    }]
                            }, {
                                "functionId": "1011",
                                "assignedPrivileges": [{
                                        "privilege": "view"
                                    }]
                            }, {
                                "functionId": "1010",
                                "assignedPrivileges": [{
                                        "privilege": "view"
                                    }]
                            }, {
                                "functionId": "1009",
                                "assignedPrivileges": [{
                                        "privilege": "create"
                                    }, {
                                        "privilege": "view"
                                    }, {
                                        "privilege": "edit"
                                    }]
                            }],
                        "validFromDate": "2019-08-16",
                        "validFromTime": "16:43:52",
                        "validUntilDate": "2019-08-23",
                        "validUntilTime": "16:43:52",
                        "type": "REGULAR"
                    }, {
                        "id": "98f8531d-9bf4-4f3f-a49b-7cf331d6eead",
                        "name": "Viewer of Product Summary and Transactions",
                        "description": "This profile will control viewing of Product Summary and Transactions",
                        "legalEntityId": "be2a6940-91f9-4cba-9c35-0b1f55f4071d",
                        "serviceAgreementId": "f34f7587-5ded-4f51-94c3-3a89d1b64539",
                        "permissions": [{
                                "functionId": "1006",
                                "assignedPrivileges": [{
                                        "privilege": "view"
                                    }]
                            }, {
                                "functionId": "1003",
                                "assignedPrivileges": [{
                                        "privilege": "view"
                                    }]
                            }],
                        "approvalId": "98f8531d-9bf4-4f3f-a49b-7cf331d6eead",
                        "validFromDate": "2019-09-17",
                        "validFromTime": "16:43:57",
                        "validUntilDate": "2019-10-25",
                        "validUntilTime": "16:55:23",
                        "type": "TEMPLATE"
                    }, {
                        "id": "98f8531d-9bf4-4f3f-a49b-7cf331d6eeae",
                        "name": "Limits Manager",
                        "description": "This profile will control managing of limits",
                        "legalEntityId": "be2a6940-91f9-4cba-9c35-0b1f55f4071d",
                        "serviceAgreementId": "f34f7587-5ded-4f51-94c3-3a89d1b64539",
                        "permissions": [{
                                "functionId": "1012",
                                "assignedPrivileges": [{
                                        "privilege": "view"
                                    }, {
                                        "privilege": "create"
                                    }, {
                                        "privilege": "edit"
                                    }, {
                                        "privilege": "delete"
                                    }]
                            }, {
                                "functionId": "1014",
                                "assignedPrivileges": [{
                                        "privilege": "view"
                                    }]
                            }],
                        "validFromDate": "2019-09-17",
                        "validFromTime": "16:43:57",
                        "validUntilDate": "2019-10-25",
                        "validUntilTime": "16:55:23",
                        "type": "TEMPLATE"
                    }, {
                        "id": "98f8531d-9bf4-4f3f-a49b-7cf331d6eeaf",
                        "name": "Permissions Approvals Manager",
                        "description": "This profile will control approving of Permissions",
                        "legalEntityId": "be2a6940-91f9-4cba-9c35-0b1f55f4071d",
                        "serviceAgreementId": "f34f7587-5ded-4f51-94c3-3a89d1b64539",
                        "permissions": [{
                                "functionId": "1009",
                                "assignedPrivileges": [{
                                        "privilege": "view"
                                    }, {
                                        "privilege": "delete"
                                    }]
                            }, {
                                "functionId": "1020",
                                "assignedPrivileges": [{
                                        "privilege": "view"
                                    }]
                            }, {
                                "functionId": "1019",
                                "assignedPrivileges": [{
                                        "privilege": "view"
                                    }]
                            }],
                        "validFromDate": "2019-09-17",
                        "validFromTime": "16:43:57",
                        "validUntilDate": "2019-10-25",
                        "validUntilTime": "16:55:23",
                        "type": "TEMPLATE"
                    }, {
                        "id": "98f8531d-9bf4-4f3f-a49b-7cf331d6feaf",
                        "name": "Salary Manager",
                        "description": "This profile will control paying salaries",
                        "legalEntityId": "be2a6940-91f9-4cba-9c35-0b1f55f4071d",
                        "serviceAgreementId": "f34f7587-5ded-4f51-94c3-3a89d1b64539",
                        "permissions": [{
                                "functionId": "1015",
                                "assignedPrivileges": [{
                                        "privilege": "view"
                                    }, {
                                        "privilege": "create"
                                    }, {
                                        "privilege": "edit"
                                    }, {
                                        "privilege": "delete"
                                    }]
                            }],
                        "validFromDate": "2019-09-17",
                        "validFromTime": "16:43:57",
                        "validUntilDate": "2019-10-25",
                        "validUntilTime": "16:55:23",
                        "type": "REGULAR"
                    }]
            },
        ]
    }]);
export const FunctionGroupsHttpServicePostFunctionGroupsMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/function-groups",
        method: "POST",
        responses: [
            {
                status: 202,
                body: {
                    "id": "0955e686d31e4216b3dd5d66161d536d"
                }
            },
            {
                status: 201,
                body: {
                    "id": "0955e686d31e4216b3dd5d66161d536d"
                }
            },
        ]
    }]);
export const FunctionGroupsHttpServicePutFunctionGroupByIdMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/function-groups/{id}",
        method: "PUT",
        responses: [
            {
                status: 202,
                body: {}
            },
            {
                status: 200,
                body: {}
            },
        ]
    }]);
export const FunctionGroupsHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/function-groups/{id}",
        method: "DELETE",
        responses: [
            {
                status: 202,
                body: {}
            },
            {
                status: 200,
                body: {}
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/function-groups/{id}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "f34f7587-5ded-4f51-94c3-3a89d1b64539",
                    "name": "Manage Entitlements",
                    "description": "This profile will control managing of entitlements",
                    "serviceAgreementId": "1642b2e1-61c4-4ce3-8bd9-877327587f4b",
                    "permissions": [{
                            "functionId": "1012",
                            "assignedPrivileges": [{
                                    "privilege": "view"
                                }]
                        }, {
                            "functionId": "1011",
                            "assignedPrivileges": [{
                                    "privilege": "view"
                                }]
                        }, {
                            "functionId": "1010",
                            "assignedPrivileges": [{
                                    "privilege": "view"
                                }]
                        }, {
                            "functionId": "1009",
                            "assignedPrivileges": [{
                                    "privilege": "create"
                                }, {
                                    "privilege": "view"
                                }, {
                                    "privilege": "edit"
                                }]
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/function-groups",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "00779897-253c-4bbb-b0c7-3c47fd0b0b08",
                        "name": "Manage Entitlements",
                        "description": "This profile will control managing of entitlements",
                        "serviceAgreementId": "51c95177-0d29-43f6-978c-be8c0b9126ab",
                        "permissions": [{
                                "functionId": "1012",
                                "assignedPrivileges": [{
                                        "privilege": "view"
                                    }]
                            }, {
                                "functionId": "1011",
                                "assignedPrivileges": [{
                                        "privilege": "view"
                                    }]
                            }, {
                                "functionId": "1010",
                                "assignedPrivileges": [{
                                        "privilege": "view"
                                    }]
                            }, {
                                "functionId": "1009",
                                "assignedPrivileges": [{
                                        "privilege": "create"
                                    }, {
                                        "privilege": "view"
                                    }, {
                                        "privilege": "edit"
                                    }]
                            }],
                        "validFromDate": "2019-08-16",
                        "validFromTime": "16:43:52",
                        "validUntilDate": "2019-08-23",
                        "validUntilTime": "16:43:52",
                        "type": "REGULAR"
                    }, {
                        "id": "98f8531d-9bf4-4f3f-a49b-7cf331d6eead",
                        "name": "Viewer of Product Summary and Transactions",
                        "description": "This profile will control viewing of Product Summary and Transactions",
                        "legalEntityId": "be2a6940-91f9-4cba-9c35-0b1f55f4071d",
                        "serviceAgreementId": "f34f7587-5ded-4f51-94c3-3a89d1b64539",
                        "permissions": [{
                                "functionId": "1006",
                                "assignedPrivileges": [{
                                        "privilege": "view"
                                    }]
                            }, {
                                "functionId": "1003",
                                "assignedPrivileges": [{
                                        "privilege": "view"
                                    }]
                            }],
                        "approvalId": "98f8531d-9bf4-4f3f-a49b-7cf331d6eead",
                        "validFromDate": "2019-09-17",
                        "validFromTime": "16:43:57",
                        "validUntilDate": "2019-10-25",
                        "validUntilTime": "16:55:23",
                        "type": "TEMPLATE"
                    }, {
                        "id": "98f8531d-9bf4-4f3f-a49b-7cf331d6eeae",
                        "name": "Limits Manager",
                        "description": "This profile will control managing of limits",
                        "legalEntityId": "be2a6940-91f9-4cba-9c35-0b1f55f4071d",
                        "serviceAgreementId": "f34f7587-5ded-4f51-94c3-3a89d1b64539",
                        "permissions": [{
                                "functionId": "1012",
                                "assignedPrivileges": [{
                                        "privilege": "view"
                                    }, {
                                        "privilege": "create"
                                    }, {
                                        "privilege": "edit"
                                    }, {
                                        "privilege": "delete"
                                    }]
                            }, {
                                "functionId": "1014",
                                "assignedPrivileges": [{
                                        "privilege": "view"
                                    }]
                            }],
                        "validFromDate": "2019-09-17",
                        "validFromTime": "16:43:57",
                        "validUntilDate": "2019-10-25",
                        "validUntilTime": "16:55:23",
                        "type": "TEMPLATE"
                    }, {
                        "id": "98f8531d-9bf4-4f3f-a49b-7cf331d6eeaf",
                        "name": "Permissions Approvals Manager",
                        "description": "This profile will control approving of Permissions",
                        "legalEntityId": "be2a6940-91f9-4cba-9c35-0b1f55f4071d",
                        "serviceAgreementId": "f34f7587-5ded-4f51-94c3-3a89d1b64539",
                        "permissions": [{
                                "functionId": "1009",
                                "assignedPrivileges": [{
                                        "privilege": "view"
                                    }, {
                                        "privilege": "delete"
                                    }]
                            }, {
                                "functionId": "1020",
                                "assignedPrivileges": [{
                                        "privilege": "view"
                                    }]
                            }, {
                                "functionId": "1019",
                                "assignedPrivileges": [{
                                        "privilege": "view"
                                    }]
                            }],
                        "validFromDate": "2019-09-17",
                        "validFromTime": "16:43:57",
                        "validUntilDate": "2019-10-25",
                        "validUntilTime": "16:55:23",
                        "type": "TEMPLATE"
                    }, {
                        "id": "98f8531d-9bf4-4f3f-a49b-7cf331d6feaf",
                        "name": "Salary Manager",
                        "description": "This profile will control paying salaries",
                        "legalEntityId": "be2a6940-91f9-4cba-9c35-0b1f55f4071d",
                        "serviceAgreementId": "f34f7587-5ded-4f51-94c3-3a89d1b64539",
                        "permissions": [{
                                "functionId": "1015",
                                "assignedPrivileges": [{
                                        "privilege": "view"
                                    }, {
                                        "privilege": "create"
                                    }, {
                                        "privilege": "edit"
                                    }, {
                                        "privilege": "delete"
                                    }]
                            }],
                        "validFromDate": "2019-09-17",
                        "validFromTime": "16:43:57",
                        "validUntilDate": "2019-10-25",
                        "validUntilTime": "16:55:23",
                        "type": "REGULAR"
                    }]
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/function-groups",
        method: "POST",
        responses: [
            {
                status: 202,
                body: {
                    "id": "0955e686d31e4216b3dd5d66161d536d"
                }
            },
            {
                status: 201,
                body: {
                    "id": "0955e686d31e4216b3dd5d66161d536d"
                }
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/function-groups/{id}",
        method: "PUT",
        responses: [
            {
                status: 202,
                body: {}
            },
            {
                status: 200,
                body: {}
            },
        ]
    },
]);
