// @backbase/accesscontrol-http-ang/esm2015/api/legalEntities.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const LegalEntitiesHttpServiceGetLegalEntitiesMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/legalentities",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "c7a382786d514262b75ab9531b749a2b",
                        "externalId": "Bank",
                        "name": "Bank",
                        "type": "BANK",
                        "isParent": true
                    }, {
                        "id": "dbc9e566d44f4cd9a1f09ad3f6729f15",
                        "externalId": "KPMG",
                        "name": "KPMG",
                        "parentId": "c7a382786d514262b75ab9531b749a2b",
                        "type": "CUSTOMER",
                        "isParent": false
                    }, {
                        "id": "77f8c4cd11c8436ba77f8ca34e6a173d",
                        "externalId": "Backbase",
                        "name": "Backbase",
                        "parentId": "c7a382786d514262b75ab9531b749a2b",
                        "type": "CUSTOMER",
                        "isParent": true
                    }, {
                        "id": "c9fc30f4e5fb492c97ed0a05f926aecc",
                        "externalId": "Backbase Amsterdam",
                        "name": "Backbase Amsterdam",
                        "parentId": "77f8c4cd11c8436ba77f8ca34e6a173d",
                        "type": "CUSTOMER",
                        "isParent": false
                    }, {
                        "id": "0f6d14f40e064deeaee351befd53d3e7",
                        "externalId": "Backbase Cardiff",
                        "name": "Backbase Cardiff",
                        "parentId": "77f8c4cd11c8436ba77f8ca34e6a173d",
                        "type": "CUSTOMER",
                        "isParent": false
                    }]
            },
        ]
    }]);
export const LegalEntitiesHttpServiceGetLegalEntityByExternalIdMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/legalentities/external/{externalId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "1",
                    "externalId": "ext1",
                    "type": "CUSTOMER",
                    "name": "Endava"
                }
            },
        ]
    }]);
export const LegalEntitiesHttpServiceGetLegalEntityByIdMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/legalentities/{legalEntityId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "e1d55a46-656a-491b-b6e0-443ff09bbe52",
                    "externalId": "ext1",
                    "name": "Endava",
                    "parentId": "e1d55a46-656a-491b-b6e0-443ff09bbe52",
                    "type": "CUSTOMER",
                    "isParent": false
                }
            },
        ]
    }]);
export const LegalEntitiesHttpServiceGetLegalEntityExternalDataMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/legalentities/external-data",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "c7a382786d514262b75ab9531b749a2b",
                        "externalId": "Bank",
                        "name": "Bank",
                        "type": "BANK"
                    }, {
                        "id": "77f8c4cd11c8436ba77f8ca34e6a173d",
                        "externalId": "Backbase",
                        "name": "Backbase",
                        "type": "CUSTOMER"
                    }, {
                        "externalId": "Backbase Amsterdam",
                        "name": "Backbase Amsterdam"
                    }, {
                        "externalId": "Backbase Cardiff",
                        "name": "Backbase Cardiff"
                    }]
            },
        ]
    }]);
export const LegalEntitiesHttpServiceGetLegalEntityForUserMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/legalentities/me",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "e1d55a46-656a-491b-b6e0-443ff09bbe52",
                    "externalId": "ext1",
                    "name": "Endava",
                    "parentId": "e1d55a46-656a-491b-b6e0-443ff09bbe52",
                    "type": "CUSTOMER",
                    "isParent": false
                }
            },
        ]
    }]);
export const LegalEntitiesHttpServiceGetMasterServiceAgreementMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/legalentities/{legalEntityId}/serviceagreements/master",
        method: "GET",
        responses: []
    }]);
export const LegalEntitiesHttpServiceGetSegmentationMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/legalentities/segmentation",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "c7a382786d514262b75ab9531b749a2b",
                        "externalId": "Bank",
                        "name": "Bank",
                        "type": "BANK",
                        "isParent": true
                    }, {
                        "id": "dbc9e566d44f4cd9a1f09ad3f6729f15",
                        "externalId": "KPMG",
                        "name": "KPMG",
                        "parentId": "c7a382786d514262b75ab9531b749a2b",
                        "type": "CUSTOMER",
                        "isParent": false
                    }, {
                        "id": "77f8c4cd11c8436ba77f8ca34e6a173d",
                        "externalId": "Backbase",
                        "name": "Backbase",
                        "parentId": "c7a382786d514262b75ab9531b749a2b",
                        "type": "CUSTOMER",
                        "isParent": true
                    }, {
                        "id": "c9fc30f4e5fb492c97ed0a05f926aecc",
                        "externalId": "Backbase Amsterdam",
                        "name": "Backbase Amsterdam",
                        "parentId": "77f8c4cd11c8436ba77f8ca34e6a173d",
                        "type": "CUSTOMER",
                        "isParent": false
                    }, {
                        "id": "0f6d14f40e064deeaee351befd53d3e7",
                        "externalId": "Backbase Cardiff",
                        "name": "Backbase Cardiff",
                        "parentId": "77f8c4cd11c8436ba77f8ca34e6a173d",
                        "type": "CUSTOMER",
                        "isParent": false
                    }]
            },
        ]
    }]);
export const LegalEntitiesHttpServicePostLegalEntitiesMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/legalentities",
        method: "POST",
        responses: [
            {
                status: 201,
                body: {
                    "id": "0955e686-d31e-4216-b3dd-5d66161d536d"
                }
            },
        ]
    }]);
export const LegalEntitiesHttpServicePostLegalEntitiesAsParticipantMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/legalentities/as-participant",
        method: "POST",
        responses: [
            {
                status: 201,
                body: {
                    "legalEntityId": "0955e686-d31e-4216-b3dd-5d66161d536d",
                    "serviceAgreementId": "0955e686-d31e-4216-b3dd-5d66161d536d"
                }
            },
        ]
    }]);
export const LegalEntitiesHttpServicePostSubEntitiesMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/legalentities/sub-entities",
        method: "POST",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "c7a382786d514262b75ab9531b749a2b",
                        "externalId": "Bank",
                        "name": "Bank",
                        "type": "BANK"
                    }, {
                        "id": "dbc9e566d44f4cd9a1f09ad3f6729f15",
                        "externalId": "KPMG",
                        "name": "KPMG",
                        "type": "CUSTOMER"
                    }, {
                        "id": "77f8c4cd11c8436ba77f8ca34e6a173d",
                        "externalId": "Backbase",
                        "name": "Backbase",
                        "type": "CUSTOMER"
                    }, {
                        "id": "c9fc30f4e5fb492c97ed0a05f926aecc",
                        "externalId": "Backbase Amsterdam",
                        "name": "Backbase Amsterdam",
                        "type": "CUSTOMER"
                    }, {
                        "id": "0f6d14f40e064deeaee351befd53d3e7",
                        "externalId": "Backbase Cardiff",
                        "name": "Backbase Cardiff",
                        "type": "CUSTOMER"
                    }]
            },
        ]
    }]);
export const LegalEntitiesHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/access-control/client-api/v2/legalentities",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "c7a382786d514262b75ab9531b749a2b",
                        "externalId": "Bank",
                        "name": "Bank",
                        "type": "BANK",
                        "isParent": true
                    }, {
                        "id": "dbc9e566d44f4cd9a1f09ad3f6729f15",
                        "externalId": "KPMG",
                        "name": "KPMG",
                        "parentId": "c7a382786d514262b75ab9531b749a2b",
                        "type": "CUSTOMER",
                        "isParent": false
                    }, {
                        "id": "77f8c4cd11c8436ba77f8ca34e6a173d",
                        "externalId": "Backbase",
                        "name": "Backbase",
                        "parentId": "c7a382786d514262b75ab9531b749a2b",
                        "type": "CUSTOMER",
                        "isParent": true
                    }, {
                        "id": "c9fc30f4e5fb492c97ed0a05f926aecc",
                        "externalId": "Backbase Amsterdam",
                        "name": "Backbase Amsterdam",
                        "parentId": "77f8c4cd11c8436ba77f8ca34e6a173d",
                        "type": "CUSTOMER",
                        "isParent": false
                    }, {
                        "id": "0f6d14f40e064deeaee351befd53d3e7",
                        "externalId": "Backbase Cardiff",
                        "name": "Backbase Cardiff",
                        "parentId": "77f8c4cd11c8436ba77f8ca34e6a173d",
                        "type": "CUSTOMER",
                        "isParent": false
                    }]
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/legalentities/external/{externalId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "1",
                    "externalId": "ext1",
                    "type": "CUSTOMER",
                    "name": "Endava"
                }
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/legalentities/{legalEntityId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "e1d55a46-656a-491b-b6e0-443ff09bbe52",
                    "externalId": "ext1",
                    "name": "Endava",
                    "parentId": "e1d55a46-656a-491b-b6e0-443ff09bbe52",
                    "type": "CUSTOMER",
                    "isParent": false
                }
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/legalentities/external-data",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "c7a382786d514262b75ab9531b749a2b",
                        "externalId": "Bank",
                        "name": "Bank",
                        "type": "BANK"
                    }, {
                        "id": "77f8c4cd11c8436ba77f8ca34e6a173d",
                        "externalId": "Backbase",
                        "name": "Backbase",
                        "type": "CUSTOMER"
                    }, {
                        "externalId": "Backbase Amsterdam",
                        "name": "Backbase Amsterdam"
                    }, {
                        "externalId": "Backbase Cardiff",
                        "name": "Backbase Cardiff"
                    }]
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/legalentities/me",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "e1d55a46-656a-491b-b6e0-443ff09bbe52",
                    "externalId": "ext1",
                    "name": "Endava",
                    "parentId": "e1d55a46-656a-491b-b6e0-443ff09bbe52",
                    "type": "CUSTOMER",
                    "isParent": false
                }
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/legalentities/{legalEntityId}/serviceagreements/master",
        method: "GET",
        responses: []
    },
    {
        urlPattern: "/access-control/client-api/v2/legalentities/segmentation",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "c7a382786d514262b75ab9531b749a2b",
                        "externalId": "Bank",
                        "name": "Bank",
                        "type": "BANK",
                        "isParent": true
                    }, {
                        "id": "dbc9e566d44f4cd9a1f09ad3f6729f15",
                        "externalId": "KPMG",
                        "name": "KPMG",
                        "parentId": "c7a382786d514262b75ab9531b749a2b",
                        "type": "CUSTOMER",
                        "isParent": false
                    }, {
                        "id": "77f8c4cd11c8436ba77f8ca34e6a173d",
                        "externalId": "Backbase",
                        "name": "Backbase",
                        "parentId": "c7a382786d514262b75ab9531b749a2b",
                        "type": "CUSTOMER",
                        "isParent": true
                    }, {
                        "id": "c9fc30f4e5fb492c97ed0a05f926aecc",
                        "externalId": "Backbase Amsterdam",
                        "name": "Backbase Amsterdam",
                        "parentId": "77f8c4cd11c8436ba77f8ca34e6a173d",
                        "type": "CUSTOMER",
                        "isParent": false
                    }, {
                        "id": "0f6d14f40e064deeaee351befd53d3e7",
                        "externalId": "Backbase Cardiff",
                        "name": "Backbase Cardiff",
                        "parentId": "77f8c4cd11c8436ba77f8ca34e6a173d",
                        "type": "CUSTOMER",
                        "isParent": false
                    }]
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/legalentities",
        method: "POST",
        responses: [
            {
                status: 201,
                body: {
                    "id": "0955e686-d31e-4216-b3dd-5d66161d536d"
                }
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/legalentities/as-participant",
        method: "POST",
        responses: [
            {
                status: 201,
                body: {
                    "legalEntityId": "0955e686-d31e-4216-b3dd-5d66161d536d",
                    "serviceAgreementId": "0955e686-d31e-4216-b3dd-5d66161d536d"
                }
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/legalentities/sub-entities",
        method: "POST",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "c7a382786d514262b75ab9531b749a2b",
                        "externalId": "Bank",
                        "name": "Bank",
                        "type": "BANK"
                    }, {
                        "id": "dbc9e566d44f4cd9a1f09ad3f6729f15",
                        "externalId": "KPMG",
                        "name": "KPMG",
                        "type": "CUSTOMER"
                    }, {
                        "id": "77f8c4cd11c8436ba77f8ca34e6a173d",
                        "externalId": "Backbase",
                        "name": "Backbase",
                        "type": "CUSTOMER"
                    }, {
                        "id": "c9fc30f4e5fb492c97ed0a05f926aecc",
                        "externalId": "Backbase Amsterdam",
                        "name": "Backbase Amsterdam",
                        "type": "CUSTOMER"
                    }, {
                        "id": "0f6d14f40e064deeaee351befd53d3e7",
                        "externalId": "Backbase Cardiff",
                        "name": "Backbase Cardiff",
                        "type": "CUSTOMER"
                    }]
            },
        ]
    },
]);
