// @backbase/accesscontrol-http-ang/esm2015/api/serviceAgreement.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const ServiceAgreementHttpServiceGetServiceAgreementMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/serviceagreements",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "951293e6521940e3ae0608f93ff0b215",
                        "externalId": "external-id-1",
                        "name": "Broker deal 1",
                        "description": "Agreement between Backbase and Apple",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96147",
                        "externalId": "external-id-2",
                        "name": "Broker deal 2",
                        "description": "Agreement between Backbase and HP",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae0a1e42bbe96147",
                        "externalId": "external-id-3",
                        "name": "Tyler Boyd Associates",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae1a1e42bbe96147",
                        "externalId": "external-id-4",
                        "name": "Tyler Boyd Associates UAE",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae2a1e42bbe96147",
                        "externalId": "external-id-5",
                        "name": "Tyler Boyd Associates US",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae3a1e42bbe96147",
                        "externalId": "external-id-6",
                        "name": "Tyler Boyd Associates US Salary Services",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae4a1e42bbe96147",
                        "externalId": "external-id-7",
                        "name": "Tyler Boyd Associates US Training",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae5a1e42bbe96147",
                        "externalId": "external-id-8",
                        "name": "Tyler Boyd Associates UK",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae6a1e42bbe96147",
                        "externalId": "external-id-9",
                        "name": "Tyler Boyd Associates UK Sales",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae7a1e42bbe96147",
                        "externalId": "external-id-10",
                        "name": "Tyler Boyd Associates UK Marketing",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae9a1e42bbe96147",
                        "externalId": "external-id-11",
                        "name": "Broker deal 11",
                        "description": "Agreement between Backbase and Microsoft",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a0e42bbe96147",
                        "externalId": "external-id-12",
                        "name": "Broker deal 12",
                        "description": "Agreement between Backbase and Packard-Bell",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a2e42bbe96147",
                        "externalId": "external-id-13",
                        "name": "Tyler Boyd Associates ROI",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a3e42bbe96147",
                        "externalId": "external-id-14",
                        "name": "Tyler Boyd Associates ROI Services",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a4e42bbe96147",
                        "externalId": "external-id-15",
                        "name": "Tyler Boyd Associates ROI Finance",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a5e42bbe96147",
                        "externalId": "external-id-16",
                        "name": "Tyler Boyd Associates ROI HR",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a6e42bbe96147",
                        "externalId": "external-id-17",
                        "name": "Tyler Boyd Associates EU",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a7e42bbe96147",
                        "externalId": "external-id-18",
                        "name": "Tyler Boyd Associates EU North",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a8e42bbe96147",
                        "externalId": "external-id-19",
                        "name": "Tyler Boyd Associates EU East",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a9e02bbe96147",
                        "externalId": "external-id-20",
                        "name": "Tyler Boyd Associates EU South",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e12bbe96147",
                        "externalId": "external-id-21",
                        "name": "Broker deal 21",
                        "description": "Agreement between Backbase and Google",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e22bbe96147",
                        "externalId": "external-id-22",
                        "name": "Broker deal 22",
                        "description": "Agreement between Backbase and Dell",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e32bbe96147",
                        "externalId": "external-id-23",
                        "name": "Tyler Boyd Associates EU West",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e52bbe96147",
                        "externalId": "external-id-24",
                        "name": "Tyler Boyd Associates EU Central",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e62bbe96147",
                        "externalId": "external-id-25",
                        "name": "Tyler Boyd Associates CAN",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e72bbe96147",
                        "externalId": "external-id-26",
                        "name": "Tyler Boyd Associates CAN Finance",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e82bbe96147",
                        "externalId": "external-id-27",
                        "name": "Tyler Boyd Associates BRA",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e92bbe96147",
                        "externalId": "external-id-28",
                        "name": "Tyler Boyd Associates URU",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e41bbe96147",
                        "externalId": "external-id-29",
                        "name": "Tyler Boyd Associates ARG",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e43bbe96147",
                        "externalId": "external-id-30",
                        "name": "Tyler Boyd Associates AUS",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }]
            },
        ]
    }]);
export const ServiceAgreementHttpServiceGetServiceAgreementAdminsMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/serviceagreements/{serviceAgreementId}/admins",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "01d70d09997b41cb83ea0e1db84626b3",
                        "externalId": "U0000011",
                        "legalEntityId": "a1198a20-5a40-4eb7-b47f-a553db958963",
                        "fullName": "David",
                        "legalEntityName": "LE-1"
                    }, {
                        "id": "bbf6a23536fa49029abf9c03f97a82fa",
                        "externalId": "U0000012",
                        "legalEntityId": "a1198a20-5a40-4eb7-b47f-a553db958963",
                        "fullName": "Paul",
                        "legalEntityName": "LE-1"
                    }, {
                        "id": "bc4ce9091eb04754a5a9b9422099f0fa",
                        "externalId": "U0000013",
                        "legalEntityId": "a1198a20-5a40-4eb7-b47f-a553db920429",
                        "fullName": "Wayne",
                        "legalEntityName": "LE-2"
                    }]
            },
        ]
    }]);
export const ServiceAgreementHttpServiceGetServiceAgreementContextMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/serviceagreements/context",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "a1198a20-5a40-4eb7-b47f-a553db920429",
                    "externalId": "external-id-1",
                    "name": "Broker deal 1",
                    "description": "Agreement between Backbase and Apple",
                    "creatorLegalEntity": "6e83bf0d-0d77-4320-b854-4a704e1d50c3",
                    "isMaster": true,
                    "status": "ENABLED",
                    "validFromDate": "2017-03-31",
                    "validFromTime": "08:30:00",
                    "validUntilDate": "2019-03-31",
                    "validUntilTime": "17:30:00"
                }
            },
        ]
    }]);
export const ServiceAgreementHttpServiceGetServiceAgreementItemMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/serviceagreements/{serviceAgreementId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "a1198a20-5a40-4eb7-b47f-a553db920429",
                    "externalId": "external-id-1",
                    "name": "Broker deal 1",
                    "description": "Agreement between Backbase and Apple",
                    "creatorLegalEntity": "6e83bf0d-0d77-4320-b854-4a704e1d50c3",
                    "isMaster": true,
                    "status": "ENABLED",
                    "validFromDate": "2017-03-31",
                    "validFromTime": "08:30:00",
                    "validUntilDate": "2019-03-31",
                    "validUntilTime": "17:30:00"
                }
            },
        ]
    }]);
export const ServiceAgreementHttpServiceGetServiceAgreementUsersMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/serviceagreements/context/users",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "01d70d09997b41cb83ea0e1db84626b3",
                        "externalId": "U0000011",
                        "legalEntityId": "a1198a20-5a40-4eb7-b47f-a553db958963",
                        "fullName": "David",
                        "legalEntityName": "LE-1"
                    }, {
                        "id": "bbf6a23536fa49029abf9c03f97a82fa",
                        "externalId": "U0000012",
                        "legalEntityId": "a1198a20-5a40-4eb7-b47f-a553db958963",
                        "fullName": "Paul",
                        "legalEntityName": "LE-1"
                    }, {
                        "id": "bc4ce9091eb04754a5a9b9422099f0fa",
                        "externalId": "U0000013",
                        "legalEntityId": "a1198a20-5a40-4eb7-b47f-a553db920429",
                        "fullName": "Wayne",
                        "legalEntityName": "LE-2"
                    }]
            },
        ]
    }]);
export const ServiceAgreementHttpServiceGetUnexposedUsersMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/serviceagreements/context/users/unexposed",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "01d70d09997b41cb83ea0e1db84626b3",
                        "externalId": "U0000011",
                        "fullName": "David",
                        "legalEntityId": "f63f3fc550794eed84bf848dbbba9bbd",
                        "legalEntityName": "LE-1"
                    }, {
                        "id": "bbf6a23536fa49029abf9c03f97a82fa",
                        "externalId": "U0000012",
                        "fullName": "Paul",
                        "legalEntityId": "f63f3fc550794eed84bf848dbbba9bbd",
                        "legalEntityName": "LE-1"
                    }, {
                        "id": "bc4ce9091eb04754a5a9b9422099f0fa",
                        "externalId": "U0000013",
                        "fullName": "Wayne",
                        "legalEntityId": "951293e6521940e3ae0608f93ff0b215",
                        "legalEntityName": "LE-2"
                    }]
            },
        ]
    }]);
export const ServiceAgreementHttpServicePostServiceAgreementMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/serviceagreements",
        method: "POST",
        responses: [
            {
                status: 202,
                body: {
                    "id": "0955e686d31e4216b3dd5d66161d536d"
                }
            },
            {
                status: 201,
                body: {
                    "id": "0955e686d31e4216b3dd5d66161d536d"
                }
            },
        ]
    }]);
export const ServiceAgreementHttpServicePostUsersAddMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/serviceagreements/context/users/add",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {}
            },
        ]
    }]);
export const ServiceAgreementHttpServicePostUsersRemoveMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/serviceagreements/context/users/remove",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {}
            },
        ]
    }]);
export const ServiceAgreementHttpServicePutAdminsMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/serviceagreements/{id}/admins",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {}
            },
        ]
    }]);
export const ServiceAgreementHttpServicePutServiceAgreementStateMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/serviceagreements/{serviceAgreementId}/state",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {}
            },
        ]
    }]);
export const ServiceAgreementHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/serviceagreements",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "951293e6521940e3ae0608f93ff0b215",
                        "externalId": "external-id-1",
                        "name": "Broker deal 1",
                        "description": "Agreement between Backbase and Apple",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96147",
                        "externalId": "external-id-2",
                        "name": "Broker deal 2",
                        "description": "Agreement between Backbase and HP",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae0a1e42bbe96147",
                        "externalId": "external-id-3",
                        "name": "Tyler Boyd Associates",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae1a1e42bbe96147",
                        "externalId": "external-id-4",
                        "name": "Tyler Boyd Associates UAE",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae2a1e42bbe96147",
                        "externalId": "external-id-5",
                        "name": "Tyler Boyd Associates US",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae3a1e42bbe96147",
                        "externalId": "external-id-6",
                        "name": "Tyler Boyd Associates US Salary Services",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae4a1e42bbe96147",
                        "externalId": "external-id-7",
                        "name": "Tyler Boyd Associates US Training",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae5a1e42bbe96147",
                        "externalId": "external-id-8",
                        "name": "Tyler Boyd Associates UK",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae6a1e42bbe96147",
                        "externalId": "external-id-9",
                        "name": "Tyler Boyd Associates UK Sales",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae7a1e42bbe96147",
                        "externalId": "external-id-10",
                        "name": "Tyler Boyd Associates UK Marketing",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae9a1e42bbe96147",
                        "externalId": "external-id-11",
                        "name": "Broker deal 11",
                        "description": "Agreement between Backbase and Microsoft",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a0e42bbe96147",
                        "externalId": "external-id-12",
                        "name": "Broker deal 12",
                        "description": "Agreement between Backbase and Packard-Bell",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a2e42bbe96147",
                        "externalId": "external-id-13",
                        "name": "Tyler Boyd Associates ROI",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a3e42bbe96147",
                        "externalId": "external-id-14",
                        "name": "Tyler Boyd Associates ROI Services",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a4e42bbe96147",
                        "externalId": "external-id-15",
                        "name": "Tyler Boyd Associates ROI Finance",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a5e42bbe96147",
                        "externalId": "external-id-16",
                        "name": "Tyler Boyd Associates ROI HR",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a6e42bbe96147",
                        "externalId": "external-id-17",
                        "name": "Tyler Boyd Associates EU",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a7e42bbe96147",
                        "externalId": "external-id-18",
                        "name": "Tyler Boyd Associates EU North",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a8e42bbe96147",
                        "externalId": "external-id-19",
                        "name": "Tyler Boyd Associates EU East",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a9e02bbe96147",
                        "externalId": "external-id-20",
                        "name": "Tyler Boyd Associates EU South",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e12bbe96147",
                        "externalId": "external-id-21",
                        "name": "Broker deal 21",
                        "description": "Agreement between Backbase and Google",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e22bbe96147",
                        "externalId": "external-id-22",
                        "name": "Broker deal 22",
                        "description": "Agreement between Backbase and Dell",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e32bbe96147",
                        "externalId": "external-id-23",
                        "name": "Tyler Boyd Associates EU West",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e52bbe96147",
                        "externalId": "external-id-24",
                        "name": "Tyler Boyd Associates EU Central",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e62bbe96147",
                        "externalId": "external-id-25",
                        "name": "Tyler Boyd Associates CAN",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e72bbe96147",
                        "externalId": "external-id-26",
                        "name": "Tyler Boyd Associates CAN Finance",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e82bbe96147",
                        "externalId": "external-id-27",
                        "name": "Tyler Boyd Associates BRA",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e92bbe96147",
                        "externalId": "external-id-28",
                        "name": "Tyler Boyd Associates URU",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e41bbe96147",
                        "externalId": "external-id-29",
                        "name": "Tyler Boyd Associates ARG",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e43bbe96147",
                        "externalId": "external-id-30",
                        "name": "Tyler Boyd Associates AUS",
                        "description": "Description of service agreement",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED"
                    }]
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/serviceagreements/{serviceAgreementId}/admins",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "01d70d09997b41cb83ea0e1db84626b3",
                        "externalId": "U0000011",
                        "legalEntityId": "a1198a20-5a40-4eb7-b47f-a553db958963",
                        "fullName": "David",
                        "legalEntityName": "LE-1"
                    }, {
                        "id": "bbf6a23536fa49029abf9c03f97a82fa",
                        "externalId": "U0000012",
                        "legalEntityId": "a1198a20-5a40-4eb7-b47f-a553db958963",
                        "fullName": "Paul",
                        "legalEntityName": "LE-1"
                    }, {
                        "id": "bc4ce9091eb04754a5a9b9422099f0fa",
                        "externalId": "U0000013",
                        "legalEntityId": "a1198a20-5a40-4eb7-b47f-a553db920429",
                        "fullName": "Wayne",
                        "legalEntityName": "LE-2"
                    }]
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/serviceagreements/context",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "a1198a20-5a40-4eb7-b47f-a553db920429",
                    "externalId": "external-id-1",
                    "name": "Broker deal 1",
                    "description": "Agreement between Backbase and Apple",
                    "creatorLegalEntity": "6e83bf0d-0d77-4320-b854-4a704e1d50c3",
                    "isMaster": true,
                    "status": "ENABLED",
                    "validFromDate": "2017-03-31",
                    "validFromTime": "08:30:00",
                    "validUntilDate": "2019-03-31",
                    "validUntilTime": "17:30:00"
                }
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/serviceagreements/{serviceAgreementId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "a1198a20-5a40-4eb7-b47f-a553db920429",
                    "externalId": "external-id-1",
                    "name": "Broker deal 1",
                    "description": "Agreement between Backbase and Apple",
                    "creatorLegalEntity": "6e83bf0d-0d77-4320-b854-4a704e1d50c3",
                    "isMaster": true,
                    "status": "ENABLED",
                    "validFromDate": "2017-03-31",
                    "validFromTime": "08:30:00",
                    "validUntilDate": "2019-03-31",
                    "validUntilTime": "17:30:00"
                }
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/serviceagreements/context/users",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "01d70d09997b41cb83ea0e1db84626b3",
                        "externalId": "U0000011",
                        "legalEntityId": "a1198a20-5a40-4eb7-b47f-a553db958963",
                        "fullName": "David",
                        "legalEntityName": "LE-1"
                    }, {
                        "id": "bbf6a23536fa49029abf9c03f97a82fa",
                        "externalId": "U0000012",
                        "legalEntityId": "a1198a20-5a40-4eb7-b47f-a553db958963",
                        "fullName": "Paul",
                        "legalEntityName": "LE-1"
                    }, {
                        "id": "bc4ce9091eb04754a5a9b9422099f0fa",
                        "externalId": "U0000013",
                        "legalEntityId": "a1198a20-5a40-4eb7-b47f-a553db920429",
                        "fullName": "Wayne",
                        "legalEntityName": "LE-2"
                    }]
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/serviceagreements/context/users/unexposed",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "01d70d09997b41cb83ea0e1db84626b3",
                        "externalId": "U0000011",
                        "fullName": "David",
                        "legalEntityId": "f63f3fc550794eed84bf848dbbba9bbd",
                        "legalEntityName": "LE-1"
                    }, {
                        "id": "bbf6a23536fa49029abf9c03f97a82fa",
                        "externalId": "U0000012",
                        "fullName": "Paul",
                        "legalEntityId": "f63f3fc550794eed84bf848dbbba9bbd",
                        "legalEntityName": "LE-1"
                    }, {
                        "id": "bc4ce9091eb04754a5a9b9422099f0fa",
                        "externalId": "U0000013",
                        "fullName": "Wayne",
                        "legalEntityId": "951293e6521940e3ae0608f93ff0b215",
                        "legalEntityName": "LE-2"
                    }]
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/serviceagreements",
        method: "POST",
        responses: [
            {
                status: 202,
                body: {
                    "id": "0955e686d31e4216b3dd5d66161d536d"
                }
            },
            {
                status: 201,
                body: {
                    "id": "0955e686d31e4216b3dd5d66161d536d"
                }
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/serviceagreements/context/users/add",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {}
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/serviceagreements/context/users/remove",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {}
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/serviceagreements/{id}/admins",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {}
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/serviceagreements/{serviceAgreementId}/state",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {}
            },
        ]
    },
]);
