// @backbase/accesscontrol-http-ang/esm2015/api/serviceAgreements.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const ServiceAgreementsHttpServiceGetAssignUsersPermissionsMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/service-agreements/{id}/users/{userId}/permissions",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "items": [{
                            "functionGroupId": "98f8531d-9bf4-4f3f-a49b-7cf331d6eead",
                            "dataGroupIds": [{
                                    "id": "138a7683-e9da-4ffa-a257-b78626790cd7"
                                }]
                        }, {
                            "functionGroupId": "00779897-253c-4bbb-b0c7-3c47fd0b0b08",
                            "dataGroupIds": []
                        }]
                }
            },
        ]
    }]);
export const ServiceAgreementsHttpServiceGetBusinessfunctionsMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/service-agreements/{id}/business-functions",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "functionId": "1026",
                        "functionCode": "us.billpay.enrolment",
                        "resource": "Billpay",
                        "name": "US Billpay Enrolment",
                        "privileges": [{
                                "privilege": "execute",
                                "supportsLimit": false
                            }, {
                                "privilege": "view",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1006",
                        "functionCode": "product.summary",
                        "resource": "Product Summary",
                        "name": "Product Summary",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1046",
                        "functionCode": "ach.debit",
                        "resource": "Payments",
                        "name": "ACH Debit",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": true
                            }, {
                                "privilege": "create",
                                "supportsLimit": true
                            }, {
                                "privilege": "edit",
                                "supportsLimit": true
                            }, {
                                "privilege": "delete",
                                "supportsLimit": true
                            }, {
                                "privilege": "approve",
                                "supportsLimit": true
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": true
                            }]
                    }, {
                        "functionId": "1013",
                        "functionCode": "audit",
                        "resource": "Audit",
                        "name": "Audit",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1043",
                        "functionCode": "sepa.credit.transfer.intc",
                        "resource": "Payments",
                        "name": "SEPA CT - Intracompany",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": true
                            }, {
                                "privilege": "create",
                                "supportsLimit": true
                            }, {
                                "privilege": "edit",
                                "supportsLimit": true
                            }, {
                                "privilege": "delete",
                                "supportsLimit": true
                            }, {
                                "privilege": "approve",
                                "supportsLimit": true
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": true
                            }]
                    }, {
                        "functionId": "1021",
                        "functionCode": "us.billpay.payments",
                        "resource": "Payments",
                        "name": "US Billpay Payments",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1011",
                        "functionCode": "manage.legalentities",
                        "resource": "Legal Entity",
                        "name": "Manage Legal Entities",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1032",
                        "functionCode": "assign.approval.policies",
                        "resource": "Approvals",
                        "name": "Assign Approval Policies",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1050",
                        "functionCode": "unlock.user",
                        "resource": "Identities",
                        "name": "Unlock User",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1049",
                        "functionCode": "lock.user",
                        "resource": "Identities",
                        "name": "Lock User",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1014",
                        "functionCode": "manage.shadow.limits",
                        "resource": "Limits",
                        "name": "Manage Shadow Limits",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1009",
                        "functionCode": "assign.permissions",
                        "resource": "Service Agreement",
                        "name": "Assign Permissions",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1037",
                        "functionCode": "batch.sepa",
                        "resource": "Batch",
                        "name": "Batch - SEPA CT",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1017",
                        "functionCode": "us.domestic.wire",
                        "resource": "Payments",
                        "name": "US Domestic Wire",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": true
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1010",
                        "functionCode": "manage.users",
                        "resource": "User",
                        "name": "Manage Users",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1022",
                        "functionCode": "us.billpay.payees",
                        "resource": "Contacts",
                        "name": "US Billpay Payees",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1019",
                        "functionCode": "manage.data.groups",
                        "resource": "Entitlements",
                        "name": "Manage Data Groups",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1015",
                        "functionCode": "intra.company.payments",
                        "resource": "Payments",
                        "name": "Intra Company Payments",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": true
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1044",
                        "functionCode": "us.domestic.wire.intc",
                        "resource": "Payments",
                        "name": "US Domestic Wire - Intracompany",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": true
                            }, {
                                "privilege": "create",
                                "supportsLimit": true
                            }, {
                                "privilege": "edit",
                                "supportsLimit": true
                            }, {
                                "privilege": "delete",
                                "supportsLimit": true
                            }, {
                                "privilege": "approve",
                                "supportsLimit": true
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": true
                            }]
                    }, {
                        "functionId": "1028",
                        "functionCode": "manage.service.agreements",
                        "resource": "Service Agreement",
                        "name": "Manage Service Agreements",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1025",
                        "functionCode": "us.billpay.payees.summary",
                        "resource": "Contacts",
                        "name": "US Billpay Payees-Summary",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1038",
                        "functionCode": "manage.messages",
                        "resource": "Message Center",
                        "name": "Manage Messages",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1045",
                        "functionCode": "us.foreign.wire.intc",
                        "resource": "Payments",
                        "name": "US Foreign Wire - Intracompany",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": true
                            }, {
                                "privilege": "create",
                                "supportsLimit": true
                            }, {
                                "privilege": "edit",
                                "supportsLimit": true
                            }, {
                                "privilege": "delete",
                                "supportsLimit": true
                            }, {
                                "privilege": "approve",
                                "supportsLimit": true
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": true
                            }]
                    }, {
                        "functionId": "1040",
                        "functionCode": "manage.default.limits",
                        "resource": "Limits",
                        "name": "Manage Default Limits",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1031",
                        "functionCode": "manage.topics",
                        "resource": "Message Center",
                        "name": "Manage Topics",
                        "privileges": [{
                                "privilege": "execute",
                                "supportsLimit": false
                            }, {
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1042",
                        "functionCode": "ach.credit.intc",
                        "resource": "Payments",
                        "name": "ACH Credit - Intracompany",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": true
                            }, {
                                "privilege": "create",
                                "supportsLimit": true
                            }, {
                                "privilege": "edit",
                                "supportsLimit": true
                            }, {
                                "privilege": "delete",
                                "supportsLimit": true
                            }, {
                                "privilege": "approve",
                                "supportsLimit": true
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": true
                            }]
                    }, {
                        "functionId": "1034",
                        "functionCode": "manage.identities",
                        "resource": "Identities",
                        "name": "Manage Identities",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1002",
                        "functionCode": "payments.sepa",
                        "resource": "Payments",
                        "name": "SEPA CT",
                        "privileges": [{
                                "privilege": "create",
                                "supportsLimit": true
                            }, {
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1023",
                        "functionCode": "us.billpay.accounts",
                        "resource": "Arrangements",
                        "name": "US Billpay Accounts",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1047",
                        "functionCode": "manage.budgets",
                        "resource": "Personal Finance Management",
                        "name": "Manage Budgets",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1033",
                        "functionCode": "manage.approval.policy.level",
                        "resource": "Approvals",
                        "name": "Manage Approval Policy and Level",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1035",
                        "functionCode": "manage.user.profiles",
                        "resource": "User Profiles",
                        "name": "Manage User Profiles",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1003",
                        "functionCode": "transactions",
                        "resource": "Transactions",
                        "name": "Transactions",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1039",
                        "functionCode": "supervise.messages",
                        "resource": "Message Center",
                        "name": "Supervise Messages",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1005",
                        "functionCode": "contacts",
                        "resource": "Contacts",
                        "name": "Contacts",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1012",
                        "functionCode": "manage.limits",
                        "resource": "Limits",
                        "name": "Manage Limits",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1041",
                        "functionCode": "ach.credit.transfer",
                        "resource": "Payments",
                        "name": "ACH Credit Transfer",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": true
                            }, {
                                "privilege": "create",
                                "supportsLimit": true
                            }, {
                                "privilege": "edit",
                                "supportsLimit": true
                            }, {
                                "privilege": "delete",
                                "supportsLimit": true
                            }, {
                                "privilege": "approve",
                                "supportsLimit": true
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": true
                            }]
                    }, {
                        "functionId": "1029",
                        "functionCode": "manage.actions.recipes",
                        "resource": "Actions",
                        "name": "Manage Action Recipes",
                        "privileges": [{
                                "privilege": "execute",
                                "supportsLimit": false
                            }, {
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1027",
                        "functionCode": "access.actions.history",
                        "resource": "Actions",
                        "name": "Access Actions History",
                        "privileges": [{
                                "privilege": "execute",
                                "supportsLimit": false
                            }, {
                                "privilege": "view",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1016",
                        "functionCode": "manage.statements",
                        "resource": "Account Statements",
                        "name": "Manage Statements",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1048",
                        "functionCode": "manage.saving.goals",
                        "resource": "Personal Finance Management",
                        "name": "Manage Saving Goals",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1030",
                        "functionCode": "manage.notifications",
                        "resource": "Notifications",
                        "name": "Manage Notifications",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1020",
                        "functionCode": "manage.function.groups",
                        "resource": "Entitlements",
                        "name": "Manage Function Groups",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1007",
                        "functionCode": "assign.users",
                        "resource": "Service Agreement",
                        "name": "Assign Users",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1018",
                        "functionCode": "us.foreign.wire",
                        "resource": "Payments",
                        "name": "US Foreign Wire",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": true
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1036",
                        "functionCode": "support.access.payments",
                        "resource": "Support Access",
                        "name": "Support Access for Payments",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1024",
                        "functionCode": "us.billpay.payees.search",
                        "resource": "Contacts",
                        "name": "US Billpay Payees-Search",
                        "privileges": [{
                                "privilege": "execute",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1051",
                        "functionCode": "manage.devices",
                        "resource": "Device",
                        "name": "Manage Devices",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1052",
                        "functionCode": "sepa.credit.transfer.closed",
                        "resource": "Payments",
                        "name": "SEPA CT - closed",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": true
                            }, {
                                "privilege": "create",
                                "supportsLimit": true
                            }, {
                                "privilege": "edit",
                                "supportsLimit": true
                            }, {
                                "privilege": "delete",
                                "supportsLimit": true
                            }, {
                                "privilege": "approve",
                                "supportsLimit": true
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": true
                            }]
                    }, {
                        "functionId": "1053",
                        "functionCode": "a2a.transfer",
                        "resource": "Payments",
                        "name": "A2A Transfer",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": true
                            }, {
                                "privilege": "create",
                                "supportsLimit": true
                            }, {
                                "privilege": "edit",
                                "supportsLimit": true
                            }, {
                                "privilege": "delete",
                                "supportsLimit": true
                            }, {
                                "privilege": "approve",
                                "supportsLimit": true
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": true
                            }]
                    }]
            },
        ]
    }]);
export const ServiceAgreementsHttpServiceGetContextServiceAgreementParticipantsMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/service-agreements/usercontext/participants",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "a1198a20-5a40-4eb7-b47f-a553db958963",
                        "externalId": "KPMG",
                        "name": "KPMG",
                        "sharingUsers": true,
                        "sharingAccounts": false
                    }, {
                        "id": "a1198a20-5a40-4eb7-b47f-a553db920429",
                        "externalId": "Backbase",
                        "name": "Backbase",
                        "sharingUsers": false,
                        "sharingAccounts": true
                    }]
            },
        ]
    }]);
export const ServiceAgreementsHttpServiceGetServiceAgreementParticipantsMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/service-agreements/{id}/participants",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "a1198a20-5a40-4eb7-b47f-a553db958963",
                        "externalId": "KPMG",
                        "name": "KPMG",
                        "sharingUsers": true,
                        "sharingAccounts": false
                    }, {
                        "id": "a1198a20-5a40-4eb7-b47f-a553db920429",
                        "externalId": "Backbase",
                        "name": "Backbase",
                        "sharingUsers": false,
                        "sharingAccounts": true
                    }]
            },
        ]
    }]);
export const ServiceAgreementsHttpServiceGetServiceAgreementUsersSAMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/service-agreements/{id}/users",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "01d70d09997b41cb83ea0e1db84626b3",
                        "externalId": "U0000011",
                        "legalEntityId": "a1198a20-5a40-4eb7-b47f-a553db958963",
                        "fullName": "David",
                        "legalEntityName": "LE-1"
                    }, {
                        "id": "bbf6a23536fa49029abf9c03f97a82fa",
                        "externalId": "U0000012",
                        "legalEntityId": "a1198a20-5a40-4eb7-b47f-a553db958963",
                        "fullName": "Paul",
                        "legalEntityName": "LE-1"
                    }, {
                        "id": "bc4ce9091eb04754a5a9b9422099f0fa",
                        "externalId": "U0000013",
                        "legalEntityId": "a1198a20-5a40-4eb7-b47f-a553db920429",
                        "fullName": "Wayne",
                        "legalEntityName": "LE-2"
                    }]
            },
        ]
    }]);
export const ServiceAgreementsHttpServiceGetServiceAgreementsMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/service-agreements/hierarchy",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "951293e6521940e3ae0608f93ff0b215",
                        "externalId": "external-id-1",
                        "name": "Broker deal 1",
                        "description": "Agreement between Backbase and Apple",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName1",
                        "numberOfParticipants": 3,
                        "validFromDate": "2017-03-31",
                        "validFromTime": "08:00:00",
                        "validUntilDate": "2019-03-31",
                        "validUntilTime": "08:00:00"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96147",
                        "externalId": "external-id-2",
                        "name": "Broker deal 2",
                        "description": "Agreement between Backbase and HP",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName",
                        "numberOfParticipants": 4,
                        "validFromDate": "2018-03-31",
                        "validFromTime": "08:00:00",
                        "validUntilDate": "2019-03-31",
                        "validUntilTime": "08:00:00"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96148",
                        "externalId": "external-id-3",
                        "name": "Broker deal 3",
                        "description": "Agreement between Backbase and Tyler Boyd Associates",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": true,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName4",
                        "numberOfParticipants": 1,
                        "validFromDate": "2018-10-31",
                        "validFromTime": "09:00:00"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96149",
                        "externalId": "external-id-4",
                        "name": "Broker deal 4",
                        "description": "Agreement between Backbase and Tyler Boyd Associates UAE",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName1",
                        "numberOfParticipants": 3,
                        "validUntilDate": "2020-06-29",
                        "validUntilTime": "20:00:00"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96150",
                        "externalId": "external-id-5",
                        "name": "Broker deal 5",
                        "description": "Agreement between Backbase and Tyler Boyd Associates US",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName",
                        "numberOfParticipants": 4,
                        "validFromDate": "2015-01-01",
                        "validFromTime": "08:00:00",
                        "validUntilDate": "2019-12-31",
                        "validUntilTime": "08:00:00"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96151",
                        "externalId": "external-id-6",
                        "name": "Broker deal 6",
                        "description": "Agreement between Backbase and Tyler Boyd Associates US Salary Services",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": true,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName4",
                        "numberOfParticipants": 1
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96152",
                        "externalId": "external-id-7",
                        "name": "Broker deal 7",
                        "description": "Agreement between Backbase and Tyler Boyd Associates US Training",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName1",
                        "numberOfParticipants": 3
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96153",
                        "externalId": "external-id-8",
                        "name": "Broker deal 8",
                        "description": "Agreement between Backbase and Tyler Boyd Associates UK",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName",
                        "numberOfParticipants": 4
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96154",
                        "externalId": "external-id-9",
                        "name": "Broker deal 9",
                        "description": "Agreement between Backbase and Tyler Boyd Associates UK Sales",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": true,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName4",
                        "numberOfParticipants": 1
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96155",
                        "externalId": "external-id-10",
                        "name": "Broker deal 10",
                        "description": "Agreement between Backbase and Tyler Boyd Associates UK Marketing",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName1",
                        "numberOfParticipants": 3
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96156",
                        "externalId": "external-id-11",
                        "name": "Broker deal 11",
                        "description": "Agreement between Backbase and Microsoft",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName1",
                        "numberOfParticipants": 3,
                        "validFromDate": "2017-03-01",
                        "validFromTime": "10:00:00"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96157",
                        "externalId": "external-id-12",
                        "name": "Broker deal 12",
                        "description": "Agreement between Backbase and Packard-Bell",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName",
                        "numberOfParticipants": 4
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96158",
                        "externalId": "external-id-13",
                        "name": "Broker deal 13",
                        "description": "Agreement between Backbase and Packard-Bell US",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": true,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName4",
                        "numberOfParticipants": 1
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96159",
                        "externalId": "external-id-14",
                        "name": "Broker deal 14",
                        "description": "Agreement between Backbase and Packard-Bell UAE",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName1",
                        "numberOfParticipants": 3
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96160",
                        "externalId": "external-id-15",
                        "name": "Broker deal 15",
                        "description": "Agreement between Backbase and Packard-Bell US Foreign Exchange",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName",
                        "numberOfParticipants": 4,
                        "validFromDate": "2017-03-31",
                        "validFromTime": "08:00:00",
                        "validUntilDate": "2019-03-31",
                        "validUntilTime": "08:00:00"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96161",
                        "externalId": "external-id-16",
                        "name": "Broker deal 16",
                        "description": "Agreement between Backbase and Packard-Bell US Salary Services",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": true,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName4",
                        "numberOfParticipants": 1
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96162",
                        "externalId": "external-id-17",
                        "name": "Broker deal 17",
                        "description": "Agreement between Backbase and Packard-Bell US Training",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName1",
                        "numberOfParticipants": 3
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96163",
                        "externalId": "external-id-18",
                        "name": "Broker deal 18",
                        "description": "Agreement between Backbase and Packard-Bell UK",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName",
                        "numberOfParticipants": 4
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96164",
                        "externalId": "external-id-19",
                        "name": "Broker deal 19",
                        "description": "Agreement between Backbase and Packard-Bell UK Sales",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": true,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName4",
                        "numberOfParticipants": 1
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96165",
                        "externalId": "external-id-20",
                        "name": "Broker deal 20",
                        "description": "Agreement between Backbase and Packard-Bell UK Marketing",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName1",
                        "numberOfParticipants": 3
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96166",
                        "externalId": "external-id-21",
                        "name": "Broker deal 21",
                        "description": "Agreement between Backbase and Jenkins",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName1",
                        "numberOfParticipants": 3
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96167",
                        "externalId": "external-id-22",
                        "name": "Broker deal 22",
                        "description": "Agreement between Backbase and Blue Ocean",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName",
                        "numberOfParticipants": 4
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96168",
                        "externalId": "external-id-23",
                        "name": "Broker deal 23",
                        "description": "Agreement between Backbase and Dell US",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": true,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName4",
                        "numberOfParticipants": 1
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96169",
                        "externalId": "external-id-24",
                        "name": "Broker deal 24",
                        "description": "Agreement between Backbase and Dell UAE",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName1",
                        "numberOfParticipants": 3
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96170",
                        "externalId": "external-id-25",
                        "name": "Broker deal 25",
                        "description": "Agreement between Backbase and Dell AUS",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName",
                        "numberOfParticipants": 4
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96171",
                        "externalId": "external-id-26",
                        "name": "Broker deal 26",
                        "description": "Agreement between Backbase and Dell US Salary Services",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": true,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName4",
                        "numberOfParticipants": 1
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96172",
                        "externalId": "external-id-27",
                        "name": "Broker deal 27",
                        "description": "Agreement between Backbase and Dell US Training",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName1",
                        "numberOfParticipants": 3
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96173",
                        "externalId": "external-id-28",
                        "name": "Broker deal 28",
                        "description": "Agreement between Backbase and Dell UK",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName",
                        "numberOfParticipants": 4
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96174",
                        "externalId": "external-id-29",
                        "name": "Broker deal 29",
                        "description": "Agreement between Backbase and Dell UK Sales",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": true,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName4",
                        "numberOfParticipants": 1
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96175",
                        "externalId": "external-id-30",
                        "name": "Broker deal 30",
                        "description": "Agreement between Backbase and Dell UK Marketing",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName1",
                        "numberOfParticipants": 3
                    }]
            },
        ]
    }]);
export const ServiceAgreementsHttpServiceGetUnexposedUsersSAMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/service-agreements/{id}/users/unexposed",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "01d70d09997b41cb83ea0e1db84626b3",
                        "externalId": "U0000011",
                        "fullName": "David",
                        "legalEntityId": "f63f3fc550794eed84bf848dbbba9bbd",
                        "legalEntityName": "LE-1"
                    }, {
                        "id": "bbf6a23536fa49029abf9c03f97a82fa",
                        "externalId": "U0000012",
                        "fullName": "Paul",
                        "legalEntityId": "f63f3fc550794eed84bf848dbbba9bbd",
                        "legalEntityName": "LE-1"
                    }, {
                        "id": "bc4ce9091eb04754a5a9b9422099f0fa",
                        "externalId": "U0000013",
                        "fullName": "Wayne",
                        "legalEntityId": "951293e6521940e3ae0608f93ff0b215",
                        "legalEntityName": "LE-2"
                    }]
            },
        ]
    }]);
export const ServiceAgreementsHttpServicePostUsersAddSAMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/service-agreements/{id}/users/add",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {}
            },
        ]
    }]);
export const ServiceAgreementsHttpServicePostUsersRemoveSAMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/service-agreements/{id}/users/remove",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {}
            },
        ]
    }]);
export const ServiceAgreementsHttpServicePutAssignUsersPermissionsMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/service-agreements/{id}/users/{userId}/permissions",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {
                    "approvalStatus": "PENDING"
                }
            },
        ]
    }]);
export const ServiceAgreementsHttpServicePutServiceAgreementSaveMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/service-agreements/{id}",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {}
            },
        ]
    }]);
export const ServiceAgreementsHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/service-agreements/{id}/users/{userId}/permissions",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "items": [{
                            "functionGroupId": "98f8531d-9bf4-4f3f-a49b-7cf331d6eead",
                            "dataGroupIds": [{
                                    "id": "138a7683-e9da-4ffa-a257-b78626790cd7"
                                }]
                        }, {
                            "functionGroupId": "00779897-253c-4bbb-b0c7-3c47fd0b0b08",
                            "dataGroupIds": []
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/service-agreements/{id}/business-functions",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "functionId": "1026",
                        "functionCode": "us.billpay.enrolment",
                        "resource": "Billpay",
                        "name": "US Billpay Enrolment",
                        "privileges": [{
                                "privilege": "execute",
                                "supportsLimit": false
                            }, {
                                "privilege": "view",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1006",
                        "functionCode": "product.summary",
                        "resource": "Product Summary",
                        "name": "Product Summary",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1046",
                        "functionCode": "ach.debit",
                        "resource": "Payments",
                        "name": "ACH Debit",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": true
                            }, {
                                "privilege": "create",
                                "supportsLimit": true
                            }, {
                                "privilege": "edit",
                                "supportsLimit": true
                            }, {
                                "privilege": "delete",
                                "supportsLimit": true
                            }, {
                                "privilege": "approve",
                                "supportsLimit": true
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": true
                            }]
                    }, {
                        "functionId": "1013",
                        "functionCode": "audit",
                        "resource": "Audit",
                        "name": "Audit",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1043",
                        "functionCode": "sepa.credit.transfer.intc",
                        "resource": "Payments",
                        "name": "SEPA CT - Intracompany",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": true
                            }, {
                                "privilege": "create",
                                "supportsLimit": true
                            }, {
                                "privilege": "edit",
                                "supportsLimit": true
                            }, {
                                "privilege": "delete",
                                "supportsLimit": true
                            }, {
                                "privilege": "approve",
                                "supportsLimit": true
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": true
                            }]
                    }, {
                        "functionId": "1021",
                        "functionCode": "us.billpay.payments",
                        "resource": "Payments",
                        "name": "US Billpay Payments",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1011",
                        "functionCode": "manage.legalentities",
                        "resource": "Legal Entity",
                        "name": "Manage Legal Entities",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1032",
                        "functionCode": "assign.approval.policies",
                        "resource": "Approvals",
                        "name": "Assign Approval Policies",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1050",
                        "functionCode": "unlock.user",
                        "resource": "Identities",
                        "name": "Unlock User",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1049",
                        "functionCode": "lock.user",
                        "resource": "Identities",
                        "name": "Lock User",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1014",
                        "functionCode": "manage.shadow.limits",
                        "resource": "Limits",
                        "name": "Manage Shadow Limits",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1009",
                        "functionCode": "assign.permissions",
                        "resource": "Service Agreement",
                        "name": "Assign Permissions",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1037",
                        "functionCode": "batch.sepa",
                        "resource": "Batch",
                        "name": "Batch - SEPA CT",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1017",
                        "functionCode": "us.domestic.wire",
                        "resource": "Payments",
                        "name": "US Domestic Wire",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": true
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1010",
                        "functionCode": "manage.users",
                        "resource": "User",
                        "name": "Manage Users",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1022",
                        "functionCode": "us.billpay.payees",
                        "resource": "Contacts",
                        "name": "US Billpay Payees",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1019",
                        "functionCode": "manage.data.groups",
                        "resource": "Entitlements",
                        "name": "Manage Data Groups",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1015",
                        "functionCode": "intra.company.payments",
                        "resource": "Payments",
                        "name": "Intra Company Payments",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": true
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1044",
                        "functionCode": "us.domestic.wire.intc",
                        "resource": "Payments",
                        "name": "US Domestic Wire - Intracompany",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": true
                            }, {
                                "privilege": "create",
                                "supportsLimit": true
                            }, {
                                "privilege": "edit",
                                "supportsLimit": true
                            }, {
                                "privilege": "delete",
                                "supportsLimit": true
                            }, {
                                "privilege": "approve",
                                "supportsLimit": true
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": true
                            }]
                    }, {
                        "functionId": "1028",
                        "functionCode": "manage.service.agreements",
                        "resource": "Service Agreement",
                        "name": "Manage Service Agreements",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1025",
                        "functionCode": "us.billpay.payees.summary",
                        "resource": "Contacts",
                        "name": "US Billpay Payees-Summary",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1038",
                        "functionCode": "manage.messages",
                        "resource": "Message Center",
                        "name": "Manage Messages",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1045",
                        "functionCode": "us.foreign.wire.intc",
                        "resource": "Payments",
                        "name": "US Foreign Wire - Intracompany",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": true
                            }, {
                                "privilege": "create",
                                "supportsLimit": true
                            }, {
                                "privilege": "edit",
                                "supportsLimit": true
                            }, {
                                "privilege": "delete",
                                "supportsLimit": true
                            }, {
                                "privilege": "approve",
                                "supportsLimit": true
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": true
                            }]
                    }, {
                        "functionId": "1040",
                        "functionCode": "manage.default.limits",
                        "resource": "Limits",
                        "name": "Manage Default Limits",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1031",
                        "functionCode": "manage.topics",
                        "resource": "Message Center",
                        "name": "Manage Topics",
                        "privileges": [{
                                "privilege": "execute",
                                "supportsLimit": false
                            }, {
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1042",
                        "functionCode": "ach.credit.intc",
                        "resource": "Payments",
                        "name": "ACH Credit - Intracompany",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": true
                            }, {
                                "privilege": "create",
                                "supportsLimit": true
                            }, {
                                "privilege": "edit",
                                "supportsLimit": true
                            }, {
                                "privilege": "delete",
                                "supportsLimit": true
                            }, {
                                "privilege": "approve",
                                "supportsLimit": true
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": true
                            }]
                    }, {
                        "functionId": "1034",
                        "functionCode": "manage.identities",
                        "resource": "Identities",
                        "name": "Manage Identities",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1002",
                        "functionCode": "payments.sepa",
                        "resource": "Payments",
                        "name": "SEPA CT",
                        "privileges": [{
                                "privilege": "create",
                                "supportsLimit": true
                            }, {
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1023",
                        "functionCode": "us.billpay.accounts",
                        "resource": "Arrangements",
                        "name": "US Billpay Accounts",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1047",
                        "functionCode": "manage.budgets",
                        "resource": "Personal Finance Management",
                        "name": "Manage Budgets",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1033",
                        "functionCode": "manage.approval.policy.level",
                        "resource": "Approvals",
                        "name": "Manage Approval Policy and Level",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1035",
                        "functionCode": "manage.user.profiles",
                        "resource": "User Profiles",
                        "name": "Manage User Profiles",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1003",
                        "functionCode": "transactions",
                        "resource": "Transactions",
                        "name": "Transactions",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1039",
                        "functionCode": "supervise.messages",
                        "resource": "Message Center",
                        "name": "Supervise Messages",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1005",
                        "functionCode": "contacts",
                        "resource": "Contacts",
                        "name": "Contacts",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1012",
                        "functionCode": "manage.limits",
                        "resource": "Limits",
                        "name": "Manage Limits",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1041",
                        "functionCode": "ach.credit.transfer",
                        "resource": "Payments",
                        "name": "ACH Credit Transfer",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": true
                            }, {
                                "privilege": "create",
                                "supportsLimit": true
                            }, {
                                "privilege": "edit",
                                "supportsLimit": true
                            }, {
                                "privilege": "delete",
                                "supportsLimit": true
                            }, {
                                "privilege": "approve",
                                "supportsLimit": true
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": true
                            }]
                    }, {
                        "functionId": "1029",
                        "functionCode": "manage.actions.recipes",
                        "resource": "Actions",
                        "name": "Manage Action Recipes",
                        "privileges": [{
                                "privilege": "execute",
                                "supportsLimit": false
                            }, {
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1027",
                        "functionCode": "access.actions.history",
                        "resource": "Actions",
                        "name": "Access Actions History",
                        "privileges": [{
                                "privilege": "execute",
                                "supportsLimit": false
                            }, {
                                "privilege": "view",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1016",
                        "functionCode": "manage.statements",
                        "resource": "Account Statements",
                        "name": "Manage Statements",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1048",
                        "functionCode": "manage.saving.goals",
                        "resource": "Personal Finance Management",
                        "name": "Manage Saving Goals",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1030",
                        "functionCode": "manage.notifications",
                        "resource": "Notifications",
                        "name": "Manage Notifications",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1020",
                        "functionCode": "manage.function.groups",
                        "resource": "Entitlements",
                        "name": "Manage Function Groups",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1007",
                        "functionCode": "assign.users",
                        "resource": "Service Agreement",
                        "name": "Assign Users",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1018",
                        "functionCode": "us.foreign.wire",
                        "resource": "Payments",
                        "name": "US Foreign Wire",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "create",
                                "supportsLimit": true
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }, {
                                "privilege": "delete",
                                "supportsLimit": false
                            }, {
                                "privilege": "approve",
                                "supportsLimit": false
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1036",
                        "functionCode": "support.access.payments",
                        "resource": "Support Access",
                        "name": "Support Access for Payments",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1024",
                        "functionCode": "us.billpay.payees.search",
                        "resource": "Contacts",
                        "name": "US Billpay Payees-Search",
                        "privileges": [{
                                "privilege": "execute",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1051",
                        "functionCode": "manage.devices",
                        "resource": "Device",
                        "name": "Manage Devices",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": false
                            }, {
                                "privilege": "edit",
                                "supportsLimit": false
                            }]
                    }, {
                        "functionId": "1052",
                        "functionCode": "sepa.credit.transfer.closed",
                        "resource": "Payments",
                        "name": "SEPA CT - closed",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": true
                            }, {
                                "privilege": "create",
                                "supportsLimit": true
                            }, {
                                "privilege": "edit",
                                "supportsLimit": true
                            }, {
                                "privilege": "delete",
                                "supportsLimit": true
                            }, {
                                "privilege": "approve",
                                "supportsLimit": true
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": true
                            }]
                    }, {
                        "functionId": "1053",
                        "functionCode": "a2a.transfer",
                        "resource": "Payments",
                        "name": "A2A Transfer",
                        "privileges": [{
                                "privilege": "view",
                                "supportsLimit": true
                            }, {
                                "privilege": "create",
                                "supportsLimit": true
                            }, {
                                "privilege": "edit",
                                "supportsLimit": true
                            }, {
                                "privilege": "delete",
                                "supportsLimit": true
                            }, {
                                "privilege": "approve",
                                "supportsLimit": true
                            }, {
                                "privilege": "cancel",
                                "supportsLimit": true
                            }]
                    }]
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/service-agreements/usercontext/participants",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "a1198a20-5a40-4eb7-b47f-a553db958963",
                        "externalId": "KPMG",
                        "name": "KPMG",
                        "sharingUsers": true,
                        "sharingAccounts": false
                    }, {
                        "id": "a1198a20-5a40-4eb7-b47f-a553db920429",
                        "externalId": "Backbase",
                        "name": "Backbase",
                        "sharingUsers": false,
                        "sharingAccounts": true
                    }]
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/service-agreements/{id}/participants",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "a1198a20-5a40-4eb7-b47f-a553db958963",
                        "externalId": "KPMG",
                        "name": "KPMG",
                        "sharingUsers": true,
                        "sharingAccounts": false
                    }, {
                        "id": "a1198a20-5a40-4eb7-b47f-a553db920429",
                        "externalId": "Backbase",
                        "name": "Backbase",
                        "sharingUsers": false,
                        "sharingAccounts": true
                    }]
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/service-agreements/{id}/users",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "01d70d09997b41cb83ea0e1db84626b3",
                        "externalId": "U0000011",
                        "legalEntityId": "a1198a20-5a40-4eb7-b47f-a553db958963",
                        "fullName": "David",
                        "legalEntityName": "LE-1"
                    }, {
                        "id": "bbf6a23536fa49029abf9c03f97a82fa",
                        "externalId": "U0000012",
                        "legalEntityId": "a1198a20-5a40-4eb7-b47f-a553db958963",
                        "fullName": "Paul",
                        "legalEntityName": "LE-1"
                    }, {
                        "id": "bc4ce9091eb04754a5a9b9422099f0fa",
                        "externalId": "U0000013",
                        "legalEntityId": "a1198a20-5a40-4eb7-b47f-a553db920429",
                        "fullName": "Wayne",
                        "legalEntityName": "LE-2"
                    }]
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/service-agreements/hierarchy",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "951293e6521940e3ae0608f93ff0b215",
                        "externalId": "external-id-1",
                        "name": "Broker deal 1",
                        "description": "Agreement between Backbase and Apple",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName1",
                        "numberOfParticipants": 3,
                        "validFromDate": "2017-03-31",
                        "validFromTime": "08:00:00",
                        "validUntilDate": "2019-03-31",
                        "validUntilTime": "08:00:00"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96147",
                        "externalId": "external-id-2",
                        "name": "Broker deal 2",
                        "description": "Agreement between Backbase and HP",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName",
                        "numberOfParticipants": 4,
                        "validFromDate": "2018-03-31",
                        "validFromTime": "08:00:00",
                        "validUntilDate": "2019-03-31",
                        "validUntilTime": "08:00:00"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96148",
                        "externalId": "external-id-3",
                        "name": "Broker deal 3",
                        "description": "Agreement between Backbase and Tyler Boyd Associates",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": true,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName4",
                        "numberOfParticipants": 1,
                        "validFromDate": "2018-10-31",
                        "validFromTime": "09:00:00"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96149",
                        "externalId": "external-id-4",
                        "name": "Broker deal 4",
                        "description": "Agreement between Backbase and Tyler Boyd Associates UAE",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName1",
                        "numberOfParticipants": 3,
                        "validUntilDate": "2020-06-29",
                        "validUntilTime": "20:00:00"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96150",
                        "externalId": "external-id-5",
                        "name": "Broker deal 5",
                        "description": "Agreement between Backbase and Tyler Boyd Associates US",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName",
                        "numberOfParticipants": 4,
                        "validFromDate": "2015-01-01",
                        "validFromTime": "08:00:00",
                        "validUntilDate": "2019-12-31",
                        "validUntilTime": "08:00:00"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96151",
                        "externalId": "external-id-6",
                        "name": "Broker deal 6",
                        "description": "Agreement between Backbase and Tyler Boyd Associates US Salary Services",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": true,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName4",
                        "numberOfParticipants": 1
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96152",
                        "externalId": "external-id-7",
                        "name": "Broker deal 7",
                        "description": "Agreement between Backbase and Tyler Boyd Associates US Training",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName1",
                        "numberOfParticipants": 3
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96153",
                        "externalId": "external-id-8",
                        "name": "Broker deal 8",
                        "description": "Agreement between Backbase and Tyler Boyd Associates UK",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName",
                        "numberOfParticipants": 4
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96154",
                        "externalId": "external-id-9",
                        "name": "Broker deal 9",
                        "description": "Agreement between Backbase and Tyler Boyd Associates UK Sales",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": true,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName4",
                        "numberOfParticipants": 1
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96155",
                        "externalId": "external-id-10",
                        "name": "Broker deal 10",
                        "description": "Agreement between Backbase and Tyler Boyd Associates UK Marketing",
                        "creatorLegalEntity": "0955e686-d31e-4216-b3dd-5d66161d536d",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName1",
                        "numberOfParticipants": 3
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96156",
                        "externalId": "external-id-11",
                        "name": "Broker deal 11",
                        "description": "Agreement between Backbase and Microsoft",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName1",
                        "numberOfParticipants": 3,
                        "validFromDate": "2017-03-01",
                        "validFromTime": "10:00:00"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96157",
                        "externalId": "external-id-12",
                        "name": "Broker deal 12",
                        "description": "Agreement between Backbase and Packard-Bell",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName",
                        "numberOfParticipants": 4
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96158",
                        "externalId": "external-id-13",
                        "name": "Broker deal 13",
                        "description": "Agreement between Backbase and Packard-Bell US",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": true,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName4",
                        "numberOfParticipants": 1
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96159",
                        "externalId": "external-id-14",
                        "name": "Broker deal 14",
                        "description": "Agreement between Backbase and Packard-Bell UAE",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName1",
                        "numberOfParticipants": 3
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96160",
                        "externalId": "external-id-15",
                        "name": "Broker deal 15",
                        "description": "Agreement between Backbase and Packard-Bell US Foreign Exchange",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName",
                        "numberOfParticipants": 4,
                        "validFromDate": "2017-03-31",
                        "validFromTime": "08:00:00",
                        "validUntilDate": "2019-03-31",
                        "validUntilTime": "08:00:00"
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96161",
                        "externalId": "external-id-16",
                        "name": "Broker deal 16",
                        "description": "Agreement between Backbase and Packard-Bell US Salary Services",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": true,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName4",
                        "numberOfParticipants": 1
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96162",
                        "externalId": "external-id-17",
                        "name": "Broker deal 17",
                        "description": "Agreement between Backbase and Packard-Bell US Training",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName1",
                        "numberOfParticipants": 3
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96163",
                        "externalId": "external-id-18",
                        "name": "Broker deal 18",
                        "description": "Agreement between Backbase and Packard-Bell UK",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName",
                        "numberOfParticipants": 4
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96164",
                        "externalId": "external-id-19",
                        "name": "Broker deal 19",
                        "description": "Agreement between Backbase and Packard-Bell UK Sales",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": true,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName4",
                        "numberOfParticipants": 1
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96165",
                        "externalId": "external-id-20",
                        "name": "Broker deal 20",
                        "description": "Agreement between Backbase and Packard-Bell UK Marketing",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName1",
                        "numberOfParticipants": 3
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96166",
                        "externalId": "external-id-21",
                        "name": "Broker deal 21",
                        "description": "Agreement between Backbase and Jenkins",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName1",
                        "numberOfParticipants": 3
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96167",
                        "externalId": "external-id-22",
                        "name": "Broker deal 22",
                        "description": "Agreement between Backbase and Blue Ocean",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName",
                        "numberOfParticipants": 4
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96168",
                        "externalId": "external-id-23",
                        "name": "Broker deal 23",
                        "description": "Agreement between Backbase and Dell US",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": true,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName4",
                        "numberOfParticipants": 1
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96169",
                        "externalId": "external-id-24",
                        "name": "Broker deal 24",
                        "description": "Agreement between Backbase and Dell UAE",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName1",
                        "numberOfParticipants": 3
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96170",
                        "externalId": "external-id-25",
                        "name": "Broker deal 25",
                        "description": "Agreement between Backbase and Dell AUS",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName",
                        "numberOfParticipants": 4
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96171",
                        "externalId": "external-id-26",
                        "name": "Broker deal 26",
                        "description": "Agreement between Backbase and Dell US Salary Services",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": true,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName4",
                        "numberOfParticipants": 1
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96172",
                        "externalId": "external-id-27",
                        "name": "Broker deal 27",
                        "description": "Agreement between Backbase and Dell US Training",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName1",
                        "numberOfParticipants": 3
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96173",
                        "externalId": "external-id-28",
                        "name": "Broker deal 28",
                        "description": "Agreement between Backbase and Dell UK",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName",
                        "numberOfParticipants": 4
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96174",
                        "externalId": "external-id-29",
                        "name": "Broker deal 29",
                        "description": "Agreement between Backbase and Dell UK Sales",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": true,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName4",
                        "numberOfParticipants": 1
                    }, {
                        "id": "9e4fb91794574b98ae8a1e42bbe96175",
                        "externalId": "external-id-30",
                        "name": "Broker deal 30",
                        "description": "Agreement between Backbase and Dell UK Marketing",
                        "creatorLegalEntity": "9371efe0-e295-47f7-822f-934532646b98",
                        "isMaster": false,
                        "status": "ENABLED",
                        "creatorLegalEntityName": "someName1",
                        "numberOfParticipants": 3
                    }]
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/service-agreements/{id}/users/unexposed",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "01d70d09997b41cb83ea0e1db84626b3",
                        "externalId": "U0000011",
                        "fullName": "David",
                        "legalEntityId": "f63f3fc550794eed84bf848dbbba9bbd",
                        "legalEntityName": "LE-1"
                    }, {
                        "id": "bbf6a23536fa49029abf9c03f97a82fa",
                        "externalId": "U0000012",
                        "fullName": "Paul",
                        "legalEntityId": "f63f3fc550794eed84bf848dbbba9bbd",
                        "legalEntityName": "LE-1"
                    }, {
                        "id": "bc4ce9091eb04754a5a9b9422099f0fa",
                        "externalId": "U0000013",
                        "fullName": "Wayne",
                        "legalEntityId": "951293e6521940e3ae0608f93ff0b215",
                        "legalEntityName": "LE-2"
                    }]
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/service-agreements/{id}/users/add",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {}
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/service-agreements/{id}/users/remove",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {}
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/service-agreements/{id}/users/{userId}/permissions",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {
                    "approvalStatus": "PENDING"
                }
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/service-agreements/{id}",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {}
            },
        ]
    },
]);
