// @backbase/accesscontrol-http-ang/esm2015/api/users.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const UsersHttpServiceGetArrangementUserPermissionCheckMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/users/user-privileges/arrangements/{id}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {}
            },
        ]
    }]);
export const UsersHttpServiceGetArrangementUserPrivilegesMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/users/user-privileges/arrangements",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "arrangementId": "9fa0961eb3d04dd3a5e48748d27e7275",
                        "privileges": [{
                                "privilege": "view"
                            }]
                    }, {
                        "arrangementId": "149f7e8d26904fc38bd0fdfc83394116",
                        "privileges": [{
                                "privilege": "view"
                            }]
                    }]
            },
        ]
    }]);
export const UsersHttpServiceGetCheckUserPermissionMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/users/user-permissions",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {}
            },
        ]
    }]);
export const UsersHttpServiceGetDataItemPermissionsContextMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/users/data-item-permissions",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "dataItem": {
                            "id": "9fa0961eb3d04dd3a5e48748d27e7275",
                            "dataType": "ARRANGEMENTS"
                        },
                        "permissions": [{
                                "resource": "Payments",
                                "businessFunction": "SEPA CT",
                                "privileges": ["view"]
                            }, {
                                "resource": "Payments",
                                "businessFunction": "Domestic Payments",
                                "privileges": ["view", "create"]
                            }]
                    }, {
                        "dataItem": {
                            "id": "9fa0961eb3d046d3a5e48748d27e7275",
                            "dataType": "ARRANGEMENTS"
                        },
                        "permissions": [{
                                "resource": "Payments",
                                "businessFunction": "Domestic Payments",
                                "privileges": ["approve"]
                            }]
                    }]
            },
        ]
    }]);
export const UsersHttpServiceGetUserPermissionsSummaryMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/users/permissions/summary",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "resource": "Contacts",
                        "function": "Contacts",
                        "permissions": {
                            "view": true,
                            "edit": true
                        }
                    }, {
                        "resource": "Payments",
                        "function": "US Domestic Wire",
                        "permissions": {
                            "create": true,
                            "view": true,
                            "approve": true
                        }
                    }, {
                        "resource": "Payments",
                        "function": "US Foreign Wire",
                        "permissions": {
                            "create": true,
                            "view": true,
                            "approve": true
                        }
                    }, {
                        "resource": "Payments",
                        "function": "SEPA CT",
                        "permissions": {
                            "create": true,
                            "view": true,
                            "approve": true
                        }
                    }, {
                        "resource": "Entitlements",
                        "function": "Manage Data Groups",
                        "permissions": {
                            "view": true,
                            "edit": true,
                            "create": true,
                            "delete": true
                        }
                    }, {
                        "resource": "Entitlements",
                        "function": "Manage Function Groups",
                        "permissions": {
                            "view": true,
                            "edit": true,
                            "create": true,
                            "delete": true
                        }
                    }, {
                        "resource": "Limits",
                        "function": "Manage Limits",
                        "permissions": {
                            "view": true,
                            "edit": true,
                            "create": true,
                            "delete": true
                        }
                    }, {
                        "resource": "Actions",
                        "function": "Manage Action Recipes",
                        "permissions": {
                            "execute": true,
                            "view": true,
                            "edit": true,
                            "create": true,
                            "delete": true
                        }
                    }, {
                        "resource": "User",
                        "function": "Manage Users",
                        "permissions": {
                            "execute": true,
                            "view": true,
                            "edit": true,
                            "create": true,
                            "delete": true
                        }
                    }, {
                        "resource": "Service Agreement",
                        "function": "Assign Users",
                        "permissions": {
                            "execute": true,
                            "view": true,
                            "edit": true,
                            "create": true,
                            "delete": true
                        }
                    }, {
                        "resource": "Service Agreement",
                        "function": "Assign Permissions",
                        "permissions": {
                            "execute": true,
                            "view": true,
                            "edit": true,
                            "create": true,
                            "delete": true,
                            "approve": true
                        }
                    }, {
                        "resource": "Service Agreement",
                        "function": "Manage Service Agreements",
                        "permissions": {
                            "execute": true,
                            "view": true,
                            "edit": true,
                            "create": true,
                            "delete": true
                        }
                    }, {
                        "resource": "Legal Entity",
                        "function": "Manage Legal Entities",
                        "permissions": {
                            "view": true
                        }
                    }, {
                        "resource": "Approvals",
                        "function": "Assign Approval Policies",
                        "permissions": {
                            "create": true,
                            "delete": true,
                            "edit": true,
                            "view": true
                        }
                    }]
            },
        ]
    }]);
export const UsersHttpServiceGetUserPrivilegesMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/users/user-privileges",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "privilege": "execute"
                    }, {
                        "privilege": "read"
                    }]
            },
        ]
    }]);
export const UsersHttpServiceGetUsersByPermissionsMocksProvider = createMocks([{
        urlPattern: "/access-control/client-api/v2/accessgroups/users/by-permissions",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "userIds": ["4e4def6f-d7dc-40ef-9031-3a62938a859c", "f7552b61-5f17-45d4-bdbe-fa3a12217bdd"]
                }
            },
        ]
    }]);
export const UsersHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/users/user-privileges/arrangements/{id}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {}
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/users/user-privileges/arrangements",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "arrangementId": "9fa0961eb3d04dd3a5e48748d27e7275",
                        "privileges": [{
                                "privilege": "view"
                            }]
                    }, {
                        "arrangementId": "149f7e8d26904fc38bd0fdfc83394116",
                        "privileges": [{
                                "privilege": "view"
                            }]
                    }]
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/users/user-permissions",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {}
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/users/data-item-permissions",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "dataItem": {
                            "id": "9fa0961eb3d04dd3a5e48748d27e7275",
                            "dataType": "ARRANGEMENTS"
                        },
                        "permissions": [{
                                "resource": "Payments",
                                "businessFunction": "SEPA CT",
                                "privileges": ["view"]
                            }, {
                                "resource": "Payments",
                                "businessFunction": "Domestic Payments",
                                "privileges": ["view", "create"]
                            }]
                    }, {
                        "dataItem": {
                            "id": "9fa0961eb3d046d3a5e48748d27e7275",
                            "dataType": "ARRANGEMENTS"
                        },
                        "permissions": [{
                                "resource": "Payments",
                                "businessFunction": "Domestic Payments",
                                "privileges": ["approve"]
                            }]
                    }]
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/users/permissions/summary",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "resource": "Contacts",
                        "function": "Contacts",
                        "permissions": {
                            "view": true,
                            "edit": true
                        }
                    }, {
                        "resource": "Payments",
                        "function": "US Domestic Wire",
                        "permissions": {
                            "create": true,
                            "view": true,
                            "approve": true
                        }
                    }, {
                        "resource": "Payments",
                        "function": "US Foreign Wire",
                        "permissions": {
                            "create": true,
                            "view": true,
                            "approve": true
                        }
                    }, {
                        "resource": "Payments",
                        "function": "SEPA CT",
                        "permissions": {
                            "create": true,
                            "view": true,
                            "approve": true
                        }
                    }, {
                        "resource": "Entitlements",
                        "function": "Manage Data Groups",
                        "permissions": {
                            "view": true,
                            "edit": true,
                            "create": true,
                            "delete": true
                        }
                    }, {
                        "resource": "Entitlements",
                        "function": "Manage Function Groups",
                        "permissions": {
                            "view": true,
                            "edit": true,
                            "create": true,
                            "delete": true
                        }
                    }, {
                        "resource": "Limits",
                        "function": "Manage Limits",
                        "permissions": {
                            "view": true,
                            "edit": true,
                            "create": true,
                            "delete": true
                        }
                    }, {
                        "resource": "Actions",
                        "function": "Manage Action Recipes",
                        "permissions": {
                            "execute": true,
                            "view": true,
                            "edit": true,
                            "create": true,
                            "delete": true
                        }
                    }, {
                        "resource": "User",
                        "function": "Manage Users",
                        "permissions": {
                            "execute": true,
                            "view": true,
                            "edit": true,
                            "create": true,
                            "delete": true
                        }
                    }, {
                        "resource": "Service Agreement",
                        "function": "Assign Users",
                        "permissions": {
                            "execute": true,
                            "view": true,
                            "edit": true,
                            "create": true,
                            "delete": true
                        }
                    }, {
                        "resource": "Service Agreement",
                        "function": "Assign Permissions",
                        "permissions": {
                            "execute": true,
                            "view": true,
                            "edit": true,
                            "create": true,
                            "delete": true,
                            "approve": true
                        }
                    }, {
                        "resource": "Service Agreement",
                        "function": "Manage Service Agreements",
                        "permissions": {
                            "execute": true,
                            "view": true,
                            "edit": true,
                            "create": true,
                            "delete": true
                        }
                    }, {
                        "resource": "Legal Entity",
                        "function": "Manage Legal Entities",
                        "permissions": {
                            "view": true
                        }
                    }, {
                        "resource": "Approvals",
                        "function": "Assign Approval Policies",
                        "permissions": {
                            "create": true,
                            "delete": true,
                            "edit": true,
                            "view": true
                        }
                    }]
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/users/user-privileges",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "privilege": "execute"
                    }, {
                        "privilege": "read"
                    }]
            },
        ]
    },
    {
        urlPattern: "/access-control/client-api/v2/accessgroups/users/by-permissions",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "userIds": ["4e4def6f-d7dc-40ef-9031-3a62938a859c", "f7552b61-5f17-45d4-bdbe-fa3a12217bdd"]
                }
            },
        ]
    },
]);
