// @backbase/accounts-transactions-journey-ang/esm2015/lib/views/accounts-manage.component.js
import { Component } from '@angular/core';
import { ActivatedRoute, Router } from '@angular/router';
import { AccountsTransactionsJourneyService } from '../accounts-transactions-journey-config.service';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "../accounts-transactions-journey-config.service";
import * as i3 from "@backbase/ui-ang/header";
import * as i4 from "@backbase/product-summary-manage-accounts-widget-ang";
export class AccountsManageComponent {
    constructor(router, route, config) {
        this.router = router;
        this.route = route;
        this.config = config;
        this.manageMyAccounts = $localize `:Manage my accounts heading|manage my accounts heading@@accounts-transactions-journey.accounts-manage.heading.manageMyAccounts:Manage my Accounts`;
        this.backToAccounts = $localize `:Back to my accounts button title|back to my accounts button title@@accounts-transactions-journey.accounts-manage.buttonTitle.backToAccounts:Back to my Accounts`;
    }
    onNavigateBack() {
        this.router.navigate(['..'], { relativeTo: this.route });
    }
}
/** @nocollapse */ AccountsManageComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountsManageComponent, deps: [{ token: i1.Router }, { token: i1.ActivatedRoute }, { token: i2.AccountsTransactionsJourneyService }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ AccountsManageComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AccountsManageComponent, selector: "bb-accounts-manage-list", ngImport: i0, template: `
    <div class="px-0 pt-4 pb-5 flex-column-reverse align-items-start">
      <button (click)="onNavigateBack()" class="bb-heading-widget__button btn btn-link pl-0 ml-0 mb-3">
        <i aria-hidden="true" class="bb-icon bb-icon-arrow-back"></i>
        <span>{{ backToAccounts }}</span>
      </button>
      <bb-header-ui
        headingType="h1"
        headingClasses="bb-heading-widget__heading"
        [heading]="manageMyAccounts"
      ></bb-header-ui>
    </div>

    <bb-product-summary-manage-accounts-widget
      [showIcons]="config.showAccountIcons"
    ></bb-product-summary-manage-accounts-widget>
  `, isInline: true, components: [{ type: i3.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i4.ProductSummaryManageAccountsWidgetComponent, selector: "bb-product-summary-manage-accounts-widget", inputs: ["businessFunction", "resourceName", "privilege", "size", "showIcons"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountsManageComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-accounts-manage-list',
                    template: `
    <div class="px-0 pt-4 pb-5 flex-column-reverse align-items-start">
      <button (click)="onNavigateBack()" class="bb-heading-widget__button btn btn-link pl-0 ml-0 mb-3">
        <i aria-hidden="true" class="bb-icon bb-icon-arrow-back"></i>
        <span>{{ backToAccounts }}</span>
      </button>
      <bb-header-ui
        headingType="h1"
        headingClasses="bb-heading-widget__heading"
        [heading]="manageMyAccounts"
      ></bb-header-ui>
    </div>

    <bb-product-summary-manage-accounts-widget
      [showIcons]="config.showAccountIcons"
    ></bb-product-summary-manage-accounts-widget>
  `,
                }]
        }], ctorParameters: function () { return [{ type: i1.Router }, { type: i1.ActivatedRoute }, { type: i2.AccountsTransactionsJourneyService }]; } });
