// @backbase/accounts-transactions-journey-ang/esm2015/lib/views/accounts-details.component.js
import { Component } from '@angular/core';
import { ProductSummaryAccountsService } from '@backbase/product-summary-common-ang';
import { AccountService } from '@backbase/product-summary-details-widget-ang';
import { AccountsTransactionsJourneyService } from '../accounts-transactions-journey-config.service';
import * as i0 from "@angular/core";
import * as i1 from "../accounts-transactions-journey-config.service";
import * as i2 from "@backbase/product-summary-details-widget-ang";
/**
 * A widget for displaying account details
 *
 * @usageNotes
 *
 * ### Display details of an account
 *
 * ```html
 * <bb-accounts-details></bb-accounts-details>
 * ```
 */
export class AccountsDetailsComponent {
    /**
     * @constructor
     * @param config
     */
    constructor(config) {
        this.config = config;
    }
}
/** @nocollapse */ AccountsDetailsComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountsDetailsComponent, deps: [{ token: i1.AccountsTransactionsJourneyService }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ AccountsDetailsComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AccountsDetailsComponent, selector: "bb-accounts-details", providers: [AccountService, ProductSummaryAccountsService], ngImport: i0, template: ` <bb-product-summary-details></bb-product-summary-details> `, isInline: true, components: [{ type: i2.ProductSummaryDetailsComponent, selector: "bb-product-summary-details" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountsDetailsComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-accounts-details',
                    template: ` <bb-product-summary-details></bb-product-summary-details> `,
                    providers: [AccountService, ProductSummaryAccountsService],
                }]
        }], ctorParameters: function () { return [{ type: i1.AccountsTransactionsJourneyService }]; } });
