// @backbase/product-summary-ui/esm2015/src/bb-product-summary-footer/product-summary-footer.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ProductSummaryFooterComponent } from './product-summary-footer.component';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { PaginationModule } from '@backbase/ui-ang/pagination';
import * as i0 from "@angular/core";
export class ProductSummaryFooterModule {
}
ProductSummaryFooterModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryFooterModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductSummaryFooterModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryFooterModule, declarations: [ProductSummaryFooterComponent], imports: [CommonModule, PaginationModule, LoadButtonModule], exports: [ProductSummaryFooterComponent] });
ProductSummaryFooterModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryFooterModule, imports: [[CommonModule, PaginationModule, LoadButtonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryFooterModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, PaginationModule, LoadButtonModule],
                    declarations: [ProductSummaryFooterComponent],
                    exports: [ProductSummaryFooterComponent],
                }]
        }] });
