// @backbase/product-summary-ui/esm2015/src/bb-product-summary-footer/product-summary-footer.component.js
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { PaginationType } from '@backbase/product-summary-common-ang';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/pagination";
import * as i2 from "@backbase/ui-ang/load-button";
import * as i3 from "@angular/common";
export class ProductSummaryFooterComponent {
    constructor() {
        /**
         * If `true`, the "First" and "Last" page links are shown.
         */
        this.boundaryLinks = true;
        /**
         * If `true`, the "Next" and "Previous" page links are shown.
         */
        this.directionLinks = false;
        /**
         * Event (Output) that emits when page prop is changed
         */
        this.pageChange = new EventEmitter();
        /**
         * Event (Output) that emits when load-more
         */
        this.loadMore = new EventEmitter();
        this.PaginationType = PaginationType;
    }
    onPageChange(page) {
        this.pageChange.emit(page);
    }
    onLoadMore() {
        if (!this.loading) {
            this.loadMore.emit();
        }
    }
}
ProductSummaryFooterComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryFooterComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ProductSummaryFooterComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ProductSummaryFooterComponent, selector: "bb-product-summary-footer-ui", inputs: { page: "page", count: "count", itemsPerPage: "itemsPerPage", paginationType: "paginationType", hasMore: "hasMore", loading: "loading", boundaryLinks: "boundaryLinks", directionLinks: "directionLinks", maxNavPages: "maxNavPages" }, outputs: { pageChange: "pageChange", loadMore: "loadMore" }, ngImport: i0, template: "<div class=\"bb-stack bb-stack--center\">\n  <bb-pagination-ui\n    *ngIf=\"paginationType === PaginationType.PAGINATION && (hasMore || (!!page && page > 0))\"\n    [boundaryLinks]=\"boundaryLinks\"\n    [directionLinks]=\"directionLinks\"\n    [itemsPerPage]=\"itemsPerPage\"\n    [maxNavPages]=\"maxNavPages\"\n    [page]=\"page\"\n    [totalItems]=\"count\"\n    (pageChange)=\"onPageChange($event)\">\n  </bb-pagination-ui>\n\n  <bb-load-button-ui\n    color=\"secondary\"\n    *ngIf=\"paginationType === PaginationType.LOAD_MORE && hasMore\"\n    [size]=\"page\"\n    [totalCount]=\"count\"\n    (paramChange)=\"onLoadMore()\"\n    [isLoading]=\"loading\">\n      <span i18n=\"Load more|Load more button@@accounts-overview-footer.pagination.button.load-more\">\n        Load more\n      </span>\n  </bb-load-button-ui>\n</div>\n", components: [{ type: i1.PaginatorComponent, selector: "bb-pagination-ui", inputs: ["page", "boundaryLinks", "directionLinks", "totalItems", "itemsPerPage", "maxNavPages"], outputs: ["pageChange"] }, { type: i2.LoadButtonComponent, selector: "bb-load-button-ui", inputs: ["isLoading", "from", "size", "circle", "totalCount"], outputs: ["paramChange"] }], directives: [{ type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryFooterComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-product-summary-footer-ui',
                    templateUrl: './product-summary-footer.component.html',
                }]
        }], propDecorators: { page: [{
                type: Input
            }], count: [{
                type: Input
            }], itemsPerPage: [{
                type: Input
            }], paginationType: [{
                type: Input
            }], hasMore: [{
                type: Input
            }], loading: [{
                type: Input
            }], boundaryLinks: [{
                type: Input
            }], directionLinks: [{
                type: Input
            }], maxNavPages: [{
                type: Input
            }], pageChange: [{
                type: Output
            }], loadMore: [{
                type: Output
            }] } });
