// @backbase/messages-common/esm2015/src/components/messages-mailout-status/messages-mailout-status.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { BadgeModule } from '@backbase/ui-ang/badge';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { MessagesMailoutStatusComponent, MessagesMailoutStatusCustomizableDirective, } from './messages-mailout-status.component';
import * as i0 from "@angular/core";
const components = [MessagesMailoutStatusComponent];
const directives = [MessagesMailoutStatusCustomizableDirective];
const uiModules = [BadgeModule, TooltipModule];
export class MessagesMailoutStatusModule {
}
MessagesMailoutStatusModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesMailoutStatusModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MessagesMailoutStatusModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesMailoutStatusModule, declarations: [MessagesMailoutStatusComponent, MessagesMailoutStatusCustomizableDirective], imports: [CommonModule, BadgeModule, TooltipModule], exports: [MessagesMailoutStatusComponent, MessagesMailoutStatusCustomizableDirective, BadgeModule, TooltipModule] });
MessagesMailoutStatusModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesMailoutStatusModule, imports: [[CommonModule, ...uiModules], BadgeModule, TooltipModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesMailoutStatusModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components, ...directives],
                    exports: [...components, ...directives, ...uiModules],
                    imports: [CommonModule, ...uiModules],
                }]
        }] });
