// @backbase/loans-collection-ui-ang/esm2015/lib/loan-info/loan-info-escrow-section/loan-info-escrow-section.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AmountModule } from '@backbase/ui-ang/amount';
import { LoanInfoEscrowSectionComponent } from './loan-info-escrow-section.component';
import { LoanInfoItemModule } from '../loan-info-item/loan-info-item.module';
import * as i0 from "@angular/core";
export class LoanInfoEscrowSectionModule {
}
/** @nocollapse */ LoanInfoEscrowSectionModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoEscrowSectionModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanInfoEscrowSectionModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoEscrowSectionModule, declarations: [LoanInfoEscrowSectionComponent], imports: [CommonModule, AmountModule, LoanInfoItemModule], exports: [LoanInfoEscrowSectionComponent] });
/** @nocollapse */ LoanInfoEscrowSectionModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoEscrowSectionModule, imports: [[CommonModule, AmountModule, LoanInfoItemModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoEscrowSectionModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanInfoEscrowSectionComponent],
                    imports: [CommonModule, AmountModule, LoanInfoItemModule],
                    exports: [LoanInfoEscrowSectionComponent],
                }]
        }] });
