// @backbase/ui-ang/esm2015/_example/ellipsis/showcase/ellipsis-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { EllipsisShowcaseComponent } from './ellipsis-showcase.component';
import { EllipsisModule } from '@backbase/ui-ang/ellipsis';
import * as i0 from "@angular/core";
export class EllipsisShowcaseModule {
}
EllipsisShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EllipsisShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
EllipsisShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EllipsisShowcaseModule, declarations: [EllipsisShowcaseComponent], imports: [CommonModule, EllipsisModule] });
EllipsisShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EllipsisShowcaseModule, imports: [[CommonModule, EllipsisModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EllipsisShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, EllipsisModule],
                    declarations: [EllipsisShowcaseComponent],
                    entryComponents: [EllipsisShowcaseComponent],
                }]
        }] });
