// @backbase/user-context-menu-widget-ang/esm2015/lib/user-context-menu-widget.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { NgbDropdownModule } from '@ng-bootstrap/ng-bootstrap';
import { AccessgroupUiModule } from '@backbase/accessgroup-ui';
import { SelectContextWidgetModule } from '@backbase/select-context-widget-ang';
import { AvatarModule } from '@backbase/ui-ang/avatar';
import { IconModule } from '@backbase/ui-ang/icon';
import { ModalModule } from '@backbase/ui-ang/modal';
import { ButtonModule } from '@backbase/ui-ang/button';
import { DropdownMenuModule } from '@backbase/ui-ang/dropdown-menu';
import { UserContextMenuInfoComponent, UserContextMenuInfoCustomizableDirective, } from './user-context-menu-widget/user-context-menu-info/user-context-menu-info.component';
import { UserContextMenuDropdownComponent, UserContextMenuDropdownCustomizableDirective, } from './user-context-menu-widget/user-context-menu-dropdown/user-context-menu-dropdown.component';
import { UserContextMenuContainerComponent, UserContextMenuContainerCustomizableDirective, } from './user-context-menu-widget/user-context-menu-container/user-context-menu-container.component';
import { UserContextMenuWidgetComponent } from './user-context-menu-widget.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const components = [
    UserContextMenuWidgetComponent,
    UserContextMenuContainerComponent,
    UserContextMenuDropdownComponent,
    UserContextMenuInfoComponent,
];
const directives = [
    UserContextMenuInfoCustomizableDirective,
    UserContextMenuDropdownCustomizableDirective,
    UserContextMenuContainerCustomizableDirective,
];
const uiModules = [
    AccessgroupUiModule,
    SelectContextWidgetModule,
    AvatarModule,
    IconModule,
    ModalModule,
    ButtonModule,
    NgbDropdownModule,
    DropdownMenuModule,
];
export class UserContextMenuWidgetModule {
}
/** @nocollapse */ UserContextMenuWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: UserContextMenuWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ UserContextMenuWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: UserContextMenuWidgetModule, declarations: [UserContextMenuWidgetComponent,
        UserContextMenuContainerComponent,
        UserContextMenuDropdownComponent,
        UserContextMenuInfoComponent, UserContextMenuInfoCustomizableDirective,
        UserContextMenuDropdownCustomizableDirective,
        UserContextMenuContainerCustomizableDirective], imports: [CommonModule, i1.BackbaseCoreModule, AccessgroupUiModule,
        SelectContextWidgetModule,
        AvatarModule,
        IconModule,
        ModalModule,
        ButtonModule,
        NgbDropdownModule,
        DropdownMenuModule], exports: [UserContextMenuWidgetComponent,
        UserContextMenuContainerComponent,
        UserContextMenuDropdownComponent,
        UserContextMenuInfoComponent, UserContextMenuInfoCustomizableDirective,
        UserContextMenuDropdownCustomizableDirective,
        UserContextMenuContainerCustomizableDirective, AccessgroupUiModule,
        SelectContextWidgetModule,
        AvatarModule,
        IconModule,
        ModalModule,
        ButtonModule,
        NgbDropdownModule,
        DropdownMenuModule] });
/** @nocollapse */ UserContextMenuWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: UserContextMenuWidgetModule, imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { UserContextMenuWidgetComponent },
            }),
            ...uiModules,
        ], AccessgroupUiModule,
        SelectContextWidgetModule,
        AvatarModule,
        IconModule,
        ModalModule,
        ButtonModule,
        NgbDropdownModule,
        DropdownMenuModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: UserContextMenuWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components, ...directives],
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { UserContextMenuWidgetComponent },
                        }),
                        ...uiModules,
                    ],
                    exports: [...components, ...directives, ...uiModules],
                }]
        }] });
