// @backbase/actions-product-notifications-settings-widget-ang/esm2015/src/components/actions-product-notifications-settings-widget/actions-product-notifications-settings-widget.component.js
import { Component, Input } from '@angular/core';
import { ActionsStoreModel } from '@backbase/actions-store-ang';
import { BehaviorSubject, Subject } from 'rxjs';
import { ActionsProductNotificationsSettingsRouterService } from '../../services/actions-product-notifications-settings-router.service';
import { ActionsProductNotificationsSettingsDataService } from '../../services/actions-product-notifications-settings-data.service';
import { ActionsProductNotificationsSettingsPropertiesService } from '../../services/actions-product-notifications-settings-properties.service';
import { secondsToMilliseconds } from '@backbase/actions-common-ang';
import { map } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "../../services/actions-product-notifications-settings-router.service";
import * as i2 from "../pages/actions-product-settings-page/actions-product-settings-page.component";
import * as i3 from "@angular/common";
export class ActionsProductNotificationsSettingsWidgetComponent {
    constructor(routerService) {
        this.routerService = routerService;
        this.specificationIDs$$ = new BehaviorSubject(undefined);
        this.specificationIDs$ = this.specificationIDs$$.asObservable();
        this.notificationDismissTime$$ = new BehaviorSubject(undefined);
        this.notificationDismissTime$ = this.notificationDismissTime$$
            .asObservable()
            .pipe(map(seconds => (seconds ? secondsToMilliseconds(seconds) : undefined)));
        this.destroy$ = new Subject();
    }
    /**
     * Input for selected account id.
     */
    set selectedAccountId(value) {
        this.routerService.selectedAccountId = value;
    }
    /**
     * Specification IDs value setter.
     * Configurable via model property.
     */
    set specificationIDs(value) {
        this.specificationIDs$$.next(value);
    }
    /**
     * Notification dismiss time value setter
     */
    set notificationDismissTime(value) {
        this.notificationDismissTime$$.next(value);
    }
    ngOnDestroy() {
        this.destroy$.next();
        this.destroy$.complete();
    }
}
ActionsProductNotificationsSettingsWidgetComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsProductNotificationsSettingsWidgetComponent, deps: [{ token: i1.ActionsProductNotificationsSettingsRouterService }], target: i0.ɵɵFactoryTarget.Component });
ActionsProductNotificationsSettingsWidgetComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.14", type: ActionsProductNotificationsSettingsWidgetComponent, selector: "bb-actions-product-notifications-settings-widget", inputs: { selectedAccountId: "selectedAccountId", specificationIDs: "specificationIDs", notificationDismissTime: "notificationDismissTime" }, providers: [
        ActionsStoreModel,
        ActionsProductNotificationsSettingsPropertiesService,
        ActionsProductNotificationsSettingsDataService,
        ActionsProductNotificationsSettingsRouterService,
    ], ngImport: i0, template: '<bb-actions-product-settings-page [specificationIDs]="specificationIDs$ | async" [notificationDismissTime]="notificationDismissTime$ | async"></bb-actions-product-settings-page>', isInline: true, components: [{ type: i2.ActionsProductSettingsPageComponent, selector: "bb-actions-product-settings-page", inputs: ["notificationDismissTime", "specificationIDs"] }], pipes: { "async": i3.AsyncPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsProductNotificationsSettingsWidgetComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-actions-product-notifications-settings-widget',
                    template: '<bb-actions-product-settings-page [specificationIDs]="specificationIDs$ | async" [notificationDismissTime]="notificationDismissTime$ | async"></bb-actions-product-settings-page>',
                    providers: [
                        ActionsStoreModel,
                        ActionsProductNotificationsSettingsPropertiesService,
                        ActionsProductNotificationsSettingsDataService,
                        ActionsProductNotificationsSettingsRouterService,
                    ],
                }]
        }], ctorParameters: function () { return [{ type: i1.ActionsProductNotificationsSettingsRouterService }]; }, propDecorators: { selectedAccountId: [{
                type: Input
            }], specificationIDs: [{
                type: Input
            }], notificationDismissTime: [{
                type: Input
            }] } });
