// @backbase/loans-collection-ui-ang/esm2015/lib/loan-summary/loan-summary-formula/loan-summary-formula.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CollapsibleCardModule } from '@backbase/ui-ang/collapsible-card';
import { IconModule } from '@backbase/ui-ang/icon';
import { MediaQueryModule } from '@backbase/ui-ang/media-query-lib';
import { LoanSummaryFormulaComponent } from './loan-summary-formula.component';
import { LoanSummaryAmountItemModule } from '../loan-summary-amount-item/loan-summary-amount-item.module';
import * as i0 from "@angular/core";
export class LoanSummaryFormulaModule {
}
/** @nocollapse */ LoanSummaryFormulaModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryFormulaModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanSummaryFormulaModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryFormulaModule, declarations: [LoanSummaryFormulaComponent], imports: [CommonModule, LoanSummaryAmountItemModule, CollapsibleCardModule, IconModule, MediaQueryModule], exports: [LoanSummaryFormulaComponent] });
/** @nocollapse */ LoanSummaryFormulaModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryFormulaModule, imports: [[CommonModule, LoanSummaryAmountItemModule, CollapsibleCardModule, IconModule, MediaQueryModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryFormulaModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanSummaryFormulaComponent],
                    imports: [CommonModule, LoanSummaryAmountItemModule, CollapsibleCardModule, IconModule, MediaQueryModule],
                    exports: [LoanSummaryFormulaComponent],
                }]
        }] });
