// @backbase/authorized-users-journey-ang/esm2015/src/authorized-users-create/authorized-users-create.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { AlertModule } from '@backbase/ui-ang/alert';
import { ButtonModule } from '@backbase/ui-ang/button';
import { CheckboxGroupModule } from '@backbase/ui-ang/checkbox-group';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { IconModule } from '@backbase/ui-ang/icon';
import { InputCheckboxModule } from '@backbase/ui-ang/input-checkbox';
import { InputEmailModule } from '@backbase/ui-ang/input-email';
import { InputPhoneModule } from '@backbase/ui-ang/input-phone';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { ModalModule } from '@backbase/ui-ang/modal';
import { PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import { StepperModule } from '@backbase/ui-ang/stepper';
import { SwitchModule } from '@backbase/ui-ang/switch';
import { AuthorizedUsersCommonModule } from '../common/authorized-users-common.module';
import { AuthorizedUsersCreateWrapperComponent } from './authorized-users-create-wrapper.component';
import { AuthorizedUsersAccountsComponent } from './components/authorized-users-accounts/authorized-users-accounts.component';
import { AuthorizedUsersCancelCreationComponent } from './components/authorized-users-cancel-creation/authorized-users-cancel-creation.component';
import { AuthorizedUsersConfirmationCompleteComponent } from './components/authorized-users-confirmation-complete/authorized-users-confirmation-complete.component';
import { AuthorizedUsersConfirmationComponent } from './components/authorized-users-confirmation/authorized-users-confirmation.component';
import { AuthorizedUserStepHandlerDirective } from './components/authorized-users-create-common/authorized-users-step-handler.directive';
import { AuthorizedUsersCreateContainerComponent } from './components/authorized-users-create-container/authorized-users-create-container.component';
import { AuthorizedUsersPermissionsCardComponent } from './components/authorized-users-permissions/authorized-users-permissions-card.component';
import { AuthorizedUsersPermissionsComponent } from './components/authorized-users-permissions/authorized-users-permissions.component';
import { AuthorizedUserEditStepperComponent } from './components/authorized-users-stepper/authorized-users-edit-stepper.component';
import { AuthorizedUsersStepperComponent } from './components/authorized-users-stepper/authorized-users-stepper.component';
import { AuthorizedUsersUserDetailsComponent } from './components/authorized-users-user-details/authorized-users-user-details.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const modules = [
    StepperModule,
    ButtonModule,
    IconModule,
    SwitchModule,
    CheckboxGroupModule,
    InputCheckboxModule,
    PaymentCardNumberModule,
    InputPhoneModule,
    InputEmailModule,
    InputTextModule,
    ReactiveFormsModule,
    FormsModule,
    InputValidationMessageModule,
    InputCheckboxModule,
    ModalModule,
    AlertModule,
    LoadingIndicatorModule,
    LoadButtonModule,
    EmptyStateModule,
    AuthorizedUsersCommonModule,
];
const components = [
    AuthorizedUsersCreateWrapperComponent,
    AuthorizedUsersStepperComponent,
    AuthorizedUsersConfirmationComponent,
    AuthorizedUsersAccountsComponent,
    AuthorizedUsersCreateContainerComponent,
    AuthorizedUsersPermissionsComponent,
    AuthorizedUsersUserDetailsComponent,
    AuthorizedUsersPermissionsCardComponent,
    AuthorizedUsersCancelCreationComponent,
    AuthorizedUserEditStepperComponent,
    AuthorizedUserStepHandlerDirective,
    AuthorizedUsersConfirmationCompleteComponent,
];
export class AuthorizedUsersCreateModule {
}
AuthorizedUsersCreateModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersCreateModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AuthorizedUsersCreateModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersCreateModule, declarations: [AuthorizedUsersCreateWrapperComponent,
        AuthorizedUsersStepperComponent,
        AuthorizedUsersConfirmationComponent,
        AuthorizedUsersAccountsComponent,
        AuthorizedUsersCreateContainerComponent,
        AuthorizedUsersPermissionsComponent,
        AuthorizedUsersUserDetailsComponent,
        AuthorizedUsersPermissionsCardComponent,
        AuthorizedUsersCancelCreationComponent,
        AuthorizedUserEditStepperComponent,
        AuthorizedUserStepHandlerDirective,
        AuthorizedUsersConfirmationCompleteComponent], imports: [CommonModule,
        FormsModule,
        ReactiveFormsModule, StepperModule,
        ButtonModule,
        IconModule,
        SwitchModule,
        CheckboxGroupModule,
        InputCheckboxModule,
        PaymentCardNumberModule,
        InputPhoneModule,
        InputEmailModule,
        InputTextModule,
        ReactiveFormsModule,
        FormsModule,
        InputValidationMessageModule,
        InputCheckboxModule,
        ModalModule,
        AlertModule,
        LoadingIndicatorModule,
        LoadButtonModule,
        EmptyStateModule,
        AuthorizedUsersCommonModule, i1.BackbaseCoreModule], exports: [AuthorizedUsersCreateWrapperComponent,
        AuthorizedUsersStepperComponent,
        AuthorizedUsersConfirmationComponent,
        AuthorizedUsersAccountsComponent,
        AuthorizedUsersCreateContainerComponent,
        AuthorizedUsersPermissionsComponent,
        AuthorizedUsersUserDetailsComponent,
        AuthorizedUsersPermissionsCardComponent,
        AuthorizedUsersCancelCreationComponent,
        AuthorizedUserEditStepperComponent,
        AuthorizedUserStepHandlerDirective,
        AuthorizedUsersConfirmationCompleteComponent, StepperModule,
        ButtonModule,
        IconModule,
        SwitchModule,
        CheckboxGroupModule,
        InputCheckboxModule,
        PaymentCardNumberModule,
        InputPhoneModule,
        InputEmailModule,
        InputTextModule,
        ReactiveFormsModule,
        FormsModule,
        InputValidationMessageModule,
        InputCheckboxModule,
        ModalModule,
        AlertModule,
        LoadingIndicatorModule,
        LoadButtonModule,
        EmptyStateModule,
        AuthorizedUsersCommonModule] });
AuthorizedUsersCreateModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersCreateModule, imports: [[
            CommonModule,
            FormsModule,
            ReactiveFormsModule,
            ...modules,
            BackbaseCoreModule.withConfig({
                classMap: { AuthorizedUsersCreateWrapperComponent },
            }),
        ], StepperModule,
        ButtonModule,
        IconModule,
        SwitchModule,
        CheckboxGroupModule,
        InputCheckboxModule,
        PaymentCardNumberModule,
        InputPhoneModule,
        InputEmailModule,
        InputTextModule,
        ReactiveFormsModule,
        FormsModule,
        InputValidationMessageModule,
        InputCheckboxModule,
        ModalModule,
        AlertModule,
        LoadingIndicatorModule,
        LoadButtonModule,
        EmptyStateModule,
        AuthorizedUsersCommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersCreateModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components],
                    imports: [
                        CommonModule,
                        FormsModule,
                        ReactiveFormsModule,
                        ...modules,
                        BackbaseCoreModule.withConfig({
                            classMap: { AuthorizedUsersCreateWrapperComponent },
                        }),
                    ],
                    exports: [...components, ...modules],
                }]
        }] });
