// @backbase/product-summary-ui/esm2015/src/bb-product-number/product-number.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ProductNumberPipeModule } from '@backbase/product-summary-common-ang';
import { ButtonModule } from '@backbase/ui-ang/button';
import { IconModule } from '@backbase/ui-ang/icon';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { ProductNumberComponent, ProductNumberCustomizableDirective } from './product-number.component';
import * as i0 from "@angular/core";
const sharedPipes = [ProductNumberPipeModule];
const components = [ProductNumberComponent];
const directives = [ProductNumberCustomizableDirective];
const uiModules = [ButtonModule, IconModule, TooltipModule];
export class ProductNumberModule {
}
ProductNumberModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductNumberModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductNumberModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductNumberModule, declarations: [ProductNumberComponent, ProductNumberCustomizableDirective], imports: [CommonModule, ProductNumberPipeModule, ButtonModule, IconModule, TooltipModule], exports: [ProductNumberPipeModule, ProductNumberComponent, ProductNumberCustomizableDirective] });
ProductNumberModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductNumberModule, imports: [[CommonModule, ...sharedPipes, ...uiModules], ProductNumberPipeModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductNumberModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...sharedPipes, ...uiModules],
                    declarations: [...components, ...directives],
                    exports: [...sharedPipes, ...components, ...directives],
                }]
        }] });
