// @backbase/manage-bill-payments-journey-ang/esm2015/src/components/billpay-payments-group/billpay-payments-group.component.js
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../../services/manage-bill-payments-navigation.service";
import * as i2 from "../billpay-payment-card/billpay-payment-card.component";
import * as i3 from "@angular/common";
import * as i4 from "@backbase/ui-ang/date-pipe";
export var DateLabel;
(function (DateLabel) {
    DateLabel["yesterday"] = "yesterday";
    DateLabel["today"] = "today";
    DateLabel["tomorrow"] = "tomorrow";
})(DateLabel || (DateLabel = {}));
export class BillpayPaymentsGroupComponent {
    /** @internal */
    constructor(router) {
        this.router = router;
    }
    getDateLabel(date) {
        const day = 1000 * 60 * 60 * 24;
        const today = new Date().setHours(0, 0, 0, 0);
        const targetDay = new Date(date).setHours(0, 0, 0, 0);
        const daysDiff = Math.round((targetDay - today) / day);
        switch (daysDiff) {
            case -1:
                return DateLabel.yesterday;
            case 0:
                return DateLabel.today;
            case 1:
                return DateLabel.tomorrow;
            default:
                return;
        }
    }
    navigateToPayment(id, isRecurring) {
        this.router.navigateToPayment(id, isRecurring);
    }
}
BillpayPaymentsGroupComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayPaymentsGroupComponent, deps: [{ token: i1.ManageBillPaymentsNavigationService }], target: i0.ɵɵFactoryTarget.Component });
BillpayPaymentsGroupComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: BillpayPaymentsGroupComponent, selector: "bb-billpay-payments-group", inputs: { paymentsGroup: "paymentsGroup" }, ngImport: i0, template: "<section>\n  <div role=\"heading\" aria-level=\"2\" attr.id=\"payments-group-heading-{{paymentsGroup?.payments[0]?.id}}\" data-role=\"date-label\" class=\"bb-text-support bb-subheader bb-block bb-block--lg\">\n    <ng-container\n      *ngIf=\"getDateLabel(paymentsGroup?.date) as date; else formattedDate\"\n      i18n=\"Date label for payments group@@billpay-payments.payments-group.date-label\"\n    >\n      { date, select, yesterday {Yesterday} today {Today} tomorrow {Tomorrow} }\n    </ng-container>\n    <ng-template #formattedDate>{{ paymentsGroup?.date | bbDate: 'MMMM d' }}</ng-template>\n  </div>\n  <ul class=\"bb-list\" attr.aria-labelledby=\"payments-group-heading-{{paymentsGroup?.payments[0]?.id}}\">\n    <li\n      class=\"bb-list__item bb-list__item--no-separator\"\n      *ngFor=\"let payment of paymentsGroup?.payments\"\n    >\n      <bb-billpay-payment-card\n        tabindex=\"0\"\n        role=\"button\"\n        [payment]=\"payment\"\n        isNegativeAmount=\"true\"\n        (click)=\"navigateToPayment(payment.id, payment.recurring)\"\n        (keydown.enter)=\"navigateToPayment(payment.id, payment.recurring)\"\n      ></bb-billpay-payment-card>\n    </li>\n  </ul>\n</section>", components: [{ type: i2.BillpayPaymentCardComponent, selector: "bb-billpay-payment-card", inputs: ["payment", "isNegativeAmount"] }], directives: [{ type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i3.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }], pipes: { "bbDate": i4.BbDatePipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayPaymentsGroupComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-billpay-payments-group',
                    templateUrl: './billpay-payments-group.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.ManageBillPaymentsNavigationService }]; }, propDecorators: { paymentsGroup: [{
                type: Input
            }] } });
