// @backbase/positive-pay-ach-v1-client-ang/bundles/backbase-positive-pay-ach-v1-client-ang.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common/http'), require('@backbase/foundation-ang/data-http')) :
    typeof define === 'function' && define.amd ? define('@backbase/positive-pay-ach-v1-client-ang', ['exports', '@angular/core', '@angular/common/http', '@backbase/foundation-ang/data-http'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["positive-pay-ach-v1-client-ang"] = {}), global.ng.core, global.ng.common.http, global.i2));
})(this, (function (exports, i0, i1, i2) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);

    var ExceptionsFilterOrderByEnum = {
        TransactionDate: 'transactionDate'
    };

    var RuleTypeEnum = {
        BLOCKER: 'BLOCKER',
        FILTER: 'FILTER'
    };

    var RulesFilterOrderByEnum = {
        RegistrationDate: 'registrationDate',
        RuleType: 'ruleType',
        StartDate: 'startDate'
    };

    var SortDirectionEnum = {
        ASC: 'ASC',
        DESC: 'DESC'
    };

    var POSITIVE_PAY_ACH_BASE_PATH = new i0.InjectionToken('POSITIVE_PAY_ACH_BASE_PATH');
    var COLLECTION_FORMATS = {
        'csv': ',',
        'tsv': '   ',
        'ssv': ' ',
        'pipes': '|'
    };

    var PositivePayAchConfiguration = /** @class */ (function () {
        function PositivePayAchConfiguration(configurationParameters) {
            if (configurationParameters === void 0) { configurationParameters = {}; }
            this.apiKeys = configurationParameters.apiKeys;
            this.username = configurationParameters.username;
            this.password = configurationParameters.password;
            this.accessToken = configurationParameters.accessToken;
            this.basePath = configurationParameters.basePath;
            this.withCredentials = configurationParameters.withCredentials;
            this.encoder = configurationParameters.encoder;
            if (configurationParameters.credentials) {
                this.credentials = configurationParameters.credentials;
            }
            else {
                this.credentials = {};
            }
        }
        PositivePayAchConfiguration.prototype.selectHeaderContentType = function (contentTypes) {
            var _this = this;
            if (contentTypes.length === 0) {
                return undefined;
            }
            var type = contentTypes.find(function (x) { return _this.isJsonMime(x); });
            if (type === undefined) {
                return contentTypes[0];
            }
            return type;
        };
        PositivePayAchConfiguration.prototype.selectHeaderAccept = function (accepts) {
            var _this = this;
            if (accepts.length === 0) {
                return undefined;
            }
            var type = accepts.find(function (x) { return _this.isJsonMime(x); });
            if (type === undefined) {
                return accepts[0];
            }
            return type;
        };
        PositivePayAchConfiguration.prototype.isJsonMime = function (mime) {
            var jsonMime = new RegExp('^(application\/json|[^;/ \t]+\/[^;/ \t]+[+]json)[ \t]*(;.*)?$', 'i');
            return mime !== null && (jsonMime.test(mime) || mime.toLowerCase() === 'application/json-patch+json');
        };
        PositivePayAchConfiguration.prototype.lookupCredential = function (key) {
            var value = this.credentials[key];
            return typeof value === 'function'
                ? value()
                : value;
        };
        return PositivePayAchConfiguration;
    }());
    var CONFIG_TOKEN = new i0.InjectionToken('api-data-module PositivePayAchConfiguration');

    var PositivePayAchApiModule = /** @class */ (function () {
        function PositivePayAchApiModule(parentModule, http, dataModulesManager, config) {
            if (parentModule) {
                throw new Error('PositivePayAchApiModule is already loaded. Import in your base AppModule only.');
            }
            if (!http) {
                throw new Error('You need to import the HttpClientModule in your AppModule! \n' +
                    'See also https://github.com/angular/angular/issues/20575');
            }
            if (dataModulesManager) {
                dataModulesManager.setModuleConfig(CONFIG_TOKEN, {
                    apiRoot: '',
                    servicePath: config.basePath || '',
                    headers: {},
                });
            }
        }
        PositivePayAchApiModule.forRoot = function (configurationFactory) {
            return {
                ngModule: PositivePayAchApiModule,
                providers: [{ provide: PositivePayAchConfiguration, useFactory: configurationFactory }]
            };
        };
        return PositivePayAchApiModule;
    }());
    PositivePayAchApiModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: PositivePayAchApiModule, deps: [{ token: PositivePayAchApiModule, optional: true, skipSelf: true }, { token: i1__namespace.HttpClient, optional: true }, { token: i2__namespace.DataModulesManager, optional: true }, { token: PositivePayAchConfiguration }], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    PositivePayAchApiModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: PositivePayAchApiModule });
    PositivePayAchApiModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: PositivePayAchApiModule, providers: [], imports: [[]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: PositivePayAchApiModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [],
                        declarations: [],
                        exports: [],
                        providers: []
                    }]
            }], ctorParameters: function () {
            return [{ type: PositivePayAchApiModule, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.SkipSelf
                        }] }, { type: i1__namespace.HttpClient, decorators: [{
                            type: i0.Optional
                        }] }, { type: i2__namespace.DataModulesManager, decorators: [{
                            type: i0.Optional
                        }] }, { type: PositivePayAchConfiguration }];
        } });

    var CustomHttpParameterCodec = /** @class */ (function () {
        function CustomHttpParameterCodec() {
        }
        CustomHttpParameterCodec.prototype.encodeKey = function (k) {
            return encodeURIComponent(k);
        };
        CustomHttpParameterCodec.prototype.encodeValue = function (v) {
            return encodeURIComponent(v);
        };
        CustomHttpParameterCodec.prototype.decodeKey = function (k) {
            return decodeURIComponent(k);
        };
        CustomHttpParameterCodec.prototype.decodeValue = function (v) {
            return decodeURIComponent(v);
        };
        return CustomHttpParameterCodec;
    }());

    var ExceptionsHttpService = /** @class */ (function () {
        function ExceptionsHttpService(httpClient, basePath, configuration) {
            this.httpClient = httpClient;
            this.basePath = '';
            this.defaultHeaders = new i1.HttpHeaders();
            this.configuration = new PositivePayAchConfiguration();
            if (configuration) {
                this.configuration = configuration;
            }
            if (typeof this.configuration.basePath !== 'string') {
                if (typeof basePath !== 'string') {
                    basePath = this.basePath;
                }
                this.configuration.basePath = basePath;
            }
            this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
        }
        ExceptionsHttpService.prototype.addToHttpParams = function (httpParams, value, key) {
            if (typeof value === "object" && value instanceof Date === false) {
                httpParams = this.addToHttpParamsRecursive(httpParams, value);
            }
            else {
                httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
            }
            return httpParams;
        };
        ExceptionsHttpService.prototype.addToHttpParamsRecursive = function (httpParams, value, key) {
            var _this = this;
            if (value == null) {
                return httpParams;
            }
            if (typeof value === "object") {
                if (Array.isArray(value)) {
                    value.forEach(function (elem) { return httpParams = _this.addToHttpParamsRecursive(httpParams, elem, key); });
                }
                else if (value instanceof Date) {
                    if (key != null) {
                        httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                    }
                    else {
                        throw Error("key may not be null if value is Date");
                    }
                }
                else {
                    Object.keys(value).forEach(function (k) { return httpParams = _this.addToHttpParamsRecursive(httpParams, value[k], key != null ? key + "." + k : k); });
                }
            }
            else if (key != null) {
                httpParams = httpParams.append(key, value);
            }
            else {
                throw Error("key may not be null if value is not object or array");
            }
            return httpParams;
        };
        ExceptionsHttpService.prototype.filterExceptions = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _body = requestParameters["body"];
            if (_body === null || _body === undefined) {
                throw new Error('Required parameter body was null or undefined when calling filterExceptions.');
            }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var consumes = [
                'application/json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected !== undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.post(this.configuration.basePath + "/client-api/v1/exceptions", _body, {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        ExceptionsHttpService.prototype.filterExceptionsUrl = function (requestParameters) {
            var _body = requestParameters["body"];
            if (_body === null || _body === undefined) {
                throw new Error('Required parameter body was null or undefined when calling filterExceptions.');
            }
            return this.configuration.basePath + "/client-api/v1/exceptions";
        };
        ExceptionsHttpService.prototype.getExceptionDetails = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _id = requestParameters["id"];
            if (_id === null || _id === undefined) {
                throw new Error('Required parameter id was null or undefined when calling getExceptionDetails.');
            }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.get(this.configuration.basePath + "/client-api/v1/exceptions/" + encodeURIComponent(String(_id)), {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        ExceptionsHttpService.prototype.getExceptionDetailsUrl = function (requestParameters) {
            var _id = requestParameters["id"];
            if (_id === null || _id === undefined) {
                throw new Error('Required parameter id was null or undefined when calling getExceptionDetails.');
            }
            return this.configuration.basePath + "/client-api/v1/exceptions/" + encodeURIComponent(String(_id));
        };
        return ExceptionsHttpService;
    }());
    ExceptionsHttpService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: ExceptionsHttpService, deps: [{ token: i1__namespace.HttpClient }, { token: POSITIVE_PAY_ACH_BASE_PATH, optional: true }, { token: PositivePayAchConfiguration, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    ExceptionsHttpService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: ExceptionsHttpService, providedIn: 'root' });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: ExceptionsHttpService, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root'
                    }]
            }], ctorParameters: function () {
            return [{ type: i1__namespace.HttpClient }, { type: undefined, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.Inject,
                            args: [POSITIVE_PAY_ACH_BASE_PATH]
                        }] }, { type: PositivePayAchConfiguration, decorators: [{
                            type: i0.Optional
                        }] }];
        } });

    var ExceptionsHttpServiceFilterExceptionsMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v1/exceptions",
            method: "POST",
            responses: [
                {
                    status: 200,
                    body: {
                        "exceptions": [{
                                "id": "df6bf97c-19cc-4056-9524-42cef5ce71b8",
                                "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4692",
                                "paymentType": "ACH Debit",
                                "originatorId": "5123456789",
                                "originatorName": "First Originator",
                                "standardEntryClass": "CIE",
                                "amount": {
                                    "amount": "100.00",
                                    "currencyCode": "USD"
                                },
                                "transactionDate": "2020-12-31T06:30:00.000-05:00",
                                "errors": ["Invalid Standard Entry Class", "Originator is blocked"]
                            }, {
                                "id": "8632554d-1d19-4876-b35a-4cef21934bc6",
                                "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4600",
                                "paymentType": "ACH Credit",
                                "originatorId": "4123456789",
                                "originatorName": "Second Originator",
                                "standardEntryClass": "CCD",
                                "amount": {
                                    "amount": "999.99",
                                    "currencyCode": "USD"
                                },
                                "transactionDate": "2020-12-31T05:30:00.000-05:00",
                                "errors": ["Invalid Standard Entry Class", "Originator is blocked"]
                            }, {
                                "id": "6c5bd090-4430-4c8a-ae3f-d50399b95bd8",
                                "arrangementId": "4cdb2224-8926-4b4d-a99f-1c9dfbbb4697",
                                "paymentType": "ACH Debit",
                                "originatorId": "5120056789",
                                "standardEntryClass": "CCD",
                                "amount": {
                                    "amount": "2657.53",
                                    "currencyCode": "USD"
                                },
                                "transactionDate": "2020-12-31T04:30:00.000-05:00",
                                "errors": ["Invalid payment type"]
                            }, {
                                "id": "1453f0e5-cb1c-4706-b774-620ad7695163",
                                "arrangementId": "5cdb2224-8926-4b4d-a99f-1c9dfbbb4696",
                                "paymentType": "ACH Debit",
                                "originatorId": "2123456789",
                                "originatorName": "Originator First",
                                "standardEntryClass": "CIE",
                                "amount": {
                                    "amount": "10000.00",
                                    "currencyCode": "USD"
                                },
                                "transactionDate": "2020-12-31T03:30:00.000-05:00",
                                "errors": ["Maximum amount exceeded"]
                            }, {
                                "id": "20d499a0-e60e-47ce-bbb4-cae2abce4dea",
                                "arrangementId": "2d94d333-471d-4cd7-84d2-24e822d5ca68",
                                "paymentType": "ACH Debit",
                                "originatorId": "1123456789",
                                "originatorName": "Originator First",
                                "standardEntryClass": "CCD",
                                "amount": {
                                    "amount": "899.00",
                                    "currencyCode": "USD"
                                },
                                "transactionDate": "2020-12-31T02:30:00.000-05:00",
                                "errors": ["Maximum amount exceeded"]
                            }]
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 500,
                    body: {
                        "message": "Description of error"
                    }
                },
            ]
        }]);
    var ExceptionsHttpServiceGetExceptionDetailsMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v1/exceptions/{id}",
            method: "GET",
            responses: [
                {
                    status: 200,
                    body: {
                        "id": "df6bf97c-19cc-4056-9524-42cef5ce71b8",
                        "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4692",
                        "paymentType": "ACH Debit",
                        "originatorId": "5123456789",
                        "originatorName": "First Originator",
                        "standardEntryClass": "CIE",
                        "amount": {
                            "amount": "100.00",
                            "currencyCode": "USD"
                        },
                        "transactionDate": "2020-12-31T06:30:00.000-05:00",
                        "errors": ["Invalid Standard Entry Class", "Originator is blocked"]
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
                {
                    status: 500,
                    body: {
                        "message": "Description of error"
                    }
                },
            ]
        }]);
    var ExceptionsHttpServiceMocksProvider = i2.createMocks([
        {
            urlPattern: "/client-api/v1/exceptions",
            method: "POST",
            responses: [
                {
                    status: 200,
                    body: {
                        "exceptions": [{
                                "id": "df6bf97c-19cc-4056-9524-42cef5ce71b8",
                                "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4692",
                                "paymentType": "ACH Debit",
                                "originatorId": "5123456789",
                                "originatorName": "First Originator",
                                "standardEntryClass": "CIE",
                                "amount": {
                                    "amount": "100.00",
                                    "currencyCode": "USD"
                                },
                                "transactionDate": "2020-12-31T06:30:00.000-05:00",
                                "errors": ["Invalid Standard Entry Class", "Originator is blocked"]
                            }, {
                                "id": "8632554d-1d19-4876-b35a-4cef21934bc6",
                                "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4600",
                                "paymentType": "ACH Credit",
                                "originatorId": "4123456789",
                                "originatorName": "Second Originator",
                                "standardEntryClass": "CCD",
                                "amount": {
                                    "amount": "999.99",
                                    "currencyCode": "USD"
                                },
                                "transactionDate": "2020-12-31T05:30:00.000-05:00",
                                "errors": ["Invalid Standard Entry Class", "Originator is blocked"]
                            }, {
                                "id": "6c5bd090-4430-4c8a-ae3f-d50399b95bd8",
                                "arrangementId": "4cdb2224-8926-4b4d-a99f-1c9dfbbb4697",
                                "paymentType": "ACH Debit",
                                "originatorId": "5120056789",
                                "standardEntryClass": "CCD",
                                "amount": {
                                    "amount": "2657.53",
                                    "currencyCode": "USD"
                                },
                                "transactionDate": "2020-12-31T04:30:00.000-05:00",
                                "errors": ["Invalid payment type"]
                            }, {
                                "id": "1453f0e5-cb1c-4706-b774-620ad7695163",
                                "arrangementId": "5cdb2224-8926-4b4d-a99f-1c9dfbbb4696",
                                "paymentType": "ACH Debit",
                                "originatorId": "2123456789",
                                "originatorName": "Originator First",
                                "standardEntryClass": "CIE",
                                "amount": {
                                    "amount": "10000.00",
                                    "currencyCode": "USD"
                                },
                                "transactionDate": "2020-12-31T03:30:00.000-05:00",
                                "errors": ["Maximum amount exceeded"]
                            }, {
                                "id": "20d499a0-e60e-47ce-bbb4-cae2abce4dea",
                                "arrangementId": "2d94d333-471d-4cd7-84d2-24e822d5ca68",
                                "paymentType": "ACH Debit",
                                "originatorId": "1123456789",
                                "originatorName": "Originator First",
                                "standardEntryClass": "CCD",
                                "amount": {
                                    "amount": "899.00",
                                    "currencyCode": "USD"
                                },
                                "transactionDate": "2020-12-31T02:30:00.000-05:00",
                                "errors": ["Maximum amount exceeded"]
                            }]
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 500,
                    body: {
                        "message": "Description of error"
                    }
                },
            ]
        },
        {
            urlPattern: "/client-api/v1/exceptions/{id}",
            method: "GET",
            responses: [
                {
                    status: 200,
                    body: {
                        "id": "df6bf97c-19cc-4056-9524-42cef5ce71b8",
                        "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4692",
                        "paymentType": "ACH Debit",
                        "originatorId": "5123456789",
                        "originatorName": "First Originator",
                        "standardEntryClass": "CIE",
                        "amount": {
                            "amount": "100.00",
                            "currencyCode": "USD"
                        },
                        "transactionDate": "2020-12-31T06:30:00.000-05:00",
                        "errors": ["Invalid Standard Entry Class", "Originator is blocked"]
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
                {
                    status: 500,
                    body: {
                        "message": "Description of error"
                    }
                },
            ]
        },
    ]);

    var RulesHttpService = /** @class */ (function () {
        function RulesHttpService(httpClient, basePath, configuration) {
            this.httpClient = httpClient;
            this.basePath = '';
            this.defaultHeaders = new i1.HttpHeaders();
            this.configuration = new PositivePayAchConfiguration();
            if (configuration) {
                this.configuration = configuration;
            }
            if (typeof this.configuration.basePath !== 'string') {
                if (typeof basePath !== 'string') {
                    basePath = this.basePath;
                }
                this.configuration.basePath = basePath;
            }
            this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
        }
        RulesHttpService.prototype.addToHttpParams = function (httpParams, value, key) {
            if (typeof value === "object" && value instanceof Date === false) {
                httpParams = this.addToHttpParamsRecursive(httpParams, value);
            }
            else {
                httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
            }
            return httpParams;
        };
        RulesHttpService.prototype.addToHttpParamsRecursive = function (httpParams, value, key) {
            var _this = this;
            if (value == null) {
                return httpParams;
            }
            if (typeof value === "object") {
                if (Array.isArray(value)) {
                    value.forEach(function (elem) { return httpParams = _this.addToHttpParamsRecursive(httpParams, elem, key); });
                }
                else if (value instanceof Date) {
                    if (key != null) {
                        httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                    }
                    else {
                        throw Error("key may not be null if value is Date");
                    }
                }
                else {
                    Object.keys(value).forEach(function (k) { return httpParams = _this.addToHttpParamsRecursive(httpParams, value[k], key != null ? key + "." + k : k); });
                }
            }
            else if (key != null) {
                httpParams = httpParams.append(key, value);
            }
            else {
                throw Error("key may not be null if value is not object or array");
            }
            return httpParams;
        };
        RulesHttpService.prototype.deleteRule = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _id = requestParameters["id"];
            if (_id === null || _id === undefined) {
                throw new Error('Required parameter id was null or undefined when calling deleteRule.');
            }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.delete(this.configuration.basePath + "/client-api/v1/rules/" + encodeURIComponent(String(_id)), {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        RulesHttpService.prototype.deleteRuleUrl = function (requestParameters) {
            var _id = requestParameters["id"];
            if (_id === null || _id === undefined) {
                throw new Error('Required parameter id was null or undefined when calling deleteRule.');
            }
            return this.configuration.basePath + "/client-api/v1/rules/" + encodeURIComponent(String(_id));
        };
        RulesHttpService.prototype.filterRules = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _body = requestParameters["body"];
            if (_body === null || _body === undefined) {
                throw new Error('Required parameter body was null or undefined when calling filterRules.');
            }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var consumes = [
                'application/json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected !== undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.post(this.configuration.basePath + "/client-api/v1/rules", _body, {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        RulesHttpService.prototype.filterRulesUrl = function (requestParameters) {
            var _body = requestParameters["body"];
            if (_body === null || _body === undefined) {
                throw new Error('Required parameter body was null or undefined when calling filterRules.');
            }
            return this.configuration.basePath + "/client-api/v1/rules";
        };
        RulesHttpService.prototype.getRuleDetails = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _id = requestParameters["id"];
            if (_id === null || _id === undefined) {
                throw new Error('Required parameter id was null or undefined when calling getRuleDetails.');
            }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.get(this.configuration.basePath + "/client-api/v1/rules/" + encodeURIComponent(String(_id)), {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        RulesHttpService.prototype.getRuleDetailsUrl = function (requestParameters) {
            var _id = requestParameters["id"];
            if (_id === null || _id === undefined) {
                throw new Error('Required parameter id was null or undefined when calling getRuleDetails.');
            }
            return this.configuration.basePath + "/client-api/v1/rules/" + encodeURIComponent(String(_id));
        };
        RulesHttpService.prototype.submitBlockerRule = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _body = requestParameters["body"];
            if (_body === null || _body === undefined) {
                throw new Error('Required parameter body was null or undefined when calling submitBlockerRule.');
            }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var consumes = [
                'application/json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected !== undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.post(this.configuration.basePath + "/client-api/v1/rules/blockers", _body, {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        RulesHttpService.prototype.submitBlockerRuleUrl = function (requestParameters) {
            var _body = requestParameters["body"];
            if (_body === null || _body === undefined) {
                throw new Error('Required parameter body was null or undefined when calling submitBlockerRule.');
            }
            return this.configuration.basePath + "/client-api/v1/rules/blockers";
        };
        RulesHttpService.prototype.submitFilterRule = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _body = requestParameters["body"];
            if (_body === null || _body === undefined) {
                throw new Error('Required parameter body was null or undefined when calling submitFilterRule.');
            }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var consumes = [
                'application/json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected !== undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.post(this.configuration.basePath + "/client-api/v1/rules/filters", _body, {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        RulesHttpService.prototype.submitFilterRuleUrl = function (requestParameters) {
            var _body = requestParameters["body"];
            if (_body === null || _body === undefined) {
                throw new Error('Required parameter body was null or undefined when calling submitFilterRule.');
            }
            return this.configuration.basePath + "/client-api/v1/rules/filters";
        };
        return RulesHttpService;
    }());
    RulesHttpService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: RulesHttpService, deps: [{ token: i1__namespace.HttpClient }, { token: POSITIVE_PAY_ACH_BASE_PATH, optional: true }, { token: PositivePayAchConfiguration, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    RulesHttpService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: RulesHttpService, providedIn: 'root' });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: RulesHttpService, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root'
                    }]
            }], ctorParameters: function () {
            return [{ type: i1__namespace.HttpClient }, { type: undefined, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.Inject,
                            args: [POSITIVE_PAY_ACH_BASE_PATH]
                        }] }, { type: PositivePayAchConfiguration, decorators: [{
                            type: i0.Optional
                        }] }];
        } });

    var RulesHttpServiceDeleteRuleMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v1/rules/{id}",
            method: "DELETE",
            responses: [
                {
                    status: 204,
                    body: ""
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
                {
                    status: 500,
                    body: {
                        "message": "Description of error"
                    }
                },
            ]
        }]);
    var RulesHttpServiceFilterRulesMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v1/rules",
            method: "POST",
            responses: [
                {
                    status: 200,
                    body: {
                        "rules": [{
                                "id": "5",
                                "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4692",
                                "ruleType": "BLOCKER",
                                "paymentType": "ACH Debit",
                                "originatorId": "5123456789",
                                "originatorName": "First Originator",
                                "standardEntryClass": ["CCD", "CIE"],
                                "startDate": "2021-12-01",
                                "endDate": "2021-12-31",
                                "registrationDate": "2020-12-31T06:30:00.000-05:00"
                            }, {
                                "id": "4",
                                "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4600",
                                "ruleType": "FILTER",
                                "paymentType": "ACH Credit",
                                "originatorId": "4123456789",
                                "originatorName": "Second Originator",
                                "standardEntryClass": ["CCD"],
                                "startDate": "2021-12-01",
                                "endDate": "2021-12-31",
                                "frequency": "Weekly",
                                "maximumAmount": {
                                    "amount": "999.99",
                                    "currencyCode": "USD"
                                },
                                "registrationDate": "2020-12-31T05:30:00.000-05:00"
                            }, {
                                "id": "3",
                                "arrangementId": "4cdb2224-8926-4b4d-a99f-1c9dfbbb4697",
                                "ruleType": "BLOCKER",
                                "paymentType": "ACH Debit",
                                "originatorId": "*",
                                "standardEntryClass": ["CCD", "CIE"],
                                "startDate": "2021-12-01",
                                "endDate": "2021-12-31",
                                "registrationDate": "2020-12-31T04:30:00.000-05:00"
                            }, {
                                "id": "2",
                                "arrangementId": "5cdb2224-8926-4b4d-a99f-1c9dfbbb4696",
                                "ruleType": "BLOCKER",
                                "paymentType": "ACH Debit",
                                "originatorId": "2123456789",
                                "originatorName": "Originator First",
                                "standardEntryClass": ["CCD", "CIE"],
                                "startDate": "2022-01-01",
                                "endDate": "2022-01-31",
                                "registrationDate": "2020-12-31T03:30:00.000-05:00"
                            }, {
                                "id": "1",
                                "arrangementId": "2d94d333-471d-4cd7-84d2-24e822d5ca68",
                                "ruleType": "BLOCKER",
                                "paymentType": "ACH Debit",
                                "originatorId": "1123456789",
                                "originatorName": "Originator First",
                                "standardEntryClass": ["CCD", "CIE"],
                                "startDate": "2021-12-01",
                                "endDate": "2021-12-31",
                                "registrationDate": "2020-12-31T02:30:00.000-05:00"
                            }]
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 500,
                    body: {
                        "message": "Description of error"
                    }
                },
            ]
        }]);
    var RulesHttpServiceGetRuleDetailsMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v1/rules/{id}",
            method: "GET",
            responses: [
                {
                    status: 200,
                    body: {
                        "id": "849d66ae-c942-2ba7-8739-c5a38a596f32",
                        "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4691",
                        "ruleType": "FILTER",
                        "standardEntryClass": ["CCD", "CIE"],
                        "originatorId": "1123456789",
                        "originatorName": "Originator First",
                        "startDate": "2021-12-01",
                        "endDate": "2021-12-31",
                        "frequency": "Once",
                        "maximumAmount": {
                            "amount": "999.99",
                            "currencyCode": "USD"
                        },
                        "paymentType": "ACH Debit",
                        "additions": {
                            "key": "value"
                        }
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
                {
                    status: 500,
                    body: {
                        "message": "Description of error"
                    }
                },
            ]
        }]);
    var RulesHttpServiceSubmitBlockerRuleMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v1/rules/blockers",
            method: "POST",
            responses: [
                {
                    status: 201,
                    body: {
                        "id": "849d66ae-c942-2ba7-8739-c5a38a596f32"
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 500,
                    body: {
                        "message": "Description of error"
                    }
                },
            ]
        }]);
    var RulesHttpServiceSubmitFilterRuleMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v1/rules/filters",
            method: "POST",
            responses: [
                {
                    status: 201,
                    body: {
                        "id": "849d66ae-c942-2ba7-8739-c5a38a596f32"
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 500,
                    body: {
                        "message": "Description of error"
                    }
                },
            ]
        }]);
    var RulesHttpServiceMocksProvider = i2.createMocks([
        {
            urlPattern: "/client-api/v1/rules/{id}",
            method: "DELETE",
            responses: [
                {
                    status: 204,
                    body: ""
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
                {
                    status: 500,
                    body: {
                        "message": "Description of error"
                    }
                },
            ]
        },
        {
            urlPattern: "/client-api/v1/rules",
            method: "POST",
            responses: [
                {
                    status: 200,
                    body: {
                        "rules": [{
                                "id": "5",
                                "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4692",
                                "ruleType": "BLOCKER",
                                "paymentType": "ACH Debit",
                                "originatorId": "5123456789",
                                "originatorName": "First Originator",
                                "standardEntryClass": ["CCD", "CIE"],
                                "startDate": "2021-12-01",
                                "endDate": "2021-12-31",
                                "registrationDate": "2020-12-31T06:30:00.000-05:00"
                            }, {
                                "id": "4",
                                "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4600",
                                "ruleType": "FILTER",
                                "paymentType": "ACH Credit",
                                "originatorId": "4123456789",
                                "originatorName": "Second Originator",
                                "standardEntryClass": ["CCD"],
                                "startDate": "2021-12-01",
                                "endDate": "2021-12-31",
                                "frequency": "Weekly",
                                "maximumAmount": {
                                    "amount": "999.99",
                                    "currencyCode": "USD"
                                },
                                "registrationDate": "2020-12-31T05:30:00.000-05:00"
                            }, {
                                "id": "3",
                                "arrangementId": "4cdb2224-8926-4b4d-a99f-1c9dfbbb4697",
                                "ruleType": "BLOCKER",
                                "paymentType": "ACH Debit",
                                "originatorId": "*",
                                "standardEntryClass": ["CCD", "CIE"],
                                "startDate": "2021-12-01",
                                "endDate": "2021-12-31",
                                "registrationDate": "2020-12-31T04:30:00.000-05:00"
                            }, {
                                "id": "2",
                                "arrangementId": "5cdb2224-8926-4b4d-a99f-1c9dfbbb4696",
                                "ruleType": "BLOCKER",
                                "paymentType": "ACH Debit",
                                "originatorId": "2123456789",
                                "originatorName": "Originator First",
                                "standardEntryClass": ["CCD", "CIE"],
                                "startDate": "2022-01-01",
                                "endDate": "2022-01-31",
                                "registrationDate": "2020-12-31T03:30:00.000-05:00"
                            }, {
                                "id": "1",
                                "arrangementId": "2d94d333-471d-4cd7-84d2-24e822d5ca68",
                                "ruleType": "BLOCKER",
                                "paymentType": "ACH Debit",
                                "originatorId": "1123456789",
                                "originatorName": "Originator First",
                                "standardEntryClass": ["CCD", "CIE"],
                                "startDate": "2021-12-01",
                                "endDate": "2021-12-31",
                                "registrationDate": "2020-12-31T02:30:00.000-05:00"
                            }]
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 500,
                    body: {
                        "message": "Description of error"
                    }
                },
            ]
        },
        {
            urlPattern: "/client-api/v1/rules/{id}",
            method: "GET",
            responses: [
                {
                    status: 200,
                    body: {
                        "id": "849d66ae-c942-2ba7-8739-c5a38a596f32",
                        "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4691",
                        "ruleType": "FILTER",
                        "standardEntryClass": ["CCD", "CIE"],
                        "originatorId": "1123456789",
                        "originatorName": "Originator First",
                        "startDate": "2021-12-01",
                        "endDate": "2021-12-31",
                        "frequency": "Once",
                        "maximumAmount": {
                            "amount": "999.99",
                            "currencyCode": "USD"
                        },
                        "paymentType": "ACH Debit",
                        "additions": {
                            "key": "value"
                        }
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
                {
                    status: 500,
                    body: {
                        "message": "Description of error"
                    }
                },
            ]
        },
        {
            urlPattern: "/client-api/v1/rules/blockers",
            method: "POST",
            responses: [
                {
                    status: 201,
                    body: {
                        "id": "849d66ae-c942-2ba7-8739-c5a38a596f32"
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 500,
                    body: {
                        "message": "Description of error"
                    }
                },
            ]
        },
        {
            urlPattern: "/client-api/v1/rules/filters",
            method: "POST",
            responses: [
                {
                    status: 201,
                    body: {
                        "id": "849d66ae-c942-2ba7-8739-c5a38a596f32"
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "key": "GENERAL_ERROR",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 500,
                    body: {
                        "message": "Description of error"
                    }
                },
            ]
        },
    ]);

    exports.COLLECTION_FORMATS = COLLECTION_FORMATS;
    exports.CONFIG_TOKEN = CONFIG_TOKEN;
    exports.ExceptionsFilterOrderByEnum = ExceptionsFilterOrderByEnum;
    exports.ExceptionsHttpService = ExceptionsHttpService;
    exports.ExceptionsHttpServiceFilterExceptionsMocksProvider = ExceptionsHttpServiceFilterExceptionsMocksProvider;
    exports.ExceptionsHttpServiceGetExceptionDetailsMocksProvider = ExceptionsHttpServiceGetExceptionDetailsMocksProvider;
    exports.ExceptionsHttpServiceMocksProvider = ExceptionsHttpServiceMocksProvider;
    exports.POSITIVE_PAY_ACH_BASE_PATH = POSITIVE_PAY_ACH_BASE_PATH;
    exports.PositivePayAchApiModule = PositivePayAchApiModule;
    exports.PositivePayAchConfiguration = PositivePayAchConfiguration;
    exports.RuleTypeEnum = RuleTypeEnum;
    exports.RulesFilterOrderByEnum = RulesFilterOrderByEnum;
    exports.RulesHttpService = RulesHttpService;
    exports.RulesHttpServiceDeleteRuleMocksProvider = RulesHttpServiceDeleteRuleMocksProvider;
    exports.RulesHttpServiceFilterRulesMocksProvider = RulesHttpServiceFilterRulesMocksProvider;
    exports.RulesHttpServiceGetRuleDetailsMocksProvider = RulesHttpServiceGetRuleDetailsMocksProvider;
    exports.RulesHttpServiceMocksProvider = RulesHttpServiceMocksProvider;
    exports.RulesHttpServiceSubmitBlockerRuleMocksProvider = RulesHttpServiceSubmitBlockerRuleMocksProvider;
    exports.RulesHttpServiceSubmitFilterRuleMocksProvider = RulesHttpServiceSubmitFilterRuleMocksProvider;
    exports.SortDirectionEnum = SortDirectionEnum;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
