// @backbase/accessgroup-ui/esm2015/lib/unassign-data-group/unassign-data-group-modal.component.js
import { Component, Directive, Output, EventEmitter, Input, ChangeDetectionStrategy } from '@angular/core';
import { BbTemplate } from '@backbase/foundation-ang/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/modal";
import * as i2 from "@angular/common";
import * as i3 from "@backbase/ui-ang/button";
export class UnassignDataGroupModalComponent {
    constructor() {
        /**
         * Component context
         */
        this.hostRef = this;
        /**
         * Flag to show modal
         */
        this.isOpen = false;
        this.activeDataGroup = 'account';
        /**
         * Event emitter that emits on cancel button click
         */
        this.cancel = new EventEmitter();
        /**
         * Event emitter that emits on confirm button click
         */
        this.confirm = new EventEmitter();
    }
    /**
     * Function to call on cancel button click
     */
    onCancel() {
        this.cancel.emit();
    }
    /**
     * Function to call on confirm button click
     */
    onConfirm() {
        this.confirm.emit();
    }
}
/** @nocollapse */ UnassignDataGroupModalComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: UnassignDataGroupModalComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ UnassignDataGroupModalComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: UnassignDataGroupModalComponent, selector: "bb-unassign-data-group-modal", inputs: { isOpen: "isOpen", activeDataGroup: "activeDataGroup", jobRole: "jobRole", accountGroupName: "accountGroupName", payeeGroupName: "payeeGroupName" }, outputs: { cancel: "cancel", confirm: "confirm" }, ngImport: i0, template: "<ng-container bbUnassignDataGroupModalCustomizable [bbHostRef]=\"hostRef\"> </ng-container>\n\n<ng-template bbUnassignDataGroupModalCustomizable let-hostRef>\n  <bb-modal-ui [isOpen]=\"hostRef.isOpen\" (cancel)=\"hostRef.onCancel()\" data-role=\"unassign-data-group-modal\">\n    <ng-container\n      *ngTemplateOutlet=\"\n        hostRef.activeDataGroup === 'account' ? unassignAccountGroupHeader : unassignPayeeHeader\n      \"></ng-container>\n    <bb-modal-body-ui>\n      <ng-template bbCustomModalBody>\n        <ng-content></ng-content>\n      </ng-template>\n    </bb-modal-body-ui>\n    <bb-modal-footer-ui>\n      <ng-template bbCustomModalFooter>\n        <div class=\"bb-button-bar\">\n          <button\n            bbButton\n            color=\"danger\"\n            class=\"bb-button-bar__button\"\n            (click)=\"hostRef.onConfirm()\"\n            i18n=\"Delete button@payee-group-delete-modal.delete-btn\"\n            data-role=\"unassign-data-group-modal-unassign-btn\">\n            Unassign\n          </button>\n          <button\n            bbButton\n            color=\"secondary\"\n            class=\"bb-button-bar__button\"\n            (click)=\"hostRef.onCancel()\"\n            i18n=\"Cancel button@payee-group-delete-modal.cancel-btn\"\n            data-role=\"unassign-data-group-modal-cancel-btn\">\n            Cancel\n          </button>\n        </div>\n      </ng-template>\n    </bb-modal-footer-ui>\n  </bb-modal-ui>\n\n  <ng-template #unassignPayeeHeader>\n    <bb-modal-header-ui\n      title=\"Unassign '{{ hostRef.payeeGroupName }}' Payee Group from {{ hostRef.accountGroupName }} in the '{{\n        hostRef.jobRole\n      }}' job role?\"\n      i18n-title=\"Title of unassign payee modal@unassign-data-group-modal.title\"\n      data-role=\"unassign-payee-modal-modal-title\"></bb-modal-header-ui>\n  </ng-template>\n\n  <ng-template #unassignAccountGroupHeader>\n    <bb-modal-header-ui\n      title=\"Unassign '{{ hostRef.accountGroupName }}' Account Group from '{{ hostRef.jobRole }}' job role?\"\n      i18n-title=\"Title of unassign account group modal@unassign-data-group-modal.title\"\n      data-role=\"unassign-account-group-modal-title\"></bb-modal-header-ui>\n  </ng-template>\n</ng-template>\n", components: [{ type: i0.forwardRef(function () { return i1.ModalComponent; }), selector: "bb-modal-ui", inputs: ["isOpen", "aria-labelledby", "aria-describedby", "modalOptions"], outputs: ["confirm", "cancel", "isOpenChange"] }, { type: i0.forwardRef(function () { return i1.ModalBodyComponent; }), selector: "bb-modal-body-ui" }, { type: i0.forwardRef(function () { return i1.ModalFooterComponent; }), selector: "bb-modal-footer-ui", inputs: ["confirmText", "cancelText"], outputs: ["confirm", "cancel"] }, { type: i0.forwardRef(function () { return i1.ModalHeaderComponent; }), selector: "bb-modal-header-ui", inputs: ["title"], outputs: ["close"] }], directives: [{ type: i0.forwardRef(function () { return UnassignDataGroupModalCustomizableDirective; }), selector: "[bbUnassignDataGroupModalCustomizable]" }, { type: i0.forwardRef(function () { return i2.NgTemplateOutlet; }), selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }, { type: i0.forwardRef(function () { return i1.CustomModalBodyDirective; }), selector: "ng-template[bbCustomModalBody]" }, { type: i0.forwardRef(function () { return i1.CustomModalFooterDirective; }), selector: "ng-template[bbCustomModalFooter]" }, { type: i0.forwardRef(function () { return i3.ButtonDirective; }), selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: UnassignDataGroupModalComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-unassign-data-group-modal',
                    templateUrl: './unassign-data-group-modal.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { isOpen: [{
                type: Input
            }], activeDataGroup: [{
                type: Input
            }], jobRole: [{
                type: Input
            }], accountGroupName: [{
                type: Input
            }], payeeGroupName: [{
                type: Input
            }], cancel: [{
                type: Output
            }], confirm: [{
                type: Output
            }] } });
export class UnassignDataGroupModalCustomizableDirective extends BbTemplate {
}
/** @nocollapse */ UnassignDataGroupModalCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: UnassignDataGroupModalCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
/** @nocollapse */ UnassignDataGroupModalCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: UnassignDataGroupModalCustomizableDirective, selector: "[bbUnassignDataGroupModalCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: UnassignDataGroupModalCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbUnassignDataGroupModalCustomizable]',
                }]
        }] });
