// @backbase/ui-ang/esm2015/_example/dropdown-single-select/examples/dropdown-single-select.example.component.js
import { Component } from '@angular/core';
import { Validators } from '@angular/forms';
import * as i0 from "@angular/core";
import * as i1 from "@angular/forms";
import * as i2 from "@backbase/ui-ang/dropdown-single-select";
import * as i3 from "@angular/common";
export class DropdownSingleSelectExampleComponent {
    constructor(formBuilder) {
        this.formBuilder = formBuilder;
        this.objectDropdownOptions = [
            { title: 'DuckDuckGo', url: 'https://www.duckduckgo.com' },
            { title: 'Google', url: 'https://google.com' },
            { title: 'Yahoo', url: 'https://yahoo.com' },
        ];
        this.stringDropdownOptions = ['DuckDuckGo', 'Google', 'Yahoo'];
        this.frequencies = [
            {
                value: 'ONCE',
                every: 1,
            },
            {
                value: 'DAILY',
                every: 1,
            },
            {
                value: 'WEEKLY',
                every: 1,
            },
            {
                value: 'BIWEEKLY',
                every: 1,
            },
            {
                value: 'MONTHLY',
                every: 1,
            },
            {
                value: 'QUARTERLY',
                every: 1,
            },
            {
                value: 'YEARLY',
                every: 1,
            },
        ];
        this.vForm = this.formBuilder.group({
            websiteStrings: [
                this.stringDropdownOptions[0],
                Validators.required,
            ],
            websiteObjects: [
                this.objectDropdownOptions[0],
                Validators.required,
            ],
            frequency: [
                {
                    value: 'ONCE',
                    every: 1,
                },
                Validators.required,
            ],
        });
    }
    compareByValue(p1, p2) {
        if (!p1 || !p2)
            return;
        return p1.value === p2.value;
    }
}
DropdownSingleSelectExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownSingleSelectExampleComponent, deps: [{ token: i1.FormBuilder }], target: i0.ɵɵFactoryTarget.Component });
DropdownSingleSelectExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: DropdownSingleSelectExampleComponent, selector: "bb-dropdown-single-select-example-ui", ngImport: i0, template: "<h4 id=\"ariaLabelledById\">Single select Dropdown</h4>\n<div id=\"ariaDescribedById\">Default with list of string as options</div>\n<div [formGroup]=\"vForm\">\n  <bb-dropdown-single-select-ui\n    label=\"Choose your search engine\"\n    placeholder=\"No search engine selected\"\n    formControlName=\"websiteStrings\"\n    [options]=\"stringDropdownOptions\"\n    [required]=\"true\"\n    [id]=\"'singleDropdown'\"\n    class=\"d-block bb-block bb-block--lg\"\n    [aria-label]=\"'single select'\"\n    [aria-labelledby]=\"'ariaLabelledById'\"\n    [aria-describedby]=\"'ariaDescribedById'\"\n    [aria-owns]=\"'singleDropdown'\"\n  >\n  </bb-dropdown-single-select-ui>\n\n  <bb-dropdown-single-select-ui\n    label=\"Choose your search engine\"\n    placeholder=\"No search engine selected\"\n    formControlName=\"websiteObjects\"\n    displayAttributePath=\"title\"\n    [options]=\"objectDropdownOptions\"\n    [required]=\"true\"\n    class=\"d-block bb-block bb-block--lg\"\n  >\n  </bb-dropdown-single-select-ui>\n\n  <bb-dropdown-single-select-ui\n    [compareWithFn]=\"compareByValue\"\n    label=\"Frequency\"\n    displayAttributePath=\"value\"\n    formControlName=\"frequency\"\n    data-role=\"frequency-selector\"\n  >\n    <bb-dropdown-single-select-option-ui\n      *ngFor=\"let frequency of frequencies\"\n      class=\"text-capitalize\"\n      [value]=\"frequency\"\n      [label]=\"frequency.value\"\n    >\n    </bb-dropdown-single-select-option-ui>\n  </bb-dropdown-single-select-ui>\n</div>\n", components: [{ type: i2.DropdownSingleSelectComponent, selector: "bb-dropdown-single-select-ui", inputs: ["placeholder", "defaultOptionAsPlaceholder", "options", "displayAttributePath", "iconName", "compareWithFn"] }, { type: i2.DropdownSingleSelectOptionComponent, selector: "bb-dropdown-single-select-option-ui", inputs: ["value", "label"] }], directives: [{ type: i1.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { type: i1.FormGroupDirective, selector: "[formGroup]", inputs: ["formGroup"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { type: i1.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { type: i1.FormControlName, selector: "[formControlName]", inputs: ["disabled", "formControlName", "ngModel"], outputs: ["ngModelChange"] }, { type: i1.RequiredValidator, selector: ":not([type=checkbox])[required][formControlName],:not([type=checkbox])[required][formControl],:not([type=checkbox])[required][ngModel]", inputs: ["required"] }, { type: i3.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownSingleSelectExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-dropdown-single-select-example-ui',
                    templateUrl: './dropdown-single-select.example.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.FormBuilder }]; } });
