// @backbase/contact-common-ang/esm2015/lib/components/contact-personal-data/contact-personal-data.module.js
import { NgModule } from '@angular/core';
import { ContactPersonalDataComponent } from './contact-personal-data.component';
import { ContactPersonalDataCustomizableDirective } from './contact-personal-data.directive';
import { FormsModule } from '@angular/forms';
import { ContactAddressDataModule } from '../contact-address-data/contact-address-data.module';
import { CommonModule } from '@angular/common';
import { NoWhiteSpaceModule } from '../../directives/no-white-space-validator/no-white-space.module';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { InputEmailModule } from '@backbase/ui-ang/input-email';
import { EmailDomainModule } from '../../directives/email-domain/email-domain.module';
import * as i0 from "@angular/core";
/**
 * Exports:
 * - ContactPersonalDataComponent
 * - ContactPersonalDataCustomizableDirective
 */
export class ContactPersonalDataModule {
}
/** @nocollapse */ ContactPersonalDataModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactPersonalDataModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ ContactPersonalDataModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactPersonalDataModule, declarations: [ContactPersonalDataComponent, ContactPersonalDataCustomizableDirective], imports: [FormsModule,
        InputTextModule,
        InputValidationMessageModule,
        InputEmailModule,
        ContactAddressDataModule,
        CommonModule,
        NoWhiteSpaceModule,
        EmailDomainModule], exports: [ContactPersonalDataComponent, ContactPersonalDataCustomizableDirective] });
/** @nocollapse */ ContactPersonalDataModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactPersonalDataModule, imports: [[
            FormsModule,
            InputTextModule,
            InputValidationMessageModule,
            InputEmailModule,
            ContactAddressDataModule,
            CommonModule,
            NoWhiteSpaceModule,
            EmailDomainModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactPersonalDataModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ContactPersonalDataComponent, ContactPersonalDataCustomizableDirective],
                    exports: [ContactPersonalDataComponent, ContactPersonalDataCustomizableDirective],
                    imports: [
                        FormsModule,
                        InputTextModule,
                        InputValidationMessageModule,
                        InputEmailModule,
                        ContactAddressDataModule,
                        CommonModule,
                        NoWhiteSpaceModule,
                        EmailDomainModule,
                    ],
                }]
        }] });
