// @backbase/notifications-badge-widget-ang/esm2015/lib/notifications-badge-widget.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { NgbDropdownModule } from '@ng-bootstrap/ng-bootstrap';
import { NotificationsBadgeWidgetComponent, NotificationsBadgeWidgetCustomizableDirective, } from './components/notifications-badge-widget.component';
import { NotificationsListModule } from './components/notifications-list/notifications-list.module';
import { NotificationsListItemModule } from './components/notifications-list-item/notifications-list-item.module';
import { BadgeModule } from '@backbase/ui-ang/badge';
import { ButtonModule } from '@backbase/ui-ang/button';
import { IconModule } from '@backbase/ui-ang/icon';
import { BadgeCounterModule } from '@backbase/ui-ang/badge-counter';
import { MediaQueryModule } from '@backbase/ui-ang/media-query-lib';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const uiModules = [BadgeModule, ButtonModule, IconModule, NgbDropdownModule, BadgeCounterModule, MediaQueryModule];
const innerModules = [NotificationsListItemModule, NotificationsListModule];
const components = [NotificationsBadgeWidgetComponent, NotificationsBadgeWidgetCustomizableDirective];
export class NotificationsBadgeWidgetModule {
}
/** @nocollapse */ NotificationsBadgeWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsBadgeWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ NotificationsBadgeWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsBadgeWidgetModule, declarations: [NotificationsBadgeWidgetComponent, NotificationsBadgeWidgetCustomizableDirective], imports: [CommonModule, i1.BackbaseCoreModule, BadgeModule, ButtonModule, IconModule, NgbDropdownModule, BadgeCounterModule, MediaQueryModule, NotificationsListItemModule, NotificationsListModule], exports: [BadgeModule, ButtonModule, IconModule, NgbDropdownModule, BadgeCounterModule, MediaQueryModule, NotificationsListItemModule, NotificationsListModule, NotificationsBadgeWidgetComponent, NotificationsBadgeWidgetCustomizableDirective] });
/** @nocollapse */ NotificationsBadgeWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsBadgeWidgetModule, imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { NotificationsBadgeWidgetComponent },
            }),
            ...uiModules,
            ...innerModules,
        ], BadgeModule, ButtonModule, IconModule, NgbDropdownModule, BadgeCounterModule, MediaQueryModule, NotificationsListItemModule, NotificationsListModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsBadgeWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { NotificationsBadgeWidgetComponent },
                        }),
                        ...uiModules,
                        ...innerModules,
                    ],
                    declarations: [...components],
                    exports: [...uiModules, ...innerModules, ...components],
                }]
        }] });
