// @backbase/manage-pockets-journey-ang/esm2015/src/manage-pockets-common/services/dom-attributes.service.js
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * Utility service for DOM manipulations
 */
export class DomAttributesService {
    constructor() {
        this.basePrefix = 'bb_manage_pockets';
        this.idCounter = 0;
    }
    nextId(prefix) {
        return `${this.basePrefix}-${!!prefix ? prefix : 'default'}-${Date.now()}-${this.idCounter++}`;
    }
}
DomAttributesService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DomAttributesService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
DomAttributesService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DomAttributesService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DomAttributesService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }] });
