// @backbase/manage-pockets-journey-ang/esm2015/src/manage-pockets-common/services/manage-pockets.service.js
import { Injectable } from '@angular/core';
import { BehaviorSubject, of } from 'rxjs';
import { catchError, map } from 'rxjs/operators';
import { AccountsNamespace } from '../model/accounts.model';
import { PocketNamespace } from '../model/pocket-item.model';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/pocket-tailor";
import * as i2 from "@backbase/data-ang/arrangements";
export class ManagePocketsService {
    constructor(pocketTailorService, productSummaryDataService) {
        this.pocketTailorService = pocketTailorService;
        this.productSummaryDataService = productSummaryDataService;
        this.pocketListApiError = new BehaviorSubject(false);
        this.pocketListApiError$ = this.pocketListApiError.asObservable();
        this.handleApiError = () => {
            this.pocketListApiError.next(true);
            return of({ pockets: [] });
        };
    }
    retrievePocketList() {
        this.pocketListApiError.next(false);
        return this.pocketTailorService.listPockets().pipe(catchError(this.handleApiError));
    }
    postPocket(newPocketItem) {
        const payload = PocketNamespace.buildNewPocketPostRequest(newPocketItem);
        return this.pocketTailorService.postPocket(payload);
    }
    retrieveAccountList() {
        const RESOURCE_NAME = 'Payments';
        const PRIVILEGE = 'create';
        const FROM_PAGES = 0;
        const PAGE_SIZE = 50;
        const BUSINESS_FUNCTION = 'A2A Transfer';
        const productSummaryParameters = {
            businessFunction: BUSINESS_FUNCTION,
            resourceName: RESOURCE_NAME,
            privilege: PRIVILEGE,
            from: FROM_PAGES,
            size: PAGE_SIZE,
        };
        return this.productSummaryDataService
            .getArrangementsByBusinessFunction(productSummaryParameters)
            .pipe(map(AccountsNamespace.formatHttpresponse));
    }
}
ManagePocketsService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsService, deps: [{ token: i1.PocketTailorClientHttpService }, { token: i2.ProductSummaryHttpService }], target: i0.ɵɵFactoryTarget.Injectable });
ManagePocketsService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.PocketTailorClientHttpService }, { type: i2.ProductSummaryHttpService }]; } });
