// @backbase/manage-pockets-journey-ang/esm2015/src/manage-pockets-common/services/payment-order.service.js
import { Injectable } from '@angular/core';
import { PaymentOrderNamespace } from '../model/payment.model';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/payment-order";
export class PaymentOrderService {
    constructor(paymentOrderService) {
        this.paymentOrderService = paymentOrderService;
    }
    transferMoney(paymentInfo) {
        const payload = PaymentOrderNamespace.buildPaymentOrderPostRequest(paymentInfo);
        return this.paymentOrderService.postPaymentOrders(payload);
    }
}
PaymentOrderService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentOrderService, deps: [{ token: i1.PaymentOrdersHttpService }], target: i0.ɵɵFactoryTarget.Injectable });
PaymentOrderService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentOrderService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentOrderService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.PaymentOrdersHttpService }]; } });
