// @backbase/connect-external-accounts-journey-ang/esm2015/lib/helpers/constants.js
export const FORM_INPUT_LENGTHS = {
    name: 30,
    number: 17,
    routingNumber: 9,
};
const externalAcc = 'EXTERNAL_ACCOUNT_VERIFICATION_';
export const RESPONSES = {
    unmatchedKey: `${externalAcc}FAILURE`,
    exceededKey: `${externalAcc}EXCEEDED_MAX_ATTEMPTS`,
    invalidAmount: `${externalAcc}INVALID_DEPOSIT_AMOUNTS`,
};
export const payverisStatuses = {
    InProcess: 'warning',
    PendingActivation: 'warning',
    FailedVerification: 'danger',
    RejectedByExternalBank: 'danger',
    DisabledForExcessiveReturns: 'secondary',
    DisabledForFraud: 'secondary',
};
export const responses = {
    unmatchedKey: `${externalAcc}FAILURE`,
    exceededKey: `${externalAcc}EXCEEDED_MAX_ATTEMPTS`,
    invalidAmount: `${externalAcc}INVALID_DEPOSIT_AMOUNTS`,
};
export const failedStatuses = {
    FailedVerification: 'FailedVerification',
    RejectedByExternalBank: 'RejectedByExternalBank',
    DisabledForExcessiveReturns: 'DisabledForExcessiveReturns',
    DisabledForFraud: 'DisabledForFraud',
};
