// @backbase/contact-common-ang/esm2015/lib/services/approval-http.service.js
import { Injectable } from '@angular/core';
import { ApprovalsHttpService } from '@backbase/data-ang/approvals';
import { ContactsError } from '../models/types';
import { catchError } from 'rxjs/operators';
import { of } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/approvals";
/**
 * Service responsible for connecting Contact service to data-ang services
 *  - `ContactService`
 * This service is dependant on data-ang.
 *
 * @see ContactService
 * @see ApprovalsHttpService,
 * @usageNotes
 *
 * ```ts
 * @NgModule({
 *   ...
 *   providers: [
 *     ...
 *     ApprovalHttpService,
 *   ],
 * })
 * export class MyContactCommonAngModule {}
 * ```
 *
 * ### Inject this service into your component
 *
 * ```ts
 * @Component({
 *   ...
 *   providers: [ApprovalHttpService],
 * })
 *  export class ContactCommonAngComponent {
 * ```
 */
export class ApprovalHttpService {
    constructor(approvalDataHttpService) {
        this.approvalDataHttpService = approvalDataHttpService;
    }
    /**
     * Fetch approval object based on approval ID.
     *
     * @param {string} approvalId - Approval ID
     * @returns {Observable<ApprovalDetails | ContactsError>} observable of response
     */
    getApproval(approvalId) {
        return this.approvalDataHttpService
            .getApprovalByApprovalId({ approvalId })
            .pipe(catchError((error) => of(new ContactsError(error))));
    }
}
/** @nocollapse */ ApprovalHttpService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ApprovalHttpService, deps: [{ token: i1.ApprovalsHttpService }], target: i0.ɵɵFactoryTarget.Injectable });
/** @nocollapse */ ApprovalHttpService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ApprovalHttpService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ApprovalHttpService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.ApprovalsHttpService }]; } });
