// @backbase/contact-common-ang/esm2015/lib/services/contact-country-list.service.js
import { Inject, Injectable } from '@angular/core';
import { BB_CONTACT_COUNTRIES_LIST } from '../config.providers';
import { of } from 'rxjs';
import { ContactHttpService } from './contact-http.service';
import { catchError, map } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "./contact-http.service";
/**
 * Service responsible for providing available countries list
 */
export class ContactCountryListService {
    constructor(countriesList, contactHttpService) {
        this.countriesList = countriesList;
        this.contactHttpService = contactHttpService;
    }
    /**
     * Returns countries configured locally
     *
     * @returns {Observable<string[]>} - stream of country lists
     */
    getFromLocalSource() {
        return of(this.countriesList);
    }
    /**
     * Return countries from external service
     *
     * @returns {Observable<string[]>} - stream of country lists
     */
    getFromExternalSource() {
        return this.contactHttpService.getSupportedCountries().pipe(map((response) => response.body || []), catchError(() => of([])));
    }
}
/** @nocollapse */ ContactCountryListService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactCountryListService, deps: [{ token: BB_CONTACT_COUNTRIES_LIST }, { token: i1.ContactHttpService }], target: i0.ɵɵFactoryTarget.Injectable });
/** @nocollapse */ ContactCountryListService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactCountryListService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactCountryListService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [BB_CONTACT_COUNTRIES_LIST]
                }] }, { type: i1.ContactHttpService }]; } });
