// @backbase/contact-common-ang/esm2015/lib/services/contact-discard-changes.service.js
import { Injectable } from '@angular/core';
import { of, Subject } from 'rxjs';
import * as i0 from "@angular/core";
/**
 * Service responsible for managing logic behind showing data discard confirmation
 */
export class ContactDiscardChangesService {
    constructor() {
        this.discard$ = new Subject();
        this.discardRequested$$ = new Subject();
        this.isFormActive = false;
        /**
         * Stream providing values when discarding data is requested from outside the form component
         */
        this.discardRequested$ = this.discardRequested$$.asObservable();
    }
    /**
     * Registers the event of introducing changes to the form
     */
    startChanges() {
        this.isFormActive = true;
    }
    /**
     * Registers the event of finishing working with the form
     */
    finishChanges() {
        this.isFormActive = false;
    }
    /**
     * Requests that data is discarded from the form
     *
     * @returns {Observable<boolean>} whether data was actually discarded fulfilling the request
     */
    requestDiscard() {
        if (this.isFormActive) {
            this.discard$.complete();
            this.discard$ = new Subject();
            this.discardRequested$$.next();
            return this.discard$.asObservable();
        }
        else {
            return of(true);
        }
    }
    /**
     * Confirms discarding data after it was requested
     */
    discard() {
        this.discard$.next(true);
        this.discard$.complete();
        this.finishChanges();
    }
    /**
     * Cancels discarding data after it was requested
     */
    cancelDiscard() {
        this.discard$.next(false);
        this.discard$.complete();
    }
}
/** @nocollapse */ ContactDiscardChangesService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactDiscardChangesService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
/** @nocollapse */ ContactDiscardChangesService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactDiscardChangesService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactDiscardChangesService, decorators: [{
            type: Injectable
        }] });
