// @backbase/accounts-transactions-journeys-shared-ui-account-ang/esm2015/lib/pipes/to-payment-card-data/to-payment-card-data.pipe.js
import { Pipe } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * Transforms data to payment-card info
 */
export class ToPaymentCardDataPipe {
    transform(debitCard) {
        return {
            number: (debitCard === null || debitCard === void 0 ? void 0 : debitCard.number) || '',
            expirationDate: (debitCard === null || debitCard === void 0 ? void 0 : debitCard.expiryDate) ? new Date(debitCard.expiryDate) : undefined,
            name: '',
            vendor: '',
            cardName: '',
        };
    }
}
/** @nocollapse */ ToPaymentCardDataPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ToPaymentCardDataPipe, deps: [], target: i0.ɵɵFactoryTarget.Pipe });
/** @nocollapse */ ToPaymentCardDataPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ToPaymentCardDataPipe, name: "bbToPaymentCardData" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ToPaymentCardDataPipe, decorators: [{
            type: Pipe,
            args: [{ name: 'bbToPaymentCardData' }]
        }] });
