// @backbase/accounts-transactions-journeys-shared-ui-account-ang/esm2015/lib/pipes/to-payment-card-data/to-payment-card-data.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ToPaymentCardDataPipe } from './to-payment-card-data.pipe';
import * as i0 from "@angular/core";
const pipes = [ToPaymentCardDataPipe];
export class ToPaymentCardDataPipeModule {
}
/** @nocollapse */ ToPaymentCardDataPipeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ToPaymentCardDataPipeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ ToPaymentCardDataPipeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ToPaymentCardDataPipeModule, declarations: [ToPaymentCardDataPipe], imports: [CommonModule], exports: [ToPaymentCardDataPipe] });
/** @nocollapse */ ToPaymentCardDataPipeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ToPaymentCardDataPipeModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ToPaymentCardDataPipeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [...pipes],
                    exports: [...pipes],
                }]
        }] });
