// @backbase/foundation-ang/esm2015/core/backbase-core/stepup/stepup.interceptor.js
import { HttpClient, } from '@angular/common/http';
import { throwError, race } from 'rxjs';
import { switchMap, catchError, map, take } from 'rxjs/operators';
/**
 * @deprecated Will be removed in v7.0.0
 */
export const createInterceptCallback = (interceptConfig) => (responseError) => {
    return (responseError.status === interceptConfig.statusCode &&
        responseError.headers.get(interceptConfig.headerAttribute) !== null);
};
/**
 * @deprecated Will be removed in v7.0.0
 */
export const isInterceptCallback = (interceptConfig) => typeof interceptConfig === 'function';
/**
 * @deprecated Will be removed in v7.0.0
 */
export const DEFAULT_STEPUP_CONFIG = {
    retryRequests: {
        headerAttribute: 'X-MFA',
    },
    interceptResponses: {
        statusCode: 401,
        headerAttribute: 'WWW-Authenticate',
    },
};
/**
 * @deprecated Will be removed in v7.0.0
 */
export class StepUpInterceptor {
    constructor(stepupService, config, injector) {
        this.stepupService = stepupService;
        this.config = config;
        this.injector = injector;
    }
    intercept(req, next) {
        const shouldIntercept = isInterceptCallback(this.config.interceptResponses)
            ? this.config.interceptResponses
            : createInterceptCallback(this.config.interceptResponses);
        return next.handle(req).pipe(catchError((error) => {
            var _a;
            if (!shouldIntercept(error)) {
                // if it's an error which shouldn't be intercepted propagate the error
                // to as if it wasn't intercepted at all.
                return throwError(error);
            }
            this.mediatorComponent = (_a = this.mediatorComponent) !== null && _a !== void 0 ? _a : this.stepupService.stepupComponent;
            // Errors which are to be intercepted should be "paused" and the stream will be handled according
            // to the event coming from a mediator component. On complete event we are going to retry the request
            // while if any other event was fired by the mediator component, we proceed with the error as usual
            const handleMediatorComponentEvent = ([event, token]) => {
                if (event === 'complete' && token) {
                    // The HttpClient cannot be injected because it is a circular dependency, so we must lazily get it only when required.
                    const http = this.injector.get(HttpClient);
                    return http.request(req.clone({ setHeaders: { [this.config.retryRequests.headerAttribute]: token } }));
                }
                else {
                    return throwError(error);
                }
            };
            // we listen to an event from mediator component in order to be able to amend
            // requests "chain", so we can repeat the request or just propagate the error
            const mediatorComponentResponse = race(this.mediatorComponent.instance.complete.pipe(map(value => ['complete', value])), this.mediatorComponent.instance.cancel.pipe(map(() => ['cancel', undefined]))).pipe(take(1), switchMap(handleMediatorComponentEvent));
            // and we trigger the component to open
            this.mediatorComponent.instance.open(error);
            return mediatorComponentResponse;
        }));
    }
}
