// @backbase/foundation-ang/esm2015/core/backbase-core/stepup/stepup.module.js
import { NgModule, ComponentFactoryResolver, ANALYZE_FOR_ENTRY_COMPONENTS, Injector, } from '@angular/core';
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { StepUpInterceptor } from './stepup.interceptor';
import { StepUpService, STEPUP_CONFIG } from './stepup.service';
/**
 * @deprecated Will be removed in v7.0.0
 */
export class StepUpModule {
    static forRoot(config) {
        return {
            ngModule: StepUpModule,
            providers: [
                {
                    provide: ANALYZE_FOR_ENTRY_COMPONENTS,
                    useValue: [config.mediatorComponentClass],
                    multi: true,
                },
                {
                    provide: STEPUP_CONFIG,
                    useValue: config,
                },
                {
                    provide: HTTP_INTERCEPTORS,
                    useClass: StepUpInterceptor,
                    deps: [StepUpService, STEPUP_CONFIG, Injector],
                    multi: true,
                },
                {
                    provide: StepUpService,
                    useClass: StepUpService,
                    deps: [ComponentFactoryResolver, STEPUP_CONFIG],
                },
            ],
        };
    }
}
StepUpModule.decorators = [
    { type: NgModule, args: [{},] }
];
