// @backbase/foundation-ang/esm2015/core/backbase-core/stepup/stepup.service.js
import { ComponentFactoryResolver, Injectable, Inject, InjectionToken, } from '@angular/core';
import { DEFAULT_STEPUP_CONFIG } from './stepup.interceptor';
/**
 * @deprecated Will be removed in v7.0.0
 */
export const STEPUP_CONFIG = new InjectionToken('Custom step-up configuration injection token');
/**
 * @deprecated Will be removed in v7.0.0
 */
export class StepUpService {
    constructor(resolver, stepupConfigParam) {
        this.resolver = resolver;
        this.stepupConfig = Object.assign(Object.assign({}, DEFAULT_STEPUP_CONFIG), stepupConfigParam);
    }
    get stepupComponent() {
        if (this.stepupComponentRef) {
            return this.stepupComponentRef;
        }
        throw new Error('No stepupComponent defined');
    }
    createStepUpComponent(stepupContainer) {
        stepupContainer.clear();
        const factory = this.resolver.resolveComponentFactory(this.stepupConfig.mediatorComponentClass);
        this.stepupComponentRef = stepupContainer.createComponent(factory);
    }
}
StepUpService.decorators = [
    { type: Injectable }
];
StepUpService.ctorParameters = () => [
    { type: ComponentFactoryResolver },
    { type: undefined, decorators: [{ type: Inject, args: [STEPUP_CONFIG,] }] }
];
