// @backbase/notifications-badge-widget-ang/esm2015/lib/components/notifications-list/notifications-list.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NotificationsListComponent, NotificationsListCustomizableDirective } from './notifications-list.component';
import { NotificationsListItemModule } from '../notifications-list-item/notifications-list-item.module';
import { NotificationDetailsModule } from '../notification-details/notification-details.module';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { ErrorCommonStateModule } from '@backbase/ui-ang/common-error-state';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { ButtonModule } from '@backbase/ui-ang/button';
import { AlertModule } from '@backbase/ui-ang/alert';
import * as i0 from "@angular/core";
const uiModules = [
    EmptyStateModule,
    ErrorCommonStateModule,
    LoadButtonModule,
    LoadingIndicatorModule,
    ButtonModule,
    AlertModule,
];
const innerModules = [NotificationsListItemModule, NotificationDetailsModule];
const components = [NotificationsListComponent, NotificationsListCustomizableDirective];
export class NotificationsListModule {
}
/** @nocollapse */ NotificationsListModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsListModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ NotificationsListModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsListModule, declarations: [NotificationsListComponent, NotificationsListCustomizableDirective], imports: [CommonModule, EmptyStateModule,
        ErrorCommonStateModule,
        LoadButtonModule,
        LoadingIndicatorModule,
        ButtonModule,
        AlertModule, NotificationsListItemModule, NotificationDetailsModule], exports: [EmptyStateModule,
        ErrorCommonStateModule,
        LoadButtonModule,
        LoadingIndicatorModule,
        ButtonModule,
        AlertModule, NotificationsListItemModule, NotificationDetailsModule, NotificationsListComponent, NotificationsListCustomizableDirective] });
/** @nocollapse */ NotificationsListModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsListModule, imports: [[CommonModule, ...uiModules, ...innerModules], EmptyStateModule,
        ErrorCommonStateModule,
        LoadButtonModule,
        LoadingIndicatorModule,
        ButtonModule,
        AlertModule, NotificationsListItemModule, NotificationDetailsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsListModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules, ...innerModules],
                    declarations: [...components],
                    exports: [...uiModules, ...innerModules, ...components],
                }]
        }] });
