// @backbase/payment-orders-ang/esm2015/lib/ui/components/payments-controls/payments-controls.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { PaymentsControlsComponent, PaymentsControlsCustomizableDirective, } from './payments-controls.component';
import { IconModule } from '@backbase/ui-ang/icon';
import { ButtonModule } from '@backbase/ui-ang/button';
import { PaymentsSearchModule } from '../payments-search/payments-search.module';
import { PaymentsFilterModule } from '../payments-filter/payments-filter.module';
import * as i0 from "@angular/core";
export class PaymentsControlsModule {
}
/** @nocollapse */ PaymentsControlsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentsControlsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ PaymentsControlsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentsControlsModule, declarations: [PaymentsControlsComponent,
        PaymentsControlsCustomizableDirective], imports: [ButtonModule,
        CommonModule,
        IconModule,
        PaymentsFilterModule,
        PaymentsSearchModule], exports: [PaymentsControlsComponent, PaymentsControlsCustomizableDirective] });
/** @nocollapse */ PaymentsControlsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentsControlsModule, imports: [[
            ButtonModule,
            CommonModule,
            IconModule,
            PaymentsFilterModule,
            PaymentsSearchModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentsControlsModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        PaymentsControlsComponent,
                        PaymentsControlsCustomizableDirective,
                    ],
                    imports: [
                        ButtonModule,
                        CommonModule,
                        IconModule,
                        PaymentsFilterModule,
                        PaymentsSearchModule,
                    ],
                    exports: [PaymentsControlsComponent, PaymentsControlsCustomizableDirective],
                }]
        }] });
