// @backbase/product-summary-ui/esm2015/src/bb-product-kind-collapsible/product-kind-collapsible.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { AmountModule } from '@backbase/ui-ang/amount';
import { IconModule } from '@backbase/ui-ang/icon';
import { KeyboardClickModule } from '@backbase/ui-ang/keyboard-click-directive';
import { ProductKindCollapsibleComponent, ProductKindCollapsibleCustomizableDirective, } from './product-kind-collapsible.component';
import { ProductKindModule } from '../bb-product-kind-ui/product-kind.module';
import { ProductKindNameModule } from '../bb-product-kind-name/bb-product-kind-name.module';
import * as i0 from "@angular/core";
const uiModules = [IconModule, AmountModule, KeyboardClickModule, NgbModule, ProductKindModule, ProductKindNameModule];
const components = [ProductKindCollapsibleComponent, ProductKindCollapsibleCustomizableDirective];
export class ProductKindCollapsibleModule {
}
ProductKindCollapsibleModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductKindCollapsibleModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductKindCollapsibleModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductKindCollapsibleModule, declarations: [ProductKindCollapsibleComponent, ProductKindCollapsibleCustomizableDirective], imports: [CommonModule, IconModule, AmountModule, KeyboardClickModule, NgbModule, ProductKindModule, ProductKindNameModule], exports: [ProductKindCollapsibleComponent, ProductKindCollapsibleCustomizableDirective, IconModule, AmountModule, KeyboardClickModule, NgbModule, ProductKindModule, ProductKindNameModule] });
ProductKindCollapsibleModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductKindCollapsibleModule, imports: [[CommonModule, ...uiModules], IconModule, AmountModule, KeyboardClickModule, NgbModule, ProductKindModule, ProductKindNameModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductKindCollapsibleModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules],
                    declarations: [...components],
                    exports: [...components, ...uiModules],
                }]
        }] });
