// @backbase/payments-collection-ui-ang/esm2015/lib/components/payments-header/payments-header.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { PaymentsHeaderComponent } from './payments-header.component';
import { HeaderModule } from '@backbase/ui-ang/header';
import { DropdownMenuModule } from '@backbase/ui-ang/dropdown-menu';
import { IconModule } from '@backbase/ui-ang/icon';
import { RouterModule } from '@angular/router';
import { HeaderDropdownComponent } from './components/header-dropdown/header-dropdown.component';
import { ButtonModule } from '@backbase/ui-ang/button';
import * as i0 from "@angular/core";
export class PaymentsHeaderModule {
}
/** @nocollapse */ PaymentsHeaderModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentsHeaderModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ PaymentsHeaderModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentsHeaderModule, declarations: [PaymentsHeaderComponent, HeaderDropdownComponent], imports: [CommonModule,
        HeaderModule,
        DropdownMenuModule,
        IconModule,
        RouterModule,
        ButtonModule], exports: [PaymentsHeaderComponent] });
/** @nocollapse */ PaymentsHeaderModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentsHeaderModule, imports: [[
            CommonModule,
            HeaderModule,
            DropdownMenuModule,
            IconModule,
            RouterModule,
            ButtonModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentsHeaderModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [PaymentsHeaderComponent, HeaderDropdownComponent],
                    exports: [PaymentsHeaderComponent],
                    imports: [
                        CommonModule,
                        HeaderModule,
                        DropdownMenuModule,
                        IconModule,
                        RouterModule,
                        ButtonModule,
                    ],
                }]
        }] });
