// @backbase/ui-ang/esm2015/fieldset/fieldset.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FieldsetComponent } from './fieldset.component';
import * as i0 from "@angular/core";
export class FieldsetModule {
}
FieldsetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FieldsetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
FieldsetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FieldsetModule, declarations: [FieldsetComponent], imports: [CommonModule], exports: [FieldsetComponent] });
FieldsetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FieldsetModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FieldsetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [FieldsetComponent],
                    exports: [FieldsetComponent],
                }]
        }] });
