// @backbase/ui-ang/esm2015/item-log/item-log.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ItemLogComponent, ItemLogRecordDirective } from './item-log.component';
import { IconModule } from '@backbase/ui-ang/icon';
import * as i0 from "@angular/core";
export class ItemLogModule {
}
ItemLogModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ItemLogModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ItemLogModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ItemLogModule, declarations: [ItemLogComponent, ItemLogRecordDirective], imports: [CommonModule, IconModule], exports: [ItemLogComponent, ItemLogRecordDirective] });
ItemLogModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ItemLogModule, imports: [[CommonModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ItemLogModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, IconModule],
                    declarations: [ItemLogComponent, ItemLogRecordDirective],
                    exports: [ItemLogComponent, ItemLogRecordDirective],
                }]
        }] });
