// @backbase/cards-management-journey-ang/esm2015/lib/guard/cards-management-travel-notice.guard.js
import { Inject, Injectable, Optional } from '@angular/core';
import { CardsManagementJourneyConfigurationToken, getDefaultConfigValues, } from '../cards-management-journey-config.service';
import * as i0 from "@angular/core";
export class CardsManagementTravelNoticeGuard {
    constructor(config) {
        this.config = config;
        this._config = getDefaultConfigValues(this.config);
    }
    canActivate() {
        return this._config.enableTravelNotice;
    }
}
/** @nocollapse */ CardsManagementTravelNoticeGuard.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardsManagementTravelNoticeGuard, deps: [{ token: CardsManagementJourneyConfigurationToken, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
/** @nocollapse */ CardsManagementTravelNoticeGuard.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardsManagementTravelNoticeGuard, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardsManagementTravelNoticeGuard, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [CardsManagementJourneyConfigurationToken]
                }] }]; } });
