// @backbase/saving-goals-http-ang/fesm2015/backbase-saving-goals-http-ang.js
import * as i0 from '@angular/core';
import { InjectionToken, NgModule, Optional, SkipSelf, Injectable, Inject } from '@angular/core';
import * as i1 from '@angular/common/http';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import * as i2 from '@backbase/foundation-ang/data-http';
import { createMocks } from '@backbase/foundation-ang/data-http';

var IdstandingorderPostRequestBody;
(function (IdstandingorderPostRequestBody) {
    IdstandingorderPostRequestBody.FrequencyEnum = {
        MONTHLY: 'MONTHLY',
        WEEKLY: 'WEEKLY',
        DAILY: 'DAILY'
    };
})(IdstandingorderPostRequestBody || (IdstandingorderPostRequestBody = {}));

const SAVING_GOALS_BASE_PATH = new InjectionToken('SAVING_GOALS_BASE_PATH');
const COLLECTION_FORMATS = {
    'csv': ',',
    'tsv': '   ',
    'ssv': ' ',
    'pipes': '|'
};

class SavingGoalsConfiguration {
    constructor(configurationParameters = {}) {
        this.apiKeys = configurationParameters.apiKeys;
        this.username = configurationParameters.username;
        this.password = configurationParameters.password;
        this.accessToken = configurationParameters.accessToken;
        this.basePath = configurationParameters.basePath;
        this.withCredentials = configurationParameters.withCredentials;
        this.encoder = configurationParameters.encoder;
        if (configurationParameters.credentials) {
            this.credentials = configurationParameters.credentials;
        }
        else {
            this.credentials = {};
        }
    }
    selectHeaderContentType(contentTypes) {
        if (contentTypes.length === 0) {
            return undefined;
        }
        const type = contentTypes.find((x) => this.isJsonMime(x));
        if (type === undefined) {
            return contentTypes[0];
        }
        return type;
    }
    selectHeaderAccept(accepts) {
        if (accepts.length === 0) {
            return undefined;
        }
        const type = accepts.find((x) => this.isJsonMime(x));
        if (type === undefined) {
            return accepts[0];
        }
        return type;
    }
    isJsonMime(mime) {
        const jsonMime = new RegExp('^(application\/json|[^;/ \t]+\/[^;/ \t]+[+]json)[ \t]*(;.*)?$', 'i');
        return mime !== null && (jsonMime.test(mime) || mime.toLowerCase() === 'application/json-patch+json');
    }
    lookupCredential(key) {
        const value = this.credentials[key];
        return typeof value === 'function'
            ? value()
            : value;
    }
}
const CONFIG_TOKEN = new InjectionToken('api-data-module SavingGoalsConfiguration');

class SavingGoalsApiModule {
    constructor(parentModule, http, dataModulesManager, config) {
        if (parentModule) {
            throw new Error('SavingGoalsApiModule is already loaded. Import in your base AppModule only.');
        }
        if (!http) {
            throw new Error('You need to import the HttpClientModule in your AppModule! \n' +
                'See also https://github.com/angular/angular/issues/20575');
        }
        if (dataModulesManager) {
            dataModulesManager.setModuleConfig(CONFIG_TOKEN, {
                apiRoot: '',
                servicePath: config.basePath || '',
                headers: {},
            });
        }
    }
    static forRoot(configurationFactory) {
        return {
            ngModule: SavingGoalsApiModule,
            providers: [{ provide: SavingGoalsConfiguration, useFactory: configurationFactory }]
        };
    }
}
SavingGoalsApiModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: SavingGoalsApiModule, deps: [{ token: SavingGoalsApiModule, optional: true, skipSelf: true }, { token: i1.HttpClient, optional: true }, { token: i2.DataModulesManager, optional: true }, { token: SavingGoalsConfiguration }], target: i0.ɵɵFactoryTarget.NgModule });
SavingGoalsApiModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: SavingGoalsApiModule });
SavingGoalsApiModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: SavingGoalsApiModule, providers: [], imports: [[]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: SavingGoalsApiModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [],
                    declarations: [],
                    exports: [],
                    providers: []
                }]
        }], ctorParameters: function () { return [{ type: SavingGoalsApiModule, decorators: [{
                    type: Optional
                }, {
                    type: SkipSelf
                }] }, { type: i1.HttpClient, decorators: [{
                    type: Optional
                }] }, { type: i2.DataModulesManager, decorators: [{
                    type: Optional
                }] }, { type: SavingGoalsConfiguration }]; } });

class CustomHttpParameterCodec {
    encodeKey(k) {
        return encodeURIComponent(k);
    }
    encodeValue(v) {
        return encodeURIComponent(v);
    }
    decodeKey(k) {
        return decodeURIComponent(k);
    }
    decodeValue(v) {
        return decodeURIComponent(v);
    }
}

class SavinggoalsHttpService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new SavingGoalsConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    deleteId(requestParameters, observe = 'body', reportProgress = false, options) {
        const _id = requestParameters["id"];
        if (_id === null || _id === undefined) {
            throw new Error('Required parameter id was null or undefined when calling deleteId.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.delete(`${this.configuration.basePath}/client-api/v2/savinggoals/${encodeURIComponent(String(_id))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    deleteIdUrl(requestParameters) {
        const _id = requestParameters["id"];
        if (_id === null || _id === undefined) {
            throw new Error('Required parameter id was null or undefined when calling deleteId.');
        }
        return `${this.configuration.basePath}/client-api/v2/savinggoals/${encodeURIComponent(String(_id))}`;
    }
    getId(requestParameters, observe = 'body', reportProgress = false, options) {
        const _id = requestParameters["id"];
        if (_id === null || _id === undefined) {
            throw new Error('Required parameter id was null or undefined when calling getId.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/savinggoals/${encodeURIComponent(String(_id))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getIdUrl(requestParameters) {
        const _id = requestParameters["id"];
        if (_id === null || _id === undefined) {
            throw new Error('Required parameter id was null or undefined when calling getId.');
        }
        return `${this.configuration.basePath}/client-api/v2/savinggoals/${encodeURIComponent(String(_id))}`;
    }
    getSavingGoals(requestParameters, observe = 'body', reportProgress = false, options) {
        const _arrangementId = requestParameters["arrangementId"];
        if (_arrangementId === null || _arrangementId === undefined) {
            throw new Error('Required parameter arrangementId was null or undefined when calling getSavingGoals.');
        }
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_arrangementId !== undefined && _arrangementId !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _arrangementId, 'arrangementId');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/savinggoals`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getSavingGoalsUrl(requestParameters) {
        const _arrangementId = requestParameters["arrangementId"];
        if (_arrangementId === null || _arrangementId === undefined) {
            throw new Error('Required parameter arrangementId was null or undefined when calling getSavingGoals.');
        }
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_arrangementId !== undefined && _arrangementId !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _arrangementId, 'arrangementId');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/savinggoals${queryString ? `?${queryString}` : ''}`;
    }
    postIdstandingorder(requestParameters, observe = 'body', reportProgress = false, options) {
        const _id = requestParameters["id"];
        if (_id === null || _id === undefined) {
            throw new Error('Required parameter id was null or undefined when calling postIdstandingorder.');
        }
        const _idstandingorderPostRequestBody = requestParameters["idstandingorderPostRequestBody"];
        if (_idstandingorderPostRequestBody === null || _idstandingorderPostRequestBody === undefined) {
            throw new Error('Required parameter idstandingorderPostRequestBody was null or undefined when calling postIdstandingorder.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v2/savinggoals/${encodeURIComponent(String(_id))}/standing-order`, _idstandingorderPostRequestBody, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    postIdstandingorderUrl(requestParameters) {
        const _id = requestParameters["id"];
        if (_id === null || _id === undefined) {
            throw new Error('Required parameter id was null or undefined when calling postIdstandingorder.');
        }
        const _idstandingorderPostRequestBody = requestParameters["idstandingorderPostRequestBody"];
        if (_idstandingorderPostRequestBody === null || _idstandingorderPostRequestBody === undefined) {
            throw new Error('Required parameter idstandingorderPostRequestBody was null or undefined when calling postIdstandingorder.');
        }
        return `${this.configuration.basePath}/client-api/v2/savinggoals/${encodeURIComponent(String(_id))}/standing-order`;
    }
    postSavingGoals(requestParameters, observe = 'body', reportProgress = false, options) {
        const _savingGoalsPostRequestBody = requestParameters["savingGoalsPostRequestBody"];
        if (_savingGoalsPostRequestBody === null || _savingGoalsPostRequestBody === undefined) {
            throw new Error('Required parameter savingGoalsPostRequestBody was null or undefined when calling postSavingGoals.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v2/savinggoals`, _savingGoalsPostRequestBody, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    postSavingGoalsUrl(requestParameters) {
        const _savingGoalsPostRequestBody = requestParameters["savingGoalsPostRequestBody"];
        if (_savingGoalsPostRequestBody === null || _savingGoalsPostRequestBody === undefined) {
            throw new Error('Required parameter savingGoalsPostRequestBody was null or undefined when calling postSavingGoals.');
        }
        return `${this.configuration.basePath}/client-api/v2/savinggoals`;
    }
    putId(requestParameters, observe = 'body', reportProgress = false, options) {
        const _id = requestParameters["id"];
        if (_id === null || _id === undefined) {
            throw new Error('Required parameter id was null or undefined when calling putId.');
        }
        const _idPutRequestBody = requestParameters["idPutRequestBody"];
        if (_idPutRequestBody === null || _idPutRequestBody === undefined) {
            throw new Error('Required parameter idPutRequestBody was null or undefined when calling putId.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v2/savinggoals/${encodeURIComponent(String(_id))}`, _idPutRequestBody, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    putIdUrl(requestParameters) {
        const _id = requestParameters["id"];
        if (_id === null || _id === undefined) {
            throw new Error('Required parameter id was null or undefined when calling putId.');
        }
        const _idPutRequestBody = requestParameters["idPutRequestBody"];
        if (_idPutRequestBody === null || _idPutRequestBody === undefined) {
            throw new Error('Required parameter idPutRequestBody was null or undefined when calling putId.');
        }
        return `${this.configuration.basePath}/client-api/v2/savinggoals/${encodeURIComponent(String(_id))}`;
    }
}
SavinggoalsHttpService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: SavinggoalsHttpService, deps: [{ token: i1.HttpClient }, { token: SAVING_GOALS_BASE_PATH, optional: true }, { token: SavingGoalsConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
SavinggoalsHttpService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: SavinggoalsHttpService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: SavinggoalsHttpService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [SAVING_GOALS_BASE_PATH]
                }] }, { type: SavingGoalsConfiguration, decorators: [{
                    type: Optional
                }] }]; } });

const SavinggoalsHttpServiceDeleteIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/savinggoals/{id}",
        method: "DELETE",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
const SavinggoalsHttpServiceGetIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/savinggoals/{id}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "1955e686-d31e-4216-b3dd-5d66161d536b",
                    "goalName": "Holiday to Hawaii",
                    "goalAmount": {
                        "amount": "2000.00",
                        "currencyCode": "EUR"
                    },
                    "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                    "amountProgress": {
                        "amount": "550.00",
                        "currencyCode": "EUR"
                    },
                    "progressPercentage": 27.5,
                    "performancePercentage": 27.5,
                    "timePercentage": 50.5,
                    "amountRemaining": {
                        "amount": "1450.00",
                        "currencyCode": "EUR"
                    },
                    "startDate": "2017-02-28",
                    "endDate": "2019-02-28",
                    "initialAmount": {
                        "amount": "10.00",
                        "currencyCode": "EUR"
                    },
                    "recipeId": "65b5c1b7-17e0-435f-a828-e2f6afec26a9"
                }
            },
            {
                status: 200,
                body: {
                    "arrangementId": "arrangementId",
                    "goalAmount": {
                        "amount": "amount",
                        "additions": {
                            "key": "additions"
                        },
                        "currencyCode": "currencyCode"
                    },
                    "additions": {
                        "key": "additions"
                    },
                    "endDate": "2000-01-23",
                    "recipeId": "recipeId",
                    "goalName": "goalName",
                    "amountRemaining": {
                        "amount": "amount",
                        "additions": {
                            "key": "additions"
                        },
                        "currencyCode": "currencyCode"
                    },
                    "initialAmount": {
                        "amount": "amount",
                        "additions": {
                            "key": "additions"
                        },
                        "currencyCode": "currencyCode"
                    },
                    "progressPercentage": 0.8008281904610115,
                    "performancePercentage": 6.027456183070403,
                    "amountProgress": {
                        "amount": "amount",
                        "additions": {
                            "key": "additions"
                        },
                        "currencyCode": "currencyCode"
                    },
                    "id": "id",
                    "timePercentage": 1.4658129805029452,
                    "startDate": "2000-01-23"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
const SavinggoalsHttpServiceGetSavingGoalsMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/savinggoals",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "1955e686-d31e-4216-b3dd-5d66161d536b",
                        "goalName": "Holiday to Hawaii",
                        "goalAmount": {
                            "amount": "2000.00",
                            "currencyCode": "EUR"
                        },
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "amountProgress": {
                            "amount": "550.00",
                            "currencyCode": "EUR"
                        },
                        "progressPercentage": 27.5,
                        "performancePercentage": 27.5,
                        "timePercentage": 50.5,
                        "amountRemaining": {
                            "amount": "1450.00",
                            "currencyCode": "EUR"
                        },
                        "startDate": "2017-02-28",
                        "endDate": "2019-02-28",
                        "initialAmount": {
                            "amount": "10.00",
                            "currencyCode": "EUR"
                        },
                        "recipeId": "65b5c1b7-17e0-435f-a828-e2f6afec26a9"
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
const SavinggoalsHttpServicePostIdstandingorderMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/savinggoals/{id}/standing-order",
        method: "POST",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
const SavinggoalsHttpServicePostSavingGoalsMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/savinggoals",
        method: "POST",
        responses: [
            {
                status: 201,
                body: {
                    "id": "0955e686-d31e-4216-b3dd-5d77161d536d"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
const SavinggoalsHttpServicePutIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/savinggoals/{id}",
        method: "PUT",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
const SavinggoalsHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v2/savinggoals/{id}",
        method: "DELETE",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/savinggoals/{id}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "1955e686-d31e-4216-b3dd-5d66161d536b",
                    "goalName": "Holiday to Hawaii",
                    "goalAmount": {
                        "amount": "2000.00",
                        "currencyCode": "EUR"
                    },
                    "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                    "amountProgress": {
                        "amount": "550.00",
                        "currencyCode": "EUR"
                    },
                    "progressPercentage": 27.5,
                    "performancePercentage": 27.5,
                    "timePercentage": 50.5,
                    "amountRemaining": {
                        "amount": "1450.00",
                        "currencyCode": "EUR"
                    },
                    "startDate": "2017-02-28",
                    "endDate": "2019-02-28",
                    "initialAmount": {
                        "amount": "10.00",
                        "currencyCode": "EUR"
                    },
                    "recipeId": "65b5c1b7-17e0-435f-a828-e2f6afec26a9"
                }
            },
            {
                status: 200,
                body: {
                    "arrangementId": "arrangementId",
                    "goalAmount": {
                        "amount": "amount",
                        "additions": {
                            "key": "additions"
                        },
                        "currencyCode": "currencyCode"
                    },
                    "additions": {
                        "key": "additions"
                    },
                    "endDate": "2000-01-23",
                    "recipeId": "recipeId",
                    "goalName": "goalName",
                    "amountRemaining": {
                        "amount": "amount",
                        "additions": {
                            "key": "additions"
                        },
                        "currencyCode": "currencyCode"
                    },
                    "initialAmount": {
                        "amount": "amount",
                        "additions": {
                            "key": "additions"
                        },
                        "currencyCode": "currencyCode"
                    },
                    "progressPercentage": 0.8008281904610115,
                    "performancePercentage": 6.027456183070403,
                    "amountProgress": {
                        "amount": "amount",
                        "additions": {
                            "key": "additions"
                        },
                        "currencyCode": "currencyCode"
                    },
                    "id": "id",
                    "timePercentage": 1.4658129805029452,
                    "startDate": "2000-01-23"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/savinggoals",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "1955e686-d31e-4216-b3dd-5d66161d536b",
                        "goalName": "Holiday to Hawaii",
                        "goalAmount": {
                            "amount": "2000.00",
                            "currencyCode": "EUR"
                        },
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "amountProgress": {
                            "amount": "550.00",
                            "currencyCode": "EUR"
                        },
                        "progressPercentage": 27.5,
                        "performancePercentage": 27.5,
                        "timePercentage": 50.5,
                        "amountRemaining": {
                            "amount": "1450.00",
                            "currencyCode": "EUR"
                        },
                        "startDate": "2017-02-28",
                        "endDate": "2019-02-28",
                        "initialAmount": {
                            "amount": "10.00",
                            "currencyCode": "EUR"
                        },
                        "recipeId": "65b5c1b7-17e0-435f-a828-e2f6afec26a9"
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/savinggoals/{id}/standing-order",
        method: "POST",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/savinggoals",
        method: "POST",
        responses: [
            {
                status: 201,
                body: {
                    "id": "0955e686-d31e-4216-b3dd-5d77161d536d"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/savinggoals/{id}",
        method: "PUT",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
]);

export { COLLECTION_FORMATS, CONFIG_TOKEN, IdstandingorderPostRequestBody, SAVING_GOALS_BASE_PATH, SavingGoalsApiModule, SavingGoalsConfiguration, SavinggoalsHttpService, SavinggoalsHttpServiceDeleteIdMocksProvider, SavinggoalsHttpServiceGetIdMocksProvider, SavinggoalsHttpServiceGetSavingGoalsMocksProvider, SavinggoalsHttpServiceMocksProvider, SavinggoalsHttpServicePostIdstandingorderMocksProvider, SavinggoalsHttpServicePostSavingGoalsMocksProvider, SavinggoalsHttpServicePutIdMocksProvider };
