// @backbase/ui-ang/esm2015/ellipsis/directives/is-content-truncated.directive.js
import { Directive, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class IsContentTruncatedDirective {
    constructor(elementRef, cd) {
        this.elementRef = elementRef;
        this.cd = cd;
    }
    ngAfterViewInit() {
        if (!this.hostRef.setIsContentTruncated) {
            throw new Error('"hostRef" should implement "CheckContentTruncationBehaviour" interface to set truncation status');
        }
        setTimeout(() => {
            this.setIsContentTruncatedInParentComponent();
        });
    }
    setIsContentTruncatedInParentComponent() {
        const isContentTruncated = this.isContentTruncated(this.elementRef.nativeElement);
        this.hostRef.setIsContentTruncated(isContentTruncated);
        this.cd.markForCheck();
    }
    isContentTruncated({ offsetHeight, scrollHeight }) {
        return offsetHeight < scrollHeight;
    }
}
IsContentTruncatedDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IsContentTruncatedDirective, deps: [{ token: i0.ElementRef }, { token: i0.ChangeDetectorRef }], target: i0.ɵɵFactoryTarget.Directive });
IsContentTruncatedDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: IsContentTruncatedDirective, selector: "[bbIsContentTruncated]", inputs: { hostRef: "hostRef" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IsContentTruncatedDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbIsContentTruncated]',
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.ChangeDetectorRef }]; }, propDecorators: { hostRef: [{
                type: Input
            }] } });
