// @backbase/loans-collection-ui-ang/esm2015/lib/payment-schedule/amortized-payment/loan-amortized-payment-schedule-list/loan-amortized-payment-schedule-list.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AmountModule } from '@backbase/ui-ang/amount';
import { BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import { KeyboardClickModule } from '@backbase/ui-ang/keyboard-click-directive';
import { LoanAmortizedPaymentScheduleListComponent } from './loan-amortized-payment-schedule-list.component';
import { LoanPaymentScheduleStatusModule } from '../../loan-payment-schedule-status/loan-payment-schedule-status.module';
import * as i0 from "@angular/core";
export class LoanAmortizedPaymentScheduleListModule {
}
/** @nocollapse */ LoanAmortizedPaymentScheduleListModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanAmortizedPaymentScheduleListModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanAmortizedPaymentScheduleListModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanAmortizedPaymentScheduleListModule, declarations: [LoanAmortizedPaymentScheduleListComponent], imports: [CommonModule, LoanPaymentScheduleStatusModule, AmountModule, BbDatePipeModule, KeyboardClickModule], exports: [LoanAmortizedPaymentScheduleListComponent] });
/** @nocollapse */ LoanAmortizedPaymentScheduleListModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanAmortizedPaymentScheduleListModule, imports: [[CommonModule, LoanPaymentScheduleStatusModule, AmountModule, BbDatePipeModule, KeyboardClickModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanAmortizedPaymentScheduleListModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanAmortizedPaymentScheduleListComponent],
                    imports: [CommonModule, LoanPaymentScheduleStatusModule, AmountModule, BbDatePipeModule, KeyboardClickModule],
                    exports: [LoanAmortizedPaymentScheduleListComponent],
                }]
        }] });
