// @backbase/ui-ang/esm2015/_example/product-item-credit-card/examples/product-item-credit-card.example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/product-item-credit-card";
export class ProductItemCreditCardExampleComponent {
    constructor() {
        this.product = {
            id: 'b387bfa5-75fd-410e-bb86-7f764f8ef69c',
            name: 'Ricardo Norton\'s Credit Card',
            currency: 'USD',
            amount: '1425.99',
            productNumber: 'CH6629039MLXHC9VOO7M9',
        };
        this.interest = {
            id: 'd2d85314-e60c-44af-ba98-8f704dad5970',
            name: 'Available Balance',
            currency: 'USD',
            amount: '880.54',
            productNumber: 'CH6629039MLXHC9VOO7M9',
        };
        this.credit = {
            id: 'e9af649e-e338-4b3e-808b-62e9feae0865',
            name: 'Credit Limit',
            currency: 'USD',
            amount: '7280.86',
        };
        this.productDetails = {
            owner: 'Richard Norton',
        };
        this.productNumberFormat = {
            length: 18,
            maskRange: [0, 0],
            segments: 4,
        };
    }
}
ProductItemCreditCardExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCreditCardExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ProductItemCreditCardExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ProductItemCreditCardExampleComponent, selector: "bb-product-item-credit-card-example-ui", ngImport: i0, template: "<bb-product-item-credit-card-ui\n  [title]=\"product.name\"\n  [amount]=\"product.amount\"\n  [productNumber]=\"product.productNumber\"\n  [currency]=\"product.currency\"\n  [productNumberFormat]=\"productNumberFormat\"\n  [mainSectionCurrency]=\"interest.currency\"\n  [mainSectionAmount]=\"interest.amount\"\n  [mainSectionTitle]=\"interest.name\"\n  [secondarySectionCurrency]=\"credit.currency\"\n  [secondarySectionAmount]=\"credit.amount\"\n  [secondarySectionTitle]=\"credit.name\"\n  [productDetailsInfo]=\"productDetails\"\n  [highlight]=\"false\"\n  [favorite]=\"false\"\n></bb-product-item-credit-card-ui>\n", components: [{ type: i1.ProductItemCreditCardComponent, selector: "bb-product-item-credit-card-ui", inputs: ["title", "productNumber", "productNumberFormat", "currency", "amount", "mainSectionCurrency", "mainSectionAmount", "mainSectionTitle", "secondarySectionCurrency", "secondarySectionAmount", "secondarySectionTitle", "highlight", "productDetailsInfo", "active", "favorite", "favoriteIconColor"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCreditCardExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-product-item-credit-card-example-ui',
                    templateUrl: './product-item-credit-card.example.component.html',
                }]
        }] });
