// @backbase/account-statement-common-ang/esm2015/src/services/journey-configuration.service.js
import { of } from 'rxjs';
import { DEFAULT_JOURNEY_CONFIGURATION } from '../constants/journey-configuration';
/**
 * Abstract class for all the journeys that require common configuration
 *
 * @template T - Type of the configuration object for journey
 * @internal
 */
export class JourneyConfigurationService {
    /**
     * Create new instance of journey service
     *
     * @param staticConfigOverrides new configuration object provided by concrete class
     */
    constructor(staticConfigOverrides) {
        this.staticConfig = Object.assign(Object.assign({}, DEFAULT_JOURNEY_CONFIGURATION), staticConfigOverrides);
    }
    /** Absolute filter datepicker limit (yyyy-mm-dd)
     *
     * @defaultValue ''
     * @returns An observable of string
     */
    get absoluteDateLimit() {
        return of(this.staticConfig.absoluteDateLimit);
    }
    /**
     * Notification dismiss time in seconds (defaults to 5)
     *
     * @defaultValue 5
     * @returns An observable of number
     */
    get notificationDismissTime() {
        return of(this.staticConfig.notificationDismissTime);
    }
    /**
     * PaginationType will decide whether the user will see load more or page number style pagination
     *
     * @defaultValue `PaginationType.LOAD_MORE`
     * @returns An observable of PaginationType enum
     */
    get paginationType() {
        return of(this.staticConfig.paginationType);
    }
    /**
     * Number of items to be fetched in single request
     *
     * @defaultValue 10
     * @returns An observable of number
     */
    get pageSize() {
        return of(this.staticConfig.pageSize);
    }
    /**
     * Relative filter datepicker limit (months)
     *
     * @defaultValue 0
     * @returns An observable of number
     */
    get relativeDateLimit() {
        return of(this.staticConfig.relativeDateLimit);
    }
    /**
     * Sorting direction (ASC, DESC)
     *
     * @defaultValue DESC
     * @returns a string
     */
    get sortingDirection() {
        return of(this.staticConfig.sortingDirection);
    }
    /**
     * Sorting order (date, description, category)
     *
     * @defaultValue date
     * @returns a string
     */
    get sortingOrder() {
        return of(this.staticConfig.sortingOrder);
    }
    /**
     * Show category
     *
     * @defaultValue true
     * @returns An observable of boolean
     */
    get showCategory() {
        return of(this.staticConfig.showCategory);
    }
}
