// @backbase/ui-ang/esm2015/_example/file-attachment/examples/file-attachment.example.component.js
import { Component } from '@angular/core';
import { of } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/file-attachment";
export class FileAttachmentExampleComponent {
    constructor() {
        this.attachment = {
            name: 'examplePDF.pdf',
            size: 10000,
            fileContent: of(new ArrayBuffer(2000000)),
        };
    }
}
FileAttachmentExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FileAttachmentExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
FileAttachmentExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: FileAttachmentExampleComponent, selector: "bb-file-attachment-example-ui", ngImport: i0, template: "<bb-file-attachment-ui\n  [name]=\"attachment.name\"\n  [size]=\"attachment.size\"\n  [fileContent]=\"attachment.fileContent\"\n></bb-file-attachment-ui>\n", components: [{ type: i1.FileAttachmentComponent, selector: "bb-file-attachment-ui", inputs: ["name", "size", "fileContent", "loading", "deletable", "disabled", "block"], outputs: ["delete"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FileAttachmentExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-file-attachment-example-ui',
                    templateUrl: './file-attachment.example.component.html',
                }]
        }] });
