// @backbase/ui-ang/esm2015/layout/bb-layout-nav-dropdown-directive/layout-nav-dropdown.directive.js
import { Directive, ContentChildren } from '@angular/core';
import { RouterLinkActive, NavigationEnd } from '@angular/router';
import { filter, first, switchMap } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
export class LayoutNavDropdownDirective {
    constructor(router, ngZone) {
        this.router = router;
        this.ngZone = ngZone;
        this.isActive = false;
    }
    ngAfterContentInit() {
        this.router.events
            .pipe(filter((event) => event instanceof NavigationEnd), switchMap((_) => this.ngZone.onStable.pipe(first())))
            .subscribe(() => {
            this.ngZone.run(() => {
                if (this.routerLinks) {
                    this.isActive = this.routerLinks.toArray().some((navLink) => navLink.isActive);
                }
            });
        });
    }
}
LayoutNavDropdownDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LayoutNavDropdownDirective, deps: [{ token: i1.Router }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Directive });
LayoutNavDropdownDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: LayoutNavDropdownDirective, selector: "[bbLayoutNavDropdown]", queries: [{ propertyName: "routerLinks", predicate: RouterLinkActive, descendants: true }], exportAs: ["bbLayoutNavDropdown"], ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LayoutNavDropdownDirective, decorators: [{
            type: Directive,
            args: [{ selector: '[bbLayoutNavDropdown]', exportAs: 'bbLayoutNavDropdown' }]
        }], ctorParameters: function () { return [{ type: i1.Router }, { type: i0.NgZone }]; }, propDecorators: { routerLinks: [{
                type: ContentChildren,
                args: [RouterLinkActive, { descendants: true }]
            }] } });
