// @backbase/product-summary-common-ang/esm2015/src/classes/responsive-container/responsive-container.component.js
import { Component, HostListener, Input } from '@angular/core';
import { SIZE_MOBILE_DEFAULT, SIZE_TABLET_DEFAULT } from './constants';
import * as i0 from "@angular/core";
export class ResponsiveContainerComponent {
    constructor(element, changeDetector) {
        /**
         * Specifies screen size: `mobile` | `tablet` | 'screen'
         *
         * @default 'screen'
         */
        this._responsiveView = 'screen';
        this._sizeTablet = SIZE_TABLET_DEFAULT;
        this._sizeMobile = SIZE_MOBILE_DEFAULT;
        this._element = element;
        this._changeDetector = changeDetector;
    }
    get responsiveView() {
        return this._customResponsiveView || this._responsiveView;
    }
    set responsiveView(value) {
        this._customResponsiveView = value;
    }
    get sizeMobile() {
        return this._sizeMobile;
    }
    set sizeMobile(value) {
        this._sizeMobile = value;
    }
    get sizeTablet() {
        return this._sizeTablet;
    }
    set sizeTablet(value) {
        this._sizeTablet = value;
    }
    /**
     * @internal
     */
    ngAfterViewInit() {
        if (!this._customResponsiveView) {
            this.checkElementSize();
            this._changeDetector.detectChanges();
        }
    }
    /**
     * Event handler on `windows:resize`
     * Checks size of the container
     */
    onResize() {
        if (!this._customResponsiveView) {
            this.checkElementSize();
        }
    }
    /**
     * Checks width of the parent container and updates `isSmallScreen$` and `isExtraSmallScreen$`
     */
    checkElementSize() {
        var _a, _b;
        const containerWidth = ((_b = (_a = this._element.nativeElement) === null || _a === void 0 ? void 0 : _a.firstElementChild) === null || _b === void 0 ? void 0 : _b.getBoundingClientRect().width) || 0;
        this._responsiveView = this.getResponsiveView(containerWidth);
    }
    getResponsiveView(containerWidth) {
        let responseView = 'screen';
        if (containerWidth <= this.sizeMobile) {
            responseView = 'mobile';
        }
        else if (containerWidth <= this.sizeTablet) {
            responseView = 'tablet';
        }
        return responseView;
    }
}
ResponsiveContainerComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ResponsiveContainerComponent, deps: [{ token: i0.ElementRef }, { token: i0.ChangeDetectorRef }], target: i0.ɵɵFactoryTarget.Component });
ResponsiveContainerComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: ResponsiveContainerComponent, selector: "ng-component", inputs: { responsiveView: "responsiveView" }, host: { listeners: { "window:resize": "onResize()" } }, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ResponsiveContainerComponent, decorators: [{
            type: Component,
            args: [{ template: '' }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.ChangeDetectorRef }]; }, propDecorators: { responsiveView: [{
                type: Input
            }], onResize: [{
                type: HostListener,
                args: ['window:resize', []]
            }] } });
