// @backbase/user-common/esm2015/src/components/input-base/input-base.js
import { Directive, Input } from '@angular/core';
import * as i0 from "@angular/core";
// eslint-disable-next-line @angular-eslint/directive-class-suffix
export class InputBase {
    constructor() {
        /**
         * A flag to determine if the form has been submitted.
         * Defaults to false.
         */
        this.hasBeenSubmitted = false;
        this.hasBlurred = false;
    }
    get fieldControl() {
        if (!this.formGroup) {
            throw new Error('formGroup must be initialised to type FormGroup');
        }
        if (!this.fieldName) {
            throw new Error('fieldName must be initialised to type FormFieldName');
        }
        return this.formGroup.controls[this.fieldName];
    }
    get hasErrors() {
        if (this.hasBeenSubmitted) {
            this.onFieldBlur();
        }
        return Boolean(this.fieldControl && this.fieldControl.touched && this.fieldControl.errors && this.hasBlurred);
    }
    onFieldBlur() {
        this.hasBlurred = true;
    }
}
InputBase.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: InputBase, deps: [], target: i0.ɵɵFactoryTarget.Directive });
InputBase.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.15", type: InputBase, selector: "[bbUserInputBase]", inputs: { formGroup: "formGroup", fieldName: "fieldName", hasBeenSubmitted: "hasBeenSubmitted" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: InputBase, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbUserInputBase]',
                }]
        }], propDecorators: { formGroup: [{
                type: Input
            }], fieldName: [{
                type: Input
            }], hasBeenSubmitted: [{
                type: Input
            }] } });
