// @backbase/ui-ang/esm2015/_example/modal/examples/modal.example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/modal";
import * as i2 from "@backbase/ui-ang/button";
export class ModalExampleComponent {
    constructor() {
        this.buttonColor = 'primary';
        this.showDialog = false;
    }
    openDialog() {
        this.showDialog = true;
    }
    closeDialog() {
        this.showDialog = false;
    }
    changeColor() {
        this.buttonColor =
            this.buttonColor === 'primary' ? 'secondary' : 'primary';
    }
    onConfirmDialog() {
        this.changeColor();
        this.closeDialog();
    }
}
ModalExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ModalExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ModalExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ModalExampleComponent, selector: "bb-modal-example-ui", ngImport: i0, template: "<div id=\"ariaDescribedById\">\n  Confirm Changing the color By Clicking on the button\n</div>\n<button\n  #button\n  bbButton\n  type=\"button\"\n  [color]=\"buttonColor\"\n  (click)=\"openDialog()\"\n  id=\"ariaLabelledByID\"\n>\n  Click here to change the color!\n</button>\n<bb-modal-ui\n  [isOpen]=\"showDialog\"\n  (cancel)=\"closeDialog()\"\n  (confirm)=\"onConfirmDialog()\"\n  [aria-describedby]=\"'ariaDescribedById'\"\n  [aria-labelledby]=\"'ariaLabelledByID'\"\n>\n  <bb-modal-header-ui\n    title=\"Confirm changing the color\"\n  ></bb-modal-header-ui>\n  <bb-modal-body-ui>\n    <ng-template bbCustomModalBody>\n      Are you sure you want to change the color?\n    </ng-template>\n  </bb-modal-body-ui>\n  <bb-modal-footer-ui confirmText=\"I Agree\" (confirm)=\"closeDialog()\">\n  </bb-modal-footer-ui>\n</bb-modal-ui>\n", components: [{ type: i1.ModalComponent, selector: "bb-modal-ui", inputs: ["isOpen", "aria-labelledby", "aria-describedby", "modalOptions"], outputs: ["confirm", "cancel", "isOpenChange"] }, { type: i1.ModalHeaderComponent, selector: "bb-modal-header-ui", inputs: ["title"], outputs: ["close"] }, { type: i1.ModalBodyComponent, selector: "bb-modal-body-ui" }, { type: i1.ModalFooterComponent, selector: "bb-modal-footer-ui", inputs: ["confirmText", "cancelText"], outputs: ["confirm", "cancel"] }], directives: [{ type: i2.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }, { type: i1.CustomModalBodyDirective, selector: "ng-template[bbCustomModalBody]" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ModalExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-modal-example-ui',
                    templateUrl: './modal.example.component.html',
                }]
        }] });
