// @backbase/ui-ang/esm2015/mode-header/mode-header.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ButtonModule } from '@backbase/ui-ang/button';
import { HeaderModule } from '@backbase/ui-ang/header';
import { IconModule } from '@backbase/ui-ang/icon';
import { ModeHeaderComponent } from './mode-header.component';
import * as i0 from "@angular/core";
export class ModeHeaderModule {
}
ModeHeaderModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ModeHeaderModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ModeHeaderModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ModeHeaderModule, declarations: [ModeHeaderComponent], imports: [CommonModule, ButtonModule, IconModule, HeaderModule], exports: [ModeHeaderComponent] });
ModeHeaderModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ModeHeaderModule, imports: [[CommonModule, ButtonModule, IconModule, HeaderModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ModeHeaderModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ButtonModule, IconModule, HeaderModule],
                    declarations: [ModeHeaderComponent],
                    exports: [ModeHeaderComponent],
                }]
        }] });
