// @backbase/approval-http-ang/esm2015/api/approvalTypes.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const ApprovalTypesHttpServiceGetAvailableApprovalTypesMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/approval-types",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "approvalTypes": [{
                            "id": "ed47a5d7-1626-4e8c-819c-8c1851cb0357",
                            "name": "A",
                            "description": "1st approval",
                            "rank": 100
                        }, {
                            "id": "5fa97eb5-277f-4b2c-b05d-f13d630e05c8",
                            "name": "B",
                            "description": "2nd approval",
                            "rank": 101
                        }, {
                            "id": "7b9459a3-03be-4b80-97da-53005bad9fb7",
                            "name": "C",
                            "description": "3rd approval",
                            "rank": 300
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
        ]
    }]);
export const ApprovalTypesHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v2/approval-types",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "approvalTypes": [{
                            "id": "ed47a5d7-1626-4e8c-819c-8c1851cb0357",
                            "name": "A",
                            "description": "1st approval",
                            "rank": 100
                        }, {
                            "id": "5fa97eb5-277f-4b2c-b05d-f13d630e05c8",
                            "name": "B",
                            "description": "2nd approval",
                            "rank": 101
                        }, {
                            "id": "7b9459a3-03be-4b80-97da-53005bad9fb7",
                            "name": "C",
                            "description": "3rd approval",
                            "rank": 300
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
        ]
    },
]);
