// @backbase/retail-loans-journey-ang/esm2015/lib/store/payment-schedule/store.service.js
import { Injectable } from '@angular/core';
import { Store } from '@ngrx/store';
import { selectAmortizedPayment, selectAmortizedPaymentCallState, selectAmortizedPayments, selectAmortizedPaymentsCallState, selectAmortizedPaymentsCurrentPage, selectAmortizedPaymentsTotalCount, } from './selectors';
import { fetchAmortizationPayment, fetchAmortizationSchedule } from './actions';
import * as i0 from "@angular/core";
import * as i1 from "@ngrx/store";
export class PaymentScheduleStoreService {
    constructor(store) {
        this.store = store;
        /**
         * Stream of api call state for fetching scheduled amortized payments
         */
        this.amortizedPaymentsCallState$ = this.store.select(selectAmortizedPaymentsCallState);
        /**
         * Stream of scheduled amortized payments
         */
        this.amortizedPayments$ = this.store.select(selectAmortizedPayments);
        /**
         * Stream of api call state for fetching scheduled amortized payment
         */
        this.amortizedPaymentCallState$ = this.store.select(selectAmortizedPaymentCallState);
        /**
         * Stream of scheduled amortized payment
         */
        this.amortizedPayment$ = this.store.select(selectAmortizedPayment);
        /**
         * Stream represents total pages count for scheduled amortized payments
         */
        this.amortizedPaymentsTotalCount$ = this.store.select(selectAmortizedPaymentsTotalCount);
        /**
         * Stream represents current loaded page for payment schedule table/list
         */
        this.amortizedPaymentsCurrentPage$ = this.store.select(selectAmortizedPaymentsCurrentPage);
    }
    /**
     * Fetch amortized payment schedule
     */
    fetchAmortizationSchedule(params) {
        this.store.dispatch(fetchAmortizationSchedule({ params }));
    }
    /**
     * Fetch amortized payment
     */
    fetchAmortizationPayment(params) {
        this.store.dispatch(fetchAmortizationPayment({ params }));
    }
}
/** @nocollapse */ PaymentScheduleStoreService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentScheduleStoreService, deps: [{ token: i1.Store }], target: i0.ɵɵFactoryTarget.Injectable });
/** @nocollapse */ PaymentScheduleStoreService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentScheduleStoreService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentScheduleStoreService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.Store }]; } });
