// @backbase/ui-ang/esm2015/progressbar/progressbar.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ProgressbarComponent } from './progressbar.component';
import { NgbProgressbarModule } from '@ng-bootstrap/ng-bootstrap';
import * as i0 from "@angular/core";
export class ProgressbarModule {
}
ProgressbarModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProgressbarModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProgressbarModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProgressbarModule, declarations: [ProgressbarComponent], imports: [CommonModule, NgbProgressbarModule], exports: [ProgressbarComponent] });
ProgressbarModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProgressbarModule, imports: [[CommonModule, NgbProgressbarModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProgressbarModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, NgbProgressbarModule],
                    declarations: [ProgressbarComponent],
                    exports: [ProgressbarComponent],
                }]
        }] });
