// @backbase/ui-ang/esm2015/progressbar/progressbar.component.js
import { Component, ChangeDetectionStrategy, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@ng-bootstrap/ng-bootstrap";
/**
 * @name ProgressbarComponent
 *
 * @description
 * Component that represents the progress of a task.
 */
export class ProgressbarComponent {
    constructor() {
        /**
         * Maximum value of the progressbar. Defaults to 100
         */
        this.max = 100;
        /**
         * A flag indicating if the current value should be displayed within the progressbar. Defaults to false
         */
        this.showValue = false;
        /**
         * A flag indicating if the progress bar should be striped. Defaults to false
         */
        this.striped = false;
        /**
         * A flag indicating if the stripes of the progress bar should be animated. Defaults to false
         */
        this.animated = false;
        /**
         * Current value of the progressbar. Defaults to 0 and should be less than max
         */
        this.value = 0;
    }
}
ProgressbarComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProgressbarComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ProgressbarComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ProgressbarComponent, selector: "bb-progressbar-ui", inputs: { height: "height", max: "max", showValue: "showValue", striped: "striped", animated: "animated", type: "type", value: "value" }, ngImport: i0, template: "<ngb-progressbar\n  class=\"bb-progress-bar\"\n  [type]=\"type\"\n  [max]=\"max\"\n  [value]=\"value\"\n  [showValue]=\"showValue\"\n  [striped]=\"striped\"\n  [animated]=\"animated\"\n  [height]=\"height\"\n>\n  <ng-content></ng-content>\n</ngb-progressbar>\n", components: [{ type: i1.NgbProgressbar, selector: "ngb-progressbar", inputs: ["value", "max", "animated", "striped", "textType", "type", "showValue", "height"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProgressbarComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-progressbar-ui',
                    templateUrl: './progressbar.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { height: [{
                type: Input
            }], max: [{
                type: Input
            }], showValue: [{
                type: Input
            }], striped: [{
                type: Input
            }], animated: [{
                type: Input
            }], type: [{
                type: Input
            }], value: [{
                type: Input
            }] } });
