// @backbase/loans-collection-ui-ang/esm2015/lib/loan-summary/loan-summary-progress-bar/loan-summary-progress-bar.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ProgressbarModule } from '@backbase/ui-ang/progressbar';
import { LoanSummaryProgressBarComponent } from './loan-summary-progress-bar.component';
import { LoanSummaryAmountItemModule } from '../loan-summary-amount-item/loan-summary-amount-item.module';
import * as i0 from "@angular/core";
export class LoanSummaryProgressBarModule {
}
/** @nocollapse */ LoanSummaryProgressBarModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryProgressBarModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanSummaryProgressBarModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryProgressBarModule, declarations: [LoanSummaryProgressBarComponent], imports: [CommonModule, LoanSummaryAmountItemModule, ProgressbarModule], exports: [LoanSummaryProgressBarComponent] });
/** @nocollapse */ LoanSummaryProgressBarModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryProgressBarModule, imports: [[CommonModule, LoanSummaryAmountItemModule, ProgressbarModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryProgressBarModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanSummaryProgressBarComponent],
                    imports: [CommonModule, LoanSummaryAmountItemModule, ProgressbarModule],
                    exports: [LoanSummaryProgressBarComponent],
                }]
        }] });
