// @backbase/ui-ang/esm2015/_example/focus/showcase/focus-showcase.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/button";
import * as i2 from "@backbase/ui-ang/focus";
let FocusShowcaseComponent = class FocusShowcaseComponent {
};
FocusShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FocusShowcaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
FocusShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: FocusShowcaseComponent, selector: "bb-focus-ui-showcase", ngImport: i0, template: "<div class=\"bb-block bb-block--xl\">\n  <h4>Automatically setting focus</h4>\n  <p>Are you sure you want to delete something?</p>\n  <div class=\"bb-stack\">\n    <div class=\"bb-stack__item\">\n      <button bbButton bbFocus>Cancel</button>\n    </div>\n    <button bbButton>OK</button>\n  </div>\n</div>\n\n<div clsas=\"bb-block bb-block--xl\">\n  <h4>Manually setting focus</h4>\n  <div class=\"bb-stack\">\n    <div class=\"bb-stack__item\">\n      <button bbButton [bbFocus]=\"\" #focusThis=\"bbFocus\">Button with bbFocus</button>\n    </div>\n    <button bbButton (click)=\"focusThis.focus()\">Click to focus the other button</button>\n  </div>\n</div>\n", directives: [{ type: i1.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }, { type: i2.FocusDirective, selector: "[bbFocus]", inputs: ["bbFocus"], exportAs: ["bbFocus"] }] });
FocusShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
], FocusShowcaseComponent);
export { FocusShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FocusShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-focus-ui-showcase',
                    templateUrl: './focus-showcase.component.html',
                }]
        }] });
