// @backbase/ui-ang/esm2015/_example/focus/showcase/focus-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FocusShowcaseComponent } from './focus-showcase.component';
import { FocusModule } from '@backbase/ui-ang/focus';
import { ButtonModule } from '@backbase/ui-ang/button';
import * as i0 from "@angular/core";
export class FocusShowcaseModule {
}
FocusShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FocusShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
FocusShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FocusShowcaseModule, declarations: [FocusShowcaseComponent], imports: [CommonModule, FocusModule, ButtonModule] });
FocusShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FocusShowcaseModule, imports: [[CommonModule, FocusModule, ButtonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FocusShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, FocusModule, ButtonModule],
                    declarations: [FocusShowcaseComponent],
                    entryComponents: [FocusShowcaseComponent],
                }]
        }] });
