// @backbase/ui-ang/esm2015/payment-card/payment-card.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PaymentCardComponent } from './payment-card.component';
import { PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import { LogoModule } from '@backbase/ui-ang/logo';
import { CardVendorModule } from '@backbase/ui-ang/card-vendor';
import * as i0 from "@angular/core";
export class PaymentCardModule {
}
PaymentCardModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PaymentCardModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardModule, declarations: [PaymentCardComponent], imports: [CommonModule, LogoModule, CardVendorModule, PaymentCardNumberModule], exports: [PaymentCardComponent] });
PaymentCardModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardModule, imports: [[CommonModule, LogoModule, CardVendorModule, PaymentCardNumberModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, LogoModule, CardVendorModule, PaymentCardNumberModule],
                    declarations: [PaymentCardComponent],
                    exports: [PaymentCardComponent],
                }]
        }] });
