// @backbase/product-summary-ui/esm2015/src/bb-product-summary-loading/bb-product-summary-loading.component.js
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/loading-indicator";
import * as i2 from "@angular/common";
/**
 * Displays a loading indicator while favourite accounts are being loaded
 *
 * @usageNotes
 *
 * ### Display details of an account
 *
 * ```html
 * <bb-product-summary-loading></bb-product-summary-loading>
 * ```
 *
 * @ngModule ProductSummaryUiModule
 */
export class ProductSummaryLoadingComponent {
    constructor() {
        /**
         * Flag indicating if list is requesting new data.
         *
         * @default false
         */
        this.isLoading = false;
        /**
         * Text to be displayed during loading.
         *
         * @default ''.
         */
        this.text = '';
    }
}
ProductSummaryLoadingComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryLoadingComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ProductSummaryLoadingComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ProductSummaryLoadingComponent, selector: "bb-product-summary-loading", inputs: { isLoading: "isLoading", text: "text", classes: "classes" }, ngImport: i0, template: "<ng-container *ngIf=\"!isLoading; else loading\">\n  <ng-content></ng-content>\n</ng-container>\n\n<ng-template #loading>\n  <div [ngClass]=\"classes\">\n    <bb-loading-indicator-ui [text]=\"text\"></bb-loading-indicator-ui>\n  </div>\n</ng-template>\n", components: [{ type: i1.LoadingIndicatorComponent, selector: "bb-loading-indicator-ui", inputs: ["text", "loaderSize", "showDelay", "hasBackground", "inline"] }], directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i2.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryLoadingComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-product-summary-loading',
                    templateUrl: 'bb-product-summary-loading.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { isLoading: [{
                type: Input
            }], text: [{
                type: Input
            }], classes: [{
                type: Input
            }] } });
