// @backbase/product-summary-ui/esm2015/src/bb-product-summary-loading/bb-product-summary-loading.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { ProductSummaryLoadingComponent } from './bb-product-summary-loading.component';
import * as i0 from "@angular/core";
export class ProductSummaryLoadingModule {
}
ProductSummaryLoadingModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryLoadingModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductSummaryLoadingModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryLoadingModule, declarations: [ProductSummaryLoadingComponent], imports: [CommonModule, LoadingIndicatorModule], exports: [ProductSummaryLoadingComponent] });
ProductSummaryLoadingModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryLoadingModule, imports: [[CommonModule, LoadingIndicatorModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryLoadingModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, LoadingIndicatorModule],
                    declarations: [ProductSummaryLoadingComponent],
                    exports: [ProductSummaryLoadingComponent],
                }]
        }] });
