// @backbase/positive-pay-check-v1-client-ang/esm2015/api/positivePayChecks.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const PositivePayChecksHttpServiceDeleteCheckMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/checks/{id}",
        method: "DELETE",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const PositivePayChecksHttpServiceFilterChecksMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/checks/filter",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "checks": [{
                            "id": "849d66ae-c942-2ba7-8739-c5a38a596f31",
                            "arrangementId": "fe9d66ae-b927-4ac7-8799-c5a38a596ff2",
                            "payeeName": "Sarah Lee",
                            "amountDetails": {
                                "currencyCode": "USD",
                                "amount": "99.99"
                            },
                            "checkNumber": "111456789123456789123456789111",
                            "issueDate": "2020-11-01",
                            "voidCheck": false,
                            "registrationDate": "2005-10-21T11:50:23.000-05:00",
                            "issueStatus": "SUBMITTED"
                        }, {
                            "id": "849d66ae-c942-2ba7-8739-c5a38a596f32",
                            "arrangementId": "aa9d66ae-b927-4ac7-8799-c5a38a53ed24",
                            "payeeName": " ",
                            "amountDetails": {
                                "currencyCode": "USD",
                                "amount": "150.54"
                            },
                            "checkNumber": "333456789123456789123456789333",
                            "issueDate": "2020-11-01",
                            "voidCheck": true,
                            "registrationDate": "2009-09-20T15:42:20.000-05:00",
                            "issueStatus": "SUBMITTED"
                        }, {
                            "id": "849d66ae-c942-2ba7-8739-c5a38a596f33",
                            "arrangementId": "c3d62af-3r27-4ac7-8799-c5a38a596asf2",
                            "payeeName": "Robin Green",
                            "amountDetails": {
                                "currencyCode": "CAD",
                                "amount": "10.89"
                            },
                            "checkNumber": "444456789123456789123456789444",
                            "issueDate": "2020-11-01",
                            "voidCheck": false,
                            "issueStatus": "SAVED"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const PositivePayChecksHttpServiceFilterExceptionsMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/exceptions/filter",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "exceptions": [{
                            "id": "849d66ae-c942-2ba7-8739-c5a38a596f32",
                            "checkNumber": "111456789123456789123456789111",
                            "accountNumber": "12312312312321213",
                            "issueDate": "2020-11-01",
                            "payeeName": "Robin Green",
                            "amountDetails": {
                                "currencyCode": "USD",
                                "amount": "150.54"
                            },
                            "errors": ["Data mismatch", "Duplicate check"],
                            "displayName": "Current account"
                        }, {
                            "id": "849d66ae-c942-2ba7-8739-c5a38a596f44",
                            "checkNumber": "111456789123456789123456789222",
                            "accountNumber": "12312312312324444",
                            "issueDate": "2021-01-05",
                            "payeeName": "Robert Plant",
                            "amountDetails": {
                                "currencyCode": "USD",
                                "amount": "120.00"
                            },
                            "errors": ["Already paid"],
                            "displayName": "Saving account"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const PositivePayChecksHttpServiceGetCountsMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/counts",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "exceptionCount": 250
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const PositivePayChecksHttpServiceGetExceptionDetailsMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/exceptions/{id}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "exceptionCheck": {
                        "id": "849d66ae-c942-2ba7-8739-c5a38a596f32",
                        "checkNumber": "111456789123456789123456789111",
                        "accountNumber": "12312312312321213",
                        "issueDate": "2020-11-01",
                        "payeeName": "Robin Green",
                        "amountDetails": {
                            "currencyCode": "USD",
                            "amount": "150.54"
                        },
                        "errors": ["Data mismatch", "Duplicate check"],
                        "checkDepositDate": "2020-12-01T10:05:28Z",
                        "exceptionDate": "2020-12-01T11:00:00Z",
                        "displayName": "Payroll account for new hires"
                    },
                    "issuedCheck": {
                        "id": "849d66ae-c942-2ba7-8739-c5a38a596f31",
                        "arrangementId": "fe9d66ae-b927-4ac7-8799-c5a38a596ff2",
                        "payeeName": "Sarah Lee",
                        "amountDetails": {
                            "currencyCode": "USD",
                            "amount": "99.99"
                        },
                        "checkNumber": "111456789123456789123456789111",
                        "issueDate": "2020-11-01",
                        "voidCheck": false,
                        "registrationDate": "2020-12-31T01:30:00.000-05:00",
                        "issueStatus": "SAVED",
                        "displayName": "Payroll account for new hires",
                        "userFullName": "Jimmy Page"
                    }
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const PositivePayChecksHttpServiceGetSettingsMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/settings",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "settings": [{
                            "subscription": "checks-positive-pay-with-payee-match",
                            "systemDecision": "Reject",
                            "additions": {
                                "costs-to-update": "$1.50 ex. taxes"
                            }
                        }, {
                            "subscription": "checks-positive-pay-without-payee-match",
                            "systemDecision": "Pay"
                        }],
                    "additions": {
                        "settingId": "1"
                    }
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const PositivePayChecksHttpServiceProcessExceptionDecisionMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/exceptions/{id}",
        method: "PATCH",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const PositivePayChecksHttpServiceStartChecksFileUploadMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/uploads/checks",
        method: "POST",
        responses: [
            {
                status: 201,
                body: {
                    "uploadId": "10"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const PositivePayChecksHttpServiceSubmitCheckMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/checks",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "id": "849d66ae-c942-2ba7-8739-c5a38a596f31",
                    "arrangementId": "fe9d66ae-b927-4ac7-8799-c5a38a596ff2",
                    "payeeName": "Sarah Lee",
                    "amountDetails": {
                        "currencyCode": "USD",
                        "amount": "99.99"
                    },
                    "checkNumber": "111456789123456789123456789111",
                    "issueDate": "2020-11-01",
                    "voidCheck": false,
                    "registrationDate": "2020-12-31T01:30:00.000",
                    "issueStatus": "SAVED"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const PositivePayChecksHttpServiceUpdateSettingsMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/settings",
        method: "PATCH",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const PositivePayChecksHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v1/checks/{id}",
        method: "DELETE",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/checks/filter",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "checks": [{
                            "id": "849d66ae-c942-2ba7-8739-c5a38a596f31",
                            "arrangementId": "fe9d66ae-b927-4ac7-8799-c5a38a596ff2",
                            "payeeName": "Sarah Lee",
                            "amountDetails": {
                                "currencyCode": "USD",
                                "amount": "99.99"
                            },
                            "checkNumber": "111456789123456789123456789111",
                            "issueDate": "2020-11-01",
                            "voidCheck": false,
                            "registrationDate": "2005-10-21T11:50:23.000-05:00",
                            "issueStatus": "SUBMITTED"
                        }, {
                            "id": "849d66ae-c942-2ba7-8739-c5a38a596f32",
                            "arrangementId": "aa9d66ae-b927-4ac7-8799-c5a38a53ed24",
                            "payeeName": " ",
                            "amountDetails": {
                                "currencyCode": "USD",
                                "amount": "150.54"
                            },
                            "checkNumber": "333456789123456789123456789333",
                            "issueDate": "2020-11-01",
                            "voidCheck": true,
                            "registrationDate": "2009-09-20T15:42:20.000-05:00",
                            "issueStatus": "SUBMITTED"
                        }, {
                            "id": "849d66ae-c942-2ba7-8739-c5a38a596f33",
                            "arrangementId": "c3d62af-3r27-4ac7-8799-c5a38a596asf2",
                            "payeeName": "Robin Green",
                            "amountDetails": {
                                "currencyCode": "CAD",
                                "amount": "10.89"
                            },
                            "checkNumber": "444456789123456789123456789444",
                            "issueDate": "2020-11-01",
                            "voidCheck": false,
                            "issueStatus": "SAVED"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/exceptions/filter",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "exceptions": [{
                            "id": "849d66ae-c942-2ba7-8739-c5a38a596f32",
                            "checkNumber": "111456789123456789123456789111",
                            "accountNumber": "12312312312321213",
                            "issueDate": "2020-11-01",
                            "payeeName": "Robin Green",
                            "amountDetails": {
                                "currencyCode": "USD",
                                "amount": "150.54"
                            },
                            "errors": ["Data mismatch", "Duplicate check"],
                            "displayName": "Current account"
                        }, {
                            "id": "849d66ae-c942-2ba7-8739-c5a38a596f44",
                            "checkNumber": "111456789123456789123456789222",
                            "accountNumber": "12312312312324444",
                            "issueDate": "2021-01-05",
                            "payeeName": "Robert Plant",
                            "amountDetails": {
                                "currencyCode": "USD",
                                "amount": "120.00"
                            },
                            "errors": ["Already paid"],
                            "displayName": "Saving account"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/counts",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "exceptionCount": 250
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/exceptions/{id}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "exceptionCheck": {
                        "id": "849d66ae-c942-2ba7-8739-c5a38a596f32",
                        "checkNumber": "111456789123456789123456789111",
                        "accountNumber": "12312312312321213",
                        "issueDate": "2020-11-01",
                        "payeeName": "Robin Green",
                        "amountDetails": {
                            "currencyCode": "USD",
                            "amount": "150.54"
                        },
                        "errors": ["Data mismatch", "Duplicate check"],
                        "checkDepositDate": "2020-12-01T10:05:28Z",
                        "exceptionDate": "2020-12-01T11:00:00Z",
                        "displayName": "Payroll account for new hires"
                    },
                    "issuedCheck": {
                        "id": "849d66ae-c942-2ba7-8739-c5a38a596f31",
                        "arrangementId": "fe9d66ae-b927-4ac7-8799-c5a38a596ff2",
                        "payeeName": "Sarah Lee",
                        "amountDetails": {
                            "currencyCode": "USD",
                            "amount": "99.99"
                        },
                        "checkNumber": "111456789123456789123456789111",
                        "issueDate": "2020-11-01",
                        "voidCheck": false,
                        "registrationDate": "2020-12-31T01:30:00.000-05:00",
                        "issueStatus": "SAVED",
                        "displayName": "Payroll account for new hires",
                        "userFullName": "Jimmy Page"
                    }
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/settings",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "settings": [{
                            "subscription": "checks-positive-pay-with-payee-match",
                            "systemDecision": "Reject",
                            "additions": {
                                "costs-to-update": "$1.50 ex. taxes"
                            }
                        }, {
                            "subscription": "checks-positive-pay-without-payee-match",
                            "systemDecision": "Pay"
                        }],
                    "additions": {
                        "settingId": "1"
                    }
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/exceptions/{id}",
        method: "PATCH",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/uploads/checks",
        method: "POST",
        responses: [
            {
                status: 201,
                body: {
                    "uploadId": "10"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/checks",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "id": "849d66ae-c942-2ba7-8739-c5a38a596f31",
                    "arrangementId": "fe9d66ae-b927-4ac7-8799-c5a38a596ff2",
                    "payeeName": "Sarah Lee",
                    "amountDetails": {
                        "currencyCode": "USD",
                        "amount": "99.99"
                    },
                    "checkNumber": "111456789123456789123456789111",
                    "issueDate": "2020-11-01",
                    "voidCheck": false,
                    "registrationDate": "2020-12-31T01:30:00.000",
                    "issueStatus": "SAVED"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/settings",
        method: "PATCH",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
]);
