// @backbase/ui-ang/esm2015/_example/switch/showcase/switch-showcase.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@angular/forms";
import * as i2 from "@backbase/ui-ang/switch";
let SwitchShowcaseComponent = class SwitchShowcaseComponent {
    constructor(formBuilder) {
        this.formBuilder = formBuilder;
        this.value = true;
        this.vForm = this.formBuilder.group({
            defaultSwitch1: false,
            defaultSwitch2: false,
            defaultSwitch3: false,
            defaultSwitch4: true,
            defaultSwitch5: { value: false, disabled: true },
            defaultSwitch6: { value: true, disabled: true },
        });
    }
};
SwitchShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SwitchShowcaseComponent, deps: [{ token: i1.FormBuilder }], target: i0.ɵɵFactoryTarget.Component });
SwitchShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: SwitchShowcaseComponent, selector: "bb-switch-ui-showcase", ngImport: i0, template: "<form [formGroup]=\"vForm\">\n  <div class=\"bb-block bb-block--md\">\n    <bb-switch-ui [id]=\"'defaultSwitch4'\" aria-label=\"Switch without visible label\" formControlName=\"defaultSwitch4\">\n    </bb-switch-ui>\n  </div>\n  <div class=\"bb-block bb-block--md\">\n    <bb-switch-ui [id]=\"'defaultSwitch5'\" aria-label=\"Disabled switch\" formControlName=\"defaultSwitch5\"> </bb-switch-ui>\n  </div>\n  <div class=\"bb-block bb-block--md\">\n    <bb-switch-ui\n      [labelPosition]=\"'right'\"\n      [label]=\"'Label'\"\n      [id]=\"'defaultSwitch3'\"\n      formControlName=\"defaultSwitch3\"\n      aria-label=\"Switch with visible label (on the right)\"\n    >\n    </bb-switch-ui>\n  </div>\n  <div>\n    <bb-switch-ui\n      [label]=\"'Label'\"\n      [id]=\"'defaultSwitch2'\"\n      formControlName=\"defaultSwitch2\"\n      aria-label=\"Switch with visible label (on the left)\"\n    >\n    </bb-switch-ui>\n  </div>\n</form>\n", components: [{ type: i2.SwitchComponent, selector: "bb-switch-ui", inputs: ["labelPosition"] }], directives: [{ type: i1.ɵNgNoValidate, selector: "form:not([ngNoForm]):not([ngNativeValidate])" }, { type: i1.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { type: i1.FormGroupDirective, selector: "[formGroup]", inputs: ["formGroup"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { type: i1.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { type: i1.FormControlName, selector: "[formControlName]", inputs: ["disabled", "formControlName", "ngModel"], outputs: ["ngModelChange"] }] });
SwitchShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
], SwitchShowcaseComponent);
export { SwitchShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SwitchShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-switch-ui-showcase',
                    templateUrl: './switch-showcase.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.FormBuilder }]; } });
