// @backbase/ui-ang/esm2015/_example/switch/showcase/switch-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SwitchShowcaseComponent } from './switch-showcase.component';
import { SwitchModule } from '@backbase/ui-ang/switch';
import { ReactiveFormsModule } from '@angular/forms';
import * as i0 from "@angular/core";
export class SwitchShowcaseModule {
}
SwitchShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SwitchShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
SwitchShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SwitchShowcaseModule, declarations: [SwitchShowcaseComponent], imports: [CommonModule, SwitchModule, ReactiveFormsModule] });
SwitchShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SwitchShowcaseModule, imports: [[CommonModule, SwitchModule, ReactiveFormsModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SwitchShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, SwitchModule, ReactiveFormsModule],
                    declarations: [SwitchShowcaseComponent],
                    entryComponents: [SwitchShowcaseComponent],
                }]
        }] });
