// @backbase/ui-ang/esm2015/_example/input-datepicker/showcase/input-datepicker-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { InputDatepickerShowcaseComponent } from './input-datepicker-showcase.component';
import { InputDatepickerModule } from '@backbase/ui-ang/input-datepicker';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import * as i0 from "@angular/core";
export class InputDatepickerShowcaseModule {
}
InputDatepickerShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputDatepickerShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputDatepickerShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputDatepickerShowcaseModule, declarations: [InputDatepickerShowcaseComponent], imports: [CommonModule, InputDatepickerModule, ReactiveFormsModule, FormsModule] });
InputDatepickerShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputDatepickerShowcaseModule, imports: [[CommonModule, InputDatepickerModule, ReactiveFormsModule, FormsModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputDatepickerShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, InputDatepickerModule, ReactiveFormsModule, FormsModule],
                    declarations: [InputDatepickerShowcaseComponent],
                    entryComponents: [InputDatepickerShowcaseComponent],
                }]
        }] });
