// @backbase/contact-common-ang/esm2015/lib/directives/iban-validator/iban-validator.directive.js
import { Directive, Inject } from '@angular/core';
import { NG_VALIDATORS } from '@angular/forms';
import { isValidIBANFormControl } from './index';
import { BB_CONTACT_IBAN_COUNTRIES_LIST } from '../../config.providers';
import { addIBANCountries } from './countrySpec';
import * as i0 from "@angular/core";
/**
 * This is a directive to validate iban
 *
 * @usageNotes
 *
 * ### validates the iban
 *
 * ```html
 * <input type="text" #ibanControl="ngModel" [(ngModel)]="iban" bbIbanValidatorUi />
 * ```
 * @ngModule IbanValidatorModule
 */
export class IbanValidatorDirective {
    constructor(ibanCountryList) {
        this.ibanCountryList = ibanCountryList;
        addIBANCountries(this.ibanCountryList);
    }
    validate(control) {
        return control && control.value ? isValidIBANFormControl(control) : null;
    }
}
/** @nocollapse */ IbanValidatorDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IbanValidatorDirective, deps: [{ token: BB_CONTACT_IBAN_COUNTRIES_LIST }], target: i0.ɵɵFactoryTarget.Directive });
/** @nocollapse */ IbanValidatorDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: IbanValidatorDirective, selector: "[bbIbanValidator]", providers: [{ provide: NG_VALIDATORS, useExisting: IbanValidatorDirective, multi: true }], ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IbanValidatorDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbIbanValidator]',
                    providers: [{ provide: NG_VALIDATORS, useExisting: IbanValidatorDirective, multi: true }],
                }]
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [BB_CONTACT_IBAN_COUNTRIES_LIST]
                }] }]; } });
