// @backbase/payment-orders-ang/esm2015/lib/ui/components/bb-schedule-label-ui/schedule-label.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ScheduleLabelComponent } from './schedule-label.component';
import * as i0 from "@angular/core";
export class ScheduleLabelModule {
}
/** @nocollapse */ ScheduleLabelModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ScheduleLabelModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ ScheduleLabelModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ScheduleLabelModule, declarations: [ScheduleLabelComponent], imports: [CommonModule], exports: [ScheduleLabelComponent] });
/** @nocollapse */ ScheduleLabelModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ScheduleLabelModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ScheduleLabelModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ScheduleLabelComponent],
                    imports: [CommonModule],
                    exports: [ScheduleLabelComponent],
                }]
        }] });
