// @backbase/ui-ang/esm2015/date-pipe/date-pipe.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { BbDatePipe } from './date-pipe.pipe';
import * as i0 from "@angular/core";
export class BbDatePipeModule {
}
BbDatePipeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BbDatePipeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
BbDatePipeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BbDatePipeModule, declarations: [BbDatePipe], imports: [CommonModule], exports: [BbDatePipe] });
BbDatePipeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BbDatePipeModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BbDatePipeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [BbDatePipe],
                    exports: [BbDatePipe],
                }]
        }] });
