// @backbase/stop-checks-journey-ang/esm2015/lib/stop-checks-journey.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { PayordStopChecksListWidgetAngModule, StopChecksListCommunicationService, } from '@backbase/payord-stop-checks-list-widget-ang';
import { StopChecksJourneyComponent } from './stop-checks-journey.component';
import { provideRoutes, RouterModule } from '@angular/router';
import { StopChecksListComponent } from './components/stop-checks-list/stop-checks-list.component';
import { StopChecksJourneyService } from './services/stop-checks-journey-config.service';
import { StopChecksCreateModalComponent } from './components/stop-checks-create-modal/stop-checks-create-modal.component';
import { PayordStopChecksWidgetAngModule } from '@backbase/payord-stop-checks-widget-ang';
import { StopChecksCreateFormComponent } from './components/stop-checks-create-form/stop-checks-create-form.component';
import { CommunicationListService } from './services/comunication-list.service';
import { StopChecksReviewFormComponent } from './components/stop-checks-review-form/stop-checks-review-form.component';
import { ModalModule } from '@backbase/ui-ang/modal';
import { HeaderModule } from '@backbase/ui-ang/header';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
export const StopChecksDefaultRoute = {
    path: '',
    component: StopChecksJourneyComponent,
    children: [
        {
            path: '',
            redirectTo: 'list',
            pathMatch: 'full',
        },
        {
            path: 'list',
            component: StopChecksListComponent,
            children: [
                {
                    path: 'form',
                    component: StopChecksCreateModalComponent,
                    children: [
                        {
                            path: '',
                            pathMatch: 'full',
                            component: StopChecksCreateFormComponent,
                        },
                        { path: 'review', component: StopChecksReviewFormComponent },
                    ],
                },
            ],
        },
    ],
};
/**
 * @deprecated Please use StopChecksDefaultRoute instead
 */
export const stopChecksRoutes = StopChecksDefaultRoute;
export class StopChecksJourneyModule {
    static forRoot(data = {
        route: StopChecksDefaultRoute,
    }) {
        return {
            ngModule: StopChecksJourneyModule,
            providers: [provideRoutes([data.route])],
        };
    }
}
/** @nocollapse */ StopChecksJourneyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StopChecksJourneyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ StopChecksJourneyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StopChecksJourneyModule, declarations: [StopChecksJourneyComponent,
        StopChecksListComponent,
        StopChecksCreateModalComponent,
        StopChecksCreateFormComponent,
        StopChecksReviewFormComponent], imports: [CommonModule,
        ModalModule, i1.BackbaseCoreModule, RouterModule,
        HeaderModule,
        PayordStopChecksListWidgetAngModule,
        PayordStopChecksWidgetAngModule,
        ModalModule] });
/** @nocollapse */ StopChecksJourneyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StopChecksJourneyModule, providers: [
        StopChecksJourneyService,
        {
            provide: StopChecksListCommunicationService,
            useClass: CommunicationListService,
        },
    ], imports: [[
            CommonModule,
            ModalModule,
            BackbaseCoreModule.withConfig({
                classMap: {
                    StopChecksJourneyComponent,
                },
            }),
            RouterModule,
            HeaderModule,
            PayordStopChecksListWidgetAngModule,
            PayordStopChecksWidgetAngModule,
            ModalModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StopChecksJourneyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        StopChecksJourneyComponent,
                        StopChecksListComponent,
                        StopChecksCreateModalComponent,
                        StopChecksCreateFormComponent,
                        StopChecksReviewFormComponent,
                    ],
                    imports: [
                        CommonModule,
                        ModalModule,
                        BackbaseCoreModule.withConfig({
                            classMap: {
                                StopChecksJourneyComponent,
                            },
                        }),
                        RouterModule,
                        HeaderModule,
                        PayordStopChecksListWidgetAngModule,
                        PayordStopChecksWidgetAngModule,
                        ModalModule,
                    ],
                    providers: [
                        StopChecksJourneyService,
                        {
                            provide: StopChecksListCommunicationService,
                            useClass: CommunicationListService,
                        },
                    ],
                }]
        }] });
