// @backbase/product-summary-details-widget-ang/esm2015/src/account-error.js
export class AccountAccessDenied {
    constructor(response) {
        this.response = response;
        this.kind = 'AccountAccessDenied';
        this.messages = ((response.error || {}).errors || []).map((e) => e.message);
    }
}
export class AccountBadRequest {
    constructor(response) {
        this.response = response;
        this.kind = 'ProductKindsBadRequest';
        this.messages = ((response.error || {}).errors || []).map((e) => e.message);
    }
}
export class AccounttNotFound {
    constructor(response) {
        this.response = response;
        this.kind = 'ProductKindstNotFound';
        this.messages = ((response.error || {}).errors || []).map((e) => e.message);
    }
}
export class AccountConnectivityError {
    constructor(response) {
        this.response = response;
        this.kind = 'ProductKindsConnectivityError';
        this.messages = ((response.error || {}).errors || []).map((e) => e.message);
    }
}
export class AccountUnknownError {
    constructor(response) {
        this.response = response;
        this.kind = 'ProductKindsUnknownError';
        this.messages = [response.message];
    }
}
export const parseError = (error) => {
    if (error.error instanceof ErrorEvent) {
        return new AccountUnknownError(error);
    }
    else {
        switch (error.status) {
            case 0:
                return new AccountConnectivityError(error);
            case 400:
                return new AccountBadRequest(error);
            case 401:
                return new AccountAccessDenied(error);
            case 403:
                return new AccountAccessDenied(error);
            case 404:
                return new AccounttNotFound(error);
            default:
                return new AccountUnknownError(error);
        }
    }
};
