// @backbase/product-summary-details-widget-ang/esm2015/src/account.service.js
import { Injectable } from '@angular/core';
import { ReplaySubject, BehaviorSubject, of } from 'rxjs';
import { distinctUntilChanged, catchError, switchMap, filter, tap } from 'rxjs/operators';
import { isEqual } from '@backbase/product-summary-common-ang';
import { parseError } from './account-error';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/product-summary-common-ang";
/**
 * Service for fetching and storing product summary account details.
 *
 * This service relies on providers from `ProductSummaryDetailsWidgetModule`.
 *
 * @see ProductSummaryDetailsWidgetModule
 *
 * @usageNotes
 *
 * ### Ensure the ProductSummaryDetailsWidgetModule is imported to your component module
 *
 * ```ts
 * @NgModule({
 *   ...
 *   imports: [
 *     ...
 *     ProductSummaryDetailsWidgetModule,
 *   ],
 * })
 * export class MyWidgetModule {}
 * ```
 *
 * ### Inject this service into your component
 *
 * ```ts
 * @Component({
 *   ...
 *   providers: [AccountService],
 * })
 *  export class MyComponent {
 * ```
 */
export class AccountService {
    constructor(productSummaryAccountService) {
        this.productSummaryAccountService = productSummaryAccountService;
        this.arrangementId = new ReplaySubject(1);
        /**
         * If the fetching of the account details failed, this property
         * holds the errors coming from the server
         */
        this.error = new BehaviorSubject(undefined);
        /**
         * Defines the loading state during the loading of data.
         */
        this.isLoading = new BehaviorSubject(false);
        /**
         * Fetches from the server and returns product details for `arrangementId`
         */
        this.productDetailsById = this.arrangementId.pipe(distinctUntilChanged((a, b) => isEqual(a, b) && !this.error.value), filter(arrangementId => !!arrangementId), tap(() => this.error.next(undefined)), tap(() => this.isLoading.next(true)), switchMap((id) => this.productSummaryAccountService.getAccountById(id).pipe(catchError((error) => {
            this.error.next(parseError(error));
            return of(undefined);
        }))), tap(() => this.isLoading.next(false)));
    }
    /**
     * Subscribes on the passed `arrangementId`
     * Causes refetching data
     *
     * @param arrangementId
     */
    loadProductDetailsFrom(arrangementId) {
        arrangementId.subscribe(this.arrangementId);
    }
    /**
     * Fetches an unmasked attribute from the server for the given attribute name
     *
     * @param requestParameters
     */
    getUnmaskedAttribute(requestParameters) {
        return this.productSummaryAccountService.getUnmaskedAttribute(requestParameters);
    }
}
AccountService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountService, deps: [{ token: i1.ProductSummaryAccountsService }], target: i0.ɵɵFactoryTarget.Injectable });
AccountService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.ProductSummaryAccountsService }]; } });
