// @backbase/product-summary-details-widget-ang/esm2015/src/product-summary-details-widget.component.js
import { Component } from '@angular/core';
import { ProductSummaryAccountsService } from '@backbase/product-summary-common-ang';
import { AccountService } from './account.service';
import * as i0 from "@angular/core";
import * as i1 from "./product-summary-details/product-summary-details.component";
/**
 * A widget for displaying account details
 *
 * @usageNotes
 *
 * ### Include this widget in your `AppModule`:
 *
 * ```typescript
 * @NgModule({
 *   imports: [
 *     ProductSummaryDetailsWidgetModule,
 *   ],
 *   ...
 * })
 * export class AppModule {}
 * ```
 *
 * ### Include this widget (standalone model)
 *
 * ```typescript
 * export const environment: Environment = {
 *   ...
 *   bootstrap: {
 *     pageModel: {
 *       name: 'root',
 *       properties: {},
 *       children: [
 *         {
 *           name: 'product-summary-account-details-widget-ang',
 *           properties: {
 *              classId: 'ProductSummaryDetailsWidgetComponent',
 *           },
 *         },
 *       ],
 *     },
 *   },
 * };
 * ```
 */
export class ProductSummaryDetailsWidgetComponent {
    constructor() {
        this.hostRef = this;
    }
}
ProductSummaryDetailsWidgetComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryDetailsWidgetComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ProductSummaryDetailsWidgetComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ProductSummaryDetailsWidgetComponent, selector: "bb-product-summary-details-widget", providers: [AccountService, ProductSummaryAccountsService], ngImport: i0, template: ` <bb-product-summary-details></bb-product-summary-details> `, isInline: true, components: [{ type: i1.ProductSummaryDetailsComponent, selector: "bb-product-summary-details" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryDetailsWidgetComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-product-summary-details-widget',
                    template: ` <bb-product-summary-details></bb-product-summary-details> `,
                    providers: [AccountService, ProductSummaryAccountsService],
                }]
        }] });
