// @backbase/product-summary-details-widget-ang/esm2015/src/product-summary-details-widget.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { ButtonModule } from '@backbase/ui-ang/button';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { AmountModule } from '@backbase/ui-ang/amount';
import { IconModule } from '@backbase/ui-ang/icon';
import { PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import { PaymentCardModule } from '@backbase/ui-ang/payment-card';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { NotificationModule } from '@backbase/ui-ang/notification';
import { ArrangementStateIndicatorModule, ProductNumberModule } from '@backbase/product-summary-ui';
import { ProductSummaryCommon, ValidationModule, DatePipeModule, ProductsExternalInfoService, } from '@backbase/product-summary-common-ang';
import { ProductSummaryDetailsComponent, ProductSummaryDetailsLoadingCustomizableDirective, ProductSummaryDetailsNoAccountSelectedCustomizableDirective, ProductSummaryDetailsViewCustomizableDirective, ProductSummaryDetailsCustomizableDirective, ProductSummaryDetailsHeaderCustomizableDirective, } from './product-summary-details/product-summary-details.component';
import { CurrentAccountDetailsComponent, ProductSummaryDetailsCurrentAccountCustomizableDirective, } from './product-summary-details/product-details-ui/current-account-details/current-account-details.component';
import { SavingsAccountDetailsComponent, ProductSummaryDetailsSavingsAccountCustomizableDirective, } from './product-summary-details/product-details-ui/savings-account-details/savings-account-details.component';
import { TermDepositDetailsComponent, ProductSummaryDetailsTermDepositCustomizableDirective, } from './product-summary-details/product-details-ui/term-deposit-details/term-deposit-details.component';
import { LoanDetailsComponent, ProductSummaryDetailsLoanCustomizableDirective, } from './product-summary-details/product-details-ui/loan-details/loan-details.component';
import { CreditCardDetailsComponent, ProductSummaryDetailsCreditCardCustomizableDirective, } from './product-summary-details/product-details-ui/credit-card-details/credit-card-details.component';
import { DebitCardDetailsComponent, ProductSummaryDetailsDebitCardCustomizableDirective, } from './product-summary-details/product-details-ui/debit-card-details/debit-card-details.component';
import { InvestmentAccountDetailsComponent, ProductSummaryDetailsInvestmentAccountCustomizableDirective, } from './product-summary-details/product-details-ui/investment-account-details/investment-account-details.component';
import { OtherDetailsComponent, ProductSummaryDetailsOtherCustomizableDirective, } from './product-summary-details/product-details-ui/other-details/other-details.component';
import { FrequencyUnitKeyComponent } from './product-summary-details/product-details-ui/frequency-unit-key.component';
import { ProductSummaryDetailsWidgetComponent } from './product-summary-details-widget.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const components = [
    ProductSummaryDetailsWidgetComponent,
    ProductSummaryDetailsComponent,
    CurrentAccountDetailsComponent,
    SavingsAccountDetailsComponent,
    TermDepositDetailsComponent,
    LoanDetailsComponent,
    CreditCardDetailsComponent,
    DebitCardDetailsComponent,
    InvestmentAccountDetailsComponent,
    OtherDetailsComponent,
    FrequencyUnitKeyComponent,
];
const directives = [
    ProductSummaryDetailsLoadingCustomizableDirective,
    ProductSummaryDetailsNoAccountSelectedCustomizableDirective,
    ProductSummaryDetailsViewCustomizableDirective,
    ProductSummaryDetailsCustomizableDirective,
    ProductSummaryDetailsHeaderCustomizableDirective,
    ProductSummaryDetailsCurrentAccountCustomizableDirective,
    ProductSummaryDetailsSavingsAccountCustomizableDirective,
    ProductSummaryDetailsTermDepositCustomizableDirective,
    ProductSummaryDetailsLoanCustomizableDirective,
    ProductSummaryDetailsCreditCardCustomizableDirective,
    ProductSummaryDetailsDebitCardCustomizableDirective,
    ProductSummaryDetailsInvestmentAccountCustomizableDirective,
    ProductSummaryDetailsOtherCustomizableDirective,
];
const uiModules = [
    AmountModule,
    ButtonModule,
    EmptyStateModule,
    IconModule,
    LoadingIndicatorModule,
    PaymentCardNumberModule,
    PaymentCardModule,
    TooltipModule,
    ArrangementStateIndicatorModule,
    ProductNumberModule,
    ValidationModule,
    NotificationModule,
    DatePipeModule,
];
const services = [ProductsExternalInfoService];
/**
 * Exports the Widgets and UI components for displaying of account details. (Retail banking)
 *
 * The following widgets are included in this module:
 *  - `ProductSummaryDetailsWidgetComponent`
 *
 * @see ProductSummaryDetailsWidgetComponent
 *
 * The following components are included in this module:
 * - `ProductSummaryDetailsComponent`,
 * - `CurrentAccountDetailsComponent`,
 * - `SavingsAccountDetailsComponent`,
 * - `TermDepositDetailsComponent`,
 * - `LoanDetailsComponent`,
 * - `CreditCardDetailsComponent`,
 * - `DebitCardDetailsComponent`,
 * - `InvestmentAccountDetailsComponent`,
 * - `OtherDetailsComponent`,
 * - `FrequencyUnitKeyComponent`,
 *
 * @see `ProductSummaryDetailsComponent`,
 * @see `CurrentAccountDetailsComponent`,
 * @see `SavingsAccountDetailsComponent`,
 * @see `TermDepositDetailsComponent`,
 * @see `LoanDetailsComponent`,
 * @see `CreditCardDetailsComponent`,
 * @see `DebitCardDetailsComponent`,
 * @see `InvestmentAccountDetailsComponent`,
 * @see `OtherDetailsComponent`,
 * @see `FrequencyUnitKeyComponent`,
 *
 * @usageNotes
 *
 * ### Include the exported widgets in your `AppModule`:
 *
 * ```typescript
 * @NgModule({
 *   imports: [
 *     ProductSummaryDetailsWidgetModule,
 *   ],
 *   ...
 * })
 * export class AppModule {}
 * ```
 */
export class ProductSummaryDetailsWidgetModule {
}
ProductSummaryDetailsWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryDetailsWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductSummaryDetailsWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryDetailsWidgetModule, declarations: [ProductSummaryDetailsWidgetComponent,
        ProductSummaryDetailsComponent,
        CurrentAccountDetailsComponent,
        SavingsAccountDetailsComponent,
        TermDepositDetailsComponent,
        LoanDetailsComponent,
        CreditCardDetailsComponent,
        DebitCardDetailsComponent,
        InvestmentAccountDetailsComponent,
        OtherDetailsComponent,
        FrequencyUnitKeyComponent, ProductSummaryDetailsLoadingCustomizableDirective,
        ProductSummaryDetailsNoAccountSelectedCustomizableDirective,
        ProductSummaryDetailsViewCustomizableDirective,
        ProductSummaryDetailsCustomizableDirective,
        ProductSummaryDetailsHeaderCustomizableDirective,
        ProductSummaryDetailsCurrentAccountCustomizableDirective,
        ProductSummaryDetailsSavingsAccountCustomizableDirective,
        ProductSummaryDetailsTermDepositCustomizableDirective,
        ProductSummaryDetailsLoanCustomizableDirective,
        ProductSummaryDetailsCreditCardCustomizableDirective,
        ProductSummaryDetailsDebitCardCustomizableDirective,
        ProductSummaryDetailsInvestmentAccountCustomizableDirective,
        ProductSummaryDetailsOtherCustomizableDirective], imports: [CommonModule, i1.BackbaseCoreModule, ProductSummaryCommon,
        DatePipeModule, AmountModule,
        ButtonModule,
        EmptyStateModule,
        IconModule,
        LoadingIndicatorModule,
        PaymentCardNumberModule,
        PaymentCardModule,
        TooltipModule,
        ArrangementStateIndicatorModule,
        ProductNumberModule,
        ValidationModule,
        NotificationModule,
        DatePipeModule], exports: [ProductSummaryDetailsWidgetComponent,
        ProductSummaryDetailsComponent,
        CurrentAccountDetailsComponent,
        SavingsAccountDetailsComponent,
        TermDepositDetailsComponent,
        LoanDetailsComponent,
        CreditCardDetailsComponent,
        DebitCardDetailsComponent,
        InvestmentAccountDetailsComponent,
        OtherDetailsComponent,
        FrequencyUnitKeyComponent, ProductSummaryDetailsLoadingCustomizableDirective,
        ProductSummaryDetailsNoAccountSelectedCustomizableDirective,
        ProductSummaryDetailsViewCustomizableDirective,
        ProductSummaryDetailsCustomizableDirective,
        ProductSummaryDetailsHeaderCustomizableDirective,
        ProductSummaryDetailsCurrentAccountCustomizableDirective,
        ProductSummaryDetailsSavingsAccountCustomizableDirective,
        ProductSummaryDetailsTermDepositCustomizableDirective,
        ProductSummaryDetailsLoanCustomizableDirective,
        ProductSummaryDetailsCreditCardCustomizableDirective,
        ProductSummaryDetailsDebitCardCustomizableDirective,
        ProductSummaryDetailsInvestmentAccountCustomizableDirective,
        ProductSummaryDetailsOtherCustomizableDirective, AmountModule,
        ButtonModule,
        EmptyStateModule,
        IconModule,
        LoadingIndicatorModule,
        PaymentCardNumberModule,
        PaymentCardModule,
        TooltipModule,
        ArrangementStateIndicatorModule,
        ProductNumberModule,
        ValidationModule,
        NotificationModule,
        DatePipeModule] });
ProductSummaryDetailsWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryDetailsWidgetModule, providers: [...services], imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { ProductSummaryDetailsWidgetComponent },
            }),
            ProductSummaryCommon,
            DatePipeModule,
            ...uiModules,
        ], AmountModule,
        ButtonModule,
        EmptyStateModule,
        IconModule,
        LoadingIndicatorModule,
        PaymentCardNumberModule,
        PaymentCardModule,
        TooltipModule,
        ArrangementStateIndicatorModule,
        ProductNumberModule,
        ValidationModule,
        NotificationModule,
        DatePipeModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryDetailsWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { ProductSummaryDetailsWidgetComponent },
                        }),
                        ProductSummaryCommon,
                        DatePipeModule,
                        ...uiModules,
                    ],
                    declarations: [...components, ...directives],
                    exports: [...components, ...directives, ...uiModules],
                    entryComponents: [ProductSummaryDetailsComponent],
                    providers: [...services],
                }]
        }] });
