// @backbase/foundation-ang/esm2015/core/backbase-core/communication/pipes/handlers/app-config.js
import { InjectionToken } from '@angular/core';
import { pluck } from 'rxjs/operators';
import { CommunicationDefinition, serializerDeserializer } from '@backbase/communication-property';
import { AppConfigService } from '../../../models/app-config.service';
import { createPipelineProcessorConfig } from '../pipeline-registry';
/**
 * @deprecated Will be removed in v7.0.0
 */
export const createAppConfigSource = (appConfig) => ({
    source: (configName) => () => appConfig.getConfig().pipe(pluck(configName)),
});
const TAG = 'app-config';
/**
 * @deprecated Will be removed in v7.0.0
 * AppConfig Communication Transport
 *
 * Can be used in standalone development to configure communication using AppConfig
 */
export class AppConfig {
    /**
     * Generate a serialized communication configuration suitable for use as a Widget Input property value.
     *
     * @param group The name of the communication group to which the Input belongs
     */
    static toProperty(group) {
        return serializerDeserializer.toProperty(new CommunicationDefinition(TAG, group));
    }
}
/**
 * @deprecated Will be removed in v7.0.0
 * Configure the `AppConfigSource`.
 *
 * Use the `AppConfig.toProperty` helper to configure this handler.
 *
 * @usageNotes
 *
 * ```json
 * {
 *   "input.myInput": "AppConfig.toProperty(configPropertyName)", // read the value of application configuration
 * }
 * ```
 */
export const appConfigProcessorConfig = createPipelineProcessorConfig(TAG, {
    provide: new InjectionToken('Application configuration pipeline handler'),
    useFactory: createAppConfigSource,
    deps: [AppConfigService],
});
