// @backbase/foundation-ang/esm2015/core/backbase-core/communication/pipes/handlers/item-navigate.js
import { ӨItemNavigationService } from '../../item-navigation.service';
import { InjectionToken } from '@angular/core';
export const createItemNavigate = (itemNavigation) => ({
    sink: (targetName) => (stdin) => {
        stdin.subscribe(() => {
            itemNavigation.navigateToItem(targetName, {});
        });
    },
});
/**
 * Configure the `ItemNavigateSink`.
 *
 * Use the string 'item-navigate' to configure this handler.
 *
 * Note: to pass route params you must use `RouteParamsStore` instead.
 *
 * @usageNotes
 *
 * Navigate between widgets
 *
 *
 * ```json
 * {
 *   "name": "widget-a",
 *   "properties": {
 *     "output.myOutput": "item-navigate widget-b"
 *   }
 * },
 * {
 *   "name": "widget-b"
 * }
 * ```
 *
 * See `RouteParamsStore` for more options.
 */
export const itemNavigateProcessorConfig = {
    name: 'item-navigate',
    provider: {
        provide: new InjectionToken('Item navigation pipeline handler'),
        useFactory: createItemNavigate,
        deps: [ӨItemNavigationService],
    },
};
