// @backbase/foundation-ang/esm2015/core/backbase-core/communication/pipes/handlers/navigation.js
import { InjectionToken } from '@angular/core';
import { combineLatest } from 'rxjs';
import { ItemTransitionService } from '../../deprecated/item-transition.service';
import { ItemModel } from '../../../models/item-model';
import { ӨRootContainerService } from '../../../models/root-container.service';
import { mergeMap, filter, map } from 'rxjs/operators';
export const pipeName = 'navigation';
export const navigationHandlerToken = new InjectionToken('Navigation handler');
export const createNavigationSink = (rootContainer, itemTransition, itemModel) => {
    return {
        sink: (outputName, config) => (stdin) => {
            if (outputName === undefined || config === undefined) {
                return;
            }
            const [targetName, targetInput] = config.split('.');
            const target$ = rootContainer.model.pipe(mergeMap(tree => tree.find(item => item.name === targetName)), filter((a) => a !== undefined));
            const targetAnscestors = target$.pipe(mergeMap(tree => tree.flatten()), map(flatNodes => flatNodes.map(node => node.value)));
            combineLatest([target$, targetAnscestors, stdin]).subscribe(([target, anscestors, payload]) => itemTransition.transition(itemModel, outputName, target.value, targetInput, anscestors, payload));
        },
    };
};
export const navigationProcessorConfig = {
    name: pipeName,
    provider: {
        provide: navigationHandlerToken,
        useFactory: createNavigationSink,
        deps: [ӨRootContainerService, ItemTransitionService, ItemModel],
    },
};
