// @backbase/foundation-ang/esm2015/core/backbase-core/communication/pipes/handlers/route-param-multi.js
import { ӨItemNavigationService } from '../../item-navigation.service';
import { InjectionToken } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { RouterNavigationService } from '../../../routing/navigation/router-navigation.service';
export const createRouteParamsMulti = (activatedRoute, routerService, itemNavigation) => ({
    sink: (targetName) => (stdin) => {
        stdin.subscribe(params => {
            if (!targetName) {
                routerService.navigate(routerService.setParams(params, activatedRoute));
            }
            else {
                itemNavigation.navigateToItem(targetName, params);
            }
        });
    },
});
/**
 * Configure the `RouteParamsMultiSink`.
 *
 * Use the string 'route-param-multi' to configure this handler.
 *
 * @usageNotes
 *
 * Store some 2-way bound states to the router as "paramName"
 *
 * ```json
 * {
 *   "name": "widget-a",
 *   "properties": {
 *     "output.myOutput": "route-param-multi",
 *     "input.myInput": "route-param input1", // assuming input1 is set via myOutput
 *     "input.myInput2": "route-param input2" // assuming input2 is set via myOutput
 *   }
 * }
 * ```
 *
 * Navigate to widget-b and pass data via route-param
 *
 * ```json
 * {
 *   "name": "widget-a",
 *   "properties": {
 *     "output.myOutput": "route-param-multi widget-b"
 *   }
 * },
 * {
 *   "name": "widget-b",
 *   "properties": {
 *     "input.myInput": "route-param input1",
 *     "input.myInput2": "route-param input2"
 *   }
 * }
 * ```
 *
 * See `RouteParamsMultiSink` for more options.
 *
 * Also see:
 *  - `routeParamsProcessorConfig`
 *  -
 */
export const routeParamsMultiProcessorConfig = {
    name: 'route-param-multi',
    provider: {
        provide: new InjectionToken('Route param multi pipeline handler'),
        useFactory: createRouteParamsMulti,
        deps: [ActivatedRoute, RouterNavigationService, ӨItemNavigationService],
    },
};
