// @backbase/foundation-ang/esm2015/core/backbase-core/communication/pipes/handlers/url.js
import { InjectionToken } from '@angular/core';
export class UrlSinkNavigator {
    go(url) {
        window.location.href = url;
    }
}
export const createUrlSink = (navigator = new UrlSinkNavigator()) => ({
    sink: (url, paramName) => (stdin) => {
        stdin.subscribe((payload) => {
            navigator.go(navigationUrlWithParam(url, paramName, payload));
        });
    },
});
function navigationUrlWithParam(navigationUrl, paramName, param) {
    if (!paramName || !param) {
        return navigationUrl;
    }
    const paramSeparator = navigationUrl.indexOf('?') === -1 ? '?' : '&';
    return `${navigationUrl}${paramSeparator}${paramName}=${param}`;
}
/**
 * Configure the `UrlSink`.
 *
 * Use the string 'url' to configure this handler.
 *
 * @usageNotes
 *
 * ```json
 * {
 *   "output.myOutput1": "url http://www.google.com/", // navigate without param
 *   "output.myOutput2": "url http://www.google.com/ query" // navigate with param
 * }
 * ```
 */
export const urlProcessorConfig = {
    name: 'url',
    provider: {
        provide: new InjectionToken('Url pipeline handler'),
        useFactory: createUrlSink,
        deps: [],
    },
};
