// @backbase/accounts-transactions-journeys-shared-ui-account-ang/esm2015/lib/pipes/cast-account-property/cast-account-property.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CastAccountPropertyPipe } from './cast-account-property.pipe';
import * as i0 from "@angular/core";
const pipes = [CastAccountPropertyPipe];
export class CastAccountPropertyPipeModule {
}
/** @nocollapse */ CastAccountPropertyPipeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CastAccountPropertyPipeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ CastAccountPropertyPipeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CastAccountPropertyPipeModule, declarations: [CastAccountPropertyPipe], imports: [CommonModule], exports: [CastAccountPropertyPipe] });
/** @nocollapse */ CastAccountPropertyPipeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CastAccountPropertyPipeModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CastAccountPropertyPipeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [...pipes],
                    exports: [...pipes],
                }]
        }] });
