// @backbase/messages-create-message-modal-widget-ang/esm2015/src/messages-create-message-modal-widget-ang.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { BaseCreateMessageFormModule, BaseCreateMessageModalModule, SharedMethodsModule, } from '@backbase/messages-common';
import { HeaderModule } from '@backbase/ui-ang/header';
import { ButtonModule } from '@backbase/ui-ang/button';
import { ModalModule } from '@backbase/ui-ang/modal';
import { MessagesCreateMessageModalCustomizableDirective, MessagesCreateMessageModalWidgetComponent, } from './messages-create-message-modal-widget-ang.component';
import { CreateMessageFormModule } from './create-message-form/create-message-form.module';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const components = [MessagesCreateMessageModalWidgetComponent];
const directives = [MessagesCreateMessageModalCustomizableDirective];
const uiModules = [ButtonModule, ModalModule, HeaderModule];
const innerModules = [CreateMessageFormModule, BaseCreateMessageFormModule];
export class MessagesCreateMessageModalWidgetModule {
}
MessagesCreateMessageModalWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesCreateMessageModalWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MessagesCreateMessageModalWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesCreateMessageModalWidgetModule, declarations: [MessagesCreateMessageModalWidgetComponent, MessagesCreateMessageModalCustomizableDirective], imports: [CommonModule, i1.BackbaseCoreModule, CreateMessageFormModule,
        SharedMethodsModule,
        BaseCreateMessageModalModule, ButtonModule, ModalModule, HeaderModule], exports: [MessagesCreateMessageModalWidgetComponent, MessagesCreateMessageModalCustomizableDirective, ButtonModule, ModalModule, HeaderModule, CreateMessageFormModule, BaseCreateMessageFormModule] });
MessagesCreateMessageModalWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesCreateMessageModalWidgetModule, imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { MessagesCreateMessageModalWidgetComponent },
            }),
            CreateMessageFormModule,
            SharedMethodsModule,
            BaseCreateMessageModalModule,
            ...uiModules,
        ], ButtonModule, ModalModule, HeaderModule, CreateMessageFormModule, BaseCreateMessageFormModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesCreateMessageModalWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { MessagesCreateMessageModalWidgetComponent },
                        }),
                        CreateMessageFormModule,
                        SharedMethodsModule,
                        BaseCreateMessageModalModule,
                        ...uiModules,
                    ],
                    declarations: [...components, ...directives],
                    exports: [...components, ...directives, ...uiModules, ...innerModules],
                }]
        }] });
