// @backbase/turnovers-widget-ang/esm2015/src/turnovers-widget-helpers.service.js
import { Injectable } from '@angular/core';
import { DEFAULT_START_DAY } from './constants';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class TurnoversWidgetHelpersService {
    constructor(datePipe) {
        this.datePipe = datePipe;
        this.extendTurnoverItem = (barData) => (Object.assign(Object.assign({}, barData), { intervalStartDate: this.datePipe.transform(barData.intervalStartDate.slice(0, 10), 'MMM') || '' }));
    }
    getChartData(turnovers, labels) {
        return (turnovers || []).map(function (barData) {
            return {
                series: [
                    {
                        name: labels.income,
                        value: Number(barData.creditAmount.amount),
                        currency: barData.creditAmount.currencyCode,
                    },
                    {
                        name: labels.spending,
                        value: Number(barData.debitAmount.amount),
                        currency: barData.debitAmount.currencyCode,
                    },
                ],
                currency: barData.balance.currencyCode,
                diff: Number(barData.balance.amount),
                badge: barData.forecast ? labels.forecast : '',
                name: barData.intervalStartDate.slice(0, 10) + (barData.forecast ? ' ' : ''),
            };
        });
    }
    periodToDate(period) {
        const date = new Date();
        // current date already covers one period
        const intervalCount = period.duration;
        switch (period.interval) {
            case 'DAY':
                date.setDate(date.getDate() - intervalCount);
                break;
            case 'WEEK':
                date.setDate(date.getDate() - intervalCount * 7);
                break;
            case 'YEAR':
                date.setFullYear(date.getFullYear() - intervalCount);
                break;
            case 'MONTH':
            default:
                date.setMonth(date.getMonth() - (intervalCount - 1));
                date.setDate(DEFAULT_START_DAY);
                break;
        }
        return date.toISOString().slice(0, 10);
    }
}
TurnoversWidgetHelpersService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TurnoversWidgetHelpersService, deps: [{ token: i1.DatePipe }], target: i0.ɵɵFactoryTarget.Injectable });
TurnoversWidgetHelpersService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TurnoversWidgetHelpersService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TurnoversWidgetHelpersService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.DatePipe }]; } });
