// @backbase/turnovers-widget-ang/esm2015/src/turnovers-widget.module.js
// General modules
import { CommonModule, CurrencyPipe, DatePipe } from '@angular/common';
import { NgModule } from '@angular/core';
// Backbase modules
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { FormatAmountModule } from '@backbase/ui-ang/amount';
import { ButtonModule } from '@backbase/ui-ang/button';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { NgxChartsModule } from '@swimlane/ngx-charts';
import { BbTurnoversLegendComponent } from './bb-turnovers-legend/bb-turnovers-legend.component';
import { BbGridPanelSeriesComponent } from './ngx-charts/common/bb-grid-panel-series.component';
import { TurnoversBarChartComponent, TurnoversChartLegendCustomizableDirective, TurnoversChartTooltipCustomizableDirective, TurnoversVerticalGroupedBarChartCustomizableDirective, TurnoversVerticalStackedBarChartCustomizableDirective, } from './turnovers-bar-chart/turnovers-bar-chart.component';
import { TurnoversVerticalBarChartComponent } from './turnovers-vertical-bar-chart/turnovers-vertical-bar-chart.component';
import { BbSeriesVerticalComponent } from './turnovers-vertical-grouped-bar-chart/bb-charts-series-vertical.component';
import { TurnoversVerticalGroupedBarChartComponent } from './turnovers-vertical-grouped-bar-chart/turnovers-vertical-grouped-bar-chart.component';
import { TurnoversVerticalStackedBarChartComponent } from './turnovers-vertical-stacked-bar-chart/turnovers-vertical-stacked-bar-chart.component';
import { TurnoversWidgetComponent } from './turnovers-widget.component';
// Components
import { TurnoversComponent, TurnoversControlsCustomizableDirective, TurnoversEmptyStateCustomizableDirective, TurnoversItemCustomizableDirective, } from './turnovers/turnovers.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const components = [
    TurnoversWidgetComponent,
    TurnoversComponent,
    TurnoversBarChartComponent,
    TurnoversVerticalBarChartComponent,
    TurnoversVerticalStackedBarChartComponent,
    TurnoversVerticalGroupedBarChartComponent,
    BbTurnoversLegendComponent,
    BbSeriesVerticalComponent,
    BbGridPanelSeriesComponent,
];
const customizables = [
    TurnoversItemCustomizableDirective,
    TurnoversControlsCustomizableDirective,
    TurnoversChartTooltipCustomizableDirective,
    TurnoversChartLegendCustomizableDirective,
    TurnoversVerticalStackedBarChartCustomizableDirective,
    TurnoversVerticalGroupedBarChartCustomizableDirective,
    TurnoversEmptyStateCustomizableDirective,
];
const modules = [FormatAmountModule, LoadingIndicatorModule, EmptyStateModule, ButtonModule];
export class TurnoversWidgetModule {
}
TurnoversWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TurnoversWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
TurnoversWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TurnoversWidgetModule, declarations: [TurnoversWidgetComponent,
        TurnoversComponent,
        TurnoversBarChartComponent,
        TurnoversVerticalBarChartComponent,
        TurnoversVerticalStackedBarChartComponent,
        TurnoversVerticalGroupedBarChartComponent,
        BbTurnoversLegendComponent,
        BbSeriesVerticalComponent,
        BbGridPanelSeriesComponent, TurnoversItemCustomizableDirective,
        TurnoversControlsCustomizableDirective,
        TurnoversChartTooltipCustomizableDirective,
        TurnoversChartLegendCustomizableDirective,
        TurnoversVerticalStackedBarChartCustomizableDirective,
        TurnoversVerticalGroupedBarChartCustomizableDirective,
        TurnoversEmptyStateCustomizableDirective], imports: [CommonModule, i1.BackbaseCoreModule, NgxChartsModule, FormatAmountModule, LoadingIndicatorModule, EmptyStateModule, ButtonModule], exports: [TurnoversWidgetComponent,
        TurnoversComponent,
        TurnoversBarChartComponent,
        TurnoversVerticalBarChartComponent,
        TurnoversVerticalStackedBarChartComponent,
        TurnoversVerticalGroupedBarChartComponent,
        BbTurnoversLegendComponent,
        BbSeriesVerticalComponent,
        BbGridPanelSeriesComponent, TurnoversItemCustomizableDirective,
        TurnoversControlsCustomizableDirective,
        TurnoversChartTooltipCustomizableDirective,
        TurnoversChartLegendCustomizableDirective,
        TurnoversVerticalStackedBarChartCustomizableDirective,
        TurnoversVerticalGroupedBarChartCustomizableDirective,
        TurnoversEmptyStateCustomizableDirective, FormatAmountModule, LoadingIndicatorModule, EmptyStateModule, ButtonModule] });
TurnoversWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TurnoversWidgetModule, providers: [DatePipe, CurrencyPipe], imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { TurnoversWidgetComponent },
            }),
            NgxChartsModule,
            ...modules,
        ], FormatAmountModule, LoadingIndicatorModule, EmptyStateModule, ButtonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TurnoversWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { TurnoversWidgetComponent },
                        }),
                        NgxChartsModule,
                        ...modules,
                    ],
                    providers: [DatePipe, CurrencyPipe],
                    entryComponents: [TurnoversComponent],
                    declarations: [...components, ...customizables],
                    exports: [...components, ...customizables, ...modules],
                }]
        }] });
