// @backbase/turnovers-widget-ang/esm2015/src/turnovers.service.js
import { Injectable } from '@angular/core';
import { map } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/transactions";
export class TurnoversService {
    constructor(turnoversDataHttpService) {
        this.turnoversDataHttpService = turnoversDataHttpService;
    }
    loadAnalysisData(params) {
        return this.turnoversDataHttpService.getTurnovers(params).pipe(map((body) => {
            return {
                turnovers: body.turnovers || [],
                forecast: (body.forecast || []).map((item) => (Object.assign(Object.assign({}, item), { forecast: true }))),
            };
        }), map((body) => {
            return (body.turnovers || []).concat(params.forecast && body.forecast ? body.forecast : []);
        }));
    }
}
TurnoversService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TurnoversService, deps: [{ token: i1.TurnoversHttpService }], target: i0.ɵɵFactoryTarget.Injectable });
TurnoversService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TurnoversService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TurnoversService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.TurnoversHttpService }]; } });
