// @backbase/turnovers-widget-ang/esm2015/src/turnovers-widget.component.js
import { Component, Input, Optional } from '@angular/core';
import { BehaviorSubject, of } from 'rxjs';
import { map } from 'rxjs/operators';
import { filterInputParameters, PROPERTIES } from './constants';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
import * as i2 from "./turnovers/turnovers.component";
import * as i3 from "@angular/common";
export class TurnoversWidgetComponent {
    constructor(itemModel) {
        this.itemModel = itemModel;
        this.forecast = this.getProperty(PROPERTIES.FORECAST).pipe(map((propertyValue) => propertyValue === true));
        this.forecastSubject = new BehaviorSubject(true);
        this.forecast$ = filterInputParameters(this.forecastSubject.asObservable(), this.forecast);
    }
    // eslint-disable-next-line
    set forecastValue(value) {
        this.forecastSubject.next(value);
    }
    getProperty(property) {
        return this.itemModel ? this.itemModel.property(property.name, property.value) : of(property.value);
    }
}
TurnoversWidgetComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TurnoversWidgetComponent, deps: [{ token: i1.ItemModel, optional: true }], target: i0.ɵɵFactoryTarget.Component });
TurnoversWidgetComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: TurnoversWidgetComponent, selector: "bb-turnovers-widget", inputs: { forecastValue: ["forecast", "forecastValue"] }, ngImport: i0, template: "<bb-turnovers\n  [forecast]=\"(forecast | async) || undefined\"\n></bb-turnovers>", components: [{ type: i2.TurnoversComponent, selector: "bb-turnovers", inputs: ["forecast"] }], pipes: { "async": i3.AsyncPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TurnoversWidgetComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-turnovers-widget',
                    templateUrl: './turnovers-widget.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.ItemModel, decorators: [{
                    type: Optional
                }] }]; }, propDecorators: { forecastValue: [{
                type: Input,
                args: ['forecast']
            }] } });
