// @backbase/ui-ang/esm2015/tab/tab.component.js
import { Component, ViewChild } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * TabComponent represents Tab item
 * that should be used inside tab group component
 */
export class TabComponent {
    /**
     * Getter for the template attached to this component
     */
    get template() {
        return this.defaultTemplate;
    }
}
TabComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TabComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
TabComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: TabComponent, selector: "bb-tab-ui", viewQueries: [{ propertyName: "defaultTemplate", first: true, predicate: ["tabItemTemplate"], descendants: true, static: true }], ngImport: i0, template: "<ng-template #tabItemTemplate>\n  <ng-content></ng-content>\n</ng-template>\n" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TabComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-tab-ui',
                    templateUrl: './tab.component.html',
                }]
        }], propDecorators: { defaultTemplate: [{
                type: ViewChild,
                args: ['tabItemTemplate', { static: true }]
            }] } });
