// @backbase/pay-bills-journey-ang/esm2015/src/pay-bills-journey.component.js
import { Component } from '@angular/core';
import { Subject } from 'rxjs';
import { takeUntil } from 'rxjs/operators';
import { PayBillsNavigationService } from './services/pay-bills-navigation.service';
import * as i0 from "@angular/core";
import * as i1 from "./services/pay-bills-navigation.service";
import * as i2 from "@angular/router";
import * as i3 from "@backbase/billpay-journeys-common";
export class PayBillsJourneyComponent {
    constructor(navigationService, route, billpayNotificationRoutingService) {
        this.navigationService = navigationService;
        this.route = route;
        this.billpayNotificationRoutingService = billpayNotificationRoutingService;
        this.destroy$ = new Subject();
    }
    ngOnDestroy() {
        this.destroy$.next();
        this.destroy$.complete();
    }
    ngOnInit() {
        this.setupNavigationSubscriptions();
    }
    setupNavigationSubscriptions() {
        this.billpayNotificationRoutingService.upcommingPaymentsSubject$
            .pipe(takeUntil(this.destroy$))
            .subscribe(() => this.navigationService.navigateToPayments());
        this.route.paramMap.pipe(takeUntil(this.destroy$)).subscribe((param) => {
            if (param.has('editRecurringPayment')) {
                this.navigationService.navigateToEditRecurringForm(param.get('editRecurringPayment'));
            }
            else if (param.has('editOneOffPayment')) {
                this.navigationService.navigateToEditOneOffForm(param.get('editOneOffPayment'));
            }
        });
        this.billpayNotificationRoutingService.managePayeesSubject$.pipe(takeUntil(this.destroy$)).subscribe(() => {
            this.navigationService.navigateToAddPayee();
        });
    }
}
PayBillsJourneyComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PayBillsJourneyComponent, deps: [{ token: i1.PayBillsNavigationService }, { token: i2.ActivatedRoute }, { token: i3.BillpayNotificationRoutingService }], target: i0.ɵɵFactoryTarget.Component });
PayBillsJourneyComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: PayBillsJourneyComponent, selector: "bb-pay-bills-journey", providers: [PayBillsNavigationService], ngImport: i0, template: "<router-outlet></router-outlet>\n", directives: [{ type: i2.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PayBillsJourneyComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-pay-bills-journey',
                    templateUrl: './pay-bills-journey.component.html',
                    providers: [PayBillsNavigationService],
                }]
        }], ctorParameters: function () { return [{ type: i1.PayBillsNavigationService }, { type: i2.ActivatedRoute }, { type: i3.BillpayNotificationRoutingService }]; } });
