// @backbase/pay-bills-journey-ang/esm2015/src/pay-bills-journey.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { provideRoutes, RouterModule } from '@angular/router';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { BillpayJourneysCommonModule, BillpayEnrolmentGuard, BillpayPayeeDataService, BillpayPaymentDataService, } from '@backbase/billpay-journeys-common';
import { PayBillsJourneyComponent } from './pay-bills-journey.component';
import { BillPayEnrolmentViewComponent } from './views/billpay-enrolment-view.component';
import { BillpayPaymentFormViewComponent } from './views/billpay-payment-form-view.component';
import { BillpayPaymentReviewViewComponent } from './views/billpay-payment-review-view.component';
import { BillpayPaymentRecurringEditFormViewComponent } from './views/billpay-payment-recurring-edit-form-view.component';
import { BillpayPaymentRecurringEditReviewViewComponent } from './views/billpay-payment-recurring-edit-review-view.component';
import { BillpayPaymentOneOffEditFormViewComponent } from './views/billpay-payment-oneoff-edit-form-view.component';
import { BillpayPaymentOneOffEditReviewViewComponent } from './views/billpay-payment-oneoff-edit-review-view.component';
import { BillpayPaymentMultipleBillsViewComponent } from './views/billpay-payment-multiple-bills-view.component';
import { PayBillsStateService } from './services/pay-bills-state.service';
import { BillpayPaymentMultipleBillsReviewViewComponent } from './views/billpay-payment-multiple-bills-review-view.component';
import { BillpayManagePaymentDeactivateGuardService } from './services/billpay-manage-payment-deactivate-guard.service';
import { PayBillsJourneyUiModule } from './components/pay-bills-journey-ui.module';
import { PayBillsJourneyConfigService } from './services/pay-bills-journey-config.service';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
export const defaultRoute = {
    path: '',
    component: PayBillsJourneyComponent,
    children: [
        { path: 'enrolment', component: BillPayEnrolmentViewComponent },
        {
            path: '',
            canActivate: [BillpayEnrolmentGuard],
            children: [
                { path: '', redirectTo: 'form', pathMatch: 'full' },
                {
                    path: 'form',
                    component: BillpayPaymentMultipleBillsViewComponent,
                    canDeactivate: [BillpayManagePaymentDeactivateGuardService],
                },
                {
                    path: 'review',
                    component: BillpayPaymentMultipleBillsReviewViewComponent,
                    canDeactivate: [BillpayManagePaymentDeactivateGuardService],
                },
                {
                    path: 'recurring/edit/:id',
                    component: BillpayPaymentRecurringEditFormViewComponent,
                    canDeactivate: [BillpayManagePaymentDeactivateGuardService],
                },
                {
                    path: 'recurring/edit/:id/review',
                    component: BillpayPaymentRecurringEditReviewViewComponent,
                    canDeactivate: [BillpayManagePaymentDeactivateGuardService],
                },
                {
                    path: 'one-off/edit/:id',
                    component: BillpayPaymentOneOffEditFormViewComponent,
                    canDeactivate: [BillpayManagePaymentDeactivateGuardService],
                },
                {
                    path: 'one-off/edit/:id/review',
                    component: BillpayPaymentOneOffEditReviewViewComponent,
                    canDeactivate: [BillpayManagePaymentDeactivateGuardService],
                },
            ],
        },
    ],
};
const components = [
    BillPayEnrolmentViewComponent,
    BillpayPaymentFormViewComponent,
    BillpayPaymentReviewViewComponent,
    BillpayPaymentRecurringEditFormViewComponent,
    BillpayPaymentRecurringEditReviewViewComponent,
    BillpayPaymentOneOffEditFormViewComponent,
    BillpayPaymentOneOffEditReviewViewComponent,
    BillpayPaymentMultipleBillsViewComponent,
    BillpayPaymentMultipleBillsReviewViewComponent,
];
export class PayBillsJourneyModule {
    static forRoot(data = { route: defaultRoute }) {
        return {
            ngModule: PayBillsJourneyModule,
            providers: [provideRoutes([data.route])],
        };
    }
}
PayBillsJourneyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PayBillsJourneyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PayBillsJourneyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PayBillsJourneyModule, declarations: [PayBillsJourneyComponent, BillPayEnrolmentViewComponent,
        BillpayPaymentFormViewComponent,
        BillpayPaymentReviewViewComponent,
        BillpayPaymentRecurringEditFormViewComponent,
        BillpayPaymentRecurringEditReviewViewComponent,
        BillpayPaymentOneOffEditFormViewComponent,
        BillpayPaymentOneOffEditReviewViewComponent,
        BillpayPaymentMultipleBillsViewComponent,
        BillpayPaymentMultipleBillsReviewViewComponent], imports: [CommonModule,
        RouterModule,
        BillpayJourneysCommonModule,
        ReactiveFormsModule,
        PayBillsJourneyUiModule,
        FormsModule,
        NgbModule, i1.BackbaseCoreModule] });
PayBillsJourneyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PayBillsJourneyModule, providers: [
        BillpayManagePaymentDeactivateGuardService,
        PayBillsStateService,
        BillpayPaymentDataService,
        PayBillsJourneyConfigService,
        BillpayPayeeDataService,
    ], imports: [[
            CommonModule,
            RouterModule,
            BillpayJourneysCommonModule,
            ReactiveFormsModule,
            PayBillsJourneyUiModule,
            FormsModule,
            NgbModule,
            BackbaseCoreModule.withConfig({
                classMap: { PayBillsJourneyComponent },
            }),
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PayBillsJourneyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [PayBillsJourneyComponent, ...components],
                    imports: [
                        CommonModule,
                        RouterModule,
                        BillpayJourneysCommonModule,
                        ReactiveFormsModule,
                        PayBillsJourneyUiModule,
                        FormsModule,
                        NgbModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { PayBillsJourneyComponent },
                        }),
                    ],
                    providers: [
                        BillpayManagePaymentDeactivateGuardService,
                        PayBillsStateService,
                        BillpayPaymentDataService,
                        PayBillsJourneyConfigService,
                        BillpayPayeeDataService,
                    ],
                }]
        }] });
