// @backbase/account-statement-common-ang/esm2015/src/components/account-statement-view-online/account-statement-view-online.component.js
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../../services/account-statement-dom-service";
import * as i2 from "@backbase/ui-ang/icon";
import * as i3 from "@backbase/ui-ang/tooltip-directive";
import * as i4 from "@backbase/ui-ang/button";
export class AccountStatementViewOnlineComponent {
    constructor(domService, renderer) {
        this.domService = domService;
        this.renderer = renderer;
        this.href = '/';
        this.describedByElementId = '';
    }
    viewOnline() {
        this.domService.simulateLinkClick({
            target: '_blank',
            href: this.href,
        }, this.renderer);
    }
}
AccountStatementViewOnlineComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementViewOnlineComponent, deps: [{ token: i1.AccountStatementDomService }, { token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component });
AccountStatementViewOnlineComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: AccountStatementViewOnlineComponent, selector: "bb-account-statement-view-online", inputs: { href: "href", describedByElementId: "describedByElementId" }, ngImport: i0, template: "<div\n  triggers=\"hover focus\"\n  bbTooltip=\"Preview\"\n  i18n-bbTooltip=\"Preview|Tooltip text for preview button@@account-statement.preview.tooltip-simple\"\n>\n  <button\n    bbButton\n    color=\"link\"\n    [circle]=\"true\"\n    aria-label=\"View online\"\n    i18n-aria-label=\"View online link aria label@@account-statement.view-online.aria-label\"\n    [attr.aria-describedby]=\"describedByElementId\"\n    (click)=\"viewOnline()\"\n    data-role=\"view-online\"\n  >\n    <bb-icon-ui name=\"open-in-new\"></bb-icon-ui>\n  </button>\n</div>\n", components: [{ type: i2.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], directives: [{ type: i3.TooltipDirective, selector: "[bbTooltip]", inputs: ["bbTooltip", "triggers"] }, { type: i4.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementViewOnlineComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-account-statement-view-online',
                    templateUrl: './account-statement-view-online.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.AccountStatementDomService }, { type: i0.Renderer2 }]; }, propDecorators: { href: [{
                type: Input
            }], describedByElementId: [{
                type: Input
            }] } });
