// @backbase/ui-ang/esm2015/_example/account-selector/showcase/account-selector-showcase.module.js
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { CommonModule } from '@angular/common';
import { AccountSelectorShowcaseComponent } from './account-selector-showcase.component';
import { AccountSelectorModule } from '@backbase/ui-ang/account-selector';
import { ButtonModule } from '@backbase/ui-ang/button';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import * as i0 from "@angular/core";
const uiModules = [AccountSelectorModule, ButtonModule, InputValidationMessageModule];
export class AccountSelectorShowcaseModule {
}
AccountSelectorShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountSelectorShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AccountSelectorShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountSelectorShowcaseModule, declarations: [AccountSelectorShowcaseComponent], imports: [CommonModule, FormsModule, ReactiveFormsModule, AccountSelectorModule, ButtonModule, InputValidationMessageModule] });
AccountSelectorShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountSelectorShowcaseModule, imports: [[CommonModule, FormsModule, ReactiveFormsModule, ...uiModules]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountSelectorShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, FormsModule, ReactiveFormsModule, ...uiModules],
                    declarations: [AccountSelectorShowcaseComponent],
                    entryComponents: [AccountSelectorShowcaseComponent],
                }]
        }] });
