// @backbase/ui-ang/esm2015/_example/input-email/showcase/input-email-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { InputEmailShowcaseComponent } from './input-email-showcase.component';
import { InputEmailModule } from '@backbase/ui-ang/input-email';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { ReactiveFormsModule } from '@angular/forms';
import * as i0 from "@angular/core";
export class InputEmailShowcaseModule {
}
InputEmailShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputEmailShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputEmailShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputEmailShowcaseModule, declarations: [InputEmailShowcaseComponent], imports: [CommonModule, InputEmailModule, InputValidationMessageModule, ReactiveFormsModule] });
InputEmailShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputEmailShowcaseModule, imports: [[CommonModule, InputEmailModule, InputValidationMessageModule, ReactiveFormsModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputEmailShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, InputEmailModule, InputValidationMessageModule, ReactiveFormsModule],
                    declarations: [InputEmailShowcaseComponent],
                    entryComponents: [InputEmailShowcaseComponent],
                }]
        }] });
