// @backbase/ui-ang/esm2015/_example/input-email/showcase/input-email-showcase.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { Validators } from '@angular/forms';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@angular/forms";
import * as i2 from "@backbase/ui-ang/input-email";
import * as i3 from "@backbase/ui-ang/input-validation-message";
import * as i4 from "@angular/common";
let InputEmailShowcaseComponent = class InputEmailShowcaseComponent {
    constructor(formBuilder) {
        this.formBuilder = formBuilder;
        this.vForm = this.formBuilder.group({
            inputEmail: ['', Validators.required],
        });
    }
    isFieldValid(fieldName) {
        const control = this.getControl(fieldName);
        return control && (control.valid || control.untouched);
    }
    getControl(fieldName) {
        var _a;
        return (_a = this.vForm) === null || _a === void 0 ? void 0 : _a.controls[fieldName];
    }
    hasEmailError(field) {
        const fieldControl = this.getControl(field);
        if (!fieldControl || !fieldControl.errors) {
            return;
        }
        return fieldControl.errors.email;
    }
    hasRequiredError(field) {
        const fieldControl = this.getControl(field);
        if (!fieldControl || !fieldControl.errors) {
            return;
        }
        return fieldControl.errors.required;
    }
};
InputEmailShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputEmailShowcaseComponent, deps: [{ token: i1.FormBuilder }], target: i0.ɵɵFactoryTarget.Component });
InputEmailShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: InputEmailShowcaseComponent, selector: "bb-input-email-ui-showcase", ngImport: i0, template: "<div class=\"row\">\n  <div class=\"col-md-6\">\n    <form [formGroup]=\"vForm\">\n      <label for=\"inputEmail\">\n        E-mail field <span class=\"bb-text-support bb-text-default\">(max {{ inputEmail?.maxLength }} characters)</span>\n      </label>\n      <bb-input-email-ui\n        #inputEmail\n        [id]=\"'inputEmail'\"\n        formControlName=\"inputEmail\"\n        [readonly]=\"false\"\n        [required]=\"true\"\n        placeholder=\"Please enter an email address...\"\n        [maxLength]=\"70\"\n        [aria-labelledby]=\"'inputEmailLabel'\"\n        [aria-describedby]=\"'inputEmailErrors'\"\n        [aria-invalid]=\"!isFieldValid('inputEmail')\"\n      >\n      </bb-input-email-ui>\n      <bb-input-validation-message-ui [showErrors]=\"!isFieldValid('inputEmail')\" id=\"inputEmailErrors\">\n        <p *ngIf=\"hasRequiredError('inputEmail')\">Field is required.</p>\n        <p *ngIf=\"hasEmailError('inputEmail')\">Email format is incorrect.</p>\n      </bb-input-validation-message-ui>\n    </form>\n  </div>\n</div>\n", components: [{ type: i2.InputEmailComponent, selector: "bb-input-email-ui", inputs: ["placeholder", "maxLength", "minLength", "readonly", "autocomplete"] }, { type: i3.InputValidationMessageComponent, selector: "bb-input-validation-message-ui", inputs: ["showErrors", "messageId"] }], directives: [{ type: i1.ɵNgNoValidate, selector: "form:not([ngNoForm]):not([ngNativeValidate])" }, { type: i1.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { type: i1.FormGroupDirective, selector: "[formGroup]", inputs: ["formGroup"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { type: i1.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { type: i1.FormControlName, selector: "[formControlName]", inputs: ["disabled", "formControlName", "ngModel"], outputs: ["ngModelChange"] }, { type: i1.RequiredValidator, selector: ":not([type=checkbox])[required][formControlName],:not([type=checkbox])[required][formControl],:not([type=checkbox])[required][ngModel]", inputs: ["required"] }, { type: i4.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
InputEmailShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
], InputEmailShowcaseComponent);
export { InputEmailShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputEmailShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-input-email-ui-showcase',
                    templateUrl: './input-email-showcase.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.FormBuilder }]; } });
