// @backbase/ui-ang/esm2015/_example/ui-ang-examples.module.js
import { AlertExampleComponent } from './alert/examples/alert.example.component';
import { AmountExampleComponent } from './amount/examples/amount.example.component';
import { AmountInputExampleComponent } from './amount-input/examples/amount-input.example.component';
import { AvatarExampleComponent } from './avatar/examples/avatar.example.component';
import { BadgeExampleComponent } from './badge/examples/badge.example.component';
import { BadgeCounterExampleComponent } from './badge-counter/examples/badge-counter.example.component';
import { ButtonExampleComponent } from './button/examples/button.example.component';
import { CardVendorExampleComponent } from './card-vendor/examples/card-vendor.example.component';
import { CharCounterExampleComponent } from './char-counter/examples/char-counter.example.component';
import { CheckboxGroupExampleComponent } from './checkbox-group/examples/checkbox-group-example.component';
import { CollapsibleAccordionExampleComponent } from './collapsible-accordion/examples/collapsible-accordion.example.component';
import { CollapsibleCardExampleComponent } from './collapsible-card/examples/collapsible-card-example.component';
import { CollapsibleExampleComponent } from './collapsible/examples/collapsible.example.component';
import { CommonErrorStateExampleComponent } from './common-error-state/examples/common-error-state.example.component';
import { CurrencyInputExampleComponent } from './currency-input/examples/currency-input.example.component';
import { DatePipeExampleComponent } from './date-pipe/examples/date-pipe.example.component';
import { DropdownMenuExampleComponent } from './dropdown-menu/examples/dropdown-menu.example.component';
import { EllipsisExampleComponent } from './ellipsis/examples/ellipsis.example.component';
import { EmptyStateExampleComponent } from './empty-state/examples/empty-state.example.component';
import { FieldsetExampleComponent } from './fieldset/examples/fieldset.example.component';
import { FileAttachmentExampleComponent } from './file-attachment/examples/file-attachment.example.component';
import { HeaderExampleComponent } from './header/examples/header.example.component';
import { IconExampleComponent } from './icon/examples/icon.example.component';
import { InputCheckboxExampleComponent } from './input-checkbox/examples/input-checkbox.example.component';
import { InputDatepickerExampleComponent } from './input-datepicker/examples/input-datepicker.example.component';
import { InputEmailExampleComponent } from './input-email/examples/input-email.example.component';
import { InputNumberExampleComponent } from './input-number/examples/input-number.example.component';
import { InputPhoneExampleComponent } from './input-phone/examples/input-phone.example.component';
import { InputRadioGroupExampleComponent } from './input-radio-group/examples/input-radio-group.example.component';
import { InputTextExampleComponent } from './input-text/examples/input-text.example.component';
import { InputValidationMessageExampleComponent } from './input-validation-message/examples/input-validation-message.example.component';
import { ListMultipleSelectExampleComponent } from './list-multiple-select/examples/list-multiple-select-example.component';
import { LoadButtonExampleComponent } from './load-button/examples/load-button.example.component';
import { LoadingIndicatorExampleComponent } from './loading-indicator/examples/loading-indicator.example.component';
import { LogoExampleComponent } from './logo/examples/logo.example.component';
import { ModalExampleComponent } from './modal/examples/modal.example.component';
import { ModeHeaderExampleComponent } from './mode-header/examples/mode-header.example.component';
import { NotificationExampleComponent } from './notification/examples/notification.example.component';
import { PaginationExampleComponent } from './pagination/examples/pagination.example.component';
import { PaymentCardExampleComponent } from './payment-card/examples/payment-card.example.component';
import { PeriodSelectorExampleComponent } from './period-selector/examples/period-selector.example.component';
import { ProductItemBasicAccountExampleComponent } from './product-item-basic-account/examples/product-item-basic-account.example.component';
import { ProductItemCreditCardExampleComponent } from './product-item-credit-card/examples/product-item-credit-card.example.component';
import { ProductItemCurrentAccountExampleComponent } from './product-item-current-account/examples/product-item-current-account.example.component';
import { ProductItemDebitCardExampleComponent } from './product-item-debit-card/examples/product-item-debit-card.example.component';
import { ProductItemInvestmentAccountExampleComponent } from './product-item-investment-account/examples/product-item-investment-account.example.component';
import { ProductItemLoanExampleComponent } from './product-item-loan/examples/product-item-loan.example.component';
import { ProductItemSavingsAccountExampleComponent } from './product-item-savings-account/examples/product-item-savings-account.example.component';
import { ProductItemTermDepositExampleComponent } from './product-item-term-deposit/examples/product-item-term-deposit.example.component';
import { ProductSelectorExampleComponent } from './product-selector/examples/product-selector.example.component';
import { RichTextEditorExampleComponent } from './rich-text-editor/examples/rich-text-editor.example.component';
import { SearchBoxExampleComponent } from './search-box/examples/search-box.example.component';
import { SwitchExampleComponent } from './switch/examples/switch.example.component';
import { TextareaExampleComponent } from './textarea/examples/textarea.example.component';
import { DropdownSingleSelectExampleComponent } from './dropdown-single-select/examples/dropdown-single-select.example.component';
import { TabExampleComponent } from './tab/examples/tab.example.component';
import { TooltipExampleComponent } from './tooltip-directive/examples/tooltip.example.component';
import { ProgessbarExampleComponent } from './progressbar/examples/progressbar.example.component';
import { TableExampleComponent } from './table/examples/table.example.component';
import { InputPasswordExampleComponent } from './input-password/examples/input-password.example.component';
import { InputInlineEditExampleComponent } from './input-inline-edit/examples/input-inline-edit.example.component';
import { StepperExampleComponent } from './stepper/examples/stepper.example.component';
import { InputRangeExampleComponent } from './input-range/examples/input-range.example.component';
import { IbanPipeExampleComponent } from './iban-pipe/examples/iban-pipe.example.component';
import { ActionStatusExampleComponent } from './action-status/examples/action-status.example.component';
import { InfiniteScrollExampleComponent } from './infinite-scroll/examples/infinite-scroll.example.component';
import { HighlightExampleComponent } from './highlight/examples/highlight.example.component';
import { PaymentCardNumberPipeExampleComponent } from './payment-card-number-pipe/examples/payment-card-number-pipe.example.component';
import { AccountNumberPipeExampleComponent } from './account-number-pipe/examples/account-number-pipe.example.component';
import { PhoneNumberPipeExampleComponent } from './phone-number-format-pipe/examples/phone-number-pipe.example.component';
import { AccountSelectorExampleComponent } from './account-selector/examples/account-selector.example.component';
import { DropdownPanelExampleComponent } from './dropdown-panel/examples/dropdown-panel-example.component';
import { DropdownMultiSelectExampleComponent } from './dropdown-multi-select/examples/dropdown-multi-select.example.component';
import { InputTimepickerExampleComponent } from './input-timepicker/examples/input-timepicker-example.component';
import { ItemLogExampleComponent } from './item-log/examples/item-log.example.component';
import { KeyboardClickExampleComponent } from './keyboard-click-directive/examples/keyboard-click.example.component';
import { SelectListExampleComponent } from './select-list/examples/select-list.example.component';
import { InputFileExampleComponent } from './input-file/examples/input-file.example.component';
import { DropdownMultiSelectModule } from '@backbase/ui-ang/dropdown-multi-select';
import { DropdownPanelModule } from '@backbase/ui-ang/dropdown-panel';
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { BackbaseUiModule } from '../ui-ang.module';
import { ValueDiffExampleComponent } from './value-diff/examples/value-diff-example.component';
import { ButtonExampleService } from './button/examples/button.example.service';
import { ConfigurationInputComponent } from './configuration/configuration-input.component';
import { ConfigurationInputService } from './configuration/configuration-input.service';
import * as i0 from "@angular/core";
const exampleComponents = [
    AccountNumberPipeExampleComponent,
    AccountSelectorExampleComponent,
    ActionStatusExampleComponent,
    AlertExampleComponent,
    AmountExampleComponent,
    AmountInputExampleComponent,
    AvatarExampleComponent,
    BadgeCounterExampleComponent,
    BadgeExampleComponent,
    ButtonExampleComponent,
    CardVendorExampleComponent,
    CharCounterExampleComponent,
    CheckboxGroupExampleComponent,
    CollapsibleAccordionExampleComponent,
    CollapsibleCardExampleComponent,
    CollapsibleExampleComponent,
    CommonErrorStateExampleComponent,
    CurrencyInputExampleComponent,
    ConfigurationInputComponent,
    DatePipeExampleComponent,
    DropdownMenuExampleComponent,
    DropdownMultiSelectExampleComponent,
    DropdownPanelExampleComponent,
    DropdownSingleSelectExampleComponent,
    EllipsisExampleComponent,
    EmptyStateExampleComponent,
    FieldsetExampleComponent,
    FileAttachmentExampleComponent,
    HeaderExampleComponent,
    HighlightExampleComponent,
    IbanPipeExampleComponent,
    IconExampleComponent,
    InfiniteScrollExampleComponent,
    InputCheckboxExampleComponent,
    InputDatepickerExampleComponent,
    InputEmailExampleComponent,
    InputFileExampleComponent,
    InputInlineEditExampleComponent,
    InputNumberExampleComponent,
    InputPasswordExampleComponent,
    InputPhoneExampleComponent,
    InputRadioGroupExampleComponent,
    InputRangeExampleComponent,
    InputTextExampleComponent,
    InputTimepickerExampleComponent,
    InputValidationMessageExampleComponent,
    ItemLogExampleComponent,
    KeyboardClickExampleComponent,
    ListMultipleSelectExampleComponent,
    LoadButtonExampleComponent,
    LoadingIndicatorExampleComponent,
    LogoExampleComponent,
    ModalExampleComponent,
    ModeHeaderExampleComponent,
    NotificationExampleComponent,
    PaginationExampleComponent,
    PaymentCardExampleComponent,
    PaymentCardNumberPipeExampleComponent,
    PeriodSelectorExampleComponent,
    PhoneNumberPipeExampleComponent,
    ProductItemBasicAccountExampleComponent,
    ProductItemCreditCardExampleComponent,
    ProductItemCurrentAccountExampleComponent,
    ProductItemDebitCardExampleComponent,
    ProductItemInvestmentAccountExampleComponent,
    ProductItemLoanExampleComponent,
    ProductItemSavingsAccountExampleComponent,
    ProductItemTermDepositExampleComponent,
    ProductSelectorExampleComponent,
    ProgessbarExampleComponent,
    RichTextEditorExampleComponent,
    SearchBoxExampleComponent,
    SelectListExampleComponent,
    StepperExampleComponent,
    SwitchExampleComponent,
    TabExampleComponent,
    TableExampleComponent,
    TextareaExampleComponent,
    TooltipExampleComponent,
    ValueDiffExampleComponent,
];
export class BackbaseUiAngExamplesModule {
}
BackbaseUiAngExamplesModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BackbaseUiAngExamplesModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
BackbaseUiAngExamplesModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BackbaseUiAngExamplesModule, declarations: [AccountNumberPipeExampleComponent,
        AccountSelectorExampleComponent,
        ActionStatusExampleComponent,
        AlertExampleComponent,
        AmountExampleComponent,
        AmountInputExampleComponent,
        AvatarExampleComponent,
        BadgeCounterExampleComponent,
        BadgeExampleComponent,
        ButtonExampleComponent,
        CardVendorExampleComponent,
        CharCounterExampleComponent,
        CheckboxGroupExampleComponent,
        CollapsibleAccordionExampleComponent,
        CollapsibleCardExampleComponent,
        CollapsibleExampleComponent,
        CommonErrorStateExampleComponent,
        CurrencyInputExampleComponent,
        ConfigurationInputComponent,
        DatePipeExampleComponent,
        DropdownMenuExampleComponent,
        DropdownMultiSelectExampleComponent,
        DropdownPanelExampleComponent,
        DropdownSingleSelectExampleComponent,
        EllipsisExampleComponent,
        EmptyStateExampleComponent,
        FieldsetExampleComponent,
        FileAttachmentExampleComponent,
        HeaderExampleComponent,
        HighlightExampleComponent,
        IbanPipeExampleComponent,
        IconExampleComponent,
        InfiniteScrollExampleComponent,
        InputCheckboxExampleComponent,
        InputDatepickerExampleComponent,
        InputEmailExampleComponent,
        InputFileExampleComponent,
        InputInlineEditExampleComponent,
        InputNumberExampleComponent,
        InputPasswordExampleComponent,
        InputPhoneExampleComponent,
        InputRadioGroupExampleComponent,
        InputRangeExampleComponent,
        InputTextExampleComponent,
        InputTimepickerExampleComponent,
        InputValidationMessageExampleComponent,
        ItemLogExampleComponent,
        KeyboardClickExampleComponent,
        ListMultipleSelectExampleComponent,
        LoadButtonExampleComponent,
        LoadingIndicatorExampleComponent,
        LogoExampleComponent,
        ModalExampleComponent,
        ModeHeaderExampleComponent,
        NotificationExampleComponent,
        PaginationExampleComponent,
        PaymentCardExampleComponent,
        PaymentCardNumberPipeExampleComponent,
        PeriodSelectorExampleComponent,
        PhoneNumberPipeExampleComponent,
        ProductItemBasicAccountExampleComponent,
        ProductItemCreditCardExampleComponent,
        ProductItemCurrentAccountExampleComponent,
        ProductItemDebitCardExampleComponent,
        ProductItemInvestmentAccountExampleComponent,
        ProductItemLoanExampleComponent,
        ProductItemSavingsAccountExampleComponent,
        ProductItemTermDepositExampleComponent,
        ProductSelectorExampleComponent,
        ProgessbarExampleComponent,
        RichTextEditorExampleComponent,
        SearchBoxExampleComponent,
        SelectListExampleComponent,
        StepperExampleComponent,
        SwitchExampleComponent,
        TabExampleComponent,
        TableExampleComponent,
        TextareaExampleComponent,
        TooltipExampleComponent,
        ValueDiffExampleComponent], imports: [CommonModule,
        FormsModule,
        ReactiveFormsModule,
        BackbaseUiModule,
        DropdownPanelModule,
        DropdownMultiSelectModule], exports: [AccountNumberPipeExampleComponent,
        AccountSelectorExampleComponent,
        ActionStatusExampleComponent,
        AlertExampleComponent,
        AmountExampleComponent,
        AmountInputExampleComponent,
        AvatarExampleComponent,
        BadgeCounterExampleComponent,
        BadgeExampleComponent,
        ButtonExampleComponent,
        CardVendorExampleComponent,
        CharCounterExampleComponent,
        CheckboxGroupExampleComponent,
        CollapsibleAccordionExampleComponent,
        CollapsibleCardExampleComponent,
        CollapsibleExampleComponent,
        CommonErrorStateExampleComponent,
        CurrencyInputExampleComponent,
        ConfigurationInputComponent,
        DatePipeExampleComponent,
        DropdownMenuExampleComponent,
        DropdownMultiSelectExampleComponent,
        DropdownPanelExampleComponent,
        DropdownSingleSelectExampleComponent,
        EllipsisExampleComponent,
        EmptyStateExampleComponent,
        FieldsetExampleComponent,
        FileAttachmentExampleComponent,
        HeaderExampleComponent,
        HighlightExampleComponent,
        IbanPipeExampleComponent,
        IconExampleComponent,
        InfiniteScrollExampleComponent,
        InputCheckboxExampleComponent,
        InputDatepickerExampleComponent,
        InputEmailExampleComponent,
        InputFileExampleComponent,
        InputInlineEditExampleComponent,
        InputNumberExampleComponent,
        InputPasswordExampleComponent,
        InputPhoneExampleComponent,
        InputRadioGroupExampleComponent,
        InputRangeExampleComponent,
        InputTextExampleComponent,
        InputTimepickerExampleComponent,
        InputValidationMessageExampleComponent,
        ItemLogExampleComponent,
        KeyboardClickExampleComponent,
        ListMultipleSelectExampleComponent,
        LoadButtonExampleComponent,
        LoadingIndicatorExampleComponent,
        LogoExampleComponent,
        ModalExampleComponent,
        ModeHeaderExampleComponent,
        NotificationExampleComponent,
        PaginationExampleComponent,
        PaymentCardExampleComponent,
        PaymentCardNumberPipeExampleComponent,
        PeriodSelectorExampleComponent,
        PhoneNumberPipeExampleComponent,
        ProductItemBasicAccountExampleComponent,
        ProductItemCreditCardExampleComponent,
        ProductItemCurrentAccountExampleComponent,
        ProductItemDebitCardExampleComponent,
        ProductItemInvestmentAccountExampleComponent,
        ProductItemLoanExampleComponent,
        ProductItemSavingsAccountExampleComponent,
        ProductItemTermDepositExampleComponent,
        ProductSelectorExampleComponent,
        ProgessbarExampleComponent,
        RichTextEditorExampleComponent,
        SearchBoxExampleComponent,
        SelectListExampleComponent,
        StepperExampleComponent,
        SwitchExampleComponent,
        TabExampleComponent,
        TableExampleComponent,
        TextareaExampleComponent,
        TooltipExampleComponent,
        ValueDiffExampleComponent] });
BackbaseUiAngExamplesModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BackbaseUiAngExamplesModule, providers: [ButtonExampleService, ConfigurationInputService], imports: [[
            CommonModule,
            FormsModule,
            ReactiveFormsModule,
            BackbaseUiModule,
            DropdownPanelModule,
            DropdownMultiSelectModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BackbaseUiAngExamplesModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        FormsModule,
                        ReactiveFormsModule,
                        BackbaseUiModule,
                        DropdownPanelModule,
                        DropdownMultiSelectModule,
                    ],
                    declarations: [...exampleComponents],
                    exports: [...exampleComponents],
                    entryComponents: [...exampleComponents],
                    providers: [ButtonExampleService, ConfigurationInputService],
                }]
        }] });
export const COMPONENTS = exampleComponents;
