// @backbase/stop-checks-http-ang/bundles/backbase-stop-checks-http-ang.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common/http'), require('@backbase/foundation-ang/data-http')) :
    typeof define === 'function' && define.amd ? define('@backbase/stop-checks-http-ang', ['exports', '@angular/core', '@angular/common/http', '@backbase/foundation-ang/data-http'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["stop-checks-http-ang"] = {}), global.ng.core, global.ng.common.http, global.i2));
})(this, (function (exports, i0, i1, i2) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);

    var SchemeName = {
        IBAN: 'IBAN',
        BBAN: 'BBAN',
        ID: 'ID'
    };

    var StopCheckStatus = {
        PENDING: 'PENDING',
        ACCEPTED: 'ACCEPTED',
        REJECTED: 'REJECTED',
        EXPIRED: 'EXPIRED'
    };

    var STOP_CHECKS_BASE_PATH = new i0.InjectionToken('STOP_CHECKS_BASE_PATH');
    var COLLECTION_FORMATS = {
        'csv': ',',
        'tsv': '   ',
        'ssv': ' ',
        'pipes': '|'
    };

    var StopChecksConfiguration = /** @class */ (function () {
        function StopChecksConfiguration(configurationParameters) {
            if (configurationParameters === void 0) { configurationParameters = {}; }
            this.apiKeys = configurationParameters.apiKeys;
            this.username = configurationParameters.username;
            this.password = configurationParameters.password;
            this.accessToken = configurationParameters.accessToken;
            this.basePath = configurationParameters.basePath;
            this.withCredentials = configurationParameters.withCredentials;
            this.encoder = configurationParameters.encoder;
            if (configurationParameters.credentials) {
                this.credentials = configurationParameters.credentials;
            }
            else {
                this.credentials = {};
            }
        }
        StopChecksConfiguration.prototype.selectHeaderContentType = function (contentTypes) {
            var _this = this;
            if (contentTypes.length === 0) {
                return undefined;
            }
            var type = contentTypes.find(function (x) { return _this.isJsonMime(x); });
            if (type === undefined) {
                return contentTypes[0];
            }
            return type;
        };
        StopChecksConfiguration.prototype.selectHeaderAccept = function (accepts) {
            var _this = this;
            if (accepts.length === 0) {
                return undefined;
            }
            var type = accepts.find(function (x) { return _this.isJsonMime(x); });
            if (type === undefined) {
                return accepts[0];
            }
            return type;
        };
        StopChecksConfiguration.prototype.isJsonMime = function (mime) {
            var jsonMime = new RegExp('^(application\/json|[^;/ \t]+\/[^;/ \t]+[+]json)[ \t]*(;.*)?$', 'i');
            return mime !== null && (jsonMime.test(mime) || mime.toLowerCase() === 'application/json-patch+json');
        };
        StopChecksConfiguration.prototype.lookupCredential = function (key) {
            var value = this.credentials[key];
            return typeof value === 'function'
                ? value()
                : value;
        };
        return StopChecksConfiguration;
    }());
    var CONFIG_TOKEN = new i0.InjectionToken('api-data-module StopChecksConfiguration');

    var StopChecksApiModule = /** @class */ (function () {
        function StopChecksApiModule(parentModule, http, dataModulesManager, config) {
            if (parentModule) {
                throw new Error('StopChecksApiModule is already loaded. Import in your base AppModule only.');
            }
            if (!http) {
                throw new Error('You need to import the HttpClientModule in your AppModule! \n' +
                    'See also https://github.com/angular/angular/issues/20575');
            }
            if (dataModulesManager) {
                dataModulesManager.setModuleConfig(CONFIG_TOKEN, {
                    apiRoot: '',
                    servicePath: config.basePath || '',
                    headers: {},
                });
            }
        }
        StopChecksApiModule.forRoot = function (configurationFactory) {
            return {
                ngModule: StopChecksApiModule,
                providers: [{ provide: StopChecksConfiguration, useFactory: configurationFactory }]
            };
        };
        return StopChecksApiModule;
    }());
    StopChecksApiModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: StopChecksApiModule, deps: [{ token: StopChecksApiModule, optional: true, skipSelf: true }, { token: i1__namespace.HttpClient, optional: true }, { token: i2__namespace.DataModulesManager, optional: true }, { token: StopChecksConfiguration }], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    StopChecksApiModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: StopChecksApiModule });
    StopChecksApiModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: StopChecksApiModule, providers: [], imports: [[]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: StopChecksApiModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [],
                        declarations: [],
                        exports: [],
                        providers: []
                    }]
            }], ctorParameters: function () {
            return [{ type: StopChecksApiModule, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.SkipSelf
                        }] }, { type: i1__namespace.HttpClient, decorators: [{
                            type: i0.Optional
                        }] }, { type: i2__namespace.DataModulesManager, decorators: [{
                            type: i0.Optional
                        }] }, { type: StopChecksConfiguration }];
        } });

    var CustomHttpParameterCodec = /** @class */ (function () {
        function CustomHttpParameterCodec() {
        }
        CustomHttpParameterCodec.prototype.encodeKey = function (k) {
            return encodeURIComponent(k);
        };
        CustomHttpParameterCodec.prototype.encodeValue = function (v) {
            return encodeURIComponent(v);
        };
        CustomHttpParameterCodec.prototype.decodeKey = function (k) {
            return decodeURIComponent(k);
        };
        CustomHttpParameterCodec.prototype.decodeValue = function (v) {
            return decodeURIComponent(v);
        };
        return CustomHttpParameterCodec;
    }());

    var StopChecksHttpService = /** @class */ (function () {
        function StopChecksHttpService(httpClient, basePath, configuration) {
            this.httpClient = httpClient;
            this.basePath = '';
            this.defaultHeaders = new i1.HttpHeaders();
            this.configuration = new StopChecksConfiguration();
            if (configuration) {
                this.configuration = configuration;
            }
            if (typeof this.configuration.basePath !== 'string') {
                if (typeof basePath !== 'string') {
                    basePath = this.basePath;
                }
                this.configuration.basePath = basePath;
            }
            this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
        }
        StopChecksHttpService.prototype.addToHttpParams = function (httpParams, value, key) {
            if (typeof value === "object" && value instanceof Date === false) {
                httpParams = this.addToHttpParamsRecursive(httpParams, value);
            }
            else {
                httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
            }
            return httpParams;
        };
        StopChecksHttpService.prototype.addToHttpParamsRecursive = function (httpParams, value, key) {
            var _this = this;
            if (value == null) {
                return httpParams;
            }
            if (typeof value === "object") {
                if (Array.isArray(value)) {
                    value.forEach(function (elem) { return httpParams = _this.addToHttpParamsRecursive(httpParams, elem, key); });
                }
                else if (value instanceof Date) {
                    if (key != null) {
                        httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                    }
                    else {
                        throw Error("key may not be null if value is Date");
                    }
                }
                else {
                    Object.keys(value).forEach(function (k) { return httpParams = _this.addToHttpParamsRecursive(httpParams, value[k], key != null ? key + "." + k : k); });
                }
            }
            else if (key != null) {
                httpParams = httpParams.append(key, value);
            }
            else {
                throw Error("key may not be null if value is not object or array");
            }
            return httpParams;
        };
        StopChecksHttpService.prototype.getStopChecks = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _from = requestParameters["from"];
            var _cursor = requestParameters["cursor"];
            var _size = requestParameters["size"];
            var queryParameters = new i1.HttpParams({ encoder: this.encoder });
            if (_from !== undefined && _from !== null) {
                queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
            }
            if (_cursor !== undefined && _cursor !== null) {
                queryParameters = this.addToHttpParams(queryParameters, _cursor, 'cursor');
            }
            if (_size !== undefined && _size !== null) {
                queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
            }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.get(this.configuration.basePath + "/client-api/v1/stop-checks", {
                params: queryParameters,
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        StopChecksHttpService.prototype.getStopChecksUrl = function (requestParameters) {
            var _from = requestParameters["from"];
            var _cursor = requestParameters["cursor"];
            var _size = requestParameters["size"];
            var queryParameters = new i1.HttpParams({ encoder: this.encoder });
            if (_from !== undefined && _from !== null) {
                queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
            }
            if (_cursor !== undefined && _cursor !== null) {
                queryParameters = this.addToHttpParams(queryParameters, _cursor, 'cursor');
            }
            if (_size !== undefined && _size !== null) {
                queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
            }
            var queryString = queryParameters.toString();
            return this.configuration.basePath + "/client-api/v1/stop-checks" + (queryString ? "?" + queryString : '');
        };
        StopChecksHttpService.prototype.postStopChecks = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _stopCheckPostRequest = requestParameters["stopCheckPostRequest"];
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var consumes = [
                'application/json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected !== undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.post(this.configuration.basePath + "/client-api/v1/stop-checks", _stopCheckPostRequest, {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        StopChecksHttpService.prototype.postStopChecksUrl = function (requestParameters) {
            var _stopCheckPostRequest = requestParameters["stopCheckPostRequest"];
            return this.configuration.basePath + "/client-api/v1/stop-checks";
        };
        return StopChecksHttpService;
    }());
    StopChecksHttpService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: StopChecksHttpService, deps: [{ token: i1__namespace.HttpClient }, { token: STOP_CHECKS_BASE_PATH, optional: true }, { token: StopChecksConfiguration, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    StopChecksHttpService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: StopChecksHttpService, providedIn: 'root' });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: StopChecksHttpService, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root'
                    }]
            }], ctorParameters: function () {
            return [{ type: i1__namespace.HttpClient }, { type: undefined, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.Inject,
                            args: [STOP_CHECKS_BASE_PATH]
                        }] }, { type: StopChecksConfiguration, decorators: [{
                            type: i0.Optional
                        }] }];
        } });

    var StopChecksHttpServiceGetStopChecksMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v1/stop-checks",
            method: "GET",
            responses: [
                {
                    status: 200,
                    body: {
                        "stopChecks": [{
                                "id": "cb881630-0860-4d2b-b0e1-ff288c531b91",
                                "originatorAccount": {
                                    "name": "Main Current",
                                    "identification": "0256108927",
                                    "schemeName": "BBAN"
                                },
                                "reason": "Invalid name",
                                "status": "REJECTED",
                                "check": {
                                    "number": "10000001337900112340"
                                },
                                "dateOfRequest": "2019-09-24T12:00:00.000Z",
                                "expirationDate": "2019-09-30"
                            }, {
                                "id": "d201e455-d744-44e5-8fc6-458be1cc99e7",
                                "originatorAccount": {
                                    "name": "Shared Current",
                                    "identification": "0256108944",
                                    "schemeName": "BBAN"
                                },
                                "reason": "Checks stolen",
                                "status": "ACCEPTED",
                                "checkNumberRange": {
                                    "firstNumber": "10000001337900112345",
                                    "lastNumber": "10000001337900112355"
                                },
                                "dateOfRequest": "2019-09-30T16:00:00.000Z",
                                "expirationDate": "2019-10-07"
                            }, {
                                "id": "b60a8955-1523-429d-bdbe-905d95225a9c",
                                "originatorAccount": {
                                    "name": "Main Checking Current",
                                    "identification": "0256108963",
                                    "schemeName": "BBAN"
                                },
                                "reason": "Insufficient funds",
                                "status": "PENDING",
                                "check": {
                                    "number": "10000001337900112344",
                                    "amount": {
                                        "currencyCode": "USD",
                                        "amount": "913.32"
                                    },
                                    "payeeName": "Jack S. P. Arrow",
                                    "dateIssued": "2018-10-18"
                                },
                                "dateOfRequest": "2019-10-24T12:00:00.000Z",
                                "expirationDate": "2019-10-30"
                            }],
                        "totalStopChecks": 3
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
            ]
        }]);
    var StopChecksHttpServicePostStopChecksMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v1/stop-checks",
            method: "POST",
            responses: [
                {
                    status: 200,
                    body: {
                        "stopCheck": {
                            "id": "cb881630-0860-4d2b-b0e1-ff288c531b91",
                            "originatorAccount": {
                                "name": "Main Current",
                                "identification": "0256108927",
                                "schemeName": "BBAN"
                            },
                            "reason": "Invalid name",
                            "status": "REJECTED",
                            "check": {
                                "number": "10000001337900112340"
                            },
                            "dateOfRequest": "2019-09-24T12:00:00.000Z",
                            "expirationDate": "2019-09-30"
                        }
                    }
                },
                {
                    status: 200,
                    body: {
                        "stopCheck": {
                            "id": "b60a8955-1523-429d-bdbe-905d95225a9c",
                            "originatorAccount": {
                                "name": "Main Checking Current",
                                "identification": "0256108963",
                                "schemeName": "BBAN"
                            },
                            "reason": "Insufficient funds",
                            "status": "PENDING",
                            "check": {
                                "number": "10000001337900112344",
                                "amount": {
                                    "currencyCode": "USD",
                                    "amount": "913.32"
                                },
                                "payeeName": "Jack S. P. Arrow",
                                "dateIssued": "2018-10-18"
                            },
                            "dateOfRequest": "2019-10-24T12:00:00.000Z",
                            "expirationDate": "2019-10-30"
                        }
                    }
                },
                {
                    status: 200,
                    body: {
                        "stopCheck": {
                            "id": "d201e455-d744-44e5-8fc6-458be1cc99e7",
                            "originatorAccount": {
                                "name": "Shared Current",
                                "identification": "0256108944",
                                "schemeName": "BBAN"
                            },
                            "reason": "Checks stolen",
                            "status": "ACCEPTED",
                            "checkNumberRange": {
                                "firstNumber": "10000001337900112345",
                                "lastNumber": "10000001337900112355"
                            },
                            "dateOfRequest": "2019-09-30T16:00:00.000Z",
                            "expirationDate": "2019-10-07"
                        }
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
            ]
        }]);
    var StopChecksHttpServiceMocksProvider = i2.createMocks([
        {
            urlPattern: "/client-api/v1/stop-checks",
            method: "GET",
            responses: [
                {
                    status: 200,
                    body: {
                        "stopChecks": [{
                                "id": "cb881630-0860-4d2b-b0e1-ff288c531b91",
                                "originatorAccount": {
                                    "name": "Main Current",
                                    "identification": "0256108927",
                                    "schemeName": "BBAN"
                                },
                                "reason": "Invalid name",
                                "status": "REJECTED",
                                "check": {
                                    "number": "10000001337900112340"
                                },
                                "dateOfRequest": "2019-09-24T12:00:00.000Z",
                                "expirationDate": "2019-09-30"
                            }, {
                                "id": "d201e455-d744-44e5-8fc6-458be1cc99e7",
                                "originatorAccount": {
                                    "name": "Shared Current",
                                    "identification": "0256108944",
                                    "schemeName": "BBAN"
                                },
                                "reason": "Checks stolen",
                                "status": "ACCEPTED",
                                "checkNumberRange": {
                                    "firstNumber": "10000001337900112345",
                                    "lastNumber": "10000001337900112355"
                                },
                                "dateOfRequest": "2019-09-30T16:00:00.000Z",
                                "expirationDate": "2019-10-07"
                            }, {
                                "id": "b60a8955-1523-429d-bdbe-905d95225a9c",
                                "originatorAccount": {
                                    "name": "Main Checking Current",
                                    "identification": "0256108963",
                                    "schemeName": "BBAN"
                                },
                                "reason": "Insufficient funds",
                                "status": "PENDING",
                                "check": {
                                    "number": "10000001337900112344",
                                    "amount": {
                                        "currencyCode": "USD",
                                        "amount": "913.32"
                                    },
                                    "payeeName": "Jack S. P. Arrow",
                                    "dateIssued": "2018-10-18"
                                },
                                "dateOfRequest": "2019-10-24T12:00:00.000Z",
                                "expirationDate": "2019-10-30"
                            }],
                        "totalStopChecks": 3
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
            ]
        },
        {
            urlPattern: "/client-api/v1/stop-checks",
            method: "POST",
            responses: [
                {
                    status: 200,
                    body: {
                        "stopCheck": {
                            "id": "cb881630-0860-4d2b-b0e1-ff288c531b91",
                            "originatorAccount": {
                                "name": "Main Current",
                                "identification": "0256108927",
                                "schemeName": "BBAN"
                            },
                            "reason": "Invalid name",
                            "status": "REJECTED",
                            "check": {
                                "number": "10000001337900112340"
                            },
                            "dateOfRequest": "2019-09-24T12:00:00.000Z",
                            "expirationDate": "2019-09-30"
                        }
                    }
                },
                {
                    status: 200,
                    body: {
                        "stopCheck": {
                            "id": "b60a8955-1523-429d-bdbe-905d95225a9c",
                            "originatorAccount": {
                                "name": "Main Checking Current",
                                "identification": "0256108963",
                                "schemeName": "BBAN"
                            },
                            "reason": "Insufficient funds",
                            "status": "PENDING",
                            "check": {
                                "number": "10000001337900112344",
                                "amount": {
                                    "currencyCode": "USD",
                                    "amount": "913.32"
                                },
                                "payeeName": "Jack S. P. Arrow",
                                "dateIssued": "2018-10-18"
                            },
                            "dateOfRequest": "2019-10-24T12:00:00.000Z",
                            "expirationDate": "2019-10-30"
                        }
                    }
                },
                {
                    status: 200,
                    body: {
                        "stopCheck": {
                            "id": "d201e455-d744-44e5-8fc6-458be1cc99e7",
                            "originatorAccount": {
                                "name": "Shared Current",
                                "identification": "0256108944",
                                "schemeName": "BBAN"
                            },
                            "reason": "Checks stolen",
                            "status": "ACCEPTED",
                            "checkNumberRange": {
                                "firstNumber": "10000001337900112345",
                                "lastNumber": "10000001337900112355"
                            },
                            "dateOfRequest": "2019-09-30T16:00:00.000Z",
                            "expirationDate": "2019-10-07"
                        }
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
            ]
        },
    ]);

    exports.COLLECTION_FORMATS = COLLECTION_FORMATS;
    exports.CONFIG_TOKEN = CONFIG_TOKEN;
    exports.STOP_CHECKS_BASE_PATH = STOP_CHECKS_BASE_PATH;
    exports.SchemeName = SchemeName;
    exports.StopCheckStatus = StopCheckStatus;
    exports.StopChecksApiModule = StopChecksApiModule;
    exports.StopChecksConfiguration = StopChecksConfiguration;
    exports.StopChecksHttpService = StopChecksHttpService;
    exports.StopChecksHttpServiceGetStopChecksMocksProvider = StopChecksHttpServiceGetStopChecksMocksProvider;
    exports.StopChecksHttpServiceMocksProvider = StopChecksHttpServiceMocksProvider;
    exports.StopChecksHttpServicePostStopChecksMocksProvider = StopChecksHttpServicePostStopChecksMocksProvider;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
