// @backbase/identity-auth/esm2015/impersonation/impersonation.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ButtonModule } from '@backbase/ui-ang/button';
import { ModeHeaderModule } from '@backbase/ui-ang/mode-header';
import { ImpersonationBannerLayoutComponent } from './banner/banner-layout/banner-layout.component';
import { ImpersonationBannerComponent } from './banner/banner.component';
import * as i0 from "@angular/core";
export class ImpersonationModule {
}
/** @nocollapse */ ImpersonationModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ImpersonationModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ ImpersonationModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ImpersonationModule, declarations: [ImpersonationBannerLayoutComponent, ImpersonationBannerComponent], imports: [CommonModule, ModeHeaderModule, ButtonModule], exports: [ImpersonationBannerComponent] });
/** @nocollapse */ ImpersonationModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ImpersonationModule, imports: [[CommonModule, ModeHeaderModule, ButtonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ImpersonationModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ModeHeaderModule, ButtonModule],
                    declarations: [ImpersonationBannerLayoutComponent, ImpersonationBannerComponent],
                    exports: [ImpersonationBannerComponent],
                }]
        }] });
