// @backbase/payment-request-widget-ang/esm2015/src/payment-request-widget/payment-request-error.js
export class PaymentRequestAccessDenied {
    constructor(error) {
        this.error = error;
        this.kind = 'PaymentRequestAccessDenied';
    }
}
export class PaymentRequestNotFound {
    constructor(error) {
        this.error = error;
        this.kind = 'PaymentRequestNotFound';
    }
}
export class PaymentRequestBadRequest {
    constructor(error) {
        this.error = error;
        this.kind = 'PaymentRequestBadRequest';
    }
}
export class PaymentRequestUnknownError {
    constructor(error) {
        this.error = error;
        this.kind = 'PaymentRequestUnknownError';
    }
}
export class PaymentRequestConnectivityError {
    constructor(error) {
        this.error = error;
        this.kind = 'PaymentRequestConnectivityError';
    }
}
export const parseError = (error) => {
    if (error.error instanceof ErrorEvent) {
        return new PaymentRequestUnknownError(error);
    }
    else {
        switch (error.status) {
            case 0:
                return new PaymentRequestConnectivityError(error);
            case 400:
                return new PaymentRequestBadRequest(error);
            case 401:
                return new PaymentRequestAccessDenied(error);
            case 403:
                return new PaymentRequestAccessDenied(error);
            case 404:
                return new PaymentRequestNotFound(error);
            default:
                return new PaymentRequestUnknownError(error);
        }
    }
};
