// @backbase/manage-payees-journey-ang/esm2015/src/views/billpay-create-manual-payee/billpay-create-manual-payee-view.component.js
import { Component, HostListener } from '@angular/core';
import { of } from 'rxjs';
import { BillpayDiscardChangesModalService, } from '@backbase/billpay-journeys-common';
import * as i0 from "@angular/core";
import * as i1 from "../../services/billpay-manage-payee-state.service";
import * as i2 from "../../services/manage-payees-navigation.service";
import * as i3 from "@backbase/billpay-journeys-common";
import * as i4 from "../../components/add-payee/billpay-manual-payee-form/billpay-manual-payee-form.component";
import * as i5 from "../../components/add-payee/billpay-cancel-new-payee-modal/billpay-cancel-new-payee-modal.component";
import * as i6 from "@angular/common";
export class BillpayCreateManualPayeeViewComponent {
    /** @internal */
    constructor(stateService, router, discardModalService) {
        this.stateService = stateService;
        this.router = router;
        this.discardModalService = discardModalService;
        this.payeeFormState = this.stateService.manualPayeeFormState;
        this.isFormTouched = false;
        this.isFindPayeeModal = false;
    }
    get isDiscardModalOpen() {
        return this.discardModalService.isDiscardModalOpen;
    }
    onWindowUnload($event) {
        if ((this.isFormTouched || this.stateService.isFormProcessing) && !this.isDiscardModalOpen) {
            $event.returnValue = true;
        }
    }
    onFormSubmitted(payee) {
        this.stateService.isFormProcessing = true;
        this.stateService.setManualPayee(payee);
        this.router.navigateToManualPayeeReview();
    }
    onBackToSearch() {
        this.isFindPayeeModal = true;
        this.router.navigateToSearch();
    }
    onFormCanceled() {
        this.router.navigateToPayeeList();
    }
    onFormValueChanged() {
        this.isFormTouched = true;
    }
    openModal() {
        return this.isFormTouched || this.stateService.isFormProcessing ? this.discardModalService.openModal() : of(true);
    }
    onModalResolve(resolvedWith) {
        if (resolvedWith)
            this.stateService.resetManualPayee();
        this.isFindPayeeModal = false;
        this.discardModalService.onModalResolve(resolvedWith);
    }
}
BillpayCreateManualPayeeViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayCreateManualPayeeViewComponent, deps: [{ token: i1.BillpayManagePayeeStateService }, { token: i2.ManagePayeesNavigationService }, { token: i3.BillpayDiscardChangesModalService }], target: i0.ɵɵFactoryTarget.Component });
BillpayCreateManualPayeeViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: BillpayCreateManualPayeeViewComponent, selector: "ng-component", host: { listeners: { "window:beforeunload": "onWindowUnload($event)" } }, providers: [BillpayDiscardChangesModalService], ngImport: i0, template: "<bb-billpay-manual-payee-form\n  [payeeFormState]=\"payeeFormState | async\"\n  (formSubmitted)=\"onFormSubmitted($event)\"\n  (formCanceled)=\"onFormCanceled()\"\n  (backToSearch)=\"onBackToSearch()\"\n  (valueChanged)=\"onFormValueChanged()\"\n>\n</bb-billpay-manual-payee-form>\n<bb-billpay-cancel-new-payee-modal\n  *ngIf=\"isDiscardModalOpen\"\n  [findPayee]=\"isFindPayeeModal\"\n  (resolve)=\"onModalResolve($event)\"\n></bb-billpay-cancel-new-payee-modal>\n", components: [{ type: i4.BillpayManualPayeeFormComponent, selector: "bb-billpay-manual-payee-form", inputs: ["payeeFormState", "editForm"], outputs: ["formSubmitted", "formCanceled", "backToSearch", "valueChanged"] }, { type: i5.BillpayCancelNewPayeeModalComponent, selector: "bb-billpay-cancel-new-payee-modal", inputs: ["findPayee"], outputs: ["resolve"] }], directives: [{ type: i6.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "async": i6.AsyncPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayCreateManualPayeeViewComponent, decorators: [{
            type: Component,
            args: [{
                    templateUrl: './billpay-create-manual-payee-view.component.html',
                    providers: [BillpayDiscardChangesModalService],
                }]
        }], ctorParameters: function () { return [{ type: i1.BillpayManagePayeeStateService }, { type: i2.ManagePayeesNavigationService }, { type: i3.BillpayDiscardChangesModalService }]; }, propDecorators: { onWindowUnload: [{
                type: HostListener,
                args: ['window:beforeunload', ['$event']]
            }] } });
