// @backbase/turnovers-journey-ang/esm2015/src/turnovers-journey.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { provideRoutes, RouterModule } from '@angular/router';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { ProductSummaryAccountSelectorWidgetModule } from '@backbase/product-summary-account-selector-widget-ang';
import { TurnoversWidgetModule } from '@backbase/turnovers-widget-ang';
import { HeaderModule } from '@backbase/ui-ang/header';
import { TurnoversJourneyComponent } from './turnovers-journey.component';
import { TurnoversViewComponent } from './views/turnovers-view.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
export const defaultRoute = {
    path: '',
    component: TurnoversJourneyComponent,
    children: [
        {
            path: '',
            component: TurnoversViewComponent,
            data: { title: 'Turnovers' },
        },
    ],
};
export class TurnoversJourneyModule {
    static forRoot(data = { route: defaultRoute }) {
        return {
            ngModule: TurnoversJourneyModule,
            providers: [provideRoutes([data.route])],
        };
    }
}
TurnoversJourneyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TurnoversJourneyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
TurnoversJourneyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TurnoversJourneyModule, declarations: [TurnoversJourneyComponent, TurnoversViewComponent], imports: [i1.BackbaseCoreModule, CommonModule,
        HeaderModule,
        ProductSummaryAccountSelectorWidgetModule,
        RouterModule,
        TurnoversWidgetModule] });
TurnoversJourneyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TurnoversJourneyModule, imports: [[
            BackbaseCoreModule.withConfig({
                classMap: { TurnoversJourneyComponent },
            }),
            CommonModule,
            HeaderModule,
            ProductSummaryAccountSelectorWidgetModule,
            RouterModule,
            TurnoversWidgetModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TurnoversJourneyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [TurnoversJourneyComponent, TurnoversViewComponent],
                    imports: [
                        BackbaseCoreModule.withConfig({
                            classMap: { TurnoversJourneyComponent },
                        }),
                        CommonModule,
                        HeaderModule,
                        ProductSummaryAccountSelectorWidgetModule,
                        RouterModule,
                        TurnoversWidgetModule,
                    ],
                }]
        }] });
