// @backbase/turnovers-journey-ang/esm2015/src/turnovers-journey-config.service.js
import { Inject, Injectable, InjectionToken, Optional } from '@angular/core';
import * as i0 from "@angular/core";
// eslint-disable-next-line
export const TurnoversJourneyConfigurationToken = new InjectionToken('TurnoversJourneyConfiguration injection token');
const configDefaults = {
    showForecast: true,
};
export class TurnoversJourneyService {
    constructor(config) {
        config = config || {};
        this._config = Object.assign(Object.assign({}, configDefaults), config);
    }
    get defaults() {
        return configDefaults;
    }
    get showForecast() {
        return this._config.showForecast;
    }
}
TurnoversJourneyService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TurnoversJourneyService, deps: [{ token: TurnoversJourneyConfigurationToken, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
TurnoversJourneyService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TurnoversJourneyService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TurnoversJourneyService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [TurnoversJourneyConfigurationToken]
                }] }]; } });
