// @backbase/turnovers-journey-ang/esm2015/src/turnovers-journey.component.js
import { Component } from '@angular/core';
import { TurnoversJourneyService } from './turnovers-journey-config.service';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
export class TurnoversJourneyComponent {
}
TurnoversJourneyComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TurnoversJourneyComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
TurnoversJourneyComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: TurnoversJourneyComponent, selector: "bb-turnovers-journey", providers: [TurnoversJourneyService], ngImport: i0, template: '<router-outlet></router-outlet>', isInline: true, directives: [{ type: i1.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TurnoversJourneyComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-turnovers-journey',
                    template: '<router-outlet></router-outlet>',
                    providers: [TurnoversJourneyService],
                }]
        }] });
