// @backbase/ui-ang/esm2015/amount/amount.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { AmountComponent } from './amount.component';
import * as i0 from "@angular/core";
export class AmountModule {
}
AmountModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AmountModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AmountModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AmountModule, declarations: [AmountComponent], imports: [CommonModule], exports: [AmountComponent] });
AmountModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AmountModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AmountModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [AmountComponent],
                    exports: [AmountComponent],
                }]
        }] });
