// @backbase/ui-ang/esm2015/ui-ang.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NgbAlertModule, NgbCollapseModule, NgbDatepickerModule, NgbDropdownModule, NgbModalModule, NgbPaginationModule, NgbTypeaheadModule, NgbTooltipModule, NgbProgressbarModule, NgbTimepickerModule, } from '@ng-bootstrap/ng-bootstrap';
import { NgSelectModule } from '@ng-select/ng-select';
import { SearchBoxModule } from '@backbase/ui-ang/search-box';
import { AvatarModule } from '@backbase/ui-ang/avatar';
import { BadgeModule } from '@backbase/ui-ang/badge';
import { ButtonModule } from '@backbase/ui-ang/button';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { CardVendorModule } from '@backbase/ui-ang/card-vendor';
import { AmountModule } from '@backbase/ui-ang/amount';
import { LogoModule } from '@backbase/ui-ang/logo';
import { ErrorCommonStateModule } from '@backbase/ui-ang/common-error-state';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { InputDatepickerModule } from '@backbase/ui-ang/input-datepicker';
import { CharCounterModule } from '@backbase/ui-ang/char-counter';
import { CurrencyInputModule } from '@backbase/ui-ang/currency-input';
import { CollapsibleModule } from '@backbase/ui-ang/collapsible';
import { CollapsibleCardModule } from '@backbase/ui-ang/collapsible-card';
import { ModalModule } from '@backbase/ui-ang/modal';
import { PaginationModule } from '@backbase/ui-ang/pagination';
import { PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import { PaymentCardModule } from '@backbase/ui-ang/payment-card';
import { AccountNumberPipeModule } from '@backbase/ui-ang/account-number-pipe';
import { PeriodSelectorModule } from '@backbase/ui-ang/period-selector';
import { SwitchModule } from '@backbase/ui-ang/switch';
import { TabModule } from '@backbase/ui-ang/tab';
import { FileAttachmentModule } from '@backbase/ui-ang/file-attachment';
import { FocusModule } from '@backbase/ui-ang/focus';
import { CollapsibleAccordionModule } from '@backbase/ui-ang/collapsible-accordion';
import { IconModule } from '@backbase/ui-ang/icon';
import { IbanModule } from '@backbase/ui-ang/iban-directive';
import { HeaderModule } from '@backbase/ui-ang/header';
import { AlertModule } from '@backbase/ui-ang/alert';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { NotificationModule } from '@backbase/ui-ang/notification';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import { InputCheckboxModule } from '@backbase/ui-ang/input-checkbox';
import { InputPhoneModule } from '@backbase/ui-ang/input-phone';
import { InputEmailModule } from '@backbase/ui-ang/input-email';
import { DropdownSingleSelectModule } from '@backbase/ui-ang/dropdown-single-select';
import { DropdownMenuModule } from '@backbase/ui-ang/dropdown-menu';
import { PhoneNumberModule } from '@backbase/ui-ang/phone-number-format-pipe';
import { ProductItemSavingsAccountModule } from '@backbase/ui-ang/product-item-savings-account';
import { ProductItemCreditCardModule } from '@backbase/ui-ang/product-item-credit-card';
import { ProductItemBasicAccountModule } from '@backbase/ui-ang/product-item-basic-account';
import { ProductItemCurrentAccountModule } from '@backbase/ui-ang/product-item-current-account';
import { ProductItemInvestmentAccountModule } from '@backbase/ui-ang/product-item-investment-account';
import { ProductItemDebitCardModule } from '@backbase/ui-ang/product-item-debit-card';
import { ProductItemTermDepositModule } from '@backbase/ui-ang/product-item-term-deposit';
import { ProductItemLoanModule } from '@backbase/ui-ang/product-item-loan';
import { FieldsetModule } from '@backbase/ui-ang/fieldset';
import { TextareaModule } from '@backbase/ui-ang/textarea';
import { ProductSelectorModule } from '@backbase/ui-ang/product-selector';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { InputNumberModule } from '@backbase/ui-ang/input-number';
import { RichTextEditorModule } from '@backbase/ui-ang/rich-text-editor';
import { InputRadioGroupModule } from '@backbase/ui-ang/input-radio-group';
import { InputFileModule } from '@backbase/ui-ang/input-file';
import { BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import { ListMultipleSelectModule } from '@backbase/ui-ang/list-multiple-select';
import { CheckboxGroupModule } from '@backbase/ui-ang/checkbox-group';
import { EllipsisModule } from '@backbase/ui-ang/ellipsis';
import { HighlightModule } from '@backbase/ui-ang/highlight';
import { ProgressbarModule } from '@backbase/ui-ang/progressbar';
import { TableModule } from '@backbase/ui-ang/table';
import { InputPasswordModule } from '@backbase/ui-ang/input-password';
import { InputInlineEditModule } from '@backbase/ui-ang/input-inline-edit';
import { StepperModule } from '@backbase/ui-ang/stepper';
import { BbIbanPipeModule } from '@backbase/ui-ang/iban-pipe';
import { ActionStatusModule } from '@backbase/ui-ang/action-status';
import { BadgeCounterModule } from '@backbase/ui-ang/badge-counter';
import { InfiniteScrollModule } from '@backbase/ui-ang/infinite-scroll';
import { AccountSelectorModule } from '@backbase/ui-ang/account-selector';
import { ItemLogModule } from '@backbase/ui-ang/item-log';
import { InputTimepickerModule } from '@backbase/ui-ang/input-timepicker';
import { KeyboardClickModule } from '@backbase/ui-ang/keyboard-click-directive';
import { SelectListModule } from '@backbase/ui-ang/select-list';
import { AmountInputModule } from '@backbase/ui-ang/amount-input';
import { ModeHeaderModule } from '@backbase/ui-ang/mode-header';
import { ControlErrorHandlerModule } from '@backbase/ui-ang/control-error-handler';
import { InputRangeModule } from '@backbase/ui-ang/input-range';
import { ValueDiffDirectiveModule } from '@backbase/ui-ang/value-diff';
import { A11yModule } from '@angular/cdk/a11y';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/future";
const ngbModules = [
    NgbAlertModule,
    NgbCollapseModule,
    NgbDatepickerModule,
    NgbDropdownModule,
    NgbModalModule,
    NgbPaginationModule,
    NgbTypeaheadModule,
    NgbTooltipModule,
    NgbProgressbarModule,
    NgbTimepickerModule,
];
const uiCompModules = [
    AccountNumberPipeModule,
    AccountSelectorModule,
    ActionStatusModule,
    AlertModule,
    AmountInputModule,
    AmountModule,
    AvatarModule,
    BadgeCounterModule,
    BadgeModule,
    BbDatePipeModule,
    BbIbanPipeModule,
    ButtonModule,
    CardVendorModule,
    CharCounterModule,
    CheckboxGroupModule,
    CollapsibleAccordionModule,
    CollapsibleCardModule,
    CollapsibleModule,
    ErrorCommonStateModule,
    CurrencyInputModule,
    DropdownMenuModule,
    DropdownSingleSelectModule,
    EllipsisModule,
    EmptyStateModule,
    FieldsetModule,
    FileAttachmentModule,
    FocusModule,
    HeaderModule,
    HighlightModule,
    IconModule,
    InfiniteScrollModule,
    InputCheckboxModule,
    InputDatepickerModule,
    InputEmailModule,
    InputFileModule,
    InputInlineEditModule,
    InputNumberModule,
    InputPasswordModule,
    InputPhoneModule,
    InputRadioGroupModule,
    InputRangeModule,
    InputTextModule,
    InputTimepickerModule,
    InputValidationMessageModule,
    ItemLogModule,
    KeyboardClickModule,
    ListMultipleSelectModule,
    LoadButtonModule,
    LoadingIndicatorModule,
    LogoModule,
    ModalModule,
    ModeHeaderModule,
    NotificationModule,
    PaginationModule,
    PaymentCardModule,
    PaymentCardNumberModule,
    PeriodSelectorModule,
    PhoneNumberModule,
    ProductItemBasicAccountModule,
    ProductItemCreditCardModule,
    ProductItemCurrentAccountModule,
    ProductItemDebitCardModule,
    ProductItemInvestmentAccountModule,
    ProductItemLoanModule,
    ProductItemSavingsAccountModule,
    ProductItemTermDepositModule,
    ProductSelectorModule,
    ProgressbarModule,
    RichTextEditorModule,
    SearchBoxModule,
    SelectListModule,
    StepperModule,
    SwitchModule,
    TabModule,
    TableModule,
    TextareaModule,
    TooltipModule,
    ControlErrorHandlerModule,
    ValueDiffDirectiveModule,
    IbanModule,
];
export class BackbaseUiModule {
    constructor(deprecationService) {
        this.deprecationService = deprecationService;
        this.deprecationService.logDeprecatedFeature(`[ui-ang]: The main entry point '@backbase/ui-ang' will be deprecated in ui-ang v8.0. Use
      the secondary entry points instead. I.e. 'import { ... } from '@backbase/ui-ang/button';'`);
        this.deprecationService.logDeprecatedFeature(`[ui-ang]:Packages '@backbase/ui-ang'from v7.0 and '@backbase/backbase-theme' from v2.318.0 will be deprecated.
      From now on, use '@backbase/web-ui' package;'`);
    }
}
BackbaseUiModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BackbaseUiModule, deps: [{ token: i1.DeprecationsService }], target: i0.ɵɵFactoryTarget.NgModule });
BackbaseUiModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BackbaseUiModule, imports: [CommonModule, NgbAlertModule,
        NgbCollapseModule,
        NgbDatepickerModule,
        NgbDropdownModule,
        NgbModalModule,
        NgbPaginationModule,
        NgbTypeaheadModule,
        NgbTooltipModule,
        NgbProgressbarModule,
        NgbTimepickerModule, AccountNumberPipeModule,
        AccountSelectorModule,
        ActionStatusModule,
        AlertModule,
        AmountInputModule,
        AmountModule,
        AvatarModule,
        BadgeCounterModule,
        BadgeModule,
        BbDatePipeModule,
        BbIbanPipeModule,
        ButtonModule,
        CardVendorModule,
        CharCounterModule,
        CheckboxGroupModule,
        CollapsibleAccordionModule,
        CollapsibleCardModule,
        CollapsibleModule,
        ErrorCommonStateModule,
        CurrencyInputModule,
        DropdownMenuModule,
        DropdownSingleSelectModule,
        EllipsisModule,
        EmptyStateModule,
        FieldsetModule,
        FileAttachmentModule,
        FocusModule,
        HeaderModule,
        HighlightModule,
        IconModule,
        InfiniteScrollModule,
        InputCheckboxModule,
        InputDatepickerModule,
        InputEmailModule,
        InputFileModule,
        InputInlineEditModule,
        InputNumberModule,
        InputPasswordModule,
        InputPhoneModule,
        InputRadioGroupModule,
        InputRangeModule,
        InputTextModule,
        InputTimepickerModule,
        InputValidationMessageModule,
        ItemLogModule,
        KeyboardClickModule,
        ListMultipleSelectModule,
        LoadButtonModule,
        LoadingIndicatorModule,
        LogoModule,
        ModalModule,
        ModeHeaderModule,
        NotificationModule,
        PaginationModule,
        PaymentCardModule,
        PaymentCardNumberModule,
        PeriodSelectorModule,
        PhoneNumberModule,
        ProductItemBasicAccountModule,
        ProductItemCreditCardModule,
        ProductItemCurrentAccountModule,
        ProductItemDebitCardModule,
        ProductItemInvestmentAccountModule,
        ProductItemLoanModule,
        ProductItemSavingsAccountModule,
        ProductItemTermDepositModule,
        ProductSelectorModule,
        ProgressbarModule,
        RichTextEditorModule,
        SearchBoxModule,
        SelectListModule,
        StepperModule,
        SwitchModule,
        TabModule,
        TableModule,
        TextareaModule,
        TooltipModule,
        ControlErrorHandlerModule,
        ValueDiffDirectiveModule,
        IbanModule, NgSelectModule, A11yModule], exports: [AccountNumberPipeModule,
        AccountSelectorModule,
        ActionStatusModule,
        AlertModule,
        AmountInputModule,
        AmountModule,
        AvatarModule,
        BadgeCounterModule,
        BadgeModule,
        BbDatePipeModule,
        BbIbanPipeModule,
        ButtonModule,
        CardVendorModule,
        CharCounterModule,
        CheckboxGroupModule,
        CollapsibleAccordionModule,
        CollapsibleCardModule,
        CollapsibleModule,
        ErrorCommonStateModule,
        CurrencyInputModule,
        DropdownMenuModule,
        DropdownSingleSelectModule,
        EllipsisModule,
        EmptyStateModule,
        FieldsetModule,
        FileAttachmentModule,
        FocusModule,
        HeaderModule,
        HighlightModule,
        IconModule,
        InfiniteScrollModule,
        InputCheckboxModule,
        InputDatepickerModule,
        InputEmailModule,
        InputFileModule,
        InputInlineEditModule,
        InputNumberModule,
        InputPasswordModule,
        InputPhoneModule,
        InputRadioGroupModule,
        InputRangeModule,
        InputTextModule,
        InputTimepickerModule,
        InputValidationMessageModule,
        ItemLogModule,
        KeyboardClickModule,
        ListMultipleSelectModule,
        LoadButtonModule,
        LoadingIndicatorModule,
        LogoModule,
        ModalModule,
        ModeHeaderModule,
        NotificationModule,
        PaginationModule,
        PaymentCardModule,
        PaymentCardNumberModule,
        PeriodSelectorModule,
        PhoneNumberModule,
        ProductItemBasicAccountModule,
        ProductItemCreditCardModule,
        ProductItemCurrentAccountModule,
        ProductItemDebitCardModule,
        ProductItemInvestmentAccountModule,
        ProductItemLoanModule,
        ProductItemSavingsAccountModule,
        ProductItemTermDepositModule,
        ProductSelectorModule,
        ProgressbarModule,
        RichTextEditorModule,
        SearchBoxModule,
        SelectListModule,
        StepperModule,
        SwitchModule,
        TabModule,
        TableModule,
        TextareaModule,
        TooltipModule,
        ControlErrorHandlerModule,
        ValueDiffDirectiveModule,
        IbanModule] });
BackbaseUiModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BackbaseUiModule, imports: [[CommonModule, ...ngbModules, ...uiCompModules, NgSelectModule, A11yModule], AccountNumberPipeModule,
        AccountSelectorModule,
        ActionStatusModule,
        AlertModule,
        AmountInputModule,
        AmountModule,
        AvatarModule,
        BadgeCounterModule,
        BadgeModule,
        BbDatePipeModule,
        BbIbanPipeModule,
        ButtonModule,
        CardVendorModule,
        CharCounterModule,
        CheckboxGroupModule,
        CollapsibleAccordionModule,
        CollapsibleCardModule,
        CollapsibleModule,
        ErrorCommonStateModule,
        CurrencyInputModule,
        DropdownMenuModule,
        DropdownSingleSelectModule,
        EllipsisModule,
        EmptyStateModule,
        FieldsetModule,
        FileAttachmentModule,
        FocusModule,
        HeaderModule,
        HighlightModule,
        IconModule,
        InfiniteScrollModule,
        InputCheckboxModule,
        InputDatepickerModule,
        InputEmailModule,
        InputFileModule,
        InputInlineEditModule,
        InputNumberModule,
        InputPasswordModule,
        InputPhoneModule,
        InputRadioGroupModule,
        InputRangeModule,
        InputTextModule,
        InputTimepickerModule,
        InputValidationMessageModule,
        ItemLogModule,
        KeyboardClickModule,
        ListMultipleSelectModule,
        LoadButtonModule,
        LoadingIndicatorModule,
        LogoModule,
        ModalModule,
        ModeHeaderModule,
        NotificationModule,
        PaginationModule,
        PaymentCardModule,
        PaymentCardNumberModule,
        PeriodSelectorModule,
        PhoneNumberModule,
        ProductItemBasicAccountModule,
        ProductItemCreditCardModule,
        ProductItemCurrentAccountModule,
        ProductItemDebitCardModule,
        ProductItemInvestmentAccountModule,
        ProductItemLoanModule,
        ProductItemSavingsAccountModule,
        ProductItemTermDepositModule,
        ProductSelectorModule,
        ProgressbarModule,
        RichTextEditorModule,
        SearchBoxModule,
        SelectListModule,
        StepperModule,
        SwitchModule,
        TabModule,
        TableModule,
        TextareaModule,
        TooltipModule,
        ControlErrorHandlerModule,
        ValueDiffDirectiveModule,
        IbanModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BackbaseUiModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...ngbModules, ...uiCompModules, NgSelectModule, A11yModule],
                    exports: [...uiCompModules],
                }]
        }], ctorParameters: function () { return [{ type: i1.DeprecationsService }]; } });
