// @backbase/actions-product-notifications-settings-widget-ang/esm2015/src/services/actions-product-notifications-settings-router.service.js
import { Injectable } from '@angular/core';
import { ReplaySubject, Subject } from 'rxjs';
import * as i0 from "@angular/core";
export class ActionsProductNotificationsSettingsRouterService {
    constructor() {
        this.productNotificationsSettingsSavedSubject = new Subject();
        /**
         * Stream of settings save events that trigger navigation.
         */
        this.settingsSavedNavigation$ = this.productNotificationsSettingsSavedSubject.asObservable();
        this.productNotificationsSettingsCancelledSubject = new Subject();
        /**
         * Stream of settings cancel events that trigger navigation.
         */
        this.settingsCancelledNavigation$ = this.productNotificationsSettingsCancelledSubject.asObservable();
        this.selectedAccountIdSubject = new ReplaySubject(1);
        /**
         * Stream with selected account IDs.
         */
        this.selectedAccountId$ = this.selectedAccountIdSubject.asObservable();
    }
    /**
     * Selected account ID setter.
     */
    set selectedAccountId(value) {
        this.selectedAccountIdSubject.next(value);
    }
    /**
     * Method to emit settings saved event in a stream.
     */
    emitSettingsSavedNavigation() {
        this.productNotificationsSettingsSavedSubject.next();
    }
    /**
     * Method to emit settings cancelled event in a stream.
     */
    emitSettingsCancelledNavigation() {
        this.productNotificationsSettingsCancelledSubject.next();
    }
}
ActionsProductNotificationsSettingsRouterService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsProductNotificationsSettingsRouterService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
ActionsProductNotificationsSettingsRouterService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsProductNotificationsSettingsRouterService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsProductNotificationsSettingsRouterService, decorators: [{
            type: Injectable
        }] });
