// @backbase/account-statement-store-ang/bundles/backbase-account-statement-store-ang.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@backbase/account-statement-common-ang'), require('@ngrx/store'), require('rxjs'), require('rxjs/operators'), require('@angular/common'), require('@backbase/data-ang/account-statements'), require('@ngrx/effects')) :
    typeof define === 'function' && define.amd ? define('@backbase/account-statement-store-ang', ['exports', '@angular/core', '@backbase/account-statement-common-ang', '@ngrx/store', 'rxjs', 'rxjs/operators', '@angular/common', '@backbase/data-ang/account-statements', '@ngrx/effects'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["account-statement-store-ang"] = {}), global.ng.core, global.i2, global.i1, global.rxjs, global.rxjs.operators, global.ng.common, global.accountStatements$1, global.i1$1));
})(this, (function (exports, i0, i2, i1, rxjs, operators, common, accountStatements$1, i1$1) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i1__namespace$1 = /*#__PURE__*/_interopNamespace(i1);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1$1);

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from, pack) {
        if (pack || arguments.length === 2)
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar)
                        ar = Array.prototype.slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
        return to.concat(ar || Array.prototype.slice.call(from));
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    var loadAccountStatementsBusinessToRequest = function (_a, widgetPreferences) {
        var accountIds = _a.accountIds, filterParams = _a.filterParams, paginationParams = _a.paginationParams, sortingParams = _a.sortingParams, actionRequestParams = __rest(_a, ["accountIds", "filterParams", "paginationParams", "sortingParams"]);
        var _b = filterParams || {}, dateFrom = _b.dateFrom, dateTo = _b.dateTo, category = _b.category;
        var _c = widgetPreferences || {}, minimalDate = _c.minimalDate, maximumDate = _c.maximumDate;
        var params = Object.assign(Object.assign(Object.assign(Object.assign(Object.assign(Object.assign({}, (accountIds && { accountIds: accountIds })), (_hasDate(dateFrom, minimalDate) && {
            dateFrom: _getFormattedDate(dateFrom, minimalDate),
        })), (_hasDate(dateTo, maximumDate) && {
            dateTo: _getFormattedDate(dateTo, maximumDate),
        })), (category && { category: category })), (paginationParams && Object.assign({}, paginationParams))), (sortingParams &&
            sortingParams.direction && {
            direction: _mapSortingDirection(sortingParams.direction),
            orderBy: sortingParams.column,
        }));
        return { accountStatementsPostRequest: params };
    };
    /**
     * Check if date or default date exist
     *
     * @param date Optional date string
     * @param defaultDate Default date object if date is not present
     * @returns boolean
     */
    var _hasDate = function (date, defaultDate) { return !!(date || defaultDate); };
    /**
     * Convert date into yyyy-MM-dd format
     *
     * @param date Optional date string
     * @param defaultDate Default date object if date is not present
     * @returns Date in format of yyyy-MM-dd
     */
    var _getFormattedDate = function (value, defaultDate) {
        var date = value || defaultDate;
        return date && common.formatDate(new Date(date), 'yyyy-MM-dd', 'en-US');
    };
    /**
     * Convert direction of type 'asc' or 'desc' to 'ASC' or 'DESC'
     *
     * @param direction of type string
     * @returns sortingDirection of type AccountStatementsPostRequest.DirectionEnum
     */
    var _mapSortingDirection = function (direction) { return direction === 'asc'
        ? accountStatements$1.AccountStatementsPostRequest.DirectionEnum.ASC
        : accountStatements$1.AccountStatementsPostRequest.DirectionEnum.DESC; };

    var loadAccountStatementsBusiness = i1.createAction('[AS Account Statements] Load Account Statements for Business', i1.props());
    var loadAccountStatementsBusinessSuccess = i1.createAction('[AS Account Statements] Load Account Statements for Business Success', i1.props());
    var loadAccountStatementsFailure = i1.createAction('[AS Account Statements] Load Account Statements Failure', i1.props());
    var updateWidgetPreferences = i1.createAction('[AS Accounts] Update Widget Preferences', i1.props());

    var accountStatementsInitialState = {
        data: [],
        totalCount: 0,
        loading: i2.LoadingState.IDLE,
        error: undefined,
        accountIds: [],
        filterParams: {
            datePeriod: i2.DatePeriods.AllTime,
        },
        widgetPreferences: {
            minimalDate: i2.unixZeroDay(),
            maximumDate: i2.today(),
        },
    };
    var smartLoadingState = function (state, action) {
        if (action.accountIds && action.accountIds !== state.accountIds)
            return i2.LoadingState.LOADING;
        if (action.filterParams && action.filterParams !== state.filterParams)
            return i2.LoadingState.LOADING_FILTERED;
        if (action.append)
            return i2.LoadingState.LOADING_MORE;
        if ((action.sortingParams && action.sortingParams !== state.sortingParams) ||
            (action.paginationParams && action.paginationParams !== state.paginationParams)) {
            return i2.LoadingState.LOADING_TABLE;
        }
        return i2.LoadingState.LOADING;
    };
    var reducer$1 = i1.createReducer(accountStatementsInitialState, i1.on(loadAccountStatementsBusiness, function (state, action) { return (Object.assign(Object.assign({}, state), { loading: smartLoadingState(state, action), error: undefined })); }), i1.on(loadAccountStatementsBusinessSuccess, function (state, action) {
        var accountStatements = action.accountStatements, accountIds = action.accountIds, filterParams = action.filterParams, paginationParams = action.paginationParams, sortingParams = action.sortingParams, totalCount = action.totalCount, append = action.append;
        var dataAppendTo = append ? state.data : [];
        return Object.assign(Object.assign(Object.assign(Object.assign(Object.assign(Object.assign({}, state), { loading: i2.LoadingState.IDLE, data: __spreadArray(__spreadArray([], __read(dataAppendTo)), __read(accountStatements)), totalCount: totalCount }), (accountIds && { accountIds: accountIds })), (filterParams && { filterParams: filterParams })), (sortingParams && { sortingParams: sortingParams })), (paginationParams && { paginationParams: paginationParams }));
    }), i1.on(loadAccountStatementsFailure, function (state, action) { return (Object.assign(Object.assign({}, state), { error: action.error, loading: i2.LoadingState.IDLE })); }), i1.on(updateWidgetPreferences, function (state, action) { return (Object.assign(Object.assign({}, state), { widgetPreferences: Object.assign(Object.assign({}, state.widgetPreferences), action.widgetPreferences) })); }));
    // eslint-disable-next-line prefer-arrow/prefer-arrow-functions
    function accountStatementsReducer(state, action) {
        return reducer$1(state, action);
    }

    var loadCategories = i1.createAction('[AS Account Statements] Load Categories for Business');
    var loadCategoriesSuccess = i1.createAction('[AS Account Statements] Load Categories for Business Success', i1.props());
    var loadCategoriesFailure = i1.createAction('[AS Account Statements] Load Categories for Business Failure', i1.props());

    var categoriesInitialState = {
        data: [],
        error: undefined,
    };
    var reducer = i1.createReducer(categoriesInitialState, i1.on(loadCategoriesSuccess, function (state, _a) {
        var categories = _a.categories;
        return (Object.assign(Object.assign({}, state), { data: categories }));
    }), i1.on(loadCategoriesFailure, function (state, _a) {
        var error = _a.error;
        return (Object.assign(Object.assign({}, state), { data: [], error: error }));
    }));
    // eslint-disable-next-line prefer-arrow/prefer-arrow-functions
    function categoriesReducer(state, action) {
        return reducer(state, action);
    }

    var initialState = {
        accountStatements: accountStatementsInitialState,
        categories: categoriesInitialState,
    };
    /** Key for feature state used in NgRx StoreModule. */
    var ACCOUNT_STATEMENTS_FEATURE_KEY = 'accountStatements';
    var accountStatementsReducers = {
        accountStatements: accountStatementsReducer,
        categories: categoriesReducer,
    };

    var selectAccountStatements = i1.createFeatureSelector(ACCOUNT_STATEMENTS_FEATURE_KEY);

    var selectAccountStatementsState = i1.createSelector(selectAccountStatements, function (accountStatementsState) { return accountStatementsState.accountStatements; });
    var accountStatements = i1.createSelector(selectAccountStatementsState, function (state) { return state.data; });
    var accountStatementsLoading = i1.createSelector(selectAccountStatementsState, function (state) { return state.loading; });
    var accountStatementsError = i1.createSelector(selectAccountStatementsState, function (state) { return state.error; });
    var accountStatementsTotalCount = i1.createSelector(selectAccountStatementsState, function (state) { return state.totalCount; });
    var accountStatementsAccountIds = i1.createSelector(selectAccountStatementsState, function (state) { return state.accountIds; });
    var accountStatementsPaginationParams = i1.createSelector(selectAccountStatementsState, function (state) { return state.paginationParams; });
    var accountStatementsSortingParams = i1.createSelector(selectAccountStatementsState, function (state) { return state.sortingParams; });
    var accountStatementsFilterParams = i1.createSelector(selectAccountStatementsState, function (state) { return state.filterParams; });
    var accountStatementsWidgetPreferences = i1.createSelector(selectAccountStatementsState, function (state) { return state.widgetPreferences; });

    var AccountStatementEffects = /** @class */ (function () {
        function AccountStatementEffects(actions$, dataService, store) {
            var _this = this;
            this.actions$ = actions$;
            this.dataService = dataService;
            this.store = store;
            this.accountStatementsBusiness$ = i1$1.createEffect(function () { return _this.actions$.pipe(i1$1.ofType(loadAccountStatementsBusiness), operators.withLatestFrom(_this.store.select(accountStatementsAccountIds), _this.store.select(accountStatementsFilterParams), _this.store.select(accountStatementsSortingParams), _this.store.select(accountStatementsPaginationParams), _this.store.select(accountStatementsWidgetPreferences)), operators.switchMap(function (_a) {
                var _b = __read(_a, 6), action = _b[0], accountIds = _b[1], filterParams = _b[2], sortingParams = _b[3], paginationParams = _b[4], widgetPreferences = _b[5];
                var append = action.append, actionRequestParams = __rest(action, ["append"]);
                return _this.dataService
                    .getAccountStatementsBusiness(loadAccountStatementsBusinessToRequest(Object.assign({ accountIds: accountIds, filterParams: filterParams, sortingParams: sortingParams, paginationParams: paginationParams }, actionRequestParams), widgetPreferences))
                    .pipe(operators.map(function (response) { return loadAccountStatementsBusinessSuccess(Object.assign({ accountStatements: response.body, totalCount: i2.getTotalCountFromHeaders(response.headers) }, action)); }), operators.catchError(function (error) { return rxjs.of(loadAccountStatementsFailure({ error: error })); }));
            })); });
        }
        return AccountStatementEffects;
    }());
    AccountStatementEffects.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: AccountStatementEffects, deps: [{ token: i1__namespace.Actions }, { token: i2__namespace.AccountStatementDataService }, { token: i1__namespace$1.Store }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    AccountStatementEffects.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: AccountStatementEffects });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: AccountStatementEffects, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () { return [{ type: i1__namespace.Actions }, { type: i2__namespace.AccountStatementDataService }, { type: i1__namespace$1.Store }]; } });

    var CategoriesEffects = /** @class */ (function () {
        function CategoriesEffects(actions$, dataService) {
            var _this = this;
            this.actions$ = actions$;
            this.dataService = dataService;
            this.loadCategories$ = i1$1.createEffect(function () { return _this.actions$.pipe(i1$1.ofType(loadCategories), operators.switchMap(function () { return _this.dataService.loadCategories().pipe(operators.map(function (response) { return loadCategoriesSuccess({ categories: response }); }), operators.catchError(function (error) { return rxjs.of(loadCategoriesFailure({ error: error })); })); })); });
        }
        return CategoriesEffects;
    }());
    CategoriesEffects.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: CategoriesEffects, deps: [{ token: i1__namespace.Actions }, { token: i2__namespace.AccountStatementDataService }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    CategoriesEffects.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: CategoriesEffects });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: CategoriesEffects, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () { return [{ type: i1__namespace.Actions }, { type: i2__namespace.AccountStatementDataService }]; } });

    var AccountStatementStoreModule = /** @class */ (function () {
        function AccountStatementStoreModule() {
        }
        return AccountStatementStoreModule;
    }());
    AccountStatementStoreModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: AccountStatementStoreModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    AccountStatementStoreModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: AccountStatementStoreModule, imports: [i1__namespace$1.StoreFeatureModule, i1__namespace.EffectsFeatureModule, i2.AccountStatementCommonModule] });
    AccountStatementStoreModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: AccountStatementStoreModule, imports: [[
                i1.StoreModule.forFeature(ACCOUNT_STATEMENTS_FEATURE_KEY, accountStatementsReducers),
                i1$1.EffectsModule.forFeature([AccountStatementEffects, CategoriesEffects]),
                i2.AccountStatementCommonModule,
            ]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: AccountStatementStoreModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [
                            i1.StoreModule.forFeature(ACCOUNT_STATEMENTS_FEATURE_KEY, accountStatementsReducers),
                            i1$1.EffectsModule.forFeature([AccountStatementEffects, CategoriesEffects]),
                            i2.AccountStatementCommonModule,
                        ],
                    }]
            }] });

    var selectCategoriesState = i1.createSelector(selectAccountStatements, function (accountStatementsState) { return accountStatementsState.categories; });
    var selectCategories = i1.createSelector(selectCategoriesState, function (state) { return state.data; });
    var selectCategoriesError = i1.createSelector(selectCategoriesState, function (state) { return state.error; });

    var AccountStatementStoreModel = /** @class */ (function () {
        function AccountStatementStoreModel(store) {
            this.store = store;
            this.accountStatements$ = this.store.select(accountStatements);
            this.accountStatementsLoading$ = this.store.select(accountStatementsLoading);
            this.accountStatementsError$ = this.store.select(accountStatementsError);
            this.accountStatementsTotalCount$ = this.store.select(accountStatementsTotalCount);
            this.accountStatementsPaginationParams$ = this.store.select(accountStatementsPaginationParams);
            this.accountStatementsSortingParams$ = this.store.select(accountStatementsSortingParams);
            this.accountStatementsFilterParams$ = this.store.select(accountStatementsFilterParams);
            this.accountStatementsWidgetPreferences$ = this.store.select(accountStatementsWidgetPreferences);
            this.categories$ = this.store.select(selectCategories);
        }
        AccountStatementStoreModel.prototype.dispatchLoadAccountStatementsBusiness = function (payload) {
            if (payload === void 0) { payload = {}; }
            this.store.dispatch(loadAccountStatementsBusiness(payload));
        };
        AccountStatementStoreModel.prototype.dispatchUpdateWidgetPreferences = function (payload) {
            this.store.dispatch(updateWidgetPreferences(payload));
        };
        AccountStatementStoreModel.prototype.dispatchLoadCategories = function () {
            this.store.dispatch(loadCategories());
        };
        return AccountStatementStoreModel;
    }());
    AccountStatementStoreModel.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: AccountStatementStoreModel, deps: [{ token: i1__namespace$1.Store }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    AccountStatementStoreModel.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: AccountStatementStoreModel });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: AccountStatementStoreModel, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () { return [{ type: i1__namespace$1.Store }]; } });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.AccountStatementStoreModel = AccountStatementStoreModel;
    exports.AccountStatementStoreModule = AccountStatementStoreModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
