// @backbase/ui-ang/esm2015/_example/checkbox-group/showcase/checkbox-group-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CheckboxGroupShowcaseComponent } from './checkbox-group-showcase.component';
import { CheckboxGroupModule } from '@backbase/ui-ang/checkbox-group';
import { InputCheckboxModule } from '@backbase/ui-ang/input-checkbox';
import { FormsModule } from '@angular/forms';
import * as i0 from "@angular/core";
export class CheckboxGroupShowcaseModule {
}
CheckboxGroupShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CheckboxGroupShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CheckboxGroupShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CheckboxGroupShowcaseModule, declarations: [CheckboxGroupShowcaseComponent], imports: [CommonModule, CheckboxGroupModule, FormsModule, InputCheckboxModule] });
CheckboxGroupShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CheckboxGroupShowcaseModule, imports: [[CommonModule, CheckboxGroupModule, FormsModule, InputCheckboxModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CheckboxGroupShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, CheckboxGroupModule, FormsModule, InputCheckboxModule],
                    declarations: [CheckboxGroupShowcaseComponent],
                    entryComponents: [CheckboxGroupShowcaseComponent],
                }]
        }] });
