// @backbase/actions-product-notifications-settings-widget-ang/esm2015/src/components/actions-product-item-card/actions-product-item-card.component.js
import { Component, Directive, Input } from '@angular/core';
import { BbTemplate } from '@backbase/foundation-ang/core';
import { ProductKindNames } from '../../models/product-kind-names';
import { firstNonEmptyString } from '@backbase/actions-common-ang';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/payment-card-number-pipe";
import * as i2 from "@backbase/ui-ang/iban-pipe";
export class ActionsProductItemCardComponent {
    constructor() {
        /**
         * Product kind names enum.
         */
        this.productKindNames = ProductKindNames;
        /**
         * Configuration that determines how a payment card number should be formatted.
         */
        this.productNumberFormat = {
            length: 0,
            maskRange: [0, 12],
            segments: 4,
        };
    }
    /**
     * Method to get account number.
     *
     * @param args - accounts array
     */
    retrieveAccountNumber(...args) {
        const account = args[0];
        switch (account.productKindName) {
            case ProductKindNames.TermDeposit:
                return firstNonEmptyString(account.IBAN);
            case ProductKindNames.Loan:
            case ProductKindNames.CreditCard:
            case ProductKindNames.InvestmentAccount:
                return firstNonEmptyString(account.creditCardAccountNumber, account.IBAN, account.BBAN);
            case ProductKindNames.DebitCard:
                return firstNonEmptyString(account.creditCardAccountNumber);
            default:
                return firstNonEmptyString(account.IBAN, account.BBAN);
        }
    }
    /**
     * Method to get account alias.
     *
     * @param account - account data
     * @returns account alias name
     */
    getAlias(account) {
        return firstNonEmptyString(account.userPreferences && account.userPreferences.alias, account.bankAlias, account.name);
    }
}
ActionsProductItemCardComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsProductItemCardComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ActionsProductItemCardComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.14", type: ActionsProductItemCardComponent, selector: "bb-actions-product-item-card", inputs: { account: "account" }, ngImport: i0, template: "<ng-container bbActionsProductItemCardCustomizableDirective [bbHostRef]=\"this\" [bbTemplateContext]=\"account\">\n</ng-container>\n\n<ng-template bbActionsProductItemCardCustomizableDirective let-hostRef let-account=\"context\">\n  <ng-container>\n    <div>\n      <div data-role=\"card-title\" class=\"bb-text-bold\" role=\"heading\" aria-level=\"2\">\n        {{ hostRef.getAlias(account) }}\n      </div>\n      <div\n        data-role=\"card-sub-title\"\n        class=\"text-small bb-text-support bb-block bb-block--xl\"\n        role=\"heading\"\n        aria-level=\"3\"\n      >\n        {{ hostRef.retrieveAccountNumber(account) | bbIban | paymentCardNumber: productNumberFormat }}\n      </div>\n    </div>\n  </ng-container>\n</ng-template>\n", directives: [{ type: i0.forwardRef(function () { return ActionsProductItemCardCustomizableDirective; }), selector: "[bbActionsProductItemCardCustomizableDirective]" }], pipes: { "paymentCardNumber": i0.forwardRef(function () { return i1.PaymentCardNumberPipe; }), "bbIban": i0.forwardRef(function () { return i2.BbIbanPipe; }) } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsProductItemCardComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-actions-product-item-card',
                    templateUrl: './actions-product-item-card.component.html',
                }]
        }], propDecorators: { account: [{
                type: Input
            }] } });
export class ActionsProductItemCardCustomizableDirective extends BbTemplate {
}
ActionsProductItemCardCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsProductItemCardCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
ActionsProductItemCardCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.14", type: ActionsProductItemCardCustomizableDirective, selector: "[bbActionsProductItemCardCustomizableDirective]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsProductItemCardCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbActionsProductItemCardCustomizableDirective]',
                }]
        }] });
