// @backbase/ui-ang/esm2015/_example/file-attachment/showcase/file-attachment-showcase.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { of } from 'rxjs';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/file-attachment";
import * as i2 from "@angular/common";
const attachment = {
    name: 'examplePDF.pdf',
    deletable: false,
    disabled: false,
    size: 10000,
    loading: false,
    fileContent: of(new ArrayBuffer(2000000)),
};
let FileAttachmentShowcaseComponent = class FileAttachmentShowcaseComponent {
    constructor() {
        this.defaultFileAttachmentsList = [
            attachment,
            Object.assign(Object.assign({}, attachment), { name: 'exampleXLS.xls' }),
            Object.assign(Object.assign({}, attachment), { name: 'exampleCSV.csv' }),
            Object.assign(Object.assign({}, attachment), { name: 'examplePNG.png' }),
            Object.assign(Object.assign({}, attachment), { name: 'exampleUnknown' }),
        ];
        this.deletableFileAttachmentsList = this.defaultFileAttachmentsList.map((defaultAttachment) => (Object.assign(Object.assign({}, defaultAttachment), { deletable: true, disabled: true })));
    }
    deleteAttachment(filesList, index) {
        filesList[index].loading = true;
        setTimeout(() => {
            filesList[index].loading = false;
            filesList.splice(index, 1);
        }, 1000);
    }
};
FileAttachmentShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FileAttachmentShowcaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
FileAttachmentShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: FileAttachmentShowcaseComponent, selector: "bb-file-attachment-ui-showcase", ngImport: i0, template: "<h4>Default downloadable file attachments</h4>\n<div class=\"row\">\n  <div class=\"col-auto\" *ngFor=\"let attachment of defaultFileAttachmentsList\">\n    <div class=\"bb-block bb-block--md\">\n      <bb-file-attachment-ui\n        [name]=\"attachment.name\"\n        [size]=\"attachment.size\"\n        [loading]=\"attachment.loading\"\n        [fileContent]=\"attachment.fileContent\"\n      ></bb-file-attachment-ui>\n    </div>\n  </div>\n</div>\n<br />\n<h4>Disabled file attachments</h4>\n<div class=\"row\">\n  <div class=\"col-auto\" *ngFor=\"let attachment of defaultFileAttachmentsList\">\n    <div class=\"bb-block bb-block--md\">\n      <bb-file-attachment-ui disabled=\"true\" [name]=\"attachment.name\" [size]=\"attachment.size\"></bb-file-attachment-ui>\n    </div>\n  </div>\n</div>\n<br />\n<h4>Deletable file attachments with disabled download</h4>\n<div class=\"row\">\n  <div class=\"col-auto\" *ngFor=\"let attachment of deletableFileAttachmentsList; let i = index\">\n    <div class=\"bb-block bb-block--md\">\n      <bb-file-attachment-ui\n        [name]=\"attachment.name\"\n        [deletable]=\"attachment.deletable\"\n        [disabled]=\"attachment.disabled\"\n        [size]=\"attachment.size\"\n        [loading]=\"attachment.loading\"\n        (delete)=\"deleteAttachment(deletableFileAttachmentsList, i)\"\n        [fileContent]=\"attachment.fileContent\"\n      ></bb-file-attachment-ui>\n    </div>\n  </div>\n</div>\n<br />\n<h4>File attachments that fill their container</h4>\n<div class=\"row\">\n  <div class=\"col-md-4\" *ngFor=\"let attachment of defaultFileAttachmentsList; let i = index\">\n    <div class=\"bb-block bb-block--md\">\n      <bb-file-attachment-ui\n        [name]=\"attachment.name\"\n        [size]=\"attachment.size\"\n        [loading]=\"attachment.loading\"\n        [fileContent]=\"attachment.fileContent\"\n        [block]=\"true\"\n      ></bb-file-attachment-ui>\n    </div>\n  </div>\n</div>\n", components: [{ type: i1.FileAttachmentComponent, selector: "bb-file-attachment-ui", inputs: ["name", "size", "fileContent", "loading", "deletable", "disabled", "block"], outputs: ["delete"] }], directives: [{ type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }] });
FileAttachmentShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
], FileAttachmentShowcaseComponent);
export { FileAttachmentShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FileAttachmentShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-file-attachment-ui-showcase',
                    templateUrl: 'file-attachment-showcase.component.html',
                }]
        }] });
