// @backbase/ui-ang/esm2015/_example/file-attachment/showcase/file-attachment-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FileAttachmentShowcaseComponent } from './file-attachment-showcase.component';
import { IconModule } from '@backbase/ui-ang/icon';
import { FileAttachmentModule } from '@backbase/ui-ang/file-attachment';
import { ButtonModule } from '@backbase/ui-ang/button';
import { KeyboardClickModule } from '@backbase/ui-ang/keyboard-click-directive';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import * as i0 from "@angular/core";
const uiModules = [ButtonModule, IconModule, KeyboardClickModule, LoadingIndicatorModule];
export class FileAttachmentShowcaseModule {
}
FileAttachmentShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FileAttachmentShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
FileAttachmentShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FileAttachmentShowcaseModule, declarations: [FileAttachmentShowcaseComponent], imports: [CommonModule, FileAttachmentModule, ButtonModule, IconModule, KeyboardClickModule, LoadingIndicatorModule] });
FileAttachmentShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FileAttachmentShowcaseModule, imports: [[CommonModule, FileAttachmentModule, uiModules]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FileAttachmentShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, FileAttachmentModule, uiModules],
                    declarations: [FileAttachmentShowcaseComponent],
                    entryComponents: [FileAttachmentShowcaseComponent],
                }]
        }] });
