// @backbase/product-summary-list-widget-ang/esm2015/src/product-summary-base.service.js
import { Injectable } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import * as i0 from "@angular/core";
/**
 * Service for fetching and storing product kinds
 *
 * This service relies on providers from `ProductSummaryListWidgetModule`.
 *
 * @see ProductSummaryListWidgetModule
 *
 * @usageNotes
 *
 * ### Ensure the ProductSummaryListWidgetModule is imported to your component module
 *
 * ```ts
 * @NgModule({
 *   ...
 *   imports: [
 *     ...
 *     ProductSummaryListWidgetModule,
 *   ],
 * })
 * export class MyWidgetModule {}
 * ```
 *
 * ### Inject this service into your component
 *
 * ```ts
 * @Component({
 *   ...
 *   providers: [ProductSummaryService],
 * })
 *  export class MyComponent {
 * ```
 */
export class ProductSummaryBaseService {
    constructor() {
        this.listParams = new BehaviorSubject({});
        this.productKindsRefresh = new BehaviorSubject(false);
        /**
         * Used to store the error occured during loading of accounts
         */
        this.error = new BehaviorSubject(undefined);
        /**
         * List of accounts fetched from the server
         */
        this.isLoading = new BehaviorSubject(false);
    }
    /**
     * @internal
     */
    ngOnDestroy() {
        this.error.complete();
        this.productKindsRefresh.complete();
    }
    /**
     * Refreshes the list of product kinds
     * Causes refetching data
     */
    refreshProductKinds() {
        this.productKindsRefresh.next(true);
    }
}
ProductSummaryBaseService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryBaseService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
ProductSummaryBaseService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryBaseService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryBaseService, decorators: [{
            type: Injectable
        }] });
