// @backbase/product-summary-list-widget-ang/esm2015/src/product-summary-list-widget.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { IconModule } from '@backbase/ui-ang/icon';
import { AmountModule } from '@backbase/ui-ang/amount';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { ProductKindModule, ProductKindCollapsibleModule, ProductNeedsAttentionModule, } from '@backbase/product-summary-ui';
import { ProductsExternalInfoService } from '@backbase/product-summary-common-ang';
import { ProductSummaryListWidgetComponent } from './product-summary-list/product-summary-list-widget.component';
import { ArrangementsListWidgetComponent, ProductSummaryListItemNeedsAttentionAccountsCustomizableDirective, } from './arrangements-list/arrangements-list-widget.component';
import { ProductSummaryBodyCustomizableDirective, ProductSummaryCustomizableDirective, ProductSummaryHeaderCustomizableDirective, ProductSummaryListItemCreditCardCustomizableDirective, ProductSummaryListItemCurrentAccountCustomizableDirective, ProductSummaryListItemCustomizableDirective, ProductSummaryListItemDebitCardCustomizableDirective, ProductSummaryListItemInvestmentAccountCustomizableDirective, ProductSummaryListItemLoanCustomizableDirective, ProductSummaryListItemOtherProductKindsCustomizableDirective, ProductSummaryListItemSavingsAccountCustomizableDirective, ProductSummaryListItemTermDepositCustomizableDirective, ProductSummaryListLoadingCustomizableDirective, } from './product-summary-list-base/product-summary-list-base.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const components = [ProductSummaryListWidgetComponent, ArrangementsListWidgetComponent];
const directives = [
    ProductSummaryCustomizableDirective,
    ProductSummaryHeaderCustomizableDirective,
    ProductSummaryBodyCustomizableDirective,
    ProductSummaryListItemCustomizableDirective,
    ProductSummaryListItemNeedsAttentionAccountsCustomizableDirective,
    ProductSummaryListItemCurrentAccountCustomizableDirective,
    ProductSummaryListItemSavingsAccountCustomizableDirective,
    ProductSummaryListItemTermDepositCustomizableDirective,
    ProductSummaryListItemLoanCustomizableDirective,
    ProductSummaryListItemCreditCardCustomizableDirective,
    ProductSummaryListItemDebitCardCustomizableDirective,
    ProductSummaryListItemInvestmentAccountCustomizableDirective,
    ProductSummaryListItemOtherProductKindsCustomizableDirective,
    ProductSummaryListLoadingCustomizableDirective,
];
const uiModules = [
    ProductKindModule,
    ProductKindCollapsibleModule,
    IconModule,
    AmountModule,
    EmptyStateModule,
    LoadingIndicatorModule,
    TooltipModule,
    ProductNeedsAttentionModule,
];
/**
 * Exports the Widgets and UI components for showing a list of accounts
 *
 * The following widgets are included in this module:
 *  - `ProductSummaryListWidgetComponent`
 *  - `ArrangementsViewWidgetComponent`
 *
 * @see ProductSummaryListWidgetComponent
 * @see ArrangementsListWidgetComponent
 *
 * @usageNotes
 *
 * ### Include the exported widgets in your `AppModule`:
 *
 * ```typescript
 * @NgModule({
 *   imports: [
 *     ProductSummaryListWidgetModule,
 *   ],
 *   ...
 * })
 * export class AppModule {}
 * ```
 */
export class ProductSummaryListWidgetModule {
}
ProductSummaryListWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryListWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductSummaryListWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryListWidgetModule, declarations: [ProductSummaryListWidgetComponent, ArrangementsListWidgetComponent, ProductSummaryCustomizableDirective,
        ProductSummaryHeaderCustomizableDirective,
        ProductSummaryBodyCustomizableDirective,
        ProductSummaryListItemCustomizableDirective,
        ProductSummaryListItemNeedsAttentionAccountsCustomizableDirective,
        ProductSummaryListItemCurrentAccountCustomizableDirective,
        ProductSummaryListItemSavingsAccountCustomizableDirective,
        ProductSummaryListItemTermDepositCustomizableDirective,
        ProductSummaryListItemLoanCustomizableDirective,
        ProductSummaryListItemCreditCardCustomizableDirective,
        ProductSummaryListItemDebitCardCustomizableDirective,
        ProductSummaryListItemInvestmentAccountCustomizableDirective,
        ProductSummaryListItemOtherProductKindsCustomizableDirective,
        ProductSummaryListLoadingCustomizableDirective], imports: [CommonModule, ProductKindModule,
        ProductKindCollapsibleModule,
        IconModule,
        AmountModule,
        EmptyStateModule,
        LoadingIndicatorModule,
        TooltipModule,
        ProductNeedsAttentionModule, i1.BackbaseCoreModule], exports: [ProductSummaryListWidgetComponent, ArrangementsListWidgetComponent, ProductSummaryCustomizableDirective,
        ProductSummaryHeaderCustomizableDirective,
        ProductSummaryBodyCustomizableDirective,
        ProductSummaryListItemCustomizableDirective,
        ProductSummaryListItemNeedsAttentionAccountsCustomizableDirective,
        ProductSummaryListItemCurrentAccountCustomizableDirective,
        ProductSummaryListItemSavingsAccountCustomizableDirective,
        ProductSummaryListItemTermDepositCustomizableDirective,
        ProductSummaryListItemLoanCustomizableDirective,
        ProductSummaryListItemCreditCardCustomizableDirective,
        ProductSummaryListItemDebitCardCustomizableDirective,
        ProductSummaryListItemInvestmentAccountCustomizableDirective,
        ProductSummaryListItemOtherProductKindsCustomizableDirective,
        ProductSummaryListLoadingCustomizableDirective, ProductKindModule,
        ProductKindCollapsibleModule,
        IconModule,
        AmountModule,
        EmptyStateModule,
        LoadingIndicatorModule,
        TooltipModule,
        ProductNeedsAttentionModule] });
ProductSummaryListWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryListWidgetModule, providers: [ProductsExternalInfoService], imports: [[
            CommonModule,
            ...uiModules,
            BackbaseCoreModule.withConfig({
                classMap: {
                    ProductSummaryListWidgetComponent,
                    ArrangementsListWidgetComponent,
                },
            }),
        ], ProductKindModule,
        ProductKindCollapsibleModule,
        IconModule,
        AmountModule,
        EmptyStateModule,
        LoadingIndicatorModule,
        TooltipModule,
        ProductNeedsAttentionModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryListWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        ...uiModules,
                        BackbaseCoreModule.withConfig({
                            classMap: {
                                ProductSummaryListWidgetComponent,
                                ArrangementsListWidgetComponent,
                            },
                        }),
                    ],
                    declarations: [...components, ...directives],
                    exports: [...components, ...directives, ...uiModules],
                    providers: [ProductsExternalInfoService],
                }]
        }] });
