// @backbase/payment-batch-template-http-ang/esm2015/api/batchTemplates.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { PAYMENT_BATCH_TEMPLATE_BASE_PATH } from '../variables';
import { PaymentBatchTemplateConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class BatchTemplatesHttpService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new PaymentBatchTemplateConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    createBatchOrderFromTemplate(requestParameters, observe = 'body', reportProgress = false, options) {
        const _batchTemplateId = requestParameters["batchTemplateId"];
        if (_batchTemplateId === null || _batchTemplateId === undefined) {
            throw new Error('Required parameter batchTemplateId was null or undefined when calling createBatchOrderFromTemplate.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v1/batch-templates/${encodeURIComponent(String(_batchTemplateId))}/batch-order`, null, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    createBatchOrderFromTemplateUrl(requestParameters) {
        const _batchTemplateId = requestParameters["batchTemplateId"];
        if (_batchTemplateId === null || _batchTemplateId === undefined) {
            throw new Error('Required parameter batchTemplateId was null or undefined when calling createBatchOrderFromTemplate.');
        }
        return `${this.configuration.basePath}/client-api/v1/batch-templates/${encodeURIComponent(String(_batchTemplateId))}/batch-order`;
    }
    deleteBatchTemplateById(requestParameters, observe = 'body', reportProgress = false, options) {
        const _batchTemplateId = requestParameters["batchTemplateId"];
        if (_batchTemplateId === null || _batchTemplateId === undefined) {
            throw new Error('Required parameter batchTemplateId was null or undefined when calling deleteBatchTemplateById.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.delete(`${this.configuration.basePath}/client-api/v1/batch-templates/${encodeURIComponent(String(_batchTemplateId))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    deleteBatchTemplateByIdUrl(requestParameters) {
        const _batchTemplateId = requestParameters["batchTemplateId"];
        if (_batchTemplateId === null || _batchTemplateId === undefined) {
            throw new Error('Required parameter batchTemplateId was null or undefined when calling deleteBatchTemplateById.');
        }
        return `${this.configuration.basePath}/client-api/v1/batch-templates/${encodeURIComponent(String(_batchTemplateId))}`;
    }
    deleteBatchTemplatePaymentById(requestParameters, observe = 'body', reportProgress = false, options) {
        const _batchTemplateId = requestParameters["batchTemplateId"];
        if (_batchTemplateId === null || _batchTemplateId === undefined) {
            throw new Error('Required parameter batchTemplateId was null or undefined when calling deleteBatchTemplatePaymentById.');
        }
        const _batchTemplatePaymentId = requestParameters["batchTemplatePaymentId"];
        if (_batchTemplatePaymentId === null || _batchTemplatePaymentId === undefined) {
            throw new Error('Required parameter batchTemplatePaymentId was null or undefined when calling deleteBatchTemplatePaymentById.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.delete(`${this.configuration.basePath}/client-api/v1/batch-templates/${encodeURIComponent(String(_batchTemplateId))}/batch-template-payments/${encodeURIComponent(String(_batchTemplatePaymentId))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    deleteBatchTemplatePaymentByIdUrl(requestParameters) {
        const _batchTemplateId = requestParameters["batchTemplateId"];
        if (_batchTemplateId === null || _batchTemplateId === undefined) {
            throw new Error('Required parameter batchTemplateId was null or undefined when calling deleteBatchTemplatePaymentById.');
        }
        const _batchTemplatePaymentId = requestParameters["batchTemplatePaymentId"];
        if (_batchTemplatePaymentId === null || _batchTemplatePaymentId === undefined) {
            throw new Error('Required parameter batchTemplatePaymentId was null or undefined when calling deleteBatchTemplatePaymentById.');
        }
        return `${this.configuration.basePath}/client-api/v1/batch-templates/${encodeURIComponent(String(_batchTemplateId))}/batch-template-payments/${encodeURIComponent(String(_batchTemplatePaymentId))}`;
    }
    getBatchTemplatePaymentsById(requestParameters, observe = 'body', reportProgress = false, options) {
        const _batchTemplateId = requestParameters["batchTemplateId"];
        if (_batchTemplateId === null || _batchTemplateId === undefined) {
            throw new Error('Required parameter batchTemplateId was null or undefined when calling getBatchTemplatePaymentsById.');
        }
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        const _privilegesToCheck = requestParameters["privilegesToCheck"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_privilegesToCheck) {
            _privilegesToCheck.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'privilegesToCheck');
            });
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/batch-templates/${encodeURIComponent(String(_batchTemplateId))}/batch-template-payments`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getBatchTemplatePaymentsByIdUrl(requestParameters) {
        const _batchTemplateId = requestParameters["batchTemplateId"];
        if (_batchTemplateId === null || _batchTemplateId === undefined) {
            throw new Error('Required parameter batchTemplateId was null or undefined when calling getBatchTemplatePaymentsById.');
        }
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        const _privilegesToCheck = requestParameters["privilegesToCheck"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_privilegesToCheck) {
            _privilegesToCheck.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'privilegesToCheck');
            });
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v1/batch-templates/${encodeURIComponent(String(_batchTemplateId))}/batch-template-payments${queryString ? `?${queryString}` : ''}`;
    }
    getBatchTemplates(requestParameters, observe = 'body', reportProgress = false, options) {
        const _query = requestParameters["query"];
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        const _batchType = requestParameters["batchType"];
        const _creditDebitMixedIndicator = requestParameters["creditDebitMixedIndicator"];
        const _privilegesToCheck = requestParameters["privilegesToCheck"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_query !== undefined && _query !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _query, 'query');
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_batchType !== undefined && _batchType !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _batchType, 'batchType');
        }
        if (_creditDebitMixedIndicator !== undefined && _creditDebitMixedIndicator !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _creditDebitMixedIndicator, 'creditDebitMixedIndicator');
        }
        if (_privilegesToCheck) {
            _privilegesToCheck.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'privilegesToCheck');
            });
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/batch-templates`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getBatchTemplatesUrl(requestParameters) {
        const _query = requestParameters["query"];
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        const _batchType = requestParameters["batchType"];
        const _creditDebitMixedIndicator = requestParameters["creditDebitMixedIndicator"];
        const _privilegesToCheck = requestParameters["privilegesToCheck"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_query !== undefined && _query !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _query, 'query');
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_batchType !== undefined && _batchType !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _batchType, 'batchType');
        }
        if (_creditDebitMixedIndicator !== undefined && _creditDebitMixedIndicator !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _creditDebitMixedIndicator, 'creditDebitMixedIndicator');
        }
        if (_privilegesToCheck) {
            _privilegesToCheck.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'privilegesToCheck');
            });
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v1/batch-templates${queryString ? `?${queryString}` : ''}`;
    }
    postBatchTemplatePayments(requestParameters, observe = 'body', reportProgress = false, options) {
        const _batchTemplateId = requestParameters["batchTemplateId"];
        if (_batchTemplateId === null || _batchTemplateId === undefined) {
            throw new Error('Required parameter batchTemplateId was null or undefined when calling postBatchTemplatePayments.');
        }
        const _batchTemplatePaymentsPostRequest = requestParameters["batchTemplatePaymentsPostRequest"];
        if (_batchTemplatePaymentsPostRequest === null || _batchTemplatePaymentsPostRequest === undefined) {
            throw new Error('Required parameter batchTemplatePaymentsPostRequest was null or undefined when calling postBatchTemplatePayments.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v1/batch-templates/${encodeURIComponent(String(_batchTemplateId))}/batch-template-payments`, _batchTemplatePaymentsPostRequest, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    postBatchTemplatePaymentsUrl(requestParameters) {
        const _batchTemplateId = requestParameters["batchTemplateId"];
        if (_batchTemplateId === null || _batchTemplateId === undefined) {
            throw new Error('Required parameter batchTemplateId was null or undefined when calling postBatchTemplatePayments.');
        }
        const _batchTemplatePaymentsPostRequest = requestParameters["batchTemplatePaymentsPostRequest"];
        if (_batchTemplatePaymentsPostRequest === null || _batchTemplatePaymentsPostRequest === undefined) {
            throw new Error('Required parameter batchTemplatePaymentsPostRequest was null or undefined when calling postBatchTemplatePayments.');
        }
        return `${this.configuration.basePath}/client-api/v1/batch-templates/${encodeURIComponent(String(_batchTemplateId))}/batch-template-payments`;
    }
    postBatchTemplates(requestParameters, observe = 'body', reportProgress = false, options) {
        const _draftOf = requestParameters["draftOf"];
        const _copyFrom = requestParameters["copyFrom"];
        const _name = requestParameters["name"];
        const _batchTemplatesPostRequest = requestParameters["batchTemplatesPostRequest"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_draftOf !== undefined && _draftOf !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _draftOf, 'draftOf');
        }
        if (_copyFrom !== undefined && _copyFrom !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _copyFrom, 'copyFrom');
        }
        if (_name !== undefined && _name !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _name, 'name');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v1/batch-templates`, _batchTemplatesPostRequest, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    postBatchTemplatesUrl(requestParameters) {
        const _draftOf = requestParameters["draftOf"];
        const _copyFrom = requestParameters["copyFrom"];
        const _name = requestParameters["name"];
        const _batchTemplatesPostRequest = requestParameters["batchTemplatesPostRequest"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_draftOf !== undefined && _draftOf !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _draftOf, 'draftOf');
        }
        if (_copyFrom !== undefined && _copyFrom !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _copyFrom, 'copyFrom');
        }
        if (_name !== undefined && _name !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _name, 'name');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v1/batch-templates${queryString ? `?${queryString}` : ''}`;
    }
    postBatchTemplatesFromAttributes(requestParameters, observe = 'body', reportProgress = false, options) {
        const _batchTemplatesAttributesPostRequest = requestParameters["batchTemplatesAttributesPostRequest"];
        if (_batchTemplatesAttributesPostRequest === null || _batchTemplatesAttributesPostRequest === undefined) {
            throw new Error('Required parameter batchTemplatesAttributesPostRequest was null or undefined when calling postBatchTemplatesFromAttributes.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v1/batch-templates/from-attributes`, _batchTemplatesAttributesPostRequest, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    postBatchTemplatesFromAttributesUrl(requestParameters) {
        const _batchTemplatesAttributesPostRequest = requestParameters["batchTemplatesAttributesPostRequest"];
        if (_batchTemplatesAttributesPostRequest === null || _batchTemplatesAttributesPostRequest === undefined) {
            throw new Error('Required parameter batchTemplatesAttributesPostRequest was null or undefined when calling postBatchTemplatesFromAttributes.');
        }
        return `${this.configuration.basePath}/client-api/v1/batch-templates/from-attributes`;
    }
    putBatchTemplatePaymentById(requestParameters, observe = 'body', reportProgress = false, options) {
        const _batchTemplateId = requestParameters["batchTemplateId"];
        if (_batchTemplateId === null || _batchTemplateId === undefined) {
            throw new Error('Required parameter batchTemplateId was null or undefined when calling putBatchTemplatePaymentById.');
        }
        const _batchTemplatePaymentId = requestParameters["batchTemplatePaymentId"];
        if (_batchTemplatePaymentId === null || _batchTemplatePaymentId === undefined) {
            throw new Error('Required parameter batchTemplatePaymentId was null or undefined when calling putBatchTemplatePaymentById.');
        }
        const _batchTemplatePaymentsPutRequest = requestParameters["batchTemplatePaymentsPutRequest"];
        if (_batchTemplatePaymentsPutRequest === null || _batchTemplatePaymentsPutRequest === undefined) {
            throw new Error('Required parameter batchTemplatePaymentsPutRequest was null or undefined when calling putBatchTemplatePaymentById.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v1/batch-templates/${encodeURIComponent(String(_batchTemplateId))}/batch-template-payments/${encodeURIComponent(String(_batchTemplatePaymentId))}`, _batchTemplatePaymentsPutRequest, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    putBatchTemplatePaymentByIdUrl(requestParameters) {
        const _batchTemplateId = requestParameters["batchTemplateId"];
        if (_batchTemplateId === null || _batchTemplateId === undefined) {
            throw new Error('Required parameter batchTemplateId was null or undefined when calling putBatchTemplatePaymentById.');
        }
        const _batchTemplatePaymentId = requestParameters["batchTemplatePaymentId"];
        if (_batchTemplatePaymentId === null || _batchTemplatePaymentId === undefined) {
            throw new Error('Required parameter batchTemplatePaymentId was null or undefined when calling putBatchTemplatePaymentById.');
        }
        const _batchTemplatePaymentsPutRequest = requestParameters["batchTemplatePaymentsPutRequest"];
        if (_batchTemplatePaymentsPutRequest === null || _batchTemplatePaymentsPutRequest === undefined) {
            throw new Error('Required parameter batchTemplatePaymentsPutRequest was null or undefined when calling putBatchTemplatePaymentById.');
        }
        return `${this.configuration.basePath}/client-api/v1/batch-templates/${encodeURIComponent(String(_batchTemplateId))}/batch-template-payments/${encodeURIComponent(String(_batchTemplatePaymentId))}`;
    }
    updateBatchTemplateById(requestParameters, observe = 'body', reportProgress = false, options) {
        const _batchTemplateId = requestParameters["batchTemplateId"];
        if (_batchTemplateId === null || _batchTemplateId === undefined) {
            throw new Error('Required parameter batchTemplateId was null or undefined when calling updateBatchTemplateById.');
        }
        const _draftTemplateId = requestParameters["draftTemplateId"];
        const _batchTemplatesPutRequest = requestParameters["batchTemplatesPutRequest"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_draftTemplateId !== undefined && _draftTemplateId !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _draftTemplateId, 'draftTemplateId');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v1/batch-templates/${encodeURIComponent(String(_batchTemplateId))}`, _batchTemplatesPutRequest, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    updateBatchTemplateByIdUrl(requestParameters) {
        const _batchTemplateId = requestParameters["batchTemplateId"];
        if (_batchTemplateId === null || _batchTemplateId === undefined) {
            throw new Error('Required parameter batchTemplateId was null or undefined when calling updateBatchTemplateById.');
        }
        const _draftTemplateId = requestParameters["draftTemplateId"];
        const _batchTemplatesPutRequest = requestParameters["batchTemplatesPutRequest"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_draftTemplateId !== undefined && _draftTemplateId !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _draftTemplateId, 'draftTemplateId');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v1/batch-templates/${encodeURIComponent(String(_batchTemplateId))}${queryString ? `?${queryString}` : ''}`;
    }
    updateBatchTemplateStatusById(requestParameters, observe = 'body', reportProgress = false, options) {
        const _batchTemplateId = requestParameters["batchTemplateId"];
        if (_batchTemplateId === null || _batchTemplateId === undefined) {
            throw new Error('Required parameter batchTemplateId was null or undefined when calling updateBatchTemplateStatusById.');
        }
        const _batchTemplatesStatusUpdateRequest = requestParameters["batchTemplatesStatusUpdateRequest"];
        if (_batchTemplatesStatusUpdateRequest === null || _batchTemplatesStatusUpdateRequest === undefined) {
            throw new Error('Required parameter batchTemplatesStatusUpdateRequest was null or undefined when calling updateBatchTemplateStatusById.');
        }
        const _dryRun = requestParameters["dryRun"];
        const _intendedCreditTransactionsCount = requestParameters["intendedCreditTransactionsCount"];
        const _intendedDebitTransactionsCount = requestParameters["intendedDebitTransactionsCount"];
        const _intendedCreditAmount = requestParameters["intendedCreditAmount"];
        const _intendedDebitAmount = requestParameters["intendedDebitAmount"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_dryRun !== undefined && _dryRun !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _dryRun, 'dryRun');
        }
        if (_intendedCreditTransactionsCount !== undefined && _intendedCreditTransactionsCount !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _intendedCreditTransactionsCount, 'intendedCreditTransactionsCount');
        }
        if (_intendedDebitTransactionsCount !== undefined && _intendedDebitTransactionsCount !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _intendedDebitTransactionsCount, 'intendedDebitTransactionsCount');
        }
        if (_intendedCreditAmount !== undefined && _intendedCreditAmount !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _intendedCreditAmount, 'intendedCreditAmount');
        }
        if (_intendedDebitAmount !== undefined && _intendedDebitAmount !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _intendedDebitAmount, 'intendedDebitAmount');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v1/batch-templates/${encodeURIComponent(String(_batchTemplateId))}/status`, _batchTemplatesStatusUpdateRequest, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    updateBatchTemplateStatusByIdUrl(requestParameters) {
        const _batchTemplateId = requestParameters["batchTemplateId"];
        if (_batchTemplateId === null || _batchTemplateId === undefined) {
            throw new Error('Required parameter batchTemplateId was null or undefined when calling updateBatchTemplateStatusById.');
        }
        const _batchTemplatesStatusUpdateRequest = requestParameters["batchTemplatesStatusUpdateRequest"];
        if (_batchTemplatesStatusUpdateRequest === null || _batchTemplatesStatusUpdateRequest === undefined) {
            throw new Error('Required parameter batchTemplatesStatusUpdateRequest was null or undefined when calling updateBatchTemplateStatusById.');
        }
        const _dryRun = requestParameters["dryRun"];
        const _intendedCreditTransactionsCount = requestParameters["intendedCreditTransactionsCount"];
        const _intendedDebitTransactionsCount = requestParameters["intendedDebitTransactionsCount"];
        const _intendedCreditAmount = requestParameters["intendedCreditAmount"];
        const _intendedDebitAmount = requestParameters["intendedDebitAmount"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_dryRun !== undefined && _dryRun !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _dryRun, 'dryRun');
        }
        if (_intendedCreditTransactionsCount !== undefined && _intendedCreditTransactionsCount !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _intendedCreditTransactionsCount, 'intendedCreditTransactionsCount');
        }
        if (_intendedDebitTransactionsCount !== undefined && _intendedDebitTransactionsCount !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _intendedDebitTransactionsCount, 'intendedDebitTransactionsCount');
        }
        if (_intendedCreditAmount !== undefined && _intendedCreditAmount !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _intendedCreditAmount, 'intendedCreditAmount');
        }
        if (_intendedDebitAmount !== undefined && _intendedDebitAmount !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _intendedDebitAmount, 'intendedDebitAmount');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v1/batch-templates/${encodeURIComponent(String(_batchTemplateId))}/status${queryString ? `?${queryString}` : ''}`;
    }
    validateBatchTemplate(requestParameters, observe = 'body', reportProgress = false, options) {
        const _batchTemplateId = requestParameters["batchTemplateId"];
        if (_batchTemplateId === null || _batchTemplateId === undefined) {
            throw new Error('Required parameter batchTemplateId was null or undefined when calling validateBatchTemplate.');
        }
        const _intendedCreditTransactionsCount = requestParameters["intendedCreditTransactionsCount"];
        const _intendedDebitTransactionsCount = requestParameters["intendedDebitTransactionsCount"];
        const _intendedCreditAmount = requestParameters["intendedCreditAmount"];
        const _intendedDebitAmount = requestParameters["intendedDebitAmount"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_intendedCreditTransactionsCount !== undefined && _intendedCreditTransactionsCount !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _intendedCreditTransactionsCount, 'intendedCreditTransactionsCount');
        }
        if (_intendedDebitTransactionsCount !== undefined && _intendedDebitTransactionsCount !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _intendedDebitTransactionsCount, 'intendedDebitTransactionsCount');
        }
        if (_intendedCreditAmount !== undefined && _intendedCreditAmount !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _intendedCreditAmount, 'intendedCreditAmount');
        }
        if (_intendedDebitAmount !== undefined && _intendedDebitAmount !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _intendedDebitAmount, 'intendedDebitAmount');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/batch-templates/${encodeURIComponent(String(_batchTemplateId))}/validation-errors`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    validateBatchTemplateUrl(requestParameters) {
        const _batchTemplateId = requestParameters["batchTemplateId"];
        if (_batchTemplateId === null || _batchTemplateId === undefined) {
            throw new Error('Required parameter batchTemplateId was null or undefined when calling validateBatchTemplate.');
        }
        const _intendedCreditTransactionsCount = requestParameters["intendedCreditTransactionsCount"];
        const _intendedDebitTransactionsCount = requestParameters["intendedDebitTransactionsCount"];
        const _intendedCreditAmount = requestParameters["intendedCreditAmount"];
        const _intendedDebitAmount = requestParameters["intendedDebitAmount"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_intendedCreditTransactionsCount !== undefined && _intendedCreditTransactionsCount !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _intendedCreditTransactionsCount, 'intendedCreditTransactionsCount');
        }
        if (_intendedDebitTransactionsCount !== undefined && _intendedDebitTransactionsCount !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _intendedDebitTransactionsCount, 'intendedDebitTransactionsCount');
        }
        if (_intendedCreditAmount !== undefined && _intendedCreditAmount !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _intendedCreditAmount, 'intendedCreditAmount');
        }
        if (_intendedDebitAmount !== undefined && _intendedDebitAmount !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _intendedDebitAmount, 'intendedDebitAmount');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v1/batch-templates/${encodeURIComponent(String(_batchTemplateId))}/validation-errors${queryString ? `?${queryString}` : ''}`;
    }
}
BatchTemplatesHttpService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BatchTemplatesHttpService, deps: [{ token: i1.HttpClient }, { token: PAYMENT_BATCH_TEMPLATE_BASE_PATH, optional: true }, { token: i2.PaymentBatchTemplateConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
BatchTemplatesHttpService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BatchTemplatesHttpService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BatchTemplatesHttpService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [PAYMENT_BATCH_TEMPLATE_BASE_PATH]
                }] }, { type: i2.PaymentBatchTemplateConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
