// @backbase/ui-ang/esm2015/_example/card-vendor/showcase/card-vendor-showcase.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/card-vendor";
let CardVendorShowcaseComponent = class CardVendorShowcaseComponent {
};
CardVendorShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardVendorShowcaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
CardVendorShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: CardVendorShowcaseComponent, selector: "bb-card-vendor-ui-showcase", ngImport: i0, template: "<div class=\"bb-stack bb-stack--center\">\n  <div [style.width.px]=\"39\" class=\"bb-stack__item\">\n    <bb-card-vendor-ui vendor=\"mastercard\"></bb-card-vendor-ui>\n  </div>\n  <div [style.width.px]=\"48\" class=\"bb-stack__item\">\n    <bb-card-vendor-ui vendor=\"visa-debit\"></bb-card-vendor-ui>\n  </div>\n  <div [style.width.px]=\"38\" class=\"bb-stack__item\">\n    <bb-card-vendor-ui vendor=\"union-pay\"></bb-card-vendor-ui>\n  </div>\n  <div [style.width.px]=\"39\" class=\"bb-stack__item\">\n    <bb-card-vendor-ui vendor=\"maestro\"></bb-card-vendor-ui>\n  </div>\n  <div [style.width.px]=\"31\" class=\"bb-stack__item\">\n    <bb-card-vendor-ui vendor=\"diners-club\"></bb-card-vendor-ui>\n  </div>\n  <div [style.width.px]=\"80\" class=\"bb-stack__item\">\n    <bb-card-vendor-ui vendor=\"visa\"></bb-card-vendor-ui>\n  </div>\n</div>\n", components: [{ type: i1.CardVendorComponent, selector: "bb-card-vendor-ui", inputs: ["vendor"] }] });
CardVendorShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
], CardVendorShowcaseComponent);
export { CardVendorShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardVendorShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-card-vendor-ui-showcase',
                    templateUrl: './card-vendor-showcase.component.html',
                }]
        }] });
