// @backbase/ui-ang/esm2015/_example/card-vendor/showcase/card-vendor-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CardVendorShowcaseComponent } from './card-vendor-showcase.component';
import { CardVendorModule } from '@backbase/ui-ang/card-vendor';
import * as i0 from "@angular/core";
export class CardVendorShowcaseModule {
}
CardVendorShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardVendorShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CardVendorShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardVendorShowcaseModule, declarations: [CardVendorShowcaseComponent], imports: [CommonModule, CardVendorModule] });
CardVendorShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardVendorShowcaseModule, imports: [[CommonModule, CardVendorModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardVendorShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, CardVendorModule],
                    declarations: [CardVendorShowcaseComponent],
                    entryComponents: [CardVendorShowcaseComponent],
                }]
        }] });
