// @backbase/cards-common-ang/esm2015/mocks/cards-get/error.js
import { HttpErrorResponse } from '@angular/common/http';
// Throw HttpErrorResponse on wrong parameters
export default (response, params) => {
    const errorCode = parseInt(params.get('error'), 10);
    if (errorCode === 400) {
        const responseBody = {
            message: 'Bad Request',
            errors: [
                {
                    message: 'Mock error message: bad request',
                    key: 'cards.api.test.key',
                    context: {
                        arg0: 0,
                    },
                },
            ],
        };
        throw new HttpErrorResponse({
            error: responseBody,
            status: 400,
        });
    }
    if (errorCode === 403) {
        const responseBody = {
            message: 'Forbidden',
            errors: [
                {
                    message: 'Mock error message: Forbidden',
                    key: 'cards.api.test.key',
                    context: {
                        arg0: 0,
                    },
                },
            ],
        };
        throw new HttpErrorResponse({
            error: responseBody,
            status: 403,
        });
    }
    return response;
};
