// @backbase/ui-ang/esm2015/_example/product-item-credit-card/showcase/product-item-credit-card-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ProductItemCreditCardShowcaseComponent } from './product-item-credit-card-showcase.component';
import { ProductItemCreditCardModule } from '@backbase/ui-ang/product-item-credit-card';
import { ProductItemBasicAccountModule } from '@backbase/ui-ang/product-item-basic-account';
import * as i0 from "@angular/core";
export class ProductItemCreditCardShowcaseModule {
}
ProductItemCreditCardShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCreditCardShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductItemCreditCardShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCreditCardShowcaseModule, declarations: [ProductItemCreditCardShowcaseComponent], imports: [CommonModule, ProductItemCreditCardModule, ProductItemBasicAccountModule] });
ProductItemCreditCardShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCreditCardShowcaseModule, imports: [[CommonModule, ProductItemCreditCardModule, ProductItemBasicAccountModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCreditCardShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ProductItemCreditCardModule, ProductItemBasicAccountModule],
                    declarations: [ProductItemCreditCardShowcaseComponent],
                    entryComponents: [ProductItemCreditCardShowcaseComponent],
                }]
        }] });
