// @backbase/product-summary-common-ang/esm2015/src/errors/accounts-overview-service-error.js
export class AccountsOverviewNotFound {
    constructor(response) {
        this.response = response;
        this.kind = 'AccountsOverviewNotFound';
        this.messages = ((response.error || {}).errors || []).map((e) => e.message);
    }
}
export class AccountsOverviewBadRequest {
    constructor(response) {
        this.response = response;
        this.kind = 'AccountsOverviewBadRequest';
        this.messages = ((response.error || {}).errors || []).map((e) => e.message);
    }
}
export class AccountsOverviewAccessDenied {
    constructor(response) {
        this.response = response;
        this.kind = 'AccountsOverviewAccessDenied';
        this.messages = ((response.error || {}).errors || []).map((e) => e.message);
    }
}
export class AccountsOverviewUnknownError {
    constructor(response) {
        this.response = response;
        this.kind = 'AccountsOverviewUnknownError';
        this.messages = [response.message];
    }
}
export class AccountsOverviewConnectivityError {
    constructor(response) {
        this.response = response;
        this.kind = 'AccountsOverviewConnectivityError';
        this.messages = ((response.error || {}).errors || []).map((e) => e.message);
    }
}
export const parseError = (error) => {
    if (error.error instanceof ErrorEvent) {
        return new AccountsOverviewUnknownError(error);
    }
    else {
        switch (error.status) {
            case 0:
                return new AccountsOverviewConnectivityError(error);
            case 401:
                return new AccountsOverviewAccessDenied(error);
            case 403:
                return new AccountsOverviewAccessDenied(error);
            case 400:
                return new AccountsOverviewBadRequest(error);
            case 404:
                return new AccountsOverviewNotFound(error);
            default:
                return new AccountsOverviewUnknownError(error);
        }
    }
};
