// @backbase/product-summary-common-ang/esm2015/src/errors/balance-history-service-errors.js
export class BalanceHistoryNotFound {
    constructor(response) {
        this.response = response;
        this.kind = 'BalanceHistoryNotFound';
        this.messages = ((response.error || {}).errors || []).map((e) => e.message);
    }
}
export class BalanceHistoryBadRequest {
    constructor(response) {
        this.response = response;
        this.kind = 'BalanceHistoryBadRequest';
        this.messages = ((response.error || {}).errors || []).map((e) => e.message);
    }
}
export class BalanceHistoryAccessDenied {
    constructor(response) {
        this.response = response;
        this.kind = 'BalanceHistoryAccessDenied';
        this.messages = ((response.error || {}).errors || []).map((e) => e.message);
    }
}
export class BalanceHistoryUnknownError {
    constructor(response) {
        this.response = response;
        this.kind = 'BalanceHistoryUnknownError';
        this.messages = [response.message];
    }
}
export class BalanceHistoryConnectivityError {
    constructor(response) {
        this.response = response;
        this.kind = 'BalanceHistoryConnectivityError';
        this.messages = ((response.error || {}).errors || []).map((e) => e.message);
    }
}
export const parseBalanceHistoryError = (error) => {
    if (error.error instanceof ErrorEvent) {
        return new BalanceHistoryUnknownError(error);
    }
    else {
        switch (error.status) {
            case 0:
                return new BalanceHistoryConnectivityError(error);
            case 401:
                return new BalanceHistoryAccessDenied(error);
            case 403:
                return new BalanceHistoryAccessDenied(error);
            case 400:
                return new BalanceHistoryBadRequest(error);
            case 404:
                return new BalanceHistoryNotFound(error);
            default:
                return new BalanceHistoryUnknownError(error);
        }
    }
};
