// @backbase/notifications-common-ang/esm2015/lib/notifications-common.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NotificationLocalizationPipe } from './pipes/notification-localization.pipe';
import { InterpolateLinkPipe } from './pipes/interpolate-link.pipe';
import { NotificationsCapabilityIconComponent, NotificationsCapabilityIconCustomizableDirective, } from './components/notifications-capability-icon/notifications-capability-icon.component';
import { IconModule } from '@backbase/ui-ang/icon';
import { IsElementContentTruncatedDirective } from './directives/is-element-content-truncated.directive';
import * as i0 from "@angular/core";
const pipes = [NotificationLocalizationPipe, InterpolateLinkPipe];
const components = [NotificationsCapabilityIconComponent];
const directives = [NotificationsCapabilityIconCustomizableDirective, IsElementContentTruncatedDirective];
const uiModules = [IconModule];
export class NotificationsCommonModule {
}
/** @nocollapse */ NotificationsCommonModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsCommonModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ NotificationsCommonModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsCommonModule, declarations: [NotificationLocalizationPipe, InterpolateLinkPipe, NotificationsCapabilityIconComponent, NotificationsCapabilityIconCustomizableDirective, IsElementContentTruncatedDirective], imports: [CommonModule, IconModule], exports: [NotificationLocalizationPipe, InterpolateLinkPipe, NotificationsCapabilityIconComponent, NotificationsCapabilityIconCustomizableDirective, IsElementContentTruncatedDirective, IconModule] });
/** @nocollapse */ NotificationsCommonModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsCommonModule, imports: [[CommonModule, ...uiModules], IconModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsCommonModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules],
                    declarations: [...pipes, ...components, ...directives],
                    exports: [...pipes, ...components, ...directives, ...uiModules],
                }]
        }] });
