// @backbase/product-summary-common-ang/esm2015/src/services/balance-history/colors-helper.js
export const getColorTrend = (index, series) => {
    const serie = series[index];
    if (serie.value === 0) {
        if (index > 0) {
            return getColorTrend(index - 1, series);
        }
        if (series.length > 1) {
            return series[1].value < 0 ? -1 : 1;
        }
    }
    if (serie.value < 0) {
        return -1;
    }
    return 1;
};
export const getCurrentSeriesName = (trend) => (trend === -1 ? 'negative' : 'positive');
export const getFirstSeries = (datum, trend) => [
    {
        name: getCurrentSeriesName(trend),
        series: [datum],
    },
];
export const createNewSeries = (result, lastResult, series, trend) => {
    const lastName = lastResult.series[lastResult.series.length - 1].name;
    const lastValue = lastResult.series[lastResult.series.length - 1].value;
    const valuesSum = Math.abs(lastValue) + Math.abs(series.value);
    const step = series.name.getTime() - lastName.getTime();
    const valuesRatio = Math.abs(lastValue) / valuesSum;
    const prevValue = lastResult.series[lastResult.series.length - 1].value;
    const crossingPoint = { name: new Date(lastName.getTime() + step * valuesRatio), value: 0 };
    const crossPointSection = prevValue ? [crossingPoint] : [];
    return [
        ...result.slice(0, -1),
        {
            name: lastResult.name,
            series: [...lastResult.series, ...crossPointSection],
        },
        {
            name: getCurrentSeriesName(trend),
            series: [crossingPoint, series],
        },
    ];
};
export const concatExistingSeries = (result, lastResult, series) => {
    const name = lastResult.name;
    return [
        ...result.slice(0, -1),
        {
            name,
            series: [...lastResult.series, series],
        },
    ];
};
export const splitDataByTrend = (series) => series.reduce((result, seriesData, i) => {
    const currentTrend = getColorTrend(i, series);
    if (i === 0) {
        return getFirstSeries(seriesData, currentTrend);
    }
    const previousTrend = getColorTrend(i - 1, series);
    const lastResult = result[result.length - 1];
    if (previousTrend === currentTrend) {
        return concatExistingSeries(result, lastResult, seriesData);
    }
    else {
        return createNewSeries(result, lastResult, seriesData, currentTrend);
    }
}, []);
