// @backbase/product-summary-common-ang/esm2015/src/pipes/date/date-updated-ago.pipe.js
import { Pipe } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/date-pipe";
/**
 * Returns localised info of date of update
 *
 * @usageNotes
 * ### Example
 * ```html
 * <p>{{ '2020-10-10' | bbDateUpdateAgo }}</p>
 * <p>{{ '2020-10-10' | bbDateUpdateAgo:'mediumDate' }}</p>
 * ```
 */
export class DateUpdatedAgoPipe {
    /**
     * @name DateUpdatedAgoPipe#constructor
     * @param datePipe - BbDatePipe
     *
     * @internal
     */
    constructor(datePipe) {
        this.datePipe = datePipe;
        this.millisecondsInPeriod = {
            day: 86400000,
            hours: 7200000,
            hour: 3600000,
            minutes: 120000,
            minute: 60000,
        };
        this.localisationByPeriod = {
            date: (date) => $localize `:@@product-summary.update-date.date:Updated on ${date}`,
            hours: (hours) => $localize `:@@product-summary.update-date.hours:Updated ${hours} hours ago`,
            hour: () => $localize `:@@product-summary.update-date.hour:Updated 1 hour ago`,
            minutes: (minutes) => $localize `:@@product-summary.update-date.minutes:Updated ${minutes} minutes ago`,
            minute: () => $localize `:@@product-summary.update-date.minute:Updated 1 minute ago`,
        };
    }
    /**
     * @name DateUpdatedAgoPipe#transform
     * @param updateTimestamp - update date
     * @param dateFormat - the format type for the date to be formatted (for update time >= 1 day ago)
     *
     * @description
     * Returns localised info of date of update
     *
     * Example:
     *  transform(hourAgo) - 'Updated 1 hour ago'
     *  transform(minuteAgo) - 'Updated 1 minute ago'
     */
    transform(updateTimestamp, dateFormat = 'longDate') {
        const millisecondsAgo = Math.max(new Date().getTime() - new Date(updateTimestamp).getTime(), 0);
        if (millisecondsAgo < this.millisecondsInPeriod.minutes) {
            return this.localisationByPeriod.minute();
        }
        else if (millisecondsAgo < this.millisecondsInPeriod.hour) {
            const minutesAgo = Math.floor(millisecondsAgo / this.millisecondsInPeriod.minute);
            return this.localisationByPeriod.minutes(minutesAgo);
        }
        else if (millisecondsAgo < this.millisecondsInPeriod.hours) {
            return this.localisationByPeriod.hour();
        }
        else if (millisecondsAgo < this.millisecondsInPeriod.day) {
            const hoursAgo = Math.floor(millisecondsAgo / this.millisecondsInPeriod.hour);
            return this.localisationByPeriod.hours(hoursAgo);
        }
        else {
            const updateDate = this.datePipe.transform(updateTimestamp, dateFormat);
            return this.localisationByPeriod.date(updateDate);
        }
    }
}
DateUpdatedAgoPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DateUpdatedAgoPipe, deps: [{ token: i1.BbDatePipe }], target: i0.ɵɵFactoryTarget.Pipe });
DateUpdatedAgoPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DateUpdatedAgoPipe, name: "bbDateUpdatedAgo" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DateUpdatedAgoPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'bbDateUpdatedAgo',
                }]
        }], ctorParameters: function () { return [{ type: i1.BbDatePipe }]; } });
