// @backbase/card-details-widget-ang/esm2015/src/components/card-details-widget/card-details-widget.component.js
import { Component, Input } from '@angular/core';
import { combineLatest } from 'rxjs';
import { map, pluck } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "../card-details-container/card-details-container.component";
export class CardDetailsWidgetComponent {
    constructor(route) {
        this.route = route;
        this.paymentCardId = this.getRouteParam(this.route, 'selectedCard').pipe(map((selectedCard) => selectedCard));
    }
    getRouteParam(route, param) {
        const paramValue = route.paramMap.pipe(pluck('params', param));
        if (!route.parent) {
            return paramValue;
        }
        // Ensure that level that has actual value keeps it from being cleared.
        return combineLatest(paramValue, this.getRouteParam(route.parent, param)).pipe(map((params) => params[0] || params[1]));
    }
}
CardDetailsWidgetComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardDetailsWidgetComponent, deps: [{ token: i1.ActivatedRoute }], target: i0.ɵɵFactoryTarget.Component });
CardDetailsWidgetComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: CardDetailsWidgetComponent, selector: "bb-card-details-widget", inputs: { notificationTtl: "notificationTtl", updateAllLimits: "updateAllLimits" }, ngImport: i0, template: "<bb-card-details-container \n  [paymentCardId]=\"paymentCardId\"\n  [notificationTtl]=\"notificationTtl\"\n  [updateAllLimits]=\"updateAllLimits\"\n></bb-card-details-container>", components: [{ type: i2.CardDetailsContainerComponent, selector: "bb-card-details-container", inputs: ["paymentCardId", "notificationTtl", "updateAllLimits"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardDetailsWidgetComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-card-details-widget',
                    templateUrl: './card-details-widget.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.ActivatedRoute }]; }, propDecorators: { notificationTtl: [{
                type: Input
            }], updateAllLimits: [{
                type: Input
            }] } });
