// @backbase/budget-journey-ang/fesm2015/backbase-budget-journey-ang.js
import { CommonModule } from '@angular/common';
import * as i0 from '@angular/core';
import { InjectionToken, Injectable, Optional, Inject, Component, NgModule } from '@angular/core';
import * as i1 from '@angular/router';
import { provideRoutes, RouterModule } from '@angular/router';
import * as i3 from '@backbase/budget-widget-ang';
import { BudgetWidgetModule } from '@backbase/budget-widget-ang';
import * as i1$1 from '@backbase/foundation-ang/core';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import * as i2 from '@backbase/ui-ang/header';
import { HeaderModule } from '@backbase/ui-ang/header';

// eslint-disable-next-line
const BudgetJourneyConfigurationToken = new InjectionToken('BudgetJourneyConfiguration injection token');
const configDefaults = {
    showPercentage: false,
    budgetSafeZoneLimit: 80,
    notificationDismissTime: 5,
    maxBudgets: null,
};
class BudgetJourneyService {
    constructor(config) {
        config = config || {};
        this._config = Object.assign(Object.assign({}, configDefaults), config);
    }
    get defaults() {
        return configDefaults;
    }
    get showPercentage() {
        return this._config.showPercentage;
    }
    get budgetSafeZoneLimit() {
        return this._config.budgetSafeZoneLimit;
    }
    get notificationDismissTime() {
        return this._config.notificationDismissTime;
    }
    get maxBudgets() {
        return this._config.maxBudgets;
    }
}
BudgetJourneyService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetJourneyService, deps: [{ token: BudgetJourneyConfigurationToken, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
BudgetJourneyService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetJourneyService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetJourneyService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [BudgetJourneyConfigurationToken]
                }] }]; } });

class BudgetJourneyComponent {
}
BudgetJourneyComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetJourneyComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
BudgetJourneyComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: BudgetJourneyComponent, selector: "bb-budget-journey", providers: [BudgetJourneyService], ngImport: i0, template: '<router-outlet></router-outlet>', isInline: true, directives: [{ type: i1.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetJourneyComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-budget-journey',
                    template: '<router-outlet></router-outlet>',
                    providers: [BudgetJourneyService],
                }]
        }] });

// TODO: remove utility classes
class BudgetViewComponent {
    constructor(config) {
        this.config = config;
    }
}
BudgetViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetViewComponent, deps: [{ token: BudgetJourneyService }], target: i0.ɵɵFactoryTarget.Component });
BudgetViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: BudgetViewComponent, selector: "bb-budgets-view", ngImport: i0, template: `
    <bb-header-ui
      headingType="h1"
      headingClasses="bb-heading-widget__heading px-0 pb-5"
      heading="Budgets"
      i18n-heading="Budgets|Heading for budget journey@@budget.journey.heading.text"
    ></bb-header-ui>
    <bb-budget-widget
      [showPercentage]="config.showPercentage"
      [budgetSafeZoneLimit]="config.budgetSafeZoneLimit"
      [notificationDismissTime]="config.notificationDismissTime"
      [maxBudgets]="config.maxBudgets"
    ></bb-budget-widget>
  `, isInline: true, components: [{ type: i2.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i3.BudgetWidgetComponent, selector: "bb-budget-widget", inputs: ["showPercentage", "budgetSafeZoneLimit", "notificationDismissTime", "maxBudgets"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetViewComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-budgets-view',
                    template: `
    <bb-header-ui
      headingType="h1"
      headingClasses="bb-heading-widget__heading px-0 pb-5"
      heading="Budgets"
      i18n-heading="Budgets|Heading for budget journey@@budget.journey.heading.text"
    ></bb-header-ui>
    <bb-budget-widget
      [showPercentage]="config.showPercentage"
      [budgetSafeZoneLimit]="config.budgetSafeZoneLimit"
      [notificationDismissTime]="config.notificationDismissTime"
      [maxBudgets]="config.maxBudgets"
    ></bb-budget-widget>
  `,
                }]
        }], ctorParameters: function () { return [{ type: BudgetJourneyService }]; } });

const defaultRoute = {
    path: '',
    component: BudgetJourneyComponent,
    children: [
        {
            path: '',
            component: BudgetViewComponent,
        },
    ],
};
class BudgetJourneyModule {
    static forRoot(data = { route: defaultRoute }) {
        return {
            ngModule: BudgetJourneyModule,
            providers: [provideRoutes([data.route])],
        };
    }
}
BudgetJourneyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetJourneyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
BudgetJourneyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetJourneyModule, declarations: [BudgetJourneyComponent, BudgetViewComponent], imports: [CommonModule, i1$1.BackbaseCoreModule, RouterModule,
        HeaderModule,
        BudgetWidgetModule] });
BudgetJourneyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetJourneyModule, imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { BudgetJourneyComponent },
            }),
            RouterModule,
            HeaderModule,
            BudgetWidgetModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetJourneyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [BudgetJourneyComponent, BudgetViewComponent],
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { BudgetJourneyComponent },
                        }),
                        RouterModule,
                        HeaderModule,
                        BudgetWidgetModule,
                    ],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { BudgetJourneyComponent, BudgetJourneyConfigurationToken, BudgetJourneyModule, BudgetJourneyService, BudgetViewComponent };
