// @backbase/quick-transfer-journey-ang/esm2015/lib/components/account-selector-base/account-selector-base.component.js
import { Component, Input } from '@angular/core';
import { LoadingState } from '../../models/state.model';
import { ViewState } from '../../models/view.model';
import { QuickTransferStoreService } from '../../store/store.service';
import * as i0 from "@angular/core";
import * as i1 from "../../store/store.service";
export class BaseAccountSelectorComponent {
    constructor(store) {
        this.store = store;
        /**
         * Stream of accountsCallState
         */
        this.accountsCallState$ = this.store.accountsCallState$;
        /**
         * Enum for loadingstate
         */
        this.loadingstate = LoadingState;
        /**
         * Stream of extAccountsCallState
         */
        this.extAccountsCallState$ = this.store.extAccountsCallState$;
    }
    /**
     * Method to navigate back to the main Form
     */
    backToForm(fromSelector) {
        this.store.setView(ViewState.FORM, fromSelector);
    }
    /**
     * Method to dispatch store action for fetching the accounts
     */
    getAccounts() {
        this.store.fetchAccounts();
    }
    /**
     * Method to dispatch store action for fetching external accounts
     */
    getExtAccounts() {
        this.store.fetchExtAccounts();
    }
}
/** @nocollapse */ BaseAccountSelectorComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BaseAccountSelectorComponent, deps: [{ token: i1.QuickTransferStoreService }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ BaseAccountSelectorComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: BaseAccountSelectorComponent, selector: "ng-component", inputs: { config: "config" }, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BaseAccountSelectorComponent, decorators: [{
            type: Component,
            args: [{
                    template: '',
                }]
        }], ctorParameters: function () { return [{ type: i1.QuickTransferStoreService }]; }, propDecorators: { config: [{
                type: Input
            }] } });
