// @backbase/cards-common-ang/esm2015/src/model/payment-card.model.js
/**
 * Possible payment card statuses.
 */
export var PaymentCardStatus;
(function (PaymentCardStatus) {
    PaymentCardStatus["Active"] = "ACTIVE";
    PaymentCardStatus["Inactive"] = "INACTIVE";
    PaymentCardStatus["Canceled"] = "CANCELED";
    PaymentCardStatus["Expired"] = "EXPIRED";
})(PaymentCardStatus || (PaymentCardStatus = {}));
/**
 * Possible payment card replacement statuses.
 */
export var PaymentCardReplacementStatus;
(function (PaymentCardReplacementStatus) {
    PaymentCardReplacementStatus["NotUnderReplacement"] = "NOT_UNDER_REPLACEMENT";
    PaymentCardReplacementStatus["UnderReplacement"] = "UNDER_REPLACEMENT";
    PaymentCardReplacementStatus["Replaced"] = "REPLACED";
})(PaymentCardReplacementStatus || (PaymentCardReplacementStatus = {}));
/**
 * Possible payment card lock statuses.
 */
export var PaymentCardLockStatus;
(function (PaymentCardLockStatus) {
    PaymentCardLockStatus["Locked"] = "LOCKED";
    PaymentCardLockStatus["Unlocked"] = "UNLOCKED";
})(PaymentCardLockStatus || (PaymentCardLockStatus = {}));
