// @backbase/ui-ang/esm2015/collapsible/collapsible.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NgbCollapseModule } from '@ng-bootstrap/ng-bootstrap';
import { CollapsibleBodyDirective, CollapsibleComponent, CollapsibleHeaderDirective } from './collapsible.component';
import * as i0 from "@angular/core";
export class CollapsibleModule {
}
CollapsibleModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CollapsibleModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CollapsibleModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CollapsibleModule, declarations: [CollapsibleComponent, CollapsibleHeaderDirective, CollapsibleBodyDirective], imports: [CommonModule, NgbCollapseModule], exports: [CollapsibleComponent, CollapsibleHeaderDirective, CollapsibleBodyDirective] });
CollapsibleModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CollapsibleModule, imports: [[CommonModule, NgbCollapseModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CollapsibleModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, NgbCollapseModule],
                    declarations: [CollapsibleComponent, CollapsibleHeaderDirective, CollapsibleBodyDirective],
                    exports: [CollapsibleComponent, CollapsibleHeaderDirective, CollapsibleBodyDirective],
                }]
        }] });
