// @backbase/ui-ang/esm2015/_example/header/showcase/header-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HeaderShowcaseComponent } from './header-showcase.component';
import { HeaderModule } from '@backbase/ui-ang/header';
import * as i0 from "@angular/core";
export class HeaderShowcaseModule {
}
HeaderShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HeaderShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
HeaderShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HeaderShowcaseModule, declarations: [HeaderShowcaseComponent], imports: [CommonModule, HeaderModule] });
HeaderShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HeaderShowcaseModule, imports: [[CommonModule, HeaderModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HeaderShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, HeaderModule],
                    declarations: [HeaderShowcaseComponent],
                    entryComponents: [HeaderShowcaseComponent],
                }]
        }] });
