// @backbase/ui-ang/esm2015/_example/header/showcase/header-showcase.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/header";
let HeaderShowcaseComponent = class HeaderShowcaseComponent {
};
HeaderShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HeaderShowcaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
HeaderShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: HeaderShowcaseComponent, selector: "bb-header-ui-showcase", ngImport: i0, template: "<bb-header-ui headingType=\"h1\" heading=\"This is H1\"></bb-header-ui>\n<bb-header-ui headingType=\"h2\" heading=\"This is H2\"></bb-header-ui>\n<bb-header-ui headingType=\"h3\" heading=\"This is H3\"></bb-header-ui>\n", components: [{ type: i1.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }] });
HeaderShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
], HeaderShowcaseComponent);
export { HeaderShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HeaderShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-header-ui-showcase',
                    templateUrl: './header-showcase.component.html',
                }]
        }] });
