// @backbase/contact-common-ang/esm2015/lib/components/confirm-action-dialog/confirm-action-dialog.module.js
import { NgModule } from '@angular/core';
import { ButtonModule } from '@backbase/ui-ang/button';
import { ModalModule } from '@backbase/ui-ang/modal';
import { ConfirmActionDialogComponent } from './confirm-action-dialog.component';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { TextareaModule } from '@backbase/ui-ang/textarea';
import * as i0 from "@angular/core";
/**
 * Exports ConfirmActionDialogComponent
 */
export class ConfirmActionDialogModule {
}
/** @nocollapse */ ConfirmActionDialogModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ConfirmActionDialogModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ ConfirmActionDialogModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ConfirmActionDialogModule, declarations: [ConfirmActionDialogComponent], imports: [ModalModule, ButtonModule, CommonModule, TextareaModule, FormsModule], exports: [ConfirmActionDialogComponent] });
/** @nocollapse */ ConfirmActionDialogModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ConfirmActionDialogModule, imports: [[ModalModule, ButtonModule, CommonModule, TextareaModule, FormsModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ConfirmActionDialogModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ConfirmActionDialogComponent],
                    exports: [ConfirmActionDialogComponent],
                    imports: [ModalModule, ButtonModule, CommonModule, TextareaModule, FormsModule],
                }]
        }] });
