// @backbase/ui-ang/esm2015/_example/select-list/showcase/select-list-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { SelectListShowcaseComponent } from './select-list-showcase.component';
import { SelectListModule } from '@backbase/ui-ang/select-list';
import * as i0 from "@angular/core";
export class SelectListShowcaseModule {
}
SelectListShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SelectListShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
SelectListShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SelectListShowcaseModule, declarations: [SelectListShowcaseComponent], imports: [CommonModule, SelectListModule, FormsModule, ReactiveFormsModule] });
SelectListShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SelectListShowcaseModule, imports: [[CommonModule, SelectListModule, FormsModule, ReactiveFormsModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SelectListShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, SelectListModule, FormsModule, ReactiveFormsModule],
                    declarations: [SelectListShowcaseComponent],
                    entryComponents: [SelectListShowcaseComponent],
                }]
        }] });
