// @backbase/manage-payees-journey-ang/esm2015/src/components/payveris-ebills-enrollment/billpay-payveris-ebills-enrolment-notification-success/billpay-payveris-ebills-enrolment-notification-success.component.js
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/notification";
export class BillpayPayverisEbillsEnrolmentNotificationSuccessComponent {
    /** @internal */
    constructor(notificationService) {
        this.notificationService = notificationService;
        /**
         * Time until the notification disappears in milliseconds.
         */
        this.ttl = 5000;
        /**
         * The modal message.
         */
        this.message = '';
    }
    ngOnInit() {
        this.notificationService.showNotification({
            header: this.header,
            message: this.message,
            dismissible: true,
            modifier: 'success',
            ttl: this.ttl,
        });
    }
}
BillpayPayverisEbillsEnrolmentNotificationSuccessComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayPayverisEbillsEnrolmentNotificationSuccessComponent, deps: [{ token: i1.NotificationService }], target: i0.ɵɵFactoryTarget.Component });
BillpayPayverisEbillsEnrolmentNotificationSuccessComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: BillpayPayverisEbillsEnrolmentNotificationSuccessComponent, selector: "bb-billpay-payveris-ebills-enrolment-notification-success", inputs: { ttl: "ttl", header: "header", message: "message" }, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayPayverisEbillsEnrolmentNotificationSuccessComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-billpay-payveris-ebills-enrolment-notification-success',
                    template: '',
                }]
        }], ctorParameters: function () { return [{ type: i1.NotificationService }]; }, propDecorators: { ttl: [{
                type: Input
            }], header: [{
                type: Input
            }], message: [{
                type: Input
            }] } });
