// @backbase/ui-ang/esm2015/dropdown-single-select/dropdown-single-select.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { IconModule } from '@backbase/ui-ang/icon';
import { FormsModule } from '@angular/forms';
import { BaseClassesModule } from '@backbase/ui-ang/base-classes';
import { FocusModule } from '@backbase/ui-ang/focus';
import { DropdownSingleSelectComponent } from './dropdown-single-select.component';
import { DropdownSingleSelectOptionComponent } from './dropdown-single-select-option.component';
import * as i0 from "@angular/core";
export class DropdownSingleSelectModule {
}
DropdownSingleSelectModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownSingleSelectModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
DropdownSingleSelectModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownSingleSelectModule, declarations: [DropdownSingleSelectComponent, DropdownSingleSelectOptionComponent], imports: [CommonModule, IconModule, FormsModule, BaseClassesModule, FocusModule], exports: [DropdownSingleSelectComponent, DropdownSingleSelectOptionComponent] });
DropdownSingleSelectModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownSingleSelectModule, imports: [[CommonModule, IconModule, FormsModule, BaseClassesModule, FocusModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownSingleSelectModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, IconModule, FormsModule, BaseClassesModule, FocusModule],
                    declarations: [DropdownSingleSelectComponent, DropdownSingleSelectOptionComponent],
                    exports: [DropdownSingleSelectComponent, DropdownSingleSelectOptionComponent],
                }]
        }] });
