// @backbase/ui-ang/esm2015/dropdown-single-select/dropdown-single-select-option.component.js
import { ChangeDetectionStrategy, Component, Inject, InjectionToken, Input, } from '@angular/core';
import * as i0 from "@angular/core";
export const BB_DROPDOWN_CHANGE_DETECTION_REF_TOKEN = new InjectionToken('BB_DROPDOWN_SINGLE_SELECT');
/**
 * @name DropdownSingleSelectOption
 *
 * @description
 * Component that represents a single option that is part of the dropdown single select component.
 */
export class DropdownSingleSelectOptionComponent {
    constructor(dropdownCDRef) {
        this.dropdownCDRef = dropdownCDRef;
    }
    // NOTES: trigger of detectChanges is needed
    // because DropdownSingleSelectComponent has changeDetection
    // strategy onPush, hence it reacts only to changes
    // of its own @Inputs, but not @Inputs() of options component
    ngOnChanges({ value = {}, label = {} }) {
        const { firstChange: firstChangeValue } = value;
        const { firstChange: firstChangeLabel } = label;
        if (firstChangeLabel && firstChangeValue) {
            return;
        }
        this.dropdownCDRef.detectChanges();
    }
}
DropdownSingleSelectOptionComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownSingleSelectOptionComponent, deps: [{ token: BB_DROPDOWN_CHANGE_DETECTION_REF_TOKEN }], target: i0.ɵɵFactoryTarget.Component });
DropdownSingleSelectOptionComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: DropdownSingleSelectOptionComponent, selector: "bb-dropdown-single-select-option-ui", inputs: { value: "value", label: "label" }, usesOnChanges: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownSingleSelectOptionComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-dropdown-single-select-option-ui',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    template: '',
                }]
        }], ctorParameters: function () { return [{ type: i0.ChangeDetectorRef, decorators: [{
                    type: Inject,
                    args: [BB_DROPDOWN_CHANGE_DETECTION_REF_TOKEN]
                }] }]; }, propDecorators: { value: [{
                type: Input
            }], label: [{
                type: Input
            }] } });
