// @backbase/pocket-tailor-http-ang/bundles/backbase-pocket-tailor-http-ang.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common/http'), require('@backbase/foundation-ang/data-http')) :
    typeof define === 'function' && define.amd ? define('@backbase/pocket-tailor-http-ang', ['exports', '@angular/core', '@angular/common/http', '@backbase/foundation-ang/data-http'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["pocket-tailor-http-ang"] = {}), global.ng.core, global.ng.common.http, global.i2));
})(this, (function (exports, i0, i1, i2) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);

    var POCKET_TAILOR_BASE_PATH = new i0.InjectionToken('POCKET_TAILOR_BASE_PATH');
    var COLLECTION_FORMATS = {
        'csv': ',',
        'tsv': '   ',
        'ssv': ' ',
        'pipes': '|'
    };

    var PocketTailorConfiguration = /** @class */ (function () {
        function PocketTailorConfiguration(configurationParameters) {
            if (configurationParameters === void 0) { configurationParameters = {}; }
            this.apiKeys = configurationParameters.apiKeys;
            this.username = configurationParameters.username;
            this.password = configurationParameters.password;
            this.accessToken = configurationParameters.accessToken;
            this.basePath = configurationParameters.basePath;
            this.withCredentials = configurationParameters.withCredentials;
            this.encoder = configurationParameters.encoder;
            if (configurationParameters.credentials) {
                this.credentials = configurationParameters.credentials;
            }
            else {
                this.credentials = {};
            }
        }
        PocketTailorConfiguration.prototype.selectHeaderContentType = function (contentTypes) {
            var _this = this;
            if (contentTypes.length === 0) {
                return undefined;
            }
            var type = contentTypes.find(function (x) { return _this.isJsonMime(x); });
            if (type === undefined) {
                return contentTypes[0];
            }
            return type;
        };
        PocketTailorConfiguration.prototype.selectHeaderAccept = function (accepts) {
            var _this = this;
            if (accepts.length === 0) {
                return undefined;
            }
            var type = accepts.find(function (x) { return _this.isJsonMime(x); });
            if (type === undefined) {
                return accepts[0];
            }
            return type;
        };
        PocketTailorConfiguration.prototype.isJsonMime = function (mime) {
            var jsonMime = new RegExp('^(application\/json|[^;/ \t]+\/[^;/ \t]+[+]json)[ \t]*(;.*)?$', 'i');
            return mime !== null && (jsonMime.test(mime) || mime.toLowerCase() === 'application/json-patch+json');
        };
        PocketTailorConfiguration.prototype.lookupCredential = function (key) {
            var value = this.credentials[key];
            return typeof value === 'function'
                ? value()
                : value;
        };
        return PocketTailorConfiguration;
    }());
    var CONFIG_TOKEN = new i0.InjectionToken('api-data-module PocketTailorConfiguration');

    var PocketTailorApiModule = /** @class */ (function () {
        function PocketTailorApiModule(parentModule, http, dataModulesManager, config) {
            if (parentModule) {
                throw new Error('PocketTailorApiModule is already loaded. Import in your base AppModule only.');
            }
            if (!http) {
                throw new Error('You need to import the HttpClientModule in your AppModule! \n' +
                    'See also https://github.com/angular/angular/issues/20575');
            }
            if (dataModulesManager) {
                dataModulesManager.setModuleConfig(CONFIG_TOKEN, {
                    apiRoot: '',
                    servicePath: config.basePath || '',
                    headers: {},
                });
            }
        }
        PocketTailorApiModule.forRoot = function (configurationFactory) {
            return {
                ngModule: PocketTailorApiModule,
                providers: [{ provide: PocketTailorConfiguration, useFactory: configurationFactory }]
            };
        };
        return PocketTailorApiModule;
    }());
    PocketTailorApiModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: PocketTailorApiModule, deps: [{ token: PocketTailorApiModule, optional: true, skipSelf: true }, { token: i1__namespace.HttpClient, optional: true }, { token: i2__namespace.DataModulesManager, optional: true }, { token: PocketTailorConfiguration }], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    PocketTailorApiModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: PocketTailorApiModule });
    PocketTailorApiModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: PocketTailorApiModule, providers: [], imports: [[]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: PocketTailorApiModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [],
                        declarations: [],
                        exports: [],
                        providers: []
                    }]
            }], ctorParameters: function () {
            return [{ type: PocketTailorApiModule, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.SkipSelf
                        }] }, { type: i1__namespace.HttpClient, decorators: [{
                            type: i0.Optional
                        }] }, { type: i2__namespace.DataModulesManager, decorators: [{
                            type: i0.Optional
                        }] }, { type: PocketTailorConfiguration }];
        } });

    var CustomHttpParameterCodec = /** @class */ (function () {
        function CustomHttpParameterCodec() {
        }
        CustomHttpParameterCodec.prototype.encodeKey = function (k) {
            return encodeURIComponent(k);
        };
        CustomHttpParameterCodec.prototype.encodeValue = function (v) {
            return encodeURIComponent(v);
        };
        CustomHttpParameterCodec.prototype.decodeKey = function (k) {
            return decodeURIComponent(k);
        };
        CustomHttpParameterCodec.prototype.decodeValue = function (v) {
            return decodeURIComponent(v);
        };
        return CustomHttpParameterCodec;
    }());

    var PocketTailorClientHttpService = /** @class */ (function () {
        function PocketTailorClientHttpService(httpClient, basePath, configuration) {
            this.httpClient = httpClient;
            this.basePath = '';
            this.defaultHeaders = new i1.HttpHeaders();
            this.configuration = new PocketTailorConfiguration();
            if (configuration) {
                this.configuration = configuration;
            }
            if (typeof this.configuration.basePath !== 'string') {
                if (typeof basePath !== 'string') {
                    basePath = this.basePath;
                }
                this.configuration.basePath = basePath;
            }
            this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
        }
        PocketTailorClientHttpService.prototype.addToHttpParams = function (httpParams, value, key) {
            if (typeof value === "object" && value instanceof Date === false) {
                httpParams = this.addToHttpParamsRecursive(httpParams, value);
            }
            else {
                httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
            }
            return httpParams;
        };
        PocketTailorClientHttpService.prototype.addToHttpParamsRecursive = function (httpParams, value, key) {
            var _this = this;
            if (value == null) {
                return httpParams;
            }
            if (typeof value === "object") {
                if (Array.isArray(value)) {
                    value.forEach(function (elem) { return httpParams = _this.addToHttpParamsRecursive(httpParams, elem, key); });
                }
                else if (value instanceof Date) {
                    if (key != null) {
                        httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                    }
                    else {
                        throw Error("key may not be null if value is Date");
                    }
                }
                else {
                    Object.keys(value).forEach(function (k) { return httpParams = _this.addToHttpParamsRecursive(httpParams, value[k], key != null ? key + "." + k : k); });
                }
            }
            else if (key != null) {
                httpParams = httpParams.append(key, value);
            }
            else {
                throw Error("key may not be null if value is not object or array");
            }
            return httpParams;
        };
        PocketTailorClientHttpService.prototype.closePocket = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _closePocketRequest = requestParameters["closePocketRequest"];
            if (_closePocketRequest === null || _closePocketRequest === undefined) {
                throw new Error('Required parameter closePocketRequest was null or undefined when calling closePocket.');
            }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var consumes = [
                'application/json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected !== undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.post(this.configuration.basePath + "/client-api/v2/pockets/closed", _closePocketRequest, {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        PocketTailorClientHttpService.prototype.closePocketUrl = function (requestParameters) {
            var _closePocketRequest = requestParameters["closePocketRequest"];
            if (_closePocketRequest === null || _closePocketRequest === undefined) {
                throw new Error('Required parameter closePocketRequest was null or undefined when calling closePocket.');
            }
            return this.configuration.basePath + "/client-api/v2/pockets/closed";
        };
        PocketTailorClientHttpService.prototype.listPockets = function (observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.get(this.configuration.basePath + "/client-api/v2/pockets", {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        PocketTailorClientHttpService.prototype.listPocketsUrl = function () {
            return this.configuration.basePath + "/client-api/v2/pockets";
        };
        PocketTailorClientHttpService.prototype.postPocket = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _pocketPostRequest = requestParameters["pocketPostRequest"];
            if (_pocketPostRequest === null || _pocketPostRequest === undefined) {
                throw new Error('Required parameter pocketPostRequest was null or undefined when calling postPocket.');
            }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var consumes = [
                'application/json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected !== undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.post(this.configuration.basePath + "/client-api/v2/pockets", _pocketPostRequest, {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        PocketTailorClientHttpService.prototype.postPocketUrl = function (requestParameters) {
            var _pocketPostRequest = requestParameters["pocketPostRequest"];
            if (_pocketPostRequest === null || _pocketPostRequest === undefined) {
                throw new Error('Required parameter pocketPostRequest was null or undefined when calling postPocket.');
            }
            return this.configuration.basePath + "/client-api/v2/pockets";
        };
        PocketTailorClientHttpService.prototype.viewPocket = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _pocketId = requestParameters["pocketId"];
            if (_pocketId === null || _pocketId === undefined) {
                throw new Error('Required parameter pocketId was null or undefined when calling viewPocket.');
            }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.get(this.configuration.basePath + "/client-api/v2/pockets/" + encodeURIComponent(String(_pocketId)), {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        PocketTailorClientHttpService.prototype.viewPocketUrl = function (requestParameters) {
            var _pocketId = requestParameters["pocketId"];
            if (_pocketId === null || _pocketId === undefined) {
                throw new Error('Required parameter pocketId was null or undefined when calling viewPocket.');
            }
            return this.configuration.basePath + "/client-api/v2/pockets/" + encodeURIComponent(String(_pocketId));
        };
        return PocketTailorClientHttpService;
    }());
    PocketTailorClientHttpService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: PocketTailorClientHttpService, deps: [{ token: i1__namespace.HttpClient }, { token: POCKET_TAILOR_BASE_PATH, optional: true }, { token: PocketTailorConfiguration, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    PocketTailorClientHttpService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: PocketTailorClientHttpService, providedIn: 'root' });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: PocketTailorClientHttpService, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root'
                    }]
            }], ctorParameters: function () {
            return [{ type: i1__namespace.HttpClient }, { type: undefined, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.Inject,
                            args: [POCKET_TAILOR_BASE_PATH]
                        }] }, { type: PocketTailorConfiguration, decorators: [{
                            type: i0.Optional
                        }] }];
        } });

    var PocketTailorClientHttpServiceClosePocketMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v2/pockets/closed",
            method: "POST",
            responses: [
                {
                    status: 204,
                    body: ""
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
                {
                    status: 500,
                    body: {
                        "message": "Description of error"
                    }
                },
            ]
        }]);
    var PocketTailorClientHttpServiceListPocketsMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v2/pockets",
            method: "GET",
            responses: [
                {
                    status: 200,
                    body: {
                        "pockets": [{
                                "id": "546abc12",
                                "arrangementId": "11da7d70-ed15-11ea-adc1-0242ac120002",
                                "name": "Travel Europe",
                                "icon": "travel",
                                "goal": {
                                    "amountCurrency": {
                                        "amount": "2000.00",
                                        "currencyCode": "EUR"
                                    },
                                    "deadline": "2021-02-28",
                                    "progress": 35.5
                                },
                                "balance": {
                                    "amount": "-20.00",
                                    "currencyCode": "EUR"
                                }
                            }, {
                                "id": "gy2345dff",
                                "arrangementId": "3cdb2224-8926-4b4d-a99f-012331c94699",
                                "name": "Christmas gifts",
                                "icon": "gift",
                                "balance": {
                                    "amount": "789.13",
                                    "currencyCode": "EUR"
                                }
                            }, {
                                "id": "cd1235dff",
                                "arrangementId": "2331c9469-09a8-4b4d-a91c946999f-01233",
                                "name": "Adventure holidays",
                                "icon": "holidays",
                                "goal": {
                                    "deadline": "2021-09-18"
                                },
                                "balance": {
                                    "amount": "89.13",
                                    "currencyCode": "EUR"
                                }
                            }, {
                                "id": "987abc12",
                                "arrangementId": "c7dgh-a7d70-ed15-11ea-fec1-0242120002",
                                "name": "Home Renovation",
                                "icon": "home",
                                "goal": {
                                    "amountCurrency": {
                                        "amount": "50000.00",
                                        "currencyCode": "EUR"
                                    },
                                    "progress": 0.5
                                },
                                "balance": {
                                    "amount": "250.00",
                                    "currencyCode": "EUR"
                                }
                            }]
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 500,
                    body: {
                        "message": "Description of error"
                    }
                },
            ]
        }]);
    var PocketTailorClientHttpServicePostPocketMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v2/pockets",
            method: "POST",
            responses: [
                {
                    status: 201,
                    body: {
                        "id": "546abc12",
                        "arrangementId": "11da7d70-ed15-11ea-adc1-0242ac120002",
                        "name": "Travel Europe",
                        "icon": "travel",
                        "goal": {
                            "amountCurrency": {
                                "amount": "2000.00",
                                "currencyCode": "EUR"
                            },
                            "deadline": "2021-02-28",
                            "progress": 35.5
                        },
                        "balance": {
                            "amount": "-20.00",
                            "currencyCode": "EUR"
                        }
                    }
                },
                {
                    status: 201,
                    body: {
                        "arrangementId": "arrangementId",
                        "goal": {
                            "progress": 0.8008281904610115,
                            "deadline": "2000-01-23",
                            "amountCurrency": {
                                "amount": "amount",
                                "additions": {
                                    "key": "additions"
                                },
                                "currencyCode": "currencyCode"
                            }
                        },
                        "balance": {
                            "amount": "amount",
                            "additions": {
                                "key": "additions"
                            },
                            "currencyCode": "currencyCode"
                        },
                        "name": "name",
                        "icon": "icon",
                        "id": "id"
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 500,
                    body: {
                        "message": "Description of error"
                    }
                },
            ]
        }]);
    var PocketTailorClientHttpServiceViewPocketMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v2/pockets/{pocketId}",
            method: "GET",
            responses: [
                {
                    status: 200,
                    body: {
                        "id": "546abc12",
                        "arrangementId": "11da7d70-ed15-11ea-adc1-0242ac120002",
                        "name": "Travel Europe",
                        "icon": "travel",
                        "goal": {
                            "amountCurrency": {
                                "amount": "2000.00",
                                "currencyCode": "EUR"
                            },
                            "deadline": "2021-02-28",
                            "progress": 35.5
                        },
                        "balance": {
                            "amount": "-20.00",
                            "currencyCode": "EUR"
                        }
                    }
                },
                {
                    status: 200,
                    body: {
                        "arrangementId": "arrangementId",
                        "goal": {
                            "progress": 0.8008281904610115,
                            "deadline": "2000-01-23",
                            "amountCurrency": {
                                "amount": "amount",
                                "additions": {
                                    "key": "additions"
                                },
                                "currencyCode": "currencyCode"
                            }
                        },
                        "balance": {
                            "amount": "amount",
                            "additions": {
                                "key": "additions"
                            },
                            "currencyCode": "currencyCode"
                        },
                        "name": "name",
                        "icon": "icon",
                        "id": "id"
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
                {
                    status: 500,
                    body: {
                        "message": "Description of error"
                    }
                },
            ]
        }]);
    var PocketTailorClientHttpServiceMocksProvider = i2.createMocks([
        {
            urlPattern: "/client-api/v2/pockets/closed",
            method: "POST",
            responses: [
                {
                    status: 204,
                    body: ""
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
                {
                    status: 500,
                    body: {
                        "message": "Description of error"
                    }
                },
            ]
        },
        {
            urlPattern: "/client-api/v2/pockets",
            method: "GET",
            responses: [
                {
                    status: 200,
                    body: {
                        "pockets": [{
                                "id": "546abc12",
                                "arrangementId": "11da7d70-ed15-11ea-adc1-0242ac120002",
                                "name": "Travel Europe",
                                "icon": "travel",
                                "goal": {
                                    "amountCurrency": {
                                        "amount": "2000.00",
                                        "currencyCode": "EUR"
                                    },
                                    "deadline": "2021-02-28",
                                    "progress": 35.5
                                },
                                "balance": {
                                    "amount": "-20.00",
                                    "currencyCode": "EUR"
                                }
                            }, {
                                "id": "gy2345dff",
                                "arrangementId": "3cdb2224-8926-4b4d-a99f-012331c94699",
                                "name": "Christmas gifts",
                                "icon": "gift",
                                "balance": {
                                    "amount": "789.13",
                                    "currencyCode": "EUR"
                                }
                            }, {
                                "id": "cd1235dff",
                                "arrangementId": "2331c9469-09a8-4b4d-a91c946999f-01233",
                                "name": "Adventure holidays",
                                "icon": "holidays",
                                "goal": {
                                    "deadline": "2021-09-18"
                                },
                                "balance": {
                                    "amount": "89.13",
                                    "currencyCode": "EUR"
                                }
                            }, {
                                "id": "987abc12",
                                "arrangementId": "c7dgh-a7d70-ed15-11ea-fec1-0242120002",
                                "name": "Home Renovation",
                                "icon": "home",
                                "goal": {
                                    "amountCurrency": {
                                        "amount": "50000.00",
                                        "currencyCode": "EUR"
                                    },
                                    "progress": 0.5
                                },
                                "balance": {
                                    "amount": "250.00",
                                    "currencyCode": "EUR"
                                }
                            }]
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 500,
                    body: {
                        "message": "Description of error"
                    }
                },
            ]
        },
        {
            urlPattern: "/client-api/v2/pockets",
            method: "POST",
            responses: [
                {
                    status: 201,
                    body: {
                        "id": "546abc12",
                        "arrangementId": "11da7d70-ed15-11ea-adc1-0242ac120002",
                        "name": "Travel Europe",
                        "icon": "travel",
                        "goal": {
                            "amountCurrency": {
                                "amount": "2000.00",
                                "currencyCode": "EUR"
                            },
                            "deadline": "2021-02-28",
                            "progress": 35.5
                        },
                        "balance": {
                            "amount": "-20.00",
                            "currencyCode": "EUR"
                        }
                    }
                },
                {
                    status: 201,
                    body: {
                        "arrangementId": "arrangementId",
                        "goal": {
                            "progress": 0.8008281904610115,
                            "deadline": "2000-01-23",
                            "amountCurrency": {
                                "amount": "amount",
                                "additions": {
                                    "key": "additions"
                                },
                                "currencyCode": "currencyCode"
                            }
                        },
                        "balance": {
                            "amount": "amount",
                            "additions": {
                                "key": "additions"
                            },
                            "currencyCode": "currencyCode"
                        },
                        "name": "name",
                        "icon": "icon",
                        "id": "id"
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 500,
                    body: {
                        "message": "Description of error"
                    }
                },
            ]
        },
        {
            urlPattern: "/client-api/v2/pockets/{pocketId}",
            method: "GET",
            responses: [
                {
                    status: 200,
                    body: {
                        "id": "546abc12",
                        "arrangementId": "11da7d70-ed15-11ea-adc1-0242ac120002",
                        "name": "Travel Europe",
                        "icon": "travel",
                        "goal": {
                            "amountCurrency": {
                                "amount": "2000.00",
                                "currencyCode": "EUR"
                            },
                            "deadline": "2021-02-28",
                            "progress": 35.5
                        },
                        "balance": {
                            "amount": "-20.00",
                            "currencyCode": "EUR"
                        }
                    }
                },
                {
                    status: 200,
                    body: {
                        "arrangementId": "arrangementId",
                        "goal": {
                            "progress": 0.8008281904610115,
                            "deadline": "2000-01-23",
                            "amountCurrency": {
                                "amount": "amount",
                                "additions": {
                                    "key": "additions"
                                },
                                "currencyCode": "currencyCode"
                            }
                        },
                        "balance": {
                            "amount": "amount",
                            "additions": {
                                "key": "additions"
                            },
                            "currencyCode": "currencyCode"
                        },
                        "name": "name",
                        "icon": "icon",
                        "id": "id"
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
                {
                    status: 500,
                    body: {
                        "message": "Description of error"
                    }
                },
            ]
        },
    ]);

    exports.COLLECTION_FORMATS = COLLECTION_FORMATS;
    exports.CONFIG_TOKEN = CONFIG_TOKEN;
    exports.POCKET_TAILOR_BASE_PATH = POCKET_TAILOR_BASE_PATH;
    exports.PocketTailorApiModule = PocketTailorApiModule;
    exports.PocketTailorClientHttpService = PocketTailorClientHttpService;
    exports.PocketTailorClientHttpServiceClosePocketMocksProvider = PocketTailorClientHttpServiceClosePocketMocksProvider;
    exports.PocketTailorClientHttpServiceListPocketsMocksProvider = PocketTailorClientHttpServiceListPocketsMocksProvider;
    exports.PocketTailorClientHttpServiceMocksProvider = PocketTailorClientHttpServiceMocksProvider;
    exports.PocketTailorClientHttpServicePostPocketMocksProvider = PocketTailorClientHttpServicePostPocketMocksProvider;
    exports.PocketTailorClientHttpServiceViewPocketMocksProvider = PocketTailorClientHttpServiceViewPocketMocksProvider;
    exports.PocketTailorConfiguration = PocketTailorConfiguration;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
