// @backbase/ui-ang/esm2015/_example/phone-number-format-pipe/showcase/phone-number-format-pipe-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { PhoneNumberFormatPipeShowcaseComponent } from './phone-number-format-pipe-showcase.component';
import { PhoneNumberModule } from '@backbase/ui-ang/phone-number-format-pipe';
import * as i0 from "@angular/core";
export class PhoneNumberFormatPipeShowcaseModule {
}
PhoneNumberFormatPipeShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PhoneNumberFormatPipeShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PhoneNumberFormatPipeShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PhoneNumberFormatPipeShowcaseModule, declarations: [PhoneNumberFormatPipeShowcaseComponent], imports: [CommonModule, PhoneNumberModule] });
PhoneNumberFormatPipeShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PhoneNumberFormatPipeShowcaseModule, imports: [[CommonModule, PhoneNumberModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PhoneNumberFormatPipeShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, PhoneNumberModule],
                    declarations: [PhoneNumberFormatPipeShowcaseComponent],
                    entryComponents: [PhoneNumberFormatPipeShowcaseComponent],
                }]
        }] });
