// @backbase/ui-ang/esm2015/_example/pagination/examples/pagination.example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/pagination";
export class PaginationExampleComponent {
    constructor() {
        this.page = 0;
        this.totalItems = 100;
    }
    onPageChange(page) {
        this.page = page;
    }
}
PaginationExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaginationExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
PaginationExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: PaginationExampleComponent, selector: "bb-pagination-example-ui", ngImport: i0, template: "<bb-pagination-ui\n  [boundaryLinks]=\"true\"\n  [directionLinks]=\"true\"\n  [itemsPerPage]=\"10\"\n  [maxNavPages]=\"2\"\n  [page]=\"0\"\n  [totalItems]=\"totalItems\"\n  (pageChange)=\"onPageChange($event)\"\n></bb-pagination-ui>\n", components: [{ type: i1.PaginatorComponent, selector: "bb-pagination-ui", inputs: ["page", "boundaryLinks", "directionLinks", "totalItems", "itemsPerPage", "maxNavPages"], outputs: ["pageChange"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaginationExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-pagination-example-ui',
                    templateUrl: './pagination.example.component.html',
                }]
        }] });
