// @backbase/consent-details-widget-ang/esm2015/src/consent-details-widget/model/consent-details-error.js
export class ConsentDetailsAccessDenied {
    constructor(error) {
        this.error = error;
        this.kind = 'ConsentDetailsAccessDenied';
    }
}
export class ConsentDetailsNotFound {
    constructor(error) {
        this.error = error;
        this.kind = 'ConsentDetailsNotFound';
    }
}
export class ConsentDetailsBadRequest {
    constructor(error) {
        this.error = error;
        this.kind = 'ConsentDetailsBadRequest';
    }
}
export class ConsentDetailsUnknownError {
    constructor(error) {
        this.error = error;
        this.kind = 'ConsentDetailsUnknownError';
    }
}
export class ConsentDetailsConnectivityError {
    constructor(error) {
        this.error = error;
        this.kind = 'ConsentDetailsConnectivityError';
    }
}
export const parseError = (error) => {
    if (error.error instanceof ErrorEvent) {
        return new ConsentDetailsUnknownError(error);
    }
    else {
        switch (error.status) {
            case 0:
                return new ConsentDetailsConnectivityError(error);
            case 400:
                return new ConsentDetailsBadRequest(error);
            case 401:
                return new ConsentDetailsAccessDenied(error);
            case 403:
                return new ConsentDetailsAccessDenied(error);
            case 404:
                return new ConsentDetailsNotFound(error);
            default:
                return new ConsentDetailsUnknownError(error);
        }
    }
};
