// @backbase/identity-self-service-journey-ang/bundles/backbase-identity-self-service-journey-ang.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('@angular/router'), require('@backbase/foundation-ang/core'), require('@backbase/identity-common-ang'), require('@backbase/user-identity-security-center-widget-ang'), require('@backbase/user-manage-profile-widget-ang'), require('@backbase/device-information-widget-ang'), require('rxjs'), require('rxjs/operators'), require('@backbase/user-http-ang'), require('@backbase/ui-ang/header'), require('@backbase/ui-ang/icon'), require('@backbase/ui-ang/button'), require('@backbase/ui-ang/notification'), require('@backbase/ui-ang/alert'), require('@angular/forms'), require('@backbase/ui-ang/input-radio-group'), require('@backbase/ui-ang/input-validation-message'), require('@backbase/ui-ang/load-button'), require('@backbase/ui-ang/modal')) :
    typeof define === 'function' && define.amd ? define('@backbase/identity-self-service-journey-ang', ['exports', '@angular/core', '@angular/common', '@angular/router', '@backbase/foundation-ang/core', '@backbase/identity-common-ang', '@backbase/user-identity-security-center-widget-ang', '@backbase/user-manage-profile-widget-ang', '@backbase/device-information-widget-ang', 'rxjs', 'rxjs/operators', '@backbase/user-http-ang', '@backbase/ui-ang/header', '@backbase/ui-ang/icon', '@backbase/ui-ang/button', '@backbase/ui-ang/notification', '@backbase/ui-ang/alert', '@angular/forms', '@backbase/ui-ang/input-radio-group', '@backbase/ui-ang/input-validation-message', '@backbase/ui-ang/load-button', '@backbase/ui-ang/modal'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["identity-self-service-journey-ang"] = {}), global.ng.core, global.ng.common, global.ng.router, global.i1$4, global["@backbase/identity-common-ang"], global["@backbase/user-identity-security-center-widget-ang"], global["@backbase/user-manage-profile-widget-ang"], global["@backbase/device-information-widget-ang"], global.rxjs, global.rxjs.operators, global.i1, global.i3, global.i5, global.i7, global.i3$2, global.i6$1, global.ng.forms, global.i2$3, global.i3$1, global.i4$1, global.i1$3));
})(this, (function (exports, i0, i6, i1$1, i1$4, i4, i2, i2$1, i2$2, rxjs, operators, i1, i3, i5, i7, i3$2, i6$1, i1$2, i2$3, i3$1, i4$1, i1$3) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i6__namespace = /*#__PURE__*/_interopNamespace(i6);
    var i1__namespace$1 = /*#__PURE__*/_interopNamespace(i1$1);
    var i1__namespace$4 = /*#__PURE__*/_interopNamespace(i1$4);
    var i4__namespace = /*#__PURE__*/_interopNamespace(i4);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i2__namespace$1 = /*#__PURE__*/_interopNamespace(i2$1);
    var i2__namespace$2 = /*#__PURE__*/_interopNamespace(i2$2);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i3__namespace = /*#__PURE__*/_interopNamespace(i3);
    var i5__namespace = /*#__PURE__*/_interopNamespace(i5);
    var i7__namespace = /*#__PURE__*/_interopNamespace(i7);
    var i3__namespace$2 = /*#__PURE__*/_interopNamespace(i3$2);
    var i6__namespace$1 = /*#__PURE__*/_interopNamespace(i6$1);
    var i1__namespace$2 = /*#__PURE__*/_interopNamespace(i1$2);
    var i2__namespace$3 = /*#__PURE__*/_interopNamespace(i2$3);
    var i3__namespace$1 = /*#__PURE__*/_interopNamespace(i3$1);
    var i4__namespace$1 = /*#__PURE__*/_interopNamespace(i4$1);
    var i1__namespace$3 = /*#__PURE__*/_interopNamespace(i1$3);

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from, pack) {
        if (pack || arguments.length === 2)
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar)
                        ar = Array.prototype.slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
        return to.concat(ar || Array.prototype.slice.call(from));
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    var IdentitySelfServiceJourneyConfigurationToken = new i0.InjectionToken('IdentitySelfServiceJourneyConfiguration injection token');

    var localizationOptions = [
        Object.freeze({
            label: $localize(templateObject_1$1 || (templateObject_1$1 = __makeTemplateObject([":Label for English language in the United States@@bb-identity-self-service-journey.lanugage.en-us.label:English (United States)"], [":Label for English language in the United States@@bb-identity-self-service-journey.lanugage.en-us.label:English (United States)"]))),
            value: 'en-US',
        }),
        Object.freeze({
            label: $localize(templateObject_2$1 || (templateObject_2$1 = __makeTemplateObject([":Label for Spanish language in the United States@@bb-identity-self-service-journey.lanugage.es-us.label:Espa\u00F1ol (Estados Unidos)"], [":Label for Spanish language in the United States@@bb-identity-self-service-journey.lanugage.es-us.label:Espa\u00F1ol (Estados Unidos)"]))),
            value: 'es-US',
        }),
    ];
    var templateObject_1$1, templateObject_2$1;

    var notificationDismissTime = 3;
    var emailAddressTypes = ['Personal', 'Work'];
    var postalAddressTypes = ['Home', 'Business', 'Mailing'];
    var phoneNumberTypes = ['Home', 'Work', 'Mobile'];
    var maxEmailAddresses = 3;
    var maxPhoneNumbers = 3;
    var maxPostalAddresses = 3;
    var showWidgetHeader = true;
    var phoneNumberCountryCode = 'CA';
    var phoneNumberDebounceTime = 1;
    var enablePhoneNumberFormatter = true;
    var showEditableLanguage = true;
    var routeParamNameUserId = 'userId';
    var configDefaults = {
        userIdentitySecurityCenter: {
            notificationDismissTime: notificationDismissTime,
        },
        userManageProfile: {
            notificationDismissTime: notificationDismissTime,
            emailAddressTypes: __spreadArray([], __read(emailAddressTypes)),
            postalAddressTypes: __spreadArray([], __read(postalAddressTypes)),
            phoneNumberTypes: __spreadArray([], __read(phoneNumberTypes)),
            maxEmailAddresses: maxEmailAddresses,
            maxPhoneNumbers: maxPhoneNumbers,
            maxPostalAddresses: maxPostalAddresses,
            showWidgetHeader: showWidgetHeader,
            phoneNumberCountryCode: phoneNumberCountryCode,
            phoneNumberDebounceTime: phoneNumberDebounceTime,
            enablePhoneNumberFormatter: enablePhoneNumberFormatter,
        },
        deviceInformation: {
            notificationDismissTime: notificationDismissTime,
            routeParamNameUserId: routeParamNameUserId,
        },
        userLocalization: {
            notificationDismissTime: notificationDismissTime,
            localizationOptions: __spreadArray([], __read(localizationOptions)),
            showEditableLanguage: showEditableLanguage,
        },
    };
    var IdentitySelfServiceJourneyConfigurationService = /** @class */ (function () {
        function IdentitySelfServiceJourneyConfigurationService(configOverrides) {
            this.config = i4.createJourneyConfiguration(configDefaults, configOverrides);
        }
        Object.defineProperty(IdentitySelfServiceJourneyConfigurationService.prototype, "userIdentitySecurityCenter", {
            get: function () {
                return this.config.userIdentitySecurityCenter;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(IdentitySelfServiceJourneyConfigurationService.prototype, "userManageProfile", {
            get: function () {
                return this.config.userManageProfile;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(IdentitySelfServiceJourneyConfigurationService.prototype, "deviceInformation", {
            get: function () {
                return this.config.deviceInformation;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(IdentitySelfServiceJourneyConfigurationService.prototype, "userLocalization", {
            get: function () {
                return this.config.userLocalization;
            },
            enumerable: false,
            configurable: true
        });
        return IdentitySelfServiceJourneyConfigurationService;
    }());
    /** @nocollapse */ IdentitySelfServiceJourneyConfigurationService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: IdentitySelfServiceJourneyConfigurationService, deps: [{ token: IdentitySelfServiceJourneyConfigurationToken, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    /** @nocollapse */ IdentitySelfServiceJourneyConfigurationService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: IdentitySelfServiceJourneyConfigurationService, providedIn: 'root' });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: IdentitySelfServiceJourneyConfigurationService, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root',
                    }]
            }], ctorParameters: function () {
            return [{ type: undefined, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.Inject,
                            args: [IdentitySelfServiceJourneyConfigurationToken]
                        }] }];
        } });

    var UserIdentitySecurityCenterViewComponent = /** @class */ (function () {
        function UserIdentitySecurityCenterViewComponent(
        /** @internal */
        configService) {
            this.configService = configService;
        }
        return UserIdentitySecurityCenterViewComponent;
    }());
    /** @nocollapse */ UserIdentitySecurityCenterViewComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: UserIdentitySecurityCenterViewComponent, deps: [{ token: IdentitySelfServiceJourneyConfigurationService }], target: i0__namespace.ɵɵFactoryTarget.Component });
    /** @nocollapse */ UserIdentitySecurityCenterViewComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: UserIdentitySecurityCenterViewComponent, selector: "bb-user-identity-security-center-view", ngImport: i0__namespace, template: "<bb-user-identity-security-center-widget\n    [notificationDismissTime]=\"this.configService.userIdentitySecurityCenter.notificationDismissTime\"\n  >\n  </bb-user-identity-security-center-widget>", isInline: true, components: [{ type: i2__namespace.UserIdentitySecurityCenterWidgetComponent, selector: "bb-user-identity-security-center-widget", inputs: ["notificationDismissTime"], outputs: ["widgetStatus"] }], changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: UserIdentitySecurityCenterViewComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-user-identity-security-center-view',
                        template: "<bb-user-identity-security-center-widget\n    [notificationDismissTime]=\"this.configService.userIdentitySecurityCenter.notificationDismissTime\"\n  >\n  </bb-user-identity-security-center-widget>",
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], ctorParameters: function () { return [{ type: IdentitySelfServiceJourneyConfigurationService }]; } });

    var UserManageProfileViewComponent = /** @class */ (function () {
        function UserManageProfileViewComponent(
        /** @internal */
        configService) {
            this.configService = configService;
        }
        return UserManageProfileViewComponent;
    }());
    /** @nocollapse */ UserManageProfileViewComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: UserManageProfileViewComponent, deps: [{ token: IdentitySelfServiceJourneyConfigurationService }], target: i0__namespace.ɵɵFactoryTarget.Component });
    /** @nocollapse */ UserManageProfileViewComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: UserManageProfileViewComponent, selector: "bb-user-manage-profile-view", ngImport: i0__namespace, template: "<bb-user-manage-profile-widget\n    [notificationDismissTime]=\"this.configService.userManageProfile.notificationDismissTime\"\n    [emailAddressTypes]=\"this.configService.userManageProfile.emailAddressTypes\"\n    [postalAddressTypes]=\"this.configService.userManageProfile.postalAddressTypes\"\n    [phoneNumberTypes]=\"this.configService.userManageProfile.phoneNumberTypes\"\n    [maxEmailAddresses]=\"this.configService.userManageProfile.maxEmailAddresses\"\n    [maxPhoneNumbers]=\"this.configService.userManageProfile.maxPhoneNumbers\"\n    [maxPostalAddresses]=\"this.configService.userManageProfile.maxPostalAddresses\"\n    [showWidgetHeader]=\"this.configService.userManageProfile.showWidgetHeader\"\n    [phoneNumberCountryCode]=\"this.configService.userManageProfile.phoneNumberCountryCode\"\n    [phoneNumberDebounceTime]=\"this.configService.userManageProfile.phoneNumberDebounceTime\"\n    [enablePhoneNumberFormatter]=\"this.configService.userManageProfile.enablePhoneNumberFormatter\"\n  >\n  </bb-user-manage-profile-widget>", isInline: true, components: [{ type: i2__namespace$1.UserManageProfileWidgetComponent, selector: "bb-user-manage-profile-widget", inputs: ["emailAddressTypes", "postalAddressTypes", "phoneNumberTypes", "maxEmailAddresses", "maxPhoneNumbers", "maxPostalAddresses", "showWidgetHeader", "phoneNumberCountryCode", "phoneNumberDebounceTime", "enablePhoneNumberFormatter"] }], changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: UserManageProfileViewComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-user-manage-profile-view',
                        template: "<bb-user-manage-profile-widget\n    [notificationDismissTime]=\"this.configService.userManageProfile.notificationDismissTime\"\n    [emailAddressTypes]=\"this.configService.userManageProfile.emailAddressTypes\"\n    [postalAddressTypes]=\"this.configService.userManageProfile.postalAddressTypes\"\n    [phoneNumberTypes]=\"this.configService.userManageProfile.phoneNumberTypes\"\n    [maxEmailAddresses]=\"this.configService.userManageProfile.maxEmailAddresses\"\n    [maxPhoneNumbers]=\"this.configService.userManageProfile.maxPhoneNumbers\"\n    [maxPostalAddresses]=\"this.configService.userManageProfile.maxPostalAddresses\"\n    [showWidgetHeader]=\"this.configService.userManageProfile.showWidgetHeader\"\n    [phoneNumberCountryCode]=\"this.configService.userManageProfile.phoneNumberCountryCode\"\n    [phoneNumberDebounceTime]=\"this.configService.userManageProfile.phoneNumberDebounceTime\"\n    [enablePhoneNumberFormatter]=\"this.configService.userManageProfile.enablePhoneNumberFormatter\"\n  >\n  </bb-user-manage-profile-widget>",
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], ctorParameters: function () { return [{ type: IdentitySelfServiceJourneyConfigurationService }]; } });

    var DeviceInformationViewComponent = /** @class */ (function () {
        function DeviceInformationViewComponent(
        /** @internal */
        configService) {
            this.configService = configService;
        }
        return DeviceInformationViewComponent;
    }());
    /** @nocollapse */ DeviceInformationViewComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: DeviceInformationViewComponent, deps: [{ token: IdentitySelfServiceJourneyConfigurationService }], target: i0__namespace.ɵɵFactoryTarget.Component });
    /** @nocollapse */ DeviceInformationViewComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: DeviceInformationViewComponent, selector: "bb-device-information-view", ngImport: i0__namespace, template: "<bb-device-information-widget\n    [dismissTimeout]=\"this.configService.deviceInformation.notificationDismissTime\"\n    [isAdminWidget]=\"false\"\n  >\n  </bb-device-information-widget>", isInline: true, components: [{ type: i2__namespace$2.DeviceInformationWidgetComponent, selector: "bb-device-information-widget", inputs: ["dismissTimeout", "isAdminWidget", "dbsUserId"] }], changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: DeviceInformationViewComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-device-information-view',
                        template: "<bb-device-information-widget\n    [dismissTimeout]=\"this.configService.deviceInformation.notificationDismissTime\"\n    [isAdminWidget]=\"false\"\n  >\n  </bb-device-information-widget>",
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], ctorParameters: function () { return [{ type: IdentitySelfServiceJourneyConfigurationService }]; } });

    // eslint-disable-next-line @typescript-eslint/ban-types
    var cacheItem = function (queryMethod) {
        var subject$ = new rxjs.BehaviorSubject(undefined);
        var update = function () { return queryMethod().pipe(operators.tap(function (payload) { return subject$.next(payload); })); };
        return {
            subject$: subject$,
            state$: subject$.pipe(operators.switchMap(function (payload) { return (payload !== undefined ? rxjs.of(payload) : update()); })),
            update: update,
        };
    };

    var findLanguageRegionOption = function (localizationOptionsConfig, preferredLanguage) {
        if (localizationOptionsConfig === void 0) { localizationOptionsConfig = []; }
        return localizationOptionsConfig.find(function (item) { return item.value === preferredLanguage; }) || {
            isDefault: true,
            value: typeof preferredLanguage === 'string' ? preferredLanguage : undefined,
            label: '',
        };
    };

    var UserLocalizationService = /** @class */ (function () {
        function UserLocalizationService(dataService, 
        /** @internal */
        configService) {
            this.dataService = dataService;
            this.configService = configService;
            this.getUserCacheItem = cacheItem(this.dataService.getOwnUser.bind(this.dataService));
            this.getUser$ = this.getUserCacheItem.state$;
            this.getUser = this.getUserCacheItem.update;
            this.defaultLocalizationOptions = localizationOptions;
            this.config = Object.freeze(Object.assign(Object.assign({}, this.configService.userLocalization), { notificationDismissTime: (this.configService.userLocalization.notificationDismissTime || 0) * 1000 }));
        }
        UserLocalizationService.prototype.updateUser = function (userVO) {
            return this.dataService.updateOwnLanguage({
                updatePreferredLanguage: {
                    preferredLanguage: userVO.preferredLanguage || '',
                },
            });
        };
        return UserLocalizationService;
    }());
    /** @nocollapse */ UserLocalizationService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: UserLocalizationService, deps: [{ token: i1__namespace.UserManagementService }, { token: IdentitySelfServiceJourneyConfigurationService }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    /** @nocollapse */ UserLocalizationService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: UserLocalizationService });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: UserLocalizationService, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () { return [{ type: i1__namespace.UserManagementService }, { type: IdentitySelfServiceJourneyConfigurationService }]; } });

    var UserLocalizationComponent = /** @class */ (function () {
        function UserLocalizationComponent() {
        }
        return UserLocalizationComponent;
    }());
    /** @nocollapse */ UserLocalizationComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: UserLocalizationComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    /** @nocollapse */ UserLocalizationComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: UserLocalizationComponent, selector: "bb-user-localization", providers: [UserLocalizationService], ngImport: i0__namespace, template: "<router-outlet></router-outlet>", isInline: true, directives: [{ type: i1__namespace$1.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }], changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: UserLocalizationComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-user-localization',
                        template: "<router-outlet></router-outlet>",
                        providers: [UserLocalizationService],
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }] });

    var userLanguageCommonViewStatePipe = function (userLocalizationService) { return rxjs.pipe(operators.switchMap(function () { return userLocalizationService.getUser$.pipe(operators.map(function (userProfile) { return ({
        loading: false,
        languageRegion: findLanguageRegionOption(userLocalizationService.config.localizationOptions, userProfile.preferredLanguage),
    }); }), operators.startWith({ loading: true }), operators.catchError(function (error) { return rxjs.of({ loading: false, error: error }); })); })); };

    var UserLocalizationCommunicationsViewComponent = /** @class */ (function () {
        function UserLocalizationCommunicationsViewComponent(router, userLocalizationService, activatedRoute) {
            this.router = router;
            this.userLocalizationService = userLocalizationService;
            this.activatedRoute = activatedRoute;
            this.viewStateSubject$ = new rxjs.BehaviorSubject(undefined);
            this.viewState$ = this.viewStateSubject$.pipe(userLanguageCommonViewStatePipe(this.userLocalizationService), operators.share());
        }
        UserLocalizationCommunicationsViewComponent.prototype.changeLanguage = function () {
            var _a, _b;
            this.router.navigate([((_b = (_a = this.activatedRoute.snapshot) === null || _a === void 0 ? void 0 : _a.data) === null || _b === void 0 ? void 0 : _b.changeLanguagePath) || '../change-language'], {
                relativeTo: this.activatedRoute,
            });
        };
        UserLocalizationCommunicationsViewComponent.prototype.getLanguageLabelFor = function (value) {
            var _a;
            return (_a = this.userLocalizationService.defaultLocalizationOptions.find(function (option) { return option.value === value; })) === null || _a === void 0 ? void 0 : _a.label;
        };
        Object.defineProperty(UserLocalizationCommunicationsViewComponent.prototype, "showEditableLanguage", {
            get: function () {
                return this.userLocalizationService.config.showEditableLanguage;
            },
            enumerable: false,
            configurable: true
        });
        return UserLocalizationCommunicationsViewComponent;
    }());
    /** @nocollapse */ UserLocalizationCommunicationsViewComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: UserLocalizationCommunicationsViewComponent, deps: [{ token: i1__namespace$1.Router }, { token: UserLocalizationService }, { token: i1__namespace$1.ActivatedRoute }], target: i0__namespace.ɵɵFactoryTarget.Component });
    /** @nocollapse */ UserLocalizationCommunicationsViewComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: UserLocalizationCommunicationsViewComponent, selector: "bb-user-localization-communications-view", ngImport: i0__namespace, template: "<div data-role=\"user-localization-communications\">\n  <div class=\"bb-block bb-block--lg\">\n    <bb-header-ui\n      headingType=\"h1\"\n      data-role=\"user-localization-communications-header\"\n      heading=\"Communications\"\n      i18n-heading=\"Header for user language communications@@bb-user-localization-communications.header\"\n    >\n    </bb-header-ui>\n  </div>\n\n  <div class=\"card\" *ngIf=\"viewState$ | async as state\">\n    <div class=\"card-body\">\n      <bb-identity-view-state-container-component [error]=\"state.error\" [loading]=\"state.loading\">\n        <div class=\"bb-block bb-block--lg\">\n          <bb-header-ui\n            headingType=\"h2\"\n            data-role=\"user-localization-communications-header-external\"\n            heading=\"External communications\"\n            i18n-heading=\"\n              Header for user language external communications@@bb-user-localization-communications.header-external\"\n          >\n          </bb-header-ui>\n        </div>\n\n        <div class=\"bb-stack\" data-role=\"user-localization-communications-item\">\n          <div class=\"bb-stack__item bb-stack__item--fill\">\n            <div class=\"bb-stack bb-stack--wrap\">\n              <div class=\"bb-stack__item\">\n                <p\n                  class=\"bb-block bb-block--sm bb-subtitle bb-text-support\"\n                  i18n=\"Label for language and region@@bb-user-localization-communications.lanugage.label\"\n                >\n                  Language and region\n                </p>\n\n                <p\n                  *ngIf=\"\n                    state.languageRegion?.label ||\n                      getLanguageLabelFor(\n                        (state.languageRegion?.isDefault && state.languageRegion?.value) || ''\n                      ) as label;\n                    else noSelectedLabel\n                  \"\n                  data-role=\"user-localization-communications-chosen-label\"\n                  i18n=\"Chosen language and region@@bb-user-localization-communications.lanugage.selected\"\n                >\n                  {{ label }}\n                </p>\n                <ng-template #noSelectedLabel>\n                  <p\n                    data-role=\"user-localization-communications-default-label\"\n                    i18n=\"No language selected label@@bb-user-localization-communications.lanugage.not-selected\"\n                  >\n                    No language selected.\n                  </p>\n                </ng-template>\n              </div>\n              <div class=\"bb-stack__break bb-stack__break--sm-down\"></div>\n              <div\n                class=\"bb-stack__item bb-stack__item--fill bb-text-align-right bb-text-align-left--sm-down break-word\"\n              >\n                <button\n                  *ngIf=\"showEditableLanguage\"\n                  bbButton\n                  data-role=\"user-localization-communications-edit-button\"\n                  aria-label=\"Change your language and region\"\n                  i18n-aria-label=\"\n                    Change your language and region aria label@@bb-user-localization-communications.action.change\"\n                  type=\"button\"\n                  color=\"secondary\"\n                  [circle]=\"true\"\n                  (click)=\"changeLanguage()\"\n                >\n                  <bb-icon-ui name=\"edit\" size=\"md\" color=\"info\"> </bb-icon-ui>\n                </button>\n              </div>\n            </div>\n          </div>\n        </div>\n      </bb-identity-view-state-container-component>\n    </div>\n  </div>\n</div>\n", components: [{ type: i3__namespace.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i4__namespace.ViewStateContainerComponent, selector: "bb-identity-view-state-container-component", inputs: ["error", "loading"] }, { type: i5__namespace.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], directives: [{ type: i6__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i7__namespace.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }], pipes: { "async": i6__namespace.AsyncPipe } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: UserLocalizationCommunicationsViewComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-user-localization-communications-view',
                        templateUrl: 'user-localization-communications-view.component.html',
                    }]
            }], ctorParameters: function () { return [{ type: i1__namespace$1.Router }, { type: UserLocalizationService }, { type: i1__namespace$1.ActivatedRoute }]; } });

    var BackButtonComponent = /** @class */ (function () {
        function BackButtonComponent() {
            /**
             * Output to track when back was clicked
             */
            this.backClick = new i0.EventEmitter();
        }
        return BackButtonComponent;
    }());
    /** @nocollapse */ BackButtonComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: BackButtonComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    /** @nocollapse */ BackButtonComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: BackButtonComponent, selector: "bb-identity-back-button", inputs: { page: "page" }, outputs: { backClick: "backClick" }, ngImport: i0__namespace, template: "<div class=\"bb-block bb-block--lg\" data-role=\"back-button-wrapper\">\n  <button bbButton color=\"link-text\" data-role=\"back-button\" type=\"button\" buttonSize=\"sm\" (click)=\"backClick.emit()\">\n    <bb-icon-ui name=\"arrow-back\"></bb-icon-ui>\n    <span i18n=\"Back to button@@bb-identity.back-button\"> Back to {{ page }} </span>\n  </button>\n</div>\n", components: [{ type: i5__namespace.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], directives: [{ type: i7__namespace.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }], changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: BackButtonComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-identity-back-button',
                        templateUrl: './back-button.component.html',
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], propDecorators: { page: [{
                    type: i0.Input
                }], backClick: [{
                    type: i0.Output
                }] } });

    var UserLocalizationChangeLanguageFormComponent = /** @class */ (function () {
        function UserLocalizationChangeLanguageFormComponent(formBuilder) {
            this.formBuilder = formBuilder;
            /**
             * Set language options
             */
            this.options = [];
            /**
             * Set save loading language
             */
            this.saveLoading = false;
            /**
             * Output for cancel clicked
             */
            this.cancelClick = new i0.EventEmitter();
            /**
             * Output for save clicked
             */
            this.saveClick = new i0.EventEmitter();
            this.ids = {
                languageRegionLabel: i4.generateId('language-region'),
            };
            this.isSubmitted = false;
        }
        UserLocalizationChangeLanguageFormComponent.prototype.ngOnInit = function () {
            this.form = this.formBuilder.group({
                languageRegion: [{ value: this.selectedLanguageCode, disabled: false }, i1$2.Validators.required],
            });
        };
        UserLocalizationChangeLanguageFormComponent.prototype.save = function () {
            this.isSubmitted = true;
            if (this.form && this.form.valid) {
                this.saveClick.emit({ preferredLanguage: this.form.value.languageRegion });
            }
        };
        Object.defineProperty(UserLocalizationChangeLanguageFormComponent.prototype, "showLanguageControlError", {
            get: function () {
                return !((this.form || {}).controls || {}).languageRegion.value && this.isSubmitted;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(UserLocalizationChangeLanguageFormComponent.prototype, "defaultLocalizationOptions", {
            get: function () {
                return localizationOptions;
            },
            enumerable: false,
            configurable: true
        });
        return UserLocalizationChangeLanguageFormComponent;
    }());
    /** @nocollapse */ UserLocalizationChangeLanguageFormComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: UserLocalizationChangeLanguageFormComponent, deps: [{ token: i1__namespace$2.FormBuilder }], target: i0__namespace.ɵɵFactoryTarget.Component });
    /** @nocollapse */ UserLocalizationChangeLanguageFormComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: UserLocalizationChangeLanguageFormComponent, selector: "bb-user-localization-change-language-form", inputs: { options: "options", selectedLanguageCode: "selectedLanguageCode", saveLoading: "saveLoading" }, outputs: { cancelClick: "cancelClick", saveClick: "saveClick" }, ngImport: i0__namespace, template: "<form *ngIf=\"form\" [formGroup]=\"form\" (submit)=\"save()\">\n  <div class=\"bb-block bb-block--lg\">\n    <bb-input-radio-group-ui\n      data-role=\"user-localization-change-language-form-radio-option-group\"\n      formControlName=\"languageRegion\"\n      label=\"Language and region\"\n      i18n-label=\"Label for language and region radio options@@bb-user-localization-change-language-form.label.lanugage\"\n      [preselect]=\"false\"\n      [required]=\"true\"\n    >\n      <ng-container *ngIf=\"(options.length && options) || defaultLocalizationOptions as localizationOptions\">\n        <bb-input-radio-ui\n          *ngFor=\"let option of localizationOptions\"\n          [value]=\"option.value\"\n          [labelTemplate]=\"radioLabelTemplate\"\n          [labelTemplateContext]=\"option\"\n        >\n        </bb-input-radio-ui>\n      </ng-container>\n    </bb-input-radio-group-ui>\n    <bb-input-validation-message-ui [showErrors]=\"showLanguageControlError\">\n      <span\n        i18n=\"\n          Language required error|Error to say that language is a required\n          field@@bb-user-localization-change-language-form.language.errors.required\"\n      >\n        You must choose a language\n      </span>\n    </bb-input-validation-message-ui>\n  </div>\n\n  <div class=\"bb-button-bar bb-button-bar--reverse\">\n    <bb-load-button-ui\n      type=\"submit\"\n      data-role=\"user-localization-change-language-form-save-button\"\n      class=\"bb-button-bar__button\"\n      [isLoading]=\"saveLoading\"\n      i18n=\"Save changes button action@@bb-user-localization-change-language-form.action.save\"\n    >\n      Save changes\n    </bb-load-button-ui>\n    <button\n      bbButton\n      data-role=\"user-localization-change-language-form-cancel-button\"\n      color=\"secondary\"\n      class=\"bb-button-bar__button\"\n      type=\"button\"\n      (click)=\"cancelClick.emit()\"\n      i18n=\"Cancel changes button action@@bb-user-localization-change-language-form.action.cancel\"\n    >\n      Cancel\n    </button>\n  </div>\n</form>\n\n<ng-template #radioLabelTemplate let-labelTemplateContext=\"labelTemplateContext\">\n  <span class=\"bb-input-radio-group__radio-label\" [lang]=\"labelTemplateContext.value\">\n    {{ labelTemplateContext.label }}\n  </span>\n</ng-template>\n", components: [{ type: i2__namespace$3.InputRadioGroupComponent, selector: "bb-input-radio-group-ui", inputs: ["horizontal", "preselect", "name", "verticalAlign"] }, { type: i2__namespace$3.InputRadioComponent, selector: "bb-input-radio-ui", inputs: ["labelTemplate", "labelTemplateContext", "value", "label", "aria-label", "aria-labelledby", "aria-describedby", "role", "aria-activedescendant", "aria-expanded", "aria-invalid", "aria-owns", "disabled"] }, { type: i3__namespace$1.InputValidationMessageComponent, selector: "bb-input-validation-message-ui", inputs: ["showErrors", "messageId"] }, { type: i4__namespace$1.LoadButtonComponent, selector: "bb-load-button-ui", inputs: ["isLoading", "from", "size", "circle", "totalCount"], outputs: ["paramChange"] }], directives: [{ type: i6__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i1__namespace$2.ɵNgNoValidate, selector: "form:not([ngNoForm]):not([ngNativeValidate])" }, { type: i1__namespace$2.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { type: i1__namespace$2.FormGroupDirective, selector: "[formGroup]", inputs: ["formGroup"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { type: i1__namespace$2.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { type: i1__namespace$2.FormControlName, selector: "[formControlName]", inputs: ["disabled", "formControlName", "ngModel"], outputs: ["ngModelChange"] }, { type: i1__namespace$2.RequiredValidator, selector: ":not([type=checkbox])[required][formControlName],:not([type=checkbox])[required][formControl],:not([type=checkbox])[required][ngModel]", inputs: ["required"] }, { type: i6__namespace.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { type: i7__namespace.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }], changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: UserLocalizationChangeLanguageFormComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-user-localization-change-language-form',
                        templateUrl: './user-localization-change-language-form.component.html',
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], ctorParameters: function () { return [{ type: i1__namespace$2.FormBuilder }]; }, propDecorators: { options: [{
                    type: i0.Input
                }], selectedLanguageCode: [{
                    type: i0.Input
                }], saveLoading: [{
                    type: i0.Input
                }], cancelClick: [{
                    type: i0.Output
                }], saveClick: [{
                    type: i0.Output
                }] } });

    var DiscardModalComponent = /** @class */ (function () {
        function DiscardModalComponent() {
            /**
             * Set whether modal show be open
             */
            this.modalOpen = false;
            /**
             * Output to track when cancel was clicked
             */
            this.cancelModalClick = new i0.EventEmitter();
            /**
             * Output to track when discard was clicked
             */
            this.discardChangesClick = new i0.EventEmitter();
        }
        return DiscardModalComponent;
    }());
    /** @nocollapse */ DiscardModalComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: DiscardModalComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    /** @nocollapse */ DiscardModalComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: DiscardModalComponent, selector: "bb-identity-discard-modal-component", inputs: { modalOpen: "modalOpen" }, outputs: { cancelModalClick: "cancelModalClick", discardChangesClick: "discardChangesClick" }, ngImport: i0__namespace, template: "<bb-modal-ui data-role=\"discard-modal\" [isOpen]=\"modalOpen\" (cancel)=\"cancelModalClick.emit()\">\n  <bb-modal-header-ui>\n    <ng-template bbCustomModalHeader>\n      <div data-role=\"discard-modal-header\">\n        <bb-header-ui\n          headingType=\"h3\"\n          i18n-heading=\"Discard changes modal title@@bb-identity-discard-modal.title\"\n          heading=\"Discard changes?\"\n        >\n        </bb-header-ui>\n      </div>\n    </ng-template>\n  </bb-modal-header-ui>\n  <bb-modal-body-ui>\n    <ng-template bbCustomModalBody i18n=\"Discard changes modal body@@bb-identity-discard-modal.body\">\n      Any unsaved changes you've made will be lost.\n    </ng-template>\n  </bb-modal-body-ui>\n  <bb-modal-footer-ui>\n    <ng-template bbCustomModalFooter>\n      <button\n        bbButton\n        data-role=\"discard-modal-cancel-button\"\n        color=\"secondary\"\n        (click)=\"cancelModalClick.emit()\"\n        i18n=\"Dismiss discard modal button@@bb-identity-discard-modal.cancel\"\n      >\n        Cancel\n      </button>\n      <button\n        bbButton\n        data-role=\"discard-modal-confirm-button\"\n        (click)=\"discardChangesClick.emit()\"\n        i18n=\"Confirm discard modal button@@bb-identity-discard-modal.confirm\"\n      >\n        Discard\n      </button>\n    </ng-template>\n  </bb-modal-footer-ui>\n</bb-modal-ui>\n", components: [{ type: i1__namespace$3.ModalComponent, selector: "bb-modal-ui", inputs: ["isOpen", "aria-labelledby", "aria-describedby", "modalOptions"], outputs: ["confirm", "cancel", "isOpenChange"] }, { type: i1__namespace$3.ModalHeaderComponent, selector: "bb-modal-header-ui", inputs: ["title"], outputs: ["close"] }, { type: i3__namespace.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i1__namespace$3.ModalBodyComponent, selector: "bb-modal-body-ui" }, { type: i1__namespace$3.ModalFooterComponent, selector: "bb-modal-footer-ui", inputs: ["confirmText", "cancelText"], outputs: ["confirm", "cancel"] }], directives: [{ type: i1__namespace$3.CustomModalHeaderDirective, selector: "ng-template[bbCustomModalHeader]" }, { type: i1__namespace$3.CustomModalBodyDirective, selector: "ng-template[bbCustomModalBody]" }, { type: i1__namespace$3.CustomModalFooterDirective, selector: "ng-template[bbCustomModalFooter]" }, { type: i7__namespace.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }], changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: DiscardModalComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-identity-discard-modal-component',
                        templateUrl: './discard-modal.component.html',
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], propDecorators: { modalOpen: [{
                    type: i0.Input
                }], cancelModalClick: [{
                    type: i0.Output
                }], discardChangesClick: [{
                    type: i0.Output
                }] } });

    var UserLocalizationChangeLanguageViewComponent = /** @class */ (function () {
        function UserLocalizationChangeLanguageViewComponent(userLocalizationService, router, notificationService, activatedRoute) {
            this.userLocalizationService = userLocalizationService;
            this.router = router;
            this.notificationService = notificationService;
            this.activatedRoute = activatedRoute;
            this.modalOpen = false;
            this.isSaveLoadingSubject$ = new rxjs.BehaviorSubject(false);
            this.viewStateSubject$ = new rxjs.BehaviorSubject(undefined);
            this.viewState$ = this.viewStateSubject$.pipe(userLanguageCommonViewStatePipe(this.userLocalizationService), operators.share());
            this.ids = {
                languageRegionLabel: i4.generateId('language-region'),
            };
        }
        UserLocalizationChangeLanguageViewComponent.prototype.saveChanges = function (_c) {
            var _this = this;
            var preferredLanguage = _c.preferredLanguage;
            this.isSaveLoadingSubject$.next(true);
            this.userLocalizationService
                .updateUser({ preferredLanguage: preferredLanguage })
                .pipe(operators.switchMap(function () { return _this.userLocalizationService.getUser(); }), operators.finalize(function () { return _this.isSaveLoadingSubject$.next(false); }))
                .subscribe({
                next: function () {
                    _this.showNotification(_this.notificationChangeLanguageSuccess, _this.notificationChangeLanguageSuccessHeader);
                    _this.navigateCommunications();
                },
                error: function () {
                    _this.showNotification(_this.notificationFailure, undefined, 'error');
                },
            });
        };
        UserLocalizationChangeLanguageViewComponent.prototype.cancelChangeLanguage = function () {
            this.modalOpen = true;
        };
        UserLocalizationChangeLanguageViewComponent.prototype.cancelModal = function () {
            this.modalOpen = false;
        };
        UserLocalizationChangeLanguageViewComponent.prototype.discardChanges = function () {
            this.modalOpen = false;
            this.navigateCommunications();
        };
        UserLocalizationChangeLanguageViewComponent.prototype.navigateCommunications = function () {
            var _a, _b;
            this.router.navigate([((_b = (_a = this.activatedRoute.snapshot) === null || _a === void 0 ? void 0 : _a.data) === null || _b === void 0 ? void 0 : _b.communicationsPath) || '../communications'], {
                relativeTo: this.activatedRoute,
            });
        };
        UserLocalizationChangeLanguageViewComponent.prototype.showNotification = function (message, header, modifier) {
            this.notificationService.showNotification({
                header: header,
                message: message,
                modifier: modifier || 'success',
                ttl: this.userLocalizationService.config.notificationDismissTime,
            });
        };
        Object.defineProperty(UserLocalizationChangeLanguageViewComponent.prototype, "localizationOptions", {
            get: function () {
                return this.userLocalizationService.config.localizationOptions;
            },
            enumerable: false,
            configurable: true
        });
        return UserLocalizationChangeLanguageViewComponent;
    }());
    /** @nocollapse */ UserLocalizationChangeLanguageViewComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: UserLocalizationChangeLanguageViewComponent, deps: [{ token: UserLocalizationService }, { token: i1__namespace$1.Router }, { token: i3__namespace$2.NotificationService }, { token: i1__namespace$1.ActivatedRoute }], target: i0__namespace.ɵɵFactoryTarget.Component });
    /** @nocollapse */ UserLocalizationChangeLanguageViewComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: UserLocalizationChangeLanguageViewComponent, selector: "bb-user-localization-change-language-view", viewQueries: [{ propertyName: "notificationChangeLanguageSuccess", first: true, predicate: ["notificationChangeLanguageSuccess"], descendants: true }, { propertyName: "notificationChangeLanguageSuccessHeader", first: true, predicate: ["notificationChangeLanguageSuccessHeader"], descendants: true }, { propertyName: "notificationFailure", first: true, predicate: ["notificationFailure"], descendants: true }], ngImport: i0__namespace, template: "<div *ngIf=\"viewState$ | async as state\" data-role=\"user-localization-change-language\">\n  <bb-identity-back-button\n    page=\"Communications\"\n    i18n-page=\"Page description for change language back button@@bb-user-localization-change-language.back-link-page\"\n    (backClick)=\"cancelChangeLanguage()\"\n  >\n  </bb-identity-back-button>\n\n  <div class=\"bb-block bb-block--lg\">\n    <bb-header-ui\n      headingType=\"h1\"\n      data-role=\"user-localization-change-language-header\"\n      heading=\"Change language and region\"\n      i18n-heading=\"Header for change language@@bb-user-localization-change-language.header\"\n    >\n    </bb-header-ui>\n  </div>\n\n  <div class=\"bb-block bb-block--lg\" *ngIf=\"!state.error\">\n    <bb-alert-ui\n      modifier=\"info\"\n      title=\"This only affects external communication including emails, SMS and push notifications.\"\n      i18n-title=\"Alert info notification for change language region@@bb-user-localization-change-language.info\"\n    >\n    </bb-alert-ui>\n  </div>\n\n  <div class=\"card\">\n    <div class=\"card-body\">\n      <bb-identity-view-state-container-component [error]=\"state.error\" [loading]=\"state.loading\">\n        <bb-user-localization-change-language-form\n          [selectedLanguageCode]=\"state.languageRegion?.value || localizationOptions[0]?.value\"\n          [options]=\"localizationOptions\"\n          [saveLoading]=\"(isSaveLoadingSubject$ | async) || false\"\n          (cancelClick)=\"cancelChangeLanguage()\"\n          (saveClick)=\"saveChanges($event)\"\n        >\n        </bb-user-localization-change-language-form>\n      </bb-identity-view-state-container-component>\n    </div>\n  </div>\n\n  <bb-identity-discard-modal-component\n    [modalOpen]=\"modalOpen\"\n    (cancelModalClick)=\"cancelModal()\"\n    (discardChangesClick)=\"discardChanges()\"\n  >\n  </bb-identity-discard-modal-component>\n\n  <ng-template #notificationChangeLanguageSuccessHeader>\n    <strong\n      data-role=\"user-localization-success-notification-header\"\n      i18n=\"\n        Language and region changed success notification\n        header@@bb-user-localization-change-language.notification.success-header\"\n    >\n      Language and region changed successfully\n    </strong>\n  </ng-template>\n\n  <ng-template #notificationChangeLanguageSuccess>\n    <span\n      data-role=\"user-localization-success-notification\"\n      i18n=\"Language and region changed success notification@@bb-user-localization-change-language.notification.success\"\n    >\n      All future emails, SMS and push notification will use your new language and region.\n    </span>\n  </ng-template>\n\n  <ng-template #notificationFailure>\n    <span\n      data-role=\"user-localization-error-notification\"\n      i18n=\"\n        An unexpected error occurred notification@@bb-user-localization-change-language.notification.unexpected-error\"\n    >\n      An unexpected error occurred\n    </span>\n  </ng-template>\n</div>\n", components: [{ type: BackButtonComponent, selector: "bb-identity-back-button", inputs: ["page"], outputs: ["backClick"] }, { type: i3__namespace.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i6__namespace$1.AlertComponent, selector: "bb-alert-ui", inputs: ["modifier", "dismissible", "title", "message"], outputs: ["close"] }, { type: i4__namespace.ViewStateContainerComponent, selector: "bb-identity-view-state-container-component", inputs: ["error", "loading"] }, { type: UserLocalizationChangeLanguageFormComponent, selector: "bb-user-localization-change-language-form", inputs: ["options", "selectedLanguageCode", "saveLoading"], outputs: ["cancelClick", "saveClick"] }, { type: DiscardModalComponent, selector: "bb-identity-discard-modal-component", inputs: ["modalOpen"], outputs: ["cancelModalClick", "discardChangesClick"] }], directives: [{ type: i6__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "async": i6__namespace.AsyncPipe } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: UserLocalizationChangeLanguageViewComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-user-localization-change-language-view',
                        templateUrl: 'user-localization-change-language-view.component.html',
                    }]
            }], ctorParameters: function () { return [{ type: UserLocalizationService }, { type: i1__namespace$1.Router }, { type: i3__namespace$2.NotificationService }, { type: i1__namespace$1.ActivatedRoute }]; }, propDecorators: { notificationChangeLanguageSuccess: [{
                    type: i0.ViewChild,
                    args: ['notificationChangeLanguageSuccess']
                }], notificationChangeLanguageSuccessHeader: [{
                    type: i0.ViewChild,
                    args: ['notificationChangeLanguageSuccessHeader']
                }], notificationFailure: [{
                    type: i0.ViewChild,
                    args: ['notificationFailure']
                }] } });

    var IdentitySelfServiceJourneyComponent = /** @class */ (function () {
        function IdentitySelfServiceJourneyComponent() {
        }
        return IdentitySelfServiceJourneyComponent;
    }());
    /** @nocollapse */ IdentitySelfServiceJourneyComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: IdentitySelfServiceJourneyComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    /** @nocollapse */ IdentitySelfServiceJourneyComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: IdentitySelfServiceJourneyComponent, selector: "bb-identity-self-service-journey", ngImport: i0__namespace, template: "<bb-routable-tab-set data-role=\"user-context-routable-tab-set\"></bb-routable-tab-set>\n", components: [{ type: i4__namespace.RoutableTabComponent, selector: "bb-routable-tab-set" }], changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: IdentitySelfServiceJourneyComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-identity-self-service-journey',
                        templateUrl: './identity-self-service-journey.component.html',
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }] });

    var uiModules$1 = [i3.HeaderModule, i7.ButtonModule, i1$3.ModalModule, i5.IconModule];
    var components$1 = [DiscardModalComponent, BackButtonComponent];
    var SelfServiceUiModule = /** @class */ (function () {
        function SelfServiceUiModule() {
        }
        return SelfServiceUiModule;
    }());
    /** @nocollapse */ SelfServiceUiModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: SelfServiceUiModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    /** @nocollapse */ SelfServiceUiModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: SelfServiceUiModule, declarations: [DiscardModalComponent, BackButtonComponent], imports: [i6.CommonModule, i3.HeaderModule, i7.ButtonModule, i1$3.ModalModule, i5.IconModule], exports: [i3.HeaderModule, i7.ButtonModule, i1$3.ModalModule, i5.IconModule, DiscardModalComponent, BackButtonComponent] });
    /** @nocollapse */ SelfServiceUiModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: SelfServiceUiModule, imports: [__spreadArray([i6.CommonModule], __read(uiModules$1)), i3.HeaderModule, i7.ButtonModule, i1$3.ModalModule, i5.IconModule] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: SelfServiceUiModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: __spreadArray([i6.CommonModule], __read(uiModules$1)),
                        declarations: __spreadArray([], __read(components$1)),
                        exports: __spreadArray(__spreadArray([], __read(uiModules$1)), __read(components$1)),
                    }]
            }] });

    var uiModules = [
        i3.HeaderModule,
        i5.IconModule,
        i7.ButtonModule,
        i4$1.LoadButtonModule,
        i6$1.AlertModule,
        i2$3.InputRadioGroupModule,
        i1$3.ModalModule,
        i3$2.NotificationModule,
        i3$1.InputValidationMessageModule,
    ];
    var components = [
        UserIdentitySecurityCenterViewComponent,
        UserManageProfileViewComponent,
        DeviceInformationViewComponent,
        UserLocalizationComponent,
        UserLocalizationCommunicationsViewComponent,
        UserLocalizationChangeLanguageViewComponent,
        UserLocalizationChangeLanguageFormComponent,
    ];
    var widgetModules = [
        i2.UserIdentitySecurityCenterWidgetModule,
        i2$1.UserManageProfileWidgetModule,
        i2$2.DeviceInformationWidgetModule,
    ];
    var SelfServiceFeaturesModule = /** @class */ (function () {
        function SelfServiceFeaturesModule() {
        }
        return SelfServiceFeaturesModule;
    }());
    /** @nocollapse */ SelfServiceFeaturesModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: SelfServiceFeaturesModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    /** @nocollapse */ SelfServiceFeaturesModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: SelfServiceFeaturesModule, declarations: [UserIdentitySecurityCenterViewComponent,
            UserManageProfileViewComponent,
            DeviceInformationViewComponent,
            UserLocalizationComponent,
            UserLocalizationCommunicationsViewComponent,
            UserLocalizationChangeLanguageViewComponent,
            UserLocalizationChangeLanguageFormComponent], imports: [i6.CommonModule,
            i1$2.ReactiveFormsModule,
            i1$1.RouterModule, i3.HeaderModule,
            i5.IconModule,
            i7.ButtonModule,
            i4$1.LoadButtonModule,
            i6$1.AlertModule,
            i2$3.InputRadioGroupModule,
            i1$3.ModalModule,
            i3$2.NotificationModule,
            i3$1.InputValidationMessageModule, i2.UserIdentitySecurityCenterWidgetModule,
            i2$1.UserManageProfileWidgetModule,
            i2$2.DeviceInformationWidgetModule, i4.IdentityCommonAngModule,
            SelfServiceUiModule], exports: [i1$1.RouterModule, UserIdentitySecurityCenterViewComponent,
            UserManageProfileViewComponent,
            DeviceInformationViewComponent,
            UserLocalizationComponent,
            UserLocalizationCommunicationsViewComponent,
            UserLocalizationChangeLanguageViewComponent,
            UserLocalizationChangeLanguageFormComponent, i2.UserIdentitySecurityCenterWidgetModule,
            i2$1.UserManageProfileWidgetModule,
            i2$2.DeviceInformationWidgetModule] });
    /** @nocollapse */ SelfServiceFeaturesModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: SelfServiceFeaturesModule, imports: [__spreadArray(__spreadArray(__spreadArray([
                i6.CommonModule,
                i1$2.ReactiveFormsModule,
                i1$1.RouterModule
            ], __read(uiModules)), __read(widgetModules)), [
                i4.IdentityCommonAngModule,
                SelfServiceUiModule,
            ]), i1$1.RouterModule, i2.UserIdentitySecurityCenterWidgetModule,
            i2$1.UserManageProfileWidgetModule,
            i2$2.DeviceInformationWidgetModule] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: SelfServiceFeaturesModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: __spreadArray(__spreadArray(__spreadArray([
                            i6.CommonModule,
                            i1$2.ReactiveFormsModule,
                            i1$1.RouterModule
                        ], __read(uiModules)), __read(widgetModules)), [
                            i4.IdentityCommonAngModule,
                            SelfServiceUiModule,
                        ]),
                        declarations: __spreadArray([], __read(components)),
                        exports: __spreadArray(__spreadArray([i1$1.RouterModule], __read(components)), __read(widgetModules)),
                    }]
            }] });

    var defaultRoute = {
        path: '',
        component: IdentitySelfServiceJourneyComponent,
        children: [
            { path: '', redirectTo: 'profile', pathMatch: 'full' },
            {
                path: 'profile',
                data: {
                    title: $localize(templateObject_1 || (templateObject_1 = __makeTemplateObject([":Tab label for managing user profile@@bb-identity-self-service-journey.tab-user-manage-profile:Profile"], [":Tab label for managing user profile@@bb-identity-self-service-journey.tab-user-manage-profile:Profile"]))),
                },
                component: UserManageProfileViewComponent,
            },
            {
                path: 'user-localization',
                data: {
                    title: $localize(templateObject_2 || (templateObject_2 = __makeTemplateObject([":Tab label for changing user preferences@@bb-identity-self-service-journey.tab-localization:Localization"], [":Tab label for changing user preferences@@bb-identity-self-service-journey.tab-localization:Localization"]))),
                },
                component: UserLocalizationComponent,
                children: [
                    { path: '', redirectTo: 'communications', pathMatch: 'full' },
                    {
                        path: 'communications',
                        component: UserLocalizationCommunicationsViewComponent,
                    },
                    {
                        path: 'change-language',
                        component: UserLocalizationChangeLanguageViewComponent,
                    },
                ],
            },
            {
                path: 'login-security',
                data: {
                    title: $localize(templateObject_3 || (templateObject_3 = __makeTemplateObject([":Tab label for login and security settings@@bb-identity-self-service-journey.tab-login-security:Login & security"], [":Tab label for login and security settings@@bb-identity-self-service-journey.tab-login-security:Login & security"]))),
                },
                component: UserIdentitySecurityCenterViewComponent,
            },
            {
                path: 'devices',
                data: { title: $localize(templateObject_4 || (templateObject_4 = __makeTemplateObject([":Tab label for managing devices@@bb-identity-self-service-journey.tab-devices:Devices"], [":Tab label for managing devices@@bb-identity-self-service-journey.tab-devices:Devices"]))) },
                component: DeviceInformationViewComponent,
            },
        ],
    };
    var IdentitySelfServiceJourneyModule = /** @class */ (function () {
        function IdentitySelfServiceJourneyModule() {
        }
        IdentitySelfServiceJourneyModule.forRoot = function (
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        data) {
            if (data === void 0) { data = { route: defaultRoute }; }
            return {
                ngModule: IdentitySelfServiceJourneyModule,
                providers: [i1$1.provideRoutes([data.route])],
            };
        };
        return IdentitySelfServiceJourneyModule;
    }());
    /** @nocollapse */ IdentitySelfServiceJourneyModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: IdentitySelfServiceJourneyModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    /** @nocollapse */ IdentitySelfServiceJourneyModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: IdentitySelfServiceJourneyModule, declarations: [IdentitySelfServiceJourneyComponent], imports: [i6.CommonModule,
            i1$1.RouterModule, i1__namespace$4.BackbaseCoreModule, i4.IdentityCommonAngModule,
            SelfServiceFeaturesModule], exports: [i1$1.RouterModule, IdentitySelfServiceJourneyComponent] });
    /** @nocollapse */ IdentitySelfServiceJourneyModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: IdentitySelfServiceJourneyModule, providers: [IdentitySelfServiceJourneyConfigurationService], imports: [[
                i6.CommonModule,
                i1$1.RouterModule,
                i1$4.BackbaseCoreModule.forRoot({
                    classMap: {
                        IdentitySelfServiceJourneyComponent: IdentitySelfServiceJourneyComponent,
                    },
                }),
                i4.IdentityCommonAngModule,
                SelfServiceFeaturesModule,
            ], i1$1.RouterModule] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: IdentitySelfServiceJourneyModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [
                            i6.CommonModule,
                            i1$1.RouterModule,
                            i1$4.BackbaseCoreModule.forRoot({
                                classMap: {
                                    IdentitySelfServiceJourneyComponent: IdentitySelfServiceJourneyComponent,
                                },
                            }),
                            i4.IdentityCommonAngModule,
                            SelfServiceFeaturesModule,
                        ],
                        declarations: [IdentitySelfServiceJourneyComponent],
                        providers: [IdentitySelfServiceJourneyConfigurationService],
                        exports: [i1$1.RouterModule, IdentitySelfServiceJourneyComponent],
                    }]
            }] });
    var templateObject_1, templateObject_2, templateObject_3, templateObject_4;

    /**
     * Shell
     */

    /**
     * Generated bundle index. Do not edit.
     */

    exports.BackButtonComponent = BackButtonComponent;
    exports.DeviceInformationViewComponent = DeviceInformationViewComponent;
    exports.DiscardModalComponent = DiscardModalComponent;
    exports.IdentitySelfServiceJourneyComponent = IdentitySelfServiceJourneyComponent;
    exports.IdentitySelfServiceJourneyConfigurationService = IdentitySelfServiceJourneyConfigurationService;
    exports.IdentitySelfServiceJourneyConfigurationToken = IdentitySelfServiceJourneyConfigurationToken;
    exports.IdentitySelfServiceJourneyModule = IdentitySelfServiceJourneyModule;
    exports.SelfServiceFeaturesModule = SelfServiceFeaturesModule;
    exports.SelfServiceUiModule = SelfServiceUiModule;
    exports.UserIdentitySecurityCenterViewComponent = UserIdentitySecurityCenterViewComponent;
    exports.UserLocalizationChangeLanguageFormComponent = UserLocalizationChangeLanguageFormComponent;
    exports.UserLocalizationChangeLanguageViewComponent = UserLocalizationChangeLanguageViewComponent;
    exports.UserLocalizationCommunicationsViewComponent = UserLocalizationCommunicationsViewComponent;
    exports.UserLocalizationComponent = UserLocalizationComponent;
    exports.UserLocalizationService = UserLocalizationService;
    exports.UserManageProfileViewComponent = UserManageProfileViewComponent;
    exports.cacheItem = cacheItem;
    exports.defaultRoute = defaultRoute;
    exports.findLanguageRegionOption = findLanguageRegionOption;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
