// @backbase/user-manage-profile-widget-ang/esm2015/src/user-profile-forms/user-profile-form-inputs/base-inputs/user-profile-form-base-inputs.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { InputCheckboxModule } from '@backbase/ui-ang/input-checkbox';
import { UserPrimaryCheckboxInputComponent, UserPrimaryCheckboxInputCustomizableDirective, } from './user-primary-checkbox-input.component';
import * as i0 from "@angular/core";
const uiModules = [InputCheckboxModule];
export class UserProfileFormBaseInputsModule {
}
UserProfileFormBaseInputsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormBaseInputsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
UserProfileFormBaseInputsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormBaseInputsModule, declarations: [UserPrimaryCheckboxInputComponent, UserPrimaryCheckboxInputCustomizableDirective], imports: [CommonModule, ReactiveFormsModule, InputCheckboxModule], exports: [UserPrimaryCheckboxInputComponent, UserPrimaryCheckboxInputCustomizableDirective, InputCheckboxModule] });
UserProfileFormBaseInputsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormBaseInputsModule, imports: [[CommonModule, ReactiveFormsModule, ...uiModules], InputCheckboxModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormBaseInputsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ReactiveFormsModule, ...uiModules],
                    declarations: [UserPrimaryCheckboxInputComponent, UserPrimaryCheckboxInputCustomizableDirective],
                    exports: [UserPrimaryCheckboxInputComponent, UserPrimaryCheckboxInputCustomizableDirective, ...uiModules],
                }]
        }] });
