// @backbase/ui-ang/esm2015/_example/rich-text-editor/showcase/rich-text-editor-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RichTextEditorShowcaseComponent } from './rich-text-editor-showcase.component';
import { RichTextEditorModule } from '@backbase/ui-ang/rich-text-editor';
import { ReactiveFormsModule } from '@angular/forms';
import * as i0 from "@angular/core";
export class RichTextEditorShowcaseModule {
}
RichTextEditorShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RichTextEditorShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
RichTextEditorShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RichTextEditorShowcaseModule, declarations: [RichTextEditorShowcaseComponent], imports: [CommonModule, RichTextEditorModule, ReactiveFormsModule] });
RichTextEditorShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RichTextEditorShowcaseModule, imports: [[CommonModule, RichTextEditorModule, ReactiveFormsModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RichTextEditorShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, RichTextEditorModule, ReactiveFormsModule],
                    declarations: [RichTextEditorShowcaseComponent],
                    entryComponents: [RichTextEditorShowcaseComponent],
                }]
        }] });
