// @backbase/payment-orders-ang/esm2015/lib/ui/services/navigation.service.js
import { Injectable } from '@angular/core';
import { RouterService } from '@backbase/foundation-ang/core';
import { Subject } from 'rxjs';
import { Routes } from '../models';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
export class NavigationService {
    constructor(routerService) {
        this.routerService = routerService;
        this.emitAfterSuccess = new Subject();
        this.afterSuccessEmitted$ = this.emitAfterSuccess.asObservable();
    }
    form() {
        return this.routerService.navigate([Routes.FORM], {
            queryParamsHandling: 'merge',
        });
    }
    review() {
        return this.routerService.navigate([Routes.REVIEW], {
            queryParamsHandling: 'merge',
        });
    }
    afterSuccess() {
        this.emitAfterSuccess.next();
    }
}
/** @nocollapse */ NavigationService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NavigationService, deps: [{ token: i1.RouterService }], target: i0.ɵɵFactoryTarget.Injectable });
/** @nocollapse */ NavigationService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NavigationService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NavigationService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.RouterService }]; } });
