// @backbase/payment-orders-ang/esm2015/lib/ui/services/state.service.js
import { Injectable } from '@angular/core';
import { FormGroup } from '@angular/forms';
import { BehaviorSubject } from 'rxjs';
import { PaymentState } from '../models';
import * as i0 from "@angular/core";
export class StateService {
    constructor() {
        this.formSubject = new BehaviorSubject(new FormGroup({}));
        this.paymentStateSubject = new BehaviorSubject(PaymentState.INITIAL);
        this.canApprovePayment = false;
        this.paymentTemplateSubject = new BehaviorSubject(undefined);
        this.paymentEditSubject = new BehaviorSubject(undefined);
    }
    get form$() {
        return this.formSubject.asObservable();
    }
    get formInstance() {
        return this.formSubject.value;
    }
    get paymentState$() {
        return this.paymentStateSubject.asObservable();
    }
    get currentState() {
        return this.paymentStateSubject.value;
    }
    get canApprove() {
        return this.canApprovePayment;
    }
    set canApprove(canApprovePayment) {
        this.canApprovePayment = canApprovePayment;
    }
    get paymentTemplateState$() {
        return this.paymentTemplateSubject.asObservable();
    }
    get paymentEditState$() {
        return this.paymentEditSubject.asObservable();
    }
    get initialEditFormValue() {
        return this.initialEditFormValueItem;
    }
    set initialEditFormValue(formValue) {
        this.initialEditFormValueItem = formValue;
    }
    saveForm(form) {
        this.formSubject.next(form);
    }
    resetForm() {
        this.formSubject.next(new FormGroup({}));
    }
    setPaymentState(paymentState) {
        this.paymentStateSubject.next(paymentState);
    }
    setPaymentTemplate(template) {
        this.paymentTemplateSubject.next(template);
    }
    setEditPayment(payment) {
        this.paymentEditSubject.next(payment);
    }
}
/** @nocollapse */ StateService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StateService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
/** @nocollapse */ StateService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StateService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StateService, decorators: [{
            type: Injectable
        }] });
