// @backbase/cash-management-http-ang/esm2015/api/sweepRules.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const SweepRulesHttpServiceGetSweepRuleBySweepIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/sweep-rules/{sweepId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": 135864,
                    "name": "Investment IS 605",
                    "description": "Two-way daily investment sweep that never expires.",
                    "sweptAccountBban": "6129172765GE6UQ2U8TYD56",
                    "targetAccountBban": "680BCNMXVMFAY60",
                    "lowerBoundAmount": 10000,
                    "upperBoundAmount": 12500,
                    "status": "active",
                    "ruleType": "two-way",
                    "schedule": {
                        "frequency": "EvryDay(1)",
                        "dayOptions": {
                            "startingAtTime": 540,
                            "intradayRepeat": {
                                "everyMinutes": 60,
                                "endingAtTime": 1260
                            }
                        }
                    },
                    "startDate": "2019-04-10T15:52:27.218Z",
                    "createdAt": "2019-04-09T15:52:27.218Z",
                    "createdBy": "Denys Haidash"
                }
            },
            {
                status: 200,
                body: {
                    "lowerBoundAmount": 6.027456183070403,
                    "additions": {
                        "key": "additions"
                    },
                    "targetAccountBban": "targetAccountBban",
                    "description": "description",
                    "sweptAccountBban": "sweptAccountBban",
                    "expiryDate": "2000-01-23T04:56:07.000+00:00",
                    "schedule": {
                        "dayOptions": {
                            "intradayRepeat": {
                                "endingAtTime": 2.3021358869347655,
                                "everyMinutes": 5.637376656633329
                            },
                            "startingAtTime": 5.962133916683182
                        },
                        "monthOptions": {
                            "startingAtTime": 3.616076749251911,
                            "monthRepeat": "day-of-the-month"
                        },
                        "frequency": "frequency",
                        "weekOptions": {
                            "weekRepeatDay": 9.301444243932576,
                            "startingAtTime": 7.061401241503109
                        }
                    },
                    "createdAt": "2000-01-23T04:56:07.000+00:00",
                    "createdBy": "createdBy",
                    "upperBoundAmount": 1.4658129805029452,
                    "name": "name",
                    "id": 0.8008281904610115,
                    "startDate": "2000-01-23T04:56:07.000+00:00",
                    "status": "status"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 409,
                body: {
                    "message": "Conflict error",
                    "errors": [{
                            "message": "The request could not be completed due to a conflict with the current state of the resource.",
                            "key": "transactions.api.billingstatus.transaction",
                            "context": {
                                "billingStatus": "BILLED"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const SweepRulesHttpServiceGetSweeprulesMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/sweep-rules",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "items": [{
                            "id": 135864,
                            "name": "Investment IS 605",
                            "description": "Two-way daily investment sweep that never expires.",
                            "sweptAccountBban": "6129172765GE6UQ2U8TYD56",
                            "targetAccountBban": "680BCNMXVMFAY60",
                            "lowerBoundAmount": 10000,
                            "upperBoundAmount": 12500,
                            "status": "entered",
                            "ruleType": "two-way",
                            "schedule": {
                                "frequency": "EvryDay(1)",
                                "dayOptions": {
                                    "startingAtTime": 540,
                                    "intradayRepeat": {
                                        "everyMinutes": 60,
                                        "endingAtTime": 1260
                                    }
                                }
                            },
                            "startDate": "2019-04-10T15:52:27.218Z",
                            "createdAt": "2019-04-09T15:52:27.218Z",
                            "createdBy": "Anna Hrechana"
                        }, {
                            "id": 246975,
                            "name": "Current CS 16",
                            "description": "Two-way monthly sweep between current accounts that expires today.",
                            "sweptAccountBban": "TRIO0212471066",
                            "targetAccountBban": "FR596129172765GE6UQ2U8TYD56",
                            "lowerBoundAmount": 500,
                            "upperBoundAmount": 1750,
                            "status": "active",
                            "ruleType": "two-way",
                            "schedule": {
                                "frequency": "EvryMonth(1)",
                                "monthOptions": {
                                    "startingAtTime": 720,
                                    "monthRepeat": "week-day-of-the-month"
                                }
                            },
                            "startDate": "2019-04-10T16:24:37.463Z",
                            "createdAt": "2019-04-09T16:24:37.463Z",
                            "createdBy": "Serhii Koshman"
                        }, {
                            "id": 137731,
                            "name": "Loan LS 31",
                            "description": "Loan sweep that takes place each 4 weeks on Sundays until manually stopped.",
                            "sweptAccountBban": "FR596129172765GE6UQ2U8TYD56",
                            "targetAccountBban": "6129172765GE6UQ2U8TYD84",
                            "upperBoundAmount": 5000,
                            "status": "paused",
                            "ruleType": "one-way",
                            "schedule": {
                                "frequency": "EvryWeek(4)",
                                "weekOptions": {
                                    "startingAtTime": 0,
                                    "weekRepeatDay": 0
                                }
                            },
                            "startDate": "2019-01-01T16:24:37.463Z",
                            "createdAt": "2019-01-01T16:23:37.463Z",
                            "createdBy": "Marian Kitsenko"
                        }, {
                            "id": 225547,
                            "name": "Savings SS 24",
                            "description": "One-time one-way sweep to accumulate savings performed on a daily basis.",
                            "sweptAccountBban": "FR596129172765GE6UQ2U8TYD56",
                            "targetAccountBban": "TRIO0212471066",
                            "upperBoundAmount": 1100,
                            "status": "rejected",
                            "ruleType": "one-way",
                            "schedule": {
                                "frequency": "EvryDay(1)",
                                "dayOptions": {
                                    "startingAtTime": 30,
                                    "intradayRepeat": {
                                        "everyMinutes": 60,
                                        "endingAtTime": 30
                                    }
                                }
                            },
                            "startDate": "2019-04-11T08:10:18.591Z",
                            "expiryDate": "2019-04-11T20:59:59.999Z",
                            "createdAt": "2019-04-10T08:10:18.591Z",
                            "createdBy": "Dmytro Topchii"
                        }, {
                            "id": 110580,
                            "name": "Debit SD 350",
                            "description": "Maintenance sweep to handle debit card account set to repeat once a week starting from next month.",
                            "sweptAccountBban": "FR596129172765GE6UQ2U8TYD56",
                            "targetAccountBban": "6129172765GE6UQ2U8TYD84",
                            "lowerBoundAmount": 1000,
                            "status": "cancelled",
                            "ruleType": "one-way",
                            "schedule": {
                                "frequency": "EvryDay(7)",
                                "dayOptions": {
                                    "startingAtTime": 540
                                }
                            },
                            "startDate": "2019-05-10T21:00:00.000Z",
                            "expiryDate": "2019-05-11T20:59:59.999Z",
                            "createdAt": "2019-05-09T21:00:00.000Z",
                            "createdBy": "Serhii Sobur"
                        }, {
                            "id": 190783,
                            "name": "Debit 500",
                            "description": "Maintenance sweep to handle debit card account set to repeat every 5 days starting from next month.",
                            "sweptAccountBban": "FR596129172765GE6UQ2U8TYD56",
                            "targetAccountBban": "6129172765GE6UQ2U8TYD84",
                            "lowerBoundAmount": 500,
                            "status": "processing",
                            "ruleType": "one-way",
                            "schedule": {
                                "frequency": "EvryDay(5)",
                                "dayOptions": {
                                    "startingAtTime": 540
                                }
                            },
                            "startDate": "2019-05-10T21:00:00.000Z",
                            "expiryDate": "2019-05-11T20:59:59.999Z",
                            "createdAt": "2019-05-09T21:00:00.000Z",
                            "createdBy": "Yevhen Lubianov"
                        }, {
                            "id": 256076,
                            "name": "Outdated CS 3",
                            "description": "Two-way monthly sweep between current accounts that have been expired.",
                            "sweptAccountBban": "TRIO0212471066",
                            "targetAccountBban": "FR596129172765GE6UQ2U8TYD56",
                            "lowerBoundAmount": 500,
                            "upperBoundAmount": 1750,
                            "status": "expired",
                            "ruleType": "two-way",
                            "schedule": {
                                "frequency": "EvryMonth(1)",
                                "monthOptions": {
                                    "startingAtTime": 720,
                                    "monthRepeat": "week-day-of-the-month"
                                }
                            },
                            "startDate": "2019-03-10T16:24:37.463Z",
                            "expiryDate": "2019-04-10T16:24:37.463Z",
                            "createdAt": "2019-03-09T16:24:37.463Z",
                            "createdBy": "Yevhenii Holub"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 409,
                body: {
                    "message": "Conflict error",
                    "errors": [{
                            "message": "The request could not be completed due to a conflict with the current state of the resource.",
                            "key": "transactions.api.billingstatus.transaction",
                            "context": {
                                "billingStatus": "BILLED"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const SweepRulesHttpServicePostCreateSweepRuleMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/sweep-rules",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "id": 16250
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 409,
                body: {
                    "message": "Conflict error",
                    "errors": [{
                            "message": "The request could not be completed due to a conflict with the current state of the resource.",
                            "key": "transactions.api.billingstatus.transaction",
                            "context": {
                                "billingStatus": "BILLED"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const SweepRulesHttpServicePostSweepRuleActivationBySweepIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/sweep-rules/{sweepId}/activate",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "id": 135864,
                    "name": "Investment IS 605",
                    "description": "Two-way daily investment sweep that never expires.",
                    "sweptAccountBban": "6129172765GE6UQ2U8TYD56",
                    "targetAccountBban": "680BCNMXVMFAY60",
                    "lowerBoundAmount": 10000,
                    "upperBoundAmount": 12500,
                    "status": "active",
                    "ruleType": "two-way",
                    "schedule": {
                        "frequency": "EvryDay(1)",
                        "dayOptions": {
                            "startingAtTime": 540,
                            "intradayRepeat": {
                                "everyMinutes": 60,
                                "endingAtTime": 1260
                            }
                        }
                    },
                    "startDate": "2019-04-10T15:52:27.218Z",
                    "createdAt": "2019-04-09T15:52:27.218Z",
                    "createdBy": "Denys Haidash"
                }
            },
            {
                status: 200,
                body: {
                    "lowerBoundAmount": 6.027456183070403,
                    "additions": {
                        "key": "additions"
                    },
                    "targetAccountBban": "targetAccountBban",
                    "description": "description",
                    "sweptAccountBban": "sweptAccountBban",
                    "expiryDate": "2000-01-23T04:56:07.000+00:00",
                    "schedule": {
                        "dayOptions": {
                            "intradayRepeat": {
                                "endingAtTime": 2.3021358869347655,
                                "everyMinutes": 5.637376656633329
                            },
                            "startingAtTime": 5.962133916683182
                        },
                        "monthOptions": {
                            "startingAtTime": 3.616076749251911,
                            "monthRepeat": "day-of-the-month"
                        },
                        "frequency": "frequency",
                        "weekOptions": {
                            "weekRepeatDay": 9.301444243932576,
                            "startingAtTime": 7.061401241503109
                        }
                    },
                    "createdAt": "2000-01-23T04:56:07.000+00:00",
                    "createdBy": "createdBy",
                    "upperBoundAmount": 1.4658129805029452,
                    "name": "name",
                    "id": 0.8008281904610115,
                    "startDate": "2000-01-23T04:56:07.000+00:00",
                    "status": "status"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 409,
                body: {
                    "message": "Conflict error",
                    "errors": [{
                            "message": "The request could not be completed due to a conflict with the current state of the resource.",
                            "key": "transactions.api.billingstatus.transaction",
                            "context": {
                                "billingStatus": "BILLED"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const SweepRulesHttpServicePostSweepRuleCancelingBySweepIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/sweep-rules/{sweepId}/cancel",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "id": 135864,
                    "name": "Investment IS 605",
                    "description": "Two-way daily investment sweep that never expires.",
                    "sweptAccountBban": "6129172765GE6UQ2U8TYD56",
                    "targetAccountBban": "680BCNMXVMFAY60",
                    "lowerBoundAmount": 10000,
                    "upperBoundAmount": 12500,
                    "status": "cancel",
                    "ruleType": "two-way",
                    "schedule": {
                        "frequency": "EvryDay(1)",
                        "dayOptions": {
                            "startingAtTime": 540,
                            "intradayRepeat": {
                                "everyMinutes": 60,
                                "endingAtTime": 1260
                            }
                        }
                    },
                    "startDate": "2019-04-10T15:52:27.218Z",
                    "createdAt": "2019-04-09T15:52:27.218Z",
                    "createdBy": "Denys Haidash"
                }
            },
            {
                status: 200,
                body: {
                    "lowerBoundAmount": 6.027456183070403,
                    "additions": {
                        "key": "additions"
                    },
                    "targetAccountBban": "targetAccountBban",
                    "description": "description",
                    "sweptAccountBban": "sweptAccountBban",
                    "expiryDate": "2000-01-23T04:56:07.000+00:00",
                    "schedule": {
                        "dayOptions": {
                            "intradayRepeat": {
                                "endingAtTime": 2.3021358869347655,
                                "everyMinutes": 5.637376656633329
                            },
                            "startingAtTime": 5.962133916683182
                        },
                        "monthOptions": {
                            "startingAtTime": 3.616076749251911,
                            "monthRepeat": "day-of-the-month"
                        },
                        "frequency": "frequency",
                        "weekOptions": {
                            "weekRepeatDay": 9.301444243932576,
                            "startingAtTime": 7.061401241503109
                        }
                    },
                    "createdAt": "2000-01-23T04:56:07.000+00:00",
                    "createdBy": "createdBy",
                    "upperBoundAmount": 1.4658129805029452,
                    "name": "name",
                    "id": 0.8008281904610115,
                    "startDate": "2000-01-23T04:56:07.000+00:00",
                    "status": "status"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 409,
                body: {
                    "message": "Conflict error",
                    "errors": [{
                            "message": "The request could not be completed due to a conflict with the current state of the resource.",
                            "key": "transactions.api.billingstatus.transaction",
                            "context": {
                                "billingStatus": "BILLED"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const SweepRulesHttpServicePostSweepRulePausingBySweepIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/sweep-rules/{sweepId}/pause",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "id": 135864,
                    "name": "Investment IS 605",
                    "description": "Two-way daily investment sweep that never expires.",
                    "sweptAccountBban": "6129172765GE6UQ2U8TYD56",
                    "targetAccountBban": "680BCNMXVMFAY60",
                    "lowerBoundAmount": 10000,
                    "upperBoundAmount": 12500,
                    "status": "pause",
                    "ruleType": "two-way",
                    "schedule": {
                        "frequency": "EvryDay(1)",
                        "dayOptions": {
                            "startingAtTime": 540,
                            "intradayRepeat": {
                                "everyMinutes": 60,
                                "endingAtTime": 1260
                            }
                        }
                    },
                    "startDate": "2019-04-10T15:52:27.218Z",
                    "createdAt": "2019-04-09T15:52:27.218Z",
                    "createdBy": "Denys Haidash"
                }
            },
            {
                status: 200,
                body: {
                    "lowerBoundAmount": 6.027456183070403,
                    "additions": {
                        "key": "additions"
                    },
                    "targetAccountBban": "targetAccountBban",
                    "description": "description",
                    "sweptAccountBban": "sweptAccountBban",
                    "expiryDate": "2000-01-23T04:56:07.000+00:00",
                    "schedule": {
                        "dayOptions": {
                            "intradayRepeat": {
                                "endingAtTime": 2.3021358869347655,
                                "everyMinutes": 5.637376656633329
                            },
                            "startingAtTime": 5.962133916683182
                        },
                        "monthOptions": {
                            "startingAtTime": 3.616076749251911,
                            "monthRepeat": "day-of-the-month"
                        },
                        "frequency": "frequency",
                        "weekOptions": {
                            "weekRepeatDay": 9.301444243932576,
                            "startingAtTime": 7.061401241503109
                        }
                    },
                    "createdAt": "2000-01-23T04:56:07.000+00:00",
                    "createdBy": "createdBy",
                    "upperBoundAmount": 1.4658129805029452,
                    "name": "name",
                    "id": 0.8008281904610115,
                    "startDate": "2000-01-23T04:56:07.000+00:00",
                    "status": "status"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 409,
                body: {
                    "message": "Conflict error",
                    "errors": [{
                            "message": "The request could not be completed due to a conflict with the current state of the resource.",
                            "key": "transactions.api.billingstatus.transaction",
                            "context": {
                                "billingStatus": "BILLED"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const SweepRulesHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v2/sweep-rules/{sweepId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": 135864,
                    "name": "Investment IS 605",
                    "description": "Two-way daily investment sweep that never expires.",
                    "sweptAccountBban": "6129172765GE6UQ2U8TYD56",
                    "targetAccountBban": "680BCNMXVMFAY60",
                    "lowerBoundAmount": 10000,
                    "upperBoundAmount": 12500,
                    "status": "active",
                    "ruleType": "two-way",
                    "schedule": {
                        "frequency": "EvryDay(1)",
                        "dayOptions": {
                            "startingAtTime": 540,
                            "intradayRepeat": {
                                "everyMinutes": 60,
                                "endingAtTime": 1260
                            }
                        }
                    },
                    "startDate": "2019-04-10T15:52:27.218Z",
                    "createdAt": "2019-04-09T15:52:27.218Z",
                    "createdBy": "Denys Haidash"
                }
            },
            {
                status: 200,
                body: {
                    "lowerBoundAmount": 6.027456183070403,
                    "additions": {
                        "key": "additions"
                    },
                    "targetAccountBban": "targetAccountBban",
                    "description": "description",
                    "sweptAccountBban": "sweptAccountBban",
                    "expiryDate": "2000-01-23T04:56:07.000+00:00",
                    "schedule": {
                        "dayOptions": {
                            "intradayRepeat": {
                                "endingAtTime": 2.3021358869347655,
                                "everyMinutes": 5.637376656633329
                            },
                            "startingAtTime": 5.962133916683182
                        },
                        "monthOptions": {
                            "startingAtTime": 3.616076749251911,
                            "monthRepeat": "day-of-the-month"
                        },
                        "frequency": "frequency",
                        "weekOptions": {
                            "weekRepeatDay": 9.301444243932576,
                            "startingAtTime": 7.061401241503109
                        }
                    },
                    "createdAt": "2000-01-23T04:56:07.000+00:00",
                    "createdBy": "createdBy",
                    "upperBoundAmount": 1.4658129805029452,
                    "name": "name",
                    "id": 0.8008281904610115,
                    "startDate": "2000-01-23T04:56:07.000+00:00",
                    "status": "status"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 409,
                body: {
                    "message": "Conflict error",
                    "errors": [{
                            "message": "The request could not be completed due to a conflict with the current state of the resource.",
                            "key": "transactions.api.billingstatus.transaction",
                            "context": {
                                "billingStatus": "BILLED"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/sweep-rules",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "items": [{
                            "id": 135864,
                            "name": "Investment IS 605",
                            "description": "Two-way daily investment sweep that never expires.",
                            "sweptAccountBban": "6129172765GE6UQ2U8TYD56",
                            "targetAccountBban": "680BCNMXVMFAY60",
                            "lowerBoundAmount": 10000,
                            "upperBoundAmount": 12500,
                            "status": "entered",
                            "ruleType": "two-way",
                            "schedule": {
                                "frequency": "EvryDay(1)",
                                "dayOptions": {
                                    "startingAtTime": 540,
                                    "intradayRepeat": {
                                        "everyMinutes": 60,
                                        "endingAtTime": 1260
                                    }
                                }
                            },
                            "startDate": "2019-04-10T15:52:27.218Z",
                            "createdAt": "2019-04-09T15:52:27.218Z",
                            "createdBy": "Anna Hrechana"
                        }, {
                            "id": 246975,
                            "name": "Current CS 16",
                            "description": "Two-way monthly sweep between current accounts that expires today.",
                            "sweptAccountBban": "TRIO0212471066",
                            "targetAccountBban": "FR596129172765GE6UQ2U8TYD56",
                            "lowerBoundAmount": 500,
                            "upperBoundAmount": 1750,
                            "status": "active",
                            "ruleType": "two-way",
                            "schedule": {
                                "frequency": "EvryMonth(1)",
                                "monthOptions": {
                                    "startingAtTime": 720,
                                    "monthRepeat": "week-day-of-the-month"
                                }
                            },
                            "startDate": "2019-04-10T16:24:37.463Z",
                            "createdAt": "2019-04-09T16:24:37.463Z",
                            "createdBy": "Serhii Koshman"
                        }, {
                            "id": 137731,
                            "name": "Loan LS 31",
                            "description": "Loan sweep that takes place each 4 weeks on Sundays until manually stopped.",
                            "sweptAccountBban": "FR596129172765GE6UQ2U8TYD56",
                            "targetAccountBban": "6129172765GE6UQ2U8TYD84",
                            "upperBoundAmount": 5000,
                            "status": "paused",
                            "ruleType": "one-way",
                            "schedule": {
                                "frequency": "EvryWeek(4)",
                                "weekOptions": {
                                    "startingAtTime": 0,
                                    "weekRepeatDay": 0
                                }
                            },
                            "startDate": "2019-01-01T16:24:37.463Z",
                            "createdAt": "2019-01-01T16:23:37.463Z",
                            "createdBy": "Marian Kitsenko"
                        }, {
                            "id": 225547,
                            "name": "Savings SS 24",
                            "description": "One-time one-way sweep to accumulate savings performed on a daily basis.",
                            "sweptAccountBban": "FR596129172765GE6UQ2U8TYD56",
                            "targetAccountBban": "TRIO0212471066",
                            "upperBoundAmount": 1100,
                            "status": "rejected",
                            "ruleType": "one-way",
                            "schedule": {
                                "frequency": "EvryDay(1)",
                                "dayOptions": {
                                    "startingAtTime": 30,
                                    "intradayRepeat": {
                                        "everyMinutes": 60,
                                        "endingAtTime": 30
                                    }
                                }
                            },
                            "startDate": "2019-04-11T08:10:18.591Z",
                            "expiryDate": "2019-04-11T20:59:59.999Z",
                            "createdAt": "2019-04-10T08:10:18.591Z",
                            "createdBy": "Dmytro Topchii"
                        }, {
                            "id": 110580,
                            "name": "Debit SD 350",
                            "description": "Maintenance sweep to handle debit card account set to repeat once a week starting from next month.",
                            "sweptAccountBban": "FR596129172765GE6UQ2U8TYD56",
                            "targetAccountBban": "6129172765GE6UQ2U8TYD84",
                            "lowerBoundAmount": 1000,
                            "status": "cancelled",
                            "ruleType": "one-way",
                            "schedule": {
                                "frequency": "EvryDay(7)",
                                "dayOptions": {
                                    "startingAtTime": 540
                                }
                            },
                            "startDate": "2019-05-10T21:00:00.000Z",
                            "expiryDate": "2019-05-11T20:59:59.999Z",
                            "createdAt": "2019-05-09T21:00:00.000Z",
                            "createdBy": "Serhii Sobur"
                        }, {
                            "id": 190783,
                            "name": "Debit 500",
                            "description": "Maintenance sweep to handle debit card account set to repeat every 5 days starting from next month.",
                            "sweptAccountBban": "FR596129172765GE6UQ2U8TYD56",
                            "targetAccountBban": "6129172765GE6UQ2U8TYD84",
                            "lowerBoundAmount": 500,
                            "status": "processing",
                            "ruleType": "one-way",
                            "schedule": {
                                "frequency": "EvryDay(5)",
                                "dayOptions": {
                                    "startingAtTime": 540
                                }
                            },
                            "startDate": "2019-05-10T21:00:00.000Z",
                            "expiryDate": "2019-05-11T20:59:59.999Z",
                            "createdAt": "2019-05-09T21:00:00.000Z",
                            "createdBy": "Yevhen Lubianov"
                        }, {
                            "id": 256076,
                            "name": "Outdated CS 3",
                            "description": "Two-way monthly sweep between current accounts that have been expired.",
                            "sweptAccountBban": "TRIO0212471066",
                            "targetAccountBban": "FR596129172765GE6UQ2U8TYD56",
                            "lowerBoundAmount": 500,
                            "upperBoundAmount": 1750,
                            "status": "expired",
                            "ruleType": "two-way",
                            "schedule": {
                                "frequency": "EvryMonth(1)",
                                "monthOptions": {
                                    "startingAtTime": 720,
                                    "monthRepeat": "week-day-of-the-month"
                                }
                            },
                            "startDate": "2019-03-10T16:24:37.463Z",
                            "expiryDate": "2019-04-10T16:24:37.463Z",
                            "createdAt": "2019-03-09T16:24:37.463Z",
                            "createdBy": "Yevhenii Holub"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 409,
                body: {
                    "message": "Conflict error",
                    "errors": [{
                            "message": "The request could not be completed due to a conflict with the current state of the resource.",
                            "key": "transactions.api.billingstatus.transaction",
                            "context": {
                                "billingStatus": "BILLED"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/sweep-rules",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "id": 16250
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 409,
                body: {
                    "message": "Conflict error",
                    "errors": [{
                            "message": "The request could not be completed due to a conflict with the current state of the resource.",
                            "key": "transactions.api.billingstatus.transaction",
                            "context": {
                                "billingStatus": "BILLED"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/sweep-rules/{sweepId}/activate",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "id": 135864,
                    "name": "Investment IS 605",
                    "description": "Two-way daily investment sweep that never expires.",
                    "sweptAccountBban": "6129172765GE6UQ2U8TYD56",
                    "targetAccountBban": "680BCNMXVMFAY60",
                    "lowerBoundAmount": 10000,
                    "upperBoundAmount": 12500,
                    "status": "active",
                    "ruleType": "two-way",
                    "schedule": {
                        "frequency": "EvryDay(1)",
                        "dayOptions": {
                            "startingAtTime": 540,
                            "intradayRepeat": {
                                "everyMinutes": 60,
                                "endingAtTime": 1260
                            }
                        }
                    },
                    "startDate": "2019-04-10T15:52:27.218Z",
                    "createdAt": "2019-04-09T15:52:27.218Z",
                    "createdBy": "Denys Haidash"
                }
            },
            {
                status: 200,
                body: {
                    "lowerBoundAmount": 6.027456183070403,
                    "additions": {
                        "key": "additions"
                    },
                    "targetAccountBban": "targetAccountBban",
                    "description": "description",
                    "sweptAccountBban": "sweptAccountBban",
                    "expiryDate": "2000-01-23T04:56:07.000+00:00",
                    "schedule": {
                        "dayOptions": {
                            "intradayRepeat": {
                                "endingAtTime": 2.3021358869347655,
                                "everyMinutes": 5.637376656633329
                            },
                            "startingAtTime": 5.962133916683182
                        },
                        "monthOptions": {
                            "startingAtTime": 3.616076749251911,
                            "monthRepeat": "day-of-the-month"
                        },
                        "frequency": "frequency",
                        "weekOptions": {
                            "weekRepeatDay": 9.301444243932576,
                            "startingAtTime": 7.061401241503109
                        }
                    },
                    "createdAt": "2000-01-23T04:56:07.000+00:00",
                    "createdBy": "createdBy",
                    "upperBoundAmount": 1.4658129805029452,
                    "name": "name",
                    "id": 0.8008281904610115,
                    "startDate": "2000-01-23T04:56:07.000+00:00",
                    "status": "status"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 409,
                body: {
                    "message": "Conflict error",
                    "errors": [{
                            "message": "The request could not be completed due to a conflict with the current state of the resource.",
                            "key": "transactions.api.billingstatus.transaction",
                            "context": {
                                "billingStatus": "BILLED"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/sweep-rules/{sweepId}/cancel",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "id": 135864,
                    "name": "Investment IS 605",
                    "description": "Two-way daily investment sweep that never expires.",
                    "sweptAccountBban": "6129172765GE6UQ2U8TYD56",
                    "targetAccountBban": "680BCNMXVMFAY60",
                    "lowerBoundAmount": 10000,
                    "upperBoundAmount": 12500,
                    "status": "cancel",
                    "ruleType": "two-way",
                    "schedule": {
                        "frequency": "EvryDay(1)",
                        "dayOptions": {
                            "startingAtTime": 540,
                            "intradayRepeat": {
                                "everyMinutes": 60,
                                "endingAtTime": 1260
                            }
                        }
                    },
                    "startDate": "2019-04-10T15:52:27.218Z",
                    "createdAt": "2019-04-09T15:52:27.218Z",
                    "createdBy": "Denys Haidash"
                }
            },
            {
                status: 200,
                body: {
                    "lowerBoundAmount": 6.027456183070403,
                    "additions": {
                        "key": "additions"
                    },
                    "targetAccountBban": "targetAccountBban",
                    "description": "description",
                    "sweptAccountBban": "sweptAccountBban",
                    "expiryDate": "2000-01-23T04:56:07.000+00:00",
                    "schedule": {
                        "dayOptions": {
                            "intradayRepeat": {
                                "endingAtTime": 2.3021358869347655,
                                "everyMinutes": 5.637376656633329
                            },
                            "startingAtTime": 5.962133916683182
                        },
                        "monthOptions": {
                            "startingAtTime": 3.616076749251911,
                            "monthRepeat": "day-of-the-month"
                        },
                        "frequency": "frequency",
                        "weekOptions": {
                            "weekRepeatDay": 9.301444243932576,
                            "startingAtTime": 7.061401241503109
                        }
                    },
                    "createdAt": "2000-01-23T04:56:07.000+00:00",
                    "createdBy": "createdBy",
                    "upperBoundAmount": 1.4658129805029452,
                    "name": "name",
                    "id": 0.8008281904610115,
                    "startDate": "2000-01-23T04:56:07.000+00:00",
                    "status": "status"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 409,
                body: {
                    "message": "Conflict error",
                    "errors": [{
                            "message": "The request could not be completed due to a conflict with the current state of the resource.",
                            "key": "transactions.api.billingstatus.transaction",
                            "context": {
                                "billingStatus": "BILLED"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/sweep-rules/{sweepId}/pause",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "id": 135864,
                    "name": "Investment IS 605",
                    "description": "Two-way daily investment sweep that never expires.",
                    "sweptAccountBban": "6129172765GE6UQ2U8TYD56",
                    "targetAccountBban": "680BCNMXVMFAY60",
                    "lowerBoundAmount": 10000,
                    "upperBoundAmount": 12500,
                    "status": "pause",
                    "ruleType": "two-way",
                    "schedule": {
                        "frequency": "EvryDay(1)",
                        "dayOptions": {
                            "startingAtTime": 540,
                            "intradayRepeat": {
                                "everyMinutes": 60,
                                "endingAtTime": 1260
                            }
                        }
                    },
                    "startDate": "2019-04-10T15:52:27.218Z",
                    "createdAt": "2019-04-09T15:52:27.218Z",
                    "createdBy": "Denys Haidash"
                }
            },
            {
                status: 200,
                body: {
                    "lowerBoundAmount": 6.027456183070403,
                    "additions": {
                        "key": "additions"
                    },
                    "targetAccountBban": "targetAccountBban",
                    "description": "description",
                    "sweptAccountBban": "sweptAccountBban",
                    "expiryDate": "2000-01-23T04:56:07.000+00:00",
                    "schedule": {
                        "dayOptions": {
                            "intradayRepeat": {
                                "endingAtTime": 2.3021358869347655,
                                "everyMinutes": 5.637376656633329
                            },
                            "startingAtTime": 5.962133916683182
                        },
                        "monthOptions": {
                            "startingAtTime": 3.616076749251911,
                            "monthRepeat": "day-of-the-month"
                        },
                        "frequency": "frequency",
                        "weekOptions": {
                            "weekRepeatDay": 9.301444243932576,
                            "startingAtTime": 7.061401241503109
                        }
                    },
                    "createdAt": "2000-01-23T04:56:07.000+00:00",
                    "createdBy": "createdBy",
                    "upperBoundAmount": 1.4658129805029452,
                    "name": "name",
                    "id": 0.8008281904610115,
                    "startDate": "2000-01-23T04:56:07.000+00:00",
                    "status": "status"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 409,
                body: {
                    "message": "Conflict error",
                    "errors": [{
                            "message": "The request could not be completed due to a conflict with the current state of the resource.",
                            "key": "transactions.api.billingstatus.transaction",
                            "context": {
                                "billingStatus": "BILLED"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
]);
