// @backbase/notifications-popups-widget-ang/esm2015/lib/notifications-popups-widget.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { NotificationsPopupsWidgetComponent } from './components/notifications-popups-widget/notifications-popups-widget.component';
import { NotificationsStickyListComponent, NotificationsStickyListCustomizableDirective, } from './components/notifications-sticky-list/notifications-sticky-list.component';
import { NotificationsStickyListItemComponent, NotificationsStickyListItemCustomizableDirective, } from './components/notifications-sticky-list-item/notifications-sticky-list-item.component';
import { NotificationsCommonModule } from '@backbase/notifications-common-ang';
import { NotificationsPopupsListComponent, NotificationsPopupsListCustomizableDirective, } from './components/notifications-popups-list/notifications-popups-list.component';
import { NotificationsPopupsItemComponent, NotificationsPopupsItemHeaderCustomizableDirective, NotificationsPopupsItemMessageCustomizableDirective, } from './components/notifications-popups-item/notifications-popups-item.component';
import { AlertModule } from '@backbase/ui-ang/alert';
import { EllipsisModule } from '@backbase/ui-ang/ellipsis';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const components = [
    NotificationsPopupsWidgetComponent,
    NotificationsStickyListComponent,
    NotificationsStickyListItemComponent,
    NotificationsPopupsListComponent,
    NotificationsPopupsItemComponent,
];
const directives = [
    NotificationsStickyListCustomizableDirective,
    NotificationsStickyListItemCustomizableDirective,
    NotificationsPopupsItemHeaderCustomizableDirective,
    NotificationsPopupsItemMessageCustomizableDirective,
    NotificationsPopupsListCustomizableDirective,
];
const uiModules = [AlertModule, EllipsisModule];
export class NotificationsPopupsWidgetModule {
}
/** @nocollapse */ NotificationsPopupsWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsPopupsWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ NotificationsPopupsWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsPopupsWidgetModule, declarations: [NotificationsPopupsWidgetComponent,
        NotificationsStickyListComponent,
        NotificationsStickyListItemComponent,
        NotificationsPopupsListComponent,
        NotificationsPopupsItemComponent, NotificationsStickyListCustomizableDirective,
        NotificationsStickyListItemCustomizableDirective,
        NotificationsPopupsItemHeaderCustomizableDirective,
        NotificationsPopupsItemMessageCustomizableDirective,
        NotificationsPopupsListCustomizableDirective], imports: [CommonModule,
        NotificationsCommonModule, i1.BackbaseCoreModule, AlertModule, EllipsisModule], exports: [NotificationsCommonModule, NotificationsPopupsWidgetComponent,
        NotificationsStickyListComponent,
        NotificationsStickyListItemComponent,
        NotificationsPopupsListComponent,
        NotificationsPopupsItemComponent, NotificationsStickyListCustomizableDirective,
        NotificationsStickyListItemCustomizableDirective,
        NotificationsPopupsItemHeaderCustomizableDirective,
        NotificationsPopupsItemMessageCustomizableDirective,
        NotificationsPopupsListCustomizableDirective, AlertModule, EllipsisModule] });
/** @nocollapse */ NotificationsPopupsWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsPopupsWidgetModule, imports: [[
            CommonModule,
            NotificationsCommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { NotificationsPopupsWidgetComponent },
            }),
            ...uiModules,
        ], NotificationsCommonModule, AlertModule, EllipsisModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsPopupsWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components, ...directives],
                    imports: [
                        CommonModule,
                        NotificationsCommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { NotificationsPopupsWidgetComponent },
                        }),
                        ...uiModules,
                    ],
                    exports: [NotificationsCommonModule, ...components, ...directives, ...uiModules],
                }]
        }] });
