// @backbase/messages-common/esm2015/src/components/base-modal/base-modal.component.js
import { Component, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
export class BaseModalComponent {
    constructor() {
        /**
         * Input that toggles opening/closing the modal
         */
        this.isOpen = false;
        /**
         * Event emitted when modal was resolved
         */
        this.confirm = new EventEmitter();
        /**
         * Event emitted when modal was dismissed
         */
        this.cancel = new EventEmitter();
        /**
         * The unique ID HTML property for each component instance.
         */
        this.id = `base-modal-component-${BaseModalComponent.counter++}`;
        this.hostRef = this;
    }
    /**
     * The method to reset the form and the cancellation modal.
     */
    onCancel() {
        this.cancel.emit();
    }
    /**
     * The method to confirm the value change.
     *
     * @param value Changed value
     */
    onConfirm(value) {
        this.confirm.emit(value);
    }
}
BaseModalComponent.counter = 0;
BaseModalComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: BaseModalComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
BaseModalComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.14", type: BaseModalComponent, selector: "ng-component", inputs: { isOpen: "isOpen" }, outputs: { confirm: "confirm", cancel: "cancel" }, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: BaseModalComponent, decorators: [{
            type: Component,
            args: [{
                    template: '',
                }]
        }], propDecorators: { isOpen: [{
                type: Input
            }], confirm: [{
                type: Output
            }], cancel: [{
                type: Output
            }] } });
