// @backbase/quick-transfer-journey-ang/esm2015/lib/ui/validation-message/validation-message.component.js
import { Component, Input } from '@angular/core';
import { FormControl } from '@angular/forms';
import { fadeSlideAnimation } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/input-validation-message";
import * as i2 from "@angular/common";
export class QuickTransferValidationMessageComponent {
    ngOnInit() {
        this.requiredMessage = this.getValidationMessage('required');
    }
    getValidationMessage(error) {
        var _a;
        const validation = (_a = this.validationMessages) === null || _a === void 0 ? void 0 : _a.find(msg => { var _a, _b; return !!((_b = (_a = this.control) === null || _a === void 0 ? void 0 : _a.errors) === null || _b === void 0 ? void 0 : _b[error || msg.name]); });
        return (validation === null || validation === void 0 ? void 0 : validation.message) || '';
    }
}
/** @nocollapse */ QuickTransferValidationMessageComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: QuickTransferValidationMessageComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ QuickTransferValidationMessageComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: QuickTransferValidationMessageComponent, selector: "bb-quick-transfer-validation-message", inputs: { control: "control", validationMessages: "validationMessages", id: "id" }, ngImport: i0, template: "<bb-input-validation-message-ui\n  data-role=\"input-error\"\n  *ngIf=\"control?.touched && control?.invalid\"\n  [showErrors]=\"control?.touched && control?.invalid\"\n  [@fadeSlideInOut]\n>\n  <ng-container *ngIf=\"control?.errors?.['required']; else invalidError\">\n    <span data-role=\"required-error\" [attr.id]=\"id\" *ngIf=\"requiredMessage\">\n      {{ requiredMessage }}\n    </span>\n  </ng-container>\n\n  <ng-template #invalidError>\n    <span data-role=\"required-error\" [attr.id]=\"id\" *ngIf=\"getValidationMessage() as validationMessage\">\n      {{ validationMessage }}</span\n    >\n  </ng-template>\n</bb-input-validation-message-ui>\n", components: [{ type: i1.InputValidationMessageComponent, selector: "bb-input-validation-message-ui", inputs: ["showErrors", "messageId"] }], directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], animations: [fadeSlideAnimation] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: QuickTransferValidationMessageComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-quick-transfer-validation-message',
                    templateUrl: './validation-message.component.html',
                    animations: [fadeSlideAnimation],
                }]
        }], propDecorators: { control: [{
                type: Input
            }], validationMessages: [{
                type: Input
            }], id: [{
                type: Input
            }] } });
