// @backbase/impersonation-v1-client-ang/esm2015/api/identityImpersonation.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const IdentityImpersonationServiceGetImpersonationAccessTokenMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/impersonation/token",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "access_token": "eyJhbGciOiJSUzI...",
                    "refresh_token": "eyJhbGciOiJSUzI..."
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 408,
                body: {
                    "message": "Request Timeout",
                    "errors": [{
                            "message": "Server request timed out",
                            "key": "common.api.providerInternalError"
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const IdentityImpersonationServiceGetImpersonationTokenReferenceMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/impersonation/reference",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "ref": "MTNkYzE4YWYtNGUxOS00YjBkLWEwMGMtZjkyODE1NzU2Y2U0"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 408,
                body: {
                    "message": "Request Timeout",
                    "errors": [{
                            "message": "Server request timed out",
                            "key": "common.api.providerInternalError"
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const IdentityImpersonationServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v1/impersonation/token",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "access_token": "eyJhbGciOiJSUzI...",
                    "refresh_token": "eyJhbGciOiJSUzI..."
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 408,
                body: {
                    "message": "Request Timeout",
                    "errors": [{
                            "message": "Server request timed out",
                            "key": "common.api.providerInternalError"
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/impersonation/reference",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "ref": "MTNkYzE4YWYtNGUxOS00YjBkLWEwMGMtZjkyODE1NzU2Y2U0"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 408,
                body: {
                    "message": "Request Timeout",
                    "errors": [{
                            "message": "Server request timed out",
                            "key": "common.api.providerInternalError"
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
]);
