// @backbase/ui-ang/esm2015/collapsible-card/collapsible-card.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { CollapsibleCardComponent } from './collapsible-card.component';
import { CollapsibleModule } from '@backbase/ui-ang/collapsible';
import { IconModule } from '@backbase/ui-ang/icon';
import { KeyboardClickModule } from '@backbase/ui-ang/keyboard-click-directive';
import * as i0 from "@angular/core";
export class CollapsibleCardModule {
}
CollapsibleCardModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CollapsibleCardModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CollapsibleCardModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CollapsibleCardModule, declarations: [CollapsibleCardComponent], imports: [CollapsibleModule, CommonModule, IconModule, KeyboardClickModule], exports: [CollapsibleCardComponent] });
CollapsibleCardModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CollapsibleCardModule, imports: [[CollapsibleModule, CommonModule, IconModule, KeyboardClickModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CollapsibleCardModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CollapsibleModule, CommonModule, IconModule, KeyboardClickModule],
                    declarations: [CollapsibleCardComponent],
                    exports: [CollapsibleCardComponent],
                }]
        }] });
