// @backbase/ui-ang/esm2015/input-validation-message/input-validation-message.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { InputValidationMessageComponent } from './input-validation-message.component';
import * as i0 from "@angular/core";
export class InputValidationMessageModule {
}
InputValidationMessageModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputValidationMessageModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputValidationMessageModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputValidationMessageModule, declarations: [InputValidationMessageComponent], imports: [CommonModule], exports: [InputValidationMessageComponent] });
InputValidationMessageModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputValidationMessageModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputValidationMessageModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [InputValidationMessageComponent],
                    exports: [InputValidationMessageComponent],
                }]
        }] });
