// @backbase/ui-ang/esm2015/input-validation-message/input-validation-message.component.js
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
/**
 * @name InputValidationMessageComponent
 *
 * @description
 * Component that displays an input validation message or helper.
 */
export class InputValidationMessageComponent {
    constructor() {
        /**
         * Whether the errors template should be shown
         */
        this.showErrors = false;
    }
}
InputValidationMessageComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputValidationMessageComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
InputValidationMessageComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: InputValidationMessageComponent, selector: "bb-input-validation-message-ui", inputs: { showErrors: "showErrors", messageId: "messageId" }, ngImport: i0, template: "<ng-container *ngIf=\"showErrors\">\n  <small class=\"bb-input-validation-message\" role=\"alert\" [attr.id]=\"messageId || null\">\n    <ng-content></ng-content>\n  </small>\n</ng-container>\n", directives: [{ type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputValidationMessageComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-input-validation-message-ui',
                    templateUrl: './input-validation-message.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { showErrors: [{
                type: Input
            }], messageId: [{
                type: Input
            }] } });
