// @backbase/product-summary-common-ang/esm2015/src/pipes/date/date.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BbDatePipe, BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import { DateUpdatedAgoPipe } from './date-updated-ago.pipe';
import * as i0 from "@angular/core";
const sharedPipes = [BbDatePipeModule];
const pipes = [DateUpdatedAgoPipe];
export class DatePipeModule {
}
DatePipeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: DatePipeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
DatePipeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: DatePipeModule, declarations: [DateUpdatedAgoPipe], imports: [CommonModule, BbDatePipeModule], exports: [DateUpdatedAgoPipe, BbDatePipeModule] });
DatePipeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: DatePipeModule, providers: [BbDatePipe], imports: [[CommonModule, ...sharedPipes], BbDatePipeModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: DatePipeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...sharedPipes],
                    declarations: [...pipes],
                    exports: [...pipes, ...sharedPipes],
                    providers: [BbDatePipe],
                }]
        }] });
