// @backbase/messages-common/esm2015/src/components/messages-manipulation-confirm-modal/messages-manipulation-confirm-modal.service.js
import { Injectable } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import { ManipulationType } from './manipulation-confirm-modal-types.model';
import * as i0 from "@angular/core";
import * as i1 from "../../services/shared-methods/shared-methods.service";
export class MessagesManipulationConfirmModalService {
    constructor(shared) {
        this.shared = shared;
        this.initialModalVal = {
            opened: false,
            type: undefined,
        };
        this.emitToggleModal = new BehaviorSubject(this.initialModalVal);
        /**
         * Toggle modal event emitter
         */
        this.emitToggleModal$ = this.emitToggleModal.asObservable();
    }
    /**
     * Toggle confirm modal
     *
     * @param shouldOpen Should open modal flag
     * @param type Manipulation type
     * @param form Form data
     * @param prevValue Previous value
     */
    toggleModal(shouldOpen, type, form, prevValue) {
        let opened = shouldOpen;
        if (type === ManipulationType.discard) {
            opened = this.shared.isFormInvalid(form, prevValue);
        }
        this.emitToggleModal.next({ opened, type });
    }
}
MessagesManipulationConfirmModalService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesManipulationConfirmModalService, deps: [{ token: i1.SharedMethodsService }], target: i0.ɵɵFactoryTarget.Injectable });
MessagesManipulationConfirmModalService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesManipulationConfirmModalService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesManipulationConfirmModalService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.SharedMethodsService }]; } });
