// @backbase/messages-common/esm2015/src/components/messages-manipulation-confirm-modal/messages-manipulation-confirm-modal.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ButtonModule } from '@backbase/ui-ang/button';
import { HeaderModule } from '@backbase/ui-ang/header';
import { ModalModule } from '@backbase/ui-ang/modal';
import { MessagesManipulationConfirmModalService } from './messages-manipulation-confirm-modal.service';
import { MessagesManipulationConfirmModalComponent, MessagesManipulationConfirmModalCustomizableDirective, } from './messages-manipulation-confirm-modal.component';
import { BaseModalComponent } from '../base-modal/base-modal.component';
import * as i0 from "@angular/core";
const components = [MessagesManipulationConfirmModalComponent];
const directives = [MessagesManipulationConfirmModalCustomizableDirective];
const uiModule = [ModalModule, HeaderModule, ButtonModule];
export class MessagesManipulationConfirmModalModule {
}
MessagesManipulationConfirmModalModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesManipulationConfirmModalModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MessagesManipulationConfirmModalModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesManipulationConfirmModalModule, declarations: [BaseModalComponent, MessagesManipulationConfirmModalComponent, MessagesManipulationConfirmModalCustomizableDirective], imports: [CommonModule, ModalModule, HeaderModule, ButtonModule], exports: [MessagesManipulationConfirmModalComponent, MessagesManipulationConfirmModalCustomizableDirective] });
MessagesManipulationConfirmModalModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesManipulationConfirmModalModule, providers: [MessagesManipulationConfirmModalService, ...uiModule], imports: [[CommonModule, ...uiModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesManipulationConfirmModalModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [BaseModalComponent, ...components, ...directives],
                    exports: [...components, ...directives],
                    imports: [CommonModule, ...uiModule],
                    providers: [MessagesManipulationConfirmModalService, ...uiModule],
                }]
        }] });
