// @backbase/card-details-widget-ang/bundles/backbase-card-details-widget-ang.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@backbase/cards-common-ang'), require('@angular/common'), require('@angular/core'), require('@angular/forms'), require('@backbase/foundation-ang/core'), require('@backbase/ui-ang/alert'), require('@backbase/ui-ang/empty-state'), require('@backbase/ui-ang/loading-indicator'), require('@backbase/ui-ang/payment-card'), require('@backbase/ui-ang/switch'), require('@backbase/ui-ang/icon'), require('@backbase/ui-ang/notification'), require('rxjs'), require('rxjs/operators'), require('@angular/router')) :
    typeof define === 'function' && define.amd ? define('@backbase/card-details-widget-ang', ['exports', '@backbase/cards-common-ang', '@angular/common', '@angular/core', '@angular/forms', '@backbase/foundation-ang/core', '@backbase/ui-ang/alert', '@backbase/ui-ang/empty-state', '@backbase/ui-ang/loading-indicator', '@backbase/ui-ang/payment-card', '@backbase/ui-ang/switch', '@backbase/ui-ang/icon', '@backbase/ui-ang/notification', 'rxjs', 'rxjs/operators', '@angular/router'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["card-details-widget-ang"] = {}), global.i1, global.ng.common, global.ng.core, global.ng.forms, global.i1$2, global.i3$1, global.i4$1, global.i3, global.i2$1, global.i2, global.i1$1, global.i2$2, global.rxjs, global.rxjs.operators, global.ng.router));
})(this, (function (exports, i1, i4, i0, i6, i1$2, i3$1, i4$1, i3, i2$1, i2, i1$1, i2$2, rxjs, operators, i1$3) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i1__namespace$1 = /*#__PURE__*/_interopNamespace(i1);
    var i4__namespace = /*#__PURE__*/_interopNamespace(i4);
    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i6__namespace = /*#__PURE__*/_interopNamespace(i6);
    var i1__namespace$2 = /*#__PURE__*/_interopNamespace(i1$2);
    var i3__namespace$1 = /*#__PURE__*/_interopNamespace(i3$1);
    var i4__namespace$1 = /*#__PURE__*/_interopNamespace(i4$1);
    var i3__namespace = /*#__PURE__*/_interopNamespace(i3);
    var i2__namespace$1 = /*#__PURE__*/_interopNamespace(i2$1);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1$1);
    var i2__namespace$2 = /*#__PURE__*/_interopNamespace(i2$2);
    var i1__namespace$3 = /*#__PURE__*/_interopNamespace(i1$3);

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from, pack) {
        if (pack || arguments.length === 2)
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar)
                        ar = Array.prototype.slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
        return to.concat(ar || Array.prototype.slice.call(from));
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    var PaymentCardControlsComponent = /** @class */ (function (_super) {
        __extends(PaymentCardControlsComponent, _super);
        function PaymentCardControlsComponent() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        Object.defineProperty(PaymentCardControlsComponent.prototype, "lockStatus", {
            get: function () {
                return this.isLocked || false;
            },
            set: function (value) {
                if (value) {
                    this.onLock();
                }
                else {
                    this.onUnlock();
                }
            },
            enumerable: false,
            configurable: true
        });
        return PaymentCardControlsComponent;
    }(i1.PaymentCardControlsBase));
    PaymentCardControlsComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: PaymentCardControlsComponent, deps: null, target: i0__namespace.ɵɵFactoryTarget.Component });
    PaymentCardControlsComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: PaymentCardControlsComponent, selector: "bb-card-details-payment-card-controls", usesInheritance: true, ngImport: i0__namespace, template: "<ng-container bbCardDetailsPaymentCardControlsCustomizable [bbHostRef]=\"hostRef\"></ng-container>\n\n<ng-template bbCardDetailsPaymentCardControlsCustomizable let-hostRef>\n  <div class=\"bb-block--lg bb-list card\" *ngIf=\"hostRef.paymentCard\">\n   \n    <ng-container *ngIf=\"hostRef.paymentCard.status === 'ACTIVE'\">\n\n      <div class=\"card-header\">\n        <h2 i18n=\"Self service heading|Heading for the Self service section@@card.selfService.heading\">\n          Self service\n        </h2>\n      </div>\n      <div class=\"card-body\">\n        <div class=\"bb-list\">\n          <div class=\"bb-list__item bb-stack\">\n            <div class=\"bb-stack bb-stack--align-top\">\n              <bb-icon-ui \n                name=\"lock\"\n                size=\"md\"\n                class=\"bb-stack__item\"\n              ></bb-icon-ui>\n              <div class=\"bb-stack__item\">\n                <h5 class=\"bb-block--no-margin\" i18n=\"Lock or unlock card|Lock or unlock card@@card.controls.lockStatus.heading\">\n                  Lock card\n                </h5>\n                <span class=\"bb-text-support text-small\" i18n=\"Lock card description|Subtitle for lock card@@card.controls.lockStatus.description\">\n                  Freeze your card temporarily\n                </span>\n              </div>\n            </div>\n    \n            <form>\n              <bb-switch-ui\n                [id]=\"hostRef.lockStatus\"\n                *ngIf=\"!hostRef.isLockLoading\"\n                aria-label=\"lock/unlock your payment card, Freeze your card temporarily\"\n                i18n-aria-label=\"lock/unlock aria-label@@card.controls.lock.switch.aria-label\"\n                name=\"lockStatus\"\n                [(ngModel)]=\"hostRef.lockStatus\"\n              >\n              </bb-switch-ui>\n              <bb-loading-indicator-ui *ngIf=\"hostRef.isLockLoading\" loaderSize=\"sm\"></bb-loading-indicator-ui>\n            </form>\n          </div>\n    \n          <div class=\"bb-list__item bb-stack bb-payment-card-self-service-row\">\n            <div class=\"bb-stack bb-stack--align-top bb-stack__item bb-payment-card-self-service-row__description\">\n              <bb-icon-ui \n                name=\"autorenew\"\n                size=\"md\"\n                class=\"bb-stack__item\"\n              ></bb-icon-ui>\n            \n              <div class=\"bb-stack__item\">\n                <h5 class=\"bb-block--no-margin\" i18n=\"Replace card|Replace card@@card.controls.replaceCard.heading\">\n                  Replace card\n                </h5>\n                <span class=\"bb-text-support text-small\" i18n=\"Replace card description|Subtitle for replace card@@card.controls.replaceCard.description\">\n                  Replace lost, stolen or damaged cards\n                </span>\n              </div>\n            </div>\n            <div class=\"bb-stack__item bb-payment-card-self-service-row__button\">\n              <bb-payment-card-initiate-replacement\n                [paymentCard]=\"hostRef.paymentCard\"\n                [isInitiateReplacementLoading]=\"hostRef.isInitiateReplacementLoading\"\n                (initiateReplacement)=\"hostRef.onInitiateReplacement($event)\"\n              ></bb-payment-card-initiate-replacement>\n            </div>\n          </div>\n    \n          <div class=\"bb-list__item bb-stack bb-payment-card-self-service-row\">\n            <div class=\"bb-stack bb-stack--align-top bb-stack__item bb-payment-card-self-service-row__description\">\n              <bb-icon-ui \n                name=\"dialpad\"\n                size=\"md\"\n                class=\"bb-stack__item\"\n              ></bb-icon-ui>\n    \n              <div class=\"bb-stack__item\">\n                <ng-container *ngIf=\"hostRef.resetPinMode === 'RESET'\">\n                  <h5 class=\"bb-block--no-margin\" i18n=\"reset PIN|Reset PIN@@card.controls.resetPin.heading\">\n                    Reset PIN\n                  </h5>\n                  <span class=\"bb-text-support text-small\" i18n=\"Reset PIN description|Subtitle for reset PIN@@card.controls.resetPin.description\">\n                    Change your PIN online\n                  </span>\n                </ng-container>\n                <ng-container *ngIf=\"hostRef.resetPinMode === 'REQUEST'\">\n                  <h5 class=\"bb-block--no-margin\" i18n=\"Request PIN|Request PIN@@card.controls.requestPin.heading\" >\n                    Request new PIN\n                  </h5>\n                  <span class=\"bb-text-support text-small\" i18n=\"Request PIN description|Subtitle for request PIN@@card.controls.requestPin.description\">\n                    Request a new PIN by mail\n                  </span>\n                </ng-container>\n              </div>\n            </div>\n            <div class=\"bb-stack__item bb-payment-card-self-service-row__button\">\n              <bb-payment-card-reset-pin \n                [paymentCard]=\"hostRef.paymentCard\" \n                [mode]=\"hostRef.resetPinMode\"\n                [isResetPinLoading]=\"hostRef.isResetPinLoading\"\n                (resetPin)=\"hostRef.onResetPin($event)\"\n                (requestPin)=\"hostRef.onRequestPin($event)\"\n              ></bb-payment-card-reset-pin>\n            </div>\n          </div>\n        </div>\n      </div>\n      \n    </ng-container>\n    \n    <ng-container *ngIf=\"hostRef.paymentCard.status === 'INACTIVE'\">\n      <div class=\"card-body\">\n        <div class=\"bb-block bb-block--lg\">\n          <h2 i18n=\"Card activation heading|Heading for the Card activation section@@card.cardActivation.heading\">\n            Card activation\n          </h2>\n          <p \n            class=\"bb-text-support\"\n            i18n=\"Activation needed|Text for activation needed@@card.controls.activate.text\"\n          >\n            In order to use this card you need to activate it first.\n          </p>\n        </div>\n        <div class=\"bb-payment-card-self-service__inactive-container\">\n          <bb-payment-card-initiate-activation\n            [paymentCard]=\"hostRef.paymentCard\"\n            [isInitiateActivationLoading]=\"hostRef.isInitiateActivationLoading\"\n            (initiateActivation)=\"hostRef.onInitiateActivation($event)\"\n          ></bb-payment-card-initiate-activation>\n        </div>\n      </div>\n      \n    </ng-container>\n  </div>\n</ng-template>\n", components: [{ type: i0__namespace.forwardRef(function () { return i1__namespace.IconComponent; }), selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i0__namespace.forwardRef(function () { return i2__namespace.SwitchComponent; }), selector: "bb-switch-ui", inputs: ["labelPosition"] }, { type: i0__namespace.forwardRef(function () { return i3__namespace.LoadingIndicatorComponent; }), selector: "bb-loading-indicator-ui", inputs: ["text", "loaderSize", "showDelay", "hasBackground", "inline"] }, { type: i0__namespace.forwardRef(function () { return i1__namespace$1.PaymentCardInitiateReplacementComponent; }), selector: "bb-payment-card-initiate-replacement", inputs: ["paymentCard", "isInitiateReplacementLoading"], outputs: ["initiateReplacement"] }, { type: i0__namespace.forwardRef(function () { return i1__namespace$1.PaymentCardResetPinComponent; }), selector: "bb-payment-card-reset-pin", inputs: ["paymentCard", "mode", "isResetPinLoading"], outputs: ["resetPin", "requestPin"] }, { type: i0__namespace.forwardRef(function () { return i1__namespace$1.PaymentCardInitiateActivationComponent; }), selector: "bb-payment-card-initiate-activation", inputs: ["paymentCard", "isInitiateActivationLoading"], outputs: ["initiateActivation"] }], directives: [{ type: i0__namespace.forwardRef(function () { return CardDetailsPaymentCardControlsCustomizableDirective; }), selector: "[bbCardDetailsPaymentCardControlsCustomizable]" }, { type: i0__namespace.forwardRef(function () { return i4__namespace.NgIf; }), selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i0__namespace.forwardRef(function () { return i6__namespace.ɵNgNoValidate; }), selector: "form:not([ngNoForm]):not([ngNativeValidate])" }, { type: i0__namespace.forwardRef(function () { return i6__namespace.NgControlStatusGroup; }), selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { type: i0__namespace.forwardRef(function () { return i6__namespace.NgForm; }), selector: "form:not([ngNoForm]):not([formGroup]),ng-form,[ngForm]", inputs: ["ngFormOptions"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { type: i0__namespace.forwardRef(function () { return i6__namespace.NgControlStatus; }), selector: "[formControlName],[ngModel],[formControl]" }, { type: i0__namespace.forwardRef(function () { return i6__namespace.NgModel; }), selector: "[ngModel]:not([formControlName]):not([formControl])", inputs: ["name", "disabled", "ngModel", "ngModelOptions"], outputs: ["ngModelChange"], exportAs: ["ngModel"] }] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: PaymentCardControlsComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-card-details-payment-card-controls',
                        templateUrl: './payment-card-controls.component.html',
                    }]
            }] });
    var CardDetailsPaymentCardControlsCustomizableDirective = /** @class */ (function (_super) {
        __extends(CardDetailsPaymentCardControlsCustomizableDirective, _super);
        function CardDetailsPaymentCardControlsCustomizableDirective() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return CardDetailsPaymentCardControlsCustomizableDirective;
    }(i1$2.BbTemplate));
    CardDetailsPaymentCardControlsCustomizableDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: CardDetailsPaymentCardControlsCustomizableDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    CardDetailsPaymentCardControlsCustomizableDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.15", type: CardDetailsPaymentCardControlsCustomizableDirective, selector: "[bbCardDetailsPaymentCardControlsCustomizable]", usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: CardDetailsPaymentCardControlsCustomizableDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: '[bbCardDetailsPaymentCardControlsCustomizable]',
                    }]
            }] });

    var PaymentCardSummaryComponent = /** @class */ (function () {
        function PaymentCardSummaryComponent() {
            this.hostRef = this;
        }
        /**
         * Translate the payment card status/lock status/replacement status into a displayable
         * PaymentCardDisplayState.
         */
        PaymentCardSummaryComponent.prototype.getDisplayState = function () {
            if (this.paymentCard && this.paymentCard.status === i1.PaymentCardStatus.Inactive) {
                return 'INACTIVE';
            }
            if (this.paymentCard && this.paymentCard.status === i1.PaymentCardStatus.Canceled) {
                return 'CANCELLED';
            }
            if (this.paymentCard && this.paymentCard.status === i1.PaymentCardStatus.Expired) {
                return 'EXPIRED';
            }
            if (this.paymentCard && this.paymentCard.lockStatus === i1.PaymentCardLockStatus.Locked) {
                return 'LOCKED';
            }
            return undefined;
        };
        PaymentCardSummaryComponent.prototype.ngOnInit = function () {
            if (typeof this.paymentCard === 'undefined') {
                throw new Error("\"paymentCard\" input is required in \"" + this.constructor.name + "\"");
            }
        };
        return PaymentCardSummaryComponent;
    }());
    PaymentCardSummaryComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: PaymentCardSummaryComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    PaymentCardSummaryComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: PaymentCardSummaryComponent, selector: "bb-payment-card-summary", inputs: { paymentCard: "paymentCard" }, ngImport: i0__namespace, template: "<ng-container bbCardDetailsPaymentCardSummaryCustomizable [bbHostRef]=\"hostRef\"></ng-container>\n\n<ng-template bbCardDetailsPaymentCardSummaryCustomizable let-hostRef>\n  <div class=\"bb-payment-card-summary\" *ngIf=\"hostRef.paymentCard\">\n    <div class=\"bb-block bb-block--xl\">\n      <div class=\"row\">\n        <div class=\"col-auto\">\n          <div class=\" bb-payment-card-component\">\n            <bb-payment-card-state-wrapper-ui [state]=\"hostRef.getDisplayState()\">\n              <bb-payment-card-ui [paymentCard]=\"hostRef.paymentCard\"></bb-payment-card-ui>\n            </bb-payment-card-state-wrapper-ui>\n          </div>\n        </div>\n        <div class=\"col-12 order-first col-sm order-sm-last\">\n          <div class=\"bb-block bb-block--xl\">\n            <h2 i18n=\"Card payment header|Card payment title@@card.summary.heading\">\n              {{ hostRef.paymentCard.name }} {{ hostRef.paymentCard.type | lowercase }} card\n            </h2>\n          </div>\n        </div>\n      </div>\n    </div>\n\n    <ng-container *ngIf=\"hostRef.paymentCard.replacementStatus === 'UNDER_REPLACEMENT'\">\n      <div class=\"bb-block bb-block--xl\">\n        <bb-alert-ui\n          modifier=\"warning\"\n          title=\"Card under replacement\"\n          i18n-title=\"Card under replacement title|Card under replacement title@@card.summary.underReplacement.title\"\n          message=\"New cards usually take 3-5 business days to arrive.\"\n          i18n-message=\"Card under replacement text|Card under replacement@@card.summary.underReplacement.subtitle\"\n        ></bb-alert-ui>\n      </div>\n    </ng-container>\n\n    <ng-container *ngIf=\"hostRef.paymentCard.deliveryStatus\">\n      <bb-payment-card-delivery-status\n        [deliveryStatus]=\"hostRef.paymentCard.deliveryStatus\"\n      ></bb-payment-card-delivery-status>\n    </ng-container>\n  </div>\n</ng-template>\n", components: [{ type: i0__namespace.forwardRef(function () { return i1__namespace$1.PaymentCardStateWrapperComponent; }), selector: "bb-payment-card-state-wrapper,bb-payment-card-state-wrapper-ui", inputs: ["state"] }, { type: i0__namespace.forwardRef(function () { return i2__namespace$1.PaymentCardComponent; }), selector: "bb-payment-card-ui", inputs: ["paymentCard", "paymentCardNumberFormat"] }, { type: i0__namespace.forwardRef(function () { return i3__namespace$1.AlertComponent; }), selector: "bb-alert-ui", inputs: ["modifier", "dismissible", "title", "message"], outputs: ["close"] }, { type: i0__namespace.forwardRef(function () { return i1__namespace$1.PaymentCardDeliveryStatusComponent; }), selector: "bb-payment-card-delivery-status", inputs: ["deliveryStatus"] }], directives: [{ type: i0__namespace.forwardRef(function () { return CardDetailsPaymentCardSummaryCustomizableDirective; }), selector: "[bbCardDetailsPaymentCardSummaryCustomizable]" }, { type: i0__namespace.forwardRef(function () { return i4__namespace.NgIf; }), selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "lowercase": i0__namespace.forwardRef(function () { return i4__namespace.LowerCasePipe; }) } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: PaymentCardSummaryComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-payment-card-summary',
                        templateUrl: './payment-card-summary.component.html',
                    }]
            }], propDecorators: { paymentCard: [{
                    type: i0.Input
                }] } });
    var CardDetailsPaymentCardSummaryCustomizableDirective = /** @class */ (function (_super) {
        __extends(CardDetailsPaymentCardSummaryCustomizableDirective, _super);
        function CardDetailsPaymentCardSummaryCustomizableDirective() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return CardDetailsPaymentCardSummaryCustomizableDirective;
    }(i1$2.BbTemplate));
    CardDetailsPaymentCardSummaryCustomizableDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: CardDetailsPaymentCardSummaryCustomizableDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    CardDetailsPaymentCardSummaryCustomizableDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.15", type: CardDetailsPaymentCardSummaryCustomizableDirective, selector: "[bbCardDetailsPaymentCardSummaryCustomizable]", usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: CardDetailsPaymentCardSummaryCustomizableDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: '[bbCardDetailsPaymentCardSummaryCustomizable]',
                    }]
            }] });

    var PaymentCardComponent = /** @class */ (function (_super) {
        __extends(PaymentCardComponent, _super);
        function PaymentCardComponent(cardsService, cd, notificationService) {
            var _this = _super.call(this, cardsService, cd, notificationService) || this;
            _this.cardsService = cardsService;
            _this.cd = cd;
            _this.notificationService = notificationService;
            _this.hostRef = _this;
            return _this;
        }
        PaymentCardComponent.prototype.ngOnInit = function () {
            var _a, _b, _c, _d;
            /* eslint-disable max-len */
            var cardDetailsLocalizations = {
                activationErrorHeader: $localize(templateObject_1 || (templateObject_1 = __makeTemplateObject([":Error@@card.details.activation.notification.errorHeader:Error"], [":Error@@card.details.activation.notification.errorHeader:Error"]))),
                activationErrorMessage: $localize(templateObject_2 || (templateObject_2 = __makeTemplateObject([":Failed to Activate Card. Unable to proceed, please try again@@card.details.activation.notification.errorBody:Failed to Activate Card. Unable to proceed, please try again."], [":Failed to Activate Card. Unable to proceed, please try again@@card.details.activation.notification.errorBody:Failed to Activate Card. Unable to proceed, please try again."]))),
                activationSuccessHeader: $localize(templateObject_3 || (templateObject_3 = __makeTemplateObject([":Card Activated@@card.details.activation.notification.successHeader:Card Activated."], [":Card Activated@@card.details.activation.notification.successHeader:Card Activated."]))),
                lockErrorHeader: $localize(templateObject_4 || (templateObject_4 = __makeTemplateObject([":Error@@card.details.lock.notification.errorHeader:Error"], [":Error@@card.details.lock.notification.errorHeader:Error"]))),
                lockErrorMessage: $localize(templateObject_5 || (templateObject_5 = __makeTemplateObject([":Card number cannot be locked, try again later|Lock Error Message@@card.details.lock.notification.errorMessage:Card number ending with ", ":Card Number: cannot be locked! Try again later."], [":Card number cannot be locked, try again later|Lock Error Message@@card.details.lock.notification.errorMessage:Card number ending with ", ":Card Number: cannot be locked! Try again later."])), (_a = this.paymentCard) === null || _a === void 0 ? void 0 : _a.number),
                lockSuccessHeader: $localize(templateObject_6 || (templateObject_6 = __makeTemplateObject([":Card locked@@card.details.lock.notification.successHeader:Card locked"], [":Card locked@@card.details.lock.notification.successHeader:Card locked"]))),
                lockSuccessMessage: $localize(templateObject_7 || (templateObject_7 = __makeTemplateObject([":Card number locked|Successful lock message@@card.details.lock.notification.successMessage:Card number ending with ", ":Card Number: has been locked."], [":Card number locked|Successful lock message@@card.details.lock.notification.successMessage:Card number ending with ", ":Card Number: has been locked."])), (_b = this.paymentCard) === null || _b === void 0 ? void 0 : _b.number),
                replacementErrorHeader: $localize(templateObject_8 || (templateObject_8 = __makeTemplateObject([":Error@@card.details.replacement.notification.errorHeader:Error"], [":Error@@card.details.replacement.notification.errorHeader:Error"]))),
                replacementErrorMessage: $localize(templateObject_9 || (templateObject_9 = __makeTemplateObject([":Can't request card replacement. Try again@@card.details.replacement.notification.errorBody:Unable to request a card replacement. Try again later."], [":Can't request card replacement. Try again@@card.details.replacement.notification.errorBody:Unable to request a card replacement. Try again later."]))),
                replacementSuccessHeader: $localize(templateObject_10 || (templateObject_10 = __makeTemplateObject([":New card ordered header@@card.details.replacement.notification.successHeader:New card ordered"], [":New card ordered header@@card.details.replacement.notification.successHeader:New card ordered"]))),
                replacementSuccessMessage: $localize(templateObject_11 || (templateObject_11 = __makeTemplateObject([":New card ordered@@card.details.replacement.notification.successBody:A new card has been ordered."], [":New card ordered@@card.details.replacement.notification.successBody:A new card has been ordered."]))),
                requestNewPinSuccessMessage: $localize(templateObject_12 || (templateObject_12 = __makeTemplateObject([":New PIN requested@@card.details.pin.notification.requestNewPinSuccessMessage:New PIN requested"], [":New PIN requested@@card.details.pin.notification.requestNewPinSuccessMessage:New PIN requested"]))),
                requestPinErrorMessage: $localize(templateObject_13 || (templateObject_13 = __makeTemplateObject([":Can't request PIN. Try again@@card.details.requestPin.notification.errorBody:Can't request PIN.  Please try again later."], [":Can't request PIN. Try again@@card.details.requestPin.notification.errorBody:Can't request PIN.  Please try again later."]))),
                resetPinSuccessMessage: $localize(templateObject_14 || (templateObject_14 = __makeTemplateObject([":PIN successfully reset@@card.details.pin.notification.resetPinSuccessMessage:PIN successfully reset"], [":PIN successfully reset@@card.details.pin.notification.resetPinSuccessMessage:PIN successfully reset"]))),
                resetPinErrorMessage: $localize(templateObject_15 || (templateObject_15 = __makeTemplateObject([":Unable to change PIN, try again@@card.details.resetPin.notification.errorBody:Unable to change PIN.  Please try again later."], [":Unable to change PIN, try again@@card.details.resetPin.notification.errorBody:Unable to change PIN.  Please try again later."]))),
                resetInvalidPinMessage: $localize(templateObject_16 || (templateObject_16 = __makeTemplateObject([":Unable to change PIN, validations not meet@@card.details.pin.notification.resetInvalidPinMessage:Unable to change PIN. New PIN and confirm don't match"], [":Unable to change PIN, validations not meet@@card.details.pin.notification.resetInvalidPinMessage:Unable to change PIN. New PIN and confirm don't match"]))),
                unlockErrorHeader: $localize(templateObject_17 || (templateObject_17 = __makeTemplateObject([":Error@@card.details.unlock.notification.errorHeader:Error"], [":Error@@card.details.unlock.notification.errorHeader:Error"]))),
                unlockErrorMessage: $localize(templateObject_18 || (templateObject_18 = __makeTemplateObject([":Card number cannot be unlocked, try again later|Unlock Error Message@@card.details.unlock.notification.errorMessage:Card number ending with ", ":Card Number: cannot be unlocked! Try again later."], [":Card number cannot be unlocked, try again later|Unlock Error Message@@card.details.unlock.notification.errorMessage:Card number ending with ", ":Card Number: cannot be unlocked! Try again later."])), (_c = this.paymentCard) === null || _c === void 0 ? void 0 : _c.number),
                unlockSuccessHeader: $localize(templateObject_19 || (templateObject_19 = __makeTemplateObject([":Card unlocked@@card.details.unlock.notification.successHeader:Card unlocked"], [":Card unlocked@@card.details.unlock.notification.successHeader:Card unlocked"]))),
                unlockSuccessMessage: $localize(templateObject_20 || (templateObject_20 = __makeTemplateObject([":Card number unlocked|Successful unlock Error Message@@card.details.unlock.notification.successMessage:Card number ending with ", ":Card Number: has been unlocked."], [":Card number unlocked|Successful unlock Error Message@@card.details.unlock.notification.successMessage:Card number ending with ", ":Card Number: has been unlocked."])), (_d = this.paymentCard) === null || _d === void 0 ? void 0 : _d.number),
                updateLimitErrorHeader: $localize(templateObject_21 || (templateObject_21 = __makeTemplateObject([":Update Limit Error Header@@card.details.limit.notification.errorHeader:"], [":Update Limit Error Header@@card.details.limit.notification.errorHeader:"]))),
                updateLimitErrorMessage: $localize(templateObject_22 || (templateObject_22 = __makeTemplateObject([":Unable to change card limit@@card.details.limit.notification.errorBody:Unable to change card limit"], [":Unable to change card limit@@card.details.limit.notification.errorBody:Unable to change card limit"]))),
                updateLimitSuccessHeader: $localize(templateObject_23 || (templateObject_23 = __makeTemplateObject([":Limit changed header@@card.details.limit.notification.successHeader:Limit changed"], [":Limit changed header@@card.details.limit.notification.successHeader:Limit changed"]))),
                updateLimitSuccessMessage: $localize(templateObject_24 || (templateObject_24 = __makeTemplateObject([":Card limit changed successfully@@card.details.limit.notification.changeLimitSuccessMessage:Card limit changed successfully"], [":Card limit changed successfully@@card.details.limit.notification.changeLimitSuccessMessage:Card limit changed successfully"]))),
            };
            this.localizations = Object.assign({}, cardDetailsLocalizations);
        };
        return PaymentCardComponent;
    }(i1.PaymentCardContainerBase));
    PaymentCardComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: PaymentCardComponent, deps: [{ token: i1__namespace$1.CardsService }, { token: i0__namespace.ChangeDetectorRef }, { token: i2__namespace$2.NotificationService }], target: i0__namespace.ɵɵFactoryTarget.Component });
    PaymentCardComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: PaymentCardComponent, selector: "bb-payment-card", usesInheritance: true, ngImport: i0__namespace, template: "<ng-container bbCardDetailsPaymentCardCustomizable [bbHostRef]=\"hostRef\"></ng-container>\n\n<ng-template bbCardDetailsPaymentCardCustomizable let-hostRef>\n  <bb-payment-card-summary\n    [paymentCard]=\"hostRef.paymentCard\"\n  ></bb-payment-card-summary>\n\n  <bb-card-details-payment-card-controls\n    [paymentCard]=\"hostRef.paymentCard\"\n    [isLockLoading]=\"hostRef.isLockLoading\"\n    [isInitiateReplacementLoading]=\"hostRef.isInitiateReplacementLoading\"\n    [isInitiateActivationLoading]=\"hostRef.isInitiateActivationLoading\"\n    [isResetPinLoading]=\"hostRef.isResetPinLoading\"\n    (updateLockStatus)=\"hostRef.onUpdateLockStatus($event)\"\n    (initiateReplacement)=\"hostRef.onInitiateReplacement($event)\"\n    (initiateActivation)=\"hostRef.onInitiateActivation($event)\"\n    (resetPin)=\"hostRef.onResetPin($event)\"\n    (requestPin)=\"hostRef.onRequestPin($event)\"\n    *ngIf=\"hostRef.paymentCard?.status !== 'CANCELED'\"\n  ></bb-card-details-payment-card-controls>\n\n  <bb-payment-card-limits\n    [paymentCard]=\"hostRef.paymentCard\"\n    [isLimitsLoading]=\"hostRef.isLimitsLoading\"\n    *ngIf=\"hostRef.paymentCard?.status === 'ACTIVE'\"\n    (updateLimit)=\"hostRef.onUpdateLimit($event)\"\n  ></bb-payment-card-limits>\n</ng-template>", components: [{ type: i0__namespace.forwardRef(function () { return PaymentCardSummaryComponent; }), selector: "bb-payment-card-summary", inputs: ["paymentCard"] }, { type: i0__namespace.forwardRef(function () { return PaymentCardControlsComponent; }), selector: "bb-card-details-payment-card-controls" }, { type: i0__namespace.forwardRef(function () { return i1__namespace$1.PaymentCardLimitsComponent; }), selector: "bb-payment-card-limits", inputs: ["paymentCard", "isLimitsLoading"], outputs: ["updateLimit"] }], directives: [{ type: i0__namespace.forwardRef(function () { return CardDetailsPaymentCardCustomizableDirective; }), selector: "[bbCardDetailsPaymentCardCustomizable]" }, { type: i0__namespace.forwardRef(function () { return i4__namespace.NgIf; }), selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: PaymentCardComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-payment-card',
                        templateUrl: './payment-card.component.html',
                    }]
            }], ctorParameters: function () { return [{ type: i1__namespace$1.CardsService }, { type: i0__namespace.ChangeDetectorRef }, { type: i2__namespace$2.NotificationService }]; } });
    var CardDetailsPaymentCardCustomizableDirective = /** @class */ (function (_super) {
        __extends(CardDetailsPaymentCardCustomizableDirective, _super);
        function CardDetailsPaymentCardCustomizableDirective() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return CardDetailsPaymentCardCustomizableDirective;
    }(i1$2.BbTemplate));
    CardDetailsPaymentCardCustomizableDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: CardDetailsPaymentCardCustomizableDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    CardDetailsPaymentCardCustomizableDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.15", type: CardDetailsPaymentCardCustomizableDirective, selector: "[bbCardDetailsPaymentCardCustomizable]", usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: CardDetailsPaymentCardCustomizableDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: '[bbCardDetailsPaymentCardCustomizable]',
                    }]
            }] });
    var templateObject_1, templateObject_2, templateObject_3, templateObject_4, templateObject_5, templateObject_6, templateObject_7, templateObject_8, templateObject_9, templateObject_10, templateObject_11, templateObject_12, templateObject_13, templateObject_14, templateObject_15, templateObject_16, templateObject_17, templateObject_18, templateObject_19, templateObject_20, templateObject_21, templateObject_22, templateObject_23, templateObject_24;

    var WidgetPropertiesService = /** @class */ (function (_super) {
        __extends(WidgetPropertiesService, _super);
        function WidgetPropertiesService(itemModel) {
            var _this = _super.call(this, itemModel) || this;
            _this.itemModel = itemModel;
            return _this;
        }
        return WidgetPropertiesService;
    }(i1.BaseWidgetPropertiesService));
    WidgetPropertiesService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: WidgetPropertiesService, deps: [{ token: i1__namespace$2.ItemModel, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    WidgetPropertiesService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: WidgetPropertiesService });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: WidgetPropertiesService, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () {
            return [{ type: i1__namespace$2.ItemModel, decorators: [{
                            type: i0.Optional
                        }] }];
        } });

    var CardDetailsContainerComponent = /** @class */ (function () {
        function CardDetailsContainerComponent(cardService, cd, widgetPropertiesService) {
            var _this = this;
            this.cardService = cardService;
            this.cd = cd;
            this.widgetPropertiesService = widgetPropertiesService;
            this.notificationTtlSubject = new rxjs.BehaviorSubject(undefined);
            this.updateAllLimitsSubject = new rxjs.BehaviorSubject(undefined);
            this.handleError = function (errorResponse) {
                _this.errors = errorResponse;
                return rxjs.of(undefined);
            };
        }
        Object.defineProperty(CardDetailsContainerComponent.prototype, "notificationTtlValue", {
            /**
             * Notification timeout timer value configuration
             */
            // eslint-disable-next-line
            set: function (value) {
                this.notificationTtlSubject.next(value);
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(CardDetailsContainerComponent.prototype, "updateAllLimitsValue", {
            /**
             * Enable to send multiple limits to the API when updating values
             */
            set: function (value) {
                this.updateAllLimitsSubject.next(value);
            },
            enumerable: false,
            configurable: true
        });
        CardDetailsContainerComponent.prototype.ngOnInit = function () {
            var _this = this;
            if (typeof this.paymentCardId === 'undefined') {
                throw new Error("\"paymentCardId\" input is required in \"" + this.constructor.name + "\"");
            }
            this.cardService.getPaymentCardFrom(this.paymentCardId);
            this.paymentCard = this.cardService.getPaymentCard().pipe(operators.tap(function () {
                _this.cd.markForCheck();
            }), operators.catchError(this.handleError));
        };
        Object.defineProperty(CardDetailsContainerComponent.prototype, "notificationTtl", {
            get: function () {
                return i1.filterInputParameters(this.notificationTtlSubject, this.widgetPropertiesService.notificationTtl);
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(CardDetailsContainerComponent.prototype, "updateAllLimits", {
            get: function () {
                return i1.filterInputParameters(this.updateAllLimitsSubject, this.widgetPropertiesService.updateAllLimits);
            },
            enumerable: false,
            configurable: true
        });
        return CardDetailsContainerComponent;
    }());
    CardDetailsContainerComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: CardDetailsContainerComponent, deps: [{ token: i1__namespace$1.CardService }, { token: i0__namespace.ChangeDetectorRef }, { token: WidgetPropertiesService, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Component });
    CardDetailsContainerComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: CardDetailsContainerComponent, selector: "bb-card-details-container", inputs: { paymentCardId: "paymentCardId", notificationTtlValue: ["notificationTtl", "notificationTtlValue"], updateAllLimitsValue: ["updateAllLimits", "updateAllLimitsValue"] }, providers: [i1.CardService, WidgetPropertiesService], ngImport: i0__namespace, template: "<ng-container *ngIf=\"paymentCardId | async; else emptyState\">\n  <ng-container *ngIf=\"!errors; else errorState\">\n    <ng-container *ngIf=\"paymentCard | async as paymentCard; else loadingState\">\n      <bb-payment-card \n        [paymentCard]=\"paymentCard\" \n        [notificationTtl]=\"(notificationTtl | async) || undefined\"\n        [updateAllLimits]=\"(updateAllLimits | async) || undefined\"\n      ></bb-payment-card>\n    </ng-container>\n  </ng-container>\n</ng-container>\n\n<ng-template #emptyState>\n  <bb-empty-state-ui title=\"No Card Selected\"></bb-empty-state-ui>\n</ng-template>\n\n<ng-template #loadingState>\n  <bb-loading-indicator-ui loaderSize=\"lg\"></bb-loading-indicator-ui>\n</ng-template>\n\n<ng-template #errorState>\n  <bb-payment-card-multiple-error-state [errors]=\"errors\"></bb-payment-card-multiple-error-state>\n</ng-template>\n", components: [{ type: PaymentCardComponent, selector: "bb-payment-card" }, { type: i4__namespace$1.EmptyStateComponent, selector: "bb-empty-state-ui", inputs: ["title", "subtitle", "showIcon", "iconClasses", "iconModifier", "iconSize", "iconColor"] }, { type: i3__namespace.LoadingIndicatorComponent, selector: "bb-loading-indicator-ui", inputs: ["text", "loaderSize", "showDelay", "hasBackground", "inline"] }, { type: i1__namespace$1.PaymentCardMultipleErrorStateComponent, selector: "bb-payment-card-multiple-error-state", inputs: ["errors"] }], directives: [{ type: i4__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "async": i4__namespace.AsyncPipe } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: CardDetailsContainerComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-card-details-container',
                        templateUrl: './card-details-container.component.html',
                        providers: [i1.CardService, WidgetPropertiesService],
                    }]
            }], ctorParameters: function () {
            return [{ type: i1__namespace$1.CardService }, { type: i0__namespace.ChangeDetectorRef }, { type: WidgetPropertiesService, decorators: [{
                            type: i0.Optional
                        }] }];
        }, propDecorators: { paymentCardId: [{
                    type: i0.Input
                }], notificationTtlValue: [{
                    type: i0.Input,
                    args: ['notificationTtl']
                }], updateAllLimitsValue: [{
                    type: i0.Input,
                    args: ['updateAllLimits']
                }] } });

    var CardDetailsWidgetComponent = /** @class */ (function () {
        function CardDetailsWidgetComponent(route) {
            this.route = route;
            this.paymentCardId = this.getRouteParam(this.route, 'selectedCard').pipe(operators.map(function (selectedCard) { return selectedCard; }));
        }
        CardDetailsWidgetComponent.prototype.getRouteParam = function (route, param) {
            var paramValue = route.paramMap.pipe(operators.pluck('params', param));
            if (!route.parent) {
                return paramValue;
            }
            // Ensure that level that has actual value keeps it from being cleared.
            return rxjs.combineLatest(paramValue, this.getRouteParam(route.parent, param)).pipe(operators.map(function (params) { return params[0] || params[1]; }));
        };
        return CardDetailsWidgetComponent;
    }());
    CardDetailsWidgetComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: CardDetailsWidgetComponent, deps: [{ token: i1__namespace$3.ActivatedRoute }], target: i0__namespace.ɵɵFactoryTarget.Component });
    CardDetailsWidgetComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: CardDetailsWidgetComponent, selector: "bb-card-details-widget", inputs: { notificationTtl: "notificationTtl", updateAllLimits: "updateAllLimits" }, ngImport: i0__namespace, template: "<bb-card-details-container \n  [paymentCardId]=\"paymentCardId\"\n  [notificationTtl]=\"notificationTtl\"\n  [updateAllLimits]=\"updateAllLimits\"\n></bb-card-details-container>", components: [{ type: CardDetailsContainerComponent, selector: "bb-card-details-container", inputs: ["paymentCardId", "notificationTtl", "updateAllLimits"] }] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: CardDetailsWidgetComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-card-details-widget',
                        templateUrl: './card-details-widget.component.html',
                    }]
            }], ctorParameters: function () { return [{ type: i1__namespace$3.ActivatedRoute }]; }, propDecorators: { notificationTtl: [{
                    type: i0.Input
                }], updateAllLimits: [{
                    type: i0.Input
                }] } });

    var modules = [
        i1.CardsCommonModule,
        i2.SwitchModule,
        i3.LoadingIndicatorModule,
        i4$1.EmptyStateModule,
        i2$1.PaymentCardModule,
        i3$1.AlertModule,
    ];
    var components = [
        CardDetailsWidgetComponent,
        CardDetailsContainerComponent,
        PaymentCardComponent,
        PaymentCardSummaryComponent,
        PaymentCardControlsComponent,
    ];
    var customizables = [
        CardDetailsPaymentCardCustomizableDirective,
        CardDetailsPaymentCardControlsCustomizableDirective,
        CardDetailsPaymentCardSummaryCustomizableDirective,
    ];
    var CardDetailsWidgetModule = /** @class */ (function () {
        function CardDetailsWidgetModule() {
        }
        return CardDetailsWidgetModule;
    }());
    CardDetailsWidgetModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: CardDetailsWidgetModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    CardDetailsWidgetModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: CardDetailsWidgetModule, declarations: [CardDetailsWidgetComponent,
            CardDetailsContainerComponent,
            PaymentCardComponent,
            PaymentCardSummaryComponent,
            PaymentCardControlsComponent, CardDetailsPaymentCardCustomizableDirective,
            CardDetailsPaymentCardControlsCustomizableDirective,
            CardDetailsPaymentCardSummaryCustomizableDirective], imports: [i4.CommonModule,
            i6.FormsModule, i1__namespace$2.BackbaseCoreModule, i1.CardsCommonModule,
            i2.SwitchModule,
            i3.LoadingIndicatorModule,
            i4$1.EmptyStateModule,
            i2$1.PaymentCardModule,
            i3$1.AlertModule], exports: [i1.CardsCommonModule,
            i2.SwitchModule,
            i3.LoadingIndicatorModule,
            i4$1.EmptyStateModule,
            i2$1.PaymentCardModule,
            i3$1.AlertModule, CardDetailsWidgetComponent,
            CardDetailsContainerComponent,
            PaymentCardComponent,
            PaymentCardSummaryComponent,
            PaymentCardControlsComponent, CardDetailsPaymentCardCustomizableDirective,
            CardDetailsPaymentCardControlsCustomizableDirective,
            CardDetailsPaymentCardSummaryCustomizableDirective] });
    CardDetailsWidgetModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: CardDetailsWidgetModule, imports: [__spreadArray([
                i4.CommonModule,
                i6.FormsModule,
                i1$2.BackbaseCoreModule.withConfig({
                    classMap: { CardDetailsWidgetComponent: CardDetailsWidgetComponent },
                })
            ], __read(modules)), i1.CardsCommonModule,
            i2.SwitchModule,
            i3.LoadingIndicatorModule,
            i4$1.EmptyStateModule,
            i2$1.PaymentCardModule,
            i3$1.AlertModule] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: CardDetailsWidgetModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: __spreadArray([
                            i4.CommonModule,
                            i6.FormsModule,
                            i1$2.BackbaseCoreModule.withConfig({
                                classMap: { CardDetailsWidgetComponent: CardDetailsWidgetComponent },
                            })
                        ], __read(modules)),
                        declarations: __spreadArray(__spreadArray([], __read(components)), __read(customizables)),
                        exports: __spreadArray(__spreadArray(__spreadArray([], __read(modules)), __read(components)), __read(customizables)),
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    Object.defineProperty(exports, 'CardsBackendError', {
        enumerable: true,
        get: function () { return i1.CardsBackendError; }
    });
    Object.defineProperty(exports, 'LimitChannelMappingConfigToken', {
        enumerable: true,
        get: function () { return i1.LimitChannelMappingConfigToken; }
    });
    Object.defineProperty(exports, 'PaymentCardLockStatus', {
        enumerable: true,
        get: function () { return i1.PaymentCardLockStatus; }
    });
    Object.defineProperty(exports, 'PaymentCardReplacementReason', {
        enumerable: true,
        get: function () { return i1.PaymentCardReplacementReason; }
    });
    Object.defineProperty(exports, 'PaymentCardReplacementStatus', {
        enumerable: true,
        get: function () { return i1.PaymentCardReplacementStatus; }
    });
    Object.defineProperty(exports, 'PaymentCardStatus', {
        enumerable: true,
        get: function () { return i1.PaymentCardStatus; }
    });
    Object.defineProperty(exports, 'ReplacementReasonMappingConfigToken', {
        enumerable: true,
        get: function () { return i1.ReplacementReasonMappingConfigToken; }
    });
    Object.defineProperty(exports, 'ReplacementStatusMappingConfigToken', {
        enumerable: true,
        get: function () { return i1.ReplacementStatusMappingConfigToken; }
    });
    Object.defineProperty(exports, 'StatusMappingConfigToken', {
        enumerable: true,
        get: function () { return i1.StatusMappingConfigToken; }
    });
    exports.CardDetailsContainerComponent = CardDetailsContainerComponent;
    exports.CardDetailsPaymentCardControlsCustomizableDirective = CardDetailsPaymentCardControlsCustomizableDirective;
    exports.CardDetailsPaymentCardCustomizableDirective = CardDetailsPaymentCardCustomizableDirective;
    exports.CardDetailsPaymentCardSummaryCustomizableDirective = CardDetailsPaymentCardSummaryCustomizableDirective;
    exports.CardDetailsWidgetComponent = CardDetailsWidgetComponent;
    exports.CardDetailsWidgetModule = CardDetailsWidgetModule;
    exports.PaymentCardComponent = PaymentCardComponent;
    exports.PaymentCardControlsComponent = PaymentCardControlsComponent;
    exports.PaymentCardSummaryComponent = PaymentCardSummaryComponent;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
