// @backbase/notifications-badge-widget-ang/esm2015/lib/components/notifications-list-item/notifications-list-item.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NotificationListItemCustomizableDirective, NotificationsListItemComponent, } from './notifications-list-item.component';
import { HumanizedDateModule } from '../humanized-date/humanized-date.module';
import { NotificationsCommonModule } from '@backbase/notifications-common-ang';
import { BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import { EllipsisModule } from '@backbase/ui-ang/ellipsis';
import * as i0 from "@angular/core";
const uiModules = [BbDatePipeModule, EllipsisModule];
const innerModules = [HumanizedDateModule];
const components = [NotificationsListItemComponent, NotificationListItemCustomizableDirective];
export class NotificationsListItemModule {
}
/** @nocollapse */ NotificationsListItemModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsListItemModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ NotificationsListItemModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsListItemModule, declarations: [NotificationsListItemComponent, NotificationListItemCustomizableDirective], imports: [CommonModule, NotificationsCommonModule, BbDatePipeModule, EllipsisModule, HumanizedDateModule], exports: [NotificationsCommonModule, BbDatePipeModule, EllipsisModule, HumanizedDateModule, NotificationsListItemComponent, NotificationListItemCustomizableDirective] });
/** @nocollapse */ NotificationsListItemModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsListItemModule, imports: [[CommonModule, NotificationsCommonModule, ...uiModules, ...innerModules], NotificationsCommonModule, BbDatePipeModule, EllipsisModule, HumanizedDateModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsListItemModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, NotificationsCommonModule, ...uiModules, ...innerModules],
                    declarations: [...components],
                    exports: [NotificationsCommonModule, ...uiModules, ...innerModules, ...components],
                }]
        }] });
