// @backbase/actions-http-ang/esm2015/api/actionRecipeSpecifications.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const ActionRecipeSpecificationsHttpServiceGetActionRecipeSpecificationsMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/action-recipe-specifications",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "actionRecipeSpecifications": [{
                            "id": "1",
                            "name": "New Transaction",
                            "userDefinable": true,
                            "origin": "Transaction",
                            "type": "newTransaction",
                            "category": "arrangementBased",
                            "actions": [{
                                    "type": "notification",
                                    "severityLevel": "INFO",
                                    "route": {
                                        "whereTo": "transaction-view",
                                        "dataTemplate": "{\"id\": \"${transactions[0].id}\", \"arrangementId\": \"${transactions[0].arrangementId}\"}"
                                    },
                                    "templates": [{
                                            "locale": "en-US",
                                            "kind": "REGULAR",
                                            "path": "/pathto/template",
                                            "repository": "repository_id"
                                        }, {
                                            "locale": "en-US",
                                            "kind": "DIGEST",
                                            "path": "/pathto/template",
                                            "repository": "repository_id"
                                        }]
                                }, {
                                    "type": "email",
                                    "templates": [{
                                            "locale": "en-US",
                                            "kind": "REGULAR",
                                            "path": "/pathto/template",
                                            "repository": "repository_id"
                                        }, {
                                            "locale": "en-US",
                                            "kind": "DIGEST",
                                            "path": "/pathto/template",
                                            "repository": "repository_id"
                                        }]
                                }, {
                                    "type": "sms",
                                    "templates": [{
                                            "locale": "en-US",
                                            "kind": "REGULAR",
                                            "path": "/pathto/template",
                                            "repository": "repository_id"
                                        }, {
                                            "locale": "en-US",
                                            "kind": "DIGEST",
                                            "path": "/pathto/template",
                                            "repository": "repository_id"
                                        }]
                                }, {
                                    "type": "push",
                                    "severityLevel": "INFO",
                                    "route": {
                                        "whereTo": "transaction-view",
                                        "dataTemplate": "{\"id\": \"${transactions[0].id}\", \"arrangementId\": \"${transactions[0].arrangementId}\"}"
                                    },
                                    "templates": [{
                                            "locale": "en-US",
                                            "kind": "REGULAR",
                                            "path": "/pathto/template",
                                            "repository": "repository_id"
                                        }, {
                                            "locale": "en-US",
                                            "kind": "DIGEST",
                                            "path": "/pathto/template",
                                            "repository": "repository_id"
                                        }]
                                }],
                            "digestEnabled": true,
                            "recipeDefaults": {
                                "amount": "0",
                                "actions": [{
                                        "type": "notification"
                                    }, {
                                        "type": "push"
                                    }]
                            }
                        }, {
                            "id": "4",
                            "name": "Account Low Balance",
                            "userDefinable": true,
                            "type": "accountLowBalance",
                            "category": "arrangementBased",
                            "actions": [{
                                    "type": "notification"
                                }, {
                                    "type": "email"
                                }, {
                                    "type": "sms"
                                }, {
                                    "type": "push"
                                }],
                            "digestEnabled": false,
                            "recipeDefaults": {
                                "amount": "200",
                                "actions": [{
                                        "type": "notification"
                                    }, {
                                        "type": "push"
                                    }],
                                "additions": {
                                    "type": "Cash Withdraw"
                                }
                            }
                        }, {
                            "id": "5",
                            "name": "Payment Status Updated",
                            "userDefinable": true,
                            "type": "paymentStatusUpdated",
                            "category": "paymentStatusBased",
                            "actions": [{
                                    "type": "notification"
                                }, {
                                    "type": "push"
                                }],
                            "digestEnabled": true,
                            "recipeDefaults": {
                                "actions": [{
                                        "type": "notification"
                                    }, {
                                        "type": "push"
                                    }]
                            }
                        }, {
                            "id": "6",
                            "name": "Party Status Updated",
                            "userDefinable": true,
                            "type": "partyStatusUpdated",
                            "category": "partyStatusBased",
                            "actions": [{
                                    "type": "notification"
                                }, {
                                    "type": "push"
                                }],
                            "digestEnabled": false,
                            "recipeDefaults": {
                                "actions": [{
                                        "type": "notification"
                                    }, {
                                        "type": "push"
                                    }]
                            }
                        }, {
                            "id": "8",
                            "name": "Saving goal",
                            "type": "savingGoal",
                            "category": "userBased",
                            "userDefinable": true,
                            "actions": [{
                                    "type": "notification"
                                }, {
                                    "type": "push"
                                }],
                            "digestEnabled": false
                        }, {
                            "id": "9",
                            "name": "Account statement is ready",
                            "type": "accountStatementIsReady",
                            "category": "arrangementBased",
                            "userDefinable": true,
                            "actions": [{
                                    "type": "email"
                                }, {
                                    "type": "notification"
                                }, {
                                    "type": "push"
                                }],
                            "digestEnabled": false
                        }, {
                            "id": "10",
                            "name": "Account Low Balance Business",
                            "type": "accountLowBalance",
                            "category": "arrangementBased",
                            "userDefinable": true,
                            "actions": [{
                                    "type": "notification"
                                }, {
                                    "type": "push"
                                }],
                            "digestEnabled": false,
                            "recipeDefaults": {
                                "amount": "0",
                                "actions": [{
                                        "type": "notification"
                                    }, {
                                        "type": "push"
                                    }]
                            }
                        }, {
                            "id": "13",
                            "name": "Payment to approve",
                            "type": "paymentToApprove",
                            "category": "paymentResourceBased",
                            "digestEnabled": false,
                            "recipeDefaults": {
                                "actions": [{
                                        "type": "notification"
                                    }, {
                                        "type": "push"
                                    }],
                                "additions": {
                                    "resource": "Payments"
                                }
                            }
                        }, {
                            "id": "14",
                            "name": "Contact to approve",
                            "type": "contactToApprove",
                            "category": "contactResourceBased",
                            "digestEnabled": false,
                            "recipeDefaults": {
                                "actions": [{
                                        "type": "notification"
                                    }, {
                                        "type": "push"
                                    }],
                                "additions": {
                                    "resource": "Contacts"
                                }
                            }
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
        ]
    }]);
export const ActionRecipeSpecificationsHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v2/action-recipe-specifications",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "actionRecipeSpecifications": [{
                            "id": "1",
                            "name": "New Transaction",
                            "userDefinable": true,
                            "origin": "Transaction",
                            "type": "newTransaction",
                            "category": "arrangementBased",
                            "actions": [{
                                    "type": "notification",
                                    "severityLevel": "INFO",
                                    "route": {
                                        "whereTo": "transaction-view",
                                        "dataTemplate": "{\"id\": \"${transactions[0].id}\", \"arrangementId\": \"${transactions[0].arrangementId}\"}"
                                    },
                                    "templates": [{
                                            "locale": "en-US",
                                            "kind": "REGULAR",
                                            "path": "/pathto/template",
                                            "repository": "repository_id"
                                        }, {
                                            "locale": "en-US",
                                            "kind": "DIGEST",
                                            "path": "/pathto/template",
                                            "repository": "repository_id"
                                        }]
                                }, {
                                    "type": "email",
                                    "templates": [{
                                            "locale": "en-US",
                                            "kind": "REGULAR",
                                            "path": "/pathto/template",
                                            "repository": "repository_id"
                                        }, {
                                            "locale": "en-US",
                                            "kind": "DIGEST",
                                            "path": "/pathto/template",
                                            "repository": "repository_id"
                                        }]
                                }, {
                                    "type": "sms",
                                    "templates": [{
                                            "locale": "en-US",
                                            "kind": "REGULAR",
                                            "path": "/pathto/template",
                                            "repository": "repository_id"
                                        }, {
                                            "locale": "en-US",
                                            "kind": "DIGEST",
                                            "path": "/pathto/template",
                                            "repository": "repository_id"
                                        }]
                                }, {
                                    "type": "push",
                                    "severityLevel": "INFO",
                                    "route": {
                                        "whereTo": "transaction-view",
                                        "dataTemplate": "{\"id\": \"${transactions[0].id}\", \"arrangementId\": \"${transactions[0].arrangementId}\"}"
                                    },
                                    "templates": [{
                                            "locale": "en-US",
                                            "kind": "REGULAR",
                                            "path": "/pathto/template",
                                            "repository": "repository_id"
                                        }, {
                                            "locale": "en-US",
                                            "kind": "DIGEST",
                                            "path": "/pathto/template",
                                            "repository": "repository_id"
                                        }]
                                }],
                            "digestEnabled": true,
                            "recipeDefaults": {
                                "amount": "0",
                                "actions": [{
                                        "type": "notification"
                                    }, {
                                        "type": "push"
                                    }]
                            }
                        }, {
                            "id": "4",
                            "name": "Account Low Balance",
                            "userDefinable": true,
                            "type": "accountLowBalance",
                            "category": "arrangementBased",
                            "actions": [{
                                    "type": "notification"
                                }, {
                                    "type": "email"
                                }, {
                                    "type": "sms"
                                }, {
                                    "type": "push"
                                }],
                            "digestEnabled": false,
                            "recipeDefaults": {
                                "amount": "200",
                                "actions": [{
                                        "type": "notification"
                                    }, {
                                        "type": "push"
                                    }],
                                "additions": {
                                    "type": "Cash Withdraw"
                                }
                            }
                        }, {
                            "id": "5",
                            "name": "Payment Status Updated",
                            "userDefinable": true,
                            "type": "paymentStatusUpdated",
                            "category": "paymentStatusBased",
                            "actions": [{
                                    "type": "notification"
                                }, {
                                    "type": "push"
                                }],
                            "digestEnabled": true,
                            "recipeDefaults": {
                                "actions": [{
                                        "type": "notification"
                                    }, {
                                        "type": "push"
                                    }]
                            }
                        }, {
                            "id": "6",
                            "name": "Party Status Updated",
                            "userDefinable": true,
                            "type": "partyStatusUpdated",
                            "category": "partyStatusBased",
                            "actions": [{
                                    "type": "notification"
                                }, {
                                    "type": "push"
                                }],
                            "digestEnabled": false,
                            "recipeDefaults": {
                                "actions": [{
                                        "type": "notification"
                                    }, {
                                        "type": "push"
                                    }]
                            }
                        }, {
                            "id": "8",
                            "name": "Saving goal",
                            "type": "savingGoal",
                            "category": "userBased",
                            "userDefinable": true,
                            "actions": [{
                                    "type": "notification"
                                }, {
                                    "type": "push"
                                }],
                            "digestEnabled": false
                        }, {
                            "id": "9",
                            "name": "Account statement is ready",
                            "type": "accountStatementIsReady",
                            "category": "arrangementBased",
                            "userDefinable": true,
                            "actions": [{
                                    "type": "email"
                                }, {
                                    "type": "notification"
                                }, {
                                    "type": "push"
                                }],
                            "digestEnabled": false
                        }, {
                            "id": "10",
                            "name": "Account Low Balance Business",
                            "type": "accountLowBalance",
                            "category": "arrangementBased",
                            "userDefinable": true,
                            "actions": [{
                                    "type": "notification"
                                }, {
                                    "type": "push"
                                }],
                            "digestEnabled": false,
                            "recipeDefaults": {
                                "amount": "0",
                                "actions": [{
                                        "type": "notification"
                                    }, {
                                        "type": "push"
                                    }]
                            }
                        }, {
                            "id": "13",
                            "name": "Payment to approve",
                            "type": "paymentToApprove",
                            "category": "paymentResourceBased",
                            "digestEnabled": false,
                            "recipeDefaults": {
                                "actions": [{
                                        "type": "notification"
                                    }, {
                                        "type": "push"
                                    }],
                                "additions": {
                                    "resource": "Payments"
                                }
                            }
                        }, {
                            "id": "14",
                            "name": "Contact to approve",
                            "type": "contactToApprove",
                            "category": "contactResourceBased",
                            "digestEnabled": false,
                            "recipeDefaults": {
                                "actions": [{
                                        "type": "notification"
                                    }, {
                                        "type": "push"
                                    }],
                                "additions": {
                                    "resource": "Contacts"
                                }
                            }
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
        ]
    },
]);
