// @backbase/actions-http-ang/esm2015/api/actionRecipes.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const ActionRecipesHttpServiceDeleteActionRecipeMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/action-recipes/{recipeId}",
        method: "DELETE",
        responses: [
            {
                status: 202,
                body: {}
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    }]);
export const ActionRecipesHttpServiceGetActionRecipeMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/action-recipes/{recipeId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "1",
                    "name": "Transaction > 100 EUR",
                    "active": true,
                    "specificationId": "1",
                    "userId": "kjordan",
                    "actions": [{
                            "type": "notification",
                            "enabled": true
                        }, {
                            "type": "email",
                            "enabled": false
                        }, {
                            "type": "sms",
                            "enabled": false
                        }, {
                            "type": "push",
                            "enabled": true
                        }],
                    "amount": "100",
                    "arrangementId": "123"
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    }]);
export const ActionRecipesHttpServiceGetActionRecipesMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/action-recipes",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "totalCount": 20,
                    "actionRecipes": [{
                            "id": "1",
                            "name": "Transaction > 100 EUR",
                            "active": true,
                            "specificationId": "1",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "100",
                            "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4691",
                            "createdOn": "2020-01-17T12:52:34.000+0000"
                        }, {
                            "id": "2",
                            "name": "Account Low Balance < 1000 EUR",
                            "active": true,
                            "specificationId": "4",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "300",
                            "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4691",
                            "createdOn": "2020-01-17T12:52:34.000+0000"
                        }, {
                            "id": "3",
                            "name": "Transaction > 300 EUR",
                            "active": true,
                            "specificationId": "1",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "300",
                            "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4600",
                            "createdOn": "2020-01-17T12:52:34.000+0000"
                        }, {
                            "id": "4",
                            "name": "Account Low Balance < 1000 EUR",
                            "active": true,
                            "specificationId": "4",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "300",
                            "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4692",
                            "createdOn": "2020-01-17T12:52:34.000+0000"
                        }, {
                            "id": "5",
                            "name": "Transaction > 100 EUR",
                            "active": true,
                            "specificationId": "1",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "100",
                            "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4600",
                            "createdOn": "2020-01-17T12:52:34.000+0000"
                        }, {
                            "id": "6",
                            "name": "Rounding rule",
                            "active": false,
                            "specificationId": "3",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "payment",
                                    "enabled": true
                                }],
                            "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4691",
                            "createdOn": "2020-01-17T12:52:34.000+0000",
                            "additions": {
                                "targetArrangementId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699"
                            }
                        }, {
                            "id": "7",
                            "name": "Account Low Balance < 1000 EUR",
                            "active": true,
                            "specificationId": "4",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "300",
                            "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4692",
                            "createdOn": "2020-01-17T12:52:34.000+0000"
                        }, {
                            "id": "8",
                            "name": "Transaction > 100 EUR",
                            "active": true,
                            "specificationId": "1",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "100",
                            "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                            "createdOn": "2020-01-17T12:52:34.000+0000"
                        }, {
                            "id": "9",
                            "name": "Account Low Balance < 1000 EUR",
                            "active": true,
                            "specificationId": "4",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "300",
                            "arrangementId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                            "createdOn": "2020-01-17T12:52:34.000+0000"
                        }, {
                            "id": "10",
                            "name": "Transaction > 100 EUR",
                            "active": true,
                            "specificationId": "1",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "100",
                            "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                            "createdOn": "2020-01-17T12:52:34.000+0000"
                        }, {
                            "id": "11",
                            "name": "Account Low Balance < 1000 EUR",
                            "active": true,
                            "specificationId": "4",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "300",
                            "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                            "createdOn": "2020-01-16T12:52:34.000+0000"
                        }, {
                            "id": "12",
                            "name": "Transaction > 100 EUR",
                            "active": true,
                            "specificationId": "1",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "100",
                            "arrangementId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                            "createdOn": "2020-01-16T12:52:34.000+0000"
                        }, {
                            "id": "13",
                            "name": "Account Low Balance < 1000 EUR",
                            "active": true,
                            "specificationId": "4",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "300",
                            "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                            "createdOn": "2020-01-16T12:52:34.000+0000"
                        }, {
                            "id": "14",
                            "name": "Transaction > 100 EUR",
                            "active": true,
                            "specificationId": "1",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "100",
                            "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                            "createdOn": "2020-01-16T12:52:34.000+0000"
                        }, {
                            "id": "15",
                            "name": "Account Low Balance < 1000 EUR",
                            "active": true,
                            "specificationId": "4",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "300",
                            "arrangementId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                            "createdOn": "2020-01-16T12:52:34.000+0000"
                        }, {
                            "id": "16",
                            "name": "Rounding rule Example",
                            "active": true,
                            "specificationId": "3",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "payment",
                                    "enabled": true
                                }],
                            "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4600",
                            "createdOn": "2020-01-16T12:52:34.000+0000",
                            "additions": {
                                "targetArrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699"
                            }
                        }, {
                            "id": "17",
                            "name": "Payment Status Rejected",
                            "active": false,
                            "specificationId": "5",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "createdOn": "2020-01-16T12:52:34.000+0000",
                            "additions": {
                                "paymentStatuses": "rejected"
                            }
                        }, {
                            "id": "18",
                            "name": "Payment Status Rejected",
                            "active": true,
                            "specificationId": "5",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "createdOn": "2020-01-16T12:52:34.000+0000",
                            "additions": {
                                "paymentStatuses": "rejected"
                            }
                        }, {
                            "id": "19",
                            "name": "Contact Status Rejected",
                            "active": false,
                            "specificationId": "6",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "createdOn": "2020-01-16T12:52:34.000+0000",
                            "additions": {
                                "partyStatuses": "rejected"
                            }
                        }, {
                            "id": "20",
                            "name": "Contact Status Rejected",
                            "active": true,
                            "specificationId": "6",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "createdOn": "2020-01-16T12:52:34.000+0000",
                            "additions": {
                                "partyStatuses": "rejected"
                            }
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
        ]
    }]);
export const ActionRecipesHttpServiceGetActionsHistoryMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/action-recipes/history",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "actionEventsHistory": [{
                            "alertName": "Low balance alert",
                            "deliveryMethod": "sms",
                            "alertDate": "2019-12-16T19:35:05.000+0000",
                            "alertText": "Warning! Your account balance is below 20 EUR.",
                            "destinations": ["+380987654321", "+380123456789"]
                        }, {
                            "alertName": "Low balance alert",
                            "deliveryMethod": "email",
                            "alertDate": "2019-12-16T19:25:05.000+0000",
                            "alertText": "Warning! Your account balance is below 20 EUR.",
                            "destinations": ["user@domain.com", "resu@domain.com"]
                        }, {
                            "alertName": "Low balance alert",
                            "deliveryMethod": "notification",
                            "alertDate": "2019-12-16T19:15:05.000+0000",
                            "alertText": "Warning! Your account balance is below 20 EUR.",
                            "destinations": ["User id - 'userId'"]
                        }, {
                            "alertName": "Party status updated alert",
                            "deliveryMethod": "notification",
                            "alertDate": "2019-12-16T19:10:05.000+0000",
                            "alertText": "Your contact 'Sarah' is active. You can find this in 'My Requests'",
                            "destinations": ["User id - 'Sarah', Service agreement id - '2b5b4bff-4135-4d18-b3fc-eb86d647dccd'"]
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
        ]
    }]);
export const ActionRecipesHttpServiceGetActionsHistoryByUserMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/action-recipes/history/{userId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "actionEventsHistory": [{
                            "alertName": "Low balance alert",
                            "deliveryMethod": "sms",
                            "alertDate": "2019-12-16T19:35:05.000+0000",
                            "alertText": "Warning! Your account balance is below 20 EUR.",
                            "destinations": ["+380987654321", "+380123456789"]
                        }, {
                            "alertName": "Low balance alert",
                            "deliveryMethod": "email",
                            "alertDate": "2019-12-16T19:25:05.000+0000",
                            "alertText": "Warning! Your account balance is below 20 EUR.",
                            "destinations": ["user@domain.com", "resu@domain.com"]
                        }, {
                            "alertName": "Low balance alert",
                            "deliveryMethod": "notification",
                            "alertDate": "2019-12-16T19:15:05.000+0000",
                            "alertText": "Warning! Your account balance is below 20 EUR.",
                            "destinations": ["User id - 'userId'"]
                        }, {
                            "alertName": "Party status updated alert",
                            "deliveryMethod": "notification",
                            "alertDate": "2019-12-16T19:10:05.000+0000",
                            "alertText": "Your contact 'Sarah' is active. You can find this in 'My Requests'",
                            "destinations": ["User id - 'Sarah', Service agreement id - '2b5b4bff-4135-4d18-b3fc-eb86d647dccd'"]
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    }]);
export const ActionRecipesHttpServiceGetPaymentStatusesMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/action-recipes/payment-statuses",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "paymentStatuses": ["DRAFT", "ENTERED", "READY", "ACCEPTED", "PROCESSED", "REJECTED", "CANCELLED"]
                }
            },
        ]
    }]);
export const ActionRecipesHttpServiceGetPaymentTypesMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/action-recipes/payment-types",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "paymentTypes": ["SEPA CT", "US Domestic Wire", "US Foreign Wire"]
                }
            },
        ]
    }]);
export const ActionRecipesHttpServicePostActionRecipesMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/action-recipes",
        method: "POST",
        responses: [
            {
                status: 202,
                body: {
                    "id": "e92bbe7d-10cf-4d7c-b7b5-eadb7632812a"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
        ]
    }]);
export const ActionRecipesHttpServicePostActivationRequestMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/action-recipes/{recipeId}/activation-request",
        method: "POST",
        responses: [
            {
                status: 202,
                body: {}
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    }]);
export const ActionRecipesHttpServicePostDeactivationRequestMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/action-recipes/{recipeId}/deactivation-request",
        method: "POST",
        responses: [
            {
                status: 202,
                body: {}
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    }]);
export const ActionRecipesHttpServicePutActionRecipeMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/action-recipes/{recipeId}",
        method: "PUT",
        responses: [
            {
                status: 202,
                body: {}
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    }]);
export const ActionRecipesHttpServicePutBatchProcessingMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/action-recipes/batch",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {
                    "actionRecipes": [{
                            "id": "1",
                            "name": "TransactionExecuted for User",
                            "active": true,
                            "specificationId": "1",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }],
                            "amount": "100",
                            "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                            "createdOn": "2020-03-18T11:36:02.000+0000",
                            "recurrence": {
                                "daysOfWeek": ["MON", "FRI"],
                                "hoursOfDay": ["9", "17"],
                                "quartzExpression": "0 0 9,17 ? * MON,FRI *"
                            }
                        }, {
                            "id": "2",
                            "name": "Account Low Balance < 1000 EUR",
                            "active": true,
                            "specificationId": "4",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": false
                                }, {
                                    "type": "email",
                                    "enabled": true
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }],
                            "amount": "300",
                            "arrangementId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                            "createdOn": "2020-03-18T11:36:02.000+0000"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    }]);
export const ActionRecipesHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v2/action-recipes/{recipeId}",
        method: "DELETE",
        responses: [
            {
                status: 202,
                body: {}
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/action-recipes/{recipeId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "1",
                    "name": "Transaction > 100 EUR",
                    "active": true,
                    "specificationId": "1",
                    "userId": "kjordan",
                    "actions": [{
                            "type": "notification",
                            "enabled": true
                        }, {
                            "type": "email",
                            "enabled": false
                        }, {
                            "type": "sms",
                            "enabled": false
                        }, {
                            "type": "push",
                            "enabled": true
                        }],
                    "amount": "100",
                    "arrangementId": "123"
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/action-recipes",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "totalCount": 20,
                    "actionRecipes": [{
                            "id": "1",
                            "name": "Transaction > 100 EUR",
                            "active": true,
                            "specificationId": "1",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "100",
                            "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4691",
                            "createdOn": "2020-01-17T12:52:34.000+0000"
                        }, {
                            "id": "2",
                            "name": "Account Low Balance < 1000 EUR",
                            "active": true,
                            "specificationId": "4",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "300",
                            "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4691",
                            "createdOn": "2020-01-17T12:52:34.000+0000"
                        }, {
                            "id": "3",
                            "name": "Transaction > 300 EUR",
                            "active": true,
                            "specificationId": "1",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "300",
                            "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4600",
                            "createdOn": "2020-01-17T12:52:34.000+0000"
                        }, {
                            "id": "4",
                            "name": "Account Low Balance < 1000 EUR",
                            "active": true,
                            "specificationId": "4",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "300",
                            "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4692",
                            "createdOn": "2020-01-17T12:52:34.000+0000"
                        }, {
                            "id": "5",
                            "name": "Transaction > 100 EUR",
                            "active": true,
                            "specificationId": "1",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "100",
                            "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4600",
                            "createdOn": "2020-01-17T12:52:34.000+0000"
                        }, {
                            "id": "6",
                            "name": "Rounding rule",
                            "active": false,
                            "specificationId": "3",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "payment",
                                    "enabled": true
                                }],
                            "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4691",
                            "createdOn": "2020-01-17T12:52:34.000+0000",
                            "additions": {
                                "targetArrangementId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699"
                            }
                        }, {
                            "id": "7",
                            "name": "Account Low Balance < 1000 EUR",
                            "active": true,
                            "specificationId": "4",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "300",
                            "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4692",
                            "createdOn": "2020-01-17T12:52:34.000+0000"
                        }, {
                            "id": "8",
                            "name": "Transaction > 100 EUR",
                            "active": true,
                            "specificationId": "1",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "100",
                            "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                            "createdOn": "2020-01-17T12:52:34.000+0000"
                        }, {
                            "id": "9",
                            "name": "Account Low Balance < 1000 EUR",
                            "active": true,
                            "specificationId": "4",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "300",
                            "arrangementId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                            "createdOn": "2020-01-17T12:52:34.000+0000"
                        }, {
                            "id": "10",
                            "name": "Transaction > 100 EUR",
                            "active": true,
                            "specificationId": "1",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "100",
                            "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                            "createdOn": "2020-01-17T12:52:34.000+0000"
                        }, {
                            "id": "11",
                            "name": "Account Low Balance < 1000 EUR",
                            "active": true,
                            "specificationId": "4",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "300",
                            "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                            "createdOn": "2020-01-16T12:52:34.000+0000"
                        }, {
                            "id": "12",
                            "name": "Transaction > 100 EUR",
                            "active": true,
                            "specificationId": "1",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "100",
                            "arrangementId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                            "createdOn": "2020-01-16T12:52:34.000+0000"
                        }, {
                            "id": "13",
                            "name": "Account Low Balance < 1000 EUR",
                            "active": true,
                            "specificationId": "4",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "300",
                            "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                            "createdOn": "2020-01-16T12:52:34.000+0000"
                        }, {
                            "id": "14",
                            "name": "Transaction > 100 EUR",
                            "active": true,
                            "specificationId": "1",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "100",
                            "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                            "createdOn": "2020-01-16T12:52:34.000+0000"
                        }, {
                            "id": "15",
                            "name": "Account Low Balance < 1000 EUR",
                            "active": true,
                            "specificationId": "4",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "amount": "300",
                            "arrangementId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                            "createdOn": "2020-01-16T12:52:34.000+0000"
                        }, {
                            "id": "16",
                            "name": "Rounding rule Example",
                            "active": true,
                            "specificationId": "3",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "payment",
                                    "enabled": true
                                }],
                            "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4600",
                            "createdOn": "2020-01-16T12:52:34.000+0000",
                            "additions": {
                                "targetArrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699"
                            }
                        }, {
                            "id": "17",
                            "name": "Payment Status Rejected",
                            "active": false,
                            "specificationId": "5",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "createdOn": "2020-01-16T12:52:34.000+0000",
                            "additions": {
                                "paymentStatuses": "rejected"
                            }
                        }, {
                            "id": "18",
                            "name": "Payment Status Rejected",
                            "active": true,
                            "specificationId": "5",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "createdOn": "2020-01-16T12:52:34.000+0000",
                            "additions": {
                                "paymentStatuses": "rejected"
                            }
                        }, {
                            "id": "19",
                            "name": "Contact Status Rejected",
                            "active": false,
                            "specificationId": "6",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": false
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "createdOn": "2020-01-16T12:52:34.000+0000",
                            "additions": {
                                "partyStatuses": "rejected"
                            }
                        }, {
                            "id": "20",
                            "name": "Contact Status Rejected",
                            "active": true,
                            "specificationId": "6",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "push",
                                    "enabled": true
                                }],
                            "createdOn": "2020-01-16T12:52:34.000+0000",
                            "additions": {
                                "partyStatuses": "rejected"
                            }
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/action-recipes/history",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "actionEventsHistory": [{
                            "alertName": "Low balance alert",
                            "deliveryMethod": "sms",
                            "alertDate": "2019-12-16T19:35:05.000+0000",
                            "alertText": "Warning! Your account balance is below 20 EUR.",
                            "destinations": ["+380987654321", "+380123456789"]
                        }, {
                            "alertName": "Low balance alert",
                            "deliveryMethod": "email",
                            "alertDate": "2019-12-16T19:25:05.000+0000",
                            "alertText": "Warning! Your account balance is below 20 EUR.",
                            "destinations": ["user@domain.com", "resu@domain.com"]
                        }, {
                            "alertName": "Low balance alert",
                            "deliveryMethod": "notification",
                            "alertDate": "2019-12-16T19:15:05.000+0000",
                            "alertText": "Warning! Your account balance is below 20 EUR.",
                            "destinations": ["User id - 'userId'"]
                        }, {
                            "alertName": "Party status updated alert",
                            "deliveryMethod": "notification",
                            "alertDate": "2019-12-16T19:10:05.000+0000",
                            "alertText": "Your contact 'Sarah' is active. You can find this in 'My Requests'",
                            "destinations": ["User id - 'Sarah', Service agreement id - '2b5b4bff-4135-4d18-b3fc-eb86d647dccd'"]
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/action-recipes/history/{userId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "actionEventsHistory": [{
                            "alertName": "Low balance alert",
                            "deliveryMethod": "sms",
                            "alertDate": "2019-12-16T19:35:05.000+0000",
                            "alertText": "Warning! Your account balance is below 20 EUR.",
                            "destinations": ["+380987654321", "+380123456789"]
                        }, {
                            "alertName": "Low balance alert",
                            "deliveryMethod": "email",
                            "alertDate": "2019-12-16T19:25:05.000+0000",
                            "alertText": "Warning! Your account balance is below 20 EUR.",
                            "destinations": ["user@domain.com", "resu@domain.com"]
                        }, {
                            "alertName": "Low balance alert",
                            "deliveryMethod": "notification",
                            "alertDate": "2019-12-16T19:15:05.000+0000",
                            "alertText": "Warning! Your account balance is below 20 EUR.",
                            "destinations": ["User id - 'userId'"]
                        }, {
                            "alertName": "Party status updated alert",
                            "deliveryMethod": "notification",
                            "alertDate": "2019-12-16T19:10:05.000+0000",
                            "alertText": "Your contact 'Sarah' is active. You can find this in 'My Requests'",
                            "destinations": ["User id - 'Sarah', Service agreement id - '2b5b4bff-4135-4d18-b3fc-eb86d647dccd'"]
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/action-recipes/payment-statuses",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "paymentStatuses": ["DRAFT", "ENTERED", "READY", "ACCEPTED", "PROCESSED", "REJECTED", "CANCELLED"]
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/action-recipes/payment-types",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "paymentTypes": ["SEPA CT", "US Domestic Wire", "US Foreign Wire"]
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/action-recipes",
        method: "POST",
        responses: [
            {
                status: 202,
                body: {
                    "id": "e92bbe7d-10cf-4d7c-b7b5-eadb7632812a"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/action-recipes/{recipeId}/activation-request",
        method: "POST",
        responses: [
            {
                status: 202,
                body: {}
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/action-recipes/{recipeId}/deactivation-request",
        method: "POST",
        responses: [
            {
                status: 202,
                body: {}
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/action-recipes/{recipeId}",
        method: "PUT",
        responses: [
            {
                status: 202,
                body: {}
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/action-recipes/batch",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {
                    "actionRecipes": [{
                            "id": "1",
                            "name": "TransactionExecuted for User",
                            "active": true,
                            "specificationId": "1",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": true
                                }, {
                                    "type": "email",
                                    "enabled": false
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }],
                            "amount": "100",
                            "arrangementId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                            "createdOn": "2020-03-18T11:36:02.000+0000",
                            "recurrence": {
                                "daysOfWeek": ["MON", "FRI"],
                                "hoursOfDay": ["9", "17"],
                                "quartzExpression": "0 0 9,17 ? * MON,FRI *"
                            }
                        }, {
                            "id": "2",
                            "name": "Account Low Balance < 1000 EUR",
                            "active": true,
                            "specificationId": "4",
                            "userId": "kjordan",
                            "actions": [{
                                    "type": "notification",
                                    "enabled": false
                                }, {
                                    "type": "email",
                                    "enabled": true
                                }, {
                                    "type": "sms",
                                    "enabled": false
                                }],
                            "amount": "300",
                            "arrangementId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                            "createdOn": "2020-03-18T11:36:02.000+0000"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    },
]);
