// @backbase/card-details-widget-ang/esm2015/src/components/card-details-container/card-details-container.component.js
import { Component, Input, Optional } from '@angular/core';
import { CardService, filterInputParameters } from '@backbase/cards-common-ang';
import { BehaviorSubject, of } from 'rxjs';
import { catchError, tap } from 'rxjs/operators';
import { WidgetPropertiesService } from '../../services/widget-properties.service';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/cards-common-ang";
import * as i2 from "../../services/widget-properties.service";
import * as i3 from "../bb-payment-card/payment-card.component";
import * as i4 from "@backbase/ui-ang/empty-state";
import * as i5 from "@backbase/ui-ang/loading-indicator";
import * as i6 from "@angular/common";
export class CardDetailsContainerComponent {
    constructor(cardService, cd, widgetPropertiesService) {
        this.cardService = cardService;
        this.cd = cd;
        this.widgetPropertiesService = widgetPropertiesService;
        this.notificationTtlSubject = new BehaviorSubject(undefined);
        this.updateAllLimitsSubject = new BehaviorSubject(undefined);
        this.handleError = (errorResponse) => {
            this.errors = errorResponse;
            return of(undefined);
        };
    }
    /**
     * Notification timeout timer value configuration
     */
    // eslint-disable-next-line
    set notificationTtlValue(value) {
        this.notificationTtlSubject.next(value);
    }
    /**
     * Enable to send multiple limits to the API when updating values
     */
    set updateAllLimitsValue(value) {
        this.updateAllLimitsSubject.next(value);
    }
    ngOnInit() {
        if (typeof this.paymentCardId === 'undefined') {
            throw new Error(`"paymentCardId" input is required in "${this.constructor.name}"`);
        }
        this.cardService.getPaymentCardFrom(this.paymentCardId);
        this.paymentCard = this.cardService.getPaymentCard().pipe(tap(() => {
            this.cd.markForCheck();
        }), catchError(this.handleError));
    }
    get notificationTtl() {
        return filterInputParameters(this.notificationTtlSubject, this.widgetPropertiesService.notificationTtl);
    }
    get updateAllLimits() {
        return filterInputParameters(this.updateAllLimitsSubject, this.widgetPropertiesService.updateAllLimits);
    }
}
CardDetailsContainerComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardDetailsContainerComponent, deps: [{ token: i1.CardService }, { token: i0.ChangeDetectorRef }, { token: i2.WidgetPropertiesService, optional: true }], target: i0.ɵɵFactoryTarget.Component });
CardDetailsContainerComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: CardDetailsContainerComponent, selector: "bb-card-details-container", inputs: { paymentCardId: "paymentCardId", notificationTtlValue: ["notificationTtl", "notificationTtlValue"], updateAllLimitsValue: ["updateAllLimits", "updateAllLimitsValue"] }, providers: [CardService, WidgetPropertiesService], ngImport: i0, template: "<ng-container *ngIf=\"paymentCardId | async; else emptyState\">\n  <ng-container *ngIf=\"!errors; else errorState\">\n    <ng-container *ngIf=\"paymentCard | async as paymentCard; else loadingState\">\n      <bb-payment-card \n        [paymentCard]=\"paymentCard\" \n        [notificationTtl]=\"(notificationTtl | async) || undefined\"\n        [updateAllLimits]=\"(updateAllLimits | async) || undefined\"\n      ></bb-payment-card>\n    </ng-container>\n  </ng-container>\n</ng-container>\n\n<ng-template #emptyState>\n  <bb-empty-state-ui title=\"No Card Selected\"></bb-empty-state-ui>\n</ng-template>\n\n<ng-template #loadingState>\n  <bb-loading-indicator-ui loaderSize=\"lg\"></bb-loading-indicator-ui>\n</ng-template>\n\n<ng-template #errorState>\n  <bb-payment-card-multiple-error-state [errors]=\"errors\"></bb-payment-card-multiple-error-state>\n</ng-template>\n", components: [{ type: i3.PaymentCardComponent, selector: "bb-payment-card" }, { type: i4.EmptyStateComponent, selector: "bb-empty-state-ui", inputs: ["title", "subtitle", "showIcon", "iconClasses", "iconModifier", "iconSize", "iconColor"] }, { type: i5.LoadingIndicatorComponent, selector: "bb-loading-indicator-ui", inputs: ["text", "loaderSize", "showDelay", "hasBackground", "inline"] }, { type: i1.PaymentCardMultipleErrorStateComponent, selector: "bb-payment-card-multiple-error-state", inputs: ["errors"] }], directives: [{ type: i6.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "async": i6.AsyncPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardDetailsContainerComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-card-details-container',
                    templateUrl: './card-details-container.component.html',
                    providers: [CardService, WidgetPropertiesService],
                }]
        }], ctorParameters: function () { return [{ type: i1.CardService }, { type: i0.ChangeDetectorRef }, { type: i2.WidgetPropertiesService, decorators: [{
                    type: Optional
                }] }]; }, propDecorators: { paymentCardId: [{
                type: Input
            }], notificationTtlValue: [{
                type: Input,
                args: ['notificationTtl']
            }], updateAllLimitsValue: [{
                type: Input,
                args: ['updateAllLimits']
            }] } });
