// @backbase/ui-ang/esm2015/_example/notification/examples/notification.example.component.js
import { Component, ViewChild } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/notification";
import * as i2 from "@backbase/ui-ang/button";
const LOAD_DELAY = 500;
export class NotificationExampleComponent {
    constructor(notificationService) {
        this.notificationService = notificationService;
    }
    checkMail() {
        setTimeout(() => this.showNewMailNotification(), LOAD_DELAY);
    }
    checkMailTemplate() {
        setTimeout(() => this.showNewMailNotificationWithTemplateMesage(), LOAD_DELAY);
    }
    showNewMailNotification() {
        this.notificationService.showNotification({
            header: 'You\'ve got new mail!',
            message: 'Backbase Release 1.2.3 is now available! What’s new in this release? ...',
            modifier: 'info',
            dismissible: true,
            primaryActionText: 'Visit website',
            primaryAction: () => {
                window.location.href = 'https://backbase.com/';
            },
        });
    }
    showNewMailNotificationWithTemplateMesage() {
        this.notificationService.showNotification({
            header: 'You\'ve got new mail!',
            message: this.messageTemplate,
            modifier: 'info',
            dismissible: true,
            primaryActionText: 'Visit website',
            primaryAction: () => {
                window.location.href = 'https://backbase.com/';
            },
        });
    }
}
NotificationExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationExampleComponent, deps: [{ token: i1.NotificationService }], target: i0.ɵɵFactoryTarget.Component });
NotificationExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: NotificationExampleComponent, selector: "bb-notification-example-ui", viewQueries: [{ propertyName: "messageTemplate", first: true, predicate: ["messageTemplate"], descendants: true, static: true }], ngImport: i0, template: "<button\n  type=\"button\"\n  (click)=\"checkMail()\"\n  bbButton\n  [buttonSize]=\"'md'\"\n  [color]=\"'primary'\"\n>\n  Check for new mails\n</button>\n\n<button\n  type=\"button\"\n  (click)=\"checkMailTemplate()\"\n  bbButton\n  [buttonSize]=\"'md'\"\n  [color]=\"'primary'\"\n>\n  Check for new mails (message template)\n</button>\n\n<ng-template #messageTemplate>\n  <p>Backbase Release <strong>3.8.0</strong> is now available!</p>\n</ng-template>\n", directives: [{ type: i2.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-notification-example-ui',
                    templateUrl: './notification.example.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.NotificationService }]; }, propDecorators: { messageTemplate: [{
                type: ViewChild,
                args: ['messageTemplate', { static: true }]
            }] } });
