// @backbase/foundation-ang/esm2015/core/backbase-core/communication/pipes/pipeline.service.js
import { Injectable, Injector } from '@angular/core';
import { PipelineRegistry } from './pipeline-registry';
import { takeUntil } from 'rxjs/operators';
import { pipeFromArray } from '../../models/observable';
const noop = () => { };
const ɵ0 = noop;
const noopSink = (stdin) => {
    stdin.subscribe(noop);
};
const ɵ1 = noopSink;
const isSourceDefinition = (definition) => definition && 'source' in definition;
const ɵ2 = isSourceDefinition;
const isPipeDefinition = (definition) => definition && 'pipe' in definition;
const ɵ3 = isPipeDefinition;
const isSinkDefinition = (definition) => definition && 'sink' in definition;
const ɵ4 = isSinkDefinition;
/**
 * @deprecated Will be removed in v7.0.0
 */
export class PipelineService {
    constructor(broker, injector) {
        this.broker = broker;
        this.injector = injector;
    }
    createSource(name, args) {
        const instance = this.getInstance(name);
        if (!isSourceDefinition(instance)) {
            throw new Error(`${name} is not a valid source`);
        }
        return instance.source(...args);
    }
    createPipe(name, args) {
        const instance = this.getInstance(name);
        if (!isPipeDefinition(instance)) {
            throw new Error(`${name} is not a valid pipe`);
        }
        return instance.pipe(...args);
    }
    createSink(name, args) {
        const instance = this.getInstance(name);
        if (isSinkDefinition(instance)) {
            return instance.sink(...args);
        }
        else if (isPipeDefinition(instance)) {
            return this.chainSink([instance.pipe(...args)], noopSink);
        }
        throw new Error(`${name} is not a valid sink`);
    }
    getInstance(name) {
        const provide = this.broker.getProcessorProviderToken(name);
        return this.injector.get(provide);
    }
    chainSink(pipes, sink) {
        return (stdin) => {
            sink(stdin.pipe(pipeFromArray(pipes)));
        };
    }
    runPipeline(pipeline, gc) {
        return pipeline.sink(pipeline.source().pipe(takeUntil(gc)));
    }
}
PipelineService.decorators = [
    { type: Injectable }
];
PipelineService.ctorParameters = () => [
    { type: PipelineRegistry },
    { type: Injector }
];
export { ɵ0, ɵ1, ɵ2, ɵ3, ɵ4 };
