// @backbase/foundation-ang/esm2015/core/backbase-core/communication/pipes/pipeline-registry.js
import { InjectionToken, Inject, Injectable } from '@angular/core';
import { flatten } from '../../models/array';
import * as i0 from "@angular/core";
/**
 * @deprecated Will be removed in v7.0.0
 */
export const CONFIG_PIPELINE_PROCESSORS = new InjectionToken('@backbase/foundation Pipeline Handlers');
/**
 * @deprecated Will be removed in v7.0.0
 * Helper service wrapping around CONFIG_PIPELINE_PROCESSORS
 */
export class PipelineRegistry {
    constructor(processorsNested) {
        const allProcessors = flatten(processorsNested);
        this.registeredHandlers = new Map(allProcessors.map(({ name, provider }) => [name, provider]));
    }
    getProcessorProvider(name) {
        const provider = this.registeredHandlers.get(name);
        if (provider === undefined) {
            throw new Error(`No provider to pipeline handler ${name}`);
        }
        return Object.assign({ deps: [] }, provider);
    }
    getProcessorProviderToken(name) {
        return this.getProcessorProvider(name).provide;
    }
}
PipelineRegistry.ɵprov = i0.ɵɵdefineInjectable({ factory: function PipelineRegistry_Factory() { return new PipelineRegistry(i0.ɵɵinject(CONFIG_PIPELINE_PROCESSORS)); }, token: PipelineRegistry, providedIn: "root" });
PipelineRegistry.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root',
            },] }
];
PipelineRegistry.ctorParameters = () => [
    { type: Array, decorators: [{ type: Inject, args: [CONFIG_PIPELINE_PROCESSORS,] }] }
];
/**
 * @deprecated Will be removed in v7.0.0
 */
export function createPipelineProcessorConfig(name, provider) {
    return {
        name,
        provider,
    };
}
