// @backbase/product-summary-common-ang/esm2015/src/services/widget-properties/product-summary-widget-properties.service.js
import { of, combineLatest, BehaviorSubject } from 'rxjs';
import { map } from 'rxjs/operators';
import { Inject, Injectable, Optional } from '@angular/core';
import { PRODUCT_SUMMARY_WIDGET_PROPERTIES } from '../../model/providers';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
/** @dynamic */
export class ProductSummaryWidgetPropertiesService {
    /**
     * @internal
     * @param defaultConfiguration
     * @param itemModel
     */
    constructor(defaultConfiguration, itemModel) {
        this.defaultConfiguration = defaultConfiguration;
        this.itemModel = itemModel;
        /**
         * Widget properties configuration
         */
        this.configuration = new BehaviorSubject({});
        /**
         * Gets a property value for each value in `propertyNames` input,
         * joins them into `WidgetProperties` structure
         */
        const propertyNames = Object.keys(defaultConfiguration);
        this.properties = combineLatest(propertyNames.map(propertyName => this.getWidgetProperty(propertyName))).pipe(map(properties => propertyNames.reduce((widgetProperties, propertyName, index) => (Object.assign(Object.assign({}, widgetProperties), { [propertyName]: properties[index] })), {})));
    }
    /**
     * Sets widget properties
     *
     * @param config
     */
    setConfiguration(config) {
        this.configuration.next(config);
    }
    /**
     * Returns observable with a widget property
     *
     * @param propertyName
     */
    getProperty(propertyName) {
        return this.properties.pipe(map(properties => properties[propertyName]));
    }
    /**
     * Returns a widget configuration property,
     * takes the value in the following order:
     * - configuration value if exists (widget input)
     * - item-model value if exists (CXP configuration)
     * - default value (constant value)
     */
    getWidgetProperty(propertyName) {
        const defaultValue = this.defaultConfiguration[propertyName];
        const configurationValue$ = this.configuration.pipe(map(configuration => configuration[propertyName]));
        const modelValue$ = this.itemModel
            ? this.itemModel.property(propertyName, defaultValue)
            : of(defaultValue);
        return combineLatest([configurationValue$, modelValue$]).pipe(map(([configurationValue, modelValue]) => (configurationValue !== undefined ? configurationValue : modelValue)));
    }
}
ProductSummaryWidgetPropertiesService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ProductSummaryWidgetPropertiesService, deps: [{ token: PRODUCT_SUMMARY_WIDGET_PROPERTIES }, { token: i1.ItemModel, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
ProductSummaryWidgetPropertiesService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ProductSummaryWidgetPropertiesService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ProductSummaryWidgetPropertiesService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [PRODUCT_SUMMARY_WIDGET_PROPERTIES]
                }] }, { type: i1.ItemModel, decorators: [{
                    type: Optional
                }] }]; } });
