// @backbase/ui-ang/esm2015/_example/product-selector/showcase/product-selector-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ProductSelectorShowcaseComponent } from './product-selector-showcase.component';
import { ProductSelectorModule } from '@backbase/ui-ang/product-selector';
import { ProductItemBasicAccountModule } from '@backbase/ui-ang/product-item-basic-account';
import { AmountModule } from '@backbase/ui-ang/amount';
import { ButtonModule } from '@backbase/ui-ang/button';
import * as i0 from "@angular/core";
export class ProductSelectorShowcaseModule {
}
ProductSelectorShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSelectorShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductSelectorShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSelectorShowcaseModule, declarations: [ProductSelectorShowcaseComponent], imports: [CommonModule, ProductSelectorModule, ProductItemBasicAccountModule, AmountModule, ButtonModule] });
ProductSelectorShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSelectorShowcaseModule, imports: [[CommonModule, ProductSelectorModule, ProductItemBasicAccountModule, AmountModule, ButtonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSelectorShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ProductSelectorModule, ProductItemBasicAccountModule, AmountModule, ButtonModule],
                    declarations: [ProductSelectorShowcaseComponent],
                    entryComponents: [ProductSelectorShowcaseComponent],
                }]
        }] });
