// @backbase/product-summary-ui/esm2015/src/bb-account-picker-ui/account-picker.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { ProductItemBasicAccountModule } from '@backbase/ui-ang/product-item-basic-account';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import { SearchBoxModule } from '@backbase/ui-ang/search-box';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { InputCheckboxModule } from '@backbase/ui-ang/input-checkbox';
import { ButtonModule } from '@backbase/ui-ang/button';
import { NgSelectModule } from '@ng-select/ng-select';
import { AccountPickerComponent, CustomSingleSelectedItemLabelDirective, CustomMultiSelectedItemsLabelDirective, CustomOptionsHeaderDirective, CustomOptionItemDirective, CustomGroupItemsHeaderDirective, CustomLoadingTemplateDirective, } from './account-picker-ui.component';
import * as i0 from "@angular/core";
const uiModules = [
    ProductItemBasicAccountModule,
    InputTextModule,
    SearchBoxModule,
    LoadingIndicatorModule,
    InputCheckboxModule,
    ButtonModule,
];
const components = [AccountPickerComponent];
const directives = [
    CustomSingleSelectedItemLabelDirective,
    CustomMultiSelectedItemsLabelDirective,
    CustomOptionsHeaderDirective,
    CustomOptionItemDirective,
    CustomGroupItemsHeaderDirective,
    CustomLoadingTemplateDirective,
];
export class AccountPickerModule {
}
AccountPickerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountPickerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AccountPickerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountPickerModule, declarations: [AccountPickerComponent, CustomSingleSelectedItemLabelDirective,
        CustomMultiSelectedItemsLabelDirective,
        CustomOptionsHeaderDirective,
        CustomOptionItemDirective,
        CustomGroupItemsHeaderDirective,
        CustomLoadingTemplateDirective], imports: [CommonModule, FormsModule, NgSelectModule, ProductItemBasicAccountModule,
        InputTextModule,
        SearchBoxModule,
        LoadingIndicatorModule,
        InputCheckboxModule,
        ButtonModule], exports: [AccountPickerComponent, CustomSingleSelectedItemLabelDirective,
        CustomMultiSelectedItemsLabelDirective,
        CustomOptionsHeaderDirective,
        CustomOptionItemDirective,
        CustomGroupItemsHeaderDirective,
        CustomLoadingTemplateDirective, ProductItemBasicAccountModule,
        InputTextModule,
        SearchBoxModule,
        LoadingIndicatorModule,
        InputCheckboxModule,
        ButtonModule, NgSelectModule] });
AccountPickerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountPickerModule, imports: [[CommonModule, FormsModule, NgSelectModule, ...uiModules], ProductItemBasicAccountModule,
        InputTextModule,
        SearchBoxModule,
        LoadingIndicatorModule,
        InputCheckboxModule,
        ButtonModule, NgSelectModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountPickerModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, FormsModule, NgSelectModule, ...uiModules],
                    declarations: [...components, ...directives],
                    exports: [...components, ...directives, ...uiModules, NgSelectModule],
                }]
        }] });
