// @backbase/ui-ang/esm2015/input-range/bb-input-range-ui.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { InputRangeDirective, InputRangeMaxDirective, InputRangeMinDirective } from './bb-input-range-ui.directive';
import * as i0 from "@angular/core";
export class InputRangeModule {
}
InputRangeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputRangeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputRangeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputRangeModule, declarations: [InputRangeDirective, InputRangeMinDirective, InputRangeMaxDirective], imports: [CommonModule], exports: [InputRangeDirective, InputRangeMinDirective, InputRangeMaxDirective] });
InputRangeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputRangeModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputRangeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    exports: [InputRangeDirective, InputRangeMinDirective, InputRangeMaxDirective],
                    declarations: [InputRangeDirective, InputRangeMinDirective, InputRangeMaxDirective],
                }]
        }] });
