// @backbase/messages-common/esm2015/src/directives/prevent-bubble-down/prevent-bubble-down.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { PreventBubbleDownDirective } from './prevent-bubble-down.directive';
import * as i0 from "@angular/core";
export class PreventBubbleDownModule {
}
PreventBubbleDownModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: PreventBubbleDownModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PreventBubbleDownModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: PreventBubbleDownModule, declarations: [PreventBubbleDownDirective], imports: [CommonModule], exports: [PreventBubbleDownDirective] });
PreventBubbleDownModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: PreventBubbleDownModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: PreventBubbleDownModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [PreventBubbleDownDirective],
                    imports: [CommonModule],
                    exports: [PreventBubbleDownDirective],
                }]
        }] });
