// @backbase/budget-widget-ang/esm2015/src/budget-widget.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { AmountModule } from '@backbase/ui-ang/amount';
import { ButtonModule } from '@backbase/ui-ang/button';
import { CssVariablesService } from '@backbase/ui-ang/css-variables-lib';
import { CurrencyInputModule } from '@backbase/ui-ang/currency-input';
import { DropdownMenuModule } from '@backbase/ui-ang/dropdown-menu';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { IconModule } from '@backbase/ui-ang/icon';
import { InputDatepickerModule } from '@backbase/ui-ang/input-datepicker';
import { InputRadioGroupModule } from '@backbase/ui-ang/input-radio-group';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { ModalModule } from '@backbase/ui-ang/modal';
import { NotificationModule } from '@backbase/ui-ang/notification';
import { StepperModule } from '@backbase/ui-ang/stepper';
import { NgbDropdownModule } from '@ng-bootstrap/ng-bootstrap';
import { PieChartModule } from '@swimlane/ngx-charts';
import { BudgetCardChartComponent } from './budget-card/budget-card-chart.component';
import { BudgetCardInfoComponent } from './budget-card/budget-card-info.component';
import { BudgetCardComponent } from './budget-card/budget-card.component';
import { BudgetCardCustomizableDirective, BudgetConfirmationModalCustomizableDirective, BudgetCreateModalCustomizableDirective, BudgetManagerComponent, BudgetsHeaderCustomizableDirective, BudgetUpdateModalCustomizableDirective, } from './budget-manager/budget-manager.component';
import { BudgetNotificationCustomizableDirective, BudgetWidgetComponent } from './budget-widget.component';
import { BudgetsHeaderComponent } from './budgets-header/budgets-header.component';
import { BudgetService } from './services/budget.service';
import { CategoriesService } from './services/categories.service';
import { UtilityService } from './services/utility.service';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const modules = [
    AmountModule,
    ButtonModule,
    CurrencyInputModule,
    DropdownMenuModule,
    EmptyStateModule,
    IconModule,
    InputDatepickerModule,
    InputRadioGroupModule,
    LoadingIndicatorModule,
    ModalModule,
    NotificationModule,
    StepperModule,
];
const components = [
    BudgetWidgetComponent,
    BudgetsHeaderComponent,
    BudgetCardComponent,
    BudgetCardChartComponent,
    BudgetCardInfoComponent,
    BudgetManagerComponent,
];
const directives = [
    BudgetsHeaderCustomizableDirective,
    BudgetCardCustomizableDirective,
    BudgetConfirmationModalCustomizableDirective,
    BudgetUpdateModalCustomizableDirective,
    BudgetCreateModalCustomizableDirective,
    BudgetNotificationCustomizableDirective,
];
export class BudgetWidgetModule {
}
BudgetWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
BudgetWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetWidgetModule, declarations: [BudgetWidgetComponent,
        BudgetsHeaderComponent,
        BudgetCardComponent,
        BudgetCardChartComponent,
        BudgetCardInfoComponent,
        BudgetManagerComponent, BudgetsHeaderCustomizableDirective,
        BudgetCardCustomizableDirective,
        BudgetConfirmationModalCustomizableDirective,
        BudgetUpdateModalCustomizableDirective,
        BudgetCreateModalCustomizableDirective,
        BudgetNotificationCustomizableDirective], imports: [CommonModule, i1.BackbaseCoreModule, ReactiveFormsModule,
        NgbDropdownModule,
        PieChartModule, AmountModule,
        ButtonModule,
        CurrencyInputModule,
        DropdownMenuModule,
        EmptyStateModule,
        IconModule,
        InputDatepickerModule,
        InputRadioGroupModule,
        LoadingIndicatorModule,
        ModalModule,
        NotificationModule,
        StepperModule], exports: [BudgetWidgetComponent,
        BudgetsHeaderComponent,
        BudgetCardComponent,
        BudgetCardChartComponent,
        BudgetCardInfoComponent,
        BudgetManagerComponent, BudgetsHeaderCustomizableDirective,
        BudgetCardCustomizableDirective,
        BudgetConfirmationModalCustomizableDirective,
        BudgetUpdateModalCustomizableDirective,
        BudgetCreateModalCustomizableDirective,
        BudgetNotificationCustomizableDirective, AmountModule,
        ButtonModule,
        CurrencyInputModule,
        DropdownMenuModule,
        EmptyStateModule,
        IconModule,
        InputDatepickerModule,
        InputRadioGroupModule,
        LoadingIndicatorModule,
        ModalModule,
        NotificationModule,
        StepperModule] });
BudgetWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetWidgetModule, providers: [BudgetService, CategoriesService, UtilityService, CssVariablesService], imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { BudgetWidgetComponent },
            }),
            ReactiveFormsModule,
            NgbDropdownModule,
            PieChartModule,
            ...modules,
        ], AmountModule,
        ButtonModule,
        CurrencyInputModule,
        DropdownMenuModule,
        EmptyStateModule,
        IconModule,
        InputDatepickerModule,
        InputRadioGroupModule,
        LoadingIndicatorModule,
        ModalModule,
        NotificationModule,
        StepperModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { BudgetWidgetComponent },
                        }),
                        ReactiveFormsModule,
                        NgbDropdownModule,
                        PieChartModule,
                        ...modules,
                    ],
                    declarations: [...components, ...directives],
                    providers: [BudgetService, CategoriesService, UtilityService, CssVariablesService],
                    exports: [...components, ...directives, ...modules],
                }]
        }] });
