// @backbase/income-spending-analysis-widget-ang/esm2015/src/bb-donut-chart-ui/donut-chart.component.js
import { ChangeDetectionStrategy, Component, HostBinding, Input, ViewChild } from '@angular/core';
import { ArrowClasses, CARET_SIZE_CSS_VAR } from './constants';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/css-variables-lib";
import * as i2 from "./ngx/pie/pie-chart.component";
import * as i3 from "@backbase/ui-ang/amount";
import * as i4 from "@angular/common";
export class DonutChartComponent {
    constructor(cssVariableService) {
        this.cssVariableService = cssVariableService;
        /**
         * Sets the threshold in which the label of each section is visible.
         */
        this.relevancePercentage = 0;
        this.classes = 'absolute-center w-100 h-100';
    }
    get template() {
        return this.tooltipTmpl || this.defaultTooltipTemplate;
    }
    expandModel(model) {
        if (!this.data) {
            return model;
        }
        const data = this.data.find((item) => item.name === model.name) || model;
        return Object.assign(data, {
            arrowClass: data.totalPortion - data.portion / 2 >= 50 ? ArrowClasses.RIGHT : ArrowClasses.LEFT,
            arrowSize: this.arrowSize,
        });
    }
    ngOnInit() {
        this.cssVariableService
            .getCssVariable(CARET_SIZE_CSS_VAR)
            .then((value) => (this.arrowSize = parseInt(value || '', 10) || 0));
    }
}
DonutChartComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DonutChartComponent, deps: [{ token: i1.CssVariablesService }], target: i0.ɵɵFactoryTarget.Component });
DonutChartComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: DonutChartComponent, selector: "bb-donut-chart-ui", inputs: { data: "data", colors: "colors", relevancePercentage: "relevancePercentage", tooltipTmpl: "tooltipTmpl", labelTmpl: "labelTmpl" }, host: { properties: { "class": "this.classes" } }, viewQueries: [{ propertyName: "defaultTooltipTemplate", first: true, predicate: ["defaultTooltipTemplate"], descendants: true, static: true }], ngImport: i0, template: "<bb-charts-pie-chart\n  [results]=\"data\"\n  [series]=\"data\"\n  [customColors]=\"colors\"\n  [doughnut]=\"true\"\n  [arcWidth]=\"0.4\"\n  [labels]=\"true\"\n  [labelThreshold]=\"relevancePercentage\"\n  [labelTemplate]=\"labelTmpl\"\n>\n  <ng-template #tooltipTemplate let-model=\"model\">\n    <ng-container *ngTemplateOutlet=\"template; context: expandModel(model)\"></ng-container>\n  </ng-template>\n</bb-charts-pie-chart>\n\n<ng-template\n  #defaultTooltipTemplate\n  let-name=\"name\"\n  let-totalAmount=\"totalAmount\"\n>\n  <h3 class=\"chart-tooltip-analysis-category\">\n    {{ name }}\n  </h3>\n  <h4 class=\"chart-tooltip-analysis-amount\">\n    <bb-amount-ui\n      [amount]=\"totalAmount.amount\"\n      [currency]=\"totalAmount.currencyCode\"\n    ></bb-amount-ui>\n  </h4>\n</ng-template>\n", components: [{ type: i2.PieChartComponent, selector: "bb-charts-pie-chart", inputs: ["series", "labels", "legend", "legendTitle", "explodeSlices", "doughnut", "arcWidth", "gradient", "activeEntries", "tooltipDisabled", "labelFormatting", "labelTemplate", "labelThreshold", "trimLabels", "maxLabelLength", "tooltipText"], outputs: ["dblclick", "select", "activate", "deactivate"] }, { type: i3.AmountComponent, selector: "bb-amount-ui", inputs: ["currency", "showPlusSign", "mapCurrency", "showPercent", "abbreviate", "decimalPlaces", "trailingZeroes", "amount"] }], directives: [{ type: i4.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DonutChartComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-donut-chart-ui',
                    templateUrl: './donut-chart.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.CssVariablesService }]; }, propDecorators: { data: [{
                type: Input
            }], colors: [{
                type: Input
            }], relevancePercentage: [{
                type: Input
            }], tooltipTmpl: [{
                type: Input
            }], labelTmpl: [{
                type: Input
            }], defaultTooltipTemplate: [{
                type: ViewChild,
                args: ['defaultTooltipTemplate', { static: true }]
            }], classes: [{
                type: HostBinding,
                args: ['class']
            }] } });
