// @backbase/income-spending-analysis-widget-ang/esm2015/src/bb-donut-chart-ui/donut-chart.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { AmountModule } from '@backbase/ui-ang/amount';
import { CssVariablesService } from '@backbase/ui-ang/css-variables-lib';
import { DonutChartComponent } from './donut-chart.component';
import { PieChartModule } from './ngx/pie/pie-chart.module';
import * as i0 from "@angular/core";
export class DonutChartModule {
}
DonutChartModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DonutChartModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
DonutChartModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DonutChartModule, declarations: [DonutChartComponent], imports: [CommonModule, AmountModule, PieChartModule], exports: [DonutChartComponent, PieChartModule, AmountModule] });
DonutChartModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DonutChartModule, providers: [CssVariablesService], imports: [[CommonModule, AmountModule, PieChartModule], PieChartModule, AmountModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DonutChartModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, AmountModule, PieChartModule],
                    declarations: [DonutChartComponent],
                    exports: [DonutChartComponent, PieChartModule, AmountModule],
                    providers: [CssVariablesService],
                }]
        }] });
