// @backbase/payment-template-http-ang/bundles/backbase-payment-template-http-ang.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common/http'), require('@backbase/foundation-ang/data-http')) :
    typeof define === 'function' && define.amd ? define('@backbase/payment-template-http-ang', ['exports', '@angular/core', '@angular/common/http', '@backbase/foundation-ang/data-http'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["payment-template-http-ang"] = {}), global.ng.core, global.ng.common.http, global.i2));
})(this, (function (exports, i0, i1, i2) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);

    var Action = {
        CREATE: 'CREATE',
        VIEW: 'VIEW'
    };

    var ChargeBearer = {
        OUR: 'OUR',
        BEN: 'BEN',
        SHA: 'SHA'
    };

    var InstructionPriority = {
        NORM: 'NORM',
        HIGH: 'HIGH'
    };

    var InvolvedPartyRole = {
        CREDITOR: 'CREDITOR',
        DEBTOR: 'DEBTOR'
    };

    var SchemeNames = {
        IBAN: 'IBAN',
        BBAN: 'BBAN',
        ID: 'ID',
        EXTERNALID: 'EXTERNAL_ID',
        EMAIL: 'EMAIL',
        MOBILE: 'MOBILE'
    };

    var PAYMENT_TEMPLATE_BASE_PATH = new i0.InjectionToken('PAYMENT_TEMPLATE_BASE_PATH');
    var COLLECTION_FORMATS = {
        'csv': ',',
        'tsv': '   ',
        'ssv': ' ',
        'pipes': '|'
    };

    var PaymentTemplateConfiguration = /** @class */ (function () {
        function PaymentTemplateConfiguration(configurationParameters) {
            if (configurationParameters === void 0) { configurationParameters = {}; }
            this.apiKeys = configurationParameters.apiKeys;
            this.username = configurationParameters.username;
            this.password = configurationParameters.password;
            this.accessToken = configurationParameters.accessToken;
            this.basePath = configurationParameters.basePath;
            this.withCredentials = configurationParameters.withCredentials;
            this.encoder = configurationParameters.encoder;
            if (configurationParameters.credentials) {
                this.credentials = configurationParameters.credentials;
            }
            else {
                this.credentials = {};
            }
        }
        PaymentTemplateConfiguration.prototype.selectHeaderContentType = function (contentTypes) {
            var _this = this;
            if (contentTypes.length === 0) {
                return undefined;
            }
            var type = contentTypes.find(function (x) { return _this.isJsonMime(x); });
            if (type === undefined) {
                return contentTypes[0];
            }
            return type;
        };
        PaymentTemplateConfiguration.prototype.selectHeaderAccept = function (accepts) {
            var _this = this;
            if (accepts.length === 0) {
                return undefined;
            }
            var type = accepts.find(function (x) { return _this.isJsonMime(x); });
            if (type === undefined) {
                return accepts[0];
            }
            return type;
        };
        PaymentTemplateConfiguration.prototype.isJsonMime = function (mime) {
            var jsonMime = new RegExp('^(application\/json|[^;/ \t]+\/[^;/ \t]+[+]json)[ \t]*(;.*)?$', 'i');
            return mime !== null && (jsonMime.test(mime) || mime.toLowerCase() === 'application/json-patch+json');
        };
        PaymentTemplateConfiguration.prototype.lookupCredential = function (key) {
            var value = this.credentials[key];
            return typeof value === 'function'
                ? value()
                : value;
        };
        return PaymentTemplateConfiguration;
    }());
    var CONFIG_TOKEN = new i0.InjectionToken('api-data-module PaymentTemplateConfiguration');

    var PaymentTemplateApiModule = /** @class */ (function () {
        function PaymentTemplateApiModule(parentModule, http, dataModulesManager, config) {
            if (parentModule) {
                throw new Error('PaymentTemplateApiModule is already loaded. Import in your base AppModule only.');
            }
            if (!http) {
                throw new Error('You need to import the HttpClientModule in your AppModule! \n' +
                    'See also https://github.com/angular/angular/issues/20575');
            }
            if (dataModulesManager) {
                dataModulesManager.setModuleConfig(CONFIG_TOKEN, {
                    apiRoot: '',
                    servicePath: config.basePath || '',
                    headers: {},
                });
            }
        }
        PaymentTemplateApiModule.forRoot = function (configurationFactory) {
            return {
                ngModule: PaymentTemplateApiModule,
                providers: [{ provide: PaymentTemplateConfiguration, useFactory: configurationFactory }]
            };
        };
        return PaymentTemplateApiModule;
    }());
    PaymentTemplateApiModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: PaymentTemplateApiModule, deps: [{ token: PaymentTemplateApiModule, optional: true, skipSelf: true }, { token: i1__namespace.HttpClient, optional: true }, { token: i2__namespace.DataModulesManager, optional: true }, { token: PaymentTemplateConfiguration }], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    PaymentTemplateApiModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: PaymentTemplateApiModule });
    PaymentTemplateApiModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: PaymentTemplateApiModule, providers: [], imports: [[]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: PaymentTemplateApiModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [],
                        declarations: [],
                        exports: [],
                        providers: []
                    }]
            }], ctorParameters: function () {
            return [{ type: PaymentTemplateApiModule, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.SkipSelf
                        }] }, { type: i1__namespace.HttpClient, decorators: [{
                            type: i0.Optional
                        }] }, { type: i2__namespace.DataModulesManager, decorators: [{
                            type: i0.Optional
                        }] }, { type: PaymentTemplateConfiguration }];
        } });

    var CustomHttpParameterCodec = /** @class */ (function () {
        function CustomHttpParameterCodec() {
        }
        CustomHttpParameterCodec.prototype.encodeKey = function (k) {
            return encodeURIComponent(k);
        };
        CustomHttpParameterCodec.prototype.encodeValue = function (v) {
            return encodeURIComponent(v);
        };
        CustomHttpParameterCodec.prototype.decodeKey = function (k) {
            return decodeURIComponent(k);
        };
        CustomHttpParameterCodec.prototype.decodeValue = function (v) {
            return decodeURIComponent(v);
        };
        return CustomHttpParameterCodec;
    }());

    var PaymentTemplatesHttpService = /** @class */ (function () {
        function PaymentTemplatesHttpService(httpClient, basePath, configuration) {
            this.httpClient = httpClient;
            this.basePath = '';
            this.defaultHeaders = new i1.HttpHeaders();
            this.configuration = new PaymentTemplateConfiguration();
            if (configuration) {
                this.configuration = configuration;
            }
            if (typeof this.configuration.basePath !== 'string') {
                if (typeof basePath !== 'string') {
                    basePath = this.basePath;
                }
                this.configuration.basePath = basePath;
            }
            this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
        }
        PaymentTemplatesHttpService.prototype.addToHttpParams = function (httpParams, value, key) {
            if (typeof value === "object" && value instanceof Date === false) {
                httpParams = this.addToHttpParamsRecursive(httpParams, value);
            }
            else {
                httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
            }
            return httpParams;
        };
        PaymentTemplatesHttpService.prototype.addToHttpParamsRecursive = function (httpParams, value, key) {
            var _this = this;
            if (value == null) {
                return httpParams;
            }
            if (typeof value === "object") {
                if (Array.isArray(value)) {
                    value.forEach(function (elem) { return httpParams = _this.addToHttpParamsRecursive(httpParams, elem, key); });
                }
                else if (value instanceof Date) {
                    if (key != null) {
                        httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                    }
                    else {
                        throw Error("key may not be null if value is Date");
                    }
                }
                else {
                    Object.keys(value).forEach(function (k) { return httpParams = _this.addToHttpParamsRecursive(httpParams, value[k], key != null ? key + "." + k : k); });
                }
            }
            else if (key != null) {
                httpParams = httpParams.append(key, value);
            }
            else {
                throw Error("key may not be null if value is not object or array");
            }
            return httpParams;
        };
        PaymentTemplatesHttpService.prototype.deletePaymentTemplateIdByPaymentTemplateId = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _paymentTemplateId = requestParameters["paymentTemplateId"];
            if (_paymentTemplateId === null || _paymentTemplateId === undefined) {
                throw new Error('Required parameter paymentTemplateId was null or undefined when calling deletePaymentTemplateIdByPaymentTemplateId.');
            }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.delete(this.configuration.basePath + "/client-api/v2/payment-templates/" + encodeURIComponent(String(_paymentTemplateId)), {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        PaymentTemplatesHttpService.prototype.deletePaymentTemplateIdByPaymentTemplateIdUrl = function (requestParameters) {
            var _paymentTemplateId = requestParameters["paymentTemplateId"];
            if (_paymentTemplateId === null || _paymentTemplateId === undefined) {
                throw new Error('Required parameter paymentTemplateId was null or undefined when calling deletePaymentTemplateIdByPaymentTemplateId.');
            }
            return this.configuration.basePath + "/client-api/v2/payment-templates/" + encodeURIComponent(String(_paymentTemplateId));
        };
        PaymentTemplatesHttpService.prototype.getPaymentTemplates = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _query = requestParameters["query"];
            var _intendedAction = requestParameters["intendedAction"];
            var _paymentType = requestParameters["paymentType"];
            var _from = requestParameters["from"];
            var _size = requestParameters["size"];
            var queryParameters = new i1.HttpParams({ encoder: this.encoder });
            if (_query !== undefined && _query !== null) {
                queryParameters = this.addToHttpParams(queryParameters, _query, 'query');
            }
            if (_intendedAction !== undefined && _intendedAction !== null) {
                queryParameters = this.addToHttpParams(queryParameters, _intendedAction, 'intendedAction');
            }
            if (_paymentType !== undefined && _paymentType !== null) {
                queryParameters = this.addToHttpParams(queryParameters, _paymentType, 'paymentType');
            }
            if (_from !== undefined && _from !== null) {
                queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
            }
            if (_size !== undefined && _size !== null) {
                queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
            }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.get(this.configuration.basePath + "/client-api/v2/payment-templates", {
                params: queryParameters,
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        PaymentTemplatesHttpService.prototype.getPaymentTemplatesUrl = function (requestParameters) {
            var _query = requestParameters["query"];
            var _intendedAction = requestParameters["intendedAction"];
            var _paymentType = requestParameters["paymentType"];
            var _from = requestParameters["from"];
            var _size = requestParameters["size"];
            var queryParameters = new i1.HttpParams({ encoder: this.encoder });
            if (_query !== undefined && _query !== null) {
                queryParameters = this.addToHttpParams(queryParameters, _query, 'query');
            }
            if (_intendedAction !== undefined && _intendedAction !== null) {
                queryParameters = this.addToHttpParams(queryParameters, _intendedAction, 'intendedAction');
            }
            if (_paymentType !== undefined && _paymentType !== null) {
                queryParameters = this.addToHttpParams(queryParameters, _paymentType, 'paymentType');
            }
            if (_from !== undefined && _from !== null) {
                queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
            }
            if (_size !== undefined && _size !== null) {
                queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
            }
            var queryString = queryParameters.toString();
            return this.configuration.basePath + "/client-api/v2/payment-templates" + (queryString ? "?" + queryString : '');
        };
        PaymentTemplatesHttpService.prototype.postPaymentTemplates = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _paymentTemplatesPostRequest = requestParameters["paymentTemplatesPostRequest"];
            if (_paymentTemplatesPostRequest === null || _paymentTemplatesPostRequest === undefined) {
                throw new Error('Required parameter paymentTemplatesPostRequest was null or undefined when calling postPaymentTemplates.');
            }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var consumes = [
                'application/json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected !== undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.post(this.configuration.basePath + "/client-api/v2/payment-templates", _paymentTemplatesPostRequest, {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        PaymentTemplatesHttpService.prototype.postPaymentTemplatesUrl = function (requestParameters) {
            var _paymentTemplatesPostRequest = requestParameters["paymentTemplatesPostRequest"];
            if (_paymentTemplatesPostRequest === null || _paymentTemplatesPostRequest === undefined) {
                throw new Error('Required parameter paymentTemplatesPostRequest was null or undefined when calling postPaymentTemplates.');
            }
            return this.configuration.basePath + "/client-api/v2/payment-templates";
        };
        PaymentTemplatesHttpService.prototype.putPaymentTemplateById = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _paymentTemplateId = requestParameters["paymentTemplateId"];
            if (_paymentTemplateId === null || _paymentTemplateId === undefined) {
                throw new Error('Required parameter paymentTemplateId was null or undefined when calling putPaymentTemplateById.');
            }
            var _paymentTemplatesPutRequest = requestParameters["paymentTemplatesPutRequest"];
            if (_paymentTemplatesPutRequest === null || _paymentTemplatesPutRequest === undefined) {
                throw new Error('Required parameter paymentTemplatesPutRequest was null or undefined when calling putPaymentTemplateById.');
            }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var consumes = [
                'application/json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected !== undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.put(this.configuration.basePath + "/client-api/v2/payment-templates/" + encodeURIComponent(String(_paymentTemplateId)), _paymentTemplatesPutRequest, {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        PaymentTemplatesHttpService.prototype.putPaymentTemplateByIdUrl = function (requestParameters) {
            var _paymentTemplateId = requestParameters["paymentTemplateId"];
            if (_paymentTemplateId === null || _paymentTemplateId === undefined) {
                throw new Error('Required parameter paymentTemplateId was null or undefined when calling putPaymentTemplateById.');
            }
            var _paymentTemplatesPutRequest = requestParameters["paymentTemplatesPutRequest"];
            if (_paymentTemplatesPutRequest === null || _paymentTemplatesPutRequest === undefined) {
                throw new Error('Required parameter paymentTemplatesPutRequest was null or undefined when calling putPaymentTemplateById.');
            }
            return this.configuration.basePath + "/client-api/v2/payment-templates/" + encodeURIComponent(String(_paymentTemplateId));
        };
        return PaymentTemplatesHttpService;
    }());
    PaymentTemplatesHttpService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: PaymentTemplatesHttpService, deps: [{ token: i1__namespace.HttpClient }, { token: PAYMENT_TEMPLATE_BASE_PATH, optional: true }, { token: PaymentTemplateConfiguration, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    PaymentTemplatesHttpService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: PaymentTemplatesHttpService, providedIn: 'root' });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: PaymentTemplatesHttpService, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root'
                    }]
            }], ctorParameters: function () {
            return [{ type: i1__namespace.HttpClient }, { type: undefined, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.Inject,
                            args: [PAYMENT_TEMPLATE_BASE_PATH]
                        }] }, { type: PaymentTemplateConfiguration, decorators: [{
                            type: i0.Optional
                        }] }];
        } });

    var PaymentTemplatesHttpServiceDeletePaymentTemplateIdByPaymentTemplateIdMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v2/payment-templates/{paymentTemplateId}",
            method: "DELETE",
            responses: [
                {
                    status: 204,
                    body: ""
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
            ]
        }]);
    var PaymentTemplatesHttpServiceGetPaymentTemplatesMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v2/payment-templates",
            method: "GET",
            responses: [
                {
                    status: 200,
                    body: [{
                            "id": "7d341c28-6714-11e7-907b-a6006ad3dba0",
                            "name": "Textile supplier invoice",
                            "createdAt": "2020-04-12T14:33:13.452Z",
                            "serviceAgreementId": "d6fdc581-1306-412a-9e40-86d851190acf",
                            "details": {
                                "paymentType": "SEPA_CREDIT_TRANSFER",
                                "originatorAccount": {
                                    "arrangementId": "729190df-a421-4937-94fd-5e1a3da132cc",
                                    "externalArrangementId": "729190421493794513132",
                                    "identification": {
                                        "identification": "NL53RABO0309349755",
                                        "schemeName": "IBAN"
                                    },
                                    "name": "Smith Bonus Savings"
                                },
                                "instructionPriority": "NORM",
                                "transferTransactionInformation": {
                                    "counterparty": {
                                        "name": "Backbase",
                                        "postalAddress": {
                                            "addressLine1": "Jacob Bontiusplaats 9, 1018LL, Amsterdam",
                                            "country": "NL"
                                        }
                                    },
                                    "counterpartyAccount": {
                                        "identification": {
                                            "identification": "FR708933019952AUNHQNQ0KZ",
                                            "schemeName": "IBAN"
                                        },
                                        "arrangementId": "fe9d66ae-b927-4ac7-8799-c5a38a596ff2",
                                        "externalArrangementId": "9669274787995385962"
                                    },
                                    "instructedAmount": {
                                        "amount": "5000.55",
                                        "currencyCode": "EUR"
                                    },
                                    "remittanceInformation": "Salary"
                                }
                            }
                        }]
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
            ]
        }]);
    var PaymentTemplatesHttpServicePostPaymentTemplatesMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v2/payment-templates",
            method: "POST",
            responses: [
                {
                    status: 201,
                    body: {
                        "template": {
                            "id": "7d341c28-6714-11e7-907b-a6006ad3dba0",
                            "name": "Salary for John",
                            "createdAt": "2020-04-12T14:33:13.452Z",
                            "serviceAgreementId": "d6fdc581-1306-412a-9e40-86d851190acf",
                            "details": {
                                "paymentType": "SEPA_CREDIT_TRANSFER",
                                "originatorAccount": {
                                    "arrangementId": "729190df-a421-4937-94fd-5e1a3da132cc",
                                    "externalArrangementId": "729190421493794513132",
                                    "identification": {
                                        "identification": "NL53RABO0309349755",
                                        "schemeName": "IBAN"
                                    }
                                },
                                "instructionPriority": "NORM",
                                "transferTransactionInformation": {
                                    "counterparty": {
                                        "name": "John Smith",
                                        "postalAddress": {
                                            "addressLine1": "Schippersstraat 15, 1011AZ, Amsterdam",
                                            "country": "NL"
                                        }
                                    },
                                    "counterpartyAccount": {
                                        "identification": {
                                            "identification": "FR708933019952AUNHQNQ0KZ",
                                            "schemeName": "IBAN"
                                        },
                                        "arrangementId": "fe9d66ae-b927-4ac7-8799-c5a38a596ff2",
                                        "externalArrangementId": "9669274787995385962"
                                    },
                                    "instructedAmount": {
                                        "amount": "5000.55",
                                        "currencyCode": "EUR"
                                    },
                                    "remittanceInformation": "Salary"
                                }
                            }
                        }
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
            ]
        }]);
    var PaymentTemplatesHttpServicePutPaymentTemplateByIdMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v2/payment-templates/{paymentTemplateId}",
            method: "PUT",
            responses: [
                {
                    status: 202,
                    body: {
                        "template": {
                            "id": "7d341c28-6714-11e7-907b-a6006ad3dba0",
                            "name": "Salary for John",
                            "createdAt": "2020-04-12T14:33:13.452Z",
                            "serviceAgreementId": "d6fdc581-1306-412a-9e40-86d851190acf",
                            "details": {
                                "paymentType": "SEPA_CREDIT_TRANSFER",
                                "originatorAccount": {
                                    "arrangementId": "729190df-a421-4937-94fd-5e1a3da132cc",
                                    "externalArrangementId": "729190421493794513132",
                                    "identification": {
                                        "identification": "NL53RABO0309349755",
                                        "schemeName": "IBAN"
                                    }
                                },
                                "instructionPriority": "NORM",
                                "transferTransactionInformation": {
                                    "counterparty": {
                                        "name": "John Smith",
                                        "postalAddress": {
                                            "addressLine1": "Schippersstraat 15, 1011AZ, Amsterdam",
                                            "country": "NL"
                                        }
                                    },
                                    "counterpartyAccount": {
                                        "identification": {
                                            "identification": "FR708933019952AUNHQNQ0KZ",
                                            "schemeName": "IBAN"
                                        },
                                        "arrangementId": "fe9d66ae-b927-4ac7-8799-c5a38a596ff2",
                                        "externalArrangementId": "9669274787995385962"
                                    },
                                    "instructedAmount": {
                                        "amount": "5000.55",
                                        "currencyCode": "EUR"
                                    },
                                    "remittanceInformation": "Salary"
                                }
                            }
                        }
                    }
                },
                {
                    status: 202,
                    body: {
                        "template": {
                            "createdAt": "2000-01-23T04:56:07.000+00:00",
                            "additions": {
                                "key": "additions"
                            },
                            "serviceAgreementId": "serviceAgreementId",
                            "name": "name",
                            "details": {
                                "transferTransactionInformation": {
                                    "correspondentBank": {
                                        "bankBranchCode": "bankBranchCode",
                                        "postalAddress": {
                                            "country": "country",
                                            "streetName": "streetName",
                                            "town": "town",
                                            "countrySubDivision": "countrySubDivision",
                                            "additions": {
                                                "key": "additions"
                                            },
                                            "addressLine1": "addressLine1",
                                            "addressLine2": "addressLine2",
                                            "postCode": "postCode"
                                        },
                                        "additions": {
                                            "key": "additions"
                                        },
                                        "name": "name",
                                        "bic": "bic"
                                    },
                                    "counterpartyBank": {
                                        "bankBranchCode": "bankBranchCode",
                                        "postalAddress": {
                                            "country": "country",
                                            "streetName": "streetName",
                                            "town": "town",
                                            "countrySubDivision": "countrySubDivision",
                                            "additions": {
                                                "key": "additions"
                                            },
                                            "addressLine1": "addressLine1",
                                            "addressLine2": "addressLine2",
                                            "postCode": "postCode"
                                        },
                                        "additions": {
                                            "key": "additions"
                                        },
                                        "name": "name",
                                        "bic": "bic"
                                    },
                                    "targetCurrency": "targetCurrency",
                                    "additions": {
                                        "key": "additions"
                                    },
                                    "purposeOfPayment": {
                                        "code": "code",
                                        "additions": {
                                            "key": "additions"
                                        },
                                        "freeText": "freeText",
                                        "description": "description"
                                    },
                                    "counterpartyAccount": {
                                        "accountType": "accountType",
                                        "selectedContact": {
                                            "accountId": "accountId",
                                            "additions": {
                                                "key": "additions"
                                            },
                                            "contactId": "contactId"
                                        }
                                    },
                                    "intermediaryBank": {
                                        "bankBranchCode": "bankBranchCode",
                                        "postalAddress": {
                                            "country": "country",
                                            "streetName": "streetName",
                                            "town": "town",
                                            "countrySubDivision": "countrySubDivision",
                                            "additions": {
                                                "key": "additions"
                                            },
                                            "addressLine1": "addressLine1",
                                            "addressLine2": "addressLine2",
                                            "postCode": "postCode"
                                        },
                                        "additions": {
                                            "key": "additions"
                                        },
                                        "name": "name",
                                        "bic": "bic"
                                    },
                                    "mandateIdentifier": "mandateIdentifier",
                                    "remittanceInformation": "remittanceInformation",
                                    "messageToBank": "messageToBank",
                                    "counterparty": {
                                        "postalAddress": {
                                            "country": "country",
                                            "streetName": "streetName",
                                            "town": "town",
                                            "countrySubDivision": "countrySubDivision",
                                            "additions": {
                                                "key": "additions"
                                            },
                                            "addressLine1": "addressLine1",
                                            "addressLine2": "addressLine2",
                                            "postCode": "postCode"
                                        },
                                        "additions": {
                                            "key": "additions"
                                        },
                                        "name": "name",
                                        "recipientId": "recipientId"
                                    },
                                    "instructedAmount": {
                                        "amount": "amount",
                                        "additions": {
                                            "key": "additions"
                                        },
                                        "currencyCode": "currencyCode"
                                    },
                                    "endToEndIdentification": "endToEndIdentification"
                                },
                                "additions": {
                                    "key": "additions"
                                },
                                "originatorAccount": {
                                    "arrangementId": "arrangementId",
                                    "identification": {
                                        "identification": "identification",
                                        "additions": {
                                            "key": "additions"
                                        }
                                    },
                                    "additions": {
                                        "key": "additions"
                                    },
                                    "name": "name",
                                    "externalArrangementId": "externalArrangementId"
                                },
                                "entryClass": "entryClass",
                                "paymentType": "paymentType"
                            },
                            "id": "id",
                            "actions": [null, null]
                        },
                        "additions": {
                            "key": "additions"
                        }
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
            ]
        }]);
    var PaymentTemplatesHttpServiceMocksProvider = i2.createMocks([
        {
            urlPattern: "/client-api/v2/payment-templates/{paymentTemplateId}",
            method: "DELETE",
            responses: [
                {
                    status: 204,
                    body: ""
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
            ]
        },
        {
            urlPattern: "/client-api/v2/payment-templates",
            method: "GET",
            responses: [
                {
                    status: 200,
                    body: [{
                            "id": "7d341c28-6714-11e7-907b-a6006ad3dba0",
                            "name": "Textile supplier invoice",
                            "createdAt": "2020-04-12T14:33:13.452Z",
                            "serviceAgreementId": "d6fdc581-1306-412a-9e40-86d851190acf",
                            "details": {
                                "paymentType": "SEPA_CREDIT_TRANSFER",
                                "originatorAccount": {
                                    "arrangementId": "729190df-a421-4937-94fd-5e1a3da132cc",
                                    "externalArrangementId": "729190421493794513132",
                                    "identification": {
                                        "identification": "NL53RABO0309349755",
                                        "schemeName": "IBAN"
                                    },
                                    "name": "Smith Bonus Savings"
                                },
                                "instructionPriority": "NORM",
                                "transferTransactionInformation": {
                                    "counterparty": {
                                        "name": "Backbase",
                                        "postalAddress": {
                                            "addressLine1": "Jacob Bontiusplaats 9, 1018LL, Amsterdam",
                                            "country": "NL"
                                        }
                                    },
                                    "counterpartyAccount": {
                                        "identification": {
                                            "identification": "FR708933019952AUNHQNQ0KZ",
                                            "schemeName": "IBAN"
                                        },
                                        "arrangementId": "fe9d66ae-b927-4ac7-8799-c5a38a596ff2",
                                        "externalArrangementId": "9669274787995385962"
                                    },
                                    "instructedAmount": {
                                        "amount": "5000.55",
                                        "currencyCode": "EUR"
                                    },
                                    "remittanceInformation": "Salary"
                                }
                            }
                        }]
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
            ]
        },
        {
            urlPattern: "/client-api/v2/payment-templates",
            method: "POST",
            responses: [
                {
                    status: 201,
                    body: {
                        "template": {
                            "id": "7d341c28-6714-11e7-907b-a6006ad3dba0",
                            "name": "Salary for John",
                            "createdAt": "2020-04-12T14:33:13.452Z",
                            "serviceAgreementId": "d6fdc581-1306-412a-9e40-86d851190acf",
                            "details": {
                                "paymentType": "SEPA_CREDIT_TRANSFER",
                                "originatorAccount": {
                                    "arrangementId": "729190df-a421-4937-94fd-5e1a3da132cc",
                                    "externalArrangementId": "729190421493794513132",
                                    "identification": {
                                        "identification": "NL53RABO0309349755",
                                        "schemeName": "IBAN"
                                    }
                                },
                                "instructionPriority": "NORM",
                                "transferTransactionInformation": {
                                    "counterparty": {
                                        "name": "John Smith",
                                        "postalAddress": {
                                            "addressLine1": "Schippersstraat 15, 1011AZ, Amsterdam",
                                            "country": "NL"
                                        }
                                    },
                                    "counterpartyAccount": {
                                        "identification": {
                                            "identification": "FR708933019952AUNHQNQ0KZ",
                                            "schemeName": "IBAN"
                                        },
                                        "arrangementId": "fe9d66ae-b927-4ac7-8799-c5a38a596ff2",
                                        "externalArrangementId": "9669274787995385962"
                                    },
                                    "instructedAmount": {
                                        "amount": "5000.55",
                                        "currencyCode": "EUR"
                                    },
                                    "remittanceInformation": "Salary"
                                }
                            }
                        }
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
            ]
        },
        {
            urlPattern: "/client-api/v2/payment-templates/{paymentTemplateId}",
            method: "PUT",
            responses: [
                {
                    status: 202,
                    body: {
                        "template": {
                            "id": "7d341c28-6714-11e7-907b-a6006ad3dba0",
                            "name": "Salary for John",
                            "createdAt": "2020-04-12T14:33:13.452Z",
                            "serviceAgreementId": "d6fdc581-1306-412a-9e40-86d851190acf",
                            "details": {
                                "paymentType": "SEPA_CREDIT_TRANSFER",
                                "originatorAccount": {
                                    "arrangementId": "729190df-a421-4937-94fd-5e1a3da132cc",
                                    "externalArrangementId": "729190421493794513132",
                                    "identification": {
                                        "identification": "NL53RABO0309349755",
                                        "schemeName": "IBAN"
                                    }
                                },
                                "instructionPriority": "NORM",
                                "transferTransactionInformation": {
                                    "counterparty": {
                                        "name": "John Smith",
                                        "postalAddress": {
                                            "addressLine1": "Schippersstraat 15, 1011AZ, Amsterdam",
                                            "country": "NL"
                                        }
                                    },
                                    "counterpartyAccount": {
                                        "identification": {
                                            "identification": "FR708933019952AUNHQNQ0KZ",
                                            "schemeName": "IBAN"
                                        },
                                        "arrangementId": "fe9d66ae-b927-4ac7-8799-c5a38a596ff2",
                                        "externalArrangementId": "9669274787995385962"
                                    },
                                    "instructedAmount": {
                                        "amount": "5000.55",
                                        "currencyCode": "EUR"
                                    },
                                    "remittanceInformation": "Salary"
                                }
                            }
                        }
                    }
                },
                {
                    status: 202,
                    body: {
                        "template": {
                            "createdAt": "2000-01-23T04:56:07.000+00:00",
                            "additions": {
                                "key": "additions"
                            },
                            "serviceAgreementId": "serviceAgreementId",
                            "name": "name",
                            "details": {
                                "transferTransactionInformation": {
                                    "correspondentBank": {
                                        "bankBranchCode": "bankBranchCode",
                                        "postalAddress": {
                                            "country": "country",
                                            "streetName": "streetName",
                                            "town": "town",
                                            "countrySubDivision": "countrySubDivision",
                                            "additions": {
                                                "key": "additions"
                                            },
                                            "addressLine1": "addressLine1",
                                            "addressLine2": "addressLine2",
                                            "postCode": "postCode"
                                        },
                                        "additions": {
                                            "key": "additions"
                                        },
                                        "name": "name",
                                        "bic": "bic"
                                    },
                                    "counterpartyBank": {
                                        "bankBranchCode": "bankBranchCode",
                                        "postalAddress": {
                                            "country": "country",
                                            "streetName": "streetName",
                                            "town": "town",
                                            "countrySubDivision": "countrySubDivision",
                                            "additions": {
                                                "key": "additions"
                                            },
                                            "addressLine1": "addressLine1",
                                            "addressLine2": "addressLine2",
                                            "postCode": "postCode"
                                        },
                                        "additions": {
                                            "key": "additions"
                                        },
                                        "name": "name",
                                        "bic": "bic"
                                    },
                                    "targetCurrency": "targetCurrency",
                                    "additions": {
                                        "key": "additions"
                                    },
                                    "purposeOfPayment": {
                                        "code": "code",
                                        "additions": {
                                            "key": "additions"
                                        },
                                        "freeText": "freeText",
                                        "description": "description"
                                    },
                                    "counterpartyAccount": {
                                        "accountType": "accountType",
                                        "selectedContact": {
                                            "accountId": "accountId",
                                            "additions": {
                                                "key": "additions"
                                            },
                                            "contactId": "contactId"
                                        }
                                    },
                                    "intermediaryBank": {
                                        "bankBranchCode": "bankBranchCode",
                                        "postalAddress": {
                                            "country": "country",
                                            "streetName": "streetName",
                                            "town": "town",
                                            "countrySubDivision": "countrySubDivision",
                                            "additions": {
                                                "key": "additions"
                                            },
                                            "addressLine1": "addressLine1",
                                            "addressLine2": "addressLine2",
                                            "postCode": "postCode"
                                        },
                                        "additions": {
                                            "key": "additions"
                                        },
                                        "name": "name",
                                        "bic": "bic"
                                    },
                                    "mandateIdentifier": "mandateIdentifier",
                                    "remittanceInformation": "remittanceInformation",
                                    "messageToBank": "messageToBank",
                                    "counterparty": {
                                        "postalAddress": {
                                            "country": "country",
                                            "streetName": "streetName",
                                            "town": "town",
                                            "countrySubDivision": "countrySubDivision",
                                            "additions": {
                                                "key": "additions"
                                            },
                                            "addressLine1": "addressLine1",
                                            "addressLine2": "addressLine2",
                                            "postCode": "postCode"
                                        },
                                        "additions": {
                                            "key": "additions"
                                        },
                                        "name": "name",
                                        "recipientId": "recipientId"
                                    },
                                    "instructedAmount": {
                                        "amount": "amount",
                                        "additions": {
                                            "key": "additions"
                                        },
                                        "currencyCode": "currencyCode"
                                    },
                                    "endToEndIdentification": "endToEndIdentification"
                                },
                                "additions": {
                                    "key": "additions"
                                },
                                "originatorAccount": {
                                    "arrangementId": "arrangementId",
                                    "identification": {
                                        "identification": "identification",
                                        "additions": {
                                            "key": "additions"
                                        }
                                    },
                                    "additions": {
                                        "key": "additions"
                                    },
                                    "name": "name",
                                    "externalArrangementId": "externalArrangementId"
                                },
                                "entryClass": "entryClass",
                                "paymentType": "paymentType"
                            },
                            "id": "id",
                            "actions": [null, null]
                        },
                        "additions": {
                            "key": "additions"
                        }
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
            ]
        },
    ]);

    exports.Action = Action;
    exports.COLLECTION_FORMATS = COLLECTION_FORMATS;
    exports.CONFIG_TOKEN = CONFIG_TOKEN;
    exports.ChargeBearer = ChargeBearer;
    exports.InstructionPriority = InstructionPriority;
    exports.InvolvedPartyRole = InvolvedPartyRole;
    exports.PAYMENT_TEMPLATE_BASE_PATH = PAYMENT_TEMPLATE_BASE_PATH;
    exports.PaymentTemplateApiModule = PaymentTemplateApiModule;
    exports.PaymentTemplateConfiguration = PaymentTemplateConfiguration;
    exports.PaymentTemplatesHttpService = PaymentTemplatesHttpService;
    exports.PaymentTemplatesHttpServiceDeletePaymentTemplateIdByPaymentTemplateIdMocksProvider = PaymentTemplatesHttpServiceDeletePaymentTemplateIdByPaymentTemplateIdMocksProvider;
    exports.PaymentTemplatesHttpServiceGetPaymentTemplatesMocksProvider = PaymentTemplatesHttpServiceGetPaymentTemplatesMocksProvider;
    exports.PaymentTemplatesHttpServiceMocksProvider = PaymentTemplatesHttpServiceMocksProvider;
    exports.PaymentTemplatesHttpServicePostPaymentTemplatesMocksProvider = PaymentTemplatesHttpServicePostPaymentTemplatesMocksProvider;
    exports.PaymentTemplatesHttpServicePutPaymentTemplateByIdMocksProvider = PaymentTemplatesHttpServicePutPaymentTemplateByIdMocksProvider;
    exports.SchemeNames = SchemeNames;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
