// @backbase/manage-upcoming-and-historical-payments-journey-ang/esm2015/lib/manage-upcoming-and-historical-payments-journey.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { provideRoutes, RouterModule } from '@angular/router';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { AmountModule } from '@backbase/ui-ang/amount';
import { BadgeModule } from '@backbase/ui-ang/badge';
import { ButtonModule } from '@backbase/ui-ang/button';
import { EllipsisModule } from '@backbase/ui-ang/ellipsis';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { HeaderModule } from '@backbase/ui-ang/header';
import { IconModule } from '@backbase/ui-ang/icon';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { ModalModule } from '@backbase/ui-ang/modal';
import { PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import { TabModule } from '@backbase/ui-ang/tab';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { UpcomingPaymentsCancelPaymentConfirmationComponent } from './components/cancel-payment-confirmation/cancel-payment-confirmation.component';
import { UpcomingPaymentsDetailsListComponent } from './components/details-list/upcoming-payments-details-list.component';
import { UpcomingPaymentsDetailsComponent } from './components/details-modal/upcoming-payments-details.component';
import { UpcomingPaymentsItemGroupComponent } from './components/item-group/upcoming-payments-item-group.component';
import { UpcomingPaymentsItemComponent } from './components/item/upcoming-payments-item.component';
import { UpcomingPaymentsListComponent } from './components/list/upcoming-payments-list.component';
import { PaymentsListContainerComponent } from './components/payment-list-container/payments-list-container.component';
import { ManageUpcomingAndHistoricalPaymentsJourneyComponent } from './manage-upcoming-and-historical-payments-journey.component';
import { ManageUpcomingAndHistoricalPaymentsJourneyConfigService } from './services/config.service';
import { ManageUpcomingAndHistoricalPaymentsNavigationService } from './services/navigation.service';
import { HistoryPaymentsViewComponent } from './views/history-payments-view/history-payments-view.component';
import { UpcomingAndHistoryPaymentsTabContainerViewComponent } from './views/payments-container-view/upcoming-and-history-payments-tab-container-view.component';
import { UpcomingPaymentsViewComponent } from './views/upcoming-payments-view/upcoming-payments-view.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const defaultRoute = {
    path: '',
    component: ManageUpcomingAndHistoricalPaymentsJourneyComponent,
    children: [
        {
            path: '',
            component: UpcomingPaymentsViewComponent,
        },
    ],
};
const components = [
    ManageUpcomingAndHistoricalPaymentsJourneyComponent,
    UpcomingPaymentsViewComponent,
    HistoryPaymentsViewComponent,
    UpcomingAndHistoryPaymentsTabContainerViewComponent,
    PaymentsListContainerComponent,
    UpcomingPaymentsListComponent,
    UpcomingPaymentsItemGroupComponent,
    UpcomingPaymentsItemComponent,
    UpcomingPaymentsDetailsComponent,
    UpcomingPaymentsDetailsListComponent,
    UpcomingPaymentsCancelPaymentConfirmationComponent,
];
const uiModules = [
    TabModule,
    IconModule,
    AmountModule,
    LoadButtonModule,
    EmptyStateModule,
    LoadingIndicatorModule,
    ModalModule,
    ButtonModule,
    HeaderModule,
    PaymentCardNumberModule,
    TooltipModule,
    EllipsisModule,
    BadgeModule,
];
export class ManageUpcomingAndHistoricalPaymentsJourneyModule {
    static forRoot(data = { route: defaultRoute }) {
        return {
            ngModule: ManageUpcomingAndHistoricalPaymentsJourneyModule,
            providers: [provideRoutes([data.route])],
        };
    }
}
/** @nocollapse */ ManageUpcomingAndHistoricalPaymentsJourneyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManageUpcomingAndHistoricalPaymentsJourneyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ ManageUpcomingAndHistoricalPaymentsJourneyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManageUpcomingAndHistoricalPaymentsJourneyModule, declarations: [ManageUpcomingAndHistoricalPaymentsJourneyComponent,
        UpcomingPaymentsViewComponent,
        HistoryPaymentsViewComponent,
        UpcomingAndHistoryPaymentsTabContainerViewComponent,
        PaymentsListContainerComponent,
        UpcomingPaymentsListComponent,
        UpcomingPaymentsItemGroupComponent,
        UpcomingPaymentsItemComponent,
        UpcomingPaymentsDetailsComponent,
        UpcomingPaymentsDetailsListComponent,
        UpcomingPaymentsCancelPaymentConfirmationComponent], imports: [CommonModule,
        RouterModule, i1.BackbaseCoreModule, TabModule,
        IconModule,
        AmountModule,
        LoadButtonModule,
        EmptyStateModule,
        LoadingIndicatorModule,
        ModalModule,
        ButtonModule,
        HeaderModule,
        PaymentCardNumberModule,
        TooltipModule,
        EllipsisModule,
        BadgeModule], exports: [TabModule,
        IconModule,
        AmountModule,
        LoadButtonModule,
        EmptyStateModule,
        LoadingIndicatorModule,
        ModalModule,
        ButtonModule,
        HeaderModule,
        PaymentCardNumberModule,
        TooltipModule,
        EllipsisModule,
        BadgeModule, ManageUpcomingAndHistoricalPaymentsJourneyComponent,
        UpcomingPaymentsViewComponent,
        HistoryPaymentsViewComponent,
        UpcomingAndHistoryPaymentsTabContainerViewComponent,
        PaymentsListContainerComponent,
        UpcomingPaymentsListComponent,
        UpcomingPaymentsItemGroupComponent,
        UpcomingPaymentsItemComponent,
        UpcomingPaymentsDetailsComponent,
        UpcomingPaymentsDetailsListComponent,
        UpcomingPaymentsCancelPaymentConfirmationComponent] });
/** @nocollapse */ ManageUpcomingAndHistoricalPaymentsJourneyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManageUpcomingAndHistoricalPaymentsJourneyModule, providers: [
        ManageUpcomingAndHistoricalPaymentsJourneyConfigService,
        ManageUpcomingAndHistoricalPaymentsNavigationService,
    ], imports: [[
            CommonModule,
            RouterModule,
            BackbaseCoreModule.withConfig({
                classMap: {
                    ManageUpcomingAndHistoricalPaymentsJourneyComponent,
                },
            }),
            ...uiModules,
        ], TabModule,
        IconModule,
        AmountModule,
        LoadButtonModule,
        EmptyStateModule,
        LoadingIndicatorModule,
        ModalModule,
        ButtonModule,
        HeaderModule,
        PaymentCardNumberModule,
        TooltipModule,
        EllipsisModule,
        BadgeModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManageUpcomingAndHistoricalPaymentsJourneyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components],
                    imports: [
                        CommonModule,
                        RouterModule,
                        BackbaseCoreModule.withConfig({
                            classMap: {
                                ManageUpcomingAndHistoricalPaymentsJourneyComponent,
                            },
                        }),
                        ...uiModules,
                    ],
                    providers: [
                        ManageUpcomingAndHistoricalPaymentsJourneyConfigService,
                        ManageUpcomingAndHistoricalPaymentsNavigationService,
                    ],
                    exports: [...uiModules, ...components],
                }]
        }] });
