// @backbase/manage-upcoming-and-historical-payments-journey-ang/esm2015/lib/manage-upcoming-and-historical-payments-journey.component.js
import { Component } from '@angular/core';
import '@angular/localize/init';
import { ManageUpcomingAndHistoricalPaymentsJourneyConfigService } from './services/config.service';
import * as i0 from "@angular/core";
import * as i1 from "./services/config.service";
import * as i2 from "@backbase/ui-ang/header";
import * as i3 from "@angular/router";
export class ManageUpcomingAndHistoricalPaymentsJourneyComponent {
    constructor(configService) {
        this.configService = configService;
    }
    get title() {
        return $localize `:@@upcoming-history-payments-journey-title:${this.configService.journeyTitle}`;
    }
}
/** @nocollapse */ ManageUpcomingAndHistoricalPaymentsJourneyComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManageUpcomingAndHistoricalPaymentsJourneyComponent, deps: [{ token: i1.ManageUpcomingAndHistoricalPaymentsJourneyConfigService }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ ManageUpcomingAndHistoricalPaymentsJourneyComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ManageUpcomingAndHistoricalPaymentsJourneyComponent, selector: "bb-manage-upcoming-and-historical-payments-journey", ngImport: i0, template: `
    <ng-container>
      <bb-header-ui
        headingType="h2"
        headingClasses="bb-heading-1 bb-block bb-block--xl"
        [heading]="title"
        i18n-heading="@@upcoming-and-history-payments-journey.header"
      ></bb-header-ui>
      <router-outlet></router-outlet>
    </ng-container>
  `, isInline: true, components: [{ type: i2.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }], directives: [{ type: i3.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManageUpcomingAndHistoricalPaymentsJourneyComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-manage-upcoming-and-historical-payments-journey',
                    template: `
    <ng-container>
      <bb-header-ui
        headingType="h2"
        headingClasses="bb-heading-1 bb-block bb-block--xl"
        [heading]="title"
        i18n-heading="@@upcoming-and-history-payments-journey.header"
      ></bb-header-ui>
      <router-outlet></router-outlet>
    </ng-container>
  `,
                }]
        }], ctorParameters: function () { return [{ type: i1.ManageUpcomingAndHistoricalPaymentsJourneyConfigService }]; } });
