// @backbase/account-statement-common-ang/esm2015/src/directives/account-statement-download-doc/account-statement-download-doc.directive.js
import { Directive, EventEmitter, HostListener, Input, Output } from '@angular/core';
import { isQueryStringExist } from '../../helpers/utils';
import * as i0 from "@angular/core";
import * as i1 from "../../services/account-statement-dom-service";
// @dynamic to suppress error with resolving Window type during compilation
export class AccountStatementDownloadDocDirective {
    constructor(domService, renderer) {
        this.domService = domService;
        this.renderer = renderer;
        /**
         * document item which is available for downloading
         */
        this.doc = {};
        /**
         * custom filename for document
         */
        this.filename = '';
        // eslint-disable-next-line @angular-eslint/no-output-rename
        this.downloadFailure = new EventEmitter();
    }
    onClick() {
        this.downloadDoc(this.doc);
    }
    downloadDoc(doc) {
        if (doc.url) {
            const url = isQueryStringExist(doc.url) ? `${doc.url}&inline=false` : `${doc.url}?inline=false`;
            this.domService.simulateLinkClick({
                href: url,
            }, this.renderer);
        }
    }
}
AccountStatementDownloadDocDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementDownloadDocDirective, deps: [{ token: i1.AccountStatementDomService }, { token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Directive });
AccountStatementDownloadDocDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.15", type: AccountStatementDownloadDocDirective, selector: "[bbAccountStatementDownloadDoc]", inputs: { doc: ["bbAccountStatementDownloadDoc", "doc"], filename: ["bbAccountStatementDownloadDocFilename", "filename"] }, outputs: { downloadFailure: "bbAccountStatementDownloadDocFailure" }, host: { listeners: { "click": "onClick()", "keydown.enter": "onClick()", "keydown.space": "onClick()" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementDownloadDocDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbAccountStatementDownloadDoc]',
                }]
        }], ctorParameters: function () { return [{ type: i1.AccountStatementDomService }, { type: i0.Renderer2 }]; }, propDecorators: { doc: [{
                type: Input,
                args: ['bbAccountStatementDownloadDoc']
            }], filename: [{
                type: Input,
                args: ['bbAccountStatementDownloadDocFilename']
            }], downloadFailure: [{
                type: Output,
                args: ['bbAccountStatementDownloadDocFailure']
            }], onClick: [{
                type: HostListener,
                args: ['click']
            }, {
                type: HostListener,
                args: ['keydown.enter']
            }, {
                type: HostListener,
                args: ['keydown.space']
            }] } });
