// @backbase/ui-ang/esm2015/_example/input-validation-message/showcase/input-validation-message-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { InputValidationMessageShowcaseComponent } from './input-validation-message-showcase.component';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import * as i0 from "@angular/core";
export class InputValidationMessageShowcaseModule {
}
InputValidationMessageShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputValidationMessageShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputValidationMessageShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputValidationMessageShowcaseModule, declarations: [InputValidationMessageShowcaseComponent], imports: [CommonModule, InputValidationMessageModule, ReactiveFormsModule, InputTextModule] });
InputValidationMessageShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputValidationMessageShowcaseModule, imports: [[CommonModule, InputValidationMessageModule, ReactiveFormsModule, InputTextModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputValidationMessageShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, InputValidationMessageModule, ReactiveFormsModule, InputTextModule],
                    declarations: [InputValidationMessageShowcaseComponent],
                    entryComponents: [InputValidationMessageShowcaseComponent],
                }]
        }] });
