// @backbase/ui-ang/esm2015/layout/layout.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { MediaQueryModule } from '@backbase/ui-ang/media-query-lib';
import { LAYOUT_CONFIG_TOKEN } from './bb-layout-lib/layout.model';
import { LayoutNavDropdownDirective } from './bb-layout-nav-dropdown-directive/layout-nav-dropdown.directive';
import * as i0 from "@angular/core";
export class LayoutModule {
    static forRoot(config) {
        return { ngModule: LayoutModule, providers: [{ provide: LAYOUT_CONFIG_TOKEN, useValue: config }] };
    }
}
LayoutModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LayoutModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LayoutModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LayoutModule, declarations: [LayoutNavDropdownDirective], imports: [CommonModule, MediaQueryModule], exports: [LayoutNavDropdownDirective] });
LayoutModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LayoutModule, imports: [[CommonModule, MediaQueryModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LayoutModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, MediaQueryModule],
                    declarations: [LayoutNavDropdownDirective],
                    exports: [LayoutNavDropdownDirective],
                }]
        }] });
