// @backbase/ui-ang/esm2015/switch/switch.module.js
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { NgModule } from '@angular/core';
import { SwitchComponent } from './switch.component';
import { InputCheckboxModule } from '@backbase/ui-ang/input-checkbox';
import * as i0 from "@angular/core";
export class SwitchModule {
}
SwitchModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SwitchModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
SwitchModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SwitchModule, declarations: [SwitchComponent], imports: [CommonModule, FormsModule, InputCheckboxModule], exports: [SwitchComponent] });
SwitchModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SwitchModule, imports: [[CommonModule, FormsModule, InputCheckboxModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SwitchModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, FormsModule, InputCheckboxModule],
                    declarations: [SwitchComponent],
                    exports: [SwitchComponent],
                }]
        }] });
