// @backbase/pocket-tailor-http-ang/esm2015/api/pocketTailorClient.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const PocketTailorClientHttpServiceClosePocketMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/pockets/closed",
        method: "POST",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const PocketTailorClientHttpServiceListPocketsMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/pockets",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "pockets": [{
                            "id": "546abc12",
                            "arrangementId": "11da7d70-ed15-11ea-adc1-0242ac120002",
                            "name": "Travel Europe",
                            "icon": "travel",
                            "goal": {
                                "amountCurrency": {
                                    "amount": "2000.00",
                                    "currencyCode": "EUR"
                                },
                                "deadline": "2021-02-28",
                                "progress": 35.5
                            },
                            "balance": {
                                "amount": "-20.00",
                                "currencyCode": "EUR"
                            }
                        }, {
                            "id": "gy2345dff",
                            "arrangementId": "3cdb2224-8926-4b4d-a99f-012331c94699",
                            "name": "Christmas gifts",
                            "icon": "gift",
                            "balance": {
                                "amount": "789.13",
                                "currencyCode": "EUR"
                            }
                        }, {
                            "id": "cd1235dff",
                            "arrangementId": "2331c9469-09a8-4b4d-a91c946999f-01233",
                            "name": "Adventure holidays",
                            "icon": "holidays",
                            "goal": {
                                "deadline": "2021-09-18"
                            },
                            "balance": {
                                "amount": "89.13",
                                "currencyCode": "EUR"
                            }
                        }, {
                            "id": "987abc12",
                            "arrangementId": "c7dgh-a7d70-ed15-11ea-fec1-0242120002",
                            "name": "Home Renovation",
                            "icon": "home",
                            "goal": {
                                "amountCurrency": {
                                    "amount": "50000.00",
                                    "currencyCode": "EUR"
                                },
                                "progress": 0.5
                            },
                            "balance": {
                                "amount": "250.00",
                                "currencyCode": "EUR"
                            }
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const PocketTailorClientHttpServicePostPocketMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/pockets",
        method: "POST",
        responses: [
            {
                status: 201,
                body: {
                    "id": "546abc12",
                    "arrangementId": "11da7d70-ed15-11ea-adc1-0242ac120002",
                    "name": "Travel Europe",
                    "icon": "travel",
                    "goal": {
                        "amountCurrency": {
                            "amount": "2000.00",
                            "currencyCode": "EUR"
                        },
                        "deadline": "2021-02-28",
                        "progress": 35.5
                    },
                    "balance": {
                        "amount": "-20.00",
                        "currencyCode": "EUR"
                    }
                }
            },
            {
                status: 201,
                body: {
                    "arrangementId": "arrangementId",
                    "goal": {
                        "progress": 0.8008281904610115,
                        "deadline": "2000-01-23",
                        "amountCurrency": {
                            "amount": "amount",
                            "additions": {
                                "key": "additions"
                            },
                            "currencyCode": "currencyCode"
                        }
                    },
                    "balance": {
                        "amount": "amount",
                        "additions": {
                            "key": "additions"
                        },
                        "currencyCode": "currencyCode"
                    },
                    "name": "name",
                    "icon": "icon",
                    "id": "id"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const PocketTailorClientHttpServiceViewPocketMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/pockets/{pocketId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "546abc12",
                    "arrangementId": "11da7d70-ed15-11ea-adc1-0242ac120002",
                    "name": "Travel Europe",
                    "icon": "travel",
                    "goal": {
                        "amountCurrency": {
                            "amount": "2000.00",
                            "currencyCode": "EUR"
                        },
                        "deadline": "2021-02-28",
                        "progress": 35.5
                    },
                    "balance": {
                        "amount": "-20.00",
                        "currencyCode": "EUR"
                    }
                }
            },
            {
                status: 200,
                body: {
                    "arrangementId": "arrangementId",
                    "goal": {
                        "progress": 0.8008281904610115,
                        "deadline": "2000-01-23",
                        "amountCurrency": {
                            "amount": "amount",
                            "additions": {
                                "key": "additions"
                            },
                            "currencyCode": "currencyCode"
                        }
                    },
                    "balance": {
                        "amount": "amount",
                        "additions": {
                            "key": "additions"
                        },
                        "currencyCode": "currencyCode"
                    },
                    "name": "name",
                    "icon": "icon",
                    "id": "id"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const PocketTailorClientHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v2/pockets/closed",
        method: "POST",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/pockets",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "pockets": [{
                            "id": "546abc12",
                            "arrangementId": "11da7d70-ed15-11ea-adc1-0242ac120002",
                            "name": "Travel Europe",
                            "icon": "travel",
                            "goal": {
                                "amountCurrency": {
                                    "amount": "2000.00",
                                    "currencyCode": "EUR"
                                },
                                "deadline": "2021-02-28",
                                "progress": 35.5
                            },
                            "balance": {
                                "amount": "-20.00",
                                "currencyCode": "EUR"
                            }
                        }, {
                            "id": "gy2345dff",
                            "arrangementId": "3cdb2224-8926-4b4d-a99f-012331c94699",
                            "name": "Christmas gifts",
                            "icon": "gift",
                            "balance": {
                                "amount": "789.13",
                                "currencyCode": "EUR"
                            }
                        }, {
                            "id": "cd1235dff",
                            "arrangementId": "2331c9469-09a8-4b4d-a91c946999f-01233",
                            "name": "Adventure holidays",
                            "icon": "holidays",
                            "goal": {
                                "deadline": "2021-09-18"
                            },
                            "balance": {
                                "amount": "89.13",
                                "currencyCode": "EUR"
                            }
                        }, {
                            "id": "987abc12",
                            "arrangementId": "c7dgh-a7d70-ed15-11ea-fec1-0242120002",
                            "name": "Home Renovation",
                            "icon": "home",
                            "goal": {
                                "amountCurrency": {
                                    "amount": "50000.00",
                                    "currencyCode": "EUR"
                                },
                                "progress": 0.5
                            },
                            "balance": {
                                "amount": "250.00",
                                "currencyCode": "EUR"
                            }
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/pockets",
        method: "POST",
        responses: [
            {
                status: 201,
                body: {
                    "id": "546abc12",
                    "arrangementId": "11da7d70-ed15-11ea-adc1-0242ac120002",
                    "name": "Travel Europe",
                    "icon": "travel",
                    "goal": {
                        "amountCurrency": {
                            "amount": "2000.00",
                            "currencyCode": "EUR"
                        },
                        "deadline": "2021-02-28",
                        "progress": 35.5
                    },
                    "balance": {
                        "amount": "-20.00",
                        "currencyCode": "EUR"
                    }
                }
            },
            {
                status: 201,
                body: {
                    "arrangementId": "arrangementId",
                    "goal": {
                        "progress": 0.8008281904610115,
                        "deadline": "2000-01-23",
                        "amountCurrency": {
                            "amount": "amount",
                            "additions": {
                                "key": "additions"
                            },
                            "currencyCode": "currencyCode"
                        }
                    },
                    "balance": {
                        "amount": "amount",
                        "additions": {
                            "key": "additions"
                        },
                        "currencyCode": "currencyCode"
                    },
                    "name": "name",
                    "icon": "icon",
                    "id": "id"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/pockets/{pocketId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "546abc12",
                    "arrangementId": "11da7d70-ed15-11ea-adc1-0242ac120002",
                    "name": "Travel Europe",
                    "icon": "travel",
                    "goal": {
                        "amountCurrency": {
                            "amount": "2000.00",
                            "currencyCode": "EUR"
                        },
                        "deadline": "2021-02-28",
                        "progress": 35.5
                    },
                    "balance": {
                        "amount": "-20.00",
                        "currencyCode": "EUR"
                    }
                }
            },
            {
                status: 200,
                body: {
                    "arrangementId": "arrangementId",
                    "goal": {
                        "progress": 0.8008281904610115,
                        "deadline": "2000-01-23",
                        "amountCurrency": {
                            "amount": "amount",
                            "additions": {
                                "key": "additions"
                            },
                            "currencyCode": "currencyCode"
                        }
                    },
                    "balance": {
                        "amount": "amount",
                        "additions": {
                            "key": "additions"
                        },
                        "currencyCode": "currencyCode"
                    },
                    "name": "name",
                    "icon": "icon",
                    "id": "id"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
]);
