// @backbase/income-spending-analysis-journey-ang/esm2015/src/views/income-spending-analysis-view.component.js
import { Component } from '@angular/core';
import { AnalysisType } from '../income-spending-analysis-journey-config.service';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "../income-spending-analysis-journey-config.service";
import * as i3 from "@backbase/ui-ang/header";
import * as i4 from "@backbase/product-summary-account-selector-widget-ang";
import * as i5 from "@backbase/income-spending-analysis-widget-ang";
import * as i6 from "@angular/common";
export class IncomeSpendingAnalysisViewComponent {
    constructor(router, route, config) {
        this.router = router;
        this.route = route;
        this.config = config;
        this.isSpendingAnalysis = this.config.analysisType === AnalysisType.SPENDING;
    }
    onItemSelect(selectedAccount) {
        this.router.navigate(['.', { selectedAccount }], { relativeTo: this.route });
    }
}
IncomeSpendingAnalysisViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IncomeSpendingAnalysisViewComponent, deps: [{ token: i1.Router }, { token: i1.ActivatedRoute }, { token: i2.IncomeSpendingAnalysisJourneyService }], target: i0.ɵɵFactoryTarget.Component });
IncomeSpendingAnalysisViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: IncomeSpendingAnalysisViewComponent, selector: "bb-income-spending-analysis-view", ngImport: i0, template: `
    <bb-header-ui
      *ngIf="isSpendingAnalysis"
      headingType="h1"
      headingClasses="bb-heading-widget__heading px-0 pb-5"
      heading="Spending Analysis"
      i18n-heading="Spending analysis|Heading for Spending analysis journey@@spending.analysis.journey.heading.text"
    ></bb-header-ui>
    <bb-header-ui
      *ngIf="!isSpendingAnalysis"
      headingType="h1"
      headingClasses="bb-heading-widget__heading px-0 pb-5"
      heading="Income Analysis"
      i18n-heading="Income analysis|Heading for Income analysis journey@@income.analysis.journey.heading.text"
    ></bb-header-ui>
    <div class="bb-block--xl">
      <bb-product-summary-account-selector-widget
        (selectedAccount)="onItemSelect($event)"
      ></bb-product-summary-account-selector-widget>
    </div>
    <div>
      <bb-income-spending-analysis-widget [analysisType]="config.analysisType"></bb-income-spending-analysis-widget>
    </div>
  `, isInline: true, components: [{ type: i3.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i4.ProductSummaryAccountSelectorWidgetComponent, selector: "bb-product-summary-account-selector-widget", inputs: ["event.refreshAccountsList", "multiple", "filterItems", "showAccountState"], outputs: ["selectedAccount", "selectedKind"] }, { type: i5.IncomeSpendingAnalysisWidgetComponent, selector: "bb-income-spending-analysis-widget", inputs: ["analysisType"] }], directives: [{ type: i6.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IncomeSpendingAnalysisViewComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-income-spending-analysis-view',
                    template: `
    <bb-header-ui
      *ngIf="isSpendingAnalysis"
      headingType="h1"
      headingClasses="bb-heading-widget__heading px-0 pb-5"
      heading="Spending Analysis"
      i18n-heading="Spending analysis|Heading for Spending analysis journey@@spending.analysis.journey.heading.text"
    ></bb-header-ui>
    <bb-header-ui
      *ngIf="!isSpendingAnalysis"
      headingType="h1"
      headingClasses="bb-heading-widget__heading px-0 pb-5"
      heading="Income Analysis"
      i18n-heading="Income analysis|Heading for Income analysis journey@@income.analysis.journey.heading.text"
    ></bb-header-ui>
    <div class="bb-block--xl">
      <bb-product-summary-account-selector-widget
        (selectedAccount)="onItemSelect($event)"
      ></bb-product-summary-account-selector-widget>
    </div>
    <div>
      <bb-income-spending-analysis-widget [analysisType]="config.analysisType"></bb-income-spending-analysis-widget>
    </div>
  `,
                }]
        }], ctorParameters: function () { return [{ type: i1.Router }, { type: i1.ActivatedRoute }, { type: i2.IncomeSpendingAnalysisJourneyService }]; } });
