// @backbase/cards-list-widget-ang/esm2015/src/cards-list-widget.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { CardsCommonModule } from '@backbase/cards-common-ang';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { IconModule } from '@backbase/ui-ang/icon';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { PaymentCardModule } from '@backbase/ui-ang/payment-card';
import { CardsListPaymentCardControlsCustomizableDirective, PaymentCardControlsComponent, } from './components/bb-payment-card-controls/payment-card-controls.component';
import { PaymentCardGroupComponent } from './components/bb-payment-card-group/payment-card-group.component';
import { CardsListPaymentCardCustomizableDirective, PaymentCardComponent, } from './components/bb-payment-card/payment-card.component';
import { CardsListPaymentCardsContainerCustomizableDirective, CardsListPaymentCardsEmptyStateContainerCustomizableDirective, PaymentCardsContainerComponent, } from './components/bb-payment-cards-container/payment-cards-container.component';
import { CardsListWidgetComponent } from './components/cards-list-widget/cards-list-widget.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const modules = [
    PaymentCardModule,
    LoadingIndicatorModule,
    LoadButtonModule,
    IconModule,
    EmptyStateModule,
    CardsCommonModule,
];
const components = [
    CardsListWidgetComponent,
    PaymentCardComponent,
    PaymentCardControlsComponent,
    PaymentCardGroupComponent,
    PaymentCardsContainerComponent,
];
const customizables = [
    CardsListPaymentCardCustomizableDirective,
    CardsListPaymentCardControlsCustomizableDirective,
    CardsListPaymentCardsContainerCustomizableDirective,
    CardsListPaymentCardsEmptyStateContainerCustomizableDirective,
];
export class CardsListWidgetModule {
}
CardsListWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardsListWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CardsListWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardsListWidgetModule, declarations: [CardsListWidgetComponent,
        PaymentCardComponent,
        PaymentCardControlsComponent,
        PaymentCardGroupComponent,
        PaymentCardsContainerComponent, CardsListPaymentCardCustomizableDirective,
        CardsListPaymentCardControlsCustomizableDirective,
        CardsListPaymentCardsContainerCustomizableDirective,
        CardsListPaymentCardsEmptyStateContainerCustomizableDirective], imports: [CommonModule, i1.BackbaseCoreModule, PaymentCardModule,
        LoadingIndicatorModule,
        LoadButtonModule,
        IconModule,
        EmptyStateModule,
        CardsCommonModule], exports: [PaymentCardModule,
        LoadingIndicatorModule,
        LoadButtonModule,
        IconModule,
        EmptyStateModule,
        CardsCommonModule, CardsListWidgetComponent,
        PaymentCardComponent,
        PaymentCardControlsComponent,
        PaymentCardGroupComponent,
        PaymentCardsContainerComponent, CardsListPaymentCardCustomizableDirective,
        CardsListPaymentCardControlsCustomizableDirective,
        CardsListPaymentCardsContainerCustomizableDirective,
        CardsListPaymentCardsEmptyStateContainerCustomizableDirective] });
CardsListWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardsListWidgetModule, providers: [], imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { CardsListWidgetComponent },
            }),
            ...modules,
        ], PaymentCardModule,
        LoadingIndicatorModule,
        LoadButtonModule,
        IconModule,
        EmptyStateModule,
        CardsCommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardsListWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { CardsListWidgetComponent },
                        }),
                        ...modules,
                    ],
                    providers: [],
                    declarations: [...components, ...customizables],
                    exports: [...modules, ...components, ...customizables],
                }]
        }] });
