// @backbase/ui-ang/esm2015/_example/avatar/showcase/avatar-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AvatarShowcaseComponent } from './avatar-showcase.component';
import { AvatarModule } from '@backbase/ui-ang/avatar';
import * as i0 from "@angular/core";
export class AvatarShowcaseModule {
}
AvatarShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AvatarShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AvatarShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AvatarShowcaseModule, declarations: [AvatarShowcaseComponent], imports: [CommonModule, AvatarModule] });
AvatarShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AvatarShowcaseModule, imports: [[CommonModule, AvatarModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AvatarShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, AvatarModule],
                    declarations: [AvatarShowcaseComponent],
                    entryComponents: [AvatarShowcaseComponent],
                }]
        }] });
