// @backbase/ui-ang/esm2015/modal/modal-templates/modal-footer.component.js
import { Component, ContentChild, Directive, EventEmitter, Input, Optional, Output } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../modal.component";
import * as i2 from "@angular/common";
export class CustomModalFooterDirective {
    constructor(templateRef) {
        this.templateRef = templateRef;
    }
}
CustomModalFooterDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomModalFooterDirective, deps: [{ token: i0.TemplateRef }], target: i0.ɵɵFactoryTarget.Directive });
CustomModalFooterDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomModalFooterDirective, selector: "ng-template[bbCustomModalFooter]", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomModalFooterDirective, decorators: [{
            type: Directive,
            args: [{ selector: 'ng-template[bbCustomModalFooter]' }]
        }], ctorParameters: function () { return [{ type: i0.TemplateRef }]; } });
export class ModalFooterComponent {
    constructor(modal) {
        this.modal = modal;
        this.hostRef = this;
        /**
         * The event that's fired after confirm button is pressed. Can be used with (click)="dialogRef.onConfirm()"
         * where dialogRef is a templateRef on <bb-dialog-ui #dialogRef />
         */
        this.confirm = new EventEmitter();
        /**
         * The event that's fired after cancel button is pressed. Can be used with (click)="dialogRef.onCancel()"
         * where dialogRef is a templateRef on <bb-dialog-ui #dialogRef />
         */
        this.cancel = new EventEmitter();
    }
    onCancel($event) {
        if (this.modal) {
            this.modal.dismissModal($event);
        }
        else {
            this.cancel.emit($event);
        }
    }
    onConfirm($event) {
        if (this.modal) {
            this.modal.closeModal($event);
        }
        else {
            this.confirm.emit($event);
        }
    }
}
ModalFooterComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ModalFooterComponent, deps: [{ token: i1.ModalComponent, optional: true }], target: i0.ɵɵFactoryTarget.Component });
ModalFooterComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ModalFooterComponent, selector: "bb-modal-footer-ui", inputs: { confirmText: "confirmText", cancelText: "cancelText" }, outputs: { confirm: "confirm", cancel: "cancel" }, queries: [{ propertyName: "customModalFooter", first: true, predicate: CustomModalFooterDirective, descendants: true, static: true }], ngImport: i0, template: "<div class=\"modal-footer\" [attr.data-role]=\"'modal-footer'\">\n  <ng-container *ngIf=\"customModalFooter; else bbDefaultModalFooter\">\n    <ng-container *ngTemplateOutlet=\"customModalFooter?.templateRef; context: hostRef\"> </ng-container>\n  </ng-container>\n</div>\n\n<ng-template #bbDefaultModalFooter>\n  <div class=\"bb-button-bar\">\n    <button\n      (click)=\"onConfirm()\"\n      class=\"bb-button-bar__button btn btn-primary btn-md\"\n      type=\"button\"\n      data-role=\"confirm\"\n    >\n      <ng-container\n        *ngIf=\"!confirmText\"\n        i18n=\"Accept|Agree with the message presented in the dialog @@bb-dialog-ui.yes\"\n      >\n        Confirm\n      </ng-container>\n      <ng-container *ngIf=\"confirmText\">{{ confirmText }}</ng-container>\n    </button>\n    <button\n      (click)=\"onCancel()\"\n      class=\"bb-button-bar__button btn btn-secondary btn-md\"\n      type=\"button\"\n      data-role=\"cancel\"\n    >\n      <ng-container\n        *ngIf=\"!cancelText\"\n        i18n=\"Reject|Do not agree with the message presented in the dialog @@bb-dialog-ui.no\"\n      >\n        Cancel\n      </ng-container>\n      <ng-container *ngIf=\"cancelText\">{{ cancelText }}</ng-container>\n    </button>\n  </div>\n</ng-template>\n", directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i2.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ModalFooterComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-modal-footer-ui',
                    templateUrl: './modal-footer.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.ModalComponent, decorators: [{
                    type: Optional
                }] }]; }, propDecorators: { customModalFooter: [{
                type: ContentChild,
                args: [CustomModalFooterDirective, { static: true }]
            }], confirmText: [{
                type: Input
            }], cancelText: [{
                type: Input
            }], confirm: [{
                type: Output
            }], cancel: [{
                type: Output
            }] } });
