// @backbase/ui-ang/esm2015/modal/modal-templates/modal-header.component.js
import { Component, ContentChild, Directive, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/services";
import * as i2 from "@angular/common";
export class CustomModalHeaderDirective {
    constructor(templateRef) {
        this.templateRef = templateRef;
    }
}
CustomModalHeaderDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomModalHeaderDirective, deps: [{ token: i0.TemplateRef }], target: i0.ɵɵFactoryTarget.Directive });
CustomModalHeaderDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomModalHeaderDirective, selector: "ng-template[bbCustomModalHeader]", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomModalHeaderDirective, decorators: [{
            type: Directive,
            args: [{ selector: 'ng-template[bbCustomModalHeader]' }]
        }], ctorParameters: function () { return [{ type: i0.TemplateRef }]; } });
export class ModalHeaderComponent {
    constructor(domAttrService) {
        this.domAttrService = domAttrService;
        this.hostRef = this;
        /**
         * Title text for the modal header. Defaults to empty string.
         */
        this.title = '';
        /**
         * The event emitted when the modal header close button is clicked.
         */
        this.close = new EventEmitter();
        this.showClose = false;
        this.headerId = this.domAttrService.generateId();
    }
    ngOnInit() {
        this.showClose = this.close.observers.length > 0;
    }
    onClose() {
        this.close.emit();
    }
}
ModalHeaderComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ModalHeaderComponent, deps: [{ token: i1.DomAttributesService }], target: i0.ɵɵFactoryTarget.Component });
ModalHeaderComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ModalHeaderComponent, selector: "bb-modal-header-ui", inputs: { title: "title" }, outputs: { close: "close" }, queries: [{ propertyName: "customModalHeader", first: true, predicate: CustomModalHeaderDirective, descendants: true, static: true }], ngImport: i0, template: "<div class=\"modal-header\" [attr.id]=\"headerId\" [attr.data-role]=\"'modal-header'\">\n  <ng-container *ngIf=\"customModalHeader; else bbDefaultModalHeader\">\n    <ng-container *ngTemplateOutlet=\"customModalHeader?.templateRef; context: hostRef\"> </ng-container>\n  </ng-container>\n</div>\n\n<ng-template #bbDefaultModalHeader>\n  <h3 class=\"modal-title\" [innerText]=\"title\"></h3>\n  <button *ngIf=\"showClose\" class=\"btn btn-unstyled\" (click)=\"onClose()\">\n    <i class=\"bicon bicon-clear bb-icon bb-icon--clear\" aria-hidden=\"true\"></i>\n  </button>\n</ng-template>\n", directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i2.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ModalHeaderComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-modal-header-ui',
                    templateUrl: './modal-header.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.DomAttributesService }]; }, propDecorators: { customModalHeader: [{
                type: ContentChild,
                args: [CustomModalHeaderDirective, { static: true }]
            }], title: [{
                type: Input
            }], close: [{
                type: Output
            }] } });
