// @backbase/ui-ang/esm2015/modal/modal-templates/modal-body.component.js
import { Component, ContentChild, Directive } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class CustomModalBodyDirective {
    constructor(templateRef) {
        this.templateRef = templateRef;
    }
}
CustomModalBodyDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomModalBodyDirective, deps: [{ token: i0.TemplateRef }], target: i0.ɵɵFactoryTarget.Directive });
CustomModalBodyDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomModalBodyDirective, selector: "ng-template[bbCustomModalBody]", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomModalBodyDirective, decorators: [{
            type: Directive,
            args: [{ selector: 'ng-template[bbCustomModalBody]' }]
        }], ctorParameters: function () { return [{ type: i0.TemplateRef }]; } });
export class ModalBodyComponent {
    constructor() {
        this.hostRef = this;
    }
}
ModalBodyComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ModalBodyComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ModalBodyComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ModalBodyComponent, selector: "bb-modal-body-ui", queries: [{ propertyName: "customModalBody", first: true, predicate: CustomModalBodyDirective, descendants: true, static: true }], ngImport: i0, template: "<div class=\"modal-body\" [attr.data-role]=\"'modal-body'\">\n  <ng-container *ngIf=\"customModalBody; else bbDefaultModalBody\">\n    <ng-container *ngTemplateOutlet=\"customModalBody?.templateRef; context: hostRef\"> </ng-container>\n  </ng-container>\n</div>\n\n<ng-template #bbDefaultModalBody> Custom Body </ng-template>\n", directives: [{ type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i1.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ModalBodyComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-modal-body-ui',
                    templateUrl: './modal-body.component.html',
                }]
        }], propDecorators: { customModalBody: [{
                type: ContentChild,
                args: [CustomModalBodyDirective, { static: true }]
            }] } });
