// @backbase/actions-store-ang/esm2015/src/actions-store-model.service.js
import { Injectable } from '@angular/core';
import { loadActionRecipes, saveActionRecipe } from './action-recipes/actions';
import { loadSpecifications } from './specifications/actions';
import { specificationsError, specificationsLoading, specifications } from './specifications/selectors';
import { actionRecipes, actionRecipesError, actionRecipesLoading } from './action-recipes/selectors';
import { specificationsWithActionRecipe } from './selectors';
import * as i0 from "@angular/core";
import * as i1 from "@ngrx/store";
export class ActionsStoreModel {
    constructor(store) {
        this.store = store;
        /**
         * Stream of specifications.
         */
        this.specifications$ = this.store.select(specifications);
        /**
         * Stream of specifications loading flags.
         */
        this.specificationsLoading$ = this.store.select(specificationsLoading);
        /**
         * Stream of specifications errors.
         */
        this.specificationsError$ = this.store.select(specificationsError);
        /**
         * Stream of action recipes.
         */
        this.actionRecipes$ = this.store.select(actionRecipes);
        /**
         * Stream of action recipes loading flags.
         */
        this.actionRecipesLoading$ = this.store.select(actionRecipesLoading);
        /**
         * Stream of action recipes errors.
         */
        this.actionRecipesError$ = this.store.select(actionRecipesError);
        /**
         * Stream of specifications with action recipe.
         */
        this.specificationsWithActionRecipe$ = this.store.select(specificationsWithActionRecipe);
    }
    /**
     * Method to dispatch load specifications.
     *
     * @param payload - load specifications payload
     */
    dispatchLoadSpecifications(payload) {
        this.store.dispatch(loadSpecifications(payload));
    }
    /**
     * Method to dispatch load action recipes.
     *
     * @param payload - load specifications payload
     */
    dispatchLoadActionRecipes(payload) {
        this.store.dispatch(loadActionRecipes(payload));
    }
    /**
     * Method to dispatch save action recipe.
     *
     * @param payload - action recipe payload
     */
    dispatchSaveActionRecipe(payload) {
        this.store.dispatch(saveActionRecipe(payload));
    }
}
ActionsStoreModel.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsStoreModel, deps: [{ token: i1.Store }], target: i0.ɵɵFactoryTarget.Injectable });
ActionsStoreModel.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsStoreModel });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsStoreModel, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.Store }]; } });
