// @backbase/ui-ang/esm2015/product-item-term-deposit/product-item-term-deposit.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { CustomProductItemTermDepositDetailsFormatterDirective, CustomProductItemTermDepositMainSectionDirective, CustomProductItemTermDepositTitleDirective, ProductItemTermDepositComponent, } from './product-item-term-deposit.component';
import { AmountModule } from '@backbase/ui-ang/amount';
import { PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import { HighlightModule } from '@backbase/ui-ang/highlight';
import { IconModule } from '@backbase/ui-ang/icon';
import * as i0 from "@angular/core";
export class ProductItemTermDepositModule {
}
ProductItemTermDepositModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemTermDepositModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductItemTermDepositModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemTermDepositModule, declarations: [ProductItemTermDepositComponent,
        CustomProductItemTermDepositTitleDirective,
        CustomProductItemTermDepositDetailsFormatterDirective,
        CustomProductItemTermDepositMainSectionDirective], imports: [CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule], exports: [ProductItemTermDepositComponent,
        CustomProductItemTermDepositTitleDirective,
        CustomProductItemTermDepositDetailsFormatterDirective,
        CustomProductItemTermDepositMainSectionDirective] });
ProductItemTermDepositModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemTermDepositModule, imports: [[CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemTermDepositModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule],
                    declarations: [
                        ProductItemTermDepositComponent,
                        CustomProductItemTermDepositTitleDirective,
                        CustomProductItemTermDepositDetailsFormatterDirective,
                        CustomProductItemTermDepositMainSectionDirective,
                    ],
                    exports: [
                        ProductItemTermDepositComponent,
                        CustomProductItemTermDepositTitleDirective,
                        CustomProductItemTermDepositDetailsFormatterDirective,
                        CustomProductItemTermDepositMainSectionDirective,
                    ],
                }]
        }] });
