// @backbase/manage-payees-journey-ang/esm2015/src/views/billpay-create-electronic-payee/billpay-create-electronic-payee-view.component.js
import { Component } from '@angular/core';
import { combineLatest, of } from 'rxjs';
import { map, switchMap, take } from 'rxjs/operators';
import { BaseElectronicPayeePageDirective } from '../../model/base-electronic-payee-page.model';
import { BillpayDiscardChangesModalService } from '@backbase/billpay-journeys-common';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "../../services/billpay-manage-payee-state.service";
import * as i3 from "@backbase/billpay-journeys-common";
import * as i4 from "../../services/manage-payees-navigation.service";
import * as i5 from "../../components/add-payee/billpay-electronic-payee-form/billpay-electronic-payee-form.component";
import * as i6 from "../../components/add-payee/billpay-cancel-new-payee-modal/billpay-cancel-new-payee-modal.component";
import * as i7 from "@angular/common";
export class BillpayCreateElectronicPayeeViewComponent extends BaseElectronicPayeePageDirective {
    /** @internal */
    constructor(route, stateService, discardModalService, router) {
        super(route, stateService, discardModalService);
        this.router = router;
        this.zipRequired$ = this.route.queryParamMap.pipe(take(1), map(params => params.get('zipRequired')), map(zipRequired => (zipRequired ? JSON.parse(zipRequired) : false)));
        this.name$ = this.route.queryParamMap.pipe(take(1), map(params => params.get('name') || ''));
        this.payee$ = this.initializePayee();
    }
    onFormSubmitted(payee) {
        this.stateService.setElectronicPayee(payee);
        this.id$.subscribe(id => this.router.navigateToElectronicPayeeReview(id, { queryParamsHandling: 'preserve' }));
    }
    onBackToSearch() {
        this.isFindPayeeModal = true;
        this.router.navigateToSearch();
    }
    onFormCanceled() {
        this.router.navigateToPayeeList();
    }
    initializePayee() {
        const payee$ = combineLatest([this.zipRequired$, this.name$]).pipe(map(([zipRequired, name]) => (Object.assign({ name, accountNumber: '', nickName: '' }, (zipRequired && { merchantZipCode: '' })))));
        return this.payeeState$.pipe(switchMap(state => (state ? of(state) : payee$)));
    }
}
BillpayCreateElectronicPayeeViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayCreateElectronicPayeeViewComponent, deps: [{ token: i1.ActivatedRoute }, { token: i2.BillpayManagePayeeStateService }, { token: i3.BillpayDiscardChangesModalService }, { token: i4.ManagePayeesNavigationService }], target: i0.ɵɵFactoryTarget.Component });
BillpayCreateElectronicPayeeViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: BillpayCreateElectronicPayeeViewComponent, selector: "ng-component", providers: [BillpayDiscardChangesModalService], usesInheritance: true, ngImport: i0, template: "<ng-container *ngIf=\"payee$ | async as payee\">\n  <bb-billpay-electronic-payee-form\n    [payee]=\"payee\"\n    [zipCodeRequired]=\"zipRequired$ | async\"\n    (formSubmitted)=\"onFormSubmitted($event)\"\n    (formCanceled)=\"onFormCanceled()\"\n    (backToSearch)=\"onBackToSearch()\"\n    (valueChanged)=\"onFormValueChanged()\"\n  >\n  </bb-billpay-electronic-payee-form>\n\n  <bb-billpay-cancel-new-payee-modal\n    *ngIf=\"isDiscardModalOpen\"\n    [findPayee]=\"isFindPayeeModal\"\n    (resolve)=\"onModalResolve($event)\"\n  ></bb-billpay-cancel-new-payee-modal>\n</ng-container>\n", components: [{ type: i5.BillpayElectronicPayeeFormComponent, selector: "bb-billpay-electronic-payee-form", inputs: ["payee", "zipCodeRequired", "backButtonShown"], outputs: ["formSubmitted", "formCanceled", "backToSearch", "valueChanged"] }, { type: i6.BillpayCancelNewPayeeModalComponent, selector: "bb-billpay-cancel-new-payee-modal", inputs: ["findPayee"], outputs: ["resolve"] }], directives: [{ type: i7.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "async": i7.AsyncPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayCreateElectronicPayeeViewComponent, decorators: [{
            type: Component,
            args: [{
                    templateUrl: './billpay-create-electronic-payee-view.component.html',
                    providers: [BillpayDiscardChangesModalService],
                }]
        }], ctorParameters: function () { return [{ type: i1.ActivatedRoute }, { type: i2.BillpayManagePayeeStateService }, { type: i3.BillpayDiscardChangesModalService }, { type: i4.ManagePayeesNavigationService }]; } });
