// @backbase/payment-orders-ang/esm2015/lib/ui/components/payments-search/payments-search.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { PaymentsSearchComponent, PaymentsSearchCustomizableDirective, } from './payments-search.component';
import { SearchBoxModule } from '@backbase/ui-ang/search-box';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import * as i0 from "@angular/core";
export class PaymentsSearchModule {
}
/** @nocollapse */ PaymentsSearchModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentsSearchModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ PaymentsSearchModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentsSearchModule, declarations: [PaymentsSearchComponent, PaymentsSearchCustomizableDirective], imports: [CommonModule, FormsModule, SearchBoxModule, ReactiveFormsModule], exports: [PaymentsSearchComponent, PaymentsSearchCustomizableDirective] });
/** @nocollapse */ PaymentsSearchModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentsSearchModule, imports: [[CommonModule, FormsModule, SearchBoxModule, ReactiveFormsModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentsSearchModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [PaymentsSearchComponent, PaymentsSearchCustomizableDirective],
                    imports: [CommonModule, FormsModule, SearchBoxModule, ReactiveFormsModule],
                    exports: [PaymentsSearchComponent, PaymentsSearchCustomizableDirective],
                }]
        }] });
