// @backbase/payment-orders-ang/esm2015/lib/ui/components/payments-search/payments-search.component.js
import { Component, Directive, EventEmitter, Input, Output, } from '@angular/core';
import { debounceTime, distinctUntilChanged, takeUntil } from 'rxjs/operators';
import { Subject } from 'rxjs';
import { DefaultDebounceTime, SearchMinLength } from '../../constants';
import { BbTemplate } from '@backbase/foundation-ang/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/search-box";
import * as i2 from "@angular/forms";
export class PaymentsSearchComponent {
    constructor() {
        /**
         * Reference to instance of customizable component.
         */
        this.hostRef = this;
        this.gc$ = new Subject();
        this.debounceSearch$ = new Subject();
        this.searchTerm = '';
        /**
         * Time delay between each search requests in milliseconds.
         * Default is 300
         */
        this.debounceTime = DefaultDebounceTime;
        /**
         * Minimum search length
         * Default is 3
         */
        this.searchMinLength = SearchMinLength;
        /**
         * Changes will be triggered on search
         */
        this.search = new EventEmitter();
    }
    ngOnInit() {
        this.debounceSearch$
            .pipe(takeUntil(this.gc$), debounceTime(this.debounceTime), distinctUntilChanged())
            .subscribe((term) => {
            if (!term || term.length >= this.searchMinLength) {
                this.search.emit(term);
            }
        });
    }
    startSearching(query) {
        this.searchTerm = query;
        this.debounceSearch$.next(this.searchTerm);
    }
    clearSearching() {
        this.searchTerm = '';
        this.debounceSearch$.next(this.searchTerm);
    }
    submitSearch() {
        if (this.searchTerm.length >= this.searchMinLength) {
            this.search.emit(this.searchTerm);
        }
    }
    ngOnDestroy() {
        this.gc$.next();
        this.debounceSearch$.complete();
        this.gc$.complete();
    }
}
/** @nocollapse */ PaymentsSearchComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentsSearchComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ PaymentsSearchComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: PaymentsSearchComponent, selector: "bb-payments-search", inputs: { debounceTime: "debounceTime", searchMinLength: "searchMinLength" }, outputs: { search: "search" }, ngImport: i0, template: "<ng-container bbPaymentsSearchCustomizable [bbHostRef]=\"hostRef\"></ng-container>\n<ng-template bbPaymentsSearchCustomizable let-hostRef>\n  <div class=\"payments-search-container\">\n    <bb-search-box-ui\n      placeholder=\"Search\"\n      i18n-placeholder=\"@@payments-list.placeholder.search\"\n      aria-label=\"Search payments\"\n      i18n-aria-label=\"@@payments-list.label.search\"\n      [showSearch]=\"true\"\n      [showClear]=\"true\"\n      (submit)=\"hostRef.submitSearch()\"\n      (clear)=\"hostRef.clearSearching()\"\n      (search)=\"$event.stopPropagation()\"\n      [ngModel]=\"hostRef.searchTerm\"\n      (ngModelChange)=\"hostRef.startSearching($event)\"\n      data-role=\"searchbox\"\n    >\n    </bb-search-box-ui>\n  </div>\n</ng-template>\n", components: [{ type: i0.forwardRef(function () { return i1.SearchBoxComponent; }), selector: "bb-search-box-ui", inputs: ["placeholder", "maxLength", "squareBorder", "typeaheadOptions", "showSearch", "showClear", "searchLabel", "clearLabel", "autocomplete"], outputs: ["submit", "clear", "valueChange"] }], directives: [{ type: i0.forwardRef(function () { return PaymentsSearchCustomizableDirective; }), selector: "[bbPaymentsSearchCustomizable]" }, { type: i0.forwardRef(function () { return i2.NgControlStatus; }), selector: "[formControlName],[ngModel],[formControl]" }, { type: i0.forwardRef(function () { return i2.NgModel; }), selector: "[ngModel]:not([formControlName]):not([formControl])", inputs: ["name", "disabled", "ngModel", "ngModelOptions"], outputs: ["ngModelChange"], exportAs: ["ngModel"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentsSearchComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-payments-search',
                    templateUrl: 'payments-search.component.html',
                }]
        }], propDecorators: { debounceTime: [{
                type: Input
            }], searchMinLength: [{
                type: Input
            }], search: [{
                type: Output
            }] } });
export class PaymentsSearchCustomizableDirective extends BbTemplate {
}
/** @nocollapse */ PaymentsSearchCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentsSearchCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
/** @nocollapse */ PaymentsSearchCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: PaymentsSearchCustomizableDirective, selector: "[bbPaymentsSearchCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentsSearchCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbPaymentsSearchCustomizable]',
                }]
        }] });
