// @backbase/income-spending-analysis-widget-ang/esm2015/src/bb-donut-chart-ui/ngx/pie/pie-label.component.js
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { trimLabel } from '@swimlane/ngx-charts';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class PieLabelComponent {
    constructor() {
        /**
         * Radius of the circle.
         */
        this.radius = 0;
        /**
         * The maximum number of labels displayed.
         */
        this.max = 1;
        /**
         * Value of the label.
         */
        this.value = 0;
        /**
         * Make the radius of each slice proportional to it's value.
         */
        this.explodeSlices = false;
        /**
         * Enable/Disable animations.
         */
        this.animations = true;
        /**
         * Enable/Disable label trimming.
         */
        this.labelTrim = true;
        /**
         * Set the label trimming size.
         */
        this.labelTrimSize = 10;
        this.isIE = /(edge|msie|trident)/i.test(navigator.userAgent);
        this.trimLabel = trimLabel;
    }
    get labelText() {
        return this.labelTrim ? trimLabel(this.label || '', this.labelTrimSize) : this.label;
    }
    get textX() {
        return this.data ? this.data.pos[0] : 0;
    }
    get textY() {
        return this.data ? this.data.pos[1] : 0;
    }
    get textTransition() {
        return this.isIE || !this.animations ? '' : 'transform 0.75s';
    }
    get styleTransform() {
        return this.isIE ? '' : `translate3d(${this.textX}px, ${this.textY}px, 0)`;
    }
    get attrTransform() {
        return !this.isIE ? '' : `translate(${this.textX}, ${this.textY})`;
    }
    ngOnInit() {
        if (typeof this.data === 'undefined') {
            throw new Error(`"data" input is required in "${this.constructor.name}"`);
        }
    }
}
PieLabelComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PieLabelComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
PieLabelComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: PieLabelComponent, selector: "g[bb-charts-pie-label]", inputs: { data: "data", radius: "radius", label: "label", color: "color", icon: "icon", max: "max", value: "value", explodeSlices: "explodeSlices", animations: "animations", labelTrim: "labelTrim", labelTrimSize: "labelTrimSize", template: "template" }, ngImport: i0, template: `
    <ng-container *ngIf="template; else defaultTemplate">
      <ng-template
        [ngTemplateOutlet]="template"
        [ngTemplateOutletContext]="{ data: data, color: color, icon: icon, label: labelText }"
      >
      </ng-template>
    </ng-container>

    <ng-template #defaultTemplate>
      <svg:g [attr.transform]="attrTransform" [style.transform]="styleTransform" [style.transition]="textTransition">
        <svg:text
          class="pie-label"
          [class.animation]="animations"
          dy=".35em"
          [style.textAnchor]="'middle'"
          [style.shapeRendering]="'crispEdges'"
        >
          {{ labelText }}
        </svg:text>
      </svg:g>
    </ng-template>
  `, isInline: true, directives: [{ type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i1.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PieLabelComponent, decorators: [{
            type: Component,
            args: [{
                    // eslint-disable-next-line @angular-eslint/component-selector
                    selector: 'g[bb-charts-pie-label]',
                    template: `
    <ng-container *ngIf="template; else defaultTemplate">
      <ng-template
        [ngTemplateOutlet]="template"
        [ngTemplateOutletContext]="{ data: data, color: color, icon: icon, label: labelText }"
      >
      </ng-template>
    </ng-container>

    <ng-template #defaultTemplate>
      <svg:g [attr.transform]="attrTransform" [style.transform]="styleTransform" [style.transition]="textTransition">
        <svg:text
          class="pie-label"
          [class.animation]="animations"
          dy=".35em"
          [style.textAnchor]="'middle'"
          [style.shapeRendering]="'crispEdges'"
        >
          {{ labelText }}
        </svg:text>
      </svg:g>
    </ng-template>
  `,
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return []; }, propDecorators: { data: [{
                type: Input
            }], radius: [{
                type: Input
            }], label: [{
                type: Input
            }], color: [{
                type: Input
            }], icon: [{
                type: Input
            }], max: [{
                type: Input
            }], value: [{
                type: Input
            }], explodeSlices: [{
                type: Input
            }], animations: [{
                type: Input
            }], labelTrim: [{
                type: Input
            }], labelTrimSize: [{
                type: Input
            }], template: [{
                type: Input
            }] } });
