// @backbase/income-spending-analysis-widget-ang/esm2015/src/bb-donut-chart-ui/ngx/pie/pie-chart.module.js
import { CommonModule, CurrencyPipe } from '@angular/common';
import { NgModule } from '@angular/core';
import { NgxChartsModule } from '@swimlane/ngx-charts';
import { TooltipModule } from '../tooltip/tooltip.module';
import { PieChartComponent } from './pie-chart.component';
import { PieLabelComponent } from './pie-label.component';
import { PieSeriesComponent } from './pie-series.component';
import * as i0 from "@angular/core";
export class PieChartModule {
}
PieChartModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PieChartModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PieChartModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PieChartModule, declarations: [PieChartComponent, PieSeriesComponent, PieLabelComponent], imports: [CommonModule, NgxChartsModule, TooltipModule], exports: [PieChartComponent, NgxChartsModule] });
PieChartModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PieChartModule, providers: [CurrencyPipe], imports: [[CommonModule, NgxChartsModule, TooltipModule], NgxChartsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PieChartModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, NgxChartsModule, TooltipModule],
                    declarations: [PieChartComponent, PieSeriesComponent, PieLabelComponent],
                    exports: [PieChartComponent, NgxChartsModule],
                    providers: [CurrencyPipe],
                    entryComponents: [PieChartComponent],
                }]
        }] });
