// @backbase/manage-pockets-journey-ang/esm2015/src/manage-pockets-journey-config.service.js
import { Inject, Injectable, InjectionToken, Optional } from '@angular/core';
import * as i0 from "@angular/core";
export const ManagePocketsJourneyConfigurationToken = new InjectionToken('ManagePocketsJourneyConfiguration injection token');
const configDefaults = {
    displayHeaders: true,
    journeyCurrency: 'EUR',
    pocketImageList: [
        'home',
        'travel',
        'rent',
        'holidays',
        'savings',
        'utilities',
        'new-car',
        'insurance',
        'new-phone',
        'entertainment',
        'gift',
        'business',
        'custom',
    ],
};
export const getDefaultConfigValues = (config) => (Object.assign(Object.assign({}, configDefaults), (config || {})));
export class ManagePocketsJourneyConfigService {
    constructor(config) {
        this._config = getDefaultConfigValues(config);
    }
    get defaults() {
        return configDefaults;
    }
    /**
     * Defines if the journey display header on each view
     */
    get displayHeaders() {
        return this._config.displayHeaders;
    }
    /**
     * Names for pockets images, used to both name the type and look for the asset
     */
    get pocketImageList() {
        return this._config.pocketImageList;
    }
    /**
     * Currency used across the whole journey
     */
    get journeyCurrency() {
        return this._config.journeyCurrency;
    }
}
ManagePocketsJourneyConfigService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsJourneyConfigService, deps: [{ token: ManagePocketsJourneyConfigurationToken, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
ManagePocketsJourneyConfigService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsJourneyConfigService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsJourneyConfigService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [ManagePocketsJourneyConfigurationToken]
                }] }]; } });
