// @backbase/manage-pockets-journey-ang/esm2015/src/manage-pockets-journey.component.js
import { Component } from '@angular/core';
import { ManagePocketsJourneyConfigService } from './manage-pockets-journey-config.service';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
export class ManagePocketsJourneyComponent {
}
ManagePocketsJourneyComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsJourneyComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ManagePocketsJourneyComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ManagePocketsJourneyComponent, selector: "bb-manage-pockets-journey", providers: [ManagePocketsJourneyConfigService], ngImport: i0, template: '<router-outlet></router-outlet>', isInline: true, directives: [{ type: i1.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsJourneyComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-manage-pockets-journey',
                    template: '<router-outlet></router-outlet>',
                    providers: [ManagePocketsJourneyConfigService],
                }]
        }] });
