// @backbase/manage-pockets-journey-ang/esm2015/src/manage-pockets-journey.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@angular/localize/init';
import { provideRoutes, RouterModule } from '@angular/router';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { HeaderModule } from '@backbase/ui-ang/header';
import { ManagePocketsCreateModule } from './manage-pockets-create/manage-pockets-create.module';
import { ManagePocketsJourneyComponent } from './manage-pockets-journey.component';
import { ManagePocketsListModule } from './manage-pockets-list/manage-pockets-list.module';
import { ManagePocketsTransfersModule } from './manage-pockets-transfers/manage-pockets-transfers.module';
import { ManagePocketsCreateViewComponent } from './views/manage-pockets-create-view.component';
import { ManagePocketsListViewComponent } from './views/manage-pockets-list-view.component';
import { ManagePocketsTransfersViewComponent } from './views/manage-pockets-transfers-view.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
export const defaultRoute = {
    path: '',
    component: ManagePocketsJourneyComponent,
    children: [
        {
            path: '',
            redirectTo: 'list',
            pathMatch: 'full',
        },
        {
            path: 'list',
            data: { title: 'Manage Pockets List' },
            component: ManagePocketsListViewComponent,
        },
        {
            path: 'create',
            data: { title: 'Create Pocket' },
            component: ManagePocketsCreateViewComponent,
        },
        {
            path: 'transfer',
            data: { title: 'Money Transfer' },
            component: ManagePocketsTransfersViewComponent,
        },
    ],
};
export class ManagePocketsJourneyModule {
    static forRoot(data = { route: defaultRoute }) {
        return {
            ngModule: ManagePocketsJourneyModule,
            providers: [provideRoutes([data.route])],
        };
    }
}
ManagePocketsJourneyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsJourneyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ManagePocketsJourneyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsJourneyModule, declarations: [ManagePocketsJourneyComponent,
        ManagePocketsListViewComponent,
        ManagePocketsCreateViewComponent,
        ManagePocketsTransfersViewComponent], imports: [CommonModule, i1.BackbaseCoreModule, RouterModule,
        HeaderModule,
        ManagePocketsListModule,
        ManagePocketsCreateModule,
        ManagePocketsTransfersModule] });
ManagePocketsJourneyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsJourneyModule, imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { ManagePocketsJourneyComponent },
            }),
            RouterModule,
            HeaderModule,
            ManagePocketsListModule,
            ManagePocketsCreateModule,
            ManagePocketsTransfersModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsJourneyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        ManagePocketsJourneyComponent,
                        ManagePocketsListViewComponent,
                        ManagePocketsCreateViewComponent,
                        ManagePocketsTransfersViewComponent,
                    ],
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { ManagePocketsJourneyComponent },
                        }),
                        RouterModule,
                        HeaderModule,
                        ManagePocketsListModule,
                        ManagePocketsCreateModule,
                        ManagePocketsTransfersModule,
                    ],
                }]
        }] });
