// @backbase/foundation-ang/esm2015/web-sdk/providers.js
import { InjectionToken } from '@angular/core';
/**
 * @deprecated Will be removed in v7.0.0. Use the `PubSubService` class instead.
 */
export const PUBSUB = new InjectionToken('web-sdk-pubsub');
/**
 * @deprecated Will be removed in v7.0.0. Use the `SetLocaleService` class instead.
 */
export const SET_LOCALE = new InjectionToken('web-sdk-set-locale');
/**
 * @deprecated Will be removed in v7.0.0
 */
export const EMBEDDED_SERVICES = new InjectionToken('embedded platform services');
/**
 * @description
 * Service to reauthenticate a user.
 *
 * @example
 * export class SigningService {
 *   constructor(
 *     @Inject(REAUTHENTICATE) private reauthenticateService: ReauthenticateService,
 *   ) {
 *     this.reauthenticate(error);
 *   }
 *
 *   private reauthenticate(error: HttpErrorResponse) {
 *     const { acrValues, scope } = this.getChallenge(error);
 *     this.reauthenticateService.reauthenticate(scope, acrValues);
 *   }
 *
 *   private getReauthChallenge(error: HttpErrorResponse) {
 *     const challenges = error.error ? error.error.challenges : [];
 *     return challenges.find(c => c.scope !== undefined);
 *   }
 * }
 */
/**
 * @deprecated Will be removed in v7.0.0
 */
export const EXTERNAL_SERVICES = new InjectionToken('external platform services');
/**
 * @deprecated Will be removed in v7.0.0
 */
export const WEBSDK_SERVICE_MAP = new InjectionToken('Web SDK Service Map');
/**
 * @deprecated Will be removed in v7.0.0
 */
export function websdkServiceFactory(webSdkServiceMap, serviceKey) {
    if (!webSdkServiceMap) {
        return undefined;
    }
    const factory = webSdkServiceMap[serviceKey];
    if (!factory) {
        return undefined;
    }
    return factory(); // TS isn't smart enough to know that factory() is T.
}
/**
 * @deprecated Will be removed in v7.0.0
 */
export function navigationFactory(webSdkServiceMap) {
    return websdkServiceFactory(webSdkServiceMap, 'navigation');
}
/**
 * @deprecated Will be removed in v7.0.0
 */
export function pubsubFactory(webSdkServiceMap) {
    return websdkServiceFactory(webSdkServiceMap, 'eventBus');
}
/**
 * @deprecated Will be removed in v7.0.0
 */
export function portalContentFactory(webSdkServiceMap) {
    return websdkServiceFactory(webSdkServiceMap, 'portalContent');
}
/**
 * @deprecated Will be removed in v7.0.0
 */
export function setLocaleFactory(webSdkServiceMap) {
    return websdkServiceFactory(webSdkServiceMap, 'setLocale');
}
