// @backbase/foundation-ang/esm2015/web-sdk/pub-sub.service.js
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
export class PubSubService {
    constructor() {
        this.getListeners = (string) => this.listenersMap.get(string) || [];
        this.listenersMap = new Map();
    }
    publish(string, payload) {
        const listeners = this.getListeners(string);
        listeners.forEach(listener => {
            try {
                listener(payload);
            }
            catch (error) {
                console.error(error);
            }
        });
    }
    subscribe(string, listener) {
        const listeners = this.getListeners(string);
        const newListeners = [...listeners, listener];
        this.listenersMap.set(string, newListeners);
        return () => {
            this.unsubscribe(string, listener);
        };
    }
    unsubscribe(string, listener) {
        const listeners = this.getListeners(string);
        const newListeners = listeners.filter(currentListener => currentListener !== listener);
        this.listenersMap.set(string, newListeners);
    }
    unsubscribeAll() {
        this.listenersMap.clear();
    }
}
PubSubService.ɵprov = i0.ɵɵdefineInjectable({ factory: function PubSubService_Factory() { return new PubSubService(); }, token: PubSubService, providedIn: "root" });
PubSubService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root',
            },] }
];
PubSubService.ctorParameters = () => [];
