// @backbase/foundation-ang/esm2015/web-sdk/public-api.js
export { WebSdkApiModule } from './web-sdk-api-ang.module';
export { PAGE_CONFIG } from './page-config';
export { PUBSUB, SET_LOCALE, EXTERNAL_SERVICES as ɵEXTERNAL_SERVICES } from './providers';
export { PORTAL_CONTENT, LOGIN, LOGOUT, SESSION, REAUTHENTICATE, NAVIGATION, AUTH_EVENTS as ɵAUTH_EVENTS, HTTP_INTERCEPTORS, REFRESH, } from './deprecated/providers';
// classes
import { PortalContentMock as _PortalContentMock } from '@backbase/foundation-ang/testing';
/**
 * @deprecated This entrypoint is deprecated use \'@backbase/foundation-ang/test\' for PortalContentMock
 */
export class PortalContentMock extends _PortalContentMock {
}
// enums
export { ReauthErrorCodes, ContentTypes, ContentType, AuthKind, } from '@backbase/lib-bb-page-bootstrap-cdn';
// typings
export { PubSubService } from './pub-sub.service';
export { SetLocaleService } from './set-locale.service';
export { WebSdkModule } from './web-sdk.module';
export { CONDITIONS, ENTITLEMENTS_CONFIG, ConditionsService, SUMMARY_PERMISSIONS_PATH, DATA_ITEMS_PERMISSIONS_PATH, } from '@backbase/foundation-ang/entitlements';
