// @backbase/foundation-ang/esm2015/web-sdk/set-locale.service.js
import { Inject, Injectable } from '@angular/core';
import { DOCUMENT } from '@angular/common';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
const COOKIE_NAME = 'bb-locale';
//TODO: Remove dynamic when Ivy compilation is enabled
//@dynamic
export class SetLocaleService {
    constructor(document) {
        this.document = document;
    }
    getCookieName() {
        return COOKIE_NAME;
    }
    setLocale(locale) {
        this.document.cookie = `${COOKIE_NAME}=${locale}; path=/`;
    }
    /**
     * @deprecated Will be removed in v7.0.0
     */
    setLocaleLegacy(locale) {
        this.setLocale(locale);
        return Promise.resolve({
            status: 200,
            statusText: 'OK',
            headers: {},
            body: '',
        });
    }
}
SetLocaleService.ɵprov = i0.ɵɵdefineInjectable({ factory: function SetLocaleService_Factory() { return new SetLocaleService(i0.ɵɵinject(i1.DOCUMENT)); }, token: SetLocaleService, providedIn: "root" });
SetLocaleService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root',
            },] }
];
SetLocaleService.ctorParameters = () => [
    { type: Document, decorators: [{ type: Inject, args: [DOCUMENT,] }] }
];
