// @backbase/product-summary-ui/esm2015/src/bb-arrangement-state-indicator-ui/arrangement-state-indicator.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { IconModule } from '@backbase/ui-ang/icon';
import { ArrangementStateIndicatorComponent, ArrangementStateIndicatorCustomizableDirective, } from './arrangement-state-indicator.component';
import * as i0 from "@angular/core";
const uiModules = [IconModule];
const components = [ArrangementStateIndicatorComponent];
const directives = [ArrangementStateIndicatorCustomizableDirective];
export class ArrangementStateIndicatorModule {
}
ArrangementStateIndicatorModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ArrangementStateIndicatorModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ArrangementStateIndicatorModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ArrangementStateIndicatorModule, declarations: [ArrangementStateIndicatorComponent, ArrangementStateIndicatorCustomizableDirective], imports: [CommonModule, IconModule], exports: [ArrangementStateIndicatorComponent, ArrangementStateIndicatorCustomizableDirective, IconModule] });
ArrangementStateIndicatorModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ArrangementStateIndicatorModule, imports: [[CommonModule, ...uiModules], IconModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ArrangementStateIndicatorModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules],
                    declarations: [...components, ...directives],
                    exports: [...components, ...directives, ...uiModules],
                }]
        }] });
