// @backbase/connect-external-accounts-journey-ang/esm2015/lib/connect-external-accounts-journey.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { provideRoutes, RouterModule } from '@angular/router';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { HeaderModule } from '@backbase/ui-ang/header';
import { ConnectExternalAccountsJourneyUiModule } from './components/connect-external-accounts-journey-ui.module';
import { ConnectExternalAccountsJourneyComponent } from './connect-external-accounts-journey.component';
import { ConnectExternalAccountsJourneyConfigService } from './services/config.service';
import { ConnectExternalAccountsViewComponent } from './views/connect-external-accounts-view.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const defaultRoute = {
    path: '',
    component: ConnectExternalAccountsJourneyComponent,
    children: [{ path: '', component: ConnectExternalAccountsViewComponent }],
};
const components = [ConnectExternalAccountsJourneyComponent, ConnectExternalAccountsViewComponent];
const uiModules = [EmptyStateModule, HeaderModule];
export class ConnectExternalAccountsJourneyModule {
    static forRoot(data = { route: defaultRoute }) {
        return {
            ngModule: ConnectExternalAccountsJourneyModule,
            providers: [provideRoutes([data.route])],
        };
    }
}
/** @nocollapse */ ConnectExternalAccountsJourneyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ConnectExternalAccountsJourneyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ ConnectExternalAccountsJourneyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ConnectExternalAccountsJourneyModule, declarations: [ConnectExternalAccountsJourneyComponent, ConnectExternalAccountsViewComponent], imports: [CommonModule,
        RouterModule, i1.BackbaseCoreModule, ConnectExternalAccountsJourneyUiModule, EmptyStateModule, HeaderModule], exports: [EmptyStateModule, HeaderModule, ConnectExternalAccountsJourneyComponent, ConnectExternalAccountsViewComponent] });
/** @nocollapse */ ConnectExternalAccountsJourneyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ConnectExternalAccountsJourneyModule, providers: [ConnectExternalAccountsJourneyConfigService], imports: [[
            CommonModule,
            RouterModule,
            BackbaseCoreModule.withConfig({
                classMap: {
                    ConnectExternalAccountsJourneyComponent,
                },
            }),
            ConnectExternalAccountsJourneyUiModule,
            ...uiModules,
        ], EmptyStateModule, HeaderModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ConnectExternalAccountsJourneyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components],
                    imports: [
                        CommonModule,
                        RouterModule,
                        BackbaseCoreModule.withConfig({
                            classMap: {
                                ConnectExternalAccountsJourneyComponent,
                            },
                        }),
                        ConnectExternalAccountsJourneyUiModule,
                        ...uiModules,
                    ],
                    exports: [...uiModules, ...components],
                    providers: [ConnectExternalAccountsJourneyConfigService],
                }]
        }] });
