// @backbase/payment-orders-ang/esm2015/lib/ui/pipes/has-value/has-value.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HasValuePipe } from './has-value.pipe';
import * as i0 from "@angular/core";
export class HasValueModule {
}
/** @nocollapse */ HasValueModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HasValueModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ HasValueModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HasValueModule, declarations: [HasValuePipe], imports: [CommonModule], exports: [HasValuePipe] });
/** @nocollapse */ HasValueModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HasValueModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HasValueModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [HasValuePipe],
                    exports: [HasValuePipe],
                    imports: [CommonModule],
                }]
        }] });
