// @backbase/payment-batch-http-ang/esm2015/api/batchOrders.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const BatchOrdersHttpServiceDeleteBatchOrderIdByBatchOrderIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/batch-orders/{batchOrderId}",
        method: "DELETE",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 409,
                body: {
                    "message": "Conflict error",
                    "errors": [{
                            "message": "The request could not be completed due to a conflict with the current state of the resource.",
                            "key": "transactions.api.billingstatus.transaction",
                            "context": {
                                "billingStatus": "BILLED"
                            }
                        }]
                }
            },
        ]
    }]);
export const BatchOrdersHttpServiceDeleteBatchPaymentByIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/batch-orders/{batchOrderId}/batch-payments/{batchPaymentId}",
        method: "DELETE",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 404,
                body: {
                    "message": "Batch order not found",
                    "errors": []
                }
            },
            {
                status: 404,
                body: {
                    "message": "Batch payment not found",
                    "errors": []
                }
            },
            {
                status: 409,
                body: {
                    "message": "Conflict error",
                    "errors": [{
                            "message": "The request could not be completed due to a conflict with the current state of the resource.",
                            "key": "transactions.api.billingstatus.transaction",
                            "context": {
                                "billingStatus": "BILLED"
                            }
                        }]
                }
            },
        ]
    }]);
export const BatchOrdersHttpServiceGetBatchOrdersMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/batch-orders",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "batchOrders": [{
                            "id": "2f3a6d96-f671-4f03-994d-8f6caaa53cab",
                            "name": "Employee salaries",
                            "type": "SEPACT",
                            "creditDebitMixedIndicator": "CREDIT",
                            "status": "ACCEPTED",
                            "totalTransactionsCount": 73500,
                            "totalCreditTransactionsCount": 73500,
                            "totalInstructedAmount": {
                                "amount": "2400123.00",
                                "currencyCode": "EUR"
                            },
                            "totalCreditInstructedAmount": {
                                "amount": "2400123.00",
                                "currencyCode": "EUR"
                            },
                            "account": {
                                "name": "Main Current",
                                "identification": {
                                    "identification": "CH8739215260566920524",
                                    "schemeName": "IBAN"
                                }
                            },
                            "requestedExecutionDate": "2019-02-18",
                            "canDelete": false,
                            "canCancel": false
                        }, {
                            "id": "33295f9c-95ec-42f9-9817-64f7d48be023",
                            "name": "Employee salaries",
                            "type": "SEPACT",
                            "creditDebitMixedIndicator": "CREDIT",
                            "status": "ACCEPTED",
                            "totalTransactionsCount": 75930,
                            "totalCreditTransactionsCount": 75930,
                            "totalInstructedAmount": {
                                "amount": "2479474.00",
                                "currencyCode": "EUR"
                            },
                            "totalCreditInstructedAmount": {
                                "amount": "2479474.00",
                                "currencyCode": "EUR"
                            },
                            "account": {
                                "name": "Main Current",
                                "identification": {
                                    "identification": "CH8739215260566920524",
                                    "schemeName": "IBAN"
                                }
                            },
                            "requestedExecutionDate": "2019-03-18",
                            "canDelete": false,
                            "canCancel": false
                        }, {
                            "id": "f3da0418-c801-4db1-ae52-597f1459b525",
                            "name": "Employee salaries",
                            "type": "SEPACT",
                            "creditDebitMixedIndicator": "CREDIT",
                            "status": "ENTERED",
                            "totalTransactionsCount": 78791,
                            "totalCreditTransactionsCount": 78791,
                            "totalInstructedAmount": {
                                "amount": "2552947.15",
                                "currencyCode": "EUR"
                            },
                            "totalCreditInstructedAmount": {
                                "amount": "2552947.15",
                                "currencyCode": "EUR"
                            },
                            "account": {
                                "name": "Main Current",
                                "identification": {
                                    "identification": "CH8739215260566920524",
                                    "schemeName": "IBAN"
                                }
                            },
                            "requestedExecutionDate": "2019-04-18",
                            "canDelete": true,
                            "canCancel": false
                        }],
                    "totalBatchOrders": 3,
                    "creditDebitMixedIndicator": "CREDIT"
                }
            },
            {
                status: 200,
                body: {
                    "batchOrders": [{
                            "id": "a8e1ba16-d9be-4dd4-92b8-e2fbfe138b71",
                            "name": "Employees salaries May",
                            "type": "SEPACT",
                            "creditDebitMixedIndicator": "CREDIT",
                            "status": "ENTERED",
                            "totalTransactionsCount": 71,
                            "totalCreditTransactionsCount": 71,
                            "totalInstructedAmount": {
                                "amount": "191056.00",
                                "currencyCode": "EUR"
                            },
                            "totalCreditInstructedAmount": {
                                "amount": "191056.00",
                                "currencyCode": "EUR"
                            },
                            "account": {
                                "name": "Salaries Account",
                                "identification": {
                                    "identification": "ST42353317399261373273497",
                                    "schemeName": "IBAN"
                                }
                            },
                            "requestedExecutionDate": "2019-05-27",
                            "approval": {
                                "id": "9bb143b1-bdb3-4ca5-9631-e02e743d6a10",
                                "userId": "23b8a2ac-e8bb-46b1-98d2-a2da71dfd646",
                                "externalUserId": "judi-miles",
                                "userFullName": "Judi Miles",
                                "serviceAgreementId": "e3e718d8-8f2e-4ef4-bf27-22580759b742",
                                "itemId": "a8e1ba16-d9be-4dd4-92b8-e2fbfe138b71",
                                "dataAccessItemId": "ac59c1a1-3729-4514-8051-5015dd0aadef",
                                "resource": "Batches",
                                "function": "SEPACT",
                                "status": "PENDING",
                                "createdAt": "2019-05-10T10:29:10Z",
                                "records": [],
                                "canApprove": true,
                                "canReject": true,
                                "finalApprover": true
                            },
                            "canDelete": false,
                            "canCancel": false
                        }],
                    "totalBatchOrders": 1,
                    "creditDebitMixedIndicator": "CREDIT"
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
        ]
    }]);
export const BatchOrdersHttpServiceGetBatchOrdersByIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/batch-orders/{batchOrderId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "2f3a6d96-f671-4f03-994d-8f6caaa53cab",
                    "name": "Employee salaries",
                    "type": "SEPACT",
                    "creditDebitMixedIndicator": "CREDIT",
                    "status": "ACCEPTED",
                    "totalTransactionsCount": 73500,
                    "totalCreditTransactionsCount": 73500,
                    "totalInstructedAmount": {
                        "amount": "2400123.00",
                        "currencyCode": "EUR"
                    },
                    "totalCreditInstructedAmount": {
                        "amount": "2400123.00",
                        "currencyCode": "EUR"
                    },
                    "account": {
                        "name": "Main Current",
                        "identification": {
                            "identification": "CH8739215260566920524",
                            "schemeName": "IBAN"
                        }
                    },
                    "requestedExecutionDate": "2019-02-18",
                    "fileName": "salaries-2019-02.csv",
                    "userFullName": "Davy Jones",
                    "uploadCompletedAt": "2019-02-22T14:08:34Z",
                    "canDelete": false,
                    "canCancel": false
                }
            },
            {
                status: 200,
                body: {
                    "id": "2fd15c69-c4da-4c58-986f-d46082057bf5",
                    "name": "End of Q2 payments",
                    "type": "SEPACT",
                    "status": "ENTERED",
                    "totalTransactionsCount": 2109,
                    "totalInstructedAmount": {
                        "amount": "280810.00",
                        "currencyCode": "EUR"
                    },
                    "account": {
                        "name": "Main",
                        "identification": {
                            "identification": "NO0741575812927",
                            "schemeName": "IBAN"
                        }
                    },
                    "requestedExecutionDate": "2019-05-31",
                    "fileName": "2019-Q2-payments.csv",
                    "userFullName": "Tommy Perrotti",
                    "uploadCompletedAt": "2019-05-16T14:09:30Z",
                    "approval": {
                        "id": "9966c07c-1547-42a5-9bc7-91986ff33993",
                        "userId": "c3608888-faa3-4f23-9c5a-269dd39caea5",
                        "externalUserId": "judi-miles",
                        "userFullName": "Judi Miles",
                        "serviceAgreementId": "64f538bf-3f5d-4dd9-8de7-ba7082da198d",
                        "itemId": "2fd15c69-c4da-4c58-986f-d46082057bf5",
                        "dataAccessItemId": "f094b702-e912-434c-8e49-12c56fc78551",
                        "resource": "Batches",
                        "function": "SEPACT",
                        "status": "PENDING",
                        "createdAt": "2019-05-16T14:09:40Z",
                        "records": [{
                                "id": "9e61d96d-fea3-44a4-8903-88d95eb266a8",
                                "userId": "f34feccc-0d0d-4365-982b-d6d1779e5aaf",
                                "externalUserId": "bigWillie",
                                "userFullName": "Willie Iles",
                                "approvalTypeId": "ece87cfe-33f9-45a4-8c51-a04a9f905c66",
                                "status": "APPROVED",
                                "createdAt": "2019-05-17T13:44:02Z"
                            }],
                        "canApprove": false,
                        "canReject": false,
                        "finalApprover": false
                    },
                    "canDelete": false,
                    "canCancel": true
                }
            },
            {
                status: 200,
                body: {
                    "id": "b438751d-6e6a-4601-8009-a82fa22bddcc",
                    "name": "Employee salaries",
                    "type": "SEPACT",
                    "status": "REJECTED",
                    "totalTransactionsCount": 9208,
                    "totalInstructedAmount": {
                        "amount": "50343009.00",
                        "currencyCode": "EUR"
                    },
                    "account": {
                        "name": "Main Account 8",
                        "identification": {
                            "identification": "NL18ABNA4956451237",
                            "schemeName": "IBAN"
                        }
                    },
                    "requestedExecutionDate": "2019-09-15",
                    "description": "Salary Sep 2019",
                    "bankStatus": "UNPROCESSABLE",
                    "reasonCode": "RJ08",
                    "reasonText": "Insufficient funds",
                    "errorDescription": "Target account has not enough funds to execute batch payments",
                    "fileName": "salaries-2019-02.csv",
                    "userFullName": "Davy Jones",
                    "uploadCompletedAt": "2019-02-22T14:08:34Z",
                    "canDelete": false,
                    "canCancel": false
                }
            },
            {
                status: 200,
                body: {
                    "id": "2f4c4bb6-c107-473a-adbd-7343b8f0ea80",
                    "name": "Salaries 2020-12",
                    "type": "ACH_CREDIT",
                    "status": "ACCEPTED",
                    "totalTransactionsCount": 401,
                    "totalInstructedAmount": {
                        "amount": "650980.25",
                        "currencyCode": "USD"
                    },
                    "account": {
                        "name": "Current Account",
                        "identification": {
                            "identification": "7649373647",
                            "schemeName": "BBAN"
                        }
                    },
                    "bankBranchCode": "113105452",
                    "companyId": "1800948594",
                    "companyName": "Your Company",
                    "requestedExecutionDate": "2020-12-18",
                    "fileName": "salaries-2020-12.ach",
                    "userFullName": "Ben Jones",
                    "uploadCompletedAt": "2020-12-15T15:28:10Z",
                    "canDelete": false,
                    "canCancel": false
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    }]);
export const BatchOrdersHttpServiceGetBatchPaymentByIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/batch-orders/{batchOrderId}/batch-payments/{batchPaymentId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "counterpartyName": "M. Frederiksen",
                    "counterpartyAccountNumber": "DE49500105174361358418",
                    "instructedAmount": {
                        "amount": "5576.00",
                        "currencyCode": "EUR"
                    },
                    "description": "Salary Feb 2019 empl.no:068572",
                    "status": "REJECTED",
                    "extendedDescription": "Addenda record for the credit payment \nAddenda record for the credit payment",
                    "bankStatus": "UNPROCESSABLE",
                    "reasonCode": "RJ05",
                    "reasonText": "Rejected by counterparty bank",
                    "errorDescription": "Target account has been closed."
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    }]);
export const BatchOrdersHttpServiceGetBatchPaymentsByIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/batch-orders/{batchOrderId}/batch-payments",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "batchPayments": [{
                            "counterpartyName": "A. Hamers",
                            "counterpartyAccountNumber": "NL62ABNA0609476343",
                            "instructedAmount": {
                                "amount": "4102.00",
                                "currencyCode": "EUR"
                            },
                            "description": "Salaray Feb 2019 empl.no:902713",
                            "extendedDescriptionSize": 99999
                        }, {
                            "counterpartyName": "E. Gutierrez",
                            "counterpartyAccountNumber": "NL50ABNA4418364501",
                            "instructedAmount": {
                                "amount": "5287.00",
                                "currencyCode": "EUR"
                            },
                            "description": "Salaray Feb 2019 empl.no:256869",
                            "extendedDescriptionSize": 0
                        }, {
                            "counterpartyName": "C.M. McAlister",
                            "counterpartyAccountNumber": "BE98735114499693",
                            "instructedAmount": {
                                "amount": "3398.00",
                                "currencyCode": "EUR"
                            },
                            "description": "Salaray Feb 2019 empl.no:100642",
                            "extendedDescriptionSize": 456768
                        }, {
                            "counterpartyName": "K.C. Bennis",
                            "counterpartyAccountNumber": "NL67ABNA5515439724",
                            "instructedAmount": {
                                "amount": "4334.00",
                                "currencyCode": "EUR"
                            },
                            "description": "Salaray Feb 2019 empl.no:191872",
                            "extendedDescriptionSize": 0
                        }, {
                            "counterpartyName": "M. Frederiksen",
                            "counterpartyAccountNumber": "DE49500105174361358418",
                            "instructedAmount": {
                                "amount": "5576.00",
                                "currencyCode": "EUR"
                            },
                            "description": "Salaray Feb 2019 empl.no:068572",
                            "extendedDescriptionSize": 123,
                            "status": "REJECTED",
                            "bankStatus": "UNPROCESSABLE",
                            "reasonCode": "RJ05",
                            "reasonText": "Rejected by counterparty bank",
                            "errorDescription": "Target account has been closed."
                        }],
                    "totalBatchPayments": 5,
                    "batchOrder": {
                        "id": "2f3a6d96-f671-4f03-994d-8f6caaa53cab",
                        "name": "Employee salaries",
                        "type": "SEPACT",
                        "status": "ACCEPTED",
                        "totalTransactionsCount": 5,
                        "totalInstructedAmount": {
                            "amount": "2400123.00",
                            "currencyCode": "EUR"
                        },
                        "account": {
                            "name": "Main Current",
                            "identification": {
                                "identification": "CH8739215260566920524",
                                "schemeName": "IBAN"
                            }
                        },
                        "requestedExecutionDate": "2019-02-18",
                        "fileName": "salaries-2019-02.csv",
                        "userFullName": "Davy Jones",
                        "uploadCompletedAt": "2019-02-22T14:08:34Z",
                        "confidentialType": "confidential type"
                    }
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    }]);
export const BatchOrdersHttpServicePostBatchOrdersMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/batch-orders",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "id": "c1263fd6-8bd3-4a0b-be02-5db5cb1d6d06",
                    "status": "DRAFT"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
        ]
    }]);
export const BatchOrdersHttpServicePostBatchPaymentsMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/batch-orders/{batchOrderId}/batch-payments",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "id": "cad9e668-15a4-44bf-832d-67e7b41387ac"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Batch order not found",
                    "errors": []
                }
            },
        ]
    }]);
export const BatchOrdersHttpServicePutApprovalsByIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/batch-orders/{batchOrderId}/approvals",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {
                    "batchOrder": {
                        "id": "72d17d02-20a9-4f0b-b2b8-f83959eb187e",
                        "name": "Salaries May 2019",
                        "type": "SEPACT",
                        "creditDebitMixedIndicator": "CREDIT",
                        "status": "ENTERED",
                        "totalTransactionsCount": 6504,
                        "totalCreditTransactionsCount": 6504,
                        "totalInstructedAmount": {
                            "amount": "14009055.00",
                            "currencyCode": "EUR"
                        },
                        "totalCreditInstructedAmount": {
                            "amount": "14009055.00",
                            "currencyCode": "EUR"
                        },
                        "account": {
                            "name": "Salaries account",
                            "identification": {
                                "identification": "NL78ABNA3470416656",
                                "schemeName": "IBAN"
                            }
                        },
                        "requestedExecutionDate": "2019-05-26",
                        "approval": {
                            "id": "00b5ca29-eddd-49f4-a898-c3f185ab9a3a",
                            "userId": "4840854b-0597-47a6-8984-87cb50f970eb",
                            "externalUserId": "judi-miles",
                            "userFullName": "Judi Miles",
                            "serviceAgreementId": "ea7b3dfd-29b8-4559-872a-91f5576b56ef",
                            "itemId": "72d17d02-20a9-4f0b-b2b8-f83959eb187e",
                            "dataAccessItemId": "606c7afa-3c8a-4210-ad25-fc83893a4e49",
                            "resource": "Batches",
                            "function": "SEPACT",
                            "status": "PENDING",
                            "createdAt": "2019-08-10T18:09:21Z",
                            "records": [],
                            "canApprove": true,
                            "canReject": true,
                            "finalApprover": false
                        }
                    },
                    "approvalStatus": "PENDING"
                }
            },
            {
                status: 403,
                body: {
                    "message": "Limits has been breached, check the report for more details",
                    "checkTime": "2019-10-31T12:12:12Z",
                    "breachReport": [{
                            "limitedEntity": [{
                                    "ref": "1234567-12312-123123",
                                    "type": "Service Agreement",
                                    "description": "Kuhic, Gislason and Kemmer. SERVICE AGREEMENT"
                                }],
                            "shadow": false,
                            "currency": "EUR",
                            "user-BBID": "oleksii",
                            "breachInfo": [{
                                    "breachType": "THRESHOLD",
                                    "breachedLimitType": "PERIODIC",
                                    "timeframe": {
                                        "period": "daily",
                                        "startTime": "2019-10-31T00:00:00Z",
                                        "endTime": "2019-10-31T23:59:59Z"
                                    },
                                    "currentConsumption": "250.0",
                                    "currentThreshold": "499.9"
                                }, {
                                    "breachType": "CONSUMPTION",
                                    "timeframe": {
                                        "period": "monthly",
                                        "startTime": "2019-10-01T00:00:00Z",
                                        "endTime": "2019-10-31T23:59:59Z"
                                    },
                                    "currentConsumption": "9950.0",
                                    "currentThreshold": "10000.0"
                                }]
                        }, {
                            "user-BBID": "oleksii",
                            "shadow": false,
                            "currency": "EUR",
                            "breachInfo": [{
                                    "breachType": "CONSUMPTION",
                                    "timeframe": {
                                        "period": "daily",
                                        "startTime": "2019-10-31T00:00:00Z",
                                        "endTime": "2019-10-31T23:59:59Z"
                                    },
                                    "currentConsumption": "500.0",
                                    "currentThreshold": "1000.0"
                                }]
                        }, {
                            "limitedEntity": [{
                                    "ref": "1234567-12312-123123",
                                    "type": "Function Access Group",
                                    "description": "Batches approvers"
                                }, {
                                    "ref": "1234567-12312-123123",
                                    "type": "Function",
                                    "description": "Domestic batches"
                                }, {
                                    "ref": "Approve",
                                    "type": "Privilege",
                                    "description": "Approve"
                                }],
                            "shadow": false,
                            "currency": "EUR",
                            "breachInfo": [{
                                    "breachType": "THRESHOLD",
                                    "timeframe": {
                                        "period": "daily",
                                        "startTime": "2019-10-31T00:00:00Z",
                                        "endTime": "2019-10-31T23:59:59Z"
                                    },
                                    "currentConsumption": "250.0",
                                    "currentThreshold": "499.9"
                                }]
                        }, {
                            "limitedEntity": [{
                                    "ref": "1234567-12312-123123",
                                    "type": "Legal Entity",
                                    "description": "Kuhic, Gislason and Kemmer"
                                }],
                            "shadow": false,
                            "currency": "EUR",
                            "breachInfo": [{
                                    "breachType": "CONSUMPTION",
                                    "timeframe": {
                                        "period": "quarterly",
                                        "startTime": "2019-10-01T00:00:00Z",
                                        "endTime": "2019-12-31T23:59:59Z"
                                    },
                                    "currentConsumption": "99950.0",
                                    "currentThreshold": "100000.0"
                                }]
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
        ]
    }]);
export const BatchOrdersHttpServicePutBatchPaymentByIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/batch-orders/{batchOrderId}/batch-payments/{batchPaymentId}",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {
                    "id": "7dc95f0a-deba-47be-8f5d-6d0454ef271c",
                    "counterpartyName": "Uriel Johnson",
                    "counterpartyAccountNumber": "HU65109180019369738837536884",
                    "instructedAmount": {
                        "amount": "830.00",
                        "currencyCode": "EUR"
                    },
                    "description": "Order 00023162"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Batch order not found",
                    "errors": []
                }
            },
            {
                status: 404,
                body: {
                    "message": "Batch payment not found",
                    "errors": []
                }
            },
        ]
    }]);
export const BatchOrdersHttpServiceUpdateBatchOrderByIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/batch-orders/{batchOrderId}",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {
                    "id": "f7e4f0bf-6080-46be-bd5d-7dad2b054495",
                    "name": "Employee salaries",
                    "type": "SEPACT",
                    "creditDebitMixedIndicator": "CREDIT",
                    "status": "CANCELLED",
                    "totalTransactionsCount": 120,
                    "totalCreditTransactionsCount": 120,
                    "totalInstructedAmount": {
                        "amount": "322038.00",
                        "currencyCode": "EUR"
                    },
                    "totalCreditInstructedAmount": {
                        "amount": "322038.00",
                        "currencyCode": "EUR"
                    },
                    "account": {
                        "name": "Main Current",
                        "identification": {
                            "identification": "NL54ABNA2366058993",
                            "schemeName": "IBAN"
                        }
                    },
                    "requestedExecutionDate": "2012-01-01",
                    "fileName": "salaries-2020-01.csv",
                    "userFullName": "John Davis",
                    "uploadCompletedAt": "2019-12-31T23:55:31Z",
                    "canDelete": false,
                    "canCancel": false,
                    "approval": {
                        "id": "ac150312-1bec-4075-934f-23600886cb4e",
                        "userId": "64350fcf-ddd1-4cdc-b725-f6c76f89f7c2",
                        "externalUserId": "jdavis",
                        "userFullName": "John Davis",
                        "serviceAgreementId": "c94dc22f-381f-4127-a9f2-db2f5a314e41",
                        "itemId": "f7e4f0bf-6080-46be-bd5d-7dad2b054495",
                        "dataAccessItemId": "e8daae6b-0276-4533-b64d-e3c73afb47a0",
                        "resource": "Batches",
                        "function": "SEPACT",
                        "status": "PENDING",
                        "createdAt": "2019-12-31T23:56:08Z",
                        "records": [{
                                "id": "9e61d96d-fea3-44a4-8903-88d95eb266a8",
                                "userId": "f34feccc-0d0d-4365-982b-d6d1779e5aaf",
                                "externalUserId": "twilliams",
                                "userFullName": "Tobi Williams",
                                "approvalTypeId": "16f655f3-3abf-438d-8351-2c667d5a3c86",
                                "status": "APPROVED",
                                "createdAt": "2019-12-31T23:57:20Z"
                            }],
                        "canApprove": false,
                        "canReject": false,
                        "finalApprover": false
                    }
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad request",
                    "errors": [{
                            "message": "The requested status ({{requestedStatus}}) is not valid. The status can be set to {{expectedStatus}}.",
                            "key": "BatchOrderCannotBeUpdated",
                            "context": {
                                "requestedStatus": "READY",
                                "expectedStatus": "CANCELLED"
                            }
                        }]
                }
            },
            {
                status: 409,
                body: {
                    "message": "Conflict",
                    "errors": [{
                            "message": "The current status of the batch order ({{currentStatus}}) does not allow cancellation.",
                            "key": "BatchOrderCannotBeCancelled",
                            "context": {
                                "currentStatus": "PROCESSED"
                            }
                        }]
                }
            },
            {
                status: 409,
                body: {
                    "message": "Conflict",
                    "errors": [{
                            "message": "Only partially approved batch orders can be cancelled.",
                            "key": "BatchOrderWithoutApprovalsCannotBeCancelled"
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    }]);
export const BatchOrdersHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v2/batch-orders/{batchOrderId}",
        method: "DELETE",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 409,
                body: {
                    "message": "Conflict error",
                    "errors": [{
                            "message": "The request could not be completed due to a conflict with the current state of the resource.",
                            "key": "transactions.api.billingstatus.transaction",
                            "context": {
                                "billingStatus": "BILLED"
                            }
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/batch-orders/{batchOrderId}/batch-payments/{batchPaymentId}",
        method: "DELETE",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 404,
                body: {
                    "message": "Batch order not found",
                    "errors": []
                }
            },
            {
                status: 404,
                body: {
                    "message": "Batch payment not found",
                    "errors": []
                }
            },
            {
                status: 409,
                body: {
                    "message": "Conflict error",
                    "errors": [{
                            "message": "The request could not be completed due to a conflict with the current state of the resource.",
                            "key": "transactions.api.billingstatus.transaction",
                            "context": {
                                "billingStatus": "BILLED"
                            }
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/batch-orders",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "batchOrders": [{
                            "id": "2f3a6d96-f671-4f03-994d-8f6caaa53cab",
                            "name": "Employee salaries",
                            "type": "SEPACT",
                            "creditDebitMixedIndicator": "CREDIT",
                            "status": "ACCEPTED",
                            "totalTransactionsCount": 73500,
                            "totalCreditTransactionsCount": 73500,
                            "totalInstructedAmount": {
                                "amount": "2400123.00",
                                "currencyCode": "EUR"
                            },
                            "totalCreditInstructedAmount": {
                                "amount": "2400123.00",
                                "currencyCode": "EUR"
                            },
                            "account": {
                                "name": "Main Current",
                                "identification": {
                                    "identification": "CH8739215260566920524",
                                    "schemeName": "IBAN"
                                }
                            },
                            "requestedExecutionDate": "2019-02-18",
                            "canDelete": false,
                            "canCancel": false
                        }, {
                            "id": "33295f9c-95ec-42f9-9817-64f7d48be023",
                            "name": "Employee salaries",
                            "type": "SEPACT",
                            "creditDebitMixedIndicator": "CREDIT",
                            "status": "ACCEPTED",
                            "totalTransactionsCount": 75930,
                            "totalCreditTransactionsCount": 75930,
                            "totalInstructedAmount": {
                                "amount": "2479474.00",
                                "currencyCode": "EUR"
                            },
                            "totalCreditInstructedAmount": {
                                "amount": "2479474.00",
                                "currencyCode": "EUR"
                            },
                            "account": {
                                "name": "Main Current",
                                "identification": {
                                    "identification": "CH8739215260566920524",
                                    "schemeName": "IBAN"
                                }
                            },
                            "requestedExecutionDate": "2019-03-18",
                            "canDelete": false,
                            "canCancel": false
                        }, {
                            "id": "f3da0418-c801-4db1-ae52-597f1459b525",
                            "name": "Employee salaries",
                            "type": "SEPACT",
                            "creditDebitMixedIndicator": "CREDIT",
                            "status": "ENTERED",
                            "totalTransactionsCount": 78791,
                            "totalCreditTransactionsCount": 78791,
                            "totalInstructedAmount": {
                                "amount": "2552947.15",
                                "currencyCode": "EUR"
                            },
                            "totalCreditInstructedAmount": {
                                "amount": "2552947.15",
                                "currencyCode": "EUR"
                            },
                            "account": {
                                "name": "Main Current",
                                "identification": {
                                    "identification": "CH8739215260566920524",
                                    "schemeName": "IBAN"
                                }
                            },
                            "requestedExecutionDate": "2019-04-18",
                            "canDelete": true,
                            "canCancel": false
                        }],
                    "totalBatchOrders": 3,
                    "creditDebitMixedIndicator": "CREDIT"
                }
            },
            {
                status: 200,
                body: {
                    "batchOrders": [{
                            "id": "a8e1ba16-d9be-4dd4-92b8-e2fbfe138b71",
                            "name": "Employees salaries May",
                            "type": "SEPACT",
                            "creditDebitMixedIndicator": "CREDIT",
                            "status": "ENTERED",
                            "totalTransactionsCount": 71,
                            "totalCreditTransactionsCount": 71,
                            "totalInstructedAmount": {
                                "amount": "191056.00",
                                "currencyCode": "EUR"
                            },
                            "totalCreditInstructedAmount": {
                                "amount": "191056.00",
                                "currencyCode": "EUR"
                            },
                            "account": {
                                "name": "Salaries Account",
                                "identification": {
                                    "identification": "ST42353317399261373273497",
                                    "schemeName": "IBAN"
                                }
                            },
                            "requestedExecutionDate": "2019-05-27",
                            "approval": {
                                "id": "9bb143b1-bdb3-4ca5-9631-e02e743d6a10",
                                "userId": "23b8a2ac-e8bb-46b1-98d2-a2da71dfd646",
                                "externalUserId": "judi-miles",
                                "userFullName": "Judi Miles",
                                "serviceAgreementId": "e3e718d8-8f2e-4ef4-bf27-22580759b742",
                                "itemId": "a8e1ba16-d9be-4dd4-92b8-e2fbfe138b71",
                                "dataAccessItemId": "ac59c1a1-3729-4514-8051-5015dd0aadef",
                                "resource": "Batches",
                                "function": "SEPACT",
                                "status": "PENDING",
                                "createdAt": "2019-05-10T10:29:10Z",
                                "records": [],
                                "canApprove": true,
                                "canReject": true,
                                "finalApprover": true
                            },
                            "canDelete": false,
                            "canCancel": false
                        }],
                    "totalBatchOrders": 1,
                    "creditDebitMixedIndicator": "CREDIT"
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/batch-orders/{batchOrderId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "2f3a6d96-f671-4f03-994d-8f6caaa53cab",
                    "name": "Employee salaries",
                    "type": "SEPACT",
                    "creditDebitMixedIndicator": "CREDIT",
                    "status": "ACCEPTED",
                    "totalTransactionsCount": 73500,
                    "totalCreditTransactionsCount": 73500,
                    "totalInstructedAmount": {
                        "amount": "2400123.00",
                        "currencyCode": "EUR"
                    },
                    "totalCreditInstructedAmount": {
                        "amount": "2400123.00",
                        "currencyCode": "EUR"
                    },
                    "account": {
                        "name": "Main Current",
                        "identification": {
                            "identification": "CH8739215260566920524",
                            "schemeName": "IBAN"
                        }
                    },
                    "requestedExecutionDate": "2019-02-18",
                    "fileName": "salaries-2019-02.csv",
                    "userFullName": "Davy Jones",
                    "uploadCompletedAt": "2019-02-22T14:08:34Z",
                    "canDelete": false,
                    "canCancel": false
                }
            },
            {
                status: 200,
                body: {
                    "id": "2fd15c69-c4da-4c58-986f-d46082057bf5",
                    "name": "End of Q2 payments",
                    "type": "SEPACT",
                    "status": "ENTERED",
                    "totalTransactionsCount": 2109,
                    "totalInstructedAmount": {
                        "amount": "280810.00",
                        "currencyCode": "EUR"
                    },
                    "account": {
                        "name": "Main",
                        "identification": {
                            "identification": "NO0741575812927",
                            "schemeName": "IBAN"
                        }
                    },
                    "requestedExecutionDate": "2019-05-31",
                    "fileName": "2019-Q2-payments.csv",
                    "userFullName": "Tommy Perrotti",
                    "uploadCompletedAt": "2019-05-16T14:09:30Z",
                    "approval": {
                        "id": "9966c07c-1547-42a5-9bc7-91986ff33993",
                        "userId": "c3608888-faa3-4f23-9c5a-269dd39caea5",
                        "externalUserId": "judi-miles",
                        "userFullName": "Judi Miles",
                        "serviceAgreementId": "64f538bf-3f5d-4dd9-8de7-ba7082da198d",
                        "itemId": "2fd15c69-c4da-4c58-986f-d46082057bf5",
                        "dataAccessItemId": "f094b702-e912-434c-8e49-12c56fc78551",
                        "resource": "Batches",
                        "function": "SEPACT",
                        "status": "PENDING",
                        "createdAt": "2019-05-16T14:09:40Z",
                        "records": [{
                                "id": "9e61d96d-fea3-44a4-8903-88d95eb266a8",
                                "userId": "f34feccc-0d0d-4365-982b-d6d1779e5aaf",
                                "externalUserId": "bigWillie",
                                "userFullName": "Willie Iles",
                                "approvalTypeId": "ece87cfe-33f9-45a4-8c51-a04a9f905c66",
                                "status": "APPROVED",
                                "createdAt": "2019-05-17T13:44:02Z"
                            }],
                        "canApprove": false,
                        "canReject": false,
                        "finalApprover": false
                    },
                    "canDelete": false,
                    "canCancel": true
                }
            },
            {
                status: 200,
                body: {
                    "id": "b438751d-6e6a-4601-8009-a82fa22bddcc",
                    "name": "Employee salaries",
                    "type": "SEPACT",
                    "status": "REJECTED",
                    "totalTransactionsCount": 9208,
                    "totalInstructedAmount": {
                        "amount": "50343009.00",
                        "currencyCode": "EUR"
                    },
                    "account": {
                        "name": "Main Account 8",
                        "identification": {
                            "identification": "NL18ABNA4956451237",
                            "schemeName": "IBAN"
                        }
                    },
                    "requestedExecutionDate": "2019-09-15",
                    "description": "Salary Sep 2019",
                    "bankStatus": "UNPROCESSABLE",
                    "reasonCode": "RJ08",
                    "reasonText": "Insufficient funds",
                    "errorDescription": "Target account has not enough funds to execute batch payments",
                    "fileName": "salaries-2019-02.csv",
                    "userFullName": "Davy Jones",
                    "uploadCompletedAt": "2019-02-22T14:08:34Z",
                    "canDelete": false,
                    "canCancel": false
                }
            },
            {
                status: 200,
                body: {
                    "id": "2f4c4bb6-c107-473a-adbd-7343b8f0ea80",
                    "name": "Salaries 2020-12",
                    "type": "ACH_CREDIT",
                    "status": "ACCEPTED",
                    "totalTransactionsCount": 401,
                    "totalInstructedAmount": {
                        "amount": "650980.25",
                        "currencyCode": "USD"
                    },
                    "account": {
                        "name": "Current Account",
                        "identification": {
                            "identification": "7649373647",
                            "schemeName": "BBAN"
                        }
                    },
                    "bankBranchCode": "113105452",
                    "companyId": "1800948594",
                    "companyName": "Your Company",
                    "requestedExecutionDate": "2020-12-18",
                    "fileName": "salaries-2020-12.ach",
                    "userFullName": "Ben Jones",
                    "uploadCompletedAt": "2020-12-15T15:28:10Z",
                    "canDelete": false,
                    "canCancel": false
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/batch-orders/{batchOrderId}/batch-payments/{batchPaymentId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "counterpartyName": "M. Frederiksen",
                    "counterpartyAccountNumber": "DE49500105174361358418",
                    "instructedAmount": {
                        "amount": "5576.00",
                        "currencyCode": "EUR"
                    },
                    "description": "Salary Feb 2019 empl.no:068572",
                    "status": "REJECTED",
                    "extendedDescription": "Addenda record for the credit payment \nAddenda record for the credit payment",
                    "bankStatus": "UNPROCESSABLE",
                    "reasonCode": "RJ05",
                    "reasonText": "Rejected by counterparty bank",
                    "errorDescription": "Target account has been closed."
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/batch-orders/{batchOrderId}/batch-payments",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "batchPayments": [{
                            "counterpartyName": "A. Hamers",
                            "counterpartyAccountNumber": "NL62ABNA0609476343",
                            "instructedAmount": {
                                "amount": "4102.00",
                                "currencyCode": "EUR"
                            },
                            "description": "Salaray Feb 2019 empl.no:902713",
                            "extendedDescriptionSize": 99999
                        }, {
                            "counterpartyName": "E. Gutierrez",
                            "counterpartyAccountNumber": "NL50ABNA4418364501",
                            "instructedAmount": {
                                "amount": "5287.00",
                                "currencyCode": "EUR"
                            },
                            "description": "Salaray Feb 2019 empl.no:256869",
                            "extendedDescriptionSize": 0
                        }, {
                            "counterpartyName": "C.M. McAlister",
                            "counterpartyAccountNumber": "BE98735114499693",
                            "instructedAmount": {
                                "amount": "3398.00",
                                "currencyCode": "EUR"
                            },
                            "description": "Salaray Feb 2019 empl.no:100642",
                            "extendedDescriptionSize": 456768
                        }, {
                            "counterpartyName": "K.C. Bennis",
                            "counterpartyAccountNumber": "NL67ABNA5515439724",
                            "instructedAmount": {
                                "amount": "4334.00",
                                "currencyCode": "EUR"
                            },
                            "description": "Salaray Feb 2019 empl.no:191872",
                            "extendedDescriptionSize": 0
                        }, {
                            "counterpartyName": "M. Frederiksen",
                            "counterpartyAccountNumber": "DE49500105174361358418",
                            "instructedAmount": {
                                "amount": "5576.00",
                                "currencyCode": "EUR"
                            },
                            "description": "Salaray Feb 2019 empl.no:068572",
                            "extendedDescriptionSize": 123,
                            "status": "REJECTED",
                            "bankStatus": "UNPROCESSABLE",
                            "reasonCode": "RJ05",
                            "reasonText": "Rejected by counterparty bank",
                            "errorDescription": "Target account has been closed."
                        }],
                    "totalBatchPayments": 5,
                    "batchOrder": {
                        "id": "2f3a6d96-f671-4f03-994d-8f6caaa53cab",
                        "name": "Employee salaries",
                        "type": "SEPACT",
                        "status": "ACCEPTED",
                        "totalTransactionsCount": 5,
                        "totalInstructedAmount": {
                            "amount": "2400123.00",
                            "currencyCode": "EUR"
                        },
                        "account": {
                            "name": "Main Current",
                            "identification": {
                                "identification": "CH8739215260566920524",
                                "schemeName": "IBAN"
                            }
                        },
                        "requestedExecutionDate": "2019-02-18",
                        "fileName": "salaries-2019-02.csv",
                        "userFullName": "Davy Jones",
                        "uploadCompletedAt": "2019-02-22T14:08:34Z",
                        "confidentialType": "confidential type"
                    }
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/batch-orders",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "id": "c1263fd6-8bd3-4a0b-be02-5db5cb1d6d06",
                    "status": "DRAFT"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/batch-orders/{batchOrderId}/batch-payments",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "id": "cad9e668-15a4-44bf-832d-67e7b41387ac"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Batch order not found",
                    "errors": []
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/batch-orders/{batchOrderId}/approvals",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {
                    "batchOrder": {
                        "id": "72d17d02-20a9-4f0b-b2b8-f83959eb187e",
                        "name": "Salaries May 2019",
                        "type": "SEPACT",
                        "creditDebitMixedIndicator": "CREDIT",
                        "status": "ENTERED",
                        "totalTransactionsCount": 6504,
                        "totalCreditTransactionsCount": 6504,
                        "totalInstructedAmount": {
                            "amount": "14009055.00",
                            "currencyCode": "EUR"
                        },
                        "totalCreditInstructedAmount": {
                            "amount": "14009055.00",
                            "currencyCode": "EUR"
                        },
                        "account": {
                            "name": "Salaries account",
                            "identification": {
                                "identification": "NL78ABNA3470416656",
                                "schemeName": "IBAN"
                            }
                        },
                        "requestedExecutionDate": "2019-05-26",
                        "approval": {
                            "id": "00b5ca29-eddd-49f4-a898-c3f185ab9a3a",
                            "userId": "4840854b-0597-47a6-8984-87cb50f970eb",
                            "externalUserId": "judi-miles",
                            "userFullName": "Judi Miles",
                            "serviceAgreementId": "ea7b3dfd-29b8-4559-872a-91f5576b56ef",
                            "itemId": "72d17d02-20a9-4f0b-b2b8-f83959eb187e",
                            "dataAccessItemId": "606c7afa-3c8a-4210-ad25-fc83893a4e49",
                            "resource": "Batches",
                            "function": "SEPACT",
                            "status": "PENDING",
                            "createdAt": "2019-08-10T18:09:21Z",
                            "records": [],
                            "canApprove": true,
                            "canReject": true,
                            "finalApprover": false
                        }
                    },
                    "approvalStatus": "PENDING"
                }
            },
            {
                status: 403,
                body: {
                    "message": "Limits has been breached, check the report for more details",
                    "checkTime": "2019-10-31T12:12:12Z",
                    "breachReport": [{
                            "limitedEntity": [{
                                    "ref": "1234567-12312-123123",
                                    "type": "Service Agreement",
                                    "description": "Kuhic, Gislason and Kemmer. SERVICE AGREEMENT"
                                }],
                            "shadow": false,
                            "currency": "EUR",
                            "user-BBID": "oleksii",
                            "breachInfo": [{
                                    "breachType": "THRESHOLD",
                                    "breachedLimitType": "PERIODIC",
                                    "timeframe": {
                                        "period": "daily",
                                        "startTime": "2019-10-31T00:00:00Z",
                                        "endTime": "2019-10-31T23:59:59Z"
                                    },
                                    "currentConsumption": "250.0",
                                    "currentThreshold": "499.9"
                                }, {
                                    "breachType": "CONSUMPTION",
                                    "timeframe": {
                                        "period": "monthly",
                                        "startTime": "2019-10-01T00:00:00Z",
                                        "endTime": "2019-10-31T23:59:59Z"
                                    },
                                    "currentConsumption": "9950.0",
                                    "currentThreshold": "10000.0"
                                }]
                        }, {
                            "user-BBID": "oleksii",
                            "shadow": false,
                            "currency": "EUR",
                            "breachInfo": [{
                                    "breachType": "CONSUMPTION",
                                    "timeframe": {
                                        "period": "daily",
                                        "startTime": "2019-10-31T00:00:00Z",
                                        "endTime": "2019-10-31T23:59:59Z"
                                    },
                                    "currentConsumption": "500.0",
                                    "currentThreshold": "1000.0"
                                }]
                        }, {
                            "limitedEntity": [{
                                    "ref": "1234567-12312-123123",
                                    "type": "Function Access Group",
                                    "description": "Batches approvers"
                                }, {
                                    "ref": "1234567-12312-123123",
                                    "type": "Function",
                                    "description": "Domestic batches"
                                }, {
                                    "ref": "Approve",
                                    "type": "Privilege",
                                    "description": "Approve"
                                }],
                            "shadow": false,
                            "currency": "EUR",
                            "breachInfo": [{
                                    "breachType": "THRESHOLD",
                                    "timeframe": {
                                        "period": "daily",
                                        "startTime": "2019-10-31T00:00:00Z",
                                        "endTime": "2019-10-31T23:59:59Z"
                                    },
                                    "currentConsumption": "250.0",
                                    "currentThreshold": "499.9"
                                }]
                        }, {
                            "limitedEntity": [{
                                    "ref": "1234567-12312-123123",
                                    "type": "Legal Entity",
                                    "description": "Kuhic, Gislason and Kemmer"
                                }],
                            "shadow": false,
                            "currency": "EUR",
                            "breachInfo": [{
                                    "breachType": "CONSUMPTION",
                                    "timeframe": {
                                        "period": "quarterly",
                                        "startTime": "2019-10-01T00:00:00Z",
                                        "endTime": "2019-12-31T23:59:59Z"
                                    },
                                    "currentConsumption": "99950.0",
                                    "currentThreshold": "100000.0"
                                }]
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/batch-orders/{batchOrderId}/batch-payments/{batchPaymentId}",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {
                    "id": "7dc95f0a-deba-47be-8f5d-6d0454ef271c",
                    "counterpartyName": "Uriel Johnson",
                    "counterpartyAccountNumber": "HU65109180019369738837536884",
                    "instructedAmount": {
                        "amount": "830.00",
                        "currencyCode": "EUR"
                    },
                    "description": "Order 00023162"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Batch order not found",
                    "errors": []
                }
            },
            {
                status: 404,
                body: {
                    "message": "Batch payment not found",
                    "errors": []
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/batch-orders/{batchOrderId}",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {
                    "id": "f7e4f0bf-6080-46be-bd5d-7dad2b054495",
                    "name": "Employee salaries",
                    "type": "SEPACT",
                    "creditDebitMixedIndicator": "CREDIT",
                    "status": "CANCELLED",
                    "totalTransactionsCount": 120,
                    "totalCreditTransactionsCount": 120,
                    "totalInstructedAmount": {
                        "amount": "322038.00",
                        "currencyCode": "EUR"
                    },
                    "totalCreditInstructedAmount": {
                        "amount": "322038.00",
                        "currencyCode": "EUR"
                    },
                    "account": {
                        "name": "Main Current",
                        "identification": {
                            "identification": "NL54ABNA2366058993",
                            "schemeName": "IBAN"
                        }
                    },
                    "requestedExecutionDate": "2012-01-01",
                    "fileName": "salaries-2020-01.csv",
                    "userFullName": "John Davis",
                    "uploadCompletedAt": "2019-12-31T23:55:31Z",
                    "canDelete": false,
                    "canCancel": false,
                    "approval": {
                        "id": "ac150312-1bec-4075-934f-23600886cb4e",
                        "userId": "64350fcf-ddd1-4cdc-b725-f6c76f89f7c2",
                        "externalUserId": "jdavis",
                        "userFullName": "John Davis",
                        "serviceAgreementId": "c94dc22f-381f-4127-a9f2-db2f5a314e41",
                        "itemId": "f7e4f0bf-6080-46be-bd5d-7dad2b054495",
                        "dataAccessItemId": "e8daae6b-0276-4533-b64d-e3c73afb47a0",
                        "resource": "Batches",
                        "function": "SEPACT",
                        "status": "PENDING",
                        "createdAt": "2019-12-31T23:56:08Z",
                        "records": [{
                                "id": "9e61d96d-fea3-44a4-8903-88d95eb266a8",
                                "userId": "f34feccc-0d0d-4365-982b-d6d1779e5aaf",
                                "externalUserId": "twilliams",
                                "userFullName": "Tobi Williams",
                                "approvalTypeId": "16f655f3-3abf-438d-8351-2c667d5a3c86",
                                "status": "APPROVED",
                                "createdAt": "2019-12-31T23:57:20Z"
                            }],
                        "canApprove": false,
                        "canReject": false,
                        "finalApprover": false
                    }
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad request",
                    "errors": [{
                            "message": "The requested status ({{requestedStatus}}) is not valid. The status can be set to {{expectedStatus}}.",
                            "key": "BatchOrderCannotBeUpdated",
                            "context": {
                                "requestedStatus": "READY",
                                "expectedStatus": "CANCELLED"
                            }
                        }]
                }
            },
            {
                status: 409,
                body: {
                    "message": "Conflict",
                    "errors": [{
                            "message": "The current status of the batch order ({{currentStatus}}) does not allow cancellation.",
                            "key": "BatchOrderCannotBeCancelled",
                            "context": {
                                "currentStatus": "PROCESSED"
                            }
                        }]
                }
            },
            {
                status: 409,
                body: {
                    "message": "Conflict",
                    "errors": [{
                            "message": "Only partially approved batch orders can be cancelled.",
                            "key": "BatchOrderWithoutApprovalsCannotBeCancelled"
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    },
]);
