// @backbase/payment-batch-http-ang/esm2015/api/batchUploads.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { PAYMENT_BATCH_BASE_PATH } from '../variables';
import { PaymentBatchConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class BatchUploadsHttpService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new PaymentBatchConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    canConsumeForm(consumes) {
        const form = 'multipart/form-data';
        for (const consume of consumes) {
            if (form === consume) {
                return true;
            }
        }
        return false;
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    createBatchOrdersForId(requestParameters, observe = 'body', reportProgress = false, options) {
        const _batchUploadId = requestParameters["batchUploadId"];
        if (_batchUploadId === null || _batchUploadId === undefined) {
            throw new Error('Required parameter batchUploadId was null or undefined when calling createBatchOrdersForId.');
        }
        const _confidentialType = requestParameters["confidentialType"];
        const _file = requestParameters["file"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_confidentialType !== undefined && _confidentialType !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _confidentialType, 'confidentialType');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'multipart/form-data'
        ];
        const canConsumeForm = this.canConsumeForm(consumes);
        let formParams;
        let useForm = false;
        let convertFormParamsToString = false;
        useForm = canConsumeForm;
        if (useForm) {
            formParams = new FormData();
        }
        else {
            formParams = new HttpParams({ encoder: this.encoder });
        }
        if (_file !== undefined) {
            formParams = formParams.append('file', _file) || formParams;
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v2/batch-uploads/${encodeURIComponent(String(_batchUploadId))}/batch-orders`, convertFormParamsToString ? formParams.toString() : formParams, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    createBatchOrdersForIdUrl(requestParameters) {
        const _batchUploadId = requestParameters["batchUploadId"];
        if (_batchUploadId === null || _batchUploadId === undefined) {
            throw new Error('Required parameter batchUploadId was null or undefined when calling createBatchOrdersForId.');
        }
        const _confidentialType = requestParameters["confidentialType"];
        const _file = requestParameters["file"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_confidentialType !== undefined && _confidentialType !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _confidentialType, 'confidentialType');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/batch-uploads/${encodeURIComponent(String(_batchUploadId))}/batch-orders${queryString ? `?${queryString}` : ''}`;
    }
    deleteBatchUploadById(requestParameters, observe = 'body', reportProgress = false, options) {
        const _batchUploadId = requestParameters["batchUploadId"];
        if (_batchUploadId === null || _batchUploadId === undefined) {
            throw new Error('Required parameter batchUploadId was null or undefined when calling deleteBatchUploadById.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.delete(`${this.configuration.basePath}/client-api/v2/batch-uploads/${encodeURIComponent(String(_batchUploadId))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    deleteBatchUploadByIdUrl(requestParameters) {
        const _batchUploadId = requestParameters["batchUploadId"];
        if (_batchUploadId === null || _batchUploadId === undefined) {
            throw new Error('Required parameter batchUploadId was null or undefined when calling deleteBatchUploadById.');
        }
        return `${this.configuration.basePath}/client-api/v2/batch-uploads/${encodeURIComponent(String(_batchUploadId))}`;
    }
    getBatchUploadById(requestParameters, observe = 'body', reportProgress = false, options) {
        const _batchUploadId = requestParameters["batchUploadId"];
        if (_batchUploadId === null || _batchUploadId === undefined) {
            throw new Error('Required parameter batchUploadId was null or undefined when calling getBatchUploadById.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/batch-uploads/${encodeURIComponent(String(_batchUploadId))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getBatchUploadByIdUrl(requestParameters) {
        const _batchUploadId = requestParameters["batchUploadId"];
        if (_batchUploadId === null || _batchUploadId === undefined) {
            throw new Error('Required parameter batchUploadId was null or undefined when calling getBatchUploadById.');
        }
        return `${this.configuration.basePath}/client-api/v2/batch-uploads/${encodeURIComponent(String(_batchUploadId))}`;
    }
    getBatchUploads(requestParameters, observe = 'body', reportProgress = false, options) {
        const _from = requestParameters["from"];
        const _cursor = requestParameters["cursor"];
        const _size = requestParameters["size"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_cursor !== undefined && _cursor !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _cursor, 'cursor');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/batch-uploads`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getBatchUploadsUrl(requestParameters) {
        const _from = requestParameters["from"];
        const _cursor = requestParameters["cursor"];
        const _size = requestParameters["size"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_cursor !== undefined && _cursor !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _cursor, 'cursor');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/batch-uploads${queryString ? `?${queryString}` : ''}`;
    }
    putBatchUploadIdByBatchUploadId(requestParameters, observe = 'body', reportProgress = false, options) {
        const _batchUploadId = requestParameters["batchUploadId"];
        if (_batchUploadId === null || _batchUploadId === undefined) {
            throw new Error('Required parameter batchUploadId was null or undefined when calling putBatchUploadIdByBatchUploadId.');
        }
        const _batchUploadPutRequest = requestParameters["batchUploadPutRequest"];
        if (_batchUploadPutRequest === null || _batchUploadPutRequest === undefined) {
            throw new Error('Required parameter batchUploadPutRequest was null or undefined when calling putBatchUploadIdByBatchUploadId.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v2/batch-uploads/${encodeURIComponent(String(_batchUploadId))}`, _batchUploadPutRequest, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    putBatchUploadIdByBatchUploadIdUrl(requestParameters) {
        const _batchUploadId = requestParameters["batchUploadId"];
        if (_batchUploadId === null || _batchUploadId === undefined) {
            throw new Error('Required parameter batchUploadId was null or undefined when calling putBatchUploadIdByBatchUploadId.');
        }
        const _batchUploadPutRequest = requestParameters["batchUploadPutRequest"];
        if (_batchUploadPutRequest === null || _batchUploadPutRequest === undefined) {
            throw new Error('Required parameter batchUploadPutRequest was null or undefined when calling putBatchUploadIdByBatchUploadId.');
        }
        return `${this.configuration.basePath}/client-api/v2/batch-uploads/${encodeURIComponent(String(_batchUploadId))}`;
    }
    startBatchUpload(requestParameters, observe = 'body', reportProgress = false, options) {
        const _startUploadBatchOrdersPostRequest = requestParameters["startUploadBatchOrdersPostRequest"];
        if (_startUploadBatchOrdersPostRequest === null || _startUploadBatchOrdersPostRequest === undefined) {
            throw new Error('Required parameter startUploadBatchOrdersPostRequest was null or undefined when calling startBatchUpload.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v2/batch-uploads`, _startUploadBatchOrdersPostRequest, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    startBatchUploadUrl(requestParameters) {
        const _startUploadBatchOrdersPostRequest = requestParameters["startUploadBatchOrdersPostRequest"];
        if (_startUploadBatchOrdersPostRequest === null || _startUploadBatchOrdersPostRequest === undefined) {
            throw new Error('Required parameter startUploadBatchOrdersPostRequest was null or undefined when calling startBatchUpload.');
        }
        return `${this.configuration.basePath}/client-api/v2/batch-uploads`;
    }
}
BatchUploadsHttpService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BatchUploadsHttpService, deps: [{ token: i1.HttpClient }, { token: PAYMENT_BATCH_BASE_PATH, optional: true }, { token: i2.PaymentBatchConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
BatchUploadsHttpService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BatchUploadsHttpService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BatchUploadsHttpService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [PAYMENT_BATCH_BASE_PATH]
                }] }, { type: i2.PaymentBatchConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
