// @backbase/payment-request-widget-ang/esm2015/src/payment-request-widget/payment-request-amount/payment-request-amount.component.js
import { Component, Directive, Input } from '@angular/core';
import { BbTemplate } from '@backbase/foundation-ang/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/amount";
export class PaymentRequestAmountComponent {
    constructor() {
        /**
         * Component context
         */
        this.hostRef = this;
        /**
         * Amount of the request
         */
        this.amount = '';
        /**
         * Currency of the request
         */
        this.currency = '';
    }
}
PaymentRequestAmountComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentRequestAmountComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
PaymentRequestAmountComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: PaymentRequestAmountComponent, selector: "bb-payment-request-amount", inputs: { amount: "amount", currency: "currency" }, ngImport: i0, template: "<ng-container\n  bbPaymentRequestAmountCustomizable\n  [bbHostRef]=\"hostRef\"\n  [bbTemplateContext]=\"{\n    amount: amount,\n    currency: currency\n  }\">\n</ng-container>\n\n<ng-template bbPaymentRequestAmountCustomizable let-hostRef let-amount=\"context.amount\" let-currency=\"context.currency\">\n  <div\n    i18n=\"Amount|Amount message@@payment-request.label.amount\"\n    data-role=\"amount-label\"\n    class=\"bb-block bb-block--sm bb-text-bold\">\n    Amount\n  </div>\n  <bb-amount-ui\n    class=\"bb-payment-request__amount\"\n    data-role=\"amount-value\"\n    [amount]=\"amount\"\n    [currency]=\"currency\"></bb-amount-ui>\n</ng-template>\n", components: [{ type: i0.forwardRef(function () { return i1.AmountComponent; }), selector: "bb-amount-ui", inputs: ["currency", "showPlusSign", "mapCurrency", "showPercent", "abbreviate", "decimalPlaces", "trailingZeroes", "amount"] }], directives: [{ type: i0.forwardRef(function () { return PaymentRequestAmountCustomizableDirective; }), selector: "[bbPaymentRequestAmountCustomizable]" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentRequestAmountComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-payment-request-amount',
                    templateUrl: './payment-request-amount.component.html',
                }]
        }], propDecorators: { amount: [{
                type: Input
            }], currency: [{
                type: Input
            }] } });
export class PaymentRequestAmountCustomizableDirective extends BbTemplate {
}
PaymentRequestAmountCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentRequestAmountCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
PaymentRequestAmountCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.15", type: PaymentRequestAmountCustomizableDirective, selector: "[bbPaymentRequestAmountCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentRequestAmountCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbPaymentRequestAmountCustomizable]',
                }]
        }] });
