// @backbase/transactions-common-ang/esm2015/lib/transactions-common.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TransactionUpdateService } from './services/transaction-update.service';
import { EnumValuesService } from './services/enum-values.service';
import { TransactionsNumberOfResultsFoundComponent, TransactionsNumberOfResultsFoundCustomizableDirective, } from './components/transactions-controls/transactions-number-of-results-found.component';
import { CategoriesService } from './services/categories.service';
import { TransactionsErrorStateCustomizableDirective, TransactionsNoAccountSelectedStateCustomizableDirective, } from './components/base-transactions-container';
import { AccountsService } from './services/accounts.service';
import { AccessibilityKeyboardDirective } from './components/transactions-controls/accessibility-keyboard.directive';
import { TransactionCheckImageComponent } from './components/transactions-controls/transaction-check-image.component';
import { TransactionGeoLocationMapComponent } from './components/transactions-controls/transaction-geo-location-map.component';
import { MapAPILoaderService } from './services/map-api-loader.service';
import { TransactionsMapUiComponent } from './components/transactions-controls/transactions-map-ui.component';
import { TransactionCategoryIconComponent } from './components/transactions-controls/transaction-category-icon.component';
import { TransactionsIconCategoriesFactoryService } from './services/transactions-icon-categories-factory.service';
import { CapitalizePipe } from './components/capitalize.pipe';
import { TransactionCheckImageItemComponent } from './components/transactions-controls/transaction-check-image-item.component';
import { DocumentViewerBackdropComponent } from './components/document-viewer-ui/document-viewer-backdrop/document-viewer-backdrop.component';
import { DocumentViewerContainerComponent } from './components/document-viewer-ui/document-viewer-container/document-viewer-container.component';
import { DocumentViewerHeaderComponent } from './components/document-viewer-ui/document-viewer-header/document-viewer-header.component';
import { DocumentViewerInteractorComponent } from './components/document-viewer-ui/document-viewer-interactor/document-viewer-interactor.component';
import { ImageViewerComponent } from './components/document-viewer-ui/image-viewer/image-viewer.component';
import { TransactionCheckImageDocumentViewerComponent } from './components/transactions-controls/transaction-check-image-document-viewer.component';
import { A11yModule } from '@angular/cdk/a11y';
import { DomAttributesService } from './services/dom-attributes.service';
import { DynamicIdDirective } from './components/dynamic-id.directive';
import { MessagesService } from './services/messages.service';
import { TransactionsInquiryDisputeComponent } from './components/transactions-controls/transactions-inquiry-dispute.component';
import { ReactiveFormsModule } from '@angular/forms';
import { TransactionsInquireAndDisputeUIService } from './services/transactions-inquire-and-dispute-ui.service';
import { AvailableFiltersConfigService } from './services/available-filters-config.service';
import { PaymentsBatchService } from './services/payments-batch.service';
import { TransactionsDisputeStatusTagComponent } from './components/transactions-controls/transaction-dispute-status-tag.component';
import { TransactionsDisputeBannerComponent } from './components/transactions-controls/transaction-dispute-banner.component';
import { ButtonModule } from '@backbase/ui-ang/button';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { DropdownMenuModule } from '@backbase/ui-ang/dropdown-menu';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { IconModule } from '@backbase/ui-ang/icon';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { InputRadioGroupModule } from '@backbase/ui-ang/input-radio-group';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { RichTextEditorModule } from '@backbase/ui-ang/rich-text-editor';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { InputFileModule } from '@backbase/ui-ang/input-file';
import { FileAttachmentModule } from '@backbase/ui-ang/file-attachment';
import { AlertModule } from '@backbase/ui-ang/alert';
import { ModalModule } from '@backbase/ui-ang/modal';
import { BadgeModule } from '@backbase/ui-ang/badge';
import * as i0 from "@angular/core";
export class TransactionsCommonModule {
}
TransactionsCommonModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TransactionsCommonModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
TransactionsCommonModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TransactionsCommonModule, declarations: [TransactionsNumberOfResultsFoundComponent,
        TransactionCheckImageComponent,
        TransactionGeoLocationMapComponent,
        TransactionsErrorStateCustomizableDirective,
        TransactionsNoAccountSelectedStateCustomizableDirective,
        AccessibilityKeyboardDirective,
        TransactionsNumberOfResultsFoundCustomizableDirective,
        TransactionsMapUiComponent,
        TransactionCategoryIconComponent,
        TransactionCheckImageItemComponent,
        CapitalizePipe,
        DocumentViewerBackdropComponent,
        DocumentViewerContainerComponent,
        DocumentViewerHeaderComponent,
        DocumentViewerInteractorComponent,
        ImageViewerComponent,
        TransactionCheckImageDocumentViewerComponent,
        DynamicIdDirective,
        TransactionsInquiryDisputeComponent,
        TransactionsDisputeStatusTagComponent,
        TransactionsDisputeBannerComponent], imports: [CommonModule,
        IconModule,
        ButtonModule,
        LoadButtonModule,
        InputRadioGroupModule,
        TooltipModule,
        ReactiveFormsModule,
        LoadingIndicatorModule,
        EmptyStateModule,
        DropdownMenuModule,
        A11yModule,
        RichTextEditorModule,
        InputValidationMessageModule,
        InputFileModule,
        FileAttachmentModule,
        AlertModule,
        ModalModule,
        BadgeModule], exports: [TransactionsNumberOfResultsFoundComponent,
        TransactionsErrorStateCustomizableDirective,
        TransactionsNoAccountSelectedStateCustomizableDirective,
        AccessibilityKeyboardDirective,
        TransactionCheckImageComponent,
        TransactionCheckImageItemComponent,
        TransactionGeoLocationMapComponent,
        TransactionsNumberOfResultsFoundCustomizableDirective,
        IconModule,
        A11yModule,
        TransactionsMapUiComponent,
        TransactionCategoryIconComponent,
        CapitalizePipe,
        DocumentViewerBackdropComponent,
        DocumentViewerContainerComponent,
        DocumentViewerHeaderComponent,
        DocumentViewerInteractorComponent,
        ImageViewerComponent,
        TransactionCheckImageDocumentViewerComponent,
        EmptyStateModule,
        LoadingIndicatorModule,
        DropdownMenuModule,
        DynamicIdDirective,
        TransactionsInquiryDisputeComponent,
        TransactionsDisputeStatusTagComponent,
        TransactionsDisputeBannerComponent] });
TransactionsCommonModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TransactionsCommonModule, providers: [
        TransactionUpdateService,
        EnumValuesService,
        CategoriesService,
        AccountsService,
        TransactionsIconCategoriesFactoryService,
        MapAPILoaderService,
        DomAttributesService,
        MessagesService,
        TransactionsInquireAndDisputeUIService,
        PaymentsBatchService,
        AvailableFiltersConfigService,
    ], imports: [[
            CommonModule,
            IconModule,
            ButtonModule,
            LoadButtonModule,
            InputRadioGroupModule,
            TooltipModule,
            ReactiveFormsModule,
            LoadingIndicatorModule,
            EmptyStateModule,
            DropdownMenuModule,
            A11yModule,
            RichTextEditorModule,
            InputValidationMessageModule,
            InputFileModule,
            FileAttachmentModule,
            AlertModule,
            ModalModule,
            BadgeModule,
        ], IconModule,
        A11yModule,
        EmptyStateModule,
        LoadingIndicatorModule,
        DropdownMenuModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TransactionsCommonModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        IconModule,
                        ButtonModule,
                        LoadButtonModule,
                        InputRadioGroupModule,
                        TooltipModule,
                        ReactiveFormsModule,
                        LoadingIndicatorModule,
                        EmptyStateModule,
                        DropdownMenuModule,
                        A11yModule,
                        RichTextEditorModule,
                        InputValidationMessageModule,
                        InputFileModule,
                        FileAttachmentModule,
                        AlertModule,
                        ModalModule,
                        BadgeModule,
                    ],
                    providers: [
                        TransactionUpdateService,
                        EnumValuesService,
                        CategoriesService,
                        AccountsService,
                        TransactionsIconCategoriesFactoryService,
                        MapAPILoaderService,
                        DomAttributesService,
                        MessagesService,
                        TransactionsInquireAndDisputeUIService,
                        PaymentsBatchService,
                        AvailableFiltersConfigService,
                    ],
                    declarations: [
                        TransactionsNumberOfResultsFoundComponent,
                        TransactionCheckImageComponent,
                        TransactionGeoLocationMapComponent,
                        TransactionsErrorStateCustomizableDirective,
                        TransactionsNoAccountSelectedStateCustomizableDirective,
                        AccessibilityKeyboardDirective,
                        TransactionsNumberOfResultsFoundCustomizableDirective,
                        TransactionsMapUiComponent,
                        TransactionCategoryIconComponent,
                        TransactionCheckImageItemComponent,
                        CapitalizePipe,
                        DocumentViewerBackdropComponent,
                        DocumentViewerContainerComponent,
                        DocumentViewerHeaderComponent,
                        DocumentViewerInteractorComponent,
                        ImageViewerComponent,
                        TransactionCheckImageDocumentViewerComponent,
                        DynamicIdDirective,
                        TransactionsInquiryDisputeComponent,
                        TransactionsDisputeStatusTagComponent,
                        TransactionsDisputeBannerComponent,
                    ],
                    exports: [
                        TransactionsNumberOfResultsFoundComponent,
                        TransactionsErrorStateCustomizableDirective,
                        TransactionsNoAccountSelectedStateCustomizableDirective,
                        AccessibilityKeyboardDirective,
                        TransactionCheckImageComponent,
                        TransactionCheckImageItemComponent,
                        TransactionGeoLocationMapComponent,
                        TransactionsNumberOfResultsFoundCustomizableDirective,
                        IconModule,
                        A11yModule,
                        TransactionsMapUiComponent,
                        TransactionCategoryIconComponent,
                        CapitalizePipe,
                        DocumentViewerBackdropComponent,
                        DocumentViewerContainerComponent,
                        DocumentViewerHeaderComponent,
                        DocumentViewerInteractorComponent,
                        ImageViewerComponent,
                        TransactionCheckImageDocumentViewerComponent,
                        EmptyStateModule,
                        LoadingIndicatorModule,
                        DropdownMenuModule,
                        DynamicIdDirective,
                        TransactionsInquiryDisputeComponent,
                        TransactionsDisputeStatusTagComponent,
                        TransactionsDisputeBannerComponent,
                    ],
                }]
        }] });
