// @backbase/product-summary-common-ang/esm2015/src/classes/product-summary-account-details-base/product-summary-account-details-base.component.js
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { MaskableAttribute } from '@backbase/data-ang/arrangements';
import * as i0 from "@angular/core";
export class ProductSummaryAccountDetailsBaseComponent {
    constructor() {
        /**
         * List of maskable attributes. Holds states of the attributes.
         */
        this.maskableAttributes = [];
        /**
         * Event when a user clicked on the `unmask-attribute-button` button
         */
        this.unmaskAttribute = new EventEmitter();
        /**
         * Event when a user clicked on the `mask-attribute-button` button
         */
        this.maskAttribute = new EventEmitter();
        /**
         * All unmaskable attributes
         */
        this.maskableAttribute = MaskableAttribute;
        /**
         * Details info sections availability
         */
        this.infoAvailability = {};
    }
    /**
     * Product type data that this component displays.
     */
    set product(data) {
        this._product = data;
        this.infoAvailability = this.getProductInfoAvailability(data);
    }
    get product() {
        return this._product;
    }
    /**
     * Returns `true` if the given `attributeName` is masked
     *
     * @param attributeName
     */
    isMasked(attributeName) {
        const maskableAttribute = this.maskableAttributes.find(attribute => attribute.attributeName === attributeName);
        return (maskableAttribute === null || maskableAttribute === void 0 ? void 0 : maskableAttribute.masked) === undefined ? true : maskableAttribute === null || maskableAttribute === void 0 ? void 0 : maskableAttribute.masked;
    }
    /**
     * Event handler for the `click` event of the unmask buttons
     * Emits `MaskableAttribute` to the `unmaskAttribute` Output
     *
     * @param attributeName
     */
    onUnmaskAttribute(attributeName) {
        if (Object.values(MaskableAttribute).includes(attributeName)) {
            this.unmaskAttribute.emit(attributeName);
        }
    }
    /**
     * Event handler for the `click` event of the mask buttons
     * Emits `MaskableAttribute` to the `maskAttribute` Output
     *
     * @param attributeName
     */
    onMaskAttribute(attributeName) {
        if (Object.values(MaskableAttribute).includes(attributeName)) {
            this.maskAttribute.emit(attributeName);
        }
    }
    // eslint-disable-next-line @typescript-eslint/no-unused-vars,no-unused-vars
    getProductInfoAvailability(product) {
        return {};
    }
}
ProductSummaryAccountDetailsBaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ProductSummaryAccountDetailsBaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ProductSummaryAccountDetailsBaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: ProductSummaryAccountDetailsBaseComponent, selector: "ng-component", inputs: { product: "product", maskableAttributes: "maskableAttributes" }, outputs: { unmaskAttribute: "unmaskAttribute", maskAttribute: "maskAttribute" }, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ProductSummaryAccountDetailsBaseComponent, decorators: [{
            type: Component,
            args: [{ template: '' }]
        }], propDecorators: { product: [{
                type: Input
            }], maskableAttributes: [{
                type: Input
            }], unmaskAttribute: [{
                type: Output
            }], maskAttribute: [{
                type: Output
            }] } });
