// @backbase/categories-management-http-ang/esm2015/api/categories.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { CATEGORIES_MANAGEMENT_BASE_PATH } from '../variables';
import { CategoriesManagementConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class CategoriesHttpService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new CategoriesManagementConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    getCategories(requestParameters, observe = 'body', reportProgress = false, options) {
        const _categoryType = requestParameters["categoryType"];
        const _level = requestParameters["level"];
        const _parentId = requestParameters["parentId"];
        const _externalIds = requestParameters["externalIds"];
        const _internalIds = requestParameters["internalIds"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_categoryType !== undefined && _categoryType !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _categoryType, 'categoryType');
        }
        if (_level !== undefined && _level !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _level, 'level');
        }
        if (_parentId !== undefined && _parentId !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _parentId, 'parentId');
        }
        if (_externalIds) {
            _externalIds.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'externalIds');
            });
        }
        if (_internalIds) {
            _internalIds.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'internalIds');
            });
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/categories`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getCategoriesUrl(requestParameters) {
        const _categoryType = requestParameters["categoryType"];
        const _level = requestParameters["level"];
        const _parentId = requestParameters["parentId"];
        const _externalIds = requestParameters["externalIds"];
        const _internalIds = requestParameters["internalIds"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_categoryType !== undefined && _categoryType !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _categoryType, 'categoryType');
        }
        if (_level !== undefined && _level !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _level, 'level');
        }
        if (_parentId !== undefined && _parentId !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _parentId, 'parentId');
        }
        if (_externalIds) {
            _externalIds.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'externalIds');
            });
        }
        if (_internalIds) {
            _internalIds.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'internalIds');
            });
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/categories${queryString ? `?${queryString}` : ''}`;
    }
    getCategory(requestParameters, observe = 'body', reportProgress = false, options) {
        const _id = requestParameters["id"];
        if (_id === null || _id === undefined) {
            throw new Error('Required parameter id was null or undefined when calling getCategory.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/categories/${encodeURIComponent(String(_id))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getCategoryUrl(requestParameters) {
        const _id = requestParameters["id"];
        if (_id === null || _id === undefined) {
            throw new Error('Required parameter id was null or undefined when calling getCategory.');
        }
        return `${this.configuration.basePath}/client-api/v2/categories/${encodeURIComponent(String(_id))}`;
    }
}
CategoriesHttpService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CategoriesHttpService, deps: [{ token: i1.HttpClient }, { token: CATEGORIES_MANAGEMENT_BASE_PATH, optional: true }, { token: i2.CategoriesManagementConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
CategoriesHttpService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CategoriesHttpService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CategoriesHttpService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [CATEGORIES_MANAGEMENT_BASE_PATH]
                }] }, { type: i2.CategoriesManagementConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
