// @backbase/contact-common-ang/esm2015/lib/components/contact-full-address/contact-full-address.module.js
import { NgModule } from '@angular/core';
import { ContactFullAddressComponent } from './contact-full-address.component';
import { ContactFullAddressCustomizableDirective } from './contact-full-address.directive';
import { CountryItemModule } from '../country-item/country-item.module';
import { CommonModule } from '@angular/common';
import * as i0 from "@angular/core";
/**
 * Exports:
 * - ContactFullAddressComponent
 * - ContactFullAddressCustomizableDirective
 */
export class ContactFullAddressModule {
}
/** @nocollapse */ ContactFullAddressModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactFullAddressModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ ContactFullAddressModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactFullAddressModule, declarations: [ContactFullAddressComponent, ContactFullAddressCustomizableDirective], imports: [CommonModule, CountryItemModule], exports: [ContactFullAddressComponent, ContactFullAddressCustomizableDirective] });
/** @nocollapse */ ContactFullAddressModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactFullAddressModule, imports: [[CommonModule, CountryItemModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactFullAddressModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ContactFullAddressComponent, ContactFullAddressCustomizableDirective],
                    exports: [ContactFullAddressComponent, ContactFullAddressCustomizableDirective],
                    imports: [CommonModule, CountryItemModule],
                }]
        }] });
