// @backbase/cards-travel-notice-widget-ang/esm2015/src/components/cards-travel-notice-list/cards-travel-notice-list.component.js
import { Component, Input, Output, EventEmitter, Directive } from '@angular/core';
import { BbTemplate } from '@backbase/foundation-ang/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/icon";
import * as i2 from "./cards-travel-notice-list-item.component";
import * as i3 from "@backbase/ui-ang/button";
import * as i4 from "@angular/common";
export class CardsTravelNoticeListComponent {
    constructor() {
        this.hostRef = this;
        /**
         * EventEmitter for triggering post event.
         */
        this.create = new EventEmitter();
        /**
         * EventEmitter for triggering delete event.
         */
        this.deleteListItem = new EventEmitter();
    }
    onShowCreateForm() {
        this.create.emit({ goToPage: 'create' });
    }
    onDeleteListItem(travelNoticeItem) {
        this.deleteListItem.emit(travelNoticeItem);
    }
}
CardsTravelNoticeListComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardsTravelNoticeListComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
CardsTravelNoticeListComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: CardsTravelNoticeListComponent, selector: "bb-cards-travel-notice-list", inputs: { travelNoticeList: "travelNoticeList", destinationList: "destinationList" }, outputs: { create: "create", deleteListItem: "deleteListItem" }, ngImport: i0, template: "<ng-container bbTravelNoticeListCustomizable [bbHostRef]=\"hostRef\"></ng-container>\n\n<ng-template bbTravelNoticeListCustomizable let-hostRef>\n  <div>\n    <div class=\"bb-text-align-right bb-block bb-block--xl bb-stack\">\n      <div class=\"bb-stack__item bb-stack__item--push-right\">\n        <button bbButton data-role=\"add-button\" (click)=\"hostRef.onShowCreateForm()\">\n          <bb-icon-ui name=\"add\"></bb-icon-ui>\n          <span i18n=\"Create new travel notice button@@cards.travel.notice.createNew.button\"> Create New</span>\n        </button>\n      </div>\n    </div>\n    <div class=\"card card-lg\">\n      <div class=\"card-body\">\n        <ng-container bbTravelNoticeListItemCustomizable [bbHostRef]=\"hostRef\"></ng-container>\n      </div>\n    </div>\n  </div>\n</ng-template>\n\n<ng-template bbTravelNoticeListItemCustomizable let-hostRef>\n  <div class=\"bb-travel-notice-list bb-list\">\n    <div class=\"bb-list__item\" *ngFor=\"let travelNoticeItem of hostRef.travelNoticeList\">\n      <bb-cards-travel-notice-list-item\n        [travelNoticeItem]=\"travelNoticeItem\"\n        [destinationList]=\"hostRef.destinationList\" \n        (delete)=\"hostRef.onDeleteListItem($event)\"\n      ></bb-cards-travel-notice-list-item>\n    </div>\n  </div>\n</ng-template>", components: [{ type: i0.forwardRef(function () { return i1.IconComponent; }), selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i0.forwardRef(function () { return i2.CardsTravelNoticeListItemComponent; }), selector: "bb-cards-travel-notice-list-item", inputs: ["travelNoticeItem", "destinationList"], outputs: ["delete"] }], directives: [{ type: i0.forwardRef(function () { return TravelNoticeListCustomizableDirective; }), selector: "[bbTravelNoticeListCustomizable]" }, { type: i0.forwardRef(function () { return i3.ButtonDirective; }), selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }, { type: i0.forwardRef(function () { return TravelNoticeListItemCustomizableDirective; }), selector: "[bbTravelNoticeListItemCustomizable]" }, { type: i0.forwardRef(function () { return i4.NgForOf; }), selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardsTravelNoticeListComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-cards-travel-notice-list',
                    templateUrl: './cards-travel-notice-list.component.html',
                }]
        }], propDecorators: { travelNoticeList: [{
                type: Input
            }], destinationList: [{
                type: Input
            }], create: [{
                type: Output
            }], deleteListItem: [{
                type: Output
            }] } });
export class TravelNoticeListCustomizableDirective extends BbTemplate {
}
TravelNoticeListCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: TravelNoticeListCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
TravelNoticeListCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.15", type: TravelNoticeListCustomizableDirective, selector: "[bbTravelNoticeListCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: TravelNoticeListCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbTravelNoticeListCustomizable]',
                }]
        }] });
export class TravelNoticeListItemCustomizableDirective extends BbTemplate {
}
TravelNoticeListItemCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: TravelNoticeListItemCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
TravelNoticeListItemCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.15", type: TravelNoticeListItemCustomizableDirective, selector: "[bbTravelNoticeListItemCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: TravelNoticeListItemCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbTravelNoticeListItemCustomizable]',
                }]
        }] });
