// @backbase/accounts-transactions-journeys-shared-ui-transform-ang/esm2015/lib/get-value-pipe/get-value.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { GetValuePipe } from './get-value.pipe';
import * as i0 from "@angular/core";
const pipes = [GetValuePipe];
export class GetValuePipeModule {
}
/** @nocollapse */ GetValuePipeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: GetValuePipeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ GetValuePipeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: GetValuePipeModule, declarations: [GetValuePipe], imports: [CommonModule], exports: [GetValuePipe] });
/** @nocollapse */ GetValuePipeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: GetValuePipeModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: GetValuePipeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [...pipes],
                    exports: [...pipes],
                }]
        }] });
