// @backbase/payment-orders-ang/esm2015/lib/ui/components/beneficiary-selector-ui/beneficiary-selector-ui.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PaymentsSrSearchResultInfoModule } from '../sr-search-result-info/sr-search-result-info.module';
import { NgbDropdownModule } from '@ng-bootstrap/ng-bootstrap';
import { BeneficiaryAccountComponent } from './components/beneficiary-account/beneficiary-account';
import { BeneficiaryContactComponent } from './components/beneficiary-contact/beneficiary-contact';
import { BeneficiarySelectorUiComponent } from './beneficiary-selector-ui.component';
import { BeneficiaryTabsComponent } from './components/beneficiary-tabs/beneficiary-tabs';
import { ReactiveFormsModule } from '@angular/forms';
import { ScrollingModule } from '@angular/cdk/scrolling';
import { BeneficiaryScrollDirective, BeneficiaryScrollItemDirective, } from './directives/beneficiary-scroll.directive';
import { BbKeyboardNavigationModule } from '../../directives/bb-keyboard-navigation.module';
import { AmountModule } from '@backbase/ui-ang/amount';
import { ButtonModule } from '@backbase/ui-ang/button';
import { DropdownMenuModule } from '@backbase/ui-ang/dropdown-menu';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { FocusModule } from '@backbase/ui-ang/focus';
import { IconModule } from '@backbase/ui-ang/icon';
import { KeyboardClickModule } from '@backbase/ui-ang/keyboard-click-directive';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { SearchBoxModule } from '@backbase/ui-ang/search-box';
import { HighlightModule } from '@backbase/ui-ang/highlight';
import * as i0 from "@angular/core";
const modules = [
    NgbDropdownModule,
    IconModule,
    SearchBoxModule,
    EmptyStateModule,
    HighlightModule,
    DropdownMenuModule,
    ButtonModule,
    LoadingIndicatorModule,
    KeyboardClickModule,
    AmountModule,
    PaymentsSrSearchResultInfoModule,
    ScrollingModule,
    BbKeyboardNavigationModule,
];
const components = [
    BeneficiarySelectorUiComponent,
    BeneficiaryTabsComponent,
    BeneficiaryContactComponent,
    BeneficiaryAccountComponent,
    BeneficiaryScrollDirective,
    BeneficiaryScrollItemDirective,
];
export class BeneficiarySelectorUIModule {
}
/** @nocollapse */ BeneficiarySelectorUIModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BeneficiarySelectorUIModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ BeneficiarySelectorUIModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BeneficiarySelectorUIModule, declarations: [BeneficiarySelectorUiComponent,
        BeneficiaryTabsComponent,
        BeneficiaryContactComponent,
        BeneficiaryAccountComponent,
        BeneficiaryScrollDirective,
        BeneficiaryScrollItemDirective], imports: [CommonModule, ReactiveFormsModule, NgbDropdownModule,
        IconModule,
        SearchBoxModule,
        EmptyStateModule,
        HighlightModule,
        DropdownMenuModule,
        ButtonModule,
        LoadingIndicatorModule,
        KeyboardClickModule,
        AmountModule,
        PaymentsSrSearchResultInfoModule,
        ScrollingModule,
        BbKeyboardNavigationModule, FocusModule], exports: [BeneficiarySelectorUiComponent, NgbDropdownModule,
        IconModule,
        SearchBoxModule,
        EmptyStateModule,
        HighlightModule,
        DropdownMenuModule,
        ButtonModule,
        LoadingIndicatorModule,
        KeyboardClickModule,
        AmountModule,
        PaymentsSrSearchResultInfoModule,
        ScrollingModule,
        BbKeyboardNavigationModule] });
/** @nocollapse */ BeneficiarySelectorUIModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BeneficiarySelectorUIModule, imports: [[CommonModule, ReactiveFormsModule, ...modules, FocusModule], NgbDropdownModule,
        IconModule,
        SearchBoxModule,
        EmptyStateModule,
        HighlightModule,
        DropdownMenuModule,
        ButtonModule,
        LoadingIndicatorModule,
        KeyboardClickModule,
        AmountModule,
        PaymentsSrSearchResultInfoModule,
        ScrollingModule,
        BbKeyboardNavigationModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BeneficiarySelectorUIModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components],
                    imports: [CommonModule, ReactiveFormsModule, ...modules, FocusModule],
                    exports: [BeneficiarySelectorUiComponent, ...modules],
                }]
        }] });
