// @backbase/portfolio-trading-http-ang/esm2015/api/tradingMarkets.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const TradingMarketsServiceGetMarketMocksProvider = createMocks([{
        urlPattern: "/api/v1/client-api/v1/trading/markets/{id}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "faa4c056-357b-11ec-8d3d-0242ac130003",
                    "name": "NASDAQ",
                    "status": "Open",
                    "iconUrl": "/client-api/v1/trading/markets/faa4c056-357b-11ec-8d3d-0242ac130003/icons"
                }
            },
            {
                status: 200,
                body: {
                    "name": "name",
                    "id": "id",
                    "iconUrl": "iconUrl"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const TradingMarketsServiceGetMarketsMocksProvider = createMocks([{
        urlPattern: "/api/v1/client-api/v1/trading/markets",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "faa4c056-357b-11ec-8d3d-0242ac130003",
                        "name": "NASDAQ",
                        "status": "Open",
                        "iconUrl": "/client-api/v1/trading/markets/faa4c056-357b-11ec-8d3d-0242ac130003/icons"
                    }, {
                        "id": "94b7c6fc-357c-11ec-8d3d-0242ac130003",
                        "name": "NYSE",
                        "status": "Close",
                        "iconUrl": "/client-api/v1/trading/markets/94b7c6fc-357c-11ec-8d3d-0242ac130003/icons"
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const TradingMarketsServiceGutMarketIconMocksProvider = createMocks([{
        urlPattern: "/api/v1/client-api/v1/trading/markets/{id}/icons",
        method: "GET",
        responses: [
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const TradingMarketsServiceMocksProvider = createMocks([
    {
        urlPattern: "/api/v1/client-api/v1/trading/markets/{id}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "faa4c056-357b-11ec-8d3d-0242ac130003",
                    "name": "NASDAQ",
                    "status": "Open",
                    "iconUrl": "/client-api/v1/trading/markets/faa4c056-357b-11ec-8d3d-0242ac130003/icons"
                }
            },
            {
                status: 200,
                body: {
                    "name": "name",
                    "id": "id",
                    "iconUrl": "iconUrl"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/api/v1/client-api/v1/trading/markets",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "faa4c056-357b-11ec-8d3d-0242ac130003",
                        "name": "NASDAQ",
                        "status": "Open",
                        "iconUrl": "/client-api/v1/trading/markets/faa4c056-357b-11ec-8d3d-0242ac130003/icons"
                    }, {
                        "id": "94b7c6fc-357c-11ec-8d3d-0242ac130003",
                        "name": "NYSE",
                        "status": "Close",
                        "iconUrl": "/client-api/v1/trading/markets/94b7c6fc-357c-11ec-8d3d-0242ac130003/icons"
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/api/v1/client-api/v1/trading/markets/{id}/icons",
        method: "GET",
        responses: [
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
]);
