// @backbase/portfolio-trading-http-ang/esm2015/api/tradingInstruments.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const TradingInstrumentsServiceAddToFavoritesMocksProvider = createMocks([{
        urlPattern: "/api/v1/client-api/v1/trading/instruments/{id}/favorite",
        method: "POST",
        responses: [
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const TradingInstrumentsServiceGetInstrumentMocksProvider = createMocks([{
        urlPattern: "/api/v1/client-api/v1/trading/instruments/{id}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "68bbeace-423e-11ec-9621-0242ac130002",
                    "name": "Apple",
                    "favorite": true,
                    "country": "USA",
                    "ticker": "AAPL",
                    "exchange": "NASDAQ",
                    "assetClassName": "Equity",
                    "subAssetClassName": "Tech Sector",
                    "code": "US0378331005",
                    "price": {
                        "value": 1247.15,
                        "currency": "USD"
                    },
                    "buyPrice": {
                        "value": 1247.15,
                        "currency": "USD"
                    },
                    "sellPrice": {
                        "value": 1249.15,
                        "currency": "USD"
                    },
                    "pricePerformance": 7.81,
                    "pricePerformanceAbs": {
                        "amount": 1.48,
                        "currencyCode": "USD"
                    },
                    "canBuy": true,
                    "canSell": false,
                    "about": {
                        "about": "about",
                        "address": "address",
                        "phoneNumber": "phoneNumber",
                        "link": "link"
                    },
                    "holdings": {
                        "totalHoldings": {
                            "quantity": 100.8,
                            "valuation": {
                                "amount": 1245.45,
                                "currency": "USD"
                            },
                            "buyPrice": {
                                "amount": 4245.45,
                                "currency": "USD"
                            },
                            "transactionsNumber": 111,
                            "unrealizedPL": {
                                "amount": 245.45,
                                "currency": "USD"
                            }
                        },
                        "holdingsPerPortfolio": [{
                                "portfolio": {
                                    "id": "68bbeace-274e-11ec-9621-0242ac130002",
                                    "name": "Trading portfolio",
                                    "iban": "NL79RABO5373380466",
                                    "alias": "My portfolio to trade"
                                },
                                "holdings": {
                                    "quantity": 40.8,
                                    "valuation": {
                                        "amount": 245.0,
                                        "currency": "USD"
                                    },
                                    "transactionsNumber": 89,
                                    "unrealizedPL": {
                                        "amount": 245.45,
                                        "currency": "USD"
                                    }
                                }
                            }, {
                                "portfolio": {
                                    "id": "72ba1654-274e-11ec-9621-0242ac130002",
                                    "name": "Trading portfolio 2",
                                    "iban": "NL79RABO5373380477",
                                    "alias": "My second portfolio to trade"
                                },
                                "holdings": {
                                    "quantity": 60.8,
                                    "valuation": {
                                        "amount": 1000.45,
                                        "currency": "USD"
                                    },
                                    "transactionsNumber": 22,
                                    "unrealizedPL": {
                                        "amount": 245.45,
                                        "currency": "USD"
                                    }
                                }
                            }]
                    },
                    "keyStatistics": {
                        "exchangeRate": 1.21,
                        "changeInNetAssetValue": 7.81,
                        "fundStatus": "open-ended fund",
                        "totalAssets": 134.3,
                        "trailingTwelveMonthsYield": 1.4,
                        "oneYearReturn": 68.97,
                        "effectiveDuration": 4.51,
                        "portfolioAssetsAllocation": "Equity",
                        "turnover": 30,
                        "creditRating": "AAA",
                        "priceToEarningsRatio": 37.59,
                        "estPriceToEarningsRatio": 31.06,
                        "priceToEarningsToGrowthRatio": 3.09,
                        "sharesOutstanding": 17.002,
                        "priceToBookRatio": 32.07,
                        "earningsPerShare": 1.39,
                        "dividendYield": 0.66,
                        "lastDividendReported": 0.2,
                        "nextEarningsAnnouncement": "2021-01-26",
                        "coupon": 5.63,
                        "maturity": "2028-01-04",
                        "yieldToMaturity": 0,
                        "duration": 7.63,
                        "couponFrequency": "annually",
                        "unrealizedPLPct": 4.14,
                        "unrealizedPL": 4.14,
                        "accruedInterestOnBond": 4.14,
                        "additionalKeyStatistics": {
                            "purchasePrice": "118.34",
                            "totalAssets": "50",
                            "coupon": "false"
                        }
                    },
                    "additions": {
                        "someKey": "someValue"
                    },
                    "iconUrl": "/client-api/v1/trading/instruments/68bbeace-423e-11ec-9621-0242ac130002/icons"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const TradingInstrumentsServiceGetInstrumentHistoryPricesMocksProvider = createMocks([{
        urlPattern: "/api/v1/client-api/v1/trading/instruments/{id}/history-prices",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "chartData": [{
                            "date": "2020-01-01",
                            "price": {
                                "amount": 125.01,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2020-02-01",
                            "price": {
                                "amount": 129.04,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2020-03-01",
                            "price": {
                                "amount": 134.18,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2020-04-01",
                            "price": {
                                "amount": 131.4,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2020-05-01",
                            "price": {
                                "amount": 120.88,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2020-06-01",
                            "price": {
                                "amount": 112.82,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2020-07-01",
                            "price": {
                                "amount": 115.54,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2020-08-01",
                            "price": {
                                "amount": 110.34,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2020-09-01",
                            "price": {
                                "amount": 106.84,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2020-10-01",
                            "price": {
                                "amount": 108.22,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2020-11-01",
                            "price": {
                                "amount": 116.5,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2020-12-01",
                            "price": {
                                "amount": 114.97,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2021-01-01",
                            "price": {
                                "amount": 124.4,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2021-02-01",
                            "price": {
                                "amount": 120.71,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2021-03-01",
                            "price": {
                                "amount": 116.6,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2021-04-01",
                            "price": {
                                "amount": 111.2,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2021-05-01",
                            "price": {
                                "amount": 115.32,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2021-06-01",
                            "price": {
                                "amount": 108.77,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2021-07-01",
                            "price": {
                                "amount": 114.95,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2021-08-01",
                            "price": {
                                "amount": 119.03,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2021-09-01",
                            "price": {
                                "amount": 118.69,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2021-10-01",
                            "price": {
                                "amount": 116.32,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2021-11-01",
                            "price": {
                                "amount": 115.97,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2021-12-01",
                            "price": {
                                "amount": 119.49,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2022-01-01",
                            "price": {
                                "amount": 118.64,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2022-02-01",
                            "price": {
                                "amount": 113.85,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2022-03-01",
                            "price": {
                                "amount": 115.17,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2022-04-01",
                            "price": {
                                "amount": 116.03,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2022-05-01",
                            "price": {
                                "amount": 116.59,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2022-06-01",
                            "price": {
                                "amount": 119.05,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2022-07-01",
                            "price": {
                                "amount": 122.72,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2022-08-01",
                            "price": {
                                "amount": 123.08,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2022-09-01",
                            "price": {
                                "amount": 122.94,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2022-10-01",
                            "price": {
                                "amount": 122.25,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2022-11-01",
                            "price": {
                                "amount": 123.75,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2022-12-01",
                            "price": {
                                "amount": 124.3,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const TradingInstrumentsServiceGetInstrumentsMocksProvider = createMocks([{
        urlPattern: "/api/v1/client-api/v1/trading/instruments",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "instruments": [{
                            "id": "117c6345-0fee-11ec-82a8-0242ac130003",
                            "name": "Apple",
                            "ticker": "AAPL",
                            "exchange": "NASDAQ",
                            "price": {
                                "value": 147.15,
                                "currency": "USD"
                            },
                            "pricePerformance": 7.81,
                            "pricePerformanceAbs": {
                                "amount": 1.48,
                                "currencyCode": "USD"
                            },
                            "iconUrl": "/client-api/v1/trading/instruments/117c6345-0fee-11ec-82a8-0242ac130003/icons"
                        }, {
                            "id": "17bbeace-274e-22ec-9621-0242ac130003",
                            "name": "Tesla",
                            "instrumentNameLong": "Tesla",
                            "ticker": "TSLA",
                            "exchange": "NASDAQ",
                            "price": {
                                "value": 127.15,
                                "currency": "USD"
                            },
                            "pricePerformance": 6.81,
                            "pricePerformanceAbs": {
                                "amount": 1.32,
                                "currencyCode": "USD"
                            },
                            "iconUrl": "/client-api/v1/trading/instruments/17bbeace-274e-22ec-9621-0242ac130003/icons"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const TradingInstrumentsServiceGetPortfoliosByInstrumentMocksProvider = createMocks([{
        urlPattern: "/api/v1/client-api/v1/trading/instruments/{id}/portfolios",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "portfolios": [{
                            "name": "Trading portfolio",
                            "alias": "My portfolio to trade",
                            "portfolioId": "68bbeace-274e-11ec-9621-0242ac130002",
                            "availableBalance": {
                                "value": 5068.3,
                                "currency": "USD"
                            },
                            "iban": "NL79RABO5373380466",
                            "canSell": true
                        }, {
                            "name": "Trading portfolio 2",
                            "alias": "My second portfolio to trade",
                            "portfolioId": "72ba1654-274e-11ec-9621-0242ac130002",
                            "availableBalance": {
                                "value": 6174.9,
                                "currency": "USD"
                            },
                            "iban": "NL79RABO5373380477",
                            "canSell": false
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const TradingInstrumentsServiceGutInstrumentIconMocksProvider = createMocks([{
        urlPattern: "/api/v1/client-api/v1/trading/instruments/{id}/icons",
        method: "GET",
        responses: [
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const TradingInstrumentsServiceRemoveFromFavoritesMocksProvider = createMocks([{
        urlPattern: "/api/v1/client-api/v1/trading/instruments/{id}/favorite",
        method: "DELETE",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const TradingInstrumentsServiceMocksProvider = createMocks([
    {
        urlPattern: "/api/v1/client-api/v1/trading/instruments/{id}/favorite",
        method: "POST",
        responses: [
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/api/v1/client-api/v1/trading/instruments/{id}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "68bbeace-423e-11ec-9621-0242ac130002",
                    "name": "Apple",
                    "favorite": true,
                    "country": "USA",
                    "ticker": "AAPL",
                    "exchange": "NASDAQ",
                    "assetClassName": "Equity",
                    "subAssetClassName": "Tech Sector",
                    "code": "US0378331005",
                    "price": {
                        "value": 1247.15,
                        "currency": "USD"
                    },
                    "buyPrice": {
                        "value": 1247.15,
                        "currency": "USD"
                    },
                    "sellPrice": {
                        "value": 1249.15,
                        "currency": "USD"
                    },
                    "pricePerformance": 7.81,
                    "pricePerformanceAbs": {
                        "amount": 1.48,
                        "currencyCode": "USD"
                    },
                    "canBuy": true,
                    "canSell": false,
                    "about": {
                        "about": "about",
                        "address": "address",
                        "phoneNumber": "phoneNumber",
                        "link": "link"
                    },
                    "holdings": {
                        "totalHoldings": {
                            "quantity": 100.8,
                            "valuation": {
                                "amount": 1245.45,
                                "currency": "USD"
                            },
                            "buyPrice": {
                                "amount": 4245.45,
                                "currency": "USD"
                            },
                            "transactionsNumber": 111,
                            "unrealizedPL": {
                                "amount": 245.45,
                                "currency": "USD"
                            }
                        },
                        "holdingsPerPortfolio": [{
                                "portfolio": {
                                    "id": "68bbeace-274e-11ec-9621-0242ac130002",
                                    "name": "Trading portfolio",
                                    "iban": "NL79RABO5373380466",
                                    "alias": "My portfolio to trade"
                                },
                                "holdings": {
                                    "quantity": 40.8,
                                    "valuation": {
                                        "amount": 245.0,
                                        "currency": "USD"
                                    },
                                    "transactionsNumber": 89,
                                    "unrealizedPL": {
                                        "amount": 245.45,
                                        "currency": "USD"
                                    }
                                }
                            }, {
                                "portfolio": {
                                    "id": "72ba1654-274e-11ec-9621-0242ac130002",
                                    "name": "Trading portfolio 2",
                                    "iban": "NL79RABO5373380477",
                                    "alias": "My second portfolio to trade"
                                },
                                "holdings": {
                                    "quantity": 60.8,
                                    "valuation": {
                                        "amount": 1000.45,
                                        "currency": "USD"
                                    },
                                    "transactionsNumber": 22,
                                    "unrealizedPL": {
                                        "amount": 245.45,
                                        "currency": "USD"
                                    }
                                }
                            }]
                    },
                    "keyStatistics": {
                        "exchangeRate": 1.21,
                        "changeInNetAssetValue": 7.81,
                        "fundStatus": "open-ended fund",
                        "totalAssets": 134.3,
                        "trailingTwelveMonthsYield": 1.4,
                        "oneYearReturn": 68.97,
                        "effectiveDuration": 4.51,
                        "portfolioAssetsAllocation": "Equity",
                        "turnover": 30,
                        "creditRating": "AAA",
                        "priceToEarningsRatio": 37.59,
                        "estPriceToEarningsRatio": 31.06,
                        "priceToEarningsToGrowthRatio": 3.09,
                        "sharesOutstanding": 17.002,
                        "priceToBookRatio": 32.07,
                        "earningsPerShare": 1.39,
                        "dividendYield": 0.66,
                        "lastDividendReported": 0.2,
                        "nextEarningsAnnouncement": "2021-01-26",
                        "coupon": 5.63,
                        "maturity": "2028-01-04",
                        "yieldToMaturity": 0,
                        "duration": 7.63,
                        "couponFrequency": "annually",
                        "unrealizedPLPct": 4.14,
                        "unrealizedPL": 4.14,
                        "accruedInterestOnBond": 4.14,
                        "additionalKeyStatistics": {
                            "purchasePrice": "118.34",
                            "totalAssets": "50",
                            "coupon": "false"
                        }
                    },
                    "additions": {
                        "someKey": "someValue"
                    },
                    "iconUrl": "/client-api/v1/trading/instruments/68bbeace-423e-11ec-9621-0242ac130002/icons"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/api/v1/client-api/v1/trading/instruments/{id}/history-prices",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "chartData": [{
                            "date": "2020-01-01",
                            "price": {
                                "amount": 125.01,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2020-02-01",
                            "price": {
                                "amount": 129.04,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2020-03-01",
                            "price": {
                                "amount": 134.18,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2020-04-01",
                            "price": {
                                "amount": 131.4,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2020-05-01",
                            "price": {
                                "amount": 120.88,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2020-06-01",
                            "price": {
                                "amount": 112.82,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2020-07-01",
                            "price": {
                                "amount": 115.54,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2020-08-01",
                            "price": {
                                "amount": 110.34,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2020-09-01",
                            "price": {
                                "amount": 106.84,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2020-10-01",
                            "price": {
                                "amount": 108.22,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2020-11-01",
                            "price": {
                                "amount": 116.5,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2020-12-01",
                            "price": {
                                "amount": 114.97,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2021-01-01",
                            "price": {
                                "amount": 124.4,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2021-02-01",
                            "price": {
                                "amount": 120.71,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2021-03-01",
                            "price": {
                                "amount": 116.6,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2021-04-01",
                            "price": {
                                "amount": 111.2,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2021-05-01",
                            "price": {
                                "amount": 115.32,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2021-06-01",
                            "price": {
                                "amount": 108.77,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2021-07-01",
                            "price": {
                                "amount": 114.95,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2021-08-01",
                            "price": {
                                "amount": 119.03,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2021-09-01",
                            "price": {
                                "amount": 118.69,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2021-10-01",
                            "price": {
                                "amount": 116.32,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2021-11-01",
                            "price": {
                                "amount": 115.97,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2021-12-01",
                            "price": {
                                "amount": 119.49,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2022-01-01",
                            "price": {
                                "amount": 118.64,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2022-02-01",
                            "price": {
                                "amount": 113.85,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2022-03-01",
                            "price": {
                                "amount": 115.17,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2022-04-01",
                            "price": {
                                "amount": 116.03,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2022-05-01",
                            "price": {
                                "amount": 116.59,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2022-06-01",
                            "price": {
                                "amount": 119.05,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2022-07-01",
                            "price": {
                                "amount": 122.72,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2022-08-01",
                            "price": {
                                "amount": 123.08,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2022-09-01",
                            "price": {
                                "amount": 122.94,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2022-10-01",
                            "price": {
                                "amount": 122.25,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2022-11-01",
                            "price": {
                                "amount": 123.75,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }, {
                            "date": "2022-12-01",
                            "price": {
                                "amount": 124.3,
                                "currency": "USD"
                            },
                            "type": "Closing"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/api/v1/client-api/v1/trading/instruments",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "instruments": [{
                            "id": "117c6345-0fee-11ec-82a8-0242ac130003",
                            "name": "Apple",
                            "ticker": "AAPL",
                            "exchange": "NASDAQ",
                            "price": {
                                "value": 147.15,
                                "currency": "USD"
                            },
                            "pricePerformance": 7.81,
                            "pricePerformanceAbs": {
                                "amount": 1.48,
                                "currencyCode": "USD"
                            },
                            "iconUrl": "/client-api/v1/trading/instruments/117c6345-0fee-11ec-82a8-0242ac130003/icons"
                        }, {
                            "id": "17bbeace-274e-22ec-9621-0242ac130003",
                            "name": "Tesla",
                            "instrumentNameLong": "Tesla",
                            "ticker": "TSLA",
                            "exchange": "NASDAQ",
                            "price": {
                                "value": 127.15,
                                "currency": "USD"
                            },
                            "pricePerformance": 6.81,
                            "pricePerformanceAbs": {
                                "amount": 1.32,
                                "currencyCode": "USD"
                            },
                            "iconUrl": "/client-api/v1/trading/instruments/17bbeace-274e-22ec-9621-0242ac130003/icons"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/api/v1/client-api/v1/trading/instruments/{id}/portfolios",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "portfolios": [{
                            "name": "Trading portfolio",
                            "alias": "My portfolio to trade",
                            "portfolioId": "68bbeace-274e-11ec-9621-0242ac130002",
                            "availableBalance": {
                                "value": 5068.3,
                                "currency": "USD"
                            },
                            "iban": "NL79RABO5373380466",
                            "canSell": true
                        }, {
                            "name": "Trading portfolio 2",
                            "alias": "My second portfolio to trade",
                            "portfolioId": "72ba1654-274e-11ec-9621-0242ac130002",
                            "availableBalance": {
                                "value": 6174.9,
                                "currency": "USD"
                            },
                            "iban": "NL79RABO5373380477",
                            "canSell": false
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/api/v1/client-api/v1/trading/instruments/{id}/icons",
        method: "GET",
        responses: [
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/api/v1/client-api/v1/trading/instruments/{id}/favorite",
        method: "DELETE",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
]);
