// @backbase/portfolio-trading-http-ang/esm2015/api/tradeOrders.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const TradeOrdersServiceCancelOrderMocksProvider = createMocks([{
        urlPattern: "/api/v1/client-api/v1/trading/orders/{id}/cancel",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "id": "55bbeace-274e-22ec-5487-0242ac130003",
                    "status": "Pending Cancel"
                }
            },
            {
                status: 200,
                body: {
                    "id": "id",
                    "rejectionReason": "rejectionReason"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const TradeOrdersServiceGetInstrumentOrderListMocksProvider = createMocks([{
        urlPattern: "/api/v1/client-api/v1/trading/instruments/{id}/orders",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "cursor": "55bbeace-274e-22ec-5487-0242ac130001",
                    "hasMoreItems": true,
                    "orders": [{
                            "id": "55bbeace-274e-22ec-5487-0242ac130003",
                            "date": "2021-09-07T12:00:00.000Z",
                            "type": "Market Order",
                            "tradeDirection": "Sell",
                            "status": "Rejected",
                            "instrument": {
                                "id": "68bbeace-423e-11ec-9621-0242ac130002",
                                "name": "Apple",
                                "ticker": "AAPL",
                                "exchange": "NASDAQ",
                                "price": {
                                    "value": 1247.15,
                                    "currency": "USD"
                                },
                                "iconUrl": "/client-api/v1/trading/instruments/68bbeace-423e-11ec-9621-0242ac130002/icons"
                            },
                            "quantity": 10.5,
                            "totalValue": {
                                "value": 13160.6,
                                "currency": "USD"
                            }
                        }, {
                            "id": "25bbeace-274e-22ec-5487-0242ac130003",
                            "date": "2021-09-07T12:00:00.000Z",
                            "type": "Market Order",
                            "tradeDirection": "Buy",
                            "status": "Filled",
                            "instrument": {
                                "id": "68bbeace-423e-11ec-9621-0242ac130002",
                                "name": "Apple",
                                "ticker": "AAPL",
                                "exchange": "NASDAQ",
                                "price": {
                                    "value": 1247.15,
                                    "currency": "USD"
                                },
                                "iconUrl": "/client-api/v1/trading/instruments/68bbeace-423e-11ec-9621-0242ac130002/icons"
                            },
                            "totalValue": {
                                "value": 13095.1,
                                "currency": "USD"
                            },
                            "quantity": 10.5
                        }, {
                            "id": "55bbeace-274e-22ec-5487-0242ac130004",
                            "date": "2022-02-15T12:00:00.000Z",
                            "type": "Limit Order",
                            "tradeDirection": "Buy",
                            "status": "Partially Filled",
                            "instrument": {
                                "id": "68bbeace-423e-11ec-9621-0242ac130002",
                                "name": "Apple",
                                "ticker": "AAPL",
                                "exchange": "NASDAQ",
                                "price": {
                                    "value": 1247.15,
                                    "currency": "USD"
                                },
                                "iconUrl": "/client-api/v1/trading/instruments/68bbeace-423e-11ec-9621-0242ac130002/icons"
                            },
                            "totalValue": {
                                "value": 13095.1,
                                "currency": "USD"
                            },
                            "quantity": 10
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const TradeOrdersServiceGetOrderMocksProvider = createMocks([{
        urlPattern: "/api/v1/client-api/v1/trading/orders/{id}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "55bbeace-274e-22ec-5487-0242ac130003",
                    "date": "2021-09-07T12:00:00.000Z",
                    "type": "Market Order",
                    "tradeDirection": "Buy",
                    "status": "Filled",
                    "instrument": {
                        "id": "24aaeace-274e-22ec-5487-0242ac130003",
                        "name": "Apple",
                        "ticker": "AAPL",
                        "exchange": "NASDAQ",
                        "price": {
                            "value": 1247.15,
                            "currency": "USD"
                        },
                        "iconUrl": "/client-api/v1/trading/instruments/24aaeace-274e-22ec-5487-0242ac130003/icons"
                    },
                    "portfolio": {
                        "id": "66bbeace-274e-55ec-5487-0242ac130008",
                        "name": "Trading portfolio 2",
                        "iban": "NL79RABO5373380477",
                        "alias": "My second portfolio to trade"
                    },
                    "value": {
                        "value": 13095.1,
                        "currency": "USD"
                    },
                    "quantity": 10.5,
                    "totalValue": {
                        "value": 13160.6,
                        "currency": "USD"
                    },
                    "commissionFee": {
                        "value": 65.5,
                        "currency": "USD"
                    },
                    "transactionId": "817c6362-0fee-11ec-82a8-0242ac130003"
                }
            },
            {
                status: 200,
                body: {
                    "id": "55bbeace-274e-22ec-5487-0242ac130003",
                    "date": "2021-09-07T12:00:00.000Z",
                    "type": "Market Order",
                    "tradeDirection": "Buy",
                    "status": "Rejected",
                    "instrument": {
                        "id": "68bbeace-423e-11ec-9621-0242ac130002",
                        "name": "Apple",
                        "ticker": "AAPL",
                        "exchange": "NASDAQ",
                        "price": {
                            "value": 1247.15,
                            "currency": "USD"
                        },
                        "iconUrl": "/client-api/v1/trading/instruments/68bbeace-423e-11ec-9621-0242ac130002/icons"
                    },
                    "portfolio": {
                        "id": "72ba1654-274e-11ec-9621-0242ac130002",
                        "name": "Trading portfolio 2",
                        "iban": "NL79RABO5373380477",
                        "alias": "My second portfolio to trade"
                    },
                    "value": {
                        "value": 13095.1,
                        "currency": "USD"
                    },
                    "quantity": 10.5,
                    "totalValue": {
                        "value": 13160.6,
                        "currency": "USD"
                    },
                    "commissionFee": {
                        "value": 65.5,
                        "currency": "USD"
                    },
                    "rejectedReason": "some reason"
                }
            },
            {
                status: 200,
                body: {
                    "id": "55bbeace-274e-22ec-5487-0242ac130004",
                    "date": "2021-02-15T12:00:00.000Z",
                    "type": "Limit Order",
                    "tradeDirection": "Buy",
                    "status": "Partially Filled",
                    "instrument": {
                        "id": "24aaeace-274e-22ec-5487-0242ac130003",
                        "name": "Apple",
                        "ticker": "AAPL",
                        "exchange": "NASDAQ",
                        "price": {
                            "value": 1247.15,
                            "currency": "USD"
                        },
                        "iconUrl": "/client-api/v1/trading/instruments/24aaeace-274e-22ec-5487-0242ac130003/icons"
                    },
                    "portfolio": {
                        "id": "66bbeace-274e-55ec-5487-0242ac130008",
                        "name": "Trading portfolio 2",
                        "iban": "NL79RABO5373380477",
                        "alias": "My second portfolio to trade"
                    },
                    "value": {
                        "value": 13095.1,
                        "currency": "USD"
                    },
                    "quantity": 10.5,
                    "totalValue": {
                        "value": 13160.6,
                        "currency": "USD"
                    },
                    "commissionFee": {
                        "value": 65.5,
                        "currency": "USD"
                    },
                    "partiallyQuantity": 5,
                    "limitPrice": {
                        "value": 1015.0,
                        "currency": "USD"
                    },
                    "partiallyOrderDetails": [{
                            "quantity": 2,
                            "value": {
                                "value": 2201.16,
                                "currency": "USD"
                            },
                            "totalValue": {
                                "value": 2211.56,
                                "currency": "USD"
                            },
                            "commissionFee": {
                                "value": 10.5,
                                "currency": "USD"
                            },
                            "instrumentPrice": {
                                "value": 1010.58,
                                "currency": "USD"
                            },
                            "date": "2022-02-14T12:00:00.000Z"
                        }, {
                            "quantity": 3,
                            "value": {
                                "value": 3037.74,
                                "currency": "USD"
                            },
                            "totalValue": {
                                "value": 3050.24,
                                "currency": "USD"
                            },
                            "commissionFee": {
                                "value": 12.5,
                                "currency": "USD"
                            },
                            "instrumentPrice": {
                                "value": 1012.58,
                                "currency": "USD"
                            },
                            "date": "2022-02-14T15:00:00.000Z"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const TradeOrdersServiceGetOrderListMocksProvider = createMocks([{
        urlPattern: "/api/v1/client-api/v1/trading/orders",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "cursor": "55bbeace-274e-22ec-5487-0242ac130001",
                    "hasMoreItems": true,
                    "orders": [{
                            "id": "55bbeace-274e-22ec-5487-0242ac130003",
                            "date": "2021-09-07T12:00:00.000Z",
                            "type": "Market Order",
                            "tradeDirection": "Sell",
                            "status": "Rejected",
                            "instrument": {
                                "id": "68bbeace-423e-11ec-9621-0242ac130002",
                                "name": "Apple",
                                "ticker": "AAPL",
                                "exchange": "NASDAQ",
                                "price": {
                                    "value": 1247.15,
                                    "currency": "USD"
                                },
                                "iconUrl": "/client-api/v1/trading/instruments/68bbeace-423e-11ec-9621-0242ac130002/icons"
                            },
                            "quantity": 10.5,
                            "totalValue": {
                                "value": 13160.6,
                                "currency": "USD"
                            }
                        }, {
                            "id": "25bbeace-274e-22ec-5487-0242ac130003",
                            "date": "2021-09-07T12:00:00.000Z",
                            "type": "Market Order",
                            "tradeDirection": "Buy",
                            "status": "Filled",
                            "instrument": {
                                "id": "68bbeace-423e-11ec-9621-0242ac130002",
                                "name": "Apple",
                                "ticker": "AAPL",
                                "exchange": "NASDAQ",
                                "price": {
                                    "value": 1247.15,
                                    "currency": "USD"
                                },
                                "iconUrl": "/client-api/v1/trading/instruments/68bbeace-423e-11ec-9621-0242ac130002/icons"
                            },
                            "totalValue": {
                                "value": 13095.1,
                                "currency": "USD"
                            },
                            "quantity": 10.5
                        }, {
                            "id": "55bbeace-274e-22ec-5487-0242ac130004",
                            "date": "2022-02-15T12:00:00.000Z",
                            "type": "Limit Order",
                            "tradeDirection": "Buy",
                            "status": "Partially Filled",
                            "instrument": {
                                "id": "68bbeace-423e-11ec-9621-0242ac130002",
                                "name": "Apple",
                                "ticker": "AAPL",
                                "exchange": "NASDAQ",
                                "price": {
                                    "value": 1247.15,
                                    "currency": "USD"
                                },
                                "iconUrl": "/client-api/v1/trading/instruments/68bbeace-423e-11ec-9621-0242ac130002/icons"
                            },
                            "totalValue": {
                                "value": 13095.1,
                                "currency": "USD"
                            },
                            "quantity": 10
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "cursor": "cursor",
                    "orders": [{
                            "date": "2000-01-23T04:56:07.000+00:00",
                            "totalValue": {
                                "amount": 0.8008281904610115,
                                "currency": "currency"
                            },
                            "quantity": 0.8008281904610115,
                            "instrument": {
                                "ticker": "ticker",
                                "price": {
                                    "amount": 0.8008281904610115,
                                    "currency": "currency"
                                },
                                "name": "name",
                                "exchange": "exchange",
                                "id": "id",
                                "iconUrl": "iconUrl"
                            },
                            "id": "id"
                        }, {
                            "date": "2000-01-23T04:56:07.000+00:00",
                            "totalValue": {
                                "amount": 0.8008281904610115,
                                "currency": "currency"
                            },
                            "quantity": 0.8008281904610115,
                            "instrument": {
                                "ticker": "ticker",
                                "price": {
                                    "amount": 0.8008281904610115,
                                    "currency": "currency"
                                },
                                "name": "name",
                                "exchange": "exchange",
                                "id": "id",
                                "iconUrl": "iconUrl"
                            },
                            "id": "id"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const TradeOrdersServiceInitOrderMocksProvider = createMocks([{
        urlPattern: "/api/v1/client-api/v1/trading/orders",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "id": "55bbeace-274e-22ec-5487-0242ac130003",
                    "type": "Market Order",
                    "tradeDirection": "Buy",
                    "status": "Draft",
                    "instrument": {
                        "id": "24aaeace-274e-22ec-5487-0242ac130003",
                        "name": "Apple",
                        "ticker": "AAPL",
                        "exchange": "NASDAQ",
                        "price": {
                            "value": 1247.15,
                            "currency": "USD"
                        },
                        "iconUrl": "/client-api/v1/trading/instruments/68bbeace-423e-11ec-9621-0242ac130002/icons"
                    },
                    "portfolio": {
                        "id": "66bbeace-274e-55ec-5487-0242ac130008",
                        "name": "Trading portfolio 2",
                        "alias": "My second portfolio to trade",
                        "availableBalance": {
                            "value": 6174.9,
                            "currency": "EUR"
                        }
                    },
                    "value": {
                        "value": 13095.1,
                        "currency": "USD"
                    },
                    "quantity": 10.5,
                    "totalValue": {
                        "value": 13160.6,
                        "currency": "USD"
                    },
                    "commissionFee": {
                        "value": 65.5,
                        "currency": "USD"
                    }
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const TradeOrdersServicePlaceOrderMocksProvider = createMocks([{
        urlPattern: "/api/v1/client-api/v1/trading/orders/{id}/place",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "id": "55bbeace-274e-22ec-5487-0242ac130003",
                    "status": "Pending"
                }
            },
            {
                status: 200,
                body: {
                    "id": "55bbeace-274e-22ec-5487-0242ac130003",
                    "status": "Rejected",
                    "rejectionReason": "rejectionReason"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const TradeOrdersServiceStatusOrderMocksProvider = createMocks([{
        urlPattern: "/api/v1/client-api/v1/trading/orders/{id}/status",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "55bbeace-274e-22ec-5487-0242ac130003",
                    "status": "Pending"
                }
            },
            {
                status: 200,
                body: {
                    "id": "55bbeace-274e-22ec-5487-0242ac130003",
                    "status": "Rejected",
                    "rejectionReason": "rejectionReason"
                }
            },
            {
                status: 200,
                body: {
                    "id": "id",
                    "rejectionReason": "rejectionReason"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const TradeOrdersServiceMocksProvider = createMocks([
    {
        urlPattern: "/api/v1/client-api/v1/trading/orders/{id}/cancel",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "id": "55bbeace-274e-22ec-5487-0242ac130003",
                    "status": "Pending Cancel"
                }
            },
            {
                status: 200,
                body: {
                    "id": "id",
                    "rejectionReason": "rejectionReason"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/api/v1/client-api/v1/trading/instruments/{id}/orders",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "cursor": "55bbeace-274e-22ec-5487-0242ac130001",
                    "hasMoreItems": true,
                    "orders": [{
                            "id": "55bbeace-274e-22ec-5487-0242ac130003",
                            "date": "2021-09-07T12:00:00.000Z",
                            "type": "Market Order",
                            "tradeDirection": "Sell",
                            "status": "Rejected",
                            "instrument": {
                                "id": "68bbeace-423e-11ec-9621-0242ac130002",
                                "name": "Apple",
                                "ticker": "AAPL",
                                "exchange": "NASDAQ",
                                "price": {
                                    "value": 1247.15,
                                    "currency": "USD"
                                },
                                "iconUrl": "/client-api/v1/trading/instruments/68bbeace-423e-11ec-9621-0242ac130002/icons"
                            },
                            "quantity": 10.5,
                            "totalValue": {
                                "value": 13160.6,
                                "currency": "USD"
                            }
                        }, {
                            "id": "25bbeace-274e-22ec-5487-0242ac130003",
                            "date": "2021-09-07T12:00:00.000Z",
                            "type": "Market Order",
                            "tradeDirection": "Buy",
                            "status": "Filled",
                            "instrument": {
                                "id": "68bbeace-423e-11ec-9621-0242ac130002",
                                "name": "Apple",
                                "ticker": "AAPL",
                                "exchange": "NASDAQ",
                                "price": {
                                    "value": 1247.15,
                                    "currency": "USD"
                                },
                                "iconUrl": "/client-api/v1/trading/instruments/68bbeace-423e-11ec-9621-0242ac130002/icons"
                            },
                            "totalValue": {
                                "value": 13095.1,
                                "currency": "USD"
                            },
                            "quantity": 10.5
                        }, {
                            "id": "55bbeace-274e-22ec-5487-0242ac130004",
                            "date": "2022-02-15T12:00:00.000Z",
                            "type": "Limit Order",
                            "tradeDirection": "Buy",
                            "status": "Partially Filled",
                            "instrument": {
                                "id": "68bbeace-423e-11ec-9621-0242ac130002",
                                "name": "Apple",
                                "ticker": "AAPL",
                                "exchange": "NASDAQ",
                                "price": {
                                    "value": 1247.15,
                                    "currency": "USD"
                                },
                                "iconUrl": "/client-api/v1/trading/instruments/68bbeace-423e-11ec-9621-0242ac130002/icons"
                            },
                            "totalValue": {
                                "value": 13095.1,
                                "currency": "USD"
                            },
                            "quantity": 10
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/api/v1/client-api/v1/trading/orders/{id}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "55bbeace-274e-22ec-5487-0242ac130003",
                    "date": "2021-09-07T12:00:00.000Z",
                    "type": "Market Order",
                    "tradeDirection": "Buy",
                    "status": "Filled",
                    "instrument": {
                        "id": "24aaeace-274e-22ec-5487-0242ac130003",
                        "name": "Apple",
                        "ticker": "AAPL",
                        "exchange": "NASDAQ",
                        "price": {
                            "value": 1247.15,
                            "currency": "USD"
                        },
                        "iconUrl": "/client-api/v1/trading/instruments/24aaeace-274e-22ec-5487-0242ac130003/icons"
                    },
                    "portfolio": {
                        "id": "66bbeace-274e-55ec-5487-0242ac130008",
                        "name": "Trading portfolio 2",
                        "iban": "NL79RABO5373380477",
                        "alias": "My second portfolio to trade"
                    },
                    "value": {
                        "value": 13095.1,
                        "currency": "USD"
                    },
                    "quantity": 10.5,
                    "totalValue": {
                        "value": 13160.6,
                        "currency": "USD"
                    },
                    "commissionFee": {
                        "value": 65.5,
                        "currency": "USD"
                    },
                    "transactionId": "817c6362-0fee-11ec-82a8-0242ac130003"
                }
            },
            {
                status: 200,
                body: {
                    "id": "55bbeace-274e-22ec-5487-0242ac130003",
                    "date": "2021-09-07T12:00:00.000Z",
                    "type": "Market Order",
                    "tradeDirection": "Buy",
                    "status": "Rejected",
                    "instrument": {
                        "id": "68bbeace-423e-11ec-9621-0242ac130002",
                        "name": "Apple",
                        "ticker": "AAPL",
                        "exchange": "NASDAQ",
                        "price": {
                            "value": 1247.15,
                            "currency": "USD"
                        },
                        "iconUrl": "/client-api/v1/trading/instruments/68bbeace-423e-11ec-9621-0242ac130002/icons"
                    },
                    "portfolio": {
                        "id": "72ba1654-274e-11ec-9621-0242ac130002",
                        "name": "Trading portfolio 2",
                        "iban": "NL79RABO5373380477",
                        "alias": "My second portfolio to trade"
                    },
                    "value": {
                        "value": 13095.1,
                        "currency": "USD"
                    },
                    "quantity": 10.5,
                    "totalValue": {
                        "value": 13160.6,
                        "currency": "USD"
                    },
                    "commissionFee": {
                        "value": 65.5,
                        "currency": "USD"
                    },
                    "rejectedReason": "some reason"
                }
            },
            {
                status: 200,
                body: {
                    "id": "55bbeace-274e-22ec-5487-0242ac130004",
                    "date": "2021-02-15T12:00:00.000Z",
                    "type": "Limit Order",
                    "tradeDirection": "Buy",
                    "status": "Partially Filled",
                    "instrument": {
                        "id": "24aaeace-274e-22ec-5487-0242ac130003",
                        "name": "Apple",
                        "ticker": "AAPL",
                        "exchange": "NASDAQ",
                        "price": {
                            "value": 1247.15,
                            "currency": "USD"
                        },
                        "iconUrl": "/client-api/v1/trading/instruments/24aaeace-274e-22ec-5487-0242ac130003/icons"
                    },
                    "portfolio": {
                        "id": "66bbeace-274e-55ec-5487-0242ac130008",
                        "name": "Trading portfolio 2",
                        "iban": "NL79RABO5373380477",
                        "alias": "My second portfolio to trade"
                    },
                    "value": {
                        "value": 13095.1,
                        "currency": "USD"
                    },
                    "quantity": 10.5,
                    "totalValue": {
                        "value": 13160.6,
                        "currency": "USD"
                    },
                    "commissionFee": {
                        "value": 65.5,
                        "currency": "USD"
                    },
                    "partiallyQuantity": 5,
                    "limitPrice": {
                        "value": 1015.0,
                        "currency": "USD"
                    },
                    "partiallyOrderDetails": [{
                            "quantity": 2,
                            "value": {
                                "value": 2201.16,
                                "currency": "USD"
                            },
                            "totalValue": {
                                "value": 2211.56,
                                "currency": "USD"
                            },
                            "commissionFee": {
                                "value": 10.5,
                                "currency": "USD"
                            },
                            "instrumentPrice": {
                                "value": 1010.58,
                                "currency": "USD"
                            },
                            "date": "2022-02-14T12:00:00.000Z"
                        }, {
                            "quantity": 3,
                            "value": {
                                "value": 3037.74,
                                "currency": "USD"
                            },
                            "totalValue": {
                                "value": 3050.24,
                                "currency": "USD"
                            },
                            "commissionFee": {
                                "value": 12.5,
                                "currency": "USD"
                            },
                            "instrumentPrice": {
                                "value": 1012.58,
                                "currency": "USD"
                            },
                            "date": "2022-02-14T15:00:00.000Z"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/api/v1/client-api/v1/trading/orders",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "cursor": "55bbeace-274e-22ec-5487-0242ac130001",
                    "hasMoreItems": true,
                    "orders": [{
                            "id": "55bbeace-274e-22ec-5487-0242ac130003",
                            "date": "2021-09-07T12:00:00.000Z",
                            "type": "Market Order",
                            "tradeDirection": "Sell",
                            "status": "Rejected",
                            "instrument": {
                                "id": "68bbeace-423e-11ec-9621-0242ac130002",
                                "name": "Apple",
                                "ticker": "AAPL",
                                "exchange": "NASDAQ",
                                "price": {
                                    "value": 1247.15,
                                    "currency": "USD"
                                },
                                "iconUrl": "/client-api/v1/trading/instruments/68bbeace-423e-11ec-9621-0242ac130002/icons"
                            },
                            "quantity": 10.5,
                            "totalValue": {
                                "value": 13160.6,
                                "currency": "USD"
                            }
                        }, {
                            "id": "25bbeace-274e-22ec-5487-0242ac130003",
                            "date": "2021-09-07T12:00:00.000Z",
                            "type": "Market Order",
                            "tradeDirection": "Buy",
                            "status": "Filled",
                            "instrument": {
                                "id": "68bbeace-423e-11ec-9621-0242ac130002",
                                "name": "Apple",
                                "ticker": "AAPL",
                                "exchange": "NASDAQ",
                                "price": {
                                    "value": 1247.15,
                                    "currency": "USD"
                                },
                                "iconUrl": "/client-api/v1/trading/instruments/68bbeace-423e-11ec-9621-0242ac130002/icons"
                            },
                            "totalValue": {
                                "value": 13095.1,
                                "currency": "USD"
                            },
                            "quantity": 10.5
                        }, {
                            "id": "55bbeace-274e-22ec-5487-0242ac130004",
                            "date": "2022-02-15T12:00:00.000Z",
                            "type": "Limit Order",
                            "tradeDirection": "Buy",
                            "status": "Partially Filled",
                            "instrument": {
                                "id": "68bbeace-423e-11ec-9621-0242ac130002",
                                "name": "Apple",
                                "ticker": "AAPL",
                                "exchange": "NASDAQ",
                                "price": {
                                    "value": 1247.15,
                                    "currency": "USD"
                                },
                                "iconUrl": "/client-api/v1/trading/instruments/68bbeace-423e-11ec-9621-0242ac130002/icons"
                            },
                            "totalValue": {
                                "value": 13095.1,
                                "currency": "USD"
                            },
                            "quantity": 10
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "cursor": "cursor",
                    "orders": [{
                            "date": "2000-01-23T04:56:07.000+00:00",
                            "totalValue": {
                                "amount": 0.8008281904610115,
                                "currency": "currency"
                            },
                            "quantity": 0.8008281904610115,
                            "instrument": {
                                "ticker": "ticker",
                                "price": {
                                    "amount": 0.8008281904610115,
                                    "currency": "currency"
                                },
                                "name": "name",
                                "exchange": "exchange",
                                "id": "id",
                                "iconUrl": "iconUrl"
                            },
                            "id": "id"
                        }, {
                            "date": "2000-01-23T04:56:07.000+00:00",
                            "totalValue": {
                                "amount": 0.8008281904610115,
                                "currency": "currency"
                            },
                            "quantity": 0.8008281904610115,
                            "instrument": {
                                "ticker": "ticker",
                                "price": {
                                    "amount": 0.8008281904610115,
                                    "currency": "currency"
                                },
                                "name": "name",
                                "exchange": "exchange",
                                "id": "id",
                                "iconUrl": "iconUrl"
                            },
                            "id": "id"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/api/v1/client-api/v1/trading/orders",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "id": "55bbeace-274e-22ec-5487-0242ac130003",
                    "type": "Market Order",
                    "tradeDirection": "Buy",
                    "status": "Draft",
                    "instrument": {
                        "id": "24aaeace-274e-22ec-5487-0242ac130003",
                        "name": "Apple",
                        "ticker": "AAPL",
                        "exchange": "NASDAQ",
                        "price": {
                            "value": 1247.15,
                            "currency": "USD"
                        },
                        "iconUrl": "/client-api/v1/trading/instruments/68bbeace-423e-11ec-9621-0242ac130002/icons"
                    },
                    "portfolio": {
                        "id": "66bbeace-274e-55ec-5487-0242ac130008",
                        "name": "Trading portfolio 2",
                        "alias": "My second portfolio to trade",
                        "availableBalance": {
                            "value": 6174.9,
                            "currency": "EUR"
                        }
                    },
                    "value": {
                        "value": 13095.1,
                        "currency": "USD"
                    },
                    "quantity": 10.5,
                    "totalValue": {
                        "value": 13160.6,
                        "currency": "USD"
                    },
                    "commissionFee": {
                        "value": 65.5,
                        "currency": "USD"
                    }
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/api/v1/client-api/v1/trading/orders/{id}/place",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "id": "55bbeace-274e-22ec-5487-0242ac130003",
                    "status": "Pending"
                }
            },
            {
                status: 200,
                body: {
                    "id": "55bbeace-274e-22ec-5487-0242ac130003",
                    "status": "Rejected",
                    "rejectionReason": "rejectionReason"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/api/v1/client-api/v1/trading/orders/{id}/status",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "55bbeace-274e-22ec-5487-0242ac130003",
                    "status": "Pending"
                }
            },
            {
                status: 200,
                body: {
                    "id": "55bbeace-274e-22ec-5487-0242ac130003",
                    "status": "Rejected",
                    "rejectionReason": "rejectionReason"
                }
            },
            {
                status: 200,
                body: {
                    "id": "id",
                    "rejectionReason": "rejectionReason"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
]);
