// @backbase/payment-orders-ang/esm2015/lib/ui/components/full-address/full-address.component.js
import { Component, Input } from '@angular/core';
import { addressToString } from '../../helpers/type-helpers';
import * as i0 from "@angular/core";
import * as i1 from "../country-name/country-name.component";
import * as i2 from "@angular/common";
export class FullAddressComponent {
    constructor() {
        this.hostRef = this;
    }
    ngOnInit() {
        if (!this.address) {
            console.warn('Please provide a valid `address` input');
        }
    }
    format() {
        return addressToString(this.address);
    }
}
/** @nocollapse */ FullAddressComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FullAddressComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ FullAddressComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: FullAddressComponent, selector: "bb-full-address", inputs: { address: "address" }, ngImport: i0, template: `
    <span class="break-word" data-role="full-address-details">
      {{ hostRef.format() }},
      <bb-country-name
        *ngIf="address?.country"
        [countryCode]="address?.country"
      ></bb-country-name>
    </span>
  `, isInline: true, components: [{ type: i1.CountryNameComponent, selector: "bb-country-name", inputs: ["countryCode"] }], directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FullAddressComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-full-address',
                    template: `
    <span class="break-word" data-role="full-address-details">
      {{ hostRef.format() }},
      <bb-country-name
        *ngIf="address?.country"
        [countryCode]="address?.country"
      ></bb-country-name>
    </span>
  `,
                }]
        }], propDecorators: { address: [{
                type: Input
            }] } });
