// @backbase/ui-ang/esm2015/_example/select-list/examples/select-list.example.component.js
import { Component } from '@angular/core';
import { FormControl, FormGroup } from '@angular/forms';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/select-list";
import * as i2 from "@angular/forms";
// Please replace relative path imports e.g. '../*' with '@backbase/ui-ang'.
export class SelectListExampleComponent {
    constructor() {
        this.items = [
            { name: 'One', value: 'one' },
            { name: 'Two', value: 'two' },
            { name: 'Three', value: 'three' },
        ];
    }
    ngOnInit() {
        this.form = new FormGroup({
            number: new FormControl([]),
        });
    }
}
SelectListExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SelectListExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
SelectListExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: SelectListExampleComponent, selector: "bb-select-list-example-ui", ngImport: i0, template: "<form [formGroup]=\"form\">\n  <h4 id=\"selectListLabel\">Numbers from one to three</h4>\n  <bb-select-list-ui\n    id=\"selectListInput\"\n    formControlName=\"number\"\n    placeholder=\"Start typing name of numbers. e.g. one\"\n    [items]=\"items\"\n    [minTermLength]=\"3\"\n    aria-labelledby=\"selectListLabel\"\n  >\n  </bb-select-list-ui>\n</form>\n", components: [{ type: i1.SelectListComponent, selector: "bb-select-list-ui", inputs: ["loading", "placeholder", "minTermLength", "maxSelectedItems", "multiple", "items", "filterEnabled"], outputs: ["valueChange", "search", "clear"] }], directives: [{ type: i2.ɵNgNoValidate, selector: "form:not([ngNoForm]):not([ngNativeValidate])" }, { type: i2.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { type: i2.FormGroupDirective, selector: "[formGroup]", inputs: ["formGroup"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { type: i2.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { type: i2.FormControlName, selector: "[formControlName]", inputs: ["disabled", "formControlName", "ngModel"], outputs: ["ngModelChange"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SelectListExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-select-list-example-ui',
                    templateUrl: './select-list.example.component.html',
                }]
        }] });
