// @backbase/user-common/esm2015/src/components/phone-number-input/phone-number-input-base.js
import { Directive, Input } from '@angular/core';
import { debounceTime, distinctUntilChanged } from 'rxjs/operators';
import { InputBase } from '../input-base/input-base';
import * as i0 from "@angular/core";
import * as i1 from "../../services/manage-profile/manage-profile-preferences.service";
import * as i2 from "@backbase/ui-ang/phone-number-format-pipe";
// eslint-disable-next-line @angular-eslint/directive-class-suffix
export class PhoneNumberInputBase extends InputBase {
    constructor(manageProfilePreferencesService, phoneNumberFormatter) {
        super();
        this.manageProfilePreferencesService = manageProfilePreferencesService;
        this.phoneNumberFormatter = phoneNumberFormatter;
        this.phoneNumberPreferencesSubscription = this.manageProfilePreferencesService
            .getPhoneNumberInputPreferences()
            .subscribe((preferences) => {
            if (preferences) {
                this.preferences = preferences;
                this.preferences.debounceTime = this.preferences.debounceTime * 1000;
            }
        });
    }
    // eslint-disable-next-line complexity
    ngOnInit() {
        if (this.enablePhoneNumberFormatter && this.phoneNumberDebounceTime === undefined) {
            throw new Error(`phoneNumberDebounceTime is required when enablePhoneNumberFormatter is true`);
        }
        if (this.enablePhoneNumberFormatter && this.phoneNumberCountryCode === undefined) {
            throw new Error(`phoneNumberCountryCode is required when enablePhoneNumberFormatter is true`);
        }
        if (this.enablePhoneNumberFormatter !== false) {
            if (this.enablePhoneNumberFormatter || (this.preferences || {}).enablePhoneNumberFormatter) {
                this.phoneNumberSubscription = this.fieldControl.valueChanges
                    .pipe(distinctUntilChanged(), debounceTime((this.phoneNumberDebounceTime && this.phoneNumberDebounceTime * 1000) || this.preference.debounceTime))
                    .subscribe((value) => {
                    value = this.phoneNumberFormatter.transform(value, this.phoneNumberCountryCode || this.preference.countryCode, true);
                    this.fieldControl.setValue(value);
                });
            }
        }
    }
    get preference() {
        if (!this.preferences) {
            throw new Error('preferences are not defined');
        }
        return this.preferences;
    }
    ngOnDestroy() {
        this.phoneNumberPreferencesSubscription.unsubscribe();
        if (this.phoneNumberSubscription) {
            this.phoneNumberSubscription.unsubscribe();
        }
    }
}
PhoneNumberInputBase.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PhoneNumberInputBase, deps: [{ token: i1.ManageProfilePreferencesService }, { token: i2.PhoneNumberPipe }], target: i0.ɵɵFactoryTarget.Directive });
PhoneNumberInputBase.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.15", type: PhoneNumberInputBase, selector: "[bbUserPhoneNumberInputBase]", inputs: { phoneNumberCountryCode: "phoneNumberCountryCode", phoneNumberDebounceTime: "phoneNumberDebounceTime", enablePhoneNumberFormatter: "enablePhoneNumberFormatter" }, usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PhoneNumberInputBase, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbUserPhoneNumberInputBase]',
                }]
        }], ctorParameters: function () { return [{ type: i1.ManageProfilePreferencesService }, { type: i2.PhoneNumberPipe }]; }, propDecorators: { phoneNumberCountryCode: [{
                type: Input
            }], phoneNumberDebounceTime: [{
                type: Input
            }], enablePhoneNumberFormatter: [{
                type: Input
            }] } });
