// @backbase/ui-ang/esm2015/iban-pipe/iban-pipe.pipe.js
import { Pipe } from '@angular/core';
import { segment } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/account-number-pipe";
/**
 * @name bbIban
 *
 * @description
 * Pipe for formatting iban
 */
export class BbIbanPipe {
    /**
     * @name BbIbanPipe#constructor
     * @param accountsDisplayingFormatService - AccountsDisplayingFormatService
     *
     * @internal
     */
    constructor(accountsDisplayingFormatService) {
        this.accountsDisplayingFormatService = accountsDisplayingFormatService;
        this.defaultFormat = this.accountsDisplayingFormatService.getDefaultFormat('iban');
    }
    /**
     * @name BbIbanPipe#transform
     * @param value - the iban that should be formatted
     *
     * @description
     * Splits the iban up with spaces
     * Applies global numbers format configuration for IBAN attribute type
     */
    transform(value) {
        return segment(value, this.defaultFormat.segments);
    }
}
BbIbanPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BbIbanPipe, deps: [{ token: i1.AccountsDisplayingFormatService }], target: i0.ɵɵFactoryTarget.Pipe });
BbIbanPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BbIbanPipe, name: "bbIban" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BbIbanPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'bbIban',
                }]
        }], ctorParameters: function () { return [{ type: i1.AccountsDisplayingFormatService }]; } });
