// @backbase/ui-ang/esm2015/iban-pipe/iban-pipe.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { BbIbanPipe } from './iban-pipe.pipe';
import * as i0 from "@angular/core";
export class BbIbanPipeModule {
}
BbIbanPipeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BbIbanPipeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
BbIbanPipeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BbIbanPipeModule, declarations: [BbIbanPipe], imports: [CommonModule], exports: [BbIbanPipe] });
BbIbanPipeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BbIbanPipeModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BbIbanPipeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [BbIbanPipe],
                    exports: [BbIbanPipe],
                }]
        }] });
