// @backbase/manage-payees-journey-ang/esm2015/src/views/billpay-create-manual-payee-review/billpay-create-manual-payee-review-view.component.js
import { Component, HostListener, ViewChild } from '@angular/core';
import { filter, switchMap, take, takeUntil, tap } from 'rxjs/operators';
import { BehaviorSubject, of, Subject } from 'rxjs';
import { BillpayDiscardChangesModalService, } from '@backbase/billpay-journeys-common';
import * as i0 from "@angular/core";
import * as i1 from "../../services/manage-payees-journey-config.service";
import * as i2 from "../../services/manage-payees-navigation.service";
import * as i3 from "../../services/billpay-manage-payee-state.service";
import * as i4 from "../../services/billpay-manage-payee-data.service";
import * as i5 from "@backbase/ui-ang/notification";
import * as i6 from "@backbase/billpay-journeys-common";
import * as i7 from "../../components/add-payee/billpay-manual-payee-review/billpay-manual-payee-review.component";
import * as i8 from "../../components/add-payee/billpay-cancel-new-payee-modal/billpay-cancel-new-payee-modal.component";
import * as i9 from "@angular/common";
import * as i10 from "@backbase/ui-ang/button";
export class CreateManualPayeeReviewViewComponent {
    /** @internal */
    constructor(config, router, stateService, dataService, notificationService, discardModalService, billpayNotificationRoutingService) {
        this.config = config;
        this.router = router;
        this.stateService = stateService;
        this.dataService = dataService;
        this.notificationService = notificationService;
        this.discardModalService = discardModalService;
        this.billpayNotificationRoutingService = billpayNotificationRoutingService;
        this.notificationHeaderTemplate = '';
        this.notificationMessageTemplate = '';
        this.payeeSubmit$ = new BehaviorSubject(undefined);
        this.destroy$ = new Subject();
    }
    get isDiscardModalOpen() {
        return this.discardModalService.isDiscardModalOpen;
    }
    ngOnInit() {
        this.stateService.manualPayeeFormState.pipe(take(1)).subscribe(payee => {
            this.payee = payee;
            if (!this.payee) {
                this.router.navigateToManualPayeeForm();
            }
        });
        this.payeeSubmit$
            .pipe(takeUntil(this.destroy$), filter(Boolean))
            .subscribe(payee => this.onSubmitPayee(payee));
    }
    ngOnDestroy() {
        this.destroy$.next();
        this.destroy$.complete();
    }
    onWindowUnload($event) {
        if (!this.isDiscardModalOpen) {
            $event.returnValue = true;
        }
    }
    submitPayee(payee) {
        this.payeeSubmit$.next(payee);
    }
    navigateBack() {
        this.router.navigateToManualPayeeForm();
    }
    cancel() {
        this.router.navigateToPayeeList();
    }
    navigateToPayee() {
        this.billpayNotificationRoutingService.navigateToPayees();
    }
    openModal() {
        return this.payeeSubmit$.pipe(switchMap(payee => (payee ? of(true) : this.discardModalService.openModal())), tap(isResolved => isResolved && this.stateService.resetManualPayee()));
    }
    onModalResolve(resolvedWith) {
        this.discardModalService.onModalResolve(resolvedWith);
    }
    onSubmitPayee(payee) {
        this.dataService.addManualPayee(payee).subscribe(() => {
            this.router.navigateToPayeeList();
            this.notificationService.showNotification({
                header: this.notificationHeaderTemplate,
                message: this.notificationMessageTemplate,
                modifier: 'success',
                ttl: this.dismissTimeout,
            });
        }, () => this.notificationService.showNotification({
            header: $localize `:@@billpay.create-manuel-payee.error-notification.header:Something went wrong`,
            message: $localize `:@@billpay.create-manuel-payee.error-notification.header:We were unable to proceed. Please try again.`,
            modifier: 'error',
            ttl: this.dismissTimeout,
        }));
    }
    get dismissTimeout() {
        if (this.config.notificationDismissTime === undefined) {
            throw new Error('Dismiss timeout value not found');
        }
        return this.config.notificationDismissTime;
    }
}
CreateManualPayeeReviewViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CreateManualPayeeReviewViewComponent, deps: [{ token: i1.ManagePayeesJourneyConfigService }, { token: i2.ManagePayeesNavigationService }, { token: i3.BillpayManagePayeeStateService }, { token: i4.BillpayManagePayeeDataService }, { token: i5.NotificationService }, { token: i6.BillpayDiscardChangesModalService }, { token: i6.BillpayNotificationRoutingService }], target: i0.ɵɵFactoryTarget.Component });
CreateManualPayeeReviewViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: CreateManualPayeeReviewViewComponent, selector: "ng-component", host: { listeners: { "window:beforeunload": "onWindowUnload($event)" } }, providers: [BillpayDiscardChangesModalService], viewQueries: [{ propertyName: "notificationHeaderTemplate", first: true, predicate: ["notificationHeader"], descendants: true }, { propertyName: "notificationMessageTemplate", first: true, predicate: ["notificationMessage"], descendants: true }], ngImport: i0, template: "<div class=\"card card-lg\" *ngIf=\"payee\">\n  <section class=\"card-body\">\n    <bb-billpay-manual-payee-review [payeeFormState]=\"payee\"> </bb-billpay-manual-payee-review>\n    <div class=\"bb-button-bar bb-button-bar--reverse\">\n      <button\n        bbButton\n        data-role=\"add-payee-button\"\n        i18n=\"Label for button|Manual payee continue button@@billpay-create-manual-payee-review.add-payee\"\n        (click)=\"submitPayee(payee)\"\n        class=\"bb-button-bar__button\"\n      >\n        Add payee\n      </button>\n      <button\n        bbButton\n        data-role=\"back-button\"\n        i18n=\"Label for button|Find a different payee button@@billpay-create-manual-payee-review.edit-payee\"\n        (click)=\"navigateBack()\"\n        class=\"bb-button-bar__button\"\n        color=\"secondary\"\n      >\n        Edit\n      </button>\n      <button\n        bbButton\n        data-role=\"cancel-button\"\n        color=\"link\"\n        class=\"bb-button-bar__button bb-button-bar__button--across\"\n        (click)=\"cancel()\"\n        i18n=\"Label for button|Cancel button@@billpay-create-manual-payee-review.cancel-payee\"\n      >\n        Cancel\n      </button>\n    </div>\n  </section>\n</div>\n\n<bb-billpay-cancel-new-payee-modal\n  *ngIf=\"isDiscardModalOpen\"\n  (resolve)=\"onModalResolve($event)\"\n></bb-billpay-cancel-new-payee-modal>\n\n<ng-template #notificationMessage>\n  <span\n    data-role=\"notification-message\"\n    i18n=\"\n      Payee notification message|Payee has created successfully\n      notification@@billpay-create-manual-payee-review.notification.message\"\n    >The payee \u2018{{ (payee.nickName && payee.nickName + ' (' + payee.name + ')') || payee.name }}\u2018 has been created\n    successfully\n  </span>\n\n  <div class=\"bb-button-bar bb-button-bar--reverse\">\n    <button\n      bbButton\n      data-role=\"notification-add-payee-button\"\n      color=\"primary\"\n      class=\"bb-button-bar__button\"\n      i18n=\"notification button|Add another payee@@billpay-create-manual-payee-review.notification.add-button\"\n      (click)=\"navigateToPayee()\"\n    >\n      Add another payee\n    </button>\n  </div>\n</ng-template>\n\n<ng-template #notificationHeader>\n  <span\n    i18n=\"\n      Payee notification header|Payee created success notification\n      header@@billpay-create-manual-payee-review.notification.header\"\n    >Payee created</span\n  >\n</ng-template>\n", components: [{ type: i7.BillpayManualPayeeReviewComponent, selector: "bb-billpay-manual-payee-review", inputs: ["payeeFormState"] }, { type: i8.BillpayCancelNewPayeeModalComponent, selector: "bb-billpay-cancel-new-payee-modal", inputs: ["findPayee"], outputs: ["resolve"] }], directives: [{ type: i9.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i10.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CreateManualPayeeReviewViewComponent, decorators: [{
            type: Component,
            args: [{
                    templateUrl: './billpay-create-manual-payee-review-view.component.html',
                    providers: [BillpayDiscardChangesModalService],
                }]
        }], ctorParameters: function () { return [{ type: i1.ManagePayeesJourneyConfigService }, { type: i2.ManagePayeesNavigationService }, { type: i3.BillpayManagePayeeStateService }, { type: i4.BillpayManagePayeeDataService }, { type: i5.NotificationService }, { type: i6.BillpayDiscardChangesModalService }, { type: i6.BillpayNotificationRoutingService }]; }, propDecorators: { notificationHeaderTemplate: [{
                type: ViewChild,
                args: ['notificationHeader']
            }], notificationMessageTemplate: [{
                type: ViewChild,
                args: ['notificationMessage']
            }], onWindowUnload: [{
                type: HostListener,
                args: ['window:beforeunload', ['$event']]
            }] } });
