// @backbase/billpay-journeys-common/esm2015/lib/billpay-journeys-common.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { ReactiveFormsModule } from '@angular/forms';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { BillpayEnrolmentService } from './services/billpay-enrolment.service';
import { BillpayEnrolmentGuard } from './services/billpay-enrolment.guard';
import { BillpayPaymentParserService } from './services/billpay-payment-parser.service';
import { BillpayPaymentScheduleService } from './services/billpay-payment-schedule.service';
import { BillpayUtilityService } from './services/billpay-utility.service';
import { BillpayPaymentDataService } from './services/billpay-payment-data.service';
import { BillpayEnrolmentComponent } from './components/billpay-enrolment/billpay-enrolment.component';
import { BillpayPaymentReviewCardComponent } from './components/billpay-payment-review-card/billpay-payment-review-card.component';
import { BillpayAddressFormComponent } from './components/billpay-address-form/billpay-address-form.component';
import { BillpayPayeeNavigateModalComponent } from './components/billpay-payee-navigate-modal/billpay-payee-navigate-modal.component';
import { BillpayPayeeDeleteModalComponent } from './components/billpay-payee-delete-modal/billpay-payee-delete-modal.component';
import { BillpayEbillsDeleteModalComponent } from './components/billpay-ebills-delete-modal/billpay-ebills-delete-modal.component';
import { AmountModule } from '@backbase/ui-ang/amount';
import { ButtonModule } from '@backbase/ui-ang/button';
import { DropdownSingleSelectModule } from '@backbase/ui-ang/dropdown-single-select';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { HeaderModule } from '@backbase/ui-ang/header';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { ModalModule } from '@backbase/ui-ang/modal';
import { NotificationModule } from '@backbase/ui-ang/notification';
import { PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import { KeyboardClickModule } from '@backbase/ui-ang/keyboard-click-directive';
import * as i0 from "@angular/core";
const uiModules = [
    ButtonModule,
    LoadingIndicatorModule,
    EmptyStateModule,
    NotificationModule,
    HeaderModule,
    AmountModule,
    PaymentCardNumberModule,
    InputValidationMessageModule,
    DropdownSingleSelectModule,
    ModalModule,
    InputTextModule,
    KeyboardClickModule,
];
const services = [
    BillpayEnrolmentService,
    BillpayEnrolmentGuard,
    BillpayPaymentParserService,
    BillpayPaymentScheduleService,
    BillpayUtilityService,
    BillpayPaymentDataService,
];
const components = [
    BillpayEnrolmentComponent,
    BillpayPaymentReviewCardComponent,
    BillpayAddressFormComponent,
    BillpayPayeeNavigateModalComponent,
    BillpayPayeeDeleteModalComponent,
    BillpayEbillsDeleteModalComponent,
];
export class BillpayJourneysCommonModule {
}
/** @nocollapse */ BillpayJourneysCommonModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BillpayJourneysCommonModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ BillpayJourneysCommonModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BillpayJourneysCommonModule, declarations: [BillpayEnrolmentComponent,
        BillpayPaymentReviewCardComponent,
        BillpayAddressFormComponent,
        BillpayPayeeNavigateModalComponent,
        BillpayPayeeDeleteModalComponent,
        BillpayEbillsDeleteModalComponent], imports: [CommonModule, RouterModule, ReactiveFormsModule, BackbaseCoreModule, ButtonModule,
        LoadingIndicatorModule,
        EmptyStateModule,
        NotificationModule,
        HeaderModule,
        AmountModule,
        PaymentCardNumberModule,
        InputValidationMessageModule,
        DropdownSingleSelectModule,
        ModalModule,
        InputTextModule,
        KeyboardClickModule], exports: [BillpayEnrolmentComponent,
        BillpayPaymentReviewCardComponent,
        BillpayAddressFormComponent,
        BillpayPayeeNavigateModalComponent,
        BillpayPayeeDeleteModalComponent,
        BillpayEbillsDeleteModalComponent, ButtonModule,
        LoadingIndicatorModule,
        EmptyStateModule,
        NotificationModule,
        HeaderModule,
        AmountModule,
        PaymentCardNumberModule,
        InputValidationMessageModule,
        DropdownSingleSelectModule,
        ModalModule,
        InputTextModule,
        KeyboardClickModule] });
/** @nocollapse */ BillpayJourneysCommonModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BillpayJourneysCommonModule, providers: [...services], imports: [[CommonModule, RouterModule, ReactiveFormsModule, BackbaseCoreModule, ...uiModules], ButtonModule,
        LoadingIndicatorModule,
        EmptyStateModule,
        NotificationModule,
        HeaderModule,
        AmountModule,
        PaymentCardNumberModule,
        InputValidationMessageModule,
        DropdownSingleSelectModule,
        ModalModule,
        InputTextModule,
        KeyboardClickModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BillpayJourneysCommonModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components],
                    imports: [CommonModule, RouterModule, ReactiveFormsModule, BackbaseCoreModule, ...uiModules],
                    providers: [...services],
                    exports: [...components, ...uiModules],
                }]
        }] });
