// @backbase/contact-common-ang/esm2015/lib/components/contact-account-data/contact-account-data.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { ContactAccountDataComponent } from './contact-account-data.component';
import { ContactAccountDataCustomizableDirective } from './contact-account-data.directive';
import { ContactAddressDataModule } from '../contact-address-data/contact-address-data.module';
import { AccountCardViewModule } from '../account-card-view/account-card-view.module';
import { NoWhiteSpaceModule } from '../../directives/no-white-space-validator/no-white-space.module';
import { IbanValidatorModule } from '../../directives/iban-validator/iban-validator.module';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { DropdownSingleSelectModule } from '@backbase/ui-ang/dropdown-single-select';
import * as i0 from "@angular/core";
/**
 * Exports:
 * - ContactAccountDataComponent
 * - ContactAccountDataCustomizableDirective
 */
export class ContactAccountDataModule {
}
/** @nocollapse */ ContactAccountDataModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactAccountDataModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ ContactAccountDataModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactAccountDataModule, declarations: [ContactAccountDataComponent, ContactAccountDataCustomizableDirective], imports: [CommonModule,
        FormsModule,
        InputTextModule,
        InputValidationMessageModule,
        DropdownSingleSelectModule,
        ContactAddressDataModule,
        AccountCardViewModule,
        NoWhiteSpaceModule,
        IbanValidatorModule], exports: [ContactAccountDataComponent, ContactAccountDataCustomizableDirective] });
/** @nocollapse */ ContactAccountDataModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactAccountDataModule, imports: [[
            CommonModule,
            FormsModule,
            InputTextModule,
            InputValidationMessageModule,
            DropdownSingleSelectModule,
            ContactAddressDataModule,
            AccountCardViewModule,
            NoWhiteSpaceModule,
            IbanValidatorModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactAccountDataModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ContactAccountDataComponent, ContactAccountDataCustomizableDirective],
                    exports: [ContactAccountDataComponent, ContactAccountDataCustomizableDirective],
                    imports: [
                        CommonModule,
                        FormsModule,
                        InputTextModule,
                        InputValidationMessageModule,
                        DropdownSingleSelectModule,
                        ContactAddressDataModule,
                        AccountCardViewModule,
                        NoWhiteSpaceModule,
                        IbanValidatorModule,
                    ],
                }]
        }] });
