// @backbase/authorized-users-journey-ang/esm2015/src/authorized-users-create/components/authorized-users-create-container/authorized-users-create-container.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
export class AuthorizedUsersCreateContainerComponent {
    constructor() {
        this.nextStep = 1;
    }
    nextStepIndex() {
        return this.nextStep++;
    }
}
AuthorizedUsersCreateContainerComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersCreateContainerComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
AuthorizedUsersCreateContainerComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AuthorizedUsersCreateContainerComponent, selector: "bb-authorized-users-create-container", ngImport: i0, template: "<div class=\"card\">\n  <div class=\"card-body modal-body\">\n    <ng-content></ng-content>\n  </div>\n</div>" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersCreateContainerComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-authorized-users-create-container',
                    templateUrl: 'authorized-users-create-container.component.html',
                }]
        }] });
