// @backbase/ui-ang/esm2015/_example/load-button/examples/load-button.example.component.js
import { Component } from '@angular/core';
import { Subject, merge, of } from 'rxjs';
import { startWith, mergeMap, map, delay, tap } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/load-button";
import * as i2 from "@backbase/ui-ang/icon";
import * as i3 from "@angular/common";
const LOADING_DURATION = 1500;
export class LoadButtonExampleComponent {
    constructor() {
        this.itemsPerPage = 4;
        this.page = 0;
        this.totalItems = 12;
        this.loadRequest = new Subject();
        this.loadProcedure = this.loadRequest.pipe(mergeMap((params) => this.loadMore(params)));
        this.isLoading = merge(this.loadRequest.pipe(map(() => true)), this.loadProcedure.pipe(map(() => false))).pipe(startWith(false));
    }
    onLoadButtonPress(params) {
        this.loadRequest.next(params);
    }
    loadMore(params) {
        console.log('Loading more...');
        return of(params).pipe(delay(LOADING_DURATION), tap(() => {
            this.page = params.from;
        }));
    }
}
LoadButtonExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoadButtonExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
LoadButtonExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: LoadButtonExampleComponent, selector: "bb-load-button-example-ui", ngImport: i0, template: "<bb-load-button-ui\n  color=\"primary\"\n  [size]=\"itemsPerPage\"\n  [from]=\"page\"\n  [totalCount]=\"totalItems\"\n  [isLoading]=\"isLoading | async\"\n  buttonSize=\"md\"\n  (paramChange)=\"onLoadButtonPress($event)\"\n>\n  <bb-icon-ui name=\"search\" size=\"lg\" cropped></bb-icon-ui>\n  Show more results...\n</bb-load-button-ui>\n", components: [{ type: i1.LoadButtonComponent, selector: "bb-load-button-ui", inputs: ["isLoading", "from", "size", "circle", "totalCount"], outputs: ["paramChange"] }, { type: i2.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], pipes: { "async": i3.AsyncPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoadButtonExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-load-button-example-ui',
                    templateUrl: './load-button.example.component.html',
                }]
        }] });
