// @backbase/payment-orders-ang/esm2015/lib/ui/pipes/account-formatter/account-formatter.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AccountFormatterPipe } from './account-formatter.pipe';
import * as i0 from "@angular/core";
export class AccountFormatterModule {
}
/** @nocollapse */ AccountFormatterModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountFormatterModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ AccountFormatterModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountFormatterModule, declarations: [AccountFormatterPipe], imports: [CommonModule], exports: [AccountFormatterPipe] });
/** @nocollapse */ AccountFormatterModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountFormatterModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountFormatterModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [AccountFormatterPipe],
                    imports: [CommonModule],
                    exports: [AccountFormatterPipe],
                }]
        }] });
