// @backbase/ui-ang/esm2015/_example/period-selector/showcase/period-selector-showcase.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/period-selector";
let PeriodSelectorShowcaseComponent = class PeriodSelectorShowcaseComponent {
    constructor() {
        this.periodStartDate = new Date('2019-02-09');
        this.periodEndDate = new Date('2019-02-21');
    }
};
PeriodSelectorShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PeriodSelectorShowcaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
PeriodSelectorShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: PeriodSelectorShowcaseComponent, selector: "bb-period-selector-ui-showcase", ngImport: i0, template: "<bb-period-selector-ui [periodStart]=\"periodStartDate\" [periodEnd]=\"periodEndDate\"></bb-period-selector-ui>\n", components: [{ type: i1.PeriodSelectorComponent, selector: "bb-period-selector-ui", inputs: ["periodStart", "periodEnd", "periodFormatters", "period"], outputs: ["periodChange"] }] });
PeriodSelectorShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
], PeriodSelectorShowcaseComponent);
export { PeriodSelectorShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PeriodSelectorShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-period-selector-ui-showcase',
                    templateUrl: './period-selector-showcase.component.html',
                }]
        }] });
