// @backbase/ui-ang/esm2015/_example/period-selector/showcase/period-selector-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { PeriodSelectorShowcaseComponent } from './period-selector-showcase.component';
import { PeriodSelectorModule } from '@backbase/ui-ang/period-selector';
import * as i0 from "@angular/core";
export class PeriodSelectorShowcaseModule {
}
PeriodSelectorShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PeriodSelectorShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PeriodSelectorShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PeriodSelectorShowcaseModule, declarations: [PeriodSelectorShowcaseComponent], imports: [CommonModule, PeriodSelectorModule] });
PeriodSelectorShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PeriodSelectorShowcaseModule, imports: [[CommonModule, PeriodSelectorModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PeriodSelectorShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, PeriodSelectorModule],
                    declarations: [PeriodSelectorShowcaseComponent],
                    entryComponents: [PeriodSelectorShowcaseComponent],
                }]
        }] });
