// @backbase/actions-common-ang/esm2015/src/helpers/time.helper.js
const AM_PM = ['AM', 'PM'];
export const DAYS_OF_WEEK = ['MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT', 'SUN'];
export function getTimeOptions(startTime, minutesStep, excludedOptions, amPmFormat) {
    const timeOptions = [];
    let index = 0;
    while (startTime < 24 * 60) {
        const hh = Math.floor(startTime / 60);
        const mm = startTime % 60;
        if (!excludedOptions.includes(startTime)) {
            timeOptions[index] = {
                label: getOptionLabel(hh, mm, amPmFormat),
                time: startTime,
            };
            index++;
        }
        startTime = startTime + minutesStep;
    }
    return timeOptions;
}
export function getHoursOfDay(scheduleTime = []) {
    const list = scheduleTime.map(time => String(time / 60));
    if (!list.length) {
        for (let i = 0; i < 24; i++) {
            list.push(`${i}`);
        }
    }
    return list;
}
export function hoursToMinutesArray(hours) {
    return (hours === null || hours === void 0 ? void 0 : hours.map(hour => Number(hour) * 60)) || [];
}
function getOptionLabel(hh, mm, amPmFormat) {
    const getFormattedTime = (hours = hh, suffix = '') => ('0' + hours).slice(-2) + '.' + ('0' + mm).slice(-2) + suffix;
    return amPmFormat ? getFormattedTime(hh % 12, AM_PM[Math.floor(hh / 12)]) : getFormattedTime();
}
