// @backbase/manage-payees-journey-ang/esm2015/src/components/add-payee/billpay-electronic-payee-search/billpay-electronic-payee-search.component.js
import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
import { of } from 'rxjs';
import { catchError, debounceTime, distinctUntilChanged, filter, map, switchMap } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "../../../services/billpay-manage-payee-data.service";
import * as i2 from "@backbase/ui-ang/search-box";
export class BillpayElectronicPayeeSearchComponent {
    /** @internal */
    constructor(dataService) {
        this.dataService = dataService;
        /**
         * Fires a payee's id when electronic payee is selected in the search box.
         */
        this.payeeSelected = new EventEmitter();
        this.errorSearching = new EventEmitter();
        this.typeaheadOptions = {
            ngbTypeahead: this.search.bind(this),
            selectItem: this.onItemSelected.bind(this),
            resultFormatter: (payee) => payee.name,
            inputFormatter: (payee) => payee.name,
        };
    }
    search(text$) {
        const payees$ = (text) => this.dataService.searchPayeesByName(text).pipe(map(({ payees }) => payees || []), catchError(() => {
            this.errorSearching.emit();
            return of([]);
        }));
        return text$.pipe(debounceTime(300), distinctUntilChanged(), filter(text => text.length >= 2), switchMap(payees$), catchError(() => of([])));
    }
    onItemSelected(event) {
        this.payeeSelected.emit(event.item);
    }
}
BillpayElectronicPayeeSearchComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayElectronicPayeeSearchComponent, deps: [{ token: i1.BillpayManagePayeeDataService }], target: i0.ɵɵFactoryTarget.Component });
BillpayElectronicPayeeSearchComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: BillpayElectronicPayeeSearchComponent, selector: "bb-billpay-electronic-payee-search", inputs: { descTextId: "descTextId" }, outputs: { payeeSelected: "payeeSelected", errorSearching: "errorSearching" }, ngImport: i0, template: "<div class=\"bb-block bb-block--xl\">\n  <bb-search-box-ui\n    [role]=\"'combobox'\"\n    showClear=\"true\"\n    [aria-describedby]=\"descTextId\"\n    label=\"Payee name\"\n    i18n-label=\"\n      Payee name|Label for electronic search input @@billpay-manage-payee.electronic-payee-search.form.search-label\"\n    clearLabel=\"Clear search\"\n    i18n-clearLabel=\"\n      The aria label used when the clear button is in focus@@billpay-manage-payee.electronic-payee-search.clear-label\"\n    [typeaheadOptions]=\"typeaheadOptions\"\n    [autocomplete]=\"'off'\"\n    autofocus\n  ></bb-search-box-ui>\n</div>\n", components: [{ type: i2.SearchBoxComponent, selector: "bb-search-box-ui", inputs: ["placeholder", "maxLength", "squareBorder", "typeaheadOptions", "showSearch", "showClear", "searchLabel", "clearLabel", "autocomplete"], outputs: ["submit", "clear", "valueChange"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayElectronicPayeeSearchComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-billpay-electronic-payee-search',
                    templateUrl: './billpay-electronic-payee-search.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.BillpayManagePayeeDataService }]; }, propDecorators: { payeeSelected: [{
                type: Output
            }], errorSearching: [{
                type: Output
            }], descTextId: [{
                type: Input
            }] } });
