// @backbase/budget-journey-ang/esm2015/src/views/budgets-view.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../budget-journey-config.service";
import * as i2 from "@backbase/ui-ang/header";
import * as i3 from "@backbase/budget-widget-ang";
// TODO: remove utility classes
export class BudgetViewComponent {
    constructor(config) {
        this.config = config;
    }
}
BudgetViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetViewComponent, deps: [{ token: i1.BudgetJourneyService }], target: i0.ɵɵFactoryTarget.Component });
BudgetViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: BudgetViewComponent, selector: "bb-budgets-view", ngImport: i0, template: `
    <bb-header-ui
      headingType="h1"
      headingClasses="bb-heading-widget__heading px-0 pb-5"
      heading="Budgets"
      i18n-heading="Budgets|Heading for budget journey@@budget.journey.heading.text"
    ></bb-header-ui>
    <bb-budget-widget
      [showPercentage]="config.showPercentage"
      [budgetSafeZoneLimit]="config.budgetSafeZoneLimit"
      [notificationDismissTime]="config.notificationDismissTime"
      [maxBudgets]="config.maxBudgets"
    ></bb-budget-widget>
  `, isInline: true, components: [{ type: i2.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i3.BudgetWidgetComponent, selector: "bb-budget-widget", inputs: ["showPercentage", "budgetSafeZoneLimit", "notificationDismissTime", "maxBudgets"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetViewComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-budgets-view',
                    template: `
    <bb-header-ui
      headingType="h1"
      headingClasses="bb-heading-widget__heading px-0 pb-5"
      heading="Budgets"
      i18n-heading="Budgets|Heading for budget journey@@budget.journey.heading.text"
    ></bb-header-ui>
    <bb-budget-widget
      [showPercentage]="config.showPercentage"
      [budgetSafeZoneLimit]="config.budgetSafeZoneLimit"
      [notificationDismissTime]="config.notificationDismissTime"
      [maxBudgets]="config.maxBudgets"
    ></bb-budget-widget>
  `,
                }]
        }], ctorParameters: function () { return [{ type: i1.BudgetJourneyService }]; } });
