// @backbase/manage-pockets-journey-ang/esm2015/src/manage-pockets-list/components/manage-pockets-list/manage-pockets-list.component.js
import { Component, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../../../manage-pockets-common/components/manage-pockets-card/manage-pockets-card.component";
import * as i2 from "@angular/common";
export class ManagePocketsListComponent {
    constructor() {
        this.pocketTransfer = new EventEmitter();
    }
    goToPocketTransfer(transferType) {
        this.pocketTransfer.emit(transferType);
    }
}
ManagePocketsListComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsListComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ManagePocketsListComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ManagePocketsListComponent, selector: "bb-manage-pockets-list", inputs: { pocketList: "pocketList" }, outputs: { pocketTransfer: "pocketTransfer" }, ngImport: i0, template: "<ng-container *ngFor=\"let pocket of pocketList\">\n  <div class=\"bb-block bb-block--xl\">\n    <bb-manage-pockets-card\n      [pocket]=\"pocket\"\n      (pocketTransfer)=\"goToPocketTransfer($event)\"\n    ></bb-manage-pockets-card>\n  </div>\n</ng-container>", components: [{ type: i1.ManagePocketsCardComponent, selector: "bb-manage-pockets-card", inputs: ["pocket"], outputs: ["pocketTransfer"] }], directives: [{ type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsListComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-manage-pockets-list',
                    templateUrl: `./manage-pockets-list.component.html`,
                }]
        }], propDecorators: { pocketList: [{
                type: Input
            }], pocketTransfer: [{
                type: Output
            }] } });
