// @backbase/user-http-ang/esm2015/api/userApprovals.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const UserApprovalsServiceApproveApprovalMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/users/approvals/{approvalId}/approve",
        method: "POST",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    }]);
export const UserApprovalsServiceGetApprovalDetailsMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/users/approvals/{approvalId}/details",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "changelogDetails": [{
                            "type": "LOCK_USER",
                            "key": "User Status",
                            "from": "Locked",
                            "to": "Enabled"
                        }],
                    "userDetails": {
                        "username": "john.doe",
                        "legalEntityName": "Bank 001",
                        "fullName": "John Doe",
                        "emailAddress": "john.doe@example.test",
                        "mobileNumber": "012345 67890",
                        "dateCreated": "2019-07-11T11:11:11.000Z"
                    }
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    }]);
export const UserApprovalsServiceRejectApprovalMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/users/approvals/{approvalId}/reject",
        method: "POST",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    }]);
export const UserApprovalsServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v2/users/approvals/{approvalId}/approve",
        method: "POST",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/users/approvals/{approvalId}/details",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "changelogDetails": [{
                            "type": "LOCK_USER",
                            "key": "User Status",
                            "from": "Locked",
                            "to": "Enabled"
                        }],
                    "userDetails": {
                        "username": "john.doe",
                        "legalEntityName": "Bank 001",
                        "fullName": "John Doe",
                        "emailAddress": "john.doe@example.test",
                        "mobileNumber": "012345 67890",
                        "dateCreated": "2019-07-11T11:11:11.000Z"
                    }
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/users/approvals/{approvalId}/reject",
        method: "POST",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    },
]);
