// @backbase/manage-pockets-journey-ang/esm2015/src/manage-pockets-common/components/manage-pockets-base-wrapper/manage-pockets-base-wrapper.component.js
import { Directive, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
export class ManagePocketsBaseWrapperComponent {
    constructor() {
        this.cancel = new EventEmitter();
        this.currentStep = 1;
        this.showDialog = false;
    }
    confirmAbort() {
        this.cancel.emit();
        this.showDialog = false;
    }
    showAbortDialog() {
        this.showDialog = true;
    }
    closeAbortDialog() {
        this.showDialog = false;
    }
    goBack() {
        this.goToStep(this.currentStep - 1);
    }
    goToStep(value) {
        this.currentStep = value;
    }
}
ManagePocketsBaseWrapperComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsBaseWrapperComponent, deps: [], target: i0.ɵɵFactoryTarget.Directive });
ManagePocketsBaseWrapperComponent.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: ManagePocketsBaseWrapperComponent, inputs: { journeyCurrency: "journeyCurrency" }, outputs: { cancel: "cancel" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsBaseWrapperComponent, decorators: [{
            type: Directive
        }], propDecorators: { journeyCurrency: [{
                type: Input
            }], cancel: [{
                type: Output
            }] } });
