// @backbase/ui-ang/esm2015/_example/input-text/showcase/input-text-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { InputTextShowcaseComponent } from './input-text-showcase.component';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import { ReactiveFormsModule } from '@angular/forms';
import * as i0 from "@angular/core";
export class InputTextShowcaseModule {
}
InputTextShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputTextShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputTextShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputTextShowcaseModule, declarations: [InputTextShowcaseComponent], imports: [CommonModule, InputTextModule, ReactiveFormsModule] });
InputTextShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputTextShowcaseModule, imports: [[CommonModule, InputTextModule, ReactiveFormsModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputTextShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, InputTextModule, ReactiveFormsModule],
                    declarations: [InputTextShowcaseComponent],
                    entryComponents: [InputTextShowcaseComponent],
                }]
        }] });
