// @backbase/ui-ang/esm2015/_example/date-pipe/showcase/date-pipe-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { DatePipeShowcaseComponent } from './date-pipe-showcase.component';
import { BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import * as i0 from "@angular/core";
export class DatePipeShowcaseModule {
}
DatePipeShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DatePipeShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
DatePipeShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DatePipeShowcaseModule, declarations: [DatePipeShowcaseComponent], imports: [CommonModule, BbDatePipeModule] });
DatePipeShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DatePipeShowcaseModule, imports: [[CommonModule, BbDatePipeModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DatePipeShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, BbDatePipeModule],
                    declarations: [DatePipeShowcaseComponent],
                    entryComponents: [DatePipeShowcaseComponent],
                }]
        }] });
