// @backbase/accounts-transactions-journeys-shared-ui-account-ang/esm2015/lib/components/account-info-frequency/account-info-frequency.component.js
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "@backbase/product-summary-common-ang";
export class AccountInfoFrequencyComponent {
    constructor() {
        this.localisationByUnit = {
            day: $localize `:@@accounts-transactions-journeys.account-frequency.day:day(s)`,
            week: $localize `:@@accounts-transactions-journeys.account-frequency.week:week(s)`,
            month: $localize `:@@accounts-transactions-journeys.account-frequency.month:month(s)`,
            year: $localize `:@@accounts-transactions-journeys.account-frequency.year:year(s)`,
        };
    }
    get unitLabel() {
        switch (this.unit) {
            case 'D':
                return this.localisationByUnit.day;
            case 'W':
                return this.localisationByUnit.week;
            case 'M':
                return this.localisationByUnit.month;
            case 'Y':
                return this.localisationByUnit.year;
            default:
                return '';
        }
    }
}
/** @nocollapse */ AccountInfoFrequencyComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountInfoFrequencyComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ AccountInfoFrequencyComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AccountInfoFrequencyComponent, selector: "bb-account-info-frequency", inputs: { value: "value", unit: "unit" }, ngImport: i0, template: "<span *ngIf=\"value | bbIsDefined\" data-role=\"account-info-frequency\">\n  {{ value }}\n  {{ unitLabel }}\n</span>\n", directives: [{ type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "bbIsDefined": i2.IsDefinedPipe }, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountInfoFrequencyComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-account-info-frequency',
                    templateUrl: './account-info-frequency.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { value: [{
                type: Input
            }], unit: [{
                type: Input
            }] } });
