// @backbase/accounts-transactions-journeys-shared-ui-account-ang/esm2015/lib/components/account-info-frequency/account-info-frequency.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ValidationModule } from '@backbase/product-summary-common-ang';
import { AccountInfoFrequencyComponent } from './account-info-frequency.component';
import * as i0 from "@angular/core";
const pipeModules = [ValidationModule];
const components = [AccountInfoFrequencyComponent];
export class AccountInfoFrequencyModule {
}
/** @nocollapse */ AccountInfoFrequencyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountInfoFrequencyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ AccountInfoFrequencyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountInfoFrequencyModule, declarations: [AccountInfoFrequencyComponent], imports: [CommonModule, ValidationModule], exports: [AccountInfoFrequencyComponent, ValidationModule] });
/** @nocollapse */ AccountInfoFrequencyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountInfoFrequencyModule, imports: [[CommonModule, ...pipeModules], ValidationModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountInfoFrequencyModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...pipeModules],
                    declarations: [...components],
                    exports: [...components, ...pipeModules],
                }]
        }] });
