// @backbase/messages-ui-ang/esm2015/lib/messages-tabs/messages-tabs.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { MessagesTabsComponent } from './messages-tabs.component';
import * as i0 from "@angular/core";
export class MessagesTabsModule {
}
/** @nocollapse */ MessagesTabsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesTabsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ MessagesTabsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesTabsModule, declarations: [MessagesTabsComponent], imports: [CommonModule, RouterModule], exports: [MessagesTabsComponent] });
/** @nocollapse */ MessagesTabsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesTabsModule, imports: [[CommonModule, RouterModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesTabsModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [MessagesTabsComponent],
                    imports: [CommonModule, RouterModule],
                    exports: [MessagesTabsComponent],
                }]
        }] });
