// @backbase/loans-collection-ui-ang/esm2015/lib/loan-list-item/loan-item-status/loan-item-status.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BadgeModule } from '@backbase/ui-ang/badge';
import { LoanItemStatusComponent } from './loan-item-status.component';
import * as i0 from "@angular/core";
export class LoanItemStatusModule {
}
/** @nocollapse */ LoanItemStatusModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemStatusModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanItemStatusModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemStatusModule, declarations: [LoanItemStatusComponent], imports: [CommonModule, BadgeModule], exports: [LoanItemStatusComponent] });
/** @nocollapse */ LoanItemStatusModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemStatusModule, imports: [[CommonModule, BadgeModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemStatusModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanItemStatusComponent],
                    imports: [CommonModule, BadgeModule],
                    exports: [LoanItemStatusComponent],
                }]
        }] });
