// @backbase/loans-collection-ui-ang/esm2015/lib/loan-list-item/loan-item-status/loan-item-status.component.js
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { loanItemStatusColorMap } from './loan-item-status.model';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/badge";
import * as i2 from "@angular/common";
export class LoanItemStatusComponent {
    constructor() {
        /**
         * Map of colors for each status
         */
        this.itemStatusColorMap = loanItemStatusColorMap;
    }
}
/** @nocollapse */ LoanItemStatusComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemStatusComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ LoanItemStatusComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: LoanItemStatusComponent, selector: "bb-loan-item-status", inputs: { status: "status", itemStatusColorMap: "itemStatusColorMap" }, ngImport: i0, template: "<ng-container *ngIf=\"status\">\n  <span class=\"sr-only\" i18n=\"Loan item status a11y prefix@@loans.loan-item.loan-status.a11y.prefix\"> Status: </span>\n  <span class=\"d-none\" i18n=\"Loan item status@@loans.loan-item.status\" #textLabel\n    >{status, select, Active {Active} Inactive {Inactive} Pending {Pending} Overdue {Overdue}}</span\n  >\n  <bb-badge-ui\n    [color]=\"itemStatusColorMap[status]\"\n    [text]=\"textLabel.textContent! | titlecase\"\n    data-role=\"bb-loans-badge\"\n  ></bb-badge-ui>\n  <span class=\"sr-only\">,</span>\n</ng-container>\n", components: [{ type: i1.BadgeComponent, selector: "bb-badge-ui", inputs: ["color", "text", "abbr", "circle", "wrap", "ariaLabel"] }], directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "titlecase": i2.TitleCasePipe }, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemStatusComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-loan-item-status',
                    templateUrl: './loan-item-status.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { status: [{
                type: Input
            }], itemStatusColorMap: [{
                type: Input
            }] } });
