// @backbase/foundation-ang/esm2015/future/deprecations.service.js
import { Injectable, InjectionToken, Inject, Optional } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * @deprecated Will be removed in v7.0.0
 */
export const LOG_DEPRECATIONS_CONFIG = new InjectionToken('@backbase/foundation Log Deprecation Warnings Configuration');
/**
 * @deprecated Will be removed in v7.0.0
 * Log deprecated features.
 *
 * Deprecations will only be logged when `logDeprecations` has been
 * configured to "true" in `BackbaseCoreConfiguration`.
 *
 * @usageNotes
 *
 * ### Example
 *
 * Log a deprecated property:
 *
 * ```
 * @Component({
 *   …
 * })
 * export class MyComponent {
 *   private myOldPropertyValue: string | undefined;
 *
 *   constructor(private deprecationsService: DeprecationsService) { }
 *
 *   @Input()
 *   set myOldProperty(prop: string | undefined) {
 *     this.deprecationsService.logDeprecatedFeature("The `myOldProperty` property of MyComponent has been deprecated");
 *     this.myOldPropertyValue = prop;
 *   }
 * }
 */
export class DeprecationsService {
    constructor(logDeprecations) {
        this.logDeprecations = !!logDeprecations;
    }
    logDeprecatedFeature(warning) {
        if (!this.logDeprecations) {
            return;
        }
        console.warn('[DEPRECATED]', warning);
    }
}
DeprecationsService.ɵprov = i0.ɵɵdefineInjectable({ factory: function DeprecationsService_Factory() { return new DeprecationsService(i0.ɵɵinject(LOG_DEPRECATIONS_CONFIG, 8)); }, token: DeprecationsService, providedIn: "root" });
DeprecationsService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root',
            },] }
];
DeprecationsService.ctorParameters = () => [
    { type: Boolean, decorators: [{ type: Optional }, { type: Inject, args: [LOG_DEPRECATIONS_CONFIG,] }] }
];
