// @backbase/income-spending-analysis-widget-ang/esm2015/src/constants.js
export var AnalysisTypes;
(function (AnalysisTypes) {
    AnalysisTypes["INCOME"] = "CRDT";
    AnalysisTypes["SPENDING"] = "DBIT";
})(AnalysisTypes || (AnalysisTypes = {}));
export const PROPERTIES = {
    analysisType: {
        name: 'analysisIndicator',
        value: AnalysisTypes.SPENDING,
    },
};
export var Interval;
(function (Interval) {
    Interval["DAY"] = "DAY";
    Interval["WEEK"] = "WEEK";
    Interval["MONTH"] = "MONTH";
    Interval["YEAR"] = "YEAR";
})(Interval || (Interval = {}));
export const DEFAULT_INTERVAL = Interval.MONTH;
export const RELEVANCE_PERCENTAGE = 2;
export const CATEGORY_COLORS_CSS_VAR = '--chart-categories-colors';
export const BAR_COLORS_CSS_VAR = '--chart-group-bar-color';
export var TrendTextClasses;
(function (TrendTextClasses) {
    TrendTextClasses["POSITIVE"] = "text-danger";
    TrendTextClasses["NEGATIVE"] = "text-success";
})(TrendTextClasses || (TrendTextClasses = {}));
export var TrendArrowClasses;
(function (TrendArrowClasses) {
    TrendArrowClasses["POSITIVE"] = "arrow-upward";
    TrendArrowClasses["NEGATIVE"] = "arrow-downward";
})(TrendArrowClasses || (TrendArrowClasses = {}));
export const PERIODS = [
    {
        interval: 'MONTH',
        duration: 1,
        default: true,
    },
    {
        interval: 'MONTH',
        duration: 6,
    },
    {
        interval: 'MONTH',
        duration: 12,
    },
];
export const DB_DATE_FORMAT = 'yyyy-MM-dd';
