// @backbase/income-spending-analysis-widget-ang/esm2015/src/income-spending-analysis.service.js
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/transactions";
export class IncomeSpendingAnalysisWidgetService {
    constructor(categoryTotalsHttpService, categoryPeriodTotalsHttpService) {
        this.categoryTotalsHttpService = categoryTotalsHttpService;
        this.categoryPeriodTotalsHttpService = categoryPeriodTotalsHttpService;
    }
    loadAnalysisData(params) {
        return this.categoryTotalsHttpService.getCategorytotals(params);
    }
    loadAnalysisPeriodData(params) {
        return this.categoryPeriodTotalsHttpService.getCategoryPeriodTotals(params);
    }
}
IncomeSpendingAnalysisWidgetService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IncomeSpendingAnalysisWidgetService, deps: [{ token: i1.CategoryTotalsHttpService }, { token: i1.CategoryPeriodTotalsHttpService }], target: i0.ɵɵFactoryTarget.Injectable });
IncomeSpendingAnalysisWidgetService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IncomeSpendingAnalysisWidgetService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IncomeSpendingAnalysisWidgetService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.CategoryTotalsHttpService }, { type: i1.CategoryPeriodTotalsHttpService }]; } });
