// @backbase/income-spending-analysis-widget-ang/esm2015/src/income-spending-analysis.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { AmountModule } from '@backbase/ui-ang/amount';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { IconModule } from '@backbase/ui-ang/icon';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { PeriodSelectorModule } from '@backbase/ui-ang/period-selector';
import { DonutChartModule } from './bb-donut-chart-ui/donut-chart.module';
import { MonthSelectorComponent } from './bb-month-selector-ui/month-selector.component';
import { IncomeSpendingAnalysisBarChartWrapperCustomizableDirective, IncomeSpendingAnalysisCategoriesListCustomizableDirective, IncomeSpendingAnalysisChartCustomizableDirective, IncomeSpendingAnalysisChartWrapperCustomizableDirective, IncomeSpendingAnalysisCustomizableDirective, IncomeSpendingAnalysisMonthSelectorCustomizableDirective, IncomeSpendingAnalysisPeriodSelectorCustomizableDirective, IncomeSpendingAnalysisWidgetComponent, } from './income-spending-analysis.component';
import { IncomeSpendingAnalysisWidgetService } from './income-spending-analysis.service';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const components = [IncomeSpendingAnalysisWidgetComponent, MonthSelectorComponent];
const customizables = [
    IncomeSpendingAnalysisCustomizableDirective,
    IncomeSpendingAnalysisPeriodSelectorCustomizableDirective,
    IncomeSpendingAnalysisMonthSelectorCustomizableDirective,
    IncomeSpendingAnalysisChartWrapperCustomizableDirective,
    IncomeSpendingAnalysisCategoriesListCustomizableDirective,
    IncomeSpendingAnalysisChartCustomizableDirective,
    IncomeSpendingAnalysisBarChartWrapperCustomizableDirective,
];
const modules = [
    AmountModule,
    DonutChartModule,
    EmptyStateModule,
    IconModule,
    LoadingIndicatorModule,
    PeriodSelectorModule,
];
export class IncomeSpendingAnalysisWidgetModule {
}
IncomeSpendingAnalysisWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IncomeSpendingAnalysisWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IncomeSpendingAnalysisWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IncomeSpendingAnalysisWidgetModule, declarations: [IncomeSpendingAnalysisWidgetComponent, MonthSelectorComponent, IncomeSpendingAnalysisCustomizableDirective,
        IncomeSpendingAnalysisPeriodSelectorCustomizableDirective,
        IncomeSpendingAnalysisMonthSelectorCustomizableDirective,
        IncomeSpendingAnalysisChartWrapperCustomizableDirective,
        IncomeSpendingAnalysisCategoriesListCustomizableDirective,
        IncomeSpendingAnalysisChartCustomizableDirective,
        IncomeSpendingAnalysisBarChartWrapperCustomizableDirective], imports: [CommonModule, i1.BackbaseCoreModule, AmountModule,
        DonutChartModule,
        EmptyStateModule,
        IconModule,
        LoadingIndicatorModule,
        PeriodSelectorModule], exports: [IncomeSpendingAnalysisWidgetComponent, MonthSelectorComponent, IncomeSpendingAnalysisCustomizableDirective,
        IncomeSpendingAnalysisPeriodSelectorCustomizableDirective,
        IncomeSpendingAnalysisMonthSelectorCustomizableDirective,
        IncomeSpendingAnalysisChartWrapperCustomizableDirective,
        IncomeSpendingAnalysisCategoriesListCustomizableDirective,
        IncomeSpendingAnalysisChartCustomizableDirective,
        IncomeSpendingAnalysisBarChartWrapperCustomizableDirective, AmountModule,
        DonutChartModule,
        EmptyStateModule,
        IconModule,
        LoadingIndicatorModule,
        PeriodSelectorModule] });
IncomeSpendingAnalysisWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IncomeSpendingAnalysisWidgetModule, providers: [IncomeSpendingAnalysisWidgetService], imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { IncomeSpendingAnalysisWidgetComponent },
            }),
            ...modules,
        ], AmountModule,
        DonutChartModule,
        EmptyStateModule,
        IconModule,
        LoadingIndicatorModule,
        PeriodSelectorModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IncomeSpendingAnalysisWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { IncomeSpendingAnalysisWidgetComponent },
                        }),
                        ...modules,
                    ],
                    declarations: [...components, ...customizables],
                    providers: [IncomeSpendingAnalysisWidgetService],
                    exports: [...components, ...customizables, ...modules],
                }]
        }] });
