// @backbase/quick-transfer-journey-ang/esm2015/lib/animations.js
import { animate, animation, style, transition, trigger, useAnimation } from '@angular/animations';
const timing = '200ms ease-in';
const distances = {
    normalX: 'translateX(0%)',
    right: 'translateX(100%)',
    left: 'translateX(-100%)',
    normalY: 'translateY(0%)',
    bottom: 'translateY(100%)',
};
const slideAnimation = animation([
    style({
        position: 'absolute',
        height: '100%',
        width: '100%',
        transform: '{{from}}',
        display: 'block',
        top: 0,
    }),
    animate(timing, style({ transform: '{{to}}' })),
]);
export function slide2D(from, to) {
    return [
        transition(':enter', [
            useAnimation(slideAnimation, {
                params: { from, to },
            }),
        ]),
        transition(':leave', [
            useAnimation(slideAnimation, {
                params: { from: to, to: from },
            }),
        ]),
    ];
}
// used to animate account selector view
export const slideLeftAnimation = trigger('slideLeft', slide2D(distances.right, distances.normalX));
// used to animate the main view "form view"
export const slideRightAnimation = trigger('slideRight', slide2D(distances.left, distances.normalX));
// used for transition of the success/fail screen
export const slideUpAnimation = trigger('slideUp', slide2D(distances.bottom, distances.normalY));
