// @backbase/quick-transfer-journey-ang/esm2015/lib/quick-transfer-journey.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { AmountModule } from '@backbase/ui-ang/amount';
import { AmountInputModule } from '@backbase/ui-ang/amount-input';
import { ButtonModule } from '@backbase/ui-ang/button';
import { CurrencyInputModule } from '@backbase/ui-ang/currency-input';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { HeaderModule } from '@backbase/ui-ang/header';
import { IconModule } from '@backbase/ui-ang/icon';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { ModalModule } from '@backbase/ui-ang/modal';
import { ProductItemBasicAccountModule } from '@backbase/ui-ang/product-item-basic-account';
import { TabModule } from '@backbase/ui-ang/tab';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { BaseAccountSelectorComponent } from './components/account-selector-base/account-selector-base.component';
import { QuickTransferAccountsComponent } from './components/quick-transfer-accounts/quick-transfer-accounts.component';
import { QuickTransferCompleteComponent } from './components/quick-transfer-complete/quick-transfer-complete.component';
import { QuickTransferConfirmationModalComponent } from './components/quick-transfer-confirmation-modal/quick-transfer-confirmation-modal.component';
import { QuickTransferFormComponent } from './components/quick-transfer-form/quick-transfer-form.component';
import { QuickTransferFromAccountComponent } from './components/quick-transfer-from-account/quick-transfer-from-account.component';
import { QuickTransferToAccountComponent } from './components/quick-transfer-to-account/quick-transfer-to-account.component';
import { QuickTransferJourneyComponent } from './quick-transfer-journey.component';
import { QuickTransferJourneyConfigService } from './services/config.service';
import { QuickTransferHttpService } from './services/quick-transfer-http.service';
import { QuickTransferStoreModule } from './store/store.module';
import { CustomProductSelectorComponent } from './ui/custom-product-selector/custom-product-selector.component';
import { QuickTransferValidationMessageComponent } from './ui/validation-message/validation-message.component';
import * as i0 from "@angular/core";
const uiModules = [
    InputValidationMessageModule,
    ButtonModule,
    HeaderModule,
    IconModule,
    CurrencyInputModule,
    TooltipModule,
    ProductItemBasicAccountModule,
    TabModule,
    LoadingIndicatorModule,
    ModalModule,
    LoadButtonModule,
    EmptyStateModule,
    AmountModule,
    AmountInputModule,
];
const components = [
    QuickTransferFormComponent,
    BaseAccountSelectorComponent,
    QuickTransferFromAccountComponent,
    QuickTransferAccountsComponent,
    QuickTransferToAccountComponent,
    QuickTransferConfirmationModalComponent,
    QuickTransferCompleteComponent,
    CustomProductSelectorComponent,
    QuickTransferJourneyComponent,
    QuickTransferValidationMessageComponent,
];
export class QuickTransferJourneyModule {
}
/** @nocollapse */ QuickTransferJourneyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: QuickTransferJourneyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ QuickTransferJourneyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: QuickTransferJourneyModule, declarations: [QuickTransferFormComponent,
        BaseAccountSelectorComponent,
        QuickTransferFromAccountComponent,
        QuickTransferAccountsComponent,
        QuickTransferToAccountComponent,
        QuickTransferConfirmationModalComponent,
        QuickTransferCompleteComponent,
        CustomProductSelectorComponent,
        QuickTransferJourneyComponent,
        QuickTransferValidationMessageComponent], imports: [CommonModule, ReactiveFormsModule, QuickTransferStoreModule, InputValidationMessageModule,
        ButtonModule,
        HeaderModule,
        IconModule,
        CurrencyInputModule,
        TooltipModule,
        ProductItemBasicAccountModule,
        TabModule,
        LoadingIndicatorModule,
        ModalModule,
        LoadButtonModule,
        EmptyStateModule,
        AmountModule,
        AmountInputModule], exports: [QuickTransferFormComponent,
        BaseAccountSelectorComponent,
        QuickTransferFromAccountComponent,
        QuickTransferAccountsComponent,
        QuickTransferToAccountComponent,
        QuickTransferConfirmationModalComponent,
        QuickTransferCompleteComponent,
        CustomProductSelectorComponent,
        QuickTransferJourneyComponent,
        QuickTransferValidationMessageComponent, QuickTransferStoreModule, InputValidationMessageModule,
        ButtonModule,
        HeaderModule,
        IconModule,
        CurrencyInputModule,
        TooltipModule,
        ProductItemBasicAccountModule,
        TabModule,
        LoadingIndicatorModule,
        ModalModule,
        LoadButtonModule,
        EmptyStateModule,
        AmountModule,
        AmountInputModule] });
/** @nocollapse */ QuickTransferJourneyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: QuickTransferJourneyModule, providers: [QuickTransferHttpService, QuickTransferJourneyConfigService], imports: [[CommonModule, ReactiveFormsModule, QuickTransferStoreModule, ...uiModules], QuickTransferStoreModule, InputValidationMessageModule,
        ButtonModule,
        HeaderModule,
        IconModule,
        CurrencyInputModule,
        TooltipModule,
        ProductItemBasicAccountModule,
        TabModule,
        LoadingIndicatorModule,
        ModalModule,
        LoadButtonModule,
        EmptyStateModule,
        AmountModule,
        AmountInputModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: QuickTransferJourneyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components],
                    imports: [CommonModule, ReactiveFormsModule, QuickTransferStoreModule, ...uiModules],
                    exports: [...components, QuickTransferStoreModule, ...uiModules],
                    providers: [QuickTransferHttpService, QuickTransferJourneyConfigService],
                }]
        }] });
