// @backbase/user-manage-profile-widget-ang/esm2015/src/user-profile-forms/user-profile-form-inputs/email-address-inputs/user-profile-form-email-address-inputs.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { DropdownSingleSelectModule } from '@backbase/ui-ang/dropdown-single-select';
import { InputEmailModule } from '@backbase/ui-ang/input-email';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { UserEmailAddressInputComponent, UserEmailAddressInputCustomizableDirective, UserEmailAddressInputErrorsCustomizableDirective, } from './user-email-address-input.component';
import { UserEmailAddressTypeErrorsCustomizableDirective, UserEmailAddressTypeInputComponent, UserEmailAddressTypeInputCustomizableDirective, } from './user-email-address-type-input.component';
import * as i0 from "@angular/core";
const uiModules = [DropdownSingleSelectModule, InputEmailModule, InputValidationMessageModule];
export class UserProfileFormEmailAddressInputsModule {
}
UserProfileFormEmailAddressInputsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormEmailAddressInputsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
UserProfileFormEmailAddressInputsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormEmailAddressInputsModule, declarations: [UserEmailAddressTypeInputComponent,
        UserEmailAddressTypeInputCustomizableDirective,
        UserEmailAddressTypeErrorsCustomizableDirective,
        UserEmailAddressInputComponent,
        UserEmailAddressInputCustomizableDirective,
        UserEmailAddressInputErrorsCustomizableDirective], imports: [CommonModule, ReactiveFormsModule, DropdownSingleSelectModule, InputEmailModule, InputValidationMessageModule], exports: [UserEmailAddressTypeInputComponent,
        UserEmailAddressTypeInputCustomizableDirective,
        UserEmailAddressTypeErrorsCustomizableDirective,
        UserEmailAddressInputComponent,
        UserEmailAddressInputCustomizableDirective,
        UserEmailAddressInputErrorsCustomizableDirective, DropdownSingleSelectModule, InputEmailModule, InputValidationMessageModule] });
UserProfileFormEmailAddressInputsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormEmailAddressInputsModule, imports: [[CommonModule, ReactiveFormsModule, ...uiModules], DropdownSingleSelectModule, InputEmailModule, InputValidationMessageModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormEmailAddressInputsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ReactiveFormsModule, ...uiModules],
                    declarations: [
                        UserEmailAddressTypeInputComponent,
                        UserEmailAddressTypeInputCustomizableDirective,
                        UserEmailAddressTypeErrorsCustomizableDirective,
                        UserEmailAddressInputComponent,
                        UserEmailAddressInputCustomizableDirective,
                        UserEmailAddressInputErrorsCustomizableDirective,
                    ],
                    exports: [
                        UserEmailAddressTypeInputComponent,
                        UserEmailAddressTypeInputCustomizableDirective,
                        UserEmailAddressTypeErrorsCustomizableDirective,
                        UserEmailAddressInputComponent,
                        UserEmailAddressInputCustomizableDirective,
                        UserEmailAddressInputErrorsCustomizableDirective,
                        ...uiModules,
                    ],
                }]
        }] });
