// @backbase/ui-ang/esm2015/empty-state/empty-state.component.js
import { Component, Input, ChangeDetectionStrategy } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/icon";
import * as i2 from "@angular/common";
/**
 * @name EmptyStateComponent
 *
 * @description
 * Component that displays an empty state.
 */
export class EmptyStateComponent {
    constructor() {
        /**
         * Whether the icon should be shown. Defaults to true
         */
        this.showIcon = true;
        /**
         * The modifier for the icon used in the component. Defaults to 'list'
         */
        this.iconModifier = 'list';
        /**
         * The size for the icon used in the component. Defaults to 'xl'
         */
        this.iconSize = 'xl';
    }
    ngOnInit() {
        if (typeof this.title === 'undefined') {
            throw new Error(`"title" input is required in "${this.constructor.name}"`);
        }
    }
}
EmptyStateComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EmptyStateComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
EmptyStateComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: EmptyStateComponent, selector: "bb-empty-state-ui", inputs: { title: "title", subtitle: "subtitle", showIcon: "showIcon", iconClasses: "iconClasses", iconModifier: "iconModifier", iconSize: "iconSize", iconColor: "iconColor" }, ngImport: i0, template: "<div class=\"bb-empty-state\">\n  <bb-icon-ui\n    class=\"bb-empty-state__icon\"\n    *ngIf=\"showIcon\"\n    [ngClass]=\"iconClasses\"\n    [name]=\"iconModifier\"\n    [size]=\"iconSize\"\n    [color]=\"iconColor\"\n    [attr.data-role]=\"'icon'\"\n  ></bb-icon-ui>\n  <p class=\"bb-empty-state__title\" data-role=\"empty-state-title\">\n    {{ title }}\n  </p>\n  <p *ngIf=\"subtitle\" class=\"bb-empty-state__message\" data-role=\"empty-state-message\" [innerHTML]=\"subtitle\"></p>\n</div>\n", components: [{ type: i1.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i2.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EmptyStateComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-empty-state-ui',
                    templateUrl: './empty-state.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { title: [{
                type: Input
            }], subtitle: [{
                type: Input
            }], showIcon: [{
                type: Input
            }], iconClasses: [{
                type: Input
            }], iconModifier: [{
                type: Input
            }], iconSize: [{
                type: Input
            }], iconColor: [{
                type: Input
            }] } });
