// @backbase/ui-ang/esm2015/empty-state/empty-state.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { IconModule } from '@backbase/ui-ang/icon';
import { EmptyStateComponent } from './empty-state.component';
import * as i0 from "@angular/core";
export class EmptyStateModule {
}
EmptyStateModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EmptyStateModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
EmptyStateModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EmptyStateModule, declarations: [EmptyStateComponent], imports: [CommonModule, IconModule], exports: [EmptyStateComponent] });
EmptyStateModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EmptyStateModule, imports: [[CommonModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EmptyStateModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, IconModule],
                    declarations: [EmptyStateComponent],
                    exports: [EmptyStateComponent],
                }]
        }] });
