// @backbase/product-summary-common-ang/esm2015/src/model/constants.js
import { AccountNavigationType } from './types';
//TODO: check if this can be like id
export const defaultArrangementName = '';
export const WidgetProperty = {
    BUSINESS_FUNCTION: 'businessFunction',
    RESOURCE_NAME: 'resourceName',
    PRIVILEGE_TYPE: 'privilege',
    SIZE: 'size',
    ACTIVE_VIEW: 'activeView',
};
export const SMALL_SCREEN_SIZE = 640;
export const EXTRA_SMALL_SCREEN_SIZE = 576;
export const DEFAULT_DEBOUNCE_TIME = 300;
export var DefaultProductKindNameList;
(function (DefaultProductKindNameList) {
    DefaultProductKindNameList["CURRENT_ACCOUNT"] = "current-account";
    DefaultProductKindNameList["SAVINGS_ACCOUNT"] = "savings-account";
    DefaultProductKindNameList["TERM_DEPOSIT"] = "term-deposit";
    DefaultProductKindNameList["LOAN"] = "loan";
    DefaultProductKindNameList["CREDIT_CARD"] = "credit-card";
    DefaultProductKindNameList["DEBIT_CARD"] = "debit-card";
    DefaultProductKindNameList["INVESTMENT_ACCOUNT"] = "investment-account";
})(DefaultProductKindNameList || (DefaultProductKindNameList = {}));
export const DefaultProductKindName = DefaultProductKindNameList.CURRENT_ACCOUNT;
export const CardProductKindName = ['credit card', 'debit card'];
/**
 * Default value for allowed navigation types per account,
 * by default all navigation types are allowed
 */
export const DefaultAllowedNavigationTypes = [
    AccountNavigationType.BALANCE_HISTORY,
    AccountNavigationType.PAYMENTS,
    AccountNavigationType.TRANSACTIONS,
    AccountNavigationType.STATEMENTS,
];
