// @backbase/product-summary-common-ang/esm2015/src/model/currency-to-country-map.js
export const CurrencyToCountryMap = {
    BDT: 'BD',
    EUR: 'EU',
    XOF: 'TG',
    BGN: 'BG',
    BAM: 'BA',
    BBD: 'BB',
    XPF: 'NC',
    BMD: 'BM',
    BND: 'BN',
    BOB: 'BO',
    BHD: 'BH',
    BIF: 'BI',
    BTN: 'BT',
    JMD: 'JM',
    NOK: 'NO',
    BWP: 'BW',
    WST: 'WS',
    USD: 'US',
    BRL: 'BR',
    BSD: 'BS',
    GBP: 'GB',
    BYR: 'BY',
    BZD: 'BZ',
    RUB: 'RU',
    RWF: 'RW',
    RSD: 'RS',
    TMT: 'TM',
    TJS: 'TJ',
    RON: 'RO',
    NZD: 'CK',
    GTQ: 'GT',
    XAF: 'TD',
    JPY: 'JP',
    GYD: 'GY',
    GEL: 'GE',
    XCD: 'AI',
    GNF: 'GN',
    GMD: 'GM',
    DKK: 'DK',
    GIP: 'GI',
    GHS: 'GH',
    OMR: 'OM',
    TND: 'TN',
    JOD: 'JO',
    HRK: 'HR',
    HTG: 'HT',
    HUF: 'HU',
    HKD: 'HK',
    HNL: 'HN',
    AUD: 'AU',
    VEF: 'VE',
    ILS: 'IL',
    PYG: 'PY',
    IQD: 'IQ',
    PAB: 'PA',
    PGK: 'PG',
    PEN: 'PE',
    PKR: 'PK',
    PHP: 'PH',
    PLN: 'PL',
    ZMK: 'ZM',
    MAD: 'MA',
    EGP: 'EG',
    ZAR: 'ZA',
    VND: 'VN',
    SBD: 'SB',
    ETB: 'ET',
    SOS: 'SO',
    ZWL: 'ZW',
    SAR: 'SA',
    ERN: 'ER',
    MDL: 'MD',
    MGA: 'MG',
    UZS: 'UZ',
    MMK: 'MM',
    MOP: 'MO',
    MNT: 'MN',
    MKD: 'MK',
    MUR: 'MU',
    MWK: 'MW',
    MVR: 'MV',
    MRO: 'MR',
    UGX: 'UG',
    TZS: 'TZ',
    MYR: 'MY',
    MXN: 'MX',
    SHP: 'SH',
    FJD: 'FJ',
    FKP: 'FK',
    NIO: 'NI',
    NAD: 'NA',
    VUV: 'VU',
    NGN: 'NG',
    NPR: 'NP',
    CHF: 'LI',
    COP: 'CO',
    CNY: 'CN',
    CLP: 'CL',
    CAD: 'CA',
    CDF: 'CD',
    CZK: 'CZ',
    CRC: 'CR',
    ANG: 'SX',
    CVE: 'CV',
    CUP: 'CU',
    SZL: 'SZ',
    SYP: 'SY',
    KGS: 'KG',
    KES: 'KE',
    SSP: 'SS',
    SRD: 'SR',
    KHR: 'KH',
    KMF: 'KM',
    STD: 'ST',
    KRW: 'KR',
    KPW: 'KP',
    KWD: 'KW',
    SLL: 'SL',
    SCR: 'SC',
    KZT: 'KZ',
    KYD: 'KY',
    SGD: 'SG',
    SEK: 'SE',
    SDG: 'SD',
    DOP: 'DO',
    DJF: 'DJ',
    YER: 'YE',
    DZD: 'DZ',
    UYU: 'UY',
    LBP: 'LB',
    LAK: 'LA',
    TWD: 'TW',
    TTD: 'TT',
    TRY: 'TR',
    LKR: 'LK',
    TOP: 'TO',
    LTL: 'LT',
    LRD: 'LR',
    LSL: 'LS',
    THB: 'TH',
    LYD: 'LY',
    AED: 'AE',
    AFN: 'AF',
    ISK: 'IS',
    IRR: 'IR',
    AMD: 'AM',
    ALL: 'AL',
    AOA: 'AO',
    ARS: 'AR',
    AWG: 'AW',
    INR: 'IN',
    AZN: 'AZ',
    IDR: 'ID',
    UAH: 'UA',
    QAR: 'QA',
    MZN: 'MZ',
};
