// @backbase/product-summary-common-ang/esm2015/src/model/test-mock.js
import { defaultArrangementName } from './constants';
import { PaginationType } from './types';
export const balanceHistorySeriesMock = {
    items: [
        {
            arrangementId: '3cdb2224-8926-4b4d-a99f-1c9dfbbb4699',
            balanceHistory: [
                {
                    dateFrom: '2017-05-10T00:00:00.000Z',
                    dateTo: '2017-06-01T00:00:00.000Z',
                    value: {
                        currency: 'EUR',
                        balance: '12',
                    },
                    valuePtc: '12',
                },
                {
                    dateFrom: '2017-05-11T00:00:00Z',
                    dateTo: '2017-06-01T00:00:00Z',
                    value: {
                        currency: 'EUR',
                        balance: '22',
                    },
                    valuePtc: '22',
                },
                {
                    dateFrom: '2017-05-23T00:00:00+0000',
                    dateTo: '2017-06-01T00:00:00+0000',
                    value: {
                        currency: 'EUR',
                        balance: '32',
                    },
                    valuePtc: '32',
                },
                {
                    dateFrom: '2017-06-01T00:00:00.000Z',
                    dateTo: '2017-06-01T00:00:00.000Z',
                    value: {
                        currency: 'EUR',
                        balance: '54',
                    },
                    valuePtc: '54',
                },
                {
                    dateFrom: '2017-06-02T00:00:00.000Z',
                    dateTo: '2017-06-01T00:00:00.000Z',
                    value: {
                        currency: 'EUR',
                        balance: '34',
                    },
                    valuePtc: '34',
                },
                {
                    dateFrom: '2017-06-12T00:00:00.000Z',
                    dateTo: '2017-06-01T00:00:00.000Z',
                    value: {
                        currency: 'EUR',
                        balance: '4',
                    },
                    valuePtc: '4',
                },
                {
                    dateFrom: '2017-06-18T00:00:00.000Z',
                    dateTo: '2017-06-01T00:00:00.000Z',
                    value: {
                        currency: 'EUR',
                        balance: '-14',
                    },
                    valuePtc: '-14',
                },
            ],
        },
    ],
};
export const balanceHistoryPaginatedMock = [
    {
        dateFrom: '2017-05-10T00:00:00.000Z',
        dateTo: '2017-06-01T00:00:00.000Z',
        value: {
            currency: 'EUR',
            balance: '12',
        },
        valuePtc: '12',
    },
    {
        dateFrom: '2017-05-11T00:00:00.000Z',
        dateTo: '2017-06-01T00:00:00.000Z',
        value: {
            currency: 'EUR',
            balance: '22',
        },
        valuePtc: '22',
    },
    {
        dateFrom: '2017-05-23T00:00:00.000Z',
        dateTo: '2017-06-01T00:00:00.000Z',
        value: {
            currency: 'EUR',
            balance: '32',
        },
        valuePtc: '32',
    },
    {
        dateFrom: '2017-06-01T00:00:00.000Z',
        dateTo: '2017-06-01T00:00:00.000Z',
        value: {
            currency: 'EUR',
            balance: '54',
        },
        valuePtc: '54',
    },
    {
        dateFrom: '2017-06-02T00:00:00.000Z',
        dateTo: '2017-07-01T00:00:00.000Z',
        value: {
            currency: 'EUR',
            balance: '34',
        },
        valuePtc: '34',
    },
    {
        dateFrom: '2017-06-12T00:00:00.000Z',
        dateTo: '2017-07-01T00:00:00.000Z',
        value: {
            currency: 'EUR',
            balance: '4',
        },
        valuePtc: '4',
    },
    {
        dateFrom: '2017-06-18T00:00:00.000Z',
        dateTo: '2017-07-01T00:00:00.000Z',
        value: {
            currency: 'EUR',
            balance: '-14',
        },
        valuePtc: '-14',
    },
];
export const balanceHistoryPaginatedMockWithCount = {
    count: 3,
    items: balanceHistoryPaginatedMock,
    params: { from: 1, paginationType: 'load-more' },
};
export const emptyBalanceHistoryPaginatedMockWithCount = {
    count: 0,
    items: [],
    params: { from: 0, paginationType: 'pagination' },
};
export const requestObjectMockSeries = {
    arrangementIds: '1',
    params: {
        dateFrom: '2017-06-18',
        dateTo: '2017-06-18',
        timePeriod: '',
        format: '',
        paginationType: 'pagination',
        from: 0,
    },
};
export const requestObjectMockPaginated = {
    arrangementId: '1',
    params: {
        dateFrom: '2017-06-18',
        dateTo: '2017-06-18',
        timePeriod: '',
        format: '',
        paginationType: 'pagination',
        from: 0,
    },
};
export const processedBalanceHistoryMock = [
    {
        name: defaultArrangementName,
        series: [
            {
                name: new Date('2017-06-01T00:00:00.000Z'),
                value: '12',
            },
            {
                name: new Date('2017-06-01T00:00:00Z'),
                value: '22',
            },
            {
                name: new Date('2017-06-01T00:00:00+0000'),
                value: '32',
            },
            {
                name: new Date('2017-06-01T00:00:00.000Z'),
                value: '54',
            },
            {
                name: new Date('2017-06-01T00:00:00.000Z'),
                value: '34',
            },
            {
                name: new Date('2017-06-01T00:00:00.000Z'),
                value: '4',
            },
            {
                name: new Date('2017-06-01T00:00:00.000Z'),
                value: 0,
            },
        ],
    },
    {
        name: defaultArrangementName,
        series: [
            {
                name: new Date('2017-06-01T00:00:00.000Z'),
                value: 0,
            },
            {
                name: new Date('2017-06-01T00:00:00.000Z'),
                value: '-14',
            },
        ],
    },
];
export const mockRequestObject = {
    businessFunction: '',
    resourceName: '',
    privilege: '',
    paginationType: PaginationType.LOAD_MORE,
};
export const mockAccountsResponse = [
    {
        id: '1cdb2224-8926-4b4d-a99f-1c9dfbbb4699',
        externalArrangementId: 'kjh4567-asd1-11e7-b114-b2f933d50000',
        externalLegalEntityId: 'COOOOO1',
        externalProductId: '1234567',
        name: 'Mr and Mrs J. Smith',
        productKindName: 'Current Account',
        bankAlias: 'Secret account',
        bookedBalance: 100.1,
        availableBalance: 100.2,
        creditLimit: 100.3,
        IBAN: 'GB29NWBK60161331926819',
        BBAN: 'BBAN',
        currency: 'EUR',
        externalTransferAllowed: true,
        urgentTransferAllowed: false,
        accruedInterest: 2.2,
        principalAmount: 100.4,
        currentInvestmentValue: 100.5,
        legalEntityIds: ['257da57a-11e4-4553-9175-54baf755069b', 'cd83683b-13f2-43d8-882b-39c9ab27d499'],
        productId: '36c8fc42-ec97-4f83-8a7c-d622625007f3',
        productTypeName: 'Current Account',
        BIC: 'BICExample1',
        bankBranchCode: 'bankBranchCode1',
        accountOpeningDate: '2016-01-28T16:41:41.090Z',
        accountInterestRate: 100.2,
        valueDateBalance: 100.1,
        creditLimitUsage: 100.3,
        creditLimitInterestRate: 100.4,
        creditLimitExpiryDate: '2019-09-28T16:41:41.090Z',
        debitCards: [
            {
                number: '4578',
                expiryDate: '2018-08-22T16:41:41.090Z',
            },
            {
                number: '3241',
                expiryDate: '2019-01-14T16:41:41.090Z',
            },
        ],
        startDate: '2016-02-28T16:41:41.090Z',
        termUnit: 'Y',
        termNumber: 50,
        maturityDate: '2017-02-28T16:41:41.090Z',
        maturityAmount: 99.5,
        autoRenewalIndicator: true,
        interestPaymentFrequencyUnit: 'M',
        interestPaymentFrequencyNumber: 15,
        interestSettlementAccount: 'interestSettlementAccount1',
        outstandingPrincipalAmount: 100.2,
        monthlyInstalmentAmount: 100.1,
        amountInArrear: 100.3,
        minimumRequiredBalance: 80.4,
        creditCardAccountNumber: '123456',
        validThru: '2019-02-28T16:41:41.090Z',
        applicableInterestRate: 101.2,
        remainingCredit: 50,
        outstandingPayment: 105.5,
        minimumPayment: 51.1,
        minimumPaymentDueDate: '2018-02-28T16:41:41.090Z',
        totalInvestmentValue: 110.2,
        accountHolderAddressLine1: 'accountHolderAddressLine11',
        accountHolderAddressLine2: 'accountHolderAddressLine12',
        creditAccount: true,
        debitAccount: true,
        visible: true,
        lastUpdateDate: '2019-09-28T16:41:41.090Z',
    },
    {
        id: '2222222-8926-4b4d-a99f-1c9dfbbb4610',
        externalArrangementId: 'kjh4567-asd1-11e7-b114-b2f933d50000',
        externalLegalEntityId: 'COOOOO1',
        externalProductId: '1234567',
        name: 'Mr and Mrs J. Smith second one',
        productKindName: 'Current Account',
        bankAlias: 'Secret account Mr and Mrs J. Smith second one',
        bookedBalance: 100.1,
        availableBalance: 100.2,
        creditLimit: 100.3,
        IBAN: 'GB29NWBK60161331926819',
        BBAN: 'BBAN',
        currency: 'EUR',
        externalTransferAllowed: true,
        urgentTransferAllowed: false,
        accruedInterest: 2.2,
        principalAmount: 100.4,
        currentInvestmentValue: 100.5,
        legalEntityIds: ['257da57a-11e4-4553-9175-54baf755069b', 'cd83683b-13f2-43d8-882b-39c9ab27d499'],
        productId: '36c8fc42-ec97-4f83-8a7c-d622625007f3',
        productTypeName: 'Current Account',
        BIC: 'BICExample2',
        bankBranchCode: 'bankBranchCode1',
        accountOpeningDate: '2016-01-28T16:41:41.090Z',
        accountInterestRate: 100.2,
        valueDateBalance: 100.1,
        creditLimitUsage: 100.3,
        creditLimitInterestRate: 100.4,
        creditLimitExpiryDate: '2019-09-28T16:41:41.090Z',
        debitCards: [
            {
                number: '6378',
                expiryDate: '2018-02-12T16:41:41.090Z',
            },
            {
                number: '5111',
                expiryDate: '2019-03-24T16:41:41.090Z',
            },
        ],
        startDate: '2016-02-28T16:41:41.090Z',
        termUnit: 'Y',
        termNumber: 50,
        maturityDate: '2017-02-28T16:41:41.090Z',
        maturityAmount: 99.5,
        autoRenewalIndicator: true,
        interestPaymentFrequencyUnit: 'M',
        interestPaymentFrequencyNumber: 15,
        interestSettlementAccount: 'interestSettlementAccount1',
        outstandingPrincipalAmount: 100.2,
        monthlyInstalmentAmount: 100.1,
        amountInArrear: 100.3,
        minimumRequiredBalance: 80.4,
        creditCardAccountNumber: '123456',
        validThru: '2019-02-28T16:41:41.090Z',
        applicableInterestRate: 101.2,
        remainingCredit: 50,
        outstandingPayment: 105.5,
        minimumPayment: 51.1,
        minimumPaymentDueDate: '2018-02-28T16:41:41.090Z',
        totalInvestmentValue: 110.2,
        accountHolderAddressLine1: 'accountHolderAddressLine11',
        accountHolderAddressLine2: 'accountHolderAddressLine12',
        creditAccount: true,
        debitAccount: true,
        visible: true,
        lastUpdateDate: '2019-09-28T16:41:41.090Z',
        userPreferences: {
            visible: true,
            alias: 'nice',
            favorite: true,
        },
    },
    {
        id: '1cdb2224-8926-4b4d-a99f-1c9dfbbb4699',
        externalArrangementId: 'kjh4567-asd1-11e7-b114-b2f933d50000',
        externalLegalEntityId: 'COOOOO1',
        externalProductId: '1234567',
        name: 'Mr and Mrs J. Smith',
        productKindName: 'Current Account',
        bankAlias: 'Secret account',
        bookedBalance: 100.1,
        availableBalance: 100.2,
        creditLimit: 100.3,
        IBAN: 'GB29NWBK60161331926819',
        BBAN: 'BBAN',
        currency: 'EUR',
        externalTransferAllowed: true,
        urgentTransferAllowed: false,
        accruedInterest: 2.2,
        principalAmount: 100.4,
        currentInvestmentValue: 100.5,
        legalEntityIds: ['257da57a-11e4-4553-9175-54baf755069b', 'cd83683b-13f2-43d8-882b-39c9ab27d499'],
        productId: '36c8fc42-ec97-4f83-8a7c-d622625007f3',
        productTypeName: 'Current Account',
        BIC: 'BICExample1',
        bankBranchCode: 'bankBranchCode1',
        accountOpeningDate: '2016-01-28T16:41:41.090Z',
        accountInterestRate: 100.2,
        valueDateBalance: 100.1,
        creditLimitUsage: 100.3,
        creditLimitInterestRate: 100.4,
        creditLimitExpiryDate: '2019-09-28T16:41:41.090Z',
        debitCards: [
            {
                number: '4578',
                expiryDate: '2018-08-22T16:41:41.090Z',
            },
            {
                number: '3241',
                expiryDate: '2019-01-14T16:41:41.090Z',
            },
        ],
        startDate: '2016-02-28T16:41:41.090Z',
        termUnit: 'Y',
        termNumber: 50,
        maturityDate: '2017-02-28T16:41:41.090Z',
        maturityAmount: 99.5,
        autoRenewalIndicator: true,
        interestPaymentFrequencyUnit: 'M',
        interestPaymentFrequencyNumber: 15,
        interestSettlementAccount: 'interestSettlementAccount1',
        outstandingPrincipalAmount: 100.2,
        monthlyInstalmentAmount: 100.1,
        amountInArrear: 100.3,
        minimumRequiredBalance: 80.4,
        creditCardAccountNumber: '123456',
        validThru: '2019-02-28T16:41:41.090Z',
        applicableInterestRate: 101.2,
        remainingCredit: 50,
        outstandingPayment: 105.5,
        minimumPayment: 51.1,
        minimumPaymentDueDate: '2018-02-28T16:41:41.090Z',
        totalInvestmentValue: 110.2,
        accountHolderAddressLine1: 'accountHolderAddressLine11',
        accountHolderAddressLine2: 'accountHolderAddressLine12',
        creditAccount: true,
        debitAccount: true,
        visible: true,
        lastUpdateDate: '2019-09-28T16:41:41.090Z',
        userPreferences: {
            alias: 'nice',
        },
    },
];
export const accountsWithCount = {
    count: 3,
    items: mockAccountsResponse,
    params: { from: 1, paginationType: 'load-more' },
};
export const balanceAggregationResponseMock = [
    {
        aggregatedBalances: [
            {
                currencyCode: 'EUR',
                amount: '3289.0',
                numberOfArrangements: 3,
            },
            {
                currencyCode: 'USD',
                amount: '2500.0',
                numberOfArrangements: 10,
            },
        ],
    },
];
