// @backbase/product-summary-common-ang/esm2015/src/model/types.js
export var HttpResponseType;
(function (HttpResponseType) {
    HttpResponseType["BODY"] = "body";
    HttpResponseType["RESPONSE"] = "response";
    HttpResponseType["EVENT"] = "event";
})(HttpResponseType || (HttpResponseType = {}));
export var BalanceHistoryExportFormat;
(function (BalanceHistoryExportFormat) {
    BalanceHistoryExportFormat["JSON"] = "application/json";
    BalanceHistoryExportFormat["CSV"] = "text/csv";
})(BalanceHistoryExportFormat || (BalanceHistoryExportFormat = {}));
export var PaginationType;
(function (PaginationType) {
    PaginationType["LOAD_MORE"] = "load-more";
    PaginationType["PAGINATION"] = "pagination";
})(PaginationType || (PaginationType = {}));
export var ProductKindNames;
(function (ProductKindNames) {
    ProductKindNames["CURRENT_ACCOUNT"] = "Current Account";
    ProductKindNames["SAVINGS_ACCOUNT"] = "Savings Account";
    ProductKindNames["TERM_DEPOSIT"] = "Term Deposit";
    ProductKindNames["LOAN"] = "Loan";
    ProductKindNames["CREDIT_CARD"] = "Credit Card";
    ProductKindNames["DEBIT_CARD"] = "Debit Card";
    ProductKindNames["INVESTMENT_ACCOUNT"] = "Investment Account";
})(ProductKindNames || (ProductKindNames = {}));
/**
 * Default values of "kindUri" per product kind
 */
export var ProductKindUri;
(function (ProductKindUri) {
    ProductKindUri["CREDIT_CARD"] = "credit-card";
    ProductKindUri["CURRENT_ACCOUNT"] = "current-account";
    ProductKindUri["DEBIT_CARD"] = "debit-card";
    ProductKindUri["INVESTMENT_ACCOUNT"] = "investment-account";
    ProductKindUri["LOAN"] = "loan";
    ProductKindUri["POCKET"] = "pocket";
    ProductKindUri["POCKET_PARENT"] = "pocket-parent";
    ProductKindUri["SAVINGS_ACCOUNT"] = "savings-account";
    ProductKindUri["TERM_DEPOSIT"] = "term-deposit";
})(ProductKindUri || (ProductKindUri = {}));
/**
 * Available navigation types from account
 */
export var AccountNavigationType;
(function (AccountNavigationType) {
    AccountNavigationType["PAYMENTS"] = "PAYMENTS";
    AccountNavigationType["TRANSACTIONS"] = "TRANSACTIONS";
    AccountNavigationType["BALANCE_HISTORY"] = "BALANCE_HISTORY";
    AccountNavigationType["STATEMENTS"] = "STATEMENTS";
})(AccountNavigationType || (AccountNavigationType = {}));
/**
 * Privileges applicable to product summary business functions
 */
export var ProductSummaryPrivilege;
(function (ProductSummaryPrivilege) {
    ProductSummaryPrivilege["VIEW"] = "view";
    ProductSummaryPrivilege["EDIT"] = "edit";
})(ProductSummaryPrivilege || (ProductSummaryPrivilege = {}));
/**
 * Displaying level of account aliases
 */
export var AccountAliasDisplayingLevel;
(function (AccountAliasDisplayingLevel) {
    AccountAliasDisplayingLevel["ACCOUNT"] = "ACCOUNT";
    AccountAliasDisplayingLevel["USER"] = "USER";
})(AccountAliasDisplayingLevel || (AccountAliasDisplayingLevel = {}));
/**
 * Duplicates validation settings from the `user-preferences` schema of fe-presentation-spec
 * TODO: remove these constants, when min/max length settings will be part of the generated client API
 */
export const UserPreferencesValidationSettings = {
    ALIAS_MAX_LENGTH: 50,
};
