// @backbase/payment-batch-upload-mapping-http-ang/esm2015/api/batchUploadMappings.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { PAYMENT_BATCH_UPLOAD_MAPPING_BASE_PATH } from '../variables';
import { PaymentBatchUploadMappingConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class BatchUploadMappingsService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new PaymentBatchUploadMappingConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    deleteUploadMapping(requestParameters, observe = 'body', reportProgress = false, options) {
        const _uploadMappingId = requestParameters["uploadMappingId"];
        if (_uploadMappingId === null || _uploadMappingId === undefined) {
            throw new Error('Required parameter uploadMappingId was null or undefined when calling deleteUploadMapping.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.delete(`${this.configuration.basePath}/client-api/v1/batch-upload-mappings/${encodeURIComponent(String(_uploadMappingId))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    deleteUploadMappingUrl(requestParameters) {
        const _uploadMappingId = requestParameters["uploadMappingId"];
        if (_uploadMappingId === null || _uploadMappingId === undefined) {
            throw new Error('Required parameter uploadMappingId was null or undefined when calling deleteUploadMapping.');
        }
        return `${this.configuration.basePath}/client-api/v1/batch-upload-mappings/${encodeURIComponent(String(_uploadMappingId))}`;
    }
    getUploadMapping(requestParameters, observe = 'body', reportProgress = false, options) {
        const _uploadMappingId = requestParameters["uploadMappingId"];
        if (_uploadMappingId === null || _uploadMappingId === undefined) {
            throw new Error('Required parameter uploadMappingId was null or undefined when calling getUploadMapping.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/batch-upload-mappings/${encodeURIComponent(String(_uploadMappingId))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getUploadMappingUrl(requestParameters) {
        const _uploadMappingId = requestParameters["uploadMappingId"];
        if (_uploadMappingId === null || _uploadMappingId === undefined) {
            throw new Error('Required parameter uploadMappingId was null or undefined when calling getUploadMapping.');
        }
        return `${this.configuration.basePath}/client-api/v1/batch-upload-mappings/${encodeURIComponent(String(_uploadMappingId))}`;
    }
    getUploadMappings(requestParameters, observe = 'body', reportProgress = false, options) {
        const _query = requestParameters["query"];
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        const _batchType = requestParameters["batchType"];
        const _fieldDelimiter = requestParameters["fieldDelimiter"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_query !== undefined && _query !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _query, 'query');
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_batchType !== undefined && _batchType !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _batchType, 'batchType');
        }
        if (_fieldDelimiter !== undefined && _fieldDelimiter !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _fieldDelimiter, 'fieldDelimiter');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/batch-upload-mappings`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getUploadMappingsUrl(requestParameters) {
        const _query = requestParameters["query"];
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        const _batchType = requestParameters["batchType"];
        const _fieldDelimiter = requestParameters["fieldDelimiter"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_query !== undefined && _query !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _query, 'query');
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_batchType !== undefined && _batchType !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _batchType, 'batchType');
        }
        if (_fieldDelimiter !== undefined && _fieldDelimiter !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _fieldDelimiter, 'fieldDelimiter');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v1/batch-upload-mappings${queryString ? `?${queryString}` : ''}`;
    }
    postUploadMapping(requestParameters, observe = 'body', reportProgress = false, options) {
        const _uploadMappingPostRequest = requestParameters["uploadMappingPostRequest"];
        if (_uploadMappingPostRequest === null || _uploadMappingPostRequest === undefined) {
            throw new Error('Required parameter uploadMappingPostRequest was null or undefined when calling postUploadMapping.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v1/batch-upload-mappings`, _uploadMappingPostRequest, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    postUploadMappingUrl(requestParameters) {
        const _uploadMappingPostRequest = requestParameters["uploadMappingPostRequest"];
        if (_uploadMappingPostRequest === null || _uploadMappingPostRequest === undefined) {
            throw new Error('Required parameter uploadMappingPostRequest was null or undefined when calling postUploadMapping.');
        }
        return `${this.configuration.basePath}/client-api/v1/batch-upload-mappings`;
    }
    putUploadMapping(requestParameters, observe = 'body', reportProgress = false, options) {
        const _uploadMappingId = requestParameters["uploadMappingId"];
        if (_uploadMappingId === null || _uploadMappingId === undefined) {
            throw new Error('Required parameter uploadMappingId was null or undefined when calling putUploadMapping.');
        }
        const _uploadMappingPutRequest = requestParameters["uploadMappingPutRequest"];
        if (_uploadMappingPutRequest === null || _uploadMappingPutRequest === undefined) {
            throw new Error('Required parameter uploadMappingPutRequest was null or undefined when calling putUploadMapping.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v1/batch-upload-mappings/${encodeURIComponent(String(_uploadMappingId))}`, _uploadMappingPutRequest, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    putUploadMappingUrl(requestParameters) {
        const _uploadMappingId = requestParameters["uploadMappingId"];
        if (_uploadMappingId === null || _uploadMappingId === undefined) {
            throw new Error('Required parameter uploadMappingId was null or undefined when calling putUploadMapping.');
        }
        const _uploadMappingPutRequest = requestParameters["uploadMappingPutRequest"];
        if (_uploadMappingPutRequest === null || _uploadMappingPutRequest === undefined) {
            throw new Error('Required parameter uploadMappingPutRequest was null or undefined when calling putUploadMapping.');
        }
        return `${this.configuration.basePath}/client-api/v1/batch-upload-mappings/${encodeURIComponent(String(_uploadMappingId))}`;
    }
}
BatchUploadMappingsService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BatchUploadMappingsService, deps: [{ token: i1.HttpClient }, { token: PAYMENT_BATCH_UPLOAD_MAPPING_BASE_PATH, optional: true }, { token: i2.PaymentBatchUploadMappingConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
BatchUploadMappingsService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BatchUploadMappingsService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BatchUploadMappingsService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [PAYMENT_BATCH_UPLOAD_MAPPING_BASE_PATH]
                }] }, { type: i2.PaymentBatchUploadMappingConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
