// @backbase/manage-bill-payments-journey-ang/esm2015/src/services/manage-bill-payments-navigation.service.js
import { Injectable, Optional } from '@angular/core';
import { ONE_OFF_PAYMENT, RECURRING_PAYMENT } from '@backbase/billpay-journeys-common';
import { PaymentsFilterStatus } from './manage-bill-payments-journey-config.service';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "../communication.service";
export class ManageBillPaymentsNavigationService {
    constructor(router, route, communicationService) {
        this.router = router;
        this.route = route;
        this.communicationService = communicationService;
    }
    navigateToPayment(id, isRecurring = false) {
        const paymentType = isRecurring ? RECURRING_PAYMENT : ONE_OFF_PAYMENT;
        this.router.navigate([id], { queryParams: { type: paymentType }, relativeTo: this.route });
    }
    navigateToPaymentsList(filter) {
        if (this.communicationService) {
            filter === PaymentsFilterStatus.HISTORICAL
                ? this.communicationService.navigateToPaymentsHistory()
                : this.communicationService.navigateToPaymentsList();
        }
        else {
            this.router.navigate(['./']);
        }
    }
    navigateToEditRecurringPayment(id) {
        var _a;
        (_a = this.communicationService) === null || _a === void 0 ? void 0 : _a.navigateToEditRecurringPayment(id);
    }
    navigateToEditOneOffPayment(id) {
        var _a;
        (_a = this.communicationService) === null || _a === void 0 ? void 0 : _a.navigateToEditOneOffPayment(id);
    }
}
ManageBillPaymentsNavigationService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageBillPaymentsNavigationService, deps: [{ token: i1.Router }, { token: i1.ActivatedRoute }, { token: i2.CommunicationService, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
ManageBillPaymentsNavigationService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageBillPaymentsNavigationService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageBillPaymentsNavigationService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.Router }, { type: i1.ActivatedRoute }, { type: i2.CommunicationService, decorators: [{
                    type: Optional
                }] }]; } });
