// @backbase/manage-bill-payments-journey-ang/esm2015/src/services/manage-bill-payments-data.service.js
import { Injectable } from '@angular/core';
import { PaymentsFilterStatus } from '../model';
import { ONE_OFF_PAYMENT, Direction } from '@backbase/billpay-journeys-common';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/billpay";
export class ManageBillPaymentsDataService {
    /** @internal */
    constructor(paymentsData, payeesData) {
        this.paymentsData = paymentsData;
        this.payeesData = payeesData;
    }
    getPayments(pageSize, paymentFilter) {
        const getDirection = (status) => status === PaymentsFilterStatus.PENDING ? Direction.ASC : Direction.DESC;
        return this.paymentsData.getBillPayPayments(Object.assign({ status: paymentFilter, direction: getDirection(paymentFilter) }, (pageSize && { size: pageSize })));
    }
    getPayment(id, paymentType = ONE_OFF_PAYMENT) {
        return paymentType === ONE_OFF_PAYMENT
            ? this.paymentsData.getPaymentById({ id })
            : this.paymentsData.getRecurringPaymentById({ id });
    }
    getPayee(id) {
        return this.payeesData.getPayeeById({ id });
    }
    deletePayment(id, paymentType = ONE_OFF_PAYMENT) {
        return paymentType === ONE_OFF_PAYMENT
            ? this.paymentsData.deletePaymentById({ id })
            : this.paymentsData.deleteRecurringPaymentById({ id });
    }
}
ManageBillPaymentsDataService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageBillPaymentsDataService, deps: [{ token: i1.BillPayPaymentsService }, { token: i1.BillPayPayeesService }], target: i0.ɵɵFactoryTarget.Injectable });
ManageBillPaymentsDataService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageBillPaymentsDataService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageBillPaymentsDataService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.BillPayPaymentsService }, { type: i1.BillPayPayeesService }]; } });
