// @backbase/manage-bill-payments-journey-ang/esm2015/src/services/manage-bill-payments-journey-config.service.js
import { Injectable, Optional, Inject, InjectionToken } from '@angular/core';
import '@angular/localize/init';
import * as i0 from "@angular/core";
export const ManageBillPaymentsJourneyConfigurationToken = new InjectionToken('ManageBillPaymentsJourneyJourneyConfiguration injection token');
export var PaymentsFilterStatus;
(function (PaymentsFilterStatus) {
    PaymentsFilterStatus["PENDING"] = "PENDING";
    PaymentsFilterStatus["HISTORICAL"] = "HISTORICAL";
    PaymentsFilterStatus["ALL"] = "ALL";
})(PaymentsFilterStatus || (PaymentsFilterStatus = {}));
const configDefaults = {
    notificationDismissTime: 5000,
    pageSize: 0,
    pageFilter: PaymentsFilterStatus.PENDING,
    accountNumberMasked: false,
    pageTitle: $localize `:@@manage-bill-payments.page-title:Pending payments`,
};
export class ManageBillPaymentsJourneyConfigService {
    constructor(config) {
        config = config || {};
        this._config = Object.assign(Object.assign({}, configDefaults), config);
    }
    get defaults() {
        return configDefaults;
    }
    get notificationDismissTime() {
        return this._config.notificationDismissTime;
    }
    get pageSize() {
        return this._config.pageSize;
    }
    get pageFilter() {
        return this._config.pageFilter;
    }
    get accountNumberMasked() {
        return this._config.accountNumberMasked;
    }
    get pageTitle() {
        return this._config.pageTitle;
    }
}
ManageBillPaymentsJourneyConfigService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageBillPaymentsJourneyConfigService, deps: [{ token: ManageBillPaymentsJourneyConfigurationToken, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
ManageBillPaymentsJourneyConfigService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageBillPaymentsJourneyConfigService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageBillPaymentsJourneyConfigService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [ManageBillPaymentsJourneyConfigurationToken]
                }] }]; } });
