// @backbase/cards-common-ang/esm2015/src/components/bb-payment-card-delivery-status/delivery-tracker.component.js
import { Component, Input } from '@angular/core';
import { PaymentCardDeliveryStepStatus } from '../../model/payment-card-delivery-status.model';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/icon";
import * as i2 from "@angular/common";
/**
 * @name DeliveryTrackerComponent
 *
 * @description
 * Component that displays the steps to fulfill an activity.
 */
export class DeliveryTrackerComponent {
    constructor() {
        /**
         * The steps that this component displays the steps of.  Mandatory.
         */
        this.steps = [];
        /**
         * Color for the icons and the lines
         */
        this.color = 'primary';
    }
    get currentStep() {
        const steps = this.steps.filter((step) => step.status === PaymentCardDeliveryStepStatus.Success);
        if (steps.length > 0) {
            return steps[steps.length - 1];
        }
        return undefined;
    }
    get stepColorClassName() {
        return `bb-delivery-tracker__step-icon-${this.color}`;
    }
    getStepClassName(step) {
        const stepClassName = step.name.toLowerCase().replace(/\W/g, '-').replace(/-{2,}/g, '-');
        return `bb-delivery-tracker__step--${stepClassName}`;
    }
    getStepIconClassName(name) {
        const stepClassName = name.toLowerCase().replace(/\W/g, '-').replace(/-{2,}/g, '-');
        return `delivery-tracker__step--${stepClassName}`;
    }
    ngOnInit() {
        if (typeof this.steps === 'undefined') {
            throw new Error(`"steps" input is required in "${this.constructor.name}"`);
        }
    }
}
DeliveryTrackerComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: DeliveryTrackerComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
DeliveryTrackerComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: DeliveryTrackerComponent, selector: "bb-delivery-tracker", inputs: { steps: "steps", color: "color" }, ngImport: i0, template: "<div class=\"bb-delivery-tracker\">\n  <div\n    class=\"bb-delivery-tracker__step\"\n    *ngFor=\"let step of steps\"\n    [ngClass]=\"[\n      step === currentStep ? 'active' : '',\n      getStepClassName(step)\n    ]\"\n  >\n    <div\n      class=\"bb-delivery-tracker__step-icon\"\n      [ngClass]=\"[\n        stepColorClassName,\n        step.status !== 'SUCCESS' ? 'bb-delivery-tracker__step-icon--disabled' : ''\n      ]\"\n    >\n      <bb-icon-ui \n        [name]=\"getStepIconClassName(step.status === 'SUCCESS' ? 'ordered' : step.name)\"\n        backgroundType=\"circle\"\n        size=\"md\"\n        [color]=\"step.status === 'SUCCESS' ? color : 'light'\"\n      ></bb-icon-ui>\n    </div>\n    <div\n      class=\"bb-delivery-tracker__step-text\"\n      [ngClass]=\"{'bb-text-support': step.status !== 'SUCCESS'}\"\n    >\n      {{ step.name }}\n    </div>\n    <div\n      class=\"bb-delivery-tracker__step-date bb-subtitle\"\n      *ngIf=\"step.stepDateTime\"\n    >\n      {{ step.stepDateTime | date }} at {{step.stepDateTime | date:'shortTime'}}\n    </div>\n  </div>\n</div>\n", components: [{ type: i1.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], directives: [{ type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { type: i2.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "date": i2.DatePipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: DeliveryTrackerComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-delivery-tracker',
                    templateUrl: './delivery-tracker.component.html',
                }]
        }], propDecorators: { steps: [{
                type: Input
            }], color: [{
                type: Input
            }] } });
