// @backbase/cards-common-ang/esm2015/src/components/bb-payment-card-delivery-status/payment-card-delivery-status.component.js
import { Component, Directive, Input } from '@angular/core';
import { BbTemplate } from '@backbase/foundation-ang/core';
import * as i0 from "@angular/core";
import * as i1 from "./delivery-tracker.component";
export class PaymentCardDeliveryStatusComponent {
    constructor() {
        /**
         * Delivery status of the payment card to be displayed.
         */
        this.deliveryStatus = { transitSteps: [] };
        this.hostRef = this;
    }
}
PaymentCardDeliveryStatusComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentCardDeliveryStatusComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
PaymentCardDeliveryStatusComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: PaymentCardDeliveryStatusComponent, selector: "bb-payment-card-delivery-status", inputs: { deliveryStatus: "deliveryStatus" }, ngImport: i0, template: "<ng-container bbCardsDeliveryStatusCustomizable [bbHostRef]=\"hostRef\"></ng-container>\n\n<ng-template bbCardsDeliveryStatusCustomizable let-hostRef>\n  <div class=\"bb-block bb-block--lg\">\n    <bb-delivery-tracker [steps]=\"hostRef.deliveryStatus.transitSteps\" color=\"success\"></bb-delivery-tracker>\n  </div>\n</ng-template>\n", components: [{ type: i0.forwardRef(function () { return i1.DeliveryTrackerComponent; }), selector: "bb-delivery-tracker", inputs: ["steps", "color"] }], directives: [{ type: i0.forwardRef(function () { return CardsDeliveryStatusCustomizableDirective; }), selector: "[bbCardsDeliveryStatusCustomizable]" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentCardDeliveryStatusComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-payment-card-delivery-status',
                    templateUrl: './payment-card-delivery-status.component.html',
                }]
        }], propDecorators: { deliveryStatus: [{
                type: Input
            }] } });
export class CardsDeliveryStatusCustomizableDirective extends BbTemplate {
}
CardsDeliveryStatusCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardsDeliveryStatusCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
CardsDeliveryStatusCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.15", type: CardsDeliveryStatusCustomizableDirective, selector: "[bbCardsDeliveryStatusCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardsDeliveryStatusCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbCardsDeliveryStatusCustomizable]',
                }]
        }] });
