// @backbase/ui-ang/esm2015/avatar/avatar.component.js
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
const MAX_INITIALS_LENGTH = 2;
export class AvatarComponent {
    constructor() {
        this.initials = '';
        /**
         * The image  of the avatar. Defaults to empty string.
         */
        this.image = '';
        /**
         * The name  of the avatar. Defaults to empty string.
         */
        this.name = '';
        /**
         * The imgAlt for provide alt in img of the avatar. Defaults to empty string.
         */
        this.imgAlt = '';
        /**
         * Size of the avatar. Possible values sm | md| lg | xl. Defaults to empty string.
         */
        this.size = '';
    }
    ngOnChanges() {
        this.initials = (this.name || '')
            .split(' ')
            .reduce((initials, word) => `${initials}${word.charAt(0)}`, '')
            .substr(0, MAX_INITIALS_LENGTH)
            .toUpperCase();
    }
}
AvatarComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AvatarComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
AvatarComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AvatarComponent, selector: "bb-avatar-ui", inputs: { image: "image", name: "name", imgAlt: "imgAlt", size: "size" }, usesOnChanges: true, ngImport: i0, template: "<div class=\"bb-avatar\" [ngClass]=\"[size ? 'bb-avatar--' + size : '', image ? 'bb-avatar--image' : '']\">\n  <span\n    *ngIf=\"!image\"\n    class=\"bb-avatar__initials\"\n    [textContent]=\"initials\"\n    data-role=\"avatar-initials\"\n    [attr.aria-label]=\"name\"\n  ></span>\n  <div\n    role=\"img\"\n    [attr.aria-label]=\"imgAlt || name\"\n    *ngIf=\"image\"\n    class=\"bb-avatar__image\"\n    data-role=\"avatar-image\"\n    [style.backgroundImage]=\"'url(' + image + ')'\"\n  ></div>\n</div>\n", directives: [{ type: i1.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AvatarComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-avatar-ui',
                    templateUrl: './avatar.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { image: [{
                type: Input
            }], name: [{
                type: Input
            }], imgAlt: [{
                type: Input
            }], size: [{
                type: Input
            }] } });
