// @backbase/user-manage-profile-widget-ang/esm2015/src/user-profile-forms/user-profile-form-inputs/phone-number-inputs/user-profile-form-phone-number-inputs.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { DropdownSingleSelectModule } from '@backbase/ui-ang/dropdown-single-select';
import { InputPhoneModule } from '@backbase/ui-ang/input-phone';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { PhoneNumberModule, PhoneNumberPipe } from '@backbase/ui-ang/phone-number-format-pipe';
import { UserPhoneNumberInputComponent, UserPhoneNumberInputCustomizableDirective, UserPhoneNumberInputErrorsCustomizableDirective, } from './user-phone-number-input.component';
import { UserPhoneNumberTypeErrorsCustomizableDirective, UserPhoneNumberTypeInputComponent, UserPhoneNumberTypeInputCustomizableDirective, } from './user-phone-number-type-input.component';
import * as i0 from "@angular/core";
const uiModules = [DropdownSingleSelectModule, InputPhoneModule, InputValidationMessageModule, PhoneNumberModule];
export class UserProfileFormPhoneNumberInputsModule {
}
UserProfileFormPhoneNumberInputsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormPhoneNumberInputsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
UserProfileFormPhoneNumberInputsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormPhoneNumberInputsModule, declarations: [UserPhoneNumberTypeInputComponent,
        UserPhoneNumberTypeInputCustomizableDirective,
        UserPhoneNumberTypeErrorsCustomizableDirective,
        UserPhoneNumberInputComponent,
        UserPhoneNumberInputCustomizableDirective,
        UserPhoneNumberInputErrorsCustomizableDirective], imports: [CommonModule, ReactiveFormsModule, DropdownSingleSelectModule, InputPhoneModule, InputValidationMessageModule, PhoneNumberModule], exports: [UserPhoneNumberTypeInputComponent,
        UserPhoneNumberTypeInputCustomizableDirective,
        UserPhoneNumberTypeErrorsCustomizableDirective,
        UserPhoneNumberInputComponent,
        UserPhoneNumberInputCustomizableDirective,
        UserPhoneNumberInputErrorsCustomizableDirective, DropdownSingleSelectModule, InputPhoneModule, InputValidationMessageModule, PhoneNumberModule] });
UserProfileFormPhoneNumberInputsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormPhoneNumberInputsModule, providers: [PhoneNumberPipe], imports: [[CommonModule, ReactiveFormsModule, ...uiModules], DropdownSingleSelectModule, InputPhoneModule, InputValidationMessageModule, PhoneNumberModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormPhoneNumberInputsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ReactiveFormsModule, ...uiModules],
                    declarations: [
                        UserPhoneNumberTypeInputComponent,
                        UserPhoneNumberTypeInputCustomizableDirective,
                        UserPhoneNumberTypeErrorsCustomizableDirective,
                        UserPhoneNumberInputComponent,
                        UserPhoneNumberInputCustomizableDirective,
                        UserPhoneNumberInputErrorsCustomizableDirective,
                    ],
                    providers: [PhoneNumberPipe],
                    exports: [
                        UserPhoneNumberTypeInputComponent,
                        UserPhoneNumberTypeInputCustomizableDirective,
                        UserPhoneNumberTypeErrorsCustomizableDirective,
                        UserPhoneNumberInputComponent,
                        UserPhoneNumberInputCustomizableDirective,
                        UserPhoneNumberInputErrorsCustomizableDirective,
                        ...uiModules,
                    ],
                }]
        }] });
