// @backbase/arrangement-manager-http-ang/esm2015/api/accounts.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { ARRANGEMENT_MANAGER_BASE_PATH } from '../variables';
import { ArrangementManagerConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class AccountsHttpService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new ArrangementManagerConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    filterAccounts(requestParameters, observe = 'body', reportProgress = false, options) {
        const _filterAccountsRequest = requestParameters["filterAccountsRequest"];
        if (_filterAccountsRequest === null || _filterAccountsRequest === undefined) {
            throw new Error('Required parameter filterAccountsRequest was null or undefined when calling filterAccounts.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v2/accounts/filter`, _filterAccountsRequest, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    filterAccountsUrl(requestParameters) {
        const _filterAccountsRequest = requestParameters["filterAccountsRequest"];
        if (_filterAccountsRequest === null || _filterAccountsRequest === undefined) {
            throw new Error('Required parameter filterAccountsRequest was null or undefined when calling filterAccounts.');
        }
        return `${this.configuration.basePath}/client-api/v2/accounts/filter`;
    }
    getBalanceHistory(requestParameters, observe = 'body', reportProgress = false, options) {
        const _arrangementIds = requestParameters["arrangementIds"];
        if (_arrangementIds === null || _arrangementIds === undefined) {
            throw new Error('Required parameter arrangementIds was null or undefined when calling getBalanceHistory.');
        }
        const _accept = requestParameters["accept"];
        const _dateFrom = requestParameters["dateFrom"];
        const _dateTo = requestParameters["dateTo"];
        const _timePeriod = requestParameters["timePeriod"];
        const _format = requestParameters["format"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_dateFrom !== undefined && _dateFrom !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _dateFrom, 'dateFrom');
        }
        if (_dateTo !== undefined && _dateTo !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _dateTo, 'dateTo');
        }
        if (_timePeriod !== undefined && _timePeriod !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _timePeriod, 'timePeriod');
        }
        if (_format !== undefined && _format !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _format, 'format');
        }
        let headers = this.defaultHeaders;
        if (_accept !== undefined && _accept !== null) {
            headers = headers.set('Accept', String(_accept));
        }
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json',
                'text/csv'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/accounts/balance-history/${encodeURIComponent(String(_arrangementIds))}`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getBalanceHistoryUrl(requestParameters) {
        const _arrangementIds = requestParameters["arrangementIds"];
        if (_arrangementIds === null || _arrangementIds === undefined) {
            throw new Error('Required parameter arrangementIds was null or undefined when calling getBalanceHistory.');
        }
        const _accept = requestParameters["accept"];
        const _dateFrom = requestParameters["dateFrom"];
        const _dateTo = requestParameters["dateTo"];
        const _timePeriod = requestParameters["timePeriod"];
        const _format = requestParameters["format"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_dateFrom !== undefined && _dateFrom !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _dateFrom, 'dateFrom');
        }
        if (_dateTo !== undefined && _dateTo !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _dateTo, 'dateTo');
        }
        if (_timePeriod !== undefined && _timePeriod !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _timePeriod, 'timePeriod');
        }
        if (_format !== undefined && _format !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _format, 'format');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/accounts/balance-history/${encodeURIComponent(String(_arrangementIds))}${queryString ? `?${queryString}` : ''}`;
    }
    getBalanceHistoryForArrangement(requestParameters, observe = 'body', reportProgress = false, options) {
        const _arrangementId = requestParameters["arrangementId"];
        if (_arrangementId === null || _arrangementId === undefined) {
            throw new Error('Required parameter arrangementId was null or undefined when calling getBalanceHistoryForArrangement.');
        }
        const _dateFrom = requestParameters["dateFrom"];
        const _dateTo = requestParameters["dateTo"];
        const _timePeriod = requestParameters["timePeriod"];
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        const _cursor = requestParameters["cursor"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_dateFrom !== undefined && _dateFrom !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _dateFrom, 'dateFrom');
        }
        if (_dateTo !== undefined && _dateTo !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _dateTo, 'dateTo');
        }
        if (_timePeriod !== undefined && _timePeriod !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _timePeriod, 'timePeriod');
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_cursor !== undefined && _cursor !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _cursor, 'cursor');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json',
                'text/csv'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/accounts/balance-history/arrangement/${encodeURIComponent(String(_arrangementId))}`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getBalanceHistoryForArrangementUrl(requestParameters) {
        const _arrangementId = requestParameters["arrangementId"];
        if (_arrangementId === null || _arrangementId === undefined) {
            throw new Error('Required parameter arrangementId was null or undefined when calling getBalanceHistoryForArrangement.');
        }
        const _dateFrom = requestParameters["dateFrom"];
        const _dateTo = requestParameters["dateTo"];
        const _timePeriod = requestParameters["timePeriod"];
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        const _cursor = requestParameters["cursor"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_dateFrom !== undefined && _dateFrom !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _dateFrom, 'dateFrom');
        }
        if (_dateTo !== undefined && _dateTo !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _dateTo, 'dateTo');
        }
        if (_timePeriod !== undefined && _timePeriod !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _timePeriod, 'timePeriod');
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_cursor !== undefined && _cursor !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _cursor, 'cursor');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/accounts/balance-history/arrangement/${encodeURIComponent(String(_arrangementId))}${queryString ? `?${queryString}` : ''}`;
    }
}
AccountsHttpService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountsHttpService, deps: [{ token: i1.HttpClient }, { token: ARRANGEMENT_MANAGER_BASE_PATH, optional: true }, { token: i2.ArrangementManagerConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
AccountsHttpService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountsHttpService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountsHttpService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [ARRANGEMENT_MANAGER_BASE_PATH]
                }] }, { type: i2.ArrangementManagerConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
