// @backbase/arrangement-manager-http-ang/esm2015/api/arrangements.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { ARRANGEMENT_MANAGER_BASE_PATH } from '../variables';
import { ArrangementManagerConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class ArrangementsHttpService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new ArrangementManagerConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    getArrangementById(requestParameters, observe = 'body', reportProgress = false, options) {
        const _arrangementId = requestParameters["arrangementId"];
        if (_arrangementId === null || _arrangementId === undefined) {
            throw new Error('Required parameter arrangementId was null or undefined when calling getArrangementById.');
        }
        const _contentLanguage = requestParameters["contentLanguage"];
        const _maskIndicator = requestParameters["maskIndicator"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_maskIndicator !== undefined && _maskIndicator !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _maskIndicator, 'maskIndicator');
        }
        let headers = this.defaultHeaders;
        if (_contentLanguage !== undefined && _contentLanguage !== null) {
            headers = headers.set('Content-Language', String(_contentLanguage));
        }
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/arrangements/${encodeURIComponent(String(_arrangementId))}`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getArrangementByIdUrl(requestParameters) {
        const _arrangementId = requestParameters["arrangementId"];
        if (_arrangementId === null || _arrangementId === undefined) {
            throw new Error('Required parameter arrangementId was null or undefined when calling getArrangementById.');
        }
        const _contentLanguage = requestParameters["contentLanguage"];
        const _maskIndicator = requestParameters["maskIndicator"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_maskIndicator !== undefined && _maskIndicator !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _maskIndicator, 'maskIndicator');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/arrangements/${encodeURIComponent(String(_arrangementId))}${queryString ? `?${queryString}` : ''}`;
    }
    getArrangementChildren(requestParameters, observe = 'body', reportProgress = false, options) {
        const _arrangementId = requestParameters["arrangementId"];
        if (_arrangementId === null || _arrangementId === undefined) {
            throw new Error('Required parameter arrangementId was null or undefined when calling getArrangementChildren.');
        }
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/arrangements/${encodeURIComponent(String(_arrangementId))}/children`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getArrangementChildrenUrl(requestParameters) {
        const _arrangementId = requestParameters["arrangementId"];
        if (_arrangementId === null || _arrangementId === undefined) {
            throw new Error('Required parameter arrangementId was null or undefined when calling getArrangementChildren.');
        }
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/arrangements/${encodeURIComponent(String(_arrangementId))}/children${queryString ? `?${queryString}` : ''}`;
    }
    getLinked(requestParameters, observe = 'body', reportProgress = false, options) {
        const _legalEntityId = requestParameters["legalEntityId"];
        if (_legalEntityId === null || _legalEntityId === undefined) {
            throw new Error('Required parameter legalEntityId was null or undefined when calling getLinked.');
        }
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_legalEntityId !== undefined && _legalEntityId !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _legalEntityId, 'legalEntityId');
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/arrangements/linked`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getLinkedUrl(requestParameters) {
        const _legalEntityId = requestParameters["legalEntityId"];
        if (_legalEntityId === null || _legalEntityId === undefined) {
            throw new Error('Required parameter legalEntityId was null or undefined when calling getLinked.');
        }
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_legalEntityId !== undefined && _legalEntityId !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _legalEntityId, 'legalEntityId');
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/arrangements/linked${queryString ? `?${queryString}` : ''}`;
    }
    getUserPreferences(requestParameters, observe = 'body', reportProgress = false, options) {
        const _arrangementId = requestParameters["arrangementId"];
        if (_arrangementId === null || _arrangementId === undefined) {
            throw new Error('Required parameter arrangementId was null or undefined when calling getUserPreferences.');
        }
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_arrangementId !== undefined && _arrangementId !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _arrangementId, 'arrangementId');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/arrangements/user-preferences`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getUserPreferencesUrl(requestParameters) {
        const _arrangementId = requestParameters["arrangementId"];
        if (_arrangementId === null || _arrangementId === undefined) {
            throw new Error('Required parameter arrangementId was null or undefined when calling getUserPreferences.');
        }
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_arrangementId !== undefined && _arrangementId !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _arrangementId, 'arrangementId');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/arrangements/user-preferences${queryString ? `?${queryString}` : ''}`;
    }
    linkExternalLegalEntity(requestParameters, observe = 'body', reportProgress = false, options) {
        const _externalLegalEntityId = requestParameters["externalLegalEntityId"];
        if (_externalLegalEntityId === null || _externalLegalEntityId === undefined) {
            throw new Error('Required parameter externalLegalEntityId was null or undefined when calling linkExternalLegalEntity.');
        }
        const _accountLinkedArrangementItem = requestParameters["accountLinkedArrangementItem"];
        if (_accountLinkedArrangementItem === null || _accountLinkedArrangementItem === undefined) {
            throw new Error('Required parameter accountLinkedArrangementItem was null or undefined when calling linkExternalLegalEntity.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v2/arrangements/link/${encodeURIComponent(String(_externalLegalEntityId))}`, _accountLinkedArrangementItem, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    linkExternalLegalEntityUrl(requestParameters) {
        const _externalLegalEntityId = requestParameters["externalLegalEntityId"];
        if (_externalLegalEntityId === null || _externalLegalEntityId === undefined) {
            throw new Error('Required parameter externalLegalEntityId was null or undefined when calling linkExternalLegalEntity.');
        }
        const _accountLinkedArrangementItem = requestParameters["accountLinkedArrangementItem"];
        if (_accountLinkedArrangementItem === null || _accountLinkedArrangementItem === undefined) {
            throw new Error('Required parameter accountLinkedArrangementItem was null or undefined when calling linkExternalLegalEntity.');
        }
        return `${this.configuration.basePath}/client-api/v2/arrangements/link/${encodeURIComponent(String(_externalLegalEntityId))}`;
    }
    unlinkExternalLegalEntity(requestParameters, observe = 'body', reportProgress = false, options) {
        const _externalLegalEntityId = requestParameters["externalLegalEntityId"];
        if (_externalLegalEntityId === null || _externalLegalEntityId === undefined) {
            throw new Error('Required parameter externalLegalEntityId was null or undefined when calling unlinkExternalLegalEntity.');
        }
        const _externalArrangementId = requestParameters["externalArrangementId"];
        if (_externalArrangementId === null || _externalArrangementId === undefined) {
            throw new Error('Required parameter externalArrangementId was null or undefined when calling unlinkExternalLegalEntity.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v2/arrangements/unlink/${encodeURIComponent(String(_externalLegalEntityId))}/${encodeURIComponent(String(_externalArrangementId))}`, null, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    unlinkExternalLegalEntityUrl(requestParameters) {
        const _externalLegalEntityId = requestParameters["externalLegalEntityId"];
        if (_externalLegalEntityId === null || _externalLegalEntityId === undefined) {
            throw new Error('Required parameter externalLegalEntityId was null or undefined when calling unlinkExternalLegalEntity.');
        }
        const _externalArrangementId = requestParameters["externalArrangementId"];
        if (_externalArrangementId === null || _externalArrangementId === undefined) {
            throw new Error('Required parameter externalArrangementId was null or undefined when calling unlinkExternalLegalEntity.');
        }
        return `${this.configuration.basePath}/client-api/v2/arrangements/unlink/${encodeURIComponent(String(_externalLegalEntityId))}/${encodeURIComponent(String(_externalArrangementId))}`;
    }
    unmaskedAttribute(requestParameters, observe = 'body', reportProgress = false, options) {
        const _arrangementId = requestParameters["arrangementId"];
        if (_arrangementId === null || _arrangementId === undefined) {
            throw new Error('Required parameter arrangementId was null or undefined when calling unmaskedAttribute.');
        }
        const _attributeName = requestParameters["attributeName"];
        if (_attributeName === null || _attributeName === undefined) {
            throw new Error('Required parameter attributeName was null or undefined when calling unmaskedAttribute.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'text/plain',
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/arrangements/${encodeURIComponent(String(_arrangementId))}/unmasked-attributes/${encodeURIComponent(String(_attributeName))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    unmaskedAttributeUrl(requestParameters) {
        const _arrangementId = requestParameters["arrangementId"];
        if (_arrangementId === null || _arrangementId === undefined) {
            throw new Error('Required parameter arrangementId was null or undefined when calling unmaskedAttribute.');
        }
        const _attributeName = requestParameters["attributeName"];
        if (_attributeName === null || _attributeName === undefined) {
            throw new Error('Required parameter attributeName was null or undefined when calling unmaskedAttribute.');
        }
        return `${this.configuration.basePath}/client-api/v2/arrangements/${encodeURIComponent(String(_arrangementId))}/unmasked-attributes/${encodeURIComponent(String(_attributeName))}`;
    }
    updateUserPreferences(requestParameters, observe = 'body', reportProgress = false, options) {
        const _accountUserPreferences = requestParameters["accountUserPreferences"];
        if (_accountUserPreferences === null || _accountUserPreferences === undefined) {
            throw new Error('Required parameter accountUserPreferences was null or undefined when calling updateUserPreferences.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v2/arrangements/user-preferences`, _accountUserPreferences, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    updateUserPreferencesUrl(requestParameters) {
        const _accountUserPreferences = requestParameters["accountUserPreferences"];
        if (_accountUserPreferences === null || _accountUserPreferences === undefined) {
            throw new Error('Required parameter accountUserPreferences was null or undefined when calling updateUserPreferences.');
        }
        return `${this.configuration.basePath}/client-api/v2/arrangements/user-preferences`;
    }
}
ArrangementsHttpService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ArrangementsHttpService, deps: [{ token: i1.HttpClient }, { token: ARRANGEMENT_MANAGER_BASE_PATH, optional: true }, { token: i2.ArrangementManagerConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
ArrangementsHttpService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ArrangementsHttpService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ArrangementsHttpService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [ARRANGEMENT_MANAGER_BASE_PATH]
                }] }, { type: i2.ArrangementManagerConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
