// @backbase/arrangement-manager-http-ang/esm2015/api/balances.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const BalancesHttpServiceGetAggregationsMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/balances/aggregations",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "aggregatedBalances": [{
                                "currencyCode": "EUR",
                                "amount": "3289.0",
                                "numberOfArrangements": 3
                            }, {
                                "currencyCode": "USD",
                                "amount": "2500.0",
                                "numberOfArrangements": 10
                            }]
                    }]
            },
            {
                status: 200,
                body: [{
                        "id": "871283172971234",
                        "productKindName": "Savings Account",
                        "balanceType": "Booked Balance",
                        "aggregatedBalances": [{
                                "currencyCode": "EUR",
                                "amount": "3289.0",
                                "numberOfArrangements": 3
                            }, {
                                "currencyCode": "USD",
                                "amount": "2500.0",
                                "numberOfArrangements": 10
                            }]
                    }, {
                        "id": "85342342387123",
                        "productKindName": "Term Deposit",
                        "balanceType": "Outstanding Principal Amount",
                        "aggregatedBalances": [{
                                "currencyCode": "USD",
                                "amount": "3543.0",
                                "numberOfArrangements": 3
                            }]
                    }]
            },
            {
                status: 200,
                body: [{
                        "aggregatedBalances": [{
                                "currencyCode": "USD",
                                "amount": "6043.0",
                                "numberOfArrangements": 14
                            }, {
                                "currencyCode": "EUR",
                                "amount": "3289.0",
                                "numberOfArrangements": 3
                            }]
                    }, {
                        "id": "871283172971234",
                        "productKindName": "Savings Account",
                        "balanceType": "Booked Balance",
                        "numberOfArrangements": 13,
                        "aggregatedBalances": [{
                                "currencyCode": "USD",
                                "amount": "2500.0",
                                "numberOfArrangements": 10
                            }, {
                                "currencyCode": "EUR",
                                "amount": "3289.0",
                                "numberOfArrangements": 3
                            }]
                    }, {
                        "id": "85342342387123",
                        "productKindName": "Term Deposit",
                        "balanceType": "OutstandingPrincipalAmount",
                        "numberOfArrangements": 4,
                        "aggregatedBalances": [{
                                "currencyCode": "USD",
                                "amount": "3543.0",
                                "numberOfArrangements": 4
                            }]
                    }]
            },
            {
                status: 200,
                body: [{
                        "aggregatedBalances": [{
                                "currencyCode": "EUR",
                                "amount": "3289.0",
                                "numberOfArrangements": 3
                            }, {
                                "currencyCode": "USD",
                                "amount": "2500.0",
                                "numberOfArrangements": 10
                            }]
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const BalancesHttpServiceGetFavoriteCurrenciesMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/balances/aggregations/favorites/currencies",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "currencyCode": "EUR",
                        "amount": "3289.0",
                        "numberOfArrangements": 3
                    }, {
                        "currencyCode": "USD",
                        "amount": "2500.0",
                        "numberOfArrangements": 10
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const BalancesHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v2/balances/aggregations",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "aggregatedBalances": [{
                                "currencyCode": "EUR",
                                "amount": "3289.0",
                                "numberOfArrangements": 3
                            }, {
                                "currencyCode": "USD",
                                "amount": "2500.0",
                                "numberOfArrangements": 10
                            }]
                    }]
            },
            {
                status: 200,
                body: [{
                        "id": "871283172971234",
                        "productKindName": "Savings Account",
                        "balanceType": "Booked Balance",
                        "aggregatedBalances": [{
                                "currencyCode": "EUR",
                                "amount": "3289.0",
                                "numberOfArrangements": 3
                            }, {
                                "currencyCode": "USD",
                                "amount": "2500.0",
                                "numberOfArrangements": 10
                            }]
                    }, {
                        "id": "85342342387123",
                        "productKindName": "Term Deposit",
                        "balanceType": "Outstanding Principal Amount",
                        "aggregatedBalances": [{
                                "currencyCode": "USD",
                                "amount": "3543.0",
                                "numberOfArrangements": 3
                            }]
                    }]
            },
            {
                status: 200,
                body: [{
                        "aggregatedBalances": [{
                                "currencyCode": "USD",
                                "amount": "6043.0",
                                "numberOfArrangements": 14
                            }, {
                                "currencyCode": "EUR",
                                "amount": "3289.0",
                                "numberOfArrangements": 3
                            }]
                    }, {
                        "id": "871283172971234",
                        "productKindName": "Savings Account",
                        "balanceType": "Booked Balance",
                        "numberOfArrangements": 13,
                        "aggregatedBalances": [{
                                "currencyCode": "USD",
                                "amount": "2500.0",
                                "numberOfArrangements": 10
                            }, {
                                "currencyCode": "EUR",
                                "amount": "3289.0",
                                "numberOfArrangements": 3
                            }]
                    }, {
                        "id": "85342342387123",
                        "productKindName": "Term Deposit",
                        "balanceType": "OutstandingPrincipalAmount",
                        "numberOfArrangements": 4,
                        "aggregatedBalances": [{
                                "currencyCode": "USD",
                                "amount": "3543.0",
                                "numberOfArrangements": 4
                            }]
                    }]
            },
            {
                status: 200,
                body: [{
                        "aggregatedBalances": [{
                                "currencyCode": "EUR",
                                "amount": "3289.0",
                                "numberOfArrangements": 3
                            }, {
                                "currencyCode": "USD",
                                "amount": "2500.0",
                                "numberOfArrangements": 10
                            }]
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/balances/aggregations/favorites/currencies",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "currencyCode": "EUR",
                        "amount": "3289.0",
                        "numberOfArrangements": 3
                    }, {
                        "currencyCode": "USD",
                        "amount": "2500.0",
                        "numberOfArrangements": 10
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
]);
