// @backbase/arrangement-manager-http-ang/esm2015/api/productKinds.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const ProductKindsHttpServiceGetProductKindsMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/product-kinds",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "productKinds": [{
                            "id": 1,
                            "externalKindId": "123123-123f",
                            "kindName": "Current Account",
                            "kindUri": "current-account",
                            "translations": [{
                                    "locale": "de_DE",
                                    "localeTranslations": [{
                                            "fieldName": "kind_name",
                                            "fieldValue": "Aktuelles Konto"
                                        }]
                                }, {
                                    "locale": "ar_JO",
                                    "localeTranslations": [{
                                            "fieldName": "kind_name",
                                            "fieldValue": "الحساب الحالي"
                                        }]
                                }]
                        }, {
                            "id": 2,
                            "externalKindId": "asdads67-asd",
                            "kindName": "Savings Account",
                            "kindUri": "savings-account",
                            "translations": [{
                                    "locale": "de_DE",
                                    "localeTranslations": [{
                                            "fieldName": "kind_name",
                                            "fieldValue": "Sparkonto"
                                        }]
                                }, {
                                    "locale": "ar_JO",
                                    "localeTranslations": [{
                                            "fieldName": "kind_name",
                                            "fieldValue": "حساب التوفير"
                                        }]
                                }]
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const ProductKindsHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v2/product-kinds",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "productKinds": [{
                            "id": 1,
                            "externalKindId": "123123-123f",
                            "kindName": "Current Account",
                            "kindUri": "current-account",
                            "translations": [{
                                    "locale": "de_DE",
                                    "localeTranslations": [{
                                            "fieldName": "kind_name",
                                            "fieldValue": "Aktuelles Konto"
                                        }]
                                }, {
                                    "locale": "ar_JO",
                                    "localeTranslations": [{
                                            "fieldName": "kind_name",
                                            "fieldValue": "الحساب الحالي"
                                        }]
                                }]
                        }, {
                            "id": 2,
                            "externalKindId": "asdads67-asd",
                            "kindName": "Savings Account",
                            "kindUri": "savings-account",
                            "translations": [{
                                    "locale": "de_DE",
                                    "localeTranslations": [{
                                            "fieldName": "kind_name",
                                            "fieldValue": "Sparkonto"
                                        }]
                                }, {
                                    "locale": "ar_JO",
                                    "localeTranslations": [{
                                            "fieldName": "kind_name",
                                            "fieldValue": "حساب التوفير"
                                        }]
                                }]
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
]);
