// @backbase/ui-ang/esm2015/_example/alert/examples/alert.example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../../configuration/configuration-input.service";
import * as i2 from "@backbase/ui-ang/alert";
import * as i3 from "../../configuration/configuration-input.component";
import * as i4 from "@angular/common";
const MILLISECONDS_PER_SECOND = 1000;
export class AlertExampleComponent {
    constructor(configurationService) {
        this.configurationService = configurationService;
        this.restoreDelay = 5;
        this.showAlert = true;
        this.alertCheckoxList = [
            {
                label: 'Dismissible',
                value: true,
                formControlName: 'dismissible',
            },
        ];
        this.alertDropdownList = [
            {
                label: 'Modifier',
                formControlName: 'modifier',
                dropdownItemList: ['info', 'warning', 'success', 'error'],
            },
        ];
        this.alertValues = {
            checkbox: {
                dismissible: true,
            },
            dropdown: {
                modifier: 'primary',
            },
            radio: {
                buttonSize: 'sm',
            },
        };
        this.configurationService.sharedField.subscribe((val) => {
            this.alertValues.checkbox.dismissible = val.checkbox.dismissible;
            this.alertValues.dropdown.modifier = val.dropdown.modifier;
        });
    }
    hideAlert() {
        this.showAlert = false;
        // Restore alert after 5 seconds for demo purposes
        setTimeout(() => this.displayAlert(), this.restoreDelay * MILLISECONDS_PER_SECOND);
    }
    displayAlert() {
        this.showAlert = true;
    }
}
AlertExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AlertExampleComponent, deps: [{ token: i1.ConfigurationInputService }], target: i0.ɵɵFactoryTarget.Component });
AlertExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AlertExampleComponent, selector: "bb-alert-example-ui", ngImport: i0, template: "<div class=\"row col-12\">\n  <div class=\"col-md-8 example-component-col\">\n    <bb-alert-ui\n      *ngIf=\"showAlert\"\n      title=\"Example information alert\"\n      [modifier]=\"alertValues.dropdown.modifier\"\n      [dismissible]=\"alertValues.checkbox.dismissible\"\n      (close)=\"hideAlert()\"\n    >\n      <p>\n        This example will be restored after {{ restoreDelay }} seconds.\n      </p>\n    </bb-alert-ui>\n  </div>\n  <div class=\"col-md-4 example-configuration-col\">\n    <bb-configuration-input-ui\n      [checkboxList]=\"alertCheckoxList\"\n      [dropdownList]=\"alertDropdownList\"\n    ></bb-configuration-input-ui>\n  </div>\n</div>\n", components: [{ type: i2.AlertComponent, selector: "bb-alert-ui", inputs: ["modifier", "dismissible", "title", "message"], outputs: ["close"] }, { type: i3.ConfigurationInputComponent, selector: "bb-configuration-input-ui", inputs: ["checkboxList", "radioList", "dropdownList"] }], directives: [{ type: i4.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AlertExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-alert-example-ui',
                    templateUrl: './alert.example.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.ConfigurationInputService }]; } });
