// @backbase/payment-orders-ang/esm2015/lib/ui/pipes/bytes-pipe/bytes.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BytesPipe } from './bytes.pipe';
import * as i0 from "@angular/core";
export class BytesModule {
}
/** @nocollapse */ BytesModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BytesModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ BytesModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BytesModule, declarations: [BytesPipe], imports: [CommonModule], exports: [BytesPipe] });
/** @nocollapse */ BytesModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BytesModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BytesModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [BytesPipe],
                    imports: [CommonModule],
                    exports: [BytesPipe],
                }]
        }] });
