// @backbase/manage-pockets-journey-ang/esm2015/src/manage-pockets-list/manage-pockets-list.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { ManagePocketsCommonModule } from '../manage-pockets-common/manage-pockets-common.module';
import { ManagePocketsEmptyListComponent } from './components/manage-pockets-empty-list/manage-pockets-empty-list.component';
import { ManagePocketsListWrapperComponent } from './components/manage-pockets-list-wrapper/manage-pockets-list-wrapper.component';
import { ManagePocketsListComponent } from './components/manage-pockets-list/manage-pockets-list.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const components = [ManagePocketsEmptyListComponent, ManagePocketsListComponent, ManagePocketsListWrapperComponent];
const modules = [LoadingIndicatorModule];
export class ManagePocketsListModule {
}
ManagePocketsListModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsListModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ManagePocketsListModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsListModule, declarations: [ManagePocketsEmptyListComponent, ManagePocketsListComponent, ManagePocketsListWrapperComponent], imports: [CommonModule,
        ManagePocketsCommonModule, LoadingIndicatorModule, i1.BackbaseCoreModule], exports: [ManagePocketsEmptyListComponent, ManagePocketsListComponent, ManagePocketsListWrapperComponent, LoadingIndicatorModule] });
ManagePocketsListModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsListModule, imports: [[
            CommonModule,
            ManagePocketsCommonModule,
            ...modules,
            BackbaseCoreModule.withConfig({
                classMap: { ManagePocketsListWrapperComponent },
            }),
        ], LoadingIndicatorModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsListModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components],
                    imports: [
                        CommonModule,
                        ManagePocketsCommonModule,
                        ...modules,
                        BackbaseCoreModule.withConfig({
                            classMap: { ManagePocketsListWrapperComponent },
                        }),
                    ],
                    exports: [...components, ...modules],
                }]
        }] });
