// @backbase/consent-ui/esm2015/src/consent-ui.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AvatarModule } from '@backbase/ui-ang/avatar';
import { IconModule } from '@backbase/ui-ang/icon';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { ThirdPartyProviderComponent } from './bb-third-party-provider-ui/third-party-provider.component';
import * as i0 from "@angular/core";
const uiModules = [AvatarModule, IconModule, TooltipModule];
export class ConsentUiModule {
}
ConsentUiModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentUiModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ConsentUiModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentUiModule, declarations: [ThirdPartyProviderComponent], imports: [CommonModule, AvatarModule, IconModule, TooltipModule], exports: [ThirdPartyProviderComponent, AvatarModule, IconModule, TooltipModule] });
ConsentUiModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentUiModule, imports: [[CommonModule, ...uiModules], AvatarModule, IconModule, TooltipModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentUiModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules],
                    declarations: [ThirdPartyProviderComponent],
                    exports: [ThirdPartyProviderComponent, ...uiModules],
                }]
        }] });
