// @backbase/cards-http-ang/esm2015/api/cards.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const CardsHttpServiceActivateMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/cards/{id}/activation",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "id": "sara-card-3",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold Card",
                    "status": "Inactive",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Sara Williams"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5237",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2020-05-04T12:52:20.906+00:00"
                            }, {
                                "name": "Processed",
                                "status": "PENDING"
                            }, {
                                "name": "In Transit",
                                "status": "PENDING"
                            }, {
                                "name": "Delivered",
                                "status": "PENDING"
                            }]
                    },
                    "limits": [{
                            "id": "sara-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000,
                            "maxAmount": 10000,
                            "minAmount": 0
                        }, {
                            "id": "sara-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000,
                            "maxAmount": 10000,
                            "minAmount": 0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/elon/value"
                }
            },
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/john/value"
                }
            },
            {
                status: 200,
                body: {
                    "id": "kim-card-2",
                    "brand": "mastercard",
                    "type": "Debit",
                    "subType": "ATM",
                    "name": "Copper",
                    "status": "Active",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Kim Sook"
                    },
                    "expiryDate": {
                        "year": "2027",
                        "month": "03"
                    },
                    "currency": "EUR",
                    "maskedNumber": "9292"
                }
            },
            {
                status: 200,
                body: {
                    "delivery": {
                        "additions": {
                            "key": "additions"
                        },
                        "transitSteps": [{
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }, {
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }]
                    },
                    "additions": {
                        "key": "additions"
                    },
                    "holder": {
                        "additions": {
                            "key": "additions"
                        },
                        "name": "name"
                    },
                    "type": "type",
                    "expiryDate": {
                        "month": "month",
                        "additions": {
                            "key": "additions"
                        },
                        "year": "year"
                    },
                    "maskedNumber": "maskedNumber",
                    "name": "name",
                    "subType": "subType",
                    "currency": "currency",
                    "id": "id",
                    "brand": "brand",
                    "replacement": {
                        "reason": "reason",
                        "replacingId": "replacingId",
                        "additions": {
                            "key": "additions"
                        },
                        "replacedById": "replacedById",
                        "status": "status"
                    },
                    "limits": [{
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }, {
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }],
                    "status": "status"
                }
            },
        ]
    }]);
export const CardsHttpServiceChangeLimitsMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/cards/{id}/limits",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/sara/value"
                }
            },
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/elon/value"
                }
            },
            {
                status: 200,
                body: {
                    "id": "john-card-1",
                    "brand": "visa",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold",
                    "status": "Active",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "John Doe"
                    },
                    "expiryDate": {
                        "year": "2030",
                        "month": "06"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5827",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "SUCCESS"
                            }, {
                                "name": "In Transit",
                                "status": "SUCCESS"
                            }, {
                                "name": "Delivered",
                                "status": "SUCCESS"
                            }]
                    },
                    "limits": [{
                            "id": "john-limit-1",
                            "channel": "ATM",
                            "frequency": "WEEKLY",
                            "amount": 1000,
                            "maxAmount": 9000,
                            "minAmount": 0
                        }, {
                            "id": "john-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 20000,
                            "maxAmount": 9000,
                            "minAmount": 0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/kim/value"
                }
            },
            {
                status: 200,
                body: {
                    "delivery": {
                        "additions": {
                            "key": "additions"
                        },
                        "transitSteps": [{
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }, {
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }]
                    },
                    "additions": {
                        "key": "additions"
                    },
                    "holder": {
                        "additions": {
                            "key": "additions"
                        },
                        "name": "name"
                    },
                    "type": "type",
                    "expiryDate": {
                        "month": "month",
                        "additions": {
                            "key": "additions"
                        },
                        "year": "year"
                    },
                    "maskedNumber": "maskedNumber",
                    "name": "name",
                    "subType": "subType",
                    "currency": "currency",
                    "id": "id",
                    "brand": "brand",
                    "replacement": {
                        "reason": "reason",
                        "replacingId": "replacingId",
                        "additions": {
                            "key": "additions"
                        },
                        "replacedById": "replacedById",
                        "status": "status"
                    },
                    "limits": [{
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }, {
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }],
                    "status": "status"
                }
            },
        ]
    }]);
export const CardsHttpServiceGetCardByIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/cards/{id}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "sara-card-1",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold Card",
                    "status": "Active",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Sara Williams"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5235",
                    "limits": [{
                            "id": "sara-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000,
                            "maxAmount": 10000,
                            "minAmount": 0
                        }, {
                            "id": "sara-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000,
                            "maxAmount": 10000,
                            "minAmount": 0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "elon-card-1",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "World Elite",
                    "status": "Active",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Elon"
                    },
                    "expiryDate": {
                        "year": "2050",
                        "month": "12"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5235",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "PENDING"
                            }, {
                                "name": "In Transit",
                                "status": "PENDING"
                            }, {
                                "name": "Delivered",
                                "status": "PENDING",
                                "stepDateTime": "2019-06-20T14:21:25.001Z"
                            }]
                    },
                    "limits": [{
                            "id": "elon-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 500000,
                            "maxAmount": 100000,
                            "minAmount": 0
                        }, {
                            "id": "elon-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 1000000,
                            "maxAmount": 800000,
                            "minAmount": 10000
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "john-card-1",
                    "brand": "visa",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold",
                    "status": "Active",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "John Doe"
                    },
                    "expiryDate": {
                        "year": "2030",
                        "month": "06"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5827",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "SUCCESS"
                            }, {
                                "name": "In Transit",
                                "status": "SUCCESS"
                            }, {
                                "name": "Delivered",
                                "status": "SUCCESS"
                            }]
                    },
                    "limits": [{
                            "id": "john-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 5000,
                            "maxAmount": 1000,
                            "minAmount": 0
                        }, {
                            "id": "john-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 20000,
                            "maxAmount": 9000,
                            "minAmount": 0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "kim-card-1",
                    "brand": "mastercard",
                    "type": "Debit",
                    "subType": "ATM",
                    "name": "Copper",
                    "status": "Inactive",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Kim Sook"
                    },
                    "expiryDate": {
                        "year": "2027",
                        "month": "03"
                    },
                    "currency": "EUR",
                    "maskedNumber": "9292"
                }
            },
            {
                status: 200,
                body: {
                    "delivery": {
                        "additions": {
                            "key": "additions"
                        },
                        "transitSteps": [{
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }, {
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }]
                    },
                    "additions": {
                        "key": "additions"
                    },
                    "holder": {
                        "additions": {
                            "key": "additions"
                        },
                        "name": "name"
                    },
                    "type": "type",
                    "expiryDate": {
                        "month": "month",
                        "additions": {
                            "key": "additions"
                        },
                        "year": "year"
                    },
                    "maskedNumber": "maskedNumber",
                    "name": "name",
                    "subType": "subType",
                    "currency": "currency",
                    "id": "id",
                    "brand": "brand",
                    "replacement": {
                        "reason": "reason",
                        "replacingId": "replacingId",
                        "additions": {
                            "key": "additions"
                        },
                        "replacedById": "replacedById",
                        "status": "status"
                    },
                    "limits": [{
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }, {
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }],
                    "status": "status"
                }
            },
        ]
    }]);
export const CardsHttpServiceGetCardsMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/cards",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "sara-card-1",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Sara Williams"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5235"
                    }, {
                        "id": "sara-card-2",
                        "brand": "visa",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Green Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Sara Williams"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "01"
                        },
                        "currency": "USD",
                        "maskedNumber": "5236"
                    }, {
                        "id": "sara-card-3",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Inactive",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Sara Williams"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5237",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2020-05-04T12:52:20.906+00:00"
                                }, {
                                    "name": "Processed",
                                    "status": "PENDING"
                                }, {
                                    "name": "In Transit",
                                    "status": "PENDING"
                                }, {
                                    "name": "Delivered",
                                    "status": "PENDING"
                                }]
                        }
                    }, {
                        "id": "sara-card-4",
                        "brand": "mastercard",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Black Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Sara Williams"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "6235"
                    }, {
                        "id": "sara-card-5",
                        "brand": "visa",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Blue Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Sara Williams"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "6236"
                    }]
            },
            {
                status: 200,
                body: [{
                        "id": "elon-card-1",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "World Elite",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Elon"
                        },
                        "expiryDate": {
                            "year": "2050",
                            "month": "12"
                        },
                        "currency": "EUR",
                        "maskedNumber": "5235",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-19T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "PENDING"
                                }, {
                                    "name": "In Transit",
                                    "status": "PENDING"
                                }, {
                                    "name": "Delivered",
                                    "status": "PENDING",
                                    "stepDateTime": "2019-06-20T14:21:25.001Z"
                                }]
                        }
                    }, {
                        "id": "elon-card-2",
                        "brand": "visa",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Infinite Privilege",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Elon"
                        },
                        "expiryDate": {
                            "year": "2045",
                            "month": "01"
                        },
                        "currency": "EUR",
                        "maskedNumber": "6782"
                    }, {
                        "id": "elon-card-3",
                        "brand": "american_express",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Reserve",
                        "status": "Cancelled",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "UnderReplacement",
                            "reason": "Lost",
                            "replacedById": "4"
                        },
                        "holder": {
                            "name": "Elon"
                        },
                        "expiryDate": {
                            "year": "2048",
                            "month": "06"
                        },
                        "currency": "EUR",
                        "maskedNumber": "1262"
                    }, {
                        "id": "elon-card-4",
                        "brand": "american_express",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Reserve",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Elon"
                        },
                        "expiryDate": {
                            "year": "2048",
                            "month": "07"
                        },
                        "currency": "EUR",
                        "maskedNumber": "1289"
                    }, {
                        "id": "elon-card-5",
                        "brand": "mastercard",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Platinum",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Elon"
                        },
                        "expiryDate": {
                            "year": "2050",
                            "month": "11"
                        },
                        "currency": "EUR",
                        "maskedNumber": "4627",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-19T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-20T14:32:20.906Z"
                                }, {
                                    "name": "In Transit",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-20T15:12:20.906Z"
                                }, {
                                    "name": "Delivered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-21T11:28:25.001Z"
                                }]
                        }
                    }, {
                        "id": "elon-card-6",
                        "brand": "american_express",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Reserve",
                        "status": "Active",
                        "lockStatus": "LOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Elon"
                        },
                        "expiryDate": {
                            "year": "2047",
                            "month": "03"
                        },
                        "currency": "EUR",
                        "maskedNumber": "8291"
                    }]
            },
            {
                status: 200,
                body: [{
                        "id": "john-card-1",
                        "brand": "visa",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "John Doe"
                        },
                        "expiryDate": {
                            "year": "2030",
                            "month": "06"
                        },
                        "currency": "EUR",
                        "maskedNumber": "5827",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-19T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "SUCCESS"
                                }, {
                                    "name": "In Transit",
                                    "status": "SUCCESS"
                                }, {
                                    "name": "Delivered",
                                    "status": "SUCCESS"
                                }]
                        },
                        "limits": [{
                                "id": "john-limit-1",
                                "channel": "ATM",
                                "frequency": "DAILY",
                                "amount": 3000,
                                "maxAmount": 9000,
                                "minAmount": 0
                            }, {
                                "id": "john-limit-2",
                                "channel": "ONLINE",
                                "frequency": "DAILY",
                                "amount": 20000,
                                "maxAmount": 9000,
                                "minAmount": 0
                            }]
                    }, {
                        "id": "john-card-2",
                        "brand": "mastercard",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Gold",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "John Doe"
                        },
                        "expiryDate": {
                            "year": "2031",
                            "month": "02"
                        },
                        "currency": "EUR",
                        "maskedNumber": "8299"
                    }]
            },
            {
                status: 200,
                body: [{
                        "id": "kim-card-1",
                        "brand": "mastercard",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Copper",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Kim Sook"
                        },
                        "expiryDate": {
                            "year": "2027",
                            "month": "03"
                        },
                        "currency": "EUR",
                        "maskedNumber": "9292"
                    }, {
                        "id": "kim-card-2",
                        "brand": "visa",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Iron",
                        "status": "Inactive",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Kim Sook"
                        },
                        "expiryDate": {
                            "year": "2010",
                            "month": "04"
                        },
                        "currency": "EUR",
                        "maskedNumber": "9494"
                    }, {
                        "id": "kim-card-3",
                        "brand": "visa",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Iron",
                        "status": "Expired",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Kim Sook"
                        },
                        "expiryDate": {
                            "year": "2010",
                            "month": "04"
                        },
                        "currency": "EUR",
                        "maskedNumber": "9494"
                    }]
            },
            {
                status: 200,
                body: []
            },
        ]
    }]);
export const CardsHttpServiceRequestPinMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/cards/{id}/pin/request",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/sara/value"
                }
            },
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/elon/value"
                }
            },
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/john/value"
                }
            },
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/kim/value"
                }
            },
            {
                status: 200,
                body: {
                    "delivery": {
                        "additions": {
                            "key": "additions"
                        },
                        "transitSteps": [{
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }, {
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }]
                    },
                    "additions": {
                        "key": "additions"
                    },
                    "holder": {
                        "additions": {
                            "key": "additions"
                        },
                        "name": "name"
                    },
                    "type": "type",
                    "expiryDate": {
                        "month": "month",
                        "additions": {
                            "key": "additions"
                        },
                        "year": "year"
                    },
                    "maskedNumber": "maskedNumber",
                    "name": "name",
                    "subType": "subType",
                    "currency": "currency",
                    "id": "id",
                    "brand": "brand",
                    "replacement": {
                        "reason": "reason",
                        "replacingId": "replacingId",
                        "additions": {
                            "key": "additions"
                        },
                        "replacedById": "replacedById",
                        "status": "status"
                    },
                    "limits": [{
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }, {
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }],
                    "status": "status"
                }
            },
        ]
    }]);
export const CardsHttpServiceRequestReplacementMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/cards/{id}/replacement",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/sara/value"
                }
            },
            {
                status: 200,
                body: {
                    "id": "elon-card-1",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "World Elite",
                    "status": "Active",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "ReplacementRequested",
                        "reason": "Damaged"
                    },
                    "holder": {
                        "name": "Elon"
                    },
                    "expiryDate": {
                        "year": "2050",
                        "month": "12"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5235",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "PENDING"
                            }]
                    }
                }
            },
            {
                status: 200,
                body: {
                    "id": "john-card-1",
                    "brand": "visa",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold",
                    "status": "Inactive",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "ReplacementRequested",
                        "reason": "Stolen"
                    },
                    "holder": {
                        "name": "John Doe"
                    },
                    "expiryDate": {
                        "year": "2030",
                        "month": "06"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5827",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "SUCCESS"
                            }, {
                                "name": "In Transit",
                                "status": "SUCCESS"
                            }, {
                                "name": "Delivered",
                                "status": "SUCCESS"
                            }]
                    }
                }
            },
            {
                status: 200,
                body: {
                    "id": "kim-card-1",
                    "brand": "mastercard",
                    "type": "Debit",
                    "subType": "ATM",
                    "name": "Copper",
                    "status": "Inactive",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "ReplacementRequested",
                        "reason": "Lost"
                    },
                    "holder": {
                        "name": "Kim Sook"
                    },
                    "expiryDate": {
                        "year": "2027",
                        "month": "03"
                    },
                    "currency": "EUR",
                    "maskedNumber": "9292"
                }
            },
            {
                status: 200,
                body: {
                    "delivery": {
                        "additions": {
                            "key": "additions"
                        },
                        "transitSteps": [{
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }, {
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }]
                    },
                    "additions": {
                        "key": "additions"
                    },
                    "holder": {
                        "additions": {
                            "key": "additions"
                        },
                        "name": "name"
                    },
                    "type": "type",
                    "expiryDate": {
                        "month": "month",
                        "additions": {
                            "key": "additions"
                        },
                        "year": "year"
                    },
                    "maskedNumber": "maskedNumber",
                    "name": "name",
                    "subType": "subType",
                    "currency": "currency",
                    "id": "id",
                    "brand": "brand",
                    "replacement": {
                        "reason": "reason",
                        "replacingId": "replacingId",
                        "additions": {
                            "key": "additions"
                        },
                        "replacedById": "replacedById",
                        "status": "status"
                    },
                    "limits": [{
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }, {
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }],
                    "status": "status"
                }
            },
        ]
    }]);
export const CardsHttpServiceResetPinMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/cards/{id}/pin/reset",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/sara/value"
                }
            },
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/elon/value"
                }
            },
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/john/value"
                }
            },
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/kim/value"
                }
            },
            {
                status: 200,
                body: {
                    "delivery": {
                        "additions": {
                            "key": "additions"
                        },
                        "transitSteps": [{
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }, {
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }]
                    },
                    "additions": {
                        "key": "additions"
                    },
                    "holder": {
                        "additions": {
                            "key": "additions"
                        },
                        "name": "name"
                    },
                    "type": "type",
                    "expiryDate": {
                        "month": "month",
                        "additions": {
                            "key": "additions"
                        },
                        "year": "year"
                    },
                    "maskedNumber": "maskedNumber",
                    "name": "name",
                    "subType": "subType",
                    "currency": "currency",
                    "id": "id",
                    "brand": "brand",
                    "replacement": {
                        "reason": "reason",
                        "replacingId": "replacingId",
                        "additions": {
                            "key": "additions"
                        },
                        "replacedById": "replacedById",
                        "status": "status"
                    },
                    "limits": [{
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }, {
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }],
                    "status": "status"
                }
            },
        ]
    }]);
export const CardsHttpServiceUpdateLockStatusMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/cards/{id}/lock-status",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/sara/value"
                }
            },
            {
                status: 200,
                body: {
                    "id": "elon-card-6",
                    "brand": "american_express",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Reserve",
                    "status": "Active",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Elon"
                    },
                    "expiryDate": {
                        "year": "2047",
                        "month": "03"
                    },
                    "currency": "EUR",
                    "maskedNumber": "8291"
                }
            },
            {
                status: 200,
                body: {
                    "id": "john-card-1",
                    "brand": "visa",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold",
                    "status": "Active",
                    "lockStatus": "LOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "John Doe"
                    },
                    "expiryDate": {
                        "year": "2030",
                        "month": "06"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5827",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "SUCCESS"
                            }, {
                                "name": "In Transit",
                                "status": "SUCCESS"
                            }, {
                                "name": "Delivered",
                                "status": "SUCCESS"
                            }]
                    },
                    "limits": [{
                            "id": "john-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 5000,
                            "maxAmount": 1000,
                            "minAmount": 0
                        }, {
                            "id": "john-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 20000,
                            "maxAmount": 9000,
                            "minAmount": 0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/kim/value"
                }
            },
            {
                status: 200,
                body: {
                    "delivery": {
                        "additions": {
                            "key": "additions"
                        },
                        "transitSteps": [{
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }, {
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }]
                    },
                    "additions": {
                        "key": "additions"
                    },
                    "holder": {
                        "additions": {
                            "key": "additions"
                        },
                        "name": "name"
                    },
                    "type": "type",
                    "expiryDate": {
                        "month": "month",
                        "additions": {
                            "key": "additions"
                        },
                        "year": "year"
                    },
                    "maskedNumber": "maskedNumber",
                    "name": "name",
                    "subType": "subType",
                    "currency": "currency",
                    "id": "id",
                    "brand": "brand",
                    "replacement": {
                        "reason": "reason",
                        "replacingId": "replacingId",
                        "additions": {
                            "key": "additions"
                        },
                        "replacedById": "replacedById",
                        "status": "status"
                    },
                    "limits": [{
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }, {
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }],
                    "status": "status"
                }
            },
        ]
    }]);
export const CardsHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v2/cards/{id}/activation",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "id": "sara-card-3",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold Card",
                    "status": "Inactive",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Sara Williams"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5237",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2020-05-04T12:52:20.906+00:00"
                            }, {
                                "name": "Processed",
                                "status": "PENDING"
                            }, {
                                "name": "In Transit",
                                "status": "PENDING"
                            }, {
                                "name": "Delivered",
                                "status": "PENDING"
                            }]
                    },
                    "limits": [{
                            "id": "sara-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000,
                            "maxAmount": 10000,
                            "minAmount": 0
                        }, {
                            "id": "sara-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000,
                            "maxAmount": 10000,
                            "minAmount": 0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/elon/value"
                }
            },
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/john/value"
                }
            },
            {
                status: 200,
                body: {
                    "id": "kim-card-2",
                    "brand": "mastercard",
                    "type": "Debit",
                    "subType": "ATM",
                    "name": "Copper",
                    "status": "Active",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Kim Sook"
                    },
                    "expiryDate": {
                        "year": "2027",
                        "month": "03"
                    },
                    "currency": "EUR",
                    "maskedNumber": "9292"
                }
            },
            {
                status: 200,
                body: {
                    "delivery": {
                        "additions": {
                            "key": "additions"
                        },
                        "transitSteps": [{
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }, {
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }]
                    },
                    "additions": {
                        "key": "additions"
                    },
                    "holder": {
                        "additions": {
                            "key": "additions"
                        },
                        "name": "name"
                    },
                    "type": "type",
                    "expiryDate": {
                        "month": "month",
                        "additions": {
                            "key": "additions"
                        },
                        "year": "year"
                    },
                    "maskedNumber": "maskedNumber",
                    "name": "name",
                    "subType": "subType",
                    "currency": "currency",
                    "id": "id",
                    "brand": "brand",
                    "replacement": {
                        "reason": "reason",
                        "replacingId": "replacingId",
                        "additions": {
                            "key": "additions"
                        },
                        "replacedById": "replacedById",
                        "status": "status"
                    },
                    "limits": [{
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }, {
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }],
                    "status": "status"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/cards/{id}/limits",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/sara/value"
                }
            },
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/elon/value"
                }
            },
            {
                status: 200,
                body: {
                    "id": "john-card-1",
                    "brand": "visa",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold",
                    "status": "Active",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "John Doe"
                    },
                    "expiryDate": {
                        "year": "2030",
                        "month": "06"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5827",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "SUCCESS"
                            }, {
                                "name": "In Transit",
                                "status": "SUCCESS"
                            }, {
                                "name": "Delivered",
                                "status": "SUCCESS"
                            }]
                    },
                    "limits": [{
                            "id": "john-limit-1",
                            "channel": "ATM",
                            "frequency": "WEEKLY",
                            "amount": 1000,
                            "maxAmount": 9000,
                            "minAmount": 0
                        }, {
                            "id": "john-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 20000,
                            "maxAmount": 9000,
                            "minAmount": 0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/kim/value"
                }
            },
            {
                status: 200,
                body: {
                    "delivery": {
                        "additions": {
                            "key": "additions"
                        },
                        "transitSteps": [{
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }, {
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }]
                    },
                    "additions": {
                        "key": "additions"
                    },
                    "holder": {
                        "additions": {
                            "key": "additions"
                        },
                        "name": "name"
                    },
                    "type": "type",
                    "expiryDate": {
                        "month": "month",
                        "additions": {
                            "key": "additions"
                        },
                        "year": "year"
                    },
                    "maskedNumber": "maskedNumber",
                    "name": "name",
                    "subType": "subType",
                    "currency": "currency",
                    "id": "id",
                    "brand": "brand",
                    "replacement": {
                        "reason": "reason",
                        "replacingId": "replacingId",
                        "additions": {
                            "key": "additions"
                        },
                        "replacedById": "replacedById",
                        "status": "status"
                    },
                    "limits": [{
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }, {
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }],
                    "status": "status"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/cards/{id}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "sara-card-1",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold Card",
                    "status": "Active",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Sara Williams"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5235",
                    "limits": [{
                            "id": "sara-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000,
                            "maxAmount": 10000,
                            "minAmount": 0
                        }, {
                            "id": "sara-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000,
                            "maxAmount": 10000,
                            "minAmount": 0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "elon-card-1",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "World Elite",
                    "status": "Active",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Elon"
                    },
                    "expiryDate": {
                        "year": "2050",
                        "month": "12"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5235",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "PENDING"
                            }, {
                                "name": "In Transit",
                                "status": "PENDING"
                            }, {
                                "name": "Delivered",
                                "status": "PENDING",
                                "stepDateTime": "2019-06-20T14:21:25.001Z"
                            }]
                    },
                    "limits": [{
                            "id": "elon-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 500000,
                            "maxAmount": 100000,
                            "minAmount": 0
                        }, {
                            "id": "elon-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 1000000,
                            "maxAmount": 800000,
                            "minAmount": 10000
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "john-card-1",
                    "brand": "visa",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold",
                    "status": "Active",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "John Doe"
                    },
                    "expiryDate": {
                        "year": "2030",
                        "month": "06"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5827",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "SUCCESS"
                            }, {
                                "name": "In Transit",
                                "status": "SUCCESS"
                            }, {
                                "name": "Delivered",
                                "status": "SUCCESS"
                            }]
                    },
                    "limits": [{
                            "id": "john-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 5000,
                            "maxAmount": 1000,
                            "minAmount": 0
                        }, {
                            "id": "john-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 20000,
                            "maxAmount": 9000,
                            "minAmount": 0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "kim-card-1",
                    "brand": "mastercard",
                    "type": "Debit",
                    "subType": "ATM",
                    "name": "Copper",
                    "status": "Inactive",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Kim Sook"
                    },
                    "expiryDate": {
                        "year": "2027",
                        "month": "03"
                    },
                    "currency": "EUR",
                    "maskedNumber": "9292"
                }
            },
            {
                status: 200,
                body: {
                    "delivery": {
                        "additions": {
                            "key": "additions"
                        },
                        "transitSteps": [{
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }, {
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }]
                    },
                    "additions": {
                        "key": "additions"
                    },
                    "holder": {
                        "additions": {
                            "key": "additions"
                        },
                        "name": "name"
                    },
                    "type": "type",
                    "expiryDate": {
                        "month": "month",
                        "additions": {
                            "key": "additions"
                        },
                        "year": "year"
                    },
                    "maskedNumber": "maskedNumber",
                    "name": "name",
                    "subType": "subType",
                    "currency": "currency",
                    "id": "id",
                    "brand": "brand",
                    "replacement": {
                        "reason": "reason",
                        "replacingId": "replacingId",
                        "additions": {
                            "key": "additions"
                        },
                        "replacedById": "replacedById",
                        "status": "status"
                    },
                    "limits": [{
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }, {
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }],
                    "status": "status"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/cards",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "sara-card-1",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Sara Williams"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5235"
                    }, {
                        "id": "sara-card-2",
                        "brand": "visa",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Green Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Sara Williams"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "01"
                        },
                        "currency": "USD",
                        "maskedNumber": "5236"
                    }, {
                        "id": "sara-card-3",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Inactive",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Sara Williams"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5237",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2020-05-04T12:52:20.906+00:00"
                                }, {
                                    "name": "Processed",
                                    "status": "PENDING"
                                }, {
                                    "name": "In Transit",
                                    "status": "PENDING"
                                }, {
                                    "name": "Delivered",
                                    "status": "PENDING"
                                }]
                        }
                    }, {
                        "id": "sara-card-4",
                        "brand": "mastercard",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Black Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Sara Williams"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "6235"
                    }, {
                        "id": "sara-card-5",
                        "brand": "visa",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Blue Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Sara Williams"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "6236"
                    }]
            },
            {
                status: 200,
                body: [{
                        "id": "elon-card-1",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "World Elite",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Elon"
                        },
                        "expiryDate": {
                            "year": "2050",
                            "month": "12"
                        },
                        "currency": "EUR",
                        "maskedNumber": "5235",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-19T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "PENDING"
                                }, {
                                    "name": "In Transit",
                                    "status": "PENDING"
                                }, {
                                    "name": "Delivered",
                                    "status": "PENDING",
                                    "stepDateTime": "2019-06-20T14:21:25.001Z"
                                }]
                        }
                    }, {
                        "id": "elon-card-2",
                        "brand": "visa",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Infinite Privilege",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Elon"
                        },
                        "expiryDate": {
                            "year": "2045",
                            "month": "01"
                        },
                        "currency": "EUR",
                        "maskedNumber": "6782"
                    }, {
                        "id": "elon-card-3",
                        "brand": "american_express",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Reserve",
                        "status": "Cancelled",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "UnderReplacement",
                            "reason": "Lost",
                            "replacedById": "4"
                        },
                        "holder": {
                            "name": "Elon"
                        },
                        "expiryDate": {
                            "year": "2048",
                            "month": "06"
                        },
                        "currency": "EUR",
                        "maskedNumber": "1262"
                    }, {
                        "id": "elon-card-4",
                        "brand": "american_express",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Reserve",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Elon"
                        },
                        "expiryDate": {
                            "year": "2048",
                            "month": "07"
                        },
                        "currency": "EUR",
                        "maskedNumber": "1289"
                    }, {
                        "id": "elon-card-5",
                        "brand": "mastercard",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Platinum",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Elon"
                        },
                        "expiryDate": {
                            "year": "2050",
                            "month": "11"
                        },
                        "currency": "EUR",
                        "maskedNumber": "4627",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-19T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-20T14:32:20.906Z"
                                }, {
                                    "name": "In Transit",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-20T15:12:20.906Z"
                                }, {
                                    "name": "Delivered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-21T11:28:25.001Z"
                                }]
                        }
                    }, {
                        "id": "elon-card-6",
                        "brand": "american_express",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Reserve",
                        "status": "Active",
                        "lockStatus": "LOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Elon"
                        },
                        "expiryDate": {
                            "year": "2047",
                            "month": "03"
                        },
                        "currency": "EUR",
                        "maskedNumber": "8291"
                    }]
            },
            {
                status: 200,
                body: [{
                        "id": "john-card-1",
                        "brand": "visa",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "John Doe"
                        },
                        "expiryDate": {
                            "year": "2030",
                            "month": "06"
                        },
                        "currency": "EUR",
                        "maskedNumber": "5827",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-19T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "SUCCESS"
                                }, {
                                    "name": "In Transit",
                                    "status": "SUCCESS"
                                }, {
                                    "name": "Delivered",
                                    "status": "SUCCESS"
                                }]
                        },
                        "limits": [{
                                "id": "john-limit-1",
                                "channel": "ATM",
                                "frequency": "DAILY",
                                "amount": 3000,
                                "maxAmount": 9000,
                                "minAmount": 0
                            }, {
                                "id": "john-limit-2",
                                "channel": "ONLINE",
                                "frequency": "DAILY",
                                "amount": 20000,
                                "maxAmount": 9000,
                                "minAmount": 0
                            }]
                    }, {
                        "id": "john-card-2",
                        "brand": "mastercard",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Gold",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "John Doe"
                        },
                        "expiryDate": {
                            "year": "2031",
                            "month": "02"
                        },
                        "currency": "EUR",
                        "maskedNumber": "8299"
                    }]
            },
            {
                status: 200,
                body: [{
                        "id": "kim-card-1",
                        "brand": "mastercard",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Copper",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Kim Sook"
                        },
                        "expiryDate": {
                            "year": "2027",
                            "month": "03"
                        },
                        "currency": "EUR",
                        "maskedNumber": "9292"
                    }, {
                        "id": "kim-card-2",
                        "brand": "visa",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Iron",
                        "status": "Inactive",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Kim Sook"
                        },
                        "expiryDate": {
                            "year": "2010",
                            "month": "04"
                        },
                        "currency": "EUR",
                        "maskedNumber": "9494"
                    }, {
                        "id": "kim-card-3",
                        "brand": "visa",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Iron",
                        "status": "Expired",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Kim Sook"
                        },
                        "expiryDate": {
                            "year": "2010",
                            "month": "04"
                        },
                        "currency": "EUR",
                        "maskedNumber": "9494"
                    }]
            },
            {
                status: 200,
                body: []
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/cards/{id}/pin/request",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/sara/value"
                }
            },
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/elon/value"
                }
            },
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/john/value"
                }
            },
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/kim/value"
                }
            },
            {
                status: 200,
                body: {
                    "delivery": {
                        "additions": {
                            "key": "additions"
                        },
                        "transitSteps": [{
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }, {
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }]
                    },
                    "additions": {
                        "key": "additions"
                    },
                    "holder": {
                        "additions": {
                            "key": "additions"
                        },
                        "name": "name"
                    },
                    "type": "type",
                    "expiryDate": {
                        "month": "month",
                        "additions": {
                            "key": "additions"
                        },
                        "year": "year"
                    },
                    "maskedNumber": "maskedNumber",
                    "name": "name",
                    "subType": "subType",
                    "currency": "currency",
                    "id": "id",
                    "brand": "brand",
                    "replacement": {
                        "reason": "reason",
                        "replacingId": "replacingId",
                        "additions": {
                            "key": "additions"
                        },
                        "replacedById": "replacedById",
                        "status": "status"
                    },
                    "limits": [{
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }, {
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }],
                    "status": "status"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/cards/{id}/replacement",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/sara/value"
                }
            },
            {
                status: 200,
                body: {
                    "id": "elon-card-1",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "World Elite",
                    "status": "Active",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "ReplacementRequested",
                        "reason": "Damaged"
                    },
                    "holder": {
                        "name": "Elon"
                    },
                    "expiryDate": {
                        "year": "2050",
                        "month": "12"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5235",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "PENDING"
                            }]
                    }
                }
            },
            {
                status: 200,
                body: {
                    "id": "john-card-1",
                    "brand": "visa",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold",
                    "status": "Inactive",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "ReplacementRequested",
                        "reason": "Stolen"
                    },
                    "holder": {
                        "name": "John Doe"
                    },
                    "expiryDate": {
                        "year": "2030",
                        "month": "06"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5827",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "SUCCESS"
                            }, {
                                "name": "In Transit",
                                "status": "SUCCESS"
                            }, {
                                "name": "Delivered",
                                "status": "SUCCESS"
                            }]
                    }
                }
            },
            {
                status: 200,
                body: {
                    "id": "kim-card-1",
                    "brand": "mastercard",
                    "type": "Debit",
                    "subType": "ATM",
                    "name": "Copper",
                    "status": "Inactive",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "ReplacementRequested",
                        "reason": "Lost"
                    },
                    "holder": {
                        "name": "Kim Sook"
                    },
                    "expiryDate": {
                        "year": "2027",
                        "month": "03"
                    },
                    "currency": "EUR",
                    "maskedNumber": "9292"
                }
            },
            {
                status: 200,
                body: {
                    "delivery": {
                        "additions": {
                            "key": "additions"
                        },
                        "transitSteps": [{
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }, {
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }]
                    },
                    "additions": {
                        "key": "additions"
                    },
                    "holder": {
                        "additions": {
                            "key": "additions"
                        },
                        "name": "name"
                    },
                    "type": "type",
                    "expiryDate": {
                        "month": "month",
                        "additions": {
                            "key": "additions"
                        },
                        "year": "year"
                    },
                    "maskedNumber": "maskedNumber",
                    "name": "name",
                    "subType": "subType",
                    "currency": "currency",
                    "id": "id",
                    "brand": "brand",
                    "replacement": {
                        "reason": "reason",
                        "replacingId": "replacingId",
                        "additions": {
                            "key": "additions"
                        },
                        "replacedById": "replacedById",
                        "status": "status"
                    },
                    "limits": [{
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }, {
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }],
                    "status": "status"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/cards/{id}/pin/reset",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/sara/value"
                }
            },
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/elon/value"
                }
            },
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/john/value"
                }
            },
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/kim/value"
                }
            },
            {
                status: 200,
                body: {
                    "delivery": {
                        "additions": {
                            "key": "additions"
                        },
                        "transitSteps": [{
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }, {
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }]
                    },
                    "additions": {
                        "key": "additions"
                    },
                    "holder": {
                        "additions": {
                            "key": "additions"
                        },
                        "name": "name"
                    },
                    "type": "type",
                    "expiryDate": {
                        "month": "month",
                        "additions": {
                            "key": "additions"
                        },
                        "year": "year"
                    },
                    "maskedNumber": "maskedNumber",
                    "name": "name",
                    "subType": "subType",
                    "currency": "currency",
                    "id": "id",
                    "brand": "brand",
                    "replacement": {
                        "reason": "reason",
                        "replacingId": "replacingId",
                        "additions": {
                            "key": "additions"
                        },
                        "replacedById": "replacedById",
                        "status": "status"
                    },
                    "limits": [{
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }, {
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }],
                    "status": "status"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/cards/{id}/lock-status",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/sara/value"
                }
            },
            {
                status: 200,
                body: {
                    "id": "elon-card-6",
                    "brand": "american_express",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Reserve",
                    "status": "Active",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Elon"
                    },
                    "expiryDate": {
                        "year": "2047",
                        "month": "03"
                    },
                    "currency": "EUR",
                    "maskedNumber": "8291"
                }
            },
            {
                status: 200,
                body: {
                    "id": "john-card-1",
                    "brand": "visa",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold",
                    "status": "Active",
                    "lockStatus": "LOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "John Doe"
                    },
                    "expiryDate": {
                        "year": "2030",
                        "month": "06"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5827",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "SUCCESS"
                            }, {
                                "name": "In Transit",
                                "status": "SUCCESS"
                            }, {
                                "name": "Delivered",
                                "status": "SUCCESS"
                            }]
                    },
                    "limits": [{
                            "id": "john-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 5000,
                            "maxAmount": 1000,
                            "minAmount": 0
                        }, {
                            "id": "john-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 20000,
                            "maxAmount": 9000,
                            "minAmount": 0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "$ref": "#/paths/~1client-api~1v2~1cards~1%7Bid%7D/get/responses/200/content/application~1json/examples/kim/value"
                }
            },
            {
                status: 200,
                body: {
                    "delivery": {
                        "additions": {
                            "key": "additions"
                        },
                        "transitSteps": [{
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }, {
                                "additions": {
                                    "key": "additions"
                                },
                                "name": "name",
                                "stepDateTime": "2000-01-23T04:56:07.000+00:00",
                                "status": "SUCCESS"
                            }]
                    },
                    "additions": {
                        "key": "additions"
                    },
                    "holder": {
                        "additions": {
                            "key": "additions"
                        },
                        "name": "name"
                    },
                    "type": "type",
                    "expiryDate": {
                        "month": "month",
                        "additions": {
                            "key": "additions"
                        },
                        "year": "year"
                    },
                    "maskedNumber": "maskedNumber",
                    "name": "name",
                    "subType": "subType",
                    "currency": "currency",
                    "id": "id",
                    "brand": "brand",
                    "replacement": {
                        "reason": "reason",
                        "replacingId": "replacingId",
                        "additions": {
                            "key": "additions"
                        },
                        "replacedById": "replacedById",
                        "status": "status"
                    },
                    "limits": [{
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }, {
                            "minAmount": 1.4658129805029452,
                            "amount": 0.8008281904610115,
                            "additions": {
                                "key": "additions"
                            },
                            "channel": "channel",
                            "id": "id",
                            "maxAmount": 6.027456183070403,
                            "frequency": "frequency"
                        }],
                    "status": "status"
                }
            },
        ]
    },
]);
