// @backbase/user-manage-profile-widget-ang/esm2015/src/user-profile-forms/user-profile-form-modals/user-modal-base.js
import { Directive, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
// eslint-disable-next-line @angular-eslint/directive-class-suffix
export class UserModalBase {
    constructor() {
        /**
         * Whether the modal is open.
         * Defaults to false.
         */
        this.isOpen = false;
        this.close = new EventEmitter();
        this.dismiss = new EventEmitter();
    }
    closeModal() {
        this.close.emit();
    }
    dismissModal() {
        this.dismiss.emit();
    }
}
UserModalBase.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserModalBase, deps: [], target: i0.ɵɵFactoryTarget.Directive });
UserModalBase.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.15", type: UserModalBase, selector: "[bbUserModalBase]", inputs: { isOpen: "isOpen" }, outputs: { close: "close", dismiss: "dismiss" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserModalBase, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbUserModalBase]',
                }]
        }], propDecorators: { isOpen: [{
                type: Input
            }], close: [{
                type: Output
            }], dismiss: [{
                type: Output
            }] } });
