// @backbase/user-manage-profile-widget-ang/esm2015/src/user-profile-forms/user-profile-form-modals/user-profile-form-modals.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { ButtonModule } from '@backbase/ui-ang/button';
import { HeaderModule } from '@backbase/ui-ang/header';
import { ModalModule } from '@backbase/ui-ang/modal';
import { NgbModalModule } from '@ng-bootstrap/ng-bootstrap';
import { UserModalBase } from './user-modal-base';
import { UserProfileFormCancelModalComponent, UserProfileFormCancelModalCustomizableDirective, } from './user-profile-form-cancel-modal.component';
import { UserProfileFormDeleteEmailAddressModalComponent, UserProfileFormDeleteEmailAddressModalCustomizableDirective, } from './user-profile-form-delete-email-address-modal.component';
import { UserProfileFormDeletePhoneAddressModalComponent, UserProfileFormDeletePhoneAddressModalCustomizableDirective, } from './user-profile-form-delete-phone-address-modal.component';
import { UserProfileFormDeletePostalAddressModalComponent, UserProfileFormDeletePostalAddressModalCustomizableDirective, } from './user-profile-form-delete-postal-address-modal.component';
import * as i0 from "@angular/core";
const uiModules = [ModalModule, ButtonModule, HeaderModule];
const components = [
    UserModalBase,
    UserProfileFormCancelModalComponent,
    UserProfileFormCancelModalCustomizableDirective,
    UserProfileFormDeleteEmailAddressModalComponent,
    UserProfileFormDeleteEmailAddressModalCustomizableDirective,
    UserProfileFormDeletePhoneAddressModalComponent,
    UserProfileFormDeletePhoneAddressModalCustomizableDirective,
    UserProfileFormDeletePostalAddressModalComponent,
    UserProfileFormDeletePostalAddressModalCustomizableDirective,
];
export class UserProfileFormModalsModule {
}
UserProfileFormModalsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormModalsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
UserProfileFormModalsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormModalsModule, declarations: [UserModalBase,
        UserProfileFormCancelModalComponent,
        UserProfileFormCancelModalCustomizableDirective,
        UserProfileFormDeleteEmailAddressModalComponent,
        UserProfileFormDeleteEmailAddressModalCustomizableDirective,
        UserProfileFormDeletePhoneAddressModalComponent,
        UserProfileFormDeletePhoneAddressModalCustomizableDirective,
        UserProfileFormDeletePostalAddressModalComponent,
        UserProfileFormDeletePostalAddressModalCustomizableDirective], imports: [CommonModule, ReactiveFormsModule, NgbModalModule, ModalModule, ButtonModule, HeaderModule], exports: [UserModalBase,
        UserProfileFormCancelModalComponent,
        UserProfileFormCancelModalCustomizableDirective,
        UserProfileFormDeleteEmailAddressModalComponent,
        UserProfileFormDeleteEmailAddressModalCustomizableDirective,
        UserProfileFormDeletePhoneAddressModalComponent,
        UserProfileFormDeletePhoneAddressModalCustomizableDirective,
        UserProfileFormDeletePostalAddressModalComponent,
        UserProfileFormDeletePostalAddressModalCustomizableDirective, ModalModule, ButtonModule, HeaderModule] });
UserProfileFormModalsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormModalsModule, imports: [[CommonModule, ReactiveFormsModule, NgbModalModule, ...uiModules], ModalModule, ButtonModule, HeaderModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormModalsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ReactiveFormsModule, NgbModalModule, ...uiModules],
                    declarations: [...components],
                    exports: [...components, ...uiModules],
                }]
        }] });
