// @backbase/contact-common-ang/esm2015/lib/pipes/sort-pipes/aria-sort-pipe.module.js
import { NgModule } from '@angular/core';
import { AriaSortPipe } from './aria-sort.pipe';
import { CommonModule } from '@angular/common';
import * as i0 from "@angular/core";
export class AriaSortPipeModule {
}
/** @nocollapse */ AriaSortPipeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AriaSortPipeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ AriaSortPipeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AriaSortPipeModule, declarations: [AriaSortPipe], imports: [CommonModule], exports: [AriaSortPipe] });
/** @nocollapse */ AriaSortPipeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AriaSortPipeModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AriaSortPipeModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [AriaSortPipe],
                    imports: [CommonModule],
                    exports: [AriaSortPipe],
                }]
        }] });
