// @backbase/contact-common-ang/esm2015/lib/pipes/sort-pipes/direction-sort-pipe.module.js
import { DirectionSortPipe } from './direction-sort.pipe';
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import * as i0 from "@angular/core";
export class DirectionSortPipeModule {
}
/** @nocollapse */ DirectionSortPipeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DirectionSortPipeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ DirectionSortPipeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DirectionSortPipeModule, declarations: [DirectionSortPipe], imports: [CommonModule], exports: [DirectionSortPipe] });
/** @nocollapse */ DirectionSortPipeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DirectionSortPipeModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DirectionSortPipeModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [DirectionSortPipe],
                    imports: [CommonModule],
                    exports: [DirectionSortPipe],
                }]
        }] });
