// @backbase/contact-common-ang/esm2015/lib/components/country-item/country-item.module.js
import { NgModule } from '@angular/core';
import { CountryItemComponent } from './country-item.component';
import { CountryItemCustomizableDirective } from './country-item.directive';
import { CommonModule } from '@angular/common';
import * as i0 from "@angular/core";
/**
 * Exports:
 * - CountryItemComponent
 * - CountryItemCustomizableDirective
 */
export class CountryItemModule {
}
/** @nocollapse */ CountryItemModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CountryItemModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ CountryItemModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CountryItemModule, declarations: [CountryItemComponent, CountryItemCustomizableDirective], imports: [CommonModule], exports: [CountryItemComponent, CountryItemCustomizableDirective] });
/** @nocollapse */ CountryItemModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CountryItemModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CountryItemModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [CountryItemComponent, CountryItemCustomizableDirective],
                    exports: [CountryItemComponent, CountryItemCustomizableDirective],
                    imports: [CommonModule],
                }]
        }] });
