// @backbase/manage-pockets-journey-ang/esm2015/src/manage-pockets-common/localization/manage-pockets-journey.localizations.js
export const POCKET_LOCALIZATIONS = {
    transfers: {
        details: {
            addMoneyContinueButton: $localize `:Review Transfer@@managePockets.transfers.details.button.continue.add:Review Transfer`,
            addMoneyError: $localize `:Your account balance is insufficient@@managePockets.transfers.details.amount.error.add.max:Your account balance is insufficient`,
            addMoneyHeader: $localize `:From Account@@managePockets.transfers.details.heading.addMoney:From Account`,
            withdrawMoneyContinueButton: $localize `:Review Withdrawal@@managePockets.transfers.details.button.continue.withdraw:Review Withdrawal`,
            withdrawMoneyError: $localize `:Your pocket balance is insufficient@@managePockets.transfers.details.amount.error.withdraw.max:Your pocket balance is insufficient`,
            withdrawMoneyHeader: $localize `:To Account@@managePockets.transfers.details.heading.withdrawMoney:To Account`,
        },
        review: {
            addMoneyHeader: $localize `:Review Your Transfer@@managePockets.transfers.review.header.add:Review Your Transfer`,
            withdrawMoneyHeader: $localize `:Review Your Withdrawal@@managePockets.transfers.review.header.withdraw:Review Your Withdrawal`,
        },
        wrapper: {
            addMoneyModalBody: $localize `:All transfer information will be permanently lost.@@managePockets.transfers.wrapper.modal.body.add:All transfer information will be permanently lost.`,
            addMoneyModalConfirm: $localize `:Discard Transfer@@managePockets.transfers.wrapper.modal.button.confirm.add:Discard Transfer`,
            addMoneyModalHeader: $localize `:Discard Transfer?@@managePockets.transfers.wrapper.modal.header.add:Discard Transfer?`,
            withdrawMoneyModalBody: $localize `:All withdrawal information will be permanently lost.@@managePockets.transfers.wrapper.modal.body.withdraw:All withdrawal information will be permanently lost.`,
            withdrawMoneyModalConfirm: $localize `:Discard Withdrawal@@managePockets.transfers.wrapper.modal.button.confirm.withdraw:Discard Withdrawal`,
            withdrawMoneyModalHeader: $localize `:Discard Withdrawal?@@managePockets.transfers.wrapper.modal.header.withdraw:Discard Withdrawal?`,
        },
    },
    views: {
        transfers: {
            addMoneyHeader: $localize `:Add Money@@managePockets.transfersView.heading.add:Add Money`,
            withdrawMoneyHeader: $localize `:Withdraw Money@@managePockets.transfersView.heading.withdraw:Withdraw Money`,
        },
    },
};
