// @backbase/identity-self-service-journey-ang/esm2015/lib/identity-self-service-journey.component.js
import { ChangeDetectionStrategy, Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/identity-common-ang";
export class IdentitySelfServiceJourneyComponent {
}
/** @nocollapse */ IdentitySelfServiceJourneyComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: IdentitySelfServiceJourneyComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ IdentitySelfServiceJourneyComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: IdentitySelfServiceJourneyComponent, selector: "bb-identity-self-service-journey", ngImport: i0, template: "<bb-routable-tab-set data-role=\"user-context-routable-tab-set\"></bb-routable-tab-set>\n", components: [{ type: i1.RoutableTabComponent, selector: "bb-routable-tab-set" }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: IdentitySelfServiceJourneyComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-identity-self-service-journey',
                    templateUrl: './identity-self-service-journey.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }] });
