// @backbase/identity-self-service-journey-ang/esm2015/lib/identity-self-service-journey.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { provideRoutes, RouterModule } from '@angular/router';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { UserIdentitySecurityCenterViewComponent } from './features/user-identity-security-center-view.component';
import { UserManageProfileViewComponent } from './features/user-manage-profile-view.component';
import { DeviceInformationViewComponent } from './features/device-information-view.component';
import { UserLocalizationComponent } from './features/user-localization/user-localization.component';
import { UserLocalizationCommunicationsViewComponent } from './features/user-localization/communications-view/user-localization-communications-view.component';
import { UserLocalizationChangeLanguageViewComponent } from './features/user-localization/change-language-view/user-localization-change-language-view.component';
import { IdentitySelfServiceJourneyComponent } from './identity-self-service-journey.component';
import { IdentityCommonAngModule } from '@backbase/identity-common-ang';
import { SelfServiceFeaturesModule } from './features/self-service-features.module';
import { IdentitySelfServiceJourneyConfigurationService } from './identity-self-service-journey.config.service';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
export const defaultRoute = {
    path: '',
    component: IdentitySelfServiceJourneyComponent,
    children: [
        { path: '', redirectTo: 'profile', pathMatch: 'full' },
        {
            path: 'profile',
            data: {
                title: $localize `:Tab label for managing user profile@@bb-identity-self-service-journey.tab-user-manage-profile:Profile`,
            },
            component: UserManageProfileViewComponent,
        },
        {
            path: 'user-localization',
            data: {
                title: $localize `:Tab label for changing user preferences@@bb-identity-self-service-journey.tab-localization:Localization`,
            },
            component: UserLocalizationComponent,
            children: [
                { path: '', redirectTo: 'communications', pathMatch: 'full' },
                {
                    path: 'communications',
                    component: UserLocalizationCommunicationsViewComponent,
                },
                {
                    path: 'change-language',
                    component: UserLocalizationChangeLanguageViewComponent,
                },
            ],
        },
        {
            path: 'login-security',
            data: {
                title: $localize `:Tab label for login and security settings@@bb-identity-self-service-journey.tab-login-security:Login & security`,
            },
            component: UserIdentitySecurityCenterViewComponent,
        },
        {
            path: 'devices',
            data: { title: $localize `:Tab label for managing devices@@bb-identity-self-service-journey.tab-devices:Devices` },
            component: DeviceInformationViewComponent,
        },
    ],
};
export class IdentitySelfServiceJourneyModule {
    static forRoot(
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    data = { route: defaultRoute }) {
        return {
            ngModule: IdentitySelfServiceJourneyModule,
            providers: [provideRoutes([data.route])],
        };
    }
}
/** @nocollapse */ IdentitySelfServiceJourneyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: IdentitySelfServiceJourneyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ IdentitySelfServiceJourneyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: IdentitySelfServiceJourneyModule, declarations: [IdentitySelfServiceJourneyComponent], imports: [CommonModule,
        RouterModule, i1.BackbaseCoreModule, IdentityCommonAngModule,
        SelfServiceFeaturesModule], exports: [RouterModule, IdentitySelfServiceJourneyComponent] });
/** @nocollapse */ IdentitySelfServiceJourneyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: IdentitySelfServiceJourneyModule, providers: [IdentitySelfServiceJourneyConfigurationService], imports: [[
            CommonModule,
            RouterModule,
            BackbaseCoreModule.forRoot({
                classMap: {
                    IdentitySelfServiceJourneyComponent,
                },
            }),
            IdentityCommonAngModule,
            SelfServiceFeaturesModule,
        ], RouterModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: IdentitySelfServiceJourneyModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        RouterModule,
                        BackbaseCoreModule.forRoot({
                            classMap: {
                                IdentitySelfServiceJourneyComponent,
                            },
                        }),
                        IdentityCommonAngModule,
                        SelfServiceFeaturesModule,
                    ],
                    declarations: [IdentitySelfServiceJourneyComponent],
                    providers: [IdentitySelfServiceJourneyConfigurationService],
                    exports: [RouterModule, IdentitySelfServiceJourneyComponent],
                }]
        }] });
