// @backbase/actions-store-ang/bundles/backbase-actions-store-ang.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('@ngrx/store'), require('@ngrx/effects'), require('rxjs'), require('rxjs/operators'), require('@backbase/actions-common-ang'), require('@backbase/ui-ang/notification')) :
    typeof define === 'function' && define.amd ? define('@backbase/actions-store-ang', ['exports', '@angular/core', '@angular/common', '@ngrx/store', '@ngrx/effects', 'rxjs', 'rxjs/operators', '@backbase/actions-common-ang', '@backbase/ui-ang/notification'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["actions-store-ang"] = {}), global.ng.core, global.ng.common, global.i1, global.i1$1, global.rxjs, global.rxjs.operators, global.i3, global.i4));
})(this, (function (exports, i0, common, i1, i1$1, rxjs, operators, i3, i4) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i1__namespace$1 = /*#__PURE__*/_interopNamespace(i1);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1$1);
    var i3__namespace = /*#__PURE__*/_interopNamespace(i3);
    var i4__namespace = /*#__PURE__*/_interopNamespace(i4);

    var loadActionRecipes = i1.createAction('[ACT Action Recipes] Load Action Recipes', i1.props());
    var loadActionRecipesSuccess = i1.createAction('[ACT Action Recipes] Load Action Recipes Success', i1.props());
    var loadActionRecipesFailed = i1.createAction('[ACT Action Recipes] Load Action Recipes Failed', i1.props());
    var saveActionRecipe = i1.createAction('[ACT Action Recipes] Save Action Recipe', i1.props());
    var saveActionRecipeSuccess = i1.createAction('[ACT Action Recipes] Save Action Recipe Success', i1.props());
    var saveActionRecipeFailure = i1.createAction('[ACT Action Recipes] Save Action Recipe Failure', i1.props());

    var selectActionRecipesState = i1.createFeatureSelector('actionRecipes');
    var actionRecipes = i1.createSelector(selectActionRecipesState, function (state) { return state.data; });
    var actionRecipesLoading = i1.createSelector(selectActionRecipesState, function (state) { return state.loading; });
    var actionRecipesError = i1.createSelector(selectActionRecipesState, function (state) { return state.error; });
    var actionRecipesSpecificationIds = i1.createSelector(selectActionRecipesState, function (state) { return state.specificationIds; });

    var ActionsRecipesEffects = /** @class */ (function () {
        function ActionsRecipesEffects(actions$, store, dataService, notificationService) {
            var _this = this;
            this.actions$ = actions$;
            this.store = store;
            this.dataService = dataService;
            this.notificationService = notificationService;
            this.actionRecipes$ = i1$1.createEffect(function () { return _this.actions$.pipe(i1$1.ofType(loadActionRecipes), operators.switchMap(function (_a) {
                var specificationIds = _a.specificationIds, arrangementId = _a.arrangementId;
                return _this.dataService.getActionRecipes(arrangementId).pipe(operators.map(function (data) { return (specificationIds ? data.filter(function (_a) {
                    var id = _a.specificationId;
                    return specificationIds.has(id);
                }) : data); }), operators.map(function (data) { return arrangementId ? data.filter(function (item) { return !item.arrangementId || item.arrangementId === arrangementId; }) : data; }), operators.map(function (data) { return loadActionRecipesSuccess({
                    actionRecipes: data || [],
                    specificationIds: specificationIds,
                }); }), operators.catchError(function (error) { return rxjs.of(loadActionRecipesFailed({ error: error })); }));
            })); });
            this.saveActionRecipe$ = i1$1.createEffect(function () { return _this.actions$.pipe(i1$1.ofType(saveActionRecipe), operators.groupBy(function (_a) {
                var actionRecipe = _a.actionRecipe;
                return actionRecipe.specificationId;
            }), operators.mergeMap(function (group) { return group.pipe(operators.switchMap(function (_a) {
                var actionRecipe = _a.actionRecipe, arrangementId = _a.arrangementId, warningNotification = _a.warningNotification, successNotification = _a.successNotification;
                return _this.dataService.saveActionRecipe(actionRecipe, arrangementId).pipe(operators.switchMap(function (_a) {
                    var actionRecipes = _a.actionRecipes;
                    if (!actionRecipes || !actionRecipes.length)
                        return rxjs.throwError(i3.EMPTY_RESPONSE_ERROR);
                    return rxjs.of(saveActionRecipeSuccess(Object.assign({ actionRecipe: Object.assign(Object.assign({}, actionRecipes[0]), (actionRecipes[0].recurrence && {
                            recurrence: i3.getMappedRecurrenceResponse(actionRecipes[0].recurrence, actionRecipe),
                        })) }, (successNotification && { notification: successNotification }))));
                }), operators.catchError(function (err) {
                    if (err.status === 409) {
                        return _this.store.select(actionRecipesSpecificationIds).pipe(operators.first(), operators.map(function (specificationIds) { return loadActionRecipes({
                            arrangementId: arrangementId,
                            specificationIds: specificationIds,
                        }); }));
                    }
                    return rxjs.of(saveActionRecipeFailure(Object.assign({ actionRecipe: i3.parseActionRecipeFormItem(actionRecipe) }, (warningNotification && { notification: warningNotification }))));
                }));
            })); })); });
            this.showNotification$ = i1$1.createEffect(function () { return rxjs.merge(_this.actions$.pipe(i1$1.ofType(saveActionRecipeSuccess)), _this.actions$.pipe(i1$1.ofType(saveActionRecipeFailure))).pipe(operators.map(function (_a) {
                var notification = _a.notification;
                if (notification) {
                    _this.notificationService.showNotification(notification);
                }
            })); }, { dispatch: false });
        }
        return ActionsRecipesEffects;
    }());
    ActionsRecipesEffects.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsRecipesEffects, deps: [{ token: i1__namespace.Actions }, { token: i1__namespace$1.Store }, { token: i3__namespace.NotificationsPreferencesBaseDataService }, { token: i4__namespace.NotificationService }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    ActionsRecipesEffects.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsRecipesEffects });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsRecipesEffects, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () { return [{ type: i1__namespace.Actions }, { type: i1__namespace$1.Store }, { type: i3__namespace.NotificationsPreferencesBaseDataService }, { type: i4__namespace.NotificationService }]; } });

    var loadSpecifications = i1.createAction('[ACT Specifications] Load Specifications', i1.props());
    var loadSpecificationsSuccess = i1.createAction('[ACT Specifications] Load Specifications Success', i1.props());
    var loadSpecificationsFailed = i1.createAction('[ACT Specifications] Load Specifications Failed', i1.props());

    var ActionsSpecificationsEffects = /** @class */ (function () {
        function ActionsSpecificationsEffects(actions$, dataService) {
            var _this = this;
            this.actions$ = actions$;
            this.dataService = dataService;
            this.specifications$ = i1$1.createEffect(function () { return _this.actions$.pipe(i1$1.ofType(loadSpecifications), operators.switchMap(function (_a) {
                var specificationIds = _a.specificationIds;
                return _this.dataService.getSpecifications().pipe(operators.map(function (data) { return (specificationIds ? data.filter(function (_a) {
                    var id = _a.id;
                    return !!id && specificationIds.has(id);
                }) : data); }), operators.map(function (data) { return loadSpecificationsSuccess({ specifications: data || [] }); }), operators.catchError(function () { return rxjs.of(loadSpecificationsFailed({ error: true })); }));
            })); });
        }
        return ActionsSpecificationsEffects;
    }());
    ActionsSpecificationsEffects.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsSpecificationsEffects, deps: [{ token: i1__namespace.Actions }, { token: i3__namespace.NotificationsPreferencesBaseDataService }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    ActionsSpecificationsEffects.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsSpecificationsEffects });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsSpecificationsEffects, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () { return [{ type: i1__namespace.Actions }, { type: i3__namespace.NotificationsPreferencesBaseDataService }]; } });

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from) {
        for (var i = 0, il = from.length, j = to.length; i < il; i++, j++)
            to[j] = from[i];
        return to;
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    var actionRecipeInitialState = {
        data: [],
        loading: false,
        error: undefined,
        specificationIds: undefined,
    };

    var _actionRecipesReducer = i1.createReducer(actionRecipeInitialState, i1.on(loadActionRecipes, function (state) { return (Object.assign(Object.assign({}, state), { loading: true, error: undefined, specificationIds: undefined })); }), i1.on(loadActionRecipesSuccess, function (state, action) { return (Object.assign(Object.assign({}, state), { data: action.actionRecipes, loading: false, specificationIds: action.specificationIds })); }), i1.on(loadActionRecipesFailed, function (state, action) { return (Object.assign(Object.assign({}, state), { error: action.error, loading: false })); }), i1.on(saveActionRecipeSuccess, function (state, action) {
        var recipe = action.actionRecipe;
        var data = state.data;
        var targetIndex = data.findIndex(function (_a) {
            var specificationId = _a.specificationId;
            return specificationId === recipe.specificationId;
        });
        var isAppend = targetIndex === -1;
        return Object.assign(Object.assign({}, state), { data: isAppend ? __spreadArray(__spreadArray([], __read(data)), [recipe]) : insertActionRecipe(data, recipe, targetIndex) });
    }), i1.on(saveActionRecipeFailure, function (state, action) {
        var recipe = action.actionRecipe;
        var data = state.data;
        var targetIndex = data.findIndex(function (_a) {
            var specificationId = _a.specificationId;
            return specificationId === recipe.specificationId;
        });
        var isAppend = targetIndex === -1;
        return Object.assign(Object.assign({}, state), { data: isAppend
                ? __spreadArray(__spreadArray([], __read(data)), [Object.assign(Object.assign({}, recipe), { active: false })]) : insertActionRecipe(data, Object.assign({}, data[targetIndex]), targetIndex) });
    }));
    function actionRecipesReducer(state, action) {
        return _actionRecipesReducer(state, action);
    }
    var insertActionRecipe = function (data, actionRecipe, index) { return __spreadArray(__spreadArray(__spreadArray([], __read(data.slice(0, index))), [
        actionRecipe
    ]), __read(data.slice(index + 1))); };

    var specificationsInitialState = {
        data: [],
        loading: false,
        error: false,
    };

    var initialState = {
        specifications: specificationsInitialState,
        actionRecipes: actionRecipeInitialState,
    };

    var _specificationsReducer = i1.createReducer(initialState.specifications, i1.on(loadSpecifications, function (state) { return (Object.assign(Object.assign({}, state), { loading: true, error: false })); }), i1.on(loadSpecificationsSuccess, function (state, action) { return (Object.assign(Object.assign({}, state), { data: action.specifications, loading: false })); }), i1.on(loadSpecificationsFailed, function (state, action) { return (Object.assign(Object.assign({}, state), { error: action.error, loading: false })); }));
    function specificationsReducer(state, action) {
        return _specificationsReducer(state, action);
    }

    var ActionsStoreModule = /** @class */ (function () {
        function ActionsStoreModule() {
        }
        return ActionsStoreModule;
    }());
    ActionsStoreModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsStoreModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ActionsStoreModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsStoreModule, imports: [common.CommonModule,
            i3.ActionsCommonModule, i1__namespace$1.StoreFeatureModule, i1__namespace$1.StoreFeatureModule, i1__namespace.EffectsFeatureModule] });
    ActionsStoreModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsStoreModule, providers: [ActionsRecipesEffects, ActionsSpecificationsEffects], imports: [[
                common.CommonModule,
                i3.ActionsCommonModule,
                i1.StoreModule.forFeature('actionRecipes', actionRecipesReducer),
                i1.StoreModule.forFeature('specifications', specificationsReducer),
                i1$1.EffectsModule.forFeature([ActionsRecipesEffects, ActionsSpecificationsEffects]),
            ]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsStoreModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        providers: [ActionsRecipesEffects, ActionsSpecificationsEffects],
                        imports: [
                            common.CommonModule,
                            i3.ActionsCommonModule,
                            i1.StoreModule.forFeature('actionRecipes', actionRecipesReducer),
                            i1.StoreModule.forFeature('specifications', specificationsReducer),
                            i1$1.EffectsModule.forFeature([ActionsRecipesEffects, ActionsSpecificationsEffects]),
                        ],
                    }]
            }] });

    var selectSpecificationsState = i1.createFeatureSelector('specifications');
    var specifications = i1.createSelector(selectSpecificationsState, function (state) { return state.data; });
    var specificationsLoading = i1.createSelector(selectSpecificationsState, function (state) { return state.loading; });
    var specificationsError = i1.createSelector(selectSpecificationsState, function (state) { return state.error; });

    var specificationsWithActionRecipe = i1.createSelector(specifications, actionRecipes, function (specs, recipes) { return i3.mergeSpecificationsWithActionRecipes(specs, recipes); });

    var ActionsStoreModel = /** @class */ (function () {
        function ActionsStoreModel(store) {
            this.store = store;
            /**
             * Stream of specifications.
             */
            this.specifications$ = this.store.select(specifications);
            /**
             * Stream of specifications loading flags.
             */
            this.specificationsLoading$ = this.store.select(specificationsLoading);
            /**
             * Stream of specifications errors.
             */
            this.specificationsError$ = this.store.select(specificationsError);
            /**
             * Stream of action recipes.
             */
            this.actionRecipes$ = this.store.select(actionRecipes);
            /**
             * Stream of action recipes loading flags.
             */
            this.actionRecipesLoading$ = this.store.select(actionRecipesLoading);
            /**
             * Stream of action recipes errors.
             */
            this.actionRecipesError$ = this.store.select(actionRecipesError);
            /**
             * Stream of specifications with action recipe.
             */
            this.specificationsWithActionRecipe$ = this.store.select(specificationsWithActionRecipe);
        }
        /**
         * Method to dispatch load specifications.
         *
         * @param payload - load specifications payload
         */
        ActionsStoreModel.prototype.dispatchLoadSpecifications = function (payload) {
            this.store.dispatch(loadSpecifications(payload));
        };
        /**
         * Method to dispatch load action recipes.
         *
         * @param payload - load specifications payload
         */
        ActionsStoreModel.prototype.dispatchLoadActionRecipes = function (payload) {
            this.store.dispatch(loadActionRecipes(payload));
        };
        /**
         * Method to dispatch save action recipe.
         *
         * @param payload - action recipe payload
         */
        ActionsStoreModel.prototype.dispatchSaveActionRecipe = function (payload) {
            this.store.dispatch(saveActionRecipe(payload));
        };
        return ActionsStoreModel;
    }());
    ActionsStoreModel.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsStoreModel, deps: [{ token: i1__namespace$1.Store }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    ActionsStoreModel.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsStoreModel });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsStoreModel, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () { return [{ type: i1__namespace$1.Store }]; } });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.ActionsStoreModel = ActionsStoreModel;
    exports.ActionsStoreModule = ActionsStoreModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
