// @backbase/product-summary-ui/esm2015/src/bb-product-summary-account-logo-ui/bb-product-summary-account-logo.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { IconModule } from '@backbase/ui-ang/icon';
import { ProductSummaryAccountLogoComponent, ProductSummaryAccountLogoCustomizableDirective, } from './bb-product-summary-account-logo.component';
import * as i0 from "@angular/core";
const uiModules = [IconModule];
const components = [ProductSummaryAccountLogoComponent];
const directives = [ProductSummaryAccountLogoCustomizableDirective];
/**
 * Exports a UI components for displaying account logo
 *
 * The following components are included in this module:
 *  - `ProductSummaryAccountLogoComponent`
 *
 * @see ProductSummaryAccountLogoComponent
 *
 * @usageNotes
 *
 * ### Include the exported widgets in your `AppModule`:
 *
 * ```typescript
 * @NgModule({
 *   imports: [
 *     ProductSummaryAccountLogoModule,
 *   ],
 *   ...
 * })
 * export class AppModule {}
 * ```
 */
export class ProductSummaryAccountLogoModule {
}
ProductSummaryAccountLogoModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryAccountLogoModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductSummaryAccountLogoModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryAccountLogoModule, declarations: [ProductSummaryAccountLogoComponent, ProductSummaryAccountLogoCustomizableDirective], imports: [CommonModule, IconModule], exports: [ProductSummaryAccountLogoComponent, ProductSummaryAccountLogoCustomizableDirective, IconModule] });
ProductSummaryAccountLogoModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryAccountLogoModule, imports: [[CommonModule, ...uiModules], IconModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryAccountLogoModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules],
                    declarations: [...components, ...directives],
                    exports: [...components, ...directives, ...uiModules],
                }]
        }] });
