// @backbase/product-summary-ui/esm2015/src/bb-product-summary-account-logo-ui/bb-product-summary-account-logo.component.js
import { Component, Directive, Input } from '@angular/core';
import { BbTemplate } from '@backbase/foundation-ang/core';
import { getProductKindIconName, ProductKindNames, toKebabCase } from '@backbase/product-summary-common-ang';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/icon";
import * as i2 from "@angular/common";
/**
 * A component for displaying an account logo.
 * In case the logo is not reachable a related icon is displayed basing on `productKindName`.
 *
 * @usageNotes
 *
 * ### Include this widget in your `AppModule`:
 *
 * ```typescript
 * @NgModule({
 *   imports: [
 *     ProductSummaryAccountLogoModule,
 *   ],
 *   ...
 * })
 * export class AppModule {}
 * ```
 *
 * ### Display details
 *
 * ```html
 * <bb-product-summary-account-logo
 *   [logoSrc]="externalBank.logo"
 *   [productKindName]="item.productKindName"
 *   [defaultProductKindName]="'Current Account'"
 * ></bb-product-summary-account-logo>
 * ```
 *
 * @ngModule ProductSummaryAccountLogoModule
 */
export class ProductSummaryAccountLogoComponent {
    constructor() {
        this.hostRef = this;
        /**
         * Source for bank logo.
         * Can be either a URL or base64 image.
         *
         * @default ''
         */
        this.logoSrc = '';
        /**
         * Product kind name to display a related icon.
         *
         * @default ''
         */
        this.productKindName = '';
        /**
         * Default product kind name to display icons
         */
        this.defaultProductKindName = ProductKindNames.CURRENT_ACCOUNT;
        /**
         * Sets icon size
         *
         * @default 'md'
         */
        this.iconSize = 'md';
        /**
         * Sets icon size
         *
         * @default 'md'
         */
        this.logoSize = 'md';
        /**
         * Indicates whether during loading of the image any errors appeared
         *
         * @default false
         */
        this.isImgError = false;
    }
    /**
     * Returns a related icon color for the given 'productKindName'
     */
    getProductKindIconColor() {
        const color = this.getProductKindIconName().slice(0, -1);
        return color === 'account' ? toKebabCase(this.defaultProductKindName) : color;
    }
    /**
     * Returns a related icon name for the given 'productKindName'
     */
    getProductKindIconName() {
        return getProductKindIconName(this.productKindName);
    }
    /**
     * Event handler for the `error` event of `img`
     * sets `isImgError` to `true`
     */
    onImgError() {
        this.isImgError = true;
    }
}
ProductSummaryAccountLogoComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryAccountLogoComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ProductSummaryAccountLogoComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ProductSummaryAccountLogoComponent, selector: "bb-product-summary-account-logo", inputs: { logoSrc: "logoSrc", productKindName: "productKindName", defaultProductKindName: "defaultProductKindName", iconColor: "iconColor", iconSize: "iconSize", logoSize: "logoSize" }, ngImport: i0, template: "<ng-container\n  bbProductSummaryAccountLogoCustomizable\n  [bbHostRef]=\"hostRef\"\n></ng-container>\n\n<ng-template\n  bbProductSummaryAccountLogoCustomizable\n  let-hostRef\n>\n  <ng-container *ngIf=\"hostRef.logoSrc && !hostRef.isImgError; else productKindIcon\">\n    <div class=\"bb-stack bb-icon bb-icon--{{hostRef.logoSize}} bb-icon__background\">\n      <img [src]=\"hostRef.logoSrc\"\n           data-role=\"product-summary-account-logo-img\"\n           aria-hidden=\"true\"\n           alt=\"\"\n           class=\"bb-product-summary-account-logo__img bb-icon__background--square\"\n           (error)=\"hostRef.onImgError()\"/>\n    </div>\n  </ng-container>\n  <ng-template #productKindIcon>\n    <bb-icon-ui data-role=\"product-summary-account-logo-icon\"\n                backgroundType=\"square\"\n                [size]=\"hostRef.iconSize\"\n                [color]=\"hostRef.iconColor || hostRef.getProductKindIconColor()\"\n                [name]=\"hostRef.getProductKindIconName()\">\n    </bb-icon-ui>\n  </ng-template>\n</ng-template>\n\n", components: [{ type: i0.forwardRef(function () { return i1.IconComponent; }), selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], directives: [{ type: i0.forwardRef(function () { return ProductSummaryAccountLogoCustomizableDirective; }), selector: "[bbProductSummaryAccountLogoCustomizable]" }, { type: i0.forwardRef(function () { return i2.NgIf; }), selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryAccountLogoComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-product-summary-account-logo',
                    templateUrl: 'bb-product-summary-account-logo.component.html',
                }]
        }], propDecorators: { logoSrc: [{
                type: Input
            }], productKindName: [{
                type: Input
            }], defaultProductKindName: [{
                type: Input
            }], iconColor: [{
                type: Input
            }], iconSize: [{
                type: Input
            }], logoSize: [{
                type: Input
            }] } });
/**
 * This is an extension slot to be used for customizing the logo
 *
 * @usageNotes
 *
 * ### Customize details
 *
 * ```html
 * <ng-template
 *   bbProductSummaryAccountLogoCustomizable
 *   let-hostRef>
 *   Your custom template here
 * </ng-template>
 * ```
 *
 * @NgModule ProductSummaryAccountDetailsWidgetModule
 */
export class ProductSummaryAccountLogoCustomizableDirective extends BbTemplate {
}
ProductSummaryAccountLogoCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryAccountLogoCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
ProductSummaryAccountLogoCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: ProductSummaryAccountLogoCustomizableDirective, selector: "[bbProductSummaryAccountLogoCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryAccountLogoCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbProductSummaryAccountLogoCustomizable]',
                }]
        }] });
