// @backbase/ui-ang/esm2015/rich-text-editor/bb-rich-text-editor-actions/rich-text-editor-image-action/rich-text-editor-image-action.component.js
import { Component } from '@angular/core';
import { RichTextEditorActionsComponent } from '../rich-text-editor-actions.component';
import * as i0 from "@angular/core";
import * as i1 from "../rich-text-editor-actions.service";
import * as i2 from "@backbase/ui-ang/icon";
import * as i3 from "@backbase/ui-ang/tooltip-directive";
import * as i4 from "@backbase/ui-ang/button";
export class RichTextEditorImageActionComponent extends RichTextEditorActionsComponent {
    constructor(ma) {
        super(ma);
    }
    onResult() {
        const url = window.prompt('Enter the image URL');
        if (url)
            super.result('insertImage', url);
    }
}
RichTextEditorImageActionComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RichTextEditorImageActionComponent, deps: [{ token: i1.RichTextEditorActionsService }], target: i0.ɵɵFactoryTarget.Component });
RichTextEditorImageActionComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: RichTextEditorImageActionComponent, selector: "bb-rich-text-editor-image-action", usesInheritance: true, ngImport: i0, template: "<span\n  bbTooltip=\"Image\"\n  i18n-bbTooltip=\"Image|Image action @@bb-rich-text-editor-ui.rich-text-editor-actions.image\"\n  container=\"body\"\n  triggers=\"hover focus\"\n  [placement]=\"actionsTooltipPlacement\"\n>\n  <button\n    bbButton\n    [circle]=\"true\"\n    color=\"link-dark\"\n    aria-label=\"Image\"\n    i18n-aria-label=\"@@bb-rich-text-editor-ui.rich-text-editor-actions.image\"\n    [disabled]=\"disabled\"\n    (click)=\"onResult()\"\n  >\n    <bb-icon-ui name=\"image\"></bb-icon-ui>\n  </button>\n</span>\n", components: [{ type: i2.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], directives: [{ type: i3.TooltipDirective, selector: "[bbTooltip]", inputs: ["bbTooltip", "triggers"] }, { type: i4.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RichTextEditorImageActionComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-rich-text-editor-image-action',
                    templateUrl: './rich-text-editor-image-action.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.RichTextEditorActionsService }]; } });
