// @backbase/messages-common/esm2015/src/components/upload-attachments/services/base-attachments.service.js
import { Subject } from 'rxjs';
import { takeUntil, tap } from 'rxjs/operators';
export class BaseUploadAttachmentsService {
}
export class FileAttachment {
    constructor(file, service) {
        this.service = service;
        /**
         * The loading flag.
         */
        this.loading = false;
        /**
         * Delete attachment subject that holds index
         */
        this.onDelete = new Subject();
        this.cancelSubject = new Subject();
        this.file = file;
        this.size = file.size;
        if (file.id) {
            this.id = file.id;
            this.name = file.filename;
        }
        else {
            this.name = file.name;
        }
    }
    /**
     * Delete attachment
     *
     * @param id Attachment id
     * @param index Attachment index
     */
    deleteAttachment(id, index) {
        if (this.id && this.id === id) {
            this.service.deleteAttachments([id]).subscribe();
        }
        else {
            this.cancelSubject.next();
        }
        this.onDelete.next(index);
    }
    /**
     * Upload attachments
     */
    uploadAttachment() {
        this.setLoading(true);
        return this.service.uploadAttachment(this.file).pipe(takeUntil(this.cancelSubject), tap(value => {
            this.id = value ? value.id : undefined;
            this.setLoading(false);
        }));
    }
    /**
     * Set loading flag
     *
     * @param flag Loaging status flag
     */
    setLoading(flag) {
        this.loading = flag;
    }
}
