// @backbase/ui-ang/esm2015/_example/payment-card-number-pipe/showcase/payment-card-number-pipe-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { PaymentCardNumberPipeShowcaseComponent } from './payment-card-number-pipe-showcase.component';
import { PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import { InputNumberModule } from '@backbase/ui-ang/input-number';
import { FormsModule } from '@angular/forms';
import * as i0 from "@angular/core";
export class PaymentCardNumberPipeShowcaseModule {
}
PaymentCardNumberPipeShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardNumberPipeShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PaymentCardNumberPipeShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardNumberPipeShowcaseModule, declarations: [PaymentCardNumberPipeShowcaseComponent], imports: [CommonModule, PaymentCardNumberModule, InputNumberModule, FormsModule] });
PaymentCardNumberPipeShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardNumberPipeShowcaseModule, imports: [[CommonModule, PaymentCardNumberModule, InputNumberModule, FormsModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardNumberPipeShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, PaymentCardNumberModule, InputNumberModule, FormsModule],
                    declarations: [PaymentCardNumberPipeShowcaseComponent],
                    entryComponents: [PaymentCardNumberPipeShowcaseComponent],
                }]
        }] });
