// @backbase/ui-ang/esm2015/_example/payment-card-number-pipe/showcase/payment-card-number-pipe-showcase.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/input-number";
import * as i2 from "@angular/forms";
import * as i3 from "@backbase/ui-ang/payment-card-number-pipe";
let PaymentCardNumberPipeShowcaseComponent = class PaymentCardNumberPipeShowcaseComponent {
    constructor() {
        this.maskStart = 0;
        this.maskEnd = 12;
        this._paymentCardNumberFormat = {};
    }
    get paymentCard() {
        return {
            name: 'Ricardo Peterson',
            number: '1234567891234567',
            vendor: 'mastercard',
            expirationDate: new Date(2021, 2),
        };
    }
    set paymentCardNumberFormat(val) {
        this._paymentCardNumberFormat = val;
    }
    get paymentCardNumberFormat() {
        return {
            length: 16,
            maskRange: [this.maskStart, this.maskEnd],
            segments: 4,
        };
    }
};
PaymentCardNumberPipeShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardNumberPipeShowcaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
PaymentCardNumberPipeShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: PaymentCardNumberPipeShowcaseComponent, selector: "bb-payment-card-number-pipe-ui-showcase", ngImport: i0, template: "<bb-input-number-ui\n  label=\"Mask Start\"\n  aria-label=\"Mask Start Index\"\n  placeholder=\"Mask Start Index\"\n  [(ngModel)]=\"maskStart\"\n>\n</bb-input-number-ui>\n<bb-input-number-ui label=\"Mask End\" aria-label=\"Mask End Index\" placeholder=\"Mask End Index\" [(ngModel)]=\"maskEnd\">\n</bb-input-number-ui>\n<hr />\n{{ paymentCard?.number | paymentCardNumber: paymentCardNumberFormat }}\n", components: [{ type: i1.InputNumberComponent, selector: "bb-input-number-ui", inputs: ["placeholder", "max", "min", "step", "readonly", "autocomplete", "ariaLabel", "allowNonDigits", "allowNegativeSign"] }], directives: [{ type: i2.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { type: i2.NgModel, selector: "[ngModel]:not([formControlName]):not([formControl])", inputs: ["name", "disabled", "ngModel", "ngModelOptions"], outputs: ["ngModelChange"], exportAs: ["ngModel"] }], pipes: { "paymentCardNumber": i3.PaymentCardNumberPipe } });
PaymentCardNumberPipeShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
], PaymentCardNumberPipeShowcaseComponent);
export { PaymentCardNumberPipeShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardNumberPipeShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-payment-card-number-pipe-ui-showcase',
                    templateUrl: './payment-card-number-pipe-showcase.component.html',
                }]
        }] });
