// @backbase/loans-collection-ui-ang/esm2015/lib/tabs/loans-tabs.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { LoansTabsComponent } from './loans-tabs.component';
import * as i0 from "@angular/core";
export class LoansTabsModule {
}
/** @nocollapse */ LoansTabsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoansTabsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoansTabsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoansTabsModule, declarations: [LoansTabsComponent], imports: [CommonModule, RouterModule], exports: [LoansTabsComponent] });
/** @nocollapse */ LoansTabsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoansTabsModule, imports: [[CommonModule, RouterModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoansTabsModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoansTabsComponent],
                    imports: [CommonModule, RouterModule],
                    exports: [LoansTabsComponent],
                }]
        }] });
