// @backbase/saving-goals-http-ang/esm2015/api/savinggoals.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const SavinggoalsHttpServiceDeleteIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/savinggoals/{id}",
        method: "DELETE",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const SavinggoalsHttpServiceGetIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/savinggoals/{id}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "1955e686-d31e-4216-b3dd-5d66161d536b",
                    "goalName": "Holiday to Hawaii",
                    "goalAmount": {
                        "amount": "2000.00",
                        "currencyCode": "EUR"
                    },
                    "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                    "amountProgress": {
                        "amount": "550.00",
                        "currencyCode": "EUR"
                    },
                    "progressPercentage": 27.5,
                    "performancePercentage": 27.5,
                    "timePercentage": 50.5,
                    "amountRemaining": {
                        "amount": "1450.00",
                        "currencyCode": "EUR"
                    },
                    "startDate": "2017-02-28",
                    "endDate": "2019-02-28",
                    "initialAmount": {
                        "amount": "10.00",
                        "currencyCode": "EUR"
                    },
                    "recipeId": "65b5c1b7-17e0-435f-a828-e2f6afec26a9"
                }
            },
            {
                status: 200,
                body: {
                    "arrangementId": "arrangementId",
                    "goalAmount": {
                        "amount": "amount",
                        "additions": {
                            "key": "additions"
                        },
                        "currencyCode": "currencyCode"
                    },
                    "additions": {
                        "key": "additions"
                    },
                    "endDate": "2000-01-23",
                    "recipeId": "recipeId",
                    "goalName": "goalName",
                    "amountRemaining": {
                        "amount": "amount",
                        "additions": {
                            "key": "additions"
                        },
                        "currencyCode": "currencyCode"
                    },
                    "initialAmount": {
                        "amount": "amount",
                        "additions": {
                            "key": "additions"
                        },
                        "currencyCode": "currencyCode"
                    },
                    "progressPercentage": 0.8008281904610115,
                    "performancePercentage": 6.027456183070403,
                    "amountProgress": {
                        "amount": "amount",
                        "additions": {
                            "key": "additions"
                        },
                        "currencyCode": "currencyCode"
                    },
                    "id": "id",
                    "timePercentage": 1.4658129805029452,
                    "startDate": "2000-01-23"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const SavinggoalsHttpServiceGetSavingGoalsMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/savinggoals",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "1955e686-d31e-4216-b3dd-5d66161d536b",
                        "goalName": "Holiday to Hawaii",
                        "goalAmount": {
                            "amount": "2000.00",
                            "currencyCode": "EUR"
                        },
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "amountProgress": {
                            "amount": "550.00",
                            "currencyCode": "EUR"
                        },
                        "progressPercentage": 27.5,
                        "performancePercentage": 27.5,
                        "timePercentage": 50.5,
                        "amountRemaining": {
                            "amount": "1450.00",
                            "currencyCode": "EUR"
                        },
                        "startDate": "2017-02-28",
                        "endDate": "2019-02-28",
                        "initialAmount": {
                            "amount": "10.00",
                            "currencyCode": "EUR"
                        },
                        "recipeId": "65b5c1b7-17e0-435f-a828-e2f6afec26a9"
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const SavinggoalsHttpServicePostIdstandingorderMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/savinggoals/{id}/standing-order",
        method: "POST",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const SavinggoalsHttpServicePostSavingGoalsMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/savinggoals",
        method: "POST",
        responses: [
            {
                status: 201,
                body: {
                    "id": "0955e686-d31e-4216-b3dd-5d77161d536d"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const SavinggoalsHttpServicePutIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/savinggoals/{id}",
        method: "PUT",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const SavinggoalsHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v2/savinggoals/{id}",
        method: "DELETE",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/savinggoals/{id}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "1955e686-d31e-4216-b3dd-5d66161d536b",
                    "goalName": "Holiday to Hawaii",
                    "goalAmount": {
                        "amount": "2000.00",
                        "currencyCode": "EUR"
                    },
                    "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                    "amountProgress": {
                        "amount": "550.00",
                        "currencyCode": "EUR"
                    },
                    "progressPercentage": 27.5,
                    "performancePercentage": 27.5,
                    "timePercentage": 50.5,
                    "amountRemaining": {
                        "amount": "1450.00",
                        "currencyCode": "EUR"
                    },
                    "startDate": "2017-02-28",
                    "endDate": "2019-02-28",
                    "initialAmount": {
                        "amount": "10.00",
                        "currencyCode": "EUR"
                    },
                    "recipeId": "65b5c1b7-17e0-435f-a828-e2f6afec26a9"
                }
            },
            {
                status: 200,
                body: {
                    "arrangementId": "arrangementId",
                    "goalAmount": {
                        "amount": "amount",
                        "additions": {
                            "key": "additions"
                        },
                        "currencyCode": "currencyCode"
                    },
                    "additions": {
                        "key": "additions"
                    },
                    "endDate": "2000-01-23",
                    "recipeId": "recipeId",
                    "goalName": "goalName",
                    "amountRemaining": {
                        "amount": "amount",
                        "additions": {
                            "key": "additions"
                        },
                        "currencyCode": "currencyCode"
                    },
                    "initialAmount": {
                        "amount": "amount",
                        "additions": {
                            "key": "additions"
                        },
                        "currencyCode": "currencyCode"
                    },
                    "progressPercentage": 0.8008281904610115,
                    "performancePercentage": 6.027456183070403,
                    "amountProgress": {
                        "amount": "amount",
                        "additions": {
                            "key": "additions"
                        },
                        "currencyCode": "currencyCode"
                    },
                    "id": "id",
                    "timePercentage": 1.4658129805029452,
                    "startDate": "2000-01-23"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/savinggoals",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "1955e686-d31e-4216-b3dd-5d66161d536b",
                        "goalName": "Holiday to Hawaii",
                        "goalAmount": {
                            "amount": "2000.00",
                            "currencyCode": "EUR"
                        },
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "amountProgress": {
                            "amount": "550.00",
                            "currencyCode": "EUR"
                        },
                        "progressPercentage": 27.5,
                        "performancePercentage": 27.5,
                        "timePercentage": 50.5,
                        "amountRemaining": {
                            "amount": "1450.00",
                            "currencyCode": "EUR"
                        },
                        "startDate": "2017-02-28",
                        "endDate": "2019-02-28",
                        "initialAmount": {
                            "amount": "10.00",
                            "currencyCode": "EUR"
                        },
                        "recipeId": "65b5c1b7-17e0-435f-a828-e2f6afec26a9"
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/savinggoals/{id}/standing-order",
        method: "POST",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/savinggoals",
        method: "POST",
        responses: [
            {
                status: 201,
                body: {
                    "id": "0955e686-d31e-4216-b3dd-5d77161d536d"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/savinggoals/{id}",
        method: "PUT",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
]);
