// @backbase/manage-pockets-journey-ang/esm2015/src/manage-pockets-common/model/payment.model.js
import { pocketFormatDate } from '../helpers/formatting.helper';
import { TransferTypeEnum } from './pocket-item.model';
export var PaymentTypeEnum;
(function (PaymentTypeEnum) {
    PaymentTypeEnum["POCKET_TRANSFER"] = "POCKET_TRANSFER";
})(PaymentTypeEnum || (PaymentTypeEnum = {}));
export var PaymentRoleEnum;
(function (PaymentRoleEnum) {
    PaymentRoleEnum["CREDITOR"] = "CREDITOR";
    PaymentRoleEnum["DEBTOR"] = "DEBTOR";
})(PaymentRoleEnum || (PaymentRoleEnum = {}));
export var SchemeNameTypeEnum;
(function (SchemeNameTypeEnum) {
    SchemeNameTypeEnum["IBAN"] = "IBAN";
    SchemeNameTypeEnum["BBAN"] = "BBAN";
    SchemeNameTypeEnum["ID"] = "ID";
    SchemeNameTypeEnum["EXTERNAL_ID"] = "EXTERNAL_ID";
    SchemeNameTypeEnum["EMAIL"] = "EMAIL";
    SchemeNameTypeEnum["MOBILE"] = "MOBILE";
})(SchemeNameTypeEnum || (SchemeNameTypeEnum = {}));
export var PaymentOrderNamespace;
(function (PaymentOrderNamespace) {
    function buildPaymentOrderPostRequest(paymentInfo) {
        var _a, _b, _c, _d, _e, _f, _g;
        return {
            paymentOrdersPost: {
                paymentType: PaymentTypeEnum.POCKET_TRANSFER,
                batchBooking: false,
                requestedExecutionDate: pocketFormatDate(),
                originatorAccount: {
                    identification: {
                        identification: ((_a = paymentInfo.account) === null || _a === void 0 ? void 0 : _a.id) || '',
                        schemeName: SchemeNameTypeEnum.ID,
                    },
                },
                transferTransactionInformation: {
                    counterparty: {
                        name: ((_c = (_b = paymentInfo.pocketInfo) === null || _b === void 0 ? void 0 : _b.pocket) === null || _c === void 0 ? void 0 : _c.name) || '',
                        role: ((_d = paymentInfo.pocketInfo) === null || _d === void 0 ? void 0 : _d.transferType) === TransferTypeEnum.ADD
                            ? PaymentRoleEnum.CREDITOR
                            : PaymentRoleEnum.DEBTOR,
                    },
                    counterpartyAccount: {
                        identification: {
                            identification: ((_f = (_e = paymentInfo.pocketInfo) === null || _e === void 0 ? void 0 : _e.pocket) === null || _f === void 0 ? void 0 : _f.arrangementId) || '',
                            schemeName: SchemeNameTypeEnum.ID,
                        },
                    },
                    instructedAmount: {
                        amount: ((_g = paymentInfo === null || paymentInfo === void 0 ? void 0 : paymentInfo.amount) === null || _g === void 0 ? void 0 : _g.toString()) || '',
                        currencyCode: (paymentInfo === null || paymentInfo === void 0 ? void 0 : paymentInfo.currencyCode) || '',
                    },
                },
            },
        };
    }
    PaymentOrderNamespace.buildPaymentOrderPostRequest = buildPaymentOrderPostRequest;
})(PaymentOrderNamespace || (PaymentOrderNamespace = {}));
