// @backbase/manage-pockets-journey-ang/esm2015/src/manage-pockets-common/model/pocket-item.model.js
import { __rest } from "tslib";
import { pocketFormatDate } from '../helpers/formatting.helper';
export var TransferTypeEnum;
(function (TransferTypeEnum) {
    TransferTypeEnum["ADD"] = "ADD";
    TransferTypeEnum["WITHDRAW"] = "WITHDRAW";
})(TransferTypeEnum || (TransferTypeEnum = {}));
export var PocketNamespace;
(function (PocketNamespace) {
    function buildNewPocketPostRequest(pocketItem) {
        const { name = '', icon = '', goal = {} } = pocketItem, others = __rest(pocketItem, ["name", "icon", "goal"]);
        return {
            pocketPostRequest: Object.assign({ name,
                icon, goal: formatGoal(goal) }, others),
        };
    }
    PocketNamespace.buildNewPocketPostRequest = buildNewPocketPostRequest;
    function formatGoal(pocketGoal) {
        var _a, _b;
        const amount = ((_a = pocketGoal === null || pocketGoal === void 0 ? void 0 : pocketGoal.amountCurrency) === null || _a === void 0 ? void 0 : _a.amount) || undefined;
        const currencyCode = ((_b = pocketGoal === null || pocketGoal === void 0 ? void 0 : pocketGoal.amountCurrency) === null || _b === void 0 ? void 0 : _b.currencyCode) || undefined;
        const deadline = (pocketGoal === null || pocketGoal === void 0 ? void 0 : pocketGoal.deadline) ? pocketFormatDate(new Date(pocketGoal.deadline)) : undefined;
        const amountCurrency = amount && currencyCode ? { amount, currencyCode } : undefined;
        return Object.assign(Object.assign({}, (!!deadline ? { deadline } : {})), (!!amountCurrency ? { amountCurrency } : {}));
    }
})(PocketNamespace || (PocketNamespace = {}));
