// @backbase/payment-orders-ang/esm2015/lib/ui/components/payments-filter/payments-filter.module.js
import { NgModule } from '@angular/core';
import { CommonModule, DatePipe } from '@angular/common';
import { PaymentsFilterComponent, PaymentsFilterCustomizableDirective, } from './payments-filter.component';
import { InputNumberModule } from '@backbase/ui-ang/input-number';
import { InputDatepickerModule } from '@backbase/ui-ang/input-datepicker';
import { DropdownMultiSelectModule } from '@backbase/ui-ang/dropdown-multi-select';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { ButtonModule } from '@backbase/ui-ang/button';
import { ReactiveFormsModule } from '@angular/forms';
import * as i0 from "@angular/core";
('@backbase/payord-form-builder-ang');
const uiModules = [
    ButtonModule,
    DropdownMultiSelectModule,
    InputDatepickerModule,
    InputValidationMessageModule,
    InputNumberModule,
];
export class PaymentsFilterModule {
}
/** @nocollapse */ PaymentsFilterModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentsFilterModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ PaymentsFilterModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentsFilterModule, declarations: [PaymentsFilterComponent, PaymentsFilterCustomizableDirective], imports: [CommonModule, ReactiveFormsModule, ButtonModule,
        DropdownMultiSelectModule,
        InputDatepickerModule,
        InputValidationMessageModule,
        InputNumberModule], exports: [PaymentsFilterComponent, PaymentsFilterCustomizableDirective] });
/** @nocollapse */ PaymentsFilterModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentsFilterModule, providers: [DatePipe], imports: [[CommonModule, ReactiveFormsModule, ...uiModules]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentsFilterModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [PaymentsFilterComponent, PaymentsFilterCustomizableDirective],
                    imports: [CommonModule, ReactiveFormsModule, ...uiModules],
                    exports: [PaymentsFilterComponent, PaymentsFilterCustomizableDirective],
                    providers: [DatePipe],
                }]
        }] });
