// @backbase/manage-pockets-journey-ang/esm2015/src/manage-pockets-create/manage-pockets-create.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { AmountInputModule } from '@backbase/ui-ang/amount-input';
import { IconModule } from '@backbase/ui-ang/icon';
import { InputDatepickerModule } from '@backbase/ui-ang/input-datepicker';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { ModalModule } from '@backbase/ui-ang/modal';
import { StepperModule } from '@backbase/ui-ang/stepper';
import { ManagePocketsCommonModule } from '../manage-pockets-common/manage-pockets-common.module';
import { ManagePocketsConfirmationComponent } from './components/manage-pockets-confirmation/manage-pockets-confirmation.component';
import { ManagePocketsCreateWrapperComponent } from './components/manage-pockets-create-wrapper/manage-pockets-create-wrapper.component';
import { ManagePocketsCreateComponent } from './components/manage-pockets-create/manage-pockets-create.component';
import { ManagePocketsDetailsComponent } from './components/manage-pockets-details/manage-pockets-details.component';
import { ManagePocketsGoalsComponent } from './components/manage-pockets-goals/manage-pockets-goals.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const components = [
    ManagePocketsCreateComponent,
    ManagePocketsCreateWrapperComponent,
    ManagePocketsDetailsComponent,
    ManagePocketsGoalsComponent,
    ManagePocketsConfirmationComponent,
];
const modules = [
    AmountInputModule,
    FormsModule,
    IconModule,
    InputDatepickerModule,
    InputTextModule,
    InputValidationMessageModule,
    LoadButtonModule,
    ModalModule,
    ReactiveFormsModule,
    StepperModule,
];
export class ManagePocketsCreateModule {
}
ManagePocketsCreateModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsCreateModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ManagePocketsCreateModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsCreateModule, declarations: [ManagePocketsCreateComponent,
        ManagePocketsCreateWrapperComponent,
        ManagePocketsDetailsComponent,
        ManagePocketsGoalsComponent,
        ManagePocketsConfirmationComponent], imports: [CommonModule,
        ManagePocketsCommonModule, AmountInputModule,
        FormsModule,
        IconModule,
        InputDatepickerModule,
        InputTextModule,
        InputValidationMessageModule,
        LoadButtonModule,
        ModalModule,
        ReactiveFormsModule,
        StepperModule, i1.BackbaseCoreModule], exports: [ManagePocketsCreateComponent,
        ManagePocketsCreateWrapperComponent,
        ManagePocketsDetailsComponent,
        ManagePocketsGoalsComponent,
        ManagePocketsConfirmationComponent, AmountInputModule,
        FormsModule,
        IconModule,
        InputDatepickerModule,
        InputTextModule,
        InputValidationMessageModule,
        LoadButtonModule,
        ModalModule,
        ReactiveFormsModule,
        StepperModule] });
ManagePocketsCreateModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsCreateModule, imports: [[
            CommonModule,
            ManagePocketsCommonModule,
            ...modules,
            BackbaseCoreModule.withConfig({
                classMap: { ManagePocketsCreateWrapperComponent },
            }),
        ], AmountInputModule,
        FormsModule,
        IconModule,
        InputDatepickerModule,
        InputTextModule,
        InputValidationMessageModule,
        LoadButtonModule,
        ModalModule,
        ReactiveFormsModule,
        StepperModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsCreateModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components],
                    imports: [
                        CommonModule,
                        ManagePocketsCommonModule,
                        ...modules,
                        BackbaseCoreModule.withConfig({
                            classMap: { ManagePocketsCreateWrapperComponent },
                        }),
                    ],
                    exports: [...components, ...modules],
                }]
        }] });
