// @backbase/financial-institution-manager-http-ang/bundles/backbase-financial-institution-manager-http-ang.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common/http'), require('@backbase/foundation-ang/data-http')) :
    typeof define === 'function' && define.amd ? define('@backbase/financial-institution-manager-http-ang', ['exports', '@angular/core', '@angular/common/http', '@backbase/foundation-ang/data-http'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["financial-institution-manager-http-ang"] = {}), global.ng.core, global.ng.common.http, global.i2));
})(this, (function (exports, i0, i1, i2) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);

    var CredentialType = {
        TEXT: 'TEXT',
        PASSWORD: 'PASSWORD'
    };

    var FINANCIAL_INSTITUTION_MANAGER_BASE_PATH = new i0.InjectionToken('FINANCIAL_INSTITUTION_MANAGER_BASE_PATH');
    var COLLECTION_FORMATS = {
        'csv': ',',
        'tsv': '   ',
        'ssv': ' ',
        'pipes': '|'
    };

    var FinancialInstitutionManagerConfiguration = /** @class */ (function () {
        function FinancialInstitutionManagerConfiguration(configurationParameters) {
            if (configurationParameters === void 0) { configurationParameters = {}; }
            this.apiKeys = configurationParameters.apiKeys;
            this.username = configurationParameters.username;
            this.password = configurationParameters.password;
            this.accessToken = configurationParameters.accessToken;
            this.basePath = configurationParameters.basePath;
            this.withCredentials = configurationParameters.withCredentials;
            this.encoder = configurationParameters.encoder;
            if (configurationParameters.credentials) {
                this.credentials = configurationParameters.credentials;
            }
            else {
                this.credentials = {};
            }
        }
        FinancialInstitutionManagerConfiguration.prototype.selectHeaderContentType = function (contentTypes) {
            var _this = this;
            if (contentTypes.length === 0) {
                return undefined;
            }
            var type = contentTypes.find(function (x) { return _this.isJsonMime(x); });
            if (type === undefined) {
                return contentTypes[0];
            }
            return type;
        };
        FinancialInstitutionManagerConfiguration.prototype.selectHeaderAccept = function (accepts) {
            var _this = this;
            if (accepts.length === 0) {
                return undefined;
            }
            var type = accepts.find(function (x) { return _this.isJsonMime(x); });
            if (type === undefined) {
                return accepts[0];
            }
            return type;
        };
        FinancialInstitutionManagerConfiguration.prototype.isJsonMime = function (mime) {
            var jsonMime = new RegExp('^(application\/json|[^;/ \t]+\/[^;/ \t]+[+]json)[ \t]*(;.*)?$', 'i');
            return mime !== null && (jsonMime.test(mime) || mime.toLowerCase() === 'application/json-patch+json');
        };
        FinancialInstitutionManagerConfiguration.prototype.lookupCredential = function (key) {
            var value = this.credentials[key];
            return typeof value === 'function'
                ? value()
                : value;
        };
        return FinancialInstitutionManagerConfiguration;
    }());
    var CONFIG_TOKEN = new i0.InjectionToken('api-data-module FinancialInstitutionManagerConfiguration');

    var FinancialInstitutionManagerApiModule = /** @class */ (function () {
        function FinancialInstitutionManagerApiModule(parentModule, http, dataModulesManager, config) {
            if (parentModule) {
                throw new Error('FinancialInstitutionManagerApiModule is already loaded. Import in your base AppModule only.');
            }
            if (!http) {
                throw new Error('You need to import the HttpClientModule in your AppModule! \n' +
                    'See also https://github.com/angular/angular/issues/20575');
            }
            if (dataModulesManager) {
                dataModulesManager.setModuleConfig(CONFIG_TOKEN, {
                    apiRoot: '',
                    servicePath: config.basePath || '',
                    headers: {},
                });
            }
        }
        FinancialInstitutionManagerApiModule.forRoot = function (configurationFactory) {
            return {
                ngModule: FinancialInstitutionManagerApiModule,
                providers: [{ provide: FinancialInstitutionManagerConfiguration, useFactory: configurationFactory }]
            };
        };
        return FinancialInstitutionManagerApiModule;
    }());
    FinancialInstitutionManagerApiModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: FinancialInstitutionManagerApiModule, deps: [{ token: FinancialInstitutionManagerApiModule, optional: true, skipSelf: true }, { token: i1__namespace.HttpClient, optional: true }, { token: i2__namespace.DataModulesManager, optional: true }, { token: FinancialInstitutionManagerConfiguration }], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    FinancialInstitutionManagerApiModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: FinancialInstitutionManagerApiModule });
    FinancialInstitutionManagerApiModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: FinancialInstitutionManagerApiModule, providers: [], imports: [[]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: FinancialInstitutionManagerApiModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [],
                        declarations: [],
                        exports: [],
                        providers: []
                    }]
            }], ctorParameters: function () {
            return [{ type: FinancialInstitutionManagerApiModule, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.SkipSelf
                        }] }, { type: i1__namespace.HttpClient, decorators: [{
                            type: i0.Optional
                        }] }, { type: i2__namespace.DataModulesManager, decorators: [{
                            type: i0.Optional
                        }] }, { type: FinancialInstitutionManagerConfiguration }];
        } });

    var CustomHttpParameterCodec = /** @class */ (function () {
        function CustomHttpParameterCodec() {
        }
        CustomHttpParameterCodec.prototype.encodeKey = function (k) {
            return encodeURIComponent(k);
        };
        CustomHttpParameterCodec.prototype.encodeValue = function (v) {
            return encodeURIComponent(v);
        };
        CustomHttpParameterCodec.prototype.decodeKey = function (k) {
            return decodeURIComponent(k);
        };
        CustomHttpParameterCodec.prototype.decodeValue = function (v) {
            return decodeURIComponent(v);
        };
        return CustomHttpParameterCodec;
    }());

    var FinancialInstitutionManagerClientHttpService = /** @class */ (function () {
        function FinancialInstitutionManagerClientHttpService(httpClient, basePath, configuration) {
            this.httpClient = httpClient;
            this.basePath = '';
            this.defaultHeaders = new i1.HttpHeaders();
            this.configuration = new FinancialInstitutionManagerConfiguration();
            if (configuration) {
                this.configuration = configuration;
            }
            if (typeof this.configuration.basePath !== 'string') {
                if (typeof basePath !== 'string') {
                    basePath = this.basePath;
                }
                this.configuration.basePath = basePath;
            }
            this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
        }
        FinancialInstitutionManagerClientHttpService.prototype.addToHttpParams = function (httpParams, value, key) {
            if (typeof value === "object" && value instanceof Date === false) {
                httpParams = this.addToHttpParamsRecursive(httpParams, value);
            }
            else {
                httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
            }
            return httpParams;
        };
        FinancialInstitutionManagerClientHttpService.prototype.addToHttpParamsRecursive = function (httpParams, value, key) {
            var _this = this;
            if (value == null) {
                return httpParams;
            }
            if (typeof value === "object") {
                if (Array.isArray(value)) {
                    value.forEach(function (elem) { return httpParams = _this.addToHttpParamsRecursive(httpParams, elem, key); });
                }
                else if (value instanceof Date) {
                    if (key != null) {
                        httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                    }
                    else {
                        throw Error("key may not be null if value is Date");
                    }
                }
                else {
                    Object.keys(value).forEach(function (k) { return httpParams = _this.addToHttpParamsRecursive(httpParams, value[k], key != null ? key + "." + k : k); });
                }
            }
            else if (key != null) {
                httpParams = httpParams.append(key, value);
            }
            else {
                throw Error("key may not be null if value is not object or array");
            }
            return httpParams;
        };
        FinancialInstitutionManagerClientHttpService.prototype.getFinancialInstitutions = function (requestParameters, observe, reportProgress, options) {
            var _this = this;
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _ids = requestParameters["ids"];
            var _from = requestParameters["from"];
            var _cursor = requestParameters["cursor"];
            var _size = requestParameters["size"];
            var queryParameters = new i1.HttpParams({ encoder: this.encoder });
            if (_ids) {
                _ids.forEach(function (element) {
                    queryParameters = _this.addToHttpParams(queryParameters, element, 'ids');
                });
            }
            if (_from !== undefined && _from !== null) {
                queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
            }
            if (_cursor !== undefined && _cursor !== null) {
                queryParameters = this.addToHttpParams(queryParameters, _cursor, 'cursor');
            }
            if (_size !== undefined && _size !== null) {
                queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
            }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.get(this.configuration.basePath + "/client-api/v1/financial-institutions", {
                params: queryParameters,
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        FinancialInstitutionManagerClientHttpService.prototype.getFinancialInstitutionsUrl = function (requestParameters) {
            var _this = this;
            var _ids = requestParameters["ids"];
            var _from = requestParameters["from"];
            var _cursor = requestParameters["cursor"];
            var _size = requestParameters["size"];
            var queryParameters = new i1.HttpParams({ encoder: this.encoder });
            if (_ids) {
                _ids.forEach(function (element) {
                    queryParameters = _this.addToHttpParams(queryParameters, element, 'ids');
                });
            }
            if (_from !== undefined && _from !== null) {
                queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
            }
            if (_cursor !== undefined && _cursor !== null) {
                queryParameters = this.addToHttpParams(queryParameters, _cursor, 'cursor');
            }
            if (_size !== undefined && _size !== null) {
                queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
            }
            var queryString = queryParameters.toString();
            return this.configuration.basePath + "/client-api/v1/financial-institutions" + (queryString ? "?" + queryString : '');
        };
        return FinancialInstitutionManagerClientHttpService;
    }());
    FinancialInstitutionManagerClientHttpService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: FinancialInstitutionManagerClientHttpService, deps: [{ token: i1__namespace.HttpClient }, { token: FINANCIAL_INSTITUTION_MANAGER_BASE_PATH, optional: true }, { token: FinancialInstitutionManagerConfiguration, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    FinancialInstitutionManagerClientHttpService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: FinancialInstitutionManagerClientHttpService, providedIn: 'root' });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: FinancialInstitutionManagerClientHttpService, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root'
                    }]
            }], ctorParameters: function () {
            return [{ type: i1__namespace.HttpClient }, { type: undefined, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.Inject,
                            args: [FINANCIAL_INSTITUTION_MANAGER_BASE_PATH]
                        }] }, { type: FinancialInstitutionManagerConfiguration, decorators: [{
                            type: i0.Optional
                        }] }];
        } });

    var FinancialInstitutionManagerClientHttpServiceGetFinancialInstitutionsMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v1/financial-institutions",
            method: "GET",
            responses: [
                {
                    status: 200,
                    body: {
                        "financialInstitutions": [{
                                "id": 1,
                                "externalId": "11da7d70-ed15-11ea-adc1-0242ac120002",
                                "name": "ABN Amro",
                                "logo": "https://landportal.org/sites/landportal.org/files/styles/220heightmax/public/abn-amro-logo-vertikaal.png",
                                "credentials": [{
                                        "name": "username",
                                        "label": "Username",
                                        "type": "TEXT"
                                    }, {
                                        "name": "password",
                                        "label": "Password",
                                        "type": "PASSWORD"
                                    }]
                            }, {
                                "id": 2,
                                "externalId": "b149f34b-de04-4d79-83f6-9605d698a7b5",
                                "name": "ING",
                                "logo": "https://www.ing.com/static/ingdotcompresentation/static/images/ING_lion_RGB_200x200.png",
                                "credentials": [{
                                        "name": "username",
                                        "label": "Username",
                                        "type": "TEXT"
                                    }, {
                                        "name": "password",
                                        "label": "Password",
                                        "type": "PASSWORD"
                                    }]
                            }]
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 500,
                    body: {
                        "message": "Description of error"
                    }
                },
            ]
        }]);
    var FinancialInstitutionManagerClientHttpServiceMocksProvider = i2.createMocks([
        {
            urlPattern: "/client-api/v1/financial-institutions",
            method: "GET",
            responses: [
                {
                    status: 200,
                    body: {
                        "financialInstitutions": [{
                                "id": 1,
                                "externalId": "11da7d70-ed15-11ea-adc1-0242ac120002",
                                "name": "ABN Amro",
                                "logo": "https://landportal.org/sites/landportal.org/files/styles/220heightmax/public/abn-amro-logo-vertikaal.png",
                                "credentials": [{
                                        "name": "username",
                                        "label": "Username",
                                        "type": "TEXT"
                                    }, {
                                        "name": "password",
                                        "label": "Password",
                                        "type": "PASSWORD"
                                    }]
                            }, {
                                "id": 2,
                                "externalId": "b149f34b-de04-4d79-83f6-9605d698a7b5",
                                "name": "ING",
                                "logo": "https://www.ing.com/static/ingdotcompresentation/static/images/ING_lion_RGB_200x200.png",
                                "credentials": [{
                                        "name": "username",
                                        "label": "Username",
                                        "type": "TEXT"
                                    }, {
                                        "name": "password",
                                        "label": "Password",
                                        "type": "PASSWORD"
                                    }]
                            }]
                    }
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 500,
                    body: {
                        "message": "Description of error"
                    }
                },
            ]
        },
    ]);

    exports.COLLECTION_FORMATS = COLLECTION_FORMATS;
    exports.CONFIG_TOKEN = CONFIG_TOKEN;
    exports.CredentialType = CredentialType;
    exports.FINANCIAL_INSTITUTION_MANAGER_BASE_PATH = FINANCIAL_INSTITUTION_MANAGER_BASE_PATH;
    exports.FinancialInstitutionManagerApiModule = FinancialInstitutionManagerApiModule;
    exports.FinancialInstitutionManagerClientHttpService = FinancialInstitutionManagerClientHttpService;
    exports.FinancialInstitutionManagerClientHttpServiceGetFinancialInstitutionsMocksProvider = FinancialInstitutionManagerClientHttpServiceGetFinancialInstitutionsMocksProvider;
    exports.FinancialInstitutionManagerClientHttpServiceMocksProvider = FinancialInstitutionManagerClientHttpServiceMocksProvider;
    exports.FinancialInstitutionManagerConfiguration = FinancialInstitutionManagerConfiguration;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
