// @backbase/budget-widget-ang/esm2015/src/budget-card/budget-card-info.component.js
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/amount";
import * as i2 from "@angular/common";
export class BudgetCardInfoComponent {
    constructor() {
        /**
         * Sets the percentage of the safe zone limit of the chart.
         */
        this.safeZoneLimit = 0;
    }
    ngOnInit() {
        if (this.data) {
            if (this.data.available) {
                this.data.available.amount = Math.round(parseFloat(this.data.available.amount)).toString();
            }
            if (this.data.spent) {
                this.data.spent.amount = Math.round(parseFloat(this.data.spent.amount)).toString();
            }
            if (this.data.spendingLimit) {
                this.data.spendingLimit.amount = Math.round(parseFloat(this.data.spendingLimit.amount)).toString();
            }
        }
    }
}
BudgetCardInfoComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetCardInfoComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
BudgetCardInfoComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: BudgetCardInfoComponent, selector: "bb-budget-card-info-ui", inputs: { data: "data", safeZoneLimit: "safeZoneLimit" }, ngImport: i0, template: "<div class=\"bb-stack bb-stack--center\">\n  <div *ngIf=\"data?.spent as spent\" class=\"bb-stack__item bb-stack__item--spacing-xl bb-text-align-center\">\n    <div i18n=\"Spent|Spent card info label@@budget.cardInfo.spent.label\">\n      Spent\n    </div>\n    <bb-amount-ui\n      class=\"bb-highlight\"\n      [amount]=\"spent.amount\"\n      [currency]=\"spent.currencyCode\"\n      [decimalPlaces]=\"0\"\n    ></bb-amount-ui>\n  </div>\n\n  <div *ngIf=\"data?.spendingLimit as spendingLimit\" class=\"bb-stack__item bb-text-align-center\">\n    <div i18n=\"Limit|Limit card info label@@budget.cardInfo.budgetLimit.label\">\n      Limit\n    </div>\n    <bb-amount-ui\n      class=\"bb-highlight\"\n      [amount]=\"spendingLimit.amount\"\n      [currency]=\"spendingLimit.currencyCode\"\n      [decimalPlaces]=\"0\"\n    ></bb-amount-ui>\n  </div>\n</div>\n", components: [{ type: i1.AmountComponent, selector: "bb-amount-ui", inputs: ["currency", "showPlusSign", "mapCurrency", "showPercent", "abbreviate", "decimalPlaces", "trailingZeroes", "amount"] }], directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetCardInfoComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-budget-card-info-ui',
                    templateUrl: './budget-card-info.component.html',
                }]
        }], propDecorators: { data: [{
                type: Input
            }], safeZoneLimit: [{
                type: Input
            }] } });
