// @backbase/user-manage-profile-widget-ang/esm2015/src/user-profile-forms/user-profile-forms.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { ButtonModule } from '@backbase/ui-ang/button';
import { IconModule } from '@backbase/ui-ang/icon';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { UserEmailAddressFormComponent, UserEmailAddressFormCustomizableDirective, UserEmailAddressFormErrorsCustomizableDirective, } from './user-email-address-form.component';
import { UserPhoneNumberFormComponent, UserPhoneNumberFormCustomizableDirective, UserPhoneNumberFormErrorsCustomizableDirective, } from './user-phone-number-form.component';
import { UserPostalAddressFormComponent, UserPostalAddressFormCustomizableDirective, UserPostalAddressFormErrorsCustomizableDirective, } from './user-postal-address-form.component';
import { UserProfileFormErrorsModule } from './user-profile-form-errors/user-profile-form-errors.module';
import { UserProfileFormBaseInputsModule } from './user-profile-form-inputs/base-inputs/user-profile-form-base-inputs.module';
import { UserProfileFormEmailAddressInputsModule } from './user-profile-form-inputs/email-address-inputs/user-profile-form-email-address-inputs.module';
import { UserProfileFormPhoneNumberInputsModule } from './user-profile-form-inputs/phone-number-inputs/user-profile-form-phone-number-inputs.module';
import { UserProfileFormPostalAddressInputsModule } from './user-profile-form-inputs/postal-address-inputs/user-profile-form-postal-address-inputs.module';
import { UserProfileFormModalsModule } from './user-profile-form-modals/user-profile-form-modals.module';
import * as i0 from "@angular/core";
const uiModules = [ButtonModule, LoadButtonModule, IconModule, ReactiveFormsModule];
const componentModules = [
    UserProfileFormBaseInputsModule,
    UserProfileFormEmailAddressInputsModule,
    UserProfileFormPhoneNumberInputsModule,
    UserProfileFormPostalAddressInputsModule,
    UserProfileFormErrorsModule,
    UserProfileFormModalsModule,
];
export class UserProfileFormsModule {
}
UserProfileFormsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
UserProfileFormsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormsModule, declarations: [UserEmailAddressFormComponent,
        UserEmailAddressFormCustomizableDirective,
        UserEmailAddressFormErrorsCustomizableDirective,
        UserPhoneNumberFormComponent,
        UserPhoneNumberFormCustomizableDirective,
        UserPhoneNumberFormErrorsCustomizableDirective,
        UserPostalAddressFormComponent,
        UserPostalAddressFormCustomizableDirective,
        UserPostalAddressFormErrorsCustomizableDirective], imports: [CommonModule, UserProfileFormBaseInputsModule,
        UserProfileFormEmailAddressInputsModule,
        UserProfileFormPhoneNumberInputsModule,
        UserProfileFormPostalAddressInputsModule,
        UserProfileFormErrorsModule,
        UserProfileFormModalsModule, ButtonModule, LoadButtonModule, IconModule, ReactiveFormsModule], exports: [UserEmailAddressFormComponent,
        UserEmailAddressFormCustomizableDirective,
        UserEmailAddressFormErrorsCustomizableDirective,
        UserPhoneNumberFormComponent,
        UserPhoneNumberFormCustomizableDirective,
        UserPhoneNumberFormErrorsCustomizableDirective,
        UserPostalAddressFormComponent,
        UserPostalAddressFormCustomizableDirective,
        UserPostalAddressFormErrorsCustomizableDirective, ButtonModule, LoadButtonModule, IconModule, ReactiveFormsModule, UserProfileFormBaseInputsModule,
        UserProfileFormEmailAddressInputsModule,
        UserProfileFormPhoneNumberInputsModule,
        UserProfileFormPostalAddressInputsModule,
        UserProfileFormErrorsModule,
        UserProfileFormModalsModule] });
UserProfileFormsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormsModule, imports: [[CommonModule, ...componentModules, ...uiModules], ButtonModule, LoadButtonModule, IconModule, ReactiveFormsModule, UserProfileFormBaseInputsModule,
        UserProfileFormEmailAddressInputsModule,
        UserProfileFormPhoneNumberInputsModule,
        UserProfileFormPostalAddressInputsModule,
        UserProfileFormErrorsModule,
        UserProfileFormModalsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...componentModules, ...uiModules],
                    declarations: [
                        UserEmailAddressFormComponent,
                        UserEmailAddressFormCustomizableDirective,
                        UserEmailAddressFormErrorsCustomizableDirective,
                        UserPhoneNumberFormComponent,
                        UserPhoneNumberFormCustomizableDirective,
                        UserPhoneNumberFormErrorsCustomizableDirective,
                        UserPostalAddressFormComponent,
                        UserPostalAddressFormCustomizableDirective,
                        UserPostalAddressFormErrorsCustomizableDirective,
                    ],
                    exports: [
                        UserEmailAddressFormComponent,
                        UserEmailAddressFormCustomizableDirective,
                        UserEmailAddressFormErrorsCustomizableDirective,
                        UserPhoneNumberFormComponent,
                        UserPhoneNumberFormCustomizableDirective,
                        UserPhoneNumberFormErrorsCustomizableDirective,
                        UserPostalAddressFormComponent,
                        UserPostalAddressFormCustomizableDirective,
                        UserPostalAddressFormErrorsCustomizableDirective,
                        ...uiModules,
                        ...componentModules,
                    ],
                }]
        }] });
