// @backbase/messages-common/esm2015/src/components/create-message/employee-mailout-review/employee-mailout-review.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ButtonModule } from '@backbase/ui-ang/button';
import { AlertModule } from '@backbase/ui-ang/alert';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { FileAttachmentModule } from '@backbase/ui-ang/file-attachment';
import { MessagesEmployeeMailoutReviewComponent, MessagesEmployeeMailoutReviewCustomizableDirective, MessagesEmployeeMailoutReviewErrorNotificationHeaderCustomizableDirective, MessagesEmployeeMailoutReviewSuccessNotificationHeaderCustomizableDirective, } from './employee-mailout-review.component';
import { MessagesEncodingService } from '../../../services/messages-encoding/messages-encoding.service';
import { MessagesBodyPreviewModule } from '../../messages-preview/messages-preview.module';
import { MessagesEmployeeMailoutReviewService } from './employee-mailout-review.service';
import { MessagesManipulationConfirmModalModule } from '../../messages-manipulation-confirm-modal/messages-manipulation-confirm-modal.module';
import { MessagesManipulationConfirmModalService } from '../../messages-manipulation-confirm-modal/messages-manipulation-confirm-modal.service';
import { SharedMethodsService } from '../../../services/shared-methods/shared-methods.service';
import * as i0 from "@angular/core";
const components = [MessagesEmployeeMailoutReviewComponent];
const directives = [
    MessagesEmployeeMailoutReviewCustomizableDirective,
    MessagesEmployeeMailoutReviewSuccessNotificationHeaderCustomizableDirective,
    MessagesEmployeeMailoutReviewErrorNotificationHeaderCustomizableDirective,
];
const uiModules = [ButtonModule, FileAttachmentModule, LoadButtonModule, AlertModule, LoadingIndicatorModule];
const innerModules = [MessagesBodyPreviewModule, MessagesManipulationConfirmModalModule];
export class MessagesEmployeeMailoutReviewModule {
}
MessagesEmployeeMailoutReviewModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesEmployeeMailoutReviewModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MessagesEmployeeMailoutReviewModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesEmployeeMailoutReviewModule, declarations: [MessagesEmployeeMailoutReviewComponent, MessagesEmployeeMailoutReviewCustomizableDirective,
        MessagesEmployeeMailoutReviewSuccessNotificationHeaderCustomizableDirective,
        MessagesEmployeeMailoutReviewErrorNotificationHeaderCustomizableDirective], imports: [CommonModule, ButtonModule, FileAttachmentModule, LoadButtonModule, AlertModule, LoadingIndicatorModule, MessagesBodyPreviewModule, MessagesManipulationConfirmModalModule], exports: [MessagesEmployeeMailoutReviewComponent, MessagesEmployeeMailoutReviewCustomizableDirective,
        MessagesEmployeeMailoutReviewSuccessNotificationHeaderCustomizableDirective,
        MessagesEmployeeMailoutReviewErrorNotificationHeaderCustomizableDirective, ButtonModule, FileAttachmentModule, LoadButtonModule, AlertModule, LoadingIndicatorModule, MessagesBodyPreviewModule, MessagesManipulationConfirmModalModule] });
MessagesEmployeeMailoutReviewModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesEmployeeMailoutReviewModule, providers: [
        MessagesEncodingService,
        MessagesEmployeeMailoutReviewService,
        MessagesManipulationConfirmModalService,
        SharedMethodsService,
    ], imports: [[CommonModule, ...uiModules, ...innerModules], ButtonModule, FileAttachmentModule, LoadButtonModule, AlertModule, LoadingIndicatorModule, MessagesBodyPreviewModule, MessagesManipulationConfirmModalModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesEmployeeMailoutReviewModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules, ...innerModules],
                    declarations: [...components, ...directives],
                    providers: [
                        MessagesEncodingService,
                        MessagesEmployeeMailoutReviewService,
                        MessagesManipulationConfirmModalService,
                        SharedMethodsService,
                    ],
                    exports: [...components, ...directives, ...uiModules, ...innerModules],
                }]
        }] });
