// @backbase/authorized-users-journey-ang/esm2015/src/authorized-users-list/components/authorized-users-list/authorized-users-list.component.js
import { Component, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/icon";
import * as i2 from "../authorized-users-list-item/authorized-users-list-item.component";
import * as i3 from "@backbase/ui-ang/button";
import * as i4 from "@angular/common";
export class AuthorizedUsersListComponent {
    constructor() {
        this.edit = new EventEmitter();
        this.remove = new EventEmitter();
        this.paused = new EventEmitter();
        this.resumed = new EventEmitter();
        this.createAuthorizedUser = new EventEmitter();
    }
    goToAuthorizedUser() {
        this.createAuthorizedUser.emit();
    }
    editPermissions(user) {
        this.edit.emit(user);
    }
    removeUser(user) {
        this.remove.emit(user);
    }
    pauseUser(user) {
        this.paused.emit(user);
    }
    resumeUser(user) {
        this.resumed.emit(user);
    }
}
AuthorizedUsersListComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersListComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
AuthorizedUsersListComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AuthorizedUsersListComponent, selector: "bb-authorized-users-list", inputs: { authorizedUsersList: "authorizedUsersList", levelsConfig: "levelsConfig" }, outputs: { edit: "edit", remove: "remove", paused: "paused", resumed: "resumed", createAuthorizedUser: "createAuthorizedUser" }, ngImport: i0, template: "<div class=\"bb-block bb-block--lg bb-text-align-right\">\n  <button bbButton data-role=\"add-button\" color=\"primary\" (click)=\"goToAuthorizedUser()\">\n    <bb-icon-ui name=\"add\"></bb-icon-ui> \n    <span i18n=\"Add new authorized user button@@authorized.users.add.button\">\n      Add Authorized User\n    </span>\n  </button>\n</div>\n<div class=\"bb-list\">\n  <div class=\"bb-list__item--no-separator\" *ngFor=\"let authorizedUsersItem of authorizedUsersList\">\n    <div class=\"bb-block bb-block--lg\">\n      <bb-authorized-users-list-item\n        [authorizedUsersItem]=\"authorizedUsersItem\"\n        [levelsConfig]=\"levelsConfig\"\n        (edit)=\"editPermissions($event)\"\n        (remove)=\"removeUser($event)\"\n        (paused)=\"pauseUser($event)\"\n        (resumed)=\"resumeUser($event)\"\n      ></bb-authorized-users-list-item>\n    </div>\n  </div>\n</div>\n", components: [{ type: i1.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i2.AuthorizedUsersListItemComponent, selector: "bb-authorized-users-list-item", inputs: ["authorizedUsersItem", "levelsConfig"], outputs: ["edit", "remove", "paused", "resumed"] }], directives: [{ type: i3.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }, { type: i4.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersListComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-authorized-users-list',
                    templateUrl: `./authorized-users-list.component.html`,
                }]
        }], propDecorators: { authorizedUsersList: [{
                type: Input
            }], levelsConfig: [{
                type: Input
            }], edit: [{
                type: Output
            }], remove: [{
                type: Output
            }], paused: [{
                type: Output
            }], resumed: [{
                type: Output
            }], createAuthorizedUser: [{
                type: Output
            }] } });
