// @backbase/product-summary-common-ang/esm2015/src/services/download-resources.service.js
import { Inject, Injectable } from '@angular/core';
import { WINDOW } from '../model/providers';
import * as i0 from "@angular/core";
export class DownloadResourcesService {
    constructor(localWindow) {
        this.localWindow = localWindow;
    }
    downloadFile(name, type, data) {
        const fileData = new Blob([data], { type });
        if (this.isFileSaverAvailable()) {
            this.downloadViaFileSaver(name, fileData);
        }
        else {
            this.downloadViaLink(name, fileData);
        }
    }
    isFileSaverAvailable() {
        const { navigator } = this.localWindow;
        // @ts-ignore
        return navigator !== undefined && navigator.msSaveOrOpenBlob !== undefined;
    }
    downloadViaFileSaver(name, data) {
        const { navigator } = this.localWindow;
        // @ts-ignore
        navigator.msSaveOrOpenBlob(data, name);
    }
    downloadViaLink(name, data) {
        const { document, URL } = this.localWindow;
        const downloadLink = document.createElement('a');
        downloadLink.href = URL.createObjectURL(data);
        downloadLink.download = name;
        downloadLink.click();
        downloadLink.remove();
    }
}
DownloadResourcesService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: DownloadResourcesService, deps: [{ token: WINDOW }], target: i0.ɵɵFactoryTarget.Injectable });
DownloadResourcesService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: DownloadResourcesService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: DownloadResourcesService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [WINDOW]
                }] }]; } });
