// @backbase/authorized-users-journey-ang/esm2015/src/authorized-users-list/authorized-users-list.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { AuthorizedUsersListWrapperComponent } from './components/authorized-users-list-wrapper/authorized-users-list-wrapper.component';
import { AuthorizedUsersEmptyListComponent } from './components/authorized-users-empty-list/authorized-users-empty-list.component';
import { AuthorizedUsersListComponent } from './components/authorized-users-list/authorized-users-list.component';
import { AuthorizedUsersListItemComponent } from './components/authorized-users-list-item/authorized-users-list-item.component';
import { NgbCollapseModule } from '@ng-bootstrap/ng-bootstrap';
import { AuthorizedUsersListStatusBadgeComponent } from './components/authorized-users-list-status-badge/authorized-users-list-status-badge.component';
import { AuthorizedUsersListConfirmationRemoveUserDialogComponent } from './components/authorized-users-list-confirmation-dialog/authorized-users-list-confirmation-remove-user-dialog.component';
import { AuthorizedUserListConfirmationPauseUserDialogComponent } from './components/authorized-users-list-confirmation-dialog/authorized-user-list-confirmation-pause-user-dialog.component';
import { AuthorizedUsersCommonModule } from '../common/authorized-users-common.module';
import { IconModule } from '@backbase/ui-ang/icon';
import { ButtonModule } from '@backbase/ui-ang/button';
import { PhoneNumberModule } from '@backbase/ui-ang/phone-number-format-pipe';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { DropdownMenuModule } from '@backbase/ui-ang/dropdown-menu';
import { ModalModule } from '@backbase/ui-ang/modal';
import { BadgeModule } from '@backbase/ui-ang/badge';
import { NotificationModule } from '@backbase/ui-ang/notification';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const components = [
    AuthorizedUsersListWrapperComponent,
    AuthorizedUsersEmptyListComponent,
    AuthorizedUsersListComponent,
    AuthorizedUsersListItemComponent,
    AuthorizedUsersListStatusBadgeComponent,
    AuthorizedUsersListConfirmationRemoveUserDialogComponent,
    AuthorizedUserListConfirmationPauseUserDialogComponent,
];
const modules = [
    ButtonModule,
    IconModule,
    PhoneNumberModule,
    LoadingIndicatorModule,
    EmptyStateModule,
    NgbCollapseModule,
    DropdownMenuModule,
    ModalModule,
    BadgeModule,
    NotificationModule,
    AuthorizedUsersCommonModule,
];
export class AuthorizedUsersListModule {
}
AuthorizedUsersListModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersListModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AuthorizedUsersListModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersListModule, declarations: [AuthorizedUsersListWrapperComponent,
        AuthorizedUsersEmptyListComponent,
        AuthorizedUsersListComponent,
        AuthorizedUsersListItemComponent,
        AuthorizedUsersListStatusBadgeComponent,
        AuthorizedUsersListConfirmationRemoveUserDialogComponent,
        AuthorizedUserListConfirmationPauseUserDialogComponent], imports: [CommonModule, i1.BackbaseCoreModule, ButtonModule,
        IconModule,
        PhoneNumberModule,
        LoadingIndicatorModule,
        EmptyStateModule,
        NgbCollapseModule,
        DropdownMenuModule,
        ModalModule,
        BadgeModule,
        NotificationModule,
        AuthorizedUsersCommonModule], exports: [ButtonModule,
        IconModule,
        PhoneNumberModule,
        LoadingIndicatorModule,
        EmptyStateModule,
        NgbCollapseModule,
        DropdownMenuModule,
        ModalModule,
        BadgeModule,
        NotificationModule,
        AuthorizedUsersCommonModule, AuthorizedUsersListWrapperComponent,
        AuthorizedUsersEmptyListComponent,
        AuthorizedUsersListComponent,
        AuthorizedUsersListItemComponent,
        AuthorizedUsersListStatusBadgeComponent,
        AuthorizedUsersListConfirmationRemoveUserDialogComponent,
        AuthorizedUserListConfirmationPauseUserDialogComponent] });
AuthorizedUsersListModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersListModule, imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { AuthorizedUsersListWrapperComponent },
            }),
            ...modules,
        ], ButtonModule,
        IconModule,
        PhoneNumberModule,
        LoadingIndicatorModule,
        EmptyStateModule,
        NgbCollapseModule,
        DropdownMenuModule,
        ModalModule,
        BadgeModule,
        NotificationModule,
        AuthorizedUsersCommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersListModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components],
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { AuthorizedUsersListWrapperComponent },
                        }),
                        ...modules,
                    ],
                    exports: [...modules, ...components],
                }]
        }] });
