// @backbase/consent-journey-ang/esm2015/src/consent-journey.routes.js
import { ConsentJourneyComponent } from './consent-journey.component';
import { PaymentRequestComponent } from './views/payment-request/payment-request.component';
import { PaymentConsentRedirectComponent } from './views/payment-consent-redirect/payment-consent-redirect.component';
import { TppConsentRedirectComponent } from './views/tpp-consent-redirect/tpp-consent-redirect.component';
import { ConsentRequestComponent } from './views/consent-request/consent-request.component';
import { ConsentsComponent } from './views/consents/consents.component';
import { ConsentDetailComponent } from './views/consents/consent-detail/consent-detail.component';
import { ConsentListComponent } from './views/consents/consent-list/consent-list.component';
import '@angular/localize/init';
export const tabs = {
    paymentRequest: () => $localize `:Payment request tab title@@consent-journey.tab.paymentRequest.title:Payment Request`,
    consentRequest: () => $localize `:Consent request tab title@@consent-journey.tab.consentRequest.title:Consent Request`,
    consents: () => $localize `:Consents tab title@@consent-journey.tab.consents.title:Consents`,
};
export const consentRoutes = {
    path: '',
    component: ConsentJourneyComponent,
    children: [
        {
            path: '',
            redirectTo: 'payment-request',
            pathMatch: 'full',
        },
        {
            path: 'payment-request',
            component: PaymentRequestComponent,
            data: { title: tabs.paymentRequest },
        },
        {
            path: 'payment-redirect',
            component: PaymentConsentRedirectComponent,
        },
        {
            path: 'consent-redirect',
            component: TppConsentRedirectComponent,
        },
        {
            path: 'consent-request',
            component: ConsentRequestComponent,
            data: { title: tabs.consentRequest },
        },
        {
            component: ConsentsComponent,
            data: { title: tabs.consents },
            path: 'consents',
            children: [
                {
                    path: '',
                    redirectTo: 'list',
                    pathMatch: 'full',
                },
                {
                    path: 'list',
                    component: ConsentListComponent,
                },
                {
                    path: 'details',
                    component: ConsentDetailComponent,
                },
            ],
        },
    ],
};
