// @backbase/consent-journey-ang/esm2015/src/consent-journey.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "@angular/common";
export class ConsentJourneyComponent {
    constructor(route) {
        this.route = route;
        const { routeConfig } = this.route.snapshot;
        this.tabs = (routeConfig === null || routeConfig === void 0 ? void 0 : routeConfig.children) ? this.initializeTabs(routeConfig) : [];
    }
    initializeTabs({ children = [] }) {
        return children
            .filter(routeItem => routeItem.path && routeItem.data)
            .map((routeItem, index) => {
            var _a;
            return ({
                title: ((_a = routeItem.data) === null || _a === void 0 ? void 0 : _a.title()) || `Tab${index + 1}`,
                route: routeItem.path || `${index + 1}`,
            });
        });
    }
}
ConsentJourneyComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentJourneyComponent, deps: [{ token: i1.ActivatedRoute }], target: i0.ɵɵFactoryTarget.Component });
ConsentJourneyComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: ConsentJourneyComponent, selector: "bb-consent-journey", ngImport: i0, template: `
    <div class="bb-tab">
      <div class="bb-tab-container__tabs nav nav-tabs">
        <a
          *ngFor="let tab of tabs"
          class="nav-link nav-item"
          role="tab"
          [routerLink]="[tab.route]"
          routerLinkActive="active">
          <span class="bb-stack__item bb-stack__item--spacing-xs">{{ tab?.title }}</span>
        </a>
      </div>
      <div class="bb-tab-container__content tab-content" role="tabpanel">
        <router-outlet></router-outlet>
      </div>
    </div>
  `, isInline: true, directives: [{ type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { type: i1.RouterLinkWithHref, selector: "a[routerLink],area[routerLink]", inputs: ["routerLink", "target", "queryParams", "fragment", "queryParamsHandling", "preserveFragment", "skipLocationChange", "replaceUrl", "state", "relativeTo"] }, { type: i1.RouterLinkActive, selector: "[routerLinkActive]", inputs: ["routerLinkActiveOptions", "routerLinkActive"], exportAs: ["routerLinkActive"] }, { type: i1.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentJourneyComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-consent-journey',
                    template: `
    <div class="bb-tab">
      <div class="bb-tab-container__tabs nav nav-tabs">
        <a
          *ngFor="let tab of tabs"
          class="nav-link nav-item"
          role="tab"
          [routerLink]="[tab.route]"
          routerLinkActive="active">
          <span class="bb-stack__item bb-stack__item--spacing-xs">{{ tab?.title }}</span>
        </a>
      </div>
      <div class="bb-tab-container__content tab-content" role="tabpanel">
        <router-outlet></router-outlet>
      </div>
    </div>
  `,
                }]
        }], ctorParameters: function () { return [{ type: i1.ActivatedRoute }]; } });
