// @backbase/consent-journey-ang/esm2015/src/consent-journey.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { provideRoutes, RouterModule } from '@angular/router';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { EntitlementsModule } from '@backbase/foundation-ang/entitlements';
import { PaymentRequestWidgetModule } from '@backbase/payment-request-widget-ang';
import { ConsentRequestWidgetModule } from '@backbase/consent-request-widget-ang';
import { ConsentListWidgetModule } from '@backbase/consent-list-widget-ang';
import { ConsentDetailsWidgetModule } from '@backbase/consent-details-widget-ang';
import { ButtonModule } from '@backbase/ui-ang/button';
import { ConsentJourneyComponent } from './consent-journey.component';
import { consentRoutes } from './consent-journey.routes';
import { ConsentJourneyConfigurationService } from './services/consent-journey-config.service';
import { PaymentRequestComponent } from './views/payment-request/payment-request.component';
import { PaymentConsentRedirectComponent } from './views/payment-consent-redirect/payment-consent-redirect.component';
import { TppConsentRedirectComponent } from './views/tpp-consent-redirect/tpp-consent-redirect.component';
import { ConsentRequestComponent } from './views/consent-request/consent-request.component';
import { ConsentsComponent } from './views/consents/consents.component';
import { ConsentDetailComponent } from './views/consents/consent-detail/consent-detail.component';
import { ConsentListComponent } from './views/consents/consent-list/consent-list.component';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "@backbase/foundation-ang/core";
const viewComponents = [
    ConsentJourneyComponent,
    PaymentRequestComponent,
    PaymentConsentRedirectComponent,
    TppConsentRedirectComponent,
    ConsentRequestComponent,
    ConsentsComponent,
    ConsentDetailComponent,
    ConsentListComponent,
];
const uiModules = [
    EntitlementsModule,
    PaymentRequestWidgetModule,
    ConsentRequestWidgetModule,
    ConsentListWidgetModule,
    ConsentDetailsWidgetModule,
];
export class ConsentJourneyModule {
    static forRoot(data = { route: consentRoutes }) {
        return {
            ngModule: ConsentJourneyModule,
            providers: [provideRoutes([data.route])],
        };
    }
}
ConsentJourneyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentJourneyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ConsentJourneyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentJourneyModule, declarations: [ConsentJourneyComponent,
        PaymentRequestComponent,
        PaymentConsentRedirectComponent,
        TppConsentRedirectComponent,
        ConsentRequestComponent,
        ConsentsComponent,
        ConsentDetailComponent,
        ConsentListComponent], imports: [CommonModule, i1.RouterModule, i2.BackbaseCoreModule, EntitlementsModule,
        PaymentRequestWidgetModule,
        ConsentRequestWidgetModule,
        ConsentListWidgetModule,
        ConsentDetailsWidgetModule, ButtonModule], exports: [ConsentJourneyComponent,
        PaymentRequestComponent,
        PaymentConsentRedirectComponent,
        TppConsentRedirectComponent,
        ConsentRequestComponent,
        ConsentsComponent,
        ConsentDetailComponent,
        ConsentListComponent, EntitlementsModule,
        PaymentRequestWidgetModule,
        ConsentRequestWidgetModule,
        ConsentListWidgetModule,
        ConsentDetailsWidgetModule] });
ConsentJourneyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentJourneyModule, providers: [ConsentJourneyConfigurationService], imports: [[
            CommonModule,
            RouterModule.forChild([consentRoutes]),
            BackbaseCoreModule.withConfig({
                classMap: { ConsentJourneyComponent: ConsentJourneyComponent },
            }),
            ...uiModules,
            ButtonModule,
        ], EntitlementsModule,
        PaymentRequestWidgetModule,
        ConsentRequestWidgetModule,
        ConsentListWidgetModule,
        ConsentDetailsWidgetModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentJourneyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: viewComponents,
                    imports: [
                        CommonModule,
                        RouterModule.forChild([consentRoutes]),
                        BackbaseCoreModule.withConfig({
                            classMap: { ConsentJourneyComponent: ConsentJourneyComponent },
                        }),
                        ...uiModules,
                        ButtonModule,
                    ],
                    providers: [ConsentJourneyConfigurationService],
                    exports: [...viewComponents, ...uiModules],
                }]
        }] });
