// @backbase/turnovers-journey-ang/bundles/backbase-turnovers-journey-ang.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core'), require('@angular/router'), require('@backbase/foundation-ang/core'), require('@backbase/product-summary-account-selector-widget-ang'), require('@backbase/turnovers-widget-ang'), require('@backbase/ui-ang/header')) :
    typeof define === 'function' && define.amd ? define('@backbase/turnovers-journey-ang', ['exports', '@angular/common', '@angular/core', '@angular/router', '@backbase/foundation-ang/core', '@backbase/product-summary-account-selector-widget-ang', '@backbase/turnovers-widget-ang', '@backbase/ui-ang/header'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["turnovers-journey-ang"] = {}), global.ng.common, global.ng.core, global.ng.router, global.i1$1, global.i4, global.i5, global.i3));
})(this, (function (exports, common, i0, i1, i1$1, i4, i5, i3) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i1__namespace$1 = /*#__PURE__*/_interopNamespace(i1$1);
    var i4__namespace = /*#__PURE__*/_interopNamespace(i4);
    var i5__namespace = /*#__PURE__*/_interopNamespace(i5);
    var i3__namespace = /*#__PURE__*/_interopNamespace(i3);

    // eslint-disable-next-line
    var TurnoversJourneyConfigurationToken = new i0.InjectionToken('TurnoversJourneyConfiguration injection token');
    var configDefaults = {
        showForecast: true,
    };
    var TurnoversJourneyService = /** @class */ (function () {
        function TurnoversJourneyService(config) {
            config = config || {};
            this._config = Object.assign(Object.assign({}, configDefaults), config);
        }
        Object.defineProperty(TurnoversJourneyService.prototype, "defaults", {
            get: function () {
                return configDefaults;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(TurnoversJourneyService.prototype, "showForecast", {
            get: function () {
                return this._config.showForecast;
            },
            enumerable: false,
            configurable: true
        });
        return TurnoversJourneyService;
    }());
    TurnoversJourneyService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: TurnoversJourneyService, deps: [{ token: TurnoversJourneyConfigurationToken, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    TurnoversJourneyService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: TurnoversJourneyService });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: TurnoversJourneyService, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () {
            return [{ type: undefined, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.Inject,
                            args: [TurnoversJourneyConfigurationToken]
                        }] }];
        } });

    var TurnoversJourneyComponent = /** @class */ (function () {
        function TurnoversJourneyComponent() {
        }
        return TurnoversJourneyComponent;
    }());
    TurnoversJourneyComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: TurnoversJourneyComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    TurnoversJourneyComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: TurnoversJourneyComponent, selector: "bb-turnovers-journey", providers: [TurnoversJourneyService], ngImport: i0__namespace, template: '<router-outlet></router-outlet>', isInline: true, directives: [{ type: i1__namespace.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: TurnoversJourneyComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-turnovers-journey',
                        template: '<router-outlet></router-outlet>',
                        providers: [TurnoversJourneyService],
                    }]
            }] });

    // TODO: remove utility classes
    var TurnoversViewComponent = /** @class */ (function () {
        function TurnoversViewComponent(router, route, config) {
            this.router = router;
            this.route = route;
            this.config = config;
        }
        TurnoversViewComponent.prototype.onItemSelect = function (selectedAccount) {
            this.router.navigate(['.', { selectedAccount: selectedAccount }], { relativeTo: this.route });
        };
        return TurnoversViewComponent;
    }());
    TurnoversViewComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: TurnoversViewComponent, deps: [{ token: i1__namespace.Router }, { token: i1__namespace.ActivatedRoute }, { token: TurnoversJourneyService }], target: i0__namespace.ɵɵFactoryTarget.Component });
    TurnoversViewComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: TurnoversViewComponent, selector: "bb-turnovers-view", ngImport: i0__namespace, template: "\n    <bb-header-ui\n      headingType=\"h1\"\n      headingClasses=\"bb-heading-widget__heading px-0 pb-5\"\n      heading=\"Cash Flow\"\n      i18n-heading=\"Cash flow chart|Heading for Turnovers journey@@turnovers.journey.heading.text\"\n    ></bb-header-ui>\n    <div class=\"bb-block--xl\">\n      <bb-product-summary-account-selector-widget\n        (selectedAccount)=\"onItemSelect($event)\"\n      ></bb-product-summary-account-selector-widget>\n    </div>\n    <div class=\"bb-card card-body\">\n      <bb-turnovers-widget [forecast]=\"config.showForecast\"></bb-turnovers-widget>\n    </div>\n  ", isInline: true, components: [{ type: i3__namespace.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i4__namespace.ProductSummaryAccountSelectorWidgetComponent, selector: "bb-product-summary-account-selector-widget", inputs: ["event.refreshAccountsList", "multiple", "filterItems", "showAccountState"], outputs: ["selectedAccount", "selectedKind"] }, { type: i5__namespace.TurnoversWidgetComponent, selector: "bb-turnovers-widget", inputs: ["forecast"] }], changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: TurnoversViewComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-turnovers-view',
                        template: "\n    <bb-header-ui\n      headingType=\"h1\"\n      headingClasses=\"bb-heading-widget__heading px-0 pb-5\"\n      heading=\"Cash Flow\"\n      i18n-heading=\"Cash flow chart|Heading for Turnovers journey@@turnovers.journey.heading.text\"\n    ></bb-header-ui>\n    <div class=\"bb-block--xl\">\n      <bb-product-summary-account-selector-widget\n        (selectedAccount)=\"onItemSelect($event)\"\n      ></bb-product-summary-account-selector-widget>\n    </div>\n    <div class=\"bb-card card-body\">\n      <bb-turnovers-widget [forecast]=\"config.showForecast\"></bb-turnovers-widget>\n    </div>\n  ",
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], ctorParameters: function () { return [{ type: i1__namespace.Router }, { type: i1__namespace.ActivatedRoute }, { type: TurnoversJourneyService }]; } });

    var defaultRoute = {
        path: '',
        component: TurnoversJourneyComponent,
        children: [
            {
                path: '',
                component: TurnoversViewComponent,
                data: { title: 'Turnovers' },
            },
        ],
    };
    var TurnoversJourneyModule = /** @class */ (function () {
        function TurnoversJourneyModule() {
        }
        TurnoversJourneyModule.forRoot = function (data) {
            if (data === void 0) { data = { route: defaultRoute }; }
            return {
                ngModule: TurnoversJourneyModule,
                providers: [i1.provideRoutes([data.route])],
            };
        };
        return TurnoversJourneyModule;
    }());
    TurnoversJourneyModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: TurnoversJourneyModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    TurnoversJourneyModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: TurnoversJourneyModule, declarations: [TurnoversJourneyComponent, TurnoversViewComponent], imports: [i1__namespace$1.BackbaseCoreModule, common.CommonModule,
            i3.HeaderModule,
            i4.ProductSummaryAccountSelectorWidgetModule,
            i1.RouterModule,
            i5.TurnoversWidgetModule] });
    TurnoversJourneyModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: TurnoversJourneyModule, imports: [[
                i1$1.BackbaseCoreModule.withConfig({
                    classMap: { TurnoversJourneyComponent: TurnoversJourneyComponent },
                }),
                common.CommonModule,
                i3.HeaderModule,
                i4.ProductSummaryAccountSelectorWidgetModule,
                i1.RouterModule,
                i5.TurnoversWidgetModule,
            ]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: TurnoversJourneyModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [TurnoversJourneyComponent, TurnoversViewComponent],
                        imports: [
                            i1$1.BackbaseCoreModule.withConfig({
                                classMap: { TurnoversJourneyComponent: TurnoversJourneyComponent },
                            }),
                            common.CommonModule,
                            i3.HeaderModule,
                            i4.ProductSummaryAccountSelectorWidgetModule,
                            i1.RouterModule,
                            i5.TurnoversWidgetModule,
                        ],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.TurnoversJourneyComponent = TurnoversJourneyComponent;
    exports.TurnoversJourneyConfigurationToken = TurnoversJourneyConfigurationToken;
    exports.TurnoversJourneyModule = TurnoversJourneyModule;
    exports.TurnoversJourneyService = TurnoversJourneyService;
    exports.TurnoversViewComponent = TurnoversViewComponent;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
