// @backbase/ui-ang/esm2015/card-vendor/card-vendor.component.js
import { Component, Input, ChangeDetectionStrategy } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
/**
 * The prefix for the vendor class name.
 */
const classNamePrefix = 'bb-card-vendor-';
const vendorToClassName = (vendor) => {
    if (typeof vendor !== 'string') {
        return '';
    }
    const classNamePostfix = vendor.trim().toLowerCase().replace(/\W/g, '-').replace(/-{2,}/g, '-');
    return `${classNamePrefix}${classNamePostfix}`;
};
const vendorToTitle = (vendor) => (vendor ? vendor.trim().toLowerCase().replace(/-/g, ' ') : '');
/**
 * @name CardVendorComponent
 *
 * @description
 * Component that displays the logo of a card vendor.
 */
export class CardVendorComponent {
    constructor() {
        this.className = '';
        this.title = '';
    }
    ngOnInit() {
        if (typeof this.vendor !== 'string') {
            throw new Error(`"vendor" input is required in "${this.constructor.name}"`);
        }
    }
    ngOnChanges(changes) {
        this.className = vendorToClassName(this.vendor);
        this.title = vendorToTitle(this.vendor);
    }
}
CardVendorComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardVendorComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
CardVendorComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: CardVendorComponent, selector: "bb-card-vendor-ui", inputs: { vendor: "vendor" }, usesOnChanges: true, ngImport: i0, template: "<div role=\"img\" [attr.aria-label]=\"title\" [title]=\"title\" class=\"bb-card-vendor\" [ngClass]=\"className\"></div>\n", directives: [{ type: i1.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardVendorComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-card-vendor-ui',
                    templateUrl: './card-vendor.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { vendor: [{
                type: Input
            }] } });
