// @backbase/ui-ang/esm2015/card-vendor/card-vendor.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { CardVendorComponent } from './card-vendor.component';
import * as i0 from "@angular/core";
export class CardVendorModule {
}
CardVendorModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardVendorModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CardVendorModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardVendorModule, declarations: [CardVendorComponent], imports: [CommonModule], exports: [CardVendorComponent] });
CardVendorModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardVendorModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardVendorModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [CardVendorComponent],
                    exports: [CardVendorComponent],
                }]
        }] });
