// @backbase/ui-ang/esm2015/product-selector/product-selector.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NgbDropdownModule } from '@ng-bootstrap/ng-bootstrap';
import { IconModule } from '@backbase/ui-ang/icon';
import { ProductSelectorComponent } from './product-selector.component';
import { ButtonModule } from '@backbase/ui-ang/button';
import * as i0 from "@angular/core";
export class ProductSelectorModule {
}
ProductSelectorModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSelectorModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductSelectorModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSelectorModule, declarations: [ProductSelectorComponent], imports: [CommonModule, NgbDropdownModule, IconModule, ButtonModule], exports: [ProductSelectorComponent] });
ProductSelectorModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSelectorModule, imports: [[CommonModule, NgbDropdownModule, IconModule, ButtonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSelectorModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, NgbDropdownModule, IconModule, ButtonModule],
                    declarations: [ProductSelectorComponent],
                    exports: [ProductSelectorComponent],
                }]
        }] });
