// @backbase/identity-self-service-journey-ang/esm2015/lib/features/device-information-view.component.js
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { IdentitySelfServiceJourneyConfigurationService } from '../identity-self-service-journey.config.service';
import * as i0 from "@angular/core";
import * as i1 from "../identity-self-service-journey.config.service";
import * as i2 from "@backbase/device-information-widget-ang";
export class DeviceInformationViewComponent {
    constructor(
    /** @internal */
    configService) {
        this.configService = configService;
    }
}
/** @nocollapse */ DeviceInformationViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: DeviceInformationViewComponent, deps: [{ token: i1.IdentitySelfServiceJourneyConfigurationService }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ DeviceInformationViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: DeviceInformationViewComponent, selector: "bb-device-information-view", ngImport: i0, template: `<bb-device-information-widget
    [dismissTimeout]="this.configService.deviceInformation.notificationDismissTime"
    [isAdminWidget]="false"
  >
  </bb-device-information-widget>`, isInline: true, components: [{ type: i2.DeviceInformationWidgetComponent, selector: "bb-device-information-widget", inputs: ["dismissTimeout", "isAdminWidget", "dbsUserId"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: DeviceInformationViewComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-device-information-view',
                    template: `<bb-device-information-widget
    [dismissTimeout]="this.configService.deviceInformation.notificationDismissTime"
    [isAdminWidget]="false"
  >
  </bb-device-information-widget>`,
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.IdentitySelfServiceJourneyConfigurationService }]; } });
