// @backbase/identity-self-service-journey-ang/esm2015/lib/features/self-service-features.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { ReactiveFormsModule } from '@angular/forms';
import { UserIdentitySecurityCenterWidgetModule } from '@backbase/user-identity-security-center-widget-ang';
import { UserManageProfileWidgetModule } from '@backbase/user-manage-profile-widget-ang';
import { DeviceInformationWidgetModule } from '@backbase/device-information-widget-ang';
import { UserIdentitySecurityCenterViewComponent } from './user-identity-security-center-view.component';
import { UserManageProfileViewComponent } from './user-manage-profile-view.component';
import { DeviceInformationViewComponent } from './device-information-view.component';
import { UserLocalizationCommunicationsViewComponent } from './user-localization/communications-view/user-localization-communications-view.component';
import { UserLocalizationChangeLanguageViewComponent } from './user-localization/change-language-view/user-localization-change-language-view.component';
import { UserLocalizationChangeLanguageFormComponent } from './user-localization/shared/components/user-localization-change-language-form.component';
import { UserLocalizationComponent } from './user-localization/user-localization.component';
import { IdentityCommonAngModule } from '@backbase/identity-common-ang';
import { SelfServiceUiModule } from '../ui/self-service-ui.module';
import { AlertModule } from '@backbase/ui-ang/alert';
import { ButtonModule } from '@backbase/ui-ang/button';
import { HeaderModule } from '@backbase/ui-ang/header';
import { IconModule } from '@backbase/ui-ang/icon';
import { InputRadioGroupModule } from '@backbase/ui-ang/input-radio-group';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { ModalModule } from '@backbase/ui-ang/modal';
import { NotificationModule } from '@backbase/ui-ang/notification';
import * as i0 from "@angular/core";
const uiModules = [
    HeaderModule,
    IconModule,
    ButtonModule,
    LoadButtonModule,
    AlertModule,
    InputRadioGroupModule,
    ModalModule,
    NotificationModule,
    InputValidationMessageModule,
];
const components = [
    UserIdentitySecurityCenterViewComponent,
    UserManageProfileViewComponent,
    DeviceInformationViewComponent,
    UserLocalizationComponent,
    UserLocalizationCommunicationsViewComponent,
    UserLocalizationChangeLanguageViewComponent,
    UserLocalizationChangeLanguageFormComponent,
];
const widgetModules = [
    UserIdentitySecurityCenterWidgetModule,
    UserManageProfileWidgetModule,
    DeviceInformationWidgetModule,
];
export class SelfServiceFeaturesModule {
}
/** @nocollapse */ SelfServiceFeaturesModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: SelfServiceFeaturesModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ SelfServiceFeaturesModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: SelfServiceFeaturesModule, declarations: [UserIdentitySecurityCenterViewComponent,
        UserManageProfileViewComponent,
        DeviceInformationViewComponent,
        UserLocalizationComponent,
        UserLocalizationCommunicationsViewComponent,
        UserLocalizationChangeLanguageViewComponent,
        UserLocalizationChangeLanguageFormComponent], imports: [CommonModule,
        ReactiveFormsModule,
        RouterModule, HeaderModule,
        IconModule,
        ButtonModule,
        LoadButtonModule,
        AlertModule,
        InputRadioGroupModule,
        ModalModule,
        NotificationModule,
        InputValidationMessageModule, UserIdentitySecurityCenterWidgetModule,
        UserManageProfileWidgetModule,
        DeviceInformationWidgetModule, IdentityCommonAngModule,
        SelfServiceUiModule], exports: [RouterModule, UserIdentitySecurityCenterViewComponent,
        UserManageProfileViewComponent,
        DeviceInformationViewComponent,
        UserLocalizationComponent,
        UserLocalizationCommunicationsViewComponent,
        UserLocalizationChangeLanguageViewComponent,
        UserLocalizationChangeLanguageFormComponent, UserIdentitySecurityCenterWidgetModule,
        UserManageProfileWidgetModule,
        DeviceInformationWidgetModule] });
/** @nocollapse */ SelfServiceFeaturesModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: SelfServiceFeaturesModule, imports: [[
            CommonModule,
            ReactiveFormsModule,
            RouterModule,
            ...uiModules,
            ...widgetModules,
            IdentityCommonAngModule,
            SelfServiceUiModule,
        ], RouterModule, UserIdentitySecurityCenterWidgetModule,
        UserManageProfileWidgetModule,
        DeviceInformationWidgetModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: SelfServiceFeaturesModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        ReactiveFormsModule,
                        RouterModule,
                        ...uiModules,
                        ...widgetModules,
                        IdentityCommonAngModule,
                        SelfServiceUiModule,
                    ],
                    declarations: [...components],
                    exports: [RouterModule, ...components, ...widgetModules],
                }]
        }] });
