// @backbase/identity-self-service-journey-ang/esm2015/lib/features/user-manage-profile-view.component.js
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { IdentitySelfServiceJourneyConfigurationService } from '../identity-self-service-journey.config.service';
import * as i0 from "@angular/core";
import * as i1 from "../identity-self-service-journey.config.service";
import * as i2 from "@backbase/user-manage-profile-widget-ang";
export class UserManageProfileViewComponent {
    constructor(
    /** @internal */
    configService) {
        this.configService = configService;
    }
}
/** @nocollapse */ UserManageProfileViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserManageProfileViewComponent, deps: [{ token: i1.IdentitySelfServiceJourneyConfigurationService }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ UserManageProfileViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: UserManageProfileViewComponent, selector: "bb-user-manage-profile-view", ngImport: i0, template: `<bb-user-manage-profile-widget
    [notificationDismissTime]="this.configService.userManageProfile.notificationDismissTime"
    [emailAddressTypes]="this.configService.userManageProfile.emailAddressTypes"
    [postalAddressTypes]="this.configService.userManageProfile.postalAddressTypes"
    [phoneNumberTypes]="this.configService.userManageProfile.phoneNumberTypes"
    [maxEmailAddresses]="this.configService.userManageProfile.maxEmailAddresses"
    [maxPhoneNumbers]="this.configService.userManageProfile.maxPhoneNumbers"
    [maxPostalAddresses]="this.configService.userManageProfile.maxPostalAddresses"
    [showWidgetHeader]="this.configService.userManageProfile.showWidgetHeader"
    [phoneNumberCountryCode]="this.configService.userManageProfile.phoneNumberCountryCode"
    [phoneNumberDebounceTime]="this.configService.userManageProfile.phoneNumberDebounceTime"
    [enablePhoneNumberFormatter]="this.configService.userManageProfile.enablePhoneNumberFormatter"
  >
  </bb-user-manage-profile-widget>`, isInline: true, components: [{ type: i2.UserManageProfileWidgetComponent, selector: "bb-user-manage-profile-widget", inputs: ["emailAddressTypes", "postalAddressTypes", "phoneNumberTypes", "maxEmailAddresses", "maxPhoneNumbers", "maxPostalAddresses", "showWidgetHeader", "phoneNumberCountryCode", "phoneNumberDebounceTime", "enablePhoneNumberFormatter"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserManageProfileViewComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-user-manage-profile-view',
                    template: `<bb-user-manage-profile-widget
    [notificationDismissTime]="this.configService.userManageProfile.notificationDismissTime"
    [emailAddressTypes]="this.configService.userManageProfile.emailAddressTypes"
    [postalAddressTypes]="this.configService.userManageProfile.postalAddressTypes"
    [phoneNumberTypes]="this.configService.userManageProfile.phoneNumberTypes"
    [maxEmailAddresses]="this.configService.userManageProfile.maxEmailAddresses"
    [maxPhoneNumbers]="this.configService.userManageProfile.maxPhoneNumbers"
    [maxPostalAddresses]="this.configService.userManageProfile.maxPostalAddresses"
    [showWidgetHeader]="this.configService.userManageProfile.showWidgetHeader"
    [phoneNumberCountryCode]="this.configService.userManageProfile.phoneNumberCountryCode"
    [phoneNumberDebounceTime]="this.configService.userManageProfile.phoneNumberDebounceTime"
    [enablePhoneNumberFormatter]="this.configService.userManageProfile.enablePhoneNumberFormatter"
  >
  </bb-user-manage-profile-widget>`,
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.IdentitySelfServiceJourneyConfigurationService }]; } });
