// @backbase/identity-self-service-journey-ang/esm2015/lib/features/user-identity-security-center-view.component.js
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { IdentitySelfServiceJourneyConfigurationService } from '../identity-self-service-journey.config.service';
import * as i0 from "@angular/core";
import * as i1 from "../identity-self-service-journey.config.service";
import * as i2 from "@backbase/user-identity-security-center-widget-ang";
export class UserIdentitySecurityCenterViewComponent {
    constructor(
    /** @internal */
    configService) {
        this.configService = configService;
    }
}
/** @nocollapse */ UserIdentitySecurityCenterViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserIdentitySecurityCenterViewComponent, deps: [{ token: i1.IdentitySelfServiceJourneyConfigurationService }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ UserIdentitySecurityCenterViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: UserIdentitySecurityCenterViewComponent, selector: "bb-user-identity-security-center-view", ngImport: i0, template: `<bb-user-identity-security-center-widget
    [notificationDismissTime]="this.configService.userIdentitySecurityCenter.notificationDismissTime"
  >
  </bb-user-identity-security-center-widget>`, isInline: true, components: [{ type: i2.UserIdentitySecurityCenterWidgetComponent, selector: "bb-user-identity-security-center-widget", inputs: ["notificationDismissTime"], outputs: ["widgetStatus"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserIdentitySecurityCenterViewComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-user-identity-security-center-view',
                    template: `<bb-user-identity-security-center-widget
    [notificationDismissTime]="this.configService.userIdentitySecurityCenter.notificationDismissTime"
  >
  </bb-user-identity-security-center-widget>`,
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.IdentitySelfServiceJourneyConfigurationService }]; } });
