// @backbase/accounts-transactions-journeys-shared-ui-account-ang/esm2015/lib/components/account-info-renewal-status/account-info-renewal-status.component.js
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class AccountInfoRenewalStatusComponent {
    constructor() {
        this.isRenewalAllowed = false;
    }
}
/** @nocollapse */ AccountInfoRenewalStatusComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountInfoRenewalStatusComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ AccountInfoRenewalStatusComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AccountInfoRenewalStatusComponent, selector: "bb-account-info-renewal-status", inputs: { isRenewalAllowed: "isRenewalAllowed" }, ngImport: i0, template: "<ng-container *ngIf=\"isRenewalAllowed; else noAutomaticRenewal\">\n  <span\n    data-role=\"account-info-renewal-status\"\n    i18n=\"Automatic Renewal@@product.details.term.deposit.label.autoRenewalIndicator.yes\"\n  >\n    Automatic Renewal\n  </span>\n</ng-container>\n\n<ng-template #noAutomaticRenewal>\n  <span\n    data-role=\"account-info-renewal-status\"\n    i18n=\"No Automatic Renewal@@product.details.term.deposit.label.autoRenewalIndicator.no\"\n  >\n    No Automatic Renewal\n  </span>\n</ng-template>\n", directives: [{ type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountInfoRenewalStatusComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-account-info-renewal-status',
                    templateUrl: './account-info-renewal-status.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { isRenewalAllowed: [{
                type: Input
            }] } });
