// @backbase/accounts-transactions-journeys-shared-ui-account-ang/esm2015/lib/components/account-info-renewal-status/account-info-renewal-status.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ProductSummaryCommon, ValidationModule } from '@backbase/product-summary-common-ang';
import { AccountInfoRenewalStatusComponent } from './account-info-renewal-status.component';
import * as i0 from "@angular/core";
const pipeModules = [ValidationModule];
const sharedModules = [ProductSummaryCommon];
const components = [AccountInfoRenewalStatusComponent];
export class AccountInfoRenewalStatusModule {
}
/** @nocollapse */ AccountInfoRenewalStatusModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountInfoRenewalStatusModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ AccountInfoRenewalStatusModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountInfoRenewalStatusModule, declarations: [AccountInfoRenewalStatusComponent], imports: [CommonModule, ProductSummaryCommon, ValidationModule], exports: [ProductSummaryCommon, AccountInfoRenewalStatusComponent, ValidationModule] });
/** @nocollapse */ AccountInfoRenewalStatusModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountInfoRenewalStatusModule, imports: [[CommonModule, ...sharedModules, ...pipeModules], ProductSummaryCommon, ValidationModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountInfoRenewalStatusModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...sharedModules, ...pipeModules],
                    declarations: [...components],
                    exports: [...sharedModules, ...components, ...pipeModules],
                }]
        }] });
