// @backbase/initiate-payment-journey-ang/esm2015/lib/services/initiate-payment-navigation.service.js
import { Injectable } from '@angular/core';
import { ActivatedRoute, Router } from '@angular/router';
import { Routes } from '@backbase/payment-orders-ang';
import { Subject } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
/** @internal */
export class InitiatePaymentNavigationService {
    constructor(router, route) {
        this.router = router;
        this.route = route;
        this.emitAfterSuccess = new Subject();
        this.routingExtras = {
            queryParamsHandling: 'merge',
            relativeTo: this.route,
            skipLocationChange: true,
        };
    }
    /**
     * Navigate to the initiate payment form
     */
    form() {
        return this.router.navigate([Routes.FORM], this.routingExtras);
    }
    /**
     * Navigate to the initiate payment review screen.
     */
    review() {
        return this.router.navigate([Routes.REVIEW], this.routingExtras);
    }
    /**
     * Method to emit success event after form is submitted
     */
    afterSuccess() {
        this.emitAfterSuccess.next();
    }
}
/** @nocollapse */ InitiatePaymentNavigationService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InitiatePaymentNavigationService, deps: [{ token: i1.Router }, { token: i1.ActivatedRoute }], target: i0.ɵɵFactoryTarget.Injectable });
/** @nocollapse */ InitiatePaymentNavigationService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InitiatePaymentNavigationService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InitiatePaymentNavigationService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.Router }, { type: i1.ActivatedRoute }]; } });
