// @backbase/initiate-payment-journey-ang/esm2015/lib/services/initiate-payment-routable-modal.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { INITIATE_PAYMENT_JOURNEY_COMMUNICATOR, } from './communication.service';
import * as i0 from "@angular/core";
/** @internal */
export class InitiatePaymentRoutableModalService {
    constructor(communicator) {
        this.communicator = communicator;
    }
    closeModal() {
        var _a;
        (_a = this.communicator) === null || _a === void 0 ? void 0 : _a.closeEvent();
    }
    openModal() {
        var _a;
        (_a = this.communicator) === null || _a === void 0 ? void 0 : _a.headerNavigationAction('new-debit-ach');
    }
}
/** @nocollapse */ InitiatePaymentRoutableModalService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InitiatePaymentRoutableModalService, deps: [{ token: INITIATE_PAYMENT_JOURNEY_COMMUNICATOR, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
/** @nocollapse */ InitiatePaymentRoutableModalService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InitiatePaymentRoutableModalService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InitiatePaymentRoutableModalService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [INITIATE_PAYMENT_JOURNEY_COMMUNICATOR]
                }] }]; } });
