// @backbase/ui-ang/esm2015/button/button.component.js
import { Component, Input, ViewChild, HostBinding } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
/**
 * @name ButtonComponent
 *
 * @description
 * Component that displays a button.
 */
export class ButtonComponent {
    constructor(elRef, renderer2) {
        this.elRef = elRef;
        this.renderer2 = renderer2;
        /**
         * This Boolean attribute lets you specify that the button should have input focus when the page
         * loads, unless the user overrides it, for example by typing in a different control.  Only one
         * form-associated element in a document can have this attribute specified.
         */
        this.autofocus = false;
        /**
         * The HTML5 button type.  Defaults to 'button'.
         */
        this.type = 'button';
        /**
         * This Boolean attribute indicates that the user cannot interact with the button.  Defaults to
         * false.
         */
        this.disabled = false;
        /**
         * Size of the button.
         */
        this.buttonSize = 'md';
        /**
         * The color for the button.  Defaults to 'primary'.
         */
        this.color = 'primary';
        /**
         * The flag to indicate whether the button should fill the container that it is in. Defaults to 'false'.
         */
        this.block = false;
    }
    moveAriaAttributes(hostEl, childEl) {
        const nodeMap = hostEl.attributes;
        // CUT every aria attribute from the host element and PASTE it into the button element
        Object.keys(nodeMap).forEach((key) => {
            const item = nodeMap.item(Number(key));
            let attrName;
            let attrVal;
            if (item) {
                attrName = item.name;
                attrVal = item.value;
                if (attrName.includes('aria')) {
                    this.renderer2.setAttribute(childEl, attrName, attrVal);
                    this.renderer2.removeAttribute(hostEl, attrName);
                }
            }
        });
    }
    ngAfterViewInit() {
        this.moveAriaAttributes(this.elRef.nativeElement, this.buttonEl && this.buttonEl.nativeElement);
    }
}
ButtonComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ButtonComponent, deps: [{ token: i0.ElementRef }, { token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component });
ButtonComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ButtonComponent, selector: "bb-button-ui", inputs: { autofocus: "autofocus", type: "type", disabled: "disabled", buttonSize: "buttonSize", color: "color", block: "block" }, host: { properties: { "class.btn-block": "this.block" } }, viewQueries: [{ propertyName: "buttonEl", first: true, predicate: ["button"], descendants: true, static: true }], ngImport: i0, template: "<button\n  #button\n  [type]=\"type\"\n  class=\"btn bb-button btn-{{ color }}\"\n  [autofocus]=\"autofocus\"\n  [disabled]=\"disabled\"\n  [class.btn-block]=\"block\"\n  [ngClass]=\"buttonSize === 'sm' ? 'btn-sm' : 'btn-md'\"\n>\n  <ng-content></ng-content>\n</button>\n", directives: [{ type: i1.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ButtonComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-button-ui',
                    templateUrl: './button.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.Renderer2 }]; }, propDecorators: { autofocus: [{
                type: Input
            }], type: [{
                type: Input
            }], disabled: [{
                type: Input
            }], buttonSize: [{
                type: Input
            }], color: [{
                type: Input
            }], block: [{
                type: HostBinding,
                args: ['class.btn-block']
            }, {
                type: Input
            }], buttonEl: [{
                type: ViewChild,
                args: ['button', { static: true }]
            }] } });
