// @backbase/ui-ang/esm2015/button/button.directive.js
import { Directive, HostBinding, Input } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * @name ButtonDirective
 *
 * @description
 * Directive for styling button.
 *
 * @a11y
 * The `bbButton` directive doesn't provide any specific properties for accessibility.
 * In case of necessity add `aria-*` attributes to the `button` element.
 */
export class ButtonDirective {
    constructor(renderer, hostElement) {
        this.renderer = renderer;
        this.hostElement = hostElement;
        /**
         * Button's type atrribute with default value.
         */
        this.type = 'button';
        /**
         * The color for the button. Defaults to 'primary'.
         * If a custom string used, then a css class is going to be added. The pattern is as follows:
         * `btn-customColor`
         */
        this.color = 'primary';
        /**
         * Size of the button.
         */
        this.buttonSize = 'md';
        /**
         * The flag to indicate whether the button should fill the container that it is in. Defaults to 'false'.
         */
        this.block = false;
        /**
         * The flag to indicate whether the button should be in a circular shape. Defaults to 'false'.
         * This will only work when there's only an icon inside the button without a text.
         */
        this.circle = false;
    }
    ngOnInit() {
        this.renderer.addClass(this.hostElement.nativeElement, 'btn');
        this.renderer.addClass(this.hostElement.nativeElement, `btn-${this.color}`);
        this.renderer.addClass(this.hostElement.nativeElement, `btn-${this.buttonSize}`);
    }
    ngOnChanges(changes) {
        if (changes.color) {
            if (changes.color.previousValue) {
                this.renderer.removeClass(this.hostElement.nativeElement, `btn-${changes.color.previousValue}`);
            }
            if (changes.color.currentValue) {
                this.renderer.addClass(this.hostElement.nativeElement, `btn-${changes.color.currentValue}`);
            }
        }
        if (changes.buttonSize) {
            if (changes.buttonSize.previousValue) {
                this.renderer.removeClass(this.hostElement.nativeElement, `btn-${changes.buttonSize.previousValue}`);
            }
            if (changes.buttonSize.currentValue) {
                this.renderer.addClass(this.hostElement.nativeElement, `btn-${changes.buttonSize.currentValue}`);
            }
        }
    }
}
ButtonDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ButtonDirective, deps: [{ token: i0.Renderer2 }, { token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive });
ButtonDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: ButtonDirective, selector: "button[bbButton]", inputs: { type: "type", color: "color", buttonSize: "buttonSize", block: "block", circle: "circle" }, host: { properties: { "type": "this.type", "class.btn-block": "this.block", "class.btn-circle": "this.circle" } }, usesOnChanges: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ButtonDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'button[bbButton]',
                }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }, { type: i0.ElementRef }]; }, propDecorators: { type: [{
                type: HostBinding,
                args: ['type']
            }, {
                type: Input
            }], color: [{
                type: Input
            }], buttonSize: [{
                type: Input
            }], block: [{
                type: HostBinding,
                args: ['class.btn-block']
            }, {
                type: Input
            }], circle: [{
                type: HostBinding,
                args: ['class.btn-circle']
            }, {
                type: Input
            }] } });
