// @backbase/accounts-transactions-journeys-shared-ui-account-ang/esm2015/lib/components/account-info-property/account-info-property.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AmountModule } from '@backbase/ui-ang/amount';
import { BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import { PaymentCardModule } from '@backbase/ui-ang/payment-card';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { ArrangementStateIndicatorModule, ProductNumberModule } from '@backbase/product-summary-ui';
import { DatePipeModule, FormatPipeModule, ProductSummaryCommon, ValidationModule, } from '@backbase/product-summary-common-ang';
import { GetDecimalsCountModule, GetValuePipeModule, ToArrayPipeModule, ToNumberPipeModule, } from '@backbase/accounts-transactions-journeys-shared-ui-transform-ang';
import { CastAccountPropertyPipeModule } from '../../pipes/cast-account-property/cast-account-property.module';
import { HasExternalInfoPipeModule } from '../../pipes/has-external-info/has-external-info.module';
import { ToPaymentCardDataPipeModule } from '../../pipes/to-payment-card-data/to-payment-card-data.module';
import { AccountInfoFrequencyModule } from '../account-info-frequency/account-info-frequency.module';
import { AccountInfoPropertyComponent } from './account-info-property.component';
import { AccountInfoRenewalStatusModule } from '../account-info-renewal-status/account-info-renewal-status.module';
import * as i0 from "@angular/core";
const uiModules = [
    AccountInfoFrequencyModule,
    AmountModule,
    ProductNumberModule,
    ArrangementStateIndicatorModule,
    TooltipModule,
    BbDatePipeModule,
    DatePipeModule,
    PaymentCardModule,
    AccountInfoRenewalStatusModule,
];
const pipeModules = [
    CastAccountPropertyPipeModule,
    GetDecimalsCountModule,
    GetValuePipeModule,
    HasExternalInfoPipeModule,
    ToArrayPipeModule,
    ToPaymentCardDataPipeModule,
    ToNumberPipeModule,
];
const sharedModules = [ProductSummaryCommon, ValidationModule, FormatPipeModule];
const components = [AccountInfoPropertyComponent];
export class AccountInfoPropertyModule {
}
/** @nocollapse */ AccountInfoPropertyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountInfoPropertyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ AccountInfoPropertyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountInfoPropertyModule, declarations: [AccountInfoPropertyComponent], imports: [CommonModule, AccountInfoFrequencyModule,
        AmountModule,
        ProductNumberModule,
        ArrangementStateIndicatorModule,
        TooltipModule,
        BbDatePipeModule,
        DatePipeModule,
        PaymentCardModule,
        AccountInfoRenewalStatusModule, ProductSummaryCommon, ValidationModule, FormatPipeModule, CastAccountPropertyPipeModule,
        GetDecimalsCountModule,
        GetValuePipeModule,
        HasExternalInfoPipeModule,
        ToArrayPipeModule,
        ToPaymentCardDataPipeModule,
        ToNumberPipeModule], exports: [AccountInfoFrequencyModule,
        AmountModule,
        ProductNumberModule,
        ArrangementStateIndicatorModule,
        TooltipModule,
        BbDatePipeModule,
        DatePipeModule,
        PaymentCardModule,
        AccountInfoRenewalStatusModule, ProductSummaryCommon, ValidationModule, FormatPipeModule, AccountInfoPropertyComponent, CastAccountPropertyPipeModule,
        GetDecimalsCountModule,
        GetValuePipeModule,
        HasExternalInfoPipeModule,
        ToArrayPipeModule,
        ToPaymentCardDataPipeModule,
        ToNumberPipeModule] });
/** @nocollapse */ AccountInfoPropertyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountInfoPropertyModule, imports: [[CommonModule, ...uiModules, ...sharedModules, ...pipeModules], AccountInfoFrequencyModule,
        AmountModule,
        ProductNumberModule,
        ArrangementStateIndicatorModule,
        TooltipModule,
        BbDatePipeModule,
        DatePipeModule,
        PaymentCardModule,
        AccountInfoRenewalStatusModule, ProductSummaryCommon, ValidationModule, FormatPipeModule, CastAccountPropertyPipeModule,
        GetDecimalsCountModule,
        GetValuePipeModule,
        HasExternalInfoPipeModule,
        ToArrayPipeModule,
        ToPaymentCardDataPipeModule,
        ToNumberPipeModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountInfoPropertyModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules, ...sharedModules, ...pipeModules],
                    declarations: [...components],
                    exports: [...uiModules, ...sharedModules, ...components, ...pipeModules],
                }]
        }] });
