// @backbase/product-summary-common-ang/esm2015/src/pipes/product-number/product-number-format.pipe.js
import { Pipe } from '@angular/core';
import { getProductNumberType } from '../../helpers/helpers';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/account-number-pipe";
/**
 * Provides numbers format per product or number type
 * a format includes masking and formatting configuration
 *
 * TODO: remove this pipe when all necessary ui-ang components support using `bbAccountNumber` pipe
 */
export class ProductNumberFormatPipe {
    /**
     * @name ProductNumberFormatPipe#constructor
     * @param accountsDisplayingFormatService - AccountsDisplayingFormatService
     *
     * @internal
     */
    constructor(accountsDisplayingFormatService) {
        this.accountsDisplayingFormatService = accountsDisplayingFormatService;
    }
    /**
     * @name ProductNumberFormatPipe#transform
     * @param product - product that should be formatted or specific product number type
     *
     * @description
     * Formats account numbers
     */
    transform(product) {
        const productType = typeof product === 'object' ? getProductNumberType(product) : product;
        const displayingFormat = this.accountsDisplayingFormatService.getDefaultFormat(productType);
        return Object.assign({ length: 0, maskRange: [0, 0] }, displayingFormat);
    }
}
ProductNumberFormatPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductNumberFormatPipe, deps: [{ token: i1.AccountsDisplayingFormatService }], target: i0.ɵɵFactoryTarget.Pipe });
ProductNumberFormatPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductNumberFormatPipe, name: "bbProductNumberFormat" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductNumberFormatPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'bbProductNumberFormat',
                }]
        }], ctorParameters: function () { return [{ type: i1.AccountsDisplayingFormatService }]; } });
