// @backbase/product-summary-common-ang/esm2015/src/pipes/product-number/product-number.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AccountNumberPipeModule } from '@backbase/ui-ang/account-number-pipe';
import { PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import { ProductNumberValuePipe } from './product-number-value.pipe';
import { ProductNumberTypePipe } from './product-number-type.pipe';
import { ProductNumberFormatPipe } from './product-number-format.pipe';
import * as i0 from "@angular/core";
const sharedPipes = [AccountNumberPipeModule, PaymentCardNumberModule];
const pipes = [ProductNumberValuePipe, ProductNumberTypePipe, ProductNumberFormatPipe];
export class ProductNumberPipeModule {
}
ProductNumberPipeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductNumberPipeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductNumberPipeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductNumberPipeModule, declarations: [ProductNumberValuePipe, ProductNumberTypePipe, ProductNumberFormatPipe], imports: [CommonModule, AccountNumberPipeModule, PaymentCardNumberModule], exports: [AccountNumberPipeModule, PaymentCardNumberModule, ProductNumberValuePipe, ProductNumberTypePipe, ProductNumberFormatPipe] });
ProductNumberPipeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductNumberPipeModule, imports: [[CommonModule, ...sharedPipes], AccountNumberPipeModule, PaymentCardNumberModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductNumberPipeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...sharedPipes],
                    declarations: [...pipes],
                    exports: [...sharedPipes, ...pipes],
                }]
        }] });
