// @backbase/accessgroup-common-ang/esm2015/lib/input-time-picker/input-time-picker.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { NgbTimepickerModule } from '@ng-bootstrap/ng-bootstrap';
import { TimePickerStringAdapter } from './input-time-picker.service';
import { AccessGroupInputTimePickerComponent } from './input-time-picker.component';
import * as i0 from "@angular/core";
export class AccessGroupInputTimePickerModule {
}
/** @nocollapse */ AccessGroupInputTimePickerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccessGroupInputTimePickerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ AccessGroupInputTimePickerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccessGroupInputTimePickerModule, declarations: [AccessGroupInputTimePickerComponent], imports: [CommonModule, NgbTimepickerModule, FormsModule, ReactiveFormsModule], exports: [AccessGroupInputTimePickerComponent] });
/** @nocollapse */ AccessGroupInputTimePickerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccessGroupInputTimePickerModule, providers: [TimePickerStringAdapter], imports: [[CommonModule, NgbTimepickerModule, FormsModule, ReactiveFormsModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccessGroupInputTimePickerModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, NgbTimepickerModule, FormsModule, ReactiveFormsModule],
                    declarations: [AccessGroupInputTimePickerComponent],
                    exports: [AccessGroupInputTimePickerComponent],
                    providers: [TimePickerStringAdapter],
                }]
        }] });
