// @backbase/ui-ang/esm2015/_example/tooltip-directive/showcase/tooltip-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TooltipShowcaseComponent } from './tooltip-showcase.component';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { IconModule } from '@backbase/ui-ang/icon';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import * as i0 from "@angular/core";
export class TooltipShowcaseModule {
}
TooltipShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TooltipShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
TooltipShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TooltipShowcaseModule, declarations: [TooltipShowcaseComponent], imports: [CommonModule, TooltipModule, IconModule, InputTextModule] });
TooltipShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TooltipShowcaseModule, imports: [[CommonModule, TooltipModule, IconModule, InputTextModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TooltipShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, TooltipModule, IconModule, InputTextModule],
                    declarations: [TooltipShowcaseComponent],
                    entryComponents: [TooltipShowcaseComponent],
                }]
        }] });
