// @backbase/ui-ang/esm2015/_example/product-item-debit-card/showcase/product-item-debit-card-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ProductItemDebitCardShowcaseComponent } from './product-item-debit-card-showcase.component';
import { ProductItemDebitCardModule } from '@backbase/ui-ang/product-item-debit-card';
import * as i0 from "@angular/core";
export class ProductItemDebitCardShowcaseModule {
}
ProductItemDebitCardShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemDebitCardShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductItemDebitCardShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemDebitCardShowcaseModule, declarations: [ProductItemDebitCardShowcaseComponent], imports: [CommonModule, ProductItemDebitCardModule] });
ProductItemDebitCardShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemDebitCardShowcaseModule, imports: [[CommonModule, ProductItemDebitCardModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemDebitCardShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ProductItemDebitCardModule],
                    declarations: [ProductItemDebitCardShowcaseComponent],
                    entryComponents: [ProductItemDebitCardShowcaseComponent],
                }]
        }] });
