// @backbase/ui-ang/esm2015/_example/payment-card/showcase/payment-card-showcase.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/payment-card";
let PaymentCardShowcaseComponent = class PaymentCardShowcaseComponent {
    get paymentCard() {
        return {
            name: 'Ricardo Peterson',
            number: '1234',
            vendor: 'mastercard',
            expirationDate: new Date(2021, 2),
        };
    }
    get paymentCardWithBackgroundImage() {
        return {
            name: 'Ricardo Peterson',
            number: '1234',
            vendor: 'mastercard',
            expirationDate: new Date(2021, 2),
            cardName: 'Blue Card',
        };
    }
    get paymentCardNumberFormat() {
        return {
            length: 16,
            maskRange: [0, 12],
            segments: 4,
        };
    }
};
PaymentCardShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardShowcaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
PaymentCardShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: PaymentCardShowcaseComponent, selector: "bb-payment-card-ui-showcase", ngImport: i0, template: "<div class=\"bb-block bb-block--xl\">\n  <h2>Default</h2>\n  <bb-payment-card-ui\n    [paymentCard]=\"paymentCard\"\n    [paymentCardNumberFormat]=\"paymentCardNumberFormat\"\n  ></bb-payment-card-ui>\n</div>\n\n<div class=\"bb-block bb-block--xl\">\n  <h2>With different background <small class=\"bb-text-support\">(controlled by card name)</small></h2>\n  <bb-payment-card-ui\n    [paymentCard]=\"paymentCardWithBackgroundImage\"\n    [paymentCardNumberFormat]=\"paymentCardNumberFormat\"\n  ></bb-payment-card-ui>\n</div>\n\n<div class=\"bb-block bb-block--xl\">\n  <h2>With state overlay <small class=\"bb-text-support\">(provided by css, not component)</small></h2>\n  <div class=\"bb-payment-card-state\">\n    <div class=\"bb-payment-card-state__container\">\n      <i\n        aria-hidden=\"true\"\n        class=\"bb-payment-card-state__icon bb-icon bb-icon-not-interested bb-icon--xxl bb-icon--white\"\n      ></i>\n      <span class=\"bb-payment-card-state__label badge badge-secondary\">Deactivated</span>\n    </div>\n    <div class=\"bb-payment-card-state__overlay\">\n      <bb-payment-card-ui\n        [paymentCard]=\"paymentCard\"\n        [paymentCardNumberFormat]=\"paymentCardNumberFormat\"\n      ></bb-payment-card-ui>\n    </div>\n  </div>\n</div>\n", components: [{ type: i1.PaymentCardComponent, selector: "bb-payment-card-ui", inputs: ["paymentCard", "paymentCardNumberFormat"] }] });
PaymentCardShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
], PaymentCardShowcaseComponent);
export { PaymentCardShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-payment-card-ui-showcase',
                    templateUrl: './payment-card-showcase.component.html',
                }]
        }] });
