// @backbase/ui-ang/esm2015/_example/payment-card/showcase/payment-card-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { PaymentCardShowcaseComponent } from './payment-card-showcase.component';
import { PaymentCardModule } from '@backbase/ui-ang/payment-card';
import * as i0 from "@angular/core";
export class PaymentCardShowcaseModule {
}
PaymentCardShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PaymentCardShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardShowcaseModule, declarations: [PaymentCardShowcaseComponent], imports: [CommonModule, PaymentCardModule] });
PaymentCardShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardShowcaseModule, imports: [[CommonModule, PaymentCardModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, PaymentCardModule],
                    declarations: [PaymentCardShowcaseComponent],
                    entryComponents: [PaymentCardShowcaseComponent],
                }]
        }] });
