// @backbase/ui-ang/esm2015/stepper/stepper-step/stepper-step.component.js
import { Component, Input, ContentChildren } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * @name StepperStepComponent
 *
 * @description
 * Component that represents a single step that is a part of the stepper component.
 */
export class StepperStepComponent {
    constructor() {
        /**
         * Marks step as current
         */
        this.isCurrent = false;
        /**
         * Marks step as checked
         */
        this.isChecked = false;
        /**
         * Marks step as active
         */
        this.isActive = false;
    }
    ngAfterContentChecked() {
        var _a;
        // TODO: workaround for angular bug https://github.com/angular/angular/issues/10098#issuecomment-235157642
        // an infinite loop takes place in banking schematics
        // it requires more investigation to see the root cause
        this.steps = (_a = this._steps) === null || _a === void 0 ? void 0 : _a.filter((item) => item !== this);
    }
}
StepperStepComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StepperStepComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
StepperStepComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: StepperStepComponent, selector: "bb-stepper-step-ui", inputs: { label: "label", stateLabel: "stateLabel", data: "data", ariaLabel: "ariaLabel", isCurrent: "isCurrent", isChecked: "isChecked", isActive: "isActive" }, queries: [{ propertyName: "_steps", predicate: StepperStepComponent }], ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StepperStepComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-stepper-step-ui',
                    template: '',
                }]
        }], propDecorators: { _steps: [{
                type: ContentChildren,
                args: [StepperStepComponent]
            }], label: [{
                type: Input
            }], stateLabel: [{
                type: Input
            }], data: [{
                type: Input
            }], ariaLabel: [{
                type: Input
            }], isCurrent: [{
                type: Input
            }], isChecked: [{
                type: Input
            }], isActive: [{
                type: Input
            }] } });
