// @backbase/payord-form-builder-ang/esm2015/lib/components/credit-debit-selector/payord-credit-debit-selector.js
import { Component } from '@angular/core';
import { DeprecationsService } from '@backbase/foundation-ang/future';
import { registerFormControl } from '../../helpers/form-helpers';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/future";
import * as i2 from "@backbase/payment-orders-ang";
import * as i3 from "@angular/common";
/**
 * @deprecated Will be removed in widget collection v3.11.0
 */
export class PayordCreditDebitSelectorComponent {
    constructor(deprecationsService) {
        this.deprecationsService = deprecationsService;
        /**
         * @TODO Remove this component in 3.11.0
         */
        deprecationsService.logDeprecatedFeature(`
      bb-payord-credit-debit-selector has been deprecated since 3.10.0, it will be removed in 3.11.0\n
      To define credit/debit options use \`creditDebitOptions\` property from \`ManualBatchTypeConfig\`
      type defined in \`@backbase/business-ang/batch\`
    `);
    }
    ngOnInit() {
        var _a, _b, _c, _d, _e;
        this.control = registerFormControl(this.group, (_a = this.config) === null || _a === void 0 ? void 0 : _a.name, (_b = this.options) === null || _b === void 0 ? void 0 : _b.validators, (_c = this.options) === null || _c === void 0 ? void 0 : _c.asyncValidators, ((_d = this.options) === null || _d === void 0 ? void 0 : _d.defaultValue) ||
            (((_e = this.options) === null || _e === void 0 ? void 0 : _e.choices) && this.options.choices[0]), this.config);
    }
}
/** @nocollapse */ PayordCreditDebitSelectorComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordCreditDebitSelectorComponent, deps: [{ token: i1.DeprecationsService }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ PayordCreditDebitSelectorComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: PayordCreditDebitSelectorComponent, selector: "bb-payord-credit-debit-selector", ngImport: i0, template: "<div class=\"bb-block bb-block--lg\">\n  <div class=\"bb-block bb-block--sm\">\n    <div *ngIf=\"options?.label\" class=\"bb-heading-3\">{{options?.label}}</div>\n    <div\n      *ngIf=\"options?.helperText\"\n      class=\"bb-dynamic-input__desc bb-text-support bb-subtitle\"\n    >\n      {{options?.helperText}}\n    </div>\n  </div>\n  <bb-payord-card-menu [items]=\"options?.choices\"></bb-payord-card-menu>\n</div>\n", components: [{ type: i2.ӨPayordCardMenuComponent, selector: "bb-payord-card-menu", inputs: ["items", "initialSelection"], outputs: ["select"] }], directives: [{ type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordCreditDebitSelectorComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-payord-credit-debit-selector',
                    templateUrl: './payord-credit-debit-selector.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.DeprecationsService }]; } });
