// @backbase/campaign-space-ang/bundles/backbase-campaign-space-ang.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('@backbase/foundation-ang/core'), require('rxjs'), require('@backbase/engagement-http-ang'), require('rxjs/operators'), require('@backbase/data-ang/engagement'), require('@backbase/foundation-ang/web-sdk'), require('@angular/common/http'), require('@backbase/foundation-ang/data-http')) :
    typeof define === 'function' && define.amd ? define('@backbase/campaign-space-ang', ['exports', '@angular/core', '@angular/common', '@backbase/foundation-ang/core', 'rxjs', '@backbase/engagement-http-ang', 'rxjs/operators', '@backbase/data-ang/engagement', '@backbase/foundation-ang/web-sdk', '@angular/common/http', '@backbase/foundation-ang/data-http'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["campaign-space-ang"] = {}), global.ng.core, global.ng.common, global.i1$1, global.rxjs, global.i1, global.rxjs.operators, global.engagement, global.webSdk, global.ng.common.http, global.dataHttp));
})(this, (function (exports, i0, i2, i1$1, rxjs, i1, operators, engagement, webSdk, http, dataHttp) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i1__namespace$1 = /*#__PURE__*/_interopNamespace(i1$1);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from, pack) {
        if (pack || arguments.length === 2)
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar)
                        ar = Array.prototype.slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
        return to.concat(ar || Array.prototype.slice.call(from));
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    var CampaignSpaceConfigurationToken = new i0.InjectionToken('CampaignSpaceConfiguration injection token');
    var staticConfig = {
        portalName: 'backbase-wc3',
        locale: 'en-US',
        designMode: false,
    };
    var CampaignService = /** @class */ (function () {
        function CampaignService(selectCampaignService, config) {
            this.selectCampaignService = selectCampaignService;
            this._config = Object.assign(Object.assign({}, staticConfig), config);
        }
        Object.defineProperty(CampaignService.prototype, "portalName", {
            get: function () {
                return this._config.portalName;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(CampaignService.prototype, "locale", {
            get: function () {
                return this._config.locale;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(CampaignService.prototype, "designMode", {
            get: function () {
                return this._config.designMode;
            },
            enumerable: false,
            configurable: true
        });
        CampaignService.prototype.selectCampaign = function (widgetName, dimensions, pageName) {
            var spaceId = this.portalName + "." + pageName + "." + widgetName;
            return this.selectCampaignService.selectCampaign({ locale: this.locale, dimensions: dimensions, spaceId: spaceId });
        };
        CampaignService.prototype.b64Decode = function (str) {
            return decodeURIComponent(atob(str)
                .split('')
                .map(function (c) {
                return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2);
            })
                .join(''));
        };
        return CampaignService;
    }());
    CampaignService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CampaignService, deps: [{ token: i1__namespace.SelectCampaignService }, { token: CampaignSpaceConfigurationToken, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    CampaignService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CampaignService });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CampaignService, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () {
            return [{ type: i1__namespace.SelectCampaignService }, { type: undefined, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.Inject,
                            args: [CampaignSpaceConfigurationToken]
                        }] }];
        } });

    var TargetUrlMode;
    (function (TargetUrlMode) {
        TargetUrlMode["SELF"] = "self";
        TargetUrlMode["BLANK"] = "blank";
    })(TargetUrlMode || (TargetUrlMode = {}));
    var TargetAttribute;
    (function (TargetAttribute) {
        TargetAttribute["self"] = "_self";
        TargetAttribute["blank"] = "_blank";
    })(TargetAttribute || (TargetAttribute = {}));

    var CampaignSpaceComponent = /** @class */ (function () {
        function CampaignSpaceComponent(campaignService, renderer, cdr, basePath) {
            this.campaignService = campaignService;
            this.renderer = renderer;
            this.cdr = cdr;
            this.basePath = basePath;
            /**
             * Component name
             */
            this.name = '';
            /**
             * Campaign banner dimensions
             */
            this.dimensions = '300x250';
            /**
             * Page name
             */
            this.pageName = 'index';
            /**
             * Is banner responsive
             */
            this.responsive = false;
            /**
             * Banner title
             */
            this.title = 'Campaign Space';
            this.designMode = this.campaignService.designMode;
            this.placeholderTitle = '';
            this.iframeContext = new rxjs.BehaviorSubject(undefined);
            this.destroy$ = new rxjs.Subject();
        }
        CampaignSpaceComponent.prototype.ngOnInit = function () {
            this.widgetStyles = this.getWidgetStyles();
            if (this.designMode) {
                this.initDesignMode();
            }
            else {
                this.initLiveMode();
            }
        };
        CampaignSpaceComponent.prototype.ngOnDestroy = function () {
            this.destroy$.next();
        };
        CampaignSpaceComponent.prototype.initDesignMode = function () {
            this.placeholderTitle = this.getPlaceholderTitle();
        };
        CampaignSpaceComponent.prototype.initLiveMode = function () {
            var _this = this;
            this.campaignService.selectCampaign(this.name, this.dimensions, this.pageName).subscribe(function (campaign) {
                if (campaign) {
                    _this.campaign = campaign;
                    _this.setTargetUrl();
                    _this.content = _this.campaignService.b64Decode(campaign.creative.content);
                    _this.iframeContext.next(_this.content);
                    _this.targetMode = TargetAttribute[campaign.targetUrlDisplayMode || 'self'];
                    _this.cdr.detectChanges();
                }
            });
        };
        CampaignSpaceComponent.prototype.setTargetUrl = function () {
            var _a, _b;
            var basePath = this.basePath.endsWith('/') ? this.basePath.slice(0, -1) : this.basePath;
            var campaignUrl = ((_a = this.campaign) === null || _a === void 0 ? void 0 : _a.targetUrl.startsWith('/'))
                ? this.campaign.targetUrl.slice(1)
                : (_b = this.campaign) === null || _b === void 0 ? void 0 : _b.targetUrl;
            this.targetUrl = basePath + "/" + campaignUrl;
        };
        CampaignSpaceComponent.prototype.iframeLoad = function () {
            var _this = this;
            this.cdr.detectChanges();
            var iframeElement = this.iframeRef.nativeElement;
            var iframeBody = iframeElement.contentDocument.body;
            this.renderer.setStyle(iframeElement, 'width', '100%');
            this.renderer.setStyle(iframeElement, 'border', '0');
            this.renderer.setStyle(iframeBody, 'margin', '0');
            this.iframeContext
                .pipe(operators.filter(Boolean), operators.tap(function (content) { return _this.renderer.setProperty(iframeBody, 'innerHTML', content); }), operators.delay(300), operators.takeUntil(this.destroy$))
                .subscribe(function () {
                var currentHeight = iframeBody.scrollHeight;
                var previousHeight = 0;
                var counter = 0;
                do {
                    _this.renderer.setStyle(iframeElement, 'height', currentHeight + "px");
                    previousHeight = currentHeight;
                    currentHeight = iframeBody.scrollHeight;
                    counter++;
                } while (previousHeight !== currentHeight && counter < 5);
                _this.cdr.detectChanges();
            });
        };
        CampaignSpaceComponent.prototype.getWidgetStyles = function () {
            var _c = __read(this.getDimensions(this.dimensions), 2), width = _c[0], height = _c[1];
            var properties = {
                width: this.responsive ? '100%' : this.toPx(width),
                height: this.toPx(height),
            };
            if (this.responsive) {
                properties.maxWidth = this.toPx(width);
            }
            return properties;
        };
        CampaignSpaceComponent.prototype.getPlaceholderTitle = function () {
            var _c = __read(this.getDimensions(this.dimensions), 2), width = _c[0], height = _c[1];
            var placeholderTitle = this.title + " - " + this.dimensions;
            if (this.responsive) {
                placeholderTitle = this.title + " - Responsive Max-width: " + width + ", Height: " + height;
            }
            return placeholderTitle;
        };
        CampaignSpaceComponent.prototype.toPx = function (value) {
            return value + 'px';
        };
        CampaignSpaceComponent.prototype.getDimensions = function (dimensions) {
            return dimensions.split('x');
        };
        return CampaignSpaceComponent;
    }());
    CampaignSpaceComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CampaignSpaceComponent, deps: [{ token: CampaignService }, { token: i0__namespace.Renderer2 }, { token: i0__namespace.ChangeDetectorRef }, { token: engagement.ENGAGEMENT_BASE_PATH, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Component });
    CampaignSpaceComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: CampaignSpaceComponent, selector: "bb-campaign-space-ang", inputs: { name: "name", dimensions: "dimensions", pageName: "pageName", responsive: "responsive", title: "title" }, providers: [CampaignService], viewQueries: [{ propertyName: "iframeRef", first: true, predicate: ["iframeRef"], descendants: true }], ngImport: i0__namespace, template: "<ng-container *ngIf=\"widgetStyles\">\n  <div *ngIf=\"!designMode && campaign\" class=\"bb-campaign-space-widget\" [ngStyle]=\"widgetStyles\">\n    <a [href]=\"targetUrl\" [ngStyle]=\"widgetStyles\" [target]=\"targetMode\" class=\"bb-campaign-space-widget__overlay\"> </a>\n    <iframe #iframeRef class=\"bb-campaign-space-widget__iframe\" [ngStyle]=\"widgetStyles\" (load)=\"iframeLoad()\">\n    </iframe>\n  </div>\n  <div *ngIf=\"designMode\" [ngStyle]=\"widgetStyles\" class=\"bb-campaign-space-widget\">\n    <div [ngStyle]=\"widgetStyles\" class=\"bb-campaign-space-widget__placeholder\">\n      <span>{{ placeholderTitle }}</span>\n    </div>\n  </div>\n</ng-container>\n", styles: [":host{position:relative}.bb-campaign-space-widget{position:relative;left:50%;transform:translate(-50%)}.bb-campaign-space-widget__overlay{position:absolute;top:0;left:0;right:0;bottom:0;z-index:1}.bb-campaign-space-widget__iframe{position:absolute;top:0;left:0;right:0;bottom:0;z-index:0}.bb-campaign-space-widget__placeholder{display:flex;justify-content:center;align-items:center;position:relative;background-color:#0000001a;border:1px solid #bfbfbf;text-align:center}\n"], directives: [{ type: i2__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i2__namespace.NgStyle, selector: "[ngStyle]", inputs: ["ngStyle"] }] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CampaignSpaceComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-campaign-space-ang',
                        templateUrl: './campaign-space.component.html',
                        styleUrls: ['./campaign-space.component.css'],
                        providers: [CampaignService],
                    }]
            }], ctorParameters: function () {
            return [{ type: CampaignService }, { type: i0__namespace.Renderer2 }, { type: i0__namespace.ChangeDetectorRef }, { type: undefined, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.Inject,
                            args: [engagement.ENGAGEMENT_BASE_PATH]
                        }] }];
        }, propDecorators: { name: [{
                    type: i0.Input
                }], dimensions: [{
                    type: i0.Input
                }], pageName: [{
                    type: i0.Input
                }], responsive: [{
                    type: i0.Input
                }], title: [{
                    type: i0.Input
                }], iframeRef: [{
                    type: i0.ViewChild,
                    args: ['iframeRef', { static: false }]
                }] } });

    var CampaignSpaceModule = /** @class */ (function () {
        function CampaignSpaceModule() {
        }
        return CampaignSpaceModule;
    }());
    CampaignSpaceModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CampaignSpaceModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    CampaignSpaceModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CampaignSpaceModule, declarations: [CampaignSpaceComponent], imports: [i2.CommonModule,
            webSdk.WebSdkApiModule,
            http.HttpClientModule, i1__namespace$1.BackbaseCoreModule], exports: [CampaignSpaceComponent] });
    CampaignSpaceModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CampaignSpaceModule, imports: [[
                i2.CommonModule,
                webSdk.WebSdkApiModule,
                http.HttpClientModule,
                i1$1.BackbaseCoreModule.withConfig({
                    classMap: { CampaignSpaceComponent: CampaignSpaceComponent },
                }),
            ]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CampaignSpaceModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [
                            i2.CommonModule,
                            webSdk.WebSdkApiModule,
                            http.HttpClientModule,
                            i1$1.BackbaseCoreModule.withConfig({
                                classMap: { CampaignSpaceComponent: CampaignSpaceComponent },
                            }),
                        ],
                        exports: [CampaignSpaceComponent],
                        declarations: [CampaignSpaceComponent],
                    }]
            }] });

    var mockCampaign = {
        targetUrl: '../../../../',
        targetUrlDisplayMode: TargetUrlMode.BLANK,
        creative: {
            uuid: '34793c5e-bc0b-454b-b97a-994159d01904',
            dimensions: '300x250',
            entryAsset: './index.html',
            basePath: '../../assets/creatives/', // ../../ to avoid the prepended gateway/api (apiRoot)
        },
    };
    /**
     * Patterns to setup mock provider.
     */
    var examples = [
        {
            urlPattern: 'gateway/api/engagement/client-api/v1/engagements/banner',
            method: 'GET',
            responses: [
                {
                    status: 200,
                    body: mockCampaign,
                    type: 'json',
                },
            ],
        },
    ];
    /**
     * Mocks provider for set campaign space widget.
     */
    var CampaignSpaceHttpMocksProvider = dataHttp.createMocks(examples);

    /**
     * Generated bundle index. Do not edit.
     */

    exports.CampaignService = CampaignService;
    exports.CampaignSpaceComponent = CampaignSpaceComponent;
    exports.CampaignSpaceConfigurationToken = CampaignSpaceConfigurationToken;
    exports.CampaignSpaceHttpMocksProvider = CampaignSpaceHttpMocksProvider;
    exports.CampaignSpaceModule = CampaignSpaceModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
