// @backbase/ui-ang/esm2015/table/directives/table-focus.directive.js
import { ContentChildren, Directive, ElementRef, HostListener, Input, } from '@angular/core';
import { TableRowDirective } from './table-row.directive';
import { getKeyCode, KEY_CODES } from '@backbase/ui-ang/util';
import { takeUntil } from 'rxjs/operators';
import { Subject } from 'rxjs';
import * as i0 from "@angular/core";
export class TableFocusDirective {
    constructor(renderer) {
        this.renderer = renderer;
        this.focusedItemIndex = -1;
        this.focusFirstRowOnChanges = false;
        this.destroy$ = new Subject();
    }
    onKeyEvent(event) {
        switch (getKeyCode(event)) {
            case KEY_CODES.DOWN:
                this.focusedItemIndex = Math.min(this.focusedItemIndex + 1, this.listItems.length - 1);
                this.focusItem();
                event.preventDefault();
                break;
            case KEY_CODES.UP:
                this.focusedItemIndex = Math.max(this.focusedItemIndex - 1, 0);
                this.focusItem();
                event.preventDefault();
                break;
            case KEY_CODES.ENTER:
                const currentActiveItem = this.listItems.find((item, index) => index === this.focusedItemIndex);
                if (currentActiveItem) {
                    currentActiveItem.nativeElement.click();
                    event.preventDefault();
                }
                break;
        }
    }
    focusItem(focusNativeElement = true) {
        this.listItems.forEach((el, index) => {
            if (index === this.focusedItemIndex) {
                this.renderer.addClass(el.nativeElement, 'active');
                this.renderer.setAttribute(el.nativeElement, 'tabIndex', '0');
                if (focusNativeElement) {
                    el.nativeElement.focus();
                }
            }
            else {
                this.renderer.removeClass(el.nativeElement, 'active');
                this.renderer.setAttribute(el.nativeElement, 'tabIndex', '-1');
            }
        });
    }
    /**
     * @internal
     */
    ngAfterViewInit() {
        // make first row in a table focusable
        this.listItems.changes.pipe(takeUntil(this.destroy$)).subscribe((items) => {
            var _a;
            if ((_a = items.first) === null || _a === void 0 ? void 0 : _a.nativeElement) {
                this.focusedItemIndex = 0;
                this.focusItem(this.focusFirstRowOnChanges);
            }
        });
    }
    /**
     * @internal
     */
    ngOnDestroy() {
        this.destroy$.next();
        this.destroy$.complete();
    }
}
TableFocusDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableFocusDirective, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Directive });
TableFocusDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: TableFocusDirective, selector: "[bbTableFocus]", inputs: { focusFirstRowOnChanges: "focusFirstRowOnChanges" }, host: { listeners: { "keydown": "onKeyEvent($event)" } }, queries: [{ propertyName: "listItems", predicate: TableRowDirective, emitDistinctChangesOnly: false, descendants: true, read: ElementRef }], ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableFocusDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbTableFocus]',
                }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { listItems: [{
                type: ContentChildren,
                args: [TableRowDirective, { read: ElementRef, descendants: true, emitDistinctChangesOnly: false }]
            }], focusFirstRowOnChanges: [{
                type: Input
            }], onKeyEvent: [{
                type: HostListener,
                args: ['keydown', ['$event']]
            }] } });
