// @backbase/ui-ang/esm2015/table/directives/table-sortable.directive.js
import { ContentChild, ContentChildren, Directive, EventEmitter, HostBinding, HostListener, Input, Output, } from '@angular/core';
import { switchMap } from 'rxjs/operators';
import { merge } from 'rxjs';
import { DropdownMenuToken, DropdownMenuOptionToken } from '@backbase/ui-ang/dropdown-menu';
import * as i0 from "@angular/core";
import * as i1 from "./table.directive";
const rotate = { asc: 'desc', desc: '', '': 'asc' };
export class TableSortableDirective {
    constructor(parentTable, renderer) {
        this.parentTable = parentTable;
        this.renderer = renderer;
        /**
         * The current sort direction. Accepts one of the values of 'asc', 'desc' or ''
         */
        this.direction = '';
        /**
         * An event emitter for on sort actions.
         */
        this.sort = new EventEmitter();
        this.sortable = true;
        this.parentTable.registerHeader(this);
    }
    get asc() {
        return this.direction === 'asc';
    }
    get desc() {
        return this.direction === 'desc';
    }
    rotate() {
        if (this.dropdown) {
            return;
        }
        this.emitSortEvent();
    }
    ngAfterViewInit() {
        if (this.dropdown) {
            this.presetDropdown(this.dropdown);
        }
        this.getMenuOptionSelectionChanges(this.bbMenuOptions).subscribe((menuOption) => this.onMenuOptionSelect(menuOption));
    }
    onMenuOptionSelect(menuOption) {
        var _a;
        const columnValue = menuOption.bbDropdownMenuOption;
        const activeMenuOption = (_a = this.bbMenuOptions) === null || _a === void 0 ? void 0 : _a.find((option) => option.active);
        // Reset currently active item
        if (activeMenuOption) {
            activeMenuOption.active = false;
        }
        // If same menu item is not clicked, then reset the direction for new menuitem
        if (activeMenuOption && columnValue !== activeMenuOption.bbDropdownMenuOption) {
            this.direction = '';
        }
        this.bbSortable = menuOption.bbDropdownMenuOption;
        this.emitSortEvent();
        this.markMenuOptionAsActive(menuOption);
        this.updateAriaSortAttribute();
    }
    /** Notify table with new sorting params */
    emitSortEvent() {
        if (this.bbSortable) {
            this.direction = rotate[this.direction];
            this.parentTable.onSort({ column: this.bbSortable, direction: this.direction });
        }
    }
    /** Combined stream of all of the bbTableMenuitem selectionChange events. */
    getMenuOptionSelectionChanges(menuOptions) {
        return menuOptions.changes.pipe(switchMap((_) => merge(...menuOptions.map((menuOption) => menuOption.selectionChange))));
    }
    /** mark the new menuitem as active only if sorting order is asc/desc */
    markMenuOptionAsActive(menuOption) {
        if (this.direction !== '') {
            menuOption.active = true;
        }
    }
    markAllMenuOptionAsInactive() {
        if (this.bbMenuOptions.length) {
            this.bbMenuOptions.forEach((menuOption) => (menuOption.active = false));
        }
    }
    updateAriaSortAttribute() {
        var _a;
        if (this.dropdown) {
            const buttonEl = (_a = this.dropdown.buttonEl) === null || _a === void 0 ? void 0 : _a.nativeElement;
            if (!this.direction) {
                this.renderer.removeAttribute(buttonEl, 'aria-sort');
            }
            else {
                this.renderer.setAttribute(buttonEl, 'aria-sort', this.getAriaSortAttribute(this.direction));
            }
        }
    }
    getAriaSortAttribute(direction) {
        return direction === 'asc' ? 'ascending' : 'descending';
    }
    /** Preset bb-dropdown button to add role for a11y and set full height and width of table column */
    presetDropdown(dropdown) {
        var _a;
        const buttonEl = (_a = dropdown.buttonEl) === null || _a === void 0 ? void 0 : _a.nativeElement;
        this.renderer.setAttribute(buttonEl, 'role', 'columnheader');
        this.renderer.setStyle(buttonEl, 'height', '100%');
        this.renderer.setStyle(buttonEl, 'width', '100%');
    }
    ngOnDestroy() {
        this.parentTable.unRegisterHeader(this);
    }
}
TableSortableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableSortableDirective, deps: [{ token: i1.TableDirective }, { token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Directive });
TableSortableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: TableSortableDirective, selector: "th[bbSortable]", inputs: { bbSortable: "bbSortable", direction: "direction" }, outputs: { sort: "sort" }, host: { listeners: { "click": "rotate()", "keyup.enter": "rotate()", "keyup.space": "rotate()" }, properties: { "class.sortable": "this.sortable", "class.asc": "this.asc", "class.desc": "this.desc" } }, queries: [{ propertyName: "dropdown", first: true, predicate: DropdownMenuToken, descendants: true }, { propertyName: "bbMenuOptions", predicate: DropdownMenuOptionToken, descendants: true }], ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableSortableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'th[bbSortable]',
                }]
        }], ctorParameters: function () { return [{ type: i1.TableDirective }, { type: i0.Renderer2 }]; }, propDecorators: { bbSortable: [{
                type: Input
            }], direction: [{
                type: Input
            }], sort: [{
                type: Output
            }], dropdown: [{
                type: ContentChild,
                args: [DropdownMenuToken]
            }], bbMenuOptions: [{
                type: ContentChildren,
                args: [DropdownMenuOptionToken, { descendants: true }]
            }], sortable: [{
                type: HostBinding,
                args: ['class.sortable']
            }], asc: [{
                type: HostBinding,
                args: ['class.asc']
            }], desc: [{
                type: HostBinding,
                args: ['class.desc']
            }], rotate: [{
                type: HostListener,
                args: ['click']
            }, {
                type: HostListener,
                args: ['keyup.enter']
            }, {
                type: HostListener,
                args: ['keyup.space']
            }] } });
