// @backbase/ui-ang/esm2015/table/directives/table.directive.js
import { Directive, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * @name TableDirective
 *
 * @description
 * Directive that displays a table.
 *
 * @a11y
 * The table directive doesn't provide any specific properties for accessibility.
 * But aria-sort tag can be set on the sortable header cells with ascending and descending values.
 *
 */
export class TableDirective {
    constructor() {
        /**
         * An event emitter for on select row action.
         */
        this.selectRow = new EventEmitter();
        /**
         * An event emitter for on row click action.
         */
        this.rowClick = new EventEmitter();
        /**
         * An event emitter for on column sort action.
         */
        this.sort = new EventEmitter();
        this.selectedRows = new Set();
        this.headers = new Set();
    }
    registerHeader(header) {
        this.headers.add(header);
    }
    unRegisterHeader(header) {
        this.headers.delete(header);
    }
    onRowClick(rowItem) {
        this.rowClick.emit(rowItem);
    }
    isSelected(row) {
        return this.selectedRows.has(row);
    }
    toggleItem(row, isChecked) {
        if (isChecked) {
            this.selectedRows.add(row);
        }
        else {
            this.selectedRows.delete(row);
        }
        this.selectRow.emit(this.selectedRows);
    }
    toggleAllItems() {
        if (this.isSelectedAll()) {
            this.selectedRows = new Set();
        }
        else {
            this.selectedRows = new Set(this.dataSource);
        }
        this.selectRow.emit(this.selectedRows);
    }
    isSelectedAll() {
        if (this.dataSource) {
            const matches = this.dataSource.filter((item) => this.selectedRows.has(item));
            return matches.length === this.dataSource.length;
        }
        return false;
    }
    isSelectedPart() {
        if (this.dataSource) {
            return this.selectedRows.size > 0 && this.selectedRows.size < this.dataSource.length;
        }
        return false;
    }
    onSort(sortEvent) {
        this.headers.forEach((header) => {
            if (header.bbSortable !== sortEvent.column) {
                header.direction = '';
                header.markAllMenuOptionAsInactive();
            }
        });
        this.sort.emit(sortEvent);
    }
}
TableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
TableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: TableDirective, selector: "table[bbTable]", inputs: { dataSource: ["bbTable", "dataSource"] }, outputs: { selectRow: "selectRow", rowClick: "rowClick", sort: "sort" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'table[bbTable]',
                }]
        }], propDecorators: { dataSource: [{
                type: Input,
                args: ['bbTable']
            }], selectRow: [{
                type: Output
            }], rowClick: [{
                type: Output
            }], sort: [{
                type: Output
            }] } });
