// @backbase/ui-ang/esm2015/table/directives/table-row.directive.js
import { Directive, HostBinding, HostListener, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./table.directive";
export class TableRowDirective {
    constructor(parentTable) {
        this.parentTable = parentTable;
    }
    get selected() {
        return this.parentTable.isSelected(this.bbRow);
    }
    onRowClick() {
        this.parentTable.onRowClick(this.bbRow);
    }
}
TableRowDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableRowDirective, deps: [{ token: i1.TableDirective }], target: i0.ɵɵFactoryTarget.Directive });
TableRowDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: TableRowDirective, selector: "tr[bbRow]", inputs: { bbRow: "bbRow" }, host: { listeners: { "click": "onRowClick()" }, properties: { "class.selected": "this.selected" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableRowDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'tr[bbRow]',
                }]
        }], ctorParameters: function () { return [{ type: i1.TableDirective }]; }, propDecorators: { bbRow: [{
                type: Input
            }], selected: [{
                type: HostBinding,
                args: ['class.selected']
            }], onRowClick: [{
                type: HostListener,
                args: ['click']
            }] } });
