// @backbase/transactions-common-ang/esm2015/src/components/transactions-controls/transactions-number-of-results-found.component.js
import { ChangeDetectionStrategy, Component, Directive, Input } from '@angular/core';
import { LoadingState } from '../../services/transactions.service';
import { BbTemplate } from '@backbase/foundation-ang/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class TransactionsNumberOfResultsFoundComponent {
    constructor() {
        /**
         * Reference to instance of customizable component.
         */
        this.hostRef = this;
        /**
         * Number of results found after filtering or searching.
         */
        this.numberOfResultsFound = 0;
        /**
         * The loading state of the transactions list.
         */
        this.loadingState = LoadingState.NotLoaded;
        /**
         * Whether filter is applied or not.
         */
        this.isFilterApplied = false;
        /**
         * Whether search is applied or not.
         */
        this.isSearchApplied = false;
    }
    /**
     * Whether loading has finished or not.
     */
    get isLoaded() {
        return this.loadingState === LoadingState.Loaded;
    }
}
TransactionsNumberOfResultsFoundComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TransactionsNumberOfResultsFoundComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
TransactionsNumberOfResultsFoundComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: TransactionsNumberOfResultsFoundComponent, selector: "bb-transactions-number-of-results-found", inputs: { numberOfResultsFound: "numberOfResultsFound", loadingState: "loadingState", isFilterApplied: "isFilterApplied", isSearchApplied: "isSearchApplied" }, ngImport: i0, template: "<ng-container\n  bbTransactionsNumberOfResultsFoundCustomizable\n  [bbHostRef]=\"hostRef\"\n></ng-container>\n\n\n<ng-template bbTransactionsNumberOfResultsFoundCustomizable let-hostRef>\n  <div\n    class=\"bb-card__header bb-subtitle\"\n    [ngClass]=\"{ 'sr-only': hostRef.numberOfResultsFound <= 0 }\"\n    i18n=\"Number of results found for the applied filter@@transactions.list.numberOfResultsFound\"\n    *ngIf=\"hostRef.isLoaded && (hostRef.isFilterApplied || hostRef.isSearchApplied)\"\n    aria-live=\"assertive\"\n  >\n    {hostRef.numberOfResultsFound, plural, =0 {No results} =1 {1 result} other {{{ hostRef.numberOfResultsFound }} results}} found\n  </div>\n</ng-template>", directives: [{ type: i0.forwardRef(function () { return TransactionsNumberOfResultsFoundCustomizableDirective; }), selector: "[bbTransactionsNumberOfResultsFoundCustomizable]" }, { type: i0.forwardRef(function () { return i1.NgIf; }), selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i0.forwardRef(function () { return i1.NgClass; }), selector: "[ngClass]", inputs: ["class", "ngClass"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TransactionsNumberOfResultsFoundComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-transactions-number-of-results-found',
                    templateUrl: 'transactions-number-of-results-found.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { numberOfResultsFound: [{
                type: Input
            }], loadingState: [{
                type: Input
            }], isFilterApplied: [{
                type: Input
            }], isSearchApplied: [{
                type: Input
            }] } });
export class TransactionsNumberOfResultsFoundCustomizableDirective extends BbTemplate {
}
TransactionsNumberOfResultsFoundCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TransactionsNumberOfResultsFoundCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
TransactionsNumberOfResultsFoundCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: TransactionsNumberOfResultsFoundCustomizableDirective, selector: "[bbTransactionsNumberOfResultsFoundCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TransactionsNumberOfResultsFoundCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbTransactionsNumberOfResultsFoundCustomizable]',
                }]
        }] });
