// @backbase/remote-config-ang/esm2015/angular/remote-config.service.js
import { Inject, Injectable, InjectionToken } from '@angular/core';
import { RemoteConfig } from '../core/remote-config';
import { RemoteConfigFetcherService } from './remote-config-fetcher.service';
export const RemoteConfigConfigurationToken = new InjectionToken('@backbase/remote-config-ang: RemoteConfigConfigurationToken');
export class RemoteConfigService {
    constructor(configuration, configFetcher) {
        this.configuration = configuration;
        this.configFetcher = configFetcher;
        this.remoteConfigClient = new RemoteConfig(this.configFetcher, {
            appName: configuration.appName,
            appVersion: configuration.appVersion,
            projectName: configuration.projectName,
            serviceRoot: configuration.serviceRoot,
        }, configuration.defaults);
        this.disabled = Boolean(configuration.disabled);
    }
    addCustomTag(tag) {
        this.remoteConfigClient.addCustomTag(tag);
    }
    clearCustomTags() {
        this.remoteConfigClient.clearCustomTags();
    }
    getValue(parameterName) {
        return this.remoteConfigClient.getValue(parameterName);
    }
    fetchAndActivate() {
        if (this.disabled) {
            return Promise.resolve();
        }
        return this.remoteConfigClient.fetchAndActivate();
    }
    removeCustomTag(tag) {
        return this.remoteConfigClient.removeCustomTag(tag);
    }
}
RemoteConfigService.decorators = [
    { type: Injectable }
];
RemoteConfigService.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Inject, args: [RemoteConfigConfigurationToken,] }] },
    { type: RemoteConfigFetcherService }
];
