// @backbase/user-manage-profile-widget-ang/esm2015/src/user-manage-profile.service.js
import { __rest } from "tslib";
import { Injectable } from '@angular/core';
import { map } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/user";
import * as i2 from "@backbase/user-common";
export class UserManageProfileService {
    constructor(data, manageProfileService) {
        this.data = data;
        this.manageProfileService = manageProfileService;
    }
    getUserProfile() {
        return this.data.getOwnUserProfile().pipe(map(this.manageProfileService.sortUserContactListsByPrimary));
    }
    createElectronicAddressRecord(electronicAddress) {
        return this.data.addOwnElectronicAddress({ electronicAddress });
    }
    updateElectronicAddressRecord(electronicAddress) {
        const { key } = electronicAddress, record = __rest(electronicAddress, ["key"]);
        return this.data.updateOwnElectronicAddress({
            electronicAddressKey: electronicAddress.key,
            electronicAddress: record,
        });
    }
    deleteElectronicAddressRecord(electronicAddress) {
        return this.data.deleteOwnElectronicAddress({ electronicAddressKey: electronicAddress.key });
    }
    createPhoneAddressRecord(phoneAddress) {
        return this.data.addOwnPhoneAddress({ phoneAddress });
    }
    updatePhoneAddressRecord(phoneAddress) {
        const { key } = phoneAddress, record = __rest(phoneAddress, ["key"]);
        return this.data.updateOwnPhoneAddress({
            phoneAddressKey: phoneAddress.key,
            phoneAddress: record,
        });
    }
    deletePhoneAddressRecord(phoneAddress) {
        return this.data.deleteOwnPhoneAddress({ phoneAddressKey: phoneAddress.key });
    }
    createPostalAddressRecord(postalAddress) {
        return this.data.addOwnPostalAddress({ postalAddress });
    }
    updatePostalAddressRecord(postalAddress) {
        const { key } = postalAddress, record = __rest(postalAddress, ["key"]);
        return this.data.updateOwnPostalAddress({
            postalAddressKey: postalAddress.key,
            postalAddress: record,
        });
    }
    deletePostalAddressRecord(postalAddress) {
        return this.data.deleteOwnPostalAddress({ postalAddressKey: postalAddress.key });
    }
}
UserManageProfileService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserManageProfileService, deps: [{ token: i1.UserProfileManagementService }, { token: i2.ManageProfileService }], target: i0.ɵɵFactoryTarget.Injectable });
UserManageProfileService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserManageProfileService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserManageProfileService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.UserProfileManagementService }, { type: i2.ManageProfileService }]; } });
