// @backbase/user-manage-profile-widget-ang/esm2015/src/user-manage-profile-widget.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { ErrorCommonStateModule } from '@backbase/ui-ang/common-error-state';
import { HeaderModule } from '@backbase/ui-ang/header';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { UserCommonModule } from '@backbase/user-common';
import { UserManageEmailAddressesModule } from './user-manage-email-addresses/user-manage-email-addresses.module';
import { UserManagePhoneNumbersModule } from './user-manage-phone-numbers/user-manage-phone-numbers.module';
import { UserManagePostalAddressesModule } from './user-manage-postal-addresses/user-manage-postal-addresses.module';
import { UserManageProfileWidgetExtensionsModule } from './user-manage-profile-widget-extensions.module';
import { UserManageProfileWidgetComponent, UserManageWidgetCustomizableDirective, UserManageWidgetTitleCustomizableDirective, } from './user-manage-profile-widget.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const uiModules = [ErrorCommonStateModule, LoadingIndicatorModule, HeaderModule];
const widgetModules = [
    UserManageProfileWidgetExtensionsModule,
    UserManagePhoneNumbersModule,
    UserManageEmailAddressesModule,
    UserManagePostalAddressesModule,
];
export class UserManageProfileWidgetModule {
}
UserManageProfileWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserManageProfileWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
UserManageProfileWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserManageProfileWidgetModule, declarations: [UserManageProfileWidgetComponent,
        UserManageWidgetCustomizableDirective,
        UserManageWidgetTitleCustomizableDirective], imports: [CommonModule, i1.BackbaseCoreModule, UserCommonModule, ErrorCommonStateModule, LoadingIndicatorModule, HeaderModule, UserManageProfileWidgetExtensionsModule,
        UserManagePhoneNumbersModule,
        UserManageEmailAddressesModule,
        UserManagePostalAddressesModule], exports: [UserManageProfileWidgetComponent,
        UserManageWidgetCustomizableDirective,
        UserManageWidgetTitleCustomizableDirective, ErrorCommonStateModule, LoadingIndicatorModule, HeaderModule, UserManageProfileWidgetExtensionsModule,
        UserManagePhoneNumbersModule,
        UserManageEmailAddressesModule,
        UserManagePostalAddressesModule] });
UserManageProfileWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserManageProfileWidgetModule, imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { UserManageProfileWidgetComponent },
            }),
            UserCommonModule,
            ...uiModules,
            ...widgetModules,
        ], ErrorCommonStateModule, LoadingIndicatorModule, HeaderModule, UserManageProfileWidgetExtensionsModule,
        UserManagePhoneNumbersModule,
        UserManageEmailAddressesModule,
        UserManagePostalAddressesModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserManageProfileWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { UserManageProfileWidgetComponent },
                        }),
                        UserCommonModule,
                        ...uiModules,
                        ...widgetModules,
                    ],
                    declarations: [
                        UserManageProfileWidgetComponent,
                        UserManageWidgetCustomizableDirective,
                        UserManageWidgetTitleCustomizableDirective,
                    ],
                    exports: [
                        UserManageProfileWidgetComponent,
                        UserManageWidgetCustomizableDirective,
                        UserManageWidgetTitleCustomizableDirective,
                        ...uiModules,
                        ...widgetModules,
                    ],
                }]
        }] });
