// @backbase/actions-common-ang/esm2015/src/components/time-picker/time-picker.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { DropdownSingleSelectModule } from '@backbase/ui-ang/dropdown-single-select';
import { ReactiveFormsModule } from '@angular/forms';
import { TimePickerComponent } from './time-picker.component';
import * as i0 from "@angular/core";
export class TimePickerModule {
}
TimePickerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: TimePickerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
TimePickerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: TimePickerModule, declarations: [TimePickerComponent], imports: [CommonModule, DropdownSingleSelectModule, ReactiveFormsModule], exports: [TimePickerComponent, DropdownSingleSelectModule, ReactiveFormsModule] });
TimePickerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: TimePickerModule, imports: [[CommonModule, DropdownSingleSelectModule, ReactiveFormsModule], DropdownSingleSelectModule, ReactiveFormsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: TimePickerModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, DropdownSingleSelectModule, ReactiveFormsModule],
                    declarations: [TimePickerComponent],
                    exports: [TimePickerComponent, DropdownSingleSelectModule, ReactiveFormsModule],
                }]
        }] });
