// @backbase/messages-common/esm2015/src/components/upload-attachments/messages-upload-attachments.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ButtonModule } from '@backbase/ui-ang/button';
import { IconModule } from '@backbase/ui-ang/icon';
import { InputFileModule } from '@backbase/ui-ang/input-file';
import { FileAttachmentModule } from '@backbase/ui-ang/file-attachment';
import { MessagesUploadAttachmentsComponent, MessagesUploadAttachmentsCustomizableDirective, } from './messages-upload-attachments.component';
import { SharedMethodsModule } from '../../services/shared-methods/shared-methods.module';
import { MessagesUploadAttachmentsService } from './services/messages-upload-attachments.service';
import * as i0 from "@angular/core";
const components = [MessagesUploadAttachmentsComponent];
const directives = [MessagesUploadAttachmentsCustomizableDirective];
const uiModules = [ButtonModule, FileAttachmentModule, IconModule, InputFileModule];
const innerModules = [SharedMethodsModule];
export class MessagesUploadAttachmentsModule {
}
MessagesUploadAttachmentsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesUploadAttachmentsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MessagesUploadAttachmentsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesUploadAttachmentsModule, declarations: [MessagesUploadAttachmentsComponent, MessagesUploadAttachmentsCustomizableDirective], imports: [CommonModule, ButtonModule, FileAttachmentModule, IconModule, InputFileModule, SharedMethodsModule], exports: [MessagesUploadAttachmentsComponent, MessagesUploadAttachmentsCustomizableDirective, ButtonModule, FileAttachmentModule, IconModule, InputFileModule, SharedMethodsModule] });
MessagesUploadAttachmentsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesUploadAttachmentsModule, providers: [MessagesUploadAttachmentsService], imports: [[CommonModule, ...uiModules, ...innerModules], ButtonModule, FileAttachmentModule, IconModule, InputFileModule, SharedMethodsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesUploadAttachmentsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules, ...innerModules],
                    declarations: [...components, ...directives],
                    exports: [...components, ...directives, ...uiModules, ...innerModules],
                    providers: [MessagesUploadAttachmentsService],
                }]
        }] });
