// @backbase/contact-manager-http-ang/esm2015/api/ibanRestrictions.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const IbanRestrictionsHttpServiceGetIbanRestrictionMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/iban-restrictions/countries/{code}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "ibanLength": 18
                }
            },
        ]
    }]);
export const IbanRestrictionsHttpServiceGetIbanRestrictionsMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/iban-restrictions/countries",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "countryCode": "NL",
                        "ibanLength": 18
                    }, {
                        "countryCode": "TR",
                        "ibanLength": 26
                    }]
            },
        ]
    }]);
export const IbanRestrictionsHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v2/iban-restrictions/countries/{code}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "ibanLength": 18
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/iban-restrictions/countries",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "countryCode": "NL",
                        "ibanLength": 18
                    }, {
                        "countryCode": "TR",
                        "ibanLength": 26
                    }]
            },
        ]
    },
]);
