// @backbase/contact-manager-http-ang/esm2015/api/supportedCountries.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const SupportedCountriesHttpServiceGetSupportedCountriesMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/supported-countries",
        method: "GET",
        responses: [
            {
                status: 200,
                body: ["AR", "AU", "AT", "BE", "BR", "CA", "CN", "CO", "HR", "CZ", "DK", "EC", "EG", "ET", "FR", "DE", "GR", "HN", "HK", "HU", "IN", "ID", "IE", "IT", "JM", "MY", "MX", "MA", "NA", "NL", "NZ", "NO", "PL", "PT", "RU", "SN"]
            },
        ]
    }]);
export const SupportedCountriesHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v2/supported-countries",
        method: "GET",
        responses: [
            {
                status: 200,
                body: ["AR", "AU", "AT", "BE", "BR", "CA", "CN", "CO", "HR", "CZ", "DK", "EC", "EG", "ET", "FR", "DE", "GR", "HN", "HK", "HU", "IN", "ID", "IE", "IT", "JM", "MY", "MX", "MA", "NA", "NL", "NZ", "NO", "PL", "PT", "RU", "SN"]
            },
        ]
    },
]);
