// @backbase/messages-common/esm2015/src/components/messages-conversation-message/messages-conversation-message.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FileAttachmentModule } from '@backbase/ui-ang/file-attachment';
import { IconModule } from '@backbase/ui-ang/icon';
import { KeyboardClickModule } from '@backbase/ui-ang/keyboard-click-directive';
import { MessagesConversationMessageComponent, MessagesConversationMessageCustomizableDirective, } from './messages-conversation-message.component';
import { MessagesBodyPreviewModule } from '../messages-preview/messages-preview.module';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import * as i0 from "@angular/core";
const uiModules = [FileAttachmentModule, IconModule, KeyboardClickModule, TooltipModule];
const innerModules = [MessagesBodyPreviewModule];
export class MessagesConversationMessageModule {
}
MessagesConversationMessageModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesConversationMessageModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MessagesConversationMessageModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesConversationMessageModule, declarations: [MessagesConversationMessageComponent, MessagesConversationMessageCustomizableDirective], imports: [CommonModule, FileAttachmentModule, IconModule, KeyboardClickModule, TooltipModule, MessagesBodyPreviewModule], exports: [MessagesConversationMessageComponent,
        MessagesConversationMessageCustomizableDirective, FileAttachmentModule, IconModule, KeyboardClickModule, TooltipModule, MessagesBodyPreviewModule] });
MessagesConversationMessageModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesConversationMessageModule, imports: [[CommonModule, ...uiModules, ...innerModules], FileAttachmentModule, IconModule, KeyboardClickModule, TooltipModule, MessagesBodyPreviewModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesConversationMessageModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [MessagesConversationMessageComponent, MessagesConversationMessageCustomizableDirective],
                    exports: [
                        MessagesConversationMessageComponent,
                        MessagesConversationMessageCustomizableDirective,
                        ...uiModules,
                        ...innerModules,
                    ],
                    imports: [CommonModule, ...uiModules, ...innerModules],
                }]
        }] });
