// @backbase/ui-ang/esm2015/_example/input-radio-group/showcase/input-radio-group-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { InputRadioGroupShowcaseComponent } from './input-radio-group-showcase.component';
import { InputRadioGroupModule } from '@backbase/ui-ang/input-radio-group';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import * as i0 from "@angular/core";
export class InputRadioGroupShowcaseModule {
}
InputRadioGroupShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputRadioGroupShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputRadioGroupShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputRadioGroupShowcaseModule, declarations: [InputRadioGroupShowcaseComponent], imports: [CommonModule, InputRadioGroupModule, ReactiveFormsModule, FormsModule] });
InputRadioGroupShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputRadioGroupShowcaseModule, imports: [[CommonModule, InputRadioGroupModule, ReactiveFormsModule, FormsModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputRadioGroupShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, InputRadioGroupModule, ReactiveFormsModule, FormsModule],
                    declarations: [InputRadioGroupShowcaseComponent],
                    entryComponents: [InputRadioGroupShowcaseComponent],
                }]
        }] });
