// @backbase/payment-orders-ang/esm2015/lib/ui/directives/bb-keyboard-navigation.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { KeyboardNavigationAnchorDirective, KeyboardNavigationDirective, KeyboardNavigationItemDirective, KeyboardNavigationListDirective, } from './bb-keyboard-navigation.directive';
import * as i0 from "@angular/core";
export class BbKeyboardNavigationModule {
}
/** @nocollapse */ BbKeyboardNavigationModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BbKeyboardNavigationModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ BbKeyboardNavigationModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BbKeyboardNavigationModule, declarations: [KeyboardNavigationItemDirective,
        KeyboardNavigationAnchorDirective,
        KeyboardNavigationListDirective,
        KeyboardNavigationDirective], imports: [CommonModule], exports: [KeyboardNavigationItemDirective,
        KeyboardNavigationAnchorDirective,
        KeyboardNavigationListDirective,
        KeyboardNavigationDirective] });
/** @nocollapse */ BbKeyboardNavigationModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BbKeyboardNavigationModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BbKeyboardNavigationModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        KeyboardNavigationItemDirective,
                        KeyboardNavigationAnchorDirective,
                        KeyboardNavigationListDirective,
                        KeyboardNavigationDirective,
                    ],
                    exports: [
                        KeyboardNavigationItemDirective,
                        KeyboardNavigationAnchorDirective,
                        KeyboardNavigationListDirective,
                        KeyboardNavigationDirective,
                    ],
                }]
        }] });
