// @backbase/payment-orders-ang/esm2015/lib/ui/directives/iban.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { IbanDirective } from './iban.directive';
import * as i0 from "@angular/core";
export class IbanModule {
}
/** @nocollapse */ IbanModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IbanModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ IbanModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IbanModule, declarations: [IbanDirective], imports: [CommonModule], exports: [IbanDirective] });
/** @nocollapse */ IbanModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IbanModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IbanModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [IbanDirective],
                    imports: [CommonModule],
                    exports: [IbanDirective],
                }]
        }] });
