// @backbase/cards-common-ang/esm2015/src/components/bb-payment-card-controls-base/payment-card-controls-base.component.js
import { Directive, EventEmitter, Input, Output } from '@angular/core';
import { ResetPinModeEnum } from '../../model/payment-card-reset-pin.model';
import { PaymentCardLockStatus } from '../../model/payment-card.model';
import * as i0 from "@angular/core";
// eslint-disable-next-line
export class PaymentCardControlsBase {
    constructor() {
        /**
         * Whether lock status is being updated.
         */
        this.isLockLoading = false;
        /**
         * Whether card replacement has been initiated.
         */
        this.isInitiateReplacementLoading = false;
        /**
         * Whether card replacement has been initiated.
         */
        this.isInitiateActivationLoading = false;
        /**
         * Whether card change pin has been initiated.
         */
        this.isResetPinLoading = false;
        /**
         * Event to update the lock status of the payment card.
         */
        this.updateLockStatus = new EventEmitter();
        /**
         * Event to initiate replacement of the payment card.
         */
        this.initiateReplacement = new EventEmitter();
        /**
         * Event to initiate activation of the payment card.
         */
        this.initiateActivation = new EventEmitter();
        /**
         * Event to initiate reset pin of the payment card.
         */
        this.resetPin = new EventEmitter();
        /**
         * Event to initiate requestPin pin of the payment card.
         */
        this.requestPin = new EventEmitter();
        this.hostRef = this;
    }
    ngOnInit() {
        if (typeof this.paymentCard === 'undefined') {
            throw new Error(`"paymentCard" input is required in "${this.constructor.name}"`);
        }
    }
    onLock() {
        this.updateLockStatus.emit(PaymentCardLockStatus.Locked);
    }
    onUnlock() {
        this.updateLockStatus.emit(PaymentCardLockStatus.Unlocked);
    }
    get isLocked() {
        return this.paymentCard && this.paymentCard.lockStatus === PaymentCardLockStatus.Locked;
    }
    get resetPinMode() {
        if (this.paymentCard) {
            if (this.paymentCard.type === 'Credit') {
                return ResetPinModeEnum.Request;
            }
            return ResetPinModeEnum.Reset;
        }
        return '';
    }
    onInitiateReplacement(replacementReason) {
        this.initiateReplacement.emit(replacementReason);
    }
    onInitiateActivation(token) {
        this.initiateActivation.emit(token);
    }
    onResetPin(token) {
        this.resetPin.emit(token);
    }
    onRequestPin(token) {
        this.requestPin.emit(token);
    }
}
PaymentCardControlsBase.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentCardControlsBase, deps: [], target: i0.ɵɵFactoryTarget.Directive });
PaymentCardControlsBase.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.15", type: PaymentCardControlsBase, inputs: { paymentCard: "paymentCard", isLockLoading: "isLockLoading", isInitiateReplacementLoading: "isInitiateReplacementLoading", isInitiateActivationLoading: "isInitiateActivationLoading", isResetPinLoading: "isResetPinLoading" }, outputs: { updateLockStatus: "updateLockStatus", initiateReplacement: "initiateReplacement", initiateActivation: "initiateActivation", resetPin: "resetPin", requestPin: "requestPin" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentCardControlsBase, decorators: [{
            type: Directive
        }], propDecorators: { paymentCard: [{
                type: Input
            }], isLockLoading: [{
                type: Input
            }], isInitiateReplacementLoading: [{
                type: Input
            }], isInitiateActivationLoading: [{
                type: Input
            }], isResetPinLoading: [{
                type: Input
            }], updateLockStatus: [{
                type: Output
            }], initiateReplacement: [{
                type: Output
            }], initiateActivation: [{
                type: Output
            }], resetPin: [{
                type: Output
            }], requestPin: [{
                type: Output
            }] } });
