// @backbase/product-summary-ui/esm2015/src/bb-account-header-ui/bb-account-header.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AmountModule } from '@backbase/ui-ang/amount';
import { IconModule } from '@backbase/ui-ang/icon';
import { InputInlineEditModule } from '@backbase/ui-ang/input-inline-edit';
import { ModalModule } from '@backbase/ui-ang/modal';
import { ValidationModule } from '@backbase/product-summary-common-ang';
import { ProductSummaryAccountLogoModule } from '../bb-product-summary-account-logo-ui/bb-product-summary-account-logo.module';
import { ProductNumberModule } from '../bb-product-number/product-number.module';
import { CreditLimitModule } from '../bb-credit-limit-ui/credit-limit.module';
import { AccountHeaderComponent } from './bb-account-header.component';
import * as i0 from "@angular/core";
const uiModules = [
    IconModule,
    ProductSummaryAccountLogoModule,
    InputInlineEditModule,
    ProductNumberModule,
    CreditLimitModule,
    AmountModule,
    ModalModule,
];
export class AccountHeaderModule {
}
AccountHeaderModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountHeaderModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AccountHeaderModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountHeaderModule, declarations: [AccountHeaderComponent], imports: [CommonModule, ValidationModule, IconModule,
        ProductSummaryAccountLogoModule,
        InputInlineEditModule,
        ProductNumberModule,
        CreditLimitModule,
        AmountModule,
        ModalModule], exports: [AccountHeaderComponent, IconModule,
        ProductSummaryAccountLogoModule,
        InputInlineEditModule,
        ProductNumberModule,
        CreditLimitModule,
        AmountModule,
        ModalModule] });
AccountHeaderModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountHeaderModule, imports: [[CommonModule, ValidationModule, ...uiModules], IconModule,
        ProductSummaryAccountLogoModule,
        InputInlineEditModule,
        ProductNumberModule,
        CreditLimitModule,
        AmountModule,
        ModalModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountHeaderModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ValidationModule, ...uiModules],
                    declarations: [AccountHeaderComponent],
                    exports: [AccountHeaderComponent, ...uiModules],
                }]
        }] });
