// @backbase/identity-self-service-journey-ang/esm2015/lib/ui/back-button.component.js
import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/icon";
import * as i2 from "@backbase/ui-ang/button";
export class BackButtonComponent {
    constructor() {
        /**
         * Output to track when back was clicked
         */
        this.backClick = new EventEmitter();
    }
}
/** @nocollapse */ BackButtonComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BackButtonComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ BackButtonComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: BackButtonComponent, selector: "bb-identity-back-button", inputs: { page: "page" }, outputs: { backClick: "backClick" }, ngImport: i0, template: "<div class=\"bb-block bb-block--lg\" data-role=\"back-button-wrapper\">\n  <button bbButton color=\"link-text\" data-role=\"back-button\" type=\"button\" buttonSize=\"sm\" (click)=\"backClick.emit()\">\n    <bb-icon-ui name=\"arrow-back\"></bb-icon-ui>\n    <span i18n=\"Back to button@@bb-identity.back-button\"> Back to {{ page }} </span>\n  </button>\n</div>\n", components: [{ type: i1.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], directives: [{ type: i2.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BackButtonComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-identity-back-button',
                    templateUrl: './back-button.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { page: [{
                type: Input
            }], backClick: [{
                type: Output
            }] } });
