// @backbase/identity-self-service-journey-ang/esm2015/lib/ui/self-service-ui.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ButtonModule } from '@backbase/ui-ang/button';
import { HeaderModule } from '@backbase/ui-ang/header';
import { ModalModule } from '@backbase/ui-ang/modal';
import { IconModule } from '@backbase/ui-ang/icon';
import { DiscardModalComponent } from './discard-modal.component';
import { BackButtonComponent } from './back-button.component';
import * as i0 from "@angular/core";
const uiModules = [HeaderModule, ButtonModule, ModalModule, IconModule];
const components = [DiscardModalComponent, BackButtonComponent];
export class SelfServiceUiModule {
}
/** @nocollapse */ SelfServiceUiModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: SelfServiceUiModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ SelfServiceUiModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: SelfServiceUiModule, declarations: [DiscardModalComponent, BackButtonComponent], imports: [CommonModule, HeaderModule, ButtonModule, ModalModule, IconModule], exports: [HeaderModule, ButtonModule, ModalModule, IconModule, DiscardModalComponent, BackButtonComponent] });
/** @nocollapse */ SelfServiceUiModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: SelfServiceUiModule, imports: [[CommonModule, ...uiModules], HeaderModule, ButtonModule, ModalModule, IconModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: SelfServiceUiModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules],
                    declarations: [...components],
                    exports: [...uiModules, ...components],
                }]
        }] });
