// @backbase/loans-collection-ui-ang/esm2015/lib/payment-schedule/amortized-payment/loan-amortized-payment-schedule-details/loan-amortized-payment-schedule-details.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AmountModule } from '@backbase/ui-ang/amount';
import { BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import { IconModule } from '@backbase/ui-ang/icon';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { LoanAmortizedPaymentScheduleDetailsComponent } from './loan-amortized-payment-schedule-details.component';
import { LoanPaymentScheduleStatusModule } from '../../loan-payment-schedule-status/loan-payment-schedule-status.module';
import { TooltipTriggerModule } from '../../../directives/tooltip-trigger/tooltip-trigger.module';
import * as i0 from "@angular/core";
export class LoanAmortizedPaymentScheduleDetailsModule {
}
/** @nocollapse */ LoanAmortizedPaymentScheduleDetailsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanAmortizedPaymentScheduleDetailsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanAmortizedPaymentScheduleDetailsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanAmortizedPaymentScheduleDetailsModule, declarations: [LoanAmortizedPaymentScheduleDetailsComponent], imports: [CommonModule,
        LoanPaymentScheduleStatusModule,
        IconModule,
        AmountModule,
        BbDatePipeModule,
        TooltipModule,
        TooltipTriggerModule], exports: [LoanAmortizedPaymentScheduleDetailsComponent] });
/** @nocollapse */ LoanAmortizedPaymentScheduleDetailsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanAmortizedPaymentScheduleDetailsModule, imports: [[
            CommonModule,
            LoanPaymentScheduleStatusModule,
            IconModule,
            AmountModule,
            BbDatePipeModule,
            TooltipModule,
            TooltipTriggerModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanAmortizedPaymentScheduleDetailsModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanAmortizedPaymentScheduleDetailsComponent],
                    imports: [
                        CommonModule,
                        LoanPaymentScheduleStatusModule,
                        IconModule,
                        AmountModule,
                        BbDatePipeModule,
                        TooltipModule,
                        TooltipTriggerModule,
                    ],
                    exports: [LoanAmortizedPaymentScheduleDetailsComponent],
                }]
        }] });
