// @backbase/payord-form-builder-ang/esm2015/lib/components/country-selector/payord-country-selector.component.js
import { Component } from '@angular/core';
import { PaymentFormFieldHooks } from '@backbase/payment-orders-ang';
import { registerFormControl } from '../../helpers/form-helpers';
import { triggerHook } from '../../helpers/hooks-helpers';
import { ActivatableFormField } from '../../models/activatable-form-field';
import { VisibilityService } from '../../services/visibility.service';
import * as i0 from "@angular/core";
import * as i1 from "../../services/visibility.service";
import * as i2 from "@backbase/payment-orders-ang";
import * as i3 from "../validation-message/validation-message.component";
import * as i4 from "@angular/common";
import * as i5 from "@angular/forms";
export class PayordCountrySelectorComponent extends ActivatableFormField {
    constructor(visibilityService) {
        super(visibilityService);
        this.visibilityService = visibilityService;
    }
    /**
     * Initialize the component.
     *
     * @returns
     */
    ngOnInit() {
        this.defaultCountry = this.options.defaultValue || '';
        this.control = registerFormControl(this.group, this.config.name, this.options.validators, this.options.asyncValidators, this.defaultCountry, this.config);
        this.toggleDependants();
        triggerHook(PaymentFormFieldHooks.onInit, this);
    }
    /**
     * Cleanup just before Angular destroys the component.
     *
     * @returns
     */
    ngOnDestroy() {
        triggerHook(PaymentFormFieldHooks.onDestroy, this);
    }
    /**
     * Selects an item
     *
     * @param country Item country
     * @returns
     */
    selectItem(country) {
        this.control.setValue(country);
        this.toggleDependants();
    }
}
/** @nocollapse */ PayordCountrySelectorComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordCountrySelectorComponent, deps: [{ token: i1.VisibilityService }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ PayordCountrySelectorComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: PayordCountrySelectorComponent, selector: "bb-payord-country-selector", usesInheritance: true, ngImport: i0, template: "<div\n  class=\"bb-dynamic-input form-group\"\n  *ngIf=\"!config?.hidden\"\n  [formGroup]=\"group\"\n  [ngClass]=\"options?.cssClasses\"\n>\n  <bb-select-country-dropdown\n    [formControlName]=\"config?.name\"\n    [countries]=\"options?.countries\"\n    [label]=\"options?.label\"\n    [helperText]=\"options?.helperText\"\n    [placeholder]=\"options?.placeholder\"\n    [searchPlaceholder]=\"options?.searchPlaceholder\"\n    [dropdownIcon]=\"options?.dropdownIcon\"\n    [selectedCountry]=\"control.value\"\n    (selectCountry)=\"selectItem($event)\"\n    [attr.data-role]=\"config?.name\"\n  ></bb-select-country-dropdown>\n\n  <bb-payord-validation-message\n    [control]=\"group?.controls[config.name]\"\n    [validationMessages]=\"options?.validationMessages\"\n    [label]=\"options?.label\"\n  ></bb-payord-validation-message>\n\n  <div\n    class=\"bb-dynamic-input__desc bb-text-support bb-subtitle\"\n    [attr.aria-hidden]=\"!options?.description\"\n    [hidden]=\"!options?.description\"\n    [textContent]=\"options?.description || ''\"\n    #descElem\n  ></div>\n</div>\n", components: [{ type: i2.ӨSelectCountryDropdownComponent, selector: "bb-select-country-dropdown", inputs: ["countries", "selectedCountry", "helperText", "placeholder", "label", "searchPlaceholder", "dropdownIcon", "isClosedPayment"], outputs: ["selectCountry"] }, { type: i3.ValidationMessageComponent, selector: "bb-payord-validation-message", inputs: ["control", "validationMessages", "label", "id", "hideGeneric"] }], directives: [{ type: i4.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i5.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { type: i5.FormGroupDirective, selector: "[formGroup]", inputs: ["formGroup"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { type: i4.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { type: i5.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { type: i5.FormControlName, selector: "[formControlName]", inputs: ["disabled", "formControlName", "ngModel"], outputs: ["ngModelChange"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordCountrySelectorComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-payord-country-selector',
                    templateUrl: 'payord-country-selector.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.VisibilityService }]; } });
