// @backbase/foundation-ang/esm2015/core/backbase-core/rendering/combined-component-factory-resolver.js
import { Injectable, ComponentFactoryResolver } from '@angular/core';
import { ItemRegistryService } from './item-registry.service';
export class CombinedComponentFactoryResolver {
    constructor(itemRegistry, rootResolver) {
        this.itemRegistry = itemRegistry;
        this.rootResolver = rootResolver;
        this.childResolvers = new Map();
    }
    registerChildResolver(classId, resolver) {
        this.childResolvers.set(classId, resolver);
    }
    resolveComponentFactory(classId) {
        const type = this.itemRegistry.getComponentValue(classId);
        if (!type) {
            throw new Error(`@backbase/foundation No component registered for ${classId}. Did you add it to the classMap?`);
        }
        const resolver = this.childResolvers.get(classId) || this.rootResolver;
        return resolver.resolveComponentFactory(type);
    }
}
CombinedComponentFactoryResolver.decorators = [
    { type: Injectable }
];
CombinedComponentFactoryResolver.ctorParameters = () => [
    { type: ItemRegistryService },
    { type: ComponentFactoryResolver }
];
