// @backbase/foundation-ang/esm2015/core/backbase-core/rendering/item-injector.service.js
import { Injector, Injectable } from '@angular/core';
import { Router, ActivatedRoute } from '@angular/router';
import { ItemModelTree, ItemModel } from '../models/item-model';
import { RouterService } from '../routing/navigation/router.service';
import { FeaturesService } from '@backbase/foundation-ang/future';
import { ɵSTORE_BRANCH, ɵSTORE_DISPATCH_CHANNEL, StoreFactory, ɵStoreBroker } from '@backbase/foundation-ang/store';
import { PipelineRegistry } from '../communication/pipes/pipeline-registry';
import { InputBrokerService } from '../communication/item-io/input-broker.service';
import { OutputBrokerService } from '../communication/item-io/output-broker.service';
import { PipelineService } from '../communication/pipes/pipeline.service';
import { uniq } from '../models/array';
import { TemplateRegistry } from '../customizable/template-registry';
import { AbstractRouterConfigService } from '../routing/config/abstract-router-config.service';
export class ItemInjectorService {
    constructor(inputBroker, outputBroker) {
        this.inputBroker = inputBroker;
        this.outputBroker = outputBroker;
    }
    createInjector(parent, item) {
        const widgetProvider = {
            provide: ItemModel,
            useValue: item.value,
        };
        const containerProvider = {
            provide: ItemModelTree,
            useValue: item,
        };
        const routerServiceProvider = {
            provide: RouterService,
            deps: [Router, ActivatedRoute, ItemModel, AbstractRouterConfigService],
        };
        /**
         * @todo: consider putting *all* configured handlers on the
         * injector, instead of only handlers used based on the model.
         *
         * This would make it possible to create higher-order handlers
         */
        const communicationPipelineProviders = uniq([
            ...this.inputBroker.getItemProviders(item.value),
            ...this.outputBroker.getItemProviders(item.value),
        ]);
        const pipelineService = {
            provide: PipelineService,
            deps: [PipelineRegistry, Injector],
        };
        return Injector.create({
            providers: [
                widgetProvider,
                containerProvider,
                routerServiceProvider,
                ...communicationPipelineProviders,
                pipelineService,
                {
                    provide: ɵSTORE_BRANCH,
                    useValue: item.value.name,
                },
                {
                    provide: ɵSTORE_DISPATCH_CHANNEL,
                    useValue: item.value.name,
                },
                {
                    provide: StoreFactory,
                    useClass: StoreFactory,
                    deps: [ɵStoreBroker, ɵSTORE_BRANCH, ɵSTORE_DISPATCH_CHANNEL, FeaturesService],
                },
                {
                    provide: TemplateRegistry,
                    useClass: TemplateRegistry,
                    deps: [],
                },
            ],
            parent,
        });
    }
}
ItemInjectorService.decorators = [
    { type: Injectable }
];
ItemInjectorService.ctorParameters = () => [
    { type: InputBrokerService },
    { type: OutputBrokerService }
];
