// @backbase/foundation-ang/esm2015/core/backbase-core/rendering/item-lifecycle.service.js
import { Injectable } from '@angular/core';
import { Subject } from 'rxjs';
import { filter } from 'rxjs/operators';
import * as i0 from "@angular/core";
export var LifecycleEventKind;
(function (LifecycleEventKind) {
    LifecycleEventKind["Create"] = "create";
    LifecycleEventKind["Remove"] = "remove";
})(LifecycleEventKind || (LifecycleEventKind = {}));
export class ItemLifecycleService {
    constructor() {
        this.lifecycleEvents = new Subject();
    }
    onCreate(itemModel, component) {
        this.lifecycleEvents.next({
            kind: LifecycleEventKind.Create,
            itemModel,
            component,
        });
    }
    onRemove(itemModel, component) {
        this.lifecycleEvents.next({
            kind: LifecycleEventKind.Remove,
            itemModel,
            component,
        });
    }
    fromItem(name) {
        return this.lifecycleEvents.pipe(filter((lcEvent) => name === lcEvent.itemModel.value.name));
    }
    whenCreate(name) {
        const nameFilter = name ? (currentName) => currentName === name : () => true;
        return this.lifecycleEvents.pipe(filter((lcEvent) => LifecycleEventKind.Create === lcEvent.kind && nameFilter(lcEvent.itemModel.value.name)));
    }
    whenRemove(name) {
        const nameFilter = name ? (currentName) => currentName === name : () => true;
        return this.lifecycleEvents.pipe(filter((lcEvent) => LifecycleEventKind.Remove === lcEvent.kind && nameFilter(lcEvent.itemModel.value.name)));
    }
}
ItemLifecycleService.ɵprov = i0.ɵɵdefineInjectable({ factory: function ItemLifecycleService_Factory() { return new ItemLifecycleService(); }, token: ItemLifecycleService, providedIn: "root" });
ItemLifecycleService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root',
            },] }
];
