// @backbase/foundation-ang/esm2015/core/backbase-core/rendering/item-registry.service.js
import { Inject, Injectable, InjectionToken, Optional } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import { filter, map, take } from 'rxjs/operators';
import { LAZY_MAP } from './lazy-config';
/**
 * ComponentRegistry configuration injection token
 */
export const CONFIG_COMPONENT_CLASS_MAP = new InjectionToken('@backbase/foundation Components Registry');
/**
 * ItemRegistryService stores a hashMap
 * of the components included into the app and
 * avaliable for usage. It's meant to be an extension
 * point which is used by a developer.
 */
export class ItemRegistryService {
    /**
     * ItemRegistryService constructor which allows to define components
     * included into application
     * @param ComponentRegistry a hashMap of the component classes to be added
     */
    constructor(classMaps, lazyConfig) {
        this.itemRegistry = new BehaviorSubject(new Map(classMaps
            ? classMaps.reduce((acc, classMap) => [...acc, ...Object.entries(classMap)], [])
            : undefined));
        this.lazyModuleConfig = (lazyConfig || []).filter(isLazyModuleConfig);
    }
    entries() {
        return Array.from(this.itemRegistry.value.entries());
    }
    add(classId, componentClass) {
        this.itemRegistry.next(new Map([...this.entries(), [classId, componentClass]]));
    }
    has(classId) {
        return this.itemRegistry.value.has(classId);
    }
    /**
     * This method allows to retrieve Component by it's classId
     * @param classId classId key to get a component class
     * @returns componentType to be used by a ComponentFactoryResolver
     */
    getComponentValue(classId) {
        return this.itemRegistry.value.get(classId);
    }
    getComponent(classId) {
        return this.itemRegistry
            .pipe(map(registry => registry.get(classId)), filter((componentClass) => componentClass !== undefined), take(1))
            .toPromise();
    }
    getModuleLoader(classId) {
        const lazyModuleConfigItem = this.lazyModuleConfig.find(config => config.module === classId);
        return lazyModuleConfigItem && lazyModuleConfigItem.loadChildren;
    }
}
ItemRegistryService.decorators = [
    { type: Injectable }
];
ItemRegistryService.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Inject, args: [CONFIG_COMPONENT_CLASS_MAP,] }, { type: Optional }] },
    { type: Array, decorators: [{ type: Inject, args: [LAZY_MAP,] }, { type: Optional }] }
];
function isLazyModuleConfig(config) {
    return !!config.module;
}
