// @backbase/quick-transfer-journey-ang/esm2015/lib/components/quick-transfer-accounts/quick-transfer-accounts.component.js
import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/product-item-basic-account";
import * as i2 from "@angular/common";
import * as i3 from "@backbase/ui-ang/button";
export class QuickTransferAccountsComponent {
    constructor() {
        /**
         * Output to inform which account or contact is selected
         */
        this.select = new EventEmitter();
    }
    /**
     * Emit event to inform parent with the picked account id
     *
     * @param id
     */
    pick(id) {
        this.select.emit(id);
    }
}
/** @nocollapse */ QuickTransferAccountsComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: QuickTransferAccountsComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ QuickTransferAccountsComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: QuickTransferAccountsComponent, selector: "bb-quick-transfer-accounts-ui", inputs: { arrangements: "arrangements", selected: "selected", selectedCounterPartyAccount: "selectedCounterPartyAccount", config: "config" }, outputs: { select: "select" }, ngImport: i0, template: "<button\n  bbButton\n  color=\"unstyled\"\n  *ngFor=\"let arrangement of arrangements\"\n  class=\"bb-quick-transfer__product-item bb-stack\"\n  [class.bb-quick-transfer--selected-bg]=\"arrangement.uuid === selected\"\n  (click)=\"pick(arrangement.uuid)\"\n  [disabled]=\"arrangement.uuid === selectedCounterPartyAccount\"\n>\n  <bb-product-item-basic-account-ui\n    [highlight]=\"config?.uiComponentProps.productItemHighlight\"\n    [title]=\"arrangement.name\"\n    [amount]=\"arrangement?.balance\"\n    [productNumber]=\"arrangement.number\"\n    [currency]=\"arrangement?.currency\"\n    [productNumberFormat]=\"config?.uiComponentProps.accountNumberFormat\"\n    [showCurrencySymbol]=\"config?.uiComponentProps.mapCurrency\"\n    [active]=\"false\"\n    [class.bb-quick-transfer--disabled]=\"arrangement.uuid === selectedCounterPartyAccount\"\n  ></bb-product-item-basic-account-ui>\n</button>\n", components: [{ type: i1.ProductItemBasicAccountComponent, selector: "bb-product-item-basic-account-ui", inputs: ["title", "productNumber", "productNumberFormat", "currency", "showCurrencySymbol", "amount", "highlight", "productDetailsInfo", "active", "favorite", "favoriteIconColor", "selected", "type", "status", "balanceLabel", "bankBranchCode"] }], directives: [{ type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { type: i3.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: QuickTransferAccountsComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-quick-transfer-accounts-ui',
                    templateUrl: './quick-transfer-accounts.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { arrangements: [{
                type: Input
            }], selected: [{
                type: Input
            }], selectedCounterPartyAccount: [{
                type: Input
            }], config: [{
                type: Input
            }], select: [{
                type: Output
            }] } });
