// @backbase/loans-collection-ui-ang/esm2015/lib/loans-initial-empty/loans-initial-empty.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ButtonModule } from '@backbase/ui-ang/button';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { LoansInitialEmptyComponent } from './loans-initial-empty.component';
import * as i0 from "@angular/core";
export class LoansInitialEmptyModule {
}
/** @nocollapse */ LoansInitialEmptyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoansInitialEmptyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoansInitialEmptyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoansInitialEmptyModule, declarations: [LoansInitialEmptyComponent], imports: [CommonModule, EmptyStateModule, ButtonModule], exports: [LoansInitialEmptyComponent] });
/** @nocollapse */ LoansInitialEmptyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoansInitialEmptyModule, imports: [[CommonModule, EmptyStateModule, ButtonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoansInitialEmptyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoansInitialEmptyComponent],
                    imports: [CommonModule, EmptyStateModule, ButtonModule],
                    exports: [LoansInitialEmptyComponent],
                }]
        }] });
