// @backbase/user-manage-profile-widget-ang/esm2015/src/user-profile-forms/user-profile-form-inputs/postal-address-inputs/user-profile-form-postal-address-inputs.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { DropdownSingleSelectModule } from '@backbase/ui-ang/dropdown-single-select';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { UserPostalAddressTypeErrorsCustomizableDirective, UserPostalAddressTypeInputComponent, UserPostalAddressTypeInputCustomizableDirective, } from './user-postal-address-type-input.component';
import { UserPostalBuildingNumberInputComponent, UserPostalBuildingNumberInputCustomizableDirective, UserPostalBuildingNumberInputErrorsCustomizableDirective, } from './user-postal-building-number-input.component';
import { UserPostalCountrySubDivisionInputComponent, UserPostalCountrySubDivisionInputCustomizableDirective, UserPostalCountrySubDivisionInputErrorsCustomizableDirective, } from './user-postal-country-sub-division-input.component';
import { UserPostalPostalCodeInputComponent, UserPostalPostalCodeInputCustomizableDirective, UserPostalPostalCodeInputErrorsCustomizableDirective, } from './user-postal-postal-code-input.component';
import { UserPostalStreetNameInputComponent, UserPostalStreetNameInputCustomizableDirective, UserPostalStreetNameInputErrorsCustomizableDirective, } from './user-postal-street-name-input.component';
import { UserPostalTownNameInputComponent, UserPostalTownNameInputCustomizableDirective, UserPostalTownNameInputErrorsCustomizableDirective, } from './user-postal-town-name-input.component';
import * as i0 from "@angular/core";
const uiModules = [DropdownSingleSelectModule, InputTextModule, InputValidationMessageModule];
export class UserProfileFormPostalAddressInputsModule {
}
UserProfileFormPostalAddressInputsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormPostalAddressInputsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
UserProfileFormPostalAddressInputsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormPostalAddressInputsModule, declarations: [UserPostalAddressTypeInputComponent,
        UserPostalAddressTypeInputCustomizableDirective,
        UserPostalAddressTypeErrorsCustomizableDirective,
        UserPostalBuildingNumberInputComponent,
        UserPostalBuildingNumberInputCustomizableDirective,
        UserPostalBuildingNumberInputErrorsCustomizableDirective,
        UserPostalCountrySubDivisionInputComponent,
        UserPostalCountrySubDivisionInputCustomizableDirective,
        UserPostalCountrySubDivisionInputErrorsCustomizableDirective,
        UserPostalPostalCodeInputComponent,
        UserPostalPostalCodeInputCustomizableDirective,
        UserPostalPostalCodeInputErrorsCustomizableDirective,
        UserPostalStreetNameInputComponent,
        UserPostalStreetNameInputCustomizableDirective,
        UserPostalStreetNameInputErrorsCustomizableDirective,
        UserPostalTownNameInputComponent,
        UserPostalTownNameInputCustomizableDirective,
        UserPostalTownNameInputErrorsCustomizableDirective], imports: [CommonModule, ReactiveFormsModule, DropdownSingleSelectModule, InputTextModule, InputValidationMessageModule], exports: [UserPostalAddressTypeInputComponent,
        UserPostalAddressTypeInputCustomizableDirective,
        UserPostalAddressTypeErrorsCustomizableDirective,
        UserPostalBuildingNumberInputComponent,
        UserPostalBuildingNumberInputCustomizableDirective,
        UserPostalBuildingNumberInputErrorsCustomizableDirective,
        UserPostalCountrySubDivisionInputComponent,
        UserPostalCountrySubDivisionInputCustomizableDirective,
        UserPostalCountrySubDivisionInputErrorsCustomizableDirective,
        UserPostalPostalCodeInputComponent,
        UserPostalPostalCodeInputCustomizableDirective,
        UserPostalPostalCodeInputErrorsCustomizableDirective,
        UserPostalStreetNameInputComponent,
        UserPostalStreetNameInputCustomizableDirective,
        UserPostalStreetNameInputErrorsCustomizableDirective,
        UserPostalTownNameInputComponent,
        UserPostalTownNameInputCustomizableDirective,
        UserPostalTownNameInputErrorsCustomizableDirective, DropdownSingleSelectModule, InputTextModule, InputValidationMessageModule] });
UserProfileFormPostalAddressInputsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormPostalAddressInputsModule, imports: [[CommonModule, ReactiveFormsModule, ...uiModules], DropdownSingleSelectModule, InputTextModule, InputValidationMessageModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormPostalAddressInputsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ReactiveFormsModule, ...uiModules],
                    declarations: [
                        UserPostalAddressTypeInputComponent,
                        UserPostalAddressTypeInputCustomizableDirective,
                        UserPostalAddressTypeErrorsCustomizableDirective,
                        UserPostalBuildingNumberInputComponent,
                        UserPostalBuildingNumberInputCustomizableDirective,
                        UserPostalBuildingNumberInputErrorsCustomizableDirective,
                        UserPostalCountrySubDivisionInputComponent,
                        UserPostalCountrySubDivisionInputCustomizableDirective,
                        UserPostalCountrySubDivisionInputErrorsCustomizableDirective,
                        UserPostalPostalCodeInputComponent,
                        UserPostalPostalCodeInputCustomizableDirective,
                        UserPostalPostalCodeInputErrorsCustomizableDirective,
                        UserPostalStreetNameInputComponent,
                        UserPostalStreetNameInputCustomizableDirective,
                        UserPostalStreetNameInputErrorsCustomizableDirective,
                        UserPostalTownNameInputComponent,
                        UserPostalTownNameInputCustomizableDirective,
                        UserPostalTownNameInputErrorsCustomizableDirective,
                    ],
                    exports: [
                        UserPostalAddressTypeInputComponent,
                        UserPostalAddressTypeInputCustomizableDirective,
                        UserPostalAddressTypeErrorsCustomizableDirective,
                        UserPostalBuildingNumberInputComponent,
                        UserPostalBuildingNumberInputCustomizableDirective,
                        UserPostalBuildingNumberInputErrorsCustomizableDirective,
                        UserPostalCountrySubDivisionInputComponent,
                        UserPostalCountrySubDivisionInputCustomizableDirective,
                        UserPostalCountrySubDivisionInputErrorsCustomizableDirective,
                        UserPostalPostalCodeInputComponent,
                        UserPostalPostalCodeInputCustomizableDirective,
                        UserPostalPostalCodeInputErrorsCustomizableDirective,
                        UserPostalStreetNameInputComponent,
                        UserPostalStreetNameInputCustomizableDirective,
                        UserPostalStreetNameInputErrorsCustomizableDirective,
                        UserPostalTownNameInputComponent,
                        UserPostalTownNameInputCustomizableDirective,
                        UserPostalTownNameInputErrorsCustomizableDirective,
                        ...uiModules,
                    ],
                }]
        }] });
