// @backbase/ui-ang/esm2015/select-list/select-list.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { NgSelectModule } from '@ng-select/ng-select';
import { SelectListComponent } from './select-list.component';
import { BaseClassesModule } from '@backbase/ui-ang/base-classes';
import * as i0 from "@angular/core";
export class SelectListModule {
}
SelectListModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SelectListModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
SelectListModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SelectListModule, declarations: [SelectListComponent], imports: [CommonModule, NgSelectModule, FormsModule, ReactiveFormsModule, BaseClassesModule], exports: [SelectListComponent] });
SelectListModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SelectListModule, imports: [[CommonModule, NgSelectModule, FormsModule, ReactiveFormsModule, BaseClassesModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SelectListModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, NgSelectModule, FormsModule, ReactiveFormsModule, BaseClassesModule],
                    declarations: [SelectListComponent],
                    exports: [SelectListComponent],
                }]
        }] });
