// @backbase/loans-collection-ui-ang/esm2015/lib/loan-list-item/loan-item-menu/loan-item-menu.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { DropdownMenuModule } from '@backbase/ui-ang/dropdown-menu';
import { LoanItemMenuComponent } from './loan-item-menu.component';
import * as i0 from "@angular/core";
export class LoanItemMenuModule {
}
/** @nocollapse */ LoanItemMenuModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemMenuModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanItemMenuModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemMenuModule, declarations: [LoanItemMenuComponent], imports: [CommonModule, DropdownMenuModule], exports: [LoanItemMenuComponent] });
/** @nocollapse */ LoanItemMenuModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemMenuModule, imports: [[CommonModule, DropdownMenuModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemMenuModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanItemMenuComponent],
                    imports: [CommonModule, DropdownMenuModule],
                    exports: [LoanItemMenuComponent],
                }]
        }] });
