// @backbase/loans-collection-ui-ang/esm2015/lib/loan-list-item/loan-item-menu/loan-item-menu.component.js
import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/dropdown-menu";
import * as i2 from "@angular/common";
export class LoanItemMenuComponent {
    constructor() {
        /**
         * placement of the dropdown
         */
        this.position = 'bottom-right';
        /**
         * Emit on menu item select
         */
        this.itemSelect = new EventEmitter();
    }
}
/** @nocollapse */ LoanItemMenuComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemMenuComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ LoanItemMenuComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: LoanItemMenuComponent, selector: "bb-loan-item-menu", inputs: { items: "items", dropDownButtonLabel: "dropDownButtonLabel", position: "position" }, outputs: { itemSelect: "itemSelect" }, ngImport: i0, template: "<bb-dropdown-menu-ui\n  icon=\"ellipsis-h\"\n  iconSize=\"md\"\n  buttonSize=\"md\"\n  btnColor=\"link-dark\"\n  [btnCircle]=\"true\"\n  [autoClose]=\"true\"\n  [dropDownButtonLabel]=\"dropDownButtonLabel\"\n  [position]=\"position\"\n>\n  <ng-template bbDropdownMenuItem>\n    <button\n      *ngFor=\"let item of items\"\n      role=\"menuitem\"\n      class=\"dropdown-item bb-loans-list-menu-item\"\n      data-role=\"bb-loans-list-item-menu__item\"\n      (click)=\"itemSelect.emit(item.key)\"\n    >\n      {{ item.label }}\n    </button>\n  </ng-template>\n</bb-dropdown-menu-ui>\n", components: [{ type: i1.DropdownMenuComponent, selector: "bb-dropdown-menu-ui", inputs: ["label", "disabled", "options", "icon", "iconSize", "iconColor", "btnColor", "btnCircle", "optionLabelKey", "fullWidth", "autoClose", "container", "position", "ariaLabel", "dropDownMenuRole", "dropDownMenuId", "buttonSize", "dropDownButtonLabel", "visibleElementDependingOnViewport", "role"], outputs: ["select"] }], directives: [{ type: i1.DropdownMenuItemDirective, selector: "ng-template[bbDropdownMenuItem]" }, { type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemMenuComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-loan-item-menu',
                    templateUrl: './loan-item-menu.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { items: [{
                type: Input
            }], dropDownButtonLabel: [{
                type: Input
            }], position: [{
                type: Input
            }], itemSelect: [{
                type: Output
            }] } });
