// @backbase/ui-ang/esm2015/_example/value-diff/showcase/value-diff-showcase.component.js
import { Component } from '@angular/core';
import { ValueDiffPosition } from '@backbase/ui-ang/value-diff';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/value-diff";
import * as i2 from "@backbase/ui-ang/button";
export class ValueDiffShowcaseComponent {
    constructor() {
        this.oldValue = 'Old Value';
        this.newValue = 'New Value';
        this.oldValueAfter = 'Old Value';
        this.newValueAfter = 'New Value';
        this.positionAfter = ValueDiffPosition.AFTER;
    }
    reset() {
        this.oldValue = 'Equal Value';
        this.newValue = 'Equal Value';
    }
    changeData() {
        this.oldValue = ValueDiffShowcaseComponent.makeId();
        this.newValue = ValueDiffShowcaseComponent.makeId();
    }
    static makeId() {
        const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
        const charactersLength = characters.length;
        let result = '';
        for (let i = 0; i < 5; i++) {
            result += characters.charAt(Math.floor(Math.random() * charactersLength));
        }
        return result;
    }
}
ValueDiffShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ValueDiffShowcaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ValueDiffShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ValueDiffShowcaseComponent, selector: "bb-value-diff-ui-showcase", ngImport: i0, template: "<div class=\"row\">\n  <div class=\"col-md-6\">\n    <h3>Default case</h3>\n    <div class=\"row\">\n      <div class=\"col-6\">\n        <div [bbValueDiff]=\"newValue\" [sr-only-old]=\"'Old value: '\" [sr-only-new]=\"'New value: '\">\n          {{ oldValue }}\n        </div>\n        <br />\n        <div class=\"bb-button-bar bb-button-bar--reverse\">\n          <button class=\"bb-button-bar__button\" (click)=\"changeData()\" bbButton buttonSize=\"sm\">Change data</button>\n          <button class=\"bb-button-bar__button\" (click)=\"reset()\" bbButton buttonSize=\"sm\">Make equal</button>\n        </div>\n      </div>\n    </div>\n  </div>\n\n  <div class=\"col-md-6\">\n    <h3>Show new value after old</h3>\n    <div class=\"row\">\n      <div class=\"col\">\n        <div [bbValueDiff]=\"newValueAfter\" [position]=\"positionAfter\">\n          {{ oldValueAfter }}\n        </div>\n      </div>\n    </div>\n  </div>\n</div>\n<hr />\n\n<div class=\"row\">\n  <div class=\"col-md-6\">\n    <h3>With arrow right</h3>\n    <div class=\"bb-stack\">\n      <span [bbValueDiff]=\"newValue\" [position]=\"positionAfter\" [showArrow]=\"true\">{{ oldValue }}</span>\n    </div>\n  </div>\n\n  <div class=\"col-md-6\">\n    <h3>With arrow left</h3>\n    <div class=\"bb-stack\">\n      <span [bbValueDiff]=\"newValueAfter\" [showArrow]=\"true\">{{ oldValueAfter }}</span>\n    </div>\n  </div>\n</div>\n", directives: [{ type: i1.ValueDiffDirective, selector: "[bbValueDiff]", inputs: ["bbValueDiff", "position", "sr-only-old", "sr-only-new", "showArrow"] }, { type: i2.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ValueDiffShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-value-diff-ui-showcase',
                    templateUrl: './value-diff-showcase.component.html',
                }]
        }] });
