// @backbase/ui-ang/esm2015/_example/value-diff/showcase/value-diff-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ValueDiffShowcaseComponent } from './value-diff-showcase.component';
import { ValueDiffDirectiveModule } from '@backbase/ui-ang/value-diff';
import { ButtonModule } from '@backbase/ui-ang/button';
import * as i0 from "@angular/core";
export class ValueDiffShowcaseModule {
}
ValueDiffShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ValueDiffShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ValueDiffShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ValueDiffShowcaseModule, declarations: [ValueDiffShowcaseComponent], imports: [CommonModule, ValueDiffDirectiveModule, ButtonModule], exports: [ValueDiffShowcaseComponent] });
ValueDiffShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ValueDiffShowcaseModule, imports: [[CommonModule, ValueDiffDirectiveModule, ButtonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ValueDiffShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ValueDiffShowcaseComponent],
                    exports: [ValueDiffShowcaseComponent],
                    imports: [CommonModule, ValueDiffDirectiveModule, ButtonModule],
                }]
        }] });
