// @backbase/ui-ang/esm2015/_example/input-inline-edit/examples/input-inline-edit.example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/forms";
import * as i2 from "@backbase/ui-ang/input-inline-edit";
export class InputInlineEditExampleComponent {
    constructor(formBuilder) {
        this.formBuilder = formBuilder;
        this.form = this.formBuilder.group({
            description: ['Monthly Rent'],
        });
    }
    onAcceptDescription(description) {
        this.form.patchValue({
            description,
        });
    }
}
InputInlineEditExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputInlineEditExampleComponent, deps: [{ token: i1.FormBuilder }], target: i0.ɵɵFactoryTarget.Component });
InputInlineEditExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: InputInlineEditExampleComponent, selector: "bb-input-inline-edit-example-ui", ngImport: i0, template: "<form [formGroup]=\"form\">\n  <bb-input-inline-edit-ui\n    [inputText]=\"form.get('description')?.value\"\n    [maxLength]=\"64\"\n    [minLength]=\"2\"\n    [required]=\"true\"\n    (accept)=\"onAcceptDescription($event)\"\n  >\n  </bb-input-inline-edit-ui>\n</form>\n", components: [{ type: i2.InputInlineEditComponent, selector: "bb-input-inline-edit-ui", inputs: ["state", "canEdit", "hasLoadingState", "inputInlineTemplate", "inputText", "maxLength", "minLength", "maxValue", "minValue", "autofocus", "ariaLabelAccept", "ariaLabelCancel", "ariaLabelEdit", "tooltipAccept", "tooltipCancel", "tooltipEdit", "autocomplete", "pattern", "hint"], outputs: ["stateChange", "cancel", "accept"] }], directives: [{ type: i1.ɵNgNoValidate, selector: "form:not([ngNoForm]):not([ngNativeValidate])" }, { type: i1.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { type: i1.FormGroupDirective, selector: "[formGroup]", inputs: ["formGroup"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputInlineEditExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-input-inline-edit-example-ui',
                    templateUrl: './input-inline-edit.example.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.FormBuilder }]; } });
