// @backbase/loans-collection-ui-ang/esm2015/lib/payment-schedule/loan-payment-schedule-status/loan-payment-schedule-status.component.js
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { PresentationAmortizationPaymentStatus } from '@backbase/data-ang/loans';
import { loanPaymentScheduleStatusColorMap, } from './loan-payment-schedule-status.model';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/badge";
import * as i2 from "@angular/common";
export class LoanPaymentScheduleStatusComponent {
    constructor() {
        /**
         * Whether the status should be truncated
         *
         * @default: false
         */
        this.truncate = false;
        /**
         * Map of colors for each status
         */
        this.statusColorMap = loanPaymentScheduleStatusColorMap;
    }
}
/** @nocollapse */ LoanPaymentScheduleStatusComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanPaymentScheduleStatusComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ LoanPaymentScheduleStatusComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: LoanPaymentScheduleStatusComponent, selector: "bb-loan-payment-schedule-status", inputs: { status: "status", truncate: "truncate", statusColorMap: "statusColorMap" }, ngImport: i0, template: "<span class=\"sr-only\" i18n=\"Loan payments schedule status a11y prefix@@loans.payment-schedule.status.a11y.prefix\">\n  Payment status:\n</span>\n<span class=\"sr-only\" i18n=\"Loan payments schedule full status@@loans.payment-schedule.full-status\" #textLabelFull\n  >{status, select, Scheduled {Scheduled} Overdue {Overdue}}</span\n>\n<span class=\"d-none\" i18n=\"Loan payments schedule short status@@loans.payment-schedule.short-status\" #textLabelShort\n  >{status, select, Scheduled {Sch} Overdue {Over}}</span\n>\n<bb-badge-ui\n  aria-hidden=\"true\"\n  [color]=\"statusColorMap[status]\"\n  [abbr]=\"(truncate ? textLabelShort.textContent : textLabelFull.textContent)! | titlecase\"\n  [text]=\"textLabelFull.textContent! | titlecase\"\n  data-role=\"bb-payment-schedule-status__badge\"\n></bb-badge-ui>\n<span class=\"sr-only\">,</span>\n", components: [{ type: i1.BadgeComponent, selector: "bb-badge-ui", inputs: ["color", "text", "abbr", "circle", "wrap", "ariaLabel"] }], pipes: { "titlecase": i2.TitleCasePipe }, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanPaymentScheduleStatusComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-loan-payment-schedule-status',
                    templateUrl: './loan-payment-schedule-status.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { status: [{
                type: Input
            }], truncate: [{
                type: Input
            }], statusColorMap: [{
                type: Input
            }] } });
