// @backbase/loans-collection-ui-ang/esm2015/lib/payment-schedule/loan-payment-schedule-status/loan-payment-schedule-status.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { LoanPaymentScheduleStatusComponent } from './loan-payment-schedule-status.component';
import { BadgeModule } from '@backbase/ui-ang/badge';
import * as i0 from "@angular/core";
export class LoanPaymentScheduleStatusModule {
}
/** @nocollapse */ LoanPaymentScheduleStatusModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanPaymentScheduleStatusModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanPaymentScheduleStatusModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanPaymentScheduleStatusModule, declarations: [LoanPaymentScheduleStatusComponent], imports: [CommonModule, BadgeModule], exports: [LoanPaymentScheduleStatusComponent] });
/** @nocollapse */ LoanPaymentScheduleStatusModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanPaymentScheduleStatusModule, imports: [[CommonModule, BadgeModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanPaymentScheduleStatusModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanPaymentScheduleStatusComponent],
                    imports: [CommonModule, BadgeModule],
                    exports: [LoanPaymentScheduleStatusComponent],
                }]
        }] });
