// @backbase/ui-ang/esm2015/_example/alert/showcase/alert-showcase.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/alert";
import * as i2 from "@backbase/ui-ang/button";
import * as i3 from "@angular/common";
let AlertShowcaseComponent = class AlertShowcaseComponent {
    constructor() {
        this.colorVariants = [
            {
                title: 'Information! Use this alert to introduce information only',
                modifier: 'info',
            },
            {
                title: 'Warning! Use this alert to show warn only',
                modifier: 'warning',
            },
            {
                title: 'Success! Use this alert to show success only',
                modifier: 'success',
            },
            {
                title: 'Error! Use this alert to show error only',
                modifier: 'error',
            },
        ];
        this.dismissibleAlerts = [
            {
                dismissed: false,
            },
            {
                dismissed: false,
            },
        ];
    }
};
AlertShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AlertShowcaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
AlertShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AlertShowcaseComponent, selector: "bb-alert-ui-showcase", ngImport: i0, template: "<div class=\"mb-4\">\n  <h4>Default</h4>\n  <bb-alert-ui title=\"Default\" data-role=\"alert-default\"></bb-alert-ui>\n</div>\n\n<div class=\"mb-4\">\n  <h4>Color variants</h4>\n  <bb-alert-ui\n    *ngFor=\"let alert of colorVariants\"\n    [title]=\"alert.title\"\n    [modifier]=\"alert.modifier\"\n    [attr.data-role]=\"'alert-' + alert.modifier\"\n  ></bb-alert-ui>\n</div>\n\n<div class=\"mb-4\">\n  <h4>Multiple lines</h4>\n  <bb-alert-ui\n    title=\"Default\"\n    modifier=\"info\"\n    message=\"Body of a sticky alert.  This alert type is placed above the site header.\"\n    data-role=\"alert-multiple-lines\"\n  ></bb-alert-ui>\n</div>\n\n<div class=\"mb-4\">\n  <h4>Dismissible</h4>\n  <bb-alert-ui\n    *ngIf=\"!dismissibleAlerts[0].dismissed\"\n    title=\"Dismissible alert\"\n    modifier=\"info\"\n    [dismissible]=\"true\"\n    (close)=\"dismissibleAlerts[0].dismissed = true\"\n    data-role=\"alert-dismissible-single-line\"\n  ></bb-alert-ui>\n  <bb-button-ui\n    *ngIf=\"dismissibleAlerts[0].dismissed\"\n    color=\"secondary\"\n    (click)=\"dismissibleAlerts[0].dismissed = false\"\n    data-role=\"alert-dismissible-single-line-restore\"\n    >Restore the alert</bb-button-ui\n  >\n\n  <bb-alert-ui\n    *ngIf=\"!dismissibleAlerts[1].dismissed\"\n    title=\"Dismissible alert\"\n    modifier=\"info\"\n    message=\"With text\"\n    [dismissible]=\"true\"\n    (close)=\"dismissibleAlerts[1].dismissed = true\"\n    data-role=\"alert-dismissible-multi-line\"\n  ></bb-alert-ui>\n  <bb-button-ui\n    *ngIf=\"dismissibleAlerts[1].dismissed\"\n    color=\"secondary\"\n    (click)=\"dismissibleAlerts[1].dismissed = false\"\n    data-role=\"alert-dismissible-multi-line-restore\"\n    >Restore the alert</bb-button-ui\n  >\n</div>\n\n<div class=\"mb-4\">\n  <h4>Custom message content template</h4>\n  <bb-alert-ui\n    *ngFor=\"let alert of colorVariants\"\n    [title]=\"alert.title\"\n    [modifier]=\"alert.modifier\"\n    [attr.data-role]=\"'alert-' + alert.modifier\"\n  >\n    Lorem ipsum dolor sit amet consectetur <a href=\"javascript:void(0);\" class=\"alert-link\">click here</a> adipisicing\n    elit. Dicta, corporis.\n  </bb-alert-ui>\n</div>\n", components: [{ type: i1.AlertComponent, selector: "bb-alert-ui", inputs: ["modifier", "dismissible", "title", "message"], outputs: ["close"] }, { type: i2.ButtonComponent, selector: "bb-button-ui", inputs: ["autofocus", "type", "disabled", "buttonSize", "color", "block"] }], directives: [{ type: i3.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
AlertShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
], AlertShowcaseComponent);
export { AlertShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AlertShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-alert-ui-showcase',
                    templateUrl: 'alert-showcase.component.html',
                }]
        }] });
