// @backbase/ui-ang/esm2015/_example/alert/showcase/alert-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AlertShowcaseComponent } from './alert-showcase.component';
import { AlertModule } from '@backbase/ui-ang/alert';
import { ButtonModule } from '@backbase/ui-ang/button';
import * as i0 from "@angular/core";
export class AlertShowcaseModule {
}
AlertShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AlertShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AlertShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AlertShowcaseModule, declarations: [AlertShowcaseComponent], imports: [CommonModule, AlertModule, ButtonModule] });
AlertShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AlertShowcaseModule, imports: [[CommonModule, AlertModule, ButtonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AlertShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, AlertModule, ButtonModule],
                    declarations: [AlertShowcaseComponent],
                    entryComponents: [AlertShowcaseComponent],
                }]
        }] });
