// @backbase/account-statement-store-ang/esm2015/src/account-statements/effects.js
import { __rest } from "tslib";
import { Injectable } from '@angular/core';
import { getTotalCountFromHeaders } from '@backbase/account-statement-common-ang';
import { of } from 'rxjs';
import { catchError, map, switchMap, withLatestFrom } from 'rxjs/operators';
import { loadAccountStatementsBusinessToRequest } from '../helpers/mappers';
import { loadAccountStatementsBusiness, loadAccountStatementsBusinessSuccess, loadAccountStatementsFailure, } from './actions';
import { accountStatementsAccountIds, accountStatementsFilterParams, accountStatementsPaginationParams, accountStatementsSortingParams, accountStatementsWidgetPreferences, } from './selectors';
import { createEffect, ofType } from '@ngrx/effects';
import * as i0 from "@angular/core";
import * as i1 from "@ngrx/effects";
import * as i2 from "@backbase/account-statement-common-ang";
import * as i3 from "@ngrx/store";
export class AccountStatementEffects {
    constructor(actions$, dataService, store) {
        this.actions$ = actions$;
        this.dataService = dataService;
        this.store = store;
        this.accountStatementsBusiness$ = createEffect(() => this.actions$.pipe(ofType(loadAccountStatementsBusiness), withLatestFrom(this.store.select(accountStatementsAccountIds), this.store.select(accountStatementsFilterParams), this.store.select(accountStatementsSortingParams), this.store.select(accountStatementsPaginationParams), this.store.select(accountStatementsWidgetPreferences)), switchMap(([action, accountIds, filterParams, sortingParams, paginationParams, widgetPreferences]) => {
            const { append } = action, actionRequestParams = __rest(action, ["append"]);
            return this.dataService
                .getAccountStatementsBusiness(loadAccountStatementsBusinessToRequest(Object.assign({ accountIds,
                filterParams,
                sortingParams,
                paginationParams }, actionRequestParams), widgetPreferences))
                .pipe(map(response => loadAccountStatementsBusinessSuccess(Object.assign({ accountStatements: response.body, totalCount: getTotalCountFromHeaders(response.headers) }, action))), catchError(error => of(loadAccountStatementsFailure({ error }))));
        })));
    }
}
AccountStatementEffects.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementEffects, deps: [{ token: i1.Actions }, { token: i2.AccountStatementDataService }, { token: i3.Store }], target: i0.ɵɵFactoryTarget.Injectable });
AccountStatementEffects.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementEffects });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementEffects, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.Actions }, { type: i2.AccountStatementDataService }, { type: i3.Store }]; } });
