// @backbase/account-statement-store-ang/esm2015/src/account-statements/reducers.js
import { DatePeriods, LoadingState, today, unixZeroDay } from '@backbase/account-statement-common-ang';
import { createReducer, on } from '@ngrx/store';
import { loadAccountStatementsBusiness, loadAccountStatementsBusinessSuccess, loadAccountStatementsFailure, updateWidgetPreferences, } from './actions';
export const accountStatementsInitialState = {
    data: [],
    totalCount: 0,
    loading: LoadingState.IDLE,
    error: undefined,
    accountIds: [],
    filterParams: {
        datePeriod: DatePeriods.AllTime,
    },
    widgetPreferences: {
        minimalDate: unixZeroDay(),
        maximumDate: today(),
    },
};
const smartLoadingState = (state, action) => {
    if (action.accountIds && action.accountIds !== state.accountIds)
        return LoadingState.LOADING;
    if (action.filterParams && action.filterParams !== state.filterParams)
        return LoadingState.LOADING_FILTERED;
    if (action.append)
        return LoadingState.LOADING_MORE;
    if ((action.sortingParams && action.sortingParams !== state.sortingParams) ||
        (action.paginationParams && action.paginationParams !== state.paginationParams)) {
        return LoadingState.LOADING_TABLE;
    }
    return LoadingState.LOADING;
};
const reducer = createReducer(accountStatementsInitialState, on(loadAccountStatementsBusiness, (state, action) => (Object.assign(Object.assign({}, state), { loading: smartLoadingState(state, action), error: undefined }))), on(loadAccountStatementsBusinessSuccess, (state, action) => {
    const { accountStatements, accountIds, filterParams, paginationParams, sortingParams, totalCount, append } = action;
    const dataAppendTo = append ? state.data : [];
    return Object.assign(Object.assign(Object.assign(Object.assign(Object.assign(Object.assign({}, state), { loading: LoadingState.IDLE, data: [...dataAppendTo, ...accountStatements], totalCount }), (accountIds && { accountIds })), (filterParams && { filterParams })), (sortingParams && { sortingParams })), (paginationParams && { paginationParams }));
}), on(loadAccountStatementsFailure, (state, action) => (Object.assign(Object.assign({}, state), { error: action.error, loading: LoadingState.IDLE }))), on(updateWidgetPreferences, (state, action) => (Object.assign(Object.assign({}, state), { widgetPreferences: Object.assign(Object.assign({}, state.widgetPreferences), action.widgetPreferences) }))));
// eslint-disable-next-line prefer-arrow/prefer-arrow-functions
export function accountStatementsReducer(state, action) {
    return reducer(state, action);
}
