// @backbase/ui-ang/esm2015/_example/logo/showcase/logo-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { LogoShowcaseComponent } from './logo-showcase.component';
import { LogoModule } from '@backbase/ui-ang/logo';
import * as i0 from "@angular/core";
export class LogoShowcaseModule {
}
LogoShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LogoShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LogoShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LogoShowcaseModule, declarations: [LogoShowcaseComponent], imports: [CommonModule, LogoModule] });
LogoShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LogoShowcaseModule, imports: [[CommonModule, LogoModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LogoShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, LogoModule],
                    declarations: [LogoShowcaseComponent],
                    entryComponents: [LogoShowcaseComponent],
                }]
        }] });
