// @backbase/ui-ang/esm2015/_example/logo/showcase/logo-showcase.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/logo";
let LogoShowcaseComponent = class LogoShowcaseComponent {
};
LogoShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LogoShowcaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
LogoShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: LogoShowcaseComponent, selector: "bb-logo-ui-showcase", ngImport: i0, template: "<h4>Emblem</h4>\n<div class=\"bb-stack bb-block bb-block--xl row\">\n  <bb-logo-ui type=\"emblem\" class=\"bb-stack__item col-2\"></bb-logo-ui>\n  <span>Backbase</span>\n</div>\n<h4>Full</h4>\n<div class=\"bb-stack bb-block bb-block--xl\">\n  <bb-logo-ui type=\"full\" class=\"bb-stack__item\"></bb-logo-ui>\n</div>\n<h4>Inverse</h4>\n<div class=\"bb-stack row\">\n  <bb-logo-ui type=\"emblem\" [isInverse]=\"true\" class=\"bb-stack__item col-2\"></bb-logo-ui>\n  <span>Backbase</span>\n</div>\n", components: [{ type: i1.LogoComponent, selector: "bb-logo-ui", inputs: ["type", "isInverse"] }] });
LogoShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
], LogoShowcaseComponent);
export { LogoShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LogoShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-logo-ui-showcase',
                    templateUrl: './logo-showcase.component.html',
                }]
        }] });
