// @backbase/contact-manager-journey-ang/esm2015/lib/views/contact-manager-type-list/contact-manager-type-list.component.js
import { ChangeDetectionStrategy, Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/contact-manager-widget-ang";
/**
 * The wrappper component will show all contacts in a list
 *
 * @usageNotes
 *
 * ### Display all contacts in a list .
 *
 * ```route
 * export const routes: Routes = [
 *   {
 *     path: '',
 *     component: ContactManagerJourneyComponent,
 *     children: [
 *       {
 *         path: '',
 *         component: ContactManagerDetailViewComponent,
 *       },
 *       {
 *         path: 'select',
 *         component: ContactManagerTypeListViewComponent,
 *       },
 *       {
 *         path: ':id',
 *         component: ContactManagerDetailViewComponent,
 *       },
 *       {
 *         path: 'edit/:id/:type',
 *         canDeactivate: [CanDeactivateGuard],
 *         component: ContactManagerFormViewComponent,
 *       },
 *     ],
 *   },
 * ];
 * };
 * @NgModule({
 *   imports: [ContactManagerJourneyModule.forRoot({ routes })],
 * })
 * export class ContactManagerJourneyBundleModule {}
 * ```
 * @ngModule ContactManagerJourneyModule
 */
export class ContactManagerTypeListViewComponent {
}
/** @nocollapse */ ContactManagerTypeListViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactManagerTypeListViewComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ ContactManagerTypeListViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ContactManagerTypeListViewComponent, selector: "bb-contact-manager-type-list-view", ngImport: i0, template: "<bb-contacts-type-list-container></bb-contacts-type-list-container>\n", components: [{ type: i1.ContactsTypeListContainerComponent, selector: "bb-contacts-type-list-container" }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactManagerTypeListViewComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-contact-manager-type-list-view',
                    templateUrl: './contact-manager-type-list.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }] });
