// @backbase/ui-ang/esm2015/product-item-basic-account/product-item-basic-account.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { CustomProductItemBasicAccountDetailsFormatterDirective, CustomProductItemBasicAccountTypeDirective, CustomProductItemBasicAccountTitleDirective, ProductItemBasicAccountComponent, } from './product-item-basic-account.component';
import { AmountModule } from '@backbase/ui-ang/amount';
import { PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import { HighlightModule } from '@backbase/ui-ang/highlight';
import { IconModule } from '@backbase/ui-ang/icon';
import { EllipsisModule } from '@backbase/ui-ang/ellipsis';
import * as i0 from "@angular/core";
export class ProductItemBasicAccountModule {
}
ProductItemBasicAccountModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemBasicAccountModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductItemBasicAccountModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemBasicAccountModule, declarations: [ProductItemBasicAccountComponent,
        CustomProductItemBasicAccountTypeDirective,
        CustomProductItemBasicAccountTitleDirective,
        CustomProductItemBasicAccountDetailsFormatterDirective], imports: [CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule, EllipsisModule], exports: [ProductItemBasicAccountComponent,
        CustomProductItemBasicAccountTypeDirective,
        CustomProductItemBasicAccountTitleDirective,
        CustomProductItemBasicAccountDetailsFormatterDirective] });
ProductItemBasicAccountModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemBasicAccountModule, imports: [[CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule, EllipsisModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemBasicAccountModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule, EllipsisModule],
                    declarations: [
                        ProductItemBasicAccountComponent,
                        CustomProductItemBasicAccountTypeDirective,
                        CustomProductItemBasicAccountTitleDirective,
                        CustomProductItemBasicAccountDetailsFormatterDirective,
                    ],
                    exports: [
                        ProductItemBasicAccountComponent,
                        CustomProductItemBasicAccountTypeDirective,
                        CustomProductItemBasicAccountTitleDirective,
                        CustomProductItemBasicAccountDetailsFormatterDirective,
                    ],
                }]
        }] });
