// @backbase/ui-ang/esm2015/notification/notification.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NgbAlertModule } from '@ng-bootstrap/ng-bootstrap';
import { IconModule } from '@backbase/ui-ang/icon';
import { NotificationComponent } from './notification.component';
import { ButtonModule } from '@backbase/ui-ang/button';
import { KeyboardClickModule } from '@backbase/ui-ang/keyboard-click-directive';
import { FocusModule } from '@backbase/ui-ang/focus';
import * as i0 from "@angular/core";
export class NotificationModule {
}
NotificationModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NotificationModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationModule, declarations: [NotificationComponent], imports: [CommonModule, NgbAlertModule, IconModule, ButtonModule, KeyboardClickModule, FocusModule], exports: [NotificationComponent] });
NotificationModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationModule, imports: [[CommonModule, NgbAlertModule, IconModule, ButtonModule, KeyboardClickModule, FocusModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, NgbAlertModule, IconModule, ButtonModule, KeyboardClickModule, FocusModule],
                    declarations: [NotificationComponent],
                    exports: [NotificationComponent],
                    entryComponents: [NotificationComponent],
                }]
        }] });
