// @backbase/retail-loans-journey-ang/esm2015/lib/store/loans/actions.js
import { createAction, props } from '@ngrx/store';
// Fetch loans
export const loadLoans = createAction('[Loans API] fetch loans', props());
export const loadLoansSuccess = createAction('[Loans API] fetch loans succeed', props());
export const loadLoansError = createAction('[Loans API] fetch loans error', props());
// Fetch loan by id
export const fetchLoanById = createAction('[Loans API] fetch loan by id', props());
export const fetchLoanByIdSuccess = createAction('[Loans API] fetch loan by id succeed', props());
export const fetchLoanByIdError = createAction('[Loans API] fetch loan by id error', props());
// Fetch amount of loans
export const fetchAmountOfLoans = createAction('[Loans API] fetch amount of loans');
export const fetchAmountOfLoansSuccess = createAction('[Loans API] fetch amount of loans succeed', props());
export const fetchAmountOfLoansError = createAction('[Loans API] fetch amount of loans error');
// Update loan alias
export const updateLoanAlias = createAction('[Product Summary API] Update alias', props());
export const updateLoanAliasSuccess = createAction('[Product Summary API] Update alias succeed', props());
export const updateLoanAliasError = createAction('[Product Summary API] Update alias error', props());
export const updateLoanAliasCancel = createAction('[Effect] Cancel update alias request');
