// @backbase/retail-loans-journey-ang/esm2015/lib/store/loans/selectors.js
import { createFeatureSelector, createSelector } from '@ngrx/store';
import { storeKey } from './reducer';
const getLoansState = createFeatureSelector(storeKey);
export const selectLoans = createSelector(getLoansState, ({ loans }) => loans);
export const selectLoansCallState = createSelector(getLoansState, ({ loansCallState }) => loansCallState);
export const selectLoan = createSelector(getLoansState, ({ loan }) => loan);
export const selectLoanCallState = createSelector(getLoansState, ({ loanCallState }) => loanCallState);
export const selectShowNavigateToOverview = createSelector(getLoansState, ({ loansTotalCount }) => Boolean(loansTotalCount && loansTotalCount > 1));
export const selectLoansCurrentPage = createSelector(getLoansState, ({ loansCurrentPage }) => loansCurrentPage);
export const selectLoansTotalCount = createSelector(getLoansState, ({ loansTotalCount }) => loansTotalCount);
export const selectAliasUpdateCallState = createSelector(getLoansState, ({ aliasUpdateCallState }) => aliasUpdateCallState);
