// @backbase/ui-ang/esm2015/focus/focus.service.js
import { Injectable } from '@angular/core';
import { Subject } from 'rxjs';
import { throttleTime } from 'rxjs/operators';
import * as i0 from "@angular/core";
const isFocusable = (el) => !!el && typeof el.focus === 'function';
export class FocusService {
    constructor() {
        this.focusSubject = new Subject();
        this.sub = this.focusSubject.pipe(throttleTime(0, undefined, { leading: true })).subscribe((focused) => {
            if (isFocusable(focused.nativeElement)) {
                focused.nativeElement.focus();
            }
        });
    }
    focus(el) {
        this.focusSubject.next(el);
    }
    ngOnDestroy() {
        this.sub.unsubscribe();
    }
}
FocusService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FocusService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
FocusService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FocusService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FocusService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }] });
