// @backbase/ui-ang/esm2015/focus/focus.directive.js
import { Directive, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./focus.service";
export class FocusDirective {
    constructor(service, el) {
        this.service = service;
        this.el = el;
        this.autofocus = true;
    }
    set bbFocus(value) {
        this.autofocus = value === '' || !!value;
    }
    ngAfterViewInit() {
        if (this.autofocus) {
            this.focus();
        }
    }
    focus() {
        this.service.focus(this.el);
    }
}
FocusDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FocusDirective, deps: [{ token: i1.FocusService }, { token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive });
FocusDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: FocusDirective, selector: "[bbFocus]", inputs: { bbFocus: "bbFocus" }, exportAs: ["bbFocus"], ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FocusDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbFocus]',
                    exportAs: 'bbFocus',
                }]
        }], ctorParameters: function () { return [{ type: i1.FocusService }, { type: i0.ElementRef }]; }, propDecorators: { bbFocus: [{
                type: Input
            }] } });
