// @backbase/ui-ang/esm2015/focus/focus.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FocusDirective } from './focus.directive';
import { FocusService } from './focus.service';
import * as i0 from "@angular/core";
export class FocusModule {
}
FocusModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FocusModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
FocusModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FocusModule, declarations: [FocusDirective], imports: [CommonModule], exports: [FocusDirective] });
FocusModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FocusModule, providers: [FocusService], imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FocusModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [FocusDirective],
                    providers: [FocusService],
                    exports: [FocusDirective],
                }]
        }] });
