// @backbase/accounts-transactions-journey-ang/bundles/backbase-accounts-transactions-journey-ang.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('@backbase/foundation-ang/core'), require('@angular/router'), require('rxjs'), require('@backbase/product-summary-common-ang'), require('@backbase/transactions-common-ang'), require('@backbase/accounts-transactions-journeys-shared-ui-account-ang'), require('@backbase/ui-ang/header'), require('@backbase/product-summary-list-widget-ang'), require('rxjs/operators'), require('@backbase/transactions-list-widget-ang'), require('@backbase/ui-ang/icon'), require('@backbase/product-summary-ui'), require('@backbase/ui-ang/button'), require('@backbase/product-summary-account-selector-widget-ang'), require('@backbase/product-summary-manage-accounts-widget-ang'), require('@backbase/product-summary-details-widget-ang'), require('@backbase/accounts-transactions-journeys-shared-util-transform-ang'), require('@backbase/accounts-transactions-journeys-shared-util-account-ang'), require('@backbase/accounts-transactions-journeys-shared-ui-transform-ang'), require('@backbase/ui-ang/amount'), require('@backbase/ui-ang/date-pipe'), require('@backbase/ui-ang/payment-card'), require('@backbase/ui-ang/tooltip-directive')) :
    typeof define === 'function' && define.amd ? define('@backbase/accounts-transactions-journey-ang', ['exports', '@angular/core', '@angular/common', '@backbase/foundation-ang/core', '@angular/router', 'rxjs', '@backbase/product-summary-common-ang', '@backbase/transactions-common-ang', '@backbase/accounts-transactions-journeys-shared-ui-account-ang', '@backbase/ui-ang/header', '@backbase/product-summary-list-widget-ang', 'rxjs/operators', '@backbase/transactions-list-widget-ang', '@backbase/ui-ang/icon', '@backbase/product-summary-ui', '@backbase/ui-ang/button', '@backbase/product-summary-account-selector-widget-ang', '@backbase/product-summary-manage-accounts-widget-ang', '@backbase/product-summary-details-widget-ang', '@backbase/accounts-transactions-journeys-shared-util-transform-ang', '@backbase/accounts-transactions-journeys-shared-util-account-ang', '@backbase/accounts-transactions-journeys-shared-ui-transform-ang', '@backbase/ui-ang/amount', '@backbase/ui-ang/date-pipe', '@backbase/ui-ang/payment-card', '@backbase/ui-ang/tooltip-directive'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["accounts-transactions-journey-ang"] = {}), global.ng.core, global.ng.common, global.i1$1, global.ng.router, global.rxjs, global.i4, global.i3, global.i1, global.i1$2, global.i2$1, global.rxjs.operators, global.i4$1, global.i5, global.i6, global.i7, global.productSummaryAccountSelectorWidgetAng, global.i4$2, global.i2$2, global.accountsTransactionsJourneysSharedUtilTransformAng, global.accountsTransactionsJourneysSharedUtilAccountAng, global.i3$1, global.amount, global.datePipe, global.paymentCard, global.tooltipDirective));
})(this, (function (exports, i0, i8, i1$1, i2, rxjs, i4, i3, i1, i1$2, i2$1, operators, i4$1, i5, i6, i7, productSummaryAccountSelectorWidgetAng, i4$2, i2$2, accountsTransactionsJourneysSharedUtilTransformAng, accountsTransactionsJourneysSharedUtilAccountAng, i3$1, amount, datePipe, paymentCard, tooltipDirective) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i8__namespace = /*#__PURE__*/_interopNamespace(i8);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1$1);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i4__namespace$1 = /*#__PURE__*/_interopNamespace(i4);
    var i3__namespace = /*#__PURE__*/_interopNamespace(i3);
    var i1__namespace$2 = /*#__PURE__*/_interopNamespace(i1);
    var i1__namespace$1 = /*#__PURE__*/_interopNamespace(i1$2);
    var i2__namespace$1 = /*#__PURE__*/_interopNamespace(i2$1);
    var i4__namespace = /*#__PURE__*/_interopNamespace(i4$1);
    var i5__namespace = /*#__PURE__*/_interopNamespace(i5);
    var i6__namespace = /*#__PURE__*/_interopNamespace(i6);
    var i7__namespace = /*#__PURE__*/_interopNamespace(i7);
    var i4__namespace$2 = /*#__PURE__*/_interopNamespace(i4$2);
    var i2__namespace$2 = /*#__PURE__*/_interopNamespace(i2$2);
    var i3__namespace$1 = /*#__PURE__*/_interopNamespace(i3$1);

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from, pack) {
        if (pack || arguments.length === 2)
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar)
                        ar = Array.prototype.slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
        return to.concat(ar || Array.prototype.slice.call(from));
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    var _a;
    var defaultConfiguration = [
        {
            title: $localize(templateObject_1$2 || (templateObject_1$2 = __makeTemplateObject([":@@product.details.other.account.label.general.section:General"], [":@@product.details.other.account.label.general.section:General"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_2$2 || (templateObject_2$2 = __makeTemplateObject([":@@product.details.other.account.label.productTypeName:Account Type"], [":@@product.details.other.account.label.productTypeName:Account Type"]))),
                    key: 'productTypeName',
                },
                {
                    type: i1.AccountInfoPropertyType.ACCOUNT_NUMBER,
                    label: $localize(templateObject_3 || (templateObject_3 = __makeTemplateObject([":@@product.details.other.account.label.account.number:Account Number"], [":@@product.details.other.account.label.account.number:Account Number"]))),
                    key: 'IBAN',
                    numberType: 'iban',
                },
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_4 || (templateObject_4 = __makeTemplateObject([":@@product.details.other.account.label.bookedBalance:Account Balance"], [":@@product.details.other.account.label.bookedBalance:Account Balance"]))),
                    key: 'bookedBalance',
                    tooltip: $localize(templateObject_5 || (templateObject_5 = __makeTemplateObject([":@@product.details.other.account.tooltip.bookedBalance:Account Balance"], [":@@product.details.other.account.tooltip.bookedBalance:Account Balance"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_6 || (templateObject_6 = __makeTemplateObject([":@@product.details.other.account.label.accountHolderNames:Account Owner(s)"], [":@@product.details.other.account.label.accountHolderNames:Account Owner(s)"]))),
                    key: 'accountHolderNames',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_7 || (templateObject_7 = __makeTemplateObject([":@@product.details.other.account.label.bankBranchCode:ACH Routing Number"], [":@@product.details.other.account.label.bankBranchCode:ACH Routing Number"]))),
                    key: 'bankBranchCode',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_8 || (templateObject_8 = __makeTemplateObject([":@@product.details.other.account.label.bankBranchCode2:FedWire Routing Number"], [":@@product.details.other.account.label.bankBranchCode2:FedWire Routing Number"]))),
                    key: 'bankBranchCode2',
                },
                {
                    type: i1.AccountInfoPropertyType.ACCOUNT_STATE,
                    label: $localize(templateObject_9 || (templateObject_9 = __makeTemplateObject([":@@product.details.other.account.label.accountStatus:Account Status"], [":@@product.details.other.account.label.accountStatus:Account Status"]))),
                    key: 'state.state',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_10 || (templateObject_10 = __makeTemplateObject([":@@product.details.other.account.label.financialInstitution.name:Bank Name"], [":@@product.details.other.account.label.financialInstitution.name:Bank Name"]))),
                    key: 'financialInstitution.name',
                },
                {
                    type: i1.AccountInfoPropertyType.UPDATE_TIME,
                    label: $localize(templateObject_11 || (templateObject_11 = __makeTemplateObject([":@@product.details.other.account.label.lastSyncDate:Last sync"], [":@@product.details.other.account.label.lastSyncDate:Last sync"]))),
                    key: 'lastSyncDate',
                },
            ],
        },
        {
            title: $localize(templateObject_12 || (templateObject_12 = __makeTemplateObject([":@@product.details.other.account.label.interest.details.section:Interest Details"], [":@@product.details.other.account.label.interest.details.section:Interest Details"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.PERCENT,
                    label: $localize(templateObject_13 || (templateObject_13 = __makeTemplateObject([":@@product.details.other.account.label.accountInterestRate:Interest Rate"], [":@@product.details.other.account.label.accountInterestRate:Interest Rate"]))),
                    key: 'accountInterestRate',
                    tooltip: $localize(templateObject_14 || (templateObject_14 = __makeTemplateObject([":@@product.details.other.account.tooltip.accountInterestRate:The annualized cost of credit or debt-capital computed as the percentage ratio of interest to the principal"], [":@@product.details.other.account.tooltip.accountInterestRate:The annualized cost of credit or debt-capital computed as the percentage ratio of interest to the principal"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_15 || (templateObject_15 = __makeTemplateObject([":@@product.details.other.account.label.accruedInterest:Accrued Interest"], [":@@product.details.other.account.label.accruedInterest:Accrued Interest"]))),
                    key: 'accruedInterest',
                    tooltip: $localize(templateObject_16 || (templateObject_16 = __makeTemplateObject([":@@product.details.other.account.tooltip.accruedInterest:The interest that is earned (credit interest) or due (debit interest) but not settled yet"], [":@@product.details.other.account.tooltip.accruedInterest:The interest that is earned (credit interest) or due (debit interest) but not settled yet"]))),
                },
            ],
        },
        {
            title: $localize(templateObject_17 || (templateObject_17 = __makeTemplateObject([":@@product.details.other.account.label.other.section:Other"], [":@@product.details.other.account.label.other.section:Other"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.DATE,
                    label: $localize(templateObject_18 || (templateObject_18 = __makeTemplateObject([":@@product.details.other.account.label.accountOpeningDate:Account Opening Date"], [":@@product.details.other.account.label.accountOpeningDate:Account Opening Date"]))),
                    key: 'accountOpeningDate',
                    tooltip: $localize(templateObject_19 || (templateObject_19 = __makeTemplateObject([":@@product.details.other.account.tooltip.accountOpeningDate:Account Opening Date"], [":@@product.details.other.account.tooltip.accountOpeningDate:Account Opening Date"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.DATE,
                    label: $localize(templateObject_20 || (templateObject_20 = __makeTemplateObject([":@@product.details.other.account.label.lastUpdateDate:Last Updated Date"], [":@@product.details.other.account.label.lastUpdateDate:Last Updated Date"]))),
                    key: 'lastUpdateDate',
                    tooltip: $localize(templateObject_21 || (templateObject_21 = __makeTemplateObject([":@@product.details.other.account.tooltip.lastUpdateDate:Last date of parameter update for the product"], [":@@product.details.other.account.tooltip.lastUpdateDate:Last date of parameter update for the product"]))),
                },
            ],
        },
    ];
    var creditCardConfiguration = [
        {
            title: $localize(templateObject_22 || (templateObject_22 = __makeTemplateObject([":@@product.details.credit.card.label.general.section:General"], [":@@product.details.credit.card.label.general.section:General"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_23 || (templateObject_23 = __makeTemplateObject([":@@product.details.credit.card.label.productTypeName:Account Type"], [":@@product.details.credit.card.label.productTypeName:Account Type"]))),
                    key: 'productTypeName',
                },
                {
                    type: i1.AccountInfoPropertyType.ACCOUNT_NUMBER,
                    label: $localize(templateObject_24 || (templateObject_24 = __makeTemplateObject([":@@product.details.credit.card.label.number:Account Number"], [":@@product.details.credit.card.label.number:Account Number"]))),
                    key: 'number',
                    numberType: 'cardNumber',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_25 || (templateObject_25 = __makeTemplateObject([":@@product.details.credit.card.label.accountHolderNames:Account Owner(s)"], [":@@product.details.credit.card.label.accountHolderNames:Account Owner(s)"]))),
                    key: 'accountHolderNames',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_26 || (templateObject_26 = __makeTemplateObject([":@@product.details.credit.card.label.cardProvider:Card Provider"], [":@@product.details.credit.card.label.cardProvider:Card Provider"]))),
                    key: 'cardDetails.cardProvider',
                    tooltip: $localize(templateObject_27 || (templateObject_27 = __makeTemplateObject([":@@product.details.credit.card.tooltip.cardProvider:Card Provider"], [":@@product.details.credit.card.tooltip.cardProvider:Card Provider"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.DATE,
                    label: $localize(templateObject_28 || (templateObject_28 = __makeTemplateObject([":@@product.details.credit.card.label.validThru:Valid thru"], [":@@product.details.credit.card.label.validThru:Valid thru"]))),
                    key: 'validThru',
                    tooltip: $localize(templateObject_29 || (templateObject_29 = __makeTemplateObject([":@@product.details.credit.card.tooltip.validThru:Expiration date of a credit card, after which is no longer valid"], [":@@product.details.credit.card.tooltip.validThru:Expiration date of a credit card, after which is no longer valid"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.ACCOUNT_STATE,
                    label: $localize(templateObject_30 || (templateObject_30 = __makeTemplateObject([":@@product.details.credit.card.label.accountStatus:Account Status"], [":@@product.details.credit.card.label.accountStatus:Account Status"]))),
                    key: 'state.state',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_31 || (templateObject_31 = __makeTemplateObject([":@@product.details.credit.card.label.financialInstitution.name:Bank Name"], [":@@product.details.credit.card.label.financialInstitution.name:Bank Name"]))),
                    key: 'financialInstitution.name',
                },
                {
                    type: i1.AccountInfoPropertyType.UPDATE_TIME,
                    label: $localize(templateObject_32 || (templateObject_32 = __makeTemplateObject([":@@product.details.credit.card.label.lastSyncDate:Last sync"], [":@@product.details.credit.card.label.lastSyncDate:Last sync"]))),
                    key: 'lastSyncDate',
                },
            ],
        },
        {
            title: $localize(templateObject_33 || (templateObject_33 = __makeTemplateObject([":@@product.details.credit.card.label.status.section:Status"], [":@@product.details.credit.card.label.status.section:Status"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_34 || (templateObject_34 = __makeTemplateObject([":@@product.details.credit.card.label.consumedAmount:Consumed Amount"], [":@@product.details.credit.card.label.consumedAmount:Consumed Amount"]))),
                    key: 'bookedBalance',
                    tooltip: $localize(templateObject_35 || (templateObject_35 = __makeTemplateObject([":@@product.details.credit.card.tooltip.consumedAmount:Consumed Amount"], [":@@product.details.credit.card.tooltip.consumedAmount:Consumed Amount"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_36 || (templateObject_36 = __makeTemplateObject([":@@product.details.credit.card.label.remainingAmount:Remaining Amount"], [":@@product.details.credit.card.label.remainingAmount:Remaining Amount"]))),
                    key: 'remainingCredit',
                    tooltip: $localize(templateObject_37 || (templateObject_37 = __makeTemplateObject([":@@product.details.credit.card.tooltip.remainingAmount:Remaining Amount"], [":@@product.details.credit.card.tooltip.remainingAmount:Remaining Amount"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_38 || (templateObject_38 = __makeTemplateObject([":@@product.details.credit.card.label.creditLimit:Credit Limit"], [":@@product.details.credit.card.label.creditLimit:Credit Limit"]))),
                    key: 'creditLimit',
                    tooltip: $localize(templateObject_39 || (templateObject_39 = __makeTemplateObject([":@@product.details.credit.card.tooltip.creditLimit:Monetary amount of the used overdraft"], [":@@product.details.credit.card.tooltip.creditLimit:Monetary amount of the used overdraft"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_40 || (templateObject_40 = __makeTemplateObject([":@@product.details.credit.card.label.statementBalance:Statement Balance"], [":@@product.details.credit.card.label.statementBalance:Statement Balance"]))),
                    key: 'cardDetails.statementBalance',
                    tooltip: $localize(templateObject_41 || (templateObject_41 = __makeTemplateObject([":@@product.details.credit.card.tooltip.statementBalance:Statement Balance"], [":@@product.details.credit.card.tooltip.statementBalance:Statement Balance"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_42 || (templateObject_42 = __makeTemplateObject([":@@product.details.credit.card.label.availableCashCredit:Available Cash Credit Limit"], [":@@product.details.credit.card.label.availableCashCredit:Available Cash Credit Limit"]))),
                    key: 'cardDetails.availableCashCredit',
                    tooltip: $localize(templateObject_43 || (templateObject_43 = __makeTemplateObject([":@@product.details.credit.card.tooltip.availableCashCredit:Available Cash Credit Limit"], [":@@product.details.credit.card.tooltip.availableCashCredit:Available Cash Credit Limit"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_44 || (templateObject_44 = __makeTemplateObject([":@@product.details.credit.card.label.cashCreditLimit:Cash Credit Limit"], [":@@product.details.credit.card.label.cashCreditLimit:Cash Credit Limit"]))),
                    key: 'cardDetails.cashCreditLimit',
                    tooltip: $localize(templateObject_45 || (templateObject_45 = __makeTemplateObject([":@@product.details.credit.card.tooltip.cashCreditLimit:Cash Credit Limit"], [":@@product.details.credit.card.tooltip.cashCreditLimit:Cash Credit Limit"]))),
                },
            ],
        },
        {
            title: $localize(templateObject_46 || (templateObject_46 = __makeTemplateObject([":@@product.details.credit.card.label.payment.section:Payment and Interest Details"], [":@@product.details.credit.card.label.payment.section:Payment and Interest Details"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_47 || (templateObject_47 = __makeTemplateObject([":@@product.details.credit.card.label.minimumPayment:Minimum Payment"], [":@@product.details.credit.card.label.minimumPayment:Minimum Payment"]))),
                    key: 'minimumPayment',
                    tooltip: $localize(templateObject_48 || (templateObject_48 = __makeTemplateObject([":@@product.details.credit.card.tooltip.minimumPayment:The minimum payment set a percentage of balance, or a fixed cash amount"], [":@@product.details.credit.card.tooltip.minimumPayment:The minimum payment set a percentage of balance, or a fixed cash amount"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.DATE,
                    label: $localize(templateObject_49 || (templateObject_49 = __makeTemplateObject([":@@product.details.credit.card.label.minimumPaymentDueDate:Minimum Payment Due Date"], [":@@product.details.credit.card.label.minimumPaymentDueDate:Minimum Payment Due Date"]))),
                    key: 'minimumPaymentDueDate',
                    tooltip: $localize(templateObject_50 || (templateObject_50 = __makeTemplateObject([":@@product.details.credit.card.tooltip.minimumPaymentDueDate:Minimum Payment Due Date shown on your monthly statement to remain in good standing"], [":@@product.details.credit.card.tooltip.minimumPaymentDueDate:Minimum Payment Due Date shown on your monthly statement to remain in good standing"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.PERCENT,
                    label: $localize(templateObject_51 || (templateObject_51 = __makeTemplateObject([":@@product.details.credit.card.label.cashAdvanceInterestRate:Cash Advance APR"], [":@@product.details.credit.card.label.cashAdvanceInterestRate:Cash Advance APR"]))),
                    key: 'interestDetails.cashAdvanceInterestRate',
                    tooltip: $localize(templateObject_52 || (templateObject_52 = __makeTemplateObject([":@@product.details.credit.card.tooltip.cashAdvanceInterestRate:Cash Advance APR"], [":@@product.details.credit.card.tooltip.cashAdvanceInterestRate:Cash Advance APR"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.PERCENT,
                    label: $localize(templateObject_53 || (templateObject_53 = __makeTemplateObject([":@@product.details.credit.card.label.applicableInterestRate:Interest Rate"], [":@@product.details.credit.card.label.applicableInterestRate:Interest Rate"]))),
                    key: 'applicableInterestRate',
                    tooltip: $localize(templateObject_54 || (templateObject_54 = __makeTemplateObject([":@@product.details.credit.card.tooltip.applicableInterestRate:The annualized cost of credit or debt-capital computed as the percentage ratio of interest to the principal"], [":@@product.details.credit.card.tooltip.applicableInterestRate:The annualized cost of credit or debt-capital computed as the percentage ratio of interest to the principal"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_55 || (templateObject_55 = __makeTemplateObject([":@@product.details.credit.card.label.lastPaymentAmount:Last Payment Amount"], [":@@product.details.credit.card.label.lastPaymentAmount:Last Payment Amount"]))),
                    key: 'cardDetails.lastPaymentAmount',
                    tooltip: $localize(templateObject_56 || (templateObject_56 = __makeTemplateObject([":@@product.details.credit.card.tooltip.lastPaymentAmount:Last Payment Amount"], [":@@product.details.credit.card.tooltip.lastPaymentAmount:Last Payment Amount"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.DATE,
                    label: $localize(templateObject_57 || (templateObject_57 = __makeTemplateObject([":@@product.details.credit.card.label.lastPaymentDate:Last Payment Date"], [":@@product.details.credit.card.label.lastPaymentDate:Last Payment Date"]))),
                    key: 'cardDetails.lastPaymentDate',
                    tooltip: $localize(templateObject_58 || (templateObject_58 = __makeTemplateObject([":@@product.details.credit.card.tooltip.lastPaymentDate:Last Payment Date"], [":@@product.details.credit.card.tooltip.lastPaymentDate:Last Payment Date"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_59 || (templateObject_59 = __makeTemplateObject([":@@product.details.credit.card.label.previousStatementBalance:Last Statement Balance"], [":@@product.details.credit.card.label.previousStatementBalance:Last Statement Balance"]))),
                    key: 'cardDetails.previousStatementBalance',
                    tooltip: $localize(templateObject_60 || (templateObject_60 = __makeTemplateObject([":@@product.details.credit.card.tooltip.previousStatementBalance:Last Statement Balance"], [":@@product.details.credit.card.tooltip.previousStatementBalance:Last Statement Balance"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.DATE,
                    label: $localize(templateObject_61 || (templateObject_61 = __makeTemplateObject([":@@product.details.credit.card.label.previousStatementDate:Last Statement Date"], [":@@product.details.credit.card.label.previousStatementDate:Last Statement Date"]))),
                    key: 'cardDetails.previousStatementDate',
                    tooltip: $localize(templateObject_62 || (templateObject_62 = __makeTemplateObject([":@@product.details.credit.card.tooltip.previousStatementDate:Last Statement Date"], [":@@product.details.credit.card.tooltip.previousStatementDate:Last Statement Date"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.PERCENT,
                    label: $localize(templateObject_63 || (templateObject_63 = __makeTemplateObject([":@@product.details.credit.card.label.penaltyInterestRate:Penalty APR"], [":@@product.details.credit.card.label.penaltyInterestRate:Penalty APR"]))),
                    key: 'interestDetails.penaltyInterestRate',
                    tooltip: $localize(templateObject_64 || (templateObject_64 = __makeTemplateObject([":@@product.details.credit.card.tooltip.penaltyInterestRate:Penalty APR"], [":@@product.details.credit.card.tooltip.penaltyInterestRate:Penalty APR"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_65 || (templateObject_65 = __makeTemplateObject([":@@product.details.credit.card.label.latePaymentFee:Late Fee"], [":@@product.details.credit.card.label.latePaymentFee:Late Fee"]))),
                    key: 'cardDetails.latePaymentFee',
                    tooltip: $localize(templateObject_66 || (templateObject_66 = __makeTemplateObject([":@@product.details.credit.card.tooltip.latePaymentFee:Late Fee"], [":@@product.details.credit.card.tooltip.latePaymentFee:Late Fee"]))),
                },
            ],
        },
        {
            title: $localize(templateObject_67 || (templateObject_67 = __makeTemplateObject([":@@product.details.credit.card.label.other.section:Other"], [":@@product.details.credit.card.label.other.section:Other"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.DATE,
                    label: $localize(templateObject_68 || (templateObject_68 = __makeTemplateObject([":@@product.details.credit.card.label.accountOpeningDate:Account Opening Date"], [":@@product.details.credit.card.label.accountOpeningDate:Account Opening Date"]))),
                    key: 'accountOpeningDate',
                    tooltip: $localize(templateObject_69 || (templateObject_69 = __makeTemplateObject([":@@product.details.credit.card.tooltip.accountOpeningDate:The date of activation of the account in the bank's system"], [":@@product.details.credit.card.tooltip.accountOpeningDate:The date of activation of the account in the bank's system"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.DATE,
                    label: $localize(templateObject_70 || (templateObject_70 = __makeTemplateObject([":@@product.details.credit.card.label.lastUpdateDate:Last Updated Date"], [":@@product.details.credit.card.label.lastUpdateDate:Last Updated Date"]))),
                    key: 'lastUpdateDate',
                    tooltip: $localize(templateObject_71 || (templateObject_71 = __makeTemplateObject([":@@product.details.credit.card.tooltip.lastUpdateDate:Last date of parameter update for the product"], [":@@product.details.credit.card.tooltip.lastUpdateDate:Last date of parameter update for the product"]))),
                },
            ],
        },
    ];
    var currentAccountConfiguration = [
        {
            title: $localize(templateObject_72 || (templateObject_72 = __makeTemplateObject([":@@product.details.current.account.label.general.section:General"], [":@@product.details.current.account.label.general.section:General"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_73 || (templateObject_73 = __makeTemplateObject([":@@product.details.current.account.label.productTypeName:Account Type"], [":@@product.details.current.account.label.productTypeName:Account Type"]))),
                    key: 'productTypeName',
                },
                {
                    type: i1.AccountInfoPropertyType.ACCOUNT_NUMBER,
                    label: $localize(templateObject_74 || (templateObject_74 = __makeTemplateObject([":@@product.details.current.account.label.iban:IBAN"], [":@@product.details.current.account.label.iban:IBAN"]))),
                    key: 'IBAN',
                    numberType: 'iban',
                },
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_75 || (templateObject_75 = __makeTemplateObject([":@@product.details.current.account.label.availableBalance:Available Balance"], [":@@product.details.current.account.label.availableBalance:Available Balance"]))),
                    key: 'availableBalance',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_76 || (templateObject_76 = __makeTemplateObject([":@@product.details.current.account.label.accountHolderNames:Account Owner(s)"], [":@@product.details.current.account.label.accountHolderNames:Account Owner(s)"]))),
                    key: 'accountHolderNames',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_77 || (templateObject_77 = __makeTemplateObject([":@@account.details.current.account.label.bankBranchCode:ACH Routing Number"], [":@@account.details.current.account.label.bankBranchCode:ACH Routing Number"]))),
                    key: 'bankBranchCode',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_78 || (templateObject_78 = __makeTemplateObject([":@@account.details.current.account.label.bankBranchCode2:FedWire Routing Number"], [":@@account.details.current.account.label.bankBranchCode2:FedWire Routing Number"]))),
                    key: 'bankBranchCode2',
                },
                {
                    type: i1.AccountInfoPropertyType.ACCOUNT_STATE,
                    label: $localize(templateObject_79 || (templateObject_79 = __makeTemplateObject([":@@product.details.current.account.label.accountStatus:Account Status"], [":@@product.details.current.account.label.accountStatus:Account Status"]))),
                    key: 'state.state',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_80 || (templateObject_80 = __makeTemplateObject([":@@product.details.current.account.label.financialInstitution.name:Bank Name"], [":@@product.details.current.account.label.financialInstitution.name:Bank Name"]))),
                    key: 'financialInstitution.name',
                },
                {
                    type: i1.AccountInfoPropertyType.UPDATE_TIME,
                    label: $localize(templateObject_81 || (templateObject_81 = __makeTemplateObject([":@@product.details.current.account.label.lastSyncDate:Last sync"], [":@@product.details.current.account.label.lastSyncDate:Last sync"]))),
                    key: 'lastSyncDate',
                },
            ],
        },
        {
            title: $localize(templateObject_82 || (templateObject_82 = __makeTemplateObject([":@@product.details.current.account.label.interest.section:Interest Details"], [":@@product.details.current.account.label.interest.section:Interest Details"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.PERCENT,
                    label: $localize(templateObject_83 || (templateObject_83 = __makeTemplateObject([":@@product.details.current.account.label.interestRate:Interest Rate"], [":@@product.details.current.account.label.interestRate:Interest Rate"]))),
                    tooltip: $localize(templateObject_84 || (templateObject_84 = __makeTemplateObject([":@@product.details.current.account.tooltip.interestRate:The annualized cost of credit or debt-capital computed as the percentage ratio of interest to the principal"], [":@@product.details.current.account.tooltip.interestRate:The annualized cost of credit or debt-capital computed as the percentage ratio of interest to the principal"]))),
                    key: 'accountInterestRate',
                },
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_85 || (templateObject_85 = __makeTemplateObject([":@@product.details.current.account.label.accruedInterest:Accrued Interest"], [":@@product.details.current.account.label.accruedInterest:Accrued Interest"]))),
                    tooltip: $localize(templateObject_86 || (templateObject_86 = __makeTemplateObject([":@@product.details.current.account.tooltip.accruedInterest:The interest that is earned (credit interest) or due (debit interest) but not settled yet"], [":@@product.details.current.account.tooltip.accruedInterest:The interest that is earned (credit interest) or due (debit interest) but not settled yet"]))),
                    key: 'accruedInterest',
                },
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_87 || (templateObject_87 = __makeTemplateObject([":@@product.details.current-account.label.account.lastYearAccruedInterest:Last year\u2019s accrued interest"], [":@@product.details.current-account.label.account.lastYearAccruedInterest:Last year\u2019s accrued interest"]))),
                    tooltip: $localize(templateObject_88 || (templateObject_88 = __makeTemplateObject([":@@product.details.current-account.tooltip.lastYearAccruedInterest:Last year\u2019s accrued interest"], [":@@product.details.current-account.tooltip.lastYearAccruedInterest:Last year\u2019s accrued interest"]))),
                    key: 'interestDetails.lastYearAccruedInterest',
                },
            ],
        },
        {
            title: $localize(templateObject_89 || (templateObject_89 = __makeTemplateObject([":@@product.details.current.account.label.overdraft.section:Overdraft Details"], [":@@product.details.current.account.label.overdraft.section:Overdraft Details"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_90 || (templateObject_90 = __makeTemplateObject([":@@product.details.current.account.label.overdraftLimit:Overdraft Limit"], [":@@product.details.current.account.label.overdraftLimit:Overdraft Limit"]))),
                    tooltip: $localize(templateObject_91 || (templateObject_91 = __makeTemplateObject([":@@product.details.current.account.tooltip.overdraftLimit:Overdraft Limit"], [":@@product.details.current.account.tooltip.overdraftLimit:Overdraft Limit"]))),
                    key: 'creditLimit',
                },
                {
                    type: i1.AccountInfoPropertyType.DATE,
                    label: $localize(templateObject_92 || (templateObject_92 = __makeTemplateObject([":@@product.details.current.account.label.overdraftexpirydate:Overdraft Expiry Date"], [":@@product.details.current.account.label.overdraftexpirydate:Overdraft Expiry Date"]))),
                    tooltip: $localize(templateObject_93 || (templateObject_93 = __makeTemplateObject([":@@product.details.current.account.tooltip.overdraftexpirydate:Expiration date of the overdraft limit"], [":@@product.details.current.account.tooltip.overdraftexpirydate:Expiration date of the overdraft limit"]))),
                    key: 'creditLimitExpiryDate',
                },
            ],
        },
        {
            title: $localize(templateObject_94 || (templateObject_94 = __makeTemplateObject([":@@product.details.current.account.label.other.section:Other"], [":@@product.details.current.account.label.other.section:Other"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.DATE,
                    label: $localize(templateObject_95 || (templateObject_95 = __makeTemplateObject([":@@product.details.current.account.label.accountOpeningDate:Account Opening Date"], [":@@product.details.current.account.label.accountOpeningDate:Account Opening Date"]))),
                    tooltip: $localize(templateObject_96 || (templateObject_96 = __makeTemplateObject([":@@product.details.current.account.tooltip.accountOpeningDate:Account Opening Date"], [":@@product.details.current.account.tooltip.accountOpeningDate:Account Opening Date"]))),
                    key: 'accountOpeningDate',
                },
                {
                    type: i1.AccountInfoPropertyType.DATE,
                    label: $localize(templateObject_97 || (templateObject_97 = __makeTemplateObject([":@@product.details.current.account.label.lastUpdateDate:Last Updated Date"], [":@@product.details.current.account.label.lastUpdateDate:Last Updated Date"]))),
                    tooltip: $localize(templateObject_98 || (templateObject_98 = __makeTemplateObject([":@@product.details.current.account.tooltip.lastUpdateDate:Last date of parameter update for the product"], [":@@product.details.current.account.tooltip.lastUpdateDate:Last date of parameter update for the product"]))),
                    key: 'lastUpdateDate',
                },
            ],
        },
        {
            title: $localize(templateObject_99 || (templateObject_99 = __makeTemplateObject([":@@product.details.current.account.label.associatedDebitCards:Associated Debit Cards"], [":@@product.details.current.account.label.associatedDebitCards:Associated Debit Cards"]))),
            hideCard: true,
            properties: [
                {
                    type: i1.AccountInfoPropertyType.DEBIT_CARDS,
                    key: 'debitCards',
                },
            ],
        },
    ];
    var debitCardConfiguration = [
        {
            title: $localize(templateObject_100 || (templateObject_100 = __makeTemplateObject([":@@product.details.debit.card.label.general.section:General"], [":@@product.details.debit.card.label.general.section:General"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_101 || (templateObject_101 = __makeTemplateObject([":@@product.details.debit.card.label.productTypeName:Account Type"], [":@@product.details.debit.card.label.productTypeName:Account Type"]))),
                    key: 'productTypeName',
                },
                {
                    type: i1.AccountInfoPropertyType.ACCOUNT_NUMBER,
                    label: $localize(templateObject_102 || (templateObject_102 = __makeTemplateObject([":@@product.details.debit.card.label.iban:IBAN"], [":@@product.details.debit.card.label.iban:IBAN"]))),
                    key: 'IBAN',
                    numberType: 'iban',
                },
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_103 || (templateObject_103 = __makeTemplateObject([":@@product.details.debit.card.label.availableBalance:Available Balance"], [":@@product.details.debit.card.label.availableBalance:Available Balance"]))),
                    key: 'availableBalance',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_104 || (templateObject_104 = __makeTemplateObject([":@@product.details.debit.card.label.accountHolderNames:Account Owner(s)"], [":@@product.details.debit.card.label.accountHolderNames:Account Owner(s)"]))),
                    key: 'accountHolderNames',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_105 || (templateObject_105 = __makeTemplateObject([":@@product.details.debit.card.label.bankBranchCode:ACH Routing Number"], [":@@product.details.debit.card.label.bankBranchCode:ACH Routing Number"]))),
                    key: 'bankBranchCode',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_106 || (templateObject_106 = __makeTemplateObject([":@@product.details.debit.card.label.bankBranchCode2:FedWire Routing Number"], [":@@product.details.debit.card.label.bankBranchCode2:FedWire Routing Number"]))),
                    key: 'bankBranchCode2',
                },
                {
                    type: i1.AccountInfoPropertyType.ACCOUNT_STATE,
                    label: $localize(templateObject_107 || (templateObject_107 = __makeTemplateObject([":@@product.details.debit.card.label.accountStatus:Account Status"], [":@@product.details.debit.card.label.accountStatus:Account Status"]))),
                    key: 'state.state',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_108 || (templateObject_108 = __makeTemplateObject([":@@product.details.debit.card.label.financialInstitution.name:Bank Name"], [":@@product.details.debit.card.label.financialInstitution.name:Bank Name"]))),
                    key: 'financialInstitution.name',
                },
                {
                    type: i1.AccountInfoPropertyType.UPDATE_TIME,
                    label: $localize(templateObject_109 || (templateObject_109 = __makeTemplateObject([":@@product.details.debit.card.label.lastSyncDate.name:Last sync"], [":@@product.details.debit.card.label.lastSyncDate.name:Last sync"]))),
                    key: 'lastSyncDate',
                },
            ],
        },
        {
            title: $localize(templateObject_110 || (templateObject_110 = __makeTemplateObject([":@@product.details.debit.card.label.other.section:Other"], [":@@product.details.debit.card.label.other.section:Other"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.DATE,
                    label: $localize(templateObject_111 || (templateObject_111 = __makeTemplateObject([":@@product.details.debit.card.label.accountOpeningDate:Account Opening Date"], [":@@product.details.debit.card.label.accountOpeningDate:Account Opening Date"]))),
                    key: 'accountOpeningDate',
                    tooltip: $localize(templateObject_112 || (templateObject_112 = __makeTemplateObject([":@@product.details.debit.card.tooltip.accountOpeningDate:Account Opening Date"], [":@@product.details.debit.card.tooltip.accountOpeningDate:Account Opening Date"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.DATE,
                    label: $localize(templateObject_113 || (templateObject_113 = __makeTemplateObject([":@@product.details.debit.card.label.lastUpdateDate:Last Updated Date"], [":@@product.details.debit.card.label.lastUpdateDate:Last Updated Date"]))),
                    key: 'lastUpdateDate',
                    tooltip: $localize(templateObject_114 || (templateObject_114 = __makeTemplateObject([":@@product.details.debit.card.tooltip.lastUpdateDate:Last date of parameter update for the product"], [":@@product.details.debit.card.tooltip.lastUpdateDate:Last date of parameter update for the product"]))),
                },
            ],
        },
    ];
    var investmentAccountConfiguration = [
        {
            title: $localize(templateObject_115 || (templateObject_115 = __makeTemplateObject([":@@product.details.investment.account.label.general.section:General"], [":@@product.details.investment.account.label.general.section:General"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_116 || (templateObject_116 = __makeTemplateObject([":@@product.details.investment.account.label.productTypeName:Account Type"], [":@@product.details.investment.account.label.productTypeName:Account Type"]))),
                    key: 'productTypeName',
                },
                {
                    type: i1.AccountInfoPropertyType.ACCOUNT_NUMBER,
                    label: $localize(templateObject_117 || (templateObject_117 = __makeTemplateObject([":@@product.details.investment.account.label.account.number:Account Number"], [":@@product.details.investment.account.label.account.number:Account Number"]))),
                    key: 'BBAN',
                    numberType: 'bban',
                },
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_118 || (templateObject_118 = __makeTemplateObject([":@@product.details.investment.account.label.totalInvestmentValue:Total Investment Value"], [":@@product.details.investment.account.label.totalInvestmentValue:Total Investment Value"]))),
                    key: 'totalInvestmentValue',
                    tooltip: $localize(templateObject_119 || (templateObject_119 = __makeTemplateObject([":@@product.details.investment.account.tooltip.totalInvestmentValue:Total Investment Value"], [":@@product.details.investment.account.tooltip.totalInvestmentValue:Total Investment Value"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_120 || (templateObject_120 = __makeTemplateObject([":@@product.details.investment.account.label.accountHolderNames:Account Owner(s)"], [":@@product.details.investment.account.label.accountHolderNames:Account Owner(s)"]))),
                    key: 'accountHolderNames',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_121 || (templateObject_121 = __makeTemplateObject([":@@product.details.investment.account.label.bankBranchCode:ACH Routing Number"], [":@@product.details.investment.account.label.bankBranchCode:ACH Routing Number"]))),
                    key: 'bankBranchCode',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_122 || (templateObject_122 = __makeTemplateObject([":@@product.details.investment.account.label.bankBranchCode2:FedWire Routing Number"], [":@@product.details.investment.account.label.bankBranchCode2:FedWire Routing Number"]))),
                    key: 'bankBranchCode2',
                },
                {
                    type: i1.AccountInfoPropertyType.ACCOUNT_STATE,
                    label: $localize(templateObject_123 || (templateObject_123 = __makeTemplateObject([":@@product.details.investment.account.label.accountStatus:Account Status"], [":@@product.details.investment.account.label.accountStatus:Account Status"]))),
                    key: 'state.state',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_124 || (templateObject_124 = __makeTemplateObject([":@@product.details.investment.account.label.financialInstitution.name:Bank Name"], [":@@product.details.investment.account.label.financialInstitution.name:Bank Name"]))),
                    key: 'financialInstitution.name',
                },
                {
                    type: i1.AccountInfoPropertyType.UPDATE_TIME,
                    label: $localize(templateObject_125 || (templateObject_125 = __makeTemplateObject([":@@product.details.investment.account.label.lastSyncDate:Last sync"], [":@@product.details.investment.account.label.lastSyncDate:Last sync"]))),
                    key: 'lastSyncDate',
                },
            ],
        },
        {
            title: $localize(templateObject_126 || (templateObject_126 = __makeTemplateObject([":@@product.details.investment.account.label.other.section:Other"], [":@@product.details.investment.account.label.other.section:Other"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.DATE,
                    label: $localize(templateObject_127 || (templateObject_127 = __makeTemplateObject([":@@product.details.investment.account.label.accountOpeningDate:Account Opening Date"], [":@@product.details.investment.account.label.accountOpeningDate:Account Opening Date"]))),
                    key: 'accountOpeningDate',
                    tooltip: $localize(templateObject_128 || (templateObject_128 = __makeTemplateObject([":@@product.details.investment.account.tooltip.accountOpeningDate:Account Opening Date"], [":@@product.details.investment.account.tooltip.accountOpeningDate:Account Opening Date"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.DATE,
                    label: $localize(templateObject_129 || (templateObject_129 = __makeTemplateObject([":@@product.details.investment.account.label.lastUpdateDate:Last Updated Date"], [":@@product.details.investment.account.label.lastUpdateDate:Last Updated Date"]))),
                    key: 'lastUpdateDate',
                    tooltip: $localize(templateObject_130 || (templateObject_130 = __makeTemplateObject([":@@product.details.investment.account.tooltip.lastUpdateDate:Last date of parameter update for the product"], [":@@product.details.investment.account.tooltip.lastUpdateDate:Last date of parameter update for the product"]))),
                },
            ],
        },
    ];
    var loanConfiguration = [
        {
            title: $localize(templateObject_131 || (templateObject_131 = __makeTemplateObject([":@@product.details.loan.label.general.section:General"], [":@@product.details.loan.label.general.section:General"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_132 || (templateObject_132 = __makeTemplateObject([":@@product.details.loan.label.productTypeName:Account Type"], [":@@product.details.loan.label.productTypeName:Account Type"]))),
                    key: 'productTypeName',
                },
                {
                    type: i1.AccountInfoPropertyType.ACCOUNT_NUMBER,
                    label: $localize(templateObject_133 || (templateObject_133 = __makeTemplateObject([":@@product.details.loan.label.account.number:Account Number"], [":@@product.details.loan.label.account.number:Account Number"]))),
                    key: 'BBAN',
                    numberType: 'bban',
                },
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_134 || (templateObject_134 = __makeTemplateObject([":@@product.details.loan.label.outstandingamount:Outstanding Amount"], [":@@product.details.loan.label.outstandingamount:Outstanding Amount"]))),
                    key: 'bookedBalance',
                    tooltip: $localize(templateObject_135 || (templateObject_135 = __makeTemplateObject([":@@product.details.loan.tooltip.outstandingamount:Outstanding Amount"], [":@@product.details.loan.tooltip.outstandingamount:Outstanding Amount"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_136 || (templateObject_136 = __makeTemplateObject([":@@product.details.loan.label.accountHolderNames:Account Owner(s)"], [":@@product.details.loan.label.accountHolderNames:Account Owner(s)"]))),
                    key: 'accountHolderNames',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_137 || (templateObject_137 = __makeTemplateObject([":@@product.details.loan.label.bankBranchCode:ACH Routing Number"], [":@@product.details.loan.label.bankBranchCode:ACH Routing Number"]))),
                    key: 'bankBranchCode',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_138 || (templateObject_138 = __makeTemplateObject([":@@product.details.loan.label.bankBranchCode2:FedWire Routing Number"], [":@@product.details.loan.label.bankBranchCode2:FedWire Routing Number"]))),
                    key: 'bankBranchCode2',
                },
                {
                    type: i1.AccountInfoPropertyType.ACCOUNT_STATE,
                    label: $localize(templateObject_139 || (templateObject_139 = __makeTemplateObject([":@@product.details.loan.label.accountStatus:Account Status"], [":@@product.details.loan.label.accountStatus:Account Status"]))),
                    key: 'state.state',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_140 || (templateObject_140 = __makeTemplateObject([":@@product.details.loan.label.financialInstitution.name:Bank Name"], [":@@product.details.loan.label.financialInstitution.name:Bank Name"]))),
                    key: 'financialInstitution.name',
                },
                {
                    type: i1.AccountInfoPropertyType.UPDATE_TIME,
                    label: $localize(templateObject_141 || (templateObject_141 = __makeTemplateObject([":@@product.details.loan.label.lastSyncDate:Last sync"], [":@@product.details.loan.label.lastSyncDate:Last sync"]))),
                    key: 'lastSyncDate',
                },
            ],
        },
        {
            title: $localize(templateObject_142 || (templateObject_142 = __makeTemplateObject([":@@product.details.loan.label.loan.details.section:Loan Details"], [":@@product.details.loan.label.loan.details.section:Loan Details"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_143 || (templateObject_143 = __makeTemplateObject([":@@product.details.loan.label.monthlyInstalmentAmount:Monthly Instalment Amount"], [":@@product.details.loan.label.monthlyInstalmentAmount:Monthly Instalment Amount"]))),
                    key: 'monthlyInstalmentAmount',
                    tooltip: $localize(templateObject_144 || (templateObject_144 = __makeTemplateObject([":@@product.details.loan.tooltip.monthlyInstalmentAmount:A fixed payment amount paid by a borrower to the bank at a specified date each calendar month"], [":@@product.details.loan.tooltip.monthlyInstalmentAmount:A fixed payment amount paid by a borrower to the bank at a specified date each calendar month"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_145 || (templateObject_145 = __makeTemplateObject([":@@product.details.loan.label.principalAmount:Principal Amount"], [":@@product.details.loan.label.principalAmount:Principal Amount"]))),
                    key: 'principalAmount',
                    tooltip: $localize(templateObject_146 || (templateObject_146 = __makeTemplateObject([":@@product.details.loan.tooltip.principalAmount:The amount that was (originally) contracted for the respective product"], [":@@product.details.loan.tooltip.principalAmount:The amount that was (originally) contracted for the respective product"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.PERCENT,
                    label: $localize(templateObject_147 || (templateObject_147 = __makeTemplateObject([":@@product.details.loan.label.accountInterestRate:Interest Rate"], [":@@product.details.loan.label.accountInterestRate:Interest Rate"]))),
                    key: 'accountInterestRate',
                    tooltip: $localize(templateObject_148 || (templateObject_148 = __makeTemplateObject([":@@product.details.loan.tooltip.accountInterestRate:The annualized cost of credit or debt-capital computed as the percentage ratio of interest to the principal"], [":@@product.details.loan.tooltip.accountInterestRate:The annualized cost of credit or debt-capital computed as the percentage ratio of interest to the principal"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.FREQUENCY,
                    label: $localize(templateObject_149 || (templateObject_149 = __makeTemplateObject([":@@product.details.loan.label.termNumber:Term"], [":@@product.details.loan.label.termNumber:Term"]))),
                    key: 'termNumber',
                    unitKey: 'termUnit',
                    tooltip: $localize(templateObject_150 || (templateObject_150 = __makeTemplateObject([":@@product.details.loan.tooltip.termNumber:The period of time and/or the interest rate arranged between Bank and customer"], [":@@product.details.loan.tooltip.termNumber:The period of time and/or the interest rate arranged between Bank and customer"]))),
                },
            ],
        },
        {
            title: $localize(templateObject_151 || (templateObject_151 = __makeTemplateObject([":@@product.details.loan.label.other.section:Other"], [":@@product.details.loan.label.other.section:Other"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.DATE,
                    label: $localize(templateObject_152 || (templateObject_152 = __makeTemplateObject([":@@product.details.loan.label.accountOpeningDate:Account Opening Date"], [":@@product.details.loan.label.accountOpeningDate:Account Opening Date"]))),
                    key: 'accountOpeningDate',
                    tooltip: $localize(templateObject_153 || (templateObject_153 = __makeTemplateObject([":@@product.details.loan.tooltip.accountOpeningDate:Account Opening Date"], [":@@product.details.loan.tooltip.accountOpeningDate:Account Opening Date"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.DATE,
                    label: $localize(templateObject_154 || (templateObject_154 = __makeTemplateObject([":@@product.details.loan.label.lastUpdateDate:Last Updated Date"], [":@@product.details.loan.label.lastUpdateDate:Last Updated Date"]))),
                    key: 'lastUpdateDate',
                    tooltip: $localize(templateObject_155 || (templateObject_155 = __makeTemplateObject([":@@product.details.loan.tooltip.lastUpdateDate:Last date of parameter update for the product"], [":@@product.details.loan.tooltip.lastUpdateDate:Last date of parameter update for the product"]))),
                },
            ],
        },
    ];
    var savingsAccountConfiguration = [
        {
            title: $localize(templateObject_156 || (templateObject_156 = __makeTemplateObject([":@@product.details.savings.account.label.general.section:General"], [":@@product.details.savings.account.label.general.section:General"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_157 || (templateObject_157 = __makeTemplateObject([":@@product.details.savings.account.label.productTypeName:Account Type"], [":@@product.details.savings.account.label.productTypeName:Account Type"]))),
                    key: 'productTypeName',
                },
                {
                    type: i1.AccountInfoPropertyType.ACCOUNT_NUMBER,
                    label: $localize(templateObject_158 || (templateObject_158 = __makeTemplateObject([":@@product.details.savings.account.label.account.number:Account Number"], [":@@product.details.savings.account.label.account.number:Account Number"]))),
                    key: 'BBAN',
                    numberType: 'bban',
                },
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_159 || (templateObject_159 = __makeTemplateObject([":@@product.details.savings.account.label.bookedBalance:Account Balance"], [":@@product.details.savings.account.label.bookedBalance:Account Balance"]))),
                    key: 'bookedBalance',
                    tooltip: $localize(templateObject_160 || (templateObject_160 = __makeTemplateObject([":@@product.details.savings.account.tooltip.bookedBalance:Account Balance"], [":@@product.details.savings.account.tooltip.bookedBalance:Account Balance"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_161 || (templateObject_161 = __makeTemplateObject([":@@product.details.savings.account.label.accountHolderNames:Account Owner(s)"], [":@@product.details.savings.account.label.accountHolderNames:Account Owner(s)"]))),
                    key: 'accountHolderNames',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_162 || (templateObject_162 = __makeTemplateObject([":@@product.details.savings.account.label.bankBranchCode:ACH Routing Number"], [":@@product.details.savings.account.label.bankBranchCode:ACH Routing Number"]))),
                    key: 'bankBranchCode',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_163 || (templateObject_163 = __makeTemplateObject([":@@product.details.savings.account.label.bankBranchCode2:FedWire Routing Number"], [":@@product.details.savings.account.label.bankBranchCode2:FedWire Routing Number"]))),
                    key: 'bankBranchCode2',
                },
                {
                    type: i1.AccountInfoPropertyType.ACCOUNT_STATE,
                    label: $localize(templateObject_164 || (templateObject_164 = __makeTemplateObject([":@@product.details.savings.account.label.accountStatus:Account Status"], [":@@product.details.savings.account.label.accountStatus:Account Status"]))),
                    key: 'state.state',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_165 || (templateObject_165 = __makeTemplateObject([":@@product.details.savings.account.label.financialInstitution.name:Bank Name"], [":@@product.details.savings.account.label.financialInstitution.name:Bank Name"]))),
                    key: 'financialInstitution.name',
                },
                {
                    type: i1.AccountInfoPropertyType.NUMBER,
                    label: $localize(templateObject_166 || (templateObject_166 = __makeTemplateObject([":@@product.details.savings.account.label.remainingPeriodicTransfers:Remaining Periodic Transfers"], [":@@product.details.savings.account.label.remainingPeriodicTransfers:Remaining Periodic Transfers"]))),
                    key: 'remainingPeriodicTransfers',
                    tooltip: $localize(templateObject_167 || (templateObject_167 = __makeTemplateObject([":@@product.details.savings.account.tooltip.remainingPeriodicTransfers:Remaining Periodic Transfers"], [":@@product.details.savings.account.tooltip.remainingPeriodicTransfers:Remaining Periodic Transfers"]))),
                    decimalPlaces: 0,
                },
                {
                    type: i1.AccountInfoPropertyType.UPDATE_TIME,
                    label: $localize(templateObject_168 || (templateObject_168 = __makeTemplateObject([":@@product.details.savings.account.label.lastSyncDate:Last sync"], [":@@product.details.savings.account.label.lastSyncDate:Last sync"]))),
                    key: 'lastSyncDate',
                },
            ],
        },
        {
            title: $localize(templateObject_169 || (templateObject_169 = __makeTemplateObject([":@@product.details.savings.account.label.interest.details.section:Interest Details"], [":@@product.details.savings.account.label.interest.details.section:Interest Details"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.PERCENT,
                    label: $localize(templateObject_170 || (templateObject_170 = __makeTemplateObject([":@@product.details.savings.account.label.accountInterestRate:Interest Rate"], [":@@product.details.savings.account.label.accountInterestRate:Interest Rate"]))),
                    key: 'accountInterestRate',
                    tooltip: $localize(templateObject_171 || (templateObject_171 = __makeTemplateObject([":@@product.details.savings.account.tooltip.accountInterestRate:The annualized cost of credit or debt-capital computed as the percentage ratio of interest to the principal"], [":@@product.details.savings.account.tooltip.accountInterestRate:The annualized cost of credit or debt-capital computed as the percentage ratio of interest to the principal"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_172 || (templateObject_172 = __makeTemplateObject([":@@product.details.savings.account.label.accruedInterest:Accrued Interest"], [":@@product.details.savings.account.label.accruedInterest:Accrued Interest"]))),
                    key: 'accruedInterest',
                    tooltip: $localize(templateObject_173 || (templateObject_173 = __makeTemplateObject([":@@product.details.savings.account.tooltip.accruedInterest:The interest that is earned (credit interest) or due (debit interest) but not settled yet"], [":@@product.details.savings.account.tooltip.accruedInterest:The interest that is earned (credit interest) or due (debit interest) but not settled yet"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_174 || (templateObject_174 = __makeTemplateObject([":@@product.details.savings.account.label.minimumRequiredBalance:Minimum Required Balance"], [":@@product.details.savings.account.label.minimumRequiredBalance:Minimum Required Balance"]))),
                    key: 'minimumRequiredBalance',
                    tooltip: $localize(templateObject_175 || (templateObject_175 = __makeTemplateObject([":@@product.details.savings.account.tooltip.minimumRequiredBalance:Minimum amount that a customer must have in an account in order to receive some sort of service, such as keeping the account open or receive interest"], [":@@product.details.savings.account.tooltip.minimumRequiredBalance:Minimum amount that a customer must have in an account in order to receive some sort of service, such as keeping the account open or receive interest"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_176 || (templateObject_176 = __makeTemplateObject([":@@product.details.savings.account.label.lastYearAccruedInterest:Last year\u2019s accrued interest"], [":@@product.details.savings.account.label.lastYearAccruedInterest:Last year\u2019s accrued interest"]))),
                    key: 'interestDetails.lastYearAccruedInterest',
                    tooltip: $localize(templateObject_177 || (templateObject_177 = __makeTemplateObject([":@@product.details.savings.account.tooltip.lastYearAccruedInterest:Last year\u2019s accrued interest"], [":@@product.details.savings.account.tooltip.lastYearAccruedInterest:Last year\u2019s accrued interest"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_178 || (templateObject_178 = __makeTemplateObject([":@@product.details.savings.account.label.dividendWithheldYTD:Interest/Dividend Withheld YTD"], [":@@product.details.savings.account.label.dividendWithheldYTD:Interest/Dividend Withheld YTD"]))),
                    key: 'interestDetails.dividendWithheldYTD',
                    tooltip: $localize(templateObject_179 || (templateObject_179 = __makeTemplateObject([":@@product.details.savings.account.tooltip.dividendWithheldYTD:Interest/Dividend Withheld YTD"], [":@@product.details.savings.account.tooltip.dividendWithheldYTD:Interest/Dividend Withheld YTD"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.PERCENT,
                    label: $localize(templateObject_180 || (templateObject_180 = __makeTemplateObject([":@@product.details.savings.account.label.annualPercentageYield:Annual Percentage Yield"], [":@@product.details.savings.account.label.annualPercentageYield:Annual Percentage Yield"]))),
                    key: 'interestDetails.annualPercentageYield',
                    tooltip: $localize(templateObject_181 || (templateObject_181 = __makeTemplateObject([":@@product.details.savings.account.tooltip.annualPercentageYield:Annual Percentage Yield"], [":@@product.details.savings.account.tooltip.annualPercentageYield:Annual Percentage Yield"]))),
                },
            ],
        },
        {
            title: $localize(templateObject_182 || (templateObject_182 = __makeTemplateObject([":@@product.details.savings.account.label.other.section:Other"], [":@@product.details.savings.account.label.other.section:Other"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.DATE,
                    label: $localize(templateObject_183 || (templateObject_183 = __makeTemplateObject([":@@product.details.savings.account.label.accountOpeningDate:Account Opening Date"], [":@@product.details.savings.account.label.accountOpeningDate:Account Opening Date"]))),
                    key: 'accountOpeningDate',
                    tooltip: $localize(templateObject_184 || (templateObject_184 = __makeTemplateObject([":@@product.details.savings.account.tooltip.accountOpeningDate:Account Opening Date"], [":@@product.details.savings.account.tooltip.accountOpeningDate:Account Opening Date"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.DATE,
                    label: $localize(templateObject_185 || (templateObject_185 = __makeTemplateObject([":@@product.details.savings.account.label.lastUpdateDate:Last Updated Date"], [":@@product.details.savings.account.label.lastUpdateDate:Last Updated Date"]))),
                    key: 'lastUpdateDate',
                    tooltip: $localize(templateObject_186 || (templateObject_186 = __makeTemplateObject([":@@product.details.savings.account.tooltip.lastUpdateDate:Last date of parameter update for the product"], [":@@product.details.savings.account.tooltip.lastUpdateDate:Last date of parameter update for the product"]))),
                },
            ],
        },
    ];
    var termDepositConfiguration = [
        {
            title: $localize(templateObject_187 || (templateObject_187 = __makeTemplateObject([":@@product.details.term.deposit.label.general.section:General"], [":@@product.details.term.deposit.label.general.section:General"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_188 || (templateObject_188 = __makeTemplateObject([":@@product.details.term.deposit.label.productTypeName:Account Type"], [":@@product.details.term.deposit.label.productTypeName:Account Type"]))),
                    key: 'productTypeName',
                },
                {
                    type: i1.AccountInfoPropertyType.ACCOUNT_NUMBER,
                    label: $localize(templateObject_189 || (templateObject_189 = __makeTemplateObject([":@@product.details.term.deposit.label.account.number:Account Number"], [":@@product.details.term.deposit.label.account.number:Account Number"]))),
                    key: 'BBAN',
                    numberType: 'bban',
                },
                {
                    type: i1.AccountInfoPropertyType.ACCOUNT_NUMBER,
                    label: $localize(templateObject_190 || (templateObject_190 = __makeTemplateObject([":@@product.details.term.deposit.label.iban:IBAN"], [":@@product.details.term.deposit.label.iban:IBAN"]))),
                    key: 'IBAN',
                    numberType: 'iban',
                },
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_191 || (templateObject_191 = __makeTemplateObject([":@@product.details.term.deposit.label.principalAmount:Principal Amount"], [":@@product.details.term.deposit.label.principalAmount:Principal Amount"]))),
                    key: 'principalAmount',
                    tooltip: $localize(templateObject_192 || (templateObject_192 = __makeTemplateObject([":@@product.details.term.deposit.tooltip.principalAmount:The amount that was (originally) contracted for the respective product"], [":@@product.details.term.deposit.tooltip.principalAmount:The amount that was (originally) contracted for the respective product"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_193 || (templateObject_193 = __makeTemplateObject([":@@product.details.term.deposit.label.bankBranchCode:ACH Routing Number"], [":@@product.details.term.deposit.label.bankBranchCode:ACH Routing Number"]))),
                    key: 'bankBranchCode',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_194 || (templateObject_194 = __makeTemplateObject([":@@product.details.term.deposit.label.bankBranchCode2:FedWire Routing Number"], [":@@product.details.term.deposit.label.bankBranchCode2:FedWire Routing Number"]))),
                    key: 'bankBranchCode2',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_195 || (templateObject_195 = __makeTemplateObject([":@@product.details.term.deposit.label.accountHolderNames:Account Owner(s)"], [":@@product.details.term.deposit.label.accountHolderNames:Account Owner(s)"]))),
                    key: 'accountHolderNames',
                },
                {
                    type: i1.AccountInfoPropertyType.ACCOUNT_STATE,
                    label: $localize(templateObject_196 || (templateObject_196 = __makeTemplateObject([":@@product.details.term.deposit.label.accountStatus:Account Status"], [":@@product.details.term.deposit.label.accountStatus:Account Status"]))),
                    key: 'state.state',
                },
                {
                    type: i1.AccountInfoPropertyType.TEXT,
                    label: $localize(templateObject_197 || (templateObject_197 = __makeTemplateObject([":@@product.details.term.deposit.label.financialInstitution.name:Bank Name"], [":@@product.details.term.deposit.label.financialInstitution.name:Bank Name"]))),
                    key: 'financialInstitution.name',
                },
                {
                    type: i1.AccountInfoPropertyType.UPDATE_TIME,
                    label: $localize(templateObject_198 || (templateObject_198 = __makeTemplateObject([":@@product.details.term.deposit.label.lastSyncDate:Last sync"], [":@@product.details.term.deposit.label.lastSyncDate:Last sync"]))),
                    key: 'lastSyncDate',
                },
            ],
        },
        {
            title: $localize(templateObject_199 || (templateObject_199 = __makeTemplateObject([":@@product.details.term.deposit.label.interest.details.section:Interest Details"], [":@@product.details.term.deposit.label.interest.details.section:Interest Details"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.PERCENT,
                    label: $localize(templateObject_200 || (templateObject_200 = __makeTemplateObject([":@@product.details.term.deposit.label.accountInterestRate:Interest Rate"], [":@@product.details.term.deposit.label.accountInterestRate:Interest Rate"]))),
                    key: 'accountInterestRate',
                    tooltip: $localize(templateObject_201 || (templateObject_201 = __makeTemplateObject([":@@product.details.term.deposit.tooltip.accountInterestRate:The annualized cost of credit or debt-capital computed as the percentage ratio of interest to the principal"], [":@@product.details.term.deposit.tooltip.accountInterestRate:The annualized cost of credit or debt-capital computed as the percentage ratio of interest to the principal"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_202 || (templateObject_202 = __makeTemplateObject([":@@product.details.term.deposit.label.accruedInterest:Accrued Interest"], [":@@product.details.term.deposit.label.accruedInterest:Accrued Interest"]))),
                    key: 'accruedInterest',
                    tooltip: $localize(templateObject_203 || (templateObject_203 = __makeTemplateObject([":@@product.details.term.deposit.tooltip.accruedInterest:The interest that is earned (credit interest) or due (debit interest) but not settled yet"], [":@@product.details.term.deposit.tooltip.accruedInterest:The interest that is earned (credit interest) or due (debit interest) but not settled yet"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.FREQUENCY,
                    label: $localize(templateObject_204 || (templateObject_204 = __makeTemplateObject([":@@product.details.term.deposit.label.interestPaymentFrequencyNumber:Interest Payment Frequency"], [":@@product.details.term.deposit.label.interestPaymentFrequencyNumber:Interest Payment Frequency"]))),
                    key: 'interestPaymentFrequencyNumber',
                    unitKey: 'interestPaymentFrequencyUnit',
                    tooltip: $localize(templateObject_205 || (templateObject_205 = __makeTemplateObject([":@@product.details.term.deposit.tooltip.interestPaymentFrequencyNumber:Number of times per year when the accumulated interest is settled to the product on a regular basis"], [":@@product.details.term.deposit.tooltip.interestPaymentFrequencyNumber:Number of times per year when the accumulated interest is settled to the product on a regular basis"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.RENEWAL_STATUS,
                    label: $localize(templateObject_206 || (templateObject_206 = __makeTemplateObject([":@@product.details.term.deposit.label.autoRenewalIndicator:Auto Renewal Indicator"], [":@@product.details.term.deposit.label.autoRenewalIndicator:Auto Renewal Indicator"]))),
                    key: 'autoRenewalIndicator',
                    tooltip: $localize(templateObject_207 || (templateObject_207 = __makeTemplateObject([":@@product.details.term.deposit.tooltip.autoRenewalIndicator:Indicates whether or not a product is to be continued after maturity automatically"], [":@@product.details.term.deposit.tooltip.autoRenewalIndicator:Indicates whether or not a product is to be continued after maturity automatically"]))),
                },
            ],
        },
        {
            title: $localize(templateObject_208 || (templateObject_208 = __makeTemplateObject([":@@product.details.term.deposit.label.maturity.details.section:Maturity Details"], [":@@product.details.term.deposit.label.maturity.details.section:Maturity Details"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.AMOUNT,
                    label: $localize(templateObject_209 || (templateObject_209 = __makeTemplateObject([":@@product.details.term.deposit.label.maturityAmount:Maturity Balance"], [":@@product.details.term.deposit.label.maturityAmount:Maturity Balance"]))),
                    key: 'maturityAmount',
                    tooltip: $localize(templateObject_210 || (templateObject_210 = __makeTemplateObject([":@@product.details.term.deposit.tooltip.maturityAmount:Amount payable at the end of a holding period of a product"], [":@@product.details.term.deposit.tooltip.maturityAmount:Amount payable at the end of a holding period of a product"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.FREQUENCY,
                    label: $localize(templateObject_211 || (templateObject_211 = __makeTemplateObject([":@@product.details.term.deposit.label.termNumber:Term"], [":@@product.details.term.deposit.label.termNumber:Term"]))),
                    key: 'termNumber',
                    unitKey: 'termUnit',
                    tooltip: $localize(templateObject_212 || (templateObject_212 = __makeTemplateObject([":@@product.details.term.deposit.tooltip.termNumber:The period of time and/or the interest rate arranged between Bank and customer"], [":@@product.details.term.deposit.tooltip.termNumber:The period of time and/or the interest rate arranged between Bank and customer"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.DATE,
                    label: $localize(templateObject_213 || (templateObject_213 = __makeTemplateObject([":@@product.details.term.deposit.label.startDate:Start Date"], [":@@product.details.term.deposit.label.startDate:Start Date"]))),
                    key: 'startDate',
                    tooltip: $localize(templateObject_214 || (templateObject_214 = __makeTemplateObject([":@@product.details.term.deposit.tooltip.startDate:Start Date"], [":@@product.details.term.deposit.tooltip.startDate:Start Date"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.DATE,
                    label: $localize(templateObject_215 || (templateObject_215 = __makeTemplateObject([":@@product.details.term.deposit.label.maturityDate:Maturity Date"], [":@@product.details.term.deposit.label.maturityDate:Maturity Date"]))),
                    key: 'maturityDate',
                    tooltip: $localize(templateObject_216 || (templateObject_216 = __makeTemplateObject([":@@product.details.term.deposit.tooltip.maturityDate:End term of a holding period"], [":@@product.details.term.deposit.tooltip.maturityDate:End term of a holding period"]))),
                },
            ],
        },
        {
            title: $localize(templateObject_217 || (templateObject_217 = __makeTemplateObject([":@@product.details.term.deposit.label.other.section:Other"], [":@@product.details.term.deposit.label.other.section:Other"]))),
            properties: [
                {
                    type: i1.AccountInfoPropertyType.DATE,
                    label: $localize(templateObject_218 || (templateObject_218 = __makeTemplateObject([":@@product.details.term.deposit.label.accountOpeningDate:Account Opening Date"], [":@@product.details.term.deposit.label.accountOpeningDate:Account Opening Date"]))),
                    key: 'accountOpeningDate',
                    tooltip: $localize(templateObject_219 || (templateObject_219 = __makeTemplateObject([":@@product.details.term.deposit.tooltip.accountOpeningDate:Account Opening Date"], [":@@product.details.term.deposit.tooltip.accountOpeningDate:Account Opening Date"]))),
                },
                {
                    type: i1.AccountInfoPropertyType.DATE,
                    label: $localize(templateObject_220 || (templateObject_220 = __makeTemplateObject([":@@product.details.term.deposit.label.lastUpdateDate:Last Updated Date"], [":@@product.details.term.deposit.label.lastUpdateDate:Last Updated Date"]))),
                    key: 'lastUpdateDate',
                    tooltip: $localize(templateObject_221 || (templateObject_221 = __makeTemplateObject([":@@product.details.term.deposit.tooltip.lastUpdateDate:Last date of parameter update for the product"], [":@@product.details.term.deposit.tooltip.lastUpdateDate:Last date of parameter update for the product"]))),
                },
            ],
        },
    ];
    /**
     * Default configuration for account info
     */
    var defaultAccountInfoConfiguration = (_a = {
            default: defaultConfiguration
        },
        _a[i4.ProductKindUri.CREDIT_CARD] = creditCardConfiguration,
        _a[i4.ProductKindUri.CURRENT_ACCOUNT] = currentAccountConfiguration,
        _a[i4.ProductKindUri.DEBIT_CARD] = debitCardConfiguration,
        _a[i4.ProductKindUri.INVESTMENT_ACCOUNT] = investmentAccountConfiguration,
        _a[i4.ProductKindUri.LOAN] = loanConfiguration,
        _a[i4.ProductKindUri.SAVINGS_ACCOUNT] = savingsAccountConfiguration,
        _a[i4.ProductKindUri.TERM_DEPOSIT] = termDepositConfiguration,
        _a);
    var templateObject_1$2, templateObject_2$2, templateObject_3, templateObject_4, templateObject_5, templateObject_6, templateObject_7, templateObject_8, templateObject_9, templateObject_10, templateObject_11, templateObject_12, templateObject_13, templateObject_14, templateObject_15, templateObject_16, templateObject_17, templateObject_18, templateObject_19, templateObject_20, templateObject_21, templateObject_22, templateObject_23, templateObject_24, templateObject_25, templateObject_26, templateObject_27, templateObject_28, templateObject_29, templateObject_30, templateObject_31, templateObject_32, templateObject_33, templateObject_34, templateObject_35, templateObject_36, templateObject_37, templateObject_38, templateObject_39, templateObject_40, templateObject_41, templateObject_42, templateObject_43, templateObject_44, templateObject_45, templateObject_46, templateObject_47, templateObject_48, templateObject_49, templateObject_50, templateObject_51, templateObject_52, templateObject_53, templateObject_54, templateObject_55, templateObject_56, templateObject_57, templateObject_58, templateObject_59, templateObject_60, templateObject_61, templateObject_62, templateObject_63, templateObject_64, templateObject_65, templateObject_66, templateObject_67, templateObject_68, templateObject_69, templateObject_70, templateObject_71, templateObject_72, templateObject_73, templateObject_74, templateObject_75, templateObject_76, templateObject_77, templateObject_78, templateObject_79, templateObject_80, templateObject_81, templateObject_82, templateObject_83, templateObject_84, templateObject_85, templateObject_86, templateObject_87, templateObject_88, templateObject_89, templateObject_90, templateObject_91, templateObject_92, templateObject_93, templateObject_94, templateObject_95, templateObject_96, templateObject_97, templateObject_98, templateObject_99, templateObject_100, templateObject_101, templateObject_102, templateObject_103, templateObject_104, templateObject_105, templateObject_106, templateObject_107, templateObject_108, templateObject_109, templateObject_110, templateObject_111, templateObject_112, templateObject_113, templateObject_114, templateObject_115, templateObject_116, templateObject_117, templateObject_118, templateObject_119, templateObject_120, templateObject_121, templateObject_122, templateObject_123, templateObject_124, templateObject_125, templateObject_126, templateObject_127, templateObject_128, templateObject_129, templateObject_130, templateObject_131, templateObject_132, templateObject_133, templateObject_134, templateObject_135, templateObject_136, templateObject_137, templateObject_138, templateObject_139, templateObject_140, templateObject_141, templateObject_142, templateObject_143, templateObject_144, templateObject_145, templateObject_146, templateObject_147, templateObject_148, templateObject_149, templateObject_150, templateObject_151, templateObject_152, templateObject_153, templateObject_154, templateObject_155, templateObject_156, templateObject_157, templateObject_158, templateObject_159, templateObject_160, templateObject_161, templateObject_162, templateObject_163, templateObject_164, templateObject_165, templateObject_166, templateObject_167, templateObject_168, templateObject_169, templateObject_170, templateObject_171, templateObject_172, templateObject_173, templateObject_174, templateObject_175, templateObject_176, templateObject_177, templateObject_178, templateObject_179, templateObject_180, templateObject_181, templateObject_182, templateObject_183, templateObject_184, templateObject_185, templateObject_186, templateObject_187, templateObject_188, templateObject_189, templateObject_190, templateObject_191, templateObject_192, templateObject_193, templateObject_194, templateObject_195, templateObject_196, templateObject_197, templateObject_198, templateObject_199, templateObject_200, templateObject_201, templateObject_202, templateObject_203, templateObject_204, templateObject_205, templateObject_206, templateObject_207, templateObject_208, templateObject_209, templateObject_210, templateObject_211, templateObject_212, templateObject_213, templateObject_214, templateObject_215, templateObject_216, templateObject_217, templateObject_218, templateObject_219, templateObject_220, templateObject_221;

    // eslint-disable-next-line
    var AccountsTransactionsJourneyConfigurationToken = new i0.InjectionToken('AccountsTransactionsJourneyConfiguration injection token');
    var configDefaults = {
        apiKey: '',
        itemsPerPage: 10,
        transactionsMaxPages: 3,
        paginationType: 'load-more',
        pendingOnTop: false,
        transactionsFilterParameters: '',
        transactionsSortOptions: 'bookingDate DESC',
        showControls: true,
        showChangeCategory: true,
        showCheckImages: false,
        getOrPostEndpoints: i3.EndpointType.GET_REQUEST,
        enableDisputeAndInquiry: true,
        disputeEligibilityDays: 120,
        disputeTransactionTypes: 'Credit/Debit Card',
        disputeByBillingStatus: true,
        disputeTopicId: '',
        inquireTopicId: '',
        availableFiltersConfig: {
            'Current Account': [
                i3.AvailableFilterEnum.Date,
                i3.AvailableFilterEnum.CreditDebitIndicator,
                i3.AvailableFilterEnum.Amount,
                i3.AvailableFilterEnum.Types,
                i3.AvailableFilterEnum.BillingStatus,
                i3.AvailableFilterEnum.CheckSerialNumber,
            ],
            'Savings Account': [
                i3.AvailableFilterEnum.Date,
                i3.AvailableFilterEnum.CreditDebitIndicator,
                i3.AvailableFilterEnum.Amount,
                i3.AvailableFilterEnum.Types,
                i3.AvailableFilterEnum.BillingStatus,
                i3.AvailableFilterEnum.CheckSerialNumber,
            ],
            'Credit Card': [
                i3.AvailableFilterEnum.Date,
                i3.AvailableFilterEnum.CreditDebitIndicator,
                i3.AvailableFilterEnum.Amount,
                i3.AvailableFilterEnum.Types,
                i3.AvailableFilterEnum.BillingStatus,
            ],
            Loan: [i3.AvailableFilterEnum.Date, i3.AvailableFilterEnum.CreditDebitIndicator, i3.AvailableFilterEnum.Amount],
            'Term Deposit': [i3.AvailableFilterEnum.Date, i3.AvailableFilterEnum.CreditDebitIndicator, i3.AvailableFilterEnum.Amount],
            'Debit Card': [i3.AvailableFilterEnum.Date, i3.AvailableFilterEnum.CreditDebitIndicator, i3.AvailableFilterEnum.Amount],
            'Investment Account': [
                i3.AvailableFilterEnum.Date,
                i3.AvailableFilterEnum.CreditDebitIndicator,
                i3.AvailableFilterEnum.Amount,
            ],
        },
        showAccountIcons: true,
        enableManageAccounts: true,
        enableTransferButton: true,
        productKindsWithGraphicalRepresentation: i4.ProductKindUri.CREDIT_CARD,
        productKindsWithExternalDetailsPage: [],
        creditLimitViewMode: 'emptied',
        accountAliasDisplayLevel: i4.AccountAliasDisplayingLevel.USER,
        accountInfoProperties: defaultAccountInfoConfiguration,
        arrangementViewsName: undefined,
    };
    var AccountsTransactionsJourneyService = /** @class */ (function () {
        function AccountsTransactionsJourneyService(config, itemModel) {
            var _this = this;
            this.itemModel = itemModel;
            config = config || {};
            this._config = Object.assign(Object.assign({}, configDefaults), config);
            // update the API key from ItemModel if it was not provided from a static config
            if (this.itemModel && (config.apiKey === undefined || config.showCheckImages === undefined)) {
                rxjs.combineLatest([
                    this.itemModel.property('apiKey', this._config.apiKey),
                    this.itemModel.property('showCheckImages', this._config.showCheckImages),
                ])
                    .subscribe({
                    next: function (values) {
                        if (config.apiKey === undefined) {
                            _this._config.apiKey = values[0];
                        }
                        if (config.showCheckImages === undefined) {
                            _this._config.showCheckImages = !!values[1];
                        }
                    },
                })
                    .unsubscribe();
            }
        }
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "defaults", {
            get: function () {
                return configDefaults;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "apiKey", {
            get: function () {
                return this._config.apiKey;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "itemsPerPage", {
            get: function () {
                return this._config.itemsPerPage;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "transactionsMaxPages", {
            get: function () {
                return this._config.transactionsMaxPages;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "paginationType", {
            get: function () {
                return this._config.paginationType;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "pendingOnTop", {
            get: function () {
                return this._config.pendingOnTop;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "transactionsFilterParameters", {
            get: function () {
                return this._config.transactionsFilterParameters;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "transactionsSortOptions", {
            get: function () {
                return this._config.transactionsSortOptions;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "showControls", {
            get: function () {
                return this._config.showControls;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "showChangeCategory", {
            get: function () {
                return this._config.showChangeCategory;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "showCheckImages", {
            get: function () {
                return this._config.showCheckImages;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "getOrPostEndpoints", {
            get: function () {
                return this._config.getOrPostEndpoints;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "enableDisputeAndInquiry", {
            get: function () {
                return this._config.enableDisputeAndInquiry;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "disputeEligibilityDays", {
            get: function () {
                return this._config.disputeEligibilityDays;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "disputeTransactionTypes", {
            get: function () {
                return this._config.disputeTransactionTypes;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "disputeByBillingStatus", {
            get: function () {
                return this._config.disputeByBillingStatus;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "disputeTopicId", {
            get: function () {
                return this._config.disputeTopicId;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "inquireTopicId", {
            get: function () {
                return this._config.inquireTopicId;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "availableFiltersConfig", {
            get: function () {
                return this._config.availableFiltersConfig;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "enableManageAccounts", {
            get: function () {
                return this._config.enableManageAccounts;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "showAccountIcons", {
            get: function () {
                return this._config.showAccountIcons;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "enableTransferButton", {
            get: function () {
                return this._config.enableTransferButton;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "productKindsWithGraphicalRepresentation", {
            get: function () {
                return this._config.productKindsWithGraphicalRepresentation;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "productKindsWithExternalDetailsPage", {
            get: function () {
                return this._config.productKindsWithExternalDetailsPage;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "creditLimitViewMode", {
            get: function () {
                return this._config.creditLimitViewMode;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "accountAliasDisplayLevel", {
            get: function () {
                return this._config.accountAliasDisplayLevel;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "accountInfoProperties", {
            /**
             * Account info properties configuration
             */
            get: function () {
                return this._config.accountInfoProperties;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(AccountsTransactionsJourneyService.prototype, "arrangementViewsName", {
            get: function () {
                return this._config.arrangementViewsName;
            },
            enumerable: false,
            configurable: true
        });
        return AccountsTransactionsJourneyService;
    }());
    /** @nocollapse */ AccountsTransactionsJourneyService.fac = i0__namespace.ngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountsTransactionsJourneyService, deps: [{ token: AccountsTransactionsJourneyConfigurationToken, optional: true }, { token: i1__namespace.ItemModel, optional: true }], target: i0__namespace.FactoryTarget.Injectable });
    /** @nocollapse */ AccountsTransactionsJourneyService.prov = i0__namespace.ngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountsTransactionsJourneyService });
    i0__namespace.ngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountsTransactionsJourneyService, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () {
            return [{ type: undefined, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.Inject,
                            args: [AccountsTransactionsJourneyConfigurationToken]
                        }] }, { type: i1__namespace.ItemModel, decorators: [{
                            type: i0.Optional
                        }] }];
        } });

    var AccountsTransactionsJourneyComponent = /** @class */ (function () {
        function AccountsTransactionsJourneyComponent() {
        }
        return AccountsTransactionsJourneyComponent;
    }());
    /** @nocollapse */ AccountsTransactionsJourneyComponent.fac = i0__namespace.ngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountsTransactionsJourneyComponent, deps: [], target: i0__namespace.FactoryTarget.Component });
    /** @nocollapse */ AccountsTransactionsJourneyComponent.cmp = i0__namespace.ngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AccountsTransactionsJourneyComponent, selector: "bb-accounts-transactions-journey", providers: [AccountsTransactionsJourneyService], ngImport: i0__namespace, template: '<router-outlet></router-outlet>', isInline: true, directives: [{ type: i2__namespace.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }] });
    i0__namespace.ngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountsTransactionsJourneyComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-accounts-transactions-journey',
                        template: '<router-outlet></router-outlet>',
                        providers: [AccountsTransactionsJourneyService],
                    }]
            }] });

    /**
     * Accounts communication service API
     */
    var AccountsCommunicationService = /** @class */ (function () {
        function AccountsCommunicationService() {
        }
        return AccountsCommunicationService;
    }());

    var AccountsListBaseComponent = /** @class */ (function () {
        /**
         * @internal
         * @param config
         * @param router
         * @param route
         * @param accountsCommunicationService
         */
        function AccountsListBaseComponent(
        /**
         * Journey configuration
         */
        config, router, route, accountsCommunicationService) {
            this.config = config;
            this.router = router;
            this.route = route;
            this.accountsCommunicationService = accountsCommunicationService;
            /**
             * Label for accounts list title
             */
            this.myAccounts = $localize(templateObject_1$1 || (templateObject_1$1 = __makeTemplateObject([":My accounts heading|my accounts heading@@accounts-transactions-journey.accounts-list.heading.myAccounts:My Accounts"], [":My accounts heading|my accounts heading@@accounts-transactions-journey.accounts-list.heading.myAccounts:My Accounts"])));
            /**
             * Label for manage accounts navigation button
             */
            this.manageAccounts = $localize(templateObject_2$1 || (templateObject_2$1 = __makeTemplateObject([":Manage accounts button title|manage accounts button title@@accounts-transactions-journey.accounts-list.buttonTitle.manageAccounts:Manage Accounts"], [":Manage accounts button title|manage accounts button title@@accounts-transactions-journey.accounts-list.buttonTitle.manageAccounts:Manage Accounts"])));
        }
        /**
         * Navigates to selected product details
         *
         * @param product
         */
        AccountsListBaseComponent.prototype.onProductSelect = function (product) {
            var id = product.id;
            var kind = this.getProductKind(product);
            if (this.accountsCommunicationService && this.productKindsWithExternalDetailsPage.includes(kind)) {
                this.accountsCommunicationService.navigateToExternalAccountDetails({
                    id: id,
                    kind: kind,
                });
            }
            else {
                this.navigateToAccountDetails(id);
            }
        };
        /**
         * Navigates to account details by id
         *
         * @param id
         */
        AccountsListBaseComponent.prototype.onAccountSelect = function (id) {
            this.navigateToAccountDetails(id);
        };
        /**
         * Navigates to account management
         */
        AccountsListBaseComponent.prototype.navigateToAccountManagement = function () {
            this.router.navigate(['../manage'], { relativeTo: this.route });
        };
        AccountsListBaseComponent.prototype.navigateToAccountDetails = function (selectedAccount) {
            this.router.navigate(['../transactions', { selectedAccount: selectedAccount }], {
                relativeTo: this.route,
            });
        };
        Object.defineProperty(AccountsListBaseComponent.prototype, "productKindsWithExternalDetailsPage", {
            get: function () {
                return Array.isArray(this.config.productKindsWithExternalDetailsPage)
                    ? this.config.productKindsWithExternalDetailsPage
                    : [this.config.productKindsWithExternalDetailsPage];
            },
            enumerable: false,
            configurable: true
        });
        AccountsListBaseComponent.prototype.getProductKind = function (product) {
            switch (product.kind) {
                case 'currentAccounts':
                    return i4.ProductKindUri.CURRENT_ACCOUNT;
                case 'savingsAccounts':
                    return i4.ProductKindUri.SAVINGS_ACCOUNT;
                case 'termDeposits':
                    return i4.ProductKindUri.TERM_DEPOSIT;
                case 'loans':
                    return i4.ProductKindUri.LOAN;
                case 'creditCards':
                    return i4.ProductKindUri.CREDIT_CARD;
                case 'debitCards':
                    return i4.ProductKindUri.DEBIT_CARD;
                case 'investmentAccounts':
                    return i4.ProductKindUri.INVESTMENT_ACCOUNT;
                default:
                    return product.kind;
            }
        };
        return AccountsListBaseComponent;
    }());
    /** @nocollapse */ AccountsListBaseComponent.fac = i0__namespace.ngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountsListBaseComponent, deps: [{ token: AccountsTransactionsJourneyService }, { token: i2__namespace.Router }, { token: i2__namespace.ActivatedRoute }, { token: AccountsCommunicationService, optional: true }], target: i0__namespace.FactoryTarget.Directive });
    /** @nocollapse */ AccountsListBaseComponent.dir = i0__namespace.ngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: AccountsListBaseComponent, ngImport: i0__namespace });
    i0__namespace.ngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountsListBaseComponent, decorators: [{
                type: i0.Directive
            }], ctorParameters: function () {
            return [{ type: AccountsTransactionsJourneyService }, { type: i2__namespace.Router }, { type: i2__namespace.ActivatedRoute }, { type: AccountsCommunicationService, decorators: [{
                            type: i0.Optional
                        }] }];
        } });
    var templateObject_1$1, templateObject_2$1;

    var AccountsListComponent = /** @class */ (function (_super) {
        __extends(AccountsListComponent, _super);
        function AccountsListComponent() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return AccountsListComponent;
    }(AccountsListBaseComponent));
    /** @nocollapse */ AccountsListComponent.fac = i0__namespace.ngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountsListComponent, deps: null, target: i0__namespace.FactoryTarget.Component });
    /** @nocollapse */ AccountsListComponent.cmp = i0__namespace.ngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AccountsListComponent, selector: "bb-accounts-list", usesInheritance: true, ngImport: i0__namespace, template: "<ng-container *ngIf=\"config.enableManageAccounts; else titleOnly\">\n  <div class=\"bb-heading-widget heading-widget px-0 flex-row align-items-center\">\n    <bb-header-ui headingType=\"h1\" headingClasses=\"bb-heading-widget__heading\" [heading]=\"myAccounts\"></bb-header-ui>\n    <button\n      (click)=\"navigateToAccountManagement()\"\n      class=\"bb-heading-widget__button btn btn-link w-auto bb-button bb-button--xs-hide-text m-0\"\n    >\n      <i aria-hidden=\"true\" class=\"bb-icon bb-icon-settings\"></i>\n      <span>{{ manageAccounts }}</span>\n    </button>\n  </div>\n</ng-container>\n\n<ng-template #titleOnly>\n  <div class=\"bb-heading-widget heading-widget\">\n    <bb-header-ui headingType=\"h1\" headingClasses=\"bb-heading-widget__heading\" [heading]=\"myAccounts\"></bb-header-ui>\n  </div>\n</ng-template>\n\n<bb-product-summary-list-widget\n  [showIcons]=\"config.showAccountIcons\"\n  (selectProduct)=\"onProductSelect($event)\"\n></bb-product-summary-list-widget>\n", components: [{ type: i1__namespace$1.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i2__namespace$1.ProductSummaryListWidgetComponent, selector: "bb-product-summary-list-widget" }], directives: [{ type: i8__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
    i0__namespace.ngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountsListComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-accounts-list',
                        templateUrl: 'accounts-list.component.html',
                    }]
            }] });

    var TransactionsHelperService = /** @class */ (function () {
        function TransactionsHelperService() {
            var _this = this;
            this.transactionIdSubject = new rxjs.BehaviorSubject('');
            this.transactionsSubject = new rxjs.BehaviorSubject(undefined);
            this.onCloseDetailSubject = new rxjs.Subject();
            this.currentTransaction = this.transactionIdSubject.pipe(operators.mergeMap(function (id) { return _this.transactionsSubject.pipe(operators.map(function (transaction) { return transaction === null || transaction === void 0 ? void 0 : transaction.find(function (x) { return id === x.id; }); })); }));
            this.onCloseDetail = this.onCloseDetailSubject.asObservable();
        }
        TransactionsHelperService.prototype.setCloseDetail = function () {
            this.onCloseDetailSubject.next(true);
        };
        TransactionsHelperService.prototype.setTransactions = function (transactions) {
            this.transactionsSubject.next(transactions);
        };
        TransactionsHelperService.prototype.setTransactionId = function (transactionId) {
            this.transactionIdSubject.next(transactionId);
        };
        return TransactionsHelperService;
    }());
    /** @nocollapse */ TransactionsHelperService.fac = i0__namespace.ngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: TransactionsHelperService, deps: [], target: i0__namespace.FactoryTarget.Injectable });
    /** @nocollapse */ TransactionsHelperService.prov = i0__namespace.ngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: TransactionsHelperService });
    i0__namespace.ngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: TransactionsHelperService, decorators: [{
                type: i0.Injectable
            }] });

    var TransactionsListComponent = /** @class */ (function () {
        function TransactionsListComponent(config, router, route, shared) {
            var _this = this;
            this.config = config;
            this.router = router;
            this.route = route;
            this.shared = shared;
            this.shared.onCloseDetail.subscribe(function (event) {
                if (event) {
                    _this.router.navigate(['../list'], { relativeTo: _this.route });
                }
            });
        }
        TransactionsListComponent.prototype.getTransaction = function (transaction) {
            this.shared.setTransactionId(transaction.id || '');
            this.router.navigate(['./detail', { transactionId: transaction.id }], {
                relativeTo: this.route,
            });
        };
        TransactionsListComponent.prototype.transactionsLoaded = function (transactionsList) {
            if (transactionsList) {
                this.shared.setTransactions(transactionsList.items);
            }
        };
        TransactionsListComponent.prototype.ngOnInit = function () {
            if (this.route.snapshot.children[0] && this.route.snapshot.children[0].params['transactionId']) {
                this.shared.setTransactionId(this.route.snapshot.children[0].params['transactionId']);
            }
        };
        return TransactionsListComponent;
    }());
    /** @nocollapse */ TransactionsListComponent.fac = i0__namespace.ngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: TransactionsListComponent, deps: [{ token: AccountsTransactionsJourneyService }, { token: i2__namespace.Router }, { token: i2__namespace.ActivatedRoute }, { token: TransactionsHelperService }], target: i0__namespace.FactoryTarget.Component });
    /** @nocollapse */ TransactionsListComponent.cmp = i0__namespace.ngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: TransactionsListComponent, selector: "bb-transactions-list-view", ngImport: i0__namespace, template: "\n    <bb-transactions-list-widget\n      [googleAPIKey]=\"config.apiKey\"\n      [pageSize]=\"config.itemsPerPage\"\n      [maxNavPages]=\"config.transactionsMaxPages\"\n      [paginationType]=\"config.paginationType\"\n      [showPendingTransactionsOnTop]=\"config.pendingOnTop\"\n      [initialFilterOptions]=\"config.transactionsFilterParameters\"\n      [initialSortOptions]=\"config.transactionsSortOptions\"\n      [showControls]=\"true\"\n      [showChangeCategory]=\"true\"\n      [showCheckImage]=\"config.showCheckImages\"\n      [getOrPostEndpoints]=\"config.getOrPostEndpoints\"\n      [enableDisputeAndInquiry]=\"config.enableDisputeAndInquiry\"\n      [disputeEligibilityDays]=\"config.disputeEligibilityDays\"\n      [disputeTransactionTypes]=\"config.disputeTransactionTypes\"\n      [disputeByBillingStatus]=\"config.disputeByBillingStatus\"\n      [inquireTopicId]=\"config.inquireTopicId\"\n      [disputeTopicId]=\"config.disputeTopicId\"\n      [availableFiltersConfig]=\"config.availableFiltersConfig\"\n      (transactionSelected)=\"getTransaction($event)\"\n      (transactionsLoaded)=\"transactionsLoaded($event)\"\n    ></bb-transactions-list-widget>\n    <router-outlet></router-outlet>\n  ", isInline: true, components: [{ type: i4__namespace.TransactionsListWidgetComponent, selector: "bb-transactions-list-widget" }], directives: [{ type: i2__namespace.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }] });
    i0__namespace.ngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: TransactionsListComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-transactions-list-view',
                        template: "\n    <bb-transactions-list-widget\n      [googleAPIKey]=\"config.apiKey\"\n      [pageSize]=\"config.itemsPerPage\"\n      [maxNavPages]=\"config.transactionsMaxPages\"\n      [paginationType]=\"config.paginationType\"\n      [showPendingTransactionsOnTop]=\"config.pendingOnTop\"\n      [initialFilterOptions]=\"config.transactionsFilterParameters\"\n      [initialSortOptions]=\"config.transactionsSortOptions\"\n      [showControls]=\"true\"\n      [showChangeCategory]=\"true\"\n      [showCheckImage]=\"config.showCheckImages\"\n      [getOrPostEndpoints]=\"config.getOrPostEndpoints\"\n      [enableDisputeAndInquiry]=\"config.enableDisputeAndInquiry\"\n      [disputeEligibilityDays]=\"config.disputeEligibilityDays\"\n      [disputeTransactionTypes]=\"config.disputeTransactionTypes\"\n      [disputeByBillingStatus]=\"config.disputeByBillingStatus\"\n      [inquireTopicId]=\"config.inquireTopicId\"\n      [disputeTopicId]=\"config.disputeTopicId\"\n      [availableFiltersConfig]=\"config.availableFiltersConfig\"\n      (transactionSelected)=\"getTransaction($event)\"\n      (transactionsLoaded)=\"transactionsLoaded($event)\"\n    ></bb-transactions-list-widget>\n    <router-outlet></router-outlet>\n  ",
                    }]
            }], ctorParameters: function () { return [{ type: AccountsTransactionsJourneyService }, { type: i2__namespace.Router }, { type: i2__namespace.ActivatedRoute }, { type: TransactionsHelperService }]; } });

    /**
     * An operator that switches to another stream, but emits data from original stream before switching
     * - subscribes to "source" stream,
     * - emits data, when "source" stream is resolved,
     * - then creates a subscription to a "mapSource" using the previously published data as an input,
     * - when "mapSource" is resolved the operator publishes an event with "mapSource" data.
     */
    var publishSwitch = function (mapSource) { return function (source) { return new rxjs.Observable(function (subscriber) {
        var subscriptions = [];
        subscriptions.push(source.subscribe(function (sourceData) {
            subscriber.next(sourceData);
            if (sourceData) {
                subscriptions.push(mapSource(sourceData).subscribe(subscriber));
            }
        }, subscriber.error));
        return function () { return subscriptions.forEach(function (subscription) { return subscription.unsubscribe(); }); };
    }); }; };

    /**
     * Quick actions communication API
     */
    var AccountsPaymentsCommunication = /** @class */ (function () {
        function AccountsPaymentsCommunication() {
        }
        return AccountsPaymentsCommunication;
    }());

    var AccountsDetailsTabComponent = /** @class */ (function () {
        function AccountsDetailsTabComponent(accountAliasDisplayingLevel, config, accountsPaymentsCommunication, route, router, accountsService, externalInfoService) {
            var _this = this;
            this.accountAliasDisplayingLevel = accountAliasDisplayingLevel;
            this.config = config;
            this.accountsPaymentsCommunication = accountsPaymentsCommunication;
            this.route = route;
            this.router = router;
            this.accountsService = accountsService;
            this.externalInfoService = externalInfoService;
            /**
             * Arrangement item to be displayed
             */
            this.accountArrangementItem$ = this.accountsService
                .getAccountById(this.getSelectedAccount())
                .pipe(publishSwitch(function (source) { return _this.externalInfoService.getProductsWithExternalInfo(source); }));
            /**
             * Returns `true` if `AccountsPaymentsCommunication` was provided
             */
            this.isCommunicationProvided = false;
            var routeConfig = this.route.snapshot.routeConfig;
            this.tabs = (routeConfig === null || routeConfig === void 0 ? void 0 : routeConfig.children) ? this.initializeTabs(routeConfig) : [];
            this.accountAliasDisplayingLevel = accountAliasDisplayingLevel || config.accountAliasDisplayLevel;
            this.isCommunicationProvided = !!accountsPaymentsCommunication;
        }
        /**
         * Specifies whether `repay` action can be initiated for provided AccountArrangementItem
         *
         * @param accountArrangementItem
         */
        AccountsDetailsTabComponent.prototype.canInitiateRepay = function (accountArrangementItem) {
            return this.isQuickActionsAllowed(accountArrangementItem) && accountArrangementItem.creditAccount === true;
        };
        /**
         * Specifies whether `cash advance` action can be initiated for provided AccountArrangementItem
         *
         * @param accountArrangementItem
         */
        AccountsDetailsTabComponent.prototype.canInitiateCashInAdvance = function (accountArrangementItem) {
            return this.isQuickActionsAllowed(accountArrangementItem) && accountArrangementItem.debitAccount === true;
        };
        /**
         * Event handler for the `repayInitiated` event of `bb-account-graphical-header`
         * Initiates the event by running the corresponding method from communication protocol
         *
         * @param arrangementId
         */
        AccountsDetailsTabComponent.prototype.onRepayInitiated = function (arrangementId) {
            if (arrangementId && this.accountsPaymentsCommunication) {
                this.accountsPaymentsCommunication.repayEvent(arrangementId);
            }
        };
        /**
         * Event handler for the `cashAdvanceInitiated` event of `bb-account-graphical-header`
         * Initiates the event by running the corresponding method from communication protocol
         *
         * @param arrangementId
         */
        AccountsDetailsTabComponent.prototype.onCashAdvanceInitiated = function (arrangementId) {
            if (arrangementId && this.accountsPaymentsCommunication) {
                this.accountsPaymentsCommunication.cashAdvanceEvent(arrangementId);
            }
        };
        /**
         * Navigates back to `My Accounts`
         */
        AccountsDetailsTabComponent.prototype.onNavigateBack = function () {
            this.router.navigate(['..'], { relativeTo: this.route });
        };
        AccountsDetailsTabComponent.prototype.isQuickActionsAllowed = function (accountArrangementItem) {
            return this.isCommunicationProvided && !i4.hasExternalInfo(accountArrangementItem);
        };
        AccountsDetailsTabComponent.prototype.initializeTabs = function (_b) {
            var _c = _b.children, children = _c === void 0 ? [] : _c;
            return children
                .filter(function (routeItem) { return routeItem.path && routeItem.data; })
                .map(function (routeItem, index) {
                var _a;
                return ({
                    title: ((_a = routeItem.data) === null || _a === void 0 ? void 0 : _a.title) || "Tab" + (index + 1),
                    route: routeItem.path || "" + (index + 1),
                });
            });
        };
        /**
         * Returns selected account from snapshot params of activatedRoute
         *
         * @private
         */
        AccountsDetailsTabComponent.prototype.getSelectedAccount = function () {
            return this.route.snapshot.params.selectedAccount;
        };
        return AccountsDetailsTabComponent;
    }());
    /** @nocollapse */ AccountsDetailsTabComponent.fac = i0__namespace.ngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountsDetailsTabComponent, deps: [{ token: i4.ACCOUNT_ALIAS_DISPLAYING_LEVEL, optional: true }, { token: AccountsTransactionsJourneyService }, { token: AccountsPaymentsCommunication, optional: true }, { token: i2__namespace.ActivatedRoute }, { token: i2__namespace.Router }, { token: i4__namespace$1.ProductSummaryAccountsService }, { token: i4__namespace$1.ProductsExternalInfoService }], target: i0__namespace.FactoryTarget.Component });
    /** @nocollapse */ AccountsDetailsTabComponent.cmp = i0__namespace.ngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AccountsDetailsTabComponent, selector: "bb-accounts-details-tab-container", providers: [i4.ProductSummaryAccountsService, i4.ProductsExternalInfoService], ngImport: i0__namespace, template: "<div class=\"pt-5 bb-block bb-block--xl\">\n  <button\n    bbButton\n    data-role=\"account-details-tab__back-navigation-button\"\n    buttonSize=\"md\"\n    color=\"link-text\"\n    class=\"bb-stack justify-content-start bb-stack--density-sm\"\n    (click)=\"onNavigateBack()\"\n  >\n    <bb-icon-ui size=\"md\" name=\"arrow-back\" class=\"bb-stack__item\"></bb-icon-ui>\n    <span\n      class=\"bb-stack__item\"\n      i18n=\"Back to my accounts|Navigation button@@accounts-details-tab-header-back-navigation-button\"\n    >\n      Back to my accounts\n    </span>\n  </button>\n</div>\n\n<ng-container *ngIf=\"accountArrangementItem$ | async as accountArrangementItem\">\n  <div\n    class=\"bb-card bb-card--lg\"\n    *ngIf=\"\n      accountArrangementItem?.product?.productKind | bbIsProductKind: config.productKindsWithGraphicalRepresentation;\n      else accountHeader\n    \"\n  >\n    <div class=\"bb-card__body\">\n      <bb-account-graphical-header\n        data-role=\"accounts-details-tab__account-graphical-header\"\n        [product]=\"accountArrangementItem\"\n        [canEditAlias]=\"false\"\n        [useShortCurrency]=\"true\"\n        [creditLimitViewMode]=\"config.creditLimitViewMode\"\n        [canRepay]=\"canInitiateRepay(accountArrangementItem)\"\n        [canCashInAdvance]=\"canInitiateCashInAdvance(accountArrangementItem)\"\n        [creditCardBackgroundType]=\"'darkred-card'\"\n        [aliasLevel]=\"accountAliasDisplayingLevel\"\n        [isLogoVisible]=\"false\"\n        (repayInitiated)=\"onRepayInitiated($event)\"\n        (cashAdvanceInitiated)=\"onCashAdvanceInitiated($event)\"\n      ></bb-account-graphical-header>\n    </div>\n  </div>\n  <ng-template #accountHeader>\n    <bb-account-header\n      data-role=\"accounts-details-tab__account-header\"\n      [product]=\"accountArrangementItem\"\n      [isBalanceTypeVisible]=\"false\"\n      [mapCurrency]=\"true\"\n      [aliasLevel]=\"accountAliasDisplayingLevel\"\n      [canEditAlias]=\"false\"\n      [isLogoVisible]=\"config.showAccountIcons\"\n    ></bb-account-header>\n  </ng-template>\n</ng-container>\n\n<div class=\"d-block\">\n  <div class=\"d-block mx-4 pt-5\">\n    <div class=\"bb-tab-container__tabs nav nav-tabs\">\n      <a *ngFor=\"let tab of tabs\" class=\"nav-link\" role=\"tab\" [routerLink]=\"[tab.route]\" routerLinkActive=\"active\">\n        <span class=\"bb-stack__item bb-stack__item--spacing-xs\">{{ tab?.title }}</span>\n      </a>\n    </div>\n  </div>\n\n  <div class=\"bb-tab-container__content tab-content\" role=\"tabpanel\">\n    <router-outlet></router-outlet>\n  </div>\n</div>\n", components: [{ type: i5__namespace.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i6__namespace.AccountGraphicalHeaderComponent, selector: "bb-account-graphical-header", inputs: ["product", "aliasLevel", "canEditAlias", "useShortCurrency", "creditLimitViewMode", "canRepay", "canCashInAdvance", "creditCardBackgroundType", "isLogoVisible"], outputs: ["aliasUpdated", "repayInitiated", "cashAdvanceInitiated"] }, { type: i6__namespace.AccountHeaderComponent, selector: "bb-account-header", inputs: ["product", "canEditAlias", "aliasLevel", "showBalance", "isBalanceTypeVisible", "isLogoVisible", "mapCurrency"], outputs: ["aliasUpdated"] }], directives: [{ type: i7__namespace.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }, { type: i8__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i8__namespace.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { type: i2__namespace.RouterLinkWithHref, selector: "a[routerLink],area[routerLink]", inputs: ["routerLink", "target", "queryParams", "fragment", "queryParamsHandling", "preserveFragment", "skipLocationChange", "replaceUrl", "state", "relativeTo"] }, { type: i2__namespace.RouterLinkActive, selector: "[routerLinkActive]", inputs: ["routerLinkActiveOptions", "routerLinkActive"], exportAs: ["routerLinkActive"] }, { type: i2__namespace.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }], pipes: { "async": i8__namespace.AsyncPipe, "bbIsProductKind": i4__namespace$1.IsProductKindPipe } });
    i0__namespace.ngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountsDetailsTabComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-accounts-details-tab-container',
                        templateUrl: './accounts-details-tab.component.html',
                        providers: [i4.ProductSummaryAccountsService, i4.ProductsExternalInfoService],
                    }]
            }], ctorParameters: function () {
            return [{ type: i4__namespace$1.AccountAliasDisplayingLevel, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.Inject,
                            args: [i4.ACCOUNT_ALIAS_DISPLAYING_LEVEL]
                        }] }, { type: AccountsTransactionsJourneyService }, { type: AccountsPaymentsCommunication, decorators: [{
                            type: i0.Optional
                        }] }, { type: i2__namespace.ActivatedRoute }, { type: i2__namespace.Router }, { type: i4__namespace$1.ProductSummaryAccountsService }, { type: i4__namespace$1.ProductsExternalInfoService }];
        } });

    /**
     * A widget for displaying account details
     *
     * @usageNotes
     *
     * ### Display details of an account
     *
     * ```html
     * <bb-accounts-details></bb-accounts-details>
     * ```
     */
    var AccountsDetailsComponent = /** @class */ (function () {
        /**
         * @constructor
         * @param config
         */
        function AccountsDetailsComponent(config) {
            this.config = config;
        }
        return AccountsDetailsComponent;
    }());
    /** @nocollapse */ AccountsDetailsComponent.fac = i0__namespace.ngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountsDetailsComponent, deps: [{ token: AccountsTransactionsJourneyService }], target: i0__namespace.FactoryTarget.Component });
    /** @nocollapse */ AccountsDetailsComponent.cmp = i0__namespace.ngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AccountsDetailsComponent, selector: "bb-accounts-details", providers: [i2$2.AccountService, i4.ProductSummaryAccountsService], ngImport: i0__namespace, template: " <bb-product-summary-details></bb-product-summary-details> ", isInline: true, components: [{ type: i2__namespace$2.ProductSummaryDetailsComponent, selector: "bb-product-summary-details" }] });
    i0__namespace.ngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountsDetailsComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-accounts-details',
                        template: " <bb-product-summary-details></bb-product-summary-details> ",
                        providers: [i2$2.AccountService, i4.ProductSummaryAccountsService],
                    }]
            }], ctorParameters: function () { return [{ type: AccountsTransactionsJourneyService }]; } });

    var AccountsManageComponent = /** @class */ (function () {
        function AccountsManageComponent(router, route, config) {
            this.router = router;
            this.route = route;
            this.config = config;
            this.manageMyAccounts = $localize(templateObject_1 || (templateObject_1 = __makeTemplateObject([":Manage my accounts heading|manage my accounts heading@@accounts-transactions-journey.accounts-manage.heading.manageMyAccounts:Manage my Accounts"], [":Manage my accounts heading|manage my accounts heading@@accounts-transactions-journey.accounts-manage.heading.manageMyAccounts:Manage my Accounts"])));
            this.backToAccounts = $localize(templateObject_2 || (templateObject_2 = __makeTemplateObject([":Back to my accounts button title|back to my accounts button title@@accounts-transactions-journey.accounts-manage.buttonTitle.backToAccounts:Back to my Accounts"], [":Back to my accounts button title|back to my accounts button title@@accounts-transactions-journey.accounts-manage.buttonTitle.backToAccounts:Back to my Accounts"])));
        }
        AccountsManageComponent.prototype.onNavigateBack = function () {
            this.router.navigate(['..'], { relativeTo: this.route });
        };
        return AccountsManageComponent;
    }());
    /** @nocollapse */ AccountsManageComponent.fac = i0__namespace.ngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountsManageComponent, deps: [{ token: i2__namespace.Router }, { token: i2__namespace.ActivatedRoute }, { token: AccountsTransactionsJourneyService }], target: i0__namespace.FactoryTarget.Component });
    /** @nocollapse */ AccountsManageComponent.cmp = i0__namespace.ngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AccountsManageComponent, selector: "bb-accounts-manage-list", ngImport: i0__namespace, template: "\n    <div class=\"px-0 pt-4 pb-5 flex-column-reverse align-items-start\">\n      <button (click)=\"onNavigateBack()\" class=\"bb-heading-widget__button btn btn-link pl-0 ml-0 mb-3\">\n        <i aria-hidden=\"true\" class=\"bb-icon bb-icon-arrow-back\"></i>\n        <span>{{ backToAccounts }}</span>\n      </button>\n      <bb-header-ui\n        headingType=\"h1\"\n        headingClasses=\"bb-heading-widget__heading\"\n        [heading]=\"manageMyAccounts\"\n      ></bb-header-ui>\n    </div>\n\n    <bb-product-summary-manage-accounts-widget\n      [showIcons]=\"config.showAccountIcons\"\n    ></bb-product-summary-manage-accounts-widget>\n  ", isInline: true, components: [{ type: i1__namespace$1.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i4__namespace$2.ProductSummaryManageAccountsWidgetComponent, selector: "bb-product-summary-manage-accounts-widget", inputs: ["businessFunction", "resourceName", "privilege", "size", "showIcons"] }] });
    i0__namespace.ngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountsManageComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-accounts-manage-list',
                        template: "\n    <div class=\"px-0 pt-4 pb-5 flex-column-reverse align-items-start\">\n      <button (click)=\"onNavigateBack()\" class=\"bb-heading-widget__button btn btn-link pl-0 ml-0 mb-3\">\n        <i aria-hidden=\"true\" class=\"bb-icon bb-icon-arrow-back\"></i>\n        <span>{{ backToAccounts }}</span>\n      </button>\n      <bb-header-ui\n        headingType=\"h1\"\n        headingClasses=\"bb-heading-widget__heading\"\n        [heading]=\"manageMyAccounts\"\n      ></bb-header-ui>\n    </div>\n\n    <bb-product-summary-manage-accounts-widget\n      [showIcons]=\"config.showAccountIcons\"\n    ></bb-product-summary-manage-accounts-widget>\n  ",
                    }]
            }], ctorParameters: function () { return [{ type: i2__namespace.Router }, { type: i2__namespace.ActivatedRoute }, { type: AccountsTransactionsJourneyService }]; } });
    var templateObject_1, templateObject_2;

    var AccountsManageGuardService = /** @class */ (function () {
        function AccountsManageGuardService(config) {
            this.config = config;
        }
        AccountsManageGuardService.prototype.canActivate = function () {
            var _a;
            return ((_a = this.config) === null || _a === void 0 ? void 0 : _a.enableManageAccounts) === undefined || !!this.config.enableManageAccounts;
        };
        return AccountsManageGuardService;
    }());
    /** @nocollapse */ AccountsManageGuardService.fac = i0__namespace.ngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountsManageGuardService, deps: [{ token: AccountsTransactionsJourneyConfigurationToken, optional: true }], target: i0__namespace.FactoryTarget.Injectable });
    /** @nocollapse */ AccountsManageGuardService.prov = i0__namespace.ngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountsManageGuardService });
    i0__namespace.ngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountsManageGuardService, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () {
            return [{ type: undefined, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.Inject,
                            args: [AccountsTransactionsJourneyConfigurationToken]
                        }] }];
        } });

    var TransactionDetailsComponent = /** @class */ (function () {
        function TransactionDetailsComponent(config, shared, transactionDetailsService, cd) {
            var _this = this;
            this.config = config;
            this.shared = shared;
            this.transactionDetailsService = transactionDetailsService;
            this.isDetailsOpen = true;
            this.unsubscribeSubject = new rxjs.Subject();
            this.transactionDetailsService
                .fromCheckImages(rxjs.of(this.config.showCheckImages))
                .pipe(operators.takeUntil(this.unsubscribeSubject))
                .subscribe(function (checkImagesList) {
                _this.checkImagesList = checkImagesList;
            });
            this.transactionDetailsService.fromInquiryAndDispute(rxjs.of(config.enableDisputeAndInquiry), rxjs.of(config.disputeByBillingStatus), rxjs.of(config.disputeEligibilityDays), rxjs.of(config.disputeTransactionTypes.split(',')));
            this.transactionDetailsService.loadingState
                .pipe(operators.takeUntil(this.unsubscribeSubject))
                .subscribe(function (transactionDetailsLoadingStatus) {
                _this.transactionDetailsLoadingStatus = transactionDetailsLoadingStatus;
                cd.markForCheck();
            });
            this.transactionDetailsService
                .fromGeolocation(rxjs.of(this.config.apiKey))
                .pipe(operators.takeUntil(this.unsubscribeSubject))
                .subscribe();
            this.inquireAndDisputeTopics = {
                inquire: this.config.inquireTopicId,
                dispute: this.config.disputeTopicId,
            };
        }
        TransactionDetailsComponent.prototype.ngOnInit = function () {
            var _this = this;
            this.currentTransaction = this.shared.currentTransaction.pipe(operators.tap(function (transaction) {
                if (transaction) {
                    _this.transactionDetailsService.selectTransaction(transaction);
                }
            }));
        };
        TransactionDetailsComponent.prototype.ngOnDestroy = function () {
            this.unsubscribeSubject.next();
            this.unsubscribeSubject.complete();
        };
        TransactionDetailsComponent.prototype.closeDialog = function () {
            this.isDetailsOpen = false;
            this.shared.setCloseDetail();
        };
        return TransactionDetailsComponent;
    }());
    /** @nocollapse */ TransactionDetailsComponent.fac = i0__namespace.ngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: TransactionDetailsComponent, deps: [{ token: AccountsTransactionsJourneyService }, { token: TransactionsHelperService }, { token: i3__namespace.TransactionDetailsService }, { token: i0__namespace.ChangeDetectorRef }], target: i0__namespace.FactoryTarget.Component });
    /** @nocollapse */ TransactionDetailsComponent.cmp = i0__namespace.ngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: TransactionDetailsComponent, selector: "bb-transactions-details", providers: [i3.TransactionDetailsService], ngImport: i0__namespace, template: "\n    <bb-transaction-details-modal\n      *ngIf=\"currentTransaction | async as transaction\"\n      [isDetailsOpen]=\"isDetailsOpen\"\n      [transaction]=\"transaction\"\n      [showChangeCategory]=\"true\"\n      [checkImagesList]=\"checkImagesList\"\n      [transactionDetailsLoadingStatus]=\"transactionDetailsLoadingStatus\"\n      [inquireAndDisputeTopics]=\"inquireAndDisputeTopics\"\n      (close)=\"closeDialog()\"\n    ></bb-transaction-details-modal>\n  ", isInline: true, components: [{ type: i4__namespace.TransactionDetailsModalComponent, selector: "bb-transaction-details-modal", inputs: ["isDetailsOpen", "transaction", "showBillingStatus", "showChangeCategory", "checkImagesList", "transactionDetailsLoadingStatus", "inquireAndDisputeTopics"], outputs: ["detailsOpen", "close"] }], directives: [{ type: i8__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "async": i8__namespace.AsyncPipe } });
    i0__namespace.ngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: TransactionDetailsComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-transactions-details',
                        providers: [i3.TransactionDetailsService],
                        template: "\n    <bb-transaction-details-modal\n      *ngIf=\"currentTransaction | async as transaction\"\n      [isDetailsOpen]=\"isDetailsOpen\"\n      [transaction]=\"transaction\"\n      [showChangeCategory]=\"true\"\n      [checkImagesList]=\"checkImagesList\"\n      [transactionDetailsLoadingStatus]=\"transactionDetailsLoadingStatus\"\n      [inquireAndDisputeTopics]=\"inquireAndDisputeTopics\"\n      (close)=\"closeDialog()\"\n    ></bb-transaction-details-modal>\n  ",
                    }]
            }], ctorParameters: function () { return [{ type: AccountsTransactionsJourneyService }, { type: TransactionsHelperService }, { type: i3__namespace.TransactionDetailsService }, { type: i0__namespace.ChangeDetectorRef }]; } });

    var AccountInfoContainerComponent = /** @class */ (function () {
        function AccountInfoContainerComponent() {
            /**
             * Account info properties configuration
             */
            this.propertiesConfiguration = [];
            /**
             * Currency displaying mode
             */
            this.currencyFormat = 'symbol';
            /**
             * Date format
             */
            this.dateFormat = 'longDate';
            /**
             * Attributes configuration
             */
            this.attributesConfiguration = [];
            /**
             * Mask attribute event emitter
             */
            this.mask = new i0.EventEmitter();
            /**
             * Sanitized properties configuration
             */
            this.propertyGroups = [];
        }
        AccountInfoContainerComponent.prototype.ngOnChanges = function (changes) {
            if (changes.account || changes.propertiesConfiguration) {
                this.propertyGroups = this.getDefinedPropertyGroups();
            }
        };
        AccountInfoContainerComponent.prototype.isPropertyMaskable = function (property) {
            var _a, _b;
            var attributeName = this.getMaskableAttributeName(property);
            return ((_b = (_a = this.account) === null || _a === void 0 ? void 0 : _a.unmaskableAttributes) !== null && _b !== void 0 ? _b : []).includes(attributeName);
        };
        AccountInfoContainerComponent.prototype.isPropertyMasked = function (property) {
            var _a;
            var attribute = this.getMaskableAttribute(property);
            return (_a = attribute.masked) !== null && _a !== void 0 ? _a : true;
        };
        AccountInfoContainerComponent.prototype.onPropertyMasked = function (property, isMasked) {
            var attribute = this.getMaskableAttribute(property);
            this.mask.emit(Object.assign(Object.assign({}, attribute), { masked: isMasked }));
        };
        AccountInfoContainerComponent.prototype.getMaskableAttribute = function (property) {
            var attributeName = this.getMaskableAttributeName(property);
            return this.attributesConfiguration.find(function (item) { return item.attributeName === attributeName; }) || { attributeName: attributeName };
        };
        AccountInfoContainerComponent.prototype.getMaskableAttributeName = function (property, defaultType) {
            if (defaultType === void 0) { defaultType = 'IBAN'; }
            return property.type === i1.AccountInfoPropertyType.ACCOUNT_NUMBER
                ? accountsTransactionsJourneysSharedUtilAccountAng.accountNumberToMaskableAttribute(property.numberType)
                : defaultType;
        };
        AccountInfoContainerComponent.prototype.getDefinedPropertyGroups = function () {
            var _this = this;
            return this.propertiesConfiguration
                .map(function (group) {
                var properties = group.properties.filter(function (property) { return _this.hasValue(property); });
                return Object.assign(Object.assign({}, group), { properties: properties });
            })
                .filter(function (group) { return group.properties.length > 0; });
        };
        AccountInfoContainerComponent.prototype.hasValue = function (property) {
            var _a;
            var value = accountsTransactionsJourneysSharedUtilTransformAng.getValue(this.account, (_a = property.key) !== null && _a !== void 0 ? _a : '');
            return Array.isArray(value) ? value.length > 0 : i4.isDefined(value);
        };
        return AccountInfoContainerComponent;
    }());
    /** @nocollapse */ AccountInfoContainerComponent.fac = i0__namespace.ngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoContainerComponent, deps: [], target: i0__namespace.FactoryTarget.Component });
    /** @nocollapse */ AccountInfoContainerComponent.cmp = i0__namespace.ngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AccountInfoContainerComponent, selector: "bb-account-info-container", inputs: { account: "account", propertiesConfiguration: "propertiesConfiguration", currencyFormat: "currencyFormat", dateFormat: "dateFormat", attributesConfiguration: "attributesConfiguration" }, outputs: { mask: "mask" }, usesOnChanges: true, ngImport: i0__namespace, template: "<div class=\"bb-list\" [attr.data-role]=\"'account-info-container-' + account?.product?.productKind?.kindUri\">\n  <ng-container *ngFor=\"let group of propertyGroups\">\n    <div class=\"bb-product-summary-details\" [class.card]=\"!group.hideCard\" data-role=\"account-info-container-group\">\n      <div\n        [ngClass]=\"{ 'card-header': !group.hideCard, 'bb-block bb-block--lg': group.hideCard }\"\n        data-role=\"account-info-container-header\"\n      >\n        <h3>{{ group.title }}</h3>\n      </div>\n      <div [class.card-body]=\"!group.hideCard\" data-role=\"account-info-container-body\">\n        <div\n          *ngFor=\"let row of group.properties | bbSplitArray: 2\"\n          class=\"bb-account-details-row bb-stack bb-stack--wrap bb-stack--align-top\"\n        >\n          <div\n            *ngFor=\"let property of row\"\n            class=\"bb-account-details-item bb-stack__item bb-stack__item--fill\"\n            [attr.data-role]=\"'account-info-container-' + property.type\"\n          >\n            <bb-account-info-property\n              [account]=\"account\"\n              [property]=\"property\"\n              [currencyFormat]=\"currencyFormat\"\n              [dateFormat]=\"dateFormat\"\n              [isMaskable]=\"isPropertyMaskable(property)\"\n              [isMasked]=\"isPropertyMasked(property)\"\n              (masked)=\"onPropertyMasked(property, $event)\"\n            ></bb-account-info-property>\n          </div>\n        </div>\n      </div>\n    </div>\n  </ng-container>\n</div>\n", components: [{ type: i1__namespace$2.AccountInfoPropertyComponent, selector: "bb-account-info-property", inputs: ["account", "property", "currencyFormat", "dateFormat", "isMaskable", "isMasked"], outputs: ["masked"] }], directives: [{ type: i8__namespace.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { type: i8__namespace.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }], pipes: { "bbSplitArray": i3__namespace$1.SplitArrayPipe }, changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoContainerComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-account-info-container',
                        templateUrl: './account-info-container.component.html',
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], propDecorators: { account: [{
                    type: i0.Input
                }], propertiesConfiguration: [{
                    type: i0.Input
                }], currencyFormat: [{
                    type: i0.Input
                }], dateFormat: [{
                    type: i0.Input
                }], attributesConfiguration: [{
                    type: i0.Input
                }], mask: [{
                    type: i0.Output
                }] } });

    /**
     * Accounts Transactions journey Info View
     */
    var AccountInfoViewComponent = /** @class */ (function () {
        /**
         * @internal
         * @param configurationService
         */
        function AccountInfoViewComponent(configurationService) {
            this.configurationService = configurationService;
        }
        /**
         * Returns properties configuration for a given account
         * @param account
         */
        AccountInfoViewComponent.prototype.getAccountProperties = function (account) {
            var _a, _b;
            var configuration = this.configurationService.accountInfoProperties;
            var kindUri = ((_b = (_a = account.product) === null || _a === void 0 ? void 0 : _a.productKind) === null || _b === void 0 ? void 0 : _b.kindUri) || 'default';
            return kindUri in configuration ? configuration[kindUri] : configuration.default;
        };
        return AccountInfoViewComponent;
    }());
    /** @nocollapse */ AccountInfoViewComponent.fac = i0__namespace.ngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoViewComponent, deps: [{ token: AccountsTransactionsJourneyService }], target: i0__namespace.FactoryTarget.Component });
    /** @nocollapse */ AccountInfoViewComponent.cmp = i0__namespace.ngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AccountInfoViewComponent, selector: "bb-account-info", providers: [i2$2.AccountService, i4.ProductSummaryAccountsService], ngImport: i0__namespace, template: "<bb-product-summary-details></bb-product-summary-details>\n\n<ng-template bbProductSummaryDetailsViewCustomizable let-hostRef let-account=\"context\">\n  <ng-container\n    bbProductSummaryDetailsHeaderCustomizable\n    [bbHostRef]=\"hostRef\"\n    [bbTemplateContext]=\"account\"\n  ></ng-container>\n  <ng-container bbProductSummaryDetailsCustomizable [bbHostRef]=\"hostRef\" [bbTemplateContext]=\"account\"></ng-container>\n  <ng-template bbProductSummaryDetailsCreditCardCustomizable>\n    <ng-container *ngTemplateOutlet=\"accountInfo; context: { hostRef: hostRef, account: account }\"></ng-container>\n  </ng-template>\n  <ng-template bbProductSummaryDetailsCurrentAccountCustomizable>\n    <ng-container *ngTemplateOutlet=\"accountInfo; context: { hostRef: hostRef, account: account }\"></ng-container>\n  </ng-template>\n  <ng-template bbProductSummaryDetailsDebitCardCustomizable>\n    <ng-container *ngTemplateOutlet=\"accountInfo; context: { hostRef: hostRef, account: account }\"></ng-container>\n  </ng-template>\n  <ng-template bbProductSummaryDetailsInvestmentAccountCustomizable>\n    <ng-container *ngTemplateOutlet=\"accountInfo; context: { hostRef: hostRef, account: account }\"></ng-container>\n  </ng-template>\n  <ng-template bbProductSummaryDetailsLoanCustomizable>\n    <ng-container *ngTemplateOutlet=\"accountInfo; context: { hostRef: hostRef, account: account }\"></ng-container>\n  </ng-template>\n  <ng-template bbProductSummaryDetailsOtherCustomizable>\n    <ng-container *ngTemplateOutlet=\"accountInfo; context: { hostRef: hostRef, account: account }\"></ng-container>\n  </ng-template>\n  <ng-template bbProductSummaryDetailsSavingsAccountCustomizable>\n    <ng-container *ngTemplateOutlet=\"accountInfo; context: { hostRef: hostRef, account: account }\"></ng-container>\n  </ng-template>\n  <ng-template bbProductSummaryDetailsTermDepositCustomizable>\n    <ng-container *ngTemplateOutlet=\"accountInfo; context: { hostRef: hostRef, account: account }\"></ng-container>\n  </ng-template>\n</ng-template>\n\n<ng-template #accountInfo let-hostRef=\"hostRef\" let-account=\"account\">\n  <bb-account-info-container\n    [account]=\"account\"\n    [propertiesConfiguration]=\"getAccountProperties(account)\"\n    [attributesConfiguration]=\"hostRef.maskableAttributes\"\n    (mask)=\"\n      $event.masked\n        ? hostRef.onMaskAttribute($event.attributeName, account)\n        : hostRef.onUnmaskAttribute($event.attributeName, account)\n    \"\n  ></bb-account-info-container>\n</ng-template>\n", components: [{ type: i2__namespace$2.ProductSummaryDetailsComponent, selector: "bb-product-summary-details" }, { type: AccountInfoContainerComponent, selector: "bb-account-info-container", inputs: ["account", "propertiesConfiguration", "currencyFormat", "dateFormat", "attributesConfiguration"], outputs: ["mask"] }], directives: [{ type: i2__namespace$2.ProductSummaryDetailsViewCustomizableDirective, selector: "[bbProductSummaryDetailsViewCustomizable]" }, { type: i2__namespace$2.ProductSummaryDetailsHeaderCustomizableDirective, selector: "[bbProductSummaryDetailsHeaderCustomizable]" }, { type: i2__namespace$2.ProductSummaryDetailsCustomizableDirective, selector: "[bbProductSummaryDetailsCustomizable]" }, { type: i2__namespace$2.ProductSummaryDetailsCreditCardCustomizableDirective, selector: "[bbProductSummaryDetailsCreditCardCustomizable]" }, { type: i8__namespace.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }, { type: i2__namespace$2.ProductSummaryDetailsCurrentAccountCustomizableDirective, selector: "[bbProductSummaryDetailsCurrentAccountCustomizable]" }, { type: i2__namespace$2.ProductSummaryDetailsDebitCardCustomizableDirective, selector: "[bbProductSummaryDetailsDebitCardCustomizable]" }, { type: i2__namespace$2.ProductSummaryDetailsInvestmentAccountCustomizableDirective, selector: "[bbProductSummaryDetailsInvestmentAccountCustomizable]" }, { type: i2__namespace$2.ProductSummaryDetailsLoanCustomizableDirective, selector: "[bbProductSummaryDetailsLoanCustomizable]" }, { type: i2__namespace$2.ProductSummaryDetailsOtherCustomizableDirective, selector: "[bbProductSummaryDetailsOtherCustomizable]" }, { type: i2__namespace$2.ProductSummaryDetailsSavingsAccountCustomizableDirective, selector: "[bbProductSummaryDetailsSavingsAccountCustomizable]" }, { type: i2__namespace$2.ProductSummaryDetailsTermDepositCustomizableDirective, selector: "[bbProductSummaryDetailsTermDepositCustomizable]" }] });
    i0__namespace.ngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoViewComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-account-info',
                        templateUrl: 'account-info-view.component.html',
                        providers: [i2$2.AccountService, i4.ProductSummaryAccountsService],
                    }]
            }], ctorParameters: function () { return [{ type: AccountsTransactionsJourneyService }]; } });

    var components = [AccountInfoViewComponent, AccountInfoContainerComponent];
    var modules = [
        i2$2.ProductSummaryDetailsWidgetModule,
        i1.AccountInfoPropertyModule,
        amount.AmountModule,
        i6.ProductNumberModule,
        i4.ValidationModule,
        i4.FormatPipeModule,
        i6.ArrangementStateIndicatorModule,
        tooltipDirective.TooltipModule,
        datePipe.BbDatePipeModule,
        i4.DatePipeModule,
        paymentCard.PaymentCardModule,
        i3$1.SplitArrayPipeModule,
        i4.ProductSummaryCommon,
    ];
    var AccountInfoViewModule = /** @class */ (function () {
        function AccountInfoViewModule() {
        }
        return AccountInfoViewModule;
    }());
    /** @nocollapse */ AccountInfoViewModule.fac = i0__namespace.ngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoViewModule, deps: [], target: i0__namespace.FactoryTarget.NgModule });
    /** @nocollapse */ AccountInfoViewModule.mod = i0__namespace.ngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoViewModule, declarations: [AccountInfoViewComponent, AccountInfoContainerComponent], imports: [i8.CommonModule, i2$2.ProductSummaryDetailsWidgetModule,
            i1.AccountInfoPropertyModule,
            amount.AmountModule,
            i6.ProductNumberModule,
            i4.ValidationModule,
            i4.FormatPipeModule,
            i6.ArrangementStateIndicatorModule,
            tooltipDirective.TooltipModule,
            datePipe.BbDatePipeModule,
            i4.DatePipeModule,
            paymentCard.PaymentCardModule,
            i3$1.SplitArrayPipeModule,
            i4.ProductSummaryCommon], exports: [i2$2.ProductSummaryDetailsWidgetModule,
            i1.AccountInfoPropertyModule,
            amount.AmountModule,
            i6.ProductNumberModule,
            i4.ValidationModule,
            i4.FormatPipeModule,
            i6.ArrangementStateIndicatorModule,
            tooltipDirective.TooltipModule,
            datePipe.BbDatePipeModule,
            i4.DatePipeModule,
            paymentCard.PaymentCardModule,
            i3$1.SplitArrayPipeModule,
            i4.ProductSummaryCommon, AccountInfoViewComponent, AccountInfoContainerComponent] });
    /** @nocollapse */ AccountInfoViewModule.inj = i0__namespace.ngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoViewModule, imports: [__spreadArray([i8.CommonModule], __read(modules)), i2$2.ProductSummaryDetailsWidgetModule,
            i1.AccountInfoPropertyModule,
            amount.AmountModule,
            i6.ProductNumberModule,
            i4.ValidationModule,
            i4.FormatPipeModule,
            i6.ArrangementStateIndicatorModule,
            tooltipDirective.TooltipModule,
            datePipe.BbDatePipeModule,
            i4.DatePipeModule,
            paymentCard.PaymentCardModule,
            i3$1.SplitArrayPipeModule,
            i4.ProductSummaryCommon] });
    i0__namespace.ngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoViewModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: __spreadArray([i8.CommonModule], __read(modules)),
                        declarations: __spreadArray([], __read(components)),
                        exports: __spreadArray(__spreadArray([], __read(modules)), __read(components)),
                    }]
            }] });

    var ArrangementsListComponent = /** @class */ (function (_super) {
        __extends(ArrangementsListComponent, _super);
        /**
         * @internal
         * @param config
         * @param router
         * @param route
         * @param accountsCommunicationService
         * @param accountsPaymentsCommunication
         */
        function ArrangementsListComponent(
        /**
         * Journey configuration
         */
        config, router, route, accountsCommunicationService, accountsPaymentsCommunication) {
            var _this = _super.call(this, config, router, route, accountsCommunicationService) || this;
            _this.accountsPaymentsCommunication = accountsPaymentsCommunication;
            return _this;
        }
        /**
         * Navigates to transfer payment
         *
         * @param arrangementId
         */
        ArrangementsListComponent.prototype.onPayNowClick = function (arrangementId) {
            var _a;
            (_a = this.accountsPaymentsCommunication) === null || _a === void 0 ? void 0 : _a.payNowEvent(arrangementId);
        };
        return ArrangementsListComponent;
    }(AccountsListBaseComponent));
    /** @nocollapse */ ArrangementsListComponent.fac = i0__namespace.ngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ArrangementsListComponent, deps: [{ token: AccountsTransactionsJourneyService }, { token: i2__namespace.Router }, { token: i2__namespace.ActivatedRoute }, { token: AccountsCommunicationService, optional: true }, { token: AccountsPaymentsCommunication, optional: true }], target: i0__namespace.FactoryTarget.Component });
    /** @nocollapse */ ArrangementsListComponent.cmp = i0__namespace.ngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ArrangementsListComponent, selector: "bb-arrangements-list", usesInheritance: true, ngImport: i0__namespace, template: "<ng-container *ngIf=\"config.enableManageAccounts; else titleOnly\">\n  <div class=\"bb-heading-widget heading-widget px-0 flex-row align-items-center\">\n    <bb-header-ui headingType=\"h1\" headingClasses=\"bb-heading-widget__heading\" [heading]=\"myAccounts\"></bb-header-ui>\n    <button\n      (click)=\"navigateToAccountManagement()\"\n      class=\"bb-heading-widget__button btn btn-link w-auto bb-button bb-button--xs-hide-text m-0\"\n    >\n      <i aria-hidden=\"true\" class=\"bb-icon bb-icon-settings\"></i>\n      <span>{{ manageAccounts }}</span>\n    </button>\n  </div>\n</ng-container>\n\n<ng-template #titleOnly>\n  <div class=\"bb-heading-widget heading-widget\">\n    <bb-header-ui headingType=\"h1\" headingClasses=\"bb-heading-widget__heading\" [heading]=\"myAccounts\"></bb-header-ui>\n  </div>\n</ng-template>\n\n<bb-arrangements-list-widget\n  [showIcons]=\"config.showAccountIcons\"\n  (selectProduct)=\"onProductSelect($event)\"\n  (payNowClick)=\"onPayNowClick($event)\"\n  [viewName]=\"config.arrangementViewsName\"\n></bb-arrangements-list-widget>\n", components: [{ type: i1__namespace$1.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i2__namespace$1.ArrangementsListWidgetComponent, selector: "bb-arrangements-list-widget", inputs: ["viewName"], outputs: ["payNowClick"] }], directives: [{ type: i8__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
    i0__namespace.ngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ArrangementsListComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-arrangements-list',
                        templateUrl: 'arrangements-list.component.html',
                    }]
            }], ctorParameters: function () {
            return [{ type: AccountsTransactionsJourneyService }, { type: i2__namespace.Router }, { type: i2__namespace.ActivatedRoute }, { type: AccountsCommunicationService, decorators: [{
                            type: i0.Optional
                        }] }, { type: AccountsPaymentsCommunication, decorators: [{
                            type: i0.Optional
                        }] }];
        } });

    var defaultRoute = {
        path: '',
        component: AccountsTransactionsJourneyComponent,
        children: [
            {
                path: '',
                redirectTo: 'list',
                pathMatch: 'full',
            },
            {
                path: 'list',
                component: AccountsListComponent,
                data: { title: 'My Accounts' },
            },
            {
                path: 'manage',
                component: AccountsManageComponent,
                canActivate: [AccountsManageGuardService],
            },
            {
                path: 'transactions',
                component: AccountsDetailsTabComponent,
                data: { title: 'Transactions' },
                children: [
                    { path: '', redirectTo: 'list', pathMatch: 'full' },
                    {
                        path: 'list',
                        component: TransactionsListComponent,
                        data: { title: 'List' },
                        children: [
                            {
                                path: 'detail',
                                component: TransactionDetailsComponent,
                            },
                        ],
                    },
                    {
                        path: 'details',
                        component: AccountInfoViewComponent,
                        data: { title: 'Details' },
                    },
                ],
            },
        ],
    };
    var views = [
        TransactionsListComponent,
        TransactionDetailsComponent,
        AccountsListComponent,
        ArrangementsListComponent,
        AccountsDetailsTabComponent,
        AccountsDetailsComponent,
        AccountsManageComponent,
    ];
    var AccountsTransactionsJourneyModule = /** @class */ (function () {
        function AccountsTransactionsJourneyModule() {
        }
        AccountsTransactionsJourneyModule.forRoot = function (data) {
            if (data === void 0) { data = { route: defaultRoute }; }
            return {
                ngModule: AccountsTransactionsJourneyModule,
                providers: [i2.provideRoutes([data.route])],
            };
        };
        return AccountsTransactionsJourneyModule;
    }());
    /** @nocollapse */ AccountsTransactionsJourneyModule.fac = i0__namespace.ngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountsTransactionsJourneyModule, deps: [], target: i0__namespace.FactoryTarget.NgModule });
    /** @nocollapse */ AccountsTransactionsJourneyModule.mod = i0__namespace.ngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountsTransactionsJourneyModule, declarations: [AccountsTransactionsJourneyComponent, TransactionsListComponent,
            TransactionDetailsComponent,
            AccountsListComponent,
            ArrangementsListComponent,
            AccountsDetailsTabComponent,
            AccountsDetailsComponent,
            AccountsManageComponent], imports: [i8.CommonModule, i1__namespace.BackbaseCoreModule, i2.RouterModule,
            AccountInfoViewModule,
            i4$1.TransactionsListWidgetModule,
            i3.TransactionsCommonModule,
            productSummaryAccountSelectorWidgetAng.ProductSummaryAccountSelectorWidgetModule,
            i4$2.ProductSummaryManageAccountsWidgetModule,
            i2$2.ProductSummaryDetailsWidgetModule,
            i2$1.ProductSummaryListWidgetModule,
            i6.AccountHeaderModule,
            i6.AccountGraphicalHeaderModule,
            i4.ValidationModule, i4__namespace.TransactionsListWidgetModule] });
    /** @nocollapse */ AccountsTransactionsJourneyModule.inj = i0__namespace.ngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountsTransactionsJourneyModule, providers: [AccountsManageGuardService, TransactionsHelperService], imports: [[
                i8.CommonModule,
                i1$1.BackbaseCoreModule.withConfig({
                    classMap: {
                        // eslint-disable-next-line
                        AccountsTransactionsJourneyComponent: AccountsTransactionsJourneyComponent,
                    },
                }),
                i2.RouterModule,
                AccountInfoViewModule,
                i4$1.TransactionsListWidgetModule,
                i3.TransactionsCommonModule,
                productSummaryAccountSelectorWidgetAng.ProductSummaryAccountSelectorWidgetModule,
                i4$2.ProductSummaryManageAccountsWidgetModule,
                i2$2.ProductSummaryDetailsWidgetModule,
                i2$1.ProductSummaryListWidgetModule,
                i6.AccountHeaderModule,
                i6.AccountGraphicalHeaderModule,
                i4.ValidationModule,
                i4$1.TransactionsListWidgetModule.forRoot({
                    showDetails: false,
                }),
            ]] });
    i0__namespace.ngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountsTransactionsJourneyModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: __spreadArray([AccountsTransactionsJourneyComponent], __read(views)),
                        imports: [
                            i8.CommonModule,
                            i1$1.BackbaseCoreModule.withConfig({
                                classMap: {
                                    // eslint-disable-next-line
                                    AccountsTransactionsJourneyComponent: AccountsTransactionsJourneyComponent,
                                },
                            }),
                            i2.RouterModule,
                            AccountInfoViewModule,
                            i4$1.TransactionsListWidgetModule,
                            i3.TransactionsCommonModule,
                            productSummaryAccountSelectorWidgetAng.ProductSummaryAccountSelectorWidgetModule,
                            i4$2.ProductSummaryManageAccountsWidgetModule,
                            i2$2.ProductSummaryDetailsWidgetModule,
                            i2$1.ProductSummaryListWidgetModule,
                            i6.AccountHeaderModule,
                            i6.AccountGraphicalHeaderModule,
                            i4.ValidationModule,
                            i4$1.TransactionsListWidgetModule.forRoot({
                                showDetails: false,
                            }),
                        ],
                        providers: [AccountsManageGuardService, TransactionsHelperService],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    Object.defineProperty(exports, 'ACCOUNT_ALIAS_DISPLAYING_LEVEL', {
        enumerable: true,
        get: function () { return i4.ACCOUNT_ALIAS_DISPLAYING_LEVEL; }
    });
    Object.defineProperty(exports, 'AccountAliasDisplayingLevel', {
        enumerable: true,
        get: function () { return i4.AccountAliasDisplayingLevel; }
    });
    Object.defineProperty(exports, 'ProductKindUri', {
        enumerable: true,
        get: function () { return i4.ProductKindUri; }
    });
    Object.defineProperty(exports, 'AccountInfoPropertyType', {
        enumerable: true,
        get: function () { return i1.AccountInfoPropertyType; }
    });
    exports.AccountInfoContainerComponent = AccountInfoContainerComponent;
    exports.AccountInfoViewComponent = AccountInfoViewComponent;
    exports.AccountInfoViewModule = AccountInfoViewModule;
    exports.AccountsCommunicationService = AccountsCommunicationService;
    exports.AccountsDetailsComponent = AccountsDetailsComponent;
    exports.AccountsDetailsTabComponent = AccountsDetailsTabComponent;
    exports.AccountsListBaseComponent = AccountsListBaseComponent;
    exports.AccountsListComponent = AccountsListComponent;
    exports.AccountsManageComponent = AccountsManageComponent;
    exports.AccountsManageGuardService = AccountsManageGuardService;
    exports.AccountsPaymentsCommunication = AccountsPaymentsCommunication;
    exports.AccountsTransactionsJourneyComponent = AccountsTransactionsJourneyComponent;
    exports.AccountsTransactionsJourneyConfigurationToken = AccountsTransactionsJourneyConfigurationToken;
    exports.AccountsTransactionsJourneyModule = AccountsTransactionsJourneyModule;
    exports.AccountsTransactionsJourneyService = AccountsTransactionsJourneyService;
    exports.ArrangementsListComponent = ArrangementsListComponent;
    exports.TransactionDetailsComponent = TransactionDetailsComponent;
    exports.TransactionsListComponent = TransactionsListComponent;
    exports.defaultAccountInfoConfiguration = defaultAccountInfoConfiguration;
    exports.publishSwitch = publishSwitch;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
