// @backbase/manage-upcoming-and-historical-payments-journey-ang/esm2015/lib/views/payments-container-view/upcoming-and-history-payments-tab-container-view.component.js
import { Component } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { ManageUpcomingAndHistoricalPaymentsNavigationService } from '../../services/navigation.service';
import * as i0 from "@angular/core";
import * as i1 from "../../services/navigation.service";
import * as i2 from "@angular/router";
import * as i3 from "@backbase/ui-ang/tab";
export class UpcomingAndHistoryPaymentsTabContainerViewComponent {
    constructor(navigationService, route) {
        this.navigationService = navigationService;
        this.route = route;
        this.initialTabSelection = 0;
    }
    ngOnInit() {
        this.initialTabSelection = this.navigationService.mapRouteToTabIndex();
    }
    onTabSelect(tabIndex) {
        if (tabIndex === 0) {
            this.navigationService.navigateToUpcoming({ relativeTo: this.route });
        }
        else {
            this.navigationService.navigateToHistory({ relativeTo: this.route });
        }
    }
}
/** @nocollapse */ UpcomingAndHistoryPaymentsTabContainerViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: UpcomingAndHistoryPaymentsTabContainerViewComponent, deps: [{ token: i1.ManageUpcomingAndHistoricalPaymentsNavigationService }, { token: i2.ActivatedRoute }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ UpcomingAndHistoryPaymentsTabContainerViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: UpcomingAndHistoryPaymentsTabContainerViewComponent, selector: "bb-upcoming-and-history-payments-tab-container-view", ngImport: i0, template: "<bb-tab-group-ui\n  [initialSelection]=\"initialTabSelection\"\n  (select)=\"onTabSelect($event)\"\n  class=\"bb-upcoming-payments-tab-container\"\n>\n  <bb-tab-ui i18n=\"@@bb-upcoming-and-history-payments-journey.upcoming-tab\">Schedule</bb-tab-ui>\n  <bb-tab-ui i18n=\"@@bb-upcoming-and-history-payments-journey.history-tab\">History</bb-tab-ui>\n</bb-tab-group-ui>\n\n<router-outlet></router-outlet>\n", components: [{ type: i3.TabGroupComponent, selector: "bb-tab-group-ui", inputs: ["initialSelection"], outputs: ["select"] }, { type: i3.TabComponent, selector: "bb-tab-ui" }], directives: [{ type: i2.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: UpcomingAndHistoryPaymentsTabContainerViewComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-upcoming-and-history-payments-tab-container-view',
                    templateUrl: 'upcoming-and-history-payments-tab-container-view.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.ManageUpcomingAndHistoricalPaymentsNavigationService }, { type: i2.ActivatedRoute }]; } });
