// @backbase/income-spending-analysis-journey-ang/esm2015/src/income-spending-analysis-journey.component.js
import { Component } from '@angular/core';
import { IncomeSpendingAnalysisJourneyService } from './income-spending-analysis-journey-config.service';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
export class IncomeSpendingAnalysisJourneyComponent {
}
IncomeSpendingAnalysisJourneyComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IncomeSpendingAnalysisJourneyComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
IncomeSpendingAnalysisJourneyComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: IncomeSpendingAnalysisJourneyComponent, selector: "bb-income-spending-analysis-journey", providers: [IncomeSpendingAnalysisJourneyService], ngImport: i0, template: '<router-outlet></router-outlet>', isInline: true, directives: [{ type: i1.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IncomeSpendingAnalysisJourneyComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-income-spending-analysis-journey',
                    template: '<router-outlet></router-outlet>',
                    providers: [IncomeSpendingAnalysisJourneyService],
                }]
        }] });
