// @backbase/income-spending-analysis-journey-ang/esm2015/src/income-spending-analysis-journey-config.service.js
import { Injectable, Optional, Inject, InjectionToken } from '@angular/core';
import * as i0 from "@angular/core";
export var AnalysisType;
(function (AnalysisType) {
    AnalysisType["INCOME"] = "CRDT";
    AnalysisType["SPENDING"] = "DBIT";
})(AnalysisType || (AnalysisType = {}));
export const IncomeSpendingAnalysisJourneyConfigurationToken = new InjectionToken('IncomeSpendingAnalysisJourneyConfiguration injection token');
const configDefaults = {
    analysisType: AnalysisType.SPENDING,
};
export class IncomeSpendingAnalysisJourneyService {
    constructor(config) {
        config = config || {};
        this._config = Object.assign(Object.assign({}, configDefaults), config);
    }
    get defaults() {
        return configDefaults;
    }
    get analysisType() {
        return this._config.analysisType;
    }
}
IncomeSpendingAnalysisJourneyService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IncomeSpendingAnalysisJourneyService, deps: [{ token: IncomeSpendingAnalysisJourneyConfigurationToken, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
IncomeSpendingAnalysisJourneyService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IncomeSpendingAnalysisJourneyService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IncomeSpendingAnalysisJourneyService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [IncomeSpendingAnalysisJourneyConfigurationToken]
                }] }]; } });
