// @backbase/income-spending-analysis-journey-ang/esm2015/src/income-spending-analysis-journey.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { provideRoutes, RouterModule } from '@angular/router';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { IncomeSpendingAnalysisWidgetModule } from '@backbase/income-spending-analysis-widget-ang';
import { ProductSummaryAccountSelectorWidgetModule } from '@backbase/product-summary-account-selector-widget-ang';
import { HeaderModule } from '@backbase/ui-ang/header';
import { IncomeSpendingAnalysisJourneyComponent } from './income-spending-analysis-journey.component';
import { IncomeSpendingAnalysisViewComponent } from './views/income-spending-analysis-view.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
export const defaultRoute = {
    path: '',
    component: IncomeSpendingAnalysisJourneyComponent,
    children: [
        {
            path: '',
            component: IncomeSpendingAnalysisViewComponent,
        },
    ],
};
export class IncomeSpendingAnalysisJourneyModule {
    static forRoot(data = { route: defaultRoute }) {
        return {
            ngModule: IncomeSpendingAnalysisJourneyModule,
            providers: [provideRoutes([data.route])],
        };
    }
}
IncomeSpendingAnalysisJourneyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IncomeSpendingAnalysisJourneyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IncomeSpendingAnalysisJourneyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IncomeSpendingAnalysisJourneyModule, declarations: [IncomeSpendingAnalysisJourneyComponent, IncomeSpendingAnalysisViewComponent], imports: [CommonModule, i1.BackbaseCoreModule, RouterModule,
        HeaderModule,
        ProductSummaryAccountSelectorWidgetModule,
        IncomeSpendingAnalysisWidgetModule] });
IncomeSpendingAnalysisJourneyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IncomeSpendingAnalysisJourneyModule, imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { IncomeSpendingAnalysisJourneyComponent },
            }),
            RouterModule,
            HeaderModule,
            ProductSummaryAccountSelectorWidgetModule,
            IncomeSpendingAnalysisWidgetModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IncomeSpendingAnalysisJourneyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [IncomeSpendingAnalysisJourneyComponent, IncomeSpendingAnalysisViewComponent],
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { IncomeSpendingAnalysisJourneyComponent },
                        }),
                        RouterModule,
                        HeaderModule,
                        ProductSummaryAccountSelectorWidgetModule,
                        IncomeSpendingAnalysisWidgetModule,
                    ],
                }]
        }] });
