// @backbase/payment-orders-ang/esm2015/lib/ui/components/separator/separator.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SeparatorComponent } from './separator.component';
import { IconModule } from '@backbase/ui-ang/icon';
import * as i0 from "@angular/core";
export class SeparatorModule {
}
/** @nocollapse */ SeparatorModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SeparatorModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ SeparatorModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SeparatorModule, declarations: [SeparatorComponent], imports: [CommonModule, IconModule], exports: [SeparatorComponent] });
/** @nocollapse */ SeparatorModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SeparatorModule, imports: [[CommonModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SeparatorModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [SeparatorComponent],
                    imports: [CommonModule, IconModule],
                    exports: [SeparatorComponent],
                }]
        }] });
