// @backbase/ui-ang/esm2015/product-item-current-account/product-item-current-account.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { CustomProductItemCurrentAccountDetailsFormatterDirective, CustomProductItemCurrentAccountMainSectionDirective, CustomProductItemCurrentAccountSecondarySectionDirective, CustomProductItemCurrentAccountTitleDirective, ProductItemCurrentAccountComponent, } from './product-item-current-account.component';
import { AmountModule } from '@backbase/ui-ang/amount';
import { PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import { HighlightModule } from '@backbase/ui-ang/highlight';
import { IconModule } from '@backbase/ui-ang/icon';
import * as i0 from "@angular/core";
export class ProductItemCurrentAccountModule {
}
ProductItemCurrentAccountModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCurrentAccountModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductItemCurrentAccountModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCurrentAccountModule, declarations: [ProductItemCurrentAccountComponent,
        CustomProductItemCurrentAccountDetailsFormatterDirective,
        CustomProductItemCurrentAccountMainSectionDirective,
        CustomProductItemCurrentAccountSecondarySectionDirective,
        CustomProductItemCurrentAccountTitleDirective], imports: [CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule], exports: [ProductItemCurrentAccountComponent,
        CustomProductItemCurrentAccountDetailsFormatterDirective,
        CustomProductItemCurrentAccountMainSectionDirective,
        CustomProductItemCurrentAccountSecondarySectionDirective,
        CustomProductItemCurrentAccountTitleDirective] });
ProductItemCurrentAccountModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCurrentAccountModule, imports: [[CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCurrentAccountModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule],
                    declarations: [
                        ProductItemCurrentAccountComponent,
                        CustomProductItemCurrentAccountDetailsFormatterDirective,
                        CustomProductItemCurrentAccountMainSectionDirective,
                        CustomProductItemCurrentAccountSecondarySectionDirective,
                        CustomProductItemCurrentAccountTitleDirective,
                    ],
                    exports: [
                        ProductItemCurrentAccountComponent,
                        CustomProductItemCurrentAccountDetailsFormatterDirective,
                        CustomProductItemCurrentAccountMainSectionDirective,
                        CustomProductItemCurrentAccountSecondarySectionDirective,
                        CustomProductItemCurrentAccountTitleDirective,
                    ],
                }]
        }] });
