// @backbase/payment-order-options-http-ang/esm2015/api/sanctionedCurrencies.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const SanctionedCurrenciesHttpServiceGetSanctionedCurrenciesMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/sanctioned-currencies",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "default": {
                        "code": "GBP",
                        "decimals": 2
                    },
                    "labeled": {
                        "normal": [{
                                "code": "AED",
                                "decimals": 2
                            }, {
                                "code": "AUD",
                                "decimals": 2
                            }, {
                                "code": "BRL",
                                "decimals": 2
                            }, {
                                "code": "CAD",
                                "decimals": 2
                            }, {
                                "code": "CNY",
                                "decimals": 2
                            }, {
                                "code": "HKD",
                                "decimals": 2
                            }, {
                                "code": "INR",
                                "decimals": 2
                            }, {
                                "code": "JPY",
                                "decimals": 0
                            }, {
                                "code": "JOD",
                                "decimals": 3
                            }, {
                                "code": "RUB",
                                "decimals": 2
                            }, {
                                "code": "SGD",
                                "decimals": 2
                            }, {
                                "code": "ZAR",
                                "decimals": 2
                            }],
                        "popular": [{
                                "code": "EUR",
                                "decimals": 2
                            }, {
                                "code": "GBP",
                                "decimals": 2
                            }, {
                                "code": "USD",
                                "decimals": 2
                            }]
                    }
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const SanctionedCurrenciesHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v1/sanctioned-currencies",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "default": {
                        "code": "GBP",
                        "decimals": 2
                    },
                    "labeled": {
                        "normal": [{
                                "code": "AED",
                                "decimals": 2
                            }, {
                                "code": "AUD",
                                "decimals": 2
                            }, {
                                "code": "BRL",
                                "decimals": 2
                            }, {
                                "code": "CAD",
                                "decimals": 2
                            }, {
                                "code": "CNY",
                                "decimals": 2
                            }, {
                                "code": "HKD",
                                "decimals": 2
                            }, {
                                "code": "INR",
                                "decimals": 2
                            }, {
                                "code": "JPY",
                                "decimals": 0
                            }, {
                                "code": "JOD",
                                "decimals": 3
                            }, {
                                "code": "RUB",
                                "decimals": 2
                            }, {
                                "code": "SGD",
                                "decimals": 2
                            }, {
                                "code": "ZAR",
                                "decimals": 2
                            }],
                        "popular": [{
                                "code": "EUR",
                                "decimals": 2
                            }, {
                                "code": "GBP",
                                "decimals": 2
                            }, {
                                "code": "USD",
                                "decimals": 2
                            }]
                    }
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
]);
