// @backbase/ui-ang/esm2015/_example/amount/showcase/amount-showcase.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { AmountShowcaseComponent } from './amount-showcase.component';
import { HighlightModule } from '@backbase/ui-ang/highlight';
import { AmountModule } from '@backbase/ui-ang/amount';
import * as i0 from "@angular/core";
export class AmountShowcaseModule {
}
AmountShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AmountShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AmountShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AmountShowcaseModule, declarations: [AmountShowcaseComponent], imports: [CommonModule, AmountModule, HighlightModule] });
AmountShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AmountShowcaseModule, imports: [[CommonModule, AmountModule, HighlightModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AmountShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, AmountModule, HighlightModule],
                    declarations: [AmountShowcaseComponent],
                    entryComponents: [AmountShowcaseComponent],
                }]
        }] });
