// @backbase/foundation-ang/esm2015/core/backbase-core/rendering/chrome/journey.component.js
import { Component } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { FeaturesService } from '@backbase/foundation-ang/future';
import { StoreFactory, ɵStoreBroker, ɵSTORE_BRANCH, ɵSTORE_DISPATCH_CHANNEL } from '@backbase/foundation-ang/store';
import { TemplateRegistry } from '../../customizable/template-registry';
import { ItemModel } from '../../models/item-model';
export function journeyItemModelFactory(route) {
    return route.snapshot.data.item;
}
export function storeNameFactory(route) {
    return route.snapshot.data.item.name;
}
export class JourneyComponent {
}
JourneyComponent.decorators = [
    { type: Component, args: [{
                selector: 'bb-journey',
                template: ` <router-outlet></router-outlet> `,
                providers: [
                    {
                        provide: ItemModel,
                        useFactory: journeyItemModelFactory,
                        deps: [ActivatedRoute],
                    },
                    TemplateRegistry,
                    {
                        provide: ɵSTORE_BRANCH,
                        useFactory: storeNameFactory,
                        deps: [ActivatedRoute],
                    },
                    {
                        provide: ɵSTORE_DISPATCH_CHANNEL,
                        useFactory: storeNameFactory,
                        deps: [ActivatedRoute],
                    },
                    {
                        provide: StoreFactory,
                        useClass: StoreFactory,
                        deps: [ɵStoreBroker, ɵSTORE_BRANCH, ɵSTORE_DISPATCH_CHANNEL, FeaturesService],
                    },
                ]
            },] }
];
