// @backbase/foundation-ang/esm2015/auth/invalid-token.interceptor.js
import { Inject, Injectable, InjectionToken, Optional } from '@angular/core';
import { HttpClient, } from '@angular/common/http';
import { throwError } from 'rxjs';
import { catchError } from 'rxjs/operators';
import { AuthService } from './deprecated/auth.service';
import { createAutoRefresh } from './auto-refresh';
import { is401AndNotReauth } from './auth-utils';
import { DOCUMENT } from '@angular/common';
export const INVALID_TOKEN_CONFIG = new InjectionToken('invalid-token-config');
export class InvalidTokenInterceptor {
    constructor(authService, httpClient, injectedConfig, document) {
        this.authService = authService;
        this.httpClient = httpClient;
        this.injectedConfig = injectedConfig;
        this.document = document;
        this.config = {
            enableAutoRefresh: false,
            redirectToLoginPage: true,
        };
        this.autoRefresh = createAutoRefresh(this.authService, this.httpClient);
        this.config = Object.assign(Object.assign({}, this.config), (this.injectedConfig || {}));
    }
    checkRedirect() {
        if (this.config.redirectToLoginPage) {
            this.authService.logout({ redirectUri: this.document.location.href });
        }
    }
    tryRefresh(req, err) {
        return this.autoRefresh.refreshAndRepeat(req).pipe(catchError(() => {
            this.checkRedirect();
            return throwError(err);
        }));
    }
    intercept(req, next) {
        return next.handle(req).pipe(catchError((err) => {
            if (is401AndNotReauth(err)) {
                if (this.config.enableAutoRefresh && !this.autoRefresh.failed) {
                    return this.tryRefresh(req, err);
                }
                this.checkRedirect();
            }
            return throwError(err);
        }));
    }
}
InvalidTokenInterceptor.decorators = [
    { type: Injectable }
];
InvalidTokenInterceptor.ctorParameters = () => [
    { type: AuthService },
    { type: HttpClient },
    { type: undefined, decorators: [{ type: Optional }, { type: Inject, args: [INVALID_TOKEN_CONFIG,] }] },
    { type: undefined, decorators: [{ type: Inject, args: [DOCUMENT,] }] }
];
