// @backbase/ui-ang/esm2015/_example/infinite-scroll/showcase/infinite-scroll-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { InfiniteScrollShowcaseComponent } from './infinite-scroll-showcase.component';
import { InfiniteScrollModule } from '@backbase/ui-ang/infinite-scroll';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { AmountModule } from '@backbase/ui-ang/amount';
import * as i0 from "@angular/core";
export class InfiniteScrollShowcaseModule {
}
InfiniteScrollShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InfiniteScrollShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InfiniteScrollShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InfiniteScrollShowcaseModule, declarations: [InfiniteScrollShowcaseComponent], imports: [CommonModule, InfiniteScrollModule, LoadingIndicatorModule, AmountModule] });
InfiniteScrollShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InfiniteScrollShowcaseModule, imports: [[CommonModule, InfiniteScrollModule, LoadingIndicatorModule, AmountModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InfiniteScrollShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, InfiniteScrollModule, LoadingIndicatorModule, AmountModule],
                    declarations: [InfiniteScrollShowcaseComponent],
                    entryComponents: [InfiniteScrollShowcaseComponent],
                }]
        }] });
