// @backbase/ui-ang/esm2015/_example/infinite-scroll/showcase/infinite-scroll-showcase.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/infinite-scroll";
import * as i2 from "@backbase/ui-ang/amount";
import * as i3 from "@backbase/ui-ang/loading-indicator";
import * as i4 from "@angular/common";
let InfiniteScrollShowcaseComponent = class InfiniteScrollShowcaseComponent {
    constructor() {
        this.items = [1, 2, 3, 4];
        this.complete = false;
        this.loading = false;
        this.page = 0;
    }
    loadMore() {
        if (!this.loading) {
            this.loading = true;
            this.page = this.page + 1;
            window.setTimeout(() => {
                this.items = Array.from({ length: this.page * 4 + 4 }).map((_, i) => i + 1);
                this.loading = false;
                this.complete = this.page >= 4;
            }, 1000);
        }
    }
};
InfiniteScrollShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InfiniteScrollShowcaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
InfiniteScrollShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: InfiniteScrollShowcaseComponent, selector: "bb-infinite-scroll-ui-showcase", ngImport: i0, template: "<div class=\"row\">\n  <div class=\"col-md-6\">\n    <div class=\"card\">\n      <div class=\"bb-infinite-scroll-container card-body\">\n        <bb-infinite-scroll-ui\n          class=\"bb-infinite-scroll\"\n          (scrollEnd)=\"loadMore()\"\n          [disableScrollEnd]=\"complete || loading\"\n        >\n          <div\n            class=\"bb-list bb-subtitle bb-list--density-sm\"\n            [attr.aria-busy]=\"loading\"\n            [class.bb-list--infinite]=\"!complete\"\n          >\n            <div *ngFor=\"let item of items\" class=\"bb-list__item\">\n              <div class=\"bb-stack\">\n                <div class=\"bb-stack__item\">Some kind of data:</div>\n                <bb-amount-ui\n                  [amount]=\"item\"\n                  [currency]=\"'USD'\"\n                  class=\"bb-stack__item bb-stack__item--push-right\"\n                ></bb-amount-ui>\n              </div>\n            </div>\n          </div>\n          <div *ngIf=\"loading\" class=\"bb-block bb-block--md\">\n            <bb-loading-indicator-ui loaderSize=\"sm\"></bb-loading-indicator-ui>\n          </div>\n          <div *ngIf=\"complete\" class=\"bb-text-align-center\">All items loaded</div>\n        </bb-infinite-scroll-ui>\n      </div>\n    </div>\n  </div>\n</div>\n", components: [{ type: i1.InfiniteScrollComponent, selector: "bb-infinite-scroll-ui", inputs: ["offset", "debounce", "disableScrollEnd"], outputs: ["scrollEnd"] }, { type: i2.AmountComponent, selector: "bb-amount-ui", inputs: ["currency", "showPlusSign", "mapCurrency", "showPercent", "abbreviate", "decimalPlaces", "trailingZeroes", "amount"] }, { type: i3.LoadingIndicatorComponent, selector: "bb-loading-indicator-ui", inputs: ["text", "loaderSize", "showDelay", "hasBackground", "inline"] }], directives: [{ type: i4.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { type: i4.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
InfiniteScrollShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
], InfiniteScrollShowcaseComponent);
export { InfiniteScrollShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InfiniteScrollShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-infinite-scroll-ui-showcase',
                    templateUrl: './infinite-scroll-showcase.component.html',
                }]
        }] });
