// @backbase/ui-ang/esm2015/_example/product-item-loan/showcase/product-item-loan-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ProductItemLoanShowcaseComponent } from './product-item-loan-showcase.component';
import { ProductItemLoanModule } from '@backbase/ui-ang/product-item-loan';
import * as i0 from "@angular/core";
export class ProductItemLoanShowcaseModule {
}
ProductItemLoanShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemLoanShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductItemLoanShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemLoanShowcaseModule, declarations: [ProductItemLoanShowcaseComponent], imports: [CommonModule, ProductItemLoanModule] });
ProductItemLoanShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemLoanShowcaseModule, imports: [[CommonModule, ProductItemLoanModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemLoanShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ProductItemLoanModule],
                    declarations: [ProductItemLoanShowcaseComponent],
                    entryComponents: [ProductItemLoanShowcaseComponent],
                }]
        }] });
