// @backbase/ui-ang/esm2015/_example/collapsible-accordion/showcase/collapsible-accordion-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CollapsibleAccordionShowcaseComponent } from './collapsible-accordion-showcase.component';
import { CollapsibleAccordionModule } from '@backbase/ui-ang/collapsible-accordion';
import { IconModule } from '@backbase/ui-ang/icon';
import { CollapsibleCardModule } from '@backbase/ui-ang/collapsible-card';
import { CollapsibleModule } from '@backbase/ui-ang/collapsible';
import { ButtonModule } from '@backbase/ui-ang/button';
import * as i0 from "@angular/core";
export class CollapsibleAccordionShowcaseModule {
}
CollapsibleAccordionShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CollapsibleAccordionShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CollapsibleAccordionShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CollapsibleAccordionShowcaseModule, declarations: [CollapsibleAccordionShowcaseComponent], imports: [CommonModule,
        CollapsibleAccordionModule,
        IconModule,
        CollapsibleCardModule,
        CollapsibleModule,
        ButtonModule] });
CollapsibleAccordionShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CollapsibleAccordionShowcaseModule, imports: [[
            CommonModule,
            CollapsibleAccordionModule,
            IconModule,
            CollapsibleCardModule,
            CollapsibleModule,
            ButtonModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CollapsibleAccordionShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        CollapsibleAccordionModule,
                        IconModule,
                        CollapsibleCardModule,
                        CollapsibleModule,
                        ButtonModule,
                    ],
                    declarations: [CollapsibleAccordionShowcaseComponent],
                    entryComponents: [CollapsibleAccordionShowcaseComponent],
                }]
        }] });
