// @backbase/consolidated-approvals-beta-http-ang/esm2015/api/contactApprovals.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const ContactApprovalsHttpServicePostFilterContactApprovalsMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/approvals-private/contacts/filter",
        method: "POST",
        responses: [
            {
                status: 200,
                body: [{
                        "contactInfo": {
                            "id": "0855e686-d31e-4216-b3dd-5d66161d5312",
                            "creatorBbId": "john@example.com",
                            "data": {
                                "id": "0855e686-d31e-4216-b3dd-5d66161d536d",
                                "name": "John Doe",
                                "alias": "John",
                                "category": "Employee",
                                "contactPerson": "Jane Doe",
                                "phoneNumber": "055512345678",
                                "emailId": "john@example.com",
                                "addressLine1": "Backbase enterprise",
                                "addressLine2": "",
                                "streetName": "Jacob Bontiusplaats 9",
                                "town": "Amsterdam",
                                "postCode": "1018 LL",
                                "countrySubDivision": "North Holland",
                                "country": "NL",
                                "accounts": [{
                                        "name": "Jane Doe",
                                        "alias": "Jane",
                                        "accountType": "SAVINGS",
                                        "IBAN": "FI21 1234 5600 0007 85",
                                        "bankName": "Test Bank",
                                        "bankAddressLine1": "QA Bank",
                                        "bankAddressLine2": "",
                                        "bankStreetName": "Jodenbreestraat 96",
                                        "bankTown": "Amsterdam",
                                        "bankPostCode": "1011NS",
                                        "bankCountrySubDivision": "North Holland",
                                        "bankCountry": "NL",
                                        "accountHolderAddressLine1": "",
                                        "accountHolderAddressLine2": "",
                                        "accountHolderStreetName": "Woestduinstraat 114-3",
                                        "accountHolderTown": "Amsterdam",
                                        "accountHolderPostCode": "1059 SP",
                                        "accountHolderCountrySubDivision": "North Holland",
                                        "accountHolderCountry": "NL"
                                    }],
                                "accessContextScope": "SA"
                            },
                            "creationTime": "2017-05-12T14:33:13.452Z",
                            "action": "CREATE",
                            "status": "ENTERED"
                        },
                        "approvalInfo": {
                            "id": "b3e4692d-6772-4fb2-9446-b8f4607f49b1",
                            "userId": "606d4532-f8d9-4a5f-840a-887baf88fa24",
                            "userFullName": "Paddy Brian",
                            "externalUserId": "paddy-brian",
                            "serviceAgreementId": "c3f62ff9-2894-47d9-a34a-2c8b54e6e9be",
                            "itemId": "0855e686-d31e-4216-b3dd-5d66161d5312",
                            "resource": "Contacts",
                            "function": "Contacts",
                            "status": "PENDING",
                            "createdAt": "2018-07-22T13:44:02Z",
                            "records": [{
                                    "id": "59f23015-9007-4391-a604-1d17e85ab3c0",
                                    "userId": "606d4532-f8d9-4a5f-840a-887baf88fa24",
                                    "serviceAgreementId": "c3f62ff9-2894-47d9-a34a-2c8b54e6e9be",
                                    "externalUserId": "robin-green",
                                    "userFullName": "Robin Green",
                                    "approvalTypeId": "ed47a5d7-1626-4e8c-819c-8c1851cb0357",
                                    "status": "APPROVED",
                                    "createdAt": "2018-01-22T13:44:02Z"
                                }],
                            "approvalTypes": [{
                                    "id": "ed47a5d7-1626-4e8c-819c-8c1851cb0357",
                                    "name": "A",
                                    "description": "1st approval",
                                    "rank": 100
                                }, {
                                    "id": "5fa97eb5-277f-4b2c-b05d-f13d630e05c8",
                                    "name": "B",
                                    "description": "2nd approval",
                                    "rank": 101
                                }],
                            "policy": {
                                "id": "6546c424-d3b1-4848-bbda-31cd44ab6915",
                                "name": "Default policy",
                                "description": "A default policy that requires two approvals of different types: A + B",
                                "items": [{
                                        "approvalTypeId": "ed47a5d7-1626-4e8c-819c-8c1851cb0357",
                                        "approvalType": {
                                            "id": "ed47a5d7-1626-4e8c-819c-8c1851cb0357",
                                            "name": "A",
                                            "description": "1st approval",
                                            "rank": 100
                                        },
                                        "numberOfApprovals": 1
                                    }, {
                                        "approvalTypeId": "5fa97eb5-277f-4b2c-b05d-f13d630e05c8",
                                        "approvalType": {
                                            "id": "5fa97eb5-277f-4b2c-b05d-f13d630e05c8",
                                            "name": "B",
                                            "description": "2nd approval",
                                            "rank": 101
                                        },
                                        "numberOfApprovals": 1
                                    }]
                            },
                            "canApprove": true,
                            "canReject": true,
                            "finalApprover": false
                        }
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const ContactApprovalsHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v2/approvals-private/contacts/filter",
        method: "POST",
        responses: [
            {
                status: 200,
                body: [{
                        "contactInfo": {
                            "id": "0855e686-d31e-4216-b3dd-5d66161d5312",
                            "creatorBbId": "john@example.com",
                            "data": {
                                "id": "0855e686-d31e-4216-b3dd-5d66161d536d",
                                "name": "John Doe",
                                "alias": "John",
                                "category": "Employee",
                                "contactPerson": "Jane Doe",
                                "phoneNumber": "055512345678",
                                "emailId": "john@example.com",
                                "addressLine1": "Backbase enterprise",
                                "addressLine2": "",
                                "streetName": "Jacob Bontiusplaats 9",
                                "town": "Amsterdam",
                                "postCode": "1018 LL",
                                "countrySubDivision": "North Holland",
                                "country": "NL",
                                "accounts": [{
                                        "name": "Jane Doe",
                                        "alias": "Jane",
                                        "accountType": "SAVINGS",
                                        "IBAN": "FI21 1234 5600 0007 85",
                                        "bankName": "Test Bank",
                                        "bankAddressLine1": "QA Bank",
                                        "bankAddressLine2": "",
                                        "bankStreetName": "Jodenbreestraat 96",
                                        "bankTown": "Amsterdam",
                                        "bankPostCode": "1011NS",
                                        "bankCountrySubDivision": "North Holland",
                                        "bankCountry": "NL",
                                        "accountHolderAddressLine1": "",
                                        "accountHolderAddressLine2": "",
                                        "accountHolderStreetName": "Woestduinstraat 114-3",
                                        "accountHolderTown": "Amsterdam",
                                        "accountHolderPostCode": "1059 SP",
                                        "accountHolderCountrySubDivision": "North Holland",
                                        "accountHolderCountry": "NL"
                                    }],
                                "accessContextScope": "SA"
                            },
                            "creationTime": "2017-05-12T14:33:13.452Z",
                            "action": "CREATE",
                            "status": "ENTERED"
                        },
                        "approvalInfo": {
                            "id": "b3e4692d-6772-4fb2-9446-b8f4607f49b1",
                            "userId": "606d4532-f8d9-4a5f-840a-887baf88fa24",
                            "userFullName": "Paddy Brian",
                            "externalUserId": "paddy-brian",
                            "serviceAgreementId": "c3f62ff9-2894-47d9-a34a-2c8b54e6e9be",
                            "itemId": "0855e686-d31e-4216-b3dd-5d66161d5312",
                            "resource": "Contacts",
                            "function": "Contacts",
                            "status": "PENDING",
                            "createdAt": "2018-07-22T13:44:02Z",
                            "records": [{
                                    "id": "59f23015-9007-4391-a604-1d17e85ab3c0",
                                    "userId": "606d4532-f8d9-4a5f-840a-887baf88fa24",
                                    "serviceAgreementId": "c3f62ff9-2894-47d9-a34a-2c8b54e6e9be",
                                    "externalUserId": "robin-green",
                                    "userFullName": "Robin Green",
                                    "approvalTypeId": "ed47a5d7-1626-4e8c-819c-8c1851cb0357",
                                    "status": "APPROVED",
                                    "createdAt": "2018-01-22T13:44:02Z"
                                }],
                            "approvalTypes": [{
                                    "id": "ed47a5d7-1626-4e8c-819c-8c1851cb0357",
                                    "name": "A",
                                    "description": "1st approval",
                                    "rank": 100
                                }, {
                                    "id": "5fa97eb5-277f-4b2c-b05d-f13d630e05c8",
                                    "name": "B",
                                    "description": "2nd approval",
                                    "rank": 101
                                }],
                            "policy": {
                                "id": "6546c424-d3b1-4848-bbda-31cd44ab6915",
                                "name": "Default policy",
                                "description": "A default policy that requires two approvals of different types: A + B",
                                "items": [{
                                        "approvalTypeId": "ed47a5d7-1626-4e8c-819c-8c1851cb0357",
                                        "approvalType": {
                                            "id": "ed47a5d7-1626-4e8c-819c-8c1851cb0357",
                                            "name": "A",
                                            "description": "1st approval",
                                            "rank": 100
                                        },
                                        "numberOfApprovals": 1
                                    }, {
                                        "approvalTypeId": "5fa97eb5-277f-4b2c-b05d-f13d630e05c8",
                                        "approvalType": {
                                            "id": "5fa97eb5-277f-4b2c-b05d-f13d630e05c8",
                                            "name": "B",
                                            "description": "2nd approval",
                                            "rank": 101
                                        },
                                        "numberOfApprovals": 1
                                    }]
                            },
                            "canApprove": true,
                            "canReject": true,
                            "finalApprover": false
                        }
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
]);
