// @backbase/consolidated-approvals-beta-http-ang/esm2015/api/paymentApprovals.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const PaymentApprovalsHttpServicePostFilterPaymentApprovalsMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/approvals-private/payments/filter",
        method: "POST",
        responses: [
            {
                status: 200,
                body: [{
                        "paymentOrderInfo": {
                            "id": "cddf639a-3cbe-4162-a925-dd8909b43bfb",
                            "approvalId": "12aacd0c-2648-4452-b04f-3fc160bda571",
                            "status": "ENTERED",
                            "createdBy": "u0001012",
                            "createdAt": "2022-04-01T11:27:59.416190Z",
                            "updatedBy": "u0001012",
                            "updatedAt": "2022-04-01T11:27:59.416190Z",
                            "actions": ["APPROVE", "REJECT"],
                            "intraLegalEntity": false,
                            "originatorAccountCurrency": "EUR",
                            "originator": {
                                "name": "vPBxYC",
                                "role": "DEBTOR",
                                "postalAddress": {
                                    "addressLine1": "nwRlOk",
                                    "addressLine2": "mzZWeZbz",
                                    "streetName": "vKlNcfHRUUEiJm",
                                    "postCode": "177135",
                                    "town": "gXNt",
                                    "countrySubDivision": "dMptgmoC",
                                    "country": "AQ"
                                }
                            },
                            "originatorAccount": {
                                "arrangementId": "903309c5-1b07-4043-b46b-4a022bdb1b41",
                                "externalArrangementId": "0000000000000001bwdCoVXvNuDrYTSw",
                                "identification": {
                                    "identification": "LT526319689655711037",
                                    "schemeName": "IBAN"
                                }
                            },
                            "batchBooking": false,
                            "instructionPriority": "NORM",
                            "requestedExecutionDate": "2022-04-01",
                            "paymentMode": "SINGLE",
                            "paymentType": "SEPA_CREDIT_TRANSFER",
                            "transferTransactionInformation": {
                                "counterparty": {
                                    "name": "Sleek Copper Shirt",
                                    "role": "CREDITOR"
                                },
                                "counterpartyAccount": {
                                    "identification": {
                                        "identification": "IE06YJVA15965129386257",
                                        "schemeName": "IBAN"
                                    }
                                },
                                "instructedAmount": {
                                    "amount": "50.00000",
                                    "currencyCode": "EUR"
                                }
                            },
                            "totalAmount": {
                                "amount": "50.00000",
                                "currencyCode": "EUR"
                            }
                        },
                        "approvalInfo": {
                            "id": "12aacd0c-2648-4452-b04f-3fc160bda571",
                            "userId": "2226d9be-4323-40e8-acec-988eab4777f3",
                            "serviceAgreementId": "8a49bc667fe4c35e017fe4e16b1f016a",
                            "itemId": "cddf639a-3cbe-4162-a925-dd8909b43bfb",
                            "dataAccessItemId": "903309c5-1b07-4043-b46b-4a022bdb1b41",
                            "resource": "Payments",
                            "function": "SEPA CT",
                            "status": "PENDING",
                            "canApprove": true,
                            "canReject": true,
                            "finalApprover": false,
                            "createdAt": "2022-04-01T11:27:58.800336Z"
                        }
                    }, {
                        "paymentOrderInfo": {
                            "id": "f4bc422a-1d7d-4a3f-aad8-52882fe2f2dd",
                            "approvalId": "98b07ac0-2360-453b-8dc9-6af7b5228bec",
                            "status": "ENTERED",
                            "createdBy": "u0001012",
                            "createdAt": "2022-04-01T11:28:00.643542Z",
                            "updatedBy": "u0001012",
                            "updatedAt": "2022-04-01T11:28:00.643542Z",
                            "actions": ["APPROVE", "REJECT"],
                            "intraLegalEntity": false,
                            "originatorAccountCurrency": "EUR",
                            "originator": {
                                "name": "vPBxYC",
                                "role": "DEBTOR",
                                "postalAddress": {
                                    "addressLine1": "nwRlOk",
                                    "addressLine2": "mzZWeZbz",
                                    "streetName": "vKlNcfHRUUEiJm",
                                    "postCode": "177135",
                                    "town": "gXNt",
                                    "countrySubDivision": "dMptgmoC",
                                    "country": "AQ"
                                }
                            },
                            "originatorAccount": {
                                "arrangementId": "903309c5-1b07-4043-b46b-4a022bdb1b41",
                                "externalArrangementId": "0000000000000001bwdCoVXvNuDrYTSw",
                                "identification": {
                                    "identification": "LT526319689655711037",
                                    "schemeName": "IBAN"
                                }
                            },
                            "batchBooking": false,
                            "instructionPriority": "NORM",
                            "requestedExecutionDate": "2022-04-01",
                            "paymentMode": "SINGLE",
                            "paymentType": "SEPA_CREDIT_TRANSFER",
                            "transferTransactionInformation": {
                                "counterparty": {
                                    "name": "Sleek Wooden Table",
                                    "role": "CREDITOR"
                                },
                                "counterpartyAccount": {
                                    "identification": {
                                        "identification": "SM37C1125812374QXIKVA5ZKH96",
                                        "schemeName": "IBAN"
                                    }
                                },
                                "instructedAmount": {
                                    "amount": "50.00000",
                                    "currencyCode": "EUR"
                                }
                            },
                            "totalAmount": {
                                "amount": "50.00000",
                                "currencyCode": "EUR"
                            }
                        },
                        "approvalInfo": {
                            "id": "98b07ac0-2360-453b-8dc9-6af7b5228bec",
                            "userId": "2226d9be-4323-40e8-acec-988eab4777f3",
                            "serviceAgreementId": "8a49bc667fe4c35e017fe4e16cc60172",
                            "itemId": "f4bc422a-1d7d-4a3f-aad8-52882fe2f2dd",
                            "dataAccessItemId": "903309c5-1b07-4043-b46b-4a022bdb1b41",
                            "resource": "Payments",
                            "function": "SEPA CT",
                            "amountDetails": {
                                "currencyCode": "EUR",
                                "amount": "50.00000"
                            },
                            "status": "PENDING",
                            "records": [{
                                    "id": "59f23015-9007-4391-a604-1d17e85ab3c0",
                                    "userId": "606d4532-f8d9-4a5f-840a-887baf88fa24",
                                    "serviceAgreementId": "c3f62ff9-2894-47d9-a34a-2c8b54e6e9be",
                                    "approvalTypeId": "8b4cf4ba-1b6d-43fe-bb44-3c762e0ba5c1",
                                    "status": "APPROVED",
                                    "createdAt": "2018-01-22T13:44:02Z"
                                }],
                            "approvalTypes": [{
                                    "id": "8b4cf4ba-1b6d-43fe-bb44-3c762e0ba5c1",
                                    "name": "A",
                                    "description": "1st approval",
                                    "rank": 100
                                }, {
                                    "id": "5fa97eb5-277f-4b2c-b05d-f13d630e05c8",
                                    "name": "B",
                                    "description": "2nd approval",
                                    "rank": 101
                                }],
                            "policy": {
                                "id": "6546c424-d3b1-4848-bbda-31cd44ab6915",
                                "name": "Default policy",
                                "description": "A default policy that requires two approvals of different types: A + B",
                                "items": [{
                                        "approvalTypeId": "8b4cf4ba-1b6d-43fe-bb44-3c762e0ba5c1",
                                        "approvalType": {
                                            "id": "8b4cf4ba-1b6d-43fe-bb44-3c762e0ba5c1",
                                            "name": "A",
                                            "description": "1st approval",
                                            "rank": 100
                                        },
                                        "numberOfApprovals": 1
                                    }, {
                                        "approvalTypeId": "5fa97eb5-277f-4b2c-b05d-f13d630e05c8",
                                        "approvalType": {
                                            "id": "5fa97eb5-277f-4b2c-b05d-f13d630e05c8",
                                            "name": "B",
                                            "description": "2nd approval",
                                            "rank": 101
                                        },
                                        "numberOfApprovals": 1
                                    }]
                            },
                            "canApprove": true,
                            "canReject": true,
                            "finalApprover": false,
                            "createdAt": "2022-04-01T11:28:00.418198Z"
                        }
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const PaymentApprovalsHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v2/approvals-private/payments/filter",
        method: "POST",
        responses: [
            {
                status: 200,
                body: [{
                        "paymentOrderInfo": {
                            "id": "cddf639a-3cbe-4162-a925-dd8909b43bfb",
                            "approvalId": "12aacd0c-2648-4452-b04f-3fc160bda571",
                            "status": "ENTERED",
                            "createdBy": "u0001012",
                            "createdAt": "2022-04-01T11:27:59.416190Z",
                            "updatedBy": "u0001012",
                            "updatedAt": "2022-04-01T11:27:59.416190Z",
                            "actions": ["APPROVE", "REJECT"],
                            "intraLegalEntity": false,
                            "originatorAccountCurrency": "EUR",
                            "originator": {
                                "name": "vPBxYC",
                                "role": "DEBTOR",
                                "postalAddress": {
                                    "addressLine1": "nwRlOk",
                                    "addressLine2": "mzZWeZbz",
                                    "streetName": "vKlNcfHRUUEiJm",
                                    "postCode": "177135",
                                    "town": "gXNt",
                                    "countrySubDivision": "dMptgmoC",
                                    "country": "AQ"
                                }
                            },
                            "originatorAccount": {
                                "arrangementId": "903309c5-1b07-4043-b46b-4a022bdb1b41",
                                "externalArrangementId": "0000000000000001bwdCoVXvNuDrYTSw",
                                "identification": {
                                    "identification": "LT526319689655711037",
                                    "schemeName": "IBAN"
                                }
                            },
                            "batchBooking": false,
                            "instructionPriority": "NORM",
                            "requestedExecutionDate": "2022-04-01",
                            "paymentMode": "SINGLE",
                            "paymentType": "SEPA_CREDIT_TRANSFER",
                            "transferTransactionInformation": {
                                "counterparty": {
                                    "name": "Sleek Copper Shirt",
                                    "role": "CREDITOR"
                                },
                                "counterpartyAccount": {
                                    "identification": {
                                        "identification": "IE06YJVA15965129386257",
                                        "schemeName": "IBAN"
                                    }
                                },
                                "instructedAmount": {
                                    "amount": "50.00000",
                                    "currencyCode": "EUR"
                                }
                            },
                            "totalAmount": {
                                "amount": "50.00000",
                                "currencyCode": "EUR"
                            }
                        },
                        "approvalInfo": {
                            "id": "12aacd0c-2648-4452-b04f-3fc160bda571",
                            "userId": "2226d9be-4323-40e8-acec-988eab4777f3",
                            "serviceAgreementId": "8a49bc667fe4c35e017fe4e16b1f016a",
                            "itemId": "cddf639a-3cbe-4162-a925-dd8909b43bfb",
                            "dataAccessItemId": "903309c5-1b07-4043-b46b-4a022bdb1b41",
                            "resource": "Payments",
                            "function": "SEPA CT",
                            "status": "PENDING",
                            "canApprove": true,
                            "canReject": true,
                            "finalApprover": false,
                            "createdAt": "2022-04-01T11:27:58.800336Z"
                        }
                    }, {
                        "paymentOrderInfo": {
                            "id": "f4bc422a-1d7d-4a3f-aad8-52882fe2f2dd",
                            "approvalId": "98b07ac0-2360-453b-8dc9-6af7b5228bec",
                            "status": "ENTERED",
                            "createdBy": "u0001012",
                            "createdAt": "2022-04-01T11:28:00.643542Z",
                            "updatedBy": "u0001012",
                            "updatedAt": "2022-04-01T11:28:00.643542Z",
                            "actions": ["APPROVE", "REJECT"],
                            "intraLegalEntity": false,
                            "originatorAccountCurrency": "EUR",
                            "originator": {
                                "name": "vPBxYC",
                                "role": "DEBTOR",
                                "postalAddress": {
                                    "addressLine1": "nwRlOk",
                                    "addressLine2": "mzZWeZbz",
                                    "streetName": "vKlNcfHRUUEiJm",
                                    "postCode": "177135",
                                    "town": "gXNt",
                                    "countrySubDivision": "dMptgmoC",
                                    "country": "AQ"
                                }
                            },
                            "originatorAccount": {
                                "arrangementId": "903309c5-1b07-4043-b46b-4a022bdb1b41",
                                "externalArrangementId": "0000000000000001bwdCoVXvNuDrYTSw",
                                "identification": {
                                    "identification": "LT526319689655711037",
                                    "schemeName": "IBAN"
                                }
                            },
                            "batchBooking": false,
                            "instructionPriority": "NORM",
                            "requestedExecutionDate": "2022-04-01",
                            "paymentMode": "SINGLE",
                            "paymentType": "SEPA_CREDIT_TRANSFER",
                            "transferTransactionInformation": {
                                "counterparty": {
                                    "name": "Sleek Wooden Table",
                                    "role": "CREDITOR"
                                },
                                "counterpartyAccount": {
                                    "identification": {
                                        "identification": "SM37C1125812374QXIKVA5ZKH96",
                                        "schemeName": "IBAN"
                                    }
                                },
                                "instructedAmount": {
                                    "amount": "50.00000",
                                    "currencyCode": "EUR"
                                }
                            },
                            "totalAmount": {
                                "amount": "50.00000",
                                "currencyCode": "EUR"
                            }
                        },
                        "approvalInfo": {
                            "id": "98b07ac0-2360-453b-8dc9-6af7b5228bec",
                            "userId": "2226d9be-4323-40e8-acec-988eab4777f3",
                            "serviceAgreementId": "8a49bc667fe4c35e017fe4e16cc60172",
                            "itemId": "f4bc422a-1d7d-4a3f-aad8-52882fe2f2dd",
                            "dataAccessItemId": "903309c5-1b07-4043-b46b-4a022bdb1b41",
                            "resource": "Payments",
                            "function": "SEPA CT",
                            "amountDetails": {
                                "currencyCode": "EUR",
                                "amount": "50.00000"
                            },
                            "status": "PENDING",
                            "records": [{
                                    "id": "59f23015-9007-4391-a604-1d17e85ab3c0",
                                    "userId": "606d4532-f8d9-4a5f-840a-887baf88fa24",
                                    "serviceAgreementId": "c3f62ff9-2894-47d9-a34a-2c8b54e6e9be",
                                    "approvalTypeId": "8b4cf4ba-1b6d-43fe-bb44-3c762e0ba5c1",
                                    "status": "APPROVED",
                                    "createdAt": "2018-01-22T13:44:02Z"
                                }],
                            "approvalTypes": [{
                                    "id": "8b4cf4ba-1b6d-43fe-bb44-3c762e0ba5c1",
                                    "name": "A",
                                    "description": "1st approval",
                                    "rank": 100
                                }, {
                                    "id": "5fa97eb5-277f-4b2c-b05d-f13d630e05c8",
                                    "name": "B",
                                    "description": "2nd approval",
                                    "rank": 101
                                }],
                            "policy": {
                                "id": "6546c424-d3b1-4848-bbda-31cd44ab6915",
                                "name": "Default policy",
                                "description": "A default policy that requires two approvals of different types: A + B",
                                "items": [{
                                        "approvalTypeId": "8b4cf4ba-1b6d-43fe-bb44-3c762e0ba5c1",
                                        "approvalType": {
                                            "id": "8b4cf4ba-1b6d-43fe-bb44-3c762e0ba5c1",
                                            "name": "A",
                                            "description": "1st approval",
                                            "rank": 100
                                        },
                                        "numberOfApprovals": 1
                                    }, {
                                        "approvalTypeId": "5fa97eb5-277f-4b2c-b05d-f13d630e05c8",
                                        "approvalType": {
                                            "id": "5fa97eb5-277f-4b2c-b05d-f13d630e05c8",
                                            "name": "B",
                                            "description": "2nd approval",
                                            "rank": 101
                                        },
                                        "numberOfApprovals": 1
                                    }]
                            },
                            "canApprove": true,
                            "canReject": true,
                            "finalApprover": false,
                            "createdAt": "2022-04-01T11:28:00.418198Z"
                        }
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
]);
