// @backbase/ui-ang/esm2015/table/table.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { TableDirective } from './directives/table.directive';
import { TableRowDirective } from './directives/table-row.directive';
import { TableSortableDirective } from './directives/table-sortable.directive';
import { TableFocusDirective } from './directives/table-focus.directive';
import * as i0 from "@angular/core";
const directives = [TableRowDirective, TableDirective, TableSortableDirective, TableFocusDirective];
export class TableModule {
}
TableModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
TableModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableModule, declarations: [TableRowDirective, TableDirective, TableSortableDirective, TableFocusDirective], imports: [CommonModule], exports: [TableRowDirective, TableDirective, TableSortableDirective, TableFocusDirective] });
TableModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [...directives],
                    exports: [...directives],
                }]
        }] });
