// @backbase/connect-external-accounts-journey-ang/esm2015/lib/components/connect-external-accounts-journey-ui.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { AlertModule } from '@backbase/ui-ang/alert';
import { BadgeModule } from '@backbase/ui-ang/badge';
import { ButtonModule } from '@backbase/ui-ang/button';
import { CollapsibleModule } from '@backbase/ui-ang/collapsible';
import { CurrencyInputModule } from '@backbase/ui-ang/currency-input';
import { DropdownMenuModule } from '@backbase/ui-ang/dropdown-menu';
import { DropdownSingleSelectModule } from '@backbase/ui-ang/dropdown-single-select';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { HeaderModule } from '@backbase/ui-ang/header';
import { IconModule } from '@backbase/ui-ang/icon';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { KeyboardClickModule } from '@backbase/ui-ang/keyboard-click-directive';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { ModalModule } from '@backbase/ui-ang/modal';
import { NotificationModule } from '@backbase/ui-ang/notification';
import { TextareaModule } from '@backbase/ui-ang/textarea';
import { AccountDetailModalComponent } from './account-details/account-details.component';
import { AccountInformationComponent } from './account-information/account-information.component';
import { AccountsListComponent } from './accounts-list/accounts-list.component';
import { ActivateExternalAccountComponent } from './activate-account-form/activate-account-form.component';
import { ConfirmationDialogComponent } from './confirmation-dialog/confirmation-dialog.component';
import { AccountDetailModalHeaderComponent } from './details-header/details-header.component';
import { EmptyListComponent } from './empty-list/empty-list.component';
import { LinkAccountInfoComponent } from './link-a2a-steps/link-a2a-steps.component';
import { LinkAccountModalComponent } from './link-account-form/link-account-form.component';
import { PendingActivationAccountDetailsComponent } from './pending-activation-account-details/pending-activation-account-details.component';
import * as i0 from "@angular/core";
const components = [
    LinkAccountModalComponent,
    LinkAccountInfoComponent,
    AccountsListComponent,
    AccountDetailModalComponent,
    ActivateExternalAccountComponent,
    EmptyListComponent,
    AccountDetailModalHeaderComponent,
    AccountInformationComponent,
    PendingActivationAccountDetailsComponent,
    ConfirmationDialogComponent,
];
const uiModules = [
    EmptyStateModule,
    ButtonModule,
    ModalModule,
    InputTextModule,
    DropdownSingleSelectModule,
    InputValidationMessageModule,
    IconModule,
    AlertModule,
    NotificationModule,
    BadgeModule,
    LoadingIndicatorModule,
    CurrencyInputModule,
    CollapsibleModule,
    DropdownMenuModule,
    HeaderModule,
    LoadButtonModule,
    AlertModule,
    TextareaModule,
    KeyboardClickModule,
];
export class ConnectExternalAccountsJourneyUiModule {
}
/** @nocollapse */ ConnectExternalAccountsJourneyUiModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ConnectExternalAccountsJourneyUiModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ ConnectExternalAccountsJourneyUiModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ConnectExternalAccountsJourneyUiModule, declarations: [LinkAccountModalComponent,
        LinkAccountInfoComponent,
        AccountsListComponent,
        AccountDetailModalComponent,
        ActivateExternalAccountComponent,
        EmptyListComponent,
        AccountDetailModalHeaderComponent,
        AccountInformationComponent,
        PendingActivationAccountDetailsComponent,
        ConfirmationDialogComponent], imports: [CommonModule, ReactiveFormsModule, EmptyStateModule,
        ButtonModule,
        ModalModule,
        InputTextModule,
        DropdownSingleSelectModule,
        InputValidationMessageModule,
        IconModule,
        AlertModule,
        NotificationModule,
        BadgeModule,
        LoadingIndicatorModule,
        CurrencyInputModule,
        CollapsibleModule,
        DropdownMenuModule,
        HeaderModule,
        LoadButtonModule,
        AlertModule,
        TextareaModule,
        KeyboardClickModule], exports: [EmptyStateModule,
        ButtonModule,
        ModalModule,
        InputTextModule,
        DropdownSingleSelectModule,
        InputValidationMessageModule,
        IconModule,
        AlertModule,
        NotificationModule,
        BadgeModule,
        LoadingIndicatorModule,
        CurrencyInputModule,
        CollapsibleModule,
        DropdownMenuModule,
        HeaderModule,
        LoadButtonModule,
        AlertModule,
        TextareaModule,
        KeyboardClickModule, LinkAccountModalComponent,
        LinkAccountInfoComponent,
        AccountsListComponent,
        AccountDetailModalComponent,
        ActivateExternalAccountComponent,
        EmptyListComponent,
        AccountDetailModalHeaderComponent,
        AccountInformationComponent,
        PendingActivationAccountDetailsComponent,
        ConfirmationDialogComponent] });
/** @nocollapse */ ConnectExternalAccountsJourneyUiModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ConnectExternalAccountsJourneyUiModule, imports: [[CommonModule, ReactiveFormsModule, ...uiModules], EmptyStateModule,
        ButtonModule,
        ModalModule,
        InputTextModule,
        DropdownSingleSelectModule,
        InputValidationMessageModule,
        IconModule,
        AlertModule,
        NotificationModule,
        BadgeModule,
        LoadingIndicatorModule,
        CurrencyInputModule,
        CollapsibleModule,
        DropdownMenuModule,
        HeaderModule,
        LoadButtonModule,
        AlertModule,
        TextareaModule,
        KeyboardClickModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ConnectExternalAccountsJourneyUiModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components],
                    imports: [CommonModule, ReactiveFormsModule, ...uiModules],
                    exports: [...uiModules, ...components],
                }]
        }] });
