// @backbase/payment-draft-http-ang/esm2015/api/paymentOrderDrafts.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { PAYMENT_DRAFT_BASE_PATH } from '../variables';
import { PaymentDraftConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class PaymentOrderDraftsHttpService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new PaymentDraftConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    deleteDraftPaymentOrder(requestParameters, observe = 'body', reportProgress = false, options) {
        const _draftPaymentOrderId = requestParameters["draftPaymentOrderId"];
        if (_draftPaymentOrderId === null || _draftPaymentOrderId === undefined) {
            throw new Error('Required parameter draftPaymentOrderId was null or undefined when calling deleteDraftPaymentOrder.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.delete(`${this.configuration.basePath}/client-api/v2/payment-order-drafts/${encodeURIComponent(String(_draftPaymentOrderId))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    deleteDraftPaymentOrderUrl(requestParameters) {
        const _draftPaymentOrderId = requestParameters["draftPaymentOrderId"];
        if (_draftPaymentOrderId === null || _draftPaymentOrderId === undefined) {
            throw new Error('Required parameter draftPaymentOrderId was null or undefined when calling deleteDraftPaymentOrder.');
        }
        return `${this.configuration.basePath}/client-api/v2/payment-order-drafts/${encodeURIComponent(String(_draftPaymentOrderId))}`;
    }
    getDraftPaymentOrder(requestParameters, observe = 'body', reportProgress = false, options) {
        const _draftPaymentOrderId = requestParameters["draftPaymentOrderId"];
        if (_draftPaymentOrderId === null || _draftPaymentOrderId === undefined) {
            throw new Error('Required parameter draftPaymentOrderId was null or undefined when calling getDraftPaymentOrder.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/payment-order-drafts/${encodeURIComponent(String(_draftPaymentOrderId))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getDraftPaymentOrderUrl(requestParameters) {
        const _draftPaymentOrderId = requestParameters["draftPaymentOrderId"];
        if (_draftPaymentOrderId === null || _draftPaymentOrderId === undefined) {
            throw new Error('Required parameter draftPaymentOrderId was null or undefined when calling getDraftPaymentOrder.');
        }
        return `${this.configuration.basePath}/client-api/v2/payment-order-drafts/${encodeURIComponent(String(_draftPaymentOrderId))}`;
    }
    getPaymentOrdersDrafts(requestParameters, observe = 'body', reportProgress = false, options) {
        const _query = requestParameters["query"];
        const _amountFrom = requestParameters["amountFrom"];
        const _amountTo = requestParameters["amountTo"];
        const _executionDate = requestParameters["executionDate"];
        const _executionDateFrom = requestParameters["executionDateFrom"];
        const _executionDateTo = requestParameters["executionDateTo"];
        const _nextExecutionDate = requestParameters["nextExecutionDate"];
        const _nextExecutionDateFrom = requestParameters["nextExecutionDateFrom"];
        const _nextExecutionDateTo = requestParameters["nextExecutionDateTo"];
        const _paymentMode = requestParameters["paymentMode"];
        const _paymentSetupId = requestParameters["paymentSetupId"];
        const _transferFrequency = requestParameters["transferFrequency"];
        const _from = requestParameters["from"];
        const _cursor = requestParameters["cursor"];
        const _size = requestParameters["size"];
        const _orderBy = requestParameters["orderBy"];
        const _direction = requestParameters["direction"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_query !== undefined && _query !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _query, 'query');
        }
        if (_amountFrom !== undefined && _amountFrom !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _amountFrom, 'amountFrom');
        }
        if (_amountTo !== undefined && _amountTo !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _amountTo, 'amountTo');
        }
        if (_executionDate !== undefined && _executionDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _executionDate, 'executionDate');
        }
        if (_executionDateFrom !== undefined && _executionDateFrom !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _executionDateFrom, 'executionDateFrom');
        }
        if (_executionDateTo !== undefined && _executionDateTo !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _executionDateTo, 'executionDateTo');
        }
        if (_nextExecutionDate !== undefined && _nextExecutionDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _nextExecutionDate, 'nextExecutionDate');
        }
        if (_nextExecutionDateFrom !== undefined && _nextExecutionDateFrom !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _nextExecutionDateFrom, 'nextExecutionDateFrom');
        }
        if (_nextExecutionDateTo !== undefined && _nextExecutionDateTo !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _nextExecutionDateTo, 'nextExecutionDateTo');
        }
        if (_paymentMode !== undefined && _paymentMode !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _paymentMode, 'paymentMode');
        }
        if (_paymentSetupId !== undefined && _paymentSetupId !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _paymentSetupId, 'paymentSetupId');
        }
        if (_transferFrequency !== undefined && _transferFrequency !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _transferFrequency, 'transferFrequency');
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_cursor !== undefined && _cursor !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _cursor, 'cursor');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_orderBy !== undefined && _orderBy !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _orderBy, 'orderBy');
        }
        if (_direction !== undefined && _direction !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _direction, 'direction');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/payment-order-drafts`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getPaymentOrdersDraftsUrl(requestParameters) {
        const _query = requestParameters["query"];
        const _amountFrom = requestParameters["amountFrom"];
        const _amountTo = requestParameters["amountTo"];
        const _executionDate = requestParameters["executionDate"];
        const _executionDateFrom = requestParameters["executionDateFrom"];
        const _executionDateTo = requestParameters["executionDateTo"];
        const _nextExecutionDate = requestParameters["nextExecutionDate"];
        const _nextExecutionDateFrom = requestParameters["nextExecutionDateFrom"];
        const _nextExecutionDateTo = requestParameters["nextExecutionDateTo"];
        const _paymentMode = requestParameters["paymentMode"];
        const _paymentSetupId = requestParameters["paymentSetupId"];
        const _transferFrequency = requestParameters["transferFrequency"];
        const _from = requestParameters["from"];
        const _cursor = requestParameters["cursor"];
        const _size = requestParameters["size"];
        const _orderBy = requestParameters["orderBy"];
        const _direction = requestParameters["direction"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_query !== undefined && _query !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _query, 'query');
        }
        if (_amountFrom !== undefined && _amountFrom !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _amountFrom, 'amountFrom');
        }
        if (_amountTo !== undefined && _amountTo !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _amountTo, 'amountTo');
        }
        if (_executionDate !== undefined && _executionDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _executionDate, 'executionDate');
        }
        if (_executionDateFrom !== undefined && _executionDateFrom !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _executionDateFrom, 'executionDateFrom');
        }
        if (_executionDateTo !== undefined && _executionDateTo !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _executionDateTo, 'executionDateTo');
        }
        if (_nextExecutionDate !== undefined && _nextExecutionDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _nextExecutionDate, 'nextExecutionDate');
        }
        if (_nextExecutionDateFrom !== undefined && _nextExecutionDateFrom !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _nextExecutionDateFrom, 'nextExecutionDateFrom');
        }
        if (_nextExecutionDateTo !== undefined && _nextExecutionDateTo !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _nextExecutionDateTo, 'nextExecutionDateTo');
        }
        if (_paymentMode !== undefined && _paymentMode !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _paymentMode, 'paymentMode');
        }
        if (_paymentSetupId !== undefined && _paymentSetupId !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _paymentSetupId, 'paymentSetupId');
        }
        if (_transferFrequency !== undefined && _transferFrequency !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _transferFrequency, 'transferFrequency');
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_cursor !== undefined && _cursor !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _cursor, 'cursor');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_orderBy !== undefined && _orderBy !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _orderBy, 'orderBy');
        }
        if (_direction !== undefined && _direction !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _direction, 'direction');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/payment-order-drafts${queryString ? `?${queryString}` : ''}`;
    }
    postPaymentOrdersDrafts(requestParameters, observe = 'body', reportProgress = false, options) {
        const _draftInitiatePaymentOrder = requestParameters["draftInitiatePaymentOrder"];
        if (_draftInitiatePaymentOrder === null || _draftInitiatePaymentOrder === undefined) {
            throw new Error('Required parameter draftInitiatePaymentOrder was null or undefined when calling postPaymentOrdersDrafts.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v2/payment-order-drafts`, _draftInitiatePaymentOrder, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    postPaymentOrdersDraftsUrl(requestParameters) {
        const _draftInitiatePaymentOrder = requestParameters["draftInitiatePaymentOrder"];
        if (_draftInitiatePaymentOrder === null || _draftInitiatePaymentOrder === undefined) {
            throw new Error('Required parameter draftInitiatePaymentOrder was null or undefined when calling postPaymentOrdersDrafts.');
        }
        return `${this.configuration.basePath}/client-api/v2/payment-order-drafts`;
    }
    putDraftPaymentOrder(requestParameters, observe = 'body', reportProgress = false, options) {
        const _draftPaymentOrderId = requestParameters["draftPaymentOrderId"];
        if (_draftPaymentOrderId === null || _draftPaymentOrderId === undefined) {
            throw new Error('Required parameter draftPaymentOrderId was null or undefined when calling putDraftPaymentOrder.');
        }
        const _draftInitiatePaymentOrderWithId = requestParameters["draftInitiatePaymentOrderWithId"];
        if (_draftInitiatePaymentOrderWithId === null || _draftInitiatePaymentOrderWithId === undefined) {
            throw new Error('Required parameter draftInitiatePaymentOrderWithId was null or undefined when calling putDraftPaymentOrder.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v2/payment-order-drafts/${encodeURIComponent(String(_draftPaymentOrderId))}`, _draftInitiatePaymentOrderWithId, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    putDraftPaymentOrderUrl(requestParameters) {
        const _draftPaymentOrderId = requestParameters["draftPaymentOrderId"];
        if (_draftPaymentOrderId === null || _draftPaymentOrderId === undefined) {
            throw new Error('Required parameter draftPaymentOrderId was null or undefined when calling putDraftPaymentOrder.');
        }
        const _draftInitiatePaymentOrderWithId = requestParameters["draftInitiatePaymentOrderWithId"];
        if (_draftInitiatePaymentOrderWithId === null || _draftInitiatePaymentOrderWithId === undefined) {
            throw new Error('Required parameter draftInitiatePaymentOrderWithId was null or undefined when calling putDraftPaymentOrder.');
        }
        return `${this.configuration.basePath}/client-api/v2/payment-order-drafts/${encodeURIComponent(String(_draftPaymentOrderId))}`;
    }
}
PaymentOrderDraftsHttpService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentOrderDraftsHttpService, deps: [{ token: i1.HttpClient }, { token: PAYMENT_DRAFT_BASE_PATH, optional: true }, { token: i2.PaymentDraftConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
PaymentOrderDraftsHttpService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentOrderDraftsHttpService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentOrderDraftsHttpService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [PAYMENT_DRAFT_BASE_PATH]
                }] }, { type: i2.PaymentDraftConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
