// @backbase/manage-upcoming-and-historical-payments-journey-ang/esm2015/lib/views/history-payments-view/history-payments-view.component.js
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { map } from 'rxjs/operators';
import { mapPaymentOrdersToUpcomingPayments } from '../../helpers/mappers';
import { ManageUpcomingAndHistoricalPaymentsJourneyConfigService as ConfigService } from '../../services/config.service';
import { PaymentService } from '../../services/payments.service';
import * as i0 from "@angular/core";
import * as i1 from "../../services/payments.service";
import * as i2 from "../../services/config.service";
import * as i3 from "../../components/payment-list-container/payments-list-container.component";
export class HistoryPaymentsViewComponent {
    constructor(dataService, config) {
        this.dataService = dataService;
        this.config = config;
        this.payments$ = this.dataService.payments$.pipe(map(payload => payload && {
            items: mapPaymentOrdersToUpcomingPayments(payload.items),
            totalCount: payload.totalCount,
        }));
    }
    ngOnInit() {
        this.dataService.loadPayments(this.config.historyPaymentParams);
    }
}
/** @nocollapse */ HistoryPaymentsViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HistoryPaymentsViewComponent, deps: [{ token: i1.PaymentService }, { token: i2.ManageUpcomingAndHistoricalPaymentsJourneyConfigService }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ HistoryPaymentsViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: HistoryPaymentsViewComponent, selector: "bb-history-payments-view", providers: [PaymentService], ngImport: i0, template: `
    <div class="card card-body">
      <bb-payments-list-container [payments$]="payments$" [isHistoryPayments]="true"> </bb-payments-list-container>
    </div>
  `, isInline: true, components: [{ type: i3.PaymentsListContainerComponent, selector: "bb-payments-list-container", inputs: ["payments$", "isHistoryPayments"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HistoryPaymentsViewComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-history-payments-view',
                    template: `
    <div class="card card-body">
      <bb-payments-list-container [payments$]="payments$" [isHistoryPayments]="true"> </bb-payments-list-container>
    </div>
  `,
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [PaymentService],
                }]
        }], ctorParameters: function () { return [{ type: i1.PaymentService }, { type: i2.ManageUpcomingAndHistoricalPaymentsJourneyConfigService }]; } });
