// @backbase/manage-payees-journey-ang/esm2015/src/manage-payees-journey.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { ManagePayeesJourneyComponent } from './manage-payees-journey.component';
import { BillpayEnrolmentGuard, BillpayJourneysCommonModule, BillpayPayeeDataService, } from '@backbase/billpay-journeys-common';
import { provideRoutes, RouterModule } from '@angular/router';
import { PayeeListViewComponent } from './views/billpay-payee-list/billpay-payee-list-view.component';
import { PayeeSummaryViewComponent } from './views/billpay-payee-summary/billpay-payee-summary-view.component';
import { EbillsEnrolmentViewComponent } from './views/billpay-ebills-enrolment/billpay-ebills-enrolment-view.component';
import { ManagePayeeViewComponent } from './views/billpay-manage-payee/billpay-manage-payee-view.component';
import { BillPayEnrolmentViewComponent } from './views/billpay-enrolment/billpay-enrolment-view.component';
import { CreateManualPayeeReviewViewComponent } from './views/billpay-create-manual-payee-review/billpay-create-manual-payee-review-view.component';
import { EditManualPayeeReviewViewComponent } from './views/billpay-edit-manual-payee-review/billpay-edit-manual-payee-review-view.component';
import { CreateElectronicPayeeReviewViewComponent } from './views/billpay-create-electronic-payee-review/billpay-create-electronic-payee-review-view.component';
import { EditElectronicPayeeReviewViewComponent } from './views/billpay-edit-electronic-payee-review/billpay-edit-electronic-payee-review-view.component';
import { ManagePayeesJourneyConfigService } from './services/manage-payees-journey-config.service';
import { ManagePayeesJourneyUiModule } from './components/manage-payees-journey-ui.module';
import { BillpayPayverisEbillsEnrolmentWidgetService } from './services/billpay-payveris-ebills-enrolment-widget.service';
import { BillpayPayeeSummaryWidgetService } from './services/billpay-payee-summary-widget.service';
import { BillpayManagePayeeDeactivateGuardService } from './services/billpay-manage-payee-deactivate-guard.service';
import { BillpaySearchPayeeViewComponent } from './views/billpay-search-payee/billpay-search-payee-view.component';
import { BillpayCreateManualPayeeViewComponent } from './views/billpay-create-manual-payee/billpay-create-manual-payee-view.component';
import { BillpayEditManualPayeeViewComponent } from './views/billpay-edit-manual-payee/billpay-edit-manual-payee-view.component';
import { BillpayCreateElectronicPayeeViewComponent } from './views/billpay-create-electronic-payee/billpay-create-electronic-payee-view.component';
import { BillpayEditElectronicPayeeViewComponent } from './views/billpay-edit-electronic-payee/billpay-edit-electronic-payee-view.component';
import { BillpayManagePayeeDataService } from './services/billpay-manage-payee-data.service';
import { BillpayManagePayeeStateService } from './services/billpay-manage-payee-state.service';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
export const defaultRoute = {
    path: '',
    component: ManagePayeesJourneyComponent,
    children: [
        { path: 'enrolment', component: BillPayEnrolmentViewComponent },
        {
            path: '',
            canActivate: [BillpayEnrolmentGuard],
            children: [
                { path: '', component: PayeeListViewComponent, pathMatch: 'full' },
                {
                    path: 'payees',
                    component: ManagePayeeViewComponent,
                    children: [
                        {
                            path: '',
                            component: BillpaySearchPayeeViewComponent,
                            pathMatch: 'full',
                        },
                        {
                            path: 'manual',
                            component: BillpayCreateManualPayeeViewComponent,
                            canDeactivate: [BillpayManagePayeeDeactivateGuardService],
                        },
                        {
                            path: 'manual/review',
                            component: CreateManualPayeeReviewViewComponent,
                            canDeactivate: [BillpayManagePayeeDeactivateGuardService],
                        },
                        {
                            path: 'manual/edit/:id',
                            component: BillpayEditManualPayeeViewComponent,
                            canDeactivate: [BillpayManagePayeeDeactivateGuardService],
                        },
                        {
                            path: 'manual/edit/:id/review',
                            component: EditManualPayeeReviewViewComponent,
                            canDeactivate: [BillpayManagePayeeDeactivateGuardService],
                        },
                        {
                            path: 'electronic/:id',
                            component: BillpayCreateElectronicPayeeViewComponent,
                            canDeactivate: [BillpayManagePayeeDeactivateGuardService],
                        },
                        {
                            path: 'electronic/:id/review',
                            component: CreateElectronicPayeeReviewViewComponent,
                            canDeactivate: [BillpayManagePayeeDeactivateGuardService],
                        },
                        {
                            path: 'electronic/edit/:id',
                            component: BillpayEditElectronicPayeeViewComponent,
                            canDeactivate: [BillpayManagePayeeDeactivateGuardService],
                        },
                        {
                            path: 'electronic/edit/:id/review',
                            component: EditElectronicPayeeReviewViewComponent,
                            canDeactivate: [BillpayManagePayeeDeactivateGuardService],
                        },
                    ],
                },
                { path: 'payee-summary', component: PayeeSummaryViewComponent },
                { path: 'ebills', component: EbillsEnrolmentViewComponent },
            ],
        },
    ],
};
export class ManagePayeesJourneyModule {
    static forRoot(data = { route: defaultRoute }) {
        return {
            ngModule: ManagePayeesJourneyModule,
            providers: [provideRoutes([data.route])],
        };
    }
}
ManagePayeesJourneyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePayeesJourneyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ManagePayeesJourneyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePayeesJourneyModule, declarations: [ManagePayeesJourneyComponent,
        PayeeListViewComponent,
        PayeeSummaryViewComponent,
        EbillsEnrolmentViewComponent,
        ManagePayeeViewComponent,
        BillPayEnrolmentViewComponent,
        CreateManualPayeeReviewViewComponent,
        EditManualPayeeReviewViewComponent,
        CreateElectronicPayeeReviewViewComponent,
        EditElectronicPayeeReviewViewComponent,
        BillpaySearchPayeeViewComponent,
        BillpayCreateManualPayeeViewComponent,
        BillpayEditManualPayeeViewComponent,
        BillpayCreateElectronicPayeeViewComponent,
        BillpayEditElectronicPayeeViewComponent], imports: [CommonModule,
        RouterModule,
        BillpayJourneysCommonModule,
        ManagePayeesJourneyUiModule, i1.BackbaseCoreModule] });
ManagePayeesJourneyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePayeesJourneyModule, providers: [
        ManagePayeesJourneyConfigService,
        BillpayPayeeDataService,
        BillpayPayverisEbillsEnrolmentWidgetService,
        BillpayPayeeSummaryWidgetService,
        BillpayManagePayeeDeactivateGuardService,
        BillpayManagePayeeDataService,
        BillpayManagePayeeStateService,
    ], imports: [[
            CommonModule,
            RouterModule,
            BillpayJourneysCommonModule,
            ManagePayeesJourneyUiModule,
            BackbaseCoreModule.withConfig({
                classMap: { ManagePayeesJourneyComponent },
            }),
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePayeesJourneyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        ManagePayeesJourneyComponent,
                        PayeeListViewComponent,
                        PayeeSummaryViewComponent,
                        EbillsEnrolmentViewComponent,
                        ManagePayeeViewComponent,
                        BillPayEnrolmentViewComponent,
                        CreateManualPayeeReviewViewComponent,
                        EditManualPayeeReviewViewComponent,
                        CreateElectronicPayeeReviewViewComponent,
                        EditElectronicPayeeReviewViewComponent,
                        BillpaySearchPayeeViewComponent,
                        BillpayCreateManualPayeeViewComponent,
                        BillpayEditManualPayeeViewComponent,
                        BillpayCreateElectronicPayeeViewComponent,
                        BillpayEditElectronicPayeeViewComponent,
                    ],
                    imports: [
                        CommonModule,
                        RouterModule,
                        BillpayJourneysCommonModule,
                        ManagePayeesJourneyUiModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { ManagePayeesJourneyComponent },
                        }),
                    ],
                    providers: [
                        ManagePayeesJourneyConfigService,
                        BillpayPayeeDataService,
                        BillpayPayverisEbillsEnrolmentWidgetService,
                        BillpayPayeeSummaryWidgetService,
                        BillpayManagePayeeDeactivateGuardService,
                        BillpayManagePayeeDataService,
                        BillpayManagePayeeStateService,
                    ],
                }]
        }] });
