// @backbase/manage-payees-journey-ang/esm2015/src/manage-payees-journey.component.js
import { Component } from '@angular/core';
import { ManagePayeesNavigationService } from './services/manage-payees-navigation.service';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
export class ManagePayeesJourneyComponent {
}
ManagePayeesJourneyComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePayeesJourneyComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ManagePayeesJourneyComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: ManagePayeesJourneyComponent, selector: "bb-manage-payees-journey", providers: [ManagePayeesNavigationService], ngImport: i0, template: `<router-outlet></router-outlet>`, isInline: true, directives: [{ type: i1.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePayeesJourneyComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-manage-payees-journey',
                    template: `<router-outlet></router-outlet>`,
                    providers: [ManagePayeesNavigationService],
                }]
        }] });
