// @backbase/cards-common-ang/esm2015/src/constants.js
import { combineLatest } from 'rxjs';
import { map } from 'rxjs/operators';
import { PaymentCardDeliveryStepStatus } from './model/payment-card-delivery-status.model';
import { PaymentCardReplacementReason, } from './model/payment-card-replacement-reason.model';
import { PaymentCardLockStatus, PaymentCardReplacementStatus, PaymentCardStatus } from './model/payment-card.model';
/**
 * Timeout for the notification
 */
export const notificationTtl = 5000;
/**
 * The default mapping of backend values to internal model representation for card status.  This
 * mapping can be overridden per app.
 */
// prettier-ignore
export const defaultStatusMapping = {
    'Active': PaymentCardStatus.Active,
    'Inactive': PaymentCardStatus.Inactive,
    'Cancelled': PaymentCardStatus.Canceled,
    'Expired': PaymentCardStatus.Expired,
};
/**
 * The default mapping of backend values to internal model representation for card replacement
 * status.  This mapping can be overridden per app.
 */
// prettier-ignore
export const defaultReplacementStatusMapping = {
    'NotUnderReplacement': PaymentCardReplacementStatus.NotUnderReplacement,
    'ReplacementRequested': PaymentCardReplacementStatus.UnderReplacement,
    'ReplacementInTransit': PaymentCardReplacementStatus.UnderReplacement,
    'ReplacementUnderCancellation': PaymentCardReplacementStatus.UnderReplacement,
    'Replaced': PaymentCardReplacementStatus.Replaced,
};
/**
 * The mapping of backend values to internal model representation for card lock status.  This
 * mapping cannot be overridden (it is hardwired in the spec).
 */
// prettier-ignore
export const defaultLockStatusMapping = {
    'LOCKED': PaymentCardLockStatus.Locked,
    'UNLOCKED': PaymentCardLockStatus.Unlocked,
};
/**
 * The mapping of backend values to internal model representation for card delivery step status.
 * This mapping cannot be overridden (it is hardwired in the spec).
 */
// prettier-ignore
export const defaultDeliveryStepStatusMapping = {
    'SUCCESS': PaymentCardDeliveryStepStatus.Success,
    'PENDING': PaymentCardDeliveryStepStatus.Pending,
    'FAILED': PaymentCardDeliveryStepStatus.Failed,
};
/**
 * The default mapping of internal model representation to backend values for card replacement
 * reason.  This mapping can be overridden per app.
 */
export const defaultReplacementReasonMapping = {
    [PaymentCardReplacementReason.Lost]: 'Lost',
    [PaymentCardReplacementReason.Stolen]: 'Stolen',
    [PaymentCardReplacementReason.Damaged]: 'Damaged',
};
// prettier-ignore
export const defaultLimitChannelMapping = {
    'ATM': { name: 'withdrawals', description: 'The amount you can withdraw from cash machines' },
    'ONLINE': { name: 'online purchases', description: 'The limit for when using your card for payments online' },
    'POS': {
        name: 'terminal payments',
        description: 'The limit for physical payments with your card (e.g. supermarkets, restaurants, etc.)',
    },
};
/**
 * Enumeration of possible widget preferences
 */
export const WidgetProperties = {
    NotificationTtl: 'notificationTtl',
    updateAllLimits: 'bb.card.updateAllLimits',
};
export const DEFAULT_PAYMENT_CARD_TYPE_GROUP = ['Credit', 'Debit'];
export const filterInputParameters = (input, prop) => combineLatest([input, prop]).pipe(map(([inputVal, propVal]) => (inputVal !== undefined ? inputVal : propVal)));
