// @backbase/ui-ang/esm2015/_example/control-error-handler/showcase/control-error-handler-showcase.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import { BackbaseUiModule } from '../../../ui-ang.module';
import { ControlErrorHandlerShowcaseComponent } from './control-error-handler-showcase.component';
import { ControlErrorHandlerModule } from '@backbase/ui-ang/control-error-handler';
import { InputWithCustomErrorComponent } from './custom-error/input-with-custom-errors-showcase.component';
import { CustomErrorShowcaseComponent } from './custom-error/custom-error-showcase.component';
import * as i0 from "@angular/core";
const components = [ControlErrorHandlerShowcaseComponent, InputWithCustomErrorComponent, CustomErrorShowcaseComponent];
export class ControlErrorHandlerShowcaseModule {
}
ControlErrorHandlerShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ControlErrorHandlerShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ControlErrorHandlerShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ControlErrorHandlerShowcaseModule, declarations: [ControlErrorHandlerShowcaseComponent, InputWithCustomErrorComponent, CustomErrorShowcaseComponent], imports: [CommonModule, ReactiveFormsModule, ControlErrorHandlerModule, BackbaseUiModule, InputTextModule], exports: [ControlErrorHandlerShowcaseComponent, InputWithCustomErrorComponent, CustomErrorShowcaseComponent] });
ControlErrorHandlerShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ControlErrorHandlerShowcaseModule, imports: [[CommonModule, ReactiveFormsModule, ControlErrorHandlerModule, BackbaseUiModule, InputTextModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ControlErrorHandlerShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ReactiveFormsModule, ControlErrorHandlerModule, BackbaseUiModule, InputTextModule],
                    declarations: [...components],
                    exports: [...components],
                    entryComponents: [ControlErrorHandlerShowcaseComponent],
                }]
        }] });
