// @backbase/manage-payees-journey-ang/esm2015/src/components/manage-payees-journey-ui.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule, FormsModule } from '@angular/forms';
import { BillpayJourneysCommonModule } from '@backbase/billpay-journeys-common';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { BillpayCancelNewPayeeModalComponent } from './add-payee/billpay-cancel-new-payee-modal/billpay-cancel-new-payee-modal.component';
import { BillpayElectronicPayeeReviewComponent } from './add-payee/billpay-electronic-payee-review/billpay-electronic-payee-review.component';
import { BillpayManualPayeeReviewComponent } from './add-payee/billpay-manual-payee-review/billpay-manual-payee-review.component';
import { BillpayPayverisEbillsEnrolmentCancelModalComponent } from './payveris-ebills-enrollment/billpay-payveris-ebills-enrolment-cancel-modal/billpay-payveris-ebills-enrolment-cancel-modal.component';
import { BillpayPayeeSummaryListComponent } from './payee-list/billpay-payee-summary-list/billpay-payee-summary-list.component';
import { BillpayPayeeSummaryItemComponent } from './payee-list/billpay-payee-summary-item/billpay-payee-summary-item.component';
import { BillpayPayeeStepperComponent } from './payee-summary/billpay-payee-stepper/billpay-payee-stepper.component';
import { BillpayPayeeInformationComponent } from './payee-summary/billpay-payee-information/billpay-payee-information.component';
import { BillpayPayverisEbillsEnrolmentSelectBillerComponent } from './payveris-ebills-enrollment/billpay-payveris-ebills-enrolment-select-biller/billpay-payveris-ebills-enrolment-select-biller.component';
import { BillpayPayverisEbillsEnrolmentMfaEntryComponent } from './payveris-ebills-enrollment/billpay-payveris-ebills-enrolment-mfa-entry/billpay-payveris-ebills-enrolment-mfa-entry.component';
import { BillpayPayverisEbillsEnrolmentLoginComponent } from './payveris-ebills-enrollment/billpay-payveris-ebills-enrolment-login/billpay-payveris-ebills-enrolment-login.component';
import { BillpayPayverisEbillsEnrolmentOtpSelectComponent } from './payveris-ebills-enrollment/billpay-payveris-ebills-enrolment-otp-select/billpay-payveris-ebills-enrolment-otp-select.component';
import { BillpayPayverisEbillsEnrolmentOtpEntryComponent } from './payveris-ebills-enrollment/billpay-payveris-ebills-enrolment-otp-entry/billpay-payveris-ebills-enrolment-otp-entry.component';
import { BillpayPayverisEbillsEnrolmentCaptchaComponent } from './payveris-ebills-enrollment/billpay-payveris-ebills-enrolment-captcha/billpay-payveris-ebills-enrolment-captcha.component';
import { BillpayPayverisEbillsEnrolmentSelectAccountComponent } from './payveris-ebills-enrollment/billpay-payveris-ebills-enrolment-select-account/billpay-payveris-ebills-enrolment-select-account.component';
import { BillpayPayverisEbillsEnrolmentErrorComponent } from './payveris-ebills-enrollment/billpay-payveris-ebills-enrolment-error/billpay-payveris-ebills-enrolment-error.component';
import { BillpayPayverisEbillsEnrolmentNotificationSuccessComponent } from './payveris-ebills-enrollment/billpay-payveris-ebills-enrolment-notification-success/billpay-payveris-ebills-enrolment-notification-success.component';
import { AlertModule } from '@backbase/ui-ang/alert';
import { BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import { CollapsibleModule } from '@backbase/ui-ang/collapsible';
import { DropdownMenuModule } from '@backbase/ui-ang/dropdown-menu';
import { IconModule } from '@backbase/ui-ang/icon';
import { InputPasswordModule } from '@backbase/ui-ang/input-password';
import { InputRadioGroupModule } from '@backbase/ui-ang/input-radio-group';
import { PhoneNumberModule } from '@backbase/ui-ang/phone-number-format-pipe';
import { BillpayCancelEditPayeeModalComponent } from './add-payee/billpay-cancel-edit-payee-modal/billpay-cancel-edit-payee-modal.component';
import { BillpayElectronicPayeeFormComponent } from './add-payee/billpay-electronic-payee-form/billpay-electronic-payee-form.component';
import { BillpayElectronicPayeeSearchComponent } from './add-payee/billpay-electronic-payee-search/billpay-electronic-payee-search.component';
import { BillpayManualPayeeFormComponent } from './add-payee/billpay-manual-payee-form/billpay-manual-payee-form.component';
import { FieldsetModule } from '@backbase/ui-ang/fieldset';
import { SearchBoxModule } from '@backbase/ui-ang/search-box';
import * as i0 from "@angular/core";
const components = [
    BillpayCancelNewPayeeModalComponent,
    BillpayElectronicPayeeReviewComponent,
    BillpayManualPayeeReviewComponent,
    BillpayPayverisEbillsEnrolmentSelectBillerComponent,
    BillpayPayverisEbillsEnrolmentCancelModalComponent,
    BillpayPayverisEbillsEnrolmentMfaEntryComponent,
    BillpayPayverisEbillsEnrolmentLoginComponent,
    BillpayPayverisEbillsEnrolmentOtpSelectComponent,
    BillpayPayverisEbillsEnrolmentOtpEntryComponent,
    BillpayPayverisEbillsEnrolmentCaptchaComponent,
    BillpayPayverisEbillsEnrolmentSelectAccountComponent,
    BillpayPayverisEbillsEnrolmentErrorComponent,
    BillpayPayeeSummaryListComponent,
    BillpayPayeeSummaryItemComponent,
    BillpayPayeeStepperComponent,
    BillpayPayeeInformationComponent,
    BillpayPayverisEbillsEnrolmentNotificationSuccessComponent,
    BillpayCancelEditPayeeModalComponent,
    BillpayElectronicPayeeFormComponent,
    BillpayElectronicPayeeSearchComponent,
    BillpayManualPayeeFormComponent,
];
const uiAngModules = [
    IconModule,
    CollapsibleModule,
    PhoneNumberModule,
    AlertModule,
    InputPasswordModule,
    InputRadioGroupModule,
    BbDatePipeModule,
    DropdownMenuModule,
    FieldsetModule,
    SearchBoxModule,
];
export class ManagePayeesJourneyUiModule {
}
ManagePayeesJourneyUiModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePayeesJourneyUiModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ManagePayeesJourneyUiModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePayeesJourneyUiModule, declarations: [BillpayCancelNewPayeeModalComponent,
        BillpayElectronicPayeeReviewComponent,
        BillpayManualPayeeReviewComponent,
        BillpayPayverisEbillsEnrolmentSelectBillerComponent,
        BillpayPayverisEbillsEnrolmentCancelModalComponent,
        BillpayPayverisEbillsEnrolmentMfaEntryComponent,
        BillpayPayverisEbillsEnrolmentLoginComponent,
        BillpayPayverisEbillsEnrolmentOtpSelectComponent,
        BillpayPayverisEbillsEnrolmentOtpEntryComponent,
        BillpayPayverisEbillsEnrolmentCaptchaComponent,
        BillpayPayverisEbillsEnrolmentSelectAccountComponent,
        BillpayPayverisEbillsEnrolmentErrorComponent,
        BillpayPayeeSummaryListComponent,
        BillpayPayeeSummaryItemComponent,
        BillpayPayeeStepperComponent,
        BillpayPayeeInformationComponent,
        BillpayPayverisEbillsEnrolmentNotificationSuccessComponent,
        BillpayCancelEditPayeeModalComponent,
        BillpayElectronicPayeeFormComponent,
        BillpayElectronicPayeeSearchComponent,
        BillpayManualPayeeFormComponent], imports: [CommonModule, ReactiveFormsModule, FormsModule, NgbModule, BillpayJourneysCommonModule, IconModule,
        CollapsibleModule,
        PhoneNumberModule,
        AlertModule,
        InputPasswordModule,
        InputRadioGroupModule,
        BbDatePipeModule,
        DropdownMenuModule,
        FieldsetModule,
        SearchBoxModule], exports: [BillpayCancelNewPayeeModalComponent,
        BillpayElectronicPayeeReviewComponent,
        BillpayManualPayeeReviewComponent,
        BillpayPayverisEbillsEnrolmentSelectBillerComponent,
        BillpayPayverisEbillsEnrolmentCancelModalComponent,
        BillpayPayverisEbillsEnrolmentMfaEntryComponent,
        BillpayPayverisEbillsEnrolmentLoginComponent,
        BillpayPayverisEbillsEnrolmentOtpSelectComponent,
        BillpayPayverisEbillsEnrolmentOtpEntryComponent,
        BillpayPayverisEbillsEnrolmentCaptchaComponent,
        BillpayPayverisEbillsEnrolmentSelectAccountComponent,
        BillpayPayverisEbillsEnrolmentErrorComponent,
        BillpayPayeeSummaryListComponent,
        BillpayPayeeSummaryItemComponent,
        BillpayPayeeStepperComponent,
        BillpayPayeeInformationComponent,
        BillpayPayverisEbillsEnrolmentNotificationSuccessComponent,
        BillpayCancelEditPayeeModalComponent,
        BillpayElectronicPayeeFormComponent,
        BillpayElectronicPayeeSearchComponent,
        BillpayManualPayeeFormComponent, IconModule,
        CollapsibleModule,
        PhoneNumberModule,
        AlertModule,
        InputPasswordModule,
        InputRadioGroupModule,
        BbDatePipeModule,
        DropdownMenuModule,
        FieldsetModule,
        SearchBoxModule] });
ManagePayeesJourneyUiModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePayeesJourneyUiModule, imports: [[CommonModule, ReactiveFormsModule, FormsModule, NgbModule, BillpayJourneysCommonModule, ...uiAngModules], IconModule,
        CollapsibleModule,
        PhoneNumberModule,
        AlertModule,
        InputPasswordModule,
        InputRadioGroupModule,
        BbDatePipeModule,
        DropdownMenuModule,
        FieldsetModule,
        SearchBoxModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePayeesJourneyUiModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components],
                    imports: [CommonModule, ReactiveFormsModule, FormsModule, NgbModule, BillpayJourneysCommonModule, ...uiAngModules],
                    exports: [...components, ...uiAngModules],
                }]
        }] });
