// @backbase/accessgroup-ui/esm2015/lib/business-function-name/business-function-name.component.js
import { Component, Input } from '@angular/core';
import { TranslationsService } from '@backbase/accessgroup-common-ang';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/accessgroup-common-ang";
export class BusinessFunctionNameComponent {
    constructor(translationsService) {
        this.translationsService = translationsService;
    }
    get name() {
        if (!this.businessFunction)
            return '';
        return this.translationsService.getBusinessFunctionName(this.businessFunction.functionCode) || this.businessFunction.name;
    }
}
/** @nocollapse */ BusinessFunctionNameComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BusinessFunctionNameComponent, deps: [{ token: i1.TranslationsService }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ BusinessFunctionNameComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: BusinessFunctionNameComponent, selector: "bb-business-function-name", inputs: { businessFunction: "businessFunction" }, ngImport: i0, template: "{{ name }}" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BusinessFunctionNameComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-business-function-name',
                    templateUrl: './business-function-name.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.TranslationsService }]; }, propDecorators: { businessFunction: [{
                type: Input
            }] } });
