// @backbase/notifications-ang/esm2015/lib/notifications-popups-ang/src/notifications-popups.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NotificationsCommonModule } from '@backbase/notifications-common-ang';
import { NotificationsPopupsWidgetModule } from '@backbase/notifications-popups-widget-ang';
import { NotificationsPopupsComponent } from './components/notifications-popups.component';
import * as i0 from "@angular/core";
const modules = [NotificationsCommonModule, NotificationsPopupsWidgetModule];
export class NotificationsPopupsModule {
}
/** @nocollapse */ NotificationsPopupsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsPopupsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ NotificationsPopupsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsPopupsModule, declarations: [NotificationsPopupsComponent], imports: [CommonModule, NotificationsCommonModule, NotificationsPopupsWidgetModule], exports: [NotificationsPopupsComponent] });
/** @nocollapse */ NotificationsPopupsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsPopupsModule, imports: [[CommonModule, ...modules]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsPopupsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...modules],
                    declarations: [NotificationsPopupsComponent],
                    exports: [NotificationsPopupsComponent],
                }]
        }] });
