// @backbase/ui-ang/esm2015/_example/product-item-investment-account/showcase/product-item-investment-account-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ProductItemInvestmentAccountShowcaseComponent } from './product-item-investment-account-showcase.component';
import { ProductItemInvestmentAccountModule } from '@backbase/ui-ang/product-item-investment-account';
import * as i0 from "@angular/core";
export class ProductItemInvestmentAccountShowcaseModule {
}
ProductItemInvestmentAccountShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemInvestmentAccountShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductItemInvestmentAccountShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemInvestmentAccountShowcaseModule, declarations: [ProductItemInvestmentAccountShowcaseComponent], imports: [CommonModule, ProductItemInvestmentAccountModule] });
ProductItemInvestmentAccountShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemInvestmentAccountShowcaseModule, imports: [[CommonModule, ProductItemInvestmentAccountModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemInvestmentAccountShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ProductItemInvestmentAccountModule],
                    declarations: [ProductItemInvestmentAccountShowcaseComponent],
                    entryComponents: [ProductItemInvestmentAccountShowcaseComponent],
                }]
        }] });
