// @backbase/budget-widget-ang/esm2015/src/services/budget.service.js
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/budgeting";
export class BudgetService {
    constructor(budgetingDataHttpService) {
        this.budgetingDataHttpService = budgetingDataHttpService;
    }
    getBudgets() {
        return this.budgetingDataHttpService.getBudgets();
    }
    createBudget(budget) {
        return this.budgetingDataHttpService.postBudgets({ budgetsPostRequestBody: budget }, 'response');
    }
    updateBudget(budgetId, budget) {
        return this.budgetingDataHttpService.putId({ id: budgetId, idPutRequestBody: budget }, 'response');
    }
    deleteBudget(budgetId) {
        return this.budgetingDataHttpService.deleteId({ id: budgetId }, 'response');
    }
}
BudgetService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetService, deps: [{ token: i1.BudgetsHttpService }], target: i0.ɵɵFactoryTarget.Injectable });
BudgetService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.BudgetsHttpService }]; } });
