// @backbase/ui-ang/esm2015/dropdown-menu/dropdown-menu.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NgbDropdownModule } from '@ng-bootstrap/ng-bootstrap';
import { IconModule } from '@backbase/ui-ang/icon';
import { ButtonModule } from '@backbase/ui-ang/button';
import { DropdownMenuOptionDirective } from './directives/dropdown-menu-option.directive';
import { DropdownLabelDirective, DropdownMenuComponent, DropdownMenuItemDirective } from './dropdown-menu.component';
import { DropdownMenuFullWidthDirective } from './directives/dropdown-menu-full-width.directive';
import * as i0 from "@angular/core";
export class DropdownMenuModule {
}
DropdownMenuModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownMenuModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
DropdownMenuModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownMenuModule, declarations: [DropdownMenuComponent,
        DropdownMenuItemDirective,
        DropdownLabelDirective,
        DropdownMenuOptionDirective,
        DropdownMenuFullWidthDirective], imports: [CommonModule, NgbDropdownModule, IconModule, ButtonModule], exports: [DropdownMenuComponent,
        DropdownMenuItemDirective,
        DropdownLabelDirective,
        DropdownMenuOptionDirective,
        DropdownMenuFullWidthDirective] });
DropdownMenuModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownMenuModule, imports: [[CommonModule, NgbDropdownModule, IconModule, ButtonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownMenuModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, NgbDropdownModule, IconModule, ButtonModule],
                    declarations: [
                        DropdownMenuComponent,
                        DropdownMenuItemDirective,
                        DropdownLabelDirective,
                        DropdownMenuOptionDirective,
                        DropdownMenuFullWidthDirective,
                    ],
                    exports: [
                        DropdownMenuComponent,
                        DropdownMenuItemDirective,
                        DropdownLabelDirective,
                        DropdownMenuOptionDirective,
                        DropdownMenuFullWidthDirective,
                    ],
                }]
        }] });
