// @backbase/data-ang/bundles/backbase-data-ang-comments.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@backbase/comments-http-ang')) :
	typeof define === 'function' && define.amd ? define('@backbase/data-ang/comments', ['exports', '@backbase/comments-http-ang'], factory) :
	(global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["data-ang"] = global.backbase["data-ang"] || {}, global.backbase["data-ang"].comments = {}), global.commentsHttpAng));
})(this, (function (exports, commentsHttpAng) { 'use strict';

	/**
	 * @deprecated The entry point `@backbase/data-ang/comments` will be removed in the future.
	 *             Use `@backbase/data-ang/comments-v1` or `@backbase/data-ang/comments-v2` instead.
	 */
	var DEPRECATED_BASE_PATH = commentsHttpAng.BASE_PATH;
	/**
	 * @deprecated The entry point `@backbase/data-ang/comments` will be removed in the future.
	 *             Use `@backbase/data-ang/comments-v1` or `@backbase/data-ang/comments-v2` instead.
	 */
	var DeprecatedThreadDto = commentsHttpAng.ThreadDto;
	/**
	 * @deprecated The entry point `@backbase/data-ang/comments` will be removed in the future.
	 *             Use `@backbase/data-ang/comments-v1` or `@backbase/data-ang/comments-v2` instead.
	 */
	var DeprecatedCommentsService = commentsHttpAng.CommentsService;
	/**
	 * @deprecated The entry point `@backbase/data-ang/comments` will be removed in the future.
	 *             Use `@backbase/data-ang/comments-v1` or `@backbase/data-ang/comments-v2` instead.
	 */
	var DEPRECATED_APIS = commentsHttpAng.APIS;
	/**
	 * @deprecated The entry point `@backbase/data-ang/comments` will be removed in the future.
	 *             Use `@backbase/data-ang/comments-v1` or `@backbase/data-ang/comments-v2` instead.
	 */
	var DeprecatedApiModule = commentsHttpAng.ApiModule;
	/**
	 * @deprecated The entry point `@backbase/data-ang/comments` will be removed in the future.
	 *             Use `@backbase/data-ang/comments-v1` or `@backbase/data-ang/comments-v2` instead.
	 */
	var DEPRECATED_COLLECTION_FORMATS = commentsHttpAng.COLLECTION_FORMATS;
	/**
	 * @deprecated The entry point `@backbase/data-ang/comments` will be removed in the future.
	 *             Use `@backbase/data-ang/comments-v1` or `@backbase/data-ang/comments-v2` instead.
	 */
	var DeprecatedConfiguration = commentsHttpAng.Configuration;

	/**
	 * Generated bundle index. Do not edit.
	 */

	exports.APIS = DEPRECATED_APIS;
	exports.ApiModule = DeprecatedApiModule;
	exports.BASE_PATH = DEPRECATED_BASE_PATH;
	exports.COLLECTION_FORMATS = DEPRECATED_COLLECTION_FORMATS;
	exports.CommentsService = DeprecatedCommentsService;
	exports.Configuration = DeprecatedConfiguration;
	exports.ThreadDto = DeprecatedThreadDto;

	Object.defineProperty(exports, '__esModule', { value: true });

}));
