// @backbase/accessgroup-ui/esm2015/lib/function-group-time-bounds/function-group-time-bounds.component.js
import { Component, Input } from '@angular/core';
import { formatDate, formatTime } from '../model/helpers';
import * as i0 from "@angular/core";
import * as i1 from "../time-bounds/time-bounds.component";
export class FunctionGroupTimeBoundsComponent {
    constructor() {
        this.validityFormat = '';
    }
    /**
     * Function group
     */
    set functionGroup(value) {
        if (value) {
            this.functionGroupValue = value;
            this.fromDate = formatDate(value.validFromDate);
            this.fromTime = formatTime(this.fromDate, value.validFromTime);
            this.untilDate = formatDate(value.validUntilDate);
            this.untilTime = formatTime(this.untilDate, value.validUntilTime);
            this.setValidityPeriod();
        }
    }
    get functionGroup() {
        return this.functionGroupValue;
    }
    setValidityPeriod() {
        this.validityFormat = '';
        if (this.fromDate && !this.untilDate) {
            this.validityFormat = 'validFromOnly';
        }
        if (!this.fromDate && this.untilDate) {
            this.validityFormat = 'validUntilOnly';
        }
        if (this.fromDate && this.untilDate) {
            this.validityFormat = 'validFromUntil';
        }
    }
}
/** @nocollapse */ FunctionGroupTimeBoundsComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FunctionGroupTimeBoundsComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ FunctionGroupTimeBoundsComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: FunctionGroupTimeBoundsComponent, selector: "bb-function-group-time-bounds", inputs: { functionGroup: "functionGroup" }, ngImport: i0, template: "<bb-time-bounds [timeBounds]=\"functionGroup\"></bb-time-bounds>\n", components: [{ type: i1.TimeBoundsComponent, selector: "bb-time-bounds", inputs: ["timeBounds"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FunctionGroupTimeBoundsComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-function-group-time-bounds',
                    templateUrl: './function-group-time-bounds.component.html',
                }]
        }], propDecorators: { functionGroup: [{
                type: Input
            }] } });
