// @backbase/ui-ang/esm2015/_example/common-error-state/showcase/common-error-state-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ErrorCommonStateModule } from '@backbase/ui-ang/common-error-state';
import { CommonErrorStateShowcaseComponent } from './common-error-state-showcase.component';
import * as i0 from "@angular/core";
export class CommonErrorStateShowcaseModule {
}
CommonErrorStateShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CommonErrorStateShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CommonErrorStateShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CommonErrorStateShowcaseModule, declarations: [CommonErrorStateShowcaseComponent], imports: [CommonModule, ErrorCommonStateModule] });
CommonErrorStateShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CommonErrorStateShowcaseModule, imports: [[CommonModule, ErrorCommonStateModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CommonErrorStateShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ErrorCommonStateModule],
                    declarations: [CommonErrorStateShowcaseComponent],
                    entryComponents: [CommonErrorStateShowcaseComponent],
                }]
        }] });
