// @backbase/manage-payees-journey-ang/esm2015/src/model/base-electronic-payee-page.model.js
import { HostListener, Directive } from '@angular/core';
import { of, Subject } from 'rxjs';
import { map, shareReplay, switchMap, take, takeUntil, tap } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "../services/billpay-manage-payee-state.service";
import * as i3 from "@backbase/billpay-journeys-common";
export class BaseElectronicPayeePageDirective {
    /** @internal */
    constructor(route, stateService, discardModalService) {
        this.route = route;
        this.stateService = stateService;
        this.discardModalService = discardModalService;
        this.isFormTouched = false;
        this.destroy$ = new Subject();
        this.isFindPayeeModal = false;
        this.id$ = this.route.paramMap.pipe(take(1), map(params => params.get('id') || ''));
        this.payeeState$ = this.stateService.electronicPayeeFormState.pipe(take(1), shareReplay(1));
    }
    onWindowUnload($event) {
        this.payeeState$.subscribe(state => {
            if ((state || this.isFormTouched) && !this.isDiscardModalOpen) {
                $event.returnValue = true;
            }
        });
    }
    get isDiscardModalOpen() {
        return this.discardModalService.isDiscardModalOpen;
    }
    onFormValueChanged() {
        this.isFormTouched = true;
    }
    openModal() {
        const openModal = () => this.discardModalService
            .openModal()
            .pipe(tap(isResolved => isResolved && this.stateService.resetElectronicPayee()));
        return this.payeeState$.pipe(switchMap(payeeState => (payeeState || this.isFormTouched ? openModal() : of(true))), takeUntil(this.destroy$));
    }
    onModalResolve(resolvedWith) {
        this.isFindPayeeModal = false;
        this.discardModalService.onModalResolve(resolvedWith);
    }
    ngOnDestroy() {
        this.destroy$.next();
        this.destroy$.complete();
    }
}
BaseElectronicPayeePageDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BaseElectronicPayeePageDirective, deps: [{ token: i1.ActivatedRoute }, { token: i2.BillpayManagePayeeStateService }, { token: i3.BillpayDiscardChangesModalService }], target: i0.ɵɵFactoryTarget.Directive });
BaseElectronicPayeePageDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.15", type: BaseElectronicPayeePageDirective, host: { listeners: { "window:beforeunload": "onWindowUnload($event)" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BaseElectronicPayeePageDirective, decorators: [{
            type: Directive
        }], ctorParameters: function () { return [{ type: i1.ActivatedRoute }, { type: i2.BillpayManagePayeeStateService }, { type: i3.BillpayDiscardChangesModalService }]; }, propDecorators: { onWindowUnload: [{
                type: HostListener,
                args: ['window:beforeunload', ['$event']]
            }] } });
