// @backbase/manage-payees-journey-ang/esm2015/src/model/payveris-enrolment-state.enum.js
export var PayverisEnrolmentState;
(function (PayverisEnrolmentState) {
    PayverisEnrolmentState[PayverisEnrolmentState["Initialize"] = 0] = "Initialize";
    PayverisEnrolmentState[PayverisEnrolmentState["Login"] = 1] = "Login";
    PayverisEnrolmentState[PayverisEnrolmentState["SelectBiller"] = 2] = "SelectBiller";
    PayverisEnrolmentState[PayverisEnrolmentState["OtpSelect"] = 3] = "OtpSelect";
    PayverisEnrolmentState[PayverisEnrolmentState["OtpEntry"] = 4] = "OtpEntry";
    PayverisEnrolmentState[PayverisEnrolmentState["Callback"] = 5] = "Callback";
    PayverisEnrolmentState[PayverisEnrolmentState["Captcha"] = 6] = "Captcha";
    PayverisEnrolmentState[PayverisEnrolmentState["Mfa"] = 7] = "Mfa";
    PayverisEnrolmentState[PayverisEnrolmentState["SelectAccounts"] = 8] = "SelectAccounts";
    PayverisEnrolmentState[PayverisEnrolmentState["Success"] = 9] = "Success";
    PayverisEnrolmentState[PayverisEnrolmentState["ErrorState"] = 10] = "ErrorState";
    PayverisEnrolmentState[PayverisEnrolmentState["Undefined"] = 11] = "Undefined";
})(PayverisEnrolmentState || (PayverisEnrolmentState = {}));
export var PayverisEnrolmentComponentState;
(function (PayverisEnrolmentComponentState) {
    PayverisEnrolmentComponentState[PayverisEnrolmentComponentState["Undefined"] = 0] = "Undefined";
    PayverisEnrolmentComponentState[PayverisEnrolmentComponentState["Ready"] = 1] = "Ready";
    PayverisEnrolmentComponentState[PayverisEnrolmentComponentState["Checking"] = 2] = "Checking";
    PayverisEnrolmentComponentState[PayverisEnrolmentComponentState["Callback"] = 3] = "Callback";
})(PayverisEnrolmentComponentState || (PayverisEnrolmentComponentState = {}));
