// @backbase/loans-collection-ui-ang/esm2015/lib/loan-summary/retail-loan-summary-overdue-banner/retail-loan-summary-overdue-banner.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AmountModule } from '@backbase/ui-ang/amount';
import { BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import { ButtonModule } from '@backbase/ui-ang/button';
import { IconModule } from '@backbase/ui-ang/icon';
import { AlertModule } from '@backbase/ui-ang/alert';
import { RetailLoanSummaryOverdueBannerComponent } from './retail-loan-summary-overdue-banner.component';
import * as i0 from "@angular/core";
export class RetailLoanSummaryOverdueBannerModule {
}
/** @nocollapse */ RetailLoanSummaryOverdueBannerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RetailLoanSummaryOverdueBannerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ RetailLoanSummaryOverdueBannerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RetailLoanSummaryOverdueBannerModule, declarations: [RetailLoanSummaryOverdueBannerComponent], imports: [CommonModule, ButtonModule, IconModule, AmountModule, BbDatePipeModule, AlertModule], exports: [RetailLoanSummaryOverdueBannerComponent] });
/** @nocollapse */ RetailLoanSummaryOverdueBannerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RetailLoanSummaryOverdueBannerModule, imports: [[CommonModule, ButtonModule, IconModule, AmountModule, BbDatePipeModule, AlertModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RetailLoanSummaryOverdueBannerModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [RetailLoanSummaryOverdueBannerComponent],
                    imports: [CommonModule, ButtonModule, IconModule, AmountModule, BbDatePipeModule, AlertModule],
                    exports: [RetailLoanSummaryOverdueBannerComponent],
                }]
        }] });
