// @backbase/ui-ang/esm2015/_example/button/showcase/button-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ButtonShowcaseComponent } from './button-showcase.component';
import { ButtonModule } from '@backbase/ui-ang/button';
import { IconModule } from '@backbase/ui-ang/icon';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import * as i0 from "@angular/core";
export class ButtonShowcaseModule {
}
ButtonShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ButtonShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ButtonShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ButtonShowcaseModule, declarations: [ButtonShowcaseComponent], imports: [ButtonModule, CommonModule, IconModule, LoadButtonModule] });
ButtonShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ButtonShowcaseModule, imports: [[ButtonModule, CommonModule, IconModule, LoadButtonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ButtonShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [ButtonModule, CommonModule, IconModule, LoadButtonModule],
                    declarations: [ButtonShowcaseComponent],
                    entryComponents: [ButtonShowcaseComponent],
                }]
        }] });
