// @backbase/ui-ang/esm2015/_example/currency-input/showcase/currency-input-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { CurrencyInputShowcaseComponent } from './currency-input-showcase.component';
import { CurrencyInputModule } from '@backbase/ui-ang/currency-input';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import * as i0 from "@angular/core";
export class CurrencyInputShowcaseModule {
}
CurrencyInputShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CurrencyInputShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CurrencyInputShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CurrencyInputShowcaseModule, declarations: [CurrencyInputShowcaseComponent], imports: [CommonModule, CurrencyInputModule, ReactiveFormsModule, FormsModule, InputValidationMessageModule] });
CurrencyInputShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CurrencyInputShowcaseModule, imports: [[CommonModule, CurrencyInputModule, ReactiveFormsModule, FormsModule, InputValidationMessageModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CurrencyInputShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, CurrencyInputModule, ReactiveFormsModule, FormsModule, InputValidationMessageModule],
                    declarations: [CurrencyInputShowcaseComponent],
                    entryComponents: [CurrencyInputShowcaseComponent],
                }]
        }] });
