// @backbase/payord-stop-checks-widget-ang/esm2015/lib/payord-stop-checks-widget-ang.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { StopCheckFormComponent, StopCheckFormComponentCustomizableDirective, } from './components/stop-check-form/stop-check-form.component';
import { StopCheckReviewComponent, StopCheckReviewComponentCustomizableDirective, } from './components/stop-check-review/stop-check-review.component';
import { StopCheckConfigProvider } from './config.providers';
import { FormContainerComponent } from './containers/form-container/form-container.component';
import { ReviewContainerComponent } from './containers/review-container/review-container.component';
import { PayordStopChecksWidgetAngComponent } from './payord-stop-checks-widget-ang.component';
import { AccountSelectorModule } from '@backbase/ui-ang/account-selector';
import { AlertModule } from '@backbase/ui-ang/alert';
import { AmountModule } from '@backbase/ui-ang/amount';
import { ButtonModule } from '@backbase/ui-ang/button';
import { CollapsibleModule } from '@backbase/ui-ang/collapsible';
import { CurrencyInputModule } from '@backbase/ui-ang/currency-input';
import { DropdownMenuModule } from '@backbase/ui-ang/dropdown-menu';
import { DropdownSingleSelectModule } from '@backbase/ui-ang/dropdown-single-select';
import { FieldsetModule } from '@backbase/ui-ang/fieldset';
import { HeaderModule } from '@backbase/ui-ang/header';
import { IconModule } from '@backbase/ui-ang/icon';
import { InputCheckboxModule } from '@backbase/ui-ang/input-checkbox';
import { InputDatepickerModule } from '@backbase/ui-ang/input-datepicker';
import { InputRadioGroupModule } from '@backbase/ui-ang/input-radio-group';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { ModalModule } from '@backbase/ui-ang/modal';
import { SearchBoxModule } from '@backbase/ui-ang/search-box';
import { TextareaModule } from '@backbase/ui-ang/textarea';
import { PaymentOrdersUiModule } from '@backbase/payment-orders-ang';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const components = [
    FormContainerComponent,
    ReviewContainerComponent,
    StopCheckFormComponent,
    StopCheckReviewComponent,
    PayordStopChecksWidgetAngComponent,
];
const customizables = [
    StopCheckFormComponentCustomizableDirective,
    StopCheckReviewComponentCustomizableDirective,
];
const uiModules = [
    AccountSelectorModule,
    AlertModule,
    AmountModule,
    ButtonModule,
    CollapsibleModule,
    CurrencyInputModule,
    DropdownMenuModule,
    DropdownSingleSelectModule,
    FieldsetModule,
    HeaderModule,
    IconModule,
    InputCheckboxModule,
    InputRadioGroupModule,
    InputTextModule,
    InputValidationMessageModule,
    ModalModule,
    SearchBoxModule,
    TextareaModule,
    InputDatepickerModule,
];
const innerModules = [PaymentOrdersUiModule];
/**
 * Exports the UI components, directives and modules for the stop checks widget
 */
export class PayordStopChecksWidgetAngModule {
}
/** @nocollapse */ PayordStopChecksWidgetAngModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordStopChecksWidgetAngModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ PayordStopChecksWidgetAngModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordStopChecksWidgetAngModule, declarations: [FormContainerComponent,
        ReviewContainerComponent,
        StopCheckFormComponent,
        StopCheckReviewComponent,
        PayordStopChecksWidgetAngComponent, StopCheckFormComponentCustomizableDirective,
        StopCheckReviewComponentCustomizableDirective], imports: [CommonModule,
        ReactiveFormsModule, i1.BackbaseCoreModule, PaymentOrdersUiModule, AccountSelectorModule,
        AlertModule,
        AmountModule,
        ButtonModule,
        CollapsibleModule,
        CurrencyInputModule,
        DropdownMenuModule,
        DropdownSingleSelectModule,
        FieldsetModule,
        HeaderModule,
        IconModule,
        InputCheckboxModule,
        InputRadioGroupModule,
        InputTextModule,
        InputValidationMessageModule,
        ModalModule,
        SearchBoxModule,
        TextareaModule,
        InputDatepickerModule], exports: [FormContainerComponent,
        ReviewContainerComponent,
        StopCheckFormComponent,
        StopCheckReviewComponent,
        PayordStopChecksWidgetAngComponent, StopCheckFormComponentCustomizableDirective,
        StopCheckReviewComponentCustomizableDirective, AccountSelectorModule,
        AlertModule,
        AmountModule,
        ButtonModule,
        CollapsibleModule,
        CurrencyInputModule,
        DropdownMenuModule,
        DropdownSingleSelectModule,
        FieldsetModule,
        HeaderModule,
        IconModule,
        InputCheckboxModule,
        InputRadioGroupModule,
        InputTextModule,
        InputValidationMessageModule,
        ModalModule,
        SearchBoxModule,
        TextareaModule,
        InputDatepickerModule, PaymentOrdersUiModule] });
/** @nocollapse */ PayordStopChecksWidgetAngModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordStopChecksWidgetAngModule, providers: [StopCheckConfigProvider], imports: [[
            CommonModule,
            ReactiveFormsModule,
            BackbaseCoreModule.withConfig({
                classMap: { PayordStopChecksWidgetAngComponent },
            }),
            ...innerModules,
            // UI-ANG
            ...uiModules,
        ], AccountSelectorModule,
        AlertModule,
        AmountModule,
        ButtonModule,
        CollapsibleModule,
        CurrencyInputModule,
        DropdownMenuModule,
        DropdownSingleSelectModule,
        FieldsetModule,
        HeaderModule,
        IconModule,
        InputCheckboxModule,
        InputRadioGroupModule,
        InputTextModule,
        InputValidationMessageModule,
        ModalModule,
        SearchBoxModule,
        TextareaModule,
        InputDatepickerModule, PaymentOrdersUiModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordStopChecksWidgetAngModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components, ...customizables],
                    entryComponents: [FormContainerComponent, ReviewContainerComponent],
                    imports: [
                        CommonModule,
                        ReactiveFormsModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { PayordStopChecksWidgetAngComponent },
                        }),
                        ...innerModules,
                        // UI-ANG
                        ...uiModules,
                    ],
                    providers: [StopCheckConfigProvider],
                    exports: [...components, ...customizables, ...uiModules, ...innerModules],
                }]
        }] });
