// @backbase/income-spending-analysis-widget-ang/esm2015/src/bb-month-selector-ui/month-selector.component.js
import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
import { PERIODS } from '../constants';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class MonthSelectorComponent {
    constructor() {
        this.hostRef = this;
        /**
         * Event emitter for selecting a tab.
         */
        // eslint-disable-next-line @angular-eslint/no-output-native
        this.change = new EventEmitter();
        /**
         * Default selected tab
         */
        this.selectedTab = PERIODS[Math.max(PERIODS.map((item) => item.default).indexOf(true), 0)].duration;
    }
    /**
     * Change selected tab and triggers the change event after
     * creating start, end dates based on selected tab
     */
    onPeriodRangeChange(num) {
        this.selectedTab = num;
        // (monthsRange - 1) to get data starting from this month and 5 months before it,
        // which will end up to 6 months in total (the same of 12 months)
        const startDate = new Date();
        startDate.setMonth(startDate.getMonth() - (num - 1), 1);
        this.change.emit({
            tabIndex: num,
            period: {
                start: startDate,
                end: new Date(),
            },
        });
    }
}
MonthSelectorComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MonthSelectorComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
MonthSelectorComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: MonthSelectorComponent, selector: "bb-month-selector-ui", inputs: { selectedTab: "selectedTab" }, outputs: { change: "change" }, ngImport: i0, template: "<div class=\"bb-block bb-block--lg bb-income-spending-analysis-periods bb-button-bar\">\n  <div class=\"btn-group btn-group-toggle\">\n      <button\n        data-role=\"1-month-btn\"\n        class=\"btn btn-secondary\"\n        [ngClass]=\"{active: selectedTab === 1}\"\n        value=\"1\"\n        (click)=\"onPeriodRangeChange(1)\"\n      >\n        <span\n          i18n=\"Data period|One month label@@incomeSpendingAnalysis.label.duration.month.one\"\n        >1</span>\n        <span\n          class=\"d-none d-md-inline\"\n          i18n=\"Data period|One month label@@incomeSpendingAnalysis.label.duration.month.one.largescreen\"\n        >&nbsp;month</span>\n        <span\n          class=\"d-md-none\"\n          i18n=\"Data period|One month label@@incomeSpendingAnalysis.label.duration.month.one.mobilescreen\"\n        >m</span>\n      </button>\n      <button\n        data-role=\"6-month-btn\"\n        class=\"btn btn-secondary\"\n        [ngClass]=\"{active: selectedTab === 6}\"\n        value=\"6\"\n        (click)=\"onPeriodRangeChange(6)\"\n      >\n        <span\n          i18n=\"Data period|One month label@@incomeSpendingAnalysis.label.duration.month.six\"\n        >6</span>\n        <span\n          class=\"d-none d-md-inline\"\n          i18n=\"Data period|One month label@@incomeSpendingAnalysis.label.duration.month.six.largescreen\"\n        >&nbsp;months</span>\n        <span\n          class=\"d-md-none\"\n          i18n=\"Data period|One month label@@incomeSpendingAnalysis.label.duration.month.six.mobilescreen\"\n        >m</span>\n      </button>\n      <button\n        data-role=\"12-month-btn\"\n        class=\"btn btn-secondary\"\n        [ngClass]=\"{active: selectedTab === 12}\"\n        value=\"12\"\n        (click)=\"onPeriodRangeChange(12)\"\n      >\n        <span\n          i18n=\"Data period|One month label@@incomeSpendingAnalysis.label.duration.month.twelve\"\n        >12</span>\n        <span\n          class=\"d-none d-md-inline\"\n          i18n=\"Data period|One month label@@incomeSpendingAnalysis.label.duration.month.twelve.largescreen\"\n        >&nbsp;months</span>\n        <span\n          class=\"d-md-none\"\n          i18n=\"Data period|One month label@@incomeSpendingAnalysis.label.duration.month.twelve.mobilescreen\"\n        >m</span>\n      </button>\n    </div>\n</div>\n", directives: [{ type: i1.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MonthSelectorComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-month-selector-ui',
                    templateUrl: './month-selector.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { change: [{
                type: Output
            }], selectedTab: [{
                type: Input
            }] } });
