// @backbase/retail-loans-journey-ang/esm2015/lib/components/base/loan-details/loan-details-base-component.js
import { Directive } from '@angular/core';
import { isErrorState, isLoadingState, stateMap } from '../../../utils/call-state.util';
import { LoansStoreService } from '../../../store/loans/store.service';
import * as i0 from "@angular/core";
import * as i1 from "../../../store/loans/store.service";
// eslint-disable-next-line @angular-eslint/directive-class-suffix
export class LoanDetailsBaseComponent {
    constructor(storeService) {
        this.storeService = storeService;
        /**
         * Loan details steam
         */
        this.loan$ = this.storeService.loan$;
        /**
         * Flag represents loading state
         */
        this.isLoading$ = this.storeService.loanCallState$.pipe(stateMap(isLoadingState));
        /**
         * Flag represents error state
         */
        this.isErrorState$ = this.storeService.loanCallState$.pipe(stateMap(isErrorState));
    }
}
/** @nocollapse */ LoanDetailsBaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanDetailsBaseComponent, deps: [{ token: i1.LoansStoreService }], target: i0.ɵɵFactoryTarget.Directive });
/** @nocollapse */ LoanDetailsBaseComponent.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: LoanDetailsBaseComponent, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanDetailsBaseComponent, decorators: [{
            type: Directive
        }], ctorParameters: function () { return [{ type: i1.LoansStoreService }]; } });
