// @backbase/product-summary-common-ang/esm2015/src/services/accounts/products-external-info.service.js
import { Injectable } from '@angular/core';
import { of } from 'rxjs';
import { map, pluck } from 'rxjs/operators';
import { hasExternalInfo } from '../../helpers/helpers';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/financial-institution-manager";
/**
 * A service for getting external info for products
 */
export class ProductsExternalInfoService {
    constructor(financialInstitutionsHttpService) {
        this.financialInstitutionsHttpService = financialInstitutionsHttpService;
    }
    getProductsWithExternalInfo(products) {
        const productsList = Array.isArray(products) ? products : [products];
        const financialInstitutionIds = this.getAssignedFinancialInstitutionIds(productsList);
        return financialInstitutionIds.length
            ? this.getFinancialInstitutions(financialInstitutionIds).pipe(map(institutions => Array.isArray(products)
                ? productsList.map(product => this.addExternalInfoToProduct(product, institutions))
                : this.addExternalInfoToProduct(products, institutions)))
            : of(products);
    }
    getAssignedFinancialInstitutionIds(items) {
        const itemsWithExternalInfo = items.filter(hasExternalInfo);
        const financialInstitutionIds = itemsWithExternalInfo.map(item => item.financialInstitutionId);
        return financialInstitutionIds.filter((item, index) => financialInstitutionIds.indexOf(item, index + 1) === -1);
    }
    addExternalInfoToProduct(product, institutions) {
        return Object.assign(Object.assign({}, product), this.getExternalInfo(product, institutions));
    }
    getExternalInfo(product, institutions) {
        const isExternalInfoAvailable = hasExternalInfo(product);
        return isExternalInfoAvailable
            ? { financialInstitution: institutions.find(institution => institution.id === product.financialInstitutionId) }
            : {};
    }
    getFinancialInstitutions(ids) {
        return this.financialInstitutionsHttpService.getFinancialInstitutions({ ids }).pipe(pluck('financialInstitutions'));
    }
}
ProductsExternalInfoService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ProductsExternalInfoService, deps: [{ token: i1.FinancialInstitutionManagerClientHttpService }], target: i0.ɵɵFactoryTarget.Injectable });
ProductsExternalInfoService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ProductsExternalInfoService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ProductsExternalInfoService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.FinancialInstitutionManagerClientHttpService }]; } });
