// @backbase/accounts-transactions-journey-ang/esm2015/lib/views/account-info/model/configuration.js
import { ProductKindUri } from '@backbase/product-summary-common-ang';
import { AccountInfoPropertyType } from '@backbase/accounts-transactions-journeys-shared-ui-account-ang';
const defaultConfiguration = [
    {
        title: $localize `:@@product.details.other.account.label.general.section:General`,
        properties: [
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.other.account.label.productTypeName:Account Type`,
                key: 'productTypeName',
            },
            {
                type: AccountInfoPropertyType.ACCOUNT_NUMBER,
                label: $localize `:@@product.details.other.account.label.account.number:Account Number`,
                key: 'IBAN',
                numberType: 'iban',
            },
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.other.account.label.bookedBalance:Account Balance`,
                key: 'bookedBalance',
                tooltip: $localize `:@@product.details.other.account.tooltip.bookedBalance:Account Balance`,
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.other.account.label.accountHolderNames:Account Owner(s)`,
                key: 'accountHolderNames',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.other.account.label.bankBranchCode:ACH Routing Number`,
                key: 'bankBranchCode',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.other.account.label.bankBranchCode2:FedWire Routing Number`,
                key: 'bankBranchCode2',
            },
            {
                type: AccountInfoPropertyType.ACCOUNT_STATE,
                label: $localize `:@@product.details.other.account.label.accountStatus:Account Status`,
                key: 'state.state',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.other.account.label.financialInstitution.name:Bank Name`,
                key: 'financialInstitution.name',
            },
            {
                type: AccountInfoPropertyType.UPDATE_TIME,
                label: $localize `:@@product.details.other.account.label.lastSyncDate:Last sync`,
                key: 'lastSyncDate',
            },
        ],
    },
    {
        title: $localize `:@@product.details.other.account.label.interest.details.section:Interest Details`,
        properties: [
            {
                type: AccountInfoPropertyType.PERCENT,
                label: $localize `:@@product.details.other.account.label.accountInterestRate:Interest Rate`,
                key: 'accountInterestRate',
                tooltip: $localize `:@@product.details.other.account.tooltip.accountInterestRate:The annualized cost of credit or debt-capital computed as the percentage ratio of interest to the principal`,
            },
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.other.account.label.accruedInterest:Accrued Interest`,
                key: 'accruedInterest',
                tooltip: $localize `:@@product.details.other.account.tooltip.accruedInterest:The interest that is earned (credit interest) or due (debit interest) but not settled yet`,
            },
        ],
    },
    {
        title: $localize `:@@product.details.other.account.label.other.section:Other`,
        properties: [
            {
                type: AccountInfoPropertyType.DATE,
                label: $localize `:@@product.details.other.account.label.accountOpeningDate:Account Opening Date`,
                key: 'accountOpeningDate',
                tooltip: $localize `:@@product.details.other.account.tooltip.accountOpeningDate:Account Opening Date`,
            },
            {
                type: AccountInfoPropertyType.DATE,
                label: $localize `:@@product.details.other.account.label.lastUpdateDate:Last Updated Date`,
                key: 'lastUpdateDate',
                tooltip: $localize `:@@product.details.other.account.tooltip.lastUpdateDate:Last date of parameter update for the product`,
            },
        ],
    },
];
const creditCardConfiguration = [
    {
        title: $localize `:@@product.details.credit.card.label.general.section:General`,
        properties: [
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.credit.card.label.productTypeName:Account Type`,
                key: 'productTypeName',
            },
            {
                type: AccountInfoPropertyType.ACCOUNT_NUMBER,
                label: $localize `:@@product.details.credit.card.label.number:Account Number`,
                key: 'number',
                numberType: 'cardNumber',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.credit.card.label.accountHolderNames:Account Owner(s)`,
                key: 'accountHolderNames',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.credit.card.label.cardProvider:Card Provider`,
                key: 'cardDetails.cardProvider',
                tooltip: $localize `:@@product.details.credit.card.tooltip.cardProvider:Card Provider`,
            },
            {
                type: AccountInfoPropertyType.DATE,
                label: $localize `:@@product.details.credit.card.label.validThru:Valid thru`,
                key: 'validThru',
                tooltip: $localize `:@@product.details.credit.card.tooltip.validThru:Expiration date of a credit card, after which is no longer valid`,
            },
            {
                type: AccountInfoPropertyType.ACCOUNT_STATE,
                label: $localize `:@@product.details.credit.card.label.accountStatus:Account Status`,
                key: 'state.state',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.credit.card.label.financialInstitution.name:Bank Name`,
                key: 'financialInstitution.name',
            },
            {
                type: AccountInfoPropertyType.UPDATE_TIME,
                label: $localize `:@@product.details.credit.card.label.lastSyncDate:Last sync`,
                key: 'lastSyncDate',
            },
        ],
    },
    {
        title: $localize `:@@product.details.credit.card.label.status.section:Status`,
        properties: [
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.credit.card.label.consumedAmount:Consumed Amount`,
                key: 'bookedBalance',
                tooltip: $localize `:@@product.details.credit.card.tooltip.consumedAmount:Consumed Amount`,
            },
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.credit.card.label.remainingAmount:Remaining Amount`,
                key: 'remainingCredit',
                tooltip: $localize `:@@product.details.credit.card.tooltip.remainingAmount:Remaining Amount`,
            },
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.credit.card.label.creditLimit:Credit Limit`,
                key: 'creditLimit',
                tooltip: $localize `:@@product.details.credit.card.tooltip.creditLimit:Monetary amount of the used overdraft`,
            },
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.credit.card.label.statementBalance:Statement Balance`,
                key: 'cardDetails.statementBalance',
                tooltip: $localize `:@@product.details.credit.card.tooltip.statementBalance:Statement Balance`,
            },
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.credit.card.label.availableCashCredit:Available Cash Credit Limit`,
                key: 'cardDetails.availableCashCredit',
                tooltip: $localize `:@@product.details.credit.card.tooltip.availableCashCredit:Available Cash Credit Limit`,
            },
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.credit.card.label.cashCreditLimit:Cash Credit Limit`,
                key: 'cardDetails.cashCreditLimit',
                tooltip: $localize `:@@product.details.credit.card.tooltip.cashCreditLimit:Cash Credit Limit`,
            },
        ],
    },
    {
        title: $localize `:@@product.details.credit.card.label.payment.section:Payment and Interest Details`,
        properties: [
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.credit.card.label.minimumPayment:Minimum Payment`,
                key: 'minimumPayment',
                tooltip: $localize `:@@product.details.credit.card.tooltip.minimumPayment:The minimum payment set a percentage of balance, or a fixed cash amount`,
            },
            {
                type: AccountInfoPropertyType.DATE,
                label: $localize `:@@product.details.credit.card.label.minimumPaymentDueDate:Minimum Payment Due Date`,
                key: 'minimumPaymentDueDate',
                tooltip: $localize `:@@product.details.credit.card.tooltip.minimumPaymentDueDate:Minimum Payment Due Date shown on your monthly statement to remain in good standing`,
            },
            {
                type: AccountInfoPropertyType.PERCENT,
                label: $localize `:@@product.details.credit.card.label.cashAdvanceInterestRate:Cash Advance APR`,
                key: 'interestDetails.cashAdvanceInterestRate',
                tooltip: $localize `:@@product.details.credit.card.tooltip.cashAdvanceInterestRate:Cash Advance APR`,
            },
            {
                type: AccountInfoPropertyType.PERCENT,
                label: $localize `:@@product.details.credit.card.label.applicableInterestRate:Interest Rate`,
                key: 'applicableInterestRate',
                tooltip: $localize `:@@product.details.credit.card.tooltip.applicableInterestRate:The annualized cost of credit or debt-capital computed as the percentage ratio of interest to the principal`,
            },
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.credit.card.label.lastPaymentAmount:Last Payment Amount`,
                key: 'cardDetails.lastPaymentAmount',
                tooltip: $localize `:@@product.details.credit.card.tooltip.lastPaymentAmount:Last Payment Amount`,
            },
            {
                type: AccountInfoPropertyType.DATE,
                label: $localize `:@@product.details.credit.card.label.lastPaymentDate:Last Payment Date`,
                key: 'cardDetails.lastPaymentDate',
                tooltip: $localize `:@@product.details.credit.card.tooltip.lastPaymentDate:Last Payment Date`,
            },
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.credit.card.label.previousStatementBalance:Last Statement Balance`,
                key: 'cardDetails.previousStatementBalance',
                tooltip: $localize `:@@product.details.credit.card.tooltip.previousStatementBalance:Last Statement Balance`,
            },
            {
                type: AccountInfoPropertyType.DATE,
                label: $localize `:@@product.details.credit.card.label.previousStatementDate:Last Statement Date`,
                key: 'cardDetails.previousStatementDate',
                tooltip: $localize `:@@product.details.credit.card.tooltip.previousStatementDate:Last Statement Date`,
            },
            {
                type: AccountInfoPropertyType.PERCENT,
                label: $localize `:@@product.details.credit.card.label.penaltyInterestRate:Penalty APR`,
                key: 'interestDetails.penaltyInterestRate',
                tooltip: $localize `:@@product.details.credit.card.tooltip.penaltyInterestRate:Penalty APR`,
            },
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.credit.card.label.latePaymentFee:Late Fee`,
                key: 'cardDetails.latePaymentFee',
                tooltip: $localize `:@@product.details.credit.card.tooltip.latePaymentFee:Late Fee`,
            },
        ],
    },
    {
        title: $localize `:@@product.details.credit.card.label.other.section:Other`,
        properties: [
            {
                type: AccountInfoPropertyType.DATE,
                label: $localize `:@@product.details.credit.card.label.accountOpeningDate:Account Opening Date`,
                key: 'accountOpeningDate',
                tooltip: $localize `:@@product.details.credit.card.tooltip.accountOpeningDate:The date of activation of the account in the bank's system`,
            },
            {
                type: AccountInfoPropertyType.DATE,
                label: $localize `:@@product.details.credit.card.label.lastUpdateDate:Last Updated Date`,
                key: 'lastUpdateDate',
                tooltip: $localize `:@@product.details.credit.card.tooltip.lastUpdateDate:Last date of parameter update for the product`,
            },
        ],
    },
];
const currentAccountConfiguration = [
    {
        title: $localize `:@@product.details.current.account.label.general.section:General`,
        properties: [
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.current.account.label.productTypeName:Account Type`,
                key: 'productTypeName',
            },
            {
                type: AccountInfoPropertyType.ACCOUNT_NUMBER,
                label: $localize `:@@product.details.current.account.label.iban:IBAN`,
                key: 'IBAN',
                numberType: 'iban',
            },
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.current.account.label.availableBalance:Available Balance`,
                key: 'availableBalance',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.current.account.label.accountHolderNames:Account Owner(s)`,
                key: 'accountHolderNames',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@account.details.current.account.label.bankBranchCode:ACH Routing Number`,
                key: 'bankBranchCode',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@account.details.current.account.label.bankBranchCode2:FedWire Routing Number`,
                key: 'bankBranchCode2',
            },
            {
                type: AccountInfoPropertyType.ACCOUNT_STATE,
                label: $localize `:@@product.details.current.account.label.accountStatus:Account Status`,
                key: 'state.state',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.current.account.label.financialInstitution.name:Bank Name`,
                key: 'financialInstitution.name',
            },
            {
                type: AccountInfoPropertyType.UPDATE_TIME,
                label: $localize `:@@product.details.current.account.label.lastSyncDate:Last sync`,
                key: 'lastSyncDate',
            },
        ],
    },
    {
        title: $localize `:@@product.details.current.account.label.interest.section:Interest Details`,
        properties: [
            {
                type: AccountInfoPropertyType.PERCENT,
                label: $localize `:@@product.details.current.account.label.interestRate:Interest Rate`,
                tooltip: $localize `:@@product.details.current.account.tooltip.interestRate:The annualized cost of credit or debt-capital computed as the percentage ratio of interest to the principal`,
                key: 'accountInterestRate',
            },
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.current.account.label.accruedInterest:Accrued Interest`,
                tooltip: $localize `:@@product.details.current.account.tooltip.accruedInterest:The interest that is earned (credit interest) or due (debit interest) but not settled yet`,
                key: 'accruedInterest',
            },
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.current-account.label.account.lastYearAccruedInterest:Last year’s accrued interest`,
                tooltip: $localize `:@@product.details.current-account.tooltip.lastYearAccruedInterest:Last year’s accrued interest`,
                key: 'interestDetails.lastYearAccruedInterest',
            },
        ],
    },
    {
        title: $localize `:@@product.details.current.account.label.overdraft.section:Overdraft Details`,
        properties: [
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.current.account.label.overdraftLimit:Overdraft Limit`,
                tooltip: $localize `:@@product.details.current.account.tooltip.overdraftLimit:Overdraft Limit`,
                key: 'creditLimit',
            },
            {
                type: AccountInfoPropertyType.DATE,
                label: $localize `:@@product.details.current.account.label.overdraftexpirydate:Overdraft Expiry Date`,
                tooltip: $localize `:@@product.details.current.account.tooltip.overdraftexpirydate:Expiration date of the overdraft limit`,
                key: 'creditLimitExpiryDate',
            },
        ],
    },
    {
        title: $localize `:@@product.details.current.account.label.other.section:Other`,
        properties: [
            {
                type: AccountInfoPropertyType.DATE,
                label: $localize `:@@product.details.current.account.label.accountOpeningDate:Account Opening Date`,
                tooltip: $localize `:@@product.details.current.account.tooltip.accountOpeningDate:Account Opening Date`,
                key: 'accountOpeningDate',
            },
            {
                type: AccountInfoPropertyType.DATE,
                label: $localize `:@@product.details.current.account.label.lastUpdateDate:Last Updated Date`,
                tooltip: $localize `:@@product.details.current.account.tooltip.lastUpdateDate:Last date of parameter update for the product`,
                key: 'lastUpdateDate',
            },
        ],
    },
    {
        title: $localize `:@@product.details.current.account.label.associatedDebitCards:Associated Debit Cards`,
        hideCard: true,
        properties: [
            {
                type: AccountInfoPropertyType.DEBIT_CARDS,
                key: 'debitCards',
            },
        ],
    },
];
const debitCardConfiguration = [
    {
        title: $localize `:@@product.details.debit.card.label.general.section:General`,
        properties: [
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.debit.card.label.productTypeName:Account Type`,
                key: 'productTypeName',
            },
            {
                type: AccountInfoPropertyType.ACCOUNT_NUMBER,
                label: $localize `:@@product.details.debit.card.label.iban:IBAN`,
                key: 'IBAN',
                numberType: 'iban',
            },
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.debit.card.label.availableBalance:Available Balance`,
                key: 'availableBalance',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.debit.card.label.accountHolderNames:Account Owner(s)`,
                key: 'accountHolderNames',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.debit.card.label.bankBranchCode:ACH Routing Number`,
                key: 'bankBranchCode',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.debit.card.label.bankBranchCode2:FedWire Routing Number`,
                key: 'bankBranchCode2',
            },
            {
                type: AccountInfoPropertyType.ACCOUNT_STATE,
                label: $localize `:@@product.details.debit.card.label.accountStatus:Account Status`,
                key: 'state.state',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.debit.card.label.financialInstitution.name:Bank Name`,
                key: 'financialInstitution.name',
            },
            {
                type: AccountInfoPropertyType.UPDATE_TIME,
                label: $localize `:@@product.details.debit.card.label.lastSyncDate.name:Last sync`,
                key: 'lastSyncDate',
            },
        ],
    },
    {
        title: $localize `:@@product.details.debit.card.label.other.section:Other`,
        properties: [
            {
                type: AccountInfoPropertyType.DATE,
                label: $localize `:@@product.details.debit.card.label.accountOpeningDate:Account Opening Date`,
                key: 'accountOpeningDate',
                tooltip: $localize `:@@product.details.debit.card.tooltip.accountOpeningDate:Account Opening Date`,
            },
            {
                type: AccountInfoPropertyType.DATE,
                label: $localize `:@@product.details.debit.card.label.lastUpdateDate:Last Updated Date`,
                key: 'lastUpdateDate',
                tooltip: $localize `:@@product.details.debit.card.tooltip.lastUpdateDate:Last date of parameter update for the product`,
            },
        ],
    },
];
const investmentAccountConfiguration = [
    {
        title: $localize `:@@product.details.investment.account.label.general.section:General`,
        properties: [
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.investment.account.label.productTypeName:Account Type`,
                key: 'productTypeName',
            },
            {
                type: AccountInfoPropertyType.ACCOUNT_NUMBER,
                label: $localize `:@@product.details.investment.account.label.account.number:Account Number`,
                key: 'BBAN',
                numberType: 'bban',
            },
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.investment.account.label.totalInvestmentValue:Total Investment Value`,
                key: 'totalInvestmentValue',
                tooltip: $localize `:@@product.details.investment.account.tooltip.totalInvestmentValue:Total Investment Value`,
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.investment.account.label.accountHolderNames:Account Owner(s)`,
                key: 'accountHolderNames',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.investment.account.label.bankBranchCode:ACH Routing Number`,
                key: 'bankBranchCode',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.investment.account.label.bankBranchCode2:FedWire Routing Number`,
                key: 'bankBranchCode2',
            },
            {
                type: AccountInfoPropertyType.ACCOUNT_STATE,
                label: $localize `:@@product.details.investment.account.label.accountStatus:Account Status`,
                key: 'state.state',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.investment.account.label.financialInstitution.name:Bank Name`,
                key: 'financialInstitution.name',
            },
            {
                type: AccountInfoPropertyType.UPDATE_TIME,
                label: $localize `:@@product.details.investment.account.label.lastSyncDate:Last sync`,
                key: 'lastSyncDate',
            },
        ],
    },
    {
        title: $localize `:@@product.details.investment.account.label.other.section:Other`,
        properties: [
            {
                type: AccountInfoPropertyType.DATE,
                label: $localize `:@@product.details.investment.account.label.accountOpeningDate:Account Opening Date`,
                key: 'accountOpeningDate',
                tooltip: $localize `:@@product.details.investment.account.tooltip.accountOpeningDate:Account Opening Date`,
            },
            {
                type: AccountInfoPropertyType.DATE,
                label: $localize `:@@product.details.investment.account.label.lastUpdateDate:Last Updated Date`,
                key: 'lastUpdateDate',
                tooltip: $localize `:@@product.details.investment.account.tooltip.lastUpdateDate:Last date of parameter update for the product`,
            },
        ],
    },
];
const loanConfiguration = [
    {
        title: $localize `:@@product.details.loan.label.general.section:General`,
        properties: [
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.loan.label.productTypeName:Account Type`,
                key: 'productTypeName',
            },
            {
                type: AccountInfoPropertyType.ACCOUNT_NUMBER,
                label: $localize `:@@product.details.loan.label.account.number:Account Number`,
                key: 'BBAN',
                numberType: 'bban',
            },
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.loan.label.outstandingamount:Outstanding Amount`,
                key: 'bookedBalance',
                tooltip: $localize `:@@product.details.loan.tooltip.outstandingamount:Outstanding Amount`,
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.loan.label.accountHolderNames:Account Owner(s)`,
                key: 'accountHolderNames',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.loan.label.bankBranchCode:ACH Routing Number`,
                key: 'bankBranchCode',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.loan.label.bankBranchCode2:FedWire Routing Number`,
                key: 'bankBranchCode2',
            },
            {
                type: AccountInfoPropertyType.ACCOUNT_STATE,
                label: $localize `:@@product.details.loan.label.accountStatus:Account Status`,
                key: 'state.state',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.loan.label.financialInstitution.name:Bank Name`,
                key: 'financialInstitution.name',
            },
            {
                type: AccountInfoPropertyType.UPDATE_TIME,
                label: $localize `:@@product.details.loan.label.lastSyncDate:Last sync`,
                key: 'lastSyncDate',
            },
        ],
    },
    {
        title: $localize `:@@product.details.loan.label.loan.details.section:Loan Details`,
        properties: [
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.loan.label.monthlyInstalmentAmount:Monthly Instalment Amount`,
                key: 'monthlyInstalmentAmount',
                tooltip: $localize `:@@product.details.loan.tooltip.monthlyInstalmentAmount:A fixed payment amount paid by a borrower to the bank at a specified date each calendar month`,
            },
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.loan.label.principalAmount:Principal Amount`,
                key: 'principalAmount',
                tooltip: $localize `:@@product.details.loan.tooltip.principalAmount:The amount that was (originally) contracted for the respective product`,
            },
            {
                type: AccountInfoPropertyType.PERCENT,
                label: $localize `:@@product.details.loan.label.accountInterestRate:Interest Rate`,
                key: 'accountInterestRate',
                tooltip: $localize `:@@product.details.loan.tooltip.accountInterestRate:The annualized cost of credit or debt-capital computed as the percentage ratio of interest to the principal`,
            },
            {
                type: AccountInfoPropertyType.FREQUENCY,
                label: $localize `:@@product.details.loan.label.termNumber:Term`,
                key: 'termNumber',
                unitKey: 'termUnit',
                tooltip: $localize `:@@product.details.loan.tooltip.termNumber:The period of time and/or the interest rate arranged between Bank and customer`,
            },
        ],
    },
    {
        title: $localize `:@@product.details.loan.label.other.section:Other`,
        properties: [
            {
                type: AccountInfoPropertyType.DATE,
                label: $localize `:@@product.details.loan.label.accountOpeningDate:Account Opening Date`,
                key: 'accountOpeningDate',
                tooltip: $localize `:@@product.details.loan.tooltip.accountOpeningDate:Account Opening Date`,
            },
            {
                type: AccountInfoPropertyType.DATE,
                label: $localize `:@@product.details.loan.label.lastUpdateDate:Last Updated Date`,
                key: 'lastUpdateDate',
                tooltip: $localize `:@@product.details.loan.tooltip.lastUpdateDate:Last date of parameter update for the product`,
            },
        ],
    },
];
const savingsAccountConfiguration = [
    {
        title: $localize `:@@product.details.savings.account.label.general.section:General`,
        properties: [
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.savings.account.label.productTypeName:Account Type`,
                key: 'productTypeName',
            },
            {
                type: AccountInfoPropertyType.ACCOUNT_NUMBER,
                label: $localize `:@@product.details.savings.account.label.account.number:Account Number`,
                key: 'BBAN',
                numberType: 'bban',
            },
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.savings.account.label.bookedBalance:Account Balance`,
                key: 'bookedBalance',
                tooltip: $localize `:@@product.details.savings.account.tooltip.bookedBalance:Account Balance`,
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.savings.account.label.accountHolderNames:Account Owner(s)`,
                key: 'accountHolderNames',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.savings.account.label.bankBranchCode:ACH Routing Number`,
                key: 'bankBranchCode',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.savings.account.label.bankBranchCode2:FedWire Routing Number`,
                key: 'bankBranchCode2',
            },
            {
                type: AccountInfoPropertyType.ACCOUNT_STATE,
                label: $localize `:@@product.details.savings.account.label.accountStatus:Account Status`,
                key: 'state.state',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.savings.account.label.financialInstitution.name:Bank Name`,
                key: 'financialInstitution.name',
            },
            {
                type: AccountInfoPropertyType.NUMBER,
                label: $localize `:@@product.details.savings.account.label.remainingPeriodicTransfers:Remaining Periodic Transfers`,
                key: 'remainingPeriodicTransfers',
                tooltip: $localize `:@@product.details.savings.account.tooltip.remainingPeriodicTransfers:Remaining Periodic Transfers`,
                decimalPlaces: 0,
            },
            {
                type: AccountInfoPropertyType.UPDATE_TIME,
                label: $localize `:@@product.details.savings.account.label.lastSyncDate:Last sync`,
                key: 'lastSyncDate',
            },
        ],
    },
    {
        title: $localize `:@@product.details.savings.account.label.interest.details.section:Interest Details`,
        properties: [
            {
                type: AccountInfoPropertyType.PERCENT,
                label: $localize `:@@product.details.savings.account.label.accountInterestRate:Interest Rate`,
                key: 'accountInterestRate',
                tooltip: $localize `:@@product.details.savings.account.tooltip.accountInterestRate:The annualized cost of credit or debt-capital computed as the percentage ratio of interest to the principal`,
            },
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.savings.account.label.accruedInterest:Accrued Interest`,
                key: 'accruedInterest',
                tooltip: $localize `:@@product.details.savings.account.tooltip.accruedInterest:The interest that is earned (credit interest) or due (debit interest) but not settled yet`,
            },
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.savings.account.label.minimumRequiredBalance:Minimum Required Balance`,
                key: 'minimumRequiredBalance',
                tooltip: $localize `:@@product.details.savings.account.tooltip.minimumRequiredBalance:Minimum amount that a customer must have in an account in order to receive some sort of service, such as keeping the account open or receive interest`,
            },
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.savings.account.label.lastYearAccruedInterest:Last year’s accrued interest`,
                key: 'interestDetails.lastYearAccruedInterest',
                tooltip: $localize `:@@product.details.savings.account.tooltip.lastYearAccruedInterest:Last year’s accrued interest`,
            },
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.savings.account.label.dividendWithheldYTD:Interest/Dividend Withheld YTD`,
                key: 'interestDetails.dividendWithheldYTD',
                tooltip: $localize `:@@product.details.savings.account.tooltip.dividendWithheldYTD:Interest/Dividend Withheld YTD`,
            },
            {
                type: AccountInfoPropertyType.PERCENT,
                label: $localize `:@@product.details.savings.account.label.annualPercentageYield:Annual Percentage Yield`,
                key: 'interestDetails.annualPercentageYield',
                tooltip: $localize `:@@product.details.savings.account.tooltip.annualPercentageYield:Annual Percentage Yield`,
            },
        ],
    },
    {
        title: $localize `:@@product.details.savings.account.label.other.section:Other`,
        properties: [
            {
                type: AccountInfoPropertyType.DATE,
                label: $localize `:@@product.details.savings.account.label.accountOpeningDate:Account Opening Date`,
                key: 'accountOpeningDate',
                tooltip: $localize `:@@product.details.savings.account.tooltip.accountOpeningDate:Account Opening Date`,
            },
            {
                type: AccountInfoPropertyType.DATE,
                label: $localize `:@@product.details.savings.account.label.lastUpdateDate:Last Updated Date`,
                key: 'lastUpdateDate',
                tooltip: $localize `:@@product.details.savings.account.tooltip.lastUpdateDate:Last date of parameter update for the product`,
            },
        ],
    },
];
const termDepositConfiguration = [
    {
        title: $localize `:@@product.details.term.deposit.label.general.section:General`,
        properties: [
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.term.deposit.label.productTypeName:Account Type`,
                key: 'productTypeName',
            },
            {
                type: AccountInfoPropertyType.ACCOUNT_NUMBER,
                label: $localize `:@@product.details.term.deposit.label.account.number:Account Number`,
                key: 'BBAN',
                numberType: 'bban',
            },
            {
                type: AccountInfoPropertyType.ACCOUNT_NUMBER,
                label: $localize `:@@product.details.term.deposit.label.iban:IBAN`,
                key: 'IBAN',
                numberType: 'iban',
            },
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.term.deposit.label.principalAmount:Principal Amount`,
                key: 'principalAmount',
                tooltip: $localize `:@@product.details.term.deposit.tooltip.principalAmount:The amount that was (originally) contracted for the respective product`,
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.term.deposit.label.bankBranchCode:ACH Routing Number`,
                key: 'bankBranchCode',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.term.deposit.label.bankBranchCode2:FedWire Routing Number`,
                key: 'bankBranchCode2',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.term.deposit.label.accountHolderNames:Account Owner(s)`,
                key: 'accountHolderNames',
            },
            {
                type: AccountInfoPropertyType.ACCOUNT_STATE,
                label: $localize `:@@product.details.term.deposit.label.accountStatus:Account Status`,
                key: 'state.state',
            },
            {
                type: AccountInfoPropertyType.TEXT,
                label: $localize `:@@product.details.term.deposit.label.financialInstitution.name:Bank Name`,
                key: 'financialInstitution.name',
            },
            {
                type: AccountInfoPropertyType.UPDATE_TIME,
                label: $localize `:@@product.details.term.deposit.label.lastSyncDate:Last sync`,
                key: 'lastSyncDate',
            },
        ],
    },
    {
        title: $localize `:@@product.details.term.deposit.label.interest.details.section:Interest Details`,
        properties: [
            {
                type: AccountInfoPropertyType.PERCENT,
                label: $localize `:@@product.details.term.deposit.label.accountInterestRate:Interest Rate`,
                key: 'accountInterestRate',
                tooltip: $localize `:@@product.details.term.deposit.tooltip.accountInterestRate:The annualized cost of credit or debt-capital computed as the percentage ratio of interest to the principal`,
            },
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.term.deposit.label.accruedInterest:Accrued Interest`,
                key: 'accruedInterest',
                tooltip: $localize `:@@product.details.term.deposit.tooltip.accruedInterest:The interest that is earned (credit interest) or due (debit interest) but not settled yet`,
            },
            {
                type: AccountInfoPropertyType.FREQUENCY,
                label: $localize `:@@product.details.term.deposit.label.interestPaymentFrequencyNumber:Interest Payment Frequency`,
                key: 'interestPaymentFrequencyNumber',
                unitKey: 'interestPaymentFrequencyUnit',
                tooltip: $localize `:@@product.details.term.deposit.tooltip.interestPaymentFrequencyNumber:Number of times per year when the accumulated interest is settled to the product on a regular basis`,
            },
            {
                type: AccountInfoPropertyType.RENEWAL_STATUS,
                label: $localize `:@@product.details.term.deposit.label.autoRenewalIndicator:Auto Renewal Indicator`,
                key: 'autoRenewalIndicator',
                tooltip: $localize `:@@product.details.term.deposit.tooltip.autoRenewalIndicator:Indicates whether or not a product is to be continued after maturity automatically`,
            },
        ],
    },
    {
        title: $localize `:@@product.details.term.deposit.label.maturity.details.section:Maturity Details`,
        properties: [
            {
                type: AccountInfoPropertyType.AMOUNT,
                label: $localize `:@@product.details.term.deposit.label.maturityAmount:Maturity Balance`,
                key: 'maturityAmount',
                tooltip: $localize `:@@product.details.term.deposit.tooltip.maturityAmount:Amount payable at the end of a holding period of a product`,
            },
            {
                type: AccountInfoPropertyType.FREQUENCY,
                label: $localize `:@@product.details.term.deposit.label.termNumber:Term`,
                key: 'termNumber',
                unitKey: 'termUnit',
                tooltip: $localize `:@@product.details.term.deposit.tooltip.termNumber:The period of time and/or the interest rate arranged between Bank and customer`,
            },
            {
                type: AccountInfoPropertyType.DATE,
                label: $localize `:@@product.details.term.deposit.label.startDate:Start Date`,
                key: 'startDate',
                tooltip: $localize `:@@product.details.term.deposit.tooltip.startDate:Start Date`,
            },
            {
                type: AccountInfoPropertyType.DATE,
                label: $localize `:@@product.details.term.deposit.label.maturityDate:Maturity Date`,
                key: 'maturityDate',
                tooltip: $localize `:@@product.details.term.deposit.tooltip.maturityDate:End term of a holding period`,
            },
        ],
    },
    {
        title: $localize `:@@product.details.term.deposit.label.other.section:Other`,
        properties: [
            {
                type: AccountInfoPropertyType.DATE,
                label: $localize `:@@product.details.term.deposit.label.accountOpeningDate:Account Opening Date`,
                key: 'accountOpeningDate',
                tooltip: $localize `:@@product.details.term.deposit.tooltip.accountOpeningDate:Account Opening Date`,
            },
            {
                type: AccountInfoPropertyType.DATE,
                label: $localize `:@@product.details.term.deposit.label.lastUpdateDate:Last Updated Date`,
                key: 'lastUpdateDate',
                tooltip: $localize `:@@product.details.term.deposit.tooltip.lastUpdateDate:Last date of parameter update for the product`,
            },
        ],
    },
];
/**
 * Default configuration for account info
 */
export const defaultAccountInfoConfiguration = {
    default: defaultConfiguration,
    [ProductKindUri.CREDIT_CARD]: creditCardConfiguration,
    [ProductKindUri.CURRENT_ACCOUNT]: currentAccountConfiguration,
    [ProductKindUri.DEBIT_CARD]: debitCardConfiguration,
    [ProductKindUri.INVESTMENT_ACCOUNT]: investmentAccountConfiguration,
    [ProductKindUri.LOAN]: loanConfiguration,
    [ProductKindUri.SAVINGS_ACCOUNT]: savingsAccountConfiguration,
    [ProductKindUri.TERM_DEPOSIT]: termDepositConfiguration,
};
