// @backbase/remote-config-ang/esm2015/core/parser/value-parser.js
import { ArrayValueTypeDescriptor, ObjectValueTypeDescriptor, ValueType, } from '../type-extractor/extracted-value-types';
import { isArray, isBoolean, isNull, isNumber, isObject, isString } from '../utils/is';
import { ParseFailure, ParseSuccess } from './parse-result';
export var ParseError;
(function (ParseError) {
    ParseError["TypeMismatch"] = "TypeMismatch";
    ParseError["UnknownPropertyType"] = "UnknownPropertyType";
})(ParseError || (ParseError = {}));
export class ValueParser {
    parseAs(rawValue, valueType) {
        if (valueType instanceof ArrayValueTypeDescriptor) {
            return this.parseArray(rawValue);
        }
        if (valueType instanceof ObjectValueTypeDescriptor) {
            return this.parseObjectAs(rawValue, valueType);
        }
        return this.parsePrimitiveAs(rawValue, valueType);
    }
    parseArray(value) {
        return isArray(value) ? ParseSuccess.create(value) : ParseFailure.create(ParseError.TypeMismatch);
    }
    parseObjectAs(rawValue, valueType) {
        if (!isObject(rawValue)) {
            return ParseFailure.create(ParseError.TypeMismatch);
        }
        const objectRawValue = rawValue;
        const result = {};
        for (const propertyName of Object.keys(valueType.properties)) {
            const propertyValueType = valueType.properties[propertyName];
            const propertyValueParseResult = this.parseAs(objectRawValue[propertyName], propertyValueType);
            if (propertyValueParseResult instanceof ParseFailure) {
                return ParseFailure.create(propertyValueParseResult.reason);
            }
            result[propertyName] = propertyValueParseResult.result;
        }
        return ParseSuccess.create(result);
    }
    parsePrimitiveAs(value, valueType) {
        if (valueType.type === ValueType.Boolean) {
            return isBoolean(value) ? ParseSuccess.create(value) : ParseFailure.create(ParseError.TypeMismatch);
        }
        if (valueType.type === ValueType.Number) {
            return isNumber(value) ? ParseSuccess.create(value) : ParseFailure.create(ParseError.TypeMismatch);
        }
        if (valueType.type === ValueType.Null) {
            return isNull(value) ? ParseSuccess.create(value) : ParseFailure.create(ParseError.TypeMismatch);
        }
        return isString(value) ? ParseSuccess.create(value) : ParseFailure.create(ParseError.TypeMismatch);
    }
}
