// @backbase/ui-ang/esm2015/_example/switch/examples/switch.example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/forms";
import * as i2 from "@backbase/ui-ang/switch";
export class SwitchExampleComponent {
    constructor(formBuilder) {
        this.formBuilder = formBuilder;
        this.vForm = this.formBuilder.group({
            notifications: true,
        });
    }
}
SwitchExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SwitchExampleComponent, deps: [{ token: i1.FormBuilder }], target: i0.ɵɵFactoryTarget.Component });
SwitchExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: SwitchExampleComponent, selector: "bb-switch-example-ui", ngImport: i0, template: "<bb-switch-ui\n  [labelPosition]=\"'right'\"\n  [label]=\"'Show notifications'\"\n  formControlName=\"notifications\"\n>\n</bb-switch-ui>\n", components: [{ type: i2.SwitchComponent, selector: "bb-switch-ui", inputs: ["labelPosition"] }], directives: [{ type: i1.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { type: i1.FormControlName, selector: "[formControlName]", inputs: ["disabled", "formControlName", "ngModel"], outputs: ["ngModelChange"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SwitchExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-switch-example-ui',
                    templateUrl: './switch.example.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.FormBuilder }]; } });
