// @backbase/actions-common-ang/esm2015/src/services/actions-notifications-settings-data.service.js
import { Injectable } from '@angular/core';
import { of, throwError } from 'rxjs';
import { map, switchMap } from 'rxjs/operators';
import { EMPTY_RESPONSE_ERROR } from '../constants/errors';
import { actionRecipeFormItemToRequest, arrangementActionRecipeToRequest, fromHttpResponse } from '../helpers/mappers';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/actions";
export const ACTION_RECIPES_LIMIT = 999999;
export class ActionsNotificationsSettingsDataService {
    constructor(dataHttpService, actionSpecificationsService) {
        this.dataHttpService = dataHttpService;
        this.actionSpecificationsService = actionSpecificationsService;
    }
    /**
     * Method to get action recipes.
     *
     * @param params - params object
     * @returns action recipe array
     */
    getActionRecipes(params = {}) {
        const requestParams = Object.assign({ limit: ACTION_RECIPES_LIMIT }, params);
        return this.dataHttpService.getActionRecipes(requestParams).pipe(map(fromHttpResponse), map(response => response.actionRecipes));
    }
    /**
     * Method to get action recipe specifications.
     *
     * @returns action recipe specifications
     */
    getSpecifications() {
        return this.actionSpecificationsService.getActionRecipeSpecifications({}).pipe(map(fromHttpResponse), map(response => response.actionRecipeSpecifications || []));
    }
    /**
     * Method to save action recipe.
     *
     * @param action - recipe form value
     * @param arrangementId - arrangement ID
     * @returns batch processing response
     */
    saveActionRecipe(actionRecipeFormItem, arrangementId) {
        const recipeToRequest = arrangementId
            ? arrangementActionRecipeToRequest(arrangementId)
            : actionRecipeFormItemToRequest;
        const batchProcessingPutRequestBody = { actionRecipes: [recipeToRequest(actionRecipeFormItem)] };
        return this.dataHttpService
            .putBatchProcessing({ batchProcessingPutRequestBody })
            .pipe(switchMap(body => (body ? of(body) : throwError(EMPTY_RESPONSE_ERROR))));
    }
}
ActionsNotificationsSettingsDataService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsNotificationsSettingsDataService, deps: [{ token: i1.ActionRecipesHttpService }, { token: i1.ActionRecipeSpecificationsHttpService }], target: i0.ɵɵFactoryTarget.Injectable });
ActionsNotificationsSettingsDataService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsNotificationsSettingsDataService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsNotificationsSettingsDataService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.ActionRecipesHttpService }, { type: i1.ActionRecipeSpecificationsHttpService }]; } });
