// @backbase/loans-collection-ui-ang/esm2015/lib/loan-list-item/loan-item-name/loan-item-name.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HeaderModule } from '@backbase/ui-ang/header';
import { LoanItemNameComponent } from './loan-item-name.component';
import * as i0 from "@angular/core";
export class LoanItemNameModule {
}
/** @nocollapse */ LoanItemNameModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemNameModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanItemNameModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemNameModule, declarations: [LoanItemNameComponent], imports: [CommonModule, HeaderModule], exports: [LoanItemNameComponent] });
/** @nocollapse */ LoanItemNameModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemNameModule, imports: [[CommonModule, HeaderModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemNameModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanItemNameComponent],
                    imports: [CommonModule, HeaderModule],
                    exports: [LoanItemNameComponent],
                }]
        }] });
