// @backbase/loans-collection-ui-ang/esm2015/lib/loan-list-item/loan-item-name/loan-item-name.component.js
import { Component, ChangeDetectionStrategy, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class LoanItemNameComponent {
}
/** @nocollapse */ LoanItemNameComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemNameComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ LoanItemNameComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: LoanItemNameComponent, selector: "bb-loan-item-name", inputs: { loanName: "loanName", headingLevel: "headingLevel" }, ngImport: i0, template: "<span class=\"sr-only\" i18n=\"Loan item name a11y prefix@@loans.loan-item.loan-name.a11y.prefix\">Name:</span>\n<span\n  role=\"heading\"\n  [attr.aria-level]=\"headingLevel\"\n  class=\"bb-heading-{{ headingLevel }} bb-text-bold bb-block bb-block--xs break-word\"\n  >{{ loanName }}</span\n>\n<span class=\"sr-only\">,</span>\n", changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemNameComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-loan-item-name',
                    templateUrl: './loan-item-name.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { loanName: [{
                type: Input
            }], headingLevel: [{
                type: Input
            }] } });
