// @backbase/ui-ang/esm2015/_example/product-item-current-account/examples/product-item-current-account.example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/product-item-current-account";
export class ProductItemCurrentAccountExampleComponent {
    constructor() {
        this.product = {
            id: 'b387bfa5-75fd-410e-bb86-7f764f8ef69c',
            name: 'Ricardo Norton\'s Account',
            currency: 'EUR',
            amount: '83652.32',
            productNumber: 'NL75INGB3400120056',
        };
        this.interest = {
            id: 'd2d85314-e60c-44af-ba98-8f704dad5970',
            name: 'Accrued Interest',
            currency: 'EUR',
            amount: '880.54',
            productNumber: 'NL75INGB3400120056',
        };
        this.credit = {
            id: 'e9af649e-e338-4b3e-808b-62e9feae0865',
            name: 'Credit Limit',
            currency: 'EUR',
            amount: '7500',
        };
        this.productDetails = {
            owner: 'Richard Norton',
        };
        this.productNumberFormat = {
            length: 18,
            maskRange: [0, 12],
            segments: 4,
        };
    }
}
ProductItemCurrentAccountExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCurrentAccountExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ProductItemCurrentAccountExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ProductItemCurrentAccountExampleComponent, selector: "bb-product-item-current-account-example-ui", ngImport: i0, template: "<bb-product-item-current-account-ui\n  [title]=\"product.name\"\n  [amount]=\"product.amount\"\n  [productNumber]=\"product.productNumber\"\n  [currency]=\"product.currency\"\n  [productNumberFormat]=\"productNumberFormat\"\n  [mainSectionCurrency]=\"interest.currency\"\n  [mainSectionAmount]=\"interest.amount\"\n  [mainSectionTitle]=\"interest.name\"\n  [secondarySectionCurrency]=\"credit.currency\"\n  [secondarySectionAmount]=\"credit.amount\"\n  [secondarySectionTitle]=\"credit.name\"\n  [productDetailsInfo]=\"productDetails\"\n  [highlight]=\"false\"\n  [favorite]=\"true\"\n>\n</bb-product-item-current-account-ui>\n", components: [{ type: i1.ProductItemCurrentAccountComponent, selector: "bb-product-item-current-account-ui", inputs: ["title", "productNumber", "productNumberFormat", "currency", "amount", "mainSectionCurrency", "mainSectionAmount", "mainSectionTitle", "secondarySectionCurrency", "secondarySectionAmount", "secondarySectionTitle", "highlight", "productDetailsInfo", "active", "favorite", "favoriteIconColor"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCurrentAccountExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-product-item-current-account-example-ui',
                    templateUrl: './product-item-current-account.example.component.html',
                }]
        }] });
