// @backbase/device-information-widget-ang/esm2015/src/device-information-widget.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { BadgeModule, ButtonModule, DropdownMenuModule, EmptyStateModule, ErrorCommonStateModule, HeaderModule, IconModule, InputTextModule, InputValidationMessageModule, LoadButtonModule, LoadingIndicatorModule, ModalModule, NotificationModule, } from '@backbase/ui-ang';
import { DeviceConfirmationModalComponent, DeviceInformationConfirmationModalCustomizableDirective, } from './components/bb-device-confirmation-modal-ui/device-confirmation-modal.component';
import { DeviceInformationComponent, DeviceInformationWidgetCustomizableDirective, } from './components/bb-device-information-ui/device-information.component';
import { DeviceListItemComponent } from './components/bb-device-list-item-ui/device-list-item.component';
import { DeviceRegisteredOnComponent } from './components/bb-device-registered-on-ui/device-registered-on.component';
import { DeviceVendorModelComponent } from './components/bb-device-vendor-model-ui/device-vendor-model.component';
import { DeviceInformationFormCustomizableDirective, ManageDeviceFormComponent, } from './components/bb-manage-device-form-ui/manage-device-form.component';
import { DeviceInformationWidgetComponent } from './device-information-widget.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const componentModules = [
    DeviceInformationComponent,
    DeviceInformationWidgetComponent,
    DeviceListItemComponent,
    ManageDeviceFormComponent,
    DeviceVendorModelComponent,
    DeviceConfirmationModalComponent,
    DeviceRegisteredOnComponent,
];
const componentDirectives = [
    DeviceInformationWidgetCustomizableDirective,
    DeviceInformationFormCustomizableDirective,
    DeviceInformationConfirmationModalCustomizableDirective,
];
const uiModules = [
    HeaderModule,
    LoadingIndicatorModule,
    ErrorCommonStateModule,
    IconModule,
    ButtonModule,
    InputValidationMessageModule,
    InputTextModule,
    LoadButtonModule,
    DropdownMenuModule,
    ModalModule,
    NotificationModule,
    ReactiveFormsModule,
    EmptyStateModule,
    BadgeModule,
];
export class DeviceInformationWidgetModule {
}
DeviceInformationWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: DeviceInformationWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
DeviceInformationWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: DeviceInformationWidgetModule, declarations: [DeviceInformationComponent,
        DeviceInformationWidgetComponent,
        DeviceListItemComponent,
        ManageDeviceFormComponent,
        DeviceVendorModelComponent,
        DeviceConfirmationModalComponent,
        DeviceRegisteredOnComponent, DeviceInformationWidgetCustomizableDirective,
        DeviceInformationFormCustomizableDirective,
        DeviceInformationConfirmationModalCustomizableDirective], imports: [CommonModule, i1.BackbaseCoreModule, HeaderModule,
        LoadingIndicatorModule,
        ErrorCommonStateModule,
        IconModule,
        ButtonModule,
        InputValidationMessageModule,
        InputTextModule,
        LoadButtonModule,
        DropdownMenuModule,
        ModalModule,
        NotificationModule,
        ReactiveFormsModule,
        EmptyStateModule,
        BadgeModule], exports: [DeviceInformationComponent,
        DeviceInformationWidgetComponent,
        DeviceListItemComponent,
        ManageDeviceFormComponent,
        DeviceVendorModelComponent,
        DeviceConfirmationModalComponent,
        DeviceRegisteredOnComponent, HeaderModule,
        LoadingIndicatorModule,
        ErrorCommonStateModule,
        IconModule,
        ButtonModule,
        InputValidationMessageModule,
        InputTextModule,
        LoadButtonModule,
        DropdownMenuModule,
        ModalModule,
        NotificationModule,
        ReactiveFormsModule,
        EmptyStateModule,
        BadgeModule, DeviceInformationWidgetCustomizableDirective,
        DeviceInformationFormCustomizableDirective,
        DeviceInformationConfirmationModalCustomizableDirective] });
DeviceInformationWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: DeviceInformationWidgetModule, imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { DeviceInformationWidgetComponent },
            }),
            ...uiModules,
        ], HeaderModule,
        LoadingIndicatorModule,
        ErrorCommonStateModule,
        IconModule,
        ButtonModule,
        InputValidationMessageModule,
        InputTextModule,
        LoadButtonModule,
        DropdownMenuModule,
        ModalModule,
        NotificationModule,
        ReactiveFormsModule,
        EmptyStateModule,
        BadgeModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: DeviceInformationWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { DeviceInformationWidgetComponent },
                        }),
                        ...uiModules,
                    ],
                    declarations: [...componentModules, ...componentDirectives],
                    exports: [...componentModules, ...uiModules, ...componentDirectives],
                }]
        }] });
