// @backbase/ui-ang/esm2015/_example/search-box/examples/search-box.example.component.js
import { Component } from '@angular/core';
import { map } from 'rxjs/operators';
import { FormControl, FormGroup } from '@angular/forms';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/search-box";
import * as i2 from "@angular/forms";
export class SearchBoxExampleComponent {
    constructor() {
        this.form = new FormGroup({
            query: new FormControl(''),
        });
        this.typeaheadOptions = {
            ngbTypeahead: this.search,
        };
    }
    startSearching(value) {
        alert(`You are searching for ${value}`);
    }
    search(text) {
        const searchItems = ['John', 'Jane', 'James', 'Joey'];
        return text.pipe(map((term) => searchItems.filter((item) => item.toLowerCase().indexOf(term.toLowerCase()) === 0)));
    }
}
SearchBoxExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SearchBoxExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
SearchBoxExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: SearchBoxExampleComponent, selector: "bb-search-box-example-ui", ngImport: i0, template: "<form [formGroup]=\"form\">\n  <h4 id=\"ariaLabelledby\">Example Search Box</h4>\n  <div class=\"bb-block bb-block--lg\">\n    <h5 id=\"ariaDescribedby\">\n      Search box with search and clear button.\n    </h5>\n    <bb-search-box-ui\n      formControlName=\"query\"\n      [showClear]=\"true\"\n      [showSearch]=\"true\"\n      placeholder=\"Search box with autocomplete. Start typing with letter 'J'\"\n      [typeaheadOptions]=\"typeaheadOptions\"\n      (submit)=\"startSearching($event)\"\n      [aria-labelledby]=\"'ariaLabelledby'\"\n      [aria-describedby]=\"'ariaDescribedby'\"\n    ></bb-search-box-ui>\n  </div>\n</form>\n", components: [{ type: i1.SearchBoxComponent, selector: "bb-search-box-ui", inputs: ["placeholder", "maxLength", "squareBorder", "typeaheadOptions", "showSearch", "showClear", "searchLabel", "clearLabel", "autocomplete"], outputs: ["submit", "clear", "valueChange"] }], directives: [{ type: i2.ɵNgNoValidate, selector: "form:not([ngNoForm]):not([ngNativeValidate])" }, { type: i2.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { type: i2.FormGroupDirective, selector: "[formGroup]", inputs: ["formGroup"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { type: i2.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { type: i2.FormControlName, selector: "[formControlName]", inputs: ["disabled", "formControlName", "ngModel"], outputs: ["ngModelChange"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SearchBoxExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-search-box-example-ui',
                    templateUrl: './search-box.example.component.html',
                }]
        }] });
