// @backbase/ui-ang/esm2015/stepper/stepper.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { IconModule } from '@backbase/ui-ang/icon';
import { CustomStepperStepDirective, StepperComponent } from './stepper.component';
import { StepperStepComponent } from './stepper-step/stepper-step.component';
import * as i0 from "@angular/core";
export class StepperModule {
}
StepperModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StepperModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
StepperModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StepperModule, declarations: [StepperComponent, StepperStepComponent, CustomStepperStepDirective], imports: [CommonModule, IconModule], exports: [StepperComponent, StepperStepComponent, CustomStepperStepDirective] });
StepperModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StepperModule, imports: [[CommonModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StepperModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, IconModule],
                    declarations: [StepperComponent, StepperStepComponent, CustomStepperStepDirective],
                    exports: [StepperComponent, StepperStepComponent, CustomStepperStepDirective],
                }]
        }] });
