// @backbase/payment-request-widget-ang/esm2015/src/payment-request-widget.component.js
import { Component, Output, EventEmitter } from '@angular/core';
import { PaymentRequestService } from './payment-request-widget/payment-request.service';
import * as i0 from "@angular/core";
import * as i1 from "./payment-request-widget/payment-request-container/payment-request-container.component";
export class PaymentRequestWidgetComponent {
    constructor() {
        /**
         * Emitter that fires after allowing payment
         */
        this.allowPaymentRequestRedirectUrl = new EventEmitter();
        /**
         * Emitter that fires after rejecting payment
         */
        this.rejectPaymentRequestRedirectUrl = new EventEmitter();
    }
    redirectToAllowUrl(url) {
        this.allowPaymentRequestRedirectUrl.next(url);
    }
    redirectToRejectUrl(url) {
        this.rejectPaymentRequestRedirectUrl.next(url);
    }
}
PaymentRequestWidgetComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentRequestWidgetComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
PaymentRequestWidgetComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: PaymentRequestWidgetComponent, selector: "bb-payment-request-widget", outputs: { allowPaymentRequestRedirectUrl: "allowPaymentRequestRedirectUrl", rejectPaymentRequestRedirectUrl: "rejectPaymentRequestRedirectUrl" }, providers: [PaymentRequestService], ngImport: i0, template: `
    <bb-payment-request-container
      (redirectToAllowUrl)="redirectToAllowUrl($event)"
      (redirectToRejectUrl)="redirectToRejectUrl($event)"></bb-payment-request-container>
  `, isInline: true, components: [{ type: i1.PaymentRequestContainerComponent, selector: "bb-payment-request-container", outputs: ["redirectToAllowUrl", "redirectToRejectUrl"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentRequestWidgetComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-payment-request-widget',
                    template: `
    <bb-payment-request-container
      (redirectToAllowUrl)="redirectToAllowUrl($event)"
      (redirectToRejectUrl)="redirectToRejectUrl($event)"></bb-payment-request-container>
  `,
                    providers: [PaymentRequestService],
                }]
        }], propDecorators: { allowPaymentRequestRedirectUrl: [{
                type: Output
            }], rejectPaymentRequestRedirectUrl: [{
                type: Output
            }] } });
