// @backbase/payment-request-widget-ang/esm2015/src/payment-request-widget.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { ConsentUiModule } from '@backbase/consent-ui';
import { ButtonModule } from '@backbase/ui-ang/button';
import { AlertModule } from '@backbase/ui-ang/alert';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { ProductItemBasicAccountModule } from '@backbase/ui-ang/product-item-basic-account';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { AmountModule } from '@backbase/ui-ang/amount';
import { NotificationModule } from '@backbase/ui-ang/notification';
import { AccountSelectorModule } from '@backbase/ui-ang/account-selector';
import { PaymentRequestAmountComponent, PaymentRequestAmountCustomizableDirective, } from './payment-request-widget/payment-request-amount/payment-request-amount.component';
import { PaymentRequestErrorComponent } from './payment-request-widget/payment-request-error/payment-request-error.component';
import { PaymentRequestTppInformationComponent, PaymentRequestTppInformationCustomizableDirective, } from './payment-request-widget/payment-request-tpp-information/payment-request-tpp-information.component';
import { PaymentRequestAccountSelectorComponent, PaymentRequestAccountSelectorCustomizableDirective, } from './payment-request-widget/payment-request-account-selector/payment-request-account-selector.component';
import { PaymentRequestContainerComponent, PaymentRequestWidgetCustomizableDirective, } from './payment-request-widget/payment-request-container/payment-request-container.component';
import { PaymentRequestWidgetComponent } from './payment-request-widget.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const components = [
    PaymentRequestWidgetComponent,
    PaymentRequestContainerComponent,
    PaymentRequestAccountSelectorComponent,
    PaymentRequestTppInformationComponent,
    PaymentRequestErrorComponent,
    PaymentRequestAmountComponent,
];
const directives = [
    PaymentRequestWidgetCustomizableDirective,
    PaymentRequestTppInformationCustomizableDirective,
    PaymentRequestAccountSelectorCustomizableDirective,
    PaymentRequestAmountCustomizableDirective,
];
const uiModules = [
    ConsentUiModule,
    ButtonModule,
    AlertModule,
    LoadingIndicatorModule,
    ProductItemBasicAccountModule,
    EmptyStateModule,
    AmountModule,
    NotificationModule,
    AccountSelectorModule,
    AmountModule,
];
export class PaymentRequestWidgetModule {
}
PaymentRequestWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentRequestWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PaymentRequestWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentRequestWidgetModule, declarations: [PaymentRequestWidgetComponent,
        PaymentRequestContainerComponent,
        PaymentRequestAccountSelectorComponent,
        PaymentRequestTppInformationComponent,
        PaymentRequestErrorComponent,
        PaymentRequestAmountComponent, PaymentRequestWidgetCustomizableDirective,
        PaymentRequestTppInformationCustomizableDirective,
        PaymentRequestAccountSelectorCustomizableDirective,
        PaymentRequestAmountCustomizableDirective], imports: [CommonModule, ConsentUiModule,
        ButtonModule,
        AlertModule,
        LoadingIndicatorModule,
        ProductItemBasicAccountModule,
        EmptyStateModule,
        AmountModule,
        NotificationModule,
        AccountSelectorModule,
        AmountModule, i1.BackbaseCoreModule], exports: [PaymentRequestWidgetComponent,
        PaymentRequestContainerComponent,
        PaymentRequestAccountSelectorComponent,
        PaymentRequestTppInformationComponent,
        PaymentRequestErrorComponent,
        PaymentRequestAmountComponent, PaymentRequestWidgetCustomizableDirective,
        PaymentRequestTppInformationCustomizableDirective,
        PaymentRequestAccountSelectorCustomizableDirective,
        PaymentRequestAmountCustomizableDirective, ConsentUiModule,
        ButtonModule,
        AlertModule,
        LoadingIndicatorModule,
        ProductItemBasicAccountModule,
        EmptyStateModule,
        AmountModule,
        NotificationModule,
        AccountSelectorModule,
        AmountModule] });
PaymentRequestWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentRequestWidgetModule, imports: [[
            CommonModule,
            ...uiModules,
            BackbaseCoreModule.withConfig({
                classMap: { PaymentRequestWidgetComponent },
            }),
        ], ConsentUiModule,
        ButtonModule,
        AlertModule,
        LoadingIndicatorModule,
        ProductItemBasicAccountModule,
        EmptyStateModule,
        AmountModule,
        NotificationModule,
        AccountSelectorModule,
        AmountModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentRequestWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        ...uiModules,
                        BackbaseCoreModule.withConfig({
                            classMap: { PaymentRequestWidgetComponent },
                        }),
                    ],
                    declarations: [...components, ...directives],
                    exports: [...components, ...directives, ...uiModules],
                }]
        }] });
