// @backbase/ui-ang/esm2015/layout/bb-layout-lib/layout.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { merge, Subject, of, EMPTY } from 'rxjs';
import { distinctUntilChanged, map, shareReplay, filter, withLatestFrom, switchMap, tap } from 'rxjs/operators';
import { LAYOUT_CONFIG_TOKEN } from './layout.model';
import { NavigationEnd } from '@angular/router';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/media-query-lib";
import * as i2 from "@angular/router";
/**
 * ## Navigation Type Horizontal
 * - On large desktop show the menu in the topbar. On mobile show a side menu
 *
 * ## Navigation Type Vertical
 * - On large desktop, show the sidebar. Toggling collapses the nav items into icons.
 * - On tablet and small desktop, collapse the nav items into icons. Toggling shows the full nav items.
 * - On mobile, don't collapse the sidebar, but hide it. Toggling shows the sidebar.
 */
export class LayoutService {
    constructor(mediaQueryService, router, config) {
        this.mediaQueryService = mediaQueryService;
        this.router = router;
        this.config = config;
        this.isHighResolutionViewSubject = new Subject();
        this.isHighResolutionView$ = this.isHighResolutionViewSubject.asObservable();
        this.manuallyExpanded$ = new Subject();
        this.isNavExpanded = false;
        /**
         * Default breakpoint used by business
         */
        if (!this.config) {
            this.config = { collapseBreakpoint: 'xl' };
        }
        /**
         * Observing the changes for screen resize
         */
        const isLowResolutionView$ = this.mediaQueryService.isMediaBreakpointMatches('max-width', this.config.collapseBreakpoint);
        /**
         * In mobile view after navigation to another route the sidebar should collapse
         */
        const collapseOnRouterNav$ = this.router.events.pipe(filter((event) => event instanceof NavigationEnd), withLatestFrom(isLowResolutionView$), switchMap(([, isLowResolutionView]) => (isLowResolutionView ? of(false) : EMPTY)));
        this.navigationExpanded$ = merge(isLowResolutionView$.pipe(map((isLowResolutionView) => {
            this.isHighResolutionViewSubject.next(!isLowResolutionView);
            return !isLowResolutionView;
        })), collapseOnRouterNav$, this.manuallyExpanded$).pipe(distinctUntilChanged(), shareReplay({ bufferSize: 1, refCount: true }), tap((isExpanded) => {
            this.isNavExpanded = isExpanded;
        }));
    }
    /**
     * Toggling navigation visibility
     */
    toggleNav() {
        this.isNavExpanded = !this.isNavExpanded;
        this.manuallyExpanded$.next(this.isNavExpanded);
    }
}
LayoutService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LayoutService, deps: [{ token: i1.MediaQueryService }, { token: i2.Router }, { token: LAYOUT_CONFIG_TOKEN, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
LayoutService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LayoutService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LayoutService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.MediaQueryService }, { type: i2.Router }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [LAYOUT_CONFIG_TOKEN]
                }] }]; } });
