// @backbase/ui-ang/esm2015/_example/tab/showcase/tab-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TabShowcaseComponent } from './tab-showcase.component';
import { TabModule } from '@backbase/ui-ang/tab';
import { BadgeCounterModule } from '@backbase/ui-ang/badge-counter';
import * as i0 from "@angular/core";
export class TabShowcaseModule {
}
TabShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TabShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
TabShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TabShowcaseModule, declarations: [TabShowcaseComponent], imports: [CommonModule, TabModule, BadgeCounterModule] });
TabShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TabShowcaseModule, imports: [[CommonModule, TabModule, BadgeCounterModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TabShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, TabModule, BadgeCounterModule],
                    declarations: [TabShowcaseComponent],
                    entryComponents: [TabShowcaseComponent],
                }]
        }] });
