// @backbase/audit-http-ang/esm2015/api/auditClient.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const AuditClientServiceGetAuditEventCategoriesMocksProvider = createMocks([{
        urlPattern: "/client-api/v3/audit-log/event-categories",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "eventCategories": [{
                            "eventCategory": "Access Control",
                            "objectTypes": [{
                                    "objectType": "Data Group",
                                    "eventActions": ["Create", "Update", "Delete"]
                                }]
                        }, {
                            "eventCategory": "Audit",
                            "objectTypes": [{
                                    "objectType": "Audit Log",
                                    "eventActions": ["Publish", "Read", "Export"]
                                }]
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const AuditClientServiceGetAuditLogMessagesMocksProvider = createMocks([{
        urlPattern: "/client-api/v3/audit-log/messages",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "totalCount": 3,
                    "auditMessages": [{
                            "timestamp": "2020-09-24T13:28:57.896Z",
                            "eventCategory": "Access Control",
                            "objectType": "Data Group",
                            "eventAction": "Update",
                            "eventDescription": "JBloggs changed Account Group 'MyAccountGroup'. Added account: ABC123",
                            "status": "Initiated",
                            "username": "JBloggs",
                            "userId": "8ad6818b682e6e7201682f853a8f0002",
                            "messageSetId": "8ab2b95874742cc201748cf9a8a106768305",
                            "eventMetaData": {
                                "AccountGroup Name": "MyAccountGroup",
                                "Account Added": "ABC123"
                            }
                        }, {
                            "timestamp": "2020-09-24T13:29:01.549Z",
                            "eventCategory": "Access Control",
                            "objectType": "Data Group",
                            "eventAction": "Update",
                            "eventDescription": "JBloggs changed Account Group 'MyAccountGroup'. Added account: ABC123",
                            "status": "Successful",
                            "ipAddress": "86.3.123.123",
                            "username": "JBloggs",
                            "userId": "8ad6818b682e6e7201682f853a8f0002",
                            "legalEntityId": "8ad6818b682e6f3c01682f8536c60198",
                            "serviceAgreementId": "8ad6818b682e6f3c01682f85381a0199",
                            "messageSetId": "8ab2b95874742cc201748cf9a8a106768305",
                            "schemaVersion": "1.2.3",
                            "userAgent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.90 Safari/537.36",
                            "eventMetaData": {
                                "AccountGroup Name": "MyAccountGroup",
                                "Account Added": "ABC123"
                            }
                        }, {
                            "timestamp": "2020-09-24T13:29:01.549Z",
                            "eventCategory": "Access Control",
                            "objectType": "Data Group",
                            "eventAction": "Update",
                            "eventDescription": "JBloggs changed Account Group 'MyAccountGroup'. Added account: ABC123",
                            "status": "Failed",
                            "ipAddress": "86.3.123.123",
                            "username": "JBloggs",
                            "userId": "8ad6818b682e6e7201682f853a8f0002",
                            "legalEntityId": "8ad6818b682e6f3c01682f8536c60198",
                            "serviceAgreementId": "8ad6818b682e6f3c01682f85381a0199",
                            "messageSetId": "8ab2b95874742cc201748cf9a8a106768305",
                            "schemaVersion": "1.2.3",
                            "userAgent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.90 Safari/537.36",
                            "error": "Cannot update, account 'ABC123' does not exist.",
                            "eventMetaData": {
                                "AccountGroup Name": "MyAccountGroup",
                                "Account Added": "ABC123"
                            }
                        }]
                }
            },
            {
                status: 200,
                body: "\"timestamp\",\"eventCategory\",\"objectType\",\"eventAction\",\"eventDescription\",\"status\",\"username\",\"userId\",\"messageSetId\",\"eventMetaData__AccountGroup Name\",\"eventMetaData__Account Added\",\"ipAddress\",\"legalEntityId\",\"serviceAgreementId\",\"schemaVersion\",\"userAgent\",\"error\"\n\"2020-09-24T13:28:57.896Z\",\"Access Control\",\"Data Group\",\"Update\",\"JBloggs changed Account Group \'MyAccountGroup\'. Added account: ABC123\",\"Initiated\",\"JBloggs\",\"8ad6818b682e6e7201682f853a8f0002\",\"8ab2b95874742cc201748cf9a8a106768305\",\"MyAccountGroup\",\"ABC123\",\"\",\"\",\"\",\"\",\"\",\"\"\n\"2020-09-24T13:29:01.549Z\",\"Access Control\",\"Data Group\",\"Update\",\"JBloggs changed Account Group \'MyAccountGroup\'. Added account: ABC123\",\"Successful\",\"JBloggs\",\"8ad6818b682e6e7201682f853a8f0002\",\"8ab2b95874742cc201748cf9a8a106768305\",\"MyAccountGroup\",\"ABC123\",\"86.3.123.123\",\"8ad6818b682e6f3c01682f8536c60198\",\"8ad6818b682e6f3c01682f85381a0199\",\"1.2.3\",\"Mozilla\/5.0 (X11; Linux x86_64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/77.0.3865.90 Safari\/537.36\",\"\"\n\"2020-09-24T13:29:01.549Z\",\"Access Control\",\"Data Group\",\"Update\",\"JBloggs changed Account Group \'MyAccountGroup\'. Added account: ABC123\",\"Failed\",\"JBloggs\",\"8ad6818b682e6e7201682f853a8f0002\",\"8ab2b95874742cc201748cf9a8a106768305\",\"MyAccountGroup\",\"ABC123\",\"86.3.123.123\",\"8ad6818b682e6f3c01682f8536c60198\",\"8ad6818b682e6f3c01682f85381a0199\",\"1.2.3\",\"Mozilla\/5.0 (X11; Linux x86_64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/77.0.3865.90 Safari\/537.36\",\"Cannot update, account \'ABC123\' does not exist.\""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const AuditClientServiceGetLegalEntitiesMocksProvider = createMocks([{
        urlPattern: "/client-api/v3/audit-log/legal-entities",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "c7a382786d514262b75ab9531b749a2b",
                        "externalId": "Bank",
                        "name": "My Bank",
                        "type": "BANK"
                    }, {
                        "id": "0f6d14f40e064deeaee351befd53d3e7",
                        "externalId": "Backbase Cardiff",
                        "name": "Backbase Cardiff",
                        "type": "CUSTOMER"
                    }]
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const AuditClientServiceGetStartAsyncExportMocksProvider = createMocks([{
        urlPattern: "/client-api/v3/audit-log/export",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "exportId": "a892e1e0-b62c-4e2b-bd24-9e2ad9786ae1"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 429,
                body: {
                    "message": "Too many requests for Async export."
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const AuditClientServiceGetStoredContentMocksProvider = createMocks([{
        urlPattern: "/client-api/v3/audit-log/stored-content",
        method: "GET",
        responses: [
            {
                status: 200,
                body: "\"timestamp\",\"eventCategory\",\"objectType\",\"eventAction\",\"eventDescription\",\"status\",\"username\",\"userId\",\"messageSetId\",\"eventMetaData__AccountGroup Name\",\"eventMetaData__Account Added\",\"ipAddress\",\"legalEntityId\",\"serviceAgreementId\",\"schemaVersion\",\"userAgent\",\"error\"\n\"2020-09-24T13:28:57.896Z\",\"Access Control\",\"Data Group\",\"Update\",\"JBloggs changed Account Group \'MyAccountGroup\'. Added account: ABC123\",\"Initiated\",\"JBloggs\",\"8ad6818b682e6e7201682f853a8f0002\",\"8ab2b95874742cc201748cf9a8a106768305\",\"MyAccountGroup\",\"ABC123\",\"\",\"\",\"\",\"\",\"\",\"\"\n\"2020-09-24T13:29:01.549Z\",\"Access Control\",\"Data Group\",\"Update\",\"JBloggs changed Account Group \'MyAccountGroup\'. Added account: ABC123\",\"Successful\",\"JBloggs\",\"8ad6818b682e6e7201682f853a8f0002\",\"8ab2b95874742cc201748cf9a8a106768305\",\"MyAccountGroup\",\"ABC123\",\"86.3.123.123\",\"8ad6818b682e6f3c01682f8536c60198\",\"8ad6818b682e6f3c01682f85381a0199\",\"1.2.3\",\"Mozilla\/5.0 (X11; Linux x86_64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/77.0.3865.90 Safari\/537.36\",\"\"\n\"2020-09-24T13:29:01.549Z\",\"Access Control\",\"Data Group\",\"Update\",\"JBloggs changed Account Group \'MyAccountGroup\'. Added account: ABC123\",\"Failed\",\"JBloggs\",\"8ad6818b682e6e7201682f853a8f0002\",\"8ab2b95874742cc201748cf9a8a106768305\",\"MyAccountGroup\",\"ABC123\",\"86.3.123.123\",\"8ad6818b682e6f3c01682f8536c60198\",\"8ad6818b682e6f3c01682f85381a0199\",\"1.2.3\",\"Mozilla\/5.0 (X11; Linux x86_64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/77.0.3865.90 Safari\/537.36\",\"Cannot update, account \'ABC123\' does not exist.\""
            },
        ]
    }]);
export const AuditClientServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v3/audit-log/event-categories",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "eventCategories": [{
                            "eventCategory": "Access Control",
                            "objectTypes": [{
                                    "objectType": "Data Group",
                                    "eventActions": ["Create", "Update", "Delete"]
                                }]
                        }, {
                            "eventCategory": "Audit",
                            "objectTypes": [{
                                    "objectType": "Audit Log",
                                    "eventActions": ["Publish", "Read", "Export"]
                                }]
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v3/audit-log/messages",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "totalCount": 3,
                    "auditMessages": [{
                            "timestamp": "2020-09-24T13:28:57.896Z",
                            "eventCategory": "Access Control",
                            "objectType": "Data Group",
                            "eventAction": "Update",
                            "eventDescription": "JBloggs changed Account Group 'MyAccountGroup'. Added account: ABC123",
                            "status": "Initiated",
                            "username": "JBloggs",
                            "userId": "8ad6818b682e6e7201682f853a8f0002",
                            "messageSetId": "8ab2b95874742cc201748cf9a8a106768305",
                            "eventMetaData": {
                                "AccountGroup Name": "MyAccountGroup",
                                "Account Added": "ABC123"
                            }
                        }, {
                            "timestamp": "2020-09-24T13:29:01.549Z",
                            "eventCategory": "Access Control",
                            "objectType": "Data Group",
                            "eventAction": "Update",
                            "eventDescription": "JBloggs changed Account Group 'MyAccountGroup'. Added account: ABC123",
                            "status": "Successful",
                            "ipAddress": "86.3.123.123",
                            "username": "JBloggs",
                            "userId": "8ad6818b682e6e7201682f853a8f0002",
                            "legalEntityId": "8ad6818b682e6f3c01682f8536c60198",
                            "serviceAgreementId": "8ad6818b682e6f3c01682f85381a0199",
                            "messageSetId": "8ab2b95874742cc201748cf9a8a106768305",
                            "schemaVersion": "1.2.3",
                            "userAgent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.90 Safari/537.36",
                            "eventMetaData": {
                                "AccountGroup Name": "MyAccountGroup",
                                "Account Added": "ABC123"
                            }
                        }, {
                            "timestamp": "2020-09-24T13:29:01.549Z",
                            "eventCategory": "Access Control",
                            "objectType": "Data Group",
                            "eventAction": "Update",
                            "eventDescription": "JBloggs changed Account Group 'MyAccountGroup'. Added account: ABC123",
                            "status": "Failed",
                            "ipAddress": "86.3.123.123",
                            "username": "JBloggs",
                            "userId": "8ad6818b682e6e7201682f853a8f0002",
                            "legalEntityId": "8ad6818b682e6f3c01682f8536c60198",
                            "serviceAgreementId": "8ad6818b682e6f3c01682f85381a0199",
                            "messageSetId": "8ab2b95874742cc201748cf9a8a106768305",
                            "schemaVersion": "1.2.3",
                            "userAgent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.90 Safari/537.36",
                            "error": "Cannot update, account 'ABC123' does not exist.",
                            "eventMetaData": {
                                "AccountGroup Name": "MyAccountGroup",
                                "Account Added": "ABC123"
                            }
                        }]
                }
            },
            {
                status: 200,
                body: "\"timestamp\",\"eventCategory\",\"objectType\",\"eventAction\",\"eventDescription\",\"status\",\"username\",\"userId\",\"messageSetId\",\"eventMetaData__AccountGroup Name\",\"eventMetaData__Account Added\",\"ipAddress\",\"legalEntityId\",\"serviceAgreementId\",\"schemaVersion\",\"userAgent\",\"error\"\n\"2020-09-24T13:28:57.896Z\",\"Access Control\",\"Data Group\",\"Update\",\"JBloggs changed Account Group \'MyAccountGroup\'. Added account: ABC123\",\"Initiated\",\"JBloggs\",\"8ad6818b682e6e7201682f853a8f0002\",\"8ab2b95874742cc201748cf9a8a106768305\",\"MyAccountGroup\",\"ABC123\",\"\",\"\",\"\",\"\",\"\",\"\"\n\"2020-09-24T13:29:01.549Z\",\"Access Control\",\"Data Group\",\"Update\",\"JBloggs changed Account Group \'MyAccountGroup\'. Added account: ABC123\",\"Successful\",\"JBloggs\",\"8ad6818b682e6e7201682f853a8f0002\",\"8ab2b95874742cc201748cf9a8a106768305\",\"MyAccountGroup\",\"ABC123\",\"86.3.123.123\",\"8ad6818b682e6f3c01682f8536c60198\",\"8ad6818b682e6f3c01682f85381a0199\",\"1.2.3\",\"Mozilla\/5.0 (X11; Linux x86_64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/77.0.3865.90 Safari\/537.36\",\"\"\n\"2020-09-24T13:29:01.549Z\",\"Access Control\",\"Data Group\",\"Update\",\"JBloggs changed Account Group \'MyAccountGroup\'. Added account: ABC123\",\"Failed\",\"JBloggs\",\"8ad6818b682e6e7201682f853a8f0002\",\"8ab2b95874742cc201748cf9a8a106768305\",\"MyAccountGroup\",\"ABC123\",\"86.3.123.123\",\"8ad6818b682e6f3c01682f8536c60198\",\"8ad6818b682e6f3c01682f85381a0199\",\"1.2.3\",\"Mozilla\/5.0 (X11; Linux x86_64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/77.0.3865.90 Safari\/537.36\",\"Cannot update, account \'ABC123\' does not exist.\""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v3/audit-log/legal-entities",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "c7a382786d514262b75ab9531b749a2b",
                        "externalId": "Bank",
                        "name": "My Bank",
                        "type": "BANK"
                    }, {
                        "id": "0f6d14f40e064deeaee351befd53d3e7",
                        "externalId": "Backbase Cardiff",
                        "name": "Backbase Cardiff",
                        "type": "CUSTOMER"
                    }]
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v3/audit-log/export",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "exportId": "a892e1e0-b62c-4e2b-bd24-9e2ad9786ae1"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 429,
                body: {
                    "message": "Too many requests for Async export."
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v3/audit-log/stored-content",
        method: "GET",
        responses: [
            {
                status: 200,
                body: "\"timestamp\",\"eventCategory\",\"objectType\",\"eventAction\",\"eventDescription\",\"status\",\"username\",\"userId\",\"messageSetId\",\"eventMetaData__AccountGroup Name\",\"eventMetaData__Account Added\",\"ipAddress\",\"legalEntityId\",\"serviceAgreementId\",\"schemaVersion\",\"userAgent\",\"error\"\n\"2020-09-24T13:28:57.896Z\",\"Access Control\",\"Data Group\",\"Update\",\"JBloggs changed Account Group \'MyAccountGroup\'. Added account: ABC123\",\"Initiated\",\"JBloggs\",\"8ad6818b682e6e7201682f853a8f0002\",\"8ab2b95874742cc201748cf9a8a106768305\",\"MyAccountGroup\",\"ABC123\",\"\",\"\",\"\",\"\",\"\",\"\"\n\"2020-09-24T13:29:01.549Z\",\"Access Control\",\"Data Group\",\"Update\",\"JBloggs changed Account Group \'MyAccountGroup\'. Added account: ABC123\",\"Successful\",\"JBloggs\",\"8ad6818b682e6e7201682f853a8f0002\",\"8ab2b95874742cc201748cf9a8a106768305\",\"MyAccountGroup\",\"ABC123\",\"86.3.123.123\",\"8ad6818b682e6f3c01682f8536c60198\",\"8ad6818b682e6f3c01682f85381a0199\",\"1.2.3\",\"Mozilla\/5.0 (X11; Linux x86_64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/77.0.3865.90 Safari\/537.36\",\"\"\n\"2020-09-24T13:29:01.549Z\",\"Access Control\",\"Data Group\",\"Update\",\"JBloggs changed Account Group \'MyAccountGroup\'. Added account: ABC123\",\"Failed\",\"JBloggs\",\"8ad6818b682e6e7201682f853a8f0002\",\"8ab2b95874742cc201748cf9a8a106768305\",\"MyAccountGroup\",\"ABC123\",\"86.3.123.123\",\"8ad6818b682e6f3c01682f8536c60198\",\"8ad6818b682e6f3c01682f85381a0199\",\"1.2.3\",\"Mozilla\/5.0 (X11; Linux x86_64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/77.0.3865.90 Safari\/537.36\",\"Cannot update, account \'ABC123\' does not exist.\""
            },
        ]
    },
]);
