// @backbase/actions-common-ang/esm2015/src/models/arrangement-based-recipe.model.js
import { EventEmitter, Input, Output, Directive } from '@angular/core';
import { Subject } from 'rxjs';
import { takeUntil } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "../services/actions-form-builder.service";
// tslint:disable-next-line:directive-class-suffix
export class ArrangementBasedRecipeModel {
    constructor(actionsFormBuilderService) {
        this.actionsFormBuilderService = actionsFormBuilderService;
        /**
         * On submit event emitter.
         */
        this.submit = new EventEmitter();
        /**
         * Specifications with actions.
         */
        this.specificationActions = [];
        /**
         * Loading flag.
         */
        this.isLoading = false;
        this.destroy$ = new Subject();
    }
    /**
     * Specification setter.
     */
    set specification(specification) {
        this.specificationValue = specification;
        this.specificationActions = specification.actions || [];
    }
    /**
     * Specification getter.
     */
    get specification() {
        return this.specificationValue;
    }
    /**
     * Method to check if card form control is expanded.
     *
     * @returns boolean value
     */
    get isCardExpanded() {
        const activeControl = this.formGroup && this.formGroup.get('active');
        return activeControl && activeControl.value && !this.isLoading;
    }
    ngOnInit() {
        this.formGroup = this.actionsFormBuilderService.buildActionRecipeFormGroup(this.specification, this.actionRecipe, this.account);
        this.formGroup.valueChanges.pipe(takeUntil(this.destroy$)).subscribe(value => {
            this.onSubmit(value);
        });
    }
    ngOnChanges(changes) {
        const { actionRecipe } = changes;
        if (this.formGroup && actionRecipe && actionRecipe.previousValue !== actionRecipe.currentValue) {
            this.formGroup.patchValue(this.actionsFormBuilderService.buildActionRecipeFormValue(this.specification, this.actionRecipe, this.account), {
                emitEvent: false,
            });
            this.isLoading = false;
        }
    }
    ngOnDestroy() {
        this.destroy$.next();
        this.destroy$.complete();
    }
    /**
     * Method to submit action recipe form.
     *
     * @param formValue - action recipe form value
     */
    onSubmit(formValue) {
        this.submit.emit(formValue);
    }
    /**
     * Method to set loading flag to true.
     */
    onActiveLoadingChange() {
        this.isLoading = true;
    }
}
ArrangementBasedRecipeModel.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ArrangementBasedRecipeModel, deps: [{ token: i1.ActionsFormBuilderService }], target: i0.ɵɵFactoryTarget.Directive });
ArrangementBasedRecipeModel.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.14", type: ArrangementBasedRecipeModel, inputs: { account: "account", actionRecipe: "actionRecipe", specification: "specification" }, outputs: { submit: "submit" }, usesOnChanges: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ArrangementBasedRecipeModel, decorators: [{
            type: Directive
        }], ctorParameters: function () { return [{ type: i1.ActionsFormBuilderService }]; }, propDecorators: { account: [{
                type: Input
            }], actionRecipe: [{
                type: Input
            }], specification: [{
                type: Input
            }], submit: [{
                type: Output
            }] } });
