// @backbase/account-statement-retail-journey-ang/esm2015/src/account-statement-retail-journey.module.js
import { NgModule } from '@angular/core';
import { provideRoutes, RouterModule } from '@angular/router';
import { accountStatementRetailJourney } from './account-statement-retail-journey.routes';
import { AccountStatementRetailJourneyComponent } from './account-statement-retail-journey.component';
import { CommonModule } from '@angular/common';
import { EntitlementsModule } from '@backbase/foundation-ang/entitlements';
import { AccountStatementListWidgetModule } from '@backbase/account-statement-list-widget-ang';
import { AccountStatementRetailViewComponent } from './view/account-statement-retail-view.component';
import { HeaderModule } from '@backbase/ui-ang/header';
import * as i0 from "@angular/core";
const components = [AccountStatementRetailJourneyComponent, AccountStatementRetailViewComponent];
const imports = [HeaderModule, AccountStatementListWidgetModule];
export class AccountStatementRetailJourneyModule {
    /**
     * Configure routes provided by application developer to the journey.
     *
     * @public
     *
     * @example
     *
     * ```
     * @NgModule({
     *  imports: [AccountStatementRetailJourneyAngModule.forRoot({route: JOURNEY_ROUTE})]
     * })
     * class MyNgModule {}
     * ```
     *
     * @param data Object containing routes for the journey.
     * @returns The new `NgModule`.
     */
    static forRoot(data = { routes: accountStatementRetailJourney }) {
        return {
            ngModule: AccountStatementRetailJourneyModule,
            providers: [provideRoutes([data.routes])],
        };
    }
}
AccountStatementRetailJourneyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementRetailJourneyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AccountStatementRetailJourneyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementRetailJourneyModule, declarations: [AccountStatementRetailJourneyComponent, AccountStatementRetailViewComponent], imports: [CommonModule, RouterModule, EntitlementsModule, AccountStatementListWidgetModule, HeaderModule, AccountStatementListWidgetModule], exports: [AccountStatementRetailJourneyComponent, AccountStatementRetailViewComponent] });
AccountStatementRetailJourneyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementRetailJourneyModule, imports: [[CommonModule, RouterModule, EntitlementsModule, AccountStatementListWidgetModule, ...imports]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementRetailJourneyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components],
                    imports: [CommonModule, RouterModule, EntitlementsModule, AccountStatementListWidgetModule, ...imports],
                    exports: [...components],
                }]
        }] });
