// @backbase/account-statement-retail-journey-ang/esm2015/src/account-statement-retail-journey-configuration.service.js
import { Inject, Injectable, InjectionToken, Optional } from '@angular/core';
import '@angular/localize/init';
import { JourneyConfigurationService } from '@backbase/account-statement-common-ang';
import { of } from 'rxjs';
import { RETAIL_JOURNEY_DEFAULT_CONFIGURATION } from './constants/retail-journey-default-configuration';
import * as i0 from "@angular/core";
/**
 * Injection token to override the journey configurations.
 * Application developer can override the journey configuration by providing their value for this token.
 *
 * @public
 *
 * @usageNotes
 *
 * ### Usage
 *
 * ```typescript
 * import { ACCOUNT_STATEMENT_RETAIL_JOURNEY_CONFIGURATION_TOKEN } from '@backbase/account-statement-retail-journey-ang';
 *
 * @NgModule({
 *  providers: [{ provide: ACCOUNT_STATEMENT_RETAIL_JOURNEY_CONFIGURATION_TOKEN, useValue: { headingType: 'h2'} }]
 * })
 * export class AppModule {}
 *
 * ```
 */
export const ACCOUNT_STATEMENT_RETAIL_JOURNEY_CONFIGURATION_TOKEN = new InjectionToken('AccountStatementRetailJourneyConfiguration injection token');
/**
 * Configuration service for the Account Statement Retail Journey Configuration Service.
 *
 * @public
 *
 * @remarks
 * This will allow the application developer to override the configurations specific to this journey via injection token @see {@link @backbase/account-statement-retail-journey-ang#ACCOUNT_STATEMENT_RETAIL_JOURNEY_CONFIGURATION_TOKEN}
 */
export class AccountStatementRetailJourneyConfigurationService extends JourneyConfigurationService {
    constructor(staticConfigOverrides) {
        super(Object.assign(Object.assign({}, RETAIL_JOURNEY_DEFAULT_CONFIGURATION), staticConfigOverrides));
    }
    /**
     * Show description
     *
     * @defaultValue true
     * @returns An observable of boolean
     */
    get showDescription() {
        return of(this.staticConfig.showDescription);
    }
    /** Combined observable with journey config properties */
    get allConfig() {
        return of(Object.assign({}, this.staticConfig));
    }
}
AccountStatementRetailJourneyConfigurationService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementRetailJourneyConfigurationService, deps: [{ token: ACCOUNT_STATEMENT_RETAIL_JOURNEY_CONFIGURATION_TOKEN, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
AccountStatementRetailJourneyConfigurationService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementRetailJourneyConfigurationService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementRetailJourneyConfigurationService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [ACCOUNT_STATEMENT_RETAIL_JOURNEY_CONFIGURATION_TOKEN]
                }] }]; } });
