// @backbase/messages-common/esm2015/src/components/messages-request-error/messages-request-error.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { MessagesRequestErrorComponent } from './messages-request-error.component';
import * as i0 from "@angular/core";
const uiModules = [EmptyStateModule];
export class MessagesRequestErrorModule {
}
MessagesRequestErrorModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesRequestErrorModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MessagesRequestErrorModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesRequestErrorModule, declarations: [MessagesRequestErrorComponent], imports: [CommonModule, EmptyStateModule], exports: [MessagesRequestErrorComponent, EmptyStateModule] });
MessagesRequestErrorModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesRequestErrorModule, imports: [[CommonModule, ...uiModules], EmptyStateModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesRequestErrorModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [MessagesRequestErrorComponent],
                    exports: [MessagesRequestErrorComponent, ...uiModules],
                    imports: [CommonModule, ...uiModules],
                }]
        }] });
