// @backbase/ui-ang/esm2015/_example/modal/showcase/modal-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ModalShowcaseComponent } from './modal-showcase.component';
import { ModalModule } from '@backbase/ui-ang/modal';
import { IconModule } from '@backbase/ui-ang/icon';
import { ButtonModule } from '@backbase/ui-ang/button';
import * as i0 from "@angular/core";
export class ModalShowcaseModule {
}
ModalShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ModalShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ModalShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ModalShowcaseModule, declarations: [ModalShowcaseComponent], imports: [CommonModule, ModalModule, IconModule, ButtonModule] });
ModalShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ModalShowcaseModule, imports: [[CommonModule, ModalModule, IconModule, ButtonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ModalShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ModalModule, IconModule, ButtonModule],
                    declarations: [ModalShowcaseComponent],
                    entryComponents: [ModalShowcaseComponent],
                }]
        }] });
