// @backbase/ui-ang/esm2015/_example/textarea/showcase/textarea-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TextareaShowcaseComponent } from './textarea-showcase.component';
import { TextareaModule } from '@backbase/ui-ang/textarea';
import { ReactiveFormsModule } from '@angular/forms';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import * as i0 from "@angular/core";
export class TextareaShowcaseModule {
}
TextareaShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TextareaShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
TextareaShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TextareaShowcaseModule, declarations: [TextareaShowcaseComponent], imports: [CommonModule, TextareaModule, ReactiveFormsModule, InputValidationMessageModule] });
TextareaShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TextareaShowcaseModule, imports: [[CommonModule, TextareaModule, ReactiveFormsModule, InputValidationMessageModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TextareaShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, TextareaModule, ReactiveFormsModule, InputValidationMessageModule],
                    declarations: [TextareaShowcaseComponent],
                    entryComponents: [TextareaShowcaseComponent],
                }]
        }] });
