// @backbase/consent-journey-ang/esm2015/src/views/consents/consent-list/consent-list.component.js
import { Component } from '@angular/core';
import { Subject } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "../../../services/consent-journey-config.service";
import * as i2 from "@angular/router";
import * as i3 from "@backbase/consent-list-widget-ang";
export class ConsentListComponent {
    constructor(config, router, activeRoute) {
        this.config = config;
        this.router = router;
        this.activeRoute = activeRoute;
        this.destroy$ = new Subject();
        this.consentId = '';
    }
    handleDetail(event) {
        this.consentId = event;
        this.router.navigate(['../details', { consentId: event }], {
            relativeTo: this.activeRoute,
        });
    }
    ngOnDestroy() {
        this.destroy$.next();
    }
}
ConsentListComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentListComponent, deps: [{ token: i1.ConsentJourneyConfigurationService }, { token: i2.Router }, { token: i2.ActivatedRoute }], target: i0.ɵɵFactoryTarget.Component });
ConsentListComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: ConsentListComponent, selector: "bb-consent-list-view", ngImport: i0, template: ` <div class="bb-card">
    <div class="bb-card__body">
      <bb-consent-list-widget (consentId)="handleDetail($event)"></bb-consent-list-widget>
    </div>
  </div>`, isInline: true, components: [{ type: i3.ConsentListWidgetComponent, selector: "bb-consent-list-widget", outputs: ["consentId"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentListComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-consent-list-view',
                    template: ` <div class="bb-card">
    <div class="bb-card__body">
      <bb-consent-list-widget (consentId)="handleDetail($event)"></bb-consent-list-widget>
    </div>
  </div>`,
                }]
        }], ctorParameters: function () { return [{ type: i1.ConsentJourneyConfigurationService }, { type: i2.Router }, { type: i2.ActivatedRoute }]; } });
