// @backbase/pay-bills-journey-ang/esm2015/src/services/pay-bills-journey-config.service.js
import { Injectable, Optional, Inject, InjectionToken } from '@angular/core';
import { accountNumberFullFormat } from '@backbase/billpay-journeys-common';
import '@angular/localize/init';
import { FrequencyKeys } from '@backbase/billpay-journeys-common';
import * as i0 from "@angular/core";
export const PayBillsJourneyConfigurationToken = new InjectionToken('PayBillsJourneyConfigurationToken injection token');
const configDefaults = {
    notificationDismissTime: 5000,
    accountNumberMasked: false,
    paymentDetailsTitle: $localize `:@@pay-bills-journey-form-payment-details-title:Payment details`,
    paymentDefaultCurrency: 'USD',
    multipleBillsMode: false,
    productNumberFormat: accountNumberFullFormat,
    frequencies: [
        { name: $localize `:@@billpay-common.frequency-label.once:Once`, value: FrequencyKeys.Once },
        { name: $localize `:@@billpay-common.frequency-label.weekly:Weekly`, value: FrequencyKeys.Weekly },
        {
            name: $localize `:@@billpay-common.frequency-label.every-two-week:Every two weeks`,
            value: FrequencyKeys.Biweekly,
        },
        { name: $localize `:@@billpay-common.frequency-label.monthly:Monthly`, value: FrequencyKeys.Monthly },
    ],
};
export class PayBillsJourneyConfigService {
    constructor(config) {
        config = config || {};
        this._config = Object.assign(Object.assign({}, configDefaults), config);
    }
    get defaults() {
        return configDefaults;
    }
    get notificationDismissTime() {
        return this._config.notificationDismissTime;
    }
    get accountNumberMasked() {
        return this._config.accountNumberMasked;
    }
    get paymentDetailsTitle() {
        return this._config.paymentDetailsTitle;
    }
    get paymentDefaultCurrency() {
        return this._config.paymentDefaultCurrency;
    }
    get multipleBillsMode() {
        return this._config.multipleBillsMode;
    }
    get productNumberFormat() {
        return this._config.productNumberFormat;
    }
    get frequencies() {
        return this._config.frequencies;
    }
}
PayBillsJourneyConfigService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PayBillsJourneyConfigService, deps: [{ token: PayBillsJourneyConfigurationToken, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
PayBillsJourneyConfigService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PayBillsJourneyConfigService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PayBillsJourneyConfigService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [PayBillsJourneyConfigurationToken]
                }] }]; } });
