// @backbase/pay-bills-journey-ang/esm2015/src/services/pay-bills-state.service.js
import { Injectable } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import * as i0 from "@angular/core";
export class PayBillsStateService {
    constructor() {
        this.paymentStateSource = new BehaviorSubject(undefined);
        this.paymentState = this.paymentStateSource.asObservable();
        this.isFormProcessingValue = false;
    }
    set isFormProcessing(value) {
        this.isFormProcessingValue = value;
    }
    get isFormProcessing() {
        return this.isFormProcessingValue;
    }
    updatePaymentState(state) {
        this.paymentStateSource.next(state);
    }
    reset() {
        this.isFormProcessing = false;
        this.updatePaymentState();
    }
}
PayBillsStateService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PayBillsStateService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
PayBillsStateService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PayBillsStateService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PayBillsStateService, decorators: [{
            type: Injectable
        }] });
