// @backbase/loans-collection-ui-ang/esm2015/lib/loan-summary/loan-summary-edit-alias/loan-summary-edit-alias-confirm-modal/loan-summary-edit-alias-confirm-modal.component.js
import { Component, ChangeDetectionStrategy, Input, Output, EventEmitter } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/modal";
import * as i2 from "@angular/common";
export class LoanSummaryEditAliasConfirmModalComponent {
    constructor() {
        /**
         * Emits on closing / dismissing the modal
         */
        this.closeModal = new EventEmitter();
        /**
         * Emits on modal confirm
         */
        this.confirmModal = new EventEmitter();
        /**
         * Modal options
         */
        this.modalOptions = {
            backdrop: 'static',
            keyboard: true,
            centered: true,
        };
    }
}
/** @nocollapse */ LoanSummaryEditAliasConfirmModalComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryEditAliasConfirmModalComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ LoanSummaryEditAliasConfirmModalComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: LoanSummaryEditAliasConfirmModalComponent, selector: "bb-loan-summary-edit-alias-confirm-modal", inputs: { isOpen: "isOpen" }, outputs: { closeModal: "closeModal", confirmModal: "confirmModal" }, ngImport: i0, template: "<bb-modal-ui\n  *ngIf=\"isOpen\"\n  [modalOptions]=\"modalOptions\"\n  [isOpen]=\"true\"\n  (cancel)=\"closeModal.emit()\"\n  (confirm)=\"confirmModal.emit()\"\n  data-role=\"bb-loan-summary-edit-alias-confirm-modal\"\n>\n  <bb-modal-header-ui\n    title=\"This alias will be visible to users who have access to this account\"\n    i18n-title=\"Loan summary edit alias confirm modal title@@loans.loan-summary-edit-alias.confirm-modal.title\"\n  >\n  </bb-modal-header-ui>\n  <bb-modal-body-ui>\n    <ng-template bbCustomModalBody>\n      <ng-container i18n=\"Loan summary edit alias confirm modal body@@loans.loan-summary-edit-alias.confirm-modal.body\">\n        Are you sure to change the account alias?\n      </ng-container>\n    </ng-template>\n  </bb-modal-body-ui>\n  <bb-modal-footer-ui\n    confirmText=\"Continue\"\n    i18n-confirmText=\"\n      Loan summary edit alias confirm modal confirm button\n      text@@loans.loan-summary-edit-alias.confirm-modal.confirm-button\"\n  ></bb-modal-footer-ui>\n</bb-modal-ui>\n", components: [{ type: i1.ModalComponent, selector: "bb-modal-ui", inputs: ["isOpen", "aria-labelledby", "aria-describedby", "modalOptions"], outputs: ["confirm", "cancel", "isOpenChange"] }, { type: i1.ModalHeaderComponent, selector: "bb-modal-header-ui", inputs: ["title"], outputs: ["close"] }, { type: i1.ModalBodyComponent, selector: "bb-modal-body-ui" }, { type: i1.ModalFooterComponent, selector: "bb-modal-footer-ui", inputs: ["confirmText", "cancelText"], outputs: ["confirm", "cancel"] }], directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i1.CustomModalBodyDirective, selector: "ng-template[bbCustomModalBody]" }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryEditAliasConfirmModalComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-loan-summary-edit-alias-confirm-modal',
                    templateUrl: './loan-summary-edit-alias-confirm-modal.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { isOpen: [{
                type: Input
            }], closeModal: [{
                type: Output
            }], confirmModal: [{
                type: Output
            }] } });
