// @backbase/loans-collection-ui-ang/esm2015/lib/loan-summary/loan-summary-edit-alias/loan-summary-edit-alias-confirm-modal/loan-summary-edit-alias-confirm-modal.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ModalModule } from '@backbase/ui-ang/modal';
import { LoanSummaryEditAliasConfirmModalComponent } from './loan-summary-edit-alias-confirm-modal.component';
import * as i0 from "@angular/core";
export class LoanSummaryEditAliasConfirmModalModule {
}
/** @nocollapse */ LoanSummaryEditAliasConfirmModalModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryEditAliasConfirmModalModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanSummaryEditAliasConfirmModalModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryEditAliasConfirmModalModule, declarations: [LoanSummaryEditAliasConfirmModalComponent], imports: [CommonModule, ModalModule], exports: [LoanSummaryEditAliasConfirmModalComponent] });
/** @nocollapse */ LoanSummaryEditAliasConfirmModalModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryEditAliasConfirmModalModule, imports: [[CommonModule, ModalModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryEditAliasConfirmModalModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanSummaryEditAliasConfirmModalComponent],
                    exports: [LoanSummaryEditAliasConfirmModalComponent],
                    imports: [CommonModule, ModalModule],
                }]
        }] });
