// @backbase/foundation-ang/esm2015/core/backbase-core/routing/components/route.component.js
import { Component } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { pluck, switchMap, filter } from 'rxjs/operators';
import { ItemModelTree } from '../../models/item-model';
import { ItemRegistryService } from '../../rendering/item-registry.service';
export class RouteComponent {
    constructor(route, tree, itemRegistry) {
        this.route = route;
        this.tree = tree;
        this.itemRegistry = itemRegistry;
        this.item = this.route.data.pipe(pluck('renderChild'), switchMap(renderChild => this.tree.find(item => item.name === renderChild)), filter((a) => a !== undefined));
    }
    static isItemInRoute(itemName, route) {
        if (!route.data || !route.data.renderChild) {
            throw new Error(`@backbase/foundation Invalid route for RouteComponent. Missing data renderChild: ${route}`);
        }
        return route.data.renderChild === itemName;
    }
    /**
     * @internal
     */
    isModule(model) {
        return !!this.itemRegistry.getModuleLoader(model.ɵclassId);
    }
}
RouteComponent.decorators = [
    { type: Component, args: [{
                selector: 'bb-route',
                template: `
    <ng-container *ngIf="item | async as item">
      <router-outlet *ngIf="isModule(item.value); else renderModel"></router-outlet>
      <ng-template #renderModel><bb-chrome [item]="item"></bb-chrome></ng-template>
    </ng-container>
  `
            },] }
];
RouteComponent.ctorParameters = () => [
    { type: ActivatedRoute },
    { type: ItemModelTree },
    { type: ItemRegistryService }
];
