// @backbase/user-manage-profile-widget-ang/esm2015/src/user-manage-email-addresses/user-manage-email-addresses.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { BadgeModule } from '@backbase/ui-ang/badge';
import { ButtonModule } from '@backbase/ui-ang/button';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { HeaderModule } from '@backbase/ui-ang/header';
import { IconModule } from '@backbase/ui-ang/icon';
import { NotificationModule } from '@backbase/ui-ang/notification';
import { UserProfileFormsModule } from '../user-profile-forms/user-profile-forms.module';
import { UserEditEmailAddressComponent, UserEditEmailAddressCustomizableDirective, UserEditEmailAddressTitleCustomizableDirective, } from './user-edit-email-address/user-edit-email-address.component';
import { UserManageEmailAddressComponent, UserManageEmailAddressCustomizableDirective, } from './user-manage-email-address.component';
import { UserManageEmailAddressesComponent, UserManageEmailAddressesCustomizableDirective, UserManageEmailAddressesTitleCustomizableDirective, } from './user-manage-email-addresses.component';
import { UserNewEmailAddressComponent, UserNewEmailAddressCustomizableDirective, UserNewEmailAddressTitleCustomizableDirective, } from './user-new-email-address/user-new-email-address.component';
import * as i0 from "@angular/core";
const uiModules = [ButtonModule, EmptyStateModule, BadgeModule, IconModule, NotificationModule, HeaderModule];
const componentModules = [UserProfileFormsModule];
export class UserManageEmailAddressesModule {
}
UserManageEmailAddressesModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserManageEmailAddressesModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
UserManageEmailAddressesModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserManageEmailAddressesModule, declarations: [UserManageEmailAddressComponent,
        UserManageEmailAddressCustomizableDirective,
        UserManageEmailAddressesComponent,
        UserManageEmailAddressesCustomizableDirective,
        UserManageEmailAddressesTitleCustomizableDirective,
        UserManageEmailAddressesComponent,
        UserNewEmailAddressComponent,
        UserNewEmailAddressCustomizableDirective,
        UserNewEmailAddressTitleCustomizableDirective,
        UserEditEmailAddressComponent,
        UserEditEmailAddressCustomizableDirective,
        UserEditEmailAddressTitleCustomizableDirective], imports: [CommonModule, UserProfileFormsModule, ButtonModule, EmptyStateModule, BadgeModule, IconModule, NotificationModule, HeaderModule], exports: [UserManageEmailAddressComponent,
        UserManageEmailAddressCustomizableDirective,
        UserManageEmailAddressesComponent,
        UserManageEmailAddressesCustomizableDirective,
        UserManageEmailAddressesTitleCustomizableDirective,
        UserManageEmailAddressesComponent,
        UserNewEmailAddressComponent,
        UserNewEmailAddressCustomizableDirective,
        UserNewEmailAddressTitleCustomizableDirective,
        UserEditEmailAddressComponent,
        UserEditEmailAddressCustomizableDirective,
        UserEditEmailAddressTitleCustomizableDirective, ButtonModule, EmptyStateModule, BadgeModule, IconModule, NotificationModule, HeaderModule, UserProfileFormsModule] });
UserManageEmailAddressesModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserManageEmailAddressesModule, imports: [[CommonModule, ...componentModules, ...uiModules], ButtonModule, EmptyStateModule, BadgeModule, IconModule, NotificationModule, HeaderModule, UserProfileFormsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserManageEmailAddressesModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...componentModules, ...uiModules],
                    declarations: [
                        UserManageEmailAddressComponent,
                        UserManageEmailAddressCustomizableDirective,
                        UserManageEmailAddressesComponent,
                        UserManageEmailAddressesCustomizableDirective,
                        UserManageEmailAddressesTitleCustomizableDirective,
                        UserManageEmailAddressesComponent,
                        UserNewEmailAddressComponent,
                        UserNewEmailAddressCustomizableDirective,
                        UserNewEmailAddressTitleCustomizableDirective,
                        UserEditEmailAddressComponent,
                        UserEditEmailAddressCustomizableDirective,
                        UserEditEmailAddressTitleCustomizableDirective,
                    ],
                    exports: [
                        UserManageEmailAddressComponent,
                        UserManageEmailAddressCustomizableDirective,
                        UserManageEmailAddressesComponent,
                        UserManageEmailAddressesCustomizableDirective,
                        UserManageEmailAddressesTitleCustomizableDirective,
                        UserManageEmailAddressesComponent,
                        UserNewEmailAddressComponent,
                        UserNewEmailAddressCustomizableDirective,
                        UserNewEmailAddressTitleCustomizableDirective,
                        UserEditEmailAddressComponent,
                        UserEditEmailAddressCustomizableDirective,
                        UserEditEmailAddressTitleCustomizableDirective,
                        ...uiModules,
                        ...componentModules,
                    ],
                }]
        }] });
