// @backbase/letter-of-credit-http-ang/esm2015/api/letterOfCreditClient.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { LETTER_OF_CREDIT_BASE_PATH, COLLECTION_FORMATS } from '../variables';
import { LetterOfCreditConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class LetterOfCreditClientService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '/api/letter-of-credit';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new LetterOfCreditConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    getLettersOfCredit(requestParameters, observe = 'body', reportProgress = false, options) {
        const _filterParams = requestParameters["filterParams"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_filterParams !== undefined && _filterParams !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _filterParams, 'filterParams');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/letters-of-credit`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getLettersOfCreditUrl(requestParameters) {
        const _filterParams = requestParameters["filterParams"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_filterParams !== undefined && _filterParams !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _filterParams, 'filterParams');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v1/letters-of-credit${queryString ? `?${queryString}` : ''}`;
    }
    getLettersOfCreditActions(requestParameters, observe = 'body', reportProgress = false, options) {
        const _ids = requestParameters["ids"];
        if (_ids === null || _ids === undefined) {
            throw new Error('Required parameter ids was null or undefined when calling getLettersOfCreditActions.');
        }
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_ids) {
            queryParameters = this.addToHttpParams(queryParameters, _ids.join(COLLECTION_FORMATS['csv']), 'ids');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/letters-of-credit/actions`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getLettersOfCreditActionsUrl(requestParameters) {
        const _ids = requestParameters["ids"];
        if (_ids === null || _ids === undefined) {
            throw new Error('Required parameter ids was null or undefined when calling getLettersOfCreditActions.');
        }
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_ids) {
            queryParameters = this.addToHttpParams(queryParameters, _ids.join(COLLECTION_FORMATS['csv']), 'ids');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v1/letters-of-credit/actions${queryString ? `?${queryString}` : ''}`;
    }
    getLettersOfCreditApprovals(requestParameters, observe = 'body', reportProgress = false, options) {
        const _filterParams = requestParameters["filterParams"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_filterParams !== undefined && _filterParams !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _filterParams, 'filterParams');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/letters-of-credit/approvals`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getLettersOfCreditApprovalsUrl(requestParameters) {
        const _filterParams = requestParameters["filterParams"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_filterParams !== undefined && _filterParams !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _filterParams, 'filterParams');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v1/letters-of-credit/approvals${queryString ? `?${queryString}` : ''}`;
    }
}
LetterOfCreditClientService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: LetterOfCreditClientService, deps: [{ token: i1.HttpClient }, { token: LETTER_OF_CREDIT_BASE_PATH, optional: true }, { token: i2.LetterOfCreditConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
LetterOfCreditClientService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: LetterOfCreditClientService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: LetterOfCreditClientService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [LETTER_OF_CREDIT_BASE_PATH]
                }] }, { type: i2.LetterOfCreditConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
