// @backbase/manage-pockets-journey-ang/esm2015/src/manage-pockets-transfers/manage-pockets-transfers.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { AccountSelectorModule } from '@backbase/ui-ang/account-selector';
import { AmountInputModule } from '@backbase/ui-ang/amount-input';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { ModalModule } from '@backbase/ui-ang/modal';
import { ManagePocketsCommonModule } from '../manage-pockets-common/manage-pockets-common.module';
import { ManagePocketsTransfersDetailsComponent } from './components/manage-pockets-transfers-details/manage-pockets-transfers-details.component';
import { ManagePocketsTransfersReviewComponent } from './components/manage-pockets-transfers-review/manage-pockets-transfers-review.component';
import { ManagePocketsTransfersWrapperComponent } from './components/manage-pockets-transfers-wrapper/manage-pockets-transfers-wrapper.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const components = [
    ManagePocketsTransfersReviewComponent,
    ManagePocketsTransfersWrapperComponent,
    ManagePocketsTransfersDetailsComponent,
];
const modules = [
    AccountSelectorModule,
    AmountInputModule,
    InputValidationMessageModule,
    FormsModule,
    LoadButtonModule,
    ModalModule,
    ReactiveFormsModule,
];
export class ManagePocketsTransfersModule {
}
ManagePocketsTransfersModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsTransfersModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ManagePocketsTransfersModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsTransfersModule, declarations: [ManagePocketsTransfersReviewComponent,
        ManagePocketsTransfersWrapperComponent,
        ManagePocketsTransfersDetailsComponent], imports: [AccountSelectorModule,
        AmountInputModule,
        InputValidationMessageModule,
        FormsModule,
        LoadButtonModule,
        ModalModule,
        ReactiveFormsModule, CommonModule,
        ManagePocketsCommonModule, i1.BackbaseCoreModule], exports: [ManagePocketsTransfersReviewComponent,
        ManagePocketsTransfersWrapperComponent,
        ManagePocketsTransfersDetailsComponent, AccountSelectorModule,
        AmountInputModule,
        InputValidationMessageModule,
        FormsModule,
        LoadButtonModule,
        ModalModule,
        ReactiveFormsModule] });
ManagePocketsTransfersModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsTransfersModule, imports: [[
            ...modules,
            CommonModule,
            ManagePocketsCommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { ManagePocketsTransfersWrapperComponent },
            }),
        ], AccountSelectorModule,
        AmountInputModule,
        InputValidationMessageModule,
        FormsModule,
        LoadButtonModule,
        ModalModule,
        ReactiveFormsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsTransfersModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components],
                    imports: [
                        ...modules,
                        CommonModule,
                        ManagePocketsCommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { ManagePocketsTransfersWrapperComponent },
                        }),
                    ],
                    exports: [...components, ...modules],
                }]
        }] });
