// @backbase/ui-ang/esm2015/_example/payment-card/examples/payment-card.example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/payment-card";
export class PaymentCardExampleComponent {
    constructor() {
        this.paymentCard = {
            name: 'Ricardo Peterson',
            number: '5678',
            vendor: 'mastercard',
            expirationDate: new Date(2021, 2),
            cardName: 'Backbase Blue Card',
        };
        this.numberFormat = {
            length: 16,
            maskRange: [0, 12],
            segments: 4,
        };
    }
}
PaymentCardExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
PaymentCardExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: PaymentCardExampleComponent, selector: "bb-payment-card-example-ui", ngImport: i0, template: "<bb-payment-card-ui\n  [paymentCard]=\"paymentCard\"\n  [paymentCardNumberFormat]=\"numberFormat\"\n></bb-payment-card-ui>\n", components: [{ type: i1.PaymentCardComponent, selector: "bb-payment-card-ui", inputs: ["paymentCard", "paymentCardNumberFormat"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-payment-card-example-ui',
                    templateUrl: './payment-card.example.component.html',
                }]
        }] });
