// @backbase/payment-order-a2a-http-ang/esm2015/api/a2aClient.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { PAYMENT_ORDER_A2A_BASE_PATH } from '../variables';
import { PaymentOrderA2aConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class A2aClientHttpService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new PaymentOrderA2aConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    _delete(requestParameters, observe = 'body', reportProgress = false, options) {
        const _externalAccountId = requestParameters["externalAccountId"];
        if (_externalAccountId === null || _externalAccountId === undefined) {
            throw new Error('Required parameter externalAccountId was null or undefined when calling _delete.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.delete(`${this.configuration.basePath}/client-api/v1/external-accounts/${encodeURIComponent(String(_externalAccountId))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    _deleteUrl(requestParameters) {
        const _externalAccountId = requestParameters["externalAccountId"];
        if (_externalAccountId === null || _externalAccountId === undefined) {
            throw new Error('Required parameter externalAccountId was null or undefined when calling _delete.');
        }
        return `${this.configuration.basePath}/client-api/v1/external-accounts/${encodeURIComponent(String(_externalAccountId))}`;
    }
    create(requestParameters, observe = 'body', reportProgress = false, options) {
        const _externalAccountPostRequest = requestParameters["externalAccountPostRequest"];
        if (_externalAccountPostRequest === null || _externalAccountPostRequest === undefined) {
            throw new Error('Required parameter externalAccountPostRequest was null or undefined when calling create.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v1/external-accounts`, _externalAccountPostRequest, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    createUrl(requestParameters) {
        const _externalAccountPostRequest = requestParameters["externalAccountPostRequest"];
        if (_externalAccountPostRequest === null || _externalAccountPostRequest === undefined) {
            throw new Error('Required parameter externalAccountPostRequest was null or undefined when calling create.');
        }
        return `${this.configuration.basePath}/client-api/v1/external-accounts`;
    }
    getAll(observe = 'body', reportProgress = false, options) {
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/external-accounts`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getAllUrl() {
        return `${this.configuration.basePath}/client-api/v1/external-accounts`;
    }
    getById(requestParameters, observe = 'body', reportProgress = false, options) {
        const _externalAccountId = requestParameters["externalAccountId"];
        if (_externalAccountId === null || _externalAccountId === undefined) {
            throw new Error('Required parameter externalAccountId was null or undefined when calling getById.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/external-accounts/${encodeURIComponent(String(_externalAccountId))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getByIdUrl(requestParameters) {
        const _externalAccountId = requestParameters["externalAccountId"];
        if (_externalAccountId === null || _externalAccountId === undefined) {
            throw new Error('Required parameter externalAccountId was null or undefined when calling getById.');
        }
        return `${this.configuration.basePath}/client-api/v1/external-accounts/${encodeURIComponent(String(_externalAccountId))}`;
    }
    update(requestParameters, observe = 'body', reportProgress = false, options) {
        const _externalAccountId = requestParameters["externalAccountId"];
        if (_externalAccountId === null || _externalAccountId === undefined) {
            throw new Error('Required parameter externalAccountId was null or undefined when calling update.');
        }
        const _externalAccountPutRequest = requestParameters["externalAccountPutRequest"];
        if (_externalAccountPutRequest === null || _externalAccountPutRequest === undefined) {
            throw new Error('Required parameter externalAccountPutRequest was null or undefined when calling update.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v1/external-accounts/${encodeURIComponent(String(_externalAccountId))}`, _externalAccountPutRequest, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    updateUrl(requestParameters) {
        const _externalAccountId = requestParameters["externalAccountId"];
        if (_externalAccountId === null || _externalAccountId === undefined) {
            throw new Error('Required parameter externalAccountId was null or undefined when calling update.');
        }
        const _externalAccountPutRequest = requestParameters["externalAccountPutRequest"];
        if (_externalAccountPutRequest === null || _externalAccountPutRequest === undefined) {
            throw new Error('Required parameter externalAccountPutRequest was null or undefined when calling update.');
        }
        return `${this.configuration.basePath}/client-api/v1/external-accounts/${encodeURIComponent(String(_externalAccountId))}`;
    }
    verify(requestParameters, observe = 'body', reportProgress = false, options) {
        const _externalAccountId = requestParameters["externalAccountId"];
        if (_externalAccountId === null || _externalAccountId === undefined) {
            throw new Error('Required parameter externalAccountId was null or undefined when calling verify.');
        }
        const _externalAccountVerificationPutRequest = requestParameters["externalAccountVerificationPutRequest"];
        if (_externalAccountVerificationPutRequest === null || _externalAccountVerificationPutRequest === undefined) {
            throw new Error('Required parameter externalAccountVerificationPutRequest was null or undefined when calling verify.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v1/external-accounts/${encodeURIComponent(String(_externalAccountId))}/account-verification`, _externalAccountVerificationPutRequest, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    verifyUrl(requestParameters) {
        const _externalAccountId = requestParameters["externalAccountId"];
        if (_externalAccountId === null || _externalAccountId === undefined) {
            throw new Error('Required parameter externalAccountId was null or undefined when calling verify.');
        }
        const _externalAccountVerificationPutRequest = requestParameters["externalAccountVerificationPutRequest"];
        if (_externalAccountVerificationPutRequest === null || _externalAccountVerificationPutRequest === undefined) {
            throw new Error('Required parameter externalAccountVerificationPutRequest was null or undefined when calling verify.');
        }
        return `${this.configuration.basePath}/client-api/v1/external-accounts/${encodeURIComponent(String(_externalAccountId))}/account-verification`;
    }
}
A2aClientHttpService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: A2aClientHttpService, deps: [{ token: i1.HttpClient }, { token: PAYMENT_ORDER_A2A_BASE_PATH, optional: true }, { token: i2.PaymentOrderA2aConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
A2aClientHttpService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: A2aClientHttpService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: A2aClientHttpService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [PAYMENT_ORDER_A2A_BASE_PATH]
                }] }, { type: i2.PaymentOrderA2aConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
