// @backbase/accounts-transactions-journeys-shared-ui-transform-ang/esm2015/lib/get-decimals-count-pipe/get-decimals-count.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { GetDecimalsCountPipe } from './get-decimals-count.pipe';
import * as i0 from "@angular/core";
const pipes = [GetDecimalsCountPipe];
export class GetDecimalsCountModule {
}
/** @nocollapse */ GetDecimalsCountModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: GetDecimalsCountModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ GetDecimalsCountModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: GetDecimalsCountModule, declarations: [GetDecimalsCountPipe], imports: [CommonModule], exports: [GetDecimalsCountPipe] });
/** @nocollapse */ GetDecimalsCountModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: GetDecimalsCountModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: GetDecimalsCountModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [...pipes],
                    exports: [...pipes],
                }]
        }] });
