// @backbase/actions-common-ang/esm2015/src/actions-common.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { NoFractionDigitsPipe } from './pipes/no-fraction-digits.pipe';
import { ActionsFormBuilderService } from './services/actions-form-builder.service';
import { ActionsNotificationsSettingsDataService } from './services/actions-notifications-settings-data.service';
import { ButtonModule } from '@backbase/ui-ang/button';
import { CollapsibleModule } from '@backbase/ui-ang/collapsible';
import { HeaderModule } from '@backbase/ui-ang/header';
import { IconModule } from '@backbase/ui-ang/icon';
import { InputCheckboxModule } from '@backbase/ui-ang/input-checkbox';
import { InputInlineEditModule } from '@backbase/ui-ang/input-inline-edit';
import { ModalModule } from '@backbase/ui-ang/modal';
import { SwitchModule } from '@backbase/ui-ang/switch';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { AmountModule } from '@backbase/ui-ang/amount';
import { ActionsNotificationChannelsComponent, ActionsNotificationChannelsCustomizableDirective, } from './components/actions-notification-channels/actions-notification-channels.component';
import { ActionsAccountBalanceRecipeFormComponent, ActionsAccountBalanceRecipeCustomizableDirective, } from './components/actions-account-balance-recipe-form/actions-account-balance-recipe-form.component';
import { ActionsTransactionsRecipeFormComponent, ActionsTransactionsRecipeCustomizableDirective, } from './components/actions-transactions-recipe-form/actions-transactions-recipe-form.component';
import { ActionsToggleRecipeFormComponent, ActionsToggleRecipeFormCustomizableDirective, } from './components/actions-toggle-recipe-form/actions-toggle-recipe-form.component';
import { SwitchWithLoadingComponent } from './components/switch-with-loading/switch-with-loading.component';
import { ActionsScheduleNotificationsComponent, ActionsScheduleNotificationsComponentCustomizableDirective, } from './components/actions-schedule-notifications/actions-schedule-notifications.component';
import { TimePickerModule } from './components/time-picker/time-picker.module';
import { ActionsNotificationsPreferencesDataService } from './services/actions-notifications-preferences-data.service';
import * as i0 from "@angular/core";
const components = [
    ActionsNotificationChannelsComponent,
    ActionsAccountBalanceRecipeFormComponent,
    ActionsTransactionsRecipeFormComponent,
    ActionsToggleRecipeFormComponent,
    SwitchWithLoadingComponent,
    ActionsScheduleNotificationsComponent,
];
const directives = [
    ActionsNotificationChannelsCustomizableDirective,
    ActionsAccountBalanceRecipeCustomizableDirective,
    ActionsTransactionsRecipeCustomizableDirective,
    ActionsToggleRecipeFormCustomizableDirective,
    ActionsScheduleNotificationsComponentCustomizableDirective,
];
const pipes = [NoFractionDigitsPipe];
const uiModules = [
    AmountModule,
    ModalModule,
    ButtonModule,
    InputCheckboxModule,
    SwitchModule,
    HeaderModule,
    CollapsibleModule,
    IconModule,
    TooltipModule,
    InputInlineEditModule,
];
export class ActionsCommonModule {
}
ActionsCommonModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsCommonModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ActionsCommonModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsCommonModule, declarations: [ActionsNotificationChannelsComponent,
        ActionsAccountBalanceRecipeFormComponent,
        ActionsTransactionsRecipeFormComponent,
        ActionsToggleRecipeFormComponent,
        SwitchWithLoadingComponent,
        ActionsScheduleNotificationsComponent, ActionsNotificationChannelsCustomizableDirective,
        ActionsAccountBalanceRecipeCustomizableDirective,
        ActionsTransactionsRecipeCustomizableDirective,
        ActionsToggleRecipeFormCustomizableDirective,
        ActionsScheduleNotificationsComponentCustomizableDirective, NoFractionDigitsPipe], imports: [CommonModule, FormsModule, ReactiveFormsModule, TimePickerModule, AmountModule,
        ModalModule,
        ButtonModule,
        InputCheckboxModule,
        SwitchModule,
        HeaderModule,
        CollapsibleModule,
        IconModule,
        TooltipModule,
        InputInlineEditModule], exports: [ActionsNotificationChannelsComponent,
        ActionsAccountBalanceRecipeFormComponent,
        ActionsTransactionsRecipeFormComponent,
        ActionsToggleRecipeFormComponent,
        SwitchWithLoadingComponent,
        ActionsScheduleNotificationsComponent, AmountModule,
        ModalModule,
        ButtonModule,
        InputCheckboxModule,
        SwitchModule,
        HeaderModule,
        CollapsibleModule,
        IconModule,
        TooltipModule,
        InputInlineEditModule, ActionsNotificationChannelsCustomizableDirective,
        ActionsAccountBalanceRecipeCustomizableDirective,
        ActionsTransactionsRecipeCustomizableDirective,
        ActionsToggleRecipeFormCustomizableDirective,
        ActionsScheduleNotificationsComponentCustomizableDirective, NoFractionDigitsPipe, TimePickerModule] });
ActionsCommonModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsCommonModule, providers: [
        ActionsFormBuilderService,
        ActionsNotificationsSettingsDataService,
        ActionsNotificationsPreferencesDataService,
    ], imports: [[CommonModule, FormsModule, ReactiveFormsModule, TimePickerModule, ...uiModules], AmountModule,
        ModalModule,
        ButtonModule,
        InputCheckboxModule,
        SwitchModule,
        HeaderModule,
        CollapsibleModule,
        IconModule,
        TooltipModule,
        InputInlineEditModule, TimePickerModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsCommonModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, FormsModule, ReactiveFormsModule, TimePickerModule, ...uiModules],
                    providers: [
                        ActionsFormBuilderService,
                        ActionsNotificationsSettingsDataService,
                        ActionsNotificationsPreferencesDataService,
                    ],
                    declarations: [...components, ...directives, ...pipes],
                    exports: [...components, ...uiModules, ...directives, ...pipes, TimePickerModule],
                }]
        }] });
