// @backbase/foundation-ang/esm2015/core/backbase-core/communication/item-navigation.service.js
import { __awaiter } from "tslib";
import { Injectable } from '@angular/core';
import { PRIMARY_OUTLET } from '@angular/router';
import { createRouteComponentStrategy } from '../routing/strategies/routing-decorators';
import { RouterConfigService } from '../routing/config/router-config.service';
import { take } from 'rxjs/operators';
import { treeBranch } from '../models/tree';
import { isRoutable } from '../routing/config/routing-tree';
import { mapLinkedList, linkedListFindLast, linkedListLast } from '../models/linked-list';
import { RouterNavigationService, } from '../routing/navigation/router-navigation.service';
/**
 * Extracts a tree branch from the component route tree up to the given item and
 * unwraps the value.
 *
 * It unwraps it so we don't end up with a `LinkedList<Tree>`, where each node
 * would have value`.value`.
 *
 * @param itemRouteTree
 * @param itemName
 */
const itemRoutesBranch = (findRoute, itemRouteTree) => {
    const branch = treeBranch(itemRouteTree, findRoute);
    if (branch === undefined) {
        return undefined;
    }
    return mapLinkedList(branch, tree => tree.value);
};
const ɵ0 = itemRoutesBranch;
/**
 * Maps a LinkedList<Route> to a NavigationRoute, to ensure each route is a
 * part of a URL segment.
 */
const mapToNavigationRoute = (routeBranch) => mapLinkedList(routeBranch, route => {
    if (route.path === undefined) {
        throw new Error(`Can't find route to item as route has no path`);
    }
    return {
        path: route.path,
        outlet: route.outlet || PRIMARY_OUTLET,
    };
});
const ɵ1 = mapToNavigationRoute;
/**
 * @internal
 */
export class ӨItemNavigationService {
    constructor(routerConfig, navigationService) {
        this.routerConfig = routerConfig;
        this.navigationService = navigationService;
    }
    navigateToItem(itemName, params) {
        return __awaiter(this, void 0, void 0, function* () {
            const isItemByName = (tree) => tree.value.item === itemName;
            return this.navigateToRoute(isItemByName, params);
        });
    }
    isItemChild(navigateToParent, childIndex) {
        return (tree, index, parent) => {
            if (parent) {
                const isChild = parent.value.item === navigateToParent;
                const isNthChild = index === childIndex;
                if (isChild && isNthChild) {
                    return true;
                }
                const isLastChild = index === parent.children.length - 1;
                const isIndexOutOfRange = childIndex > parent.children.length - 1;
                if (isChild && isLastChild && isIndexOutOfRange) {
                    return true;
                }
            }
            const noChildren = tree.children.length === 0;
            const isParent = tree.value.item === navigateToParent;
            if (isParent && noChildren) {
                return true;
            }
            return false;
        };
    }
    navigateToChild(navigateToParent, childIndex, params) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.navigateToRoute(this.isItemChild(navigateToParent, childIndex), params);
        });
    }
    navigateToRoute(findRoute, params) {
        return __awaiter(this, void 0, void 0, function* () {
            const route = yield this.routeToItem(findRoute);
            if (!route) {
                // If there is no route, then this item should already be visible, as it means
                // it's somewhere in the component tree above any routable item.
                if (Object.keys(params).length) {
                    // If trying to pass params, this can't work, so throw
                    return Promise.reject(`Can't pass params to an item that isn't within a routable item`);
                }
                // Otherwise, there is no need to navigate, navigation has failed
                return Promise.resolve(false);
            }
            const navigationRoute = mapToNavigationRoute(route);
            // Set the params on the last path in the route.
            linkedListLast(navigationRoute).value.params = params;
            return this.navigationService.navigate(this.navigationService.createUrlTree(navigationRoute));
        });
    }
    /**
     * Search the component tree for a given item, and return the route to that item.
     *
     * @param itemName The CXP instance name of the item to route to.
     */
    routeToItem(findRoute) {
        // Take a snapshot of the current state of the route tree.
        return this.routerConfig.ɵrouteTree
            .pipe(take(1))
            .toPromise()
            .then((routeTree) => this.findItemRoute(findRoute, routeTree));
    }
    /**
     * Breadth-first-search of routes, to find a route to a given item.
     *
     * 1. Find the branch to this item in the component route tree.
     * 2. Once found, find the closest routable parent.
     * 3. Take the child of the routable parent.
     * 4. For each route in the parent, use the "RouteComponentStrategy" to determine
     *      which route the route-child is rendered in.
     * 5. Once the route is found, flatten the routes of the componentRouteTree up to that route.
     *
     * @param itemName
     * @param routes
     */
    findItemRoute(findRoute, componentRouteTree) {
        // 1. Find the branch containing the item.
        const branch = itemRoutesBranch(findRoute, componentRouteTree);
        if (!branch) {
            // item not in tree
            throw new Error(`No matching route found in model.`);
        }
        const foundNode = linkedListLast(branch);
        // 2. Find the closest *routable* parent (depth-first search, so it searches from TAIL to HEAD)
        const isParent = (node) => node.item !== foundNode.value.item;
        const isRoutableParent = (node) => isRoutable(node) && isParent(node);
        const routableNode = linkedListFindLast(branch, isRoutableParent);
        if (!routableNode) {
            return undefined;
        }
        // 3. Take the child of the routable node.
        const routableNodeChild = routableNode.next;
        // 4. Determine which route the child is rendered in.
        const foundItemInRoute = routableNodeChild &&
            routableNode.value.routes.find(route => this.isItemInRoute(routableNodeChild.value.item, route));
        // 5. Find the branch in the route tree up to that route (similar to step 1).
        // Except that "Routes" is really like an array of trees, so we loop through them.
        for (const rootRoute of componentRouteTree.value.routes) {
            const routeBranch = treeBranch(rootRoute, route => route === foundItemInRoute);
            if (routeBranch) {
                return routeBranch;
            }
        }
        return undefined;
    }
    isItemInRoute(itemName, route) {
        if (!route.component) {
            return false;
        }
        return createRouteComponentStrategy(route.component).isItemInRoute(itemName, route);
    }
}
ӨItemNavigationService.decorators = [
    { type: Injectable }
];
ӨItemNavigationService.ctorParameters = () => [
    { type: RouterConfigService },
    { type: RouterNavigationService }
];
export { ɵ0, ɵ1 };
