// @backbase/account-statement-store-ang/esm2015/src/account-statement-store-model.service.js
import { Injectable } from '@angular/core';
import { accountStatements, accountStatementsError, accountStatementsFilterParams, accountStatementsLoading, accountStatementsPaginationParams, accountStatementsSortingParams, accountStatementsTotalCount, accountStatementsWidgetPreferences, } from './account-statements/selectors';
import { selectCategories } from './categories/selectors';
import { loadAccountStatementsBusiness, updateWidgetPreferences } from './account-statements/actions';
import { loadCategories } from './categories/actions';
import * as i0 from "@angular/core";
import * as i1 from "@ngrx/store";
export class AccountStatementStoreModel {
    constructor(store) {
        this.store = store;
        this.accountStatements$ = this.store.select(accountStatements);
        this.accountStatementsLoading$ = this.store.select(accountStatementsLoading);
        this.accountStatementsError$ = this.store.select(accountStatementsError);
        this.accountStatementsTotalCount$ = this.store.select(accountStatementsTotalCount);
        this.accountStatementsPaginationParams$ = this.store.select(accountStatementsPaginationParams);
        this.accountStatementsSortingParams$ = this.store.select(accountStatementsSortingParams);
        this.accountStatementsFilterParams$ = this.store.select(accountStatementsFilterParams);
        this.accountStatementsWidgetPreferences$ = this.store.select(accountStatementsWidgetPreferences);
        this.categories$ = this.store.select(selectCategories);
    }
    dispatchLoadAccountStatementsBusiness(payload = {}) {
        this.store.dispatch(loadAccountStatementsBusiness(payload));
    }
    dispatchUpdateWidgetPreferences(payload) {
        this.store.dispatch(updateWidgetPreferences(payload));
    }
    dispatchLoadCategories() {
        this.store.dispatch(loadCategories());
    }
}
AccountStatementStoreModel.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementStoreModel, deps: [{ token: i1.Store }], target: i0.ɵɵFactoryTarget.Injectable });
AccountStatementStoreModel.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementStoreModel });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementStoreModel, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.Store }]; } });
