// @backbase/product-summary-common-ang/esm2015/src/services/user/product-summary-user-permissions.service.js
import { Injectable } from '@angular/core';
import { map, shareReplay } from 'rxjs/operators';
import { AccountNavigationType, ProductSummaryPrivilege } from '../../model/types';
import { DefaultAllowedNavigationTypes } from '../../model/constants';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/accesscontrol";
export class ProductSummaryUserPermissionsService {
    constructor(usersDataService) {
        this.usersDataService = usersDataService;
        this.productSummaryPermissionParameters = {
            resourceName: 'Product Summary',
            functionName: 'Product Summary',
        };
        this.manageArrangementsPermissionParameters = {
            resourceName: 'Product Summary',
            functionName: 'Manage Arrangement Alias',
        };
        this.productSummaryPrivileges = this.usersDataService.getUserPrivileges(this.productSummaryPermissionParameters).pipe(map(privileges => privileges.map(({ privilege }) => privilege)), shareReplay());
        this.managerArrangementsPrivileges = this.usersDataService.getUserPrivileges(this.manageArrangementsPermissionParameters).pipe(map(privileges => privileges.map(({ privilege }) => privilege)), shareReplay());
        this.hasFullViewPermission = this.productSummaryPrivileges.pipe(map(privileges => privileges.includes(ProductSummaryPrivilege.VIEW)), shareReplay());
        this.hasFullEditPermission = this.productSummaryPrivileges.pipe(map(privileges => privileges.includes(ProductSummaryPrivilege.EDIT)), shareReplay());
        this.hasArrangementAliasEditPermission = this.managerArrangementsPrivileges.pipe(map(privileges => privileges.includes(ProductSummaryPrivilege.EDIT)), shareReplay());
        this.allowedAccountNavigationTypes = this.hasFullViewPermission.pipe(map(hasFullViewPermission => hasFullViewPermission
            ? DefaultAllowedNavigationTypes
            : DefaultAllowedNavigationTypes.filter(type => type !== AccountNavigationType.BALANCE_HISTORY)), shareReplay());
    }
}
ProductSummaryUserPermissionsService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ProductSummaryUserPermissionsService, deps: [{ token: i1.UsersHttpService }], target: i0.ɵɵFactoryTarget.Injectable });
ProductSummaryUserPermissionsService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ProductSummaryUserPermissionsService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ProductSummaryUserPermissionsService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.UsersHttpService }]; } });
