// @backbase/ui-ang/esm2015/dropdown-panel/dropdown-panel.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NgbDropdownModule } from '@ng-bootstrap/ng-bootstrap';
import { DropdownMenuDirective, DropdownPanelComponent, DropdownToggleDirective } from './dropdown-panel.component';
import * as i0 from "@angular/core";
export class DropdownPanelModule {
}
DropdownPanelModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownPanelModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
DropdownPanelModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownPanelModule, declarations: [DropdownPanelComponent, DropdownMenuDirective, DropdownToggleDirective], imports: [CommonModule, NgbDropdownModule], exports: [DropdownPanelComponent, DropdownToggleDirective, DropdownMenuDirective] });
DropdownPanelModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownPanelModule, imports: [[CommonModule, NgbDropdownModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownPanelModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, NgbDropdownModule],
                    declarations: [DropdownPanelComponent, DropdownMenuDirective, DropdownToggleDirective],
                    exports: [DropdownPanelComponent, DropdownToggleDirective, DropdownMenuDirective],
                }]
        }] });
