// @backbase/accessgroup-common-ang/esm2015/lib/manage-limits-modal/manage-limits-modal.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { ModalModule } from '@backbase/ui-ang/modal';
import { AmountModule } from '@backbase/ui-ang/amount';
import { IconModule } from '@backbase/ui-ang/icon';
import { ButtonModule } from '@backbase/ui-ang/button';
import { EllipsisModule } from '@backbase/ui-ang/ellipsis';
import { AlertModule } from '@backbase/ui-ang/alert';
import { KeyboardClickModule } from '@backbase/ui-ang/keyboard-click-directive';
import { HeaderModule } from '@backbase/ui-ang/header';
import { DropdownSingleSelectModule } from '@backbase/ui-ang/dropdown-single-select';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { CurrencyInputModule } from '@backbase/ui-ang/currency-input';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { LimitsService } from './services/limits.service';
import { EditLimitComponent, EditLimitCustomizableDirective } from './edit-limit/edit-limit.component';
import { CurrencyService } from './services/currency.service';
import { ManageLimitsModalHeaderComponent, ManageLimitsModalHeaderCustomizableDirective, } from './manage-limits-modal-header/manage-limits-modal-header.component';
import { LimitBoundItemComponent, LimitBoundItemCustomizableDirective, } from './manage-limits-modal-view-body/limit-bound-item/limit-bound-item.component';
import { ManageLimitsModalViewBodyComponent, ManageLimitsModalViewBodyCustomizableDirective, } from './manage-limits-modal-view-body/manage-limits-modal-view-body.component';
import { ManageLimitsModalComponent, ManageLimitsModalCustomizableDirective } from './manage-limits-modal.component';
import * as i0 from "@angular/core";
const uiModules = [
    ModalModule,
    AmountModule,
    HeaderModule,
    IconModule,
    ButtonModule,
    DropdownSingleSelectModule,
    EllipsisModule,
    AlertModule,
    KeyboardClickModule,
    TooltipModule,
    LoadingIndicatorModule,
    CurrencyInputModule,
    LoadButtonModule,
    InputValidationMessageModule,
];
const services = [CurrencyService, LimitsService];
const components = [
    ManageLimitsModalComponent,
    ManageLimitsModalCustomizableDirective,
    ManageLimitsModalViewBodyComponent,
    ManageLimitsModalViewBodyCustomizableDirective,
    LimitBoundItemComponent,
    LimitBoundItemCustomizableDirective,
    ManageLimitsModalHeaderComponent,
    ManageLimitsModalHeaderCustomizableDirective,
    EditLimitComponent,
    EditLimitCustomizableDirective,
];
export class ManageLimitsModalModule {
}
/** @nocollapse */ ManageLimitsModalModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManageLimitsModalModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ ManageLimitsModalModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManageLimitsModalModule, declarations: [ManageLimitsModalComponent,
        ManageLimitsModalCustomizableDirective,
        ManageLimitsModalViewBodyComponent,
        ManageLimitsModalViewBodyCustomizableDirective,
        LimitBoundItemComponent,
        LimitBoundItemCustomizableDirective,
        ManageLimitsModalHeaderComponent,
        ManageLimitsModalHeaderCustomizableDirective,
        EditLimitComponent,
        EditLimitCustomizableDirective], imports: [CommonModule, ReactiveFormsModule, ModalModule,
        AmountModule,
        HeaderModule,
        IconModule,
        ButtonModule,
        DropdownSingleSelectModule,
        EllipsisModule,
        AlertModule,
        KeyboardClickModule,
        TooltipModule,
        LoadingIndicatorModule,
        CurrencyInputModule,
        LoadButtonModule,
        InputValidationMessageModule], exports: [ReactiveFormsModule, ManageLimitsModalComponent,
        ManageLimitsModalCustomizableDirective,
        ManageLimitsModalViewBodyComponent,
        ManageLimitsModalViewBodyCustomizableDirective,
        LimitBoundItemComponent,
        LimitBoundItemCustomizableDirective,
        ManageLimitsModalHeaderComponent,
        ManageLimitsModalHeaderCustomizableDirective,
        EditLimitComponent,
        EditLimitCustomizableDirective, ModalModule,
        AmountModule,
        HeaderModule,
        IconModule,
        ButtonModule,
        DropdownSingleSelectModule,
        EllipsisModule,
        AlertModule,
        KeyboardClickModule,
        TooltipModule,
        LoadingIndicatorModule,
        CurrencyInputModule,
        LoadButtonModule,
        InputValidationMessageModule] });
/** @nocollapse */ ManageLimitsModalModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManageLimitsModalModule, providers: [...services], imports: [[CommonModule, ReactiveFormsModule, ...uiModules], ReactiveFormsModule, ModalModule,
        AmountModule,
        HeaderModule,
        IconModule,
        ButtonModule,
        DropdownSingleSelectModule,
        EllipsisModule,
        AlertModule,
        KeyboardClickModule,
        TooltipModule,
        LoadingIndicatorModule,
        CurrencyInputModule,
        LoadButtonModule,
        InputValidationMessageModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManageLimitsModalModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ReactiveFormsModule, ...uiModules],
                    declarations: [...components],
                    providers: [...services],
                    exports: [ReactiveFormsModule, ...components, ...uiModules],
                }]
        }] });
