// @backbase/ui-ang/esm2015/_example/input-number/showcase/input-number-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { InputNumberShowcaseComponent } from './input-number-showcase.component';
import { InputNumberModule } from '@backbase/ui-ang/input-number';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { ReactiveFormsModule } from '@angular/forms';
import * as i0 from "@angular/core";
export class InputNumberShowcaseModule {
}
InputNumberShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputNumberShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputNumberShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputNumberShowcaseModule, declarations: [InputNumberShowcaseComponent], imports: [CommonModule, InputNumberModule, InputValidationMessageModule, ReactiveFormsModule] });
InputNumberShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputNumberShowcaseModule, imports: [[CommonModule, InputNumberModule, InputValidationMessageModule, ReactiveFormsModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputNumberShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, InputNumberModule, InputValidationMessageModule, ReactiveFormsModule],
                    declarations: [InputNumberShowcaseComponent],
                    entryComponents: [InputNumberShowcaseComponent],
                }]
        }] });
