// @backbase/actions-product-notifications-settings-widget-ang/bundles/backbase-actions-product-notifications-settings-widget-ang.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core'), require('@backbase/actions-common-ang'), require('@backbase/actions-store-ang'), require('@backbase/foundation-ang/core'), require('@backbase/ui-ang/button'), require('@backbase/ui-ang/empty-state'), require('@backbase/ui-ang/iban-pipe'), require('@backbase/ui-ang/amount'), require('@backbase/ui-ang/loading-indicator'), require('@backbase/ui-ang/payment-card-number-pipe'), require('@backbase/ui-ang/product-item-basic-account'), require('@backbase/ui-ang/product-item-credit-card'), require('@backbase/ui-ang/product-item-current-account'), require('@backbase/ui-ang/product-item-debit-card'), require('@backbase/ui-ang/product-item-investment-account'), require('@backbase/ui-ang/product-item-loan'), require('@backbase/ui-ang/product-item-savings-account'), require('@backbase/ui-ang/product-item-term-deposit'), require('rxjs'), require('rxjs/operators'), require('@backbase/data-ang/arrangements')) :
    typeof define === 'function' && define.amd ? define('@backbase/actions-product-notifications-settings-widget-ang', ['exports', '@angular/common', '@angular/core', '@backbase/actions-common-ang', '@backbase/actions-store-ang', '@backbase/foundation-ang/core', '@backbase/ui-ang/button', '@backbase/ui-ang/empty-state', '@backbase/ui-ang/iban-pipe', '@backbase/ui-ang/amount', '@backbase/ui-ang/loading-indicator', '@backbase/ui-ang/payment-card-number-pipe', '@backbase/ui-ang/product-item-basic-account', '@backbase/ui-ang/product-item-credit-card', '@backbase/ui-ang/product-item-current-account', '@backbase/ui-ang/product-item-debit-card', '@backbase/ui-ang/product-item-investment-account', '@backbase/ui-ang/product-item-loan', '@backbase/ui-ang/product-item-savings-account', '@backbase/ui-ang/product-item-term-deposit', 'rxjs', 'rxjs/operators', '@backbase/data-ang/arrangements'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["actions-product-notifications-settings-widget-ang"] = {}), global.ng.common, global.ng.core, global.i6, global.i1$3, global.i1$1, global.button, global.i8, global.i2, global.amount, global.i7, global.i1, global.productItemBasicAccount, global.productItemCreditCard, global.productItemCurrentAccount, global.productItemDebitCard, global.productItemInvestmentAccount, global.productItemLoan, global.productItemSavingsAccount, global.productItemTermDeposit, global.rxjs, global.rxjs.operators, global.i1$2));
})(this, (function (exports, i9, i0, i6, i1$3, i1$1, button, i8, i2, amount, i7, i1, productItemBasicAccount, productItemCreditCard, productItemCurrentAccount, productItemDebitCard, productItemInvestmentAccount, productItemLoan, productItemSavingsAccount, productItemTermDeposit, rxjs, operators, i1$2) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i9__namespace = /*#__PURE__*/_interopNamespace(i9);
    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i6__namespace = /*#__PURE__*/_interopNamespace(i6);
    var i1__namespace$2 = /*#__PURE__*/_interopNamespace(i1$3);
    var i1__namespace$3 = /*#__PURE__*/_interopNamespace(i1$1);
    var i8__namespace = /*#__PURE__*/_interopNamespace(i8);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i7__namespace = /*#__PURE__*/_interopNamespace(i7);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i1__namespace$1 = /*#__PURE__*/_interopNamespace(i1$2);

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from) {
        for (var i = 0, il = from.length, j = to.length; i < il; i++, j++)
            to[j] = from[i];
        return to;
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    var ProductKindNames;
    (function (ProductKindNames) {
        ProductKindNames["CurrentAccount"] = "Current Account";
        ProductKindNames["SavingsAccount"] = "Savings Account";
        ProductKindNames["TermDeposit"] = "Term Deposit";
        ProductKindNames["Loan"] = "Loan";
        ProductKindNames["CreditCard"] = "Credit Card";
        ProductKindNames["DebitCard"] = "Debit Card";
        ProductKindNames["InvestmentAccount"] = "Investment Account";
    })(ProductKindNames || (ProductKindNames = {}));

    var ActionsProductItemCardComponent = /** @class */ (function () {
        function ActionsProductItemCardComponent() {
            /**
             * Product kind names enum.
             */
            this.productKindNames = ProductKindNames;
            /**
             * Configuration that determines how a payment card number should be formatted.
             */
            this.productNumberFormat = {
                length: 0,
                maskRange: [0, 12],
                segments: 4,
            };
        }
        /**
         * Method to get account number.
         *
         * @param args - accounts array
         */
        ActionsProductItemCardComponent.prototype.retrieveAccountNumber = function () {
            var args = [];
            for (var _i = 0; _i < arguments.length; _i++) {
                args[_i] = arguments[_i];
            }
            var account = args[0];
            switch (account.productKindName) {
                case ProductKindNames.TermDeposit:
                    return i6.firstNonEmptyString(account.IBAN);
                case ProductKindNames.Loan:
                case ProductKindNames.CreditCard:
                case ProductKindNames.InvestmentAccount:
                    return i6.firstNonEmptyString(account.creditCardAccountNumber, account.IBAN, account.BBAN);
                case ProductKindNames.DebitCard:
                    return i6.firstNonEmptyString(account.creditCardAccountNumber);
                default:
                    return i6.firstNonEmptyString(account.IBAN, account.BBAN);
            }
        };
        /**
         * Method to get account alias.
         *
         * @param account - account data
         * @returns account alias name
         */
        ActionsProductItemCardComponent.prototype.getAlias = function (account) {
            return i6.firstNonEmptyString(account.userPreferences && account.userPreferences.alias, account.bankAlias, account.name);
        };
        return ActionsProductItemCardComponent;
    }());
    ActionsProductItemCardComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductItemCardComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    ActionsProductItemCardComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.14", type: ActionsProductItemCardComponent, selector: "bb-actions-product-item-card", inputs: { account: "account" }, ngImport: i0__namespace, template: "<ng-container bbActionsProductItemCardCustomizableDirective [bbHostRef]=\"this\" [bbTemplateContext]=\"account\">\n</ng-container>\n\n<ng-template bbActionsProductItemCardCustomizableDirective let-hostRef let-account=\"context\">\n  <ng-container>\n    <div>\n      <div data-role=\"card-title\" class=\"bb-text-bold\" role=\"heading\" aria-level=\"2\">\n        {{ hostRef.getAlias(account) }}\n      </div>\n      <div\n        data-role=\"card-sub-title\"\n        class=\"text-small bb-text-support bb-block bb-block--xl\"\n        role=\"heading\"\n        aria-level=\"3\"\n      >\n        {{ hostRef.retrieveAccountNumber(account) | bbIban | paymentCardNumber: productNumberFormat }}\n      </div>\n    </div>\n  </ng-container>\n</ng-template>\n", directives: [{ type: i0__namespace.forwardRef(function () { return ActionsProductItemCardCustomizableDirective; }), selector: "[bbActionsProductItemCardCustomizableDirective]" }], pipes: { "paymentCardNumber": i0__namespace.forwardRef(function () { return i1__namespace.PaymentCardNumberPipe; }), "bbIban": i0__namespace.forwardRef(function () { return i2__namespace.BbIbanPipe; }) } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductItemCardComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-actions-product-item-card',
                        templateUrl: './actions-product-item-card.component.html',
                    }]
            }], propDecorators: { account: [{
                    type: i0.Input
                }] } });
    var ActionsProductItemCardCustomizableDirective = /** @class */ (function (_super) {
        __extends(ActionsProductItemCardCustomizableDirective, _super);
        function ActionsProductItemCardCustomizableDirective() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return ActionsProductItemCardCustomizableDirective;
    }(i1$1.BbTemplate));
    ActionsProductItemCardCustomizableDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductItemCardCustomizableDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    ActionsProductItemCardCustomizableDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.14", type: ActionsProductItemCardCustomizableDirective, selector: "[bbActionsProductItemCardCustomizableDirective]", usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductItemCardCustomizableDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: '[bbActionsProductItemCardCustomizableDirective]',
                    }]
            }] });

    var ActionsProductNotificationsSettingsRouterService = /** @class */ (function () {
        function ActionsProductNotificationsSettingsRouterService() {
            this.productNotificationsSettingsSavedSubject = new rxjs.Subject();
            /**
             * Stream of settings save events that trigger navigation.
             */
            this.settingsSavedNavigation$ = this.productNotificationsSettingsSavedSubject.asObservable();
            this.productNotificationsSettingsCancelledSubject = new rxjs.Subject();
            /**
             * Stream of settings cancel events that trigger navigation.
             */
            this.settingsCancelledNavigation$ = this.productNotificationsSettingsCancelledSubject.asObservable();
            this.selectedAccountIdSubject = new rxjs.ReplaySubject(1);
            /**
             * Stream with selected account IDs.
             */
            this.selectedAccountId$ = this.selectedAccountIdSubject.asObservable();
        }
        Object.defineProperty(ActionsProductNotificationsSettingsRouterService.prototype, "selectedAccountId", {
            /**
             * Selected account ID setter.
             */
            set: function (value) {
                this.selectedAccountIdSubject.next(value);
            },
            enumerable: false,
            configurable: true
        });
        /**
         * Method to emit settings saved event in a stream.
         */
        ActionsProductNotificationsSettingsRouterService.prototype.emitSettingsSavedNavigation = function () {
            this.productNotificationsSettingsSavedSubject.next();
        };
        /**
         * Method to emit settings cancelled event in a stream.
         */
        ActionsProductNotificationsSettingsRouterService.prototype.emitSettingsCancelledNavigation = function () {
            this.productNotificationsSettingsCancelledSubject.next();
        };
        return ActionsProductNotificationsSettingsRouterService;
    }());
    ActionsProductNotificationsSettingsRouterService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductNotificationsSettingsRouterService, deps: [], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    ActionsProductNotificationsSettingsRouterService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductNotificationsSettingsRouterService });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductNotificationsSettingsRouterService, decorators: [{
                type: i0.Injectable
            }] });

    var ActionsProductNotificationsSettingsDataService = /** @class */ (function () {
        function ActionsProductNotificationsSettingsDataService(accountHttpService) {
            this.accountHttpService = accountHttpService;
        }
        /**
         * Method to get account by ID.
         *
         * @param arrangementId - arrangement ID
         * @returns account data
         */
        ActionsProductNotificationsSettingsDataService.prototype.getAccountById = function (arrangementId) {
            return this.accountHttpService.getArrangementById({ arrangementId: arrangementId }).pipe(operators.map(i6.fromHttpResponse));
        };
        return ActionsProductNotificationsSettingsDataService;
    }());
    ActionsProductNotificationsSettingsDataService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductNotificationsSettingsDataService, deps: [{ token: i1__namespace$1.ArrangementsHttpService }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    ActionsProductNotificationsSettingsDataService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductNotificationsSettingsDataService });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductNotificationsSettingsDataService, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () { return [{ type: i1__namespace$1.ArrangementsHttpService }]; } });

    var DEFAULT_SPECIFICATION_IDS = '5, 6, 13, 14';
    var ActionsProductNotificationsSettingsPropertiesService = /** @class */ (function (_super) {
        __extends(ActionsProductNotificationsSettingsPropertiesService, _super);
        function ActionsProductNotificationsSettingsPropertiesService() {
            return _super.call(this) || this;
        }
        /**
         * Method to get specification IDs.
         *
         * @returns specification IDs array
         */
        ActionsProductNotificationsSettingsPropertiesService.prototype.getSpecificationIDs = function () {
            return rxjs.of(DEFAULT_SPECIFICATION_IDS).pipe(operators.map(function (data) { return data.replace(/\s/g, '').split(','); }));
        };
        return ActionsProductNotificationsSettingsPropertiesService;
    }(i6.ActionsBasePropertiesService));
    ActionsProductNotificationsSettingsPropertiesService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductNotificationsSettingsPropertiesService, deps: [], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    ActionsProductNotificationsSettingsPropertiesService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductNotificationsSettingsPropertiesService });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductNotificationsSettingsPropertiesService, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () { return []; } });

    var ActionsProductSettingsPageComponent = /** @class */ (function () {
        function ActionsProductSettingsPageComponent(store, routerService, widgetDataService, widgetPropertiesService) {
            var _this = this;
            this.store = store;
            this.routerService = routerService;
            this.widgetDataService = widgetDataService;
            this.widgetPropertiesService = widgetPropertiesService;
            this.notificationDismissTime$$ = new rxjs.BehaviorSubject(undefined);
            this.specificationIDs$$ = new rxjs.BehaviorSubject(undefined);
            /**
             * Stream of Specifications with action recipe.
             */
            this.specificationsWithActionRecipe$ = this.store.specificationsWithActionRecipe$;
            this.hasAccountLoadingError = new rxjs.BehaviorSubject(false);
            /**
             * Stream of loading errors.
             */
            this.loadingError$ = rxjs.combineLatest([
                this.store.actionRecipesError$,
                this.store.specificationsError$,
                this.hasAccountLoadingError,
            ]).pipe(operators.map(function (errors) { return errors.find(Boolean); }));
            /**
             * Specification IDs enum.
             */
            this.specificationIds = i6.SpecificationIds;
            this.accountId$ = this.routerService.selectedAccountId$;
            this.account$ = this.accountId$.pipe(operators.switchMap(function (id) { return _this.getAccount(id); }), operators.publishReplay(), operators.refCount());
            this.isLoading$ = rxjs.combineLatest([
                this.store.actionRecipesLoading$,
                this.store.specificationsLoading$,
                this.account$.pipe(operators.mapTo(false), operators.startWith(true)),
            ]).pipe(operators.map(function (loadings) { return loadings.some(Boolean); }));
        }
        Object.defineProperty(ActionsProductSettingsPageComponent.prototype, "notificationDismissTime", {
            /**
             * Notification dismiss time value setter.
             * Configurable via model property.
             */
            set: function (value) {
                this.notificationDismissTime$$.next(value);
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(ActionsProductSettingsPageComponent.prototype, "specificationIDs", {
            /**
             * Specification IDs value setter.
             * Configurable via model property.
             */
            set: function (value) {
                this.specificationIDs$$.next(value);
            },
            enumerable: false,
            configurable: true
        });
        ActionsProductSettingsPageComponent.prototype.ngOnInit = function () {
            var _this = this;
            var specificationIdsStream = i6.getPropertyValue(this.specificationIDs$$, this.widgetPropertiesService.getSpecificationIDs());
            this.accountId$.pipe(operators.first(), operators.withLatestFrom(specificationIdsStream)).subscribe(function (_a) {
                var _b = __read(_a, 2), accountId = _b[0], specificationIDs = _b[1];
                var specificationIDsSet = new Set(specificationIDs);
                _this.store.dispatchLoadSpecifications({ specificationIds: specificationIDsSet });
                _this.store.dispatchLoadActionRecipes({
                    specificationIds: specificationIDsSet,
                    arrangementId: accountId,
                });
            });
        };
        /**
         * Method to submit action recipe form.
         *
         * @param formValue - action recipe form value
         */
        ActionsProductSettingsPageComponent.prototype.onSubmit = function (formValue) {
            var _this = this;
            var notificationDismissTime$ = i6.getPropertyValue(this.notificationDismissTime$$, this.widgetPropertiesService.getNotificationDismissTimeProperty());
            rxjs.combineLatest([this.accountId$, notificationDismissTime$])
                .pipe(operators.first())
                .subscribe(function (_a) {
                var _b = __read(_a, 2), arrangementId = _b[0], ttl = _b[1];
                var errorNotification = {
                    header: Object.create(_this.errorNotificationHeaderTemplate),
                    message: Object.create(_this.errorNotificationMessageTemplate),
                    modifier: 'error',
                    ttl: ttl,
                };
                _this.store.dispatchSaveActionRecipe({
                    actionRecipe: formValue,
                    arrangementId: arrangementId,
                    warningNotification: errorNotification,
                });
            });
        };
        ActionsProductSettingsPageComponent.prototype.getAccount = function (id) {
            var _this = this;
            return this.widgetDataService.getAccountById(id).pipe(operators.catchError(function () {
                _this.hasAccountLoadingError.next(true);
                return rxjs.of({});
            }));
        };
        return ActionsProductSettingsPageComponent;
    }());
    ActionsProductSettingsPageComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductSettingsPageComponent, deps: [{ token: i1__namespace$2.ActionsStoreModel }, { token: ActionsProductNotificationsSettingsRouterService }, { token: ActionsProductNotificationsSettingsDataService }, { token: ActionsProductNotificationsSettingsPropertiesService }], target: i0__namespace.ɵɵFactoryTarget.Component });
    ActionsProductSettingsPageComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.14", type: ActionsProductSettingsPageComponent, selector: "bb-actions-product-settings-page", inputs: { notificationDismissTime: "notificationDismissTime", specificationIDs: "specificationIDs" }, viewQueries: [{ propertyName: "errorNotificationHeaderTemplate", first: true, predicate: ["errorNotificationHeader"], descendants: true }, { propertyName: "errorNotificationMessageTemplate", first: true, predicate: ["errorNotificationMessage"], descendants: true }], ngImport: i0__namespace, template: "<ng-container\n  bbActionsProductSettingsPageCustomizable\n  [bbHostRef]=\"this\"\n  [bbTemplateContext]=\"{\n    account: account$ | async,\n    loadingError: loadingError$ | async,\n    loading: isLoading$ | async,\n    specificationsWithAction: specificationsWithActionRecipe$ | async,\n    specificationIds: specificationIds\n  }\"\n></ng-container>\n\n<ng-template\n  bbActionsProductSettingsPageCustomizable\n  let-host\n  let-actions=\"context.actions\"\n  let-specifications=\"context.specifications\"\n  let-account=\"context.account\"\n  let-loadingError=\"context.loadingError\"\n  let-loading=\"context.loading\"\n  let-specificationsWithAction=\"context.specificationsWithAction\"\n  let-specificationIds=\"context.specificationIds\"\n>\n  <ng-container *ngIf=\"!loading; else loadingState\">\n    <ng-container *ngIf=\"!loadingError; else errorState\">\n      <bb-actions-product-item-card [account]=\"account\"></bb-actions-product-item-card>\n\n      <div class=\"bb-block bb-block--xl\">\n        <bb-actions-account-balance-recipe-form\n          *ngIf=\"specificationsWithAction[specificationIds.AccountBalance] as specificationWithAction\"\n          [account]=\"account\"\n          [specification]=\"specificationWithAction.specification\"\n          [actionRecipe]=\"specificationWithAction.actionRecipe\"\n          (submit)=\"host.onSubmit($event)\"\n        ></bb-actions-account-balance-recipe-form>\n      </div>\n\n      <div class=\"bb-block bb-block--xl\">\n        <bb-actions-transactions-recipe-form\n          *ngIf=\"specificationsWithAction[specificationIds.Transactions] as specificationWithAction\"\n          [specification]=\"specificationWithAction.specification\"\n          [actionRecipe]=\"specificationWithAction.actionRecipe\"\n          (submit)=\"host.onSubmit($event)\"\n        ></bb-actions-transactions-recipe-form>\n      </div>\n    </ng-container>\n  </ng-container>\n\n  <ng-template #loadingState>\n    <div class=\"card\">\n      <bb-loading-indicator-ui\n        class=\"bb-state-container\"\n        loaderSize=\"lg\"\n        text=\"Loading...\"\n        i18n-text=\"Loading... text@@actions-product-settings-page.title\"\n      ></bb-loading-indicator-ui>\n    </div>\n  </ng-template>\n\n  <ng-template #errorState>\n    <div class=\"card\">\n      <bb-empty-state-ui\n        class=\"bb-state-container\"\n        data-role=\"error-state\"\n        iconSize=\"xxl\"\n        iconModifier=\"error-outline\"\n        title=\"Unknown Error\"\n        i18n-title=\"\n          Actions product notifications settings are not loading title@@actions-product-settings-page.error-state.title\"\n        subtitle=\"Try to reload the page or contact the system administrator.\"\n        i18n-subtitle=\"\n          Actions product notifications settings are not loading\n          subtitle@@actions-product-settings-page.error-state.subtitle\"\n      >\n      </bb-empty-state-ui>\n    </div>\n  </ng-template>\n</ng-template>\n\n<ng-template #errorNotificationHeader>\n  <ng-container bbActionsProductSettingsPageErrorNotificationHeaderCustomizable [bbHostRef]=\"this\"> </ng-container>\n</ng-template>\n\n<ng-template #errorNotificationMessage>\n  <ng-container bbActionsProductSettingsPageErrorNotificationMessageCustomizable [bbHostRef]=\"this\"> </ng-container>\n</ng-template>\n\n<ng-template bbActionsProductSettingsPageErrorNotificationHeaderCustomizable>\n  <span i18n=\"Couldn't save changes@@actions-product-settings-page.error-notification.header\"\n    >Couldn't make changes</span\n  >\n</ng-template>\n\n<ng-template bbActionsProductSettingsPageErrorNotificationMessageCustomizable>\n  <div i18n=\"Please try again later@@actions-product-settings-page.error-notification.message\">\n    Please try again later\n  </div>\n</ng-template>\n", components: [{ type: i0__namespace.forwardRef(function () { return ActionsProductItemCardComponent; }), selector: "bb-actions-product-item-card", inputs: ["account"] }, { type: i0__namespace.forwardRef(function () { return i6__namespace.ӨActionsAccountBalanceRecipeFormComponent; }), selector: "bb-actions-account-balance-recipe-form" }, { type: i0__namespace.forwardRef(function () { return i6__namespace.ӨActionsTransactionsRecipeFormComponent; }), selector: "bb-actions-transactions-recipe-form" }, { type: i0__namespace.forwardRef(function () { return i7__namespace.LoadingIndicatorComponent; }), selector: "bb-loading-indicator-ui", inputs: ["text", "loaderSize", "showDelay", "hasBackground", "inline"] }, { type: i0__namespace.forwardRef(function () { return i8__namespace.EmptyStateComponent; }), selector: "bb-empty-state-ui", inputs: ["title", "subtitle", "showIcon", "iconClasses", "iconModifier", "iconSize", "iconColor"] }], directives: [{ type: i0__namespace.forwardRef(function () { return ActionsProductSettingsPageCustomizableDirective; }), selector: "[bbActionsProductSettingsPageCustomizable]" }, { type: i0__namespace.forwardRef(function () { return i9__namespace.NgIf; }), selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i0__namespace.forwardRef(function () { return ActionsProductSettingsPageErrorNotificationHeaderCustomizableDirective; }), selector: "[bbActionsProductSettingsPageErrorNotificationHeaderCustomizable]" }, { type: i0__namespace.forwardRef(function () { return ActionsProductSettingsPageErrorNotificationMessageCustomizableDirective; }), selector: "[bbActionsProductSettingsPageErrorNotificationMessageCustomizable]" }], pipes: { "async": i0__namespace.forwardRef(function () { return i9__namespace.AsyncPipe; }) } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductSettingsPageComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-actions-product-settings-page',
                        templateUrl: './actions-product-settings-page.component.html',
                    }]
            }], ctorParameters: function () { return [{ type: i1__namespace$2.ActionsStoreModel }, { type: ActionsProductNotificationsSettingsRouterService }, { type: ActionsProductNotificationsSettingsDataService }, { type: ActionsProductNotificationsSettingsPropertiesService }]; }, propDecorators: { notificationDismissTime: [{
                    type: i0.Input
                }], specificationIDs: [{
                    type: i0.Input
                }], errorNotificationHeaderTemplate: [{
                    type: i0.ViewChild,
                    args: ['errorNotificationHeader', { static: false }]
                }], errorNotificationMessageTemplate: [{
                    type: i0.ViewChild,
                    args: ['errorNotificationMessage', { static: false }]
                }] } });
    var ActionsProductSettingsPageCustomizableDirective = /** @class */ (function (_super) {
        __extends(ActionsProductSettingsPageCustomizableDirective, _super);
        function ActionsProductSettingsPageCustomizableDirective() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return ActionsProductSettingsPageCustomizableDirective;
    }(i1$1.BbTemplate));
    ActionsProductSettingsPageCustomizableDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductSettingsPageCustomizableDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    ActionsProductSettingsPageCustomizableDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.14", type: ActionsProductSettingsPageCustomizableDirective, selector: "[bbActionsProductSettingsPageCustomizable]", usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductSettingsPageCustomizableDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: '[bbActionsProductSettingsPageCustomizable]',
                    }]
            }] });
    var ActionsProductSettingsPageErrorNotificationHeaderCustomizableDirective = /** @class */ (function (_super) {
        __extends(ActionsProductSettingsPageErrorNotificationHeaderCustomizableDirective, _super);
        function ActionsProductSettingsPageErrorNotificationHeaderCustomizableDirective() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return ActionsProductSettingsPageErrorNotificationHeaderCustomizableDirective;
    }(i1$1.BbTemplate));
    ActionsProductSettingsPageErrorNotificationHeaderCustomizableDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductSettingsPageErrorNotificationHeaderCustomizableDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    ActionsProductSettingsPageErrorNotificationHeaderCustomizableDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.14", type: ActionsProductSettingsPageErrorNotificationHeaderCustomizableDirective, selector: "[bbActionsProductSettingsPageErrorNotificationHeaderCustomizable]", usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductSettingsPageErrorNotificationHeaderCustomizableDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: '[bbActionsProductSettingsPageErrorNotificationHeaderCustomizable]',
                    }]
            }] });
    var ActionsProductSettingsPageErrorNotificationMessageCustomizableDirective = /** @class */ (function (_super) {
        __extends(ActionsProductSettingsPageErrorNotificationMessageCustomizableDirective, _super);
        function ActionsProductSettingsPageErrorNotificationMessageCustomizableDirective() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return ActionsProductSettingsPageErrorNotificationMessageCustomizableDirective;
    }(i1$1.BbTemplate));
    ActionsProductSettingsPageErrorNotificationMessageCustomizableDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductSettingsPageErrorNotificationMessageCustomizableDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    ActionsProductSettingsPageErrorNotificationMessageCustomizableDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.14", type: ActionsProductSettingsPageErrorNotificationMessageCustomizableDirective, selector: "[bbActionsProductSettingsPageErrorNotificationMessageCustomizable]", usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductSettingsPageErrorNotificationMessageCustomizableDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: '[bbActionsProductSettingsPageErrorNotificationMessageCustomizable]',
                    }]
            }] });

    var ActionsProductNotificationsSettingsWidgetComponent = /** @class */ (function () {
        function ActionsProductNotificationsSettingsWidgetComponent(routerService) {
            this.routerService = routerService;
            this.specificationIDs$$ = new rxjs.BehaviorSubject(undefined);
            this.specificationIDs$ = this.specificationIDs$$.asObservable();
            this.notificationDismissTime$$ = new rxjs.BehaviorSubject(undefined);
            this.notificationDismissTime$ = this.notificationDismissTime$$
                .asObservable()
                .pipe(operators.map(function (seconds) { return (seconds ? i6.secondsToMilliseconds(seconds) : undefined); }));
            this.destroy$ = new rxjs.Subject();
        }
        Object.defineProperty(ActionsProductNotificationsSettingsWidgetComponent.prototype, "selectedAccountId", {
            /**
             * Input for selected account id.
             */
            set: function (value) {
                this.routerService.selectedAccountId = value;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(ActionsProductNotificationsSettingsWidgetComponent.prototype, "specificationIDs", {
            /**
             * Specification IDs value setter.
             * Configurable via model property.
             */
            set: function (value) {
                this.specificationIDs$$.next(value);
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(ActionsProductNotificationsSettingsWidgetComponent.prototype, "notificationDismissTime", {
            /**
             * Notification dismiss time value setter
             */
            set: function (value) {
                this.notificationDismissTime$$.next(value);
            },
            enumerable: false,
            configurable: true
        });
        ActionsProductNotificationsSettingsWidgetComponent.prototype.ngOnDestroy = function () {
            this.destroy$.next();
            this.destroy$.complete();
        };
        return ActionsProductNotificationsSettingsWidgetComponent;
    }());
    ActionsProductNotificationsSettingsWidgetComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductNotificationsSettingsWidgetComponent, deps: [{ token: ActionsProductNotificationsSettingsRouterService }], target: i0__namespace.ɵɵFactoryTarget.Component });
    ActionsProductNotificationsSettingsWidgetComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.14", type: ActionsProductNotificationsSettingsWidgetComponent, selector: "bb-actions-product-notifications-settings-widget", inputs: { selectedAccountId: "selectedAccountId", specificationIDs: "specificationIDs", notificationDismissTime: "notificationDismissTime" }, providers: [
            i1$3.ActionsStoreModel,
            ActionsProductNotificationsSettingsPropertiesService,
            ActionsProductNotificationsSettingsDataService,
            ActionsProductNotificationsSettingsRouterService,
        ], ngImport: i0__namespace, template: '<bb-actions-product-settings-page [specificationIDs]="specificationIDs$ | async" [notificationDismissTime]="notificationDismissTime$ | async"></bb-actions-product-settings-page>', isInline: true, components: [{ type: ActionsProductSettingsPageComponent, selector: "bb-actions-product-settings-page", inputs: ["notificationDismissTime", "specificationIDs"] }], pipes: { "async": i9__namespace.AsyncPipe } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductNotificationsSettingsWidgetComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-actions-product-notifications-settings-widget',
                        template: '<bb-actions-product-settings-page [specificationIDs]="specificationIDs$ | async" [notificationDismissTime]="notificationDismissTime$ | async"></bb-actions-product-settings-page>',
                        providers: [
                            i1$3.ActionsStoreModel,
                            ActionsProductNotificationsSettingsPropertiesService,
                            ActionsProductNotificationsSettingsDataService,
                            ActionsProductNotificationsSettingsRouterService,
                        ],
                    }]
            }], ctorParameters: function () { return [{ type: ActionsProductNotificationsSettingsRouterService }]; }, propDecorators: { selectedAccountId: [{
                    type: i0.Input
                }], specificationIDs: [{
                    type: i0.Input
                }], notificationDismissTime: [{
                    type: i0.Input
                }] } });

    var pages = [ActionsProductSettingsPageComponent];
    var components = __spreadArray(__spreadArray([], __read(pages)), [
        ActionsProductNotificationsSettingsWidgetComponent,
        ActionsProductSettingsPageComponent,
        ActionsProductItemCardComponent,
    ]);
    var directives = [
        ActionsProductItemCardCustomizableDirective,
        ActionsProductSettingsPageCustomizableDirective,
        ActionsProductSettingsPageErrorNotificationHeaderCustomizableDirective,
        ActionsProductSettingsPageErrorNotificationMessageCustomizableDirective,
    ];
    var uiModules = [
        i8.EmptyStateModule,
        i7.LoadingIndicatorModule,
        i2.BbIbanPipeModule,
        productItemCurrentAccount.ProductItemCurrentAccountModule,
        amount.AmountModule,
        button.ButtonModule,
        productItemSavingsAccount.ProductItemSavingsAccountModule,
        productItemTermDeposit.ProductItemTermDepositModule,
        productItemLoan.ProductItemLoanModule,
        productItemCreditCard.ProductItemCreditCardModule,
        productItemDebitCard.ProductItemDebitCardModule,
        productItemInvestmentAccount.ProductItemInvestmentAccountModule,
        productItemBasicAccount.ProductItemBasicAccountModule,
        i1.PaymentCardNumberModule,
    ];
    var ActionsProductNotificationsSettingsWidgetModule = /** @class */ (function () {
        function ActionsProductNotificationsSettingsWidgetModule() {
        }
        return ActionsProductNotificationsSettingsWidgetModule;
    }());
    ActionsProductNotificationsSettingsWidgetModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductNotificationsSettingsWidgetModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ActionsProductNotificationsSettingsWidgetModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductNotificationsSettingsWidgetModule, declarations: [ActionsProductSettingsPageComponent, ActionsProductNotificationsSettingsWidgetComponent,
            ActionsProductSettingsPageComponent,
            ActionsProductItemCardComponent, ActionsProductItemCardCustomizableDirective,
            ActionsProductSettingsPageCustomizableDirective,
            ActionsProductSettingsPageErrorNotificationHeaderCustomizableDirective,
            ActionsProductSettingsPageErrorNotificationMessageCustomizableDirective], imports: [i9.CommonModule,
            i6.ActionsCommonModule,
            i1$3.ActionsStoreModule, i1__namespace$3.BackbaseCoreModule, i8.EmptyStateModule,
            i7.LoadingIndicatorModule,
            i2.BbIbanPipeModule,
            productItemCurrentAccount.ProductItemCurrentAccountModule,
            amount.AmountModule,
            button.ButtonModule,
            productItemSavingsAccount.ProductItemSavingsAccountModule,
            productItemTermDeposit.ProductItemTermDepositModule,
            productItemLoan.ProductItemLoanModule,
            productItemCreditCard.ProductItemCreditCardModule,
            productItemDebitCard.ProductItemDebitCardModule,
            productItemInvestmentAccount.ProductItemInvestmentAccountModule,
            productItemBasicAccount.ProductItemBasicAccountModule,
            i1.PaymentCardNumberModule], exports: [ActionsProductSettingsPageComponent, ActionsProductNotificationsSettingsWidgetComponent,
            ActionsProductSettingsPageComponent,
            ActionsProductItemCardComponent, ActionsProductItemCardCustomizableDirective,
            ActionsProductSettingsPageCustomizableDirective,
            ActionsProductSettingsPageErrorNotificationHeaderCustomizableDirective,
            ActionsProductSettingsPageErrorNotificationMessageCustomizableDirective, i8.EmptyStateModule,
            i7.LoadingIndicatorModule,
            i2.BbIbanPipeModule,
            productItemCurrentAccount.ProductItemCurrentAccountModule,
            amount.AmountModule,
            button.ButtonModule,
            productItemSavingsAccount.ProductItemSavingsAccountModule,
            productItemTermDeposit.ProductItemTermDepositModule,
            productItemLoan.ProductItemLoanModule,
            productItemCreditCard.ProductItemCreditCardModule,
            productItemDebitCard.ProductItemDebitCardModule,
            productItemInvestmentAccount.ProductItemInvestmentAccountModule,
            productItemBasicAccount.ProductItemBasicAccountModule,
            i1.PaymentCardNumberModule] });
    ActionsProductNotificationsSettingsWidgetModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductNotificationsSettingsWidgetModule, imports: [__spreadArray([
                i9.CommonModule,
                i6.ActionsCommonModule,
                i1$3.ActionsStoreModule,
                i1$1.BackbaseCoreModule.withConfig({
                    classMap: { ActionsProductNotificationsSettingsWidgetComponent: ActionsProductNotificationsSettingsWidgetComponent },
                })
            ], __read(uiModules)), i8.EmptyStateModule,
            i7.LoadingIndicatorModule,
            i2.BbIbanPipeModule,
            productItemCurrentAccount.ProductItemCurrentAccountModule,
            amount.AmountModule,
            button.ButtonModule,
            productItemSavingsAccount.ProductItemSavingsAccountModule,
            productItemTermDeposit.ProductItemTermDepositModule,
            productItemLoan.ProductItemLoanModule,
            productItemCreditCard.ProductItemCreditCardModule,
            productItemDebitCard.ProductItemDebitCardModule,
            productItemInvestmentAccount.ProductItemInvestmentAccountModule,
            productItemBasicAccount.ProductItemBasicAccountModule,
            i1.PaymentCardNumberModule] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: ActionsProductNotificationsSettingsWidgetModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: __spreadArray(__spreadArray([], __read(components)), __read(directives)),
                        imports: __spreadArray([
                            i9.CommonModule,
                            i6.ActionsCommonModule,
                            i1$3.ActionsStoreModule,
                            i1$1.BackbaseCoreModule.withConfig({
                                classMap: { ActionsProductNotificationsSettingsWidgetComponent: ActionsProductNotificationsSettingsWidgetComponent },
                            })
                        ], __read(uiModules)),
                        entryComponents: __spreadArray([], __read(pages)),
                        exports: __spreadArray(__spreadArray(__spreadArray([], __read(components)), __read(directives)), __read(uiModules)),
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.ActionsProductNotificationsSettingsWidgetComponent = ActionsProductNotificationsSettingsWidgetComponent;
    exports.ActionsProductNotificationsSettingsWidgetModule = ActionsProductNotificationsSettingsWidgetModule;
    exports.ActionsProductSettingsPageComponent = ActionsProductSettingsPageComponent;
    exports.ActionsProductSettingsPageCustomizableDirective = ActionsProductSettingsPageCustomizableDirective;
    exports.ActionsProductSettingsPageErrorNotificationHeaderCustomizableDirective = ActionsProductSettingsPageErrorNotificationHeaderCustomizableDirective;
    exports.ActionsProductSettingsPageErrorNotificationMessageCustomizableDirective = ActionsProductSettingsPageErrorNotificationMessageCustomizableDirective;
    exports["ӨActionsProductItemCardComponent"] = ActionsProductItemCardComponent;
    exports["ӨActionsProductItemCardCustomizableDirective"] = ActionsProductItemCardCustomizableDirective;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
