// @backbase/places-journey-ang/esm2015/lib/services/places-journey-config.service.js
import { Injectable, Optional, Inject, InjectionToken } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
export const PlacesJourneyConfigurationToken = new InjectionToken('PlacesJourneyConfiguration injection token');
export var PlacesMeasurementSystem;
(function (PlacesMeasurementSystem) {
    PlacesMeasurementSystem["METRIC"] = "metric";
    PlacesMeasurementSystem["US_IMPERIAL"] = "imperial";
})(PlacesMeasurementSystem || (PlacesMeasurementSystem = {}));
const configDefaults = {
    apiKey: '',
    placeTypes: {},
    mapZoom: 15,
    latitude: 52.3760992,
    longitude: 4.900779,
    radius: 2,
    limitList: 100,
    measurementSystem: PlacesMeasurementSystem.METRIC,
    deElevatedHeader: false,
};
export class PlacesJourneyConfigService {
    constructor(config, itemModel) {
        this.itemModel = itemModel;
        config = config || {};
        this._config = Object.assign(Object.assign({}, configDefaults), config);
        // update the API key from ItemModel if it was not provided from a static config
        if (this.itemModel && config.apiKey === undefined) {
            this.itemModel
                .property('apiKey', this._config.apiKey)
                .subscribe({
                next: (key) => {
                    this._config.apiKey = key;
                },
            })
                .unsubscribe();
        }
    }
    get defaults() {
        return configDefaults;
    }
    /**
     * API key for Google Maps
     */
    get apiKey() {
        return this._config.apiKey;
    }
    /**
     * Place types configuration. Used to set up marker image and the filter label for any place type available in the service
     */
    get placeTypes() {
        return this._config.placeTypes;
    }
    /**
     * Default zoom value for a map
     */
    get mapZoom() {
        return this._config.mapZoom;
    }
    /**
     * Default map latitude
     */
    get latitude() {
        return this._config.latitude;
    }
    /**
     * Default map longitude
     */
    get longitude() {
        return this._config.longitude;
    }
    /**
     * Default radius in which the locations are being searched while map is not loaded
     */
    get radius() {
        return this._config.radius;
    }
    /**
     * Maximum amount of locations to display
     */
    get limitList() {
        return this._config.limitList;
    }
    /**
     * Measurement system
     */
    get measurementSystem() {
        return this._config.measurementSystem;
    }
    /**
     * Dev config to enable de-elevated header
     */
    get deElevatedHeader() {
        return this._config.deElevatedHeader;
    }
}
PlacesJourneyConfigService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PlacesJourneyConfigService, deps: [{ token: PlacesJourneyConfigurationToken, optional: true }, { token: i1.ItemModel, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
PlacesJourneyConfigService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PlacesJourneyConfigService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PlacesJourneyConfigService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [PlacesJourneyConfigurationToken]
                }] }, { type: i1.ItemModel, decorators: [{
                    type: Optional
                }] }]; } });
