// @backbase/ui-ang/fesm2015/backbase-ui-ang-logo.js
import * as i1 from '@angular/common';
import { CommonModule } from '@angular/common';
import * as i0 from '@angular/core';
import { Component, Input, NgModule } from '@angular/core';

/**
 * @name LogoComponent
 *
 * @description
 * Component that displays the company logo.
 */
class LogoComponent {
    constructor() {
        /**
         * The type of the logo.  Defaults to 'full'.
         */
        this.type = 'full';
        /**
         * Flag to indicate whether to use the inverse colors or not.  Defaults to false.
         */
        this.isInverse = false;
    }
    /**
     * @name LogoComponent#isEmblem
     * @returns Whether this component should show the emblem only or the full logo.
     *
     * @description
     * Returns whether this component should show the emblem only or the full logo.
     */
    get isEmblem() {
        return this.type === 'emblem';
    }
    ngOnInit() { }
}
LogoComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LogoComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
LogoComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: LogoComponent, selector: "bb-logo-ui", inputs: { type: "type", isInverse: "isInverse" }, ngImport: i0, template: "<div\n  class=\"bb-logo\"\n  [ngClass]=\"{\n    'bb-logo--emblem': isEmblem,\n    'bb-logo--inverse': isInverse\n  }\"\n></div>\n", directives: [{ type: i1.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LogoComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-logo-ui',
                    templateUrl: './logo.component.html',
                }]
        }], ctorParameters: function () { return []; }, propDecorators: { type: [{
                type: Input
            }], isInverse: [{
                type: Input
            }] } });

class LogoModule {
}
LogoModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LogoModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LogoModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LogoModule, declarations: [LogoComponent], imports: [CommonModule], exports: [LogoComponent] });
LogoModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LogoModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LogoModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [LogoComponent],
                    exports: [LogoComponent],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { LogoComponent, LogoModule };
