// @backbase/ui-ang/fesm2015/backbase-ui-ang-badge.js
import * as i1 from '@angular/common';
import { CommonModule } from '@angular/common';
import * as i0 from '@angular/core';
import { Component, Input, NgModule } from '@angular/core';

/**
 * @name BadgeComponent
 *
 * @description
 * Component that displays a badge.
 */
class BadgeComponent {
    constructor() {
        /**
         * The (background) color of the badge.  Defaults to 'primary'.
         */
        this.color = 'primary';
        /**
         * The abbreviation to show when using circle styling.
         */
        this.abbr = false;
        /**
         * Whether circle styling should be applied.
         */
        this.circle = false;
        /**
         * Whether text wrap styling should be applied.
         */
        this.wrap = false;
    }
    ngOnInit() {
        if (typeof this.text === 'undefined') {
            throw new Error(`"text" input is required in "${this.constructor.name}"`);
        }
    }
}
BadgeComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BadgeComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
BadgeComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: BadgeComponent, selector: "bb-badge-ui", inputs: { color: "color", text: "text", abbr: "abbr", circle: "circle", wrap: "wrap", ariaLabel: "ariaLabel" }, ngImport: i0, template: "<span\n  class=\"badge break-word\"\n  [ngClass]=\"[color ? 'badge-' + color : '', circle ? 'badge-pill' : '', wrap ? 'text-wrap' : '']\"\n>\n  <ng-container *ngIf=\"!abbr\">\n    {{ text }}\n  </ng-container>\n  <abbr [attr.aria-label]=\"ariaLabel\" title=\"{{ text }}\" *ngIf=\"abbr\">\n    {{ abbr === true && text ? text[0] : abbr }}\n  </abbr>\n</span>\n", directives: [{ type: i1.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BadgeComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-badge-ui',
                    templateUrl: './badge.component.html',
                }]
        }], propDecorators: { color: [{
                type: Input
            }], text: [{
                type: Input
            }], abbr: [{
                type: Input
            }], circle: [{
                type: Input
            }], wrap: [{
                type: Input
            }], ariaLabel: [{
                type: Input
            }] } });

class BadgeModule {
}
BadgeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BadgeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
BadgeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BadgeModule, declarations: [BadgeComponent], imports: [CommonModule], exports: [BadgeComponent] });
BadgeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BadgeModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BadgeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [BadgeComponent],
                    exports: [BadgeComponent],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { BadgeComponent, BadgeModule };
