// @backbase/ui-ang/fesm2015/backbase-ui-ang-product-item-debit-card.js
import * as i2 from '@angular/common';
import { CommonModule } from '@angular/common';
import * as i0 from '@angular/core';
import { Directive, Component, ContentChild, Input, NgModule } from '@angular/core';
import * as i3 from '@backbase/ui-ang/payment-card-number-pipe';
import { DefaultPaymentCardNumberFormat, PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import * as i1 from '@backbase/ui-ang/icon';
import { IconModule } from '@backbase/ui-ang/icon';
import { AmountModule } from '@backbase/ui-ang/amount';

class CustomProductItemDebitCardTitleDirective {
    constructor(templateRef) {
        this.templateRef = templateRef;
    }
}
CustomProductItemDebitCardTitleDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomProductItemDebitCardTitleDirective, deps: [{ token: i0.TemplateRef }], target: i0.ɵɵFactoryTarget.Directive });
CustomProductItemDebitCardTitleDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomProductItemDebitCardTitleDirective, selector: "ng-template[bbCustomProductItemDebitCardTitle]", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomProductItemDebitCardTitleDirective, decorators: [{
            type: Directive,
            args: [{ selector: 'ng-template[bbCustomProductItemDebitCardTitle]' }]
        }], ctorParameters: function () { return [{ type: i0.TemplateRef }]; } });
class ProductItemDebitCardComponent {
    constructor() {
        this.hostRef = this;
        /**
         * Configuration of how the product number should be formatted.
         */
        this.productNumberFormat = DefaultPaymentCardNumberFormat;
        /**
         * The active state of card (it adds active class for blue line)
         */
        this.active = true;
        /**
         * The favorite state of card (add favorite icon next to title)
         */
        this.favorite = false;
        /**
         * The favorite icon color (define star icon color)
         */
        this.favoriteIconColor = 'primary';
    }
}
ProductItemDebitCardComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemDebitCardComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ProductItemDebitCardComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ProductItemDebitCardComponent, selector: "bb-product-item-debit-card-ui", inputs: { title: "title", productNumber: "productNumber", productNumberFormat: "productNumberFormat", highlight: "highlight", active: "active", productDetailsInfo: "productDetailsInfo", favorite: "favorite", favoriteIconColor: "favoriteIconColor" }, queries: [{ propertyName: "customProductItemDebitCardTitle", first: true, predicate: CustomProductItemDebitCardTitleDirective, descendants: true, static: true }], ngImport: i0, template: "<div [class.active]=\"active\" class=\"bb-product-item product-item\">\n  <div class=\"product-item-content\">\n    <ng-container *ngIf=\"customProductItemDebitCardTitle; else defaultProductItemDebitCardTitle\">\n      <ng-container *ngTemplateOutlet=\"customProductItemDebitCardTitle?.templateRef; context: hostRef\"> </ng-container>\n    </ng-container>\n\n    <div *ngIf=\"productDetailsInfo?.owners\" class=\"bb-product-item__owners\">\n      <span data-role=\"productDetailsInfo-owners\">\n        {{ productDetailsInfo?.owners }}\n      </span>\n    </div>\n  </div>\n</div>\n\n<ng-template #defaultProductItemDebitCardTitle>\n  <div class=\"bb-account-info\">\n    <div data-role=\"card-title\" class=\"bb-account-info__title\">\n      {{ title }}\n      <bb-icon-ui\n        *ngIf=\"favorite\"\n        size=\"sm\"\n        class=\"bb-account-info__icon\"\n        [color]=\"favoriteIconColor\"\n        data-role=\"favorite-icon\"\n        name=\"star\"\n      >\n      </bb-icon-ui>\n    </div>\n    <div\n      *ngIf=\"productNumber\"\n      data-role=\"card-sub-title\"\n      class=\"bb-subtitle bb-text-support bb-account-info__product-number\"\n    >\n      {{ productNumber | paymentCardNumber: productNumberFormat }}\n    </div>\n  </div>\n</ng-template>\n", components: [{ type: i1.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i2.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }], pipes: { "paymentCardNumber": i3.PaymentCardNumberPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemDebitCardComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-product-item-debit-card-ui',
                    templateUrl: './product-item-debit-card.component.html',
                }]
        }], propDecorators: { customProductItemDebitCardTitle: [{
                type: ContentChild,
                args: [CustomProductItemDebitCardTitleDirective, { static: true }]
            }], title: [{
                type: Input
            }], productNumber: [{
                type: Input
            }], productNumberFormat: [{
                type: Input
            }], highlight: [{
                type: Input
            }], active: [{
                type: Input
            }], productDetailsInfo: [{
                type: Input
            }], favorite: [{
                type: Input
            }], favoriteIconColor: [{
                type: Input
            }] } });

class ProductItemDebitCardModule {
}
ProductItemDebitCardModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemDebitCardModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductItemDebitCardModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemDebitCardModule, declarations: [ProductItemDebitCardComponent, CustomProductItemDebitCardTitleDirective], imports: [CommonModule, AmountModule, PaymentCardNumberModule, IconModule], exports: [ProductItemDebitCardComponent, CustomProductItemDebitCardTitleDirective] });
ProductItemDebitCardModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemDebitCardModule, imports: [[CommonModule, AmountModule, PaymentCardNumberModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemDebitCardModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, AmountModule, PaymentCardNumberModule, IconModule],
                    declarations: [ProductItemDebitCardComponent, CustomProductItemDebitCardTitleDirective],
                    exports: [ProductItemDebitCardComponent, CustomProductItemDebitCardTitleDirective],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { CustomProductItemDebitCardTitleDirective, ProductItemDebitCardComponent, ProductItemDebitCardModule };
