// @backbase/ui-ang/fesm2015/backbase-ui-ang-product-item-investment-account.js
import * as i3 from '@angular/common';
import { CommonModule } from '@angular/common';
import * as i0 from '@angular/core';
import { Directive, Component, ContentChild, Input, NgModule } from '@angular/core';
import * as i5 from '@backbase/ui-ang/payment-card-number-pipe';
import { DefaultPaymentCardNumberFormat, PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import * as i1 from '@backbase/ui-ang/icon';
import { IconModule } from '@backbase/ui-ang/icon';
import * as i2 from '@backbase/ui-ang/amount';
import { AmountModule } from '@backbase/ui-ang/amount';
import * as i4 from '@backbase/ui-ang/highlight';
import { HighlightModule } from '@backbase/ui-ang/highlight';

class CustomProductItemInvestmentAccountTitleDirective {
    constructor(templateRef) {
        this.templateRef = templateRef;
    }
}
CustomProductItemInvestmentAccountTitleDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomProductItemInvestmentAccountTitleDirective, deps: [{ token: i0.TemplateRef }], target: i0.ɵɵFactoryTarget.Directive });
CustomProductItemInvestmentAccountTitleDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomProductItemInvestmentAccountTitleDirective, selector: "ng-template[bbCustomProductItemInvestmentAccountTitle]", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomProductItemInvestmentAccountTitleDirective, decorators: [{
            type: Directive,
            args: [{ selector: 'ng-template[bbCustomProductItemInvestmentAccountTitle]' }]
        }], ctorParameters: function () { return [{ type: i0.TemplateRef }]; } });
class CustomProductItemInvestmentAccountDetailsFormatterDirective {
    constructor(templateRef) {
        this.templateRef = templateRef;
    }
}
CustomProductItemInvestmentAccountDetailsFormatterDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomProductItemInvestmentAccountDetailsFormatterDirective, deps: [{ token: i0.TemplateRef }], target: i0.ɵɵFactoryTarget.Directive });
CustomProductItemInvestmentAccountDetailsFormatterDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomProductItemInvestmentAccountDetailsFormatterDirective, selector: "ng-template[bbCustomProductItemInvestmentAccountDetailsFormatter]", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomProductItemInvestmentAccountDetailsFormatterDirective, decorators: [{
            type: Directive,
            args: [{ selector: 'ng-template[bbCustomProductItemInvestmentAccountDetailsFormatter]' }]
        }], ctorParameters: function () { return [{ type: i0.TemplateRef }]; } });
class ProductItemInvestmentAccountComponent {
    constructor() {
        this.hostRef = this;
        /**
         * Configuration of how the product number should be formatted.
         */
        this.productNumberFormat = DefaultPaymentCardNumberFormat;
        /**
         * The active state of card (it adds active class for blue line)
         */
        this.active = true;
        /**
         * The favorite state of card (add favorite icon next to title)
         */
        this.favorite = false;
        /**
         * The favorite icon color (define star icon color)
         */
        this.favoriteIconColor = 'primary';
    }
}
ProductItemInvestmentAccountComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemInvestmentAccountComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ProductItemInvestmentAccountComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ProductItemInvestmentAccountComponent, selector: "bb-product-item-investment-account-ui", inputs: { title: "title", productNumber: "productNumber", productNumberFormat: "productNumberFormat", currency: "currency", amount: "amount", highlight: "highlight", productDetailsInfo: "productDetailsInfo", active: "active", favorite: "favorite", favoriteIconColor: "favoriteIconColor" }, queries: [{ propertyName: "customProductItemInvestmentAccountTitle", first: true, predicate: CustomProductItemInvestmentAccountTitleDirective, descendants: true, static: true }, { propertyName: "customProductItemInvestmentAccountDetailsFormatter", first: true, predicate: CustomProductItemInvestmentAccountDetailsFormatterDirective, descendants: true, static: true }], ngImport: i0, template: "<div [class.active]=\"active\" class=\"bb-product-item product-item\">\n  <div class=\"product-item-content\">\n    <div class=\"bb-stack bb-stack--wrap\">\n      <div class=\"bb-stack__item\">\n        <ng-container *ngIf=\"customProductItemInvestmentAccountTitle; else defaultProductItemInvestmentAccountTitle\">\n          <ng-container *ngTemplateOutlet=\"customProductItemInvestmentAccountTitle?.templateRef; context: hostRef\">\n          </ng-container>\n        </ng-container>\n      </div>\n      <div class=\"bb-stack__break bb-stack__break--xs\"></div>\n      <div class=\"bb-stack__item bb-stack__item--push-right bb-stack__item--spacing-none bb-product-item__details\">\n        <ng-container\n          *ngIf=\"\n            customProductItemInvestmentAccountDetailsFormatter;\n            else defaultProductItemInvestmentAccountDetailsFormatter\n          \"\n        >\n          <ng-container\n            *ngTemplateOutlet=\"customProductItemInvestmentAccountDetailsFormatter?.templateRef; context: hostRef\"\n          >\n          </ng-container>\n        </ng-container>\n      </div>\n    </div>\n    <div *ngIf=\"productDetailsInfo?.owners\" class=\"bb-product-item__owners\">\n      <span data-role=\"productDetailsInfo-owners\" class=\"bb-subtitle\">\n        {{ productDetailsInfo?.owners }}\n      </span>\n    </div>\n  </div>\n\n  <ng-template #defaultProductItemInvestmentAccountTitle>\n    <div class=\"bb-account-info\">\n      <div data-role=\"card-title\" class=\"bb-account-info__title\">\n        {{ title }}\n        <bb-icon-ui\n          *ngIf=\"favorite\"\n          size=\"sm\"\n          class=\"bb-account-info__icon\"\n          [color]=\"favoriteIconColor\"\n          data-role=\"favorite-icon\"\n          name=\"star\"\n        >\n        </bb-icon-ui>\n      </div>\n      <div\n        *ngIf=\"productNumber\"\n        data-role=\"card-sub-title\"\n        class=\"bb-subtitle bb-text-support bb-account-info__product-number\"\n      >\n        {{ productNumber | paymentCardNumber: productNumberFormat }}\n      </div>\n    </div>\n  </ng-template>\n\n  <ng-template #defaultProductItemInvestmentAccountDetailsFormatter>\n    <bb-amount-ui\n      class=\"bb-highlight bb-product-item__amount\"\n      [amount]=\"amount\"\n      [currency]=\"currency\"\n      [bbHighlight]=\"highlight\"\n    ></bb-amount-ui>\n  </ng-template>\n</div>\n", components: [{ type: i1.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i2.AmountComponent, selector: "bb-amount-ui", inputs: ["currency", "showPlusSign", "mapCurrency", "showPercent", "abbreviate", "decimalPlaces", "trailingZeroes", "amount"] }], directives: [{ type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i3.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }, { type: i4.HighlightDirective, selector: "[bbHighlight]", inputs: ["amount", "bbHighlight", "negativeClass", "positiveClass"] }], pipes: { "paymentCardNumber": i5.PaymentCardNumberPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemInvestmentAccountComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-product-item-investment-account-ui',
                    templateUrl: './product-item-investment-account.component.html',
                }]
        }], propDecorators: { customProductItemInvestmentAccountTitle: [{
                type: ContentChild,
                args: [CustomProductItemInvestmentAccountTitleDirective, { static: true }]
            }], customProductItemInvestmentAccountDetailsFormatter: [{
                type: ContentChild,
                args: [CustomProductItemInvestmentAccountDetailsFormatterDirective, { static: true }]
            }], title: [{
                type: Input
            }], productNumber: [{
                type: Input
            }], productNumberFormat: [{
                type: Input
            }], currency: [{
                type: Input
            }], amount: [{
                type: Input
            }], highlight: [{
                type: Input
            }], productDetailsInfo: [{
                type: Input
            }], active: [{
                type: Input
            }], favorite: [{
                type: Input
            }], favoriteIconColor: [{
                type: Input
            }] } });

class ProductItemInvestmentAccountModule {
}
ProductItemInvestmentAccountModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemInvestmentAccountModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductItemInvestmentAccountModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemInvestmentAccountModule, declarations: [ProductItemInvestmentAccountComponent,
        CustomProductItemInvestmentAccountTitleDirective,
        CustomProductItemInvestmentAccountDetailsFormatterDirective], imports: [CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule], exports: [ProductItemInvestmentAccountComponent,
        CustomProductItemInvestmentAccountTitleDirective,
        CustomProductItemInvestmentAccountDetailsFormatterDirective] });
ProductItemInvestmentAccountModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemInvestmentAccountModule, imports: [[CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemInvestmentAccountModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule],
                    declarations: [
                        ProductItemInvestmentAccountComponent,
                        CustomProductItemInvestmentAccountTitleDirective,
                        CustomProductItemInvestmentAccountDetailsFormatterDirective,
                    ],
                    exports: [
                        ProductItemInvestmentAccountComponent,
                        CustomProductItemInvestmentAccountTitleDirective,
                        CustomProductItemInvestmentAccountDetailsFormatterDirective,
                    ],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { CustomProductItemInvestmentAccountDetailsFormatterDirective, CustomProductItemInvestmentAccountTitleDirective, ProductItemInvestmentAccountComponent, ProductItemInvestmentAccountModule };
