// @backbase/ui-ang/fesm2015/backbase-ui-ang-iban-pipe.js
import { CommonModule } from '@angular/common';
import * as i0 from '@angular/core';
import { Pipe, NgModule } from '@angular/core';
import { segment } from '@backbase/ui-ang/util';
import * as i1 from '@backbase/ui-ang/account-number-pipe';

/**
 * @name bbIban
 *
 * @description
 * Pipe for formatting iban
 */
class BbIbanPipe {
    /**
     * @name BbIbanPipe#constructor
     * @param accountsDisplayingFormatService - AccountsDisplayingFormatService
     *
     * @internal
     */
    constructor(accountsDisplayingFormatService) {
        this.accountsDisplayingFormatService = accountsDisplayingFormatService;
        this.defaultFormat = this.accountsDisplayingFormatService.getDefaultFormat('iban');
    }
    /**
     * @name BbIbanPipe#transform
     * @param value - the iban that should be formatted
     *
     * @description
     * Splits the iban up with spaces
     * Applies global numbers format configuration for IBAN attribute type
     */
    transform(value) {
        return segment(value, this.defaultFormat.segments);
    }
}
BbIbanPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BbIbanPipe, deps: [{ token: i1.AccountsDisplayingFormatService }], target: i0.ɵɵFactoryTarget.Pipe });
BbIbanPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BbIbanPipe, name: "bbIban" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BbIbanPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'bbIban',
                }]
        }], ctorParameters: function () { return [{ type: i1.AccountsDisplayingFormatService }]; } });

class BbIbanPipeModule {
}
BbIbanPipeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BbIbanPipeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
BbIbanPipeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BbIbanPipeModule, declarations: [BbIbanPipe], imports: [CommonModule], exports: [BbIbanPipe] });
BbIbanPipeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BbIbanPipeModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BbIbanPipeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [BbIbanPipe],
                    exports: [BbIbanPipe],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { BbIbanPipe, BbIbanPipeModule };
