// @backbase/ui-ang/fesm2015/backbase-ui-ang-card-vendor.js
import * as i1 from '@angular/common';
import { CommonModule } from '@angular/common';
import * as i0 from '@angular/core';
import { Component, ChangeDetectionStrategy, Input, NgModule } from '@angular/core';

/**
 * The prefix for the vendor class name.
 */
const classNamePrefix = 'bb-card-vendor-';
const vendorToClassName = (vendor) => {
    if (typeof vendor !== 'string') {
        return '';
    }
    const classNamePostfix = vendor.trim().toLowerCase().replace(/\W/g, '-').replace(/-{2,}/g, '-');
    return `${classNamePrefix}${classNamePostfix}`;
};
const vendorToTitle = (vendor) => (vendor ? vendor.trim().toLowerCase().replace(/-/g, ' ') : '');
/**
 * @name CardVendorComponent
 *
 * @description
 * Component that displays the logo of a card vendor.
 */
class CardVendorComponent {
    constructor() {
        this.className = '';
        this.title = '';
    }
    ngOnInit() {
        if (typeof this.vendor !== 'string') {
            throw new Error(`"vendor" input is required in "${this.constructor.name}"`);
        }
    }
    ngOnChanges(changes) {
        this.className = vendorToClassName(this.vendor);
        this.title = vendorToTitle(this.vendor);
    }
}
CardVendorComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardVendorComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
CardVendorComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: CardVendorComponent, selector: "bb-card-vendor-ui", inputs: { vendor: "vendor" }, usesOnChanges: true, ngImport: i0, template: "<div role=\"img\" [attr.aria-label]=\"title\" [title]=\"title\" class=\"bb-card-vendor\" [ngClass]=\"className\"></div>\n", directives: [{ type: i1.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardVendorComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-card-vendor-ui',
                    templateUrl: './card-vendor.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { vendor: [{
                type: Input
            }] } });

class CardVendorModule {
}
CardVendorModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardVendorModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CardVendorModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardVendorModule, declarations: [CardVendorComponent], imports: [CommonModule], exports: [CardVendorComponent] });
CardVendorModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardVendorModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardVendorModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [CardVendorComponent],
                    exports: [CardVendorComponent],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { CardVendorComponent, CardVendorModule };
