// @backbase/ui-ang/fesm2015/backbase-ui-ang-header.js
import * as i1 from '@angular/common';
import { CommonModule } from '@angular/common';
import * as i0 from '@angular/core';
import { Component, Input, NgModule } from '@angular/core';

/**
 * @name HeaderComponent
 *
 * @description
 * Component that displays a header.
 */
class HeaderComponent {
}
HeaderComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HeaderComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
HeaderComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: HeaderComponent, selector: "bb-header-ui", inputs: { headingClasses: "headingClasses", headingType: "headingType", heading: "heading" }, ngImport: i0, template: "<ng-container [ngSwitch]=\"headingType\">\n  <h1 *ngSwitchCase=\"'h1'\" [ngClass]=\"headingClasses\" data-role=\"headings\">\n    {{ heading }}\n  </h1>\n  <h2 *ngSwitchCase=\"'h2'\" [ngClass]=\"headingClasses\" data-role=\"headings\">\n    {{ heading }}\n  </h2>\n  <h3 *ngSwitchCase=\"'h3'\" [ngClass]=\"headingClasses\" data-role=\"headings\">\n    {{ heading }}\n  </h3>\n  <h4 *ngSwitchCase=\"'h4'\" [ngClass]=\"headingClasses\" data-role=\"headings\">\n    {{ heading }}\n  </h4>\n  <h5 *ngSwitchCase=\"'h5'\" [ngClass]=\"headingClasses\" data-role=\"headings\">\n    {{ heading }}\n  </h5>\n  <h6 *ngSwitchCase=\"'h6'\" [ngClass]=\"headingClasses\" data-role=\"headings\">\n    {{ heading }}\n  </h6>\n</ng-container>\n", directives: [{ type: i1.NgSwitch, selector: "[ngSwitch]", inputs: ["ngSwitch"] }, { type: i1.NgSwitchCase, selector: "[ngSwitchCase]", inputs: ["ngSwitchCase"] }, { type: i1.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HeaderComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-header-ui',
                    templateUrl: './header.component.html',
                    styles: [],
                }]
        }], propDecorators: { headingClasses: [{
                type: Input
            }], headingType: [{
                type: Input
            }], heading: [{
                type: Input
            }] } });

class HeaderModule {
}
HeaderModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HeaderModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
HeaderModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HeaderModule, declarations: [HeaderComponent], imports: [CommonModule], exports: [HeaderComponent] });
HeaderModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HeaderModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HeaderModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [HeaderComponent],
                    exports: [HeaderComponent],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { HeaderComponent, HeaderModule };
