// @backbase/ui-ang/fesm2015/backbase-ui-ang-input-validation-message.js
import * as i1 from '@angular/common';
import { CommonModule } from '@angular/common';
import * as i0 from '@angular/core';
import { Component, ChangeDetectionStrategy, Input, NgModule } from '@angular/core';

/**
 * @name InputValidationMessageComponent
 *
 * @description
 * Component that displays an input validation message or helper.
 */
class InputValidationMessageComponent {
    constructor() {
        /**
         * Whether the errors template should be shown
         */
        this.showErrors = false;
    }
}
InputValidationMessageComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputValidationMessageComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
InputValidationMessageComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: InputValidationMessageComponent, selector: "bb-input-validation-message-ui", inputs: { showErrors: "showErrors", messageId: "messageId" }, ngImport: i0, template: "<ng-container *ngIf=\"showErrors\">\n  <small class=\"bb-input-validation-message\" role=\"alert\" [attr.id]=\"messageId || null\">\n    <ng-content></ng-content>\n  </small>\n</ng-container>\n", directives: [{ type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputValidationMessageComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-input-validation-message-ui',
                    templateUrl: './input-validation-message.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { showErrors: [{
                type: Input
            }], messageId: [{
                type: Input
            }] } });

class InputValidationMessageModule {
}
InputValidationMessageModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputValidationMessageModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputValidationMessageModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputValidationMessageModule, declarations: [InputValidationMessageComponent], imports: [CommonModule], exports: [InputValidationMessageComponent] });
InputValidationMessageModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputValidationMessageModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputValidationMessageModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [InputValidationMessageComponent],
                    exports: [InputValidationMessageComponent],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { InputValidationMessageComponent, InputValidationMessageModule };
