// @backbase/ui-ang/fesm2015/backbase-ui-ang-fieldset.js
import * as i0 from '@angular/core';
import { Component, ChangeDetectionStrategy, Input, NgModule } from '@angular/core';
import * as i1 from '@angular/common';
import { CommonModule } from '@angular/common';

/**
 * @name FieldsetComponent
 *
 * @description
 * Component that displays a fieldset group.
 */
class FieldsetComponent {
    constructor() {
        /**
         * Set disabled state of all form items within fieldset
         */
        this.disabled = false;
    }
}
FieldsetComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FieldsetComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
FieldsetComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: FieldsetComponent, selector: "bb-fieldset-ui", inputs: { heading: "heading", disabled: "disabled" }, ngImport: i0, template: "<fieldset [disabled]=\"disabled\" class=\"bb-fieldset\">\n  <legend *ngIf=\"heading\" class=\"bb-fieldset__heading\">\n    {{ heading }}\n  </legend>\n  <ng-content></ng-content>\n</fieldset>\n", directives: [{ type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FieldsetComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-fieldset-ui',
                    templateUrl: './fieldset.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { heading: [{
                type: Input
            }], disabled: [{
                type: Input
            }] } });

class FieldsetModule {
}
FieldsetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FieldsetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
FieldsetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FieldsetModule, declarations: [FieldsetComponent], imports: [CommonModule], exports: [FieldsetComponent] });
FieldsetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FieldsetModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FieldsetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [FieldsetComponent],
                    exports: [FieldsetComponent],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { FieldsetComponent, FieldsetModule };
