// @backbase/ui-ang/fesm2015/backbase-ui-ang-product-item-term-deposit.js
import * as i0 from '@angular/core';
import { Directive, Component, ContentChild, Input, NgModule } from '@angular/core';
import * as i5 from '@backbase/ui-ang/payment-card-number-pipe';
import { DefaultPaymentCardNumberFormat, PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import * as i1 from '@backbase/ui-ang/icon';
import { IconModule } from '@backbase/ui-ang/icon';
import * as i2 from '@backbase/ui-ang/amount';
import { AmountModule } from '@backbase/ui-ang/amount';
import * as i3 from '@angular/common';
import { CommonModule } from '@angular/common';
import * as i4 from '@backbase/ui-ang/highlight';
import { HighlightModule } from '@backbase/ui-ang/highlight';

class CustomProductItemTermDepositTitleDirective {
    constructor(templateRef) {
        this.templateRef = templateRef;
    }
}
CustomProductItemTermDepositTitleDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomProductItemTermDepositTitleDirective, deps: [{ token: i0.TemplateRef }], target: i0.ɵɵFactoryTarget.Directive });
CustomProductItemTermDepositTitleDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomProductItemTermDepositTitleDirective, selector: "ng-template[bbCustomProductItemTermDepositTitle]", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomProductItemTermDepositTitleDirective, decorators: [{
            type: Directive,
            args: [{ selector: 'ng-template[bbCustomProductItemTermDepositTitle]' }]
        }], ctorParameters: function () { return [{ type: i0.TemplateRef }]; } });
class CustomProductItemTermDepositDetailsFormatterDirective {
    constructor(templateRef) {
        this.templateRef = templateRef;
    }
}
CustomProductItemTermDepositDetailsFormatterDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomProductItemTermDepositDetailsFormatterDirective, deps: [{ token: i0.TemplateRef }], target: i0.ɵɵFactoryTarget.Directive });
CustomProductItemTermDepositDetailsFormatterDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomProductItemTermDepositDetailsFormatterDirective, selector: "ng-template[bbCustomProductItemTermDepositDetailsFormatter]", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomProductItemTermDepositDetailsFormatterDirective, decorators: [{
            type: Directive,
            args: [{ selector: 'ng-template[bbCustomProductItemTermDepositDetailsFormatter]' }]
        }], ctorParameters: function () { return [{ type: i0.TemplateRef }]; } });
class CustomProductItemTermDepositMainSectionDirective {
    constructor(templateRef) {
        this.templateRef = templateRef;
    }
}
CustomProductItemTermDepositMainSectionDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomProductItemTermDepositMainSectionDirective, deps: [{ token: i0.TemplateRef }], target: i0.ɵɵFactoryTarget.Directive });
CustomProductItemTermDepositMainSectionDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomProductItemTermDepositMainSectionDirective, selector: "ng-template[bbCustomProductItemTermDepositMainSection]", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomProductItemTermDepositMainSectionDirective, decorators: [{
            type: Directive,
            args: [{ selector: 'ng-template[bbCustomProductItemTermDepositMainSection]' }]
        }], ctorParameters: function () { return [{ type: i0.TemplateRef }]; } });
class ProductItemTermDepositComponent {
    constructor() {
        this.hostRef = this;
        /**
         * Configuration of how the product number should be formatted.
         */
        this.productNumberFormat = DefaultPaymentCardNumberFormat;
        /**
         * The active state of card (it adds active class for blue line)
         */
        this.active = true;
        /**
         * The favorite state of card (add favorite icon next to title)
         */
        this.favorite = false;
        /**
         * The favorite icon color (define star icon color)
         */
        this.favoriteIconColor = 'primary';
    }
}
ProductItemTermDepositComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemTermDepositComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ProductItemTermDepositComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ProductItemTermDepositComponent, selector: "bb-product-item-term-deposit-ui", inputs: { title: "title", productNumber: "productNumber", productNumberFormat: "productNumberFormat", currency: "currency", amount: "amount", mainSectionCurrency: "mainSectionCurrency", mainSectionAmount: "mainSectionAmount", mainSectionTitle: "mainSectionTitle", highlight: "highlight", productDetailsInfo: "productDetailsInfo", active: "active", favorite: "favorite", favoriteIconColor: "favoriteIconColor" }, queries: [{ propertyName: "customProductItemTermDepositTitle", first: true, predicate: CustomProductItemTermDepositTitleDirective, descendants: true, static: true }, { propertyName: "customProductItemTermDepositDetailsFormatter", first: true, predicate: CustomProductItemTermDepositDetailsFormatterDirective, descendants: true, static: true }, { propertyName: "customProductItemTermDepositMainSection", first: true, predicate: CustomProductItemTermDepositMainSectionDirective, descendants: true, static: true }], ngImport: i0, template: "<div [class.active]=\"active\" class=\"bb-product-item product-item\">\n  <div class=\"product-item-content\">\n    <div class=\"bb-stack bb-stack--wrap\" [ngClass]=\"{ 'bb-block bb-block--md': mainSectionAmount }\">\n      <div class=\"bb-stack__item\">\n        <ng-container *ngIf=\"customProductItemTermDepositTitle; else defaultProductItemTermDepositTitle\">\n          <ng-container *ngTemplateOutlet=\"customProductItemTermDepositTitle?.templateRef; context: hostRef\">\n          </ng-container>\n        </ng-container>\n      </div>\n      <div class=\"bb-stack__break bb-stack__break--xs\"></div>\n      <div class=\"bb-stack__item bb-stack__item--push-right bb-stack__item--spacing-none bb-product-item__details\">\n        <ng-container\n          *ngIf=\"customProductItemTermDepositDetailsFormatter; else defaultProductItemTermDepositDetailsFormatter\"\n        >\n          <ng-container *ngTemplateOutlet=\"customProductItemTermDepositDetailsFormatter?.templateRef; context: hostRef\">\n          </ng-container>\n        </ng-container>\n      </div>\n    </div>\n\n    <ng-container *ngIf=\"customProductItemTermDepositMainSection; else defaultProductItemTermDepositMainSection\">\n      <ng-container *ngTemplateOutlet=\"customProductItemTermDepositMainSection?.templateRef; context: hostRef\">\n      </ng-container>\n    </ng-container>\n\n    <div *ngIf=\"productDetailsInfo?.owners\" class=\"bb-text-default bb-product-item__owners\">\n      <span data-role=\"productDetailsInfo-owners\">\n        {{ productDetailsInfo?.owners }}\n      </span>\n    </div>\n  </div>\n</div>\n\n<ng-template #defaultProductItemTermDepositTitle>\n  <div data-role=\"card-title\" class=\"bb-text-bold\">\n    {{ title }}\n    <bb-icon-ui *ngIf=\"favorite\" size=\"sm\" [color]=\"favoriteIconColor\" data-role=\"favorite-icon\" name=\"star\">\n    </bb-icon-ui>\n  </div>\n  <div\n    *ngIf=\"productNumber\"\n    data-role=\"card-sub-title\"\n    class=\"bb-subtitle bb-text-support bb-account-info__product-number\"\n  >\n    {{ productNumber | paymentCardNumber: productNumberFormat }}\n  </div>\n</ng-template>\n\n<ng-template #defaultProductItemTermDepositDetailsFormatter>\n  <bb-amount-ui\n    class=\"bb-highlight bb-product-item__amount\"\n    [amount]=\"amount\"\n    [currency]=\"currency\"\n    [bbHighlight]=\"highlight\"\n  ></bb-amount-ui>\n</ng-template>\n\n<ng-template #defaultProductItemTermDepositMainSection>\n  <div *ngIf=\"mainSectionAmount\" class=\"bb-stack bb-stack--wrap bb-subtitle bb-text-support\">\n    <span class=\"bb-stack__item bb-product-item__main-section-title\">\n      {{ mainSectionTitle }}\n    </span>\n    <bb-amount-ui\n      [amount]=\"mainSectionAmount\"\n      [currency]=\"mainSectionCurrency\"\n      [bbHighlight]=\"highlight\"\n      class=\"\n        bb-stack__item bb-stack__item--push-right bb-stack__item--spacing-none\n        bb-product-item__main-section-amount\n      \"\n    ></bb-amount-ui>\n  </div>\n</ng-template>\n", components: [{ type: i1.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i2.AmountComponent, selector: "bb-amount-ui", inputs: ["currency", "showPlusSign", "mapCurrency", "showPercent", "abbreviate", "decimalPlaces", "trailingZeroes", "amount"] }], directives: [{ type: i3.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i3.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }, { type: i4.HighlightDirective, selector: "[bbHighlight]", inputs: ["amount", "bbHighlight", "negativeClass", "positiveClass"] }], pipes: { "paymentCardNumber": i5.PaymentCardNumberPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemTermDepositComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-product-item-term-deposit-ui',
                    templateUrl: './product-item-term-deposit.component.html',
                }]
        }], propDecorators: { customProductItemTermDepositTitle: [{
                type: ContentChild,
                args: [CustomProductItemTermDepositTitleDirective, { static: true }]
            }], customProductItemTermDepositDetailsFormatter: [{
                type: ContentChild,
                args: [CustomProductItemTermDepositDetailsFormatterDirective, { static: true }]
            }], customProductItemTermDepositMainSection: [{
                type: ContentChild,
                args: [CustomProductItemTermDepositMainSectionDirective, { static: true }]
            }], title: [{
                type: Input
            }], productNumber: [{
                type: Input
            }], productNumberFormat: [{
                type: Input
            }], currency: [{
                type: Input
            }], amount: [{
                type: Input
            }], mainSectionCurrency: [{
                type: Input
            }], mainSectionAmount: [{
                type: Input
            }], mainSectionTitle: [{
                type: Input
            }], highlight: [{
                type: Input
            }], productDetailsInfo: [{
                type: Input
            }], active: [{
                type: Input
            }], favorite: [{
                type: Input
            }], favoriteIconColor: [{
                type: Input
            }] } });

class ProductItemTermDepositModule {
}
ProductItemTermDepositModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemTermDepositModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductItemTermDepositModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemTermDepositModule, declarations: [ProductItemTermDepositComponent,
        CustomProductItemTermDepositTitleDirective,
        CustomProductItemTermDepositDetailsFormatterDirective,
        CustomProductItemTermDepositMainSectionDirective], imports: [CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule], exports: [ProductItemTermDepositComponent,
        CustomProductItemTermDepositTitleDirective,
        CustomProductItemTermDepositDetailsFormatterDirective,
        CustomProductItemTermDepositMainSectionDirective] });
ProductItemTermDepositModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemTermDepositModule, imports: [[CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemTermDepositModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule],
                    declarations: [
                        ProductItemTermDepositComponent,
                        CustomProductItemTermDepositTitleDirective,
                        CustomProductItemTermDepositDetailsFormatterDirective,
                        CustomProductItemTermDepositMainSectionDirective,
                    ],
                    exports: [
                        ProductItemTermDepositComponent,
                        CustomProductItemTermDepositTitleDirective,
                        CustomProductItemTermDepositDetailsFormatterDirective,
                        CustomProductItemTermDepositMainSectionDirective,
                    ],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { CustomProductItemTermDepositDetailsFormatterDirective, CustomProductItemTermDepositMainSectionDirective, CustomProductItemTermDepositTitleDirective, ProductItemTermDepositComponent, ProductItemTermDepositModule };
