// @backbase/ui-ang/fesm2015/backbase-ui-ang-product-item-loan.js
import * as i3 from '@angular/common';
import { CommonModule } from '@angular/common';
import * as i0 from '@angular/core';
import { Directive, Component, ContentChild, Input, NgModule } from '@angular/core';
import * as i5 from '@backbase/ui-ang/payment-card-number-pipe';
import { DefaultPaymentCardNumberFormat, PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import * as i1 from '@backbase/ui-ang/icon';
import { IconModule } from '@backbase/ui-ang/icon';
import * as i2 from '@backbase/ui-ang/amount';
import { AmountModule } from '@backbase/ui-ang/amount';
import * as i4 from '@backbase/ui-ang/highlight';
import { HighlightModule } from '@backbase/ui-ang/highlight';

class CustomProductItemLoanTitleDirective {
    constructor(templateRef) {
        this.templateRef = templateRef;
    }
}
CustomProductItemLoanTitleDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomProductItemLoanTitleDirective, deps: [{ token: i0.TemplateRef }], target: i0.ɵɵFactoryTarget.Directive });
CustomProductItemLoanTitleDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomProductItemLoanTitleDirective, selector: "ng-template[bbCustomProductItemLoanTitle]", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomProductItemLoanTitleDirective, decorators: [{
            type: Directive,
            args: [{ selector: 'ng-template[bbCustomProductItemLoanTitle]' }]
        }], ctorParameters: function () { return [{ type: i0.TemplateRef }]; } });
class CustomProductItemLoanDetailsFormatterDirective {
    constructor(templateRef) {
        this.templateRef = templateRef;
    }
}
CustomProductItemLoanDetailsFormatterDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomProductItemLoanDetailsFormatterDirective, deps: [{ token: i0.TemplateRef }], target: i0.ɵɵFactoryTarget.Directive });
CustomProductItemLoanDetailsFormatterDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomProductItemLoanDetailsFormatterDirective, selector: "ng-template[bbCustomProductItemLoanDetailsFormatter]", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomProductItemLoanDetailsFormatterDirective, decorators: [{
            type: Directive,
            args: [{ selector: 'ng-template[bbCustomProductItemLoanDetailsFormatter]' }]
        }], ctorParameters: function () { return [{ type: i0.TemplateRef }]; } });
class ProductItemLoanComponent {
    constructor() {
        this.hostRef = this;
        /**
         * Configuration of how the product number should be formatted.
         */
        this.productNumberFormat = DefaultPaymentCardNumberFormat;
        /**
         * The active state of card (it adds active class for blue line)
         */
        this.active = true;
        /**
         * The favorite state of card (add favorite icon next to title)
         */
        this.favorite = false;
        /**
         * The favorite icon color (define star icon color)
         */
        this.favoriteIconColor = 'primary';
    }
}
ProductItemLoanComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemLoanComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ProductItemLoanComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ProductItemLoanComponent, selector: "bb-product-item-loan-ui", inputs: { title: "title", productNumber: "productNumber", productNumberFormat: "productNumberFormat", currency: "currency", amount: "amount", highlight: "highlight", productDetailsInfo: "productDetailsInfo", active: "active", favorite: "favorite", favoriteIconColor: "favoriteIconColor" }, queries: [{ propertyName: "customProductItemLoanTitle", first: true, predicate: CustomProductItemLoanTitleDirective, descendants: true, static: true }, { propertyName: "customProductItemLoanDetailsFormatter", first: true, predicate: CustomProductItemLoanDetailsFormatterDirective, descendants: true, static: true }], ngImport: i0, template: "<div [class.active]=\"active\" class=\"bb-product-item product-item\">\n  <div class=\"product-item-content\">\n    <div class=\"bb-stack bb-stack--wrap\">\n      <div class=\"bb-stack__item\">\n        <ng-container *ngIf=\"customProductItemLoanTitle; else defaultProductItemLoanTitle\">\n          <ng-container *ngTemplateOutlet=\"customProductItemLoanTitle?.templateRef; context: hostRef\"> </ng-container>\n        </ng-container>\n      </div>\n      <div class=\"bb-stack__break bb-stack__break--xs\"></div>\n      <div class=\"bb-stack__item bb-stack__item--push-right bb-stack__item--spacing-none bb-product-item__details\">\n        <ng-container *ngIf=\"customProductItemLoanDetailsFormatter; else defaultProductItemLoanDetailsFormatter\">\n          <ng-container *ngTemplateOutlet=\"customProductItemLoanDetailsFormatter?.templateRef; context: hostRef\">\n          </ng-container>\n        </ng-container>\n      </div>\n    </div>\n    <div *ngIf=\"productDetailsInfo?.owners\" class=\"bb-product-item__owners\">\n      <span data-role=\"productDetailsInfo-owners\">\n        {{ productDetailsInfo?.owners }}\n      </span>\n    </div>\n  </div>\n</div>\n\n<ng-template #defaultProductItemLoanTitle>\n  <div class=\"bb-account-info\">\n    <div data-role=\"card-title\" class=\"bb-account-info__title\">\n      {{ title }}\n      <bb-icon-ui\n        *ngIf=\"favorite\"\n        size=\"sm\"\n        class=\"bb-account-info__icon\"\n        [color]=\"favoriteIconColor\"\n        data-role=\"favorite-icon\"\n        name=\"star\"\n      >\n      </bb-icon-ui>\n    </div>\n    <div\n      *ngIf=\"productNumber\"\n      data-role=\"card-sub-title\"\n      class=\"bb-text-support bb-subtitle bb-account-info__product-number\"\n    >\n      {{ productNumber | paymentCardNumber: productNumberFormat }}\n    </div>\n  </div>\n</ng-template>\n\n<ng-template #defaultProductItemLoanDetailsFormatter>\n  <bb-amount-ui\n    class=\"bb-highlight bb-product-item__amount\"\n    [amount]=\"amount\"\n    [currency]=\"currency\"\n    [bbHighlight]=\"highlight\"\n  ></bb-amount-ui>\n</ng-template>\n", components: [{ type: i1.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i2.AmountComponent, selector: "bb-amount-ui", inputs: ["currency", "showPlusSign", "mapCurrency", "showPercent", "abbreviate", "decimalPlaces", "trailingZeroes", "amount"] }], directives: [{ type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i3.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }, { type: i4.HighlightDirective, selector: "[bbHighlight]", inputs: ["amount", "bbHighlight", "negativeClass", "positiveClass"] }], pipes: { "paymentCardNumber": i5.PaymentCardNumberPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemLoanComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-product-item-loan-ui',
                    templateUrl: './product-item-loan.component.html',
                }]
        }], propDecorators: { customProductItemLoanTitle: [{
                type: ContentChild,
                args: [CustomProductItemLoanTitleDirective, { static: true }]
            }], customProductItemLoanDetailsFormatter: [{
                type: ContentChild,
                args: [CustomProductItemLoanDetailsFormatterDirective, { static: true }]
            }], title: [{
                type: Input
            }], productNumber: [{
                type: Input
            }], productNumberFormat: [{
                type: Input
            }], currency: [{
                type: Input
            }], amount: [{
                type: Input
            }], highlight: [{
                type: Input
            }], productDetailsInfo: [{
                type: Input
            }], active: [{
                type: Input
            }], favorite: [{
                type: Input
            }], favoriteIconColor: [{
                type: Input
            }] } });

class ProductItemLoanModule {
}
ProductItemLoanModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemLoanModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductItemLoanModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemLoanModule, declarations: [ProductItemLoanComponent,
        CustomProductItemLoanTitleDirective,
        CustomProductItemLoanDetailsFormatterDirective], imports: [CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule], exports: [ProductItemLoanComponent,
        CustomProductItemLoanTitleDirective,
        CustomProductItemLoanDetailsFormatterDirective] });
ProductItemLoanModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemLoanModule, imports: [[CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemLoanModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule],
                    declarations: [
                        ProductItemLoanComponent,
                        CustomProductItemLoanTitleDirective,
                        CustomProductItemLoanDetailsFormatterDirective,
                    ],
                    exports: [
                        ProductItemLoanComponent,
                        CustomProductItemLoanTitleDirective,
                        CustomProductItemLoanDetailsFormatterDirective,
                    ],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { CustomProductItemLoanDetailsFormatterDirective, CustomProductItemLoanTitleDirective, ProductItemLoanComponent, ProductItemLoanModule };
