// @backbase/ui-ang/fesm2015/backbase-ui-ang-payment-card.js
import * as i3 from '@angular/common';
import { CommonModule } from '@angular/common';
import * as i0 from '@angular/core';
import { Component, ChangeDetectionStrategy, Input, NgModule } from '@angular/core';
import * as i4 from '@backbase/ui-ang/payment-card-number-pipe';
import { DefaultPaymentCardNumberFormat, PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import * as i1 from '@backbase/ui-ang/logo';
import { LogoModule } from '@backbase/ui-ang/logo';
import * as i2 from '@backbase/ui-ang/card-vendor';
import { CardVendorModule } from '@backbase/ui-ang/card-vendor';

/**
 * The prefix for the payment card class names.
 */
const classNamePrefix = 'bb-payment-card-';
class PaymentCardComponent {
    constructor() {
        /**
         * Configuration of how the card number should be formatted.
         */
        this.paymentCardNumberFormat = DefaultPaymentCardNumberFormat;
    }
    ngOnInit() {
        if (typeof this.paymentCard === 'undefined') {
            throw new Error(`"paymentCard" input is required in "${this.constructor.name}"`);
        }
    }
    get classNames() {
        if (this.paymentCard && this.paymentCard.cardName) {
            const classNamePostfix = this.paymentCard.cardName
                .toLowerCase()
                .replace(/[^a-zA-Z0-9 ]/g, ' ')
                .trim()
                .replace(/\s/g, '-')
                .replace(/-{2,}/g, '-');
            return classNamePostfix ? `${classNamePrefix}${classNamePostfix}` : undefined;
        }
        return undefined;
    }
}
PaymentCardComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
PaymentCardComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: PaymentCardComponent, selector: "bb-payment-card-ui", inputs: { paymentCard: "paymentCard", paymentCardNumberFormat: "paymentCardNumberFormat" }, ngImport: i0, template: "<div class=\"bb-payment-card\" [ngClass]=\"classNames\">\n  <div class=\"bb-payment-card__container\">\n    <div class=\"bb-payment-card__logo\">\n      <bb-logo-ui type=\"emblem\" [isInverse]=\"true\"></bb-logo-ui>\n    </div>\n    <div class=\"bb-payment-card__vendor\">\n      <bb-card-vendor-ui [vendor]=\"paymentCard?.vendor\"></bb-card-vendor-ui>\n    </div>\n    <div class=\"bb-payment-card__number\">\n      {{ paymentCard?.number | paymentCardNumber: paymentCardNumberFormat }}\n    </div>\n    <div class=\"bb-payment-card__name\">\n      {{ paymentCard?.name }}\n    </div>\n    <div class=\"bb-payment-card__expiration-date\">\n      <div class=\"bb-payment-card__expiration-date-label\" i18n=\"Expiration date label@@paymentCard.expirationDateLabel\">\n        VALID THRU\n      </div>\n      <span class=\"bb-payment-card__expiration-date-date\">\n        {{ paymentCard?.expirationDate | date: 'MM/yy' }}\n      </span>\n    </div>\n  </div>\n</div>\n", components: [{ type: i1.LogoComponent, selector: "bb-logo-ui", inputs: ["type", "isInverse"] }, { type: i2.CardVendorComponent, selector: "bb-card-vendor-ui", inputs: ["vendor"] }], directives: [{ type: i3.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }], pipes: { "paymentCardNumber": i4.PaymentCardNumberPipe, "date": i3.DatePipe }, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-payment-card-ui',
                    templateUrl: './payment-card.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return []; }, propDecorators: { paymentCard: [{
                type: Input
            }], paymentCardNumberFormat: [{
                type: Input
            }] } });

class PaymentCardModule {
}
PaymentCardModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PaymentCardModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardModule, declarations: [PaymentCardComponent], imports: [CommonModule, LogoModule, CardVendorModule, PaymentCardNumberModule], exports: [PaymentCardComponent] });
PaymentCardModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardModule, imports: [[CommonModule, LogoModule, CardVendorModule, PaymentCardNumberModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, LogoModule, CardVendorModule, PaymentCardNumberModule],
                    declarations: [PaymentCardComponent],
                    exports: [PaymentCardComponent],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { PaymentCardComponent, PaymentCardModule };
