// @backbase/ui-ang/fesm2015/backbase-ui-ang-phone-number-format-pipe.js
import { CommonModule } from '@angular/common';
import * as i0 from '@angular/core';
import { Pipe, NgModule } from '@angular/core';
import { AsYouType, parsePhoneNumber } from '@backbase/phone-number-lib';

/**
 * @name PhoneNumberFormatPipe
 *
 * @description
 * Pipe for formatting phone numbers to various international formats
 */
class PhoneNumberPipe {
    /**
     * @name PhoneNumberFormatPipe#transform
     * @param value - the phone number string to transform
     * @param countryCode - the two letter CDLR code for the country to format to
     * @param isInputElement - whether the value was entered in an input element
     *
     * @description
     * Formats a phone number to any entered country code international format
     */
    transform(value, countryCode, isInputElement) {
        try {
            if (isInputElement) {
                value = new AsYouType(countryCode).input(value.toString());
            }
            return parsePhoneNumber(value.toString(), countryCode).formatInternational();
        }
        catch (e) {
            return value.toString();
        }
    }
}
PhoneNumberPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PhoneNumberPipe, deps: [], target: i0.ɵɵFactoryTarget.Pipe });
PhoneNumberPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PhoneNumberPipe, name: "phoneNumber" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PhoneNumberPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'phoneNumber',
                }]
        }] });

class PhoneNumberModule {
}
PhoneNumberModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PhoneNumberModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PhoneNumberModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PhoneNumberModule, declarations: [PhoneNumberPipe], imports: [CommonModule], exports: [PhoneNumberPipe] });
PhoneNumberModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PhoneNumberModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PhoneNumberModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [PhoneNumberPipe],
                    exports: [PhoneNumberPipe],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { PhoneNumberModule, PhoneNumberPipe };
