// @backbase/ui-ang/fesm2015/backbase-ui-ang-iban-directive.js
import * as i0 from '@angular/core';
import { Directive, NgModule } from '@angular/core';
import { NG_VALIDATORS } from '@angular/forms';
import { isValidIBANFormControl } from '@backbase/ui-ang/iban-lib';

class IbanValidatorDirective {
    // emits invalidIban
    validate(control) {
        return control && control.value ? isValidIBANFormControl(control.value) : null;
    }
}
IbanValidatorDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IbanValidatorDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
IbanValidatorDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: IbanValidatorDirective, selector: "[bbIbanValidatorDir]", providers: [{ provide: NG_VALIDATORS, useExisting: IbanValidatorDirective, multi: true }], ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IbanValidatorDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbIbanValidatorDir]',
                    providers: [{ provide: NG_VALIDATORS, useExisting: IbanValidatorDirective, multi: true }],
                }]
        }] });

class IbanModule {
}
IbanModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IbanModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IbanModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IbanModule, declarations: [IbanValidatorDirective], exports: [IbanValidatorDirective] });
IbanModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IbanModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IbanModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [IbanValidatorDirective],
                    exports: [IbanValidatorDirective],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { IbanModule, IbanValidatorDirective };
