// @backbase/ui-ang/fesm2015/backbase-ui-ang-action-status.js
import * as i0 from '@angular/core';
import { Component, Input, NgModule } from '@angular/core';
import * as i3 from '@angular/common';
import { CommonModule } from '@angular/common';
import * as i1 from '@backbase/ui-ang/loading-indicator';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import * as i2 from '@backbase/ui-ang/icon';
import { IconModule } from '@backbase/ui-ang/icon';

/**
 * @name ActionStatusComponent
 *
 * @description
 * Component that displays a loading indicator or icon to indicate an action response.
 */
class ActionStatusComponent {
    isLoading() {
        return this.state === 'loading';
    }
    getColor() {
        switch (this.state) {
            case 'message':
                return 'info';
            case 'error':
                return 'danger';
            default:
                return this.state;
        }
    }
}
ActionStatusComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ActionStatusComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ActionStatusComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ActionStatusComponent, selector: "bb-action-status-ui", inputs: { state: "state" }, ngImport: i0, template: "<div class=\"bb-action-status\">\n  <bb-loading-indicator-ui *ngIf=\"isLoading()\" loaderSize=\"sm\"></bb-loading-indicator-ui>\n  <bb-icon-ui *ngIf=\"!isLoading() && state\" size=\"md\" [name]=\"state\" [color]=\"getColor()\"></bb-icon-ui>\n</div>\n", components: [{ type: i1.LoadingIndicatorComponent, selector: "bb-loading-indicator-ui", inputs: ["text", "loaderSize", "showDelay", "hasBackground", "inline"] }, { type: i2.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], directives: [{ type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ActionStatusComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-action-status-ui',
                    templateUrl: './action-status.component.html',
                }]
        }], propDecorators: { state: [{
                type: Input
            }] } });

class ActionStatusModule {
}
ActionStatusModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ActionStatusModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ActionStatusModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ActionStatusModule, declarations: [ActionStatusComponent], imports: [CommonModule, LoadingIndicatorModule, IconModule], exports: [ActionStatusComponent] });
ActionStatusModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ActionStatusModule, imports: [[CommonModule, LoadingIndicatorModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ActionStatusModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, LoadingIndicatorModule, IconModule],
                    declarations: [ActionStatusComponent],
                    exports: [ActionStatusComponent],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { ActionStatusComponent, ActionStatusModule };
