// @backbase/ui-ang/fesm2015/backbase-ui-ang-product-item-savings-account.js
import * as i0 from '@angular/core';
import { Directive, Component, ContentChild, Input, NgModule } from '@angular/core';
import * as i5 from '@backbase/ui-ang/payment-card-number-pipe';
import { DefaultPaymentCardNumberFormat, PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import * as i1 from '@backbase/ui-ang/icon';
import { IconModule } from '@backbase/ui-ang/icon';
import * as i2 from '@backbase/ui-ang/amount';
import { AmountModule } from '@backbase/ui-ang/amount';
import * as i3 from '@angular/common';
import { CommonModule } from '@angular/common';
import * as i4 from '@backbase/ui-ang/highlight';
import { HighlightModule } from '@backbase/ui-ang/highlight';

class CustomProductItemSavingsAccountTitleDirective {
    constructor(templateRef) {
        this.templateRef = templateRef;
    }
}
CustomProductItemSavingsAccountTitleDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomProductItemSavingsAccountTitleDirective, deps: [{ token: i0.TemplateRef }], target: i0.ɵɵFactoryTarget.Directive });
CustomProductItemSavingsAccountTitleDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomProductItemSavingsAccountTitleDirective, selector: "ng-template[bbCustomProductItemSavingsAccountTitle]", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomProductItemSavingsAccountTitleDirective, decorators: [{
            type: Directive,
            args: [{ selector: 'ng-template[bbCustomProductItemSavingsAccountTitle]' }]
        }], ctorParameters: function () { return [{ type: i0.TemplateRef }]; } });
class CustomProductItemSavingsAccountDetailsFormatterDirective {
    constructor(templateRef) {
        this.templateRef = templateRef;
    }
}
CustomProductItemSavingsAccountDetailsFormatterDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomProductItemSavingsAccountDetailsFormatterDirective, deps: [{ token: i0.TemplateRef }], target: i0.ɵɵFactoryTarget.Directive });
CustomProductItemSavingsAccountDetailsFormatterDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomProductItemSavingsAccountDetailsFormatterDirective, selector: "ng-template[bbCustomProductItemSavingsAccountDetailsFormatter]", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomProductItemSavingsAccountDetailsFormatterDirective, decorators: [{
            type: Directive,
            args: [{ selector: 'ng-template[bbCustomProductItemSavingsAccountDetailsFormatter]' }]
        }], ctorParameters: function () { return [{ type: i0.TemplateRef }]; } });
class CustomProductItemSavingsAccountMainSectionDirective {
    constructor(templateRef) {
        this.templateRef = templateRef;
    }
}
CustomProductItemSavingsAccountMainSectionDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomProductItemSavingsAccountMainSectionDirective, deps: [{ token: i0.TemplateRef }], target: i0.ɵɵFactoryTarget.Directive });
CustomProductItemSavingsAccountMainSectionDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomProductItemSavingsAccountMainSectionDirective, selector: "ng-template[bbCustomProductItemSavingsAccountMainSection]", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomProductItemSavingsAccountMainSectionDirective, decorators: [{
            type: Directive,
            args: [{ selector: 'ng-template[bbCustomProductItemSavingsAccountMainSection]' }]
        }], ctorParameters: function () { return [{ type: i0.TemplateRef }]; } });
class ProductItemSavingsAccountComponent {
    constructor() {
        this.hostRef = this;
        /**
         * Configuration of how the card number should be formatted.
         */
        this.productNumberFormat = DefaultPaymentCardNumberFormat;
        /**
         * The active state of card (it adds active class for blue line)
         */
        this.active = true;
        /**
         * The favorite state of card (add favorite icon next to title)
         */
        this.favorite = false;
        /**
         * The favorite icon color (define star icon color)
         */
        this.favoriteIconColor = 'primary';
    }
}
ProductItemSavingsAccountComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemSavingsAccountComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ProductItemSavingsAccountComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ProductItemSavingsAccountComponent, selector: "bb-product-item-savings-account-ui", inputs: { title: "title", productNumber: "productNumber", productNumberFormat: "productNumberFormat", currency: "currency", amount: "amount", mainSectionCurrency: "mainSectionCurrency", mainSectionAmount: "mainSectionAmount", mainSectionTitle: "mainSectionTitle", highlight: "highlight", productDetailsInfo: "productDetailsInfo", active: "active", favorite: "favorite", favoriteIconColor: "favoriteIconColor" }, queries: [{ propertyName: "customProductItemSavingsAccountTitle", first: true, predicate: CustomProductItemSavingsAccountTitleDirective, descendants: true, static: true }, { propertyName: "customProductItemSavingsAccountDetailsFormatter", first: true, predicate: CustomProductItemSavingsAccountDetailsFormatterDirective, descendants: true, static: true }, { propertyName: "customProductItemSavingsAccountMainSection", first: true, predicate: CustomProductItemSavingsAccountMainSectionDirective, descendants: true, static: true }], ngImport: i0, template: "<div [class.active]=\"active\" class=\"bb-product-item product-item\">\n  <div class=\"product-item-content\">\n    <div\n      class=\"bb-stack bb-stack--wrap\"\n      [ngClass]=\"{\n        'bb-block bb-block--md': mainSectionAmount\n      }\"\n    >\n      <div class=\"bb-stack__item\">\n        <ng-container *ngIf=\"customProductItemSavingsAccountTitle; else defaultProductItemSavingsAccountTitle\">\n          <ng-container\n            *ngTemplateOutlet=\"customProductItemSavingsAccountTitle?.templateRef; context: hostRef\"\n          ></ng-container>\n        </ng-container>\n      </div>\n      <div class=\"bb-stack__break bb-stack__break--xs\"></div>\n      <div class=\"bb-stack__item bb-stack__item--push-right bb-stack__item--spacing-none bb-product-item__details\">\n        <ng-container\n          *ngIf=\"customProductItemSavingsAccountDetailsFormatter; else defaultProductItemSavingsAccountDetailsFormatter\"\n        >\n          <ng-container\n            *ngTemplateOutlet=\"customProductItemSavingsAccountDetailsFormatter?.templateRef; context: hostRef\"\n          >\n          </ng-container>\n        </ng-container>\n      </div>\n    </div>\n\n    <ng-container *ngIf=\"customProductItemSavingsAccountMainSection; else defaultProductItemSavingsAccountMainSection\">\n      <ng-container *ngTemplateOutlet=\"customProductItemSavingsAccountMainSection?.templateRef; context: hostRef\">\n      </ng-container>\n    </ng-container>\n\n    <div *ngIf=\"productDetailsInfo?.owners\" class=\"bb-product-item__owners\">\n      <span data-role=\"productDetailsInfo-owners\">\n        {{ productDetailsInfo?.owners }}\n      </span>\n    </div>\n  </div>\n</div>\n\n<ng-template #defaultProductItemSavingsAccountTitle>\n  <div class=\"bb-account-info\">\n    <div data-role=\"card-title\" class=\"bb-account-info__title\">\n      {{ title }}\n      <bb-icon-ui\n        *ngIf=\"favorite\"\n        size=\"sm\"\n        [color]=\"favoriteIconColor\"\n        class=\"bb-account-info__icon\"\n        data-role=\"favorite-icon\"\n        name=\"star\"\n      >\n      </bb-icon-ui>\n    </div>\n    <div\n      *ngIf=\"productNumber\"\n      data-role=\"card-sub-title\"\n      class=\"bb-subtitle bb-text-support bb-account-info__product-number\"\n    >\n      {{ productNumber | paymentCardNumber: productNumberFormat }}\n    </div>\n  </div>\n</ng-template>\n\n<ng-template #defaultProductItemSavingsAccountDetailsFormatter>\n  <bb-amount-ui\n    class=\"bb-highlight bb-product-item__amount\"\n    [amount]=\"amount\"\n    [currency]=\"currency\"\n    [bbHighlight]=\"highlight\"\n  ></bb-amount-ui>\n</ng-template>\n\n<ng-template #defaultProductItemSavingsAccountMainSection>\n  <div *ngIf=\"mainSectionAmount\" class=\"bb-stack bb-stack--wrap bb-subtitle bb-text-support\">\n    <span class=\"bb-stack__item bb-product-item__main-section-title\">\n      {{ mainSectionTitle }}\n    </span>\n    <bb-amount-ui\n      [amount]=\"mainSectionAmount\"\n      [currency]=\"mainSectionCurrency\"\n      [bbHighlight]=\"highlight\"\n      class=\"\n        bb-stack__item bb-stack__item--push-right bb-stack__item--spacing-none\n        bb-subtitle\n        bb-product-item__main-section-amount\n      \"\n    ></bb-amount-ui>\n  </div>\n</ng-template>\n", components: [{ type: i1.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i2.AmountComponent, selector: "bb-amount-ui", inputs: ["currency", "showPlusSign", "mapCurrency", "showPercent", "abbreviate", "decimalPlaces", "trailingZeroes", "amount"] }], directives: [{ type: i3.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i3.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }, { type: i4.HighlightDirective, selector: "[bbHighlight]", inputs: ["amount", "bbHighlight", "negativeClass", "positiveClass"] }], pipes: { "paymentCardNumber": i5.PaymentCardNumberPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemSavingsAccountComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-product-item-savings-account-ui',
                    templateUrl: './product-item-savings-account.component.html',
                }]
        }], propDecorators: { customProductItemSavingsAccountTitle: [{
                type: ContentChild,
                args: [CustomProductItemSavingsAccountTitleDirective, { static: true }]
            }], customProductItemSavingsAccountDetailsFormatter: [{
                type: ContentChild,
                args: [CustomProductItemSavingsAccountDetailsFormatterDirective, { static: true }]
            }], customProductItemSavingsAccountMainSection: [{
                type: ContentChild,
                args: [CustomProductItemSavingsAccountMainSectionDirective, { static: true }]
            }], title: [{
                type: Input
            }], productNumber: [{
                type: Input
            }], productNumberFormat: [{
                type: Input
            }], currency: [{
                type: Input
            }], amount: [{
                type: Input
            }], mainSectionCurrency: [{
                type: Input
            }], mainSectionAmount: [{
                type: Input
            }], mainSectionTitle: [{
                type: Input
            }], highlight: [{
                type: Input
            }], productDetailsInfo: [{
                type: Input
            }], active: [{
                type: Input
            }], favorite: [{
                type: Input
            }], favoriteIconColor: [{
                type: Input
            }] } });

class ProductItemSavingsAccountModule {
}
ProductItemSavingsAccountModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemSavingsAccountModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductItemSavingsAccountModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemSavingsAccountModule, declarations: [ProductItemSavingsAccountComponent,
        CustomProductItemSavingsAccountTitleDirective,
        CustomProductItemSavingsAccountDetailsFormatterDirective,
        CustomProductItemSavingsAccountMainSectionDirective], imports: [CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule], exports: [ProductItemSavingsAccountComponent,
        CustomProductItemSavingsAccountTitleDirective,
        CustomProductItemSavingsAccountDetailsFormatterDirective,
        CustomProductItemSavingsAccountMainSectionDirective] });
ProductItemSavingsAccountModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemSavingsAccountModule, imports: [[CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemSavingsAccountModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule],
                    declarations: [
                        ProductItemSavingsAccountComponent,
                        CustomProductItemSavingsAccountTitleDirective,
                        CustomProductItemSavingsAccountDetailsFormatterDirective,
                        CustomProductItemSavingsAccountMainSectionDirective,
                    ],
                    exports: [
                        ProductItemSavingsAccountComponent,
                        CustomProductItemSavingsAccountTitleDirective,
                        CustomProductItemSavingsAccountDetailsFormatterDirective,
                        CustomProductItemSavingsAccountMainSectionDirective,
                    ],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { CustomProductItemSavingsAccountDetailsFormatterDirective, CustomProductItemSavingsAccountMainSectionDirective, CustomProductItemSavingsAccountTitleDirective, ProductItemSavingsAccountComponent, ProductItemSavingsAccountModule };
