// @backbase/ui-ang/esm2015/_example/currency-input/examples/currency-input.example.component.js
import { Component } from '@angular/core';
import { FormControl } from '@angular/forms';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/currency-input";
import * as i2 from "@angular/forms";
export class CurrencyInputExampleComponent {
    constructor() {
        this.amount = new FormControl({
            amount: '123.32',
            currency: 'EUR',
        });
        this.currencies = ['USD', 'EUR'];
    }
}
CurrencyInputExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CurrencyInputExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
CurrencyInputExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: CurrencyInputExampleComponent, selector: "bb-currency-input-example-ui", ngImport: i0, template: "<bb-currency-input-ui\n  [formControl]=\"amount\"\n  [currencies]=\"currencies\"\n  [allowNegativeValue]=\"true\"\n  resultType=\"string\"\n></bb-currency-input-ui>\n", components: [{ type: i1.CurrencyInputComponent, selector: "bb-currency-input-ui", inputs: ["decMaxLen", "decMinLen", "placeholder", "resultType", "integer", "required", "readonly", "disabled", "integerLabelSrOnly", "decimalLabelSrOnly", "currencies", "isRightAligned", "integerId", "decimalId", "integerLabel", "decimalLabel", "currencyLabel", "allowNegativeValue", "currencyListAriaLabel", "currencyListWithOnItemAriaLabel", "aria-describedby", "aria-labelledby", "aria-invalid"], outputs: ["blur"] }], directives: [{ type: i2.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { type: i2.FormControlDirective, selector: "[formControl]", inputs: ["disabled", "formControl", "ngModel"], outputs: ["ngModelChange"], exportAs: ["ngForm"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CurrencyInputExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-currency-input-example-ui',
                    templateUrl: './currency-input.example.component.html',
                }]
        }] });
