// @backbase/consent-details-widget-ang/esm2015/src/consent-details-widget/consent-details.service.js
import { Injectable } from '@angular/core';
import { BehaviorSubject, combineLatest, of, ReplaySubject } from 'rxjs';
import { catchError, distinctUntilChanged, filter, pluck, switchMap, take, tap } from 'rxjs/operators';
import { parseError } from './model/consent-details-error';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/consent";
import * as i2 from "@backbase/ui-ang/notification";
export class ConsentDetailsService {
    constructor(consentDataService, notification) {
        this.consentDataService = consentDataService;
        this.notification = notification;
        this.consentId = new ReplaySubject(1);
        this.revokingComplete = new BehaviorSubject(true);
        this.loading = new BehaviorSubject(true);
        this.error = new BehaviorSubject(undefined);
        this.consentDetails = combineLatest([this.consentId, this.revokingComplete]).pipe(filter(([id]) => !!id), distinctUntilChanged(), tap(() => this.loading.next(true)), switchMap(([id]) => this.getConsentDetailsById(id)), tap(() => this.error.next(undefined)), catchError((error) => {
            this.error.next(error);
            return of(undefined);
        }), tap(() => this.loading.next(false)));
    }
    getConsentFrom(consentId) {
        consentId.subscribe(this.consentId);
    }
    requestRevoke(errorTemplateRef, successTemplateRef) {
        return this.consentId.pipe(take(1), filter((id) => !!id), switchMap((id) => this.revokeConsentById(id)), tap(() => {
            this.revokingComplete.next(true);
            this.showNotification(successTemplateRef, 'success');
        }), catchError(() => {
            this.showNotification(errorTemplateRef, 'error');
            return of(undefined);
        }));
    }
    getConsentDetailsById(id) {
        return this.consentDataService.getIdByIdgetId({ id }, 'response').pipe(pluck('body'), catchError((error) => {
            throw parseError(error);
        }));
    }
    revokeConsentById(id) {
        return this.consentDataService.postRevokeById({ id }, 'response').pipe(catchError((error) => {
            throw parseError(error);
        }));
    }
    showNotification(templateRef, modifier) {
        this.notification.showNotification({
            header: templateRef,
            message: '',
            modifier,
        });
    }
}
ConsentDetailsService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentDetailsService, deps: [{ token: i1.ConsentsHttpService }, { token: i2.NotificationService }], target: i0.ɵɵFactoryTarget.Injectable });
ConsentDetailsService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentDetailsService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentDetailsService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.ConsentsHttpService }, { type: i2.NotificationService }]; } });
