// @backbase/ui-ang/esm2015/rich-text-editor/bb-rich-text-editor-actions/rich-text-editor-link-action/rich-text-editor-link-action.component.js
import { Component, EventEmitter, Inject, Input, Output } from '@angular/core';
import { RichTextEditorActionsComponent } from '../rich-text-editor-actions.component';
import { DOCUMENT } from '@angular/common';
import * as i0 from "@angular/core";
import * as i1 from "../rich-text-editor-actions.service";
import * as i2 from "@backbase/ui-ang/icon";
import * as i3 from "@backbase/ui-ang/modal";
import * as i4 from "@backbase/ui-ang/tooltip-directive";
import * as i5 from "@backbase/ui-ang/button";
import * as i6 from "@angular/forms";
/**
 * @name RichTextEditorLinkActionComponent
 *
 * @dynamic (to suppress error with resolving statics declarations during compilation)
 */
export class RichTextEditorLinkActionComponent extends RichTextEditorActionsComponent {
    constructor(ma, document) {
        super(ma);
        this.document = document;
        /**
         * Event emitted after link was added
         */
        this.linkAdded = new EventEmitter();
        /**
         * Flag to determine whether the add link modal is open
         */
        this.modalOpened = false;
        /**
         * Selection link url
         */
        this.selectionUrl = '';
    }
    /**
     * Open add link modal
     */
    openModal() {
        var _a, _b, _c, _d;
        this.modalOpened = true;
        this.setSelectionData();
        if (((_a = this.selection) === null || _a === void 0 ? void 0 : _a.rangeCount) && ((_b = this.selection) === null || _b === void 0 ? void 0 : _b.rangeCount) > 0) {
            this.selectionUrl =
                ((_c = this.linkedSelection) === null || _c === void 0 ? void 0 : _c.nodeName) === 'A' && ((_d = this.linkedSelection) === null || _d === void 0 ? void 0 : _d.href) !== ''
                    ? this.linkedSelection.href
                    : 'http://';
        }
    }
    /**
     * On add link modal confirmation
     */
    onResult() {
        this.restoreSelection();
        this.selectionUrl = this.selectionUrl.replace(/\s/g, '');
        if (this.selectionUrl.length === 0) {
            this.document.execCommand('unlink', false);
        }
        else {
            this.insertLink();
        }
        this.closeModal();
    }
    /**
     * Close add link modal
     */
    closeModal() {
        this.modalOpened = false;
        this.linkAdded.emit();
    }
    setSelectionData() {
        var _a;
        this.selection = this.document.getSelection();
        this.range = (_a = this.selection) === null || _a === void 0 ? void 0 : _a.getRangeAt(0);
        this.selectionUrl = '';
    }
    restoreSelection() {
        var _a, _b;
        if (this.range) {
            (_a = this.selection) === null || _a === void 0 ? void 0 : _a.removeAllRanges();
            (_b = this.selection) === null || _b === void 0 ? void 0 : _b.addRange(this.range);
        }
    }
    insertLink() {
        var _a;
        if (!/^(http:\/\/|https:\/\/)/.test(this.selectionUrl)) {
            this.selectionUrl = 'http://' + this.selectionUrl;
        }
        if (((_a = this.range) === null || _a === void 0 ? void 0 : _a.toString()) !== '') {
            this.document.execCommand('createLink', false, this.selectionUrl);
        }
        else {
            this.document.execCommand('insertHTML', false, this.linkElement);
        }
    }
    get linkElement() {
        var _a;
        const innerText = ((_a = this.selection) === null || _a === void 0 ? void 0 : _a.toString()) || this.selectionUrl;
        const element = this.document.createElement('a');
        element.setAttribute('href', this.selectionUrl);
        element.setAttribute('target', '_blank');
        element.innerHTML = innerText;
        return element.outerHTML;
    }
    get linkedSelection() {
        var _a, _b, _c, _d, _e, _f, _g;
        return (((_c = (_b = (_a = this.range) === null || _a === void 0 ? void 0 : _a.commonAncestorContainer) === null || _b === void 0 ? void 0 : _b.parentElement) === null || _c === void 0 ? void 0 : _c.closest('a')) ||
            ((_g = (_f = (_e = (_d = this.range) === null || _d === void 0 ? void 0 : _d.commonAncestorContainer) === null || _e === void 0 ? void 0 : _e.children) === null || _f === void 0 ? void 0 : _f[0]) === null || _g === void 0 ? void 0 : _g.closest('a')));
    }
}
RichTextEditorLinkActionComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RichTextEditorLinkActionComponent, deps: [{ token: i1.RichTextEditorActionsService }, { token: DOCUMENT }], target: i0.ɵɵFactoryTarget.Component });
RichTextEditorLinkActionComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: RichTextEditorLinkActionComponent, selector: "bb-rich-text-editor-link-action", inputs: { contentClassNames: "contentClassNames" }, outputs: { linkAdded: "linkAdded" }, usesInheritance: true, ngImport: i0, template: "<span\n  bbTooltip=\"Link\"\n  i18n-bbTooltip=\"Link|Link action @@bb-rich-text-editor-ui.rich-text-editor-actions.link\"\n  container=\"body\"\n  triggers=\"hover focus\"\n  [placement]=\"actionsTooltipPlacement\"\n>\n  <button\n    bbButton\n    [circle]=\"true\"\n    color=\"link-dark\"\n    aria-label=\"Link\"\n    i18n-aria-label=\"@@bb-rich-text-editor-ui.rich-text-editor-actions.link\"\n    [disabled]=\"disabled\"\n    (click)=\"openModal()\"\n    data-role=\"link-action\"\n  >\n    <bb-icon-ui name=\"link\"></bb-icon-ui>\n  </button>\n</span>\n\n<bb-modal-ui [isOpen]=\"modalOpened\" (cancel)=\"closeModal()\" (confirm)=\"onResult()\">\n  <form data-role=\"add-link-form\">\n    <bb-modal-header-ui\n      title=\"Add link\"\n      i18n-title=\"Add link|Add link title@@bb-rich-text-editor-ui.rich-text-editor-link-modal.title\"\n    ></bb-modal-header-ui>\n    <bb-modal-body-ui>\n      <ng-template bbCustomModalBody>\n        <label\n          class=\"bb-label\"\n          for=\"link-input\"\n          i18n=\"Enter the link|Enter the link title@@bb-rich-text-editor-ui.rich-text-editor-link-modal.input-label\"\n        >\n          Enter the URL Link\n        </label>\n        <input\n          type=\"text\"\n          class=\"form-control\"\n          [(ngModel)]=\"selectionUrl\"\n          name=\"url\"\n          id=\"link-input\"\n          data-role=\"link-input\"\n        />\n      </ng-template>\n    </bb-modal-body-ui>\n    <bb-modal-footer-ui #footerRef data-role=\"link-modal-footer\">\n      <ng-template bbCustomModalFooter>\n        <div class=\"bb-button-bar\">\n          <button\n            bbButton\n            class=\"bb-button-bar__button\"\n            (click)=\"onResult()\"\n            type=\"submit\"\n            data-role=\"save-link\"\n            i18n=\"Save a link|Save a link button@@bb-rich-text-editor-ui.rich-text-editor-link-modal.save-button\"\n          >\n            Save Link\n          </button>\n          <button\n            bbButton\n            class=\"bb-button-bar__button\"\n            (click)=\"closeModal()\"\n            color=\"secondary\"\n            data-role=\"cancel-link-saving\"\n            i18n=\"Cancel|Cancel button@@bb-rich-text-editor-ui.rich-text-editor-link-modal.cancel-button\"\n          >\n            Cancel\n          </button>\n        </div>\n      </ng-template>\n    </bb-modal-footer-ui>\n  </form>\n</bb-modal-ui>\n", components: [{ type: i2.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i3.ModalComponent, selector: "bb-modal-ui", inputs: ["isOpen", "aria-labelledby", "aria-describedby", "modalOptions"], outputs: ["confirm", "cancel", "isOpenChange"] }, { type: i3.ModalHeaderComponent, selector: "bb-modal-header-ui", inputs: ["title"], outputs: ["close"] }, { type: i3.ModalBodyComponent, selector: "bb-modal-body-ui" }, { type: i3.ModalFooterComponent, selector: "bb-modal-footer-ui", inputs: ["confirmText", "cancelText"], outputs: ["confirm", "cancel"] }], directives: [{ type: i4.TooltipDirective, selector: "[bbTooltip]", inputs: ["bbTooltip", "triggers"] }, { type: i5.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }, { type: i6.ɵNgNoValidate, selector: "form:not([ngNoForm]):not([ngNativeValidate])" }, { type: i6.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { type: i6.NgForm, selector: "form:not([ngNoForm]):not([formGroup]),ng-form,[ngForm]", inputs: ["ngFormOptions"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { type: i3.CustomModalBodyDirective, selector: "ng-template[bbCustomModalBody]" }, { type: i6.DefaultValueAccessor, selector: "input:not([type=checkbox])[formControlName],textarea[formControlName],input:not([type=checkbox])[formControl],textarea[formControl],input:not([type=checkbox])[ngModel],textarea[ngModel],[ngDefaultControl]" }, { type: i6.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { type: i6.NgModel, selector: "[ngModel]:not([formControlName]):not([formControl])", inputs: ["name", "disabled", "ngModel", "ngModelOptions"], outputs: ["ngModelChange"], exportAs: ["ngModel"] }, { type: i3.CustomModalFooterDirective, selector: "ng-template[bbCustomModalFooter]" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RichTextEditorLinkActionComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-rich-text-editor-link-action',
                    templateUrl: './rich-text-editor-link-action.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.RichTextEditorActionsService }, { type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }]; }, propDecorators: { linkAdded: [{
                type: Output
            }], contentClassNames: [{
                type: Input
            }] } });
