// @backbase/user-identity-security-center-widget-ang/esm2015/src/change-password-form/change-password-form.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { AlertModule } from '@backbase/ui-ang/alert';
import { ButtonModule } from '@backbase/ui-ang/button';
import { ErrorCommonStateModule } from '@backbase/ui-ang/common-error-state';
import { HeaderModule } from '@backbase/ui-ang/header';
import { InputPasswordModule } from '@backbase/ui-ang/input-password';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { ModalModule } from '@backbase/ui-ang/modal';
import { ChangePasswordFormComponent, UserIdentitySecurityCenterPasswordFormCustomizableDirective, UserIdentitySecurityCenterPasswordFormPasswordPolicyCustomizableDirective, } from './change-password-form.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const passwordForm = [
    ChangePasswordFormComponent,
    UserIdentitySecurityCenterPasswordFormPasswordPolicyCustomizableDirective,
    UserIdentitySecurityCenterPasswordFormCustomizableDirective,
];
const uiModules = [
    AlertModule,
    ButtonModule,
    ErrorCommonStateModule,
    HeaderModule,
    LoadingIndicatorModule,
    ModalModule,
    InputPasswordModule,
    InputValidationMessageModule,
    ReactiveFormsModule,
    LoadButtonModule,
];
export class ChangePasswordFormModule {
}
ChangePasswordFormModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ChangePasswordFormModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ChangePasswordFormModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ChangePasswordFormModule, declarations: [ChangePasswordFormComponent,
        UserIdentitySecurityCenterPasswordFormPasswordPolicyCustomizableDirective,
        UserIdentitySecurityCenterPasswordFormCustomizableDirective], imports: [CommonModule, i1.BackbaseCoreModule, AlertModule,
        ButtonModule,
        ErrorCommonStateModule,
        HeaderModule,
        LoadingIndicatorModule,
        ModalModule,
        InputPasswordModule,
        InputValidationMessageModule,
        ReactiveFormsModule,
        LoadButtonModule], exports: [AlertModule,
        ButtonModule,
        ErrorCommonStateModule,
        HeaderModule,
        LoadingIndicatorModule,
        ModalModule,
        InputPasswordModule,
        InputValidationMessageModule,
        ReactiveFormsModule,
        LoadButtonModule, ChangePasswordFormComponent,
        UserIdentitySecurityCenterPasswordFormPasswordPolicyCustomizableDirective,
        UserIdentitySecurityCenterPasswordFormCustomizableDirective] });
ChangePasswordFormModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ChangePasswordFormModule, imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { ChangePasswordFormComponent: ChangePasswordFormComponent },
            }),
            ...uiModules,
        ], AlertModule,
        ButtonModule,
        ErrorCommonStateModule,
        HeaderModule,
        LoadingIndicatorModule,
        ModalModule,
        InputPasswordModule,
        InputValidationMessageModule,
        ReactiveFormsModule,
        LoadButtonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ChangePasswordFormModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { ChangePasswordFormComponent: ChangePasswordFormComponent },
                        }),
                        ...uiModules,
                    ],
                    declarations: [...passwordForm],
                    exports: [...uiModules, ...passwordForm],
                }]
        }] });
