// @backbase/accessgroup-ui/esm2015/lib/accessgroup-ui.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { ButtonModule } from '@backbase/ui-ang/button';
import { HeaderModule } from '@backbase/ui-ang/header';
import { IconModule } from '@backbase/ui-ang/icon';
import { ModalModule } from '@backbase/ui-ang/modal';
import { TableModule } from '@backbase/ui-ang/table';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { KeyboardClickModule } from '@backbase/ui-ang/keyboard-click-directive';
import { InputCheckboxModule } from '@backbase/ui-ang/input-checkbox';
import { CollapsibleModule } from '@backbase/ui-ang/collapsible';
import { EllipsisModule } from '@backbase/ui-ang/ellipsis';
import { PaginationModule } from '@backbase/ui-ang/pagination';
import { BadgeModule } from '@backbase/ui-ang/badge';
import { AmountModule } from '@backbase/ui-ang/amount';
import { AvatarModule } from '@backbase/ui-ang/avatar';
import { CurrencyInputModule } from '@backbase/ui-ang/currency-input';
import { DropdownSingleSelectModule } from '@backbase/ui-ang/dropdown-single-select';
import { SwitchModule } from '@backbase/ui-ang/switch';
import { AccessgroupErrorComponent } from './accessgroup-error/component/accessgroup-error.component';
import { AccessgroupHeaderComponent } from './accessgroup-header/accessgroup-header.component';
import { AccessgroupApprovalLogComponent } from './accessgroup-approval-log/accessgroup-approval-log.component';
import { BusinessFunctionGroupNameComponent } from './business-function-group-name/business-function-group-name.component';
import { BusinessFunctionNameComponent } from './business-function-name/business-function-name.component';
import { BusinessFunctionPrivilegesNameComponent } from './business-function-privileges-name/business-function-privileges-name.component';
import { DataGroupItemsTableComponent } from './data-group-items-table/data-group-items-table.component';
import { DiscardChangesModalComponent } from './discard-changes-modal/discard-changes-modal.component';
import { FunctionGroupTimeBoundsComponent } from './function-group-time-bounds/function-group-time-bounds.component';
import { PrivilegeIndicatorListComponent } from './privilege-indicator-list/privilege-indicator-list.component';
import { PrivilegeIndicatorComponent } from './privilege-indicator/privilege-indicator.component';
import { SelectItemsModalModule } from './select-items-modal/select-items-modal.module';
import { TimeBoundsComponent } from './time-bounds/time-bounds.component';
import { UserContextIconComponent } from './user-context-icon/user-context-icon.component';
import { UserPermissionsItemBusinessFunctionDirective, UserPermissionsItemBusinessFunctionGroupNameDirective, UserPermissionsItemComponent, } from './user-permissions-item/user-permissions-item.component';
import { PayeeGroupTableRowComponent } from './payee-group-table-row/payee-group-table-row.component';
import { PayeeItemDetailComponent, PayeeItemDetailCustomizableDirective, } from './payee-item-detail/payee-item-detail.component';
import { UnassignDataGroupModalComponent, UnassignDataGroupModalCustomizableDirective, } from './unassign-data-group/unassign-data-group-modal.component';
import { PayeeGroupDetailModalComponent, PayeeGroupDetailModalCustomizableDirective, } from './payee-group-detail-modal/payee-group-detail-modal.component';
import { AccountGroupDetailModalComponent, AccountGroupDetailModalCustomizableDirective, } from './account-group-detail-modal/account-group-detail-modal.component';
import { SelfApproveModalComponent, SelfApproveModalCustomizableDirective, } from './self-approval-modal/self-approve-modal.component';
import * as i0 from "@angular/core";
const uiModules = [
    IconModule,
    TooltipModule,
    SelectItemsModalModule,
    ButtonModule,
    ModalModule,
    TableModule,
    HeaderModule,
    KeyboardClickModule,
    InputCheckboxModule,
    CollapsibleModule,
    EllipsisModule,
    BadgeModule,
    PaginationModule,
    SwitchModule,
    AmountModule,
    AvatarModule,
    CurrencyInputModule,
    DropdownSingleSelectModule,
];
const directives = [
    UserPermissionsItemBusinessFunctionDirective,
    UserPermissionsItemBusinessFunctionGroupNameDirective,
];
const uiComponents = [
    BusinessFunctionGroupNameComponent,
    BusinessFunctionNameComponent,
    PrivilegeIndicatorComponent,
    UserContextIconComponent,
    DiscardChangesModalComponent,
    DataGroupItemsTableComponent,
    PrivilegeIndicatorListComponent,
    BusinessFunctionPrivilegesNameComponent,
    UserPermissionsItemComponent,
    FunctionGroupTimeBoundsComponent,
    AccessgroupErrorComponent,
    AccessgroupHeaderComponent,
    AccessgroupApprovalLogComponent,
    TimeBoundsComponent,
    PayeeGroupTableRowComponent,
    PayeeItemDetailComponent,
    PayeeItemDetailCustomizableDirective,
    UnassignDataGroupModalComponent,
    UnassignDataGroupModalCustomizableDirective,
    PayeeGroupDetailModalComponent,
    PayeeGroupDetailModalCustomizableDirective,
    AccountGroupDetailModalComponent,
    AccountGroupDetailModalCustomizableDirective,
    SelfApproveModalComponent,
    SelfApproveModalCustomizableDirective,
];
const formsModules = [FormsModule, ReactiveFormsModule];
export class AccessgroupUiModule {
}
/** @nocollapse */ AccessgroupUiModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccessgroupUiModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ AccessgroupUiModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccessgroupUiModule, declarations: [BusinessFunctionGroupNameComponent,
        BusinessFunctionNameComponent,
        PrivilegeIndicatorComponent,
        UserContextIconComponent,
        DiscardChangesModalComponent,
        DataGroupItemsTableComponent,
        PrivilegeIndicatorListComponent,
        BusinessFunctionPrivilegesNameComponent,
        UserPermissionsItemComponent,
        FunctionGroupTimeBoundsComponent,
        AccessgroupErrorComponent,
        AccessgroupHeaderComponent,
        AccessgroupApprovalLogComponent,
        TimeBoundsComponent,
        PayeeGroupTableRowComponent,
        PayeeItemDetailComponent,
        PayeeItemDetailCustomizableDirective,
        UnassignDataGroupModalComponent,
        UnassignDataGroupModalCustomizableDirective,
        PayeeGroupDetailModalComponent,
        PayeeGroupDetailModalCustomizableDirective,
        AccountGroupDetailModalComponent,
        AccountGroupDetailModalCustomizableDirective,
        SelfApproveModalComponent,
        SelfApproveModalCustomizableDirective, UserPermissionsItemBusinessFunctionDirective,
        UserPermissionsItemBusinessFunctionGroupNameDirective], imports: [CommonModule, IconModule,
        TooltipModule,
        SelectItemsModalModule,
        ButtonModule,
        ModalModule,
        TableModule,
        HeaderModule,
        KeyboardClickModule,
        InputCheckboxModule,
        CollapsibleModule,
        EllipsisModule,
        BadgeModule,
        PaginationModule,
        SwitchModule,
        AmountModule,
        AvatarModule,
        CurrencyInputModule,
        DropdownSingleSelectModule, FormsModule, ReactiveFormsModule], exports: [BusinessFunctionGroupNameComponent,
        BusinessFunctionNameComponent,
        PrivilegeIndicatorComponent,
        UserContextIconComponent,
        DiscardChangesModalComponent,
        DataGroupItemsTableComponent,
        PrivilegeIndicatorListComponent,
        BusinessFunctionPrivilegesNameComponent,
        UserPermissionsItemComponent,
        FunctionGroupTimeBoundsComponent,
        AccessgroupErrorComponent,
        AccessgroupHeaderComponent,
        AccessgroupApprovalLogComponent,
        TimeBoundsComponent,
        PayeeGroupTableRowComponent,
        PayeeItemDetailComponent,
        PayeeItemDetailCustomizableDirective,
        UnassignDataGroupModalComponent,
        UnassignDataGroupModalCustomizableDirective,
        PayeeGroupDetailModalComponent,
        PayeeGroupDetailModalCustomizableDirective,
        AccountGroupDetailModalComponent,
        AccountGroupDetailModalCustomizableDirective,
        SelfApproveModalComponent,
        SelfApproveModalCustomizableDirective, IconModule,
        TooltipModule,
        SelectItemsModalModule,
        ButtonModule,
        ModalModule,
        TableModule,
        HeaderModule,
        KeyboardClickModule,
        InputCheckboxModule,
        CollapsibleModule,
        EllipsisModule,
        BadgeModule,
        PaginationModule,
        SwitchModule,
        AmountModule,
        AvatarModule,
        CurrencyInputModule,
        DropdownSingleSelectModule, UserPermissionsItemBusinessFunctionDirective,
        UserPermissionsItemBusinessFunctionGroupNameDirective, FormsModule, ReactiveFormsModule] });
/** @nocollapse */ AccessgroupUiModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccessgroupUiModule, imports: [[CommonModule, ...uiModules, ...formsModules], IconModule,
        TooltipModule,
        SelectItemsModalModule,
        ButtonModule,
        ModalModule,
        TableModule,
        HeaderModule,
        KeyboardClickModule,
        InputCheckboxModule,
        CollapsibleModule,
        EllipsisModule,
        BadgeModule,
        PaginationModule,
        SwitchModule,
        AmountModule,
        AvatarModule,
        CurrencyInputModule,
        DropdownSingleSelectModule, FormsModule, ReactiveFormsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccessgroupUiModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules, ...formsModules],
                    declarations: [...uiComponents, ...directives],
                    exports: [...uiComponents, ...uiModules, ...directives, ...formsModules],
                }]
        }] });
