// @backbase/loans-collection-ui-ang/esm2015/lib/heading/loans-heading.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { ButtonModule } from '@backbase/ui-ang/button';
import { HeaderModule } from '@backbase/ui-ang/header';
import { LoansHeadingComponent } from './loans-heading.component';
import * as i0 from "@angular/core";
export class LoansHeadingModule {
}
/** @nocollapse */ LoansHeadingModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoansHeadingModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoansHeadingModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoansHeadingModule, declarations: [LoansHeadingComponent], imports: [CommonModule, RouterModule, HeaderModule, ButtonModule], exports: [LoansHeadingComponent] });
/** @nocollapse */ LoansHeadingModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoansHeadingModule, imports: [[CommonModule, RouterModule, HeaderModule, ButtonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoansHeadingModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoansHeadingComponent],
                    imports: [CommonModule, RouterModule, HeaderModule, ButtonModule],
                    exports: [LoansHeadingComponent],
                }]
        }] });
