// @backbase/loans-collection-ui-ang/esm2015/lib/heading/loans-heading.component.js
import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/header";
import * as i2 from "@angular/common";
export class LoansHeadingComponent {
    constructor() {
        /**
         * Heading navigation button press action
         */
        this.navigationButtonAction = new EventEmitter();
        /**
         * Heading main button press action
         */
        this.mainButtonAction = new EventEmitter();
    }
}
/** @nocollapse */ LoansHeadingComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoansHeadingComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ LoansHeadingComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: LoansHeadingComponent, selector: "bb-loans-heading", inputs: { headingTitle: "headingTitle", mainButtonTitle: "mainButtonTitle", showMainButton: "showMainButton", showNavigationButton: "showNavigationButton", navigationButtonTitle: "navigationButtonTitle" }, outputs: { navigationButtonAction: "navigationButtonAction", mainButtonAction: "mainButtonAction" }, ngImport: i0, template: "<div\n  class=\"bb-heading-widget bb-heading-widget--de-elevated\"\n  data-role=\"loans-heading\"\n  [class.bb-heading-widget--button-above]=\"showNavigationButton\"\n>\n  <bb-header-ui\n    headingType=\"h1\"\n    headingClasses=\"bb-heading-widget__heading text-white\"\n    [heading]=\"headingTitle\"\n  ></bb-header-ui>\n\n  <button\n    *ngIf=\"showNavigationButton\"\n    data-role=\"loans-heading-button__navigation\"\n    (click)=\"navigationButtonAction.emit()\"\n    class=\"bb-heading-widget__button btn btn-link-text-light btn-sm bb-block bb-block--xl\"\n  >\n    <i aria-hidden=\"true\" class=\"bb-icon bb-icon-arrow-back bb-icon--sm\"></i>\n    <span>{{ navigationButtonTitle }}</span>\n  </button>\n\n  <button\n    *ngIf=\"showMainButton\"\n    data-role=\"loans-heading-button__main\"\n    (click)=\"mainButtonAction.emit()\"\n    class=\"bb-heading-widget__button btn btn-primary\"\n  >\n    <i aria-hidden=\"true\" class=\"bb-icon bb-icon-add\"></i>\n    <span>{{ mainButtonTitle }}</span>\n  </button>\n</div>\n", components: [{ type: i1.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }], directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoansHeadingComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-loans-heading',
                    templateUrl: './loans-heading.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { headingTitle: [{
                type: Input
            }], mainButtonTitle: [{
                type: Input
            }], showMainButton: [{
                type: Input
            }], showNavigationButton: [{
                type: Input
            }], navigationButtonTitle: [{
                type: Input
            }], navigationButtonAction: [{
                type: Output
            }], mainButtonAction: [{
                type: Output
            }] } });
