// @backbase/authorized-users-journey-ang/esm2015/src/common/authorized-users-common.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { IconModule } from '@backbase/ui-ang/icon';
import { PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import { TableModule } from '@backbase/ui-ang/table';
import { AuthorizedUsersPermissionsLevelsTableComponent } from './components/permissions-table/permissions-table.component';
import { UserDataService } from './services/user-data.service';
import { UsersUIService } from './services/users-ui.service';
import { AuthorizedUsersAccessibilityKeyboardDirective } from './utils/accessibility-keyboard.directive';
import * as i0 from "@angular/core";
const modules = [IconModule, TableModule, PaymentCardNumberModule];
const declarations = [AuthorizedUsersPermissionsLevelsTableComponent, AuthorizedUsersAccessibilityKeyboardDirective];
export class AuthorizedUsersCommonModule {
}
AuthorizedUsersCommonModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersCommonModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AuthorizedUsersCommonModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersCommonModule, declarations: [AuthorizedUsersPermissionsLevelsTableComponent, AuthorizedUsersAccessibilityKeyboardDirective], imports: [CommonModule, IconModule, TableModule, PaymentCardNumberModule], exports: [IconModule, TableModule, PaymentCardNumberModule, AuthorizedUsersPermissionsLevelsTableComponent, AuthorizedUsersAccessibilityKeyboardDirective] });
AuthorizedUsersCommonModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersCommonModule, providers: [UserDataService, UsersUIService], imports: [[CommonModule, ...modules], IconModule, TableModule, PaymentCardNumberModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersCommonModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...modules],
                    providers: [UserDataService, UsersUIService],
                    declarations: [...declarations],
                    exports: [...modules, ...declarations],
                }]
        }] });
