// @backbase/ui-ang/esm2015/_example/input-password/showcase/input-password-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { InputPasswordShowcaseComponent } from './input-password-showcase.component';
import { InputPasswordModule } from '@backbase/ui-ang/input-password';
import { ReactiveFormsModule } from '@angular/forms';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import * as i0 from "@angular/core";
export class InputPasswordShowcaseModule {
}
InputPasswordShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputPasswordShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputPasswordShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputPasswordShowcaseModule, declarations: [InputPasswordShowcaseComponent], imports: [CommonModule, InputPasswordModule, ReactiveFormsModule, InputValidationMessageModule] });
InputPasswordShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputPasswordShowcaseModule, imports: [[CommonModule, InputPasswordModule, ReactiveFormsModule, InputValidationMessageModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputPasswordShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, InputPasswordModule, ReactiveFormsModule, InputValidationMessageModule],
                    declarations: [InputPasswordShowcaseComponent],
                    entryComponents: [InputPasswordShowcaseComponent],
                }]
        }] });
