// @backbase/payments-collection-ui-ang/esm2015/lib/containers/initiate-payment/components/review-payment-details/review-payment.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PaymentOrdersUiModule } from '@backbase/payment-orders-ang';
import { ReviewPaymentContainerComponent } from './review-payment.component';
import { AdaptedReviewComponent } from '../adapted-review/adapted-review-container/adapted-review.component';
import { CounterpartyDetailsComponent } from '../adapted-review/counterparty-details/counterparty-details.component';
import { CreditorAccountComponent } from '../adapted-review/creditor-account/creditor-account.component';
import { CreditorDetailsComponent } from '../adapted-review/creditor-details/creditor-details.component';
import { InitiatorDetailsComponent } from '../adapted-review/initiator-details/initiator-details.component';
import { RemittanceDetailsComponent } from '../adapted-review/remittance-details/remittance-details.component';
import { ScheduleDetailsComponent } from '../adapted-review/schedule-details/schedule-details.component';
import { ModalModule } from '@backbase/ui-ang/modal';
import { DropdownMenuModule } from '@backbase/ui-ang/dropdown-menu';
import { KeyboardClickModule } from '@backbase/ui-ang/keyboard-click-directive';
import { NotificationModule } from '@backbase/ui-ang/notification';
import { AlertModule } from '@backbase/ui-ang/alert';
import { HeaderModule } from '@backbase/ui-ang/header';
import { FocusModule } from '@backbase/ui-ang/focus';
import * as i0 from "@angular/core";
const components = [
    ReviewPaymentContainerComponent,
    RemittanceDetailsComponent,
    CounterpartyDetailsComponent,
    InitiatorDetailsComponent,
    ScheduleDetailsComponent,
    CreditorAccountComponent,
    CreditorDetailsComponent,
    AdaptedReviewComponent,
];
const uiAngComponentModules = [
    ModalModule,
    DropdownMenuModule,
    KeyboardClickModule,
    NotificationModule,
    AlertModule,
    HeaderModule,
    FocusModule,
];
export class ReviewPaymentModule {
}
/** @nocollapse */ ReviewPaymentModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ReviewPaymentModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ ReviewPaymentModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ReviewPaymentModule, declarations: [ReviewPaymentContainerComponent,
        RemittanceDetailsComponent,
        CounterpartyDetailsComponent,
        InitiatorDetailsComponent,
        ScheduleDetailsComponent,
        CreditorAccountComponent,
        CreditorDetailsComponent,
        AdaptedReviewComponent], imports: [CommonModule, PaymentOrdersUiModule, ModalModule,
        DropdownMenuModule,
        KeyboardClickModule,
        NotificationModule,
        AlertModule,
        HeaderModule,
        FocusModule], exports: [ReviewPaymentContainerComponent,
        RemittanceDetailsComponent,
        CounterpartyDetailsComponent,
        InitiatorDetailsComponent,
        ScheduleDetailsComponent,
        CreditorAccountComponent,
        CreditorDetailsComponent,
        AdaptedReviewComponent, ModalModule,
        DropdownMenuModule,
        KeyboardClickModule,
        NotificationModule,
        AlertModule,
        HeaderModule,
        FocusModule] });
/** @nocollapse */ ReviewPaymentModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ReviewPaymentModule, imports: [[CommonModule, PaymentOrdersUiModule, ...uiAngComponentModules], ModalModule,
        DropdownMenuModule,
        KeyboardClickModule,
        NotificationModule,
        AlertModule,
        HeaderModule,
        FocusModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ReviewPaymentModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, PaymentOrdersUiModule, ...uiAngComponentModules],
                    exports: [...components, ...uiAngComponentModules],
                    declarations: [...components],
                }]
        }] });
