// @backbase/ui-ang/esm2015/input-password/input-password.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { InputPasswordComponent } from './input-password.component';
import { BaseClassesModule } from '@backbase/ui-ang/base-classes';
import { FocusModule } from '@backbase/ui-ang/focus';
import { IconModule } from '@backbase/ui-ang/icon';
import { ButtonModule } from '@backbase/ui-ang/button';
import * as i0 from "@angular/core";
export class InputPasswordModule {
}
InputPasswordModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputPasswordModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputPasswordModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputPasswordModule, declarations: [InputPasswordComponent], imports: [CommonModule, FormsModule, BaseClassesModule, FocusModule, IconModule, ButtonModule], exports: [InputPasswordComponent] });
InputPasswordModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputPasswordModule, imports: [[CommonModule, FormsModule, BaseClassesModule, FocusModule, IconModule, ButtonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputPasswordModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, FormsModule, BaseClassesModule, FocusModule, IconModule, ButtonModule],
                    declarations: [InputPasswordComponent],
                    exports: [InputPasswordComponent],
                }]
        }] });
