// @backbase/ui-ang/esm2015/_example/account-selector/examples/account-selector.example.component.js
import { Component } from '@angular/core';
import { accounts } from './accounts';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/account-selector";
export class AccountSelectorExampleComponent {
    constructor() {
        this.items = accounts;
        this.selectedItems = this.items[0];
    }
    onChange($event) {
        const count = Array.isArray($event)
            ? $event.length
            : $event
                ? 1
                : 0;
        console.log(`${count} items selected`);
    }
    onSearch($event) {
        console.log(`Search for query: ${$event}`);
    }
    onClearSearch() {
        console.log('Search cleared');
    }
    onScrollToEnd() {
        console.log('Scrolled to end');
    }
}
AccountSelectorExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountSelectorExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
AccountSelectorExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AccountSelectorExampleComponent, selector: "bb-account-selector-example-ui", ngImport: i0, template: "<h2 id=\"ariaLabelledById\">Account Selector</h2>\n<div id=\"ariaDescribedById\">\n  Account selector Component that supports single select, multi select\n  and auto complete\n</div>\n<bb-account-selector-ui\n  [items]=\"items\"\n  [disableScrollEnd]=\"false\"\n  [multiple]=\"true\"\n  [closeOnSelect]=\"true\"\n  [filterItems]=\"true\"\n  [selectedItems]=\"selectedItems\"\n  [dropdownPosition]=\"'bottom'\"\n  (change)=\"onChange($event)\"\n  (clearSearch)=\"onClearSearch()\"\n  (scrollToEnd)=\"onScrollToEnd()\"\n  (search)=\"onSearch($event)\"\n  placeholder=\"Select Accounts\"\n  [aria-describedby]=\"'ariaLabelledById'\"\n  [aria-labelledby]=\"'ariaDescribedById'\"\n>\n</bb-account-selector-ui>\n", components: [{ type: i1.AccountSelectorComponent, selector: "bb-account-selector-ui", inputs: ["items", "disabled", "notFoundText", "showError", "productNumberFormat", "selectedItems", "multiple", "closeOnSelect", "searchable", "clearable", "placeholder", "loading", "dropdownPosition", "disableScrollEnd", "highlight", "filterItems", "markFirst", "internalFiltering", "virtualScroll", "required", "typeahead", "minTermLength", "typeToSearchText", "aria-describedby", "aria-labelledby", "aria-invalid", "searchFunc", "showCurrencySymbol", "showBankBranchCode", "groupByFn", "groupValueFn", "size", "compareItemsWith"], outputs: ["search", "clearSearch", "scroll", "scrollToEnd", "remove", "change", "filterChange", "focus", "blur"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountSelectorExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-account-selector-example-ui',
                    templateUrl: './account-selector.example.component.html',
                }]
        }] });
