// @backbase/ui-ang/esm2015/_example/table/showcase/table-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TableShowcaseComponent } from './table-showcase.component';
import { TableModule } from '@backbase/ui-ang/table';
import { PaginationModule } from '@backbase/ui-ang/pagination';
import { BadgeModule } from '@backbase/ui-ang/badge';
import { DropdownMenuModule } from '@backbase/ui-ang/dropdown-menu';
import { IconModule } from '@backbase/ui-ang/icon';
import * as i0 from "@angular/core";
export class TableShowcaseModule {
}
TableShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
TableShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableShowcaseModule, declarations: [TableShowcaseComponent], imports: [CommonModule, TableModule, IconModule, DropdownMenuModule, PaginationModule, BadgeModule] });
TableShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableShowcaseModule, imports: [[CommonModule, TableModule, IconModule, DropdownMenuModule, PaginationModule, BadgeModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, TableModule, IconModule, DropdownMenuModule, PaginationModule, BadgeModule],
                    declarations: [TableShowcaseComponent],
                    entryComponents: [TableShowcaseComponent],
                }]
        }] });
