// @backbase/ui-ang/esm2015/_example/badge-counter/showcase/badge-counter-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BadgeCounterShowcaseComponent } from './badge-counter-showcase.component';
import { BadgeCounterModule } from '@backbase/ui-ang/badge-counter';
import * as i0 from "@angular/core";
export class BadgeCounterShowcaseModule {
}
BadgeCounterShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BadgeCounterShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
BadgeCounterShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BadgeCounterShowcaseModule, declarations: [BadgeCounterShowcaseComponent], imports: [CommonModule, BadgeCounterModule] });
BadgeCounterShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BadgeCounterShowcaseModule, imports: [[CommonModule, BadgeCounterModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BadgeCounterShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, BadgeCounterModule],
                    declarations: [BadgeCounterShowcaseComponent],
                    entryComponents: [BadgeCounterShowcaseComponent],
                }]
        }] });
