// @backbase/ui-ang/esm2015/_example/badge-counter/showcase/badge-counter-showcase.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/badge-counter";
import * as i2 from "@angular/common";
let BadgeCounterShowcaseComponent = class BadgeCounterShowcaseComponent {
    constructor() {
        this.badgeCounterShowCaseSetup = [
            {
                type: 'success',
                count: 5,
            },
            {
                type: 'danger',
                count: 18,
            },
            {
                type: 'danger',
                count: 1900,
            },
            {
                type: 'danger',
                count: 100,
                maxCount: 99,
            },
        ];
    }
};
BadgeCounterShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BadgeCounterShowcaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
BadgeCounterShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: BadgeCounterShowcaseComponent, selector: "bb-badge-counter-ui-showcase", ngImport: i0, template: "<div class=\"bb-stack bb-block bb-block--md\">\n  <bb-badge-counter-ui\n    *ngFor=\"let badge of badgeCounterShowCaseSetup\"\n    [count]=\"badge.count\"\n    [maxCount]=\"badge.maxCount\"\n    [color]=\"badge.type\"\n    class=\"bb-stack__item\"\n  ></bb-badge-counter-ui>\n</div>\n", components: [{ type: i1.BadgeCounterComponent, selector: "bb-badge-counter-ui", inputs: ["color", "count", "maxCount"] }], directives: [{ type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }] });
BadgeCounterShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
], BadgeCounterShowcaseComponent);
export { BadgeCounterShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BadgeCounterShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-badge-counter-ui-showcase',
                    templateUrl: 'badge-counter-showcase.component.html',
                }]
        }] });
