// @backbase/ui-ang/esm2015/search-box/search-box.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { NgbTypeaheadModule } from '@ng-bootstrap/ng-bootstrap';
import { SearchBoxComponent } from './search-box.component';
import { ButtonModule } from '@backbase/ui-ang/button';
import { IconModule } from '@backbase/ui-ang/icon';
import { BaseClassesModule } from '@backbase/ui-ang/base-classes';
import { FocusModule } from '@backbase/ui-ang/focus';
import * as i0 from "@angular/core";
export class SearchBoxModule {
}
SearchBoxModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SearchBoxModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
SearchBoxModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SearchBoxModule, declarations: [SearchBoxComponent], imports: [CommonModule, NgbTypeaheadModule, ButtonModule, IconModule, FormsModule, BaseClassesModule, FocusModule], exports: [SearchBoxComponent] });
SearchBoxModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SearchBoxModule, imports: [[CommonModule, NgbTypeaheadModule, ButtonModule, IconModule, FormsModule, BaseClassesModule, FocusModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SearchBoxModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, NgbTypeaheadModule, ButtonModule, IconModule, FormsModule, BaseClassesModule, FocusModule],
                    declarations: [SearchBoxComponent],
                    exports: [SearchBoxComponent],
                }]
        }] });
