// @backbase/foundation-ang/esm2015/core/backbase-core/routing/config/outlet-map.js
import { PRIMARY_OUTLET } from '@angular/router';
import { Tree } from '../../models/tree';
import { flatMap } from '../../models/array';
/**
 * Special type to flag an outlet as having a collision. This
 * enables us to create the outlet tree, and name the outlet, but
 * also flag the item as having a collision with another item.
 */
export class OutletCollision {
    constructor(outletName, itemName, collisionWith) {
        this.outletName = outletName;
        this.itemName = itemName;
        this.collisionWith = collisionWith;
    }
}
const isValidOutlet = (outlet) => outlet !== undefined && outlet !== '';
const ɵ0 = isValidOutlet;
const outletCollision = (outletItem, collisionWith) => {
    return {
        item: outletItem.item,
        outlet: new OutletCollision(`ɵ${outletItem.item}`, outletItem.item, collisionWith.item),
    };
};
const ɵ1 = outletCollision;
const flattenOutletTreeNode = (node) => {
    const flattenedChildren = flatMap(flattenOutletTreeNode, node.children);
    if (node.value.outlet !== undefined) {
        // not routable
        // Don't flatten this node (just it's children)
        return [new Tree({ item: node.value.name, outlet: node.value.outlet }, flattenedChildren)];
    }
    else {
        // Flatten this node
        return flattenedChildren.reduce((acc, nextChild) => {
            const collisionWith = acc.find(child => child.value.outlet === nextChild.value.outlet);
            const newChild = collisionWith
                ? new Tree(outletCollision(nextChild.value, collisionWith.value), nextChild.children)
                : nextChild;
            return [...acc, newChild];
        }, []);
    }
};
const ɵ2 = flattenOutletTreeNode;
/**
 * Special case for flattening the root node. If it can't be flattened into a tree with a single
 * root node, we create a special root node, so there is only ever a single tree.
 */
const flattenOutletTree = (outletTree) => {
    return {
        rootItem: outletTree.value.name,
        children: flattenOutletTreeNode(outletTree),
    };
};
const ɵ3 = flattenOutletTree;
const mapItemToOutlet = (item) => {
    if (!item.routingStrategy) {
        return {
            name: item.name,
            outlet: undefined,
        };
    }
    return {
        name: item.name,
        outlet: isValidOutlet(item.outlet) ? item.outlet : PRIMARY_OUTLET,
    };
};
const ɵ4 = mapItemToOutlet;
const outletTreeToMap = (outletTree) => {
    const entries = flatMap(tree => tree.flatten(), outletTree.children) // flatten and merge the root outlet trees
        .map((node) => [node.value.item, node.value.outlet]);
    return new Map(entries);
};
const ɵ5 = outletTreeToMap;
export const outletMap = (itemTree) => {
    return outletTreeToMap(flattenOutletTree(itemTree.map(mapItemToOutlet)));
};
export { ɵ0, ɵ1, ɵ2, ɵ3, ɵ4, ɵ5 };
