// @backbase/foundation-ang/esm2015/core/backbase-core/routing/config/router-config.service.js
import { Injectable } from '@angular/core';
import { map, switchMap, shareReplay, distinctUntilChanged, withLatestFrom } from 'rxjs/operators';
import { ӨRootContainerService } from '../../models/root-container.service';
import { OutletCollision, outletMap } from './outlet-map';
import { RouteGuardsRegistryService } from '../guards/route-guards-registry.service';
import { ItemRegistryService } from '../../rendering/item-registry.service';
import { routableItemTree } from './routable-item-tree';
import { routingTree } from './routing-tree';
import { AbstractRouterConfigService } from './abstract-router-config.service';
import * as i0 from "@angular/core";
import * as i1 from "../../rendering/item-registry.service";
import * as i2 from "../guards/route-guards-registry.service";
import * as i3 from "../../models/root-container.service";
/**
 * @deprecated Will be removed in v7.0.0
 */
export class RouterConfigService extends AbstractRouterConfigService {
    constructor(itemRegistry, routeGuardsRegistry, rootContainer) {
        super();
        this.itemRegistry = itemRegistry;
        this.routeGuardsRegistry = routeGuardsRegistry;
        this.rootContainer = rootContainer;
        this.routableItemTree = this.rootContainer.model.pipe(switchMap(model => model.toTree()), switchMap(model => routableItemTree(this.itemRegistry, this.routeGuardsRegistry, model)), shareReplay(1));
        this.outlets = this.routableItemTree.pipe(map(outletMap), shareReplay(1));
        /**
         * Convert the component tree to a routing tree
         *
         * @internal
         */
        this.ɵrouteTree = this.outlets.pipe(withLatestFrom(this.routableItemTree), map(([outlets, tree]) => routingTree(tree, outlets)), shareReplay(1));
        /**
         * The dynamically generated Angular routes for the SPA, based on the model of the app.
         */
        this.routes = this.ɵrouteTree.pipe(map((routeTrees) => routeTrees.value.routes), shareReplay(1));
    }
    /**
     * Get the router-outlet name for the given item in the model tree.
     *
     * @internal
     */
    ɵitemOutletName(item) {
        return this.outlets.pipe(map(outlets => outlets.get(item.name)), map(outlet => (outlet instanceof OutletCollision ? outlet.outletName : outlet)), distinctUntilChanged());
    }
}
RouterConfigService.ɵprov = i0.ɵɵdefineInjectable({ factory: function RouterConfigService_Factory() { return new RouterConfigService(i0.ɵɵinject(i1.ItemRegistryService), i0.ɵɵinject(i2.RouteGuardsRegistryService), i0.ɵɵinject(i3.ӨRootContainerService)); }, token: RouterConfigService, providedIn: "root" });
RouterConfigService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root',
            },] }
];
RouterConfigService.ctorParameters = () => [
    { type: ItemRegistryService },
    { type: RouteGuardsRegistryService },
    { type: ӨRootContainerService }
];
