// @backbase/notifications-common-ang/esm2015/lib/pipes/notification-localization.pipe.js
import { Inject, LOCALE_ID, Pipe } from '@angular/core';
import * as i0 from "@angular/core";
export class NotificationLocalizationPipe {
    constructor(locale) {
        this.locale = locale;
    }
    transform(notification, property) {
        return (((notification.translations || {})[this.locale] || notification)[property] || '');
    }
}
/** @nocollapse */ NotificationLocalizationPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationLocalizationPipe, deps: [{ token: LOCALE_ID }], target: i0.ɵɵFactoryTarget.Pipe });
/** @nocollapse */ NotificationLocalizationPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationLocalizationPipe, name: "notificationLocalization" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationLocalizationPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'notificationLocalization',
                }]
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [LOCALE_ID]
                }] }]; } });
