// @backbase/contact-common-ang/esm2015/lib/models/types.js
export var LoadMoreMethod;
(function (LoadMoreMethod) {
    LoadMoreMethod["LOAD_MORE"] = "load-more";
    LoadMoreMethod["PAGINATION"] = "pagination";
})(LoadMoreMethod || (LoadMoreMethod = {}));
export var ContactActions;
(function (ContactActions) {
    ContactActions[ContactActions["Load"] = 0] = "Load";
    ContactActions[ContactActions["Refresh"] = 1] = "Refresh";
    ContactActions[ContactActions["LoadMore"] = 2] = "LoadMore";
    ContactActions[ContactActions["ChangePage"] = 3] = "ChangePage";
    ContactActions[ContactActions["Search"] = 4] = "Search";
    ContactActions[ContactActions["Sort"] = 5] = "Sort";
})(ContactActions || (ContactActions = {}));
export class ContactsError {
    constructor(errorResponse) {
        this.errorResponse = errorResponse;
        if (this.errorResponse) {
            const { status, error } = this.errorResponse;
            this.status = status;
            if (error) {
                this.message = error.message || '';
                this.errors = error.errors || [];
                this.additions = error.additions || {};
            }
        }
    }
}
export var ViewType;
(function (ViewType) {
    ViewType["CARD"] = "CARD";
    ViewType["FORM"] = "FORM";
})(ViewType || (ViewType = {}));
export var InputGroupName;
(function (InputGroupName) {
    InputGroupName["CONTACT"] = "contactDetails";
    InputGroupName["ACCOUNT"] = "accountDetails";
    InputGroupName["ADDRESS"] = "addressDetails";
})(InputGroupName || (InputGroupName = {}));
export var InputName;
(function (InputName) {
    InputName["COPY_CONTACT_ADDRESS"] = "copyContactAddress";
})(InputName || (InputName = {}));
export var AddressType;
(function (AddressType) {
    AddressType["CONTACT"] = "";
    AddressType["BANK"] = "bank";
    AddressType["ACCOUNTHOLDER"] = "accountHolder";
})(AddressType || (AddressType = {}));
export class ContactFormGroupItem {
    constructor() {
        this.viewType = ViewType.FORM;
        this.data = {};
        this.bankAddressViewType = ViewType.FORM;
        this.accountHolderAddressViewType = ViewType.FORM;
        this.isAddBankAddress = false;
        this.isAddAccountHolderAddress = false;
        this.accountHolderAddress = {};
        this.bankAddress = {};
        this.isCopyAddress = false;
        this.isContactAddressCopied = false;
    }
}
export var AccessContextScope;
(function (AccessContextScope) {
    AccessContextScope["USER"] = "USER";
    AccessContextScope["SA"] = "SA";
    AccessContextScope["LE"] = "LE";
})(AccessContextScope || (AccessContextScope = {}));
export var ActiveContactStatus;
(function (ActiveContactStatus) {
    ActiveContactStatus["ACTIVE"] = "ACTIVE";
    ActiveContactStatus["PENDING_APPROVAL"] = "PENDING_APPROVAL";
})(ActiveContactStatus || (ActiveContactStatus = {}));
export var ContactUpdateDeleteStatus;
(function (ContactUpdateDeleteStatus) {
    ContactUpdateDeleteStatus["APPROVAL_REQUIRED"] = "APPROVAL_REQUIRED";
    ContactUpdateDeleteStatus["REQUEST_ACCEPTED"] = "REQUEST_ACCEPTED";
    ContactUpdateDeleteStatus["REJECTED"] = "REJECTED";
    ContactUpdateDeleteStatus["PENDING"] = "PENDING";
    ContactUpdateDeleteStatus["APPROVED"] = "APPROVED";
    ContactUpdateDeleteStatus["CANCELLED"] = "CANCELLED";
})(ContactUpdateDeleteStatus || (ContactUpdateDeleteStatus = {}));
export var ApprovalStatus;
(function (ApprovalStatus) {
    ApprovalStatus["OPEN"] = "OPEN";
    ApprovalStatus["CLOSED"] = "CLOSED";
    ApprovalStatus["INVALID"] = "INVALID";
    ApprovalStatus["VALID"] = "VALID";
    ApprovalStatus["ENTERED"] = "ENTERED";
    ApprovalStatus["READY"] = "READY";
    ApprovalStatus["ACCEPTED"] = "ACCEPTED";
    ApprovalStatus["REJECTED"] = "REJECTED";
})(ApprovalStatus || (ApprovalStatus = {}));
