// @backbase/foundation-ang/esm2015/core/backbase-core/rendering/area/area.component.js
import { Component, HostBinding, Input } from '@angular/core';
import { ItemRegistryService } from '../item-registry.service';
/**
 * AreaComponent is a wrapper for ChromeComponent,
 * which defines the area for dynamic components rendering
 */
export class AreaComponent {
    constructor(itemRegistry) {
        this.itemRegistry = itemRegistry;
        /**
         * name attribute of the area,
         * represented also as a "data-area"
         * attribute in the corresponding node
         */
        this.name = '0';
        this.children = [];
    }
    /**
     * @internal
     */
    isModule(model) {
        return !!this.itemRegistry.getModuleLoader(model.ɵclassId);
    }
}
AreaComponent.decorators = [
    { type: Component, args: [{
                selector: 'bb-area',
                template: `
    <ng-container *ngFor="let item of children">
      <router-outlet *ngIf="isModule(item.value); else renderModel"></router-outlet>
      <ng-template #renderModel><bb-chrome [item]="item"></bb-chrome></ng-template>
      <ng-container> </ng-container
    ></ng-container>
  `,
                styles: [':host { display: block; }']
            },] }
];
AreaComponent.ctorParameters = () => [
    { type: ItemRegistryService }
];
AreaComponent.propDecorators = {
    name: [{ type: HostBinding, args: ['attr.data-area',] }, { type: Input }],
    children: [{ type: Input }]
};
