// @backbase/manage-payees-journey-ang/esm2015/src/directives/base-electronic-payee-review-page.model.js
import { HostListener, Directive } from '@angular/core';
import { of, Subject } from 'rxjs';
import { filter, map, shareReplay, switchMap, take, takeUntil, tap } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "../services/billpay-manage-payee-state.service";
import * as i3 from "@backbase/billpay-journeys-common";
import * as i4 from "../services/manage-payees-journey-config.service";
export class BaseElectronicPayeeReviewPageModelDirective {
    /** @internal */
    constructor(route, stateService, discardModalService, config) {
        this.route = route;
        this.stateService = stateService;
        this.discardModalService = discardModalService;
        this.config = config;
        this.destroy$ = new Subject();
        this.payee$ = this.stateService.electronicPayeeFormState.pipe(take(1), shareReplay(1));
        this.id$ = this.route.paramMap.pipe(take(1), map(params => params.get('id') || ''));
    }
    ngOnInit() {
        this.payee$
            .pipe(filter(payee => !payee), switchMap(() => this.id$))
            .subscribe(id => {
            this.navigateBack(id);
        });
        this.payeeSubmit$.pipe(takeUntil(this.destroy$), filter(Boolean)).subscribe(payee => this.onSubmitPayee(payee));
    }
    ngOnDestroy() {
        this.destroy$.next();
        this.destroy$.complete();
    }
    onWindowUnload($event) {
        if (!this.isDiscardModalOpen) {
            $event.returnValue = true;
        }
    }
    get isDiscardModalOpen() {
        return this.discardModalService.isDiscardModalOpen;
    }
    get dismissTimeout() {
        if (this.config.notificationDismissTime === undefined) {
            throw new Error('Dismiss timeout value not found');
        }
        return this.config.notificationDismissTime;
    }
    openModal() {
        return this.payeeSubmit$.pipe(switchMap(payee => (payee ? of(true) : this.discardModalService.openModal())), tap(isResolved => isResolved && this.stateService.resetElectronicPayee()));
    }
    onModalResolve(resolvedWith) {
        this.discardModalService.onModalResolve(resolvedWith);
    }
}
BaseElectronicPayeeReviewPageModelDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BaseElectronicPayeeReviewPageModelDirective, deps: [{ token: i1.ActivatedRoute }, { token: i2.BillpayManagePayeeStateService }, { token: i3.BillpayDiscardChangesModalService }, { token: i4.ManagePayeesJourneyConfigService }], target: i0.ɵɵFactoryTarget.Directive });
BaseElectronicPayeeReviewPageModelDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.15", type: BaseElectronicPayeeReviewPageModelDirective, host: { listeners: { "window:beforeunload": "onWindowUnload($event)" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BaseElectronicPayeeReviewPageModelDirective, decorators: [{
            type: Directive
        }], ctorParameters: function () { return [{ type: i1.ActivatedRoute }, { type: i2.BillpayManagePayeeStateService }, { type: i3.BillpayDiscardChangesModalService }, { type: i4.ManagePayeesJourneyConfigService }]; }, propDecorators: { onWindowUnload: [{
                type: HostListener,
                args: ['window:beforeunload', ['$event']]
            }] } });
