// @backbase/messages-client-inbox-journey-ang/esm2015/src/messages-client-inbox-journey.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MessagesClientInboxJourneyComponent } from './messages-client-inbox-journey.component';
import { provideRoutes, RouterModule } from '@angular/router';
import { BackbaseCoreModule, TemplateRegistry } from '@backbase/foundation-ang/core';
import { MessagesCreateMessageModalWidgetModule } from '@backbase/messages-create-message-modal-widget-ang';
import { MessagesClientInboxListComponent } from './components/messages-client-inbox-list/messages-client-inbox-list.component';
import { MessagesConversationsListWidgetModule } from '@backbase/messages-conversations-list-widget-ang';
import { MessagesConversationThreadWidgetModule } from '@backbase/messages-conversation-thread-widget-ang';
import { MessagesClientInboxThreadComponent } from './components/messages-client-inbox-thread/messages-client-inbox-thread.component';
import { MessagesManageConversationsComponent } from './components/messages-manage-conversations/messages-manage-conversations.component';
import { defaultRoute } from './messages-client-inbox-journey.routes';
import { MessagesClientInboxJourneyConfigService } from './messages-client-inbox-journey-config.service';
import { MessagesCollectionUiAngModule } from '@backbase/messages-ui-ang';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const components = [
    MessagesClientInboxJourneyComponent,
    MessagesClientInboxListComponent,
    MessagesClientInboxThreadComponent,
    MessagesManageConversationsComponent,
];
export class MessagesClientInboxJourneyModule {
    static forRoot(data = { routes: defaultRoute }) {
        return {
            ngModule: MessagesClientInboxJourneyModule,
            providers: [provideRoutes([data.routes])],
        };
    }
}
MessagesClientInboxJourneyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesClientInboxJourneyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MessagesClientInboxJourneyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesClientInboxJourneyModule, declarations: [MessagesClientInboxJourneyComponent,
        MessagesClientInboxListComponent,
        MessagesClientInboxThreadComponent,
        MessagesManageConversationsComponent], imports: [CommonModule,
        RouterModule,
        MessagesCreateMessageModalWidgetModule,
        MessagesConversationsListWidgetModule,
        MessagesConversationThreadWidgetModule,
        MessagesCollectionUiAngModule, i1.BackbaseCoreModule] });
MessagesClientInboxJourneyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesClientInboxJourneyModule, providers: [MessagesClientInboxJourneyConfigService, TemplateRegistry], imports: [[
            CommonModule,
            RouterModule,
            MessagesCreateMessageModalWidgetModule,
            MessagesConversationsListWidgetModule,
            MessagesConversationThreadWidgetModule,
            MessagesCollectionUiAngModule,
            BackbaseCoreModule.withConfig({
                classMap: {
                    MessagesClientInboxJourneyComponent,
                },
            }),
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesClientInboxJourneyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components],
                    imports: [
                        CommonModule,
                        RouterModule,
                        MessagesCreateMessageModalWidgetModule,
                        MessagesConversationsListWidgetModule,
                        MessagesConversationThreadWidgetModule,
                        MessagesCollectionUiAngModule,
                        BackbaseCoreModule.withConfig({
                            classMap: {
                                MessagesClientInboxJourneyComponent,
                            },
                        }),
                    ],
                    providers: [MessagesClientInboxJourneyConfigService, TemplateRegistry],
                }]
        }] });
