// @backbase/messages-client-inbox-journey-ang/esm2015/src/messages-client-inbox-journey.routes.js
import '@angular/localize/init';
import { MessagesClientInboxJourneyComponent } from './messages-client-inbox-journey.component';
import { MessagesClientInboxListComponent } from './components/messages-client-inbox-list/messages-client-inbox-list.component';
import { MessagesClientInboxThreadComponent } from './components/messages-client-inbox-thread/messages-client-inbox-thread.component';
import { MessagesManageConversationsComponent } from './components/messages-manage-conversations/messages-manage-conversations.component';
export const mailboxTypeTitles = {
    inbox: $localize `:Client Inbox Journey - Inbox mailbox type@@messages-client-inbox-journey.inbox.mailbox-type:Inbox`,
    // eslint-disable-next-line max-len
    outbox: $localize `:Client Inbox Journey - Outbox mailbox type@@messages-client-inbox-journey.outbox.mailbox-type:Outbox`,
    // eslint-disable-next-line max-len
    drafts: $localize `:Client Inbox Journey - Drafts mailbox type@@messages-client-inbox-journey.drafts.mailbox-type:Drafts`,
};
export const defaultRoute = {
    path: '',
    component: MessagesClientInboxJourneyComponent,
    children: [
        { path: '', redirectTo: 'inbox/list' },
        {
            path: 'inbox',
            data: { title: mailboxTypeTitles.inbox },
            component: MessagesManageConversationsComponent,
            children: [
                { path: '', redirectTo: 'list', pathMatch: 'full' },
                { path: 'list', component: MessagesClientInboxListComponent, data: { mailboxType: 'inbox' } },
                {
                    path: 'conversation',
                    component: MessagesClientInboxThreadComponent,
                    data: { mailboxTypeTitle: mailboxTypeTitles.inbox },
                },
            ],
        },
        {
            path: 'outbox',
            data: { title: mailboxTypeTitles.outbox },
            component: MessagesManageConversationsComponent,
            children: [
                { path: '', redirectTo: 'list', pathMatch: 'full' },
                { path: 'list', component: MessagesClientInboxListComponent, data: { mailboxType: 'outbox' } },
                {
                    path: 'conversation',
                    component: MessagesClientInboxThreadComponent,
                    data: { mailboxTypeTitle: mailboxTypeTitles.outbox },
                },
            ],
        },
        {
            path: 'drafts',
            data: { title: mailboxTypeTitles.drafts },
            component: MessagesManageConversationsComponent,
            children: [
                { path: '', redirectTo: 'list', pathMatch: 'full' },
                { path: 'list', component: MessagesClientInboxListComponent, data: { mailboxType: 'drafts' } },
                {
                    path: 'conversation',
                    component: MessagesClientInboxThreadComponent,
                    data: { mailboxTypeTitle: mailboxTypeTitles.drafts },
                },
            ],
        },
    ],
};
