// @backbase/messages-client-inbox-journey-ang/esm2015/src/messages-client-inbox-journey.component.js
import { ChangeDetectionStrategy, Component, Inject } from '@angular/core';
import '@angular/localize/init';
import { PUBSUB } from '@backbase/foundation-ang/web-sdk';
import * as i0 from "@angular/core";
import * as i1 from "./messages-client-inbox-journey-config.service";
import * as i2 from "@angular/router";
import * as i3 from "@backbase/messages-create-message-modal-widget-ang";
import * as i4 from "@backbase/messages-ui-ang";
export class MessagesClientInboxJourneyComponent {
    constructor(config, eventBus, route) {
        this.config = config;
        this.eventBus = eventBus;
        this.route = route;
        /**
         * Collection of Tabs to be rendered according to the journey layout and routing.
         */
        this.tabs = [];
        const routeConfig = this.route.snapshot.routeConfig;
        if (routeConfig && routeConfig.children) {
            this.tabs = (routeConfig === null || routeConfig === void 0 ? void 0 : routeConfig.children) ? this.initializeTabs(routeConfig) : [];
        }
    }
    initializeTabs({ children = [] }) {
        return children
            .filter(routeItem => routeItem.path && routeItem.data)
            .map((routeItem, index) => {
            var _a;
            return ({
                title: ((_a = routeItem.data) === null || _a === void 0 ? void 0 : _a.title) ||
                    $localize `Default unnamed tab title@@messages-client-inbox-journey.default-tab.title:Tab${index + 1}`,
                route: routeItem.path || `${index + 1}`,
            });
        });
    }
    /**
     * Handler for the header button to open the upload modal
     */
    openComposeModal() {
        this.eventBus.publish(this.config.createMessageOpenEventName, undefined);
    }
}
MessagesClientInboxJourneyComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesClientInboxJourneyComponent, deps: [{ token: i1.MessagesClientInboxJourneyConfigService }, { token: PUBSUB }, { token: i2.ActivatedRoute }], target: i0.ɵɵFactoryTarget.Component });
MessagesClientInboxJourneyComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.14", type: MessagesClientInboxJourneyComponent, selector: "bb-messages-client-inbox-journey", ngImport: i0, template: "<bb-messages-create-message-modal-widget-ang\n  [maxAttachmentSize]=\"config.maxAttachmentSize\"\n  [maxSubjectLength]=\"config.maxSubjectLength\"\n  [maxMessageLength]=\"config.maxMessageLength\"\n  [modalHeader]=\"config.modalHeader\"\n  [hideComposeButton]=\"config.hideComposeButton\"\n  [createMessageOpenEventName]=\"config.createMessageOpenEventName\"\n  [createMessageClosedEventName]=\"config.createMessageClosedEventName\"\n>\n</bb-messages-create-message-modal-widget-ang>\n<bb-messages-heading\n  [headingTitle]=\"config.headingTitle\"\n  [headingType]=\"config.headingType\"\n  [headingWrapperClasses]=\"config.headingWrapperClasses\"\n  [buttonText]=\"config.buttonText\"\n  (buttonAction)=\"openComposeModal()\"\n>\n</bb-messages-heading>\n<div class=\"bb-messages bb-messages__tabs-wrapper\">\n  <bb-messages-tabs data-role=\"client-inbox__tabs\" [tabs]=\"tabs\"></bb-messages-tabs>\n</div>\n", components: [{ type: i3.MessagesCreateMessageModalWidgetComponent, selector: "bb-messages-create-message-modal-widget-ang" }, { type: i4.MessagesHeadingComponent, selector: "bb-messages-heading", inputs: ["headingTitle", "headingType", "headingClasses", "headingWrapperClasses", "buttonText", "showButton"], outputs: ["buttonAction"] }, { type: i4.MessagesTabsComponent, selector: "bb-messages-tabs", inputs: ["tabs"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesClientInboxJourneyComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-messages-client-inbox-journey',
                    templateUrl: './messages-client-inbox-journey.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.MessagesClientInboxJourneyConfigService }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [PUBSUB]
                }] }, { type: i2.ActivatedRoute }]; } });
