// @backbase/consent-journey-ang/esm2015/src/views/payment-request/payment-request.component.js
import { Component } from '@angular/core';
import { Subject } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "../../services/consent-journey-config.service";
import * as i2 from "@angular/router";
import * as i3 from "@backbase/payment-request-widget-ang";
import * as i4 from "@backbase/ui-ang/button";
export class PaymentRequestComponent {
    constructor(config, router, activeRoute) {
        this.config = config;
        this.router = router;
        this.activeRoute = activeRoute;
        this.destroy$ = new Subject();
    }
    requestPayment() {
        this.router.navigate([], {
            queryParams: {
                paymentRequestId: '1231313',
            },
            skipLocationChange: false,
            relativeTo: this.activeRoute,
        });
    }
    handleAllowRequest(event) {
        this.router.navigate(['../payment-redirect', { allowPaymentRequestRedirectUrl: event }], {
            relativeTo: this.activeRoute,
        });
    }
    handleRejectRequest(event) {
        this.router.navigate(['../payment-redirect', { rejectPaymentRequestRedirectUrl: event }], {
            relativeTo: this.activeRoute,
        });
    }
    ngOnDestroy() {
        this.destroy$.next();
    }
}
PaymentRequestComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentRequestComponent, deps: [{ token: i1.ConsentJourneyConfigurationService }, { token: i2.Router }, { token: i2.ActivatedRoute }], target: i0.ɵɵFactoryTarget.Component });
PaymentRequestComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: PaymentRequestComponent, selector: "bb-payment-request-view", ngImport: i0, template: `
    <div class="container--fixed-width-small mx-auto">
      <div class="bb-card">
        <div class="bb-card__body">
          <div class="mb-5 d-flex justify-content-center">
            <button
              bbButton
              (click)="requestPayment()"
              i18n="Simulate Payment Request@@consent-journey.label.simulate-payment-request">
              Pay with Backbase
            </button>
          </div>
          <bb-payment-request-widget
            (allowPaymentRequestRedirectUrl)="handleAllowRequest($event)"
            (rejectPaymentRequestRedirectUrl)="handleRejectRequest($event)"></bb-payment-request-widget>
        </div>
      </div>
    </div>
  `, isInline: true, components: [{ type: i3.PaymentRequestWidgetComponent, selector: "bb-payment-request-widget", outputs: ["allowPaymentRequestRedirectUrl", "rejectPaymentRequestRedirectUrl"] }], directives: [{ type: i4.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentRequestComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-payment-request-view',
                    template: `
    <div class="container--fixed-width-small mx-auto">
      <div class="bb-card">
        <div class="bb-card__body">
          <div class="mb-5 d-flex justify-content-center">
            <button
              bbButton
              (click)="requestPayment()"
              i18n="Simulate Payment Request@@consent-journey.label.simulate-payment-request">
              Pay with Backbase
            </button>
          </div>
          <bb-payment-request-widget
            (allowPaymentRequestRedirectUrl)="handleAllowRequest($event)"
            (rejectPaymentRequestRedirectUrl)="handleRejectRequest($event)"></bb-payment-request-widget>
        </div>
      </div>
    </div>
  `,
                }]
        }], ctorParameters: function () { return [{ type: i1.ConsentJourneyConfigurationService }, { type: i2.Router }, { type: i2.ActivatedRoute }]; } });
