// @backbase/loans-collection-ui-ang/esm2015/lib/loan-info/loan-info-item/loan-info-item.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AmountModule } from '@backbase/ui-ang/amount';
import { BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import { LoanInfoItemComponent } from './loan-info-item.component';
import * as i0 from "@angular/core";
export class LoanInfoItemModule {
}
/** @nocollapse */ LoanInfoItemModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoItemModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanInfoItemModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoItemModule, declarations: [LoanInfoItemComponent], imports: [CommonModule, AmountModule, BbDatePipeModule], exports: [LoanInfoItemComponent] });
/** @nocollapse */ LoanInfoItemModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoItemModule, imports: [[CommonModule, AmountModule, BbDatePipeModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoItemModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanInfoItemComponent],
                    imports: [CommonModule, AmountModule, BbDatePipeModule],
                    exports: [LoanInfoItemComponent],
                }]
        }] });
