// @backbase/product-summary-ui/esm2015/src/bb-line-chart-ui/line-chart.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AmountModule } from '@backbase/ui-ang/amount';
import { LineChartModule as SwimlineLineChartModule } from '@swimlane/ngx-charts';
import { LineChartComponent, BalanceHistoryLineChartCustomizableDirective, BalanceHistoryLineChartTooltipCustomizableDirective, BalanceHistoryLineChartTooltipSeriesCustomizableDirective, } from './line-chart.component';
import { LineChartLegendComponent, BalanceHistoryLineChartLegendCustomizableDirective, } from './line-chart-legend.component';
import { ChartLegendModule } from '../bb-chart-legend-ui/chart-legend.module';
import * as i0 from "@angular/core";
const uiModules = [AmountModule, ChartLegendModule];
const declarations = [
    LineChartComponent,
    LineChartLegendComponent,
    BalanceHistoryLineChartCustomizableDirective,
    BalanceHistoryLineChartTooltipCustomizableDirective,
    BalanceHistoryLineChartTooltipSeriesCustomizableDirective,
    BalanceHistoryLineChartCustomizableDirective,
    BalanceHistoryLineChartTooltipCustomizableDirective,
    BalanceHistoryLineChartTooltipSeriesCustomizableDirective,
    BalanceHistoryLineChartLegendCustomizableDirective,
];
export class LineChartModule {
}
LineChartModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LineChartModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LineChartModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LineChartModule, declarations: [LineChartComponent,
        LineChartLegendComponent,
        BalanceHistoryLineChartCustomizableDirective,
        BalanceHistoryLineChartTooltipCustomizableDirective,
        BalanceHistoryLineChartTooltipSeriesCustomizableDirective,
        BalanceHistoryLineChartCustomizableDirective,
        BalanceHistoryLineChartTooltipCustomizableDirective,
        BalanceHistoryLineChartTooltipSeriesCustomizableDirective,
        BalanceHistoryLineChartLegendCustomizableDirective], imports: [CommonModule, SwimlineLineChartModule, AmountModule, ChartLegendModule], exports: [LineChartComponent,
        LineChartLegendComponent,
        BalanceHistoryLineChartCustomizableDirective,
        BalanceHistoryLineChartTooltipCustomizableDirective,
        BalanceHistoryLineChartTooltipSeriesCustomizableDirective,
        BalanceHistoryLineChartCustomizableDirective,
        BalanceHistoryLineChartTooltipCustomizableDirective,
        BalanceHistoryLineChartTooltipSeriesCustomizableDirective,
        BalanceHistoryLineChartLegendCustomizableDirective, AmountModule, ChartLegendModule] });
LineChartModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LineChartModule, imports: [[CommonModule, SwimlineLineChartModule, ...uiModules], AmountModule, ChartLegendModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LineChartModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, SwimlineLineChartModule, ...uiModules],
                    declarations: [...declarations],
                    exports: [...declarations, ...uiModules],
                }]
        }] });
