// @backbase/messages-common/esm2015/src/components/conversation-thread/component/base-conversation-thread.component.js
import { Component, EventEmitter, Input, Output, ViewChild } from '@angular/core';
import { FormControl, FormGroup } from '@angular/forms';
import { BehaviorSubject, combineLatest, EMPTY, Subject, of } from 'rxjs';
import { catchError, switchMap, takeUntil, tap } from 'rxjs/operators';
import { ManipulationType, } from '../../messages-manipulation-confirm-modal/manipulation-confirm-modal-types.model';
import { ReadUnreadStatusType } from '../models/conversations.model';
import { BaseConversationThreadDraftsComponent } from './base-conversation-thread-drafts.component';
import { UnknownIdError } from '../../../utils/unknown-id-error';
import { shareReplayOnce } from '../../../helpers/rxjs-operators';
import * as i0 from "@angular/core";
import * as i1 from "../services/abstract-base-messages-conversation-thread.service";
import * as i2 from "@angular/router";
import * as i3 from "../../../services/shared-methods/shared-methods.service";
import * as i4 from "../../messages-manipulation-confirm-modal/messages-manipulation-confirm-modal.service";
import * as i5 from "../../upload-attachments/services/messages-upload-attachments.service";
import * as i6 from "../../../services/messages-encoding/messages-encoding.service";
export class BaseConversationThreadComponent extends BaseConversationThreadDraftsComponent {
    constructor(apiThreadService, activatedRoute, shared, confirmModalService, uploadService, encodingService) {
        super(apiThreadService, shared);
        this.apiThreadService = apiThreadService;
        this.activatedRoute = activatedRoute;
        this.shared = shared;
        this.confirmModalService = confirmModalService;
        this.uploadService = uploadService;
        this.encodingService = encodingService;
        /**
         * Event emitted when this thread should be closed or navigation should be changed to previous or another widget
         */
        this.back = new EventEmitter();
        /**
         * The conversation thread ID.
         */
        this.id = `base-conversation-thread-component-${BaseConversationThreadComponent.counter++}`;
        /**
         * The manipulation types enum (`resolve`, `unresolve`, `assign`, `unassign`, `discard`,
         * `discardDraft`, `discardPreview`, `deleteTopic`, `deleteMailout`).
         */
        this.manipulationTypes = ManipulationType;
        this.unsubscribe = new Subject();
        this.replyPressed = new Subject();
        this.conversationId$ = new BehaviorSubject(undefined);
        /**
         * The loading flag for the data fetched through the HTTP request.
         */
        this.loading = true;
        /**
         * The flag defining if the reply is opened or not.
         */
        this.replyOpened = false;
        /**
         * The flag defining if the topic alert is closed or not.
         */
        this.topicAlertClosed = false;
        /**
         * The modal for action confirmation.
         */
        this.confirmationModal = {
            opened: false,
            type: undefined,
        };
        /**
         * The reply form.
         */
        this.replyFormGroup = new FormGroup({
            body: new FormControl(),
            attachments: new FormControl(),
        });
        /**
         * The form loading flag.
         */
        this.formLoading = false;
        this.conversationId = this.getConversationId();
        this.markAsReadUnreadSubject = new BehaviorSubject(ReadUnreadStatusType.READ);
        this.markAsReadUnreadStream = this.getMarkAsReadUnreadStream();
    }
    /**
     * Selected conversation id setter
     */
    set selectedConversationId(value) {
        this.conversationId$.next(value);
    }
    /**
     * Conversation id getter
     */
    get conversationIdValue() {
        return this.conversationId$.getValue();
    }
    ngOnDestroy() {
        this.unsubscribe.next();
        this.unsubscribe.complete();
    }
    /**
     * The method to save the draft.
     */
    onSaveDraft() {
        if (this.replyFormGroup.valid) {
            this.saveDraft(this.conversationIdValue ? this.conversationIdValue : '', this.convertFormValue())
                .pipe(takeUntil(this.unsubscribe))
                .subscribe();
        }
    }
    /**
     * The method that is used for component initialization to subscribe for the read-unread status and
     * to toggle the reply form updates.
     */
    initSubscriptions() {
        this.markAsReadUnreadStream.pipe(takeUntil(this.unsubscribe)).subscribe();
        this.confirmModalService.emitToggleModal$.pipe(takeUntil(this.unsubscribe)).subscribe(({ opened, type }) => {
            this.confirmationModal = { opened, type };
            if (!opened && (type === ManipulationType.discard || type === ManipulationType.discardDraft)) {
                this.toggleReplyForm(false);
            }
        });
    }
    /**
     * The method to navigate back.
     */
    navigateBack() {
        this.back.emit();
    }
    /**
     * The method to toggle the reply form.
     *
     * @param flag Toggle status
     */
    toggleReplyForm(flag) {
        this.replyOpened = flag;
        if (!flag) {
            this.uploadService.reset(true);
            this.replyFormGroup.reset();
        }
        else {
            setTimeout(() => this.richTextEditor && this.richTextEditor.labelClick());
        }
    }
    /**
     * The method to submit the reply.
     */
    reply() {
        this.replyFormGroup.markAsDirty();
        this.replyFormGroup.markAllAsTouched();
        if (this.replyFormGroup.valid) {
            this.replyPressed.next(this.convertFormValue());
        }
    }
    /**
     * The method that defines whether the control is valid or not.
     *
     * @param field Field name
     * @returns Is form field valid flag
     */
    isInvalidControl(field) {
        const control = this.getControl(field);
        return !!control && control.touched && control.invalid;
    }
    /**
     * The method to get any errors generated by failing validation, or null if there are no errors.
     *
     * @param field Field name
     * @param type Error type
     */
    hasError(field, type) {
        const control = this.getControl(field);
        return control && control.errors && control.errors[type];
    }
    getConversationId() {
        return this.conversationId$.pipe(switchMap(id => {
            if (!id) {
                this.errorResponse = new UnknownIdError();
                return EMPTY;
            }
            return of(id);
        }), shareReplayOnce());
    }
    getMarkAsReadUnreadStream() {
        return combineLatest([this.markAsReadUnreadSubject, this.conversationId]).pipe(switchMap(([status, conversationId]) => this.apiThreadService.markConversationsAsReadUnread({ status, ids: [conversationId] }).pipe(tap(() => status === ReadUnreadStatusType.UNREAD && this.navigateBack()), catchError(error => {
            this.shared.openNotification(this.shared.getErrorMessage(error), 'error');
            return EMPTY;
        }))));
    }
    convertFormValue() {
        const { body, attachments } = this.replyFormGroup.value;
        return Object.assign({ body: this.encodingService.b64Encode(body) }, ((attachments === null || attachments === void 0 ? void 0 : attachments.length) && { attachments: attachments.map(item => item.id) }));
    }
    getControl(field) {
        var _a;
        return (_a = this.replyFormGroup) === null || _a === void 0 ? void 0 : _a.controls[field];
    }
}
BaseConversationThreadComponent.counter = 0;
BaseConversationThreadComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: BaseConversationThreadComponent, deps: [{ token: i1.AbstractBaseMessagesConversationThreadService }, { token: i2.ActivatedRoute }, { token: i3.SharedMethodsService }, { token: i4.MessagesManipulationConfirmModalService }, { token: i5.MessagesUploadAttachmentsService }, { token: i6.MessagesEncodingService }], target: i0.ɵɵFactoryTarget.Component });
BaseConversationThreadComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.14", type: BaseConversationThreadComponent, selector: "ng-component", inputs: { selectedConversationId: "selectedConversationId" }, outputs: { back: "back" }, viewQueries: [{ propertyName: "replySuccessTpl", first: true, predicate: ["conversationReplySuccess,messageReplySuccess"], descendants: true }, { propertyName: "deleteSuccessTpl", first: true, predicate: ["conversationDeleteSuccess,messageDeleteSuccess"], descendants: true }, { propertyName: "richTextEditor", first: true, predicate: ["richTextEditor"], descendants: true }], usesInheritance: true, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: BaseConversationThreadComponent, decorators: [{
            type: Component,
            args: [{
                    template: '',
                }]
        }], ctorParameters: function () { return [{ type: i1.AbstractBaseMessagesConversationThreadService }, { type: i2.ActivatedRoute }, { type: i3.SharedMethodsService }, { type: i4.MessagesManipulationConfirmModalService }, { type: i5.MessagesUploadAttachmentsService }, { type: i6.MessagesEncodingService }]; }, propDecorators: { back: [{
                type: Output
            }], selectedConversationId: [{
                type: Input
            }], replySuccessTpl: [{
                type: ViewChild,
                args: ['conversationReplySuccess,messageReplySuccess']
            }], deleteSuccessTpl: [{
                type: ViewChild,
                args: ['conversationDeleteSuccess,messageDeleteSuccess']
            }], richTextEditor: [{
                type: ViewChild,
                args: ['richTextEditor']
            }] } });
