// @backbase/messages-common/esm2015/src/components/conversation-thread/component/base-conversation-thread-drafts.component.js
import { Component, ViewChild } from '@angular/core';
import { EMPTY, of } from 'rxjs';
import { catchError, switchMap, tap } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "../services/abstract-base-messages-conversation-thread.service";
import * as i2 from "../../../services/shared-methods/shared-methods.service";
export class BaseConversationThreadDraftsComponent {
    constructor(apiThreadService, shared) {
        this.apiThreadService = apiThreadService;
        this.shared = shared;
    }
    saveDraft(conversationId, body) {
        return of(this.draftResponse).pipe(switchMap(draft => (draft ? this.updateDraft(conversationId, body) : this.createDraft(conversationId, body))), tap(() => this.shared.openNotification(this.draftSaveSuccessTpl, 'success')), catchError(error => {
            this.shared.openNotification(this.shared.getErrorMessage(error), 'error');
            return EMPTY;
        }));
    }
    deleteDraft() {
        if (this.apiThreadService.deleteDraft && this.draftResponse) {
            return this.apiThreadService.deleteDraft(this.draftResponse.id).pipe(tap(() => {
                this.draftResponse = undefined;
                this.shared.openNotification(this.draftDeleteSuccessTpl, 'success');
            }), catchError(error => {
                this.shared.openNotification(this.shared.getErrorMessage(error), 'error');
                return EMPTY;
            }));
        }
        return of(undefined);
    }
    createDraft(conversationId, body) {
        if (this.apiThreadService.createDraft) {
            return this.apiThreadService.createDraft(conversationId, body).pipe(tap(res => (this.draftResponse = {
                id: res ? res.id : '',
                createdAt: new Date().toString(),
            })));
        }
        return of(undefined);
    }
    updateDraft(conversationId, body) {
        if (this.apiThreadService.updateDraft) {
            return this.apiThreadService.updateDraft(conversationId, body);
        }
        return of(undefined);
    }
}
BaseConversationThreadDraftsComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: BaseConversationThreadDraftsComponent, deps: [{ token: i1.AbstractBaseMessagesConversationThreadService }, { token: i2.SharedMethodsService }], target: i0.ɵɵFactoryTarget.Component });
BaseConversationThreadDraftsComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.14", type: BaseConversationThreadDraftsComponent, selector: "ng-component", viewQueries: [{ propertyName: "draftSaveSuccessTpl", first: true, predicate: ["draftSaveSuccess"], descendants: true }, { propertyName: "draftDeleteSuccessTpl", first: true, predicate: ["draftDeleteSuccess"], descendants: true }], ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: BaseConversationThreadDraftsComponent, decorators: [{
            type: Component,
            args: [{
                    template: '',
                }]
        }], ctorParameters: function () { return [{ type: i1.AbstractBaseMessagesConversationThreadService }, { type: i2.SharedMethodsService }]; }, propDecorators: { draftSaveSuccessTpl: [{
                type: ViewChild,
                args: ['draftSaveSuccess']
            }], draftDeleteSuccessTpl: [{
                type: ViewChild,
                args: ['draftDeleteSuccess']
            }] } });
