// @backbase/payment-orders-ang/esm2015/lib/ui/components/beneficiary-selector-ui/components/beneficiary-contact/beneficiary-contact.js
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/icon";
import * as i2 from "@angular/common";
export class BeneficiaryContactComponent {
    /**
     * Contact item
     */
    set contact(item) {
        this._contact = item;
        if (item === null || item === void 0 ? void 0 : item.accounts[0]) {
            this.accountNumber =
                item.accounts[0].accountNumber || item.accounts[0].IBAN || '';
        }
    }
    get contact() {
        return this._contact;
    }
}
/** @nocollapse */ BeneficiaryContactComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BeneficiaryContactComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ BeneficiaryContactComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: BeneficiaryContactComponent, selector: "bb-beneficiary-contact", inputs: { contact: "contact" }, ngImport: i0, template: "<div class=\"bb-stack\">\n  <div class=\"d-none d-md-block bb-stack__item\">\n    <span class=\"sr-only\" i18n=\"@@beneficiary.selector.ui.group.contacts\"\n      >Contact</span\n    >\n    <bb-icon-ui size=\"xl\" color=\"primary\" class=\"bb-stack\" name=\"contacts\">\n    </bb-icon-ui>\n  </div>\n\n  <div class=\"text-left\">\n    <span\n      class=\"sr-only\"\n      i18n=\"@@beneficiary.selector.ui.contact.account.name.label\"\n      id=\"contact_accountNameLabel\"\n    >\n      Account name:\n    </span>\n    <div class=\"bb-text-bold\" data-role=\"beneficiary-account-name-list-item\">\n      {{ contact?.name }}\n      <ng-container *ngIf=\"contact?.accountName\">\n        - {{contact?.accountName}}</ng-container\n      >\n    </div>\n\n    <span\n      class=\"sr-only\"\n      i18n=\"@@beneficiary.selector.ui.contact.account.number.label\"\n      id=\"contact_accountNumberLabel\"\n    >\n      ; Account number:\n    </span>\n    <div\n      class=\"text-dark\"\n      *ngIf=\"accountNumber\"\n      data-role=\"beneficiary-account-number-list-item\"\n    >\n      {{ accountNumber }}\n    </div>\n  </div>\n</div>\n", components: [{ type: i1.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BeneficiaryContactComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-beneficiary-contact',
                    templateUrl: 'beneficiary-contact.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { contact: [{
                type: Input
            }] } });
