// @backbase/stop-checks-journey-ang/esm2015/lib/services/navigation.service.js
import { Injectable } from '@angular/core';
import { ActivatedRoute, Router } from '@angular/router';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
/** @internal */
export class StopChecksNavigationService {
    constructor(router, route) {
        this.router = router;
        this.route = route;
        this.defaultRouteParams = { relativeTo: this.route };
    }
    form(extras = {}) {
        this.router.navigate(['..', 'form'], Object.assign(Object.assign({}, this.defaultRouteParams), extras));
    }
    review(extras = {}) {
        this.router.navigate(['review'], Object.assign(Object.assign({}, this.defaultRouteParams), extras));
    }
}
/** @nocollapse */ StopChecksNavigationService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StopChecksNavigationService, deps: [{ token: i1.Router }, { token: i1.ActivatedRoute }], target: i0.ɵɵFactoryTarget.Injectable });
/** @nocollapse */ StopChecksNavigationService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StopChecksNavigationService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StopChecksNavigationService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.Router }, { type: i1.ActivatedRoute }]; } });
