// @backbase/stop-checks-journey-ang/esm2015/lib/services/stop-checks-journey-config.service.js
import { Injectable, Optional, Inject, InjectionToken } from '@angular/core';
import { LoadMoreMethod } from '@backbase/payment-orders-ang';
import * as i0 from "@angular/core";
export const StopChecksJourneyConfigurationToken = new InjectionToken('StopChecksJourneyConfiguration injection token');
export const stopChecksDefaultConfig = {
    currencies: ['USD'],
    pageSize: 50,
    maxNavPages: 3,
    notificationDismissTime: 3,
    paginationType: LoadMoreMethod.PAGINATION,
    checksRangeNumber: 0,
    shouldDisplayHeading: true,
    schemeName: 'BBAN',
};
export class StopChecksJourneyService {
    constructor(config) {
        this.config = config;
        this._config = Object.assign(Object.assign({}, stopChecksDefaultConfig), (config || {}));
    }
    get currencies() {
        return this._config.currencies;
    }
    get pageSize() {
        return this._config.pageSize;
    }
    get schemeName() {
        return this._config.schemeName;
    }
    get maxNavPages() {
        return this._config.maxNavPages;
    }
    get notificationDismissTime() {
        return this._config.notificationDismissTime;
    }
    get paginationType() {
        return this._config.paginationType;
    }
    get checksRangeNumber() {
        return this._config.checksRangeNumber;
    }
    get shouldDisplayHeading() {
        return this._config.shouldDisplayHeading;
    }
    get configuration() {
        return this._config;
    }
}
/** @nocollapse */ StopChecksJourneyService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StopChecksJourneyService, deps: [{ token: StopChecksJourneyConfigurationToken, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
/** @nocollapse */ StopChecksJourneyService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StopChecksJourneyService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StopChecksJourneyService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [StopChecksJourneyConfigurationToken]
                }] }]; } });
