// @backbase/budget-journey-ang/esm2015/src/budget-journey.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { provideRoutes, RouterModule } from '@angular/router';
import { BudgetWidgetModule } from '@backbase/budget-widget-ang';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { HeaderModule } from '@backbase/ui-ang/header';
import { BudgetJourneyComponent } from './budget-journey.component';
import { BudgetViewComponent } from './views/budgets-view.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
export const defaultRoute = {
    path: '',
    component: BudgetJourneyComponent,
    children: [
        {
            path: '',
            component: BudgetViewComponent,
        },
    ],
};
export class BudgetJourneyModule {
    static forRoot(data = { route: defaultRoute }) {
        return {
            ngModule: BudgetJourneyModule,
            providers: [provideRoutes([data.route])],
        };
    }
}
BudgetJourneyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetJourneyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
BudgetJourneyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetJourneyModule, declarations: [BudgetJourneyComponent, BudgetViewComponent], imports: [CommonModule, i1.BackbaseCoreModule, RouterModule,
        HeaderModule,
        BudgetWidgetModule] });
BudgetJourneyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetJourneyModule, imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { BudgetJourneyComponent },
            }),
            RouterModule,
            HeaderModule,
            BudgetWidgetModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetJourneyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [BudgetJourneyComponent, BudgetViewComponent],
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { BudgetJourneyComponent },
                        }),
                        RouterModule,
                        HeaderModule,
                        BudgetWidgetModule,
                    ],
                }]
        }] });
