// @backbase/budget-journey-ang/esm2015/src/budget-journey-config.service.js
import { Inject, Injectable, InjectionToken, Optional } from '@angular/core';
import * as i0 from "@angular/core";
// eslint-disable-next-line
export const BudgetJourneyConfigurationToken = new InjectionToken('BudgetJourneyConfiguration injection token');
const configDefaults = {
    showPercentage: false,
    budgetSafeZoneLimit: 80,
    notificationDismissTime: 5,
    maxBudgets: null,
};
export class BudgetJourneyService {
    constructor(config) {
        config = config || {};
        this._config = Object.assign(Object.assign({}, configDefaults), config);
    }
    get defaults() {
        return configDefaults;
    }
    get showPercentage() {
        return this._config.showPercentage;
    }
    get budgetSafeZoneLimit() {
        return this._config.budgetSafeZoneLimit;
    }
    get notificationDismissTime() {
        return this._config.notificationDismissTime;
    }
    get maxBudgets() {
        return this._config.maxBudgets;
    }
}
BudgetJourneyService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetJourneyService, deps: [{ token: BudgetJourneyConfigurationToken, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
BudgetJourneyService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetJourneyService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetJourneyService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [BudgetJourneyConfigurationToken]
                }] }]; } });
