// @backbase/budget-journey-ang/esm2015/src/budget-journey.component.js
import { Component } from '@angular/core';
import { BudgetJourneyService } from './budget-journey-config.service';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
export class BudgetJourneyComponent {
}
BudgetJourneyComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetJourneyComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
BudgetJourneyComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: BudgetJourneyComponent, selector: "bb-budget-journey", providers: [BudgetJourneyService], ngImport: i0, template: '<router-outlet></router-outlet>', isInline: true, directives: [{ type: i1.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BudgetJourneyComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-budget-journey',
                    template: '<router-outlet></router-outlet>',
                    providers: [BudgetJourneyService],
                }]
        }] });
