// @backbase/payment-orders-ang/esm2015/lib/configs/payment-types/sepa.js
import '@angular/localize/init';
import { Validators } from '@angular/forms';
import { PaymentComponents, } from '../models/form-items.model';
import { PaymentBaseFields, CounterPartyFields, RemittanceInfoFields, } from '../models/form-item-names.model';
import { defaultInitiatorConfig as initiator, defaultScheduleConfig as schedule, } from './default-config';
import { SepaCountries } from '../constants';
import { ibanCountryValidator } from '../helpers/form-validation';
const counterParty = {
    name: PaymentBaseFields.counterparty,
    fields: [
        {
            name: PaymentComponents.beneficiaryGroupHeader,
            type: PaymentComponents.beneficiaryGroupHeader,
            options: {
                label: $localize `:@@sepa-config.beneficiary.header-label:To`,
                hasDynamicTitle: true,
            },
        },
        {
            name: CounterPartyFields.name,
            type: PaymentComponents.beneficiarySelector,
            options: {
                placeholder: $localize `:@@sepa-config.beneficiary-selector-placeholder:Select a beneficiary`,
                saveNewContactLabel: $localize `:@@sepa-config.new-contact-label:Save as a new contact`,
                cssClasses: ['col-12'],
                validationMessages: [
                    {
                        message: $localize `:@@sepa-config.same-accounts-selected-error-message:Beneficiary is same as selected Debit account`,
                        name: 'sameDebitCreditAccountsSelected',
                    },
                ],
            },
        },
        {
            name: CounterPartyFields.accountNumber,
            type: PaymentComponents.text,
            options: {
                validators: [Validators.required, ibanCountryValidator(SepaCountries)],
                label: $localize `:@@sepa-config.beneficiary.iban-label:IBAN`,
                isIban: true,
                cssClasses: ['d-inline-block', 'col-md-6', 'align-top'],
                validationMessages: [
                    {
                        message: $localize `:@@sepa-config.iban-country-error-message:Invalid IBAN`,
                        name: 'invalidIbanCountry',
                    },
                ],
            },
        },
    ],
};
const remittanceInfo = {
    name: PaymentBaseFields.remittanceInfo,
    title: $localize `:@@sepa-config.remittance-info-title:Payment details`,
    fields: [
        {
            type: PaymentComponents.amount,
            name: RemittanceInfoFields.amountCurrencyGroup,
            options: {
                label: $localize `:@@sepa-config.amount-label:Amount`,
                currencies: ['EUR'],
                cssClasses: ['col-12'],
            },
        },
    ],
};
const paymentInformation = {
    name: 'paymentInformation',
    title: $localize `:@@sepa-config.information-title:Payment information`,
    fields: [
        {
            name: CounterPartyFields.remittanceInformation,
            type: PaymentComponents.textarea,
            options: {
                label: $localize `:@@sepa-config.remittance-info-label:Description`,
                helperText: $localize `:@@sepa-config.remittance-info-helper-text: (optional)`,
                placeholder: $localize `:@@sepa-config.remittance-info-placeholder:Enter payment description`,
                showCharCounter: true,
                minLength: 0,
                rows: 3,
                maxLength: 140,
                cols: 50,
                cssClasses: ['d-inline-block', 'col-md-8'],
                mapToField: PaymentBaseFields.counterparty,
            },
        },
    ],
};
export const SEPA = {
    fields: [
        initiator,
        remittanceInfo,
        counterParty,
        paymentInformation,
        schedule,
    ],
    name: $localize `:@@sepa-config.name:SEPA CT Transfer`,
    businessFunction: 'SEPA CT',
    paymentType: 'SEPA_CREDIT_TRANSFER',
    options: {
        closedPaymentBusFns: ['SEPA CT - closed,SEPA CT - Intracompany'],
    },
};
