// @backbase/payment-orders-ang/esm2015/lib/configs/payment-types/uk-faster-payment.js
import '@angular/localize/init';
import { Validators } from '@angular/forms';
import { PaymentComponents, } from '../models/form-items.model';
import { PaymentBaseFields, CounterPartyFields, RemittanceInfoFields, } from '../models/form-item-names.model';
import { defaultInitiatorConfig as initiator, defaultScheduleConfig as schedule, } from './default-config';
const counterParty = {
    name: PaymentBaseFields.counterparty,
    fields: [
        {
            name: PaymentComponents.beneficiaryGroupHeader,
            type: PaymentComponents.beneficiaryGroupHeader,
            options: {
                label: $localize `:@@fps-config.beneficiary.header-label:To`,
                hasDynamicTitle: true,
            },
        },
        {
            name: CounterPartyFields.name,
            type: PaymentComponents.beneficiarySelector,
            options: {
                placeholder: $localize `:@@fps-config.beneficiary-selector-placeholder:Select a beneficiary`,
                saveNewContactLabel: $localize `:@@fps-config.new-contact-label:Save as a new contact`,
                cssClasses: ['col-12'],
                validationMessages: [
                    {
                        message: $localize `:@@fps-config.same-accounts-selected-error-message:Beneficiary is same as selected Debit account`,
                        name: 'sameDebitCreditAccountsSelected',
                    },
                ],
            },
        },
        {
            name: CounterPartyFields.accountNumber,
            type: PaymentComponents.text,
            options: {
                validators: [Validators.required],
                label: $localize `:@@fps-config.beneficiary.account-number-label:Account number`,
                cssClasses: ['d-inline-block', 'col-md-6', 'align-top'],
            },
        },
        {
            name: CounterPartyFields.creditorBank,
            fields: [
                {
                    name: CounterPartyFields.bankBranchCode,
                    type: PaymentComponents.text,
                    options: {
                        label: $localize `:@@fps-config.sort-code-label:Sort code`,
                        cssClasses: ['d-inline-block', 'col-md-6', 'align-top'],
                        validators: [Validators.required],
                    },
                },
            ],
        },
    ],
};
const remittanceInfo = {
    name: PaymentBaseFields.remittanceInfo,
    title: $localize `:@@fps-config.remittance-info-title:Payment details`,
    fields: [
        {
            type: PaymentComponents.amount,
            name: RemittanceInfoFields.amountCurrencyGroup,
            options: {
                label: $localize `:@@fps-config.amount-label:Amount`,
                currencies: ['GBP'],
                cssClasses: ['col-md-8'],
            },
        },
    ],
};
const paymentInformation = {
    name: 'paymentInformation',
    title: $localize `:@@fps-config.information-title:Payment information`,
    fields: [
        {
            name: CounterPartyFields.remittanceInformation,
            type: PaymentComponents.textarea,
            options: {
                label: $localize `:@@fps-config.remittance-info-label:Description`,
                placeholder: $localize `:@@fps-config.remittance-info-placeholder:Enter payment description`,
                showCharCounter: true,
                minLength: 0,
                rows: 3,
                maxLength: 140,
                cols: 50,
                cssClasses: ['d-inline-block', 'col-12'],
                mapToField: PaymentBaseFields.counterparty,
            },
        },
    ],
};
export const UK_FASTER_PAYMENT = {
    fields: [
        initiator,
        remittanceInfo,
        counterParty,
        paymentInformation,
        schedule,
    ],
    name: $localize `:@@fps-config.name:UK Faster Payment`,
    paymentType: 'FASTER_PAYMENT',
    businessFunction: 'UK Faster Payments',
};
