// @backbase/payment-orders-ang/esm2015/lib/configs/payment-types/us-ach-credit.js
import '@angular/localize/init';
import { Validators } from '@angular/forms';
import { PaymentComponents, } from '../models/form-items.model';
import { PaymentBaseFields, CounterPartyFields, RemittanceInfoFields, InitiatorFields, } from '../models/form-item-names.model';
import { routingTransitNumberValidator } from '../../ui/helpers/form-validation';
import { defaultScheduleConfig as schedule } from './default-config';
export const initiator = {
    name: PaymentBaseFields.initiator,
    title: $localize `:@@ach-config.initiator-title:From account`,
    fields: [
        {
            type: PaymentComponents.debitAccountSelector,
            name: InitiatorFields.initiatorAccountGroup,
            options: {
                placeholder: $localize `:@@ach-config.initiator-placeholder:Select an account`,
                cssClasses: ['col-12'],
                validationMessages: [
                    {
                        message: $localize `:@@ach-config.invalid-account-message:Debit account is invalid`,
                        name: 'missingInList',
                    },
                    {
                        message: $localize `:@@ach-config.closed-permission-error-message:You can not do payments to new beneficiary from this account.`,
                        name: 'hasOnlyClosedPaymentPermission',
                    },
                ],
            },
        },
    ],
};
const counterParty = {
    name: PaymentBaseFields.counterparty,
    fields: [
        {
            name: PaymentComponents.beneficiaryGroupHeader,
            type: PaymentComponents.beneficiaryGroupHeader,
            options: {
                label: $localize `:@@ach-config.beneficiary.header-label:Transfer to`,
                hasDynamicTitle: true,
            },
        },
        {
            name: CounterPartyFields.name,
            type: PaymentComponents.beneficiarySelector,
            options: {
                placeholder: $localize `:@@ach-config.beneficiary-selector-placeholder:Enter beneficiary's name or account`,
                cssClasses: ['col-12'],
                saveNewContactLabel: $localize `:@@ach-config.new-contact-label:Save as a new contact`,
                label: $localize `:@@ach-config.beneficiary-selector-label:Beneficiary name`,
                dropdownIcon: 'perm-contact-calendar',
                dependants: [
                    CounterPartyFields.accountNumber,
                    CounterPartyFields.accountType,
                    CounterPartyFields.creditorBank,
                    CounterPartyFields.standardEntryClass,
                    CounterPartyFields.recipientId,
                ],
                validationMessages: [
                    {
                        message: $localize `:@@ach-config.same-accounts-selected-error-message:Beneficiary is same as selected Debit account`,
                        name: 'sameDebitCreditAccountsSelected',
                    },
                ],
            },
        },
        {
            name: CounterPartyFields.accountNumber,
            type: PaymentComponents.text,
            options: {
                validators: [Validators.required],
                label: $localize `:@@ach-config.beneficiary.account-number-label:Account number`,
                maxLength: 34,
                cssClasses: ['d-inline-block', 'col-md-6', 'align-top'],
            },
        },
        {
            name: CounterPartyFields.accountType,
            type: PaymentComponents.select,
            options: {
                label: $localize `:@@ach-config.beneficiary.account-type-label:Account type`,
                validators: [Validators.required],
                options: [
                    {
                        label: $localize `:@@ach-config.beneficiary.savings-account-label:Savings`,
                        value: 'SAVINGS',
                    },
                    {
                        label: $localize `:@@ach-config.beneficiary.checking-account-label:Checking`,
                        value: 'CHECKING',
                    },
                ],
                placeholder: $localize `:@@ach-config.beneficiary-account-type-placeholder:Select an account type`,
                cssClasses: ['d-inline-block', 'col-md-6', 'align-top'],
            },
        },
        {
            name: CounterPartyFields.creditorBank,
            fields: [
                {
                    name: CounterPartyFields.bankBranchCode,
                    type: PaymentComponents.text,
                    options: {
                        label: $localize `:@@ach-config.branch-code-label:Routing transit number (ABA)`,
                        maxLength: 9,
                        cssClasses: ['d-inline-block', 'col-md-6', 'align-top'],
                        validators: [Validators.required, routingTransitNumberValidator],
                    },
                },
                {
                    name: CounterPartyFields.standardEntryClass,
                    type: PaymentComponents.select,
                    options: {
                        label: $localize `:@@ach-config.entry-class-label:Standard entry class (SEC)`,
                        validators: [Validators.required],
                        placeholder: $localize `:@@ach-config.entry-class-placeholder:Select SEC type`,
                        cssClasses: ['d-inline-block', 'col-md-6', 'align-top'],
                        mapToField: PaymentBaseFields.counterparty,
                        options: [
                            {
                                label: $localize `:@@ach-config.ppd-class-label:Personal (PPD)`,
                                value: 'PPD',
                            },
                            {
                                label: $localize `:@@ach-config.ccd-class-label:Business (CCD)`,
                                value: 'CCD',
                            },
                        ],
                    },
                },
            ],
        },
        {
            name: CounterPartyFields.recipientId,
            type: PaymentComponents.text,
            options: {
                label: $localize `:@@ach-config.recipient-id-label:Recipient ID`,
                helperText: $localize `:@@ach-config.recipient-id-helper-text: (optional)`,
                cssClasses: ['d-inline-block', 'col-md-6', 'align-top'],
                maxLength: 15,
                showCharCounter: true,
            },
        },
    ],
};
const remittanceInfo = {
    name: PaymentBaseFields.remittanceInfo,
    title: $localize `:@@ach-config.remittance-info-title:Payment details`,
    fields: [
        {
            type: PaymentComponents.amount,
            name: RemittanceInfoFields.amountCurrencyGroup,
            options: {
                label: $localize `:@@ach-config.amount-label:Amount`,
                currencies: ['USD'],
                cssClasses: ['col-md-8'],
                validationMessages: [
                    {
                        name: 'invalidAmount',
                        message: $localize `:@@ach-config.invalid-amount-message:Amount is required`,
                    },
                ],
            },
        },
        {
            type: PaymentComponents.checkbox,
            name: RemittanceInfoFields.urgent,
            options: {
                label: $localize `:@@ach-config.urgent-payment-label:This payment is urgent`,
                cssClasses: ['col-md-8'],
                hideInWizardMode: true,
            },
        },
        {
            name: RemittanceInfoFields.paymentReference,
            type: PaymentComponents.text,
            options: {
                label: $localize `:@@ach-config.reference-label:Payment reference`,
                helperText: $localize `:@@ach-config.reference-placeholder: (optional)`,
                showCharCounter: true,
                maxLength: 35,
                cssClasses: ['d-inline-block', 'col-md-8'],
            },
        },
    ],
};
export const US_ACH_CREDIT = {
    fields: [initiator, remittanceInfo, counterParty, schedule],
    name: $localize `:@@ach-config.name:ACH Transfer`,
    businessFunction: 'ACH Credit Transfer',
    paymentType: 'ACH_CREDIT',
};
