// @backbase/ui-ang/esm2015/_example/progressbar/showcase/progressbar-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ProgressbarShowcaseComponent } from './progressbar-showcase.component';
import { ProgressbarModule } from '@backbase/ui-ang/progressbar';
import * as i0 from "@angular/core";
export class ProgressbarShowcaseModule {
}
ProgressbarShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProgressbarShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProgressbarShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProgressbarShowcaseModule, declarations: [ProgressbarShowcaseComponent], imports: [CommonModule, ProgressbarModule] });
ProgressbarShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProgressbarShowcaseModule, imports: [[CommonModule, ProgressbarModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProgressbarShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ProgressbarModule],
                    declarations: [ProgressbarShowcaseComponent],
                    entryComponents: [ProgressbarShowcaseComponent],
                }]
        }] });
