// @backbase/contact-common-ang/esm2015/lib/directives/email-domain/email-domain-validator.directive.js
import { Directive } from '@angular/core';
import { NG_VALIDATORS } from '@angular/forms';
import * as i0 from "@angular/core";
/**
 * This is a directive to validate presence of email top-level domain
 *
 * @usageNotes
 *
 * ```html
 * <input type="email" #ibanControl="ngModel" [(ngModel)]="iban" bbDomainValidator />
 * ```
 * @ngModule EmailDomainModule
 */
export class EmailDomainValidatorDirective {
    constructor() {
        this.topLevelDomainRegex = /@.*\.(([a-zA-Z])|([a-zA-Z0-9][a-zA-Z0-9.\-]*[a-zA-Z0-9]))$/;
    }
    /**
     * Performs validation
     *
     * @param {AbstractControl} control form control
     * @returns {Record<string, any>?} form control error object
     */
    validate(control) {
        const value = control === null || control === void 0 ? void 0 : control.value;
        if (!value)
            return null;
        return this.topLevelDomainRegex.test(value) ? null : { email: { value } };
    }
}
/** @nocollapse */ EmailDomainValidatorDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EmailDomainValidatorDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
/** @nocollapse */ EmailDomainValidatorDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: EmailDomainValidatorDirective, selector: "[bbEmailDomainValidator]", providers: [{ provide: NG_VALIDATORS, useExisting: EmailDomainValidatorDirective, multi: true }], ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EmailDomainValidatorDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbEmailDomainValidator]',
                    providers: [{ provide: NG_VALIDATORS, useExisting: EmailDomainValidatorDirective, multi: true }],
                }]
        }] });
