// @backbase/ui-ang/esm2015/_example/control-error-handler/showcase/custom-error/custom-error-showcase.component.js
import { Component } from '@angular/core';
import { animate, style, transition, trigger } from '@angular/animations';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class CustomErrorShowcaseComponent {
    constructor() {
        this.errorId = 'testID';
    }
}
CustomErrorShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomErrorShowcaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
CustomErrorShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: CustomErrorShowcaseComponent, selector: "bb-custom-error-showcase", ngImport: i0, template: `
    <small @fade [id]="errorId" *ngIf="text" role="alert" class="bb-input-validation-message">
      Custom error component - {{ text }}
    </small>
  `, isInline: true, directives: [{ type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], animations: [
        trigger('fade', [
            transition('void => *', [style({ opacity: 0, height: '0px' }), animate(150, style({ opacity: 1, height: '*' }))]),
        ]),
    ] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CustomErrorShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-custom-error-showcase',
                    template: `
    <small @fade [id]="errorId" *ngIf="text" role="alert" class="bb-input-validation-message">
      Custom error component - {{ text }}
    </small>
  `,
                    animations: [
                        trigger('fade', [
                            transition('void => *', [style({ opacity: 0, height: '0px' }), animate(150, style({ opacity: 1, height: '*' }))]),
                        ]),
                    ],
                }]
        }] });
export const customErrorMessagesShowcase = { required: () => 'My Field is required' };
