// @backbase/ui-ang/esm2015/_example/control-error-handler/showcase/custom-error/input-with-custom-errors-showcase.component.js
import { Component } from '@angular/core';
import { FormControl, Validators } from '@angular/forms';
import { BB_DYNAMIC_VALIDATION_ERROR_TMPL, BB_VALIDATION_ERRORS } from '@backbase/ui-ang/control-error-handler';
import { customErrorMessagesShowcase, CustomErrorShowcaseComponent } from './custom-error-showcase.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/input-text";
import * as i2 from "@backbase/ui-ang/control-error-handler";
import * as i3 from "@angular/forms";
export class InputWithCustomErrorComponent {
    constructor() {
        this.inputField = new FormControl(null, [Validators.required]);
    }
}
InputWithCustomErrorComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputWithCustomErrorComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
InputWithCustomErrorComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: InputWithCustomErrorComponent, selector: "bb-input-with-custom-error-showcase", providers: [
        {
            provide: BB_VALIDATION_ERRORS,
            useValue: customErrorMessagesShowcase,
        },
        {
            provide: BB_DYNAMIC_VALIDATION_ERROR_TMPL,
            useValue: CustomErrorShowcaseComponent,
        },
    ], ngImport: i0, template: `
    <div>
      <bb-input-text-ui
        bbFormControl
        label="Input with custom error component using InjectionToken"
        [formControl]="inputField"
        placeholder="Please enter a value, at most 35 chars..."
      >
      </bb-input-text-ui>
    </div>
  `, isInline: true, components: [{ type: i1.InputTextComponent, selector: "bb-input-text-ui", inputs: ["placeholder", "maxLength", "minLength", "autocomplete", "readonly", "mask", "maskValidator", "showCharCounter"], outputs: ["valueChange"] }], directives: [{ type: i2.ValidationErrorsDirective, selector: "[bbFormControl]", inputs: ["errorLabels", "showError", "inputSelector", "label", "errorComponent"] }, { type: i3.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { type: i3.FormControlDirective, selector: "[formControl]", inputs: ["disabled", "formControl", "ngModel"], outputs: ["ngModelChange"], exportAs: ["ngForm"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputWithCustomErrorComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-input-with-custom-error-showcase',
                    template: `
    <div>
      <bb-input-text-ui
        bbFormControl
        label="Input with custom error component using InjectionToken"
        [formControl]="inputField"
        placeholder="Please enter a value, at most 35 chars..."
      >
      </bb-input-text-ui>
    </div>
  `,
                    providers: [
                        {
                            provide: BB_VALIDATION_ERRORS,
                            useValue: customErrorMessagesShowcase,
                        },
                        {
                            provide: BB_DYNAMIC_VALIDATION_ERROR_TMPL,
                            useValue: CustomErrorShowcaseComponent,
                        },
                    ],
                }]
        }] });
