// @backbase/loans-collection-ui-ang/esm2015/lib/main-container/loans-main-container.component.js
import { ChangeDetectionStrategy, Component, Input, TemplateRef } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/loading-indicator";
import * as i2 from "@backbase/ui-ang/empty-state";
import * as i3 from "@angular/common";
export class LoansMainContainerComponent {
    constructor() {
        /**
         * Flag represents error state
         *
         * @default: false
         */
        this.isErrorState = false;
        /**
         * Flag represents loading  state
         *
         * @default: false
         */
        this.isLoadingState = false;
        /**
         * Flag represents empty state
         *
         * @default: false
         */
        this.isEmptyState = false;
        /**
         * Flag will add the additional class for state containers to have some min height
         * Applicable for empty error and loading default states only
         *
         * @default: false
         */
        this.containerHeightFitToContent = false;
        /**
         * Classes for inner content element wrapper
         *
         * @default: card-body
         */
        this.contentWrapperClasses = 'bb-card__body';
        /**
         * Classes for root wrapper element
         */
        this.rootWrapperClasses = 'bb-card';
    }
}
/** @nocollapse */ LoansMainContainerComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoansMainContainerComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ LoansMainContainerComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: LoansMainContainerComponent, selector: "bb-loans-main-container", inputs: { isErrorState: "isErrorState", isLoadingState: "isLoadingState", isEmptyState: "isEmptyState", containerHeightFitToContent: "containerHeightFitToContent", contentWrapperClasses: "contentWrapperClasses", rootWrapperClasses: "rootWrapperClasses", customErrorState: "customErrorState", customLoadingState: "customLoadingState", customEmptyState: "customEmptyState" }, ngImport: i0, template: "<div [ngClass]=\"rootWrapperClasses\">\n  <div [ngClass]=\"contentWrapperClasses\">\n    <ng-container *ngIf=\"!isErrorState; else customErrorState || errorStateDefault\">\n      <ng-container *ngIf=\"!isLoadingState; else customLoadingState || loadingStateDefault\">\n        <ng-container *ngIf=\"!isEmptyState; else customEmptyState || emptyStateDefault\">\n          <ng-content></ng-content>\n        </ng-container>\n      </ng-container>\n    </ng-container>\n  </div>\n</div>\n\n<ng-template #loadingStateDefault>\n  <div\n    class=\"bb-state-container\"\n    [class.bb-state-container--min-height]=\"!containerHeightFitToContent\"\n    data-role=\"loans-loading-state-default\"\n  >\n    <bb-loading-indicator-ui loaderSize=\"md\"></bb-loading-indicator-ui>\n  </div>\n</ng-template>\n\n<ng-template #errorStateDefault>\n  <div\n    class=\"bb-state-container\"\n    [class.bb-state-container--min-height]=\"!containerHeightFitToContent\"\n    data-role=\"loans-error-state-default\"\n  >\n    <bb-empty-state-ui\n      title=\"Unknown Error\"\n      i18n-title=\"Error state title@@loans.state.error.title\"\n      subtitle=\"Try to reload the page or contact the system administrator.\"\n      i18n-subtitle=\"Error state subtitle@@loans.state.error.subtitle\"\n      iconSize=\"xxl\"\n      iconModifier=\"error-outline\"\n    >\n    </bb-empty-state-ui>\n  </div>\n</ng-template>\n\n<ng-template #emptyStateDefault>\n  <div\n    class=\"bb-state-container\"\n    [class.bb-state-container--min-height]=\"!containerHeightFitToContent\"\n    data-role=\"loans-empty-state-default\"\n  >\n    <bb-empty-state-ui\n      title=\"No Data\"\n      i18n-title=\"Empty state title@@loans.state.empty.title\"\n      iconModifier=\"currency-trade\"\n    ></bb-empty-state-ui>\n  </div>\n</ng-template>\n", components: [{ type: i1.LoadingIndicatorComponent, selector: "bb-loading-indicator-ui", inputs: ["text", "loaderSize", "showDelay", "hasBackground", "inline"] }, { type: i2.EmptyStateComponent, selector: "bb-empty-state-ui", inputs: ["title", "subtitle", "showIcon", "iconClasses", "iconModifier", "iconSize", "iconColor"] }], directives: [{ type: i3.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoansMainContainerComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-loans-main-container',
                    templateUrl: './loans-main-container.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { isErrorState: [{
                type: Input
            }], isLoadingState: [{
                type: Input
            }], isEmptyState: [{
                type: Input
            }], containerHeightFitToContent: [{
                type: Input
            }], contentWrapperClasses: [{
                type: Input
            }], rootWrapperClasses: [{
                type: Input
            }], customErrorState: [{
                type: Input
            }], customLoadingState: [{
                type: Input
            }], customEmptyState: [{
                type: Input
            }] } });
