// @backbase/loans-collection-ui-ang/esm2015/lib/main-container/loans-main-container.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { LoansMainContainerComponent } from './loans-main-container.component';
import * as i0 from "@angular/core";
export class LoansMainContainerModule {
}
/** @nocollapse */ LoansMainContainerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoansMainContainerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoansMainContainerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoansMainContainerModule, declarations: [LoansMainContainerComponent], imports: [CommonModule, LoadingIndicatorModule, EmptyStateModule], exports: [LoansMainContainerComponent] });
/** @nocollapse */ LoansMainContainerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoansMainContainerModule, imports: [[CommonModule, LoadingIndicatorModule, EmptyStateModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoansMainContainerModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoansMainContainerComponent],
                    imports: [CommonModule, LoadingIndicatorModule, EmptyStateModule],
                    exports: [LoansMainContainerComponent],
                }]
        }] });
