// @backbase/cards-common-ang/esm2015/src/services/card.service.js
import { Injectable } from '@angular/core';
import { ReplaySubject } from 'rxjs';
import { catchError, distinctUntilChanged, filter, map, publishReplay, refCount, switchMap } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/cards";
import * as i2 from "./cards-backend.service";
const cacheRequest = (request) => (source) => source.pipe(switchMap(request), publishReplay(1), refCount());
/**
 * @name CardService
 *
 * @description
 * Service for retrieving and updating a single card from the backend.
 */
export class CardService {
    constructor(cardsDataHttpService, cardsBackendService) {
        this.cardsDataHttpService = cardsDataHttpService;
        this.cardsBackendService = cardsBackendService;
        this.paymentCardId = new ReplaySubject(1);
        this.paymentCard = this.paymentCardId.pipe(distinctUntilChanged(), filter((paymentCardId) => typeof paymentCardId !== 'undefined'), cacheRequest((paymentCardId) => this.cardsDataHttpService.getCardById({ id: paymentCardId }, 'response').pipe(map((response) => {
            if (response.body === null) {
                throw new Error('Some error message');
            }
            return this.cardsBackendService.itemToModel(response.body);
        }), catchError((error) => {
            throw this.cardsBackendService.parseError(error);
        }))));
    }
    getPaymentCardFrom(paymentCardId) {
        paymentCardId.subscribe(this.paymentCardId);
    }
    getPaymentCard() {
        return this.paymentCard;
    }
}
CardService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardService, deps: [{ token: i1.CardsHttpService }, { token: i2.CardsBackendService }], target: i0.ɵɵFactoryTarget.Injectable });
CardService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.CardsHttpService }, { type: i2.CardsBackendService }]; } });
