// @backbase/ui-ang/esm2015/phone-number-format-pipe/phone-number-format.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PhoneNumberPipe } from './phone-number-format.pipe';
import * as i0 from "@angular/core";
export class PhoneNumberModule {
}
PhoneNumberModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PhoneNumberModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PhoneNumberModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PhoneNumberModule, declarations: [PhoneNumberPipe], imports: [CommonModule], exports: [PhoneNumberPipe] });
PhoneNumberModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PhoneNumberModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PhoneNumberModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [PhoneNumberPipe],
                    exports: [PhoneNumberPipe],
                }]
        }] });
