// @backbase/payment-orders-ang/esm2015/lib/ui/components/card-menu/card-menu.component.js
import { Component, Input, Output, EventEmitter, } from '@angular/core';
import { NgbTooltipConfig } from '@ng-bootstrap/ng-bootstrap';
import * as i0 from "@angular/core";
import * as i1 from "@ng-bootstrap/ng-bootstrap";
import * as i2 from "@backbase/ui-ang/ellipsis";
import * as i3 from "@angular/common";
import * as i4 from "@backbase/ui-ang/keyboard-click-directive";
export class PayordCardMenuComponent {
    constructor(config) {
        /**
         * List of tab items.
         * Default is empty list.
         */
        this.items = [];
        /**
         * Initial selected tab item.
         * Default is `0`.
         */
        this.initialSelection = 0;
        /**
         * Select event emitted when a new tab item has been selected.
         */
        // eslint-disable-next-line @angular-eslint/no-output-native
        this.select = new EventEmitter();
        this.activeItem = 0;
        config.container = 'body';
    }
    ngAfterViewInit() {
        if (this.items) {
            this.activeItem = this.items[this.initialSelection]
                ? this.initialSelection
                : 0;
        }
    }
    onSelect(index) {
        this.activeItem = index;
        this.select.emit(this.items[index]);
    }
    trackByName(index, item) {
        return item.name;
    }
}
/** @nocollapse */ PayordCardMenuComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordCardMenuComponent, deps: [{ token: i1.NgbTooltipConfig }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ PayordCardMenuComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: PayordCardMenuComponent, selector: "bb-payord-card-menu", inputs: { items: "items", initialSelection: "initialSelection" }, outputs: { select: "select" }, providers: [NgbTooltipConfig], ngImport: i0, template: "<div class=\"bb-card-menu bb-stack manual-batches-type-select overflow-auto\">\n  <div\n    class=\"bb-card-menu__option bb-stack__item manual-batches-type-select__button\"\n    [class.selected]=\"i === activeItem\"\n    role=\"button\"\n    tabindex=\"0\"\n    (bbKeyboardClick)=\"onSelect(i)\"\n    *ngFor=\"let item of items; trackBy: trackByName; let i = index\"\n  >\n    <bb-ellipsis-ui\n      [text]=\"item.name\"\n      tooltipTriggers=\"hover\"\n      class=\"bb-text-bold\"\n    >\n    </bb-ellipsis-ui>\n    <ng-container *ngIf=\"item.description\">\n      <bb-ellipsis-ui\n        [text]=\"item.description\"\n        tooltipTriggers=\"hover\"\n        [lineClamp]=\"2\"\n        class=\"bb-subheader bb-subheader--regular\"\n      ></bb-ellipsis-ui>\n    </ng-container>\n  </div>\n</div>\n", components: [{ type: i2.EllipsisComponent, selector: "bb-ellipsis-ui", inputs: ["text", "tooltipTriggers", "lineClamp"] }], directives: [{ type: i3.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { type: i4.KeyboardClickDirective, selector: "[bbKeyboardClick]", outputs: ["bbKeyboardClick"] }, { type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordCardMenuComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-payord-card-menu',
                    templateUrl: 'card-menu.component.html',
                    providers: [NgbTooltipConfig],
                }]
        }], ctorParameters: function () { return [{ type: i1.NgbTooltipConfig }]; }, propDecorators: { items: [{
                type: Input
            }], initialSelection: [{
                type: Input
            }], select: [{
                type: Output
            }] } });
