// @backbase/user-manage-profile-widget-ang/esm2015/src/user-profile-forms/user-profile-form-errors/user-profile-form-errors.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { AlertModule } from '@backbase/ui-ang/alert';
import { UserProfileFormErrorsComponent } from './user-profile-form-errors.component';
import * as i0 from "@angular/core";
const uiModules = [AlertModule];
export class UserProfileFormErrorsModule {
}
UserProfileFormErrorsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormErrorsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
UserProfileFormErrorsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormErrorsModule, declarations: [UserProfileFormErrorsComponent], imports: [CommonModule, AlertModule], exports: [UserProfileFormErrorsComponent, AlertModule] });
UserProfileFormErrorsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormErrorsModule, imports: [[CommonModule, ...uiModules], AlertModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormErrorsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules],
                    declarations: [UserProfileFormErrorsComponent],
                    exports: [UserProfileFormErrorsComponent, ...uiModules],
                }]
        }] });
