// @backbase/user-manage-profile-widget-ang/esm2015/src/user-profile-forms/user-profile-form-errors/user-profile-form-errors.component.js
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/alert";
import * as i2 from "@angular/common";
export class UserProfileFormErrorsComponent {
    get hasErrors() {
        if (!this.formGroup) {
            throw new Error('formGroup must be initialised to type FormGroup');
        }
        return Boolean(this.formGroup && this.formGroup.touched && this.hasBeenSubmittedWithErrors);
    }
}
UserProfileFormErrorsComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormErrorsComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
UserProfileFormErrorsComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: UserProfileFormErrorsComponent, selector: "bb-user-profile-form-errors", inputs: { formGroup: "formGroup", hasBeenSubmittedWithErrors: "hasBeenSubmittedWithErrors" }, ngImport: i0, template: "<bb-alert-ui *ngIf=\"hasErrors\" i18n-title=\"Form errors title|Title for the form errors list@@bb-user-email-address-form.errors.title\"\n  title=\"There is a problem, please check the form\">\n  <ng-content></ng-content>\n</bb-alert-ui>", components: [{ type: i1.AlertComponent, selector: "bb-alert-ui", inputs: ["modifier", "dismissible", "title", "message"], outputs: ["close"] }], directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserProfileFormErrorsComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-user-profile-form-errors',
                    templateUrl: './user-profile-form-errors.component.html',
                }]
        }], propDecorators: { formGroup: [{
                type: Input
            }], hasBeenSubmittedWithErrors: [{
                type: Input
            }] } });
