// @backbase/user-common/esm2015/src/components/manage-profile/manage-email-addresses/manage-email-address.component.js
import { Component, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
export class ManageEmailAddressComponent {
    constructor() {
        this.updateProfile = new EventEmitter();
        this.manageEmail = new EventEmitter();
        this.cancelManageEmail = new EventEmitter();
        this.managingEmailAddress = false;
    }
    handleManageEmailAddress() {
        if (this.disableManage) {
            return;
        }
        this.managingEmailAddress = true;
        this.manageEmail.emit();
    }
    handleCloseEditEmailAddressAfterSave() {
        this.updateProfile.emit();
    }
    handleCloseEditEmailAddress() {
        this.managingEmailAddress = false;
        this.disableManage = false;
        this.cancelManageEmail.emit();
    }
}
ManageEmailAddressComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageEmailAddressComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ManageEmailAddressComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: ManageEmailAddressComponent, selector: "ng-component", inputs: { electronicAddress: "electronicAddress", emailAddressCount: "emailAddressCount", emailAddressTypes: "emailAddressTypes", disableManage: "disableManage", dismissTimeout: "dismissTimeout" }, outputs: { updateProfile: "updateProfile", manageEmail: "manageEmail", cancelManageEmail: "cancelManageEmail" }, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageEmailAddressComponent, decorators: [{
            type: Component,
            args: [{
                    template: '',
                }]
        }], propDecorators: { electronicAddress: [{
                type: Input
            }], emailAddressCount: [{
                type: Input
            }], emailAddressTypes: [{
                type: Input
            }], disableManage: [{
                type: Input
            }], dismissTimeout: [{
                type: Input
            }], updateProfile: [{
                type: Output
            }], manageEmail: [{
                type: Output
            }], cancelManageEmail: [{
                type: Output
            }] } });
