// @backbase/user-common/esm2015/src/components/manage-profile/manage-email-addresses/manage-email-addresses.component.js
import { Component, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
export class ManageEmailAddressesComponent {
    constructor() {
        this.updateProfile = new EventEmitter();
        this.manageEmail = new EventEmitter();
        this.cancelManageEmail = new EventEmitter();
        this.isManagingEmailAddressIndex = undefined;
        this.isAddingEmailAddress = false;
        this.isAddButtonAvailable = true;
    }
    ngOnInit() {
        if (this.emailAddresses && this.maxEmailAddresses) {
            this.isAddButtonAvailable = this.emailAddresses.length < this.maxEmailAddresses;
        }
    }
    handleManageEmailOpen(index) {
        this.isManagingEmailAddressIndex = index;
        this.manageEmail.emit();
    }
    handleManageEmailClosed() {
        this.isManagingEmailAddressIndex = undefined;
        this.isAddingEmailAddress = false;
        this.cancelManageEmail.emit();
    }
    handleAddNewEmailAddress() {
        if (this.disableManage) {
            return;
        }
        this.isAddingEmailAddress = true;
        this.manageEmail.emit();
    }
    updateUserProfileData() {
        this.updateProfile.emit();
    }
}
ManageEmailAddressesComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageEmailAddressesComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ManageEmailAddressesComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: ManageEmailAddressesComponent, selector: "ng-component", inputs: { disableManage: "disableManage", emailAddresses: "emailAddresses", emailAddressTypes: "emailAddressTypes", maxEmailAddresses: "maxEmailAddresses", dismissTimeout: "dismissTimeout" }, outputs: { updateProfile: "updateProfile", manageEmail: "manageEmail", cancelManageEmail: "cancelManageEmail" }, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageEmailAddressesComponent, decorators: [{
            type: Component,
            args: [{
                    template: '',
                }]
        }], propDecorators: { disableManage: [{
                type: Input
            }], emailAddresses: [{
                type: Input
            }], emailAddressTypes: [{
                type: Input
            }], maxEmailAddresses: [{
                type: Input
            }], dismissTimeout: [{
                type: Input
            }], updateProfile: [{
                type: Output
            }], manageEmail: [{
                type: Output
            }], cancelManageEmail: [{
                type: Output
            }] } });
