// @backbase/user-common/esm2015/src/components/manage-profile/manage-email-addresses/new-email-address.component.js
import { Component, Input } from '@angular/core';
import { NewAddressComponent } from '../new-address/new-address.component';
import * as i0 from "@angular/core";
export class NewEmailAddressComponent extends NewAddressComponent {
    ngOnInit() {
        this.newEmailAddressData = {
            type: this.emailAddressTypes && this.emailAddressTypes.length ? this.emailAddressTypes[0] : '',
            address: '',
        };
    }
}
NewEmailAddressComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: NewEmailAddressComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
NewEmailAddressComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: NewEmailAddressComponent, selector: "ng-component", inputs: { emailAddressTypes: "emailAddressTypes" }, usesInheritance: true, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: NewEmailAddressComponent, decorators: [{
            type: Component,
            args: [{
                    template: '',
                }]
        }], propDecorators: { emailAddressTypes: [{
                type: Input
            }] } });
