// @backbase/data-ang/fesm2015/backbase-data-ang-comments.js
import { BASE_PATH, ThreadDto, CommentsService, APIS, ApiModule, COLLECTION_FORMATS, Configuration } from '@backbase/comments-http-ang';

/**
 * @deprecated The entry point `@backbase/data-ang/comments` will be removed in the future.
 *             Use `@backbase/data-ang/comments-v1` or `@backbase/data-ang/comments-v2` instead.
 */
const DEPRECATED_BASE_PATH = BASE_PATH;
/**
 * @deprecated The entry point `@backbase/data-ang/comments` will be removed in the future.
 *             Use `@backbase/data-ang/comments-v1` or `@backbase/data-ang/comments-v2` instead.
 */
const DeprecatedThreadDto = ThreadDto;
/**
 * @deprecated The entry point `@backbase/data-ang/comments` will be removed in the future.
 *             Use `@backbase/data-ang/comments-v1` or `@backbase/data-ang/comments-v2` instead.
 */
const DeprecatedCommentsService = CommentsService;
/**
 * @deprecated The entry point `@backbase/data-ang/comments` will be removed in the future.
 *             Use `@backbase/data-ang/comments-v1` or `@backbase/data-ang/comments-v2` instead.
 */
const DEPRECATED_APIS = APIS;
/**
 * @deprecated The entry point `@backbase/data-ang/comments` will be removed in the future.
 *             Use `@backbase/data-ang/comments-v1` or `@backbase/data-ang/comments-v2` instead.
 */
const DeprecatedApiModule = ApiModule;
/**
 * @deprecated The entry point `@backbase/data-ang/comments` will be removed in the future.
 *             Use `@backbase/data-ang/comments-v1` or `@backbase/data-ang/comments-v2` instead.
 */
const DEPRECATED_COLLECTION_FORMATS = COLLECTION_FORMATS;
/**
 * @deprecated The entry point `@backbase/data-ang/comments` will be removed in the future.
 *             Use `@backbase/data-ang/comments-v1` or `@backbase/data-ang/comments-v2` instead.
 */
const DeprecatedConfiguration = Configuration;

/**
 * Generated bundle index. Do not edit.
 */

export { DEPRECATED_APIS as APIS, DeprecatedApiModule as ApiModule, DEPRECATED_BASE_PATH as BASE_PATH, DEPRECATED_COLLECTION_FORMATS as COLLECTION_FORMATS, DeprecatedCommentsService as CommentsService, DeprecatedConfiguration as Configuration, DeprecatedThreadDto as ThreadDto };
