// @backbase/product-summary-ui/esm2015/src/bb-credit-limit-ui/credit-limit.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AmountModule } from '@backbase/ui-ang/amount';
import { BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import { FormatPipeModule, ValidationModule } from '@backbase/product-summary-common-ang';
import { CreditLimitComponent } from './credit-limit.component';
import { CreditLimitAttributeComponent } from './credit-limit-attribute/credit-limit-attribute.component';
import { CreditLimitStatusBarComponent } from './credit-limit-status-bar/credit-limit-status-bar.component';
import { CreditLimitMinimumPaymentComponent } from './credit-limit-minimum-payment/credit-limit-minimum-payment.component';
import * as i0 from "@angular/core";
const uiModules = [AmountModule, BbDatePipeModule, FormatPipeModule, ValidationModule];
const components = [
    CreditLimitComponent,
    CreditLimitAttributeComponent,
    CreditLimitStatusBarComponent,
    CreditLimitMinimumPaymentComponent,
];
export class CreditLimitModule {
}
CreditLimitModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CreditLimitModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CreditLimitModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CreditLimitModule, declarations: [CreditLimitComponent,
        CreditLimitAttributeComponent,
        CreditLimitStatusBarComponent,
        CreditLimitMinimumPaymentComponent], imports: [CommonModule, AmountModule, BbDatePipeModule, FormatPipeModule, ValidationModule], exports: [CreditLimitComponent,
        CreditLimitAttributeComponent,
        CreditLimitStatusBarComponent,
        CreditLimitMinimumPaymentComponent] });
CreditLimitModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CreditLimitModule, imports: [[CommonModule, ...uiModules]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CreditLimitModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules],
                    declarations: [...components],
                    exports: [...components],
                }]
        }] });
