// @backbase/ui-ang/esm2015/_example/dropdown-panel/showcase/dropdown-panel-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { DropdownPanelShowcaseComponent } from './dropdown-panel-showcase.component';
import { DropdownPanelModule } from '@backbase/ui-ang/dropdown-panel';
import { IconModule } from '@backbase/ui-ang/icon';
import * as i0 from "@angular/core";
export class DropdownPanelShowcaseModule {
}
DropdownPanelShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownPanelShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
DropdownPanelShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownPanelShowcaseModule, declarations: [DropdownPanelShowcaseComponent], imports: [CommonModule, DropdownPanelModule, IconModule] });
DropdownPanelShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownPanelShowcaseModule, imports: [[CommonModule, DropdownPanelModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownPanelShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, DropdownPanelModule, IconModule],
                    declarations: [DropdownPanelShowcaseComponent],
                    entryComponents: [DropdownPanelShowcaseComponent],
                }]
        }] });
