// @backbase/ui-ang/esm2015/logo/logo.component.js
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
/**
 * @name LogoComponent
 *
 * @description
 * Component that displays the company logo.
 */
export class LogoComponent {
    constructor() {
        /**
         * The type of the logo.  Defaults to 'full'.
         */
        this.type = 'full';
        /**
         * Flag to indicate whether to use the inverse colors or not.  Defaults to false.
         */
        this.isInverse = false;
    }
    /**
     * @name LogoComponent#isEmblem
     * @returns Whether this component should show the emblem only or the full logo.
     *
     * @description
     * Returns whether this component should show the emblem only or the full logo.
     */
    get isEmblem() {
        return this.type === 'emblem';
    }
    ngOnInit() { }
}
LogoComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LogoComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
LogoComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: LogoComponent, selector: "bb-logo-ui", inputs: { type: "type", isInverse: "isInverse" }, ngImport: i0, template: "<div\n  class=\"bb-logo\"\n  [ngClass]=\"{\n    'bb-logo--emblem': isEmblem,\n    'bb-logo--inverse': isInverse\n  }\"\n></div>\n", directives: [{ type: i1.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LogoComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-logo-ui',
                    templateUrl: './logo.component.html',
                }]
        }], ctorParameters: function () { return []; }, propDecorators: { type: [{
                type: Input
            }], isInverse: [{
                type: Input
            }] } });
