// @backbase/ui-ang/esm2015/logo/logo.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { LogoComponent } from './logo.component';
import * as i0 from "@angular/core";
export class LogoModule {
}
LogoModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LogoModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LogoModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LogoModule, declarations: [LogoComponent], imports: [CommonModule], exports: [LogoComponent] });
LogoModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LogoModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LogoModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [LogoComponent],
                    exports: [LogoComponent],
                }]
        }] });
