// @backbase/ui-ang/esm2015/account-number-pipe/account-number.pipe.js
import { Pipe } from '@angular/core';
import { segment } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "./accounts-displaying-format.service";
/**
 * Formats account numbers, applies different format configurations per number type
 */
export class AccountNumberPipe {
    /**
     * @name AccountNumberPipe#constructor
     * @param accountsDisplayingFormatService - AccountsDisplayingFormatService
     *
     * @internal
     */
    constructor(accountsDisplayingFormatService) {
        this.accountsDisplayingFormatService = accountsDisplayingFormatService;
    }
    /**
     * @name AccountNumberPipe#transform
     * @param value - the string to transform
     * @param configuration - type of the value (iban, bban or number) or a custom numbers format
     * @param cardProviderName - card provider name
     *
     * @description
     * Formats account numbers
     *
     * to use with a global accountsFormat configuration
     * - '12345678' | bbAccountNumber:'iban'
     * to apply a custom configuration
     * - '12345678' | bbAccountNumber:{ segments: [2, 4, 2] }
     * to apply a 'visa' card number configuration
     * - '12345678' | bbAccountNumber:'cardNumber':'visa'
     */
    transform(value, configuration, cardProviderName) {
        const accountNumber = value.toString();
        const displayingFormat = typeof configuration === 'object'
            ? configuration
            : this.accountsDisplayingFormatService.getDefaultFormat(configuration, cardProviderName);
        return segment(accountNumber, displayingFormat.segments);
    }
}
AccountNumberPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountNumberPipe, deps: [{ token: i1.AccountsDisplayingFormatService }], target: i0.ɵɵFactoryTarget.Pipe });
AccountNumberPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountNumberPipe, name: "bbAccountNumber" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountNumberPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'bbAccountNumber',
                }]
        }], ctorParameters: function () { return [{ type: i1.AccountsDisplayingFormatService }]; } });
