// @backbase/ui-ang/esm2015/account-number-pipe/accounts-displaying-format.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { ACCOUNTS_DISPLAYING_FORMAT, DefaultAccountsDisplayingFormat, } from './constants';
import * as i0 from "@angular/core";
/**
 * A configuration service for account numbers displaying format
 */
export class AccountsDisplayingFormatService {
    /**
     * @name AccountsDisplayingFormatService#constructor
     * @param accountsDisplayingFormat - (Optional) AccountsDisplayingFormatConfiguration
     *
     * @internal
     */
    constructor(accountsDisplayingFormat) {
        this.accountsDisplayingFormat = accountsDisplayingFormat;
    }
    /**
     * @name AccountsDisplayingFormatService#getDefaultFormat
     * @param attributeType - numbers format attributeType
     * @param cardProviderName - card provider name
     *
     * @description
     * Gets default account numbers format per attribute type
     */
    getDefaultFormat(attributeType, cardProviderName) {
        var _a, _b;
        if (attributeType === 'cardNumber') {
            return this.getCardNumberFormat(cardProviderName);
        }
        return (_b = (_a = this.accountsDisplayingFormat) === null || _a === void 0 ? void 0 : _a[attributeType]) !== null && _b !== void 0 ? _b : DefaultAccountsDisplayingFormat;
    }
    getCardNumberFormat(cardProviderName) {
        var _a;
        const cardNumber = (_a = this.accountsDisplayingFormat) === null || _a === void 0 ? void 0 : _a.cardNumber;
        const providersConfiguration = Object.entries((cardNumber === null || cardNumber === void 0 ? void 0 : cardNumber.cardProviders) || {}).reduce((configuration, entry) => (Object.assign(Object.assign({}, configuration), { [entry[0].toLowerCase()]: entry[1] })), {});
        if (cardProviderName && providersConfiguration[cardProviderName.toLowerCase()]) {
            return providersConfiguration[cardProviderName.toLowerCase()];
        }
        return cardNumber !== null && cardNumber !== void 0 ? cardNumber : DefaultAccountsDisplayingFormat;
    }
}
AccountsDisplayingFormatService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountsDisplayingFormatService, deps: [{ token: ACCOUNTS_DISPLAYING_FORMAT, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
AccountsDisplayingFormatService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountsDisplayingFormatService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountsDisplayingFormatService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [ACCOUNTS_DISPLAYING_FORMAT]
                }] }]; } });
