// @backbase/ui-ang/esm2015/_example/value-diff/examples/value-diff-example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/value-diff";
export class ValueDiffExampleComponent {
    constructor() {
        this.oldValue = 'Old Value';
        this.newValue = 'New Value';
    }
}
ValueDiffExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ValueDiffExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ValueDiffExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ValueDiffExampleComponent, selector: "bb-value-diff-ui-example", ngImport: i0, template: "<div [bbValueDiff]=\"newValue\">\n  {{ oldValue }}\n</div>\n", directives: [{ type: i1.ValueDiffDirective, selector: "[bbValueDiff]", inputs: ["bbValueDiff", "position", "sr-only-old", "sr-only-new", "showArrow"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ValueDiffExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-value-diff-ui-example',
                    templateUrl: './value-diff-example.component.html',
                }]
        }] });
