// @backbase/payment-orders-ang/esm2015/lib/ui/pipes/relative-date/relative-date.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RelativeDatePipe } from './relative-date.pipe';
import * as i0 from "@angular/core";
export class RelativeDateModule {
}
/** @nocollapse */ RelativeDateModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RelativeDateModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ RelativeDateModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RelativeDateModule, declarations: [RelativeDatePipe], imports: [CommonModule], exports: [RelativeDatePipe] });
/** @nocollapse */ RelativeDateModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RelativeDateModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RelativeDateModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [RelativeDatePipe],
                    imports: [CommonModule],
                    exports: [RelativeDatePipe],
                }]
        }] });
