// @backbase/identity-auth/esm2015/lib/auth.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ActivityMonitorService } from './activity-monitor';
import { AuthService, WINDOW } from './auth';
import { ɵɵOidcAuthService } from './oidc-auth';
import * as i0 from "@angular/core";
export function _window() {
    return window;
}
export class IdentityAuthModule {
}
/** @nocollapse */ IdentityAuthModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: IdentityAuthModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ IdentityAuthModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: IdentityAuthModule, imports: [CommonModule] });
/** @nocollapse */ IdentityAuthModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: IdentityAuthModule, providers: [
        {
            provide: WINDOW,
            useValue: _window,
        },
        AuthService,
        ɵɵOidcAuthService,
        ActivityMonitorService,
    ], imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: IdentityAuthModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    providers: [
                        {
                            provide: WINDOW,
                            useValue: _window,
                        },
                        AuthService,
                        ɵɵOidcAuthService,
                        ActivityMonitorService,
                    ],
                }]
        }] });
