// @backbase/foundation-ang/esm2015/data-http/custom-encoder.service.js
import { HttpParams } from '@angular/common/http';
import { InjectionToken } from '@angular/core';
export const normalizeHttpParameter = (accum, [key, value]) => {
    if (value === undefined) {
        return accum;
    }
    if (typeof value === 'number') {
        return Object.assign(Object.assign({}, accum), { [key]: String(value) });
    }
    return Object.assign(Object.assign({}, accum), { [key]: value });
};
export const EXTRA_ENCODE_URI_PARAMS = {
    name: 'EXTRA_ENCODE_URI_PARAMS',
    description: 'Correctly encode certain symbols in data module http params (eg encode "+" to "%2B) to fix a known angular bug (#18261)',
};
/**
 * Filter undefined params, and convert numbers to strings.
 */
export const normalizeHttpParameters = (params) => Object.entries(params).reduce(normalizeHttpParameter, {});
export class CustomEncoder {
    encodeKey(key) {
        return encodeURIComponent(key);
    }
    encodeValue(value) {
        return encodeURIComponent(value);
    }
    decodeKey(key) {
        return decodeURIComponent(key);
    }
    decodeValue(value) {
        return decodeURIComponent(value);
    }
}
/**
 * Injection token for the `HttpParamsFactory` utility function.
 *
 * @usageNotes
 *
 * ### Example
 *
 * ```
 * export class MyDataService {
 *   constructor(private readonly http: HttpClient,
 *   @Inject(HTTP_PARAMS_FACTORY)
 *   private readonly toHttpParams: HttpParamsFactory,
 *   ){}
 *
 *   makeRequest(params: HttpParameters): Observable<HttpResponse<any>> {
 *     return this.http.request('get', '/api', {
 *       params: this.toHttpParams(params),
 *     });
 *   }
 * }
 * ```
 */
export const HTTP_PARAMS_FACTORY = new InjectionToken('Http Param Factory');
export const httpParamsFactoryFactory = (featuresService) => {
    // We can't type options correctly, because angular is not exporting HttpParamsOptions
    // https://github.com/angular/angular/issues/20276
    const httpParamsOptions = featuresService.isEnabled(EXTRA_ENCODE_URI_PARAMS) ? { encoder: new CustomEncoder() } : {};
    return (fromObject) => new HttpParams(Object.assign(Object.assign({}, httpParamsOptions), { fromObject: normalizeHttpParameters(fromObject) }));
};
