// @backbase/foundation-ang/esm2015/data-http/data-http.module.js
import { NgModule } from '@angular/core';
import { cacheControlInterceptorProvider } from './cache-control.interceptor';
import { DATA_HTTP_CONFIG, DATA_HTTP_CUSTOM_CONFIG } from './data-http-config';
import { requestCacheInterceptorProvider } from './request-cache.interceptor';
import { PAGE_CONFIG } from '@backbase/foundation-ang/web-sdk';
import { HTTP_PARAMS_FACTORY, httpParamsFactoryFactory, EXTRA_ENCODE_URI_PARAMS } from './custom-encoder.service';
import { FutureFeaturesModule, FeaturesService } from '@backbase/foundation-ang/core';
/**
 * Returns merged data http configuration object
 * @param pageConfigService
 * @param config
 */
export function dataHttpConfigFactory(pageConfigService, config = {}) {
    return Object.assign({ headers: {}, apiRoot: pageConfigService.apiRoot }, config);
}
const ɵ0 = httpParamsFactoryFactory;
export class DataHttpModule {
    /**
     * Creates a module with http configuration for data modules.
     *
     * @usageNotes
     *
     * ### Example
     *
     * ```
     * @NgModule({
     *   imports: [
     *     DataHttpModule.forRoot({
     *       rootApi: '/custom-base-url/',
     *       headers: {
     *         'X-custom': 'foo',
     *       }
     *     })
     *   ]
     * })
     * class MyAppModule {}
     * ```
     */
    static forRoot(config = {}) {
        return {
            ngModule: DataHttpModule,
            providers: [
                requestCacheInterceptorProvider,
                cacheControlInterceptorProvider,
                {
                    provide: DATA_HTTP_CUSTOM_CONFIG,
                    useValue: config,
                },
                {
                    provide: DATA_HTTP_CONFIG,
                    useFactory: dataHttpConfigFactory,
                    deps: [PAGE_CONFIG, DATA_HTTP_CUSTOM_CONFIG],
                },
            ],
        };
    }
}
DataHttpModule.decorators = [
    { type: NgModule, args: [{
                imports: [FutureFeaturesModule.allowOptIn([EXTRA_ENCODE_URI_PARAMS])],
                providers: [
                    {
                        provide: DATA_HTTP_CONFIG,
                        useFactory: dataHttpConfigFactory,
                        deps: [PAGE_CONFIG],
                    },
                    {
                        provide: HTTP_PARAMS_FACTORY,
                        useFactory: ɵ0,
                        deps: [FeaturesService],
                    },
                ],
            },] }
];
export { ɵ0 };
