// @backbase/foundation-ang/esm2015/data-http/data-module-manager.js
import { Inject, Injectable, InjectionToken, Injector, Optional } from '@angular/core';
import { fromExamples, MOCKS_EXAMPLES } from './mocks';
import { joinUrl } from './url';
import * as i0 from "@angular/core";
import * as i1 from "./mocks";
const DEFAULT_CONFIG_TOKEN = new InjectionToken(`@backbase/foundation default config token`);
export class DataModulesManager {
    constructor(injector, mockData) {
        this.injector = injector;
        this._mocks = [];
        this.modulesConfig = new Map();
        this.addMockExamples(mockData || []);
    }
    setModuleConfig(token, config) {
        if (this.modulesConfig.has(token)) {
            const current = this.modulesConfig.get(token);
            this.modulesConfig.set(token, Object.assign(Object.assign({}, current), { config }));
        }
        else {
            this.modulesConfig.set(token, {
                config,
            });
        }
        this.updateMocks();
    }
    getMocks() {
        return this._mocks;
    }
    addMockExamples(mockData) {
        var _a, _b, _c;
        for (const { token, examples } of mockData) {
            const configToken = token || DEFAULT_CONFIG_TOKEN;
            const config = (_a = this.injector.get(configToken, null)) !== null && _a !== void 0 ? _a : undefined;
            const allExamples = configToken === DEFAULT_CONFIG_TOKEN
                ? examples.concat((_c = (_b = this.modulesConfig.get(configToken)) === null || _b === void 0 ? void 0 : _b.examples) !== null && _c !== void 0 ? _c : [])
                : examples;
            this.modulesConfig.set(configToken, {
                examples: allExamples,
                config,
            });
        }
        this.updateMocks();
    }
    updateMocks() {
        const mockExamples = [];
        for (const { config, examples } of this.modulesConfig.values()) {
            mockExamples.push(...(examples || []).map(example => this.createURL(example, config)));
        }
        this._mocks = fromExamples(mockExamples);
    }
    createURL(example, serviceConfig) {
        return serviceConfig
            ? Object.assign(Object.assign({}, example), { urlPattern: joinUrl(serviceConfig.apiRoot, serviceConfig.servicePath, example.urlPattern) }) : example;
    }
}
DataModulesManager.ɵprov = i0.ɵɵdefineInjectable({ factory: function DataModulesManager_Factory() { return new DataModulesManager(i0.ɵɵinject(i0.INJECTOR), i0.ɵɵinject(i1.MOCKS_EXAMPLES, 8)); }, token: DataModulesManager, providedIn: "root" });
DataModulesManager.decorators = [
    { type: Injectable, args: [{ providedIn: 'root' },] }
];
DataModulesManager.ctorParameters = () => [
    { type: Injector },
    { type: Array, decorators: [{ type: Optional }, { type: Inject, args: [MOCKS_EXAMPLES,] }] }
];
