// @backbase/foundation-ang/esm2015/data-http/mocks.js
import { InjectionToken } from '@angular/core';
const DEFAULT_RETURN_STATUS = 200;
export var MocksItem;
(function (MocksItem) {
    const placeholderPattern = /\{[^\}]*\}/g;
    function placeholdersCount(item) {
        return (item.urlPattern.match(placeholderPattern) || []).length;
    }
    function placeholderIndex(item) {
        const regex = new RegExp(placeholderPattern);
        const found = regex.exec(item.urlPattern);
        return found == null ? item.urlPattern.length : found.index;
    }
    function comparePlaceHolderCount(a, b) {
        return placeholdersCount(a) - placeholdersCount(b);
    }
    function comparePlaceholderIndex(a, b) {
        return placeholderIndex(b) - placeholderIndex(a);
    }
    function compareSpecificity(a, b) {
        return comparePlaceHolderCount(a, b) || comparePlaceholderIndex(a, b);
    }
    MocksItem.compareSpecificity = compareSpecificity;
})(MocksItem || (MocksItem = {}));
export function fromExamples(examples) {
    return examples.map(exampleItemToMockItem).sort(MocksItem.compareSpecificity);
}
function exampleItemToMockItem(example) {
    let defaultResponseItem;
    if (example.responses && example.responses.length) {
        defaultResponseItem =
            example.responses.find((response) => isSuccessStatus(response.status)) ||
                example.responses[0];
    }
    return Object.assign({ statusToReturn: defaultResponseItem ? defaultResponseItem.status : DEFAULT_RETURN_STATUS }, example);
}
export function isSuccessStatus(status) {
    return status >= 200 && status <= 299;
}
export const MOCKS_EXAMPLES = new InjectionToken(`@backbase/foundation Mocks Examples`);
