// @backbase/foundation-ang/esm2015/data-http/pagination.service.js
import { HttpParams } from '@angular/common/http';
var PageParams;
(function (PageParams) {
    function equal(a, b) {
        return a.size === b.size && a.from === b.from;
    }
    PageParams.equal = equal;
    function toHttpParams(params) {
        return new HttpParams({
            fromObject: {
                from: Math.floor(params.from).toString(),
                size: Math.floor(params.size).toString(),
            },
        });
    }
    PageParams.toHttpParams = toHttpParams;
})(PageParams || (PageParams = {}));
var CursorParams;
(function (CursorParams) {
    function equal(a, b) {
        return a.size === b.size && a.cursor === b.cursor;
    }
    CursorParams.equal = equal;
    function toHttpParams(params) {
        return new HttpParams({
            fromObject: Object.assign({ from: params.cursor }, (params.size ? { size: Math.floor(params.size).toString() } : {})),
        });
    }
    CursorParams.toHttpParams = toHttpParams;
})(CursorParams || (CursorParams = {}));
export var PaginationParams;
(function (PaginationParams) {
    function equal(a, b) {
        const cursorsEqual = isCursorParams(a) && isCursorParams(b) && CursorParams.equal(a, b);
        const pagesEqual = isPageParams(a) && isPageParams(b) && PageParams.equal(a, b);
        return cursorsEqual || pagesEqual;
    }
    PaginationParams.equal = equal;
    function isPageParams(params) {
        return params.from !== undefined;
    }
    PaginationParams.isPageParams = isPageParams;
    function isCursorParams(params) {
        return params.cursor !== undefined;
    }
    PaginationParams.isCursorParams = isCursorParams;
    function toHttpParams(params) {
        return isPageParams(params) ? PageParams.toHttpParams(params) : CursorParams.toHttpParams(params);
    }
    PaginationParams.toHttpParams = toHttpParams;
})(PaginationParams || (PaginationParams = {}));
