// @backbase/foundation-ang/esm2015/data-http/request-cache.service.js
import { Injectable } from '@angular/core';
const second = 1000;
export class RequestCacheService {
    constructor() {
        this.ttl = 1 * second;
        this.results = new Map();
    }
    get(req) {
        this.garbageCollect();
        const cached = this.results.get(this.key(req));
        if (cached === undefined) {
            return undefined;
        }
        return cached.value;
    }
    set(request, response) {
        this.results.set(this.key(request), this.value(response));
    }
    key(request) {
        return JSON.stringify({
            url: request.urlWithParams,
        });
    }
    value(response) {
        return {
            expireAt: this.expiryTime(),
            value: response,
        };
    }
    expiryTime() {
        return Date.now() + this.ttl;
    }
    isExpired(value) {
        return Date.now() > value.expireAt;
    }
    garbageCollect() {
        this.results.forEach((value, key) => {
            if (this.isExpired(value)) {
                this.results.delete(key);
            }
        });
    }
}
RequestCacheService.decorators = [
    { type: Injectable }
];
