// @backbase/foundation-ang/esm2015/data-http/request-cache.interceptor.js
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { RequestCacheService } from './request-cache.service';
export class RequestCacheInterceptor {
    constructor(cache) {
        this.cache = cache;
    }
    intercept(req, next) {
        if (!this.isCachable(req)) {
            return next.handle(req);
        }
        return this.cachedResponse(req) || this.refreshResponse(req, next);
    }
    isCachable(req) {
        return req.method === 'GET';
    }
    cachedResponse(req) {
        return this.cache.get(req);
    }
    refreshResponse(req, next) {
        const response = next.handle(req);
        this.cache.set(req, response);
        return response;
    }
}
RequestCacheInterceptor.decorators = [
    { type: Injectable }
];
RequestCacheInterceptor.ctorParameters = () => [
    { type: RequestCacheService }
];
export const requestCacheInterceptorProvider = [
    RequestCacheService,
    {
        provide: HTTP_INTERCEPTORS,
        useClass: RequestCacheInterceptor,
        multi: true,
    },
];
