// @backbase/ui-ang/esm2015/_example/action-status/examples/action-status.example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/action-status";
export class ActionStatusExampleComponent {
    constructor() {
        this.actionState = 'success';
    }
}
ActionStatusExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ActionStatusExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ActionStatusExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ActionStatusExampleComponent, selector: "bb-action-status-example-ui", ngImport: i0, template: "<bb-action-status-ui [state]=\"actionState\"></bb-action-status-ui>\n", components: [{ type: i1.ActionStatusComponent, selector: "bb-action-status-ui", inputs: ["state"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ActionStatusExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-action-status-example-ui',
                    templateUrl: './action-status.example.component.html',
                }]
        }] });
