// @backbase/payment-orders-ang/esm2015/lib/ui/components/status-badge/status-badge.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { PayordStatusBadgeAbbreviationCustomizableDirective, PayordStatusBadgeComponent, PayordStatusBadgeCustomizableDirective, PayordStatusBadgeLabelCustomizableDirective, } from './status-badge.component';
import { BadgeModule } from '@backbase/ui-ang/badge';
import * as i0 from "@angular/core";
const customizables = [
    PayordStatusBadgeCustomizableDirective,
    PayordStatusBadgeAbbreviationCustomizableDirective,
    PayordStatusBadgeLabelCustomizableDirective,
];
export class PayordStatusBadgeModule {
}
/** @nocollapse */ PayordStatusBadgeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordStatusBadgeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ PayordStatusBadgeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordStatusBadgeModule, declarations: [PayordStatusBadgeComponent, PayordStatusBadgeCustomizableDirective,
        PayordStatusBadgeAbbreviationCustomizableDirective,
        PayordStatusBadgeLabelCustomizableDirective], imports: [CommonModule, BadgeModule], exports: [BadgeModule, PayordStatusBadgeComponent, PayordStatusBadgeCustomizableDirective,
        PayordStatusBadgeAbbreviationCustomizableDirective,
        PayordStatusBadgeLabelCustomizableDirective] });
/** @nocollapse */ PayordStatusBadgeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordStatusBadgeModule, imports: [[CommonModule, BadgeModule], BadgeModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordStatusBadgeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, BadgeModule],
                    declarations: [PayordStatusBadgeComponent, ...customizables],
                    exports: [BadgeModule, PayordStatusBadgeComponent, ...customizables],
                }]
        }] });
