// @backbase/accessgroup-ui/esm2015/lib/privilege-indicator/privilege-indicator.component.js
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/icon";
import * as i2 from "@angular/common";
import * as i3 from "@backbase/ui-ang/tooltip-directive";
export class PrivilegeIndicatorComponent {
    constructor() {
        /**
         * Flag to indicate if the privilege is applicable. Default is false
         */
        this.applicable = false;
        /**
         * Flag to indicate if the privilege is assigned. Default is false
         */
        this.assigned = false;
        /**
         * Icon class name for changed indicator. Default is empty string
         */
        this.iconClass = '';
        /**
         * Color of privilege indicator, different color when item is removed in approval log. Default is empty string
         */
        this.color = '';
    }
}
/** @nocollapse */ PrivilegeIndicatorComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PrivilegeIndicatorComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ PrivilegeIndicatorComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: PrivilegeIndicatorComponent, selector: "bb-privilege-indicator", inputs: { applicable: "applicable", assigned: "assigned", iconClass: "iconClass", color: "color" }, ngImport: i0, template: "<ng-container *ngIf=\"applicable\">\n  <bb-icon-ui\n    *ngIf=\"assigned\"\n    [color]=\"color || 'success'\"\n    name=\"check\"\n    aria-label=\"Permitted\"\n    i18n-aria-label=\"Permitted aria label@@function-group-list.details.permission.aria.permitted\"\n    bbTooltip=\"Permitted\"\n    i18n-bbTooltip=\"Permitted tooltip@@function-group-list.details.permission.tooltip.permitted\"\n    triggers=\"hover\"\n    [class]=\"iconClass\"></bb-icon-ui>\n  <bb-icon-ui\n    *ngIf=\"!assigned\"\n    [color]=\"color || 'muted'\"\n    name=\"times\"\n    aria-label=\"Not permitted\"\n    i18n-aria-label=\"Not permitted aria label@@function-group-list.details.permission.aria.not-permitted\"\n    bbTooltip=\"Not permitted\"\n    i18n-bbTooltip=\"Not permitted tooltip@@function-group-list.details.permission.tooltip.not-permitted\"\n    triggers=\"hover\"\n    [class]=\"iconClass\"></bb-icon-ui>\n</ng-container>\n<ng-container *ngIf=\"!applicable\">\n  <bb-icon-ui\n    name=\"remove\"\n    [color]=\"color || 'muted'\"\n    aria-label=\"Not applicable for this business function\"\n    i18n-aria-label=\"Not applicable aria label@@function-group-list.details.permission.aria.not-applicable\"\n    bbTooltip=\"Not applicable for this function\"\n    i18n-bbTooltip=\"Not applicable tooltip@@function-group-list.details.permission.tooltip.not-applicable\"\n    triggers=\"hover\"></bb-icon-ui>\n</ng-container>\n", components: [{ type: i1.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i3.TooltipDirective, selector: "[bbTooltip]", inputs: ["bbTooltip", "triggers"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PrivilegeIndicatorComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-privilege-indicator',
                    templateUrl: './privilege-indicator.component.html',
                }]
        }], propDecorators: { applicable: [{
                type: Input
            }], assigned: [{
                type: Input
            }], iconClass: [{
                type: Input
            }], color: [{
                type: Input
            }] } });
