// @backbase/payment-orders-ang/esm2015/lib/ui/components/routable-modal/bb-routable-modal-service.js
import { Injectable } from '@angular/core';
import { Router } from '@angular/router';
import { RoutableModalParamName } from '../../models/types';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
export class RoutableModalService {
    constructor(router) {
        this.router = router;
    }
    openModal(modalName) {
        if (modalName) {
            this.router.navigate([], {
                queryParams: { [RoutableModalParamName]: modalName },
                queryParamsHandling: 'merge',
            });
        }
    }
    closeModal() {
        this.router.navigate([], {
            queryParams: { [RoutableModalParamName]: null },
            queryParamsHandling: 'merge',
        });
    }
}
/** @nocollapse */ RoutableModalService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RoutableModalService, deps: [{ token: i1.Router }], target: i0.ɵɵFactoryTarget.Injectable });
/** @nocollapse */ RoutableModalService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RoutableModalService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RoutableModalService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.Router }]; } });
