// @backbase/ui-ang/esm2015/base-classes/input-base/input-base.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { InputBaseComponent } from './input-base.component';
import * as i0 from "@angular/core";
export class InputBaseModule {
}
InputBaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputBaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputBaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputBaseModule, declarations: [InputBaseComponent], imports: [CommonModule], exports: [InputBaseComponent] });
InputBaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputBaseModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputBaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [InputBaseComponent],
                    exports: [InputBaseComponent],
                }]
        }] });
