// @backbase/foundation-ang/esm2015/core/backbase-core/communication/item-io/output-broker.service.js
import { Injectable } from '@angular/core';
import { take } from 'rxjs/operators';
import { ItemLifecycleService } from '../../rendering/item-lifecycle.service';
import { createComponentPipe } from '../pipes/handlers/component';
import { PipelineRegistry } from '../pipes/pipeline-registry';
import { PipelineService } from '../pipes/pipeline.service';
import { getProcessorNames, parseProperties } from './broker-helpers';
/**
 * The output broker service creates a communication pipeline
 * using the component output as a source. The sink of the pipeline
 * is determined by the item model.
 */
export class OutputBrokerService {
    constructor(itemLifecycle, pipelineRegistry) {
        this.itemLifecycle = itemLifecycle;
        this.pipelineRegistry = pipelineRegistry;
    }
    bootstrap() {
        this.itemLifecycle.whenCreate().subscribe(({ itemModel, component }) => {
            const gc$ = this.itemLifecycle.whenRemove(itemModel.value.name).pipe(take(1));
            this.runOutputPipelines(itemModel, component, gc$);
        });
    }
    runOutputPipelines(itemModel, ref, gc$) {
        const pipelineService = ref.injector.get(PipelineService);
        parseProperties(itemModel.value.ɵoutputProperties).forEach(outputPipesConfig => {
            const sinkCommand = outputPipesConfig.pipeline;
            const sinkFn = pipelineService.createSink(sinkCommand.processorName, sinkCommand.args);
            const pipeline = {
                source: this.outputSource(ref, outputPipesConfig.propertyName),
                sink: sinkFn,
            };
            if (!pipeline.source()) {
                throw new Error(`${ref.componentType.name} does not have an output named '${outputPipesConfig.propertyName}'`);
            }
            pipelineService.runPipeline(pipeline, gc$);
        });
    }
    outputSource(ref, outputName) {
        return createComponentPipe().source(ref, outputName);
    }
    getItemProviders(item) {
        return getProcessorNames(item.ɵoutputProperties).map(processorName => {
            try {
                return this.pipelineRegistry.getProcessorProvider(processorName);
            }
            catch (message) {
                throw new Error(`Output '${processorName}' on ${item.name} is misconfigured. ${message}`);
            }
        });
    }
}
OutputBrokerService.decorators = [
    { type: Injectable }
];
OutputBrokerService.ctorParameters = () => [
    { type: ItemLifecycleService },
    { type: PipelineRegistry }
];
export function createOutputBootstrap(outputBroker) {
    return outputBroker.bootstrap.bind(outputBroker);
}
