// @backbase/foundation-ang/esm2015/core/backbase-core/communication/item-io/parse-model.js
import { serializerDeserializer } from '@backbase/communication-property';
const parsePipe = (pipeString) => {
    const communicationDefinition = serializerDeserializer.fromProperty(pipeString);
    if (communicationDefinition !== undefined) {
        return {
            processorName: communicationDefinition.transport,
            args: [communicationDefinition.param],
        };
    }
    else {
        const parts = pipeString.split(' ');
        const processorName = parts.shift();
        return {
            processorName,
            args: parts,
        };
    }
};
const ɵ0 = parsePipe;
const LEGACY_HANDLERS = ['debug', 'navigation', 'navigation-url', 'route-param', 'model-property'];
const isPipelineConfig = (propertyValue) => {
    const handlerName = propertyValue.split(':')[0];
    return !LEGACY_HANDLERS.includes(handlerName);
};
const ɵ1 = isPipelineConfig;
const upgradeToPipelineConfig = (propertyName, propertyValue) => {
    const colonIndex = propertyValue.indexOf(':');
    const hasArg = colonIndex !== -1;
    const handlerName = hasArg ? propertyValue.substr(0, colonIndex) : propertyValue;
    const handlerArg = hasArg ? propertyValue.substr(colonIndex + 1) : '';
    const handlerToPipeMapping = {
        'navigation-url': () => `url ${handlerArg} ${propertyName}`,
        navigation: () => `navigation ${propertyName} ${handlerArg}`,
        'model-property': () => `model ${handlerArg}`,
    };
    return (handlerToPipeMapping[handlerName] || (() => `${handlerName} ${handlerArg}`))().trim();
};
const ɵ2 = upgradeToPipelineConfig;
const normalizePropertyValue = (propertyName, value) => {
    if (!isPipelineConfig(value)) {
        return upgradeToPipelineConfig(propertyName, value);
    }
    return value;
};
const ɵ3 = normalizePropertyValue;
export const parseInputOutputPropertyValue = (propertyName, propertyValue) => {
    const normalPropertyValue = normalizePropertyValue(propertyName, propertyValue);
    const streamHandlers = normalPropertyValue.trim();
    return parsePipe(streamHandlers);
};
export const inputOutputProperties = (properties) => Object.entries(properties).map(([propertyName, value]) => ({
    propertyName,
    pipeline: parseInputOutputPropertyValue(propertyName, `${value}`),
}));
export { ɵ0, ɵ1, ɵ2, ɵ3 };
