// @backbase/ui-ang/esm2015/keyboard-click-directive/keyboard-click.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { KeyboardClickDirective } from './keyboard-click.directive';
import * as i0 from "@angular/core";
export class KeyboardClickModule {
}
KeyboardClickModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: KeyboardClickModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
KeyboardClickModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: KeyboardClickModule, declarations: [KeyboardClickDirective], imports: [CommonModule], exports: [KeyboardClickDirective] });
KeyboardClickModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: KeyboardClickModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: KeyboardClickModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [KeyboardClickDirective],
                    exports: [KeyboardClickDirective],
                }]
        }] });
