// @backbase/ui-ang/esm2015/_example/load-button/showcase/load-button-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { LoadButtonShowcaseComponent } from './load-button-showcase.component';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { IconModule } from '@backbase/ui-ang/icon';
import * as i0 from "@angular/core";
export class LoadButtonShowcaseModule {
}
LoadButtonShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoadButtonShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LoadButtonShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoadButtonShowcaseModule, declarations: [LoadButtonShowcaseComponent], imports: [CommonModule, LoadButtonModule, IconModule] });
LoadButtonShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoadButtonShowcaseModule, imports: [[CommonModule, LoadButtonModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoadButtonShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, LoadButtonModule, IconModule],
                    declarations: [LoadButtonShowcaseComponent],
                    entryComponents: [LoadButtonShowcaseComponent],
                }]
        }] });
