// @backbase/ui-ang/esm2015/_example/load-button/showcase/load-button-showcase.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/load-button";
import * as i2 from "@backbase/ui-ang/icon";
let LoadButtonShowcaseComponent = class LoadButtonShowcaseComponent {
    constructor() {
        this.showLoadingIndicator = false;
        this.isButtonLoading = false;
        this.isLoadMore = [false, false, false];
        this.loadMoreParams = [{ from: 0 }, { from: 0 }, { from: 0 }];
        this.localDisabled = false;
    }
    onLoadButtonPress() {
        if (this.isButtonLoading) {
            return;
        }
        this.isButtonLoading = true;
        setTimeout(() => {
            this.isButtonLoading = false;
        }, 1500);
    }
    setLoadingParams(params) {
        this.isLoadMore[0] = true;
        this.loadMoreParams[0] = params;
        window.setTimeout(() => {
            this.isLoadMore[0] = false;
        }, 1500);
    }
    setLoadingParamsWithIcon(params) {
        this.isLoadMore[1] = true;
        this.loadMoreParams[1] = params;
        window.setTimeout(() => {
            this.isLoadMore[1] = false;
        }, 1500);
    }
    setLoadingParamsOnlyIcon(params) {
        this.isLoadMore[2] = true;
        this.loadMoreParams[2] = params;
        window.setTimeout(() => {
            this.isLoadMore[2] = false;
        }, 1500);
    }
};
LoadButtonShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoadButtonShowcaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
LoadButtonShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: LoadButtonShowcaseComponent, selector: "bb-load-button-ui-showcase", ngImport: i0, template: "<div class=\"bb-toolbar\">\n  <div class=\"text-center\">\n    <bb-load-button-ui\n      color=\"primary\"\n      [isLoading]=\"isLoadMore[0]\"\n      [size]=\"8\"\n      [from]=\"loadMoreParams[0]?.from\"\n      [totalCount]=\"48\"\n      (paramChange)=\"setLoadingParams($event)\"\n      class=\"m-3\"\n      [disabled]=\"localDisabled\"\n    >\n      Load more\n    </bb-load-button-ui>\n    <p class=\"mt-3\">Counter (max 5): {{ loadMoreParams[0]?.from }}</p>\n    <button (click)=\"localDisabled = !localDisabled\">disable load button</button>\n  </div>\n  <div class=\"text-center\">\n    <bb-load-button-ui\n      class=\"m-3\"\n      color=\"primary\"\n      [isLoading]=\"isLoadMore[1]\"\n      [size]=\"8\"\n      [from]=\"loadMoreParams[1]?.from\"\n      [totalCount]=\"48\"\n      (paramChange)=\"setLoadingParamsWithIcon($event)\"\n    >\n      <span>Download file</span>\n      <bb-icon-ui name=\"download\" size=\"lg\" cropped></bb-icon-ui>\n    </bb-load-button-ui>\n    <p class=\"mt-3\">Counter (max 5): {{ loadMoreParams[1]?.from }}</p>\n  </div>\n  <div class=\"text-center\">\n    <bb-load-button-ui\n      class=\"m-3\"\n      color=\"primary\"\n      [isLoading]=\"isLoadMore[2]\"\n      [size]=\"8\"\n      [from]=\"loadMoreParams[2]?.from\"\n      [totalCount]=\"48\"\n      (paramChange)=\"setLoadingParamsOnlyIcon($event)\"\n      [circle]=\"true\"\n    >\n      <bb-icon-ui name=\"search\" size=\"lg\"></bb-icon-ui>\n    </bb-load-button-ui>\n    <p class=\"mt-3\">Counter (max 5): {{ loadMoreParams[2]?.from }}</p>\n  </div>\n</div>\n", components: [{ type: i1.LoadButtonComponent, selector: "bb-load-button-ui", inputs: ["isLoading", "from", "size", "circle", "totalCount"], outputs: ["paramChange"] }, { type: i2.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }] });
LoadButtonShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
], LoadButtonShowcaseComponent);
export { LoadButtonShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoadButtonShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-load-button-ui-showcase',
                    templateUrl: './load-button-showcase.component.html',
                }]
        }] });
