// @backbase/transactions-list-widget-ang/esm2015/lib/transactions-list-widget.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BackbaseCoreModule, FutureFeaturesModule } from '@backbase/foundation-ang/core';
import { ReactiveFormsModule } from '@angular/forms';
import { TransactionsListComponent, TransactionsListEmptyStateCustomizableDirective, TransactionsListListCustomizableDirective, TransactionsListPaginationCustomizableDirective, } from './components/transactions-list/transactions-list.component';
import { TransactionsListItemGroupComponent, TransactionsListItemGroupCustomizableDirective, } from './components/transactions-list/transactions-list-item-group.component';
import { TransactionsListItemComponent, TransactionsListItemCustomizableDirective, TransactionsListItemHeaderCustomizableDirective, } from './components/transactions-list/transactions-list-item.component';
import { TransactionHeaderComponent } from './components/transaction/transaction-header.component';
import { TransactionDetailsComponent, TransactionsListItemDetailsCustomizableDirective, TransactionsListConfirmationModalCustomizableDirective, } from './components/transaction/transaction-details.component';
import { TransactionControlsComponent } from './components/transaction/transaction-controls.component';
import { TransactionsControlsComponent, TransactionsListControlsButtonsCustomizableDirective, TransactionsListControlsCustomizableDirective, } from './components/transactions-controls/transactions-controls.component';
import { TransactionsFilterComponent, TransactionsListFilterCustomizableDirective, } from './components/transactions-controls/transactions-filter.component';
import { TransactionsContainerComponent } from './components/transactions-container.component';
import { TransactionsListWidgetComponent } from './components/transactions-list-widget.component';
import { TransactionNotesComponent } from './components/transaction/transaction-notes.component';
import { showDetailsConfigToken, TransactionsCommonModule, } from '@backbase/transactions-common-ang';
import { TransactionDetailCategoryComponent, TransactionDetailCategoryCustomizableDirective, } from './components/transaction/transaction-detail-category.component';
import { TransactionsCategoriesComponent, TransactionsCategoriesCustomizableDirective, } from './components/transactions-categories/transactions-categories.component';
import { TransactionsListPendingTransactionsComponent, PendingTransactionsListCustomizableDirective, } from './components/transactions-list/transactions-list-pending-transactions.component';
import { TransactionDetailsModalComponent } from './components/transaction/transaction-details-modal.component';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { RouterModule } from '@angular/router';
import { AmountModule } from '@backbase/ui-ang/amount';
import { ButtonModule } from '@backbase/ui-ang/button';
import { DropdownSingleSelectModule } from '@backbase/ui-ang/dropdown-single-select';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { IconModule } from '@backbase/ui-ang/icon';
import { InputDatepickerModule } from '@backbase/ui-ang/input-datepicker';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { ModalModule } from '@backbase/ui-ang/modal';
import { PaginatorModule } from '@backbase/ui-ang/pagination';
import { SearchBoxModule } from '@backbase/ui-ang/search-box';
import { FocusModule } from '@backbase/ui-ang/focus';
import { NotificationModule } from '@backbase/ui-ang/notification';
import { DropdownMenuModule } from '@backbase/ui-ang/dropdown-menu';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { AlertModule } from '@backbase/ui-ang/alert';
import { HeaderModule } from '@backbase/ui-ang/header';
import { BadgeModule } from '@backbase/ui-ang/badge';
import { InputNumberModule } from '@backbase/ui-ang/input-number';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { TextareaModule } from '@backbase/ui-ang/textarea';
import { TRANSACTIONS_DETAILS_SHARE_MODAL } from './constants';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const modules = [
    TransactionsCommonModule,
    AmountModule,
    ButtonModule,
    DropdownSingleSelectModule,
    EmptyStateModule,
    IconModule,
    InputDatepickerModule,
    InputNumberModule,
    InputValidationMessageModule,
    LoadButtonModule,
    LoadingIndicatorModule,
    ModalModule,
    PaginatorModule,
    SearchBoxModule,
    FocusModule,
    NotificationModule,
    DropdownMenuModule,
    TooltipModule,
    ButtonModule,
    AlertModule,
    HeaderModule,
    BadgeModule,
    TextareaModule,
    NgbModule,
];
const components = [
    TransactionsListWidgetComponent,
    TransactionsContainerComponent,
    TransactionsControlsComponent,
    TransactionsFilterComponent,
    TransactionsListComponent,
    TransactionsListItemComponent,
    TransactionsListItemGroupComponent,
    TransactionHeaderComponent,
    TransactionDetailsComponent,
    TransactionControlsComponent,
    TransactionNotesComponent,
    TransactionDetailCategoryComponent,
    TransactionsCategoriesComponent,
    TransactionsListPendingTransactionsComponent,
    TransactionDetailsModalComponent,
];
const customizables = [
    TransactionsListFilterCustomizableDirective,
    TransactionsListControlsCustomizableDirective,
    TransactionsListControlsButtonsCustomizableDirective,
    TransactionsListListCustomizableDirective,
    TransactionsListPaginationCustomizableDirective,
    TransactionsListItemCustomizableDirective,
    TransactionsListItemGroupCustomizableDirective,
    TransactionsListItemHeaderCustomizableDirective,
    TransactionsListItemDetailsCustomizableDirective,
    TransactionsListConfirmationModalCustomizableDirective,
    TransactionsListEmptyStateCustomizableDirective,
    TransactionDetailCategoryCustomizableDirective,
    TransactionsCategoriesCustomizableDirective,
    PendingTransactionsListCustomizableDirective,
];
export class TransactionsListWidgetModule {
    static forRoot(config) {
        return {
            ngModule: TransactionsListWidgetModule,
            providers: [
                {
                    provide: showDetailsConfigToken,
                    useValue: config.showDetails,
                },
            ],
        };
    }
}
TransactionsListWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TransactionsListWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
TransactionsListWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TransactionsListWidgetModule, declarations: [TransactionsListWidgetComponent,
        TransactionsContainerComponent,
        TransactionsControlsComponent,
        TransactionsFilterComponent,
        TransactionsListComponent,
        TransactionsListItemComponent,
        TransactionsListItemGroupComponent,
        TransactionHeaderComponent,
        TransactionDetailsComponent,
        TransactionControlsComponent,
        TransactionNotesComponent,
        TransactionDetailCategoryComponent,
        TransactionsCategoriesComponent,
        TransactionsListPendingTransactionsComponent,
        TransactionDetailsModalComponent, TransactionsListFilterCustomizableDirective,
        TransactionsListControlsCustomizableDirective,
        TransactionsListControlsButtonsCustomizableDirective,
        TransactionsListListCustomizableDirective,
        TransactionsListPaginationCustomizableDirective,
        TransactionsListItemCustomizableDirective,
        TransactionsListItemGroupCustomizableDirective,
        TransactionsListItemHeaderCustomizableDirective,
        TransactionsListItemDetailsCustomizableDirective,
        TransactionsListConfirmationModalCustomizableDirective,
        TransactionsListEmptyStateCustomizableDirective,
        TransactionDetailCategoryCustomizableDirective,
        TransactionsCategoriesCustomizableDirective,
        PendingTransactionsListCustomizableDirective], imports: [RouterModule,
        CommonModule,
        ReactiveFormsModule, i1.BackbaseCoreModule, i1.FutureFeaturesModule, TransactionsCommonModule,
        AmountModule,
        ButtonModule,
        DropdownSingleSelectModule,
        EmptyStateModule,
        IconModule,
        InputDatepickerModule,
        InputNumberModule,
        InputValidationMessageModule,
        LoadButtonModule,
        LoadingIndicatorModule,
        ModalModule,
        PaginatorModule,
        SearchBoxModule,
        FocusModule,
        NotificationModule,
        DropdownMenuModule,
        TooltipModule,
        ButtonModule,
        AlertModule,
        HeaderModule,
        BadgeModule,
        TextareaModule,
        NgbModule], exports: [TransactionsCommonModule,
        AmountModule,
        ButtonModule,
        DropdownSingleSelectModule,
        EmptyStateModule,
        IconModule,
        InputDatepickerModule,
        InputNumberModule,
        InputValidationMessageModule,
        LoadButtonModule,
        LoadingIndicatorModule,
        ModalModule,
        PaginatorModule,
        SearchBoxModule,
        FocusModule,
        NotificationModule,
        DropdownMenuModule,
        TooltipModule,
        ButtonModule,
        AlertModule,
        HeaderModule,
        BadgeModule,
        TextareaModule,
        NgbModule, TransactionsListWidgetComponent,
        TransactionsContainerComponent,
        TransactionsControlsComponent,
        TransactionsFilterComponent,
        TransactionsListComponent,
        TransactionsListItemComponent,
        TransactionsListItemGroupComponent,
        TransactionHeaderComponent,
        TransactionDetailsComponent,
        TransactionControlsComponent,
        TransactionNotesComponent,
        TransactionDetailCategoryComponent,
        TransactionsCategoriesComponent,
        TransactionsListPendingTransactionsComponent,
        TransactionDetailsModalComponent, TransactionsListFilterCustomizableDirective,
        TransactionsListControlsCustomizableDirective,
        TransactionsListControlsButtonsCustomizableDirective,
        TransactionsListListCustomizableDirective,
        TransactionsListPaginationCustomizableDirective,
        TransactionsListItemCustomizableDirective,
        TransactionsListItemGroupCustomizableDirective,
        TransactionsListItemHeaderCustomizableDirective,
        TransactionsListItemDetailsCustomizableDirective,
        TransactionsListConfirmationModalCustomizableDirective,
        TransactionsListEmptyStateCustomizableDirective,
        TransactionDetailCategoryCustomizableDirective,
        TransactionsCategoriesCustomizableDirective,
        PendingTransactionsListCustomizableDirective] });
TransactionsListWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TransactionsListWidgetModule, imports: [[
            RouterModule,
            CommonModule,
            ReactiveFormsModule,
            BackbaseCoreModule.withConfig({
                classMap: { TransactionsListWidgetComponent },
            }),
            FutureFeaturesModule.allowOptIn([TRANSACTIONS_DETAILS_SHARE_MODAL]),
            ...modules,
        ], TransactionsCommonModule,
        AmountModule,
        ButtonModule,
        DropdownSingleSelectModule,
        EmptyStateModule,
        IconModule,
        InputDatepickerModule,
        InputNumberModule,
        InputValidationMessageModule,
        LoadButtonModule,
        LoadingIndicatorModule,
        ModalModule,
        PaginatorModule,
        SearchBoxModule,
        FocusModule,
        NotificationModule,
        DropdownMenuModule,
        TooltipModule,
        ButtonModule,
        AlertModule,
        HeaderModule,
        BadgeModule,
        TextareaModule,
        NgbModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TransactionsListWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        RouterModule,
                        CommonModule,
                        ReactiveFormsModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { TransactionsListWidgetComponent },
                        }),
                        FutureFeaturesModule.allowOptIn([TRANSACTIONS_DETAILS_SHARE_MODAL]),
                        ...modules,
                    ],
                    declarations: [...components, ...customizables],
                    exports: [...modules, ...components, ...customizables],
                    entryComponents: [TransactionsContainerComponent],
                }]
        }] });
