// @backbase/payment-orders-ang/esm2015/lib/ui/components/sr-search-result-info/sr-search-result-info.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PaymentsSrSearchResultInfoComponent } from './sr-search-result-info.component';
import * as i0 from "@angular/core";
/**
 * Exports UI components and modules for payments common
 */
export class PaymentsSrSearchResultInfoModule {
}
/** @nocollapse */ PaymentsSrSearchResultInfoModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentsSrSearchResultInfoModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ PaymentsSrSearchResultInfoModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentsSrSearchResultInfoModule, declarations: [PaymentsSrSearchResultInfoComponent], imports: [CommonModule], exports: [PaymentsSrSearchResultInfoComponent] });
/** @nocollapse */ PaymentsSrSearchResultInfoModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentsSrSearchResultInfoModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentsSrSearchResultInfoModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [PaymentsSrSearchResultInfoComponent],
                    exports: [PaymentsSrSearchResultInfoComponent],
                }]
        }] });
