// @backbase/payord-form-builder-ang/esm2015/lib/components/alert/payord-alert.component.js
import { Component } from '@angular/core';
import { PaymentFormFieldHooks, } from '@backbase/payment-orders-ang';
import { triggerHook } from '../../helpers/hooks-helpers';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/alert";
import * as i2 from "@angular/common";
export class PayordAlertComponent {
    /**
     * Initializes the component.
     *
     * @returns
     */
    ngOnInit() {
        triggerHook(PaymentFormFieldHooks.onInit, this);
    }
    /**
     * Cleanups just before Angular destroys the component.
     *
     * @returns
     */
    ngOnDestroy() {
        triggerHook(PaymentFormFieldHooks.onDestroy, this);
    }
}
/** @nocollapse */ PayordAlertComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordAlertComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ PayordAlertComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: PayordAlertComponent, selector: "bb-payord-alert", ngImport: i0, template: "<ng-container *ngIf=\"!config?.hidden\">\n  <div [ngClass]=\"options?.cssClasses\" role=\"alert\">\n    <bb-alert-ui\n      *ngIf=\"options?.title && options?.modifier\"\n      [title]=\"options?.title\"\n      [modifier]=\"options?.modifier\"\n      [dismissible]=\"options?.dismissible\"\n    >\n      <p class=\"bb-block bb-block--no-margin\">{{ options?.description }}</p>\n    </bb-alert-ui>\n  </div>\n</ng-container>\n", components: [{ type: i1.AlertComponent, selector: "bb-alert-ui", inputs: ["modifier", "dismissible", "title", "message"], outputs: ["close"] }], directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i2.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordAlertComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-payord-alert',
                    templateUrl: './payord-alert.component.html',
                }]
        }] });
