// @backbase/ui-ang/esm2015/input-text/input-text.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { IconModule } from '@backbase/ui-ang/icon';
import { FormsModule } from '@angular/forms';
import { BaseClassesModule } from '@backbase/ui-ang/base-classes';
import { CharCounterModule } from '@backbase/ui-ang/char-counter';
import { InputTextComponent } from './input-text.component';
import { FocusModule } from '@backbase/ui-ang/focus';
import { NgxMaskModule } from 'ngx-mask';
import * as i0 from "@angular/core";
import * as i1 from "ngx-mask";
export class InputTextModule {
}
InputTextModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputTextModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputTextModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputTextModule, declarations: [InputTextComponent], imports: [CommonModule,
        IconModule,
        FormsModule,
        BaseClassesModule,
        FocusModule,
        CharCounterModule, i1.NgxMaskModule], exports: [InputTextComponent] });
InputTextModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputTextModule, imports: [[
            CommonModule,
            IconModule,
            FormsModule,
            BaseClassesModule,
            FocusModule,
            CharCounterModule,
            NgxMaskModule.forRoot(),
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputTextModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        IconModule,
                        FormsModule,
                        BaseClassesModule,
                        FocusModule,
                        CharCounterModule,
                        NgxMaskModule.forRoot(),
                    ],
                    declarations: [InputTextComponent],
                    exports: [InputTextComponent],
                }]
        }] });
