// @backbase/account-statement-common-ang/esm2015/mocks/account-statements-response.mock.js
export const accountStatementsResponseMock = [
    {
        accountId: '1cdb2224-8926-4b4d-a99f-1c9dfbbb4699',
        accountName: 'The Walt Disney Company',
        accountNumber: 'UA85 3996 2200 0000 0260 0123 3566 1',
        date: '2016-02-15',
        category: 'Credit Card',
        description: 'Credit Card statement - February 2016',
        documents: [
            {
                url: 'http://server.com/documentsdepo/uid_1',
                contentType: 'application/pdf',
            },
        ],
    },
    {
        accountId: '1cdb2224-8926-4b4d-a99f-1c9dfbbb4699',
        accountName: 'The Walt Disney Company',
        accountNumber: 'UA85 3996 2200 0000 0260 0123 3566 1',
        date: '2016-03-15',
        category: 'Credit Card',
        description: 'Credit Card statement - March 2016',
        documents: [
            {
                uid: 'uid_2',
                contentType: 'application/pdf',
            },
        ],
    },
    {
        accountId: '1cdb2224-8926-4b4d-a99f-1c9dfbbb4699',
        accountName: 'The Walt Disney Company',
        accountNumber: 'UA85 3996 2200 0000 0260 0123 3566 1',
        date: '2016-04-15',
        category: 'Credit Card',
        description: 'Credit Card statement - April 2016',
        documents: [
            {
                uid: 'uid_3',
                contentType: 'application/pdf',
            },
        ],
    },
    {
        accountId: '1cdb2224-8926-4b4d-a99f-1c9dfbbb4699',
        accountName: 'The Walt Disney Company',
        accountNumber: 'UA85 3996 2200 0000 0260 0123 3566 1',
        date: '2016-05-15',
        category: 'Credit Card',
        description: 'Credit Card statement - May 2016',
        documents: [
            {
                url: 'http://server.com/documentsdepo/uid_4',
                contentType: 'application/pdf',
            },
        ],
    },
    {
        accountId: '1cdb2224-8926-4b4d-a99f-1c9dfbbb4699',
        accountName: 'The Walt Disney Company',
        accountNumber: 'UA85 3996 2200 0000 0260 0123 3566 1',
        date: '2016-06-15',
        category: 'Credit Card',
        description: 'Credit Card statement - June 2016',
        documents: [
            {
                uid: 'uid_5',
                contentType: 'application/pdf',
            },
        ],
    },
];
