// @backbase/account-statement-common-ang/esm2015/mocks/accounts-response.mock.js
export const accountsResponseMock = [
    {
        id: '1cdb2224-8926-4b4d-a99f-1c9dfbbb4691',
        externalArrangementId: 'bcf10f4d-4b2f-4413-9bab-31ff693608b5',
        externalLegalEntityId: 'c7a382786d514262b75ab9531b749a2b',
        externalProductId: 'fade7867-533e-465e-90cb-e41675c54400',
        name: 'Mr and Mrs J. Smith',
        bankAlias: 'Our joined account',
        sourceId: 'LJBASI2XXXX',
        bookedBalance: 1000,
        availableBalance: 1500,
        creditLimit: 442.12,
        // eslint-disable-next-line @typescript-eslint/naming-convention
        IBAN: 'CY3887370130MJFTJ3B8Y9W7IGRO',
        // eslint-disable-next-line @typescript-eslint/naming-convention
        BBAN: '8873701303897',
        currency: 'AED',
        externalTransferAllowed: true,
        urgentTransferAllowed: true,
        accruedInterest: 0.54,
        // eslint-disable-next-line id-blacklist
        number: 'PANS',
        principalAmount: 620.54,
        currentInvestmentValue: 0.16,
        legalEntityIds: ['257da57a-11e4-4553-9175-54baf755069b', 'cd83683b-13f2-43d8-882b-39c9ab27d499'],
        productId: '36c8fc42-ec97-4f83-8a7c-d622625007f3',
        productNumber: 'ffdd939c-ac4a-4441-ae47-70a7259899e7',
        productKindName: 'Current Account',
        productTypeName: 'Current Account',
        // eslint-disable-next-line @typescript-eslint/naming-convention
        BIC: 'AABAFI22',
        bankBranchCode: 'bankBranchCode',
        visible: false,
        accountOpeningDate: '2016-01-28T16:41:41.090Z',
        accountInterestRate: 100.2,
        valueDateBalance: 100.1,
        creditLimitUsage: 100.3,
        creditLimitInterestRate: 100.4,
        creditLimitExpiryDate: '2019-09-28T16:41:41.090Z',
        startDate: '2016-02-28T16:41:41.090Z',
        termUnit: 'Y',
        termNumber: 50,
        interestPaymentFrequencyUnit: 'M',
        interestPaymentFrequencyNumber: 15,
        maturityDate: '2017-02-28T16:41:41.090Z',
        maturityAmount: 99.5,
        autoRenewalIndicator: true,
        interestSettlementAccount: 'interestSettlementAccount1',
        outstandingPrincipalAmount: 100.2,
        monthlyInstalmentAmount: 100.1,
        amountInArrear: 100.3,
        minimumRequiredBalance: 80.4,
        creditCardAccountNumber: '123456',
        validThru: '2019-02-28T16:41:41.090Z',
        applicableInterestRate: 101.2,
        remainingCredit: 50,
        outstandingPayment: 105.5,
        minimumPayment: 51.1,
        minimumPaymentDueDate: '2018-02-28T16:41:41.090Z',
        totalInvestmentValue: 110.2,
        debitCards: [
            {
                // eslint-disable-next-line id-blacklist
                number: '4578',
                expiryDate: '2020-08-22',
                cardId: 'id1',
                cardholderName: 'John Doe',
                cardType: 'Visa Electron',
                cardStatus: 'Active',
            },
            {
                // eslint-disable-next-line id-blacklist
                number: '3241',
                expiryDate: '2019-01-14',
                cardId: 'id2',
                cardholderName: 'John Doe',
                cardType: 'Visa',
                cardStatus: 'Active',
            },
        ],
        accountHolderAddressLine1: 'accountHolderAddressLine11',
        accountHolderAddressLine2: 'accountHolderAddressLine12',
        accountHolderStreetName: 'accountHolderStreetName1',
        town: 'Paris',
        postCode: '2000',
        countrySubDivision: 'countrySubDivision1',
        accountHolderNames: 'Danthe Mohr',
        accountHolderCountry: 'FR',
        creditAccount: true,
        debitAccount: true,
        lastUpdateDate: '2016-01-28T16:41:41.090Z',
        favorite: true,
        userPreferences: {
            alias: 'Our joined account',
            visible: true,
            favorite: false,
        },
        product: {
            externalId: 'externalProductidId',
            externalTypeId: 'externalProductTypeId',
            typeName: 'Current Account',
            productKind: {
                id: 1,
                externalKindId: 'kind1',
                kindName: 'Current Account',
                kindUri: 'current-account',
            },
        },
        state: {
            externalStateId: 'externalStateId1',
            state: 'Active',
        },
    },
    {
        id: '1cdb2224-8926-4b4d-a99f-1c9dfbbb4692',
        externalArrangementId: 'bcf10f4d-4b2f-4413-9bab-31ff693608b5',
        externalLegalEntityId: 'c7a382786d514262b75ab9531b749a2b',
        externalProductId: 'fade7867-533e-465e-90cb-e41675c54400',
        name: 'Howlin Wolf',
        bankAlias: 'Royalties account',
        sourceId: 'LJBASI2XXXX',
        bookedBalance: 15000,
        availableBalance: 21500,
        creditLimit: 3442.12,
        // eslint-disable-next-line @typescript-eslint/naming-convention
        IBAN: 'GB60OHSF29521903589133',
        // eslint-disable-next-line @typescript-eslint/naming-convention
        BBAN: '029521903589133',
        currency: 'AED',
        externalTransferAllowed: true,
        urgentTransferAllowed: true,
        accruedInterest: 0.54,
        // eslint-disable-next-line id-blacklist
        number: 'PANS',
        principalAmount: 620.54,
        currentInvestmentValue: 0.16,
        legalEntityIds: ['257da57a-11e4-4553-9175-54baf755069b', 'cd83683b-13f2-43d8-882b-39c9ab27d499'],
        productId: '36c8fc42-ec97-4f83-8a7c-d622625007f3',
        productNumber: 'ffdd939c-ac4a-4441-ae47-70a7259899e7',
        productKindName: 'Current Account',
        productTypeName: 'Current Account',
        // eslint-disable-next-line @typescript-eslint/naming-convention
        BIC: 'AABAFI22',
        bankBranchCode: 'bankBranchCode',
        visible: false,
        accountOpeningDate: '2016-01-28T16:41:41.090Z',
        accountInterestRate: 100.2,
        valueDateBalance: 100.1,
        creditLimitUsage: 100.3,
        creditLimitInterestRate: 100.4,
        creditLimitExpiryDate: '2019-09-28T16:41:41.090Z',
        startDate: '2016-02-28T16:41:41.090Z',
        termUnit: 'Y',
        termNumber: 50,
        interestPaymentFrequencyUnit: 'M',
        interestPaymentFrequencyNumber: 15,
        maturityDate: '2017-02-28T16:41:41.090Z',
        maturityAmount: 99.5,
        autoRenewalIndicator: true,
        interestSettlementAccount: 'interestSettlementAccount1',
        outstandingPrincipalAmount: 100.2,
        monthlyInstalmentAmount: 100.1,
        amountInArrear: 100.3,
        minimumRequiredBalance: 80.4,
        creditCardAccountNumber: '123456',
        validThru: '2019-02-28T16:41:41.090Z',
        applicableInterestRate: 101.2,
        remainingCredit: 50,
        outstandingPayment: 105.5,
        minimumPayment: 51.1,
        minimumPaymentDueDate: '2018-02-28T16:41:41.090Z',
        totalInvestmentValue: 110.2,
        debitCards: [
            {
                // eslint-disable-next-line id-blacklist
                number: '4578',
                expiryDate: '2020-08-22',
                cardId: 'id1',
                cardholderName: 'John Doe',
                cardType: 'Visa Electron',
                cardStatus: 'Active',
            },
            {
                // eslint-disable-next-line id-blacklist
                number: '3241',
                expiryDate: '2019-01-14',
                cardId: 'id2',
                cardholderName: 'John Doe',
                cardType: 'Visa',
                cardStatus: 'Active',
            },
        ],
        accountHolderAddressLine1: 'accountHolderAddressLine11',
        accountHolderAddressLine2: 'accountHolderAddressLine12',
        accountHolderStreetName: 'accountHolderStreetName1',
        town: 'Skopje',
        postCode: '2000',
        countrySubDivision: 'countrySubDivision1',
        accountHolderNames: 'Toso Malerot',
        accountHolderCountry: 'MK',
        creditAccount: true,
        debitAccount: true,
        lastUpdateDate: '2016-01-28T16:41:41.090Z',
        favorite: true,
        userPreferences: {
            alias: 'Royalties account',
            visible: true,
            favorite: false,
        },
        product: {
            externalId: 'externalProductidId',
            externalTypeId: 'externalProductTypeId',
            typeName: 'Current Account',
            productKind: {
                id: 1,
                externalKindId: 'kind1',
                kindName: 'Current Account',
                kindUri: 'current-account',
            },
        },
        state: {
            externalStateId: 'externalStateId2',
            state: 'InActive',
        },
    },
    {
        id: '3cdb2224-8926-4b4d-a99f-1c9dfbbb4600',
        externalArrangementId: 'bcf10f4d-4b2f-4413-9bab-31ff693608b5',
        externalLegalEntityId: 'c7a382786d514262b75ab9531b749a2b',
        externalProductId: 'fade7867-533e-465e-90cb-e41675c54400',
        name: 'Company saving account',
        bankAlias: 'Langosh, Okuneva and Spencer',
        sourceId: 'LJBASI2XXXX',
        bookedBalance: 10550000,
        availableBalance: 1500,
        // eslint-disable-next-line @typescript-eslint/naming-convention
        IBAN: 'IT09R0442267398V9QB48EV3THQ',
        // eslint-disable-next-line @typescript-eslint/naming-convention
        BBAN: '904422673989483',
        currency: 'EUR',
        externalTransferAllowed: true,
        urgentTransferAllowed: true,
        accruedInterest: 36.16,
        // eslint-disable-next-line id-blacklist
        number: 'PANS',
        principalAmount: 620.54,
        currentInvestmentValue: 0.16,
        legalEntityIds: ['257da57a-11e4-4553-9175-54baf755069b', 'cd83683b-13f2-43d8-882b-39c9ab27d499'],
        productId: '36c8fc42-ec97-4f83-8a7c-d622625007f3',
        productNumber: 'ffdd939c-ac4a-4441-ae47-70a7259899e7',
        productKindName: 'Savings Account',
        productTypeName: 'productTypeName',
        // eslint-disable-next-line @typescript-eslint/naming-convention
        BIC: 'AABAFI22',
        bankBranchCode: 'bankBranchCode',
        visible: false,
        accountOpeningDate: '2016-01-28T16:41:41.090Z',
        accountInterestRate: 100.2,
        valueDateBalance: 100.1,
        creditLimitUsage: 100.3,
        creditLimitInterestRate: 100.4,
        creditLimitExpiryDate: '2019-09-28T16:41:41.090Z',
        startDate: '2016-02-28T16:41:41.090Z',
        termUnit: 'Y',
        termNumber: 50,
        interestPaymentFrequencyUnit: 'M',
        interestPaymentFrequencyNumber: 15,
        maturityDate: '2017-02-28T16:41:41.090Z',
        maturityAmount: 99.5,
        autoRenewalIndicator: true,
        interestSettlementAccount: 'interestSettlementAccount1',
        outstandingPrincipalAmount: 100.2,
        monthlyInstalmentAmount: 100.1,
        amountInArrear: 100.3,
        minimumRequiredBalance: 80.4,
        creditCardAccountNumber: '123456',
        validThru: '2019-02-28T16:41:41.090Z',
        applicableInterestRate: 101.2,
        remainingCredit: 50,
        outstandingPayment: 105.5,
        minimumPayment: 51.1,
        minimumPaymentDueDate: '2018-02-28T16:41:41.090Z',
        totalInvestmentValue: 110.2,
        debitCards: [
            {
                // eslint-disable-next-line id-blacklist
                number: '4578',
                expiryDate: '2020-08-22',
                cardId: 'id1',
                cardholderName: 'John Doe',
                cardType: 'Visa Electron',
                cardStatus: 'Active',
            },
            {
                // eslint-disable-next-line id-blacklist
                number: '3241',
                expiryDate: '2019-01-14',
                cardId: 'id2',
                cardholderName: 'John Doe',
                cardType: 'Visa',
                cardStatus: 'Active',
            },
        ],
        accountHolderAddressLine1: 'accountHolderAddressLine11',
        accountHolderAddressLine2: 'accountHolderAddressLine12',
        accountHolderStreetName: 'accountHolderStreetName1',
        town: 'Skopje',
        postCode: '2000',
        countrySubDivision: 'countrySubDivision1',
        accountHolderNames: 'Mile Panika',
        accountHolderCountry: 'MK',
        creditAccount: true,
        debitAccount: true,
        lastUpdateDate: '2016-01-28T16:41:41.090Z',
        favorite: true,
        userPreferences: {
            alias: 'Langosh, Okuneva and Spencer',
            visible: true,
            favorite: false,
        },
        product: {
            externalId: 'externalProductidId',
            externalTypeId: 'externalProductTypeId',
            typeName: 'Saving Account',
            productKind: {
                id: 1,
                externalKindId: 'kind1',
                kindName: 'Saving Account',
                kindUri: 'Saving-account',
            },
        },
        state: {
            externalStateId: 'externalStateId2',
            state: 'Open',
        },
    },
];
export const accountsDataMock = [
    {
        id: '1',
        name: 'Account Name',
        identifier: '1234 5678 9012',
        amount: 1,
        currency: 'EUR',
    },
    {
        id: '2',
        name: 'Account Name 2',
        identifier: '5678 9012 1234',
        amount: 2,
        currency: 'USD',
    },
];
