// @backbase/consent-request-widget-ang/esm2015/src/consent-request-widget/consent-request-error.js
export class ConsentRequestAccessDenied {
    constructor(error) {
        this.error = error;
        this.kind = 'ConsentRequestAccessDenied';
    }
}
export class ConsentRequestNotFound {
    constructor(error) {
        this.error = error;
        this.kind = 'ConsentRequestNotFound';
    }
}
export class ConsentRequestBadRequest {
    constructor(error) {
        this.error = error;
        this.kind = 'ConsentRequestBadRequest';
    }
}
export class ConsentRequestUnknownError {
    constructor(error) {
        this.error = error;
        this.kind = 'ConsentRequestUnknownError';
    }
}
export class ConsentRequestConnectivityError {
    constructor(error) {
        this.error = error;
        this.kind = 'ConsentRequestConnectivityError';
    }
}
export const parseError = (error) => {
    if (error.error instanceof ErrorEvent) {
        return new ConsentRequestUnknownError(error);
    }
    else {
        switch (error.status) {
            case 0:
                return new ConsentRequestConnectivityError(error);
            case 400:
                return new ConsentRequestBadRequest(error);
            case 401:
                return new ConsentRequestAccessDenied(error);
            case 403:
                return new ConsentRequestAccessDenied(error);
            case 404:
                return new ConsentRequestNotFound(error);
            default:
                return new ConsentRequestUnknownError(error);
        }
    }
};
