// @backbase/consent-request-widget-ang/esm2015/src/consent-request-widget/consent-request-preferences.service.js
import { Injectable, Optional } from '@angular/core';
import { map } from 'rxjs/operators';
import { BehaviorSubject, combineLatest, of } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
export class ConsentRequestPreferencesService {
    constructor(itemModel) {
        this.itemModel = itemModel;
        /**
         * Stream of multiAccountSelection property (based on CXP value)
         */
        this.multiAccountSelectionItemModel = this.getItemModelProperty('multiAccountSelection', false).pipe(map(Boolean));
        /**
         * Stream of multiAccountSelection property (based on INPUT value)
         */
        this.multiAccountSelectionInput = new BehaviorSubject(false);
        /**
         * Stream of multiAccountSelection property (based on BOTH values)
         */
        this.multiAccountSelection$ = this.getProperty(this.multiAccountSelectionInput, this.multiAccountSelectionItemModel);
    }
    getProperty(input$, itemModel) {
        return combineLatest([input$, itemModel]).pipe(map(([value, valueIM]) => (value === undefined ? valueIM : value)));
    }
    getItemModelProperty(name, defaultVal) {
        var _a;
        return ((_a = this.itemModel) === null || _a === void 0 ? void 0 : _a.property(name, defaultVal)) || of(defaultVal);
    }
}
ConsentRequestPreferencesService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentRequestPreferencesService, deps: [{ token: i1.ItemModel, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
ConsentRequestPreferencesService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentRequestPreferencesService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentRequestPreferencesService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.ItemModel, decorators: [{
                    type: Optional
                }] }]; } });
