// @backbase/user-manage-profile-widget-ang/esm2015/src/user-manage-phone-numbers/user-manage-phone-numbers.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { BadgeModule } from '@backbase/ui-ang/badge';
import { ButtonModule } from '@backbase/ui-ang/button';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { HeaderModule } from '@backbase/ui-ang/header';
import { IconModule } from '@backbase/ui-ang/icon';
import { NotificationModule } from '@backbase/ui-ang/notification';
import { PhoneNumberModule } from '@backbase/ui-ang/phone-number-format-pipe';
import { UserProfileFormsModule } from '../user-profile-forms/user-profile-forms.module';
import { UserEditPhoneNumberComponent, UserEditPhoneNumberCustomizableDirective, UserEditPhoneNumberTitleCustomizableDirective, } from './user-edit-phone-number/user-edit-phone-number.component';
import { UserManagePhoneNumberComponent, UserManagePhoneNumberCustomizableDirective, } from './user-manage-phone-number.component';
import { UserManagePhoneNumbersComponent, UserManagePhoneNumbersCustomizableDirective, UserManagePhoneNumbersTitleCustomizableDirective, } from './user-manage-phone-numbers.component';
import { UserNewPhoneNumberComponent, UserNewPhoneNumberCustomizableDirective, UserNewPhoneNumberTitleCustomizableDirective, } from './user-new-phone-number/user-new-phone-number.component';
import * as i0 from "@angular/core";
const uiModules = [
    ButtonModule,
    EmptyStateModule,
    BadgeModule,
    IconModule,
    NotificationModule,
    PhoneNumberModule,
    HeaderModule,
];
const componentModules = [UserProfileFormsModule];
export class UserManagePhoneNumbersModule {
}
UserManagePhoneNumbersModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserManagePhoneNumbersModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
UserManagePhoneNumbersModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserManagePhoneNumbersModule, declarations: [UserManagePhoneNumberComponent,
        UserManagePhoneNumberCustomizableDirective,
        UserManagePhoneNumbersComponent,
        UserManagePhoneNumbersCustomizableDirective,
        UserManagePhoneNumbersTitleCustomizableDirective,
        UserNewPhoneNumberComponent,
        UserNewPhoneNumberCustomizableDirective,
        UserNewPhoneNumberTitleCustomizableDirective,
        UserEditPhoneNumberComponent,
        UserEditPhoneNumberCustomizableDirective,
        UserEditPhoneNumberTitleCustomizableDirective], imports: [CommonModule, UserProfileFormsModule, ButtonModule,
        EmptyStateModule,
        BadgeModule,
        IconModule,
        NotificationModule,
        PhoneNumberModule,
        HeaderModule], exports: [UserManagePhoneNumberComponent,
        UserManagePhoneNumberCustomizableDirective,
        UserManagePhoneNumbersComponent,
        UserManagePhoneNumbersCustomizableDirective,
        UserManagePhoneNumbersTitleCustomizableDirective,
        UserNewPhoneNumberComponent,
        UserNewPhoneNumberCustomizableDirective,
        UserNewPhoneNumberTitleCustomizableDirective,
        UserEditPhoneNumberComponent,
        UserEditPhoneNumberCustomizableDirective,
        UserEditPhoneNumberTitleCustomizableDirective, ButtonModule,
        EmptyStateModule,
        BadgeModule,
        IconModule,
        NotificationModule,
        PhoneNumberModule,
        HeaderModule, UserProfileFormsModule] });
UserManagePhoneNumbersModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserManagePhoneNumbersModule, imports: [[CommonModule, ...componentModules, ...uiModules], ButtonModule,
        EmptyStateModule,
        BadgeModule,
        IconModule,
        NotificationModule,
        PhoneNumberModule,
        HeaderModule, UserProfileFormsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserManagePhoneNumbersModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...componentModules, ...uiModules],
                    declarations: [
                        UserManagePhoneNumberComponent,
                        UserManagePhoneNumberCustomizableDirective,
                        UserManagePhoneNumbersComponent,
                        UserManagePhoneNumbersCustomizableDirective,
                        UserManagePhoneNumbersTitleCustomizableDirective,
                        UserNewPhoneNumberComponent,
                        UserNewPhoneNumberCustomizableDirective,
                        UserNewPhoneNumberTitleCustomizableDirective,
                        UserEditPhoneNumberComponent,
                        UserEditPhoneNumberCustomizableDirective,
                        UserEditPhoneNumberTitleCustomizableDirective,
                    ],
                    exports: [
                        UserManagePhoneNumberComponent,
                        UserManagePhoneNumberCustomizableDirective,
                        UserManagePhoneNumbersComponent,
                        UserManagePhoneNumbersCustomizableDirective,
                        UserManagePhoneNumbersTitleCustomizableDirective,
                        UserNewPhoneNumberComponent,
                        UserNewPhoneNumberCustomizableDirective,
                        UserNewPhoneNumberTitleCustomizableDirective,
                        UserEditPhoneNumberComponent,
                        UserEditPhoneNumberCustomizableDirective,
                        UserEditPhoneNumberTitleCustomizableDirective,
                        ...uiModules,
                        ...componentModules,
                    ],
                }]
        }] });
