// @backbase/manage-payees-journey-ang/esm2015/src/views/billpay-edit-manual-payee/billpay-edit-manual-payee-view.component.js
import { Component, HostListener } from '@angular/core';
import { switchMap, map, catchError, tap } from 'rxjs/operators';
import { of, combineLatest } from 'rxjs';
import { BillpayDiscardChangesModalService, } from '@backbase/billpay-journeys-common';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "../../services/billpay-manage-payee-state.service";
import * as i3 from "../../services/billpay-manage-payee-data.service";
import * as i4 from "../../services/manage-payees-navigation.service";
import * as i5 from "@backbase/billpay-journeys-common";
import * as i6 from "../../components/add-payee/billpay-manual-payee-form/billpay-manual-payee-form.component";
import * as i7 from "../../components/add-payee/billpay-cancel-edit-payee-modal/billpay-cancel-edit-payee-modal.component";
import * as i8 from "@backbase/ui-ang/loading-indicator";
import * as i9 from "@backbase/ui-ang/empty-state";
import * as i10 from "@angular/common";
export class BillpayEditManualPayeeViewComponent {
    /** @internal */
    constructor(route, stateService, dataService, router, discardModalService) {
        this.route = route;
        this.stateService = stateService;
        this.dataService = dataService;
        this.router = router;
        this.discardModalService = discardModalService;
        this.hasLoadingError = false;
        this.isFormTouched = false;
        this.payeeId = '';
    }
    ngOnInit() {
        this.payeeFormState = combineLatest([
            this.route.paramMap.pipe(map(params => params.get('id') || '')),
            this.stateService.manualPayeeFormState,
        ]).pipe(tap(([id]) => (this.payeeId = id)), switchMap(([id, state]) => (state && of(state)) ||
            this.dataService.getManualPayee(id).pipe(map(payee => payee && this.stateService.toManualFormState(payee)), catchError(() => {
                this.hasLoadingError = true;
                return of({});
            }))));
    }
    get isDiscardModalOpen() {
        return this.discardModalService.isDiscardModalOpen;
    }
    onWindowUnload($event) {
        if ((this.isFormTouched || this.stateService.isFormProcessing) && !this.isDiscardModalOpen) {
            $event.returnValue = true;
        }
    }
    onFormSubmitted(payee) {
        this.stateService.isFormProcessing = true;
        this.stateService.setManualPayee(payee);
        this.router.navigateToEditManualReview(this.payeeId);
    }
    onFormValueChanged() {
        this.isFormTouched = true;
    }
    onFormCanceled() {
        this.router.navigateToPayeeList();
    }
    openModal() {
        return this.isFormTouched || this.stateService.isFormProcessing ? this.discardModalService.openModal() : of(true);
    }
    onModalResolve(resolvedWith) {
        this.discardModalService.onModalResolve(resolvedWith);
        if (resolvedWith) {
            this.router.navigateToPayeeList();
        }
    }
}
BillpayEditManualPayeeViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayEditManualPayeeViewComponent, deps: [{ token: i1.ActivatedRoute }, { token: i2.BillpayManagePayeeStateService }, { token: i3.BillpayManagePayeeDataService }, { token: i4.ManagePayeesNavigationService }, { token: i5.BillpayDiscardChangesModalService }], target: i0.ɵɵFactoryTarget.Component });
BillpayEditManualPayeeViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: BillpayEditManualPayeeViewComponent, selector: "ng-component", host: { listeners: { "window:beforeunload": "onWindowUnload($event)" } }, providers: [BillpayDiscardChangesModalService], ngImport: i0, template: "<ng-container *ngIf=\"payeeFormState | async as payeeState; else loadingState\">\n  <ng-container *ngIf=\"!hasLoadingError; else errorState\">\n    <bb-billpay-manual-payee-form\n      [editForm]=\"true\"\n      [payeeFormState]=\"payeeState\"\n      (formSubmitted)=\"onFormSubmitted($event)\"\n      (formCanceled)=\"onFormCanceled()\"\n      (valueChanged)=\"onFormValueChanged()\"\n    >\n    </bb-billpay-manual-payee-form>\n  </ng-container>\n</ng-container>\n\n<bb-billpay-cancel-edit-payee-modal\n  *ngIf=\"isDiscardModalOpen\"\n  (resolve)=\"onModalResolve($event)\"\n></bb-billpay-cancel-edit-payee-modal>\n\n<ng-template #loadingState>\n  <bb-loading-indicator-ui loaderSize=\"lg\"> </bb-loading-indicator-ui>\n</ng-template>\n\n<ng-template #errorState>\n  <bb-empty-state-ui\n    data-role=\"error-state\"\n    iconSize=\"xxl\"\n    iconModifier=\"error\"\n    subtitle=\"Try to reload the page or contact support\"\n    i18n-subtitle=\"\n      Try to reload the page or contact support|Edit manual payee\n      subtitle@@bb-billpay-manage-payee-widget-ang.loading.subtitle\"\n    title=\"Payee is not loading\"\n    i18n-title=\"Payee is not loading|Edit manual payee loading title@@bb-billpay-manage-payee-widget-ang.loading.title\"\n  >\n  </bb-empty-state-ui>\n</ng-template> \n", components: [{ type: i6.BillpayManualPayeeFormComponent, selector: "bb-billpay-manual-payee-form", inputs: ["payeeFormState", "editForm"], outputs: ["formSubmitted", "formCanceled", "backToSearch", "valueChanged"] }, { type: i7.BillpayCancelEditPayeeModalComponent, selector: "bb-billpay-cancel-edit-payee-modal", outputs: ["resolve"] }, { type: i8.LoadingIndicatorComponent, selector: "bb-loading-indicator-ui", inputs: ["text", "loaderSize", "showDelay", "hasBackground", "inline"] }, { type: i9.EmptyStateComponent, selector: "bb-empty-state-ui", inputs: ["title", "subtitle", "showIcon", "iconClasses", "iconModifier", "iconSize", "iconColor"] }], directives: [{ type: i10.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "async": i10.AsyncPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayEditManualPayeeViewComponent, decorators: [{
            type: Component,
            args: [{
                    templateUrl: './billpay-edit-manual-payee-view.component.html',
                    providers: [BillpayDiscardChangesModalService],
                }]
        }], ctorParameters: function () { return [{ type: i1.ActivatedRoute }, { type: i2.BillpayManagePayeeStateService }, { type: i3.BillpayManagePayeeDataService }, { type: i4.ManagePayeesNavigationService }, { type: i5.BillpayDiscardChangesModalService }]; }, propDecorators: { onWindowUnload: [{
                type: HostListener,
                args: ['window:beforeunload', ['$event']]
            }] } });
