// @backbase/authorized-users-journey-ang/esm2015/src/authorized-users-create/components/authorized-users-create-common/authorized-users-step-handler.directive.js
import { Directive, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../authorized-users-create-container/authorized-users-create-container.component";
export class AuthorizedUserStepHandlerDirective {
    constructor(templateRef, viewContainerRef, parent) {
        this.templateRef = templateRef;
        this.viewContainerRef = viewContainerRef;
        this.parent = parent;
        this._currentStep = 0;
    }
    showOrHideStep() {
        if (this._currentStep === this.step) {
            this.viewContainerRef.createEmbeddedView(this.templateRef);
        }
        else {
            this.viewContainerRef.clear();
        }
    }
    set currentStep(value) {
        this._currentStep = value;
        this.showOrHideStep();
    }
    ngOnInit() {
        this.step = this._currentStep > 0 ? this.parent.nextStepIndex() : undefined;
        this.showOrHideStep();
    }
}
AuthorizedUserStepHandlerDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUserStepHandlerDirective, deps: [{ token: i0.TemplateRef }, { token: i0.ViewContainerRef }, { token: i1.AuthorizedUsersCreateContainerComponent }], target: i0.ɵɵFactoryTarget.Directive });
AuthorizedUserStepHandlerDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: AuthorizedUserStepHandlerDirective, selector: "[bbAuthorizedUsersStepHandler]", inputs: { currentStep: ["bbAuthorizedUsersStepHandler", "currentStep"] }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUserStepHandlerDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbAuthorizedUsersStepHandler]',
                }]
        }], ctorParameters: function () { return [{ type: i0.TemplateRef }, { type: i0.ViewContainerRef }, { type: i1.AuthorizedUsersCreateContainerComponent }]; }, propDecorators: { currentStep: [{
                type: Input,
                args: ['bbAuthorizedUsersStepHandler']
            }] } });
