// @backbase/authorized-users-journey-ang/esm2015/src/authorized-users-create/components/authorized-users-create-common/base-authorized-users-create-step.js
import { Output, EventEmitter, Input, Directive } from '@angular/core';
import * as i0 from "@angular/core";
// eslint-disable-next-line
export class BaseAuthorizedUsersCreateStep {
    constructor() {
        this.next = new EventEmitter();
        this.back = new EventEmitter();
        this.saveOnly = new EventEmitter();
        this.cancel = new EventEmitter();
        this.isFirstStep = false;
    }
    goNext(authorizedUser) {
        this.next.emit(authorizedUser);
    }
    goBack() {
        this.back.emit();
    }
    saveChanges(authorizedUser) {
        this.saveOnly.emit(authorizedUser);
    }
    abort() {
        this.cancel.emit();
    }
    ngOnInit() {
        if (!this.authorizedUser) {
            throw new Error(`"authorizedUser" input is required in "${this.constructor.name}"`);
        }
    }
}
BaseAuthorizedUsersCreateStep.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BaseAuthorizedUsersCreateStep, deps: [], target: i0.ɵɵFactoryTarget.Directive });
BaseAuthorizedUsersCreateStep.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: BaseAuthorizedUsersCreateStep, inputs: { authorizedUser: "authorizedUser", isFirstStep: "isFirstStep", enableAccountLevelFullAccessPermission: "enableAccountLevelFullAccessPermission", levelsConfig: "levelsConfig", permissionsConfig: "permissionsConfig" }, outputs: { next: "next", back: "back", saveOnly: "saveOnly", cancel: "cancel" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BaseAuthorizedUsersCreateStep, decorators: [{
            type: Directive
        }], propDecorators: { next: [{
                type: Output
            }], back: [{
                type: Output
            }], saveOnly: [{
                type: Output
            }], cancel: [{
                type: Output
            }], authorizedUser: [{
                type: Input
            }], isFirstStep: [{
                type: Input
            }], enableAccountLevelFullAccessPermission: [{
                type: Input
            }], levelsConfig: [{
                type: Input
            }], permissionsConfig: [{
                type: Input
            }] } });
