// @backbase/consent-journey-ang/esm2015/src/views/consents/consent-detail/consent-detail.component.js
import { Component } from '@angular/core';
import { Subject } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "../../../services/consent-journey-config.service";
import * as i2 from "@angular/router";
import * as i3 from "@backbase/consent-details-widget-ang";
export class ConsentDetailComponent {
    constructor(config, router, activeRoute) {
        this.config = config;
        this.router = router;
        this.activeRoute = activeRoute;
        this.destroy$ = new Subject();
    }
    handleBack() {
        this.router.navigate(['../list'], {
            relativeTo: this.activeRoute,
        });
    }
    ngOnDestroy() {
        this.destroy$.next();
    }
}
ConsentDetailComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentDetailComponent, deps: [{ token: i1.ConsentJourneyConfigurationService }, { token: i2.Router }, { token: i2.ActivatedRoute }], target: i0.ɵɵFactoryTarget.Component });
ConsentDetailComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: ConsentDetailComponent, selector: "bb-consent-detail-view", ngImport: i0, template: ` <div class="bb-card">
    <div class="bb-card__body">
      <div class="justify-content-start" data-role="consent-heading">
        <button
          data-role="consent-heading-button__navigation"
          (click)="handleBack()"
          class="btn btn-link bb-block bb-block--xl">
          <i aria-hidden="true" class="bb-icon bb-icon-chevron-left"></i>
          <span i18n="Consent back button title@@consent-journey.detail.button.title">Back to list</span>
        </button>
      </div>
      <bb-consent-details-widget></bb-consent-details-widget>
    </div>
  </div>`, isInline: true, components: [{ type: i3.ConsentDetailsWidgetComponent, selector: "bb-consent-details-widget" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentDetailComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-consent-detail-view',
                    template: ` <div class="bb-card">
    <div class="bb-card__body">
      <div class="justify-content-start" data-role="consent-heading">
        <button
          data-role="consent-heading-button__navigation"
          (click)="handleBack()"
          class="btn btn-link bb-block bb-block--xl">
          <i aria-hidden="true" class="bb-icon bb-icon-chevron-left"></i>
          <span i18n="Consent back button title@@consent-journey.detail.button.title">Back to list</span>
        </button>
      </div>
      <bb-consent-details-widget></bb-consent-details-widget>
    </div>
  </div>`,
                }]
        }], ctorParameters: function () { return [{ type: i1.ConsentJourneyConfigurationService }, { type: i2.Router }, { type: i2.ActivatedRoute }]; } });
