// @backbase/ui-ang/esm2015/_example/char-counter/showcase/char-counter-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { CharCounterShowcaseComponent } from './char-counter-showcase.component';
import { CharCounterModule } from '@backbase/ui-ang/char-counter';
import { TextareaModule } from '@backbase/ui-ang/textarea';
import * as i0 from "@angular/core";
export class CharCounterShowcaseModule {
}
CharCounterShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CharCounterShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CharCounterShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CharCounterShowcaseModule, declarations: [CharCounterShowcaseComponent], imports: [CommonModule, CharCounterModule, ReactiveFormsModule, TextareaModule] });
CharCounterShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CharCounterShowcaseModule, imports: [[CommonModule, CharCounterModule, ReactiveFormsModule, TextareaModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CharCounterShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, CharCounterModule, ReactiveFormsModule, TextareaModule],
                    declarations: [CharCounterShowcaseComponent],
                    entryComponents: [CharCounterShowcaseComponent],
                }]
        }] });
