// @backbase/messages-conversations-list-widget-ang/esm2015/src/messages-conversations-list-widget-ang.service.js
import { Inject, Injectable } from '@angular/core';
import { map, tap } from 'rxjs/operators';
import { PUBSUB } from '@backbase/foundation-ang/web-sdk';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/messages-v5-http-ang";
import * as i2 from "@backbase/messages-common";
const DefaultConversationsGetReqParams = {
    direction: 'DESC',
    from: 0,
    size: 10,
    mailboxType: '',
};
const bbEventUnreadConversationCountUpdate = 'bb.event.unread-conversation-count.update';
export class MessagesConversationsListWidgetService {
    constructor(messagesService, sharedMethodsService, pubSub) {
        this.messagesService = messagesService;
        this.sharedMethodsService = sharedMethodsService;
        this.pubSub = pubSub;
    }
    deleteConversations(ids) {
        return this.messagesService
            .deleteConversationThreads({ ids })
            .pipe(tap(() => this.pubSub.publish(bbEventUnreadConversationCountUpdate, undefined)));
    }
    deleteDrafts(ids) {
        return this.messagesService.deleteMessageDrafts({ ids });
    }
    getConversations(params = DefaultConversationsGetReqParams) {
        return this.messagesService
            .getConversationThreads(params, 'response')
            .pipe(map((response) => this.sharedMethodsService.processData(response)));
    }
    getDrafts({ from, size }) {
        return this.messagesService
            .getMessageDrafts({ from, size }, 'response')
            .pipe(map((response) => this.sharedMethodsService.processData(response)));
    }
    markConversationsAsReadUnread(params) {
        return this.messagesService
            .putMarkAsReadUnread({ markAsReadUnreadPutRequestBody: params })
            .pipe(tap(() => this.pubSub.publish(bbEventUnreadConversationCountUpdate, undefined)));
    }
}
MessagesConversationsListWidgetService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesConversationsListWidgetService, deps: [{ token: i1.MessagecenterHttpService }, { token: i2.SharedMethodsService }, { token: PUBSUB }], target: i0.ɵɵFactoryTarget.Injectable });
MessagesConversationsListWidgetService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesConversationsListWidgetService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesConversationsListWidgetService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.MessagecenterHttpService }, { type: i2.SharedMethodsService }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [PUBSUB]
                }] }]; } });
