// @backbase/messages-conversations-list-widget-ang/esm2015/src/messages-conversations-list-widget-ang.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { MessagesRequestErrorModule, SharedMethodsModule } from '@backbase/messages-common';
import { BadgeModule } from '@backbase/ui-ang/badge';
import { InputCheckboxModule } from '@backbase/ui-ang/input-checkbox';
import { DropdownMenuModule } from '@backbase/ui-ang/dropdown-menu';
import { CheckboxGroupModule } from '@backbase/ui-ang/checkbox-group';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { IconModule } from '@backbase/ui-ang/icon';
import { ButtonModule } from '@backbase/ui-ang/button';
import { ModalModule } from '@backbase/ui-ang/modal';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { KeyboardClickModule } from '@backbase/ui-ang/keyboard-click-directive';
import { MessagesConversationsListCustomizableDirective, MessagesConversationsListWidgetComponent, } from './messages-conversations-list-widget-ang.component';
import { MessagesConversationsListWidgetService } from './messages-conversations-list-widget-ang.service';
import { ConversationsListTableComponent, ConversationsListTableCustomizableDirective, } from './components/conversations-list-table/conversations-list-table.component';
import { ConversationsListTableHeadComponent, ConversationsListTableHeadCustomizableDirective, } from './components/conversations-list-table-head/conversations-list-table-head.component';
import { ConversationsListTableRowComponent, ConversationsListTableRowCustomizableDirective, } from './components/conversations-list-table-row/conversations-list-table-row.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const components = [
    MessagesConversationsListWidgetComponent,
    ConversationsListTableComponent,
    ConversationsListTableHeadComponent,
    ConversationsListTableRowComponent,
];
const directives = [
    MessagesConversationsListCustomizableDirective,
    ConversationsListTableCustomizableDirective,
    ConversationsListTableHeadCustomizableDirective,
    ConversationsListTableRowCustomizableDirective,
];
const uiModules = [
    BadgeModule,
    ButtonModule,
    CheckboxGroupModule,
    DropdownMenuModule,
    ModalModule,
    IconModule,
    InputCheckboxModule,
    LoadButtonModule,
    EmptyStateModule,
    LoadingIndicatorModule,
    TooltipModule,
];
const innerModules = [MessagesRequestErrorModule];
export class MessagesConversationsListWidgetModule {
}
MessagesConversationsListWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesConversationsListWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MessagesConversationsListWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesConversationsListWidgetModule, declarations: [MessagesConversationsListWidgetComponent,
        ConversationsListTableComponent,
        ConversationsListTableHeadComponent,
        ConversationsListTableRowComponent, MessagesConversationsListCustomizableDirective,
        ConversationsListTableCustomizableDirective,
        ConversationsListTableHeadCustomizableDirective,
        ConversationsListTableRowCustomizableDirective], imports: [CommonModule, i1.BackbaseCoreModule, KeyboardClickModule,
        SharedMethodsModule,
        FormsModule, BadgeModule,
        ButtonModule,
        CheckboxGroupModule,
        DropdownMenuModule,
        ModalModule,
        IconModule,
        InputCheckboxModule,
        LoadButtonModule,
        EmptyStateModule,
        LoadingIndicatorModule,
        TooltipModule, MessagesRequestErrorModule], exports: [MessagesConversationsListWidgetComponent,
        ConversationsListTableComponent,
        ConversationsListTableHeadComponent,
        ConversationsListTableRowComponent, MessagesConversationsListCustomizableDirective,
        ConversationsListTableCustomizableDirective,
        ConversationsListTableHeadCustomizableDirective,
        ConversationsListTableRowCustomizableDirective, BadgeModule,
        ButtonModule,
        CheckboxGroupModule,
        DropdownMenuModule,
        ModalModule,
        IconModule,
        InputCheckboxModule,
        LoadButtonModule,
        EmptyStateModule,
        LoadingIndicatorModule,
        TooltipModule, MessagesRequestErrorModule] });
MessagesConversationsListWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesConversationsListWidgetModule, providers: [MessagesConversationsListWidgetService], imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { MessagesConversationsListWidgetComponent },
            }),
            KeyboardClickModule,
            SharedMethodsModule,
            FormsModule,
            ...uiModules,
            ...innerModules,
        ], BadgeModule,
        ButtonModule,
        CheckboxGroupModule,
        DropdownMenuModule,
        ModalModule,
        IconModule,
        InputCheckboxModule,
        LoadButtonModule,
        EmptyStateModule,
        LoadingIndicatorModule,
        TooltipModule, MessagesRequestErrorModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesConversationsListWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { MessagesConversationsListWidgetComponent },
                        }),
                        KeyboardClickModule,
                        SharedMethodsModule,
                        FormsModule,
                        ...uiModules,
                        ...innerModules,
                    ],
                    providers: [MessagesConversationsListWidgetService],
                    declarations: [...components, ...directives],
                    exports: [...components, ...directives, ...uiModules, ...innerModules],
                }]
        }] });
