// @backbase/manage-payees-journey-ang/esm2015/src/views/billpay-manage-payee/billpay-manage-payee-view.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../../services/manage-payees-navigation.service";
import * as i2 from "@angular/router";
import * as i3 from "@backbase/ui-ang/icon";
import * as i4 from "@backbase/ui-ang/header";
import * as i5 from "@backbase/ui-ang/button";
export class ManagePayeeViewComponent {
    constructor(routerService, route) {
        this.routerService = routerService;
        this.route = route;
        this.editMode = false;
    }
    ngOnInit() {
        var _a;
        this.editMode = !!((_a = this.route.firstChild) === null || _a === void 0 ? void 0 : _a.snapshot.url.some(a => a.path === 'edit'));
    }
    gotoPayeeList() {
        this.routerService.navigateToPayeeList({ relativeTo: this.route.parent });
    }
}
ManagePayeeViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePayeeViewComponent, deps: [{ token: i1.ManagePayeesNavigationService }, { token: i2.ActivatedRoute }], target: i0.ɵɵFactoryTarget.Component });
ManagePayeeViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: ManagePayeeViewComponent, selector: "bb-billpay-payee-list-view", ngImport: i0, template: `
    <div class="bb-block bb-block--lg">
      <button bbButton color="link" class="bb-block bb-block--md pl-0" (click)="gotoPayeeList()">
        <bb-icon-ui name="arrow-back" size="md"></bb-icon-ui>
        <span i18n="button label|Back button@@billpay-payees-journey.payee-summary.back">Back to Payee List</span>
      </button>
      <bb-header-ui
        headingType="h1"
        [heading]="editMode ? 'Edit Payee' : 'Add Payee'"
        i18n-heading="@@billpay-payees-journey.add-payee"
      >
      </bb-header-ui>
    </div>
    <router-outlet></router-outlet>
  `, isInline: true, components: [{ type: i3.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i4.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }], directives: [{ type: i5.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }, { type: i2.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePayeeViewComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-billpay-payee-list-view',
                    template: `
    <div class="bb-block bb-block--lg">
      <button bbButton color="link" class="bb-block bb-block--md pl-0" (click)="gotoPayeeList()">
        <bb-icon-ui name="arrow-back" size="md"></bb-icon-ui>
        <span i18n="button label|Back button@@billpay-payees-journey.payee-summary.back">Back to Payee List</span>
      </button>
      <bb-header-ui
        headingType="h1"
        [heading]="editMode ? 'Edit Payee' : 'Add Payee'"
        i18n-heading="@@billpay-payees-journey.add-payee"
      >
      </bb-header-ui>
    </div>
    <router-outlet></router-outlet>
  `,
                }]
        }], ctorParameters: function () { return [{ type: i1.ManagePayeesNavigationService }, { type: i2.ActivatedRoute }]; } });
