// @backbase/ui-ang/esm2015/_example/badge/showcase/badge-showcase.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/badge";
import * as i2 from "@angular/common";
let BadgeShowcaseComponent = class BadgeShowcaseComponent {
    constructor() {
        this.badgeShowCaseSetup = [
            {
                type: 'warning',
                text: {
                    single: 'A',
                    double: 'AA',
                },
            },
            {
                type: 'danger',
                text: {
                    single: 'A',
                    double: 'AA',
                },
            },
            {
                type: 'info',
                text: {
                    single: 'A',
                    double: 'AA',
                },
            },
            {
                type: 'success',
                text: {
                    single: 'A',
                    double: 'AA',
                },
            },
            {
                type: 'secondary',
                text: {
                    single: 'A',
                    double: 'AA',
                },
            },
        ];
    }
};
BadgeShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BadgeShowcaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
BadgeShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: BadgeShowcaseComponent, selector: "bb-badge-ui-showcase", ngImport: i0, template: "<div class=\"mb-4\">\n  <h4>Single character</h4>\n  <div *ngFor=\"let badgeType of badgeShowCaseSetup\" class=\"d-inline-block mb-4 mr-3\">\n    <bb-badge-ui\n      color=\"{{ badgeType.type }}\"\n      text=\"{{ badgeType.text.single }}\"\n      [attr.data-role]=\"'badge-' + badgeType.type + '-single'\"\n    >\n    </bb-badge-ui>\n  </div>\n</div>\n\n<div class=\"mb-4\">\n  <h4>Double character</h4>\n  <div *ngFor=\"let badgeType of badgeShowCaseSetup\" class=\"d-inline-block mb-4 mr-3\">\n    <bb-badge-ui\n      color=\"{{ badgeType.type }}\"\n      text=\"{{ badgeType.text.double }}\"\n      [attr.data-role]=\"'badge-' + badgeType.type + '-double'\"\n    >\n    </bb-badge-ui>\n  </div>\n</div>\n\n<div class=\"mb-4\">\n  <h4>Text</h4>\n  <div *ngFor=\"let badgeType of badgeShowCaseSetup\" class=\"d-inline-block mb-4 mr-3\">\n    <bb-badge-ui\n      color=\"{{ badgeType.type }}\"\n      text=\"{{ badgeType.type | titlecase }}\"\n      [attr.data-role]=\"'badge-' + badgeType.type + '-text'\"\n    >\n    </bb-badge-ui>\n  </div>\n</div>\n\n<div class=\"mb-4\">\n  <h4>Abbreviation</h4>\n  <div *ngFor=\"let badgeType of badgeShowCaseSetup\" class=\"d-inline-block mb-4 mr-3\">\n    <bb-badge-ui\n      color=\"{{ badgeType.type }}\"\n      text=\"{{ badgeType.type | titlecase }}\"\n      ariaLabel=\"{{ badgeType.type }}\"\n      [abbr]=\"true\"\n      [attr.data-role]=\"'badge-' + badgeType.type + '-abbr'\"\n    >\n    </bb-badge-ui>\n  </div>\n</div>\n", components: [{ type: i1.BadgeComponent, selector: "bb-badge-ui", inputs: ["color", "text", "abbr", "circle", "wrap", "ariaLabel"] }], directives: [{ type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }], pipes: { "titlecase": i2.TitleCasePipe } });
BadgeShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
], BadgeShowcaseComponent);
export { BadgeShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BadgeShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-badge-ui-showcase',
                    templateUrl: './badge-showcase.component.html',
                }]
        }] });
