// @backbase/ui-ang/esm2015/_example/badge/showcase/badge-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BadgeShowcaseComponent } from './badge-showcase.component';
import { BadgeModule } from '@backbase/ui-ang/badge';
import * as i0 from "@angular/core";
export class BadgeShowcaseModule {
}
BadgeShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BadgeShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
BadgeShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BadgeShowcaseModule, declarations: [BadgeShowcaseComponent], imports: [CommonModule, BadgeModule] });
BadgeShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BadgeShowcaseModule, imports: [[CommonModule, BadgeModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BadgeShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, BadgeModule],
                    declarations: [BadgeShowcaseComponent],
                    entryComponents: [BadgeShowcaseComponent],
                }]
        }] });
