// @backbase/user-common/esm2015/src/components/bb-cancel-modal/cancel-modal.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ButtonModule } from '@backbase/ui-ang/button';
import { HeaderModule } from '@backbase/ui-ang/header';
import { ModalModule } from '@backbase/ui-ang/modal';
import { CancelModalComponent } from './cancel-modal.component';
import * as i0 from "@angular/core";
const uiModules = [ButtonModule, ModalModule, HeaderModule];
export class CancelModalModule {
}
CancelModalModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CancelModalModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CancelModalModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CancelModalModule, declarations: [CancelModalComponent], imports: [CommonModule, ButtonModule, ModalModule, HeaderModule], exports: [CancelModalComponent, ButtonModule, ModalModule, HeaderModule] });
CancelModalModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CancelModalModule, imports: [[CommonModule, ...uiModules], ButtonModule, ModalModule, HeaderModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CancelModalModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules],
                    declarations: [CancelModalComponent],
                    exports: [CancelModalComponent, ...uiModules],
                }]
        }] });
