// @backbase/user-identity-security-center-widget-ang/esm2015/src/user-identity-security-center-widget.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { AlertModule } from '@backbase/ui-ang/alert';
import { ButtonModule } from '@backbase/ui-ang/button';
import { ErrorCommonStateModule } from '@backbase/ui-ang/common-error-state';
import { HeaderModule } from '@backbase/ui-ang/header';
import { InputPasswordModule } from '@backbase/ui-ang/input-password';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { ModalModule } from '@backbase/ui-ang/modal';
import { UserCommonModule } from '@backbase/user-common';
import { ChangePasswordFormModule } from './change-password-form/change-password-form.module';
import { UserIdentitySecurityCenterCustomizableDirective, UserIdentitySecurityCenterWidgetComponent, } from './user-identity-security-center-widget.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const uiModules = [
    AlertModule,
    ButtonModule,
    ErrorCommonStateModule,
    HeaderModule,
    LoadingIndicatorModule,
    ModalModule,
    InputPasswordModule,
    InputValidationMessageModule,
    ReactiveFormsModule,
];
const widgetModules = [ChangePasswordFormModule];
export class UserIdentitySecurityCenterWidgetModule {
}
UserIdentitySecurityCenterWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserIdentitySecurityCenterWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
UserIdentitySecurityCenterWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserIdentitySecurityCenterWidgetModule, declarations: [UserIdentitySecurityCenterWidgetComponent, UserIdentitySecurityCenterCustomizableDirective], imports: [CommonModule, i1.BackbaseCoreModule, UserCommonModule, AlertModule,
        ButtonModule,
        ErrorCommonStateModule,
        HeaderModule,
        LoadingIndicatorModule,
        ModalModule,
        InputPasswordModule,
        InputValidationMessageModule,
        ReactiveFormsModule, ChangePasswordFormModule], exports: [UserIdentitySecurityCenterWidgetComponent,
        UserIdentitySecurityCenterCustomizableDirective, AlertModule,
        ButtonModule,
        ErrorCommonStateModule,
        HeaderModule,
        LoadingIndicatorModule,
        ModalModule,
        InputPasswordModule,
        InputValidationMessageModule,
        ReactiveFormsModule, ChangePasswordFormModule] });
UserIdentitySecurityCenterWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserIdentitySecurityCenterWidgetModule, imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { UserIdentitySecurityCenterWidgetComponent },
            }),
            UserCommonModule,
            ...uiModules,
            ...widgetModules,
        ], AlertModule,
        ButtonModule,
        ErrorCommonStateModule,
        HeaderModule,
        LoadingIndicatorModule,
        ModalModule,
        InputPasswordModule,
        InputValidationMessageModule,
        ReactiveFormsModule, ChangePasswordFormModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserIdentitySecurityCenterWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { UserIdentitySecurityCenterWidgetComponent },
                        }),
                        UserCommonModule,
                        ...uiModules,
                        ...widgetModules,
                    ],
                    declarations: [UserIdentitySecurityCenterWidgetComponent, UserIdentitySecurityCenterCustomizableDirective],
                    exports: [
                        UserIdentitySecurityCenterWidgetComponent,
                        UserIdentitySecurityCenterCustomizableDirective,
                        ...uiModules,
                        ...widgetModules,
                    ],
                }]
        }] });
