// @backbase/pay-bills-journey-ang/esm2015/src/views/billpay-payment-multiple-bills-view.component.js
import { Component, ViewChild } from '@angular/core';
import { BillpayDiscardChangesModalService, } from '@backbase/billpay-journeys-common';
import * as i0 from "@angular/core";
import * as i1 from "../services/pay-bills-navigation.service";
import * as i2 from "@backbase/ui-ang/header";
import * as i3 from "../components/billpay-payment-multiple-bills/billpay-payment-multiple-bills.component";
import * as i4 from "@backbase/ui-ang/button";
export class BillpayPaymentMultipleBillsViewComponent {
    constructor(navigationService) {
        this.navigationService = navigationService;
    }
    openModal() {
        return this.child.openModal();
    }
    addPayee() {
        var _a;
        (_a = this.navigationService) === null || _a === void 0 ? void 0 : _a.navigateToAddPayee();
    }
}
BillpayPaymentMultipleBillsViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayPaymentMultipleBillsViewComponent, deps: [{ token: i1.PayBillsNavigationService }], target: i0.ɵɵFactoryTarget.Component });
BillpayPaymentMultipleBillsViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: BillpayPaymentMultipleBillsViewComponent, selector: "ng-component", providers: [BillpayDiscardChangesModalService], viewQueries: [{ propertyName: "child", first: true, predicate: ["multipleBillsReview"], descendants: true }], ngImport: i0, template: `
    <div class="bb-stack bb-block bb-block--lg">
      <div class="bb-stack__item bb-stack__item--fill">
        <bb-header-ui
          headingType="h1"
          heading="Pay Bills"
          i18n-heading="@@pay-bills-journey.title.pay-bills"
          data-role="multiple-bills-widget-title"
        >
        </bb-header-ui>
        <p i18n="@@pay-bills-journey.pay-payee.header-description" data-role="multiple-bills-widget-title-info">
          Add payees if you would like to pay a single bill or multiple bills
        </p>
      </div>
      <div class="bb-stack__item">
        <button bbButton color="secondary" data-role="pay-a-bill-add-payee" (click)="addPayee()">
          <i class="bb-icon bb-icon-add"></i>
          <span class="bb-accounts-overview-table__header-row" i18n="@@pay-bills-journey.pay-bills.add-payee-button"
            >Add Payee</span
          >
        </button>
      </div>
    </div>
    <bb-billpay-payment-multiple-bills #multipleBillsReview></bb-billpay-payment-multiple-bills>
  `, isInline: true, components: [{ type: i2.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i3.BillpayPaymentMultipleBillsComponent, selector: "bb-billpay-payment-multiple-bills" }], directives: [{ type: i4.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayPaymentMultipleBillsViewComponent, decorators: [{
            type: Component,
            args: [{
                    template: `
    <div class="bb-stack bb-block bb-block--lg">
      <div class="bb-stack__item bb-stack__item--fill">
        <bb-header-ui
          headingType="h1"
          heading="Pay Bills"
          i18n-heading="@@pay-bills-journey.title.pay-bills"
          data-role="multiple-bills-widget-title"
        >
        </bb-header-ui>
        <p i18n="@@pay-bills-journey.pay-payee.header-description" data-role="multiple-bills-widget-title-info">
          Add payees if you would like to pay a single bill or multiple bills
        </p>
      </div>
      <div class="bb-stack__item">
        <button bbButton color="secondary" data-role="pay-a-bill-add-payee" (click)="addPayee()">
          <i class="bb-icon bb-icon-add"></i>
          <span class="bb-accounts-overview-table__header-row" i18n="@@pay-bills-journey.pay-bills.add-payee-button"
            >Add Payee</span
          >
        </button>
      </div>
    </div>
    <bb-billpay-payment-multiple-bills #multipleBillsReview></bb-billpay-payment-multiple-bills>
  `,
                    providers: [BillpayDiscardChangesModalService],
                }]
        }], ctorParameters: function () { return [{ type: i1.PayBillsNavigationService }]; }, propDecorators: { child: [{
                type: ViewChild,
                args: ['multipleBillsReview']
            }] } });
