// @backbase/pay-bills-journey-ang/esm2015/src/views/billpay-payment-recurring-edit-form-view.component.js
import { Component, ViewChild } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../services/pay-bills-journey-config.service";
import * as i2 from "../services/pay-bills-navigation.service";
import * as i3 from "../components/billpay-payment-single-bill/billpay-payment-recurring-edit-form-container/billpay-payment-recurring-edit-form-container.component";
import * as i4 from "@backbase/ui-ang/button";
export class BillpayPaymentRecurringEditFormViewComponent {
    constructor(config, routerService) {
        this.config = config;
        this.routerService = routerService;
    }
    openModal() {
        return this.child.openModal();
    }
    goBack() {
        this.routerService.navigateToPayments();
    }
}
BillpayPaymentRecurringEditFormViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayPaymentRecurringEditFormViewComponent, deps: [{ token: i1.PayBillsJourneyConfigService }, { token: i2.PayBillsNavigationService }], target: i0.ɵɵFactoryTarget.Component });
BillpayPaymentRecurringEditFormViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: BillpayPaymentRecurringEditFormViewComponent, selector: "ng-component", viewQueries: [{ propertyName: "child", first: true, predicate: ["paymentRecurringEditForm"], descendants: true }], ngImport: i0, template: `
    <div class="bb-block">
      <button bbButton color="link" class="bb-block bb-block--md pl-0" (click)="goBack()">
        <i aria-hidden="true" class="bb-icon bb-icon-arrow-back"></i>
        <span i18n="button label|Back button@@pay-bills-journey.payments-list.back">Back to payments list</span>
      </button>
    </div>
    <bb-billpay-payment-recurring-edit-form-container
      [accountNumberMasked]="config.accountNumberMasked"
      [paymentDetailsTitle]="config.paymentDetailsTitle"
      [frequencies]="config.frequencies"
      #paymentRecurringEditForm
    ></bb-billpay-payment-recurring-edit-form-container>
  `, isInline: true, components: [{ type: i3.BillpayPaymentRecurringEditFormContainerComponent, selector: "bb-billpay-payment-recurring-edit-form-container", inputs: ["frequencies"] }], directives: [{ type: i4.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayPaymentRecurringEditFormViewComponent, decorators: [{
            type: Component,
            args: [{
                    template: `
    <div class="bb-block">
      <button bbButton color="link" class="bb-block bb-block--md pl-0" (click)="goBack()">
        <i aria-hidden="true" class="bb-icon bb-icon-arrow-back"></i>
        <span i18n="button label|Back button@@pay-bills-journey.payments-list.back">Back to payments list</span>
      </button>
    </div>
    <bb-billpay-payment-recurring-edit-form-container
      [accountNumberMasked]="config.accountNumberMasked"
      [paymentDetailsTitle]="config.paymentDetailsTitle"
      [frequencies]="config.frequencies"
      #paymentRecurringEditForm
    ></bb-billpay-payment-recurring-edit-form-container>
  `,
                }]
        }], ctorParameters: function () { return [{ type: i1.PayBillsJourneyConfigService }, { type: i2.PayBillsNavigationService }]; }, propDecorators: { child: [{
                type: ViewChild,
                args: ['paymentRecurringEditForm']
            }] } });
