// @backbase/pay-bills-journey-ang/esm2015/src/views/billpay-payment-form-view.component.js
import { Component, ViewChild } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../services/pay-bills-journey-config.service";
import * as i2 from "../services/pay-bills-navigation.service";
import * as i3 from "@backbase/ui-ang/header";
import * as i4 from "../components/billpay-payment-single-bill/billpay-payment-form-container/billpay-payment-form-container.component";
import * as i5 from "@backbase/ui-ang/button";
export class BillpayPaymentFormViewComponent {
    constructor(config, routerService) {
        this.config = config;
        this.routerService = routerService;
    }
    openModal() {
        return this.child.openModal();
    }
    goBack() {
        this.routerService.navigateToPayeeList();
    }
}
BillpayPaymentFormViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayPaymentFormViewComponent, deps: [{ token: i1.PayBillsJourneyConfigService }, { token: i2.PayBillsNavigationService }], target: i0.ɵɵFactoryTarget.Component });
BillpayPaymentFormViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: BillpayPaymentFormViewComponent, selector: "ng-component", viewQueries: [{ propertyName: "child", first: true, predicate: ["paymentFormContainer"], descendants: true }], ngImport: i0, template: `
    <div class="bb-stack">
      <div class="bb-block bb-block--sm bb-stack__item bb-stack__item--fill">
        <button bbButton color="link" class="bb-block bb-block--md pl-0" (click)="goBack()">
          <i aria-hidden="true" class="bb-icon bb-icon-arrow-back"></i>
          <span i18n="button label|Back button@@pay-bills-journey.payee-list.back">Back to Payee List</span>
        </button>
        <bb-header-ui
          headingType="h1"
          heading="Make a payment"
          i18n-heading="@@pay-bills-journey.title.make-a-payment"
          data-role="pay-a-bill-journey-title"
        >
        </bb-header-ui>
        <p i18n="@@pay-bills-journey.pay-payee.header-description" data-role="multiple-bills-widget-title-info">
          Add payees if you would like to pay a single bill or multiple bills
        </p>
      </div>
    </div>
    <bb-billpay-payment-form-container
      [accountNumberMasked]="config.accountNumberMasked"
      [paymentDetailsTitle]="config.paymentDetailsTitle"
      [paymentDefaultCurrency]="config.paymentDefaultCurrency"
      [frequencies]="config.frequencies"
      #paymentFormContainer
    ></bb-billpay-payment-form-container>
  `, isInline: true, components: [{ type: i3.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i4.BillpayPaymentFormContainerComponent, selector: "bb-billpay-payment-form-container", inputs: ["frequencies", "accountNumberMasked", "paymentDetailsTitle", "paymentDefaultCurrency"] }], directives: [{ type: i5.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayPaymentFormViewComponent, decorators: [{
            type: Component,
            args: [{
                    template: `
    <div class="bb-stack">
      <div class="bb-block bb-block--sm bb-stack__item bb-stack__item--fill">
        <button bbButton color="link" class="bb-block bb-block--md pl-0" (click)="goBack()">
          <i aria-hidden="true" class="bb-icon bb-icon-arrow-back"></i>
          <span i18n="button label|Back button@@pay-bills-journey.payee-list.back">Back to Payee List</span>
        </button>
        <bb-header-ui
          headingType="h1"
          heading="Make a payment"
          i18n-heading="@@pay-bills-journey.title.make-a-payment"
          data-role="pay-a-bill-journey-title"
        >
        </bb-header-ui>
        <p i18n="@@pay-bills-journey.pay-payee.header-description" data-role="multiple-bills-widget-title-info">
          Add payees if you would like to pay a single bill or multiple bills
        </p>
      </div>
    </div>
    <bb-billpay-payment-form-container
      [accountNumberMasked]="config.accountNumberMasked"
      [paymentDetailsTitle]="config.paymentDetailsTitle"
      [paymentDefaultCurrency]="config.paymentDefaultCurrency"
      [frequencies]="config.frequencies"
      #paymentFormContainer
    ></bb-billpay-payment-form-container>
  `,
                }]
        }], ctorParameters: function () { return [{ type: i1.PayBillsJourneyConfigService }, { type: i2.PayBillsNavigationService }]; }, propDecorators: { child: [{
                type: ViewChild,
                args: ['paymentFormContainer']
            }] } });
