// @backbase/pay-bills-journey-ang/esm2015/src/views/billpay-enrolment-view.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../services/pay-bills-journey-config.service";
import * as i2 from "@backbase/ui-ang/header";
import * as i3 from "@backbase/billpay-journeys-common";
export class BillPayEnrolmentViewComponent {
    constructor(config) {
        this.config = config;
    }
}
BillPayEnrolmentViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillPayEnrolmentViewComponent, deps: [{ token: i1.PayBillsJourneyConfigService }], target: i0.ɵɵFactoryTarget.Component });
BillPayEnrolmentViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: BillPayEnrolmentViewComponent, selector: "bb-billpay-enrolment-view", ngImport: i0, template: `
    <div class="bb-block bb-block--lg">
      <bb-header-ui
        headingType="h1"
        heading="Enrol in bill pay"
        i18n-heading="@@billpay-manage-payee-journey.enrolment-heading"
      ></bb-header-ui>
    </div>
    <bb-billpay-enrolment [notificationDismissTime]="config?.notificationDismissTime"></bb-billpay-enrolment>
  `, isInline: true, components: [{ type: i2.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i3.BillpayEnrolmentComponent, selector: "bb-billpay-enrolment", inputs: ["notificationDismissTime"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillPayEnrolmentViewComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-billpay-enrolment-view',
                    template: `
    <div class="bb-block bb-block--lg">
      <bb-header-ui
        headingType="h1"
        heading="Enrol in bill pay"
        i18n-heading="@@billpay-manage-payee-journey.enrolment-heading"
      ></bb-header-ui>
    </div>
    <bb-billpay-enrolment [notificationDismissTime]="config?.notificationDismissTime"></bb-billpay-enrolment>
  `,
                }]
        }], ctorParameters: function () { return [{ type: i1.PayBillsJourneyConfigService }]; } });
