// @backbase/pay-bills-journey-ang/esm2015/src/views/billpay-payment-multiple-bills-review-view.component.js
import { Component, ViewChild } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/header";
import * as i2 from "../components/billpay-payment-multiple-bills/billpay-payment-multiple-ebills-review/billpay-payment-multiple-bills-review.component";
import * as i3 from "@backbase/ui-ang/button";
export class BillpayPaymentMultipleBillsReviewViewComponent {
    openModal() {
        return this.child.openModal();
    }
    navigateToForm() {
        this.child.onCancel();
    }
}
BillpayPaymentMultipleBillsReviewViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayPaymentMultipleBillsReviewViewComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
BillpayPaymentMultipleBillsReviewViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: BillpayPaymentMultipleBillsReviewViewComponent, selector: "ng-component", viewQueries: [{ propertyName: "child", first: true, predicate: ["multipleBillsReview"], descendants: true }], ngImport: i0, template: `
    <div class="bb-block bb-block--lg bb-stack">
      <div class="bb-stack__item bb-stack__item--fill">
        <button
          bbButton
          color="link"
          class="bb-block bb-block--md pl-0"
          data-role="pay-a-bill-journey-back"
          (click)="navigateToForm()"
        >
          <i aria-hidden="true" class="bb-icon bb-icon-arrow-back"></i>
          <span i18n="button label|Back button@@pay-bills-journey.back-to-multiple-bills">Back To Pay Bills</span>
        </button>
        <bb-header-ui
          headingType="h1"
          heading="Review Your Payments"
          i18n-heading="@@pay-bills-journey.title.review-bills"
          data-role="pay-a-bill-journey-title"
        >
        </bb-header-ui>
      </div>
    </div>
    <bb-billpay-payment-multiple-bills-review #multipleBillsReview></bb-billpay-payment-multiple-bills-review>
  `, isInline: true, components: [{ type: i1.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i2.BillpayPaymentMultipleBillsReviewComponent, selector: "bb-billpay-payment-multiple-bills-review" }], directives: [{ type: i3.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayPaymentMultipleBillsReviewViewComponent, decorators: [{
            type: Component,
            args: [{
                    template: `
    <div class="bb-block bb-block--lg bb-stack">
      <div class="bb-stack__item bb-stack__item--fill">
        <button
          bbButton
          color="link"
          class="bb-block bb-block--md pl-0"
          data-role="pay-a-bill-journey-back"
          (click)="navigateToForm()"
        >
          <i aria-hidden="true" class="bb-icon bb-icon-arrow-back"></i>
          <span i18n="button label|Back button@@pay-bills-journey.back-to-multiple-bills">Back To Pay Bills</span>
        </button>
        <bb-header-ui
          headingType="h1"
          heading="Review Your Payments"
          i18n-heading="@@pay-bills-journey.title.review-bills"
          data-role="pay-a-bill-journey-title"
        >
        </bb-header-ui>
      </div>
    </div>
    <bb-billpay-payment-multiple-bills-review #multipleBillsReview></bb-billpay-payment-multiple-bills-review>
  `,
                }]
        }], propDecorators: { child: [{
                type: ViewChild,
                args: ['multipleBillsReview']
            }] } });
