// @backbase/ui-ang/esm2015/_example/input-range/showcase/input-range-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { InputRangeShowcaseComponent } from './input-range-showcase.component';
import { InputRangeModule } from '@backbase/ui-ang/input-range';
import { InputNumberModule } from '@backbase/ui-ang/input-number';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { ButtonModule } from '@backbase/ui-ang/button';
import * as i0 from "@angular/core";
export class InputRangeShowcaseModule {
}
InputRangeShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputRangeShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputRangeShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputRangeShowcaseModule, declarations: [InputRangeShowcaseComponent], imports: [CommonModule,
        InputRangeModule,
        InputNumberModule,
        ReactiveFormsModule,
        InputValidationMessageModule,
        ButtonModule,
        FormsModule] });
InputRangeShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputRangeShowcaseModule, imports: [[
            CommonModule,
            InputRangeModule,
            InputNumberModule,
            ReactiveFormsModule,
            InputValidationMessageModule,
            ButtonModule,
            FormsModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputRangeShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        InputRangeModule,
                        InputNumberModule,
                        ReactiveFormsModule,
                        InputValidationMessageModule,
                        ButtonModule,
                        FormsModule,
                    ],
                    declarations: [InputRangeShowcaseComponent],
                    entryComponents: [InputRangeShowcaseComponent],
                }]
        }] });
