// @backbase/user-context-menu-widget-ang/esm2015/lib/user-context-menu-widget/user-context-menu-widget-properties.service.js
import { Injectable, Optional } from '@angular/core';
import { ItemModel } from '@backbase/foundation-ang/core';
import { BehaviorSubject, combineLatest, of } from 'rxjs';
import { map } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
export const WidgetProperty = {
    PAGE_SIZE: 'pageSize',
    INCLUDE_CURRENT_CONTEXT: 'includeCurrentContext',
    SHOW_SEARCH_AFTER: 'showSearchAfter',
    SHOW_SINGLE_CONTEXT: 'showSingleContext',
    REDIRECT_PAGE: 'selectContextRedirectPage',
};
export const DefaultWidgetProperties = {
    pageSize: 7,
    includeCurrentContext: true,
    showSearchAfter: 7,
    selectContextRedirectPage: '',
    showSingleContext: true,
};
export class UserContextMenuWidgetPropertiesService {
    constructor(itemModel) {
        this.itemModel = itemModel;
        this.pageSizeSubject$ = new BehaviorSubject(undefined);
        this.includeCurrentContextSubject$ = new BehaviorSubject(undefined);
        this.showSearchAfterSubject$ = new BehaviorSubject(undefined);
        this.selectContextRedirectPageSubject$ = new BehaviorSubject(undefined);
        this.showSingleContextSubject$ = new BehaviorSubject(undefined);
        this.pageSize = this.itemModel
            ? this.itemModel
                .property(WidgetProperty.PAGE_SIZE, '7')
                .pipe(map(propertyValue => parseInt(propertyValue, 10)))
            : of(DefaultWidgetProperties.pageSize);
        this.includeCurrentContext = this.itemModel
            ? this.itemModel
                .property(WidgetProperty.INCLUDE_CURRENT_CONTEXT, 'true')
                .pipe(map(propertyValue => `${propertyValue}` === 'true'))
            : of(DefaultWidgetProperties.includeCurrentContext);
        this.showSearchAfter = this.itemModel
            ? this.itemModel
                .property(WidgetProperty.SHOW_SEARCH_AFTER, '7')
                .pipe(map(propertyValue => parseInt(propertyValue, 10)))
            : of(DefaultWidgetProperties.showSearchAfter);
        this.selectContextRedirectPage = this.itemModel
            ? this.itemModel.property(WidgetProperty.REDIRECT_PAGE, '')
            : of(DefaultWidgetProperties.selectContextRedirectPage);
        this.showSingleContext = this.itemModel
            ? this.itemModel
                .property(WidgetProperty.SHOW_SINGLE_CONTEXT, 'true')
                .pipe(map(propertyValue => `${propertyValue}` === 'true'))
            : of(DefaultWidgetProperties.showSingleContext);
        this.pageSize$ = this.getProperty(this.pageSizeSubject$, this.pageSize).pipe(map(Number));
        this.includeCurrentContext$ = this.getProperty(this.includeCurrentContextSubject$, this.includeCurrentContext).pipe(map(this.toBoolean));
        this.showSearchAfter$ = this.getProperty(this.showSearchAfterSubject$, this.showSearchAfter).pipe(map(Number));
        this.selectContextRedirectPage$ = this.getProperty(this.selectContextRedirectPageSubject$, this.selectContextRedirectPage);
        this.showSingleContext$ = this.getProperty(this.showSingleContextSubject$, this.showSingleContext);
        this.properties$ = combineLatest([
            this.pageSize$,
            this.includeCurrentContext$,
            this.showSearchAfter$,
            this.selectContextRedirectPage$,
            this.showSingleContext$,
        ]).pipe(map(([pageSize, includeCurrentContext, showSearchAfter, selectContextRedirectPage, showSingleContext]) => ({
            pageSize,
            includeCurrentContext,
            showSearchAfter,
            selectContextRedirectPage,
            showSingleContext,
        })));
    }
    getProperty(input$, itemModel) {
        return combineLatest([input$, itemModel]).pipe(map(([value, valueIM]) => (value === undefined ? valueIM : value)));
    }
    toBoolean(value) {
        return value === 'true' || value === true;
    }
    setPageSize(value) {
        this.pageSizeSubject$.next(value);
    }
    setIncludeCurrentContext(value) {
        this.includeCurrentContextSubject$.next(value);
    }
    setShowSearchAfter(value) {
        this.showSearchAfterSubject$.next(value);
    }
    setSelectContextRedirectPage(value) {
        this.selectContextRedirectPageSubject$.next(value);
    }
    setShowSingleContext(value) {
        this.showSingleContextSubject$.next(value);
    }
}
/** @nocollapse */ UserContextMenuWidgetPropertiesService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: UserContextMenuWidgetPropertiesService, deps: [{ token: i1.ItemModel, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
/** @nocollapse */ UserContextMenuWidgetPropertiesService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: UserContextMenuWidgetPropertiesService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: UserContextMenuWidgetPropertiesService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.ItemModel, decorators: [{
                    type: Optional
                }] }]; } });
