// @backbase/payment-orders-ang/esm2015/lib/ui/components/currency-amount-selector/dom-attributes.service.js
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * Utility service for DOM manipulations
 */
export class DomAttributesService {
    constructor() {
        this.idPrefix = 'bb_element_';
    }
    generateId() {
        return `${this.idPrefix}${DomAttributesService.nextId++}`;
    }
}
DomAttributesService.nextId = 0;
/** @nocollapse */ DomAttributesService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DomAttributesService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
/** @nocollapse */ DomAttributesService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DomAttributesService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DomAttributesService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }] });
