// @backbase/ui-ang/esm2015/input-datepicker/validation/max-date-validator/max-date-validator.directive.js
import { Directive, forwardRef, Input } from '@angular/core';
import { NG_VALIDATORS } from '@angular/forms';
import * as i0 from "@angular/core";
export class MaxDateValidatorDirective {
    constructor() {
        this._maxDate = Number.MAX_VALUE;
        this.onValidatorChange = () => { };
    }
    set maxDate(date) {
        if (!date) {
            this._maxDate = Number.MAX_VALUE;
        }
        else {
            const maxDate = typeof date === 'string' ? new Date(date) : new Date(date.year, date.month - 1, date.day);
            this._maxDate = maxDate.getTime();
        }
        this.onValidatorChange();
    }
    registerOnValidatorChange(fn) {
        this.onValidatorChange = fn;
    }
    validate(control) {
        const value = control.value && control.value.to ? control.value.to : control.value;
        if (!value) {
            return null;
        }
        return new Date(value).getTime() <= this._maxDate
            ? // eslint-disable-next-line no-null/no-null
                null
            : { dateGreaterThanMaxDate: true };
    }
    ngOnDestroy() {
        this.maxDate = undefined;
    }
}
MaxDateValidatorDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MaxDateValidatorDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
MaxDateValidatorDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: MaxDateValidatorDirective, selector: "bb-input-datepicker-ui[maxDate]", inputs: { maxDate: "maxDate" }, providers: [
        {
            provide: NG_VALIDATORS,
            multi: true,
            useExisting: forwardRef(() => MaxDateValidatorDirective),
        },
    ], ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MaxDateValidatorDirective, decorators: [{
            type: Directive,
            args: [{
                    // eslint-disable-next-line @angular-eslint/directive-selector
                    selector: 'bb-input-datepicker-ui[maxDate]',
                    providers: [
                        {
                            provide: NG_VALIDATORS,
                            multi: true,
                            useExisting: forwardRef(() => MaxDateValidatorDirective),
                        },
                    ],
                }]
        }], ctorParameters: function () { return []; }, propDecorators: { maxDate: [{
                type: Input
            }] } });
