// @backbase/consent-journey-ang/esm2015/src/services/consent-journey-config.service.js
import { Injectable, Optional, Inject, InjectionToken } from '@angular/core';
import * as i0 from "@angular/core";
export const ConsentJourneyConfigurationToken = new InjectionToken('ConsentJourneyConfigurationToken injection token');
export const consentDefaultConfig = {
    multiAccountSelection: false,
};
export class ConsentJourneyConfigurationService {
    constructor(config) {
        this.config = config;
        this.config = this.config || {};
        this._config = Object.assign(Object.assign({}, consentDefaultConfig), config);
    }
    get multiAccountSelection() {
        return this._config.multiAccountSelection;
    }
}
ConsentJourneyConfigurationService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentJourneyConfigurationService, deps: [{ token: ConsentJourneyConfigurationToken, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
ConsentJourneyConfigurationService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentJourneyConfigurationService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentJourneyConfigurationService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [ConsentJourneyConfigurationToken]
                }] }]; } });
