// @backbase/stop-checks-journey-ang/esm2015/lib/components/stop-checks-list/stop-checks-list.component.js
import { Component } from '@angular/core';
import '@angular/localize/init';
import { StopChecksJourneyService } from '../../services/stop-checks-journey-config.service';
import * as i0 from "@angular/core";
import * as i1 from "../../services/stop-checks-journey-config.service";
import * as i2 from "@backbase/payord-stop-checks-list-widget-ang";
import * as i3 from "@angular/router";
export class StopChecksListComponent {
    constructor(config) {
        this.config = config;
    }
}
/** @nocollapse */ StopChecksListComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StopChecksListComponent, deps: [{ token: i1.StopChecksJourneyService }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ StopChecksListComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: StopChecksListComponent, selector: "bb-stop-checks-list-wrapper", ngImport: i0, template: "<bb-payord-stop-checks-list-widget-ang\n  [paginationType]=\"config.paginationType\"\n  [pageSize]=\"config.pageSize\"\n  [maxNavPages]=\"config.maxNavPages\"\n>\n</bb-payord-stop-checks-list-widget-ang>\n<router-outlet></router-outlet>\n", components: [{ type: i2.PayordStopChecksListWidgetAngComponent, selector: "bb-payord-stop-checks-list-widget-ang", inputs: ["pageSize", "maxNavPages", "paginationType"] }], directives: [{ type: i3.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StopChecksListComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-stop-checks-list-wrapper',
                    templateUrl: './stop-checks-list.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.StopChecksJourneyService }]; } });
