// @backbase/user-common/esm2015/src/components/manage-profile/manage-profile-form/manage-profile-postal-address-form.component.js
import { Component } from '@angular/core';
import { Validators } from '@angular/forms';
import { ManageProfileFormComponent } from './manage-profile-form.component';
import * as i0 from "@angular/core";
import * as i1 from "@angular/forms";
export class ManageProfilePostalAddressFormComponent extends ManageProfileFormComponent {
    constructor(formBuilder) {
        super();
        this.formBuilder = formBuilder;
    }
    markFormAsTouched() {
        this.form.controls.type.markAsTouched();
        this.form.controls.primary.markAsTouched();
        this.form.controls.buildingNumber.markAsTouched();
        this.form.controls.streetName.markAsTouched();
        this.form.controls.townName.markAsTouched();
        this.form.controls.postalCode.markAsTouched();
        this.form.controls.countrySubDivision.markAsTouched();
    }
    createForm() {
        const { key, type, primary, buildingNumber, streetName, townName, postalCode, countrySubDivision } = this
            .data;
        this.formData = this.formBuilder.group({
            key,
            type: [type, Validators.required],
            primary,
            buildingNumber: [buildingNumber, Validators.required],
            streetName,
            townName: [townName, Validators.required],
            postalCode: [postalCode, Validators.required],
            countrySubDivision: [countrySubDivision, Validators.required],
        });
    }
}
ManageProfilePostalAddressFormComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageProfilePostalAddressFormComponent, deps: [{ token: i1.FormBuilder }], target: i0.ɵɵFactoryTarget.Component });
ManageProfilePostalAddressFormComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: ManageProfilePostalAddressFormComponent, selector: "ng-component", usesInheritance: true, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageProfilePostalAddressFormComponent, decorators: [{
            type: Component,
            args: [{
                    template: '',
                }]
        }], ctorParameters: function () { return [{ type: i1.FormBuilder }]; } });
