// @backbase/user-common/esm2015/src/components/manage-profile/manage-profile-form/manage-profile-email-address-form.component.js
import { Component } from '@angular/core';
import { Validators } from '@angular/forms';
import { ManageProfileFormComponent } from './manage-profile-form.component';
import * as i0 from "@angular/core";
import * as i1 from "@angular/forms";
export class ManageProfileEmailAddressFormComponent extends ManageProfileFormComponent {
    constructor(formBuilder) {
        super();
        this.formBuilder = formBuilder;
    }
    markFormAsTouched() {
        this.form.controls.type.markAsTouched();
        this.form.controls.address.markAsTouched();
        this.form.controls.primary.markAsTouched();
    }
    createForm() {
        const { key, type, address, primary } = this.data;
        this.formData = this.formBuilder.group({
            key,
            type: [type, Validators.required],
            address: [address, Validators.required],
            primary,
        });
    }
}
ManageProfileEmailAddressFormComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageProfileEmailAddressFormComponent, deps: [{ token: i1.FormBuilder }], target: i0.ɵɵFactoryTarget.Component });
ManageProfileEmailAddressFormComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: ManageProfileEmailAddressFormComponent, selector: "ng-component", usesInheritance: true, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageProfileEmailAddressFormComponent, decorators: [{
            type: Component,
            args: [{
                    template: '',
                }]
        }], ctorParameters: function () { return [{ type: i1.FormBuilder }]; } });
