// @backbase/user-common/esm2015/src/components/manage-profile/manage-profile-form/manage-profile-form.component.js
import { Component, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
export class ManageProfileFormComponent {
    constructor() {
        /**
         * Whether the form is being submitted.
         * Defaults to false.
         */
        this.isSubmitting = false;
        this.cancel = new EventEmitter();
        this.delete = new EventEmitter();
        this.save = new EventEmitter();
        this.showFormErrors = false;
        this.showDeleteButton = false;
        this.isCancelModalOpen = false;
        this.isDeleteModalOpen = false;
        this.hasBeenSubmitted = false;
    }
    ngOnInit() {
        if (this.data.key) {
            this.showDeleteButton = !!(this.count && this.count > 1);
        }
        this.createForm();
    }
    get form() {
        if (!this.formData) {
            throw new Error('formData is not defined');
        }
        return this.formData;
    }
    handleSubmit() {
        this.hasBeenSubmitted = true;
        this.markFormAsTouched();
        this.showErrorsOnForm();
        if (!this.form.valid) {
            return;
        }
        this.form.updateValueAndValidity();
        this.save.emit(this.form.value);
    }
    handleCancelClick() {
        if (this.form.touched) {
            this.isCancelModalOpen = true;
        }
        else {
            this.cancel.emit();
        }
    }
    closeCancelModal() {
        this.isCancelModalOpen = false;
        this.cancel.emit();
    }
    dismissCancelModal() {
        this.isCancelModalOpen = false;
        this.markFormAsTouched();
        this.showErrorsOnForm();
    }
    handleDelete() {
        this.isDeleteModalOpen = true;
    }
    closeDeleteModal() {
        this.isDeleteModalOpen = false;
        this.delete.emit(this.form.value);
    }
    dismissDeleteModal() {
        this.isDeleteModalOpen = false;
        this.markFormAsTouched();
        this.showErrorsOnForm();
    }
    showErrorsOnForm() {
        this.showFormErrors = !this.form.valid;
    }
    markFormAsTouched() { }
    createForm() { }
}
ManageProfileFormComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageProfileFormComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ManageProfileFormComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: ManageProfileFormComponent, selector: "ng-component", inputs: { types: "types", data: "data", count: "count", isSubmitting: "isSubmitting" }, outputs: { cancel: "cancel", delete: "delete", save: "save" }, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageProfileFormComponent, decorators: [{
            type: Component,
            args: [{
                    template: '',
                }]
        }], propDecorators: { types: [{
                type: Input
            }], data: [{
                type: Input
            }], count: [{
                type: Input
            }], isSubmitting: [{
                type: Input
            }], cancel: [{
                type: Output
            }], delete: [{
                type: Output
            }], save: [{
                type: Output
            }] } });
