// @backbase/product-summary-common-ang/esm2015/src/pipes/format/format.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AccountNumberPipeModule } from '@backbase/ui-ang/account-number-pipe';
import { PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import { AbsoluteValuePipe } from './absolute-value.pipe';
import * as i0 from "@angular/core";
const sharedPipes = [AccountNumberPipeModule, PaymentCardNumberModule];
const pipes = [AbsoluteValuePipe];
export class FormatPipeModule {
}
FormatPipeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: FormatPipeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
FormatPipeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: FormatPipeModule, declarations: [AbsoluteValuePipe], imports: [CommonModule, AccountNumberPipeModule, PaymentCardNumberModule], exports: [AccountNumberPipeModule, PaymentCardNumberModule, AbsoluteValuePipe] });
FormatPipeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: FormatPipeModule, imports: [[CommonModule, ...sharedPipes], AccountNumberPipeModule, PaymentCardNumberModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: FormatPipeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...sharedPipes],
                    declarations: [...pipes],
                    exports: [...sharedPipes, ...pipes],
                }]
        }] });
