// @backbase/authorized-users-journey-ang/esm2015/src/services/authorized-users-journey-config.service.js
import { Injectable, Optional, Inject, InjectionToken } from '@angular/core';
import { DEFAULT_UI_LEVELS_CONFIG, DEFAULT_UI_PERMISSIONS_CONFIG } from '../common/constants';
import { LevelsMappingConfigImplementation } from '../common/services/default-level-mapping-config.service';
import * as i0 from "@angular/core";
export const AuthorizedUsersJourneyConfigurationToken = new InjectionToken('AuthorizedUsersJourneyConfiguration injection token');
export var FullAccessGranularity;
(function (FullAccessGranularity) {
    FullAccessGranularity["PER_ACCOUNT"] = "individual-control";
    FullAccessGranularity["SAME_FOR_ALL_ACCOUNTS"] = "all-at-once";
})(FullAccessGranularity || (FullAccessGranularity = {}));
const configDefaults = {
    fullAccess: FullAccessGranularity.SAME_FOR_ALL_ACCOUNTS,
    levelsConfig: {
        mappings: new LevelsMappingConfigImplementation(),
        uiConfig: Object.assign(Object.assign({}, DEFAULT_UI_LEVELS_CONFIG), DEFAULT_UI_PERMISSIONS_CONFIG),
    },
};
export class AuthorizedUsersJourneyConfigService {
    constructor(config) {
        config = config || {};
        this._config = Object.assign(Object.assign({}, configDefaults), config);
    }
    get defaults() {
        return configDefaults;
    }
    /**
     * Full access permission. Defines how the account owner can set access permissions: for all accounts at once, or for each account individually.
     */
    get fullAccess() {
        return this._config.fullAccess;
    }
    /**
     * Custom UI configuration for the permission levels and their permission sets.
     * For more details check https://community.backbase.com/documentation/Retail-Banking/latest/authorized_user_create_reference#create_authorized_users_configuration
     */
    get levelsConfig() {
        return this._config.levelsConfig;
    }
}
AuthorizedUsersJourneyConfigService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersJourneyConfigService, deps: [{ token: AuthorizedUsersJourneyConfigurationToken, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
AuthorizedUsersJourneyConfigService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersJourneyConfigService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersJourneyConfigService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [AuthorizedUsersJourneyConfigurationToken]
                }] }]; } });
