// @backbase/messages-client-inbox-journey-ang/esm2015/src/components/messages-client-inbox-list/messages-client-inbox-list.component.js
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "../../messages-client-inbox-journey-config.service";
import * as i2 from "@angular/router";
import * as i3 from "@backbase/messages-conversations-list-widget-ang";
import * as i4 from "@angular/common";
export class MessagesClientInboxListComponent {
    constructor(config, route, router) {
        var _a;
        this.config = config;
        this.route = route;
        this.router = router;
        this.mailboxType$$ = new BehaviorSubject(undefined);
        /**
         * Mailbox type title
         */
        this.mailboxType$ = this.mailboxType$$.asObservable();
        /**
         * List of shown tabs
         */
        this.tabs = [];
        const routeConfig = this.route.snapshot.routeConfig;
        this.mailboxType$$.next((_a = routeConfig === null || routeConfig === void 0 ? void 0 : routeConfig.data) === null || _a === void 0 ? void 0 : _a.mailboxType);
    }
    /**
     * Open conversation thread
     */
    openConversation(id) {
        this.router.navigate(['../conversation', { id }], { relativeTo: this.route });
    }
}
MessagesClientInboxListComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesClientInboxListComponent, deps: [{ token: i1.MessagesClientInboxJourneyConfigService }, { token: i2.ActivatedRoute }, { token: i2.Router }], target: i0.ɵɵFactoryTarget.Component });
MessagesClientInboxListComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.14", type: MessagesClientInboxListComponent, selector: "bb-messages-client-inbox-list", ngImport: i0, template: "<bb-messages-conversations-list-widget-ang\n  [mailboxType]=\"mailboxType$ | async\"\n  (conversationId)=\"openConversation($event)\"\n  [itemsPerPage]=\"config.itemsPerPage\"\n  [customerServiceTitle]=\"config.customerServiceTitle\"\n  [createMessageClosedEventName]=\"config.createMessageClosedEventName\"\n>\n</bb-messages-conversations-list-widget-ang>\n", components: [{ type: i3.MessagesConversationsListWidgetComponent, selector: "bb-messages-conversations-list-widget-ang", inputs: ["mailboxType", "itemsPerPage", "createMessageClosedEventName", "customerServiceTitle"] }], pipes: { "async": i4.AsyncPipe }, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesClientInboxListComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-messages-client-inbox-list',
                    templateUrl: './messages-client-inbox-list.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.MessagesClientInboxJourneyConfigService }, { type: i2.ActivatedRoute }, { type: i2.Router }]; } });
