// @backbase/quick-transfer-journey-ang/esm2015/lib/utils/arrangement-mapper.js
/**
 * Method to extract ProductKind array from ProductsummaryGET response
 *
 * @param res
 */
export const responseToProductList = (res) => {
    return res === null || res === void 0 ? void 0 : res.map(record => {
        let balance;
        switch (record.productKindName) {
            case 'Term Deposit':
                balance = record.principalAmount;
                break;
            case 'Investment Account':
                balance = record.currentInvestmentValue;
                break;
            default:
                balance = record.bookedBalance;
                break;
        }
        return {
            uuid: record.id,
            balance: balance || 0,
            name: record.displayName,
            currency: record.currency,
            number: record.IBAN || record.BBAN || record.BIC || record.number,
            externalTransferAllowed: record.externalTransferAllowed,
            productKindName: record.productKindName,
            defaultRecord: record,
        };
    });
};
