// @backbase/ui-ang/esm2015/input-email/input-email.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { IconModule } from '@backbase/ui-ang/icon';
import { FormsModule } from '@angular/forms';
import { BaseClassesModule } from '@backbase/ui-ang/base-classes';
import { InputEmailComponent } from './input-email.component';
import { FocusModule } from '@backbase/ui-ang/focus';
import * as i0 from "@angular/core";
export class InputEmailModule {
}
InputEmailModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputEmailModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputEmailModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputEmailModule, declarations: [InputEmailComponent], imports: [CommonModule, IconModule, FormsModule, BaseClassesModule, FocusModule], exports: [InputEmailComponent] });
InputEmailModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputEmailModule, imports: [[CommonModule, IconModule, FormsModule, BaseClassesModule, FocusModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputEmailModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, IconModule, FormsModule, BaseClassesModule, FocusModule],
                    declarations: [InputEmailComponent],
                    exports: [InputEmailComponent],
                }]
        }] });
