// @backbase/device-information-widget-ang/esm2015/src/services/admin-device-information.service.js
import { Injectable } from '@angular/core';
import { map } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/device";
export class AdminDeviceInformationService {
    constructor(deviceManagementService) {
        this.deviceManagementService = deviceManagementService;
        this.dbsUserId = '';
    }
    getDevices() {
        return this.deviceManagementService
            .getDevicesByDbsUserId({ dbsUserId: this.dbsUserId })
            .pipe(map(response => response.devices));
    }
    updateDevice(deviceId, patchDeviceRequest) {
        return this.deviceManagementService.patchUserDeviceByDbsUserIdAndDeviceId({
            dbsUserId: this.dbsUserId,
            deviceId,
            patchDeviceRequest,
        });
    }
    removeDevice(deviceId) {
        return this.deviceManagementService.deleteUserDeviceByDeviceId({
            dbsUserId: this.dbsUserId,
            deviceId,
        });
    }
    set dbsUserIdInput(userId) {
        this.dbsUserId = userId;
    }
    get dbsUserIdInput() {
        return this.dbsUserId;
    }
}
AdminDeviceInformationService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AdminDeviceInformationService, deps: [{ token: i1.ManageOtherUsersDevicesService }], target: i0.ɵɵFactoryTarget.Injectable });
AdminDeviceInformationService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AdminDeviceInformationService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AdminDeviceInformationService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.ManageOtherUsersDevicesService }]; } });
