// @backbase/device-information-widget-ang/esm2015/src/services/device-information-preference.service.js
import { Injectable, Optional } from '@angular/core';
import { of } from 'rxjs';
import { map } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const defaultProperties = {
    dismissTimeout: 3,
    isAdminWidget: false,
};
export class DeviceInformationPreferenceService {
    constructor(model) {
        this.model = model;
    }
    getTimeoutPreference() {
        return this.model
            ? this.model.property('dismissTimeout', defaultProperties.dismissTimeout)
            : of(defaultProperties.dismissTimeout);
    }
    getAdminPreference() {
        return this.model
            ? this.model
                .property('isAdminWidget', JSON.stringify(defaultProperties.isAdminWidget))
                .pipe(map((value) => !!value && value !== 'false'))
            : of(defaultProperties.isAdminWidget);
    }
}
DeviceInformationPreferenceService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: DeviceInformationPreferenceService, deps: [{ token: i1.ItemModel, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
DeviceInformationPreferenceService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: DeviceInformationPreferenceService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: DeviceInformationPreferenceService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.ItemModel, decorators: [{
                    type: Optional
                }] }]; } });
