// @backbase/device-information-widget-ang/esm2015/src/services/user-device-information.service.js
import { Injectable } from '@angular/core';
import { map } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/device";
export class UserDeviceInformationService {
    constructor(deviceManagementService) {
        this.deviceManagementService = deviceManagementService;
    }
    getDevices() {
        return this.deviceManagementService.getMyDevices().pipe(map(response => response.devices));
    }
    updateDevice(deviceId, patchDeviceRequest) {
        return this.deviceManagementService.patchMyDeviceByDeviceId({
            deviceId,
            patchDeviceRequest,
        });
    }
    removeDevice(deviceId) {
        return this.deviceManagementService.deleteMyDeviceByDeviceId({ deviceId });
    }
}
UserDeviceInformationService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserDeviceInformationService, deps: [{ token: i1.ManageMyDevicesService }], target: i0.ɵɵFactoryTarget.Injectable });
UserDeviceInformationService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserDeviceInformationService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserDeviceInformationService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.ManageMyDevicesService }]; } });
