// @backbase/ui-ang/esm2015/rich-text-editor/rich-text-editor.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RichTextEditorActionsDirective, RichTextEditorComponent } from './rich-text-editor.component';
import { ButtonModule } from '@backbase/ui-ang/button';
import { IconModule } from '@backbase/ui-ang/icon';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { RichTextEditorBoldActionComponent } from './bb-rich-text-editor-actions/rich-text-editor-bold-action/rich-text-editor-bold-action.component';
import { RichTextEditorActionsService } from './bb-rich-text-editor-actions/rich-text-editor-actions.service';
import { RichTextEditorItalicActionComponent } from './bb-rich-text-editor-actions/rich-text-editor-italic-action/rich-text-editor-italic-action.component';
import { RichTextEditorUnderlineActionComponent } from './bb-rich-text-editor-actions/rich-text-editor-underline-action/rich-text-editor-underline-action.component';
import { RichTextEditorStrikethroughActionComponent } from './bb-rich-text-editor-actions/rich-text-editor-strikethrough-action/rich-text-editor-strikethrough-action.component';
import { RichTextEditorOlistActionComponent } from './bb-rich-text-editor-actions/rich-text-editor-olist-action/rich-text-editor-olist-action.component';
import { RichTextEditorUlistActionComponent } from './bb-rich-text-editor-actions/rich-text-editor-ulist-action/rich-text-editor-ulist-action.component';
import { RichTextEditorLinkActionComponent } from './bb-rich-text-editor-actions/rich-text-editor-link-action/rich-text-editor-link-action.component';
import { RichTextEditorParagraphActionComponent } from './bb-rich-text-editor-actions/rich-text-editor-paragraph-action/rich-text-editor-paragraph-action.component';
import { RichTextEditorCodeActionComponent } from './bb-rich-text-editor-actions/rich-text-editor-code-action/rich-text-editor-code-action.component';
import { RichTextEditorImageActionComponent } from './bb-rich-text-editor-actions/rich-text-editor-image-action/rich-text-editor-image-action.component';
import { RichTextEditorActionsComponent } from './bb-rich-text-editor-actions/rich-text-editor-actions.component';
import { ModalModule } from '@backbase/ui-ang/modal';
import { FormsModule } from '@angular/forms';
import * as i0 from "@angular/core";
export class RichTextEditorModule {
}
RichTextEditorModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RichTextEditorModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
RichTextEditorModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RichTextEditorModule, declarations: [RichTextEditorComponent,
        RichTextEditorActionsDirective,
        RichTextEditorActionsComponent,
        RichTextEditorBoldActionComponent,
        RichTextEditorCodeActionComponent,
        RichTextEditorImageActionComponent,
        RichTextEditorItalicActionComponent,
        RichTextEditorLinkActionComponent,
        RichTextEditorOlistActionComponent,
        RichTextEditorParagraphActionComponent,
        RichTextEditorStrikethroughActionComponent,
        RichTextEditorUlistActionComponent,
        RichTextEditorUnderlineActionComponent], imports: [CommonModule, ButtonModule, IconModule, ModalModule, FormsModule, TooltipModule], exports: [RichTextEditorComponent,
        RichTextEditorActionsDirective,
        RichTextEditorActionsComponent,
        RichTextEditorBoldActionComponent,
        RichTextEditorCodeActionComponent,
        RichTextEditorImageActionComponent,
        RichTextEditorItalicActionComponent,
        RichTextEditorLinkActionComponent,
        RichTextEditorOlistActionComponent,
        RichTextEditorParagraphActionComponent,
        RichTextEditorStrikethroughActionComponent,
        RichTextEditorUlistActionComponent,
        RichTextEditorUnderlineActionComponent] });
RichTextEditorModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RichTextEditorModule, providers: [RichTextEditorActionsService], imports: [[CommonModule, ButtonModule, IconModule, ModalModule, FormsModule, TooltipModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RichTextEditorModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ButtonModule, IconModule, ModalModule, FormsModule, TooltipModule],
                    declarations: [
                        RichTextEditorComponent,
                        RichTextEditorActionsDirective,
                        RichTextEditorActionsComponent,
                        RichTextEditorBoldActionComponent,
                        RichTextEditorCodeActionComponent,
                        RichTextEditorImageActionComponent,
                        RichTextEditorItalicActionComponent,
                        RichTextEditorLinkActionComponent,
                        RichTextEditorOlistActionComponent,
                        RichTextEditorParagraphActionComponent,
                        RichTextEditorStrikethroughActionComponent,
                        RichTextEditorUlistActionComponent,
                        RichTextEditorUnderlineActionComponent,
                    ],
                    exports: [
                        RichTextEditorComponent,
                        RichTextEditorActionsDirective,
                        RichTextEditorActionsComponent,
                        RichTextEditorBoldActionComponent,
                        RichTextEditorCodeActionComponent,
                        RichTextEditorImageActionComponent,
                        RichTextEditorItalicActionComponent,
                        RichTextEditorLinkActionComponent,
                        RichTextEditorOlistActionComponent,
                        RichTextEditorParagraphActionComponent,
                        RichTextEditorStrikethroughActionComponent,
                        RichTextEditorUlistActionComponent,
                        RichTextEditorUnderlineActionComponent,
                    ],
                    providers: [RichTextEditorActionsService],
                }]
        }] });
