// @backbase/enrollment-manager-v1-client-ang/esm2015/api/enrollment.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { ENROLLMENT_MANAGER_BASE_PATH } from '../variables';
import { EnrollmentManagerConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class EnrollmentService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '/client-api/v1';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new EnrollmentManagerConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    createEnrollment(requestParameters, observe = 'body', reportProgress = false, options) {
        const _enrollmentCreation = requestParameters["enrollmentCreation"];
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v1/enrollments`, _enrollmentCreation, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    createEnrollmentUrl(requestParameters) {
        const _enrollmentCreation = requestParameters["enrollmentCreation"];
        return `${this.configuration.basePath}/client-api/v1/enrollments`;
    }
    getChallengeForEnrollment(requestParameters, observe = 'body', reportProgress = false, options) {
        const _enrollmentId = requestParameters["enrollmentId"];
        if (_enrollmentId === null || _enrollmentId === undefined) {
            throw new Error('Required parameter enrollmentId was null or undefined when calling getChallengeForEnrollment.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/enrollments/${encodeURIComponent(String(_enrollmentId))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getChallengeForEnrollmentUrl(requestParameters) {
        const _enrollmentId = requestParameters["enrollmentId"];
        if (_enrollmentId === null || _enrollmentId === undefined) {
            throw new Error('Required parameter enrollmentId was null or undefined when calling getChallengeForEnrollment.');
        }
        return `${this.configuration.basePath}/client-api/v1/enrollments/${encodeURIComponent(String(_enrollmentId))}`;
    }
    getProvisioningStatusForEnrollment(requestParameters, observe = 'body', reportProgress = false, options) {
        const _enrollmentId = requestParameters["enrollmentId"];
        if (_enrollmentId === null || _enrollmentId === undefined) {
            throw new Error('Required parameter enrollmentId was null or undefined when calling getProvisioningStatusForEnrollment.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/enrollments/${encodeURIComponent(String(_enrollmentId))}/provisioning/status`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getProvisioningStatusForEnrollmentUrl(requestParameters) {
        const _enrollmentId = requestParameters["enrollmentId"];
        if (_enrollmentId === null || _enrollmentId === undefined) {
            throw new Error('Required parameter enrollmentId was null or undefined when calling getProvisioningStatusForEnrollment.');
        }
        return `${this.configuration.basePath}/client-api/v1/enrollments/${encodeURIComponent(String(_enrollmentId))}/provisioning/status`;
    }
    handleChallengeResponse(requestParameters, observe = 'body', reportProgress = false, options) {
        const _enrollmentId = requestParameters["enrollmentId"];
        if (_enrollmentId === null || _enrollmentId === undefined) {
            throw new Error('Required parameter enrollmentId was null or undefined when calling handleChallengeResponse.');
        }
        const _challengeResponse = requestParameters["challengeResponse"];
        if (_challengeResponse === null || _challengeResponse === undefined) {
            throw new Error('Required parameter challengeResponse was null or undefined when calling handleChallengeResponse.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v1/enrollments/${encodeURIComponent(String(_enrollmentId))}/response`, _challengeResponse, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    handleChallengeResponseUrl(requestParameters) {
        const _enrollmentId = requestParameters["enrollmentId"];
        if (_enrollmentId === null || _enrollmentId === undefined) {
            throw new Error('Required parameter enrollmentId was null or undefined when calling handleChallengeResponse.');
        }
        const _challengeResponse = requestParameters["challengeResponse"];
        if (_challengeResponse === null || _challengeResponse === undefined) {
            throw new Error('Required parameter challengeResponse was null or undefined when calling handleChallengeResponse.');
        }
        return `${this.configuration.basePath}/client-api/v1/enrollments/${encodeURIComponent(String(_enrollmentId))}/response`;
    }
}
EnrollmentService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: EnrollmentService, deps: [{ token: i1.HttpClient }, { token: ENROLLMENT_MANAGER_BASE_PATH, optional: true }, { token: i2.EnrollmentManagerConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
EnrollmentService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: EnrollmentService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: EnrollmentService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [ENROLLMENT_MANAGER_BASE_PATH]
                }] }, { type: i2.EnrollmentManagerConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
