// @backbase/enrollment-manager-v1-client-ang/esm2015/api/enrollment.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const EnrollmentServiceCreateEnrollmentMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/client-api/v1/enrollments",
        method: "POST",
        responses: [
            {
                status: 201,
                body: {
                    "challengeType": "INSTRUCTIONS",
                    "challenge": "2740f44c-c4ab-42a4-bbf1-1ad46b53b743",
                    "enrollmentId": "eb101331-f6cd-4d44-81c8-7b325619a362"
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const EnrollmentServiceGetChallengeForEnrollmentMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/client-api/v1/enrollments/{enrollmentId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "challengeType": "INSTRUCTIONS",
                    "challenge": "2740f44c-c4ab-42a4-bbf1-1ad46b53b743",
                    "enrollmentId": "eb101331-f6cd-4d44-81c8-7b325619a362"
                }
            },
            {
                status: 200,
                body: {
                    "enrollmentId": "enrollmentId",
                    "challenge": "challenge"
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const EnrollmentServiceGetProvisioningStatusForEnrollmentMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/client-api/v1/enrollments/{enrollmentId}/provisioning/status",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "provisioningStatus": "COMPLETE"
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const EnrollmentServiceHandleChallengeResponseMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/client-api/v1/enrollments/{enrollmentId}/response",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "challengeType": "INSTRUCTIONS",
                    "challenge": "2740f44c-c4ab-42a4-bbf1-1ad46b53b743",
                    "enrollmentId": "eb101331-f6cd-4d44-81c8-7b325619a362"
                }
            },
            {
                status: 200,
                body: {
                    "challengeType": "TWOFASELECTION",
                    "challenge": "2740f44c-c4ab-42a4-bbf1-1ad46b53b743",
                    "enrollmentId": "eb101331-f6cd-4d44-81c8-7b325619a362",
                    "challengePayload": {
                        "otpChannels": [{
                                "channelType": "EMAIL",
                                "channelValue": "xxxxxxxyz@backbase.com"
                            }, {
                                "channelType": "SMS",
                                "channelValue": "xxxxxx789"
                            }]
                    }
                }
            },
            {
                status: 200,
                body: {
                    "challengeType": "TWOFAENTRY",
                    "challenge": "2740f44c-c4ab-42a4-bbf1-1ad46b53b743",
                    "enrollmentId": "eb101331-f6cd-4d44-81c8-7b325619a362",
                    "challengePayload": {
                        "resendDelay": 60,
                        "resendsRemaining": 1,
                        "attemptsRemaining": 2,
                        "otpSize": 6,
                        "otpChannel": {
                            "channelType": "EMAIL",
                            "channelValue": "xxxxxxxyz@backbase.com"
                        }
                    }
                }
            },
            {
                status: 200,
                body: {
                    "challengeType": "PASSWORD",
                    "challenge": "2740f44c-c4ab-42a4-bbf1-1ad46b53b743",
                    "enrollmentId": "eb101331-f6cd-4d44-81c8-7b325619a362",
                    "challengePayload": {
                        "redirectUrl": "https://identity-url-here/auth/realms/backbase/protocol/openid-connect/auth?redirect_uri=http://completion-url-here&client_id=self-enrollment&response_type=code&scope=openid"
                    }
                }
            },
            {
                status: 200,
                body: {
                    "enrollmentId": "enrollmentId",
                    "challenge": "challenge"
                }
            },
            {
                status: 400,
                body: {
                    "challengeType": "INSTRUCTIONS",
                    "challenge": "2740f44c-c4ab-42a4-bbf1-1ad46b53b743",
                    "enrollmentId": "eb101331-f6cd-4d44-81c8-7b325619a362"
                }
            },
            {
                status: 400,
                body: {
                    "challengeType": "TWOFASELECTION",
                    "challenge": "2740f44c-c4ab-42a4-bbf1-1ad46b53b743",
                    "enrollmentId": "eb101331-f6cd-4d44-81c8-7b325619a362",
                    "challengePayload": {
                        "otpChannels": [{
                                "channelType": "EMAIL",
                                "channelValue": "xxxxxxxyz@backbase.com"
                            }, {
                                "channelType": "SMS",
                                "channelValue": "xxxxxx789"
                            }]
                    }
                }
            },
            {
                status: 400,
                body: {
                    "challengeType": "TWOFAENTRY",
                    "challenge": "2740f44c-c4ab-42a4-bbf1-1ad46b53b743",
                    "enrollmentId": "eb101331-f6cd-4d44-81c8-7b325619a362",
                    "challengePayload": {
                        "resendDelay": 60,
                        "resendsRemaining": 1,
                        "attemptsRemaining": 2,
                        "otpSize": 6,
                        "otpChannel": {
                            "channelType": "EMAIL",
                            "channelValue": "xxxxxxxyz@backbase.com"
                        }
                    }
                }
            },
            {
                status: 400,
                body: {
                    "enrollmentId": "enrollmentId",
                    "challenge": "challenge"
                }
            },
            {
                status: 409,
                body: {
                    "challengeType": "INSTRUCTIONS",
                    "challenge": "2740f44c-c4ab-42a4-bbf1-1ad46b53b743",
                    "enrollmentId": "eb101331-f6cd-4d44-81c8-7b325619a362"
                }
            },
            {
                status: 409,
                body: {
                    "challengeType": "TWOFASELECTION",
                    "challenge": "2740f44c-c4ab-42a4-bbf1-1ad46b53b743",
                    "enrollmentId": "eb101331-f6cd-4d44-81c8-7b325619a362",
                    "challengePayload": {
                        "otpChannels": [{
                                "channelType": "EMAIL",
                                "channelValue": "xxxxxxxyz@backbase.com"
                            }, {
                                "channelType": "SMS",
                                "channelValue": "xxxxxx789"
                            }]
                    }
                }
            },
            {
                status: 409,
                body: {
                    "challengeType": "TWOFAENTRY",
                    "challenge": "2740f44c-c4ab-42a4-bbf1-1ad46b53b743",
                    "enrollmentId": "eb101331-f6cd-4d44-81c8-7b325619a362",
                    "challengePayload": {
                        "resendDelay": 60,
                        "resendsRemaining": 1,
                        "attemptsRemaining": 2,
                        "otpSize": 6,
                        "otpChannel": {
                            "channelType": "EMAIL",
                            "channelValue": "xxxxxxxyz@backbase.com"
                        }
                    }
                }
            },
            {
                status: 409,
                body: {
                    "enrollmentId": "enrollmentId",
                    "challenge": "challenge"
                }
            },
            {
                status: 500,
                body: {
                    "enrollmentId": "enrollmentId",
                    "challenge": "challenge"
                }
            },
        ]
    }]);
export const EnrollmentServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v1/client-api/v1/enrollments",
        method: "POST",
        responses: [
            {
                status: 201,
                body: {
                    "challengeType": "INSTRUCTIONS",
                    "challenge": "2740f44c-c4ab-42a4-bbf1-1ad46b53b743",
                    "enrollmentId": "eb101331-f6cd-4d44-81c8-7b325619a362"
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/client-api/v1/enrollments/{enrollmentId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "challengeType": "INSTRUCTIONS",
                    "challenge": "2740f44c-c4ab-42a4-bbf1-1ad46b53b743",
                    "enrollmentId": "eb101331-f6cd-4d44-81c8-7b325619a362"
                }
            },
            {
                status: 200,
                body: {
                    "enrollmentId": "enrollmentId",
                    "challenge": "challenge"
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/client-api/v1/enrollments/{enrollmentId}/provisioning/status",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "provisioningStatus": "COMPLETE"
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/client-api/v1/enrollments/{enrollmentId}/response",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "challengeType": "INSTRUCTIONS",
                    "challenge": "2740f44c-c4ab-42a4-bbf1-1ad46b53b743",
                    "enrollmentId": "eb101331-f6cd-4d44-81c8-7b325619a362"
                }
            },
            {
                status: 200,
                body: {
                    "challengeType": "TWOFASELECTION",
                    "challenge": "2740f44c-c4ab-42a4-bbf1-1ad46b53b743",
                    "enrollmentId": "eb101331-f6cd-4d44-81c8-7b325619a362",
                    "challengePayload": {
                        "otpChannels": [{
                                "channelType": "EMAIL",
                                "channelValue": "xxxxxxxyz@backbase.com"
                            }, {
                                "channelType": "SMS",
                                "channelValue": "xxxxxx789"
                            }]
                    }
                }
            },
            {
                status: 200,
                body: {
                    "challengeType": "TWOFAENTRY",
                    "challenge": "2740f44c-c4ab-42a4-bbf1-1ad46b53b743",
                    "enrollmentId": "eb101331-f6cd-4d44-81c8-7b325619a362",
                    "challengePayload": {
                        "resendDelay": 60,
                        "resendsRemaining": 1,
                        "attemptsRemaining": 2,
                        "otpSize": 6,
                        "otpChannel": {
                            "channelType": "EMAIL",
                            "channelValue": "xxxxxxxyz@backbase.com"
                        }
                    }
                }
            },
            {
                status: 200,
                body: {
                    "challengeType": "PASSWORD",
                    "challenge": "2740f44c-c4ab-42a4-bbf1-1ad46b53b743",
                    "enrollmentId": "eb101331-f6cd-4d44-81c8-7b325619a362",
                    "challengePayload": {
                        "redirectUrl": "https://identity-url-here/auth/realms/backbase/protocol/openid-connect/auth?redirect_uri=http://completion-url-here&client_id=self-enrollment&response_type=code&scope=openid"
                    }
                }
            },
            {
                status: 200,
                body: {
                    "enrollmentId": "enrollmentId",
                    "challenge": "challenge"
                }
            },
            {
                status: 400,
                body: {
                    "challengeType": "INSTRUCTIONS",
                    "challenge": "2740f44c-c4ab-42a4-bbf1-1ad46b53b743",
                    "enrollmentId": "eb101331-f6cd-4d44-81c8-7b325619a362"
                }
            },
            {
                status: 400,
                body: {
                    "challengeType": "TWOFASELECTION",
                    "challenge": "2740f44c-c4ab-42a4-bbf1-1ad46b53b743",
                    "enrollmentId": "eb101331-f6cd-4d44-81c8-7b325619a362",
                    "challengePayload": {
                        "otpChannels": [{
                                "channelType": "EMAIL",
                                "channelValue": "xxxxxxxyz@backbase.com"
                            }, {
                                "channelType": "SMS",
                                "channelValue": "xxxxxx789"
                            }]
                    }
                }
            },
            {
                status: 400,
                body: {
                    "challengeType": "TWOFAENTRY",
                    "challenge": "2740f44c-c4ab-42a4-bbf1-1ad46b53b743",
                    "enrollmentId": "eb101331-f6cd-4d44-81c8-7b325619a362",
                    "challengePayload": {
                        "resendDelay": 60,
                        "resendsRemaining": 1,
                        "attemptsRemaining": 2,
                        "otpSize": 6,
                        "otpChannel": {
                            "channelType": "EMAIL",
                            "channelValue": "xxxxxxxyz@backbase.com"
                        }
                    }
                }
            },
            {
                status: 400,
                body: {
                    "enrollmentId": "enrollmentId",
                    "challenge": "challenge"
                }
            },
            {
                status: 409,
                body: {
                    "challengeType": "INSTRUCTIONS",
                    "challenge": "2740f44c-c4ab-42a4-bbf1-1ad46b53b743",
                    "enrollmentId": "eb101331-f6cd-4d44-81c8-7b325619a362"
                }
            },
            {
                status: 409,
                body: {
                    "challengeType": "TWOFASELECTION",
                    "challenge": "2740f44c-c4ab-42a4-bbf1-1ad46b53b743",
                    "enrollmentId": "eb101331-f6cd-4d44-81c8-7b325619a362",
                    "challengePayload": {
                        "otpChannels": [{
                                "channelType": "EMAIL",
                                "channelValue": "xxxxxxxyz@backbase.com"
                            }, {
                                "channelType": "SMS",
                                "channelValue": "xxxxxx789"
                            }]
                    }
                }
            },
            {
                status: 409,
                body: {
                    "challengeType": "TWOFAENTRY",
                    "challenge": "2740f44c-c4ab-42a4-bbf1-1ad46b53b743",
                    "enrollmentId": "eb101331-f6cd-4d44-81c8-7b325619a362",
                    "challengePayload": {
                        "resendDelay": 60,
                        "resendsRemaining": 1,
                        "attemptsRemaining": 2,
                        "otpSize": 6,
                        "otpChannel": {
                            "channelType": "EMAIL",
                            "channelValue": "xxxxxxxyz@backbase.com"
                        }
                    }
                }
            },
            {
                status: 409,
                body: {
                    "enrollmentId": "enrollmentId",
                    "challenge": "challenge"
                }
            },
            {
                status: 500,
                body: {
                    "enrollmentId": "enrollmentId",
                    "challenge": "challenge"
                }
            },
        ]
    },
]);
