// @backbase/payord-stop-checks-list-widget-ang/esm2015/lib/payord-stop-checks-list-widget-ang.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { PayordStopChecksListWidgetAngComponent } from './payord-stop-checks-list-widget-ang.component';
import { StopChecksListContainerComponent, StopChecksListContainerCustomizableDirective, } from './containers/stop-checks-list-container.component';
import { StopChecksEmptyListComponent, StopChecksEmptyListCustomizableDirective, } from './components/empty-list/empty-list.component';
import { StopChecksListComponent, StopChecksListCustomizableDirective, } from './components/list/stop-checks-list.component';
import { StopChecksDetailsComponent, StopChecksDetailsCustomizableDirective, } from './components/details-modal/stop-checks-details.component';
import { AmountModule } from '@backbase/ui-ang/amount';
import { BadgeModule } from '@backbase/ui-ang/badge';
import { ButtonModule } from '@backbase/ui-ang/button';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { ErrorCommonStateModule } from '@backbase/ui-ang/common-error-state';
import { IconModule } from '@backbase/ui-ang/icon';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { ModalModule } from '@backbase/ui-ang/modal';
import { TableModule } from '@backbase/ui-ang/table';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { HeaderModule } from '@backbase/ui-ang/header';
import { PaginationModule } from '@backbase/ui-ang/pagination';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { PaymentOrdersUiModule } from '@backbase/payment-orders-ang';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const components = [
    PayordStopChecksListWidgetAngComponent,
    StopChecksListContainerComponent,
    StopChecksEmptyListComponent,
    StopChecksListComponent,
    StopChecksDetailsComponent,
];
const directives = [
    StopChecksListContainerCustomizableDirective,
    StopChecksEmptyListCustomizableDirective,
    StopChecksListCustomizableDirective,
    StopChecksDetailsCustomizableDirective,
];
const uiModules = [
    EmptyStateModule,
    ButtonModule,
    LoadingIndicatorModule,
    TableModule,
    BadgeModule,
    IconModule,
    ErrorCommonStateModule,
    AmountModule,
    TooltipModule,
    ModalModule,
    HeaderModule,
    PaginationModule,
    LoadButtonModule,
];
/**
 * Exports the UI components, directives and modules for the stop checks list widget
 */
export class PayordStopChecksListWidgetAngModule {
}
/** @nocollapse */ PayordStopChecksListWidgetAngModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordStopChecksListWidgetAngModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ PayordStopChecksListWidgetAngModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordStopChecksListWidgetAngModule, declarations: [PayordStopChecksListWidgetAngComponent,
        StopChecksListContainerComponent,
        StopChecksEmptyListComponent,
        StopChecksListComponent,
        StopChecksDetailsComponent, StopChecksListContainerCustomizableDirective,
        StopChecksEmptyListCustomizableDirective,
        StopChecksListCustomizableDirective,
        StopChecksDetailsCustomizableDirective], imports: [CommonModule, i1.BackbaseCoreModule, PaymentOrdersUiModule, EmptyStateModule,
        ButtonModule,
        LoadingIndicatorModule,
        TableModule,
        BadgeModule,
        IconModule,
        ErrorCommonStateModule,
        AmountModule,
        TooltipModule,
        ModalModule,
        HeaderModule,
        PaginationModule,
        LoadButtonModule], exports: [PayordStopChecksListWidgetAngComponent,
        StopChecksListContainerComponent,
        StopChecksEmptyListComponent,
        StopChecksListComponent,
        StopChecksDetailsComponent, StopChecksListContainerCustomizableDirective,
        StopChecksEmptyListCustomizableDirective,
        StopChecksListCustomizableDirective,
        StopChecksDetailsCustomizableDirective, PaymentOrdersUiModule, EmptyStateModule,
        ButtonModule,
        LoadingIndicatorModule,
        TableModule,
        BadgeModule,
        IconModule,
        ErrorCommonStateModule,
        AmountModule,
        TooltipModule,
        ModalModule,
        HeaderModule,
        PaginationModule,
        LoadButtonModule] });
/** @nocollapse */ PayordStopChecksListWidgetAngModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordStopChecksListWidgetAngModule, imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { PayordStopChecksListWidgetAngComponent },
            }),
            PaymentOrdersUiModule,
            ...uiModules,
        ], PaymentOrdersUiModule, EmptyStateModule,
        ButtonModule,
        LoadingIndicatorModule,
        TableModule,
        BadgeModule,
        IconModule,
        ErrorCommonStateModule,
        AmountModule,
        TooltipModule,
        ModalModule,
        HeaderModule,
        PaginationModule,
        LoadButtonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordStopChecksListWidgetAngModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { PayordStopChecksListWidgetAngComponent },
                        }),
                        PaymentOrdersUiModule,
                        ...uiModules,
                    ],
                    declarations: [...components, ...directives],
                    exports: [...components, ...directives, PaymentOrdersUiModule, ...uiModules],
                }]
        }] });
