// @backbase/payord-stop-checks-list-widget-ang/esm2015/lib/payord-stop-checks-list-widget-ang.component.js
import { Component, Input } from '@angular/core';
import { LoadMoreMethod } from '@backbase/payment-orders-ang';
import { StopChecksListPropertiesService } from './services/stop-checks-list-properties.service';
import { StopChecksListService } from './services/stop-checks-list.service';
import * as i0 from "@angular/core";
import * as i1 from "./services/stop-checks-list-properties.service";
import * as i2 from "./containers/stop-checks-list-container.component";
export class PayordStopChecksListWidgetAngComponent {
    constructor(preferences) {
        this.preferences = preferences;
    }
    /**
     * Number of payments per page
     */
    set pageSize(value) {
        this.preferences.setPageSize(value);
    }
    /**
     * Max. number of navigation pages
     */
    set maxNavPages(value) {
        this.preferences.setMaxNavPages(value);
    }
    /**
     * Number of payments per page
     */
    set paginationType(value) {
        this.preferences.setPaginationType(value);
    }
}
/** @nocollapse */ PayordStopChecksListWidgetAngComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordStopChecksListWidgetAngComponent, deps: [{ token: i1.StopChecksListPropertiesService }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ PayordStopChecksListWidgetAngComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: PayordStopChecksListWidgetAngComponent, selector: "bb-payord-stop-checks-list-widget-ang", inputs: { pageSize: "pageSize", maxNavPages: "maxNavPages", paginationType: "paginationType" }, providers: [StopChecksListService, StopChecksListPropertiesService], ngImport: i0, template: ` <bb-stop-checks-list-container></bb-stop-checks-list-container> `, isInline: true, components: [{ type: i2.StopChecksListContainerComponent, selector: "bb-stop-checks-list-container" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordStopChecksListWidgetAngComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-payord-stop-checks-list-widget-ang',
                    template: ` <bb-stop-checks-list-container></bb-stop-checks-list-container> `,
                    providers: [StopChecksListService, StopChecksListPropertiesService],
                }]
        }], ctorParameters: function () { return [{ type: i1.StopChecksListPropertiesService }]; }, propDecorators: { pageSize: [{
                type: Input
            }], maxNavPages: [{
                type: Input
            }], paginationType: [{
                type: Input
            }] } });
