// @backbase/ui-ang/esm2015/dropdown-menu/directives/dropdown-menu-option.directive.js
import { Directive, EventEmitter, HostBinding, HostListener, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
export class DropdownMenuOptionToken {
}
export class DropdownMenuOptionDirective extends DropdownMenuOptionToken {
    constructor() {
        super(...arguments);
        /**
         * Boolean value to add/remove the active class on menuitem
         *
         * @default false
         */
        this.active = false;
        /**
         * Event that is emitted when user clicks on menuitem
         */
        this.selectionChange = new EventEmitter();
    }
    onSelect() {
        this.selectionChange.emit(this);
    }
}
DropdownMenuOptionDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownMenuOptionDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
DropdownMenuOptionDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: DropdownMenuOptionDirective, selector: "[bbDropdownMenuOption]", inputs: { bbDropdownMenuOption: "bbDropdownMenuOption", active: "active" }, outputs: { selectionChange: "selectionChange" }, host: { listeners: { "click": "onSelect()", "keydown.enter": "onSelect()", "keydown.space": "onSelect()" }, properties: { "class.active": "this.active" } }, providers: [
        {
            provide: DropdownMenuOptionToken,
            useExisting: DropdownMenuOptionDirective,
        },
    ], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownMenuOptionDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbDropdownMenuOption]',
                    providers: [
                        {
                            provide: DropdownMenuOptionToken,
                            useExisting: DropdownMenuOptionDirective,
                        },
                    ],
                }]
        }], propDecorators: { bbDropdownMenuOption: [{
                type: Input
            }], active: [{
                type: HostBinding,
                args: ['class.active']
            }, {
                type: Input
            }], selectionChange: [{
                type: Output
            }], onSelect: [{
                type: HostListener,
                args: ['click']
            }, {
                type: HostListener,
                args: ['keydown.enter']
            }, {
                type: HostListener,
                args: ['keydown.space']
            }] } });
