// @backbase/ui-ang/esm2015/dropdown-menu/directives/dropdown-menu-full-width.directive.js
import { Directive, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class DropdownMenuFullWidthDirective {
    constructor(elementRef, renderer) {
        this.elementRef = elementRef;
        this.renderer = renderer;
        /**
         * If true it will stretch the dropdown menu to 100% width
         */
        this.bbDropdownToggleFullWidth = false;
    }
    ngAfterViewInit() {
        const dropdownBtn = this.elementRef.nativeElement.querySelector('[ngbDropdownToggle]');
        const dropdownMenu = this.elementRef.nativeElement.querySelector('[ngbDropdownMenu]');
        const splitBtnWrapper = dropdownBtn === null || dropdownBtn === void 0 ? void 0 : dropdownBtn.closest('div.btn-group');
        if (this.bbDropdownToggleFullWidth) {
            this.dropdownMenuChanges = new MutationObserver((mutations) => {
                mutations.forEach((mutation) => {
                    if (mutation.attributeName === 'class' && dropdownMenu.classList.contains('show')) {
                        const minWidth = splitBtnWrapper ? splitBtnWrapper.clientWidth : dropdownBtn.clientWidth;
                        this.renderer.setStyle(dropdownMenu, 'min-width', `${minWidth}px`);
                    }
                });
            });
            this.dropdownMenuChanges.observe(dropdownMenu, {
                attributes: true,
            });
        }
    }
    ngOnDestroy() {
        var _a;
        (_a = this.dropdownMenuChanges) === null || _a === void 0 ? void 0 : _a.disconnect();
    }
}
DropdownMenuFullWidthDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownMenuFullWidthDirective, deps: [{ token: i0.ElementRef }, { token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Directive });
DropdownMenuFullWidthDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: DropdownMenuFullWidthDirective, selector: "[bbDropdownToggleFullWidth]", inputs: { bbDropdownToggleFullWidth: "bbDropdownToggleFullWidth" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownMenuFullWidthDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbDropdownToggleFullWidth]',
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.Renderer2 }]; }, propDecorators: { bbDropdownToggleFullWidth: [{
                type: Input
            }] } });
