// @backbase/ui-ang/esm2015/pagination/pagination.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NgbPaginationModule } from '@ng-bootstrap/ng-bootstrap';
import { PaginatorComponent } from './pagination.component';
import { IconModule } from '@backbase/ui-ang/icon';
import * as i0 from "@angular/core";
export class PaginationModule {
}
PaginationModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaginationModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PaginationModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaginationModule, declarations: [PaginatorComponent], imports: [CommonModule, NgbPaginationModule, IconModule], exports: [PaginatorComponent] });
PaginationModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaginationModule, imports: [[CommonModule, NgbPaginationModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaginationModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, NgbPaginationModule, IconModule],
                    declarations: [PaginatorComponent],
                    exports: [PaginatorComponent],
                }]
        }] });
