// @backbase/foundation-ang/esm2015/core/backbase-core/communication/deprecated/flows.service.js
import { Injectable, InjectionToken, Inject, Optional } from '@angular/core';
import { Graph } from '../../models/graph';
export const CONFIG_FLOWS = new InjectionToken('@backbase/foundation Communcation Flow Configuration');
export class FlowsService {
    constructor(flowsConfig) {
        this.transitions = new Graph();
        if (flowsConfig !== null) {
            flowsConfig.forEach(flows => {
                flows.forEach(flow => {
                    this.addFlow(flow);
                });
            });
        }
    }
    addFlow(config) {
        const currentEdge = this.transitions.getEdge(config.output.classId, config.input.classId);
        const flow = {
            output: config.output.outputName,
            input: config.input.inputName,
            map: config.mapPayload,
        };
        const edge = currentEdge ? [...currentEdge, flow] : [flow];
        this.transitions.connect(config.output.classId, config.input.classId, edge);
    }
    getFlows(from, fromOutput, to, toInput) {
        const flows = to.reduce((acc, classId) => [...acc, ...(this.transitions.getEdge(from, classId) || [])], []);
        return flows.filter(flow => flow.output === fromOutput && (toInput === undefined || flow.input === undefined || toInput === flow.input));
    }
}
FlowsService.decorators = [
    { type: Injectable }
];
FlowsService.ctorParameters = () => [
    { type: Array, decorators: [{ type: Optional }, { type: Inject, args: [CONFIG_FLOWS,] }] }
];
