// @backbase/product-summary-ui/esm2015/src/bb-account-graphical-header-ui/account-graphical-header.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import { ButtonModule } from '@backbase/ui-ang/button';
import { PaymentCardModule } from '@backbase/ui-ang/payment-card';
import { FormatPipeModule, ProductNumberPipeModule, ValidationModule } from '@backbase/product-summary-common-ang';
import { AccountHeaderModule } from '../bb-account-header-ui/bb-account-header.module';
import { CreditLimitModule } from '../bb-credit-limit-ui/credit-limit.module';
import { AccountGraphicalHeaderComponent } from './account-graphical-header.component';
import * as i0 from "@angular/core";
const uiModules = [
    BbDatePipeModule,
    ButtonModule,
    PaymentCardModule,
    CreditLimitModule,
    AccountHeaderModule,
    ProductNumberPipeModule,
    ValidationModule,
    FormatPipeModule,
];
const components = [AccountGraphicalHeaderComponent];
export class AccountGraphicalHeaderModule {
}
AccountGraphicalHeaderModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountGraphicalHeaderModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AccountGraphicalHeaderModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountGraphicalHeaderModule, declarations: [AccountGraphicalHeaderComponent], imports: [CommonModule, BbDatePipeModule,
        ButtonModule,
        PaymentCardModule,
        CreditLimitModule,
        AccountHeaderModule,
        ProductNumberPipeModule,
        ValidationModule,
        FormatPipeModule], exports: [AccountGraphicalHeaderComponent] });
AccountGraphicalHeaderModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountGraphicalHeaderModule, imports: [[CommonModule, ...uiModules]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountGraphicalHeaderModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules],
                    declarations: [...components],
                    exports: [...components],
                }]
        }] });
