// @backbase/initiate-payment-journey-ang/esm2015/lib/initiate-payment-journey.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { provideRoutes, RouterModule } from '@angular/router';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { PaymentOptionsService, PaymentOrderHttpService, PayordPaymentErrorsConfigProvider, Routes, } from '@backbase/payment-orders-ang';
import { CreatePaymentFormComponent, CreatePaymentFormModule, CreatePaymentService, ReviewPaymentContainerComponent, ReviewPaymentModule, } from '@backbase/payments-collection-ui-ang';
import { PayordFormBuilderAngModule } from '@backbase/payord-form-builder-ang';
import { InitiatePaymentJourneyComponent } from './initiate-payment-journey.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
export const InitiatePaymentDefaultRoute = {
    path: '',
    component: InitiatePaymentJourneyComponent,
    children: [
        { path: '', redirectTo: Routes.FORM, pathMatch: 'full' },
        { path: Routes.FORM, component: CreatePaymentFormComponent },
        { path: Routes.REVIEW, component: ReviewPaymentContainerComponent },
    ],
};
export class InitiatePaymentJourneyModule {
    static forRoot(data = {
        route: InitiatePaymentDefaultRoute,
    }) {
        return {
            ngModule: InitiatePaymentJourneyModule,
            providers: [provideRoutes([data.route])],
        };
    }
}
/** @nocollapse */ InitiatePaymentJourneyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InitiatePaymentJourneyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ InitiatePaymentJourneyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InitiatePaymentJourneyModule, declarations: [InitiatePaymentJourneyComponent], imports: [CommonModule,
        RouterModule,
        ReviewPaymentModule,
        CreatePaymentFormModule,
        PayordFormBuilderAngModule, i1.BackbaseCoreModule] });
/** @nocollapse */ InitiatePaymentJourneyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InitiatePaymentJourneyModule, providers: [
        CreatePaymentService,
        PaymentOptionsService,
        PaymentOrderHttpService,
        PayordPaymentErrorsConfigProvider,
    ], imports: [[
            CommonModule,
            RouterModule,
            ReviewPaymentModule,
            CreatePaymentFormModule,
            PayordFormBuilderAngModule,
            BackbaseCoreModule.withConfig({
                classMap: {
                    InitiatePaymentJourneyComponent,
                },
            }),
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InitiatePaymentJourneyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [InitiatePaymentJourneyComponent],
                    imports: [
                        CommonModule,
                        RouterModule,
                        ReviewPaymentModule,
                        CreatePaymentFormModule,
                        PayordFormBuilderAngModule,
                        BackbaseCoreModule.withConfig({
                            classMap: {
                                InitiatePaymentJourneyComponent,
                            },
                        }),
                    ],
                    providers: [
                        CreatePaymentService,
                        PaymentOptionsService,
                        PaymentOrderHttpService,
                        PayordPaymentErrorsConfigProvider,
                    ],
                }]
        }] });
