// @backbase/initiate-payment-journey-ang/esm2015/lib/initiate-payment-journey.component.js
import { Component, Inject, Optional } from '@angular/core';
import { NavigationService, RoutableModalService, StateService, } from '@backbase/payment-orders-ang';
import { InitiatePaymentNavigationService } from './services/initiate-payment-navigation.service';
import { InitiatePaymentRoutableModalService } from './services/initiate-payment-routable-modal.service';
import { INITIATE_PAYMENT_JOURNEY_COMMUNICATOR, } from './services/communication.service';
import { InitiatePaymentConfigService } from './services/initiate-payment-config.service';
import { CreatePaymentConfigSetupService } from '@backbase/payments-collection-ui-ang';
import * as i0 from "@angular/core";
import * as i1 from "./services/initiate-payment-config.service";
import * as i2 from "@angular/router";
export class InitiatePaymentJourneyComponent {
    constructor(configService, communicator) {
        this.configService = configService;
        this.communicator = communicator;
        if (communicator) {
            communicator.init({
                setupData(data) {
                    configService.setup(data);
                },
            });
        }
        else {
            configService.setup();
        }
    }
    ngOnDestroy() {
        this.communicator.reset();
    }
}
/** @nocollapse */ InitiatePaymentJourneyComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InitiatePaymentJourneyComponent, deps: [{ token: i1.InitiatePaymentConfigService }, { token: INITIATE_PAYMENT_JOURNEY_COMMUNICATOR, optional: true }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ InitiatePaymentJourneyComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: InitiatePaymentJourneyComponent, selector: "bb-initiate-payment-journey", providers: [
        StateService,
        InitiatePaymentConfigService,
        CreatePaymentConfigSetupService,
        { provide: NavigationService, useClass: InitiatePaymentNavigationService },
        {
            provide: RoutableModalService,
            useClass: InitiatePaymentRoutableModalService,
        },
    ], ngImport: i0, template: ` <router-outlet></router-outlet>`, isInline: true, directives: [{ type: i2.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InitiatePaymentJourneyComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-initiate-payment-journey',
                    template: ` <router-outlet></router-outlet>`,
                    providers: [
                        StateService,
                        InitiatePaymentConfigService,
                        CreatePaymentConfigSetupService,
                        { provide: NavigationService, useClass: InitiatePaymentNavigationService },
                        {
                            provide: RoutableModalService,
                            useClass: InitiatePaymentRoutableModalService,
                        },
                    ],
                }]
        }], ctorParameters: function () { return [{ type: i1.InitiatePaymentConfigService }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [INITIATE_PAYMENT_JOURNEY_COMMUNICATOR]
                }] }]; } });
