// @backbase/billpay-journeys-common/esm2015/lib/services/billpay-enrolment.guard.js
import { Injectable } from '@angular/core';
import { Router } from '@angular/router';
import { map } from 'rxjs/operators';
import { ENROLMENT_PATH } from '../constants';
import { BillpayEnrolmentService } from './billpay-enrolment.service';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "./billpay-enrolment.service";
export class BillpayEnrolmentGuard {
    constructor(router, enrolmentService) {
        this.router = router;
        this.enrolmentService = enrolmentService;
    }
    canActivate(route, state) {
        return this.enrolmentService.isEnrolled().pipe(map((enrolled) => {
            const currentRoute = this.enrolmentService.getCurrentRoute();
            if (!enrolled && !(currentRoute === null || currentRoute === void 0 ? void 0 : currentRoute.endsWith(ENROLMENT_PATH))) {
                this.enrolmentService.previousRoute.next(state.url);
                const parentRoutes = route.pathFromRoot.map(snapshot => { var _a; return (_a = snapshot === null || snapshot === void 0 ? void 0 : snapshot.routeConfig) === null || _a === void 0 ? void 0 : _a.path; }).filter(item => !!item);
                this.router.navigate([...parentRoutes, ENROLMENT_PATH]);
                return false;
            }
            return true;
        }));
    }
}
/** @nocollapse */ BillpayEnrolmentGuard.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BillpayEnrolmentGuard, deps: [{ token: i1.Router }, { token: i2.BillpayEnrolmentService }], target: i0.ɵɵFactoryTarget.Injectable });
/** @nocollapse */ BillpayEnrolmentGuard.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BillpayEnrolmentGuard, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BillpayEnrolmentGuard, decorators: [{
            type: Injectable,
            args: [{ providedIn: 'root' }]
        }], ctorParameters: function () { return [{ type: i1.Router }, { type: i2.BillpayEnrolmentService }]; } });
