// @backbase/billpay-journeys-common/esm2015/lib/services/billpay-enrolment.service.js
import { Injectable } from '@angular/core';
import { BillPayEnrolmentService } from '@backbase/data-ang/billpay';
import { catchError, map } from 'rxjs/operators';
import { BehaviorSubject, of } from 'rxjs';
import { ActivatedRoute, Router } from '@angular/router';
import { ENROLMENT_PATH } from '../constants';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/billpay";
import * as i2 from "@angular/router";
export class BillpayEnrolmentService {
    constructor(dataService, route, router) {
        this.dataService = dataService;
        this.route = route;
        this.router = router;
        this.previousRoute = new BehaviorSubject(undefined);
        this.getRoutePath = (route, path) => {
            if (route.snapshot && route.snapshot.routeConfig && route.snapshot.routeConfig.path) {
                path += '/' + route.snapshot.routeConfig.path;
            }
            if (!route.firstChild) {
                return path;
            }
            return this.getRoutePath(route.firstChild, path);
        };
    }
    enrolUser() {
        return this.dataService.postBillPayEnrolment('response').pipe(map(response => response.status === 204), catchError(error => of(error.status === 204)));
    }
    isEnrolled() {
        return this.dataService.getBillPayEnrolment().pipe(map(body => body.enrolment.enrolled), catchError(() => of(false)));
    }
    getCurrentRoute() {
        return this.getRoutePath(this.route, '');
    }
    redirectUser(previousRoute) {
        if (previousRoute && !previousRoute.includes(ENROLMENT_PATH)) {
            this.router.navigateByUrl(previousRoute);
        }
        else {
            this.router.navigate(['./']);
        }
    }
}
/** @nocollapse */ BillpayEnrolmentService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BillpayEnrolmentService, deps: [{ token: i1.BillPayEnrolmentService }, { token: i2.ActivatedRoute }, { token: i2.Router }], target: i0.ɵɵFactoryTarget.Injectable });
/** @nocollapse */ BillpayEnrolmentService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BillpayEnrolmentService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BillpayEnrolmentService, decorators: [{
            type: Injectable,
            args: [{ providedIn: 'root' }]
        }], ctorParameters: function () { return [{ type: i1.BillPayEnrolmentService }, { type: i2.ActivatedRoute }, { type: i2.Router }]; } });
