// @backbase/messages-common/esm2015/src/components/messages-preview/messages-preview.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MessagesBodyPreviewComponent } from './messages-preview.component';
import * as i0 from "@angular/core";
export class MessagesBodyPreviewModule {
}
MessagesBodyPreviewModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesBodyPreviewModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MessagesBodyPreviewModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesBodyPreviewModule, declarations: [MessagesBodyPreviewComponent], imports: [CommonModule], exports: [MessagesBodyPreviewComponent] });
MessagesBodyPreviewModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesBodyPreviewModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesBodyPreviewModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [MessagesBodyPreviewComponent],
                    exports: [MessagesBodyPreviewComponent],
                    imports: [CommonModule],
                }]
        }] });
