// @backbase/messages-common/esm2015/src/components/messages-preview/messages-preview.component.js
import { Component, Input, ViewChild } from '@angular/core';
import { BehaviorSubject, Subject } from 'rxjs';
import { delay, filter, takeUntil, tap } from 'rxjs/operators';
import * as i0 from "@angular/core";
export class MessagesBodyPreviewComponent {
    constructor(renderer, cd) {
        this.renderer = renderer;
        this.cd = cd;
        /**
         * Flag to control visibility
         */
        this.isVisible = false;
        this.iframeContext = new BehaviorSubject(undefined);
        this.destroy = new Subject();
    }
    /**
     * Body from body message
     */
    set body(val) {
        this.iframeContext.next(val);
    }
    ngOnDestroy() {
        this.destroy.next();
    }
    /**
     * Load preview iframe
     */
    iframeLoad() {
        const iframeElement = this.iframeRef.nativeElement;
        const iframeBody = iframeElement.contentDocument.body;
        this.renderer.setStyle(iframeElement, 'width', '100%');
        this.renderer.setStyle(iframeElement, 'border', '0');
        this.renderer.setStyle(iframeBody, 'margin', '0');
        this.iframeContext
            .pipe(filter(Boolean), tap(content => this.renderer.setProperty(iframeBody, 'innerHTML', content)), delay(300), takeUntil(this.destroy))
            .subscribe(() => {
            let currentHeight = iframeBody.scrollHeight;
            let previousHeight = 0;
            let counter = 0;
            /* Several iterations are needed to set iframe height after recalculating template body height
            that contains elements with height or size set in relative units. */
            do {
                this.renderer.setStyle(iframeElement, 'height', `${currentHeight}px`);
                previousHeight = currentHeight;
                currentHeight = iframeBody.scrollHeight;
                counter++;
            } while (previousHeight !== currentHeight && counter < 5);
            this.isVisible = true;
            this.cd.detectChanges();
        });
    }
}
MessagesBodyPreviewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesBodyPreviewComponent, deps: [{ token: i0.Renderer2 }, { token: i0.ChangeDetectorRef }], target: i0.ɵɵFactoryTarget.Component });
MessagesBodyPreviewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.14", type: MessagesBodyPreviewComponent, selector: "bb-messages-body-preview, bb-message-preview", inputs: { body: "body" }, viewQueries: [{ propertyName: "iframeRef", first: true, predicate: ["iframeRef"], descendants: true, static: true }], ngImport: i0, template: "<iframe\n  #iframeRef\n  title=\"Message preview\"\n  i18n-title=\"Message preview title@@messages.message-preview-title\"\n  [style.visibility]=\"isVisible ? 'visible' : 'hidden'\"\n  (load)=\"iframeLoad()\"\n></iframe>\n" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesBodyPreviewComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-messages-body-preview, bb-message-preview',
                    templateUrl: './messages-preview.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }, { type: i0.ChangeDetectorRef }]; }, propDecorators: { body: [{
                type: Input
            }], iframeRef: [{
                type: ViewChild,
                args: ['iframeRef', { static: true }]
            }] } });
