// @backbase/contact-manager-widget-ang/esm2015/lib/contact-manager.module.js
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ContactManagerWidgetComponent } from './contact-manager.component';
import { ContactsListContainerComponent } from './containers/contacts-list-container/contacts-list-container.component';
import { ContactsEmptySearchStateCustomizableDirective, ContactsEmptyStateCustomizableDirective, ContactsErrorStateCustomizableDirective, ContactsListContainerCustomizableDirective, ContactsLoadingCustomizableDirective, ContactsLoadMoreCustomizableDirective, ContactsTopSectionCustomizableDirective, } from './containers/contacts-list-container/contacts-list-container.directive';
import { ContactsListWidgetComponent } from './components/contacts-list/contact-list.component';
import { ContactsListItemComponent } from './components/contacts-list/list-item/contact-list-item.component';
import { ContactsFormComponent, ContactsFormCustomizableDirective, } from './components/contacts-form/contacts-form.component';
import { ContactsFormContainerComponent } from './containers/contacts-form-container/contacts-form-container.component';
import { ContactsDetailComponent, ContactsDetailCustomizableDirective, } from './components/contacts-detail/contacts-detail.component';
import { ContactsDetailContainerComponent } from './containers/contacts-detail-container/contacts-detail-container.component';
import { CanDeactivateGuard } from './services/can-deactivate-guard.service';
import { ContactsService } from './services/contacts.service';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { RouterModule } from '@angular/router';
import { ConfirmActionDialogModule, IbanValidatorModule, NoWhiteSpaceModule, EmailDomainModule, } from '@backbase/contact-common-ang';
import { NgbDropdownModule, NgbTypeaheadModule } from '@ng-bootstrap/ng-bootstrap';
import { ContactConfigProviders } from './config.providers';
import { ContactSelectAccountTypeComponent } from './components/contact-select-account-type/contact-select-account-type.component';
import { ContactsTypeListContainerComponent } from './containers/contacts-type-list-container/contacts-type-list-container.component';
import { ContactsCreateSuccessNotificationCustomizableDirective } from './directives/contacts-create-success-notification-customizable.directive';
import { ContactsCreateErrorNotificationCustomizableDirective } from './directives/contacts-create-error-notification-customizable.directive';
import { ContactsDeleteConfirmationCustomizableDirective } from './directives/contacts-delete-confirmation-customizable.directive';
import { ContactsDeleteErrorNotificationCustomizableDirective } from './directives/contacts-delete-error-notification-customizable.directive';
import { ContactsDeleteSuccessNotificationCustomizableDirective } from './directives/contacts-delete-success-notification-customizable.directive';
import { ContactsEditDiscardConfirmationCustomizableDirective } from './directives/contacts-edit-discard-confirmation-customizable.directive';
import { ContactsEditErrorNotificationCustomizableDirective } from './directives/contacts-edit-error-notification-customizable.directive';
import { ContactsEditSuccessNotificationCustomizableDirective } from './directives/contacts-edit-success-notification-customizable.directive';
import { AvatarModule } from '@backbase/ui-ang/avatar';
import { BbIbanPipeModule } from '@backbase/ui-ang/iban-pipe';
import { ButtonModule } from '@backbase/ui-ang/button';
import { DropdownMenuModule } from '@backbase/ui-ang/dropdown-menu';
import { EllipsisModule } from '@backbase/ui-ang/ellipsis';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { ErrorCommonStateModule } from '@backbase/ui-ang/common-error-state';
import { HeaderModule } from '@backbase/ui-ang/header';
import { IconModule } from '@backbase/ui-ang/icon';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { NotificationModule } from '@backbase/ui-ang/notification';
import { PaginatorModule } from '@backbase/ui-ang/pagination';
import { SearchBoxModule } from '@backbase/ui-ang/search-box';
import { InputEmailModule } from '@backbase/ui-ang/input-email';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const uiModules = [
    ButtonModule,
    IconModule,
    NotificationModule,
    FormsModule,
    InputTextModule,
    InputValidationMessageModule,
    HeaderModule,
    EmptyStateModule,
    AvatarModule,
    PaginatorModule,
    DropdownMenuModule,
    EllipsisModule,
    LoadingIndicatorModule,
    ErrorCommonStateModule,
    NgbDropdownModule,
    NgbTypeaheadModule,
    ReactiveFormsModule,
    BbIbanPipeModule,
    SearchBoxModule,
    InputEmailModule,
    LoadButtonModule,
];
const dependentModules = [IbanValidatorModule, NoWhiteSpaceModule, ConfirmActionDialogModule, EmailDomainModule];
const components = [
    ContactManagerWidgetComponent,
    ContactsListWidgetComponent,
    ContactsListItemComponent,
    ContactsListContainerComponent,
    ContactsDetailComponent,
    ContactsDetailContainerComponent,
    ContactsFormComponent,
    ContactsFormContainerComponent,
    ContactSelectAccountTypeComponent,
    ContactsTypeListContainerComponent,
];
const customizationDirectives = [
    ContactsListContainerCustomizableDirective,
    ContactsTopSectionCustomizableDirective,
    ContactsDetailCustomizableDirective,
    ContactsFormCustomizableDirective,
    ContactsLoadMoreCustomizableDirective,
    ContactsLoadingCustomizableDirective,
    ContactsEmptyStateCustomizableDirective,
    ContactsEmptySearchStateCustomizableDirective,
    ContactsErrorStateCustomizableDirective,
    ContactsCreateErrorNotificationCustomizableDirective,
    ContactsCreateSuccessNotificationCustomizableDirective,
    ContactsDeleteConfirmationCustomizableDirective,
    ContactsDeleteErrorNotificationCustomizableDirective,
    ContactsDeleteSuccessNotificationCustomizableDirective,
    ContactsEditDiscardConfirmationCustomizableDirective,
    ContactsEditErrorNotificationCustomizableDirective,
    ContactsEditSuccessNotificationCustomizableDirective,
];
/**
 * Exports the Widgets and UI components for managing contacts.
 *
 * The following widgets are included in this module:
 *  - `ContactManagerWidgetComponent`
 *
 * @see ContactManagerWidgetComponent
 * @usageNotes
 *
 * ### Include the exported widgets in your `AppModule`:
 *
 * ```typescript
 * @NgModule({
 *   imports: [
 *     ContactManagerWidgetModule,
 *   ],
 *  providers:[{
 *   provide: CONTACT_MANAGER_BASE_PATH,
 *   useFactory: getBasePath('contact-manager'),
 *   deps: [PAGE_CONFIG], },
 * {
 *   provide: ACCESS_CONTROL_BASE_PATH,
 *   useFactory: getBasePath('access-control'),
 *   deps: [PAGE_CONFIG], }]
 *   ...
 * })
 * export class AppModule {}
 * ```
 */
export class ContactManagerWidgetModule {
}
/** @nocollapse */ ContactManagerWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactManagerWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ ContactManagerWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactManagerWidgetModule, declarations: [ContactManagerWidgetComponent,
        ContactsListWidgetComponent,
        ContactsListItemComponent,
        ContactsListContainerComponent,
        ContactsDetailComponent,
        ContactsDetailContainerComponent,
        ContactsFormComponent,
        ContactsFormContainerComponent,
        ContactSelectAccountTypeComponent,
        ContactsTypeListContainerComponent, ContactsListContainerCustomizableDirective,
        ContactsTopSectionCustomizableDirective,
        ContactsDetailCustomizableDirective,
        ContactsFormCustomizableDirective,
        ContactsLoadMoreCustomizableDirective,
        ContactsLoadingCustomizableDirective,
        ContactsEmptyStateCustomizableDirective,
        ContactsEmptySearchStateCustomizableDirective,
        ContactsErrorStateCustomizableDirective,
        ContactsCreateErrorNotificationCustomizableDirective,
        ContactsCreateSuccessNotificationCustomizableDirective,
        ContactsDeleteConfirmationCustomizableDirective,
        ContactsDeleteErrorNotificationCustomizableDirective,
        ContactsDeleteSuccessNotificationCustomizableDirective,
        ContactsEditDiscardConfirmationCustomizableDirective,
        ContactsEditErrorNotificationCustomizableDirective,
        ContactsEditSuccessNotificationCustomizableDirective], imports: [CommonModule,
        RouterModule, IbanValidatorModule, NoWhiteSpaceModule, ConfirmActionDialogModule, EmailDomainModule, ButtonModule,
        IconModule,
        NotificationModule,
        FormsModule,
        InputTextModule,
        InputValidationMessageModule,
        HeaderModule,
        EmptyStateModule,
        AvatarModule,
        PaginatorModule,
        DropdownMenuModule,
        EllipsisModule,
        LoadingIndicatorModule,
        ErrorCommonStateModule,
        NgbDropdownModule,
        NgbTypeaheadModule,
        ReactiveFormsModule,
        BbIbanPipeModule,
        SearchBoxModule,
        InputEmailModule,
        LoadButtonModule, i1.BackbaseCoreModule], exports: [ContactManagerWidgetComponent,
        ContactsListWidgetComponent,
        ContactsListItemComponent,
        ContactsListContainerComponent,
        ContactsDetailComponent,
        ContactsDetailContainerComponent,
        ContactsFormComponent,
        ContactsFormContainerComponent,
        ContactSelectAccountTypeComponent,
        ContactsTypeListContainerComponent, ContactsListContainerCustomizableDirective,
        ContactsTopSectionCustomizableDirective,
        ContactsDetailCustomizableDirective,
        ContactsFormCustomizableDirective,
        ContactsLoadMoreCustomizableDirective,
        ContactsLoadingCustomizableDirective,
        ContactsEmptyStateCustomizableDirective,
        ContactsEmptySearchStateCustomizableDirective,
        ContactsErrorStateCustomizableDirective,
        ContactsCreateErrorNotificationCustomizableDirective,
        ContactsCreateSuccessNotificationCustomizableDirective,
        ContactsDeleteConfirmationCustomizableDirective,
        ContactsDeleteErrorNotificationCustomizableDirective,
        ContactsDeleteSuccessNotificationCustomizableDirective,
        ContactsEditDiscardConfirmationCustomizableDirective,
        ContactsEditErrorNotificationCustomizableDirective,
        ContactsEditSuccessNotificationCustomizableDirective, IbanValidatorModule, NoWhiteSpaceModule, ConfirmActionDialogModule, EmailDomainModule] });
/** @nocollapse */ ContactManagerWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactManagerWidgetModule, providers: [CanDeactivateGuard, ContactsService, ...ContactConfigProviders], imports: [[
            CommonModule,
            RouterModule,
            ...dependentModules,
            ...uiModules,
            BackbaseCoreModule.withConfig({
                classMap: { ContactManagerWidgetComponent },
            }),
        ], IbanValidatorModule, NoWhiteSpaceModule, ConfirmActionDialogModule, EmailDomainModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactManagerWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        RouterModule,
                        ...dependentModules,
                        ...uiModules,
                        BackbaseCoreModule.withConfig({
                            classMap: { ContactManagerWidgetComponent },
                        }),
                    ],
                    declarations: [...components, ...customizationDirectives],
                    providers: [CanDeactivateGuard, ContactsService, ...ContactConfigProviders],
                    exports: [...components, ...customizationDirectives, ...dependentModules],
                    entryComponents: [
                        ContactsDetailContainerComponent,
                        ContactsListContainerComponent,
                        ContactsFormContainerComponent,
                        ContactsTypeListContainerComponent,
                    ],
                }]
        }] });
