// @backbase/accessgroup-common-ang/esm2015/lib/model/test-mock.js
export const functionAccessGroups = [
    {
        id: 'f34f7587-5ded-4f51-94c3-3a89d1b64538',
        name: 'Manage Entitlements',
        description: 'This profile will control managing of entitlements',
        serviceAgreementId: '1642b2e1-61c4-4ce3-8bd9-877327587f4b',
        validFromDate: '2019-08-16',
        validFromTime: '16:43:52',
        validUntilDate: '2019-08-23',
        validUntilTime: '16:43:52',
        permissions: [
            {
                functionId: '1011',
                assignedPrivileges: [
                    {
                        privilege: 'view',
                    },
                ],
            },
            {
                functionId: '1010',
                assignedPrivileges: [
                    {
                        privilege: 'execute',
                    },
                ],
            },
            {
                functionId: '1009',
                assignedPrivileges: [
                    {
                        privilege: 'create',
                    },
                    {
                        privilege: 'view',
                    },
                    {
                        privilege: 'edit',
                    },
                ],
            },
        ],
        type: 'REGULAR',
    },
    {
        id: 'f34f7587-5ded-4f51-94c3-3a89d1b64539',
        name: 'Manage Something Completely Different',
        description: 'And now... something completely different!',
        serviceAgreementId: '1642b2e1-61c4-4ce3-8bd9-877327587f4b',
        permissions: [
            {
                functionId: '1002',
                assignedPrivileges: [
                    {
                        privilege: 'view',
                    },
                ],
            },
            {
                functionId: '1011',
                assignedPrivileges: [
                    {
                        privilege: 'view',
                    },
                ],
            },
            {
                functionId: '1010',
                assignedPrivileges: [
                    {
                        privilege: 'execute',
                    },
                ],
            },
            {
                functionId: '1009',
                assignedPrivileges: [
                    {
                        privilege: 'create',
                    },
                    {
                        privilege: 'view',
                    },
                    {
                        privilege: 'edit',
                    },
                ],
            },
        ],
        type: 'TEMPLATE',
    },
];
export const businessFunctions = [
    {
        functionId: '1002',
        functionCode: 'payments.sepa',
        resource: 'Payments',
        name: 'SEPA CT',
        privileges: [
            { privilege: 'view', supportsLimit: false },
            { privilege: 'edit', supportsLimit: false },
            { privilege: 'delete', supportsLimit: false },
            { privilege: 'create', supportsLimit: true },
            { privilege: 'approve', supportsLimit: false },
            { privilege: 'cancel', supportsLimit: false },
        ],
    },
    {
        functionId: '1009',
        functionCode: 'assign.permissions',
        resource: 'Service Agreement',
        name: 'Assign Permissions',
        privileges: [
            { privilege: 'approve', supportsLimit: false },
            { privilege: 'view', supportsLimit: false },
            { privilege: 'create', supportsLimit: false },
            { privilege: 'edit', supportsLimit: false },
        ],
    },
    {
        functionId: '1010',
        functionCode: 'manage.users',
        resource: 'User',
        name: 'Manage Users',
        privileges: [{ privilege: 'view', supportsLimit: false }],
    },
    {
        functionId: '1011',
        functionCode: 'manage.legalentities',
        resource: 'Legal Entity',
        name: 'Manage Legal Entities',
        privileges: [
            { privilege: 'create', supportsLimit: false },
            { privilege: 'edit', supportsLimit: false },
            { privilege: 'delete', supportsLimit: false },
            { privilege: 'approve', supportsLimit: false },
            { privilege: 'view', supportsLimit: false },
        ],
    },
];
export const serviceAgreement = {
    id: 'a1198a205a404eb7b47fa553db920429',
    name: 'Online Banking Service Agreement',
    description: 'Service Agreement between Online Banking and Backbase',
    isMaster: false,
    creatorLegalEntity: 'a1198a205a404eb7b47fa553db920429',
    status: 'ENABLED',
};
export const participants = [
    {
        id: 'a1198a20-5a40-4eb7-b47f-a553db958963',
        externalId: 'KPMG',
        name: 'KPMG',
        sharingUsers: true,
        sharingAccounts: false,
    },
    {
        id: 'a1198a20-5a40-4eb7-b47f-a553db920429',
        externalId: 'Backbase',
        name: 'Backbase',
        sharingUsers: false,
        sharingAccounts: true,
    },
];
export const serviceAgreementWithParticipants = {
    serviceAgreement,
    participants,
};
export const productSummaryMockResponse = [
    {
        id: 'e6659eea893c4da48f9b7fd425de4677',
        name: 'YOUTH SAVINGS ACCOUNT',
        bankAlias: 'YOUTH SAVINGS',
        bookedBalance: 33434.0,
        availableBalance: 3423.0,
        creditLimit: 100.0,
        IBAN: 'GB79BARC20040432885594',
        BBAN: '30041010050500013L01234',
        currency: 'USD',
        externalTransferAllowed: true,
        urgentTransferAllowed: false,
        accruedInterest: 1000.0,
        number: '0000',
        principalAmount: 100.0,
        currentInvestmentValue: 100.0,
        legalEntityIds: ['8a48e4116e11f5b6016e224929450218'],
        productId: '8a48e7546e11f733016e22493e3f4024',
        productNumber: '10-23/2168-9601',
        productKindName: 'Savings Account',
        productTypeName: 'Savings Account',
        BIC: 'COBADEFF',
        bankBranchCode: '0260-0959-3',
        visible: true,
        accountOpeningDate: '2017-05-12T00:00:00.000+0000',
        accountInterestRate: 11.0,
        valueDateBalance: 1000.0,
        creditLimitUsage: 2000.0,
        creditLimitInterestRate: 3.5,
        creditLimitExpiryDate: '2017-10-10T00:00:00.000+0000',
        startDate: '2017-12-12T00:00:00.000+0000',
        termUnit: 'D',
        termNumber: 12,
        interestPaymentFrequencyUnit: 'M',
        interestPaymentFrequencyNumber: 3333,
        maturityDate: '2019-07-07T00:00:00.000+0000',
        maturityAmount: 4000.0,
        autoRenewalIndicator: true,
        interestSettlementAccount: '30041010050500013L01234',
        outstandingPrincipalAmount: 9412.0,
        monthlyInstalmentAmount: 23.23,
        amountInArrear: 11.2,
        minimumRequiredBalance: 500.0,
        creditCardAccountNumber: '30569309025904',
        validThru: '2019-09-09T00:00:00.000+0000',
        applicableInterestRate: 333.1,
        remainingCredit: 4500.0,
        outstandingPayment: 2000.2,
        minimumPayment: 200.6,
        minimumPaymentDueDate: '2017-11-04T00:00:00.000+0000',
        totalInvestmentValue: 1000.0,
        debitCards: [
            { number: '2', expiryDate: '2017-11-11' },
            { number: '1', expiryDate: '2016-12-12' },
        ],
        postCode: '20',
        accountHolderCountry: 'MK',
        creditAccount: true,
        debitAccount: false,
        lastUpdateDate: '2018-11-11T00:00:00.000+0000',
        product: {
            externalId: '2',
            externalTypeId: 'Savings Account Type',
            typeName: 'Savings Account',
            productKind: { id: 2, externalKindId: 'kind2', kindName: 'Savings Account', kindUri: 'savings-account' },
        },
    },
    {
        id: '8eb58b1d048d4112a2f7591d465c3f14',
        name: 'YOUTH DEPOSIT ACCOUNT',
        bankAlias: 'YOUTH DEPOSIT',
        bookedBalance: 33434.0,
        availableBalance: 3423.0,
        creditLimit: 100.0,
        IBAN: 'GB70BARC20040149989815',
        BBAN: '30041010050500013L01234',
        currency: 'USD',
        externalTransferAllowed: true,
        urgentTransferAllowed: false,
        accruedInterest: 1000.0,
        number: '0000',
        principalAmount: 100.0,
        currentInvestmentValue: 100.0,
        legalEntityIds: ['8a48e4116e11f5b6016e224929450218'],
        productId: '8a48e7546e11f733016e224941ff4028',
        productNumber: '10-23/2168-9601',
        productKindName: 'Term Deposit',
        productTypeName: 'One Year Deposit',
        BIC: 'COBADEFF',
        bankBranchCode: '0260-0959-3',
        visible: true,
        accountOpeningDate: '2017-05-12T00:00:00.000+0000',
        accountInterestRate: 11.0,
        valueDateBalance: 1000.0,
        creditLimitUsage: 2000.0,
        creditLimitInterestRate: 3.5,
        creditLimitExpiryDate: '2017-10-10T00:00:00.000+0000',
        startDate: '2017-12-12T00:00:00.000+0000',
        termUnit: 'D',
        termNumber: 12,
        interestPaymentFrequencyUnit: 'M',
        interestPaymentFrequencyNumber: 3333,
        maturityDate: '2019-07-07T00:00:00.000+0000',
        maturityAmount: 4000.0,
        autoRenewalIndicator: true,
        interestSettlementAccount: '30041010050500013L01234',
        outstandingPrincipalAmount: 9412.0,
        monthlyInstalmentAmount: 23.23,
        amountInArrear: 11.2,
        minimumRequiredBalance: 500.0,
        creditCardAccountNumber: '30569309025904',
        validThru: '2019-09-09T00:00:00.000+0000',
        applicableInterestRate: 333.1,
        remainingCredit: 4500.0,
        outstandingPayment: 2000.2,
        minimumPayment: 200.6,
        minimumPaymentDueDate: '2017-11-04T00:00:00.000+0000',
        totalInvestmentValue: 1000.0,
        debitCards: [
            { number: '2', expiryDate: '2017-11-11' },
            { number: '1', expiryDate: '2016-12-12' },
        ],
        postCode: '20',
        accountHolderCountry: 'MK',
        creditAccount: true,
        debitAccount: false,
        lastUpdateDate: '2018-11-11T00:00:00.000+0000',
        product: {
            externalId: '6',
            externalTypeId: 'Term Deposit Type',
            typeName: 'One Year Deposit',
            productKind: { id: 6, externalKindId: 'kind6', kindName: 'Term Deposit', kindUri: 'term-deposit' },
        },
    },
];
