// @backbase/user-common/esm2015/src/components/manage-profile/manage-phone-numbers/manage-phone-number.component.js
import { Component, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../../../services/manage-profile/manage-profile-preferences.service";
export class ManagePhoneNumberComponent {
    constructor(manageProfilePreferencesService) {
        this.manageProfilePreferencesService = manageProfilePreferencesService;
        this.updateProfile = new EventEmitter();
        this.managePhoneNumber = new EventEmitter();
        this.cancelManagePhoneNumber = new EventEmitter();
        this.managingPhoneNumber = false;
        /** @deprecated */
        this.phoneNumberPreferences = this.manageProfilePreferencesService.getPhoneNumberInputPreferences();
    }
    handleManagePhoneNumber() {
        if (this.disableManage) {
            return;
        }
        this.managingPhoneNumber = true;
        this.managePhoneNumber.emit();
    }
    handleCloseEditPhoneNumberAfterSave() {
        this.updateProfile.emit();
    }
    handleCloseEditPhoneNumber() {
        this.managingPhoneNumber = false;
        this.disableManage = false;
        this.cancelManagePhoneNumber.emit();
    }
}
ManagePhoneNumberComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePhoneNumberComponent, deps: [{ token: i1.ManageProfilePreferencesService }], target: i0.ɵɵFactoryTarget.Component });
ManagePhoneNumberComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: ManagePhoneNumberComponent, selector: "ng-component", inputs: { phoneNumber: "phoneNumber", phoneNumberCount: "phoneNumberCount", phoneNumberTypes: "phoneNumberTypes", disableManage: "disableManage", dismissTimeout: "dismissTimeout" }, outputs: { updateProfile: "updateProfile", managePhoneNumber: "managePhoneNumber", cancelManagePhoneNumber: "cancelManagePhoneNumber" }, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePhoneNumberComponent, decorators: [{
            type: Component,
            args: [{
                    template: '',
                }]
        }], ctorParameters: function () { return [{ type: i1.ManageProfilePreferencesService }]; }, propDecorators: { phoneNumber: [{
                type: Input
            }], phoneNumberCount: [{
                type: Input
            }], phoneNumberTypes: [{
                type: Input
            }], disableManage: [{
                type: Input
            }], dismissTimeout: [{
                type: Input
            }], updateProfile: [{
                type: Output
            }], managePhoneNumber: [{
                type: Output
            }], cancelManagePhoneNumber: [{
                type: Output
            }] } });
