// @backbase/user-common/esm2015/src/components/manage-profile/manage-phone-numbers/manage-phone-numbers.component.js
import { Component, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
export class ManagePhoneNumbersComponent {
    constructor() {
        this.updateProfile = new EventEmitter();
        this.managePhoneNumber = new EventEmitter();
        this.cancelManagePhoneNumber = new EventEmitter();
        this.isManagingPhoneNumberIndex = undefined;
        this.isAddingPhoneNumber = false;
        this.isAddButtonAvailable = true;
    }
    ngOnInit() {
        if (this.phoneNumbers && this.maxPhoneNumbers) {
            this.isAddButtonAvailable = this.phoneNumbers.length < this.maxPhoneNumbers;
        }
    }
    handleManagePhoneNumberOpen(index) {
        this.isManagingPhoneNumberIndex = index;
        this.managePhoneNumber.emit();
    }
    handleAddNewPhoneNumber() {
        if (this.disableManage) {
            return;
        }
        this.isAddingPhoneNumber = true;
        this.managePhoneNumber.emit();
    }
    handleManagePhoneNumberClosed() {
        this.isManagingPhoneNumberIndex = undefined;
        this.isAddingPhoneNumber = false;
        this.cancelManagePhoneNumber.emit();
    }
    updateUserProfileData() {
        this.updateProfile.emit();
    }
}
ManagePhoneNumbersComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePhoneNumbersComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ManagePhoneNumbersComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: ManagePhoneNumbersComponent, selector: "ng-component", inputs: { phoneNumbers: "phoneNumbers", phoneNumberTypes: "phoneNumberTypes", maxPhoneNumbers: "maxPhoneNumbers", disableManage: "disableManage", dismissTimeout: "dismissTimeout" }, outputs: { updateProfile: "updateProfile", managePhoneNumber: "managePhoneNumber", cancelManagePhoneNumber: "cancelManagePhoneNumber" }, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePhoneNumbersComponent, decorators: [{
            type: Component,
            args: [{
                    template: '',
                }]
        }], propDecorators: { phoneNumbers: [{
                type: Input
            }], phoneNumberTypes: [{
                type: Input
            }], maxPhoneNumbers: [{
                type: Input
            }], disableManage: [{
                type: Input
            }], dismissTimeout: [{
                type: Input
            }], updateProfile: [{
                type: Output
            }], managePhoneNumber: [{
                type: Output
            }], cancelManagePhoneNumber: [{
                type: Output
            }] } });
