// @backbase/user-common/esm2015/src/components/manage-profile/manage-phone-numbers/new-phone-number.component.js
import { Component, Input } from '@angular/core';
import { NewAddressComponent } from '../new-address/new-address.component';
import * as i0 from "@angular/core";
export class NewPhoneNumberComponent extends NewAddressComponent {
    ngOnInit() {
        this.newPhoneNumberData = {
            type: this.phoneNumberTypes && this.phoneNumberTypes.length ? this.phoneNumberTypes[0] : '',
            number: '',
        };
    }
}
NewPhoneNumberComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: NewPhoneNumberComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
NewPhoneNumberComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: NewPhoneNumberComponent, selector: "ng-component", inputs: { phoneNumberTypes: "phoneNumberTypes" }, usesInheritance: true, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: NewPhoneNumberComponent, decorators: [{
            type: Component,
            args: [{
                    template: '',
                }]
        }], propDecorators: { phoneNumberTypes: [{
                type: Input
            }] } });
