// @backbase/accounts-transactions-journeys-shared-ui-transform-ang/esm2015/lib/split-array-pipe/split-array.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SplitArrayPipe } from './split-array.pipe';
import * as i0 from "@angular/core";
const pipes = [SplitArrayPipe];
export class SplitArrayPipeModule {
}
/** @nocollapse */ SplitArrayPipeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SplitArrayPipeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ SplitArrayPipeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SplitArrayPipeModule, declarations: [SplitArrayPipe], imports: [CommonModule], exports: [SplitArrayPipe] });
/** @nocollapse */ SplitArrayPipeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SplitArrayPipeModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SplitArrayPipeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [...pipes],
                    exports: [...pipes],
                }]
        }] });
