// @backbase/foundation-ang/esm2015/store/actions.js
import { filter } from 'rxjs/operators';
/**
 * @deprecated Will be removed in v7.0.0
 */
export const createChannelAction = (action, channel) => ({
    type: action.type,
    channel,
    action,
});
const isChannelAction = (action) => 'channel' in action && 'action' in action;
const ɵ0 = isChannelAction;
/**
 * @deprecated Will be removed in v7.0.0
 */
export const toChannelAction = (action) => isChannelAction(action) ? action : createChannelAction(action);
/**
 * @deprecated Will be removed in v7.0.0
 */
export const isActionForChannel = (action, channels) => {
    return !isChannelAction(action) || action.channel === undefined || channels.includes(action.channel);
};
/**
 * @deprecated Will be removed in v7.0.0
 * Stream operator to filter the `Actions` stream to only pass `Action`s that match the given `type`.
 */
export const ofType = (type) => typeof type === 'string'
    ? (actions) => actions.pipe(filter((action) => action.type === type))
    : (actions) => actions.pipe(filter((action) => action.type === type.type));
/**
 * @deprecated Will be removed in v7.0.0
 */
export { createAction, props } from '@ngrx/store';
/**
 * @deprecated Will be removed in v7.0.0
 */
export const typePropertyIsNotAllowedMsg = 'type property is not allowed in action creators';
export { ɵ0 };
