// @backbase/foundation-ang/esm2015/store/reducers.js
import { isActionForChannel, toChannelAction } from './actions';
/**
 * @deprecated Will be removed in v7.0.0
 */
export const InitializeStateActionType = '[Backbase Reducer] INITIALIZE REDUCER';
/**
 * @deprecated Will be removed in v7.0.0
 */
export class InitializeState {
    constructor() {
        this.type = InitializeStateActionType;
    }
}
/**
 * @deprecated Will be removed in v7.0.0
 */
export const metaChannelReducer = (channels) => {
    // create a meta reducer (takes a reducer, and returns a new reducer)
    return (reducer) => {
        // create a function to initialize the state for this reducer
        const initializeState = () => reducer(undefined, new InitializeState());
        // return the new reducer, which filters actions not for this "channel"
        return (state = initializeState(), action) => {
            const channelAction = toChannelAction(action);
            if (isActionForChannel(action, channels)) {
                return reducer(state, channelAction.action);
            }
            return state;
        };
    };
};
export { createReducer, on } from '@ngrx/store';
