// @backbase/foundation-ang/esm2015/store/store-factory.service.js
import { Inject, InjectionToken } from '@angular/core';
import { StoreBroker } from './store-broker.service';
import { FeaturesService } from '@backbase/foundation-ang/future';
/**
 * @deprecated Will be removed in v7.0.0
 */
export const ENFORCE_INJECTOR_FOR_CREATE_STORE = {
    name: 'ENFORCE_INJECTOR_FOR_CREATE_STORE',
    description: 'you have to provide widget injector for createStore to make sure your widget support lazy loading',
};
/**
 * @deprecated Will be removed in v7.0.0
 * Configuration token that defines the "branch" of the application state that the store is responsible for.
 *
 * @internal
 */
export const STORE_BRANCH = new InjectionToken('@backbase store branch');
/**
 * @deprecated Will be removed in v7.0.0
 * Configuration token that defines the "channel" on which the store will dispatch any events it produces.
 *
 * @internal
 */
export const STORE_DISPATCH_CHANNEL = new InjectionToken('@backbase store dispatch channel');
/**
 * @deprecated Will be removed in v7.0.0
 * Create a scoped store on a particular branch/channel
 */
export class StoreFactory {
    constructor(storeBroker, branch, dispatchChannel, featuresService) {
        this.storeBroker = storeBroker;
        this.branch = branch;
        this.dispatchChannel = dispatchChannel;
        this.featuresService = featuresService;
    }
    createStore(storeName, reducer, effects, injector) {
        if (!injector && this.featuresService.isEnabled(ENFORCE_INJECTOR_FOR_CREATE_STORE)) {
            throw new Error(ENFORCE_INJECTOR_FOR_CREATE_STORE.description);
        }
        if (!injector) {
            console.warn(ENFORCE_INJECTOR_FOR_CREATE_STORE.description);
        }
        return this.storeBroker.initStore(storeName, reducer, effects, this.branch, this.dispatchChannel, undefined, injector);
    }
}
StoreFactory.ctorParameters = () => [
    { type: StoreBroker },
    { type: undefined, decorators: [{ type: Inject, args: [STORE_BRANCH,] }] },
    { type: undefined, decorators: [{ type: Inject, args: [STORE_DISPATCH_CHANNEL,] }] },
    { type: FeaturesService }
];
