// @backbase/foundation-ang/esm2015/store/store.module.js
import { NgModule } from '@angular/core';
import { EffectsModule as NgrxEffectsModule } from '@ngrx/effects';
import { StoreModule as NgrxStoreModule } from '@ngrx/store';
import { FeaturesService } from '@backbase/foundation-ang/future';
import { FutureFeaturesModule } from '@backbase/foundation-ang/future';
import { StoreBroker } from './store-broker.service';
import { ENFORCE_INJECTOR_FOR_CREATE_STORE } from './store-factory.service';
/**
 * @deprecated Will be removed in v7.0.0
 */
export const ALLOW_NGRX_IN_ROOT = {
    name: 'ALLOW_NGRX_IN_ROOT',
    description: 'StoreModule is deprecated - use BackbaseStoreModule instead',
};
/**
 * @deprecated
 */
export class StoreModule {
    constructor(featuresService) {
        if (featuresService.isEnabled(ALLOW_NGRX_IN_ROOT)) {
            throw new Error('StoreModule is deprecated - use BackbaseStoreModule instead');
        }
    }
}
StoreModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    NgrxEffectsModule.forRoot([]),
                    NgrxStoreModule.forRoot({}),
                    FutureFeaturesModule.allowOptIn([ALLOW_NGRX_IN_ROOT, ENFORCE_INJECTOR_FOR_CREATE_STORE]),
                ],
                providers: [StoreBroker],
            },] }
];
StoreModule.ctorParameters = () => [
    { type: FeaturesService }
];
/**
 * @deprecated Will be removed in v7.0.0
 */
export class BackbaseStoreModule {
}
BackbaseStoreModule.decorators = [
    { type: NgModule, args: [{
                imports: [NgrxEffectsModule, NgrxStoreModule, FutureFeaturesModule.allowOptIn([ENFORCE_INJECTOR_FOR_CREATE_STORE])],
            },] }
];
