// @backbase/notifications-common-ang/esm2015/index.js
export { NotificationsCommonModule } from './lib/notifications-common.module';
export * from './lib/constants/pubsub-events';
export * from './lib/models/notification-origins';
export * from './lib/models/notification-routing';
export * from './lib/models/notification-item';
export * from './lib/models/notification-stream-item';
export * from './lib/models/unread-notifications-count';
export * from './lib/models/badge-force-update-event.model';
export * from './lib/models/check-truncation.model';
export { BaseWidgetPropertiesService } from './lib/services/base-widget-properties.service';
export * from './lib/helpers/format-date';
export { enterZone, leaveZone } from './lib/helpers/ng-zone';
export { propertiesListToArray, shareReplayOnce } from './lib/helpers/rxjs-operators';
export { isElementContentTruncated } from './lib/helpers/is-truncated';
export { modelItemToBoolean, modelItemToArray, secondsToMilliseconds } from './lib/helpers/type-helpers';
export { getPropertyValue } from './lib/helpers/item-model-property';
export { NotificationsCapabilityIconComponent, NotificationsCapabilityIconCustomizableDirective, } from './lib/components/notifications-capability-icon/notifications-capability-icon.component';
export { NotificationLocalizationPipe as ӨNotificationLocalizationPipe } from './lib/pipes/notification-localization.pipe';
export { InterpolateLinkPipe as ӨInterpolateLinkPipe } from './lib/pipes/interpolate-link.pipe';
export { IsElementContentTruncatedDirective as ӨIsElementContentTruncatedDirective } from './lib/directives/is-element-content-truncated.directive';
