// @backbase/ui-ang/esm2015/loading-indicator/loading-indicator.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { LoadingIndicatorComponent } from './loading-indicator.component';
import * as i0 from "@angular/core";
export class LoadingIndicatorModule {
}
LoadingIndicatorModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoadingIndicatorModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LoadingIndicatorModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoadingIndicatorModule, declarations: [LoadingIndicatorComponent], imports: [CommonModule], exports: [LoadingIndicatorComponent] });
LoadingIndicatorModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoadingIndicatorModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoadingIndicatorModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [LoadingIndicatorComponent],
                    exports: [LoadingIndicatorComponent],
                }]
        }] });
