// @backbase/ui-ang/esm2015/_example/pagination/showcase/pagination-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { PaginationShowcaseComponent } from './pagination-showcase.component';
import { PaginationModule } from '@backbase/ui-ang/pagination';
import * as i0 from "@angular/core";
export class PaginationShowcaseModule {
}
PaginationShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaginationShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PaginationShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaginationShowcaseModule, declarations: [PaginationShowcaseComponent], imports: [CommonModule, PaginationModule] });
PaginationShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaginationShowcaseModule, imports: [[CommonModule, PaginationModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaginationShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, PaginationModule],
                    declarations: [PaginationShowcaseComponent],
                    entryComponents: [PaginationShowcaseComponent],
                }]
        }] });
