// @backbase/ui-ang/esm2015/_example/pagination/showcase/pagination-showcase.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/pagination";
let PaginationShowcaseComponent = class PaginationShowcaseComponent {
    constructor() {
        this.paginatorOne = {
            page: 0,
            totalItems: 30,
            maxNavPages: 3,
            type: 'paginator-small-untruncated',
        };
        this.paginatorTwo = {
            page: 7,
            totalItems: 100,
            maxNavPages: 5,
            type: 'paginator-small-left-truncated',
        };
        this.paginatorThree = {
            page: 0,
            totalItems: 90,
            maxNavPages: 4,
            type: 'paginator-small-right-truncated',
        };
        this.paginatorFour = {
            page: 1020 / 10,
            totalItems: 1020,
            maxNavPages: 5,
            type: 'paginator-big-left-truncated',
        };
    }
    get paginator() {
        return {
            boundaryLinks: true,
            directionLinks: true,
            itemsPerPage: 10,
            onPageChange(page) {
                console.log('onPageChange', page);
            },
        };
    }
};
PaginationShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaginationShowcaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
PaginationShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: PaginationShowcaseComponent, selector: "bb-pagination-ui-showcase", ngImport: i0, template: "<h4>Small numbers untruncated</h4>\n<bb-pagination-ui\n  class=\"bb-pagination d-flex justify-content-start align-items-center mt-4 mb-3\"\n  [page]=\"paginatorOne.page\"\n  [boundaryLinks]=\"paginator.boundaryLinks\"\n  [directionLinks]=\"paginator.directionLinks\"\n  [totalItems]=\"paginatorOne.totalItems\"\n  [itemsPerPage]=\"paginator.itemsPerPage\"\n  [maxNavPages]=\"paginatorOne.maxNavPages\"\n  (pageChange)=\"paginator.onPageChange($event)\"\n  [attr.data-role]=\"paginatorOne.type\"\n></bb-pagination-ui>\n\n<h4>Small numbers left truncated</h4>\n<bb-pagination-ui\n  class=\"bb-pagination d-flex justify-content-start align-items-center mt-4 mb-3\"\n  [page]=\"paginatorTwo.page\"\n  [boundaryLinks]=\"paginator.boundaryLinks\"\n  [directionLinks]=\"paginator.directionLinks\"\n  [totalItems]=\"paginatorTwo.totalItems\"\n  [itemsPerPage]=\"paginator.itemsPerPage\"\n  [maxNavPages]=\"paginatorTwo.maxNavPages\"\n  (pageChange)=\"paginator.onPageChange($event)\"\n  [attr.data-role]=\"paginatorTwo.type\"\n></bb-pagination-ui>\n\n<h4>Small numbers right truncated</h4>\n<bb-pagination-ui\n  class=\"bb-pagination d-flex justify-content-start align-items-center mt-4 mb-3\"\n  [page]=\"paginatorThree.page\"\n  [boundaryLinks]=\"paginator.boundaryLinks\"\n  [directionLinks]=\"paginator.directionLinks\"\n  [totalItems]=\"paginatorThree.totalItems\"\n  [itemsPerPage]=\"paginator.itemsPerPage\"\n  [maxNavPages]=\"paginatorThree.maxNavPages\"\n  (pageChange)=\"paginator.onPageChange($event)\"\n  [attr.data-role]=\"paginatorThree.type\"\n></bb-pagination-ui>\n\n<h4>Big numbers left truncated</h4>\n<bb-pagination-ui\n  class=\"bb-pagination d-flex justify-content-start align-items-center mt-4 mb-3\"\n  [page]=\"paginatorFour.page\"\n  [boundaryLinks]=\"paginator.boundaryLinks\"\n  [directionLinks]=\"paginator.directionLinks\"\n  [totalItems]=\"paginatorFour.totalItems\"\n  [itemsPerPage]=\"paginator.itemsPerPage\"\n  [maxNavPages]=\"paginatorFour.maxNavPages\"\n  (pageChange)=\"paginator.onPageChange($event)\"\n  [attr.data-role]=\"paginatorFour.type\"\n></bb-pagination-ui>\n", components: [{ type: i1.PaginatorComponent, selector: "bb-pagination-ui", inputs: ["page", "boundaryLinks", "directionLinks", "totalItems", "itemsPerPage", "maxNavPages"], outputs: ["pageChange"] }] });
PaginationShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
], PaginationShowcaseComponent);
export { PaginationShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaginationShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-pagination-ui-showcase',
                    templateUrl: 'pagination-showcase.component.html',
                }]
        }] });
