// @backbase/ui-ang/esm2015/_example/dropdown-multi-select/showcase/dropdown-multi-select-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { DropdownMultiSelectShowcaseComponent } from './dropdown-multi-select-showcase.component';
import { DropdownMultiSelectModule } from '@backbase/ui-ang/dropdown-multi-select';
import { ReactiveFormsModule } from '@angular/forms';
import { ButtonModule } from '@backbase/ui-ang/button';
import * as i0 from "@angular/core";
export class DropdownMultiSelectShowcaseModule {
}
DropdownMultiSelectShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownMultiSelectShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
DropdownMultiSelectShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownMultiSelectShowcaseModule, declarations: [DropdownMultiSelectShowcaseComponent], imports: [CommonModule, DropdownMultiSelectModule, ReactiveFormsModule, ButtonModule] });
DropdownMultiSelectShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownMultiSelectShowcaseModule, imports: [[CommonModule, DropdownMultiSelectModule, ReactiveFormsModule, ButtonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownMultiSelectShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, DropdownMultiSelectModule, ReactiveFormsModule, ButtonModule],
                    declarations: [DropdownMultiSelectShowcaseComponent],
                    entryComponents: [DropdownMultiSelectShowcaseComponent],
                }]
        }] });
