// @backbase/identity-common-ang/esm2015/lib/ui/identity-common-ui-ang.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { ErrorCommonStateModule, LoadingIndicatorModule } from '@backbase/ui-ang';
import { ViewStateContainerComponent } from './view-state-container.component';
import { RoutableTabComponent } from './routable-tab-component';
import * as i0 from "@angular/core";
const uiModules = [ErrorCommonStateModule, LoadingIndicatorModule];
const components = [ViewStateContainerComponent, RoutableTabComponent];
export class IdentityCommonUiAngModule {
}
/** @nocollapse */ IdentityCommonUiAngModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: IdentityCommonUiAngModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ IdentityCommonUiAngModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: IdentityCommonUiAngModule, declarations: [ViewStateContainerComponent, RoutableTabComponent], imports: [CommonModule, RouterModule, ErrorCommonStateModule, LoadingIndicatorModule], exports: [ErrorCommonStateModule, LoadingIndicatorModule, ViewStateContainerComponent, RoutableTabComponent] });
/** @nocollapse */ IdentityCommonUiAngModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: IdentityCommonUiAngModule, imports: [[CommonModule, RouterModule, ...uiModules], ErrorCommonStateModule, LoadingIndicatorModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: IdentityCommonUiAngModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, RouterModule, ...uiModules],
                    declarations: [...components],
                    exports: [...uiModules, ...components],
                }]
        }] });
