// @backbase/identity-common-ang/esm2015/lib/ui/routable-tab-component.js
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "@angular/common";
export class RoutableTabComponent {
    constructor(route) {
        this.route = route;
    }
    ngOnInit() {
        var _a;
        const routeConfig = this.route.snapshot.routeConfig;
        if (routeConfig && ((_a = routeConfig.children) === null || _a === void 0 ? void 0 : _a.length)) {
            this.tabs = this.createTabsFromRouteChildren(routeConfig.children);
        }
    }
    createTabsFromRouteChildren(routeChildren) {
        return routeChildren
            .filter((routeItem) => { var _a; return routeItem.path && ((_a = routeItem.data) === null || _a === void 0 ? void 0 : _a.title); })
            .map((routeItem) => ({
            title: routeItem.data.title,
            route: routeItem.path,
            dataRole: routeItem.data.title.split(' ').join('-').toLowerCase(),
        }));
    }
}
/** @nocollapse */ RoutableTabComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: RoutableTabComponent, deps: [{ token: i1.ActivatedRoute }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ RoutableTabComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: RoutableTabComponent, selector: "bb-routable-tab-set", ngImport: i0, template: "<div class=\"bb-tab-container__tabs nav nav-tabs\">\n  <a\n    *ngFor=\"let tab of tabs\"\n    [attr.data-role]=\"tab?.dataRole\"\n    class=\"nav-link\"\n    role=\"tab\"\n    [routerLink]=\"[tab.route]\"\n    routerLinkActive=\"active\"\n  >\n    <span class=\"bb-stack__item bb-stack__item--spacing-xs\">{{ tab?.title }}</span>\n  </a>\n</div>\n\n<div class=\"bb-tab-container__content tab-content\" role=\"tabpanel\">\n  <router-outlet></router-outlet>\n</div>\n", directives: [{ type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { type: i1.RouterLinkWithHref, selector: "a[routerLink],area[routerLink]", inputs: ["routerLink", "target", "queryParams", "fragment", "queryParamsHandling", "preserveFragment", "skipLocationChange", "replaceUrl", "state", "relativeTo"] }, { type: i1.RouterLinkActive, selector: "[routerLinkActive]", inputs: ["routerLinkActiveOptions", "routerLinkActive"], exportAs: ["routerLinkActive"] }, { type: i1.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: RoutableTabComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-routable-tab-set',
                    templateUrl: './routable-tab-component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.ActivatedRoute }]; } });
