// @backbase/product-summary-ui/esm2015/src/bb-product-item-card-ui/product-item-card.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { IconModule } from '@backbase/ui-ang/icon';
import { DropdownMenuModule } from '@backbase/ui-ang/dropdown-menu';
import { PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import { AmountModule } from '@backbase/ui-ang/amount';
import { ButtonModule } from '@backbase/ui-ang/button';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { KeyboardClickModule } from '@backbase/ui-ang/keyboard-click-directive';
import { DomAttributesService } from '@backbase/product-summary-common-ang';
import { ProductItemCardComponent } from './product-item-card-ui.component';
import { ProductKindNameModule } from '../bb-product-kind-name/bb-product-kind-name.module';
import { ProductNumberModule } from '../bb-product-number/product-number.module';
import * as i0 from "@angular/core";
const uiModules = [
    IconModule,
    DropdownMenuModule,
    PaymentCardNumberModule,
    AmountModule,
    ButtonModule,
    TooltipModule,
    ProductKindNameModule,
    KeyboardClickModule,
    ProductNumberModule,
];
const components = [ProductItemCardComponent];
export class ProductItemCardModule {
}
ProductItemCardModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCardModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductItemCardModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCardModule, declarations: [ProductItemCardComponent], imports: [CommonModule, IconModule,
        DropdownMenuModule,
        PaymentCardNumberModule,
        AmountModule,
        ButtonModule,
        TooltipModule,
        ProductKindNameModule,
        KeyboardClickModule,
        ProductNumberModule], exports: [ProductItemCardComponent, IconModule,
        DropdownMenuModule,
        PaymentCardNumberModule,
        AmountModule,
        ButtonModule,
        TooltipModule,
        ProductKindNameModule,
        KeyboardClickModule,
        ProductNumberModule] });
ProductItemCardModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCardModule, providers: [DomAttributesService], imports: [[CommonModule, ...uiModules], IconModule,
        DropdownMenuModule,
        PaymentCardNumberModule,
        AmountModule,
        ButtonModule,
        TooltipModule,
        ProductKindNameModule,
        KeyboardClickModule,
        ProductNumberModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCardModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules],
                    providers: [DomAttributesService],
                    declarations: [...components],
                    exports: [...components, ...uiModules],
                }]
        }] });
