// @backbase/loans-collection-ui-ang/esm2015/lib/loan-info/loan-info-documents-section/loan-info-documents-section.component.js
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import '@angular/localize/init';
import { LoansHttpService } from '@backbase/data-ang/loans';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/loans";
import * as i2 from "@backbase/ui-ang/icon";
import * as i3 from "@backbase/ui-ang/ellipsis";
import * as i4 from "@angular/common";
import * as i5 from "@backbase/ui-ang/button";
import * as i6 from "../loan-info-custom-link/loan-info-custom-link.directive";
export class LoanInfoDocumentsSectionComponent {
    constructor(loansHttpService) {
        this.loansHttpService = loansHttpService;
        this.defaultDocumentName = $localize `:Loan info document default name@@loans.loan-info.document.default-name:Document`;
    }
    /**
     * Loan item
     */
    set loan(value) {
        this.updateLoanDocuments(value);
    }
    /**
     * Document name tag id
     *
     * @param id
     */
    getDocumentNameId(id) {
        return `bb-loan-info-document-${id}`;
    }
    /**
     * Updates loan documents collection
     *
     * @private
     */
    updateLoanDocuments(loan) {
        var _a;
        const loanId = loan.id;
        let id = 1;
        this.loanDocuments = (_a = loan.documents) === null || _a === void 0 ? void 0 : _a.reduce((accum, { documentId, contentType, name }) => {
            if (documentId) {
                const isPreviewAvailable = contentType === 'application/pdf';
                accum.push({
                    name: (name === null || name === void 0 ? void 0 : name.trim()) || `${this.defaultDocumentName} ${id++}`,
                    downloadUrl: this.getLoanDocumentUrl(loanId, documentId),
                    previewUrl: isPreviewAvailable ? this.getLoanDocumentUrl(loanId, documentId, true) : undefined,
                });
            }
            return accum;
        }, []);
    }
    /**
     * Construct formatted url for preview or download
     *
     * @private
     */
    getLoanDocumentUrl(loanId, documentId, inline) {
        return this.loansHttpService.downloadDocumentByLoanIdAndDocumentIdUrl({
            loanId,
            documentId,
            inline,
        });
    }
}
/** @nocollapse */ LoanInfoDocumentsSectionComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoDocumentsSectionComponent, deps: [{ token: i1.LoansHttpService }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ LoanInfoDocumentsSectionComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: LoanInfoDocumentsSectionComponent, selector: "bb-loan-info-documents-section", inputs: { loan: "loan" }, ngImport: i0, template: "<ul class=\"bb-list bb-list--no-separator bb-list--density-none\">\n  <li\n    *ngFor=\"let document of loanDocuments; let index = index\"\n    class=\"bb-list__item bb-stack bb-loans-loan-info-documents-section\"\n    [class.bb-loans-loan-info-documents-section--no-preview]=\"!document.previewUrl\"\n    data-role=\"loan-info-document\"\n  >\n    <bb-icon-ui class=\"bb-stack__item bb-stack__item--spacing-sm\" name=\"description\" color=\"neutral-dark\"></bb-icon-ui>\n    <bb-ellipsis-ui\n      [id]=\"getDocumentNameId(index)\"\n      class=\"bb-stack__item bb-stack__item--fill bb-loans-loan-info-documents-section__name\"\n      [text]=\"document.name\"\n      tooltipTriggers=\"hover click\"\n      data-role=\"loan-info-document__name\"\n    ></bb-ellipsis-ui>\n    <button\n      *ngIf=\"document.previewUrl as previewUrl\"\n      class=\"bb-stack__item d-print-none\"\n      bbButton\n      color=\"link\"\n      type=\"button\"\n      [circle]=\"true\"\n      [bbLoanInfoCustomLink]=\"previewUrl\"\n      bbLoanInfoCustomLinkTarget=\"_blank\"\n      [attr.aria-describedby]=\"getDocumentNameId(index)\"\n      i18n-aria-label=\"Loan info preview document button aria label@@loans.loan-info.preview-document.button.aria-label\"\n      aria-label=\"View online\"\n      data-role=\"loan-info-document__preview-btn\"\n    >\n      <bb-icon-ui name=\"open-in-new\"></bb-icon-ui>\n    </button>\n    <button\n      class=\"bb-stack__item d-print-none\"\n      bbButton\n      color=\"link\"\n      type=\"button\"\n      [circle]=\"true\"\n      [bbLoanInfoCustomLink]=\"document.downloadUrl\"\n      [bbLoanInfoCustomLinkDownloadName]=\"document.name\"\n      [attr.aria-describedby]=\"getDocumentNameId(index)\"\n      i18n-aria-label=\"\n        Loan info download document button aria label@@loans.loan-info.download-document.button.aria-label\"\n      aria-label=\"Download\"\n      data-role=\"loan-info-document__download-btn\"\n    >\n      <bb-icon-ui name=\"get-app\"></bb-icon-ui>\n    </button>\n  </li>\n</ul>\n", components: [{ type: i2.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i3.EllipsisComponent, selector: "bb-ellipsis-ui", inputs: ["text", "tooltipTriggers", "lineClamp"] }], directives: [{ type: i4.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { type: i4.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i5.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }, { type: i6.LoanInfoCustomLinkDirective, selector: "[bbLoanInfoCustomLink]", inputs: ["bbLoanInfoCustomLink", "bbLoanInfoCustomLinkDownloadName", "bbLoanInfoCustomLinkTarget"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoDocumentsSectionComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-loan-info-documents-section',
                    templateUrl: './loan-info-documents-section.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.LoansHttpService }]; }, propDecorators: { loan: [{
                type: Input
            }] } });
