// @backbase/loans-collection-ui-ang/esm2015/lib/loan-info/loan-info-documents-section/loan-info-documents-section.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ButtonModule } from '@backbase/ui-ang/button';
import { EllipsisModule } from '@backbase/ui-ang/ellipsis';
import { IconModule } from '@backbase/ui-ang/icon';
import { LoanInfoDocumentsSectionComponent } from './loan-info-documents-section.component';
import { LoanInfoCustomLinkModule } from '../loan-info-custom-link/loan-info-custom-link.module';
import * as i0 from "@angular/core";
export class LoanInfoDocumentsSectionModule {
}
/** @nocollapse */ LoanInfoDocumentsSectionModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoDocumentsSectionModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanInfoDocumentsSectionModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoDocumentsSectionModule, declarations: [LoanInfoDocumentsSectionComponent], imports: [CommonModule, EllipsisModule, ButtonModule, IconModule, LoanInfoCustomLinkModule], exports: [LoanInfoDocumentsSectionComponent] });
/** @nocollapse */ LoanInfoDocumentsSectionModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoDocumentsSectionModule, imports: [[CommonModule, EllipsisModule, ButtonModule, IconModule, LoanInfoCustomLinkModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoDocumentsSectionModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanInfoDocumentsSectionComponent],
                    imports: [CommonModule, EllipsisModule, ButtonModule, IconModule, LoanInfoCustomLinkModule],
                    exports: [LoanInfoDocumentsSectionComponent],
                }]
        }] });
