// @backbase/ui-ang/esm2015/product-item-savings-account/product-item-savings-account.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { CustomProductItemSavingsAccountDetailsFormatterDirective, CustomProductItemSavingsAccountMainSectionDirective, CustomProductItemSavingsAccountTitleDirective, ProductItemSavingsAccountComponent, } from './product-item-savings-account.component';
import { AmountModule } from '@backbase/ui-ang/amount';
import { PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import { HighlightModule } from '@backbase/ui-ang/highlight';
import { IconModule } from '@backbase/ui-ang/icon';
import * as i0 from "@angular/core";
export class ProductItemSavingsAccountModule {
}
ProductItemSavingsAccountModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemSavingsAccountModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductItemSavingsAccountModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemSavingsAccountModule, declarations: [ProductItemSavingsAccountComponent,
        CustomProductItemSavingsAccountTitleDirective,
        CustomProductItemSavingsAccountDetailsFormatterDirective,
        CustomProductItemSavingsAccountMainSectionDirective], imports: [CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule], exports: [ProductItemSavingsAccountComponent,
        CustomProductItemSavingsAccountTitleDirective,
        CustomProductItemSavingsAccountDetailsFormatterDirective,
        CustomProductItemSavingsAccountMainSectionDirective] });
ProductItemSavingsAccountModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemSavingsAccountModule, imports: [[CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemSavingsAccountModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule],
                    declarations: [
                        ProductItemSavingsAccountComponent,
                        CustomProductItemSavingsAccountTitleDirective,
                        CustomProductItemSavingsAccountDetailsFormatterDirective,
                        CustomProductItemSavingsAccountMainSectionDirective,
                    ],
                    exports: [
                        ProductItemSavingsAccountComponent,
                        CustomProductItemSavingsAccountTitleDirective,
                        CustomProductItemSavingsAccountDetailsFormatterDirective,
                        CustomProductItemSavingsAccountMainSectionDirective,
                    ],
                }]
        }] });
