// @backbase/consent-request-widget-ang/esm2015/src/consent-request-widget.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { ConsentUiModule } from '@backbase/consent-ui';
import { ReactiveFormsModule } from '@angular/forms';
import { ButtonModule } from '@backbase/ui-ang/button';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { ProductItemBasicAccountModule } from '@backbase/ui-ang/product-item-basic-account';
import { AlertModule } from '@backbase/ui-ang/alert';
import { BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import { NotificationModule } from '@backbase/ui-ang/notification';
import { AccountSelectorModule } from '@backbase/ui-ang/account-selector';
import { AmountModule } from '@backbase/ui-ang/amount';
import { SearchBoxModule } from '@backbase/ui-ang/search-box';
import { HighlightModule } from '@backbase/ui-ang/highlight';
import { ConsentRequestContainerComponent, ConsentRequestWidgetCustomizableDirective, } from './consent-request-widget/consent-request-container/consent-request-container.component';
import { ConsentRequestErrorComponent } from './consent-request-widget/consent-request-error/consent-request-error.component';
import { ConsentRequestPermissionsInformationComponent, ConsentRequestPermissionsInformationCustomizableDirective, } from './consent-request-widget/consent-request-permissions-information/consent-request-permissions-information.component';
import { ConsentRequestAccountSelectorComponent, ConsentRequestAccountSelectorCustomizableDirective, } from './consent-request-widget/consent-request-account-selector/consent-request-account-selector.component';
import { ConsentRequestTppInformationComponent, ConsentRequestTppInformationCustomizableDirective, } from './consent-request-widget/consent-request-tpp-information/consent-request-tpp-information.component';
import { ConsentRequestWidgetComponent } from './consent-request-widget.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const components = [
    ConsentRequestWidgetComponent,
    ConsentRequestErrorComponent,
    ConsentRequestPermissionsInformationComponent,
    ConsentRequestTppInformationComponent,
    ConsentRequestAccountSelectorComponent,
    ConsentRequestContainerComponent,
];
const directives = [
    ConsentRequestWidgetCustomizableDirective,
    ConsentRequestPermissionsInformationCustomizableDirective,
    ConsentRequestTppInformationCustomizableDirective,
    ConsentRequestAccountSelectorCustomizableDirective,
];
const uiModules = [
    ConsentUiModule,
    ButtonModule,
    EmptyStateModule,
    ProductItemBasicAccountModule,
    AccountSelectorModule,
    AlertModule,
    LoadingIndicatorModule,
    BbDatePipeModule,
    NotificationModule,
    AmountModule,
    SearchBoxModule,
    HighlightModule,
];
export class ConsentRequestWidgetModule {
}
ConsentRequestWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentRequestWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ConsentRequestWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentRequestWidgetModule, declarations: [ConsentRequestWidgetComponent,
        ConsentRequestErrorComponent,
        ConsentRequestPermissionsInformationComponent,
        ConsentRequestTppInformationComponent,
        ConsentRequestAccountSelectorComponent,
        ConsentRequestContainerComponent, ConsentRequestWidgetCustomizableDirective,
        ConsentRequestPermissionsInformationCustomizableDirective,
        ConsentRequestTppInformationCustomizableDirective,
        ConsentRequestAccountSelectorCustomizableDirective], imports: [CommonModule, ConsentUiModule,
        ButtonModule,
        EmptyStateModule,
        ProductItemBasicAccountModule,
        AccountSelectorModule,
        AlertModule,
        LoadingIndicatorModule,
        BbDatePipeModule,
        NotificationModule,
        AmountModule,
        SearchBoxModule,
        HighlightModule, i1.BackbaseCoreModule, ReactiveFormsModule], exports: [ConsentRequestWidgetComponent,
        ConsentRequestErrorComponent,
        ConsentRequestPermissionsInformationComponent,
        ConsentRequestTppInformationComponent,
        ConsentRequestAccountSelectorComponent,
        ConsentRequestContainerComponent, ConsentRequestWidgetCustomizableDirective,
        ConsentRequestPermissionsInformationCustomizableDirective,
        ConsentRequestTppInformationCustomizableDirective,
        ConsentRequestAccountSelectorCustomizableDirective, ConsentUiModule,
        ButtonModule,
        EmptyStateModule,
        ProductItemBasicAccountModule,
        AccountSelectorModule,
        AlertModule,
        LoadingIndicatorModule,
        BbDatePipeModule,
        NotificationModule,
        AmountModule,
        SearchBoxModule,
        HighlightModule] });
ConsentRequestWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentRequestWidgetModule, imports: [[
            CommonModule,
            ...uiModules,
            BackbaseCoreModule.withConfig({
                classMap: { ConsentRequestWidgetComponent },
            }),
            ReactiveFormsModule,
        ], ConsentUiModule,
        ButtonModule,
        EmptyStateModule,
        ProductItemBasicAccountModule,
        AccountSelectorModule,
        AlertModule,
        LoadingIndicatorModule,
        BbDatePipeModule,
        NotificationModule,
        AmountModule,
        SearchBoxModule,
        HighlightModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentRequestWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        ...uiModules,
                        BackbaseCoreModule.withConfig({
                            classMap: { ConsentRequestWidgetComponent },
                        }),
                        ReactiveFormsModule,
                    ],
                    declarations: [...components, ...directives],
                    exports: [...components, ...directives, ...uiModules],
                }]
        }] });
