// @backbase/consent-request-widget-ang/esm2015/src/consent-request-widget.component.js
import { Component, Output, EventEmitter, Input } from '@angular/core';
import { ConsentRequestPreferencesService } from './consent-request-widget/consent-request-preferences.service';
import { ConsentRequestService } from './consent-request-widget/consent-request.service';
import * as i0 from "@angular/core";
import * as i1 from "./consent-request-widget/consent-request-preferences.service";
import * as i2 from "./consent-request-widget/consent-request-container/consent-request-container.component";
export class ConsentRequestWidgetComponent {
    constructor(consentRequestPreferencesService) {
        this.consentRequestPreferencesService = consentRequestPreferencesService;
        /**
         * Emitter that fires when allowing consent
         */
        this.allowConsentRequestRedirectUrl = new EventEmitter();
        /**
         * Emitter that fires when rejecting consent
         */
        this.rejectConsentRequestRedirectUrl = new EventEmitter();
    }
    /**
     * Allow multi account selection
     */
    set multiAccountSelection(value) {
        this.consentRequestPreferencesService.multiAccountSelectionInput.next(value);
    }
    redirectToAllowUrl(url) {
        this.allowConsentRequestRedirectUrl.next(url);
    }
    redirectToRejectUrl(url) {
        this.rejectConsentRequestRedirectUrl.next(url);
    }
}
ConsentRequestWidgetComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentRequestWidgetComponent, deps: [{ token: i1.ConsentRequestPreferencesService }], target: i0.ɵɵFactoryTarget.Component });
ConsentRequestWidgetComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: ConsentRequestWidgetComponent, selector: "bb-consent-request-widget", inputs: { multiAccountSelection: "multiAccountSelection" }, outputs: { allowConsentRequestRedirectUrl: "allowConsentRequestRedirectUrl", rejectConsentRequestRedirectUrl: "rejectConsentRequestRedirectUrl" }, providers: [ConsentRequestService, ConsentRequestPreferencesService], ngImport: i0, template: `
    <bb-consent-request-container
      (redirectToAllowUrl)="redirectToAllowUrl($event)"
      (redirectToRejectUrl)="redirectToRejectUrl($event)"></bb-consent-request-container>
  `, isInline: true, components: [{ type: i2.ConsentRequestContainerComponent, selector: "bb-consent-request-container", outputs: ["redirectToAllowUrl", "redirectToRejectUrl"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentRequestWidgetComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-consent-request-widget',
                    template: `
    <bb-consent-request-container
      (redirectToAllowUrl)="redirectToAllowUrl($event)"
      (redirectToRejectUrl)="redirectToRejectUrl($event)"></bb-consent-request-container>
  `,
                    providers: [ConsentRequestService, ConsentRequestPreferencesService],
                }]
        }], ctorParameters: function () { return [{ type: i1.ConsentRequestPreferencesService }]; }, propDecorators: { multiAccountSelection: [{
                type: Input
            }], allowConsentRequestRedirectUrl: [{
                type: Output
            }], rejectConsentRequestRedirectUrl: [{
                type: Output
            }] } });
