// @backbase/tradingfx-http-ang/esm2015/api/trading.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { FOREX_BASE_PATH } from '../variables';
import { ForexConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class TradingHttpService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new ForexConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    getCurrencypair(requestParameters, observe = 'body', reportProgress = false, options) {
        const _query = requestParameters["query"];
        const _sort = requestParameters["sort"];
        const _favorite = requestParameters["favorite"];
        const _from = requestParameters["from"];
        const _cursor = requestParameters["cursor"];
        const _size = requestParameters["size"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_query !== undefined && _query !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _query, 'query');
        }
        if (_sort !== undefined && _sort !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _sort, 'sort');
        }
        if (_favorite !== undefined && _favorite !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _favorite, 'favorite');
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_cursor !== undefined && _cursor !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _cursor, 'cursor');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/trading/currencypair`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getCurrencypairUrl(requestParameters) {
        const _query = requestParameters["query"];
        const _sort = requestParameters["sort"];
        const _favorite = requestParameters["favorite"];
        const _from = requestParameters["from"];
        const _cursor = requestParameters["cursor"];
        const _size = requestParameters["size"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_query !== undefined && _query !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _query, 'query');
        }
        if (_sort !== undefined && _sort !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _sort, 'sort');
        }
        if (_favorite !== undefined && _favorite !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _favorite, 'favorite');
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_cursor !== undefined && _cursor !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _cursor, 'cursor');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/trading/currencypair${queryString ? `?${queryString}` : ''}`;
    }
    getCurrencypairIdByCurrencypairId(requestParameters, observe = 'body', reportProgress = false, options) {
        const _currencypairId = requestParameters["currencypairId"];
        if (_currencypairId === null || _currencypairId === undefined) {
            throw new Error('Required parameter currencypairId was null or undefined when calling getCurrencypairIdByCurrencypairId.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/trading/currencypair/${encodeURIComponent(String(_currencypairId))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getCurrencypairIdByCurrencypairIdUrl(requestParameters) {
        const _currencypairId = requestParameters["currencypairId"];
        if (_currencypairId === null || _currencypairId === undefined) {
            throw new Error('Required parameter currencypairId was null or undefined when calling getCurrencypairIdByCurrencypairId.');
        }
        return `${this.configuration.basePath}/client-api/v2/trading/currencypair/${encodeURIComponent(String(_currencypairId))}`;
    }
    getCurrencyrate(requestParameters, observe = 'body', reportProgress = false, options) {
        const _currencypair = requestParameters["currencypair"];
        const _sort = requestParameters["sort"];
        const _favorite = requestParameters["favorite"];
        const _from = requestParameters["from"];
        const _cursor = requestParameters["cursor"];
        const _size = requestParameters["size"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_currencypair) {
            _currencypair.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'currencypair');
            });
        }
        if (_sort !== undefined && _sort !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _sort, 'sort');
        }
        if (_favorite !== undefined && _favorite !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _favorite, 'favorite');
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_cursor !== undefined && _cursor !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _cursor, 'cursor');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/trading/currencyrate`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getCurrencyrateUrl(requestParameters) {
        const _currencypair = requestParameters["currencypair"];
        const _sort = requestParameters["sort"];
        const _favorite = requestParameters["favorite"];
        const _from = requestParameters["from"];
        const _cursor = requestParameters["cursor"];
        const _size = requestParameters["size"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_currencypair) {
            _currencypair.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'currencypair');
            });
        }
        if (_sort !== undefined && _sort !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _sort, 'sort');
        }
        if (_favorite !== undefined && _favorite !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _favorite, 'favorite');
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_cursor !== undefined && _cursor !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _cursor, 'cursor');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/trading/currencyrate${queryString ? `?${queryString}` : ''}`;
    }
    postCurrencyPairFavoriteByCurrencypairId(requestParameters, observe = 'body', reportProgress = false, options) {
        const _currencypairId = requestParameters["currencypairId"];
        if (_currencypairId === null || _currencypairId === undefined) {
            throw new Error('Required parameter currencypairId was null or undefined when calling postCurrencyPairFavoriteByCurrencypairId.');
        }
        const _traidingCurrencypairFavorite = requestParameters["traidingCurrencypairFavorite"];
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v2/trading/currencypair/${encodeURIComponent(String(_currencypairId))}/favorite`, _traidingCurrencypairFavorite, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    postCurrencyPairFavoriteByCurrencypairIdUrl(requestParameters) {
        const _currencypairId = requestParameters["currencypairId"];
        if (_currencypairId === null || _currencypairId === undefined) {
            throw new Error('Required parameter currencypairId was null or undefined when calling postCurrencyPairFavoriteByCurrencypairId.');
        }
        const _traidingCurrencypairFavorite = requestParameters["traidingCurrencypairFavorite"];
        return `${this.configuration.basePath}/client-api/v2/trading/currencypair/${encodeURIComponent(String(_currencypairId))}/favorite`;
    }
    postCurrencyPairTransactionByCurrencypairId(requestParameters, observe = 'body', reportProgress = false, options) {
        const _currencypairId = requestParameters["currencypairId"];
        if (_currencypairId === null || _currencypairId === undefined) {
            throw new Error('Required parameter currencypairId was null or undefined when calling postCurrencyPairTransactionByCurrencypairId.');
        }
        const _tradingCurrencyPairItemPost = requestParameters["tradingCurrencyPairItemPost"];
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v2/trading/currencypair/${encodeURIComponent(String(_currencypairId))}`, _tradingCurrencyPairItemPost, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    postCurrencyPairTransactionByCurrencypairIdUrl(requestParameters) {
        const _currencypairId = requestParameters["currencypairId"];
        if (_currencypairId === null || _currencypairId === undefined) {
            throw new Error('Required parameter currencypairId was null or undefined when calling postCurrencyPairTransactionByCurrencypairId.');
        }
        const _tradingCurrencyPairItemPost = requestParameters["tradingCurrencyPairItemPost"];
        return `${this.configuration.basePath}/client-api/v2/trading/currencypair/${encodeURIComponent(String(_currencypairId))}`;
    }
}
TradingHttpService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: TradingHttpService, deps: [{ token: i1.HttpClient }, { token: FOREX_BASE_PATH, optional: true }, { token: i2.ForexConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
TradingHttpService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: TradingHttpService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: TradingHttpService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [FOREX_BASE_PATH]
                }] }, { type: i2.ForexConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
