// @backbase/notifications-ang/esm2015/lib/notifications-badge-ang/src/notifications-badge.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NotificationsCommonModule } from '@backbase/notifications-common-ang';
import { NotificationsBadgeWidgetModule } from '@backbase/notifications-badge-widget-ang';
import { NotificationsBadgeComponent } from './components/notifications-badge.component';
import * as i0 from "@angular/core";
const modules = [NotificationsCommonModule, NotificationsBadgeWidgetModule];
export class NotificationsBadgeModule {
}
/** @nocollapse */ NotificationsBadgeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsBadgeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ NotificationsBadgeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsBadgeModule, declarations: [NotificationsBadgeComponent], imports: [CommonModule, NotificationsCommonModule, NotificationsBadgeWidgetModule], exports: [NotificationsBadgeComponent] });
/** @nocollapse */ NotificationsBadgeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsBadgeModule, imports: [[CommonModule, ...modules]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsBadgeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...modules],
                    declarations: [NotificationsBadgeComponent],
                    exports: [NotificationsBadgeComponent],
                }]
        }] });
