// @backbase/loans-collection-ui-ang/esm2015/lib/loan-info/loan-info-custom-link/loan-info-custom-link.directive.js
import { Directive, HostListener, Inject, Input, Renderer2 } from '@angular/core';
import { DOCUMENT } from '@angular/common';
import * as i0 from "@angular/core";
// @dynamic
export class LoanInfoCustomLinkDirective {
    constructor(document, renderer) {
        this.document = document;
        this.renderer = renderer;
        /**
         * Link target
         *
         * @default '_self'
         */
        this.target = '_self';
    }
    simulateLinkClick() {
        const linkEl = this.renderer.createElement('a');
        this.renderer.setAttribute(linkEl, 'href', this.linkUrl);
        this.renderer.setAttribute(linkEl, 'target', this.target);
        if (this.downloadName) {
            this.renderer.setAttribute(linkEl, 'download', this.downloadName);
        }
        this.renderer.appendChild(this.document.body, linkEl);
        linkEl.click();
        this.renderer.removeChild(this.document.body, linkEl);
    }
}
/** @nocollapse */ LoanInfoCustomLinkDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoCustomLinkDirective, deps: [{ token: DOCUMENT }, { token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Directive });
/** @nocollapse */ LoanInfoCustomLinkDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: LoanInfoCustomLinkDirective, selector: "[bbLoanInfoCustomLink]", inputs: { linkUrl: ["bbLoanInfoCustomLink", "linkUrl"], downloadName: ["bbLoanInfoCustomLinkDownloadName", "downloadName"], target: ["bbLoanInfoCustomLinkTarget", "target"] }, host: { listeners: { "click": "simulateLinkClick()", "keydown.enter": "simulateLinkClick()", "keydown.space": "simulateLinkClick()" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoCustomLinkDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbLoanInfoCustomLink]',
                }]
        }], ctorParameters: function () { return [{ type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }, { type: i0.Renderer2 }]; }, propDecorators: { linkUrl: [{
                type: Input,
                args: ['bbLoanInfoCustomLink']
            }], downloadName: [{
                type: Input,
                args: ['bbLoanInfoCustomLinkDownloadName']
            }], target: [{
                type: Input,
                args: ['bbLoanInfoCustomLinkTarget']
            }], simulateLinkClick: [{
                type: HostListener,
                args: ['click']
            }, {
                type: HostListener,
                args: ['keydown.enter']
            }, {
                type: HostListener,
                args: ['keydown.space']
            }] } });
