// @backbase/loans-collection-ui-ang/esm2015/lib/loan-info/loan-info-custom-link/loan-info-custom-link.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { LoanInfoCustomLinkDirective } from './loan-info-custom-link.directive';
import * as i0 from "@angular/core";
export class LoanInfoCustomLinkModule {
}
/** @nocollapse */ LoanInfoCustomLinkModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoCustomLinkModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanInfoCustomLinkModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoCustomLinkModule, declarations: [LoanInfoCustomLinkDirective], imports: [CommonModule], exports: [LoanInfoCustomLinkDirective] });
/** @nocollapse */ LoanInfoCustomLinkModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoCustomLinkModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoCustomLinkModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanInfoCustomLinkDirective],
                    imports: [CommonModule],
                    exports: [LoanInfoCustomLinkDirective],
                }]
        }] });
