// @backbase/ui-ang/esm2015/_example/period-selector/examples/period-selector.example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/period-selector";
const dateTimeToDate = (isoDateTime) => isoDateTime.replace(/T.+/, '');
export class PeriodSelectorExampleComponent {
    constructor() {
        this.startDate = new Date('2018-08-01');
        this.endDate = new Date('2018-08-31');
        this.periodFormatters = {
            start: (date) => dateTimeToDate(date.toISOString()),
            end: (date) => `End: ${dateTimeToDate(date.toISOString())}`,
        };
    }
}
PeriodSelectorExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PeriodSelectorExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
PeriodSelectorExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: PeriodSelectorExampleComponent, selector: "bb-period-selector-example-ui", ngImport: i0, template: "<bb-period-selector-ui\n  periodSize=\"month\"\n  [periodStart]=\"startDate\"\n  [periodEnd]=\"endDate\"\n  [periodFormatters]=\"periodFormatters\"\n></bb-period-selector-ui>\n", components: [{ type: i1.PeriodSelectorComponent, selector: "bb-period-selector-ui", inputs: ["periodStart", "periodEnd", "periodFormatters", "period"], outputs: ["periodChange"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PeriodSelectorExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-period-selector-example-ui',
                    templateUrl: './period-selector.example.component.html',
                }]
        }] });
