// @backbase/payment-orders-ang/esm2015/lib/ui/components/table-header-checkbox/table-header-checkbox.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TableHeaderCheckboxComponent } from './table-header-checkbox.component';
import * as i0 from "@angular/core";
export class TableHeaderCheckboxModule {
}
/** @nocollapse */ TableHeaderCheckboxModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableHeaderCheckboxModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ TableHeaderCheckboxModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableHeaderCheckboxModule, declarations: [TableHeaderCheckboxComponent], imports: [CommonModule], exports: [TableHeaderCheckboxComponent] });
/** @nocollapse */ TableHeaderCheckboxModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableHeaderCheckboxModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableHeaderCheckboxModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [TableHeaderCheckboxComponent],
                    imports: [CommonModule],
                    exports: [TableHeaderCheckboxComponent],
                }]
        }] });
