// @backbase/notifications-popups-widget-ang/esm2015/lib/components/notifications-popups-list/notifications-popups-list.component.js
import { Component, Directive } from '@angular/core';
import { BbTemplate } from '@backbase/foundation-ang/core/';
import { map } from 'rxjs/operators';
import { NotificationsPopupsDataService } from '../../services/notifications-popups-data.service';
import * as i0 from "@angular/core";
import * as i1 from "../../services/notifications-popups-data.service";
import * as i2 from "../notifications-popups-item/notifications-popups-item.component";
import * as i3 from "@angular/common";
export const DEFAULT_NOTIFICATION_TIMEOUT = 5;
export class NotificationsPopupsListComponent {
    constructor(dataService) {
        this.dataService = dataService;
        /**
         * Notification popups stream data in reversed order
         */
        this.popups$ = this.dataService.popupsNotificationsStream$.pipe(map(this.parsePopupsData));
    }
    /**
     * Method to parse popups stream data in reversed order
     * for correct rendering by show notificaton service
     */
    parsePopupsData(array) {
        return array.reverse();
    }
    /**
     * Method to get close notification timeout value for each popup
     *
     * @param index List item index
     * @param length List length
     * @returns Number of seconds before notification popup close
     */
    getNotificationTimeoutByIndex(index, length) {
        return DEFAULT_NOTIFICATION_TIMEOUT + length - index - 1;
    }
}
/** @nocollapse */ NotificationsPopupsListComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsPopupsListComponent, deps: [{ token: i1.NotificationsPopupsDataService }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ NotificationsPopupsListComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: NotificationsPopupsListComponent, selector: "bb-notifications-popups-list", ngImport: i0, template: "<ng-container\n  bbNotificationsPopupsListCustomizable\n  [bbHostRef]=\"this\"\n  [bbTemplateContext]=\"popups$ | async\"\n></ng-container>\n\n<ng-template bbNotificationsPopupsListCustomizable let-hostRef let-notifications=\"context\">\n  <bb-notifications-popups-item\n    *ngFor=\"let notification of notifications; index as i\"\n    [notification]=\"notification\"\n    [notificationTimeout]=\"hostRef.getNotificationTimeoutByIndex(i, notifications.length)\"\n  >\n  </bb-notifications-popups-item>\n</ng-template>\n", components: [{ type: i0.forwardRef(function () { return i2.NotificationsPopupsItemComponent; }), selector: "bb-notifications-popups-item", inputs: ["notification", "notificationTimeout"] }], directives: [{ type: i0.forwardRef(function () { return NotificationsPopupsListCustomizableDirective; }), selector: "[bbNotificationsPopupsListCustomizable]" }, { type: i0.forwardRef(function () { return i3.NgForOf; }), selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }], pipes: { "async": i0.forwardRef(function () { return i3.AsyncPipe; }) } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsPopupsListComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-notifications-popups-list',
                    templateUrl: './notifications-popups-list.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.NotificationsPopupsDataService }]; } });
export class NotificationsPopupsListCustomizableDirective extends BbTemplate {
}
/** @nocollapse */ NotificationsPopupsListCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsPopupsListCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
/** @nocollapse */ NotificationsPopupsListCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: NotificationsPopupsListCustomizableDirective, selector: "[bbNotificationsPopupsListCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsPopupsListCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbNotificationsPopupsListCustomizable]',
                }]
        }] });
