// @backbase/ui-ang/esm2015/_example/item-log/examples/item-log.example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/item-log";
export class ItemLogExampleComponent {
    constructor() {
        this.logRecords = [
            {
                userFullName: 'Green Goblet',
                status: 'REJECTED',
                createdAt: '2018-01-23T13:44:02Z',
                comment: 'Limit exceeded',
            },
            {
                userFullName: 'Robin Green',
                status: 'APPROVED',
                createdAt: '2018-01-22T13:44:02Z',
            },
        ];
    }
}
ItemLogExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ItemLogExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ItemLogExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ItemLogExampleComponent, selector: "bb-item-log-example-ui", ngImport: i0, template: "<bb-item-log-ui\n  [records]=\"logRecords\"\n  [isPending]=\"true\"\n></bb-item-log-ui>\n", components: [{ type: i1.ItemLogComponent, selector: "bb-item-log-ui", inputs: ["isPending", "records"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ItemLogExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-item-log-example-ui',
                    templateUrl: './item-log.example.component.html',
                }]
        }] });
