// @backbase/account-statement-common-ang/esm2015/src/account-statement-common.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { AccountStatementAccountSelectorComponent, AccountStatementAccountSelectorCustomizableDirective, } from './components/account-statement-account-selector/account-statement-account-selector.component';
import { AccountStatementDownloadsComponent, AccountStatementDownloadsCustomizableDirective, } from './components/account-statement-downloads/account-statement-downloads.component';
import { AccountStatementViewOnlineComponent } from './components/account-statement-view-online/account-statement-view-online.component';
import { AccountStatementDownloadDocDirective } from './directives/account-statement-download-doc/account-statement-download-doc.directive';
import { AccountStatementDataService } from './services/account-statement-data.service';
import { AccountStatementDomService } from './services/account-statement-dom-service';
import { AccountStatementPaginationComponent } from './components/account-statement-pagination/account-statement-pagination.component';
import { AccountSelectorModule } from '@backbase/ui-ang/account-selector';
import { ButtonModule } from '@backbase/ui-ang/button';
import { DropdownMenuModule } from '@backbase/ui-ang/dropdown-menu';
import { IconModule } from '@backbase/ui-ang/icon';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { PaginationModule } from '@backbase/ui-ang/pagination';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import * as i0 from "@angular/core";
const components = [
    AccountStatementViewOnlineComponent,
    AccountStatementDownloadsComponent,
    AccountStatementAccountSelectorComponent,
    AccountStatementPaginationComponent,
];
const directives = [
    AccountStatementDownloadDocDirective,
    AccountStatementDownloadsCustomizableDirective,
    AccountStatementAccountSelectorCustomizableDirective,
];
const uiModules = [
    ButtonModule,
    DropdownMenuModule,
    IconModule,
    TooltipModule,
    AccountSelectorModule,
    LoadButtonModule,
    PaginationModule,
];
export class AccountStatementCommonModule {
}
AccountStatementCommonModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementCommonModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AccountStatementCommonModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementCommonModule, declarations: [AccountStatementViewOnlineComponent,
        AccountStatementDownloadsComponent,
        AccountStatementAccountSelectorComponent,
        AccountStatementPaginationComponent, AccountStatementDownloadDocDirective,
        AccountStatementDownloadsCustomizableDirective,
        AccountStatementAccountSelectorCustomizableDirective], imports: [CommonModule, ButtonModule,
        DropdownMenuModule,
        IconModule,
        TooltipModule,
        AccountSelectorModule,
        LoadButtonModule,
        PaginationModule], exports: [AccountStatementViewOnlineComponent,
        AccountStatementDownloadsComponent,
        AccountStatementAccountSelectorComponent,
        AccountStatementPaginationComponent, AccountStatementDownloadDocDirective,
        AccountStatementDownloadsCustomizableDirective,
        AccountStatementAccountSelectorCustomizableDirective, ButtonModule,
        DropdownMenuModule,
        IconModule,
        TooltipModule,
        AccountSelectorModule,
        LoadButtonModule,
        PaginationModule] });
AccountStatementCommonModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementCommonModule, providers: [AccountStatementDataService, AccountStatementDomService], imports: [[CommonModule, ...uiModules], ButtonModule,
        DropdownMenuModule,
        IconModule,
        TooltipModule,
        AccountSelectorModule,
        LoadButtonModule,
        PaginationModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementCommonModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules],
                    declarations: [...components, ...directives],
                    providers: [AccountStatementDataService, AccountStatementDomService],
                    exports: [...components, ...directives, ...uiModules],
                }]
        }] });
