// @backbase/product-summary-common-ang/esm2015/src/pipes/validation/validation.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { IsDefinedPipe } from './is-defined.pipe';
import { IsInArrayPipe } from './is-in-array.pipe';
import { IsPercentValue } from './is-percent-value.pipe';
import { IsProductKindPipe } from './is-product-kind.pipe';
import * as i0 from "@angular/core";
const pipes = [IsDefinedPipe, IsInArrayPipe, IsPercentValue, IsProductKindPipe];
export class ValidationModule {
}
ValidationModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ValidationModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ValidationModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ValidationModule, declarations: [IsDefinedPipe, IsInArrayPipe, IsPercentValue, IsProductKindPipe], imports: [CommonModule], exports: [IsDefinedPipe, IsInArrayPipe, IsPercentValue, IsProductKindPipe] });
ValidationModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ValidationModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ValidationModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [...pipes],
                    exports: [...pipes],
                }]
        }] });
