// @backbase/accessgroup-ui/esm2015/lib/job-role-detail-modal/job-role-detail-modal.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ModalModule } from '@backbase/ui-ang/modal';
import { IconModule } from '@backbase/ui-ang/icon';
import { ButtonModule } from '@backbase/ui-ang/button';
import { HeaderModule } from '@backbase/ui-ang/header';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { defaultGetGroupForBusinessFunction, defaultOrderedBusinessFunctionGroups, } from '@backbase/accessgroup-common-ang';
import { AccessgroupUiModule } from '../accessgroup-ui.module';
import { JobRoleDetailModalComponent, JobRoleDetailModalCustomizableDirective, } from './job-role-detail-modal.component';
import { JobRoleDetailPrivilegeIndicatorListComponent, JobRoleDetailPrivilegeIndicatorListCustomizableDirective, } from './job-role-detail-privilege-indicator-list/job-role-detail-privilege-indicator-list.component';
import { JobRoleDetailBusinessFunctionGroupService } from './services/job-role-detail-business-function-group.service';
import { BusinessFunctionsService } from './services/business-functions.service';
import { JOB_ROLE_DETAIL_MODAL_BUSINESS_FUNCTION_GROUP_ORDER, JOB_ROLE_DETAIL_MODAL_BUSINESS_FUNCTION_TO_GROUP_MAPPER, } from './models/types';
import * as i0 from "@angular/core";
const uiModules = [ModalModule, HeaderModule, IconModule, ButtonModule, LoadingIndicatorModule];
const services = [
    JobRoleDetailBusinessFunctionGroupService,
    BusinessFunctionsService,
    { provide: JOB_ROLE_DETAIL_MODAL_BUSINESS_FUNCTION_TO_GROUP_MAPPER, useValue: defaultGetGroupForBusinessFunction },
    { provide: JOB_ROLE_DETAIL_MODAL_BUSINESS_FUNCTION_GROUP_ORDER, useValue: defaultOrderedBusinessFunctionGroups },
];
const components = [
    JobRoleDetailModalComponent,
    JobRoleDetailModalCustomizableDirective,
    JobRoleDetailPrivilegeIndicatorListComponent,
    JobRoleDetailPrivilegeIndicatorListCustomizableDirective,
];
export class JobRoleDetailModalModule {
}
/** @nocollapse */ JobRoleDetailModalModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: JobRoleDetailModalModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ JobRoleDetailModalModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: JobRoleDetailModalModule, declarations: [JobRoleDetailModalComponent,
        JobRoleDetailModalCustomizableDirective,
        JobRoleDetailPrivilegeIndicatorListComponent,
        JobRoleDetailPrivilegeIndicatorListCustomizableDirective], imports: [CommonModule, ModalModule, HeaderModule, IconModule, ButtonModule, LoadingIndicatorModule, AccessgroupUiModule], exports: [JobRoleDetailModalComponent,
        JobRoleDetailModalCustomizableDirective,
        JobRoleDetailPrivilegeIndicatorListComponent,
        JobRoleDetailPrivilegeIndicatorListCustomizableDirective, ModalModule, HeaderModule, IconModule, ButtonModule, LoadingIndicatorModule] });
/** @nocollapse */ JobRoleDetailModalModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: JobRoleDetailModalModule, providers: [...services], imports: [[CommonModule, ...uiModules, AccessgroupUiModule], ModalModule, HeaderModule, IconModule, ButtonModule, LoadingIndicatorModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: JobRoleDetailModalModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules, AccessgroupUiModule],
                    declarations: [...components],
                    providers: [...services],
                    exports: [...components, ...uiModules],
                }]
        }] });
