// @backbase/ui-ang/esm2015/_example/dropdown-menu/showcase/dropdown-menu-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { DropdownMenuShowcaseComponent } from './dropdown-menu-showcase.component';
import { DropdownMenuModule } from '@backbase/ui-ang/dropdown-menu';
import { IconModule } from '@backbase/ui-ang/icon';
import { ButtonModule } from '@backbase/ui-ang/button';
import * as i0 from "@angular/core";
export class DropdownMenuShowcaseModule {
}
DropdownMenuShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownMenuShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
DropdownMenuShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownMenuShowcaseModule, declarations: [DropdownMenuShowcaseComponent], imports: [CommonModule, DropdownMenuModule, IconModule, ButtonModule] });
DropdownMenuShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownMenuShowcaseModule, imports: [[CommonModule, DropdownMenuModule, IconModule, ButtonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownMenuShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, DropdownMenuModule, IconModule, ButtonModule],
                    declarations: [DropdownMenuShowcaseComponent],
                    entryComponents: [DropdownMenuShowcaseComponent],
                }]
        }] });
