// @backbase/ui-ang/esm2015/_example/mode-header/showcase/mode-header-showcase.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { BadgeModule } from '@backbase/ui-ang/badge';
import { ButtonModule } from '@backbase/ui-ang/button';
import { DropdownMenuModule } from '@backbase/ui-ang/dropdown-menu';
import { IconModule } from '@backbase/ui-ang/icon';
import { ModeHeaderModule } from '@backbase/ui-ang/mode-header';
import { ModeHeaderShowcaseComponent } from './mode-header-showcase.component';
import * as i0 from "@angular/core";
const uiModules = [ModeHeaderModule, IconModule, BadgeModule, DropdownMenuModule, ButtonModule];
export class ModeHeaderShowcaseModule {
}
ModeHeaderShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ModeHeaderShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ModeHeaderShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ModeHeaderShowcaseModule, declarations: [ModeHeaderShowcaseComponent], imports: [CommonModule, ModeHeaderModule, IconModule, BadgeModule, DropdownMenuModule, ButtonModule] });
ModeHeaderShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ModeHeaderShowcaseModule, imports: [[CommonModule, ...uiModules]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ModeHeaderShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules],
                    declarations: [ModeHeaderShowcaseComponent],
                    entryComponents: [ModeHeaderShowcaseComponent],
                }]
        }] });
