// @backbase/ui-ang/esm2015/file-attachment/file-attachment.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FileAttachmentComponent } from './file-attachment.component';
import { ButtonModule } from '@backbase/ui-ang/button';
import { KeyboardClickModule } from '@backbase/ui-ang/keyboard-click-directive';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { IconModule } from '@backbase/ui-ang/icon';
import * as i0 from "@angular/core";
const uiModules = [ButtonModule, IconModule, KeyboardClickModule, LoadingIndicatorModule];
export class FileAttachmentModule {
}
FileAttachmentModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FileAttachmentModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
FileAttachmentModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FileAttachmentModule, declarations: [FileAttachmentComponent], imports: [CommonModule, ButtonModule, IconModule, KeyboardClickModule, LoadingIndicatorModule], exports: [FileAttachmentComponent] });
FileAttachmentModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FileAttachmentModule, imports: [[CommonModule, ...uiModules]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FileAttachmentModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules],
                    declarations: [FileAttachmentComponent],
                    exports: [FileAttachmentComponent],
                }]
        }] });
