// @backbase/payord-stop-checks-list-widget-ang/esm2015/lib/components/empty-list/empty-list.component.js
import { ChangeDetectionStrategy, Component, Directive, Input, } from '@angular/core';
import { BbTemplate } from '@backbase/foundation-ang/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/empty-state";
import * as i2 from "@angular/common";
export class StopChecksEmptyListComponent {
    constructor() {
        /**
         * Reference to instance of customizable component.
         */
        this.hostRef = this;
        /**
         * Indicates if empty result caused by user search
         * Default value is false
         */
        this.isSearch = false;
    }
}
/** @nocollapse */ StopChecksEmptyListComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StopChecksEmptyListComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ StopChecksEmptyListComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: StopChecksEmptyListComponent, selector: "bb-empty-stop-checks-list", inputs: { isSearch: "isSearch" }, ngImport: i0, template: `
    <ng-container
      bbStopChecksEmptyListCustomizableDirective
      [bbHostRef]="hostRef"
    >
    </ng-container>
    <ng-template bbStopChecksEmptyListCustomizableDirective let-hostRef>
      <div class="bb-state-container">
        <bb-empty-state-ui
          *ngIf="!hostRef.isSearch"
          title="No stop check requests"
          i18n-title="@@stop-checks.no.stop.checks.title"
          subtitle="A stop check payment alerts us that you want to prevent a check you have written from being cashed."
          i18n-subtitle="@@stop-checks.no.stop.checks.subTitle"
          iconSize="xl"
          iconModifier="payments"
        ></bb-empty-state-ui>
      </div>
    </ng-template>
  `, isInline: true, components: [{ type: i0.forwardRef(function () { return i1.EmptyStateComponent; }), selector: "bb-empty-state-ui", inputs: ["title", "subtitle", "showIcon", "iconClasses", "iconModifier", "iconSize", "iconColor"] }], directives: [{ type: i0.forwardRef(function () { return StopChecksEmptyListCustomizableDirective; }), selector: "[bbStopChecksEmptyListCustomizableDirective]" }, { type: i0.forwardRef(function () { return i2.NgIf; }), selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StopChecksEmptyListComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-empty-stop-checks-list',
                    template: `
    <ng-container
      bbStopChecksEmptyListCustomizableDirective
      [bbHostRef]="hostRef"
    >
    </ng-container>
    <ng-template bbStopChecksEmptyListCustomizableDirective let-hostRef>
      <div class="bb-state-container">
        <bb-empty-state-ui
          *ngIf="!hostRef.isSearch"
          title="No stop check requests"
          i18n-title="@@stop-checks.no.stop.checks.title"
          subtitle="A stop check payment alerts us that you want to prevent a check you have written from being cashed."
          i18n-subtitle="@@stop-checks.no.stop.checks.subTitle"
          iconSize="xl"
          iconModifier="payments"
        ></bb-empty-state-ui>
      </div>
    </ng-template>
  `,
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { isSearch: [{
                type: Input
            }] } });
export class StopChecksEmptyListCustomizableDirective extends BbTemplate {
}
/** @nocollapse */ StopChecksEmptyListCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StopChecksEmptyListCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
/** @nocollapse */ StopChecksEmptyListCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: StopChecksEmptyListCustomizableDirective, selector: "[bbStopChecksEmptyListCustomizableDirective]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StopChecksEmptyListCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbStopChecksEmptyListCustomizableDirective]',
                }]
        }] });
