// @backbase/accessgroup-ui/esm2015/lib/discard-changes-modal/discard-changes-modal.component.js
import { Component, Input, Output, EventEmitter } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/modal";
import * as i2 from "@backbase/ui-ang/button";
export class DiscardChangesModalComponent {
    constructor() {
        /**
         * Is modal open
         */
        this.isOpen = false;
        /**
         * Modal title
         */
        this.title = '';
        /**
         * Label for the discard changes button
         */
        this.discardButtonLabel = '';
        /**
         * Label for the cancel button
         */
        this.cancelButtonLabel = '';
        /**
         * Event emitter that emits the chosen action.
         * true to continue with action, false to cancel
         */
        this.resolve = new EventEmitter();
    }
    cancel() {
        this.resolve.emit(false);
    }
    discard() {
        this.resolve.emit(true);
    }
}
/** @nocollapse */ DiscardChangesModalComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DiscardChangesModalComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ DiscardChangesModalComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: DiscardChangesModalComponent, selector: "bb-discard-changes-modal", inputs: { isOpen: "isOpen", title: "title", discardButtonLabel: "discardButtonLabel", cancelButtonLabel: "cancelButtonLabel" }, outputs: { resolve: "resolve" }, ngImport: i0, template: "<bb-modal-ui [isOpen]=\"isOpen\" (cancel)=\"cancel()\" data-role=\"discard-changes-modal\">\n  <bb-modal-header-ui data-role=\"discard-changes-modal-title\" [title]=\"title\"></bb-modal-header-ui>\n  <bb-modal-body-ui>\n    <ng-template bbCustomModalBody>\n      <ng-content></ng-content>\n    </ng-template>\n  </bb-modal-body-ui>\n  <bb-modal-footer-ui>\n    <ng-template bbCustomModalFooter>\n      <div class=\"bb-button-bar\">\n        <button\n          bbButton\n          color=\"danger\"\n          class=\"bb-button-bar__button\"\n          (click)=\"discard()\"\n          data-role=\"discard-changes-modal-discard-button\">\n          {{ discardButtonLabel }}\n        </button>\n        <button\n          bbButton\n          color=\"secondary\"\n          class=\"bb-button-bar__button\"\n          (click)=\"cancel()\"\n          data-role=\"discard-changes-modal-cancel-button\">\n          {{ cancelButtonLabel }}\n        </button>\n      </div>\n    </ng-template>\n  </bb-modal-footer-ui>\n</bb-modal-ui>\n", components: [{ type: i1.ModalComponent, selector: "bb-modal-ui", inputs: ["isOpen", "aria-labelledby", "aria-describedby", "modalOptions"], outputs: ["confirm", "cancel", "isOpenChange"] }, { type: i1.ModalHeaderComponent, selector: "bb-modal-header-ui", inputs: ["title"], outputs: ["close"] }, { type: i1.ModalBodyComponent, selector: "bb-modal-body-ui" }, { type: i1.ModalFooterComponent, selector: "bb-modal-footer-ui", inputs: ["confirmText", "cancelText"], outputs: ["confirm", "cancel"] }], directives: [{ type: i1.CustomModalBodyDirective, selector: "ng-template[bbCustomModalBody]" }, { type: i1.CustomModalFooterDirective, selector: "ng-template[bbCustomModalFooter]" }, { type: i2.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DiscardChangesModalComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-discard-changes-modal',
                    templateUrl: './discard-changes-modal.component.html',
                }]
        }], propDecorators: { isOpen: [{
                type: Input
            }], title: [{
                type: Input
            }], discardButtonLabel: [{
                type: Input
            }], cancelButtonLabel: [{
                type: Input
            }], resolve: [{
                type: Output
            }] } });
