// @backbase/ui-ang/esm2015/payment-card-number-pipe/constants.js
/**
 * Default payment card number format.
 */
export const DefaultPaymentCardNumberFormat = {
    length: 16,
    maskRange: [0, 12],
    segments: 4,
};
/**
 * Lookup object containing payment card number formats for common payment card
 * vendors.
 */
export const PaymentCardNumberVendors = {
    mastercard: {
        length: 16,
        maskRange: [0, 12],
        segments: 4,
    },
    visa: {
        length: 16,
        maskRange: [0, 12],
        segments: 4,
    },
    'visa-debit': {
        length: 16,
        maskRange: [0, 12],
        segments: 4,
    },
    'diners-club': {
        length: 16,
        maskRange: [0, 12],
        segments: 4,
    },
    'american-express': {
        length: 15,
        maskRange: [0, 11],
        segments: [4, 6, 5],
    },
};
/**
 * A feature flag that enables numbers masking synchronization with backend
 * If set FE should not apply any masking and rely on the backend masking behavior
 *
 * @internal
 */
export const SYNCHRONIZE_NUMBERS_MASKING = {
    name: 'SYNCHRONIZE_NUMBERS_MASKING',
    description: 'Enables accounts masking synchronization with backend',
};
