// @backbase/ui-ang/esm2015/payment-card-number-pipe/payment-card-number.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FutureFeaturesModule } from '@backbase/foundation-ang/future';
import { PaymentCardNumberPipe } from './payment-card-number.pipe';
import { SYNCHRONIZE_NUMBERS_MASKING } from './constants';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/future";
export class PaymentCardNumberModule {
}
PaymentCardNumberModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardNumberModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PaymentCardNumberModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardNumberModule, declarations: [PaymentCardNumberPipe], imports: [CommonModule, i1.FutureFeaturesModule], exports: [PaymentCardNumberPipe] });
PaymentCardNumberModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardNumberModule, imports: [[CommonModule, FutureFeaturesModule.allowOptIn([SYNCHRONIZE_NUMBERS_MASKING])]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardNumberModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, FutureFeaturesModule.allowOptIn([SYNCHRONIZE_NUMBERS_MASKING])],
                    declarations: [PaymentCardNumberPipe],
                    exports: [PaymentCardNumberPipe],
                }]
        }] });
