// @backbase/loans-collection-ui-ang/esm2015/lib/loan-info/loan-info-term-item/loan-info-term-item.component.js
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class LoanInfoTermItemComponent {
}
/** @nocollapse */ LoanInfoTermItemComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoTermItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ LoanInfoTermItemComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: LoanInfoTermItemComponent, selector: "bb-loan-info-term-item", inputs: { loan: "loan" }, ngImport: i0, template: "<div class=\"bb-text-bold bb-block bb-block--xs\" i18n=\"Loan info term title@@loans.loan-info.term.title\">Term</div>\n<div i18n=\"Loan info term value@@loans.loan-info.term.value\" data-role=\"loan-info-term-value\">\n  {{ loan.termCount }} {loan.termUnit, select, Day {day} Week {week} Month {month} Year {year}}{loan.termCount, plural,\n  =1 {} other {s}}\n</div>\n", changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoTermItemComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-loan-info-term-item',
                    templateUrl: './loan-info-term-item.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { loan: [{
                type: Input
            }] } });
