// @backbase/loans-collection-ui-ang/esm2015/lib/loan-info/loan-info-term-item/loan-info-term-item.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { LoanInfoTermItemComponent } from './loan-info-term-item.component';
import * as i0 from "@angular/core";
export class LoanInfoTermItemModule {
}
/** @nocollapse */ LoanInfoTermItemModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoTermItemModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanInfoTermItemModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoTermItemModule, declarations: [LoanInfoTermItemComponent], imports: [CommonModule], exports: [LoanInfoTermItemComponent] });
/** @nocollapse */ LoanInfoTermItemModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoTermItemModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoTermItemModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanInfoTermItemComponent],
                    imports: [CommonModule],
                    exports: [LoanInfoTermItemComponent],
                }]
        }] });
