// @backbase/ui-ang/esm2015/_example/fieldset/showcase/fieldset-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FieldsetShowcaseComponent } from './fieldset-showcase.component';
import { FieldsetModule } from '@backbase/ui-ang/fieldset';
import { ReactiveFormsModule } from '@angular/forms';
import { InputPasswordModule } from '@backbase/ui-ang/input-password';
import { CurrencyInputModule } from '@backbase/ui-ang/currency-input';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import * as i0 from "@angular/core";
export class FieldsetShowcaseModule {
}
FieldsetShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FieldsetShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
FieldsetShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FieldsetShowcaseModule, declarations: [FieldsetShowcaseComponent], imports: [CommonModule,
        FieldsetModule,
        ReactiveFormsModule,
        InputPasswordModule,
        CurrencyInputModule,
        InputTextModule] });
FieldsetShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FieldsetShowcaseModule, imports: [[
            CommonModule,
            FieldsetModule,
            ReactiveFormsModule,
            InputPasswordModule,
            CurrencyInputModule,
            InputTextModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FieldsetShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        FieldsetModule,
                        ReactiveFormsModule,
                        InputPasswordModule,
                        CurrencyInputModule,
                        InputTextModule,
                    ],
                    declarations: [FieldsetShowcaseComponent],
                    entryComponents: [FieldsetShowcaseComponent],
                }]
        }] });
