// @backbase/ui-ang/esm2015/input-timepicker/input-timepicker.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { NgbTimepickerModule } from '@ng-bootstrap/ng-bootstrap';
import { BaseClassesModule } from '@backbase/ui-ang/base-classes';
import { TimepickerStringAdapter } from './input-timepicker.service';
import { InputTimepickerComponent } from './input-timepicker.component';
import * as i0 from "@angular/core";
export class InputTimepickerModule {
}
InputTimepickerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputTimepickerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputTimepickerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputTimepickerModule, declarations: [InputTimepickerComponent], imports: [CommonModule, NgbTimepickerModule, FormsModule, ReactiveFormsModule, BaseClassesModule], exports: [InputTimepickerComponent] });
InputTimepickerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputTimepickerModule, providers: [TimepickerStringAdapter], imports: [[CommonModule, NgbTimepickerModule, FormsModule, ReactiveFormsModule, BaseClassesModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputTimepickerModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, NgbTimepickerModule, FormsModule, ReactiveFormsModule, BaseClassesModule],
                    declarations: [InputTimepickerComponent],
                    exports: [InputTimepickerComponent],
                    providers: [TimepickerStringAdapter],
                }]
        }] });
