// @backbase/ui-ang/esm2015/product-item-credit-card/product-item-credit-card.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { CustomProductItemCreditCardDetailsFormatterDirective, CustomProductItemCreditCardMainSectionDirective, CustomProductItemCreditCardSecondarySectionDirective, CustomProductItemCreditCardTitleDirective, ProductItemCreditCardComponent, } from './product-item-credit-card.component';
import { AmountModule } from '@backbase/ui-ang/amount';
import { PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import { HighlightModule } from '@backbase/ui-ang/highlight';
import { IconModule } from '@backbase/ui-ang/icon';
import * as i0 from "@angular/core";
export class ProductItemCreditCardModule {
}
ProductItemCreditCardModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCreditCardModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductItemCreditCardModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCreditCardModule, declarations: [ProductItemCreditCardComponent,
        CustomProductItemCreditCardTitleDirective,
        CustomProductItemCreditCardDetailsFormatterDirective,
        CustomProductItemCreditCardMainSectionDirective,
        CustomProductItemCreditCardSecondarySectionDirective], imports: [CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule], exports: [ProductItemCreditCardComponent,
        CustomProductItemCreditCardTitleDirective,
        CustomProductItemCreditCardDetailsFormatterDirective,
        CustomProductItemCreditCardMainSectionDirective,
        CustomProductItemCreditCardSecondarySectionDirective] });
ProductItemCreditCardModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCreditCardModule, imports: [[CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCreditCardModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule],
                    declarations: [
                        ProductItemCreditCardComponent,
                        CustomProductItemCreditCardTitleDirective,
                        CustomProductItemCreditCardDetailsFormatterDirective,
                        CustomProductItemCreditCardMainSectionDirective,
                        CustomProductItemCreditCardSecondarySectionDirective,
                    ],
                    exports: [
                        ProductItemCreditCardComponent,
                        CustomProductItemCreditCardTitleDirective,
                        CustomProductItemCreditCardDetailsFormatterDirective,
                        CustomProductItemCreditCardMainSectionDirective,
                        CustomProductItemCreditCardSecondarySectionDirective,
                    ],
                }]
        }] });
