// @backbase/loans-collection-ui-ang/esm2015/lib/loan-info/loan-info.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ButtonModule } from '@backbase/ui-ang/button';
import { HeaderModule } from '@backbase/ui-ang/header';
import { IconModule } from '@backbase/ui-ang/icon';
import { LoanInfoComponent } from './loan-info.component';
import { LoanInfoItemModule } from './loan-info-item/loan-info-item.module';
import { LoanInfoPrintButtonModule } from './loan-info-print-button/loan-info-print-button.module';
import { LoanInfoCustomSectionDirective } from './loan-info-custom-section/loan-info-custom-section.directive';
import * as i0 from "@angular/core";
const components = [LoanInfoComponent, LoanInfoCustomSectionDirective];
const uiModules = [ButtonModule, IconModule, HeaderModule];
const modules = [LoanInfoPrintButtonModule, LoanInfoItemModule];
export class LoanInfoModule {
}
/** @nocollapse */ LoanInfoModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanInfoModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoModule, declarations: [LoanInfoComponent, LoanInfoCustomSectionDirective], imports: [CommonModule, ButtonModule, IconModule, HeaderModule, LoanInfoPrintButtonModule, LoanInfoItemModule], exports: [LoanInfoComponent, LoanInfoCustomSectionDirective] });
/** @nocollapse */ LoanInfoModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoModule, imports: [[CommonModule, ...uiModules, ...modules]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components],
                    imports: [CommonModule, ...uiModules, ...modules],
                    exports: [...components],
                }]
        }] });
