// @backbase/loans-collection-ui-ang/esm2015/lib/loan-summary/loan-summary-container.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ButtonModule } from '@backbase/ui-ang/button';
import { IconModule } from '@backbase/ui-ang/icon';
import { LoanItemStatusModule } from '../loan-list-item/loan-item-status/loan-item-status.module';
import { LoanItemTypeModule } from '../loan-list-item/loan-item-type/loan-item-type.module';
import { LoanItemNameModule } from '../loan-list-item/loan-item-name/loan-item-name.module';
import { LoanItemPaymentInfoModule } from '../loan-list-item/loan-item-payment-info/loan-item-payment-info.module';
import { LoanSummaryOverdueBannerModule } from './loan-summary-overdue-banner/loan-summary-overdue-banner.module';
import { LoanSummaryProgressBarModule } from './loan-summary-progress-bar/loan-summary-progress-bar.module';
import { LoanSummaryAmountItemModule } from './loan-summary-amount-item/loan-summary-amount-item.module';
import { LoanSummaryFormulaModule } from './loan-summary-formula/loan-summary-formula.module';
import { LoanSummaryContainerComponent } from './loan-summary-container.component';
import { LoanSummaryEditAliasModule } from './loan-summary-edit-alias/loan-summary-edit-alias.module';
import { LoanItemMenuModule } from '../loan-list-item/loan-item-menu/loan-item-menu.module';
import * as i0 from "@angular/core";
export class LoanSummaryContainerModule {
}
/** @nocollapse */ LoanSummaryContainerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryContainerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanSummaryContainerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryContainerModule, declarations: [LoanSummaryContainerComponent], imports: [CommonModule,
        IconModule,
        ButtonModule,
        LoanItemNameModule,
        LoanItemTypeModule,
        LoanItemStatusModule,
        LoanSummaryFormulaModule,
        LoanItemPaymentInfoModule,
        LoanSummaryAmountItemModule,
        LoanSummaryOverdueBannerModule,
        LoanSummaryProgressBarModule,
        LoanSummaryEditAliasModule,
        LoanItemMenuModule], exports: [LoanSummaryContainerComponent] });
/** @nocollapse */ LoanSummaryContainerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryContainerModule, imports: [[
            CommonModule,
            IconModule,
            ButtonModule,
            LoanItemNameModule,
            LoanItemTypeModule,
            LoanItemStatusModule,
            LoanSummaryFormulaModule,
            LoanItemPaymentInfoModule,
            LoanSummaryAmountItemModule,
            LoanSummaryOverdueBannerModule,
            LoanSummaryProgressBarModule,
            LoanSummaryEditAliasModule,
            LoanItemMenuModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryContainerModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanSummaryContainerComponent],
                    imports: [
                        CommonModule,
                        IconModule,
                        ButtonModule,
                        LoanItemNameModule,
                        LoanItemTypeModule,
                        LoanItemStatusModule,
                        LoanSummaryFormulaModule,
                        LoanItemPaymentInfoModule,
                        LoanSummaryAmountItemModule,
                        LoanSummaryOverdueBannerModule,
                        LoanSummaryProgressBarModule,
                        LoanSummaryEditAliasModule,
                        LoanItemMenuModule,
                    ],
                    exports: [LoanSummaryContainerComponent],
                }]
        }] });
