// @backbase/product-summary-ui/esm2015/src/bb-account-state-ui/account-state-ui.component.js
import { Component, Directive, Input } from '@angular/core';
import { BbTemplate } from '@backbase/foundation-ang/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/alert";
import * as i2 from "@angular/common";
export class AccountStateComponent {
    constructor() {
        this.hostRef = this;
        /**
         * Product summary account state indicator display flag.
         */
        this.isDisplayingAlert = true;
        this.isAlertCLosed = false;
    }
    /**
     * @internal
     */
    ngOnChanges() {
        this.isAlertCLosed = false;
    }
    closeAlert() {
        this.isAlertCLosed = true;
    }
}
AccountStateComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountStateComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
AccountStateComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AccountStateComponent, selector: "bb-account-state-ui", inputs: { isDisplayingAlert: "isDisplayingAlert", item: "item" }, usesOnChanges: true, ngImport: i0, template: "<ng-container\nbbAccountStateCustomizable\n  [bbHostRef]=\"hostRef\"\n  [bbTemplateContext]=\"{item:item}\">\n</ng-container>\n\n<ng-template bbAccountStateCustomizable\n  let-hostRef\n  let-item=\"context\">\n  <div class=\"row mt-3\" *ngIf=\"hostRef.isDisplayingAlert && !hostRef.isAlertCLosed\">\n    <div class=\"col-12\">\n      <bb-alert-ui\n        data-role=\"alert-danger\"\n        [title]=\"''\"\n        [modifier]=\"hostRef.item.severity\"\n        [dismissible]=\"true\"\n        (close)=\"hostRef.closeAlert()\">\n          <ng-content></ng-content>\n      </bb-alert-ui>\n    </div>\n  </div>\n</ng-template>\n", components: [{ type: i0.forwardRef(function () { return i1.AlertComponent; }), selector: "bb-alert-ui", inputs: ["modifier", "dismissible", "title", "message"], outputs: ["close"] }], directives: [{ type: i0.forwardRef(function () { return AccountStateCustomizableDirective; }), selector: "[bbAccountStateCustomizable]" }, { type: i0.forwardRef(function () { return i2.NgIf; }), selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountStateComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-account-state-ui',
                    templateUrl: 'account-state-ui.component.html',
                }]
        }], propDecorators: { isDisplayingAlert: [{
                type: Input
            }], item: [{
                type: Input
            }] } });
export class AccountStateCustomizableDirective extends BbTemplate {
}
AccountStateCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountStateCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
AccountStateCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: AccountStateCustomizableDirective, selector: "[bbAccountStateCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountStateCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbAccountStateCustomizable]',
                }]
        }] });
