// @backbase/product-summary-ui/esm2015/src/bb-account-state-ui/account-state.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AlertModule } from '@backbase/ui-ang/alert';
import { AccountStateComponent, AccountStateCustomizableDirective } from './account-state-ui.component';
import * as i0 from "@angular/core";
const uiModules = [AlertModule];
const components = [AccountStateComponent];
const directives = [AccountStateCustomizableDirective];
export class AccountStateModule {
}
AccountStateModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountStateModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AccountStateModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountStateModule, declarations: [AccountStateComponent, AccountStateCustomizableDirective], imports: [CommonModule, AlertModule], exports: [AccountStateComponent, AccountStateCustomizableDirective, AlertModule] });
AccountStateModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountStateModule, imports: [[CommonModule, ...uiModules], AlertModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountStateModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules],
                    declarations: [...components, ...directives],
                    exports: [...components, ...directives, ...uiModules],
                }]
        }] });
