// @backbase/consent-details-widget-ang/esm2015/src/consent-details-widget.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { ConsentUiModule } from '@backbase/consent-ui';
import { CollapsibleModule } from '@backbase/ui-ang/collapsible';
import { PhoneNumberModule } from '@backbase/ui-ang/phone-number-format-pipe';
import { ProductItemBasicAccountModule } from '@backbase/ui-ang/product-item-basic-account';
import { BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import { IconModule } from '@backbase/ui-ang/icon';
import { ModalModule } from '@backbase/ui-ang/modal';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { BadgeModule } from '@backbase/ui-ang/badge';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { ButtonModule } from '@backbase/ui-ang/button';
import { NotificationModule } from '@backbase/ui-ang/notification';
import { ConsentDetailsService } from './consent-details-widget/consent-details.service';
import { ConsentDetailsErrorComponent } from './consent-details-widget/consent-details-error/consent-details-error.component';
import { ConsentDetailsHeaderComponent, ConsentDetailsHeaderCustomizableDirective, } from './consent-details-widget/consent-details-header/consent-details-header.component';
import { ConsentDetailsContainerComponent, ConsentDetailsContainerCustomizableDirective, } from './consent-details-widget/consent-details-container/consent-details-container.component';
import { ConsentDetailsPermissionsListComponent, ConsentDetailsPermissionsListCustomizableDirective, } from './consent-details-widget/consent-details-permissions-list/consent-details-permissions-list.component';
import { ConsentDetailsAccountsListComponent, ConsentDetailsAccountsListCustomizableDirective, } from './consent-details-widget/consent-details-accounts-list/consent-details-accounts-list.component';
import { ConsentDetailsTppDetailsComponent, ConsentDetailsTppDetailsCustomizableDirective, } from './consent-details-widget/consent-details-tpp-details/consent-details-tpp-details.component';
import { ConsentDetailsWidgetComponent } from './consent-details-widget.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const components = [
    ConsentDetailsWidgetComponent,
    ConsentDetailsTppDetailsComponent,
    ConsentDetailsAccountsListComponent,
    ConsentDetailsPermissionsListComponent,
    ConsentDetailsContainerComponent,
    ConsentDetailsHeaderComponent,
    ConsentDetailsErrorComponent,
];
const directives = [
    ConsentDetailsContainerCustomizableDirective,
    ConsentDetailsHeaderCustomizableDirective,
    ConsentDetailsTppDetailsCustomizableDirective,
    ConsentDetailsPermissionsListCustomizableDirective,
    ConsentDetailsAccountsListCustomizableDirective,
];
const uiModules = [
    ConsentUiModule,
    CollapsibleModule,
    PhoneNumberModule,
    ProductItemBasicAccountModule,
    BbDatePipeModule,
    IconModule,
    ModalModule,
    LoadingIndicatorModule,
    BadgeModule,
    EmptyStateModule,
    ButtonModule,
    NotificationModule,
];
export class ConsentDetailsWidgetModule {
}
ConsentDetailsWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentDetailsWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ConsentDetailsWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentDetailsWidgetModule, declarations: [ConsentDetailsWidgetComponent,
        ConsentDetailsTppDetailsComponent,
        ConsentDetailsAccountsListComponent,
        ConsentDetailsPermissionsListComponent,
        ConsentDetailsContainerComponent,
        ConsentDetailsHeaderComponent,
        ConsentDetailsErrorComponent, ConsentDetailsContainerCustomizableDirective,
        ConsentDetailsHeaderCustomizableDirective,
        ConsentDetailsTppDetailsCustomizableDirective,
        ConsentDetailsPermissionsListCustomizableDirective,
        ConsentDetailsAccountsListCustomizableDirective], imports: [CommonModule, i1.BackbaseCoreModule, ConsentUiModule,
        CollapsibleModule,
        PhoneNumberModule,
        ProductItemBasicAccountModule,
        BbDatePipeModule,
        IconModule,
        ModalModule,
        LoadingIndicatorModule,
        BadgeModule,
        EmptyStateModule,
        ButtonModule,
        NotificationModule], exports: [ConsentDetailsWidgetComponent,
        ConsentDetailsTppDetailsComponent,
        ConsentDetailsAccountsListComponent,
        ConsentDetailsPermissionsListComponent,
        ConsentDetailsContainerComponent,
        ConsentDetailsHeaderComponent,
        ConsentDetailsErrorComponent, ConsentDetailsContainerCustomizableDirective,
        ConsentDetailsHeaderCustomizableDirective,
        ConsentDetailsTppDetailsCustomizableDirective,
        ConsentDetailsPermissionsListCustomizableDirective,
        ConsentDetailsAccountsListCustomizableDirective, ConsentUiModule,
        CollapsibleModule,
        PhoneNumberModule,
        ProductItemBasicAccountModule,
        BbDatePipeModule,
        IconModule,
        ModalModule,
        LoadingIndicatorModule,
        BadgeModule,
        EmptyStateModule,
        ButtonModule,
        NotificationModule] });
ConsentDetailsWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentDetailsWidgetModule, providers: [ConsentDetailsService], imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { ConsentDetailsWidgetComponent },
            }),
            ...uiModules,
        ], ConsentUiModule,
        CollapsibleModule,
        PhoneNumberModule,
        ProductItemBasicAccountModule,
        BbDatePipeModule,
        IconModule,
        ModalModule,
        LoadingIndicatorModule,
        BadgeModule,
        EmptyStateModule,
        ButtonModule,
        NotificationModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentDetailsWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { ConsentDetailsWidgetComponent },
                        }),
                        ...uiModules,
                    ],
                    declarations: [...components, ...directives],
                    exports: [...components, ...directives, ...uiModules],
                    providers: [ConsentDetailsService],
                }]
        }] });
