// @backbase/ui-ang/esm2015/_example/payment-card-number-pipe/examples/payment-card-number-pipe.example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/payment-card-number-pipe";
export class PaymentCardNumberPipeExampleComponent {
    constructor() {
        this.productNumber = 'NL75INGB3400120056';
        this.productNumberFormat = {
            length: 18,
            maskRange: [0, 12],
            segments: 4,
        };
    }
}
PaymentCardNumberPipeExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardNumberPipeExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
PaymentCardNumberPipeExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: PaymentCardNumberPipeExampleComponent, selector: "bb-payment-card-number-pipe-example-ui", ngImport: i0, template: "<p>{{ productNumber | paymentCardNumber: productNumberFormat }}</p>\n", pipes: { "paymentCardNumber": i1.PaymentCardNumberPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentCardNumberPipeExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-payment-card-number-pipe-example-ui',
                    templateUrl: './payment-card-number-pipe.example.component.html',
                }]
        }] });
