// @backbase/comments-v2-client-ang/esm2015/api/comments.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { COMMENTS_BASE_PATH } from '../variables';
import { CommentsConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class CommentsService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '/client-api/comments/v2';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new CommentsConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    addPost(requestParameters, observe = 'body', reportProgress = false, options) {
        const _threadId = requestParameters["threadId"];
        if (_threadId === null || _threadId === undefined) {
            throw new Error('Required parameter threadId was null or undefined when calling addPost.');
        }
        const _postAddDto = requestParameters["postAddDto"];
        if (_postAddDto === null || _postAddDto === undefined) {
            throw new Error('Required parameter postAddDto was null or undefined when calling addPost.');
        }
        let headers = this.defaultHeaders;
        let credential;
        credential = this.configuration.lookupCredential('bearerAuth');
        if (credential) {
            headers = headers.set('Authorization', 'Bearer ' + credential);
        }
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/comments/v2/threads/${encodeURIComponent(String(_threadId))}/posts`, _postAddDto, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    addPostUrl(requestParameters) {
        const _threadId = requestParameters["threadId"];
        if (_threadId === null || _threadId === undefined) {
            throw new Error('Required parameter threadId was null or undefined when calling addPost.');
        }
        const _postAddDto = requestParameters["postAddDto"];
        if (_postAddDto === null || _postAddDto === undefined) {
            throw new Error('Required parameter postAddDto was null or undefined when calling addPost.');
        }
        return `${this.configuration.basePath}/client-api/comments/v2/threads/${encodeURIComponent(String(_threadId))}/posts`;
    }
    addThread(requestParameters, observe = 'body', reportProgress = false, options) {
        const _threadAddDto = requestParameters["threadAddDto"];
        if (_threadAddDto === null || _threadAddDto === undefined) {
            throw new Error('Required parameter threadAddDto was null or undefined when calling addThread.');
        }
        let headers = this.defaultHeaders;
        let credential;
        credential = this.configuration.lookupCredential('bearerAuth');
        if (credential) {
            headers = headers.set('Authorization', 'Bearer ' + credential);
        }
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/comments/v2/threads`, _threadAddDto, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    addThreadUrl(requestParameters) {
        const _threadAddDto = requestParameters["threadAddDto"];
        if (_threadAddDto === null || _threadAddDto === undefined) {
            throw new Error('Required parameter threadAddDto was null or undefined when calling addThread.');
        }
        return `${this.configuration.basePath}/client-api/comments/v2/threads`;
    }
    deletePost(requestParameters, observe = 'body', reportProgress = false, options) {
        const _threadId = requestParameters["threadId"];
        if (_threadId === null || _threadId === undefined) {
            throw new Error('Required parameter threadId was null or undefined when calling deletePost.');
        }
        const _postId = requestParameters["postId"];
        if (_postId === null || _postId === undefined) {
            throw new Error('Required parameter postId was null or undefined when calling deletePost.');
        }
        let headers = this.defaultHeaders;
        let credential;
        credential = this.configuration.lookupCredential('bearerAuth');
        if (credential) {
            headers = headers.set('Authorization', 'Bearer ' + credential);
        }
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.delete(`${this.configuration.basePath}/client-api/comments/v2/threads/${encodeURIComponent(String(_threadId))}/posts/${encodeURIComponent(String(_postId))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    deletePostUrl(requestParameters) {
        const _threadId = requestParameters["threadId"];
        if (_threadId === null || _threadId === undefined) {
            throw new Error('Required parameter threadId was null or undefined when calling deletePost.');
        }
        const _postId = requestParameters["postId"];
        if (_postId === null || _postId === undefined) {
            throw new Error('Required parameter postId was null or undefined when calling deletePost.');
        }
        return `${this.configuration.basePath}/client-api/comments/v2/threads/${encodeURIComponent(String(_threadId))}/posts/${encodeURIComponent(String(_postId))}`;
    }
    getPostList(requestParameters, observe = 'body', reportProgress = false, options) {
        const _threadId = requestParameters["threadId"];
        if (_threadId === null || _threadId === undefined) {
            throw new Error('Required parameter threadId was null or undefined when calling getPostList.');
        }
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        let headers = this.defaultHeaders;
        let credential;
        credential = this.configuration.lookupCredential('bearerAuth');
        if (credential) {
            headers = headers.set('Authorization', 'Bearer ' + credential);
        }
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/comments/v2/threads/${encodeURIComponent(String(_threadId))}/posts`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getPostListUrl(requestParameters) {
        const _threadId = requestParameters["threadId"];
        if (_threadId === null || _threadId === undefined) {
            throw new Error('Required parameter threadId was null or undefined when calling getPostList.');
        }
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/comments/v2/threads/${encodeURIComponent(String(_threadId))}/posts${queryString ? `?${queryString}` : ''}`;
    }
    getPostsByAssociationsList(requestParameters, observe = 'body', reportProgress = false, options) {
        const _entityFilter = requestParameters["entityFilter"];
        if (_entityFilter === null || _entityFilter === undefined) {
            throw new Error('Required parameter entityFilter was null or undefined when calling getPostsByAssociationsList.');
        }
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_entityFilter !== undefined && _entityFilter !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _entityFilter, 'entityFilter');
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        let headers = this.defaultHeaders;
        let credential;
        credential = this.configuration.lookupCredential('bearerAuth');
        if (credential) {
            headers = headers.set('Authorization', 'Bearer ' + credential);
        }
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/comments/v2/posts`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getPostsByAssociationsListUrl(requestParameters) {
        const _entityFilter = requestParameters["entityFilter"];
        if (_entityFilter === null || _entityFilter === undefined) {
            throw new Error('Required parameter entityFilter was null or undefined when calling getPostsByAssociationsList.');
        }
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_entityFilter !== undefined && _entityFilter !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _entityFilter, 'entityFilter');
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/comments/v2/posts${queryString ? `?${queryString}` : ''}`;
    }
    getThreadList(requestParameters, observe = 'body', reportProgress = false, options) {
        const _entityFilter = requestParameters["entityFilter"];
        if (_entityFilter === null || _entityFilter === undefined) {
            throw new Error('Required parameter entityFilter was null or undefined when calling getThreadList.');
        }
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_entityFilter !== undefined && _entityFilter !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _entityFilter, 'entityFilter');
        }
        let headers = this.defaultHeaders;
        let credential;
        credential = this.configuration.lookupCredential('bearerAuth');
        if (credential) {
            headers = headers.set('Authorization', 'Bearer ' + credential);
        }
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/comments/v2/threads`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getThreadListUrl(requestParameters) {
        const _entityFilter = requestParameters["entityFilter"];
        if (_entityFilter === null || _entityFilter === undefined) {
            throw new Error('Required parameter entityFilter was null or undefined when calling getThreadList.');
        }
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_entityFilter !== undefined && _entityFilter !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _entityFilter, 'entityFilter');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/comments/v2/threads${queryString ? `?${queryString}` : ''}`;
    }
    updatePost(requestParameters, observe = 'body', reportProgress = false, options) {
        const _threadId = requestParameters["threadId"];
        if (_threadId === null || _threadId === undefined) {
            throw new Error('Required parameter threadId was null or undefined when calling updatePost.');
        }
        const _postId = requestParameters["postId"];
        if (_postId === null || _postId === undefined) {
            throw new Error('Required parameter postId was null or undefined when calling updatePost.');
        }
        const _postUpdateDto = requestParameters["postUpdateDto"];
        if (_postUpdateDto === null || _postUpdateDto === undefined) {
            throw new Error('Required parameter postUpdateDto was null or undefined when calling updatePost.');
        }
        let headers = this.defaultHeaders;
        let credential;
        credential = this.configuration.lookupCredential('bearerAuth');
        if (credential) {
            headers = headers.set('Authorization', 'Bearer ' + credential);
        }
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/comments/v2/threads/${encodeURIComponent(String(_threadId))}/posts/${encodeURIComponent(String(_postId))}`, _postUpdateDto, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    updatePostUrl(requestParameters) {
        const _threadId = requestParameters["threadId"];
        if (_threadId === null || _threadId === undefined) {
            throw new Error('Required parameter threadId was null or undefined when calling updatePost.');
        }
        const _postId = requestParameters["postId"];
        if (_postId === null || _postId === undefined) {
            throw new Error('Required parameter postId was null or undefined when calling updatePost.');
        }
        const _postUpdateDto = requestParameters["postUpdateDto"];
        if (_postUpdateDto === null || _postUpdateDto === undefined) {
            throw new Error('Required parameter postUpdateDto was null or undefined when calling updatePost.');
        }
        return `${this.configuration.basePath}/client-api/comments/v2/threads/${encodeURIComponent(String(_threadId))}/posts/${encodeURIComponent(String(_postId))}`;
    }
}
CommentsService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CommentsService, deps: [{ token: i1.HttpClient }, { token: COMMENTS_BASE_PATH, optional: true }, { token: i2.CommentsConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
CommentsService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CommentsService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CommentsService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [COMMENTS_BASE_PATH]
                }] }, { type: i2.CommentsConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
