// @backbase/ui-ang/esm2015/_example/input-timepicker/showcase/input-timepicker-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { InputTimepickerShowcaseComponent } from './input-timepicker-showcase.component';
import { InputTimepickerModule } from '@backbase/ui-ang/input-timepicker';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import * as i0 from "@angular/core";
export class InputTimepickerShowcaseModule {
}
InputTimepickerShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputTimepickerShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputTimepickerShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputTimepickerShowcaseModule, declarations: [InputTimepickerShowcaseComponent], imports: [CommonModule, InputTimepickerModule, InputValidationMessageModule, ReactiveFormsModule, FormsModule] });
InputTimepickerShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputTimepickerShowcaseModule, imports: [[CommonModule, InputTimepickerModule, InputValidationMessageModule, ReactiveFormsModule, FormsModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputTimepickerShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, InputTimepickerModule, InputValidationMessageModule, ReactiveFormsModule, FormsModule],
                    declarations: [InputTimepickerShowcaseComponent],
                    entryComponents: [InputTimepickerShowcaseComponent],
                }]
        }] });
