// @backbase/ui-ang/esm2015/_example/input-timepicker/showcase/input-timepicker-showcase.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@angular/forms";
import * as i2 from "@backbase/ui-ang/input-timepicker";
import * as i3 from "@backbase/ui-ang/input-validation-message";
import * as i4 from "@angular/common";
let InputTimepickerShowcaseComponent = class InputTimepickerShowcaseComponent {
    constructor(formBuilder) {
        this.formBuilder = formBuilder;
        this.vForm = this.formBuilder.group({
            time: [''],
        });
        this.timeModel = '';
        this.min = '13:00:00';
        this.max = '14:30:00';
    }
    hasError(type) {
        const time = this.vForm.get('time');
        if (time !== null && time.errors) {
            return !!time.errors[type];
        }
        return false;
    }
    showErrors() {
        const time = this.vForm.get('time');
        if (time !== null) {
            return time.errors && !time.valid;
        }
        return null;
    }
};
InputTimepickerShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputTimepickerShowcaseComponent, deps: [{ token: i1.FormBuilder }], target: i0.ɵɵFactoryTarget.Component });
InputTimepickerShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: InputTimepickerShowcaseComponent, selector: "bb-input-timepicker-ui-showcase", ngImport: i0, template: "<div class=\"bb-block bb-block--lg\">\n  <h2>Timepicker with 10 minute step and validation errors</h2>\n  <form [formGroup]=\"vForm\">\n    <bb-input-timepicker-ui\n      id=\"timePicker-validation\"\n      [aria-describedby]=\"'input-timepicker-example-error'\"\n      [readonly]=\"false\"\n      [minuteStep]=\"10\"\n      formControlName=\"time\"\n      [min]=\"min\"\n      [max]=\"max\"\n    >\n    </bb-input-timepicker-ui>\n\n    <bb-input-validation-message-ui [showErrors]=\"showErrors()\" [id]=\"'input-timepicker-example-error'\">\n      <span *ngIf=\"hasError('minTime')\">Time should be after {{ min }}</span>\n      <span *ngIf=\"hasError('maxTime')\">Time should not exceed {{ max }}</span>\n      <span *ngIf=\"hasError('minMaxTime')\">Time should be between {{ min }} and {{ max }}</span>\n    </bb-input-validation-message-ui>\n  </form>\n</div>\n<div class=\"bb-block bb-block--lg\">\n  <h2>Readonly timepicker</h2>\n  <bb-input-timepicker-ui id=\"timepicker-readonly\" [readonly]=\"true\"></bb-input-timepicker-ui>\n</div>\n<div class=\"bb-block bb-block--lg\">\n  <h2>Disabled timepicker</h2>\n  <bb-input-timepicker-ui id=\"timepicker-readonly\" [disabled]=\"true\"></bb-input-timepicker-ui>\n</div>\n<div class=\"bb-block bb-block--lg\">\n  <h2>Small timepicker</h2>\n  <bb-input-timepicker-ui id=\"timepicker-readonly-small\" [size]=\"'small'\"></bb-input-timepicker-ui>\n</div>\n<div class=\"bb-block bb-block--lg\">\n  <h2>Timepicker with seconds and meridian</h2>\n  <bb-input-timepicker-ui\n    id=\"timepicker-meridian\"\n    [readonly]=\"false\"\n    [seconds]=\"true\"\n    clock=\"12h\"\n    [(ngModel)]=\"timeModel\"\n    [min]=\"min\"\n  >\n  </bb-input-timepicker-ui>\n</div>\n", components: [{ type: i2.InputTimepickerComponent, selector: "bb-input-timepicker-ui", inputs: ["hourStep", "clock", "minuteStep", "readonly", "seconds", "secondStep", "min", "max", "autocomplete"] }, { type: i3.InputValidationMessageComponent, selector: "bb-input-validation-message-ui", inputs: ["showErrors", "messageId"] }], directives: [{ type: i1.ɵNgNoValidate, selector: "form:not([ngNoForm]):not([ngNativeValidate])" }, { type: i1.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { type: i1.FormGroupDirective, selector: "[formGroup]", inputs: ["formGroup"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { type: i1.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { type: i1.FormControlName, selector: "[formControlName]", inputs: ["disabled", "formControlName", "ngModel"], outputs: ["ngModelChange"] }, { type: i4.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i1.NgModel, selector: "[ngModel]:not([formControlName]):not([formControl])", inputs: ["name", "disabled", "ngModel", "ngModelOptions"], outputs: ["ngModelChange"], exportAs: ["ngModel"] }] });
InputTimepickerShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
    /**
     * @name InputTimepickerShowcaseComponent
     *
     * @description
     * Component that displays multiple examples with the timepicker component.
     */
], InputTimepickerShowcaseComponent);
export { InputTimepickerShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputTimepickerShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-input-timepicker-ui-showcase',
                    templateUrl: './input-timepicker-showcase.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.FormBuilder }]; } });
