// @backbase/product-summary-account-selector-widget-ang/esm2015/src/product-summary.service.js
import { Injectable } from '@angular/core';
import { BehaviorSubject, of, combineLatest } from 'rxjs';
import { isProductVisible, cacheRequest } from '@backbase/product-summary-common-ang';
import { parseError,
/* eslint-enable */
 } from './product-summary-error';
import { catchError, distinctUntilChanged, map } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/arrangements";
const defaultListParams = {};
/**
 * Service for fetching and storing accounts and properties from model.xml
 *
 * This service relies on providers from `ProductSummaryAccountSelectorWidgetModule`.
 *
 * @see ProductSummaryAccountSelectorWidgetModule
 *
 * @usageNotes
 *
 * ### Ensure the ProductSummaryAccountSelectorWidgetModule is imported to your component module
 *
 * ```ts
 * @NgModule({
 *   ...
 *   imports: [
 *     ...
 *     ProductSummaryAccountSelectorWidgetModule,
 *   ],
 * })
 * export class MyWidgetModule {}
 * ```
 *
 * ### Inject this service into your component
 *
 * ```ts
 * @Component({
 *   ...
 *   providers: [ProductSummaryService],
 * })
 *  export class MyComponent {
 * ```
 */
export class ProductSummaryService {
    /**
     * Constructor
     *
     * @param productSummaryDataService
     */
    constructor(productSummaryDataService) {
        this.productSummaryDataService = productSummaryDataService;
        /**
         * Used to store the error occured during loading of accounts
         */
        this.error = new BehaviorSubject(undefined);
        this.listParams = new BehaviorSubject(defaultListParams);
        this.listRefresh = new BehaviorSubject(false);
        /**
         * List of accounts fetched from the server
         */
        this.accountsList = combineLatest([
            this.listParams.pipe(distinctUntilChanged()),
            this.listRefresh,
        ]).pipe(cacheRequest(([params]) => this.getProductSummary(params)));
    }
    /**
     * Refreshes the list of accounts
     */
    refreshAccountsList() {
        this.listRefresh.next(true);
    }
    /**
     * @internal
     */
    ngOnDestroy() {
        this.error.complete();
    }
    getProductSummary(params) {
        return this.productSummaryDataService.getProductSummary(params).pipe(map((res) => this.responseToProductList(res)), catchError((error) => {
            this.error.next(parseError(error));
            return of(undefined);
        }));
    }
    extractProducts(memo, { products }) {
        return memo.concat(products);
    }
    toProductKindList(res) {
        return Object.entries(res)
            .filter(([key]) => key !== 'aggregatedBalance')
            .map(([kind, value]) => {
            if (Array.isArray(value)) {
                value.products = value.reduce(this.extractProducts, []);
            }
            let products = [];
            if (value.products && Array.isArray(value.products)) {
                products = value.products.filter(isProductVisible).map((product) => (Object.assign(Object.assign({}, product), { kind })));
            }
            return { products };
        });
    }
    responseToProductList(res) {
        return this.toProductKindList(res).reduce(this.extractProducts, []);
    }
}
ProductSummaryService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryService, deps: [{ token: i1.ProductSummaryHttpService }], target: i0.ɵɵFactoryTarget.Injectable });
ProductSummaryService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.ProductSummaryHttpService }]; } });
