// @backbase/product-summary-account-selector-widget-ang/esm2015/src/product-summary-error.js
export class ProductKindsBadRequest {
    constructor(response) {
        this.response = response;
        this.kind = 'ProductKindsBadRequest';
        this.messages = ((response.error || {}).errors || []).map((e) => e.message);
    }
}
export class ProductKindsUnknownError {
    constructor(response) {
        this.response = response;
        this.kind = 'ProductKindsUnknownError';
        this.messages = [response.message];
    }
}
export class ProductKindsAccessDenied {
    constructor(response) {
        this.response = response;
        this.kind = 'ProductKindsAccessDenied';
        this.messages = ((response.error || {}).errors || []).map((e) => e.message);
    }
}
export const parseError = (error) => {
    if (error.error instanceof ErrorEvent) {
        return new ProductKindsUnknownError(error);
    }
    else {
        switch (error.status) {
            case 400:
                return new ProductKindsBadRequest(error);
            case 403:
                return new ProductKindsAccessDenied(error);
            case 401:
                return new ProductKindsAccessDenied(error);
            default:
                return new ProductKindsUnknownError(error);
        }
    }
};
