// @backbase/ui-ang/esm2015/input-checkbox/input-checkbox.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { IconModule } from '@backbase/ui-ang/icon';
import { FormsModule } from '@angular/forms';
import { BaseClassesModule } from '@backbase/ui-ang/base-classes';
import { InputCheckboxComponent } from './input-checkbox.component';
import * as i0 from "@angular/core";
export class InputCheckboxModule {
}
InputCheckboxModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputCheckboxModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputCheckboxModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputCheckboxModule, declarations: [InputCheckboxComponent], imports: [CommonModule, IconModule, FormsModule, BaseClassesModule], exports: [InputCheckboxComponent] });
InputCheckboxModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputCheckboxModule, imports: [[CommonModule, IconModule, FormsModule, BaseClassesModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputCheckboxModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, IconModule, FormsModule, BaseClassesModule],
                    declarations: [InputCheckboxComponent],
                    exports: [InputCheckboxComponent],
                }]
        }] });
