// @backbase/transactions-common-ang/esm2015/src/model/transactions-check-images.model.js
export var TransactionsCheckImageType;
(function (TransactionsCheckImageType) {
    TransactionsCheckImageType["PNG"] = "image/png";
    TransactionsCheckImageType["JPEG"] = "image/jpeg";
})(TransactionsCheckImageType || (TransactionsCheckImageType = {}));
export var TransactionsCheckImageSide;
(function (TransactionsCheckImageSide) {
    TransactionsCheckImageSide["FRONT"] = "FRONT";
    TransactionsCheckImageSide["BACK"] = "BACK";
})(TransactionsCheckImageSide || (TransactionsCheckImageSide = {}));
export var TransactionsCheckImages;
(function (TransactionsCheckImages) {
    function processImages(imageObject) {
        return {
            side: imageObject.side,
            mimeType: imageObject.mimeType,
            content: imageObject.content,
        };
    }
    function fromHttpResponse(response) {
        const imagesArray = (response.body && response.body.images) || [];
        const mappedImages = imagesArray.map(processImages);
        return {
            images: mappedImages,
        };
    }
    TransactionsCheckImages.fromHttpResponse = fromHttpResponse;
})(TransactionsCheckImages || (TransactionsCheckImages = {}));
