// @backbase/ui-ang/esm2015/_example/button/examples/button.example.service.js
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../../configuration/configuration-input.service";
export class ButtonExampleService {
    constructor(configurationService) {
        this.configurationService = configurationService;
        this.btnCheckboxList = [
            {
                label: 'Block',
                value: false,
                formControlName: 'block',
            },
            {
                label: 'Disabled',
                value: false,
                formControlName: 'disabled',
            },
        ];
        this.btnRadioList = [
            {
                label: 'Button Size',
                formControlName: 'buttonSize',
                radioItems: [
                    {
                        label: 'Small',
                        value: 'sm',
                    },
                    {
                        label: 'Medium',
                        value: 'md',
                    },
                ],
            },
        ];
        this.btnDropdownList = [
            {
                label: 'Color',
                formControlName: 'color',
                dropdownItemList: ['primary', 'secondary', 'success', 'danger', 'warning', 'info', 'link', 'link-text'],
            },
        ];
        this.btnValues = {
            checkbox: {
                block: false,
                disabled: false,
            },
            radio: {
                buttonSize: 'sm',
            },
            dropdown: {
                color: 'primary',
            },
        };
        this.configurationService.sharedField.subscribe((val) => {
            this.btnValues.checkbox.block = val.checkbox.block;
            this.btnValues.checkbox.disabled = val.checkbox.disabled;
            this.btnValues.radio.buttonSize = val.radio.buttonSize;
            this.btnValues.dropdown.color = val.dropdown.color;
        });
    }
}
ButtonExampleService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ButtonExampleService, deps: [{ token: i1.ConfigurationInputService }], target: i0.ɵɵFactoryTarget.Injectable });
ButtonExampleService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ButtonExampleService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ButtonExampleService, decorators: [{
            type: Injectable,
            args: [{ providedIn: 'root' }]
        }], ctorParameters: function () { return [{ type: i1.ConfigurationInputService }]; } });
