// @backbase/product-summary-manage-accounts-widget-ang/esm2015/src/manage-account-error.js
export class ManageAccountNotFound {
    constructor(response) {
        this.response = response;
        this.kind = 'ManageAccountNotFound';
        this.messages = ((response.error || {}).errors || []).map((e) => e.message);
    }
}
export class ManageAccountBadRequest {
    constructor(response) {
        this.response = response;
        this.kind = 'ManageAccountBadRequest';
        this.messages = ((response.error || {}).errors || []).map((e) => e.message);
    }
}
export class ManageAccountAccessDenied {
    constructor(response) {
        this.response = response;
        this.kind = 'ManageAccountAccessDenied';
        this.messages = ((response.error || {}).errors || []).map((e) => e.message);
    }
}
export class ManageAccountUnknownError {
    constructor(response) {
        this.response = response;
        this.kind = 'ManageAccountUnknownError';
        this.messages = [response.message];
    }
}
export class ManageAccountConnectivityError {
    constructor(response) {
        this.response = response;
        this.kind = 'ManageAccountConnectivityError';
        this.messages = ((response.error || {}).errors || []).map((e) => e.message);
    }
}
export const parseError = (error) => {
    if (error.error instanceof ErrorEvent) {
        return new ManageAccountUnknownError(error);
    }
    else {
        switch (error.status) {
            case 0:
                return new ManageAccountConnectivityError(error);
            case 401:
                return new ManageAccountAccessDenied(error);
            case 403:
                return new ManageAccountAccessDenied(error);
            case 400:
                return new ManageAccountBadRequest(error);
            case 404:
                return new ManageAccountNotFound(error);
            default:
                return new ManageAccountUnknownError(error);
        }
    }
};
