// @backbase/product-summary-manage-accounts-widget-ang/esm2015/src/manage-accounts-widget-properties.service.js
import { Injectable } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import * as i0 from "@angular/core";
const defaultProperties = {
    businessFunction: 'Product Summary',
    resourceName: 'Product Summary',
    privilege: 'view',
    size: 20,
};
/**
 * State management for fetching and storing widget properties from its inputs
 *
 * This service relies on providers from `ProductSummaryManageAccountsWidgetModule`.
 *
 * @see ProductSummaryManageAccountsWidgetModule
 *
 * @usageNotes
 *
 * ### Ensure the ProductSummaryManageAccountsWidgetModule is imported to your component module
 *
 * ```ts
 * @NgModule({
 *   ...
 *   imports: [
 *     ...
 *     ProductSummaryManageAccountsWidgetModule,
 *   ],
 * })
 * export class MyWidgetModule {}
 * ```
 *
 * ### Inject this service into your component
 *
 * ```ts
 * @Component({
 *   ...
 *   providers: [ManageAccountsWidgetPropertiesService],
 * })
 *  export class MyComponent {
 * ```
 */
export class ManageAccountsWidgetPropertiesService {
    constructor() {
        this.propertiesConfig$ = new BehaviorSubject(defaultProperties);
    }
    get widgetProperties() {
        return this.propertiesConfig$;
    }
    setConfig(config) {
        this.propertiesConfig$.next({
            businessFunction: config.businessFunction || defaultProperties.businessFunction,
            resourceName: config.resourceName || defaultProperties.resourceName,
            privilege: config.privilege || defaultProperties.privilege,
            size: config.size || defaultProperties.size,
        });
    }
}
ManageAccountsWidgetPropertiesService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManageAccountsWidgetPropertiesService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
ManageAccountsWidgetPropertiesService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManageAccountsWidgetPropertiesService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManageAccountsWidgetPropertiesService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return []; } });
