// @backbase/product-summary-manage-accounts-widget-ang/esm2015/src/product-summary-manage-accounts-widget.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { InputInlineEditModule } from '@backbase/ui-ang/input-inline-edit';
import { PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import { AmountModule } from '@backbase/ui-ang/amount';
import { SwitchModule } from '@backbase/ui-ang/switch';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { NotificationModule } from '@backbase/ui-ang/notification';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { ArrangementStateIndicatorModule, ProductSummaryAccountLogoModule, ProductSummaryFooterModule, ProductKindNameModule, ProductNumberModule, } from '@backbase/product-summary-ui';
import { ProductsExternalInfoService, ProductSummaryCommon, ValidationModule, } from '@backbase/product-summary-common-ang';
import { ManageAccountsService } from './manage.service';
import { ProductSummaryManageAccountsWidgetComponent, ProductSummaryManageAccountCustomizableDirective, ProductSummaryManageAccountLoadingCustomizableDirective, ManageAccountsFooterCustomizableDirective, } from './product-summary-manage-accounts-widget/product-summary-manage-accounts-widget.component';
import { ManageAccountItemComponent, ManageAccountItemCustomizableDirective, } from './manage-account-item/manage-account-item.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const components = [ProductSummaryManageAccountsWidgetComponent, ManageAccountItemComponent];
const directives = [
    ProductSummaryManageAccountCustomizableDirective,
    ManageAccountItemCustomizableDirective,
    ProductSummaryManageAccountLoadingCustomizableDirective,
    ManageAccountsFooterCustomizableDirective,
];
const uiModules = [
    EmptyStateModule,
    LoadingIndicatorModule,
    InputInlineEditModule,
    PaymentCardNumberModule,
    AmountModule,
    SwitchModule,
    NotificationModule,
    ArrangementStateIndicatorModule,
    ProductSummaryAccountLogoModule,
    ProductSummaryFooterModule,
    ProductKindNameModule,
    ProductNumberModule,
    TooltipModule,
    ValidationModule,
];
/**
 * Exports the Widgets and UI components for managing accounts
 *
 * The following widgets are included in this module:
 *  - `ProductSummaryManageAccountsWidgetComponent`
 *
 * @see ProductSummaryManageAccountsWidgetComponent
 *
 * The following components are included in this module:
 * - `ManageAccountItemComponent`,
 *
 * @see `ManageAccountItemComponent`,
 *
 * @usageNotes
 *
 * ### Include the exported widgets in your `AppModule`:
 *
 * ```typescript
 * @NgModule({
 *   imports: [
 *     ProductSummaryManageAccountsWidgetModule,
 *   ],
 *   ...
 * })
 * export class AppModule {}
 * ```
 */
export class ProductSummaryManageAccountsWidgetModule {
}
ProductSummaryManageAccountsWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryManageAccountsWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductSummaryManageAccountsWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryManageAccountsWidgetModule, declarations: [ProductSummaryManageAccountsWidgetComponent, ManageAccountItemComponent, ProductSummaryManageAccountCustomizableDirective,
        ManageAccountItemCustomizableDirective,
        ProductSummaryManageAccountLoadingCustomizableDirective,
        ManageAccountsFooterCustomizableDirective], imports: [CommonModule,
        FormsModule, i1.BackbaseCoreModule, ProductSummaryCommon, EmptyStateModule,
        LoadingIndicatorModule,
        InputInlineEditModule,
        PaymentCardNumberModule,
        AmountModule,
        SwitchModule,
        NotificationModule,
        ArrangementStateIndicatorModule,
        ProductSummaryAccountLogoModule,
        ProductSummaryFooterModule,
        ProductKindNameModule,
        ProductNumberModule,
        TooltipModule,
        ValidationModule], exports: [ProductSummaryManageAccountsWidgetComponent, ManageAccountItemComponent, ProductSummaryManageAccountCustomizableDirective,
        ManageAccountItemCustomizableDirective,
        ProductSummaryManageAccountLoadingCustomizableDirective,
        ManageAccountsFooterCustomizableDirective, EmptyStateModule,
        LoadingIndicatorModule,
        InputInlineEditModule,
        PaymentCardNumberModule,
        AmountModule,
        SwitchModule,
        NotificationModule,
        ArrangementStateIndicatorModule,
        ProductSummaryAccountLogoModule,
        ProductSummaryFooterModule,
        ProductKindNameModule,
        ProductNumberModule,
        TooltipModule,
        ValidationModule] });
ProductSummaryManageAccountsWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryManageAccountsWidgetModule, providers: [ManageAccountsService, ProductsExternalInfoService], imports: [[
            CommonModule,
            FormsModule,
            BackbaseCoreModule.withConfig({
                classMap: {
                    ProductSummaryManageAccountsWidgetComponent,
                },
            }),
            ProductSummaryCommon,
            ...uiModules,
        ], EmptyStateModule,
        LoadingIndicatorModule,
        InputInlineEditModule,
        PaymentCardNumberModule,
        AmountModule,
        SwitchModule,
        NotificationModule,
        ArrangementStateIndicatorModule,
        ProductSummaryAccountLogoModule,
        ProductSummaryFooterModule,
        ProductKindNameModule,
        ProductNumberModule,
        TooltipModule,
        ValidationModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryManageAccountsWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        FormsModule,
                        BackbaseCoreModule.withConfig({
                            classMap: {
                                ProductSummaryManageAccountsWidgetComponent,
                            },
                        }),
                        ProductSummaryCommon,
                        ...uiModules,
                    ],
                    declarations: [...components, ...directives],
                    exports: [...components, ...directives, ...uiModules],
                    providers: [ManageAccountsService, ProductsExternalInfoService],
                }]
        }] });
