// @backbase/messages-ui-ang/fesm2015/backbase-messages-ui-ang.js
import * as i0 from '@angular/core';
import { EventEmitter, Component, ChangeDetectionStrategy, Input, Output, NgModule } from '@angular/core';
import * as i2 from '@angular/common';
import { CommonModule } from '@angular/common';
import * as i2$1 from '@angular/router';
import { RouterModule } from '@angular/router';
import { ButtonModule } from '@backbase/ui-ang/button';
import * as i1 from '@backbase/ui-ang/header';
import { HeaderModule } from '@backbase/ui-ang/header';

class MessagesHeadingComponent {
    constructor() {
        /**
         * Heading title
         */
        this.headingTitle = '';
        /**
         * Heading type
         */
        this.headingType = 'h1';
        /**
         * Custom heading classes
         */
        this.headingClasses = '';
        /**
         * Custom heading wrapper classes
         */
        this.headingWrapperClasses = '';
        /**
         * Heading button text
         */
        this.buttonText = '';
        /**
         * Flag indicates heading button visibility
         */
        this.showButton = true;
        /**
         * Heading heading button press action
         */
        this.buttonAction = new EventEmitter();
    }
    onButtonClick() {
        this.buttonAction.emit();
    }
}
/** @nocollapse */ MessagesHeadingComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesHeadingComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ MessagesHeadingComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: MessagesHeadingComponent, selector: "bb-messages-heading", inputs: { headingTitle: "headingTitle", headingType: "headingType", headingClasses: "headingClasses", headingWrapperClasses: "headingWrapperClasses", buttonText: "buttonText", showButton: "showButton" }, outputs: { buttonAction: "buttonAction" }, ngImport: i0, template: "<div [ngClass]=\"'bb-heading-widget ' + headingWrapperClasses\" data-role=\"messages-heading\">\n  <bb-header-ui\n    [headingType]=\"headingType\"\n    headingClasses=\"bb-heading-widget__heading\"\n    [heading]=\"headingTitle\"\n  ></bb-header-ui>\n  <button\n    *ngIf=\"showButton\"\n    data-role=\"messages-heading-button\"\n    (click)=\"onButtonClick()\"\n    class=\"bb-heading-widget__button btn btn-primary\"\n  >\n    <span>{{ buttonText }}</span>\n  </button>\n</div>\n", components: [{ type: i1.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }], directives: [{ type: i2.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesHeadingComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-messages-heading',
                    templateUrl: './messages-heading.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { headingTitle: [{
                type: Input
            }], headingType: [{
                type: Input
            }], headingClasses: [{
                type: Input
            }], headingWrapperClasses: [{
                type: Input
            }], buttonText: [{
                type: Input
            }], showButton: [{
                type: Input
            }], buttonAction: [{
                type: Output
            }] } });

class MessagesHeadingModule {
}
/** @nocollapse */ MessagesHeadingModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesHeadingModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ MessagesHeadingModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesHeadingModule, declarations: [MessagesHeadingComponent], imports: [CommonModule, RouterModule, HeaderModule, ButtonModule], exports: [MessagesHeadingComponent] });
/** @nocollapse */ MessagesHeadingModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesHeadingModule, imports: [[CommonModule, RouterModule, HeaderModule, ButtonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesHeadingModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [MessagesHeadingComponent],
                    imports: [CommonModule, RouterModule, HeaderModule, ButtonModule],
                    exports: [MessagesHeadingComponent],
                }]
        }] });

class MessagesTabsComponent {
}
/** @nocollapse */ MessagesTabsComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesTabsComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ MessagesTabsComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: MessagesTabsComponent, selector: "bb-messages-tabs", inputs: { tabs: "tabs" }, ngImport: i0, template: "<div class=\"bb-tab-container\" data-role=\"messages-tabs-container\">\n  <div class=\"bb-tab-container__tabs nav nav-tabs\" role=\"tablist\">\n    <a\n      *ngFor=\"let tab of tabs\"\n      [routerLink]=\"tab.route\"\n      routerLinkActive=\"active\"\n      class=\"nav-link nav-item\"\n      data-role=\"messages-tab-link\"\n      role=\"tab\"\n    >\n      {{ tab.title }}\n    </a>\n  </div>\n</div>\n\n<div class=\"bb-tab-container__content tab-content\" role=\"tabpanel\" aria-live=\"polite\" aria-atomic=\"false\">\n  <router-outlet></router-outlet>\n</div>\n", directives: [{ type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { type: i2$1.RouterLinkWithHref, selector: "a[routerLink],area[routerLink]", inputs: ["routerLink", "target", "queryParams", "fragment", "queryParamsHandling", "preserveFragment", "skipLocationChange", "replaceUrl", "state", "relativeTo"] }, { type: i2$1.RouterLinkActive, selector: "[routerLinkActive]", inputs: ["routerLinkActiveOptions", "routerLinkActive"], exportAs: ["routerLinkActive"] }, { type: i2$1.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesTabsComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-messages-tabs',
                    templateUrl: './messages-tabs.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { tabs: [{
                type: Input
            }] } });

class MessagesTabsModule {
}
/** @nocollapse */ MessagesTabsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesTabsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ MessagesTabsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesTabsModule, declarations: [MessagesTabsComponent], imports: [CommonModule, RouterModule], exports: [MessagesTabsComponent] });
/** @nocollapse */ MessagesTabsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesTabsModule, imports: [[CommonModule, RouterModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesTabsModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [MessagesTabsComponent],
                    imports: [CommonModule, RouterModule],
                    exports: [MessagesTabsComponent],
                }]
        }] });

const modules = [MessagesHeadingModule, MessagesTabsModule];
class MessagesCollectionUiAngModule {
}
/** @nocollapse */ MessagesCollectionUiAngModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesCollectionUiAngModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ MessagesCollectionUiAngModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesCollectionUiAngModule, imports: [CommonModule], exports: [MessagesHeadingModule, MessagesTabsModule] });
/** @nocollapse */ MessagesCollectionUiAngModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesCollectionUiAngModule, imports: [[CommonModule], MessagesHeadingModule, MessagesTabsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesCollectionUiAngModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    exports: [...modules],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { MessagesCollectionUiAngModule, MessagesHeadingComponent, MessagesHeadingModule, MessagesTabsComponent, MessagesTabsModule };
