// @backbase/manage-payees-journey-ang/esm2015/src/services/billpay-manage-payee-state.service.js
import { Injectable } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import * as i0 from "@angular/core";
export class BillpayManagePayeeStateService {
    constructor() {
        this.manualPayeeStateSource$ = new BehaviorSubject(undefined);
        this.manualPayeeFormState = this.manualPayeeStateSource$.asObservable();
        this.electronicPayeeStateSource = new BehaviorSubject(undefined);
        this.electronicPayeeFormState = this.electronicPayeeStateSource.asObservable();
        this.isFormProcessing = false;
    }
    setManualPayee(state) {
        this.manualPayeeStateSource$.next(state);
    }
    toManualFormState(data) {
        return Object.assign(Object.assign({ name: data.name, address: data.address, phoneNumber: data.phoneNumber || '' }, (data.accountNumber && { accountNumber: data.accountNumber })), (data.nickName && { nickName: data.nickName }));
    }
    resetManualPayee() {
        this.setManualPayee();
    }
    setElectronicPayee(state) {
        this.electronicPayeeStateSource.next(state);
    }
    resetElectronicPayee() {
        this.setElectronicPayee();
    }
    electronicGetResponseToForm(payeeResponse) {
        const { name, nickName, accountNumber } = payeeResponse.payee;
        return {
            name,
            nickName,
            accountNumber,
        };
    }
}
BillpayManagePayeeStateService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayManagePayeeStateService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
BillpayManagePayeeStateService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayManagePayeeStateService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayManagePayeeStateService, decorators: [{
            type: Injectable
        }] });
