// @backbase/manage-payees-journey-ang/esm2015/src/services/billpay-payee-summary-widget.service.js
import { Injectable } from '@angular/core';
import { catchError, map, tap } from 'rxjs/operators';
import { BehaviorSubject, EMPTY } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/billpay";
export const defaultDeleteParams = {
    cancelPendingPayments: false,
};
export class BillpayPayeeSummaryWidgetService {
    /** @internal */
    constructor(payeesSummaryData, payeesData) {
        this.payeesSummaryData = payeesSummaryData;
        this.payeesData = payeesData;
        this.currentPayeeSummaryLoadingState = new BehaviorSubject('');
    }
    getPayees() {
        this.currentPayeeSummaryLoadingState.next('loading');
        return this.payeesSummaryData.getBillPayPayeesSummary().pipe(map(this.getPayeeSummaries), tap(data => sortPayees(data)), tap(() => this.currentPayeeSummaryLoadingState.next('')), catchError(() => {
            this.currentPayeeSummaryLoadingState.next('error');
            return EMPTY;
        }));
    }
    deletePayee(id, electronic) {
        return electronic
            ? this.payeesData.deleteElectronicPayeeById(Object.assign(Object.assign({}, defaultDeleteParams), { id }))
            : this.payeesData.deletePayeeById(Object.assign(Object.assign({}, defaultDeleteParams), { id }));
    }
    getFullSummary(id) {
        return this.payeesData.getPayeeById({ id }).pipe(map(a => a.payee));
    }
    getPayeeSummaries(response) {
        // eslint-disable-next-line no-null/no-null
        if (response.payeeSummaries === null) {
            throw new Error('Response body not set');
        }
        return response.payeeSummaries ? response.payeeSummaries : [];
    }
    get payeesSummaryLoadingState() {
        return this.currentPayeeSummaryLoadingState;
    }
}
BillpayPayeeSummaryWidgetService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayPayeeSummaryWidgetService, deps: [{ token: i1.BillPayPayeesSummaryService }, { token: i1.BillPayPayeesService }], target: i0.ɵɵFactoryTarget.Injectable });
BillpayPayeeSummaryWidgetService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayPayeeSummaryWidgetService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayPayeeSummaryWidgetService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.BillPayPayeesSummaryService }, { type: i1.BillPayPayeesService }]; } });
function getPayeeDisplayName(payee) {
    return (payee.nickName || payee.name).toLowerCase();
}
function isEbillEnabled(payee) {
    return payee.ebill ? payee.ebill.enabled : false;
}
function sortPayees(data) {
    if (data) {
        data.sort(function (payee1, payee2) {
            if (isEbillEnabled(payee1) && !isEbillEnabled(payee2))
                return -1;
            if (!isEbillEnabled(payee1) && isEbillEnabled(payee2))
                return 1;
            return getPayeeDisplayName(payee1) > getPayeeDisplayName(payee2) ? 1 : -1;
        });
    }
}
