// @backbase/manage-payees-journey-ang/esm2015/src/services/manage-payees-navigation.service.js
import { Injectable, Optional } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "./manage-payees-journey-config.service";
import * as i3 from "../communication.service";
export class ManagePayeesNavigationService {
    constructor(router, route, config, communicationService) {
        this.router = router;
        this.route = route;
        this.config = config;
        this.communicationService = communicationService;
        this.defaultRelative = { relativeTo: this.route };
    }
    navigateToPayeeList(extras) {
        var _a;
        if (this.config.multipleBillsMode) {
            (_a = this.communicationService) === null || _a === void 0 ? void 0 : _a.navigateToMultiplePaymentsForm();
        }
        else {
            this.router.navigate(['./'], extras || this.defaultRelative);
        }
    }
    navigateToSearch(extras) {
        this.router.navigate(['payees'], extras || this.defaultRelative);
    }
    navigateToManualPayeeForm(extras) {
        this.router.navigate(['payees/manual'], extras || this.defaultRelative);
    }
    navigateToManualPayeeReview(extras) {
        this.router.navigate(['payees/manual/review'], extras || this.defaultRelative);
    }
    navigateToEditManualPayeeForm(id, extras) {
        this.router.navigate(['payees/manual', 'edit', id], extras || this.defaultRelative);
    }
    navigateToEditManualReview(id, extras) {
        this.router.navigate(['payees/manual/edit', id, 'review'], extras || this.defaultRelative);
    }
    navigateToElectronicPayeeForm(id, extras) {
        this.router.navigate(['payees/electronic', id], Object.assign(Object.assign({}, this.defaultRelative), extras));
    }
    navigateToElectronicPayeeReview(id, extras) {
        this.router.navigate(['payees/electronic', id, 'review'], Object.assign(Object.assign({}, this.defaultRelative), extras));
    }
    navigateToEditElectronicPayeeForm(id, extras) {
        this.router.navigate(['payees/electronic', 'edit', id], extras || this.defaultRelative);
    }
    navigateToEditElectronicPayeeReview(id, extras) {
        this.router.navigate(['payees/electronic', 'edit', id, 'review'], extras || this.defaultRelative);
    }
    navigateToPayeeSummary(payeeID, extras) {
        this.router.navigate(['payee-summary', { payeeID }], extras || this.defaultRelative);
    }
    navigateToPayveris(payeeID, extras) {
        this.router.navigate(['ebills', { payeeID }], extras || this.defaultRelative);
    }
    // External Navigations
    navigateToRecurringPayment(id) {
        var _a;
        (_a = this.communicationService) === null || _a === void 0 ? void 0 : _a.navigateToRecurringPayment(id);
    }
    navigateToOneOffPayment(id) {
        var _a;
        (_a = this.communicationService) === null || _a === void 0 ? void 0 : _a.navigateToOneOffPayment(id);
    }
    navigateToPendingPayments() {
        var _a;
        (_a = this.communicationService) === null || _a === void 0 ? void 0 : _a.navigateToPaymentsList();
    }
}
ManagePayeesNavigationService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePayeesNavigationService, deps: [{ token: i1.Router }, { token: i1.ActivatedRoute }, { token: i2.ManagePayeesJourneyConfigService }, { token: i3.CommunicationService, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
ManagePayeesNavigationService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePayeesNavigationService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePayeesNavigationService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.Router }, { type: i1.ActivatedRoute }, { type: i2.ManagePayeesJourneyConfigService }, { type: i3.CommunicationService, decorators: [{
                    type: Optional
                }] }]; } });
