// @backbase/manage-payees-journey-ang/esm2015/src/services/manage-payees-journey-config.service.js
import { Injectable, Optional, Inject, InjectionToken } from '@angular/core';
import * as i0 from "@angular/core";
export const ManagePayeesJourneyConfigurationToken = new InjectionToken('ManagePayeesJourneyConfigurationToken injection token');
const configDefaults = {
    notificationDismissTime: 5000,
    multipleBillsMode: false,
};
export class ManagePayeesJourneyConfigService {
    constructor(config) {
        config = config || {};
        this._config = Object.assign(Object.assign({}, configDefaults), config);
    }
    get defaults() {
        return configDefaults;
    }
    get notificationDismissTime() {
        return this._config.notificationDismissTime;
    }
    get multipleBillsMode() {
        return this._config.multipleBillsMode;
    }
}
ManagePayeesJourneyConfigService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePayeesJourneyConfigService, deps: [{ token: ManagePayeesJourneyConfigurationToken, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
ManagePayeesJourneyConfigService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePayeesJourneyConfigService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePayeesJourneyConfigService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [ManagePayeesJourneyConfigurationToken]
                }] }]; } });
