// @backbase/ui-ang/esm2015/control-error-handler/directives/form-submit.directive.js
import { Directive } from '@angular/core';
import { fromEvent } from 'rxjs';
import { shareReplay } from 'rxjs/operators';
import * as i0 from "@angular/core";
export class FormSubmitDirective {
    constructor(elementRef) {
        this.elementRef = elementRef;
        this.element = this.elementRef.nativeElement;
        this.submit = fromEvent(this.element, 'submit').pipe(shareReplay({ bufferSize: 1, refCount: true }));
    }
}
FormSubmitDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FormSubmitDirective, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive });
FormSubmitDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: FormSubmitDirective, selector: "form[bbFormSubmit]", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: FormSubmitDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'form[bbFormSubmit]',
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; } });
