// @backbase/cards-travel-notice-widget-ang/esm2015/src/services/travel-notice-dates-helper.service.js
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
export class TravelNoticeDatesHelperService {
    isPastDate(date) {
        const time = new Date(date);
        return this.trucanteDate(time) < this.trucanteDate(new Date());
    }
    isPastDateOrToday(date) {
        return this.isPastDate(date) || this.areDateEqual(new Date(), new Date(date));
    }
    isUpcomingDate(date) {
        const time = new Date(date);
        return this.trucanteDate(time) > this.trucanteDate(new Date());
    }
    isUpcomingDateOrToday(date) {
        return this.isUpcomingDate(date) || this.areDateEqual(new Date(), new Date(date));
    }
    isDateInBetween(date1, date2) {
        return this.isPastDateOrToday(date1) && this.isUpcomingDateOrToday(date2);
    }
    areDateEqual(date1, date2) {
        return this.trucanteDate(date1) === this.trucanteDate(date2);
    }
    trucanteDate(date) {
        date.setHours(0);
        date.setMinutes(0);
        date.setSeconds(0);
        date.setMilliseconds(0);
        return date.getTime();
    }
}
TravelNoticeDatesHelperService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: TravelNoticeDatesHelperService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
TravelNoticeDatesHelperService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: TravelNoticeDatesHelperService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: TravelNoticeDatesHelperService, decorators: [{
            type: Injectable
        }] });
