// @backbase/ui-ang/esm2015/dropdown-multi-select/dropdown-multi-select-option.component.js
import { ChangeDetectionStrategy, Component, Inject, InjectionToken, Input, } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/services";
export const BB_MULTI_SELECT_CHANGE_DETECTION_REF_TOKEN = new InjectionToken('BB_MULTI_SELECT_SINGLE_SELECT');
/**
 * @name MultiSelectSingleSelectOption
 *
 * @description
 * Component that represents a single option that is part of the multi select component.
 */
export class DropdownMultiSelectOptionComponent {
    constructor(dropdownCDRef, domAttributesService) {
        this.dropdownCDRef = dropdownCDRef;
        this.domAttributesService = domAttributesService;
        /**
         * The value of the native option element
         */
        this.value = '';
        /**
         * The label of the option item
         */
        this.label = '';
        this.state = false;
        this._id = this.domAttributesService.generateId();
    }
    /**
     * The id for the input. Defaults to unique string.
     */
    set id(value) {
        this._id = value;
    }
    get id() {
        return this._id;
    }
    // NOTES: trigger of detectChanges is needed
    // because DropdownSingleSelectComponent has changeDetection
    // strategy onPush, hence it reacts only to changes
    // of its own @Inputs, but not @Inputs() of options component
    ngOnChanges({ value = {}, label = {} }) {
        const { firstChange: firstChangeValue } = value;
        const { firstChange: firstChangeLabel } = label;
        if (firstChangeLabel && firstChangeValue) {
            return;
        }
        this.dropdownCDRef.detectChanges();
    }
}
DropdownMultiSelectOptionComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownMultiSelectOptionComponent, deps: [{ token: BB_MULTI_SELECT_CHANGE_DETECTION_REF_TOKEN }, { token: i1.DomAttributesService }], target: i0.ɵɵFactoryTarget.Component });
DropdownMultiSelectOptionComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: DropdownMultiSelectOptionComponent, selector: "bb-dropdown-multi-select-option-ui", inputs: { id: "id", value: "value", label: "label", state: "state" }, usesOnChanges: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownMultiSelectOptionComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-dropdown-multi-select-option-ui',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    template: '',
                }]
        }], ctorParameters: function () { return [{ type: i0.ChangeDetectorRef, decorators: [{
                    type: Inject,
                    args: [BB_MULTI_SELECT_CHANGE_DETECTION_REF_TOKEN]
                }] }, { type: i1.DomAttributesService }]; }, propDecorators: { id: [{
                type: Input
            }], value: [{
                type: Input
            }], label: [{
                type: Input
            }], state: [{
                type: Input
            }] } });
