// @backbase/ui-ang/esm2015/dropdown-multi-select/dropdown-multi-select.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { IconModule } from '@backbase/ui-ang/icon';
import { ReactiveFormsModule } from '@angular/forms';
import { DropdownMultiSelectComponent } from './dropdown-multi-select.component';
import { InputCheckboxModule } from '@backbase/ui-ang/input-checkbox';
import { DropdownPanelModule } from '@backbase/ui-ang/dropdown-panel';
import { DropdownMultiSelectOptionComponent } from './dropdown-multi-select-option.component';
import { ButtonModule } from '@backbase/ui-ang/button';
import * as i0 from "@angular/core";
export class DropdownMultiSelectModule {
}
DropdownMultiSelectModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownMultiSelectModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
DropdownMultiSelectModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownMultiSelectModule, declarations: [DropdownMultiSelectComponent, DropdownMultiSelectOptionComponent], imports: [CommonModule, IconModule, InputCheckboxModule, DropdownPanelModule, ReactiveFormsModule, ButtonModule], exports: [DropdownMultiSelectComponent, DropdownMultiSelectOptionComponent] });
DropdownMultiSelectModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownMultiSelectModule, imports: [[CommonModule, IconModule, InputCheckboxModule, DropdownPanelModule, ReactiveFormsModule, ButtonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownMultiSelectModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, IconModule, InputCheckboxModule, DropdownPanelModule, ReactiveFormsModule, ButtonModule],
                    declarations: [DropdownMultiSelectComponent, DropdownMultiSelectOptionComponent],
                    exports: [DropdownMultiSelectComponent, DropdownMultiSelectOptionComponent],
                }]
        }] });
