// @backbase/payord-stop-checks-widget-ang/esm2015/lib/store/store.js
import { StopCheckRequestType } from './../models/data';
export var ActionTypes;
(function (ActionTypes) {
    ActionTypes["INIT_REQUEST"] = "[STOP CHECK] Initiate request";
    ActionTypes["REVIEW_REQUEST"] = "[STOP CHECK] Review request";
    ActionTypes["EDIT_REQUEST"] = "[STOP CHECK] Edit request";
    ActionTypes["MARK_REQUEST_AS_SUBMITTED"] = "[STOP CHECK] Mark request as submitted";
    ActionTypes["DISCARD"] = "[STOP CHECK] Discard review";
})(ActionTypes || (ActionTypes = {}));
export var RequestStatus;
(function (RequestStatus) {
    RequestStatus["INIT"] = "INIT";
    RequestStatus["REVIEW"] = "REVIEW";
    RequestStatus["REQUEST_IS_SUBMITTED"] = "REQUEST_IS_SUBMITTED";
    RequestStatus["EDIT_MODE"] = "EDIT_MODE";
    RequestStatus["DISCARD"] = "DISCARD";
})(RequestStatus || (RequestStatus = {}));
export const state = {
    requestInfo: {
        payFromAccount: {
            id: '',
            name: '',
            balance: 0,
            currency: '',
            accountNumber: '',
            accountType: 'BBAN',
        },
        checkNumber: '',
        firstCheckNumber: '',
        lastCheckNumber: '',
        stopRangeType: StopCheckRequestType.SINGLE,
        amountGroup: undefined,
        payeeName: '',
        dateIssued: '',
        expirationDate: '',
        stopReason: '',
        otherReason: '',
    },
    status: RequestStatus.INIT,
};
