// @backbase/loans-http-ang/esm2015/api/paymentOptions.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { LOANS_BASE_PATH } from '../variables';
import { LoansConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class PaymentOptionsHttpService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new LoansConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    getPaymentOptions(requestParameters, observe = 'body', reportProgress = false, options) {
        const _loanId = requestParameters["loanId"];
        if (_loanId === null || _loanId === undefined) {
            throw new Error('Required parameter loanId was null or undefined when calling getPaymentOptions.');
        }
        const _operationType = requestParameters["operationType"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_operationType) {
            _operationType.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'operationType');
            });
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/loans/${encodeURIComponent(String(_loanId))}/payment-options`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getPaymentOptionsUrl(requestParameters) {
        const _loanId = requestParameters["loanId"];
        if (_loanId === null || _loanId === undefined) {
            throw new Error('Required parameter loanId was null or undefined when calling getPaymentOptions.');
        }
        const _operationType = requestParameters["operationType"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_operationType) {
            _operationType.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'operationType');
            });
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v1/loans/${encodeURIComponent(String(_loanId))}/payment-options${queryString ? `?${queryString}` : ''}`;
    }
}
PaymentOptionsHttpService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentOptionsHttpService, deps: [{ token: i1.HttpClient }, { token: LOANS_BASE_PATH, optional: true }, { token: i2.LoansConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
PaymentOptionsHttpService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentOptionsHttpService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentOptionsHttpService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [LOANS_BASE_PATH]
                }] }, { type: i2.LoansConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
