// @backbase/loans-http-ang/esm2015/api/paymentOptions.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const PaymentOptionsHttpServiceGetPaymentOptionsMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/loans/{loanId}/payment-options",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "$schema": "../schemas/payment-options.yaml",
                    "paymentOptions": [{
                            "name": "LoanPayment",
                            "operationType": "Credit",
                            "allowed": true
                        }, {
                            "name": "LoanAdvance",
                            "operationType": "Debit",
                            "allowed": false,
                            "errorCodes": ["draw.amount.is.not.allowed"]
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "paymentOptions": [{
                            "allowed": true,
                            "errorCodes": ["errorCodes", "errorCodes"],
                            "name": "name"
                        }, {
                            "allowed": true,
                            "errorCodes": ["errorCodes", "errorCodes"],
                            "name": "name"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const PaymentOptionsHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v1/loans/{loanId}/payment-options",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "$schema": "../schemas/payment-options.yaml",
                    "paymentOptions": [{
                            "name": "LoanPayment",
                            "operationType": "Credit",
                            "allowed": true
                        }, {
                            "name": "LoanAdvance",
                            "operationType": "Debit",
                            "allowed": false,
                            "errorCodes": ["draw.amount.is.not.allowed"]
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "paymentOptions": [{
                            "allowed": true,
                            "errorCodes": ["errorCodes", "errorCodes"],
                            "name": "name"
                        }, {
                            "allowed": true,
                            "errorCodes": ["errorCodes", "errorCodes"],
                            "name": "name"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
]);
