// @backbase/loans-http-ang/esm2015/api/payment.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const PaymentHttpServiceCalculatePaymentAmountMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/loans/{loanId}/calculate-payment-amount",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "$schema": "../schemas/calculated-payment-amount.yaml",
                    "paymentOption": "LOAN_PAYMENT",
                    "paymentDate": "2021-12-22",
                    "currentPaymentAmount": "100.00000",
                    "fees": [{
                            "feeType": "Transaction fee",
                            "amount": "11.00000",
                            "sameTransaction": false
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const PaymentHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v1/loans/{loanId}/calculate-payment-amount",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "$schema": "../schemas/calculated-payment-amount.yaml",
                    "paymentOption": "LOAN_PAYMENT",
                    "paymentDate": "2021-12-22",
                    "currentPaymentAmount": "100.00000",
                    "fees": [{
                            "feeType": "Transaction fee",
                            "amount": "11.00000",
                            "sameTransaction": false
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
]);
