// @backbase/account-statement-list-widget-ang/esm2015/src/services/account-statements-list-widget-properties.service.js
import { Injectable, Optional } from '@angular/core';
import { BehaviorSubject, combineLatest, of } from 'rxjs';
import { map } from 'rxjs/operators';
import { AccountStatementBasePropertiesService, modelItemToBoolean, PaginationType, } from '@backbase/account-statement-common-ang';
import { ListWidgetProperties } from '../constants/widget-properties';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
export class AccountStatementListWidgetPropertiesService extends AccountStatementBasePropertiesService {
    constructor(itemModel) {
        super(itemModel);
        this.showDescription$$ = new BehaviorSubject(undefined);
        /* eslint-disable @typescript-eslint/member-ordering */
        this.showDescription$ = this.getWidgetProperty$(this.showDescription$$, this.showDescriptionIM$);
        this.widgetProperties$ = combineLatest([
            this.pageSize$,
            this.sortingOrder$,
            this.sortingDirection$,
            this.notificationDismissTime$,
            this.showCategory$,
            this.showDescription$,
            this.relativeDateLimit$,
            this.absoluteDateLimit$,
            this.paginationType$,
        ]).pipe(map(([pageSize, sortingOrder, sortingDirection, notificationDismissTimeProperty, showCategory, showDescription, relativeDateLimit, absoluteDateLimit, paginationType,]) => ({
            pageSize,
            sortingOrder,
            sortingDirection,
            notificationDismissTimeProperty,
            showCategory,
            showDescription,
            relativeDateLimit,
            absoluteDateLimit,
            paginationType,
        })));
    }
    /* eslint-enable @typescript-eslint/member-ordering */
    setShowDescription(value) {
        this.showDescription$$.next(value);
    }
    getPaginationTypeIM$() {
        var _a;
        const defaultValue = PaginationType.LoadMore;
        return ((_a = this.itemModel) === null || _a === void 0 ? void 0 : _a.property(ListWidgetProperties.PAGINATION_TYPE, defaultValue)) || of(defaultValue);
    }
    get showDescriptionIM$() {
        var _a;
        const defaultValue = true;
        return (((_a = this.itemModel) === null || _a === void 0 ? void 0 : _a.property(ListWidgetProperties.SHOW_DESCRIPTION, defaultValue).pipe(map(modelItemToBoolean))) ||
            of(defaultValue));
    }
}
AccountStatementListWidgetPropertiesService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementListWidgetPropertiesService, deps: [{ token: i1.ItemModel, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
AccountStatementListWidgetPropertiesService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementListWidgetPropertiesService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementListWidgetPropertiesService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.ItemModel, decorators: [{
                    type: Optional
                }] }]; } });
