// @backbase/ui-ang/esm2015/_example/iban-pipe/showcase/iban-pipe-showcase.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/iban-pipe";
let IbanPipeShowcaseComponent = class IbanPipeShowcaseComponent {
    constructor() {
        this.iban = 'NL81TRIO0212471066';
    }
};
IbanPipeShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IbanPipeShowcaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
IbanPipeShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: IbanPipeShowcaseComponent, selector: "bb-iban-pipe-ui-showcase", ngImport: i0, template: "<div><strong>IBAN number formated:</strong> {{ iban | bbIban }}</div>\n", pipes: { "bbIban": i1.BbIbanPipe } });
IbanPipeShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
], IbanPipeShowcaseComponent);
export { IbanPipeShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IbanPipeShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-iban-pipe-ui-showcase',
                    templateUrl: './iban-pipe-showcase.component.html',
                }]
        }] });
