// @backbase/ui-ang/esm2015/_example/iban-pipe/showcase/iban-pipe-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { IbanPipeShowcaseComponent } from './iban-pipe-showcase.component';
import { BbIbanPipeModule } from '@backbase/ui-ang/iban-pipe';
import * as i0 from "@angular/core";
export class IbanPipeShowcaseModule {
}
IbanPipeShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IbanPipeShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IbanPipeShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IbanPipeShowcaseModule, declarations: [IbanPipeShowcaseComponent], imports: [CommonModule, BbIbanPipeModule] });
IbanPipeShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IbanPipeShowcaseModule, imports: [[CommonModule, BbIbanPipeModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IbanPipeShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, BbIbanPipeModule],
                    declarations: [IbanPipeShowcaseComponent],
                    entryComponents: [IbanPipeShowcaseComponent],
                }]
        }] });
