// @backbase/messages-common/esm2015/src/components/create-message/employee-mailout-form/employee-create-mailout-form.service.js
import { Injectable } from '@angular/core';
import { of } from 'rxjs';
import { catchError } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/messages-v5-http-ang";
export class EmployeeCreateMailoutFormService {
    constructor(messagesService, employeeService) {
        this.messagesService = messagesService;
        this.employeeService = employeeService;
        this.maxPaginationItemsNum = 65535;
    }
    /**
     * Get topics
     *
     * @returns Get topics stream
     */
    getTopics() {
        return this.employeeService
            .getTopicsEmployee({ size: this.maxPaginationItemsNum, subscribed: true })
            .pipe(catchError(() => of([])));
    }
    /**
     * Create message
     *
     * @param body Create message request body
     * @returns Create message stream
     */
    createMessage(body) {
        return this.messagesService.postMailout({ mailoutPostRequestBody: body });
    }
}
EmployeeCreateMailoutFormService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: EmployeeCreateMailoutFormService, deps: [{ token: i1.MailoutHttpService }, { token: i1.EmployeeHttpService }], target: i0.ɵɵFactoryTarget.Injectable });
EmployeeCreateMailoutFormService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: EmployeeCreateMailoutFormService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: EmployeeCreateMailoutFormService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.MailoutHttpService }, { type: i1.EmployeeHttpService }]; } });
