// @backbase/messages-common/esm2015/src/components/create-message/employee-mailout-form/employee-create-mailout-form.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { ButtonModule } from '@backbase/ui-ang/button';
import { DropdownSingleSelectModule } from '@backbase/ui-ang/dropdown-single-select';
import { InputCheckboxModule } from '@backbase/ui-ang/input-checkbox';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { InputFileModule } from '@backbase/ui-ang/input-file';
import { FileAttachmentModule } from '@backbase/ui-ang/file-attachment';
import { RichTextEditorModule } from '@backbase/ui-ang/rich-text-editor';
import { EmployeeCreateMailoutFormComponent, EmployeeCreateMailoutFormCustomizableDirective, EmployeeCreateMailoutFormErrorNotificationHeaderCustomizableDirective, EmployeeCreateMailoutFormSuccessNotificationHeaderCustomizableDirective, } from './employee-create-mailout-form.component';
import { EmployeeCreateMailoutFormService } from './employee-create-mailout-form.service';
import { MessagesManipulationConfirmModalModule } from '../../messages-manipulation-confirm-modal/messages-manipulation-confirm-modal.module';
import { MessagesUploadAttachmentsModule } from '../../upload-attachments/messages-upload-attachments.module';
import { PreventBubbleDownModule } from '../../../directives/prevent-bubble-down/prevent-bubble-down.module';
import * as i0 from "@angular/core";
const components = [EmployeeCreateMailoutFormComponent];
const directives = [
    EmployeeCreateMailoutFormCustomizableDirective,
    EmployeeCreateMailoutFormErrorNotificationHeaderCustomizableDirective,
    EmployeeCreateMailoutFormSuccessNotificationHeaderCustomizableDirective,
];
const uiModules = [
    DropdownSingleSelectModule,
    FileAttachmentModule,
    InputFileModule,
    InputValidationMessageModule,
    InputTextModule,
    RichTextEditorModule,
    ButtonModule,
    InputCheckboxModule,
    LoadButtonModule,
];
const innerModules = [ReactiveFormsModule, MessagesManipulationConfirmModalModule, MessagesUploadAttachmentsModule];
export class EmployeeCreateMailoutFormModule {
}
EmployeeCreateMailoutFormModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: EmployeeCreateMailoutFormModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
EmployeeCreateMailoutFormModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: EmployeeCreateMailoutFormModule, declarations: [EmployeeCreateMailoutFormComponent, EmployeeCreateMailoutFormCustomizableDirective,
        EmployeeCreateMailoutFormErrorNotificationHeaderCustomizableDirective,
        EmployeeCreateMailoutFormSuccessNotificationHeaderCustomizableDirective], imports: [CommonModule, PreventBubbleDownModule, DropdownSingleSelectModule,
        FileAttachmentModule,
        InputFileModule,
        InputValidationMessageModule,
        InputTextModule,
        RichTextEditorModule,
        ButtonModule,
        InputCheckboxModule,
        LoadButtonModule, ReactiveFormsModule, MessagesManipulationConfirmModalModule, MessagesUploadAttachmentsModule], exports: [EmployeeCreateMailoutFormComponent, EmployeeCreateMailoutFormCustomizableDirective,
        EmployeeCreateMailoutFormErrorNotificationHeaderCustomizableDirective,
        EmployeeCreateMailoutFormSuccessNotificationHeaderCustomizableDirective, DropdownSingleSelectModule,
        FileAttachmentModule,
        InputFileModule,
        InputValidationMessageModule,
        InputTextModule,
        RichTextEditorModule,
        ButtonModule,
        InputCheckboxModule,
        LoadButtonModule, ReactiveFormsModule, MessagesManipulationConfirmModalModule, MessagesUploadAttachmentsModule] });
EmployeeCreateMailoutFormModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: EmployeeCreateMailoutFormModule, providers: [EmployeeCreateMailoutFormService], imports: [[CommonModule, PreventBubbleDownModule, ...uiModules, ...innerModules], DropdownSingleSelectModule,
        FileAttachmentModule,
        InputFileModule,
        InputValidationMessageModule,
        InputTextModule,
        RichTextEditorModule,
        ButtonModule,
        InputCheckboxModule,
        LoadButtonModule, ReactiveFormsModule, MessagesManipulationConfirmModalModule, MessagesUploadAttachmentsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: EmployeeCreateMailoutFormModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, PreventBubbleDownModule, ...uiModules, ...innerModules],
                    declarations: [...components, ...directives],
                    providers: [EmployeeCreateMailoutFormService],
                    exports: [...components, ...directives, ...uiModules, ...innerModules],
                }]
        }] });
