// @backbase/payment-orders-ang/esm2015/lib/ui/directives/export-pdf/export-pdf.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ExportPdfDirective } from './export-pdf.directive';
import { NotificationService } from '@backbase/ui-ang/notification';
import * as i0 from "@angular/core";
export class ExportPdfModule {
}
/** @nocollapse */ ExportPdfModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ExportPdfModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ ExportPdfModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ExportPdfModule, declarations: [ExportPdfDirective], imports: [CommonModule], exports: [ExportPdfDirective] });
/** @nocollapse */ ExportPdfModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ExportPdfModule, providers: [NotificationService], imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ExportPdfModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ExportPdfDirective],
                    imports: [CommonModule],
                    exports: [ExportPdfDirective],
                    providers: [NotificationService],
                }]
        }] });
