// @backbase/actions-retail-notification-preferences-journey-ang/esm2015/src/actions-retail-notification-preferences-journey-config.service.js
import { Inject, Injectable, InjectionToken, Optional } from '@angular/core';
import { secondsToMilliseconds } from '@backbase/actions-common-ang';
import { of } from 'rxjs';
import { map } from 'rxjs/operators';
import * as i0 from "@angular/core";
export const ActionsRetailNotificationPreferencesJourneyToken = new InjectionToken('ActionsRetailNotificationPreferencesJourneyConfiguration injection token');
export function provideConfig(defaults) {
    return {
        provide: defaults,
        useFactory: (custom) => (Object.assign(Object.assign({}, new defaults()), custom)),
        deps: [ActionsRetailNotificationPreferencesJourneyToken],
    };
}
export const configDefaults = {
    notificationDismissTime: 5,
    specificationIDs: '1, 4',
    apiMode: 'engagements',
};
export class ActionsRetailNotificationPreferencesJourneyConfigService {
    constructor(staticConfigOverrides) {
        this.staticConfig = Object.assign(Object.assign({}, configDefaults), staticConfigOverrides);
    }
    /**
     * Method to get notification dismiss time preference value.
     *
     * @returns notification dismiss time in seconds
     */
    get notificationDismissTime$() {
        return of(this.staticConfig.notificationDismissTime).pipe(map(Number), map(number => (isNaN(number) ? configDefaults.notificationDismissTime : number)), map(secondsToMilliseconds));
    }
    /**
     * Method to get specification IDs.
     *
     * @returns specification IDs array
     */
    get specificationIDs$() {
        return of(this.staticConfig.specificationIDs).pipe(map(data => data.replace(/\s/g, '').split(',')));
    }
    /**
     * Method to get api mode.
     *
     * @returns api mode string
     */
    get apiMode() {
        return this.staticConfig.apiMode;
    }
}
ActionsRetailNotificationPreferencesJourneyConfigService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsRetailNotificationPreferencesJourneyConfigService, deps: [{ token: ActionsRetailNotificationPreferencesJourneyToken, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
ActionsRetailNotificationPreferencesJourneyConfigService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsRetailNotificationPreferencesJourneyConfigService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsRetailNotificationPreferencesJourneyConfigService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [ActionsRetailNotificationPreferencesJourneyToken]
                }] }]; } });
