// @backbase/actions-retail-notification-preferences-journey-ang/esm2015/src/actions-retail-notification-preferences-journey.module.js
import { NgModule, Injector } from '@angular/core';
import { CommonModule } from '@angular/common';
import { provideRoutes, RouterModule } from '@angular/router';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { ActionsProductNotificationsSettingsWidgetModule } from '@backbase/actions-product-notifications-settings-widget-ang';
import { ActionsRetailNotificationPreferencesViewComponent } from './actions-retail-notification-preferences-view.component';
import { ProductSummaryListWidgetModule } from '@backbase/product-summary-list-widget-ang';
import { HeaderModule } from '@backbase/ui-ang/header';
import { ProductNotificationsSettingsComponent } from './components/product-notifications-settings/product-notifications-settings.component';
import { routes } from './actions-retail-notification-preferences.routes';
import { ActionsRetailNotificationPreferencesJourneyConfigService } from './actions-retail-notification-preferences-journey-config.service';
import { NotificationsPreferencesBaseDataService } from '@backbase/actions-common-ang';
import { ActionNotificationsPreferencesDataFactory } from './services/actions-notifications-preferences-data.factory';
import * as i0 from "@angular/core";
export class ActionsRetailNotificationPreferencesJourneyModule {
    static forRoot(data = { routes }) {
        return {
            ngModule: ActionsRetailNotificationPreferencesJourneyModule,
            providers: [provideRoutes([data.routes])],
        };
    }
}
ActionsRetailNotificationPreferencesJourneyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsRetailNotificationPreferencesJourneyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ActionsRetailNotificationPreferencesJourneyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsRetailNotificationPreferencesJourneyModule, declarations: [ActionsRetailNotificationPreferencesViewComponent, ProductNotificationsSettingsComponent], imports: [CommonModule,
        HeaderModule,
        BackbaseCoreModule,
        RouterModule,
        ActionsProductNotificationsSettingsWidgetModule,
        ProductSummaryListWidgetModule] });
ActionsRetailNotificationPreferencesJourneyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsRetailNotificationPreferencesJourneyModule, providers: [
        ActionsRetailNotificationPreferencesJourneyConfigService,
        {
            provide: NotificationsPreferencesBaseDataService,
            useFactory: ActionNotificationsPreferencesDataFactory,
            deps: [Injector],
        },
    ], imports: [[
            CommonModule,
            HeaderModule,
            BackbaseCoreModule,
            RouterModule,
            ActionsProductNotificationsSettingsWidgetModule,
            ProductSummaryListWidgetModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsRetailNotificationPreferencesJourneyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ActionsRetailNotificationPreferencesViewComponent, ProductNotificationsSettingsComponent],
                    imports: [
                        CommonModule,
                        HeaderModule,
                        BackbaseCoreModule,
                        RouterModule,
                        ActionsProductNotificationsSettingsWidgetModule,
                        ProductSummaryListWidgetModule,
                    ],
                    providers: [
                        ActionsRetailNotificationPreferencesJourneyConfigService,
                        {
                            provide: NotificationsPreferencesBaseDataService,
                            useFactory: ActionNotificationsPreferencesDataFactory,
                            deps: [Injector],
                        },
                    ],
                }]
        }] });
