// @backbase/contact-common-ang/esm2015/lib/components/approval-log/approval-log-collapsible.component.js
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/collapsible";
import * as i2 from "@backbase/ui-ang/icon";
import * as i3 from "./approval-log.component";
/**
 * Component for a collapsible section with an approval log of a contact.
 *
 * @usageNotes
 *
 * ```html
 * <bb-approval-log-collapsible
 *   [isPending]="hostRef.approvalDetails?.approvalDetails?.status === 'PENDING'"
 *   [records]="hostRef.approvalDetails?.approvalDetails?.records"
 * ></bb-approval-log-collapsible>
 * ```
 * @ngModule ApprovalLogModule
 */
export class ApprovalLogCollapsibleComponent {
    constructor() {
        /**
         * Approval log records of a contact.
         */
        this.records = [];
        /**
         * Whether contact is currently pending approval
         */
        this.isPending = false;
    }
}
/** @nocollapse */ ApprovalLogCollapsibleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ApprovalLogCollapsibleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ ApprovalLogCollapsibleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ApprovalLogCollapsibleComponent, selector: "bb-approval-log-collapsible", inputs: { records: "records", isPending: "isPending" }, ngImport: i0, template: `
    <bb-collapsible-ui [attr.data-role]="'approval-log-toggle'">
      <ng-template bbCollapsibleHeader let-toggle="toggle" let-isOpen="isOpen">
        <div class="bb-block bb-stack">
          <div class="bb-stack__item bb-stack__item--fill">
            <h3 i18n="@@contact-manager-pending-approvals-details.approvalLog">Approval log</h3>
          </div>
          <div class="bb-stack__item bb-stack__item--push-right">
            <button
              (click)="toggle()"
              bbButton
              color="outline"
              aria-label="Approval log toggle"
              i18n-aria-label="@@bb-approval-log.toggle.label"
              data-role="approval-log-toggle-button"
              bbTooltip="Toggle"
              triggers="hover focus"
              i18n-bbTooltip="@@contact-manager-pending-approvals-details.approval-log-toggle.tooltip"
            >
              <bb-icon-ui [name]="isOpen ? 'toggle-up' : 'toggle-down'"></bb-icon-ui>
            </button>
          </div>
        </div>
        <hr />
      </ng-template>
      <ng-template bbCollapsibleBody>
        <div aria-live="assertive">
          <bb-approval-log [isPending]="isPending" [records]="records"> </bb-approval-log>
        </div>
      </ng-template>
    </bb-collapsible-ui>
  `, isInline: true, components: [{ type: i1.CollapsibleComponent, selector: "bb-collapsible-ui", inputs: ["recreateBodyContent", "hasChevron", "isOpen"], outputs: ["isOpenChange"] }, { type: i2.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i3.ApprovalLogComponent, selector: "bb-approval-log", inputs: ["records", "isPending"] }], directives: [{ type: i1.CollapsibleHeaderDirective, selector: "ng-template[bbCollapsibleHeader]" }, { type: i1.CollapsibleBodyDirective, selector: "ng-template[bbCollapsibleBody]" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ApprovalLogCollapsibleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-approval-log-collapsible',
                    template: `
    <bb-collapsible-ui [attr.data-role]="'approval-log-toggle'">
      <ng-template bbCollapsibleHeader let-toggle="toggle" let-isOpen="isOpen">
        <div class="bb-block bb-stack">
          <div class="bb-stack__item bb-stack__item--fill">
            <h3 i18n="@@contact-manager-pending-approvals-details.approvalLog">Approval log</h3>
          </div>
          <div class="bb-stack__item bb-stack__item--push-right">
            <button
              (click)="toggle()"
              bbButton
              color="outline"
              aria-label="Approval log toggle"
              i18n-aria-label="@@bb-approval-log.toggle.label"
              data-role="approval-log-toggle-button"
              bbTooltip="Toggle"
              triggers="hover focus"
              i18n-bbTooltip="@@contact-manager-pending-approvals-details.approval-log-toggle.tooltip"
            >
              <bb-icon-ui [name]="isOpen ? 'toggle-up' : 'toggle-down'"></bb-icon-ui>
            </button>
          </div>
        </div>
        <hr />
      </ng-template>
      <ng-template bbCollapsibleBody>
        <div aria-live="assertive">
          <bb-approval-log [isPending]="isPending" [records]="records"> </bb-approval-log>
        </div>
      </ng-template>
    </bb-collapsible-ui>
  `,
                }]
        }], propDecorators: { records: [{
                type: Input
            }], isPending: [{
                type: Input
            }] } });
