// @backbase/contact-common-ang/esm2015/lib/components/approval-log/approval-log.component.js
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/item-log";
import * as i2 from "@backbase/ui-ang/icon";
import * as i3 from "@angular/common";
/**
 * Component for approval log of a contact.
 *
 * @usageNotes
 *
 * ```html
 * <bb-approval-log
 *   [isPending]="hostRef.approvalDetails?.approvalDetails?.status === 'PENDING'"
 *   [records]="hostRef.approvalDetails?.approvalDetails?.records"
 * ></bb-approval-log>
 * ```
 * @ngModule ApprovalLogModule
 */
export class ApprovalLogComponent {
    constructor() {
        /**
         * Approval log records of a contact.
         */
        this.records = [];
        /**
         * Whether contact is currently pending approval
         */
        this.isPending = false;
    }
}
/** @nocollapse */ ApprovalLogComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ApprovalLogComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ ApprovalLogComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ApprovalLogComponent, selector: "bb-approval-log", inputs: { records: "records", isPending: "isPending" }, ngImport: i0, template: "<bb-item-log-ui [records]=\"records\" [isPending]=\"isPending\">\n  <ng-template bbItemLogRecord let-itemLogRecord>\n    <div class=\"bb-stack bb-stack--align-top\">\n      <bb-icon-ui\n        class=\"bb-item-log-record__icon\"\n        size=\"md\"\n        [name]=\"itemLogRecord?.status === 'REJECTED' ? 'cancel' : 'success'\"\n        [color]=\"itemLogRecord?.status === 'REJECTED' ? 'danger' : 'success'\"\n      >\n      </bb-icon-ui>\n      <div class=\"bb-item-log-record__details\">\n        <span class=\"bb-text-bold\" i18n=\"@@contact-approval-log.status\">\n          { itemLogRecord?.status, select, REJECTED {Rejected} APPROVED {Approved} other {} }\n        </span>\n        <span class=\"text-small\">\n          <ng-container i18n=\"@@contact-approval-log.label\"> by {{ itemLogRecord.userFullName }}</ng-container>\n          <ng-container *ngIf=\"itemLogRecord?.externalUserId\"> ({{ itemLogRecord.externalUserId }})</ng-container>\n        </span>\n        <div *ngIf=\"itemLogRecord?.createdAt as createdAt\" class=\"bb-item-log-record__time\">\n          <time> {{ createdAt | date }}</time>\n          <ng-container i18n=\"@@contact-approval-log-created-at.label\"> at</ng-container>\n          <time> {{ createdAt | date: 'shortTime' }}</time>\n        </div>\n        <div *ngIf=\"itemLogRecord?.comment\" class=\"media\">\n          <div class=\"bb-inline-stack__item\">\n            <bb-icon-ui color=\"secondary\" name=\"forum\" size=\"sm\"></bb-icon-ui>\n          </div>\n          <div class=\"media-body break-word text-small\">{{ itemLogRecord?.comment }}</div>\n        </div>\n      </div>\n    </div>\n  </ng-template>\n</bb-item-log-ui>\n", components: [{ type: i1.ItemLogComponent, selector: "bb-item-log-ui", inputs: ["isPending", "records"] }, { type: i2.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], directives: [{ type: i1.ItemLogRecordDirective, selector: "ng-template[bbItemLogRecord]" }, { type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "date": i3.DatePipe }, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ApprovalLogComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-approval-log',
                    templateUrl: 'approval-log.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { records: [{
                type: Input
            }], isPending: [{
                type: Input
            }] } });
