// @backbase/contact-common-ang/esm2015/lib/components/approval-log/approval-log.module.js
import { NgModule } from '@angular/core';
import { ApprovalLogComponent } from './approval-log.component';
import { CommonModule } from '@angular/common';
import { ItemLogModule } from '@backbase/ui-ang/item-log';
import { IconModule } from '@backbase/ui-ang/icon';
import { ApprovalLogCollapsibleComponent } from './approval-log-collapsible.component';
import { CollapsibleModule } from '@backbase/ui-ang/collapsible';
import * as i0 from "@angular/core";
/**
 * Exports:
 * - ApprovalLogComponent
 * - ApprovalLogCollapsibleComponent
 */
export class ApprovalLogModule {
}
/** @nocollapse */ ApprovalLogModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ApprovalLogModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ ApprovalLogModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ApprovalLogModule, declarations: [ApprovalLogComponent, ApprovalLogCollapsibleComponent], imports: [ItemLogModule, IconModule, CommonModule, CollapsibleModule], exports: [ApprovalLogComponent, ApprovalLogCollapsibleComponent] });
/** @nocollapse */ ApprovalLogModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ApprovalLogModule, imports: [[ItemLogModule, IconModule, CommonModule, CollapsibleModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ApprovalLogModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ApprovalLogComponent, ApprovalLogCollapsibleComponent],
                    imports: [ItemLogModule, IconModule, CommonModule, CollapsibleModule],
                    exports: [ApprovalLogComponent, ApprovalLogCollapsibleComponent],
                }]
        }] });
