// @backbase/ui-ang/esm2015/_example/collapsible-card/showcase/collapsible-card-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CollapsibleCardShowcaseComponent } from './collapsible-card-showcase.component';
import { CollapsibleCardModule } from '@backbase/ui-ang/collapsible-card';
import * as i0 from "@angular/core";
export class CollapsibleCardShowcaseModule {
}
CollapsibleCardShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CollapsibleCardShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CollapsibleCardShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CollapsibleCardShowcaseModule, declarations: [CollapsibleCardShowcaseComponent], imports: [CommonModule, CollapsibleCardModule] });
CollapsibleCardShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CollapsibleCardShowcaseModule, imports: [[CommonModule, CollapsibleCardModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CollapsibleCardShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, CollapsibleCardModule],
                    declarations: [CollapsibleCardShowcaseComponent],
                    entryComponents: [CollapsibleCardShowcaseComponent],
                }]
        }] });
