// @backbase/initiate-payment-journey-ang/bundles/backbase-initiate-payment-journey-ang.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core'), require('@angular/router'), require('@backbase/foundation-ang/core'), require('@backbase/payment-orders-ang'), require('@backbase/payments-collection-ui-ang'), require('@backbase/payord-form-builder-ang'), require('rxjs')) :
    typeof define === 'function' && define.amd ? define('@backbase/initiate-payment-journey-ang', ['exports', '@angular/common', '@angular/core', '@angular/router', '@backbase/foundation-ang/core', '@backbase/payment-orders-ang', '@backbase/payments-collection-ui-ang', '@backbase/payord-form-builder-ang', 'rxjs'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["initiate-payment-journey-ang"] = {}), global.ng.common, global.ng.core, global.ng.router, global.i1$2, global.i1, global.i3, global.i2, global.rxjs));
})(this, (function (exports, common, i0, i1$1, i1$2, i1, i3, i2, rxjs) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1$1);
    var i1__namespace$2 = /*#__PURE__*/_interopNamespace(i1$2);
    var i1__namespace$1 = /*#__PURE__*/_interopNamespace(i1);
    var i3__namespace = /*#__PURE__*/_interopNamespace(i3);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);

    /** @internal */
    var InitiatePaymentNavigationService = /** @class */ (function () {
        function InitiatePaymentNavigationService(router, route) {
            this.router = router;
            this.route = route;
            this.emitAfterSuccess = new rxjs.Subject();
            this.routingExtras = {
                queryParamsHandling: 'merge',
                relativeTo: this.route,
                skipLocationChange: true,
            };
        }
        /**
         * Navigate to the initiate payment form
         */
        InitiatePaymentNavigationService.prototype.form = function () {
            return this.router.navigate([i1.Routes.FORM], this.routingExtras);
        };
        /**
         * Navigate to the initiate payment review screen.
         */
        InitiatePaymentNavigationService.prototype.review = function () {
            return this.router.navigate([i1.Routes.REVIEW], this.routingExtras);
        };
        /**
         * Method to emit success event after form is submitted
         */
        InitiatePaymentNavigationService.prototype.afterSuccess = function () {
            this.emitAfterSuccess.next();
        };
        return InitiatePaymentNavigationService;
    }());
    /** @nocollapse */ InitiatePaymentNavigationService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: InitiatePaymentNavigationService, deps: [{ token: i1__namespace.Router }, { token: i1__namespace.ActivatedRoute }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    /** @nocollapse */ InitiatePaymentNavigationService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: InitiatePaymentNavigationService });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: InitiatePaymentNavigationService, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () { return [{ type: i1__namespace.Router }, { type: i1__namespace.ActivatedRoute }]; } });

    var INITIATE_PAYMENT_JOURNEY_COMMUNICATOR = new i0.InjectionToken('bb-initiate-payment-journey-communicator');

    /** @internal */
    var InitiatePaymentRoutableModalService = /** @class */ (function () {
        function InitiatePaymentRoutableModalService(communicator) {
            this.communicator = communicator;
        }
        InitiatePaymentRoutableModalService.prototype.closeModal = function () {
            var _a;
            (_a = this.communicator) === null || _a === void 0 ? void 0 : _a.closeEvent();
        };
        InitiatePaymentRoutableModalService.prototype.openModal = function () {
            var _a;
            (_a = this.communicator) === null || _a === void 0 ? void 0 : _a.headerNavigationAction('new-debit-ach');
        };
        return InitiatePaymentRoutableModalService;
    }());
    /** @nocollapse */ InitiatePaymentRoutableModalService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: InitiatePaymentRoutableModalService, deps: [{ token: INITIATE_PAYMENT_JOURNEY_COMMUNICATOR, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    /** @nocollapse */ InitiatePaymentRoutableModalService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: InitiatePaymentRoutableModalService });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: InitiatePaymentRoutableModalService, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () {
            return [{ type: undefined, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.Inject,
                            args: [INITIATE_PAYMENT_JOURNEY_COMMUNICATOR]
                        }] }];
        } });

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from, pack) {
        if (pack || arguments.length === 2)
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar)
                        ar = Array.prototype.slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
        return to.concat(ar || Array.prototype.slice.call(from));
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    function getPaymentConfig(paymentType) {
        switch (i1.normalizePaymentType(paymentType)) {
            case 'SEPA_CREDIT_TRANSFER':
            case 'SEPA_CT':
            case 'SEPA':
                return i1.SEPA;
            case 'ACH_CREDIT':
            case 'US_ACH_CREDIT':
                return i1.US_ACH_CREDIT;
            case 'US_ACH_DEBIT':
            case 'ACH_DEBIT':
                return i1.ACH_DEBIT;
            case 'FASTER_PAYMENT':
                return i1.UK_FASTER_PAYMENT;
            case 'CHAPS':
                return i1.CHAPS;
            case 'US_DOMESTIC_WIRE':
                return i1.US_DOMESTIC_WIRE;
            case 'INTERNAL_TRANSFER':
                return i1.INTERNAL_TRANSFER;
            case 'INTRABANK_TRANSFER':
                return i1.INTRABANK_TRANSFER;
            case 'INTERNAL_TRANSFER_BUSINESS':
                return i1.INTERNAL_TRANSFER_BUSINESS;
            case 'P2P_TRANSFER':
                return i1.P2P_TRANSFER;
            case 'INTERNATIONAL_TRANSFER':
                return i1.INTERNATIONAL_TRANSFER;
        }
        return i1.US_DOMESTIC_WIRE;
    }
    function getBusinessFunction(paymentType) {
        switch (i1.normalizePaymentType(paymentType)) {
            case 'SEPA_CREDIT_TRANSFER':
            case 'SEPA_CT':
            case 'SEPA':
                return 'SEPA CT';
            case 'ACH_DEBIT':
                return 'ACH Debit';
            case 'ACH_CREDIT':
            case 'US_ACH_CREDIT':
            case 'CUSTOM_ACH_CREDIT':
                return 'ACH Credit Transfer';
            case 'CHAPS':
                return 'UK CHAPS';
            case 'FASTER_PAYMENT':
                return 'UK Faster Payments';
            case 'US_DOMESTIC_WIRE':
                return 'US Domestic Wire';
            case 'INTERNAL_TRANSFER':
            case 'INTERNAL_TRANSFER_BUSINESS':
            case 'INTRABANK_TRANSFER':
                return 'A2A Transfer';
            case 'INTERNATIONAL_TRANSFER':
                return 'US Foreign Wire';
            case 'P2P_TRANSFER':
                return 'P2P Transfer';
        }
        return 'US Domestic Wire';
    }
    function getPaymentTypeTitle(paymentType) {
        switch (i1.normalizePaymentType(paymentType)) {
            case 'SEPA_CREDIT_TRANSFER':
            case 'SEPA_CT':
            case 'SEPA':
                return $localize(templateObject_1 || (templateObject_1 = __makeTemplateObject([":@@initiate-payment.payment-title.payment-type-sepa:SEPA"], [":@@initiate-payment.payment-title.payment-type-sepa:SEPA"])));
            case 'ACH_DEBIT':
            case 'ACH_CREDIT':
            case 'US_ACH_CREDIT':
            case 'CUSTOM_ACH_CREDIT':
                return $localize(templateObject_2 || (templateObject_2 = __makeTemplateObject([":@@initiate-payment.payment-title.payment-type-ach:ACH"], [":@@initiate-payment.payment-title.payment-type-ach:ACH"])));
            case 'US_DOMESTIC_WIRE':
                return $localize(templateObject_3 || (templateObject_3 = __makeTemplateObject([":@@initiate-payment.payment-title.payment-type-wire:Wire"], [":@@initiate-payment.payment-title.payment-type-wire:Wire"])));
            case 'INTERNAL_TRANSFER':
            case 'INTERNAL_TRANSFER_BUSINESS':
            case 'INTRABANK_TRANSFER':
                return $localize(templateObject_4 || (templateObject_4 = __makeTemplateObject([":@@initiate-payment.payment-title.payment-type-internal:Internal"], [":@@initiate-payment.payment-title.payment-type-internal:Internal"])));
            case 'INTERNATIONAL_TRANSFER':
                return $localize(templateObject_5 || (templateObject_5 = __makeTemplateObject([":@@initiate-payment.payment-title.payment-type-international-wire:International Wire"], [":@@initiate-payment.payment-title.payment-type-international-wire:International Wire"])));
            case 'P2P_TRANSFER':
                return $localize(templateObject_6 || (templateObject_6 = __makeTemplateObject([":@@initiate-payment.payment-title.payment-type-p2p:P2P"], [":@@initiate-payment.payment-title.payment-type-p2p:P2P"])));
        }
        return $localize(templateObject_7 || (templateObject_7 = __makeTemplateObject([":@@initiate-payment.payment-title.payment-type-default:"], [":@@initiate-payment.payment-title.payment-type-default:"])));
    }
    function defaultHeaderSetup(_a) {
        var paymentType = _a.paymentType, paymentMode = _a.paymentMode, state = _a.state, isDebit = _a.isDebit;
        var payment = isDebit
            ? $localize(templateObject_8 || (templateObject_8 = __makeTemplateObject([":@@initiate-payment.payment-title.debit: Debit"], [":@@initiate-payment.payment-title.debit: Debit"]))) : '';
        var title = "" + getPaymentTypeTitle(paymentType) + payment;
        if (paymentMode === null || paymentMode === void 0 ? void 0 : paymentMode.includes('EDIT')) {
            title = $localize(templateObject_9 || (templateObject_9 = __makeTemplateObject([":@@initiate-payment.payment-title.edit-mode:Edit ", ""], [":@@initiate-payment.payment-title.edit-mode:Edit ", ""])), title);
        }
        else {
            title = $localize(templateObject_10 || (templateObject_10 = __makeTemplateObject([":@@initiate-payment.payment-title.new-mode:New ", ""], [":@@initiate-payment.payment-title.new-mode:New ", ""])), title);
        }
        if (paymentMode === null || paymentMode === void 0 ? void 0 : paymentMode.includes('TEMPLATE')) {
            title = $localize(templateObject_11 || (templateObject_11 = __makeTemplateObject([":@@initiate-payment.payment-title.template:", " Template"], [":@@initiate-payment.payment-title.template:", " Template"])), title);
        }
        else if (!isDebit) {
            title = $localize(templateObject_12 || (templateObject_12 = __makeTemplateObject([":@@initiate-payment.payment-title.payment:", " Payment"], [":@@initiate-payment.payment-title.payment:", " Payment"])), title);
        }
        if (state === i1.InitiatePaymentHeaderStates.REVIEW) {
            title = $localize(templateObject_13 || (templateObject_13 = __makeTemplateObject([":@@initiate-payment.payment-title.review-state:", " - Review"], [":@@initiate-payment.payment-title.review-state:", " - Review"])), title);
        }
        else if (state === i1.InitiatePaymentHeaderStates.SUBMITTED) {
            title = $localize(templateObject_14 || (templateObject_14 = __makeTemplateObject([":@@initiate-payment.payment-title.submitted-state:", " - Submitted"], [":@@initiate-payment.payment-title.submitted-state:", " - Submitted"])), title);
        }
        return title;
    }
    var templateObject_1, templateObject_2, templateObject_3, templateObject_4, templateObject_5, templateObject_6, templateObject_7, templateObject_8, templateObject_9, templateObject_10, templateObject_11, templateObject_12, templateObject_13, templateObject_14;

    var InitiatePaymentDefaultOptions = {
        pageSize: 50,
        isModalView: true,
        pollInterval: 1000,
        defaultCountry: 'US',
        enableApprovals: true,
        defaultScheme: 'BBAN',
        isTemplateMode: false,
        isDebitPayment: false,
        isErrorTitleDisplayed: true,
        enableNewPaymentButton: false,
        checkClosedPaymentsAccess: false,
        enableSavePaymentAsTemplate: true,
        isOneOffToRecurrentAllowed: false,
        isRecurrentToOneOffAllowed: false,
        enablePaymentTemplateSelector: true,
        accessContextScope: i1.AccessContextScope.SA,
        successEventName: 'bb.success.create.payment',
        header: defaultHeaderSetup,
    };
    var INITIATE_PAYMENT_CONFIG = new i0.InjectionToken('INITIATE_PAYMENT_CONFIG');

    var InitiatePaymentConfigService = /** @class */ (function () {
        function InitiatePaymentConfigService(stateService, componentsFactory, paymentOrderHttpService, createPaymentSetup, initiatePaymentConfig) {
            this.stateService = stateService;
            this.componentsFactory = componentsFactory;
            this.paymentOrderHttpService = paymentOrderHttpService;
            this.createPaymentSetup = createPaymentSetup;
            this.initiatePaymentConfig = initiatePaymentConfig;
        }
        /**
         * @param {TriggerInitiatePaymentPayload} [data] - Initiate payment payload
         */
        InitiatePaymentConfigService.prototype.setup = function (data) {
            var _a;
            if (data) {
                ((_a = data.options) === null || _a === void 0 ? void 0 : _a.paymentMode) !== i1.PaymentMode.CREATE_PAYMENT
                    ? this.applyData(data)
                    : this.applyConfigWithData(data);
            }
            else {
                this.applyConfig();
            }
        };
        InitiatePaymentConfigService.prototype.applyConfig = function () {
            var _b = this.initiatePaymentConfig, paymentTypes = _b.paymentTypes, hooks = _b.hooks, options = _b.options, businessFunctions = _b.businessFunctions, baseFields = _b.baseFields, fetchPaymentOptionsListener = _b.fetchPaymentOptionsListener, customFields = _b.customFields;
            this.createPaymentSetup.setHooks(hooks);
            this.createPaymentSetup.setConfigs(paymentTypes);
            this.createPaymentSetup.setOptions(Object.assign(Object.assign({}, InitiatePaymentDefaultOptions), (options || {})));
            this.createPaymentSetup.setBaseFields(baseFields);
            this.createPaymentSetup.setFetchPaymentOptionsListener(fetchPaymentOptionsListener);
            this.paymentOrderHttpService.businessFunction = businessFunctions.join(',');
            this.componentsFactory.registerCustomFields(customFields);
            this.componentsFactory.registerCustomFieldsForPaymentConfigs(paymentTypes);
        };
        InitiatePaymentConfigService.prototype.applyData = function (data) {
            var _a;
            var template = data.template;
            var paymentType = template
                ? template.details.paymentType
                : (_a = data.payment) === null || _a === void 0 ? void 0 : _a.paymentType;
            this.paymentOrderHttpService.businessFunction = getBusinessFunction(paymentType);
            if (template) {
                this.stateService.setPaymentTemplate(data.template);
            }
            else {
                this.stateService.setEditPayment(data.payment);
            }
            this.createPaymentSetup.setConfigs([
                Object.assign({}, getPaymentConfig(paymentType)),
            ]);
            this.createPaymentSetup.setOptions(Object.assign(Object.assign({}, InitiatePaymentDefaultOptions), (data.options || {})));
        };
        InitiatePaymentConfigService.prototype.applyConfigWithData = function (data) {
            var template = data.template;
            if (template) {
                this.stateService.setPaymentTemplate(data.template);
            }
            else {
                this.stateService.setEditPayment(data.payment);
            }
            this.applyConfig();
        };
        return InitiatePaymentConfigService;
    }());
    /** @nocollapse */ InitiatePaymentConfigService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: InitiatePaymentConfigService, deps: [{ token: i1__namespace$1.StateService }, { token: i2__namespace.ComponentsFactoryService }, { token: i1__namespace$1.PaymentOrderHttpService }, { token: i3__namespace.CreatePaymentConfigSetupService }, { token: INITIATE_PAYMENT_CONFIG }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    /** @nocollapse */ InitiatePaymentConfigService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: InitiatePaymentConfigService });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: InitiatePaymentConfigService, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () {
            return [{ type: i1__namespace$1.StateService }, { type: i2__namespace.ComponentsFactoryService }, { type: i1__namespace$1.PaymentOrderHttpService }, { type: i3__namespace.CreatePaymentConfigSetupService }, { type: undefined, decorators: [{
                            type: i0.Inject,
                            args: [INITIATE_PAYMENT_CONFIG]
                        }] }];
        } });

    var InitiatePaymentJourneyComponent = /** @class */ (function () {
        function InitiatePaymentJourneyComponent(configService, communicator) {
            this.configService = configService;
            this.communicator = communicator;
            if (communicator) {
                communicator.init({
                    setupData: function (data) {
                        configService.setup(data);
                    },
                });
            }
            else {
                configService.setup();
            }
        }
        InitiatePaymentJourneyComponent.prototype.ngOnDestroy = function () {
            this.communicator.reset();
        };
        return InitiatePaymentJourneyComponent;
    }());
    /** @nocollapse */ InitiatePaymentJourneyComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: InitiatePaymentJourneyComponent, deps: [{ token: InitiatePaymentConfigService }, { token: INITIATE_PAYMENT_JOURNEY_COMMUNICATOR, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Component });
    /** @nocollapse */ InitiatePaymentJourneyComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: InitiatePaymentJourneyComponent, selector: "bb-initiate-payment-journey", providers: [
            i1.StateService,
            InitiatePaymentConfigService,
            i3.CreatePaymentConfigSetupService,
            { provide: i1.NavigationService, useClass: InitiatePaymentNavigationService },
            {
                provide: i1.RoutableModalService,
                useClass: InitiatePaymentRoutableModalService,
            },
        ], ngImport: i0__namespace, template: " <router-outlet></router-outlet>", isInline: true, directives: [{ type: i1__namespace.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: InitiatePaymentJourneyComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-initiate-payment-journey',
                        template: " <router-outlet></router-outlet>",
                        providers: [
                            i1.StateService,
                            InitiatePaymentConfigService,
                            i3.CreatePaymentConfigSetupService,
                            { provide: i1.NavigationService, useClass: InitiatePaymentNavigationService },
                            {
                                provide: i1.RoutableModalService,
                                useClass: InitiatePaymentRoutableModalService,
                            },
                        ],
                    }]
            }], ctorParameters: function () {
            return [{ type: InitiatePaymentConfigService }, { type: undefined, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.Inject,
                            args: [INITIATE_PAYMENT_JOURNEY_COMMUNICATOR]
                        }] }];
        } });

    var InitiatePaymentDefaultRoute = {
        path: '',
        component: InitiatePaymentJourneyComponent,
        children: [
            { path: '', redirectTo: i1.Routes.FORM, pathMatch: 'full' },
            { path: i1.Routes.FORM, component: i3.CreatePaymentFormComponent },
            { path: i1.Routes.REVIEW, component: i3.ReviewPaymentContainerComponent },
        ],
    };
    var InitiatePaymentJourneyModule = /** @class */ (function () {
        function InitiatePaymentJourneyModule() {
        }
        InitiatePaymentJourneyModule.forRoot = function (data) {
            if (data === void 0) { data = {
                route: InitiatePaymentDefaultRoute,
            }; }
            return {
                ngModule: InitiatePaymentJourneyModule,
                providers: [i1$1.provideRoutes([data.route])],
            };
        };
        return InitiatePaymentJourneyModule;
    }());
    /** @nocollapse */ InitiatePaymentJourneyModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: InitiatePaymentJourneyModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    /** @nocollapse */ InitiatePaymentJourneyModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: InitiatePaymentJourneyModule, declarations: [InitiatePaymentJourneyComponent], imports: [common.CommonModule,
            i1$1.RouterModule,
            i3.ReviewPaymentModule,
            i3.CreatePaymentFormModule,
            i2.PayordFormBuilderAngModule, i1__namespace$2.BackbaseCoreModule] });
    /** @nocollapse */ InitiatePaymentJourneyModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: InitiatePaymentJourneyModule, providers: [
            i3.CreatePaymentService,
            i1.PaymentOptionsService,
            i1.PaymentOrderHttpService,
            i1.PayordPaymentErrorsConfigProvider,
        ], imports: [[
                common.CommonModule,
                i1$1.RouterModule,
                i3.ReviewPaymentModule,
                i3.CreatePaymentFormModule,
                i2.PayordFormBuilderAngModule,
                i1$2.BackbaseCoreModule.withConfig({
                    classMap: {
                        InitiatePaymentJourneyComponent: InitiatePaymentJourneyComponent,
                    },
                }),
            ]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: InitiatePaymentJourneyModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [InitiatePaymentJourneyComponent],
                        imports: [
                            common.CommonModule,
                            i1$1.RouterModule,
                            i3.ReviewPaymentModule,
                            i3.CreatePaymentFormModule,
                            i2.PayordFormBuilderAngModule,
                            i1$2.BackbaseCoreModule.withConfig({
                                classMap: {
                                    InitiatePaymentJourneyComponent: InitiatePaymentJourneyComponent,
                                },
                            }),
                        ],
                        providers: [
                            i3.CreatePaymentService,
                            i1.PaymentOptionsService,
                            i1.PaymentOrderHttpService,
                            i1.PayordPaymentErrorsConfigProvider,
                        ],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    Object.defineProperty(exports, 'AccessContextScope', {
        enumerable: true,
        get: function () { return i1.AccessContextScope; }
    });
    Object.defineProperty(exports, 'CounterPartyFields', {
        enumerable: true,
        get: function () { return i1.CounterPartyFields; }
    });
    Object.defineProperty(exports, 'Frequencies', {
        enumerable: true,
        get: function () { return i1.Frequencies; }
    });
    Object.defineProperty(exports, 'INTERNAL_TRANSFER', {
        enumerable: true,
        get: function () { return i1.INTERNAL_TRANSFER; }
    });
    Object.defineProperty(exports, 'INTERNAL_TRANSFER_BUSINESS', {
        enumerable: true,
        get: function () { return i1.INTERNAL_TRANSFER_BUSINESS; }
    });
    Object.defineProperty(exports, 'INTERNATIONAL_TRANSFER', {
        enumerable: true,
        get: function () { return i1.INTERNATIONAL_TRANSFER; }
    });
    Object.defineProperty(exports, 'INTRABANK_TRANSFER', {
        enumerable: true,
        get: function () { return i1.INTRABANK_TRANSFER; }
    });
    Object.defineProperty(exports, 'InitiatorFields', {
        enumerable: true,
        get: function () { return i1.InitiatorFields; }
    });
    Object.defineProperty(exports, 'P2P_TRANSFER', {
        enumerable: true,
        get: function () { return i1.P2P_TRANSFER; }
    });
    Object.defineProperty(exports, 'PaymentBaseFields', {
        enumerable: true,
        get: function () { return i1.PaymentBaseFields; }
    });
    Object.defineProperty(exports, 'PaymentComponents', {
        enumerable: true,
        get: function () { return i1.PaymentComponents; }
    });
    Object.defineProperty(exports, 'PaymentFormFieldHooks', {
        enumerable: true,
        get: function () { return i1.PaymentFormFieldHooks; }
    });
    Object.defineProperty(exports, 'PaymentHooks', {
        enumerable: true,
        get: function () { return i1.PaymentHooks; }
    });
    Object.defineProperty(exports, 'PaymentMode', {
        enumerable: true,
        get: function () { return i1.PaymentMode; }
    });
    Object.defineProperty(exports, 'PayordOmniPaymentConfigProvider', {
        enumerable: true,
        get: function () { return i1.PayordOmniPaymentConfigProvider; }
    });
    Object.defineProperty(exports, 'RemittanceInfoFields', {
        enumerable: true,
        get: function () { return i1.RemittanceInfoFields; }
    });
    Object.defineProperty(exports, 'ReviewScreens', {
        enumerable: true,
        get: function () { return i1.ReviewScreens; }
    });
    Object.defineProperty(exports, 'SEPA', {
        enumerable: true,
        get: function () { return i1.SEPA; }
    });
    Object.defineProperty(exports, 'ScheduleEndType', {
        enumerable: true,
        get: function () { return i1.ScheduleEndType; }
    });
    Object.defineProperty(exports, 'ScheduleFields', {
        enumerable: true,
        get: function () { return i1.ScheduleFields; }
    });
    Object.defineProperty(exports, 'US_ACH_CREDIT', {
        enumerable: true,
        get: function () { return i1.US_ACH_CREDIT; }
    });
    Object.defineProperty(exports, 'US_DOMESTIC_WIRE', {
        enumerable: true,
        get: function () { return i1.US_DOMESTIC_WIRE; }
    });
    exports.INITIATE_PAYMENT_CONFIG = INITIATE_PAYMENT_CONFIG;
    exports.INITIATE_PAYMENT_JOURNEY_COMMUNICATOR = INITIATE_PAYMENT_JOURNEY_COMMUNICATOR;
    exports.InitiatePaymentConfigService = InitiatePaymentConfigService;
    exports.InitiatePaymentDefaultOptions = InitiatePaymentDefaultOptions;
    exports.InitiatePaymentDefaultRoute = InitiatePaymentDefaultRoute;
    exports.InitiatePaymentJourneyComponent = InitiatePaymentJourneyComponent;
    exports.InitiatePaymentJourneyModule = InitiatePaymentJourneyModule;
    exports.InitiatePaymentNavigationService = InitiatePaymentNavigationService;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
