// @backbase/ui-ang/esm2015/_example/amount-input/showcase/amount-input-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AmountInputShowcaseComponent } from './amount-input-showcase.component';
import { AmountInputModule } from '@backbase/ui-ang/amount-input';
import { ReactiveFormsModule } from '@angular/forms';
import * as i0 from "@angular/core";
export class AmountInputShowcaseModule {
}
AmountInputShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AmountInputShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AmountInputShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AmountInputShowcaseModule, declarations: [AmountInputShowcaseComponent], imports: [CommonModule, AmountInputModule, ReactiveFormsModule] });
AmountInputShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AmountInputShowcaseModule, imports: [[CommonModule, AmountInputModule, ReactiveFormsModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AmountInputShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, AmountInputModule, ReactiveFormsModule],
                    declarations: [AmountInputShowcaseComponent],
                    entryComponents: [AmountInputShowcaseComponent],
                }]
        }] });
