// @backbase/foundation-ang/schematics/utils/ng-utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.findPropertyInAstObject = exports.getWorkspace = exports.getAppModulePath = exports.NodeDependencyType = exports.addPackageJsonDependency = exports.NoopChange = exports.RemoveChange = exports.ReplaceChange = exports.InsertChange = exports.addImportToModule = void 0;
var ast_utils_1 = require("@schematics/angular/utility/ast-utils");
Object.defineProperty(exports, "addImportToModule", { enumerable: true, get: function () { return ast_utils_1.addImportToModule; } });
var change_1 = require("@schematics/angular/utility/change");
Object.defineProperty(exports, "InsertChange", { enumerable: true, get: function () { return change_1.InsertChange; } });
Object.defineProperty(exports, "ReplaceChange", { enumerable: true, get: function () { return change_1.ReplaceChange; } });
Object.defineProperty(exports, "RemoveChange", { enumerable: true, get: function () { return change_1.RemoveChange; } });
Object.defineProperty(exports, "NoopChange", { enumerable: true, get: function () { return change_1.NoopChange; } });
var dependencies_1 = require("@schematics/angular/utility/dependencies");
Object.defineProperty(exports, "addPackageJsonDependency", { enumerable: true, get: function () { return dependencies_1.addPackageJsonDependency; } });
Object.defineProperty(exports, "NodeDependencyType", { enumerable: true, get: function () { return dependencies_1.NodeDependencyType; } });
var ng_ast_utils_1 = require("@schematics/angular/utility/ng-ast-utils");
Object.defineProperty(exports, "getAppModulePath", { enumerable: true, get: function () { return ng_ast_utils_1.getAppModulePath; } });
var workspace_1 = require("@schematics/angular/utility/workspace");
Object.defineProperty(exports, "getWorkspace", { enumerable: true, get: function () { return workspace_1.getWorkspace; } });
function findPropertyInAstObject(node, propertyName) {
    let maybeNode = null;
    for (const property of node.properties) {
        if (property.key.value == propertyName) {
            maybeNode = property.value;
        }
    }
    return maybeNode;
}
exports.findPropertyInAstObject = findPropertyInAstObject;
