// @backbase/places-journey-ang/esm2015/lib/components/bb-map-ui/map-search.component.js
import { Component, ChangeDetectionStrategy, Input, Output, EventEmitter, ViewChild, } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./map-api-loader.service";
import * as i2 from "@backbase/ui-ang/search-box";
export class MapSearchComponent {
    constructor(loader) {
        this.loader = loader;
        /**
         * Map autocomplete options object.
         */
        this.options = {};
        /**
         * Map API configuration.
         */
        this.config = {};
        /**
         * EventEmitter for triggering a mapSearch event.
         */
        this.mapSearch = new EventEmitter();
    }
    onSearchButtonClick() {
        const input = this.searchBox.inputField;
        if (!input) {
            return;
        }
        // select first result and do the search
        google.maps.event.trigger(input.nativeElement, 'focus', {});
        google.maps.event.trigger(input.nativeElement, 'keydown', {
            keyCode: 40,
            stopPropagation: () => undefined,
            preventDefault: () => undefined,
        });
        google.maps.event.trigger(input.nativeElement, 'keydown', { keyCode: 13 }); // enter
    }
    ngOnInit() {
        this.loader.config = this.config;
        this.loader.load().then(() => {
            const input = this.searchBox.inputField;
            if (!input) {
                return;
            }
            const autocomplete = new google.maps.places.Autocomplete(input.nativeElement, this.options);
            autocomplete.addListener('place_changed', () => {
                const place = autocomplete.getPlace();
                if (place.geometry) {
                    this.mapSearch.emit(place);
                }
            });
        });
    }
}
MapSearchComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MapSearchComponent, deps: [{ token: i1.MapAPILoaderService }], target: i0.ɵɵFactoryTarget.Component });
MapSearchComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: MapSearchComponent, selector: "bb-map-search-ui", inputs: { options: "options", config: "config" }, outputs: { mapSearch: "mapSearch" }, viewQueries: [{ propertyName: "searchBox", first: true, predicate: ["bbMapsearch"], descendants: true, static: true }], ngImport: i0, template: "<div class=\"bb-block bb-block--md\">\n  <bb-search-box-ui\n    placeholder=\"Search Branch and Street Names\"\n    i18n-placeholder=\"Search for Location@@map-ui.search.placeholder\"\n    aria-label=\"Search Branch and Street Name\"\n    i18n-aria-label=\"\n      Search Branch and Street Name aria-label@@map.search.ui.aria-label\"\n    [showSearch]=\"true\"\n    (submit)=\"onSearchButtonClick()\"\n    #bbMapsearch\n  >\n  </bb-search-box-ui>\n</div>\n", components: [{ type: i2.SearchBoxComponent, selector: "bb-search-box-ui", inputs: ["placeholder", "maxLength", "squareBorder", "typeaheadOptions", "showSearch", "showClear", "searchLabel", "clearLabel", "autocomplete"], outputs: ["submit", "clear", "valueChange"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MapSearchComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-map-search-ui',
                    templateUrl: './map-search.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.MapAPILoaderService }]; }, propDecorators: { options: [{
                type: Input
            }], config: [{
                type: Input
            }], mapSearch: [{
                type: Output
            }], searchBox: [{
                type: ViewChild,
                args: ['bbMapsearch', { static: true }]
            }] } });
