// @backbase/places-journey-ang/esm2015/lib/components/bb-map-ui/map.component.js
import { Component, Input, Output, EventEmitter, ViewChild, ChangeDetectionStrategy, } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./map-helpers.service";
export class MapComponent {
    constructor(helpers) {
        this.helpers = helpers;
        /**
         * Map options object.
         */
        this.options = {};
        /**
         * Map API configuration.
         */
        this.config = {};
        /**
         * EventEmitter for triggering a mapReady event.
         */
        this.mapReady = new EventEmitter();
    }
    get mapHelpers() {
        return this.helpers;
    }
    ngOnInit() {
        this.helpers
            .createMap(this.mapContainer.nativeElement, this.options, this.config)
            .then(() => this.mapReady.emit(this));
    }
}
MapComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MapComponent, deps: [{ token: i1.MapHelpersService }], target: i0.ɵɵFactoryTarget.Component });
MapComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: MapComponent, selector: "bb-map-ui", inputs: { options: "options", config: "config" }, outputs: { mapReady: "mapReady" }, viewQueries: [{ propertyName: "mapContainer", first: true, predicate: ["bbMapContainer"], descendants: true, static: true }], ngImport: i0, template: "<ng-content select=\"[bbMapHeader]\"></ng-content>\n<div #bbMapContainer class=\"bb-map\"></div>\n<ng-content select=\"[bbMapFooter]\"></ng-content>\n", changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MapComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-map-ui',
                    templateUrl: './map.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.MapHelpersService }]; }, propDecorators: { options: [{
                type: Input
            }], config: [{
                type: Input
            }], mapReady: [{
                type: Output
            }], mapContainer: [{
                type: ViewChild,
                args: ['bbMapContainer', { static: true }]
            }] } });
