// @backbase/places-journey-ang/esm2015/lib/components/bb-map-ui/map.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MapAPILoaderService } from './map-api-loader.service';
import { MapHelpersService } from './map-helpers.service';
import { MapComponent } from './map.component';
import { MapSearchComponent } from './map-search.component';
import { SearchBoxModule } from '@backbase/ui-ang/search-box';
import { ButtonModule } from '@backbase/ui-ang/button';
import { IconModule } from '@backbase/ui-ang/icon';
import * as i0 from "@angular/core";
export class MapModule {
}
MapModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MapModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MapModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MapModule, declarations: [MapComponent, MapSearchComponent], imports: [CommonModule, SearchBoxModule, ButtonModule, IconModule], exports: [MapComponent, MapSearchComponent] });
MapModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MapModule, providers: [MapAPILoaderService, MapHelpersService], imports: [[CommonModule, SearchBoxModule, ButtonModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MapModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, SearchBoxModule, ButtonModule, IconModule],
                    declarations: [MapComponent, MapSearchComponent],
                    exports: [MapComponent, MapSearchComponent],
                    providers: [MapAPILoaderService, MapHelpersService],
                }]
        }] });
