// @backbase/accounts-transactions-journey-ang/esm2015/lib/accounts-transactions-journey.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { provideRoutes, RouterModule } from '@angular/router';
import { AccountsTransactionsJourneyComponent } from './accounts-transactions-journey.component';
import { AccountsListComponent } from './views/accounts-list/accounts-list.component';
import { TransactionsListComponent } from './views/transactions-list.component';
import { AccountsDetailsTabComponent } from './views/accounts-details-tab/accounts-details-tab.component';
import { ProductSummaryAccountSelectorWidgetModule } from '@backbase/product-summary-account-selector-widget-ang';
import { ProductSummaryListWidgetModule } from '@backbase/product-summary-list-widget-ang';
import { ProductSummaryManageAccountsWidgetModule } from '@backbase/product-summary-manage-accounts-widget-ang';
import { TransactionsListWidgetModule } from '@backbase/transactions-list-widget-ang';
import { TransactionsCommonModule } from '@backbase/transactions-common-ang';
import { ProductSummaryDetailsWidgetModule } from '@backbase/product-summary-details-widget-ang';
import { AccountsDetailsComponent } from './views/accounts-details.component';
import { AccountsManageComponent } from './views/accounts-manage.component';
import { AccountsManageGuardService } from './accounts-manage-guard.service';
import { TransactionDetailsComponent } from './views/transactions-details.component';
import { TransactionsHelperService } from './transactions-details-helper.service';
import { AccountGraphicalHeaderModule, AccountHeaderModule } from '@backbase/product-summary-ui';
import { ValidationModule } from '@backbase/product-summary-common-ang';
import { AccountInfoViewComponent } from './views/account-info/account-info-view.component';
import { AccountInfoViewModule } from './views/account-info/account-info-view.module';
import { ArrangementsListComponent } from './views/arrangements-list/arrangements-list.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
import * as i2 from "@backbase/transactions-list-widget-ang";
export const defaultRoute = {
    path: '',
    component: AccountsTransactionsJourneyComponent,
    children: [
        {
            path: '',
            redirectTo: 'list',
            pathMatch: 'full',
        },
        {
            path: 'list',
            component: AccountsListComponent,
            data: { title: 'My Accounts' },
        },
        {
            path: 'manage',
            component: AccountsManageComponent,
            canActivate: [AccountsManageGuardService],
        },
        {
            path: 'transactions',
            component: AccountsDetailsTabComponent,
            data: { title: 'Transactions' },
            children: [
                { path: '', redirectTo: 'list', pathMatch: 'full' },
                {
                    path: 'list',
                    component: TransactionsListComponent,
                    data: { title: 'List' },
                    children: [
                        {
                            path: 'detail',
                            component: TransactionDetailsComponent,
                        },
                    ],
                },
                {
                    path: 'details',
                    component: AccountInfoViewComponent,
                    data: { title: 'Details' },
                },
            ],
        },
    ],
};
const views = [
    TransactionsListComponent,
    TransactionDetailsComponent,
    AccountsListComponent,
    ArrangementsListComponent,
    AccountsDetailsTabComponent,
    AccountsDetailsComponent,
    AccountsManageComponent,
];
export class AccountsTransactionsJourneyModule {
    static forRoot(data = { route: defaultRoute }) {
        return {
            ngModule: AccountsTransactionsJourneyModule,
            providers: [provideRoutes([data.route])],
        };
    }
}
/** @nocollapse */ AccountsTransactionsJourneyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountsTransactionsJourneyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ AccountsTransactionsJourneyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountsTransactionsJourneyModule, declarations: [AccountsTransactionsJourneyComponent, TransactionsListComponent,
        TransactionDetailsComponent,
        AccountsListComponent,
        ArrangementsListComponent,
        AccountsDetailsTabComponent,
        AccountsDetailsComponent,
        AccountsManageComponent], imports: [CommonModule, i1.BackbaseCoreModule, RouterModule,
        AccountInfoViewModule,
        TransactionsListWidgetModule,
        TransactionsCommonModule,
        ProductSummaryAccountSelectorWidgetModule,
        ProductSummaryManageAccountsWidgetModule,
        ProductSummaryDetailsWidgetModule,
        ProductSummaryListWidgetModule,
        AccountHeaderModule,
        AccountGraphicalHeaderModule,
        ValidationModule, i2.TransactionsListWidgetModule] });
/** @nocollapse */ AccountsTransactionsJourneyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountsTransactionsJourneyModule, providers: [AccountsManageGuardService, TransactionsHelperService], imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                classMap: {
                    // eslint-disable-next-line
                    AccountsTransactionsJourneyComponent,
                },
            }),
            RouterModule,
            AccountInfoViewModule,
            TransactionsListWidgetModule,
            TransactionsCommonModule,
            ProductSummaryAccountSelectorWidgetModule,
            ProductSummaryManageAccountsWidgetModule,
            ProductSummaryDetailsWidgetModule,
            ProductSummaryListWidgetModule,
            AccountHeaderModule,
            AccountGraphicalHeaderModule,
            ValidationModule,
            TransactionsListWidgetModule.forRoot({
                showDetails: false,
            }),
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountsTransactionsJourneyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [AccountsTransactionsJourneyComponent, ...views],
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: {
                                // eslint-disable-next-line
                                AccountsTransactionsJourneyComponent,
                            },
                        }),
                        RouterModule,
                        AccountInfoViewModule,
                        TransactionsListWidgetModule,
                        TransactionsCommonModule,
                        ProductSummaryAccountSelectorWidgetModule,
                        ProductSummaryManageAccountsWidgetModule,
                        ProductSummaryDetailsWidgetModule,
                        ProductSummaryListWidgetModule,
                        AccountHeaderModule,
                        AccountGraphicalHeaderModule,
                        ValidationModule,
                        TransactionsListWidgetModule.forRoot({
                            showDetails: false,
                        }),
                    ],
                    providers: [AccountsManageGuardService, TransactionsHelperService],
                }]
        }] });
