// @backbase/accounts-transactions-journey-ang/esm2015/lib/accounts-transactions-journey.component.js
import { Component } from '@angular/core';
import { AccountsTransactionsJourneyService } from './accounts-transactions-journey-config.service';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
export class AccountsTransactionsJourneyComponent {
}
/** @nocollapse */ AccountsTransactionsJourneyComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountsTransactionsJourneyComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ AccountsTransactionsJourneyComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AccountsTransactionsJourneyComponent, selector: "bb-accounts-transactions-journey", providers: [AccountsTransactionsJourneyService], ngImport: i0, template: '<router-outlet></router-outlet>', isInline: true, directives: [{ type: i1.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountsTransactionsJourneyComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-accounts-transactions-journey',
                    template: '<router-outlet></router-outlet>',
                    providers: [AccountsTransactionsJourneyService],
                }]
        }] });
