// @backbase/ui-ang/esm2015/_example/iban-pipe/examples/iban-pipe.example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/iban-pipe";
export class IbanPipeExampleComponent {
    constructor() {
        this.productNumber = 'NL81TRIO0212471066';
    }
}
IbanPipeExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IbanPipeExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
IbanPipeExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: IbanPipeExampleComponent, selector: "bb-iban-pipe-example-ui", ngImport: i0, template: "<div>\n  {{ productNumber | bbIban }}\n</div>\n", pipes: { "bbIban": i1.BbIbanPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: IbanPipeExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-iban-pipe-example-ui',
                    templateUrl: './iban-pipe.example.component.html',
                }]
        }] });
