// @backbase/pay-bills-journey-ang/esm2015/src/components/pay-bills-journey-ui.module.js
import { BillpayPaymentMultipleBillsComponent } from '../components/billpay-payment-multiple-bills/billpay-payment-multiple-bills.component';
import { BillpayPaymentMultipleBillsSumBoardComponent } from '../components/billpay-payment-multiple-bills/billpay-payment-multiple-bills-sum-board/billpay-payment-multiple-bills-sum-board.component';
import { BillpayPaymentMultipleBillsListItemComponent } from '../components/billpay-payment-multiple-bills/billpay-payment-multiple-bills-list-item/billpay-payment-multiple-bills-list-item.component';
import { BillpayPaymentMultipleBillsListComponent } from '../components/billpay-payment-multiple-bills/billpay-payment-multiple-bills-list/billpay-payment-multiple-bills-list.component';
import { BillpayPaymentFrequencyComponent } from '../components/form-components/billpay-payment-frequency/billpay-payment-frequency.component';
import { BillpayPaymentStartDateComponent } from '../components/form-components/billpay-payment-start-date/billpay-payment-start-date.component';
import { BillpayPaymentEndDateComponent } from '../components/form-components/billpay-payment-end-date/billpay-payment-end-date.component';
import { BillpayPaymentMultipleBillsReviewComponent } from '../components/billpay-payment-multiple-bills/billpay-payment-multiple-ebills-review/billpay-payment-multiple-bills-review.component';
import { BillpayPaymentMultipleBillsReviewItemComponent } from '../components/billpay-payment-multiple-bills/billpay-payment-multiple-ebills-review-item/billpay-payment-multiple-bills-review-item.component';
import { BillpayPaymentMultipleBillsSuccessComponent } from '../components/billpay-payment-multiple-bills/billpay-payment-multiple-ebills-success/billpay-payment-multiple-bills-success.component';
import { BillpayPaymentCompactProductSelectorComponent } from '../components/form-components/billpay-payment-compact-product-selector/billpay-payment-compact-product-selector.component';
import { BillpayPaymentMultipleBillsReviewTableComponent } from '../components/billpay-payment-multiple-bills/billpay-payment-multiple-ebills-review-table/billpay-payment-multiple-bills-review-table.component';
import { BillpayPaymentFormContainerComponent } from '../components/billpay-payment-single-bill/billpay-payment-form-container/billpay-payment-form-container.component';
import { BillpayEbillPaymentUIComponent } from './billpay-payment-single-bill/billpay-ebill-payment-ui/billpay-ebill-payment-ui.component';
import { ScheduleComponent } from './billpay-payment-single-bill/bb-schedule-ui/schedule.component';
import { ScheduleLabelComponent } from './billpay-payment-single-bill/bb-schedule-label-ui/schedule-label.component';
import { BillpayPaymentDiscardChangesModalComponent } from './billpay-payment-single-bill/billpay-payment-discard-changes-modal/billpay-payment-discard-changes-modal.component';
import { BillpayPaymentFormComponent } from './billpay-payment-single-bill/billpay-payment-form/billpay-payment-form.component';
import { BillpayPaymentRecurringEditReviewContainerComponent } from './billpay-payment-single-bill/billpay-payment-recurring-edit-review-container/billpay-payment-recurring-edit-review-container.component';
import { BillpayPaymentReviewContainerComponent } from './billpay-payment-single-bill/billpay-payment-review-container/billpay-payment-review-container.component';
import { BillpayPaymentRecurringEditFormContainerComponent } from './billpay-payment-single-bill/billpay-payment-recurring-edit-form-container/billpay-payment-recurring-edit-form-container.component';
import { BillpayPaymentOneOffEditReviewContainerComponent } from './billpay-payment-single-bill/billpay-payment-one-off-edit-review-container/billpay-payment-one-off-edit-review-container.component';
import { BillpayPaymentOneOffEditFormContainerComponent } from './billpay-payment-single-bill/billpay-payment-one-off-edit-form-container/billpay-payment-one-off-edit-form-container.component';
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { BillpayJourneysCommonModule } from '@backbase/billpay-journeys-common';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { AlertModule } from '@backbase/ui-ang/alert';
import { AmountInputModule } from '@backbase/ui-ang/amount-input';
import { AmountModule } from '@backbase/ui-ang/amount';
import { BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import { ButtonModule } from '@backbase/ui-ang/button';
import { CharCounterModule } from '@backbase/ui-ang/char-counter';
import { CollapsibleModule } from '@backbase/ui-ang/collapsible';
import { CurrencyInputModule } from '@backbase/ui-ang/currency-input';
import { DropdownSingleSelectModule } from '@backbase/ui-ang/dropdown-single-select';
import { DropdownMenuModule } from '@backbase/ui-ang/dropdown-menu';
import { EllipsisModule } from '@backbase/ui-ang/ellipsis';
import { HeaderModule } from '@backbase/ui-ang/header';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { FieldsetModule } from '@backbase/ui-ang/fieldset';
import { IconModule } from '@backbase/ui-ang/icon';
import { InputCheckboxModule } from '@backbase/ui-ang/input-checkbox';
import { InputDatepickerModule } from '@backbase/ui-ang/input-datepicker';
import { InputNumberModule } from '@backbase/ui-ang/input-number';
import { InputRadioGroupModule } from '@backbase/ui-ang/input-radio-group';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { ModalModule } from '@backbase/ui-ang/modal';
import { ProductItemBasicAccountModule } from '@backbase/ui-ang/product-item-basic-account';
import { ProductItemCurrentAccountModule } from '@backbase/ui-ang/product-item-current-account';
import { ProductSelectorModule } from '@backbase/ui-ang/product-selector';
import { TextareaModule } from '@backbase/ui-ang/textarea';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import * as i0 from "@angular/core";
const uiAngModules = [
    LoadingIndicatorModule,
    BbDatePipeModule,
    EmptyStateModule,
    ModalModule,
    DropdownSingleSelectModule,
    InputRadioGroupModule,
    InputNumberModule,
    ButtonModule,
    InputCheckboxModule,
    ProductItemCurrentAccountModule,
    ProductItemBasicAccountModule,
    ProductSelectorModule,
    CurrencyInputModule,
    CharCounterModule,
    IconModule,
    FieldsetModule,
    AlertModule,
    InputValidationMessageModule,
    InputDatepickerModule,
    CollapsibleModule,
    TextareaModule,
    AmountInputModule,
    AlertModule,
    AmountModule,
    DropdownMenuModule,
    EllipsisModule,
    HeaderModule,
    LoadButtonModule,
    TooltipModule,
];
const components = [
    BillpayPaymentMultipleBillsComponent,
    BillpayPaymentMultipleBillsSumBoardComponent,
    BillpayPaymentMultipleBillsListItemComponent,
    BillpayPaymentMultipleBillsListComponent,
    BillpayPaymentMultipleBillsReviewComponent,
    BillpayPaymentMultipleBillsReviewItemComponent,
    BillpayPaymentMultipleBillsReviewTableComponent,
    BillpayPaymentMultipleBillsSuccessComponent,
    BillpayPaymentFrequencyComponent,
    BillpayPaymentStartDateComponent,
    BillpayPaymentEndDateComponent,
    BillpayPaymentCompactProductSelectorComponent,
    BillpayPaymentFormContainerComponent,
    BillpayEbillPaymentUIComponent,
    ScheduleComponent,
    ScheduleLabelComponent,
    BillpayPaymentDiscardChangesModalComponent,
    BillpayPaymentFormComponent,
    BillpayPaymentRecurringEditReviewContainerComponent,
    BillpayPaymentReviewContainerComponent,
    BillpayPaymentRecurringEditFormContainerComponent,
    BillpayPaymentOneOffEditReviewContainerComponent,
    BillpayPaymentOneOffEditFormContainerComponent,
];
export class PayBillsJourneyUiModule {
}
PayBillsJourneyUiModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PayBillsJourneyUiModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PayBillsJourneyUiModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PayBillsJourneyUiModule, declarations: [BillpayPaymentMultipleBillsComponent,
        BillpayPaymentMultipleBillsSumBoardComponent,
        BillpayPaymentMultipleBillsListItemComponent,
        BillpayPaymentMultipleBillsListComponent,
        BillpayPaymentMultipleBillsReviewComponent,
        BillpayPaymentMultipleBillsReviewItemComponent,
        BillpayPaymentMultipleBillsReviewTableComponent,
        BillpayPaymentMultipleBillsSuccessComponent,
        BillpayPaymentFrequencyComponent,
        BillpayPaymentStartDateComponent,
        BillpayPaymentEndDateComponent,
        BillpayPaymentCompactProductSelectorComponent,
        BillpayPaymentFormContainerComponent,
        BillpayEbillPaymentUIComponent,
        ScheduleComponent,
        ScheduleLabelComponent,
        BillpayPaymentDiscardChangesModalComponent,
        BillpayPaymentFormComponent,
        BillpayPaymentRecurringEditReviewContainerComponent,
        BillpayPaymentReviewContainerComponent,
        BillpayPaymentRecurringEditFormContainerComponent,
        BillpayPaymentOneOffEditReviewContainerComponent,
        BillpayPaymentOneOffEditFormContainerComponent], imports: [CommonModule, ReactiveFormsModule, NgbModule, BillpayJourneysCommonModule, LoadingIndicatorModule,
        BbDatePipeModule,
        EmptyStateModule,
        ModalModule,
        DropdownSingleSelectModule,
        InputRadioGroupModule,
        InputNumberModule,
        ButtonModule,
        InputCheckboxModule,
        ProductItemCurrentAccountModule,
        ProductItemBasicAccountModule,
        ProductSelectorModule,
        CurrencyInputModule,
        CharCounterModule,
        IconModule,
        FieldsetModule,
        AlertModule,
        InputValidationMessageModule,
        InputDatepickerModule,
        CollapsibleModule,
        TextareaModule,
        AmountInputModule,
        AlertModule,
        AmountModule,
        DropdownMenuModule,
        EllipsisModule,
        HeaderModule,
        LoadButtonModule,
        TooltipModule], exports: [BillpayPaymentMultipleBillsComponent,
        BillpayPaymentMultipleBillsSumBoardComponent,
        BillpayPaymentMultipleBillsListItemComponent,
        BillpayPaymentMultipleBillsListComponent,
        BillpayPaymentMultipleBillsReviewComponent,
        BillpayPaymentMultipleBillsReviewItemComponent,
        BillpayPaymentMultipleBillsReviewTableComponent,
        BillpayPaymentMultipleBillsSuccessComponent,
        BillpayPaymentFrequencyComponent,
        BillpayPaymentStartDateComponent,
        BillpayPaymentEndDateComponent,
        BillpayPaymentCompactProductSelectorComponent,
        BillpayPaymentFormContainerComponent,
        BillpayEbillPaymentUIComponent,
        ScheduleComponent,
        ScheduleLabelComponent,
        BillpayPaymentDiscardChangesModalComponent,
        BillpayPaymentFormComponent,
        BillpayPaymentRecurringEditReviewContainerComponent,
        BillpayPaymentReviewContainerComponent,
        BillpayPaymentRecurringEditFormContainerComponent,
        BillpayPaymentOneOffEditReviewContainerComponent,
        BillpayPaymentOneOffEditFormContainerComponent, LoadingIndicatorModule,
        BbDatePipeModule,
        EmptyStateModule,
        ModalModule,
        DropdownSingleSelectModule,
        InputRadioGroupModule,
        InputNumberModule,
        ButtonModule,
        InputCheckboxModule,
        ProductItemCurrentAccountModule,
        ProductItemBasicAccountModule,
        ProductSelectorModule,
        CurrencyInputModule,
        CharCounterModule,
        IconModule,
        FieldsetModule,
        AlertModule,
        InputValidationMessageModule,
        InputDatepickerModule,
        CollapsibleModule,
        TextareaModule,
        AmountInputModule,
        AlertModule,
        AmountModule,
        DropdownMenuModule,
        EllipsisModule,
        HeaderModule,
        LoadButtonModule,
        TooltipModule] });
PayBillsJourneyUiModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PayBillsJourneyUiModule, imports: [[CommonModule, ReactiveFormsModule, NgbModule, BillpayJourneysCommonModule, ...uiAngModules], LoadingIndicatorModule,
        BbDatePipeModule,
        EmptyStateModule,
        ModalModule,
        DropdownSingleSelectModule,
        InputRadioGroupModule,
        InputNumberModule,
        ButtonModule,
        InputCheckboxModule,
        ProductItemCurrentAccountModule,
        ProductItemBasicAccountModule,
        ProductSelectorModule,
        CurrencyInputModule,
        CharCounterModule,
        IconModule,
        FieldsetModule,
        AlertModule,
        InputValidationMessageModule,
        InputDatepickerModule,
        CollapsibleModule,
        TextareaModule,
        AmountInputModule,
        AlertModule,
        AmountModule,
        DropdownMenuModule,
        EllipsisModule,
        HeaderModule,
        LoadButtonModule,
        TooltipModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PayBillsJourneyUiModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components],
                    imports: [CommonModule, ReactiveFormsModule, NgbModule, BillpayJourneysCommonModule, ...uiAngModules],
                    exports: [...components, ...uiAngModules],
                }]
        }] });
