// @backbase/messages-common/esm2015/src/services/messages-employee-conversation-counters/messages-employee-conversation-counters.service.js
import { Inject, Injectable } from '@angular/core';
import { merge, of, Subject, timer } from 'rxjs';
import { concatMap, switchMap } from 'rxjs/operators';
import { PUBSUB } from '@backbase/foundation-ang/web-sdk';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/messages-v5-http-ang";
const MessagesPollPeriod = 5000;
const bbEventUnreadConversationCountUpdate = 'bb.event.unread-conversation-count.update';
export class MessagesEmployeeConversationCountersService {
    constructor(service, pubSub) {
        this.service = service;
        this.pubSub = pubSub;
        this.resetSubject = new Subject();
        this.subscriptionHandler = this.reset.bind(this);
        this.pubSub.subscribe(bbEventUnreadConversationCountUpdate, this.subscriptionHandler);
    }
    /**
     * Conversations count poll source
     */
    source() {
        return () => merge(this.resetSubject, of(undefined)).pipe(switchMap(() => timer(0, MessagesPollPeriod).pipe(concatMap(() => this.getConversationCounters()))));
    }
    /**
     * Get conversations count
     *
     * @returns Get conversation counters stream
     */
    getConversationCounters() {
        return this.service.getConversationsCounter();
    }
    reset() {
        this.resetSubject.next();
    }
}
MessagesEmployeeConversationCountersService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesEmployeeConversationCountersService, deps: [{ token: i1.EmployeeHttpService }, { token: PUBSUB }], target: i0.ɵɵFactoryTarget.Injectable });
MessagesEmployeeConversationCountersService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesEmployeeConversationCountersService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesEmployeeConversationCountersService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.EmployeeHttpService }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [PUBSUB]
                }] }]; } });
