// @backbase/messages-common/esm2015/src/services/messages-employee-conversation-counters/messages-employee-conversation-counters.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { PUBSUB } from '@backbase/foundation-ang/web-sdk';
import { EmployeeHttpService } from '@backbase/messages-v5-http-ang';
import { MessagesEmployeeConversationCountersService } from './messages-employee-conversation-counters.service';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
export class MessagesEmployeeConversationCountersModule {
}
MessagesEmployeeConversationCountersModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesEmployeeConversationCountersModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MessagesEmployeeConversationCountersModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesEmployeeConversationCountersModule, imports: [CommonModule, i1.BackbaseCoreModule] });
MessagesEmployeeConversationCountersModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesEmployeeConversationCountersModule, providers: [MessagesEmployeeConversationCountersService], imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                pipelineProcessors: [
                    {
                        name: 'tab-counters-badge',
                        provider: {
                            provide: MessagesEmployeeConversationCountersService,
                            useClass: MessagesEmployeeConversationCountersService,
                            deps: [EmployeeHttpService, PUBSUB],
                        },
                    },
                ],
            }),
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesEmployeeConversationCountersModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            pipelineProcessors: [
                                {
                                    name: 'tab-counters-badge',
                                    provider: {
                                        provide: MessagesEmployeeConversationCountersService,
                                        useClass: MessagesEmployeeConversationCountersService,
                                        deps: [EmployeeHttpService, PUBSUB],
                                    },
                                },
                            ],
                        }),
                    ],
                    providers: [MessagesEmployeeConversationCountersService],
                }]
        }] });
