// @backbase/budget-journey-ang/bundles/backbase-budget-journey-ang.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core'), require('@angular/router'), require('@backbase/budget-widget-ang'), require('@backbase/foundation-ang/core'), require('@backbase/ui-ang/header')) :
    typeof define === 'function' && define.amd ? define('@backbase/budget-journey-ang', ['exports', '@angular/common', '@angular/core', '@angular/router', '@backbase/budget-widget-ang', '@backbase/foundation-ang/core', '@backbase/ui-ang/header'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["budget-journey-ang"] = {}), global.ng.common, global.ng.core, global.ng.router, global.i3, global.i1$1, global.i2));
})(this, (function (exports, common, i0, i1, i3, i1$1, i2) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i3__namespace = /*#__PURE__*/_interopNamespace(i3);
    var i1__namespace$1 = /*#__PURE__*/_interopNamespace(i1$1);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);

    // eslint-disable-next-line
    var BudgetJourneyConfigurationToken = new i0.InjectionToken('BudgetJourneyConfiguration injection token');
    var configDefaults = {
        showPercentage: false,
        budgetSafeZoneLimit: 80,
        notificationDismissTime: 5,
        maxBudgets: null,
    };
    var BudgetJourneyService = /** @class */ (function () {
        function BudgetJourneyService(config) {
            config = config || {};
            this._config = Object.assign(Object.assign({}, configDefaults), config);
        }
        Object.defineProperty(BudgetJourneyService.prototype, "defaults", {
            get: function () {
                return configDefaults;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(BudgetJourneyService.prototype, "showPercentage", {
            get: function () {
                return this._config.showPercentage;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(BudgetJourneyService.prototype, "budgetSafeZoneLimit", {
            get: function () {
                return this._config.budgetSafeZoneLimit;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(BudgetJourneyService.prototype, "notificationDismissTime", {
            get: function () {
                return this._config.notificationDismissTime;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(BudgetJourneyService.prototype, "maxBudgets", {
            get: function () {
                return this._config.maxBudgets;
            },
            enumerable: false,
            configurable: true
        });
        return BudgetJourneyService;
    }());
    BudgetJourneyService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BudgetJourneyService, deps: [{ token: BudgetJourneyConfigurationToken, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    BudgetJourneyService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BudgetJourneyService });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BudgetJourneyService, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () {
            return [{ type: undefined, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.Inject,
                            args: [BudgetJourneyConfigurationToken]
                        }] }];
        } });

    var BudgetJourneyComponent = /** @class */ (function () {
        function BudgetJourneyComponent() {
        }
        return BudgetJourneyComponent;
    }());
    BudgetJourneyComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BudgetJourneyComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    BudgetJourneyComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: BudgetJourneyComponent, selector: "bb-budget-journey", providers: [BudgetJourneyService], ngImport: i0__namespace, template: '<router-outlet></router-outlet>', isInline: true, directives: [{ type: i1__namespace.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BudgetJourneyComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-budget-journey',
                        template: '<router-outlet></router-outlet>',
                        providers: [BudgetJourneyService],
                    }]
            }] });

    // TODO: remove utility classes
    var BudgetViewComponent = /** @class */ (function () {
        function BudgetViewComponent(config) {
            this.config = config;
        }
        return BudgetViewComponent;
    }());
    BudgetViewComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BudgetViewComponent, deps: [{ token: BudgetJourneyService }], target: i0__namespace.ɵɵFactoryTarget.Component });
    BudgetViewComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: BudgetViewComponent, selector: "bb-budgets-view", ngImport: i0__namespace, template: "\n    <bb-header-ui\n      headingType=\"h1\"\n      headingClasses=\"bb-heading-widget__heading px-0 pb-5\"\n      heading=\"Budgets\"\n      i18n-heading=\"Budgets|Heading for budget journey@@budget.journey.heading.text\"\n    ></bb-header-ui>\n    <bb-budget-widget\n      [showPercentage]=\"config.showPercentage\"\n      [budgetSafeZoneLimit]=\"config.budgetSafeZoneLimit\"\n      [notificationDismissTime]=\"config.notificationDismissTime\"\n      [maxBudgets]=\"config.maxBudgets\"\n    ></bb-budget-widget>\n  ", isInline: true, components: [{ type: i2__namespace.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i3__namespace.BudgetWidgetComponent, selector: "bb-budget-widget", inputs: ["showPercentage", "budgetSafeZoneLimit", "notificationDismissTime", "maxBudgets"] }] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BudgetViewComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-budgets-view',
                        template: "\n    <bb-header-ui\n      headingType=\"h1\"\n      headingClasses=\"bb-heading-widget__heading px-0 pb-5\"\n      heading=\"Budgets\"\n      i18n-heading=\"Budgets|Heading for budget journey@@budget.journey.heading.text\"\n    ></bb-header-ui>\n    <bb-budget-widget\n      [showPercentage]=\"config.showPercentage\"\n      [budgetSafeZoneLimit]=\"config.budgetSafeZoneLimit\"\n      [notificationDismissTime]=\"config.notificationDismissTime\"\n      [maxBudgets]=\"config.maxBudgets\"\n    ></bb-budget-widget>\n  ",
                    }]
            }], ctorParameters: function () { return [{ type: BudgetJourneyService }]; } });

    var defaultRoute = {
        path: '',
        component: BudgetJourneyComponent,
        children: [
            {
                path: '',
                component: BudgetViewComponent,
            },
        ],
    };
    var BudgetJourneyModule = /** @class */ (function () {
        function BudgetJourneyModule() {
        }
        BudgetJourneyModule.forRoot = function (data) {
            if (data === void 0) { data = { route: defaultRoute }; }
            return {
                ngModule: BudgetJourneyModule,
                providers: [i1.provideRoutes([data.route])],
            };
        };
        return BudgetJourneyModule;
    }());
    BudgetJourneyModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BudgetJourneyModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    BudgetJourneyModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BudgetJourneyModule, declarations: [BudgetJourneyComponent, BudgetViewComponent], imports: [common.CommonModule, i1__namespace$1.BackbaseCoreModule, i1.RouterModule,
            i2.HeaderModule,
            i3.BudgetWidgetModule] });
    BudgetJourneyModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BudgetJourneyModule, imports: [[
                common.CommonModule,
                i1$1.BackbaseCoreModule.withConfig({
                    classMap: { BudgetJourneyComponent: BudgetJourneyComponent },
                }),
                i1.RouterModule,
                i2.HeaderModule,
                i3.BudgetWidgetModule,
            ]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BudgetJourneyModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [BudgetJourneyComponent, BudgetViewComponent],
                        imports: [
                            common.CommonModule,
                            i1$1.BackbaseCoreModule.withConfig({
                                classMap: { BudgetJourneyComponent: BudgetJourneyComponent },
                            }),
                            i1.RouterModule,
                            i2.HeaderModule,
                            i3.BudgetWidgetModule,
                        ],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.BudgetJourneyComponent = BudgetJourneyComponent;
    exports.BudgetJourneyConfigurationToken = BudgetJourneyConfigurationToken;
    exports.BudgetJourneyModule = BudgetJourneyModule;
    exports.BudgetJourneyService = BudgetJourneyService;
    exports.BudgetViewComponent = BudgetViewComponent;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
