// @backbase/payment-orders-ang/esm2015/lib/ui/payment-orders-ui.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { PaymentsSrSearchResultInfoModule } from './components/sr-search-result-info/sr-search-result-info.module';
import { AddressDetailsComponent } from './components/address-details/address-details.component';
import { FullAddressComponent } from './components/full-address/full-address.component';
import { CurrencySelectorComponent } from './components/currency-selector/currency-selector.component';
import { CurrencyAmountSelectorComponent } from './components/currency-amount-selector/currency-amount-selector.component';
import { PayordDynamicLabelComponent } from './components/dynamic-label/dynamic-label.component';
import { PaymentDebitorDetailsCardComponent } from './components/debitor-details-card/debitor-details-card.component';
import { PaymentCreditorDetailsCardComponent } from './components/creditor-details-card/creditor-details-card.component';
import { PayordScheduleDetailsComponent } from './components/schedule-details/schedule-details.component';
import { PayordCounterpartyDetailsComponent } from './components/counterparty-details/counterparty-details.component';
import { PayordStatusComponent } from './components/status-card/status-card.component';
import { PayordAmountDetailsComponent } from './components/amount-details/amount-details.component';
import { FlagIconComponent } from './components/flag-icon/bb-flag-icon';
import { CountryNameComponent } from './components/country-name/country-name.component';
import { CountrySelectorComponent } from './components/country-selector/country-selector.component';
import { SelectCountryDropdownComponent } from './components/select-country-dropdown/select-country-dropdown.component';
import { NgbDropdownModule } from '@ng-bootstrap/ng-bootstrap';
import { PaymentTemplateComponent } from './components/create-payment-template-modal/create-template-form-modal.component';
import { BreachLimitDetailsModalComponent } from './components/breach-limit-details-modal/breach-limit-details.component';
import { PayordCardMenuComponent } from './components/card-menu/card-menu.component';
import { SelectPurposeDropdownComponent } from './components/select-purpose-dropdown/select-purpose-dropdown.component';
import { RoutableModalComponent } from './components/routable-modal/bb-routable-modal.component';
import { HorizontalListSelectorComponent } from './components/horizontal-list-selector/horizontal-list-selector.component';
import { PaymentTemplateSelectorComponent } from './components/payment-template-selector/payment-template-selector.component';
import { BeneficiarySelectorUIModule } from './components/beneficiary-selector-ui/beneficiary-selector-ui.module';
import { PurposeOfPaymentDetailComponent } from './components/purpose-of-payment-detail/purpose-of-payment-detail.component';
import { PaymentsTableComponent } from './components/bb-table-component/bb-table-payments.component';
import { PaymentsListComponent } from './components/bb-list-payments-component/bb-list-payments.component';
import { ScheduleLabelModule } from './components/bb-schedule-label-ui/schedule-label.module';
import { ConfirmationDialogModule } from './components/confirmation-dialog/confirmation-dialog.module';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import { ButtonModule } from '@backbase/ui-ang/button';
import { IconModule } from '@backbase/ui-ang/icon';
import { DropdownMenuModule } from '@backbase/ui-ang/dropdown-menu';
import { DropdownSingleSelectModule } from '@backbase/ui-ang/dropdown-single-select';
import { InfiniteScrollModule } from '@backbase/ui-ang/infinite-scroll';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { AvatarModule } from '@backbase/ui-ang/avatar';
import { AmountModule } from '@backbase/ui-ang/amount';
import { SearchBoxModule } from '@backbase/ui-ang/search-box';
import { ProductSelectorModule } from '@backbase/ui-ang/product-selector';
import { KeyboardClickModule } from '@backbase/ui-ang/keyboard-click-directive';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { ModalModule } from '@backbase/ui-ang/modal';
import { HeaderModule } from '@backbase/ui-ang/header';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { EllipsisModule } from '@backbase/ui-ang/ellipsis';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { HighlightModule } from '@backbase/ui-ang/highlight';
import { PaginationModule } from '@backbase/ui-ang/pagination';
import { TableModule } from '@backbase/ui-ang/table';
import { PaymentsControlsModule } from './components/payments-controls/payments-controls.module';
import { PaymentsFilterModule } from './components/payments-filter/payments-filter.module';
import { PaymentsSearchModule } from './components/payments-search/payments-search.module';
import { ScheduleModule } from './components/schedule/schedule.module';
import { SeparatorModule } from './components/separator/separator.module';
import { PayordStatusBadgeModule } from './components/status-badge/status-badge.module';
import { TableCheckboxModule } from './components/table-checkbox/table-checkbox.module';
import { TableHeaderCheckboxModule } from './components/table-header-checkbox/table-header-checkbox.module';
import { IbanModule } from './directives/iban.module';
import { AccountFormatterModule } from './pipes/account-formatter/account-formatter.module';
import { BytesModule } from './pipes/bytes-pipe/bytes.module';
import { BBHighlightModule } from './pipes/highlight/highlight.module';
import { RelativeDateModule } from './pipes/relative-date/relative-date.module';
import { ExportPdfModule } from './directives/export-pdf/export-pdf.module';
import * as i0 from "@angular/core";
const components = [
    AddressDetailsComponent,
    FullAddressComponent,
    PaymentDebitorDetailsCardComponent,
    PaymentCreditorDetailsCardComponent,
    PayordCounterpartyDetailsComponent,
    PayordScheduleDetailsComponent,
    PayordStatusComponent,
    PayordAmountDetailsComponent,
    PayordDynamicLabelComponent,
    FlagIconComponent,
    CountryNameComponent,
    CountrySelectorComponent,
    PayordDynamicLabelComponent,
    FlagIconComponent,
    CountryNameComponent,
    SelectCountryDropdownComponent,
    CurrencySelectorComponent,
    CurrencyAmountSelectorComponent,
    PaymentTemplateComponent,
    BreachLimitDetailsModalComponent,
    PayordCardMenuComponent,
    SelectPurposeDropdownComponent,
    RoutableModalComponent,
    HorizontalListSelectorComponent,
    PaymentTemplateSelectorComponent,
    PurposeOfPaymentDetailComponent,
    PaymentsTableComponent,
    PaymentsListComponent,
];
const modules = [
    InputTextModule,
    ButtonModule,
    IconModule,
    DropdownMenuModule,
    DropdownSingleSelectModule,
    InfiniteScrollModule,
    InputValidationMessageModule,
    EmptyStateModule,
    SearchBoxModule,
    NgbDropdownModule,
    AvatarModule,
    AmountModule,
    SearchBoxModule,
    ProductSelectorModule,
    KeyboardClickModule,
    EmptyStateModule,
    ModalModule,
    HeaderModule,
    LoadButtonModule,
    LoadingIndicatorModule,
    EllipsisModule,
    TooltipModule,
    PaginationModule,
    TableModule,
    HighlightModule,
    BeneficiarySelectorUIModule,
    PaymentsSrSearchResultInfoModule,
    ScheduleLabelModule,
    ConfirmationDialogModule,
    PaymentsControlsModule,
    PaymentsFilterModule,
    PaymentsSearchModule,
    ScheduleModule,
    SeparatorModule,
    PayordStatusBadgeModule,
    TableHeaderCheckboxModule,
    TableCheckboxModule,
    IbanModule,
    AccountFormatterModule,
    BytesModule,
    BBHighlightModule,
    RelativeDateModule,
    ExportPdfModule,
];
export class PaymentOrdersUiModule {
}
/** @nocollapse */ PaymentOrdersUiModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentOrdersUiModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ PaymentOrdersUiModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentOrdersUiModule, declarations: [AddressDetailsComponent,
        FullAddressComponent,
        PaymentDebitorDetailsCardComponent,
        PaymentCreditorDetailsCardComponent,
        PayordCounterpartyDetailsComponent,
        PayordScheduleDetailsComponent,
        PayordStatusComponent,
        PayordAmountDetailsComponent,
        PayordDynamicLabelComponent,
        FlagIconComponent,
        CountryNameComponent,
        CountrySelectorComponent,
        PayordDynamicLabelComponent,
        FlagIconComponent,
        CountryNameComponent,
        SelectCountryDropdownComponent,
        CurrencySelectorComponent,
        CurrencyAmountSelectorComponent,
        PaymentTemplateComponent,
        BreachLimitDetailsModalComponent,
        PayordCardMenuComponent,
        SelectPurposeDropdownComponent,
        RoutableModalComponent,
        HorizontalListSelectorComponent,
        PaymentTemplateSelectorComponent,
        PurposeOfPaymentDetailComponent,
        PaymentsTableComponent,
        PaymentsListComponent], imports: [CommonModule, FormsModule, ReactiveFormsModule, InputTextModule,
        ButtonModule,
        IconModule,
        DropdownMenuModule,
        DropdownSingleSelectModule,
        InfiniteScrollModule,
        InputValidationMessageModule,
        EmptyStateModule,
        SearchBoxModule,
        NgbDropdownModule,
        AvatarModule,
        AmountModule,
        SearchBoxModule,
        ProductSelectorModule,
        KeyboardClickModule,
        EmptyStateModule,
        ModalModule,
        HeaderModule,
        LoadButtonModule,
        LoadingIndicatorModule,
        EllipsisModule,
        TooltipModule,
        PaginationModule,
        TableModule,
        HighlightModule,
        BeneficiarySelectorUIModule,
        PaymentsSrSearchResultInfoModule,
        ScheduleLabelModule,
        ConfirmationDialogModule,
        PaymentsControlsModule,
        PaymentsFilterModule,
        PaymentsSearchModule,
        ScheduleModule,
        SeparatorModule,
        PayordStatusBadgeModule,
        TableHeaderCheckboxModule,
        TableCheckboxModule,
        IbanModule,
        AccountFormatterModule,
        BytesModule,
        BBHighlightModule,
        RelativeDateModule,
        ExportPdfModule], exports: [AddressDetailsComponent,
        FullAddressComponent,
        PaymentDebitorDetailsCardComponent,
        PaymentCreditorDetailsCardComponent,
        PayordCounterpartyDetailsComponent,
        PayordScheduleDetailsComponent,
        PayordStatusComponent,
        PayordAmountDetailsComponent,
        PayordDynamicLabelComponent,
        FlagIconComponent,
        CountryNameComponent,
        CountrySelectorComponent,
        PayordDynamicLabelComponent,
        FlagIconComponent,
        CountryNameComponent,
        SelectCountryDropdownComponent,
        CurrencySelectorComponent,
        CurrencyAmountSelectorComponent,
        PaymentTemplateComponent,
        BreachLimitDetailsModalComponent,
        PayordCardMenuComponent,
        SelectPurposeDropdownComponent,
        RoutableModalComponent,
        HorizontalListSelectorComponent,
        PaymentTemplateSelectorComponent,
        PurposeOfPaymentDetailComponent,
        PaymentsTableComponent,
        PaymentsListComponent, InputTextModule,
        ButtonModule,
        IconModule,
        DropdownMenuModule,
        DropdownSingleSelectModule,
        InfiniteScrollModule,
        InputValidationMessageModule,
        EmptyStateModule,
        SearchBoxModule,
        NgbDropdownModule,
        AvatarModule,
        AmountModule,
        SearchBoxModule,
        ProductSelectorModule,
        KeyboardClickModule,
        EmptyStateModule,
        ModalModule,
        HeaderModule,
        LoadButtonModule,
        LoadingIndicatorModule,
        EllipsisModule,
        TooltipModule,
        PaginationModule,
        TableModule,
        HighlightModule,
        BeneficiarySelectorUIModule,
        PaymentsSrSearchResultInfoModule,
        ScheduleLabelModule,
        ConfirmationDialogModule,
        PaymentsControlsModule,
        PaymentsFilterModule,
        PaymentsSearchModule,
        ScheduleModule,
        SeparatorModule,
        PayordStatusBadgeModule,
        TableHeaderCheckboxModule,
        TableCheckboxModule,
        IbanModule,
        AccountFormatterModule,
        BytesModule,
        BBHighlightModule,
        RelativeDateModule,
        ExportPdfModule] });
/** @nocollapse */ PaymentOrdersUiModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentOrdersUiModule, imports: [[CommonModule, FormsModule, ReactiveFormsModule, ...modules], InputTextModule,
        ButtonModule,
        IconModule,
        DropdownMenuModule,
        DropdownSingleSelectModule,
        InfiniteScrollModule,
        InputValidationMessageModule,
        EmptyStateModule,
        SearchBoxModule,
        NgbDropdownModule,
        AvatarModule,
        AmountModule,
        SearchBoxModule,
        ProductSelectorModule,
        KeyboardClickModule,
        EmptyStateModule,
        ModalModule,
        HeaderModule,
        LoadButtonModule,
        LoadingIndicatorModule,
        EllipsisModule,
        TooltipModule,
        PaginationModule,
        TableModule,
        HighlightModule,
        BeneficiarySelectorUIModule,
        PaymentsSrSearchResultInfoModule,
        ScheduleLabelModule,
        ConfirmationDialogModule,
        PaymentsControlsModule,
        PaymentsFilterModule,
        PaymentsSearchModule,
        ScheduleModule,
        SeparatorModule,
        PayordStatusBadgeModule,
        TableHeaderCheckboxModule,
        TableCheckboxModule,
        IbanModule,
        AccountFormatterModule,
        BytesModule,
        BBHighlightModule,
        RelativeDateModule,
        ExportPdfModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PaymentOrdersUiModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, FormsModule, ReactiveFormsModule, ...modules],
                    declarations: [...components],
                    exports: [...components, ...modules],
                }]
        }] });
