// @backbase/foundation-ang/esm2015/entitlements/guards/entitlements.guard.js
import { __awaiter } from "tslib";
import { Injectable } from '@angular/core';
import { Router, } from '@angular/router';
import { ConditionsService } from '../services/conditions.service';
import { Observable } from 'rxjs';
import { first } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "../services/conditions.service";
import * as i2 from "@angular/router";
export class EntitlementsGuard {
    constructor(conditions, router) {
        this.conditions = conditions;
        this.router = router;
        this.userHasEntitlements = (entitlements) => __awaiter(this, void 0, void 0, function* () {
            try {
                if (!entitlements || (yield this.conditions.resolveEntitlements(entitlements))) {
                    return true;
                }
            }
            catch (err) {
                console.error(err);
            }
            return false;
        });
    }
    resolveEntitlements(entitlements, redirectTo, route, state) {
        return __awaiter(this, void 0, void 0, function* () {
            if (yield this.userHasEntitlements(entitlements)) {
                return true;
            }
            if (typeof redirectTo === 'function') {
                redirectTo = yield this.getRedirection(redirectTo, route, state);
            }
            if (redirectTo) {
                return this.router.parseUrl(redirectTo);
            }
            return false;
        });
    }
    getRedirection(getRedirectTo, route, state) {
        if (!route || !state) {
            return undefined;
        }
        const redirectTo = getRedirectTo(this.userHasEntitlements, route, state);
        if (redirectTo instanceof Observable) {
            return redirectTo.pipe(first()).toPromise();
        }
        return redirectTo;
    }
    canActivateChild(childRoute, state) {
        const { entitlements, redirectTo } = childRoute.data;
        return this.resolveEntitlements(entitlements, redirectTo, childRoute, state);
    }
    canLoad(route, segments) {
        var _a, _b;
        const entitlements = (_a = route.data) === null || _a === void 0 ? void 0 : _a.entitlements;
        const redirectTo = (_b = route.data) === null || _b === void 0 ? void 0 : _b.redirectTo;
        return this.resolveEntitlements(entitlements, redirectTo, route, segments);
    }
    canActivate(route, state) {
        const { entitlements, redirectTo } = route.data;
        return this.resolveEntitlements(entitlements, redirectTo, route, state);
    }
}
EntitlementsGuard.ɵprov = i0.ɵɵdefineInjectable({ factory: function EntitlementsGuard_Factory() { return new EntitlementsGuard(i0.ɵɵinject(i1.ConditionsService), i0.ɵɵinject(i2.Router)); }, token: EntitlementsGuard, providedIn: "root" });
EntitlementsGuard.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root',
            },] }
];
EntitlementsGuard.ctorParameters = () => [
    { type: ConditionsService },
    { type: Router }
];
