// @backbase/messages-common/esm2015/src/components/conversations-list/filter-conversations-list/filter-conversations-list.component.js
import { Component, Directive, EventEmitter, Input, Output } from '@angular/core';
import { BehaviorSubject, combineLatest, EMPTY, Subject } from 'rxjs';
import { catchError, debounceTime, distinctUntilChanged, map, switchMap, take, takeUntil, tap } from 'rxjs/operators';
import { BbTemplate } from '@backbase/foundation-ang/core';
import { topicPlaceholderConfig } from '../../../utils/topic-placeholder-text-helper';
import * as i0 from "@angular/core";
import * as i1 from "../model/filter-service.model";
import * as i2 from "@angular/forms";
import * as i3 from "../../../services/shared-methods/shared-methods.service";
import * as i4 from "@backbase/ui-ang/icon";
import * as i5 from "@backbase/ui-ang/dropdown-single-select";
import * as i6 from "@backbase/ui-ang/search-box";
import * as i7 from "@backbase/ui-ang/input-datepicker";
import * as i8 from "@angular/common";
import * as i9 from "@backbase/ui-ang/button";
export class FilterConversationsListComponent {
    constructor(service, fb, shared) {
        this.service = service;
        this.fb = fb;
        this.shared = shared;
        /**
         * Configs for topic selector placeholder
         */
        this.topicPlaceholderConfig = topicPlaceholderConfig;
        /**
         * Event emitter for apply action
         */
        this.apply = new EventEmitter();
        /**
         * Event emitter for clear action
         */
        this.clear = new EventEmitter();
        /**
         * The filters form reference.
         */
        this.filterForm = this.fb.group({
            date: [''],
            topic: [''],
            user: [''],
        });
        /**
         * The applied flag.
         */
        this.applied = false;
        /**
         * The opened flag.
         */
        this.opened = false;
        /**
         * The stream with the topic list.
         */
        this.topics$ = new Subject();
        /**
         * The method to get the typeahead options.
         */
        this.typeaheadOptions = {
            ngbTypeahead: this.searchUsers.bind(this),
            resultFormatter: (item) => item.fullName,
            inputFormatter: (item) => item.fullName,
        };
        this.filterParams$ = new BehaviorSubject({});
        this.unsubscribe$ = new Subject();
    }
    /**
     * Observable filter params to init component with
     */
    set filterParams(value) {
        if (value && this.checkFilterParams(this.filterParams$.getValue(), value)) {
            this.applied = true;
            this.filterParams$.next(value);
        }
    }
    ngOnInit() {
        combineLatest([this.filterParams$, this.topics$])
            .pipe(switchMap(([params, topics]) => {
            const { topicId, endDate, startDate, userId } = params;
            if (topics && topicId) {
                this.filterForm.controls['topic'].setValue(topics.find((topic) => topic.id === topicId));
            }
            if (startDate && endDate) {
                this.filterForm.controls['date'].setValue({
                    from: new Date(startDate).toISOString(),
                    to: new Date(endDate).toISOString(),
                });
            }
            if (userId && this.service.getUserById) {
                return this.service
                    .getUserById(userId)
                    .pipe(tap(user => user && this.filterForm.controls['user'].setValue(Object.assign(Object.assign({}, user), { id: params.userId }))));
            }
            return EMPTY;
        }), takeUntil(this.unsubscribe$))
            .subscribe();
        this.service
            .getTopics()
            .pipe(take(1))
            .subscribe(topics => {
            this.topics$.next(topics);
            this.topics$.complete();
        });
    }
    /**
     * The method to submit the form.
     */
    submit() {
        const applyObject = this.convertToFilterParams(this.filterForm);
        this.applied = !!Object.keys(applyObject).length;
        this.opened = false;
        this.apply.emit(applyObject);
    }
    /**
     * The method to reset the form.
     *
     * @param apply Flag to submit or not the form
     */
    reset(apply = false) {
        if (apply) {
            this.filterForm.valueChanges.pipe(take(1)).subscribe(() => {
                this.submit();
            });
        }
        this.filterForm.reset();
    }
    /**
     * The method to open or close the filter form.
     */
    triggerOpened() {
        this.opened = !this.opened;
    }
    ngOnDestroy() {
        this.unsubscribe$.next();
        this.unsubscribe$.complete();
    }
    convertToFilterParams(form) {
        return Object.assign(Object.assign(Object.assign({}, (form.value.topic && { topicId: form.value.topic.id })), (form.value.user && { userId: form.value.user.id })), (form.value.date && {
            startDate: this.formatDate(form.value.date.from),
            endDate: this.formatDate(form.value.date.to),
        }));
    }
    formatDate(date) {
        const d = new Date(date);
        const year = d.getFullYear();
        const month = d.getMonth() + 1;
        const dt = d.getDate();
        const finalDt = dt < 10 ? `0${dt}` : dt.toString();
        const finalMonth = month < 10 ? `0${month}` : month.toString();
        return `${year}-${finalMonth}-${finalDt}`;
    }
    checkFilterParams(curr, next) {
        if (!Object.keys(next).length || !(next && (next.userId || next.topicId || (next.startDate && next.endDate)))) {
            return false;
        }
        for (const key in next) {
            if (curr[key] && curr[key] === next[key]) {
                return false;
            }
        }
        return true;
    }
    searchUsers(text) {
        return text.pipe(debounceTime(500), distinctUntilChanged(), switchMap(value => {
            if (!this.service.getUsers || !value.length) {
                return EMPTY;
            }
            else {
                return this.service.getUsers(value).pipe(map(response => response && response.identities), catchError(err => {
                    this.shared.openNotification(this.shared.getErrorMessage(err), 'error');
                    return EMPTY;
                }));
            }
        }));
    }
}
FilterConversationsListComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: FilterConversationsListComponent, deps: [{ token: i1.FilterService }, { token: i2.FormBuilder }, { token: i3.SharedMethodsService }], target: i0.ɵɵFactoryTarget.Component });
FilterConversationsListComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.14", type: FilterConversationsListComponent, selector: "bb-messages-filter-conversation-list", inputs: { headerClass: "headerClass", bodyClass: "bodyClass", filterParams: "filterParams" }, outputs: { apply: "apply", clear: "clear" }, ngImport: i0, template: "<ng-container\n  bbFilterConversationsListCustomizable\n  [bbHostRef]=\"this\"\n  [bbTemplateContext]=\"{\n    topics: topics$ | async\n  }\"\n></ng-container>\n\n<ng-template bbFilterConversationsListCustomizable let-host let-topics=\"context.topics\">\n  <div [ngClass]=\"host.headerClass\" data-role=\"filters-header\">\n    <div class=\"bb-stack\">\n      <div class=\"bb-stack__item\">\n        <div class=\"bb-button-bar bb-button-bar--reverse\">\n          <button\n            class=\"bb-button-bar__button\"\n            bbButton\n            [color]=\"host.applied ? 'primary' : 'secondary'\"\n            [ngClass]=\"{ active: host.opened }\"\n            (click)=\"host.triggerOpened()\"\n            data-role=\"filter-button\"\n          >\n            <bb-icon-ui [cropped]=\"true\" name=\"filter\"></bb-icon-ui>\n            <span\n              *ngIf=\"host.applied; else filtersTextDefault\"\n              i18n=\"\n                Filters applied|Text for filter button if filter was\n                applied@@messages.messagesCommon.conversationsListFilterConversation.filtersAppliedText\"\n              >Filters applied</span\n            >\n          </button>\n          <button\n            class=\"bb-button-bar__button\"\n            bbButton\n            color=\"link\"\n            *ngIf=\"host.applied\"\n            i18n=\"Clear filters@@messages.messagesCommon.conversationsListFilterConversation.clearFilterssButtonText\"\n            (click)=\"host.reset(true)\"\n            data-role=\"clear-header-button\"\n          >\n            Clear filters\n          </button>\n        </div>\n      </div>\n      <div class=\"bb-stack__item bb-stack__item--push-right\">\n        <ng-content></ng-content>\n      </div>\n    </div>\n  </div>\n  <div *ngIf=\"host.opened\" [ngClass]=\"host.bodyClass\" data-role=\"filters-form-container\">\n    <form [formGroup]=\"host.filterForm\" (submit)=\"host.submit()\">\n      <div class=\"form-row form-group\">\n        <div class=\"col-4\">\n          <bb-dropdown-single-select-ui\n            *ngIf=\"\n              topics?.length\n                ? topicPlaceholderConfig.selectTopicPlaceholder\n                : topicPlaceholderConfig.noTopicsPlaceholder as topicPlaceholder\n            \"\n            label=\"Topic\"\n            data-role=\"topics\"\n            i18n-label=\"Topic label@@messages.messagesCommon.conversationsListFilterConversation.topicLabel\"\n            [options]=\"topics\"\n            displayAttributePath=\"name\"\n            [attr.placeholder]=\"topicPlaceholder\"\n            [placeholder]=\"topicPlaceholder\"\n            formControlName=\"topic\"\n          ></bb-dropdown-single-select-ui>\n        </div>\n        <div class=\"col-4\">\n          <bb-search-box-ui\n            label=\"Customer\"\n            i18n-label=\"\n              Customer label text@@messages.messagesCommon.conversationsListFilterConversation.customerLabelText\"\n            formControlName=\"user\"\n            placeholder=\"Search by name\"\n            i18n-placeholder=\"\n              Search by name\n              placeholder@@messages.messagesCommon.conversationsListFilterConversation.searchByNamePlaceholderText\"\n            [typeaheadOptions]=\"host.typeaheadOptions\"\n          ></bb-search-box-ui>\n        </div>\n        <div class=\"col-4\">\n          <bb-input-datepicker-ui\n            label=\"Date\"\n            i18n-label=\"Date label text@@messages.messagesCommon.conversationsListFilterConversation.dateLabelText\"\n            formControlName=\"date\"\n            [rangeSelection]=\"true\"\n          ></bb-input-datepicker-ui>\n        </div>\n      </div>\n      <div class=\"bb-button-bar\">\n        <button\n          bbButton\n          class=\"bb-button-bar__button\"\n          color=\"secondary\"\n          i18n=\"Close@@messages.messagesCommon.conversationsListFilterConversation.closeButtonText\"\n          (click)=\"host.triggerOpened()\"\n          data-role=\"close-filter\"\n        >\n          Close\n        </button>\n        <button\n          bbButton\n          class=\"bb-button-bar__button\"\n          type=\"submit\"\n          i18n=\"Apply@@messages.messagesCommon.conversationsListFilterConversation.applyButtonText\"\n          data-role=\"apply-filter\"\n        >\n          Apply\n        </button>\n        <button\n          bbButton\n          class=\"bb-button-bar__button\"\n          color=\"link\"\n          i18n=\"Clear All@@messages.messagesCommon.conversationsListFilterConversation.clearAllButtonText\"\n          (click)=\"host.reset()\"\n          data-role=\"clear-filter\"\n        >\n          Clear All\n        </button>\n      </div>\n    </form>\n  </div>\n\n  <ng-template #filtersTextDefault>\n    <span\n      i18n=\"\n        Filters|Text for filter button if filter was\n        applied@@messages.messagesCommon.conversationsListFilterConversation.filtersText\"\n      >Filters</span\n    >\n  </ng-template>\n</ng-template>\n", components: [{ type: i0.forwardRef(function () { return i4.IconComponent; }), selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i0.forwardRef(function () { return i5.DropdownSingleSelectComponent; }), selector: "bb-dropdown-single-select-ui", inputs: ["placeholder", "defaultOptionAsPlaceholder", "options", "displayAttributePath", "iconName", "compareWithFn"] }, { type: i0.forwardRef(function () { return i6.SearchBoxComponent; }), selector: "bb-search-box-ui", inputs: ["placeholder", "maxLength", "squareBorder", "typeaheadOptions", "showSearch", "showClear", "searchLabel", "clearLabel", "autocomplete"], outputs: ["submit", "clear", "valueChange"] }, { type: i0.forwardRef(function () { return i7.InputDatepickerComponent; }), selector: "bb-input-datepicker-ui", inputs: ["rangeSelection", "rangeSelectionSplit", "clickOpen", "focusOpen", "placement", "datepickerDescribedby", "ariaLabelForButton", "minDate", "maxDate", "markDisabled", "icon", "iconSize", "iconColor", "btnColor", "ariaLabel", "navigation", "autocomplete", "firstDayOfWeek", "displayMonths", "placeholder", "overrideDateFormat"], outputs: ["focusedDate"] }], directives: [{ type: i0.forwardRef(function () { return FilterConversationsListCustomizableDirective; }), selector: "[bbFilterConversationsListCustomizable]" }, { type: i0.forwardRef(function () { return i8.NgClass; }), selector: "[ngClass]", inputs: ["class", "ngClass"] }, { type: i0.forwardRef(function () { return i9.ButtonDirective; }), selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }, { type: i0.forwardRef(function () { return i8.NgIf; }), selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i0.forwardRef(function () { return i2.ɵNgNoValidate; }), selector: "form:not([ngNoForm]):not([ngNativeValidate])" }, { type: i0.forwardRef(function () { return i2.NgControlStatusGroup; }), selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { type: i0.forwardRef(function () { return i2.FormGroupDirective; }), selector: "[formGroup]", inputs: ["formGroup"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { type: i0.forwardRef(function () { return i2.NgControlStatus; }), selector: "[formControlName],[ngModel],[formControl]" }, { type: i0.forwardRef(function () { return i2.FormControlName; }), selector: "[formControlName]", inputs: ["disabled", "formControlName", "ngModel"], outputs: ["ngModelChange"] }], pipes: { "async": i0.forwardRef(function () { return i8.AsyncPipe; }) } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: FilterConversationsListComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-messages-filter-conversation-list',
                    templateUrl: './filter-conversations-list.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.FilterService }, { type: i2.FormBuilder }, { type: i3.SharedMethodsService }]; }, propDecorators: { headerClass: [{
                type: Input
            }], bodyClass: [{
                type: Input
            }], filterParams: [{
                type: Input
            }], apply: [{
                type: Output
            }], clear: [{
                type: Output
            }] } });
export class FilterConversationsListCustomizableDirective extends BbTemplate {
}
FilterConversationsListCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: FilterConversationsListCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
FilterConversationsListCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.14", type: FilterConversationsListCustomizableDirective, selector: "[bbFilterConversationsListCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: FilterConversationsListCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbFilterConversationsListCustomizable]',
                }]
        }] });
