// @backbase/contact-common-ang/esm2015/lib/directives/no-white-space-validator/no-white-space.directive.js
import { Directive } from '@angular/core';
import { NG_VALIDATORS } from '@angular/forms';
import * as i0 from "@angular/core";
/**
 * This is a directive to validate white spaces
 *
 * @usageNotes
 *
 * ### validates the white spaces
 *
 * ```html
 *  <input type="text" #nameControl="ngModel" [(ngModel)]="name" bbNoWhiteSpace />
 * ```
 * @ngModule NoWhiteSpaceModule
 */
export class NoWhiteSpaceValidatorDirective {
    validate(control) {
        return noWhiteSpaceValidator(control);
    }
}
/** @nocollapse */ NoWhiteSpaceValidatorDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NoWhiteSpaceValidatorDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
/** @nocollapse */ NoWhiteSpaceValidatorDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: NoWhiteSpaceValidatorDirective, selector: "[bbNoWhiteSpace]", providers: [{ provide: NG_VALIDATORS, useExisting: NoWhiteSpaceValidatorDirective, multi: true }], ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NoWhiteSpaceValidatorDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbNoWhiteSpace]',
                    providers: [{ provide: NG_VALIDATORS, useExisting: NoWhiteSpaceValidatorDirective, multi: true }],
                }]
        }] });
export const noWhiteSpaceValidator = (control) => {
    const isWhitespace = (control.value || '').trim().length === 0;
    return isWhitespace ? { required: true } : null;
};
