// @backbase/transactions-common-ang/esm2015/lib/model/transactions-list-options.model.js
export var OnLoadAction;
(function (OnLoadAction) {
    OnLoadAction[OnLoadAction["Replace"] = 0] = "Replace";
    OnLoadAction[OnLoadAction["Append"] = 1] = "Append";
    OnLoadAction[OnLoadAction["ReplacePage"] = 2] = "ReplacePage";
})(OnLoadAction || (OnLoadAction = {}));
export var TransactionState;
(function (TransactionState) {
    TransactionState["completed"] = "COMPLETED";
    TransactionState["uncompleted"] = "UNCOMPLETED";
})(TransactionState || (TransactionState = {}));
export const defaultAccountOptions = {
    account: {},
};
export const defaultSearchOptions = {
    search: {},
};
export const defaultFilterOptions = {
    filter: {},
};
export const defaultSortOptions = {};
export const defaultPaginationOptions = {
    pagination: {
        from: 0,
        size: 10,
    },
};
export const defaultInitialListOptions = Object.assign(Object.assign(Object.assign(Object.assign(Object.assign(Object.assign({}, defaultAccountOptions), defaultSearchOptions), defaultFilterOptions), defaultSortOptions), defaultPaginationOptions), { onLoad: OnLoadAction.Replace });
export const defaultInitialExportOptions = Object.assign(Object.assign(Object.assign(Object.assign(Object.assign({}, defaultAccountOptions), defaultSearchOptions), defaultFilterOptions), defaultSortOptions), { exportType: 'csv', locale: 'en-US' });
