// @backbase/transactions-common-ang/esm2015/lib/model/transactions-details.model.js
export var TransactionsDetailsLoadingState;
(function (TransactionsDetailsLoadingState) {
    TransactionsDetailsLoadingState[TransactionsDetailsLoadingState["error"] = -1] = "error";
    TransactionsDetailsLoadingState[TransactionsDetailsLoadingState["serverError"] = -2] = "serverError";
    TransactionsDetailsLoadingState[TransactionsDetailsLoadingState["badRequestError"] = -3] = "badRequestError";
    TransactionsDetailsLoadingState[TransactionsDetailsLoadingState["notFoundError"] = -4] = "notFoundError";
    TransactionsDetailsLoadingState[TransactionsDetailsLoadingState["loading"] = 0] = "loading";
    TransactionsDetailsLoadingState[TransactionsDetailsLoadingState["done"] = 1] = "done";
    TransactionsDetailsLoadingState[TransactionsDetailsLoadingState["notLoaded"] = 2] = "notLoaded";
    TransactionsDetailsLoadingState[TransactionsDetailsLoadingState["notStarted"] = 3] = "notStarted";
})(TransactionsDetailsLoadingState || (TransactionsDetailsLoadingState = {}));
export var DisputeStatusEnum;
(function (DisputeStatusEnum) {
    DisputeStatusEnum["Refunded"] = "REFUNDED";
    DisputeStatusEnum["Undisputed"] = "UNDISPUTED";
    DisputeStatusEnum["Pending"] = "PENDING";
    DisputeStatusEnum["Rejected"] = "REJECTED";
})(DisputeStatusEnum || (DisputeStatusEnum = {}));
