// @backbase/transactions-common-ang/esm2015/lib/model/transactions-get-params.model.js
import { TransactionsFilterOptions } from './transactions-filter-options.model';
/* eslint-disable no-redeclare*/
// eslint-disable-next-line @typescript-eslint/no-namespace
export var TransactionsGetParams;
(function (TransactionsGetParams) {
    // eslint-disable-next-line complexity
    TransactionsGetParams.toHttpGetParams = (getParams) => {
        const params = Object.assign({}, TransactionsFilterOptions.toHttpGetParams(getParams.filter));
        if (getParams.account && getParams.account.arrangementId) {
            params['arrangementsIds'] = Array.isArray(getParams.account.arrangementId)
                ? getParams.account.arrangementId
                : [getParams.account.arrangementId];
        }
        if (getParams.search && getParams.search.query) {
            params['query'] = getParams.search.query;
        }
        if (getParams.pagination) {
            params['from'] = getParams.pagination.from;
            params['size'] = getParams.pagination.size;
        }
        if (getParams.sort) {
            params['orderBy'] = getParams.sort.orderBy.toString();
            params['direction'] = getParams.sort.direction.toString();
        }
        if (getParams.exportType) {
            params['exportType'] = getParams.exportType;
        }
        if (getParams.locale) {
            params['locale'] = getParams.locale;
        }
        if (getParams.state) {
            params['state'] = getParams.state;
        }
        return params;
    };
})(TransactionsGetParams || (TransactionsGetParams = {}));
/* eslint-enable no-redeclare*/
