// @backbase/ui-ang/esm2015/_example/action-status/showcase/action-status-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ActionStatusShowcaseComponent } from './action-status-showcase.component';
import { ActionStatusModule } from '@backbase/ui-ang/action-status';
import * as i0 from "@angular/core";
export class ActionStatusShowcaseModule {
}
ActionStatusShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ActionStatusShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ActionStatusShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ActionStatusShowcaseModule, declarations: [ActionStatusShowcaseComponent], imports: [CommonModule, ActionStatusModule] });
ActionStatusShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ActionStatusShowcaseModule, imports: [[CommonModule, ActionStatusModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ActionStatusShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ActionStatusModule],
                    declarations: [ActionStatusShowcaseComponent],
                    entryComponents: [ActionStatusShowcaseComponent],
                }]
        }] });
