// @backbase/ui-ang/esm2015/_example/action-status/showcase/action-status-showcase.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/action-status";
let ActionStatusShowcaseComponent = class ActionStatusShowcaseComponent {
    constructor() {
        this.successExample = 'success';
        this.errorExample = 'error';
        this.warningExample = 'warning';
        this.infoExample = 'info';
    }
    ngOnInit() {
        this.setStatesToLoading();
    }
    setStatesToLoading() {
        setTimeout(() => {
            this.successExample = 'loading';
            this.errorExample = 'loading';
            this.warningExample = 'loading';
            this.infoExample = 'loading';
            this.setStatesToDone();
        }, 2000);
    }
    setStatesToDone() {
        setTimeout(() => {
            this.successExample = 'success';
            this.errorExample = 'error';
            this.warningExample = 'warning';
            this.infoExample = 'info';
            this.setStatesToLoading();
        }, 2000);
    }
};
ActionStatusShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ActionStatusShowcaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ActionStatusShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ActionStatusShowcaseComponent, selector: "bb-action-status-ui-showcase", ngImport: i0, template: "<div class=\"bb-block bb-block--lg\">\n  <h2>Success</h2>\n  <bb-action-status-ui [state]=\"successExample\"></bb-action-status-ui>\n</div>\n<div class=\"bb-block bb-block--lg\">\n  <h2>Error</h2>\n  <bb-action-status-ui [state]=\"errorExample\"></bb-action-status-ui>\n</div>\n<div class=\"bb-block bb-block--lg\">\n  <h2>Warning</h2>\n  <bb-action-status-ui [state]=\"warningExample\"></bb-action-status-ui>\n</div>\n<div class=\"bb-block bb-block--lg\">\n  <h2>Info</h2>\n  <bb-action-status-ui [state]=\"infoExample\"></bb-action-status-ui>\n</div>\n", components: [{ type: i1.ActionStatusComponent, selector: "bb-action-status-ui", inputs: ["state"] }] });
ActionStatusShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
], ActionStatusShowcaseComponent);
export { ActionStatusShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ActionStatusShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-action-status-ui-showcase',
                    templateUrl: './action-status-showcase.component.html',
                }]
        }] });
