// @backbase/product-summary-manage-accounts-widget-ang/esm2015/src/product-summary-manage-accounts-widget/product-summary-manage-accounts-widget.component.js
import { Component, Directive, ViewChild, Input, } from '@angular/core';
import { BbTemplate } from '@backbase/foundation-ang/core';
import { BehaviorSubject, combineLatest, Observable, Subject } from 'rxjs';
import { PaginationType, SMALL_SCREEN_SIZE, processErrorMessages, UserPreferencesValidationSettings, hasExternalInfo, } from '@backbase/product-summary-common-ang';
import { map, takeUntil } from 'rxjs/operators';
import { WidgetProperty } from '../model/constants';
import { ManageAccountsWidgetPropertiesService } from '../manage-accounts-widget-properties.service';
import * as i0 from "@angular/core";
import * as i1 from "../manage.service";
import * as i2 from "../manage-accounts-widget-properties.service";
import * as i3 from "@backbase/product-summary-common-ang";
import * as i4 from "../manage-account-item/manage-account-item.component";
import * as i5 from "@backbase/ui-ang/empty-state";
import * as i6 from "@backbase/ui-ang/loading-indicator";
import * as i7 from "@backbase/product-summary-ui";
import * as i8 from "@angular/common";
/**
 * Widget for displaying and managing accounts
 *
 * The following extension slots are available in this component:
 *
 * - `ProductSummaryManageAccountCustomizableDirective`
 * - `ProductSummaryManageAccountLoadingCustomizableDirective`
 * - `ManageAccountsFooterCustomizableDirective`
 *
 * @see `AccountsOverviewWidgetCustomizableDirective`
 * @see `ProductSummaryManageAccountLoadingCustomizableDirective`
 * @see `ManageAccountsFooterCustomizableDirective`
 *
 * @usageNotes
 *
 * ### Include this widget in your `AppModule`:
 *
 * ```typescript
 * @NgModule({
 *   imports: [
 *     ProductSummaryManageAccountsWidgetModule,
 *   ],
 *   ...
 * })
 * export class AppModule {}
 * ```
 *
 * ### Include this widget (standalone model)
 *
 * ```typescript
 * export const environment: Environment = {
 *   ...
 *   bootstrap: {
 *     pageModel: {
 *       name: 'root',
 *       properties: {},
 *       children: [
 *         {
 *           name: 'product-summary-accounts-overview-widget-ang',
 *           properties: {
 *              classId: 'ProductSummaryManageAccountsWidgetComponent',
 *              businessFunction: 'Product Summary',
 *              resourceName: 'Product Summary',
 *              privilege: 'view',
 *              size: 20,
 *              route: 'manage-accounts',
 *           },
 *         },
 *       ],
 *     },
 *   },
 * };
 *
 * @ngModule ProductSummaryManageAccountsWidgetModule
 * ```
 */
export class ProductSummaryManageAccountsWidgetComponent {
    /**
     * Constructor
     *
     * @param service
     * @param widgetPropertiesService
     * @param productsExternalInfoService
     * @param el
     */
    constructor(service, widgetPropertiesService, productsExternalInfoService, el) {
        this.service = service;
        this.widgetPropertiesService = widgetPropertiesService;
        this.productsExternalInfoService = productsExternalInfoService;
        this.el = el;
        /**
         * Subject for the garbage collector
         */
        this.gc$ = new Subject();
        /**
         * @internal
         */
        this.hostRef = this;
        /**
         * ViewChild for `notificationError`
         */
        this.notificationErrorTemplate = '';
        /**
         * ViewChild for `notificationSuccess`
         */
        this.notificationSuccessTemplate = '';
        /**
         * Contains errors occurred during fetching accounts
         */
        this.error = this.service.error;
        /**
         * If errors occurred during patching arrangements record,
         * holds the errors coming from the server
         */
        this.updateError = this.service.updateError;
        /**
         * Available product summary items.
         */
        this.accounts = new Observable(subscriber => {
            const subscriptions = [];
            subscriptions.push(this.service.accountList.subscribe((accounts) => {
                const products = (accounts === null || accounts === void 0 ? void 0 : accounts.items) || [];
                subscriber.next(accounts);
                if (products.some(hasExternalInfo)) {
                    subscriptions.push(this.productsExternalInfoService
                        .getProductsWithExternalInfo(products)
                        .pipe(map(productsWithExternalInfo => (Object.assign(Object.assign({}, accounts), { items: productsWithExternalInfo }))))
                        .subscribe(subscriber));
                }
            }, subscriber.error));
            return () => subscriptions.forEach(subscription => subscription.unsubscribe());
        });
        /**
         * Whether the widget is currently fetching data from the server
         */
        this.loading = this.service.loading;
        /**
         * Maximum length of product item aliases
         */
        this.aliasMaxLength = UserPreferencesValidationSettings.ALIAS_MAX_LENGTH;
        this.isSmallScreen = false;
        /**
         * Default business function
         */
        this.businessFunction = 'Product Summary';
        /**
         * Default resource name
         */
        this.resourceName = 'Product Summary';
        /**
         * "Default privilege type
         */
        this.privilege = 'view';
        /**
         * Number of accounts per page
         */
        this.size = 20;
        /**
         * Defines whether product kind icons should be displayed
         *
         * @default true
         */
        this.showIcons = true;
        /**
         * Default pagination parameters
         */
        this.paginationParams = new BehaviorSubject({
            from: 0,
            size: 0,
            businessFunction: '',
            resourceName: '',
            privilege: '',
            paginationType: PaginationType.PAGINATION,
        });
        this.requestParams = combineLatest(this.paginationParams, this.widgetPropertiesService.widgetProperties).pipe(takeUntil(this.gc$), map(([paginationParams, widgetProperties]) => ({
            from: paginationParams.from || 0,
            size: widgetProperties.size,
            businessFunction: widgetProperties.businessFunction,
            resourceName: widgetProperties.resourceName,
            privilege: widgetProperties.privilege,
            paginationType: paginationParams.paginationType,
        })));
        /**
         * Joins the specified error messages to display on the template
         *
         * @param messages
         */
        this.processErrorMessages = processErrorMessages;
    }
    /**
     * @internal
     */
    ngAfterViewInit() {
        this.widgetPropertiesService.setConfig({
            businessFunction: this.businessFunction,
            resourceName: this.resourceName,
            privilege: this.privilege,
            size: this.size,
        });
        const containerWidth = this.el.nativeElement.children[0].getBoundingClientRect().width;
        this.isSmallScreen = containerWidth <= SMALL_SCREEN_SIZE;
        this.widgetPropertiesService.widgetProperties
            .pipe(takeUntil(this.gc$))
            .subscribe(widgetProperties => this.initWidgetProperties(widgetProperties));
    }
    /**
     * Cleanups just before Angular destroys the component.
     *
     * @returns
     */
    ngOnDestroy() {
        this.gc$.next();
        this.gc$.complete();
    }
    initWidgetProperties(properties) {
        const paginationType = this.isSmallScreen ? PaginationType.LOAD_MORE : PaginationType.PAGINATION;
        this.paginationParams.next(Object.assign(Object.assign({}, this.paginationParams.getValue()), { size: properties.size, paginationType }));
        this.service.getAccountsFrom(this.requestParams);
    }
    /**
     * Toggles visibility of product summary item.
     *
     * @param item
     */
    toggleVisibility(item) {
        this.service.toggleVisibility(item, this.notificationErrorTemplate, this.notificationSuccessTemplate);
    }
    /**
     * Updates alias of product summary item, by id.
     *
     * @param id
     * @param alias
     */
    updateAlias(id, alias = '') {
        this.service.updateAlias(id, alias, this.notificationErrorTemplate, this.notificationSuccessTemplate);
    }
    /**
     * Event handler for the `pageChange` event of `bb-product-summary-footer-ui`
     * emits the new `page` value to the `paginationParams`
     *
     * @param page
     */
    onPageChange(page) {
        this.paginationParams.next(Object.assign(Object.assign({}, this.paginationParams.getValue()), { from: page }));
    }
    /**
     * Event handler for the `loadMore` event of `bb-product-summary-footer-ui`
     * updates the `from` property of `paginationParams`
     */
    onLoadMore() {
        if (!this.loading.getValue()) {
            this.paginationParams.next(Object.assign(Object.assign({}, this.paginationParams.value), { from: this.paginationParams.value.from ? this.paginationParams.value.from + 1 : 1 }));
        }
    }
}
ProductSummaryManageAccountsWidgetComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryManageAccountsWidgetComponent, deps: [{ token: i1.ManageAccountsService }, { token: i2.ManageAccountsWidgetPropertiesService }, { token: i3.ProductsExternalInfoService }, { token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Component });
ProductSummaryManageAccountsWidgetComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ProductSummaryManageAccountsWidgetComponent, selector: "bb-product-summary-manage-accounts-widget", inputs: { businessFunction: "businessFunction", resourceName: "resourceName", privilege: "privilege", size: "size", showIcons: "showIcons" }, providers: [ManageAccountsWidgetPropertiesService], viewQueries: [{ propertyName: "notificationErrorTemplate", first: true, predicate: ["notificationError"], descendants: true, static: true }, { propertyName: "notificationSuccessTemplate", first: true, predicate: ["notificationSuccess"], descendants: true, static: true }], ngImport: i0, template: "<div class=\"bb-block bb-block--full-width\">\n  <ng-container *ngIf=\"!(hostRef.error | async | bbIsDefined) && (accounts | async) as accounts; else loadingTemplate\">\n    <ng-container *ngIf=\"((hostRef.loading | async) === false); else loadingTemplate\">\n      <ng-container bbManageAccountCustomizable\n                    [bbHostRef]=\"hostRef\"\n                    [bbTemplateContext]=\"accounts\">\n      </ng-container>\n    </ng-container>\n  </ng-container>\n</div>\n\n<!-- Manage Accounts template -->\n<ng-template\n  bbManageAccountCustomizable\n  let-hostRef\n  let-accounts=\"context\">\n  <div class=\"bb-block bb-block--xl\"\n       data-role=\"product-summary-manage-accounts-initial-message\"\n       i18n=\"Manage Products info|Manage Products info header label@@product.summary.manage.accounts.header.info\">\n    Below you can define which products will be shown in \"My accounts\" and/or edit their account names.\n  </div>\n\n  <div class=\"card card-lg\">\n    <div class=\"card-body\">\n      <div class=\"bb-list bb-list--density-md\">\n        <div class=\"bb-list__item\"\n             *ngFor=\"let item of accounts.items\">\n          <bb-manage-account-item\n            [item]=\"item\"\n            [productAliasMaxLength]=\"hostRef.aliasMaxLength\"\n            [showIcon]=\"this.showIcons\"\n            (toggleVisibility)=\"hostRef.toggleVisibility($event)\"\n            (updateAlias)=\"hostRef.updateAlias(item.id, $event)\"\n          ></bb-manage-account-item>\n        </div>\n      </div>\n    </div>\n    <div class=\"card-footer\">\n      <ng-container\n        bbManageAccountsFooterCustomizable\n        [bbHostRef]=\"hostRef\"\n        [bbTemplateContext]=\"{\n            paginationParams: hostRef.paginationParams | async,\n            accounts: accounts\n          }\"\n      >\n      </ng-container>\n    </div>\n  </div>\n</ng-template>\n\n<ng-template #notificationError>\n  <span i18n=\"Failed to update Product|Error notification message@@product-summary-manage.notification.error\">\n    Failed to update Product\n  </span>\n  <span *ngIf=\"updateError | async as updateError\">\n    {{processErrorMessages(updateError.messages)}}\n  </span>\n</ng-template>\n\n<ng-template #notificationSuccess>\n  <span i18n=\"Successfully Updated|Success notification message@@product-summary-manage.notification.success\">\n    Successfully Updated\n  </span>\n</ng-template>\n\n\n<!-- Product summary loading template -->\n<ng-template #loadingTemplate>\n  <ng-container bbManageAccountLoadingCustomizable\n                [bbHostRef]=\"hostRef\">\n  </ng-container>\n</ng-template>\n\n<ng-template\n  bbManageAccountLoadingCustomizable\n  let-hostRef>\n  <ng-container *ngIf=\"hostRef.error | async as error; else loading\">\n    <bb-empty-state-ui\n      iconSize=\"xxl\"\n      iconModifier=\"error-outline\"\n      iconClasses=\"text-muted\"\n      title=\"\"\n      [subtitle]=\"hostRef.processErrorMessages(error.messages)\"\n    ></bb-empty-state-ui>\n  </ng-container>\n  <ng-template #loading>\n    <bb-loading-indicator-ui\n      text=\"Loading Accounts...\"\n      i18n-text=\"Loading Accounts...|Text for loading indicator@@product.details.loadingAccounts\"\n    ></bb-loading-indicator-ui>\n  </ng-template>\n</ng-template>\n\n<ng-template\n  bbManageAccountsFooterCustomizable\n  let-hostRef\n  let-accounts=\"context.accounts\"\n  let-paginationParams=\"context.paginationParams\">\n  <bb-product-summary-footer-ui\n    [page]=\"paginationParams?.from\"\n    [count]=\"accounts?.count\"\n    [hasMore]=\"accounts?.items?.length !== accounts?.count\"\n    [loading]=\"hostRef.loading | async\"\n    [paginationType]=\"paginationParams?.paginationType\"\n    [itemsPerPage]=\"paginationParams?.size\"\n    (pageChange)=\"hostRef.onPageChange($event)\"\n    (loadMore)=\"hostRef.onLoadMore($event)\"\n  ></bb-product-summary-footer-ui>\n</ng-template>\n", components: [{ type: i0.forwardRef(function () { return i4.ManageAccountItemComponent; }), selector: "bb-manage-account-item", inputs: ["item", "productAliasMaxLength", "showIcon"], outputs: ["toggleVisibility", "updateAlias"] }, { type: i0.forwardRef(function () { return i5.EmptyStateComponent; }), selector: "bb-empty-state-ui", inputs: ["title", "subtitle", "showIcon", "iconClasses", "iconModifier", "iconSize", "iconColor"] }, { type: i0.forwardRef(function () { return i6.LoadingIndicatorComponent; }), selector: "bb-loading-indicator-ui", inputs: ["text", "loaderSize", "showDelay", "hasBackground", "inline"] }, { type: i0.forwardRef(function () { return i7.ProductSummaryFooterComponent; }), selector: "bb-product-summary-footer-ui", inputs: ["page", "count", "itemsPerPage", "paginationType", "hasMore", "loading", "boundaryLinks", "directionLinks", "maxNavPages"], outputs: ["pageChange", "loadMore"] }], directives: [{ type: i0.forwardRef(function () { return i8.NgIf; }), selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i0.forwardRef(function () { return ProductSummaryManageAccountCustomizableDirective; }), selector: "[bbManageAccountCustomizable]" }, { type: i0.forwardRef(function () { return i8.NgForOf; }), selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { type: i0.forwardRef(function () { return ManageAccountsFooterCustomizableDirective; }), selector: "[bbManageAccountsFooterCustomizable]" }, { type: i0.forwardRef(function () { return ProductSummaryManageAccountLoadingCustomizableDirective; }), selector: "[bbManageAccountLoadingCustomizable]" }], pipes: { "bbIsDefined": i0.forwardRef(function () { return i3.IsDefinedPipe; }), "async": i0.forwardRef(function () { return i8.AsyncPipe; }) } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryManageAccountsWidgetComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-product-summary-manage-accounts-widget',
                    templateUrl: 'product-summary-manage-accounts-widget.component.html',
                    providers: [ManageAccountsWidgetPropertiesService],
                }]
        }], ctorParameters: function () { return [{ type: i1.ManageAccountsService }, { type: i2.ManageAccountsWidgetPropertiesService }, { type: i3.ProductsExternalInfoService }, { type: i0.ElementRef }]; }, propDecorators: { notificationErrorTemplate: [{
                type: ViewChild,
                args: ['notificationError', { static: true }]
            }], notificationSuccessTemplate: [{
                type: ViewChild,
                args: ['notificationSuccess', { static: true }]
            }], businessFunction: [{
                type: Input,
                args: [WidgetProperty.BUSINESS_FUNCTION]
            }], resourceName: [{
                type: Input,
                args: [WidgetProperty.RESOURCE_NAME]
            }], privilege: [{
                type: Input,
                args: [WidgetProperty.PRIVILEGE_TYPE]
            }], size: [{
                type: Input,
                args: [WidgetProperty.SIZE]
            }], showIcons: [{
                type: Input,
                args: [WidgetProperty.SHOW_ICONS]
            }] } });
/**
 * This is an extension slot to be used for customizing the entire widget
 *
 * @usageNotes
 *
 * ### Customize the widget
 *
 * ```html
 * <ng-template
 *   bbManageAccountCustomizable
 *   let-hostRef
 *   let-accounts="context">
 *   Your custom template here
 * </ng-template>
 * ```
 *
 * @NgModule ProductSummaryManageAccountsWidgetModule
 */
export class ProductSummaryManageAccountCustomizableDirective extends BbTemplate {
}
ProductSummaryManageAccountCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryManageAccountCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
ProductSummaryManageAccountCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: ProductSummaryManageAccountCustomizableDirective, selector: "[bbManageAccountCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryManageAccountCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbManageAccountCustomizable]',
                }]
        }] });
/**
 * This is an extension slot to be used for customizing the loading indicator
 *
 * @usageNotes
 *
 * ### Customize the loading indicator
 *
 * ```html
 * <ng-template
 *   bbManageAccountLoadingCustomizable
 *   let-hostRef>
 *   Your custom template here
 * </ng-template>
 * ```
 *
 * @NgModule ProductSummaryManageAccountsWidgetModule
 */
export class ProductSummaryManageAccountLoadingCustomizableDirective extends BbTemplate {
}
ProductSummaryManageAccountLoadingCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryManageAccountLoadingCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
ProductSummaryManageAccountLoadingCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: ProductSummaryManageAccountLoadingCustomizableDirective, selector: "[bbManageAccountLoadingCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductSummaryManageAccountLoadingCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbManageAccountLoadingCustomizable]',
                }]
        }] });
/**
 * This is an extension slot to be used for customizing the footer
 *
 * @usageNotes
 *
 * ### Customize the footer
 *
 * ```html
 * <ng-template
 *   bbManageAccountsFooterCustomizable
 *   let-hostRef
 *   let-accounts="context.accounts"
 *   let-paginationParams="context.paginationParams">
 *   Your custom template here
 * </ng-template>
 * ```
 *
 * @NgModule ProductSummaryManageAccountsWidgetModule
 */
export class ManageAccountsFooterCustomizableDirective extends BbTemplate {
}
ManageAccountsFooterCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManageAccountsFooterCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
ManageAccountsFooterCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: ManageAccountsFooterCustomizableDirective, selector: "[bbManageAccountsFooterCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManageAccountsFooterCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbManageAccountsFooterCustomizable]',
                }]
        }] });
