// @backbase/accessgroup-ui/esm2015/lib/select-items-modal/select-items-modal.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { ButtonModule } from '@backbase/ui-ang/button';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { InfiniteScrollModule } from '@backbase/ui-ang/infinite-scroll';
import { InputCheckboxModule } from '@backbase/ui-ang/input-checkbox';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { ModalModule } from '@backbase/ui-ang/modal';
import { SearchBoxModule } from '@backbase/ui-ang/search-box';
import { SelectItemsEmptyDirective, SelectItemsErrorDirective, SelectItemsItemDirective, SelectItemsItemHeaderDirective, SelectItemsLoadingDirective, SelectItemsModalComponent, SelectItemsSearchDirective, } from './select-items-modal.component';
import * as i0 from "@angular/core";
const uiModules = [
    ModalModule,
    SearchBoxModule,
    ButtonModule,
    InputCheckboxModule,
    InfiniteScrollModule,
    LoadingIndicatorModule,
    EmptyStateModule,
];
const uiComponents = [
    SelectItemsModalComponent,
    SelectItemsItemHeaderDirective,
    SelectItemsItemDirective,
    SelectItemsLoadingDirective,
    SelectItemsSearchDirective,
    SelectItemsErrorDirective,
    SelectItemsEmptyDirective,
];
export class SelectItemsModalModule {
}
/** @nocollapse */ SelectItemsModalModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SelectItemsModalModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ SelectItemsModalModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SelectItemsModalModule, declarations: [SelectItemsModalComponent,
        SelectItemsItemHeaderDirective,
        SelectItemsItemDirective,
        SelectItemsLoadingDirective,
        SelectItemsSearchDirective,
        SelectItemsErrorDirective,
        SelectItemsEmptyDirective], imports: [CommonModule, FormsModule, ModalModule,
        SearchBoxModule,
        ButtonModule,
        InputCheckboxModule,
        InfiniteScrollModule,
        LoadingIndicatorModule,
        EmptyStateModule], exports: [ModalModule,
        SearchBoxModule,
        ButtonModule,
        InputCheckboxModule,
        InfiniteScrollModule,
        LoadingIndicatorModule,
        EmptyStateModule, SelectItemsModalComponent,
        SelectItemsItemHeaderDirective,
        SelectItemsItemDirective,
        SelectItemsLoadingDirective,
        SelectItemsSearchDirective,
        SelectItemsErrorDirective,
        SelectItemsEmptyDirective] });
/** @nocollapse */ SelectItemsModalModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SelectItemsModalModule, imports: [[CommonModule, FormsModule, ...uiModules], ModalModule,
        SearchBoxModule,
        ButtonModule,
        InputCheckboxModule,
        InfiniteScrollModule,
        LoadingIndicatorModule,
        EmptyStateModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: SelectItemsModalModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...uiComponents],
                    imports: [CommonModule, FormsModule, ...uiModules],
                    exports: [...uiModules, ...uiComponents],
                }]
        }] });
