// @backbase/ui-ang/esm2015/header/header.component.js
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
/**
 * @name HeaderComponent
 *
 * @description
 * Component that displays a header.
 */
export class HeaderComponent {
}
HeaderComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HeaderComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
HeaderComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: HeaderComponent, selector: "bb-header-ui", inputs: { headingClasses: "headingClasses", headingType: "headingType", heading: "heading" }, ngImport: i0, template: "<ng-container [ngSwitch]=\"headingType\">\n  <h1 *ngSwitchCase=\"'h1'\" [ngClass]=\"headingClasses\" data-role=\"headings\">\n    {{ heading }}\n  </h1>\n  <h2 *ngSwitchCase=\"'h2'\" [ngClass]=\"headingClasses\" data-role=\"headings\">\n    {{ heading }}\n  </h2>\n  <h3 *ngSwitchCase=\"'h3'\" [ngClass]=\"headingClasses\" data-role=\"headings\">\n    {{ heading }}\n  </h3>\n  <h4 *ngSwitchCase=\"'h4'\" [ngClass]=\"headingClasses\" data-role=\"headings\">\n    {{ heading }}\n  </h4>\n  <h5 *ngSwitchCase=\"'h5'\" [ngClass]=\"headingClasses\" data-role=\"headings\">\n    {{ heading }}\n  </h5>\n  <h6 *ngSwitchCase=\"'h6'\" [ngClass]=\"headingClasses\" data-role=\"headings\">\n    {{ heading }}\n  </h6>\n</ng-container>\n", directives: [{ type: i1.NgSwitch, selector: "[ngSwitch]", inputs: ["ngSwitch"] }, { type: i1.NgSwitchCase, selector: "[ngSwitchCase]", inputs: ["ngSwitchCase"] }, { type: i1.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HeaderComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-header-ui',
                    templateUrl: './header.component.html',
                    styles: [],
                }]
        }], propDecorators: { headingClasses: [{
                type: Input
            }], headingType: [{
                type: Input
            }], heading: [{
                type: Input
            }] } });
