// @backbase/messages-ui-ang/esm2015/lib/messages-heading/messages-heading.component.js
import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/header";
import * as i2 from "@angular/common";
export class MessagesHeadingComponent {
    constructor() {
        /**
         * Heading title
         */
        this.headingTitle = '';
        /**
         * Heading type
         */
        this.headingType = 'h1';
        /**
         * Custom heading classes
         */
        this.headingClasses = '';
        /**
         * Custom heading wrapper classes
         */
        this.headingWrapperClasses = '';
        /**
         * Heading button text
         */
        this.buttonText = '';
        /**
         * Flag indicates heading button visibility
         */
        this.showButton = true;
        /**
         * Heading heading button press action
         */
        this.buttonAction = new EventEmitter();
    }
    onButtonClick() {
        this.buttonAction.emit();
    }
}
/** @nocollapse */ MessagesHeadingComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesHeadingComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ MessagesHeadingComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: MessagesHeadingComponent, selector: "bb-messages-heading", inputs: { headingTitle: "headingTitle", headingType: "headingType", headingClasses: "headingClasses", headingWrapperClasses: "headingWrapperClasses", buttonText: "buttonText", showButton: "showButton" }, outputs: { buttonAction: "buttonAction" }, ngImport: i0, template: "<div [ngClass]=\"'bb-heading-widget ' + headingWrapperClasses\" data-role=\"messages-heading\">\n  <bb-header-ui\n    [headingType]=\"headingType\"\n    headingClasses=\"bb-heading-widget__heading\"\n    [heading]=\"headingTitle\"\n  ></bb-header-ui>\n  <button\n    *ngIf=\"showButton\"\n    data-role=\"messages-heading-button\"\n    (click)=\"onButtonClick()\"\n    class=\"bb-heading-widget__button btn btn-primary\"\n  >\n    <span>{{ buttonText }}</span>\n  </button>\n</div>\n", components: [{ type: i1.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }], directives: [{ type: i2.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesHeadingComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-messages-heading',
                    templateUrl: './messages-heading.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { headingTitle: [{
                type: Input
            }], headingType: [{
                type: Input
            }], headingClasses: [{
                type: Input
            }], headingWrapperClasses: [{
                type: Input
            }], buttonText: [{
                type: Input
            }], showButton: [{
                type: Input
            }], buttonAction: [{
                type: Output
            }] } });
