// @backbase/messages-ui-ang/esm2015/lib/messages-heading/messages-heading.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { ButtonModule } from '@backbase/ui-ang/button';
import { HeaderModule } from '@backbase/ui-ang/header';
import { MessagesHeadingComponent } from './messages-heading.component';
import * as i0 from "@angular/core";
export class MessagesHeadingModule {
}
/** @nocollapse */ MessagesHeadingModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesHeadingModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ MessagesHeadingModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesHeadingModule, declarations: [MessagesHeadingComponent], imports: [CommonModule, RouterModule, HeaderModule, ButtonModule], exports: [MessagesHeadingComponent] });
/** @nocollapse */ MessagesHeadingModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesHeadingModule, imports: [[CommonModule, RouterModule, HeaderModule, ButtonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesHeadingModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [MessagesHeadingComponent],
                    imports: [CommonModule, RouterModule, HeaderModule, ButtonModule],
                    exports: [MessagesHeadingComponent],
                }]
        }] });
