// @backbase/manage-pockets-journey-ang/esm2015/src/manage-pockets-common/components/manage-pockets-base-step/manage-pockets-base-step.js
import { Directive, EventEmitter, Input, Output } from '@angular/core';
import { POCKET_LOCALIZATIONS } from '../../localization/manage-pockets-journey.localizations';
import { TransferTypeEnum } from '../../model/pocket-item.model';
import * as i0 from "@angular/core";
// eslint-disable-next-line
export class ManagePocketsBaseStep {
    constructor() {
        this.back = new EventEmitter();
        this.cancel = new EventEmitter();
        this.pocketLocalizations = POCKET_LOCALIZATIONS;
        this.transferTypesEnum = TransferTypeEnum;
    }
    goBack() {
        this.back.emit();
    }
    abort() {
        this.cancel.emit();
    }
    isFieldValid(form, fieldName) {
        const controlField = form === null || form === void 0 ? void 0 : form.get(fieldName || '');
        return !!(controlField === null || controlField === void 0 ? void 0 : controlField.valid) || !!(controlField === null || controlField === void 0 ? void 0 : controlField.untouched);
    }
    getFieldErrors(form, field) {
        var _a;
        return (_a = form === null || form === void 0 ? void 0 : form.get(field || '')) === null || _a === void 0 ? void 0 : _a.errors;
    }
}
ManagePocketsBaseStep.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsBaseStep, deps: [], target: i0.ɵɵFactoryTarget.Directive });
ManagePocketsBaseStep.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: ManagePocketsBaseStep, inputs: { journeyCurrency: "journeyCurrency", partialNewPayment: "partialNewPayment" }, outputs: { back: "back", cancel: "cancel" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsBaseStep, decorators: [{
            type: Directive
        }], propDecorators: { back: [{
                type: Output
            }], cancel: [{
                type: Output
            }], journeyCurrency: [{
                type: Input
            }], partialNewPayment: [{
                type: Input
            }] } });
