// @backbase/card-details-widget-ang/esm2015/src/card-details-widget.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { CardsCommonModule } from '@backbase/cards-common-ang';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { AlertModule } from '@backbase/ui-ang/alert';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { PaymentCardModule } from '@backbase/ui-ang/payment-card';
import { SwitchModule } from '@backbase/ui-ang/switch';
import { CardDetailsPaymentCardControlsCustomizableDirective, PaymentCardControlsComponent, } from './components/bb-payment-card-controls/payment-card-controls.component';
import { CardDetailsPaymentCardSummaryCustomizableDirective, PaymentCardSummaryComponent, } from './components/bb-payment-card-summary/payment-card-summary.component';
import { CardDetailsPaymentCardCustomizableDirective, PaymentCardComponent, } from './components/bb-payment-card/payment-card.component';
import { CardDetailsContainerComponent } from './components/card-details-container/card-details-container.component';
import { CardDetailsWidgetComponent } from './components/card-details-widget/card-details-widget.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const modules = [
    CardsCommonModule,
    SwitchModule,
    LoadingIndicatorModule,
    EmptyStateModule,
    PaymentCardModule,
    AlertModule,
];
const components = [
    CardDetailsWidgetComponent,
    CardDetailsContainerComponent,
    PaymentCardComponent,
    PaymentCardSummaryComponent,
    PaymentCardControlsComponent,
];
const customizables = [
    CardDetailsPaymentCardCustomizableDirective,
    CardDetailsPaymentCardControlsCustomizableDirective,
    CardDetailsPaymentCardSummaryCustomizableDirective,
];
export class CardDetailsWidgetModule {
}
CardDetailsWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardDetailsWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CardDetailsWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardDetailsWidgetModule, declarations: [CardDetailsWidgetComponent,
        CardDetailsContainerComponent,
        PaymentCardComponent,
        PaymentCardSummaryComponent,
        PaymentCardControlsComponent, CardDetailsPaymentCardCustomizableDirective,
        CardDetailsPaymentCardControlsCustomizableDirective,
        CardDetailsPaymentCardSummaryCustomizableDirective], imports: [CommonModule,
        FormsModule, i1.BackbaseCoreModule, CardsCommonModule,
        SwitchModule,
        LoadingIndicatorModule,
        EmptyStateModule,
        PaymentCardModule,
        AlertModule], exports: [CardsCommonModule,
        SwitchModule,
        LoadingIndicatorModule,
        EmptyStateModule,
        PaymentCardModule,
        AlertModule, CardDetailsWidgetComponent,
        CardDetailsContainerComponent,
        PaymentCardComponent,
        PaymentCardSummaryComponent,
        PaymentCardControlsComponent, CardDetailsPaymentCardCustomizableDirective,
        CardDetailsPaymentCardControlsCustomizableDirective,
        CardDetailsPaymentCardSummaryCustomizableDirective] });
CardDetailsWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardDetailsWidgetModule, imports: [[
            CommonModule,
            FormsModule,
            BackbaseCoreModule.withConfig({
                classMap: { CardDetailsWidgetComponent },
            }),
            ...modules,
        ], CardsCommonModule,
        SwitchModule,
        LoadingIndicatorModule,
        EmptyStateModule,
        PaymentCardModule,
        AlertModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardDetailsWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        FormsModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { CardDetailsWidgetComponent },
                        }),
                        ...modules,
                    ],
                    declarations: [...components, ...customizables],
                    exports: [...modules, ...components, ...customizables],
                }]
        }] });
