// @backbase/billpay-http-ang/esm2015/api/billPaySearch.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { BILLPAY_BASE_PATH } from '../variables';
import { BillpayConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class BillPaySearchService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new BillpayConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    getBillPaySearch(requestParameters, observe = 'body', reportProgress = false, options) {
        const _name = requestParameters["name"];
        if (_name === null || _name === undefined) {
            throw new Error('Required parameter name was null or undefined when calling getBillPaySearch.');
        }
        const _zipCode = requestParameters["zipCode"];
        const _accountNumber = requestParameters["accountNumber"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_name !== undefined && _name !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _name, 'name');
        }
        if (_zipCode !== undefined && _zipCode !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _zipCode, 'zipCode');
        }
        if (_accountNumber !== undefined && _accountNumber !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _accountNumber, 'accountNumber');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/bill-pay/payees-search`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getBillPaySearchUrl(requestParameters) {
        const _name = requestParameters["name"];
        if (_name === null || _name === undefined) {
            throw new Error('Required parameter name was null or undefined when calling getBillPaySearch.');
        }
        const _zipCode = requestParameters["zipCode"];
        const _accountNumber = requestParameters["accountNumber"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_name !== undefined && _name !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _name, 'name');
        }
        if (_zipCode !== undefined && _zipCode !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _zipCode, 'zipCode');
        }
        if (_accountNumber !== undefined && _accountNumber !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _accountNumber, 'accountNumber');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/bill-pay/payees-search${queryString ? `?${queryString}` : ''}`;
    }
}
BillPaySearchService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillPaySearchService, deps: [{ token: i1.HttpClient }, { token: BILLPAY_BASE_PATH, optional: true }, { token: i2.BillpayConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
BillPaySearchService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillPaySearchService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillPaySearchService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [BILLPAY_BASE_PATH]
                }] }, { type: i2.BillpayConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
