// @backbase/billpay-http-ang/esm2015/api/billPayPayveris.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { BILLPAY_BASE_PATH } from '../variables';
import { BillpayConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class BillPayPayverisService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new BillpayConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    deleteBillPayPayverisEbillsEnrolment(requestParameters, observe = 'body', reportProgress = false, options) {
        const _payeeID = requestParameters["payeeID"];
        if (_payeeID === null || _payeeID === undefined) {
            throw new Error('Required parameter payeeID was null or undefined when calling deleteBillPayPayverisEbillsEnrolment.');
        }
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_payeeID !== undefined && _payeeID !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _payeeID, 'payeeID');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.delete(`${this.configuration.basePath}/client-api/v2/bill-pay/payveris/ebills/enrolment`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    deleteBillPayPayverisEbillsEnrolmentUrl(requestParameters) {
        const _payeeID = requestParameters["payeeID"];
        if (_payeeID === null || _payeeID === undefined) {
            throw new Error('Required parameter payeeID was null or undefined when calling deleteBillPayPayverisEbillsEnrolment.');
        }
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_payeeID !== undefined && _payeeID !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _payeeID, 'payeeID');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/bill-pay/payveris/ebills/enrolment${queryString ? `?${queryString}` : ''}`;
    }
    postBillPayPayverisEbillsEnrolment(requestParameters, observe = 'body', reportProgress = false, options) {
        const _billPayPayverisEbillsEnrolmentPostRequestBody = requestParameters["billPayPayverisEbillsEnrolmentPostRequestBody"];
        if (_billPayPayverisEbillsEnrolmentPostRequestBody === null || _billPayPayverisEbillsEnrolmentPostRequestBody === undefined) {
            throw new Error('Required parameter billPayPayverisEbillsEnrolmentPostRequestBody was null or undefined when calling postBillPayPayverisEbillsEnrolment.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v2/bill-pay/payveris/ebills/enrolment`, _billPayPayverisEbillsEnrolmentPostRequestBody, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    postBillPayPayverisEbillsEnrolmentUrl(requestParameters) {
        const _billPayPayverisEbillsEnrolmentPostRequestBody = requestParameters["billPayPayverisEbillsEnrolmentPostRequestBody"];
        if (_billPayPayverisEbillsEnrolmentPostRequestBody === null || _billPayPayverisEbillsEnrolmentPostRequestBody === undefined) {
            throw new Error('Required parameter billPayPayverisEbillsEnrolmentPostRequestBody was null or undefined when calling postBillPayPayverisEbillsEnrolment.');
        }
        return `${this.configuration.basePath}/client-api/v2/bill-pay/payveris/ebills/enrolment`;
    }
    postBillPayPayverisResolveEbillErrors(requestParameters, observe = 'body', reportProgress = false, options) {
        const _billPayPayverisResolveEbillErrorsPostRequestBody = requestParameters["billPayPayverisResolveEbillErrorsPostRequestBody"];
        if (_billPayPayverisResolveEbillErrorsPostRequestBody === null || _billPayPayverisResolveEbillErrorsPostRequestBody === undefined) {
            throw new Error('Required parameter billPayPayverisResolveEbillErrorsPostRequestBody was null or undefined when calling postBillPayPayverisResolveEbillErrors.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v2/bill-pay/payveris/ebills/resolve-errors`, _billPayPayverisResolveEbillErrorsPostRequestBody, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    postBillPayPayverisResolveEbillErrorsUrl(requestParameters) {
        const _billPayPayverisResolveEbillErrorsPostRequestBody = requestParameters["billPayPayverisResolveEbillErrorsPostRequestBody"];
        if (_billPayPayverisResolveEbillErrorsPostRequestBody === null || _billPayPayverisResolveEbillErrorsPostRequestBody === undefined) {
            throw new Error('Required parameter billPayPayverisResolveEbillErrorsPostRequestBody was null or undefined when calling postBillPayPayverisResolveEbillErrors.');
        }
        return `${this.configuration.basePath}/client-api/v2/bill-pay/payveris/ebills/resolve-errors`;
    }
}
BillPayPayverisService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillPayPayverisService, deps: [{ token: i1.HttpClient }, { token: BILLPAY_BASE_PATH, optional: true }, { token: i2.BillpayConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
BillPayPayverisService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillPayPayverisService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillPayPayverisService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [BILLPAY_BASE_PATH]
                }] }, { type: i2.BillpayConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
