// @backbase/connect-external-accounts-journey-ang/esm2015/lib/services/config.service.js
import { Inject, Injectable, InjectionToken, Optional } from '@angular/core';
import * as i0 from "@angular/core";
export const ConnectExternalAccountsJourneyConfigurationToken = new InjectionToken('ConnectExternalAccountsJourneyConfigurationToken injection token');
const configDefaults = {
    currencies: ['USD'],
    dismissMessageTime: 3000,
};
export class ConnectExternalAccountsJourneyConfigService {
    constructor(config) {
        this.config = config;
        this._config = Object.assign(Object.assign({}, configDefaults), (this.config || {}));
    }
    get defaults() {
        return configDefaults;
    }
    get currencies() {
        return this._config.currencies;
    }
    get dismissMessageTime() {
        return this._config.dismissMessageTime;
    }
}
/** @nocollapse */ ConnectExternalAccountsJourneyConfigService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ConnectExternalAccountsJourneyConfigService, deps: [{ token: ConnectExternalAccountsJourneyConfigurationToken, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
/** @nocollapse */ ConnectExternalAccountsJourneyConfigService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ConnectExternalAccountsJourneyConfigService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ConnectExternalAccountsJourneyConfigService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [ConnectExternalAccountsJourneyConfigurationToken]
                }] }]; } });
