// @backbase/ui-ang/esm2015/checkbox-group/checkbox-group.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { InputCheckboxModule } from '@backbase/ui-ang/input-checkbox';
import { FormsModule } from '@angular/forms';
import { BaseClassesModule } from '@backbase/ui-ang/base-classes';
import { CheckboxGroupComponent } from './checkbox-group.component';
import * as i0 from "@angular/core";
export class CheckboxGroupModule {
}
CheckboxGroupModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CheckboxGroupModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CheckboxGroupModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CheckboxGroupModule, declarations: [CheckboxGroupComponent], imports: [CommonModule, InputCheckboxModule, FormsModule, BaseClassesModule], exports: [CheckboxGroupComponent] });
CheckboxGroupModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CheckboxGroupModule, imports: [[CommonModule, InputCheckboxModule, FormsModule, BaseClassesModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CheckboxGroupModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, InputCheckboxModule, FormsModule, BaseClassesModule],
                    declarations: [CheckboxGroupComponent],
                    exports: [CheckboxGroupComponent],
                }]
        }] });
