// @backbase/accessgroup-common-ang/esm2015/lib/services/user-permissions.service.js
import { Injectable } from '@angular/core';
import { UsersHttpService } from '@backbase/data-ang/accesscontrol';
import { map, shareReplay } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/accesscontrol";
export class UserPermissionsService {
    constructor(http) {
        this.http = http;
        /**
         * Current user permissions
         */
        this.userPermissions$ = this.getUserPermissions().pipe(shareReplay(1, 1000));
        /**
         * Current user Limits permissions
         */
        this.limitsPermissions$ = this.getPermissions('Limits', 'Manage Limits');
        /**
         * Payee group permissions
         */
        this.payeeGroupPermissions$ = this.getPermissions('Entitlements', 'Manage Data Groups');
        /**
         * Current user Shadow Limits permissions
         */
        this.shadowLimitsPermissions$ = this.getPermissions('Limits', 'Manage Shadow Limits');
        /**
         * Wizard permissions
         */
        this.entitlementsWizardPermissions$ = this.getPermissions('Entitlements', 'Quick User Setup');
    }
    /**
     * Return permissions based on provided resource and function names
     * @param resource: string - resource name
     * @param functionName: string - function name
     */
    getPermissions(resource, functionName) {
        return this.userPermissions$.pipe(
        // @ts-ignore
        map((data) => {
            // NOTE: Search key hardcoded by the purpose
            const limitsPerm = data.find(perm => perm.resource === resource && perm.function === functionName);
            return limitsPerm || undefined;
        }), map((perm) => (perm && perm.permissions) || {}), shareReplay(1));
    }
    /**
     * Return PermissionsSummary request stream
     */
    getUserPermissions() {
        return this.http.getUserPermissionsSummary();
    }
}
/** @nocollapse */ UserPermissionsService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: UserPermissionsService, deps: [{ token: i1.UsersHttpService }], target: i0.ɵɵFactoryTarget.Injectable });
/** @nocollapse */ UserPermissionsService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: UserPermissionsService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: UserPermissionsService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.UsersHttpService }]; } });
