// @backbase/arrangement-manager-beta-http-ang/esm2015/model/orderByField.js
export const OrderByField = {
    Name: 'name',
    BankAlias: 'bankAlias',
    BookedBalance: 'bookedBalance',
    AvailableBalance: 'availableBalance',
    CreditLimit: 'creditLimit',
    DisplayName: 'displayName',
    IBAN: 'IBAN',
    BBAN: 'BBAN',
    BIC: 'BIC',
    Currency: 'currency',
    ExternalTransferAllowed: 'externalTransferAllowed',
    UrgentTransferAllowed: 'urgentTransferAllowed',
    AccruedInterest: 'accruedInterest',
    PrincipalAmount: 'principalAmount',
    CurrentInvestmentValue: 'currentInvestmentValue',
    ProductNumber: 'productNumber',
    AccountOpeningDate: 'accountOpeningDate',
    AccountInterestRate: 'accountInterestRate',
    ValueDateBalance: 'valueDateBalance',
    CreditLimitUsage: 'creditLimitUsage',
    CreditLimitInterestRate: 'creditLimitInterestRate',
    CreditLimitExpiryDate: 'creditLimitExpiryDate',
    StartDate: 'startDate',
    TermUnit: 'termUnit',
    TermNumber: 'termNumber',
    MaturityDate: 'maturityDate',
    MaturityAmount: 'maturityAmount',
    AutoRenewalIndicator: 'autoRenewalIndicator',
    InterestPaymentFrequencyUnit: 'interestPaymentFrequencyUnit',
    InterestPaymentFrequencyNumber: 'interestPaymentFrequencyNumber',
    InterestSettlementAccount: 'interestSettlementAccount',
    OutstandingPrincipalAmount: 'outstandingPrincipalAmount',
    MonthlyInstalmentAmount: 'monthlyInstalmentAmount',
    AmountInArrear: 'amountInArrear',
    MinimumRequiredBalance: 'minimumRequiredBalance',
    CreditCardAccountNumber: 'creditCardAccountNumber',
    ValidThru: 'validThru',
    ApplicableInterestRate: 'applicableInterestRate',
    RemainingCredit: 'remainingCredit',
    OutstandingPayment: 'outstandingPayment',
    MinimumPayment: 'minimumPayment',
    MinimumPaymentDueDate: 'minimumPaymentDueDate',
    TotalInvestmentValue: 'totalInvestmentValue',
    AccountHolderAddressLine1: 'accountHolderAddressLine1',
    AccountHolderAddressLine2: 'accountHolderAddressLine2',
    AccountHolderStreetName: 'accountHolderStreetName',
    PostCode: 'postCode',
    Town: 'town',
    CountrySubDivision: 'countrySubDivision',
    AccountHolderCountry: 'accountHolderCountry',
    CreditAccount: 'creditAccount',
    DebitAccount: 'debitAccount',
    ProductKindName: 'productKindName',
    ProductTypeName: 'productTypeName',
    SourceId: 'sourceId'
};
