// @backbase/messages-common/esm2015/src/components/create-message/employee-form/services/create-message-form-implementation.service.js
import { Injectable } from '@angular/core';
import { of } from 'rxjs';
import { catchError } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/messages-v5-http-ang";
export class CreateMessageFormImplementationService {
    constructor(employeeService) {
        this.employeeService = employeeService;
        this.maxPaginationItemsNum = 65535;
    }
    /**
     * Get topics
     *
     * @returns Employee topics stream
     */
    getTopics() {
        return this.employeeService
            .getTopicsEmployee({ size: this.maxPaginationItemsNum, subscribed: true })
            .pipe(catchError(() => of([])));
    }
    /**
     * Create message
     *
     * @param params Create message request params
     * @returns Create message stream
     */
    createMessage(params) {
        return this.employeeService.postMessagesEmployee({ presentationMessageEmployeePostRequestBody: params });
    }
}
CreateMessageFormImplementationService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: CreateMessageFormImplementationService, deps: [{ token: i1.EmployeeHttpService }], target: i0.ɵɵFactoryTarget.Injectable });
CreateMessageFormImplementationService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: CreateMessageFormImplementationService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: CreateMessageFormImplementationService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.EmployeeHttpService }]; } });
