// @backbase/product-summary-ui/esm2015/src/bb-product-item-custom-title-ui/bb-product-item-custom-title-ui.component.js
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { Subject } from 'rxjs';
import { distinctUntilChanged } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "../bb-product-summary-account-logo-ui/bb-product-summary-account-logo.component";
import * as i2 from "@backbase/ui-ang/ellipsis";
import * as i3 from "@backbase/ui-ang/icon";
import * as i4 from "../bb-arrangement-state-indicator-ui/arrangement-state-indicator.component";
import * as i5 from "../bb-product-number/product-number.component";
import * as i6 from "@angular/common";
import * as i7 from "@backbase/ui-ang/keyboard-click-directive";
import * as i8 from "@backbase/ui-ang/tooltip-directive";
/**
 * Displays product item title
 *
 * @usageNotes
 *
 * ```html
 * <bb-product-item-custom-title-ui
 *   [title]="'title'"
 *   [productNumber]="'12345678'"
 *   [productNumberType]="'bban'"
 *   [favorite]="true"
 * ></bb-product-item-custom-title-ui>
 * ```
 *
 * @ngModule ProductItemCustomModule
 */
export class ProductItemCustomTitleComponent {
    constructor() {
        /**
         * The card Title value. Default is empty string
         *
         * @default ''
         */
        this.title = '';
        /**
         * Specifies if title can be focused and selected
         *
         * @default false
         */
        this.titleInteractive = false;
        /**
         * Specifies if logo should be displayed
         *
         * @default false
         */
        this.showLogo = true;
        /**
         * Source for bank logo.
         * Can be either a URL or base64 image.
         *
         * @default ''
         */
        this.logoSrc = '';
        /**
         * Product kind name to display a related logo icon.
         *
         * @default ''
         */
        this.logoKind = '';
        /**
         * Specifies if external title should be displayed.
         *
         * @default false
         */
        this.showExternalTitle = true;
        /**
         * The favorite state of card (add favorite icon next to title)
         *
         * @default false
         */
        this.favorite = false;
        /**
         * The favorite icon color (define star icon color)
         *
         * @default 'primary'
         */
        this.favoriteIconColor = 'primary';
        /**
         * The product Number. Default is empty string
         *
         * @default ''
         */
        this.productNumber = '';
        /**
         * State name to display
         * Supported values: 'active', 'inactive', 'closed'
         * If the state name is not supported it will be shown as it is
         *
         * @default ''
         */
        this.state = '';
        /**
         * selectTitle emits when custom title is selected
         */
        this.selectCustomTitle = new EventEmitter();
        this.uniqueSelectCustomTitleEvent = new Subject();
        this.subscriptions = [];
    }
    /**
     * @internal
     */
    ngOnInit() {
        const uniqueSelectCustomTitleSubscription = this.uniqueSelectCustomTitleEvent
            .pipe(distinctUntilChanged())
            .subscribe(event => this.selectCustomTitle.emit(event));
        this.subscriptions.push(uniqueSelectCustomTitleSubscription);
    }
    /**
     * @internal
     */
    ngOnDestroy() {
        this.subscriptions.forEach(subscription => subscription.unsubscribe());
        this.uniqueSelectCustomTitleEvent.complete();
    }
    /**
     * emits "selectCustomTitle" event
     */
    onSelectCustomTitle(event) {
        this.uniqueSelectCustomTitleEvent.next(event);
    }
    /**
     * Returns financial institution name
     */
    getExternalTitle() {
        var _a;
        if (this.showExternalTitle) {
            return ((_a = this.externalTitle) !== null && _a !== void 0 ? _a : $localize `:@@product-summary-ui.product-item-custom-title.external-title:External account`);
        }
        return undefined;
    }
}
ProductItemCustomTitleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCustomTitleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ProductItemCustomTitleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ProductItemCustomTitleComponent, selector: "bb-product-item-custom-title-ui", inputs: { title: "title", titleInteractive: "titleInteractive", showLogo: "showLogo", logoSrc: "logoSrc", logoKind: "logoKind", showExternalTitle: "showExternalTitle", externalTitle: "externalTitle", favorite: "favorite", favoriteIconColor: "favoriteIconColor", productNumber: "productNumber", productNumberType: "productNumberType", state: "state" }, outputs: { selectCustomTitle: "selectCustomTitle" }, ngImport: i0, template: "<div class=\"bb-account-info\" (click)=\"onSelectCustomTitle($event)\">\n  <ng-container *ngIf=\"showLogo && (logoSrc || logoKind); else titleWithoutLogo\">\n    <div class=\"bb-stack bb-stack--align-top bb-stack--density-sm bb-block bb-block--xs\">\n      <bb-product-summary-account-logo\n        *ngIf=\"logoSrc || logoKind\"\n        class=\"bb-stack__item\"\n        [logoSrc]=\"logoSrc\"\n        [productKindName]=\"logoKind\"\n        iconColor=\"secondary\"\n        iconSize=\"sm\"\n        logoSize=\"sm\"\n      ></bb-product-summary-account-logo>\n      <div *ngIf=\"showExternalTitle; else accountTitle\" class=\"bb-stack__item\">\n        <ng-container *ngTemplateOutlet=\"accountExternalTitle\"></ng-container>\n      </div>\n    </div>\n    <ng-container *ngIf=\"showExternalTitle\">\n      <ng-container *ngTemplateOutlet=\"accountTitle\"></ng-container>\n    </ng-container>\n  </ng-container>\n\n  <ng-template #titleWithoutLogo>\n    <ng-container *ngTemplateOutlet=\"accountExternalTitle\"></ng-container>\n    <ng-container *ngTemplateOutlet=\"accountTitle\"></ng-container>\n  </ng-template>\n\n  <ng-template #accountExternalTitle>\n    <div *ngIf=\"showExternalTitle\" class=\"bb-subtitle bb-text-semi-bold bb-text-support bb-block bb-block--xs\">\n      <bb-ellipsis-ui [text]=\"getExternalTitle()\" lineClamp=\"1\"></bb-ellipsis-ui>\n    </div>\n  </ng-template>\n\n  <ng-template #accountTitle>\n    <div class=\"bb-stack bb-account-info__title bb-block bb-block--xs\" *ngIf=\"title || favorite || state\">\n      <div class=\"bb-stack__item bb-stack__item--spacing-sm\" data-role=\"card-title\">\n        <bb-ellipsis-ui\n          *ngIf=\"titleInteractive; else staticTitle\"\n          [text]=\"title\"\n          role=\"link\"\n          tabindex=\"0\"\n          lineClamp=\"1\"\n          (bbKeyboardClick)=\"onSelectCustomTitle($event)\"\n        ></bb-ellipsis-ui>\n        <ng-template #staticTitle>\n          <bb-ellipsis-ui [text]=\"title\" lineClamp=\"1\"></bb-ellipsis-ui>\n        </ng-template>\n      </div>\n      <bb-icon-ui\n        *ngIf=\"favorite\"\n        class=\"bb-stack__item bb-account-info__icon\"\n        size=\"sm\"\n        [color]=\"favoriteIconColor\"\n        data-role=\"favorite-icon\"\n        name=\"star\"\n        tabindex=\"0\"\n        aria-label=\"Favorite account\"\n        i18n-aria-label=\"Favorite account|Icon aria-label favorite account\n                @@product-summary.product-item-title-custom.icon.favorite.aria-label\"\n        bbTooltip=\"Favorite account\"\n        triggers=\"hover focus\"\n        i18n-bbTooltip=\"Favorite account|Icon tooltip favorite account\n                @@product-summary.product-item-title-custom.icon.favorite.tooltip\"\n      ></bb-icon-ui>\n      <bb-arrangement-state-indicator\n        *ngIf=\"state\"\n        [state]=\"state\"\n        class=\"bb-stack__item\"\n        data-role=\"arrangementState\"\n      ></bb-arrangement-state-indicator>\n    </div>\n  </ng-template>\n\n  <div\n      (mousedown)=\"$event.stopPropagation()\"\n      *ngIf=\"productNumber\"\n      data-role=\"card-sub-title\"\n      class=\"bb-account-info__product-number\"\n    >\n    <bb-ellipsis-ui>\n      <bb-product-number\n        [number]=\"productNumber\"\n        [type]=\"productNumberType\"\n      ></bb-product-number>\n    </bb-ellipsis-ui>\n  </div>\n</div>\n\n", components: [{ type: i1.ProductSummaryAccountLogoComponent, selector: "bb-product-summary-account-logo", inputs: ["logoSrc", "productKindName", "defaultProductKindName", "iconColor", "iconSize", "logoSize"] }, { type: i2.EllipsisComponent, selector: "bb-ellipsis-ui", inputs: ["text", "tooltipTriggers", "lineClamp"] }, { type: i3.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i4.ArrangementStateIndicatorComponent, selector: "bb-arrangement-state-indicator", inputs: ["state", "showActive", "showIcon", "hideStateTextForMobile"] }, { type: i5.ProductNumberComponent, selector: "bb-product-number", inputs: ["number", "product", "format", "type", "unmaskable", "masked"], outputs: ["mask", "unmask"] }], directives: [{ type: i6.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i6.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }, { type: i7.KeyboardClickDirective, selector: "[bbKeyboardClick]", outputs: ["bbKeyboardClick"] }, { type: i8.TooltipDirective, selector: "[bbTooltip]", inputs: ["bbTooltip", "triggers"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCustomTitleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-product-item-custom-title-ui',
                    templateUrl: './bb-product-item-custom-title-ui.component.html',
                }]
        }], propDecorators: { title: [{
                type: Input
            }], titleInteractive: [{
                type: Input
            }], showLogo: [{
                type: Input
            }], logoSrc: [{
                type: Input
            }], logoKind: [{
                type: Input
            }], showExternalTitle: [{
                type: Input
            }], externalTitle: [{
                type: Input
            }], favorite: [{
                type: Input
            }], favoriteIconColor: [{
                type: Input
            }], productNumber: [{
                type: Input
            }], productNumberType: [{
                type: Input
            }], state: [{
                type: Input
            }], selectCustomTitle: [{
                type: Output
            }] } });
