// @backbase/product-summary-ui/esm2015/src/bb-product-item-custom-title-ui/product-item-custom.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { IconModule } from '@backbase/ui-ang/icon';
import { PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import { EllipsisModule } from '@backbase/ui-ang/ellipsis';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { KeyboardClickModule } from '@backbase/ui-ang/keyboard-click-directive';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { ProductItemCustomTitleComponent } from './bb-product-item-custom-title-ui.component';
import { ArrangementStateIndicatorModule } from '../bb-arrangement-state-indicator-ui/arrangement-state-indicator.module';
import { ProductNumberModule } from '../bb-product-number/product-number.module';
import { ProductSummaryAccountLogoModule } from '../bb-product-summary-account-logo-ui/bb-product-summary-account-logo.module';
import * as i0 from "@angular/core";
const uiModules = [
    IconModule,
    PaymentCardNumberModule,
    ArrangementStateIndicatorModule,
    EllipsisModule,
    TooltipModule,
    ProductNumberModule,
    KeyboardClickModule,
    ProductSummaryAccountLogoModule,
];
const components = [ProductItemCustomTitleComponent];
export class ProductItemCustomModule {
}
ProductItemCustomModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCustomModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductItemCustomModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCustomModule, declarations: [ProductItemCustomTitleComponent], imports: [CommonModule, BackbaseCoreModule, IconModule,
        PaymentCardNumberModule,
        ArrangementStateIndicatorModule,
        EllipsisModule,
        TooltipModule,
        ProductNumberModule,
        KeyboardClickModule,
        ProductSummaryAccountLogoModule], exports: [ProductItemCustomTitleComponent, IconModule,
        PaymentCardNumberModule,
        ArrangementStateIndicatorModule,
        EllipsisModule,
        TooltipModule,
        ProductNumberModule,
        KeyboardClickModule,
        ProductSummaryAccountLogoModule] });
ProductItemCustomModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCustomModule, imports: [[CommonModule, BackbaseCoreModule, ...uiModules], IconModule,
        PaymentCardNumberModule,
        ArrangementStateIndicatorModule,
        EllipsisModule,
        TooltipModule,
        ProductNumberModule,
        KeyboardClickModule,
        ProductSummaryAccountLogoModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCustomModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, BackbaseCoreModule, ...uiModules],
                    declarations: [...components],
                    exports: [...components, ...uiModules],
                }]
        }] });
