// @backbase/pay-bills-journey-ang/esm2015/src/components/form-components/billpay-payment-frequency/billpay-payment-frequency.component.js
import { Component, ChangeDetectionStrategy, Input } from '@angular/core';
import { BillpayUtilityService as Utility, formControlOptions, FrequencyKeys } from '@backbase/billpay-journeys-common';
import { Subject } from 'rxjs';
import { takeUntil } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/dropdown-single-select";
import * as i2 from "@angular/forms";
import * as i3 from "@angular/common";
export class BillpayPaymentFrequencyComponent {
    constructor() {
        this.destroy$ = new Subject();
        this.id = Utility.generateId();
    }
    ngOnInit() {
        this.processFrequencyValueChanges();
    }
    ngOnDestroy() {
        this.destroy$.next();
        this.destroy$.complete();
    }
    processFrequencyValueChanges() {
        if (!this.scheduleForm)
            return;
        const endTypeControl = this.scheduleForm.controls['endType'];
        const endDateControl = this.scheduleForm.controls['endDate'];
        const repeatControl = this.scheduleForm.controls['repeat'];
        const startDateControl = this.scheduleForm.controls['startDate'];
        const paymentServiceControl = this.scheduleForm.controls['paymentService'];
        const frequencyControl = this.scheduleForm.controls['frequency'];
        const paymentServiceDateControl = this.scheduleForm.controls['paymentServiceDate'];
        const sendDateControl = this.scheduleForm.controls['sendDate'];
        if (startDateControl && frequencyControl && paymentServiceControl && endTypeControl && endDateControl) {
            frequencyControl.valueChanges.pipe(takeUntil(this.destroy$)).subscribe(value => {
                if (value === FrequencyKeys.Once) {
                    paymentServiceControl.enable(formControlOptions);
                    sendDateControl.enable(formControlOptions);
                    paymentServiceControl.updateValueAndValidity();
                    paymentServiceDateControl.enable(formControlOptions);
                    endDateControl.disable(formControlOptions);
                    endTypeControl.disable(formControlOptions);
                    repeatControl.disable(formControlOptions);
                }
                else {
                    paymentServiceControl.disable(formControlOptions);
                    paymentServiceDateControl.disable(formControlOptions);
                    sendDateControl.disable(formControlOptions);
                    startDateControl.enable(formControlOptions);
                    endDateControl.enable(formControlOptions);
                    repeatControl.enable(formControlOptions);
                    endTypeControl.enable();
                }
            });
        }
    }
}
BillpayPaymentFrequencyComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayPaymentFrequencyComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
BillpayPaymentFrequencyComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: BillpayPaymentFrequencyComponent, selector: "bb-billpay-payment-frequency", inputs: { scheduleForm: "scheduleForm", frequencies: "frequencies" }, ngImport: i0, template: "<div class=\"bb-form-field bb-form-field--md bb-block bb-block--md\">\n  <label\n    data-role=\"billpay-payment-frequency-label\"\n    class=\"form-label\"\n    i18n=\"@@billpay.billpay-payment-schedule.label.frequency\"\n    [id]=\"'billpay-payment-frequency-label' + id\"\n  >\n    Frequency\n  </label>\n  <bb-dropdown-single-select-ui\n    [formControl]=\"scheduleForm?.get('frequency')\"\n    [aria-labelledby]=\"'billpay-payment-frequency-label' + id\"\n    data-role=\"billpay-payment-frequency\"\n  >\n    <bb-dropdown-single-select-option-ui\n      *ngFor=\"let frequency of frequencies\"\n      [value]=\"frequency.value\"\n      [label]=\"frequency.name\"\n    >\n    </bb-dropdown-single-select-option-ui>\n  </bb-dropdown-single-select-ui>\n</div>\n", components: [{ type: i1.DropdownSingleSelectComponent, selector: "bb-dropdown-single-select-ui", inputs: ["placeholder", "defaultOptionAsPlaceholder", "options", "displayAttributePath", "iconName", "compareWithFn"] }, { type: i1.DropdownSingleSelectOptionComponent, selector: "bb-dropdown-single-select-option-ui", inputs: ["value", "label"] }], directives: [{ type: i2.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { type: i2.FormControlDirective, selector: "[formControl]", inputs: ["disabled", "formControl", "ngModel"], outputs: ["ngModelChange"], exportAs: ["ngForm"] }, { type: i3.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayPaymentFrequencyComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-billpay-payment-frequency',
                    templateUrl: './billpay-payment-frequency.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { scheduleForm: [{
                type: Input
            }], frequencies: [{
                type: Input
            }] } });
