// @backbase/campaign-space-ang/esm2015/src/campaign-space.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { CampaignSpaceComponent } from './campaign-space.component';
import { WebSdkApiModule } from '@backbase/foundation-ang/web-sdk';
import { HttpClientModule } from '@angular/common/http';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
export class CampaignSpaceModule {
}
CampaignSpaceModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CampaignSpaceModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CampaignSpaceModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CampaignSpaceModule, declarations: [CampaignSpaceComponent], imports: [CommonModule,
        WebSdkApiModule,
        HttpClientModule, i1.BackbaseCoreModule], exports: [CampaignSpaceComponent] });
CampaignSpaceModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CampaignSpaceModule, imports: [[
            CommonModule,
            WebSdkApiModule,
            HttpClientModule,
            BackbaseCoreModule.withConfig({
                classMap: { CampaignSpaceComponent: CampaignSpaceComponent },
            }),
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CampaignSpaceModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        WebSdkApiModule,
                        HttpClientModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { CampaignSpaceComponent: CampaignSpaceComponent },
                        }),
                    ],
                    exports: [CampaignSpaceComponent],
                    declarations: [CampaignSpaceComponent],
                }]
        }] });
