// @backbase/payments-collection-ui-ang/esm2015/lib/containers/initiate-payment/components/create-payment-form/create-payment-form.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CreatePaymentFormComponent } from './create-payment-form.component';
import { PayordFormBuilderAngModule } from '@backbase/payord-form-builder-ang';
import * as i0 from "@angular/core";
const components = [CreatePaymentFormComponent];
export class CreatePaymentFormModule {
}
/** @nocollapse */ CreatePaymentFormModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CreatePaymentFormModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ CreatePaymentFormModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CreatePaymentFormModule, declarations: [CreatePaymentFormComponent], imports: [CommonModule, PayordFormBuilderAngModule], exports: [CreatePaymentFormComponent] });
/** @nocollapse */ CreatePaymentFormModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CreatePaymentFormModule, imports: [[CommonModule, PayordFormBuilderAngModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CreatePaymentFormModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, PayordFormBuilderAngModule],
                    declarations: [...components],
                    exports: [...components],
                }]
        }] });
