// @backbase/ui-ang/esm2015/_example/product-item-savings-account/showcase/product-item-savings-account-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ProductItemSavingsAccountShowcaseComponent } from './product-item-savings-account-showcase.component';
import { ProductItemSavingsAccountModule } from '@backbase/ui-ang/product-item-savings-account';
import * as i0 from "@angular/core";
export class ProductItemSavingsAccountShowcaseModule {
}
ProductItemSavingsAccountShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemSavingsAccountShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductItemSavingsAccountShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemSavingsAccountShowcaseModule, declarations: [ProductItemSavingsAccountShowcaseComponent], imports: [CommonModule, ProductItemSavingsAccountModule] });
ProductItemSavingsAccountShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemSavingsAccountShowcaseModule, imports: [[CommonModule, ProductItemSavingsAccountModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemSavingsAccountShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ProductItemSavingsAccountModule],
                    declarations: [ProductItemSavingsAccountShowcaseComponent],
                    entryComponents: [ProductItemSavingsAccountShowcaseComponent],
                }]
        }] });
