// @backbase/user-common/esm2015/src/components/manage-profile/manage-postal-addresses/manage-postal-address.component.js
import { Component, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
export class ManagePostalAddressComponent {
    constructor() {
        this.updateProfile = new EventEmitter();
        this.managePostalAddress = new EventEmitter();
        this.cancelManagePostalAddress = new EventEmitter();
        this.managingPostalAddress = false;
    }
    handleManagePostalAddress() {
        if (this.disableManage) {
            return;
        }
        this.managingPostalAddress = true;
        this.managePostalAddress.emit();
    }
    handleCloseEditPostalAddressAfterSave() {
        this.updateProfile.emit();
    }
    handleCloseEditPostalAddress() {
        this.managingPostalAddress = false;
        this.disableManage = false;
        this.cancelManagePostalAddress.emit();
    }
}
ManagePostalAddressComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePostalAddressComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ManagePostalAddressComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: ManagePostalAddressComponent, selector: "ng-component", inputs: { postalAddress: "postalAddress", postalAddressCount: "postalAddressCount", postalAddressTypes: "postalAddressTypes", disableManage: "disableManage", dismissTimeout: "dismissTimeout" }, outputs: { updateProfile: "updateProfile", managePostalAddress: "managePostalAddress", cancelManagePostalAddress: "cancelManagePostalAddress" }, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePostalAddressComponent, decorators: [{
            type: Component,
            args: [{
                    template: '',
                }]
        }], propDecorators: { postalAddress: [{
                type: Input
            }], postalAddressCount: [{
                type: Input
            }], postalAddressTypes: [{
                type: Input
            }], disableManage: [{
                type: Input
            }], dismissTimeout: [{
                type: Input
            }], updateProfile: [{
                type: Output
            }], managePostalAddress: [{
                type: Output
            }], cancelManagePostalAddress: [{
                type: Output
            }] } });
