// @backbase/user-common/esm2015/src/components/manage-profile/manage-postal-addresses/manage-postal-addresses.component.js
import { Component, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
export class ManagePostalAddressesComponent {
    constructor() {
        this.updateProfile = new EventEmitter();
        this.managePostalAddress = new EventEmitter();
        this.cancelManagePostalAddress = new EventEmitter();
        this.isManagingPostalAddressIndex = undefined;
        this.isAddingPostalAddress = false;
        this.isAddButtonAvailable = true;
    }
    ngOnInit() {
        if (this.postalAddresses && this.maxPostalAddresses) {
            this.isAddButtonAvailable = this.postalAddresses.length < this.maxPostalAddresses;
        }
    }
    handleManagePostalAddressOpen(index) {
        this.isManagingPostalAddressIndex = index;
        this.managePostalAddress.emit();
    }
    handleAddNewPostalAddress() {
        if (this.disableManage) {
            return;
        }
        this.isAddingPostalAddress = true;
        this.managePostalAddress.emit();
    }
    handleManagePostalAddressClosed() {
        this.isManagingPostalAddressIndex = undefined;
        this.isAddingPostalAddress = false;
        this.cancelManagePostalAddress.emit();
    }
    updateUserProfileData() {
        this.updateProfile.emit();
    }
}
ManagePostalAddressesComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePostalAddressesComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ManagePostalAddressesComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: ManagePostalAddressesComponent, selector: "ng-component", inputs: { postalAddresses: "postalAddresses", postalAddressTypes: "postalAddressTypes", maxPostalAddresses: "maxPostalAddresses", disableManage: "disableManage", dismissTimeout: "dismissTimeout" }, outputs: { updateProfile: "updateProfile", managePostalAddress: "managePostalAddress", cancelManagePostalAddress: "cancelManagePostalAddress" }, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManagePostalAddressesComponent, decorators: [{
            type: Component,
            args: [{
                    template: '',
                }]
        }], propDecorators: { postalAddresses: [{
                type: Input
            }], postalAddressTypes: [{
                type: Input
            }], maxPostalAddresses: [{
                type: Input
            }], disableManage: [{
                type: Input
            }], dismissTimeout: [{
                type: Input
            }], updateProfile: [{
                type: Output
            }], managePostalAddress: [{
                type: Output
            }], cancelManagePostalAddress: [{
                type: Output
            }] } });
