// @backbase/user-common/esm2015/src/components/manage-profile/manage-postal-addresses/new-postal-address.component.js
import { Component, Input } from '@angular/core';
import { NewAddressComponent } from '../new-address/new-address.component';
import * as i0 from "@angular/core";
export class NewPostalAddressComponent extends NewAddressComponent {
    ngOnInit() {
        this.newPostalAddressData = {
            type: this.postalAddressTypes && this.postalAddressTypes.length ? this.postalAddressTypes[0] : '',
        };
    }
}
NewPostalAddressComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: NewPostalAddressComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
NewPostalAddressComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: NewPostalAddressComponent, selector: "ng-component", inputs: { postalAddressTypes: "postalAddressTypes" }, usesInheritance: true, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: NewPostalAddressComponent, decorators: [{
            type: Component,
            args: [{
                    template: '',
                }]
        }], propDecorators: { postalAddressTypes: [{
                type: Input
            }] } });
