// @backbase/manage-upcoming-and-historical-payments-journey-ang/esm2015/lib/services/navigation.service.js
import { Injectable } from '@angular/core';
import { ActivatedRoute, Router } from '@angular/router';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
export class ManageUpcomingAndHistoricalPaymentsNavigationService {
    constructor(router, route) {
        this.router = router;
        this.route = route;
    }
    /**
     * Navigation to history payments view tab
     */
    navigateToHistory(extras) {
        this.router.navigate(['history'], Object.assign({ relativeTo: this.route }, (extras || {})));
    }
    /**
     * Navigation to upcoming payments view tab
     */
    navigateToUpcoming(extras) {
        this.router.navigate(['upcoming'], Object.assign({ relativeTo: this.route }, (extras || {})));
    }
    /**
     * Method to return correct tab index in order to activate it
     */
    mapRouteToTabIndex() {
        const routeSections = this.router.url.split('/');
        return routeSections[routeSections.length - 1] === 'history' ? 1 : 0;
    }
}
/** @nocollapse */ ManageUpcomingAndHistoricalPaymentsNavigationService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManageUpcomingAndHistoricalPaymentsNavigationService, deps: [{ token: i1.Router }, { token: i1.ActivatedRoute }], target: i0.ɵɵFactoryTarget.Injectable });
/** @nocollapse */ ManageUpcomingAndHistoricalPaymentsNavigationService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManageUpcomingAndHistoricalPaymentsNavigationService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManageUpcomingAndHistoricalPaymentsNavigationService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.Router }, { type: i1.ActivatedRoute }]; } });
