// @backbase/ui-ang/esm2015/_example/product-item-term-deposit/examples/product-item-term-deposit.example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/product-item-term-deposit";
export class ProductItemTermDepositExampleComponent {
    constructor() {
        this.product = {
            id: 'b387bfa5-75fd-410e-bb86-7f764f8ef69c',
            name: 'Ricardo Norton`s Account',
            currency: 'EUR',
            amount: '83652.32',
            productNumber: 'NL75INGB3400120056',
        };
        this.interest = {
            id: 'd2d85314-e60c-44af-ba98-8f704dad5970',
            name: 'Accrued Interest',
            currency: 'EUR',
            amount: '880.54',
            productNumber: 'NL75INGB3400120056',
        };
        this.productDetails = {
            owner: 'Richard Norton',
        };
        this.productNumberFormat = {
            length: 18,
            maskRange: [0, 12],
            segments: 4,
        };
    }
}
ProductItemTermDepositExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemTermDepositExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ProductItemTermDepositExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ProductItemTermDepositExampleComponent, selector: "bb-product-item-term-deposit-example-ui", ngImport: i0, template: "<bb-product-item-term-deposit-ui\n  [title]=\"product.name\"\n  [amount]=\"product.amount\"\n  [productNumber]=\"product.productNumber\"\n  [currency]=\"product.currency\"\n  [productNumberFormat]=\"productNumberFormat\"\n  [mainSectionCurrency]=\"interest.currency\"\n  [mainSectionAmount]=\"interest.amount\"\n  [mainSectionTitle]=\"interest.name\"\n  [productDetailsInfo]=\"productDetails\"\n  [highlight]=\"false\"\n  [favorite]=\"true\"\n></bb-product-item-term-deposit-ui>\n", components: [{ type: i1.ProductItemTermDepositComponent, selector: "bb-product-item-term-deposit-ui", inputs: ["title", "productNumber", "productNumberFormat", "currency", "amount", "mainSectionCurrency", "mainSectionAmount", "mainSectionTitle", "highlight", "productDetailsInfo", "active", "favorite", "favoriteIconColor"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemTermDepositExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-product-item-term-deposit-example-ui',
                    templateUrl: './product-item-term-deposit.example.component.html',
                }]
        }] });
