// @backbase/accessgroup-common-ang/esm2015/lib/helpers/business-function-group.constants.js
export const defaultGetGroupForBusinessFunction = (
//@ts-ignore
businessFunction, defaultGroup) => defaultGroup;
export const defaultOrderedBusinessFunctionGroups = [
    'accountsAndPayments',
    'beneficiaries',
    'actions',
    'messagesAndNotifications',
    'entitlements',
    'identityManagement',
    'supportAccess',
    'flow',
    'employee',
    'portfolio',
    'content',
    'engagements',
    'realTimeCommunication',
    'financeManagement',
    'communicationPreferences',
    'remoteConfig',
    'other',
];
export const defaultBusinessFunctionGroups = {
    accountsAndPayments: [
        'Product Summary',
        'Transactions',
        'Payments',
        'Arrangements',
        'Billpay',
        'Account Statements',
        'Batch',
        'Cash Flow',
        'Pockets',
        'External Accounts',
    ],
    beneficiaries: ['Contacts'],
    actions: ['Actions'],
    messagesAndNotifications: ['Message Center', 'Notifications', 'Actions'],
    entitlements: [
        'Legal Entity',
        'User',
        'Service Agreement',
        'Entitlements',
        'Limits',
        'Approvals',
        'Audit',
        'User Profiles',
        'Manage Authorized Users',
    ],
    identityManagement: ['Identities', 'Device'],
    supportAccess: ['Support Access'],
    flow: ['Flow'],
    employee: ['Employee'],
    portfolio: ['Portfolio'],
    content: ['Content'],
    engagements: ['Engagements'],
    realTimeCommunication: ['Real Time Communications'],
    financeManagement: ['Personal Finance Management'],
    communicationPreferences: ['Communication Preferences'],
    remoteConfig: ['Remote Config'],
    other: [],
};
