// @backbase/ui-ang/esm2015/_example/input-file/showcase/input-file-showcase.module.js
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { InputFileModule } from '@backbase/ui-ang/input-file';
import { InputFileShowcaseComponent } from './input-file-showcase.component';
import * as i0 from "@angular/core";
export class InputFileShowCaseModule {
}
InputFileShowCaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputFileShowCaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputFileShowCaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputFileShowCaseModule, declarations: [InputFileShowcaseComponent], imports: [InputFileModule, ReactiveFormsModule] });
InputFileShowCaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputFileShowCaseModule, imports: [[InputFileModule, ReactiveFormsModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputFileShowCaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [InputFileModule, ReactiveFormsModule],
                    declarations: [InputFileShowcaseComponent],
                    entryComponents: [InputFileShowcaseComponent],
                }]
        }] });
