// @backbase/cashflow-http-ang/esm2015/api/accountingPlatforms.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const AccountingPlatformsHttpServiceGetAccountingPlatformsMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/accounting-platforms",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "quickbooks",
                        "logoUrl": "https://quickbooks.intuit.com/content/dam/intuit/quickbooks/common/qb_thumb.png",
                        "name": "QuickBooks"
                    }, {
                        "id": "xero",
                        "logoUrl": "https://www.xero.com/etc/designs/xero-cms/clientlib/assets/img/logo/logo-xero-blue.svg",
                        "name": "Xero"
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const AccountingPlatformsHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v1/accounting-platforms",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "quickbooks",
                        "logoUrl": "https://quickbooks.intuit.com/content/dam/intuit/quickbooks/common/qb_thumb.png",
                        "name": "QuickBooks"
                    }, {
                        "id": "xero",
                        "logoUrl": "https://www.xero.com/etc/designs/xero-cms/clientlib/assets/img/logo/logo-xero-blue.svg",
                        "name": "Xero"
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
]);
