// @backbase/cashflow-http-ang/esm2015/api/forecasts.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const ForecastsHttpServiceGetForecastByIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/forecasts/{forecastId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "configurationId": "58c844ae-2bf1-419e-aeb6-487f6a18c92b",
                    "syncDatetime": "2020-11-05T11:56:15Z",
                    "currentBalance": "10575882.77283",
                    "currencyCode": "USD",
                    "history": [{
                            "startDate": "2020-11-03",
                            "endDate": "2020-11-03",
                            "openingBalance": "-20500.00000",
                            "closingBalance": "-15500.00000",
                            "netChange": "5000.00000"
                        }, {
                            "startDate": "2020-11-04",
                            "endDate": "2020-11-04",
                            "openingBalance": "-15500.00000",
                            "closingBalance": "0",
                            "netChange": "15500.00000"
                        }],
                    "projection": [{
                            "startDate": "2020-11-05",
                            "endDate": "2020-11-05",
                            "openingBalance": "0",
                            "closingBalance": "6682.33000",
                            "netChange": "6682.33000",
                            "categoryBalanceChanges": [{
                                    "categoryName": "Expense",
                                    "balanceChange": "-1412.67000",
                                    "accountBalanceChanges": [{
                                            "accountId": "8c48192a-723f-4f76-b3a7-c2be76c3bdc8",
                                            "balanceChange": "-1085.00000"
                                        }, {
                                            "accountId": "225f6f2a-445e-48b3-88b5-e3f9daae059a",
                                            "balanceChange": "-86.44000"
                                        }, {
                                            "accountId": "5d8860e2-a624-4009-b9a9-6f7d9add93d5",
                                            "balanceChange": "-241.23000"
                                        }]
                                }, {
                                    "categoryName": "Income",
                                    "balanceChange": "8300.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "f934846d-8461-44dd-89be-3d4cc7b59a53",
                                            "balanceChange": "8300.00000"
                                        }]
                                }, {
                                    "categoryName": "Uncategorised",
                                    "balanceChange": "-205.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "da439b63-4dc2-45c4-9cf8-d4984413403e",
                                            "balanceChange": "-205.00000"
                                        }]
                                }]
                        }, {
                            "startDate": "2020-11-06",
                            "endDate": "2020-11-06",
                            "openingBalance": "6682.33000",
                            "closingBalance": "2337.33000",
                            "netChange": "-4345.00000",
                            "categoryBalanceChanges": [{
                                    "categoryName": "Expense",
                                    "balanceChange": "-14840.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "cfeb44c9-9e25-4fbc-9648-a49b7cf82dad",
                                            "balanceChange": "-14840.00000"
                                        }]
                                }, {
                                    "categoryName": "Income",
                                    "balanceChange": "10495.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "f934846d-8461-44dd-89be-3d4cc7b59a53",
                                            "balanceChange": "10495.00000"
                                        }]
                                }]
                        }, {
                            "startDate": "2020-11-07",
                            "endDate": "2020-11-07",
                            "openingBalance": "2337.33000",
                            "closingBalance": "-447.67000",
                            "netChange": "-2785.00000",
                            "categoryBalanceChanges": [{
                                    "categoryName": "Expense",
                                    "balanceChange": "-17785.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "0d9d51d4-b347-480b-88f3-6107f0ba343f",
                                            "balanceChange": "-755.00000"
                                        }, {
                                            "accountId": "8fca197a-3242-4795-8b06-48d821315ad1",
                                            "balanceChange": "-17030.00000"
                                        }]
                                }, {
                                    "categoryName": "Income",
                                    "balanceChange": "15000.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "1158aa19-894e-40ea-a54f-f2c6808c879a",
                                            "balanceChange": "15000.00000"
                                        }]
                                }]
                        }, {
                            "startDate": "2020-11-08",
                            "endDate": "2020-11-08",
                            "openingBalance": "-447.67000",
                            "closingBalance": "38332.33000",
                            "netChange": "38780.00000",
                            "categoryBalanceChanges": [{
                                    "categoryName": "Expense",
                                    "balanceChange": "-200.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "8fca197a-3242-4795-8b06-48d821315ad1",
                                            "balanceChange": "-200.00000"
                                        }]
                                }, {
                                    "categoryName": "Income",
                                    "balanceChange": "38980.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "f934846d-8461-44dd-89be-3d4cc7b59a53",
                                            "balanceChange": "38980.00000"
                                        }]
                                }]
                        }, {
                            "startDate": "2020-11-09",
                            "endDate": "2020-11-09",
                            "openingBalance": "38332.33000",
                            "closingBalance": "62587.33000",
                            "netChange": "24255.00000",
                            "categoryBalanceChanges": [{
                                    "categoryName": "Expense",
                                    "balanceChange": "-945.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "8fca197a-3242-4795-8b06-48d821315ad1",
                                            "balanceChange": "-945.00000"
                                        }]
                                }, {
                                    "categoryName": "Income",
                                    "balanceChange": "25200.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "f934846d-8461-44dd-89be-3d4cc7b59a53",
                                            "balanceChange": "25200.00000"
                                        }]
                                }]
                        }, {
                            "startDate": "2020-11-10",
                            "endDate": "2020-11-10",
                            "openingBalance": "62587.33000",
                            "closingBalance": "76977.33000",
                            "netChange": "14390.00000",
                            "categoryBalanceChanges": [{
                                    "categoryName": "Expense",
                                    "balanceChange": "-12760.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "b427bbbe-d104-46c5-b6a6-c13e37e5533f",
                                            "balanceChange": "-12760.00000"
                                        }]
                                }, {
                                    "categoryName": "Income",
                                    "balanceChange": "27150.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "1158aa19-894e-40ea-a54f-f2c6808c879a",
                                            "balanceChange": "27150.00000"
                                        }]
                                }]
                        }, {
                            "startDate": "2020-11-11",
                            "endDate": "2020-11-11",
                            "openingBalance": "76977.33000",
                            "closingBalance": "72327.33000",
                            "netChange": "-4650.00000",
                            "categoryBalanceChanges": [{
                                    "categoryName": "Asset",
                                    "balanceChange": "-18900.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "7e776b86-151e-4a4a-a60a-f36e682e2a50",
                                            "balanceChange": "-18900.00000"
                                        }]
                                }, {
                                    "categoryName": "Income",
                                    "balanceChange": "14250.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "dc2af3cd-9c29-4c23-b081-0ceac31d4406",
                                            "balanceChange": "14250.00000"
                                        }]
                                }]
                        }, {
                            "startDate": "2020-11-12",
                            "endDate": "2020-11-12",
                            "openingBalance": "72327.33000",
                            "closingBalance": "86627.33000",
                            "netChange": "14300.00000",
                            "categoryBalanceChanges": [{
                                    "categoryName": "Expense",
                                    "balanceChange": "-4600.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "ecff9789-dff3-44a7-8ffa-72fa0af8a7ec",
                                            "balanceChange": "-4600.00000"
                                        }]
                                }, {
                                    "categoryName": "Income",
                                    "balanceChange": "18900.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "f934846d-8461-44dd-89be-3d4cc7b59a53",
                                            "balanceChange": "18900.00000"
                                        }]
                                }]
                        }, {
                            "startDate": "2020-11-13",
                            "endDate": "2020-11-13",
                            "openingBalance": "86627.33000",
                            "closingBalance": "87837.33000",
                            "netChange": "1210.00000",
                            "categoryBalanceChanges": [{
                                    "categoryName": "Expense",
                                    "balanceChange": "-5800.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "ecff9789-dff3-44a7-8ffa-72fa0af8a7ec",
                                            "balanceChange": "-5800.00000"
                                        }]
                                }, {
                                    "categoryName": "Income",
                                    "balanceChange": "7010.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "9e66abda-69bd-460f-81e2-d540cd2bcf4b",
                                            "balanceChange": "7010.00000"
                                        }]
                                }]
                        }, {
                            "startDate": "2020-11-14",
                            "endDate": "2020-11-14",
                            "openingBalance": "87837.33000",
                            "closingBalance": "109937.33000",
                            "netChange": "22100.00000",
                            "categoryBalanceChanges": [{
                                    "categoryName": "Expense",
                                    "balanceChange": "-900.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "4339c2ee-4040-4996-acf5-810b6c8eb337",
                                            "balanceChange": "-300.00000"
                                        }, {
                                            "accountId": "3b353731-f571-4854-8240-0d49cc81fecc",
                                            "balanceChange": "-600.00000"
                                        }]
                                }, {
                                    "categoryName": "Income",
                                    "balanceChange": "23000.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "f934846d-8461-44dd-89be-3d4cc7b59a53",
                                            "balanceChange": "23000.00000"
                                        }]
                                }]
                        }, {
                            "startDate": "2020-11-15",
                            "endDate": "2020-11-15",
                            "openingBalance": "109937.33000",
                            "closingBalance": "109937.33000",
                            "netChange": "0.00000",
                            "categoryBalanceChanges": []
                        }, {
                            "startDate": "2020-11-16",
                            "endDate": "2020-11-16",
                            "openingBalance": "109937.33000",
                            "closingBalance": "109937.33000",
                            "netChange": "0.00000",
                            "categoryBalanceChanges": []
                        }, {
                            "startDate": "2020-11-17",
                            "endDate": "2020-11-17",
                            "openingBalance": "109937.33000",
                            "closingBalance": "109937.33000",
                            "netChange": "0.00000",
                            "categoryBalanceChanges": []
                        }, {
                            "startDate": "2020-11-18",
                            "endDate": "2020-11-18",
                            "openingBalance": "109937.33000",
                            "closingBalance": "109937.33000",
                            "netChange": "0.00000",
                            "categoryBalanceChanges": []
                        }],
                    "chartOfAccounts": [{
                            "id": "d37dcefc-f5ef-482d-b59a-bad2af1cb9ad",
                            "name": "Accounts Receivable (A/R)",
                            "nominalCode": "84",
                            "category": "Asset"
                        }, {
                            "id": "77b9ba20-924e-4be8-8077-87278297d915",
                            "name": "Checking",
                            "nominalCode": "35",
                            "category": "Asset"
                        }, {
                            "id": "7e776b86-151e-4a4a-a60a-f36e682e2a50",
                            "name": "Depreciation",
                            "nominalCode": "39",
                            "category": "Asset"
                        }, {
                            "id": "1f4efcfd-4e03-49d4-b7a4-3f976f2045db",
                            "name": "Inventory Asset",
                            "nominalCode": "81",
                            "category": "Asset"
                        }, {
                            "id": "271c798c-031c-4cfc-828c-d40e37fbb003",
                            "name": "Original Cost",
                            "nominalCode": "38",
                            "category": "Asset"
                        }, {
                            "id": "e987c68b-6fd9-4a86-bc9b-bac1040fbbbf",
                            "name": "Prepaid Expenses",
                            "nominalCode": "3",
                            "category": "Asset"
                        }, {
                            "id": "497559c7-60bf-476c-8b16-09b8952a9722",
                            "name": "Savings",
                            "nominalCode": "36",
                            "category": "Asset"
                        }, {
                            "id": "9e68b349-72af-4cf3-8ea8-abd76f31064e",
                            "name": "Truck",
                            "nominalCode": "37",
                            "category": "Asset"
                        }, {
                            "id": "5f4450f5-7d1b-4cc3-be31-af9df6488a50",
                            "name": "Uncategorized Asset",
                            "nominalCode": "32",
                            "category": "Asset"
                        }, {
                            "id": "1b953b3a-0ea7-4b87-b5ca-33598b68546b",
                            "name": "Undeposited Funds",
                            "nominalCode": "4",
                            "category": "Asset"
                        }, {
                            "id": "6b718262-6099-4f32-9795-7aae2a25eac7",
                            "name": "Opening Balance Equity",
                            "nominalCode": "34",
                            "category": "Equity"
                        }, {
                            "id": "22a92103-3ee1-41fa-9722-1fc209caa2c2",
                            "name": "Retained Earnings",
                            "nominalCode": "2",
                            "category": "Equity"
                        }, {
                            "id": "8c48192a-723f-4f76-b3a7-c2be76c3bdc8",
                            "name": "Accounting",
                            "nominalCode": "69",
                            "category": "Expense"
                        }, {
                            "id": "d4721d9a-d77b-4361-9807-09331ffcb6af",
                            "name": "Advertising",
                            "nominalCode": "7",
                            "category": "Expense"
                        }, {
                            "id": "b427bbbe-d104-46c5-b6a6-c13e37e5533f",
                            "name": "Automobile",
                            "nominalCode": "55",
                            "category": "Expense"
                        }, {
                            "id": "771aad7c-70d2-4f7e-9309-0ab33e11e589",
                            "name": "Bank Charges",
                            "nominalCode": "8",
                            "category": "Expense"
                        }, {
                            "id": "3d9747ce-afbf-4ff1-b23d-559206c5998c",
                            "name": "Bookkeeper",
                            "nominalCode": "70",
                            "category": "Expense"
                        }, {
                            "id": "2c7ba8c3-8c1d-4e95-b641-e78cbba8c45c",
                            "name": "Building Repairs",
                            "nominalCode": "73",
                            "category": "Expense"
                        }, {
                            "id": "4d0fca62-d1e0-48fe-8481-9410f25ea4bf",
                            "name": "Commissions & fees",
                            "nominalCode": "9",
                            "category": "Expense"
                        }, {
                            "id": "4339c2ee-4040-4996-acf5-810b6c8eb337",
                            "name": "Computer Repairs",
                            "nominalCode": "74",
                            "category": "Expense"
                        }, {
                            "id": "5270f4e2-d0e0-407d-aa12-87954a3d3e74",
                            "name": "Cost of Goods Sold",
                            "nominalCode": "80",
                            "category": "Expense"
                        }, {
                            "id": "f421c967-88eb-43c1-b19e-aa03f15f67ce",
                            "name": "Cost of Labor",
                            "nominalCode": "59",
                            "category": "Expense"
                        }, {
                            "id": "1137cd94-52b1-45b5-a573-1723c89a35f2",
                            "name": "Decks and Patios",
                            "nominalCode": "64",
                            "category": "Expense"
                        }, {
                            "id": "264c45c4-408e-4d80-ab8e-be4a3babb6c8",
                            "name": "Depreciation",
                            "nominalCode": "40",
                            "category": "Expense"
                        }, {
                            "id": "ccfb434a-b289-441c-99a2-30e05a572057",
                            "name": "Disposal Fees",
                            "nominalCode": "28",
                            "category": "Expense"
                        }, {
                            "id": "9c400ca0-009e-475a-abc6-d5a76154aa0a",
                            "name": "Dues & Subscriptions",
                            "nominalCode": "10",
                            "category": "Expense"
                        }, {
                            "id": "74d34736-f4a0-4638-9465-4f5254ddb1b5",
                            "name": "Equipment Rental",
                            "nominalCode": "29",
                            "category": "Expense"
                        }, {
                            "id": "6b90b48e-d8b6-4dc5-8ed4-eb509141e39d",
                            "name": "Equipment Rental",
                            "nominalCode": "62",
                            "category": "Expense"
                        }, {
                            "id": "0d9d51d4-b347-480b-88f3-6107f0ba343f",
                            "name": "Equipment Repairs",
                            "nominalCode": "75",
                            "category": "Expense"
                        }, {
                            "id": "f697f236-19a0-4c4f-be80-9e9f02ec7a6f",
                            "name": "Fountain and Garden Lighting",
                            "nominalCode": "65",
                            "category": "Expense"
                        }, {
                            "id": "7877e117-f0ac-4142-841f-84e062f77128",
                            "name": "Fuel",
                            "nominalCode": "56",
                            "category": "Expense"
                        }, {
                            "id": "225f6f2a-445e-48b3-88b5-e3f9daae059a",
                            "name": "Gas and Electric",
                            "nominalCode": "76",
                            "category": "Expense"
                        }, {
                            "id": "35f1c9d8-2ce4-4f9e-9b34-ac7a0d802645",
                            "name": "Installation",
                            "nominalCode": "60",
                            "category": "Expense"
                        }, {
                            "id": "5d8860e2-a624-4009-b9a9-6f7d9add93d5",
                            "name": "Insurance",
                            "nominalCode": "11",
                            "category": "Expense"
                        }, {
                            "id": "ee810af9-3b00-4de7-9204-c0cbc9bca5f8",
                            "name": "Job Expenses",
                            "nominalCode": "58",
                            "category": "Expense"
                        }, {
                            "id": "ecc43720-0fb4-4b20-9561-73f78021a3fa",
                            "name": "Job Materials",
                            "nominalCode": "63",
                            "category": "Expense"
                        }, {
                            "id": "4a959dfe-428f-48c2-88f6-efc155f2a19c",
                            "name": "Lawyer",
                            "nominalCode": "71",
                            "category": "Expense"
                        }, {
                            "id": "ecff9789-dff3-44a7-8ffa-72fa0af8a7ec",
                            "name": "Legal & Professional Fees",
                            "nominalCode": "12",
                            "category": "Expense"
                        }, {
                            "id": "6333a502-088a-4314-a8d1-05e5cf61e023",
                            "name": "Maintenance and Repair",
                            "nominalCode": "72",
                            "category": "Expense"
                        }, {
                            "id": "4edf486e-f6a7-45a8-9c0b-693675d0b65a",
                            "name": "Maintenance and Repairs",
                            "nominalCode": "61",
                            "category": "Expense"
                        }, {
                            "id": "cfeb44c9-9e25-4fbc-9648-a49b7cf82dad",
                            "name": "Meals and Entertainment",
                            "nominalCode": "13",
                            "category": "Expense"
                        }, {
                            "id": "b2e758c5-92f5-4f52-b915-cc8ea0357868",
                            "name": "Miscellaneous",
                            "nominalCode": "14",
                            "category": "Expense"
                        }, {
                            "id": "1e870df7-7255-41e2-8220-e24c301af038",
                            "name": "Office Expenses",
                            "nominalCode": "15",
                            "category": "Expense"
                        }, {
                            "id": "180cc81e-5621-45f1-8b2f-a7b1aa20cee7",
                            "name": "Penalties & Settlements",
                            "nominalCode": "27",
                            "category": "Expense"
                        }, {
                            "id": "ba81920a-23b7-4634-946f-47f0067d9e1a",
                            "name": "Permits",
                            "nominalCode": "68",
                            "category": "Expense"
                        }, {
                            "id": "c4b4f130-c536-4b83-bbc7-15a7eda2fa67",
                            "name": "Plants and Soil",
                            "nominalCode": "66",
                            "category": "Expense"
                        }, {
                            "id": "8fca197a-3242-4795-8b06-48d821315ad1",
                            "name": "Promotional",
                            "nominalCode": "16",
                            "category": "Expense"
                        }, {
                            "id": "22ed0181-f38c-4a2d-a84e-c5505eb5f14c",
                            "name": "Purchases",
                            "nominalCode": "78",
                            "category": "Expense"
                        }, {
                            "id": "3b353731-f571-4854-8240-0d49cc81fecc",
                            "name": "Rent or Lease",
                            "nominalCode": "17",
                            "category": "Expense"
                        }, {
                            "id": "8e03d0a6-f172-40bd-adf1-3834e9734267",
                            "name": "Repair & Maintenance (deleted)",
                            "nominalCode": "18",
                            "category": "Expense"
                        }, {
                            "id": "bc366bf7-30db-4856-9f65-06ffc01596f5",
                            "name": "Sprinklers and Drip Systems",
                            "nominalCode": "67",
                            "category": "Expense"
                        }, {
                            "id": "c9205c03-ea6a-48a0-be07-733f894db4d8",
                            "name": "Stationery & Printing",
                            "nominalCode": "19",
                            "category": "Expense"
                        }, {
                            "id": "d61afb14-0da1-43ba-935e-4b4a08a012fe",
                            "name": "Supplies",
                            "nominalCode": "20",
                            "category": "Expense"
                        }, {
                            "id": "bb4a5e05-87bd-4373-9b8a-b1dbbd6ec3c8",
                            "name": "Taxes & Licenses",
                            "nominalCode": "21",
                            "category": "Expense"
                        }, {
                            "id": "acf2e2dd-cc98-4c16-a705-d0010b593083",
                            "name": "Telephone",
                            "nominalCode": "77",
                            "category": "Expense"
                        }, {
                            "id": "0e0f7942-612f-481c-8166-ea33fe398a74",
                            "name": "Travel",
                            "nominalCode": "22",
                            "category": "Expense"
                        }, {
                            "id": "1d8e3740-6e22-4908-b37e-7de8bd9a012c",
                            "name": "Travel Meals",
                            "nominalCode": "23",
                            "category": "Expense"
                        }, {
                            "id": "fb3ced8a-8bd5-41e6-8f6d-7c74fcb5aaea",
                            "name": "Unapplied Cash Bill Payment Expense",
                            "nominalCode": "88",
                            "category": "Expense"
                        }, {
                            "id": "3d8fed9c-8726-44fe-bdb0-da49d9457a8d",
                            "name": "Uncategorized Expense",
                            "nominalCode": "31",
                            "category": "Expense"
                        }, {
                            "id": "ca5ea429-6cc0-4870-8513-a4e696b312d4",
                            "name": "Utilities",
                            "nominalCode": "24",
                            "category": "Expense"
                        }, {
                            "id": "a3ad5046-6a41-4b55-9072-66e5872fba62",
                            "name": "Workers Compensation",
                            "nominalCode": "57",
                            "category": "Expense"
                        }, {
                            "id": "773e43aa-8505-48b9-94ef-d7bc5786609a",
                            "name": "Billable Expense Income",
                            "nominalCode": "85",
                            "category": "Income"
                        }, {
                            "id": "026dccb9-d8f6-4116-a3ff-872d0cd628f3",
                            "name": "Decks and Patios",
                            "nominalCode": "47",
                            "category": "Income"
                        }, {
                            "id": "f934846d-8461-44dd-89be-3d4cc7b59a53",
                            "name": "Design income",
                            "nominalCode": "82",
                            "category": "Income"
                        }, {
                            "id": "587bc9dd-553d-4b3d-88d8-5256eedfb499",
                            "name": "Discounts given",
                            "nominalCode": "86",
                            "category": "Income"
                        }, {
                            "id": "ef961774-568d-4746-a790-fa8e22907db5",
                            "name": "Fees Billed",
                            "nominalCode": "5",
                            "category": "Income"
                        }, {
                            "id": "1158aa19-894e-40ea-a54f-f2c6808c879a",
                            "name": "Fountains and Garden Lighting",
                            "nominalCode": "48",
                            "category": "Income"
                        }, {
                            "id": "442e062b-8585-4f14-9763-8990c9fd6087",
                            "name": "Installation",
                            "nominalCode": "52",
                            "category": "Income"
                        }, {
                            "id": "81860ebb-997f-47b6-a1ff-30988af2520d",
                            "name": "Interest Earned",
                            "nominalCode": "25",
                            "category": "Income"
                        }, {
                            "id": "1bb12d54-9d81-4b63-b1c1-af419e47a687",
                            "name": "Job Materials",
                            "nominalCode": "46",
                            "category": "Income"
                        }, {
                            "id": "c9927ae1-eaeb-4b02-baeb-d6bcd64a20ff",
                            "name": "Labor",
                            "nominalCode": "51",
                            "category": "Income"
                        }, {
                            "id": "dc2af3cd-9c29-4c23-b081-0ceac31d4406",
                            "name": "Landscaping Services",
                            "nominalCode": "45",
                            "category": "Income"
                        }, {
                            "id": "9e66abda-69bd-460f-81e2-d540cd2bcf4b",
                            "name": "Maintenance and Repair",
                            "nominalCode": "53",
                            "category": "Income"
                        }, {
                            "id": "bbb2a004-1783-424f-8aac-d7b71d3a9ff5",
                            "name": "Other Income",
                            "nominalCode": "83",
                            "category": "Income"
                        }, {
                            "id": "45b174a0-45be-4134-95ee-27b751f29b6e",
                            "name": "Other Portfolio Income",
                            "nominalCode": "26",
                            "category": "Income"
                        }, {
                            "id": "de6e8ce5-90fc-4379-9972-951b94897fcc",
                            "name": "Pest Control Services",
                            "nominalCode": "54",
                            "category": "Income"
                        }, {
                            "id": "b94b7aca-07a3-4e49-b06c-9a89fb84793d",
                            "name": "Plants and Soil",
                            "nominalCode": "49",
                            "category": "Income"
                        }, {
                            "id": "700def49-ca84-4fa5-83d3-3de9cf393200",
                            "name": "Refunds-Allowances",
                            "nominalCode": "6",
                            "category": "Income"
                        }, {
                            "id": "b15090b5-bcea-4655-a7aa-861e6084ca8b",
                            "name": "Sales of Product Income",
                            "nominalCode": "79",
                            "category": "Income"
                        }, {
                            "id": "cfd72132-9cdc-4ca6-a106-f35d8f6bd727",
                            "name": "Services",
                            "nominalCode": "1",
                            "category": "Income"
                        }, {
                            "id": "6d9c8716-fee0-4da2-a3dc-3abd5afbfdba",
                            "name": "Sprinklers and Drip Systems",
                            "nominalCode": "50",
                            "category": "Income"
                        }, {
                            "id": "c9c3eff9-0968-4d1f-b321-c2499c079014",
                            "name": "Unapplied Cash Payment Income",
                            "nominalCode": "87",
                            "category": "Income"
                        }, {
                            "id": "1152c41c-3dec-44e7-9fc6-b9463b9017f5",
                            "name": "Uncategorized Income",
                            "nominalCode": "30",
                            "category": "Income"
                        }, {
                            "id": "a4163fda-a2c0-4d12-93cb-e016f049fee6",
                            "name": "Accounts Payable (A/P)",
                            "nominalCode": "33",
                            "category": "Liability"
                        }, {
                            "id": "e2521f1a-ab18-483f-89d9-ea0434f00989",
                            "name": "Arizona Dept. of Revenue Payable",
                            "nominalCode": "89",
                            "category": "Liability"
                        }, {
                            "id": "7ebcc80a-06b9-4007-a33d-2f0365694a4d",
                            "name": "Board of Equalization Payable",
                            "nominalCode": "90",
                            "category": "Liability"
                        }, {
                            "id": "26dfb943-b6fb-484b-82b5-85eca47fe07c",
                            "name": "Loan Payable",
                            "nominalCode": "43",
                            "category": "Liability"
                        }, {
                            "id": "109b5a2f-8b10-4953-a388-2934f4f60a6e",
                            "name": "Mastercard",
                            "nominalCode": "41",
                            "category": "Liability"
                        }, {
                            "id": "e11719b8-0ec3-4bd7-943a-72f575ef7890",
                            "name": "Notes Payable",
                            "nominalCode": "44",
                            "category": "Liability"
                        }, {
                            "id": "484ef212-372c-41bc-9a0b-9a3b64c78218",
                            "name": "Visa",
                            "nominalCode": "42",
                            "category": "Liability"
                        }, {
                            "id": "da439b63-4dc2-45c4-9cf8-d4984413403e",
                            "name": "Unknown",
                            "description": "Amounts that are not attributed to any specific account.",
                            "category": "Uncategorised"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const ForecastsHttpServiceGetForecastsMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/forecasts",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "references": [{
                            "id": "58c844ae-2bf1-419e-aeb6-487f6a18c92b"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const ForecastsHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v1/forecasts/{forecastId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "configurationId": "58c844ae-2bf1-419e-aeb6-487f6a18c92b",
                    "syncDatetime": "2020-11-05T11:56:15Z",
                    "currentBalance": "10575882.77283",
                    "currencyCode": "USD",
                    "history": [{
                            "startDate": "2020-11-03",
                            "endDate": "2020-11-03",
                            "openingBalance": "-20500.00000",
                            "closingBalance": "-15500.00000",
                            "netChange": "5000.00000"
                        }, {
                            "startDate": "2020-11-04",
                            "endDate": "2020-11-04",
                            "openingBalance": "-15500.00000",
                            "closingBalance": "0",
                            "netChange": "15500.00000"
                        }],
                    "projection": [{
                            "startDate": "2020-11-05",
                            "endDate": "2020-11-05",
                            "openingBalance": "0",
                            "closingBalance": "6682.33000",
                            "netChange": "6682.33000",
                            "categoryBalanceChanges": [{
                                    "categoryName": "Expense",
                                    "balanceChange": "-1412.67000",
                                    "accountBalanceChanges": [{
                                            "accountId": "8c48192a-723f-4f76-b3a7-c2be76c3bdc8",
                                            "balanceChange": "-1085.00000"
                                        }, {
                                            "accountId": "225f6f2a-445e-48b3-88b5-e3f9daae059a",
                                            "balanceChange": "-86.44000"
                                        }, {
                                            "accountId": "5d8860e2-a624-4009-b9a9-6f7d9add93d5",
                                            "balanceChange": "-241.23000"
                                        }]
                                }, {
                                    "categoryName": "Income",
                                    "balanceChange": "8300.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "f934846d-8461-44dd-89be-3d4cc7b59a53",
                                            "balanceChange": "8300.00000"
                                        }]
                                }, {
                                    "categoryName": "Uncategorised",
                                    "balanceChange": "-205.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "da439b63-4dc2-45c4-9cf8-d4984413403e",
                                            "balanceChange": "-205.00000"
                                        }]
                                }]
                        }, {
                            "startDate": "2020-11-06",
                            "endDate": "2020-11-06",
                            "openingBalance": "6682.33000",
                            "closingBalance": "2337.33000",
                            "netChange": "-4345.00000",
                            "categoryBalanceChanges": [{
                                    "categoryName": "Expense",
                                    "balanceChange": "-14840.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "cfeb44c9-9e25-4fbc-9648-a49b7cf82dad",
                                            "balanceChange": "-14840.00000"
                                        }]
                                }, {
                                    "categoryName": "Income",
                                    "balanceChange": "10495.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "f934846d-8461-44dd-89be-3d4cc7b59a53",
                                            "balanceChange": "10495.00000"
                                        }]
                                }]
                        }, {
                            "startDate": "2020-11-07",
                            "endDate": "2020-11-07",
                            "openingBalance": "2337.33000",
                            "closingBalance": "-447.67000",
                            "netChange": "-2785.00000",
                            "categoryBalanceChanges": [{
                                    "categoryName": "Expense",
                                    "balanceChange": "-17785.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "0d9d51d4-b347-480b-88f3-6107f0ba343f",
                                            "balanceChange": "-755.00000"
                                        }, {
                                            "accountId": "8fca197a-3242-4795-8b06-48d821315ad1",
                                            "balanceChange": "-17030.00000"
                                        }]
                                }, {
                                    "categoryName": "Income",
                                    "balanceChange": "15000.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "1158aa19-894e-40ea-a54f-f2c6808c879a",
                                            "balanceChange": "15000.00000"
                                        }]
                                }]
                        }, {
                            "startDate": "2020-11-08",
                            "endDate": "2020-11-08",
                            "openingBalance": "-447.67000",
                            "closingBalance": "38332.33000",
                            "netChange": "38780.00000",
                            "categoryBalanceChanges": [{
                                    "categoryName": "Expense",
                                    "balanceChange": "-200.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "8fca197a-3242-4795-8b06-48d821315ad1",
                                            "balanceChange": "-200.00000"
                                        }]
                                }, {
                                    "categoryName": "Income",
                                    "balanceChange": "38980.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "f934846d-8461-44dd-89be-3d4cc7b59a53",
                                            "balanceChange": "38980.00000"
                                        }]
                                }]
                        }, {
                            "startDate": "2020-11-09",
                            "endDate": "2020-11-09",
                            "openingBalance": "38332.33000",
                            "closingBalance": "62587.33000",
                            "netChange": "24255.00000",
                            "categoryBalanceChanges": [{
                                    "categoryName": "Expense",
                                    "balanceChange": "-945.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "8fca197a-3242-4795-8b06-48d821315ad1",
                                            "balanceChange": "-945.00000"
                                        }]
                                }, {
                                    "categoryName": "Income",
                                    "balanceChange": "25200.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "f934846d-8461-44dd-89be-3d4cc7b59a53",
                                            "balanceChange": "25200.00000"
                                        }]
                                }]
                        }, {
                            "startDate": "2020-11-10",
                            "endDate": "2020-11-10",
                            "openingBalance": "62587.33000",
                            "closingBalance": "76977.33000",
                            "netChange": "14390.00000",
                            "categoryBalanceChanges": [{
                                    "categoryName": "Expense",
                                    "balanceChange": "-12760.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "b427bbbe-d104-46c5-b6a6-c13e37e5533f",
                                            "balanceChange": "-12760.00000"
                                        }]
                                }, {
                                    "categoryName": "Income",
                                    "balanceChange": "27150.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "1158aa19-894e-40ea-a54f-f2c6808c879a",
                                            "balanceChange": "27150.00000"
                                        }]
                                }]
                        }, {
                            "startDate": "2020-11-11",
                            "endDate": "2020-11-11",
                            "openingBalance": "76977.33000",
                            "closingBalance": "72327.33000",
                            "netChange": "-4650.00000",
                            "categoryBalanceChanges": [{
                                    "categoryName": "Asset",
                                    "balanceChange": "-18900.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "7e776b86-151e-4a4a-a60a-f36e682e2a50",
                                            "balanceChange": "-18900.00000"
                                        }]
                                }, {
                                    "categoryName": "Income",
                                    "balanceChange": "14250.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "dc2af3cd-9c29-4c23-b081-0ceac31d4406",
                                            "balanceChange": "14250.00000"
                                        }]
                                }]
                        }, {
                            "startDate": "2020-11-12",
                            "endDate": "2020-11-12",
                            "openingBalance": "72327.33000",
                            "closingBalance": "86627.33000",
                            "netChange": "14300.00000",
                            "categoryBalanceChanges": [{
                                    "categoryName": "Expense",
                                    "balanceChange": "-4600.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "ecff9789-dff3-44a7-8ffa-72fa0af8a7ec",
                                            "balanceChange": "-4600.00000"
                                        }]
                                }, {
                                    "categoryName": "Income",
                                    "balanceChange": "18900.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "f934846d-8461-44dd-89be-3d4cc7b59a53",
                                            "balanceChange": "18900.00000"
                                        }]
                                }]
                        }, {
                            "startDate": "2020-11-13",
                            "endDate": "2020-11-13",
                            "openingBalance": "86627.33000",
                            "closingBalance": "87837.33000",
                            "netChange": "1210.00000",
                            "categoryBalanceChanges": [{
                                    "categoryName": "Expense",
                                    "balanceChange": "-5800.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "ecff9789-dff3-44a7-8ffa-72fa0af8a7ec",
                                            "balanceChange": "-5800.00000"
                                        }]
                                }, {
                                    "categoryName": "Income",
                                    "balanceChange": "7010.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "9e66abda-69bd-460f-81e2-d540cd2bcf4b",
                                            "balanceChange": "7010.00000"
                                        }]
                                }]
                        }, {
                            "startDate": "2020-11-14",
                            "endDate": "2020-11-14",
                            "openingBalance": "87837.33000",
                            "closingBalance": "109937.33000",
                            "netChange": "22100.00000",
                            "categoryBalanceChanges": [{
                                    "categoryName": "Expense",
                                    "balanceChange": "-900.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "4339c2ee-4040-4996-acf5-810b6c8eb337",
                                            "balanceChange": "-300.00000"
                                        }, {
                                            "accountId": "3b353731-f571-4854-8240-0d49cc81fecc",
                                            "balanceChange": "-600.00000"
                                        }]
                                }, {
                                    "categoryName": "Income",
                                    "balanceChange": "23000.00000",
                                    "accountBalanceChanges": [{
                                            "accountId": "f934846d-8461-44dd-89be-3d4cc7b59a53",
                                            "balanceChange": "23000.00000"
                                        }]
                                }]
                        }, {
                            "startDate": "2020-11-15",
                            "endDate": "2020-11-15",
                            "openingBalance": "109937.33000",
                            "closingBalance": "109937.33000",
                            "netChange": "0.00000",
                            "categoryBalanceChanges": []
                        }, {
                            "startDate": "2020-11-16",
                            "endDate": "2020-11-16",
                            "openingBalance": "109937.33000",
                            "closingBalance": "109937.33000",
                            "netChange": "0.00000",
                            "categoryBalanceChanges": []
                        }, {
                            "startDate": "2020-11-17",
                            "endDate": "2020-11-17",
                            "openingBalance": "109937.33000",
                            "closingBalance": "109937.33000",
                            "netChange": "0.00000",
                            "categoryBalanceChanges": []
                        }, {
                            "startDate": "2020-11-18",
                            "endDate": "2020-11-18",
                            "openingBalance": "109937.33000",
                            "closingBalance": "109937.33000",
                            "netChange": "0.00000",
                            "categoryBalanceChanges": []
                        }],
                    "chartOfAccounts": [{
                            "id": "d37dcefc-f5ef-482d-b59a-bad2af1cb9ad",
                            "name": "Accounts Receivable (A/R)",
                            "nominalCode": "84",
                            "category": "Asset"
                        }, {
                            "id": "77b9ba20-924e-4be8-8077-87278297d915",
                            "name": "Checking",
                            "nominalCode": "35",
                            "category": "Asset"
                        }, {
                            "id": "7e776b86-151e-4a4a-a60a-f36e682e2a50",
                            "name": "Depreciation",
                            "nominalCode": "39",
                            "category": "Asset"
                        }, {
                            "id": "1f4efcfd-4e03-49d4-b7a4-3f976f2045db",
                            "name": "Inventory Asset",
                            "nominalCode": "81",
                            "category": "Asset"
                        }, {
                            "id": "271c798c-031c-4cfc-828c-d40e37fbb003",
                            "name": "Original Cost",
                            "nominalCode": "38",
                            "category": "Asset"
                        }, {
                            "id": "e987c68b-6fd9-4a86-bc9b-bac1040fbbbf",
                            "name": "Prepaid Expenses",
                            "nominalCode": "3",
                            "category": "Asset"
                        }, {
                            "id": "497559c7-60bf-476c-8b16-09b8952a9722",
                            "name": "Savings",
                            "nominalCode": "36",
                            "category": "Asset"
                        }, {
                            "id": "9e68b349-72af-4cf3-8ea8-abd76f31064e",
                            "name": "Truck",
                            "nominalCode": "37",
                            "category": "Asset"
                        }, {
                            "id": "5f4450f5-7d1b-4cc3-be31-af9df6488a50",
                            "name": "Uncategorized Asset",
                            "nominalCode": "32",
                            "category": "Asset"
                        }, {
                            "id": "1b953b3a-0ea7-4b87-b5ca-33598b68546b",
                            "name": "Undeposited Funds",
                            "nominalCode": "4",
                            "category": "Asset"
                        }, {
                            "id": "6b718262-6099-4f32-9795-7aae2a25eac7",
                            "name": "Opening Balance Equity",
                            "nominalCode": "34",
                            "category": "Equity"
                        }, {
                            "id": "22a92103-3ee1-41fa-9722-1fc209caa2c2",
                            "name": "Retained Earnings",
                            "nominalCode": "2",
                            "category": "Equity"
                        }, {
                            "id": "8c48192a-723f-4f76-b3a7-c2be76c3bdc8",
                            "name": "Accounting",
                            "nominalCode": "69",
                            "category": "Expense"
                        }, {
                            "id": "d4721d9a-d77b-4361-9807-09331ffcb6af",
                            "name": "Advertising",
                            "nominalCode": "7",
                            "category": "Expense"
                        }, {
                            "id": "b427bbbe-d104-46c5-b6a6-c13e37e5533f",
                            "name": "Automobile",
                            "nominalCode": "55",
                            "category": "Expense"
                        }, {
                            "id": "771aad7c-70d2-4f7e-9309-0ab33e11e589",
                            "name": "Bank Charges",
                            "nominalCode": "8",
                            "category": "Expense"
                        }, {
                            "id": "3d9747ce-afbf-4ff1-b23d-559206c5998c",
                            "name": "Bookkeeper",
                            "nominalCode": "70",
                            "category": "Expense"
                        }, {
                            "id": "2c7ba8c3-8c1d-4e95-b641-e78cbba8c45c",
                            "name": "Building Repairs",
                            "nominalCode": "73",
                            "category": "Expense"
                        }, {
                            "id": "4d0fca62-d1e0-48fe-8481-9410f25ea4bf",
                            "name": "Commissions & fees",
                            "nominalCode": "9",
                            "category": "Expense"
                        }, {
                            "id": "4339c2ee-4040-4996-acf5-810b6c8eb337",
                            "name": "Computer Repairs",
                            "nominalCode": "74",
                            "category": "Expense"
                        }, {
                            "id": "5270f4e2-d0e0-407d-aa12-87954a3d3e74",
                            "name": "Cost of Goods Sold",
                            "nominalCode": "80",
                            "category": "Expense"
                        }, {
                            "id": "f421c967-88eb-43c1-b19e-aa03f15f67ce",
                            "name": "Cost of Labor",
                            "nominalCode": "59",
                            "category": "Expense"
                        }, {
                            "id": "1137cd94-52b1-45b5-a573-1723c89a35f2",
                            "name": "Decks and Patios",
                            "nominalCode": "64",
                            "category": "Expense"
                        }, {
                            "id": "264c45c4-408e-4d80-ab8e-be4a3babb6c8",
                            "name": "Depreciation",
                            "nominalCode": "40",
                            "category": "Expense"
                        }, {
                            "id": "ccfb434a-b289-441c-99a2-30e05a572057",
                            "name": "Disposal Fees",
                            "nominalCode": "28",
                            "category": "Expense"
                        }, {
                            "id": "9c400ca0-009e-475a-abc6-d5a76154aa0a",
                            "name": "Dues & Subscriptions",
                            "nominalCode": "10",
                            "category": "Expense"
                        }, {
                            "id": "74d34736-f4a0-4638-9465-4f5254ddb1b5",
                            "name": "Equipment Rental",
                            "nominalCode": "29",
                            "category": "Expense"
                        }, {
                            "id": "6b90b48e-d8b6-4dc5-8ed4-eb509141e39d",
                            "name": "Equipment Rental",
                            "nominalCode": "62",
                            "category": "Expense"
                        }, {
                            "id": "0d9d51d4-b347-480b-88f3-6107f0ba343f",
                            "name": "Equipment Repairs",
                            "nominalCode": "75",
                            "category": "Expense"
                        }, {
                            "id": "f697f236-19a0-4c4f-be80-9e9f02ec7a6f",
                            "name": "Fountain and Garden Lighting",
                            "nominalCode": "65",
                            "category": "Expense"
                        }, {
                            "id": "7877e117-f0ac-4142-841f-84e062f77128",
                            "name": "Fuel",
                            "nominalCode": "56",
                            "category": "Expense"
                        }, {
                            "id": "225f6f2a-445e-48b3-88b5-e3f9daae059a",
                            "name": "Gas and Electric",
                            "nominalCode": "76",
                            "category": "Expense"
                        }, {
                            "id": "35f1c9d8-2ce4-4f9e-9b34-ac7a0d802645",
                            "name": "Installation",
                            "nominalCode": "60",
                            "category": "Expense"
                        }, {
                            "id": "5d8860e2-a624-4009-b9a9-6f7d9add93d5",
                            "name": "Insurance",
                            "nominalCode": "11",
                            "category": "Expense"
                        }, {
                            "id": "ee810af9-3b00-4de7-9204-c0cbc9bca5f8",
                            "name": "Job Expenses",
                            "nominalCode": "58",
                            "category": "Expense"
                        }, {
                            "id": "ecc43720-0fb4-4b20-9561-73f78021a3fa",
                            "name": "Job Materials",
                            "nominalCode": "63",
                            "category": "Expense"
                        }, {
                            "id": "4a959dfe-428f-48c2-88f6-efc155f2a19c",
                            "name": "Lawyer",
                            "nominalCode": "71",
                            "category": "Expense"
                        }, {
                            "id": "ecff9789-dff3-44a7-8ffa-72fa0af8a7ec",
                            "name": "Legal & Professional Fees",
                            "nominalCode": "12",
                            "category": "Expense"
                        }, {
                            "id": "6333a502-088a-4314-a8d1-05e5cf61e023",
                            "name": "Maintenance and Repair",
                            "nominalCode": "72",
                            "category": "Expense"
                        }, {
                            "id": "4edf486e-f6a7-45a8-9c0b-693675d0b65a",
                            "name": "Maintenance and Repairs",
                            "nominalCode": "61",
                            "category": "Expense"
                        }, {
                            "id": "cfeb44c9-9e25-4fbc-9648-a49b7cf82dad",
                            "name": "Meals and Entertainment",
                            "nominalCode": "13",
                            "category": "Expense"
                        }, {
                            "id": "b2e758c5-92f5-4f52-b915-cc8ea0357868",
                            "name": "Miscellaneous",
                            "nominalCode": "14",
                            "category": "Expense"
                        }, {
                            "id": "1e870df7-7255-41e2-8220-e24c301af038",
                            "name": "Office Expenses",
                            "nominalCode": "15",
                            "category": "Expense"
                        }, {
                            "id": "180cc81e-5621-45f1-8b2f-a7b1aa20cee7",
                            "name": "Penalties & Settlements",
                            "nominalCode": "27",
                            "category": "Expense"
                        }, {
                            "id": "ba81920a-23b7-4634-946f-47f0067d9e1a",
                            "name": "Permits",
                            "nominalCode": "68",
                            "category": "Expense"
                        }, {
                            "id": "c4b4f130-c536-4b83-bbc7-15a7eda2fa67",
                            "name": "Plants and Soil",
                            "nominalCode": "66",
                            "category": "Expense"
                        }, {
                            "id": "8fca197a-3242-4795-8b06-48d821315ad1",
                            "name": "Promotional",
                            "nominalCode": "16",
                            "category": "Expense"
                        }, {
                            "id": "22ed0181-f38c-4a2d-a84e-c5505eb5f14c",
                            "name": "Purchases",
                            "nominalCode": "78",
                            "category": "Expense"
                        }, {
                            "id": "3b353731-f571-4854-8240-0d49cc81fecc",
                            "name": "Rent or Lease",
                            "nominalCode": "17",
                            "category": "Expense"
                        }, {
                            "id": "8e03d0a6-f172-40bd-adf1-3834e9734267",
                            "name": "Repair & Maintenance (deleted)",
                            "nominalCode": "18",
                            "category": "Expense"
                        }, {
                            "id": "bc366bf7-30db-4856-9f65-06ffc01596f5",
                            "name": "Sprinklers and Drip Systems",
                            "nominalCode": "67",
                            "category": "Expense"
                        }, {
                            "id": "c9205c03-ea6a-48a0-be07-733f894db4d8",
                            "name": "Stationery & Printing",
                            "nominalCode": "19",
                            "category": "Expense"
                        }, {
                            "id": "d61afb14-0da1-43ba-935e-4b4a08a012fe",
                            "name": "Supplies",
                            "nominalCode": "20",
                            "category": "Expense"
                        }, {
                            "id": "bb4a5e05-87bd-4373-9b8a-b1dbbd6ec3c8",
                            "name": "Taxes & Licenses",
                            "nominalCode": "21",
                            "category": "Expense"
                        }, {
                            "id": "acf2e2dd-cc98-4c16-a705-d0010b593083",
                            "name": "Telephone",
                            "nominalCode": "77",
                            "category": "Expense"
                        }, {
                            "id": "0e0f7942-612f-481c-8166-ea33fe398a74",
                            "name": "Travel",
                            "nominalCode": "22",
                            "category": "Expense"
                        }, {
                            "id": "1d8e3740-6e22-4908-b37e-7de8bd9a012c",
                            "name": "Travel Meals",
                            "nominalCode": "23",
                            "category": "Expense"
                        }, {
                            "id": "fb3ced8a-8bd5-41e6-8f6d-7c74fcb5aaea",
                            "name": "Unapplied Cash Bill Payment Expense",
                            "nominalCode": "88",
                            "category": "Expense"
                        }, {
                            "id": "3d8fed9c-8726-44fe-bdb0-da49d9457a8d",
                            "name": "Uncategorized Expense",
                            "nominalCode": "31",
                            "category": "Expense"
                        }, {
                            "id": "ca5ea429-6cc0-4870-8513-a4e696b312d4",
                            "name": "Utilities",
                            "nominalCode": "24",
                            "category": "Expense"
                        }, {
                            "id": "a3ad5046-6a41-4b55-9072-66e5872fba62",
                            "name": "Workers Compensation",
                            "nominalCode": "57",
                            "category": "Expense"
                        }, {
                            "id": "773e43aa-8505-48b9-94ef-d7bc5786609a",
                            "name": "Billable Expense Income",
                            "nominalCode": "85",
                            "category": "Income"
                        }, {
                            "id": "026dccb9-d8f6-4116-a3ff-872d0cd628f3",
                            "name": "Decks and Patios",
                            "nominalCode": "47",
                            "category": "Income"
                        }, {
                            "id": "f934846d-8461-44dd-89be-3d4cc7b59a53",
                            "name": "Design income",
                            "nominalCode": "82",
                            "category": "Income"
                        }, {
                            "id": "587bc9dd-553d-4b3d-88d8-5256eedfb499",
                            "name": "Discounts given",
                            "nominalCode": "86",
                            "category": "Income"
                        }, {
                            "id": "ef961774-568d-4746-a790-fa8e22907db5",
                            "name": "Fees Billed",
                            "nominalCode": "5",
                            "category": "Income"
                        }, {
                            "id": "1158aa19-894e-40ea-a54f-f2c6808c879a",
                            "name": "Fountains and Garden Lighting",
                            "nominalCode": "48",
                            "category": "Income"
                        }, {
                            "id": "442e062b-8585-4f14-9763-8990c9fd6087",
                            "name": "Installation",
                            "nominalCode": "52",
                            "category": "Income"
                        }, {
                            "id": "81860ebb-997f-47b6-a1ff-30988af2520d",
                            "name": "Interest Earned",
                            "nominalCode": "25",
                            "category": "Income"
                        }, {
                            "id": "1bb12d54-9d81-4b63-b1c1-af419e47a687",
                            "name": "Job Materials",
                            "nominalCode": "46",
                            "category": "Income"
                        }, {
                            "id": "c9927ae1-eaeb-4b02-baeb-d6bcd64a20ff",
                            "name": "Labor",
                            "nominalCode": "51",
                            "category": "Income"
                        }, {
                            "id": "dc2af3cd-9c29-4c23-b081-0ceac31d4406",
                            "name": "Landscaping Services",
                            "nominalCode": "45",
                            "category": "Income"
                        }, {
                            "id": "9e66abda-69bd-460f-81e2-d540cd2bcf4b",
                            "name": "Maintenance and Repair",
                            "nominalCode": "53",
                            "category": "Income"
                        }, {
                            "id": "bbb2a004-1783-424f-8aac-d7b71d3a9ff5",
                            "name": "Other Income",
                            "nominalCode": "83",
                            "category": "Income"
                        }, {
                            "id": "45b174a0-45be-4134-95ee-27b751f29b6e",
                            "name": "Other Portfolio Income",
                            "nominalCode": "26",
                            "category": "Income"
                        }, {
                            "id": "de6e8ce5-90fc-4379-9972-951b94897fcc",
                            "name": "Pest Control Services",
                            "nominalCode": "54",
                            "category": "Income"
                        }, {
                            "id": "b94b7aca-07a3-4e49-b06c-9a89fb84793d",
                            "name": "Plants and Soil",
                            "nominalCode": "49",
                            "category": "Income"
                        }, {
                            "id": "700def49-ca84-4fa5-83d3-3de9cf393200",
                            "name": "Refunds-Allowances",
                            "nominalCode": "6",
                            "category": "Income"
                        }, {
                            "id": "b15090b5-bcea-4655-a7aa-861e6084ca8b",
                            "name": "Sales of Product Income",
                            "nominalCode": "79",
                            "category": "Income"
                        }, {
                            "id": "cfd72132-9cdc-4ca6-a106-f35d8f6bd727",
                            "name": "Services",
                            "nominalCode": "1",
                            "category": "Income"
                        }, {
                            "id": "6d9c8716-fee0-4da2-a3dc-3abd5afbfdba",
                            "name": "Sprinklers and Drip Systems",
                            "nominalCode": "50",
                            "category": "Income"
                        }, {
                            "id": "c9c3eff9-0968-4d1f-b321-c2499c079014",
                            "name": "Unapplied Cash Payment Income",
                            "nominalCode": "87",
                            "category": "Income"
                        }, {
                            "id": "1152c41c-3dec-44e7-9fc6-b9463b9017f5",
                            "name": "Uncategorized Income",
                            "nominalCode": "30",
                            "category": "Income"
                        }, {
                            "id": "a4163fda-a2c0-4d12-93cb-e016f049fee6",
                            "name": "Accounts Payable (A/P)",
                            "nominalCode": "33",
                            "category": "Liability"
                        }, {
                            "id": "e2521f1a-ab18-483f-89d9-ea0434f00989",
                            "name": "Arizona Dept. of Revenue Payable",
                            "nominalCode": "89",
                            "category": "Liability"
                        }, {
                            "id": "7ebcc80a-06b9-4007-a33d-2f0365694a4d",
                            "name": "Board of Equalization Payable",
                            "nominalCode": "90",
                            "category": "Liability"
                        }, {
                            "id": "26dfb943-b6fb-484b-82b5-85eca47fe07c",
                            "name": "Loan Payable",
                            "nominalCode": "43",
                            "category": "Liability"
                        }, {
                            "id": "109b5a2f-8b10-4953-a388-2934f4f60a6e",
                            "name": "Mastercard",
                            "nominalCode": "41",
                            "category": "Liability"
                        }, {
                            "id": "e11719b8-0ec3-4bd7-943a-72f575ef7890",
                            "name": "Notes Payable",
                            "nominalCode": "44",
                            "category": "Liability"
                        }, {
                            "id": "484ef212-372c-41bc-9a0b-9a3b64c78218",
                            "name": "Visa",
                            "nominalCode": "42",
                            "category": "Liability"
                        }, {
                            "id": "da439b63-4dc2-45c4-9cf8-d4984413403e",
                            "name": "Unknown",
                            "description": "Amounts that are not attributed to any specific account.",
                            "category": "Uncategorised"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/forecasts",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "references": [{
                            "id": "58c844ae-2bf1-419e-aeb6-487f6a18c92b"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
]);
