// @backbase/cashflow-http-ang/esm2015/api/forecastConfigurations.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { CASH_FLOW_BASE_PATH } from '../variables';
import { CashFlowConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class ForecastConfigurationsHttpService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new CashFlowConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    getForecastConfigurationById(requestParameters, observe = 'body', reportProgress = false, options) {
        const _configurationId = requestParameters["configurationId"];
        if (_configurationId === null || _configurationId === undefined) {
            throw new Error('Required parameter configurationId was null or undefined when calling getForecastConfigurationById.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/forecast-configurations/${encodeURIComponent(String(_configurationId))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getForecastConfigurationByIdUrl(requestParameters) {
        const _configurationId = requestParameters["configurationId"];
        if (_configurationId === null || _configurationId === undefined) {
            throw new Error('Required parameter configurationId was null or undefined when calling getForecastConfigurationById.');
        }
        return `${this.configuration.basePath}/client-api/v1/forecast-configurations/${encodeURIComponent(String(_configurationId))}`;
    }
    getForecastConfigurations(observe = 'body', reportProgress = false, options) {
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/forecast-configurations`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getForecastConfigurationsUrl() {
        return `${this.configuration.basePath}/client-api/v1/forecast-configurations`;
    }
    postForecastConfigurations(requestParameters, observe = 'body', reportProgress = false, options) {
        const _presentationForecastConfiguration = requestParameters["presentationForecastConfiguration"];
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v1/forecast-configurations`, _presentationForecastConfiguration, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    postForecastConfigurationsUrl(requestParameters) {
        const _presentationForecastConfiguration = requestParameters["presentationForecastConfiguration"];
        return `${this.configuration.basePath}/client-api/v1/forecast-configurations`;
    }
    putForecastConfigurationById(requestParameters, observe = 'body', reportProgress = false, options) {
        const _configurationId = requestParameters["configurationId"];
        if (_configurationId === null || _configurationId === undefined) {
            throw new Error('Required parameter configurationId was null or undefined when calling putForecastConfigurationById.');
        }
        const _presentationForecastConfiguration = requestParameters["presentationForecastConfiguration"];
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v1/forecast-configurations/${encodeURIComponent(String(_configurationId))}`, _presentationForecastConfiguration, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    putForecastConfigurationByIdUrl(requestParameters) {
        const _configurationId = requestParameters["configurationId"];
        if (_configurationId === null || _configurationId === undefined) {
            throw new Error('Required parameter configurationId was null or undefined when calling putForecastConfigurationById.');
        }
        const _presentationForecastConfiguration = requestParameters["presentationForecastConfiguration"];
        return `${this.configuration.basePath}/client-api/v1/forecast-configurations/${encodeURIComponent(String(_configurationId))}`;
    }
}
ForecastConfigurationsHttpService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ForecastConfigurationsHttpService, deps: [{ token: i1.HttpClient }, { token: CASH_FLOW_BASE_PATH, optional: true }, { token: i2.CashFlowConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
ForecastConfigurationsHttpService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ForecastConfigurationsHttpService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ForecastConfigurationsHttpService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [CASH_FLOW_BASE_PATH]
                }] }, { type: i2.CashFlowConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
