// @backbase/cashflow-http-ang/esm2015/api/receivables.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const ReceivablesHttpServiceGetReceivableByIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/receivables/{receivableId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "status": "Submitted",
                    "id": "08849602-262b-40b4-a7be-ecd9db00d048",
                    "referenceNumber": "1050",
                    "issueDate": "2020-08-13T00:00:00Z",
                    "dueDate": "2020-12-02T00:00:00Z",
                    "expectedDate": "2020-12-02T00:00:00Z",
                    "dueAmount": "9055.30000",
                    "paidAmount": "0.00000",
                    "totalAmount": "9055.30000",
                    "currencyCode": "USD",
                    "counterparty": {
                        "id": "85c069ea-4f65-45de-9e56-a3f3592e0db7",
                        "name": "Cool Cars",
                        "type": "Customer"
                    },
                    "chartOfAccounts": [{
                            "id": "f934846d-8461-44dd-89be-3d4cc7b59a53",
                            "name": "Design income",
                            "nominalCode": "82",
                            "category": "Income"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const ReceivablesHttpServiceGetReceivablesMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/receivables",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "syncDatetime": "2020-01-31T12:12:12Z",
                    "receivables": [{
                            "id": "1234-5678-9012-3457",
                            "dueDate": "2020-01-31T12:12:12Z",
                            "expectedDate": "2020-02-13T12:12:12Z",
                            "dueAmount": "2.00",
                            "paidAmount": "0.00",
                            "totalAmount": "2.00",
                            "currencyCode": "USD",
                            "referenceNumber": "123-invoice-number",
                            "issueDate": "2020-01-20T12:12:12Z",
                            "counterparty": {
                                "id": "1234-4444-4444-3333",
                                "name": "Richard Brautigan",
                                "type": "Customer"
                            },
                            "chartOfAccounts": [{
                                    "id": "4a0a5e6d-bb31-463a-802f-65a7eff70536",
                                    "name": "Sales",
                                    "category": "Income"
                                }, {
                                    "id": "b068a7bb-917d-4484-8dd6-71cb2a9c9723",
                                    "name": "Accounts Receivable",
                                    "category": "Asset"
                                }],
                            "status": "Draft"
                        }, {
                            "id": "1234-5678-9012-3458",
                            "dueDate": "2020-01-31T12:12:12Z",
                            "expectedDate": "2020-03-14T12:12:12Z",
                            "dueAmount": "10.00",
                            "paidAmount": "0.00",
                            "totalAmount": "10.00",
                            "currencyCode": "CAD",
                            "referenceNumber": "125-invoice-number",
                            "issueDate": "2020-01-20T12:12:12Z",
                            "counterparty": {
                                "id": "1234-4444-4443-3334",
                                "name": "Walt Whitman",
                                "type": "Customer"
                            },
                            "chartOfAccounts": [{
                                    "id": "2th355ar-bb31-463a-802f-65a7eff70536",
                                    "name": "Unknown",
                                    "category": "Uncategorised"
                                }, {
                                    "id": "abr256dr-917d-4484-8dd6-71cb2a9c9723",
                                    "name": "Accounts Receivable",
                                    "category": "Asset"
                                }, {
                                    "id": "4a0a5e6d-bb31-463a-802f-1hnf56at0536",
                                    "name": "Sales",
                                    "category": "Income"
                                }],
                            "status": "Submitted"
                        }, {
                            "id": "1234-5678-9012-3459",
                            "dueDate": "2019-12-31T12:12:12Z",
                            "expectedDate": "2020-04-14T12:12:12Z",
                            "dueAmount": "155.00",
                            "paidAmount": "145.00",
                            "totalAmount": "300.00",
                            "currencyCode": "EUR",
                            "referenceNumber": "124-invoice-number",
                            "issueDate": "2019-10-10T12:12:12Z",
                            "counterparty": {
                                "id": "1234-4444-4443-3335",
                                "name": "Paul Celan",
                                "type": "Customer"
                            },
                            "chartOfAccounts": [{
                                    "id": "3gt6jka1-bb31-463a-802f-65a7eff70536",
                                    "name": "Sales",
                                    "category": "Income"
                                }],
                            "status": "PartiallyPaid"
                        }, {
                            "id": "1234-5678-9012-3460",
                            "dueDate": "2020-01-31T12:12:12Z",
                            "expectedDate": "2020-05-15T12:12:12Z",
                            "dueAmount": "20020.00",
                            "paidAmount": "0.00",
                            "totalAmount": "20020.00",
                            "currencyCode": "USD",
                            "referenceNumber": "123-invoice-number",
                            "issueDate": "2020-01-20T12:12:12Z",
                            "counterparty": {
                                "id": "1234-4444-4444-3336",
                                "name": "Sylvia Plath",
                                "type": "Customer"
                            },
                            "chartOfAccounts": [{
                                    "id": "bg2ya541-bb31-463a-802f-65a7eff70536",
                                    "name": "Unknown",
                                    "category": "Uncategorised"
                                }],
                            "status": "Submitted"
                        }, {
                            "id": "1234-5678-9012-3461",
                            "dueDate": "2020-02-28T12:12:12Z",
                            "expectedDate": "2020-06-16T12:12:12Z",
                            "dueAmount": "5000.00",
                            "paidAmount": "5000.00",
                            "totalAmount": "10000.00",
                            "currencyCode": "USD",
                            "referenceNumber": "125-invoice-number",
                            "issueDate": "2019-12-20T12:12:12Z",
                            "counterparty": {
                                "id": "1234-4444-4443-3337",
                                "name": "Jeff Beck",
                                "type": "Customer"
                            },
                            "chartOfAccounts": [{
                                    "id": "aa54f12y-bb31-463a-802f-65a7eff70536",
                                    "name": "Accounts Receivable",
                                    "category": "Asset"
                                }],
                            "status": "PartiallyPaid"
                        }, {
                            "id": "1234-5678-9012-3462",
                            "dueDate": "2019-12-21T12:12:12Z",
                            "expectedDate": "2020-07-17T12:12:12Z",
                            "dueAmount": "0",
                            "paidAmount": "10000.00",
                            "totalAmount": "10000.00",
                            "currencyCode": "RUB",
                            "referenceNumber": "124-invoice-number",
                            "issueDate": "2019-11-10T12:12:12Z",
                            "counterparty": {
                                "id": "1234-4444-4443-3338",
                                "name": "Boris Ryzhyi",
                                "type": "Customer"
                            },
                            "chartOfAccounts": [{
                                    "id": "h1t45dre-bb31-463a-802f-65a7eff70536",
                                    "name": "Accounts Receivable",
                                    "category": "Asset"
                                }],
                            "status": "Paid"
                        }, {
                            "id": "1234-5678-9012-3463",
                            "dueDate": "2020-01-19T12:12:12Z",
                            "expectedDate": "2020-08-18T12:12:12Z",
                            "dueAmount": "120000.00",
                            "paidAmount": "0.00",
                            "totalAmount": "120000.00",
                            "currencyCode": "CAD",
                            "referenceNumber": "125-invoice-number",
                            "issueDate": "2019-11-20T12:12:12Z",
                            "counterparty": {
                                "id": "1234-4444-4443-3339",
                                "name": "Emily Dickinson",
                                "type": "Customer"
                            },
                            "chartOfAccounts": [{
                                    "id": "11bg3rp7-bb31-463a-802f-65a7ef1htr56",
                                    "name": "Sales",
                                    "category": "Income"
                                }],
                            "status": "Submitted"
                        }, {
                            "id": "1234-5678-9012-3464",
                            "dueDate": "2019-12-26T12:12:12Z",
                            "expectedDate": "2020-09-19T12:12:12Z",
                            "dueAmount": "10000.00",
                            "paidAmount": "0.00",
                            "totalAmount": "10000.00",
                            "currencyCode": "EUR",
                            "referenceNumber": "124-invoice-number",
                            "issueDate": "2019-10-15T12:12:12Z",
                            "counterparty": {
                                "id": "1234-4444-4443-3340",
                                "name": "Mary Jane",
                                "type": "Customer"
                            },
                            "chartOfAccounts": [{
                                    "id": "aa5gpm67-bb31-463a-802f-65a7ef1ht1gt",
                                    "name": "Sales",
                                    "category": "Income"
                                }, {
                                    "id": "ba784pt1-bb31-463a-802f-65a7ef1htr56",
                                    "name": "Unknown",
                                    "category": "Uncategorised"
                                }],
                            "status": "PartiallyPaid"
                        }, {
                            "id": "1234-5678-9012-3465",
                            "dueDate": "2019-11-30T12:12:12Z",
                            "expectedDate": "2020-10-20T12:12:12Z",
                            "dueAmount": "20020.00",
                            "paidAmount": "0.00",
                            "totalAmount": "20020.00",
                            "currencyCode": "USD",
                            "referenceNumber": "123-invoice-number",
                            "issueDate": "2019-10-20T12:12:12Z",
                            "counterparty": {
                                "id": "1234-4444-4444-3341",
                                "name": "William Yeats",
                                "type": "Customer"
                            },
                            "chartOfAccounts": [{
                                    "id": "ee79mn61-bb31-463a-802f-65a7ef1htr56",
                                    "name": "Sales",
                                    "category": "Income"
                                }],
                            "status": "Draft"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const ReceivablesHttpServicePutExpectedDateByReceivableIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/receivables/{receivableId}/expected-date",
        method: "PUT",
        responses: [
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const ReceivablesHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v1/receivables/{receivableId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "status": "Submitted",
                    "id": "08849602-262b-40b4-a7be-ecd9db00d048",
                    "referenceNumber": "1050",
                    "issueDate": "2020-08-13T00:00:00Z",
                    "dueDate": "2020-12-02T00:00:00Z",
                    "expectedDate": "2020-12-02T00:00:00Z",
                    "dueAmount": "9055.30000",
                    "paidAmount": "0.00000",
                    "totalAmount": "9055.30000",
                    "currencyCode": "USD",
                    "counterparty": {
                        "id": "85c069ea-4f65-45de-9e56-a3f3592e0db7",
                        "name": "Cool Cars",
                        "type": "Customer"
                    },
                    "chartOfAccounts": [{
                            "id": "f934846d-8461-44dd-89be-3d4cc7b59a53",
                            "name": "Design income",
                            "nominalCode": "82",
                            "category": "Income"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/receivables",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "syncDatetime": "2020-01-31T12:12:12Z",
                    "receivables": [{
                            "id": "1234-5678-9012-3457",
                            "dueDate": "2020-01-31T12:12:12Z",
                            "expectedDate": "2020-02-13T12:12:12Z",
                            "dueAmount": "2.00",
                            "paidAmount": "0.00",
                            "totalAmount": "2.00",
                            "currencyCode": "USD",
                            "referenceNumber": "123-invoice-number",
                            "issueDate": "2020-01-20T12:12:12Z",
                            "counterparty": {
                                "id": "1234-4444-4444-3333",
                                "name": "Richard Brautigan",
                                "type": "Customer"
                            },
                            "chartOfAccounts": [{
                                    "id": "4a0a5e6d-bb31-463a-802f-65a7eff70536",
                                    "name": "Sales",
                                    "category": "Income"
                                }, {
                                    "id": "b068a7bb-917d-4484-8dd6-71cb2a9c9723",
                                    "name": "Accounts Receivable",
                                    "category": "Asset"
                                }],
                            "status": "Draft"
                        }, {
                            "id": "1234-5678-9012-3458",
                            "dueDate": "2020-01-31T12:12:12Z",
                            "expectedDate": "2020-03-14T12:12:12Z",
                            "dueAmount": "10.00",
                            "paidAmount": "0.00",
                            "totalAmount": "10.00",
                            "currencyCode": "CAD",
                            "referenceNumber": "125-invoice-number",
                            "issueDate": "2020-01-20T12:12:12Z",
                            "counterparty": {
                                "id": "1234-4444-4443-3334",
                                "name": "Walt Whitman",
                                "type": "Customer"
                            },
                            "chartOfAccounts": [{
                                    "id": "2th355ar-bb31-463a-802f-65a7eff70536",
                                    "name": "Unknown",
                                    "category": "Uncategorised"
                                }, {
                                    "id": "abr256dr-917d-4484-8dd6-71cb2a9c9723",
                                    "name": "Accounts Receivable",
                                    "category": "Asset"
                                }, {
                                    "id": "4a0a5e6d-bb31-463a-802f-1hnf56at0536",
                                    "name": "Sales",
                                    "category": "Income"
                                }],
                            "status": "Submitted"
                        }, {
                            "id": "1234-5678-9012-3459",
                            "dueDate": "2019-12-31T12:12:12Z",
                            "expectedDate": "2020-04-14T12:12:12Z",
                            "dueAmount": "155.00",
                            "paidAmount": "145.00",
                            "totalAmount": "300.00",
                            "currencyCode": "EUR",
                            "referenceNumber": "124-invoice-number",
                            "issueDate": "2019-10-10T12:12:12Z",
                            "counterparty": {
                                "id": "1234-4444-4443-3335",
                                "name": "Paul Celan",
                                "type": "Customer"
                            },
                            "chartOfAccounts": [{
                                    "id": "3gt6jka1-bb31-463a-802f-65a7eff70536",
                                    "name": "Sales",
                                    "category": "Income"
                                }],
                            "status": "PartiallyPaid"
                        }, {
                            "id": "1234-5678-9012-3460",
                            "dueDate": "2020-01-31T12:12:12Z",
                            "expectedDate": "2020-05-15T12:12:12Z",
                            "dueAmount": "20020.00",
                            "paidAmount": "0.00",
                            "totalAmount": "20020.00",
                            "currencyCode": "USD",
                            "referenceNumber": "123-invoice-number",
                            "issueDate": "2020-01-20T12:12:12Z",
                            "counterparty": {
                                "id": "1234-4444-4444-3336",
                                "name": "Sylvia Plath",
                                "type": "Customer"
                            },
                            "chartOfAccounts": [{
                                    "id": "bg2ya541-bb31-463a-802f-65a7eff70536",
                                    "name": "Unknown",
                                    "category": "Uncategorised"
                                }],
                            "status": "Submitted"
                        }, {
                            "id": "1234-5678-9012-3461",
                            "dueDate": "2020-02-28T12:12:12Z",
                            "expectedDate": "2020-06-16T12:12:12Z",
                            "dueAmount": "5000.00",
                            "paidAmount": "5000.00",
                            "totalAmount": "10000.00",
                            "currencyCode": "USD",
                            "referenceNumber": "125-invoice-number",
                            "issueDate": "2019-12-20T12:12:12Z",
                            "counterparty": {
                                "id": "1234-4444-4443-3337",
                                "name": "Jeff Beck",
                                "type": "Customer"
                            },
                            "chartOfAccounts": [{
                                    "id": "aa54f12y-bb31-463a-802f-65a7eff70536",
                                    "name": "Accounts Receivable",
                                    "category": "Asset"
                                }],
                            "status": "PartiallyPaid"
                        }, {
                            "id": "1234-5678-9012-3462",
                            "dueDate": "2019-12-21T12:12:12Z",
                            "expectedDate": "2020-07-17T12:12:12Z",
                            "dueAmount": "0",
                            "paidAmount": "10000.00",
                            "totalAmount": "10000.00",
                            "currencyCode": "RUB",
                            "referenceNumber": "124-invoice-number",
                            "issueDate": "2019-11-10T12:12:12Z",
                            "counterparty": {
                                "id": "1234-4444-4443-3338",
                                "name": "Boris Ryzhyi",
                                "type": "Customer"
                            },
                            "chartOfAccounts": [{
                                    "id": "h1t45dre-bb31-463a-802f-65a7eff70536",
                                    "name": "Accounts Receivable",
                                    "category": "Asset"
                                }],
                            "status": "Paid"
                        }, {
                            "id": "1234-5678-9012-3463",
                            "dueDate": "2020-01-19T12:12:12Z",
                            "expectedDate": "2020-08-18T12:12:12Z",
                            "dueAmount": "120000.00",
                            "paidAmount": "0.00",
                            "totalAmount": "120000.00",
                            "currencyCode": "CAD",
                            "referenceNumber": "125-invoice-number",
                            "issueDate": "2019-11-20T12:12:12Z",
                            "counterparty": {
                                "id": "1234-4444-4443-3339",
                                "name": "Emily Dickinson",
                                "type": "Customer"
                            },
                            "chartOfAccounts": [{
                                    "id": "11bg3rp7-bb31-463a-802f-65a7ef1htr56",
                                    "name": "Sales",
                                    "category": "Income"
                                }],
                            "status": "Submitted"
                        }, {
                            "id": "1234-5678-9012-3464",
                            "dueDate": "2019-12-26T12:12:12Z",
                            "expectedDate": "2020-09-19T12:12:12Z",
                            "dueAmount": "10000.00",
                            "paidAmount": "0.00",
                            "totalAmount": "10000.00",
                            "currencyCode": "EUR",
                            "referenceNumber": "124-invoice-number",
                            "issueDate": "2019-10-15T12:12:12Z",
                            "counterparty": {
                                "id": "1234-4444-4443-3340",
                                "name": "Mary Jane",
                                "type": "Customer"
                            },
                            "chartOfAccounts": [{
                                    "id": "aa5gpm67-bb31-463a-802f-65a7ef1ht1gt",
                                    "name": "Sales",
                                    "category": "Income"
                                }, {
                                    "id": "ba784pt1-bb31-463a-802f-65a7ef1htr56",
                                    "name": "Unknown",
                                    "category": "Uncategorised"
                                }],
                            "status": "PartiallyPaid"
                        }, {
                            "id": "1234-5678-9012-3465",
                            "dueDate": "2019-11-30T12:12:12Z",
                            "expectedDate": "2020-10-20T12:12:12Z",
                            "dueAmount": "20020.00",
                            "paidAmount": "0.00",
                            "totalAmount": "20020.00",
                            "currencyCode": "USD",
                            "referenceNumber": "123-invoice-number",
                            "issueDate": "2019-10-20T12:12:12Z",
                            "counterparty": {
                                "id": "1234-4444-4444-3341",
                                "name": "William Yeats",
                                "type": "Customer"
                            },
                            "chartOfAccounts": [{
                                    "id": "ee79mn61-bb31-463a-802f-65a7ef1htr56",
                                    "name": "Sales",
                                    "category": "Income"
                                }],
                            "status": "Draft"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/receivables/{receivableId}/expected-date",
        method: "PUT",
        responses: [
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
]);
