// @backbase/account-statement-retail-journey-ang/esm2015/src/view/account-statement-retail-view.component.js
import { Component } from '@angular/core';
import { combineLatest } from 'rxjs';
import { map } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "../account-statement-retail-journey-configuration.service";
import * as i2 from "@backbase/account-statement-list-widget-ang";
import * as i3 from "@angular/common";
export class AccountStatementRetailViewComponent {
    constructor(configService) {
        this.configService = configService;
        /**
         * View model for Account Statements Retail view
         */
        this.vm$ = combineLatest([
            this.configService.allConfig,
        ]).pipe(map(([config]) => ({ config })));
    }
}
AccountStatementRetailViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementRetailViewComponent, deps: [{ token: i1.AccountStatementRetailJourneyConfigurationService }], target: i0.ɵɵFactoryTarget.Component });
AccountStatementRetailViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: AccountStatementRetailViewComponent, selector: "bb-account-statement-retail-view", ngImport: i0, template: "<bb-account-statement-list-widget\n  *ngIf=\"vm$ | async as vm\"\n  [absoluteDateLimit]=\"vm.config.absoluteDateLimit\"\n  [notificationDismissTime]=\"vm.config.notificationDismissTime\"\n  [pageSize]=\"vm.config.pageSize\"\n  [paginationType]=\"vm.config.paginationType\"\n  [relativeDateLimit]=\"vm.config.relativeDateLimit\"\n  [showCategory]=\"vm.config.showCategory\"\n  [showDescription]=\"vm.config.showDescription\"\n  [sortingDirection]=\"vm.config.sortingDirection\"\n  [sortingOrder]=\"vm.config.sortingOrder\"\n>\n</bb-account-statement-list-widget>\n", components: [{ type: i2.AccountStatementListWidgetComponent, selector: "bb-account-statement-list-widget", inputs: ["showDescription"] }], directives: [{ type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "async": i3.AsyncPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementRetailViewComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-account-statement-retail-view',
                    templateUrl: './account-statement-retail-view.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.AccountStatementRetailJourneyConfigurationService }]; } });
