// @backbase/product-summary-ui/esm2015/src/bb-area-chart-ui/area-chart-legend.component.js
import { Component, Directive, Input } from '@angular/core';
import { BbTemplate } from '@backbase/foundation-ang/core';
import { findByName } from '@backbase/product-summary-common-ang';
import { CustomColorName } from '../types/chart';
import * as i0 from "@angular/core";
import * as i1 from "../bb-chart-legend-ui/chart-legend.component";
export class AreaChartLegendComponent {
    constructor() {
        this.hostRef = this;
        /**
         * The chart data (data series) that this component is using to calculate min max
         */
        this.dataSeries = [];
        /**
         * The currency data
         */
        this.currency = '';
        /**
         * Set custom colors for negative and positive areas
         */
        this.customColors = [];
    }
    getMax() {
        return Math.max(...this.values);
    }
    getMin() {
        return Math.min(...this.values);
    }
    getPositiveColor() {
        const customColor = findByName(this.customColors, CustomColorName.POSITIVE);
        return customColor ? customColor.value : '';
    }
    getNegativeColor() {
        const customColor = findByName(this.customColors, CustomColorName.NEGATIVE);
        return customColor ? customColor.value : '';
    }
    get values() {
        const allDateSeries = this.dataSeries.reduce((memo, { series }) => memo.concat(series), []);
        return allDateSeries.map((date) => date.value);
    }
}
AreaChartLegendComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AreaChartLegendComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
AreaChartLegendComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AreaChartLegendComponent, selector: "bb-area-chart-legend-ui", inputs: { dataSeries: "dataSeries", currency: "currency", customColors: "customColors" }, ngImport: i0, template: "<ng-container\n  bbBalanceHistoryAreaChartLegendCustomizable\n  [bbHostRef]=\"hostRef\"\n  [bbTemplateContext]=\"{\n    dataSeries: dataSeries,\n    customColors: customColors\n  }\"\n></ng-container>\n\n<ng-template\n  bbBalanceHistoryAreaChartLegendCustomizable\n  let-hostRef\n  let-dataSeries=\"context.dataSeries\"\n  let-customColors=\"context.customColors\"\n>\n  <bb-chart-legend-ui\n    [currency]=\"hostRef.currency\"\n    [maxValue]=\"hostRef.getMax()\"\n    [minValue]=\"hostRef.getMin()\"\n    [positiveColor]=\"hostRef.getPositiveColor()\"\n    [negativeColor]=\"hostRef.getNegativeColor()\"\n  ></bb-chart-legend-ui>\n</ng-template>\n", components: [{ type: i0.forwardRef(function () { return i1.ChartLegendComponent; }), selector: "bb-chart-legend-ui", inputs: ["currency", "maxValue", "minValue", "positiveColor", "negativeColor"] }], directives: [{ type: i0.forwardRef(function () { return BalanceHistoryAreaChartLegendCustomizableDirective; }), selector: "[bbBalanceHistoryAreaChartLegendCustomizable]" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AreaChartLegendComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-area-chart-legend-ui',
                    templateUrl: 'area-chart-legend.component.html',
                }]
        }], propDecorators: { dataSeries: [{
                type: Input
            }], currency: [{
                type: Input
            }], customColors: [{
                type: Input
            }] } });
export class BalanceHistoryAreaChartLegendCustomizableDirective extends BbTemplate {
}
BalanceHistoryAreaChartLegendCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BalanceHistoryAreaChartLegendCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
BalanceHistoryAreaChartLegendCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: BalanceHistoryAreaChartLegendCustomizableDirective, selector: "[bbBalanceHistoryAreaChartLegendCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BalanceHistoryAreaChartLegendCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbBalanceHistoryAreaChartLegendCustomizable]',
                }]
        }] });
