// @backbase/product-summary-ui/esm2015/src/bb-area-chart-ui/area-chart.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { IconModule } from '@backbase/ui-ang/icon';
import { AmountModule } from '@backbase/ui-ang/amount';
import { AreaChartModule as SwimlineAreaChartModule } from '@swimlane/ngx-charts';
import { AreaChartLegendComponent, BalanceHistoryAreaChartLegendCustomizableDirective, } from './area-chart-legend.component';
import { AreaChartComponent, BalanceHistoryAreaChartCustomizableDirective } from './area-chart.component';
import { ChartLegendModule } from '../bb-chart-legend-ui/chart-legend.module';
import * as i0 from "@angular/core";
const uiModules = [IconModule, AmountModule, ChartLegendModule];
const components = [AreaChartLegendComponent, AreaChartComponent];
const directives = [BalanceHistoryAreaChartLegendCustomizableDirective, BalanceHistoryAreaChartCustomizableDirective];
export class AreaChartModule {
}
AreaChartModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AreaChartModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AreaChartModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AreaChartModule, declarations: [AreaChartLegendComponent, AreaChartComponent, BalanceHistoryAreaChartLegendCustomizableDirective, BalanceHistoryAreaChartCustomizableDirective], imports: [CommonModule, SwimlineAreaChartModule, IconModule, AmountModule, ChartLegendModule], exports: [AreaChartLegendComponent, AreaChartComponent, BalanceHistoryAreaChartLegendCustomizableDirective, BalanceHistoryAreaChartCustomizableDirective, IconModule, AmountModule, ChartLegendModule] });
AreaChartModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AreaChartModule, imports: [[CommonModule, SwimlineAreaChartModule, ...uiModules], IconModule, AmountModule, ChartLegendModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AreaChartModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, SwimlineAreaChartModule, ...uiModules],
                    declarations: [...components, ...directives],
                    exports: [...components, ...directives, ...uiModules],
                }]
        }] });
