// @backbase/payment-orders-ang/esm2015/lib/ui/constants/model.js
export const contactsPageSize = 1000;
export const arrangementsPageSize = 100;
export const dataItemPermissionParams = {
    resourceName: 'Payments',
    privilege: 'create',
    dataGroupType: 'ARRANGEMENTS',
};
export var BeneficiaryList;
(function (BeneficiaryList) {
    BeneficiaryList["CONTACTS"] = "contacts";
    BeneficiaryList["ACCOUNTS"] = "accounts";
})(BeneficiaryList || (BeneficiaryList = {}));
export const defaultBeneficiaryTabsConfig = [
    BeneficiaryList.CONTACTS,
    BeneficiaryList.ACCOUNTS,
];
export var ListState;
(function (ListState) {
    ListState["EMPTY"] = "empty";
    ListState["LIST"] = "list";
    ListState["FULL_LIST"] = "full-list";
    ListState["LOADING"] = "loading";
    ListState["FETCHING"] = "fetching";
    ListState["FETCHING_FAILURE"] = "fetching-failure";
    ListState["NO_RESULT"] = "no-result";
    ListState["ERROR"] = "error";
})(ListState || (ListState = {}));
export var PaymentTypeIcon;
(function (PaymentTypeIcon) {
    PaymentTypeIcon["INTERNATIONAL"] = "language";
    PaymentTypeIcon["DOMESTIC"] = "accounts";
    PaymentTypeIcon["INTERNAL"] = "domain";
})(PaymentTypeIcon || (PaymentTypeIcon = {}));
