// @backbase/loans-collection-ui-ang/esm2015/lib/loan-list-item/loan-item-payment-info/loan-item-payment-info.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AmountModule } from '@backbase/ui-ang/amount';
import { BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import { AlertModule } from '@backbase/ui-ang/alert';
import { LoanItemPaymentInfoComponent } from './loan-item-payment-info.component';
import { RetailLoanItemPaymentInfoComponent } from './retail-loan-item-payment-info/retail-loan-item-payment-info.component';
import * as i0 from "@angular/core";
export class LoanItemPaymentInfoModule {
}
/** @nocollapse */ LoanItemPaymentInfoModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemPaymentInfoModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanItemPaymentInfoModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemPaymentInfoModule, declarations: [LoanItemPaymentInfoComponent, RetailLoanItemPaymentInfoComponent], imports: [CommonModule, AmountModule, BbDatePipeModule, AlertModule], exports: [LoanItemPaymentInfoComponent, RetailLoanItemPaymentInfoComponent] });
/** @nocollapse */ LoanItemPaymentInfoModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemPaymentInfoModule, imports: [[CommonModule, AmountModule, BbDatePipeModule, AlertModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemPaymentInfoModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanItemPaymentInfoComponent, RetailLoanItemPaymentInfoComponent],
                    imports: [CommonModule, AmountModule, BbDatePipeModule, AlertModule],
                    exports: [LoanItemPaymentInfoComponent, RetailLoanItemPaymentInfoComponent],
                }]
        }] });
