// @backbase/ui-ang/esm2015/_example/list-multiple-select/showcase/list-multiple-select-showcase.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@angular/forms";
import * as i2 from "@backbase/ui-ang/list-multiple-select";
let ListMultipleSelectShowcaseComponent = class ListMultipleSelectShowcaseComponent {
    constructor(formBuilder) {
        this.formBuilder = formBuilder;
        this.listMultipleSelectItemsConfig = {
            'List Item 1': {},
            'List Item 2': {
                items: {
                    q2: {
                        icon: 'warning',
                    },
                },
            },
        };
        this.vForm = this.formBuilder.group({
            listMultipleSelect: this.formBuilder.group({
                'List item 1': true,
                'List item 2': this.formBuilder.group({
                    'Sub item 1': false,
                    'Sub item 2': true,
                    'Sub item 3': this.formBuilder.group({
                        'Sub sub item 1': { value: true, disabled: false },
                        'Disabled sub item 2': { value: false, disabled: true },
                    }),
                }),
            }),
        });
    }
    onListMultipleSelectClearAll($event) {
        $event.preventDefault();
        if (this.vForm) {
            this.vForm.controls['listMultipleSelect'].patchValue({
                i1: false,
                i2: false,
                i3: {
                    q1: false,
                    q2: false,
                },
            });
        }
    }
};
ListMultipleSelectShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ListMultipleSelectShowcaseComponent, deps: [{ token: i1.FormBuilder }], target: i0.ɵɵFactoryTarget.Component });
ListMultipleSelectShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ListMultipleSelectShowcaseComponent, selector: "bb-list-multiple-select-ui-showcase", ngImport: i0, template: "<form [formGroup]=\"vForm\">\n  <bb-list-multiple-select-ui\n    formGroupName=\"listMultipleSelect\"\n    [config]=\"listMultipleSelectItemsConfig\"\n    aria-label=\"List multiple select\"\n  >\n    <ng-template bbListMultipleSelectAction>\n      <a href=\"\" (click)=\"onListMultipleSelectClearAll($event)\">Clear All</a>\n    </ng-template>\n  </bb-list-multiple-select-ui>\n</form>\n", components: [{ type: i2.ListMultipleSelectComponent, selector: "bb-list-multiple-select-ui", inputs: ["config"] }], directives: [{ type: i1.ɵNgNoValidate, selector: "form:not([ngNoForm]):not([ngNativeValidate])" }, { type: i1.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { type: i1.FormGroupDirective, selector: "[formGroup]", inputs: ["formGroup"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { type: i1.FormGroupName, selector: "[formGroupName]", inputs: ["formGroupName"] }, { type: i2.ListMultipleSelectActionDirective, selector: "ng-template[bbListMultipleSelectAction]" }] });
ListMultipleSelectShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
], ListMultipleSelectShowcaseComponent);
export { ListMultipleSelectShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ListMultipleSelectShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-list-multiple-select-ui-showcase',
                    templateUrl: './list-multiple-select-showcase.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.FormBuilder }]; } });
