// @backbase/ui-ang/esm2015/_example/list-multiple-select/showcase/list-multiple-select-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ListMultipleSelectShowcaseComponent } from './list-multiple-select-showcase.component';
import { ListMultipleSelectModule } from '@backbase/ui-ang/list-multiple-select';
import { ReactiveFormsModule } from '@angular/forms';
import * as i0 from "@angular/core";
export class ListMultipleSelectShowcaseModule {
}
ListMultipleSelectShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ListMultipleSelectShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ListMultipleSelectShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ListMultipleSelectShowcaseModule, declarations: [ListMultipleSelectShowcaseComponent], imports: [CommonModule, ListMultipleSelectModule, ReactiveFormsModule] });
ListMultipleSelectShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ListMultipleSelectShowcaseModule, imports: [[CommonModule, ListMultipleSelectModule, ReactiveFormsModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ListMultipleSelectShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ListMultipleSelectModule, ReactiveFormsModule],
                    declarations: [ListMultipleSelectShowcaseComponent],
                    entryComponents: [ListMultipleSelectShowcaseComponent],
                }]
        }] });
