// @backbase/contact-common-ang/esm2015/lib/components/contact-item-details/contact-item-details.module.js
import { NgModule } from '@angular/core';
import { ContactItemDetailsComponent } from './contact-item-details.component';
import { ContactItemDetailsCustomizableDirective } from './contact-item-details.directive';
import { CommonModule } from '@angular/common';
import { ContactFullAddressModule } from '../contact-full-address/contact-full-address.module';
import { HeaderModule } from '@backbase/ui-ang/header';
import { AccountCardViewModule } from '../account-card-view/account-card-view.module';
import * as i0 from "@angular/core";
export class ContactItemDetailsModule {
}
/** @nocollapse */ ContactItemDetailsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactItemDetailsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ ContactItemDetailsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactItemDetailsModule, declarations: [ContactItemDetailsComponent, ContactItemDetailsCustomizableDirective], imports: [CommonModule, HeaderModule, ContactFullAddressModule, AccountCardViewModule], exports: [ContactItemDetailsComponent, ContactItemDetailsCustomizableDirective] });
/** @nocollapse */ ContactItemDetailsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactItemDetailsModule, imports: [[CommonModule, HeaderModule, ContactFullAddressModule, AccountCardViewModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactItemDetailsModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ContactItemDetailsComponent, ContactItemDetailsCustomizableDirective],
                    exports: [ContactItemDetailsComponent, ContactItemDetailsCustomizableDirective],
                    imports: [CommonModule, HeaderModule, ContactFullAddressModule, AccountCardViewModule],
                }]
        }] });
