// @backbase/payment-orders-ang/esm2015/lib/ui/components/table-checkbox/table-checkbox.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TableCheckboxComponent } from './table-checkbox.component';
import * as i0 from "@angular/core";
export class TableCheckboxModule {
}
/** @nocollapse */ TableCheckboxModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableCheckboxModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ TableCheckboxModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableCheckboxModule, declarations: [TableCheckboxComponent], imports: [CommonModule], exports: [TableCheckboxComponent] });
/** @nocollapse */ TableCheckboxModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableCheckboxModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableCheckboxModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [TableCheckboxComponent],
                    imports: [CommonModule],
                    exports: [TableCheckboxComponent],
                }]
        }] });
