// @backbase/payment-orders-ang/esm2015/lib/ui/components/table-checkbox/table-checkbox.component.js
import { Component, ElementRef, Input, ViewChild } from '@angular/core';
import { TableDirective } from '@backbase/ui-ang/table';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/table";
export class TableCheckboxComponent {
    constructor(parentTable) {
        this.parentTable = parentTable;
    }
    get rowItem() {
        return this.row || this.rowId || undefined;
    }
    get isChecked() {
        return this.parentTable.isSelected(this.rowItem);
    }
    onClick(event) {
        event.stopPropagation();
        if (this.checkboxInput) {
            if (event.target !== this.checkboxInput.nativeElement) {
                this.checkboxInput.nativeElement.checked =
                    !this.checkboxInput.nativeElement.checked;
            }
            this.parentTable.toggleItem(this.rowItem, this.checkboxInput.nativeElement.checked);
        }
    }
}
/** @nocollapse */ TableCheckboxComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableCheckboxComponent, deps: [{ token: i1.TableDirective }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ TableCheckboxComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: TableCheckboxComponent, selector: "bb-table-checkbox", inputs: { row: "row", rowId: "rowId" }, viewQueries: [{ propertyName: "checkboxInput", first: true, predicate: ["checkbox"], descendants: true, read: ElementRef, static: true }], ngImport: i0, template: "<div class=\"bb-input-checkbox bb-stack\" (click)=\"onClick($event)\">\n  <input\n    #checkbox\n    type=\"checkbox\"\n    [checked]=\"isChecked\"\n    aria-label=\"select item\"\n    class=\"bb-input-checkbox__input\"\n    i18n-aria-label=\"@@bb-table-checkbox.aria-label\"\n  />\n  <span class=\"bb-input-checkbox__content\"></span>\n</div>\n" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TableCheckboxComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-table-checkbox',
                    templateUrl: './table-checkbox.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.TableDirective }]; }, propDecorators: { row: [{
                type: Input
            }], rowId: [{
                type: Input
            }], checkboxInput: [{
                type: ViewChild,
                args: ['checkbox', { read: ElementRef, static: true }]
            }] } });
