// @backbase/notifications-badge-widget-ang/esm2015/lib/services/notifications-router.service.js
import { Injectable } from '@angular/core';
import { Subject } from 'rxjs';
import * as i0 from "@angular/core";
export class NotificationsRouterService {
    constructor() {
        this.notificationsSettingsOpenedSubject = new Subject();
        /**
         * Settings opened navigation subject
         */
        this.settingsOpenedNavigation$ = this.notificationsSettingsOpenedSubject.asObservable();
        this.notificationRoutingSubject = new Subject();
        /**
         * Notification routing subject
         */
        this.notificationRouting$ = this.notificationRoutingSubject.asObservable();
    }
    /**
     * Emit notification settings navigation
     */
    emitSettingsOpenedNavigation() {
        this.notificationsSettingsOpenedSubject.next();
    }
    /**
     * Emit notification routing
     *
     * @param routing Routing subject
     */
    emitNotificationRouting(routing) {
        this.notificationRoutingSubject.next(routing);
    }
}
/** @nocollapse */ NotificationsRouterService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsRouterService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
/** @nocollapse */ NotificationsRouterService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsRouterService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsRouterService, decorators: [{
            type: Injectable
        }] });
