// @backbase/actions-common-ang/esm2015/src/components/switch-with-loading/switch-with-loading.component.js
import { Component, forwardRef, Input } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/switch";
import * as i2 from "@angular/common";
import * as i3 from "@angular/forms";
export class SwitchWithLoadingComponent {
    constructor() {
        /**
         * Loading indicator presentation marker.
         */
        this.isLoading = false;
        /**
         * ControlValueAccessor on change handler.
         */
        this.onChange = () => { };
        /**
         * ControlValueAccessor on touch handler.
         */
        this.onTouch = () => { };
    }
    /**
     * ControlValueAccessor value setter.
     */
    set value(val) {
        this._value = val;
        this.onChange(val);
        this.onTouch(val);
    }
    /**
     * ControlValueAccessor value getter.
     */
    get value() {
        return this._value;
    }
    /**
     * Method from ControlValueAccessor to write a new value to the element.
     *
     * @param value - new value
     */
    writeValue(value) {
        this._value = value;
    }
    /**
     * Method from ControlValueAccessor to register on change handler.
     *
     * @param fn - on change handler function
     */
    registerOnChange(fn) {
        this.onChange = fn;
    }
    /**
     * Method from ControlValueAccessor to register on touched handler.
     *
     * @param fn - on touched handler function
     */
    registerOnTouched(fn) {
        this.onTouch = fn;
    }
}
SwitchWithLoadingComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: SwitchWithLoadingComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
SwitchWithLoadingComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.14", type: SwitchWithLoadingComponent, selector: "bb-switch-with-loading", inputs: { isLoading: "isLoading", ariaLabel: ["aria-label", "ariaLabel"] }, providers: [
        {
            provide: NG_VALUE_ACCESSOR,
            useExisting: forwardRef(() => SwitchWithLoadingComponent),
            multi: true,
        },
    ], ngImport: i0, template: "<bb-switch-ui\n  [aria-label]=\"ariaLabel\"\n  [(ngModel)]=\"value\"\n  (blur)=\"onTouch()\"\n  *ngIf=\"!isLoading; else loadingIndicator\"\n></bb-switch-ui>\n<ng-template #loadingIndicator>\n  <div class=\"bb-loading-indicator bb-loading-indicator--sm\" data-role=\"loading-indicator\">\n    <svg aria-label=\"Loading indicator\" class=\"bb-loading-indicator__circle\" preserveAspectRatio=\"xMinYMin meet\">\n      <circle class=\"bb-loading-indicator__path\" cx=\"50%\" cy=\"50%\" fill=\"none\" r=\"40%\" stroke=\"currentColor\"></circle>\n    </svg>\n  </div>\n</ng-template>\n", components: [{ type: i1.SwitchComponent, selector: "bb-switch-ui", inputs: ["labelPosition"] }], directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i3.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { type: i3.NgModel, selector: "[ngModel]:not([formControlName]):not([formControl])", inputs: ["name", "disabled", "ngModel", "ngModelOptions"], outputs: ["ngModelChange"], exportAs: ["ngModel"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: SwitchWithLoadingComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-switch-with-loading',
                    templateUrl: './switch-with-loading.component.html',
                    providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            useExisting: forwardRef(() => SwitchWithLoadingComponent),
                            multi: true,
                        },
                    ],
                }]
        }], propDecorators: { isLoading: [{
                type: Input
            }], ariaLabel: [{
                type: Input,
                args: ['aria-label']
            }] } });
