// @backbase/transactions-common-ang/esm2015/lib/services/enum-values.service.js
import { Injectable } from '@angular/core';
import { TransactionClientHttpService } from '@backbase/data-ang/transactions';
import { throwError } from 'rxjs';
import { map, catchError } from 'rxjs/operators';
import { TransactionsEnumValue } from '../model/transactions-enum-value.model';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/transactions";
export class EnumValuesService {
    constructor(transactionsDataHttpService) {
        this.transactionsDataHttpService = transactionsDataHttpService;
        this.enumValues = {};
    }
    handleError(error) {
        if (error.error instanceof ErrorEvent) {
            // A client-side or network error occurred.  Handle it accordingly.
            console.error('An error occurred:', error.error.message);
        }
        else {
            // The backend returned an unsuccessful response code.
            // The response body may contain clues as to what went wrong.
            console.error(`Backend returned code ${error.status}, ` + `body was: ${error.error}`);
        }
        // Return an observable with a user-facing error message.
        return throwError('Something bad happened; please try again later.');
    }
    loadEnumValues(attributeName) {
        return this.transactionsDataHttpService
            .getEnumValuesByAttributeName({ attributeName }, 'response')
            .pipe(map(TransactionsEnumValue.fromHttpResponse), catchError(this.handleError));
    }
    getEnumValues(attributeName) {
        if (typeof this.enumValues[attributeName] === 'undefined') {
            this.enumValues[attributeName] = this.loadEnumValues(attributeName);
        }
        return this.enumValues[attributeName];
    }
}
EnumValuesService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EnumValuesService, deps: [{ token: i1.TransactionClientHttpService }], target: i0.ɵɵFactoryTarget.Injectable });
EnumValuesService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EnumValuesService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EnumValuesService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.TransactionClientHttpService }]; } });
