// @backbase/transactions-common-ang/esm2015/lib/services/messages.service.js
import { Injectable } from '@angular/core';
import { MessagecenterHttpService } from '@backbase/data-ang/messages';
import { of } from 'rxjs';
import { map } from 'rxjs/operators';
import { b64Encode } from '../constants';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/messages";
export class MessagesService {
    constructor(messageCenterHttpService) {
        this.messageCenterHttpService = messageCenterHttpService;
        // TODO: default message is out of scope
        this.defaultMessage = of({
            body: b64Encode('test message'),
            subject: 'no subject',
            topic: 'no topic',
            sender: {
                name: 'test sender',
            },
        });
    }
    saveMessages(message) {
        return this.messageCenterHttpService.postMessages({
            presentationMessagePostRequestBody: message,
        });
    }
    saveAttachment(attachment) {
        return this.messageCenterHttpService.postMessageAttachments({ attachment }).pipe(map((data) => data.id));
    }
    deleteAttachments(attachmentIds) {
        return this.messageCenterHttpService
            .deleteMessageAttachments({ ids: attachmentIds })
            .pipe(map((data) => data.deleted));
    }
    createDefaultBody(transaction) {
        var _a, _b;
        return `transactionID: ${transaction.id}
referenceNumber: ${transaction.reference}
arranagementID: ${transaction.arrangementId}
bookingDate: ${transaction.bookingDate}
amount: ${(_a = transaction.transactionAmountCurrency) === null || _a === void 0 ? void 0 : _a.amount}
merchantName: ${(_b = transaction.merchant) === null || _b === void 0 ? void 0 : _b.name}
counterpartyName: ${transaction.counterPartyName}
description: ${transaction.description}
type: ${transaction.type}`;
    }
}
MessagesService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesService, deps: [{ token: i1.MessagecenterHttpService }], target: i0.ɵɵFactoryTarget.Injectable });
MessagesService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MessagesService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.MessagecenterHttpService }]; } });
