// @backbase/identity-auth/esm2015/impersonation/banner/banner-layout/banner-layout.component.js
import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/mode-header";
import * as i2 from "@backbase/ui-ang/button";
export class ImpersonationBannerLayoutComponent {
    constructor() {
        /**
         * Emits when the impersonation session stop button is clicked.
         */
        this.stop = new EventEmitter();
    }
}
/** @nocollapse */ ImpersonationBannerLayoutComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ImpersonationBannerLayoutComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ ImpersonationBannerLayoutComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: ImpersonationBannerLayoutComponent, selector: "bb-impersonation-banner-layout", inputs: { user: "user" }, outputs: { stop: "stop" }, ngImport: i0, template: "<bb-mode-header-ui\n  i18n-title=\"\n    Header that displays who is being emulated when the banner appears during an emulation session\n    @@bb-impersonation-banner.layout.title\"\n  title=\"Emulating {{ user?.fullName }}\"\n  data-role=\"impersonation-banner\"\n  [showCloseButton]=\"false\"\n>\n  <button\n    mode-header-status\n    bbButton\n    type=\"button\"\n    buttonSize=\"sm\"\n    color=\"danger\"\n    data-role=\"impersonation-banner-stop-button\"\n    i18n=\"Stop button used to end the emulation session @@bb-impersonation-banner.layout.button.stop\"\n    (click)=\"stop.emit()\"\n  >\n    Stop\n  </button>\n</bb-mode-header-ui>\n", components: [{ type: i1.ModeHeaderComponent, selector: "bb-mode-header-ui", inputs: ["title", "secondaryTitle", "showCloseButton", "closeButtonAriaLabel", "reverseTitleOrder"], outputs: ["close"] }], directives: [{ type: i2.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ImpersonationBannerLayoutComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-impersonation-banner-layout',
                    templateUrl: 'banner-layout.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { user: [{
                type: Input
            }], stop: [{
                type: Output
            }] } });
