// @backbase/loans-collection-ui-ang/esm2015/lib/payment-schedule/amortized-payment/loan-amortized-payment-schedule-table/loan-amortized-payment-schedule-table.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AmountModule } from '@backbase/ui-ang/amount';
import { BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import { TableModule } from '@backbase/ui-ang/table';
import { LoanPaymentScheduleStatusModule } from '../../loan-payment-schedule-status/loan-payment-schedule-status.module';
import { LoanAmortizedPaymentScheduleTableComponent } from './loan-amortized-payment-schedule-table.component';
import { AccessibilityKeyboardModule } from '../../../directives/accessibility-keyboard/accessibility-keyboard.module';
import * as i0 from "@angular/core";
export class LoanAmortizedPaymentScheduleTableModule {
}
/** @nocollapse */ LoanAmortizedPaymentScheduleTableModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanAmortizedPaymentScheduleTableModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanAmortizedPaymentScheduleTableModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanAmortizedPaymentScheduleTableModule, declarations: [LoanAmortizedPaymentScheduleTableComponent], imports: [CommonModule,
        TableModule,
        AmountModule,
        BbDatePipeModule,
        LoanPaymentScheduleStatusModule,
        AccessibilityKeyboardModule], exports: [LoanAmortizedPaymentScheduleTableComponent] });
/** @nocollapse */ LoanAmortizedPaymentScheduleTableModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanAmortizedPaymentScheduleTableModule, imports: [[
            CommonModule,
            TableModule,
            AmountModule,
            BbDatePipeModule,
            LoanPaymentScheduleStatusModule,
            AccessibilityKeyboardModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanAmortizedPaymentScheduleTableModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanAmortizedPaymentScheduleTableComponent],
                    imports: [
                        CommonModule,
                        TableModule,
                        AmountModule,
                        BbDatePipeModule,
                        LoanPaymentScheduleStatusModule,
                        AccessibilityKeyboardModule,
                    ],
                    exports: [LoanAmortizedPaymentScheduleTableComponent],
                }]
        }] });
