// @backbase/contact-common-ang/esm2015/lib/components/debounce-search/debounce-search.module.js
import { NgModule } from '@angular/core';
import { DebounceSearchComponent } from './debounce-search.component';
import { FormsModule } from '@angular/forms';
import { SearchBoxModule } from '@backbase/ui-ang/search-box';
import * as i0 from "@angular/core";
/**
 * Exports:
 * - DebounceSearchComponent
 */
export class DebounceSearchModule {
}
/** @nocollapse */ DebounceSearchModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DebounceSearchModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ DebounceSearchModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DebounceSearchModule, declarations: [DebounceSearchComponent], imports: [FormsModule, SearchBoxModule], exports: [DebounceSearchComponent] });
/** @nocollapse */ DebounceSearchModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DebounceSearchModule, imports: [[FormsModule, SearchBoxModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DebounceSearchModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [DebounceSearchComponent],
                    exports: [DebounceSearchComponent],
                    imports: [FormsModule, SearchBoxModule],
                }]
        }] });
