// @backbase/campaign-space-ang/esm2015/src/services/campaign.service.js
import { Injectable, Inject, InjectionToken, Optional } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/engagement-http-ang";
export const CampaignSpaceConfigurationToken = new InjectionToken('CampaignSpaceConfiguration injection token');
const staticConfig = {
    portalName: 'backbase-wc3',
    locale: 'en-US',
    designMode: false,
};
export class CampaignService {
    constructor(selectCampaignService, config) {
        this.selectCampaignService = selectCampaignService;
        this._config = Object.assign(Object.assign({}, staticConfig), config);
    }
    get portalName() {
        return this._config.portalName;
    }
    get locale() {
        return this._config.locale;
    }
    get designMode() {
        return this._config.designMode;
    }
    selectCampaign(widgetName, dimensions, pageName) {
        const spaceId = `${this.portalName}.${pageName}.${widgetName}`;
        return this.selectCampaignService.selectCampaign({ locale: this.locale, dimensions, spaceId });
    }
    b64Decode(str) {
        return decodeURIComponent(atob(str)
            .split('')
            .map(function (c) {
            return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2);
        })
            .join(''));
    }
}
CampaignService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CampaignService, deps: [{ token: i1.SelectCampaignService }, { token: CampaignSpaceConfigurationToken, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
CampaignService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CampaignService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CampaignService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.SelectCampaignService }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [CampaignSpaceConfigurationToken]
                }] }]; } });
