// @backbase/ui-ang/esm2015/account-selector/account-selector.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { NgSelectModule } from '@ng-select/ng-select';
import { ProductItemBasicAccountModule } from '@backbase/ui-ang/product-item-basic-account';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import { InputCheckboxModule } from '@backbase/ui-ang/input-checkbox';
import { SearchBoxModule } from '@backbase/ui-ang/search-box';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { ButtonModule } from '@backbase/ui-ang/button';
import { AccountSelectorComponent, CustomGroupItemsHeaderDirective, CustomLoadingTemplateDirective, CustomMultiSelectedItemsLabelDirective, CustomOptionItemDirective, CustomOptionsHeaderDirective, CustomSingleSelectedItemLabelDirective, } from './account-selector.component';
import { EllipsisModule } from '@backbase/ui-ang/ellipsis';
import * as i0 from "@angular/core";
export class AccountSelectorModule {
}
AccountSelectorModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountSelectorModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AccountSelectorModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountSelectorModule, declarations: [AccountSelectorComponent,
        CustomGroupItemsHeaderDirective,
        CustomLoadingTemplateDirective,
        CustomMultiSelectedItemsLabelDirective,
        CustomOptionItemDirective,
        CustomOptionsHeaderDirective,
        CustomSingleSelectedItemLabelDirective], imports: [CommonModule,
        NgSelectModule,
        FormsModule,
        ProductItemBasicAccountModule,
        InputTextModule,
        InputCheckboxModule,
        SearchBoxModule,
        LoadingIndicatorModule,
        ButtonModule,
        EllipsisModule], exports: [AccountSelectorComponent,
        CustomGroupItemsHeaderDirective,
        CustomLoadingTemplateDirective,
        CustomMultiSelectedItemsLabelDirective,
        CustomOptionItemDirective,
        CustomOptionsHeaderDirective,
        CustomSingleSelectedItemLabelDirective] });
AccountSelectorModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountSelectorModule, imports: [[
            CommonModule,
            NgSelectModule,
            FormsModule,
            ProductItemBasicAccountModule,
            InputTextModule,
            InputCheckboxModule,
            SearchBoxModule,
            LoadingIndicatorModule,
            ButtonModule,
            EllipsisModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountSelectorModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        NgSelectModule,
                        FormsModule,
                        ProductItemBasicAccountModule,
                        InputTextModule,
                        InputCheckboxModule,
                        SearchBoxModule,
                        LoadingIndicatorModule,
                        ButtonModule,
                        EllipsisModule,
                    ],
                    declarations: [
                        AccountSelectorComponent,
                        CustomGroupItemsHeaderDirective,
                        CustomLoadingTemplateDirective,
                        CustomMultiSelectedItemsLabelDirective,
                        CustomOptionItemDirective,
                        CustomOptionsHeaderDirective,
                        CustomSingleSelectedItemLabelDirective,
                    ],
                    exports: [
                        AccountSelectorComponent,
                        CustomGroupItemsHeaderDirective,
                        CustomLoadingTemplateDirective,
                        CustomMultiSelectedItemsLabelDirective,
                        CustomOptionItemDirective,
                        CustomOptionsHeaderDirective,
                        CustomSingleSelectedItemLabelDirective,
                    ],
                }]
        }] });
