// @backbase/messages-common/esm2015/src/components/conversation-thread/base-messages-conversation-thread.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BaseConversationThreadComponent } from './component/base-conversation-thread.component';
import { BaseConversationThreadDraftsComponent } from './component/base-conversation-thread-drafts.component';
import * as i0 from "@angular/core";
export class BaseMessagesConversationThreadModule {
}
BaseMessagesConversationThreadModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: BaseMessagesConversationThreadModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
BaseMessagesConversationThreadModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: BaseMessagesConversationThreadModule, declarations: [BaseConversationThreadComponent, BaseConversationThreadDraftsComponent], imports: [CommonModule], exports: [BaseConversationThreadComponent, BaseConversationThreadDraftsComponent] });
BaseMessagesConversationThreadModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: BaseMessagesConversationThreadModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: BaseMessagesConversationThreadModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [BaseConversationThreadComponent, BaseConversationThreadDraftsComponent],
                    imports: [CommonModule],
                    exports: [BaseConversationThreadComponent, BaseConversationThreadDraftsComponent],
                }]
        }] });
