// @backbase/account-statement-common-ang/esm2015/src/components/account-statement-pagination/account-statement-pagination.component.js
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { PaginationType } from '../../models/pagination-type.model';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/pagination";
import * as i2 from "@backbase/ui-ang/load-button";
import * as i3 from "@angular/common";
export class AccountStatementPaginationComponent {
    constructor() {
        /**
         * Event (Output) that emits when page prop is changed
         */
        this.pageChange = new EventEmitter();
        /**
         * Event (Output) that emits when load-more
         */
        this.loadMore = new EventEmitter();
        this.paginationTypeEnum = PaginationType;
    }
    onPageChange(page) {
        this.pageChange.emit(page);
    }
    onLoadMore(loadMoreParams) {
        if (!this.loading) {
            this.loadMore.emit(loadMoreParams.from);
        }
    }
}
AccountStatementPaginationComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementPaginationComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
AccountStatementPaginationComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: AccountStatementPaginationComponent, selector: "bb-account-statement-pagination", inputs: { page: "page", from: "from", count: "count", itemsPerPage: "itemsPerPage", paginationType: "paginationType", hasMore: "hasMore", loading: "loading" }, outputs: { pageChange: "pageChange", loadMore: "loadMore" }, ngImport: i0, template: "<div class=\"bb-block bb-block--xl bb-text-align-center\">\n  <bb-pagination-ui\n    *ngIf=\"paginationType === paginationTypeEnum.Pagination && (hasMore || (!!page && page > 0))\"\n    [page]=\"page || 0\"\n    [totalItems]=\"count || 0\"\n    [itemsPerPage]=\"itemsPerPage || 0\"\n    [boundaryLinks]=\"true\"\n    [directionLinks]=\"true\"\n    [maxNavPages]=\"3\"\n    (pageChange)=\"onPageChange($event)\"\n  >\n  </bb-pagination-ui>\n\n  <bb-load-button-ui\n    color=\"link\"\n    *ngIf=\"paginationType === paginationTypeEnum.LoadMore\"\n    [from]=\"page || 0\"\n    [size]=\"itemsPerPage || 0\"\n    [totalCount]=\"count || 0\"\n    [isLoading]=\"loading || false\"\n    (paramChange)=\"onLoadMore($event)\"\n    data-role=\"load-more-button\"\n  >\n    <span i18n=\"Load more|Load More button text@@accountStatement.list.load-more\">Load more</span>\n  </bb-load-button-ui>\n</div>\n", components: [{ type: i1.PaginatorComponent, selector: "bb-pagination-ui", inputs: ["page", "boundaryLinks", "directionLinks", "totalItems", "itemsPerPage", "maxNavPages"], outputs: ["pageChange"] }, { type: i2.LoadButtonComponent, selector: "bb-load-button-ui", inputs: ["isLoading", "from", "size", "circle", "totalCount"], outputs: ["paramChange"] }], directives: [{ type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementPaginationComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-account-statement-pagination',
                    templateUrl: './account-statement-pagination.component.html',
                }]
        }], propDecorators: { page: [{
                type: Input
            }], from: [{
                type: Input
            }], count: [{
                type: Input
            }], itemsPerPage: [{
                type: Input
            }], paginationType: [{
                type: Input
            }], hasMore: [{
                type: Input
            }], loading: [{
                type: Input
            }], pageChange: [{
                type: Output
            }], loadMore: [{
                type: Output
            }] } });
