// @backbase/ui-ang/esm2015/_example/keyboard-click-directive/examples/keyboard-click.example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/notification";
import * as i2 from "@backbase/ui-ang/button";
import * as i3 from "@backbase/ui-ang/keyboard-click-directive";
export class KeyboardClickExampleComponent {
    constructor(notificationService) {
        this.notificationService = notificationService;
    }
    action() {
        this.notificationService.showNotification({
            message: 'bbKeyboardClick event',
            modifier: 'info',
        });
    }
}
KeyboardClickExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: KeyboardClickExampleComponent, deps: [{ token: i1.NotificationService }], target: i0.ɵɵFactoryTarget.Component });
KeyboardClickExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: KeyboardClickExampleComponent, selector: "bb-keyboard-click-example-ui", ngImport: i0, template: "<button bbButton (bbKeyboardClick)=\"action()\">Click Me!</button>\n", directives: [{ type: i2.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }, { type: i3.KeyboardClickDirective, selector: "[bbKeyboardClick]", outputs: ["bbKeyboardClick"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: KeyboardClickExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-keyboard-click-example-ui',
                    templateUrl: './keyboard-click.example.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.NotificationService }]; } });
