// @backbase/ui-ang/esm2015/value-diff/bb-value-diff.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ValueDiffDirective } from './bb-value-diff.directive';
import * as i0 from "@angular/core";
export class ValueDiffDirectiveModule {
}
ValueDiffDirectiveModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ValueDiffDirectiveModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ValueDiffDirectiveModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ValueDiffDirectiveModule, declarations: [ValueDiffDirective], imports: [CommonModule], exports: [ValueDiffDirective] });
ValueDiffDirectiveModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ValueDiffDirectiveModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ValueDiffDirectiveModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [ValueDiffDirective],
                    exports: [ValueDiffDirective],
                }]
        }] });
