// @backbase/messages-conversation-thread-widget-ang/esm2015/src/messages-conversation-thread-widget-ang.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { BaseMessagesConversationThreadModule, MessagesConversationMessageModule, MessagesManipulationConfirmModalModule, MessagesRequestErrorModule, MessagesUploadAttachmentsModule, } from '@backbase/messages-common';
import { AlertModule } from '@backbase/ui-ang/alert';
import { BadgeModule } from '@backbase/ui-ang/badge';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { IconModule } from '@backbase/ui-ang/icon';
import { ButtonModule } from '@backbase/ui-ang/button';
import { ModalModule } from '@backbase/ui-ang/modal';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { RichTextEditorModule } from '@backbase/ui-ang/rich-text-editor';
import { MessagesConversationThreadCustomizableDirective, MessagesConversationThreadWidgetComponent, } from './messages-conversation-thread-widget-ang.component';
import { MessagesConversationThreadWidgetService } from './services/messages-conversation-thread-widget-ang.service';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const components = [MessagesConversationThreadWidgetComponent];
const directives = [MessagesConversationThreadCustomizableDirective];
const uiModules = [
    ButtonModule,
    BadgeModule,
    IconModule,
    RichTextEditorModule,
    ModalModule,
    EmptyStateModule,
    LoadingIndicatorModule,
    TooltipModule,
    LoadButtonModule,
    AlertModule,
    InputValidationMessageModule,
];
const innerModules = [
    BaseMessagesConversationThreadModule,
    MessagesManipulationConfirmModalModule,
    MessagesUploadAttachmentsModule,
    MessagesRequestErrorModule,
    MessagesConversationMessageModule,
];
export class MessagesConversationThreadWidgetModule {
}
MessagesConversationThreadWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesConversationThreadWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MessagesConversationThreadWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesConversationThreadWidgetModule, declarations: [MessagesConversationThreadWidgetComponent, MessagesConversationThreadCustomizableDirective], imports: [CommonModule, i1.BackbaseCoreModule, ReactiveFormsModule, BaseMessagesConversationThreadModule,
        MessagesManipulationConfirmModalModule,
        MessagesUploadAttachmentsModule,
        MessagesRequestErrorModule,
        MessagesConversationMessageModule, ButtonModule,
        BadgeModule,
        IconModule,
        RichTextEditorModule,
        ModalModule,
        EmptyStateModule,
        LoadingIndicatorModule,
        TooltipModule,
        LoadButtonModule,
        AlertModule,
        InputValidationMessageModule], exports: [ReactiveFormsModule, MessagesConversationThreadWidgetComponent, MessagesConversationThreadCustomizableDirective, ButtonModule,
        BadgeModule,
        IconModule,
        RichTextEditorModule,
        ModalModule,
        EmptyStateModule,
        LoadingIndicatorModule,
        TooltipModule,
        LoadButtonModule,
        AlertModule,
        InputValidationMessageModule, BaseMessagesConversationThreadModule,
        MessagesManipulationConfirmModalModule,
        MessagesUploadAttachmentsModule,
        MessagesRequestErrorModule,
        MessagesConversationMessageModule] });
MessagesConversationThreadWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesConversationThreadWidgetModule, providers: [MessagesConversationThreadWidgetService], imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { MessagesConversationThreadWidgetComponent },
            }),
            ReactiveFormsModule,
            ...innerModules,
            ...uiModules,
        ], ReactiveFormsModule, ButtonModule,
        BadgeModule,
        IconModule,
        RichTextEditorModule,
        ModalModule,
        EmptyStateModule,
        LoadingIndicatorModule,
        TooltipModule,
        LoadButtonModule,
        AlertModule,
        InputValidationMessageModule, BaseMessagesConversationThreadModule,
        MessagesManipulationConfirmModalModule,
        MessagesUploadAttachmentsModule,
        MessagesRequestErrorModule,
        MessagesConversationMessageModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesConversationThreadWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { MessagesConversationThreadWidgetComponent },
                        }),
                        ReactiveFormsModule,
                        ...innerModules,
                        ...uiModules,
                    ],
                    declarations: [...components, ...directives],
                    providers: [MessagesConversationThreadWidgetService],
                    exports: [ReactiveFormsModule, ...components, ...directives, ...uiModules, ...innerModules],
                }]
        }] });
