// @backbase/foundation-ang/esm2015/core/backbase-core/models/array.js
export const map = (callbackfn, arr) => arr.map(callbackfn);
export const flatten = (arr) => arr.reduce((acc, a) => [...acc, ...a], []);
export const flatMap = (callbackfn, arr) => flatten(map(callbackfn, arr));
export const uniq = (array) => Array.from(new Set(array));
export const hasDuplicates = (array) => array.length !== uniq(array).length;
/**
 * Creates a new list out of the two supplied by pairing up equally-positioned
 * items from both lists. The returned list is the length of the longer of two
 * lists. If one of the lists is shorter, the values will be set to "undefined".
 *
 * @param left
 * @param right
 */
export const pairs = (left, right) => {
    const max = Math.max(left.length, right.length);
    let idx = 0;
    const returnZip = [];
    while (idx < max) {
        returnZip[idx] = [left[idx], right[idx]];
        idx++;
    }
    return returnZip;
};
/**
 * This helper checks if the value is array of strings.
 * @param value a value to be evaluated
 * @returns true for array of strings and false for any other values
 */
export const isArrayOfStrings = (value) => Array.isArray(value) && value.every(item => typeof item === 'string');
