// @backbase/foundation-ang/esm2015/core/backbase-core/models/item-model.js
import { distinctUntilChanged, map, pluck, scan, shareReplay, startWith } from 'rxjs/operators';
import { equal, propertiesByPrefix, propertiesAsPairs } from '../../web-sdk/properties';
import { ObservableTree } from './tree';
/**
 * @deprecated Will be removed in v7.0.0
 */
export class ItemModel {
    constructor(name, initialProperties, propertyUpdates) {
        this.name = name;
        const applyPropertyUpdate = (currentProperties, update) => {
            return Object.assign(Object.assign({}, currentProperties), update);
        };
        this.properties = propertyUpdates.pipe(startWith(initialProperties), scan(applyPropertyUpdate), shareReplay(1));
        if (initialProperties.classId === undefined) {
            throw new Error(`Missing classId property for ${name}`);
        }
        this.ɵclassId = initialProperties.classId;
        this.ɵoutputProperties = this.outputProperties(initialProperties);
        this.ɵinputProperties = this.inputProperties(initialProperties);
        this.ɵproperties = this.nonIOProperties(initialProperties);
    }
    property(propertyName, defaultValue) {
        return this.properties.pipe(pluck(propertyName), map(propertyValue => (propertyValue === undefined || propertyValue === null ? defaultValue : propertyValue)), distinctUntilChanged((a, b) => {
            if (a === undefined && b === undefined) {
                return true;
            }
            if (a === undefined) {
                return false;
            }
            if (b === undefined) {
                return false;
            }
            return equal(a, b);
        }));
    }
    outputProperties(properties) {
        return propertiesByPrefix('output.', properties);
    }
    inputProperties(properties) {
        return propertiesByPrefix('input.', properties);
    }
    nonIOProperties(properties) {
        return propertiesAsPairs(pairs => pairs.filter(([name]) => !name.startsWith('input.') && !name.startsWith('output.')), properties);
    }
}
/**
 * @deprecated Will be removed in v7.0.0
 */
export class ItemModelTree extends ObservableTree {
}
