// @backbase/actions-product-notifications-settings-widget-ang/esm2015/src/actions-product-notifications-settings-widget.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ActionsCommonModule } from '@backbase/actions-common-ang';
import { ActionsStoreModule } from '@backbase/actions-store-ang';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { ButtonModule } from '@backbase/ui-ang/button';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { BbIbanPipeModule } from '@backbase/ui-ang/iban-pipe';
import { AmountModule } from '@backbase/ui-ang/amount';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import { ProductItemBasicAccountModule } from '@backbase/ui-ang/product-item-basic-account';
import { ProductItemCreditCardModule } from '@backbase/ui-ang/product-item-credit-card';
import { ProductItemCurrentAccountModule } from '@backbase/ui-ang/product-item-current-account';
import { ProductItemDebitCardModule } from '@backbase/ui-ang/product-item-debit-card';
import { ProductItemInvestmentAccountModule } from '@backbase/ui-ang/product-item-investment-account';
import { ProductItemLoanModule } from '@backbase/ui-ang/product-item-loan';
import { ProductItemSavingsAccountModule } from '@backbase/ui-ang/product-item-savings-account';
import { ProductItemTermDepositModule } from '@backbase/ui-ang/product-item-term-deposit';
import { ActionsProductItemCardComponent, ActionsProductItemCardCustomizableDirective, } from './components/actions-product-item-card/actions-product-item-card.component';
import { ActionsProductNotificationsSettingsWidgetComponent } from './components/actions-product-notifications-settings-widget/actions-product-notifications-settings-widget.component';
import { ActionsProductSettingsPageComponent, ActionsProductSettingsPageCustomizableDirective, ActionsProductSettingsPageErrorNotificationHeaderCustomizableDirective, ActionsProductSettingsPageErrorNotificationMessageCustomizableDirective, } from './components/pages/actions-product-settings-page/actions-product-settings-page.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const pages = [ActionsProductSettingsPageComponent];
const components = [
    ...pages,
    ActionsProductNotificationsSettingsWidgetComponent,
    ActionsProductSettingsPageComponent,
    ActionsProductItemCardComponent,
];
const directives = [
    ActionsProductItemCardCustomizableDirective,
    ActionsProductSettingsPageCustomizableDirective,
    ActionsProductSettingsPageErrorNotificationHeaderCustomizableDirective,
    ActionsProductSettingsPageErrorNotificationMessageCustomizableDirective,
];
const uiModules = [
    EmptyStateModule,
    LoadingIndicatorModule,
    BbIbanPipeModule,
    ProductItemCurrentAccountModule,
    AmountModule,
    ButtonModule,
    ProductItemSavingsAccountModule,
    ProductItemTermDepositModule,
    ProductItemLoanModule,
    ProductItemCreditCardModule,
    ProductItemDebitCardModule,
    ProductItemInvestmentAccountModule,
    ProductItemBasicAccountModule,
    PaymentCardNumberModule,
];
export class ActionsProductNotificationsSettingsWidgetModule {
}
ActionsProductNotificationsSettingsWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsProductNotificationsSettingsWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ActionsProductNotificationsSettingsWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsProductNotificationsSettingsWidgetModule, declarations: [ActionsProductSettingsPageComponent, ActionsProductNotificationsSettingsWidgetComponent,
        ActionsProductSettingsPageComponent,
        ActionsProductItemCardComponent, ActionsProductItemCardCustomizableDirective,
        ActionsProductSettingsPageCustomizableDirective,
        ActionsProductSettingsPageErrorNotificationHeaderCustomizableDirective,
        ActionsProductSettingsPageErrorNotificationMessageCustomizableDirective], imports: [CommonModule,
        ActionsCommonModule,
        ActionsStoreModule, i1.BackbaseCoreModule, EmptyStateModule,
        LoadingIndicatorModule,
        BbIbanPipeModule,
        ProductItemCurrentAccountModule,
        AmountModule,
        ButtonModule,
        ProductItemSavingsAccountModule,
        ProductItemTermDepositModule,
        ProductItemLoanModule,
        ProductItemCreditCardModule,
        ProductItemDebitCardModule,
        ProductItemInvestmentAccountModule,
        ProductItemBasicAccountModule,
        PaymentCardNumberModule], exports: [ActionsProductSettingsPageComponent, ActionsProductNotificationsSettingsWidgetComponent,
        ActionsProductSettingsPageComponent,
        ActionsProductItemCardComponent, ActionsProductItemCardCustomizableDirective,
        ActionsProductSettingsPageCustomizableDirective,
        ActionsProductSettingsPageErrorNotificationHeaderCustomizableDirective,
        ActionsProductSettingsPageErrorNotificationMessageCustomizableDirective, EmptyStateModule,
        LoadingIndicatorModule,
        BbIbanPipeModule,
        ProductItemCurrentAccountModule,
        AmountModule,
        ButtonModule,
        ProductItemSavingsAccountModule,
        ProductItemTermDepositModule,
        ProductItemLoanModule,
        ProductItemCreditCardModule,
        ProductItemDebitCardModule,
        ProductItemInvestmentAccountModule,
        ProductItemBasicAccountModule,
        PaymentCardNumberModule] });
ActionsProductNotificationsSettingsWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsProductNotificationsSettingsWidgetModule, imports: [[
            CommonModule,
            ActionsCommonModule,
            ActionsStoreModule,
            BackbaseCoreModule.withConfig({
                classMap: { ActionsProductNotificationsSettingsWidgetComponent },
            }),
            ...uiModules,
        ], EmptyStateModule,
        LoadingIndicatorModule,
        BbIbanPipeModule,
        ProductItemCurrentAccountModule,
        AmountModule,
        ButtonModule,
        ProductItemSavingsAccountModule,
        ProductItemTermDepositModule,
        ProductItemLoanModule,
        ProductItemCreditCardModule,
        ProductItemDebitCardModule,
        ProductItemInvestmentAccountModule,
        ProductItemBasicAccountModule,
        PaymentCardNumberModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: ActionsProductNotificationsSettingsWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components, ...directives],
                    imports: [
                        CommonModule,
                        ActionsCommonModule,
                        ActionsStoreModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { ActionsProductNotificationsSettingsWidgetComponent },
                        }),
                        ...uiModules,
                    ],
                    entryComponents: [...pages],
                    exports: [...components, ...directives, ...uiModules],
                }]
        }] });
