// @backbase/loans-collection-ui-ang/esm2015/lib/loan-list-item/loan-list-item.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AmountModule } from '@backbase/ui-ang/amount';
import { BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import { IconModule } from '@backbase/ui-ang/icon';
import { KeyboardClickModule } from '@backbase/ui-ang/keyboard-click-directive';
import { LoanItemTypeModule } from './loan-item-type/loan-item-type.module';
import { LoanItemStatusModule } from './loan-item-status/loan-item-status.module';
import { LoanItemNameModule } from './loan-item-name/loan-item-name.module';
import { LoanListItemComponent } from './loan-list-item.component';
import { LoanItemPaymentInfoModule } from './loan-item-payment-info/loan-item-payment-info.module';
import * as i0 from "@angular/core";
const innerModules = [LoanItemTypeModule, LoanItemStatusModule, LoanItemNameModule, LoanItemPaymentInfoModule];
const uiModules = [IconModule, BbDatePipeModule, AmountModule, KeyboardClickModule];
export class LoanListItemModule {
}
/** @nocollapse */ LoanListItemModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanListItemModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanListItemModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanListItemModule, declarations: [LoanListItemComponent], imports: [CommonModule, IconModule, BbDatePipeModule, AmountModule, KeyboardClickModule, LoanItemTypeModule, LoanItemStatusModule, LoanItemNameModule, LoanItemPaymentInfoModule], exports: [LoanListItemComponent] });
/** @nocollapse */ LoanListItemModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanListItemModule, imports: [[CommonModule, ...uiModules, ...innerModules]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanListItemModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanListItemComponent],
                    imports: [CommonModule, ...uiModules, ...innerModules],
                    exports: [LoanListItemComponent],
                }]
        }] });
