// @backbase/identity-self-service-journey-ang/esm2015/lib/features/user-localization/user-localization.component.js
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { UserLocalizationService } from './user-localization.service';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
export class UserLocalizationComponent {
}
/** @nocollapse */ UserLocalizationComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserLocalizationComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ UserLocalizationComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: UserLocalizationComponent, selector: "bb-user-localization", providers: [UserLocalizationService], ngImport: i0, template: `<router-outlet></router-outlet>`, isInline: true, directives: [{ type: i1.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserLocalizationComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-user-localization',
                    template: `<router-outlet></router-outlet>`,
                    providers: [UserLocalizationService],
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }] });
