// @backbase/messages-client-inbox-journey-ang/esm2015/src/components/messages-client-inbox-thread/messages-client-inbox-thread.component.js
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { BehaviorSubject, Subject } from 'rxjs';
import { distinctUntilChanged, takeUntil } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "../../messages-client-inbox-journey-config.service";
import * as i2 from "@angular/router";
import * as i3 from "@backbase/messages-conversation-thread-widget-ang";
import * as i4 from "@angular/common";
export class MessagesClientInboxThreadComponent {
    constructor(config, route, router) {
        var _a;
        this.config = config;
        this.route = route;
        this.router = router;
        this.conversationId$$ = new BehaviorSubject('');
        /**
         * Selected conversation id stream
         */
        this.conversationId$ = this.conversationId$$.pipe(distinctUntilChanged());
        this.destroy$$ = new Subject();
        const routeConfig = this.route.snapshot.routeConfig;
        this.mailboxTypeTitle = (_a = routeConfig === null || routeConfig === void 0 ? void 0 : routeConfig.data) === null || _a === void 0 ? void 0 : _a.mailboxTypeTitle;
    }
    ngOnInit() {
        this.route.params.pipe(takeUntil(this.destroy$$)).subscribe(({ id }) => {
            this.conversationId$$.next(id || '');
        });
    }
    /**
     * Navigates back
     */
    navigateBack() {
        this.router.navigate([`../`], { relativeTo: this.route });
    }
    ngOnDestroy() {
        this.destroy$$.next();
    }
}
MessagesClientInboxThreadComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesClientInboxThreadComponent, deps: [{ token: i1.MessagesClientInboxJourneyConfigService }, { token: i2.ActivatedRoute }, { token: i2.Router }], target: i0.ɵɵFactoryTarget.Component });
MessagesClientInboxThreadComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.14", type: MessagesClientInboxThreadComponent, selector: "bb-messages-client-inbox-thread", ngImport: i0, template: "<bb-messages-conversation-thread-widget-ang\n  [selectedConversationId]=\"conversationId$ | async\"\n  [customerServiceTitle]=\"config.customerServiceTitle\"\n  [replyMessageMaxLength]=\"config.replyMessageMaxLength\"\n  [maxAttachmentSize]=\"config.maxAttachmentSize\"\n  [mailboxTypeTitle]=\"mailboxTypeTitle\"\n  (back)=\"navigateBack()\"\n></bb-messages-conversation-thread-widget-ang>\n", components: [{ type: i3.MessagesConversationThreadWidgetComponent, selector: "bb-messages-conversation-thread-widget-ang", inputs: ["mailboxTypeTitle", "maxAttachmentSize", "replyMessageMaxLength", "customerServiceTitle"] }], pipes: { "async": i4.AsyncPipe }, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: MessagesClientInboxThreadComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-messages-client-inbox-thread',
                    templateUrl: './messages-client-inbox-thread.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.MessagesClientInboxJourneyConfigService }, { type: i2.ActivatedRoute }, { type: i2.Router }]; } });
