// @backbase/ui-ang/esm2015/_example/input-phone/showcase/input-phone-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { InputPhoneShowcaseComponent } from './input-phone-showcase.component';
import { InputPhoneModule } from '@backbase/ui-ang/input-phone';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { ReactiveFormsModule } from '@angular/forms';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import { PhoneNumberModule } from '@backbase/ui-ang/phone-number-format-pipe';
import * as i0 from "@angular/core";
export class InputPhoneShowcaseModule {
}
InputPhoneShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputPhoneShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputPhoneShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputPhoneShowcaseModule, declarations: [InputPhoneShowcaseComponent], imports: [CommonModule,
        InputPhoneModule,
        InputTextModule,
        InputValidationMessageModule,
        ReactiveFormsModule,
        PhoneNumberModule] });
InputPhoneShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputPhoneShowcaseModule, imports: [[
            CommonModule,
            InputPhoneModule,
            InputTextModule,
            InputValidationMessageModule,
            ReactiveFormsModule,
            PhoneNumberModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputPhoneShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        InputPhoneModule,
                        InputTextModule,
                        InputValidationMessageModule,
                        ReactiveFormsModule,
                        PhoneNumberModule,
                    ],
                    declarations: [InputPhoneShowcaseComponent],
                    entryComponents: [InputPhoneShowcaseComponent],
                }]
        }] });
