// @backbase/foundation-ang/esm2015/core/backbase-core/session-timeout/session-timeout.service.js
import { ComponentFactoryResolver, Injectable, Inject, InjectionToken, } from '@angular/core';
export const SESSION_TIMEOUT_CONFIG = new InjectionToken('Custom session timeout configuration injection token');
/**
 * The session timeout service creates a session timeout component instance within the root.
 */
export class SessionTimeoutService {
    constructor(resolver, sessionTimeoutConfig) {
        this.resolver = resolver;
        this.sessionTimeoutConfig = sessionTimeoutConfig;
    }
    get sessionTimeoutComponent() {
        if (this.sessionTimeoutComponentRef) {
            return this.sessionTimeoutComponentRef;
        }
        throw new Error('No sessionTimeoutComponent defined');
    }
    createSessionTimeoutComponent(sessionTimeoutContainer) {
        sessionTimeoutContainer.clear();
        const factory = this.resolver.resolveComponentFactory(this.sessionTimeoutConfig.sessionTimeoutComponentClass);
        this.sessionTimeoutComponentRef = sessionTimeoutContainer.createComponent(factory);
        this.sessionTimeoutComponentRef.instance.inactivityModalTime = this.sessionTimeoutConfig.inactivityModalTime;
    }
}
SessionTimeoutService.decorators = [
    { type: Injectable }
];
SessionTimeoutService.ctorParameters = () => [
    { type: ComponentFactoryResolver },
    { type: undefined, decorators: [{ type: Inject, args: [SESSION_TIMEOUT_CONFIG,] }] }
];
