// @backbase/foundation-ang/esm2015/core/backbase-core/session-timeout/session-timeout.module.js
import { NgModule, ANALYZE_FOR_ENTRY_COMPONENTS, ComponentFactoryResolver } from '@angular/core';
import { SessionTimeoutService, SESSION_TIMEOUT_CONFIG } from './session-timeout.service';
export class SessionTimeoutModule {
    /**
     * Creates a configured SessionTimeoutModule.
     *
     * This should be called only once per app.
     *
     * @usageNotes
     *
     * ### Example
     *
     * ```
     * @NgModule({
     *   imports: [
     *     SessionTimeoutModule.forRoot({
     *       classMap: {
     *         sessionTimeoutComponentClass: MyCustomSessionTimeoutComponent,
     *         inactivityModalTime: 300,
     *       }
     *     })
     *   ]
     * })
     * class AppModule {}
     * ```
     */
    static forRoot(config) {
        return {
            ngModule: SessionTimeoutModule,
            providers: [
                {
                    provide: ANALYZE_FOR_ENTRY_COMPONENTS,
                    useValue: config.sessionTimeoutComponentClass,
                    multi: true,
                },
                {
                    provide: SESSION_TIMEOUT_CONFIG,
                    useValue: config,
                },
                {
                    provide: SessionTimeoutService,
                    useClass: SessionTimeoutService,
                    deps: [ComponentFactoryResolver, SESSION_TIMEOUT_CONFIG],
                },
            ],
        };
    }
}
SessionTimeoutModule.decorators = [
    { type: NgModule, args: [{},] }
];
