// @backbase/ui-ang/esm2015/textarea/textarea.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { BaseClassesModule } from '@backbase/ui-ang/base-classes';
import { CharCounterModule } from '@backbase/ui-ang/char-counter';
import { TextareaComponent } from './textarea.component';
import { FocusModule } from '@backbase/ui-ang/focus';
import * as i0 from "@angular/core";
export class TextareaModule {
}
TextareaModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TextareaModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
TextareaModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TextareaModule, declarations: [TextareaComponent], imports: [CommonModule, FormsModule, BaseClassesModule, CharCounterModule, FocusModule], exports: [TextareaComponent] });
TextareaModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TextareaModule, imports: [[CommonModule, FormsModule, BaseClassesModule, CharCounterModule, FocusModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TextareaModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, FormsModule, BaseClassesModule, CharCounterModule, FocusModule],
                    declarations: [TextareaComponent],
                    exports: [TextareaComponent],
                }]
        }] });
