// @backbase/ui-ang/esm2015/_example/stepper/showcase/stepper-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { StepperShowcaseComponent } from './stepper-showcase.component';
import { StepperModule } from '@backbase/ui-ang/stepper';
import * as i0 from "@angular/core";
export class StepperShowcaseModule {
}
StepperShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StepperShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
StepperShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StepperShowcaseModule, declarations: [StepperShowcaseComponent], imports: [CommonModule, StepperModule] });
StepperShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StepperShowcaseModule, imports: [[CommonModule, StepperModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: StepperShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, StepperModule],
                    declarations: [StepperShowcaseComponent],
                    entryComponents: [StepperShowcaseComponent],
                }]
        }] });
