// @backbase/user-manage-profile-widget-ang/esm2015/src/user-manage-postal-addresses/user-manage-postal-addresses.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { BadgeModule } from '@backbase/ui-ang/badge';
import { ButtonModule } from '@backbase/ui-ang/button';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { HeaderModule } from '@backbase/ui-ang/header';
import { IconModule } from '@backbase/ui-ang/icon';
import { NotificationModule } from '@backbase/ui-ang/notification';
import { UserProfileFormsModule } from '../user-profile-forms/user-profile-forms.module';
import { UserEditPostalAddressComponent, UserEditPostalAddressCustomizableDirective, UserEditPostalAddressTitleCustomizableDirective, } from './user-edit-postal-address/user-edit-postal-address.component';
import { UserManagePostalAddressComponent, UserManagePostalAddressCustomizableDirective, } from './user-manage-postal-address.component';
import { UserManagePostalAddressesComponent, UserManagePostalAddressesCustomizableDirective, UserManagePostalAddressesTitleCustomizableDirective, } from './user-manage-postal-addresses.component';
import { UserNewPostalAddressComponent, UserNewPostalAddressCustomizableDirective, UserNewPostalAddressTitleCustomizableDirective, } from './user-new-postal-address/user-new-postal-address.component';
import * as i0 from "@angular/core";
const uiModules = [ButtonModule, EmptyStateModule, BadgeModule, IconModule, NotificationModule, HeaderModule];
const componentModules = [UserProfileFormsModule];
export class UserManagePostalAddressesModule {
}
UserManagePostalAddressesModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserManagePostalAddressesModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
UserManagePostalAddressesModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserManagePostalAddressesModule, declarations: [UserManagePostalAddressComponent,
        UserManagePostalAddressCustomizableDirective,
        UserManagePostalAddressesComponent,
        UserManagePostalAddressesCustomizableDirective,
        UserManagePostalAddressesTitleCustomizableDirective,
        UserNewPostalAddressComponent,
        UserNewPostalAddressCustomizableDirective,
        UserNewPostalAddressTitleCustomizableDirective,
        UserEditPostalAddressComponent,
        UserEditPostalAddressCustomizableDirective,
        UserEditPostalAddressTitleCustomizableDirective], imports: [CommonModule, UserProfileFormsModule, ButtonModule, EmptyStateModule, BadgeModule, IconModule, NotificationModule, HeaderModule], exports: [UserManagePostalAddressComponent,
        UserManagePostalAddressCustomizableDirective,
        UserManagePostalAddressesComponent,
        UserManagePostalAddressesCustomizableDirective,
        UserManagePostalAddressesTitleCustomizableDirective,
        UserNewPostalAddressComponent,
        UserNewPostalAddressCustomizableDirective,
        UserNewPostalAddressTitleCustomizableDirective,
        UserEditPostalAddressComponent,
        UserEditPostalAddressCustomizableDirective,
        UserEditPostalAddressTitleCustomizableDirective, ButtonModule, EmptyStateModule, BadgeModule, IconModule, NotificationModule, HeaderModule, UserProfileFormsModule] });
UserManagePostalAddressesModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserManagePostalAddressesModule, imports: [[CommonModule, ...componentModules, ...uiModules], ButtonModule, EmptyStateModule, BadgeModule, IconModule, NotificationModule, HeaderModule, UserProfileFormsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserManagePostalAddressesModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...componentModules, ...uiModules],
                    declarations: [
                        UserManagePostalAddressComponent,
                        UserManagePostalAddressCustomizableDirective,
                        UserManagePostalAddressesComponent,
                        UserManagePostalAddressesCustomizableDirective,
                        UserManagePostalAddressesTitleCustomizableDirective,
                        UserNewPostalAddressComponent,
                        UserNewPostalAddressCustomizableDirective,
                        UserNewPostalAddressTitleCustomizableDirective,
                        UserEditPostalAddressComponent,
                        UserEditPostalAddressCustomizableDirective,
                        UserEditPostalAddressTitleCustomizableDirective,
                    ],
                    exports: [
                        UserManagePostalAddressComponent,
                        UserManagePostalAddressCustomizableDirective,
                        UserManagePostalAddressesComponent,
                        UserManagePostalAddressesCustomizableDirective,
                        UserManagePostalAddressesTitleCustomizableDirective,
                        UserNewPostalAddressComponent,
                        UserNewPostalAddressCustomizableDirective,
                        UserNewPostalAddressTitleCustomizableDirective,
                        UserEditPostalAddressComponent,
                        UserEditPostalAddressCustomizableDirective,
                        UserEditPostalAddressTitleCustomizableDirective,
                        ...uiModules,
                        ...componentModules,
                    ],
                }]
        }] });
