// @backbase/authorized-users-journey-ang/esm2015/src/views/authorized-users-list.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "@backbase/ui-ang/header";
import * as i3 from "../authorized-users-list/components/authorized-users-list-wrapper/authorized-users-list-wrapper.component";
export class AuthorizedUsersListViewComponent {
    constructor(router, route) {
        this.router = router;
        this.route = route;
    }
    onUserSelected(id) {
        this.router.navigate(['../create', id ? { selectedUser: id } : {}], { relativeTo: this.route });
    }
}
AuthorizedUsersListViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersListViewComponent, deps: [{ token: i1.Router }, { token: i1.ActivatedRoute }], target: i0.ɵɵFactoryTarget.Component });
AuthorizedUsersListViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AuthorizedUsersListViewComponent, selector: "bb-authorized-users-list-view", ngImport: i0, template: `
    <bb-header-ui
      headingType="h1"
      headingClasses="bb-heading-widget__heading px-0 pb-5"
      heading="Authorized Users"
      i18n-heading="
        Authorized Users|Heading for the Authorized Users List View@@authorized-users.journey.list.heading.text"
    ></bb-header-ui>
    <bb-authorized-users-list-wrapper (selectedUser)="onUserSelected($event)"></bb-authorized-users-list-wrapper>
  `, isInline: true, components: [{ type: i2.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i3.AuthorizedUsersListWrapperComponent, selector: "bb-authorized-users-list-wrapper", outputs: ["selectedUser"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersListViewComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-authorized-users-list-view',
                    template: `
    <bb-header-ui
      headingType="h1"
      headingClasses="bb-heading-widget__heading px-0 pb-5"
      heading="Authorized Users"
      i18n-heading="
        Authorized Users|Heading for the Authorized Users List View@@authorized-users.journey.list.heading.text"
    ></bb-header-ui>
    <bb-authorized-users-list-wrapper (selectedUser)="onUserSelected($event)"></bb-authorized-users-list-wrapper>
  `,
                }]
        }], ctorParameters: function () { return [{ type: i1.Router }, { type: i1.ActivatedRoute }]; } });
