// @backbase/authorized-users-journey-ang/esm2015/src/views/authorized-user-create.component.js
import { Component } from '@angular/core';
import { map } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "../services/authorized-users-journey-config.service";
import * as i2 from "@angular/router";
import * as i3 from "@backbase/ui-ang/header";
import * as i4 from "../authorized-users-create/authorized-users-create-wrapper.component";
import * as i5 from "@angular/common";
export class AuthorizedUserCreateViewComponent {
    constructor(config, router, route) {
        this.config = config;
        this.router = router;
        this.route = route;
        this.selectedUser = this.route.params.pipe(map((params) => params['selectedUser']));
    }
    onClose() {
        this.router.navigate(['../list'], { relativeTo: this.route });
    }
}
AuthorizedUserCreateViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUserCreateViewComponent, deps: [{ token: i1.AuthorizedUsersJourneyConfigService }, { token: i2.Router }, { token: i2.ActivatedRoute }], target: i0.ɵɵFactoryTarget.Component });
AuthorizedUserCreateViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AuthorizedUserCreateViewComponent, selector: "bb-authorized-user-create-view", ngImport: i0, template: `
    <bb-header-ui
      headingType="h1"
      headingClasses="bb-heading-widget__heading px-0 pb-5"
      heading="Authorized User"
      i18n-heading="
        Authorized User|Heading for the Create Authorized User View@@authorized-users.journey.create.heading.text"
    ></bb-header-ui>
    <bb-authorized-users-create-wrapper
      [selectedUser]="(selectedUser | async) || undefined"
      [enableAccountLevelFullAccessPermission]="config.fullAccess"
      (showAuthorizedUsers)="onClose()"
    ></bb-authorized-users-create-wrapper>
  `, isInline: true, components: [{ type: i3.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i4.AuthorizedUsersCreateWrapperComponent, selector: "bb-authorized-users-create-wrapper", inputs: ["selectedUser", "enableAccountLevelFullAccessPermission"], outputs: ["showAuthorizedUsers"] }], pipes: { "async": i5.AsyncPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUserCreateViewComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-authorized-user-create-view',
                    template: `
    <bb-header-ui
      headingType="h1"
      headingClasses="bb-heading-widget__heading px-0 pb-5"
      heading="Authorized User"
      i18n-heading="
        Authorized User|Heading for the Create Authorized User View@@authorized-users.journey.create.heading.text"
    ></bb-header-ui>
    <bb-authorized-users-create-wrapper
      [selectedUser]="(selectedUser | async) || undefined"
      [enableAccountLevelFullAccessPermission]="config.fullAccess"
      (showAuthorizedUsers)="onClose()"
    ></bb-authorized-users-create-wrapper>
  `,
                }]
        }], ctorParameters: function () { return [{ type: i1.AuthorizedUsersJourneyConfigService }, { type: i2.Router }, { type: i2.ActivatedRoute }]; } });
