// @backbase/user-common/esm2015/src/components/user-common.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { InputBase } from './input-base/input-base';
import { EditAddressComponent } from './manage-profile/edit-address/edit-address.component';
import { EditEmailAddressComponent } from './manage-profile/manage-email-addresses/edit-email-address.component';
import { ManageEmailAddressComponent } from './manage-profile/manage-email-addresses/manage-email-address.component';
import { ManageEmailAddressesComponent } from './manage-profile/manage-email-addresses/manage-email-addresses.component';
import { NewEmailAddressComponent } from './manage-profile/manage-email-addresses/new-email-address.component';
import { EditPhoneNumberComponent } from './manage-profile/manage-phone-numbers/edit-phone-number.component';
import { ManagePhoneNumberComponent } from './manage-profile/manage-phone-numbers/manage-phone-number.component';
import { ManagePhoneNumbersComponent } from './manage-profile/manage-phone-numbers/manage-phone-numbers.component';
import { NewPhoneNumberComponent } from './manage-profile/manage-phone-numbers/new-phone-number.component';
import { EditPostalAddressComponent } from './manage-profile/manage-postal-addresses/edit-postal-address.component';
import { ManagePostalAddressComponent } from './manage-profile/manage-postal-addresses/manage-postal-address.component';
import { ManagePostalAddressesComponent } from './manage-profile/manage-postal-addresses/manage-postal-addresses.component';
import { NewPostalAddressComponent } from './manage-profile/manage-postal-addresses/new-postal-address.component';
import { ManageProfileEmailAddressFormComponent } from './manage-profile/manage-profile-form/manage-profile-email-address-form.component';
import { ManageProfileFormComponent } from './manage-profile/manage-profile-form/manage-profile-form.component';
import { ManageProfilePhoneNumberFormComponent } from './manage-profile/manage-profile-form/manage-profile-phone-number-form.component';
import { ManageProfilePostalAddressFormComponent } from './manage-profile/manage-profile-form/manage-profile-postal-address-form.component';
import { ManageProfileComponent } from './manage-profile/manage-profile.component';
import { NewAddressComponent } from './manage-profile/new-address/new-address.component';
import { PhoneNumberInputBase } from './phone-number-input/phone-number-input-base';
import * as i0 from "@angular/core";
const components = [
    EditAddressComponent,
    EditEmailAddressComponent,
    EditPhoneNumberComponent,
    EditPostalAddressComponent,
    InputBase,
    ManageEmailAddressesComponent,
    ManageEmailAddressComponent,
    ManagePhoneNumbersComponent,
    ManagePhoneNumberComponent,
    ManagePostalAddressesComponent,
    ManagePostalAddressComponent,
    ManageProfileComponent,
    ManageProfileComponent,
    ManageProfileFormComponent,
    ManageProfileEmailAddressFormComponent,
    ManageProfilePhoneNumberFormComponent,
    ManageProfilePostalAddressFormComponent,
    NewAddressComponent,
    NewEmailAddressComponent,
    NewPhoneNumberComponent,
    NewPostalAddressComponent,
    PhoneNumberInputBase,
];
export class UserCommonModule {
}
UserCommonModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserCommonModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
UserCommonModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserCommonModule, declarations: [EditAddressComponent,
        EditEmailAddressComponent,
        EditPhoneNumberComponent,
        EditPostalAddressComponent,
        InputBase,
        ManageEmailAddressesComponent,
        ManageEmailAddressComponent,
        ManagePhoneNumbersComponent,
        ManagePhoneNumberComponent,
        ManagePostalAddressesComponent,
        ManagePostalAddressComponent,
        ManageProfileComponent,
        ManageProfileComponent,
        ManageProfileFormComponent,
        ManageProfileEmailAddressFormComponent,
        ManageProfilePhoneNumberFormComponent,
        ManageProfilePostalAddressFormComponent,
        NewAddressComponent,
        NewEmailAddressComponent,
        NewPhoneNumberComponent,
        NewPostalAddressComponent,
        PhoneNumberInputBase], imports: [CommonModule], exports: [EditAddressComponent,
        EditEmailAddressComponent,
        EditPhoneNumberComponent,
        EditPostalAddressComponent,
        InputBase,
        ManageEmailAddressesComponent,
        ManageEmailAddressComponent,
        ManagePhoneNumbersComponent,
        ManagePhoneNumberComponent,
        ManagePostalAddressesComponent,
        ManagePostalAddressComponent,
        ManageProfileComponent,
        ManageProfileComponent,
        ManageProfileFormComponent,
        ManageProfileEmailAddressFormComponent,
        ManageProfilePhoneNumberFormComponent,
        ManageProfilePostalAddressFormComponent,
        NewAddressComponent,
        NewEmailAddressComponent,
        NewPhoneNumberComponent,
        NewPostalAddressComponent,
        PhoneNumberInputBase] });
UserCommonModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserCommonModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserCommonModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [...components],
                    exports: [...components],
                }]
        }] });
