// @backbase/ui-ang/esm2015/input-datepicker/input-datepicker-adapter.js
import { Injectable } from '@angular/core';
import { NgbDateAdapter } from '@ng-bootstrap/ng-bootstrap';
import * as i0 from "@angular/core";
const isValidNgbDateStructProp = (struct, propName) => struct ? Object.prototype.hasOwnProperty.call(struct, propName) && typeof struct[propName] === 'number' : false;
export const isNgbDateStruct = (value) => isValidNgbDateStructProp(value, 'day') &&
    isValidNgbDateStructProp(value, 'month') &&
    isValidNgbDateStructProp(value, 'year');
export class NgbDateStringAdapter extends NgbDateAdapter {
    fromModel(dateString) {
        if (dateString) {
            const d = new Date(dateString);
            const year = d.getFullYear();
            if (year) {
                return {
                    year: year,
                    month: d.getMonth() + 1,
                    day: d.getDate(),
                };
            }
        }
        return null;
    }
    toModel(date) {
        if (!date) {
            return '';
        }
        else if (typeof date === 'string') {
            return date;
        }
        else if (isNgbDateStruct(date)) {
            return new Date(date.year, date.month - 1, date.day).toISOString();
        }
        return Object.prototype.toString.call(date);
    }
}
NgbDateStringAdapter.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NgbDateStringAdapter, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
NgbDateStringAdapter.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NgbDateStringAdapter });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NgbDateStringAdapter, decorators: [{
            type: Injectable
        }] });
