// @backbase/ui-ang/esm2015/input-datepicker/input-datepicker-i18n.js
import { Injectable, Inject, LOCALE_ID, InjectionToken, Optional } from '@angular/core';
import { FormStyle, getLocaleDayNames, getLocaleMonthNames, TranslationWidth } from '@angular/common';
import { NgbDatepickerI18n } from '@ng-bootstrap/ng-bootstrap';
import * as i0 from "@angular/core";
import * as i1 from "@ng-bootstrap/ng-bootstrap";
export const DATEPICKER_I18 = new InjectionToken('datepickerI18');
export class NgbDatepickerI18nDefault extends NgbDatepickerI18n {
    constructor(locale, datepickerI18) {
        super();
        this.datepickerI18 = datepickerI18;
        const weekdaysStartingOnSunday = getLocaleDayNames(locale, FormStyle.Standalone, TranslationWidth.Short);
        this.weekdaysShort = weekdaysStartingOnSunday.map((day, index) => weekdaysStartingOnSunday[(index + 1) % 7]);
        this.monthsShort = getLocaleMonthNames(locale, FormStyle.Standalone, TranslationWidth.Abbreviated);
        this.monthsFull = getLocaleMonthNames(locale, FormStyle.Standalone, TranslationWidth.Wide);
    }
    getWeekdayLabel(weekday) {
        return this.datepickerI18 ? this.datepickerI18.getWeekdayLabel(weekday) : this.weekdaysShort[weekday - 1];
    }
    getMonthShortName(month) {
        return this.datepickerI18 ? this.datepickerI18.getMonthShortName(month) : this.monthsShort[month - 1];
    }
    getMonthFullName(month) {
        return this.datepickerI18 ? this.datepickerI18.getMonthFullName(month) : this.monthsFull[month - 1];
    }
    getDayAriaLabel(date) {
        // @todo: use i18n date pipes?
        return this.datepickerI18 ? this.datepickerI18.getDayAriaLabel(date) : `${date.day}-${date.month}-${date.year}`;
    }
}
NgbDatepickerI18nDefault.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NgbDatepickerI18nDefault, deps: [{ token: LOCALE_ID }, { token: DATEPICKER_I18, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
NgbDatepickerI18nDefault.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NgbDatepickerI18nDefault });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NgbDatepickerI18nDefault, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [LOCALE_ID]
                }] }, { type: i1.NgbDatepickerI18n, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [DATEPICKER_I18]
                }] }]; } });
