// @backbase/ui-ang/esm2015/input-datepicker/input-datepicker.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { NgbDatepickerModule } from '@ng-bootstrap/ng-bootstrap';
import { BaseClassesModule } from '@backbase/ui-ang/base-classes';
import { ButtonModule } from '@backbase/ui-ang/button';
import { IconModule } from '@backbase/ui-ang/icon';
import { InputDatepickerComponent } from './input-datepicker.component';
import { MinDateValidatorDirective } from './validation/min-date-validator/min-date-validator.directive';
import { MaxDateValidatorDirective } from './validation/max-date-validator/max-date-validator.directive';
import { InputDatepickerDayTemplateSingleComponent } from './input-datepicker-day-template-single.component';
import { InputDatepickerDayTemplateRangeComponent } from './input-datepicker-day-template-range.component';
import * as i0 from "@angular/core";
export { MinDateValidatorDirective } from './validation/min-date-validator/min-date-validator.directive';
export { MaxDateValidatorDirective } from './validation/max-date-validator/max-date-validator.directive';
export class InputDatepickerModule {
}
InputDatepickerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputDatepickerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputDatepickerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputDatepickerModule, declarations: [InputDatepickerComponent,
        MinDateValidatorDirective,
        MaxDateValidatorDirective,
        InputDatepickerDayTemplateSingleComponent,
        InputDatepickerDayTemplateRangeComponent], imports: [CommonModule,
        ReactiveFormsModule,
        FormsModule,
        NgbDatepickerModule,
        ButtonModule,
        IconModule,
        BaseClassesModule], exports: [InputDatepickerComponent,
        MinDateValidatorDirective,
        MaxDateValidatorDirective,
        /**
         * @deprecated
         * Exporting `FormsModule` is deprecated
         **/
        FormsModule] });
InputDatepickerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputDatepickerModule, imports: [[
            CommonModule,
            ReactiveFormsModule,
            FormsModule,
            NgbDatepickerModule,
            ButtonModule,
            IconModule,
            BaseClassesModule,
        ], 
        /**
         * @deprecated
         * Exporting `FormsModule` is deprecated
         **/
        FormsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputDatepickerModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        ReactiveFormsModule,
                        FormsModule,
                        NgbDatepickerModule,
                        ButtonModule,
                        IconModule,
                        BaseClassesModule,
                    ],
                    declarations: [
                        InputDatepickerComponent,
                        MinDateValidatorDirective,
                        MaxDateValidatorDirective,
                        InputDatepickerDayTemplateSingleComponent,
                        InputDatepickerDayTemplateRangeComponent,
                    ],
                    exports: [
                        InputDatepickerComponent,
                        MinDateValidatorDirective,
                        MaxDateValidatorDirective,
                        /**
                         * @deprecated
                         * Exporting `FormsModule` is deprecated
                         **/
                        FormsModule,
                    ],
                }]
        }] });
