// @backbase/payment-orders-ang/esm2015/lib/configs/models/form-item-names.model.js
export var PaymentBaseFields;
(function (PaymentBaseFields) {
    PaymentBaseFields["initiator"] = "initiator";
    PaymentBaseFields["counterparty"] = "counterparty";
    PaymentBaseFields["remittanceInfo"] = "remittanceInfo";
    PaymentBaseFields["schedule"] = "schedule";
    PaymentBaseFields["additions"] = "additions";
})(PaymentBaseFields || (PaymentBaseFields = {}));
export var PaymentOptionFields;
(function (PaymentOptionFields) {
    PaymentOptionFields["selectPaymentOption"] = "selectPaymentOption";
    PaymentOptionFields["paymentCost"] = "paymentCost";
    PaymentOptionFields["chargeBearer"] = "chargeBearer";
    PaymentOptionFields["chargeBearerOptions"] = "chargeBearerOptions";
    PaymentOptionFields["transferFee"] = "transferFee";
    PaymentOptionFields["exchangeRate"] = "exchangeRate";
    PaymentOptionFields["inputFormFromSettings"] = "inputFormFromSettings";
    PaymentOptionFields["paymentTypeName"] = "paymentTypeName";
})(PaymentOptionFields || (PaymentOptionFields = {}));
export var InitiatorFields;
(function (InitiatorFields) {
    InitiatorFields["initiatorAccountGroup"] = "debitorAccount";
    InitiatorFields["id"] = "id";
    InitiatorFields["name"] = "name";
    InitiatorFields["alias"] = "alias";
    InitiatorFields["accountNumber"] = "accountNumber";
    InitiatorFields["accountType"] = "accountType";
    InitiatorFields["currency"] = "currency";
})(InitiatorFields || (InitiatorFields = {}));
export var AccountField;
(function (AccountField) {
    AccountField["id"] = "id";
    AccountField["name"] = "name";
    AccountField["accountNumber"] = "accountNumber";
    AccountField["accountType"] = "accountType";
    AccountField["currency"] = "currency";
    AccountField["bankBranchCode"] = "bankBranchCode";
    AccountField["balance"] = "balance";
    AccountField["number"] = "number";
})(AccountField || (AccountField = {}));
export var CounterPartyFields;
(function (CounterPartyFields) {
    CounterPartyFields["counterPartyAccountGroup"] = "creditorAccount";
    CounterPartyFields["standardEntryClass"] = "entryClass";
    CounterPartyFields["creditor"] = "creditor";
    CounterPartyFields["creditorBank"] = "creditorBank";
    CounterPartyFields["bankName"] = "bankName";
    CounterPartyFields["bic"] = "bic";
    CounterPartyFields["bankBranchCode"] = "bankBranchCode";
    CounterPartyFields["correspondentBank"] = "correspondentBank";
    CounterPartyFields["intermediaryBank"] = "intermediaryBank";
    CounterPartyFields["schemeName"] = "schemeName";
    CounterPartyFields["recipientId"] = "recipientId";
    CounterPartyFields["accountType"] = "accountType";
    CounterPartyFields["accountNumber"] = "accountNumber";
    CounterPartyFields["name"] = "name";
    CounterPartyFields["alias"] = "alias";
    CounterPartyFields["id"] = "id";
    CounterPartyFields["postalAddress"] = "postalAddress";
    CounterPartyFields["messageToBank"] = "messageToBank";
    CounterPartyFields["remittanceInformation"] = "remittanceInformation";
    CounterPartyFields["targetCurrency"] = "targetCurrency";
    CounterPartyFields["endToEndIdentification"] = "endToEndIdentification";
    CounterPartyFields["mandateIdentifier"] = "mandateIdentifier";
    CounterPartyFields["role"] = "role";
    CounterPartyFields["selectedContact"] = "selectedContact";
    CounterPartyFields["counterpartyAccountAdditions"] = "counterpartyAccountAdditions";
    CounterPartyFields["additions"] = "additions";
    CounterPartyFields["country"] = "country";
    CounterPartyFields["saveNewContact"] = "saveNewContact";
    CounterPartyFields["enableAdditionalBankDetails"] = "enableAdditionalBankDetails";
})(CounterPartyFields || (CounterPartyFields = {}));
export var PaymentTemplateFields;
(function (PaymentTemplateFields) {
    PaymentTemplateFields["templateName"] = "templateName";
    PaymentTemplateFields["paymentTemplateId"] = "paymentTemplateId";
})(PaymentTemplateFields || (PaymentTemplateFields = {}));
export var RemittanceInfoFields;
(function (RemittanceInfoFields) {
    RemittanceInfoFields["amountCurrencyGroup"] = "amountGroup";
    RemittanceInfoFields["amount"] = "amount";
    RemittanceInfoFields["currency"] = "currency";
    RemittanceInfoFields["urgent"] = "urgent";
    RemittanceInfoFields["paymentReference"] = "paymentReference";
    RemittanceInfoFields["description"] = "description";
    RemittanceInfoFields["amountOptions"] = "amountOptions";
    RemittanceInfoFields["amountOptionLabel"] = "amountOptionLabel";
    RemittanceInfoFields["purposeOfPayment"] = "purposeOfPayment";
})(RemittanceInfoFields || (RemittanceInfoFields = {}));
export var ScheduleFields;
(function (ScheduleFields) {
    ScheduleFields["scheduleGroup"] = "scheduleGroup";
    ScheduleFields["frequency"] = "frequency";
    ScheduleFields["endType"] = "endType";
    ScheduleFields["every"] = "every";
    ScheduleFields["on"] = "on";
    ScheduleFields["startDate"] = "startDate";
    ScheduleFields["endDate"] = "endDate";
    ScheduleFields["repeat"] = "repeat";
    ScheduleFields["frequencySwitch"] = "frequencySwitch";
    ScheduleFields["scheduleInfo"] = "scheduleInfo";
})(ScheduleFields || (ScheduleFields = {}));
export var HiddenFields;
(function (HiddenFields) {
    HiddenFields["isManualEdit"] = "isManualEdit";
    HiddenFields["isClosedPayment"] = "isClosedPayment";
    HiddenFields["enableBeneficiary"] = "enableBeneficiary";
    HiddenFields["beneficiaryAccountName"] = "beneficiaryAccountName";
    HiddenFields["showSaveNewContact"] = "showSaveNewContact";
    HiddenFields["selectedBeneficiary"] = "selectedBeneficiary";
    HiddenFields["hideBeneficiaryFields"] = "hideBeneficiaryFields";
    HiddenFields["defaultCountry"] = "defaultCountry";
    HiddenFields["isPaymentCopy"] = "isPaymentCopy";
    HiddenFields["isDebitPayment"] = "isDebitPayment";
    HiddenFields["defaultBankCodeSetting"] = "defaultBankCodeSetting";
    HiddenFields["sanctionedCountries"] = "sanctionedCountries";
})(HiddenFields || (HiddenFields = {}));
export var PaymentFormFieldHooks;
(function (PaymentFormFieldHooks) {
    PaymentFormFieldHooks["onInit"] = "onInit";
    PaymentFormFieldHooks["onDestroy"] = "onDestroy";
})(PaymentFormFieldHooks || (PaymentFormFieldHooks = {}));
export var PaymentHooks;
(function (PaymentHooks) {
    PaymentHooks["onInit"] = "onInit";
    PaymentHooks["onDestroy"] = "onDestroy";
    PaymentHooks["onSave"] = "onSave";
    PaymentHooks["onSubmit"] = "onSubmit";
})(PaymentHooks || (PaymentHooks = {}));
/**
 * @deprecated Will be removed in widget collection v3.11.0
 */
export var MultiPaymentHooks;
(function (MultiPaymentHooks) {
    MultiPaymentHooks["onClick"] = "onClick";
    MultiPaymentHooks["onBlur"] = "onBlur";
})(MultiPaymentHooks || (MultiPaymentHooks = {}));
/**
 * @deprecated Will be removed in widget collection v3.11.0
 */
export var StepperAction;
(function (StepperAction) {
    StepperAction["NEXT"] = "next";
    StepperAction["BACK"] = "back";
    StepperAction["SUBMIT"] = "submit";
})(StepperAction || (StepperAction = {}));
/**
 * @deprecated Will be removed in widget collection v3.11.0
 */
export var MultiPaymentFormCardFields;
(function (MultiPaymentFormCardFields) {
    MultiPaymentFormCardFields["counterpartyName"] = "counterpartyName";
    MultiPaymentFormCardFields["amount"] = "amount";
    MultiPaymentFormCardFields["description"] = "description";
})(MultiPaymentFormCardFields || (MultiPaymentFormCardFields = {}));
/**
 * @deprecated Will be removed in widget collection v3.11.0
 */
export var MultiPaymentEvents;
(function (MultiPaymentEvents) {
    MultiPaymentEvents["paymentBlur"] = "payment-blur";
    MultiPaymentEvents["deletePayment"] = "delete-payment";
})(MultiPaymentEvents || (MultiPaymentEvents = {}));
/**
 * @deprecated Will be removed in widget collection v3.11.0
 */
export var multiPaymnetValidationRules;
(function (multiPaymnetValidationRules) {
    multiPaymnetValidationRules["batchNameIsMandatory"] = "batchInfo.name";
    multiPaymnetValidationRules["requestedExecutionDateIsMandatory"] = "batchInfo.requestedExecutionDate";
    multiPaymnetValidationRules["requestedExecutionDateIsValid"] = "batchInfo.requestedExecutionDate";
    multiPaymnetValidationRules["batchhasatleastonecredittransaction"] = "counterparty.debitorAccount";
    multiPaymnetValidationRules["totalcredittransactionscountisequaltointendedtransactionscount"] = "counterparty.debitorAccount";
    multiPaymnetValidationRules["totalcreditamountisequaltototalintendedcreditamount"] = "counterparty.debitorAccount";
    multiPaymnetValidationRules["originatoraccountnumberismandatory"] = "initiator";
    multiPaymnetValidationRules["userhasaccesstooriginatoraccount"] = "initiator";
    multiPaymnetValidationRules["originatoraccountisallowedtodebit"] = "initiator";
    multiPaymnetValidationRules["counterpartyaccountnumberismandatory"] = "counterpartyAccountNumber";
    multiPaymnetValidationRules["userhasaccesstocounterpartyaccount"] = "counterpartyAccountNumber";
    multiPaymnetValidationRules["counterpartyaccountnumberisnotsameasoriginatoraccount"] = "counterpartyAccountNumber";
    multiPaymnetValidationRules["counterpartyaccountisallowedtocredit"] = "counterpartyAccountNumber";
    multiPaymnetValidationRules["batchpaymentiscredit"] = "counterpartyAccountNumber";
    multiPaymnetValidationRules["batchpaymentamountisgreaterthanzero"] = "amount";
    multiPaymnetValidationRules["batchpaymentcurrencyisvalid"] = "amount.currency";
})(multiPaymnetValidationRules || (multiPaymnetValidationRules = {}));
