// @backbase/accounts-transactions-journeys-shared-ui-transform-ang/esm2015/lib/to-array-pipe/to-array.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ToArrayPipe } from './to-array.pipe';
import * as i0 from "@angular/core";
const pipes = [ToArrayPipe];
export class ToArrayPipeModule {
}
/** @nocollapse */ ToArrayPipeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ToArrayPipeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ ToArrayPipeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ToArrayPipeModule, declarations: [ToArrayPipe], imports: [CommonModule], exports: [ToArrayPipe] });
/** @nocollapse */ ToArrayPipeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ToArrayPipeModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ToArrayPipeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [...pipes],
                    exports: [...pipes],
                }]
        }] });
