// @backbase/contact-common-ang/esm2015/lib/components/contact-address-data/contact-address-data.module.js
import { NgModule } from '@angular/core';
import { ContactAddressDataComponent } from './contact-address-data.component';
import { ContactAddressDataCustomizableDirective } from './contact-address-data.directive';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { CountryItemModule } from '../country-item/country-item.module';
import { ContactFullAddressModule } from '../contact-full-address/contact-full-address.module';
import { HeaderModule } from '@backbase/ui-ang/header';
import { IconModule } from '@backbase/ui-ang/icon';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import { DropdownSingleSelectModule } from '@backbase/ui-ang/dropdown-single-select';
import { InputCheckboxModule } from '@backbase/ui-ang/input-checkbox';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { ButtonModule } from '@backbase/ui-ang/button';
import { CamelCaseModule } from '../../pipes/camel-case/camel-case.module';
import * as i0 from "@angular/core";
/**
 * Exports:
 * - ContactAddressDataComponent
 * - ContactAddressDataCustomizableDirective
 */
export class ContactAddressDataModule {
}
/** @nocollapse */ ContactAddressDataModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactAddressDataModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ ContactAddressDataModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactAddressDataModule, declarations: [ContactAddressDataComponent, ContactAddressDataCustomizableDirective], imports: [CommonModule,
        FormsModule,
        IconModule,
        HeaderModule,
        InputTextModule,
        DropdownSingleSelectModule,
        CountryItemModule,
        InputCheckboxModule,
        ButtonModule,
        ContactFullAddressModule,
        TooltipModule,
        CamelCaseModule], exports: [ContactAddressDataComponent, ContactAddressDataCustomizableDirective] });
/** @nocollapse */ ContactAddressDataModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactAddressDataModule, imports: [[
            CommonModule,
            FormsModule,
            IconModule,
            HeaderModule,
            InputTextModule,
            DropdownSingleSelectModule,
            CountryItemModule,
            InputCheckboxModule,
            ButtonModule,
            ContactFullAddressModule,
            TooltipModule,
            CamelCaseModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactAddressDataModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ContactAddressDataComponent, ContactAddressDataCustomizableDirective],
                    exports: [ContactAddressDataComponent, ContactAddressDataCustomizableDirective],
                    imports: [
                        CommonModule,
                        FormsModule,
                        IconModule,
                        HeaderModule,
                        InputTextModule,
                        DropdownSingleSelectModule,
                        CountryItemModule,
                        InputCheckboxModule,
                        ButtonModule,
                        ContactFullAddressModule,
                        TooltipModule,
                        CamelCaseModule,
                    ],
                }]
        }] });
