// @backbase/loans-collection-ui-ang/esm2015/lib/loan-summary/loan-summary-edit-alias/loan-summary-edit-alias.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { InputInlineEditModule } from '@backbase/ui-ang/input-inline-edit';
import { LoanSummaryEditAliasComponent } from './loan-summary-edit-alias.component';
import { LoanItemNameModule } from '../../loan-list-item/loan-item-name/loan-item-name.module';
import { LoanSummaryEditAliasConfirmModalModule } from './loan-summary-edit-alias-confirm-modal/loan-summary-edit-alias-confirm-modal.module';
import * as i0 from "@angular/core";
export class LoanSummaryEditAliasModule {
}
/** @nocollapse */ LoanSummaryEditAliasModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryEditAliasModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanSummaryEditAliasModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryEditAliasModule, declarations: [LoanSummaryEditAliasComponent], imports: [CommonModule, InputInlineEditModule, LoanItemNameModule, LoanSummaryEditAliasConfirmModalModule], exports: [LoanSummaryEditAliasComponent] });
/** @nocollapse */ LoanSummaryEditAliasModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryEditAliasModule, imports: [[CommonModule, InputInlineEditModule, LoanItemNameModule, LoanSummaryEditAliasConfirmModalModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryEditAliasModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanSummaryEditAliasComponent],
                    imports: [CommonModule, InputInlineEditModule, LoanItemNameModule, LoanSummaryEditAliasConfirmModalModule],
                    exports: [LoanSummaryEditAliasComponent],
                }]
        }] });
