// @backbase/ui-ang/esm2015/input-datepicker/validation/min-date-validator/min-date-validator.directive.js
import { Directive, forwardRef, Input } from '@angular/core';
import { NG_VALIDATORS } from '@angular/forms';
import * as i0 from "@angular/core";
export class MinDateValidatorDirective {
    constructor() {
        this._minDate = -Number.MAX_VALUE;
        this.onValidatorChange = () => { };
    }
    set minDate(date) {
        if (!date) {
            this._minDate = -Number.MAX_VALUE;
        }
        else {
            const minDate = typeof date === 'string' ? new Date(date) : new Date(date.year, date.month - 1, date.day);
            this._minDate = minDate.getTime();
        }
        this.onValidatorChange();
    }
    registerOnValidatorChange(fn) {
        this.onValidatorChange = fn;
    }
    validate(control) {
        const value = control.value && control.value.from ? control.value.from : control.value;
        if (!value) {
            return null;
        }
        return new Date(value).getTime() >= this._minDate
            ? // eslint-disable-next-line no-null/no-null
                null
            : { dateLessThanMinDate: true };
    }
    ngOnDestroy() {
        this.minDate = undefined;
    }
}
MinDateValidatorDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MinDateValidatorDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
MinDateValidatorDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: MinDateValidatorDirective, selector: "bb-input-datepicker-ui[minDate]", inputs: { minDate: "minDate" }, providers: [
        {
            provide: NG_VALIDATORS,
            multi: true,
            useExisting: forwardRef(() => MinDateValidatorDirective),
        },
    ], ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: MinDateValidatorDirective, decorators: [{
            type: Directive,
            args: [{
                    // eslint-disable-next-line @angular-eslint/directive-selector
                    selector: 'bb-input-datepicker-ui[minDate]',
                    providers: [
                        {
                            provide: NG_VALIDATORS,
                            multi: true,
                            useExisting: forwardRef(() => MinDateValidatorDirective),
                        },
                    ],
                }]
        }], ctorParameters: function () { return []; }, propDecorators: { minDate: [{
                type: Input
            }] } });
