// @backbase/authorized-users-journey-ang/esm2015/src/common/services/user-data.service.js
import { Injectable } from '@angular/core';
import { AuthorizedUser, AuthorizedUsersList, } from '../model/authorized-users.model';
import { combineLatest } from 'rxjs';
import { map } from 'rxjs/operators';
import { AccountList } from '../model/account-list.model';
import { ServiceAgreementRequest } from '../model/service-agreement.model';
import { Status } from '@backbase/data-ang/accesscontrol';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/accesscontrol";
import * as i2 from "@backbase/data-ang/authorized-users";
import * as i3 from "@backbase/data-ang/arrangements";
import * as i4 from "./users-ui.service";
export class UserDataService {
    constructor(accessGroupDataHttpService, userDataHttpService, productSummaryDataHttpService, usersUIService) {
        this.accessGroupDataHttpService = accessGroupDataHttpService;
        this.userDataHttpService = userDataHttpService;
        this.productSummaryDataHttpService = productSummaryDataHttpService;
        this.usersUIService = usersUIService;
        this.statusMapping = this.usersUIService.statusMappingValue;
        this.accounts = this.productSummaryDataHttpService
            .getArrangementsByBusinessFunction({
            businessFunction: 'Product Summary',
            resourceName: 'Product Summary',
            privilege: 'view',
            size: 1000000,
        })
            .pipe(map(AccountList.fromHttpResponse));
        this.authorizedUsersList = this.userDataHttpService
            .getAuthorizedUsers()
            .pipe(map(AuthorizedUsersList.buildHttpResponse(this.statusMapping)));
    }
    setStatusForServiceAgreements(userSaid, status) {
        return this.accessGroupDataHttpService.putServiceAgreementState({
            serviceAgreementId: userSaid,
            serviceAgreementStatePut: ServiceAgreementRequest.toHttpRequest(status),
        });
    }
    updateAuthorizedUserStatus(id, status) {
        return this.userDataHttpService.updateStatusForAuthorizedUser({
            authorizedUserId: id,
            updateAuthorizedUserStatus: { enabled: status === Status.ENABLED },
        });
    }
    addAuthorizedUser(user) {
        return this.userDataHttpService.createAuthorizedUser({
            postAuthorizedUser: AuthorizedUser.buildHttpRequest(this.statusMapping).post(user),
        });
    }
    retrieveAuthorizedUser() {
        return combineLatest([this.authorizedUsersList, this.accounts]).pipe(map(([authorizedUsers, accounts]) => AuthorizedUsersList.mapData(authorizedUsers, accounts)));
    }
    getAllArrangements() {
        return this.accounts;
    }
    updateAuthorizedUser(id, permissions) {
        return this.userDataHttpService.updatePermissionsForAuthorizedUser({
            authorizedUserId: id,
            putAuthorizedUserPermissions: AuthorizedUser.buildHttpRequest(this.statusMapping).put(permissions),
        });
    }
    removeAuthorizedUser(id) {
        return this.userDataHttpService.deleteAuthorizedUser({
            authorizedUserId: id,
        });
    }
}
UserDataService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: UserDataService, deps: [{ token: i1.ServiceAgreementHttpService }, { token: i2.AuthorizedUserService }, { token: i3.ProductSummaryHttpService }, { token: i4.UsersUIService }], target: i0.ɵɵFactoryTarget.Injectable });
UserDataService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: UserDataService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: UserDataService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.ServiceAgreementHttpService }, { type: i2.AuthorizedUserService }, { type: i3.ProductSummaryHttpService }, { type: i4.UsersUIService }]; } });
