// @backbase/accounts-transactions-journey-ang/esm2015/lib/views/account-info/account-info-view.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AmountModule } from '@backbase/ui-ang/amount';
import { BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import { PaymentCardModule } from '@backbase/ui-ang/payment-card';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { ArrangementStateIndicatorModule, ProductNumberModule } from '@backbase/product-summary-ui';
import { DatePipeModule, FormatPipeModule, ProductSummaryCommon, ValidationModule, } from '@backbase/product-summary-common-ang';
import { ProductSummaryDetailsWidgetModule } from '@backbase/product-summary-details-widget-ang';
import { AccountInfoPropertyModule } from '@backbase/accounts-transactions-journeys-shared-ui-account-ang';
import { SplitArrayPipeModule } from '@backbase/accounts-transactions-journeys-shared-ui-transform-ang';
import { AccountInfoContainerComponent } from './account-info-container/account-info-container.component';
import { AccountInfoViewComponent } from './account-info-view.component';
import * as i0 from "@angular/core";
const components = [AccountInfoViewComponent, AccountInfoContainerComponent];
const modules = [
    ProductSummaryDetailsWidgetModule,
    AccountInfoPropertyModule,
    AmountModule,
    ProductNumberModule,
    ValidationModule,
    FormatPipeModule,
    ArrangementStateIndicatorModule,
    TooltipModule,
    BbDatePipeModule,
    DatePipeModule,
    PaymentCardModule,
    SplitArrayPipeModule,
    ProductSummaryCommon,
];
export class AccountInfoViewModule {
}
/** @nocollapse */ AccountInfoViewModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountInfoViewModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ AccountInfoViewModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountInfoViewModule, declarations: [AccountInfoViewComponent, AccountInfoContainerComponent], imports: [CommonModule, ProductSummaryDetailsWidgetModule,
        AccountInfoPropertyModule,
        AmountModule,
        ProductNumberModule,
        ValidationModule,
        FormatPipeModule,
        ArrangementStateIndicatorModule,
        TooltipModule,
        BbDatePipeModule,
        DatePipeModule,
        PaymentCardModule,
        SplitArrayPipeModule,
        ProductSummaryCommon], exports: [ProductSummaryDetailsWidgetModule,
        AccountInfoPropertyModule,
        AmountModule,
        ProductNumberModule,
        ValidationModule,
        FormatPipeModule,
        ArrangementStateIndicatorModule,
        TooltipModule,
        BbDatePipeModule,
        DatePipeModule,
        PaymentCardModule,
        SplitArrayPipeModule,
        ProductSummaryCommon, AccountInfoViewComponent, AccountInfoContainerComponent] });
/** @nocollapse */ AccountInfoViewModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountInfoViewModule, imports: [[CommonModule, ...modules], ProductSummaryDetailsWidgetModule,
        AccountInfoPropertyModule,
        AmountModule,
        ProductNumberModule,
        ValidationModule,
        FormatPipeModule,
        ArrangementStateIndicatorModule,
        TooltipModule,
        BbDatePipeModule,
        DatePipeModule,
        PaymentCardModule,
        SplitArrayPipeModule,
        ProductSummaryCommon] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountInfoViewModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...modules],
                    declarations: [...components],
                    exports: [...modules, ...components],
                }]
        }] });
