// @backbase/loans-collection-ui-ang/esm2015/lib/directives/tooltip-trigger/tooltip-trigger.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TooltipTriggerDirective } from './tooltip-trigger.directive';
import * as i0 from "@angular/core";
export class TooltipTriggerModule {
}
/** @nocollapse */ TooltipTriggerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TooltipTriggerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ TooltipTriggerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TooltipTriggerModule, declarations: [TooltipTriggerDirective], imports: [CommonModule], exports: [TooltipTriggerDirective] });
/** @nocollapse */ TooltipTriggerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TooltipTriggerModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TooltipTriggerModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [TooltipTriggerDirective],
                    imports: [CommonModule],
                    exports: [TooltipTriggerDirective],
                }]
        }] });
