// @backbase/account-statement-business-widget-ang/esm2015/src/account-statement-business-widget.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { AccountStatementStoreModule } from '@backbase/account-statement-store-ang';
import { AccountStatementCommonModule } from '@backbase/account-statement-common-ang';
import { AccountStatementBusinessWidgetComponent, AccountStatementBusinessWidgetCustomizableDirective, } from './components/account-statement-business-widget/account-statement-business-widget.component';
import { AccountStatementBusinessDownloadFailureNotificationHeaderCustomizableDirective, AccountStatementBusinessTableComponent, AccountStatementBusinessTableCustomizableDirective, } from './components/account-statement-business-table/account-statement-business-table.component';
import { AccountStatementViewOnlineBusinessWrapperComponent } from './components/account-statement-view-online-business-wrapper/account-statement-view-online-business-wrapper.component';
import { AccountStatementBusinessStatementsContainerComponent, AccountStatementBusinessStatementsContainerCustomizableDirective, } from './components/account-statement-business-statements-container/account-statement-business-statements-container.component';
import { AccountStatementBusinessFiltersComponent, AccountStatementBusinessFiltersCustomizableDirective, } from './components/account-statement-business-filters/account-statement-business-filters.component';
import { ReactiveFormsModule } from '@angular/forms';
import { AccountSelectorModule } from '@backbase/ui-ang/account-selector';
import { TableModule } from '@backbase/ui-ang/table';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { CollapsibleModule } from '@backbase/ui-ang/collapsible';
import { InputDatepickerModule } from '@backbase/ui-ang/input-datepicker';
import { InputValidationMessageModule } from '@backbase/ui-ang/input-validation-message';
import { PaginationModule } from '@backbase/ui-ang/pagination';
import { DropdownSingleSelectModule } from '@backbase/ui-ang/dropdown-single-select';
import { DropdownMultiSelectModule } from '@backbase/ui-ang/dropdown-multi-select';
import { BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const components = [
    AccountStatementBusinessWidgetComponent,
    AccountStatementBusinessTableComponent,
    AccountStatementViewOnlineBusinessWrapperComponent,
    AccountStatementBusinessStatementsContainerComponent,
    AccountStatementBusinessFiltersComponent,
];
const directives = [
    AccountStatementBusinessTableCustomizableDirective,
    AccountStatementBusinessWidgetCustomizableDirective,
    AccountStatementBusinessStatementsContainerCustomizableDirective,
    AccountStatementBusinessDownloadFailureNotificationHeaderCustomizableDirective,
    AccountStatementBusinessFiltersCustomizableDirective,
];
const uiModules = [
    TableModule,
    AccountSelectorModule,
    LoadButtonModule,
    EmptyStateModule,
    LoadingIndicatorModule,
    CollapsibleModule,
    InputDatepickerModule,
    InputValidationMessageModule,
    PaginationModule,
    DropdownSingleSelectModule,
    BbDatePipeModule,
    DropdownMultiSelectModule,
];
export class AccountStatementBusinessWidgetModule {
}
AccountStatementBusinessWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementBusinessWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AccountStatementBusinessWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementBusinessWidgetModule, declarations: [AccountStatementBusinessWidgetComponent,
        AccountStatementBusinessTableComponent,
        AccountStatementViewOnlineBusinessWrapperComponent,
        AccountStatementBusinessStatementsContainerComponent,
        AccountStatementBusinessFiltersComponent, AccountStatementBusinessTableCustomizableDirective,
        AccountStatementBusinessWidgetCustomizableDirective,
        AccountStatementBusinessStatementsContainerCustomizableDirective,
        AccountStatementBusinessDownloadFailureNotificationHeaderCustomizableDirective,
        AccountStatementBusinessFiltersCustomizableDirective], imports: [TableModule,
        AccountSelectorModule,
        LoadButtonModule,
        EmptyStateModule,
        LoadingIndicatorModule,
        CollapsibleModule,
        InputDatepickerModule,
        InputValidationMessageModule,
        PaginationModule,
        DropdownSingleSelectModule,
        BbDatePipeModule,
        DropdownMultiSelectModule, CommonModule,
        ReactiveFormsModule, i1.BackbaseCoreModule, AccountStatementStoreModule,
        AccountStatementCommonModule], exports: [AccountStatementCommonModule, AccountStatementBusinessWidgetComponent,
        AccountStatementBusinessTableComponent,
        AccountStatementViewOnlineBusinessWrapperComponent,
        AccountStatementBusinessStatementsContainerComponent,
        AccountStatementBusinessFiltersComponent, AccountStatementBusinessTableCustomizableDirective,
        AccountStatementBusinessWidgetCustomizableDirective,
        AccountStatementBusinessStatementsContainerCustomizableDirective,
        AccountStatementBusinessDownloadFailureNotificationHeaderCustomizableDirective,
        AccountStatementBusinessFiltersCustomizableDirective, TableModule,
        AccountSelectorModule,
        LoadButtonModule,
        EmptyStateModule,
        LoadingIndicatorModule,
        CollapsibleModule,
        InputDatepickerModule,
        InputValidationMessageModule,
        PaginationModule,
        DropdownSingleSelectModule,
        BbDatePipeModule,
        DropdownMultiSelectModule] });
AccountStatementBusinessWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementBusinessWidgetModule, imports: [[
            ...uiModules,
            CommonModule,
            ReactiveFormsModule,
            BackbaseCoreModule.withConfig({
                // eslint-disable-next-line @typescript-eslint/naming-convention
                classMap: { AccountStatementBusinessWidgetComponent },
            }),
            AccountStatementStoreModule,
            AccountStatementCommonModule,
        ], AccountStatementCommonModule, TableModule,
        AccountSelectorModule,
        LoadButtonModule,
        EmptyStateModule,
        LoadingIndicatorModule,
        CollapsibleModule,
        InputDatepickerModule,
        InputValidationMessageModule,
        PaginationModule,
        DropdownSingleSelectModule,
        BbDatePipeModule,
        DropdownMultiSelectModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementBusinessWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components, ...directives],
                    imports: [
                        ...uiModules,
                        CommonModule,
                        ReactiveFormsModule,
                        BackbaseCoreModule.withConfig({
                            // eslint-disable-next-line @typescript-eslint/naming-convention
                            classMap: { AccountStatementBusinessWidgetComponent },
                        }),
                        AccountStatementStoreModule,
                        AccountStatementCommonModule,
                    ],
                    exports: [AccountStatementCommonModule, ...components, ...directives, ...uiModules],
                }]
        }] });
