// @backbase/ui-ang/esm2015/_example/product-item-current-account/showcase/product-item-current-account-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ProductItemCurrentAccountShowcaseComponent } from './product-item-current-account-showcase.component';
import { ProductItemCurrentAccountModule } from '@backbase/ui-ang/product-item-current-account';
import { ProductItemBasicAccountModule } from '@backbase/ui-ang/product-item-basic-account';
import * as i0 from "@angular/core";
export class ProductItemCurrentAccountShowcaseModule {
}
ProductItemCurrentAccountShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCurrentAccountShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductItemCurrentAccountShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCurrentAccountShowcaseModule, declarations: [ProductItemCurrentAccountShowcaseComponent], imports: [CommonModule, ProductItemCurrentAccountModule, ProductItemBasicAccountModule] });
ProductItemCurrentAccountShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCurrentAccountShowcaseModule, imports: [[CommonModule, ProductItemCurrentAccountModule, ProductItemBasicAccountModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemCurrentAccountShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ProductItemCurrentAccountModule, ProductItemBasicAccountModule],
                    declarations: [ProductItemCurrentAccountShowcaseComponent],
                    entryComponents: [ProductItemCurrentAccountShowcaseComponent],
                }]
        }] });
