// @backbase/transactions-common-ang/esm2015/src/services/accounts.service.js
import { Injectable } from '@angular/core';
import { map } from 'rxjs/operators';
import { fromArrangementsResponse } from '../model/account-list.model';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/arrangements";
export class AccountsService {
    constructor(productSummaryDataHttpService) {
        this.productSummaryDataHttpService = productSummaryDataHttpService;
    }
    getAllArrangements() {
        return this.productSummaryDataHttpService
            .getArrangementsByBusinessFunction({
            businessFunction: 'Product Summary',
            resourceName: 'Product Summary',
            privilege: 'view',
            size: 1000000,
        }, 'response')
            .pipe(map(fromArrangementsResponse));
    }
    getProductSummary() {
        return this.productSummaryDataHttpService.getArrangementsByBusinessFunction({
            businessFunction: 'Product Summary',
            resourceName: 'Product Summary',
            privilege: 'view',
            size: 1000000,
        }, 'body');
    }
}
AccountsService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountsService, deps: [{ token: i1.ProductSummaryHttpService }], target: i0.ɵɵFactoryTarget.Injectable });
AccountsService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountsService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccountsService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.ProductSummaryHttpService }]; } });
