// @backbase/foundation-ang/esm2015/core/backbase-core/routing/strategies/default-container-strategy.js
import { RouteComponent } from '../components/route.component';
export class DefaultContainerStrategy {
    itemRoutePath(route, index) {
        if (route) {
            return route;
        }
        return (index + 1).toString();
    }
    // routes is guaranteed to have 1 item, but I don't know how to type that
    defaultRoute(routes) {
        const defaultRoute = routes[0];
        // Default behaviour, redirect to the first tab (primary route).
        return {
            path: '',
            redirectTo: defaultRoute.path,
            pathMatch: 'full',
        };
    }
    itemRoute(child, childrenRoutes, index) {
        return {
            path: this.itemRoutePath(child.route, index),
            component: RouteComponent,
            canActivate: child.canActivate,
            data: {
                renderChild: child.name,
                custom: child.routeData,
            },
            children: childrenRoutes,
        };
    }
    findChildrenRoutes(childrenItemRoutes, item) {
        const itemRoutes = childrenItemRoutes.find(child => child.item === item.name);
        return itemRoutes ? itemRoutes.routes : [];
    }
    childrenRoutes(children, childrenItemRoutes) {
        return children.map((child, index) => this.itemRoute(child, this.findChildrenRoutes(childrenItemRoutes, child), index));
    }
    getRoutes(item, children, childrenRoutes) {
        const childRoutes = this.childrenRoutes(children, childrenRoutes);
        const defaultRoutes = childRoutes.length && item.hasDefault ? [this.defaultRoute(childRoutes)] : [];
        const routes = [...childRoutes, ...defaultRoutes];
        if (item.canActivateChild.length) {
            return [
                {
                    path: '',
                    canActivateChild: item.canActivateChild,
                    children: routes,
                },
            ];
        }
        else {
            return routes;
        }
    }
}
