// @backbase/ui-ang/bundles/backbase-ui-ang-modal.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core'), require('@angular/cdk/a11y'), require('@ng-bootstrap/ng-bootstrap'), require('@backbase/ui-ang/util'), require('@backbase/ui-ang/services'), require('@backbase/foundation-ang/future')) :
    typeof define === 'function' && define.amd ? define('@backbase/ui-ang/modal', ['exports', '@angular/common', '@angular/core', '@angular/cdk/a11y', '@ng-bootstrap/ng-bootstrap', '@backbase/ui-ang/util', '@backbase/ui-ang/services', '@backbase/foundation-ang/future'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["ui-ang"] = global.backbase["ui-ang"] || {}, global.backbase["ui-ang"].modal = {}), global.ng.common, global.ng.core, global.ng.cdk.a11y, global.i1$1, global.backbase["ui-ang"].util, global.backbase["ui-ang"].services, global.i3));
})(this, (function (exports, i2, i0, i4, i1$1, util, i1, i3) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i4__namespace = /*#__PURE__*/_interopNamespace(i4);
    var i1__namespace$1 = /*#__PURE__*/_interopNamespace(i1$1);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i3__namespace = /*#__PURE__*/_interopNamespace(i3);

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from, pack) {
        if (pack || arguments.length === 2)
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar)
                        ar = Array.prototype.slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
        return to.concat(ar || Array.prototype.slice.call(from));
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    var CustomModalHeaderDirective = /** @class */ (function () {
        function CustomModalHeaderDirective(templateRef) {
            this.templateRef = templateRef;
        }
        return CustomModalHeaderDirective;
    }());
    CustomModalHeaderDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomModalHeaderDirective, deps: [{ token: i0__namespace.TemplateRef }], target: i0__namespace.ɵɵFactoryTarget.Directive });
    CustomModalHeaderDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomModalHeaderDirective, selector: "ng-template[bbCustomModalHeader]", ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomModalHeaderDirective, decorators: [{
                type: i0.Directive,
                args: [{ selector: 'ng-template[bbCustomModalHeader]' }]
            }], ctorParameters: function () { return [{ type: i0__namespace.TemplateRef }]; } });
    var ModalHeaderComponent = /** @class */ (function () {
        function ModalHeaderComponent(domAttrService) {
            this.domAttrService = domAttrService;
            this.hostRef = this;
            /**
             * Title text for the modal header. Defaults to empty string.
             */
            this.title = '';
            /**
             * The event emitted when the modal header close button is clicked.
             */
            this.close = new i0.EventEmitter();
            this.showClose = false;
            this.headerId = this.domAttrService.generateId();
        }
        ModalHeaderComponent.prototype.ngOnInit = function () {
            this.showClose = this.close.observers.length > 0;
        };
        ModalHeaderComponent.prototype.onClose = function () {
            this.close.emit();
        };
        return ModalHeaderComponent;
    }());
    ModalHeaderComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ModalHeaderComponent, deps: [{ token: i1__namespace.DomAttributesService }], target: i0__namespace.ɵɵFactoryTarget.Component });
    ModalHeaderComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ModalHeaderComponent, selector: "bb-modal-header-ui", inputs: { title: "title" }, outputs: { close: "close" }, queries: [{ propertyName: "customModalHeader", first: true, predicate: CustomModalHeaderDirective, descendants: true, static: true }], ngImport: i0__namespace, template: "<div class=\"modal-header\" [attr.id]=\"headerId\" [attr.data-role]=\"'modal-header'\">\n  <ng-container *ngIf=\"customModalHeader; else bbDefaultModalHeader\">\n    <ng-container *ngTemplateOutlet=\"customModalHeader?.templateRef; context: hostRef\"> </ng-container>\n  </ng-container>\n</div>\n\n<ng-template #bbDefaultModalHeader>\n  <h3 class=\"modal-title\" [innerText]=\"title\"></h3>\n  <button *ngIf=\"showClose\" class=\"btn btn-unstyled\" (click)=\"onClose()\">\n    <i class=\"bicon bicon-clear bb-icon bb-icon--clear\" aria-hidden=\"true\"></i>\n  </button>\n</ng-template>\n", directives: [{ type: i2__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i2__namespace.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ModalHeaderComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-modal-header-ui',
                        templateUrl: './modal-header.component.html',
                    }]
            }], ctorParameters: function () { return [{ type: i1__namespace.DomAttributesService }]; }, propDecorators: { customModalHeader: [{
                    type: i0.ContentChild,
                    args: [CustomModalHeaderDirective, { static: true }]
                }], title: [{
                    type: i0.Input
                }], close: [{
                    type: i0.Output
                }] } });

    /**
     * @name ModalComponent
     *
     * @description
     * Component that displays a modal window.
     */
    var ModalComponent = /** @class */ (function () {
        function ModalComponent(modalService, renderer, domAttrService, depricatedService, 
        // cannot use type `Document` due to compilation issues caused by `strictMetadataEmit`
        document) {
            this.modalService = modalService;
            this.renderer = renderer;
            this.domAttrService = domAttrService;
            this.depricatedService = depricatedService;
            this.document = document;
            this.id = this.domAttrService.generateId();
            /**
             * The flag to show the dialog window.  Defaults to 'false'.
             */
            this.isOpen = false;
            /**
             * Dialog options. Defaults to empty object.
             */
            this.modalOptions = {};
            /**
             * The event that's fired after confirm button is pressed. Can be used with
             * (click)="dialogRef.onConfirm()" where dialogRef is a templateRef on <bb-dialog-ui #dialogRef />
             */
            this.confirm = new i0.EventEmitter();
            /**
             * The event that's fired after cancel button is pressed. Can be used with
             * (click)="dialogRef.onCancel()" where dialogRef is a templateRef on <bb-dialog-ui #dialogRef />
             */
            this.cancel = new i0.EventEmitter();
            /**
             * Used for two way binding with the isOpen Input. If used, backdrop
             * and esc dismiss cases will be handled automatically.
             */
            this.isOpenChange = new i0.EventEmitter();
            this.emitConfirm = true;
        }
        ModalComponent.prototype.ngOnChanges = function () {
            var _this = this;
            if (this.isOpen) {
                this.setConfirm(true);
                // needs to be inside setTimeout due to the
                // https://github.com/angular/angular/issues/15634
                setTimeout(function () {
                    var headerId = _this.modalHeader ? _this.modalHeader.headerId : '';
                    _this.ariaLabelledby = _this.ariaLabelledby || headerId;
                    _this.modalOptions.ariaDescribedBy = _this.ariaDescribedby
                        ? _this.ariaDescribedby
                        : _this.modalOptions.ariaDescribedBy;
                    _this.modalOptions.ariaLabelledBy = _this.ariaLabelledby ? _this.ariaLabelledby : _this.modalOptions.ariaLabelledBy;
                    _this.modalRef = _this.modalService.open(_this.modalContent, _this.modalOptions);
                    _this.modalRef.result.then(function (result) {
                        if (_this.emitConfirm) {
                            _this.confirm.emit(result);
                        }
                    }, function (reason) { return _this.cancel.emit(_this.getDismissReason(reason)); });
                }, 0);
                // remove role=document from dialog, because we need have role=dialog, bootstrap accessibility bug
                setTimeout(function () {
                    var modalContent = _this.document.getElementsByClassName('modal-dialog');
                    var modal = _this.document.getElementsByClassName('modal');
                    modalContent === null || modalContent === void 0 ? void 0 : modalContent[0].removeAttribute('role');
                    modal === null || modal === void 0 ? void 0 : modal[0].scrollTo(0, 0);
                }, 0);
            }
            else if (this.modalRef) {
                this.setConfirm(false);
                this.modalRef.close();
            }
        };
        ModalComponent.prototype.onElementViewInit = function (el) {
            this.copyAriaAttrsToElement(el);
        };
        ModalComponent.prototype.copyAriaAttrsToElement = function (el) {
            this.depricatedService.logDeprecatedFeature('`copyAriaAttrsToElement` of `ModalComponent` is deprecated.');
            var headerId = this.modalHeader ? this.modalHeader.headerId : '';
            var dialogHostNativeElement = this.modalRef['_windowCmptRef'].instance._elRef.nativeElement;
            if (headerId && !this.modalOptions.ariaLabelledBy) {
                this.renderer.setAttribute(dialogHostNativeElement, 'aria-labelledby', headerId);
            }
            if (this.ariaDescribedby && !this.modalOptions.ariaDescribedBy) {
                this.renderer.setAttribute(dialogHostNativeElement, 'aria-describedby', this.ariaDescribedby);
            }
        };
        ModalComponent.prototype.getDismissReason = function (reason) {
            this.isOpenChange.emit(false);
            return reason;
        };
        ModalComponent.prototype.dismissModal = function (reason) {
            if (this.modalRef)
                this.modalRef.dismiss(reason);
        };
        ModalComponent.prototype.closeModal = function (reason) {
            if (this.modalRef)
                this.modalRef.close(reason);
        };
        ModalComponent.prototype.setConfirm = function (flag) {
            this.emitConfirm = flag;
        };
        ModalComponent.prototype.ngOnDestroy = function () {
            this.closeModal();
        };
        ModalComponent.prototype.trapKeyEvent = function (event) {
            if (util.getKeyCode(event) !== util.KEY_CODES.ESCAPE) {
                event.stopPropagation();
            }
        };
        return ModalComponent;
    }());
    ModalComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ModalComponent, deps: [{ token: i1__namespace$1.NgbModal }, { token: i0__namespace.Renderer2 }, { token: i1__namespace.DomAttributesService }, { token: i3__namespace.DeprecationsService }, { token: i2.DOCUMENT }], target: i0__namespace.ɵɵFactoryTarget.Component });
    ModalComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ModalComponent, selector: "bb-modal-ui", inputs: { isOpen: "isOpen", ariaLabelledby: ["aria-labelledby", "ariaLabelledby"], ariaDescribedby: ["aria-describedby", "ariaDescribedby"], modalOptions: "modalOptions" }, outputs: { confirm: "confirm", cancel: "cancel", isOpenChange: "isOpenChange" }, queries: [{ propertyName: "modalHeader", first: true, predicate: ModalHeaderComponent, descendants: true }], viewQueries: [{ propertyName: "modalContent", first: true, predicate: ["modalContent"], descendants: true, static: true }], usesOnChanges: true, ngImport: i0__namespace, template: "<ng-template #modalContent let-close=\"close\">\n  <div\n    class=\"modal-content-container\"\n    (keydown)=\"trapKeyEvent($event)\"\n    (keyup)=\"trapKeyEvent($event)\"\n    cdkTrapFocus\n    [cdkTrapFocusAutoCapture]=\"true\"\n    (bbElementViewInit)=\"onElementViewInit($event)\"\n  >\n    <ng-content></ng-content>\n  </div>\n</ng-template>\n", directives: [{ type: i0__namespace.forwardRef(function () { return i4__namespace.CdkTrapFocus; }), selector: "[cdkTrapFocus]", inputs: ["cdkTrapFocus", "cdkTrapFocusAutoCapture"], exportAs: ["cdkTrapFocus"] }, { type: i0__namespace.forwardRef(function () { return ElementViewInitDirective; }), selector: "[bbElementViewInit]", outputs: ["bbElementViewInit"] }] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ModalComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-modal-ui',
                        templateUrl: './modal.component.html',
                    }]
            }], ctorParameters: function () {
            return [{ type: i1__namespace$1.NgbModal }, { type: i0__namespace.Renderer2 }, { type: i1__namespace.DomAttributesService }, { type: i3__namespace.DeprecationsService }, { type: undefined, decorators: [{
                            type: i0.Inject,
                            args: [i2.DOCUMENT]
                        }] }];
        }, propDecorators: { isOpen: [{
                    type: i0.Input
                }], ariaLabelledby: [{
                    type: i0.Input,
                    args: ['aria-labelledby']
                }], ariaDescribedby: [{
                    type: i0.Input,
                    args: ['aria-describedby']
                }], modalOptions: [{
                    type: i0.Input
                }], confirm: [{
                    type: i0.Output
                }], cancel: [{
                    type: i0.Output
                }], isOpenChange: [{
                    type: i0.Output
                }], modalContent: [{
                    type: i0.ViewChild,
                    args: ['modalContent', { static: true }]
                }], modalHeader: [{
                    type: i0.ContentChild,
                    args: [ModalHeaderComponent]
                }] } });
    var ElementViewInitDirective = /** @class */ (function () {
        function ElementViewInitDirective(elem) {
            this.elem = elem;
            this.bbElementViewInit = new i0.EventEmitter();
        }
        ElementViewInitDirective.prototype.ngAfterViewInit = function () {
            this.bbElementViewInit.emit(this.elem);
        };
        return ElementViewInitDirective;
    }());
    ElementViewInitDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ElementViewInitDirective, deps: [{ token: i0__namespace.ElementRef }], target: i0__namespace.ɵɵFactoryTarget.Directive });
    ElementViewInitDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: ElementViewInitDirective, selector: "[bbElementViewInit]", outputs: { bbElementViewInit: "bbElementViewInit" }, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ElementViewInitDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: '[bbElementViewInit]',
                    }]
            }], ctorParameters: function () { return [{ type: i0__namespace.ElementRef }]; }, propDecorators: { bbElementViewInit: [{
                    type: i0.Output
                }] } });

    var CustomModalFooterDirective = /** @class */ (function () {
        function CustomModalFooterDirective(templateRef) {
            this.templateRef = templateRef;
        }
        return CustomModalFooterDirective;
    }());
    CustomModalFooterDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomModalFooterDirective, deps: [{ token: i0__namespace.TemplateRef }], target: i0__namespace.ɵɵFactoryTarget.Directive });
    CustomModalFooterDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomModalFooterDirective, selector: "ng-template[bbCustomModalFooter]", ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomModalFooterDirective, decorators: [{
                type: i0.Directive,
                args: [{ selector: 'ng-template[bbCustomModalFooter]' }]
            }], ctorParameters: function () { return [{ type: i0__namespace.TemplateRef }]; } });
    var ModalFooterComponent = /** @class */ (function () {
        function ModalFooterComponent(modal) {
            this.modal = modal;
            this.hostRef = this;
            /**
             * The event that's fired after confirm button is pressed. Can be used with (click)="dialogRef.onConfirm()"
             * where dialogRef is a templateRef on <bb-dialog-ui #dialogRef />
             */
            this.confirm = new i0.EventEmitter();
            /**
             * The event that's fired after cancel button is pressed. Can be used with (click)="dialogRef.onCancel()"
             * where dialogRef is a templateRef on <bb-dialog-ui #dialogRef />
             */
            this.cancel = new i0.EventEmitter();
        }
        ModalFooterComponent.prototype.onCancel = function ($event) {
            if (this.modal) {
                this.modal.dismissModal($event);
            }
            else {
                this.cancel.emit($event);
            }
        };
        ModalFooterComponent.prototype.onConfirm = function ($event) {
            if (this.modal) {
                this.modal.closeModal($event);
            }
            else {
                this.confirm.emit($event);
            }
        };
        return ModalFooterComponent;
    }());
    ModalFooterComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ModalFooterComponent, deps: [{ token: ModalComponent, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Component });
    ModalFooterComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ModalFooterComponent, selector: "bb-modal-footer-ui", inputs: { confirmText: "confirmText", cancelText: "cancelText" }, outputs: { confirm: "confirm", cancel: "cancel" }, queries: [{ propertyName: "customModalFooter", first: true, predicate: CustomModalFooterDirective, descendants: true, static: true }], ngImport: i0__namespace, template: "<div class=\"modal-footer\" [attr.data-role]=\"'modal-footer'\">\n  <ng-container *ngIf=\"customModalFooter; else bbDefaultModalFooter\">\n    <ng-container *ngTemplateOutlet=\"customModalFooter?.templateRef; context: hostRef\"> </ng-container>\n  </ng-container>\n</div>\n\n<ng-template #bbDefaultModalFooter>\n  <div class=\"bb-button-bar\">\n    <button\n      (click)=\"onConfirm()\"\n      class=\"bb-button-bar__button btn btn-primary btn-md\"\n      type=\"button\"\n      data-role=\"confirm\"\n    >\n      <ng-container\n        *ngIf=\"!confirmText\"\n        i18n=\"Accept|Agree with the message presented in the dialog @@bb-dialog-ui.yes\"\n      >\n        Confirm\n      </ng-container>\n      <ng-container *ngIf=\"confirmText\">{{ confirmText }}</ng-container>\n    </button>\n    <button\n      (click)=\"onCancel()\"\n      class=\"bb-button-bar__button btn btn-secondary btn-md\"\n      type=\"button\"\n      data-role=\"cancel\"\n    >\n      <ng-container\n        *ngIf=\"!cancelText\"\n        i18n=\"Reject|Do not agree with the message presented in the dialog @@bb-dialog-ui.no\"\n      >\n        Cancel\n      </ng-container>\n      <ng-container *ngIf=\"cancelText\">{{ cancelText }}</ng-container>\n    </button>\n  </div>\n</ng-template>\n", directives: [{ type: i2__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i2__namespace.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ModalFooterComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-modal-footer-ui',
                        templateUrl: './modal-footer.component.html',
                    }]
            }], ctorParameters: function () {
            return [{ type: ModalComponent, decorators: [{
                            type: i0.Optional
                        }] }];
        }, propDecorators: { customModalFooter: [{
                    type: i0.ContentChild,
                    args: [CustomModalFooterDirective, { static: true }]
                }], confirmText: [{
                    type: i0.Input
                }], cancelText: [{
                    type: i0.Input
                }], confirm: [{
                    type: i0.Output
                }], cancel: [{
                    type: i0.Output
                }] } });

    var CustomModalBodyDirective = /** @class */ (function () {
        function CustomModalBodyDirective(templateRef) {
            this.templateRef = templateRef;
        }
        return CustomModalBodyDirective;
    }());
    CustomModalBodyDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomModalBodyDirective, deps: [{ token: i0__namespace.TemplateRef }], target: i0__namespace.ɵɵFactoryTarget.Directive });
    CustomModalBodyDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomModalBodyDirective, selector: "ng-template[bbCustomModalBody]", ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomModalBodyDirective, decorators: [{
                type: i0.Directive,
                args: [{ selector: 'ng-template[bbCustomModalBody]' }]
            }], ctorParameters: function () { return [{ type: i0__namespace.TemplateRef }]; } });
    var ModalBodyComponent = /** @class */ (function () {
        function ModalBodyComponent() {
            this.hostRef = this;
        }
        return ModalBodyComponent;
    }());
    ModalBodyComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ModalBodyComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    ModalBodyComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ModalBodyComponent, selector: "bb-modal-body-ui", queries: [{ propertyName: "customModalBody", first: true, predicate: CustomModalBodyDirective, descendants: true, static: true }], ngImport: i0__namespace, template: "<div class=\"modal-body\" [attr.data-role]=\"'modal-body'\">\n  <ng-container *ngIf=\"customModalBody; else bbDefaultModalBody\">\n    <ng-container *ngTemplateOutlet=\"customModalBody?.templateRef; context: hostRef\"> </ng-container>\n  </ng-container>\n</div>\n\n<ng-template #bbDefaultModalBody> Custom Body </ng-template>\n", directives: [{ type: i2__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i2__namespace.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ModalBodyComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-modal-body-ui',
                        templateUrl: './modal-body.component.html',
                    }]
            }], propDecorators: { customModalBody: [{
                    type: i0.ContentChild,
                    args: [CustomModalBodyDirective, { static: true }]
                }] } });

    var components = [
        ModalComponent,
        ModalFooterComponent,
        CustomModalFooterDirective,
        ModalBodyComponent,
        CustomModalBodyDirective,
        ModalHeaderComponent,
        CustomModalHeaderDirective,
    ];
    var ModalModule = /** @class */ (function () {
        function ModalModule() {
        }
        return ModalModule;
    }());
    ModalModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ModalModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ModalModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ModalModule, declarations: [ModalComponent,
            ModalFooterComponent,
            CustomModalFooterDirective,
            ModalBodyComponent,
            CustomModalBodyDirective,
            ModalHeaderComponent,
            CustomModalHeaderDirective, ElementViewInitDirective], imports: [i2.CommonModule, i1$1.NgbModalModule, i4.A11yModule], exports: [ModalComponent,
            ModalFooterComponent,
            CustomModalFooterDirective,
            ModalBodyComponent,
            CustomModalBodyDirective,
            ModalHeaderComponent,
            CustomModalHeaderDirective] });
    ModalModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ModalModule, imports: [[i2.CommonModule, i1$1.NgbModalModule, i4.A11yModule]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ModalModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [i2.CommonModule, i1$1.NgbModalModule, i4.A11yModule],
                        declarations: __spreadArray(__spreadArray([], __read(components)), [ElementViewInitDirective]),
                        exports: __spreadArray([], __read(components)),
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.CustomModalBodyDirective = CustomModalBodyDirective;
    exports.CustomModalFooterDirective = CustomModalFooterDirective;
    exports.CustomModalHeaderDirective = CustomModalHeaderDirective;
    exports.ModalBodyComponent = ModalBodyComponent;
    exports.ModalComponent = ModalComponent;
    exports.ModalFooterComponent = ModalFooterComponent;
    exports.ModalHeaderComponent = ModalHeaderComponent;
    exports.ModalModule = ModalModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
