// @backbase/ui-ang/bundles/backbase-ui-ang-badge-counter.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('@backbase/ui-ang/badge-counter', ['exports', '@angular/common', '@angular/core'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["ui-ang"] = global.backbase["ui-ang"] || {}, global.backbase["ui-ang"]["badge-counter"] = {}), global.ng.common, global.ng.core));
})(this, (function (exports, i1, i0) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);

    /**
     * @name BadgeCounterComponent
     *
     * @description
     * Component that displays a counter badge.
     */
    var BadgeCounterComponent = /** @class */ (function () {
        function BadgeCounterComponent() {
            /**
             * The (background) color of the badge. Defaults to 'primary'.
             */
            this.color = 'primary';
            /**
             * The count to show in the badge. Defaults to 0 and badge will be hidden if count equals 0.
             */
            this.count = 0;
            /**
             * The max count value to limit text inside badge. Defaults to 999.
             */
            this.maxCount = 999;
        }
        BadgeCounterComponent.prototype.ngOnChanges = function () {
            var value = this.maxCount && this.maxCount > 0 && this.count && this.maxCount < this.count ? this.maxCount + "+" : this.count;
            this.countValue = value ? value.toString() : undefined;
        };
        return BadgeCounterComponent;
    }());
    BadgeCounterComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BadgeCounterComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    BadgeCounterComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: BadgeCounterComponent, selector: "bb-badge-counter-ui", inputs: { color: "color", count: "count", maxCount: "maxCount" }, usesOnChanges: true, ngImport: i0__namespace, template: "<div class=\"bb-badge-counter\">\n  <span class=\"badge badge-counter\" *ngIf=\"countValue\" [ngClass]=\"[color ? 'badge-counter-' + color : '']\">\n    {{ countValue }}\n  </span>\n</div>\n", directives: [{ type: i1__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i1__namespace.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BadgeCounterComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-badge-counter-ui',
                        templateUrl: './badge-counter.component.html',
                    }]
            }], propDecorators: { color: [{
                    type: i0.Input
                }], count: [{
                    type: i0.Input
                }], maxCount: [{
                    type: i0.Input
                }] } });

    var BadgeCounterModule = /** @class */ (function () {
        function BadgeCounterModule() {
        }
        return BadgeCounterModule;
    }());
    BadgeCounterModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BadgeCounterModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    BadgeCounterModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BadgeCounterModule, declarations: [BadgeCounterComponent], imports: [i1.CommonModule], exports: [BadgeCounterComponent] });
    BadgeCounterModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BadgeCounterModule, imports: [[i1.CommonModule]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BadgeCounterModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [i1.CommonModule],
                        declarations: [BadgeCounterComponent],
                        exports: [BadgeCounterComponent],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.BadgeCounterComponent = BadgeCounterComponent;
    exports.BadgeCounterModule = BadgeCounterModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
