// @backbase/ui-ang/bundles/backbase-ui-ang-header.umd.js
(function (global, factory) {
        typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core')) :
        typeof define === 'function' && define.amd ? define('@backbase/ui-ang/header', ['exports', '@angular/common', '@angular/core'], factory) :
        (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["ui-ang"] = global.backbase["ui-ang"] || {}, global.backbase["ui-ang"].header = {}), global.ng.common, global.ng.core));
})(this, (function (exports, i1, i0) { 'use strict';

        function _interopNamespace(e) {
                if (e && e.__esModule) return e;
                var n = Object.create(null);
                if (e) {
                        Object.keys(e).forEach(function (k) {
                                if (k !== 'default') {
                                        var d = Object.getOwnPropertyDescriptor(e, k);
                                        Object.defineProperty(n, k, d.get ? d : {
                                                enumerable: true,
                                                get: function () { return e[k]; }
                                        });
                                }
                        });
                }
                n["default"] = e;
                return Object.freeze(n);
        }

        var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
        var i0__namespace = /*#__PURE__*/_interopNamespace(i0);

        /**
         * @name HeaderComponent
         *
         * @description
         * Component that displays a header.
         */
        var HeaderComponent = /** @class */ (function () {
            function HeaderComponent() {
            }
            return HeaderComponent;
        }());
        HeaderComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: HeaderComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
        HeaderComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: HeaderComponent, selector: "bb-header-ui", inputs: { headingClasses: "headingClasses", headingType: "headingType", heading: "heading" }, ngImport: i0__namespace, template: "<ng-container [ngSwitch]=\"headingType\">\n  <h1 *ngSwitchCase=\"'h1'\" [ngClass]=\"headingClasses\" data-role=\"headings\">\n    {{ heading }}\n  </h1>\n  <h2 *ngSwitchCase=\"'h2'\" [ngClass]=\"headingClasses\" data-role=\"headings\">\n    {{ heading }}\n  </h2>\n  <h3 *ngSwitchCase=\"'h3'\" [ngClass]=\"headingClasses\" data-role=\"headings\">\n    {{ heading }}\n  </h3>\n  <h4 *ngSwitchCase=\"'h4'\" [ngClass]=\"headingClasses\" data-role=\"headings\">\n    {{ heading }}\n  </h4>\n  <h5 *ngSwitchCase=\"'h5'\" [ngClass]=\"headingClasses\" data-role=\"headings\">\n    {{ heading }}\n  </h5>\n  <h6 *ngSwitchCase=\"'h6'\" [ngClass]=\"headingClasses\" data-role=\"headings\">\n    {{ heading }}\n  </h6>\n</ng-container>\n", directives: [{ type: i1__namespace.NgSwitch, selector: "[ngSwitch]", inputs: ["ngSwitch"] }, { type: i1__namespace.NgSwitchCase, selector: "[ngSwitchCase]", inputs: ["ngSwitchCase"] }, { type: i1__namespace.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }] });
        i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: HeaderComponent, decorators: [{
                    type: i0.Component,
                    args: [{
                            selector: 'bb-header-ui',
                            templateUrl: './header.component.html',
                            styles: [],
                        }]
                }], propDecorators: { headingClasses: [{
                        type: i0.Input
                    }], headingType: [{
                        type: i0.Input
                    }], heading: [{
                        type: i0.Input
                    }] } });

        var HeaderModule = /** @class */ (function () {
            function HeaderModule() {
            }
            return HeaderModule;
        }());
        HeaderModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: HeaderModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
        HeaderModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: HeaderModule, declarations: [HeaderComponent], imports: [i1.CommonModule], exports: [HeaderComponent] });
        HeaderModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: HeaderModule, imports: [[i1.CommonModule]] });
        i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: HeaderModule, decorators: [{
                    type: i0.NgModule,
                    args: [{
                            imports: [i1.CommonModule],
                            declarations: [HeaderComponent],
                            exports: [HeaderComponent],
                        }]
                }] });

        /**
         * Generated bundle index. Do not edit.
         */

        exports.HeaderComponent = HeaderComponent;
        exports.HeaderModule = HeaderModule;

        Object.defineProperty(exports, '__esModule', { value: true });

}));
