// @backbase/ui-ang/bundles/backbase-ui-ang-stepper.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core'), require('@backbase/ui-ang/icon')) :
    typeof define === 'function' && define.amd ? define('@backbase/ui-ang/stepper', ['exports', '@angular/common', '@angular/core', '@backbase/ui-ang/icon'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["ui-ang"] = global.backbase["ui-ang"] || {}, global.backbase["ui-ang"].stepper = {}), global.ng.common, global.ng.core, global.backbase["ui-ang"].icon));
})(this, (function (exports, i2, i0, i1) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);

    /**
     * @name StepperStepComponent
     *
     * @description
     * Component that represents a single step that is a part of the stepper component.
     */
    var StepperStepComponent = /** @class */ (function () {
        function StepperStepComponent() {
            /**
             * Marks step as current
             */
            this.isCurrent = false;
            /**
             * Marks step as checked
             */
            this.isChecked = false;
            /**
             * Marks step as active
             */
            this.isActive = false;
        }
        StepperStepComponent.prototype.ngAfterContentChecked = function () {
            var _this = this;
            var _a;
            // TODO: workaround for angular bug https://github.com/angular/angular/issues/10098#issuecomment-235157642
            // an infinite loop takes place in banking schematics
            // it requires more investigation to see the root cause
            this.steps = (_a = this._steps) === null || _a === void 0 ? void 0 : _a.filter(function (item) { return item !== _this; });
        };
        return StepperStepComponent;
    }());
    StepperStepComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StepperStepComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    StepperStepComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: StepperStepComponent, selector: "bb-stepper-step-ui", inputs: { label: "label", stateLabel: "stateLabel", data: "data", ariaLabel: "ariaLabel", isCurrent: "isCurrent", isChecked: "isChecked", isActive: "isActive" }, queries: [{ propertyName: "_steps", predicate: StepperStepComponent }], ngImport: i0__namespace, template: '', isInline: true });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StepperStepComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-stepper-step-ui',
                        template: '',
                    }]
            }], propDecorators: { _steps: [{
                    type: i0.ContentChildren,
                    args: [StepperStepComponent]
                }], label: [{
                    type: i0.Input
                }], stateLabel: [{
                    type: i0.Input
                }], data: [{
                    type: i0.Input
                }], ariaLabel: [{
                    type: i0.Input
                }], isCurrent: [{
                    type: i0.Input
                }], isChecked: [{
                    type: i0.Input
                }], isActive: [{
                    type: i0.Input
                }] } });

    var CustomStepperStepDirective = /** @class */ (function () {
        function CustomStepperStepDirective(templateRef) {
            this.templateRef = templateRef;
        }
        return CustomStepperStepDirective;
    }());
    CustomStepperStepDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomStepperStepDirective, deps: [{ token: i0__namespace.TemplateRef }], target: i0__namespace.ɵɵFactoryTarget.Directive });
    CustomStepperStepDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomStepperStepDirective, selector: "ng-template[bbCustomStepperStep]", ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomStepperStepDirective, decorators: [{
                type: i0.Directive,
                args: [{ selector: 'ng-template[bbCustomStepperStep]' }]
            }], ctorParameters: function () { return [{ type: i0__namespace.TemplateRef }]; } });
    /**
     * @name StepperComponent
     *
     * @description
     * Component that represents a navigation stepper.
     */
    var StepperComponent = /** @class */ (function () {
        function StepperComponent() {
            /**
             * Output of clicked link
             */
            this.select = new i0.EventEmitter();
        }
        Object.defineProperty(StepperComponent.prototype, "currentStep", {
            /**
             * Returns the current selected step
             */
            get: function () {
                return this.steps && this.steps.find(function (step) { return step.isCurrent; });
            },
            enumerable: false,
            configurable: true
        });
        /**
         * Returns the current selected sub-step
         */
        StepperComponent.prototype.hasCurrentChildStep = function (_a) {
            var steps = _a.steps;
            return steps === null || steps === void 0 ? void 0 : steps.find(function (_a) {
                var isCurrent = _a.isCurrent;
                return isCurrent;
            });
        };
        StepperComponent.prototype.onClickHandler = function (data) {
            this.select.emit(data);
        };
        return StepperComponent;
    }());
    StepperComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StepperComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    StepperComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: StepperComponent, selector: "bb-stepper-ui", outputs: { select: "select" }, queries: [{ propertyName: "stepperStepTemplate", first: true, predicate: CustomStepperStepDirective, descendants: true, read: i0.TemplateRef, static: true }, { propertyName: "steps", predicate: StepperStepComponent }], ngImport: i0__namespace, template: "<div [class.bb-stepper__label]=\"currentStep?.label\" class=\"bb-stepper\">\n  <ng-container *ngFor=\"let step of steps; last as isLast; index as i\">\n    <ng-container\n      *ngTemplateOutlet=\"\n        stepperStepTemplate || defaultStepperStepTemplate;\n        context: {\n          $implicit: step,\n          number: i + 1,\n          isLast: isLast,\n          isChild: false\n        }\n      \"\n    >\n    </ng-container>\n\n    <div *ngIf=\"!isLast\" aria-hidden=\"true\" class=\"bb-stepper__separator\"></div>\n  </ng-container>\n</div>\n\n<ng-template #defaultStepperStepTemplate let-step let-number=\"number\" let-isLast=\"isLast\" let-isChild=\"isChild\">\n  <div\n    [class.bb-step--active]=\"step.isActive\"\n    [class.bb-step--current]=\"step.isCurrent && (isChild || !hasCurrentChildStep(step))\"\n    [class.bb-step--child]=\"isChild\"\n    [class.bb-step--last]=\"isLast\"\n    [class.bb-step--container]=\"step.steps?.length\"\n    [attr.aria-label]=\"step.ariaLabel\"\n    [attr.aria-current]=\"step.isActive ? 'step' : null\"\n    class=\"bb-step\"\n  >\n    <div class=\"bb-step__icon\">\n      <ng-container *ngIf=\"step.isCurrent || !step.isChecked; else checkIcon\">\n        {{ number }}\n      </ng-container>\n    </div>\n    <div *ngIf=\"step.label\" [title]=\"step.label\" [class.text-support]=\"!step.isActive\" class=\"bb-step__label\">\n      <a *ngIf=\"step.data && !step.isCurrent\" (click)=\"onClickHandler(step.data)\" href=\"javascript:void(0)\">\n        <ng-container *ngTemplateOutlet=\"simpleStep; context: { $implicit: step }\"></ng-container>\n      </a>\n      <ng-container *ngIf=\"!step.data || step.isCurrent\">\n        <ng-container *ngTemplateOutlet=\"simpleStep; context: { $implicit: step }\"></ng-container>\n      </ng-container>\n    </div>\n  </div>\n  <ng-container *ngFor=\"let subStep of step.steps; last as isLastSubStep\">\n    <ng-container\n      *ngTemplateOutlet=\"\n        defaultStepperStepTemplate;\n        context: {\n          $implicit: subStep,\n          isChild: true,\n          isLast: !isLast && isLastSubStep\n        }\n      \"\n    >\n    </ng-container>\n  </ng-container>\n</ng-template>\n\n<ng-template #checkIcon>\n  <bb-icon-ui name=\"check\" size=\"sm\"></bb-icon-ui>\n</ng-template>\n\n<ng-template #simpleStep let-step>\n  {{ step.label }}\n  <span *ngIf=\"!step.steps?.length\" [title]=\"step.stateLabel\" class=\"bb-step__state-label\">{{ step.stateLabel }}</span>\n</ng-template>\n", components: [{ type: i1__namespace.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], directives: [{ type: i2__namespace.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { type: i2__namespace.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }, { type: i2__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StepperComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-stepper-ui',
                        templateUrl: './stepper.component.html',
                    }]
            }], propDecorators: { steps: [{
                    type: i0.ContentChildren,
                    args: [StepperStepComponent]
                }], stepperStepTemplate: [{
                    type: i0.ContentChild,
                    args: [CustomStepperStepDirective, { read: i0.TemplateRef, static: true }]
                }], select: [{
                    type: i0.Output
                }] } });

    var StepperModule = /** @class */ (function () {
        function StepperModule() {
        }
        return StepperModule;
    }());
    StepperModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StepperModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    StepperModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StepperModule, declarations: [StepperComponent, StepperStepComponent, CustomStepperStepDirective], imports: [i2.CommonModule, i1.IconModule], exports: [StepperComponent, StepperStepComponent, CustomStepperStepDirective] });
    StepperModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StepperModule, imports: [[i2.CommonModule, i1.IconModule]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: StepperModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [i2.CommonModule, i1.IconModule],
                        declarations: [StepperComponent, StepperStepComponent, CustomStepperStepDirective],
                        exports: [StepperComponent, StepperStepComponent, CustomStepperStepDirective],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.CustomStepperStepDirective = CustomStepperStepDirective;
    exports.StepperComponent = StepperComponent;
    exports.StepperModule = StepperModule;
    exports.StepperStepComponent = StepperStepComponent;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
