// @backbase/ui-ang/bundles/backbase-ui-ang-common-error-state.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core'), require('@angular/common/http'), require('@backbase/ui-ang/empty-state')) :
    typeof define === 'function' && define.amd ? define('@backbase/ui-ang/common-error-state', ['exports', '@angular/common', '@angular/core', '@angular/common/http', '@backbase/ui-ang/empty-state'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["ui-ang"] = global.backbase["ui-ang"] || {}, global.backbase["ui-ang"]["common-error-state"] = {}), global.ng.common, global.ng.core, global.ng.common.http, global.backbase["ui-ang"]["empty-state"]));
})(this, (function (exports, i2, i0, http, i1) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);

    /**
     * @name ErrorCommonStateComponent
     *
     * @description
     * Component that displays error state based on error value that was provided.
     */
    var ErrorCommonStateComponent = /** @class */ (function () {
        function ErrorCommonStateComponent() {
            /**
             * Error that should be displayed. Could be HttpErrorResponse, Error, Template reference or string.
             * Default value empty string.
             */
            this.error = '';
        }
        ErrorCommonStateComponent.prototype.getErrorTemplate = function () {
            if (this.error instanceof http.HttpErrorResponse) {
                switch (this.error.status) {
                    case 400:
                        return this.errorBadRequest;
                    case 401:
                        return this.errorUnauthorized;
                    case 403:
                        return this.errorForbidden;
                    case 404:
                        return this.errorNotFound;
                    case 0:
                        return this.errorConnectivityTemplate;
                    default:
                        return this.errorUnexpectedTemplate;
                }
            }
            else if (this.error instanceof Error) {
                return this.errorMessageTemplate;
            }
            else if (typeof this.error === 'string') {
                return this.errorTextTemplate;
            }
            else if (this.error instanceof i0.TemplateRef) {
                return this.errorCustomTemplate;
            }
            return undefined;
        };
        ErrorCommonStateComponent.prototype.getMessage = function () {
            return this.error instanceof Error ? this.error.message : this.error;
        };
        return ErrorCommonStateComponent;
    }());
    ErrorCommonStateComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ErrorCommonStateComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    ErrorCommonStateComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ErrorCommonStateComponent, selector: "bb-common-error-state-ui", inputs: { error: "error" }, viewQueries: [{ propertyName: "errorBadRequest", first: true, predicate: ["errorBadRequest"], descendants: true, static: true }, { propertyName: "errorUnauthorized", first: true, predicate: ["errorUnauthorized"], descendants: true, static: true }, { propertyName: "errorForbidden", first: true, predicate: ["errorForbidden"], descendants: true, static: true }, { propertyName: "errorNotFound", first: true, predicate: ["errorNotFound"], descendants: true, static: true }, { propertyName: "errorConnectivityTemplate", first: true, predicate: ["errorConnectivityTemplate"], descendants: true, static: true }, { propertyName: "errorUnexpectedTemplate", first: true, predicate: ["errorUnexpectedTemplate"], descendants: true, static: true }, { propertyName: "errorMessageTemplate", first: true, predicate: ["errorMessageTemplate"], descendants: true, static: true }, { propertyName: "errorTextTemplate", first: true, predicate: ["errorTextTemplate"], descendants: true, static: true }, { propertyName: "errorCustomTemplate", first: true, predicate: ["errorCustomTemplate"], descendants: true, static: true }], ngImport: i0__namespace, template: "<div class=\"bb-common-error-state\">\n  <ng-container *ngTemplateOutlet=\"getErrorTemplate()\"></ng-container>\n</div>\n\n<ng-template #errorBadRequest>\n  <bb-empty-state-ui\n    iconModifier=\"user\"\n    i18n-title=\"User action error title@@errorCommonState.error.badRequest.title\"\n    title=\"Bad request error\"\n    i18n-subtitle=\"User action error message@@errorCommonState.error.badRequest.message\"\n    subtitle=\"There is a problem with the action you are trying to perform.\"\n  ></bb-empty-state-ui>\n</ng-template>\n\n<ng-template #errorUnauthorized>\n  <bb-empty-state-ui\n    iconModifier=\"user\"\n    i18n-title=\"Unauthorized error title@@errorCommonState.error.unauthorized.title\"\n    title=\"Unauthorized error\"\n    i18n-subtitle=\"Unauthorized error message@@errorCommonState.error.unauthorized.message\"\n    subtitle=\"You must authenticate yourself to get the requested response.\"\n  ></bb-empty-state-ui>\n</ng-template>\n\n<ng-template #errorForbidden>\n  <bb-empty-state-ui\n    iconModifier=\"user\"\n    i18n-title=\"Forbidden error title@@errorCommonState.error.forbidden.title\"\n    title=\"Forbidden error\"\n    i18n-subtitle=\"Forbidden error message@@errorCommonState.error.forbidden.message\"\n    subtitle=\"You do not have access rights to the content.\"\n  ></bb-empty-state-ui>\n</ng-template>\n\n<ng-template #errorNotFound>\n  <bb-empty-state-ui\n    iconModifier=\"error\"\n    i18n-title=\"Not found error title@@errorCommonState.error.notFound.title\"\n    title=\"Not found error\"\n    i18n-subtitle=\"Not found error message@@errorCommonState.error.notFound.message\"\n    subtitle=\"No data was found.\"\n  ></bb-empty-state-ui>\n</ng-template>\n\n<ng-template #errorConnectivityTemplate>\n  <bb-empty-state-ui\n    iconModifier=\"error\"\n    i18n-title=\"Internet connection error title@@errorCommonState.error.connectivity.title\"\n    title=\"Internet connection error\"\n    i18n-subtitle=\"Internet connection error message@@errorCommonState.error.connectivity.message\"\n    subtitle=\"Are you connected to the internet? Please check your connection and try again.\"\n  ></bb-empty-state-ui>\n</ng-template>\n\n<ng-template #errorUnexpectedTemplate>\n  <bb-empty-state-ui\n    iconModifier=\"unexpected\"\n    i18n-title=\"Unexpected error title@@errorCommonState.error.unexpected.title\"\n    title=\"Unexpected error\"\n    i18n-subtitle=\"Unexpected error message@@errorCommonState.error.unexpected.message\"\n    subtitle=\"Something's gone wrong with our server. Please try again later.\"\n  ></bb-empty-state-ui>\n</ng-template>\n\n<ng-template #errorMessageTemplate>\n  <bb-empty-state-ui\n    iconModifier=\"error\"\n    i18n-title=\"Unknown error title@@errorCommonState.error.unknown.title\"\n    title=\"Unknown error\"\n    [subtitle]=\"getMessage()\"\n  ></bb-empty-state-ui>\n</ng-template>\n\n<ng-template #errorTextTemplate>\n  <bb-empty-state-ui [title]=\"getMessage()\" iconModifier=\"error\"></bb-empty-state-ui>\n</ng-template>\n\n<ng-template #errorCustomTemplate>\n  <ng-container *ngTemplateOutlet=\"error\"></ng-container>\n</ng-template>\n", components: [{ type: i1__namespace.EmptyStateComponent, selector: "bb-empty-state-ui", inputs: ["title", "subtitle", "showIcon", "iconClasses", "iconModifier", "iconSize", "iconColor"] }], directives: [{ type: i2__namespace.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ErrorCommonStateComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-common-error-state-ui',
                        templateUrl: 'common-error-state.component.html',
                    }]
            }], propDecorators: { error: [{
                    type: i0.Input
                }], errorBadRequest: [{
                    type: i0.ViewChild,
                    args: ['errorBadRequest', { static: true }]
                }], errorUnauthorized: [{
                    type: i0.ViewChild,
                    args: ['errorUnauthorized', { static: true }]
                }], errorForbidden: [{
                    type: i0.ViewChild,
                    args: ['errorForbidden', { static: true }]
                }], errorNotFound: [{
                    type: i0.ViewChild,
                    args: ['errorNotFound', { static: true }]
                }], errorConnectivityTemplate: [{
                    type: i0.ViewChild,
                    args: ['errorConnectivityTemplate', { static: true }]
                }], errorUnexpectedTemplate: [{
                    type: i0.ViewChild,
                    args: ['errorUnexpectedTemplate', { static: true }]
                }], errorMessageTemplate: [{
                    type: i0.ViewChild,
                    args: ['errorMessageTemplate', { static: true }]
                }], errorTextTemplate: [{
                    type: i0.ViewChild,
                    args: ['errorTextTemplate', { static: true }]
                }], errorCustomTemplate: [{
                    type: i0.ViewChild,
                    args: ['errorCustomTemplate', { static: true }]
                }] } });

    var ErrorCommonStateModule = /** @class */ (function () {
        function ErrorCommonStateModule() {
        }
        return ErrorCommonStateModule;
    }());
    ErrorCommonStateModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ErrorCommonStateModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ErrorCommonStateModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ErrorCommonStateModule, declarations: [ErrorCommonStateComponent], imports: [i2.CommonModule, i1.EmptyStateModule], exports: [ErrorCommonStateComponent] });
    ErrorCommonStateModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ErrorCommonStateModule, imports: [[i2.CommonModule, i1.EmptyStateModule]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ErrorCommonStateModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [i2.CommonModule, i1.EmptyStateModule],
                        declarations: [ErrorCommonStateComponent],
                        exports: [ErrorCommonStateComponent],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.ErrorCommonStateComponent = ErrorCommonStateComponent;
    exports.ErrorCommonStateModule = ErrorCommonStateModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
