// @backbase/ui-ang/bundles/backbase-ui-ang-product-item-debit-card.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core'), require('@backbase/ui-ang/payment-card-number-pipe'), require('@backbase/ui-ang/icon'), require('@backbase/ui-ang/amount')) :
    typeof define === 'function' && define.amd ? define('@backbase/ui-ang/product-item-debit-card', ['exports', '@angular/common', '@angular/core', '@backbase/ui-ang/payment-card-number-pipe', '@backbase/ui-ang/icon', '@backbase/ui-ang/amount'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["ui-ang"] = global.backbase["ui-ang"] || {}, global.backbase["ui-ang"]["product-item-debit-card"] = {}), global.ng.common, global.ng.core, global.backbase["ui-ang"]["payment-card-number-pipe"], global.backbase["ui-ang"].icon, global.backbase["ui-ang"].amount));
})(this, (function (exports, i2, i0, i3, i1, amount) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i3__namespace = /*#__PURE__*/_interopNamespace(i3);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);

    var CustomProductItemDebitCardTitleDirective = /** @class */ (function () {
        function CustomProductItemDebitCardTitleDirective(templateRef) {
            this.templateRef = templateRef;
        }
        return CustomProductItemDebitCardTitleDirective;
    }());
    CustomProductItemDebitCardTitleDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomProductItemDebitCardTitleDirective, deps: [{ token: i0__namespace.TemplateRef }], target: i0__namespace.ɵɵFactoryTarget.Directive });
    CustomProductItemDebitCardTitleDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomProductItemDebitCardTitleDirective, selector: "ng-template[bbCustomProductItemDebitCardTitle]", ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomProductItemDebitCardTitleDirective, decorators: [{
                type: i0.Directive,
                args: [{ selector: 'ng-template[bbCustomProductItemDebitCardTitle]' }]
            }], ctorParameters: function () { return [{ type: i0__namespace.TemplateRef }]; } });
    var ProductItemDebitCardComponent = /** @class */ (function () {
        function ProductItemDebitCardComponent() {
            this.hostRef = this;
            /**
             * Configuration of how the product number should be formatted.
             */
            this.productNumberFormat = i3.DefaultPaymentCardNumberFormat;
            /**
             * The active state of card (it adds active class for blue line)
             */
            this.active = true;
            /**
             * The favorite state of card (add favorite icon next to title)
             */
            this.favorite = false;
            /**
             * The favorite icon color (define star icon color)
             */
            this.favoriteIconColor = 'primary';
        }
        return ProductItemDebitCardComponent;
    }());
    ProductItemDebitCardComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ProductItemDebitCardComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    ProductItemDebitCardComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ProductItemDebitCardComponent, selector: "bb-product-item-debit-card-ui", inputs: { title: "title", productNumber: "productNumber", productNumberFormat: "productNumberFormat", highlight: "highlight", active: "active", productDetailsInfo: "productDetailsInfo", favorite: "favorite", favoriteIconColor: "favoriteIconColor" }, queries: [{ propertyName: "customProductItemDebitCardTitle", first: true, predicate: CustomProductItemDebitCardTitleDirective, descendants: true, static: true }], ngImport: i0__namespace, template: "<div [class.active]=\"active\" class=\"bb-product-item product-item\">\n  <div class=\"product-item-content\">\n    <ng-container *ngIf=\"customProductItemDebitCardTitle; else defaultProductItemDebitCardTitle\">\n      <ng-container *ngTemplateOutlet=\"customProductItemDebitCardTitle?.templateRef; context: hostRef\"> </ng-container>\n    </ng-container>\n\n    <div *ngIf=\"productDetailsInfo?.owners\" class=\"bb-product-item__owners\">\n      <span data-role=\"productDetailsInfo-owners\">\n        {{ productDetailsInfo?.owners }}\n      </span>\n    </div>\n  </div>\n</div>\n\n<ng-template #defaultProductItemDebitCardTitle>\n  <div class=\"bb-account-info\">\n    <div data-role=\"card-title\" class=\"bb-account-info__title\">\n      {{ title }}\n      <bb-icon-ui\n        *ngIf=\"favorite\"\n        size=\"sm\"\n        class=\"bb-account-info__icon\"\n        [color]=\"favoriteIconColor\"\n        data-role=\"favorite-icon\"\n        name=\"star\"\n      >\n      </bb-icon-ui>\n    </div>\n    <div\n      *ngIf=\"productNumber\"\n      data-role=\"card-sub-title\"\n      class=\"bb-subtitle bb-text-support bb-account-info__product-number\"\n    >\n      {{ productNumber | paymentCardNumber: productNumberFormat }}\n    </div>\n  </div>\n</ng-template>\n", components: [{ type: i1__namespace.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], directives: [{ type: i2__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i2__namespace.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }], pipes: { "paymentCardNumber": i3__namespace.PaymentCardNumberPipe } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ProductItemDebitCardComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-product-item-debit-card-ui',
                        templateUrl: './product-item-debit-card.component.html',
                    }]
            }], propDecorators: { customProductItemDebitCardTitle: [{
                    type: i0.ContentChild,
                    args: [CustomProductItemDebitCardTitleDirective, { static: true }]
                }], title: [{
                    type: i0.Input
                }], productNumber: [{
                    type: i0.Input
                }], productNumberFormat: [{
                    type: i0.Input
                }], highlight: [{
                    type: i0.Input
                }], active: [{
                    type: i0.Input
                }], productDetailsInfo: [{
                    type: i0.Input
                }], favorite: [{
                    type: i0.Input
                }], favoriteIconColor: [{
                    type: i0.Input
                }] } });

    var ProductItemDebitCardModule = /** @class */ (function () {
        function ProductItemDebitCardModule() {
        }
        return ProductItemDebitCardModule;
    }());
    ProductItemDebitCardModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ProductItemDebitCardModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ProductItemDebitCardModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ProductItemDebitCardModule, declarations: [ProductItemDebitCardComponent, CustomProductItemDebitCardTitleDirective], imports: [i2.CommonModule, amount.AmountModule, i3.PaymentCardNumberModule, i1.IconModule], exports: [ProductItemDebitCardComponent, CustomProductItemDebitCardTitleDirective] });
    ProductItemDebitCardModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ProductItemDebitCardModule, imports: [[i2.CommonModule, amount.AmountModule, i3.PaymentCardNumberModule, i1.IconModule]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ProductItemDebitCardModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [i2.CommonModule, amount.AmountModule, i3.PaymentCardNumberModule, i1.IconModule],
                        declarations: [ProductItemDebitCardComponent, CustomProductItemDebitCardTitleDirective],
                        exports: [ProductItemDebitCardComponent, CustomProductItemDebitCardTitleDirective],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.CustomProductItemDebitCardTitleDirective = CustomProductItemDebitCardTitleDirective;
    exports.ProductItemDebitCardComponent = ProductItemDebitCardComponent;
    exports.ProductItemDebitCardModule = ProductItemDebitCardModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
