// @backbase/ui-ang/bundles/backbase-ui-ang-product-item-savings-account.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@backbase/ui-ang/payment-card-number-pipe'), require('@backbase/ui-ang/icon'), require('@backbase/ui-ang/amount'), require('@angular/common'), require('@backbase/ui-ang/highlight')) :
    typeof define === 'function' && define.amd ? define('@backbase/ui-ang/product-item-savings-account', ['exports', '@angular/core', '@backbase/ui-ang/payment-card-number-pipe', '@backbase/ui-ang/icon', '@backbase/ui-ang/amount', '@angular/common', '@backbase/ui-ang/highlight'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["ui-ang"] = global.backbase["ui-ang"] || {}, global.backbase["ui-ang"]["product-item-savings-account"] = {}), global.ng.core, global.backbase["ui-ang"]["payment-card-number-pipe"], global.backbase["ui-ang"].icon, global.backbase["ui-ang"].amount, global.ng.common, global.backbase["ui-ang"].highlight));
})(this, (function (exports, i0, i5, i1, i2, i3, i4) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i5__namespace = /*#__PURE__*/_interopNamespace(i5);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i3__namespace = /*#__PURE__*/_interopNamespace(i3);
    var i4__namespace = /*#__PURE__*/_interopNamespace(i4);

    var CustomProductItemSavingsAccountTitleDirective = /** @class */ (function () {
        function CustomProductItemSavingsAccountTitleDirective(templateRef) {
            this.templateRef = templateRef;
        }
        return CustomProductItemSavingsAccountTitleDirective;
    }());
    CustomProductItemSavingsAccountTitleDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomProductItemSavingsAccountTitleDirective, deps: [{ token: i0__namespace.TemplateRef }], target: i0__namespace.ɵɵFactoryTarget.Directive });
    CustomProductItemSavingsAccountTitleDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomProductItemSavingsAccountTitleDirective, selector: "ng-template[bbCustomProductItemSavingsAccountTitle]", ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomProductItemSavingsAccountTitleDirective, decorators: [{
                type: i0.Directive,
                args: [{ selector: 'ng-template[bbCustomProductItemSavingsAccountTitle]' }]
            }], ctorParameters: function () { return [{ type: i0__namespace.TemplateRef }]; } });
    var CustomProductItemSavingsAccountDetailsFormatterDirective = /** @class */ (function () {
        function CustomProductItemSavingsAccountDetailsFormatterDirective(templateRef) {
            this.templateRef = templateRef;
        }
        return CustomProductItemSavingsAccountDetailsFormatterDirective;
    }());
    CustomProductItemSavingsAccountDetailsFormatterDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomProductItemSavingsAccountDetailsFormatterDirective, deps: [{ token: i0__namespace.TemplateRef }], target: i0__namespace.ɵɵFactoryTarget.Directive });
    CustomProductItemSavingsAccountDetailsFormatterDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomProductItemSavingsAccountDetailsFormatterDirective, selector: "ng-template[bbCustomProductItemSavingsAccountDetailsFormatter]", ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomProductItemSavingsAccountDetailsFormatterDirective, decorators: [{
                type: i0.Directive,
                args: [{ selector: 'ng-template[bbCustomProductItemSavingsAccountDetailsFormatter]' }]
            }], ctorParameters: function () { return [{ type: i0__namespace.TemplateRef }]; } });
    var CustomProductItemSavingsAccountMainSectionDirective = /** @class */ (function () {
        function CustomProductItemSavingsAccountMainSectionDirective(templateRef) {
            this.templateRef = templateRef;
        }
        return CustomProductItemSavingsAccountMainSectionDirective;
    }());
    CustomProductItemSavingsAccountMainSectionDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomProductItemSavingsAccountMainSectionDirective, deps: [{ token: i0__namespace.TemplateRef }], target: i0__namespace.ɵɵFactoryTarget.Directive });
    CustomProductItemSavingsAccountMainSectionDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomProductItemSavingsAccountMainSectionDirective, selector: "ng-template[bbCustomProductItemSavingsAccountMainSection]", ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomProductItemSavingsAccountMainSectionDirective, decorators: [{
                type: i0.Directive,
                args: [{ selector: 'ng-template[bbCustomProductItemSavingsAccountMainSection]' }]
            }], ctorParameters: function () { return [{ type: i0__namespace.TemplateRef }]; } });
    var ProductItemSavingsAccountComponent = /** @class */ (function () {
        function ProductItemSavingsAccountComponent() {
            this.hostRef = this;
            /**
             * Configuration of how the card number should be formatted.
             */
            this.productNumberFormat = i5.DefaultPaymentCardNumberFormat;
            /**
             * The active state of card (it adds active class for blue line)
             */
            this.active = true;
            /**
             * The favorite state of card (add favorite icon next to title)
             */
            this.favorite = false;
            /**
             * The favorite icon color (define star icon color)
             */
            this.favoriteIconColor = 'primary';
        }
        return ProductItemSavingsAccountComponent;
    }());
    ProductItemSavingsAccountComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ProductItemSavingsAccountComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    ProductItemSavingsAccountComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ProductItemSavingsAccountComponent, selector: "bb-product-item-savings-account-ui", inputs: { title: "title", productNumber: "productNumber", productNumberFormat: "productNumberFormat", currency: "currency", amount: "amount", mainSectionCurrency: "mainSectionCurrency", mainSectionAmount: "mainSectionAmount", mainSectionTitle: "mainSectionTitle", highlight: "highlight", productDetailsInfo: "productDetailsInfo", active: "active", favorite: "favorite", favoriteIconColor: "favoriteIconColor" }, queries: [{ propertyName: "customProductItemSavingsAccountTitle", first: true, predicate: CustomProductItemSavingsAccountTitleDirective, descendants: true, static: true }, { propertyName: "customProductItemSavingsAccountDetailsFormatter", first: true, predicate: CustomProductItemSavingsAccountDetailsFormatterDirective, descendants: true, static: true }, { propertyName: "customProductItemSavingsAccountMainSection", first: true, predicate: CustomProductItemSavingsAccountMainSectionDirective, descendants: true, static: true }], ngImport: i0__namespace, template: "<div [class.active]=\"active\" class=\"bb-product-item product-item\">\n  <div class=\"product-item-content\">\n    <div\n      class=\"bb-stack bb-stack--wrap\"\n      [ngClass]=\"{\n        'bb-block bb-block--md': mainSectionAmount\n      }\"\n    >\n      <div class=\"bb-stack__item\">\n        <ng-container *ngIf=\"customProductItemSavingsAccountTitle; else defaultProductItemSavingsAccountTitle\">\n          <ng-container\n            *ngTemplateOutlet=\"customProductItemSavingsAccountTitle?.templateRef; context: hostRef\"\n          ></ng-container>\n        </ng-container>\n      </div>\n      <div class=\"bb-stack__break bb-stack__break--xs\"></div>\n      <div class=\"bb-stack__item bb-stack__item--push-right bb-stack__item--spacing-none bb-product-item__details\">\n        <ng-container\n          *ngIf=\"customProductItemSavingsAccountDetailsFormatter; else defaultProductItemSavingsAccountDetailsFormatter\"\n        >\n          <ng-container\n            *ngTemplateOutlet=\"customProductItemSavingsAccountDetailsFormatter?.templateRef; context: hostRef\"\n          >\n          </ng-container>\n        </ng-container>\n      </div>\n    </div>\n\n    <ng-container *ngIf=\"customProductItemSavingsAccountMainSection; else defaultProductItemSavingsAccountMainSection\">\n      <ng-container *ngTemplateOutlet=\"customProductItemSavingsAccountMainSection?.templateRef; context: hostRef\">\n      </ng-container>\n    </ng-container>\n\n    <div *ngIf=\"productDetailsInfo?.owners\" class=\"bb-product-item__owners\">\n      <span data-role=\"productDetailsInfo-owners\">\n        {{ productDetailsInfo?.owners }}\n      </span>\n    </div>\n  </div>\n</div>\n\n<ng-template #defaultProductItemSavingsAccountTitle>\n  <div class=\"bb-account-info\">\n    <div data-role=\"card-title\" class=\"bb-account-info__title\">\n      {{ title }}\n      <bb-icon-ui\n        *ngIf=\"favorite\"\n        size=\"sm\"\n        [color]=\"favoriteIconColor\"\n        class=\"bb-account-info__icon\"\n        data-role=\"favorite-icon\"\n        name=\"star\"\n      >\n      </bb-icon-ui>\n    </div>\n    <div\n      *ngIf=\"productNumber\"\n      data-role=\"card-sub-title\"\n      class=\"bb-subtitle bb-text-support bb-account-info__product-number\"\n    >\n      {{ productNumber | paymentCardNumber: productNumberFormat }}\n    </div>\n  </div>\n</ng-template>\n\n<ng-template #defaultProductItemSavingsAccountDetailsFormatter>\n  <bb-amount-ui\n    class=\"bb-highlight bb-product-item__amount\"\n    [amount]=\"amount\"\n    [currency]=\"currency\"\n    [bbHighlight]=\"highlight\"\n  ></bb-amount-ui>\n</ng-template>\n\n<ng-template #defaultProductItemSavingsAccountMainSection>\n  <div *ngIf=\"mainSectionAmount\" class=\"bb-stack bb-stack--wrap bb-subtitle bb-text-support\">\n    <span class=\"bb-stack__item bb-product-item__main-section-title\">\n      {{ mainSectionTitle }}\n    </span>\n    <bb-amount-ui\n      [amount]=\"mainSectionAmount\"\n      [currency]=\"mainSectionCurrency\"\n      [bbHighlight]=\"highlight\"\n      class=\"\n        bb-stack__item bb-stack__item--push-right bb-stack__item--spacing-none\n        bb-subtitle\n        bb-product-item__main-section-amount\n      \"\n    ></bb-amount-ui>\n  </div>\n</ng-template>\n", components: [{ type: i1__namespace.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i2__namespace.AmountComponent, selector: "bb-amount-ui", inputs: ["currency", "showPlusSign", "mapCurrency", "showPercent", "abbreviate", "decimalPlaces", "trailingZeroes", "amount"] }], directives: [{ type: i3__namespace.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { type: i3__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i3__namespace.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }, { type: i4__namespace.HighlightDirective, selector: "[bbHighlight]", inputs: ["amount", "bbHighlight", "negativeClass", "positiveClass"] }], pipes: { "paymentCardNumber": i5__namespace.PaymentCardNumberPipe } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ProductItemSavingsAccountComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-product-item-savings-account-ui',
                        templateUrl: './product-item-savings-account.component.html',
                    }]
            }], propDecorators: { customProductItemSavingsAccountTitle: [{
                    type: i0.ContentChild,
                    args: [CustomProductItemSavingsAccountTitleDirective, { static: true }]
                }], customProductItemSavingsAccountDetailsFormatter: [{
                    type: i0.ContentChild,
                    args: [CustomProductItemSavingsAccountDetailsFormatterDirective, { static: true }]
                }], customProductItemSavingsAccountMainSection: [{
                    type: i0.ContentChild,
                    args: [CustomProductItemSavingsAccountMainSectionDirective, { static: true }]
                }], title: [{
                    type: i0.Input
                }], productNumber: [{
                    type: i0.Input
                }], productNumberFormat: [{
                    type: i0.Input
                }], currency: [{
                    type: i0.Input
                }], amount: [{
                    type: i0.Input
                }], mainSectionCurrency: [{
                    type: i0.Input
                }], mainSectionAmount: [{
                    type: i0.Input
                }], mainSectionTitle: [{
                    type: i0.Input
                }], highlight: [{
                    type: i0.Input
                }], productDetailsInfo: [{
                    type: i0.Input
                }], active: [{
                    type: i0.Input
                }], favorite: [{
                    type: i0.Input
                }], favoriteIconColor: [{
                    type: i0.Input
                }] } });

    var ProductItemSavingsAccountModule = /** @class */ (function () {
        function ProductItemSavingsAccountModule() {
        }
        return ProductItemSavingsAccountModule;
    }());
    ProductItemSavingsAccountModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ProductItemSavingsAccountModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ProductItemSavingsAccountModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ProductItemSavingsAccountModule, declarations: [ProductItemSavingsAccountComponent,
            CustomProductItemSavingsAccountTitleDirective,
            CustomProductItemSavingsAccountDetailsFormatterDirective,
            CustomProductItemSavingsAccountMainSectionDirective], imports: [i3.CommonModule, i2.AmountModule, i5.PaymentCardNumberModule, i4.HighlightModule, i1.IconModule], exports: [ProductItemSavingsAccountComponent,
            CustomProductItemSavingsAccountTitleDirective,
            CustomProductItemSavingsAccountDetailsFormatterDirective,
            CustomProductItemSavingsAccountMainSectionDirective] });
    ProductItemSavingsAccountModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ProductItemSavingsAccountModule, imports: [[i3.CommonModule, i2.AmountModule, i5.PaymentCardNumberModule, i4.HighlightModule, i1.IconModule]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ProductItemSavingsAccountModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [i3.CommonModule, i2.AmountModule, i5.PaymentCardNumberModule, i4.HighlightModule, i1.IconModule],
                        declarations: [
                            ProductItemSavingsAccountComponent,
                            CustomProductItemSavingsAccountTitleDirective,
                            CustomProductItemSavingsAccountDetailsFormatterDirective,
                            CustomProductItemSavingsAccountMainSectionDirective,
                        ],
                        exports: [
                            ProductItemSavingsAccountComponent,
                            CustomProductItemSavingsAccountTitleDirective,
                            CustomProductItemSavingsAccountDetailsFormatterDirective,
                            CustomProductItemSavingsAccountMainSectionDirective,
                        ],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.CustomProductItemSavingsAccountDetailsFormatterDirective = CustomProductItemSavingsAccountDetailsFormatterDirective;
    exports.CustomProductItemSavingsAccountMainSectionDirective = CustomProductItemSavingsAccountMainSectionDirective;
    exports.CustomProductItemSavingsAccountTitleDirective = CustomProductItemSavingsAccountTitleDirective;
    exports.ProductItemSavingsAccountComponent = ProductItemSavingsAccountComponent;
    exports.ProductItemSavingsAccountModule = ProductItemSavingsAccountModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
