// @backbase/ui-ang/bundles/backbase-ui-ang-iban-directive.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/forms'), require('@backbase/ui-ang/iban-lib')) :
    typeof define === 'function' && define.amd ? define('@backbase/ui-ang/iban-directive', ['exports', '@angular/core', '@angular/forms', '@backbase/ui-ang/iban-lib'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["ui-ang"] = global.backbase["ui-ang"] || {}, global.backbase["ui-ang"]["iban-directive"] = {}), global.ng.core, global.ng.forms, global.backbase["ui-ang"]["iban-lib"]));
})(this, (function (exports, i0, forms, ibanLib) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);

    var IbanValidatorDirective = /** @class */ (function () {
        function IbanValidatorDirective() {
        }
        // emits invalidIban
        IbanValidatorDirective.prototype.validate = function (control) {
            return control && control.value ? ibanLib.isValidIBANFormControl(control.value) : null;
        };
        return IbanValidatorDirective;
    }());
    IbanValidatorDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: IbanValidatorDirective, deps: [], target: i0__namespace.ɵɵFactoryTarget.Directive });
    IbanValidatorDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: IbanValidatorDirective, selector: "[bbIbanValidatorDir]", providers: [{ provide: forms.NG_VALIDATORS, useExisting: IbanValidatorDirective, multi: true }], ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: IbanValidatorDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: '[bbIbanValidatorDir]',
                        providers: [{ provide: forms.NG_VALIDATORS, useExisting: IbanValidatorDirective, multi: true }],
                    }]
            }] });

    var IbanModule = /** @class */ (function () {
        function IbanModule() {
        }
        return IbanModule;
    }());
    IbanModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: IbanModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    IbanModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: IbanModule, declarations: [IbanValidatorDirective], exports: [IbanValidatorDirective] });
    IbanModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: IbanModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: IbanModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [IbanValidatorDirective],
                        exports: [IbanValidatorDirective],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.IbanModule = IbanModule;
    exports.IbanValidatorDirective = IbanValidatorDirective;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
