// @backbase/ui-ang/bundles/backbase-ui-ang-product-item-term-deposit.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@backbase/ui-ang/payment-card-number-pipe'), require('@backbase/ui-ang/icon'), require('@backbase/ui-ang/amount'), require('@angular/common'), require('@backbase/ui-ang/highlight')) :
    typeof define === 'function' && define.amd ? define('@backbase/ui-ang/product-item-term-deposit', ['exports', '@angular/core', '@backbase/ui-ang/payment-card-number-pipe', '@backbase/ui-ang/icon', '@backbase/ui-ang/amount', '@angular/common', '@backbase/ui-ang/highlight'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["ui-ang"] = global.backbase["ui-ang"] || {}, global.backbase["ui-ang"]["product-item-term-deposit"] = {}), global.ng.core, global.backbase["ui-ang"]["payment-card-number-pipe"], global.backbase["ui-ang"].icon, global.backbase["ui-ang"].amount, global.ng.common, global.backbase["ui-ang"].highlight));
})(this, (function (exports, i0, i5, i1, i2, i3, i4) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i5__namespace = /*#__PURE__*/_interopNamespace(i5);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i3__namespace = /*#__PURE__*/_interopNamespace(i3);
    var i4__namespace = /*#__PURE__*/_interopNamespace(i4);

    var CustomProductItemTermDepositTitleDirective = /** @class */ (function () {
        function CustomProductItemTermDepositTitleDirective(templateRef) {
            this.templateRef = templateRef;
        }
        return CustomProductItemTermDepositTitleDirective;
    }());
    CustomProductItemTermDepositTitleDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomProductItemTermDepositTitleDirective, deps: [{ token: i0__namespace.TemplateRef }], target: i0__namespace.ɵɵFactoryTarget.Directive });
    CustomProductItemTermDepositTitleDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomProductItemTermDepositTitleDirective, selector: "ng-template[bbCustomProductItemTermDepositTitle]", ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomProductItemTermDepositTitleDirective, decorators: [{
                type: i0.Directive,
                args: [{ selector: 'ng-template[bbCustomProductItemTermDepositTitle]' }]
            }], ctorParameters: function () { return [{ type: i0__namespace.TemplateRef }]; } });
    var CustomProductItemTermDepositDetailsFormatterDirective = /** @class */ (function () {
        function CustomProductItemTermDepositDetailsFormatterDirective(templateRef) {
            this.templateRef = templateRef;
        }
        return CustomProductItemTermDepositDetailsFormatterDirective;
    }());
    CustomProductItemTermDepositDetailsFormatterDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomProductItemTermDepositDetailsFormatterDirective, deps: [{ token: i0__namespace.TemplateRef }], target: i0__namespace.ɵɵFactoryTarget.Directive });
    CustomProductItemTermDepositDetailsFormatterDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomProductItemTermDepositDetailsFormatterDirective, selector: "ng-template[bbCustomProductItemTermDepositDetailsFormatter]", ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomProductItemTermDepositDetailsFormatterDirective, decorators: [{
                type: i0.Directive,
                args: [{ selector: 'ng-template[bbCustomProductItemTermDepositDetailsFormatter]' }]
            }], ctorParameters: function () { return [{ type: i0__namespace.TemplateRef }]; } });
    var CustomProductItemTermDepositMainSectionDirective = /** @class */ (function () {
        function CustomProductItemTermDepositMainSectionDirective(templateRef) {
            this.templateRef = templateRef;
        }
        return CustomProductItemTermDepositMainSectionDirective;
    }());
    CustomProductItemTermDepositMainSectionDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomProductItemTermDepositMainSectionDirective, deps: [{ token: i0__namespace.TemplateRef }], target: i0__namespace.ɵɵFactoryTarget.Directive });
    CustomProductItemTermDepositMainSectionDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomProductItemTermDepositMainSectionDirective, selector: "ng-template[bbCustomProductItemTermDepositMainSection]", ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomProductItemTermDepositMainSectionDirective, decorators: [{
                type: i0.Directive,
                args: [{ selector: 'ng-template[bbCustomProductItemTermDepositMainSection]' }]
            }], ctorParameters: function () { return [{ type: i0__namespace.TemplateRef }]; } });
    var ProductItemTermDepositComponent = /** @class */ (function () {
        function ProductItemTermDepositComponent() {
            this.hostRef = this;
            /**
             * Configuration of how the product number should be formatted.
             */
            this.productNumberFormat = i5.DefaultPaymentCardNumberFormat;
            /**
             * The active state of card (it adds active class for blue line)
             */
            this.active = true;
            /**
             * The favorite state of card (add favorite icon next to title)
             */
            this.favorite = false;
            /**
             * The favorite icon color (define star icon color)
             */
            this.favoriteIconColor = 'primary';
        }
        return ProductItemTermDepositComponent;
    }());
    ProductItemTermDepositComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ProductItemTermDepositComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    ProductItemTermDepositComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ProductItemTermDepositComponent, selector: "bb-product-item-term-deposit-ui", inputs: { title: "title", productNumber: "productNumber", productNumberFormat: "productNumberFormat", currency: "currency", amount: "amount", mainSectionCurrency: "mainSectionCurrency", mainSectionAmount: "mainSectionAmount", mainSectionTitle: "mainSectionTitle", highlight: "highlight", productDetailsInfo: "productDetailsInfo", active: "active", favorite: "favorite", favoriteIconColor: "favoriteIconColor" }, queries: [{ propertyName: "customProductItemTermDepositTitle", first: true, predicate: CustomProductItemTermDepositTitleDirective, descendants: true, static: true }, { propertyName: "customProductItemTermDepositDetailsFormatter", first: true, predicate: CustomProductItemTermDepositDetailsFormatterDirective, descendants: true, static: true }, { propertyName: "customProductItemTermDepositMainSection", first: true, predicate: CustomProductItemTermDepositMainSectionDirective, descendants: true, static: true }], ngImport: i0__namespace, template: "<div [class.active]=\"active\" class=\"bb-product-item product-item\">\n  <div class=\"product-item-content\">\n    <div class=\"bb-stack bb-stack--wrap\" [ngClass]=\"{ 'bb-block bb-block--md': mainSectionAmount }\">\n      <div class=\"bb-stack__item\">\n        <ng-container *ngIf=\"customProductItemTermDepositTitle; else defaultProductItemTermDepositTitle\">\n          <ng-container *ngTemplateOutlet=\"customProductItemTermDepositTitle?.templateRef; context: hostRef\">\n          </ng-container>\n        </ng-container>\n      </div>\n      <div class=\"bb-stack__break bb-stack__break--xs\"></div>\n      <div class=\"bb-stack__item bb-stack__item--push-right bb-stack__item--spacing-none bb-product-item__details\">\n        <ng-container\n          *ngIf=\"customProductItemTermDepositDetailsFormatter; else defaultProductItemTermDepositDetailsFormatter\"\n        >\n          <ng-container *ngTemplateOutlet=\"customProductItemTermDepositDetailsFormatter?.templateRef; context: hostRef\">\n          </ng-container>\n        </ng-container>\n      </div>\n    </div>\n\n    <ng-container *ngIf=\"customProductItemTermDepositMainSection; else defaultProductItemTermDepositMainSection\">\n      <ng-container *ngTemplateOutlet=\"customProductItemTermDepositMainSection?.templateRef; context: hostRef\">\n      </ng-container>\n    </ng-container>\n\n    <div *ngIf=\"productDetailsInfo?.owners\" class=\"bb-text-default bb-product-item__owners\">\n      <span data-role=\"productDetailsInfo-owners\">\n        {{ productDetailsInfo?.owners }}\n      </span>\n    </div>\n  </div>\n</div>\n\n<ng-template #defaultProductItemTermDepositTitle>\n  <div data-role=\"card-title\" class=\"bb-text-bold\">\n    {{ title }}\n    <bb-icon-ui *ngIf=\"favorite\" size=\"sm\" [color]=\"favoriteIconColor\" data-role=\"favorite-icon\" name=\"star\">\n    </bb-icon-ui>\n  </div>\n  <div\n    *ngIf=\"productNumber\"\n    data-role=\"card-sub-title\"\n    class=\"bb-subtitle bb-text-support bb-account-info__product-number\"\n  >\n    {{ productNumber | paymentCardNumber: productNumberFormat }}\n  </div>\n</ng-template>\n\n<ng-template #defaultProductItemTermDepositDetailsFormatter>\n  <bb-amount-ui\n    class=\"bb-highlight bb-product-item__amount\"\n    [amount]=\"amount\"\n    [currency]=\"currency\"\n    [bbHighlight]=\"highlight\"\n  ></bb-amount-ui>\n</ng-template>\n\n<ng-template #defaultProductItemTermDepositMainSection>\n  <div *ngIf=\"mainSectionAmount\" class=\"bb-stack bb-stack--wrap bb-subtitle bb-text-support\">\n    <span class=\"bb-stack__item bb-product-item__main-section-title\">\n      {{ mainSectionTitle }}\n    </span>\n    <bb-amount-ui\n      [amount]=\"mainSectionAmount\"\n      [currency]=\"mainSectionCurrency\"\n      [bbHighlight]=\"highlight\"\n      class=\"\n        bb-stack__item bb-stack__item--push-right bb-stack__item--spacing-none\n        bb-product-item__main-section-amount\n      \"\n    ></bb-amount-ui>\n  </div>\n</ng-template>\n", components: [{ type: i1__namespace.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i2__namespace.AmountComponent, selector: "bb-amount-ui", inputs: ["currency", "showPlusSign", "mapCurrency", "showPercent", "abbreviate", "decimalPlaces", "trailingZeroes", "amount"] }], directives: [{ type: i3__namespace.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { type: i3__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i3__namespace.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }, { type: i4__namespace.HighlightDirective, selector: "[bbHighlight]", inputs: ["amount", "bbHighlight", "negativeClass", "positiveClass"] }], pipes: { "paymentCardNumber": i5__namespace.PaymentCardNumberPipe } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ProductItemTermDepositComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-product-item-term-deposit-ui',
                        templateUrl: './product-item-term-deposit.component.html',
                    }]
            }], propDecorators: { customProductItemTermDepositTitle: [{
                    type: i0.ContentChild,
                    args: [CustomProductItemTermDepositTitleDirective, { static: true }]
                }], customProductItemTermDepositDetailsFormatter: [{
                    type: i0.ContentChild,
                    args: [CustomProductItemTermDepositDetailsFormatterDirective, { static: true }]
                }], customProductItemTermDepositMainSection: [{
                    type: i0.ContentChild,
                    args: [CustomProductItemTermDepositMainSectionDirective, { static: true }]
                }], title: [{
                    type: i0.Input
                }], productNumber: [{
                    type: i0.Input
                }], productNumberFormat: [{
                    type: i0.Input
                }], currency: [{
                    type: i0.Input
                }], amount: [{
                    type: i0.Input
                }], mainSectionCurrency: [{
                    type: i0.Input
                }], mainSectionAmount: [{
                    type: i0.Input
                }], mainSectionTitle: [{
                    type: i0.Input
                }], highlight: [{
                    type: i0.Input
                }], productDetailsInfo: [{
                    type: i0.Input
                }], active: [{
                    type: i0.Input
                }], favorite: [{
                    type: i0.Input
                }], favoriteIconColor: [{
                    type: i0.Input
                }] } });

    var ProductItemTermDepositModule = /** @class */ (function () {
        function ProductItemTermDepositModule() {
        }
        return ProductItemTermDepositModule;
    }());
    ProductItemTermDepositModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ProductItemTermDepositModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ProductItemTermDepositModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ProductItemTermDepositModule, declarations: [ProductItemTermDepositComponent,
            CustomProductItemTermDepositTitleDirective,
            CustomProductItemTermDepositDetailsFormatterDirective,
            CustomProductItemTermDepositMainSectionDirective], imports: [i3.CommonModule, i2.AmountModule, i5.PaymentCardNumberModule, i4.HighlightModule, i1.IconModule], exports: [ProductItemTermDepositComponent,
            CustomProductItemTermDepositTitleDirective,
            CustomProductItemTermDepositDetailsFormatterDirective,
            CustomProductItemTermDepositMainSectionDirective] });
    ProductItemTermDepositModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ProductItemTermDepositModule, imports: [[i3.CommonModule, i2.AmountModule, i5.PaymentCardNumberModule, i4.HighlightModule, i1.IconModule]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ProductItemTermDepositModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [i3.CommonModule, i2.AmountModule, i5.PaymentCardNumberModule, i4.HighlightModule, i1.IconModule],
                        declarations: [
                            ProductItemTermDepositComponent,
                            CustomProductItemTermDepositTitleDirective,
                            CustomProductItemTermDepositDetailsFormatterDirective,
                            CustomProductItemTermDepositMainSectionDirective,
                        ],
                        exports: [
                            ProductItemTermDepositComponent,
                            CustomProductItemTermDepositTitleDirective,
                            CustomProductItemTermDepositDetailsFormatterDirective,
                            CustomProductItemTermDepositMainSectionDirective,
                        ],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.CustomProductItemTermDepositDetailsFormatterDirective = CustomProductItemTermDepositDetailsFormatterDirective;
    exports.CustomProductItemTermDepositMainSectionDirective = CustomProductItemTermDepositMainSectionDirective;
    exports.CustomProductItemTermDepositTitleDirective = CustomProductItemTermDepositTitleDirective;
    exports.ProductItemTermDepositComponent = ProductItemTermDepositComponent;
    exports.ProductItemTermDepositModule = ProductItemTermDepositModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
