// @backbase/ui-ang/bundles/backbase-ui-ang-iban-pipe.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core'), require('@backbase/ui-ang/util'), require('@backbase/ui-ang/account-number-pipe')) :
    typeof define === 'function' && define.amd ? define('@backbase/ui-ang/iban-pipe', ['exports', '@angular/common', '@angular/core', '@backbase/ui-ang/util', '@backbase/ui-ang/account-number-pipe'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["ui-ang"] = global.backbase["ui-ang"] || {}, global.backbase["ui-ang"]["iban-pipe"] = {}), global.ng.common, global.ng.core, global.backbase["ui-ang"].util, global.backbase["ui-ang"]["account-number-pipe"]));
})(this, (function (exports, common, i0, util, i1) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);

    /**
     * @name bbIban
     *
     * @description
     * Pipe for formatting iban
     */
    var BbIbanPipe = /** @class */ (function () {
        /**
         * @name BbIbanPipe#constructor
         * @param accountsDisplayingFormatService - AccountsDisplayingFormatService
         *
         * @internal
         */
        function BbIbanPipe(accountsDisplayingFormatService) {
            this.accountsDisplayingFormatService = accountsDisplayingFormatService;
            this.defaultFormat = this.accountsDisplayingFormatService.getDefaultFormat('iban');
        }
        /**
         * @name BbIbanPipe#transform
         * @param value - the iban that should be formatted
         *
         * @description
         * Splits the iban up with spaces
         * Applies global numbers format configuration for IBAN attribute type
         */
        BbIbanPipe.prototype.transform = function (value) {
            return util.segment(value, this.defaultFormat.segments);
        };
        return BbIbanPipe;
    }());
    BbIbanPipe.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BbIbanPipe, deps: [{ token: i1__namespace.AccountsDisplayingFormatService }], target: i0__namespace.ɵɵFactoryTarget.Pipe });
    BbIbanPipe.ɵpipe = i0__namespace.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BbIbanPipe, name: "bbIban" });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BbIbanPipe, decorators: [{
                type: i0.Pipe,
                args: [{
                        name: 'bbIban',
                    }]
            }], ctorParameters: function () { return [{ type: i1__namespace.AccountsDisplayingFormatService }]; } });

    var BbIbanPipeModule = /** @class */ (function () {
        function BbIbanPipeModule() {
        }
        return BbIbanPipeModule;
    }());
    BbIbanPipeModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BbIbanPipeModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    BbIbanPipeModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BbIbanPipeModule, declarations: [BbIbanPipe], imports: [common.CommonModule], exports: [BbIbanPipe] });
    BbIbanPipeModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BbIbanPipeModule, imports: [[common.CommonModule]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: BbIbanPipeModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [common.CommonModule],
                        declarations: [BbIbanPipe],
                        exports: [BbIbanPipe],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.BbIbanPipe = BbIbanPipe;
    exports.BbIbanPipeModule = BbIbanPipeModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
