// @backbase/ui-ang/bundles/backbase-ui-ang-avatar.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('@backbase/ui-ang/avatar', ['exports', '@angular/common', '@angular/core'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["ui-ang"] = global.backbase["ui-ang"] || {}, global.backbase["ui-ang"].avatar = {}), global.ng.common, global.ng.core));
})(this, (function (exports, i1, i0) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);

    var MAX_INITIALS_LENGTH = 2;
    var AvatarComponent = /** @class */ (function () {
        function AvatarComponent() {
            this.initials = '';
            /**
             * The image  of the avatar. Defaults to empty string.
             */
            this.image = '';
            /**
             * The name  of the avatar. Defaults to empty string.
             */
            this.name = '';
            /**
             * The imgAlt for provide alt in img of the avatar. Defaults to empty string.
             */
            this.imgAlt = '';
            /**
             * Size of the avatar. Possible values sm | md| lg | xl. Defaults to empty string.
             */
            this.size = '';
        }
        AvatarComponent.prototype.ngOnChanges = function () {
            this.initials = (this.name || '')
                .split(' ')
                .reduce(function (initials, word) { return "" + initials + word.charAt(0); }, '')
                .substr(0, MAX_INITIALS_LENGTH)
                .toUpperCase();
        };
        return AvatarComponent;
    }());
    AvatarComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AvatarComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    AvatarComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AvatarComponent, selector: "bb-avatar-ui", inputs: { image: "image", name: "name", imgAlt: "imgAlt", size: "size" }, usesOnChanges: true, ngImport: i0__namespace, template: "<div class=\"bb-avatar\" [ngClass]=\"[size ? 'bb-avatar--' + size : '', image ? 'bb-avatar--image' : '']\">\n  <span\n    *ngIf=\"!image\"\n    class=\"bb-avatar__initials\"\n    [textContent]=\"initials\"\n    data-role=\"avatar-initials\"\n    [attr.aria-label]=\"name\"\n  ></span>\n  <div\n    role=\"img\"\n    [attr.aria-label]=\"imgAlt || name\"\n    *ngIf=\"image\"\n    class=\"bb-avatar__image\"\n    data-role=\"avatar-image\"\n    [style.backgroundImage]=\"'url(' + image + ')'\"\n  ></div>\n</div>\n", directives: [{ type: i1__namespace.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { type: i1__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AvatarComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-avatar-ui',
                        templateUrl: './avatar.component.html',
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], propDecorators: { image: [{
                    type: i0.Input
                }], name: [{
                    type: i0.Input
                }], imgAlt: [{
                    type: i0.Input
                }], size: [{
                    type: i0.Input
                }] } });

    var AvatarModule = /** @class */ (function () {
        function AvatarModule() {
        }
        return AvatarModule;
    }());
    AvatarModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AvatarModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    AvatarModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AvatarModule, declarations: [AvatarComponent], imports: [i1.CommonModule], exports: [AvatarComponent] });
    AvatarModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AvatarModule, imports: [[i1.CommonModule]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AvatarModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [i1.CommonModule],
                        declarations: [AvatarComponent],
                        exports: [AvatarComponent],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.AvatarComponent = AvatarComponent;
    exports.AvatarModule = AvatarModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
