// @backbase/ui-ang/bundles/backbase-ui-ang-input-validation-message.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('@backbase/ui-ang/input-validation-message', ['exports', '@angular/common', '@angular/core'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["ui-ang"] = global.backbase["ui-ang"] || {}, global.backbase["ui-ang"]["input-validation-message"] = {}), global.ng.common, global.ng.core));
})(this, (function (exports, i1, i0) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);

    /**
     * @name InputValidationMessageComponent
     *
     * @description
     * Component that displays an input validation message or helper.
     */
    var InputValidationMessageComponent = /** @class */ (function () {
        function InputValidationMessageComponent() {
            /**
             * Whether the errors template should be shown
             */
            this.showErrors = false;
        }
        return InputValidationMessageComponent;
    }());
    InputValidationMessageComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: InputValidationMessageComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    InputValidationMessageComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: InputValidationMessageComponent, selector: "bb-input-validation-message-ui", inputs: { showErrors: "showErrors", messageId: "messageId" }, ngImport: i0__namespace, template: "<ng-container *ngIf=\"showErrors\">\n  <small class=\"bb-input-validation-message\" role=\"alert\" [attr.id]=\"messageId || null\">\n    <ng-content></ng-content>\n  </small>\n</ng-container>\n", directives: [{ type: i1__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: InputValidationMessageComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-input-validation-message-ui',
                        templateUrl: './input-validation-message.component.html',
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], propDecorators: { showErrors: [{
                    type: i0.Input
                }], messageId: [{
                    type: i0.Input
                }] } });

    var InputValidationMessageModule = /** @class */ (function () {
        function InputValidationMessageModule() {
        }
        return InputValidationMessageModule;
    }());
    InputValidationMessageModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: InputValidationMessageModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    InputValidationMessageModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: InputValidationMessageModule, declarations: [InputValidationMessageComponent], imports: [i1.CommonModule], exports: [InputValidationMessageComponent] });
    InputValidationMessageModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: InputValidationMessageModule, imports: [[i1.CommonModule]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: InputValidationMessageModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [i1.CommonModule],
                        declarations: [InputValidationMessageComponent],
                        exports: [InputValidationMessageComponent],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.InputValidationMessageComponent = InputValidationMessageComponent;
    exports.InputValidationMessageModule = InputValidationMessageModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
