// @backbase/ui-ang/bundles/backbase-ui-ang-payment-card.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core'), require('@backbase/ui-ang/payment-card-number-pipe'), require('@backbase/ui-ang/logo'), require('@backbase/ui-ang/card-vendor')) :
    typeof define === 'function' && define.amd ? define('@backbase/ui-ang/payment-card', ['exports', '@angular/common', '@angular/core', '@backbase/ui-ang/payment-card-number-pipe', '@backbase/ui-ang/logo', '@backbase/ui-ang/card-vendor'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["ui-ang"] = global.backbase["ui-ang"] || {}, global.backbase["ui-ang"]["payment-card"] = {}), global.ng.common, global.ng.core, global.backbase["ui-ang"]["payment-card-number-pipe"], global.backbase["ui-ang"].logo, global.backbase["ui-ang"]["card-vendor"]));
})(this, (function (exports, i3, i0, i4, i1, i2) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i3__namespace = /*#__PURE__*/_interopNamespace(i3);
    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i4__namespace = /*#__PURE__*/_interopNamespace(i4);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);

    /**
     * The prefix for the payment card class names.
     */
    var classNamePrefix = 'bb-payment-card-';
    var PaymentCardComponent = /** @class */ (function () {
        function PaymentCardComponent() {
            /**
             * Configuration of how the card number should be formatted.
             */
            this.paymentCardNumberFormat = i4.DefaultPaymentCardNumberFormat;
        }
        PaymentCardComponent.prototype.ngOnInit = function () {
            if (typeof this.paymentCard === 'undefined') {
                throw new Error("\"paymentCard\" input is required in \"" + this.constructor.name + "\"");
            }
        };
        Object.defineProperty(PaymentCardComponent.prototype, "classNames", {
            get: function () {
                if (this.paymentCard && this.paymentCard.cardName) {
                    var classNamePostfix = this.paymentCard.cardName
                        .toLowerCase()
                        .replace(/[^a-zA-Z0-9 ]/g, ' ')
                        .trim()
                        .replace(/\s/g, '-')
                        .replace(/-{2,}/g, '-');
                    return classNamePostfix ? "" + classNamePrefix + classNamePostfix : undefined;
                }
                return undefined;
            },
            enumerable: false,
            configurable: true
        });
        return PaymentCardComponent;
    }());
    PaymentCardComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: PaymentCardComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    PaymentCardComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: PaymentCardComponent, selector: "bb-payment-card-ui", inputs: { paymentCard: "paymentCard", paymentCardNumberFormat: "paymentCardNumberFormat" }, ngImport: i0__namespace, template: "<div class=\"bb-payment-card\" [ngClass]=\"classNames\">\n  <div class=\"bb-payment-card__container\">\n    <div class=\"bb-payment-card__logo\">\n      <bb-logo-ui type=\"emblem\" [isInverse]=\"true\"></bb-logo-ui>\n    </div>\n    <div class=\"bb-payment-card__vendor\">\n      <bb-card-vendor-ui [vendor]=\"paymentCard?.vendor\"></bb-card-vendor-ui>\n    </div>\n    <div class=\"bb-payment-card__number\">\n      {{ paymentCard?.number | paymentCardNumber: paymentCardNumberFormat }}\n    </div>\n    <div class=\"bb-payment-card__name\">\n      {{ paymentCard?.name }}\n    </div>\n    <div class=\"bb-payment-card__expiration-date\">\n      <div class=\"bb-payment-card__expiration-date-label\" i18n=\"Expiration date label@@paymentCard.expirationDateLabel\">\n        VALID THRU\n      </div>\n      <span class=\"bb-payment-card__expiration-date-date\">\n        {{ paymentCard?.expirationDate | date: 'MM/yy' }}\n      </span>\n    </div>\n  </div>\n</div>\n", components: [{ type: i1__namespace.LogoComponent, selector: "bb-logo-ui", inputs: ["type", "isInverse"] }, { type: i2__namespace.CardVendorComponent, selector: "bb-card-vendor-ui", inputs: ["vendor"] }], directives: [{ type: i3__namespace.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }], pipes: { "paymentCardNumber": i4__namespace.PaymentCardNumberPipe, "date": i3__namespace.DatePipe }, changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: PaymentCardComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-payment-card-ui',
                        templateUrl: './payment-card.component.html',
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], ctorParameters: function () { return []; }, propDecorators: { paymentCard: [{
                    type: i0.Input
                }], paymentCardNumberFormat: [{
                    type: i0.Input
                }] } });

    var PaymentCardModule = /** @class */ (function () {
        function PaymentCardModule() {
        }
        return PaymentCardModule;
    }());
    PaymentCardModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: PaymentCardModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    PaymentCardModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: PaymentCardModule, declarations: [PaymentCardComponent], imports: [i3.CommonModule, i1.LogoModule, i2.CardVendorModule, i4.PaymentCardNumberModule], exports: [PaymentCardComponent] });
    PaymentCardModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: PaymentCardModule, imports: [[i3.CommonModule, i1.LogoModule, i2.CardVendorModule, i4.PaymentCardNumberModule]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: PaymentCardModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [i3.CommonModule, i1.LogoModule, i2.CardVendorModule, i4.PaymentCardNumberModule],
                        declarations: [PaymentCardComponent],
                        exports: [PaymentCardComponent],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.PaymentCardComponent = PaymentCardComponent;
    exports.PaymentCardModule = PaymentCardModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
