// @backbase/ui-ang/bundles/backbase-ui-ang-card-vendor.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('@backbase/ui-ang/card-vendor', ['exports', '@angular/common', '@angular/core'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["ui-ang"] = global.backbase["ui-ang"] || {}, global.backbase["ui-ang"]["card-vendor"] = {}), global.ng.common, global.ng.core));
})(this, (function (exports, i1, i0) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);

    /**
     * The prefix for the vendor class name.
     */
    var classNamePrefix = 'bb-card-vendor-';
    var vendorToClassName = function (vendor) {
        if (typeof vendor !== 'string') {
            return '';
        }
        var classNamePostfix = vendor.trim().toLowerCase().replace(/\W/g, '-').replace(/-{2,}/g, '-');
        return "" + classNamePrefix + classNamePostfix;
    };
    var vendorToTitle = function (vendor) { return (vendor ? vendor.trim().toLowerCase().replace(/-/g, ' ') : ''); };
    /**
     * @name CardVendorComponent
     *
     * @description
     * Component that displays the logo of a card vendor.
     */
    var CardVendorComponent = /** @class */ (function () {
        function CardVendorComponent() {
            this.className = '';
            this.title = '';
        }
        CardVendorComponent.prototype.ngOnInit = function () {
            if (typeof this.vendor !== 'string') {
                throw new Error("\"vendor\" input is required in \"" + this.constructor.name + "\"");
            }
        };
        CardVendorComponent.prototype.ngOnChanges = function (changes) {
            this.className = vendorToClassName(this.vendor);
            this.title = vendorToTitle(this.vendor);
        };
        return CardVendorComponent;
    }());
    CardVendorComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CardVendorComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    CardVendorComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: CardVendorComponent, selector: "bb-card-vendor-ui", inputs: { vendor: "vendor" }, usesOnChanges: true, ngImport: i0__namespace, template: "<div role=\"img\" [attr.aria-label]=\"title\" [title]=\"title\" class=\"bb-card-vendor\" [ngClass]=\"className\"></div>\n", directives: [{ type: i1__namespace.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }], changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CardVendorComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-card-vendor-ui',
                        templateUrl: './card-vendor.component.html',
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], propDecorators: { vendor: [{
                    type: i0.Input
                }] } });

    var CardVendorModule = /** @class */ (function () {
        function CardVendorModule() {
        }
        return CardVendorModule;
    }());
    CardVendorModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CardVendorModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    CardVendorModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CardVendorModule, declarations: [CardVendorComponent], imports: [i1.CommonModule], exports: [CardVendorComponent] });
    CardVendorModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CardVendorModule, imports: [[i1.CommonModule]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CardVendorModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [i1.CommonModule],
                        declarations: [CardVendorComponent],
                        exports: [CardVendorComponent],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.CardVendorComponent = CardVendorComponent;
    exports.CardVendorModule = CardVendorModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
