// @backbase/ui-ang/bundles/backbase-ui-ang-product-item-investment-account.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core'), require('@backbase/ui-ang/payment-card-number-pipe'), require('@backbase/ui-ang/icon'), require('@backbase/ui-ang/amount'), require('@backbase/ui-ang/highlight')) :
    typeof define === 'function' && define.amd ? define('@backbase/ui-ang/product-item-investment-account', ['exports', '@angular/common', '@angular/core', '@backbase/ui-ang/payment-card-number-pipe', '@backbase/ui-ang/icon', '@backbase/ui-ang/amount', '@backbase/ui-ang/highlight'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["ui-ang"] = global.backbase["ui-ang"] || {}, global.backbase["ui-ang"]["product-item-investment-account"] = {}), global.ng.common, global.ng.core, global.backbase["ui-ang"]["payment-card-number-pipe"], global.backbase["ui-ang"].icon, global.backbase["ui-ang"].amount, global.backbase["ui-ang"].highlight));
})(this, (function (exports, i3, i0, i5, i1, i2, i4) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i3__namespace = /*#__PURE__*/_interopNamespace(i3);
    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i5__namespace = /*#__PURE__*/_interopNamespace(i5);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i4__namespace = /*#__PURE__*/_interopNamespace(i4);

    var CustomProductItemInvestmentAccountTitleDirective = /** @class */ (function () {
        function CustomProductItemInvestmentAccountTitleDirective(templateRef) {
            this.templateRef = templateRef;
        }
        return CustomProductItemInvestmentAccountTitleDirective;
    }());
    CustomProductItemInvestmentAccountTitleDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomProductItemInvestmentAccountTitleDirective, deps: [{ token: i0__namespace.TemplateRef }], target: i0__namespace.ɵɵFactoryTarget.Directive });
    CustomProductItemInvestmentAccountTitleDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomProductItemInvestmentAccountTitleDirective, selector: "ng-template[bbCustomProductItemInvestmentAccountTitle]", ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomProductItemInvestmentAccountTitleDirective, decorators: [{
                type: i0.Directive,
                args: [{ selector: 'ng-template[bbCustomProductItemInvestmentAccountTitle]' }]
            }], ctorParameters: function () { return [{ type: i0__namespace.TemplateRef }]; } });
    var CustomProductItemInvestmentAccountDetailsFormatterDirective = /** @class */ (function () {
        function CustomProductItemInvestmentAccountDetailsFormatterDirective(templateRef) {
            this.templateRef = templateRef;
        }
        return CustomProductItemInvestmentAccountDetailsFormatterDirective;
    }());
    CustomProductItemInvestmentAccountDetailsFormatterDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomProductItemInvestmentAccountDetailsFormatterDirective, deps: [{ token: i0__namespace.TemplateRef }], target: i0__namespace.ɵɵFactoryTarget.Directive });
    CustomProductItemInvestmentAccountDetailsFormatterDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: CustomProductItemInvestmentAccountDetailsFormatterDirective, selector: "ng-template[bbCustomProductItemInvestmentAccountDetailsFormatter]", ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CustomProductItemInvestmentAccountDetailsFormatterDirective, decorators: [{
                type: i0.Directive,
                args: [{ selector: 'ng-template[bbCustomProductItemInvestmentAccountDetailsFormatter]' }]
            }], ctorParameters: function () { return [{ type: i0__namespace.TemplateRef }]; } });
    var ProductItemInvestmentAccountComponent = /** @class */ (function () {
        function ProductItemInvestmentAccountComponent() {
            this.hostRef = this;
            /**
             * Configuration of how the product number should be formatted.
             */
            this.productNumberFormat = i5.DefaultPaymentCardNumberFormat;
            /**
             * The active state of card (it adds active class for blue line)
             */
            this.active = true;
            /**
             * The favorite state of card (add favorite icon next to title)
             */
            this.favorite = false;
            /**
             * The favorite icon color (define star icon color)
             */
            this.favoriteIconColor = 'primary';
        }
        return ProductItemInvestmentAccountComponent;
    }());
    ProductItemInvestmentAccountComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ProductItemInvestmentAccountComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    ProductItemInvestmentAccountComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ProductItemInvestmentAccountComponent, selector: "bb-product-item-investment-account-ui", inputs: { title: "title", productNumber: "productNumber", productNumberFormat: "productNumberFormat", currency: "currency", amount: "amount", highlight: "highlight", productDetailsInfo: "productDetailsInfo", active: "active", favorite: "favorite", favoriteIconColor: "favoriteIconColor" }, queries: [{ propertyName: "customProductItemInvestmentAccountTitle", first: true, predicate: CustomProductItemInvestmentAccountTitleDirective, descendants: true, static: true }, { propertyName: "customProductItemInvestmentAccountDetailsFormatter", first: true, predicate: CustomProductItemInvestmentAccountDetailsFormatterDirective, descendants: true, static: true }], ngImport: i0__namespace, template: "<div [class.active]=\"active\" class=\"bb-product-item product-item\">\n  <div class=\"product-item-content\">\n    <div class=\"bb-stack bb-stack--wrap\">\n      <div class=\"bb-stack__item\">\n        <ng-container *ngIf=\"customProductItemInvestmentAccountTitle; else defaultProductItemInvestmentAccountTitle\">\n          <ng-container *ngTemplateOutlet=\"customProductItemInvestmentAccountTitle?.templateRef; context: hostRef\">\n          </ng-container>\n        </ng-container>\n      </div>\n      <div class=\"bb-stack__break bb-stack__break--xs\"></div>\n      <div class=\"bb-stack__item bb-stack__item--push-right bb-stack__item--spacing-none bb-product-item__details\">\n        <ng-container\n          *ngIf=\"\n            customProductItemInvestmentAccountDetailsFormatter;\n            else defaultProductItemInvestmentAccountDetailsFormatter\n          \"\n        >\n          <ng-container\n            *ngTemplateOutlet=\"customProductItemInvestmentAccountDetailsFormatter?.templateRef; context: hostRef\"\n          >\n          </ng-container>\n        </ng-container>\n      </div>\n    </div>\n    <div *ngIf=\"productDetailsInfo?.owners\" class=\"bb-product-item__owners\">\n      <span data-role=\"productDetailsInfo-owners\" class=\"bb-subtitle\">\n        {{ productDetailsInfo?.owners }}\n      </span>\n    </div>\n  </div>\n\n  <ng-template #defaultProductItemInvestmentAccountTitle>\n    <div class=\"bb-account-info\">\n      <div data-role=\"card-title\" class=\"bb-account-info__title\">\n        {{ title }}\n        <bb-icon-ui\n          *ngIf=\"favorite\"\n          size=\"sm\"\n          class=\"bb-account-info__icon\"\n          [color]=\"favoriteIconColor\"\n          data-role=\"favorite-icon\"\n          name=\"star\"\n        >\n        </bb-icon-ui>\n      </div>\n      <div\n        *ngIf=\"productNumber\"\n        data-role=\"card-sub-title\"\n        class=\"bb-subtitle bb-text-support bb-account-info__product-number\"\n      >\n        {{ productNumber | paymentCardNumber: productNumberFormat }}\n      </div>\n    </div>\n  </ng-template>\n\n  <ng-template #defaultProductItemInvestmentAccountDetailsFormatter>\n    <bb-amount-ui\n      class=\"bb-highlight bb-product-item__amount\"\n      [amount]=\"amount\"\n      [currency]=\"currency\"\n      [bbHighlight]=\"highlight\"\n    ></bb-amount-ui>\n  </ng-template>\n</div>\n", components: [{ type: i1__namespace.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i2__namespace.AmountComponent, selector: "bb-amount-ui", inputs: ["currency", "showPlusSign", "mapCurrency", "showPercent", "abbreviate", "decimalPlaces", "trailingZeroes", "amount"] }], directives: [{ type: i3__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i3__namespace.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }, { type: i4__namespace.HighlightDirective, selector: "[bbHighlight]", inputs: ["amount", "bbHighlight", "negativeClass", "positiveClass"] }], pipes: { "paymentCardNumber": i5__namespace.PaymentCardNumberPipe } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ProductItemInvestmentAccountComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-product-item-investment-account-ui',
                        templateUrl: './product-item-investment-account.component.html',
                    }]
            }], propDecorators: { customProductItemInvestmentAccountTitle: [{
                    type: i0.ContentChild,
                    args: [CustomProductItemInvestmentAccountTitleDirective, { static: true }]
                }], customProductItemInvestmentAccountDetailsFormatter: [{
                    type: i0.ContentChild,
                    args: [CustomProductItemInvestmentAccountDetailsFormatterDirective, { static: true }]
                }], title: [{
                    type: i0.Input
                }], productNumber: [{
                    type: i0.Input
                }], productNumberFormat: [{
                    type: i0.Input
                }], currency: [{
                    type: i0.Input
                }], amount: [{
                    type: i0.Input
                }], highlight: [{
                    type: i0.Input
                }], productDetailsInfo: [{
                    type: i0.Input
                }], active: [{
                    type: i0.Input
                }], favorite: [{
                    type: i0.Input
                }], favoriteIconColor: [{
                    type: i0.Input
                }] } });

    var ProductItemInvestmentAccountModule = /** @class */ (function () {
        function ProductItemInvestmentAccountModule() {
        }
        return ProductItemInvestmentAccountModule;
    }());
    ProductItemInvestmentAccountModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ProductItemInvestmentAccountModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ProductItemInvestmentAccountModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ProductItemInvestmentAccountModule, declarations: [ProductItemInvestmentAccountComponent,
            CustomProductItemInvestmentAccountTitleDirective,
            CustomProductItemInvestmentAccountDetailsFormatterDirective], imports: [i3.CommonModule, i2.AmountModule, i5.PaymentCardNumberModule, i4.HighlightModule, i1.IconModule], exports: [ProductItemInvestmentAccountComponent,
            CustomProductItemInvestmentAccountTitleDirective,
            CustomProductItemInvestmentAccountDetailsFormatterDirective] });
    ProductItemInvestmentAccountModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ProductItemInvestmentAccountModule, imports: [[i3.CommonModule, i2.AmountModule, i5.PaymentCardNumberModule, i4.HighlightModule, i1.IconModule]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ProductItemInvestmentAccountModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [i3.CommonModule, i2.AmountModule, i5.PaymentCardNumberModule, i4.HighlightModule, i1.IconModule],
                        declarations: [
                            ProductItemInvestmentAccountComponent,
                            CustomProductItemInvestmentAccountTitleDirective,
                            CustomProductItemInvestmentAccountDetailsFormatterDirective,
                        ],
                        exports: [
                            ProductItemInvestmentAccountComponent,
                            CustomProductItemInvestmentAccountTitleDirective,
                            CustomProductItemInvestmentAccountDetailsFormatterDirective,
                        ],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.CustomProductItemInvestmentAccountDetailsFormatterDirective = CustomProductItemInvestmentAccountDetailsFormatterDirective;
    exports.CustomProductItemInvestmentAccountTitleDirective = CustomProductItemInvestmentAccountTitleDirective;
    exports.ProductItemInvestmentAccountComponent = ProductItemInvestmentAccountComponent;
    exports.ProductItemInvestmentAccountModule = ProductItemInvestmentAccountModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
