// @backbase/ui-ang/bundles/backbase-ui-ang-empty-state.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core'), require('@backbase/ui-ang/icon')) :
    typeof define === 'function' && define.amd ? define('@backbase/ui-ang/empty-state', ['exports', '@angular/common', '@angular/core', '@backbase/ui-ang/icon'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["ui-ang"] = global.backbase["ui-ang"] || {}, global.backbase["ui-ang"]["empty-state"] = {}), global.ng.common, global.ng.core, global.backbase["ui-ang"].icon));
})(this, (function (exports, i2, i0, i1) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);

    /**
     * @name EmptyStateComponent
     *
     * @description
     * Component that displays an empty state.
     */
    var EmptyStateComponent = /** @class */ (function () {
        function EmptyStateComponent() {
            /**
             * Whether the icon should be shown. Defaults to true
             */
            this.showIcon = true;
            /**
             * The modifier for the icon used in the component. Defaults to 'list'
             */
            this.iconModifier = 'list';
            /**
             * The size for the icon used in the component. Defaults to 'xl'
             */
            this.iconSize = 'xl';
        }
        EmptyStateComponent.prototype.ngOnInit = function () {
            if (typeof this.title === 'undefined') {
                throw new Error("\"title\" input is required in \"" + this.constructor.name + "\"");
            }
        };
        return EmptyStateComponent;
    }());
    EmptyStateComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: EmptyStateComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    EmptyStateComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: EmptyStateComponent, selector: "bb-empty-state-ui", inputs: { title: "title", subtitle: "subtitle", showIcon: "showIcon", iconClasses: "iconClasses", iconModifier: "iconModifier", iconSize: "iconSize", iconColor: "iconColor" }, ngImport: i0__namespace, template: "<div class=\"bb-empty-state\">\n  <bb-icon-ui\n    class=\"bb-empty-state__icon\"\n    *ngIf=\"showIcon\"\n    [ngClass]=\"iconClasses\"\n    [name]=\"iconModifier\"\n    [size]=\"iconSize\"\n    [color]=\"iconColor\"\n    [attr.data-role]=\"'icon'\"\n  ></bb-icon-ui>\n  <p class=\"bb-empty-state__title\" data-role=\"empty-state-title\">\n    {{ title }}\n  </p>\n  <p *ngIf=\"subtitle\" class=\"bb-empty-state__message\" data-role=\"empty-state-message\" [innerHTML]=\"subtitle\"></p>\n</div>\n", components: [{ type: i1__namespace.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], directives: [{ type: i2__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i2__namespace.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }], changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: EmptyStateComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-empty-state-ui',
                        templateUrl: './empty-state.component.html',
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], propDecorators: { title: [{
                    type: i0.Input
                }], subtitle: [{
                    type: i0.Input
                }], showIcon: [{
                    type: i0.Input
                }], iconClasses: [{
                    type: i0.Input
                }], iconModifier: [{
                    type: i0.Input
                }], iconSize: [{
                    type: i0.Input
                }], iconColor: [{
                    type: i0.Input
                }] } });

    var EmptyStateModule = /** @class */ (function () {
        function EmptyStateModule() {
        }
        return EmptyStateModule;
    }());
    EmptyStateModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: EmptyStateModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    EmptyStateModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: EmptyStateModule, declarations: [EmptyStateComponent], imports: [i2.CommonModule, i1.IconModule], exports: [EmptyStateComponent] });
    EmptyStateModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: EmptyStateModule, imports: [[i2.CommonModule, i1.IconModule]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: EmptyStateModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [i2.CommonModule, i1.IconModule],
                        declarations: [EmptyStateComponent],
                        exports: [EmptyStateComponent],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.EmptyStateComponent = EmptyStateComponent;
    exports.EmptyStateModule = EmptyStateModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
