// @backbase/ui-ang/bundles/backbase-ui-ang-payment-card-number-pipe.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core'), require('@backbase/foundation-ang/future'), require('@backbase/ui-ang/util'), require('@backbase/ui-ang/account-number-pipe')) :
    typeof define === 'function' && define.amd ? define('@backbase/ui-ang/payment-card-number-pipe', ['exports', '@angular/common', '@angular/core', '@backbase/foundation-ang/future', '@backbase/ui-ang/util', '@backbase/ui-ang/account-number-pipe'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["ui-ang"] = global.backbase["ui-ang"] || {}, global.backbase["ui-ang"]["payment-card-number-pipe"] = {}), global.ng.common, global.ng.core, global.i2, global.backbase["ui-ang"].util, global.backbase["ui-ang"]["account-number-pipe"]));
})(this, (function (exports, common, i0, i2, util, i1) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from, pack) {
        if (pack || arguments.length === 2)
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar)
                        ar = Array.prototype.slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
        return to.concat(ar || Array.prototype.slice.call(from));
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    /**
     * Default payment card number format.
     */
    var DefaultPaymentCardNumberFormat = {
        length: 16,
        maskRange: [0, 12],
        segments: 4,
    };
    /**
     * Lookup object containing payment card number formats for common payment card
     * vendors.
     */
    var PaymentCardNumberVendors = {
        mastercard: {
            length: 16,
            maskRange: [0, 12],
            segments: 4,
        },
        visa: {
            length: 16,
            maskRange: [0, 12],
            segments: 4,
        },
        'visa-debit': {
            length: 16,
            maskRange: [0, 12],
            segments: 4,
        },
        'diners-club': {
            length: 16,
            maskRange: [0, 12],
            segments: 4,
        },
        'american-express': {
            length: 15,
            maskRange: [0, 11],
            segments: [4, 6, 5],
        },
    };
    /**
     * A feature flag that enables numbers masking synchronization with backend
     * If set FE should not apply any masking and rely on the backend masking behavior
     *
     * @internal
     */
    var SYNCHRONIZE_NUMBERS_MASKING = {
        name: 'SYNCHRONIZE_NUMBERS_MASKING',
        description: 'Enables accounts masking synchronization with backend',
    };

    /**
     * @name pad
     * @inner
     * @param value - the string to pad
     * @param length - the length to pad the string to
     * @param ch - the character to pad the string width
     * @param leftPad - whether to pad left (true) or pad right (false)
     *
     * @description
     * Pad a string to specified length with specified character
     */
    function pad(value, length, ch, leftPad) {
        if (ch === void 0) { ch = '0'; }
        if (leftPad === void 0) { leftPad = true; }
        var paddingSize = Math.max(length - value.length, 0);
        var padding = ch.repeat(paddingSize);
        return leftPad ? padding + value : value + padding;
    }
    /**
     * @name mask
     * @inner
     * @param value - the string to mask
     * @param indexStart - the index to start masking from
     * @param indexEnd - the index to end masking
     * @param maskChar - the character to mask with
     *
     * @description
     * Mask part of a string with specified character
     */
    function mask(value, _a, maskChar) {
        var _b = __read(_a, 2), indexStart = _b[0], indexEnd = _b[1];
        if (maskChar === void 0) { maskChar = '*'; }
        var maskedString = '';
        if (indexEnd < 0) {
            var charsToShow = value.slice(indexEnd);
            maskedString = maskChar.repeat(value.length - charsToShow.length) + charsToShow;
        }
        else {
            var maskStr = maskChar.repeat(indexEnd - indexStart);
            maskedString = value.substring(0, indexStart) + maskStr + value.substring(indexEnd);
        }
        return maskedString;
    }
    var PaymentCardNumberPipe = /** @class */ (function () {
        /**
         * @name PaymentCardNumberPipe#constructor
         * @param accountsDisplayingFormatService - AccountsDisplayingFormatService
         * @param featuresService - FeaturesService
         *
         * @internal
         */
        function PaymentCardNumberPipe(accountsDisplayingFormatService, featuresService) {
            this.accountsDisplayingFormatService = accountsDisplayingFormatService;
            this.featuresService = featuresService;
            this.defaultFormat = this.accountsDisplayingFormatService.getDefaultFormat('cardNumber');
        }
        /**
         * @name PaymentCardNumberPipe#transform
         * @param value - the string to transform
         * @param config - the formatting configuration
         * @param maskChar - the character to mask with, deprecated from 6.0.0
         *
         * @description
         * Formats a card number, mask it and split it into segments
         * - do not set config parameter to use with a global accountsFormat configuration
         * - use config parameter to apply a custom accountsFormat configuration
         */
        PaymentCardNumberPipe.prototype.transform = function (value, config, maskChar) {
            if (maskChar === void 0) { maskChar = '•'; }
            var displayingFormat = config !== null && config !== void 0 ? config : this.defaultFormat;
            var formattedNumber = value.toString();
            if (config && !this.featuresService.isEnabled(SYNCHRONIZE_NUMBERS_MASKING)) {
                formattedNumber = pad(formattedNumber, config.length, maskChar);
                formattedNumber = mask(formattedNumber, config.maskRange, maskChar);
            }
            return util.segment(formattedNumber, displayingFormat.segments);
        };
        return PaymentCardNumberPipe;
    }());
    PaymentCardNumberPipe.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: PaymentCardNumberPipe, deps: [{ token: i1__namespace.AccountsDisplayingFormatService }, { token: i2__namespace.FeaturesService }], target: i0__namespace.ɵɵFactoryTarget.Pipe });
    PaymentCardNumberPipe.ɵpipe = i0__namespace.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: PaymentCardNumberPipe, name: "paymentCardNumber" });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: PaymentCardNumberPipe, decorators: [{
                type: i0.Pipe,
                args: [{
                        name: 'paymentCardNumber',
                    }]
            }], ctorParameters: function () { return [{ type: i1__namespace.AccountsDisplayingFormatService }, { type: i2__namespace.FeaturesService }]; } });

    var PaymentCardNumberModule = /** @class */ (function () {
        function PaymentCardNumberModule() {
        }
        return PaymentCardNumberModule;
    }());
    PaymentCardNumberModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: PaymentCardNumberModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    PaymentCardNumberModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: PaymentCardNumberModule, declarations: [PaymentCardNumberPipe], imports: [common.CommonModule, i2__namespace.FutureFeaturesModule], exports: [PaymentCardNumberPipe] });
    PaymentCardNumberModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: PaymentCardNumberModule, imports: [[common.CommonModule, i2.FutureFeaturesModule.allowOptIn([SYNCHRONIZE_NUMBERS_MASKING])]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: PaymentCardNumberModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [common.CommonModule, i2.FutureFeaturesModule.allowOptIn([SYNCHRONIZE_NUMBERS_MASKING])],
                        declarations: [PaymentCardNumberPipe],
                        exports: [PaymentCardNumberPipe],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.DefaultPaymentCardNumberFormat = DefaultPaymentCardNumberFormat;
    exports.PaymentCardNumberModule = PaymentCardNumberModule;
    exports.PaymentCardNumberPipe = PaymentCardNumberPipe;
    exports.SYNCHRONIZE_NUMBERS_MASKING = SYNCHRONIZE_NUMBERS_MASKING;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
