// @backbase/ui-ang/bundles/backbase-ui-ang-file-attachment.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('rxjs'), require('rxjs/operators'), require('@backbase/ui-ang/icon'), require('@backbase/ui-ang/loading-indicator'), require('@backbase/ui-ang/keyboard-click-directive'), require('@backbase/ui-ang/button')) :
    typeof define === 'function' && define.amd ? define('@backbase/ui-ang/file-attachment', ['exports', '@angular/core', '@angular/common', 'rxjs', 'rxjs/operators', '@backbase/ui-ang/icon', '@backbase/ui-ang/loading-indicator', '@backbase/ui-ang/keyboard-click-directive', '@backbase/ui-ang/button'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["ui-ang"] = global.backbase["ui-ang"] || {}, global.backbase["ui-ang"]["file-attachment"] = {}), global.ng.core, global.ng.common, global.rxjs, global.rxjs.operators, global.backbase["ui-ang"].icon, global.backbase["ui-ang"]["loading-indicator"], global.backbase["ui-ang"]["keyboard-click-directive"], global.backbase["ui-ang"].button));
})(this, (function (exports, i0, i4, rxjs, operators, i1, i2, i3, i5) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i4__namespace = /*#__PURE__*/_interopNamespace(i4);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i3__namespace = /*#__PURE__*/_interopNamespace(i3);
    var i5__namespace = /*#__PURE__*/_interopNamespace(i5);

    /**
     * @name FileAttachmentComponent
     *
     * @description
     * Component displays file attachment.
     *
     * @dynamic (to suppress error with resolving statics declarations during compilation)
     */
    var FileAttachmentComponent = /** @class */ (function () {
        function FileAttachmentComponent(document, renderer) {
            this.document = document;
            this.renderer = renderer;
            /**
             * Loading indicator flag. Defaults to false.
             */
            this.loading = false;
            /**
             * Show delete button flag. Defaults to false.
             */
            this.deletable = false;
            /**
             * Disabled state flag. Defaults to false.
             */
            this.disabled = false;
            /**
             * The flag to indicate whether the file-attachment should fill the container that it is in. Defaults to 'false'.
             */
            this.block = false;
            /**
             * Event emitted on delete button click.
             */
            this.delete = new i0.EventEmitter();
            /**
             * File type which is determined by file extension. Defaults to 'unknown'.
             */
            this.fileType = 'unknown';
        }
        Object.defineProperty(FileAttachmentComponent.prototype, "name", {
            /**
             * Get name.
             */
            get: function () {
                return this._name;
            },
            /**
             * Set name (required).
             */
            set: function (value) {
                if (value === null || value === void 0 ? void 0 : value.trim()) {
                    this._name = value;
                    this.fileType = this.getFileType(value);
                }
                else {
                    throw new Error("\"name\" input is invalid in \"" + this.constructor.name + "\"");
                }
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(FileAttachmentComponent.prototype, "size", {
            /**
             * Set file size (required).
             */
            set: function (value) {
                if (value !== undefined) {
                    var valueAsNumber = Math.trunc(Number(value));
                    this._sizeInUnits = this.formatBytes(valueAsNumber, 2);
                }
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(FileAttachmentComponent.prototype, "sizeInUnits", {
            /**
             * Get file size in information units (Bytes, Kbs, etc.).
             */
            get: function () {
                return this._sizeInUnits;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(FileAttachmentComponent.prototype, "fileContent", {
            /**
             * Get file content.
             */
            get: function () {
                return this._fileContent;
            },
            /**
             * Set file content (required if disabled property is not set to true).
             */
            set: function (value) {
                if (value && rxjs.isObservable(value)) {
                    this._fileContent = value;
                }
                else {
                    throw new Error("\"fileContent\" stream input is invalid in \"" + this.constructor.name + "\"");
                }
            },
            enumerable: false,
            configurable: true
        });
        FileAttachmentComponent.prototype.ngOnInit = function () {
            if (!this._name) {
                throw new Error("\"name\" input is required in \"" + this.constructor.name + "\"");
            }
            if (!this.disabled && !this._fileContent) {
                throw new Error("\"fileContent\" input is required in \"" + this.constructor.name + "\"");
            }
        };
        /**
         * Download attachment method.
         *
         * @param $event Download button click event
         */
        FileAttachmentComponent.prototype.onDownload = function ($event) {
            var _this = this;
            $event.stopPropagation();
            if (this.fileContent) {
                this.fileContent.pipe(operators.take(1)).subscribe(function (response) {
                    if (!_this.document.defaultView) {
                        return;
                    }
                    var window = _this.document.defaultView;
                    var blob = new window.Blob([response]);
                    if (window.navigator.msSaveBlob) {
                        window.navigator.msSaveBlob(blob, _this.name);
                    }
                    else {
                        var fileURL = window.URL.createObjectURL(blob);
                        var link = _this.renderer.createElement('a');
                        link.setAttribute('href', fileURL);
                        link.setAttribute('target', '_blank');
                        link.setAttribute('download', _this.name || '');
                        link.click();
                        URL.revokeObjectURL(fileURL);
                    }
                });
            }
        };
        /**
         * Delete attachment event emitter.
         *
         * @param event Delete button click event.
         */
        FileAttachmentComponent.prototype.onDelete = function (event) {
            event.stopPropagation();
            this.delete.emit();
        };
        FileAttachmentComponent.prototype.getFileType = function (fileName) {
            return fileName.includes('.') ? fileName.toLowerCase().split('.').pop() || 'unknown' : 'unknown';
        };
        FileAttachmentComponent.prototype.formatBytes = function (bytes, decimals) {
            if (bytes === 0) {
                return {
                    value: 0,
                    unit: 'Bytes',
                };
            }
            var k = 1024;
            var dm = decimals <= 0 ? 0 : decimals || 2;
            var sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB'];
            var i = Math.floor(Math.log(bytes) / Math.log(k));
            return {
                value: parseFloat((bytes / Math.pow(k, i)).toFixed(dm)),
                unit: sizes[i],
            };
        };
        return FileAttachmentComponent;
    }());
    FileAttachmentComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: FileAttachmentComponent, deps: [{ token: i4.DOCUMENT }, { token: i0__namespace.Renderer2 }], target: i0__namespace.ɵɵFactoryTarget.Component });
    FileAttachmentComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: FileAttachmentComponent, selector: "bb-file-attachment-ui", inputs: { name: "name", size: "size", fileContent: "fileContent", loading: "loading", deletable: "deletable", disabled: "disabled", block: "block" }, outputs: { delete: "delete" }, ngImport: i0__namespace, template: "<div\n  class=\"bb-attachment-ui bb-card bb-card--sm\"\n  [class.bb-attachment-ui--block]=\"block\"\n  [attr.role]=\"disabled || deletable ? null : 'button'\"\n  [tabindex]=\"disabled ? -1 : 0\"\n  data-role=\"download-attachment\"\n  aria-describedby=\"file-attachment-card-body\"\n  (bbKeyboardClick)=\"!disabled && onDownload($event)\"\n>\n  <div class=\"bb-stack bb-stack--align-top bb-card__body\" id=\"file-attachment-card-body\">\n    <div class=\"bb-stack__item\">\n      <em\n        data-role=\"file-type-icon\"\n        [attr.aria-label]=\"fileType\"\n        [ngClass]=\"'bb-icon-file-' + fileType\"\n        class=\"bb-icon bb-icon--md bb-message-attachment__content\"\n        [class.bb-message-attachment__content--disabled]=\"loading\"\n      ></em>\n    </div>\n    <div class=\"bb-stack__item text-truncate\" [class.bb-message-attachment__content--disabled]=\"loading\">\n      <span class=\"bb-block bb-block--xs\" data-role=\"attachment-name\">{{ name }}</span>\n      <div *ngIf=\"sizeInUnits\">\n        <span class=\"bb-text-support\" data-role=\"attachment-size\"\n          >{{ sizeInUnits.value }}\n          <span i18n=\"File size|Attachment file size in units@@fileAttachment.fileSize\"\n            >{ sizeInUnits.unit, select, Bytes {Bytes} KB {KB} MB {MB} GB {GB} TB {TB} }</span\n          ></span\n        >\n      </div>\n    </div>\n    <div *ngIf=\"loading || deletable\" class=\"bb-stack__item bb-stack__item--push-right\">\n      <button\n        bbButton\n        color=\"link-text\"\n        *ngIf=\"deletable && !loading\"\n        class=\"bb-text-support\"\n        aria-label=\"Delete file\"\n        i18n-aria-label=\"Delete file button@@fileAttachment.deleteButton.ariaLabel\"\n        data-role=\"file-attachment-delete-button\"\n        (click)=\"onDelete($event)\"\n      >\n        <bb-icon-ui name=\"times\"></bb-icon-ui>\n      </button>\n\n      <bb-loading-indicator-ui\n        *ngIf=\"loading\"\n        class=\"bb-message-attachment__loader\"\n        loaderSize=\"sm\"\n        aria-label=\"Loading\"\n        i18n-aria-label=\"Loading|File attachment loading indicator@@fileAttachment.loadingIndicator.ariaLabel\"\n      >\n      </bb-loading-indicator-ui>\n    </div>\n  </div>\n</div>\n", components: [{ type: i1__namespace.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i2__namespace.LoadingIndicatorComponent, selector: "bb-loading-indicator-ui", inputs: ["text", "loaderSize", "showDelay", "hasBackground", "inline"] }], directives: [{ type: i3__namespace.KeyboardClickDirective, selector: "[bbKeyboardClick]", outputs: ["bbKeyboardClick"] }, { type: i4__namespace.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { type: i4__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i5__namespace.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }], changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: FileAttachmentComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-file-attachment-ui',
                        templateUrl: './file-attachment.component.html',
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], ctorParameters: function () {
            return [{ type: Document, decorators: [{
                            type: i0.Inject,
                            args: [i4.DOCUMENT]
                        }] }, { type: i0__namespace.Renderer2 }];
        }, propDecorators: { name: [{
                    type: i0.Input
                }], size: [{
                    type: i0.Input
                }], fileContent: [{
                    type: i0.Input
                }], loading: [{
                    type: i0.Input
                }], deletable: [{
                    type: i0.Input
                }], disabled: [{
                    type: i0.Input
                }], block: [{
                    type: i0.Input
                }], delete: [{
                    type: i0.Output
                }] } });

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from, pack) {
        if (pack || arguments.length === 2)
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar)
                        ar = Array.prototype.slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
        return to.concat(ar || Array.prototype.slice.call(from));
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    var uiModules = [i5.ButtonModule, i1.IconModule, i3.KeyboardClickModule, i2.LoadingIndicatorModule];
    var FileAttachmentModule = /** @class */ (function () {
        function FileAttachmentModule() {
        }
        return FileAttachmentModule;
    }());
    FileAttachmentModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: FileAttachmentModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    FileAttachmentModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: FileAttachmentModule, declarations: [FileAttachmentComponent], imports: [i4.CommonModule, i5.ButtonModule, i1.IconModule, i3.KeyboardClickModule, i2.LoadingIndicatorModule], exports: [FileAttachmentComponent] });
    FileAttachmentModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: FileAttachmentModule, imports: [__spreadArray([i4.CommonModule], __read(uiModules))] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: FileAttachmentModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: __spreadArray([i4.CommonModule], __read(uiModules)),
                        declarations: [FileAttachmentComponent],
                        exports: [FileAttachmentComponent],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.FileAttachmentComponent = FileAttachmentComponent;
    exports.FileAttachmentModule = FileAttachmentModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
