// @backbase/payment-orders-ang/esm2015/lib/ui/pipes/highlight/highlight.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BBHighlightPipe } from './highlight.pipe';
import * as i0 from "@angular/core";
export class BBHighlightModule {
}
/** @nocollapse */ BBHighlightModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BBHighlightModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ BBHighlightModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BBHighlightModule, declarations: [BBHighlightPipe], imports: [CommonModule], exports: [BBHighlightPipe] });
/** @nocollapse */ BBHighlightModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BBHighlightModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BBHighlightModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [BBHighlightPipe],
                    imports: [CommonModule],
                    exports: [BBHighlightPipe],
                }]
        }] });
