// @backbase/ui-ang/esm2015/_example/loading-indicator/showcase/loading-indicator-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { LoadingIndicatorShowcaseComponent } from './loading-indicator-showcase.component';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import * as i0 from "@angular/core";
export class LoadingIndicatorShowcaseModule {
}
LoadingIndicatorShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoadingIndicatorShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LoadingIndicatorShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoadingIndicatorShowcaseModule, declarations: [LoadingIndicatorShowcaseComponent], imports: [CommonModule, LoadingIndicatorModule] });
LoadingIndicatorShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoadingIndicatorShowcaseModule, imports: [[CommonModule, LoadingIndicatorModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoadingIndicatorShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, LoadingIndicatorModule],
                    declarations: [LoadingIndicatorShowcaseComponent],
                    entryComponents: [LoadingIndicatorShowcaseComponent],
                }]
        }] });
