// @backbase/ui-ang/esm2015/_example/loading-indicator/showcase/loading-indicator-showcase.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/loading-indicator";
import * as i2 from "@angular/common";
let LoadingIndicatorShowcaseComponent = class LoadingIndicatorShowcaseComponent {
    constructor() {
        this.loadingIndicators = [
            {
                title: 'Small',
                size: 'sm',
                text: null,
            },
            {
                title: 'Medium',
                size: 'md',
                text: null,
            },
            {
                title: 'Large',
                size: 'lg',
                text: null,
            },
            {
                title: 'Page loader',
                size: 'lg',
                text: 'Loading...',
            },
        ];
    }
};
LoadingIndicatorShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoadingIndicatorShowcaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
LoadingIndicatorShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: LoadingIndicatorShowcaseComponent, selector: "bb-loading-indicator-ui-showcase", ngImport: i0, template: "<ng-container *ngFor=\"let loadingIndicator of loadingIndicators\">\n  <div class=\"bb-block bb-block--lg\">\n    <h4 class=\"bb-block bb-block--lg\">{{ loadingIndicator.title }}</h4>\n    <bb-loading-indicator-ui\n      [loaderSize]=\"loadingIndicator.size\"\n      [text]=\"loadingIndicator.text\"\n    ></bb-loading-indicator-ui>\n  </div>\n</ng-container>\n\n<div class=\"bb-block bb-block--lg\">\n  <h4 class=\"bb-block bb-block--lg\">Inline</h4>\n  <p>\n    Some text\n    <bb-loading-indicator-ui [inline]=\"true\" loaderSize=\"sm\"></bb-loading-indicator-ui>\n  </p>\n</div>\n\n<div class=\"bb-block bb-block--lg\">\n  <h4 class=\"bb-block bb-block--lg\">Inline with text</h4>\n  <p>\n    Lorem ipsum\n    <bb-loading-indicator-ui [inline]=\"true\" loaderSize=\"sm\" text=\"Loading...\"></bb-loading-indicator-ui>\n  </p>\n</div>\n", components: [{ type: i1.LoadingIndicatorComponent, selector: "bb-loading-indicator-ui", inputs: ["text", "loaderSize", "showDelay", "hasBackground", "inline"] }], directives: [{ type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }] });
LoadingIndicatorShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
], LoadingIndicatorShowcaseComponent);
export { LoadingIndicatorShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoadingIndicatorShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-loading-indicator-ui-showcase',
                    templateUrl: './loading-indicator-showcase.component.html',
                }]
        }] });
