// @backbase/accounts-transactions-journeys-shared-ui-account-ang/esm2015/lib/pipes/has-external-info/has-external-info.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HasExternalInfoPipe } from './has-external-info.pipe';
import * as i0 from "@angular/core";
const pipes = [HasExternalInfoPipe];
export class HasExternalInfoPipeModule {
}
/** @nocollapse */ HasExternalInfoPipeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HasExternalInfoPipeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ HasExternalInfoPipeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HasExternalInfoPipeModule, declarations: [HasExternalInfoPipe], imports: [CommonModule], exports: [HasExternalInfoPipe] });
/** @nocollapse */ HasExternalInfoPipeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HasExternalInfoPipeModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HasExternalInfoPipeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [...pipes],
                    exports: [...pipes],
                }]
        }] });
