// @backbase/user-common/esm2015/src/services/user-data/user-data.service.js
import { Injectable } from '@angular/core';
import { of } from 'rxjs';
import { catchError, map } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/user";
export class UserDataService {
    constructor(userManagementService, identityManagementService) {
        this.userManagementService = userManagementService;
        this.identityManagementService = identityManagementService;
    }
    checkUsernameAvailability(externalId) {
        return this.userManagementService.checkUserExists({ externalId }, 'response').pipe(map(this.isNotFound), catchError((error) => of(this.isNotFound(error))));
    }
    createUser(createIdentityRequest) {
        return this.identityManagementService.createIdentity({ createIdentityRequest });
    }
    getPendingApprovalsByInternalIdAndType(internalId, type) {
        return this.getPendingApprovalsByInternalId(internalId).pipe(map((approvals) => approvals.filter((approval) => approval.type === type)));
    }
    getPendingApprovalsByInternalId(internalId) {
        return this.identityManagementService.getIdentityPendingApprovals({ internalId });
    }
    getUsersIdentitiesSessionsByInternalId(internalId) {
        return this.identityManagementService.getSessions({ internalId });
    }
    updateUsersIdentitiesSessionsLogoutRecordByInternalId(internalId, sessionIds = []) {
        return this.identityManagementService.logoutSessions({
            internalId,
            identityLogoutRequest: { sessionIds },
        });
    }
    updateUserActions(internalId, actions) {
        return this.identityManagementService.sendRequiredActions({
            internalId,
            identityRequiredActionsRequest: { actions },
        });
    }
    updateUserRecordActions(internalId, updateIdentityRequest) {
        return this.identityManagementService.updateIdentity({
            internalId,
            updateIdentityRequest,
        });
    }
    changeUserPassword(existingPassword, newPassword) {
        return this.identityManagementService.changePassword({
            changePasswordRequest: {
                existingPassword,
                newPassword,
            },
        });
    }
    getUserDetails(internalId) {
        return this.identityManagementService.getIdentity({ internalId });
    }
    isNotFound(response) {
        return response.status === 404;
    }
}
UserDataService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserDataService, deps: [{ token: i1.UserManagementService }, { token: i1.IdentityManagementService }], target: i0.ɵɵFactoryTarget.Injectable });
UserDataService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserDataService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserDataService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.UserManagementService }, { type: i1.IdentityManagementService }]; } });
