// @backbase/payord-stop-checks-widget-ang/esm2015/lib/models/data.js
export const arrangementsPageSize = 100;
export const currencyFallback = 'USD';
export var RequestStatus;
(function (RequestStatus) {
    RequestStatus["INIT"] = "INIT";
    RequestStatus["REVIEW"] = "REVIEW";
    RequestStatus["REQUEST_IS_SUBMITTED"] = "REQUEST_IS_SUBMITTED";
    RequestStatus["EDIT_MODE"] = "EDIT_MODE";
})(RequestStatus || (RequestStatus = {}));
export var StopCheckReason;
(function (StopCheckReason) {
    StopCheckReason["INSUFFICIENT_FUNDS"] = "Insufficient funds";
    StopCheckReason["INCORRECT_AMOUNT"] = "Incorrect amount or other details";
    StopCheckReason["LOST_OR_STOLEN"] = "Lost or stolen checks or check books";
    StopCheckReason["WRONG_ADDRESS"] = "Checks mailed to wrong address";
    StopCheckReason["DISPUTE_OVER_PURCHASES"] = "Dispute over purchases";
    StopCheckReason["LOST_IN_MAIL"] = "Lost in Mail";
    StopCheckReason["OTHER"] = "Other";
})(StopCheckReason || (StopCheckReason = {}));
export var StopCheckRequestType;
(function (StopCheckRequestType) {
    StopCheckRequestType["SINGLE"] = "single";
    StopCheckRequestType["MULTIPLE"] = "multiple";
})(StopCheckRequestType || (StopCheckRequestType = {}));
