// @backbase/notifications-http-ang/esm2015/api/notifications.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { NOTIFICATIONS_BASE_PATH } from '../variables';
import { NotificationsConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class NotificationsHttpService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new NotificationsConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    deleteNotification(requestParameters, observe = 'body', reportProgress = false, options) {
        const _id = requestParameters["id"];
        if (_id === null || _id === undefined) {
            throw new Error('Required parameter id was null or undefined when calling deleteNotification.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.delete(`${this.configuration.basePath}/client-api/v2/notifications/${encodeURIComponent(String(_id))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    deleteNotificationUrl(requestParameters) {
        const _id = requestParameters["id"];
        if (_id === null || _id === undefined) {
            throw new Error('Required parameter id was null or undefined when calling deleteNotification.');
        }
        return `${this.configuration.basePath}/client-api/v2/notifications/${encodeURIComponent(String(_id))}`;
    }
    getCategory(observe = 'body', reportProgress = false, options) {
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/notifications/categories`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getCategoryUrl() {
        return `${this.configuration.basePath}/client-api/v2/notifications/categories`;
    }
    getNotifications(requestParameters, observe = 'body', reportProgress = false, options) {
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        const _fromDate = requestParameters["fromDate"];
        const _toDate = requestParameters["toDate"];
        const _levels = requestParameters["levels"];
        const _read = requestParameters["read"];
        const _origins = requestParameters["origins"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_fromDate !== undefined && _fromDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _fromDate, 'fromDate');
        }
        if (_toDate !== undefined && _toDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _toDate, 'toDate');
        }
        if (_levels) {
            _levels.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'levels');
            });
        }
        if (_read !== undefined && _read !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _read, 'read');
        }
        if (_origins) {
            _origins.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'origins');
            });
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/notifications`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getNotificationsUrl(requestParameters) {
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        const _fromDate = requestParameters["fromDate"];
        const _toDate = requestParameters["toDate"];
        const _levels = requestParameters["levels"];
        const _read = requestParameters["read"];
        const _origins = requestParameters["origins"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_fromDate !== undefined && _fromDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _fromDate, 'fromDate');
        }
        if (_toDate !== undefined && _toDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _toDate, 'toDate');
        }
        if (_levels) {
            _levels.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'levels');
            });
        }
        if (_read !== undefined && _read !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _read, 'read');
        }
        if (_origins) {
            _origins.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'origins');
            });
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/notifications${queryString ? `?${queryString}` : ''}`;
    }
    getNotificationsDigest(observe = 'body', reportProgress = false, options) {
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/notifications/stream/digest`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getNotificationsDigestUrl() {
        return `${this.configuration.basePath}/client-api/v2/notifications/stream/digest`;
    }
    getNotificationsStream(requestParameters, observe = 'body', reportProgress = false, options) {
        const _interval = requestParameters["interval"];
        const _levels = requestParameters["levels"];
        const _origins = requestParameters["origins"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_interval !== undefined && _interval !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _interval, 'interval');
        }
        if (_levels !== undefined && _levels !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _levels, 'levels');
        }
        if (_origins) {
            _origins.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'origins');
            });
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/notifications/stream`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getNotificationsStreamUrl(requestParameters) {
        const _interval = requestParameters["interval"];
        const _levels = requestParameters["levels"];
        const _origins = requestParameters["origins"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_interval !== undefined && _interval !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _interval, 'interval');
        }
        if (_levels !== undefined && _levels !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _levels, 'levels');
        }
        if (_origins) {
            _origins.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'origins');
            });
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/notifications/stream${queryString ? `?${queryString}` : ''}`;
    }
    getUnreadCount(requestParameters, observe = 'body', reportProgress = false, options) {
        const _levels = requestParameters["levels"];
        const _origins = requestParameters["origins"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_levels) {
            _levels.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'levels');
            });
        }
        if (_origins) {
            _origins.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'origins');
            });
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/notifications/unread-count`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getUnreadCountUrl(requestParameters) {
        const _levels = requestParameters["levels"];
        const _origins = requestParameters["origins"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_levels) {
            _levels.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'levels');
            });
        }
        if (_origins) {
            _origins.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'origins');
            });
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/notifications/unread-count${queryString ? `?${queryString}` : ''}`;
    }
    putMarkAsRead(requestParameters, observe = 'body', reportProgress = false, options) {
        const _id = requestParameters["id"];
        if (_id === null || _id === undefined) {
            throw new Error('Required parameter id was null or undefined when calling putMarkAsRead.');
        }
        const _markAsReadPutRequestBody = requestParameters["markAsReadPutRequestBody"];
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v2/notifications/${encodeURIComponent(String(_id))}/read`, _markAsReadPutRequestBody, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    putMarkAsReadUrl(requestParameters) {
        const _id = requestParameters["id"];
        if (_id === null || _id === undefined) {
            throw new Error('Required parameter id was null or undefined when calling putMarkAsRead.');
        }
        const _markAsReadPutRequestBody = requestParameters["markAsReadPutRequestBody"];
        return `${this.configuration.basePath}/client-api/v2/notifications/${encodeURIComponent(String(_id))}/read`;
    }
    putRead(observe = 'body', reportProgress = false, options) {
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v2/notifications/read`, null, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    putReadUrl() {
        return `${this.configuration.basePath}/client-api/v2/notifications/read`;
    }
}
NotificationsHttpService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: NotificationsHttpService, deps: [{ token: i1.HttpClient }, { token: NOTIFICATIONS_BASE_PATH, optional: true }, { token: i2.NotificationsConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
NotificationsHttpService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: NotificationsHttpService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: NotificationsHttpService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [NOTIFICATIONS_BASE_PATH]
                }] }, { type: i2.NotificationsConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
