// @backbase/consent-list-widget-ang/esm2015/src/consent-list-widget.component.js
import { Component, Directive, Output, EventEmitter } from '@angular/core';
import { BbTemplate } from '@backbase/foundation-ang/core';
import * as i0 from "@angular/core";
import * as i1 from "./consent-list-widget.service";
import * as i2 from "./consent-list/consent-list.component";
import * as i3 from "@backbase/ui-ang/loading-indicator";
import * as i4 from "./consent-list-error/consent-list-error.component";
import * as i5 from "@backbase/ui-ang/empty-state";
import * as i6 from "@angular/common";
export class ConsentListWidgetComponent {
    constructor(service) {
        this.service = service;
        /**
         * Component context
         */
        this.hostRef = this;
        /**
         * Emitter that is fired when selecting consent
         */
        this.consentId = new EventEmitter();
        /**
         * Stream of loading spinner visibility flag
         */
        this.loading = this.service.loading;
        /**
         * Stream of Consent list errors
         */
        this.error = this.service.error;
        /**
         * Stream of consent items
         */
        this.consents = this.service.consents;
    }
    /**
     * Select consent click event handler.
     * @param consentId - consent ID
     */
    selectConsent(consentId) {
        this.consentId.emit(consentId);
    }
}
ConsentListWidgetComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentListWidgetComponent, deps: [{ token: i1.ConsentListService }], target: i0.ɵɵFactoryTarget.Component });
ConsentListWidgetComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: ConsentListWidgetComponent, selector: "bb-consent-list-widget", outputs: { consentId: "consentId" }, ngImport: i0, template: "<ng-container bbConsentListWidgetCustomizable [bbHostRef]=\"hostRef\" [bbTemplateContext]=\"consents | async\">\n</ng-container>\n\n<ng-template bbConsentListWidgetCustomizable let-hostRef let-consents=\"context\">\n  <ng-container *ngIf=\"consents; else loadingTemplate\">\n    <ng-container *ngIf=\"!(hostRef.loading | async) as loading; else loadingTemplate\">\n      <ng-container *ngIf=\"!(hostRef.error | async); else errorTemplate\">\n        <ng-container *ngIf=\"consents?.length > 0; else emptyTemplate\">\n          <bb-consent-list-component [consentsList]=\"consents\" (selectConsent)=\"hostRef.selectConsent($event)\">\n          </bb-consent-list-component>\n        </ng-container>\n      </ng-container>\n    </ng-container>\n  </ng-container>\n\n  <ng-template #loadingTemplate>\n    <div class=\"bb-state-container\">\n      <bb-loading-indicator-ui></bb-loading-indicator-ui>\n    </div>\n  </ng-template>\n\n  <ng-template #errorTemplate>\n    <div class=\"bb-state-container\">\n      <bb-consent-list-error [errorKind]=\"$any(hostRef.error | async)?.kind\"> </bb-consent-list-error>\n    </div>\n  </ng-template>\n\n  <ng-template #emptyTemplate>\n    <div class=\"bb-state-container\">\n      <bb-empty-state-ui\n        i18n-title=\"No Consents|No Consents@@consent-list-widget.list-empty.title\"\n        title=\"No Consents\"\n        i18n-subtitle=\"\n          You haven't given any consents|You haven't given any consents to third\n          parties@@consent-list-widget.list-empty.subtitle\"\n        subtitle=\"You haven't given any consents to third parties\"\n        iconModifier=\"assignment-turned-in\">\n      </bb-empty-state-ui>\n    </div>\n  </ng-template>\n</ng-template>\n", components: [{ type: i0.forwardRef(function () { return i2.ConsentListComponent; }), selector: "bb-consent-list-component", inputs: ["consentsList"], outputs: ["selectConsent"] }, { type: i0.forwardRef(function () { return i3.LoadingIndicatorComponent; }), selector: "bb-loading-indicator-ui", inputs: ["text", "loaderSize", "showDelay", "hasBackground", "inline"] }, { type: i0.forwardRef(function () { return i4.ConsentListErrorComponent; }), selector: "bb-consent-list-error", inputs: ["errorKind"] }, { type: i0.forwardRef(function () { return i5.EmptyStateComponent; }), selector: "bb-empty-state-ui", inputs: ["title", "subtitle", "showIcon", "iconClasses", "iconModifier", "iconSize", "iconColor"] }], directives: [{ type: i0.forwardRef(function () { return ConsentListWidgetCustomizableDirective; }), selector: "[bbConsentListWidgetCustomizable]" }, { type: i0.forwardRef(function () { return i6.NgIf; }), selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "async": i0.forwardRef(function () { return i6.AsyncPipe; }) } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentListWidgetComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-consent-list-widget',
                    templateUrl: './consent-list-widget.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.ConsentListService }]; }, propDecorators: { consentId: [{
                type: Output
            }] } });
export class ConsentListWidgetCustomizableDirective extends BbTemplate {
}
ConsentListWidgetCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentListWidgetCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
ConsentListWidgetCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.15", type: ConsentListWidgetCustomizableDirective, selector: "[bbConsentListWidgetCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentListWidgetCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbConsentListWidgetCustomizable]',
                }]
        }] });
