// @backbase/consent-list-widget-ang/esm2015/src/consent-list-widget.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { ConsentUiModule } from '@backbase/consent-ui';
import { BadgeModule } from '@backbase/ui-ang/badge';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { IconModule } from '@backbase/ui-ang/icon';
import { BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import { ConsentListWidgetComponent, ConsentListWidgetCustomizableDirective } from './consent-list-widget.component';
import { ConsentListComponent } from './consent-list/consent-list.component';
import { ConsentItemComponent, ConsentListItemCustomizableDirective } from './consent-item/consent-item.component';
import { ConsentListErrorComponent } from './consent-list-error/consent-list-error.component';
import { ConsentListService } from './consent-list-widget.service';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const components = [ConsentListWidgetComponent, ConsentListComponent, ConsentItemComponent, ConsentListErrorComponent];
const directives = [ConsentListWidgetCustomizableDirective, ConsentListItemCustomizableDirective];
const uiModules = [
    ConsentUiModule,
    BadgeModule,
    EmptyStateModule,
    LoadingIndicatorModule,
    IconModule,
    BbDatePipeModule,
];
export class ConsentListWidgetModule {
}
ConsentListWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentListWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ConsentListWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentListWidgetModule, declarations: [ConsentListWidgetComponent, ConsentListComponent, ConsentItemComponent, ConsentListErrorComponent, ConsentListWidgetCustomizableDirective, ConsentListItemCustomizableDirective], imports: [CommonModule, ConsentUiModule,
        BadgeModule,
        EmptyStateModule,
        LoadingIndicatorModule,
        IconModule,
        BbDatePipeModule, i1.BackbaseCoreModule], exports: [ConsentListWidgetComponent, ConsentListComponent, ConsentItemComponent, ConsentListErrorComponent, ConsentListWidgetCustomizableDirective, ConsentListItemCustomizableDirective, ConsentUiModule,
        BadgeModule,
        EmptyStateModule,
        LoadingIndicatorModule,
        IconModule,
        BbDatePipeModule] });
ConsentListWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentListWidgetModule, providers: [ConsentListService], imports: [[
            CommonModule,
            ...uiModules,
            BackbaseCoreModule.withConfig({
                classMap: { ConsentListWidgetComponent },
            }),
        ], ConsentUiModule,
        BadgeModule,
        EmptyStateModule,
        LoadingIndicatorModule,
        IconModule,
        BbDatePipeModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentListWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        ...uiModules,
                        BackbaseCoreModule.withConfig({
                            classMap: { ConsentListWidgetComponent },
                        }),
                    ],
                    exports: [...components, ...directives, ...uiModules],
                    providers: [ConsentListService],
                    declarations: [...components, ...directives],
                }]
        }] });
