// @backbase/consent-list-widget-ang/esm2015/src/consent-list-error.js
export class ConsentListAccessDenied {
    constructor(error) {
        this.error = error;
        this.kind = 'ConsentListAccessDenied';
    }
}
export class ConsentListBadRequest {
    constructor(error) {
        this.error = error;
        this.kind = 'ConsentListBadRequest';
    }
}
export class ConsentListUnknownError {
    constructor(error) {
        this.error = error;
        this.kind = 'ConsentListUnknownError';
    }
}
export class ConsentListConnectivityError {
    constructor(error) {
        this.error = error;
        this.kind = 'ConsentListConnectivityError';
    }
}
export const parseError = (error) => {
    if (error.error instanceof ErrorEvent) {
        return new ConsentListUnknownError(error);
    }
    else {
        switch (error.status) {
            case 0:
                return new ConsentListConnectivityError(error);
            case 400:
                return new ConsentListBadRequest(error);
            case 401:
                return new ConsentListAccessDenied(error);
            case 403:
                return new ConsentListAccessDenied(error);
            default:
                return new ConsentListUnknownError(error);
        }
    }
};
