// @backbase/consent-list-widget-ang/esm2015/src/consent-list-widget.service.js
import { Injectable } from '@angular/core';
import { of, BehaviorSubject } from 'rxjs';
import { catchError, tap, pluck } from 'rxjs/operators';
import { parseError, } from './consent-list-error';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/data-ang/consent";
export class ConsentListService {
    constructor(consentHttpService) {
        this.consentHttpService = consentHttpService;
        this.loading = new BehaviorSubject(true);
        this.error = new BehaviorSubject(undefined);
        this.consents = this.getConsentsList().pipe(tap(() => this.error.next(undefined)), catchError((error) => {
            this.error.next(error);
            return of([]);
        }), tap(() => this.loading.next(false)));
    }
    getConsentsList() {
        return this.consentHttpService.getConsents('response').pipe(pluck('body'), catchError((error) => {
            throw parseError(error);
        }));
    }
}
ConsentListService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentListService, deps: [{ token: i1.ConsentsHttpService }], target: i0.ɵɵFactoryTarget.Injectable });
ConsentListService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentListService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentListService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.ConsentsHttpService }]; } });
