// @backbase/loans-collection-ui-ang/esm2015/lib/payment-schedule/loan-payment-schedule.component.js
import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
import { LoanPaymentScheduleViewType } from './loan-payment-schedule.model';
import * as i0 from "@angular/core";
import * as i1 from "./amortized-payment/loan-amortized-payment-schedule-table/loan-amortized-payment-schedule-table.component";
import * as i2 from "./amortized-payment/loan-amortized-payment-schedule-list/loan-amortized-payment-schedule-list.component";
import * as i3 from "@angular/common";
export class LoanPaymentScheduleComponent {
    constructor() {
        /**
         * Schedule payment view type
         */
        this.viewType = LoanPaymentScheduleViewType.Table;
        /**
         * Emits on item selection
         */
        this.selectItem = new EventEmitter();
        /**
         * Available view types
         */
        this.LoanPaymentScheduleViewType = LoanPaymentScheduleViewType;
    }
}
/** @nocollapse */ LoanPaymentScheduleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanPaymentScheduleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ LoanPaymentScheduleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: LoanPaymentScheduleComponent, selector: "bb-loan-payment-schedule", inputs: { items: "items", viewType: "viewType" }, outputs: { selectItem: "selectItem" }, ngImport: i0, template: "<ng-container [ngSwitch]=\"viewType\">\n  <bb-loan-amortized-payment-schedule-table\n    *ngSwitchCase=\"LoanPaymentScheduleViewType.Table\"\n    [items]=\"items\"\n    (selectItem)=\"selectItem.emit($event)\"\n  >\n  </bb-loan-amortized-payment-schedule-table>\n  <bb-loan-amortized-payment-schedule-list\n    *ngSwitchCase=\"LoanPaymentScheduleViewType.List\"\n    [items]=\"items\"\n    (selectItem)=\"selectItem.emit($event)\"\n  ></bb-loan-amortized-payment-schedule-list>\n</ng-container>\n", components: [{ type: i1.LoanAmortizedPaymentScheduleTableComponent, selector: "bb-loan-amortized-payment-schedule-table", inputs: ["showPaymentStatus"] }, { type: i2.LoanAmortizedPaymentScheduleListComponent, selector: "bb-loan-amortized-payment-schedule-list" }], directives: [{ type: i3.NgSwitch, selector: "[ngSwitch]", inputs: ["ngSwitch"] }, { type: i3.NgSwitchCase, selector: "[ngSwitchCase]", inputs: ["ngSwitchCase"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanPaymentScheduleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-loan-payment-schedule',
                    templateUrl: './loan-payment-schedule.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { items: [{
                type: Input
            }], viewType: [{
                type: Input
            }], selectItem: [{
                type: Output
            }] } });
