// @backbase/loans-collection-ui-ang/esm2015/lib/payment-schedule/loan-payment-schedule.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { LoanPaymentScheduleComponent } from './loan-payment-schedule.component';
import { LoanPaymentScheduleStatusModule } from './loan-payment-schedule-status/loan-payment-schedule-status.module';
import { LoanAmortizedPaymentScheduleTableModule } from './amortized-payment/loan-amortized-payment-schedule-table/loan-amortized-payment-schedule-table.module';
import { LoanAmortizedPaymentScheduleListModule } from './amortized-payment/loan-amortized-payment-schedule-list/loan-amortized-payment-schedule-list.module';
import * as i0 from "@angular/core";
export class LoanPaymentScheduleModule {
}
/** @nocollapse */ LoanPaymentScheduleModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanPaymentScheduleModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanPaymentScheduleModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanPaymentScheduleModule, declarations: [LoanPaymentScheduleComponent], imports: [CommonModule,
        LoanAmortizedPaymentScheduleTableModule,
        LoanPaymentScheduleStatusModule,
        LoanAmortizedPaymentScheduleListModule], exports: [LoanPaymentScheduleComponent,
        LoanAmortizedPaymentScheduleTableModule,
        LoanAmortizedPaymentScheduleListModule] });
/** @nocollapse */ LoanPaymentScheduleModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanPaymentScheduleModule, imports: [[
            CommonModule,
            LoanAmortizedPaymentScheduleTableModule,
            LoanPaymentScheduleStatusModule,
            LoanAmortizedPaymentScheduleListModule,
        ], LoanAmortizedPaymentScheduleTableModule,
        LoanAmortizedPaymentScheduleListModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanPaymentScheduleModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanPaymentScheduleComponent],
                    imports: [
                        CommonModule,
                        LoanAmortizedPaymentScheduleTableModule,
                        LoanPaymentScheduleStatusModule,
                        LoanAmortizedPaymentScheduleListModule,
                    ],
                    exports: [
                        LoanPaymentScheduleComponent,
                        LoanAmortizedPaymentScheduleTableModule,
                        LoanAmortizedPaymentScheduleListModule,
                    ],
                }]
        }] });
