// @backbase/ui-ang/esm2015/_example/input-inline-edit/showcase/input-inline-edit-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { InputInlineEditShowcaseComponent } from './input-inline-edit-showcase.component';
import { InputInlineEditModule } from '@backbase/ui-ang/input-inline-edit';
import * as i0 from "@angular/core";
export class InputInlineEditShowcaseModule {
}
InputInlineEditShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputInlineEditShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputInlineEditShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputInlineEditShowcaseModule, declarations: [InputInlineEditShowcaseComponent], imports: [CommonModule, InputInlineEditModule] });
InputInlineEditShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputInlineEditShowcaseModule, imports: [[CommonModule, InputInlineEditModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputInlineEditShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, InputInlineEditModule],
                    declarations: [InputInlineEditShowcaseComponent],
                    entryComponents: [InputInlineEditShowcaseComponent],
                }]
        }] });
