// @backbase/ui-ang/esm2015/_example/input-inline-edit/showcase/input-inline-edit-showcase.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import { InputInlineEditState } from '@backbase/ui-ang/input-inline-edit';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/input-inline-edit";
let InputInlineEditShowcaseComponent = class InputInlineEditShowcaseComponent {
    constructor() {
        this.editInlineModel = 'Here some text that can be modified';
        this.editInlineModelWithLoading = 'Default text';
        this.inlineEditState = InputInlineEditState.IDLE;
    }
    onCancel() { }
    onAccept(item) {
        this.editInlineModel = item;
    }
    onAcceptWithLoading(value) {
        this.inlineEditState = InputInlineEditState.LOADING;
        setTimeout(() => {
            this.inlineEditState = InputInlineEditState.IDLE;
            this.editInlineModelWithLoading = value;
        }, 5000);
    }
};
InputInlineEditShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputInlineEditShowcaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
InputInlineEditShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: InputInlineEditShowcaseComponent, selector: "bb-input-inline-edit-ui-showcase", ngImport: i0, template: "<div class=\"bb-block bb-block--md\">\n  <h3>Inline Edit</h3>\n  <bb-input-inline-edit-ui\n    [inputText]=\"editInlineModel\"\n    (cancel)=\"onCancel()\"\n    (accept)=\"onAccept($event)\"\n    [maxLength]=\"64\"\n    [minLength]=\"2\"\n    [required]=\"true\"\n    hint=\"This is visible to the whole company\"\n  >\n  </bb-input-inline-edit-ui>\n</div>\n<div class=\"bb-block bb-block--md\">\n  <h3>Inline edit with loading state</h3>\n  <bb-input-inline-edit-ui\n    [hasLoadingState]=\"true\"\n    [inputText]=\"editInlineModelWithLoading\"\n    (accept)=\"onAcceptWithLoading($event)\"\n    [maxLength]=\"64\"\n    [(state)]=\"inlineEditState\"\n    hint=\"This is visible to the whole company\"\n  >\n  </bb-input-inline-edit-ui>\n</div>\n", components: [{ type: i1.InputInlineEditComponent, selector: "bb-input-inline-edit-ui", inputs: ["state", "canEdit", "hasLoadingState", "inputInlineTemplate", "inputText", "maxLength", "minLength", "maxValue", "minValue", "autofocus", "ariaLabelAccept", "ariaLabelCancel", "ariaLabelEdit", "tooltipAccept", "tooltipCancel", "tooltipEdit", "autocomplete", "pattern", "hint"], outputs: ["stateChange", "cancel", "accept"] }] });
InputInlineEditShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
], InputInlineEditShowcaseComponent);
export { InputInlineEditShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputInlineEditShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-input-inline-edit-ui-showcase',
                    templateUrl: './input-inline-edit-showcase.component.html',
                }]
        }] });
