// @backbase/foundation-ang/esm2015/entitlements/directives/if-entitlements.directive.js
import { ChangeDetectorRef, Directive, Input, TemplateRef, ViewContainerRef } from '@angular/core';
import { ConditionsService } from '../services/conditions.service';
/**
 * @description
 * A structural directive that conditionally includes a template based on the value of
 * an expression coerced to Boolean based on a entitlements triplets validation.
 * When the expression evaluates to true, Angular renders the template
 * Angular renders the template provided in an optional `else` clause. The default
 * template for the `else` clause is blank.
 *
 * @usageNotes
 *
 * ### Example
 * ```
 * <div *bbIfEntitlements="'ProductSummary.ProductSummary.edit'">Content to render when condition is true.</div>
 *
 * <div *bbIfEntitlements="'ProductSummary.ProductSummary.edit'; else alternative">Content to render when condition is true.</div>
 * <ng-template #alternative>
 *   <div>alternative content</div>
 * </ng-template>
 * ```
 */
export class IfEntitlementsDirective {
    constructor(conditions, templateRef, viewContainerRef, cd) {
        this.conditions = conditions;
        this.templateRef = templateRef;
        this.viewContainerRef = viewContainerRef;
        this.cd = cd;
    }
    /**
     * Entitlement to evaluate
     */
    set bbIfEntitlements(value) {
        const entitlements = value.trim();
        this.conditions.resolveEntitlements(entitlements).then(canAccess => {
            if (canAccess) {
                this.renderMainView();
            }
            else {
                this.renderAlternativeView();
            }
            this.cd.markForCheck();
        }, () => {
            console.error(`The element cannot be displayed because the identifier is invalid: "${value}"`);
            this.viewContainerRef.clear();
            this.cd.markForCheck();
        });
    }
    /**
     * Alternative template to show in the else clause
     */
    set bbIfEntitlementsElse(value) {
        this.elseTemplateRef = value;
    }
    renderMainView() {
        this.viewContainerRef.createEmbeddedView(this.templateRef);
    }
    renderAlternativeView() {
        this.viewContainerRef.clear();
        if (this.elseTemplateRef) {
            this.viewContainerRef.createEmbeddedView(this.elseTemplateRef);
        }
    }
}
IfEntitlementsDirective.decorators = [
    { type: Directive, args: [{
                selector: '[bbIfEntitlements]',
            },] }
];
IfEntitlementsDirective.ctorParameters = () => [
    { type: ConditionsService },
    { type: TemplateRef },
    { type: ViewContainerRef },
    { type: ChangeDetectorRef }
];
IfEntitlementsDirective.propDecorators = {
    bbIfEntitlements: [{ type: Input }],
    bbIfEntitlementsElse: [{ type: Input }]
};
