// @backbase/ui-ang/esm2015/_example/notification/showcase/notification-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NotificationShowcaseComponent } from './notification-showcase.component';
import { NotificationModule } from '@backbase/ui-ang/notification';
import { ButtonModule } from '@backbase/ui-ang/button';
import * as i0 from "@angular/core";
export class NotificationShowcaseModule {
}
NotificationShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NotificationShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationShowcaseModule, declarations: [NotificationShowcaseComponent], imports: [CommonModule, NotificationModule, ButtonModule] });
NotificationShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationShowcaseModule, imports: [[CommonModule, NotificationModule, ButtonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, NotificationModule, ButtonModule],
                    declarations: [NotificationShowcaseComponent],
                    entryComponents: [NotificationShowcaseComponent],
                }]
        }] });
