// @backbase/product-summary-ui/esm2015/src/bb-credit-limit-ui/credit-limit-attribute/credit-limit-attribute.component.js
import { Component, Input, ViewChild } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/amount";
import * as i2 from "@angular/common";
export class CreditLimitAttributeComponent {
    constructor() {
        /**
         * Custom value template reference
         */
        this.contentRef = '';
        /**
         * Attribute title
         */
        this.title = '';
        /**
         * Defines whether to color the amount field
         *
         * @default false
         */
        this.isAmountColored = false;
        /**
         * Whether currency should be displayed as a sign.
         *
         * @default true.
         */
        this.useShortCurrency = true;
        /**
         * Specifies whether to use small screen layout
         *
         * @default false.
         */
        this.isSmallScreen = false;
        /**
         * Specifies whether to use extra small screen layout
         *
         * @default false.
         */
        this.isExtraSmallScreen = false;
        /**
         * Specifies whether container size is small
         */
        this.responsiveView = 'screen';
    }
}
CreditLimitAttributeComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CreditLimitAttributeComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
CreditLimitAttributeComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: CreditLimitAttributeComponent, selector: "bb-credit-limit-attribute", inputs: { title: "title", color: "color", currency: "currency", amount: "amount", isAmountColored: "isAmountColored", useShortCurrency: "useShortCurrency", isSmallScreen: "isSmallScreen", isExtraSmallScreen: "isExtraSmallScreen", responsiveView: "responsiveView" }, viewQueries: [{ propertyName: "contentRef", first: true, predicate: ["contentRef"], descendants: true }], ngImport: i0, template: "<div [ngClass]=\"{\n  'bb-stack justify-content-between': responsiveView === 'tablet' || responsiveView === 'mobile',\n  'bb-stack--align-top': responsiveView === 'mobile'\n}\">\n  <!-- title -->\n  <div class=\"bb-text-support bb-subtitle\"\n       [ngClass]=\"{\n         'bb-block bb-block--xs': responsiveView === 'screen',\n         'bb-stack__item': responsiveView === 'tablet' || responsiveView === 'mobile'\n       }\"\n  >\n    <div *ngIf=\"color\"\n         class=\"\n         rounded-circle\n         bb-credit-limit-attribute__color-mark\n         bb-credit-limit__color--{{color}}\n       \"></div>\n    <span data-role=\"credit-limit-attribute-title\">{{title}}</span>\n  </div>\n\n  <div [ngClass]=\"{\n         'bb-stack__item': responsiveView === 'tablet' || responsiveView === 'mobile'\n  }\">\n    <!-- default value -->\n    <bb-amount-ui\n            *ngIf=\"!contentRef && amount\"\n            data-role=\"credit-limit-attribute-value\"\n            class=\"bb-stack__item bb-text-semi-bold {{ isAmountColored ? 'bb-credit-limit-attribute__value--' + color : '' }}\"\n            [class.bb-credit-limit-attribute__value]=\"color\"\n            [ngClass]=\"{\n              'bb-subtitle': responsiveView === 'mobile'\n            }\"\n            [currency]=\"currency\"\n            [amount]=\"amount\"\n            [mapCurrency]=\"useShortCurrency\"\n    ></bb-amount-ui>\n\n    <!-- custom value -->\n    <ng-content #contentRef></ng-content>\n  </div>\n</div>\n", components: [{ type: i1.AmountComponent, selector: "bb-amount-ui", inputs: ["currency", "showPlusSign", "mapCurrency", "showPercent", "abbreviate", "decimalPlaces", "trailingZeroes", "amount"] }], directives: [{ type: i2.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CreditLimitAttributeComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-credit-limit-attribute',
                    templateUrl: './credit-limit-attribute.component.html',
                }]
        }], propDecorators: { contentRef: [{
                type: ViewChild,
                args: ['contentRef']
            }], title: [{
                type: Input
            }], color: [{
                type: Input
            }], currency: [{
                type: Input
            }], amount: [{
                type: Input
            }], isAmountColored: [{
                type: Input
            }], useShortCurrency: [{
                type: Input
            }], isSmallScreen: [{
                type: Input
            }], isExtraSmallScreen: [{
                type: Input
            }], responsiveView: [{
                type: Input
            }] } });
