// @backbase/notifications-badge-widget-ang/esm2015/lib/components/notification-details/notification-details.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NotificationDetailsComponent, NotificationDetailsCustomizableDirective, } from './notification-details.component';
import { NotificationDetailsItemComponent, NotificationDetailsItemCustomizableDirective, } from './notification-details-item/notification-details-item.component';
import { HumanizedDateModule } from '../humanized-date/humanized-date.module';
import { NotificationsCommonModule } from '@backbase/notifications-common-ang';
import { ButtonModule } from '@backbase/ui-ang/button';
import { FocusModule } from '@backbase/ui-ang/focus';
import { EllipsisModule } from '@backbase/ui-ang/ellipsis';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import * as i0 from "@angular/core";
const uiModules = [ButtonModule, FocusModule, EllipsisModule, TooltipModule];
const innerModules = [HumanizedDateModule];
const components = [
    NotificationDetailsComponent,
    NotificationDetailsCustomizableDirective,
    NotificationDetailsItemComponent,
    NotificationDetailsItemCustomizableDirective,
];
export class NotificationDetailsModule {
}
/** @nocollapse */ NotificationDetailsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationDetailsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ NotificationDetailsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationDetailsModule, declarations: [NotificationDetailsComponent,
        NotificationDetailsCustomizableDirective,
        NotificationDetailsItemComponent,
        NotificationDetailsItemCustomizableDirective], imports: [CommonModule, NotificationsCommonModule, ButtonModule, FocusModule, EllipsisModule, TooltipModule, HumanizedDateModule], exports: [NotificationsCommonModule, ButtonModule, FocusModule, EllipsisModule, TooltipModule, HumanizedDateModule, NotificationDetailsComponent,
        NotificationDetailsCustomizableDirective,
        NotificationDetailsItemComponent,
        NotificationDetailsItemCustomizableDirective] });
/** @nocollapse */ NotificationDetailsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationDetailsModule, imports: [[CommonModule, NotificationsCommonModule, ...uiModules, ...innerModules], NotificationsCommonModule, ButtonModule, FocusModule, EllipsisModule, TooltipModule, HumanizedDateModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationDetailsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, NotificationsCommonModule, ...uiModules, ...innerModules],
                    declarations: [...components],
                    exports: [NotificationsCommonModule, ...uiModules, ...innerModules, ...components],
                }]
        }] });
