// @backbase/product-summary-ui/esm2015/src/bb-credit-limit-ui/credit-limit-minimum-payment/credit-limit-minimum-payment.component.js
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../credit-limit-attribute/credit-limit-attribute.component";
import * as i2 from "@backbase/ui-ang/amount";
import * as i3 from "@angular/common";
import * as i4 from "@backbase/ui-ang/date-pipe";
/**
 * Displays a credit limit minimum payment info
 *
 * @usageNotes
 *
 * ```html
 * <bb-credit-limit-minimum-payment
 *   [minimumPayment]="100"
 *   [minimumPaymentDueDate]="'2021-03-15T16:41:41.090Z'"
 *   [currency]="EUR"
 * ></bb-credit-limit-minimum-payment>
 * ```
 *
 * @ngModule CreditLimitModule
 */
export class CreditLimitMinimumPaymentComponent {
    constructor() {
        /**
         * Whether currency should be displayed as a sign.
         *
         * @default true.
         */
        this.useShortCurrency = true;
        /**
         * Specifies whether to use small screen layout
         *
         * @default false.
         */
        this.isSmallScreen = false;
        /**
         * Specifies whether to use extra small screen layout
         *
         * @default false.
         */
        this.isExtraSmallScreen = false;
        /**
         * Specifies whether container size is small
         */
        this.responsiveView = 'screen';
    }
}
CreditLimitMinimumPaymentComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CreditLimitMinimumPaymentComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
CreditLimitMinimumPaymentComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: CreditLimitMinimumPaymentComponent, selector: "bb-credit-limit-minimum-payment", inputs: { minimumPayment: "minimumPayment", minimumPaymentDueDate: "minimumPaymentDueDate", currency: "currency", useShortCurrency: "useShortCurrency", isSmallScreen: "isSmallScreen", isExtraSmallScreen: "isExtraSmallScreen", responsiveView: "responsiveView" }, ngImport: i0, template: "<bb-credit-limit-attribute\n  class=\"bb-text-align-right--md-up\"\n  data-role=\"credit-limit-minimum-payment\"\n  title=\"Minimum Payment\"\n  i18n-title=\"Minimum Payment|Credit limit attribute: Minimum Payment@@bb-credit-limit-attribute-minimum-payment\"\n  [responsiveView]=\"responsiveView\"\n>\n  <div class=\"bb-stack bb-stack--wrap bb-stack--density-xs bb-text-semi-bold\"\n       [ngClass]=\"{\n         'bb-subtitle, justify-content-end': responsiveView === 'mobile' || responsiveView === 'tablet'\n       }\"\n       data-role=\"credit-limit-minimum-payment-value\"\n  >\n    <bb-amount-ui\n      class=\"bb-stack__item\"\n      [currency]=\"currency\"\n      [amount]=\"minimumPayment\"\n      [mapCurrency]=\"useShortCurrency\">\n    </bb-amount-ui>\n\n    <ng-container *ngIf=\"minimumPaymentDueDate\">\n      <div class=\"bb-stack__break bb-stack__break--xs-down\"></div>\n      <span class=\"bb-stack__item\"\n            i18n=\"due|Minimum payment: due to date@@bb-credit-limit-attribute-minimum-payment-due\">\n        due\n      </span>\n      <span class=\"bb-stack__item\">\n        {{minimumPaymentDueDate | bbDate}}\n      </span>\n    </ng-container>\n  </div>\n</bb-credit-limit-attribute>\n", components: [{ type: i1.CreditLimitAttributeComponent, selector: "bb-credit-limit-attribute", inputs: ["title", "color", "currency", "amount", "isAmountColored", "useShortCurrency", "isSmallScreen", "isExtraSmallScreen", "responsiveView"] }, { type: i2.AmountComponent, selector: "bb-amount-ui", inputs: ["currency", "showPlusSign", "mapCurrency", "showPercent", "abbreviate", "decimalPlaces", "trailingZeroes", "amount"] }], directives: [{ type: i3.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "bbDate": i4.BbDatePipe }, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CreditLimitMinimumPaymentComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-credit-limit-minimum-payment',
                    templateUrl: './credit-limit-minimum-payment.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { minimumPayment: [{
                type: Input
            }], minimumPaymentDueDate: [{
                type: Input
            }], currency: [{
                type: Input
            }], useShortCurrency: [{
                type: Input
            }], isSmallScreen: [{
                type: Input
            }], isExtraSmallScreen: [{
                type: Input
            }], responsiveView: [{
                type: Input
            }] } });
