// @backbase/loans-collection-ui-ang/esm2015/lib/loan-summary/loan-summary-overdue-banner/loan-summary-overdue-banner.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AmountModule } from '@backbase/ui-ang/amount';
import { BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import { ButtonModule } from '@backbase/ui-ang/button';
import { IconModule } from '@backbase/ui-ang/icon';
import { LoanSummaryOverdueBannerComponent } from './loan-summary-overdue-banner.component';
import * as i0 from "@angular/core";
export class LoanSummaryOverdueBannerModule {
}
/** @nocollapse */ LoanSummaryOverdueBannerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryOverdueBannerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanSummaryOverdueBannerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryOverdueBannerModule, declarations: [LoanSummaryOverdueBannerComponent], imports: [CommonModule, ButtonModule, IconModule, AmountModule, BbDatePipeModule], exports: [LoanSummaryOverdueBannerComponent] });
/** @nocollapse */ LoanSummaryOverdueBannerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryOverdueBannerModule, imports: [[CommonModule, ButtonModule, IconModule, AmountModule, BbDatePipeModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanSummaryOverdueBannerModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanSummaryOverdueBannerComponent],
                    imports: [CommonModule, ButtonModule, IconModule, AmountModule, BbDatePipeModule],
                    exports: [LoanSummaryOverdueBannerComponent],
                }]
        }] });
