// @backbase/ui-ang/esm2015/_example/input-phone/examples/input-phone.example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/forms";
import * as i2 from "@backbase/ui-ang/input-phone";
export class InputPhoneExampleComponent {
    constructor(formBuilder) {
        this.formBuilder = formBuilder;
        this.vForm = this.formBuilder.group({
            workPhone: '4151234567',
        });
        this.country = [
            {
                countryCode: '+91',
                isoCode: 'IN',
                countryName: 'India',
            },
        ];
        this.countries = [
            {
                countryCode: '+1',
                isoCode: 'US',
                countryName: 'United States',
            },
            {
                countryCode: '+31',
                isoCode: 'NL',
                countryName: 'Netherlands',
            },
            {
                countryCode: '+34',
                isoCode: 'NZ',
                countryName: 'New Zealand',
            },
            {
                countryCode: '+355',
                isoCode: 'AL',
                countryName: 'Albania',
            },
            {
                countryCode: '+91',
                isoCode: 'IN',
                countryName: 'India',
            },
            {
                countryCode: '+55',
                isoCode: 'BR',
                countryName: 'Brazil',
            },
            {
                countryCode: '+420',
                isoCode: 'CZ',
                countryName: 'Czech Republic',
            },
            {
                countryCode: '+49',
                isoCode: 'DE',
                countryName: 'Germany',
            },
        ];
    }
}
InputPhoneExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputPhoneExampleComponent, deps: [{ token: i1.FormBuilder }], target: i0.ɵɵFactoryTarget.Component });
InputPhoneExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: InputPhoneExampleComponent, selector: "bb-input-phone-example-ui", ngImport: i0, template: "<form [formGroup]=\"vForm\">\n  <bb-input-phone-ui\n    label=\"Work Phone\"\n    formControlName=\"workPhone\"\n    placeholder=\"Phone number without country code\"\n    autoComplete=\"tel-national\"\n  ></bb-input-phone-ui>\n\n  <bb-input-phone-ui\n    [id]=\"'phoneNumber1'\"\n    label=\"Phone Number\"\n    formControlName=\"workPhone\"\n    placeholder=\"Enter a phone number\"\n    [required]=\"true\"\n    [enableCountryCode]=\"true\"\n    defaultCountryIsoCode=\"BR\"\n    [countryList]=\"countries\"\n  >\n  </bb-input-phone-ui>\n\n  <bb-input-phone-ui\n    [id]=\"'phoneNumber2'\"\n    label=\"Phone Number\"\n    formControlName=\"workPhone\"\n    placeholder=\"Enter a phone number\"\n    [required]=\"true\"\n    [enableCountryCode]=\"true\"\n    [countryList]=\"country\"\n  >\n  </bb-input-phone-ui>\n</form>\n", components: [{ type: i2.InputPhoneComponent, selector: "bb-input-phone-ui", inputs: ["placeholder", "maxLength", "minLength", "readonly", "ariaLabel", "autocomplete", "countryList", "dropdownPosition", "clearable", "searchable", "defaultCountryIsoCode", "enableCountryCode"] }], directives: [{ type: i1.ɵNgNoValidate, selector: "form:not([ngNoForm]):not([ngNativeValidate])" }, { type: i1.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { type: i1.FormGroupDirective, selector: "[formGroup]", inputs: ["formGroup"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { type: i1.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { type: i1.FormControlName, selector: "[formControlName]", inputs: ["disabled", "formControlName", "ngModel"], outputs: ["ngModelChange"] }, { type: i1.RequiredValidator, selector: ":not([type=checkbox])[required][formControlName],:not([type=checkbox])[required][formControl],:not([type=checkbox])[required][ngModel]", inputs: ["required"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputPhoneExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-input-phone-example-ui',
                    templateUrl: './input-phone.example.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.FormBuilder }]; } });
