// @backbase/ui-ang/esm2015/_example/product-item-basic-account/examples/product-item-basic-account.example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/product-item-basic-account";
export class ProductItemBasicAccountExampleComponent {
    constructor() {
        this.product = {
            id: 'b387bfa5-75fd-410e-bb86-7f764f8ef69c',
            name: 'Ricardo Norton\'s Savings Account',
            currency: 'EUR',
            amount: '83652.32',
            productNumber: 'NL75INGB3400120056',
        };
        this.productNumberFormat = {
            length: 18,
            maskRange: [0, 0],
            segments: 4,
        };
    }
}
ProductItemBasicAccountExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemBasicAccountExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ProductItemBasicAccountExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ProductItemBasicAccountExampleComponent, selector: "bb-product-item-basic-account-example-ui", ngImport: i0, template: "<bb-product-item-basic-account-ui\n  [title]=\"product.name\"\n  [amount]=\"product.amount\"\n  [productNumber]=\"product.productNumber\"\n  [currency]=\"product.currency\"\n  [productNumberFormat]=\"productNumberFormat\"\n  [highlight]=\"false\"\n  [favorite]=\"false\"\n  [active]=\"false\"\n  balanceLabel=\"Available balance\"\n></bb-product-item-basic-account-ui>\n", components: [{ type: i1.ProductItemBasicAccountComponent, selector: "bb-product-item-basic-account-ui", inputs: ["title", "productNumber", "productNumberFormat", "currency", "showCurrencySymbol", "amount", "highlight", "productDetailsInfo", "active", "favorite", "favoriteIconColor", "selected", "type", "status", "balanceLabel", "bankBranchCode"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemBasicAccountExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-product-item-basic-account-example-ui',
                    templateUrl: './product-item-basic-account.example.component.html',
                }]
        }] });
