// @backbase/engagement-http-ang/esm2015/api/campaignMetrics.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { ENGAGEMENT_BASE_PATH } from '../variables';
import { EngagementConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class CampaignMetricsService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '/engagements';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new EngagementConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    getCampaignMetrics(requestParameters, observe = 'body', reportProgress = false, options) {
        const _campaignId = requestParameters["campaignId"];
        if (_campaignId === null || _campaignId === undefined) {
            throw new Error('Required parameter campaignId was null or undefined when calling getCampaignMetrics.');
        }
        const _channelId = requestParameters["channelId"];
        if (_channelId === null || _channelId === undefined) {
            throw new Error('Required parameter channelId was null or undefined when calling getCampaignMetrics.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/engagements/custom/${encodeURIComponent(String(_campaignId))}/channels/${encodeURIComponent(String(_channelId))}/metrics`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getCampaignMetricsUrl(requestParameters) {
        const _campaignId = requestParameters["campaignId"];
        if (_campaignId === null || _campaignId === undefined) {
            throw new Error('Required parameter campaignId was null or undefined when calling getCampaignMetrics.');
        }
        const _channelId = requestParameters["channelId"];
        if (_channelId === null || _channelId === undefined) {
            throw new Error('Required parameter channelId was null or undefined when calling getCampaignMetrics.');
        }
        return `${this.configuration.basePath}/client-api/v1/engagements/custom/${encodeURIComponent(String(_campaignId))}/channels/${encodeURIComponent(String(_channelId))}/metrics`;
    }
    redirectCampaign(requestParameters, observe = 'body', reportProgress = false, options) {
        const _utmContent = requestParameters["utmContent"];
        if (_utmContent === null || _utmContent === undefined) {
            throw new Error('Required parameter utmContent was null or undefined when calling redirectCampaign.');
        }
        const _utmBanner = requestParameters["utmBanner"];
        if (_utmBanner === null || _utmBanner === undefined) {
            throw new Error('Required parameter utmBanner was null or undefined when calling redirectCampaign.');
        }
        const _key = requestParameters["key"];
        if (_key === null || _key === undefined) {
            throw new Error('Required parameter key was null or undefined when calling redirectCampaign.');
        }
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_utmContent !== undefined && _utmContent !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _utmContent, 'utm_content');
        }
        if (_utmBanner !== undefined && _utmBanner !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _utmBanner, 'utm_banner');
        }
        if (_key !== undefined && _key !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _key, 'key');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/banner/redirect`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    redirectCampaignUrl(requestParameters) {
        const _utmContent = requestParameters["utmContent"];
        if (_utmContent === null || _utmContent === undefined) {
            throw new Error('Required parameter utmContent was null or undefined when calling redirectCampaign.');
        }
        const _utmBanner = requestParameters["utmBanner"];
        if (_utmBanner === null || _utmBanner === undefined) {
            throw new Error('Required parameter utmBanner was null or undefined when calling redirectCampaign.');
        }
        const _key = requestParameters["key"];
        if (_key === null || _key === undefined) {
            throw new Error('Required parameter key was null or undefined when calling redirectCampaign.');
        }
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_utmContent !== undefined && _utmContent !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _utmContent, 'utm_content');
        }
        if (_utmBanner !== undefined && _utmBanner !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _utmBanner, 'utm_banner');
        }
        if (_key !== undefined && _key !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _key, 'key');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v1/banner/redirect${queryString ? `?${queryString}` : ''}`;
    }
}
CampaignMetricsService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CampaignMetricsService, deps: [{ token: i1.HttpClient }, { token: ENGAGEMENT_BASE_PATH, optional: true }, { token: i2.EngagementConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
CampaignMetricsService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CampaignMetricsService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CampaignMetricsService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [ENGAGEMENT_BASE_PATH]
                }] }, { type: i2.EngagementConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
