// @backbase/engagement-http-ang/esm2015/api/campaign.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { ENGAGEMENT_BASE_PATH, COLLECTION_FORMATS } from '../variables';
import { EngagementConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class CampaignService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '/engagements';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new EngagementConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    createCampaign(requestParameters, observe = 'body', reportProgress = false, options) {
        const _campaignRequestResource = requestParameters["campaignRequestResource"];
        if (_campaignRequestResource === null || _campaignRequestResource === undefined) {
            throw new Error('Required parameter campaignRequestResource was null or undefined when calling createCampaign.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v1/engagements/custom`, _campaignRequestResource, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    createCampaignUrl(requestParameters) {
        const _campaignRequestResource = requestParameters["campaignRequestResource"];
        if (_campaignRequestResource === null || _campaignRequestResource === undefined) {
            throw new Error('Required parameter campaignRequestResource was null or undefined when calling createCampaign.');
        }
        return `${this.configuration.basePath}/client-api/v1/engagements/custom`;
    }
    deleteCampaign(requestParameters, observe = 'body', reportProgress = false, options) {
        const _campaignId = requestParameters["campaignId"];
        if (_campaignId === null || _campaignId === undefined) {
            throw new Error('Required parameter campaignId was null or undefined when calling deleteCampaign.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.delete(`${this.configuration.basePath}/client-api/v1/engagements/custom/${encodeURIComponent(String(_campaignId))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    deleteCampaignUrl(requestParameters) {
        const _campaignId = requestParameters["campaignId"];
        if (_campaignId === null || _campaignId === undefined) {
            throw new Error('Required parameter campaignId was null or undefined when calling deleteCampaign.');
        }
        return `${this.configuration.basePath}/client-api/v1/engagements/custom/${encodeURIComponent(String(_campaignId))}`;
    }
    disableCampaign(requestParameters, observe = 'body', reportProgress = false, options) {
        const _campaignId = requestParameters["campaignId"];
        if (_campaignId === null || _campaignId === undefined) {
            throw new Error('Required parameter campaignId was null or undefined when calling disableCampaign.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v1/engagements/custom/${encodeURIComponent(String(_campaignId))}/disable`, null, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    disableCampaignUrl(requestParameters) {
        const _campaignId = requestParameters["campaignId"];
        if (_campaignId === null || _campaignId === undefined) {
            throw new Error('Required parameter campaignId was null or undefined when calling disableCampaign.');
        }
        return `${this.configuration.basePath}/client-api/v1/engagements/custom/${encodeURIComponent(String(_campaignId))}/disable`;
    }
    enableCampaign(requestParameters, observe = 'body', reportProgress = false, options) {
        const _campaignId = requestParameters["campaignId"];
        if (_campaignId === null || _campaignId === undefined) {
            throw new Error('Required parameter campaignId was null or undefined when calling enableCampaign.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v1/engagements/custom/${encodeURIComponent(String(_campaignId))}/enable`, null, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    enableCampaignUrl(requestParameters) {
        const _campaignId = requestParameters["campaignId"];
        if (_campaignId === null || _campaignId === undefined) {
            throw new Error('Required parameter campaignId was null or undefined when calling enableCampaign.');
        }
        return `${this.configuration.basePath}/client-api/v1/engagements/custom/${encodeURIComponent(String(_campaignId))}/enable`;
    }
    finalizeCampaign(requestParameters, observe = 'body', reportProgress = false, options) {
        const _campaignId = requestParameters["campaignId"];
        if (_campaignId === null || _campaignId === undefined) {
            throw new Error('Required parameter campaignId was null or undefined when calling finalizeCampaign.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v1/engagements/custom/${encodeURIComponent(String(_campaignId))}/final`, null, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    finalizeCampaignUrl(requestParameters) {
        const _campaignId = requestParameters["campaignId"];
        if (_campaignId === null || _campaignId === undefined) {
            throw new Error('Required parameter campaignId was null or undefined when calling finalizeCampaign.');
        }
        return `${this.configuration.basePath}/client-api/v1/engagements/custom/${encodeURIComponent(String(_campaignId))}/final`;
    }
    getCampaignByCampaignId(requestParameters, observe = 'body', reportProgress = false, options) {
        const _campaignId = requestParameters["campaignId"];
        if (_campaignId === null || _campaignId === undefined) {
            throw new Error('Required parameter campaignId was null or undefined when calling getCampaignByCampaignId.');
        }
        const _status = requestParameters["status"];
        const _version = requestParameters["version"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_status) {
            queryParameters = this.addToHttpParams(queryParameters, _status.join(COLLECTION_FORMATS['csv']), 'status');
        }
        if (_version !== undefined && _version !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _version, 'version');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/engagements/custom/${encodeURIComponent(String(_campaignId))}`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getCampaignByCampaignIdUrl(requestParameters) {
        const _campaignId = requestParameters["campaignId"];
        if (_campaignId === null || _campaignId === undefined) {
            throw new Error('Required parameter campaignId was null or undefined when calling getCampaignByCampaignId.');
        }
        const _status = requestParameters["status"];
        const _version = requestParameters["version"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_status) {
            queryParameters = this.addToHttpParams(queryParameters, _status.join(COLLECTION_FORMATS['csv']), 'status');
        }
        if (_version !== undefined && _version !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _version, 'version');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v1/engagements/custom/${encodeURIComponent(String(_campaignId))}${queryString ? `?${queryString}` : ''}`;
    }
    getCampaigns(requestParameters, observe = 'body', reportProgress = false, options) {
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        const _endDateAfter = requestParameters["endDateAfter"];
        const _endDateBefore = requestParameters["endDateBefore"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_endDateAfter !== undefined && _endDateAfter !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _endDateAfter, 'endDateAfter');
        }
        if (_endDateBefore !== undefined && _endDateBefore !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _endDateBefore, 'endDateBefore');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/engagements/custom`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getCampaignsUrl(requestParameters) {
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        const _endDateAfter = requestParameters["endDateAfter"];
        const _endDateBefore = requestParameters["endDateBefore"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_endDateAfter !== undefined && _endDateAfter !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _endDateAfter, 'endDateAfter');
        }
        if (_endDateBefore !== undefined && _endDateBefore !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _endDateBefore, 'endDateBefore');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v1/engagements/custom${queryString ? `?${queryString}` : ''}`;
    }
    updateCampaign(requestParameters, observe = 'body', reportProgress = false, options) {
        const _campaignId = requestParameters["campaignId"];
        if (_campaignId === null || _campaignId === undefined) {
            throw new Error('Required parameter campaignId was null or undefined when calling updateCampaign.');
        }
        const _campaignRequestResource = requestParameters["campaignRequestResource"];
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v1/engagements/custom/${encodeURIComponent(String(_campaignId))}`, _campaignRequestResource, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    updateCampaignUrl(requestParameters) {
        const _campaignId = requestParameters["campaignId"];
        if (_campaignId === null || _campaignId === undefined) {
            throw new Error('Required parameter campaignId was null or undefined when calling updateCampaign.');
        }
        const _campaignRequestResource = requestParameters["campaignRequestResource"];
        return `${this.configuration.basePath}/client-api/v1/engagements/custom/${encodeURIComponent(String(_campaignId))}`;
    }
}
CampaignService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CampaignService, deps: [{ token: i1.HttpClient }, { token: ENGAGEMENT_BASE_PATH, optional: true }, { token: i2.EngagementConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
CampaignService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CampaignService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CampaignService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [ENGAGEMENT_BASE_PATH]
                }] }, { type: i2.EngagementConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
