// @backbase/engagement-http-ang/esm2015/api/notificationPreference.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const NotificationPreferenceServiceCreateNotificationPreferenceMocksProvider = createMocks([{
        urlPattern: "/engagements/client-api/v1/notification-preferences",
        method: "POST",
        responses: [
            {
                status: 201,
                body: {
                    "$ref": "#/components/examples/NotificationPreferenceResponseExample/value"
                }
            },
            {
                status: 201,
                body: {
                    "channels": [{
                            "userCanChange": true,
                            "channel": "email",
                            "enabled": true
                        }, {
                            "userCanChange": true,
                            "channel": "email",
                            "enabled": true
                        }],
                    "lastUpdatedAt": "2000-01-23T04:56:07.000+00:00",
                    "generalNotificationId": "generalNotificationId",
                    "active": true,
                    "id": "046b6c7f-0b8a-43b9-b35d-6489e6daee91",
                    "createdOn": "2000-01-23T04:56:07.000+00:00",
                    "userConditions": {
                        "eRef": "6be7c465-37a7-4f1e-98b7-15d6c636c566",
                        "eType": "arrangement",
                        "conditions": {
                            "amount": 100
                        }
                    }
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
        ]
    }]);
export const NotificationPreferenceServiceGetNotificationPreferencesMocksProvider = createMocks([{
        urlPattern: "/engagements/client-api/v1/notification-preferences",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "value": {
                        "totalCount": 2,
                        "notificationPreferences": [{
                                "$ref": "#/components/examples/NotificationPreferenceGetExample2/value"
                            }, {
                                "$ref": "#/components/examples/NotificationPreferenceGetExample3/value"
                            }]
                    }
                }
            },
            {
                status: 200,
                body: {
                    "value": {
                        "totalCount": 2,
                        "notificationPreferences": [{
                                "$ref": "#/components/examples/NotificationPreferenceGetExample1/value"
                            }, {
                                "$ref": "#/components/examples/NotificationPreferenceGetExample2/value"
                            }]
                    }
                }
            },
        ]
    }]);
export const NotificationPreferenceServiceUpdateNotificationPreferenceMocksProvider = createMocks([{
        urlPattern: "/engagements/client-api/v1/notification-preferences/{id}",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {
                    "$ref": "#/components/examples/NotificationPreferenceResponseExample/value"
                }
            },
            {
                status: 200,
                body: {
                    "channels": [{
                            "userCanChange": true,
                            "channel": "email",
                            "enabled": true
                        }, {
                            "userCanChange": true,
                            "channel": "email",
                            "enabled": true
                        }],
                    "lastUpdatedAt": "2000-01-23T04:56:07.000+00:00",
                    "generalNotificationId": "generalNotificationId",
                    "active": true,
                    "id": "046b6c7f-0b8a-43b9-b35d-6489e6daee91",
                    "createdOn": "2000-01-23T04:56:07.000+00:00",
                    "userConditions": {
                        "eRef": "6be7c465-37a7-4f1e-98b7-15d6c636c566",
                        "eType": "arrangement",
                        "conditions": {
                            "amount": 100
                        }
                    }
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
        ]
    }]);
export const NotificationPreferenceServiceMocksProvider = createMocks([
    {
        urlPattern: "/engagements/client-api/v1/notification-preferences",
        method: "POST",
        responses: [
            {
                status: 201,
                body: {
                    "$ref": "#/components/examples/NotificationPreferenceResponseExample/value"
                }
            },
            {
                status: 201,
                body: {
                    "channels": [{
                            "userCanChange": true,
                            "channel": "email",
                            "enabled": true
                        }, {
                            "userCanChange": true,
                            "channel": "email",
                            "enabled": true
                        }],
                    "lastUpdatedAt": "2000-01-23T04:56:07.000+00:00",
                    "generalNotificationId": "generalNotificationId",
                    "active": true,
                    "id": "046b6c7f-0b8a-43b9-b35d-6489e6daee91",
                    "createdOn": "2000-01-23T04:56:07.000+00:00",
                    "userConditions": {
                        "eRef": "6be7c465-37a7-4f1e-98b7-15d6c636c566",
                        "eType": "arrangement",
                        "conditions": {
                            "amount": 100
                        }
                    }
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/engagements/client-api/v1/notification-preferences",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "value": {
                        "totalCount": 2,
                        "notificationPreferences": [{
                                "$ref": "#/components/examples/NotificationPreferenceGetExample2/value"
                            }, {
                                "$ref": "#/components/examples/NotificationPreferenceGetExample3/value"
                            }]
                    }
                }
            },
            {
                status: 200,
                body: {
                    "value": {
                        "totalCount": 2,
                        "notificationPreferences": [{
                                "$ref": "#/components/examples/NotificationPreferenceGetExample1/value"
                            }, {
                                "$ref": "#/components/examples/NotificationPreferenceGetExample2/value"
                            }]
                    }
                }
            },
        ]
    },
    {
        urlPattern: "/engagements/client-api/v1/notification-preferences/{id}",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {
                    "$ref": "#/components/examples/NotificationPreferenceResponseExample/value"
                }
            },
            {
                status: 200,
                body: {
                    "channels": [{
                            "userCanChange": true,
                            "channel": "email",
                            "enabled": true
                        }, {
                            "userCanChange": true,
                            "channel": "email",
                            "enabled": true
                        }],
                    "lastUpdatedAt": "2000-01-23T04:56:07.000+00:00",
                    "generalNotificationId": "generalNotificationId",
                    "active": true,
                    "id": "046b6c7f-0b8a-43b9-b35d-6489e6daee91",
                    "createdOn": "2000-01-23T04:56:07.000+00:00",
                    "userConditions": {
                        "eRef": "6be7c465-37a7-4f1e-98b7-15d6c636c566",
                        "eType": "arrangement",
                        "conditions": {
                            "amount": 100
                        }
                    }
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
        ]
    },
]);
