// @backbase/engagement-http-ang/esm2015/api/notificationPreference.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { ENGAGEMENT_BASE_PATH } from '../variables';
import { EngagementConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class NotificationPreferenceService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '/engagements';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new EngagementConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    createNotificationPreference(requestParameters, observe = 'body', reportProgress = false, options) {
        const _notificationPreferencePostRequest = requestParameters["notificationPreferencePostRequest"];
        if (_notificationPreferencePostRequest === null || _notificationPreferencePostRequest === undefined) {
            throw new Error('Required parameter notificationPreferencePostRequest was null or undefined when calling createNotificationPreference.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v1/notification-preferences`, _notificationPreferencePostRequest, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    createNotificationPreferenceUrl(requestParameters) {
        const _notificationPreferencePostRequest = requestParameters["notificationPreferencePostRequest"];
        if (_notificationPreferencePostRequest === null || _notificationPreferencePostRequest === undefined) {
            throw new Error('Required parameter notificationPreferencePostRequest was null or undefined when calling createNotificationPreference.');
        }
        return `${this.configuration.basePath}/client-api/v1/notification-preferences`;
    }
    getNotificationPreferences(requestParameters, observe = 'body', reportProgress = false, options) {
        const _generalNotificationId = requestParameters["generalNotificationId"];
        const _active = requestParameters["active"];
        const _entityId = requestParameters["entityId"];
        const _entityType = requestParameters["entityType"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_generalNotificationId !== undefined && _generalNotificationId !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _generalNotificationId, 'generalNotificationId');
        }
        if (_active !== undefined && _active !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _active, 'active');
        }
        if (_entityId !== undefined && _entityId !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _entityId, 'entityId');
        }
        if (_entityType !== undefined && _entityType !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _entityType, 'entityType');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/notification-preferences`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getNotificationPreferencesUrl(requestParameters) {
        const _generalNotificationId = requestParameters["generalNotificationId"];
        const _active = requestParameters["active"];
        const _entityId = requestParameters["entityId"];
        const _entityType = requestParameters["entityType"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_generalNotificationId !== undefined && _generalNotificationId !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _generalNotificationId, 'generalNotificationId');
        }
        if (_active !== undefined && _active !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _active, 'active');
        }
        if (_entityId !== undefined && _entityId !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _entityId, 'entityId');
        }
        if (_entityType !== undefined && _entityType !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _entityType, 'entityType');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v1/notification-preferences${queryString ? `?${queryString}` : ''}`;
    }
    updateNotificationPreference(requestParameters, observe = 'body', reportProgress = false, options) {
        const _id = requestParameters["id"];
        if (_id === null || _id === undefined) {
            throw new Error('Required parameter id was null or undefined when calling updateNotificationPreference.');
        }
        const _notificationPreferencePutRequest = requestParameters["notificationPreferencePutRequest"];
        if (_notificationPreferencePutRequest === null || _notificationPreferencePutRequest === undefined) {
            throw new Error('Required parameter notificationPreferencePutRequest was null or undefined when calling updateNotificationPreference.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v1/notification-preferences/${encodeURIComponent(String(_id))}`, _notificationPreferencePutRequest, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    updateNotificationPreferenceUrl(requestParameters) {
        const _id = requestParameters["id"];
        if (_id === null || _id === undefined) {
            throw new Error('Required parameter id was null or undefined when calling updateNotificationPreference.');
        }
        const _notificationPreferencePutRequest = requestParameters["notificationPreferencePutRequest"];
        if (_notificationPreferencePutRequest === null || _notificationPreferencePutRequest === undefined) {
            throw new Error('Required parameter notificationPreferencePutRequest was null or undefined when calling updateNotificationPreference.');
        }
        return `${this.configuration.basePath}/client-api/v1/notification-preferences/${encodeURIComponent(String(_id))}`;
    }
}
NotificationPreferenceService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: NotificationPreferenceService, deps: [{ token: i1.HttpClient }, { token: ENGAGEMENT_BASE_PATH, optional: true }, { token: i2.EngagementConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
NotificationPreferenceService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: NotificationPreferenceService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: NotificationPreferenceService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [ENGAGEMENT_BASE_PATH]
                }] }, { type: i2.EngagementConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
