// @backbase/transactions-list-widget-ang/esm2015/lib/components/transaction/transaction-controls.component.js
import { Component, Input, Output, EventEmitter, ChangeDetectionStrategy } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/icon";
import * as i2 from "@angular/common";
import * as i3 from "@backbase/ui-ang/button";
import * as i4 from "@backbase/ui-ang/focus";
export class TransactionControlsComponent {
    constructor() {
        /**
         * Whether the notes editing form is shown or not.
         */
        this.isEditingNotes = false;
        /**
         * EventEmitter for triggering an isEditingNotesChange event.
         */
        this.isEditingNotesChange = new EventEmitter();
    }
    onAddNotes() {
        this.isEditingNotesChange.emit(true);
    }
    ngOnInit() {
        if (typeof this.transaction === 'undefined') {
            throw new Error(`"transaction" input is required in "${this.constructor.name}"`);
        }
    }
}
TransactionControlsComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TransactionControlsComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
TransactionControlsComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: TransactionControlsComponent, selector: "bb-transaction-controls", inputs: { transaction: "transaction", isEditingNotes: "isEditingNotes" }, outputs: { isEditingNotesChange: "isEditingNotesChange" }, ngImport: i0, template: "<button bbButton color=\"link\" (click)=\"onAddNotes()\" *ngIf=\"!isEditingNotes && !transaction?.notes\" bbFocus>\n  <bb-icon-ui name=\"note-add\"></bb-icon-ui>\n  <span class=\"sr-only\" i18n=\"Add a note to the current transaction@@transactions.details.addNote\">Add a note</span>\n</button>\n", components: [{ type: i1.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i3.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }, { type: i4.FocusDirective, selector: "[bbFocus]", inputs: ["bbFocus"], exportAs: ["bbFocus"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TransactionControlsComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-transaction-controls',
                    templateUrl: './transaction-controls.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { transaction: [{
                type: Input
            }], isEditingNotes: [{
                type: Input
            }], isEditingNotesChange: [{
                type: Output
            }] } });
