// @backbase/pay-bills-journey-ang/esm2015/src/directives/base-payment-edit-form-container.model.js
import { HostListener, Directive, Input } from '@angular/core';
import { of, Subject } from 'rxjs';
import { map, shareReplay, take, takeUntil } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/billpay-journeys-common";
import * as i2 from "../services/pay-bills-state.service";
import * as i3 from "@angular/router";
import * as i4 from "../services/pay-bills-navigation.service";
export class BasePaymentEditFormContainerDirective {
    /** @internal */
    constructor(dataService, stateService, route, discardModalService, routerService) {
        this.dataService = dataService;
        this.stateService = stateService;
        this.route = route;
        this.discardModalService = discardModalService;
        this.routerService = routerService;
        this.destroy$ = new Subject();
        this.isFormTouched = false;
        this.id$ = this.route.paramMap.pipe(take(1), map(params => params.get('id') || ''));
        this.payeesAndAccounts$ = this.dataService.getPayeesAndAccounts().pipe(shareReplay(1), takeUntil(this.destroy$));
    }
    onWindowUnload($event) {
        if ((this.isFormTouched || this.stateService.isFormProcessing) && !this.isDiscardModalOpen) {
            $event.returnValue = true;
        }
    }
    ngOnDestroy() {
        this.destroy$.next();
        this.destroy$.complete();
    }
    get isDiscardModalOpen() {
        return this.discardModalService.isDiscardModalOpen;
    }
    openModal() {
        return this.isFormTouched || this.stateService.isFormProcessing ? this.discardModalService.openModal() : of(true);
    }
    onModalResolve(isConfirmed) {
        this.discardModalService.onModalResolve(isConfirmed);
    }
    onFormValueChanged() {
        this.isFormTouched = true;
    }
    onCancelForm() {
        this.routerService.navigateToPayeeList();
    }
}
BasePaymentEditFormContainerDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BasePaymentEditFormContainerDirective, deps: [{ token: i1.BillpayPaymentDataService }, { token: i2.PayBillsStateService }, { token: i3.ActivatedRoute }, { token: i1.BillpayDiscardChangesModalService }, { token: i4.PayBillsNavigationService }], target: i0.ɵɵFactoryTarget.Directive });
BasePaymentEditFormContainerDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.15", type: BasePaymentEditFormContainerDirective, inputs: { accountNumberMasked: "accountNumberMasked", paymentDetailsTitle: "paymentDetailsTitle" }, host: { listeners: { "window:beforeunload": "onWindowUnload($event)" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BasePaymentEditFormContainerDirective, decorators: [{
            type: Directive
        }], ctorParameters: function () { return [{ type: i1.BillpayPaymentDataService }, { type: i2.PayBillsStateService }, { type: i3.ActivatedRoute }, { type: i1.BillpayDiscardChangesModalService }, { type: i4.PayBillsNavigationService }]; }, propDecorators: { accountNumberMasked: [{
                type: Input
            }], paymentDetailsTitle: [{
                type: Input
            }], onWindowUnload: [{
                type: HostListener,
                args: ['window:beforeunload', ['$event']]
            }] } });
