// @backbase/device-information-widget-ang/esm2015/src/components/bb-device-registered-on-ui/device-registered-on.component.js
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class DeviceRegisteredOnComponent {
}
DeviceRegisteredOnComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: DeviceRegisteredOnComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
DeviceRegisteredOnComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: DeviceRegisteredOnComponent, selector: "bb-device-registered-on-ui", inputs: { createdDate: "createdDate" }, ngImport: i0, template: "<div\n  *ngIf=\"createdDate\"\n  data-role=\"device-registered-on\"\n  i18n=\"Text describing when the device was registered@@bb-device-registered-on-ui.registration-date.helper\">\n  Registered on: {{ createdDate | date :'medium' }}\n</div>", directives: [{ type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "date": i1.DatePipe }, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: DeviceRegisteredOnComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-device-registered-on-ui',
                    templateUrl: './device-registered-on.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { createdDate: [{
                type: Input
            }] } });
