// @backbase/payord-form-builder-ang/esm2015/lib/payord-form-builder-ang.module.js
import { CommonModule, DatePipe, CurrencyPipe } from '@angular/common';
import { CUSTOM_ELEMENTS_SCHEMA, NgModule } from '@angular/core';
import { ReactiveFormsModule, FormsModule } from '@angular/forms';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { PayordAddressComponent } from './components/address/payord-address.component';
import { PayordAmountComponent } from './components/amount/payord-amount.component';
import { PayordBeneficiarySelectorComponent } from './components/beneficiary-selector/payord-beneficiary-selector.component';
import { PayordCheckboxComponent } from './components/checkbox/payord-checkbox.component';
import { PayordDebitAccountSelectorComponent } from './components/debit-account-selector/payord-debit-account-selector.component';
import { PayordRadioComponent } from './components/radio/payord-radio.component';
import { PayordScheduleComponent } from './components/schedule/payord-schedule.component';
import { PayordSelectComponent } from './components/select/payord-select.component';
import { PayordTextAreaComponent } from './components/textarea/payord-textarea.component';
import { PayordTextboxComponent } from './components/textbox/payord-text.component';
import { PayordFormComponent } from './containers/payord-form/payord-form.component';
import { PaymentFormFieldDirective } from './directives/form-field.directive';
import { PayordGroupComponent } from './components/group/payord-group.component';
import { PayordSwitchComponent } from './components/switch/payord-switch.component';
import { PaymentOrdersUiModule } from '@backbase/payment-orders-ang';
import { PayordDateComponent } from './components/date/payord-date.component';
import { PayordNumberComponent } from './components/number/payord-number.component';
import { ValidationMessageComponent } from './components/validation-message/validation-message.component';
import { PayordPhoneComponent } from './components/phone/payord-phone.component';
import { PayordSelectPaymentOptionComponent } from './components/select-payment-option/payord-select-payment-option';
import { PayordPaymentCostComponent } from './components/payment-cost/payment-cost.component';
import { AdaptiveFormComponent } from './containers/adaptive-form/adaptive-form.component';
import { PayordInternalAccountSelectorComponent } from './components/internal-account-selector/payord-internal-account-selector.component';
import { PayordCompactAmountComponent } from './components/compact-amount/payord-compact-amount.component';
import { PayordButtonGroupComponent } from './components/button-group/payord-button-group.component';
import { PayordScheduleHeaderComponent } from './components/schedule-header/payord-schedule-header';
import { PayordSanctionedCurrencyAmountComponent } from './components/sanctioned-currency-input/payord-sanctioned-currency-amount.component';
import { PayordCountrySelectorComponent } from './components/country-selector/payord-country-selector.component';
import { PayordChargeBearerComponent } from './components/charge-bearer/charge-bearer.component';
import { PayordSanctionedCountrySelectorComponent } from './components/sanctioned-country-selector/payord-sanctioned-country-selector.component';
import { MultiPaymentCardComponent } from './components/multi-payment-card/multi-payment-card.component';
import { MultiPaymentFormComponent } from './containers/multi-payment-form/multi-payment-form.component';
import { PayordCreditDebitSelectorComponent } from './components/credit-debit-selector/payord-credit-debit-selector';
import { PayordInternalAmountOptionsComponent } from './components/internal-amount-options/payord-internal-amount-options.component';
import { PayordHeaderComponent } from './components/header/payord-header.component';
import { PayordPurposeOfPaymentComponent } from './components/purpose-of-payment/payord-purpose-of-payment.component';
import { PayordAlertComponent } from './components/alert/payord-alert.component';
import { ErrorAlertComponent } from './components/error-alert/error-alert.component';
import { PayordScheduleSwitchComponent } from './components/schedule-switch/payord-schedule-switch.component';
import { PayordBeneficiaryGroupHeaderComponent } from './components/beneficiary-group-header/beneficiary-group-header.component';
import { MultiPaymentAccountSelectorComponent } from './components/account-selector/multipayment-account-selector.component';
import { PayordContactSelectorComponent } from './components/contact-selector/payord-contact-selector.component';
import { DynamicAriaAttributeDirective } from './directives/dynamic-aria-attribute.directive';
import { VisibilityService } from './services/visibility.service';
import { ComponentsFactoryService } from './services/components-factory.service';
import { InputTextModule } from '@backbase/ui-ang/input-text';
import { InputRadioGroupModule } from '@backbase/ui-ang/input-radio-group';
import { InputCheckboxModule } from '@backbase/ui-ang/input-checkbox';
import { InputDatepickerModule } from '@backbase/ui-ang/input-datepicker';
import { CurrencyInputModule } from '@backbase/ui-ang/currency-input';
import { TextareaModule } from '@backbase/ui-ang/textarea';
import { SwitchModule } from '@backbase/ui-ang/switch';
import { DropdownSingleSelectModule } from '@backbase/ui-ang/dropdown-single-select';
import { AccountSelectorModule } from '@backbase/ui-ang/account-selector';
import { CollapsibleModule } from '@backbase/ui-ang/collapsible';
import { AlertModule } from '@backbase/ui-ang/alert';
import { AmountModule } from '@backbase/ui-ang/amount';
import { FieldsetModule } from '@backbase/ui-ang/fieldset';
import { LoadButtonModule } from '@backbase/ui-ang/load-button';
import { InputNumberModule } from '@backbase/ui-ang/input-number';
import { InputPhoneModule } from '@backbase/ui-ang/input-phone';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { ProductSelectorModule } from '@backbase/ui-ang/product-selector';
import { ProductItemBasicAccountModule } from '@backbase/ui-ang/product-item-basic-account';
import { ButtonModule } from '@backbase/ui-ang/button';
import { IconModule } from '@backbase/ui-ang/icon';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { StepperModule } from '@backbase/ui-ang/stepper';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { AmountInputModule } from '@backbase/ui-ang/amount-input';
import * as i0 from "@angular/core";
const entryComponents = [
    PayordAddressComponent,
    PayordAmountComponent,
    PayordBeneficiarySelectorComponent,
    PayordCheckboxComponent,
    PayordDebitAccountSelectorComponent,
    PayordRadioComponent,
    PayordScheduleComponent,
    PayordSelectComponent,
    PayordTextAreaComponent,
    PayordTextboxComponent,
    PayordSwitchComponent,
    PayordDateComponent,
    PayordNumberComponent,
    PayordPhoneComponent,
    ValidationMessageComponent,
    PayordSelectPaymentOptionComponent,
    PayordPaymentCostComponent,
    PayordInternalAccountSelectorComponent,
    PayordCompactAmountComponent,
    PayordButtonGroupComponent,
    PayordScheduleHeaderComponent,
    PayordSanctionedCurrencyAmountComponent,
    PayordCountrySelectorComponent,
    PayordChargeBearerComponent,
    PayordSanctionedCountrySelectorComponent,
    PayordChargeBearerComponent,
    MultiPaymentCardComponent,
    MultiPaymentFormComponent,
    PayordCreditDebitSelectorComponent,
    PayordInternalAmountOptionsComponent,
    PayordHeaderComponent,
    PayordPurposeOfPaymentComponent,
    PayordAlertComponent,
    ErrorAlertComponent,
    PayordScheduleSwitchComponent,
    PayordBeneficiaryGroupHeaderComponent,
    MultiPaymentAccountSelectorComponent,
    PayordContactSelectorComponent,
];
const modules = [
    InputTextModule,
    InputRadioGroupModule,
    InputCheckboxModule,
    InputDatepickerModule,
    InputNumberModule,
    InputPhoneModule,
    CurrencyInputModule,
    TextareaModule,
    SwitchModule,
    DropdownSingleSelectModule,
    FormsModule,
    ReactiveFormsModule,
    InputRadioGroupModule,
    AccountSelectorModule,
    CollapsibleModule,
    AlertModule,
    PaymentOrdersUiModule,
    AmountModule,
    FieldsetModule,
    LoadButtonModule,
    LoadingIndicatorModule,
    ProductSelectorModule,
    ProductItemBasicAccountModule,
    ButtonModule,
    IconModule,
    EmptyStateModule,
    StepperModule,
    TooltipModule,
    AmountInputModule,
];
const components = [
    PayordFormComponent,
    PayordGroupComponent,
    AdaptiveFormComponent,
    ...entryComponents,
];
const directives = [PaymentFormFieldDirective, DynamicAriaAttributeDirective];
/**
 * Exports the UI components, directives and modules for the form builder
 */
export class PayordFormBuilderAngModule {
}
/** @nocollapse */ PayordFormBuilderAngModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordFormBuilderAngModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ PayordFormBuilderAngModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordFormBuilderAngModule, declarations: [PayordFormComponent,
        PayordGroupComponent,
        AdaptiveFormComponent, PayordAddressComponent,
        PayordAmountComponent,
        PayordBeneficiarySelectorComponent,
        PayordCheckboxComponent,
        PayordDebitAccountSelectorComponent,
        PayordRadioComponent,
        PayordScheduleComponent,
        PayordSelectComponent,
        PayordTextAreaComponent,
        PayordTextboxComponent,
        PayordSwitchComponent,
        PayordDateComponent,
        PayordNumberComponent,
        PayordPhoneComponent,
        ValidationMessageComponent,
        PayordSelectPaymentOptionComponent,
        PayordPaymentCostComponent,
        PayordInternalAccountSelectorComponent,
        PayordCompactAmountComponent,
        PayordButtonGroupComponent,
        PayordScheduleHeaderComponent,
        PayordSanctionedCurrencyAmountComponent,
        PayordCountrySelectorComponent,
        PayordChargeBearerComponent,
        PayordSanctionedCountrySelectorComponent,
        PayordChargeBearerComponent,
        MultiPaymentCardComponent,
        MultiPaymentFormComponent,
        PayordCreditDebitSelectorComponent,
        PayordInternalAmountOptionsComponent,
        PayordHeaderComponent,
        PayordPurposeOfPaymentComponent,
        PayordAlertComponent,
        ErrorAlertComponent,
        PayordScheduleSwitchComponent,
        PayordBeneficiaryGroupHeaderComponent,
        MultiPaymentAccountSelectorComponent,
        PayordContactSelectorComponent, PaymentFormFieldDirective, DynamicAriaAttributeDirective], imports: [CommonModule, BackbaseCoreModule, InputTextModule,
        InputRadioGroupModule,
        InputCheckboxModule,
        InputDatepickerModule,
        InputNumberModule,
        InputPhoneModule,
        CurrencyInputModule,
        TextareaModule,
        SwitchModule,
        DropdownSingleSelectModule,
        FormsModule,
        ReactiveFormsModule,
        InputRadioGroupModule,
        AccountSelectorModule,
        CollapsibleModule,
        AlertModule,
        PaymentOrdersUiModule,
        AmountModule,
        FieldsetModule,
        LoadButtonModule,
        LoadingIndicatorModule,
        ProductSelectorModule,
        ProductItemBasicAccountModule,
        ButtonModule,
        IconModule,
        EmptyStateModule,
        StepperModule,
        TooltipModule,
        AmountInputModule], exports: [PayordFormComponent,
        PayordGroupComponent,
        AdaptiveFormComponent, PayordAddressComponent,
        PayordAmountComponent,
        PayordBeneficiarySelectorComponent,
        PayordCheckboxComponent,
        PayordDebitAccountSelectorComponent,
        PayordRadioComponent,
        PayordScheduleComponent,
        PayordSelectComponent,
        PayordTextAreaComponent,
        PayordTextboxComponent,
        PayordSwitchComponent,
        PayordDateComponent,
        PayordNumberComponent,
        PayordPhoneComponent,
        ValidationMessageComponent,
        PayordSelectPaymentOptionComponent,
        PayordPaymentCostComponent,
        PayordInternalAccountSelectorComponent,
        PayordCompactAmountComponent,
        PayordButtonGroupComponent,
        PayordScheduleHeaderComponent,
        PayordSanctionedCurrencyAmountComponent,
        PayordCountrySelectorComponent,
        PayordChargeBearerComponent,
        PayordSanctionedCountrySelectorComponent,
        PayordChargeBearerComponent,
        MultiPaymentCardComponent,
        MultiPaymentFormComponent,
        PayordCreditDebitSelectorComponent,
        PayordInternalAmountOptionsComponent,
        PayordHeaderComponent,
        PayordPurposeOfPaymentComponent,
        PayordAlertComponent,
        ErrorAlertComponent,
        PayordScheduleSwitchComponent,
        PayordBeneficiaryGroupHeaderComponent,
        MultiPaymentAccountSelectorComponent,
        PayordContactSelectorComponent, PaymentFormFieldDirective, DynamicAriaAttributeDirective, InputTextModule,
        InputRadioGroupModule,
        InputCheckboxModule,
        InputDatepickerModule,
        InputNumberModule,
        InputPhoneModule,
        CurrencyInputModule,
        TextareaModule,
        SwitchModule,
        DropdownSingleSelectModule,
        FormsModule,
        ReactiveFormsModule,
        InputRadioGroupModule,
        AccountSelectorModule,
        CollapsibleModule,
        AlertModule,
        PaymentOrdersUiModule,
        AmountModule,
        FieldsetModule,
        LoadButtonModule,
        LoadingIndicatorModule,
        ProductSelectorModule,
        ProductItemBasicAccountModule,
        ButtonModule,
        IconModule,
        EmptyStateModule,
        StepperModule,
        TooltipModule,
        AmountInputModule] });
/** @nocollapse */ PayordFormBuilderAngModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordFormBuilderAngModule, providers: [
        DatePipe,
        CurrencyPipe,
        VisibilityService,
        ComponentsFactoryService,
    ], imports: [[CommonModule, BackbaseCoreModule, ...modules], InputTextModule,
        InputRadioGroupModule,
        InputCheckboxModule,
        InputDatepickerModule,
        InputNumberModule,
        InputPhoneModule,
        CurrencyInputModule,
        TextareaModule,
        SwitchModule,
        DropdownSingleSelectModule,
        FormsModule,
        ReactiveFormsModule,
        InputRadioGroupModule,
        AccountSelectorModule,
        CollapsibleModule,
        AlertModule,
        PaymentOrdersUiModule,
        AmountModule,
        FieldsetModule,
        LoadButtonModule,
        LoadingIndicatorModule,
        ProductSelectorModule,
        ProductItemBasicAccountModule,
        ButtonModule,
        IconModule,
        EmptyStateModule,
        StepperModule,
        TooltipModule,
        AmountInputModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordFormBuilderAngModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components, ...directives],
                    exports: [...components, ...directives, ...modules],
                    entryComponents,
                    imports: [CommonModule, BackbaseCoreModule, ...modules],
                    providers: [
                        DatePipe,
                        CurrencyPipe,
                        VisibilityService,
                        ComponentsFactoryService,
                    ],
                    schemas: [CUSTOM_ELEMENTS_SCHEMA],
                }]
        }] });
