// @backbase/cards-common-ang/esm2015/src/components/bb-payment-card-limits/payment-card-limits.component.js
import { Component, Directive, EventEmitter, Input, Output } from '@angular/core';
import { BbTemplate } from '@backbase/foundation-ang/core';
import * as i0 from "@angular/core";
import * as i1 from "./payment-card-limit.component";
import * as i2 from "@backbase/ui-ang/loading-indicator";
import * as i3 from "@angular/common";
export class PaymentCardLimitsComponent {
    constructor() {
        /**
         * Event to update limit of payment card;
         */
        this.updateLimit = new EventEmitter();
        /**
         * Reference to instance of customizable component.
         */
        this.hostRef = this;
    }
    onUpdateLimit(limit) {
        if (this.paymentCard) {
            const limits = this.paymentCard.limits;
            if (limits) {
                this.updateLimit.emit(limits.map((item) => {
                    let amount = item.amount;
                    if (limit.cardLimit && item.id === limit.cardLimit.id) {
                        amount = +limit.amount;
                    }
                    return {
                        amount,
                        id: item.id,
                        frequency: item.frequency,
                    };
                }));
            }
        }
    }
}
PaymentCardLimitsComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentCardLimitsComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
PaymentCardLimitsComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: PaymentCardLimitsComponent, selector: "bb-payment-card-limits", inputs: { paymentCard: "paymentCard", isLimitsLoading: "isLimitsLoading" }, outputs: { updateLimit: "updateLimit" }, ngImport: i0, template: "<ng-container bbCardsUpdateLimitsCustomizable [bbHostRef]=\"hostRef\"></ng-container>\n<ng-template bbCardsUpdateLimitsCustomizable let-hostRef>\n  <div class=\"card bb-list\" *ngIf=\"hostRef.paymentCard?.limits\">\n    <div class=\"card-header\">\n      <h2 i18n=\"Limits update heading|update card limits header@@card.updateLimit.heading\">\n        Limits\n      </h2>\n    </div>\n    <div class=\"card-body\">\n      <div class=\"bb-list\" *ngIf=\"!hostRef.isLimitsLoading; else loadingTemplate\">\n        <div class=\"bb-list__item\" *ngFor=\"let limit of hostRef.paymentCard.limits\">\n          <bb-payment-card-limit \n            [limit]=\"limit\"\n            [currency]=\"hostRef.paymentCard.currency\"\n            (updateLimit)=\"hostRef.onUpdateLimit($event)\"\n          ></bb-payment-card-limit>\n        </div>\n      </div>\n    </div>\n  </div>\n</ng-template>\n\n<ng-template #loadingTemplate>\n  <bb-loading-indicator-ui loaderSize=\"lg\"></bb-loading-indicator-ui>\n</ng-template>", components: [{ type: i0.forwardRef(function () { return i1.PaymentCardLimitComponent; }), selector: "bb-payment-card-limit", inputs: ["currency", "limit"], outputs: ["updateLimit"] }, { type: i0.forwardRef(function () { return i2.LoadingIndicatorComponent; }), selector: "bb-loading-indicator-ui", inputs: ["text", "loaderSize", "showDelay", "hasBackground", "inline"] }], directives: [{ type: i0.forwardRef(function () { return CardsUpdateLimitsCustomizableDirective; }), selector: "[bbCardsUpdateLimitsCustomizable]" }, { type: i0.forwardRef(function () { return i3.NgIf; }), selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i0.forwardRef(function () { return i3.NgForOf; }), selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentCardLimitsComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-payment-card-limits',
                    templateUrl: './payment-card-limits.component.html',
                }]
        }], propDecorators: { paymentCard: [{
                type: Input
            }], isLimitsLoading: [{
                type: Input
            }], updateLimit: [{
                type: Output
            }] } });
export class CardsUpdateLimitsCustomizableDirective extends BbTemplate {
}
CardsUpdateLimitsCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardsUpdateLimitsCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
CardsUpdateLimitsCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.15", type: CardsUpdateLimitsCustomizableDirective, selector: "[bbCardsUpdateLimitsCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardsUpdateLimitsCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbCardsUpdateLimitsCustomizable]',
                }]
        }] });
