// @backbase/cards-travel-notice-widget-ang/esm2015/src/components/cards-travel-notice-create/cards-travel-notice-country-selector.component.js
import { Component, forwardRef, Input } from '@angular/core';
import { FormControl, NG_VALUE_ACCESSOR } from '@angular/forms';
import * as i0 from "@angular/core";
import * as i1 from "@ng-select/ng-select";
import * as i2 from "@angular/forms";
export class CardsTravelNoticeCountrySelectorComponent {
    constructor() {
        this.destionationListControl = new FormControl([]);
        this.updateCountryList = (destinations) => {
            this.onChange(destinations.map((item) => ({
                country: item.parent ? item.parent : item.value,
                region: item.parent && item.value,
            })));
        };
        this.onChange = () => { };
        this.onTouched = () => { };
    }
    // eslint-disable-next-line
    set destinationList(value) {
        this.destinationsDisplay = [
            ...value.map((item) => ({
                name: item.name,
                value: item.code,
            })),
            ...value
                .filter((item) => item.regions !== undefined)
                .reduce((acc, { regions = [], code: countryCode }) => [
                ...acc,
                ...regions.map((region) => ({
                    name: `${region.name} (${countryCode})`,
                    value: region.code,
                    parent: countryCode,
                })),
            ], []),
        ].sort((left, right) => {
            if (left.name === right.name) {
                return 0;
            }
            return left.name > right.name ? 1 : -1;
        });
        this.fullDestinationList = value;
    }
    get destinationList() {
        return this.fullDestinationList || [];
    }
    get destinationListDisplay() {
        return this.destinationsDisplay || [];
    }
    writeValue(countries) {
        this.destionationListControl.setValue(countries.map(({ country, region }) => (region ? `${country}:${region}` : country)));
    }
    registerOnChange(fn) {
        this.onChange = fn;
    }
    registerOnTouched(fn) {
        this.onTouched = fn;
    }
    ngOnInit() {
        this.countryListSubscription = this.destionationListControl.valueChanges.subscribe(this.updateCountryList);
    }
    ngOnDestroy() {
        if (this.countryListSubscription) {
            this.countryListSubscription.unsubscribe();
        }
    }
}
CardsTravelNoticeCountrySelectorComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardsTravelNoticeCountrySelectorComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
CardsTravelNoticeCountrySelectorComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: CardsTravelNoticeCountrySelectorComponent, selector: "bb-cards-travel-notice-country-selector", inputs: { destinationList: "destinationList" }, providers: [
        {
            provide: NG_VALUE_ACCESSOR,
            // eslint-disable-next-line @angular-eslint/no-forward-ref
            useExisting: forwardRef(() => CardsTravelNoticeCountrySelectorComponent),
            multi: true,
        },
    ], ngImport: i0, template: "<ng-select \n  [items]=\"destinationListDisplay\"\n  class=\"bb-country-selector\"\n  [hideSelected]=\"true\"\n  [multiple]=\"true\"\n  bindLabel=\"name\"\n  [formControl]=\"destionationListControl\"\n  aria-label=\"Search or select countries\"\n  i18n-aria-label=\"Search or select countries aria-label@@cards.create.travel.notice.country.dropdown.tagSelector.aria-label\"\n  placeholder=\"Search or select countries\"\n  i18n-placeholder=\"Search or select countries@@cards.create.travel.notice.country.dropdown.tagSelector.placeholder\"\n></ng-select>", components: [{ type: i1.NgSelectComponent, selector: "ng-select", inputs: ["markFirst", "dropdownPosition", "loading", "closeOnSelect", "hideSelected", "selectOnTab", "bufferAmount", "selectableGroup", "selectableGroupAsModel", "searchFn", "trackByFn", "clearOnBackspace", "labelForId", "inputAttrs", "readonly", "searchWhileComposing", "minTermLength", "editableSearchTerm", "keyDownFn", "multiple", "addTag", "searchable", "clearable", "isOpen", "items", "compareWith", "clearSearchOnAdd", "bindLabel", "placeholder", "notFoundText", "typeToSearchText", "addTagText", "loadingText", "clearAllText", "virtualScroll", "openOnEnter", "appendTo", "bindValue", "appearance", "maxSelectedItems", "groupBy", "groupValue", "tabIndex", "typeahead"], outputs: ["blur", "focus", "change", "open", "close", "search", "clear", "add", "remove", "scroll", "scrollToEnd"] }], directives: [{ type: i2.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { type: i2.FormControlDirective, selector: "[formControl]", inputs: ["disabled", "formControl", "ngModel"], outputs: ["ngModelChange"], exportAs: ["ngForm"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardsTravelNoticeCountrySelectorComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-cards-travel-notice-country-selector',
                    templateUrl: 'cards-travel-notice-country-selector.component.html',
                    providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            // eslint-disable-next-line @angular-eslint/no-forward-ref
                            useExisting: forwardRef(() => CardsTravelNoticeCountrySelectorComponent),
                            multi: true,
                        },
                    ],
                }]
        }], propDecorators: { destinationList: [{
                type: Input
            }] } });
