// @backbase/notifications-popups-widget-ang/esm2015/lib/components/notifications-sticky-list-item/notifications-sticky-list-item.component.js
import { Component, Directive, EventEmitter, Input, Output } from '@angular/core';
import { BbTemplate } from '@backbase/foundation-ang/core';
import { parseNotificationSeverityLevel } from '../../helpers/mappers';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/alert";
import * as i2 from "@backbase/notifications-common-ang";
export class NotificationsStickyListItemComponent {
    constructor() {
        /**
         * Notification stream item close event
         * Emits notification stream item id
         */
        // eslint-disable-next-line @angular-eslint/no-output-native
        this.close = new EventEmitter();
    }
    /**
     * Method to handle close event from alert component
     *
     * @param notification Notification data
     */
    onClose(notification) {
        this.close.emit(notification);
    }
    /**
     * Method to parse severity level into alert modifier
     *
     * @param level Severity level
     * @returns Notification severity level type
     */
    parseNotificationSeverityLevel(level) {
        return parseNotificationSeverityLevel(level);
    }
}
/** @nocollapse */ NotificationsStickyListItemComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsStickyListItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ NotificationsStickyListItemComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: NotificationsStickyListItemComponent, selector: "bb-notifications-sticky-list-item", inputs: { notification: "notification" }, outputs: { close: "close" }, ngImport: i0, template: "<ng-container\n  bbNotificationsStickyListItemCustomizable\n  [bbHostRef]=\"this\"\n  [bbTemplateContext]=\"notification\"\n></ng-container>\n\n<ng-template bbNotificationsStickyListItemCustomizable let-host let-notification=\"context\">\n  <bb-alert-ui\n    [dismissible]=\"true\"\n    [modifier]=\"host.parseNotificationSeverityLevel(notification?.level)\"\n    [title]=\"notification?.title || ''\"\n    (close)=\"host.onClose(notification)\"\n  >\n    <div\n      class=\"bb-subtitle\"\n      data-role=\"notification-sticky-message\"\n      [innerHTML]=\"notification?.message | interpolateLink: notification:true:'alert-link'\"\n    ></div>\n  </bb-alert-ui>\n</ng-template>\n", components: [{ type: i0.forwardRef(function () { return i1.AlertComponent; }), selector: "bb-alert-ui", inputs: ["modifier", "dismissible", "title", "message"], outputs: ["close"] }], directives: [{ type: i0.forwardRef(function () { return NotificationsStickyListItemCustomizableDirective; }), selector: "[bbNotificationsStickyListItemCustomizable]" }], pipes: { "interpolateLink": i0.forwardRef(function () { return i2.ӨInterpolateLinkPipe; }) } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsStickyListItemComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-notifications-sticky-list-item',
                    templateUrl: './notifications-sticky-list-item.component.html',
                }]
        }], propDecorators: { notification: [{
                type: Input
            }], close: [{
                type: Output
            }] } });
export class NotificationsStickyListItemCustomizableDirective extends BbTemplate {
}
/** @nocollapse */ NotificationsStickyListItemCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsStickyListItemCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
/** @nocollapse */ NotificationsStickyListItemCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: NotificationsStickyListItemCustomizableDirective, selector: "[bbNotificationsStickyListItemCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: NotificationsStickyListItemCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbNotificationsStickyListItemCustomizable]',
                }]
        }] });
