// @backbase/product-summary-ui/esm2015/src/bb-product-kind-ui/product-kind.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ProductItemCurrentAccountModule } from '@backbase/ui-ang/product-item-current-account';
import { ProductItemSavingsAccountModule } from '@backbase/ui-ang/product-item-savings-account';
import { ProductItemTermDepositModule } from '@backbase/ui-ang/product-item-term-deposit';
import { ProductItemLoanModule } from '@backbase/ui-ang/product-item-loan';
import { ProductItemDebitCardModule } from '@backbase/ui-ang/product-item-debit-card';
import { ProductItemCreditCardModule } from '@backbase/ui-ang/product-item-credit-card';
import { ProductItemInvestmentAccountModule } from '@backbase/ui-ang/product-item-investment-account';
import { ProductItemBasicAccountModule } from '@backbase/ui-ang/product-item-basic-account';
import { KeyboardClickModule } from '@backbase/ui-ang/keyboard-click-directive';
import { AmountModule } from '@backbase/ui-ang/amount';
import { DatePipeModule, ValidationModule } from '@backbase/product-summary-common-ang';
import { ProductKindComponent, ProductKindCustomizableDirective } from './product-kind-ui.component';
import { ProductItemCustomModule } from '../bb-product-item-custom-title-ui/product-item-custom.module';
import { ProductNumberModule } from '../bb-product-number/product-number.module';
import { ProductSummaryAccountLogoModule } from '../bb-product-summary-account-logo-ui/bb-product-summary-account-logo.module';
import * as i0 from "@angular/core";
const uiModules = [
    ProductItemCurrentAccountModule,
    ProductItemSavingsAccountModule,
    ProductItemTermDepositModule,
    ProductItemLoanModule,
    ProductItemCreditCardModule,
    ProductItemDebitCardModule,
    ProductItemInvestmentAccountModule,
    ProductItemBasicAccountModule,
    ProductItemCustomModule,
    KeyboardClickModule,
    ProductNumberModule,
    ProductSummaryAccountLogoModule,
    AmountModule,
    ValidationModule,
    DatePipeModule,
];
const components = [ProductKindComponent];
const directives = [ProductKindCustomizableDirective];
export class ProductKindModule {
}
ProductKindModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductKindModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductKindModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductKindModule, declarations: [ProductKindComponent, ProductKindCustomizableDirective], imports: [CommonModule, ProductItemCurrentAccountModule,
        ProductItemSavingsAccountModule,
        ProductItemTermDepositModule,
        ProductItemLoanModule,
        ProductItemCreditCardModule,
        ProductItemDebitCardModule,
        ProductItemInvestmentAccountModule,
        ProductItemBasicAccountModule,
        ProductItemCustomModule,
        KeyboardClickModule,
        ProductNumberModule,
        ProductSummaryAccountLogoModule,
        AmountModule,
        ValidationModule,
        DatePipeModule], exports: [ProductKindComponent, ProductKindCustomizableDirective, ProductItemCurrentAccountModule,
        ProductItemSavingsAccountModule,
        ProductItemTermDepositModule,
        ProductItemLoanModule,
        ProductItemCreditCardModule,
        ProductItemDebitCardModule,
        ProductItemInvestmentAccountModule,
        ProductItemBasicAccountModule,
        ProductItemCustomModule,
        KeyboardClickModule,
        ProductNumberModule,
        ProductSummaryAccountLogoModule,
        AmountModule,
        ValidationModule,
        DatePipeModule] });
ProductKindModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductKindModule, imports: [[CommonModule, ...uiModules], ProductItemCurrentAccountModule,
        ProductItemSavingsAccountModule,
        ProductItemTermDepositModule,
        ProductItemLoanModule,
        ProductItemCreditCardModule,
        ProductItemDebitCardModule,
        ProductItemInvestmentAccountModule,
        ProductItemBasicAccountModule,
        ProductItemCustomModule,
        KeyboardClickModule,
        ProductNumberModule,
        ProductSummaryAccountLogoModule,
        AmountModule,
        ValidationModule,
        DatePipeModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductKindModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules],
                    declarations: [...components, ...directives],
                    exports: [...components, ...directives, ...uiModules],
                }]
        }] });
