// @backbase/payment-orders-ang/esm2015/lib/ui/models/index.js
export class PaymentOrdersBadRequest {
    constructor(payment) {
        this.payment = payment;
        this.kind = 'PaymentOrdersBadRequest';
    }
}
export class UnknownError {
    constructor(origin) {
        this.origin = origin;
        this.kind = 'UnknownError';
    }
}
export var Routes;
(function (Routes) {
    Routes["FORM"] = "form";
    Routes["REVIEW"] = "review";
})(Routes || (Routes = {}));
export var PaymentState;
(function (PaymentState) {
    PaymentState["INITIAL"] = "INITIAL";
    PaymentState["REVIEW"] = "REVIEW";
    PaymentState["EDIT"] = "EDIT";
})(PaymentState || (PaymentState = {}));
export var ExecutionTimeFrame;
(function (ExecutionTimeFrame) {
    ExecutionTimeFrame["HOUR"] = "HOUR";
    ExecutionTimeFrame["CALENDAR_DAY"] = "CALENDAR_DAY";
    ExecutionTimeFrame["BUSINESS_DAY"] = "BUSINESS_DAY";
})(ExecutionTimeFrame || (ExecutionTimeFrame = {}));
export var ChargeBearer;
(function (ChargeBearer) {
    ChargeBearer["OUR"] = "OUR";
    ChargeBearer["BEN"] = "BEN";
    ChargeBearer["SHA"] = "SHA";
})(ChargeBearer || (ChargeBearer = {}));
export var ModalState;
(function (ModalState) {
    ModalState[ModalState["Preview"] = 0] = "Preview";
    ModalState[ModalState["Hidden"] = 1] = "Hidden";
})(ModalState || (ModalState = {}));
export var ProgressStatus;
(function (ProgressStatus) {
    ProgressStatus["CONFIRMATION_PENDING"] = "CONFIRMATION_PENDING";
    ProgressStatus["USER_DECLINED"] = "USER_DECLINED";
    ProgressStatus["SYSTEM_DECLINED"] = "SYSTEM_DECLINED";
    ProgressStatus["PENDING_APPROVAL"] = "PENDING_APPROVAL";
    ProgressStatus["SENDING"] = "SENDING";
    ProgressStatus["ACCEPTED"] = "ACCEPTED";
    ProgressStatus["REJECTED"] = "REJECTED";
    ProgressStatus["PROCESSED"] = "PROCESSED";
    ProgressStatus["NO_PROGRESS"] = "NO_PROGRESS";
})(ProgressStatus || (ProgressStatus = {}));
export var ManualBatchState;
(function (ManualBatchState) {
    ManualBatchState["CLOSED"] = "CLOSED";
    ManualBatchState["INITIAL"] = "INITIAL";
    ManualBatchState["DRAFT"] = "DRAFT";
    ManualBatchState["REVIEW"] = "REVIEW";
    ManualBatchState["VALID"] = "VALID";
})(ManualBatchState || (ManualBatchState = {}));
export var PaymentSubmitStatuses;
(function (PaymentSubmitStatuses) {
    PaymentSubmitStatuses["SUCCESS"] = "success";
    PaymentSubmitStatuses["ERROR"] = "error";
    PaymentSubmitStatuses["PENDING"] = "pending";
    PaymentSubmitStatuses["PENDING_EDITED"] = "pending_edited";
})(PaymentSubmitStatuses || (PaymentSubmitStatuses = {}));
export var ScrollToElementBehavior;
(function (ScrollToElementBehavior) {
    ScrollToElementBehavior["AUTO"] = "auto";
    ScrollToElementBehavior["SMOOTH"] = "smooth";
})(ScrollToElementBehavior || (ScrollToElementBehavior = {}));
export var ScrollToElementBlock;
(function (ScrollToElementBlock) {
    ScrollToElementBlock["START"] = "start";
    ScrollToElementBlock["CENTER"] = "center";
    ScrollToElementBlock["END"] = "end";
    ScrollToElementBlock["NEAREST"] = "nearest";
})(ScrollToElementBlock || (ScrollToElementBlock = {}));
export var ScrollToElementInline;
(function (ScrollToElementInline) {
    ScrollToElementInline["START"] = "start";
    ScrollToElementInline["CENTER"] = "center";
    ScrollToElementInline["END"] = "end";
    ScrollToElementInline["NEAREST"] = "nearest";
})(ScrollToElementInline || (ScrollToElementInline = {}));
