// @backbase/payment-orders-ang/esm2015/lib/ui/models/types.js
import { PaymentOrdersHttpService, } from '@backbase/data-ang/payment-order';
import { BatchOrderError } from '../constants/constants';
export { PaymentOrdersHttpService as PaymentOrdersService, };
export var ViewType;
(function (ViewType) {
    ViewType["CARD"] = "CARD";
    ViewType["FORM"] = "FORM";
    ViewType["BUTTON"] = "BUTTON";
})(ViewType || (ViewType = {}));
export class BreachLimitError {
    constructor(reportError) {
        this.reportError = reportError;
        this.kind = BatchOrderError.BREACH_LIMIT;
    }
}
export var PaymentTemplateAction;
(function (PaymentTemplateAction) {
    PaymentTemplateAction["CREATE"] = "CREATE";
    PaymentTemplateAction["VIEW"] = "VIEW";
})(PaymentTemplateAction || (PaymentTemplateAction = {}));
/** @deprecated Use `CreditDebitMixedIndicator` from `@backbase/data-ang/payment-batch` instead */
export var CreditDebitMixedIndicator;
(function (CreditDebitMixedIndicator) {
    CreditDebitMixedIndicator["CREDIT"] = "CREDIT";
    CreditDebitMixedIndicator["DEBIT"] = "DEBIT";
    CreditDebitMixedIndicator["MIXED"] = "MIXED";
})(CreditDebitMixedIndicator || (CreditDebitMixedIndicator = {}));
export const RoutableModalParamName = 'modalOpened';
