// @backbase/accessgroup-ui/esm2015/lib/business-function-group-name/business-function-group-name.component.js
import { Component, Input } from '@angular/core';
import { TranslationsService } from '@backbase/accessgroup-common-ang';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/accessgroup-common-ang";
export class BusinessFunctionGroupNameComponent {
    constructor(translationsService) {
        this.translationsService = translationsService;
    }
    get groupName() {
        if (!this.groupKey)
            return '';
        return this.translationsService.getBusinessFunctionGroupName(this.groupKey) || this.groupKey;
    }
}
/** @nocollapse */ BusinessFunctionGroupNameComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BusinessFunctionGroupNameComponent, deps: [{ token: i1.TranslationsService }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ BusinessFunctionGroupNameComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: BusinessFunctionGroupNameComponent, selector: "bb-business-function-group-name", inputs: { groupKey: "groupKey" }, ngImport: i0, template: "{{ groupName }}" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BusinessFunctionGroupNameComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-business-function-group-name',
                    templateUrl: './business-function-group-name.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.TranslationsService }]; }, propDecorators: { groupKey: [{
                type: Input
            }] } });
