// @backbase/ui-ang/esm2015/collapsible-accordion/collapsible-accordion.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { CollapsibleAccordionComponent } from './collapsible-accordion.component';
import * as i0 from "@angular/core";
export class CollapsibleAccordionModule {
}
CollapsibleAccordionModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CollapsibleAccordionModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CollapsibleAccordionModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CollapsibleAccordionModule, declarations: [CollapsibleAccordionComponent], imports: [CommonModule], exports: [CollapsibleAccordionComponent] });
CollapsibleAccordionModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CollapsibleAccordionModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CollapsibleAccordionModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [CollapsibleAccordionComponent],
                    exports: [CollapsibleAccordionComponent],
                }]
        }] });
