// @backbase/accounts-transactions-journeys-shared-ui-account-ang/bundles/backbase-accounts-transactions-journeys-shared-ui-account-ang.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('@backbase/product-summary-common-ang'), require('@backbase/ui-ang/amount'), require('@backbase/ui-ang/date-pipe'), require('@backbase/ui-ang/payment-card'), require('@backbase/ui-ang/tooltip-directive'), require('@backbase/product-summary-ui'), require('@backbase/accounts-transactions-journeys-shared-ui-transform-ang'), require('@backbase/ui-ang/icon')) :
    typeof define === 'function' && define.amd ? define('@backbase/accounts-transactions-journeys-shared-ui-account-ang', ['exports', '@angular/core', '@angular/common', '@backbase/product-summary-common-ang', '@backbase/ui-ang/amount', '@backbase/ui-ang/date-pipe', '@backbase/ui-ang/payment-card', '@backbase/ui-ang/tooltip-directive', '@backbase/product-summary-ui', '@backbase/accounts-transactions-journeys-shared-ui-transform-ang', '@backbase/ui-ang/icon'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["accounts-transactions-journeys-shared-ui-account-ang"] = {}), global.ng.core, global.ng.common, global.i12, global.i3, global.i13, global.i4, global.i8, global.i2, global.i10, global.i1));
})(this, (function (exports, i0, i7, i12, i3, i13, i4, i8, i2, i10, i1) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i7__namespace = /*#__PURE__*/_interopNamespace(i7);
    var i12__namespace = /*#__PURE__*/_interopNamespace(i12);
    var i3__namespace = /*#__PURE__*/_interopNamespace(i3);
    var i13__namespace = /*#__PURE__*/_interopNamespace(i13);
    var i4__namespace = /*#__PURE__*/_interopNamespace(i4);
    var i8__namespace = /*#__PURE__*/_interopNamespace(i8);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i10__namespace = /*#__PURE__*/_interopNamespace(i10);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from, pack) {
        if (pack || arguments.length === 2)
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar)
                        ar = Array.prototype.slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
        return to.concat(ar || Array.prototype.slice.call(from));
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    /**
     * Transforms data to payment-card info
     */
    var ToPaymentCardDataPipe = /** @class */ (function () {
        function ToPaymentCardDataPipe() {
        }
        ToPaymentCardDataPipe.prototype.transform = function (debitCard) {
            return {
                number: (debitCard === null || debitCard === void 0 ? void 0 : debitCard.number) || '',
                expirationDate: (debitCard === null || debitCard === void 0 ? void 0 : debitCard.expiryDate) ? new Date(debitCard.expiryDate) : undefined,
                name: '',
                vendor: '',
                cardName: '',
            };
        };
        return ToPaymentCardDataPipe;
    }());
    /** @nocollapse */ ToPaymentCardDataPipe.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ToPaymentCardDataPipe, deps: [], target: i0__namespace.ɵɵFactoryTarget.Pipe });
    /** @nocollapse */ ToPaymentCardDataPipe.ɵpipe = i0__namespace.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ToPaymentCardDataPipe, name: "bbToPaymentCardData" });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ToPaymentCardDataPipe, decorators: [{
                type: i0.Pipe,
                args: [{ name: 'bbToPaymentCardData' }]
            }] });

    var pipes$2 = [ToPaymentCardDataPipe];
    var ToPaymentCardDataPipeModule = /** @class */ (function () {
        function ToPaymentCardDataPipeModule() {
        }
        return ToPaymentCardDataPipeModule;
    }());
    /** @nocollapse */ ToPaymentCardDataPipeModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ToPaymentCardDataPipeModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    /** @nocollapse */ ToPaymentCardDataPipeModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ToPaymentCardDataPipeModule, declarations: [ToPaymentCardDataPipe], imports: [i7.CommonModule], exports: [ToPaymentCardDataPipe] });
    /** @nocollapse */ ToPaymentCardDataPipeModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ToPaymentCardDataPipeModule, imports: [[i7.CommonModule]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: ToPaymentCardDataPipeModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [i7.CommonModule],
                        declarations: __spreadArray([], __read(pipes$2)),
                        exports: __spreadArray([], __read(pipes$2)),
                    }]
            }] });

    /**
     * Verifies if account has external info
     */
    var HasExternalInfoPipe = /** @class */ (function () {
        function HasExternalInfoPipe() {
        }
        HasExternalInfoPipe.prototype.transform = function (account) {
            return account ? i12.hasExternalInfo(account) : false;
        };
        return HasExternalInfoPipe;
    }());
    /** @nocollapse */ HasExternalInfoPipe.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: HasExternalInfoPipe, deps: [], target: i0__namespace.ɵɵFactoryTarget.Pipe });
    /** @nocollapse */ HasExternalInfoPipe.ɵpipe = i0__namespace.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: HasExternalInfoPipe, name: "bbHasExternalInfo" });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: HasExternalInfoPipe, decorators: [{
                type: i0.Pipe,
                args: [{
                        name: 'bbHasExternalInfo',
                    }]
            }] });

    var pipes$1 = [HasExternalInfoPipe];
    var HasExternalInfoPipeModule = /** @class */ (function () {
        function HasExternalInfoPipeModule() {
        }
        return HasExternalInfoPipeModule;
    }());
    /** @nocollapse */ HasExternalInfoPipeModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: HasExternalInfoPipeModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    /** @nocollapse */ HasExternalInfoPipeModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: HasExternalInfoPipeModule, declarations: [HasExternalInfoPipe], imports: [i7.CommonModule], exports: [HasExternalInfoPipe] });
    /** @nocollapse */ HasExternalInfoPipeModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: HasExternalInfoPipeModule, imports: [[i7.CommonModule]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: HasExternalInfoPipeModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [i7.CommonModule],
                        declarations: __spreadArray([], __read(pipes$1)),
                        exports: __spreadArray([], __read(pipes$1)),
                    }]
            }] });

    /**
     * Casts a property to a given type
     * {{ property | bbCastAccountProperty:propertyType }}
     */
    var CastAccountPropertyPipe = /** @class */ (function () {
        function CastAccountPropertyPipe() {
        }
        CastAccountPropertyPipe.prototype.transform = function (property, 
        // eslint-disable-next-line @typescript-eslint/no-unused-vars, no-unused-vars
        castType) {
            return property;
        };
        return CastAccountPropertyPipe;
    }());
    /** @nocollapse */ CastAccountPropertyPipe.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CastAccountPropertyPipe, deps: [], target: i0__namespace.ɵɵFactoryTarget.Pipe });
    /** @nocollapse */ CastAccountPropertyPipe.ɵpipe = i0__namespace.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CastAccountPropertyPipe, name: "bbCastAccountProperty" });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CastAccountPropertyPipe, decorators: [{
                type: i0.Pipe,
                args: [{ name: 'bbCastAccountProperty' }]
            }] });

    var pipes = [CastAccountPropertyPipe];
    var CastAccountPropertyPipeModule = /** @class */ (function () {
        function CastAccountPropertyPipeModule() {
        }
        return CastAccountPropertyPipeModule;
    }());
    /** @nocollapse */ CastAccountPropertyPipeModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CastAccountPropertyPipeModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    /** @nocollapse */ CastAccountPropertyPipeModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CastAccountPropertyPipeModule, declarations: [CastAccountPropertyPipe], imports: [i7.CommonModule], exports: [CastAccountPropertyPipe] });
    /** @nocollapse */ CastAccountPropertyPipeModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CastAccountPropertyPipeModule, imports: [[i7.CommonModule]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: CastAccountPropertyPipeModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [i7.CommonModule],
                        declarations: __spreadArray([], __read(pipes)),
                        exports: __spreadArray([], __read(pipes)),
                    }]
            }] });

    var AccountInfoFrequencyComponent = /** @class */ (function () {
        function AccountInfoFrequencyComponent() {
            this.localisationByUnit = {
                day: $localize(templateObject_1 || (templateObject_1 = __makeTemplateObject([":@@accounts-transactions-journeys.account-frequency.day:day(s)"], [":@@accounts-transactions-journeys.account-frequency.day:day(s)"]))),
                week: $localize(templateObject_2 || (templateObject_2 = __makeTemplateObject([":@@accounts-transactions-journeys.account-frequency.week:week(s)"], [":@@accounts-transactions-journeys.account-frequency.week:week(s)"]))),
                month: $localize(templateObject_3 || (templateObject_3 = __makeTemplateObject([":@@accounts-transactions-journeys.account-frequency.month:month(s)"], [":@@accounts-transactions-journeys.account-frequency.month:month(s)"]))),
                year: $localize(templateObject_4 || (templateObject_4 = __makeTemplateObject([":@@accounts-transactions-journeys.account-frequency.year:year(s)"], [":@@accounts-transactions-journeys.account-frequency.year:year(s)"]))),
            };
        }
        Object.defineProperty(AccountInfoFrequencyComponent.prototype, "unitLabel", {
            get: function () {
                switch (this.unit) {
                    case 'D':
                        return this.localisationByUnit.day;
                    case 'W':
                        return this.localisationByUnit.week;
                    case 'M':
                        return this.localisationByUnit.month;
                    case 'Y':
                        return this.localisationByUnit.year;
                    default:
                        return '';
                }
            },
            enumerable: false,
            configurable: true
        });
        return AccountInfoFrequencyComponent;
    }());
    /** @nocollapse */ AccountInfoFrequencyComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoFrequencyComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    /** @nocollapse */ AccountInfoFrequencyComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AccountInfoFrequencyComponent, selector: "bb-account-info-frequency", inputs: { value: "value", unit: "unit" }, ngImport: i0__namespace, template: "<span *ngIf=\"value | bbIsDefined\" data-role=\"account-info-frequency\">\n  {{ value }}\n  {{ unitLabel }}\n</span>\n", directives: [{ type: i7__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "bbIsDefined": i12__namespace.IsDefinedPipe }, changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoFrequencyComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-account-info-frequency',
                        templateUrl: './account-info-frequency.component.html',
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], propDecorators: { value: [{
                    type: i0.Input
                }], unit: [{
                    type: i0.Input
                }] } });
    var templateObject_1, templateObject_2, templateObject_3, templateObject_4;

    var pipeModules$3 = [i12.ValidationModule];
    var components$2 = [AccountInfoFrequencyComponent];
    var AccountInfoFrequencyModule = /** @class */ (function () {
        function AccountInfoFrequencyModule() {
        }
        return AccountInfoFrequencyModule;
    }());
    /** @nocollapse */ AccountInfoFrequencyModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoFrequencyModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    /** @nocollapse */ AccountInfoFrequencyModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoFrequencyModule, declarations: [AccountInfoFrequencyComponent], imports: [i7.CommonModule, i12.ValidationModule], exports: [AccountInfoFrequencyComponent, i12.ValidationModule] });
    /** @nocollapse */ AccountInfoFrequencyModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoFrequencyModule, imports: [__spreadArray([i7.CommonModule], __read(pipeModules$3)), i12.ValidationModule] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoFrequencyModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: __spreadArray([i7.CommonModule], __read(pipeModules$3)),
                        declarations: __spreadArray([], __read(components$2)),
                        exports: __spreadArray(__spreadArray([], __read(components$2)), __read(pipeModules$3)),
                    }]
            }] });

    /**
     * Account info property type
     */
    exports.AccountInfoPropertyType = void 0;
    (function (AccountInfoPropertyType) {
        AccountInfoPropertyType["ACCOUNT_NUMBER"] = "ACCOUNT_NUMBER";
        AccountInfoPropertyType["ACCOUNT_STATE"] = "ACCOUNT_STATE";
        AccountInfoPropertyType["AMOUNT"] = "AMOUNT";
        AccountInfoPropertyType["DATE"] = "DATE";
        AccountInfoPropertyType["DEBIT_CARDS"] = "DEBIT_CARDS";
        AccountInfoPropertyType["FREQUENCY"] = "FREQUENCY";
        AccountInfoPropertyType["NUMBER"] = "NUMBER";
        AccountInfoPropertyType["PERCENT"] = "PERCENT";
        AccountInfoPropertyType["RENEWAL_STATUS"] = "RENEWAL_STATUS";
        AccountInfoPropertyType["TEXT"] = "TEXT";
        AccountInfoPropertyType["UPDATE_TIME"] = "UPDATE_TIME";
    })(exports.AccountInfoPropertyType || (exports.AccountInfoPropertyType = {}));

    var AccountInfoRenewalStatusComponent = /** @class */ (function () {
        function AccountInfoRenewalStatusComponent() {
            this.isRenewalAllowed = false;
        }
        return AccountInfoRenewalStatusComponent;
    }());
    /** @nocollapse */ AccountInfoRenewalStatusComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoRenewalStatusComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    /** @nocollapse */ AccountInfoRenewalStatusComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AccountInfoRenewalStatusComponent, selector: "bb-account-info-renewal-status", inputs: { isRenewalAllowed: "isRenewalAllowed" }, ngImport: i0__namespace, template: "<ng-container *ngIf=\"isRenewalAllowed; else noAutomaticRenewal\">\n  <span\n    data-role=\"account-info-renewal-status\"\n    i18n=\"Automatic Renewal@@product.details.term.deposit.label.autoRenewalIndicator.yes\"\n  >\n    Automatic Renewal\n  </span>\n</ng-container>\n\n<ng-template #noAutomaticRenewal>\n  <span\n    data-role=\"account-info-renewal-status\"\n    i18n=\"No Automatic Renewal@@product.details.term.deposit.label.autoRenewalIndicator.no\"\n  >\n    No Automatic Renewal\n  </span>\n</ng-template>\n", directives: [{ type: i7__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoRenewalStatusComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-account-info-renewal-status',
                        templateUrl: './account-info-renewal-status.component.html',
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], propDecorators: { isRenewalAllowed: [{
                    type: i0.Input
                }] } });

    var AccountInfoPropertyComponent = /** @class */ (function () {
        function AccountInfoPropertyComponent() {
            this.currencyFormat = 'code';
            this.dateFormat = 'longDate';
            this.isMaskable = false;
            this.isMasked = false;
            this.masked = new i0.EventEmitter();
            this.propertyType = exports.AccountInfoPropertyType;
        }
        return AccountInfoPropertyComponent;
    }());
    /** @nocollapse */ AccountInfoPropertyComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoPropertyComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    /** @nocollapse */ AccountInfoPropertyComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: AccountInfoPropertyComponent, selector: "bb-account-info-property", inputs: { account: "account", property: "property", currencyFormat: "currencyFormat", dateFormat: "dateFormat", isMaskable: "isMaskable", isMasked: "isMasked" }, outputs: { masked: "masked" }, ngImport: i0__namespace, template: "<div\n  class=\"bb-account-details-item bb-stack__item bb-stack__item--fill\"\n  [attr.data-role]=\"'account-info-property-' + property?.key\"\n>\n  <div\n    *ngIf=\"property?.label\"\n    class=\"bb-block bb-block--sm bb-stack bb-stack--density-sm\"\n    [attr.data-role]=\"'account-info-property-' + property?.key + '-label'\"\n  >\n    <span class=\"bb-stack__item bb-text-bold\">\n      {{ property?.label }}\n    </span>\n    <bb-icon-ui\n      *ngIf=\"property?.tooltip\"\n      class=\"bb-stack__item bb-text-support\"\n      data-role=\"toggle-tooltip-icon\"\n      tabindex=\"0\"\n      name=\"info\"\n      size=\"sm\"\n      aria-label=\"Toggle tooltip for {{ property?.label }}\"\n      i18n-aria-label=\"Toggle tooltip icon aria label@@product-summary.account-details-property.tooltip-icon.aria-label\"\n      [bbTooltip]=\"property?.tooltip\"\n      triggers=\"focus hover\"\n    ></bb-icon-ui>\n  </div>\n\n  <div [attr.data-role]=\"'account-info-property-' + property?.key + '-value'\">\n    <ng-container [ngSwitch]=\"property?.type\">\n      <ng-container *ngSwitchCase=\"propertyType.ACCOUNT_NUMBER\">\n        <ng-container *ngTemplateOutlet=\"accountNumberValue\"></ng-container>\n      </ng-container>\n      <ng-container *ngSwitchCase=\"propertyType.ACCOUNT_STATE\">\n        <ng-container *ngTemplateOutlet=\"accountStateValue\"></ng-container>\n      </ng-container>\n      <ng-container *ngSwitchCase=\"propertyType.AMOUNT\">\n        <ng-container *ngTemplateOutlet=\"amountOrPercentValue\"></ng-container>\n      </ng-container>\n      <ng-container *ngSwitchCase=\"propertyType.DATE\">\n        <ng-container *ngTemplateOutlet=\"dateValue\"></ng-container>\n      </ng-container>\n      <ng-container *ngSwitchCase=\"propertyType.DEBIT_CARDS\">\n        <ng-container *ngTemplateOutlet=\"debitCardsValue\"></ng-container>\n      </ng-container>\n      <ng-container *ngSwitchCase=\"propertyType.FREQUENCY\">\n        <ng-container *ngTemplateOutlet=\"termValue\"></ng-container>\n      </ng-container>\n      <ng-container *ngSwitchCase=\"propertyType.NUMBER\">\n        <ng-container *ngTemplateOutlet=\"numberOrPercentValue\"></ng-container>\n      </ng-container>\n      <ng-container *ngSwitchCase=\"propertyType.PERCENT\">\n        <ng-container *ngTemplateOutlet=\"percentValue\"></ng-container>\n      </ng-container>\n      <ng-container *ngSwitchCase=\"propertyType.RENEWAL_STATUS\">\n        <ng-container *ngTemplateOutlet=\"renewalStatus\"></ng-container>\n      </ng-container>\n      <ng-container *ngSwitchCase=\"propertyType.TEXT\">\n        <ng-container *ngTemplateOutlet=\"textValue\"></ng-container>\n      </ng-container>\n      <ng-container *ngSwitchCase=\"propertyType.UPDATE_TIME\">\n        <ng-container *ngTemplateOutlet=\"updateTimeValue\"></ng-container>\n      </ng-container>\n    </ng-container>\n  </div>\n</div>\n\n<ng-template #accountNumberValue>\n  <ng-container *ngIf=\"property | bbCastAccountProperty: propertyType.ACCOUNT_NUMBER as valueProperty\">\n    <span aria-live=\"assertive\">\n      <bb-product-number\n        [type]=\"valueProperty.numberType\"\n        [number]=\"account | bbGetValue: valueProperty.key\"\n        [unmaskable]=\"isMaskable && !valueProperty.hideMaskIndicator && !(account | bbHasExternalInfo)\"\n        [masked]=\"isMasked\"\n        (unmask)=\"masked.emit(false)\"\n        (mask)=\"masked.emit(true)\"\n      ></bb-product-number>\n    </span>\n  </ng-container>\n</ng-template>\n\n<ng-template #accountStateValue>\n  <ng-container *ngIf=\"property | bbCastAccountProperty: propertyType.ACCOUNT_STATE as valueProperty\">\n    <bb-arrangement-state-indicator\n      *ngIf=\"account | bbGetValue: valueProperty.key as state\"\n      [state]=\"state\"\n      [showIcon]=\"valueProperty.showIcon || false\"\n      [showActive]=\"true\"\n    ></bb-arrangement-state-indicator>\n  </ng-container>\n</ng-template>\n\n<ng-template #amountOrPercentValue>\n  <ng-container *ngIf=\"property | bbCastAccountProperty: propertyType.AMOUNT as valueProperty\">\n    <ng-container *ngIf=\"account | bbGetValue: valueProperty.key | bbIsPercentValue; else amountValue\">\n      <ng-container *ngTemplateOutlet=\"percentValue\"></ng-container>\n    </ng-container>\n  </ng-container>\n</ng-template>\n\n<ng-template #amountValue>\n  <ng-container *ngIf=\"property | bbCastAccountProperty: propertyType.AMOUNT as valueProperty\">\n    <bb-amount-ui\n      *ngIf=\"account | bbGetValue: valueProperty.key as amount\"\n      [amount]=\"amount\"\n      [currency]=\"valueProperty.currency || account?.currency || ''\"\n      [mapCurrency]=\"(valueProperty.currencyFormat || currencyFormat) === 'symbol'\"\n    ></bb-amount-ui>\n  </ng-container>\n</ng-template>\n\n<ng-template #dateValue>\n  <ng-container *ngIf=\"property | bbCastAccountProperty: propertyType.DATE as valueProperty\">\n    {{ account | bbGetValue: valueProperty.key | bbDate: valueProperty.dateFormat || dateFormat }}\n  </ng-container>\n</ng-template>\n\n<ng-template #debitCardsValue>\n  <ng-container *ngIf=\"property | bbCastAccountProperty: propertyType.DEBIT_CARDS as valueProperty\">\n    <div class=\"bb-stack bb-stack--density-xl bb-stack--wrap\">\n      <bb-payment-card-ui\n        class=\"bb-stack__item bb-block bb-block--md\"\n        *ngFor=\"let debitCardData of account | bbGetValue: valueProperty.key | bbToArray\"\n        [paymentCard]=\"debitCardData | bbToPaymentCardData\"\n        [paymentCardNumberFormat]=\"'cardNumber' | bbProductNumberFormat\"\n      ></bb-payment-card-ui>\n    </div>\n  </ng-container>\n</ng-template>\n\n<ng-template #termValue>\n  <ng-container *ngIf=\"property | bbCastAccountProperty: propertyType.FREQUENCY as valueProperty\">\n    <bb-account-info-frequency\n      [value]=\"account | bbGetValue: valueProperty.key\"\n      [unit]=\"account | bbGetValue: valueProperty.unitKey\"\n    ></bb-account-info-frequency>\n  </ng-container>\n</ng-template>\n\n<ng-template #numberOrPercentValue>\n  <ng-container *ngIf=\"property | bbCastAccountProperty: propertyType.NUMBER as valueProperty\">\n    <ng-container *ngIf=\"account | bbGetValue: valueProperty.key | bbIsPercentValue; else numberValue\">\n      <ng-container *ngTemplateOutlet=\"percentValue\"></ng-container>\n    </ng-container>\n  </ng-container>\n</ng-template>\n\n<ng-template #numberValue>\n  <ng-container *ngIf=\"property | bbCastAccountProperty: propertyType.NUMBER as valueProperty\">\n    <bb-amount-ui\n      *ngIf=\"account | bbGetValue: valueProperty.key | bbToNumber as amount\"\n      [amount]=\"amount\"\n      [decimalPlaces]=\"valueProperty.decimalPlaces\"\n    ></bb-amount-ui>\n  </ng-container>\n</ng-template>\n\n<ng-template #percentValue>\n  <ng-container *ngIf=\"property | bbCastAccountProperty: propertyType.PERCENT as valueProperty\">\n    <bb-amount-ui\n      *ngIf=\"account | bbGetValue: valueProperty.key | bbToNumber as amount\"\n      [amount]=\"amount\"\n      [showPercent]=\"true\"\n    ></bb-amount-ui>\n  </ng-container>\n</ng-template>\n\n<ng-template #renewalStatus>\n  <ng-container *ngIf=\"property | bbCastAccountProperty: propertyType.RENEWAL_STATUS as valueProperty\">\n    <bb-account-info-renewal-status\n      [isRenewalAllowed]=\"(account | bbGetValue: valueProperty.key) || false\"\n    ></bb-account-info-renewal-status>\n  </ng-container>\n</ng-template>\n\n<ng-template #textValue>\n  <ng-container *ngIf=\"property | bbCastAccountProperty: propertyType.TEXT as valueProperty\">\n    {{ account | bbGetValue: valueProperty.key }}\n  </ng-container>\n</ng-template>\n\n<ng-template #updateTimeValue>\n  <ng-container *ngIf=\"property | bbCastAccountProperty: propertyType.UPDATE_TIME as valueProperty\">\n    {{ (account | bbGetValue: valueProperty.key) || '' | bbDateUpdatedAgo: valueProperty.dateFormat || dateFormat }}\n  </ng-container>\n</ng-template>\n", components: [{ type: i1__namespace.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i2__namespace.ProductNumberComponent, selector: "bb-product-number", inputs: ["number", "product", "format", "type", "unmaskable", "masked"], outputs: ["mask", "unmask"] }, { type: i2__namespace.ArrangementStateIndicatorComponent, selector: "bb-arrangement-state-indicator", inputs: ["state", "showActive", "showIcon", "hideStateTextForMobile"] }, { type: i3__namespace.AmountComponent, selector: "bb-amount-ui", inputs: ["currency", "showPlusSign", "mapCurrency", "showPercent", "abbreviate", "decimalPlaces", "trailingZeroes", "amount"] }, { type: i4__namespace.PaymentCardComponent, selector: "bb-payment-card-ui", inputs: ["paymentCard", "paymentCardNumberFormat"] }, { type: AccountInfoFrequencyComponent, selector: "bb-account-info-frequency", inputs: ["value", "unit"] }, { type: AccountInfoRenewalStatusComponent, selector: "bb-account-info-renewal-status", inputs: ["isRenewalAllowed"] }], directives: [{ type: i7__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i8__namespace.TooltipDirective, selector: "[bbTooltip]", inputs: ["bbTooltip", "triggers"] }, { type: i7__namespace.NgSwitch, selector: "[ngSwitch]", inputs: ["ngSwitch"] }, { type: i7__namespace.NgSwitchCase, selector: "[ngSwitchCase]", inputs: ["ngSwitchCase"] }, { type: i7__namespace.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }, { type: i7__namespace.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }], pipes: { "bbCastAccountProperty": CastAccountPropertyPipe, "bbGetValue": i10__namespace.GetValuePipe, "bbHasExternalInfo": HasExternalInfoPipe, "bbIsPercentValue": i12__namespace.IsPercentValue, "bbDate": i13__namespace.BbDatePipe, "bbToPaymentCardData": ToPaymentCardDataPipe, "bbProductNumberFormat": i12__namespace.ProductNumberFormatPipe, "bbToArray": i10__namespace.ToArrayPipe, "bbToNumber": i10__namespace.ToNumberPipe, "bbDateUpdatedAgo": i12__namespace.DateUpdatedAgoPipe }, changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoPropertyComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'bb-account-info-property',
                        templateUrl: './account-info-property.component.html',
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], propDecorators: { account: [{
                    type: i0.Input
                }], property: [{
                    type: i0.Input
                }], currencyFormat: [{
                    type: i0.Input
                }], dateFormat: [{
                    type: i0.Input
                }], isMaskable: [{
                    type: i0.Input
                }], isMasked: [{
                    type: i0.Input
                }], masked: [{
                    type: i0.Output
                }] } });

    var pipeModules$2 = [i12.ValidationModule];
    var sharedModules$1 = [i12.ProductSummaryCommon];
    var components$1 = [AccountInfoRenewalStatusComponent];
    var AccountInfoRenewalStatusModule = /** @class */ (function () {
        function AccountInfoRenewalStatusModule() {
        }
        return AccountInfoRenewalStatusModule;
    }());
    /** @nocollapse */ AccountInfoRenewalStatusModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoRenewalStatusModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    /** @nocollapse */ AccountInfoRenewalStatusModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoRenewalStatusModule, declarations: [AccountInfoRenewalStatusComponent], imports: [i7.CommonModule, i12.ProductSummaryCommon, i12.ValidationModule], exports: [i12.ProductSummaryCommon, AccountInfoRenewalStatusComponent, i12.ValidationModule] });
    /** @nocollapse */ AccountInfoRenewalStatusModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoRenewalStatusModule, imports: [__spreadArray(__spreadArray([i7.CommonModule], __read(sharedModules$1)), __read(pipeModules$2)), i12.ProductSummaryCommon, i12.ValidationModule] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoRenewalStatusModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: __spreadArray(__spreadArray([i7.CommonModule], __read(sharedModules$1)), __read(pipeModules$2)),
                        declarations: __spreadArray([], __read(components$1)),
                        exports: __spreadArray(__spreadArray(__spreadArray([], __read(sharedModules$1)), __read(components$1)), __read(pipeModules$2)),
                    }]
            }] });

    var uiModules$1 = [
        AccountInfoFrequencyModule,
        i3.AmountModule,
        i2.ProductNumberModule,
        i2.ArrangementStateIndicatorModule,
        i8.TooltipModule,
        i13.BbDatePipeModule,
        i12.DatePipeModule,
        i4.PaymentCardModule,
        AccountInfoRenewalStatusModule,
    ];
    var pipeModules$1 = [
        CastAccountPropertyPipeModule,
        i10.GetDecimalsCountModule,
        i10.GetValuePipeModule,
        HasExternalInfoPipeModule,
        i10.ToArrayPipeModule,
        ToPaymentCardDataPipeModule,
        i10.ToNumberPipeModule,
    ];
    var sharedModules = [i12.ProductSummaryCommon, i12.ValidationModule, i12.FormatPipeModule];
    var components = [AccountInfoPropertyComponent];
    var AccountInfoPropertyModule = /** @class */ (function () {
        function AccountInfoPropertyModule() {
        }
        return AccountInfoPropertyModule;
    }());
    /** @nocollapse */ AccountInfoPropertyModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoPropertyModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    /** @nocollapse */ AccountInfoPropertyModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoPropertyModule, declarations: [AccountInfoPropertyComponent], imports: [i7.CommonModule, AccountInfoFrequencyModule,
            i3.AmountModule,
            i2.ProductNumberModule,
            i2.ArrangementStateIndicatorModule,
            i8.TooltipModule,
            i13.BbDatePipeModule,
            i12.DatePipeModule,
            i4.PaymentCardModule,
            AccountInfoRenewalStatusModule, i12.ProductSummaryCommon, i12.ValidationModule, i12.FormatPipeModule, CastAccountPropertyPipeModule,
            i10.GetDecimalsCountModule,
            i10.GetValuePipeModule,
            HasExternalInfoPipeModule,
            i10.ToArrayPipeModule,
            ToPaymentCardDataPipeModule,
            i10.ToNumberPipeModule], exports: [AccountInfoFrequencyModule,
            i3.AmountModule,
            i2.ProductNumberModule,
            i2.ArrangementStateIndicatorModule,
            i8.TooltipModule,
            i13.BbDatePipeModule,
            i12.DatePipeModule,
            i4.PaymentCardModule,
            AccountInfoRenewalStatusModule, i12.ProductSummaryCommon, i12.ValidationModule, i12.FormatPipeModule, AccountInfoPropertyComponent, CastAccountPropertyPipeModule,
            i10.GetDecimalsCountModule,
            i10.GetValuePipeModule,
            HasExternalInfoPipeModule,
            i10.ToArrayPipeModule,
            ToPaymentCardDataPipeModule,
            i10.ToNumberPipeModule] });
    /** @nocollapse */ AccountInfoPropertyModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoPropertyModule, imports: [__spreadArray(__spreadArray(__spreadArray([i7.CommonModule], __read(uiModules$1)), __read(sharedModules)), __read(pipeModules$1)), AccountInfoFrequencyModule,
            i3.AmountModule,
            i2.ProductNumberModule,
            i2.ArrangementStateIndicatorModule,
            i8.TooltipModule,
            i13.BbDatePipeModule,
            i12.DatePipeModule,
            i4.PaymentCardModule,
            AccountInfoRenewalStatusModule, i12.ProductSummaryCommon, i12.ValidationModule, i12.FormatPipeModule, CastAccountPropertyPipeModule,
            i10.GetDecimalsCountModule,
            i10.GetValuePipeModule,
            HasExternalInfoPipeModule,
            i10.ToArrayPipeModule,
            ToPaymentCardDataPipeModule,
            i10.ToNumberPipeModule] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: AccountInfoPropertyModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: __spreadArray(__spreadArray(__spreadArray([i7.CommonModule], __read(uiModules$1)), __read(sharedModules)), __read(pipeModules$1)),
                        declarations: __spreadArray([], __read(components)),
                        exports: __spreadArray(__spreadArray(__spreadArray(__spreadArray([], __read(uiModules$1)), __read(sharedModules)), __read(components)), __read(pipeModules$1)),
                    }]
            }] });

    var pipeModules = [ToPaymentCardDataPipeModule, HasExternalInfoPipeModule, CastAccountPropertyPipeModule];
    var uiModules = [AccountInfoPropertyModule, AccountInfoFrequencyModule];
    var SharedUiAccountModule = /** @class */ (function () {
        function SharedUiAccountModule() {
        }
        return SharedUiAccountModule;
    }());
    /** @nocollapse */ SharedUiAccountModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: SharedUiAccountModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    /** @nocollapse */ SharedUiAccountModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: SharedUiAccountModule, imports: [i7.CommonModule, ToPaymentCardDataPipeModule, HasExternalInfoPipeModule, CastAccountPropertyPipeModule, AccountInfoPropertyModule, AccountInfoFrequencyModule], exports: [ToPaymentCardDataPipeModule, HasExternalInfoPipeModule, CastAccountPropertyPipeModule, AccountInfoPropertyModule, AccountInfoFrequencyModule] });
    /** @nocollapse */ SharedUiAccountModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: SharedUiAccountModule, imports: [__spreadArray(__spreadArray([i7.CommonModule], __read(pipeModules)), __read(uiModules)), ToPaymentCardDataPipeModule, HasExternalInfoPipeModule, CastAccountPropertyPipeModule, AccountInfoPropertyModule, AccountInfoFrequencyModule] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0__namespace, type: SharedUiAccountModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: __spreadArray(__spreadArray([i7.CommonModule], __read(pipeModules)), __read(uiModules)),
                        exports: __spreadArray(__spreadArray([], __read(pipeModules)), __read(uiModules)),
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.AccountInfoFrequencyComponent = AccountInfoFrequencyComponent;
    exports.AccountInfoFrequencyModule = AccountInfoFrequencyModule;
    exports.AccountInfoPropertyComponent = AccountInfoPropertyComponent;
    exports.AccountInfoPropertyModule = AccountInfoPropertyModule;
    exports.AccountInfoRenewalStatusComponent = AccountInfoRenewalStatusComponent;
    exports.AccountInfoRenewalStatusModule = AccountInfoRenewalStatusModule;
    exports.CastAccountPropertyPipe = CastAccountPropertyPipe;
    exports.CastAccountPropertyPipeModule = CastAccountPropertyPipeModule;
    exports.HasExternalInfoPipe = HasExternalInfoPipe;
    exports.HasExternalInfoPipeModule = HasExternalInfoPipeModule;
    exports.SharedUiAccountModule = SharedUiAccountModule;
    exports.ToPaymentCardDataPipe = ToPaymentCardDataPipe;
    exports.ToPaymentCardDataPipeModule = ToPaymentCardDataPipeModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
