// @backbase/ui-ang/esm2015/_example/empty-state/showcase/empty-state-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { EmptyStateShowcaseComponent } from './empty-state-showcase.component';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { ButtonModule } from '@backbase/ui-ang/button';
import * as i0 from "@angular/core";
export class EmptyStateShowcaseModule {
}
EmptyStateShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EmptyStateShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
EmptyStateShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EmptyStateShowcaseModule, declarations: [EmptyStateShowcaseComponent], imports: [CommonModule, EmptyStateModule, ButtonModule] });
EmptyStateShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EmptyStateShowcaseModule, imports: [[CommonModule, EmptyStateModule, ButtonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EmptyStateShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, EmptyStateModule, ButtonModule],
                    declarations: [EmptyStateShowcaseComponent],
                    entryComponents: [EmptyStateShowcaseComponent],
                }]
        }] });
