// @backbase/manage-pockets-journey-ang/esm2015/src/manage-pockets-common/manage-pockets-common.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { AmountModule } from '@backbase/ui-ang/amount';
import { ButtonModule } from '@backbase/ui-ang/button';
import { DropdownMenuModule } from '@backbase/ui-ang/dropdown-menu';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { IconModule } from '@backbase/ui-ang/icon';
import { ProgressbarModule } from '@backbase/ui-ang/progressbar';
import { ManagePocketsCardComponent } from './components/manage-pockets-card/manage-pockets-card.component';
import { AccessibilityKeyboardDirective } from './directives/accessibility-keyboard.directive';
import { DynamicIdDirective } from './directives/dynamic-id.directive';
import { RegexReplacePipe } from './pipes/regex-replace.pipe';
import { DomAttributesService } from './services/dom-attributes.service';
import { ManagePocketsService } from './services/manage-pockets.service';
import * as i0 from "@angular/core";
const modules = [AmountModule, ButtonModule, DropdownMenuModule, EmptyStateModule, IconModule, ProgressbarModule];
const components = [AccessibilityKeyboardDirective, DynamicIdDirective, ManagePocketsCardComponent, RegexReplacePipe];
export class ManagePocketsCommonModule {
}
ManagePocketsCommonModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsCommonModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ManagePocketsCommonModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsCommonModule, declarations: [AccessibilityKeyboardDirective, DynamicIdDirective, ManagePocketsCardComponent, RegexReplacePipe], imports: [CommonModule, AmountModule, ButtonModule, DropdownMenuModule, EmptyStateModule, IconModule, ProgressbarModule], exports: [AccessibilityKeyboardDirective, DynamicIdDirective, ManagePocketsCardComponent, RegexReplacePipe, AmountModule, ButtonModule, DropdownMenuModule, EmptyStateModule, IconModule, ProgressbarModule] });
ManagePocketsCommonModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsCommonModule, providers: [ManagePocketsService, DomAttributesService], imports: [[CommonModule, ...modules], AmountModule, ButtonModule, DropdownMenuModule, EmptyStateModule, IconModule, ProgressbarModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ManagePocketsCommonModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...modules],
                    providers: [ManagePocketsService, DomAttributesService],
                    declarations: [...components],
                    exports: [...components, ...modules],
                }]
        }] });
