// @backbase/manage-payees-journey-ang/esm2015/src/components/payee-list/billpay-payee-summary-list/billpay-payee-summary-list.component.js
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../billpay-payee-summary-item/billpay-payee-summary-item.component";
import * as i2 from "@angular/common";
export class BillpayPayeeSummaryListComponent {
    constructor() {
        /**
         * Array of payee summary details.
         */
        this.payeeSummaries = [];
    }
    ngForTrackByIdFn(payee) {
        return payee.id;
    }
}
BillpayPayeeSummaryListComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayPayeeSummaryListComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
BillpayPayeeSummaryListComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: BillpayPayeeSummaryListComponent, selector: "bb-billpay-payee-summary-list", inputs: { payeeSummaries: "payeeSummaries" }, ngImport: i0, template: "<ul class=\"bb-list\" data-role=\"list-group\">\n  <li *ngFor=\"let payeeSummary of payeeSummaries; trackBy: ngForTrackByIdFn\" class=\"bb-list__item\">\n    <bb-billpay-payee-summary-item [payeeSummary]=\"payeeSummary\"></bb-billpay-payee-summary-item>\n  </li>\n</ul>\n", components: [{ type: i1.BillpayPayeeSummaryItemComponent, selector: "bb-billpay-payee-summary-item" }], directives: [{ type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: BillpayPayeeSummaryListComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-billpay-payee-summary-list',
                    templateUrl: './billpay-payee-summary-list.component.html',
                }]
        }], propDecorators: { payeeSummaries: [{
                type: Input
            }] } });
