// @backbase/transactions-common-ang/esm2015/lib/components/document-viewer-ui/document-viewer-backdrop/document-viewer-backdrop.component.js
import { Component, Input, ChangeDetectionStrategy } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class DocumentViewerBackdropComponent {
    get containerStyles() {
        var _a, _b;
        return ((_b = (_a = this.options) === null || _a === void 0 ? void 0 : _a.container) === null || _b === void 0 ? void 0 : _b.styles) || null;
    }
    get hasCustomStyles() {
        var _a, _b;
        return Boolean((_b = (_a = this.options) === null || _a === void 0 ? void 0 : _a.container) === null || _b === void 0 ? void 0 : _b.styles);
    }
}
DocumentViewerBackdropComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DocumentViewerBackdropComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
DocumentViewerBackdropComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: DocumentViewerBackdropComponent, selector: "bb-document-viewer-backdrop", inputs: { options: "options" }, ngImport: i0, template: "<div\n  class=\"bb-document-viewer-backdrop modal-backdrop show\"\n  aria-hidden=\"true\"\n  [class.bb-document-viewer-backdrop-default-off]=\"hasCustomStyles\"\n  [ngStyle]=\"containerStyles\"\n></div>\n", directives: [{ type: i1.NgStyle, selector: "[ngStyle]", inputs: ["ngStyle"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DocumentViewerBackdropComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-document-viewer-backdrop',
                    templateUrl: './document-viewer-backdrop.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { options: [{
                type: Input
            }] } });
