// @backbase/ui-ang/esm2015/_example/dropdown-menu/examples/dropdown-menu.example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/dropdown-menu";
import * as i2 from "@backbase/ui-ang/icon";
export class DropdownMenuExampleComponent {
    constructor() {
        this.dropdownMenuOptions = [
            { label: '30 minutes', duration: 1800 },
            { label: '1 hour', duration: 3600 },
            { label: '1 day', duration: 86400 },
        ];
        this.dropdownMenuOptionsLabel = this.dropdownMenuOptions.map((item) => item.label);
    }
    onButtonDropdownSelect(label) {
        const item = this.dropdownMenuOptions.find((option) => option.label === label);
        if (item) {
            this.snooze(item.duration);
        }
    }
    snooze(duration) {
        alert(`Snooze for ${duration} seconds`);
    }
}
DropdownMenuExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownMenuExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
DropdownMenuExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: DropdownMenuExampleComponent, selector: "bb-dropdown-menu-example-ui", ngImport: i0, template: "<bb-dropdown-menu-ui\n  container=\"body\"\n  position=\"bottom\"\n  icon=\"caret-down\"\n  btnColor=\"primary\"\n  iconSize=\"lg\"\n  [options]=\"dropdownMenuOptionsLabel\"\n  (select)=\"onButtonDropdownSelect($event)\"\n>\n  <ng-template bbDropdownLabel>\n    <bb-icon-ui name=\"time\" size=\"md\"></bb-icon-ui>\n    &nbsp;\n    <span>Snooze notification</span>\n  </ng-template>\n</bb-dropdown-menu-ui>\n", components: [{ type: i1.DropdownMenuComponent, selector: "bb-dropdown-menu-ui", inputs: ["label", "disabled", "options", "icon", "iconSize", "iconColor", "btnColor", "btnCircle", "optionLabelKey", "fullWidth", "autoClose", "container", "position", "ariaLabel", "dropDownMenuRole", "dropDownMenuId", "buttonSize", "dropDownButtonLabel", "visibleElementDependingOnViewport", "role"], outputs: ["select"] }, { type: i2.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], directives: [{ type: i1.DropdownLabelDirective, selector: "ng-template[bbDropdownLabel]" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DropdownMenuExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-dropdown-menu-example-ui',
                    templateUrl: './dropdown-menu.example.component.html',
                }]
        }] });
