// @backbase/loans-collection-ui-ang/esm2015/lib/loan-summary/retail-loan-summary-container/retail-loan-summary-container.component.js
import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
import { drawAmountMenuItem, payDownMenuItem, } from '../loan-summary-container.model';
import { LoanConfigBaseComponent } from '../../base/loan-config-base.component';
import { getLoanAlias, isLoanOverdue } from '../../utils/loan.util';
import { MediaQueryService } from '@backbase/ui-ang/media-query-lib';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/media-query-lib";
import * as i2 from "../retail-loan-summary-overdue-banner/retail-loan-summary-overdue-banner.component";
import * as i3 from "../../loan-list-item/loan-item-payment-info/retail-loan-item-payment-info/retail-loan-item-payment-info.component";
import * as i4 from "@backbase/ui-ang/icon";
import * as i5 from "../../loan-list-item/loan-item-name/loan-item-name.component";
import * as i6 from "../../loan-list-item/loan-item-type/loan-item-type.component";
import * as i7 from "../retail-loan-summary-progress-bar/retail-loan-summary-progress-bar.component";
import * as i8 from "../loan-summary-amount-item/loan-summary-amount-item.component";
import * as i9 from "../../loan-list-item/loan-item-menu/loan-item-menu.component";
import * as i10 from "../loan-summary-formula/loan-summary-formula.component";
import * as i11 from "@angular/common";
import * as i12 from "@backbase/ui-ang/button";
export class RetailLoanSummaryContainerComponent extends LoanConfigBaseComponent {
    constructor(mediaQueryService) {
        super();
        this.mediaQueryService = mediaQueryService;
        /**
         * Indicator to show/hide `payDown` button
         *
         * @default true
         */
        this.showPayDownButton = true;
        /**
         * Indicator to show/hide `payDown` button
         *
         * @default true
         */
        this.showDrawAmountButton = true;
        /**
         * Indicator to show/hide `summary formula`
         *
         * @default true
         */
        this.showFormula = true;
        /**
         * Indicator to show `interest rate` over or below the progress bar
         *
         * @default false
         */
        this.showInterestRateOnTop = false;
        /**
         * Draw amount event emitter
         */
        this.drawAmount = new EventEmitter();
        /**
         * Pay down event emitter
         */
        this.payDown = new EventEmitter();
        /**
         * Emits on accepting new alias for save
         */
        this.updateAlias = new EventEmitter();
        /**
         * Emits in canceling current alias update
         */
        this.cancelAliasUpdate = new EventEmitter();
        /**
         * Flag represents the screen resolution wider that `md` breakpoint
         */
        this.isWiderThanMobile$ = this.mediaQueryService.isMediaBreakpointMatches('min-width', 'md');
    }
    /**
     * @internal
     */
    ngOnChanges() {
        this.showOverdueBanner = isLoanOverdue(this.loan);
        this.loanAlias = getLoanAlias(this.loan);
        this.menuItems = this.getMenuItems();
        this.loanIconName = this.dataConfig.icons[this.loan.viewType];
    }
    /**
     * Select an item for the context menu
     *
     * @param itemKey
     */
    onSelectMenuItem(itemKey) {
        switch (itemKey) {
            case 'payDown':
                this.payDown.emit();
                break;
            case 'drawAmount':
                this.drawAmount.emit();
        }
    }
    getMenuItems() {
        const menuItems = [];
        if (this.showPayDownButton) {
            menuItems.push(payDownMenuItem);
        }
        if (this.showDrawAmountButton) {
            menuItems.push(drawAmountMenuItem);
        }
        return menuItems;
    }
}
/** @nocollapse */ RetailLoanSummaryContainerComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RetailLoanSummaryContainerComponent, deps: [{ token: i1.MediaQueryService }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ RetailLoanSummaryContainerComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: RetailLoanSummaryContainerComponent, selector: "bb-retail-loan-summary-container", inputs: { showPayDownButton: "showPayDownButton", showDrawAmountButton: "showDrawAmountButton", showFormula: "showFormula", showInterestRateOnTop: "showInterestRateOnTop", owedAmountFocus: "owedAmountFocus" }, outputs: { drawAmount: "drawAmount", payDown: "payDown", updateAlias: "updateAlias", cancelAliasUpdate: "cancelAliasUpdate" }, usesInheritance: true, usesOnChanges: true, ngImport: i0, template: "<bb-retail-loan-summary-overdue-banner\n  *ngIf=\"showOverdueBanner\"\n  [overdueAmount]=\"loan.inArrearsAmount\"\n  [overdueDate]=\"loan.inArrearsDate\"\n  [currencyCode]=\"loan.currencyCode\"\n  [overduePaymentsCount]=\"loan.overduePaymentsCount\"\n></bb-retail-loan-summary-overdue-banner>\n<div class=\"bb-block\">\n  <bb-retail-loan-item-payment-info\n    [loan]=\"loan\"\n    [dataConfig]=\"dataConfig.paymentInfo\"\n    class=\"bb-subtitle\"\n  ></bb-retail-loan-item-payment-info>\n</div>\n<div class=\"bb-stack bb-block--lg\">\n  <bb-icon-ui\n    [name]=\"loanIconName\"\n    class=\"bb-stack__item bb-stack__item--align-top bb-d-lg-inline-flex bb-d-none\"\n    data-role=\"loans-summary__loan-icon\"\n    backgroundType=\"square\"\n    color=\"secondary\"\n    size=\"lg\"\n  ></bb-icon-ui>\n  <div class=\"bb-stack__item bb-stack__item--fill\">\n    <bb-loan-item-name\n      [headingLevel]=\"2\"\n      [loanName]=\"loanAlias\"\n      class=\"bb-stack__item bb-stack__item--spacing-sm\"\n    ></bb-loan-item-name>\n    <div>\n      <bb-loan-item-type [loanType]=\"loan.loanTypeAlias\" [useSubtitleClass]=\"true\"></bb-loan-item-type>\n    </div>\n  </div>\n</div>\n\n<div class=\"bb-block bb-block--lg\">\n  <bb-retail-loan-summary-progress-bar\n    [loan]=\"loan\"\n    [dataConfig]=\"dataConfig.progressBar\"\n    [owedAmountFocus]=\"owedAmountFocus\"\n  >\n    <bb-loan-summary-amount-item\n      *ngIf=\"showInterestRateOnTop\"\n      [amount]=\"loan.interestRate\"\n      [showPercent]=\"true\"\n      title=\"Interest rate\"\n      i18n-title=\"Loan summary interest rate title@@loans.loan-summary.interest-rate.title\"\n      titleClasses=\"bb-text-support\"\n    >\n    </bb-loan-summary-amount-item>\n  </bb-retail-loan-summary-progress-bar>\n</div>\n\n<div class=\"bb-block bb-stack justify-content-between\">\n  <div class=\"bb-stack__item bb-stack__item--no-shrink bb-stack__item--align-top\">\n    <ng-container *ngIf=\"isWiderThanMobile$ | async; else contextMenuRef\">\n      <button\n        *ngIf=\"showPayDownButton\"\n        (click)=\"payDown.emit()\"\n        class=\"bb-stack__item bb-inline-stack bb-inline-stack--density-sm\"\n        color=\"primary\"\n        bbButton\n        data-role=\"loans-summary__pay-down-btn\"\n      >\n        <bb-icon-ui name=\"call-received\" class=\"bb-inline-stack__item\"></bb-icon-ui>\n        <span\n          class=\"bb-inline-stack__item\"\n          i18n=\"Loan summary pay down button text@@loans.loan-summary.button.pay-down.text\"\n        >\n          Pay Down\n        </span>\n      </button>\n      <button\n        *ngIf=\"showDrawAmountButton\"\n        (click)=\"drawAmount.emit()\"\n        class=\"bb-stack__item bb-inline-stack bb-inline-stack--density-sm\"\n        color=\"secondary\"\n        bbButton\n        data-role=\"loans-summary__draw-amount-btn\"\n      >\n        <bb-icon-ui name=\"savings-accounts\" class=\"bb-inline-stack__item\"></bb-icon-ui>\n        <span\n          class=\"bb-inline-stack__item\"\n          i18n=\"Loan summary draw amount button text@@loans.loan-summary.button.draw-amount.text\"\n        >\n          Draw Amount\n        </span>\n      </button>\n    </ng-container>\n\n    <ng-template #contextMenuRef>\n      <bb-loan-item-menu\n        data-role=\"loans-summary__menu-items\"\n        *ngIf=\"menuItems.length > 0\"\n        [items]=\"menuItems\"\n        [position]=\"'bottom-left'\"\n        (itemSelect)=\"onSelectMenuItem($event)\"\n      ></bb-loan-item-menu>\n    </ng-template>\n  </div>\n\n  <div *ngIf=\"!showInterestRateOnTop\" class=\"bb-inline-stack bb-text-align-center\">\n    <bb-loan-summary-amount-item\n      [amount]=\"loan.interestRate\"\n      [showPercent]=\"true\"\n      title=\"Interest rate\"\n      i18n-title=\"Loan summary interest rate title@@loans.loan-summary.interest-rate.title\"\n      titleClasses=\"bb-text-support\"\n    >\n    </bb-loan-summary-amount-item>\n  </div>\n</div>\n\n<bb-loan-summary-formula *ngIf=\"showFormula\" [loan]=\"loan\" [dataConfig]=\"dataConfig.formula\"></bb-loan-summary-formula>\n", components: [{ type: i2.RetailLoanSummaryOverdueBannerComponent, selector: "bb-retail-loan-summary-overdue-banner", inputs: ["overdueAmount", "overdueDate", "overduePaymentsCount", "currencyCode"] }, { type: i3.RetailLoanItemPaymentInfoComponent, selector: "bb-retail-loan-item-payment-info" }, { type: i4.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i5.LoanItemNameComponent, selector: "bb-loan-item-name", inputs: ["loanName", "headingLevel"] }, { type: i6.LoanItemTypeComponent, selector: "bb-loan-item-type", inputs: ["loanType", "useSubtitleClass"] }, { type: i7.RetailLoanSummaryProgressBarComponent, selector: "bb-retail-loan-summary-progress-bar", inputs: ["owedAmountFocus"] }, { type: i8.LoanSummaryAmountItemComponent, selector: "bb-loan-summary-amount-item", inputs: ["title", "amount", "currency", "isCurrencySymbolUsed", "showPercent", "titleClasses", "amountClasses", "totalAmount", "showLabel"] }, { type: i9.LoanItemMenuComponent, selector: "bb-loan-item-menu", inputs: ["items", "dropDownButtonLabel", "position"], outputs: ["itemSelect"] }, { type: i10.LoanSummaryFormulaComponent, selector: "bb-loan-summary-formula" }], directives: [{ type: i11.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i12.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }], pipes: { "async": i11.AsyncPipe }, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RetailLoanSummaryContainerComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-retail-loan-summary-container',
                    templateUrl: './retail-loan-summary-container.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.MediaQueryService }]; }, propDecorators: { showPayDownButton: [{
                type: Input
            }], showDrawAmountButton: [{
                type: Input
            }], showFormula: [{
                type: Input
            }], showInterestRateOnTop: [{
                type: Input
            }], owedAmountFocus: [{
                type: Input
            }], drawAmount: [{
                type: Output
            }], payDown: [{
                type: Output
            }], updateAlias: [{
                type: Output
            }], cancelAliasUpdate: [{
                type: Output
            }] } });
