// @backbase/stop-checks-http-ang/fesm2015/backbase-stop-checks-http-ang.js
import * as i0 from '@angular/core';
import { InjectionToken, NgModule, Optional, SkipSelf, Injectable, Inject } from '@angular/core';
import * as i1 from '@angular/common/http';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import * as i2 from '@backbase/foundation-ang/data-http';
import { createMocks } from '@backbase/foundation-ang/data-http';

const SchemeName = {
    IBAN: 'IBAN',
    BBAN: 'BBAN',
    ID: 'ID'
};

const StopCheckStatus = {
    PENDING: 'PENDING',
    ACCEPTED: 'ACCEPTED',
    REJECTED: 'REJECTED',
    EXPIRED: 'EXPIRED'
};

const STOP_CHECKS_BASE_PATH = new InjectionToken('STOP_CHECKS_BASE_PATH');
const COLLECTION_FORMATS = {
    'csv': ',',
    'tsv': '   ',
    'ssv': ' ',
    'pipes': '|'
};

class StopChecksConfiguration {
    constructor(configurationParameters = {}) {
        this.apiKeys = configurationParameters.apiKeys;
        this.username = configurationParameters.username;
        this.password = configurationParameters.password;
        this.accessToken = configurationParameters.accessToken;
        this.basePath = configurationParameters.basePath;
        this.withCredentials = configurationParameters.withCredentials;
        this.encoder = configurationParameters.encoder;
        if (configurationParameters.credentials) {
            this.credentials = configurationParameters.credentials;
        }
        else {
            this.credentials = {};
        }
    }
    selectHeaderContentType(contentTypes) {
        if (contentTypes.length === 0) {
            return undefined;
        }
        const type = contentTypes.find((x) => this.isJsonMime(x));
        if (type === undefined) {
            return contentTypes[0];
        }
        return type;
    }
    selectHeaderAccept(accepts) {
        if (accepts.length === 0) {
            return undefined;
        }
        const type = accepts.find((x) => this.isJsonMime(x));
        if (type === undefined) {
            return accepts[0];
        }
        return type;
    }
    isJsonMime(mime) {
        const jsonMime = new RegExp('^(application\/json|[^;/ \t]+\/[^;/ \t]+[+]json)[ \t]*(;.*)?$', 'i');
        return mime !== null && (jsonMime.test(mime) || mime.toLowerCase() === 'application/json-patch+json');
    }
    lookupCredential(key) {
        const value = this.credentials[key];
        return typeof value === 'function'
            ? value()
            : value;
    }
}
const CONFIG_TOKEN = new InjectionToken('api-data-module StopChecksConfiguration');

class StopChecksApiModule {
    constructor(parentModule, http, dataModulesManager, config) {
        if (parentModule) {
            throw new Error('StopChecksApiModule is already loaded. Import in your base AppModule only.');
        }
        if (!http) {
            throw new Error('You need to import the HttpClientModule in your AppModule! \n' +
                'See also https://github.com/angular/angular/issues/20575');
        }
        if (dataModulesManager) {
            dataModulesManager.setModuleConfig(CONFIG_TOKEN, {
                apiRoot: '',
                servicePath: config.basePath || '',
                headers: {},
            });
        }
    }
    static forRoot(configurationFactory) {
        return {
            ngModule: StopChecksApiModule,
            providers: [{ provide: StopChecksConfiguration, useFactory: configurationFactory }]
        };
    }
}
StopChecksApiModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: StopChecksApiModule, deps: [{ token: StopChecksApiModule, optional: true, skipSelf: true }, { token: i1.HttpClient, optional: true }, { token: i2.DataModulesManager, optional: true }, { token: StopChecksConfiguration }], target: i0.ɵɵFactoryTarget.NgModule });
StopChecksApiModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: StopChecksApiModule });
StopChecksApiModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: StopChecksApiModule, providers: [], imports: [[]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: StopChecksApiModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [],
                    declarations: [],
                    exports: [],
                    providers: []
                }]
        }], ctorParameters: function () { return [{ type: StopChecksApiModule, decorators: [{
                    type: Optional
                }, {
                    type: SkipSelf
                }] }, { type: i1.HttpClient, decorators: [{
                    type: Optional
                }] }, { type: i2.DataModulesManager, decorators: [{
                    type: Optional
                }] }, { type: StopChecksConfiguration }]; } });

class CustomHttpParameterCodec {
    encodeKey(k) {
        return encodeURIComponent(k);
    }
    encodeValue(v) {
        return encodeURIComponent(v);
    }
    decodeKey(k) {
        return decodeURIComponent(k);
    }
    decodeValue(v) {
        return decodeURIComponent(v);
    }
}

class StopChecksHttpService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new StopChecksConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    getStopChecks(requestParameters, observe = 'body', reportProgress = false, options) {
        const _from = requestParameters["from"];
        const _cursor = requestParameters["cursor"];
        const _size = requestParameters["size"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_cursor !== undefined && _cursor !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _cursor, 'cursor');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/stop-checks`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getStopChecksUrl(requestParameters) {
        const _from = requestParameters["from"];
        const _cursor = requestParameters["cursor"];
        const _size = requestParameters["size"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_cursor !== undefined && _cursor !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _cursor, 'cursor');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v1/stop-checks${queryString ? `?${queryString}` : ''}`;
    }
    postStopChecks(requestParameters, observe = 'body', reportProgress = false, options) {
        const _stopCheckPostRequest = requestParameters["stopCheckPostRequest"];
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v1/stop-checks`, _stopCheckPostRequest, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    postStopChecksUrl(requestParameters) {
        const _stopCheckPostRequest = requestParameters["stopCheckPostRequest"];
        return `${this.configuration.basePath}/client-api/v1/stop-checks`;
    }
}
StopChecksHttpService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: StopChecksHttpService, deps: [{ token: i1.HttpClient }, { token: STOP_CHECKS_BASE_PATH, optional: true }, { token: StopChecksConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
StopChecksHttpService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: StopChecksHttpService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: StopChecksHttpService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [STOP_CHECKS_BASE_PATH]
                }] }, { type: StopChecksConfiguration, decorators: [{
                    type: Optional
                }] }]; } });

const StopChecksHttpServiceGetStopChecksMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/stop-checks",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "stopChecks": [{
                            "id": "cb881630-0860-4d2b-b0e1-ff288c531b91",
                            "originatorAccount": {
                                "name": "Main Current",
                                "identification": "0256108927",
                                "schemeName": "BBAN"
                            },
                            "reason": "Invalid name",
                            "status": "REJECTED",
                            "check": {
                                "number": "10000001337900112340"
                            },
                            "dateOfRequest": "2019-09-24T12:00:00.000Z",
                            "expirationDate": "2019-09-30"
                        }, {
                            "id": "d201e455-d744-44e5-8fc6-458be1cc99e7",
                            "originatorAccount": {
                                "name": "Shared Current",
                                "identification": "0256108944",
                                "schemeName": "BBAN"
                            },
                            "reason": "Checks stolen",
                            "status": "ACCEPTED",
                            "checkNumberRange": {
                                "firstNumber": "10000001337900112345",
                                "lastNumber": "10000001337900112355"
                            },
                            "dateOfRequest": "2019-09-30T16:00:00.000Z",
                            "expirationDate": "2019-10-07"
                        }, {
                            "id": "b60a8955-1523-429d-bdbe-905d95225a9c",
                            "originatorAccount": {
                                "name": "Main Checking Current",
                                "identification": "0256108963",
                                "schemeName": "BBAN"
                            },
                            "reason": "Insufficient funds",
                            "status": "PENDING",
                            "check": {
                                "number": "10000001337900112344",
                                "amount": {
                                    "currencyCode": "USD",
                                    "amount": "913.32"
                                },
                                "payeeName": "Jack S. P. Arrow",
                                "dateIssued": "2018-10-18"
                            },
                            "dateOfRequest": "2019-10-24T12:00:00.000Z",
                            "expirationDate": "2019-10-30"
                        }],
                    "totalStopChecks": 3
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
        ]
    }]);
const StopChecksHttpServicePostStopChecksMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/stop-checks",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "stopCheck": {
                        "id": "cb881630-0860-4d2b-b0e1-ff288c531b91",
                        "originatorAccount": {
                            "name": "Main Current",
                            "identification": "0256108927",
                            "schemeName": "BBAN"
                        },
                        "reason": "Invalid name",
                        "status": "REJECTED",
                        "check": {
                            "number": "10000001337900112340"
                        },
                        "dateOfRequest": "2019-09-24T12:00:00.000Z",
                        "expirationDate": "2019-09-30"
                    }
                }
            },
            {
                status: 200,
                body: {
                    "stopCheck": {
                        "id": "b60a8955-1523-429d-bdbe-905d95225a9c",
                        "originatorAccount": {
                            "name": "Main Checking Current",
                            "identification": "0256108963",
                            "schemeName": "BBAN"
                        },
                        "reason": "Insufficient funds",
                        "status": "PENDING",
                        "check": {
                            "number": "10000001337900112344",
                            "amount": {
                                "currencyCode": "USD",
                                "amount": "913.32"
                            },
                            "payeeName": "Jack S. P. Arrow",
                            "dateIssued": "2018-10-18"
                        },
                        "dateOfRequest": "2019-10-24T12:00:00.000Z",
                        "expirationDate": "2019-10-30"
                    }
                }
            },
            {
                status: 200,
                body: {
                    "stopCheck": {
                        "id": "d201e455-d744-44e5-8fc6-458be1cc99e7",
                        "originatorAccount": {
                            "name": "Shared Current",
                            "identification": "0256108944",
                            "schemeName": "BBAN"
                        },
                        "reason": "Checks stolen",
                        "status": "ACCEPTED",
                        "checkNumberRange": {
                            "firstNumber": "10000001337900112345",
                            "lastNumber": "10000001337900112355"
                        },
                        "dateOfRequest": "2019-09-30T16:00:00.000Z",
                        "expirationDate": "2019-10-07"
                    }
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
        ]
    }]);
const StopChecksHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v1/stop-checks",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "stopChecks": [{
                            "id": "cb881630-0860-4d2b-b0e1-ff288c531b91",
                            "originatorAccount": {
                                "name": "Main Current",
                                "identification": "0256108927",
                                "schemeName": "BBAN"
                            },
                            "reason": "Invalid name",
                            "status": "REJECTED",
                            "check": {
                                "number": "10000001337900112340"
                            },
                            "dateOfRequest": "2019-09-24T12:00:00.000Z",
                            "expirationDate": "2019-09-30"
                        }, {
                            "id": "d201e455-d744-44e5-8fc6-458be1cc99e7",
                            "originatorAccount": {
                                "name": "Shared Current",
                                "identification": "0256108944",
                                "schemeName": "BBAN"
                            },
                            "reason": "Checks stolen",
                            "status": "ACCEPTED",
                            "checkNumberRange": {
                                "firstNumber": "10000001337900112345",
                                "lastNumber": "10000001337900112355"
                            },
                            "dateOfRequest": "2019-09-30T16:00:00.000Z",
                            "expirationDate": "2019-10-07"
                        }, {
                            "id": "b60a8955-1523-429d-bdbe-905d95225a9c",
                            "originatorAccount": {
                                "name": "Main Checking Current",
                                "identification": "0256108963",
                                "schemeName": "BBAN"
                            },
                            "reason": "Insufficient funds",
                            "status": "PENDING",
                            "check": {
                                "number": "10000001337900112344",
                                "amount": {
                                    "currencyCode": "USD",
                                    "amount": "913.32"
                                },
                                "payeeName": "Jack S. P. Arrow",
                                "dateIssued": "2018-10-18"
                            },
                            "dateOfRequest": "2019-10-24T12:00:00.000Z",
                            "expirationDate": "2019-10-30"
                        }],
                    "totalStopChecks": 3
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/stop-checks",
        method: "POST",
        responses: [
            {
                status: 200,
                body: {
                    "stopCheck": {
                        "id": "cb881630-0860-4d2b-b0e1-ff288c531b91",
                        "originatorAccount": {
                            "name": "Main Current",
                            "identification": "0256108927",
                            "schemeName": "BBAN"
                        },
                        "reason": "Invalid name",
                        "status": "REJECTED",
                        "check": {
                            "number": "10000001337900112340"
                        },
                        "dateOfRequest": "2019-09-24T12:00:00.000Z",
                        "expirationDate": "2019-09-30"
                    }
                }
            },
            {
                status: 200,
                body: {
                    "stopCheck": {
                        "id": "b60a8955-1523-429d-bdbe-905d95225a9c",
                        "originatorAccount": {
                            "name": "Main Checking Current",
                            "identification": "0256108963",
                            "schemeName": "BBAN"
                        },
                        "reason": "Insufficient funds",
                        "status": "PENDING",
                        "check": {
                            "number": "10000001337900112344",
                            "amount": {
                                "currencyCode": "USD",
                                "amount": "913.32"
                            },
                            "payeeName": "Jack S. P. Arrow",
                            "dateIssued": "2018-10-18"
                        },
                        "dateOfRequest": "2019-10-24T12:00:00.000Z",
                        "expirationDate": "2019-10-30"
                    }
                }
            },
            {
                status: 200,
                body: {
                    "stopCheck": {
                        "id": "d201e455-d744-44e5-8fc6-458be1cc99e7",
                        "originatorAccount": {
                            "name": "Shared Current",
                            "identification": "0256108944",
                            "schemeName": "BBAN"
                        },
                        "reason": "Checks stolen",
                        "status": "ACCEPTED",
                        "checkNumberRange": {
                            "firstNumber": "10000001337900112345",
                            "lastNumber": "10000001337900112355"
                        },
                        "dateOfRequest": "2019-09-30T16:00:00.000Z",
                        "expirationDate": "2019-10-07"
                    }
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
        ]
    },
]);

export { COLLECTION_FORMATS, CONFIG_TOKEN, STOP_CHECKS_BASE_PATH, SchemeName, StopCheckStatus, StopChecksApiModule, StopChecksConfiguration, StopChecksHttpService, StopChecksHttpServiceGetStopChecksMocksProvider, StopChecksHttpServiceMocksProvider, StopChecksHttpServicePostStopChecksMocksProvider };
