// @backbase/ui-ang/esm2015/input-phone/input-phone.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NgSelectModule } from '@ng-select/ng-select';
import { IconModule } from '@backbase/ui-ang/icon';
import { FormsModule } from '@angular/forms';
import { BaseClassesModule } from '@backbase/ui-ang/base-classes';
import { InputPhoneComponent } from './input-phone.component';
import { FocusModule } from '@backbase/ui-ang/focus';
import { SearchBoxModule } from '@backbase/ui-ang/search-box';
import { CountryFlagPipe } from './country-flag-pipe';
import * as i0 from "@angular/core";
export class InputPhoneModule {
}
InputPhoneModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputPhoneModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputPhoneModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputPhoneModule, declarations: [InputPhoneComponent, CountryFlagPipe], imports: [CommonModule, IconModule, FormsModule, BaseClassesModule, FocusModule, NgSelectModule, SearchBoxModule], exports: [InputPhoneComponent] });
InputPhoneModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputPhoneModule, imports: [[CommonModule, IconModule, FormsModule, BaseClassesModule, FocusModule, NgSelectModule, SearchBoxModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputPhoneModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, IconModule, FormsModule, BaseClassesModule, FocusModule, NgSelectModule, SearchBoxModule],
                    declarations: [InputPhoneComponent, CountryFlagPipe],
                    exports: [InputPhoneComponent],
                }]
        }] });
