// @backbase/ui-ang/esm2015/_example/highlight/showcase/highlight-showcase.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/amount";
import * as i2 from "@backbase/ui-ang/highlight";
let HighlightShowcaseComponent = class HighlightShowcaseComponent {
};
HighlightShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HighlightShowcaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
HighlightShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: HighlightShowcaseComponent, selector: "bb-highlight-ui-showcase", ngImport: i0, template: "<div class=\"bb-block bb-block--xl\">\n  <h4>Default</h4>\n  <bb-amount-ui [amount]=\"12345.47\" bbHighlight></bb-amount-ui>\n</div>\n<div class=\"bb-block bb-block--xl\">\n  <h4>Highlight with positive and negative</h4>\n  <bb-amount-ui\n    [amount]=\"12345.47\"\n    positiveClass=\"bb-amount--positive\"\n    negativeClass=\"bb-amount--negative\"\n    bbHighlight\n  ></bb-amount-ui>\n  <br />\n  <bb-amount-ui\n    [amount]=\"-12345.47\"\n    positiveClass=\"bb-amount--positive\"\n    negativeClass=\"bb-amount--negative\"\n    bbHighlight\n  ></bb-amount-ui>\n</div>\n<div>\n  <h4>Highlight off</h4>\n  <bb-amount-ui [amount]=\"12345.47\" [bbHighlight]=\"false\"></bb-amount-ui>\n</div>\n", components: [{ type: i1.AmountComponent, selector: "bb-amount-ui", inputs: ["currency", "showPlusSign", "mapCurrency", "showPercent", "abbreviate", "decimalPlaces", "trailingZeroes", "amount"] }], directives: [{ type: i2.HighlightDirective, selector: "[bbHighlight]", inputs: ["amount", "bbHighlight", "negativeClass", "positiveClass"] }] });
HighlightShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
], HighlightShowcaseComponent);
export { HighlightShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HighlightShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-highlight-ui-showcase',
                    templateUrl: './highlight-showcase.component.html',
                }]
        }] });
