// @backbase/ui-ang/esm2015/_example/highlight/showcase/highlight-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HighlightShowcaseComponent } from './highlight-showcase.component';
import { HighlightModule } from '@backbase/ui-ang/highlight';
import { AmountModule } from '@backbase/ui-ang/amount';
import * as i0 from "@angular/core";
export class HighlightShowcaseModule {
}
HighlightShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HighlightShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
HighlightShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HighlightShowcaseModule, declarations: [HighlightShowcaseComponent], imports: [CommonModule, HighlightModule, AmountModule] });
HighlightShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HighlightShowcaseModule, imports: [[CommonModule, HighlightModule, AmountModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HighlightShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, HighlightModule, AmountModule],
                    declarations: [HighlightShowcaseComponent],
                    entryComponents: [HighlightShowcaseComponent],
                }]
        }] });
