// @backbase/loans-collection-ui-ang/esm2015/lib/loan-summary/retail-loan-summary-progress-bar/retail-loan-summary-progress-bar.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ProgressbarModule } from '@backbase/ui-ang/progressbar';
import { RetailLoanSummaryProgressBarComponent } from './retail-loan-summary-progress-bar.component';
import { LoanSummaryAmountItemModule } from '../loan-summary-amount-item/loan-summary-amount-item.module';
import * as i0 from "@angular/core";
export class RetailLoanSummaryProgressBarModule {
}
/** @nocollapse */ RetailLoanSummaryProgressBarModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RetailLoanSummaryProgressBarModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ RetailLoanSummaryProgressBarModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RetailLoanSummaryProgressBarModule, declarations: [RetailLoanSummaryProgressBarComponent], imports: [CommonModule, LoanSummaryAmountItemModule, ProgressbarModule], exports: [RetailLoanSummaryProgressBarComponent] });
/** @nocollapse */ RetailLoanSummaryProgressBarModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RetailLoanSummaryProgressBarModule, imports: [[CommonModule, LoanSummaryAmountItemModule, ProgressbarModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RetailLoanSummaryProgressBarModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [RetailLoanSummaryProgressBarComponent],
                    imports: [CommonModule, LoanSummaryAmountItemModule, ProgressbarModule],
                    exports: [RetailLoanSummaryProgressBarComponent],
                }]
        }] });
