// @backbase/foundation-ang/esm2015/core/backbase-core/rendering/root/root.component.js
import { Component, ChangeDetectionStrategy, Optional, NgZone, ViewChild, ViewContainerRef, Inject, } from '@angular/core';
import { flatMap } from 'rxjs/operators';
import { ӨRootContainerService } from '../../models/root-container.service';
import { ӨItemNavigationService } from '../../communication/item-navigation.service';
import { StepUpService } from '../../stepup/stepup.service';
import { SessionTimeoutService } from '../../session-timeout/session-timeout.service';
import { PUBSUB } from '@backbase/foundation-ang/web-sdk';
export class RootComponent {
    constructor(rootContainer, zone, itemNavigation, events, stepupService, sessionTimeoutService) {
        this.rootContainer = rootContainer;
        this.zone = zone;
        this.itemNavigation = itemNavigation;
        this.events = events;
        this.stepupService = stepupService;
        this.sessionTimeoutService = sessionTimeoutService;
        this.children = this.rootContainer.model.pipe(flatMap(state => state.children));
        this.eventListener = ({ name }) => {
            // [FIXME] This zone should almost certainly be handled in the EventBus, but can't be currently
            // because we only provide an interface not a wrapper
            this.zone.run(() => {
                this.itemNavigation.navigateToItem(name, {});
            });
        };
    }
    ngOnInit() {
        if (this.events) {
            this.events.subscribe('navigation.viewItem', this.eventListener);
        }
        if (this.stepupService && this.stepupContainer) {
            this.stepupService.createStepUpComponent(this.stepupContainer);
        }
        if (this.sessionTimeoutService && this.sessionTimeoutContainer) {
            this.sessionTimeoutService.createSessionTimeoutComponent(this.sessionTimeoutContainer);
        }
    }
    ngOnDestroy() {
        if (this.events) {
            this.events.unsubscribe('navigation.viewItem', this.eventListener);
        }
    }
}
RootComponent.decorators = [
    { type: Component, args: [{
                selector: 'bb-root',
                template: `
    <bb-area [children]="children | async"></bb-area>
    <template #stepupContainer></template>
    <template #sessionTimeoutContainer></template>
  `,
                changeDetection: ChangeDetectionStrategy.OnPush
            },] }
];
RootComponent.ctorParameters = () => [
    { type: ӨRootContainerService },
    { type: NgZone },
    { type: ӨItemNavigationService },
    { type: undefined, decorators: [{ type: Optional }, { type: Inject, args: [PUBSUB,] }] },
    { type: StepUpService, decorators: [{ type: Optional }] },
    { type: SessionTimeoutService, decorators: [{ type: Optional }] }
];
RootComponent.propDecorators = {
    stepupContainer: [{ type: ViewChild, args: ['stepupContainer', { read: ViewContainerRef, static: true },] }],
    sessionTimeoutContainer: [{ type: ViewChild, args: ['sessionTimeoutContainer', { read: ViewContainerRef, static: true },] }]
};
