// @backbase/ui-ang/esm2015/input-radio-group/input-radio.component.js
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * @name InputRadioComponent
 *
 * @a11y Current component provide option to pass needed accessibility
 * attributes. You need to take care of properties that are required in your case :
 *  - role
 *  - aria-activedescendant
 *  - aria-describedby
 *  - aria-expanded
 *  - aria-invalid
 *  - aria-label
 *  - aria-labelledby
 *  - aria-owns
 *
 * @description
 * Component that represents a single radio that is part of the radio group component.
 */
export class InputRadioComponent {
    constructor() {
        this.disabled = false;
    }
}
InputRadioComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputRadioComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
InputRadioComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: InputRadioComponent, selector: "bb-input-radio-ui", inputs: { labelTemplate: "labelTemplate", labelTemplateContext: "labelTemplateContext", value: "value", label: "label", ariaLabel: ["aria-label", "ariaLabel"], ariaLabelledby: ["aria-labelledby", "ariaLabelledby"], ariaDescribedby: ["aria-describedby", "ariaDescribedby"], role: "role", ariaActivedescendant: ["aria-activedescendant", "ariaActivedescendant"], ariaExpanded: ["aria-expanded", "ariaExpanded"], ariaInvalid: ["aria-invalid", "ariaInvalid"], ariaOwns: ["aria-owns", "ariaOwns"], disabled: "disabled" }, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputRadioComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-input-radio-ui',
                    template: '',
                }]
        }], propDecorators: { labelTemplate: [{
                type: Input
            }], labelTemplateContext: [{
                type: Input
            }], value: [{
                type: Input
            }], label: [{
                type: Input
            }], ariaLabel: [{
                type: Input,
                args: ['aria-label']
            }], ariaLabelledby: [{
                type: Input,
                args: ['aria-labelledby']
            }], ariaDescribedby: [{
                type: Input,
                args: ['aria-describedby']
            }], role: [{
                type: Input
            }], ariaActivedescendant: [{
                type: Input,
                args: ['aria-activedescendant']
            }], ariaExpanded: [{
                type: Input,
                args: ['aria-expanded']
            }], ariaInvalid: [{
                type: Input,
                args: ['aria-invalid']
            }], ariaOwns: [{
                type: Input,
                args: ['aria-owns']
            }], disabled: [{
                type: Input
            }] } });
