// @backbase/ui-ang/esm2015/input-radio-group/input-radio-group.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { InputRadioGroupComponent } from './input-radio-group.component';
import { InputRadioComponent } from './input-radio.component';
import { NgbButtonsModule } from '@ng-bootstrap/ng-bootstrap';
import { FormsModule } from '@angular/forms';
import * as i0 from "@angular/core";
export class InputRadioGroupModule {
}
InputRadioGroupModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputRadioGroupModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputRadioGroupModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputRadioGroupModule, declarations: [InputRadioGroupComponent, InputRadioComponent], imports: [CommonModule, NgbButtonsModule, FormsModule], exports: [InputRadioGroupComponent, InputRadioComponent] });
InputRadioGroupModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputRadioGroupModule, imports: [[CommonModule, NgbButtonsModule, FormsModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: InputRadioGroupModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, NgbButtonsModule, FormsModule],
                    declarations: [InputRadioGroupComponent, InputRadioComponent],
                    exports: [InputRadioGroupComponent, InputRadioComponent],
                }]
        }] });
