// @backbase/ui-ang/esm2015/amount-input/amount-input.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { BaseClassesModule } from '@backbase/ui-ang/base-classes';
import { AmountInputComponent } from './amount-input.component';
import * as i0 from "@angular/core";
export class AmountInputModule {
}
AmountInputModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AmountInputModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AmountInputModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AmountInputModule, declarations: [AmountInputComponent], imports: [CommonModule, FormsModule, BaseClassesModule], exports: [AmountInputComponent] });
AmountInputModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AmountInputModule, imports: [[CommonModule, FormsModule, BaseClassesModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AmountInputModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, FormsModule, BaseClassesModule],
                    declarations: [AmountInputComponent],
                    exports: [AmountInputComponent],
                }]
        }] });
