// @backbase/ui-ang/esm2015/product-item-investment-account/product-item-investment-account.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { CustomProductItemInvestmentAccountDetailsFormatterDirective, CustomProductItemInvestmentAccountTitleDirective, ProductItemInvestmentAccountComponent, } from './product-item-investment-account.component';
import { AmountModule } from '@backbase/ui-ang/amount';
import { PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import { HighlightModule } from '@backbase/ui-ang/highlight';
import { IconModule } from '@backbase/ui-ang/icon';
import * as i0 from "@angular/core";
export class ProductItemInvestmentAccountModule {
}
ProductItemInvestmentAccountModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemInvestmentAccountModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductItemInvestmentAccountModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemInvestmentAccountModule, declarations: [ProductItemInvestmentAccountComponent,
        CustomProductItemInvestmentAccountTitleDirective,
        CustomProductItemInvestmentAccountDetailsFormatterDirective], imports: [CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule], exports: [ProductItemInvestmentAccountComponent,
        CustomProductItemInvestmentAccountTitleDirective,
        CustomProductItemInvestmentAccountDetailsFormatterDirective] });
ProductItemInvestmentAccountModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemInvestmentAccountModule, imports: [[CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemInvestmentAccountModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule],
                    declarations: [
                        ProductItemInvestmentAccountComponent,
                        CustomProductItemInvestmentAccountTitleDirective,
                        CustomProductItemInvestmentAccountDetailsFormatterDirective,
                    ],
                    exports: [
                        ProductItemInvestmentAccountComponent,
                        CustomProductItemInvestmentAccountTitleDirective,
                        CustomProductItemInvestmentAccountDetailsFormatterDirective,
                    ],
                }]
        }] });
