// @backbase/manage-bill-payments-journey-ang/esm2015/src/manage-bill-payments-journey.component.js
import { Component } from '@angular/core';
import { ManageBillPaymentsNavigationService } from './services/manage-bill-payments-navigation.service';
import { ManageBillPaymentsJourneyConfigService } from './services/manage-bill-payments-journey-config.service';
import { ManageBillPaymentsDataService } from './services/manage-bill-payments-data.service';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
export class ManageBillPaymentsJourneyComponent {
}
ManageBillPaymentsJourneyComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageBillPaymentsJourneyComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ManageBillPaymentsJourneyComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: ManageBillPaymentsJourneyComponent, selector: "bb-manage-bill-payments-journey", providers: [
        ManageBillPaymentsJourneyConfigService,
        ManageBillPaymentsDataService,
        ManageBillPaymentsNavigationService,
    ], ngImport: i0, template: `<router-outlet></router-outlet>`, isInline: true, directives: [{ type: i1.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageBillPaymentsJourneyComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-manage-bill-payments-journey',
                    template: `<router-outlet></router-outlet>`,
                    providers: [
                        ManageBillPaymentsJourneyConfigService,
                        ManageBillPaymentsDataService,
                        ManageBillPaymentsNavigationService,
                    ],
                }]
        }] });
