// @backbase/manage-bill-payments-journey-ang/esm2015/src/manage-bill-payments-journey.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { provideRoutes, RouterModule } from '@angular/router';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { BillpayEnrolmentGuard, BillpayJourneysCommonModule } from '@backbase/billpay-journeys-common';
import { ManageBillPaymentsNavigationService } from './services/manage-bill-payments-navigation.service';
import { ManageBillPaymentsDataService } from './services/manage-bill-payments-data.service';
import { BillPayPaymentsListViewComponent } from './views/billpay-payments-list-view/billpay-payments-list-view.component';
import { BillPayPaymentDetailsViewComponent } from './views/billpay-payment-details-view/billpay-payment-details-view.component';
import { BillPayEnrolmentViewComponent } from './views/billpay-enrolment-view/billpay-enrolment-view.component';
import { BillpayPaymentDetailsComponent } from './components/billpay-payment-details/billpay-payment-details.component';
import { BillpayPaymentStatusBadgeComponent } from './components/billpay-payment-status-badge/billpay-payment-status-badge.component';
import { BillpayPaymentsDeleteModalComponent } from './components/billpay-payments-delete-modal/billpay-payments-delete-modal.component';
import { ManageBillPaymentsJourneyComponent } from './manage-bill-payments-journey.component';
import { BillpayPaymentsGroupComponent } from './components/billpay-payments-group/billpay-payments-group.component';
import { BillpayPaymentCardComponent } from './components/billpay-payment-card/billpay-payment-card.component';
import { AlertModule } from '@backbase/ui-ang/alert';
import { BadgeModule } from '@backbase/ui-ang/badge';
import { BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import { IconModule } from '@backbase/ui-ang/icon';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const uiModules = [IconModule, AlertModule, BadgeModule, BbDatePipeModule];
const services = [ManageBillPaymentsNavigationService, ManageBillPaymentsDataService];
const components = [
    ManageBillPaymentsJourneyComponent,
    BillPayPaymentsListViewComponent,
    BillPayPaymentDetailsViewComponent,
    BillPayEnrolmentViewComponent,
    BillpayPaymentDetailsComponent,
    BillpayPaymentStatusBadgeComponent,
    BillpayPaymentsDeleteModalComponent,
    BillpayPaymentsGroupComponent,
    BillpayPaymentCardComponent,
];
export const defaultRoute = {
    path: '',
    component: ManageBillPaymentsJourneyComponent,
    children: [
        { path: 'enrolment', component: BillPayEnrolmentViewComponent },
        {
            path: '',
            canActivate: [BillpayEnrolmentGuard],
            children: [
                { path: '', component: BillPayPaymentsListViewComponent, pathMatch: 'full' },
                { path: ':id', component: BillPayPaymentDetailsViewComponent },
            ],
        },
    ],
};
export class ManageBillPaymentsJourneyModule {
    static forRoot(data = { route: defaultRoute }) {
        return {
            ngModule: ManageBillPaymentsJourneyModule,
            providers: [provideRoutes([data.route])],
        };
    }
}
ManageBillPaymentsJourneyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageBillPaymentsJourneyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ManageBillPaymentsJourneyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageBillPaymentsJourneyModule, declarations: [ManageBillPaymentsJourneyComponent,
        BillPayPaymentsListViewComponent,
        BillPayPaymentDetailsViewComponent,
        BillPayEnrolmentViewComponent,
        BillpayPaymentDetailsComponent,
        BillpayPaymentStatusBadgeComponent,
        BillpayPaymentsDeleteModalComponent,
        BillpayPaymentsGroupComponent,
        BillpayPaymentCardComponent], imports: [CommonModule,
        RouterModule,
        BillpayJourneysCommonModule, i1.BackbaseCoreModule, IconModule, AlertModule, BadgeModule, BbDatePipeModule] });
ManageBillPaymentsJourneyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageBillPaymentsJourneyModule, providers: [...services], imports: [[
            CommonModule,
            RouterModule,
            BillpayJourneysCommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { ManageBillPaymentsJourneyComponent },
            }),
            ...uiModules,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageBillPaymentsJourneyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...components],
                    imports: [
                        CommonModule,
                        RouterModule,
                        BillpayJourneysCommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { ManageBillPaymentsJourneyComponent },
                        }),
                        ...uiModules,
                    ],
                    providers: [...services],
                }]
        }] });
