// @backbase/turnovers-journey-ang/esm2015/src/views/turnovers-view.component.js
import { ChangeDetectionStrategy, Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "../turnovers-journey-config.service";
import * as i3 from "@backbase/ui-ang/header";
import * as i4 from "@backbase/product-summary-account-selector-widget-ang";
import * as i5 from "@backbase/turnovers-widget-ang";
// TODO: remove utility classes
export class TurnoversViewComponent {
    constructor(router, route, config) {
        this.router = router;
        this.route = route;
        this.config = config;
    }
    onItemSelect(selectedAccount) {
        this.router.navigate(['.', { selectedAccount }], { relativeTo: this.route });
    }
}
TurnoversViewComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TurnoversViewComponent, deps: [{ token: i1.Router }, { token: i1.ActivatedRoute }, { token: i2.TurnoversJourneyService }], target: i0.ɵɵFactoryTarget.Component });
TurnoversViewComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: TurnoversViewComponent, selector: "bb-turnovers-view", ngImport: i0, template: `
    <bb-header-ui
      headingType="h1"
      headingClasses="bb-heading-widget__heading px-0 pb-5"
      heading="Cash Flow"
      i18n-heading="Cash flow chart|Heading for Turnovers journey@@turnovers.journey.heading.text"
    ></bb-header-ui>
    <div class="bb-block--xl">
      <bb-product-summary-account-selector-widget
        (selectedAccount)="onItemSelect($event)"
      ></bb-product-summary-account-selector-widget>
    </div>
    <div class="bb-card card-body">
      <bb-turnovers-widget [forecast]="config.showForecast"></bb-turnovers-widget>
    </div>
  `, isInline: true, components: [{ type: i3.HeaderComponent, selector: "bb-header-ui", inputs: ["headingClasses", "headingType", "heading"] }, { type: i4.ProductSummaryAccountSelectorWidgetComponent, selector: "bb-product-summary-account-selector-widget", inputs: ["event.refreshAccountsList", "multiple", "filterItems", "showAccountState"], outputs: ["selectedAccount", "selectedKind"] }, { type: i5.TurnoversWidgetComponent, selector: "bb-turnovers-widget", inputs: ["forecast"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TurnoversViewComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-turnovers-view',
                    template: `
    <bb-header-ui
      headingType="h1"
      headingClasses="bb-heading-widget__heading px-0 pb-5"
      heading="Cash Flow"
      i18n-heading="Cash flow chart|Heading for Turnovers journey@@turnovers.journey.heading.text"
    ></bb-header-ui>
    <div class="bb-block--xl">
      <bb-product-summary-account-selector-widget
        (selectedAccount)="onItemSelect($event)"
      ></bb-product-summary-account-selector-widget>
    </div>
    <div class="bb-card card-body">
      <bb-turnovers-widget [forecast]="config.showForecast"></bb-turnovers-widget>
    </div>
  `,
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.Router }, { type: i1.ActivatedRoute }, { type: i2.TurnoversJourneyService }]; } });
