// @backbase/loans-collection-ui-ang/esm2015/lib/loan-list-item/loan-item-type/loan-item-type.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { LoanItemTypeComponent } from './loan-item-type.component';
import * as i0 from "@angular/core";
export class LoanItemTypeModule {
}
/** @nocollapse */ LoanItemTypeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemTypeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanItemTypeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemTypeModule, declarations: [LoanItemTypeComponent], imports: [CommonModule], exports: [LoanItemTypeComponent] });
/** @nocollapse */ LoanItemTypeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemTypeModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemTypeModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanItemTypeComponent],
                    imports: [CommonModule],
                    exports: [LoanItemTypeComponent],
                }]
        }] });
