// @backbase/loans-collection-ui-ang/esm2015/lib/loan-list-item/loan-item-type/loan-item-type.component.js
import { Component, ChangeDetectionStrategy, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class LoanItemTypeComponent {
}
/** @nocollapse */ LoanItemTypeComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemTypeComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ LoanItemTypeComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: LoanItemTypeComponent, selector: "bb-loan-item-type", inputs: { loanType: "loanType", useSubtitleClass: "useSubtitleClass" }, ngImport: i0, template: "<span class=\"sr-only\" i18n=\"Loan item type a11y prefix@@loans.loan-item.loan-type.a11y.prefix\"> Type: </span>\n<span\n  data-role=\"bb-loans-loan-type\"\n  class=\"bb-text-support bb-loans-margin-right-sm\"\n  [ngClass]=\"{\n    'bb-subheader': !useSubtitleClass,\n    'bb-subtitle': useSubtitleClass\n  }\"\n>\n  {{ loanType }}\n</span>\n<span class=\"sr-only\">,</span>\n", directives: [{ type: i1.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanItemTypeComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-loan-item-type',
                    templateUrl: './loan-item-type.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { loanType: [{
                type: Input
            }], useSubtitleClass: [{
                type: Input
            }] } });
