// @backbase/connect-external-accounts-journey-ang/esm2015/lib/components/empty-list/empty-list.component.js
import { Component, EventEmitter, Output } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/empty-state";
import * as i2 from "@backbase/ui-ang/button";
export class EmptyListComponent {
    constructor() {
        /**
         * Open modal event emitter
         */
        this.openModal = new EventEmitter();
    }
    openAccountModal() {
        this.openModal.emit();
    }
}
/** @nocollapse */ EmptyListComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EmptyListComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ EmptyListComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: EmptyListComponent, selector: "bb-a2a-empty-list", outputs: { openModal: "openModal" }, ngImport: i0, template: "<div class=\"bb-state-container card\">\n  <bb-empty-state-ui\n    class=\"bb-empty-state__container bb-block bb-block--md\"\n    i18n-title=\"@@link-account.state.noLinkedAccount.title\"\n    title=\"You have not connected any account yet\"\n    i18n-subtitle=\"@@link-account.state.noLinkedAccount.subtitle\"\n    subtitle=\"You need to connect your accounts before starting free fees transfers.\"\n    iconSize=\"xxl\"\n    iconModifier=\"swap-horiz\"\n  >\n  </bb-empty-state-ui>\n  <div class=\"bb-text-align-center bb-block bb-block--md\">\n    <button\n      bbButton\n      (click)=\"openAccountModal()\"\n      data-role=\"payord-a2a-linked-external-account-action-btn\"\n      buttonSize=\"md\"\n    >\n      <span\n        i18n=\"@@payments.link-account.dialog.action.open\"\n        data-role=\"payord-a2a-linked-external-account-action-btn-text\"\n      >\n        Connect account\n      </span>\n    </button>\n  </div>\n</div>\n", components: [{ type: i1.EmptyStateComponent, selector: "bb-empty-state-ui", inputs: ["title", "subtitle", "showIcon", "iconClasses", "iconModifier", "iconSize", "iconColor"] }], directives: [{ type: i2.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EmptyListComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-a2a-empty-list',
                    templateUrl: './empty-list.component.html',
                }]
        }], propDecorators: { openModal: [{
                type: Output
            }] } });
