// @backbase/ui-ang/esm2015/rich-text-editor/bb-rich-text-editor-actions/rich-text-editor-actions.component.js
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./rich-text-editor-actions.service";
export class RichTextEditorActionsComponent {
    constructor(ma) {
        this.ma = ma;
        /**
         * This Boolean attribute indicates that the user cannot interact with the button.  Defaults to
         * false.
         */
        this.disabled = false;
        /**
         * Placement of the action tooltip relative to the action button: auto | top | bottom | left | right.
         */
        this.actionsTooltipPlacement = 'auto';
    }
    result(command, value) {
        this.ma.exec(command, value);
    }
}
RichTextEditorActionsComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RichTextEditorActionsComponent, deps: [{ token: i1.RichTextEditorActionsService }], target: i0.ɵɵFactoryTarget.Component });
RichTextEditorActionsComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: RichTextEditorActionsComponent, selector: "ng-component", inputs: { disabled: "disabled", actionsTooltipPlacement: "actionsTooltipPlacement" }, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RichTextEditorActionsComponent, decorators: [{
            type: Component,
            args: [{
                    template: '',
                }]
        }], ctorParameters: function () { return [{ type: i1.RichTextEditorActionsService }]; }, propDecorators: { disabled: [{
                type: Input
            }], actionsTooltipPlacement: [{
                type: Input
            }] } });
