// @backbase/ui-ang/esm2015/badge-counter/badge-counter.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { BadgeCounterComponent } from './badge-counter.component';
import * as i0 from "@angular/core";
export class BadgeCounterModule {
}
BadgeCounterModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BadgeCounterModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
BadgeCounterModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BadgeCounterModule, declarations: [BadgeCounterComponent], imports: [CommonModule], exports: [BadgeCounterComponent] });
BadgeCounterModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BadgeCounterModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BadgeCounterModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [BadgeCounterComponent],
                    exports: [BadgeCounterComponent],
                }]
        }] });
