// @backbase/ui-ang/esm2015/badge-counter/badge-counter.component.js
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
/**
 * @name BadgeCounterComponent
 *
 * @description
 * Component that displays a counter badge.
 */
export class BadgeCounterComponent {
    constructor() {
        /**
         * The (background) color of the badge. Defaults to 'primary'.
         */
        this.color = 'primary';
        /**
         * The count to show in the badge. Defaults to 0 and badge will be hidden if count equals 0.
         */
        this.count = 0;
        /**
         * The max count value to limit text inside badge. Defaults to 999.
         */
        this.maxCount = 999;
    }
    ngOnChanges() {
        const value = this.maxCount && this.maxCount > 0 && this.count && this.maxCount < this.count ? `${this.maxCount}+` : this.count;
        this.countValue = value ? value.toString() : undefined;
    }
}
BadgeCounterComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BadgeCounterComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
BadgeCounterComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: BadgeCounterComponent, selector: "bb-badge-counter-ui", inputs: { color: "color", count: "count", maxCount: "maxCount" }, usesOnChanges: true, ngImport: i0, template: "<div class=\"bb-badge-counter\">\n  <span class=\"badge badge-counter\" *ngIf=\"countValue\" [ngClass]=\"[color ? 'badge-counter-' + color : '']\">\n    {{ countValue }}\n  </span>\n</div>\n", directives: [{ type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i1.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: BadgeCounterComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-badge-counter-ui',
                    templateUrl: './badge-counter.component.html',
                }]
        }], propDecorators: { color: [{
                type: Input
            }], count: [{
                type: Input
            }], maxCount: [{
                type: Input
            }] } });
