// @backbase/consent-list-widget-ang/esm2015/src/consent-list/consent-list.component.js
import { Component, Input, Output, EventEmitter } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../consent-item/consent-item.component";
import * as i2 from "@angular/common";
export class ConsentListComponent {
    constructor() {
        /**
         * List of consents
         */
        this.consentsList = [];
        /**
         * Emitter that fires when you select consent
         */
        this.selectConsent = new EventEmitter();
    }
    onConsentSelect(consentId) {
        this.selectConsent.emit(consentId);
    }
}
ConsentListComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentListComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ConsentListComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: ConsentListComponent, selector: "bb-consent-list-component", inputs: { consentsList: "consentsList" }, outputs: { selectConsent: "selectConsent" }, ngImport: i0, template: "<div class=\"list-group list-group-flush bb-consent-list\">\n  <div\n    *ngFor=\"let consent of consentsList; let i = index\"\n    role=\"button\"\n    tabindex=\"0\"\n    [attr.data-role]=\"'consent-list-item-' + consent.tpp.name\"\n    class=\"list-group-item list-group-item-action bb-consent-list__item\"\n    (keyup.enter)=\"onConsentSelect(consent.id)\"\n    (click)=\"onConsentSelect(consent.id)\">\n    <bb-consent-item-component [consent]=\"consent\"> </bb-consent-item-component>\n  </div>\n</div>\n", components: [{ type: i1.ConsentItemComponent, selector: "bb-consent-item-component", inputs: ["consent"] }], directives: [{ type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConsentListComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-consent-list-component',
                    templateUrl: './consent-list.component.html',
                }]
        }], propDecorators: { consentsList: [{
                type: Input
            }], selectConsent: [{
                type: Output
            }] } });
