// @backbase/ui-ang/esm2015/list-multiple-select/list-multiple-select.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { ListMultipleSelectComponent, ListMultipleSelectActionDirective, ListMultipleSelectItemDirective, } from './list-multiple-select.component';
import { InputCheckboxModule } from '@backbase/ui-ang/input-checkbox';
import { IconModule } from '@backbase/ui-ang/icon';
import { CheckboxGroupModule } from '@backbase/ui-ang/checkbox-group';
import * as i0 from "@angular/core";
export class ListMultipleSelectModule {
}
ListMultipleSelectModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ListMultipleSelectModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ListMultipleSelectModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ListMultipleSelectModule, declarations: [ListMultipleSelectComponent, ListMultipleSelectActionDirective, ListMultipleSelectItemDirective], imports: [CommonModule, ReactiveFormsModule, InputCheckboxModule, IconModule, CheckboxGroupModule], exports: [ListMultipleSelectComponent, ListMultipleSelectActionDirective, ListMultipleSelectItemDirective] });
ListMultipleSelectModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ListMultipleSelectModule, imports: [[CommonModule, ReactiveFormsModule, InputCheckboxModule, IconModule, CheckboxGroupModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ListMultipleSelectModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ReactiveFormsModule, InputCheckboxModule, IconModule, CheckboxGroupModule],
                    declarations: [ListMultipleSelectComponent, ListMultipleSelectActionDirective, ListMultipleSelectItemDirective],
                    exports: [ListMultipleSelectComponent, ListMultipleSelectActionDirective, ListMultipleSelectItemDirective],
                }]
        }] });
