// @backbase/ui-ang/esm2015/_example/product-item-basic-account/showcase/product-item-basic-account-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ProductItemBasicAccountShowcaseComponent } from './product-item-basic-account-showcase.component';
import { ProductItemBasicAccountModule } from '@backbase/ui-ang/product-item-basic-account';
import * as i0 from "@angular/core";
export class ProductItemBasicAccountShowcaseModule {
}
ProductItemBasicAccountShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemBasicAccountShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductItemBasicAccountShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemBasicAccountShowcaseModule, declarations: [ProductItemBasicAccountShowcaseComponent], imports: [CommonModule, ProductItemBasicAccountModule] });
ProductItemBasicAccountShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemBasicAccountShowcaseModule, imports: [[CommonModule, ProductItemBasicAccountModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemBasicAccountShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ProductItemBasicAccountModule],
                    declarations: [ProductItemBasicAccountShowcaseComponent],
                    entryComponents: [ProductItemBasicAccountShowcaseComponent],
                }]
        }] });
