// @backbase/notifications-badge-widget-ang/esm2015/lib/components/humanized-date/humanized-date.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import { HumanizedDateComponent, HumanizedDateCustomizableDirective } from './humanized-date.component';
import * as i0 from "@angular/core";
export class HumanizedDateModule {
}
/** @nocollapse */ HumanizedDateModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HumanizedDateModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ HumanizedDateModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HumanizedDateModule, declarations: [HumanizedDateComponent, HumanizedDateCustomizableDirective], imports: [CommonModule, BbDatePipeModule], exports: [HumanizedDateComponent, HumanizedDateCustomizableDirective] });
/** @nocollapse */ HumanizedDateModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HumanizedDateModule, imports: [[CommonModule, BbDatePipeModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: HumanizedDateModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, BbDatePipeModule],
                    declarations: [HumanizedDateComponent, HumanizedDateCustomizableDirective],
                    exports: [HumanizedDateComponent, HumanizedDateCustomizableDirective],
                }]
        }] });
