// @backbase/loans-collection-ui-ang/esm2015/lib/directives/accessibility-keyboard/accessibility-keyboard.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AccessibilityKeyboardDirective } from './accessibility-keyboard.directive';
import * as i0 from "@angular/core";
export class AccessibilityKeyboardModule {
}
/** @nocollapse */ AccessibilityKeyboardModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccessibilityKeyboardModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ AccessibilityKeyboardModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccessibilityKeyboardModule, declarations: [AccessibilityKeyboardDirective], imports: [CommonModule], exports: [AccessibilityKeyboardDirective] });
/** @nocollapse */ AccessibilityKeyboardModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccessibilityKeyboardModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AccessibilityKeyboardModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [AccessibilityKeyboardDirective],
                    imports: [CommonModule],
                    exports: [AccessibilityKeyboardDirective],
                }]
        }] });
