// @backbase/authorized-users-journey-ang/esm2015/src/authorized-users-journey.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { provideRoutes, RouterModule } from '@angular/router';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { HeaderModule } from '@backbase/ui-ang/header';
import { AuthorizedUsersCreateModule } from './authorized-users-create/authorized-users-create.module';
import { AuthorizedUsersJourneyComponent } from './authorized-users-journey.component';
import { AuthorizedUsersListModule } from './authorized-users-list/authorized-users-list.module';
import { levelsMappingConfigToken } from './common/services/users-ui.service';
import { AuthorizedUsersJourneyConfigService } from './services/authorized-users-journey-config.service';
import { AuthorizedUserCreateViewComponent } from './views/authorized-user-create.component';
import { AuthorizedUsersListViewComponent } from './views/authorized-users-list.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
import * as i2 from "@angular/router";
export const defaultRoute = {
    path: '',
    component: AuthorizedUsersJourneyComponent,
    children: [
        {
            path: '',
            redirectTo: 'list',
            pathMatch: 'full',
        },
        {
            path: 'list',
            data: { title: 'Authorized users' },
            component: AuthorizedUsersListViewComponent,
        },
        {
            path: 'create',
            data: { title: 'Add authorized user' },
            component: AuthorizedUserCreateViewComponent,
        },
    ],
};
export function getLevelsConfig(config) {
    return config.levelsConfig;
}
export class AuthorizedUsersJourneyModule {
    static forRoot(data = { route: defaultRoute }) {
        return {
            ngModule: AuthorizedUsersJourneyModule,
            providers: [
                provideRoutes([data.route]),
                {
                    provide: levelsMappingConfigToken,
                    useFactory: getLevelsConfig,
                    deps: [AuthorizedUsersJourneyConfigService],
                },
            ],
        };
    }
}
AuthorizedUsersJourneyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersJourneyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AuthorizedUsersJourneyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersJourneyModule, declarations: [AuthorizedUsersJourneyComponent, AuthorizedUsersListViewComponent, AuthorizedUserCreateViewComponent], imports: [CommonModule, i1.BackbaseCoreModule, i2.RouterModule, HeaderModule,
        AuthorizedUsersListModule,
        AuthorizedUsersCreateModule] });
AuthorizedUsersJourneyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersJourneyModule, providers: [AuthorizedUsersJourneyConfigService], imports: [[
            CommonModule,
            BackbaseCoreModule.withConfig({
                classMap: { AuthorizedUsersJourneyComponent },
            }),
            RouterModule.forChild([]),
            HeaderModule,
            AuthorizedUsersListModule,
            AuthorizedUsersCreateModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AuthorizedUsersJourneyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [AuthorizedUsersJourneyComponent, AuthorizedUsersListViewComponent, AuthorizedUserCreateViewComponent],
                    imports: [
                        CommonModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { AuthorizedUsersJourneyComponent },
                        }),
                        RouterModule.forChild([]),
                        HeaderModule,
                        AuthorizedUsersListModule,
                        AuthorizedUsersCreateModule,
                    ],
                    providers: [AuthorizedUsersJourneyConfigService],
                }]
        }] });
