// @backbase/ui-ang/esm2015/_example/common-error-state/examples/common-error-state.example.component.js
import { Component } from '@angular/core';
import { HttpErrorResponse } from '@angular/common/http';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/common-error-state";
export class CommonErrorStateExampleComponent {
    constructor() {
        this.pageNotFound = new HttpErrorResponse({ status: 404 });
    }
}
CommonErrorStateExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CommonErrorStateExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
CommonErrorStateExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: CommonErrorStateExampleComponent, selector: "bb-common-error-state-example-ui", ngImport: i0, template: "<bb-common-error-state-ui\n  [error]=\"pageNotFound\"\n></bb-common-error-state-ui>\n", components: [{ type: i1.ErrorCommonStateComponent, selector: "bb-common-error-state-ui", inputs: ["error"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CommonErrorStateExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-common-error-state-example-ui',
                    templateUrl: './common-error-state.example.component.html',
                }]
        }] });
