// @backbase/ui-ang/esm2015/_example/item-log/showcase/item-log-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ItemLogShowcaseComponent } from './item-log-showcase.component';
import { ItemLogModule } from '@backbase/ui-ang/item-log';
import { IconModule } from '@backbase/ui-ang/icon';
import * as i0 from "@angular/core";
export class ItemLogShowcaseModule {
}
ItemLogShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ItemLogShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ItemLogShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ItemLogShowcaseModule, declarations: [ItemLogShowcaseComponent], imports: [CommonModule, ItemLogModule, IconModule] });
ItemLogShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ItemLogShowcaseModule, imports: [[CommonModule, ItemLogModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ItemLogShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ItemLogModule, IconModule],
                    declarations: [ItemLogShowcaseComponent],
                    entryComponents: [ItemLogShowcaseComponent],
                }]
        }] });
