// @backbase/ui-ang/esm2015/_example/item-log/showcase/item-log-showcase.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { BackbaseDocs } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/item-log";
import * as i2 from "@backbase/ui-ang/icon";
import * as i3 from "@angular/common";
let ItemLogShowcaseComponent = class ItemLogShowcaseComponent {
    constructor() {
        this.pendingRecords = [
            {
                userFullName: 'Robin Green',
                status: 'APPROVED',
                createdAt: '2018-01-22T13:44:02Z',
            },
        ];
        this.rejectedRecords = [
            {
                userFullName: 'Green Goblet',
                status: 'REJECTED',
                createdAt: '2018-01-23T13:44:02Z',
                reason: 'Limit exceeded',
            },
            {
                userFullName: 'Robin Green',
                status: 'APPROVED',
                createdAt: '2018-01-22T13:44:02Z',
            },
        ];
        this.successRecords = [
            {
                userFullName: 'Green Goblet',
                status: 'APPROVED',
                createdAt: '2018-01-23T13:44:02Z',
            },
            {
                userFullName: 'Robin Green',
                status: 'APPROVED',
                createdAt: '2018-01-22T13:44:02Z',
            },
        ];
    }
};
ItemLogShowcaseComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ItemLogShowcaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ItemLogShowcaseComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ItemLogShowcaseComponent, selector: "bb-item-log-ui-showcase", ngImport: i0, template: "<div class=\"bb-block bb-block--lg\">\n  <h2>Pending item log</h2>\n  <bb-item-log-ui [records]=\"pendingRecords\" [isPending]=\"true\"></bb-item-log-ui>\n</div>\n<div class=\"bb-block bb-block--lg\">\n  <h2>Success item log</h2>\n  <bb-item-log-ui [records]=\"successRecords\"></bb-item-log-ui>\n</div>\n<div class=\"bb-block bb-block--lg\">\n  <h2>Rejected item log</h2>\n  <bb-item-log-ui [records]=\"rejectedRecords\"></bb-item-log-ui>\n</div>\n<div class=\"bb-block bb-block--lg\">\n  <h2>Custom item log with description</h2>\n  <bb-item-log-ui [records]=\"rejectedRecords\">\n    <ng-template bbItemLogRecord let-record>\n      <bb-icon-ui\n        class=\"bb-item-log-record__icon\"\n        size=\"md\"\n        [name]=\"record?.status === 'REJECTED' ? 'cancel' : 'success'\"\n        [color]=\"record?.status === 'REJECTED' ? 'danger' : 'success'\"\n      ></bb-icon-ui>\n      <div class=\"bb-item-log-record__details\">\n        <span class=\"bb-text-bold\">\n          {{ record?.status | titlecase }}\n        </span>\n        <ng-container> by {{ record?.userFullName }}</ng-container>\n        <div class=\"bb-item-log-record__time\">\n          <time> {{ record?.createdAt | date }}</time>\n          <ng-container> at</ng-container>\n          <time> {{ record?.createdAt | date: 'shortTime' }}</time>\n        </div>\n        <div class=\"bb-stack\" *ngIf=\"record?.reason\">\n          <bb-icon-ui color=\"secondary\" name=\"forum\" size=\"md\" class=\"bb-stack__item\"></bb-icon-ui>\n          <div>\n            User hasn\u2019t posted all his/her details to complete the payment. <br />\n            Unfortunately, we will need to request more payment details now <br />\n            to approve this transaction.\n          </div>\n        </div>\n      </div>\n    </ng-template>\n  </bb-item-log-ui>\n</div>\n", components: [{ type: i1.ItemLogComponent, selector: "bb-item-log-ui", inputs: ["isPending", "records"] }, { type: i2.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], directives: [{ type: i1.ItemLogRecordDirective, selector: "ng-template[bbItemLogRecord]" }, { type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "titlecase": i3.TitleCasePipe, "date": i3.DatePipe } });
ItemLogShowcaseComponent = __decorate([
    BackbaseDocs({
        isTheme2Stable: true,
    })
], ItemLogShowcaseComponent);
export { ItemLogShowcaseComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ItemLogShowcaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-item-log-ui-showcase',
                    templateUrl: './item-log-showcase.component.html',
                }]
        }] });
