// @backbase/product-summary-ui/esm2015/src/bb-product-needs-attention/product-needs-attention.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { AmountModule } from '@backbase/ui-ang/amount';
import { IconModule } from '@backbase/ui-ang/icon';
import { KeyboardClickModule } from '@backbase/ui-ang/keyboard-click-directive';
import { ProductNeedsAttentionComponent, ProductNeedsAttentionCustomizableDirective, } from './product-needs-attention.component';
import { ProductNeedsAttentionItemModule } from '../bb-product-needs-attention-item/product-needs-attention-item.module';
import * as i0 from "@angular/core";
const uiModules = [IconModule, AmountModule, KeyboardClickModule, NgbModule, ProductNeedsAttentionItemModule];
const components = [ProductNeedsAttentionComponent, ProductNeedsAttentionCustomizableDirective];
export class ProductNeedsAttentionModule {
}
ProductNeedsAttentionModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductNeedsAttentionModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductNeedsAttentionModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductNeedsAttentionModule, declarations: [ProductNeedsAttentionComponent, ProductNeedsAttentionCustomizableDirective], imports: [CommonModule, IconModule, AmountModule, KeyboardClickModule, NgbModule, ProductNeedsAttentionItemModule], exports: [ProductNeedsAttentionComponent, ProductNeedsAttentionCustomizableDirective, IconModule, AmountModule, KeyboardClickModule, NgbModule, ProductNeedsAttentionItemModule] });
ProductNeedsAttentionModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductNeedsAttentionModule, imports: [[CommonModule, ...uiModules], IconModule, AmountModule, KeyboardClickModule, NgbModule, ProductNeedsAttentionItemModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductNeedsAttentionModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules],
                    declarations: [...components],
                    exports: [...components, ...uiModules],
                }]
        }] });
