// @backbase/product-summary-ui/esm2015/src/bb-product-needs-attention/product-needs-attention.component.js
import { Component, ChangeDetectionStrategy, Input, Output, EventEmitter, Directive } from '@angular/core';
import { BbTemplate } from '@backbase/foundation-ang/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/icon";
import * as i2 from "../bb-product-needs-attention-item/product-needs-attention-item.component";
import * as i3 from "@backbase/ui-ang/keyboard-click-directive";
import * as i4 from "@ng-bootstrap/ng-bootstrap";
import * as i5 from "@angular/common";
export class ProductNeedsAttentionComponent {
    constructor() {
        /**
         * Emits the id of clicked account
         */
        this.selectAccount = new EventEmitter();
        /**
         * Emits selected product
         */
        this.selectProduct = new EventEmitter();
        /**
         * Emits the boolean value to toggle the expand/collapse state of the collapsable
         */
        this.isExpandedChange = new EventEmitter();
        /**
         * Emits arrangement id when pay now is clicked
         */
        this.payNowClick = new EventEmitter();
        this.hostRef = this;
    }
    onSelectAccount(id) {
        this.selectAccount.emit(id);
    }
    toggleExpanded() {
        this.isExpandedChange.emit(!this.isExpanded);
    }
}
ProductNeedsAttentionComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductNeedsAttentionComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ProductNeedsAttentionComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ProductNeedsAttentionComponent, selector: "bb-product-needs-attention-ui", inputs: { isExpanded: "isExpanded", showIcons: "showIcons", iconName: "iconName", productKind: "productKind" }, outputs: { selectAccount: "selectAccount", selectProduct: "selectProduct", isExpandedChange: "isExpandedChange", payNowClick: "payNowClick" }, ngImport: i0, template: "<ng-container\n  bbProductNeedsAttentionCustomizable\n  [bbHostRef]=\"hostRef\"\n></ng-container>\n\n<ng-template\n  bbProductNeedsAttentionCustomizable\n  let-hostRef\n>\n  <div class=\"bb-product-needs-attention card card-body\">\n    <div\n      role=\"button\"\n      tabindex=\"0\"\n      [attr.aria-expanded]=\"!hostRef.isExpanded\"\n      class=\"bb-product-needs-attention__header\"\n      data-role=\"needs-attention-expand-button\"\n      [class.bb-product-kind__header--collapsed]=\"hostRef.isExpanded\"\n      (bbKeyboardClick)=\"hostRef.toggleExpanded()\"\n    >\n      <div class=\"bb-product-needs-attention__header-title bb-stack\">\n        <bb-icon-ui\n          name=\"error\"\n          color=\"danger\"\n          data-role=\"needs-attention-title-icon\"\n          class=\"bb-product-needs-attention-title-icon bb-stack__item bb-stack__item--spacing-sm\"\n        ></bb-icon-ui>\n        <span \n          data-role=\"needs-attention-title-text\"\n          class=\"bb-product-needs-attention-title-text bb-stack__item\"\n        >Needs Your Attention</span>\n      </div>\n\n      <div class=\"bb-product-needs-attention__header-icon\">\n        <bb-icon-ui\n          class=\"bb-product-needs-attention-header-icon-chevron\"\n          data-role=\"needs-attention-header-icon-chevron\"\n          name=\"toggle-{{!hostRef.isExpanded ? 'up' : 'down'}}\"\n        ></bb-icon-ui>\n      </div>\n    </div>\n\n    <div\n      class=\"bb-product-needs-attention__item-group\"\n      [ngbCollapse]=\"hostRef.isExpanded\"\n    >\n      <div\n        class=\"bb-product-needs-attention-item bb-list__item\"\n        *ngFor=\"let item of hostRef.productKind?.products\"\n      >\n        <bb-product-needs-attention-item-ui\n          data-role=\"needs-attention-product-item\"\n          [item]=\"item\"\n          [showLogo]=\"hostRef.showIcons\"\n          (selectProductKind)=\"hostRef.selectProduct.emit(item); hostRef.onSelectAccount($event.id);\"\n          (payNowClick)=\"hostRef.payNowClick.emit($event);\"\n        ></bb-product-needs-attention-item-ui>\n      </div>\n    </div>\n  </div>\n</ng-template>\n", components: [{ type: i0.forwardRef(function () { return i1.IconComponent; }), selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i0.forwardRef(function () { return i2.ProductNeedsAttentionItemComponent; }), selector: "bb-product-needs-attention-item-ui", inputs: ["item", "showLogo"], outputs: ["selectProductKind", "payNowClick"] }], directives: [{ type: i0.forwardRef(function () { return ProductNeedsAttentionCustomizableDirective; }), selector: "[bbProductNeedsAttentionCustomizable]" }, { type: i0.forwardRef(function () { return i3.KeyboardClickDirective; }), selector: "[bbKeyboardClick]", outputs: ["bbKeyboardClick"] }, { type: i0.forwardRef(function () { return i4.NgbCollapse; }), selector: "[ngbCollapse]", inputs: ["ngbCollapse", "animation"], outputs: ["ngbCollapseChange", "shown", "hidden"], exportAs: ["ngbCollapse"] }, { type: i0.forwardRef(function () { return i5.NgForOf; }), selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductNeedsAttentionComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-product-needs-attention-ui',
                    templateUrl: './product-needs-attention.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { isExpanded: [{
                type: Input
            }], showIcons: [{
                type: Input
            }], iconName: [{
                type: Input
            }], productKind: [{
                type: Input
            }], selectAccount: [{
                type: Output
            }], selectProduct: [{
                type: Output
            }], isExpandedChange: [{
                type: Output
            }], payNowClick: [{
                type: Output
            }] } });
export class ProductNeedsAttentionCustomizableDirective extends BbTemplate {
}
ProductNeedsAttentionCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductNeedsAttentionCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
ProductNeedsAttentionCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: ProductNeedsAttentionCustomizableDirective, selector: "[bbProductNeedsAttentionCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductNeedsAttentionCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbProductNeedsAttentionCustomizable]',
                }]
        }] });
