// @backbase/ui-ang/esm2015/alert/alert.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NgbAlertModule } from '@ng-bootstrap/ng-bootstrap';
import { IconModule } from '@backbase/ui-ang/icon';
import { AlertComponent } from './alert.component';
import * as i0 from "@angular/core";
export class AlertModule {
}
AlertModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AlertModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AlertModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AlertModule, declarations: [AlertComponent], imports: [CommonModule, NgbAlertModule, IconModule], exports: [AlertComponent] });
AlertModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AlertModule, imports: [[CommonModule, NgbAlertModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: AlertModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, NgbAlertModule, IconModule],
                    declarations: [AlertComponent],
                    exports: [AlertComponent],
                }]
        }] });
