// @backbase/ui-ang/esm2015/char-counter/char-counter.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { CharCounterComponent } from './char-counter.component';
import * as i0 from "@angular/core";
export class CharCounterModule {
}
CharCounterModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CharCounterModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CharCounterModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CharCounterModule, declarations: [CharCounterComponent], imports: [CommonModule], exports: [CharCounterComponent] });
CharCounterModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CharCounterModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CharCounterModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [CharCounterComponent],
                    exports: [CharCounterComponent],
                }]
        }] });
