// @backbase/foundation-ang/esm2015/core/backbase-core/routing/navigation/router.service.js
import { Router, ActivatedRoute, PRIMARY_OUTLET } from '@angular/router';
import { Injectable } from '@angular/core';
import { take } from 'rxjs/operators';
import { ItemModel } from '../../models/item-model';
import { AbstractRouterConfigService } from '../config/abstract-router-config.service';
export class RouterService {
    /**
     * Backbase Angular Router abstraction.
     * See https://angular.io/api/router/Router
     *
     * The purpose for Backbase abstracting around the default Angular Router is
     * when linking a to a bb-router-outlet. The outlet name for a bb-router-outlet
     * is dynamically generated by a property on the model, so you can use the
     * Backbase Router service to automatically apply the outletName to the navigation
     * commands.
     *
     * @param router Angular Router
     * @param route Angular ActivatedRoute
     * @param itemModel Model of the Backbase Widget or Container
     */
    constructor(router, route, itemModel, routerConfig) {
        this.router = router;
        this.route = route;
        this.itemModel = itemModel;
        this.routerConfig = routerConfig;
        this.outletName = this.routerConfig.ɵitemOutletName(this.itemModel);
    }
    /**
     * Navigate based on the provided array of commands and a starting point.
     * If no starting route is provided, the navigation is absolute.
     * See Angular Router.navigate
     */
    navigate(commands, extras = {}) {
        return this.createUrlTree(commands, extras).then(urlTree => this.router.navigateByUrl(this.router.serializeUrl(urlTree), extras));
    }
    /**
     * Applies an array of commands to the current url tree and creates a new url tree.
     * See Angular Router.createUrlTree
     */
    createUrlTree(commands, extras = {}) {
        return this.outletName
            .pipe(take(1))
            .toPromise()
            .then(outletName => this.outletCommands(outletName, commands))
            .then(cmnds => this.router.createUrlTree(cmnds, Object.assign({ relativeTo: this.route }, extras)));
    }
    outletCommands(outletName, commands) {
        if (outletName == null || outletName === PRIMARY_OUTLET) {
            return commands;
        }
        else {
            return [{ outlets: { [outletName]: commands } }];
        }
    }
}
RouterService.decorators = [
    { type: Injectable }
];
RouterService.ctorParameters = () => [
    { type: Router },
    { type: ActivatedRoute },
    { type: ItemModel },
    { type: AbstractRouterConfigService }
];
