// @backbase/ui-ang/esm2015/_example/collapsible/showcase/collapsible-showcase.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CollapsibleShowcaseComponent } from './collapsible-showcase.component';
import { CollapsibleModule } from '@backbase/ui-ang/collapsible';
import { IconModule } from '@backbase/ui-ang/icon';
import * as i0 from "@angular/core";
export class CollapsibleShowcaseModule {
}
CollapsibleShowcaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CollapsibleShowcaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CollapsibleShowcaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CollapsibleShowcaseModule, declarations: [CollapsibleShowcaseComponent], imports: [CommonModule, CollapsibleModule, IconModule] });
CollapsibleShowcaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CollapsibleShowcaseModule, imports: [[CommonModule, CollapsibleModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CollapsibleShowcaseModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, CollapsibleModule, IconModule],
                    declarations: [CollapsibleShowcaseComponent],
                    entryComponents: [CollapsibleShowcaseComponent],
                }]
        }] });
