// @backbase/ui-ang/esm2015/modal/modal.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { A11yModule } from '@angular/cdk/a11y';
import { NgbModalModule } from '@ng-bootstrap/ng-bootstrap';
import { ElementViewInitDirective, ModalComponent } from './modal.component';
import { CustomModalFooterDirective, ModalFooterComponent } from './modal-templates/modal-footer.component';
import { CustomModalBodyDirective, ModalBodyComponent } from './modal-templates/modal-body.component';
import { CustomModalHeaderDirective, ModalHeaderComponent } from './modal-templates/modal-header.component';
import * as i0 from "@angular/core";
const components = [
    ModalComponent,
    ModalFooterComponent,
    CustomModalFooterDirective,
    ModalBodyComponent,
    CustomModalBodyDirective,
    ModalHeaderComponent,
    CustomModalHeaderDirective,
];
export class ModalModule {
}
ModalModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ModalModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ModalModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ModalModule, declarations: [ModalComponent,
        ModalFooterComponent,
        CustomModalFooterDirective,
        ModalBodyComponent,
        CustomModalBodyDirective,
        ModalHeaderComponent,
        CustomModalHeaderDirective, ElementViewInitDirective], imports: [CommonModule, NgbModalModule, A11yModule], exports: [ModalComponent,
        ModalFooterComponent,
        CustomModalFooterDirective,
        ModalBodyComponent,
        CustomModalBodyDirective,
        ModalHeaderComponent,
        CustomModalHeaderDirective] });
ModalModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ModalModule, imports: [[CommonModule, NgbModalModule, A11yModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ModalModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, NgbModalModule, A11yModule],
                    declarations: [...components, ElementViewInitDirective],
                    exports: [...components],
                }]
        }] });
