// @backbase/ui-ang/esm2015/ellipsis/ellipsis.module.js
import { NgModule } from '@angular/core';
import { EllipsisComponent } from './ellipsis.component';
import { TooltipModule } from '@backbase/ui-ang/tooltip-directive';
import { CommonModule } from '@angular/common';
import { IsContentTruncatedDirective } from './directives/is-content-truncated.directive';
import * as i0 from "@angular/core";
const directives = [IsContentTruncatedDirective];
export class EllipsisModule {
}
EllipsisModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EllipsisModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
EllipsisModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EllipsisModule, declarations: [EllipsisComponent, IsContentTruncatedDirective], imports: [TooltipModule, CommonModule], exports: [EllipsisComponent, IsContentTruncatedDirective] });
EllipsisModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EllipsisModule, imports: [[TooltipModule, CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: EllipsisModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [TooltipModule, CommonModule],
                    declarations: [EllipsisComponent, ...directives],
                    exports: [EllipsisComponent, ...directives],
                }]
        }] });
