// @backbase/user-common/esm2015/src/components/manage-profile/manage-profile.component.js
import { Component, Input } from '@angular/core';
import { BehaviorSubject, Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "../../services/manage-profile/manage-profile-preferences.service";
export class ManageProfileComponent {
    constructor(manageProfilePreferencesService, cd) {
        this.manageProfilePreferencesService = manageProfilePreferencesService;
        this.cd = cd;
        this.loadingState = true;
        this.loadingErrorState = false;
        this.disableAddressManaging = false;
        /** @deprecated */
        this.widgetPreferences = this.manageProfilePreferencesService.getWidgetPreferences();
        this.userProfileSubject = new BehaviorSubject(undefined);
        this.handleProfileError = (error) => {
            this.loadingState = false;
            this.loadingErrorState = true;
            this.loadingError = error;
            this.cd.markForCheck();
            return new Observable();
        };
        this.onProfileUpdate = () => {
            this.loadingState = true;
            this.loadingError = undefined;
        };
    }
    refreshProfileData() {
        this.handleManageAddressClose();
        this.userProfileSubject.next(undefined);
    }
    handleManageAddressOpen() {
        this.disableAddressManaging = true;
    }
    handleManageAddressClose() {
        this.disableAddressManaging = false;
    }
    setUserProfileObservable() {
        return this.userProfileSubject.asObservable().pipe(map(() => ({
            fullName: '',
        })));
    }
    ngOnInit() {
        this.userProfileObservable = this.setUserProfileObservable();
        this.userProfileSubscription = this.userProfileObservable.subscribe((userProfileData) => {
            this.userProfile = userProfileData;
            this.loadingState = false;
            this.loadingErrorState = false;
            this.cd.markForCheck();
        });
    }
    ngOnDestroy() {
        if (this.userProfileSubscription) {
            this.userProfileSubscription.unsubscribe();
        }
    }
}
ManageProfileComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageProfileComponent, deps: [{ token: i1.ManageProfilePreferencesService }, { token: i0.ChangeDetectorRef }], target: i0.ɵɵFactoryTarget.Component });
ManageProfileComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: ManageProfileComponent, selector: "ng-component", inputs: { notificationDismissTime: "notificationDismissTime" }, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ManageProfileComponent, decorators: [{
            type: Component,
            args: [{
                    template: '',
                }]
        }], ctorParameters: function () { return [{ type: i1.ManageProfilePreferencesService }, { type: i0.ChangeDetectorRef }]; }, propDecorators: { notificationDismissTime: [{
                type: Input
            }] } });
