// @backbase/ui-ang/esm2015/control-error-handler/components/validation-message/validation-message.component.js
import { Component } from '@angular/core';
import { errorMessageAnimation } from '@backbase/ui-ang/util';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
let _uniqueId = 0;
/**
 * @name ValidationMessageComponent
 *
 * @description
 * Dynamic component that displays a control validation message dependence on validation.
 */
export class ValidationMessageComponent {
    constructor() {
        this.errorId = `bb-error-message-${_uniqueId++}`;
    }
}
ValidationMessageComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ValidationMessageComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ValidationMessageComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ValidationMessageComponent, selector: "bb-validation-message-ui", ngImport: i0, template: "<small @errorMessageAnimation *ngIf=\"text\" [id]=\"errorId\" role=\"alert\" class=\"bb-input-validation-message\">\n  {{ text }}\n</small>\n", directives: [{ type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], animations: [errorMessageAnimation] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ValidationMessageComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-validation-message-ui',
                    templateUrl: 'validation-message.component.html',
                    animations: [errorMessageAnimation],
                }]
        }] });
