// @backbase/ui-ang/esm2015/_example/list-multiple-select/examples/list-multiple-select-example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/forms";
import * as i2 from "@backbase/ui-ang/list-multiple-select";
export class ListMultipleSelectExampleComponent {
    constructor(formBuilder) {
        this.formBuilder = formBuilder;
        this.listMultipleSelectItemsConfig = {
            i2: {
                icon: 'warning',
                iconColor: 'warning',
            },
            i3: {
                icon: 'warning',
                iconColor: 'secondary',
                items: {
                    q2: {
                        icon: 'warning',
                    },
                },
            },
        };
        this.vForm = this.formBuilder.group({
            listMultipleSelect: this.formBuilder.group({
                i1: true,
                i2: true,
                i3: this.formBuilder.group({
                    q1: false,
                    q2: true,
                    q3: this.formBuilder.group({
                        w1: { value: true, disabled: true },
                        w2: { value: false, disabled: true },
                    }),
                }),
            }),
        });
    }
    onListMultipleSelectClearAll($event) {
        $event.preventDefault();
        if (this.vForm) {
            this.vForm.controls['listMultipleSelect'].patchValue({
                i1: false,
                i2: false,
                i3: {
                    q1: false,
                    q2: false,
                },
            });
        }
    }
}
ListMultipleSelectExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ListMultipleSelectExampleComponent, deps: [{ token: i1.FormBuilder }], target: i0.ɵɵFactoryTarget.Component });
ListMultipleSelectExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ListMultipleSelectExampleComponent, selector: "bb-list-multiple-select-example-ui", ngImport: i0, template: "<form class=\"col-6\" [formGroup]=\"vForm\">\n  <bb-list-multiple-select-ui\n    formGroupName=\"listMultipleSelect\"\n    [config]=\"listMultipleSelectItemsConfig\"\n    aria-label=\"List multiple select\"\n  >\n    <ng-template bbListMultipleSelectAction>\n      <a href=\"\" (click)=\"onListMultipleSelectClearAll($event)\">Clear All</a>\n    </ng-template>\n  </bb-list-multiple-select-ui>\n</form>\n", components: [{ type: i2.ListMultipleSelectComponent, selector: "bb-list-multiple-select-ui", inputs: ["config"] }], directives: [{ type: i1.ɵNgNoValidate, selector: "form:not([ngNoForm]):not([ngNativeValidate])" }, { type: i1.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { type: i1.FormGroupDirective, selector: "[formGroup]", inputs: ["formGroup"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { type: i1.FormGroupName, selector: "[formGroupName]", inputs: ["formGroupName"] }, { type: i2.ListMultipleSelectActionDirective, selector: "ng-template[bbListMultipleSelectAction]" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ListMultipleSelectExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-list-multiple-select-example-ui',
                    templateUrl: './list-multiple-select-example.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.FormBuilder }]; } });
