// @backbase/contact-common-ang/esm2015/lib/constants.js
import { AccessContextScope } from './models/types';
export const Countries = [
    'AR',
    'AU',
    'AT',
    'BE',
    'BR',
    'CA',
    'CN',
    'CO',
    'HR',
    'CZ',
    'DK',
    'EC',
    'EG',
    'ET',
    'FR',
    'DE',
    'GR',
    'HN',
    'HK',
    'HU',
    'IN',
    'ID',
    'IE',
    'IT',
    'JM',
    'MY',
    'MX',
    'MA',
    'NA',
    'NL',
    'NZ',
    'NO',
    'PL',
    'PT',
    'RU',
    'SN',
    'SG',
    'ZA',
    'ES',
    'LK',
    'SE',
    'TH',
    'TR',
    'UG',
    'AE',
    'GB',
    'US',
    'ZW',
];
export const defaultAccountTypes = ['Savings', 'Checking'];
export const defaultAccessContextScope = AccessContextScope.USER;
export var ApprovalStatusColorMap;
(function (ApprovalStatusColorMap) {
    ApprovalStatusColorMap["OPEN"] = "success";
    ApprovalStatusColorMap["CLOSED"] = "danger";
    ApprovalStatusColorMap["INVALID"] = "danger";
    ApprovalStatusColorMap["VALID"] = "success";
    ApprovalStatusColorMap["ENTERED"] = "primary";
    ApprovalStatusColorMap["READY"] = "primary";
    ApprovalStatusColorMap["ACCEPTED"] = "warning";
    ApprovalStatusColorMap["REJECTED"] = "danger";
    ApprovalStatusColorMap["PENDING"] = "primary";
    ApprovalStatusColorMap["APPROVED"] = "success";
    ApprovalStatusColorMap["CANCELLED"] = "warning";
})(ApprovalStatusColorMap || (ApprovalStatusColorMap = {}));
export var ContactApprovalStatusMap;
(function (ContactApprovalStatusMap) {
    ContactApprovalStatusMap["CANCELLED"] = "CANCELLED";
    ContactApprovalStatusMap["REJECTED"] = "REJECTED";
    ContactApprovalStatusMap["PENDING"] = "ENTERED";
    ContactApprovalStatusMap["APPROVED"] = "APPROVED";
})(ContactApprovalStatusMap || (ContactApprovalStatusMap = {}));
export var Preferences;
(function (Preferences) {
    Preferences[Preferences["PAGE_SIZE"] = 20] = "PAGE_SIZE";
    Preferences[Preferences["MAX_NAV_SIZE"] = 3] = "MAX_NAV_SIZE";
    Preferences[Preferences["DEBOUNCE_TIME"] = 300] = "DEBOUNCE_TIME";
    Preferences["LOAD_MORE_METHOD"] = "pagination";
    Preferences[Preferences["NOTIFICATION_DISMISS_TIME"] = 5000] = "NOTIFICATION_DISMISS_TIME";
    Preferences["ACCESS_CONTEXT_SCOPE"] = "SA";
})(Preferences || (Preferences = {}));
export const IbanMask = 'SS00 AAAA AAAA AAAA AAAA AAAA AAAA AAAA AA';
