// @backbase/contact-common-ang/esm2015/lib/pipes/relative-date/relative-date.pipe.js
import { Pipe } from '@angular/core';
import { DatePipe } from '@angular/common';
import { getRelativeDate } from '../../helpers/date';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export var RelativeDate;
(function (RelativeDate) {
    RelativeDate["Yesterday"] = "Yesterday";
    RelativeDate["Today"] = "Today";
    RelativeDate["Tomorrow"] = "Tomorrow";
})(RelativeDate || (RelativeDate = {}));
/**
 * Pipe is used to show the date in a specific format
 *
 * @usageNotes
 *
 *
 * ```ts
 * @NgModule({
 *   ...
 *   imports: [
 *     ...
 *   ],
 *  declarations: RelativeDatePipe
 * })
 * export class MyContactCommonAngModule {}
 * ```
 *
 * ### Can be used in html
 *
 * ```html
 * {{todayDate | relativeDate: 'mediumTime'}}
 * ```
 * @ngModule RelativeDateModule
 */
export class RelativeDatePipe {
    constructor(datePipe) {
        this.datePipe = datePipe;
        this.getRelativeDateHelper = getRelativeDate;
    }
    transform(value, dateFormat) {
        let result;
        const yesterday = this.getRelativeDateHelper(-1);
        const today = this.getRelativeDateHelper();
        const tomorrow = this.getRelativeDateHelper(1);
        switch (new Date(value).toISOString().slice(0, 10)) {
            case yesterday:
                result = RelativeDate.Yesterday;
                break;
            case today:
                result = RelativeDate.Today;
                break;
            case tomorrow:
                result = RelativeDate.Tomorrow;
                break;
            default:
                result = this.datePipe.transform(value, dateFormat);
        }
        return result;
    }
}
/** @nocollapse */ RelativeDatePipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RelativeDatePipe, deps: [{ token: i1.DatePipe }], target: i0.ɵɵFactoryTarget.Pipe });
/** @nocollapse */ RelativeDatePipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RelativeDatePipe, name: "relativeDate" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: RelativeDatePipe, decorators: [{
            type: Pipe,
            args: [{ name: 'relativeDate' }]
        }], ctorParameters: function () { return [{ type: i1.DatePipe }]; } });
