// @backbase/payment-template-http-ang/esm2015/api/paymentTemplates.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { PAYMENT_TEMPLATE_BASE_PATH } from '../variables';
import { PaymentTemplateConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class PaymentTemplatesHttpService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new PaymentTemplateConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    deletePaymentTemplateIdByPaymentTemplateId(requestParameters, observe = 'body', reportProgress = false, options) {
        const _paymentTemplateId = requestParameters["paymentTemplateId"];
        if (_paymentTemplateId === null || _paymentTemplateId === undefined) {
            throw new Error('Required parameter paymentTemplateId was null or undefined when calling deletePaymentTemplateIdByPaymentTemplateId.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.delete(`${this.configuration.basePath}/client-api/v2/payment-templates/${encodeURIComponent(String(_paymentTemplateId))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    deletePaymentTemplateIdByPaymentTemplateIdUrl(requestParameters) {
        const _paymentTemplateId = requestParameters["paymentTemplateId"];
        if (_paymentTemplateId === null || _paymentTemplateId === undefined) {
            throw new Error('Required parameter paymentTemplateId was null or undefined when calling deletePaymentTemplateIdByPaymentTemplateId.');
        }
        return `${this.configuration.basePath}/client-api/v2/payment-templates/${encodeURIComponent(String(_paymentTemplateId))}`;
    }
    getPaymentTemplates(requestParameters, observe = 'body', reportProgress = false, options) {
        const _query = requestParameters["query"];
        const _intendedAction = requestParameters["intendedAction"];
        const _paymentType = requestParameters["paymentType"];
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_query !== undefined && _query !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _query, 'query');
        }
        if (_intendedAction !== undefined && _intendedAction !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _intendedAction, 'intendedAction');
        }
        if (_paymentType !== undefined && _paymentType !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _paymentType, 'paymentType');
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/payment-templates`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getPaymentTemplatesUrl(requestParameters) {
        const _query = requestParameters["query"];
        const _intendedAction = requestParameters["intendedAction"];
        const _paymentType = requestParameters["paymentType"];
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_query !== undefined && _query !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _query, 'query');
        }
        if (_intendedAction !== undefined && _intendedAction !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _intendedAction, 'intendedAction');
        }
        if (_paymentType !== undefined && _paymentType !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _paymentType, 'paymentType');
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/payment-templates${queryString ? `?${queryString}` : ''}`;
    }
    postPaymentTemplates(requestParameters, observe = 'body', reportProgress = false, options) {
        const _paymentTemplatesPostRequest = requestParameters["paymentTemplatesPostRequest"];
        if (_paymentTemplatesPostRequest === null || _paymentTemplatesPostRequest === undefined) {
            throw new Error('Required parameter paymentTemplatesPostRequest was null or undefined when calling postPaymentTemplates.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v2/payment-templates`, _paymentTemplatesPostRequest, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    postPaymentTemplatesUrl(requestParameters) {
        const _paymentTemplatesPostRequest = requestParameters["paymentTemplatesPostRequest"];
        if (_paymentTemplatesPostRequest === null || _paymentTemplatesPostRequest === undefined) {
            throw new Error('Required parameter paymentTemplatesPostRequest was null or undefined when calling postPaymentTemplates.');
        }
        return `${this.configuration.basePath}/client-api/v2/payment-templates`;
    }
    putPaymentTemplateById(requestParameters, observe = 'body', reportProgress = false, options) {
        const _paymentTemplateId = requestParameters["paymentTemplateId"];
        if (_paymentTemplateId === null || _paymentTemplateId === undefined) {
            throw new Error('Required parameter paymentTemplateId was null or undefined when calling putPaymentTemplateById.');
        }
        const _paymentTemplatesPutRequest = requestParameters["paymentTemplatesPutRequest"];
        if (_paymentTemplatesPutRequest === null || _paymentTemplatesPutRequest === undefined) {
            throw new Error('Required parameter paymentTemplatesPutRequest was null or undefined when calling putPaymentTemplateById.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v2/payment-templates/${encodeURIComponent(String(_paymentTemplateId))}`, _paymentTemplatesPutRequest, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    putPaymentTemplateByIdUrl(requestParameters) {
        const _paymentTemplateId = requestParameters["paymentTemplateId"];
        if (_paymentTemplateId === null || _paymentTemplateId === undefined) {
            throw new Error('Required parameter paymentTemplateId was null or undefined when calling putPaymentTemplateById.');
        }
        const _paymentTemplatesPutRequest = requestParameters["paymentTemplatesPutRequest"];
        if (_paymentTemplatesPutRequest === null || _paymentTemplatesPutRequest === undefined) {
            throw new Error('Required parameter paymentTemplatesPutRequest was null or undefined when calling putPaymentTemplateById.');
        }
        return `${this.configuration.basePath}/client-api/v2/payment-templates/${encodeURIComponent(String(_paymentTemplateId))}`;
    }
}
PaymentTemplatesHttpService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentTemplatesHttpService, deps: [{ token: i1.HttpClient }, { token: PAYMENT_TEMPLATE_BASE_PATH, optional: true }, { token: i2.PaymentTemplateConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
PaymentTemplatesHttpService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentTemplatesHttpService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PaymentTemplatesHttpService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [PAYMENT_TEMPLATE_BASE_PATH]
                }] }, { type: i2.PaymentTemplateConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
