// @backbase/places-journey-ang/esm2015/lib/places-journey.component.js
import { Component } from '@angular/core';
import { PlacesJourneyConfigService } from './services/places-journey-config.service';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
export class PlacesJourneyComponent {
}
PlacesJourneyComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PlacesJourneyComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
PlacesJourneyComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: PlacesJourneyComponent, selector: "bb-places-journey", providers: [PlacesJourneyConfigService], ngImport: i0, template: '<router-outlet></router-outlet>', isInline: true, directives: [{ type: i1.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PlacesJourneyComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-places-journey',
                    template: '<router-outlet></router-outlet>',
                    providers: [PlacesJourneyConfigService],
                }]
        }] });
