// @backbase/places-journey-ang/esm2015/lib/places-journey.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { provideRoutes, RouterModule } from '@angular/router';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { PlacesJourneyComponent } from './places-journey.component';
import { PlacesViewComponent } from './views/places-view.component';
import { PlacesListComponent } from './components/places-list/places-list.component';
import { MapWrapperComponent } from './components/map-wrapper/map-wrapper.component';
import { PlaceDetailsComponent } from './components/place-details/place-details.component';
import { WorkTimesComponent } from './components/work-times/work-times.component';
import { KeyboardAccessibilityDirective } from './directives/keyboard-accessibility.directive';
import { MapModule } from './components/bb-map-ui/map.module';
import { FormsModule } from '@angular/forms';
import { PlacesService } from './services/places.service';
import { HeaderModule } from '@backbase/ui-ang/header';
import { EmptyStateModule } from '@backbase/ui-ang/empty-state';
import { ErrorCommonStateModule } from '@backbase/ui-ang/common-error-state';
import { LoadingIndicatorModule } from '@backbase/ui-ang/loading-indicator';
import { IconModule } from '@backbase/ui-ang/icon';
import { ButtonModule } from '@backbase/ui-ang/button';
import { CollapsibleModule } from '@backbase/ui-ang/collapsible';
import { CollapsibleAccordionModule } from '@backbase/ui-ang/collapsible-accordion';
import { InputCheckboxModule } from '@backbase/ui-ang/input-checkbox';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
export const defaultRoute = {
    path: '',
    component: PlacesJourneyComponent,
    children: [
        {
            path: '',
            component: PlacesViewComponent,
            data: { title: 'Places' },
        },
    ],
};
const components = [
    PlacesListComponent,
    MapWrapperComponent,
    PlaceDetailsComponent,
    WorkTimesComponent,
    KeyboardAccessibilityDirective,
];
const modules = [
    HeaderModule,
    EmptyStateModule,
    ErrorCommonStateModule,
    LoadingIndicatorModule,
    IconModule,
    ButtonModule,
    MapModule,
    CollapsibleModule,
    CollapsibleAccordionModule,
    InputCheckboxModule,
];
export class PlacesJourneyModule {
    static forRoot(data = { route: defaultRoute }) {
        return {
            ngModule: PlacesJourneyModule,
            providers: [provideRoutes([data.route])],
        };
    }
}
PlacesJourneyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PlacesJourneyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PlacesJourneyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PlacesJourneyModule, declarations: [PlacesJourneyComponent, PlacesViewComponent, PlacesListComponent,
        MapWrapperComponent,
        PlaceDetailsComponent,
        WorkTimesComponent,
        KeyboardAccessibilityDirective], imports: [CommonModule,
        FormsModule, i1.BackbaseCoreModule, RouterModule, HeaderModule,
        EmptyStateModule,
        ErrorCommonStateModule,
        LoadingIndicatorModule,
        IconModule,
        ButtonModule,
        MapModule,
        CollapsibleModule,
        CollapsibleAccordionModule,
        InputCheckboxModule] });
PlacesJourneyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PlacesJourneyModule, providers: [PlacesService], imports: [[
            CommonModule,
            FormsModule,
            BackbaseCoreModule.withConfig({
                classMap: { PlacesJourneyComponent },
            }),
            RouterModule,
            ...modules,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PlacesJourneyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [PlacesJourneyComponent, PlacesViewComponent, ...components],
                    imports: [
                        CommonModule,
                        FormsModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { PlacesJourneyComponent },
                        }),
                        RouterModule,
                        ...modules,
                    ],
                    providers: [PlacesService],
                }]
        }] });
