// @backbase/account-statements-http-ang/fesm2015/backbase-account-statements-http-ang.js
import * as i0 from '@angular/core';
import { InjectionToken, NgModule, Optional, SkipSelf, Injectable, Inject } from '@angular/core';
import * as i1 from '@angular/common/http';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import * as i2 from '@backbase/foundation-ang/data-http';
import { createMocks } from '@backbase/foundation-ang/data-http';

var AccountStatementsPostRequest;
(function (AccountStatementsPostRequest) {
    AccountStatementsPostRequest.DirectionEnum = {
        ASC: 'ASC',
        DESC: 'DESC'
    };
})(AccountStatementsPostRequest || (AccountStatementsPostRequest = {}));

const ACCOUNT_STATEMENT_BASE_PATH = new InjectionToken('ACCOUNT_STATEMENT_BASE_PATH');
const COLLECTION_FORMATS = {
    'csv': ',',
    'tsv': '   ',
    'ssv': ' ',
    'pipes': '|'
};

class AccountStatementConfiguration {
    constructor(configurationParameters = {}) {
        this.apiKeys = configurationParameters.apiKeys;
        this.username = configurationParameters.username;
        this.password = configurationParameters.password;
        this.accessToken = configurationParameters.accessToken;
        this.basePath = configurationParameters.basePath;
        this.withCredentials = configurationParameters.withCredentials;
        this.encoder = configurationParameters.encoder;
        if (configurationParameters.credentials) {
            this.credentials = configurationParameters.credentials;
        }
        else {
            this.credentials = {};
        }
    }
    selectHeaderContentType(contentTypes) {
        if (contentTypes.length === 0) {
            return undefined;
        }
        const type = contentTypes.find((x) => this.isJsonMime(x));
        if (type === undefined) {
            return contentTypes[0];
        }
        return type;
    }
    selectHeaderAccept(accepts) {
        if (accepts.length === 0) {
            return undefined;
        }
        const type = accepts.find((x) => this.isJsonMime(x));
        if (type === undefined) {
            return accepts[0];
        }
        return type;
    }
    isJsonMime(mime) {
        const jsonMime = new RegExp('^(application\/json|[^;/ \t]+\/[^;/ \t]+[+]json)[ \t]*(;.*)?$', 'i');
        return mime !== null && (jsonMime.test(mime) || mime.toLowerCase() === 'application/json-patch+json');
    }
    lookupCredential(key) {
        const value = this.credentials[key];
        return typeof value === 'function'
            ? value()
            : value;
    }
}
const CONFIG_TOKEN = new InjectionToken('api-data-module AccountStatementConfiguration');

class AccountStatementApiModule {
    constructor(parentModule, http, dataModulesManager, config) {
        if (parentModule) {
            throw new Error('AccountStatementApiModule is already loaded. Import in your base AppModule only.');
        }
        if (!http) {
            throw new Error('You need to import the HttpClientModule in your AppModule! \n' +
                'See also https://github.com/angular/angular/issues/20575');
        }
        if (dataModulesManager) {
            dataModulesManager.setModuleConfig(CONFIG_TOKEN, {
                apiRoot: '',
                servicePath: config.basePath || '',
                headers: {},
            });
        }
    }
    static forRoot(configurationFactory) {
        return {
            ngModule: AccountStatementApiModule,
            providers: [{ provide: AccountStatementConfiguration, useFactory: configurationFactory }]
        };
    }
}
AccountStatementApiModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementApiModule, deps: [{ token: AccountStatementApiModule, optional: true, skipSelf: true }, { token: i1.HttpClient, optional: true }, { token: i2.DataModulesManager, optional: true }, { token: AccountStatementConfiguration }], target: i0.ɵɵFactoryTarget.NgModule });
AccountStatementApiModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementApiModule });
AccountStatementApiModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementApiModule, providers: [], imports: [[]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementApiModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [],
                    declarations: [],
                    exports: [],
                    providers: []
                }]
        }], ctorParameters: function () { return [{ type: AccountStatementApiModule, decorators: [{
                    type: Optional
                }, {
                    type: SkipSelf
                }] }, { type: i1.HttpClient, decorators: [{
                    type: Optional
                }] }, { type: i2.DataModulesManager, decorators: [{
                    type: Optional
                }] }, { type: AccountStatementConfiguration }]; } });

class CustomHttpParameterCodec {
    encodeKey(k) {
        return encodeURIComponent(k);
    }
    encodeValue(v) {
        return encodeURIComponent(v);
    }
    decodeKey(k) {
        return decodeURIComponent(k);
    }
    decodeValue(v) {
        return decodeURIComponent(v);
    }
}

class AccountStatementHttpService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new AccountStatementConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    downloadAccountStatements(requestParameters, observe = 'body', reportProgress = false, options) {
        const _uid = requestParameters["uid"];
        if (_uid === null || _uid === undefined) {
            throw new Error('Required parameter uid was null or undefined when calling downloadAccountStatements.');
        }
        const _inline = requestParameters["inline"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_inline !== undefined && _inline !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _inline, 'inline');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'binary/octet-stream',
                'application/pdf',
                'text/plain',
                'text/csv',
                'application/rtf',
                'application/msword',
                'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
                'application/vnd.ms-excel',
                'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
                'application/xml',
                'image/png',
                'image/jpeg',
                'image/gif',
                'image/tiff',
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/account/statements/download/${encodeURIComponent(String(_uid))}`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    downloadAccountStatementsUrl(requestParameters) {
        const _uid = requestParameters["uid"];
        if (_uid === null || _uid === undefined) {
            throw new Error('Required parameter uid was null or undefined when calling downloadAccountStatements.');
        }
        const _inline = requestParameters["inline"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_inline !== undefined && _inline !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _inline, 'inline');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/account/statements/download/${encodeURIComponent(String(_uid))}${queryString ? `?${queryString}` : ''}`;
    }
    getAccountStatements(requestParameters, observe = 'body', reportProgress = false, options) {
        const _accountId = requestParameters["accountId"];
        const _dateFrom = requestParameters["dateFrom"];
        const _dateTo = requestParameters["dateTo"];
        const _category = requestParameters["category"];
        const _from = requestParameters["from"];
        const _cursor = requestParameters["cursor"];
        const _size = requestParameters["size"];
        const _orderBy = requestParameters["orderBy"];
        const _direction = requestParameters["direction"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_accountId !== undefined && _accountId !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _accountId, 'accountId');
        }
        if (_dateFrom !== undefined && _dateFrom !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _dateFrom, 'dateFrom');
        }
        if (_dateTo !== undefined && _dateTo !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _dateTo, 'dateTo');
        }
        if (_category) {
            _category.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'category');
            });
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_cursor !== undefined && _cursor !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _cursor, 'cursor');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_orderBy !== undefined && _orderBy !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _orderBy, 'orderBy');
        }
        if (_direction !== undefined && _direction !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _direction, 'direction');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/account/statements`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getAccountStatementsUrl(requestParameters) {
        const _accountId = requestParameters["accountId"];
        const _dateFrom = requestParameters["dateFrom"];
        const _dateTo = requestParameters["dateTo"];
        const _category = requestParameters["category"];
        const _from = requestParameters["from"];
        const _cursor = requestParameters["cursor"];
        const _size = requestParameters["size"];
        const _orderBy = requestParameters["orderBy"];
        const _direction = requestParameters["direction"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_accountId !== undefined && _accountId !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _accountId, 'accountId');
        }
        if (_dateFrom !== undefined && _dateFrom !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _dateFrom, 'dateFrom');
        }
        if (_dateTo !== undefined && _dateTo !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _dateTo, 'dateTo');
        }
        if (_category) {
            _category.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'category');
            });
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_cursor !== undefined && _cursor !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _cursor, 'cursor');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_orderBy !== undefined && _orderBy !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _orderBy, 'orderBy');
        }
        if (_direction !== undefined && _direction !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _direction, 'direction');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v2/account/statements${queryString ? `?${queryString}` : ''}`;
    }
    getCategories(observe = 'body', reportProgress = false, options) {
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v2/account/statements/categories`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getCategoriesUrl() {
        return `${this.configuration.basePath}/client-api/v2/account/statements/categories`;
    }
    postAccountStatements(requestParameters, observe = 'body', reportProgress = false, options) {
        const _accountStatementsPostRequest = requestParameters["accountStatementsPostRequest"];
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v2/account/statements`, _accountStatementsPostRequest, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    postAccountStatementsUrl(requestParameters) {
        const _accountStatementsPostRequest = requestParameters["accountStatementsPostRequest"];
        return `${this.configuration.basePath}/client-api/v2/account/statements`;
    }
}
AccountStatementHttpService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementHttpService, deps: [{ token: i1.HttpClient }, { token: ACCOUNT_STATEMENT_BASE_PATH, optional: true }, { token: AccountStatementConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
AccountStatementHttpService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementHttpService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AccountStatementHttpService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [ACCOUNT_STATEMENT_BASE_PATH]
                }] }, { type: AccountStatementConfiguration, decorators: [{
                    type: Optional
                }] }]; } });

const AccountStatementHttpServiceDownloadAccountStatementsMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/account/statements/download/{uid}",
        method: "GET",
        responses: [
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
const AccountStatementHttpServiceGetAccountStatementsMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/account/statements",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-02-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - February 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_1",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-03-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - March 2016",
                        "documents": [{
                                "uid": "uid_2",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-04-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - April 2016",
                        "documents": [{
                                "uid": "uid_3",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-05-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - May 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_4",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-06-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - June 2016",
                        "documents": [{
                                "uid": "uid_5",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-07-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - July 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_6",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-08-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - August 2016",
                        "documents": [{
                                "uid": "uid_7",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-09-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - September 2016",
                        "documents": [{
                                "uid": "uid_8",
                                "url": "",
                                "contentType": ""
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-01-31",
                        "category": "Account",
                        "description": "Account statement - January 2017",
                        "documents": [{
                                "uid": "uid_9_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_9_2",
                                "contentType": "text/csv"
                            }, {
                                "uid": "uid_9_3",
                                "url": "http://server.com/documentsdepo/uid_9_3",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-02-28",
                        "category": "Account",
                        "description": "Account statement - February 2017",
                        "documents": [{
                                "uid": "uid_10_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_10_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-03-31",
                        "category": "Account",
                        "description": "Account statement - March 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_11_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_11_2",
                                "url": "http://server.com/documentsdepo/uid_11_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-04-30",
                        "category": "Account",
                        "description": "Account statement - April 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_12_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_12_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-05-31",
                        "category": "Account",
                        "description": "Account statement - May 2017",
                        "documents": [{
                                "uid": "uid_13_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_13_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-06-30",
                        "category": "Account",
                        "description": "Account statement - June 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_14_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_14_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-07-31",
                        "category": "Account",
                        "description": "Account statement - July 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_15_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_15_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-08-31",
                        "category": "Account",
                        "description": "Account statement - August 2017",
                        "documents": [{
                                "uid": "uid_16_1",
                                "url": "http://server.com/documentsdepo/uid_16_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_16_2",
                                "url": "http://server.com/documentsdepo/uid_16_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-09-30",
                        "description": "Statement - September 2017",
                        "documents": [{
                                "uid": "uid_17_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-10-31",
                        "category": "Account",
                        "description": "Account statement - October 2017",
                        "documents": [{
                                "uid": "uid_18_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_18_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-11-30",
                        "category": "Account",
                        "description": "Account statement - November 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_19_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_19_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-12-31",
                        "category": "Account",
                        "description": "Account statement - December 2017",
                        "documents": [{
                                "uid": "uid_20_1",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-05-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - May 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_21",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-06-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - June 2017",
                        "documents": [{
                                "uid": "uid_22",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-07-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - July 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_23",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-08-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - August 2017",
                        "documents": [{
                                "uid": "uid_24",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-09-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - September 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_25",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-10-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - October 2017",
                        "documents": [{
                                "uid": "uid_26",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-11-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - November 2017",
                        "documents": [{
                                "uid": "uid_27",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-12-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - December 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_28",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2018-01-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - January 2018",
                        "documents": [{
                                "uid": "uid_29",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-12-31",
                        "category": "Tax Forms",
                        "description": "Tax form - 2017",
                        "documents": [{
                                "uid": "uid_30",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-02-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - February 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_31",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-03-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - March 2016",
                        "documents": [{
                                "uid": "uid_32",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-04-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - April 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_33",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-05-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - May 2016",
                        "documents": [{
                                "uid": "uid_34",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-06-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - June 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_35",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-07-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - July 2016",
                        "documents": [{
                                "uid": "uid_36",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-08-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - August 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_37",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-09-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - September 2016",
                        "documents": [{
                                "uid": "uid_38",
                                "url": "",
                                "contentType": ""
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-01-31",
                        "category": "Account",
                        "description": "Account statement - January 2017",
                        "documents": [{
                                "uid": "",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_39_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-02-28",
                        "category": "Account",
                        "description": "Account statement - February 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_40_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-03-31",
                        "category": "Account",
                        "description": "Account statement - March 2017",
                        "documents": [{
                                "uid": "uid_41_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_41_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-04-30",
                        "category": "Account",
                        "description": "Account statement - April 2017",
                        "documents": [{
                                "uid": "uid_42_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_42_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-05-31",
                        "category": "Account",
                        "description": "Account statement - May 2017",
                        "documents": [{
                                "uid": "uid_43_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_43_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-06-30",
                        "category": "Account",
                        "description": "Account statement - June 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_44_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_44_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-07-31",
                        "category": "Account",
                        "description": "Account statement - July 2017",
                        "documents": [{
                                "uid": "uid_45_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_45_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-08-31",
                        "category": "Account",
                        "description": "Account statement - August 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_46_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_46_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-09-30",
                        "description": "Statement - September 2017",
                        "documents": [{
                                "uid": "uid_47_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-10-31",
                        "category": "Account",
                        "description": "Account statement - October 2017",
                        "documents": [{
                                "uid": "uid_48_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_48_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-11-30",
                        "category": "Account",
                        "description": "Account statement - November 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_49_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_49_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-12-31",
                        "category": "Account",
                        "description": "Account statement - December 2017",
                        "documents": [{
                                "uid": "uid_50_1",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-05-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - May 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_51",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-06-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - June 2017",
                        "documents": [{
                                "uid": "uid_52",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-06-20",
                        "category": "Mortgage",
                        "description": "Mortgage statement - July 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_53",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-08-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - August 2017",
                        "documents": [{
                                "uid": "uid_54",
                                "contentType": "application/pdf"
                            }]
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
const AccountStatementHttpServiceGetCategoriesMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/account/statements/categories",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "categories": ["Account", "Credit Card", "Mortgage", "Tax Form", "LOC", "EELOC"],
                    "additions": {
                        "categoryId": "1"
                    }
                }
            },
        ]
    }]);
const AccountStatementHttpServicePostAccountStatementsMocksProvider = createMocks([{
        urlPattern: "/client-api/v2/account/statements",
        method: "POST",
        responses: [
            {
                status: 200,
                body: [{
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-02-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - February 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_1",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-03-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - March 2016",
                        "documents": [{
                                "uid": "uid_2",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-04-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - April 2016",
                        "documents": [{
                                "uid": "uid_3",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-05-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - May 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_4",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-06-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - June 2016",
                        "documents": [{
                                "uid": "uid_5",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-07-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - July 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_6",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-08-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - August 2016",
                        "documents": [{
                                "uid": "uid_7",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-09-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - September 2016",
                        "documents": [{
                                "uid": "uid_8",
                                "url": "",
                                "contentType": ""
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-01-31",
                        "category": "Account",
                        "description": "Account statement - January 2017",
                        "documents": [{
                                "uid": "uid_9_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_9_2",
                                "contentType": "text/csv"
                            }, {
                                "uid": "uid_9_3",
                                "url": "http://server.com/documentsdepo/uid_9_3",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-02-28",
                        "category": "Account",
                        "description": "Account statement - February 2017",
                        "documents": [{
                                "uid": "uid_10_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_10_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-03-31",
                        "category": "Account",
                        "description": "Account statement - March 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_11_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_11_2",
                                "url": "http://server.com/documentsdepo/uid_11_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-04-30",
                        "category": "Account",
                        "description": "Account statement - April 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_12_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_12_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-05-31",
                        "category": "Account",
                        "description": "Account statement - May 2017",
                        "documents": [{
                                "uid": "uid_13_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_13_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-06-30",
                        "category": "Account",
                        "description": "Account statement - June 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_14_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_14_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-07-31",
                        "category": "Account",
                        "description": "Account statement - July 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_15_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_15_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-08-31",
                        "category": "Account",
                        "description": "Account statement - August 2017",
                        "documents": [{
                                "uid": "uid_16_1",
                                "url": "http://server.com/documentsdepo/uid_16_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_16_2",
                                "url": "http://server.com/documentsdepo/uid_16_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-09-30",
                        "description": "Statement - September 2017",
                        "documents": [{
                                "uid": "uid_17_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-10-31",
                        "category": "Account",
                        "description": "Account statement - October 2017",
                        "documents": [{
                                "uid": "uid_18_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_18_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-11-30",
                        "category": "Account",
                        "description": "Account statement - November 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_19_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_19_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-12-31",
                        "category": "Account",
                        "description": "Account statement - December 2017",
                        "documents": [{
                                "uid": "uid_20_1",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-05-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - May 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_21",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-06-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - June 2017",
                        "documents": [{
                                "uid": "uid_22",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-07-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - July 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_23",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-08-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - August 2017",
                        "documents": [{
                                "uid": "uid_24",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-09-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - September 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_25",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-10-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - October 2017",
                        "documents": [{
                                "uid": "uid_26",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-11-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - November 2017",
                        "documents": [{
                                "uid": "uid_27",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-12-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - December 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_28",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2018-01-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - January 2018",
                        "documents": [{
                                "uid": "uid_29",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-12-31",
                        "category": "Tax Forms",
                        "description": "Tax form - 2017",
                        "documents": [{
                                "uid": "uid_30",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-02-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - February 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_31",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-03-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - March 2016",
                        "documents": [{
                                "uid": "uid_32",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-04-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - April 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_33",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-05-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - May 2016",
                        "documents": [{
                                "uid": "uid_34",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-06-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - June 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_35",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-07-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - July 2016",
                        "documents": [{
                                "uid": "uid_36",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-08-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - August 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_37",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-09-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - September 2016",
                        "documents": [{
                                "uid": "uid_38",
                                "url": "",
                                "contentType": ""
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-01-31",
                        "category": "Account",
                        "description": "Account statement - January 2017",
                        "documents": [{
                                "uid": "",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_39_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-02-28",
                        "category": "Account",
                        "description": "Account statement - February 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_40_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-03-31",
                        "category": "Account",
                        "description": "Account statement - March 2017",
                        "documents": [{
                                "uid": "uid_41_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_41_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-04-30",
                        "category": "Account",
                        "description": "Account statement - April 2017",
                        "documents": [{
                                "uid": "uid_42_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_42_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-05-31",
                        "category": "Account",
                        "description": "Account statement - May 2017",
                        "documents": [{
                                "uid": "uid_43_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_43_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-06-30",
                        "category": "Account",
                        "description": "Account statement - June 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_44_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_44_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-07-31",
                        "category": "Account",
                        "description": "Account statement - July 2017",
                        "documents": [{
                                "uid": "uid_45_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_45_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-08-31",
                        "category": "Account",
                        "description": "Account statement - August 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_46_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_46_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-09-30",
                        "description": "Statement - September 2017",
                        "documents": [{
                                "uid": "uid_47_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-10-31",
                        "category": "Account",
                        "description": "Account statement - October 2017",
                        "documents": [{
                                "uid": "uid_48_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_48_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-11-30",
                        "category": "Account",
                        "description": "Account statement - November 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_49_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_49_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-12-31",
                        "category": "Account",
                        "description": "Account statement - December 2017",
                        "documents": [{
                                "uid": "uid_50_1",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-05-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - May 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_51",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-06-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - June 2017",
                        "documents": [{
                                "uid": "uid_52",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-06-20",
                        "category": "Mortgage",
                        "description": "Mortgage statement - July 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_53",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-08-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - August 2017",
                        "documents": [{
                                "uid": "uid_54",
                                "contentType": "application/pdf"
                            }]
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
const AccountStatementHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v2/account/statements/download/{uid}",
        method: "GET",
        responses: [
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/account/statements",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-02-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - February 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_1",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-03-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - March 2016",
                        "documents": [{
                                "uid": "uid_2",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-04-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - April 2016",
                        "documents": [{
                                "uid": "uid_3",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-05-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - May 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_4",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-06-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - June 2016",
                        "documents": [{
                                "uid": "uid_5",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-07-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - July 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_6",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-08-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - August 2016",
                        "documents": [{
                                "uid": "uid_7",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-09-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - September 2016",
                        "documents": [{
                                "uid": "uid_8",
                                "url": "",
                                "contentType": ""
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-01-31",
                        "category": "Account",
                        "description": "Account statement - January 2017",
                        "documents": [{
                                "uid": "uid_9_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_9_2",
                                "contentType": "text/csv"
                            }, {
                                "uid": "uid_9_3",
                                "url": "http://server.com/documentsdepo/uid_9_3",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-02-28",
                        "category": "Account",
                        "description": "Account statement - February 2017",
                        "documents": [{
                                "uid": "uid_10_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_10_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-03-31",
                        "category": "Account",
                        "description": "Account statement - March 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_11_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_11_2",
                                "url": "http://server.com/documentsdepo/uid_11_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-04-30",
                        "category": "Account",
                        "description": "Account statement - April 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_12_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_12_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-05-31",
                        "category": "Account",
                        "description": "Account statement - May 2017",
                        "documents": [{
                                "uid": "uid_13_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_13_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-06-30",
                        "category": "Account",
                        "description": "Account statement - June 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_14_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_14_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-07-31",
                        "category": "Account",
                        "description": "Account statement - July 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_15_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_15_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-08-31",
                        "category": "Account",
                        "description": "Account statement - August 2017",
                        "documents": [{
                                "uid": "uid_16_1",
                                "url": "http://server.com/documentsdepo/uid_16_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_16_2",
                                "url": "http://server.com/documentsdepo/uid_16_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-09-30",
                        "description": "Statement - September 2017",
                        "documents": [{
                                "uid": "uid_17_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-10-31",
                        "category": "Account",
                        "description": "Account statement - October 2017",
                        "documents": [{
                                "uid": "uid_18_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_18_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-11-30",
                        "category": "Account",
                        "description": "Account statement - November 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_19_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_19_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-12-31",
                        "category": "Account",
                        "description": "Account statement - December 2017",
                        "documents": [{
                                "uid": "uid_20_1",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-05-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - May 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_21",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-06-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - June 2017",
                        "documents": [{
                                "uid": "uid_22",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-07-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - July 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_23",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-08-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - August 2017",
                        "documents": [{
                                "uid": "uid_24",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-09-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - September 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_25",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-10-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - October 2017",
                        "documents": [{
                                "uid": "uid_26",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-11-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - November 2017",
                        "documents": [{
                                "uid": "uid_27",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-12-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - December 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_28",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2018-01-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - January 2018",
                        "documents": [{
                                "uid": "uid_29",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-12-31",
                        "category": "Tax Forms",
                        "description": "Tax form - 2017",
                        "documents": [{
                                "uid": "uid_30",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-02-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - February 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_31",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-03-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - March 2016",
                        "documents": [{
                                "uid": "uid_32",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-04-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - April 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_33",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-05-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - May 2016",
                        "documents": [{
                                "uid": "uid_34",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-06-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - June 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_35",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-07-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - July 2016",
                        "documents": [{
                                "uid": "uid_36",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-08-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - August 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_37",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-09-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - September 2016",
                        "documents": [{
                                "uid": "uid_38",
                                "url": "",
                                "contentType": ""
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-01-31",
                        "category": "Account",
                        "description": "Account statement - January 2017",
                        "documents": [{
                                "uid": "",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_39_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-02-28",
                        "category": "Account",
                        "description": "Account statement - February 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_40_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-03-31",
                        "category": "Account",
                        "description": "Account statement - March 2017",
                        "documents": [{
                                "uid": "uid_41_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_41_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-04-30",
                        "category": "Account",
                        "description": "Account statement - April 2017",
                        "documents": [{
                                "uid": "uid_42_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_42_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-05-31",
                        "category": "Account",
                        "description": "Account statement - May 2017",
                        "documents": [{
                                "uid": "uid_43_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_43_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-06-30",
                        "category": "Account",
                        "description": "Account statement - June 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_44_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_44_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-07-31",
                        "category": "Account",
                        "description": "Account statement - July 2017",
                        "documents": [{
                                "uid": "uid_45_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_45_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-08-31",
                        "category": "Account",
                        "description": "Account statement - August 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_46_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_46_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-09-30",
                        "description": "Statement - September 2017",
                        "documents": [{
                                "uid": "uid_47_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-10-31",
                        "category": "Account",
                        "description": "Account statement - October 2017",
                        "documents": [{
                                "uid": "uid_48_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_48_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-11-30",
                        "category": "Account",
                        "description": "Account statement - November 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_49_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_49_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-12-31",
                        "category": "Account",
                        "description": "Account statement - December 2017",
                        "documents": [{
                                "uid": "uid_50_1",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-05-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - May 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_51",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-06-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - June 2017",
                        "documents": [{
                                "uid": "uid_52",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-06-20",
                        "category": "Mortgage",
                        "description": "Mortgage statement - July 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_53",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-08-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - August 2017",
                        "documents": [{
                                "uid": "uid_54",
                                "contentType": "application/pdf"
                            }]
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/account/statements/categories",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "categories": ["Account", "Credit Card", "Mortgage", "Tax Form", "LOC", "EELOC"],
                    "additions": {
                        "categoryId": "1"
                    }
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v2/account/statements",
        method: "POST",
        responses: [
            {
                status: 200,
                body: [{
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-02-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - February 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_1",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-03-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - March 2016",
                        "documents": [{
                                "uid": "uid_2",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-04-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - April 2016",
                        "documents": [{
                                "uid": "uid_3",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-05-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - May 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_4",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-06-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - June 2016",
                        "documents": [{
                                "uid": "uid_5",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-07-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - July 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_6",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-08-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - August 2016",
                        "documents": [{
                                "uid": "uid_7",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-09-15",
                        "category": "Credit Card",
                        "description": "Credit Card statement - September 2016",
                        "documents": [{
                                "uid": "uid_8",
                                "url": "",
                                "contentType": ""
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-01-31",
                        "category": "Account",
                        "description": "Account statement - January 2017",
                        "documents": [{
                                "uid": "uid_9_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_9_2",
                                "contentType": "text/csv"
                            }, {
                                "uid": "uid_9_3",
                                "url": "http://server.com/documentsdepo/uid_9_3",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-02-28",
                        "category": "Account",
                        "description": "Account statement - February 2017",
                        "documents": [{
                                "uid": "uid_10_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_10_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-03-31",
                        "category": "Account",
                        "description": "Account statement - March 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_11_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_11_2",
                                "url": "http://server.com/documentsdepo/uid_11_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-04-30",
                        "category": "Account",
                        "description": "Account statement - April 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_12_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_12_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-05-31",
                        "category": "Account",
                        "description": "Account statement - May 2017",
                        "documents": [{
                                "uid": "uid_13_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_13_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-06-30",
                        "category": "Account",
                        "description": "Account statement - June 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_14_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_14_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-07-31",
                        "category": "Account",
                        "description": "Account statement - July 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_15_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_15_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-08-31",
                        "category": "Account",
                        "description": "Account statement - August 2017",
                        "documents": [{
                                "uid": "uid_16_1",
                                "url": "http://server.com/documentsdepo/uid_16_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_16_2",
                                "url": "http://server.com/documentsdepo/uid_16_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-09-30",
                        "description": "Statement - September 2017",
                        "documents": [{
                                "uid": "uid_17_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-10-31",
                        "category": "Account",
                        "description": "Account statement - October 2017",
                        "documents": [{
                                "uid": "uid_18_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_18_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-11-30",
                        "category": "Account",
                        "description": "Account statement - November 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_19_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_19_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-12-31",
                        "category": "Account",
                        "description": "Account statement - December 2017",
                        "documents": [{
                                "uid": "uid_20_1",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-05-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - May 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_21",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-06-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - June 2017",
                        "documents": [{
                                "uid": "uid_22",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-07-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - July 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_23",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-08-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - August 2017",
                        "documents": [{
                                "uid": "uid_24",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-09-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - September 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_25",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-10-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - October 2017",
                        "documents": [{
                                "uid": "uid_26",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-11-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - November 2017",
                        "documents": [{
                                "uid": "uid_27",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-12-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - December 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_28",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2018-01-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - January 2018",
                        "documents": [{
                                "uid": "uid_29",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-12-31",
                        "category": "Tax Forms",
                        "description": "Tax form - 2017",
                        "documents": [{
                                "uid": "uid_30",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-02-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - February 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_31",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-03-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - March 2016",
                        "documents": [{
                                "uid": "uid_32",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-04-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - April 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_33",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-05-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - May 2016",
                        "documents": [{
                                "uid": "uid_34",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-06-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - June 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_35",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-07-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - July 2016",
                        "documents": [{
                                "uid": "uid_36",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-08-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - August 2016",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_37",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2016-09-09",
                        "category": "Credit Card",
                        "description": "Credit Card statement - September 2016",
                        "documents": [{
                                "uid": "uid_38",
                                "url": "",
                                "contentType": ""
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-01-31",
                        "category": "Account",
                        "description": "Account statement - January 2017",
                        "documents": [{
                                "uid": "",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_39_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-02-28",
                        "category": "Account",
                        "description": "Account statement - February 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_40_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-03-31",
                        "category": "Account",
                        "description": "Account statement - March 2017",
                        "documents": [{
                                "uid": "uid_41_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_41_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-04-30",
                        "category": "Account",
                        "description": "Account statement - April 2017",
                        "documents": [{
                                "uid": "uid_42_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_42_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-05-31",
                        "category": "Account",
                        "description": "Account statement - May 2017",
                        "documents": [{
                                "uid": "uid_43_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_43_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-06-30",
                        "category": "Account",
                        "description": "Account statement - June 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_44_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_44_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-07-31",
                        "category": "Account",
                        "description": "Account statement - July 2017",
                        "documents": [{
                                "uid": "uid_45_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_45_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-08-31",
                        "category": "Account",
                        "description": "Account statement - August 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_46_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_46_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-09-30",
                        "description": "Statement - September 2017",
                        "documents": [{
                                "uid": "uid_47_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-10-31",
                        "category": "Account",
                        "description": "Account statement - October 2017",
                        "documents": [{
                                "uid": "uid_48_1",
                                "contentType": "application/pdf"
                            }, {
                                "uid": "uid_48_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-11-30",
                        "category": "Account",
                        "description": "Account statement - November 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_49_1",
                                "contentType": "application/pdf"
                            }, {
                                "url": "http://server.com/documentsdepo/uid_49_2",
                                "contentType": "text/csv"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-12-31",
                        "category": "Account",
                        "description": "Account statement - December 2017",
                        "documents": [{
                                "uid": "uid_50_1",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-05-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - May 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_51",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-06-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - June 2017",
                        "documents": [{
                                "uid": "uid_52",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-06-20",
                        "category": "Mortgage",
                        "description": "Mortgage statement - July 2017",
                        "documents": [{
                                "url": "http://server.com/documentsdepo/uid_53",
                                "contentType": "application/pdf"
                            }]
                    }, {
                        "accountId": "6cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "accountName": "The Walt Disney Company",
                        "accountNumber": "UA85 3996 2200 0000 0260 0123 3566 1",
                        "date": "2017-08-19",
                        "category": "Mortgage",
                        "description": "Mortgage statement - August 2017",
                        "documents": [{
                                "uid": "uid_54",
                                "contentType": "application/pdf"
                            }]
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
]);

export { ACCOUNT_STATEMENT_BASE_PATH, AccountStatementApiModule, AccountStatementConfiguration, AccountStatementHttpService, AccountStatementHttpServiceDownloadAccountStatementsMocksProvider, AccountStatementHttpServiceGetAccountStatementsMocksProvider, AccountStatementHttpServiceGetCategoriesMocksProvider, AccountStatementHttpServiceMocksProvider, AccountStatementHttpServicePostAccountStatementsMocksProvider, AccountStatementsPostRequest, COLLECTION_FORMATS, CONFIG_TOKEN };
