// @backbase/employee-http-ang/esm2015/api/cards.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const CardsServiceActivateMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/employee/users/{userId}/cards/{cardId}/activation",
        method: "POST",
        responses: [
            {
                status: 201,
                body: {
                    "id": "sara-card-3",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold Card",
                    "status": "Inactive",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Sara Williams"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5237",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2020-05-04T12:52:20.906+00:00"
                            }, {
                                "name": "Processed",
                                "status": "PENDING"
                            }, {
                                "name": "In Transit",
                                "status": "PENDING"
                            }, {
                                "name": "Delivered",
                                "status": "PENDING"
                            }]
                    },
                    "limits": [{
                            "id": "sara-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "sara-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 201,
                body: {
                    "id": "elon-card-1",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "World Elite",
                    "status": "Active",
                    "statusEnriched": "OPEN",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Elon"
                    },
                    "expiryDate": {
                        "year": "2050",
                        "month": "12"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5235",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "PENDING"
                            }, {
                                "name": "In Transit",
                                "status": "PENDING"
                            }, {
                                "name": "Delivered",
                                "status": "PENDING",
                                "stepDateTime": "2019-06-20T14:21:25.001Z"
                            }]
                    },
                    "limits": [{
                            "id": "elon-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 500000.0,
                            "maxAmount": 100000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "elon-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 1000000.0,
                            "maxAmount": 800000.0,
                            "minAmount": 10000.0
                        }]
                }
            },
            {
                status: 201,
                body: {
                    "id": "john-card-1",
                    "brand": "visa",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold",
                    "status": "Active",
                    "statusEnriched": "OPEN",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "John Doe"
                    },
                    "expiryDate": {
                        "year": "2030",
                        "month": "06"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5827",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "SUCCESS"
                            }, {
                                "name": "In Transit",
                                "status": "SUCCESS"
                            }, {
                                "name": "Delivered",
                                "status": "SUCCESS"
                            }]
                    },
                    "limits": [{
                            "id": "john-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 5000.0,
                            "maxAmount": 1000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "john-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 20000.0,
                            "maxAmount": 9000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 201,
                body: {
                    "id": "kim-card-2",
                    "brand": "mastercard",
                    "type": "Debit",
                    "subType": "ATM",
                    "name": "Copper",
                    "status": "Active",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Kim Sook"
                    },
                    "expiryDate": {
                        "year": "2027",
                        "month": "03"
                    },
                    "currency": "EUR",
                    "maskedNumber": "9292"
                }
            },
        ]
    }]);
export const CardsServiceChangeLimitsMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/employee/users/{userId}/cards/{cardId}/limits",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {
                    "id": "sara-card-1",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold Card",
                    "status": "Active",
                    "statusEnriched": "OPEN",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Sara Williams"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5235",
                    "limits": [{
                            "id": "sara-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "sara-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "elon-card-1",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "World Elite",
                    "status": "Active",
                    "statusEnriched": "OPEN",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Elon"
                    },
                    "expiryDate": {
                        "year": "2050",
                        "month": "12"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5235",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "PENDING"
                            }, {
                                "name": "In Transit",
                                "status": "PENDING"
                            }, {
                                "name": "Delivered",
                                "status": "PENDING",
                                "stepDateTime": "2019-06-20T14:21:25.001Z"
                            }]
                    },
                    "limits": [{
                            "id": "elon-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 500000.0,
                            "maxAmount": 100000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "elon-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 1000000.0,
                            "maxAmount": 800000.0,
                            "minAmount": 10000.0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "john-card-1",
                    "brand": "visa",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold",
                    "status": "Active",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "John Doe"
                    },
                    "expiryDate": {
                        "year": "2030",
                        "month": "06"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5827",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "SUCCESS"
                            }, {
                                "name": "In Transit",
                                "status": "SUCCESS"
                            }, {
                                "name": "Delivered",
                                "status": "SUCCESS"
                            }]
                    },
                    "limits": [{
                            "id": "john-limit-1",
                            "channel": "ATM",
                            "frequency": "WEEKLY",
                            "amount": 1000.0,
                            "maxAmount": 9000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "john-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 20000.0,
                            "maxAmount": 9000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "kim-card-1",
                    "brand": "mastercard",
                    "type": "Debit",
                    "subType": "ATM",
                    "name": "Copper",
                    "status": "Inactive",
                    "statusEnriched": "OPEN",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Kim Sook"
                    },
                    "expiryDate": {
                        "year": "2027",
                        "month": "03"
                    },
                    "currency": "EUR",
                    "maskedNumber": "9292"
                }
            },
        ]
    }]);
export const CardsServiceGetCardByIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/employee/users/{userId}/cards/{cardId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "sara-card-1",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold Card",
                    "status": "Active",
                    "statusEnriched": "OPEN",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Sara Williams"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5235",
                    "limits": [{
                            "id": "sara-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "sara-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "elon-card-1",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "World Elite",
                    "status": "Active",
                    "statusEnriched": "OPEN",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Elon"
                    },
                    "expiryDate": {
                        "year": "2050",
                        "month": "12"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5235",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "PENDING"
                            }, {
                                "name": "In Transit",
                                "status": "PENDING"
                            }, {
                                "name": "Delivered",
                                "status": "PENDING",
                                "stepDateTime": "2019-06-20T14:21:25.001Z"
                            }]
                    },
                    "limits": [{
                            "id": "elon-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 500000.0,
                            "maxAmount": 100000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "elon-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 1000000.0,
                            "maxAmount": 800000.0,
                            "minAmount": 10000.0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "john-card-1",
                    "brand": "visa",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold",
                    "status": "Active",
                    "statusEnriched": "OPEN",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "John Doe"
                    },
                    "expiryDate": {
                        "year": "2030",
                        "month": "06"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5827",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "SUCCESS"
                            }, {
                                "name": "In Transit",
                                "status": "SUCCESS"
                            }, {
                                "name": "Delivered",
                                "status": "SUCCESS"
                            }]
                    },
                    "limits": [{
                            "id": "john-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 5000.0,
                            "maxAmount": 1000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "john-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 20000.0,
                            "maxAmount": 9000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "kim-card-1",
                    "brand": "mastercard",
                    "type": "Debit",
                    "subType": "ATM",
                    "name": "Copper",
                    "status": "Inactive",
                    "statusEnriched": "OPEN",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Kim Sook"
                    },
                    "expiryDate": {
                        "year": "2027",
                        "month": "03"
                    },
                    "currency": "EUR",
                    "maskedNumber": "9292"
                }
            },
            {
                status: 200,
                body: {
                    "id": "expired-card-3",
                    "brand": "visa debit",
                    "type": "Debit",
                    "subType": "ATM",
                    "name": "Blue Card",
                    "status": "Expired",
                    "lockStatus": "LOCKED",
                    "replacement": {
                        "status": "UnderReplacement",
                        "reason": "Lost",
                        "replacedById": "4"
                    },
                    "holder": {
                        "name": "Jason Carter"
                    },
                    "expiryDate": {
                        "year": "2048",
                        "month": "06"
                    },
                    "currency": "EUR",
                    "maskedNumber": "1262",
                    "limits": [{
                            "id": "expired-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "expired-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 15000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "inactive-card-4",
                    "brand": "union pay",
                    "type": "Debit",
                    "subType": "ATM",
                    "name": "Black Card",
                    "status": "Inactive",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Jessie Carter"
                    },
                    "expiryDate": {
                        "year": "2048",
                        "month": "07"
                    },
                    "currency": "GBP",
                    "maskedNumber": "1289",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2020-05-04T12:52:20.906+00:00"
                            }, {
                                "name": "Processed",
                                "status": "PENDING"
                            }, {
                                "name": "In Transit",
                                "status": "PENDING"
                            }, {
                                "name": "Delivered",
                                "status": "PENDING"
                            }]
                    },
                    "limits": [{
                            "id": "inactive-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "inactive-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "locked-card-10",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Green Card",
                    "status": "Active",
                    "lockStatus": "LOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Jason Carter"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5235",
                    "limits": [{
                            "id": "locked-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "locked-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 15000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "value": {
                        "ref": "examples/unusable/card-item.json"
                    }
                }
            },
            {
                status: 200,
                body: {
                    "id": "damaged-card-9",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold Card",
                    "status": "Active",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "ReplacementRequested",
                        "reason": "Damaged"
                    },
                    "holder": {
                        "name": "Jessie Carter"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5235",
                    "limits": [{
                            "id": "damaged-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "damaged-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 15000.0,
                            "minAmount": 0.0
                        }],
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2021-07-01T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "PENDING"
                            }]
                    }
                }
            },
            {
                status: 200,
                body: {
                    "id": "replaced-card-8",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold Card",
                    "status": "Canceled",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "Replaced",
                        "reason": "Lost"
                    },
                    "holder": {
                        "name": "Jason Carter"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5235",
                    "limits": [{
                            "id": "replaced-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "replaced-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 15000.0,
                            "minAmount": 0.0
                        }],
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2021-07-01T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "SUCCESS"
                            }, {
                                "name": "In Transit",
                                "status": "SUCCESS"
                            }, {
                                "name": "Delivered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-20T14:21:25.001Z"
                            }]
                    }
                }
            },
            {
                status: 200,
                body: {
                    "id": "replacing-card-10",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold Card",
                    "status": "Inactive",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "Replacing",
                        "reason": "Lost"
                    },
                    "holder": {
                        "name": "Randy Carter"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5235",
                    "limits": [{
                            "id": "replacing-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "replacing-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 15000.0,
                            "minAmount": 0.0
                        }],
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2021-07-01T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "SUCCESS"
                            }, {
                                "name": "In Transit",
                                "status": "PENDING"
                            }]
                    }
                }
            },
            {
                status: 200,
                body: {
                    "id": "stolen-card-11",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold Card",
                    "status": "Canceled",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "Replacing",
                        "reason": "Stolen"
                    },
                    "holder": {
                        "name": "Lisa Carter"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5235",
                    "limits": [{
                            "id": "stolen-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "stolen-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 15000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "replacement-card-9",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold Card",
                    "status": "Inactive",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "ReplacementRequested",
                        "reason": "Lost"
                    },
                    "holder": {
                        "name": "Lisa Carter"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5235",
                    "limits": [{
                            "id": "replacement-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "replacement-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 15000.0,
                            "minAmount": 0.0
                        }],
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2021-07-01T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "PENDING"
                            }]
                    }
                }
            },
            {
                status: 200,
                body: {
                    "id": "fraud-card-4",
                    "brand": "union pay",
                    "type": "Debit",
                    "subType": "ATM",
                    "name": "Black Card",
                    "status": "Fraud",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Jessie Carter"
                    },
                    "expiryDate": {
                        "year": "2048",
                        "month": "07"
                    },
                    "currency": "GBP",
                    "maskedNumber": "1289",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2020-05-04T12:52:20.906+00:00"
                            }, {
                                "name": "Processed",
                                "status": "SUCCESS",
                                "stepDateTime": "2020-05-06T12:52:20.906+00:00"
                            }, {
                                "name": "In Transit",
                                "status": "SUCCESS",
                                "stepDateTime": "2020-05-07T12:52:20.906+00:00"
                            }, {
                                "name": "Delivered",
                                "status": "SUCCESS",
                                "stepDateTime": "2020-05-09T12:52:20.906+00:00"
                            }]
                    },
                    "limits": [{
                            "id": "fraud-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "fraud-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }]
                }
            },
        ]
    }]);
export const CardsServiceGetCardsMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/employee/users/{userId}/cards",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "sara-card-1",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Sara Williams"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5235"
                    }, {
                        "id": "sara-card-2",
                        "brand": "visa",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Green Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Sara Williams"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "01"
                        },
                        "currency": "USD",
                        "maskedNumber": "5236"
                    }, {
                        "id": "sara-card-3",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Inactive",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Sara Williams"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5237",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2020-05-04T12:52:20.906+00:00"
                                }, {
                                    "name": "Processed",
                                    "status": "PENDING"
                                }, {
                                    "name": "In Transit",
                                    "status": "PENDING"
                                }, {
                                    "name": "Delivered",
                                    "status": "PENDING"
                                }]
                        }
                    }, {
                        "id": "sara-card-4",
                        "brand": "mastercard",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Black Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Sara Williams"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "6235"
                    }, {
                        "id": "sara-card-5",
                        "brand": "visa",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Blue Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Sara Williams"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "6236"
                    }]
            },
            {
                status: 200,
                body: [{
                        "id": "elon-card-1",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "World Elite",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Elon"
                        },
                        "expiryDate": {
                            "year": "2050",
                            "month": "12"
                        },
                        "currency": "EUR",
                        "maskedNumber": "5235",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-19T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "PENDING"
                                }, {
                                    "name": "In Transit",
                                    "status": "PENDING"
                                }, {
                                    "name": "Delivered",
                                    "status": "PENDING",
                                    "stepDateTime": "2019-06-20T14:21:25.001Z"
                                }]
                        }
                    }, {
                        "id": "elon-card-2",
                        "brand": "visa",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Infinite Privilege",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Elon"
                        },
                        "expiryDate": {
                            "year": "2045",
                            "month": "01"
                        },
                        "currency": "EUR",
                        "maskedNumber": "6782"
                    }, {
                        "id": "elon-card-3",
                        "brand": "american_express",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Reserve",
                        "status": "Cancelled",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "UnderReplacement",
                            "reason": "Lost",
                            "replacedById": "4"
                        },
                        "holder": {
                            "name": "Elon"
                        },
                        "expiryDate": {
                            "year": "2048",
                            "month": "06"
                        },
                        "currency": "EUR",
                        "maskedNumber": "1262"
                    }, {
                        "id": "elon-card-4",
                        "brand": "american_express",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Reserve",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Elon"
                        },
                        "expiryDate": {
                            "year": "2048",
                            "month": "07"
                        },
                        "currency": "EUR",
                        "maskedNumber": "1289"
                    }, {
                        "id": "elon-card-5",
                        "brand": "mastercard",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Platinum",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Elon"
                        },
                        "expiryDate": {
                            "year": "2050",
                            "month": "11"
                        },
                        "currency": "EUR",
                        "maskedNumber": "4627",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-19T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-20T14:32:20.906Z"
                                }, {
                                    "name": "In Transit",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-20T15:12:20.906Z"
                                }, {
                                    "name": "Delivered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-21T11:28:25.001Z"
                                }]
                        }
                    }, {
                        "id": "elon-card-6",
                        "brand": "american_express",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Reserve",
                        "status": "Active",
                        "lockStatus": "LOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Elon"
                        },
                        "expiryDate": {
                            "year": "2047",
                            "month": "03"
                        },
                        "currency": "EUR",
                        "maskedNumber": "8291"
                    }]
            },
            {
                status: 200,
                body: [{
                        "id": "john-card-1",
                        "brand": "visa",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "John Doe"
                        },
                        "expiryDate": {
                            "year": "2030",
                            "month": "06"
                        },
                        "currency": "EUR",
                        "maskedNumber": "5827",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-19T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "SUCCESS"
                                }, {
                                    "name": "In Transit",
                                    "status": "SUCCESS"
                                }, {
                                    "name": "Delivered",
                                    "status": "SUCCESS"
                                }]
                        },
                        "limits": [{
                                "id": "john-limit-1",
                                "channel": "ATM",
                                "frequency": "DAILY",
                                "amount": 3000.0,
                                "maxAmount": 9000.0,
                                "minAmount": 0.0
                            }, {
                                "id": "john-limit-2",
                                "channel": "ONLINE",
                                "frequency": "DAILY",
                                "amount": 20000.0,
                                "maxAmount": 9000.0,
                                "minAmount": 0.0
                            }]
                    }, {
                        "id": "john-card-2",
                        "brand": "mastercard",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Gold",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "John Doe"
                        },
                        "expiryDate": {
                            "year": "2031",
                            "month": "02"
                        },
                        "currency": "EUR",
                        "maskedNumber": "8299"
                    }]
            },
            {
                status: 200,
                body: [{
                        "id": "kim-card-1",
                        "brand": "mastercard",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Copper",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Kim Sook"
                        },
                        "expiryDate": {
                            "year": "2027",
                            "month": "03"
                        },
                        "currency": "EUR",
                        "maskedNumber": "9292"
                    }, {
                        "id": "kim-card-2",
                        "brand": "visa",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Iron",
                        "status": "Inactive",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Kim Sook"
                        },
                        "expiryDate": {
                            "year": "2010",
                            "month": "04"
                        },
                        "currency": "EUR",
                        "maskedNumber": "9494"
                    }, {
                        "id": "kim-card-3",
                        "brand": "visa",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Iron",
                        "status": "Expired",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Kim Sook"
                        },
                        "expiryDate": {
                            "year": "2010",
                            "month": "04"
                        },
                        "currency": "EUR",
                        "maskedNumber": "9494"
                    }]
            },
            {
                status: 200,
                body: []
            },
            {
                status: 200,
                body: [{
                        "id": "current-card-1",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Jason Carter"
                        },
                        "expiryDate": {
                            "year": "2050",
                            "month": "12"
                        },
                        "currency": "EUR",
                        "maskedNumber": "5235",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-19T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "PENDING"
                                }, {
                                    "name": "In Transit",
                                    "status": "PENDING"
                                }, {
                                    "name": "Delivered",
                                    "status": "PENDING",
                                    "stepDateTime": "2019-06-20T14:21:25.001Z"
                                }]
                        }
                    }, {
                        "id": "current-card-2",
                        "brand": "visa",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Green Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Jessie Carter"
                        },
                        "expiryDate": {
                            "year": "2045",
                            "month": "01"
                        },
                        "currency": "EUR",
                        "maskedNumber": "6782"
                    }, {
                        "id": "current-card-3",
                        "brand": "american express",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Blue Card",
                        "status": "Inactive",
                        "lockStatus": "LOCKED",
                        "replacement": {
                            "status": "UnderReplacement",
                            "reason": "Lost",
                            "replacedById": "4"
                        },
                        "holder": {
                            "name": "Jason Carter"
                        },
                        "expiryDate": {
                            "year": "2048",
                            "month": "06"
                        },
                        "currency": "EUR",
                        "maskedNumber": "1262"
                    }, {
                        "id": "current-card-4",
                        "brand": "union pay",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Black Card",
                        "status": "Inactive",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Jessie Carter"
                        },
                        "expiryDate": {
                            "year": "2048",
                            "month": "07"
                        },
                        "currency": "GBP",
                        "maskedNumber": "1289"
                    }, {
                        "id": "current-card-5",
                        "brand": "maestro",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Platinum",
                        "status": "Active",
                        "lockStatus": "LOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Jessie Carter"
                        },
                        "expiryDate": {
                            "year": "2050",
                            "month": "11"
                        },
                        "currency": "USD",
                        "maskedNumber": "4627",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-19T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-20T14:32:20.906Z"
                                }, {
                                    "name": "In Transit",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-20T15:12:20.906Z"
                                }, {
                                    "name": "Delivered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-21T11:28:25.001Z"
                                }]
                        }
                    }, {
                        "id": "current-card-6",
                        "brand": "diners club",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "White Card",
                        "status": "Active",
                        "lockStatus": "LOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Lisa Carter"
                        },
                        "expiryDate": {
                            "year": "2047",
                            "month": "03"
                        },
                        "currency": "GBP",
                        "maskedNumber": "8291"
                    }, {
                        "id": "current-card-7",
                        "brand": "visa debit",
                        "type": "debit",
                        "subType": "ATM",
                        "name": "Platinum Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Randy Carter"
                        },
                        "expiryDate": {
                            "year": "2047",
                            "month": "03"
                        },
                        "currency": "EUR",
                        "maskedNumber": "8291"
                    }, {
                        "id": "current-card-8",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Jason Carter"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5235"
                    }, {
                        "id": "current-card-9",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "ReplacementRequested",
                            "reason": "Damaged"
                        },
                        "holder": {
                            "name": "Jessie Carter"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5235",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2021-07-01T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "PENDING"
                                }]
                        }
                    }, {
                        "id": "current-card-10",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Green Card",
                        "status": "Active",
                        "lockStatus": "LOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Jason Carter"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5235"
                    }, {
                        "id": "current-card-11",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Inactive",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "Replaced",
                            "reason": "Lost"
                        },
                        "holder": {
                            "name": "Lisa Carter"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5235",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2021-07-01T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "PENDING"
                                }]
                        }
                    }, {
                        "id": "current-card-12",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Inactive",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "Replacing",
                            "reason": "Stolen"
                        },
                        "holder": {
                            "name": "Lisa Carter"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5235"
                    }, {
                        "id": "current-card-13",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Unknown",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Lisa Carter"
                        },
                        "expiryDate": {
                            "year": "2030",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5299"
                    }, {
                        "id": "current-card-13",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Fraud",
                        "lockStatus": "LOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Jason Carter"
                        },
                        "expiryDate": {
                            "year": "2035",
                            "month": "01"
                        },
                        "currency": "GBP",
                        "maskedNumber": "1399"
                    }]
            },
            {
                status: 200,
                body: [{
                        "id": "closed-card-1",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Canceled",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Jason Carter"
                        },
                        "expiryDate": {
                            "year": "2050",
                            "month": "12"
                        },
                        "currency": "EUR",
                        "maskedNumber": "5235",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-19T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "PENDING"
                                }, {
                                    "name": "In Transit",
                                    "status": "PENDING"
                                }, {
                                    "name": "Delivered",
                                    "status": "PENDING",
                                    "stepDateTime": "2019-06-20T14:21:25.001Z"
                                }]
                        }
                    }, {
                        "id": "closed-card-2",
                        "brand": "visa",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Green Card",
                        "status": "Canceled",
                        "lockStatus": "LOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Jessie Carter"
                        },
                        "expiryDate": {
                            "year": "2045",
                            "month": "01"
                        },
                        "currency": "EUR",
                        "maskedNumber": "6782"
                    }, {
                        "id": "closed-card-3",
                        "brand": "american express",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Blue Card",
                        "status": "Expired",
                        "lockStatus": "LOCKED",
                        "replacement": {
                            "status": "UnderReplacement",
                            "reason": "Lost",
                            "replacedById": "4"
                        },
                        "holder": {
                            "name": "Jason Carter"
                        },
                        "expiryDate": {
                            "year": "2048",
                            "month": "06"
                        },
                        "currency": "EUR",
                        "maskedNumber": "1262"
                    }, {
                        "id": "closed-card-4",
                        "brand": "union pay",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Black Card",
                        "status": "Expired",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Jessie Carter"
                        },
                        "expiryDate": {
                            "year": "2048",
                            "month": "07"
                        },
                        "currency": "GBP",
                        "maskedNumber": "1289"
                    }, {
                        "id": "closed-card-5",
                        "brand": "maestro",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Platinum",
                        "status": "Expired",
                        "lockStatus": "LOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Jessie Carter"
                        },
                        "expiryDate": {
                            "year": "2050",
                            "month": "11"
                        },
                        "currency": "USD",
                        "maskedNumber": "4627",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-19T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-20T14:32:20.906Z"
                                }, {
                                    "name": "In Transit",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-20T15:12:20.906Z"
                                }, {
                                    "name": "Delivered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-21T11:28:25.001Z"
                                }]
                        }
                    }, {
                        "id": "closed-card-6",
                        "brand": "diners club",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "White Card",
                        "status": "Canceled",
                        "lockStatus": "LOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Lisa Carter"
                        },
                        "expiryDate": {
                            "year": "2047",
                            "month": "03"
                        },
                        "currency": "GBP",
                        "maskedNumber": "8291"
                    }, {
                        "id": "closed-card-7",
                        "brand": "visa debit",
                        "type": "debit",
                        "subType": "ATM",
                        "name": "Platinum Card",
                        "status": "Canceled",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Randy Carter"
                        },
                        "expiryDate": {
                            "year": "2047",
                            "month": "03"
                        },
                        "currency": "EUR",
                        "maskedNumber": "8291"
                    }, {
                        "id": "closed-card-8",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Canceled",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "Replaced",
                            "reason": "Lost"
                        },
                        "holder": {
                            "name": "Jason Carter"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5235",
                        "limits": [{
                                "id": "closed-limit-1",
                                "channel": "ATM",
                                "frequency": "DAILY",
                                "amount": 3000.0,
                                "maxAmount": 10000.0,
                                "minAmount": 0.0
                            }, {
                                "id": "closed-limit-2",
                                "channel": "ONLINE",
                                "frequency": "DAILY",
                                "amount": 2000.0,
                                "maxAmount": 15000.0,
                                "minAmount": 0.0
                            }],
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2021-07-01T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "PENDING"
                                }]
                        }
                    }, {
                        "id": "closed-card-9",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Canceled",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "ReplacementRequested",
                            "reason": "Lost"
                        },
                        "holder": {
                            "name": "Lisa Carter"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5235",
                        "limits": [{
                                "id": "closed-limit-1",
                                "channel": "ATM",
                                "frequency": "DAILY",
                                "amount": 3000.0,
                                "maxAmount": 10000.0,
                                "minAmount": 0.0
                            }, {
                                "id": "closed-limit-2",
                                "channel": "ONLINE",
                                "frequency": "DAILY",
                                "amount": 2000.0,
                                "maxAmount": 15000.0,
                                "minAmount": 0.0
                            }],
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2021-07-01T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "PENDING"
                                }]
                        }
                    }, {
                        "id": "closed-card-10",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Canceled",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "ReplacementRequested",
                            "reason": "Lost"
                        },
                        "holder": {
                            "name": "Randy Carter"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5235",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2021-07-01T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "PENDING"
                                }]
                        }
                    }, {
                        "id": "closed-card-11",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Canceled",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "Replacing",
                            "reason": "Stolen"
                        },
                        "holder": {
                            "name": "Lisa Carter"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5235"
                    }]
            },
        ]
    }]);
export const CardsServiceRequestReplacementMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/employee/users/{userId}/cards/{cardId}/replacement",
        method: "POST",
        responses: [
            {
                status: 201,
                body: {
                    "id": "sara-card-1",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold Card",
                    "status": "Active",
                    "statusEnriched": "OPEN",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Sara Williams"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5235",
                    "limits": [{
                            "id": "sara-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "sara-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 201,
                body: {
                    "id": "elon-card-1",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "World Elite",
                    "status": "Active",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "ReplacementRequested",
                        "reason": "Damaged"
                    },
                    "holder": {
                        "name": "Elon"
                    },
                    "expiryDate": {
                        "year": "2050",
                        "month": "12"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5235",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "PENDING"
                            }]
                    }
                }
            },
            {
                status: 201,
                body: {
                    "id": "john-card-1",
                    "brand": "visa",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold",
                    "status": "Inactive",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "ReplacementRequested",
                        "reason": "Stolen"
                    },
                    "holder": {
                        "name": "John Doe"
                    },
                    "expiryDate": {
                        "year": "2030",
                        "month": "06"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5827",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "SUCCESS"
                            }, {
                                "name": "In Transit",
                                "status": "SUCCESS"
                            }, {
                                "name": "Delivered",
                                "status": "SUCCESS"
                            }]
                    }
                }
            },
            {
                status: 201,
                body: {
                    "id": "kim-card-1",
                    "brand": "mastercard",
                    "type": "Debit",
                    "subType": "ATM",
                    "name": "Copper",
                    "status": "Inactive",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "ReplacementRequested",
                        "reason": "Lost"
                    },
                    "holder": {
                        "name": "Kim Sook"
                    },
                    "expiryDate": {
                        "year": "2027",
                        "month": "03"
                    },
                    "currency": "EUR",
                    "maskedNumber": "9292"
                }
            },
        ]
    }]);
export const CardsServiceUpdateLockStatusMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/employee/users/{userId}/cards/{cardId}/lock-status",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {
                    "id": "sara-card-1",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold Card",
                    "status": "Active",
                    "statusEnriched": "OPEN",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Sara Williams"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5235",
                    "limits": [{
                            "id": "sara-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "sara-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "elon-card-6",
                    "brand": "american_express",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Reserve",
                    "status": "Active",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Elon"
                    },
                    "expiryDate": {
                        "year": "2047",
                        "month": "03"
                    },
                    "currency": "EUR",
                    "maskedNumber": "8291"
                }
            },
            {
                status: 200,
                body: {
                    "id": "john-card-1",
                    "brand": "visa",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold",
                    "status": "Active",
                    "lockStatus": "LOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "John Doe"
                    },
                    "expiryDate": {
                        "year": "2030",
                        "month": "06"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5827",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "SUCCESS"
                            }, {
                                "name": "In Transit",
                                "status": "SUCCESS"
                            }, {
                                "name": "Delivered",
                                "status": "SUCCESS"
                            }]
                    },
                    "limits": [{
                            "id": "john-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 5000.0,
                            "maxAmount": 1000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "john-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 20000.0,
                            "maxAmount": 9000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "kim-card-1",
                    "brand": "mastercard",
                    "type": "Debit",
                    "subType": "ATM",
                    "name": "Copper",
                    "status": "Inactive",
                    "statusEnriched": "OPEN",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Kim Sook"
                    },
                    "expiryDate": {
                        "year": "2027",
                        "month": "03"
                    },
                    "currency": "EUR",
                    "maskedNumber": "9292"
                }
            },
        ]
    }]);
export const CardsServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v1/employee/users/{userId}/cards/{cardId}/activation",
        method: "POST",
        responses: [
            {
                status: 201,
                body: {
                    "id": "sara-card-3",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold Card",
                    "status": "Inactive",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Sara Williams"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5237",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2020-05-04T12:52:20.906+00:00"
                            }, {
                                "name": "Processed",
                                "status": "PENDING"
                            }, {
                                "name": "In Transit",
                                "status": "PENDING"
                            }, {
                                "name": "Delivered",
                                "status": "PENDING"
                            }]
                    },
                    "limits": [{
                            "id": "sara-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "sara-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 201,
                body: {
                    "id": "elon-card-1",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "World Elite",
                    "status": "Active",
                    "statusEnriched": "OPEN",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Elon"
                    },
                    "expiryDate": {
                        "year": "2050",
                        "month": "12"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5235",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "PENDING"
                            }, {
                                "name": "In Transit",
                                "status": "PENDING"
                            }, {
                                "name": "Delivered",
                                "status": "PENDING",
                                "stepDateTime": "2019-06-20T14:21:25.001Z"
                            }]
                    },
                    "limits": [{
                            "id": "elon-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 500000.0,
                            "maxAmount": 100000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "elon-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 1000000.0,
                            "maxAmount": 800000.0,
                            "minAmount": 10000.0
                        }]
                }
            },
            {
                status: 201,
                body: {
                    "id": "john-card-1",
                    "brand": "visa",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold",
                    "status": "Active",
                    "statusEnriched": "OPEN",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "John Doe"
                    },
                    "expiryDate": {
                        "year": "2030",
                        "month": "06"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5827",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "SUCCESS"
                            }, {
                                "name": "In Transit",
                                "status": "SUCCESS"
                            }, {
                                "name": "Delivered",
                                "status": "SUCCESS"
                            }]
                    },
                    "limits": [{
                            "id": "john-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 5000.0,
                            "maxAmount": 1000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "john-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 20000.0,
                            "maxAmount": 9000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 201,
                body: {
                    "id": "kim-card-2",
                    "brand": "mastercard",
                    "type": "Debit",
                    "subType": "ATM",
                    "name": "Copper",
                    "status": "Active",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Kim Sook"
                    },
                    "expiryDate": {
                        "year": "2027",
                        "month": "03"
                    },
                    "currency": "EUR",
                    "maskedNumber": "9292"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/employee/users/{userId}/cards/{cardId}/limits",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {
                    "id": "sara-card-1",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold Card",
                    "status": "Active",
                    "statusEnriched": "OPEN",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Sara Williams"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5235",
                    "limits": [{
                            "id": "sara-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "sara-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "elon-card-1",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "World Elite",
                    "status": "Active",
                    "statusEnriched": "OPEN",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Elon"
                    },
                    "expiryDate": {
                        "year": "2050",
                        "month": "12"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5235",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "PENDING"
                            }, {
                                "name": "In Transit",
                                "status": "PENDING"
                            }, {
                                "name": "Delivered",
                                "status": "PENDING",
                                "stepDateTime": "2019-06-20T14:21:25.001Z"
                            }]
                    },
                    "limits": [{
                            "id": "elon-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 500000.0,
                            "maxAmount": 100000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "elon-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 1000000.0,
                            "maxAmount": 800000.0,
                            "minAmount": 10000.0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "john-card-1",
                    "brand": "visa",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold",
                    "status": "Active",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "John Doe"
                    },
                    "expiryDate": {
                        "year": "2030",
                        "month": "06"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5827",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "SUCCESS"
                            }, {
                                "name": "In Transit",
                                "status": "SUCCESS"
                            }, {
                                "name": "Delivered",
                                "status": "SUCCESS"
                            }]
                    },
                    "limits": [{
                            "id": "john-limit-1",
                            "channel": "ATM",
                            "frequency": "WEEKLY",
                            "amount": 1000.0,
                            "maxAmount": 9000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "john-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 20000.0,
                            "maxAmount": 9000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "kim-card-1",
                    "brand": "mastercard",
                    "type": "Debit",
                    "subType": "ATM",
                    "name": "Copper",
                    "status": "Inactive",
                    "statusEnriched": "OPEN",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Kim Sook"
                    },
                    "expiryDate": {
                        "year": "2027",
                        "month": "03"
                    },
                    "currency": "EUR",
                    "maskedNumber": "9292"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/employee/users/{userId}/cards/{cardId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "sara-card-1",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold Card",
                    "status": "Active",
                    "statusEnriched": "OPEN",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Sara Williams"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5235",
                    "limits": [{
                            "id": "sara-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "sara-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "elon-card-1",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "World Elite",
                    "status": "Active",
                    "statusEnriched": "OPEN",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Elon"
                    },
                    "expiryDate": {
                        "year": "2050",
                        "month": "12"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5235",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "PENDING"
                            }, {
                                "name": "In Transit",
                                "status": "PENDING"
                            }, {
                                "name": "Delivered",
                                "status": "PENDING",
                                "stepDateTime": "2019-06-20T14:21:25.001Z"
                            }]
                    },
                    "limits": [{
                            "id": "elon-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 500000.0,
                            "maxAmount": 100000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "elon-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 1000000.0,
                            "maxAmount": 800000.0,
                            "minAmount": 10000.0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "john-card-1",
                    "brand": "visa",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold",
                    "status": "Active",
                    "statusEnriched": "OPEN",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "John Doe"
                    },
                    "expiryDate": {
                        "year": "2030",
                        "month": "06"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5827",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "SUCCESS"
                            }, {
                                "name": "In Transit",
                                "status": "SUCCESS"
                            }, {
                                "name": "Delivered",
                                "status": "SUCCESS"
                            }]
                    },
                    "limits": [{
                            "id": "john-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 5000.0,
                            "maxAmount": 1000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "john-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 20000.0,
                            "maxAmount": 9000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "kim-card-1",
                    "brand": "mastercard",
                    "type": "Debit",
                    "subType": "ATM",
                    "name": "Copper",
                    "status": "Inactive",
                    "statusEnriched": "OPEN",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Kim Sook"
                    },
                    "expiryDate": {
                        "year": "2027",
                        "month": "03"
                    },
                    "currency": "EUR",
                    "maskedNumber": "9292"
                }
            },
            {
                status: 200,
                body: {
                    "id": "expired-card-3",
                    "brand": "visa debit",
                    "type": "Debit",
                    "subType": "ATM",
                    "name": "Blue Card",
                    "status": "Expired",
                    "lockStatus": "LOCKED",
                    "replacement": {
                        "status": "UnderReplacement",
                        "reason": "Lost",
                        "replacedById": "4"
                    },
                    "holder": {
                        "name": "Jason Carter"
                    },
                    "expiryDate": {
                        "year": "2048",
                        "month": "06"
                    },
                    "currency": "EUR",
                    "maskedNumber": "1262",
                    "limits": [{
                            "id": "expired-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "expired-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 15000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "inactive-card-4",
                    "brand": "union pay",
                    "type": "Debit",
                    "subType": "ATM",
                    "name": "Black Card",
                    "status": "Inactive",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Jessie Carter"
                    },
                    "expiryDate": {
                        "year": "2048",
                        "month": "07"
                    },
                    "currency": "GBP",
                    "maskedNumber": "1289",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2020-05-04T12:52:20.906+00:00"
                            }, {
                                "name": "Processed",
                                "status": "PENDING"
                            }, {
                                "name": "In Transit",
                                "status": "PENDING"
                            }, {
                                "name": "Delivered",
                                "status": "PENDING"
                            }]
                    },
                    "limits": [{
                            "id": "inactive-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "inactive-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "locked-card-10",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Green Card",
                    "status": "Active",
                    "lockStatus": "LOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Jason Carter"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5235",
                    "limits": [{
                            "id": "locked-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "locked-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 15000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "value": {
                        "ref": "examples/unusable/card-item.json"
                    }
                }
            },
            {
                status: 200,
                body: {
                    "id": "damaged-card-9",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold Card",
                    "status": "Active",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "ReplacementRequested",
                        "reason": "Damaged"
                    },
                    "holder": {
                        "name": "Jessie Carter"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5235",
                    "limits": [{
                            "id": "damaged-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "damaged-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 15000.0,
                            "minAmount": 0.0
                        }],
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2021-07-01T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "PENDING"
                            }]
                    }
                }
            },
            {
                status: 200,
                body: {
                    "id": "replaced-card-8",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold Card",
                    "status": "Canceled",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "Replaced",
                        "reason": "Lost"
                    },
                    "holder": {
                        "name": "Jason Carter"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5235",
                    "limits": [{
                            "id": "replaced-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "replaced-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 15000.0,
                            "minAmount": 0.0
                        }],
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2021-07-01T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "SUCCESS"
                            }, {
                                "name": "In Transit",
                                "status": "SUCCESS"
                            }, {
                                "name": "Delivered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-20T14:21:25.001Z"
                            }]
                    }
                }
            },
            {
                status: 200,
                body: {
                    "id": "replacing-card-10",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold Card",
                    "status": "Inactive",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "Replacing",
                        "reason": "Lost"
                    },
                    "holder": {
                        "name": "Randy Carter"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5235",
                    "limits": [{
                            "id": "replacing-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "replacing-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 15000.0,
                            "minAmount": 0.0
                        }],
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2021-07-01T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "SUCCESS"
                            }, {
                                "name": "In Transit",
                                "status": "PENDING"
                            }]
                    }
                }
            },
            {
                status: 200,
                body: {
                    "id": "stolen-card-11",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold Card",
                    "status": "Canceled",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "Replacing",
                        "reason": "Stolen"
                    },
                    "holder": {
                        "name": "Lisa Carter"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5235",
                    "limits": [{
                            "id": "stolen-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "stolen-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 15000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "replacement-card-9",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold Card",
                    "status": "Inactive",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "ReplacementRequested",
                        "reason": "Lost"
                    },
                    "holder": {
                        "name": "Lisa Carter"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5235",
                    "limits": [{
                            "id": "replacement-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "replacement-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 15000.0,
                            "minAmount": 0.0
                        }],
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2021-07-01T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "PENDING"
                            }]
                    }
                }
            },
            {
                status: 200,
                body: {
                    "id": "fraud-card-4",
                    "brand": "union pay",
                    "type": "Debit",
                    "subType": "ATM",
                    "name": "Black Card",
                    "status": "Fraud",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Jessie Carter"
                    },
                    "expiryDate": {
                        "year": "2048",
                        "month": "07"
                    },
                    "currency": "GBP",
                    "maskedNumber": "1289",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2020-05-04T12:52:20.906+00:00"
                            }, {
                                "name": "Processed",
                                "status": "SUCCESS",
                                "stepDateTime": "2020-05-06T12:52:20.906+00:00"
                            }, {
                                "name": "In Transit",
                                "status": "SUCCESS",
                                "stepDateTime": "2020-05-07T12:52:20.906+00:00"
                            }, {
                                "name": "Delivered",
                                "status": "SUCCESS",
                                "stepDateTime": "2020-05-09T12:52:20.906+00:00"
                            }]
                    },
                    "limits": [{
                            "id": "fraud-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "fraud-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/employee/users/{userId}/cards",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "sara-card-1",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Sara Williams"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5235"
                    }, {
                        "id": "sara-card-2",
                        "brand": "visa",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Green Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Sara Williams"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "01"
                        },
                        "currency": "USD",
                        "maskedNumber": "5236"
                    }, {
                        "id": "sara-card-3",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Inactive",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Sara Williams"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5237",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2020-05-04T12:52:20.906+00:00"
                                }, {
                                    "name": "Processed",
                                    "status": "PENDING"
                                }, {
                                    "name": "In Transit",
                                    "status": "PENDING"
                                }, {
                                    "name": "Delivered",
                                    "status": "PENDING"
                                }]
                        }
                    }, {
                        "id": "sara-card-4",
                        "brand": "mastercard",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Black Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Sara Williams"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "6235"
                    }, {
                        "id": "sara-card-5",
                        "brand": "visa",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Blue Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Sara Williams"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "6236"
                    }]
            },
            {
                status: 200,
                body: [{
                        "id": "elon-card-1",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "World Elite",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Elon"
                        },
                        "expiryDate": {
                            "year": "2050",
                            "month": "12"
                        },
                        "currency": "EUR",
                        "maskedNumber": "5235",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-19T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "PENDING"
                                }, {
                                    "name": "In Transit",
                                    "status": "PENDING"
                                }, {
                                    "name": "Delivered",
                                    "status": "PENDING",
                                    "stepDateTime": "2019-06-20T14:21:25.001Z"
                                }]
                        }
                    }, {
                        "id": "elon-card-2",
                        "brand": "visa",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Infinite Privilege",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Elon"
                        },
                        "expiryDate": {
                            "year": "2045",
                            "month": "01"
                        },
                        "currency": "EUR",
                        "maskedNumber": "6782"
                    }, {
                        "id": "elon-card-3",
                        "brand": "american_express",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Reserve",
                        "status": "Cancelled",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "UnderReplacement",
                            "reason": "Lost",
                            "replacedById": "4"
                        },
                        "holder": {
                            "name": "Elon"
                        },
                        "expiryDate": {
                            "year": "2048",
                            "month": "06"
                        },
                        "currency": "EUR",
                        "maskedNumber": "1262"
                    }, {
                        "id": "elon-card-4",
                        "brand": "american_express",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Reserve",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Elon"
                        },
                        "expiryDate": {
                            "year": "2048",
                            "month": "07"
                        },
                        "currency": "EUR",
                        "maskedNumber": "1289"
                    }, {
                        "id": "elon-card-5",
                        "brand": "mastercard",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Platinum",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Elon"
                        },
                        "expiryDate": {
                            "year": "2050",
                            "month": "11"
                        },
                        "currency": "EUR",
                        "maskedNumber": "4627",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-19T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-20T14:32:20.906Z"
                                }, {
                                    "name": "In Transit",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-20T15:12:20.906Z"
                                }, {
                                    "name": "Delivered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-21T11:28:25.001Z"
                                }]
                        }
                    }, {
                        "id": "elon-card-6",
                        "brand": "american_express",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Reserve",
                        "status": "Active",
                        "lockStatus": "LOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Elon"
                        },
                        "expiryDate": {
                            "year": "2047",
                            "month": "03"
                        },
                        "currency": "EUR",
                        "maskedNumber": "8291"
                    }]
            },
            {
                status: 200,
                body: [{
                        "id": "john-card-1",
                        "brand": "visa",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "John Doe"
                        },
                        "expiryDate": {
                            "year": "2030",
                            "month": "06"
                        },
                        "currency": "EUR",
                        "maskedNumber": "5827",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-19T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "SUCCESS"
                                }, {
                                    "name": "In Transit",
                                    "status": "SUCCESS"
                                }, {
                                    "name": "Delivered",
                                    "status": "SUCCESS"
                                }]
                        },
                        "limits": [{
                                "id": "john-limit-1",
                                "channel": "ATM",
                                "frequency": "DAILY",
                                "amount": 3000.0,
                                "maxAmount": 9000.0,
                                "minAmount": 0.0
                            }, {
                                "id": "john-limit-2",
                                "channel": "ONLINE",
                                "frequency": "DAILY",
                                "amount": 20000.0,
                                "maxAmount": 9000.0,
                                "minAmount": 0.0
                            }]
                    }, {
                        "id": "john-card-2",
                        "brand": "mastercard",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Gold",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "John Doe"
                        },
                        "expiryDate": {
                            "year": "2031",
                            "month": "02"
                        },
                        "currency": "EUR",
                        "maskedNumber": "8299"
                    }]
            },
            {
                status: 200,
                body: [{
                        "id": "kim-card-1",
                        "brand": "mastercard",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Copper",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Kim Sook"
                        },
                        "expiryDate": {
                            "year": "2027",
                            "month": "03"
                        },
                        "currency": "EUR",
                        "maskedNumber": "9292"
                    }, {
                        "id": "kim-card-2",
                        "brand": "visa",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Iron",
                        "status": "Inactive",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Kim Sook"
                        },
                        "expiryDate": {
                            "year": "2010",
                            "month": "04"
                        },
                        "currency": "EUR",
                        "maskedNumber": "9494"
                    }, {
                        "id": "kim-card-3",
                        "brand": "visa",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Iron",
                        "status": "Expired",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Kim Sook"
                        },
                        "expiryDate": {
                            "year": "2010",
                            "month": "04"
                        },
                        "currency": "EUR",
                        "maskedNumber": "9494"
                    }]
            },
            {
                status: 200,
                body: []
            },
            {
                status: 200,
                body: [{
                        "id": "current-card-1",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Jason Carter"
                        },
                        "expiryDate": {
                            "year": "2050",
                            "month": "12"
                        },
                        "currency": "EUR",
                        "maskedNumber": "5235",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-19T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "PENDING"
                                }, {
                                    "name": "In Transit",
                                    "status": "PENDING"
                                }, {
                                    "name": "Delivered",
                                    "status": "PENDING",
                                    "stepDateTime": "2019-06-20T14:21:25.001Z"
                                }]
                        }
                    }, {
                        "id": "current-card-2",
                        "brand": "visa",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Green Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Jessie Carter"
                        },
                        "expiryDate": {
                            "year": "2045",
                            "month": "01"
                        },
                        "currency": "EUR",
                        "maskedNumber": "6782"
                    }, {
                        "id": "current-card-3",
                        "brand": "american express",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Blue Card",
                        "status": "Inactive",
                        "lockStatus": "LOCKED",
                        "replacement": {
                            "status": "UnderReplacement",
                            "reason": "Lost",
                            "replacedById": "4"
                        },
                        "holder": {
                            "name": "Jason Carter"
                        },
                        "expiryDate": {
                            "year": "2048",
                            "month": "06"
                        },
                        "currency": "EUR",
                        "maskedNumber": "1262"
                    }, {
                        "id": "current-card-4",
                        "brand": "union pay",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Black Card",
                        "status": "Inactive",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Jessie Carter"
                        },
                        "expiryDate": {
                            "year": "2048",
                            "month": "07"
                        },
                        "currency": "GBP",
                        "maskedNumber": "1289"
                    }, {
                        "id": "current-card-5",
                        "brand": "maestro",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Platinum",
                        "status": "Active",
                        "lockStatus": "LOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Jessie Carter"
                        },
                        "expiryDate": {
                            "year": "2050",
                            "month": "11"
                        },
                        "currency": "USD",
                        "maskedNumber": "4627",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-19T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-20T14:32:20.906Z"
                                }, {
                                    "name": "In Transit",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-20T15:12:20.906Z"
                                }, {
                                    "name": "Delivered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-21T11:28:25.001Z"
                                }]
                        }
                    }, {
                        "id": "current-card-6",
                        "brand": "diners club",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "White Card",
                        "status": "Active",
                        "lockStatus": "LOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Lisa Carter"
                        },
                        "expiryDate": {
                            "year": "2047",
                            "month": "03"
                        },
                        "currency": "GBP",
                        "maskedNumber": "8291"
                    }, {
                        "id": "current-card-7",
                        "brand": "visa debit",
                        "type": "debit",
                        "subType": "ATM",
                        "name": "Platinum Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Randy Carter"
                        },
                        "expiryDate": {
                            "year": "2047",
                            "month": "03"
                        },
                        "currency": "EUR",
                        "maskedNumber": "8291"
                    }, {
                        "id": "current-card-8",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Jason Carter"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5235"
                    }, {
                        "id": "current-card-9",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Active",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "ReplacementRequested",
                            "reason": "Damaged"
                        },
                        "holder": {
                            "name": "Jessie Carter"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5235",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2021-07-01T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "PENDING"
                                }]
                        }
                    }, {
                        "id": "current-card-10",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Green Card",
                        "status": "Active",
                        "lockStatus": "LOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Jason Carter"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5235"
                    }, {
                        "id": "current-card-11",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Inactive",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "Replaced",
                            "reason": "Lost"
                        },
                        "holder": {
                            "name": "Lisa Carter"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5235",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2021-07-01T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "PENDING"
                                }]
                        }
                    }, {
                        "id": "current-card-12",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Inactive",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "Replacing",
                            "reason": "Stolen"
                        },
                        "holder": {
                            "name": "Lisa Carter"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5235"
                    }, {
                        "id": "current-card-13",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Unknown",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Lisa Carter"
                        },
                        "expiryDate": {
                            "year": "2030",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5299"
                    }, {
                        "id": "current-card-13",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Fraud",
                        "lockStatus": "LOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Jason Carter"
                        },
                        "expiryDate": {
                            "year": "2035",
                            "month": "01"
                        },
                        "currency": "GBP",
                        "maskedNumber": "1399"
                    }]
            },
            {
                status: 200,
                body: [{
                        "id": "closed-card-1",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Canceled",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Jason Carter"
                        },
                        "expiryDate": {
                            "year": "2050",
                            "month": "12"
                        },
                        "currency": "EUR",
                        "maskedNumber": "5235",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-19T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "PENDING"
                                }, {
                                    "name": "In Transit",
                                    "status": "PENDING"
                                }, {
                                    "name": "Delivered",
                                    "status": "PENDING",
                                    "stepDateTime": "2019-06-20T14:21:25.001Z"
                                }]
                        }
                    }, {
                        "id": "closed-card-2",
                        "brand": "visa",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Green Card",
                        "status": "Canceled",
                        "lockStatus": "LOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Jessie Carter"
                        },
                        "expiryDate": {
                            "year": "2045",
                            "month": "01"
                        },
                        "currency": "EUR",
                        "maskedNumber": "6782"
                    }, {
                        "id": "closed-card-3",
                        "brand": "american express",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Blue Card",
                        "status": "Expired",
                        "lockStatus": "LOCKED",
                        "replacement": {
                            "status": "UnderReplacement",
                            "reason": "Lost",
                            "replacedById": "4"
                        },
                        "holder": {
                            "name": "Jason Carter"
                        },
                        "expiryDate": {
                            "year": "2048",
                            "month": "06"
                        },
                        "currency": "EUR",
                        "maskedNumber": "1262"
                    }, {
                        "id": "closed-card-4",
                        "brand": "union pay",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Black Card",
                        "status": "Expired",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Jessie Carter"
                        },
                        "expiryDate": {
                            "year": "2048",
                            "month": "07"
                        },
                        "currency": "GBP",
                        "maskedNumber": "1289"
                    }, {
                        "id": "closed-card-5",
                        "brand": "maestro",
                        "type": "Debit",
                        "subType": "ATM",
                        "name": "Platinum",
                        "status": "Expired",
                        "lockStatus": "LOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Jessie Carter"
                        },
                        "expiryDate": {
                            "year": "2050",
                            "month": "11"
                        },
                        "currency": "USD",
                        "maskedNumber": "4627",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-19T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-20T14:32:20.906Z"
                                }, {
                                    "name": "In Transit",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-20T15:12:20.906Z"
                                }, {
                                    "name": "Delivered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2019-06-21T11:28:25.001Z"
                                }]
                        }
                    }, {
                        "id": "closed-card-6",
                        "brand": "diners club",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "White Card",
                        "status": "Canceled",
                        "lockStatus": "LOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Lisa Carter"
                        },
                        "expiryDate": {
                            "year": "2047",
                            "month": "03"
                        },
                        "currency": "GBP",
                        "maskedNumber": "8291"
                    }, {
                        "id": "closed-card-7",
                        "brand": "visa debit",
                        "type": "debit",
                        "subType": "ATM",
                        "name": "Platinum Card",
                        "status": "Canceled",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "NotUnderReplacement"
                        },
                        "holder": {
                            "name": "Randy Carter"
                        },
                        "expiryDate": {
                            "year": "2047",
                            "month": "03"
                        },
                        "currency": "EUR",
                        "maskedNumber": "8291"
                    }, {
                        "id": "closed-card-8",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Canceled",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "Replaced",
                            "reason": "Lost"
                        },
                        "holder": {
                            "name": "Jason Carter"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5235",
                        "limits": [{
                                "id": "closed-limit-1",
                                "channel": "ATM",
                                "frequency": "DAILY",
                                "amount": 3000.0,
                                "maxAmount": 10000.0,
                                "minAmount": 0.0
                            }, {
                                "id": "closed-limit-2",
                                "channel": "ONLINE",
                                "frequency": "DAILY",
                                "amount": 2000.0,
                                "maxAmount": 15000.0,
                                "minAmount": 0.0
                            }],
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2021-07-01T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "PENDING"
                                }]
                        }
                    }, {
                        "id": "closed-card-9",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Canceled",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "ReplacementRequested",
                            "reason": "Lost"
                        },
                        "holder": {
                            "name": "Lisa Carter"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5235",
                        "limits": [{
                                "id": "closed-limit-1",
                                "channel": "ATM",
                                "frequency": "DAILY",
                                "amount": 3000.0,
                                "maxAmount": 10000.0,
                                "minAmount": 0.0
                            }, {
                                "id": "closed-limit-2",
                                "channel": "ONLINE",
                                "frequency": "DAILY",
                                "amount": 2000.0,
                                "maxAmount": 15000.0,
                                "minAmount": 0.0
                            }],
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2021-07-01T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "PENDING"
                                }]
                        }
                    }, {
                        "id": "closed-card-10",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Canceled",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "ReplacementRequested",
                            "reason": "Lost"
                        },
                        "holder": {
                            "name": "Randy Carter"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5235",
                        "delivery": {
                            "transitSteps": [{
                                    "name": "Ordered",
                                    "status": "SUCCESS",
                                    "stepDateTime": "2021-07-01T12:52:20.906Z"
                                }, {
                                    "name": "Processed",
                                    "status": "PENDING"
                                }]
                        }
                    }, {
                        "id": "closed-card-11",
                        "brand": "mastercard",
                        "type": "Credit",
                        "subType": "ATM",
                        "name": "Gold Card",
                        "status": "Canceled",
                        "lockStatus": "UNLOCKED",
                        "replacement": {
                            "status": "Replacing",
                            "reason": "Stolen"
                        },
                        "holder": {
                            "name": "Lisa Carter"
                        },
                        "expiryDate": {
                            "year": "2022",
                            "month": "12"
                        },
                        "currency": "USD",
                        "maskedNumber": "5235"
                    }]
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/employee/users/{userId}/cards/{cardId}/replacement",
        method: "POST",
        responses: [
            {
                status: 201,
                body: {
                    "id": "sara-card-1",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold Card",
                    "status": "Active",
                    "statusEnriched": "OPEN",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Sara Williams"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5235",
                    "limits": [{
                            "id": "sara-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "sara-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 201,
                body: {
                    "id": "elon-card-1",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "World Elite",
                    "status": "Active",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "ReplacementRequested",
                        "reason": "Damaged"
                    },
                    "holder": {
                        "name": "Elon"
                    },
                    "expiryDate": {
                        "year": "2050",
                        "month": "12"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5235",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "PENDING"
                            }]
                    }
                }
            },
            {
                status: 201,
                body: {
                    "id": "john-card-1",
                    "brand": "visa",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold",
                    "status": "Inactive",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "ReplacementRequested",
                        "reason": "Stolen"
                    },
                    "holder": {
                        "name": "John Doe"
                    },
                    "expiryDate": {
                        "year": "2030",
                        "month": "06"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5827",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "SUCCESS"
                            }, {
                                "name": "In Transit",
                                "status": "SUCCESS"
                            }, {
                                "name": "Delivered",
                                "status": "SUCCESS"
                            }]
                    }
                }
            },
            {
                status: 201,
                body: {
                    "id": "kim-card-1",
                    "brand": "mastercard",
                    "type": "Debit",
                    "subType": "ATM",
                    "name": "Copper",
                    "status": "Inactive",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "ReplacementRequested",
                        "reason": "Lost"
                    },
                    "holder": {
                        "name": "Kim Sook"
                    },
                    "expiryDate": {
                        "year": "2027",
                        "month": "03"
                    },
                    "currency": "EUR",
                    "maskedNumber": "9292"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/employee/users/{userId}/cards/{cardId}/lock-status",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {
                    "id": "sara-card-1",
                    "brand": "mastercard",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold Card",
                    "status": "Active",
                    "statusEnriched": "OPEN",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Sara Williams"
                    },
                    "expiryDate": {
                        "year": "2022",
                        "month": "12"
                    },
                    "currency": "USD",
                    "maskedNumber": "5235",
                    "limits": [{
                            "id": "sara-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 3000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "sara-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 2000.0,
                            "maxAmount": 10000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "elon-card-6",
                    "brand": "american_express",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Reserve",
                    "status": "Active",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Elon"
                    },
                    "expiryDate": {
                        "year": "2047",
                        "month": "03"
                    },
                    "currency": "EUR",
                    "maskedNumber": "8291"
                }
            },
            {
                status: 200,
                body: {
                    "id": "john-card-1",
                    "brand": "visa",
                    "type": "Credit",
                    "subType": "ATM",
                    "name": "Gold",
                    "status": "Active",
                    "lockStatus": "LOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "John Doe"
                    },
                    "expiryDate": {
                        "year": "2030",
                        "month": "06"
                    },
                    "currency": "EUR",
                    "maskedNumber": "5827",
                    "delivery": {
                        "transitSteps": [{
                                "name": "Ordered",
                                "status": "SUCCESS",
                                "stepDateTime": "2019-06-19T12:52:20.906Z"
                            }, {
                                "name": "Processed",
                                "status": "SUCCESS"
                            }, {
                                "name": "In Transit",
                                "status": "SUCCESS"
                            }, {
                                "name": "Delivered",
                                "status": "SUCCESS"
                            }]
                    },
                    "limits": [{
                            "id": "john-limit-1",
                            "channel": "ATM",
                            "frequency": "DAILY",
                            "amount": 5000.0,
                            "maxAmount": 1000.0,
                            "minAmount": 0.0
                        }, {
                            "id": "john-limit-2",
                            "channel": "ONLINE",
                            "frequency": "DAILY",
                            "amount": 20000.0,
                            "maxAmount": 9000.0,
                            "minAmount": 0.0
                        }]
                }
            },
            {
                status: 200,
                body: {
                    "id": "kim-card-1",
                    "brand": "mastercard",
                    "type": "Debit",
                    "subType": "ATM",
                    "name": "Copper",
                    "status": "Inactive",
                    "statusEnriched": "OPEN",
                    "lockStatus": "UNLOCKED",
                    "replacement": {
                        "status": "NotUnderReplacement"
                    },
                    "holder": {
                        "name": "Kim Sook"
                    },
                    "expiryDate": {
                        "year": "2027",
                        "month": "03"
                    },
                    "currency": "EUR",
                    "maskedNumber": "9292"
                }
            },
        ]
    },
]);
