// @backbase/employee-http-ang/esm2015/api/cards.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { EMPLOYEE_BASE_PATH } from '../variables';
import { EmployeeConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class CardsService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new EmployeeConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    activate(requestParameters, observe = 'body', reportProgress = false, options) {
        const _userId = requestParameters["userId"];
        if (_userId === null || _userId === undefined) {
            throw new Error('Required parameter userId was null or undefined when calling activate.');
        }
        const _cardId = requestParameters["cardId"];
        if (_cardId === null || _cardId === undefined) {
            throw new Error('Required parameter cardId was null or undefined when calling activate.');
        }
        const _body = requestParameters["body"];
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v1/employee/users/${encodeURIComponent(String(_userId))}/cards/${encodeURIComponent(String(_cardId))}/activation`, _body, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    activateUrl(requestParameters) {
        const _userId = requestParameters["userId"];
        if (_userId === null || _userId === undefined) {
            throw new Error('Required parameter userId was null or undefined when calling activate.');
        }
        const _cardId = requestParameters["cardId"];
        if (_cardId === null || _cardId === undefined) {
            throw new Error('Required parameter cardId was null or undefined when calling activate.');
        }
        const _body = requestParameters["body"];
        return `${this.configuration.basePath}/client-api/v1/employee/users/${encodeURIComponent(String(_userId))}/cards/${encodeURIComponent(String(_cardId))}/activation`;
    }
    changeLimits(requestParameters, observe = 'body', reportProgress = false, options) {
        const _userId = requestParameters["userId"];
        if (_userId === null || _userId === undefined) {
            throw new Error('Required parameter userId was null or undefined when calling changeLimits.');
        }
        const _cardId = requestParameters["cardId"];
        if (_cardId === null || _cardId === undefined) {
            throw new Error('Required parameter cardId was null or undefined when calling changeLimits.');
        }
        const _body = requestParameters["body"];
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v1/employee/users/${encodeURIComponent(String(_userId))}/cards/${encodeURIComponent(String(_cardId))}/limits`, _body, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    changeLimitsUrl(requestParameters) {
        const _userId = requestParameters["userId"];
        if (_userId === null || _userId === undefined) {
            throw new Error('Required parameter userId was null or undefined when calling changeLimits.');
        }
        const _cardId = requestParameters["cardId"];
        if (_cardId === null || _cardId === undefined) {
            throw new Error('Required parameter cardId was null or undefined when calling changeLimits.');
        }
        const _body = requestParameters["body"];
        return `${this.configuration.basePath}/client-api/v1/employee/users/${encodeURIComponent(String(_userId))}/cards/${encodeURIComponent(String(_cardId))}/limits`;
    }
    getCardById(requestParameters, observe = 'body', reportProgress = false, options) {
        const _userId = requestParameters["userId"];
        if (_userId === null || _userId === undefined) {
            throw new Error('Required parameter userId was null or undefined when calling getCardById.');
        }
        const _cardId = requestParameters["cardId"];
        if (_cardId === null || _cardId === undefined) {
            throw new Error('Required parameter cardId was null or undefined when calling getCardById.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/employee/users/${encodeURIComponent(String(_userId))}/cards/${encodeURIComponent(String(_cardId))}`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getCardByIdUrl(requestParameters) {
        const _userId = requestParameters["userId"];
        if (_userId === null || _userId === undefined) {
            throw new Error('Required parameter userId was null or undefined when calling getCardById.');
        }
        const _cardId = requestParameters["cardId"];
        if (_cardId === null || _cardId === undefined) {
            throw new Error('Required parameter cardId was null or undefined when calling getCardById.');
        }
        return `${this.configuration.basePath}/client-api/v1/employee/users/${encodeURIComponent(String(_userId))}/cards/${encodeURIComponent(String(_cardId))}`;
    }
    getCards(requestParameters, observe = 'body', reportProgress = false, options) {
        const _userId = requestParameters["userId"];
        if (_userId === null || _userId === undefined) {
            throw new Error('Required parameter userId was null or undefined when calling getCards.');
        }
        const _cardIds = requestParameters["cardIds"];
        const _status = requestParameters["status"];
        const _statusEnriched = requestParameters["statusEnriched"];
        const _types = requestParameters["types"];
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_cardIds) {
            _cardIds.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'cardIds');
            });
        }
        if (_status) {
            _status.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'status');
            });
        }
        if (_statusEnriched !== undefined && _statusEnriched !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _statusEnriched, 'statusEnriched');
        }
        if (_types) {
            _types.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'types');
            });
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/employee/users/${encodeURIComponent(String(_userId))}/cards`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getCardsUrl(requestParameters) {
        const _userId = requestParameters["userId"];
        if (_userId === null || _userId === undefined) {
            throw new Error('Required parameter userId was null or undefined when calling getCards.');
        }
        const _cardIds = requestParameters["cardIds"];
        const _status = requestParameters["status"];
        const _statusEnriched = requestParameters["statusEnriched"];
        const _types = requestParameters["types"];
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_cardIds) {
            _cardIds.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'cardIds');
            });
        }
        if (_status) {
            _status.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'status');
            });
        }
        if (_statusEnriched !== undefined && _statusEnriched !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _statusEnriched, 'statusEnriched');
        }
        if (_types) {
            _types.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'types');
            });
        }
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v1/employee/users/${encodeURIComponent(String(_userId))}/cards${queryString ? `?${queryString}` : ''}`;
    }
    requestReplacement(requestParameters, observe = 'body', reportProgress = false, options) {
        const _userId = requestParameters["userId"];
        if (_userId === null || _userId === undefined) {
            throw new Error('Required parameter userId was null or undefined when calling requestReplacement.');
        }
        const _cardId = requestParameters["cardId"];
        if (_cardId === null || _cardId === undefined) {
            throw new Error('Required parameter cardId was null or undefined when calling requestReplacement.');
        }
        const _body = requestParameters["body"];
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v1/employee/users/${encodeURIComponent(String(_userId))}/cards/${encodeURIComponent(String(_cardId))}/replacement`, _body, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    requestReplacementUrl(requestParameters) {
        const _userId = requestParameters["userId"];
        if (_userId === null || _userId === undefined) {
            throw new Error('Required parameter userId was null or undefined when calling requestReplacement.');
        }
        const _cardId = requestParameters["cardId"];
        if (_cardId === null || _cardId === undefined) {
            throw new Error('Required parameter cardId was null or undefined when calling requestReplacement.');
        }
        const _body = requestParameters["body"];
        return `${this.configuration.basePath}/client-api/v1/employee/users/${encodeURIComponent(String(_userId))}/cards/${encodeURIComponent(String(_cardId))}/replacement`;
    }
    updateLockStatus(requestParameters, observe = 'body', reportProgress = false, options) {
        const _userId = requestParameters["userId"];
        if (_userId === null || _userId === undefined) {
            throw new Error('Required parameter userId was null or undefined when calling updateLockStatus.');
        }
        const _cardId = requestParameters["cardId"];
        if (_cardId === null || _cardId === undefined) {
            throw new Error('Required parameter cardId was null or undefined when calling updateLockStatus.');
        }
        const _body = requestParameters["body"];
        if (_body === null || _body === undefined) {
            throw new Error('Required parameter body was null or undefined when calling updateLockStatus.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v1/employee/users/${encodeURIComponent(String(_userId))}/cards/${encodeURIComponent(String(_cardId))}/lock-status`, _body, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    updateLockStatusUrl(requestParameters) {
        const _userId = requestParameters["userId"];
        if (_userId === null || _userId === undefined) {
            throw new Error('Required parameter userId was null or undefined when calling updateLockStatus.');
        }
        const _cardId = requestParameters["cardId"];
        if (_cardId === null || _cardId === undefined) {
            throw new Error('Required parameter cardId was null or undefined when calling updateLockStatus.');
        }
        const _body = requestParameters["body"];
        if (_body === null || _body === undefined) {
            throw new Error('Required parameter body was null or undefined when calling updateLockStatus.');
        }
        return `${this.configuration.basePath}/client-api/v1/employee/users/${encodeURIComponent(String(_userId))}/cards/${encodeURIComponent(String(_cardId))}/lock-status`;
    }
}
CardsService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardsService, deps: [{ token: i1.HttpClient }, { token: EMPLOYEE_BASE_PATH, optional: true }, { token: i2.EmployeeConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
CardsService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardsService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CardsService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [EMPLOYEE_BASE_PATH]
                }] }, { type: i2.EmployeeConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
