// @backbase/employee-http-ang/esm2015/api/serviceAgreements.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const ServiceAgreementsServiceGetServiceAgreementsMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/employee/users/{userId}/service-agreements",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "serviceAgreements": [{
                            "name": "SA Name 1",
                            "id": "8b4fc047-c98a-4e58-975a-6d3f2bde4ebe",
                            "master": true,
                            "description": "SA 1 Description"
                        }, {
                            "name": "SA Name 2",
                            "id": "8b4fc047-c98a-4e58-975a-6d3f222de4ebe",
                            "master": true
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
        ]
    }]);
export const ServiceAgreementsServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v1/employee/users/{userId}/service-agreements",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "serviceAgreements": [{
                            "name": "SA Name 1",
                            "id": "8b4fc047-c98a-4e58-975a-6d3f2bde4ebe",
                            "master": true,
                            "description": "SA 1 Description"
                        }, {
                            "name": "SA Name 2",
                            "id": "8b4fc047-c98a-4e58-975a-6d3f222de4ebe",
                            "master": true
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
        ]
    },
]);
