// @backbase/employee-http-ang/esm2015/api/smartInsights.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const SmartInsightsServiceGetSmartInsightByIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/employee/users/{userId}/smart-insights",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "userAccount": {
                        "locked": false
                    },
                    "devices": {
                        "lockedDevices": [{
                                "friendlyName": "Mark's Phone",
                                "vendor": "Google",
                                "model": "Pixel 3",
                                "id": "4b1a0300-ea8f-4008-a6bd-e1c6cea5f93e"
                            }, {
                                "friendlyName": "Mark's Work Phone",
                                "vendor": "Apple",
                                "model": "12",
                                "id": "b7e7c546-60f0-4234-aeb8-021b88e73880"
                            }]
                    },
                    "actions": {
                        "requiredActions": ["VERIFY_EMAIL", "ACCEPT_TERMS_AND_CONDITIONS"]
                    }
                }
            },
            {
                status: 200,
                body: {
                    "userAccount": {
                        "errors": [{
                                "message": "Internal Service Error"
                            }]
                    },
                    "devices": {
                        "lockedDevices": [{
                                "friendlyName": "Mark's Phone",
                                "vendor": "Google",
                                "model": "Pixel 3",
                                "id": "4b1a0300-ea8f-4008-a6bd-e1c6cea5f93e"
                            }, {
                                "friendlyName": "Mark's Work Phone",
                                "vendor": "Apple",
                                "model": "12",
                                "id": "b7e7c546-60f0-4234-aeb8-021b88e73880"
                            }]
                    },
                    "actions": {
                        "errors": [{
                                "message": "Internal Service Error"
                            }]
                    }
                }
            },
            {
                status: 200,
                body: {
                    "userAccount": {
                        "locked": true
                    },
                    "devices": {
                        "errors": [{
                                "message": "Internal Service Error"
                            }]
                    },
                    "actions": {
                        "requiredActions": ["VERIFY_EMAIL", "ACCEPT_TERMS_AND_CONDITIONS"]
                    }
                }
            },
            {
                status: 200,
                body: {
                    "userAccountLocked": {
                        "locked": false
                    },
                    "devices": {
                        "lockedDevices": []
                    },
                    "actions": {
                        "requiredActions": []
                    }
                }
            },
        ]
    }]);
export const SmartInsightsServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v1/employee/users/{userId}/smart-insights",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "userAccount": {
                        "locked": false
                    },
                    "devices": {
                        "lockedDevices": [{
                                "friendlyName": "Mark's Phone",
                                "vendor": "Google",
                                "model": "Pixel 3",
                                "id": "4b1a0300-ea8f-4008-a6bd-e1c6cea5f93e"
                            }, {
                                "friendlyName": "Mark's Work Phone",
                                "vendor": "Apple",
                                "model": "12",
                                "id": "b7e7c546-60f0-4234-aeb8-021b88e73880"
                            }]
                    },
                    "actions": {
                        "requiredActions": ["VERIFY_EMAIL", "ACCEPT_TERMS_AND_CONDITIONS"]
                    }
                }
            },
            {
                status: 200,
                body: {
                    "userAccount": {
                        "errors": [{
                                "message": "Internal Service Error"
                            }]
                    },
                    "devices": {
                        "lockedDevices": [{
                                "friendlyName": "Mark's Phone",
                                "vendor": "Google",
                                "model": "Pixel 3",
                                "id": "4b1a0300-ea8f-4008-a6bd-e1c6cea5f93e"
                            }, {
                                "friendlyName": "Mark's Work Phone",
                                "vendor": "Apple",
                                "model": "12",
                                "id": "b7e7c546-60f0-4234-aeb8-021b88e73880"
                            }]
                    },
                    "actions": {
                        "errors": [{
                                "message": "Internal Service Error"
                            }]
                    }
                }
            },
            {
                status: 200,
                body: {
                    "userAccount": {
                        "locked": true
                    },
                    "devices": {
                        "errors": [{
                                "message": "Internal Service Error"
                            }]
                    },
                    "actions": {
                        "requiredActions": ["VERIFY_EMAIL", "ACCEPT_TERMS_AND_CONDITIONS"]
                    }
                }
            },
            {
                status: 200,
                body: {
                    "userAccountLocked": {
                        "locked": false
                    },
                    "devices": {
                        "lockedDevices": []
                    },
                    "actions": {
                        "requiredActions": []
                    }
                }
            },
        ]
    },
]);
