// @backbase/employee-http-ang/esm2015/api/transactions.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const TransactionsServiceGetTransactionMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/employee/users/{userId}/service-agreements/{serviceAgreementId}/transactions/{transactionId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c371",
                    "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                    "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                    "description": "www.starbucks.com",
                    "typeGroup": "Payment",
                    "type": "Card Payment",
                    "category": "Food Drinks",
                    "bookingDate": "2019-05-29",
                    "valueDate": "2019-05-29",
                    "creditDebitIndicator": "DBIT",
                    "transactionAmountCurrency": {
                        "amount": "7.5",
                        "currencyCode": "EUR"
                    },
                    "counterPartyName": "Starbucks",
                    "counterPartyAccountNumber": "NL91ABNA0423164312",
                    "counterPartyCity": "Amsterdam",
                    "counterPartyAddress": "Zuidplein 10, 1077 XV",
                    "additions": {
                        "settlementDate": "2020-06-01",
                        "additionalCharges": "183.56",
                        "purpose": "vacation",
                        "purposeCode": "1",
                        "fees": "62",
                        "feeReferenceId": "FEE-01",
                        "ultimateDebtorName": "John Doe",
                        "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                        "ultimateDebtorId": "e7a2da343421",
                        "ultimateCreditorId": "df8e1cac",
                        "ultimateCreditorName": "Julia Pence",
                        "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                        "MMCCode": "V"
                    }
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    }]);
export const TransactionsServiceGetTransactionsMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/employee/users/{userId}/service-agreements/{serviceAgreementId}/transactions",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c371",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "www.starbucks.com",
                        "typeGroup": "Payment",
                        "type": "Card Payment",
                        "category": "Food Drinks",
                        "bookingDate": "2019-05-29",
                        "valueDate": "2019-05-29",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "7.5",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Starbucks",
                        "counterPartyAccountNumber": "NL91ABNA0423164312",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Zuidplein 10, 1077 XV",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c372",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "www.medimarket.com",
                        "typeGroup": "Payment",
                        "type": "Card Payment",
                        "category": "Shopping",
                        "bookingDate": "2019-05-25",
                        "valueDate": "2019-05-25",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "320.95",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "MediaMarkt",
                        "counterPartyAccountNumber": "NL91ABNA0541164300",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Zuidplein 10, 1077 XV",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c373",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "albertheijn.nl",
                        "typeGroup": "Payment",
                        "type": "Card Payment",
                        "category": "Shopping",
                        "bookingDate": "2019-05-20",
                        "valueDate": "2019-05-20",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "919.95",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Albert Heijn",
                        "counterPartyAccountNumber": "BE68539007547034",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Stationsplein, 1012 AB",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c3731214",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Internet Banking",
                        "typeGroup": "Payment",
                        "type": "International Payment",
                        "category": "Transfers",
                        "bookingDate": "2019-05-15",
                        "valueDate": "2019-05-16",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "887.31",
                            "currencyCode": "EUR"
                        },
                        "currencyExchangeRate": 0.8945,
                        "counterPartyName": "Tusk Williams B.V",
                        "counterPartyAccountNumber": "BE68539007547034",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Stationsplein, 1012 AB",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c373c342",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Internet Banking",
                        "typeGroup": "Payment",
                        "type": "SEPA Payment",
                        "category": "Income",
                        "bookingDate": "2019-05-22",
                        "valueDate": "2019-05-22",
                        "creditDebitIndicator": "CRDT",
                        "transactionAmountCurrency": {
                            "amount": "1919.95",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Backbase B.V",
                        "counterPartyAccountNumber": "NL40ABNA0541164350",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Jacob Bontiusplaats 9, 1018 LL",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c374",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Internet Banking",
                        "typeGroup": "Payment",
                        "type": "SEPA Payment",
                        "category": "Shopping",
                        "bookingDate": "2019-04-28",
                        "valueDate": "2019-04-28",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "111.95",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Eneco Bank11",
                        "counterPartyAccountNumber": "BA391290079401028494",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Stationsplein, 1012 AB",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c375",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "albertheijn.nl",
                        "typeGroup": "Payment",
                        "type": "Card Payment",
                        "category": "Shopping",
                        "bookingDate": "2019-04-24",
                        "valueDate": "2019-08-24",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "86.54",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Albert Heijn",
                        "counterPartyAccountNumber": "BG80BNBG96611020345678",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Stationsplein, 1012 AB",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c376",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Standing Order",
                        "typeGroup": "Payment",
                        "type": "SEPA Payment",
                        "category": "Other Income",
                        "bookingDate": "2019-04-15",
                        "valueDate": "2019-04-15",
                        "creditDebitIndicator": "CRDT",
                        "transactionAmountCurrency": {
                            "amount": "5200",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Simons & Co. B.V.",
                        "counterPartyAccountNumber": "HR1210010051863000160",
                        "counterPartyCity": "Utrecht",
                        "counterPartyAddress": "Vendelier 71. 3905PD",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c377",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Internet Banking",
                        "typeGroup": "Payment",
                        "type": "Payment Sent",
                        "category": "Transfers",
                        "bookingDate": "2019-04-15",
                        "valueDate": "2019-04-15",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "10",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Sally Lambert",
                        "counterPartyAccountNumber": "CY17002001280000001200527600",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Stationsplein, 1012 AB",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c378",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "www.medimarket.com",
                        "typeGroup": "Payment",
                        "type": "Card Payment",
                        "category": "Shopping",
                        "bookingDate": "2019-03-13",
                        "valueDate": "2019-03-13",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "199.95",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "MediaMarkt",
                        "counterPartyAccountNumber": "CZ6508000000192000145399",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Stationsplein, 1012 AB",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c379",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Internet Banking",
                        "typeGroup": "Payment",
                        "type": "SEPA Payment",
                        "category": "Transfers",
                        "bookingDate": "2019-02-13",
                        "valueDate": "2019-02-13",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "25",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "John White",
                        "counterPartyAccountNumber": "DK5000400440116243",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Stationsplein, 1012 AB",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c380",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "www.starbucks.com",
                        "typeGroup": "Payment",
                        "type": "Card Payment",
                        "category": "Food Drinks",
                        "bookingDate": "2019-02-13",
                        "valueDate": "2019-02-13",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "7.5",
                            "currencyCode": "EUR"
                        },
                        "counterPartyAddress": "Zuidplein 10, 1077 XV",
                        "counterPartyCity": "Amsterdam",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c381",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Internet Banking",
                        "typeGroup": "Payment",
                        "type": "Card Payment",
                        "category": "Shopping",
                        "bookingDate": "2019-02-13",
                        "valueDate": "2019-02-14",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "66.54",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Eneco Bank8",
                        "counterPartyAccountNumber": "FO1464600009692713",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Stationsplein, 1012 AB",
                        "imageType": "albert",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c382",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Standing Order",
                        "typeGroup": "Payment",
                        "type": "SEPA Payment",
                        "category": "Category",
                        "bookingDate": "2019-01-10",
                        "valueDate": "2019-01-10",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "400",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Simons & Co. B.V.",
                        "counterPartyAccountNumber": "FI2112345600000785",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Stationsplein, 1012 AB",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c383",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Internet Banking",
                        "typeGroup": "Payment",
                        "type": "SEPA Payment",
                        "category": "Transfers",
                        "bookingDate": "2018-10-01",
                        "valueDate": "2018-10-01",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "31",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Sally Lambert",
                        "counterPartyAccountNumber": "FR1420041010050500013M02606",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Stationsplein, 1012 AB",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c384",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Standing Order",
                        "typeGroup": "Payment",
                        "type": "Payment Received",
                        "category": "Other Income",
                        "bookingDate": "2018-08-15",
                        "valueDate": "2018-08-15",
                        "creditDebitIndicator": "CRDT",
                        "transactionAmountCurrency": {
                            "amount": "25",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Eneco Bank12",
                        "counterPartyAccountNumber": "GE29NB0000000101904917",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Stationsplein, 1012 AB",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c385",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Internet Banking",
                        "typeGroup": "Payment",
                        "type": "SEPA Payment",
                        "category": "Transfers",
                        "bookingDate": "2018-10-13",
                        "valueDate": "2018-08-15",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "640",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Sally Lambert",
                        "counterPartyAccountNumber": "DE89370400440532013000",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Stationsplein, 1012 AB",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c386",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Internet Banking",
                        "typeGroup": "Payment",
                        "type": "SEPA Payment",
                        "category": "Income",
                        "bookingDate": "2018-07-13",
                        "valueDate": "2018-07-15",
                        "creditDebitIndicator": "CRDT",
                        "transactionAmountCurrency": {
                            "amount": "7200",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Simons & Co. B.V.",
                        "counterPartyAccountNumber": "GI75NWBK000000007099453",
                        "counterPartyCity": "Utrecht",
                        "counterPartyAddress": "Vendelier 71. 3905PD",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c387",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Internet Banking",
                        "typeGroup": "Payment",
                        "type": "SEPA Payment",
                        "bookingDate": "2018-07-13",
                        "valueDate": "2018-07-15",
                        "creditDebitIndicator": "CRDT",
                        "transactionAmountCurrency": {
                            "amount": "7200",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Simons & Co. B.V.",
                        "counterPartyAccountNumber": "GI75NWBK000000007099453",
                        "counterPartyCity": "Utrecht",
                        "counterPartyAddress": "Vendelier 71. 3905PD",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
        ]
    }]);
export const TransactionsServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v1/employee/users/{userId}/service-agreements/{serviceAgreementId}/transactions/{transactionId}",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c371",
                    "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                    "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                    "description": "www.starbucks.com",
                    "typeGroup": "Payment",
                    "type": "Card Payment",
                    "category": "Food Drinks",
                    "bookingDate": "2019-05-29",
                    "valueDate": "2019-05-29",
                    "creditDebitIndicator": "DBIT",
                    "transactionAmountCurrency": {
                        "amount": "7.5",
                        "currencyCode": "EUR"
                    },
                    "counterPartyName": "Starbucks",
                    "counterPartyAccountNumber": "NL91ABNA0423164312",
                    "counterPartyCity": "Amsterdam",
                    "counterPartyAddress": "Zuidplein 10, 1077 XV",
                    "additions": {
                        "settlementDate": "2020-06-01",
                        "additionalCharges": "183.56",
                        "purpose": "vacation",
                        "purposeCode": "1",
                        "fees": "62",
                        "feeReferenceId": "FEE-01",
                        "ultimateDebtorName": "John Doe",
                        "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                        "ultimateDebtorId": "e7a2da343421",
                        "ultimateCreditorId": "df8e1cac",
                        "ultimateCreditorName": "Julia Pence",
                        "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                        "MMCCode": "V"
                    }
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/employee/users/{userId}/service-agreements/{serviceAgreementId}/transactions",
        method: "GET",
        responses: [
            {
                status: 200,
                body: [{
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c371",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "www.starbucks.com",
                        "typeGroup": "Payment",
                        "type": "Card Payment",
                        "category": "Food Drinks",
                        "bookingDate": "2019-05-29",
                        "valueDate": "2019-05-29",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "7.5",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Starbucks",
                        "counterPartyAccountNumber": "NL91ABNA0423164312",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Zuidplein 10, 1077 XV",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c372",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "www.medimarket.com",
                        "typeGroup": "Payment",
                        "type": "Card Payment",
                        "category": "Shopping",
                        "bookingDate": "2019-05-25",
                        "valueDate": "2019-05-25",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "320.95",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "MediaMarkt",
                        "counterPartyAccountNumber": "NL91ABNA0541164300",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Zuidplein 10, 1077 XV",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c373",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "albertheijn.nl",
                        "typeGroup": "Payment",
                        "type": "Card Payment",
                        "category": "Shopping",
                        "bookingDate": "2019-05-20",
                        "valueDate": "2019-05-20",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "919.95",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Albert Heijn",
                        "counterPartyAccountNumber": "BE68539007547034",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Stationsplein, 1012 AB",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c3731214",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Internet Banking",
                        "typeGroup": "Payment",
                        "type": "International Payment",
                        "category": "Transfers",
                        "bookingDate": "2019-05-15",
                        "valueDate": "2019-05-16",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "887.31",
                            "currencyCode": "EUR"
                        },
                        "currencyExchangeRate": 0.8945,
                        "counterPartyName": "Tusk Williams B.V",
                        "counterPartyAccountNumber": "BE68539007547034",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Stationsplein, 1012 AB",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c373c342",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Internet Banking",
                        "typeGroup": "Payment",
                        "type": "SEPA Payment",
                        "category": "Income",
                        "bookingDate": "2019-05-22",
                        "valueDate": "2019-05-22",
                        "creditDebitIndicator": "CRDT",
                        "transactionAmountCurrency": {
                            "amount": "1919.95",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Backbase B.V",
                        "counterPartyAccountNumber": "NL40ABNA0541164350",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Jacob Bontiusplaats 9, 1018 LL",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c374",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Internet Banking",
                        "typeGroup": "Payment",
                        "type": "SEPA Payment",
                        "category": "Shopping",
                        "bookingDate": "2019-04-28",
                        "valueDate": "2019-04-28",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "111.95",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Eneco Bank11",
                        "counterPartyAccountNumber": "BA391290079401028494",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Stationsplein, 1012 AB",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c375",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "albertheijn.nl",
                        "typeGroup": "Payment",
                        "type": "Card Payment",
                        "category": "Shopping",
                        "bookingDate": "2019-04-24",
                        "valueDate": "2019-08-24",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "86.54",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Albert Heijn",
                        "counterPartyAccountNumber": "BG80BNBG96611020345678",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Stationsplein, 1012 AB",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c376",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Standing Order",
                        "typeGroup": "Payment",
                        "type": "SEPA Payment",
                        "category": "Other Income",
                        "bookingDate": "2019-04-15",
                        "valueDate": "2019-04-15",
                        "creditDebitIndicator": "CRDT",
                        "transactionAmountCurrency": {
                            "amount": "5200",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Simons & Co. B.V.",
                        "counterPartyAccountNumber": "HR1210010051863000160",
                        "counterPartyCity": "Utrecht",
                        "counterPartyAddress": "Vendelier 71. 3905PD",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c377",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Internet Banking",
                        "typeGroup": "Payment",
                        "type": "Payment Sent",
                        "category": "Transfers",
                        "bookingDate": "2019-04-15",
                        "valueDate": "2019-04-15",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "10",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Sally Lambert",
                        "counterPartyAccountNumber": "CY17002001280000001200527600",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Stationsplein, 1012 AB",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c378",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "www.medimarket.com",
                        "typeGroup": "Payment",
                        "type": "Card Payment",
                        "category": "Shopping",
                        "bookingDate": "2019-03-13",
                        "valueDate": "2019-03-13",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "199.95",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "MediaMarkt",
                        "counterPartyAccountNumber": "CZ6508000000192000145399",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Stationsplein, 1012 AB",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c379",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Internet Banking",
                        "typeGroup": "Payment",
                        "type": "SEPA Payment",
                        "category": "Transfers",
                        "bookingDate": "2019-02-13",
                        "valueDate": "2019-02-13",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "25",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "John White",
                        "counterPartyAccountNumber": "DK5000400440116243",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Stationsplein, 1012 AB",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c380",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "www.starbucks.com",
                        "typeGroup": "Payment",
                        "type": "Card Payment",
                        "category": "Food Drinks",
                        "bookingDate": "2019-02-13",
                        "valueDate": "2019-02-13",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "7.5",
                            "currencyCode": "EUR"
                        },
                        "counterPartyAddress": "Zuidplein 10, 1077 XV",
                        "counterPartyCity": "Amsterdam",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c381",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Internet Banking",
                        "typeGroup": "Payment",
                        "type": "Card Payment",
                        "category": "Shopping",
                        "bookingDate": "2019-02-13",
                        "valueDate": "2019-02-14",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "66.54",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Eneco Bank8",
                        "counterPartyAccountNumber": "FO1464600009692713",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Stationsplein, 1012 AB",
                        "imageType": "albert",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c382",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Standing Order",
                        "typeGroup": "Payment",
                        "type": "SEPA Payment",
                        "category": "Category",
                        "bookingDate": "2019-01-10",
                        "valueDate": "2019-01-10",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "400",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Simons & Co. B.V.",
                        "counterPartyAccountNumber": "FI2112345600000785",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Stationsplein, 1012 AB",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c383",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Internet Banking",
                        "typeGroup": "Payment",
                        "type": "SEPA Payment",
                        "category": "Transfers",
                        "bookingDate": "2018-10-01",
                        "valueDate": "2018-10-01",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "31",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Sally Lambert",
                        "counterPartyAccountNumber": "FR1420041010050500013M02606",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Stationsplein, 1012 AB",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c384",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Standing Order",
                        "typeGroup": "Payment",
                        "type": "Payment Received",
                        "category": "Other Income",
                        "bookingDate": "2018-08-15",
                        "valueDate": "2018-08-15",
                        "creditDebitIndicator": "CRDT",
                        "transactionAmountCurrency": {
                            "amount": "25",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Eneco Bank12",
                        "counterPartyAccountNumber": "GE29NB0000000101904917",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Stationsplein, 1012 AB",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c385",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Internet Banking",
                        "typeGroup": "Payment",
                        "type": "SEPA Payment",
                        "category": "Transfers",
                        "bookingDate": "2018-10-13",
                        "valueDate": "2018-08-15",
                        "creditDebitIndicator": "DBIT",
                        "transactionAmountCurrency": {
                            "amount": "640",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Sally Lambert",
                        "counterPartyAccountNumber": "DE89370400440532013000",
                        "counterPartyCity": "Amsterdam",
                        "counterPartyAddress": "Stationsplein, 1012 AB",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c386",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Internet Banking",
                        "typeGroup": "Payment",
                        "type": "SEPA Payment",
                        "category": "Income",
                        "bookingDate": "2018-07-13",
                        "valueDate": "2018-07-15",
                        "creditDebitIndicator": "CRDT",
                        "transactionAmountCurrency": {
                            "amount": "7200",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Simons & Co. B.V.",
                        "counterPartyAccountNumber": "GI75NWBK000000007099453",
                        "counterPartyCity": "Utrecht",
                        "counterPartyAddress": "Vendelier 71. 3905PD",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }, {
                        "id": "ec9ec6fd-a99c-4cee-a73e-51f84b11c387",
                        "arrangementId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4699",
                        "reference": "7e8f5718-946c-4a86-a8bb-d903a1a20db8",
                        "description": "Internet Banking",
                        "typeGroup": "Payment",
                        "type": "SEPA Payment",
                        "bookingDate": "2018-07-13",
                        "valueDate": "2018-07-15",
                        "creditDebitIndicator": "CRDT",
                        "transactionAmountCurrency": {
                            "amount": "7200",
                            "currencyCode": "EUR"
                        },
                        "counterPartyName": "Simons & Co. B.V.",
                        "counterPartyAccountNumber": "GI75NWBK000000007099453",
                        "counterPartyCity": "Utrecht",
                        "counterPartyAddress": "Vendelier 71. 3905PD",
                        "additions": {
                            "settlementDate": "2020-06-01",
                            "additionalCharges": "183.56",
                            "purpose": "vacation",
                            "purposeCode": "1",
                            "fees": "62",
                            "feeReferenceId": "FEE-01",
                            "ultimateDebtorName": "John Doe",
                            "debtorId": "df8e1cac-e483-4ad1-87e6-e7a2da343421",
                            "ultimateDebtorId": "e7a2da343421",
                            "ultimateCreditorId": "df8e1cac",
                            "ultimateCreditorName": "Julia Pence",
                            "debtorReference": "[{'name':'CLEO V MONROE SR','role':'PRIMARY'},{'name':'LANIE Q PIERCE','role':'JOINT'}]",
                            "MMCCode": "V"
                        }
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
        ]
    },
]);
