// @backbase/ui-ang/esm2015/_example/date-pipe/examples/date-pipe.example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/date-pipe";
export class DatePipeExampleComponent {
    constructor() {
        this.exampleDate = '2018-12-12T14:29:14.000Z';
    }
}
DatePipeExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DatePipeExampleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
DatePipeExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: DatePipeExampleComponent, selector: "bb-date-pipe-example-ui", ngImport: i0, template: "<p>\n  {{ exampleDate | bbDate: 'bbShort':'GMT-0700':'en-US' }}\n</p>\n", pipes: { "bbDate": i1.BbDatePipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: DatePipeExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-date-pipe-example-ui',
                    templateUrl: './date-pipe.example.component.html',
                }]
        }] });
