// @backbase/ui-ang/esm2015/period-selector/period-selector.module.js
import { NgModule } from '@angular/core';
import { CommonModule, DatePipe } from '@angular/common';
import { PeriodSelectorComponent } from './period-selector.component';
import { IconModule } from '@backbase/ui-ang/icon';
import { ButtonModule } from '@backbase/ui-ang/button';
import * as i0 from "@angular/core";
export class PeriodSelectorModule {
}
PeriodSelectorModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PeriodSelectorModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PeriodSelectorModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PeriodSelectorModule, declarations: [PeriodSelectorComponent], imports: [CommonModule, IconModule, ButtonModule], exports: [PeriodSelectorComponent] });
PeriodSelectorModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PeriodSelectorModule, providers: [DatePipe], imports: [[CommonModule, IconModule, ButtonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PeriodSelectorModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, IconModule, ButtonModule],
                    declarations: [PeriodSelectorComponent],
                    exports: [PeriodSelectorComponent],
                    providers: [DatePipe],
                }]
        }] });
