// @backbase/product-summary-ui/esm2015/src/bb-chart-legend-ui/chart-legend.component.js
import { Component, Directive, Input } from '@angular/core';
import { BbTemplate } from '@backbase/foundation-ang/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/icon";
import * as i2 from "@backbase/ui-ang/amount";
export class ChartLegendComponent {
    constructor() {
        this.hostRef = this;
        /**
         * Currency data
         */
        this.currency = '';
        /**
         * Maximum value
         */
        this.maxValue = '';
        /**
         * Minimum value
         */
        this.minValue = '';
        /**
         * Color for positive area
         */
        this.positiveColor = '';
        /**
         * Color for negative area
         */
        this.negativeColor = '';
    }
}
ChartLegendComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ChartLegendComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ChartLegendComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ChartLegendComponent, selector: "bb-chart-legend-ui", inputs: { currency: "currency", maxValue: "maxValue", minValue: "minValue", positiveColor: "positiveColor", negativeColor: "negativeColor" }, ngImport: i0, template: "<ng-container\n  bbChartLegendCustomizable\n  [bbHostRef]=\"hostRef\"\n></ng-container>\n\n<ng-template\n  bbChartLegendCustomizable\n  let-hostRef\n>\n  <div class=\"bb-subtitle bb-stack bb-stack--center bb-stack--wrap\"\n       data-role=\"area-chart-legend\">\n    <div class=\"bb-stack__item bb-stack__item--spacing-lg bb-stack__break--sm-down\">\n      <div class=\"bb-stack bb-stack--density-sm\">\n        <bb-icon-ui\n          class=\"bb-stack__item\"\n          data-role=\"area-chart-legend-positive-color\"\n          name=\"circle\"\n          size=\"sm\"\n          [style.color]=\"hostRef.positiveColor\"\n        ></bb-icon-ui>\n        <span\n          class=\"bb-stack__item\"\n          i18n=\"Max balance@@product-summary-balance-history-chart.max-balance\"\n        >\n            Max Balance:\n          </span>\n        <bb-amount-ui\n          class=\"bb-stack__item bb-text-bold\"\n          data-role=\"area-chart-legend-max-value\"\n          [currency]=\"hostRef.currency\"\n          [amount]=\"hostRef.maxValue\"\n        ></bb-amount-ui>\n      </div>\n    </div>\n    <div class=\"bb-stack__break bb-stack__break--spacing-sm bb-stack__break--sm-down\"></div>\n    <div class=\"bb-stack__item bb-stack__item--spacing-lg bb-stack__break--sm-down\">\n      <div class=\"bb-stack bb-stack--density-sm\">\n        <bb-icon-ui\n          class=\"bb-stack__item\"\n          data-role=\"area-chart-legend-negative-color\"\n          name=\"circle\"\n          size=\"sm\"\n          [style.color]=\"hostRef.negativeColor\"\n        ></bb-icon-ui>\n        <span\n          class=\"bb-stack__item\"\n          i18n=\"Min balance@@product-summary-balance-history-chart.min-balance\"\n        >\n            Min Balance:\n          </span>\n        <bb-amount-ui\n          class=\"bb-stack__item bb-text-bold\"\n          data-role=\"area-chart-legend-min-value\"\n          [currency]=\"hostRef.currency\"\n          [amount]=\"hostRef.minValue\"\n        ></bb-amount-ui>\n      </div>\n    </div>\n  </div>\n\n</ng-template>\n\n", components: [{ type: i0.forwardRef(function () { return i1.IconComponent; }), selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i0.forwardRef(function () { return i2.AmountComponent; }), selector: "bb-amount-ui", inputs: ["currency", "showPlusSign", "mapCurrency", "showPercent", "abbreviate", "decimalPlaces", "trailingZeroes", "amount"] }], directives: [{ type: i0.forwardRef(function () { return ChartLegendCustomizableDirective; }), selector: "[bbChartLegendCustomizable]" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ChartLegendComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-chart-legend-ui',
                    templateUrl: 'chart-legend.component.html',
                }]
        }], propDecorators: { currency: [{
                type: Input
            }], maxValue: [{
                type: Input
            }], minValue: [{
                type: Input
            }], positiveColor: [{
                type: Input
            }], negativeColor: [{
                type: Input
            }] } });
export class ChartLegendCustomizableDirective extends BbTemplate {
}
ChartLegendCustomizableDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ChartLegendCustomizableDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
ChartLegendCustomizableDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.16", type: ChartLegendCustomizableDirective, selector: "[bbChartLegendCustomizable]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ChartLegendCustomizableDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[bbChartLegendCustomizable]',
                }]
        }] });
