// @backbase/product-summary-ui/esm2015/src/bb-chart-legend-ui/chart-legend.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { IconModule } from '@backbase/ui-ang/icon';
import { AmountModule } from '@backbase/ui-ang/amount';
import { ChartLegendCustomizableDirective, ChartLegendComponent } from './chart-legend.component';
import * as i0 from "@angular/core";
const uiModules = [IconModule, AmountModule];
const components = [ChartLegendComponent];
const directives = [ChartLegendCustomizableDirective];
export class ChartLegendModule {
}
ChartLegendModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ChartLegendModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ChartLegendModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ChartLegendModule, declarations: [ChartLegendComponent, ChartLegendCustomizableDirective], imports: [CommonModule, IconModule, AmountModule], exports: [ChartLegendComponent, ChartLegendCustomizableDirective, IconModule, AmountModule] });
ChartLegendModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ChartLegendModule, imports: [[CommonModule, ...uiModules], IconModule, AmountModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ChartLegendModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ...uiModules],
                    declarations: [...components, ...directives],
                    exports: [...components, ...directives, ...uiModules],
                }]
        }] });
