// @backbase/contact-manager-journey-ang/esm2015/lib/contact-manager-journey.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { provideRoutes, RouterModule } from '@angular/router';
import { ContactManagerJourneyConfigurationService } from './services/contact-manager-journey-config.service';
import { contactManagerRoutes } from './contact-manager-journey.routes';
import { ContactManagerWidgetModule } from '@backbase/contact-manager-widget-ang';
import { ContactManagerJourneyComponent } from './contact-manager-journey.component';
import { ContactManagerDetailViewComponent } from './views/contact-manager-detail/contact-manager-detail.component';
import { ContactManagerTypeListViewComponent } from './views/contact-manager-type-list/contact-manager-type-list.component';
import { ContactManagerFormViewComponent } from './views/contact-manager-form/contact-manager-form.component';
import { HeaderModule } from '@backbase/ui-ang/header';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
/**
 * Journey for contacts in retails apps
 *
 * @usageNotes
 *
 * ### Include the exported widgets in your `AppModule`:
 *
 * ```typescript
 * export const routes: Routes = [
 *   {
 *     path: '',
 *     component: ContactManagerJourneyComponent,
 *     children: [
 *       {
 *         path: '',
 *         component: ContactManagerDetailViewComponent,
 *       },
 *       {
 *         path: 'select',
 *         component: ContactManagerTypeListViewComponent,
 *       },
 *       {
 *         path: ':id',
 *         component: ContactManagerDetailViewComponent,
 *       },
 *       {
 *         path: 'edit/:id/:type',
 *         canDeactivate: [CanDeactivateGuard],
 *         component: ContactManagerFormViewComponent,
 *       },
 *     ],
 *   },
 * ];
 * };
 * @NgModule({
 * imports: [ContactManagerJourneyModule.forRoot({ route })],
 *  providers: [ {
 *   provide: ContactManagerJourneyConfigurationToken,
 *   useValue: {
 *     createContactEmail: true,
 *     createContactIBAN: true,
 *     createContactAccount: true,
 *     createContactPhone: true,
 *   } as Partial<ContactManagerJourneyConfig>,
 * },],
 * })
 * export class AppModule {}
 * ```
 */
export class ContactManagerJourneyModule {
    static forRoot(data = { routes: contactManagerRoutes }) {
        return {
            ngModule: ContactManagerJourneyModule,
            providers: [provideRoutes(data.routes)],
        };
    }
}
/** @nocollapse */ ContactManagerJourneyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactManagerJourneyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ ContactManagerJourneyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactManagerJourneyModule, declarations: [ContactManagerJourneyComponent,
        ContactManagerDetailViewComponent,
        ContactManagerTypeListViewComponent,
        ContactManagerFormViewComponent], imports: [CommonModule, ContactManagerWidgetModule, HeaderModule, i1.RouterModule] });
/** @nocollapse */ ContactManagerJourneyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactManagerJourneyModule, providers: [ContactManagerJourneyConfigurationService], imports: [[CommonModule, ContactManagerWidgetModule, HeaderModule, RouterModule.forChild(contactManagerRoutes)]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ContactManagerJourneyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        ContactManagerJourneyComponent,
                        ContactManagerDetailViewComponent,
                        ContactManagerTypeListViewComponent,
                        ContactManagerFormViewComponent,
                    ],
                    imports: [CommonModule, ContactManagerWidgetModule, HeaderModule, RouterModule.forChild(contactManagerRoutes)],
                    providers: [ContactManagerJourneyConfigurationService],
                }]
        }] });
