// @backbase/quick-transfer-journey-ang/esm2015/lib/store/actions.js
import { createAction, props } from '@ngrx/store';
export const accountsFetch = createAction('[Accounts API] fetching accounts');
export const contactsFetch = createAction('[Contacts API] fetching contacts');
export const externalAccountsFetch = createAction('[Payments API] fetching external accounts');
// Success
export const accountsFetchSuccess = createAction('[Accounts API] accounts fetched success', props());
export const contactsFetchSuccess = createAction('[Contacts API] contacts fetched success', props());
export const extAccountsFetchSuccess = createAction('[Payments API] external accounts fetched success', props());
// Error
export const accountsFetchError = createAction('[Accounts API] accounts fetching error', props());
export const contactsFetchError = createAction('[Contacts API] contacts fetching error', props());
export const extAccountsFetchError = createAction('[Payments API] external accounts fetching error', props());
export const openView = createAction('[Quick Transfer Widget] current view changed', props());
export const setPayeeAccount = createAction('[Quick Transfer Widget] payee account picked up', props());
export const setFromAccount = createAction('[Quick Transfer Widget] From Account changed', props());
export const sendTransferOrder = createAction('[Payments Order API] send transfer order', props());
export const transferOrderSuccess = createAction('[Payments Order API] transfer order is successful');
export const transferOrderError = createAction('[Payments Order API] transfer order is failed', props());
export const setAmount = createAction('[Quick Transfer Widget] Transfer Amount Set', props());
export const setModalOpen = createAction('[Quick Transfer Widget] Confirmation Modal Open', props());
export const resetWidget = createAction('[Quick Transfer Widget] Reset widget', props());
export const resetFocus = createAction('[Quick Transfer Widget] Reset Focus Element');
export const setFieldTouched = createAction('[Quick Transfer Widget] Set Field Touched', props());
export const setDefaultCurrency = createAction('[Quick Transfer Widget] Default Currency changed', props());
