// @backbase/quick-transfer-journey-ang/esm2015/lib/store/selectors.js
import { createFeatureSelector, createSelector } from '@ngrx/store';
import { FEATURE_STORE_SELECTOR } from '../constants/initital-state';
const child = createFeatureSelector(FEATURE_STORE_SELECTOR);
export const selectFromAccount = createSelector(child, (state) => state['fromAccount']);
export const selectDebitAccounts = createSelector(child, (state) => state['debitAccounts']);
export const selectCreditAccounts = createSelector(child, (state) => state['creditAccounts']);
export const selectExtAccounts = createSelector(child, (state) => state['extAccounts']);
export const selectContacts = createSelector(child, (state) => state['contacts']);
export const selectAccountsCallState = createSelector(child, (state) => state['accountsCallState']);
export const selectContactsCallState = createSelector(child, (state) => state['contactsCallState']);
export const selectExtAccountsCallState = createSelector(child, (state) => state['extAccountsCallState']);
export const selectTransferOrderCallState = createSelector(child, (state) => state['transferOrderCallState']);
export const selectViewState = createSelector(child, (state) => state['viewState']);
export const selectPayeeAccount = createSelector(child, (state) => state['payeeTo']);
export const selectAmount = createSelector(child, (state) => state['amount']);
export const selectIsModalOpen = createSelector(child, (state) => state['isModalOpen']);
export const selectTouchedFields = createSelector(child, (state) => state['touchedFields']);
export const selectFocusElement = createSelector(child, (state) => state['focusElement']);
