// @backbase/quick-transfer-journey-ang/esm2015/lib/store/store.module.js
import { NgModule } from '@angular/core';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule } from '@ngrx/store';
import { FEATURE_STORE_SELECTOR } from '../constants/initital-state';
import { QuickTransferEffects } from './effects';
import { quickTransferReducer } from './reducer';
import * as i0 from "@angular/core";
import * as i1 from "@ngrx/store";
import * as i2 from "@ngrx/effects";
export class QuickTransferStoreModule {
}
/** @nocollapse */ QuickTransferStoreModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: QuickTransferStoreModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ QuickTransferStoreModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: QuickTransferStoreModule, imports: [i1.StoreFeatureModule, i2.EffectsFeatureModule] });
/** @nocollapse */ QuickTransferStoreModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: QuickTransferStoreModule, providers: [QuickTransferEffects], imports: [[
            StoreModule.forFeature(FEATURE_STORE_SELECTOR, quickTransferReducer),
            EffectsModule.forFeature([QuickTransferEffects]),
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: QuickTransferStoreModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [],
                    imports: [
                        StoreModule.forFeature(FEATURE_STORE_SELECTOR, quickTransferReducer),
                        EffectsModule.forFeature([QuickTransferEffects]),
                    ],
                    providers: [QuickTransferEffects],
                }]
        }] });
