// @backbase/ui-ang/esm2015/product-item-loan/product-item-loan.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { CustomProductItemLoanDetailsFormatterDirective, CustomProductItemLoanTitleDirective, ProductItemLoanComponent, } from './product-item-loan.component';
import { AmountModule } from '@backbase/ui-ang/amount';
import { PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import { HighlightModule } from '@backbase/ui-ang/highlight';
import { IconModule } from '@backbase/ui-ang/icon';
import * as i0 from "@angular/core";
export class ProductItemLoanModule {
}
ProductItemLoanModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemLoanModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductItemLoanModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemLoanModule, declarations: [ProductItemLoanComponent,
        CustomProductItemLoanTitleDirective,
        CustomProductItemLoanDetailsFormatterDirective], imports: [CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule], exports: [ProductItemLoanComponent,
        CustomProductItemLoanTitleDirective,
        CustomProductItemLoanDetailsFormatterDirective] });
ProductItemLoanModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemLoanModule, imports: [[CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemLoanModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, AmountModule, PaymentCardNumberModule, HighlightModule, IconModule],
                    declarations: [
                        ProductItemLoanComponent,
                        CustomProductItemLoanTitleDirective,
                        CustomProductItemLoanDetailsFormatterDirective,
                    ],
                    exports: [
                        ProductItemLoanComponent,
                        CustomProductItemLoanTitleDirective,
                        CustomProductItemLoanDetailsFormatterDirective,
                    ],
                }]
        }] });
