// @backbase/foundation-ang/esm2015/auth/deprecated/auth.guard.js
import { Inject, Injectable } from '@angular/core';
import { DOCUMENT, Location } from '@angular/common';
import { take, tap } from 'rxjs/operators';
import { AuthService } from './auth.service';
import * as i0 from "@angular/core";
import * as i1 from "./auth.service";
import * as i2 from "@angular/common";
/**
 * @deprecated Will be removed in v7.0.0
 */
export class AuthGuard {
    constructor(auth, location, document) {
        this.auth = auth;
        this.location = location;
        this.document = document;
    }
    canLoad() {
        return this.auth.isAuthenticated.pipe(take(1));
    }
    canActivate() {
        return this.redirectIfUnauthenticated();
    }
    canActivateChild() {
        return this.redirectIfUnauthenticated();
    }
    redirectIfUnauthenticated() {
        return this.auth.isAuthenticated.pipe(tap(loggedIn => {
            if (!loggedIn) {
                // Use an actual location path with hash instead of `RouterStateSnapshot.url`
                // because `Keycloak` uses `history.replaceState` on initialization to remove
                // auth data from the path.
                const actualPath = this.location.path(true);
                this.auth.login({
                    redirectUri: this.document.location.origin + this.location.prepareExternalUrl(actualPath),
                });
            }
        }));
    }
}
AuthGuard.ɵprov = i0.ɵɵdefineInjectable({ factory: function AuthGuard_Factory() { return new AuthGuard(i0.ɵɵinject(i1.AuthService), i0.ɵɵinject(i2.Location), i0.ɵɵinject(i2.DOCUMENT)); }, token: AuthGuard, providedIn: "root" });
AuthGuard.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root',
            },] }
];
AuthGuard.ctorParameters = () => [
    { type: AuthService },
    { type: Location },
    { type: undefined, decorators: [{ type: Inject, args: [DOCUMENT,] }] }
];
