// @backbase/messages-conversation-thread-widget-ang/esm2015/src/services/widget-properties.service.js
import { Injectable, Optional } from '@angular/core';
import { defaultModelObservable } from '@backbase/messages-common';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
const DefaultWidgetProperties = {
    customerServiceTitle: 'Customer Service',
    mailboxTypeTitle: 'Inbox',
    maxAttachmentSize: '10',
    replyMessageMaxLength: '300',
};
export class WidgetPropertiesService {
    constructor(itemModel) {
        this.itemModel = itemModel;
    }
    /**
     * Customer service title
     */
    get customerServiceTitle() {
        if (!this.itemModel) {
            return defaultModelObservable(DefaultWidgetProperties.customerServiceTitle);
        }
        return this.itemModel.property('customerServiceTitle', DefaultWidgetProperties.customerServiceTitle);
    }
    /**
     * Mailbox type title
     */
    get mailboxTypeTitle() {
        if (!this.itemModel) {
            return defaultModelObservable(DefaultWidgetProperties.mailboxTypeTitle);
        }
        return this.itemModel.property('mailboxTypeTitle', DefaultWidgetProperties.mailboxTypeTitle);
    }
    /**
     * Max attachment size
     */
    get maxAttachmentSize() {
        if (!this.itemModel) {
            return defaultModelObservable(DefaultWidgetProperties.maxAttachmentSize);
        }
        return this.itemModel.property('maxAttachmentSize', DefaultWidgetProperties.maxAttachmentSize);
    }
    /**
     * Reply message max length
     */
    get replyMessageMaxLength() {
        if (!this.itemModel) {
            return defaultModelObservable(DefaultWidgetProperties.replyMessageMaxLength);
        }
        return this.itemModel.property('replyMessageMaxLength', DefaultWidgetProperties.replyMessageMaxLength);
    }
}
WidgetPropertiesService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: WidgetPropertiesService, deps: [{ token: i1.ItemModel, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
WidgetPropertiesService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: WidgetPropertiesService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0, type: WidgetPropertiesService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.ItemModel, decorators: [{
                    type: Optional
                }] }]; } });
