// @backbase/loans-collection-ui-ang/esm2015/lib/loan-info/loan-info-collateral-section/loan-info-collateral-section.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AmountModule } from '@backbase/ui-ang/amount';
import { BbDatePipeModule } from '@backbase/ui-ang/date-pipe';
import { LoanInfoCollateralSectionComponent } from './loan-info-collateral-section.component';
import * as i0 from "@angular/core";
export class LoanInfoCollateralSectionModule {
}
/** @nocollapse */ LoanInfoCollateralSectionModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoCollateralSectionModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LoanInfoCollateralSectionModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoCollateralSectionModule, declarations: [LoanInfoCollateralSectionComponent], imports: [CommonModule, AmountModule, BbDatePipeModule], exports: [LoanInfoCollateralSectionComponent] });
/** @nocollapse */ LoanInfoCollateralSectionModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoCollateralSectionModule, imports: [[CommonModule, AmountModule, BbDatePipeModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: LoanInfoCollateralSectionModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LoanInfoCollateralSectionComponent],
                    imports: [CommonModule, AmountModule, BbDatePipeModule],
                    exports: [LoanInfoCollateralSectionComponent],
                }]
        }] });
