// @backbase/payment-orders-ang/esm2015/lib/ui/components/country-selector/country-selector.component.js
import { Component, EventEmitter, forwardRef, Input, Output, } from '@angular/core';
import { FormControl, NG_VALUE_ACCESSOR, } from '@angular/forms';
import { CountriesMap } from '../../constants';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/icon";
import * as i2 from "@backbase/ui-ang/search-box";
import * as i3 from "../flag-icon/bb-flag-icon";
import * as i4 from "../country-name/country-name.component";
import * as i5 from "@backbase/ui-ang/empty-state";
import * as i6 from "@ng-bootstrap/ng-bootstrap";
import * as i7 from "@angular/common";
import * as i8 from "@backbase/ui-ang/button";
import * as i9 from "@angular/forms";
import * as i10 from "@backbase/ui-ang/keyboard-click-directive";
export class CountrySelectorComponent {
    constructor() {
        this._selectedCountry = '';
        this.search = new FormControl();
        this.filteredCountries = [];
        /** */
        this.selectedCountry = '';
        /** */
        this.helperText = '';
        /** */
        this.placeholder = '';
        /** */
        this.label = '';
        /** */
        this.searchPlaceholder = '';
        /** */
        this.dropdownIcon = 'toggle-down';
        /** */
        this.selectCountry = new EventEmitter();
    }
    /** */
    set countries(countries) {
        this._countries = countries || [];
        this.setupCountries();
    }
    setupCountries() {
        this.countryEntries = this._countries.map((country) => [
            country,
            CountriesMap[country],
        ]);
        this.filteredCountries = [...this._countries];
    }
    selectItem(country) {
        this.selectedCountry = country;
        this.selectCountry.emit(country);
    }
    handleSearch(query = '') {
        if (query) {
            this.filteredCountries = this.countryEntries.reduce((acc, [key, country]) => {
                if (key.toLocaleLowerCase().indexOf(query.toLocaleLowerCase()) > -1 ||
                    country.toLocaleLowerCase().indexOf(query.toLocaleLowerCase()) > -1) {
                    acc.push(key);
                }
                return acc;
            }, []);
        }
        else {
            this.filteredCountries = this._countries;
        }
    }
    onChange(_value) { }
    onTouched() { }
    writeValue(_value) { }
    onBlur() { }
    registerOnChange(fn) {
        this.onChange = fn;
    }
    registerOnTouched(fn) {
        this.onTouched = fn;
    }
}
/** @nocollapse */ CountrySelectorComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CountrySelectorComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ CountrySelectorComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: CountrySelectorComponent, selector: "bb-country-selector", inputs: { countries: "countries", selectedCountry: "selectedCountry", helperText: "helperText", placeholder: "placeholder", label: "label", searchPlaceholder: "searchPlaceholder", dropdownIcon: "dropdownIcon" }, outputs: { selectCountry: "selectCountry" }, providers: [
        {
            provide: NG_VALUE_ACCESSOR,
            multi: true,
            useExisting: forwardRef((() => CountrySelectorComponent)),
        },
    ], ngImport: i0, template: "<div\n  class=\"bb-country-selector\"\n  ngbDropdown\n  #dropdown=\"ngbDropdown\"\n  display=\"static\"\n>\n  <div class=\"d-inline-block bb-label\" (click)=\"btn.focus()\" *ngIf=\"label\">\n    {{ label }}\n  </div>\n  <span\n    class=\"bb-dynamic-input__helper-text bb-text-support\"\n    *ngIf=\"helperText\"\n    [id]=\"searchElem?.id + '__helper-text'\"\n    [textContent]=\"helperText\"\n  >\n  </span>\n  <div>\n    <button\n      bbButton\n      ngbDropdownToggle\n      color=\"unstyled\"\n      (blur)=\"onBlur()\"\n      data-role=\"open-country-dropdown-list-button\"\n      [attr.aria-label]=\"label\"\n      class=\"bb-country-selector__toggle form-control w-100 d-flex justify-content-between bb-text-support\"\n      #btn\n    >\n      <ng-container *ngIf=\"selectedCountry\">\n        <ng-container\n          *ngTemplateOutlet=\"\n            countryTmpl;\n            context: { $implicit: selectedCountry }\n          \"\n        ></ng-container>\n      </ng-container>\n      <div *ngIf=\"!selectedCountry\">\n        <ng-container\n          *ngIf=\"placeholder; else defaultPlaceHolder\"\n        ></ng-container>\n        {{ placeholder }}\n      </div>\n      <bb-icon-ui [name]=\"dropdownIcon || 'toggle-down'\"></bb-icon-ui>\n    </button>\n  </div>\n\n  <div ngbDropdownMenu class=\"dropdown-menu dropdown-menu-right w-100\">\n    <div class=\"bb-block bb-block--md px-3\">\n      <bb-search-box-ui\n        (focus)=\"dropdown.open()\"\n        [formControl]=\"search\"\n        (clear)=\"handleSearch('')\"\n        data-role=\"country-search\"\n        role=\"searchbox\"\n        (ngModelChange)=\"handleSearch(search.value)\"\n        [placeholder]=\"searchPlaceholder || 'Search'\"\n        [showSearch]=\"true\"\n        [showClear]=\"true\"\n        [autocomplete]=\"'off'\"\n        #searchElem\n      >\n      </bb-search-box-ui>\n    </div>\n    <div *ngIf=\"filteredCountries?.length\" class=\"pre-scrollable\">\n      <button\n        *ngFor=\"let country of filteredCountries\"\n        role=\"menuitem\"\n        type=\"button\"\n        class=\"d-sm-flex px-3 py-2 d-flex-row align-items-center btn--unstyled\"\n        (bbKeyboardClick)=\"selectItem(country); dropdown.close()\"\n        ngbDropdownItem\n      >\n        <ng-container\n          *ngTemplateOutlet=\"countryTmpl; context: { $implicit: country }\"\n        ></ng-container>\n      </button>\n    </div>\n    <ng-container *ngIf=\"!filteredCountries?.length && search?.value\">\n      <ng-container *ngTemplateOutlet=\"emptySearch\"></ng-container>\n    </ng-container>\n  </div>\n</div>\n\n<ng-template let-code #countryTmpl>\n  <div class=\"d-flex justify-content-between align-items-center w-100\">\n    <div class=\"d-flex align-items-center\">\n      <bb-flag-icon\n        class=\"mr-2 d-flex\"\n        [countryCode]=\"code\"\n        [flagSize]=\"'sm'\"\n      ></bb-flag-icon>\n      <bb-country-name [countryCode]=\"code\"></bb-country-name>\n    </div>\n    <div class=\"bb-text-support\">{{ code }}</div>\n  </div>\n</ng-template>\n\n<ng-template #emptySearch>\n  <div class=\"card-body\">\n    <bb-empty-state-ui\n      data-role=\"no-templates-found\"\n      title=\"No Results Found\"\n      subtitle=\"We can\u2019t find what you\u2019re looking for. Please try a different criteria.\"\n      iconSize=\"xxl\"\n      i18n-subtitle=\"@@bb-country-selector.empty-state-subtitle\"\n      i18n-title=\"@@bb-country-selector.empty-state-title\"\n      iconModifier=\"search\"\n      [showIcon]=\"true\"\n    ></bb-empty-state-ui>\n  </div>\n</ng-template>\n\n<ng-template #defaultPlaceHolder>\n  <ng-container i18n>- select a country -</ng-container>\n</ng-template>\n", components: [{ type: i1.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }, { type: i2.SearchBoxComponent, selector: "bb-search-box-ui", inputs: ["placeholder", "maxLength", "squareBorder", "typeaheadOptions", "showSearch", "showClear", "searchLabel", "clearLabel", "autocomplete"], outputs: ["submit", "clear", "valueChange"] }, { type: i3.FlagIconComponent, selector: "bb-flag-icon", inputs: ["flagSize", "countryCode"] }, { type: i4.CountryNameComponent, selector: "bb-country-name", inputs: ["countryCode"] }, { type: i5.EmptyStateComponent, selector: "bb-empty-state-ui", inputs: ["title", "subtitle", "showIcon", "iconClasses", "iconModifier", "iconSize", "iconColor"] }], directives: [{ type: i6.NgbDropdown, selector: "[ngbDropdown]", inputs: ["open", "placement", "container", "autoClose", "display", "dropdownClass"], outputs: ["openChange"], exportAs: ["ngbDropdown"] }, { type: i7.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i8.ButtonDirective, selector: "button[bbButton]", inputs: ["type", "color", "buttonSize", "block", "circle"] }, { type: i6.NgbDropdownToggle, selector: "[ngbDropdownToggle]" }, { type: i7.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }, { type: i6.NgbDropdownMenu, selector: "[ngbDropdownMenu]" }, { type: i9.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { type: i9.FormControlDirective, selector: "[formControl]", inputs: ["disabled", "formControl", "ngModel"], outputs: ["ngModelChange"], exportAs: ["ngForm"] }, { type: i7.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { type: i6.NgbDropdownItem, selector: "[ngbDropdownItem]", inputs: ["disabled"] }, { type: i10.KeyboardClickDirective, selector: "[bbKeyboardClick]", outputs: ["bbKeyboardClick"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CountrySelectorComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-country-selector',
                    templateUrl: './country-selector.component.html',
                    providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            multi: true,
                            useExisting: forwardRef((() => CountrySelectorComponent)),
                        },
                    ],
                }]
        }], propDecorators: { countries: [{
                type: Input
            }], selectedCountry: [{
                type: Input
            }], helperText: [{
                type: Input
            }], placeholder: [{
                type: Input
            }], label: [{
                type: Input
            }], searchPlaceholder: [{
                type: Input
            }], dropdownIcon: [{
                type: Input
            }], selectCountry: [{
                type: Output
            }] } });
