// @backbase/payment-orders-ang/esm2015/lib/ui/components/status-card/status-card.component.js
import { Component, Input } from '@angular/core';
import { PaymentSubmitStatuses } from '../../models';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/icon";
import * as i2 from "@angular/common";
export class PayordStatusComponent {
    constructor() {
        /**
         * Status of the payment. Defaults to success.
         */
        this.status = PaymentSubmitStatuses.SUCCESS;
        /**
         * Message to be displayed. Defaults to empty string.
         */
        this.message = '';
        /**
         * Flag to denote debit mode.
         */
        this.isDebitPayment = false;
    }
}
/** @nocollapse */ PayordStatusComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordStatusComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ PayordStatusComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: PayordStatusComponent, selector: "bb-payord-status-card", inputs: { status: "status", message: "message", isDebitPayment: "isDebitPayment" }, ngImport: i0, template: "<div\n  class=\"bb-payment-status-card\"\n  [ngClass]=\"\n    status === 'error'\n      ? 'bb-payment-status-card--danger'\n      : 'bb-payment-status-card--success'\n  \"\n>\n  <bb-icon-ui\n    class=\"bb-payment-status-card__icon\"\n    size=\"xxl\"\n    [color]=\"status === 'error' ? 'danger' : 'success'\"\n    [name]=\"status === 'error' ? 'error-outline' : 'success'\"\n  ></bb-icon-ui>\n  <div\n    class=\"break-word col-12 bb-text-align-center\"\n    *ngIf=\"message\"\n    data-role=\"payment-status-msg\"\n  >\n    {{ message }}\n  </div>\n  <div\n    class=\"break-word col-12 bb-text-align-center\"\n    *ngIf=\"!message\"\n    data-role=\"payment-status-msg\"\n  >\n    <ng-container\n      *ngIf=\"!isDebitPayment\"\n      i18n=\"@@bb-payment-status-card-payment-status-message\"\n    >\n      { status, select, error {Failed to submit payment. Please try again later}\n      pending {Payment is submitted successfully, and pending further\n      approval(s)} pending_edited {Edited payment submitted successfully} other\n      {Payment submitted successfully} }\n    </ng-container>\n    <ng-container\n      *ngIf=\"isDebitPayment\"\n      i18n=\"@@bb-payment-status-card-debit-status-message\"\n    >\n      { status, select, error {Failed to submit debit request. Please try again\n      later} pending {Debit request is submitted successfully, and pending\n      further approval(s)} pending_edited {Edited debit request submitted\n      successfully} other {Debit request submitted successfully} }\n    </ng-container>\n  </div>\n</div>\n", components: [{ type: i1.IconComponent, selector: "bb-icon-ui", inputs: ["name", "inverse", "size", "color", "animate", "aria-label", "cropped", "backgroundType"] }], directives: [{ type: i2.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: PayordStatusComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-payord-status-card',
                    templateUrl: './status-card.component.html',
                }]
        }], propDecorators: { status: [{
                type: Input
            }], message: [{
                type: Input
            }], isDebitPayment: [{
                type: Input
            }] } });
