// @backbase/portfolio-summary-instrument-view-http-ang/esm2015/api/instrumentView.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const InstrumentViewHttpServiceGetInstrumentChartDataByPositionIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/positions/{positionID}/instruments/charts",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "chartData": [{
                            "date": "2020-01-01",
                            "price": 125.01,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2020-02-01",
                            "price": 129.04,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2020-03-01",
                            "price": 134.18,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2020-04-01",
                            "price": 131.4,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2020-05-01",
                            "price": 120.88,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2020-06-01",
                            "price": 112.82,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2020-07-01",
                            "price": 115.54,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2020-08-01",
                            "price": 110.34,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2020-09-01",
                            "price": 106.84,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2020-10-01",
                            "price": 108.22,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2020-11-01",
                            "price": 116.5,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2020-12-01",
                            "price": 114.97,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2021-01-01",
                            "price": 124.4,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2021-02-01",
                            "price": 120.71,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2021-03-01",
                            "price": 116.6,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2021-04-01",
                            "price": 111.2,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2021-05-01",
                            "price": 115.32,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2021-06-01",
                            "price": 108.77,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2021-07-01",
                            "price": 114.95,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2021-08-01",
                            "price": 119.03,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2021-09-01",
                            "price": 118.69,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2021-10-01",
                            "price": 116.32,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2021-11-01",
                            "price": 115.97,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2021-12-01",
                            "price": 119.49,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2022-01-01",
                            "price": 118.64,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2022-02-01",
                            "price": 113.85,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2022-03-01",
                            "price": 115.17,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2022-04-01",
                            "price": 116.03,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2022-05-01",
                            "price": 116.59,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2022-06-01",
                            "price": 119.05,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2022-07-01",
                            "price": 122.72,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2022-08-01",
                            "price": 123.08,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2022-09-01",
                            "price": 122.94,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2022-10-01",
                            "price": 122.25,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2022-11-01",
                            "price": 123.75,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2022-12-01",
                            "price": 124.3,
                            "currency": "USD",
                            "type": "closing"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const InstrumentViewHttpServiceGetInstrumentViewByPositionIdMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/positions/{positionID}/instruments/views",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "f6739265-aed1-47b6-8423-63518ce9caa5",
                    "name": "Apple",
                    "code": "US0378331005",
                    "absolutePerformance": 1.46,
                    "relativePerformance": 1.2,
                    "price": 123.24,
                    "currency": "USD",
                    "pricePerformance": 7.81,
                    "pricePerformanceAbs": {
                        "amount": 1.48,
                        "currency": "USD"
                    },
                    "assetClassName": "Equity",
                    "subAssetClassName": "Tech Sector",
                    "ticker": "AAPL",
                    "exchange": "NASDAQ",
                    "countryCode": "USA",
                    "about": {
                        "about": "about",
                        "address": "address",
                        "phoneNumber": "phoneNumber",
                        "link": "link"
                    },
                    "keyStatistics": {
                        "country": "United States of America",
                        "purchasePrice": 118.34,
                        "exchangeRate": 1.21,
                        "unrealizedPLPct": 4.14,
                        "unrealizedPL": 4.9,
                        "changeInNetAssetValue": 7.81,
                        "fundStatus": "open-ended fund",
                        "totalAssets": 134.3,
                        "trailingTwelveMonthsYield": 1.4,
                        "oneYearReturn": 68.97,
                        "effectiveDuration": 4.51,
                        "portfolioAssetsAllocation": "Equity",
                        "quantity": 187,
                        "turnover": 30,
                        "creditRating": "AAA",
                        "priceToEarningsRatio": 37.59,
                        "estPriceToEarningsRatio": 31.06,
                        "priceToEarningsToGrowthRatio": 3.09,
                        "sharesOutstanding": 17.002,
                        "priceToBookRatio": 32.07,
                        "earningsPerShare": 1.39,
                        "dividendYield": 0.66,
                        "lastDividendReported": 0.2,
                        "nextEarningsAnnouncement": "2021-01-26",
                        "coupon": 5.63,
                        "accruedInterestOnBond": 5.52,
                        "maturity": "2028-01-04",
                        "yieldToMaturity": 0,
                        "duration": 7.63,
                        "couponFrequency": "annually",
                        "additionalKeyStatistics": {
                            "purchasePrice": "118.34",
                            "totalAssets": "50",
                            "coupon": "false"
                        }
                    },
                    "additions": {
                        "someKey": "someValue"
                    }
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const InstrumentViewHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v1/positions/{positionID}/instruments/charts",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "chartData": [{
                            "date": "2020-01-01",
                            "price": 125.01,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2020-02-01",
                            "price": 129.04,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2020-03-01",
                            "price": 134.18,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2020-04-01",
                            "price": 131.4,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2020-05-01",
                            "price": 120.88,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2020-06-01",
                            "price": 112.82,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2020-07-01",
                            "price": 115.54,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2020-08-01",
                            "price": 110.34,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2020-09-01",
                            "price": 106.84,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2020-10-01",
                            "price": 108.22,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2020-11-01",
                            "price": 116.5,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2020-12-01",
                            "price": 114.97,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2021-01-01",
                            "price": 124.4,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2021-02-01",
                            "price": 120.71,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2021-03-01",
                            "price": 116.6,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2021-04-01",
                            "price": 111.2,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2021-05-01",
                            "price": 115.32,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2021-06-01",
                            "price": 108.77,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2021-07-01",
                            "price": 114.95,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2021-08-01",
                            "price": 119.03,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2021-09-01",
                            "price": 118.69,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2021-10-01",
                            "price": 116.32,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2021-11-01",
                            "price": 115.97,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2021-12-01",
                            "price": 119.49,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2022-01-01",
                            "price": 118.64,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2022-02-01",
                            "price": 113.85,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2022-03-01",
                            "price": 115.17,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2022-04-01",
                            "price": 116.03,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2022-05-01",
                            "price": 116.59,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2022-06-01",
                            "price": 119.05,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2022-07-01",
                            "price": 122.72,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2022-08-01",
                            "price": 123.08,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2022-09-01",
                            "price": 122.94,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2022-10-01",
                            "price": 122.25,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2022-11-01",
                            "price": 123.75,
                            "currency": "USD",
                            "type": "closing"
                        }, {
                            "date": "2022-12-01",
                            "price": 124.3,
                            "currency": "USD",
                            "type": "closing"
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/positions/{positionID}/instruments/views",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "id": "f6739265-aed1-47b6-8423-63518ce9caa5",
                    "name": "Apple",
                    "code": "US0378331005",
                    "absolutePerformance": 1.46,
                    "relativePerformance": 1.2,
                    "price": 123.24,
                    "currency": "USD",
                    "pricePerformance": 7.81,
                    "pricePerformanceAbs": {
                        "amount": 1.48,
                        "currency": "USD"
                    },
                    "assetClassName": "Equity",
                    "subAssetClassName": "Tech Sector",
                    "ticker": "AAPL",
                    "exchange": "NASDAQ",
                    "countryCode": "USA",
                    "about": {
                        "about": "about",
                        "address": "address",
                        "phoneNumber": "phoneNumber",
                        "link": "link"
                    },
                    "keyStatistics": {
                        "country": "United States of America",
                        "purchasePrice": 118.34,
                        "exchangeRate": 1.21,
                        "unrealizedPLPct": 4.14,
                        "unrealizedPL": 4.9,
                        "changeInNetAssetValue": 7.81,
                        "fundStatus": "open-ended fund",
                        "totalAssets": 134.3,
                        "trailingTwelveMonthsYield": 1.4,
                        "oneYearReturn": 68.97,
                        "effectiveDuration": 4.51,
                        "portfolioAssetsAllocation": "Equity",
                        "quantity": 187,
                        "turnover": 30,
                        "creditRating": "AAA",
                        "priceToEarningsRatio": 37.59,
                        "estPriceToEarningsRatio": 31.06,
                        "priceToEarningsToGrowthRatio": 3.09,
                        "sharesOutstanding": 17.002,
                        "priceToBookRatio": 32.07,
                        "earningsPerShare": 1.39,
                        "dividendYield": 0.66,
                        "lastDividendReported": 0.2,
                        "nextEarningsAnnouncement": "2021-01-26",
                        "coupon": 5.63,
                        "accruedInterestOnBond": 5.52,
                        "maturity": "2028-01-04",
                        "yieldToMaturity": 0,
                        "duration": 7.63,
                        "couponFrequency": "annually",
                        "additionalKeyStatistics": {
                            "purchasePrice": "118.34",
                            "totalAssets": "50",
                            "coupon": "false"
                        }
                    },
                    "additions": {
                        "someKey": "someValue"
                    }
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
]);
