// @backbase/portfolio-summary-instrument-view-http-ang/esm2015/api/instrumentView.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { PORTFOLIO_SUMMARY_INSTRUMENT_VIEW_BASE_PATH } from '../variables';
import { PortfolioSummaryInstrumentViewConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class InstrumentViewHttpService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new PortfolioSummaryInstrumentViewConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    getInstrumentChartDataByPositionId(requestParameters, observe = 'body', reportProgress = false, options) {
        const _positionID = requestParameters["positionID"];
        if (_positionID === null || _positionID === undefined) {
            throw new Error('Required parameter positionID was null or undefined when calling getInstrumentChartDataByPositionId.');
        }
        const _fromDate = requestParameters["fromDate"];
        if (_fromDate === null || _fromDate === undefined) {
            throw new Error('Required parameter fromDate was null or undefined when calling getInstrumentChartDataByPositionId.');
        }
        const _toDate = requestParameters["toDate"];
        if (_toDate === null || _toDate === undefined) {
            throw new Error('Required parameter toDate was null or undefined when calling getInstrumentChartDataByPositionId.');
        }
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_fromDate !== undefined && _fromDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _fromDate, 'fromDate');
        }
        if (_toDate !== undefined && _toDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _toDate, 'toDate');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/positions/${encodeURIComponent(String(_positionID))}/instruments/charts`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getInstrumentChartDataByPositionIdUrl(requestParameters) {
        const _positionID = requestParameters["positionID"];
        if (_positionID === null || _positionID === undefined) {
            throw new Error('Required parameter positionID was null or undefined when calling getInstrumentChartDataByPositionId.');
        }
        const _fromDate = requestParameters["fromDate"];
        if (_fromDate === null || _fromDate === undefined) {
            throw new Error('Required parameter fromDate was null or undefined when calling getInstrumentChartDataByPositionId.');
        }
        const _toDate = requestParameters["toDate"];
        if (_toDate === null || _toDate === undefined) {
            throw new Error('Required parameter toDate was null or undefined when calling getInstrumentChartDataByPositionId.');
        }
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_fromDate !== undefined && _fromDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _fromDate, 'fromDate');
        }
        if (_toDate !== undefined && _toDate !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _toDate, 'toDate');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v1/positions/${encodeURIComponent(String(_positionID))}/instruments/charts${queryString ? `?${queryString}` : ''}`;
    }
    getInstrumentViewByPositionId(requestParameters, observe = 'body', reportProgress = false, options) {
        const _positionID = requestParameters["positionID"];
        if (_positionID === null || _positionID === undefined) {
            throw new Error('Required parameter positionID was null or undefined when calling getInstrumentViewByPositionId.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/positions/${encodeURIComponent(String(_positionID))}/instruments/views`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getInstrumentViewByPositionIdUrl(requestParameters) {
        const _positionID = requestParameters["positionID"];
        if (_positionID === null || _positionID === undefined) {
            throw new Error('Required parameter positionID was null or undefined when calling getInstrumentViewByPositionId.');
        }
        return `${this.configuration.basePath}/client-api/v1/positions/${encodeURIComponent(String(_positionID))}/instruments/views`;
    }
}
InstrumentViewHttpService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: InstrumentViewHttpService, deps: [{ token: i1.HttpClient }, { token: PORTFOLIO_SUMMARY_INSTRUMENT_VIEW_BASE_PATH, optional: true }, { token: i2.PortfolioSummaryInstrumentViewConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
InstrumentViewHttpService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: InstrumentViewHttpService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: InstrumentViewHttpService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [PORTFOLIO_SUMMARY_INSTRUMENT_VIEW_BASE_PATH]
                }] }, { type: i2.PortfolioSummaryInstrumentViewConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
