// @backbase/portfolio-summary-instrument-view-http-ang/esm2015/api/portfolioPositions.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const PortfolioPositionsHttpServiceExportPortfolioPositionsMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/portfolios/{portfolioId}/positions/export",
        method: "GET",
        responses: [
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const PortfolioPositionsHttpServiceListPortfolioPositionsMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/portfolios/{portfolioId}/positions",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "positions": [{
                            "id": "51c9d6e7-c3fa-40e2-a500-4b4e2f11cbd3",
                            "instrument": {
                                "id": "6163a37e-6148-46ac-bd46-f62f563dfa7e",
                                "name": "Apples and Pears",
                                "code": "UAPL",
                                "ticker": "UAPL",
                                "exchange": "NASDAQ",
                                "currency": "UAH",
                                "available": true,
                                "assetClass": {
                                    "id": "6153577f-ad4c-47a4-bab2-b2e7c7b172cd",
                                    "name": "Equities"
                                }
                            },
                            "quantity": 120,
                            "price": {
                                "currency": "UAH",
                                "amount": 1030
                            },
                            "exchangeRate": 0.99,
                            "valuation": {
                                "currency": "USD",
                                "amount": 1028
                            },
                            "accruedInterest": {
                                "amount": 12,
                                "currency": "USD"
                            },
                            "costPrice": {
                                "amount": 1001,
                                "currency": "UAH"
                            },
                            "costExchangeRate": 1.01,
                            "unrealizedPL": {
                                "amount": 0.2,
                                "currency": "USD"
                            },
                            "unrealizedPLPct": 0.02,
                            "assetClassPct": 0.4,
                            "portfolioPct": 0.33,
                            "country": {
                                "id": "87a87cfc-81a2-4193-aafb-fa08e699ba70",
                                "name": "Ukraine",
                                "code": "UA"
                            },
                            "additions": {
                                "someKey": "someValue"
                            }
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const PortfolioPositionsHttpServiceStreamPortfolioPositionsMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/portfolios/{portfolioId}/positions/stream",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "cursor": "554e02e5-67fb-4434-9c97-91fde8ec3ce4",
                    "hasMoreItems": false,
                    "positions": [{
                            "id": "554e02e5-67fb-4434-9c97-91fde8ec3ce4",
                            "region": {
                                "id": "96821684-70c0-46a9-8e05-2c1f0526cdd9",
                                "name": "Europe"
                            },
                            "country": {
                                "id": "51a95a45-63e0-4e43-9619-8ffb679f5d9b",
                                "name": "Ukraine",
                                "code": "UA"
                            },
                            "instrument": {
                                "id": "3e35104b-03e9-42be-b58f-8ee5d23d6ca8",
                                "name": "Apples and Plums",
                                "ticker": "UAPL",
                                "exchange": "NASDAQ",
                                "assetClass": {
                                    "id": "6153577f-ad4c-47a4-bab2-b2e7c7b172cd",
                                    "name": "Equities"
                                }
                            },
                            "assetClassPct": 0.24,
                            "valuation": {
                                "amount": 43630,
                                "currency": "USD"
                            },
                            "additions": {
                                "someKey": "someValue"
                            }
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const PortfolioPositionsHttpServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v1/portfolios/{portfolioId}/positions/export",
        method: "GET",
        responses: [
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/portfolios/{portfolioId}/positions",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "positions": [{
                            "id": "51c9d6e7-c3fa-40e2-a500-4b4e2f11cbd3",
                            "instrument": {
                                "id": "6163a37e-6148-46ac-bd46-f62f563dfa7e",
                                "name": "Apples and Pears",
                                "code": "UAPL",
                                "ticker": "UAPL",
                                "exchange": "NASDAQ",
                                "currency": "UAH",
                                "available": true,
                                "assetClass": {
                                    "id": "6153577f-ad4c-47a4-bab2-b2e7c7b172cd",
                                    "name": "Equities"
                                }
                            },
                            "quantity": 120,
                            "price": {
                                "currency": "UAH",
                                "amount": 1030
                            },
                            "exchangeRate": 0.99,
                            "valuation": {
                                "currency": "USD",
                                "amount": 1028
                            },
                            "accruedInterest": {
                                "amount": 12,
                                "currency": "USD"
                            },
                            "costPrice": {
                                "amount": 1001,
                                "currency": "UAH"
                            },
                            "costExchangeRate": 1.01,
                            "unrealizedPL": {
                                "amount": 0.2,
                                "currency": "USD"
                            },
                            "unrealizedPLPct": 0.02,
                            "assetClassPct": 0.4,
                            "portfolioPct": 0.33,
                            "country": {
                                "id": "87a87cfc-81a2-4193-aafb-fa08e699ba70",
                                "name": "Ukraine",
                                "code": "UA"
                            },
                            "additions": {
                                "someKey": "someValue"
                            }
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/portfolios/{portfolioId}/positions/stream",
        method: "GET",
        responses: [
            {
                status: 200,
                body: {
                    "cursor": "554e02e5-67fb-4434-9c97-91fde8ec3ce4",
                    "hasMoreItems": false,
                    "positions": [{
                            "id": "554e02e5-67fb-4434-9c97-91fde8ec3ce4",
                            "region": {
                                "id": "96821684-70c0-46a9-8e05-2c1f0526cdd9",
                                "name": "Europe"
                            },
                            "country": {
                                "id": "51a95a45-63e0-4e43-9619-8ffb679f5d9b",
                                "name": "Ukraine",
                                "code": "UA"
                            },
                            "instrument": {
                                "id": "3e35104b-03e9-42be-b58f-8ee5d23d6ca8",
                                "name": "Apples and Plums",
                                "ticker": "UAPL",
                                "exchange": "NASDAQ",
                                "assetClass": {
                                    "id": "6153577f-ad4c-47a4-bab2-b2e7c7b172cd",
                                    "name": "Equities"
                                }
                            },
                            "assetClassPct": 0.24,
                            "valuation": {
                                "amount": 43630,
                                "currency": "USD"
                            },
                            "additions": {
                                "someKey": "someValue"
                            }
                        }]
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
]);
