// @backbase/portfolio-summary-instrument-view-http-ang/esm2015/api/portfolioPositions.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { PORTFOLIO_SUMMARY_INSTRUMENT_VIEW_BASE_PATH } from '../variables';
import { PortfolioSummaryInstrumentViewConfiguration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class PortfolioPositionsHttpService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new PortfolioSummaryInstrumentViewConfiguration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    exportPortfolioPositions(requestParameters, observe = 'body', reportProgress = false, options) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling exportPortfolioPositions.');
        }
        const _fileType = requestParameters["fileType"];
        if (_fileType === null || _fileType === undefined) {
            throw new Error('Required parameter fileType was null or undefined when calling exportPortfolioPositions.');
        }
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_fileType !== undefined && _fileType !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _fileType, 'fileType');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'text/csv',
                'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/portfolios/${encodeURIComponent(String(_portfolioId))}/positions/export`, {
            params: queryParameters,
            responseType: "blob",
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    exportPortfolioPositionsUrl(requestParameters) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling exportPortfolioPositions.');
        }
        const _fileType = requestParameters["fileType"];
        if (_fileType === null || _fileType === undefined) {
            throw new Error('Required parameter fileType was null or undefined when calling exportPortfolioPositions.');
        }
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_fileType !== undefined && _fileType !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _fileType, 'fileType');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v1/portfolios/${encodeURIComponent(String(_portfolioId))}/positions/export${queryString ? `?${queryString}` : ''}`;
    }
    listPortfolioPositions(requestParameters, observe = 'body', reportProgress = false, options) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling listPortfolioPositions.');
        }
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        const _orderBy = requestParameters["orderBy"];
        const _direction = requestParameters["direction"];
        const _currencyCode = requestParameters["currencyCode"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_orderBy !== undefined && _orderBy !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _orderBy, 'orderBy');
        }
        if (_direction !== undefined && _direction !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _direction, 'direction');
        }
        if (_currencyCode !== undefined && _currencyCode !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _currencyCode, 'currencyCode');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/portfolios/${encodeURIComponent(String(_portfolioId))}/positions`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    listPortfolioPositionsUrl(requestParameters) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling listPortfolioPositions.');
        }
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        const _orderBy = requestParameters["orderBy"];
        const _direction = requestParameters["direction"];
        const _currencyCode = requestParameters["currencyCode"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_orderBy !== undefined && _orderBy !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _orderBy, 'orderBy');
        }
        if (_direction !== undefined && _direction !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _direction, 'direction');
        }
        if (_currencyCode !== undefined && _currencyCode !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _currencyCode, 'currencyCode');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v1/portfolios/${encodeURIComponent(String(_portfolioId))}/positions${queryString ? `?${queryString}` : ''}`;
    }
    streamPortfolioPositions(requestParameters, observe = 'body', reportProgress = false, options) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling streamPortfolioPositions.');
        }
        const _cursor = requestParameters["cursor"];
        const _size = requestParameters["size"];
        const _search = requestParameters["search"];
        const _currencyCode = requestParameters["currencyCode"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_cursor !== undefined && _cursor !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _cursor, 'cursor');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_search !== undefined && _search !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _search, 'search');
        }
        if (_currencyCode !== undefined && _currencyCode !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _currencyCode, 'currencyCode');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/portfolios/${encodeURIComponent(String(_portfolioId))}/positions/stream`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    streamPortfolioPositionsUrl(requestParameters) {
        const _portfolioId = requestParameters["portfolioId"];
        if (_portfolioId === null || _portfolioId === undefined) {
            throw new Error('Required parameter portfolioId was null or undefined when calling streamPortfolioPositions.');
        }
        const _cursor = requestParameters["cursor"];
        const _size = requestParameters["size"];
        const _search = requestParameters["search"];
        const _currencyCode = requestParameters["currencyCode"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_cursor !== undefined && _cursor !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _cursor, 'cursor');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_search !== undefined && _search !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _search, 'search');
        }
        if (_currencyCode !== undefined && _currencyCode !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _currencyCode, 'currencyCode');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v1/portfolios/${encodeURIComponent(String(_portfolioId))}/positions/stream${queryString ? `?${queryString}` : ''}`;
    }
}
PortfolioPositionsHttpService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PortfolioPositionsHttpService, deps: [{ token: i1.HttpClient }, { token: PORTFOLIO_SUMMARY_INSTRUMENT_VIEW_BASE_PATH, optional: true }, { token: i2.PortfolioSummaryInstrumentViewConfiguration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
PortfolioPositionsHttpService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PortfolioPositionsHttpService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: PortfolioPositionsHttpService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [PORTFOLIO_SUMMARY_INSTRUMENT_VIEW_BASE_PATH]
                }] }, { type: i2.PortfolioSummaryInstrumentViewConfiguration, decorators: [{
                    type: Optional
                }] }]; } });
