// @backbase/rtc-v1-client-ang/esm2015/api/conversations.service.js
import { Inject, Injectable, Optional } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { CustomHttpParameterCodec } from '../encoder';
import { RTC_V1_BASE_PATH } from '../variables';
import { RtcV1Configuration } from '../configuration';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "../configuration";
export class ConversationsService {
    constructor(httpClient, basePath, configuration) {
        this.httpClient = httpClient;
        this.basePath = '';
        this.defaultHeaders = new HttpHeaders();
        this.configuration = new RtcV1Configuration();
        if (configuration) {
            this.configuration = configuration;
        }
        if (typeof this.configuration.basePath !== 'string') {
            if (typeof basePath !== 'string') {
                basePath = this.basePath;
            }
            this.configuration.basePath = basePath;
        }
        this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
    }
    addToHttpParams(httpParams, value, key) {
        if (typeof value === "object" && value instanceof Date === false) {
            httpParams = this.addToHttpParamsRecursive(httpParams, value);
        }
        else {
            httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
        }
        return httpParams;
    }
    addToHttpParamsRecursive(httpParams, value, key) {
        if (value == null) {
            return httpParams;
        }
        if (typeof value === "object") {
            if (Array.isArray(value)) {
                value.forEach(elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
            }
            else if (value instanceof Date) {
                if (key != null) {
                    httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                }
                else {
                    throw Error("key may not be null if value is Date");
                }
            }
            else {
                Object.keys(value).forEach(k => httpParams = this.addToHttpParamsRecursive(httpParams, value[k], key != null ? `${key}.${k}` : k));
            }
        }
        else if (key != null) {
            httpParams = httpParams.append(key, value);
        }
        else {
            throw Error("key may not be null if value is not object or array");
        }
        return httpParams;
    }
    createConversation(observe = 'body', reportProgress = false, options) {
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.post(`${this.configuration.basePath}/client-api/v1/rtc/conversations`, null, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    createConversationUrl() {
        return `${this.configuration.basePath}/client-api/v1/rtc/conversations`;
    }
    endCall(requestParameters, observe = 'body', reportProgress = false, options) {
        const _conversationId = requestParameters["conversationId"];
        if (_conversationId === null || _conversationId === undefined) {
            throw new Error('Required parameter conversationId was null or undefined when calling endCall.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.delete(`${this.configuration.basePath}/client-api/v1/rtc/conversations/${encodeURIComponent(String(_conversationId))}/call`, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    endCallUrl(requestParameters) {
        const _conversationId = requestParameters["conversationId"];
        if (_conversationId === null || _conversationId === undefined) {
            throw new Error('Required parameter conversationId was null or undefined when calling endCall.');
        }
        return `${this.configuration.basePath}/client-api/v1/rtc/conversations/${encodeURIComponent(String(_conversationId))}/call`;
    }
    getConversations(requestParameters, observe = 'body', reportProgress = false, options) {
        const _userId = requestParameters["userId"];
        if (_userId === null || _userId === undefined) {
            throw new Error('Required parameter userId was null or undefined when calling getConversations.');
        }
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        const _conversationIds = requestParameters["conversationIds"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_conversationIds) {
            _conversationIds.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'conversationIds');
            });
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/rtc/users/${encodeURIComponent(String(_userId))}/conversations`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getConversationsUrl(requestParameters) {
        const _userId = requestParameters["userId"];
        if (_userId === null || _userId === undefined) {
            throw new Error('Required parameter userId was null or undefined when calling getConversations.');
        }
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        const _conversationIds = requestParameters["conversationIds"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        if (_conversationIds) {
            _conversationIds.forEach((element) => {
                queryParameters = this.addToHttpParams(queryParameters, element, 'conversationIds');
            });
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v1/rtc/users/${encodeURIComponent(String(_userId))}/conversations${queryString ? `?${queryString}` : ''}`;
    }
    getMessages(requestParameters, observe = 'body', reportProgress = false, options) {
        const _userId = requestParameters["userId"];
        if (_userId === null || _userId === undefined) {
            throw new Error('Required parameter userId was null or undefined when calling getMessages.');
        }
        const _conversationId = requestParameters["conversationId"];
        if (_conversationId === null || _conversationId === undefined) {
            throw new Error('Required parameter conversationId was null or undefined when calling getMessages.');
        }
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.get(`${this.configuration.basePath}/client-api/v1/rtc/users/${encodeURIComponent(String(_userId))}/conversations/${encodeURIComponent(String(_conversationId))}/messages`, {
            params: queryParameters,
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    getMessagesUrl(requestParameters) {
        const _userId = requestParameters["userId"];
        if (_userId === null || _userId === undefined) {
            throw new Error('Required parameter userId was null or undefined when calling getMessages.');
        }
        const _conversationId = requestParameters["conversationId"];
        if (_conversationId === null || _conversationId === undefined) {
            throw new Error('Required parameter conversationId was null or undefined when calling getMessages.');
        }
        const _from = requestParameters["from"];
        const _size = requestParameters["size"];
        let queryParameters = new HttpParams({ encoder: this.encoder });
        if (_from !== undefined && _from !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _from, 'from');
        }
        if (_size !== undefined && _size !== null) {
            queryParameters = this.addToHttpParams(queryParameters, _size, 'size');
        }
        const queryString = queryParameters.toString();
        return `${this.configuration.basePath}/client-api/v1/rtc/users/${encodeURIComponent(String(_userId))}/conversations/${encodeURIComponent(String(_conversationId))}/messages${queryString ? `?${queryString}` : ''}`;
    }
    initiateCall(requestParameters, observe = 'body', reportProgress = false, options) {
        const _conversationId = requestParameters["conversationId"];
        if (_conversationId === null || _conversationId === undefined) {
            throw new Error('Required parameter conversationId was null or undefined when calling initiateCall.');
        }
        const _callRequest = requestParameters["callRequest"];
        if (_callRequest === null || _callRequest === undefined) {
            throw new Error('Required parameter callRequest was null or undefined when calling initiateCall.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v1/rtc/conversations/${encodeURIComponent(String(_conversationId))}/call`, _callRequest, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    initiateCallUrl(requestParameters) {
        const _conversationId = requestParameters["conversationId"];
        if (_conversationId === null || _conversationId === undefined) {
            throw new Error('Required parameter conversationId was null or undefined when calling initiateCall.');
        }
        const _callRequest = requestParameters["callRequest"];
        if (_callRequest === null || _callRequest === undefined) {
            throw new Error('Required parameter callRequest was null or undefined when calling initiateCall.');
        }
        return `${this.configuration.basePath}/client-api/v1/rtc/conversations/${encodeURIComponent(String(_conversationId))}/call`;
    }
    updateConversationStatus(requestParameters, observe = 'body', reportProgress = false, options) {
        const _conversationId = requestParameters["conversationId"];
        if (_conversationId === null || _conversationId === undefined) {
            throw new Error('Required parameter conversationId was null or undefined when calling updateConversationStatus.');
        }
        const _conversationStatusUpdate = requestParameters["conversationStatusUpdate"];
        if (_conversationStatusUpdate === null || _conversationStatusUpdate === undefined) {
            throw new Error('Required parameter conversationStatusUpdate was null or undefined when calling updateConversationStatus.');
        }
        let headers = this.defaultHeaders;
        let httpHeaderAcceptSelected = options && options.httpHeaderAccept;
        if (httpHeaderAcceptSelected === undefined) {
            const httpHeaderAccepts = [
                'application/json'
            ];
            httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        }
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }
        const consumes = [
            'application/json'
        ];
        const httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
        if (httpContentTypeSelected !== undefined) {
            headers = headers.set('Content-Type', httpContentTypeSelected);
        }
        let responseType = 'json';
        if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
            responseType = 'text';
        }
        return this.httpClient.put(`${this.configuration.basePath}/client-api/v1/rtc/conversations/${encodeURIComponent(String(_conversationId))}/status`, _conversationStatusUpdate, {
            responseType: responseType,
            withCredentials: this.configuration.withCredentials,
            headers: headers,
            observe: observe,
            reportProgress: reportProgress
        });
    }
    updateConversationStatusUrl(requestParameters) {
        const _conversationId = requestParameters["conversationId"];
        if (_conversationId === null || _conversationId === undefined) {
            throw new Error('Required parameter conversationId was null or undefined when calling updateConversationStatus.');
        }
        const _conversationStatusUpdate = requestParameters["conversationStatusUpdate"];
        if (_conversationStatusUpdate === null || _conversationStatusUpdate === undefined) {
            throw new Error('Required parameter conversationStatusUpdate was null or undefined when calling updateConversationStatus.');
        }
        return `${this.configuration.basePath}/client-api/v1/rtc/conversations/${encodeURIComponent(String(_conversationId))}/status`;
    }
}
ConversationsService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConversationsService, deps: [{ token: i1.HttpClient }, { token: RTC_V1_BASE_PATH, optional: true }, { token: i2.RtcV1Configuration, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
ConversationsService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConversationsService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ConversationsService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [RTC_V1_BASE_PATH]
                }] }, { type: i2.RtcV1Configuration, decorators: [{
                    type: Optional
                }] }]; } });
