// @backbase/rtc-v1-client-ang/esm2015/api/conversations.service.mocks.js
import { createMocks } from '@backbase/foundation-ang/data-http';
export const ConversationsServiceCreateConversationMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/rtc/conversations",
        method: "POST",
        responses: [
            {
                status: 201,
                body: {
                    "id": "CH0c17647bed78463291d80a72acc7c58c"
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const ConversationsServiceEndCallMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/rtc/conversations/{conversationId}/call",
        method: "DELETE",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const ConversationsServiceGetConversationsMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/rtc/users/{userId}/conversations",
        method: "GET",
        responses: [
            {
                status: 200,
                body: []
            },
            {
                status: 200,
                body: [{
                        "id": "CH1",
                        "dateCreated": "2021-09-16T22:18:37Z",
                        "participants": [{
                                "id": "employee1",
                                "preferredName": "John Smith"
                            }, {
                                "id": "supervisor1",
                                "preferredName": "David Jones"
                            }]
                    }, {
                        "id": "CH2",
                        "dateCreated": "2021-09-16T20:18:37Z",
                        "participants": [{
                                "id": "employee1",
                                "preferredName": "John Smith"
                            }]
                    }, {
                        "id": "CH3",
                        "dateCreated": "2019-12-26T22:18:37Z",
                        "participants": []
                    }, {
                        "id": "CH4",
                        "dateCreated": "2019-12-25T22:18:37Z",
                        "participants": [{
                                "id": "employee2"
                            }]
                    }, {
                        "id": "CH524234",
                        "dateCreated": "2017-12-26T22:18:37Z",
                        "participants": [{
                                "id": "employee1",
                                "preferredName": "John Smith"
                            }, {
                                "id": "supervisor1",
                                "preferredName": "David Jones"
                            }]
                    }, {
                        "id": "CH6",
                        "dateCreated": "2016-12-26T22:18:37Z",
                        "participants": [{
                                "id": "employee1"
                            }]
                    }, {
                        "id": "CH7",
                        "dateCreated": "2015-12-26T22:18:37Z",
                        "participants": [{
                                "id": "employee1"
                            }]
                    }, {
                        "id": "CH8",
                        "dateCreated": "2014-12-26T22:18:37Z",
                        "participants": [{
                                "id": "employee8"
                            }]
                    }, {
                        "id": "CH9",
                        "dateCreated": "2014-10-26T22:18:37Z",
                        "participants": [{
                                "id": "employee9"
                            }]
                    }, {
                        "id": "CH10",
                        "dateCreated": "2014-09-26T22:18:37Z",
                        "participants": [{
                                "id": "employee10"
                            }]
                    }]
            },
            {
                status: 200,
                body: [{
                        "id": "CH1",
                        "dateCreated": "2020-12-16T22:18:37Z",
                        "participants": [{
                                "id": "employee1",
                                "preferredName": "John Smith"
                            }, {
                                "id": "supervisor1",
                                "preferredName": "David Jones"
                            }]
                    }, {
                        "id": "CH2",
                        "dateCreated": "2019-12-16T22:18:37Z",
                        "participants": [{
                                "id": "employee1",
                                "preferredName": "John Smith"
                            }]
                    }, {
                        "id": "CH3",
                        "dateCreated": "2019-11-26T22:18:37Z",
                        "participants": []
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const ConversationsServiceGetMessagesMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/rtc/users/{userId}/conversations/{conversationId}/messages",
        method: "GET",
        responses: [
            {
                status: 200,
                body: []
            },
            {
                status: 200,
                body: [{
                        "id": "IM426d0acb198c43df8b42ee6e206fddb9",
                        "type": "CONVERSATION_ACCEPTED",
                        "body": null,
                        "media": [],
                        "authorId": "admin",
                        "authorPreferredName": "Kjiro",
                        "dateCreated": "2021-11-16T20:08:30Z",
                        "dateUpdated": "2021-11-16T20:08:30Z"
                    }, {
                        "id": "IMa7443cccfc9946a2810ab019396ce15c",
                        "type": "MESSAGE",
                        "body": "asdf",
                        "media": [],
                        "authorId": "admin",
                        "authorPreferredName": "Kjiro",
                        "dateCreated": "2021-11-16T20:08:32Z",
                        "dateUpdated": "2021-11-16T20:08:32Z"
                    }, {
                        "id": "IM81f239c9222b4f548c394489ace20ecc",
                        "type": "MESSAGE",
                        "body": "123123",
                        "media": [],
                        "authorId": "admin",
                        "authorPreferredName": "Kjiro",
                        "dateCreated": "2021-11-16T23:11:32Z",
                        "dateUpdated": "2021-11-16T23:11:32Z"
                    }, {
                        "id": "IM079552e6556c4ed6825b84597fa7c691",
                        "type": "MESSAGE",
                        "body": null,
                        "media": [{
                                "id": "MEd726969bac4ea9a8ea0ed1b4881d982f",
                                "contentType": "image/png",
                                "fileName": null,
                                "size": 226933
                            }],
                        "authorId": "customer",
                        "authorPreferredName": null,
                        "dateCreated": "2021-11-17T00:32:29Z",
                        "dateUpdated": "2021-11-17T00:32:29Z"
                    }, {
                        "id": "IM079552e6556c4ed6825b84597fa7c780",
                        "type": "CONVERSATION_ENDED",
                        "body": null,
                        "media": [],
                        "authorId": "admin",
                        "authorPreferredName": "Kjiro",
                        "dateCreated": "2021-11-17T00:52:29Z",
                        "dateUpdated": "2021-11-17T00:52:29Z"
                    }]
            },
            {
                status: 200,
                body: [{
                        "id": "IM426d0acb198c43df8b42ee6e206fddb9",
                        "type": "CONVERSATION_ACCEPTED",
                        "body": null,
                        "media": [],
                        "authorId": "admin",
                        "authorPreferredName": "Kjiro",
                        "dateCreated": "2021-11-16T20:08:30Z",
                        "dateUpdated": "2021-11-16T20:08:30Z"
                    }, {
                        "id": "IMa7443cccfc9946a2810ab019396ce15c",
                        "type": "MESSAGE",
                        "body": "asdf",
                        "media": [],
                        "authorId": "admin",
                        "authorPreferredName": "Kjiro",
                        "dateCreated": "2021-11-16T20:08:32Z",
                        "dateUpdated": "2021-11-16T20:08:32Z"
                    }, {
                        "id": "IM81f239c9222b4f548c394489ace20ecc",
                        "type": "MESSAGE",
                        "body": "123123",
                        "media": [],
                        "authorId": "admin",
                        "authorPreferredName": "Kjiro",
                        "dateCreated": "2021-11-16T23:11:32Z",
                        "dateUpdated": "2021-11-16T23:11:32Z"
                    }, {
                        "id": "IM079552e6556c4ed6825b84597fa7c691",
                        "type": "MESSAGE",
                        "body": null,
                        "media": [{
                                "id": "MEd726969bac4ea9a8ea0ed1b4881d982f",
                                "contentType": "image/png",
                                "fileName": null,
                                "size": 226933
                            }],
                        "authorId": "customer",
                        "authorPreferredName": null,
                        "dateCreated": "2021-11-17T00:32:29Z",
                        "dateUpdated": "2021-11-17T00:32:29Z"
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const ConversationsServiceInitiateCallMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/rtc/conversations/{conversationId}/call",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {
                    "conversationToken": "eyJjdHkiOiJ0d2lsaW8tZnBhO3Y9MSIsInR5cCI6IkpXVCIsImFsZyI6IkhTMjU2In0.eyJpc3MiOiJTSzRmNjhlMDg2ZjY1NzVlYTY1NTAyM2E0MDBkNDNmNzNhIiwiZXhwIjoxNjI2MzcxMTE4LCJncmFudHMiOnsiaWRlbnRpdHkiOiJBTk9OWU1PVVMiLCJjaGF0Ijp7InNlcnZpY2Vfc2lkIjoiSVNhYjlkMzFkNzdmZTg0MjJmOWFlNTJiNGM2NTVjMTNhNyJ9fSwianRpIjoiU0s0ZjY4ZTA4NmY2NTc1ZWE2NTUwMjNhNDAwZDQzZjczYS0xNjI2MzQyMjcyIiwic3ViIjoiQUM0YmM1N2RjYTZmOWY3MTdlNDdkOWRlYzc4NGRjOTJhMCJ9.MFtIBvMfl3oYHOILsPMYklhdUuhsdwviTV_Pxdxi_6g",
                    "callTimeout": 120
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const ConversationsServiceUpdateConversationStatusMocksProvider = createMocks([{
        urlPattern: "/client-api/v1/rtc/conversations/{conversationId}/status",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {
                    "conversationId": "CH0c17647bed78463291d80a72acc7c58c",
                    "status": "assigned"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    }]);
export const ConversationsServiceMocksProvider = createMocks([
    {
        urlPattern: "/client-api/v1/rtc/conversations",
        method: "POST",
        responses: [
            {
                status: 201,
                body: {
                    "id": "CH0c17647bed78463291d80a72acc7c58c"
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/rtc/conversations/{conversationId}/call",
        method: "DELETE",
        responses: [
            {
                status: 204,
                body: ""
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/rtc/users/{userId}/conversations",
        method: "GET",
        responses: [
            {
                status: 200,
                body: []
            },
            {
                status: 200,
                body: [{
                        "id": "CH1",
                        "dateCreated": "2021-09-16T22:18:37Z",
                        "participants": [{
                                "id": "employee1",
                                "preferredName": "John Smith"
                            }, {
                                "id": "supervisor1",
                                "preferredName": "David Jones"
                            }]
                    }, {
                        "id": "CH2",
                        "dateCreated": "2021-09-16T20:18:37Z",
                        "participants": [{
                                "id": "employee1",
                                "preferredName": "John Smith"
                            }]
                    }, {
                        "id": "CH3",
                        "dateCreated": "2019-12-26T22:18:37Z",
                        "participants": []
                    }, {
                        "id": "CH4",
                        "dateCreated": "2019-12-25T22:18:37Z",
                        "participants": [{
                                "id": "employee2"
                            }]
                    }, {
                        "id": "CH524234",
                        "dateCreated": "2017-12-26T22:18:37Z",
                        "participants": [{
                                "id": "employee1",
                                "preferredName": "John Smith"
                            }, {
                                "id": "supervisor1",
                                "preferredName": "David Jones"
                            }]
                    }, {
                        "id": "CH6",
                        "dateCreated": "2016-12-26T22:18:37Z",
                        "participants": [{
                                "id": "employee1"
                            }]
                    }, {
                        "id": "CH7",
                        "dateCreated": "2015-12-26T22:18:37Z",
                        "participants": [{
                                "id": "employee1"
                            }]
                    }, {
                        "id": "CH8",
                        "dateCreated": "2014-12-26T22:18:37Z",
                        "participants": [{
                                "id": "employee8"
                            }]
                    }, {
                        "id": "CH9",
                        "dateCreated": "2014-10-26T22:18:37Z",
                        "participants": [{
                                "id": "employee9"
                            }]
                    }, {
                        "id": "CH10",
                        "dateCreated": "2014-09-26T22:18:37Z",
                        "participants": [{
                                "id": "employee10"
                            }]
                    }]
            },
            {
                status: 200,
                body: [{
                        "id": "CH1",
                        "dateCreated": "2020-12-16T22:18:37Z",
                        "participants": [{
                                "id": "employee1",
                                "preferredName": "John Smith"
                            }, {
                                "id": "supervisor1",
                                "preferredName": "David Jones"
                            }]
                    }, {
                        "id": "CH2",
                        "dateCreated": "2019-12-16T22:18:37Z",
                        "participants": [{
                                "id": "employee1",
                                "preferredName": "John Smith"
                            }]
                    }, {
                        "id": "CH3",
                        "dateCreated": "2019-11-26T22:18:37Z",
                        "participants": []
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/rtc/users/{userId}/conversations/{conversationId}/messages",
        method: "GET",
        responses: [
            {
                status: 200,
                body: []
            },
            {
                status: 200,
                body: [{
                        "id": "IM426d0acb198c43df8b42ee6e206fddb9",
                        "type": "CONVERSATION_ACCEPTED",
                        "body": null,
                        "media": [],
                        "authorId": "admin",
                        "authorPreferredName": "Kjiro",
                        "dateCreated": "2021-11-16T20:08:30Z",
                        "dateUpdated": "2021-11-16T20:08:30Z"
                    }, {
                        "id": "IMa7443cccfc9946a2810ab019396ce15c",
                        "type": "MESSAGE",
                        "body": "asdf",
                        "media": [],
                        "authorId": "admin",
                        "authorPreferredName": "Kjiro",
                        "dateCreated": "2021-11-16T20:08:32Z",
                        "dateUpdated": "2021-11-16T20:08:32Z"
                    }, {
                        "id": "IM81f239c9222b4f548c394489ace20ecc",
                        "type": "MESSAGE",
                        "body": "123123",
                        "media": [],
                        "authorId": "admin",
                        "authorPreferredName": "Kjiro",
                        "dateCreated": "2021-11-16T23:11:32Z",
                        "dateUpdated": "2021-11-16T23:11:32Z"
                    }, {
                        "id": "IM079552e6556c4ed6825b84597fa7c691",
                        "type": "MESSAGE",
                        "body": null,
                        "media": [{
                                "id": "MEd726969bac4ea9a8ea0ed1b4881d982f",
                                "contentType": "image/png",
                                "fileName": null,
                                "size": 226933
                            }],
                        "authorId": "customer",
                        "authorPreferredName": null,
                        "dateCreated": "2021-11-17T00:32:29Z",
                        "dateUpdated": "2021-11-17T00:32:29Z"
                    }, {
                        "id": "IM079552e6556c4ed6825b84597fa7c780",
                        "type": "CONVERSATION_ENDED",
                        "body": null,
                        "media": [],
                        "authorId": "admin",
                        "authorPreferredName": "Kjiro",
                        "dateCreated": "2021-11-17T00:52:29Z",
                        "dateUpdated": "2021-11-17T00:52:29Z"
                    }]
            },
            {
                status: 200,
                body: [{
                        "id": "IM426d0acb198c43df8b42ee6e206fddb9",
                        "type": "CONVERSATION_ACCEPTED",
                        "body": null,
                        "media": [],
                        "authorId": "admin",
                        "authorPreferredName": "Kjiro",
                        "dateCreated": "2021-11-16T20:08:30Z",
                        "dateUpdated": "2021-11-16T20:08:30Z"
                    }, {
                        "id": "IMa7443cccfc9946a2810ab019396ce15c",
                        "type": "MESSAGE",
                        "body": "asdf",
                        "media": [],
                        "authorId": "admin",
                        "authorPreferredName": "Kjiro",
                        "dateCreated": "2021-11-16T20:08:32Z",
                        "dateUpdated": "2021-11-16T20:08:32Z"
                    }, {
                        "id": "IM81f239c9222b4f548c394489ace20ecc",
                        "type": "MESSAGE",
                        "body": "123123",
                        "media": [],
                        "authorId": "admin",
                        "authorPreferredName": "Kjiro",
                        "dateCreated": "2021-11-16T23:11:32Z",
                        "dateUpdated": "2021-11-16T23:11:32Z"
                    }, {
                        "id": "IM079552e6556c4ed6825b84597fa7c691",
                        "type": "MESSAGE",
                        "body": null,
                        "media": [{
                                "id": "MEd726969bac4ea9a8ea0ed1b4881d982f",
                                "contentType": "image/png",
                                "fileName": null,
                                "size": 226933
                            }],
                        "authorId": "customer",
                        "authorPreferredName": null,
                        "dateCreated": "2021-11-17T00:32:29Z",
                        "dateUpdated": "2021-11-17T00:32:29Z"
                    }]
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/rtc/conversations/{conversationId}/call",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {
                    "conversationToken": "eyJjdHkiOiJ0d2lsaW8tZnBhO3Y9MSIsInR5cCI6IkpXVCIsImFsZyI6IkhTMjU2In0.eyJpc3MiOiJTSzRmNjhlMDg2ZjY1NzVlYTY1NTAyM2E0MDBkNDNmNzNhIiwiZXhwIjoxNjI2MzcxMTE4LCJncmFudHMiOnsiaWRlbnRpdHkiOiJBTk9OWU1PVVMiLCJjaGF0Ijp7InNlcnZpY2Vfc2lkIjoiSVNhYjlkMzFkNzdmZTg0MjJmOWFlNTJiNGM2NTVjMTNhNyJ9fSwianRpIjoiU0s0ZjY4ZTA4NmY2NTc1ZWE2NTUwMjNhNDAwZDQzZjczYS0xNjI2MzQyMjcyIiwic3ViIjoiQUM0YmM1N2RjYTZmOWY3MTdlNDdkOWRlYzc4NGRjOTJhMCJ9.MFtIBvMfl3oYHOILsPMYklhdUuhsdwviTV_Pxdxi_6g",
                    "callTimeout": 120
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
    {
        urlPattern: "/client-api/v1/rtc/conversations/{conversationId}/status",
        method: "PUT",
        responses: [
            {
                status: 200,
                body: {
                    "conversationId": "CH0c17647bed78463291d80a72acc7c58c",
                    "status": "assigned"
                }
            },
            {
                status: 400,
                body: {
                    "message": "Bad Request",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Value Exceeded. Must be between {min} and {max}.",
                            "key": "common.api.shoesize",
                            "context": {
                                "max": "50",
                                "min": "1"
                            }
                        }]
                }
            },
            {
                status: 401,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to invalid credentials.",
                            "key": "common.api.token",
                            "context": {
                                "accessToken": "expired"
                            }
                        }]
                }
            },
            {
                status: 403,
                body: {
                    "message": "Access to requested resource denied.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Resource access denied due to an insufficient user quota of {quota}.",
                            "key": "common.api.quota",
                            "context": {
                                "quota": "someQuota"
                            }
                        }]
                }
            },
            {
                status: 404,
                body: {
                    "message": "Resource not found.",
                    "key": "GENERAL_ERROR",
                    "errors": [{
                            "message": "Unable to find the resource requested resource: {resource}.",
                            "key": "common.api.resource",
                            "context": {
                                "resource": "aResource"
                            }
                        }]
                }
            },
            {
                status: 500,
                body: {
                    "message": "Description of error"
                }
            },
        ]
    },
]);
