// @backbase/ui-ang/esm2015/product-item-debit-card/product-item-debit-card.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { CustomProductItemDebitCardTitleDirective, ProductItemDebitCardComponent, } from './product-item-debit-card.component';
import { AmountModule } from '@backbase/ui-ang/amount';
import { PaymentCardNumberModule } from '@backbase/ui-ang/payment-card-number-pipe';
import { IconModule } from '@backbase/ui-ang/icon';
import * as i0 from "@angular/core";
export class ProductItemDebitCardModule {
}
ProductItemDebitCardModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemDebitCardModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductItemDebitCardModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemDebitCardModule, declarations: [ProductItemDebitCardComponent, CustomProductItemDebitCardTitleDirective], imports: [CommonModule, AmountModule, PaymentCardNumberModule, IconModule], exports: [ProductItemDebitCardComponent, CustomProductItemDebitCardTitleDirective] });
ProductItemDebitCardModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemDebitCardModule, imports: [[CommonModule, AmountModule, PaymentCardNumberModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ProductItemDebitCardModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, AmountModule, PaymentCardNumberModule, IconModule],
                    declarations: [ProductItemDebitCardComponent, CustomProductItemDebitCardTitleDirective],
                    exports: [ProductItemDebitCardComponent, CustomProductItemDebitCardTitleDirective],
                }]
        }] });
