// @backbase/authorized-user-http-ang/bundles/backbase-authorized-user-http-ang.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common/http'), require('@backbase/foundation-ang/data-http')) :
    typeof define === 'function' && define.amd ? define('@backbase/authorized-user-http-ang', ['exports', '@angular/core', '@angular/common/http', '@backbase/foundation-ang/data-http'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.backbase = global.backbase || {}, global.backbase["authorized-user-http-ang"] = {}), global.ng.core, global.ng.common.http, global.i2));
})(this, (function (exports, i0, i1, i2) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);

    exports.GetAuthorizedUserItem = void 0;
    (function (GetAuthorizedUserItem) {
        GetAuthorizedUserItem.StatusEnum = {
            ACTIVE: 'ACTIVE',
            AWAITINGREGISTRATION: 'AWAITING_REGISTRATION',
            PAUSED: 'PAUSED'
        };
    })(exports.GetAuthorizedUserItem || (exports.GetAuthorizedUserItem = {}));

    var AUTHORIZED_USER_BASE_PATH = new i0.InjectionToken('AUTHORIZED_USER_BASE_PATH');
    var COLLECTION_FORMATS = {
        'csv': ',',
        'tsv': '   ',
        'ssv': ' ',
        'pipes': '|'
    };

    var AuthorizedUserConfiguration = /** @class */ (function () {
        function AuthorizedUserConfiguration(configurationParameters) {
            if (configurationParameters === void 0) { configurationParameters = {}; }
            this.apiKeys = configurationParameters.apiKeys;
            this.username = configurationParameters.username;
            this.password = configurationParameters.password;
            this.accessToken = configurationParameters.accessToken;
            this.basePath = configurationParameters.basePath;
            this.withCredentials = configurationParameters.withCredentials;
            this.encoder = configurationParameters.encoder;
            if (configurationParameters.credentials) {
                this.credentials = configurationParameters.credentials;
            }
            else {
                this.credentials = {};
            }
        }
        AuthorizedUserConfiguration.prototype.selectHeaderContentType = function (contentTypes) {
            var _this = this;
            if (contentTypes.length === 0) {
                return undefined;
            }
            var type = contentTypes.find(function (x) { return _this.isJsonMime(x); });
            if (type === undefined) {
                return contentTypes[0];
            }
            return type;
        };
        AuthorizedUserConfiguration.prototype.selectHeaderAccept = function (accepts) {
            var _this = this;
            if (accepts.length === 0) {
                return undefined;
            }
            var type = accepts.find(function (x) { return _this.isJsonMime(x); });
            if (type === undefined) {
                return accepts[0];
            }
            return type;
        };
        AuthorizedUserConfiguration.prototype.isJsonMime = function (mime) {
            var jsonMime = new RegExp('^(application\/json|[^;/ \t]+\/[^;/ \t]+[+]json)[ \t]*(;.*)?$', 'i');
            return mime !== null && (jsonMime.test(mime) || mime.toLowerCase() === 'application/json-patch+json');
        };
        AuthorizedUserConfiguration.prototype.lookupCredential = function (key) {
            var value = this.credentials[key];
            return typeof value === 'function'
                ? value()
                : value;
        };
        return AuthorizedUserConfiguration;
    }());
    var CONFIG_TOKEN = new i0.InjectionToken('api-data-module AuthorizedUserConfiguration');

    var AuthorizedUserApiModule = /** @class */ (function () {
        function AuthorizedUserApiModule(parentModule, http, dataModulesManager, config) {
            if (parentModule) {
                throw new Error('AuthorizedUserApiModule is already loaded. Import in your base AppModule only.');
            }
            if (!http) {
                throw new Error('You need to import the HttpClientModule in your AppModule! \n' +
                    'See also https://github.com/angular/angular/issues/20575');
            }
            if (dataModulesManager) {
                dataModulesManager.setModuleConfig(CONFIG_TOKEN, {
                    apiRoot: '',
                    servicePath: config.basePath || '',
                    headers: {},
                });
            }
        }
        AuthorizedUserApiModule.forRoot = function (configurationFactory) {
            return {
                ngModule: AuthorizedUserApiModule,
                providers: [{ provide: AuthorizedUserConfiguration, useFactory: configurationFactory }]
            };
        };
        return AuthorizedUserApiModule;
    }());
    AuthorizedUserApiModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: AuthorizedUserApiModule, deps: [{ token: AuthorizedUserApiModule, optional: true, skipSelf: true }, { token: i1__namespace.HttpClient, optional: true }, { token: i2__namespace.DataModulesManager, optional: true }, { token: AuthorizedUserConfiguration }], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    AuthorizedUserApiModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: AuthorizedUserApiModule });
    AuthorizedUserApiModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: AuthorizedUserApiModule, providers: [], imports: [[]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: AuthorizedUserApiModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [],
                        declarations: [],
                        exports: [],
                        providers: []
                    }]
            }], ctorParameters: function () {
            return [{ type: AuthorizedUserApiModule, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.SkipSelf
                        }] }, { type: i1__namespace.HttpClient, decorators: [{
                            type: i0.Optional
                        }] }, { type: i2__namespace.DataModulesManager, decorators: [{
                            type: i0.Optional
                        }] }, { type: AuthorizedUserConfiguration }];
        } });

    var CustomHttpParameterCodec = /** @class */ (function () {
        function CustomHttpParameterCodec() {
        }
        CustomHttpParameterCodec.prototype.encodeKey = function (k) {
            return encodeURIComponent(k);
        };
        CustomHttpParameterCodec.prototype.encodeValue = function (v) {
            return encodeURIComponent(v);
        };
        CustomHttpParameterCodec.prototype.decodeKey = function (k) {
            return decodeURIComponent(k);
        };
        CustomHttpParameterCodec.prototype.decodeValue = function (v) {
            return decodeURIComponent(v);
        };
        return CustomHttpParameterCodec;
    }());

    var AuthorizedUserService = /** @class */ (function () {
        function AuthorizedUserService(httpClient, basePath, configuration) {
            this.httpClient = httpClient;
            this.basePath = '';
            this.defaultHeaders = new i1.HttpHeaders();
            this.configuration = new AuthorizedUserConfiguration();
            if (configuration) {
                this.configuration = configuration;
            }
            if (typeof this.configuration.basePath !== 'string') {
                if (typeof basePath !== 'string') {
                    basePath = this.basePath;
                }
                this.configuration.basePath = basePath;
            }
            this.encoder = this.configuration.encoder || new CustomHttpParameterCodec();
        }
        AuthorizedUserService.prototype.addToHttpParams = function (httpParams, value, key) {
            if (typeof value === "object" && value instanceof Date === false) {
                httpParams = this.addToHttpParamsRecursive(httpParams, value);
            }
            else {
                httpParams = this.addToHttpParamsRecursive(httpParams, value, key);
            }
            return httpParams;
        };
        AuthorizedUserService.prototype.addToHttpParamsRecursive = function (httpParams, value, key) {
            var _this = this;
            if (value == null) {
                return httpParams;
            }
            if (typeof value === "object") {
                if (Array.isArray(value)) {
                    value.forEach(function (elem) { return httpParams = _this.addToHttpParamsRecursive(httpParams, elem, key); });
                }
                else if (value instanceof Date) {
                    if (key != null) {
                        httpParams = httpParams.append(key, value.toISOString().substr(0, 10));
                    }
                    else {
                        throw Error("key may not be null if value is Date");
                    }
                }
                else {
                    Object.keys(value).forEach(function (k) { return httpParams = _this.addToHttpParamsRecursive(httpParams, value[k], key != null ? key + "." + k : k); });
                }
            }
            else if (key != null) {
                httpParams = httpParams.append(key, value);
            }
            else {
                throw Error("key may not be null if value is not object or array");
            }
            return httpParams;
        };
        AuthorizedUserService.prototype.createAuthorizedUser = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _postAuthorizedUser = requestParameters["postAuthorizedUser"];
            if (_postAuthorizedUser === null || _postAuthorizedUser === undefined) {
                throw new Error('Required parameter postAuthorizedUser was null or undefined when calling createAuthorizedUser.');
            }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var consumes = [
                'application/json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected !== undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.post(this.configuration.basePath + "/client-api/v2/users/me/authorized-users", _postAuthorizedUser, {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        AuthorizedUserService.prototype.createAuthorizedUserUrl = function (requestParameters) {
            var _postAuthorizedUser = requestParameters["postAuthorizedUser"];
            if (_postAuthorizedUser === null || _postAuthorizedUser === undefined) {
                throw new Error('Required parameter postAuthorizedUser was null or undefined when calling createAuthorizedUser.');
            }
            return this.configuration.basePath + "/client-api/v2/users/me/authorized-users";
        };
        AuthorizedUserService.prototype.deleteAuthorizedUser = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _authorizedUserId = requestParameters["authorizedUserId"];
            if (_authorizedUserId === null || _authorizedUserId === undefined) {
                throw new Error('Required parameter authorizedUserId was null or undefined when calling deleteAuthorizedUser.');
            }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.delete(this.configuration.basePath + "/client-api/v2/users/me/authorized-users/" + encodeURIComponent(String(_authorizedUserId)), {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        AuthorizedUserService.prototype.deleteAuthorizedUserUrl = function (requestParameters) {
            var _authorizedUserId = requestParameters["authorizedUserId"];
            if (_authorizedUserId === null || _authorizedUserId === undefined) {
                throw new Error('Required parameter authorizedUserId was null or undefined when calling deleteAuthorizedUser.');
            }
            return this.configuration.basePath + "/client-api/v2/users/me/authorized-users/" + encodeURIComponent(String(_authorizedUserId));
        };
        AuthorizedUserService.prototype.getAuthorizedUsers = function (observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.get(this.configuration.basePath + "/client-api/v2/users/me/authorized-users", {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        AuthorizedUserService.prototype.getAuthorizedUsersUrl = function () {
            return this.configuration.basePath + "/client-api/v2/users/me/authorized-users";
        };
        AuthorizedUserService.prototype.getPermissionsByAuthorizedUserId = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _authorizedUserId = requestParameters["authorizedUserId"];
            if (_authorizedUserId === null || _authorizedUserId === undefined) {
                throw new Error('Required parameter authorizedUserId was null or undefined when calling getPermissionsByAuthorizedUserId.');
            }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.get(this.configuration.basePath + "/client-api/v2/users/me/authorized-users/" + encodeURIComponent(String(_authorizedUserId)) + "/permissions", {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        AuthorizedUserService.prototype.getPermissionsByAuthorizedUserIdUrl = function (requestParameters) {
            var _authorizedUserId = requestParameters["authorizedUserId"];
            if (_authorizedUserId === null || _authorizedUserId === undefined) {
                throw new Error('Required parameter authorizedUserId was null or undefined when calling getPermissionsByAuthorizedUserId.');
            }
            return this.configuration.basePath + "/client-api/v2/users/me/authorized-users/" + encodeURIComponent(String(_authorizedUserId)) + "/permissions";
        };
        AuthorizedUserService.prototype.updatePermissionsForAuthorizedUser = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _authorizedUserId = requestParameters["authorizedUserId"];
            if (_authorizedUserId === null || _authorizedUserId === undefined) {
                throw new Error('Required parameter authorizedUserId was null or undefined when calling updatePermissionsForAuthorizedUser.');
            }
            var _putAuthorizedUserPermissions = requestParameters["putAuthorizedUserPermissions"];
            if (_putAuthorizedUserPermissions === null || _putAuthorizedUserPermissions === undefined) {
                throw new Error('Required parameter putAuthorizedUserPermissions was null or undefined when calling updatePermissionsForAuthorizedUser.');
            }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var consumes = [
                'application/json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected !== undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.put(this.configuration.basePath + "/client-api/v2/users/me/authorized-users/" + encodeURIComponent(String(_authorizedUserId)) + "/permissions", _putAuthorizedUserPermissions, {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        AuthorizedUserService.prototype.updatePermissionsForAuthorizedUserUrl = function (requestParameters) {
            var _authorizedUserId = requestParameters["authorizedUserId"];
            if (_authorizedUserId === null || _authorizedUserId === undefined) {
                throw new Error('Required parameter authorizedUserId was null or undefined when calling updatePermissionsForAuthorizedUser.');
            }
            var _putAuthorizedUserPermissions = requestParameters["putAuthorizedUserPermissions"];
            if (_putAuthorizedUserPermissions === null || _putAuthorizedUserPermissions === undefined) {
                throw new Error('Required parameter putAuthorizedUserPermissions was null or undefined when calling updatePermissionsForAuthorizedUser.');
            }
            return this.configuration.basePath + "/client-api/v2/users/me/authorized-users/" + encodeURIComponent(String(_authorizedUserId)) + "/permissions";
        };
        AuthorizedUserService.prototype.updateStatusForAuthorizedUser = function (requestParameters, observe, reportProgress, options) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var _authorizedUserId = requestParameters["authorizedUserId"];
            if (_authorizedUserId === null || _authorizedUserId === undefined) {
                throw new Error('Required parameter authorizedUserId was null or undefined when calling updateStatusForAuthorizedUser.');
            }
            var _updateAuthorizedUserStatus = requestParameters["updateAuthorizedUserStatus"];
            if (_updateAuthorizedUserStatus === null || _updateAuthorizedUserStatus === undefined) {
                throw new Error('Required parameter updateAuthorizedUserStatus was null or undefined when calling updateStatusForAuthorizedUser.');
            }
            var headers = this.defaultHeaders;
            var httpHeaderAcceptSelected = options && options.httpHeaderAccept;
            if (httpHeaderAcceptSelected === undefined) {
                var httpHeaderAccepts = [
                    'application/json'
                ];
                httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            }
            if (httpHeaderAcceptSelected !== undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            var consumes = [
                'application/json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected !== undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            var responseType = 'json';
            if (httpHeaderAcceptSelected && httpHeaderAcceptSelected.startsWith('text')) {
                responseType = 'text';
            }
            return this.httpClient.put(this.configuration.basePath + "/client-api/v2/users/me/authorized-users/" + encodeURIComponent(String(_authorizedUserId)) + "/status", _updateAuthorizedUserStatus, {
                responseType: responseType,
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        AuthorizedUserService.prototype.updateStatusForAuthorizedUserUrl = function (requestParameters) {
            var _authorizedUserId = requestParameters["authorizedUserId"];
            if (_authorizedUserId === null || _authorizedUserId === undefined) {
                throw new Error('Required parameter authorizedUserId was null or undefined when calling updateStatusForAuthorizedUser.');
            }
            var _updateAuthorizedUserStatus = requestParameters["updateAuthorizedUserStatus"];
            if (_updateAuthorizedUserStatus === null || _updateAuthorizedUserStatus === undefined) {
                throw new Error('Required parameter updateAuthorizedUserStatus was null or undefined when calling updateStatusForAuthorizedUser.');
            }
            return this.configuration.basePath + "/client-api/v2/users/me/authorized-users/" + encodeURIComponent(String(_authorizedUserId)) + "/status";
        };
        return AuthorizedUserService;
    }());
    AuthorizedUserService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: AuthorizedUserService, deps: [{ token: i1__namespace.HttpClient }, { token: AUTHORIZED_USER_BASE_PATH, optional: true }, { token: AuthorizedUserConfiguration, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    AuthorizedUserService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: AuthorizedUserService, providedIn: 'root' });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0__namespace, type: AuthorizedUserService, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root'
                    }]
            }], ctorParameters: function () {
            return [{ type: i1__namespace.HttpClient }, { type: undefined, decorators: [{
                            type: i0.Optional
                        }, {
                            type: i0.Inject,
                            args: [AUTHORIZED_USER_BASE_PATH]
                        }] }, { type: AuthorizedUserConfiguration, decorators: [{
                            type: i0.Optional
                        }] }];
        } });

    var AuthorizedUserServiceCreateAuthorizedUserMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v2/users/me/authorized-users",
            method: "POST",
            responses: [
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
            ]
        }]);
    var AuthorizedUserServiceDeleteAuthorizedUserMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v2/users/me/authorized-users/{authorizedUserId}",
            method: "DELETE",
            responses: [
                {
                    status: 204,
                    body: ""
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
            ]
        }]);
    var AuthorizedUserServiceGetAuthorizedUsersMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v2/users/me/authorized-users",
            method: "GET",
            responses: [
                {
                    status: 200,
                    body: [{
                            "id": "376069ae-f628-4569-8bb0-ec1fce83ebe9",
                            "said": "53251f6b-f081-4e22-a977-93efaa47e449",
                            "firstName": "John",
                            "lastName": "Doe",
                            "email": "john.doe@email.com",
                            "mobileNumber": "+44 012345 67890",
                            "username": "john.doe.145764",
                            "status": "AWAITING_REGISTRATION",
                            "permissions": [{
                                    "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4691",
                                    "level": "BASIC-001"
                                }, {
                                    "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4692",
                                    "level": "PARTIAL-002"
                                }, {
                                    "accountId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4600",
                                    "level": "FULL-003"
                                }]
                        }]
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
            ]
        }]);
    var AuthorizedUserServiceGetPermissionsByAuthorizedUserIdMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v2/users/me/authorized-users/{authorizedUserId}/permissions",
            method: "GET",
            responses: [
                {
                    status: 200,
                    body: [{
                            "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4691",
                            "level": "BASIC-001"
                        }, {
                            "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4692",
                            "level": "PARTIAL-002"
                        }, {
                            "accountId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4600",
                            "level": "FULL-003"
                        }]
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
            ]
        }]);
    var AuthorizedUserServiceUpdatePermissionsForAuthorizedUserMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v2/users/me/authorized-users/{authorizedUserId}/permissions",
            method: "PUT",
            responses: [
                {
                    status: 204,
                    body: ""
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
            ]
        }]);
    var AuthorizedUserServiceUpdateStatusForAuthorizedUserMocksProvider = i2.createMocks([{
            urlPattern: "/client-api/v2/users/me/authorized-users/{authorizedUserId}/status",
            method: "PUT",
            responses: [
                {
                    status: 204,
                    body: ""
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
            ]
        }]);
    var AuthorizedUserServiceMocksProvider = i2.createMocks([
        {
            urlPattern: "/client-api/v2/users/me/authorized-users",
            method: "POST",
            responses: [
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
            ]
        },
        {
            urlPattern: "/client-api/v2/users/me/authorized-users/{authorizedUserId}",
            method: "DELETE",
            responses: [
                {
                    status: 204,
                    body: ""
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
            ]
        },
        {
            urlPattern: "/client-api/v2/users/me/authorized-users",
            method: "GET",
            responses: [
                {
                    status: 200,
                    body: [{
                            "id": "376069ae-f628-4569-8bb0-ec1fce83ebe9",
                            "said": "53251f6b-f081-4e22-a977-93efaa47e449",
                            "firstName": "John",
                            "lastName": "Doe",
                            "email": "john.doe@email.com",
                            "mobileNumber": "+44 012345 67890",
                            "username": "john.doe.145764",
                            "status": "AWAITING_REGISTRATION",
                            "permissions": [{
                                    "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4691",
                                    "level": "BASIC-001"
                                }, {
                                    "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4692",
                                    "level": "PARTIAL-002"
                                }, {
                                    "accountId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4600",
                                    "level": "FULL-003"
                                }]
                        }]
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
            ]
        },
        {
            urlPattern: "/client-api/v2/users/me/authorized-users/{authorizedUserId}/permissions",
            method: "GET",
            responses: [
                {
                    status: 200,
                    body: [{
                            "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4691",
                            "level": "BASIC-001"
                        }, {
                            "accountId": "1cdb2224-8926-4b4d-a99f-1c9dfbbb4692",
                            "level": "PARTIAL-002"
                        }, {
                            "accountId": "3cdb2224-8926-4b4d-a99f-1c9dfbbb4600",
                            "level": "FULL-003"
                        }]
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
            ]
        },
        {
            urlPattern: "/client-api/v2/users/me/authorized-users/{authorizedUserId}/permissions",
            method: "PUT",
            responses: [
                {
                    status: 204,
                    body: ""
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
            ]
        },
        {
            urlPattern: "/client-api/v2/users/me/authorized-users/{authorizedUserId}/status",
            method: "PUT",
            responses: [
                {
                    status: 204,
                    body: ""
                },
                {
                    status: 400,
                    body: {
                        "message": "Bad Request",
                        "errors": [{
                                "message": "Value Exceeded. Must be between {min} and {max}.",
                                "key": "common.api.shoesize",
                                "context": {
                                    "max": "50",
                                    "min": "1"
                                }
                            }]
                    }
                },
                {
                    status: 401,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to invalid credentials.",
                                "key": "common.api.token",
                                "context": {
                                    "accessToken": "expired"
                                }
                            }]
                    }
                },
                {
                    status: 403,
                    body: {
                        "message": "Access to requested resource denied.",
                        "errors": [{
                                "message": "Resource access denied due to an insufficient user quota of {quota}.",
                                "key": "common.api.quota",
                                "context": {
                                    "quota": "someQuota"
                                }
                            }]
                    }
                },
                {
                    status: 404,
                    body: {
                        "message": "Resource not found.",
                        "errors": [{
                                "message": "Unable to find the resource requested resource: {resource}.",
                                "key": "common.api.resource",
                                "context": {
                                    "resource": "aResource"
                                }
                            }]
                    }
                },
            ]
        },
    ]);

    exports.AUTHORIZED_USER_BASE_PATH = AUTHORIZED_USER_BASE_PATH;
    exports.AuthorizedUserApiModule = AuthorizedUserApiModule;
    exports.AuthorizedUserConfiguration = AuthorizedUserConfiguration;
    exports.AuthorizedUserService = AuthorizedUserService;
    exports.AuthorizedUserServiceCreateAuthorizedUserMocksProvider = AuthorizedUserServiceCreateAuthorizedUserMocksProvider;
    exports.AuthorizedUserServiceDeleteAuthorizedUserMocksProvider = AuthorizedUserServiceDeleteAuthorizedUserMocksProvider;
    exports.AuthorizedUserServiceGetAuthorizedUsersMocksProvider = AuthorizedUserServiceGetAuthorizedUsersMocksProvider;
    exports.AuthorizedUserServiceGetPermissionsByAuthorizedUserIdMocksProvider = AuthorizedUserServiceGetPermissionsByAuthorizedUserIdMocksProvider;
    exports.AuthorizedUserServiceMocksProvider = AuthorizedUserServiceMocksProvider;
    exports.AuthorizedUserServiceUpdatePermissionsForAuthorizedUserMocksProvider = AuthorizedUserServiceUpdatePermissionsForAuthorizedUserMocksProvider;
    exports.AuthorizedUserServiceUpdateStatusForAuthorizedUserMocksProvider = AuthorizedUserServiceUpdateStatusForAuthorizedUserMocksProvider;
    exports.COLLECTION_FORMATS = COLLECTION_FORMATS;
    exports.CONFIG_TOKEN = CONFIG_TOKEN;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
