// @backbase/ui-ang/esm2015/_example/tab/examples/tab.example.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/ui-ang/notification";
import * as i2 from "@backbase/ui-ang/tab";
export class TabExampleComponent {
    constructor(notificationService) {
        this.notificationService = notificationService;
    }
    onSelect(index) {
        this.notificationService.showNotification({
            header: 'Tab Navigation',
            message: `Navigate to Tab #${index + 1}'`,
            modifier: 'info',
        });
    }
    onClick($event) {
        $event.stopPropagation();
        $event.preventDefault();
    }
}
TabExampleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TabExampleComponent, deps: [{ token: i1.NotificationService }], target: i0.ɵɵFactoryTarget.Component });
TabExampleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: TabExampleComponent, selector: "bb-tab-example-ui", ngImport: i0, template: "<bb-tab-group-ui\n  [initialSelection]=\"0\"\n  (click)=\"onClick($event)\"\n  (select)=\"onSelect($event)\"\n>\n  <bb-tab-ui>First</bb-tab-ui>\n  <bb-tab-ui>Second</bb-tab-ui>\n  <bb-tab-ui>Third</bb-tab-ui>\n</bb-tab-group-ui>\n", components: [{ type: i2.TabGroupComponent, selector: "bb-tab-group-ui", inputs: ["initialSelection"], outputs: ["select"] }, { type: i2.TabComponent, selector: "bb-tab-ui" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: TabExampleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-tab-example-ui',
                    templateUrl: './tab.example.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.NotificationService }]; } });
