// @backbase/cards-management-journey-ang/esm2015/lib/cards-management-journey-config.service.js
import { Injectable, Optional, Inject, InjectionToken } from '@angular/core';
import * as i0 from "@angular/core";
export const CardsManagementJourneyConfigurationToken = new InjectionToken('CardsManagementJourneyConfiguration injection token');
const configDefaults = {
    notificationTtl: 5000,
    groupByPaymentCardTypes: 'Credit,Debit',
    enableTravelNotice: true,
    deElevatedHeader: false,
};
export const getDefaultConfigValues = (config) => (Object.assign(Object.assign({}, configDefaults), (config || {})));
export class CardsManagementJourneyConfigService {
    constructor(config) {
        this._config = getDefaultConfigValues(config);
    }
    get defaults() {
        return configDefaults;
    }
    /**
     * Notification timeout timer value configuration
     */
    get notificationTtl() {
        return this._config.notificationTtl;
    }
    /**
     * Types of cards configuration value
     */
    get groupByPaymentCardTypes() {
        return this._config.groupByPaymentCardTypes;
    }
    /**
     * Dev config to enable Travel notice
     */
    get enableTravelNotice() {
        return this._config.enableTravelNotice;
    }
    /**
     * Dev config to enable de-elevated header
     */
    get deElevatedHeader() {
        return this._config.deElevatedHeader;
    }
}
/** @nocollapse */ CardsManagementJourneyConfigService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardsManagementJourneyConfigService, deps: [{ token: CardsManagementJourneyConfigurationToken, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
/** @nocollapse */ CardsManagementJourneyConfigService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardsManagementJourneyConfigService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardsManagementJourneyConfigService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [CardsManagementJourneyConfigurationToken]
                }] }]; } });
