// @backbase/cards-management-journey-ang/esm2015/lib/cards-management-journey.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { provideRoutes, RouterModule } from '@angular/router';
import { CardDetailsWidgetModule } from '@backbase/card-details-widget-ang';
import { CardsListWidgetModule } from '@backbase/cards-list-widget-ang';
import { CardsTravelNoticeModule } from '@backbase/cards-travel-notice-widget-ang';
import { BackbaseCoreModule } from '@backbase/foundation-ang/core';
import { HeaderModule } from '@backbase/ui-ang/header';
import { CardsManagementJourneyComponent } from './cards-management-journey.component';
import { CardsManagementTravelNoticeGuard } from './guard/cards-management-travel-notice.guard';
import { CardDetailViewComponent } from './views/card-detail-view.component';
import { CardsListViewComponent } from './views/cards-list-view.component';
import { CardsTravelNoticeViewComponent } from './views/cards-travel-notice-view.component';
import * as i0 from "@angular/core";
import * as i1 from "@backbase/foundation-ang/core";
export const defaultRoute = {
    path: '',
    component: CardsManagementJourneyComponent,
    children: [
        {
            path: '',
            redirectTo: 'list',
            pathMatch: 'full',
        },
        {
            path: 'list',
            component: CardsListViewComponent,
        },
        {
            path: 'details',
            component: CardDetailViewComponent,
        },
        {
            path: 'travel-notice',
            component: CardsTravelNoticeViewComponent,
            canActivate: [CardsManagementTravelNoticeGuard],
        },
    ],
};
export class CardsManagementJourneyModule {
    static forRoot(data = { route: defaultRoute }) {
        return {
            ngModule: CardsManagementJourneyModule,
            providers: [provideRoutes([data.route])],
        };
    }
}
/** @nocollapse */ CardsManagementJourneyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardsManagementJourneyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ CardsManagementJourneyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardsManagementJourneyModule, declarations: [CardsManagementJourneyComponent,
        CardsListViewComponent,
        CardDetailViewComponent,
        CardsTravelNoticeViewComponent], imports: [CommonModule,
        RouterModule, i1.BackbaseCoreModule, CardsListWidgetModule,
        CardDetailsWidgetModule,
        CardsTravelNoticeModule,
        ReactiveFormsModule,
        HeaderModule] });
/** @nocollapse */ CardsManagementJourneyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardsManagementJourneyModule, providers: [CardsManagementTravelNoticeGuard], imports: [[
            CommonModule,
            RouterModule,
            BackbaseCoreModule.withConfig({
                classMap: { CardsManagementJourneyComponent },
            }),
            CardsListWidgetModule,
            CardDetailsWidgetModule,
            CardsTravelNoticeModule,
            ReactiveFormsModule,
            HeaderModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardsManagementJourneyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        CardsManagementJourneyComponent,
                        CardsListViewComponent,
                        CardDetailViewComponent,
                        CardsTravelNoticeViewComponent,
                    ],
                    imports: [
                        CommonModule,
                        RouterModule,
                        BackbaseCoreModule.withConfig({
                            classMap: { CardsManagementJourneyComponent },
                        }),
                        CardsListWidgetModule,
                        CardDetailsWidgetModule,
                        CardsTravelNoticeModule,
                        ReactiveFormsModule,
                        HeaderModule,
                    ],
                    providers: [CardsManagementTravelNoticeGuard],
                }]
        }] });
