// @backbase/cards-management-journey-ang/esm2015/lib/cards-management-journey.component.js
import { Component } from '@angular/core';
import { CardsManagementJourneyConfigService } from './cards-management-journey-config.service';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
export class CardsManagementJourneyComponent {
}
/** @nocollapse */ CardsManagementJourneyComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardsManagementJourneyComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ CardsManagementJourneyComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: CardsManagementJourneyComponent, selector: "bb-cards-management-journey", providers: [CardsManagementJourneyConfigService], ngImport: i0, template: ` <router-outlet></router-outlet> `, isInline: true, directives: [{ type: i1.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: CardsManagementJourneyComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bb-cards-management-journey',
                    template: ` <router-outlet></router-outlet> `,
                    providers: [CardsManagementJourneyConfigService],
                }]
        }] });
